<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ABACAS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ABACAS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">ABACAS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
abacas - Algorithm Based Automatic Contiguation of Assembled Sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>abacas</b> <b>-r</b> <i>ref</i> <b>-q</b> <i>qs</i> <b>-p</b> <i>prog</i>
  [OPTIONS]
<div class="Pp"></div>
OR
<div class="Pp"></div>
<b>abacas</b> <b>-r</b> <i>ref</i> <b>-q</b> <i>psf</i> <b>-e</b>
<dl class="Bl-tag">
  <dt class="It-tag"><i>ref</i></dt>
  <dd class="It-tag">reference sequence in a single fasta file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>qs</i></dt>
  <dd class="It-tag">contigs in multi-fasta format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>rog</i></dt>
  <dd class="It-tag">MUMmer program to use: 'nucmer' or 'promer'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>psf</i></dt>
  <dd class="It-tag">pseudomolecule/ordered sequence file in fasta format</dd>
</dl>
<div class="Pp"></div>
<b>OPTIONS</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">print usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b></dt>
  <dd class="It-tag">use default nucmer/promer parameters</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">int minimum length of exact matching word (nucmer default =
      12, promer default = 4)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b></dt>
  <dd class="It-tag">print ordered contigs to file in multifasta format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b></dt>
  <dd class="It-tag">print contigs in bin to file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b></dt>
  <dd class="It-tag">print a pseudomolecule without &quot;N&quot;s</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b></dt>
  <dd class="It-tag">int mimimum percent identity [default 40]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b></dt>
  <dd class="It-tag">int mimimum contig coverage [default 40]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b></dt>
  <dd class="It-tag">int minimum contig coverage difference [default 1]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b></dt>
  <dd class="It-tag">int minimum contig length [default 1]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b></dt>
  <dd class="It-tag">run tblastx on contigs that are not mapped</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">string (file name) print uncovered regions (gaps) on
      reference to file name</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b></dt>
  <dd class="It-tag">append contigs in bin to the pseudomolecule</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b></dt>
  <dd class="It-tag">prefix output files will have this prefix</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P</b></dt>
  <dd class="It-tag">pick primer sets to close gaps</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b></dt>
  <dd class="It-tag">int number of flanking bases on either side of a gap for
      primer design (default 350)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b></dt>
  <dd class="It-tag">int Run mummer [default 1, use <b>-R</b> 0 to avoid running
      mummer]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b></dt>
  <dd class="It-tag">Escape contig ordering i.e. go to primer design</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">Reference sequence is circular
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
ABACAS is intended to rapidly contiguate (align, order, orientate), visualize
  and design primers to close gaps on shotgun assembled contigs based on a
  reference sequence.
<div class="Pp"></div>
ABACAS uses MUMmer to find alignment positions and identify syntenies of
  assembled contigs against the reference. The output is then processed to
  generate a pseudomolecule taking overlapping contigs and gaps in to account.
  ABACAS generates a comparision file that can be used to visualize ordered and
  oriented contigs in ACT. Synteny is represented by red bars where colour
  intensity decreases with lower values of percent identity between comparable
  blocks. Information on contigs such as the orientation, percent identity,
  coverage and overlap with other contigs can also be visualized by loading the
  outputted feature file on ACT.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
ABACAS IS Copyright (C) 2008-10 The Wellcome Trust Sanger Institute, Cambridge,
  UK.
<div class="Pp"></div>
This manual page was written by Andreas Tille &lt;tille@debian.org&gt;, for the
  Debian project (and may be used by others).</div>
<table class="foot">
  <tr>
    <td class="foot-date">2011-02-11</td>
    <td class="foot-os">1.3.1</td>
  </tr>
</table>
</body>
</html>
