<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:47:11 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>abyss-pe(1) User Commands abyss-pe(1)</p>

<p style="margin-top: 1em">NAME <br>
abyss-pe - assemble reads into contigs</p>

<p style="margin-top: 1em">SYNOPSIS <br>
abyss-pe [OPTION]... [PARAMETER=VALUE]...
[MAKE_TARGET]...</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Assemble the reads of the input files into contigs. The
reads may be in FASTA, FASTQ, qseq, export, SRA, SAM or BAM
format and may be compressed with gz, bz2 or xz and may be
<br>
tarred.</p>

<p style="margin-top: 1em">abyss-pe is a Makefile script.
Any options of make may also be used with abyss-pe.</p>

<p style="margin-top: 1em">Parameters of abyss-pe <br>
name, JOB_NAME <br>
The name of this assembly. The resulting scaffolds will be
stored in ${name}-scaffolds.fa.</p>

<p style="margin-top: 1em">in input files. Use this
variable when assembling data from a single library.</p>

<p style="margin-top: 1em">lib a quoted list of
whitespace-separated paired-end library names. Use this
varible when assembling data from multiple paired-end
libraries. For each library name in lib, <br>
the user must define a variable on the command line with the
same name, which indicates the read files for that library.
See EXAMPLES below for a concrete example of <br>
usage.</p>

<p style="margin-top: 1em">pe list of paired-end libraries
that will be used only for merging unitigs into contigs and
will not contribute toward the consensus sequence.</p>

<p style="margin-top: 1em">mp list of mate-pair libraries
that will be used for scaffolding. Mate-pair libraries do
not contribute toward the consensus sequence.</p>

<p style="margin-top: 1em">long list of long sequence
libraries that will be used for rescaffolding. long sequence
libraries do not contribute toward the consensus
sequence.</p>

<p style="margin-top: 1em">se files containing single-end
reads</p>

<p style="margin-top: 1em">a maximum number of branches of
a bubble [2]</p>

<p style="margin-top: 1em">b maximum length of a bubble
(bp) [&quot;&quot;] <br>
abyss-pe has two bubble popping stages. The default limits
are 3*k bp for ABYSS and 10000 bp for PopBubbles.</p>

<p style="margin-top: 1em">c minimum mean k-mer coverage of
a unitig [sqrt(median)]</p>

<p style="margin-top: 1em">d allowable error of a distance
estimate (bp) [6]</p>

<p style="margin-top: 1em">e minimum erosion k-mer coverage
[round(sqrt(median))]</p>

<p style="margin-top: 1em">E minimum erosion k-mer coverage
per strand [1 if sqrt(median) &gt; 2 else 0]</p>

<p style="margin-top: 1em">j number of threads [2]</p>

<p style="margin-top: 1em">k size of a k-mer (when K is not
set) or the span of a k-mer pair (when K is set)</p>

<p style="margin-top: 1em">K size of a single k-mer in a
k-mer pair (bp)</p>

<p style="margin-top: 1em">l minimum alignment length of a
read (bp) [40]</p>

<p style="margin-top: 1em">m minimum overlap of two unitigs
(bp) [30]</p>

<p style="margin-top: 1em">n minimum number of pairs
required for building contigs [10]</p>

<p style="margin-top: 1em">N minimum number of pairs
required for building scaffolds [n]</p>

<p style="margin-top: 1em">p minimum sequence identity of a
bubble [0.9]</p>

<p style="margin-top: 1em">q minimum base quality when
trimming [3] <br>
Trim bases from the ends of reads whose quality is less
q.</p>

<p style="margin-top: 1em">Q minimum base quality [0] <br>
Mask all bases of reads whose quality is less than Q as
&lsquo;N&rsquo;.</p>

<p style="margin-top: 1em">s minimum unitig size required
for building contigs (bp) [1000] <br>
The seed length should be at least twice the value of k. If
more sequence is assembled than the expected genome size,
try increasing s.</p>

<p style="margin-top: 1em">S minimum contig size required
for building scaffolds (bp) [1000-10000]</p>

<p style="margin-top: 1em">SS SS=--SS to assemble in
strand-specific mode <br>
Requires that all libraries are strand-specific RNA-Seq
libraries. Assumes that the first read in a read pair is
reveresed WRT the transcripts sequenced.</p>

<p style="margin-top: 1em">t maximum length of blunt
contigs to trim [k]</p>

<p style="margin-top: 1em">v v=-v to enable verbose
logging</p>

<p style="margin-top: 1em">np, NSLOTS <br>
the number of processes of an MPI assembly</p>

<p style="margin-top: 1em">mpirun the path to mpirun</p>

<p style="margin-top: 1em">aligner <br>
The program to use to align the reads to the contigs [map].
<br>
Permitted values are: map, kaligner, bwa, bwasw, bowtie,
bowtie2, dida. See the DIDA section below for further info
on the dida option.</p>

<p style="margin-top: 1em">cs convert colour-space contigs
to nucleotide contigs following assembly</p>

<p style="margin-top: 1em">Options of make <br>
-n, --dry-run <br>
Print the commands that would be executed, but do not
execute them.</p>

<p style="margin-top: 1em">Make targets for abyss-pe <br>
default <br>
Equivalent to &lsquo;scaffolds scaffolds-dot
stats&rsquo;.</p>

<p style="margin-top: 1em">unitigs <br>
Assemble unitigs.</p>

<p style="margin-top: 1em">unitigs-dot <br>
Output the unitig overlap graph.</p>

<p style="margin-top: 1em">pe-sam Map paired-end reads to
the unitigs and output a SAM file. SAM file will only
contain reads mapping to different contigs, and the read ID,
sequence and quality strings <br>
will be replaced with &rsquo;*&rsquo; characters.</p>

<p style="margin-top: 1em">pe-bam Map paired-end reads to
the unitigs and output a BAM file. BAM file will only
contain reads mapping to different contigs, and the read ID,
sequence and quality strings <br>
will be replaced with &rsquo;*&rsquo; characters.</p>

<p style="margin-top: 1em">pe-index <br>
Generate an index of the unitigs used by abyss-map.</p>

<p style="margin-top: 1em">contigs <br>
Assemble contigs.</p>

<p style="margin-top: 1em">contigs-dot <br>
Output the contig overlap graph.</p>

<p style="margin-top: 1em">mp-sam Map mate-pair reads to
the contigs and output a SAM file. SAM file will only
contain reads mapping to different contigs, and the read ID,
sequence and quality strings will <br>
be replaced with &rsquo;*&rsquo; characters.</p>

<p style="margin-top: 1em">mp-bam Map mate-pair reads to
the contigs and output a BAM file. BAM file will only
contain reads mapping to different contigs, and the read ID,
sequence and quality strings will <br>
be replaced with &rsquo;*&rsquo; characters.</p>

<p style="margin-top: 1em">mp-index <br>
Generate an index of the contigs used by abyss-map.</p>

<p style="margin-top: 1em">scaffolds <br>
Assemble scaffolds.</p>

<p style="margin-top: 1em">scaffolds-dot <br>
Output the scaffold overlap graph.</p>

<p style="margin-top: 1em">scaftigs <br>
Break scaffolds and generate AGP file.</p>

<p style="margin-top: 1em">long-scaffs <br>
Rescaffold using RNA-Seq assembled contigs.</p>

<p style="margin-top: 1em">long-scaffs-dot <br>
Output the RNA scaffold overlap graph.</p>

<p style="margin-top: 1em">stats Display assembly
contiguity statistics.</p>

<p style="margin-top: 1em">clean Remove intermediate
files.</p>

<p style="margin-top: 1em">version <br>
Display the version of abyss-pe.</p>

<p style="margin-top: 1em">versions <br>
Display the versions of all programs used by abyss-pe.</p>

<p style="margin-top: 1em">help Display a helpful
message.</p>

<p style="margin-top: 1em">DIDA <br>
ABySS supports the use of DIDA (Distributed Indexing
Dispatched Alignment), an MPI-based alignment framework for
computing sequence alignments across multiple machines. To
use <br>
DIDA with ABySS, first download and install DIDA from
http://www.bcgsc.ca/platform/bioinfo/software/dida, then
specify &lsquo;dida&lsquo; as the value of the aligner
parameter to abyss-pe.</p>

<p style="margin-top: 1em">DIDA-related abyss-pe parameters
<br>
DIDA_MPIRUN <br>
The &lsquo;mpirun&lsquo; command used to run DIDA jobs.</p>

<p style="margin-top: 1em">DIDA_RUN_OPTIONS <br>
Runtime options such as number of threads per MPI rank and
values for environment variables (e.g. PATH,
LD_LIBRARY_PATH). Run &lsquo;abyss-dida --help&lsquo; for a
list of available <br>
options.</p>

<p style="margin-top: 1em">DIDA_OPTIONS <br>
Options that are passed directly to the DIDA binary. For
example, this can be used to control the minimum alignment
length threshold. Run &lsquo;dida-wrapper --help&lsquo; for
a list <br>
of available options.</p>

<p style="margin-top: 1em">MPI COMPATIBILITY <br>
Due to its use of multi-threading, DIDA has known
deadlocking issues with OpenMPI. Using the MPICH MPI library
is strongly recommmended when running assemblies with DIDA.
Test&acirc; <br>
ing was done with MPICH 3.1.3, compiled with
--enable-threads=funneled.</p>

<p style="margin-top: 1em">EXAMPLE <br>
The recommended runtime configuration for DIDA is 1 MPI rank
per machine and 1 thread per CPU core. For example, to run
an assembly across 3 cluster nodes with 12 cores each,
do:</p>

<p style="margin-top: 1em">abyss-pe k=64 name=ecoli
in=&rsquo;reads1.fa reads2.fa&rsquo; aligner=dida
DIDA_RUN_OPTIONS=&rsquo;-j12&rsquo;
DIDA_MPIRUN=&rsquo;mpirun -np 3 -ppn 1 -bind-to
board&rsquo;</p>

<p style="margin-top: 1em">This example uses the MPICH
command line options for &lsquo;mpirun&lsquo;. Here,
&lsquo;-np 3&lsquo; indicates the number of MPI ranks,
&lsquo;-ppn 1&lsquo; indicates the number of MPI ranks per
&quot;node&quot;, and <br>
&lsquo;-bind-to board&lsquo; defines a &quot;node&quot; to
be a motherboard (i.e. a full machine).</p>

<p style="margin-top: 1em">ENVIRONMENT VARIABLES <br>
Any parameter that may be specified on the command line may
also be specified in an environment variable.</p>

<p style="margin-top: 1em">PATH must contain the directory
where the ABySS executables are installed. Use
&lsquo;abyss-pe versions&lsquo; to check that PATH is
configured correctly.</p>

<p style="margin-top: 1em">TMPDIR specifies a directory to
use for temporary files</p>

<p style="margin-top: 1em">Scheduler integration <br>
ABySS integrates well with cluster job schedulers, such as:
<br>
* SGE (Sun Grid Engine) <br>
* Portable Batch System (PBS) <br>
* Load Sharing Facility (LSF) <br>
* IBM LoadLeveler</p>

<p style="margin-top: 1em">The SGE environment variables
JOB_NAME, SGE_TASK_ID and NSLOTS may be used to specify the
parameters name, k and np, respectively, and similarly for
other schedulers.</p>

<p style="margin-top: 1em">EXAMPLES <br>
One paired-end library <br>
abyss-pe k=64 name=ecoli in=&rsquo;reads1.fa
reads2.fa&rsquo;</p>

<p style="margin-top: 1em">Multiple paired-end libraries
<br>
abyss-pe k=64 name=ecoli lib=&rsquo;lib1 lib2&rsquo;
lib1=&rsquo;lib1_1.fa lib1_2.fa&rsquo; lib2=&rsquo;lib2_1.fa
lib2_2.fa&rsquo; se=&rsquo;se1.fa se2.fa&rsquo;</p>

<p style="margin-top: 1em">Paired-end and mate-pair
libraries <br>
abyss-pe k=64 name=ecoli lib=&rsquo;pe1 pe2&rsquo;
mp=&rsquo;mp1 mp2&rsquo; pe1=&rsquo;pe1_1.fa pe1_2.fa&rsquo;
pe2=&rsquo;pe2_1.fa pe2_2.fa&rsquo; mp1=&rsquo;mp1_1.fa
mp1_2.fa&rsquo; mp2=&rsquo;mp2_1.fa mp2_2.fa&rsquo;
se=&rsquo;se1.fa se2.fa&rsquo;</p>

<p style="margin-top: 1em">Including RNA-Seq assemblies
<br>
abyss-pe k=64 name=ecoli lib=pe1 mp=mp1 long=long1
pe1=&rsquo;pe1_1.fa pe1_2.fa&rsquo; mp1=&rsquo;mp1_1.fa
mp1_2.fa&rsquo; long1=long1.fa</p>

<p style="margin-top: 1em">MPI <br>
abyss-pe np=8 k=64 name=ecoli in=&rsquo;reads1.fa
reads2.fa&rsquo;</p>

<p style="margin-top: 1em">SGE <br>
qsub -N ecoli -t 64 -pe openmpi 8 abyss-pe n=10
in=&rsquo;reads1.fa reads2.fa&rsquo;</p>

<p style="margin-top: 1em">SEE ALSO <br>
make(1), ABYSS(1)</p>

<p style="margin-top: 1em">AUTHOR <br>
Written by Shaun Jackman.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;abyss-users@googlegroups.com&gt;.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright 2015 Canada&rsquo;s Michael Smith Genome Sciences
Centre</p>

<p style="margin-top: 1em">abyss-pe (ABySS) 2.0.2 2015-May
abyss-pe(1)</p>
<hr>
</body>
</html>
