<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ABYSS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ABYSS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">ABYSS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
ABYSS - assemble short reads into contigs
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>ABYSS</b> [ <i>OPTION</i>]... <i>FILE</i>...
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Assemble all input files, FILE, which may be in FASTA, FASTQ, qseq, export, SRA,
  SAM or BAM format and may be compressed with gz, bz2 or xz and may be tarred.
<div style="height: 1.00em;">&#x00A0;</div>
Users wishing to run a full assembly with ABySS are recommended to use abyss-pe
  as their entry point rather than the ABYSS program. abyss-pe is a Makefile
  that coordinates many of the ABySS tools (including ABYSS) in order to run the
  full ABySS assembly pipeline. abyss-pe is capable of doing many things that
  ABYSS is not, such as leveraging the distance information provided by paired
  end reads and mate pair reads.
<dl class="Bl-tag">
  <dt class="It-tag"><b>--chastity</b></dt>
  <dd class="It-tag">discard unchaste reads [default]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-chastity</b></dt>
  <dd class="It-tag">do not discard unchaste reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trim-masked</b></dt>
  <dd class="It-tag">trim masked bases from the ends of reads [default]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-trim-masked</b></dt>
  <dd class="It-tag">do not trim masked bases from the ends of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b>, <b>--trim-quality</b>=<i>THRESHOLD</i></dt>
  <dd class="It-tag">trim bases from the ends of reads whose quality is less
      than the threshold</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b>, <b>--mask-quality</b>=<i>THRESHOLD</i></dt>
  <dd class="It-tag">mask all bases in reads whose quality is less than the
      threshold</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--standard-quality</b></dt>
  <dd class="It-tag">zero quality is `!' (33)
    <div>&#x00A0;</div>
    default for FASTQ and SAM files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--illumina-quality</b></dt>
  <dd class="It-tag">zero quality is `@' (64)
    <div>&#x00A0;</div>
    default for qseq and export files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>, <b>--out</b>=<i>FILE</i></dt>
  <dd class="It-tag">write the contigs to FILE</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b>, <b>--kmer</b>=<i>KMER_SIZE</i></dt>
  <dd class="It-tag">k-mer size</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>, <b>--trim-length</b>=<i>TRIM_LENGTH</i></dt>
  <dd class="It-tag">maximum length of dangling edges to trim</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>, <b>--coverage</b>=<i>COVERAGE</i></dt>
  <dd class="It-tag">remove contigs with mean k-mer coverage less than this
      threshold</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b>, <b>--bubbles</b>=<i>N</i></dt>
  <dd class="It-tag">pop bubbles shorter than N bp (default: 3*k).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b>0, <b>--no-bubbles</b></dt>
  <dd class="It-tag">do not pop bubbles</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>, <b>--erode</b>=<i>COVERAGE</i></dt>
  <dd class="It-tag">erode bases at the ends of blunt contigs with coverage less
      than this threshold</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b>, <b>--erode-strand</b>=<i>COVERAGE</i></dt>
  <dd class="It-tag">erode bases at the ends of blunt contigs with coverage less
      than this threshold on either strand</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--coverage-hist</b>=<i>FILE</i></dt>
  <dd class="It-tag">record the k-mer coverage histogram in FILE</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b>, <b>--graph</b>=<i>FILE</i></dt>
  <dd class="It-tag">generate a graph in dot format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>, <b>--snp</b>=<i>FILE</i></dt>
  <dd class="It-tag">record popped bubbles in FILE</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--verbose</b></dt>
  <dd class="It-tag">display verbose output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">display this help and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">output version information and exit</dd>
</dl>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Written by Jared Simpson and Shaun Jackman.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;abyss-users@bcgsc.ca&gt;.
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright 2015 Canada's Michael Smith Genome Sciences Centre</div>
<table class="foot">
  <tr>
    <td class="foot-date">2015-May</td>
    <td class="foot-os">ABYSS (ABySS) 2.0.2</td>
  </tr>
</table>
</body>
</html>
