<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:47:10 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ABYSS(1) User Commands ABYSS(1)</p>

<p style="margin-top: 1em">NAME <br>
ABYSS - assemble short reads into contigs</p>

<p style="margin-top: 1em">SYNOPSIS <br>
ABYSS [OPTION]... FILE...</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Assemble all input files, FILE, which may be in FASTA,
FASTQ, qseq, export, SRA, SAM or BAM format and may be
compressed with gz, bz2 or xz and may be tarred.</p>

<p style="margin-top: 1em">Users wishing to run a full
assembly with ABySS are recommended to use abyss-pe as their
entry point rather than the ABYSS program. abyss-pe is a
Makefile that coordinates many <br>
of the ABySS tools (including ABYSS) in order to run the
full ABySS assembly pipeline. abyss-pe is capable of doing
many things that ABYSS is not, such as leveraging the
dis&acirc; <br>
tance information provided by paired end reads and mate pair
reads.</p>

<p style="margin-top: 1em">--chastity <br>
discard unchaste reads [default]</p>

<p style="margin-top: 1em">--no-chastity <br>
do not discard unchaste reads</p>

<p style="margin-top: 1em">--trim-masked <br>
trim masked bases from the ends of reads [default]</p>

<p style="margin-top: 1em">--no-trim-masked <br>
do not trim masked bases from the ends of reads</p>

<p style="margin-top: 1em">-q, --trim-quality=THRESHOLD
<br>
trim bases from the ends of reads whose quality is less than
the threshold</p>

<p style="margin-top: 1em">-Q, --mask-quality=THRESHOLD
<br>
mask all bases in reads whose quality is less than the
threshold</p>

<p style="margin-top: 1em">--standard-quality <br>
zero quality is &lsquo;!&rsquo; (33) <br>
default for FASTQ and SAM files</p>

<p style="margin-top: 1em">--illumina-quality <br>
zero quality is &lsquo;@&rsquo; (64) <br>
default for qseq and export files</p>

<p style="margin-top: 1em">-o, --out=FILE <br>
write the contigs to FILE</p>

<p style="margin-top: 1em">-k, --kmer=KMER_SIZE <br>
k-mer size</p>

<p style="margin-top: 1em">-t, --trim-length=TRIM_LENGTH
<br>
maximum length of dangling edges to trim</p>

<p style="margin-top: 1em">-c, --coverage=COVERAGE <br>
remove contigs with mean k-mer coverage less than this
threshold</p>

<p style="margin-top: 1em">-b, --bubbles=N <br>
pop bubbles shorter than N bp (default: 3*k).</p>

<p style="margin-top: 1em">-b0, --no-bubbles <br>
do not pop bubbles</p>

<p style="margin-top: 1em">-e, --erode=COVERAGE <br>
erode bases at the ends of blunt contigs with coverage less
than this threshold</p>

<p style="margin-top: 1em">-E, --erode-strand=COVERAGE <br>
erode bases at the ends of blunt contigs with coverage less
than this threshold on either strand</p>

<p style="margin-top: 1em">--coverage-hist=FILE <br>
record the k-mer coverage histogram in FILE</p>

<p style="margin-top: 1em">-g, --graph=FILE <br>
generate a graph in dot format</p>

<p style="margin-top: 1em">-s, --snp=FILE <br>
record popped bubbles in FILE</p>

<p style="margin-top: 1em">-v, --verbose <br>
display verbose output</p>

<p style="margin-top: 1em">--help display this help and
exit</p>

<p style="margin-top: 1em">--version <br>
output version information and exit</p>

<p style="margin-top: 1em">AUTHOR <br>
Written by Jared Simpson and Shaun Jackman.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;abyss-users@bcgsc.ca&gt;.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright 2015 Canada&rsquo;s Michael Smith Genome Sciences
Centre</p>

<p style="margin-top: 1em">ABYSS (ABySS) 2.0.2 2015-May
ABYSS(1)</p>
<hr>
</body>
</html>
