<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>abyss-pe(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">abyss-pe(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">abyss-pe(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
abyss-pe - assemble reads into contigs
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>abyss-pe</b> [ <i>OPTION</i>]... [<i>PARAMETER</i>=<i>VALUE</i>]...
  [<i>MAKE_TARGET</i>]...
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Assemble the reads of the input files into contigs. The reads may be in FASTA,
  FASTQ, qseq, export, SRA, SAM or BAM format and may be compressed with gz, bz2
  or xz and may be tarred.
<div style="height: 1.00em;">&#x00A0;</div>
abyss-pe is a Makefile script. Any options of make may also be used with
  abyss-pe.
<div style="height: 1.00em;">&#x00A0;</div>
<h2 class="Ss" title="Ss" id="Parameters_of_abyss-pe"><a class="selflink" href="#Parameters_of_abyss-pe">Parameters
  of abyss-pe</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>name</b>, <b>JOB_NAME</b></dt>
  <dd class="It-tag">The name of this assembly. The resulting scaffolds will be
      stored in ${name}-scaffolds.fa.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>in</b></dt>
  <dd class="It-tag">input files. Use this variable when assembling data from a
      single library.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>lib</b></dt>
  <dd class="It-tag">a quoted list of whitespace-separated paired-end library
      names. Use this varible when assembling data from multiple paired-end
      libraries. For each library name in lib, the user must define a variable
      on the command line with the same name, which indicates the read files for
      that library. See <b>EXAMPLES</b> below for a concrete example of
    usage.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pe</b></dt>
  <dd class="It-tag">list of paired-end libraries that will be used only for
      merging unitigs into contigs and will not contribute toward the consensus
      sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mp</b></dt>
  <dd class="It-tag">list of mate-pair libraries that will be used for
      scaffolding. Mate-pair libraries do not contribute toward the consensus
      sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>long</b></dt>
  <dd class="It-tag">list of long sequence libraries that will be used for
      rescaffolding. long sequence libraries do not contribute toward the
      consensus sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>se</b></dt>
  <dd class="It-tag">files containing single-end reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>a</b></dt>
  <dd class="It-tag">maximum number of branches of a bubble [2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>b</b></dt>
  <dd class="It-tag">maximum length of a bubble (bp) [&quot;&quot;]
    <div>&#x00A0;</div>
    abyss-pe has two bubble popping stages. The default limits are 3*k bp for
      ABYSS and 10000 bp for PopBubbles.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>c</b></dt>
  <dd class="It-tag">minimum mean k-mer coverage of a unitig [sqrt(median)]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>d</b></dt>
  <dd class="It-tag">allowable error of a distance estimate (bp) [6]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>e</b></dt>
  <dd class="It-tag">minimum erosion k-mer coverage [round(sqrt(median))]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>E</b></dt>
  <dd class="It-tag">minimum erosion k-mer coverage per strand [1 if
      sqrt(median) &gt; 2 else 0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>j</b></dt>
  <dd class="It-tag">number of threads [2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>k</b></dt>
  <dd class="It-tag">size of a k-mer (when K is not set) or the span of a k-mer
      pair (when K is set)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>K</b></dt>
  <dd class="It-tag">size of a single k-mer in a k-mer pair (bp)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>l</b></dt>
  <dd class="It-tag">minimum alignment length of a read (bp) [40]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>m</b></dt>
  <dd class="It-tag">minimum overlap of two unitigs (bp) [30]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>n</b></dt>
  <dd class="It-tag">minimum number of pairs required for building contigs
    [10]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>N</b></dt>
  <dd class="It-tag">minimum number of pairs required for building scaffolds
    [n]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>p</b></dt>
  <dd class="It-tag">minimum sequence identity of a bubble [0.9]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>q</b></dt>
  <dd class="It-tag">minimum base quality when trimming [3]
    <div>&#x00A0;</div>
    Trim bases from the ends of reads whose quality is less q.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Q</b></dt>
  <dd class="It-tag">minimum base quality [0]
    <div>&#x00A0;</div>
    Mask all bases of reads whose quality is less than Q as `N'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>s</b></dt>
  <dd class="It-tag">minimum unitig size required for building contigs (bp)
      [1000]
    <div>&#x00A0;</div>
    The seed length should be at least twice the value of k. If more sequence is
      assembled than the expected genome size, try increasing s.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>S</b></dt>
  <dd class="It-tag">minimum contig size required for building scaffolds (bp)
      [1000-10000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>SS</b></dt>
  <dd class="It-tag">SS=--SS to assemble in strand-specific mode
    <div>&#x00A0;</div>
    Requires that all libraries are strand-specific RNA-Seq libraries. Assumes
      that the first read in a read pair is reveresed WRT the transcripts
      sequenced.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>t</b></dt>
  <dd class="It-tag">maximum length of blunt contigs to trim [k]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>v</b></dt>
  <dd class="It-tag">v=-v to enable verbose logging</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>np</b>, <b>NSLOTS</b></dt>
  <dd class="It-tag">the number of processes of an MPI assembly</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mpirun</b></dt>
  <dd class="It-tag">the path to mpirun</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>aligner</b></dt>
  <dd class="It-tag">The program to use to align the reads to the contigs [map].
    <div>&#x00A0;</div>
    Permitted values are: map, kaligner, bwa, bwasw, bowtie, bowtie2, dida. See
      the <b>DIDA</b> section below for further info on the dida option.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cs</b></dt>
  <dd class="It-tag">convert colour-space contigs to nucleotide contigs
      following assembly</dd>
</dl>
<h2 class="Ss" title="Ss" id="Options_of_make"><a class="selflink" href="#Options_of_make">Options
  of make</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b>, <b>--dry-run</b></dt>
  <dd class="It-tag">Print the commands that would be executed, but do not
      execute them.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Make_targets_for_abyss-pe"><a class="selflink" href="#Make_targets_for_abyss-pe">Make
  targets for abyss-pe</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>default</b></dt>
  <dd class="It-tag">Equivalent to `scaffolds scaffolds-dot stats'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>unitigs</b></dt>
  <dd class="It-tag">Assemble unitigs.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>unitigs-dot</b></dt>
  <dd class="It-tag">Output the unitig overlap graph.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pe-sam</b></dt>
  <dd class="It-tag">Map paired-end reads to the unitigs and output a SAM file.
      SAM file will only contain reads mapping to different contigs, and the
      read ID, sequence and quality strings will be replaced with '*'
      characters.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pe-bam</b></dt>
  <dd class="It-tag">Map paired-end reads to the unitigs and output a BAM file.
      BAM file will only contain reads mapping to different contigs, and the
      read ID, sequence and quality strings will be replaced with '*'
      characters.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pe-index</b></dt>
  <dd class="It-tag">Generate an index of the unitigs used by abyss-map.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>contigs</b></dt>
  <dd class="It-tag">Assemble contigs.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>contigs-dot</b></dt>
  <dd class="It-tag">Output the contig overlap graph.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mp-sam</b></dt>
  <dd class="It-tag">Map mate-pair reads to the contigs and output a SAM file.
      SAM file will only contain reads mapping to different contigs, and the
      read ID, sequence and quality strings will be replaced with '*'
      characters.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mp-bam</b></dt>
  <dd class="It-tag">Map mate-pair reads to the contigs and output a BAM file.
      BAM file will only contain reads mapping to different contigs, and the
      read ID, sequence and quality strings will be replaced with '*'
      characters.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mp-index</b></dt>
  <dd class="It-tag">Generate an index of the contigs used by abyss-map.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>scaffolds</b></dt>
  <dd class="It-tag">Assemble scaffolds.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>scaffolds-dot</b></dt>
  <dd class="It-tag">Output the scaffold overlap graph.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>scaftigs</b></dt>
  <dd class="It-tag">Break scaffolds and generate AGP file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>long-scaffs</b></dt>
  <dd class="It-tag">Rescaffold using RNA-Seq assembled contigs.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>long-scaffs-dot</b></dt>
  <dd class="It-tag">Output the RNA scaffold overlap graph.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>stats</b></dt>
  <dd class="It-tag">Display assembly contiguity statistics.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>clean</b></dt>
  <dd class="It-tag">Remove intermediate files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>version</b></dt>
  <dd class="It-tag">Display the version of abyss-pe.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>versions</b></dt>
  <dd class="It-tag">Display the versions of all programs used by abyss-pe.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>help</b></dt>
  <dd class="It-tag">Display a helpful message.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="DIDA"><a class="selflink" href="#DIDA">DIDA</a></h1>
ABySS supports the use of DIDA (Distributed Indexing Dispatched Alignment), an
  MPI-based alignment framework for computing sequence alignments across
  multiple machines. To use DIDA with ABySS, first download and install DIDA
  from http://www.bcgsc.ca/platform/bioinfo/software/dida, then specify `dida`
  as the value of the <b>aligner</b> parameter to <b>abyss-pe</b>.
<div style="height: 1.00em;">&#x00A0;</div>
<h2 class="Ss" title="Ss" id="DIDA-related_abyss-pe_parameters"><a class="selflink" href="#DIDA-related_abyss-pe_parameters">DIDA-related
  abyss-pe parameters</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>DIDA_MPIRUN</b></dt>
  <dd class="It-tag">The `mpirun` command used to run DIDA jobs.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>DIDA_RUN_OPTIONS</b></dt>
  <dd class="It-tag">Runtime options such as number of threads per MPI rank and
      values for environment variables (e.g. PATH, LD_LIBRARY_PATH). Run
      `abyss-dida --help` for a list of available options.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>DIDA_OPTIONS</b></dt>
  <dd class="It-tag">Options that are passed directly to the DIDA binary. For
      example, this can be used to control the minimum alignment length
      threshold. Run `dida-wrapper --help` for a list of available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="MPI_COMPATIBILITY"><a class="selflink" href="#MPI_COMPATIBILITY">MPI
  COMPATIBILITY</a></h2>
Due to its use of multi-threading, DIDA has known deadlocking issues with
  OpenMPI. Using the MPICH MPI library is strongly recommmended when running
  assemblies with DIDA. Testing was done with MPICH 3.1.3, compiled with
  --enable-threads=funneled.
<div style="height: 1.00em;">&#x00A0;</div>
<h2 class="Ss" title="Ss" id="EXAMPLE"><a class="selflink" href="#EXAMPLE">EXAMPLE</a></h2>
The recommended runtime configuration for DIDA is 1 MPI rank per machine and 1
  thread per CPU core. For example, to run an assembly across 3 cluster nodes
  with 12 cores each, do:
<div style="height: 1.00em;">&#x00A0;</div>
	abyss-pe k=64 name=ecoli in='reads1.fa reads2.fa' aligner=dida
  DIDA_RUN_OPTIONS='-j12' DIDA_MPIRUN='mpirun -np 3 -ppn 1 -bind-to board'
<div style="height: 1.00em;">&#x00A0;</div>
This example uses the MPICH command line options for `mpirun`. Here, `-np 3`
  indicates the number of MPI ranks, `-ppn 1` indicates the number of MPI ranks
  per &quot;node&quot;, and `-bind-to board` defines a &quot;node&quot; to be a
  motherboard (i.e. a full machine).
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="ENVIRONMENT_VARIABLES"><a class="selflink" href="#ENVIRONMENT_VARIABLES">ENVIRONMENT
  VARIABLES</a></h1>
Any parameter that may be specified on the command line may also be specified in
  an environment variable.
<dl class="Bl-tag">
  <dt class="It-tag"><b>PATH</b></dt>
  <dd class="It-tag">must contain the directory where the ABySS executables are
      installed. Use `abyss-pe versions` to check that PATH is configured
      correctly.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>TMPDIR</b></dt>
  <dd class="It-tag">specifies a directory to use for temporary files</dd>
</dl>
<h2 class="Ss" title="Ss" id="Scheduler_integration"><a class="selflink" href="#Scheduler_integration">Scheduler
  integration</a></h2>
ABySS integrates well with cluster job schedulers, such as:
<br/>
 * SGE (Sun Grid Engine)
<br/>
 * Portable Batch System (PBS)
<br/>
 * Load Sharing Facility (LSF)
<br/>
 * IBM LoadLeveler
<div style="height: 1.00em;">&#x00A0;</div>
The SGE environment variables JOB_NAME, SGE_TASK_ID and NSLOTS may be used to
  specify the parameters name, k and np, respectively, and similarly for other
  schedulers.
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<h2 class="Ss" title="Ss" id="One_paired-end_library"><a class="selflink" href="#One_paired-end_library">One
  paired-end library</a></h2>
<br/>
 abyss-pe k=64 name=ecoli in='reads1.fa reads2.fa'
<h2 class="Ss" title="Ss" id="Multiple_paired-end_libraries"><a class="selflink" href="#Multiple_paired-end_libraries">Multiple
  paired-end libraries</a></h2>
<br/>
 abyss-pe k=64 name=ecoli lib='lib1 lib2' \
<div>&#x00A0;</div>
	lib1='lib1_1.fa lib1_2.fa' lib2='lib2_1.fa lib2_2.fa' \
<div>&#x00A0;</div>
	se='se1.fa se2.fa'
<h2 class="Ss" title="Ss" id="Paired-end_and_mate-pair_libraries"><a class="selflink" href="#Paired-end_and_mate-pair_libraries">Paired-end
  and mate-pair libraries</a></h2>
<br/>
 abyss-pe k=64 name=ecoli lib='pe1 pe2' mp='mp1 mp2' \
<div>&#x00A0;</div>
	pe1='pe1_1.fa pe1_2.fa' pe2='pe2_1.fa pe2_2.fa' \
<div>&#x00A0;</div>
	mp1='mp1_1.fa mp1_2.fa' mp2='mp2_1.fa mp2_2.fa' \
<div>&#x00A0;</div>
	se='se1.fa se2.fa'
<h2 class="Ss" title="Ss" id="Including_RNA-Seq_assemblies"><a class="selflink" href="#Including_RNA-Seq_assemblies">Including
  RNA-Seq assemblies</a></h2>
<br/>
 abyss-pe k=64 name=ecoli lib=pe1 mp=mp1 long=long1 \
<div>&#x00A0;</div>
	pe1='pe1_1.fa pe1_2.fa' mp1='mp1_1.fa mp1_2.fa' \
<div>&#x00A0;</div>
	long1=long1.fa
<h2 class="Ss" title="Ss" id="MPI"><a class="selflink" href="#MPI">MPI</a></h2>
<br/>
 abyss-pe np=8 k=64 name=ecoli in='reads1.fa reads2.fa'
<h2 class="Ss" title="Ss" id="SGE"><a class="selflink" href="#SGE">SGE</a></h2>
<br/>
 qsub -N ecoli -t 64 -pe openmpi 8 \
<div>&#x00A0;</div>
	abyss-pe n=10 in='reads1.fa reads2.fa'
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
make(1), ABYSS(1)
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Written by Shaun Jackman.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;abyss-users@googlegroups.com&gt;.
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright 2015 Canada's Michael Smith Genome Sciences Centre</div>
<table class="foot">
  <tr>
    <td class="foot-date">2015-May</td>
    <td class="foot-os">abyss-pe (ABySS) 2.0.2</td>
  </tr>
</table>
</body>
</html>
