<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ADAPTERREMOVAL(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ADAPTERREMOVAL(1)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">ADAPTERREMOVAL(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
AdapterRemoval - Remove adapters from sequences in either single end or paired
  end experiments
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>AdapterRemoval</b> --file1 filename [--file2 filename] [--interleaved]
  [--interleaved-input] [--interleaved-output] [--combined-output] [--basename
  filename] [--identify-adapters] [--trimns] [--maxns max] [--trimqualities]
  [--minquality minimum] [--collapse] [--version] [--mm mismatchrate]
  [--minlength len] [--minalignmentlength len] [--qualitybase base]
  [--qualitybase-output base] [--shift num] [--adapter1 sequence] [--adapter2
  sequence] [--adapter-list filename] [--barcode-list filename] [--barcode-mm
  num] [--barcode-mm-r1 num] [--barcode-mm-r2 num] [--demultiplex-only]
  [--output1 filename] [--output2 filename] [--singleton filename]
  [--outputcollapsed filename] [--outputcollapsedtruncated filename]
  [--discarded filename] [--settings filename] [--seed seed] [--gzip]
  [--gzip-level level] [--threads num] [--version] [--help]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>AdapterRemoval</b> reads either one FASTQ file (single ended mode) or two
  FASTQ files (paired ended mode). It removes the residual adapter sequence from
  the reads and optionally trims Ns from the reads, and low qualities bases
  using the quality string, and collapses overlapping paired ended mates into
  one read. Reads are discarded if the remaining genomic part is too short, or
  if the read contains more than an (user specified) amount of amigious
  nucleotides ('N'). These operations may be combined with simultaneous
  demultiplexing.
<div class="Pp"></div>
Alternatively, <b>AdapterRemoval</b> may attempt to reconstruct a consensus
  adapter sequences from paired-ended data, in order to allow the identification
  of the adapter sequences originally used, and thereby ensure proper trimming
  of these reads.
<div class="Pp"></div>
The reads and adapters are transformed to upper case for comparison. It is
  assumed that the letter 'N' is used for an unknown nucleotide, but in case the
  program encounters a '.' in the sequence, they will be treated as (and
  translated into) Ns. The program tries to check for invalid input and / or
  nonsensical combinations of parameters but please report strange behaviour,
  bugs and such to MSchubert@snm.ku.dk
<div class="Pp"></div>
If you use this program, please cite the paper: Stinus Lindgreen (2012):
  AdapterRemoval: easy cleaning of next-generation sequencing reads. BMC Res
  Notes, 5:337. doi: 10.1186/1756-0500-5-337.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--file1</b> <i>filename</i></dt>
  <dd class="It-tag">Read FASTQ reads from file <i>filename</i>. This contains
      either the single ended (SE) reads or, if paired ended, the mate 1 reads.
      If running in paired end mode, both file1 and file2 must be set. The file
      may optionally be gzip or bzip2 compressed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--file2</b> <i>filename</i></dt>
  <dd class="It-tag">Read FASTQ file <i>filename</i> containing mate 2 reads for
      a paired end run. If specified, --file1 must also be set. The file may
      optionally be gzip or bzip2 compressed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--interleaved</b></dt>
  <dd class="It-tag">Enables <i>--interleaved-input</i> and
      <i>--interleaved-output</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--interleaved-input</b></dt>
  <dd class="It-tag">If set, input is expected to be a single FASTQ file
      specified using <i>--file1</i>, in which pairs of paired-end reads are
      listed one after each other (read1/1, read1/2, read2/1, read2/2,
    etc.).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--interleaved-ouput</b></dt>
  <dd class="It-tag">If set, and AdapterRemoval is processing paired-end reads,
      retained pairs of reads are written to a single FASTQ file, one pair after
      each other (read1/1, read1/2, read2/1, read2/2, etc.). By default, this
      file is named <i>basename.paired.truncated</i>, but this may be changed
      using the <i>--output1</i> option.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--combined-output</b></dt>
  <dd class="It-tag">If set, all reads are written to the same file(s),
      specified by --output1 and --output2. Each read is further marked by
      either a &quot;PASSED&quot; or a &quot;FAILED&quot; flag, and any read
      that has been FAILED (including the mate for collapsed reads) are replaced
      with a single 'N' with Phred score 0. This option can be combined with
      --interleaved / --interleaved-output to write all reads to a single output
      file specified with --output1.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--basename</b> <i>filename</i></dt>
  <dd class="It-tag">Determines the default filename for output files, unless
      overridden using the specific output file settings. For single-ended mode,
      the following filenames are used: <i>basename.truncated</i>,
      <i>basename.discarded</i>, and <i>basename.settings</i>. In paired end
      mode, the following filenames are used: <i>basename.pair1.truncated</i>,
      <i>basename.pair2.truncated</i>, <i>basename.singleton.truncated</i>,
      <i>basename.discarded</i>, and <i>basename.settings</i>. If collapsing of
      reads is enabled for paired ended mode, the following filenames are also
      used: <i>basename.collapsed</i>, and <i>basename.collapsed.truncated</i>.
      The default basename is <i>your_output</i>. If gzip compression is
      enabled, the extension &quot;.gz&quot; is added to all files but the
      <i>filename.settings</i> file, while the extension &quot;.bz2&quot; is
      used if bzip2 compression is enabled.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--identify-adapters</b></dt>
  <dd class="It-tag">For paired ended reads only. In this mode, AdapterRemoval
      will attempt to reconstruct the adapter sequences used for a set of paired
      ended reads, by locating fully overlapping read-pairs, and generating a
      consensus sequence from the bases identified as adapter sequence. The
      minimum overlap is controlled by <i>minalignmentlength</i>. The values
      passed to the --adapter1 and --adapter2 command-line options are used for
      visual comparison with the consensus sequence, but otherwise not used in
      the consensus building.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trimns</b></dt>
  <dd class="It-tag">Remove stretches of Ns from the output reads in both the 5'
      and 3' end. If quality trimming is also enabled, stretches of mixed
      low-quality bases and/or Ns are trimmed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxns</b> <i>max</i></dt>
  <dd class="It-tag">If a read has more than <i>max</i> Ns after trimming, it is
      discarded (default is not to use).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trimqualities</b></dt>
  <dd class="It-tag">Remove consecutive stretches of low quality bases
      (threshold set by <i>minquality</i>) from both the 5' and 3' end of the
      reads. All bases with <i>minquality</i> or lower are trimmed. If trimming
      of Ns is also enabled, stretches of mixed low-quality bases and/or Ns are
      trimmed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minquality</b> <i>minimum</i></dt>
  <dd class="It-tag">Set the threshold for trimming low quality bases. Default
      is 2. The minimum can be set with or without the Phred quality base.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--collapse</b></dt>
  <dd class="It-tag">In paired-end mode, if the two mates overlap, collapse the
      two reads into one read by merging the two and recalculating the quality
      scores. In single-end mode, this instead attempts to identify templates
      for which the entire sequence is available. In both cases, complete
      &quot;collapsed&quot; reads are written with a 'M_' name prefix, and
      &quot;collapsed&quot; reads which are trimmed due to quality settings are
      written with a 'MT_' name prefix. The overlap needs to be at least
      <i>minalignmentlength</i> nucleotides, with a maximum number of mismatches
      determined by <i>mm</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mm</b> <i>mismatchrate</i></dt>
  <dd class="It-tag">The allowed fraction of mismatches allowed in the aligned
      region. If 0 &lt; <i>mismatchrate</i> &lt; 1, the rate is used directly.
      If <i>mismatchrate</i> &gt; 1, the rate is set to 1/<i>mismatchrate</i>.
      The default setting is 3, corresponding to a maximum mismatch rate of
    1/3.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minlength</b> <i>len</i></dt>
  <dd class="It-tag">The minimum length required after trimming and adapter
      removal. Reads shorter than <i>len</i> are discarded. Default is 15
      nucleotides.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minalignmentlength</b> <i>len</i></dt>
  <dd class="It-tag">The minimum overlap between mate 1 and mate 2 before the
      reads are collapsed into one, when collapsing paired end reads, or when
      attempting to identify complete template sequences in single-end mode.
      Default is 11 nucleotides.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--qualitybase</b> <i>base</i></dt>
  <dd class="It-tag">The base of the quality score - either '64' for Phred+Phred
      (i.e., Illumina 1.3+ and 1.5+) or '33' for Phred+33 (Illumina 1.8+). In
      addition, the value 'solexa' may be used to specify reads with Solexa
      encoded scores. Default is 33.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--qualitybase-output</b> <i>base</i></dt>
  <dd class="It-tag">The base of the quality score for reads written by
      AdapterRemoval - either '64' for Phred+Phred (i.e., Illumina 1.3+ and
      1.5+) or '33' for Phred+33 (Illumina 1.8+). In addition, the value
      'solexa' may be used to specify reads with Solexa encoded scores. However,
      note that quality scores are represented using PHRED scores internally,
      and conversion to and from Solexa scores therefore result in a loss of
      information. The default corresponds to the value given for
    --qualitybase.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--shift</b> <i>num</i></dt>
  <dd class="It-tag">To allow for missing bases in the 5' end of the read, the
      program can let the alignment slip <i>num</i> bases in the 5' end. This
      corresponds to starting the alignment maximum <i>num</i> nucleotides in
      read2 (for paired end) or the adapter (for single end). The default shift
      valule is 2.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--adapter1</b> <i>sequence</i></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--adapter2</b> <i>sequence</i></dt>
  <dd class="It-tag">Specify the adapter sequences that you wish to trim. The
      Adapter #2 sequence is only used when trimming paired-ended data.
    <div style="height: 1.00em;">&#x00A0;</div>
    The Adapter #1 and Adapter #2 sequences are expected to be found in the mate
      1 and the mate 2 reads respectively, while ignoring any difference in case
      and treating Ns as wildcards. The default sequences are
    <div style="height: 1.00em;">&#x00A0;</div>
    Adapter #1: AGATCGGAAGAGCACACGTCTGAACTCCAGTCACNNNNNNATCTCGTATGCCGTCTTCTGCTTG
    <div style="height: 1.00em;">&#x00A0;</div>
    Adapter #2: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
    <div style="height: 1.00em;">&#x00A0;</div>
    Assuming these were the adapters used to generate our data, we should
      therefore see these in the FASTQ files:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
  $ grep -i &quot;AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC......ATCTCGTATGCCGTCTTCTGCTTG&quot; file1.fq
  B&lt;AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGAATCTCGTATGCCGTCTTCTGCTTG&gt;AAAAAAAAACAAGAAT
  CTGGAGTTCB&lt;AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGAATCTCGTATGCCGTCTTCTGCTTG&gt;AAAAAAA
  GGB&lt;AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGAATCTCGTATGCCGTCTTCTGCTTG&gt;CAAATTGAAAACAC
  ...
  $ grep -i &quot;AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT&quot; file2.fq
  CB&lt;AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT&gt;CAAAAAAAGAAAAACATCTTG
  GAACTCCAGB&lt;AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT&gt;CAAAAAAAATAGA
  GAACTB&lt;AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT&gt;CAAAAACATAAGACCTA
  ...
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    Note that --adapter1 and --adapter2 replaces the --pcr[12] options of
      <b>AdapterRemoval</b> v1.x, for which the --pcr2 sequence was expected to
      be reverse complemented compared --adaper2. Using the --pcr[12] options is
      not recommended!</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--adapter-list</b> <i>filename</i></dt>
  <dd class="It-tag">Read one or more PCR sequences from a table. The first two
      columns (separated by whitespace) of each line in the file are expected to
      correspond to values passed to --adapter1 and --adapter2. In single ended
      mode, only column one is required. Lines starting with '#' are ignored.
      When multiple PCR sequences or sequence pairs are specified,
      AdapterRemoval will try each adapter (pair) listed in the table, and
      select the best aligning adapters for each read processed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--barcode-list</b> <i>filename</i></dt>
  <dd class="It-tag">Read a table of one or two fixed-length barcodes and
      perform demultiplexing of single or double indexed reads. The table is
      expected to contain 2 or 3 columns, the first of which represent the name
      of a given sample, and the second and third of which represent the mate 1
      and (optionally) the mate 2 barcode sequence:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
    $ cat barcodes.txt
    sample_1 ATGCGGA TGAATCT
    sample_2 ATGGATT ATAGTGA
    sample_7 CAAAACT TCGCTGC
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    Results are written to ${basename}.${sample_name}.*, using the default names
      for other output files. A setting file with statistics is written for each
      sample at ${basename}.${sample_name}.settings, as is a setting file
      containing the demultiplexing statistics, at ${basename}.settings.
    <div style="height: 1.00em;">&#x00A0;</div>
    When demultiplexing is used, the barcode identified for a given read is
      automatically added to the adapter sequence, in order to ensure that
      overlapping reads are correctly trimmed. The .settings file represents
      this by showing the reverse complemented) barcode sequence added to the
      --adapter1 and --adapter2 sequences, followed by an underscore (shown here
      for barcodes pair ATGCGGA / TGAATCT):
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
    [Adapter sequences]
    Adapter1[0]: AGATTCA_AGATCGGAAGAGCACACGTCTGAACTCCAGTCACNNNNNNATCTCGTATGCCGTCTTCTGCTTG
    Adapter2[0]: TCCGCAT_AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    Note that the sequence added to each adapter is the reverse complement of
      the barcode sequence of the other mate, as this sequence is expected to be
      found immediately before the adapter sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--barcode-mm</b> <i>num</i></dt>
  <dd class="It-tag">The maximum number of mismatches allowed for barcodes, when
      counting mismatches in <i>both</i> the mate 1 and mate 2 barcodes. In
      conjunction with the --barcode-mm-r1 and --barcode-mm-r2, this allows
      fine-grained control over the barcode comparisons. If not set, this value
      is set to the sum of --barcode-mm-r1 and --barcode-mm-r2.
    <div style="height: 1.00em;">&#x00A0;</div>
    For example, to allow one mismatch in either the mate 1 or the mate 2
      barcode, one might specify --barcode-mm 1; to allow a mismatch in the mate
      1 and / or the mate 2 barcode, one might specify --barcode-mm 2
      --barcode-mm-r1 1 --barcode-mm-r2 1, and so on.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--barcode-mm-r1</b> <i>num</i></dt>
  <dd class="It-tag">The maximum number of mismatches allowed in the mate 1
      barcode; if not set, this number is equal to the value of --barcode-mm.
      This number cannot exceed the value specified for --barcode-mm.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--barcode-mm-r2</b> <i>num</i></dt>
  <dd class="It-tag">The maximum number of mismatches allowed in the mate 1
      barcode; if not set, this number is equal to the value of --barcode-mm.
      This number cannot exceed the value specified for --barcode-mm.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--demultiplex-only</b> <i>num</i></dt>
  <dd class="It-tag">Only carry out demultiplexing, using the list of barcodes
      supplied using --barcode-list. Note that trimming and filtering options do
      not apply to this mode of operation.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--output1</b> <i>file</i></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--output2</b> <i>file</i></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--singleton</b> <i>file</i></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outputcollapsed</b> <i>file</i></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outputcollapsedtruncated</b> <i>file</i></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--discarded</b> <i>file</i></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--settings</b> <i>file</i></dt>
  <dd class="It-tag">Instead of using the default behaviour where the program
      automatically generates the files needed, you can specify where each type
      of output is directed. This can be files, pipes etc. thus making it
      possible to easily zip the output on the fly. Default files are still
      generated if nothing else is specified.
    <div style="height: 1.00em;">&#x00A0;</div>
    The types of output in single end mode are:
    <div style="height: 1.00em;">&#x00A0;</div>
     <i>output1</i> contains the trimmed reads.
    <div style="height: 1.00em;">&#x00A0;</div>
    The types of output in paired end mode are:
    <div style="height: 1.00em;">&#x00A0;</div>
     <i>output1</i> contains trimmed mate1 reads.
    <div style="height: 1.00em;">&#x00A0;</div>
     <i>output2</i> contains trimmed mate2 reads.
    <div style="height: 1.00em;">&#x00A0;</div>
     <i>singleton</i> contains all reads where the other mate in a pair is
      discarded.
    <div style="height: 1.00em;">&#x00A0;</div>
     <i>outputcollapsed</i> Contains pairs that overlap and are collapsed into a
      single read (if <i>--collapse</i> is used). The reads are renamed with an
      @M_ prefix.
    <div style="height: 1.00em;">&#x00A0;</div>
     <i>outputcollapsedtruncated</i> Contains pairs that overlap and are
      collapsed into a single read (if <i>--collapse</i> is used) and have
      further been trimmed due to Ns and/or low quality nucleotides in the 5' or
      3' end. The reads are renamed with an @MT_ prefix.
    <div style="height: 1.00em;">&#x00A0;</div>
    The types of output in both single end and paired end mode are:
    <div style="height: 1.00em;">&#x00A0;</div>
     <i>discarded</i> contains all reads that are discarded by the program.
    <div style="height: 1.00em;">&#x00A0;</div>
     <i>settings</i> contains information on the parameters used in the run as
      well as overall statistics on the reads after trimming such as average
      length.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b> <i>seed</i></dt>
  <dd class="It-tag">When collaping reads at positions where the two reads
      differ, and the quality of the bases are identical, AdapterRemoval will
      select a random base. This option specifies the seed used for the random
      number generator used by AdapterRemoval. This value is also written to the
      settings file. Note that setting the seed is not reliable in multithreaded
      mode, since the order of operations is non-deterministic.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gzip</b></dt>
  <dd class="It-tag">If set, all FASTQ files written by AdapterRemoval will be
      gzip compressed using the compression level specified using
      <i>--gzip-level</i>. The extension &quot;.gz&quot; is added to files for
      which no filename was given on the commandline.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gzip-level</b></dt>
  <dd class="It-tag">Determines the compression level used when gzip'ing FASTQ
      files. Must be a value in the range 0 to 9, with 0 disabling compression
      and 9 being the best compression. Defaults to 6.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bzip2</b></dt>
  <dd class="It-tag">If set, all FASTQ files written by AdapterRemoval will be
      bzip2 compressed using the compression level specified using
      <i>--bzip2-level</i>. The extension &quot;.bz2&quot; is added to files for
      which no filename was given on the commandline.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bzip2-level</b></dt>
  <dd class="It-tag">Determines the compression level used when bzip2'ing FASTQ
      files. Must be a value in the range 1 to 9, with 9 being the best
      compression. Defaults to 9.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--threads</b></dt>
  <dd class="It-tag">Maximum number of threads to use for current run; note that
      file IO is single-threaded, regardless of the number of threads
    specified.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">Output the version of the program.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">Output the summary of available command-line options,
      including default values and/or values specified on the command-line.</dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLE:_Single_end_experiment"><a class="selflink" href="#EXAMPLE:_Single_end_experiment">EXAMPLE:
  Single end experiment</a></h1>
The following command removes adapters from the file <i>reads_1.fq</i> trims
  both Ns and low quality bases from the reads, and gzip compresses the
  resulting files. The --basename option is used to specify the prefix for
  output files.
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --file1 reads_1.fq --basename output_single --trimns --trimqualities --gzip
</pre>
<div class="Pp"></div>
Since --gzip and --basename is specified, the trimmed FASTQ reads are written to
  <i>output_single.truncated.gz</i>, the dicarded FASTQ reads are written to
  <i>output_single.discarded.gz</i>, and settings and summary statistics are
  written to <i>output_single.settings</i>.
<div class="Pp"></div>
Note that by default, AdapterRemoval does not require a minimum number of bases
  overlapping with the adapter sequence, before reads are trimmed. This may
  result in an excess of very short (1 - 3 bp) 3' fragments being falsely
  identified as adapter sequences, and trimmed. This behavior may be changed
  using the --minadapteroverlap option, which allows the specification of a
  minimum number of bases (excluding Ns) that must be aligned to carry trimming.
  For example, use --minadapteroverlap 3 to require an overlap of at least 3 bp.
<h1 class="Sh" title="Sh" id="EXAMPLE:_Paired_end_experiment."><a class="selflink" href="#EXAMPLE:_Paired_end_experiment.">EXAMPLE:
  Paired end experiment.</a></h1>
The following command removes adapters from a paired-end reads, where the mate 1
  and mate 2 reads are kept in files <i>reads_1.fq</i> and <i>reads_2.fq</i>,
  respectively. The reads are trimmed for both Ns and low quality bases, and
  overlapping reads (at least 11 nucleotides, per default) are merged
  (collapsed):
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --file1 reads_1.fq --file2 reads_2.fq --basename output_paired --trimns --trimqualities --collapse
</pre>
<div class="Pp"></div>
This command generates the files <i>output_paired.pair1.truncated</i> and
  <i>output_paired.pair2.truncated</i>, which contain trimmed pairs of reads
  which were not collapsed, <i>output_paired.singleton.truncated</i> containing
  reads where one mate was discarded, <i>output_paired.collapsed</i> containing
  merged reads, and <i>output_paired.collapsed.truncated</i> containing merged
  reads that have been trimmed due to the --trimns or --trimqualities options.
  Finally, the <i>output_paired.discarded</i> and <i>output_paired.settings</i>
  files correspond to those of the single-end run.
<h1 class="Sh" title="Sh" id="EXAMPLE:_Interleaved_FASTQ_reads."><a class="selflink" href="#EXAMPLE:_Interleaved_FASTQ_reads.">EXAMPLE:
  Interleaved FASTQ reads.</a></h1>
AdapterRemoval is able to read and write paired-end reads stored in a single,
  so-called interleaved FASTQ file (one pair at a time, first mate 1, then mate
  2). This is accomplished by specifying the location of the file using
  <i>--file1</i> and *also* setting the <i>--interleaved</i> command-line
  option:
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --interleaved --file1 interleaved.fq --basename output_interleaved
</pre>
<div class="Pp"></div>
Other than taking just a single input file, this mode operates almost exactly
  like paired end trimming (as described above); the mode differs only in that
  paired reads are not written to a 'pair1' and a 'pair2' file, but instead
  these are instead written to a single, interleaved file, named 'paired'. The
  location of this file is controlled using the <i>--output1</i> option.
  Enabling either reading or writing of interleaved FASTQ files, both not both,
  can be accomplished by specifying the either of the <i>--interleaved-input</i>
  and <i>--interleaved-output</i> options, both of which are enabled by the
  <i>--interleaved</i> option.
<h1 class="Sh" title="Sh" id="EXAMPLE:_Different_quality_score_encodings."><a class="selflink" href="#EXAMPLE:_Different_quality_score_encodings.">EXAMPLE:
  Different quality score encodings.</a></h1>
By default, AdapterRemoval expects the quality scores in FASTQ reads to be
  Phred+33 encoded, meaning that the error probabilities are encoded as
  (char)('!' - 10 * log10(p)). Most data will be encoded using Phred+33, but
  Phred+64 and 'Solexa' encoded quality scores are also supported. These are
  selected by specifying the <i>--qualitybase</i> command-line option
  (specifying either '33', '64', or 'solexa')::
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --qualitybase 64 --file1 reads_q64.fq --basename phred_64_encoded
</pre>
<div class="Pp"></div>
By default, reads are written using the *same* encoding as the input. If a
  different encoding is desired, this may be accomplished using the
  <i>--qualitybase-output</i> option:
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --qualitybase 64 --qualitybase-output 33 --file1 reads_q64.fq --basename phred_33_encoded
</pre>
<div class="Pp"></div>
Note furthermore that AdapterRemoval by default only expects quality scores in
  the range 0 - 41 (or -5 to 41 in the case of Solexa encoded scores). If input
  data using a different maximum quality score is to be processed, or if the
  desired maximum quality score of collapsed reads is greater than 41, then this
  limit may be increased using the <i>--qualitymax</i> option:
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --qualitymax 50 --file1 reads_1.fq --file2 reads_2.fq --collapsed --basename collapsed_q50
</pre>
<div class="Pp"></div>
For a detailed overview of Phred encoding schemes currently and previously in
  use, see e.g. the Wikipedia article on the subject:
  https://en.wikipedia.org/wiki/FASTQ_format#Encoding
<h1 class="Sh" title="Sh" id="EXAMPLE:_Paired_end_reads_containing_multiple,_distinct_adapter_pairs."><a class="selflink" href="#EXAMPLE:_Paired_end_reads_containing_multiple,_distinct_adapter_pairs.">EXAMPLE:
  Paired end reads containing multiple, distinct adapter pairs.</a></h1>
It is possible to trim data that contains multiple adapter pairs, by providing a
  one or two-column table containing possible adapter combinations (for
  single-end and paired-end trimming, respectively; see e.g.
  <i>examples/adapters.txt</i>):
<div class="Pp"></div>
<pre>
    $ cat adapters.txt
    AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCACCTAATCTCGTATGCCGTCTTCTGCTTG    AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
    AAACTTGCTCTGTGCCCGCTCCGTATGTCACAACAGTGCGTGTATCACCTCAATGCAGGACTCA    GATCGGGAGTAATTTGGAGGCAGTAGTTCGTCGAAACTCGGAGCGTCTTTAGCAGGAG
    CTAATTTGCCGTAGCGACGTACTTCAGCCTCCAGGAATTGGACCCTTACGCACACGCATTCATG    TACCGTGAAAGGTGCGCTTAGTGGCATATGCGTTAAGAGCTAGGTAACGGTCTGGAGG
    GTTCATACGACGACGACCAATGGCACACTTATCCGGTACTTGCGTTTCAATGCGCATGCCCCAT    TAAGAAACTCGGAGTTTGGCCTGCGAGGTAGCTTGGGTGTTATGAAGAACGGCATGCG
    CCATGCCCCGAAGATTCCTATACCCTTAAGGTCGCAATTGTTCGAGTAAGCTGTACGCGCCCAT    GTTGCATTGACCCGAAGGGCTCGATGTTTAGGGAGGTCAGAAGTTGAGCGGGTTCAAA
</pre>
<div class="Pp"></div>
This table is then specified using the <i>--adapter-list</i> option:
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --file1 reads_1.fq --file2 reads_2.fq --basename output_multi --trimns --trimqualities --collapse --adapter-list adapters.txt
</pre>
<div class="Pp"></div>
The resulting .summary file contains an overview of how frequently each adapter
  (pair) was used.
<div class="Pp"></div>
Note that in the case of paired-end adapters, AdapterRemoval considers only the
  combinations of adapters specified in the table, one combination per row. For
  single-end trimming, only the first column of the table file is required, and
  the list may therefore take the form of a file containing one sequence per
  line.
<h1 class="Sh" title="Sh" id="EXAMPLE:_Identifying_adapter_sequences_from_paired-ended_reads"><a class="selflink" href="#EXAMPLE:_Identifying_adapter_sequences_from_paired-ended_reads">EXAMPLE:
  Identifying adapter sequences from paired-ended reads</a></h1>
If we did not know the adapter sequences for paired-end reads, AdapterRemoval
  may be used to generate a consensus adapter sequence based on fragments
  identified as belonging to the adapters through pairwise alignments of the
  reads, provided that the data set contains only a single adpater sequence (not
  counting differences in index sequences).
<div class="Pp"></div>
In the following example, the identified adapters corresponds to the default
  adapter sequences with a poly-A tail resulting from sequencing past the end of
  the insert + templates. It is not necessary to specify this tail when using
  the <i>--adapter1</i> or <i>--adapter2</i> command-line options. The
  characters shown under each of the consensus sequences represented the
  phred-encoded fraction of bases identical to the consensus base, with adapter
  1 containing the index CACCTA:
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --identify-adapters --file1 reads_1.fq --file2 reads_2.fq
    Attemping to identify adapter sequences ...
    Processed a total of 1,000 reads in 0.0s; 129,000 reads per second on average ...
       Found 394 overlapping pairs ...
       Of which 119 contained adapter sequence(s) ...
    Printing adapter sequences, including poly-A tails:
      --adapter1:  AGATCGGAAGAGCACACGTCTGAACTCCAGTCACNNNNNNATCTCGTATGCCGTCTTCTGCTTG
                   ||||||||||||||||||||||||||||||||||******||||||||||||||||||||||||
       Consensus:  AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCACCTAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAAAAAAAAAAAAA
         Quality:  55200522544444/4411330333330222222/1.1.1.1111100-00000///..+....--*-)),,+++++++**(('%%%$
        Top 5 most common 9-bp 5'-kmers:
                1: AGATCGGAA = 96.00% (96)
                2: AGATGGGAA =  1.00% (1)
                3: AGCTCGGAA =  1.00% (1)
                4: AGAGCGAAA =  1.00% (1)
                5: AGATCGGGA =  1.00% (1)
      --adapter2:  AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
                   ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
       Consensus:  AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
         Quality:  525555555144141441430333303.2/22-2/-1..11111110--00000///..+....--*-),,,+++++++**(%'%%%$
        Top 5 most common 9-bp 5'-kmers:
                1: AGATCGGAA = 100.00% (100)
</pre>
<div class="Pp"></div>
No files are generated from running the adapter identification step.
<div class="Pp"></div>
The consensus sequences inferred are compared to those specified using the
  <i>--adapter1</i> and <i>--adapter2</i> command-line options, or with the
  default values for these if no values have been given (as in this case). Pipes
  (|) indicate matches between the provided sequences and the consensus
  sequence, and &quot;*&quot; indicate the presence of unspecified bases (Ns).
<h1 class="Sh" title="Sh" id="EXAMPLE:_Demultiplexing_of_paired_end_reads"><a class="selflink" href="#EXAMPLE:_Demultiplexing_of_paired_end_reads">EXAMPLE:
  Demultiplexing of paired end reads</a></h1>
As of version 2.1, AdapterRemoval supports simultanious demultiplexing and
  adapter trimming; demultiplexing is carried out using a simple comparison
  between the specified barcode sequences and the first N bases of the reads,
  corresponding to the length of the barcodes. Reads identified as containing a
  specific barcode or pair of barcodes are then trimmed using adapter sequences
  including these barcodes.
<div class="Pp"></div>
Demultiplexing is enabled by creating a table of barcodes, the first column of
  which species the sample name (using characters [a-zA-Z0-9_]) and the second
  and (optional) third columns specifies the mate 1 and mate 2 barcode
  sequences.
<div class="Pp"></div>
For example, a table of barcodes from a double-indexed run might be as follows
  (see <i>examples/barcodes.txt</i>):
<div class="Pp"></div>
<pre>
    $ cat barcodes.txt
    sample_1 ATGCGGA TGAATCT
    sample_2 ATGGATT ATAGTGA
    sample_7 CAAAACT TCGCTGC
</pre>
<div class="Pp"></div>
In the case of single-read reads, only the first two columns are required.
  AdapterRemoval is invoked with the <i>--barcode-list</i> option, specifying
  the path to this table:
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --file1 demux_1.fq --file2 demux_2.fq --basename output_dumux --barcode-list barcodes.txt
</pre>
<div class="Pp"></div>
This generates a set of output files for each sample specified in the barcode
  table, using the basename ( <i>--basename</i>) as the prefix, followed by a
  dot and the sample name, followed by a dot and the default name for a given
  file type. For example, the output files for sample_2 would be
<div class="Pp"></div>
<pre>
    output_demux.sample_2.discarded
    output_demux.sample_2.pair1.truncated
    output_demux.sample_2.pair2.truncated
    output_demux.sample_2.settings
    output_demux.sample_2.singleton.truncated
</pre>
<div class="Pp"></div>
The settings files generated for each sample summarizes the reads for that
  sample only; in addition, a basename.settings file is generated which
  summarizes the number and proportion of reads identified as belonging to each
  sample.
<div class="Pp"></div>
The maximum number of mismatches allowed when comparing barocdes is controlled
  using the options <i>--barcode-mmI</i>, <i>--barcode-mm-r1</i>, and
  <i>--barcode-mm-r2</i>, which specify the maximum number of mismatches total,
  and the maximum number of mismatches for the mate 1 and mate 2 barcodes
  respectively. Thus, if mm_1(i) and mm_2(i) represents the number of mismatches
  observed for barcode-pair i for a given pair of reads, these options require
  that
<div class="Pp"></div>
<pre>
   1. mm_1(i) &lt;= --barcode-mm-r1
   2. mm_2(i) &lt;= --barcode-mm-r2
   3. mm_1(i) + mm_2(i) &lt;= --barcode-mm
</pre>
<div class="Pp"></div>
As of version 2.2, AdapterRemoval can furthermore be used to demultiplex reads
  without carrying out other forms of read trimming. This is accomplished by
  specifying the <i>--demultiplex-only</i> option:
<div class="Pp"></div>
<pre>
    $ AdapterRemoval --file1 demux_1.fq --file2 demux_2.fq --basename output_only_demux --barcode-list barcodes.txt --demultiplex-only
</pre>
<div class="Pp"></div>
Trimming and filtering related options to not apply to this mode (&quot;TRIMMING
  SETTINGS&quot; when viewing 'AdapterRemoval --help'), but compression (
  <i>--gzip</i>, <i>--bzip2</i>), multi-threading (<i>--threads</i>),
  interleaving ( <i>--interleaved</i>, etc.) and other such options may be used
  in conjunction with --demultiplex-only.
<h1 class="Sh" title="Sh" id="EXIT_STATUS"><a class="selflink" href="#EXIT_STATUS">EXIT
  STATUS</a></h1>
0 if everything worked as planned, a non-zero value otherwise.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to Mikkel Schubert &lt;MSchubert@snm.ku.dk&gt;.
<div class="Pp"></div>
Your bugreport should always include:
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The output of <b>AdapterRemoval --version</b>. If you are
      not running the latest released version you should specify why you believe
      the problem is not fixed in that version.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">A complete example that others can run that shows the
      problem.</dd>
</dl>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Copyright (C) 2011 Stinus Lindgreen &lt;stinus@binf.ku.dk&gt;.
<div class="Pp"></div>
Parts of the manual was written by Ole Tange &lt;tange@binf.ku.dk&gt;.
<div class="Pp"></div>
Parts of the manual was written by Mikkel Schubert &lt;MSchubert@snm.ku.dk&gt;.
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2011 Stinus Lindgreen &lt;stinus@binf.ku.dk&gt;.
<div class="Pp"></div>
Copyright (C) 2014 Mikkel Schubert &lt;MSchubert@snm.ku.dk&gt;.
<div class="Pp"></div>
This program is free software; you can redistribute it and/or modify it under
  the terms of the GNU General Public License as published by the Free Software
  Foundation; either version 3 of the License, or at your option any later
  version.
<div class="Pp"></div>
This program is distributed in the hope that it will be useful, but WITHOUT ANY
  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
  A PARTICULAR PURPOSE. See the GNU General Public License for more details.
<div class="Pp"></div>
You should have received a copy of the GNU General Public License along with
  this program. If not, see &lt;http://www.gnu.org/licenses/&gt;.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-12-04</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
