<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:48:20 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ALIEN_HUNTER(1) User Commands ALIEN_HUNTER(1)</p>

<p style="margin-top: 1em">NAME <br>
alien_hunter - Interpolated Variable Order Motifs for
identification of horizontally acquired DNA</p>

<p style="margin-top: 1em">SYNOPSIS <br>
alien_hunter &lt;sequence.file&gt; &lt;output.file&gt;
[Options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Alien_hunter is an application for the prediction of
putative Horizontal Gene Transfer (HGT) events with the
implementation of Interpolated Variable Order Motifs
(IVOMs). An IVOM <br>
approach exploits compositional biases using variable order
motif distributions and captures more reliably the local
composition of a sequence compared to fixed-order methods.
<br>
Optionally the predictions can be parsed into a 2-state 2nd
order Hidden Markov Model (HMM), in a change-point detection
framework, to optimize the localization of the boundaries
<br>
of the predicted regions. The predictions (embl format) can
be automatically loaded into Artemis genome viewer freely
available at: http://www.sanger.ac.uk/Software/Artemis/.</p>

<p style="margin-top: 1em">INPUT: raw genomic sequence
PREDICTION: HGT regions based on Interpolated Variable Order
Motifs (IVOMs) arguments:</p>

<p style="margin-top: 1em">&lt;sequence.file&gt; <br>
raw genomic sequence</p>

<p style="margin-top: 1em">&lt;output.file&gt; <br>
filename for output data</p>

<p style="margin-top: 1em">Optional arguments:</p>

<p style="margin-top: 1em">-a to load the prediction file
into Artemis</p>

<p style="margin-top: 1em">-c optimize predicted boundaries
with a change-point detection 2 state 2nd order HMM</p>

<p style="margin-top: 1em">Output files:</p>

<p style="margin-top: 1em">&lt;output.file&gt; <br>
predictions (tab file) in embl format</p>

<p style="margin-top: 1em">&lt;output.file&gt;.opt <br>
optimized (HMM) predictions (tab file) in embl format</p>

<p style="margin-top: 1em">&lt;output.file&gt;.plot <br>
predictions in Artemis User Plot format to be loaded
manually using Graph -&gt; Add User Plot...</p>

<p style="margin-top: 1em">&lt;output.file&gt;.opt.plot
<br>
optimized (HMM) predictions in Artemis User Plot format to
be loaded manually using Graph -&gt; Add User Plot...</p>

<p style="margin-top: 1em">&lt;output.file&gt;.sco <br>
the scores over all the sliding windows - for score
distribution check</p>

<p style="margin-top: 1em">Note: Predictions that overlap
with rRNA operon are mentioned in the note qualifier</p>

<p style="margin-top: 1em">SEE ALSO <br>
The manuscript describing the alien_hunter algorithm is
available from Bioinformatics: Interpolated variable order
motifs for identification of horizontally acquired DNA:
revis&acirc; <br>
iting the Salmonella pathogenicity islands. Vernikos GS,
Parkhill J Bioinformatics. 2006;. PMID: 16837528</p>

<p style="margin-top: 1em">AUTHOR <br>
This manual page was written by Andreas Tille
&lt;tille@debian.com&gt; for the Debian(TM) system (but may
be used by others). This man page is released under the same
conditions as the <br>
software, that is GPL.</p>

<p style="margin-top: 1em">alien_hunter 1.7 October 2009
ALIEN_HUNTER(1)</p>
<hr>
</body>
</html>
