<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ALIEN_HUNTER(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ALIEN_HUNTER(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">ALIEN_HUNTER(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
alien_hunter - Interpolated Variable Order Motifs for identification of
  horizontally acquired DNA
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>alien_hunter <i>&lt;sequence.file&gt;</i> <i>&lt;output.file&gt;</i>
  [<i>Options</i>]</b>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Alien_hunter is an application for the prediction of putative Horizontal Gene
  Transfer (HGT) events with the implementation of Interpolated Variable Order
  Motifs (IVOMs). An IVOM approach exploits compositional biases using variable
  order motif distributions and captures more reliably the local composition of
  a sequence compared to fixed-order methods. Optionally the predictions can be
  parsed into a 2-state 2nd order Hidden Markov Model (HMM), in a change-point
  detection framework, to optimize the localization of the boundaries of the
  predicted regions. The predictions (embl format) can be automatically loaded
  into Artemis genome viewer freely available at:
  http://www.sanger.ac.uk/Software/Artemis/.
<div class="Pp"></div>
INPUT: raw genomic sequence PREDICTION: HGT regions based on Interpolated
  Variable Order Motifs (IVOMs) arguments:
<dl class="Bl-tag">
  <dt class="It-tag">&lt;sequence.file&gt;</dt>
  <dd class="It-tag">raw genomic sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;output.file&gt;</dt>
  <dd class="It-tag">filename for output data
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
Optional arguments:
<dl class="Bl-tag">
  <dt class="It-tag">-a</dt>
  <dd class="It-tag">to load the prediction file into Artemis</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-c</dt>
  <dd class="It-tag">optimize predicted boundaries with a change-point detection
      2 state 2nd order HMM
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
Output files:
<dl class="Bl-tag">
  <dt class="It-tag">&lt;output.file&gt;</dt>
  <dd class="It-tag">predictions (tab file) in embl format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;output.file&gt;.opt</dt>
  <dd class="It-tag">optimized (HMM) predictions (tab file) in embl format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;output.file&gt;.plot</dt>
  <dd class="It-tag">predictions in Artemis User Plot format to be loaded
      manually using Graph -&gt; Add User Plot...</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;output.file&gt;.opt.plot</dt>
  <dd class="It-tag">optimized (HMM) predictions in Artemis User Plot format to
      be loaded manually using Graph -&gt; Add User Plot...</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;output.file&gt;.sco</dt>
  <dd class="It-tag">the scores over all the sliding windows - for score
      distribution check
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
Note: Predictions that overlap with rRNA operon are mentioned in the note
  qualifier
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The manuscript describing the alien_hunter algorithm is available from
  Bioinformatics: Interpolated variable order motifs for identification of
  horizontally acquired DNA: revisiting the Salmonella pathogenicity islands.
  Vernikos GS, Parkhill J Bioinformatics. 2006;. PMID: 16837528
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manual page was written by Andreas Tille &lt;tille@debian.com&gt; for the
  Debian(TM) system (but may be used by others). This man page is released under
  the same conditions as the software, that is GPL.</div>
<table class="foot">
  <tr>
    <td class="foot-date">October 2009</td>
    <td class="foot-os">alien_hunter 1.7</td>
  </tr>
</table>
</body>
</html>
