<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:48:22 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>alimask(1) HMMER Manual alimask(1)</p>

<p style="margin-top: 1em">NAME <br>
alimask - Add mask line to a multiple sequence alignment</p>

<p style="margin-top: 1em">SYNOPSIS <br>
alimask [options] &lt;msafile&gt; &lt;postmsafile&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
alimask is used to apply a mask line to a multiple sequence
alignment, based on provided alignment or model coordinates.
When hmmbuild receives a masked alignment as input, it <br>
produces a profile model in which the emission probabilities
at masked positions are set to match the background
frequency, rather than being set based on observed
frequencies in <br>
the alignment. Position-specific insertion and deletion
rates are not altered, even in masked regions. alimask
autodetects input format, and produces masked alignments in
<br>
Stockholm format. &lt;msafile&gt; may contain only one
sequence alignment.</p>

<p style="margin-top: 1em">A common motivation for masking
a region in an alignment is that the region contains a
simple tandem repeat that is observed to cause an
unacceptably high rate of false positive <br>
hits.</p>

<p style="margin-top: 1em">In the simplest case, a mask
range is given in coordinates relative to the input
alignment, using --alirange &lt;s&gt;. However it is more
often the case that the region to be masked <br>
has been identified in coordinates relative to the profile
model (e.g. based on recognizing a simple repeat pattern in
false hit alignments or in the HMM logo). Not all
align&acirc; <br>
ment columns are converted to match state positions in the
profile (see the --symfrac flag for hmmbuild for
discussion), so model positions do not necessarily match up
to align&acirc; <br>
ment column positions. To remove the burden of converting
model positions to alignment positions, alimask accepts the
mask range input in model coordinates as well, using
--mod&acirc; <br>
elrange &lt;s&gt;. When using this flag, alimask determines
which alignment positions would be identified by hmmbuild as
match states, a process that requires that all hmmbuild
flags <br>
impacting that decision be supplied to alimask. It is for
this reason that many of the hmmbuild flags are also used by
alimask.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
all available options.</p>

<p style="margin-top: 1em">-o &lt;f&gt; Direct the summary
output to file &lt;f&gt;, rather than to stdout.</p>

<p style="margin-top: 1em">OPTIONS FOR SPECIFYING MASK
RANGE <br>
A single mask range is given as a dash-separated pair, like
--modelrange 10-20 and multiple ranges may be submitted as a
comma-separated list, --modelrange 10-20,30-42.</p>

<p style="margin-top: 1em">--modelrange &lt;s&gt; <br>
Supply the given range(s) in model coordinates.</p>

<p style="margin-top: 1em">--alirange &lt;s&gt; <br>
Supply the given range(s) in alignment coordinates.</p>

<p style="margin-top: 1em">--apendmask <br>
Add to the existing mask found with the alignment. The
default is to overwrite any existing mask.</p>

<p style="margin-top: 1em">--model2ali &lt;s&gt; <br>
Rather than actually produce the masked alignment, simply
print model range(s) corresponding to input alignment
range(s).</p>

<p style="margin-top: 1em">--ali2model &lt;s&gt; <br>
Rather than actually produce the masked alignment, simply
print alignment range(s) corresponding to input model
range(s).</p>

<p style="margin-top: 1em">OPTIONS FOR SPECIFYING THE
ALPHABET <br>
The alphabet type (amino, DNA, or RNA) is autodetected by
default, by looking at the composition of the msafile.
Autodetection is normally quite reliable, but occasionally
<br>
alphabet type may be ambiguous and autodetection can fail
(for instance, on tiny toy alignments of just a few
residues). To avoid this, or to increase robustness in
automated <br>
analysis pipelines, you may specify the alphabet type of
msafile with these options.</p>

<p style="margin-top: 1em">--amino <br>
Specify that all sequences in msafile are proteins.</p>

<p style="margin-top: 1em">--dna Specify that all sequences
in msafile are DNAs.</p>

<p style="margin-top: 1em">--rna Specify that all sequences
in msafile are RNAs.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING PROFILE
CONSTRUCTION <br>
These options control how consensus columns are defined in
an alignment.</p>

<p style="margin-top: 1em">--fast Define consensus columns
as those that have a fraction &gt;= symfrac of residues as
opposed to gaps. (See below for the --symfrac option.) This
is the default.</p>

<p style="margin-top: 1em">--hand Define consensus columns
in next profile using reference annotation to the multiple
alignment. This allows you to define any consensus columns
you like.</p>

<p style="margin-top: 1em">--symfrac &lt;x&gt; <br>
Define the residue fraction threshold necessary to define a
consensus column when using the --fast option. The default
is 0.5. The symbol fraction in each column is calcu&acirc;
<br>
lated after taking relative sequence weighting into account,
and ignoring gap characters corresponding to ends of
sequence fragments (as opposed to internal inser&acirc; <br>
tions/deletions). Setting this to 0.0 means that every
alignment column will be assigned as consensus, which may be
useful in some cases. Setting it to 1.0 means that <br>
only columns that include 0 gaps (internal
insertions/deletions) will be assigned as consensus.</p>

<p style="margin-top: 1em">--fragthresh &lt;x&gt; <br>
We only want to count terminal gaps as deletions if the
aligned sequence is known to be full-length, not if it is a
fragment (for instance, because only part of it was <br>
sequenced). HMMER uses a simple rule to infer fragments: if
the sequence length L is less than or equal to a fraction
&lt;x&gt; times the alignment length in columns, then the
<br>
sequence is handled as a fragment. The default is 0.5.
Setting --fragthresh0 will define no (nonempty) sequence as
a fragment; you might want to do this if you know <br>
you&rsquo;ve got a carefully curated alignment of
full-length sequences. Setting --fragthresh1 will define all
sequences as fragments; you might want to do this if you
know <br>
your alignment is entirely composed of fragments, such as
translated short reads in metagenomic shotgun data.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING RELATIVE
WEIGHTS <br>
HMMER uses an ad hoc sequence weighting algorithm to
downweight closely related sequences and upweight distantly
related ones. This has the effect of making models less
biased by <br>
uneven phylogenetic representation. For example, two
identical sequences would typically each receive half the
weight that one sequence would. These options control which
algo&acirc; <br>
rithm gets used.</p>

<p style="margin-top: 1em">--wpb Use the Henikoff
position-based sequence weighting scheme [Henikoff and
Henikoff, J. Mol. Biol. 243:574, 1994]. This is the
default.</p>

<p style="margin-top: 1em">--wgsc Use the
Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et
al, J. Mol. Biol. 235:1067, 1994].</p>

<p style="margin-top: 1em">--wblosum <br>
Use the same clustering scheme that was used to weight data
in calculating BLOSUM subsitution matrices [Henikoff and
Henikoff, Proc. Natl. Acad. Sci 89:10915, 1992]. <br>
Sequences are single-linkage clustered at an identity
threshold (default 0.62; see --wid) and within each cluster
of c sequences, each sequence gets relative weight 1/c.</p>

<p style="margin-top: 1em">--wnone <br>
No relative weights. All sequences are assigned uniform
weight.</p>

<p style="margin-top: 1em">--wid &lt;x&gt; <br>
Sets the identity threshold used by single-linkage
clustering when using --wblosum. Invalid with any other
weighting scheme. Default is 0.62.</p>

<p style="margin-top: 1em">OTHER OPTIONS <br>
--informat &lt;s&gt; <br>
Declare that the input msafile is in format &lt;s&gt;.
Currently the accepted multiple alignment sequence file
formats include Stockholm, Aligned FASTA, Clustal, NCBI PSI-
<br>
BLAST, PHYLIP, Selex, and UCSC SAM A2M. Default is to
autodetect the format of the file.</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Seed the random number generator with &lt;n&gt;, an integer
&gt;= 0. If &lt;n&gt; is nonzero, any stochastic simulations
will be reproducible; the same command will give the same
<br>
results. If &lt;n&gt; is 0, the random number generator is
seeded arbitrarily, and stochastic simulations will vary
from run to run of the same command. The default seed is
<br>
42.</p>

<p style="margin-top: 1em">SEE ALSO <br>
See hmmer(1) for a master man page with a list of all the
individual man pages for programs in the HMMER package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
alimask(1)</p>
<hr>
</body>
</html>
