<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>alimask(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">alimask(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">alimask(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
alimask - Add mask line to a multiple sequence alignment
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>alimask</b> <i>[options]</i> <i>&lt;msafile&gt;</i>
  <i>&lt;postmsafile&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>alimask</b> is used to apply a mask line to a multiple sequence alignment,
  based on provided alignment or model coordinates. When <b>hmmbuild</b>
  receives a masked alignment as input, it produces a profile model in which the
  emission probabilities at masked positions are set to match the background
  frequency, rather than being set based on observed frequencies in the
  alignment. Position-specific insertion and deletion rates are not altered,
  even in masked regions. <b>alimask</b> autodetects input format, and produces
  masked alignments in Stockholm format. <i>&lt;msafile&gt;</i> may contain only
  one sequence alignment.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
A common motivation for masking a region in an alignment is that the region
  contains a simple tandem repeat that is observed to cause an unacceptably high
  rate of false positive hits.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
In the simplest case, a mask range is given in coordinates relative to the input
  alignment, using <b>--alirange</b><i> &lt;s&gt;</i><b>.</b> However it is more
  often the case that the region to be masked has been identified in coordinates
  relative to the profile model (e.g. based on recognizing a simple repeat
  pattern in false hit alignments or in the HMM logo). Not all alignment columns
  are converted to match state positions in the profile (see the
  <i>--symfrac</i> flag for <b>hmmbuild</b> for discussion), so model positions
  do not necessarily match up to alignment column positions. To remove the
  burden of converting model positions to alignment positions, <b>alimask</b>
  accepts the mask range input in model coordinates as well, using
  <b>--modelrange</b><i> &lt;s&gt;</i><b>.</b> When using this flag,
  <i>alimask</i> determines which alignment positions would be identified by
  <b>hmmbuild</b> as match states, a process that requires that all
  <b>hmmbuild</b> flags impacting that decision be supplied to <b>alimask</b>.
  It is for this reason that many of the <b>hmmbuild</b> flags are also used by
  <b>alimask</b>.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and all
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Direct the summary output to file <i>&lt;f&gt;</i>, rather
      than to <i>stdout</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_SPECIFYING_MASK_RANGE"><a class="selflink" href="#OPTIONS_FOR_SPECIFYING_MASK_RANGE">OPTIONS
  FOR SPECIFYING MASK RANGE</a></h1>
A single mask range is given as a dash-separated pair, like <i>--modelrange
  10-20</i> and multiple ranges may be submitted as a comma-separated list,
  <i>--modelrange 10-20,30-42</i>.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--modelrange</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Supply the given range(s) in model coordinates.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--alirange</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Supply the given range(s) in alignment coordinates.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--apendmask</b></dt>
  <dd class="It-tag">Add to the existing mask found with the alignment. The
      default is to overwrite any existing mask.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--model2ali</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Rather than actually produce the masked alignment, simply
      print model range(s) corresponding to input alignment range(s).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ali2model</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Rather than actually produce the masked alignment, simply
      print alignment range(s) corresponding to input model range(s).
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_SPECIFYING_THE_ALPHABET"><a class="selflink" href="#OPTIONS_FOR_SPECIFYING_THE_ALPHABET">OPTIONS
  FOR SPECIFYING THE ALPHABET</a></h1>
The alphabet type (amino, DNA, or RNA) is autodetected by default, by looking at
  the composition of the <i>msafile</i>. Autodetection is normally quite
  reliable, but occasionally alphabet type may be ambiguous and autodetection
  can fail (for instance, on tiny toy alignments of just a few residues). To
  avoid this, or to increase robustness in automated analysis pipelines, you may
  specify the alphabet type of <i>msafile</i> with these options.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--amino</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>msafile</i> are proteins.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dna</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>msafile</i> are DNAs.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rna</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>msafile</i> are RNAs.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_PROFILE_CONSTRUCTION"><a class="selflink" href="#OPTIONS_CONTROLLING_PROFILE_CONSTRUCTION">OPTIONS
  CONTROLLING PROFILE CONSTRUCTION</a></h1>
These options control how consensus columns are defined in an alignment.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fast</b></dt>
  <dd class="It-tag">Define consensus columns as those that have a fraction
      &gt;= <b>symfrac</b> of residues as opposed to gaps. (See below for the
      <b>--symfrac</b> option.) This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hand</b></dt>
  <dd class="It-tag">Define consensus columns in next profile using reference
      annotation to the multiple alignment. This allows you to define any
      consensus columns you like.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--symfrac</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Define the residue fraction threshold necessary to define a
      consensus column when using the <b>--fast</b> option. The default is 0.5.
      The symbol fraction in each column is calculated after taking relative
      sequence weighting into account, and ignoring gap characters corresponding
      to ends of sequence fragments (as opposed to internal
      insertions/deletions). Setting this to 0.0 means that every alignment
      column will be assigned as consensus, which may be useful in some cases.
      Setting it to 1.0 means that only columns that include 0 gaps (internal
      insertions/deletions) will be assigned as consensus.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fragthresh</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">We only want to count terminal gaps as deletions if the
      aligned sequence is known to be full-length, not if it is a fragment (for
      instance, because only part of it was sequenced). HMMER uses a simple rule
      to infer fragments: if the sequence length L is less than or equal to a
      fraction <i>&lt;x&gt;</i> times the alignment length in columns, then the
      sequence is handled as a fragment. The default is 0.5. Setting
      <b>--fragthresh</b><i>0</i> will define no (nonempty) sequence as a
      fragment; you might want to do this if you know you've got a carefully
      curated alignment of full-length sequences. Setting
      <b>--fragthresh</b><i>1</i> will define all sequences as fragments; you
      might want to do this if you know your alignment is entirely composed of
      fragments, such as translated short reads in metagenomic shotgun data.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_RELATIVE_WEIGHTS"><a class="selflink" href="#OPTIONS_CONTROLLING_RELATIVE_WEIGHTS">OPTIONS
  CONTROLLING RELATIVE WEIGHTS</a></h1>
HMMER uses an ad hoc sequence weighting algorithm to downweight closely related
  sequences and upweight distantly related ones. This has the effect of making
  models less biased by uneven phylogenetic representation. For example, two
  identical sequences would typically each receive half the weight that one
  sequence would. These options control which algorithm gets used.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wpb</b></dt>
  <dd class="It-tag">Use the Henikoff position-based sequence weighting scheme
      [Henikoff and Henikoff, J. Mol. Biol. 243:574, 1994]. This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wgsc</b></dt>
  <dd class="It-tag">Use the Gerstein/Sonnhammer/Chothia weighting algorithm
      [Gerstein et al, J. Mol. Biol. 235:1067, 1994].
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wblosum</b></dt>
  <dd class="It-tag">Use the same clustering scheme that was used to weight data
      in calculating BLOSUM subsitution matrices [Henikoff and Henikoff, Proc.
      Natl. Acad. Sci 89:10915, 1992]. Sequences are single-linkage clustered at
      an identity threshold (default 0.62; see <i>--wid</i>) and within each
      cluster of c sequences, each sequence gets relative weight 1/c.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wnone</b></dt>
  <dd class="It-tag">No relative weights. All sequences are assigned uniform
      weight.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wid</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the identity threshold used by single-linkage
      clustering when using <i>--wblosum</i>. Invalid with any other weighting
      scheme. Default is 0.62.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_OPTIONS"><a class="selflink" href="#OTHER_OPTIONS">OTHER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Declare that the input <i>msafile</i> is in format
      <i>&lt;s&gt;</i>. Currently the accepted multiple alignment sequence file
      formats include Stockholm, Aligned FASTA, Clustal, NCBI PSI-BLAST, PHYLIP,
      Selex, and UCSC SAM A2M. Default is to autodetect the format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Seed the random number generator with <i>&lt;n&gt;</i>, an
      integer &gt;= 0. If <i>&lt;n&gt;</i> is nonzero, any stochastic
      simulations will be reproducible; the same command will give the same
      results. If <i>&lt;n&gt;</i> is 0, the random number generator is seeded
      arbitrarily, and stochastic simulations will vary from run to run of the
      same command. The default seed is 42.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>hmmer(1)</b> for a master man page with a list of all the individual man
  pages for programs in the HMMER package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your HMMER
  distribution (Userguide.pdf); or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2015 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2015</td>
    <td class="foot-os">HMMER 3.1b2</td>
  </tr>
</table>
</body>
</html>
