<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:48:22 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>alistat(1) Biosquid Manual alistat(1)</p>

<p style="margin-top: 1em">NAME <br>
alistat - show statistics for a multiple alignment file</p>

<p style="margin-top: 1em">SYNOPSIS <br>
alistat [options] alignfile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
alistat reads a multiple sequence alignment from the file
alignfile in any supported format (including SELEX, GCG MSF,
and CLUSTAL), and shows a number of simple statistics about
<br>
it. These statistics include the name of the format, the
number of sequences, the total number of residues, the
average and range of the sequence lengths, the alignment
length <br>
(e.g. including gap characters).</p>

<p style="margin-top: 1em">Also shown are some percent
identities. A percent pairwise alignment identity is defined
as (idents / MIN(len1, len2)) where idents is the number of
exact identities and len1, <br>
len2 are the unaligned lengths of the two sequences. The
&quot;average percent identity&quot;, &quot;most related
pair&quot;, and &quot;most unrelated pair&quot; of the
alignment are the average, maximum, and <br>
minimum of all (N)(N-1)/2 pairs, respectively. The
&quot;most distant seq&quot; is calculated by finding the
maximum pairwise identity (best relative) for all N
sequences, then finding <br>
the minimum of these N numbers (hence, the most outlying
sequence).</p>

<p style="margin-top: 1em">OPTIONS <br>
-a Show additional verbose information: a table with one
line per sequence showing name, length, and its highest and
lowest pairwise identity. These lines are prefixed with a
<br>
* character to enable easily grep&rsquo;ing them out and
sorting them. For example, alistat -a foo.slx | grep * |
sort -n +3 gives a ranked list of the most distant sequences
in <br>
the alignment. Incompatible with the -f option.</p>

<p style="margin-top: 1em">-f Fast; use a sampling method
to estimate the average %id. When this option is chosen,
alistat doesn&rsquo;t show the other three pairwise identity
numbers. This option is use&acirc; <br>
ful for very large alignments, for which the full (N)(N-1)
calculation of all pairs would be prohibitive (e.g.
Pfam&rsquo;s GP120 alignment, with over 10,000 sequences).
Incom&acirc; <br>
patible with the -a option.</p>

<p style="margin-top: 1em">-h Print brief help; includes
version number and summary of all options, including expert
options.</p>

<p style="margin-top: 1em">-q be quiet - suppress the
verbose header (program name, release number and date, the
parameters and options in effect).</p>

<p style="margin-top: 1em">-B (Babelfish). Autodetect and
read a sequence file format other than the default (FASTA).
Almost any common sequence file format is recognized
(including Genbank, EMBL, <br>
SWISS-PROT, PIR, and GCG unaligned sequence formats, and
Stockholm, GCG MSF, and Clustal alignment formats). See the
printed documentation for a complete list of supported <br>
formats.</p>

<p style="margin-top: 1em">EXPERT OPTIONS <br>
--informat &lt;s&gt; <br>
Specify that the sequence file is in format &lt;s&gt;,
rather than the default FASTA format. Common examples
include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF, or
<br>
PHYLIP; see the printed documentation for a complete list of
accepted format names. This option overrides the default
format (FASTA) and the -B Babelfish autodetection <br>
option.</p>

<p style="margin-top: 1em">SEE ALSO <br>
afetch(1), compalign(1), compstruct(1), revcomp(1),
seqsplit(1), seqstat(1), sfetch(1), shuffle(1), sindex(1),
sreformat(1), stranslate(1), weight(1).</p>

<p style="margin-top: 1em">AUTHOR <br>
Biosquid and its documentation are Copyright (C) 1992-2003
HHMI/Washington University School of Medicine Freely
distributed under the GNU General Public License (GPL) See
COPYING <br>
in the source code distribution for more details, or contact
me.</p>

<p style="margin-top: 1em">Sean Eddy <br>
HHMI/Department of Genetics <br>
Washington University School of Medicine <br>
4444 Forest Park Blvd., Box 8510 <br>
St Louis, MO 63108 USA <br>
Phone: 1-314-362-7666 <br>
FAX : 1-314-362-2157 <br>
Email: eddy@genetics.wustl.edu</p>

<p style="margin-top: 1em">Biosquid 1.9g January 2003
alistat(1)</p>
<hr>
</body>
</html>
