<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>alistat(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">alistat(1)</td>
    <td class="head-vol">Biosquid Manual</td>
    <td class="head-rtitle">alistat(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">alistat - show statistics for a multiple alignment
    file</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>alistat</b> <i>[options]</i> <i>alignfile</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>alistat</b> reads a multiple sequence alignment from the file
  <i>alignfile</i> in any supported format (including SELEX, GCG MSF, and
  CLUSTAL), and shows a number of simple statistics about it. These statistics
  include the name of the format, the number of sequences, the total number of
  residues, the average and range of the sequence lengths, the alignment length
  (e.g. including gap characters).
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Also shown are some percent identities. A percent pairwise alignment identity is
  defined as <i>(idents / MIN(len1, len2))</i> where <i>idents</i> is the number
  of exact identities and <i>len1, len2</i> are the unaligned lengths of the two
  sequences. The &quot;average percent identity&quot;, &quot;most related
  pair&quot;, and &quot;most unrelated pair&quot; of the alignment are the
  average, maximum, and minimum of all (N)(N-1)/2 pairs, respectively. The
  &quot;most distant seq&quot; is calculated by finding the maximum pairwise
  identity (best relative) for all N sequences, then finding the minimum of
  these N numbers (hence, the most outlying sequence).
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b></dt>
  <dd class="It-tag">Show additional verbose information: a table with one line
      per sequence showing name, length, and its highest and lowest pairwise
      identity. These lines are prefixed with a * character to enable easily
      <b>grep'</b>ing them out and sorting them. For example, <i>alistat -a
      foo.slx | grep * | sort -n +3</i> gives a ranked list of the most distant
      sequences in the alignment. Incompatible with the <b>-f</b> option.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b></dt>
  <dd class="It-tag">Fast; use a sampling method to estimate the average %id.
      When this option is chosen, <b>alistat</b> doesn't show the other three
      pairwise identity numbers. This option is useful for very large
      alignments, for which the full (N)(N-1) calculation of all pairs would be
      prohibitive (e.g. Pfam's GP120 alignment, with over 10,000 sequences).
      Incompatible with the <b>-a</b> option.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Print brief help; includes version number and summary of
      all options, including expert options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b></dt>
  <dd class="It-tag">be quiet - suppress the verbose header (program name,
      release number and date, the parameters and options in effect).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b></dt>
  <dd class="It-tag">(Babelfish). Autodetect and read a sequence file format
      other than the default (FASTA). Almost any common sequence file format is
      recognized (including Genbank, EMBL, SWISS-PROT, PIR, and GCG unaligned
      sequence formats, and Stockholm, GCG MSF, and Clustal alignment formats).
      See the printed documentation for a complete list of supported formats.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXPERT_OPTIONS"><a class="selflink" href="#EXPERT_OPTIONS">EXPERT
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Specify that the sequence file is in format
      <i>&lt;s&gt;,</i> rather than the default FASTA format. Common examples
      include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF, or PHYLIP; see
      the printed documentation for a complete list of accepted format names.
      This option overrides the default format (FASTA) and the <i>-B</i>
      Babelfish autodetection option.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>afetch</b>(1), <b>compalign</b>(1), <b>compstruct</b>(1), <b>revcomp</b>(1),
  <b>seqsplit</b>(1), <b>seqstat</b>(1), <b>sfetch</b>(1), <b>shuffle</b>(1),
  <b>sindex</b>(1), <b>sreformat</b>(1), <b>stranslate</b>(1), <b>weight</b>(1).
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Biosquid and its documentation are Copyright (C) 1992-2003 HHMI/Washington
  University School of Medicine Freely distributed under the GNU General Public
  License (GPL) See COPYING in the source code distribution for more details, or
  contact me.
<div style="height: 1.00em;">&#x00A0;</div>
<pre>
Sean Eddy
HHMI/Department of Genetics
Washington University School of Medicine
4444 Forest Park Blvd., Box 8510
St Louis, MO 63108 USA
Phone: 1-314-362-7666
FAX  : 1-314-362-2157
Email: eddy@genetics.wustl.edu
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2003</td>
    <td class="foot-os">Biosquid 1.9g</td>
  </tr>
</table>
</body>
</html>
