<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:48:24 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ALLVERSUSALL(1e) EMBOSS Manual for Debian
ALLVERSUSALL(1e)</p>

<p style="margin-top: 1em">NAME <br>
allversusall - Sequence similarity data from all-versus-all
comparison.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
allversusall -seqinpath dirlist [-matrix matrixf] [-gapopen
float] [-gapextend float] -datoutdir outdir -logfile
outfile</p>

<p style="margin-top: 1em">allversusall -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
allversusall is a command line program from EMBOSS
(&acirc;the European Molecular Biology Open Software
Suite&acirc;). It is part of the &quot;Utils:Database
creation&quot; command group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-seqinpath dirlist <br>
Default value: ./</p>

<p style="margin-top: 1em">-matrix matrixf <br>
This option specifies the residue substitution matrix that
is used for sequence comparison. Default value:
EBLOSUM62</p>

<p style="margin-top: 1em">Required section <br>
Additional section <br>
-gapopen float <br>
This option specifies the gap insertion penalty. The gap
insertion penalty is the score taken away when a gap is
created. The best value depends on the choice of comparison
<br>
matrix. The default value assumes you are using the
EBLOSUM62 matrix for protein sequences, and the EDNAFULL
matrix for nucleotide sequences. Default value: 10</p>

<p style="margin-top: 1em">-gapextend float <br>
This option specifies the gap extension penalty. The gap
extension, penalty is added to the standard gap penalty for
each base or residue in the gap. This is how long gaps <br>
are penalized. Usually you will expect a few long gaps
rather than many short gaps, so the gap extension penalty
should be lower than the gap penalty. An exception is where
<br>
one or both sequences are single reads with possible
sequencing errors in which case you would expect many single
base gaps. You can get this result by setting the gap open
<br>
penalty to zero (or very low) and using the gap extension
penalty to control gap scoring. Default value: 0.5</p>

<p style="margin-top: 1em">Output section <br>
-datoutdir outdir <br>
This option specifies the location of sequence similarity
data files (output). Default value: ./</p>

<p style="margin-top: 1em">-logfile outfile <br>
This option specifies the name of ALLVERSUSALL log file
(output). The log file contains messages about any errors
arising while ALLVERSUSALL ran. Default value: <br>
allversusall.log</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
allversusall is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">DOMALIGN 0.1.0+20100721
08/11/2010 ALLVERSUSALL(1e)</p>
<hr>
</body>
</html>
