<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:48:51 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ALTER-SEQUENCE-ALIGNMENT(1) User Commands
ALTER-SEQUENCE-ALIGNMENT(1)</p>

<p style="margin-top: 1em">NAME <br>
alter-sequence-alignment - genomic sequences ALignment
Transformation EnviRonment</p>

<p style="margin-top: 1em">SYNOPSIS <br>
alter-sequence-alignment [option] [sequence]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ALTER (ALignment Transformation EnviRonment) is a tool to
transform <br>
between multiple sequence alignment formats. ALTER focuses
on the <br>
specifications of mainstream alignment and analysis programs
rather than <br>
on the conversion among more or less specific formats.</p>

<p style="margin-top: 1em">OPTIONS <br>
-c (--collapse) <br>
Collapse sequences to haplotypes.</p>

<p style="margin-top: 1em">-cg (--collapseGaps) <br>
Treat gaps as missing data when collapsing.</p>

<p style="margin-top: 1em">-cl (--collapseLimit) N <br>
Connection limit (sequences differing at &lt;= l sites will
be collapsed) (default is l=0).</p>

<p style="margin-top: 1em">-cm (--collapseMissing) <br>
Count missing data as differences when collapsing.</p>

<p style="margin-top: 1em">-i (--input) FILE <br>
Input file.</p>

<p style="margin-top: 1em">-ia (--inputAutodetect) <br>
Autodetect format (other input options are omitted).</p>

<p style="margin-top: 1em">-if (--inputFormat) VAL <br>
Input format (ALN, FASTA, GDE, MEGA, MSF, NEXUS, PHYLIP or
PIR).</p>

<p style="margin-top: 1em">-io (--inputOS) VAL <br>
Input operating system (Linux, MacOS or Windows)</p>

<p style="margin-top: 1em">-ip (--inputProgram) VAL <br>
Input program (Clustal, MAFFT, MUSCLE, PROBCONS or
TCoffee).</p>

<p style="margin-top: 1em">-o (--output) FILE <br>
Output file.</p>

<p style="margin-top: 1em">-of (--outputFormat) VAL <br>
Output format (ALN, FASTA, GDE, MEGA, MSF, NEXUS, PHYLIP or
PIR).</p>

<p style="margin-top: 1em">-ol (--outputLowerCase) <br>
Lowe case output.</p>

<p style="margin-top: 1em">-om (--outputMatch) <br>
Output match characters.</p>

<p style="margin-top: 1em">-on (--outputResidueNumbers)
<br>
Output residue numbers (only ALN format).</p>

<p style="margin-top: 1em">-oo (--outputOS) VAL <br>
Output operating system (Linux, MacOS or Windows).</p>

<p style="margin-top: 1em">-op (--outputProgram) VAL <br>
Output program (jModelTest, MrBayes, PAML, PAUP, PhyML,
ProtTest, RAxML, TCS, CodABC, BioEdit, MEGA, dnaSP, Se-Al,
Mesquite, SplitsTree, Clustal , MAFFT, MUSCLE, PROBCONS,
<br>
TCoffee, Gblocks, SeaView, trimAl or GENERAL)</p>

<p style="margin-top: 1em">-os (--outputSequential) <br>
Sequential output (only NEXUS and PHYLIP formats).</p>

<p style="margin-top: 1em">alter-sequence-alignment 1.3.3
September 2015 ALTER-SEQUENCE-ALIGNMENT(1)</p>
<hr>
</body>
</html>
