<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ALTREE(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ALTREE(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">ALTREE(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
altree - Association and Localisation tests using phylogenetic Trees
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
altree [options]
<div class="Pp"></div>
<pre>
 Options:
    --version        program version
    --short-help|h   brief help message
    --help           help message with options descriptions
    --man            full documentation
    --association|a  perform the association test
    --s-localisation|l   perform the localisation using the S character
    --first-input-file|i result_file from phylogeny reconstruction programs
    --second-input-file|j file containing the nb of cases/controls carrying an haplotype
    --output-file|o output_file
    --data-type|t DNA|SNP
    --data-qual|q qualitative|quantitative
    --outgroup outgroup_name
    --remove-outgroup 
    --tree-building-program|p phylip|paup|paml
    --splitmode|s nosplit|chi2split
    --no-prolongation
    --chi2-threshold|n value
    --permutations|r number
    --number-of-trees-to-analyse number
    --tree-to-analyse number
    --s-site-number number
    --s-site-characters ancestral state -&gt; derived state
    --co-evo|e simple|double
    --print-tree 
    --anc-seq ancestral sequence (only with phylip)
    --nb-files number of input files to analyse (only for association test)
</pre>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">Print the program version and exits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--short-help</b></dt>
  <dd class="It-tag">Print a brief help message and exits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">Print a help message with options descriptions and
    exits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--man</b></dt>
  <dd class="It-tag">Prints the manual page and exits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--association|a</b></dt>
  <dd class="It-tag">Perform the association test</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--s-localisation|l</b></dt>
  <dd class="It-tag">Localise the susceptibility locus using the
      &quot;S-character method&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--first-input-file|i</b> <i>result_file</i></dt>
  <dd class="It-tag">Input file 1 (paup, phylip or paml results file). If
      several input files are analysed, their names must be separated by colons.
      Example: input1:input2 etc</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--second-input-file|j</b> <i>correspond_file</i></dt>
  <dd class="It-tag">Input file 2, default <i>correspond.txt</i>. The number of
      input file 2 must be the same as the number of input file 1. The name of
      the different input file 2 must be separated by colons</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--output-file|o</b> <i>outfile</i></dt>
  <dd class="It-tag">Output file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--data-type|t</b> &quot;DNA&quot;|&quot;SNP&quot;</dt>
  <dd class="It-tag">Type of data: DNA (ATGCU) or SNP (0-1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--data-qual|q</b>
    &quot;qualitative&quot;|&quot;quantitative&quot;</dt>
  <dd class="It-tag">Analyse qualitative (case/control) or quantitative
    data</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outgroup</b> outgroup</dt>
  <dd class="It-tag">Root the tree with this outgroup</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--remove-outgroup</b></dt>
  <dd class="It-tag">Remove the outgroup of the tree before performing the
    tests</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tree-building-program|p</b>
    &quot;phylip&quot;|&quot;paup&quot;|&quot;paml&quot;</dt>
  <dd class="It-tag">Phylogeny reconstruction program</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--splitmode|s</b>
    &quot;nosplit&quot;|&quot;chi2split&quot;</dt>
  <dd class="It-tag">how tests are performed from a level to another</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-prolongation</b></dt>
  <dd class="It-tag">No prolongation of branches in the tree</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--chi2-threshold|n</b> value</dt>
  <dd class="It-tag">Significance threshold for chi2 (default value 0.01)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--permutations|r</b> number</dt>
  <dd class="It-tag">Number of permutations used to calculate exact p_values
      (Only for association test)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--number-of-trees-to-analyse</b> number</dt>
  <dd class="It-tag">Number of trees to analyse in the localisation analysis
      (only for localisation method using S-character)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tree-to-analyse</b> number</dt>
  <dd class="It-tag">With this option, you can specify the tree to use (instead
      of random). Can be used several times to specify multiple trees.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--s-site-number</b> number</dt>
  <dd class="It-tag">Number of the S character site in the sequence (only for
      localisation method using S-character)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--s-site-characters</b> transition</dt>
  <dd class="It-tag">Character states for the S character: ancestral state -&gt;
      derived state ex: G-&gt;C or 0-&gt;1 (only for localisation method using
      S-character)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--co-evo|e</b>
    &quot;simple&quot;|&quot;double&quot;</dt>
  <dd class="It-tag">Type of co-evolution indice
    <br/>
     simple: only the anc -&gt; der transition of S is used
    <br/>
     double: the two possible transitions are used</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--print-tree</b></dt>
  <dd class="It-tag">If this option is selected, the tree will be printed to the
      output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--anc-seq</b> anc_seq</dt>
  <dd class="It-tag">With this option, you can specify the ancestral sequence.
      This option is only useful when the tree is reconstructed using the mix
      program of phylip with the ancestral states specified in the file
      &quot;ancestors&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nb-files</b> number</dt>
  <dd class="It-tag">With this option, you specify the number of input files (1
      and 2) to analyse This option only works for the association test. Be
      careful if the number of trees is not the same for the different input
      files: if the chosen tree doesn't exist in one file, the program wil not
      work correctly</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>This program</b> performs
<div class="Pp"></div>
(a) an association test between a candidate gene and disease or a quantitative
  trait
<div class="Pp"></div>
(b) a localsation tests: it allows to detect which SNP is involved in the
  determinism of the disease or the quantitative trait
<div class="Pp"></div>
These two tests are based on the analysis of haplotype phylogenetic trees.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-10-24</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
