<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:48:52 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ALTREE(1p) User Contributed Perl Documentation
ALTREE(1p)</p>

<p style="margin-top: 1em">NAME <br>
altree - Association and Localisation tests using
phylogenetic Trees</p>

<p style="margin-top: 1em">SYNOPSIS <br>
altree [options]</p>

<p style="margin-top: 1em">Options: <br>
--version program version <br>
--short-help|h brief help message <br>
--help help message with options descriptions <br>
--man full documentation <br>
--association|a perform the association test <br>
--s-localisation|l perform the localisation using the S
character <br>
--first-input-file|i result_file from phylogeny
reconstruction programs <br>
--second-input-file|j file containing the nb of
cases/controls carrying an haplotype <br>
--output-file|o output_file <br>
--data-type|t DNA|SNP <br>
--data-qual|q qualitative|quantitative <br>
--outgroup outgroup_name <br>
--remove-outgroup <br>
--tree-building-program|p phylip|paup|paml <br>
--splitmode|s nosplit|chi2split <br>
--no-prolongation <br>
--chi2-threshold|n value <br>
--permutations|r number <br>
--number-of-trees-to-analyse number <br>
--tree-to-analyse number <br>
--s-site-number number <br>
--s-site-characters ancestral state -&gt; derived state <br>
--co-evo|e simple|double <br>
--print-tree <br>
--anc-seq ancestral sequence (only with phylip) <br>
--nb-files number of input files to analyse (only for
association test)</p>

<p style="margin-top: 1em">OPTIONS <br>
--version <br>
Print the program version and exits.</p>

<p style="margin-top: 1em">--short-help <br>
Print a brief help message and exits.</p>

<p style="margin-top: 1em">--help Print a help message with
options descriptions and exits.</p>

<p style="margin-top: 1em">--man Prints the manual page and
exits.</p>

<p style="margin-top: 1em">--association|a <br>
Perform the association test</p>

<p style="margin-top: 1em">--s-localisation|l <br>
Localise the susceptibility locus using the
&quot;S-character method&quot;</p>

<p style="margin-top: 1em">--first-input-file|i result_file
<br>
Input file 1 (paup, phylip or paml results file). If several
input files are analysed, their names must be separated by
colons. Example: input1:input2 etc</p>

<p style="margin-top: 1em">--second-input-file|j
correspond_file <br>
Input file 2, default correspond.txt. The number of input
file 2 must be the same as the number of input file 1. The
name of the different input file 2 must be <br>
separated by colons</p>

<p style="margin-top: 1em">--output-file|o outfile <br>
Output file</p>

<p style="margin-top: 1em">--data-type|t
&quot;DNA&quot;|&quot;SNP&quot; <br>
Type of data: DNA (ATGCU) or SNP (0-1)</p>

<p style="margin-top: 1em">--data-qual|q
&quot;qualitative&quot;|&quot;quantitative&quot; <br>
Analyse qualitative (case/control) or quantitative data</p>

<p style="margin-top: 1em">--outgroup outgroup <br>
Root the tree with this outgroup</p>

<p style="margin-top: 1em">--remove-outgroup <br>
Remove the outgroup of the tree before performing the
tests</p>

<p style="margin-top: 1em">--tree-building-program|p
&quot;phylip&quot;|&quot;paup&quot;|&quot;paml&quot; <br>
Phylogeny reconstruction program</p>

<p style="margin-top: 1em">--splitmode|s
&quot;nosplit&quot;|&quot;chi2split&quot; <br>
how tests are performed from a level to another</p>

<p style="margin-top: 1em">--no-prolongation <br>
No prolongation of branches in the tree</p>

<p style="margin-top: 1em">--chi2-threshold|n value <br>
Significance threshold for chi2 (default value 0.01)</p>

<p style="margin-top: 1em">--permutations|r number <br>
Number of permutations used to calculate exact p_values
(Only for association test)</p>

<p style="margin-top: 1em">--number-of-trees-to-analyse
number <br>
Number of trees to analyse in the localisation analysis
(only for localisation method using S-character)</p>

<p style="margin-top: 1em">--tree-to-analyse number <br>
With this option, you can specify the tree to use (instead
of random). Can be used several times to specify multiple
trees.</p>

<p style="margin-top: 1em">--s-site-number number <br>
Number of the S character site in the sequence (only for
localisation method using S-character)</p>

<p style="margin-top: 1em">--s-site-characters transition
<br>
Character states for the S character: ancestral state -&gt;
derived state ex: G-&gt;C or 0-&gt;1 (only for localisation
method using S-character)</p>

<p style="margin-top: 1em">--co-evo|e
&quot;simple&quot;|&quot;double&quot; <br>
Type of co-evolution indice <br>
simple: only the anc -&gt; der transition of S is used <br>
double: the two possible transitions are used</p>

<p style="margin-top: 1em">--print-tree <br>
If this option is selected, the tree will be printed to the
output</p>

<p style="margin-top: 1em">--anc-seq anc_seq <br>
With this option, you can specify the ancestral sequence.
This option is only useful when the tree is reconstructed
using the mix program of phylip with the ancestral <br>
states specified in the file &quot;ancestors&quot;</p>

<p style="margin-top: 1em">--nb-files number <br>
With this option, you specify the number of input files (1
and 2) to analyse This option only works for the association
test. Be careful if the number of trees is not <br>
the same for the different input files: if the chosen tree
doesn&rsquo;t exist in one file, the program wil not work
correctly</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This program performs</p>

<p style="margin-top: 1em">(a) an association test between
a candidate gene and disease or a quantitative trait</p>

<p style="margin-top: 1em">(b) a localsation tests: it
allows to detect which SNP is involved in the determinism of
the disease or the quantitative trait</p>

<p style="margin-top: 1em">These two tests are based on the
analysis of haplotype phylogenetic trees.</p>

<p style="margin-top: 1em">perl v5.24.1 2016-10-24
ALTREE(1p)</p>
<hr>
</body>
</html>
