<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:48:52 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ALTREE-CONVERT(1p) User Contributed Perl Documentation
ALTREE-CONVERT(1p)</p>

<p style="margin-top: 1em">NAME <br>
altree-convert - Title...</p>

<p style="margin-top: 1em">SYNOPSIS <br>
altree-convert [options]</p>

<p style="margin-top: 1em">Options: <br>
--version program version <br>
--short-help|h brief help message <br>
--help help message with options descriptions <br>
--man full documentation <br>
--first-input-file|i input file 1 <br>
--second-input-file|j input file 2 (not mandatory) <br>
--output-file|o output file <br>
--case-control-output|c output containing the nb
cases/controls <br>
--data-type|t DNA|NUM <br>
--phylo-prog|p PAUP|PHYLIP <br>
--reconstruct-prog|r PHASE|FAMHAP <br>
--data-quali|q Type of data: qualitative or quantitative
<br>
--nbind-threshold|s Minimum number of individuals ecquired
to keep an haplotype</p>

<p style="margin-top: 1em">OPTIONS <br>
--version <br>
Print the program version and exits.</p>

<p style="margin-top: 1em">--short-help <br>
Print a brief help message and exits.</p>

<p style="margin-top: 1em">--help Print a help message with
options descriptions and exits.</p>

<p style="margin-top: 1em">--man Prints the manual page and
exits.</p>

<p style="margin-top: 1em">--first-input-file|i <br>
Input file 1 (output of the haplotype reconstruction
program)</p>

<p style="margin-top: 1em">--second-input-file|j <br>
Input file 2 (second output of famhap or file containig the
disease status)</p>

<p style="margin-top: 1em">--output-file|o <br>
Output file</p>

<p style="margin-top: 1em">--case-control-output|c <br>
Output file containing the number of cases and controls
carrying each haplotype</p>

<p style="margin-top: 1em">--data-type|t
&quot;DNA&quot;|&quot;SNP&quot; <br>
Type of data: DNA (ATGCU) or SNP (0-1)</p>

<p style="margin-top: 1em">--phylo-prog|p
&quot;phylip&quot;|&quot;paup&quot; <br>
Phylogeny reconstruction program</p>

<p style="margin-top: 1em">reconstruct-prog|r
&quot;famhap|phase&quot; <br>
Haplotype reconstruction program</p>

<p style="margin-top: 1em">data-quali|q
&quot;qualitative|quantitative&quot; <br>
Type of data analyzed</p>

<p style="margin-top: 1em">nbind-threshold|s <br>
Minimal number of individuals carrying an haplotype
recquired to keep it for further analysis. If you want to
keep all haplotypes, you must affect 0 to this variable</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This program will read the given input file(s) and generate
an input file for the phylogenetic reconstruction software
paup or phylip/paml</p>

<p style="margin-top: 1em">perl v5.24.1 2016-10-24
ALTREE-CONVERT(1p)</p>
<hr>
</body>
</html>
