<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>AMAP(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">AMAP(1)</td>
    <td class="head-vol">User Manual</td>
    <td class="head-rtitle">AMAP(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
amap - Protein multiple alignment by sequence annealing
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>amap</b>
  [<b></b><i>OPTION</i>] [<i>MFAFILE</i>] [<i>MFAFILE</i>]</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
AMAP is a tool to perform multiple alignment of peptidic sequences. It utilizes
  posterior decoding, and a sequence-annealing alignment, instead of the
  traditional progressive alignment method. It is the only alignment program
  that allows one to control the sensitivity / specificity tradeoff. It is based
  on the ProbCons source code, but uses alignment metric accuracy and eliminates
  the consistency transformation.
<div class="Pp"></div>
In its default configuration, AMAP is tuned to maximize the expected Alignment
  Metric Accuracy (AMA) score - a new alignment accuracy measure, based on a
  metric for the multiple-alignment space, which integrates sensitivity and
  specificity into a single balanced measure. AMA is defined as the fraction of
  correctly aligned residues (either to another residue or to a gap) out of the
  total number of residues in all the sequences.
<div class="Pp"></div>
<b>amap</b> aligns sequences provided in MFA format. This format consists of
  multiple sequences. Each sequence in MFA format begins with a single-line
  description, followed by lines of sequence data. The description line is
  distinguished from the sequence data by a greater-than (&#x201C;&gt;&#x201D;)
  symbol in the first column.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<b>-clustalw</b>
<div style="margin-left: 4.00ex;">use CLUSTALW output format instead of
  MFA</div>
<div class="Pp"></div>
<b>-c</b> <b>--consistency</b> <i>REPS</i>
<div style="margin-left: 4.00ex;">use 0 &lt;= <i>REPS</i> &lt;= 5 (default: 0)
  passes of consistency transformation</div>
<div class="Pp"></div>
<b>-ir</b> <b>--iterative-refinement</b> <i>REPS</i>
<div style="margin-left: 4.00ex;">use 0 &lt;= <i>REPS</i> &lt;=1000 (default: 0)
  passes of iterative-refinement</div>
<div class="Pp"></div>
<b>-pre</b> <b>--pre-training</b> <i>REPS</i>
<div style="margin-left: 4.00ex;">use 0 &lt;= <i>REPS</i> &lt;= 20 (default: 0)
  rounds of pretraining</div>
<div class="Pp"></div>
<b>-pairs</b>
<div style="margin-left: 4.00ex;">generate all-pairs pairwise alignments</div>
<div class="Pp"></div>
<b>-viterbi</b>
<div style="margin-left: 4.00ex;">use Viterbi algorithm to generate all pairs
  (automatically enables <b>-pairs</b>)</div>
<div class="Pp"></div>
<b>-v</b> <b>--verbose</b>
<div style="margin-left: 4.00ex;">Report progress while aligning (default:
  off)</div>
<div class="Pp"></div>
<b>-annot</b> <i>FILENAME</i>
<div style="margin-left: 4.00ex;">write annotation for multiple alignment to
  <i>FILENAME</i></div>
<div class="Pp"></div>
<b>-t</b> <b>--train</b> <i>FILENAME</i>
<div style="margin-left: 4.00ex;">compute EM transition probabilities, store in
  <i>FILENAME</i> (default: no training)</div>
<div class="Pp"></div>
<b>-e</b> <b>--emissions</b>
<div style="margin-left: 4.00ex;">also reestimate emission probabilities
  (default: off)</div>
<div class="Pp"></div>
<b>-p</b> <b>--paramfile</b> <i>FILENAME</i>
<div style="margin-left: 4.00ex;">read parameters from <i>FILENAME</i> (default:
  )</div>
<div class="Pp"></div>
<b>-a</b> <b>--alignment-order</b>
<div style="margin-left: 4.00ex;">print sequences in alignment order rather than
  input order (default: off)</div>
<div class="Pp"></div>
<b>-g</b> <b>--gap-factor</b> <i>GF</i>
<div style="margin-left: 4.00ex;">use <i>GF</i> as the gap-factor parameter, set
  to 0 for best sensitivity, higher values for better specificity (default:
  0.5)</div>
<div class="Pp"></div>
<b>-w</b> <b>--edge-weight-threshold</b> <i>W</i>
<div style="margin-left: 4.00ex;">stop the sequence annealing process when best
  edge has lower weight than <i>W</i>, set to 0 for best sensitivity, higher
  values for better specificity (default: 0)</div>
<div class="Pp"></div>
<b>-prog</b> <b>--progressive</b>
<div style="margin-left: 4.00ex;">use progressive alignment instead of sequence
  annealing alignment (default: off)</div>
<div class="Pp"></div>
<b>-noreorder</b> <b>--no-edge-reordering</b>
<div style="margin-left: 4.00ex;">disable reordering of edges during sequence
  annealing alignment (default: off)</div>
<div class="Pp"></div>
<b>-maxstep</b> <b>--use-max-stepsize</b>
<div style="margin-left: 4.00ex;">use maximum improvement step size instead of
  tGf edge ranking (default: off)</div>
<div class="Pp"></div>
<b>-print</b> <b>--print-posteriors</b>
<div style="margin-left: 4.00ex;">only print the posterior probability matrices
  (default: off)</div>
<div class="Pp"></div>
<b>-gui</b> <i>START</i> <i>STEP</i>
<div style="margin-left: 4.00ex;">print output for the AMAP Display Java based
  GUI (default: ) starting at weight <i>START</i> (default: infinity) with step
  size <i>STEP</i> (default: )</div>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
To run AMAP with the default options change to the <i>align</i> directory and
  type:
<div class="Pp"></div>
<b>% amap &lt;multi-fasta-file-name&gt;</b>
<div class="Pp"></div>
If no file name is provided the list of options are printed.
<div class="Pp"></div>
In order to use the AMAP Display run AMAP with the -gui option, and save the
  output to a file, then use the file as the input to AmapDisplay. For example,
  type:
<div class="Pp"></div>
<b>% align/amap -gui examples/BB12020.tfa &gt; examples/BB12020.tfa.out</b>
<div class="Pp"></div>
<b>% java -jar display/AmapDisplay.jar examples/BB12020.tfa.out</b>
<div class="Pp"></div>
(on Debian systems, the <i>examples</i> directory is in
  <i>/usr/share/doc/amap-align/examples</i>
<h1 class="Sh" title="Sh" id="NOTE"><a class="selflink" href="#NOTE">NOTE</a></h1>
In older versions ( &lt;&#x00A0;2.0-1) of the package for Debian(TM) systems,
  the <b>amap</b> command was renamed <b>amap-align</b> because there was
  already another tool called <b>amap</b> (which performs some computer network
  diagnostics). A symbolic link <b>amap-align</b> is still provided for upgrade
  purposes but will be removed in Debian releases posterior to Etch
  (Debian&#x00A0;4.0).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The current version of AMAP uses the PROBCONS 1.09 code base for some of the
  input/output procedures, and for the calculation of posterior probabilities
  (see PROBCONS.README in <i>/usr/share/doc/amap-align/</i>). Future releases
  might implement the algorithm using a new independent code base.
<div class="Pp"></div>
On Debian(TM) systems, <b>probcons</b>(1) is available in the probcons package.
<h1 class="Sh" title="Sh" id="REFERENCES"><a class="selflink" href="#REFERENCES">REFERENCES</a></h1>
For more details on AMAP and AMA, see Schwartz, Ariel S., Myers, Eugene W., and
  Pachter, Lior. Alignment Metric Accuracy (Submitted for publication). For more
  details on sequence-annealing, see Schwartz, Ariel S. and Pachter, Lior.
  Multiple Alignment by Sequence Annealing (Submitted for publication).
<div class="Pp"></div>
PROBCONS was published in Do, C.B., Mahabhashyam, M.S.P., Brudno, M., and
  Batzoglou, S. 2005. PROBCONS: Probabilistic Consistency-based Multiple
  Sequence Alignment. Genome Research 15: 330-340.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<b>Ariel Schwartz</b> &lt;sariel@cs.berkeley.edu&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Upstream author of AMAP</dd>
</dl>
<div class="Pp"></div>
<b>Chuong Do</b>
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Wrote Probcons, on which AMAP is based.</dd>
</dl>
<div class="Pp"></div>
<b>Charles Plessy&gt;</b> &lt;charles-debian-nospam@plessy.org&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Wrote this manpage in DocBook XML for the Debian
      distribution.</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
AMAP, PROBCONS, and this manual page have been made freely available as PUBLIC
  DOMAIN software and hence are not subject to copyright in the United States.
  This system and/or any portion of the source code may be used, modified, or
  redistributed without restrictions. AMAP, PROBCONS and this manual page are
  distributed WITHOUT WARRANTY, express or implied. The authors accept NO LEGAL
  LIABILITY OR RESPONSIBILITY for loss due to reliance on the program.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">03/31/2008</td>
    <td class="foot-os">amap 2.2</td>
  </tr>
</table>
</body>
</html>
