<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:48:53 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>AMAP(1) User Manual AMAP(1)</p>

<p style="margin-top: 1em">NAME <br>
amap - Protein multiple alignment by sequence annealing</p>

<p style="margin-top: 1em">SYNOPSIS <br>
amap [OPTION] [MFAFILE] [MFAFILE]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
AMAP is a tool to perform multiple alignment of peptidic
sequences. It utilizes posterior decoding, and a
sequence-annealing alignment, instead of the traditional
progressive <br>
alignment method. It is the only alignment program that
allows one to control the sensitivity / specificity
tradeoff. It is based on the ProbCons source code, but uses
alignment <br>
metric accuracy and eliminates the consistency
transformation.</p>

<p style="margin-top: 1em">In its default configuration,
AMAP is tuned to maximize the expected Alignment Metric
Accuracy (AMA) score - a new alignment accuracy measure,
based on a metric for the <br>
multiple-alignment space, which integrates sensitivity and
specificity into a single balanced measure. AMA is defined
as the fraction of correctly aligned residues (either to
<br>
another residue or to a gap) out of the total number of
residues in all the sequences.</p>

<p style="margin-top: 1em">amap aligns sequences provided
in MFA format. This format consists of multiple sequences.
Each sequence in MFA format begins with a single-line
description, followed by lines of <br>
sequence data. The description line is distinguished from
the sequence data by a greater-than (&acirc;&gt;&acirc;)
symbol in the first column.</p>

<p style="margin-top: 1em">OPTIONS <br>
-clustalw <br>
use CLUSTALW output format instead of MFA</p>

<p style="margin-top: 1em">-c --consistency REPS <br>
use 0 &lt;= REPS &lt;= 5 (default: 0) passes of consistency
transformation</p>

<p style="margin-top: 1em">-ir --iterative-refinement REPS
<br>
use 0 &lt;= REPS &lt;=1000 (default: 0) passes of
iterative-refinement</p>

<p style="margin-top: 1em">-pre --pre-training REPS <br>
use 0 &lt;= REPS &lt;= 20 (default: 0) rounds of
pretraining</p>

<p style="margin-top: 1em">-pairs <br>
generate all-pairs pairwise alignments</p>

<p style="margin-top: 1em">-viterbi <br>
use Viterbi algorithm to generate all pairs (automatically
enables -pairs)</p>

<p style="margin-top: 1em">-v --verbose <br>
Report progress while aligning (default: off)</p>

<p style="margin-top: 1em">-annot FILENAME <br>
write annotation for multiple alignment to FILENAME</p>

<p style="margin-top: 1em">-t --train FILENAME <br>
compute EM transition probabilities, store in FILENAME
(default: no training)</p>

<p style="margin-top: 1em">-e --emissions <br>
also reestimate emission probabilities (default: off)</p>

<p style="margin-top: 1em">-p --paramfile FILENAME <br>
read parameters from FILENAME (default: )</p>

<p style="margin-top: 1em">-a --alignment-order <br>
print sequences in alignment order rather than input order
(default: off)</p>

<p style="margin-top: 1em">-g --gap-factor GF <br>
use GF as the gap-factor parameter, set to 0 for best
sensitivity, higher values for better specificity (default:
0.5)</p>

<p style="margin-top: 1em">-w --edge-weight-threshold W
<br>
stop the sequence annealing process when best edge has lower
weight than W, set to 0 for best sensitivity, higher values
for better specificity (default: 0)</p>

<p style="margin-top: 1em">-prog --progressive <br>
use progressive alignment instead of sequence annealing
alignment (default: off)</p>

<p style="margin-top: 1em">-noreorder --no-edge-reordering
<br>
disable reordering of edges during sequence annealing
alignment (default: off)</p>

<p style="margin-top: 1em">-maxstep --use-max-stepsize <br>
use maximum improvement step size instead of tGf edge
ranking (default: off)</p>

<p style="margin-top: 1em">-print --print-posteriors <br>
only print the posterior probability matrices (default:
off)</p>

<p style="margin-top: 1em">-gui START STEP <br>
print output for the AMAP Display Java based GUI (default: )
starting at weight START (default: infinity) with step size
STEP (default: )</p>

<p style="margin-top: 1em">EXAMPLES <br>
To run AMAP with the default options change to the align
directory and type:</p>

<p style="margin-top: 1em">% amap
&lt;multi-fasta-file-name&gt;</p>

<p style="margin-top: 1em">If no file name is provided the
list of options are printed.</p>

<p style="margin-top: 1em">In order to use the AMAP Display
run AMAP with the -gui option, and save the output to a
file, then use the file as the input to AmapDisplay. For
example, type:</p>

<p style="margin-top: 1em">% align/amap -gui
examples/BB12020.tfa &gt; examples/BB12020.tfa.out</p>

<p style="margin-top: 1em">% java -jar
display/AmapDisplay.jar examples/BB12020.tfa.out</p>

<p style="margin-top: 1em">(on Debian systems, the examples
directory is in /usr/share/doc/amap-align/examples</p>

<p style="margin-top: 1em">NOTE <br>
In older versions ( &lt; 2.0-1) of the package for
Debian(TM) systems, the amap command was renamed amap-align
because there was already another tool called amap (which
performs <br>
some computer network diagnostics). A symbolic link
amap-align is still provided for upgrade purposes but will
be removed in Debian releases posterior to Etch (Debian
4.0).</p>

<p style="margin-top: 1em">SEE ALSO <br>
The current version of AMAP uses the PROBCONS 1.09 code base
for some of the input/output procedures, and for the
calculation of posterior probabilities (see PROBCONS.README
in <br>
/usr/share/doc/amap-align/). Future releases might implement
the algorithm using a new independent code base.</p>

<p style="margin-top: 1em">On Debian(TM) systems,
probcons(1) is available in the probcons package.</p>

<p style="margin-top: 1em">REFERENCES <br>
For more details on AMAP and AMA, see Schwartz, Ariel S.,
Myers, Eugene W., and Pachter, Lior. Alignment Metric
Accuracy (Submitted for publication). For more details on
<br>
sequence-annealing, see Schwartz, Ariel S. and Pachter,
Lior. Multiple Alignment by Sequence Annealing (Submitted
for publication).</p>

<p style="margin-top: 1em">PROBCONS was published in Do,
C.B., Mahabhashyam, M.S.P., Brudno, M., and Batzoglou, S.
2005. PROBCONS: Probabilistic Consistency-based Multiple
Sequence Alignment. Genome <br>
Research 15: 330-340.</p>

<p style="margin-top: 1em">AUTHORS <br>
Ariel Schwartz &lt;sariel@cs.berkeley.edu&gt; <br>
Upstream author of AMAP</p>

<p style="margin-top: 1em">Chuong Do <br>
Wrote Probcons, on which AMAP is based.</p>

<p style="margin-top: 1em">Charles Plessy&gt;
&lt;charles-debian-nospam@plessy.org&gt; <br>
Wrote this manpage in DocBook XML for the Debian
distribution.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
AMAP, PROBCONS, and this manual page have been made freely
available as PUBLIC DOMAIN software and hence are not
subject to copyright in the United States. This system
and/or any <br>
portion of the source code may be used, modified, or
redistributed without restrictions. AMAP, PROBCONS and this
manual page are distributed WITHOUT WARRANTY, express or
implied. <br>
The authors accept NO LEGAL LIABILITY OR RESPONSIBILITY for
loss due to reliance on the program.</p>

<p style="margin-top: 1em">amap 2.2 03/31/2008 AMAP(1)</p>
<hr>
</body>
</html>
