<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:49:00 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>AMPLICONNOISE(1) AmpliconNoise Documentation
AMPLICONNOISE(1)</p>

<p style="margin-top: 1em">NAME <br>
AmpliconNoise - remove noise from high throughput nucleotide
sequence data</p>

<p style="margin-top: 1em">VERSION <br>
This documentation refers to version 1.22</p>

<p style="margin-top: 1em">SYNOPSIS <br>
See /usr/share/doc/ampliconnoise/Doc.pdf.gz for details of
how to run.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The following tools are included. Most of them have an MPI
equivalent, for example SeqNoise has an equivalent SeqNoiseM
which can be used with mpirun.</p>

<p style="margin-top: 1em">FastaUnique - dereplicates fasta
file <br>
-in string input file name <br>
Options:</p>

<p style="margin-top: 1em">FCluster <br>
-in string distance input file name <br>
-out string output file stub <br>
Options: <br>
-r resolution <br>
-a average linkage <br>
-w use weights <br>
-i read identifiers <br>
-s scale dist.</p>

<p style="margin-top: 1em">NDist - pairwise
Needleman-Wunsch sequence distance matrix from a fasta file
<br>
-in string fata file name <br>
Options: <br>
-i output identifiers</p>

<p style="margin-top: 1em">Perseus - slays monsters <br>
-sin string seq file name <br>
Options: <br>
-tin string reference sequence file <br>
-a output alignments <br>
-d use imbalance <br>
-rin string lookup file name</p>

<p style="margin-top: 1em">PyroDist - pairwise distance
matrix from flowgrams <br>
-in string flow file name <br>
-out stub out file stub <br>
Options: <br>
-ni no index in dat file <br>
-rin string lookup file name</p>

<p style="margin-top: 1em">PyroNoise - clusters flowgrams
without alignments <br>
-din string flow file name <br>
-out string cluster input file name <br>
-lin string list file <br>
Options: <br>
-v verbose <br>
-c double initial cut-off <br>
-ni no index in dat file <br>
-s double precision <br>
-rin file lookup file name</p>

<p style="margin-top: 1em">SeqDist - pairwise distance
matrix from a fasta file <br>
-in string fasta file name <br>
Options: <br>
-i output identifiers <br>
-rin string lookup file name</p>

<p style="margin-top: 1em">SeqNoise - clusters sequences
<br>
-in string sequence file name <br>
-din string distance matrix file name <br>
-out string cluster input file name <br>
-lin string list file <br>
Options: <br>
-min mapping file <br>
-v verbose <br>
-c double initial cut-off <br>
-s double precision <br>
-rin string lookup file name</p>

<p style="margin-top: 1em">SplitClusterEven <br>
-din string dat filename <br>
-min string map filename <br>
-tin string tree filename <br>
-s split size <br>
-m min size</p>

<p style="margin-top: 1em">AUTHOR <br>
All software by Chris Quince (quince@civil.gla.ac.uk) This
manpage by Tim Booth (tbooth@ceh.ac.uk)</p>

<p style="margin-top: 1em">LICENCE AND COPYRIGHT <br>
Copyright (c) 2009 (quince@civil.gla.ac.uk). All rights
reserved.</p>

<p style="margin-top: 1em">Released under the Lesser
GPL.</p>

<p style="margin-top: 1em">Permission is granted for anyone
to copy, use, or modify these programs and documents for
purposes of research or education, provided this copyright
notice is retained, and note <br>
is made of any changes that have been made.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-04-12
AMPLICONNOISE(1)</p>
<hr>
</body>
</html>
