<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>AMPLICONNOISE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">AMPLICONNOISE(1)</td>
    <td class="head-vol">AmpliconNoise Documentation</td>
    <td class="head-rtitle">AMPLICONNOISE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
AmpliconNoise - remove noise from high throughput nucleotide sequence data
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This documentation refers to version 1.22
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
See <b>/usr/share/doc/ampliconnoise/Doc.pdf.gz</b> for details of how to run.
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
The following tools are included. Most of them have an MPI equivalent, for
  example SeqNoise has an equivalent SeqNoiseM which can be used with mpirun.
<div class="Pp"></div>
<b>FastaUnique</b> - dereplicates fasta file
<br/>
 -in string input file name
<br/>
 Options:
<div class="Pp"></div>
<b>FCluster</b>
<br/>
 -in string distance input file name
<br/>
 -out string output file stub
<br/>
 Options:
<br/>
 -r resolution
<br/>
 -a average linkage
<br/>
 -w use weights
<br/>
 -i read identifiers
<br/>
 -s scale dist.
<div class="Pp"></div>
<b>NDist</b> - pairwise Needleman-Wunsch sequence distance matrix from a fasta
  file
<br/>
 -in string fata file name
<br/>
 Options:
<br/>
 -i output identifiers
<div class="Pp"></div>
<b>Perseus</b> - slays monsters
<br/>
 -sin string seq file name
<br/>
 Options:
<br/>
 -tin string reference sequence file
<br/>
 -a output alignments
<br/>
 -d use imbalance
<br/>
 -rin string lookup file name
<div class="Pp"></div>
<b>PyroDist</b> - pairwise distance matrix from flowgrams
<br/>
 -in string flow file name
<br/>
 -out stub out file stub
<br/>
 Options:
<br/>
 -ni no index in dat file
<br/>
 -rin string lookup file name
<div class="Pp"></div>
<b>PyroNoise</b> - clusters flowgrams without alignments
<br/>
 -din string flow file name
<br/>
 -out string cluster input file name
<br/>
 -lin string list file
<br/>
 Options:
<br/>
 -v verbose
<br/>
 -c double initial cut-off
<br/>
 -ni no index in dat file
<br/>
 -s double precision
<br/>
 -rin file lookup file name
<div class="Pp"></div>
<b>SeqDist</b> - pairwise distance matrix from a fasta file
<br/>
 -in string fasta file name
<br/>
 Options:
<br/>
 -i output identifiers
<br/>
 -rin string lookup file name
<div class="Pp"></div>
<b>SeqNoise</b> - clusters sequences
<br/>
 -in string sequence file name
<br/>
 -din string distance matrix file name
<br/>
 -out string cluster input file name
<br/>
 -lin string list file
<br/>
 Options:
<br/>
 -min mapping file
<br/>
 -v verbose
<br/>
 -c double initial cut-off
<br/>
 -s double precision
<br/>
 -rin string lookup file name
<div class="Pp"></div>
<b>SplitClusterEven</b>
<br/>
 -din string dat filename
<br/>
 -min string map filename
<br/>
 -tin string tree filename
<br/>
 -s split size
<br/>
 -m min size
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
All software by Chris Quince (quince@civil.gla.ac.uk) This manpage by Tim Booth
  (tbooth@ceh.ac.uk)
<h1 class="Sh" title="Sh" id="LICENCE_AND_COPYRIGHT"><a class="selflink" href="#LICENCE_AND_COPYRIGHT">LICENCE
  AND COPYRIGHT</a></h1>
Copyright (c) 2009 (quince@civil.gla.ac.uk). All rights reserved.
<div class="Pp"></div>
Released under the Lesser GPL.
<div class="Pp"></div>
Permission is granted for anyone to copy, use, or modify these programs and
  documents for purposes of research or education, provided this copyright
  notice is retained, and note is made of any changes that have been made.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-04-12</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
