<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ANDI(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ANDI(1)</td>
    <td class="head-vol">andi manual</td>
    <td class="head-rtitle">ANDI(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
andi - estimates evolutionary distance
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>andi</b> [ <i>-jlv</i>] [<i>-b INT</i>] [<i>-p FLOAT</i>] [<i>-m MODEL</i>]
  [<i>-t INT</i>] <i>FILES</i>...
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>andi</b> estimates the evolutionary distance between
    closely related genomes. For this <b>andi</b> reads the input sequences from
    <i>FASTA</i> files and computes the pairwise anchor distance. The idea
    behind this is explained in a paper by Haubold et al. (see below).</dt>
  <dd class="It-tag"></dd>
</dl>
<h1 class="Sh" title="Sh" id="OUTPUT"><a class="selflink" href="#OUTPUT">OUTPUT</a></h1>
The output is a symmetrical distance matrix in <i>PHYLIP</i> format, with each
  entry representing divergence with a positive real number. A distance of zero
  means that two sequences are identical, whereas other values are estimates for
  the nucleotide substitution rate (Jukes-Cantor corrected). For technical
  reasons the comparison might fail and no estimate can be computed. In such
  cases <i>nan</i> is printed. This either means that the input sequences were
  too short (&lt;200bp) or too diverse (K&gt;0.5) for our method to work
  properly.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b>, <b>--bootstrap</b> &lt;INT&gt;</dt>
  <dd class="It-tag">Compute multiple distance matrices, with <i>n-1</i>
      bootstrapped from the first. See the paper Kl&#x00F6;tzl &amp; Haubold
      (2016, in review) for a detailed explanation.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-j</b>, <b>--join</b></dt>
  <dd class="It-tag">Use this mode if each of your <i>FASTA</i> files represents
      one assembly with numerous contigs. <b>andi</b> will then treat all of the
      contained sequences per file as a single genome. In this mode at least one
      filename must be provided via command line arguments. For the output the
      filename is used to identify each sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b>, <b>--low-memory</b></dt>
  <dd class="It-tag">In multithreaded mode, <b>andi</b> requires memory linear
      to the amount of threads. The low memory mode changes this to a constant
      demand independent from the used number of threads. Unfortunately, this
      comes at a significant runtime cost.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>, <b>--model</b> &lt;Raw|JC|Kimura&gt;</dt>
  <dd class="It-tag">Different models of nucleotide evolution are supported. By
      default the Jukes-Cantor correction is used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> &lt;FLOAT&gt;</dt>
  <dd class="It-tag">Significance of an anchor pair; default: 0.05.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>, <b>--threads</b> &lt;INT&gt;</dt>
  <dd class="It-tag">The number of threads to be used; by default, all available
      processors are used.
    <div>&#x00A0;</div>
    Multithreading is only available if <b>andi</b> was compiled with OpenMP
      support.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--verbose</b></dt>
  <dd class="It-tag">Prints additional information. Apply multiple times for
      extra verboseness.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">Prints the synopsis and an explanation of available
      options.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">Outputs version information and acknowledgments.</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright &#x00A9; 2014 - 2016 Fabian Kl&#x00F6;tzl License GPLv3+: GNU GPL
  version 3 or later.
<div>&#x00A0;</div>
This is free software: you are free to change and redistribute it. There is NO
  WARRANTY, to the extent permitted by law. The full license text is available
  at &lt;http://gnu.org/licenses/gpl.html&gt;.
<h1 class="Sh" title="Sh" id="ACKNOWLEDGMENTS"><a class="selflink" href="#ACKNOWLEDGMENTS">ACKNOWLEDGMENTS</a></h1>
1) andi: Haubold, B. Kl&#x00F6;tzl, F. and Pfaffelhuber, P. (2015). andi: Fast
  and accurate estimation of evolutionary distances between closely related
  genomes
<div>&#x00A0;</div>
2) Algorithms: Ohlebusch, E. (2013). Bioinformatics Algorithms. Sequence
  Analysis, Genome Rearrangements, and Phylogenetic Reconstruction. pp 118f.
<div>&#x00A0;</div>
3) SA construction: Mori, Y. (2005). Short description of improved two-stage
  suffix sorting algorithm.
  http://homepage3.nifty.com/wpage/software/itssort.txt
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
<h2 class="Ss" title="Ss" id="Reporting_Bugs"><a class="selflink" href="#Reporting_Bugs">Reporting
  Bugs</a></h2>
Please report bugs to &lt;kloetzl@evolbio.mpg.de&gt; or at
  &lt;https://github.com/EvolBioInf/andi&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-03-05</td>
    <td class="foot-os">0.10</td>
  </tr>
</table>
</body>
</html>
