<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:49:07 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ANDI(1) andi manual ANDI(1)</p>

<p style="margin-top: 1em">NAME <br>
andi - estimates evolutionary distance</p>

<p style="margin-top: 1em">SYNOPSIS <br>
andi [-jlv] [-b INT] [-p FLOAT] [-m MODEL] [-t INT]
FILES...</p>

<p style="margin-top: 1em">DESCRIPTION <br>
andi estimates the evolutionary distance between closely
related genomes. For this andi reads the input sequences
from FASTA files and computes the pairwise anchor distance.
The <br>
idea behind this is explained in a paper by Haubold et al.
(see below).</p>

<p style="margin-top: 1em">OUTPUT <br>
The output is a symmetrical distance matrix in PHYLIP
format, with each entry representing divergence with a
positive real number. A distance of zero means that two
sequences are <br>
identical, whereas other values are estimates for the
nucleotide substitution rate (Jukes-Cantor corrected). For
technical reasons the comparison might fail and no estimate
can <br>
be computed. In such cases nan is printed. This either means
that the input sequences were too short (&lt;200bp) or too
diverse (K&gt;0.5) for our method to work properly.</p>

<p style="margin-top: 1em">OPTIONS <br>
-b, --bootstrap &lt;INT&gt; <br>
Compute multiple distance matrices, with n-1 bootstrapped
from the first. See the paper Kl&Atilde;&para;tzl &amp;
Haubold (2016, in review) for a detailed explanation.</p>

<p style="margin-top: 1em">-j, --join <br>
Use this mode if each of your FASTA files represents one
assembly with numerous contigs. andi will then treat all of
the contained sequences per file as a single genome. <br>
In this mode at least one filename must be provided via
command line arguments. For the output the filename is used
to identify each sequence.</p>

<p style="margin-top: 1em">-l, --low-memory <br>
In multithreaded mode, andi requires memory linear to the
amount of threads. The low memory mode changes this to a
constant demand independent from the used number of <br>
threads. Unfortunately, this comes at a significant runtime
cost.</p>

<p style="margin-top: 1em">-m, --model
&lt;Raw|JC|Kimura&gt; <br>
Different models of nucleotide evolution are supported. By
default the Jukes-Cantor correction is used.</p>

<p style="margin-top: 1em">-p &lt;FLOAT&gt; <br>
Significance of an anchor pair; default: 0.05.</p>

<p style="margin-top: 1em">-t, --threads &lt;INT&gt; <br>
The number of threads to be used; by default, all available
processors are used. <br>
Multithreading is only available if andi was compiled with
OpenMP support.</p>

<p style="margin-top: 1em">-v, --verbose <br>
Prints additional information. Apply multiple times for
extra verboseness.</p>

<p style="margin-top: 1em">-h, --help <br>
Prints the synopsis and an explanation of available
options.</p>

<p style="margin-top: 1em">--version <br>
Outputs version information and acknowledgments.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 2014 - 2016 Fabian
Kl&Atilde;&para;tzl License GPLv3+: GNU GPL version 3 or
later. <br>
This is free software: you are free to change and
redistribute it. There is NO WARRANTY, to the extent
permitted by law. The full license text is available at <br>
&lt;http://gnu.org/licenses/gpl.html&gt;.</p>

<p style="margin-top: 1em">ACKNOWLEDGMENTS <br>
1) andi: Haubold, B. Kl&Atilde;&para;tzl, F. and
Pfaffelhuber, P. (2015). andi: Fast and accurate estimation
of evolutionary distances between closely related genomes
<br>
2) Algorithms: Ohlebusch, E. (2013). Bioinformatics
Algorithms. Sequence Analysis, Genome Rearrangements, and
Phylogenetic Reconstruction. pp 118f. <br>
3) SA construction: Mori, Y. (2005). Short description of
improved two-stage suffix sorting algorithm.
http://homepage3.nifty.com/wpage/software/itssort.txt</p>

<p style="margin-top: 1em">BUGS <br>
Reporting Bugs <br>
Please report bugs to &lt;kloetzl@evolbio.mpg.de&gt; or at
&lt;https://github.com/EvolBioInf/andi&gt;.</p>

<p style="margin-top: 1em">0.10 2016-03-05 ANDI(1)</p>
<hr>
</body>
</html>
