<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ANFO(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ANFO(1)</td>
    <td class="head-vol">User Manuals</td>
    <td class="head-rtitle">ANFO(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
anfo - find best alignment of short reads to database
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>anfo [</b> <i>option</i> <b>|</b> <i>file</i> <b>... ]</b>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>anfo</b> aligns (short) sequencing reads to a (gigabase sized) database. It
  uses a heuristic seeding method, but then applies a genuine aligner that
  allows gaps, understands damage patterns in ancient dna and produces an easy
  to interpret score.
<div style="height: 1.00em;">&#x00A0;</div>
Input files can be any variety of FastA or FastQ files, or a native <b>anfo</b>
  binary file, optinally compressed using <b>gzip</b>or<b>bzip2</b>. The file
  format is automatically recognized and other formats may be added.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">-V, --version</dt>
  <dd class="It-tag">Print version number and exit.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-o file, --output file</dt>
  <dd class="It-tag">Write output to <i>file</i>. <i>file</i> will be written in
      the native <b>anfo</b> binary format, which can be operated upon using
      <b>anfo-tool</b> or the bindings to <b>guile</b>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-c conffile, --config conffile</dt>
  <dd class="It-tag">Read configuration from <i>conffile</i>. This file
      configures which indices are used, and by extension to which genomes an
      alignment is made, what parameters to use in the aligner and can set
      paths. See the example file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-p num, --threads num</dt>
  <dd class="It-tag">Start <i>num</i> threads for alignment. One such thread per
      processor core is usually best.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-x num, --ixthreads num</dt>
  <dd class="It-tag">Start <i>num</i> threads for indexing. Indexing normally
      uses less cpu power than alignment, so fewer indexers than aligners is
      normally best.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--solexa-scale</dt>
  <dd class="It-tag">When reading FastQ files, use the Solexa scale
      (log-odds-ratios) instead of the standard Phread scale (probabilities). If
      you use Solexa/Illumina sequencers, refer to your documentation whether
      you need this. Else you don't.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--fastq-origin ori</dt>
  <dd class="It-tag">Set origin for FastQ decoding to <i>ori</i>. The standard
      and default is 33, but it must be set to 64 for some versions of the
      Solexa/Illumina software. If you use Solexa/Illumina sequencers, refer to
      your documentation whether you need this. Else you don't.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-q, --quiet</dt>
  <dd class="It-tag">Suppress all output except fatal errors.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-v, --verbose</dt>
  <dd class="It-tag">Print a progress indicator during operation.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="ENVIRONMENT"><a class="selflink" href="#ENVIRONMENT">ENVIRONMENT</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">ANFO_PATH</dt>
  <dd class="It-tag">Colon separated list of directories searched for genome and
      index files.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="FILES"><a class="selflink" href="#FILES">FILES</a></h1>
<i>/etc/popt</i>
<div style="margin-left: 5.00ex;">The system wide configuration file for
  <b>popt</b>(3). <b>anfo</b> identifies itself as &quot;anfo&quot; to
  popt.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<i>~/.popt</i>
<div style="margin-left: 5.00ex;">Per user configuration file for
  <b>popt</b>(3).</div>
<div style="height: 1.00em;">&#x00A0;</div>
<i>/usr/share/doc/anfo/example/*.cfg</i>
<div style="margin-left: 5.00ex;">Example configuration files to be used with
  <b>anfo</b>.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
None known.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Udo Stenzel &lt;udo_stenzel@eva.mpg.de&gt;
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
http://bioinf.eva.mpg.de/anfo
<div style="height: 1.00em;">&#x00A0;</div>
<b>anfo-tool</b>(1), <b>anfo-sge</b>(1), <b>fa2dna</b>(1), <b>dnaindex</b>(1),
  <b>popt</b>(3), <b>fasta</b>(5)
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">OCTOBER 2009</td>
    <td class="foot-os">Applications</td>
  </tr>
</table>
</body>
</html>
