<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:49:08 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p style="margin-top: 1em">ANFO-TOOL(1) User Manuals
ANFO-TOOL(1)</p>

<p style="margin-top: 1em">NAME <br>
anfo-tool - process native ANFO binary files</p>

<p style="margin-top: 1em">SYNOPSIS <br>
anfo-tool [ option | pattern ... ]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
anfo-tool is used to filter, process and convert the files
created by anfo. Every pattern on the command line is
wildcard expanded, then for every input file (or the
standard <br>
input, if no pattern is given), anfo-tool builds a chain of
input filters, it then merges these input streams in one of
several ways, splits the result up into multiple output <br>
streams, each of which can have a chain of output filter
applied.</p>

<p style="margin-top: 1em">OPTIONS <br>
General Options <br>
These options apply globally and modify the behavior of the
whole program. They can be placed anywhere in the command
line.</p>

<p style="margin-top: 1em">-V, --version <br>
Print version number and exit.</p>

<p style="margin-top: 1em">-q, --quiet <br>
Suppress all output except fatal error messages.</p>

<p style="margin-top: 1em">-v, --verbose <br>
Produce more output, including progress indicators for most
operations.</p>

<p style="margin-top: 1em">--debug <br>
Produce debugging output in addition to progress
information.</p>

<p style="margin-top: 1em">-n, --dry-run <br>
Parse command line, optionally print a description of the
intended operations, then exit.</p>

<p style="margin-top: 1em">--vmem X <br>
Limit virtual memory to X megabytes. If memory runs out,
anfo-tool tries to free up memory by forgetting about big
files, e.g. genomes. Use this option to avoid swapping <br>
or out-of-memory conditions when operations involve big or
multiple genomes.</p>

<p style="margin-top: 1em">Setting Parameters <br>
A parameter can be set multiple times on the command line
and will overwrite previous settings. Any filter option that
needs a parameter picks up the last definition that <br>
appeared before the filter option.</p>

<p style="margin-top: 1em">--set-slope S <br>
Set the slope parameter to S. The slope is used together
with the intercept where filters apply to alignment scores;
alignments scoring no worse than slope * (length - <br>
intercept) are considered good. The default is 7.5.</p>

<p style="margin-top: 1em">--set-intercept L <br>
Set the intercept parameter to L. The intercept is used
together with the slope where filters apply to alignment
scores; alignments scoring no worse than slope * (length
<br>
- intercept) are considered good. The default is 20.</p>

<p style="margin-top: 1em">--set-context C <br>
Set the context parameter to C. The context is the number of
surrounding bases of the reference included when printing
alignments in text form. The default is 0.</p>

<p style="margin-top: 1em">--set-genome G <br>
Set the genome parameter to G. Many filters will only
consider the best alignments to this specific genome if it
is set. If no genome is set, the globally best alignment
<br>
is used.</p>

<p style="margin-top: 1em">--clear-genome <br>
Clear the genome parameter. Filters apply to the globally
best alignment afterwards.</p>

<p style="margin-top: 1em">Filter Options <br>
Filters can be applied before merging the inputs or after
splitting the back up.</p>

<p style="margin-top: 1em">-s, --sort-pos=n <br>
Sort by alignment position while buffering no more than n
MiB in memory. If a genome is set, alignments to that genome
are used.</p>

<p style="margin-top: 1em">-S, --sort-name=n <br>
Sort by read name while buffering no more than n MiB in
memory.</p>

<p style="margin-top: 1em">-l, --filter-length=L <br>
Retain alignments only for reads of at least L bases length.
The reads themselves are kept.</p>

<p style="margin-top: 1em">-f, --filter-score <br>
Retain alignments only if their score is good enough.
Usesslopeandintercept.</p>

<p style="margin-top: 1em">--filter-mapq=Q <br>
Remove alignments with mapping quality below Q.</p>

<p style="margin-top: 1em">-h, --filter-hit=SEQ <br>
Keep only reads that have a hit to a sequence named SEQ. If
SEQ is empty, reads are kept if they have any hit. If the
genome parameter is set, only hits to that genome <br>
count.</p>

<p style="margin-top: 1em">--delete-hit=SEQ <br>
Delete alignments to SEQ. If SEQ is empty, all alignments
are deleted. If the genome parameter is set, only alignments
to that genome are deleted.</p>

<p style="margin-top: 1em">--filter-qual=Q <br>
Mask out bases with quality below Q. Such a base is replaced
by the N ambiguity code.</p>

<p style="margin-top: 1em">--multiplicity=N <br>
Keep only reads of molecules that have been sequenced at
least N times. Reads are considered to come from the same
original molecule if their aligned coordinates are <br>
identical.</p>

<p style="margin-top: 1em">--subsample=F <br>
Subsample a fraction F of the results. Every read is
independently and randomly choosen to be kept or not.</p>

<p style="margin-top: 1em">--inside-regions=FILE <br>
Read a list of regions from FILE, then keep only alignments
that overlap an annotated region.</p>

<p style="margin-top: 1em">--outside-regions=FILE <br>
Read a list of regions from FILE, then keep only alignments
that do not overlap an annotated region.</p>

<p style="margin-top: 1em">Special Filters <br>
-d, --rmdup=Q <br>
Remove PCR duplicates, clamp quality scores to Q. Two reads
are considered to be duplicates, if their aligned
coordinates are identical. If a genome is set, the best <br>
alignment to that genome is used, else the globally best
alignment. Both alignments must be good, as determined by
slopeandintercept. For a set of duplicates, a consen&acirc;
<br>
sus is called, generally increasing the quality scores. If a
resulting quality score exceeds Q, it is set to Q. This
filter requires the input to be sorted by alignment <br>
coordinate on the selected genome.</p>

<p style="margin-top: 1em">--duct-tape=NAME Duct-tape
overlapping alignments into contigs and call a consensus for
them. If a genome is set, alignments to that genome are
used, else the globally <br>
best alignments. This filter requires input to be sorted by
alignment coordinate on the genome. Output is a set of
contigs, every position gets assigned a consensus <br>
base, a quality score and likelihoods for every possible
diallele. (It is called duct-taping because it kind of looks
like an assembly, but is not nearly as solid.)</p>

<p style="margin-top: 1em">--edit-header=ED <br>
Invoke the editor ED on the text representaion of the
stream&rsquo;s header. This can be used to clean up header
that have accumulated too much cruft.</p>

<p style="margin-top: 1em">Merging Filters <br>
Exactly one merging filter should be given on the command
line, all filter options occuring before that are part of
the input filter chains, all further filters become output
<br>
chains. If no merging filter is given, --concat is assumed,
and all filters are input filters.</p>

<p style="margin-top: 1em">-c, --concat <br>
Concatenate all input streams in the order they appear on
the command line.</p>

<p style="margin-top: 1em">-m, --merge <br>
Merge sorted input streams, producing a sorted result. All
inputs must be sorted in the same way.</p>

<p style="margin-top: 1em">-j, --join <br>
Join input streams and retain the single best hits to each
genome. Every input stream must contain a record for every
read, reads are buffered in memory until all of <br>
their hits are collected. This way, joining works well if
all inputs are nearly in the same order. If reads are
missing from some streams, joining them will waste
mem&acirc; <br>
ory.</p>

<p style="margin-top: 1em">--mega-merge <br>
Merge many streams such as those produced by running
anfo-sge. Streams that operated on the same reads are
joined, then everything is merged.</p>

<p style="margin-top: 1em">Output Options <br>
If an output option is given on the command line, the
current output filter chain is ended and a new one is
started. If no output option is given, a textual
representation of <br>
the final stream is written to stdout. All output options
accept - to write to stdout.</p>

<p style="margin-top: 1em">-o, --output FILE <br>
Write native binary stream (a compressed protobuf message)
to FILE. Writing a binary stream and reading it back in is
lossless.</p>

<p style="margin-top: 1em">--output-text FILE <br>
Write protobuf text stream to FILE. If the necessary genomes
are available, a textual representation of the alignments is
included. If the context parameter is set, that <br>
many additional bases of the reference upstream and
downstream from the alignment are included.</p>

<p style="margin-top: 1em">--output-sam=FILE <br>
Write alignments in SAM format to FILE.</p>

<p style="margin-top: 1em">--output-glz FILE <br>
Write contigs in GLZ 0.9 format to FILE. Generating GLZ only
works after application of --duct-tape, every contigs
becomes a GLZ record.</p>

<p style="margin-top: 1em">--output-3aln FILE <br>
Write contigs in a table based format to FILE. The format is
still subject to change, see the source code for detailed
documentation.</p>

<p style="margin-top: 1em">--output-fasta FILE <br>
Write alignments(!) in FastA format to FILE. Alignments are
writte as pair of reference and query sequence, aligned
coordinates are indicated in the description of the <br>
query sequence. If the context parameter is set, that many
additional bases of the reference upstream and downstream
from the alignment are included. This format is not <br>
suggested for any serious use, it exists to support legacy
applications.</p>

<p style="margin-top: 1em">--output-fastq FILE <br>
Write sequences(!) in FastQ format to FILE. Writing FastQ
effectively reconstitutes the input to ANFO if no filtering
was done on the results.</p>

<p style="margin-top: 1em">--output-table FILE <br>
Write per-alignment statistics to FILE. The file has three
colums:&Atilde;sequence length, alignment score, difference
to next best alignment. It is mainly useful to ana&acirc;
<br>
lyze/visualize the distribution of alignment scores.</p>

<p style="margin-top: 1em">--stats FILE <br>
Write simple statistics to FILE. This results in some simple
summary statistics of a whole stream: number of aligned
sequences, average length, GC content.</p>

<p style="margin-top: 1em">ENVIRONMENT <br>
ANFO_PATH <br>
Colon separated list of directories searched for genome and
index files.</p>

<p style="margin-top: 1em">ANFO_TEMP <br>
Temporary space used for sorting of large files.</p>

<p style="margin-top: 1em">FILES <br>
/etc/popt <br>
The system wide configuration file for popt(3). anfo-tool
identifies itself as &quot;anfo-tool&quot; to popt.</p>

<p style="margin-top: 1em">~/.popt <br>
Per user configuration file for popt(3).</p>

<p style="margin-top: 1em">BUGS <br>
The command line of this tools is way too complicated and
its semantics are counterintuitive. Using anfo-tool is
probably best avoided in most cases, the guile bindings
should <br>
provide a much more scalable and easier to understand
interface.</p>

<p style="margin-top: 1em">AUTHOR <br>
Udo Stenzel &lt;udo_stenzel@eva.mpg.de&gt;</p>

<p style="margin-top: 1em">SEE ALSO <br>
anfo(1), fa2dna(1) popt(3), fasta(5)</p>

<p style="margin-top: 1em">Applications OCTOBER 2009
ANFO-TOOL(1)</p>
<hr>
</body>
</html>
