<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:49:08 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p style="margin-top: 1em">ANFO(1) User Manuals ANFO(1)</p>

<p style="margin-top: 1em">NAME <br>
anfo - find best alignment of short reads to database</p>

<p style="margin-top: 1em">SYNOPSIS <br>
anfo [ option | file ... ]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
anfo aligns (short) sequencing reads to a (gigabase sized)
database. It uses a heuristic seeding method, but then
applies a genuine aligner that allows gaps, understands
damage <br>
patterns in ancient dna and produces an easy to interpret
score.</p>

<p style="margin-top: 1em">Input files can be any variety
of FastA or FastQ files, or a native anfo binary file,
optinally compressed using gziporbzip2. The file format is
automatically recognized and <br>
other formats may be added.</p>

<p style="margin-top: 1em">OPTIONS <br>
-V, --version <br>
Print version number and exit.</p>

<p style="margin-top: 1em">-o file, --output file <br>
Write output to file. file will be written in the native
anfo binary format, which can be operated upon using
anfo-tool or the bindings to guile.</p>

<p style="margin-top: 1em">-c conffile, --config conffile
<br>
Read configuration from conffile. This file configures which
indices are used, and by extension to which genomes an
alignment is made, what parameters to use in the <br>
aligner and can set paths. See the example file.</p>

<p style="margin-top: 1em">-p num, --threads num <br>
Start num threads for alignment. One such thread per
processor core is usually best.</p>

<p style="margin-top: 1em">-x num, --ixthreads num <br>
Start num threads for indexing. Indexing normally uses less
cpu power than alignment, so fewer indexers than aligners is
normally best.</p>

<p style="margin-top: 1em">--solexa-scale <br>
When reading FastQ files, use the Solexa scale
(log-odds-ratios) instead of the standard Phread scale
(probabilities). If you use Solexa/Illumina sequencers,
refer to <br>
your documentation whether you need this. Else you
don&rsquo;t.</p>

<p style="margin-top: 1em">--fastq-origin ori <br>
Set origin for FastQ decoding to ori. The standard and
default is 33, but it must be set to 64 for some versions of
the Solexa/Illumina software. If you use Solexa/Illu&acirc;
<br>
mina sequencers, refer to your documentation whether you
need this. Else you don&rsquo;t.</p>

<p style="margin-top: 1em">-q, --quiet <br>
Suppress all output except fatal errors.</p>

<p style="margin-top: 1em">-v, --verbose <br>
Print a progress indicator during operation.</p>

<p style="margin-top: 1em">ENVIRONMENT <br>
ANFO_PATH <br>
Colon separated list of directories searched for genome and
index files.</p>

<p style="margin-top: 1em">FILES <br>
/etc/popt <br>
The system wide configuration file for popt(3). anfo
identifies itself as &quot;anfo&quot; to popt.</p>

<p style="margin-top: 1em">~/.popt <br>
Per user configuration file for popt(3).</p>


<p style="margin-top: 1em">/usr/share/doc/anfo/example/*.cfg
<br>
Example configuration files to be used with anfo.</p>

<p style="margin-top: 1em">BUGS <br>
None known.</p>

<p style="margin-top: 1em">AUTHOR <br>
Udo Stenzel &lt;udo_stenzel@eva.mpg.de&gt;</p>

<p style="margin-top: 1em">SEE ALSO <br>
http://bioinf.eva.mpg.de/anfo</p>

<p style="margin-top: 1em">anfo-tool(1), anfo-sge(1),
fa2dna(1), dnaindex(1), popt(3), fasta(5)</p>

<p style="margin-top: 1em">Applications OCTOBER 2009
ANFO(1)</p>
<hr>
</body>
</html>
