<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ANFO-TOOL(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ANFO-TOOL(1)</td>
    <td class="head-vol">User Manuals</td>
    <td class="head-rtitle">ANFO-TOOL(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
anfo-tool - process native ANFO binary files
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>anfo-tool [</b> <i>option</i> <b>|</b> <i>pattern</i> <b>... ]</b>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>anfo-tool</b> is used to filter, process and convert the files created by
  <b>anfo</b>. Every pattern on the command line is wildcard expanded, then for
  every input file (or the standard input, if no pattern is given),
  <b>anfo-tool</b> builds a chain of input filters, it then merges these input
  streams in one of several ways, splits the result up into multiple output
  streams, each of which can have a chain of output filter applied.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="General_Options"><a class="selflink" href="#General_Options">General
  Options</a></h2>
These options apply globally and modify the behavior of the whole program. They
  can be placed anywhere in the command line.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">-V, --version</dt>
  <dd class="It-tag">Print version number and exit.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-q, --quiet</dt>
  <dd class="It-tag">Suppress all output except fatal error messages.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-v, --verbose</dt>
  <dd class="It-tag">Produce more output, including progress indicators for most
      operations.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--debug</dt>
  <dd class="It-tag">Produce debugging output in addition to progress
      information.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-n, --dry-run</dt>
  <dd class="It-tag">Parse command line, optionally print a description of the
      intended operations, then exit.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--vmem X</dt>
  <dd class="It-tag">Limit virtual memory to <i>X</i> megabytes. If memory runs
      out, <i>anfo-tool</i> tries to free up memory by forgetting about big
      files, e.g. genomes. Use this option to avoid swapping or out-of-memory
      conditions when operations involve big or multiple genomes.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Setting_Parameters"><a class="selflink" href="#Setting_Parameters">Setting
  Parameters</a></h2>
A parameter can be set multiple times on the command line and will overwrite
  previous settings. Any filter option that needs a parameter picks up the last
  definition that appeared before the filter option.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">--set-slope S</dt>
  <dd class="It-tag">Set the slope parameter to <i>S</i>. The <i>slope</i> is
      used together with the <i>intercept</i> where filters apply to alignment
      scores; alignments scoring no worse than <i>slope * (length -
      intercept)</i> are considered good. The default is 7.5.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--set-intercept L</dt>
  <dd class="It-tag">Set the intercept parameter to <i>L</i>. The
      <i>intercept</i> is used together with the <i>slope</i> where filters
      apply to alignment scores; alignments scoring no worse than <i>slope *
      (length - intercept)</i> are considered good. The default is 20.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--set-context C</dt>
  <dd class="It-tag">Set the context parameter to <i>C</i>. The context is the
      number of surrounding bases of the reference included when printing
      alignments in text form. The default is 0.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--set-genome G</dt>
  <dd class="It-tag">Set the genome parameter to <i>G</i>. Many filters will
      only consider the best alignments to this specific genome if it is set. If
      no genome is set, the globally best alignment is used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--clear-genome</dt>
  <dd class="It-tag">Clear the genome parameter. Filters apply to the globally
      best alignment afterwards.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Filter_Options"><a class="selflink" href="#Filter_Options">Filter
  Options</a></h2>
Filters can be applied before merging the inputs or after splitting the back up.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">-s, --sort-pos=n</dt>
  <dd class="It-tag">Sort by alignment position while buffering no more than
      <i>n</i> MiB in memory. If a genome is set, alignments to that genome are
      used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-S, --sort-name=n</dt>
  <dd class="It-tag">Sort by read name while buffering no more than <i>n</i> MiB
      in memory.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-l, --filter-length=L</dt>
  <dd class="It-tag">Retain alignments only for reads of at least <i>L</i> bases
      length. The reads themselves are kept.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-f, --filter-score</dt>
  <dd class="It-tag">Retain alignments only if their score is good enough.
      <i>Uses</i>slope<i>and</i>intercept<i>.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--filter-mapq=Q</dt>
  <dd class="It-tag">Remove alignments with mapping quality below <i>Q</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-h, --filter-hit=SEQ</dt>
  <dd class="It-tag">Keep only reads that have a hit to a sequence named
      <i>SEQ</i>. If <i>SEQ</i> is empty, reads are kept if they have any hit.
      If the <i>genome</i> parameter is set, only hits to that genome count.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--delete-hit=SEQ</dt>
  <dd class="It-tag">Delete alignments to <i>SEQ</i>. If <i>SEQ</i> is empty,
      all alignments are deleted. If the <i>genome</i> parameter is set, only
      alignments to that genome are deleted.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--filter-qual=Q</dt>
  <dd class="It-tag">Mask out bases with quality below <i>Q</i>. Such a base is
      replaced by the <i>N</i> ambiguity code.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--multiplicity=N</dt>
  <dd class="It-tag">Keep only reads of molecules that have been sequenced at
      least <i>N</i> times. Reads are considered to come from the same original
      molecule if their aligned coordinates are identical.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--subsample=F</dt>
  <dd class="It-tag">Subsample a fraction <i>F</i> of the results. Every read is
      independently and randomly choosen to be kept or not.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--inside-regions=FILE</dt>
  <dd class="It-tag">Read a list of regions from <i>FILE</i>, then keep only
      alignments that overlap an annotated region.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--outside-regions=FILE</dt>
  <dd class="It-tag">Read a list of regions from <i>FILE</i>, then keep only
      alignments that do not overlap an annotated region.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Special_Filters"><a class="selflink" href="#Special_Filters">Special
  Filters</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">-d, --rmdup=Q</dt>
  <dd class="It-tag">Remove PCR duplicates, clamp quality scores to <i>Q</i>.
      Two reads are considered to be duplicates, if their aligned coordinates
      are identical. If a <i>genome</i> is set, the best alignment to that
      genome is used, else the globally best alignment. Both alignments must be
      good, as determined by <i>slope</i>and<i>intercept</i>. For a set of
      duplicates, a consensus is called, generally increasing the quality
      scores. If a resulting quality score exceeds <i>Q</i>, it is set to
      <i>Q</i>. This filter requires the input to be sorted by alignment
      coordinate on the selected <i>genome</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    <i>--duct-tape=NAME</i> Duct-tape overlapping alignments into contigs and
      call a consensus for them. If a <i>genome</i> is set, alignments to that
      genome are used, else the globally best alignments. This filter requires
      input to be sorted by alignment coordinate on the genome. Output is a set
      of contigs, every position gets assigned a consensus base, a quality score
      and likelihoods for every possible diallele. (It is called duct-taping
      because it kind of looks like an assembly, but is not nearly as solid.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--edit-header=ED</dt>
  <dd class="It-tag">Invoke the editor <i>ED</i> on the text representaion of
      the stream's header. This can be used to clean up header that have
      accumulated too much cruft.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Merging_Filters"><a class="selflink" href="#Merging_Filters">Merging
  Filters</a></h2>
Exactly one merging filter should be given on the command line, all filter
  options occuring before that are part of the input filter chains, all further
  filters become output chains. If no merging filter is given, <i>--concat</i>
  is assumed, and all filters are input filters.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">-c, --concat</dt>
  <dd class="It-tag">Concatenate all input streams in the order they appear on
      the command line.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-m, --merge</dt>
  <dd class="It-tag">Merge sorted input streams, producing a sorted result. All
      inputs must be sorted in the same way.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-j, --join</dt>
  <dd class="It-tag">Join input streams and retain the single best hits to each
      genome. Every input stream must contain a record for every read, reads are
      buffered in memory until all of their hits are collected. This way,
      joining works well if all inputs are nearly in the same order. If reads
      are missing from some streams, joining them will waste memory.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--mega-merge</dt>
  <dd class="It-tag">Merge many streams such as those produced by running
      <b>anfo-sge</b>. Streams that operated on the same reads are joined, then
      everything is merged.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Output_Options"><a class="selflink" href="#Output_Options">Output
  Options</a></h2>
If an output option is given on the command line, the current output filter
  chain is ended and a new one is started. If no output option is given, a
  textual representation of the final stream is written to <i>stdout</i>. All
  output options accept <i>-</i> to write to stdout.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">-o, --output FILE</dt>
  <dd class="It-tag">Write native binary stream (a compressed protobuf message)
      to <i>FILE</i>. Writing a binary stream and reading it back in is
      lossless.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--output-text FILE</dt>
  <dd class="It-tag">Write protobuf text stream to <i>FILE</i>. If the necessary
      genomes are available, a textual representation of the alignments is
      included. If the <i>context</i> parameter is set, that many additional
      bases of the reference upstream and downstream from the alignment are
      included.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--output-sam=FILE</dt>
  <dd class="It-tag">Write alignments in SAM format to <i>FILE</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--output-glz FILE</dt>
  <dd class="It-tag">Write contigs in GLZ 0.9 format to <i>FILE</i>. Generating
      GLZ only works after application of <i>--duct-tape</i>, every contigs
      becomes a GLZ record.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--output-3aln FILE</dt>
  <dd class="It-tag">Write contigs in a table based format to <i>FILE</i>. The
      format is still subject to change, see the source code for detailed
      documentation.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--output-fasta FILE</dt>
  <dd class="It-tag">Write alignments(!) in FastA format to <i>FILE</i>.
      Alignments are writte as pair of reference and query sequence, aligned
      coordinates are indicated in the description of the query sequence. If the
      <i>context</i> parameter is set, that many additional bases of the
      reference upstream and downstream from the alignment are included. This
      format is not suggested for any serious use, it exists to support legacy
      applications.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--output-fastq FILE</dt>
  <dd class="It-tag">Write sequences(!) in FastQ format to <i>FILE</i>. Writing
      FastQ effectively reconstitutes the input to <b>ANFO</b> if no filtering
      was done on the results.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--output-table FILE</dt>
  <dd class="It-tag">Write per-alignment statistics to <i>FILE</i>. The file has
      three colums:&#x00C2;sequence length, alignment score, difference to next
      best alignment. It is mainly useful to analyze/visualize the distribution
      of alignment scores.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--stats FILE</dt>
  <dd class="It-tag">Write simple statistics to <i>FILE</i>. This results in
      some simple summary statistics of a whole stream: number of aligned
      sequences, average length, GC content.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="ENVIRONMENT"><a class="selflink" href="#ENVIRONMENT">ENVIRONMENT</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">ANFO_PATH</dt>
  <dd class="It-tag">Colon separated list of directories searched for genome and
      index files.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">ANFO_TEMP</dt>
  <dd class="It-tag">Temporary space used for sorting of large files.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="FILES"><a class="selflink" href="#FILES">FILES</a></h1>
<i>/etc/popt</i>
<div style="margin-left: 5.00ex;">The system wide configuration file for
  <b>popt</b>(3). <b>anfo-tool</b> identifies itself as &quot;anfo-tool&quot; to
  popt.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<i>~/.popt</i>
<div style="margin-left: 5.00ex;">Per user configuration file for
  <b>popt</b>(3).</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
The command line of this tools is way too complicated and its semantics are
  counterintuitive. Using <b>anfo-tool</b> is probably best avoided in most
  cases, the <b>guile</b> bindings should provide a much more scalable and
  easier to understand interface.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Udo Stenzel &lt;udo_stenzel@eva.mpg.de&gt;
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>anfo</b>(1), <b>fa2dna</b>(1) <b>popt</b>(3), <b>fasta</b>(5)
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">OCTOBER 2009</td>
    <td class="foot-os">Applications</td>
  </tr>
</table>
</body>
</html>
