<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ARAGORN(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ARAGORN(1)</td>
    <td class="head-vol">&#x00A0;</td>
    <td class="head-rtitle">ARAGORN(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
aragorn - detect tRNA genes in nucleotide sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>aragorn</b> [<i>OPTION</i>]... <i>FILE</i>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<b>-m</b>
<div style="margin-left: 4.00ex;">Search for tmRNA genes.</div>
<div class="Pp"></div>
<b>-t</b>
<div style="margin-left: 4.00ex;">Search for tRNA genes. By default, all are
  detected. If one of <b>-m</b> or <b>-t</b> is specified, then the other is not
  detected unless specified as well.</div>
<div class="Pp"></div>
<b>-mt</b>
<div style="margin-left: 4.00ex;">Search for Metazoan mitochondrial tRNA genes.
  tRNA genes with introns not detected. <b>-i</b>, <b>-sr</b> switches ignored.
  Composite Metazoan mitochondrial genetic code used.</div>
<div class="Pp"></div>
<b>-mtmam</b>
<div style="margin-left: 4.00ex;">Search for Mammalian mitochondrial tRNA genes.
  <b>-i</b>, <b>-sr</b> switches ignored. <b>-tv</b> switch set. Mammalian
  mitochondrial genetic code used.</div>
<div class="Pp"></div>
<b>-mtx</b>
<div style="margin-left: 4.00ex;">Same as <b>-mt</b> but low scoring tRNA genes
  are not reported.</div>
<div class="Pp"></div>
<b>-mtd</b>
<div style="margin-left: 4.00ex;">Overlapping metazoan mitochondrial tRNA genes
  on opposite strands are reported.</div>
<div class="Pp"></div>
<b>-gc</b>[<i>num</i>]
<div style="margin-left: 4.00ex;">Use the GenBank transl_table = [<i>num</i>]
  genetic code. Individual modifications can be appended using
  <i>,BBB</i>=&lt;aa&gt; B = A,C,G, or T. &lt;aa&gt; is the three letter code
  for an amino-acid. More than one modification can be specified. eg
  <b>-gcvert</b>,aga=Trp,agg=Trp uses the Vertebrate Mitochondrial code and the
  codons AGA and AGG changed to Tryptophan.</div>
<div class="Pp"></div>
<b>-gcstd</b>
<div style="margin-left: 4.00ex;">Use standard genetic code.</div>
<div class="Pp"></div>
<b>-gcmet</b>
<div style="margin-left: 4.00ex;">Use composite Metazoan mitochondrial genetic
  code.</div>
<div class="Pp"></div>
<b>-gcvert</b>
<div style="margin-left: 4.00ex;">Use Vertebrate mitochondrial genetic
  code.</div>
<div class="Pp"></div>
<b>-gcinvert</b>
<div style="margin-left: 4.00ex;">Use Invertebrate mitochondrial genetic
  code.</div>
<div class="Pp"></div>
<b>-gcyeast</b>
<div style="margin-left: 4.00ex;">Use Yeast mitochondrial genetic code.</div>
<div class="Pp"></div>
<b>-gcprot</b>
<div style="margin-left: 4.00ex;">Use Mold/Protozoan/Coelenterate mitochondrial
  genetic code.</div>
<div class="Pp"></div>
<b>-gcciliate</b>
<div style="margin-left: 4.00ex;">Use Ciliate genetic code.</div>
<div class="Pp"></div>
<b>-gcflatworm</b>
<div style="margin-left: 4.00ex;">Use Echinoderm/Flatworm mitochondrial genetic
  code</div>
<div class="Pp"></div>
<b>-gceuplot</b>
<div style="margin-left: 4.00ex;">Use Euplotid genetic code.</div>
<div class="Pp"></div>
<b>-gcbact</b>
<div style="margin-left: 4.00ex;">Use Bacterial/Plant Chloroplast genetic
  code.</div>
<div class="Pp"></div>
<b>-gcaltyeast</b>
<div style="margin-left: 4.00ex;">Use alternative Yeast genetic code.</div>
<div class="Pp"></div>
<b>-gcascid</b>
<div style="margin-left: 4.00ex;">Use Ascidian Mitochondrial genetic code.</div>
<div class="Pp"></div>
<b>-gcaltflat</b>
<div style="margin-left: 4.00ex;">Use alternative Flatworm Mitochondrial genetic
  code.</div>
<div class="Pp"></div>
<b>-gcblep</b>
<div style="margin-left: 4.00ex;">Use Blepharisma genetic code.</div>
<div class="Pp"></div>
<b>-gcchloroph</b>
<div style="margin-left: 4.00ex;">Use Chlorophycean Mitochondrial genetic
  code.</div>
<div class="Pp"></div>
<b>-gctrem</b>
<div style="margin-left: 4.00ex;">Use Trematode Mitochondrial genetic
  code.</div>
<div class="Pp"></div>
<b>-gcscen</b>
<div style="margin-left: 4.00ex;">Use Scenedesmus obliquus Mitochondrial genetic
  code.</div>
<div class="Pp"></div>
<b>-gcthraust</b>
<div style="margin-left: 4.00ex;">Use Thraustochytrium Mitochondrial genetic
  code.</div>
<div class="Pp"></div>
<b>-tv</b>
<div style="margin-left: 4.00ex;">Do not search for mitochondrial TV replacement
  loop tRNA genes. Only relevant if <b>-mt</b> used.</div>
<div class="Pp"></div>
<b>-c7</b>
<div style="margin-left: 4.00ex;">Search for tRNA genes with 7 base C-loops
  only.</div>
<div class="Pp"></div>
<b>-i</b>
<div style="margin-left: 4.00ex;">Search for tRNA genes with introns in
  anticodon loop with maximum length 3000 bases. Minimum intron length is 0
  bases. Ignored if <b>-m</b> is specified.</div>
<div class="Pp"></div>
<b>-i</b>[<i>max</i>]
<div style="margin-left: 4.00ex;">Search for tRNA genes with introns in
  anticodon loop with maximum length [ <i>max</i>] bases. Minimum intron length
  is 0 bases. Ignored if <b>-m</b> is specified.</div>
<div class="Pp"></div>
<b>-i</b>[<i>min</i>],[<i>max</i>]
<div style="margin-left: 4.00ex;">Search for tRNA genes with introns in
  anticodon loop with maximum length [ <i>max</i>] bases, and minimum length
  [<i>min</i>] bases. Ignored if <b>-m</b> is specified.</div>
<div class="Pp"></div>
<b>-io</b>
<div style="margin-left: 4.00ex;">Same as <b>-i</b>, but allow tRNA genes with
  long introns to overlap shorter tRNA genes.</div>
<div class="Pp"></div>
<b>-if</b>
<div style="margin-left: 4.00ex;">Same as <b>-i</b>, but fix intron between
  positions 37 and 38 on C-loop (one base after anticodon).</div>
<div class="Pp"></div>
<b>-ifo</b>
<div style="margin-left: 4.00ex;">Same as <b>-if</b> and <b>-io</b>
  combined.</div>
<div class="Pp"></div>
<b>-ir</b>
<div style="margin-left: 4.00ex;">Same as <b>-i</b>, but report tRNA genes with
  minimum length [ <i>min</i>] bases rather than search for tRNA genes with
  minimum length [ <i>min</i>] bases. With this switch, [<i>min</i>] acts as an
  output filter, minimum intron length for searching is still 0 bases.</div>
<div class="Pp"></div>
<b>-c</b>
<div style="margin-left: 4.00ex;">Assume that each sequence has a circular
  topology. Search wraps around each end. Default setting.</div>
<div class="Pp"></div>
<b>-l</b>
<div style="margin-left: 4.00ex;">Assume that each sequence has a linear
  topology. Search does not wrap.</div>
<div class="Pp"></div>
<b>-d</b>
<div style="margin-left: 4.00ex;">Double. Search both strands of each sequence.
  Default setting.</div>
<div class="Pp"></div>
<b>-s</b> or <b>-s+</b>
<div style="margin-left: 4.00ex;">Single. Do not search the complementary
  (antisense) strand of each sequence.</div>
<div class="Pp"></div>
<b>-sc</b> or <b>-s-</b>
<div style="margin-left: 4.00ex;">Single complementary. Do not search the sense
  strand of each sequence.</div>
<div class="Pp"></div>
<b>-ps</b>
<div style="margin-left: 4.00ex;">Lower scoring thresholds to 95% of default
  levels.</div>
<div class="Pp"></div>
<b>-ps</b>[<i>num</i>]
<div style="margin-left: 4.00ex;">Change scoring thresholds to [<i>num</i>]
  percent of default levels.</div>
<div class="Pp"></div>
<b>-rp</b>
<div style="margin-left: 4.00ex;">Flag possible pseudogenes (score &lt; 100 or
  tRNA anticodon loop &lt;&gt; 7 bases long). Note that genes with score &lt;
  100 will not be detected or flagged if scoring thresholds are not also changed
  to below 100% (see -ps switch).</div>
<div class="Pp"></div>
<b>-seq</b>
<div style="margin-left: 4.00ex;">Print out primary sequence.</div>
<div class="Pp"></div>
<b>-br</b>
<div style="margin-left: 4.00ex;">Show secondary structure of tRNA gene primary
  sequence using round brackets.</div>
<div class="Pp"></div>
<b>-fasta</b>
<div style="margin-left: 4.00ex;">Print out primary sequence in fasta
  format.</div>
<div class="Pp"></div>
<b>-fo</b>
<div style="margin-left: 4.00ex;">Print out primary sequence in fasta format
  only (no secondary structure).</div>
<div class="Pp"></div>
<b>-fon</b>
<div style="margin-left: 4.00ex;">Same as <b>-fo</b>, with sequence and gene
  numbering in header.</div>
<div class="Pp"></div>
<b>-fos</b>
<div style="margin-left: 4.00ex;">Same as <b>-fo</b>, with no spaces in
  header.</div>
<div class="Pp"></div>
<b>-fons</b>
<div style="margin-left: 4.00ex;">Same as <b>-fo</b>, with sequence and gene
  numbering, but no spaces.</div>
<div class="Pp"></div>
<b>-w</b>
<div style="margin-left: 4.00ex;">Print out in Batch mode.</div>
<div class="Pp"></div>
<b>-ss</b>
<div style="margin-left: 4.00ex;">Use the stricter canonical 1-2 bp spacer1 and
  1 bp spacer2. Ignored if <b>-mt</b> set. Default is to allow 3 bp spacer1 and
  0-2 bp spacer2, which may degrade selectivity.</div>
<div class="Pp"></div>
<b>-v</b>
<div style="margin-left: 4.00ex;">Verbose. Prints out information during search
  to STDERR.</div>
<div class="Pp"></div>
<b>-a</b>
<div style="margin-left: 4.00ex;">Print out tRNA domain for tmRNA genes.</div>
<div class="Pp"></div>
<b>-a7</b>
<div style="margin-left: 4.00ex;">Restrict tRNA astem length to a maximum of 7
  bases</div>
<div class="Pp"></div>
<b>-aa</b>
<div style="margin-left: 4.00ex;">Display message if predicted iso-acceptor
  species does not match species in sequence name (if present).</div>
<div class="Pp"></div>
<b>-j</b>
<div style="margin-left: 4.00ex;">Display 4-base sequence on 3' end of astem
  regardless of predicted amino-acyl acceptor length.</div>
<div class="Pp"></div>
<b>-jr</b>
<div style="margin-left: 4.00ex;">Allow some divergence of 3' amino-acyl
  acceptor sequence from NCCA.</div>
<div class="Pp"></div>
<b>-jr4</b>
<div style="margin-left: 4.00ex;">Allow some divergence of 3' amino-acyl
  acceptor sequence from NCCA, and display 4 bases.</div>
<div class="Pp"></div>
<b>-q</b>
<div style="margin-left: 4.00ex;">Do not print configuration line (which
  switches and files were used).</div>
<div class="Pp"></div>
<b>-rn</b>
<div style="margin-left: 4.00ex;">Repeat sequence name before summary
  information.</div>
<div class="Pp"></div>
<b>-O</b> [<i>outfile</i>]
<div style="margin-left: 4.00ex;">Print output to <i>. If ['outfile</i>] already
  exists, it is overwritten. By default all output goes to stdout.</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
aragorn detects tRNA, mtRNA, and tmRNA genes. A minimum requirement is at least
  a 32 bit compiler architecture (variable types int and unsigned int are at
  least 4 bytes long).
<div style="height: 1.00em;">&#x00A0;</div>
[ <i>FILE</i>] is assumed to contain one or more sequences in FASTA format.
  Results of the search are printed to STDOUT. All switches are optional and
  case-insensitive. Unless -i is specified, tRNA genes containing introns are
  not detected.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
Bjorn Canback &lt;bcanback@acgt.se&gt;, Dean Laslett &lt;gaiaquark@gmail.com&gt;
<h1 class="Sh" title="Sh" id="REFERENCES"><a class="selflink" href="#REFERENCES">REFERENCES</a></h1>
Laslett, D. and Canback, B. (2004) ARAGORN, a program for the detection of
  transfer RNA and transfer-messenger RNA genes in nucleotide sequences Nucleic
  Acids Research, 32;11-16
<div style="height: 1.00em;">&#x00A0;</div>
Laslett, D. and Canback, B. (2008) ARWEN: a program to detect tRNA genes in
  metazoan mitochondrial nucleotide sequences Bioinformatics, 24(2);
  172-175.</div>
<table class="foot">
  <tr>
    <td class="foot-date">02/24/2013</td>
    <td class="foot-os">&#x00A0;</td>
  </tr>
</table>
</body>
</html>
