<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:49:58 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ARAGORN(1) ARAGORN(1)</p>

<p style="margin-top: 1em">NAME <br>
aragorn - detect tRNA genes in nucleotide sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
aragorn [OPTION]... FILE</p>

<p style="margin-top: 1em">OPTIONS <br>
-m <br>
Search for tmRNA genes.</p>

<p style="margin-top: 1em">-t <br>
Search for tRNA genes. By default, all are detected. If one
of -m or -t is specified, then the other is not detected
unless specified as well.</p>

<p style="margin-top: 1em">-mt <br>
Search for Metazoan mitochondrial tRNA genes. tRNA genes
with introns not detected. -i, -sr switches ignored.
Composite Metazoan mitochondrial genetic code used.</p>

<p style="margin-top: 1em">-mtmam <br>
Search for Mammalian mitochondrial tRNA genes. -i, -sr
switches ignored. -tv switch set. Mammalian mitochondrial
genetic code used.</p>

<p style="margin-top: 1em">-mtx <br>
Same as -mt but low scoring tRNA genes are not reported.</p>

<p style="margin-top: 1em">-mtd <br>
Overlapping metazoan mitochondrial tRNA genes on opposite
strands are reported.</p>

<p style="margin-top: 1em">-gc[num] <br>
Use the GenBank transl_table = [num] genetic code.
Individual modifications can be appended using
,BBB=&lt;aa&gt; B = A,C,G, or T. &lt;aa&gt; is the three
letter code for an amino-acid. <br>
More than one modification can be specified. eg
-gcvert,aga=Trp,agg=Trp uses the Vertebrate Mitochondrial
code and the codons AGA and AGG changed to Tryptophan.</p>

<p style="margin-top: 1em">-gcstd <br>
Use standard genetic code.</p>

<p style="margin-top: 1em">-gcmet <br>
Use composite Metazoan mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gcvert <br>
Use Vertebrate mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gcinvert <br>
Use Invertebrate mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gcyeast <br>
Use Yeast mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gcprot <br>
Use Mold/Protozoan/Coelenterate mitochondrial genetic
code.</p>

<p style="margin-top: 1em">-gcciliate <br>
Use Ciliate genetic code.</p>

<p style="margin-top: 1em">-gcflatworm <br>
Use Echinoderm/Flatworm mitochondrial genetic code</p>

<p style="margin-top: 1em">-gceuplot <br>
Use Euplotid genetic code.</p>

<p style="margin-top: 1em">-gcbact <br>
Use Bacterial/Plant Chloroplast genetic code.</p>

<p style="margin-top: 1em">-gcaltyeast <br>
Use alternative Yeast genetic code.</p>

<p style="margin-top: 1em">-gcascid <br>
Use Ascidian Mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gcaltflat <br>
Use alternative Flatworm Mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gcblep <br>
Use Blepharisma genetic code.</p>

<p style="margin-top: 1em">-gcchloroph <br>
Use Chlorophycean Mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gctrem <br>
Use Trematode Mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gcscen <br>
Use Scenedesmus obliquus Mitochondrial genetic code.</p>

<p style="margin-top: 1em">-gcthraust <br>
Use Thraustochytrium Mitochondrial genetic code.</p>

<p style="margin-top: 1em">-tv <br>
Do not search for mitochondrial TV replacement loop tRNA
genes. Only relevant if -mt used.</p>

<p style="margin-top: 1em">-c7 <br>
Search for tRNA genes with 7 base C-loops only.</p>

<p style="margin-top: 1em">-i <br>
Search for tRNA genes with introns in anticodon loop with
maximum length 3000 bases. Minimum intron length is 0 bases.
Ignored if -m is specified.</p>

<p style="margin-top: 1em">-i[max] <br>
Search for tRNA genes with introns in anticodon loop with
maximum length [max] bases. Minimum intron length is 0
bases. Ignored if -m is specified.</p>

<p style="margin-top: 1em">-i[min],[max] <br>
Search for tRNA genes with introns in anticodon loop with
maximum length [max] bases, and minimum length [min] bases.
Ignored if -m is specified.</p>

<p style="margin-top: 1em">-io <br>
Same as -i, but allow tRNA genes with long introns to
overlap shorter tRNA genes.</p>

<p style="margin-top: 1em">-if <br>
Same as -i, but fix intron between positions 37 and 38 on
C-loop (one base after anticodon).</p>

<p style="margin-top: 1em">-ifo <br>
Same as -if and -io combined.</p>

<p style="margin-top: 1em">-ir <br>
Same as -i, but report tRNA genes with minimum length [min]
bases rather than search for tRNA genes with minimum length
[min] bases. With this switch, [min] acts as an output <br>
filter, minimum intron length for searching is still 0
bases.</p>

<p style="margin-top: 1em">-c <br>
Assume that each sequence has a circular topology. Search
wraps around each end. Default setting.</p>

<p style="margin-top: 1em">-l <br>
Assume that each sequence has a linear topology. Search does
not wrap.</p>

<p style="margin-top: 1em">-d <br>
Double. Search both strands of each sequence. Default
setting.</p>

<p style="margin-top: 1em">-s or -s+ <br>
Single. Do not search the complementary (antisense) strand
of each sequence.</p>

<p style="margin-top: 1em">-sc or -s- <br>
Single complementary. Do not search the sense strand of each
sequence.</p>

<p style="margin-top: 1em">-ps <br>
Lower scoring thresholds to 95% of default levels.</p>

<p style="margin-top: 1em">-ps[num] <br>
Change scoring thresholds to [num] percent of default
levels.</p>

<p style="margin-top: 1em">-rp <br>
Flag possible pseudogenes (score &lt; 100 or tRNA anticodon
loop &lt;&gt; 7 bases long). Note that genes with score &lt;
100 will not be detected or flagged if scoring thresholds
are not <br>
also changed to below 100% (see -ps switch).</p>

<p style="margin-top: 1em">-seq <br>
Print out primary sequence.</p>

<p style="margin-top: 1em">-br <br>
Show secondary structure of tRNA gene primary sequence using
round brackets.</p>

<p style="margin-top: 1em">-fasta <br>
Print out primary sequence in fasta format.</p>

<p style="margin-top: 1em">-fo <br>
Print out primary sequence in fasta format only (no
secondary structure).</p>

<p style="margin-top: 1em">-fon <br>
Same as -fo, with sequence and gene numbering in header.</p>

<p style="margin-top: 1em">-fos <br>
Same as -fo, with no spaces in header.</p>

<p style="margin-top: 1em">-fons <br>
Same as -fo, with sequence and gene numbering, but no
spaces.</p>

<p style="margin-top: 1em">-w <br>
Print out in Batch mode.</p>

<p style="margin-top: 1em">-ss <br>
Use the stricter canonical 1-2 bp spacer1 and 1 bp spacer2.
Ignored if -mt set. Default is to allow 3 bp spacer1 and 0-2
bp spacer2, which may degrade selectivity.</p>

<p style="margin-top: 1em">-v <br>
Verbose. Prints out information during search to STDERR.</p>

<p style="margin-top: 1em">-a <br>
Print out tRNA domain for tmRNA genes.</p>

<p style="margin-top: 1em">-a7 <br>
Restrict tRNA astem length to a maximum of 7 bases</p>

<p style="margin-top: 1em">-aa <br>
Display message if predicted iso-acceptor species does not
match species in sequence name (if present).</p>

<p style="margin-top: 1em">-j <br>
Display 4-base sequence on 3&rsquo; end of astem regardless
of predicted amino-acyl acceptor length.</p>

<p style="margin-top: 1em">-jr <br>
Allow some divergence of 3&rsquo; amino-acyl acceptor
sequence from NCCA.</p>

<p style="margin-top: 1em">-jr4 <br>
Allow some divergence of 3&rsquo; amino-acyl acceptor
sequence from NCCA, and display 4 bases.</p>

<p style="margin-top: 1em">-q <br>
Do not print configuration line (which switches and files
were used).</p>

<p style="margin-top: 1em">-rn <br>
Repeat sequence name before summary information.</p>

<p style="margin-top: 1em">-O [outfile] <br>
Print output to . If [&rsquo;outfile] already exists, it is
overwritten. By default all output goes to stdout.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
aragorn detects tRNA, mtRNA, and tmRNA genes. A minimum
requirement is at least a 32 bit compiler architecture
(variable types int and unsigned int are at least 4 bytes
long).</p>

<p style="margin-top: 1em">[FILE] is assumed to contain one
or more sequences in FASTA format. Results of the search are
printed to STDOUT. All switches are optional and
case-insensitive. Unless -i is <br>
specified, tRNA genes containing introns are not
detected.</p>

<p style="margin-top: 1em">AUTHORS <br>
Bjorn Canback &lt;bcanback@acgt.se&gt;, Dean Laslett
&lt;gaiaquark@gmail.com&gt;</p>

<p style="margin-top: 1em">REFERENCES <br>
Laslett, D. and Canback, B. (2004) ARAGORN, a program for
the detection of transfer RNA and transfer-messenger RNA
genes in nucleotide sequences Nucleic Acids Research,
32;11-16</p>

<p style="margin-top: 1em">Laslett, D. and Canback, B.
(2008) ARWEN: a program to detect tRNA genes in metazoan
mitochondrial nucleotide sequences Bioinformatics, 24(2);
172-175.</p>

<p style="margin-top: 1em">02/24/2013 ARAGORN(1)</p>
<hr>
</body>
</html>
