<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:50:08 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ARDEN-CREATE(1) User Commands ARDEN-CREATE(1)</p>

<p style="margin-top: 1em">NAME <br>
arden-create - generate an artificial reference genome (AR)
from a given input</p>

<p style="margin-top: 1em">SYNOPSIS <br>
arden-create [options] [OUTPUTFOLDER] [INPUT FASTA]...</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Script to generate an artificial reference genome (AR) from
a given input. The AR fulfills the following conditions by
default settings:</p>

<p style="margin-top: 1em">1 has a nucleotide distribution
equal to the input genome</p>

<p style="margin-top: 1em">2 has an amino acid (aa)
distribution equal to the input genome</p>

<p style="margin-top: 1em">3 keeps the aa neighborhood
similar to the neighborhood in input genome</p>

<p style="margin-top: 1em">4 protects start and stop codons
from mutations</p>

<p style="margin-top: 1em">OPTIONS <br>
Required Arguments:</p>

<p style="margin-top: 1em">OUTPUTFOLDER <br>
complete path to output destination folder</p>

<p style="margin-top: 1em">INPUT FASTA <br>
Single sequence Fasta file.</p>

<p style="margin-top: 1em">--version <br>
show program&rsquo;s version number and exit</p>

<p style="margin-top: 1em">-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">-d DISTANCE, --distance=DISTANCE
<br>
distance between 2 mutations on DNA level. The minimum
distance will then be d-3. [default: 15]</p>

<p style="margin-top: 1em">-o ORF, --orf=ORF <br>
1- protect ORF structure from mutations. 0 - allow mutations
in start / stop codon [default: 1]</p>

<p style="margin-top: 1em">-r REVSUB, --revsub=REVSUB <br>
1 - reverse substitution if no suitable counterpart was
found (balanced mutation). 0 - keep the unbalanced
mutations.[default: 1]</p>

<p style="margin-top: 1em">-m RANDOM, --random=RANDOM <br>
1/0 variable. 1 - shuffled starting positions for the</p>

<p style="margin-top: 1em">mutations. 0 - linear mutation
[default: 1]</p>

<p style="margin-top: 1em">-s SORF, --saveorfs=SORF <br>
Save found start and stop codons in a pickle file (can be
loaded if the input fasta is used again if a filename is
specified). [default: ]</p>

<p style="margin-top: 1em">-l LORF, --loadorfs=LORF <br>
Specify filename to a previously created pickle file
(contains positions of start/stop codons) [default: ]</p>

<p style="margin-top: 1em">-n NAME, --name=NAME <br>
Specify name which will be used as header. [default: ]</p>

<p style="margin-top: 1em">-p PEXAMPLES,
--pexamples=PEXAMPLES <br>
set to 1 if you want to print examples</p>

<p style="margin-top: 1em">arden-create 2013-04-25 February
2014 ARDEN-CREATE(1)</p>
<hr>
</body>
</html>
