<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:50:16 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ARIBA(1) ARIBA(1)</p>

<p style="margin-top: 1em">NAME <br>
ariba - Antibiotic Resistance Identification By Assembly</p>

<p style="margin-top: 1em">SYNOPSIS <br>
ariba &lt;subtool&gt; [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ariba is the main executable for the ARIBA software, a tool
that identifies antibiotic resistance genes by running local
assemblies.</p>

<p style="margin-top: 1em">In general, the input is a FASTA
file of reference sequences (can be a mix of genes and
noncoding sequences) and paired sequencing reads. ARIBA
reports which of the reference <br>
sequence were found, plus detailed information on the
quality of the assemblies and any variants between the
sequencing reads and the reference sequences.</p>

<p style="margin-top: 1em">The actual functionality is
provided by the subtools listed below.</p>

<p style="margin-top: 1em">Subtools <br>
getref <br>
download reference data</p>

<p style="margin-top: 1em">prepareref <br>
prepare reference data for input to &quot;run&quot;</p>

<p style="margin-top: 1em">run <br>
run the ARIBA local assembly pipeline</p>

<p style="margin-top: 1em">refquery <br>
get cluster or sequence info from prepareref output</p>

<p style="margin-top: 1em">reportfilter <br>
Filter report.tsv file</p>

<p style="margin-top: 1em">summary <br>
summarise multiple reports made by &quot;run&quot;</p>

<p style="margin-top: 1em">flag <br>
translate the meaning of a flag</p>

<p style="margin-top: 1em">aln2meta <br>
converts multi-aln fasta and SNPs to metadata</p>

<p style="margin-top: 1em">test <br>
run small built-in test dataset</p>

<p style="margin-top: 1em">version <br>
get versions and exit</p>

<p style="margin-top: 1em">Subtool help <br>
You can run</p>

<p style="margin-top: 1em">ariba &lt;subtool&gt; --help</p>

<p style="margin-top: 1em">to show more information about
how to run each subtool.</p>

<p style="margin-top: 1em">BUGS <br>
Please report bugs to the ARIBA issue tracker:
&lt;https://github.com/sanger-pathogens/ariba/issues&gt;</p>
 
<p style="margin-top: 1em">ARIBA(1)</p>
<hr>
</body>
</html>
