<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:52:16 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ART(1) General Commands Manual ART(1)</p>

<p style="margin-top: 1em">NAME <br>
art - Artemis genome browser and annotation tool</p>

<p style="margin-top: 1em">SYNOPSIS <br>
art [options] SEQUENCE_FILE [+FEATURE_FILE...]</p>

<p style="margin-top: 1em">OPTIONS <br>
SEQUENCE_FILE An EMBL, GenBank, FASTA, or GFF3 file <br>
FEATURE_FILE An Artemis TAB file, or GFF file</p>

<p style="margin-top: 1em">-options FILE Read a text file
of options from FILE <br>
-debug Run using the debugging JVM instead <br>
-fast | -fast64 Use the FastVM (hp Tru64 UNIX) with 32/64
bit pointers</p>

<p style="margin-top: 1em">-Dblack_belt_mode=? Keep warning
messages to a minimum [true,false] <br>
-Doffset=XXX Open viewer at base position XXX [integer &gt;=
1] <br>
-Duserplot=FILE[,FILE2] Open one or more userplots <br>
-Dloguserplot=FILE[,FILE2] Open one or more userplots, take
log(data) <br>
-Dbam=FILE[,FILE2,...] Open one or more BAM, VCF or BCF
files <br>
-DbamClone=n Open all BAMs in multiple (n &gt; 1) panels
<br>
-Dbam[1,2,..]=FILE[,FILE2,..] Open BAMs in separate panels
<br>
-Dshow_snps Show SNP marks in BamView <br>
-Dshow_snp_plot Open SNP plot in BamView <br>
-Dshow_cov_plot Open coverage plot in BamView <br>
-Dshow_forward_lines=? Hide/show forward frame lines
[true,false] <br>
-Dshow_reverse_lines=? Hide/show reverse frame lines
[true,false] <br>
-Dchado=&quot;h:p/d?u&quot; Get Artemis to open this CHADO
database <br>
-Dread_only Open CHADO database read-only</p>

<p style="margin-top: 1em">EXAMPLES <br>
% art AJ006275.embl <br>
% art contigs.fa +annotation.gff +islands.tab <br>
% art -Dblack_belt_mode=true -Dbam=ecoli_hiseq.bam
E_coli_K12.gbk <br>
% art -Duserplot=repeatmap.plot,geecee.plot Plasmid.gff3</p>

<p style="margin-top: 1em">SEE ALSO <br>
http://www.sanger.ac.uk/resources/software/artemis/</p>

<p style="margin-top: 1em">act(1), dnaplotter(1)</p>

<p style="margin-top: 1em">February 2014 ART(1)</p>
<hr>
</body>
</html>
