<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ART(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ART(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">ART(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
art - Artemis genome browser and annotation tool
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>art</b> [<i>options</i>] <i>SEQUENCE_FILE</i> [<i>+FEATURE_FILE</i>...]
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<br/>
 SEQUENCE_FILE An EMBL, GenBank, FASTA, or GFF3 file
<br/>
 FEATURE_FILE An Artemis TAB file, or GFF file
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -options FILE Read a text file of options from FILE
<br/>
 -debug Run using the debugging JVM instead
<br/>
 -fast | -fast64 Use the FastVM (hp Tru64 UNIX) with 32/64 bit pointers
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -Dblack_belt_mode=? Keep warning messages to a minimum [true,false]
<br/>
 -Doffset=XXX Open viewer at base position XXX [integer &gt;= 1]
<br/>
 -Duserplot=FILE[,FILE2] Open one or more userplots
<br/>
 -Dloguserplot=FILE[,FILE2] Open one or more userplots, take log(data)
<br/>
 -Dbam=FILE[,FILE2,...] Open one or more BAM, VCF or BCF files
<br/>
 -DbamClone=n Open all BAMs in multiple (n &gt; 1) panels
<br/>
 -Dbam[1,2,..]=FILE[,FILE2,..] Open BAMs in separate panels
<br/>
 -Dshow_snps Show SNP marks in BamView
<br/>
 -Dshow_snp_plot Open SNP plot in BamView
<br/>
 -Dshow_cov_plot Open coverage plot in BamView
<br/>
 -Dshow_forward_lines=? Hide/show forward frame lines [true,false]
<br/>
 -Dshow_reverse_lines=? Hide/show reverse frame lines [true,false]
<br/>
 -Dchado=&quot;h:p/d?u&quot; Get Artemis to open this CHADO database
<br/>
 -Dread_only Open CHADO database read-only
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<br/>
 % art AJ006275.embl
<br/>
 % art contigs.fa +annotation.gff +islands.tab
<br/>
 % art -Dblack_belt_mode=true -Dbam=ecoli_hiseq.bam E_coli_K12.gbk
<br/>
 % art -Duserplot=repeatmap.plot,geecee.plot Plasmid.gff3
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<br/>
 http://www.sanger.ac.uk/resources/software/artemis/
<div style="height: 1.00em;">&#x00A0;</div>
<b>act</b>(1), <b>dnaplotter</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2014</td>
    <td class="foot-os"></td>
  </tr>
</table>
</body>
</html>
