<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ART_SOLID(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ART_SOLID(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">ART_SOLID(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
art_SOLiD - Simulation of Applied Biosystems SOLiD Sequencing
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
ART is a set of simulation tools to generate synthetic next-generation
  sequencing reads. ART simulates sequencing reads by mimicking real sequencing
  process with empirical error models or quality profiles summarized from large
  recalibrated sequencing data.
<div class="Pp"></div>
art_SOLiD can be used for Simulation of Applied Biosystems SOLiD Sequencing.
<h1 class="Sh" title="Sh" id="USAGE"><a class="selflink" href="#USAGE">USAGE</a></h1>
<h2 class="Ss" title="Ss" id="SINGLE-END_(F3_READ)_SIMULATION"><a class="selflink" href="#SINGLE-END_(F3_READ)_SIMULATION">SINGLE-END
  (F3 READ) SIMULATION</a></h2>
<b>art_SOLiD</b> [ options ] &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt;
  &lt;LEN_READ&gt; &lt;FOLD_COVERAGE&gt;
<h2 class="Ss" title="Ss" id="MATE-PAIR_READS_(F3-R3_PAIR)_SIMULATION"><a class="selflink" href="#MATE-PAIR_READS_(F3-R3_PAIR)_SIMULATION">MATE-PAIR
  READS (F3-R3 PAIR) SIMULATION</a></h2>
<b>art_SOLiD</b> [ options ] &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt;
  &lt;LEN_READ&gt; &lt;FOLD_COVERAGE&gt; &lt;MEAN_FRAG_LEN&gt; &lt;STD_DEV&gt;
<h2 class="Ss" title="Ss" id="PAIRED-END_READS_(F3-F5_PAIR)_SIMULATION"><a class="selflink" href="#PAIRED-END_READS_(F3-F5_PAIR)_SIMULATION">PAIRED-END
  READS (F3-F5 PAIR) SIMULATION</a></h2>
<b>art_SOLiD</b> [ options ] &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt;
  &lt;LEN_READ_F3&gt; &lt;LEN_READ_F5&gt; &lt;FOLD_COVERAGE&gt;
  &lt;MEAN_FRAG_LEN&gt; &lt;STD_DEV&gt;
<h2 class="Ss" title="Ss" id="AMPLICON_SEQUENCING_SIMULATION"><a class="selflink" href="#AMPLICON_SEQUENCING_SIMULATION">AMPLICON
  SEQUENCING SIMULATION</a></h2>
<b>art_SOLiD</b> [ options ] <b>-A</b> s &lt;INPUT_SEQ_FILE&gt;
  &lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt; &lt;READS_PER_AMPLICON&gt;
<div class="Pp"></div>
<b>art_SOLiD</b> [ options ] <b>-A</b> m &lt;INPUT_SEQ_FILE&gt;
  &lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt; &lt;READ_PAIRS_PER_AMPLICON&gt;
<div class="Pp"></div>
<b>art_SOLiD</b> [ options ] <b>-A</b> p &lt;INPUT_SEQ_FILE&gt;
  &lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ_F3&gt; &lt;LEN_READ_F5&gt;
  &lt;READ_PAIRS_PER_AMPLICON&gt;
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="MANDATORY_OPTIONS"><a class="selflink" href="#MANDATORY_OPTIONS">MANDATORY
  OPTIONS</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">INPUT_SEQ_FILE - filename of DNA/RNA reference sequences in
    FASTA format</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">OUTPUT_FILE_PREFIX - prefix or directory for all output
    read data files</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">FOLD_COVERAGE - fold of read coverage over the reference
    sequences</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">LEN_READ - length of F3/R3 reads</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">LEN_READ_F3 - length of F3 reads for paired-end read
    simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">LEN_READ_F5 - length of F5 reads for paired-end read
    simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">READS_PER_AMPLICON - number of reads per amplicon</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">READ_PAIRS_PER_AMPLICON - number of read pairs per
    amplicon</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">MEAN_FRAG_LEN - mean DNA/RNA fragment size for
    matepair/paired-end read simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">STD_DEV - standard deviation of the DNA/RNA fragment sizes
    for matepair/paired-end read simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<h2 class="Ss" title="Ss" id="OPTIONAL_PARAMETERS"><a class="selflink" href="#OPTIONAL_PARAMETERS">OPTIONAL
  PARAMETERS</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b> specify the read type for amplicon sequencing
    simulation (s:single-end, m: matepair, p: paired-end)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> indicate to use CIGAR 'M' instead of '=/X' for
    alignment match/mismatch</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> indicate to generate a SAM alignment file</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> specify the random seed for the simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> specify the scale factor adjusting error rate
    (e.g., <b>-f</b> 0 for zero-error rate simulation)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> specify user's own read profile for
    simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">1) singl-end 25bp reads simulation at 10X coverage</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_SOLiD <b>-s</b> seq_reference.fa ./outdir/single_dat 25
      10</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">2) singl-end 75bp reads simulation at 20X coverage with
    user's error profile</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_SOLiD <b>-s</b> <b>-p</b>
      ../SOLiD_profiles/profile_pseudo ./seq_reference.fa ./dat_userProfile 75
      20</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">3) matepair 35bp (F3-R3) reads simulation at 20X coverage
    with DNA/RNA MEAN fragment size 2000bp and STD 50</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_SOLiD <b>-s</b> seq_reference.fa ./outdir/matepair_dat
      35 20 2000 50</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">4) matepair reads simulation with a fixed random seed</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_SOLiD <b>-r</b> 777 <b>-s</b> seq_reference.fa
      ./outdir/matepair_fs 50 10 1500 50</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">5) paired-end reads (75bp F3, 35bp F5) simulation with the
    MEAN fragment size 250 and STD 10 at 20X coverage</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_SOLiD <b>-s</b> seq_reference.fa ./outdir/paired_dat 75
      35 50 250 10</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">6) amplicon sequencing with 25bp single-end reads at 100
    reads per amplicon</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_SOLiD <b>-A</b> s <b>-s</b> amp_reference.fa
      ./outdir/amp_single 25 100</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">7) amplicon sequencing with 50bp matepair reads at 80 read
    pairs per amplicon</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_SOLiD <b>-A</b> m <b>-s</b> amp_reference.fa
      ./outdir/amp_matepair 50 80</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">8) amplicon sequencing with paired-end reads (35bp F3, 25bp
    F5 reads) at 50 pairs per amplicon</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_SOLiD <b>-A</b> p <b>-s</b> amp_reference.fa
      ./outdir/amp_pair 35 25 50</dd>
</dl>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2016</td>
    <td class="foot-os">art_SOLiD 3.19.15</td>
  </tr>
</table>
</body>
</html>
