<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:52:17 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ART_454(1) User Commands ART_454(1)</p>

<p style="margin-top: 1em">NAME <br>
art_454 - Simulation of 454 Pyrosequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ART is a set of simulation tools to generate synthetic
next-generation sequencing reads. ART simulates sequencing
reads by mimicking real sequencing process with empirical
error <br>
models or quality profiles summarized from large
recalibrated sequencing data.</p>

<p style="margin-top: 1em">art_454 can be used for
Simulation of 454 Pyrosequencing.</p>

<p style="margin-top: 1em">USAGE <br>
SINGLE-END SIMULATION <br>
art_454 [-s] [-a ] [-t] [-r rand_seed] [ -p read_profile ] [
-c num_flow_cycles ] &lt;INPUT_SEQ_FILE&gt;
&lt;OUTPUT_FILE_PREFIX&gt; &lt;FOLD_COVERAGE&gt;</p>

<p style="margin-top: 1em">PAIRED-END SIMULATION <br>
art_454 <br>
[-s] [-a ] [-t] [-r rand_seed] [ -p read_profile ] [ -c
num_flow_cycles ] &lt;INPUT_SEQ_FILE&gt;
&lt;OUTPUT_FILE_PREFIX&gt; &lt;FOLD_COVERAGE&gt;
&lt;MEAN_FRAG_LEN&gt; &lt;STD_DEV&gt;</p>

<p style="margin-top: 1em">AMPLICON SEQUENCING SIMULATION
<br>
art_454 [-s] [-a ] [-t] [-r rand_seed] [ -p read_profile ] [
-c num_flow_cycles ] &lt;-A|-B&gt; &lt;INPUT_SEQ_FILE&gt;
&lt;OUTPUT_FILE_PREFIX&gt;
&lt;#_READS/#_READ_PAIRS_PER_AMPLICON&gt;</p>

<p style="margin-top: 1em">OPTIONS <br>
MANDATORY OPTIONS <br>
INPUT_SEQ_FILE - the filename of DNA/RNA reference sequences
in FASTA format</p>

<p style="margin-top: 1em">OUTPUT_FILE_PREFIX - the prefix
or directory of output read data file (*.fq) and read
alignment file (*.aln)</p>

<p style="margin-top: 1em">FOLD_COVERAGE - the fold of read
coverage over the reference sequences</p>

<p style="margin-top: 1em">MEAN_FRAG_LEN - the average DNA
fragment size for paired-end read simulation</p>

<p style="margin-top: 1em">STD_DEV - the standard deviation
of the DNA fragment size for paired-end read simulation</p>

<p style="margin-top: 1em">#READS_PER_AMPLICON - number of
reads per amplicon (for 5&rsquo;end amplicon sequencing)</p>

<p style="margin-top: 1em">#READ_PAIRS_PER_AMPLICON -
number of read pairs per amplicon (for two-end amplicon
sequencing)</p>

<p style="margin-top: 1em">OPTIONAL PARAMETERS <br>
-A indicate to perform single-end amplicon sequencing
simulation</p>

<p style="margin-top: 1em">-B indicate to perform
paired-end amplicon sequencing simulation</p>

<p style="margin-top: 1em">-M indicate to use CIGAR
&rsquo;M&rsquo; instead of &rsquo;=/X&rsquo; for alignment
match/mismatch</p>

<p style="margin-top: 1em">-a indicate to output the ALN
alignment file</p>

<p style="margin-top: 1em">-s indicate to output the SAM
alignment file</p>

<p style="margin-top: 1em">-d print out warning messages
for debugging</p>

<p style="margin-top: 1em">-t indicate to simulate reads
from the built-in GS FLX Titanium profile [default: GS FLX
profile]</p>

<p style="margin-top: 1em">-r specify a fixed random seed
for the simulation (to generate two identical datasets from
two different runs)</p>

<p style="margin-top: 1em">-c specify the number of flow
cycles by the sequencer [ default: 100 for GS-FLX, and 200
for GS-FLX Titanium ]</p>

<p style="margin-top: 1em">-p specify user&rsquo;s own read
profile for simulation</p>

<p style="margin-top: 1em">NOTE: the name of a read profile
is the directory containing read profile data files. please
read the REAME file about the format of 454 read profile
data files and. and the <br>
default filenames of these data files.</p>

<p style="margin-top: 1em">EXAMPLES <br>
1) singl-end simulation with 20X coverage</p>

<p style="margin-top: 1em">art_454 -s seq_reference.fa
./outdir/single_dat 20</p>

<p style="margin-top: 1em">2) paired-end simulation with
the mean fragment size 1500 and STD 20 using GS FLX Titanium
platform</p>

<p style="margin-top: 1em">art_454 -s -t seq_reference.fa
./outdir/paired_dat 10 1500 20</p>

<p style="margin-top: 1em">3) paired-end simulation with a
fixed random seed</p>

<p style="margin-top: 1em">art_454 -s -r 777
seq_reference.fa ./outdir/paired_fxSeed 10 2500 50</p>

<p style="margin-top: 1em">4) single-end amplicon
sequencing with 10 reads per amplicon</p>

<p style="margin-top: 1em">art_454 -A -s amplicon_ref.fa
./outdir/amp_single 10</p>

<p style="margin-top: 1em">5) paired-end amplicon
sequencing with 10 read pairs per amplicon</p>

<p style="margin-top: 1em">art_454 -B -s amplicon_ref.fa
./outdir/amp_paired 10</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">art_454 3.19.15 February 2016
ART_454(1)</p>
<hr>
</body>
</html>
