<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ART_454(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ART_454(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">ART_454(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
art_454 - Simulation of 454 Pyrosequencing
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
ART is a set of simulation tools to generate synthetic next-generation
  sequencing reads. ART simulates sequencing reads by mimicking real sequencing
  process with empirical error models or quality profiles summarized from large
  recalibrated sequencing data.
<div class="Pp"></div>
art_454 can be used for Simulation of 454 Pyrosequencing.
<h1 class="Sh" title="Sh" id="USAGE"><a class="selflink" href="#USAGE">USAGE</a></h1>
<h2 class="Ss" title="Ss" id="SINGLE-END_SIMULATION"><a class="selflink" href="#SINGLE-END_SIMULATION">SINGLE-END
  SIMULATION</a></h2>
<b>art_454</b> [-s] [-a ] [-t] [-r rand_seed] [ <b>-p</b> read_profile ] [
  <b>-c</b> num_flow_cycles ] &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt;
  &lt;FOLD_COVERAGE&gt;
<h2 class="Ss" title="Ss" id="PAIRED-END_SIMULATION"><a class="selflink" href="#PAIRED-END_SIMULATION">PAIRED-END
  SIMULATION</a></h2>
<b>art_454</b>
<br/>
 [-s] [-a ] [-t] [-r rand_seed] [ <b>-p</b> read_profile ] [ <b>-c</b>
  num_flow_cycles ] &lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt;
  &lt;FOLD_COVERAGE&gt; &lt;MEAN_FRAG_LEN&gt; &lt;STD_DEV&gt;
<h2 class="Ss" title="Ss" id="AMPLICON_SEQUENCING_SIMULATION"><a class="selflink" href="#AMPLICON_SEQUENCING_SIMULATION">AMPLICON
  SEQUENCING SIMULATION</a></h2>
<b>art_454</b> [-s] [-a ] [-t] [-r rand_seed] [ <b>-p</b> read_profile ] [
  <b>-c</b> num_flow_cycles ] &lt;-A|-B&gt; &lt;INPUT_SEQ_FILE&gt;
  &lt;OUTPUT_FILE_PREFIX&gt; &lt;#_READS/#_READ_PAIRS_PER_AMPLICON&gt;
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="MANDATORY_OPTIONS"><a class="selflink" href="#MANDATORY_OPTIONS">MANDATORY
  OPTIONS</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">INPUT_SEQ_FILE - the filename of DNA/RNA reference
    sequences in FASTA format</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">OUTPUT_FILE_PREFIX - the prefix or directory of output read
    data file (*.fq) and read alignment file (*.aln)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">FOLD_COVERAGE - the fold of read coverage over the
    reference sequences</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">MEAN_FRAG_LEN - the average DNA fragment size for
    paired-end read simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">STD_DEV - the standard deviation of the DNA fragment size
    for paired-end read simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">#READS_PER_AMPLICON - number of reads per amplicon (for
    5'end amplicon sequencing)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">#READ_PAIRS_PER_AMPLICON - number of read pairs per
    amplicon (for two-end amplicon sequencing)</dt>
  <dd class="It-tag"></dd>
</dl>
<h2 class="Ss" title="Ss" id="OPTIONAL_PARAMETERS"><a class="selflink" href="#OPTIONAL_PARAMETERS">OPTIONAL
  PARAMETERS</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b> indicate to perform single-end amplicon
    sequencing simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b> indicate to perform paired-end amplicon
    sequencing simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> indicate to use CIGAR 'M' instead of '=/X' for
    alignment match/mismatch</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> indicate to output the ALN alignment file</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> indicate to output the SAM alignment file</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> print out warning messages for debugging</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> indicate to simulate reads from the built-in GS
    FLX Titanium profile [default: GS FLX profile]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> specify a fixed random seed for the simulation
    (to generate two identical datasets from two different runs)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b> specify the number of flow cycles by the
    sequencer [ default: 100 for GS-FLX, and 200 for GS-FLX Titanium ]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> specify user's own read profile for
    simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<div class="Pp"></div>
NOTE: the name of a read profile is the directory containing read profile data
  files. please read the REAME file about the format of 454 read profile data
  files and. and the default filenames of these data files.
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">1) singl-end simulation with 20X coverage</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_454 <b>-s</b> seq_reference.fa ./outdir/single_dat
    20</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">2) paired-end simulation with the mean fragment size 1500
    and STD 20 using GS FLX Titanium platform</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_454 <b>-s</b> <b>-t</b> seq_reference.fa
      ./outdir/paired_dat 10 1500 20</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">3) paired-end simulation with a fixed random seed</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_454 <b>-s</b> <b>-r</b> 777 seq_reference.fa
      ./outdir/paired_fxSeed 10 2500 50</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">4) single-end amplicon sequencing with 10 reads per
    amplicon</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_454 <b>-A</b> <b>-s</b> amplicon_ref.fa
      ./outdir/amp_single 10</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">5) paired-end amplicon sequencing with 10 read pairs per
    amplicon</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_454 <b>-B</b> <b>-s</b> amplicon_ref.fa
      ./outdir/amp_paired 10</dd>
</dl>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2016</td>
    <td class="foot-os">art_454 3.19.15</td>
  </tr>
</table>
</body>
</html>
