<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:52:18 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ART_ILLUMINA(1) User Commands ART_ILLUMINA(1)</p>

<p style="margin-top: 1em">NAME <br>
art_illumina - Simulation of Illumina sequencers</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ART is a set of simulation tools to generate synthetic
next-generation sequencing reads. ART simulates sequencing
reads by mimicking real sequencing process with empirical
error <br>
models or quality profiles summarized from large
recalibrated sequencing data.</p>

<p style="margin-top: 1em">art_illumina can be used for
Simulation of Illumina sequencers</p>

<p style="margin-top: 1em">USAGE <br>
art_illumina [options] -sam -i &lt;seq_ref_file&gt; -l
&lt;read_length&gt; -f &lt;fold_coverage&gt; -ss
&lt;sequencing_system&gt; -o &lt;outfile_prefix&gt;</p>

<p style="margin-top: 1em">art_illumina [options] -sam -i
&lt;seq_ref_file&gt; -l &lt;read_length&gt; -f
&lt;fold_coverage&gt; -o &lt;outfile_prefix&gt;</p>

<p style="margin-top: 1em">art_illumina [options] -sam -i
&lt;seq_ref_file&gt; -l &lt;read_length&gt; -c
&lt;total_num_reads&gt; -o &lt;outfile_prefix&gt;</p>

<p style="margin-top: 1em">art_illumina [options] -sam -i
&lt;seq_ref_file&gt; -l &lt;read_length&gt; -f
&lt;fold_coverage&gt; -m &lt;mean_fragsize&gt; -s
&lt;std_fragsize&gt; -o &lt;outfile_prefix&gt;</p>

<p style="margin-top: 1em">art_illumina [options] -sam -i
&lt;seq_ref_file&gt; -l &lt;read_length&gt; -c
&lt;total_num_reads&gt; -m &lt;mean_fragsize&gt; -s
&lt;std_fragsize&gt; -o &lt;outfile_prefix&gt;</p>

<p style="margin-top: 1em">OPTIONS <br>
-1 --qprof1 <br>
the first-read quality profile</p>

<p style="margin-top: 1em">-2 --qprof2 <br>
the second-read quality profile</p>

<p style="margin-top: 1em">-amp --amplicon amplicon
sequencing simulation</p>

<p style="margin-top: 1em">-c --rcount <br>
total number of reads/read pairs to be generated [per
amplicon if for amplicon simulation](not be used together
with -f/--fcov)</p>

<p style="margin-top: 1em">-d --id <br>
the prefix identification tag for read ID</p>

<p style="margin-top: 1em">-ef --errfree <br>
indicate to generate the zero sequencing errors SAM file as
well the regular one</p>

<p style="margin-top: 1em">NOTE: the reads in the
zero-error SAM file have the same alignment positions as
those in the regular SAM file, but have no sequencing
errors</p>

<p style="margin-top: 1em">-f --fcov <br>
the fold of read coverage to be simulated or number of
reads/read pairs generated for each amplicon</p>

<p style="margin-top: 1em">-h --help <br>
print out usage information</p>

<p style="margin-top: 1em">-i --in <br>
the filename of input DNA/RNA reference</p>

<p style="margin-top: 1em">-ir --insRate <br>
the first-read insertion rate (default: 0.00009)</p>

<p style="margin-top: 1em">-ir2 --insRate2 the second-read
insertion rate (default: 0.00015)</p>

<p style="margin-top: 1em">-dr --delRate <br>
the first-read deletion rate (default: 0.00011)</p>

<p style="margin-top: 1em">-dr2 --delRate2 the second-read
deletion rate (default: 0.00023)</p>

<p style="margin-top: 1em">-l --len <br>
the length of reads to be simulated</p>

<p style="margin-top: 1em">-m --mflen <br>
the mean size of DNA/RNA fragments for paired-end
simulations</p>

<p style="margin-top: 1em">-mp --matepair indicate a
mate-pair read simulation</p>

<p style="margin-top: 1em">-nf --maskN <br>
the cutoff frequency of &rsquo;N&rsquo; in a window size of
the read length for masking genomic regions</p>

<p style="margin-top: 1em">NOTE: default: &rsquo;-nf
1&rsquo; to mask all regions with &rsquo;N&rsquo;. Use
&rsquo;-nf 0&rsquo; to turn off masking</p>

<p style="margin-top: 1em">-na --noALN <br>
do not output ALN alignment file</p>

<p style="margin-top: 1em">-o --out <br>
the prefix of output filename</p>

<p style="margin-top: 1em">-p --paired <br>
indicate a paired-end read simulation or to generate reads
from both ends of amplicons</p>

<p style="margin-top: 1em">NOTE: art will automatically
switch to a mate-pair simulation if the given mean fragment
size &gt;= 2000</p>

<p style="margin-top: 1em">-q --quiet <br>
turn off end of run summary</p>

<p style="margin-top: 1em">-qs --qShift <br>
the amount to shift every first-read quality score by</p>

<p style="margin-top: 1em">-qs2 --qShift2 <br>
the amount to shift every second-read quality score by</p>

<p style="margin-top: 1em">NOTE: For -qs/-qs2 option, a
positive number will shift up quality scores (the max is 93)
that reduce substitution sequencing errors and a negative
number will shift down <br>
quality scores that increase sequencing errors. If shifting
scores by x, the error rate will be 1/(10^(x/10)) of the
default profile.</p>

<p style="margin-top: 1em">-rs --rndSeed <br>
the seed for random number generator (default: system time
in second)</p>

<p style="margin-top: 1em">NOTE: using a fixed seed to
generate two identical datasets from different runs</p>

<p style="margin-top: 1em">-s --sdev <br>
the standard deviation of DNA/RNA fragment size for
paired-end simulations.</p>

<p style="margin-top: 1em">-sam --samout <br>
indicate to generate SAM alignment file</p>

<p style="margin-top: 1em">-sp --sepProf <br>
indicate to use separate quality profiles for different
bases (ATGC)</p>

<p style="margin-top: 1em">-ss --seqSys <br>
The name of Illumina sequencing system of the built-in
profile used for simulation</p>

<p style="margin-top: 1em">NOTE: sequencing system id names
are:</p>

<p style="margin-top: 1em">GA1 - Genome Analyzer I, GA2 -
Genome Analyzer II</p>

<p style="margin-top: 1em">HS10 - HiSeq 1000, HS20 - HiSeq
2000, HS25 - HiSeq 2500, MS - MiSeq</p>

<p style="margin-top: 1em">-M --cigarM <br>
indicate to use CIGAR &rsquo;M&rsquo; instead of
&rsquo;=/X&rsquo; for alignment match/mismatch</p>

<p style="margin-top: 1em">NOTES <br>
* ART by default selects a built-in quality score profile
according to the read length specified for the run.</p>

<p style="margin-top: 1em">* For single-end simulation, ART
requires input sequence file, outputfile prefix, read
length, and read count/fold coverage.</p>

<p style="margin-top: 1em">* For paired-end simulation
(except for amplicon sequencing), ART also requires the
parameter values of</p>

<p style="margin-top: 1em">the mean and standard deviation
of DNA/RNA fragment lengths</p>

<p style="margin-top: 1em">EXAMPLES <br>
1) single-end read simulation</p>

<p style="margin-top: 1em">art_illumina -sam -i
reference.fa -l 150 -ss HS25 -f 10 -o single_dat</p>

<p style="margin-top: 1em">2) paired-end read
simulation</p>

<p style="margin-top: 1em">art_illumina -sam -i
reference.fa -p -l 150 -ss HS25 -f 20 -m 200 -s 10 -o
paired_dat</p>

<p style="margin-top: 1em">3) mate-pair read simulation</p>

<p style="margin-top: 1em">art_illumina -sam -i
reference.fa -mp -l 50 -f 20 -m 2500 -s 50 -o
matepair_dat</p>

<p style="margin-top: 1em">4) amplicon sequencing
simulation with 5&rsquo; end single-end reads</p>

<p style="margin-top: 1em">art_illumina -amp -sam -na -i
amp_reference.fa -l 50 -f 10 -o amplicon_5end_dat</p>

<p style="margin-top: 1em">5) amplicon sequencing
simulation with paired-end reads</p>

<p style="margin-top: 1em">art_illumina -amp -p -sam -na -i
amp_reference.fa -l 50 -f 10 -o amplicon_pair_dat</p>

<p style="margin-top: 1em">6) amplicon sequencing
simulation with matepair reads</p>

<p style="margin-top: 1em">art_illumina -amp -mp -sam -na
-i amp_reference.fa -l 50 -f 10 -o amplicon_mate_dat</p>

<p style="margin-top: 1em">7) generate an extra SAM file
with zero-sequencing errors for a paired-end read
simulation</p>

<p style="margin-top: 1em">art_illumina -ef -i reference.fa
-p -l 50 -f 20 -m 200 -s 10 -o paired_twosam_dat</p>

<p style="margin-top: 1em">8) reduce the substitution error
rate to one 10th of the default profile</p>

<p style="margin-top: 1em">art_illumina -i reference.fa -qs
10 -qs2 10 -l 50 -f 10 -p -m 500 -s 10 -sam -o
reduce_error</p>

<p style="margin-top: 1em">9) turn off the masking of
genomic regions with unknown nucleotides &rsquo;N&rsquo;</p>

<p style="margin-top: 1em">art_illumina -nf 0 -sam -i
reference.fa -p -l 50 -f 20 -m 200 -s 10 -o
paired_nomask</p>

<p style="margin-top: 1em">10) masking genomic regions with
&gt;=5 &rsquo;N&rsquo;s within the read length 50</p>

<p style="margin-top: 1em">art_illumina -nf 5 -sam -i
reference.fa -p -l 50 -f 20 -m 200 -s 10 -o
paired_maskN5</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">art_illumina 3.19.15 February
2016 ART_ILLUMINA(1)</p>
<hr>
</body>
</html>
