<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:52:19 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ART_SOLID(1) User Commands ART_SOLID(1)</p>

<p style="margin-top: 1em">NAME <br>
art_SOLiD - Simulation of Applied Biosystems SOLiD
Sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ART is a set of simulation tools to generate synthetic
next-generation sequencing reads. ART simulates sequencing
reads by mimicking real sequencing process with empirical
error <br>
models or quality profiles summarized from large
recalibrated sequencing data.</p>

<p style="margin-top: 1em">art_SOLiD can be used for
Simulation of Applied Biosystems SOLiD Sequencing.</p>

<p style="margin-top: 1em">USAGE <br>
SINGLE-END (F3 READ) SIMULATION <br>
art_SOLiD [ options ] &lt;INPUT_SEQ_FILE&gt;
&lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt;
&lt;FOLD_COVERAGE&gt;</p>

<p style="margin-top: 1em">MATE-PAIR READS (F3-R3 PAIR)
SIMULATION <br>
art_SOLiD [ options ] &lt;INPUT_SEQ_FILE&gt;
&lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt;
&lt;FOLD_COVERAGE&gt; &lt;MEAN_FRAG_LEN&gt;
&lt;STD_DEV&gt;</p>

<p style="margin-top: 1em">PAIRED-END READS (F3-F5 PAIR)
SIMULATION <br>
art_SOLiD [ options ] &lt;INPUT_SEQ_FILE&gt;
&lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ_F3&gt;
&lt;LEN_READ_F5&gt; &lt;FOLD_COVERAGE&gt;
&lt;MEAN_FRAG_LEN&gt; &lt;STD_DEV&gt;</p>

<p style="margin-top: 1em">AMPLICON SEQUENCING SIMULATION
<br>
art_SOLiD [ options ] -A s &lt;INPUT_SEQ_FILE&gt;
&lt;OUTPUT_FILE_PREFIX&gt; &lt;LEN_READ&gt;
&lt;READS_PER_AMPLICON&gt;</p>

<p style="margin-top: 1em">art_SOLiD [ options ] -A m
&lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt;
&lt;LEN_READ&gt; &lt;READ_PAIRS_PER_AMPLICON&gt;</p>

<p style="margin-top: 1em">art_SOLiD [ options ] -A p
&lt;INPUT_SEQ_FILE&gt; &lt;OUTPUT_FILE_PREFIX&gt;
&lt;LEN_READ_F3&gt; &lt;LEN_READ_F5&gt;
&lt;READ_PAIRS_PER_AMPLICON&gt;</p>

<p style="margin-top: 1em">OPTIONS <br>
MANDATORY OPTIONS <br>
INPUT_SEQ_FILE - filename of DNA/RNA reference sequences in
FASTA format</p>

<p style="margin-top: 1em">OUTPUT_FILE_PREFIX - prefix or
directory for all output read data files</p>

<p style="margin-top: 1em">FOLD_COVERAGE - fold of read
coverage over the reference sequences</p>

<p style="margin-top: 1em">LEN_READ - length of F3/R3
reads</p>

<p style="margin-top: 1em">LEN_READ_F3 - length of F3 reads
for paired-end read simulation</p>

<p style="margin-top: 1em">LEN_READ_F5 - length of F5 reads
for paired-end read simulation</p>

<p style="margin-top: 1em">READS_PER_AMPLICON - number of
reads per amplicon</p>

<p style="margin-top: 1em">READ_PAIRS_PER_AMPLICON - number
of read pairs per amplicon</p>

<p style="margin-top: 1em">MEAN_FRAG_LEN - mean DNA/RNA
fragment size for matepair/paired-end read simulation</p>

<p style="margin-top: 1em">STD_DEV - standard deviation of
the DNA/RNA fragment sizes for matepair/paired-end read
simulation</p>

<p style="margin-top: 1em">OPTIONAL PARAMETERS <br>
-A specify the read type for amplicon sequencing simulation
(s:single-end, m: matepair, p: paired-end)</p>

<p style="margin-top: 1em">-M indicate to use CIGAR
&rsquo;M&rsquo; instead of &rsquo;=/X&rsquo; for alignment
match/mismatch</p>

<p style="margin-top: 1em">-s indicate to generate a SAM
alignment file</p>

<p style="margin-top: 1em">-r specify the random seed for
the simulation</p>

<p style="margin-top: 1em">-f specify the scale factor
adjusting error rate (e.g., -f 0 for zero-error rate
simulation)</p>

<p style="margin-top: 1em">-p specify user&rsquo;s own read
profile for simulation</p>

<p style="margin-top: 1em">EXAMPLES <br>
1) singl-end 25bp reads simulation at 10X coverage</p>

<p style="margin-top: 1em">art_SOLiD -s seq_reference.fa
./outdir/single_dat 25 10</p>

<p style="margin-top: 1em">2) singl-end 75bp reads
simulation at 20X coverage with user&rsquo;s error
profile</p>

<p style="margin-top: 1em">art_SOLiD -s -p
../SOLiD_profiles/profile_pseudo ./seq_reference.fa
./dat_userProfile 75 20</p>

<p style="margin-top: 1em">3) matepair 35bp (F3-R3) reads
simulation at 20X coverage with DNA/RNA MEAN fragment size
2000bp and STD 50</p>

<p style="margin-top: 1em">art_SOLiD -s seq_reference.fa
./outdir/matepair_dat 35 20 2000 50</p>

<p style="margin-top: 1em">4) matepair reads simulation
with a fixed random seed</p>

<p style="margin-top: 1em">art_SOLiD -r 777 -s
seq_reference.fa ./outdir/matepair_fs 50 10 1500 50</p>

<p style="margin-top: 1em">5) paired-end reads (75bp F3,
35bp F5) simulation with the MEAN fragment size 250 and STD
10 at 20X coverage</p>

<p style="margin-top: 1em">art_SOLiD -s seq_reference.fa
./outdir/paired_dat 75 35 50 250 10</p>

<p style="margin-top: 1em">6) amplicon sequencing with 25bp
single-end reads at 100 reads per amplicon</p>

<p style="margin-top: 1em">art_SOLiD -A s -s
amp_reference.fa ./outdir/amp_single 25 100</p>

<p style="margin-top: 1em">7) amplicon sequencing with 50bp
matepair reads at 80 read pairs per amplicon</p>

<p style="margin-top: 1em">art_SOLiD -A m -s
amp_reference.fa ./outdir/amp_matepair 50 80</p>

<p style="margin-top: 1em">8) amplicon sequencing with
paired-end reads (35bp F3, 25bp F5 reads) at 50 pairs per
amplicon</p>

<p style="margin-top: 1em">art_SOLiD -A p -s
amp_reference.fa ./outdir/amp_pair 35 25 50</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">art_SOLiD 3.19.15 February 2016
ART_SOLID(1)</p>
<hr>
</body>
</html>
