<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ART_ILLUMINA(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ART_ILLUMINA(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">ART_ILLUMINA(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
art_illumina - Simulation of Illumina sequencers
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
ART is a set of simulation tools to generate synthetic next-generation
  sequencing reads. ART simulates sequencing reads by mimicking real sequencing
  process with empirical error models or quality profiles summarized from large
  recalibrated sequencing data.
<div class="Pp"></div>
art_illumina can be used for Simulation of Illumina sequencers
<h1 class="Sh" title="Sh" id="USAGE"><a class="selflink" href="#USAGE">USAGE</a></h1>
<b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt;
  <b>-l</b> &lt;read_length&gt; <b>-f</b> &lt;fold_coverage&gt; <b>-ss</b>
  &lt;sequencing_system&gt; <b>-o</b> &lt;outfile_prefix&gt;
<div class="Pp"></div>
<b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt;
  <b>-l</b> &lt;read_length&gt; <b>-f</b> &lt;fold_coverage&gt; <b>-o</b>
  &lt;outfile_prefix&gt;
<div class="Pp"></div>
<b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt;
  <b>-l</b> &lt;read_length&gt; <b>-c</b> &lt;total_num_reads&gt; <b>-o</b>
  &lt;outfile_prefix&gt;
<div class="Pp"></div>
<b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt;
  <b>-l</b> &lt;read_length&gt; <b>-f</b> &lt;fold_coverage&gt; <b>-m</b>
  &lt;mean_fragsize&gt; <b>-s</b> &lt;std_fragsize&gt; <b>-o</b>
  &lt;outfile_prefix&gt;
<div class="Pp"></div>
<b>art_illumina</b> [options] <b>-sam</b> <b>-i</b> &lt;seq_ref_file&gt;
  <b>-l</b> &lt;read_length&gt; <b>-c</b> &lt;total_num_reads&gt; <b>-m</b>
  &lt;mean_fragsize&gt; <b>-s</b> &lt;std_fragsize&gt; <b>-o</b>
  &lt;outfile_prefix&gt;
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b> <b>--qprof1</b></dt>
  <dd class="It-tag">the first-read quality profile</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-2</b> <b>--qprof2</b></dt>
  <dd class="It-tag">the second-read quality profile</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-amp</b>
  <b>--amplicon</b> amplicon sequencing simulation</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b> <b>--rcount</b></dt>
  <dd class="It-tag">total number of reads/read pairs to be generated [per
      amplicon if for amplicon simulation](not be used together with
      <b>-f</b>/--fcov)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> <b>--id</b></dt>
  <dd class="It-tag">the prefix identification tag for read ID</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ef</b> <b>--errfree</b></dt>
  <dd class="It-tag">indicate to generate the zero sequencing errors SAM file as
      well the regular one</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">NOTE: the reads in the zero-error SAM file have the same
      alignment positions as those in the regular SAM file, but have no
      sequencing errors</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> <b>--fcov</b></dt>
  <dd class="It-tag">the fold of read coverage to be simulated or number of
      reads/read pairs generated for each amplicon</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b> <b>--help</b></dt>
  <dd class="It-tag">print out usage information</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> <b>--in</b></dt>
  <dd class="It-tag">the filename of input DNA/RNA reference</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ir</b> <b>--insRate</b></dt>
  <dd class="It-tag">the first-read insertion rate (default: 0.00009)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-ir2</b>
  <b>--insRate2</b> the second-read insertion rate (default: 0.00015)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-dr</b> <b>--delRate</b></dt>
  <dd class="It-tag">the first-read deletion rate (default: 0.00011)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-dr2</b>
  <b>--delRate2</b> the second-read deletion rate (default: 0.00023)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> <b>--len</b></dt>
  <dd class="It-tag">the length of reads to be simulated</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> <b>--mflen</b></dt>
  <dd class="It-tag">the mean size of DNA/RNA fragments for paired-end
      simulations</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-mp</b>
  <b>--matepair</b> indicate a mate-pair read simulation</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nf</b> <b>--maskN</b></dt>
  <dd class="It-tag">the cutoff frequency of 'N' in a window size of the read
      length for masking genomic regions</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">NOTE: default: '-nf 1' to mask all regions with 'N'. Use
      '-nf 0' to turn off masking</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-na</b> <b>--noALN</b></dt>
  <dd class="It-tag">do not output ALN alignment file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> <b>--out</b></dt>
  <dd class="It-tag">the prefix of output filename</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> <b>--paired</b></dt>
  <dd class="It-tag">indicate a paired-end read simulation or to generate reads
      from both ends of amplicons</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">NOTE: art will automatically switch to a mate-pair
      simulation if the given mean fragment size &gt;= 2000</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> <b>--quiet</b></dt>
  <dd class="It-tag">turn off end of run summary</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qs</b> <b>--qShift</b></dt>
  <dd class="It-tag">the amount to shift every first-read quality score by</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qs2</b> <b>--qShift2</b></dt>
  <dd class="It-tag">the amount to shift every second-read quality score by</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">NOTE: For <b>-qs</b>/-qs2 option, a positive number will
      shift up quality scores (the max is 93) that reduce substitution
      sequencing errors and a negative number will shift down quality scores
      that increase sequencing errors. If shifting scores by x, the error rate
      will be 1/(10^(x/10)) of the default profile.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rs</b> <b>--rndSeed</b></dt>
  <dd class="It-tag">the seed for random number generator (default: system time
      in second)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">NOTE: using a fixed seed to generate two identical datasets
      from different runs</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> <b>--sdev</b></dt>
  <dd class="It-tag">the standard deviation of DNA/RNA fragment size for
      paired-end simulations.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sam</b> <b>--samout</b></dt>
  <dd class="It-tag">indicate to generate SAM alignment file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sp</b> <b>--sepProf</b></dt>
  <dd class="It-tag">indicate to use separate quality profiles for different
      bases (ATGC)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ss</b> <b>--seqSys</b></dt>
  <dd class="It-tag">The name of Illumina sequencing system of the built-in
      profile used for simulation</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">NOTE: sequencing system id names are:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">GA1 - Genome Analyzer I, GA2 - Genome Analyzer II</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">HS10 - HiSeq 1000, HS20 - HiSeq 2000, HS25 - HiSeq 2500, MS
    - MiSeq</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> <b>--cigarM</b></dt>
  <dd class="It-tag">indicate to use CIGAR 'M' instead of '=/X' for alignment
      match/mismatch</dd>
</dl>
<h1 class="Sh" title="Sh" id="NOTES"><a class="selflink" href="#NOTES">NOTES</a></h1>
* ART by default selects a built-in quality score profile according to the read
  length specified for the run.
<div class="Pp"></div>
* For single-end simulation, ART requires input sequence file, outputfile
  prefix, read length, and read count/fold coverage.
<div class="Pp"></div>
* For paired-end simulation (except for amplicon sequencing), ART also requires
  the parameter values of
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">the mean and standard deviation of DNA/RNA fragment
    lengths</dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">1) single-end read simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-sam</b> <b>-i</b> reference.fa <b>-l</b>
      150 <b>-ss</b> HS25 <b>-f</b> 10 <b>-o</b> single_dat</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">2) paired-end read simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-sam</b> <b>-i</b> reference.fa <b>-p</b>
      <b>-l</b> 150 <b>-ss</b> HS25 <b>-f</b> 20 <b>-m</b> 200 <b>-s</b> 10
      <b>-o</b> paired_dat</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">3) mate-pair read simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-sam</b> <b>-i</b> reference.fa <b>-mp</b>
      <b>-l</b> 50 <b>-f</b> 20 <b>-m</b> 2500 <b>-s</b> 50 <b>-o</b>
      matepair_dat</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">4) amplicon sequencing simulation with 5' end single-end
    reads</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-amp</b> <b>-sam</b> <b>-na</b> <b>-i</b>
      amp_reference.fa <b>-l</b> 50 <b>-f</b> 10 <b>-o</b>
    amplicon_5end_dat</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">5) amplicon sequencing simulation with paired-end
    reads</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-amp</b> <b>-p</b> <b>-sam</b> <b>-na</b>
      <b>-i</b> amp_reference.fa <b>-l</b> 50 <b>-f</b> 10 <b>-o</b>
      amplicon_pair_dat</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">6) amplicon sequencing simulation with matepair reads</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-amp</b> <b>-mp</b> <b>-sam</b> <b>-na</b>
      <b>-i</b> amp_reference.fa <b>-l</b> 50 <b>-f</b> 10 <b>-o</b>
      amplicon_mate_dat</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">7) generate an extra SAM file with zero-sequencing errors
    for a paired-end read simulation</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-ef</b> <b>-i</b> reference.fa <b>-p</b>
      <b>-l</b> 50 <b>-f</b> 20 <b>-m</b> 200 <b>-s</b> 10 <b>-o</b>
      paired_twosam_dat</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">8) reduce the substitution error rate to one 10th of the
    default profile</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-i</b> reference.fa <b>-qs</b> 10
      <b>-qs2</b> 10 <b>-l</b> 50 <b>-f</b> 10 <b>-p</b> <b>-m</b> 500 <b>-s</b>
      10 <b>-sam</b> <b>-o</b> reduce_error</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">9) turn off the masking of genomic regions with unknown
    nucleotides 'N'</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-nf</b> 0 <b>-sam</b> <b>-i</b>
      reference.fa <b>-p</b> <b>-l</b> 50 <b>-f</b> 20 <b>-m</b> 200 <b>-s</b>
      10 <b>-o</b> paired_nomask</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">10) masking genomic regions with &gt;=5 'N's within the
    read length 50</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">art_illumina <b>-nf</b> 5 <b>-sam</b> <b>-i</b>
      reference.fa <b>-p</b> <b>-l</b> 50 <b>-f</b> 20 <b>-m</b> 200 <b>-s</b>
      10 <b>-o</b> paired_maskN5</dd>
</dl>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2016</td>
    <td class="foot-os">art_illumina 3.19.15</td>
  </tr>
</table>
</body>
</html>
