<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:52:18 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ARTFASTQGENERATOR(1) User Commands
ARTFASTQGENERATOR(1)</p>

<p style="margin-top: 1em">NAME <br>
artfastqgenerator - outputs artificial FASTQ files derived
from a reference genome</p>

<p style="margin-top: 1em">SYNOPSIS <br>
artfastqgenerator -O &lt;outputPath&gt; -R
&lt;referenceGenomePath&gt; -S
&lt;startSequenceIdentifier&gt; -F1
&lt;fastq1ForQualityScores&gt; -F2
&lt;fastq2ForQualityScores&gt; -CMGCS
&lt;coverageMeanGCcon&acirc; <br>
tentSpread&gt; -CMP &lt;coverageMeanPeak&gt; -CMPGC
&lt;coverageMeanPeakGCcontent&gt; -CSD &lt;coverageSD&gt; -E
&lt;endSequenceIdentifier&gt; -GCC
&lt;GCcontentBasedCoverage&gt; -GCR
&lt;GCcontentRegionSize&gt; -L <br>
&lt;logRegionStats&gt; -N &lt;nucleobaseBufferSize&gt; -OF
&lt;outputFormat&gt; -RCNF &lt;readsContainingNfilter&gt;
-RL &lt;readLength&gt; -SE &lt;simulateErrorInRead&gt; -TLM
&lt;templateLengthMean&gt; -TLSD &lt;template&acirc; <br>
LengthSD&gt; -URQS &lt;useRealQualityScores&gt; -X
&lt;xStart&gt; -Y &lt;yStart&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ArtificialFastqGenerator takes the reference genome (in
FASTA format) as input and outputs artificial FASTQ files in
the Sanger format. It can accept Phred base quality scores
<br>
from existing FASTQ files, and use them to simulate
sequencing errors. Since the artificial FASTQs are derived
from the reference genome, the reference genome provides a
gold- <br>
standard for calling variants (Single Nucleotide
Polymorphisms (SNPs) and insertions and deletions (indels)).
This enables evaluation of a Next Generation Sequencing
(NGS) analy&acirc; <br>
sis pipeline which aligns reads to the reference genome and
then calls the variants.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Print usage help.</p>

<p style="margin-top: 1em">-O, &lt;outputPath&gt; <br>
Path for the artificial fastq and log files, including their
base name (must be specified).</p>

<p style="margin-top: 1em">-R, &lt;referenceGenomePath&gt;
<br>
Reference genome sequence file, (must be specified).</p>

<p style="margin-top: 1em">-S,
&lt;startSequenceIdentifier&gt; <br>
Prefix of the sequence identifier in the reference after
which read generation should begin (must be specified).</p>

<p style="margin-top: 1em">-F1,
&lt;fastq1ForQualityScores&gt; <br>
First fastq file to use for real quality scores, (must be
specified if useRealQualityScores = true).</p>

<p style="margin-top: 1em">-F2,
&lt;fastq2ForQualityScores&gt; <br>
Second fastq file to use for real quality scores, (must be
specified if useRealQualityScores = true).</p>

<p style="margin-top: 1em">-CMGCS,
&lt;coverageMeanGCcontentSpread&gt; <br>
The spread of coverage mean given GC content (default =
0.22).</p>

<p style="margin-top: 1em">-CMP, &lt;coverageMeanPeak&gt;
<br>
The peak coverage mean for a region (default = 37.7).</p>

<p style="margin-top: 1em">-CMPGC,
&lt;coverageMeanPeakGCcontent&gt; <br>
The GC content for regions with peak coverage mean (default
= 0.45).</p>

<p style="margin-top: 1em">-CSD, &lt;coverageSD&gt; <br>
The coverage standard deviation divided by the mean (default
= 0.2).</p>

<p style="margin-top: 1em">-E,
&lt;endSequenceIdentifier&gt; <br>
Prefix of the sequence identifier in the reference where
read generation should stop, (default = end of file).</p>

<p style="margin-top: 1em">-GCC,
&lt;GCcontentBasedCoverage&gt; <br>
Whether nucleobase coverage is biased by GC content (default
= true).</p>

<p style="margin-top: 1em">-GCR,
&lt;GCcontentRegionSize&gt; <br>
Region size in nucleobases for which to calculate GC
content, (default = 150).</p>

<p style="margin-top: 1em">-L, &lt;logRegionStats&gt; <br>
The region size as a multiple of -NBS for which summary
coverage statistics are recorded (default = 2).</p>

<p style="margin-top: 1em">-N, &lt;nucleobaseBufferSize&gt;
<br>
The number of reference sequence nucleobases to buffer in
memory, (default = 5000).</p>

<p style="margin-top: 1em">-OF, &lt;outputFormat&gt; <br>
&rsquo;default&rsquo;: standard fastq output;
&rsquo;debug_nucleobases(_nuc|read_ids)&rsquo;:
debugging.</p>

<p style="margin-top: 1em">-RCNF,
&lt;readsContainingNfilter&gt; <br>
Filter out no &quot;N-containing&quot; reads (0),
&quot;all-N&quot; reads (1), &quot;at-least-1-N&quot; reads
(2), (default = 0).</p>

<p style="margin-top: 1em">-RL, &lt;readLength&gt; <br>
The length of each read, (default = 76).</p>

<p style="margin-top: 1em">-SE, &lt;simulateErrorInRead&gt;
<br>
Whether to simulate error in the read based on the quality
scores, (default = false).</p>

<p style="margin-top: 1em">-TLM, &lt;templateLengthMean&gt;
<br>
The mean DNA template length, (default = 210).</p>

<p style="margin-top: 1em">-TLSD, &lt;templateLengthSD&gt;
<br>
The standard deviation of the DNA template length, (default
= 60).</p>

<p style="margin-top: 1em">-URQS,
&lt;useRealQualityScores&gt; <br>
Whether to use real quality scores from existing fastq files
or set all to the maximum, (default = false).</p>

<p style="margin-top: 1em">-X, &lt;xStart&gt; <br>
The first read&rsquo;s X coordinate, (default = 1000).</p>

<p style="margin-top: 1em">-Y, &lt;yStart&gt; <br>
The first read&rsquo;s Y coordinate, (default = 1000).</p>

<p style="margin-top: 1em">BUGS <br>
Any bugs should be reported to
Matthew.Frampton@icr.ac.uk</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">artfastqgenerator 0.0.20150519
February 2016 ARTFASTQGENERATOR(1)</p>
<hr>
</body>
</html>
