<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ARTFASTQGENERATOR(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ARTFASTQGENERATOR(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">ARTFASTQGENERATOR(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
artfastqgenerator - outputs artificial FASTQ files derived from a reference
  genome
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>artfastqgenerator</b> <b>-O</b> &lt;outputPath&gt; <b>-R</b>
  &lt;referenceGenomePath&gt; <b>-S</b> &lt;startSequenceIdentifier&gt;
  <b>-F1</b> &lt;fastq1ForQualityScores&gt; <b>-F2</b>
  &lt;fastq2ForQualityScores&gt; <b>-CMGCS</b>
  &lt;coverageMeanGCcontentSpread&gt; <b>-CMP</b> &lt;coverageMeanPeak&gt;
  <b>-CMPGC</b> &lt;coverageMeanPeakGCcontent&gt; <b>-CSD</b> &lt;coverageSD&gt;
  <b>-E</b> &lt;endSequenceIdentifier&gt; <b>-GCC</b>
  &lt;GCcontentBasedCoverage&gt; <b>-GCR</b> &lt;GCcontentRegionSize&gt;
  <b>-L</b> &lt;logRegionStats&gt; <b>-N</b> &lt;nucleobaseBufferSize&gt;
  <b>-OF</b> &lt;outputFormat&gt; <b>-RCNF</b> &lt;readsContainingNfilter&gt;
  <b>-RL</b> &lt;readLength&gt; <b>-SE</b> &lt;simulateErrorInRead&gt;
  <b>-TLM</b> &lt;templateLengthMean&gt; <b>-TLSD</b> &lt;templateLengthSD&gt;
  <b>-URQS</b> &lt;useRealQualityScores&gt; <b>-X</b> &lt;xStart&gt; <b>-Y</b>
  &lt;yStart&gt;
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
ArtificialFastqGenerator takes the reference genome (in FASTA format) as input
  and outputs artificial FASTQ files in the Sanger format. It can accept Phred
  base quality scores from existing FASTQ files, and use them to simulate
  sequencing errors. Since the artificial FASTQs are derived from the reference
  genome, the reference genome provides a gold-standard for calling variants
  (Single Nucleotide Polymorphisms (SNPs) and insertions and deletions
  (indels)). This enables evaluation of a Next Generation Sequencing (NGS)
  analysis pipeline which aligns reads to the reference genome and then calls
  the variants.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Print usage help.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-O</b>, &lt;outputPath&gt;</dt>
  <dd class="It-tag">Path for the artificial fastq and log files, including
      their base name (must be specified).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b>, &lt;referenceGenomePath&gt;</dt>
  <dd class="It-tag">Reference genome sequence file, (must be specified).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b>, &lt;startSequenceIdentifier&gt;</dt>
  <dd class="It-tag">Prefix of the sequence identifier in the reference after
      which read generation should begin (must be specified).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F1</b>, &lt;fastq1ForQualityScores&gt;</dt>
  <dd class="It-tag">First fastq file to use for real quality scores, (must be
      specified if useRealQualityScores = true).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F2</b>, &lt;fastq2ForQualityScores&gt;</dt>
  <dd class="It-tag">Second fastq file to use for real quality scores, (must be
      specified if useRealQualityScores = true).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-CMGCS</b>, &lt;coverageMeanGCcontentSpread&gt;</dt>
  <dd class="It-tag">The spread of coverage mean given GC content (default =
      0.22).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-CMP</b>, &lt;coverageMeanPeak&gt;</dt>
  <dd class="It-tag">The peak coverage mean for a region (default = 37.7).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-CMPGC</b>, &lt;coverageMeanPeakGCcontent&gt;</dt>
  <dd class="It-tag">The GC content for regions with peak coverage mean (default
      = 0.45).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-CSD</b>, &lt;coverageSD&gt;</dt>
  <dd class="It-tag">The coverage standard deviation divided by the mean
      (default = 0.2).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b>, &lt;endSequenceIdentifier&gt;</dt>
  <dd class="It-tag">Prefix of the sequence identifier in the reference where
      read generation should stop, (default = end of file).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-GCC</b>, &lt;GCcontentBasedCoverage&gt;</dt>
  <dd class="It-tag">Whether nucleobase coverage is biased by GC content
      (default = true).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-GCR</b>, &lt;GCcontentRegionSize&gt;</dt>
  <dd class="It-tag">Region size in nucleobases for which to calculate GC
      content, (default = 150).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b>, &lt;logRegionStats&gt;</dt>
  <dd class="It-tag">The region size as a multiple of <b>-NBS</b> for which
      summary coverage statistics are recorded (default = 2).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b>, &lt;nucleobaseBufferSize&gt;</dt>
  <dd class="It-tag">The number of reference sequence nucleobases to buffer in
      memory, (default = 5000).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-OF</b>, &lt;outputFormat&gt;</dt>
  <dd class="It-tag">
    <br/>
     'default': standard fastq output; 'debug_nucleobases(_nuc|read_ids)':
      debugging.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-RCNF</b>, &lt;readsContainingNfilter&gt;</dt>
  <dd class="It-tag">Filter out no &quot;N-containing&quot; reads (0),
      &quot;all-N&quot; reads (1), &quot;at-least-1-N&quot; reads (2), (default
      = 0).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-RL</b>, &lt;readLength&gt;</dt>
  <dd class="It-tag">The length of each read, (default = 76).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-SE</b>, &lt;simulateErrorInRead&gt;</dt>
  <dd class="It-tag">Whether to simulate error in the read based on the quality
      scores, (default = false).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-TLM</b>, &lt;templateLengthMean&gt;</dt>
  <dd class="It-tag">The mean DNA template length, (default = 210).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-TLSD</b>, &lt;templateLengthSD&gt;</dt>
  <dd class="It-tag">The standard deviation of the DNA template length, (default
      = 60).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-URQS</b>, &lt;useRealQualityScores&gt;</dt>
  <dd class="It-tag">Whether to use real quality scores from existing fastq
      files or set all to the maximum, (default = false).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-X</b>, &lt;xStart&gt;</dt>
  <dd class="It-tag">The first read's X coordinate, (default = 1000).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Y</b>, &lt;yStart&gt;</dt>
  <dd class="It-tag">The first read's Y coordinate, (default = 1000).</dd>
</dl>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Any bugs should be reported to Matthew.Frampton@icr.ac.uk
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2016</td>
    <td class="foot-os">artfastqgenerator 0.0.20150519</td>
  </tr>
</table>
</body>
</html>
