<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:53:02 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>AUGUSTUS(1) AUGUSTUS(1)</p>

<p style="margin-top: 1em">NAME <br>
augustus - a gene prediction tool</p>

<p style="margin-top: 1em">SYNOPSIS <br>
augustus [parameters] --species=SPECIES queryfilename</p>

<p style="margin-top: 1em">OPTIONS <br>
The mandatory option &rsquo;queryfilename&rsquo; specifies
the filename (including relative path) to the file
containing the query sequence(s) in FASTA format.</p>

<p style="margin-top: 1em">SPECIES is an identifier for the
species. Use --species=help to see a list.</p>

<p style="margin-top: 1em">Further parameters: <br>
--strand=both, --strand=forward or --strand=backward <br>
define the strand to search on</p>

<p style="margin-top: 1em">--genemodel=genemodel <br>
where genemodel is one of:</p>

<p style="margin-top: 1em">&Acirc;&middot; partial - allow
prediction of incomplete genes at the sequence boundaries
(default)</p>

<p style="margin-top: 1em">&Acirc;&middot; intronless -
only predict single-exon genes like in prokaryotes and some
eukaryotes</p>

<p style="margin-top: 1em">&Acirc;&middot; complete - only
predict complete genes</p>

<p style="margin-top: 1em">&Acirc;&middot; atleastone -
predict at least one complete gene</p>

<p style="margin-top: 1em">&Acirc;&middot; exactlyone -
predict exactly one complete gene</p>

<p style="margin-top: 1em">--singlestrand=true <br>
predict genes independently on each strand, allow
overlapping genes on opposite strands. This option is turned
off by default.</p>

<p style="margin-top: 1em">*--hintsfile=hintsfilenam*e <br>
When this option is used the prediction considering hints
(extrinsic information) is turned on. hintsfilename contains
the hints in gff format.</p>

<p style="margin-top: 1em">--AUGUSTUS_CONFIG_PATH=path <br>
path to config directory (if not specified as environment
variable)</p>


<p style="margin-top: 1em">--alternatives-from-evidence=true/false
<br>
report alternative transcripts when they are suggested by
hints</p>


<p style="margin-top: 1em">--alternatives-from-sampling=true/false
<br>
report alternative transcripts generated through
probabilistic sampling</p>

<p style="margin-top: 1em">--maxtracks=n <br>
For a description of these parameters see section 4 of
README.TXT.</p>

<p style="margin-top: 1em">--proteinprofile=filename <br>
When this option is used the prediction will consider the
protein profile provided as parameter. The protein profile
extension is described in section 7 of README.TXT.</p>

<p style="margin-top: 1em">--progress=true <br>
show a progressmeter</p>

<p style="margin-top: 1em">--gff3=on/off <br>
output in gff3 format</p>

<p style="margin-top: 1em">--predictionStart=A,
--predictionEnd=B <br>
A and B define the range of the sequence for which
predictions should be found.</p>

<p style="margin-top: 1em">--UTR=on/off <br>
predict the untranslated regions in addition to the coding
sequence. This currently works only for a subset of
species.</p>

<p style="margin-top: 1em">--noInFrameStop=true/false <br>
Do not report transcripts with in-frame stop codons.
Otherwise, intron-spanning stop codons could occur. Default:
false</p>

<p style="margin-top: 1em">--noprediction=true/false <br>
If true and input is in genbank format, no prediction is
made. Useful for getting the annotated protein
sequences.</p>

<p style="margin-top: 1em">--uniqueGeneId=true/false <br>
If true, output gene identifyers like this: seqname.gN</p>

<p style="margin-top: 1em">--paramlist <br>
Show full list of supported parameters.</p>

<p style="margin-top: 1em">AUTHORS <br>
AUGUSTUS was written by M. Stanke, O. Keller, S.
K&Atilde;&para;nig, L. Gerischer and L. Romoth.</p>

<p style="margin-top: 1em">ADDITIONAL DOCUMENTATION <br>
An exhaustive documentation can be found in the file
/usr/share/augustus/README.TXT.</p>
 
<p style="margin-top: 1em">AUGUSTUS(1)</p>
<hr>
</body>
</html>
