<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>AUGUSTUS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">AUGUSTUS(1)</td>
    <td class="head-vol">&#x00A0;</td>
    <td class="head-rtitle">AUGUSTUS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
augustus - a gene prediction tool
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>augustus</b> [parameters] --species=SPECIES queryfilename
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
The mandatory option 'queryfilename' specifies the filename (including relative
  path) to the file containing the query sequence(s) in FASTA format.
<div style="height: 1.00em;">&#x00A0;</div>
SPECIES is an identifier for the species. Use <b>--species=help</b> to see a
  list.
<h2 class="Ss" title="Ss" id="Further_parameters:"><a class="selflink" href="#Further_parameters:">Further
  parameters:</a></h2>
<b>--strand=both</b>, <b>--strand=forward</b> or <b>--strand=backward</b>
<div style="margin-left: 4.00ex;">define the strand to search on</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--genemodel=genemodel</b>
<div style="margin-left: 4.00ex;">where genemodel is one of:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;partial - allow prediction of
  incomplete genes at the sequence boundaries (default)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;intronless - only predict single-exon
  genes like in prokaryotes and some eukaryotes</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;complete - only predict complete
  genes</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;atleastone - predict at least one
  complete gene</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;exactlyone - predict exactly one
  complete gene</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--singlestrand=true</b>
<div style="margin-left: 4.00ex;">predict genes independently on each strand,
  allow overlapping genes on opposite strands. This option is turned off by
  default.</div>
<div style="height: 1.00em;">&#x00A0;</div>
*--hintsfile=hintsfilenam*e
<div style="margin-left: 4.00ex;">When this option is used the prediction
  considering hints (extrinsic information) is turned on. hintsfilename contains
  the hints in gff format.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--AUGUSTUS_CONFIG_PATH=path</b>
<div style="margin-left: 4.00ex;">path to config directory (if not specified as
  environment variable)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--alternatives-from-evidence=true/false</b>
<div style="margin-left: 4.00ex;">report alternative transcripts when they are
  suggested by hints</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--alternatives-from-sampling=true/false</b>
<div style="margin-left: 4.00ex;">report alternative transcripts generated
  through probabilistic sampling</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--maxtracks=n</b>
<div style="margin-left: 4.00ex;">For a description of these parameters see
  section 4 of README.TXT.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--proteinprofile=filename</b>
<div style="margin-left: 4.00ex;">When this option is used the prediction will
  consider the protein profile provided as parameter. The protein profile
  extension is described in section 7 of README.TXT.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--progress=true</b>
<div style="margin-left: 4.00ex;">show a progressmeter</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--gff3=on/off</b>
<div style="margin-left: 4.00ex;">output in gff3 format</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--predictionStart=A</b>, <b>--predictionEnd=B</b>
<div style="margin-left: 4.00ex;">A and B define the range of the sequence for
  which predictions should be found.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--UTR=on/off</b>
<div style="margin-left: 4.00ex;">predict the untranslated regions in addition
  to the coding sequence. This currently works only for a subset of
  species.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--noInFrameStop=true/false</b>
<div style="margin-left: 4.00ex;">Do not report transcripts with in-frame stop
  codons. Otherwise, intron-spanning stop codons could occur. Default:
  false</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--noprediction=true/false</b>
<div style="margin-left: 4.00ex;">If true and input is in genbank format, no
  prediction is made. Useful for getting the annotated protein sequences.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--uniqueGeneId=true/false</b>
<div style="margin-left: 4.00ex;">If true, output gene identifyers like this:
  seqname.gN</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--paramlist</b>
<div style="margin-left: 4.00ex;">Show full list of supported parameters.</div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
AUGUSTUS was written by M. Stanke, O. Keller, S. K&#x00F6;nig, L. Gerischer and
  L. Romoth.
<h1 class="Sh" title="Sh" id="ADDITIONAL_DOCUMENTATION"><a class="selflink" href="#ADDITIONAL_DOCUMENTATION">ADDITIONAL
  DOCUMENTATION</a></h1>
An exhaustive documentation can be found in the file
  /usr/share/augustus/README.TXT.</div>
<table class="foot">
  <tr>
    <td class="foot-date"></td>
    <td class="foot-os">&#x00A0;</td>
  </tr>
</table>
</body>
</html>
