<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>AXE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">AXE(1)</td>
    <td class="head-vol">axe</td>
    <td class="head-rtitle">AXE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
axe - axe Documentation
<div style="height: 1.00em;">&#x00A0;</div>
Axe is a read de-multiplexer, useful in situations where sequence reads contain
  the barcodes that uniquely distinguish samples. Axe uses a rapid and accurate
  algorithm based on hamming mismatch tries to competitively match the prefix of
  a sequencing read against a set of barcodes. Axe supports combinatorial
  barcoding schemes.
<div style="height: 1.00em;">&#x00A0;</div>
Contents:
<h1 class="Sh" title="Sh" id="AXE_USAGE"><a class="selflink" href="#AXE_USAGE">AXE
  USAGE</a></h1>
<b>NOTE:</b>
<div style="margin-left: 0.00ex;">
<div style="margin-left: 3.50ex;">For arcane reasons, the name of the <b>axe</b>
  binary changed to <b>axe-demux</b> with version 0.3.0. Apologies for the
  inconvenience, this was required to make <b>axe</b> installable in Debian and
  its derivatives. Command-line usage did not change.</div>
<br/>
</div>
<br/>
<div style="height: 1.00em;">&#x00A0;</div>
Axe has several usage modes. The primary distinction is between the two
  alternate barcoding schemes, single and combinatorial barcoding. Single
  barcode matching is used when only the first read contains barcode sequences.
  Combinatorial barcoding is used when both reads in a read pair contain
  independent (typically different) barcode sequences.
<div style="height: 1.00em;">&#x00A0;</div>
For concise reference, the command-line usage of <b>axe-demux</b> is reproduced
  below:
<div style="margin-left: 0.00ex;">
<div style="margin-left: 3.50ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<pre>
USAGE:
axe-demux [-mzc2pt] -b (-f [-r] | -i) (-F [-R] | -I)
axe-demux -h
axe-demux -v
<div class="Pp"></div>
OPTIONS:
    -m, --mismatch	Maximum hamming distance mismatch. [int, default 1]
    -z, --ziplevel	Gzip compression level, or 0 for plain text [int, default 0]
    -c, --combinatorial	Use combinatorial barcode matching. [flag, default OFF]
    -p, --permissive	Don't error on barcode mismatch confict, matching only
                    	exactly for conficting barcodes. [flag, default OFF]
    -2, --trim-r2	Trim barcode from R2 read as well as R1. [flag, default OFF]
    -b, --barcodes	Barcode file. See --help for example. [file]
    -f, --fwd-in	Input forward read. [file]
    -F, --fwd-out	Output forward read prefix. [file]
    -r, --rev-in	Input reverse read. [file]
    -R, --rev-out	Output reverse read prefix. [file]
    -i, --ilfq-in	Input interleaved paired reads. [file]
    -I, --ilfq-out	Output interleaved paired reads prefix. [file]
    -t, --table-file	Output a summary table of demultiplexing statistics to file. [file]
    -h, --help		Print this usage plus additional help.
    -V, --version	Print version string.
    -v, --verbose	Be more verbose. Additive, -vv is more vebose than -v.
    -q, --quiet		Be very quiet.
<div class="Pp"></div>
</pre>
</div>
<br/>
</div>
<br/>
<h2 class="Ss" title="Ss" id="Inputs_and_Outputs"><a class="selflink" href="#Inputs_and_Outputs">Inputs
  and Outputs</a></h2>
Regardless of read mode, three input and output schemes are supported:
  single-end reads, paired reads (separate R1 and R2 files) and interleaved
  paired reads (one file, with R1 and R2 as consecutive reads). If single end
  reads are inputted, they must be output as single end reads. If either paired
  or interleaved paired reads are read, they can be output as either paired
  reads or interleaved paired reads. This applies to both successfully
  de-multiplexed reads and reads that could not be de-multiplexed.
<div style="height: 1.00em;">&#x00A0;</div>
The <b>-z</b> flag can be used to specify that outputs should be compressed
  using gzip compression. The <b>-z</b> flag takes an integer argument between 0
  (the default) and 9, where 0 indicates plain text output ( <b>gzopen</b> mode
  &quot;wT&quot;), and 1-9 indicate that the respective compression level should
  be used, where 1 is fastest and 9 is most compact.
<div style="height: 1.00em;">&#x00A0;</div>
The output flags should be prefixes that are used to generate the output file
  name based on the barcode's (or barcode pair's) ID. The names are generated
  as: <b>prefix</b> + <b>_</b> + <b>barcode ID</b> + <b>_</b> + <b>read
  number</b> + <b>.extension</b>. The output file for reads that could not be
  demultiplexed is <b>prefix</b> + <b>_</b> + <b>unknown</b> + <b>_</b> +
  <b>read number</b> + <b>.extension</b>. The read number is omitted unless the
  paired read file scheme is used, and is &quot;il&quot; for interleaved output.
  The extension is &quot;fastq&quot;; &quot;.gz&quot; is appended to the
  extension if the <b>-z</b> flag is used.
<div style="margin-left: 0.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>The corresponding CLI flags are:</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 7.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag"><b>-f</b> and <b>-F</b>: Single end or paired R1 file input
      and output respectively.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag"><b>-r</b> and <b>-R</b>: Paired R2 file input and
    output.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag"><b>-i</b> and <b>-I</b>: Interleaved paired input and
      output.</dd>
</dl>
</div>
<br/>
</div>
<br/>
<h2 class="Ss" title="Ss" id="The_barcode_file"><a class="selflink" href="#The_barcode_file">The
  barcode file</a></h2>
The barcode file is a tab-separated file with an optional header. It is
  mandatory, and is always supplied using the <b>-b</b> command line flag. The
  exact format is dependent on barcoding mode, and is described further in the
  sections below. If a header is present, the header line must start with either
  <i>Barcode</i> or <b>barcode</b>, or it will be interpreted as a barcode line,
  leading to a parsing error. Any line starting with ';' or '#' is ignored,
  allowing comments to be added in line with barcodes. Please ensure that the
  software used to produce the barcode uses ASCII encoding, and does not insert
  a Byte-order Mark (BoM) as many text editors can silently use Unicode-based
  encoding schemes. I recommend the use of <i>LibreOffice Calc</i> (part of a
  free and open source office suite) to generate barcode tables; Microsoft Excel
  can also be used.
<h2 class="Ss" title="Ss" id="Mismatch_level_selection"><a class="selflink" href="#Mismatch_level_selection">Mismatch
  level selection</a></h2>
Independent of barcode mode, the <b>-m</b> flag is used to select the maximum
  allowable hamming distance between a read's prefix and a barcode to be
  considered as a match. As &quot;mutated&quot; barcodes must be unique, a
  hamming distance of one is the default as typically barcodes are designed to
  differ by a hamming distance of at least two. Optionally, (using the <b>-p</b>
  flag), axe will allow selective mismatch levels, where, if clashes are
  observed, the barcode will only be matched exactly. This allows one to process
  datasets with barcodes that don't have a sufficiently high distance between
  them.
<h2 class="Ss" title="Ss" id="Single_barcode_mode"><a class="selflink" href="#Single_barcode_mode">Single
  barcode mode</a></h2>
Single barcode mode is the default mode of operation. Barcodes are matched
  against read one (hereafter the forward read), and the barcode is trimmed from
  only the forward read, unless the <b>-2</b> command line flag is given, in
  which case a prefix the same length as the matched barcode is also trimmed
  from the second or reverse read. Note that sequence of this second read is not
  checked before trimming.
<div style="height: 1.00em;">&#x00A0;</div>
In single barcode mode, the barcode file has two columns: <b>Barcode</b> and
  <b>ID</b>.
<h2 class="Ss" title="Ss" id="Combinatorial_barcode_mode"><a class="selflink" href="#Combinatorial_barcode_mode">Combinatorial
  barcode mode</a></h2>
Combinatorial barcode mode is activated by giving the <b>-c</b> flag on the
  command line. Forward read barcodes are matched against the forward read, and
  reverse read barcodes are matched against the reverse read. The optimal
  barcodes are selected independently, and the barcode pair is selected from
  these two barcodes. The respective barcodes are trimmed from both reads; the
  <b>-2</b> command line flag has no effect in combinatorial barcode mode.
<div style="height: 1.00em;">&#x00A0;</div>
In combinatorial barcode mode, the barcode file has three columns:
  <b>Barcode1</b>, <b>Barcode2</b> and <b>ID</b>. Individual barcodes can occur
  many times within the forward and reverse barcodes, but barcode pairs must be
  unique combinations.
<h2 class="Ss" title="Ss" id="The_Demultipexing_Statistics_File"><a class="selflink" href="#The_Demultipexing_Statistics_File">The
  Demultipexing Statistics File</a></h2>
The <b>-t</b> option allows the output of per-sample read counts to a
  tab-separated file. The file will have a header describing its format, and
  includes a line for unbarcoded reads.
<h1 class="Sh" title="Sh" id="AXE'S_MATCHING_ALGORITHM"><a class="selflink" href="#AXE'S_MATCHING_ALGORITHM">AXE'S
  MATCHING ALGORITHM</a></h1>
Axe uses an algorithm based on longest-prefix-in-trie matching to match a
  variable length from the start of each read against a set of 'mutated'
  barcodes.
<h2 class="Ss" title="Ss" id="Hamming_distance_matching"><a class="selflink" href="#Hamming_distance_matching">Hamming
  distance matching</a></h2>
While for most applications in high-throughput sequencing hamming distances are
  a frowned-upon metric, it is typical for HTS read barcodes to be designed to
  tolerate a certain level of hamming mismatches. Given these sequences are
  short and typically occur at the 5' end of reads, insertions and deletions
  rarely need be considered, and the increased rate of assignment of reads with
  many errors is offset by the risk of falsely assigning barcodes to an
  incorrect sample. In any case, reads with more than 1-2 sequencing errors in
  their first several bases are likely to be poor quality, and will simply be
  filtered out during downstream quality control.
<h2 class="Ss" title="Ss" id="Hamming_mismatch_tries"><a class="selflink" href="#Hamming_mismatch_tries">Hamming
  mismatch tries</a></h2>
Typically, reads are matched to a set of barcodes by calculating the hamming
  distance between the barcode, and the first l bases of a read for a barcode of
  length l. The &quot;correct&quot; barcode is then selected by recording either
  the barcode with the lowest hamming distance to the read (competitive
  matching) or by simply accepting the first barcode with a hamming distance
  below a certain threshold. These approaches are both very computationally
  expensive, and can have lower accuracy than the algorithm I propose.
  Additionally, implementations of these methods rarely handle barcodes of
  differing length and combinatorial barcoding well, if at all.
<div style="height: 1.00em;">&#x00A0;</div>
Central to Axe's algorithm is the concept of hamming-mismatch tries. A trie is a
  N-ary tree for an N letter alphabet. In the case of high-throughput sequencing
  reads, we have the alphabet <b>AGCT</b>, corresponding to the four nucleotides
  of DNA, plus <b>N</b>, used to represent ambiguous base calls. Instead of
  matching each barcode to each read, we pre-calculate all allowable sequences
  at each mismatch level, and store these in level-wise tries. For example, to
  match to a hamming distance of 2, we create three tries: One containing all
  barcodes, verbatim, and two tries where every sequence within a hamming
  distance of 1 and 2 of each barcode respectively. Hereafter, these tries are
  referred to as the 0, 1 and 2-mm tries, for a hamming distance (mismatch) of
  0, 1 and 2. Then, we find the longest prefix in each sequence read in the 0mm
  trie. If this prefix is not a valid leaf in the 0mm trie, we find the longest
  prefix in the 1mm trie, and so on for all tries in ascending order. If no
  prefix of the read is a complete sequence in any trie, the read is assigned to
  an &quot;non-barcoded&quot; output file.
<div style="height: 1.00em;">&#x00A0;</div>
This algorithm ensures optimal barcode matching in many ways, but is also
  extremely fast. In situations with barcodes of differing length, we ensure
  that the <i>longest</i> acceptable barcode at a given hamming distance is
  chosen; assuming that sequence is random after the barcode, the probability of
  false assignments using this method is low. We also ensure that short perfect
  matches are preferred to longer inexact matches, as we firstly only consider
  barcodes with no error, then 1 error, and so on. This ensures that reads with
  barcodes that are followed by random sequence that happens to inexactly match
  a longer barcode in the set are not falsely assigned to this longer barcode.
<div style="height: 1.00em;">&#x00A0;</div>
The speed of this algorithm is largely due to the constant time matching
  algorithm with respect to the number of barcodes to match. The time taken to
  match each read is proportional instead to the length of the barcodes, as for
  a barcode of length l, at most l + 1 trie level descents are required to find
  an entry in the trie. As this length is more-or-less constant and small, the
  overall complexity of axe's algorithm is O(n) for n reads, as opposed to O(nm)
  for n reads and m barcodes as is typical for traditional matching algorithms
<div style="margin-left: 0.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">genindex</dd>
</dl>
</div>
<br/>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Kevin Murray
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
2016, Kevin Murray</div>
<table class="foot">
  <tr>
    <td class="foot-date">December 1, 2016</td>
    <td class="foot-os">0.3.2</td>
  </tr>
</table>
</body>
</html>
