<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BAITFILTER-V1.0.5(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BAITFILTER-V1.0.5(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">BAITFILTER-V1.0.5(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
BaitFilter-v1.0.5 - manual page for BaitFilter-v1.0.5
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
USAGE:
<dl class="Bl-tag">
  <dt class="It-tag">./BaitFilter-v1.0.5</dt>
  <dd class="It-tag"><b>-i</b> &lt;string&gt; [-o &lt;string&gt;] [-c
      &lt;string&gt;] [-m &lt;string&gt;] [--blast-second-hit-evalue
      &lt;floating point number&gt;] [--blast-first-hit-evalue &lt;floating
      point number&gt;] [--blast-min-hit-coverage-of-baits-in-tiling-stack
      &lt;floating point number&gt;] [--ref-blast-db &lt;string&gt;]
      [--blast-extra-commandline &lt;string&gt;] [--blast-evalue-cutoff
      &lt;floating point number&gt;] [-B &lt;string&gt;] [-t &lt;positive
      integer&gt;] [--ID-prefix &lt;string&gt;] [--verbosity &lt;unsigned
      integer&gt;] [-S] [--] [--version] [-h]</dd>
</dl>
<div class="Pp"></div>
Where:
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-i</b>
  &lt;string&gt;, <b>--input-bait-file-name</b> &lt;string&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag">(required)</dt>
  <dd class="It-tag">Name of the input bait locus file. This is the bait
    file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">obtained from the Bait-Fisher program.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  &lt;string&gt;, <b>--output-bait-file-name</b> &lt;string&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Name of the output bait file. This is the file the contains
      the filtered bait loci and the baits.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-c</b>
  &lt;string&gt;, <b>--convert</b> &lt;string&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Allows the user to produce the final output file for the
      bait producing company. In this mode, BaitFilter reads the input bait file
      and instead of doing a filtering step, it produces a costumn bait file
      that can be uploaded to the baits producing company. In order to avoid
      confuction a filtering step cannot be done in the same run as the
      conversion. If you want to filter a bait file and convert the output, you
      will need to call this program twice, first to do the filtering and second
      to do the conversion. Allowed conversion parameters currently are:
      &quot;Agilent-Germany&quot;.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">New output formats can be added upon request. Please
      contact the author: Christoph Mayer, Email: Mayer Christoph
      &lt;c.mayer.zfmk@uni-bonn.de&gt;</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-m</b>
  &lt;string&gt;, <b>--mode</b> &lt;string&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Appart form the input file this is the most essential
      option. This option specifies which filter mode Bait-Filter uses. (See the
      user manual for more details):</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ab&quot;:</dt>
  <dd class="It-tag">Retain only the best bait locus for each alignemntfile</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">(e.g. gene) when using the optimality criterion</dt>
  <dd class="It-tag">to</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">minimize the total number of required baits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;as&quot;:</dt>
  <dd class="It-tag">Retain only the best bait locus for each alignemntfile</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">(e.g. gene) when using the optimality criterion</dt>
  <dd class="It-tag">to</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">maximize the number of sequences the result is based
    on.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;fb&quot;:</dt>
  <dd class="It-tag">Retain only the best bait locus for each feature (e.g.
    CDS)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">when using the optimality criterion</dt>
  <dd class="It-tag">to minimize the total</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">number of required baits. Only applicable if alignment
      cutting has been used in Bait-Fisher.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;fs&quot;:</dt>
  <dd class="It-tag">Retain only the best bait locus for each feature (e.g.
    CDS)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">when using the optimality criterion</dt>
  <dd class="It-tag">to maximize the number</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">of sequences the result is based on. Only applicable if
      alignment cutting has been used in Bait-Fisher.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&quot;blast-a&quot;: Remove all bait loci of ALIGNMENTs for
      which one or more baits have multiple good hits to a reference
    genome.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&quot;blast-f&quot;: Remove all bait loci of FEATUREs for
      which one or more baits have multiple good hits to a reference
    genome.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&quot;blast-l&quot;: Remove bait LOCI that contain a bait
      that hos multiple good hits to a reference genome.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;thin-b&quot;:</dt>
  <dd class="It-tag">Thin out a bait file to every Nth bait region, by
    finding</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">the start position that minimizes the number of baits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;thin-s&quot;:</dt>
  <dd class="It-tag">Thin out a bait file to every Nth bait region, by
    finding</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">the start position that maximizes the number of
    sequences.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--blast-second-hit-evalue</b>
  &lt;floating point number&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Maximum evalue for the second hit. A bait is characterized
      to bind ambiguously, if we have two good hits. This option is the evalue
      threshold for the second hit.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--blast-first-hit-evalue</b>
  &lt;floating point number&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Maximum evalue for the first hit. A bait is characterized
      to bind ambiguously, if we have two good hits. This option is the evalue
      threshold for the first hit.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--blast-min-hit-coverage-of-baits-in-tiling-stack</b>
  &lt;floating point</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">number&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Specifies a minimum hit coverage for the primary hit which
      at least one bait has to have in each tiling stack. Otherwise the bait
      region is discarded. If not specified, no hit coverage is checked. This
      parameter can only be used in conjunction with other filters. Since the
      order in which the coverage filter and the other filters are applied
      matters, the order is defined as follows: For the mode options: ab, as,
      fb, fs the coverage is checked before determining the optimal bait region.
      For the mode options: blast-a, blast-f, blast-l the hit coverage is
      checked after filtering for baits with multiple good hits to the reference
      genome.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--ref-blast-db</b>
  &lt;string&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Base name to a blast data base file. This name is passed to
      the blast command. This is the name of the fasta file of your reference
      genome. IMPORTANT: The makeblastdb program has to be called before
      starting the Bait-Filter program. makeblastdb takes the fasta file and
      creates data base files out of it.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--blast-extra-commandline</b>
  &lt;string&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">When invoking the blast command, extra commandline
      parameters can be passed to the blast command. As an example with this
      option it is possible to specifiy the number of threads the blast command
      should use.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--blast-evalue-cutoff</b>
  &lt;floating point number&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">When invoking the blast command, a default value of twice
      the <b>--</b>&lt;blast-first-hit-evalue is used. This should guarantee
      that all hits necessary for the blast filter are found. However, for the
      coverage filtering a smaller threshold might be necessary. This can be
      specified here.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-B</b>
  &lt;string&gt;, <b>--blast-executable</b> &lt;string&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Name of or path+name to the blast executable. Default:
      nblast. Minimum version number: Blast+ 2.2.x</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-t</b>
  &lt;positive integer&gt;, <b>--thinning-step-width</b> &lt;positive
  integer&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Thin out the bait file by retaining only every Nth bait
      region. This option specified the step width N. If one of the moded
      thin-b, thin-s is active, this option is required, otherwise it is not
      allowed to set this parameter.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--ID-prefix</b>
  &lt;string&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">In the conversion mode Agilent-Germany each converted file
      should get a unique ProdeID prefix, since even among multiple files,
      ProbeIDs are not allowed to be identical. This this option the user is
      able to specifiy a prefix string to all probe IDs in this file.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--verbosity</b>
  &lt;unsigned integer&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">The verbosity option controls the amount of information
      Bait-Filter writes to the console while running. 0: report only error
      messages that lead to exiting the program. 1: report also wanrings, 2:
      report also progress, 3: report more detailed progress, &gt;10: debug
      output. 10000: all possible dignostic output.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-S</b>,
  <b>--stats</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Compute stats for the input file and report these. This
      mode is automatically used for all modes specified with <b>-m</b> or the
      conversion mode specified with <b>-c</b>.The purpose of the <b>-S</b>
      option is to compute stats without having to filter or convert the input
      file. In particular, the <b>-S</b> mode does not require specifying an
      output file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">This option has no effect if combined with the <b>-m</b> or
      <b>-c</b> modes.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--</b>,
  <b>--ignore_rest</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Ignores the rest of the labeled arguments following this
      flag.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--version</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Displays version information and exits.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-h</b>,
  <b>--help</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Displays usage information and exits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">This program can be used to produce the final output file
      for creating baits, or it can be used to filter bait loci obtained from
      the Bait-Fisher program according to different criteria. The bait file
      produced by BaitFisher computes a tiling desing for each possible starting
      position. The purpuse of BaitFilter is to determine for each target
      alignment/gene/feature the optimal bait region. As input it requires the
      bait file generated by the BaitFisher program or a BaitFile generated by a
      previous filtering run of BaitFilter. This bait file is specified with the
      <b>-i</b> command line parameter (see below). Furthermore, the user has to
      specifiy an output file name with the <b>-o</b> parameter and a filter
      mode with the <b>-m</b> parameter.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">To convert a file to a customn output format, see the
      <b>-c</b> option below.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">To compute stats of an input file, see the <b>-S</b> option
      below.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">The different filter modes provided by BaitFilter are the
      following:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">1a) Retain only the best bait locus per alignment file.
      Criterion: Minimize number of required baits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">1b) Retain only the best bait locus per alignment file.
      Criterion: Maximize number of sequenences.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">2a) Retain only best bait locus per feature (requires that
      features were selected in Bait-Fisher). Criterion: Minimize number of
      required baits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">2b) Retain only best bait locus per feature (requires that
      features were selected in Bait-Fisher). Criterion: Maximize number of
      sequenences.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">3) Use a blast search of the bait sequences against a
      reference genome to detect putative non-unique target loci. Non unique
      target sites will have multiple good hits against the reference genome.
      Furthermore, a minimum coverage of the best blast hit of bait sequence
      against the genome can be specified. Note that all blast modes require
      additional command line parameters! These modes remove bait regions for
      which multiple blast hits where found. Different versions of this mode are
      available:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">3a) If a single bait is not unique, remove all bait regions
      from the current gene.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">3b) If a single bait is not unique, remove all bait regions
      from the current feature (if applicable).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">3c) If a single bait is not unique, remove only the bait
      region that contains this bait.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">4) Thin out the given bait file: Retain only every Nth bait
      region, where N has to be specified by the user. Two submodes are
      available:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">4a) Thin out bait regions by retaining only every Nth bait
      region in a bait file. The starting offset will by chosen such that the
      number of required baits is minimized.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">4b) Thin out bait regions by retaining only every Nth bait
      region in a bait file. The starting offset will by chosen such that the
      number of sequences the result is baised on is maximized.</dd>
</dl>
<div class="Pp"></div>
./BaitFilter-v1.0.5 version: 1.0.5
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The full documentation for <b>BaitFilter-v1.0.5</b> is maintained as a Texinfo
  manual. If the <b>info</b> and <b>BaitFilter-v1.0.5</b> programs are properly
  installed at your site, the command
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>info BaitFilter-v1.0.5</b></dd>
</dl>
<div class="Pp"></div>
should give you access to the complete manual.</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">BaitFilter-v1.0.5 </td>
  </tr>
</table>
</body>
</html>
