<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:54:05 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BAITFILTER-V1.0.5(1) User Commands
BAITFILTER-V1.0.5(1)</p>

<p style="margin-top: 1em">NAME <br>
BaitFilter-v1.0.5 - manual page for BaitFilter-v1.0.5</p>

<p style="margin-top: 1em">DESCRIPTION <br>
USAGE:</p>

<p style="margin-top: 1em">./BaitFilter-v1.0.5 <br>
-i &lt;string&gt; [-o &lt;string&gt;] [-c &lt;string&gt;]
[-m &lt;string&gt;] [--blast-second-hit-evalue &lt;floating
point number&gt;] [--blast-first-hit-evalue &lt;floating
point number&gt;] <br>
[--blast-min-hit-coverage-of-baits-in-tiling-stack
&lt;floating point number&gt;] [--ref-blast-db
&lt;string&gt;] [--blast-extra-commandline &lt;string&gt;]
[--blast-evalue-cutoff &lt;floating <br>
point number&gt;] [-B &lt;string&gt;] [-t &lt;positive
integer&gt;] [--ID-prefix &lt;string&gt;] [--verbosity
&lt;unsigned integer&gt;] [-S] [--] [--version] [-h]</p>

<p style="margin-top: 1em">Where:</p>

<p style="margin-top: 1em">-i &lt;string&gt;,
--input-bait-file-name &lt;string&gt;</p>

<p style="margin-top: 1em">(required) <br>
Name of the input bait locus file. This is the bait file</p>

<p style="margin-top: 1em">obtained from the Bait-Fisher
program.</p>

<p style="margin-top: 1em">-o &lt;string&gt;,
--output-bait-file-name &lt;string&gt;</p>

<p style="margin-top: 1em">Name of the output bait file.
This is the file the contains the filtered bait loci and the
baits.</p>

<p style="margin-top: 1em">-c &lt;string&gt;, --convert
&lt;string&gt;</p>

<p style="margin-top: 1em">Allows the user to produce the
final output file for the bait producing company. In this
mode, BaitFilter reads the input bait file and instead of
doing a filtering step, <br>
it produces a costumn bait file that can be uploaded to the
baits producing company. In order to avoid confuction a
filtering step cannot be done in the same run as the <br>
conversion. If you want to filter a bait file and convert
the output, you will need to call this program twice, first
to do the filtering and second to do the conversion. <br>
Allowed conversion parameters currently are:
&quot;Agilent-Germany&quot;.</p>

<p style="margin-top: 1em">New output formats can be added
upon request. Please contact the author: Christoph Mayer,
Email: Mayer Christoph &lt;c.mayer.zfmk@uni-bonn.de&gt;</p>

<p style="margin-top: 1em">-m &lt;string&gt;, --mode
&lt;string&gt;</p>

<p style="margin-top: 1em">Appart form the input file this
is the most essential option. This option specifies which
filter mode Bait-Filter uses. (See the user manual for more
details):</p>

<p style="margin-top: 1em">&quot;ab&quot;: Retain only the
best bait locus for each alignemntfile</p>

<p style="margin-top: 1em">(e.g. gene) when using the
optimality criterion <br>
to</p>

<p style="margin-top: 1em">minimize the total number of
required baits.</p>

<p style="margin-top: 1em">&quot;as&quot;: Retain only the
best bait locus for each alignemntfile</p>

<p style="margin-top: 1em">(e.g. gene) when using the
optimality criterion <br>
to</p>

<p style="margin-top: 1em">maximize the number of sequences
the result is based on.</p>

<p style="margin-top: 1em">&quot;fb&quot;: Retain only the
best bait locus for each feature (e.g. CDS)</p>

<p style="margin-top: 1em">when using the optimality
criterion <br>
to minimize the total</p>

<p style="margin-top: 1em">number of required baits. Only
applicable if alignment cutting has been used in
Bait-Fisher.</p>

<p style="margin-top: 1em">&quot;fs&quot;: Retain only the
best bait locus for each feature (e.g. CDS)</p>

<p style="margin-top: 1em">when using the optimality
criterion <br>
to maximize the number</p>

<p style="margin-top: 1em">of sequences the result is based
on. Only applicable if alignment cutting has been used in
Bait-Fisher.</p>

<p style="margin-top: 1em">&quot;blast-a&quot;: Remove all
bait loci of ALIGNMENTs for which one or more baits have
multiple good hits to a reference genome.</p>

<p style="margin-top: 1em">&quot;blast-f&quot;: Remove all
bait loci of FEATUREs for which one or more baits have
multiple good hits to a reference genome.</p>

<p style="margin-top: 1em">&quot;blast-l&quot;: Remove bait
LOCI that contain a bait that hos multiple good hits to a
reference genome.</p>

<p style="margin-top: 1em">&quot;thin-b&quot;: <br>
Thin out a bait file to every Nth bait region, by
finding</p>

<p style="margin-top: 1em">the start position that
minimizes the number of baits.</p>

<p style="margin-top: 1em">&quot;thin-s&quot;: <br>
Thin out a bait file to every Nth bait region, by
finding</p>

<p style="margin-top: 1em">the start position that
maximizes the number of sequences.</p>

<p style="margin-top: 1em">--blast-second-hit-evalue
&lt;floating point number&gt;</p>

<p style="margin-top: 1em">Maximum evalue for the second
hit. A bait is characterized to bind ambiguously, if we have
two good hits. This option is the evalue threshold for the
second hit.</p>

<p style="margin-top: 1em">--blast-first-hit-evalue
&lt;floating point number&gt;</p>

<p style="margin-top: 1em">Maximum evalue for the first
hit. A bait is characterized to bind ambiguously, if we have
two good hits. This option is the evalue threshold for the
first hit.</p>


<p style="margin-top: 1em">--blast-min-hit-coverage-of-baits-in-tiling-stack
&lt;floating point</p>

<p style="margin-top: 1em">number&gt;</p>

<p style="margin-top: 1em">Specifies a minimum hit coverage
for the primary hit which at least one bait has to have in
each tiling stack. Otherwise the bait region is discarded.
If not specified, no <br>
hit coverage is checked. This parameter can only be used in
conjunction with other filters. Since the order in which the
coverage filter and the other filters are applied <br>
matters, the order is defined as follows: For the mode
options: ab, as, fb, fs the coverage is checked before
determining the optimal bait region. For the mode options:
<br>
blast-a, blast-f, blast-l the hit coverage is checked after
filtering for baits with multiple good hits to the reference
genome.</p>

<p style="margin-top: 1em">--ref-blast-db
&lt;string&gt;</p>

<p style="margin-top: 1em">Base name to a blast data base
file. This name is passed to the blast command. This is the
name of the fasta file of your reference genome. IMPORTANT:
The makeblastdb <br>
program has to be called before starting the Bait-Filter
program. makeblastdb takes the fasta file and creates data
base files out of it.</p>

<p style="margin-top: 1em">--blast-extra-commandline
&lt;string&gt;</p>

<p style="margin-top: 1em">When invoking the blast command,
extra commandline parameters can be passed to the blast
command. As an example with this option it is possible to
specifiy the number of <br>
threads the blast command should use.</p>

<p style="margin-top: 1em">--blast-evalue-cutoff
&lt;floating point number&gt;</p>

<p style="margin-top: 1em">When invoking the blast command,
a default value of twice the --&lt;blast-first-hit-evalue is
used. This should guarantee that all hits necessary for the
blast filter are <br>
found. However, for the coverage filtering a smaller
threshold might be necessary. This can be specified
here.</p>

<p style="margin-top: 1em">-B &lt;string&gt;,
--blast-executable &lt;string&gt;</p>

<p style="margin-top: 1em">Name of or path+name to the
blast executable. Default: nblast. Minimum version number:
Blast+ 2.2.x</p>

<p style="margin-top: 1em">-t &lt;positive integer&gt;,
--thinning-step-width &lt;positive integer&gt;</p>

<p style="margin-top: 1em">Thin out the bait file by
retaining only every Nth bait region. This option specified
the step width N. If one of the moded thin-b, thin-s is
active, this option is <br>
required, otherwise it is not allowed to set this
parameter.</p>

<p style="margin-top: 1em">--ID-prefix &lt;string&gt;</p>

<p style="margin-top: 1em">In the conversion mode
Agilent-Germany each converted file should get a unique
ProdeID prefix, since even among multiple files, ProbeIDs
are not allowed to be identical. <br>
This this option the user is able to specifiy a prefix
string to all probe IDs in this file.</p>

<p style="margin-top: 1em">--verbosity &lt;unsigned
integer&gt;</p>

<p style="margin-top: 1em">The verbosity option controls
the amount of information Bait-Filter writes to the console
while running. 0: report only error messages that lead to
exiting the program. 1: <br>
report also wanrings, 2: report also progress, 3: report
more detailed progress, &gt;10: debug output. 10000: all
possible dignostic output.</p>

<p style="margin-top: 1em">-S, --stats</p>

<p style="margin-top: 1em">Compute stats for the input file
and report these. This mode is automatically used for all
modes specified with -m or the conversion mode specified
with -c.The purpose of <br>
the -S option is to compute stats without having to filter
or convert the input file. In particular, the -S mode does
not require specifying an output file.</p>

<p style="margin-top: 1em">This option has no effect if
combined with the -m or -c modes.</p>

<p style="margin-top: 1em">--, --ignore_rest</p>

<p style="margin-top: 1em">Ignores the rest of the labeled
arguments following this flag.</p>

<p style="margin-top: 1em">--version</p>

<p style="margin-top: 1em">Displays version information and
exits.</p>

<p style="margin-top: 1em">-h, --help</p>

<p style="margin-top: 1em">Displays usage information and
exits.</p>

<p style="margin-top: 1em">This program can be used to
produce the final output file for creating baits, or it can
be used to filter bait loci obtained from the Bait-Fisher
program according to dif&acirc; <br>
ferent criteria. The bait file produced by BaitFisher
computes a tiling desing for each possible starting
position. The purpuse of BaitFilter is to determine for each
tar&acirc; <br>
get alignment/gene/feature the optimal bait region. As input
it requires the bait file generated by the BaitFisher
program or a BaitFile generated by a previous filtering <br>
run of BaitFilter. This bait file is specified with the -i
command line parameter (see below). Furthermore, the user
has to specifiy an output file name with the -o param&acirc;
<br>
eter and a filter mode with the -m parameter.</p>

<p style="margin-top: 1em">To convert a file to a customn
output format, see the -c option below.</p>

<p style="margin-top: 1em">To compute stats of an input
file, see the -S option below.</p>

<p style="margin-top: 1em">The different filter modes
provided by BaitFilter are the following:</p>

<p style="margin-top: 1em">1a) Retain only the best bait
locus per alignment file. Criterion: Minimize number of
required baits.</p>

<p style="margin-top: 1em">1b) Retain only the best bait
locus per alignment file. Criterion: Maximize number of
sequenences.</p>

<p style="margin-top: 1em">2a) Retain only best bait locus
per feature (requires that features were selected in
Bait-Fisher). Criterion: Minimize number of required
baits.</p>

<p style="margin-top: 1em">2b) Retain only best bait locus
per feature (requires that features were selected in
Bait-Fisher). Criterion: Maximize number of sequenences.</p>

<p style="margin-top: 1em">3) Use a blast search of the
bait sequences against a reference genome to detect putative
non-unique target loci. Non unique target sites will have
multiple good hits <br>
against the reference genome. Furthermore, a minimum
coverage of the best blast hit of bait sequence against the
genome can be specified. Note that all blast modes <br>
require additional command line parameters! These modes
remove bait regions for which multiple blast hits where
found. Different versions of this mode are available:</p>

<p style="margin-top: 1em">3a) If a single bait is not
unique, remove all bait regions from the current gene.</p>

<p style="margin-top: 1em">3b) If a single bait is not
unique, remove all bait regions from the current feature (if
applicable).</p>

<p style="margin-top: 1em">3c) If a single bait is not
unique, remove only the bait region that contains this
bait.</p>

<p style="margin-top: 1em">4) Thin out the given bait file:
Retain only every Nth bait region, where N has to be
specified by the user. Two submodes are available:</p>

<p style="margin-top: 1em">4a) Thin out bait regions by
retaining only every Nth bait region in a bait file. The
starting offset will by chosen such that the number of
required baits is minimized.</p>

<p style="margin-top: 1em">4b) Thin out bait regions by
retaining only every Nth bait region in a bait file. The
starting offset will by chosen such that the number of
sequences the result is baised <br>
on is maximized.</p>

<p style="margin-top: 1em">./BaitFilter-v1.0.5 version:
1.0.5</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation for BaitFilter-v1.0.5 is maintained
as a Texinfo manual. If the info and BaitFilter-v1.0.5
programs are properly installed at your site, the
command</p>

<p style="margin-top: 1em">info BaitFilter-v1.0.5</p>

<p style="margin-top: 1em">should give you access to the
complete manual.</p>

<p style="margin-top: 1em">BaitFilter-v1.0.5 January 2017
BAITFILTER-V1.0.5(1)</p>
<hr>
</body>
</html>
