<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BAM2HINTS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BAM2HINTS(1)</td>
    <td class="head-vol">&#x00A0;</td>
    <td class="head-rtitle">BAM2HINTS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bam2hints - convert mRNA-to-genome alignments in BAM format into a hint file for
  AUGUSTUS
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>bam2hints</b> [parameters] --in=example.bam --out=hints.gff
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
The input BAM file must be sorted by target (=genome) sequence names and within
  the sequences by begin coordinates.
<h2 class="Ss" title="Ss" id="Parameters:"><a class="selflink" href="#Parameters:">Parameters:</a></h2>
<b>--priority=n</b>/<b>-p</b>
<div style="margin-left: 4.00ex;">priority of hint group (set to 4)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--maxgaplen=n</b>/<b>-g</b>
<div style="margin-left: 4.00ex;">gaps at most this length are simply closed
  (set to 14)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--minintronlen=n</b>/<b>-m</b>
<div style="margin-left: 4.00ex;">alignments with gaps shorter than this and
  longer than maxgaplen are discarded (set to 32)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--maxintronlen=n</b>/<b>-M</b>
<div style="margin-left: 4.00ex;">alignments with longer gaps are discarded (set
  to 350000)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--MinEndBlockLen=n</b>/<b>-b</b>
<div style="margin-left: 4.00ex;">minimum length of a 'dangling' exon (set to
  8)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--maxQgaplen=n</b>/<b>-q</b>
<div style="margin-left: 4.00ex;">maximum length of gap in query (cDNA) sequence
  (set to 5)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--exonhints</b>/<b>-x</b>
<div style="margin-left: 4.00ex;">compute exonpart, exon and splice site hints
  in addition to intron hints (set to 0=Off). You should generate exonpart hints
  from RNA-Seq using wiggle (.wig) input to wig2hints.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--ep_cutoff=n</b>/<b>-e</b>
<div style="margin-left: 4.00ex;">this many bp are cut off of each exonpart hint
  at end of alignment (set to 10)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--source=s</b>/<b>-s</b>
<div style="margin-left: 4.00ex;">source identifier (set to 'E')</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--intronsonly</b>/<b>-I</b>
<div style="margin-left: 4.00ex;">only retrieve intron hints (e.g. because the
  exon(part) hints are retrieved by converting to a wig track, set to 1=On).
  Deprecated as this is the default now.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--nomult</b>/<b>-n</b>
<div style="margin-left: 4.00ex;">do not summarize multiple identical intron
  hints to a single one (set to 0=Off)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--remove_redundant</b>/<b>-r</b>
<div style="margin-left: 4.00ex;">only keep the strongest hint for a region (set
  to 0=Off)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--maxcoverage=n</b>/<b>-C</b>
<div style="margin-left: 4.00ex;">maximal number of hints at a given position
  (0: filtering deactivated). A high value causes long running time of AUGUSTUS
  in regions with thousands of cDNA alignments. (set to 0)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--ssOn</b>/<b>-S</b>
<div style="margin-left: 4.00ex;">include splice site (dss, ass) hints in output
  (set to 0=Off)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--trunkSS</b>/<b>-T</b>
<div style="margin-left: 4.00ex;">include splice sites hints from the ends of a
  truncated alignment (contig too short, set to 0=Off)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--score=f</b>/<b>-s</b>
<div style="margin-left: 4.00ex;">fill this number in in the score column (set
  to 0)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--maxgenelen=n</b>/<b>-G</b>
<div style="margin-left: 4.00ex;">alignments of the same clone are considered to
  be of the same gene if not separated by more than this (set to 400000).
  Alignments that span more than this are ignored, but better filter long
  introns through an alignment program.</div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
AUGUSTUS was written by M. Stanke, O. Keller, S. K&#x00F6;nig, L. Gerischer and
  L. Romoth.
<h1 class="Sh" title="Sh" id="ADDITIONAL_DOCUMENTATION"><a class="selflink" href="#ADDITIONAL_DOCUMENTATION">ADDITIONAL
  DOCUMENTATION</a></h1>
An exhaustive documentation can be found in the file
  /usr/share/augustus/README.TXT.</div>
<table class="foot">
  <tr>
    <td class="foot-date"></td>
    <td class="foot-os">&#x00A0;</td>
  </tr>
</table>
</body>
</html>
