<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:54:07 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BAM2HINTS(1) BAM2HINTS(1)</p>

<p style="margin-top: 1em">NAME <br>
bam2hints - convert mRNA-to-genome alignments in BAM format
into a hint file for AUGUSTUS</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bam2hints [parameters] --in=example.bam --out=hints.gff</p>

<p style="margin-top: 1em">OPTIONS <br>
The input BAM file must be sorted by target (=genome)
sequence names and within the sequences by begin
coordinates.</p>

<p style="margin-top: 1em">Parameters: <br>
--priority=n/-p <br>
priority of hint group (set to 4)</p>

<p style="margin-top: 1em">--maxgaplen=n/-g <br>
gaps at most this length are simply closed (set to 14)</p>

<p style="margin-top: 1em">--minintronlen=n/-m <br>
alignments with gaps shorter than this and longer than
maxgaplen are discarded (set to 32)</p>

<p style="margin-top: 1em">--maxintronlen=n/-M <br>
alignments with longer gaps are discarded (set to
350000)</p>

<p style="margin-top: 1em">--MinEndBlockLen=n/-b <br>
minimum length of a &rsquo;dangling&rsquo; exon (set to
8)</p>

<p style="margin-top: 1em">--maxQgaplen=n/-q <br>
maximum length of gap in query (cDNA) sequence (set to
5)</p>

<p style="margin-top: 1em">--exonhints/-x <br>
compute exonpart, exon and splice site hints in addition to
intron hints (set to 0=Off). You should generate exonpart
hints from RNA-Seq using wiggle (.wig) input to <br>
wig2hints.</p>

<p style="margin-top: 1em">--ep_cutoff=n/-e <br>
this many bp are cut off of each exonpart hint at end of
alignment (set to 10)</p>

<p style="margin-top: 1em">--source=s/-s <br>
source identifier (set to &rsquo;E&rsquo;)</p>

<p style="margin-top: 1em">--intronsonly/-I <br>
only retrieve intron hints (e.g. because the exon(part)
hints are retrieved by converting to a wig track, set to
1=On). Deprecated as this is the default now.</p>

<p style="margin-top: 1em">--nomult/-n <br>
do not summarize multiple identical intron hints to a single
one (set to 0=Off)</p>

<p style="margin-top: 1em">--remove_redundant/-r <br>
only keep the strongest hint for a region (set to 0=Off)</p>

<p style="margin-top: 1em">--maxcoverage=n/-C <br>
maximal number of hints at a given position (0: filtering
deactivated). A high value causes long running time of
AUGUSTUS in regions with thousands of cDNA alignments. (set
<br>
to 0)</p>

<p style="margin-top: 1em">--ssOn/-S <br>
include splice site (dss, ass) hints in output (set to
0=Off)</p>

<p style="margin-top: 1em">--trunkSS/-T <br>
include splice sites hints from the ends of a truncated
alignment (contig too short, set to 0=Off)</p>

<p style="margin-top: 1em">--score=f/-s <br>
fill this number in in the score column (set to 0)</p>

<p style="margin-top: 1em">--maxgenelen=n/-G <br>
alignments of the same clone are considered to be of the
same gene if not separated by more than this (set to
400000). Alignments that span more than this are ignored,
but <br>
better filter long introns through an alignment program.</p>

<p style="margin-top: 1em">AUTHORS <br>
AUGUSTUS was written by M. Stanke, O. Keller, S.
K&Atilde;&para;nig, L. Gerischer and L. Romoth.</p>

<p style="margin-top: 1em">ADDITIONAL DOCUMENTATION <br>
An exhaustive documentation can be found in the file
/usr/share/augustus/README.TXT.</p>
 
<p style="margin-top: 1em">BAM2HINTS(1)</p>
<hr>
</body>
</html>
