<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:54:07 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BAMTOOLS(1) User Commands BAMTOOLS(1)</p>

<p style="margin-top: 1em">NAME <br>
bamtools - toolkit for manipulating BAM (genome alignment)
files</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bamtools [--help] COMMAND [ARGS]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
BamTools facilitates research analysis and data management
using BAM files. It copes with the enormous amount of data
produced by current sequencing technologies that is
typi&acirc; <br>
cally stored in compressed, binary formats that are not
easily handled by the text-based parsers commonly used in
bioinformatics research.</p>

<p style="margin-top: 1em">OPTIONS <br>
convert <br>
Converts between BAM and a number of other formats</p>

<p style="margin-top: 1em">count Prints number of
alignments in BAM file(s)</p>

<p style="margin-top: 1em">coverage <br>
Prints coverage statistics from the input BAM file</p>

<p style="margin-top: 1em">filter Filters BAM file(s) by
user-specified criteria</p>

<p style="margin-top: 1em">header Prints BAM header
information</p>

<p style="margin-top: 1em">index Generates index for BAM
file</p>

<p style="margin-top: 1em">merge Merge multiple BAM files
into single file</p>

<p style="margin-top: 1em">random Select random alignments
from existing BAM file(s), intended more as a testing
tool.</p>

<p style="margin-top: 1em">resolve <br>
Resolves paired-end reads (marking the IsProperPair flag as
needed)</p>

<p style="margin-top: 1em">revert Removes duplicate marks
and restores original base qualities</p>

<p style="margin-top: 1em">sort Sorts the BAM file
according to some criteria</p>

<p style="margin-top: 1em">split Splits a BAM file on
user-specified property, creating a new BAM output file for
each value found</p>

<p style="margin-top: 1em">stats Prints some basic
statistics from input BAM file(s)</p>

<p style="margin-top: 1em">See &rsquo;bamtools help
COMMAND&rsquo; for more information on a specific
command.</p>

<p style="margin-top: 1em">bamtools 2.3.0+dfsg April 2014
BAMTOOLS(1)</p>
<hr>
</body>
</html>
