<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BAMTOOLS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BAMTOOLS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">BAMTOOLS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bamtools - toolkit for manipulating BAM (genome alignment) files
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>bamtools</b> [--help] COMMAND [ARGS]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
BamTools facilitates research analysis and data management using BAM files. It
  copes with the enormous amount of data produced by current sequencing
  technologies that is typically stored in compressed, binary formats that are
  not easily handled by the text-based parsers commonly used in bioinformatics
  research.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">convert</dt>
  <dd class="It-tag">Converts between BAM and a number of other formats</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">count</dt>
  <dd class="It-tag">Prints number of alignments in BAM file(s)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">coverage</dt>
  <dd class="It-tag">Prints coverage statistics from the input BAM file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">filter</dt>
  <dd class="It-tag">Filters BAM file(s) by user-specified criteria</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">header</dt>
  <dd class="It-tag">Prints BAM header information</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">index</dt>
  <dd class="It-tag">Generates index for BAM file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">merge</dt>
  <dd class="It-tag">Merge multiple BAM files into single file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">random</dt>
  <dd class="It-tag">Select random alignments from existing BAM file(s),
      intended more as a testing tool.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">resolve</dt>
  <dd class="It-tag">Resolves paired-end reads (marking the IsProperPair flag as
      needed)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">revert</dt>
  <dd class="It-tag">Removes duplicate marks and restores original base
      qualities</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">sort</dt>
  <dd class="It-tag">Sorts the BAM file according to some criteria</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">split</dt>
  <dd class="It-tag">Splits a BAM file on user-specified property, creating a
      new BAM output file for each value found</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">stats</dt>
  <dd class="It-tag">Prints some basic statistics from input BAM file(s)</dd>
</dl>
<div class="Pp"></div>
See 'bamtools help COMMAND' for more information on a specific command.</div>
<table class="foot">
  <tr>
    <td class="foot-date">April 2014</td>
    <td class="foot-os">bamtools 2.3.0+dfsg</td>
  </tr>
</table>
</body>
</html>
