<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BCFTOOLS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BCFTOOLS(1)</td>
    <td class="head-vol">&#x00A0;</td>
    <td class="head-rtitle">BCFTOOLS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bcftools - utilities for variant calling and manipulating VCFs and BCFs.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>bcftools</b> [--version|--version-only] [--help] [<i>COMMAND</i>]
  [<i>OPTIONS</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
BCFtools is a set of utilities that manipulate variant calls in the Variant Call
  Format (VCF) and its binary counterpart BCF. All commands work transparently
  with both VCFs and BCFs, both uncompressed and BGZF-compressed.
<div style="height: 1.00em;">&#x00A0;</div>
Most commands accept VCF, bgzipped VCF and BCF with filetype detected
  automatically even when streaming from a pipe. Indexed VCF and BCF will work
  in all situations. Un-indexed VCF and BCF and streams will work in most, but
  not all situations. In general, whenever multiple VCFs are read
  simultaneously, they must be indexed and therefore also compressed.
<div style="height: 1.00em;">&#x00A0;</div>
BCFtools is designed to work on a stream. It regards an input file &quot;-&quot;
  as the standard input (stdin) and outputs to the standard output (stdout).
  Several commands can thus be combined with Unix pipes.
<h2 class="Ss" title="Ss" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h2>
This manual page was last updated <b>2016-04-18 14:18 BST</b> and refers to
  bcftools git version <b>1.3-36-g47e811c+</b>.
<h2 class="Ss" title="Ss" id="BCF1"><a class="selflink" href="#BCF1">BCF1</a></h2>
The BCF1 format output by versions of samtools &lt;= 0.1.19 is <b>not</b>
  compatible with this version of bcftools. To read BCF1 files one can use the
  view command from old versions of bcftools packaged with samtools versions
  &lt;= 0.1.19 to convert to VCF, which can then be read by this version of
  bcftools.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    samtools-0.1.19/bcftools/bcftools view file.bcf1 | bcftools view
</pre>
</div>
<h2 class="Ss" title="Ss" id="VARIANT_CALLING"><a class="selflink" href="#VARIANT_CALLING">VARIANT
  CALLING</a></h2>
See <i>bcftools call</i> for variant calling from the output of the <i>samtools
  mpileup</i> command. In versions of samtools &lt;= 0.1.19 calling was done
  with <i>bcftools view</i>. Users are now required to choose between the old
  samtools calling model ( <i>-c/--consensus-caller</i>) and the new
  multiallelic calling model ( <i>-m/--multiallelic-caller</i>). The
  multiallelic calling model is recommended for most tasks.
<h1 class="Sh" title="Sh" id="LIST_OF_COMMANDS"><a class="selflink" href="#LIST_OF_COMMANDS">LIST
  OF COMMANDS</a></h1>
For a full list of available commands, run <b>bcftools</b> without arguments.
  For a full list of available options, run <b>bcftools</b> <i>COMMAND</i>
  without arguments.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>annotate</b> .. edit VCF files,
  add or remove annotations</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>call</b> .. SNP/indel calling
  (former &quot;view&quot;)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>cnv</b> .. Copy Number Variation
  caller</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>concat</b> .. concatenate VCF/BCF
  files from the same set of samples</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>consensus</b> .. create consensus
  sequence by applying VCF variants</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>convert</b> .. convert VCF/BCF to
  other formats and back</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>filter</b> .. filter VCF/BCF files
  using fixed thresholds</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>gtcheck</b> .. check sample
  concordance, detect sample swaps and contamination</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>index</b> .. index VCF/BCF</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>isec</b> .. intersections of
  VCF/BCF files</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>merge</b> .. merge VCF/BCF files
  files from non-overlapping sample sets</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>norm</b> .. normalize indels</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>plugin</b> .. run user-defined
  plugin</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>polysomy</b> .. detect
  contaminations and whole-chromosome aberrations</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>query</b> .. transform VCF/BCF
  into user-defined formats</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>reheader</b> .. modify VCF/BCF
  header, change sample names</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>roh</b> .. identify runs of
  homo/auto-zygosity</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>stats</b> .. produce VCF/BCF stats
  (former vcfcheck)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>view</b> .. subset, filter and
  convert VCF and BCF files</div>
<h1 class="Sh" title="Sh" id="LIST_OF_SCRIPTS"><a class="selflink" href="#LIST_OF_SCRIPTS">LIST
  OF SCRIPTS</a></h1>
Some helper scripts are bundled with the bcftools code.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022; <b>plot-vcfstats</b> .. plots the
  output of <b>stats</b></div>
<h1 class="Sh" title="Sh" id="COMMANDS_AND_OPTIONS"><a class="selflink" href="#COMMANDS_AND_OPTIONS">COMMANDS
  AND OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Common_Options"><a class="selflink" href="#Common_Options">Common
  Options</a></h2>
The following options are common to many bcftools commands. See usage for
  specific commands to see if they apply.
<div class="Pp"></div>
<i>FILE</i>
<div style="margin-left: 4.00ex;">Files can be both VCF or BCF, uncompressed or
  BGZF-compressed. The file &quot;-&quot; is interpreted as standard input. Some
  tools may require tabix- or CSI-indexed files.</div>
<div class="Pp"></div>
<b>-c, --collapse</b>
  <i>snps</i>|<i>indels</i>|<i>both</i>|<i>all</i>|<i>some</i>|
  <i>none</i>|<i>id</i>
<div style="margin-left: 4.00ex;">Controls how to treat records with duplicate
  positions and defines compatible records across multiple input files. Here by
  &quot;compatible&quot; we mean records which should be considered as identical
  by the tools. For example, when performing line intersections, the desire may
  be to consider as identical all sites with matching positions ( <b>bcftools
  isec -c</b> <i>all</i>), or only sites with matching variant type (<b>bcftools
  isec -c</b> <i>snps</i>&#x00A0; <b>-c</b> <i>indels</i>), or only sites with
  all alleles identical ( <b>bcftools isec -c</b> <i>none</i>).
<div class="Pp"></div>
<i>none</i>
<div style="margin-left: 4.00ex;">only records with identical REF and ALT
  alleles are compatible</div>
<div class="Pp"></div>
<i>some</i>
<div style="margin-left: 4.00ex;">only records where some subset of ALT alleles
  match are compatible</div>
<div class="Pp"></div>
<i>all</i>
<div style="margin-left: 4.00ex;">all records are compatible, regardless of
  whether the ALT alleles match or not. In the case of records with the same
  position, only the first will be considered and appear on output.</div>
<div class="Pp"></div>
<i>snps</i>
<div style="margin-left: 4.00ex;">any SNP records are compatible, regardless of
  whether the ALT alleles match or not. For duplicate positions, only the first
  SNP record will be considered and appear on output.</div>
<div class="Pp"></div>
<i>indels</i>
<div style="margin-left: 4.00ex;">all indel records are compatible, regardless
  of whether the REF and ALT alleles match or not. For duplicate positions, only
  the first indel record will be considered and appear on output.</div>
<div class="Pp"></div>
<i>both</i>
<div style="margin-left: 4.00ex;">abbreviation of &quot;<b>-c</b>
  <i>indels</i>&#x00A0; <b>-c</b> <i>snps</i>&quot;</div>
<div class="Pp"></div>
<i>id</i>
<div style="margin-left: 4.00ex;">only records with identical ID column are
  compatible. Supported by <b>bcftools merge</b> only.</div>
</div>
<div class="Pp"></div>
<b>-f, --apply-filters</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">Skip sites where FILTER column does not
  contain any of the strings listed in <i>LIST</i>. For example, to include only
  sites which have no filters set, use <b>-f</b> <i>.,PASS</i>.</div>
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">Do not append version and command line
  information to the output VCF header.</div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">When output consists of a single stream, write
  it to <i>FILE</i> rather than to standard output, where it is written by
  default.</div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">Output compressed BCF (<i>b</i>), uncompressed
  BCF ( <i>u</i>), compressed VCF (<i>z</i>), uncompressed VCF (<i>v</i>). Use
  the -Ou option when piping between bcftools subcommands to speed up
  performance by removing unnecessary compression/decompression and
  VCF&#x2190;&#x2192;BCF conversion.</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">Comma-separated list of regions, see also
  <b>-R, --regions-file</b>. Note that <b>-r</b> cannot be used in combination
  with <b>-R</b>.</div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">Regions can be specified either on command
  line or in a VCF, BED, or tab-delimited file (the default). The columns of the
  tab-delimited file are: CHROM, POS, and, optionally, POS_TO, where positions
  are 1-based and inclusive. The columns of the tab-delimited BED file are also
  CHROM, POS and POS_TO (trailing columns are ignored), but coordinates are
  0-based, half-open. To indicate that a file be treated as BED rather than the
  1-based tab-delimited file, the file must have the &quot;.bed&quot; or
  &quot;.bed.gz&quot; suffix (case-insensitive). Uncompressed files are stored
  in memory, while bgzip-compressed and tabix-indexed region files are streamed.
  Note that sequence names must match exactly, &quot;chr20&quot; is not the same
  as &quot;20&quot;. Also note that chromosome ordering in <i>FILE</i> will be
  respected, the VCF will be processed in the order in which chromosomes first
  appear in <i>FILE</i>. However, within chromosomes, the VCF will always be
  processed in ascending genomic coordinate order no matter what order they
  appear in <i>FILE</i>. Note that overlapping regions in <i>FILE</i> can result
  in duplicated out of order positions in the output. This option requires
  indexed VCF/BCF files. Note that <b>-R</b> cannot be used in combination with
  <b>-r</b>.</div>
<div class="Pp"></div>
<b>-s, --samples</b> [^]<i>LIST</i>
<div style="margin-left: 4.00ex;">Comma-separated list of samples to include or
  exclude if prefixed with &quot;^&quot;. Note that in general tags such as
  INFO/AC, INFO/AN, etc are not updated to correspond to the subset samples.
  <b>bcftools view</b> is the exception where some tags will be updated (unless
  the <b>-I, --no-update</b> option is used; see <b>bcftools view</b>
  documentation). To use updated tags for the subset in another command one can
  pipe from <b>view</b> into that command. For example:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools view -Ou -s sample1,sample2 file.vcf | bcftools query -f %INFO/AC\t%INFO/AN\n
</pre>
</div>
<div class="Pp"></div>
<b>-S, --samples-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">File of sample names to include or exclude if
  prefixed with &quot;^&quot;. One sample per line. See also the note above for
  the <b>-s, --samples</b> option. The command <b>bcftools call</b> accepts an
  optional second column indicating ploidy (0, 1 or 2) or sex (as defined by
  <b>--ploidy</b>, for example &quot;F&quot; or &quot;M&quot;), and can parse
  also PED files. If the second column is not present, the sex &quot;F&quot; is
  assumed. With <b>bcftools call</b><b> -C</b> <i>trio</i>, PED file is
  expected. File formats examples:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    sample1    1
    sample2    2
    sample3    2
<div class="Pp"></div>
  or
<div class="Pp"></div>
    sample1    M
    sample2    F
    sample3    F
<div class="Pp"></div>
  or a .ped file (here is shown a minimum working example, the first column is
  ignored and the last indicates sex: 1=male, 2=female)
<div class="Pp"></div>
    ignored daughterA fatherA motherA 2
    ignored sonB fatherB motherB 1
</pre>
</div>
<div class="Pp"></div>
<b>-t, --targets</b>
  [^]<i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">Similar as <b>-r, --regions</b>, but the next
  position is accessed by streaming the whole VCF/BCF rather than using the
  tbi/csi index. Both <b>-r</b> and <b>-t</b> options can be applied
  simultaneously: <b>-r</b> uses the index to jump to a region and <b>-t</b>
  discards positions which are not in the targets. Unlike <b>-r</b>, targets can
  be prefixed with &quot;^&quot; to request logical complement. For example,
  &quot;^X,Y,MT&quot; indicates that sequences X, Y and MT should be skipped.
  Yet another difference between the two is that <b>-r</b> checks both start and
  end positions of indels, whereas <b>-t</b> checks start positions only. Note
  that <b>-t</b> cannot be used in combination with <b>-T</b>.</div>
<div class="Pp"></div>
<b>-T, --targets-file</b> [^]<i>FILE</i>
<div style="margin-left: 4.00ex;">Same <b>-t, --targets</b>, but reads regions
  from a file. Note that <b>-T</b> cannot be used in combination with
  <b>-t</b>.</div>
<div class="Pp"></div>
<div style="margin-left: 4.00ex;">With the <b>call -C</b> <i>alleles</i>
  command, third column of the targets file must be comma-separated list of
  alleles, starting with the reference allele. Note that the file must be
  compressed and index. Such a file can be easily created from a VCF
  using:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools query -f'%CHROM\t%POS\t%REF,%ALT\n' file.vcf | bgzip -c &gt; als.tsv.gz &amp;&amp; tabix -s1 -b2 -e2 als.tsv.gz
</pre>
</div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">Number of output compression threads to use in
  addition to main thread. Only used when <i>--output-type</i> is <i>b</i> or
  <i>z</i>. Default: 0.</div>
<h2 class="Ss" title="Ss" id="bcftools_annotate_[OPTIONS]_FILE"><a class="selflink" href="#bcftools_annotate_[OPTIONS]_FILE">bcftools
  annotate <i>[OPTIONS]</i> <i>FILE</i></a></h2>
Add or remove annotations.
<div class="Pp"></div>
<b>-a, --annotations</b> <i>file</i>
<div style="margin-left: 4.00ex;">Bgzip-compressed and tabix-indexed file with
  annotations. The file can be VCF, BED, or a tab-delimited file with mandatory
  columns CHROM, POS (or, alternatively, FROM and TO), optional columns REF and
  ALT, and arbitrary number of annotation columns. BED files are expected to
  have the &quot;.bed&quot; or &quot;.bed.gz&quot; suffix (case-insensitive),
  otherwise a tab-delimited file is assumed. Note that in case of tab-delimited
  file, the coordinates POS, FROM and TO are one-based and inclusive. When REF
  and ALT are present, only matching VCF records will be annotated. When
  multiple ALT alleles are present in the annotation file (given as
  comma-separated list of alleles), at least one must match one of the alleles
  in the corresponding VCF record. Similarly, at least one alternate allele from
  a multi-allelic VCF record must be present in the annotation file. Note that
  flag types, such as &quot;INFO/FLAG&quot;, can be annotated by including a
  field with the value &quot;1&quot; to set the flag, &quot;0&quot; to remove
  it, or &quot;.&quot; to keep existing flags. See also <b>-c, --columns</b> and
  <b>-h, --header-lines</b>.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    # Sample annotation file with columns CHROM, POS, STRING_TAG, NUMERIC_TAG
    1  752566  SomeString      5
    1  798959  SomeOtherString 6
    # etc.
</pre>
</div>
<div class="Pp"></div>
<b>-c, --columns</b> <i>list</i>
<div style="margin-left: 4.00ex;">Comma-separated list of columns or tags to
  carry over from the annotation file (see also <b>-a, --annotations</b>). If
  the annotation file is not a VCF/BCF, <i>list</i> describes the columns of the
  annotation file and must include CHROM, POS (or, alternatively, FROM and TO),
  and optionally REF and ALT. Unused columns which should be ignored can be
  indicated by &quot;-&quot;. If the annotation file is a VCF/BCF, only the
  edited columns/tags must be present and their order does not matter. The
  columns ID, QUAL, FILTER, INFO and FORMAT can be edited, where INFO tags can
  be written both as &quot;INFO/TAG&quot; or simply &quot;TAG&quot;, and FORMAT
  tags can be written as &quot;FORMAT/TAG&quot; or &quot;FMT/TAG&quot;. To carry
  over all INFO annotations, use &quot;INFO&quot;. To add all INFO annotations
  except &quot;TAG&quot;, use &quot;^INFO/TAG&quot;. By default, existing values
  are replaced. To add annotations without overwriting existing values (that is,
  to add missing tags or add values to existing tags with missing values), use
  &quot;+TAG&quot; instead of &quot;TAG&quot;. To append to existing values
  (rather than replacing or leaving untouched), use &quot;=TAG&quot; (instead of
  &quot;TAG&quot; or &quot;+TAG&quot;). To replace only existing values without
  modifying missing annotations, use &quot;-TAG&quot;. If the annotation file is
  not a VCF/BCF, all new annotations must be defined via <b>-h,
  --header-lines</b>.</div>
<div class="Pp"></div>
<b>-e, --exclude</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">exclude sites for which <i>EXPRESSION</i> is
  true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-h, --header-lines</b> <i>file</i>
<div style="margin-left: 4.00ex;">Lines to append to the VCF header, see also
  <b>-c, --columns</b> and <b>-a, --annotations</b>. For example:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    ##INFO=&lt;ID=NUMERIC_TAG,Number=1,Type=Integer,Description=&quot;Example header line&quot;&gt;
    ##INFO=&lt;ID=STRING_TAG,Number=1,Type=String,Description=&quot;Yet another header line&quot;&gt;
</pre>
</div>
<div class="Pp"></div>
<b>-I, --set-id</b> [+]<i>FORMAT</i>
<div style="margin-left: 4.00ex;">assign ID on the fly. The format is the same
  as in the <b>query</b> command (see below). By default all existing IDs are
  replaced. If the format string is preceded by &quot;+&quot;, only missing IDs
  will be set. For example, one can use</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools annotate --set-id +'%CHROM\_%POS\_%REF\_%FIRST_ALT' file.vcf
</pre>
</div>
<div class="Pp"></div>
<b>-i, --include</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">include only sites for which <i>EXPRESSION</i>
  is true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-m, --mark-sites</b> <i>TAG</i>
<div style="margin-left: 4.00ex;">annotate sites which are present
  (&quot;+&quot;) or absent (&quot;-&quot;) in the <b>-a</b> file with a new
  INFO/TAG flag</div>
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--rename-chrs</b> <i>file</i>
<div style="margin-left: 4.00ex;">rename chromosomes according to the map in
  <i>file</i>, with &quot;old_name new_name\n&quot; pairs separated by
  whitespaces, each on a separate line.</div>
<div class="Pp"></div>
<b>-s, --samples</b> [^]<i>LIST</i>
<div style="margin-left: 4.00ex;">subset of samples to annotate, see also
  <b>Common Options</b></div>
<div class="Pp"></div>
<b>-S, --samples-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">subset of samples to annotate. If the samples
  are named differently in the target VCF and the <b>-a, --annotations</b> VCF,
  the name mapping can be given as &quot;src_name dst_name\n&quot;, separated by
  whitespaces, each pair on a separate line.</div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-x, --remove</b> <i>list</i>
<div style="margin-left: 4.00ex;">List of annotations to remove. Use
  &quot;FILTER&quot; to remove all filters or &quot;FILTER/SomeFilter&quot; to
  remove a specific filter. Similarly, &quot;INFO&quot; can be used to remove
  all INFO tags and &quot;FORMAT&quot; to remove all FORMAT tags except GT. To
  remove all INFO tags except &quot;FOO&quot; and &quot;BAR&quot;, use
  &quot;^INFO/FOO,INFO/BAR&quot; (and similarly for FORMAT and FILTER).
  &quot;INFO&quot; can be abbreviated to &quot;INF&quot; and &quot;FORMAT&quot;
  to &quot;FMT&quot;.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>Examples:</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    # Remove three fields
    bcftools annotate -x ID,INFO/DP,FORMAT/DP file.vcf.gz
<div class="Pp"></div>
    # Remove all INFO fields and all FORMAT fields except for GT and PL
    bcftools annotate -x INFO,^FORMAT/GT,FORMAT/PL file.vcf
<div class="Pp"></div>
    # Add ID, QUAL and INFO/TAG, not replacing TAG if already present
    bcftools annotate -a src.bcf -c ID,QUAL,+TAG dst.bcf
<div class="Pp"></div>
    # Carry over all INFO and FORMAT annotations except FORMAT/GT
    bcftools annotate -a src.bcf -c INFO,^FORMAT/GT dst.bcf
<div class="Pp"></div>
    # Annotate from a tab-delimited file with six columns (the fifth is ignored),
    # first indexing with tabix. The coordinates are 1-based.
    tabix -s1 -b2 -e2 annots.tab.gz
    bcftools annotate -a annots.tab.gz -h annots.hdr -c CHROM,POS,REF,ALT,-,TAG file.vcf
<div class="Pp"></div>
    # Annotate from a tab-delimited file with regions (1-based coordinates, inclusive)
    tabix -s1 -b2 -e3 annots.tab.gz
    bcftools annotate -a annots.tab.gz -h annots.hdr -c CHROM,FROM,TO,TAG inut.vcf
<div class="Pp"></div>
    # Annotate from a bed file (0-based coordinates, half-closed, half-open intervals)
    bcftools annotate -a annots.bed.gz -h annots.hdr -c CHROM,FROM,TO,TAG input.vcf
</pre>
</div>
<h2 class="Ss" title="Ss" id="bcftools_cnv_[OPTIONS]_FILE"><a class="selflink" href="#bcftools_cnv_[OPTIONS]_FILE">bcftools
  cnv <i>[OPTIONS]</i> <i>FILE</i></a></h2>
Copy number variation caller, requires a VCF annotated with the
  Illumina&#x2019;s B-allele frequency (BAF) and Log R Ratio intensity (LRR)
  values. The HMM considers the following copy number states: CN 2 (normal), 1
  (single-copy loss), 0 (complete loss), 3 (single-copy gain).
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>General Options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-c, --control-sample</b> <i>string</i>
<div style="margin-left: 4.00ex;">optional control sample name. If given,
  pairwise calling is performed and the <b>-P</b> option can be used</div>
<div class="Pp"></div>
<b>-f, --AF-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">read allele frequencies from a tab-delimited
  file with the columns CHR,POS,REF,ALT,AF</div>
<div class="Pp"></div>
*-o, --output-dir <i>path</i>
<div style="margin-left: 4.00ex;">output directory</div>
<div class="Pp"></div>
*-p, --plot-threshold <i>float</i>
<div style="margin-left: 4.00ex;">call <b>matplotlib</b> to produce plots for
  chromosomes with quality at least <i>float</i>, useful for visual inspection
  of the calls. With <b>-p 0</b>, plots for all chromosomes will be generated.
  If not given, a <b>matplotlib</b> script will be created but not called.</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --query-sample</b> <i>string</i>
<div style="margin-left: 4.00ex;">query samply name</div>
<div class="Pp"></div>
<b>-t, --targets</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>HMM Options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-a, --aberrant</b> <i>float</i>[,<i>float</i>]
<div style="margin-left: 4.00ex;">fraction of aberrant cells in query and
  control. The hallmark of duplications and contaminations is the BAF value of
  heterozygous markers which is dependent on the fraction of aberrant cells.
  Sensitivity to smaller fractions of cells can be increased by setting
  <b>-a</b> to a lower value. Note however, that this comes at the cost of
  increased false discovery rate.</div>
<div class="Pp"></div>
<b>-b, --BAF-weight</b> <i>float</i>
<div style="margin-left: 4.00ex;">relative contribution from BAF</div>
<div class="Pp"></div>
<b>d, --BAF-dev</b> <i>float</i>[,<i>float</i>]
<div style="margin-left: 4.00ex;">expected BAF deviation in query and control,
  i.e. the noise observed in the data.</div>
<div class="Pp"></div>
<b>-e, --err-prob</b> <i>float</i>
<div style="margin-left: 4.00ex;">uniform error probability</div>
<div class="Pp"></div>
<b>-l, --LRR-weight</b> <i>float</i>
<div style="margin-left: 4.00ex;">relative contribution from LRR. With noisy
  data, this option can have big effect on the number of calls produced. In
  truly random noise (such as in simulated data), the value should be set high
  (1.0), but in the presence of systematic noise when LRR are not informative,
  lower values result in cleaner calls (0.2).</div>
<div class="Pp"></div>
<b>-L, --LRR-smooth-win</b> <i>int</i>
<div style="margin-left: 4.00ex;">reduce LRR noise by applying moving average
  given this window size</div>
<div class="Pp"></div>
<b>-O, --optimize</b> <i>float</i>
<div style="margin-left: 4.00ex;">iteratively estimate the fraction of aberrant
  cells, down to the given fraction. Lowering this value from the default 1.0 to
  say, 0.3, can help discover more events but also increases noise</div>
<div class="Pp"></div>
<b>-P, --same-prob</b> <i>float</i>
<div style="margin-left: 4.00ex;">the prior probability of the query and the
  control sample being the same. Setting to 0 calls both independently, setting
  to 1 forces the same copy number state in both.</div>
<div class="Pp"></div>
<b>-x, --xy-prob</b> <i>float</i>
<div style="margin-left: 4.00ex;">the HMM probability of transition to another
  copy number state. Increasing this values leads to smaller and more frequent
  calls.</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_call_[OPTIONS]_FILE"><a class="selflink" href="#bcftools_call_[OPTIONS]_FILE">bcftools
  call <i>[OPTIONS]</i> <i>FILE</i></a></h2>
This command replaces the former <b>bcftools view</b> caller. Some of the
  original functionality has been temporarily lost in the process of transition
  under htslib, but will be added back on popular demand. The original calling
  model can be invoked with the <b>-c</b> option.
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>File format options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--ploidy</b> <i>ASSEMBLY</i>[<i>?</i>]
<div style="margin-left: 4.00ex;">predefined ploidy, use <i>list</i> (or any
  other unused word) to print a list of all predefined assemblies. Append a
  question mark to print the actual definition. See also
  <b>--ploidy-file</b>.</div>
<div class="Pp"></div>
<b>--ploidy-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">ploidy definition given as a
  space/tab-delimited list of CHROM, FROM, TO, SEX, PLOIDY. The SEX codes are
  arbitrary and correspond to the ones used by <b>--samples-file</b>. The
  default ploidy can be given using the starred records (see below), unlisted
  regions have ploidy 2. The default ploidy definition is</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    X 1 60000 M 1
    X 2699521 154931043 M 1
    Y 1 59373566 M 1
    Y 1 59373566 F 0
    MT 1 16569 M 1
    MT 1 16569 F 1
    *  * *     M 2
    *  * *     F 2
</pre>
</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --samples</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-S, --samples-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-t, --targets</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Input/output options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-A, --keep-alts</b>
<div style="margin-left: 4.00ex;">output all alternate alleles present in the
  alignments even if they do not appear in any of the genotypes</div>
<div class="Pp"></div>
<b>-f, --format-fields</b> <i>list</i>
<div style="margin-left: 4.00ex;">comma-separated list of FORMAT fields to
  output for each sample. Currently GQ and GP fields are supported. For
  convenience, the fields can be given as lower case letters.</div>
<div class="Pp"></div>
<b>-g, --gvcf</b> <i>INT</i>
<div style="margin-left: 4.00ex;">output also gVCF blocks of homozygous REF
  calls. The parameter <i>INT</i> is the minimum per-sample depth required to
  include a site in the non-variant block.</div>
<div class="Pp"></div>
<b>-i, --insert-missed</b> <i>INT</i>
<div style="margin-left: 4.00ex;">output also sites missed by mpileup but
  present in <b>-T, --targets-file</b>.</div>
<div class="Pp"></div>
<b>-M, --keep-masked-ref</b>
<div style="margin-left: 4.00ex;">output sites where REF allele is N</div>
<div class="Pp"></div>
<b>-V, --skip-variants</b> <i>snps</i>|<i>indels</i>
<div style="margin-left: 4.00ex;">skip indel/SNP sites</div>
<div class="Pp"></div>
<b>-v, --variants-only</b>
<div style="margin-left: 4.00ex;">output variant sites only</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Consensus/variant calling options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-c, --consensus-caller</b>
<div style="margin-left: 4.00ex;">the original <b>samtools</b>/<b>bcftools</b>
  calling method (conflicts with <b>-m</b>)</div>
<div class="Pp"></div>
<b>-C, --constrain</b> <i>alleles</i>|<i>trio</i>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<i>alleles</i>
<div style="margin-left: 4.00ex;">call genotypes given alleles. See also <b>-T,
  --targets-file</b>.</div>
<div class="Pp"></div>
<i>trio</i>
<div style="margin-left: 4.00ex;">call genotypes given the father-mother-child
  constraint. See also <b>-s, --samples</b> and <b>-n, --novel-rate</b>.</div>
</div>
<div class="Pp"></div>
<b>-m, --multiallelic-caller</b>
<div style="margin-left: 4.00ex;">alternative modelfor multiallelic and
  rare-variant calling designed to overcome known limitations in <b>-c</b>
  calling model (conflicts with <b>-c</b>)</div>
<div class="Pp"></div>
<b>-n, --novel-rate</b> <i>float</i>[,...]
<div style="margin-left: 4.00ex;">likelihood of novel mutation for constrained
  <b>-C</b> <i>trio</i> calling. The trio genotype calling maximizes likelihood
  of a particular combination of genotypes for father, mother and the child
  P(F=i,M=j,C=k) = P(unconstrained) * Pn + P(constrained) * (1-Pn). By providing
  three values, the mutation rate Pn is set explicitly for SNPs, deletions and
  insertions, respectively. If two values are given, the first is interpreted as
  the mutation rate of SNPs and the second is used to calculate the mutation
  rate of indels according to their length as Pn= <i>float</i>*exp(-a-b*len),
  where a=22.8689, b=0.2994 for insertions and a=21.9313, b=0.2856 for deletions
  [pubmed:23975140]. If only one value is given, the same mutation rate Pn is
  used for SNPs and indels.</div>
<div class="Pp"></div>
<b>-p, --pval-threshold</b> <i>float</i>
<div style="margin-left: 4.00ex;">with <b>-c</b>, accept variant if P(ref|D)
  &lt; <i>float</i>.</div>
<div class="Pp"></div>
<b>-P, --prior</b> <i>float</i>
<div style="margin-left: 4.00ex;">expected substitution rate, or 0 to disable
  the prior.</div>
<div class="Pp"></div>
<b>-t, --targets</b>
  <i>file</i>|<i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-X, --chromosome-X</b>
<div style="margin-left: 4.00ex;">haploid output for male samples (requires PED
  file with <b>-s</b>)</div>
<div class="Pp"></div>
<b>-Y, --chromosome-Y</b>
<div style="margin-left: 4.00ex;">haploid output for males and skips females
  (requires PED file with <b>-s</b>)</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_concat_[OPTIONS]_FILE1_FILE2_[...]"><a class="selflink" href="#bcftools_concat_[OPTIONS]_FILE1_FILE2_[...]">bcftools
  concat <i>[OPTIONS]</i> <i>FILE1</i> <i>FILE2</i> [...]</a></h2>
Concatenate or combine VCF/BCF files. All source files must have the same sample
  columns appearing in the same order. Can be used, for example, to concatenate
  chromosome VCFs into one VCF, or combine a SNP VCF and an indel VCF into one.
  The input files must be sorted by chr and position. The files must be given in
  the correct order to produce sorted VCF on output unless the <b>-a,
  --allow-overlaps</b> option is specified. With the --naive option, the files
  are concatenated without being recompressed, which is very fast but dangerous
  if the BCF headers differ.
<div class="Pp"></div>
<b>-a, --allow-overlaps</b>
<div style="margin-left: 4.00ex;">First coordinate of the next file can precede
  last record of the current file.</div>
<div class="Pp"></div>
<b>-c, --compact-PS</b>
<div style="margin-left: 4.00ex;">Do not output PS tag at each site, only at the
  start of a new phase set block.</div>
<div class="Pp"></div>
<b>-d, --rm-dups</b>
  <i>snps</i>|<i>indels</i>|<i>both</i>|<i>all</i>|<i>none</i>
<div style="margin-left: 4.00ex;">Output duplicate records of specified type
  present in multiple files only once. Requires <b>-a,
  --allow-overlaps</b>.</div>
<div class="Pp"></div>
<b>-D, --remove-duplicates</b>
<div style="margin-left: 4.00ex;">Alias for <b>-d none</b></div>
<div class="Pp"></div>
<b>-f, --file-list</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">Read the list of files from a file.</div>
<div class="Pp"></div>
<b>-l, --ligate</b>
<div style="margin-left: 4.00ex;">Ligate phased VCFs by matching phase at
  overlapping haplotypes</div>
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-n, --naive</b>
<div style="margin-left: 4.00ex;">Concatenate BCF files without recompression.
  This is very fast but requires that all files have the same headers. This is
  because all tags and chromosome names in the BCF body rely on the implicit
  order of the contig and tag definitions in the header. Currently no sanity
  checks are in place and only works for compressed BCF files. Dangerous, use
  with caution.</div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-q, --min-PQ</b> <i>INT</i>
<div style="margin-left: 4.00ex;">Break phase set if phasing quality is lower
  than <i>INT</i></div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b>. Requires <b>-a,
  --allow-overlaps</b>.</div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b>. Requires <b>-a,
  --allow-overlaps</b>.</div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<h2 class="Ss" title="Ss" id="bcftools_consensus_[OPTIONS]_FILE"><a class="selflink" href="#bcftools_consensus_[OPTIONS]_FILE">bcftools
  consensus <i>[OPTIONS]</i> <i>FILE</i></a></h2>
Create consensus sequence by applying VCF variants to a reference fasta file.
<div class="Pp"></div>
<b>-f, --fasta-ref</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">reference sequence in fasta format</div>
<div class="Pp"></div>
<b>-H, --haplotype</b> <i>1</i>|<i>2</i>
<div style="margin-left: 4.00ex;">apply variants for the given haplotype. This
  option requires <b>-s</b>, unless exactly one sample is present in the
  VCF</div>
<div class="Pp"></div>
<b>-i, --iupac-codes</b>
<div style="margin-left: 4.00ex;">output variants in the form of IUPAC ambiguity
  codes</div>
<div class="Pp"></div>
<b>-m, --mask</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">BED file or TAB file with regions to be
  replaced with N. See discussion of <b>--regions-file</b> in <b>Common
  Options</b> for file format details.</div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">write output to a file</div>
<div class="Pp"></div>
<b>-s, --sample</b> <i>NAME</i>
<div style="margin-left: 4.00ex;">apply variants of the given sample</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>Examples:</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    # Apply variants present in sample &quot;NA001&quot;, output IUPAC codes for hets
    bcftools consensus -i -s NA001 -f in.fa in.vcf.gz &gt; out.fa
<div class="Pp"></div>
    # Create consensus for one region. The fasta header lines are then expected
    # in the form &quot;&gt;chr:from-to&quot;.
    samtools faidx ref.fa 8:11870-11890 | bcftools consensus in.vcf.gz -o out.fa
</pre>
</div>
<h2 class="Ss" title="Ss" id="bcftools_convert_[OPTIONS]_FILE"><a class="selflink" href="#bcftools_convert_[OPTIONS]_FILE">bcftools
  convert <i>[OPTIONS]</i> <i>FILE</i></a></h2>
<b>VCF input options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-e, --exclude</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">exclude sites for which <i>EXPRESSION</i> is
  true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-i, --include</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">include only sites for which <i>EXPRESSION</i>
  is true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --samples</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-S, --samples-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-t, --targets</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>VCF output options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>GEN/SAMPLE conversion:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-G, --gensample2vcf</b> <i>prefix</i> or <i>gen-file</i>,<i>sample-file</i>
<div style="margin-left: 4.00ex;">convert IMPUTE2 output to VCF. The second
  column must be of the form &quot;CHROM:POS_REF_ALT&quot; to detect possible
  strand swaps; IMPUTE2 leaves the first one empty (&quot;--&quot;) when sites
  from reference panel are filled in. See also <b>-g</b> below.</div>
<div class="Pp"></div>
<b>-g, --gensample</b> <i>prefix</i> or <i>gen-file</i>,<i>sample-file</i>
<div style="margin-left: 4.00ex;">convert from VCF to gen/sample format used by
  IMPUTE2 and SHAPEIT. The columns of .gen file format are ID1,ID2,POS,A,B
  followed by three genotype probabilities P(AA), P(AB), P(BB) for each sample.
  In order to prevent strand swaps, the program uses IDs of the form
  &quot;CHROM:POS_REF_ALT&quot;. For example:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
  .gen
  ----
  1:111485207_G_A 1:111485207_G_A 111485207 G A 0 1 0 0 1 0
  1:111494194_C_T 1:111494194_C_T 111494194 C T 0 1 0 0 0 1
<div class="Pp"></div>
  .sample
  -------
  ID_1 ID_2 missing
  0 0 0
  sample1 sample1 0
  sample2 sample2 0
</pre>
</div>
<div class="Pp"></div>
<b>--tag</b> <i>STRING</i>
<div style="margin-left: 4.00ex;">tag to take values for .gen file:
  GT,PL,GL,GP</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>gVCF conversion:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>--gvcf2vcf</b>
<div style="margin-left: 4.00ex;">convert gVCF to VCF, expanding REF blocks into
  sites. Only sites with FILTER set to &quot;PASS&quot; or &quot;.&quot; will be
  expanded.</div>
<div class="Pp"></div>
<b>-f, --fasta-ref</b> <i>file</i>
<div style="margin-left: 4.00ex;">reference sequence in fasta format. Must be
  indexed with samtools faidx</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>HAPS/SAMPLE conversion:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>--hapsample2vcf</b> <i>prefix</i> or <i>haps-file</i>,<i>sample-file</i>
<div style="margin-left: 4.00ex;">convert from haps/sample format to VCF. The
  columns of .haps file are similar to .gen file above, but there are only two
  haplotype columns per sample. Note that the first column of the haps file is
  expected to be in the form &quot;CHR:POS_REF_ALT(_END)?&quot;, with the _END
  being optional for defining the INFO/END tag when ALT is a symbolic allele,
  for example:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
  .haps
  ----
  1:111485207_G_A rsID1 111485207 G A 0 1 0 0
  1:111494194_C_T rsID2 111494194 C T 0 1 0 0
  1:111495231_A_&lt;DEL&gt;_111495784 rsID3 111495231 A &lt;DEL&gt; 0 0 1 0
</pre>
</div>
<div class="Pp"></div>
<b>--hapsample</b> <i>prefix</i> or <i>haps-file</i>,<i>sample-file</i>
<div style="margin-left: 4.00ex;">convert from VCF to haps/sample format used by
  IMPUTE2 and SHAPEIT. The columns of .haps file begin with ID,RSID,POS,REF,ALT.
  In order to prevent strand swaps, the program uses IDs of the form
  &quot;CHROM:POS_REF_ALT&quot;.</div>
<div class="Pp"></div>
<b>--haploid2diploid</b>
<div style="margin-left: 4.00ex;">with <b>-h</b> option converts haploid
  genotypes to homozygous diploid genotypes. For example, the program will print
  <i>0 0</i> instead of the default <i>0 -</i>. This is useful for programs
  which do not handle haploid genotypes correctly.</div>
<div class="Pp"></div>
<b>--vcf-ids</b>
<div style="margin-left: 4.00ex;">output VCF IDs instead of
  &quot;CHROM:POS_REF_ALT&quot; IDs</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>HAPS/LEGEND/SAMPLE conversion:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-H, --haplegendsample2vcf</b> <i>prefix</i> or
  <i>haps-file</i>,<i>legend-file</i>, <i>sample-file</i>
<div style="margin-left: 4.00ex;">convert from haps/legend/sample format used by
  IMPUTE2 to VCF, see also <b>-h, --hapslegendsample</b> below.</div>
<div class="Pp"></div>
<b>-h, --haplegendsample</b> <i>prefix</i> or
  <i>haps-file</i>,<i>legend-file</i>, <i>sample-file</i>
<div style="margin-left: 4.00ex;">convert from VCF to haps/legend/sample format
  used by IMPUTE2 and SHAPEIT. The columns of .legend file ID,POS,REF,ALT. In
  order to prevent strand swaps, the program uses IDs of the form
  &quot;CHROM:POS_REF_ALT&quot;. The .sample file is quite basic at the moment
  with columns for population, group and sex expected to be edited by the user.
  For example:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
  .haps
  -----
  0 1 0 0 1 0
  0 1 0 0 0 1
<div class="Pp"></div>
  .legend
  -------
  id position a0 a1
  1:111485207_G_A 111485207 G A
  1:111494194_C_T 111494194 C T
<div class="Pp"></div>
  .sample
  -------
  sample population group sex
  sample1 sample1 sample1 2
  sample2 sample2 sample2 2
</pre>
</div>
<div class="Pp"></div>
<b>--haploid2diploid</b>
<div style="margin-left: 4.00ex;">with <b>-h</b> option converts haploid
  genotypes to homozygous diploid genotypes. For example, the program will print
  <i>0 0</i> instead of the default <i>0 -</i>. This is useful for programs
  which do not handle haploid genotypes correctly.</div>
<div class="Pp"></div>
<b>--vcf-ids</b>
<div style="margin-left: 4.00ex;">output VCF IDs instead of
  &quot;CHROM:POS_REF_ALT&quot; IDs</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>TSV conversion:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>--tsv2vcf</b> <i>file</i>
<div style="margin-left: 4.00ex;">convert from TSV (tab-separated values) format
  (such as generated by 23andMe) to VCF. The input file fields can be tab- or
  space- delimited</div>
<div class="Pp"></div>
<b>-c, --columns</b> <i>list</i>
<div style="margin-left: 4.00ex;">comma-separated list of fields in the input
  file. In the current version, the fields CHROM, POS, ID, and AA are expected
  and can appear in arbitrary order, columns which should be ignored in the
  input file can be indicated by &quot;-&quot;. The AA field lists alleles on
  the forward reference strand, for example &quot;CC&quot; or &quot;CT&quot; for
  diploid genotypes or &quot;C&quot; for haploid genotypes (sex chromosomes).
  Insertions and deletions are not supported yet, missing data can be indicated
  with &quot;--&quot;.</div>
<div class="Pp"></div>
<b>-f, --fasta-ref</b> <i>file</i>
<div style="margin-left: 4.00ex;">reference sequence in fasta format. Must be
  indexed with samtools faidx</div>
<div class="Pp"></div>
<b>-s, --samples</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">list of sample names. See <b>Common
  Options</b></div>
<div class="Pp"></div>
<b>-S, --samples-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">file of sample names. See <b>Common
  Options</b></div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>Example:</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
# Convert 23andme results into VCF
bcftools convert -c ID,CHROM,POS,AA -s SampleName -f 23andme-ref.fa --tsv2vcf 23andme.txt -Oz -o out.vcf.gz
</pre>
</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_filter_[OPTIONS]_FILE"><a class="selflink" href="#bcftools_filter_[OPTIONS]_FILE">bcftools
  filter <i>[OPTIONS]</i> <i>FILE</i></a></h2>
Apply fixed-threshold filters.
<div class="Pp"></div>
<b>-e, --exclude</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">exclude sites for which <i>EXPRESSION</i> is
  true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-g, --SnpGap</b> <i>INT</i>
<div style="margin-left: 4.00ex;">filter SNPs within <i>INT</i> base pairs of an
  indel. The following example demonstrates the logic of <b>--SnpGap</b>
  <i>3</i> applied on a deletion and an insertion:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
The SNPs at positions 1 and 7 are filtered, positions 0 and 8 are not:
         0123456789
    ref  .G.GT..G..
    del  .A.G-..A..
Here the positions 1 and 6 are filtered, 0 and 7 are not:
         0123-456789
    ref  .G.G-..G..
    ins  .A.GT..A..
</pre>
</div>
<div class="Pp"></div>
<b>-G, --IndelGap</b> <i>INT</i>
<div style="margin-left: 4.00ex;">filter clusters of indels separated by
  <i>INT</i> or fewer base pairs allowing only one to pass. The following
  example demonstrates the logic of <b>--IndelGap</b> <i>2</i> applied on a
  deletion and an insertion:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
The second indel is filtered:
         012345678901
    ref  .GT.GT..GT..
    del  .G-.G-..G-..
And similarly here, the second is filtered:
         01 23 456 78
    ref  .A-.A-..A-..
    ins  .AT.AT..AT..
</pre>
</div>
<div class="Pp"></div>
<b>-i, --include</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">include only sites for which <i>EXPRESSION</i>
  is true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-m, --mode</b> [<i>+x</i>]
<div style="margin-left: 4.00ex;">define behaviour at sites with existing FILTER
  annotations. The default mode replaces existing filters of failed sites with a
  new FILTER string while leaving sites which pass untouched when non-empty and
  setting to &quot;PASS&quot; when the FILTER string is absent. The
  &quot;+&quot; mode appends new FILTER strings of failed sites instead of
  replacing them. The &quot;x&quot; mode resets filters of sites which pass to
  &quot;PASS&quot;. Modes &quot;+&quot; and &quot;x&quot; can both be set.</div>
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --soft-filter</b> <i>STRING</i>|<i>+</i>
<div style="margin-left: 4.00ex;">annotate FILTER column with <i>STRING</i> or,
  with <i>+</i>, a unique filter name generated by the program
  (&quot;Filter%d&quot;).</div>
<div class="Pp"></div>
<b>-S, --set-GTs</b> <i>.</i>|<i>0</i>
<div style="margin-left: 4.00ex;">set genotypes of failed samples to missing
  value ( <i>.</i>) or reference allele (<i>0</i>)</div>
<div class="Pp"></div>
<b>-t, --targets</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<h2 class="Ss" title="Ss" id="bcftools_gtcheck_[OPTIONS]_[-g_genotypes.vcf.gz]_query.vcf.gz"><a class="selflink" href="#bcftools_gtcheck_[OPTIONS]_[-g_genotypes.vcf.gz]_query.vcf.gz">bcftools
  gtcheck [ <i>OPTIONS</i>] [-g <i>genotypes.vcf.gz</i>]
  <i>query.vcf.gz</i></a></h2>
Checks sample identity or, without <b>-g</b>, multi-sample cross-check is
  performed.
<div class="Pp"></div>
<b>-a, --all-sites</b>
<div style="margin-left: 4.00ex;">output for all sites</div>
<div class="Pp"></div>
<b>-g, --genotypes</b> <i>genotypes.vcf.gz</i>
<div style="margin-left: 4.00ex;">reference genotypes to compare against</div>
<div class="Pp"></div>
<b>-G, --GTs-only</b> <i>INT</i>
<div style="margin-left: 4.00ex;">use genotypes (GT) instead of genotype
  likelihoods (PL). When set to 1, reported discordance is the number of
  non-matching GTs, otherwise the number <i>INT</i> is interpreted as
  phred-scaled likelihood of unobserved genotypes.</div>
<div class="Pp"></div>
<b>-H, --homs-only</b>
<div style="margin-left: 4.00ex;">consider only genotypes which are homozygous
  in both <i>genotypes</i> and <i>query</i> VCF. This may be useful with low
  coverage data.</div>
<div class="Pp"></div>
<b>-p, --plot</b> <i>PREFIX</i>
<div style="margin-left: 4.00ex;">produce plots</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --query-sample</b> <i>STRING</i>
<div style="margin-left: 4.00ex;">query sample in <i>query.vcf.gz</i>. By
  default, the first sample is checked.</div>
<div class="Pp"></div>
<b>-S, --target-sample</b> <i>STRING</i>
<div style="margin-left: 4.00ex;">target sample in the <b>-g</b> file, used only
  for plotting, not for analysis</div>
<div class="Pp"></div>
<b>-t, --targets</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Output files format:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
CN, Discordance
<div style="margin-left: 4.00ex;">Pairwise discordance for all sample pairs is
  calculated as</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
        \sum_s { min_G { PL_a(G) + PL_b(G) } },
</pre>
</div>
<div class="Pp"></div>
<div style="margin-left: 4.00ex;">where the sum runs over all sites <i>s</i> and
  <i>G</i> is the the most likely genotype shared by both samples <i>a</i> and
  <i>b</i>. When PL field is not present, a constant value <i>99</i> is used for
  the unseen genotypes. With <b>-G</b>, the value <i>1</i> can be used instead;
  the discordance value then gives exactly the number of differing
  genotypes.</div>
<div class="Pp"></div>
SM, Average Discordance
<div style="margin-left: 4.00ex;">Average discordance between sample <i>a</i>
  and all other samples.</div>
<div class="Pp"></div>
SM, Average Depth
<div style="margin-left: 4.00ex;">Average depth at evaluated sites, or 1 if
  FORMAT/DP field is not present.</div>
<div class="Pp"></div>
SM, Average Number of sites
<div style="margin-left: 4.00ex;">The average number of sites used to calculate
  the discordance. In other words, the average number of non-missing
  PLs/genotypes seen both samples.</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_index_[OPTIONS]_&lt;in.bcf&gt;|&lt;in.vcf.gz&gt;"><a class="selflink" href="#bcftools_index_[OPTIONS]_&lt;in.bcf&gt;|&lt;in.vcf.gz&gt;">bcftools
  index [ <i>OPTIONS</i>] <i>&lt;in.bcf&gt;|&lt;in.vcf.gz&gt;</i></a></h2>
Creates index for bgzip compressed VCF/BCF files for random access. CSI
  (coordinate-sorted index) is created by default. The CSI format supports
  indexing of chromosomes up to length 2^31. TBI (tabix index) index files,
  which support chromosome lengths up to 2^29, can be created by using the
  <i>-t/--tbi</i> option or using the <i>tabix</i> program packaged with htslib.
  When loading an index file, bcftools will try the CSI first and then the TBI.
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Indexing options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-c, --csi</b>
<div style="margin-left: 4.00ex;">generate CSI-format index for VCF/BCF files
  [default]</div>
<div class="Pp"></div>
<b>-f, --force</b>
<div style="margin-left: 4.00ex;">overwrite index if it already exists</div>
<div class="Pp"></div>
<b>-m, --min-shift </b><b></b><i>INT</i>
<div style="margin-left: 4.00ex;">set minimal interval size for CSI indices to
  2^INT; default: 14</div>
<div class="Pp"></div>
<b>-t, --tbi</b>
<div style="margin-left: 4.00ex;">generate TBI-format index for VCF files</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Stats options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-n, --nrecords</b>
<div style="margin-left: 4.00ex;">print the number of records based on the CSI
  or TBI index files</div>
<div class="Pp"></div>
<b>-s, --stats</b>
<div style="margin-left: 4.00ex;">Print per contig stats based on the CSI or TBI
  index files. Output format is three tab-delimited columns listing the contig
  name, contig length ( <i>.</i> if unknown) and number of records for the
  contig. Contigs with zero records are not printed.</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_isec_[OPTIONS]_A.vcf.gz_B.vcf.gz_[...]"><a class="selflink" href="#bcftools_isec_[OPTIONS]_A.vcf.gz_B.vcf.gz_[...]">bcftools
  isec [ <i>OPTIONS</i>] <i>A.vcf.gz</i> <i>B.vcf.gz</i> [...]</a></h2>
Creates intersections, unions and complements of VCF files. Depending on the
  options, the program can output records from one (or more) files which have
  (or do not have) corresponding records with the same position in the other
  files.
<div class="Pp"></div>
<b>-c, --collapse</b>
  <i>snps</i>|<i>indels</i>|<i>both</i>|<i>all</i>|<i>some</i>| <i>none</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-C, --complement</b>
<div style="margin-left: 4.00ex;">output positions present only in the first
  file but missing in the others</div>
<div class="Pp"></div>
<b>-e, --exclude</b> <i>-</i>|<i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">exclude sites for which <i>EXPRESSION</i> is
  true. If <b>-e</b> (or <b>-i</b>) appears only once, the same filtering
  expression will be applied to all input files. Otherwise, <b>-e</b> or
  <b>-i</b> must be given for each input file. To indicate that no filtering
  should be performed on a file, use &quot;-&quot; in place of
  <i>EXPRESSION</i>, as shown in the example below. For valid expressions see
  <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-f, --apply-filters</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-i, --include</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">include only sites for which <i>EXPRESSION</i>
  is true. See discussion of <b>-e, --exclude</b> above.</div>
<div class="Pp"></div>
<b>-n, --nfiles</b> [+-=]<i>INT</i>|~<i>BITMAP</i>
<div style="margin-left: 4.00ex;">output positions present in this many (=),
  this many or more (+), this many or fewer (-), or the exact same (~)
  files</div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b>. When several files
  are being output, their names are controlled via <b>-p</b> instead.</div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-p, --prefix</b> <i>DIR</i>
<div style="margin-left: 4.00ex;">if given, subset each of the input files
  accordingly. See also <b>-w</b>.</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-t, --targets</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-w, --write</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">list of input files to output given as 1-based
  indices. With <b>-p</b> and no <b>-w</b>, all files are written.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Examples:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
Create intersection and complements of two sets saving the output in dir/*
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools isec -p dir A.vcf.gz B.vcf.gz
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
Filter sites in A and B (but not in C) and create intersection
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools isec -e'MAF&lt;0.01' -i'dbSNP=1' -e- A.vcf.gz B.vcf.gz C.vcf.gz -p dir
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
Extract and write records from A shared by both A and B using exact allele match
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools isec -p dir -n=2 -w1 A.vcf.gz B.vcf.gz
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
Extract records private to A or B comparing by position only
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools isec -p dir -n-1 -c all A.vcf.gz B.vcf.gz
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
Print a list of records which are present in A and B but not in C and D
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools isec -n~1100 -c all A.vcf.gz B.vcf.gz C.vcf.gz D.vcf.gz
</pre>
</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_merge_[OPTIONS]_A.vcf.gz_B.vcf.gz_[...]"><a class="selflink" href="#bcftools_merge_[OPTIONS]_A.vcf.gz_B.vcf.gz_[...]">bcftools
  merge [ <i>OPTIONS</i>] <i>A.vcf.gz</i> <i>B.vcf.gz</i> [...]</a></h2>
Merge multiple VCF/BCF files from non-overlapping sample sets to create one
  multi-sample file. For example, when merging file <i>A.vcf.gz</i> containing
  samples <i>S1</i>, <i>S2</i> and <i>S3</i> and file <i>B.vcf.gz</i> containing
  samples <i>S3</i> and <i>S4</i>, the output file will contain four samples
  named <i>S1</i>, <i>S2</i>, <i>S3</i>, <i>2:S3</i> and <i>S4</i>.
<div style="height: 1.00em;">&#x00A0;</div>
Note that it is responsibility of the user to ensure that the sample names are
  unique across all files. If they are not, the program will exit with an error
  unless the option <b>--force-samples</b> is given. The sample names can be
  also given explicitly using the <b>--print-header</b> and <b>--use-header</b>
  options.
<div style="height: 1.00em;">&#x00A0;</div>
Note that only records from different files can be merged, never from the same
  file. For &quot;vertical&quot; merge take a look at <b>bcftools norm</b>
  instead.
<div class="Pp"></div>
<b>--force-samples</b>
<div style="margin-left: 4.00ex;">if the merged files contain duplicate samples
  names, proceed anyway. Duplicate sample names will be resolved by prepending
  index of the file as it appeared on the command line to the conflicting sample
  name (see <i>2:S3</i> in the above example).</div>
<div class="Pp"></div>
<b>--print-header</b>
<div style="margin-left: 4.00ex;">print only merged header and exit</div>
<div class="Pp"></div>
<b>--use-header</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">use the VCF header in the provided text
  <i>FILE</i></div>
<div class="Pp"></div>
<b>-f, --apply-filters</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-i, --info-rules</b> <i>-</i>|<i>TAG:METHOD</i>[,...]
<div style="margin-left: 4.00ex;">Rules for merging INFO fields (scalars or
  vectors) or <i>-</i> to disable the default rules. <i>METHOD</i> is one of
  <i>sum</i>, <i>avg</i>, <i>min</i>, <i>max</i>, <i>join</i>. Default is
  <i>DP:sum,DP4:sum</i> if these fields exist in the input files. Fields with no
  specified rule will take the value from the first input file. The merged QUAL
  value is currently set to the maximum. This behaviour is not user controllable
  at the moment.</div>
<div class="Pp"></div>
<b>-l, --file-list</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">read file names from <i>FILE</i></div>
<div class="Pp"></div>
<b>-m, --merge</b>
  <i>snps</i>|<i>indels</i>|<i>both</i>|<i>all</i>|<i>none</i>|<i>id</i>
<div style="margin-left: 4.00ex;">The option controls what types of multiallelic
  records can be created:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
-m none   ..  no new multiallelics, output multiple records instead
-m snps   ..  allow multiallelic SNP records
-m indels ..  allow multiallelic indel records
-m both   ..  both SNP and indel records can be multiallelic
-m all    ..  SNP records can be merged with indel records
-m id     ..  merge by ID
</pre>
</div>
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<h2 class="Ss" title="Ss" id="bcftools_norm_[OPTIONS]_file.vcf.gz"><a class="selflink" href="#bcftools_norm_[OPTIONS]_file.vcf.gz">bcftools
  norm [ <i>OPTIONS</i>] <i>file.vcf.gz</i></a></h2>
Left-align and normalize indels, check if REF alleles match the reference, split
  multiallelic sites into multiple rows; recover multiallelics from multiple
  rows. Left-alignment and normalization will only be applied if the
  <b>--fasta-ref</b> option is supplied.
<div class="Pp"></div>
<b>-c, --check-ref</b> <i>e</i>|<i>w</i>|<i>x</i>|<i>s</i>
<div style="margin-left: 4.00ex;">what to do when incorrect or missing REF
  allele is encountered: exit ( <i>e</i>), warn (<i>w</i>), exclude (<i>x</i>),
  or set/fix ( <i>s</i>) bad sites. The <i>w</i> option can be combined with
  <i>x</i> and <i>s</i>. Note that <i>s</i> can swap alleles and will update
  genotypes (GT) and AC counts, but will not attempt to fix PL or other
  fields.</div>
<div class="Pp"></div>
<b>-d, --rm-dup</b> <i>snps</i>|<i>indels</i>|<i>both</i>|<i>all</i>|<i>none</i>
<div style="margin-left: 4.00ex;">If a record is present in multiple files,
  output only the first instance, see <b>--collapse</b> in <b>Common
  Options</b>. Requires <b>-a, --allow-overlaps</b>.</div>
<div class="Pp"></div>
<b>-D, --remove-duplicates</b>
<div style="margin-left: 4.00ex;">If a record is present in multiple files,
  output only the first instance. Alias for <b>-d none</b>. Requires <b>-a,
  --allow-overlaps</b>.</div>
<div class="Pp"></div>
<b>-f, --fasta-ref</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">reference sequence. Supplying this option will
  turn on left-alignment and normalization, however, see also the
  <b>--do-not-normalize</b> option below.</div>
<div class="Pp"></div>
<b>-m, --multiallelics</b>
  &#x2190;|+&gt;[<i>snps</i>|<i>indels</i>|<i>both</i>|<i>any</i>]
<div style="margin-left: 4.00ex;">split multiallelic sites into biallelic
  records ( <i>-</i>) or join biallelic sites into multiallelic records
  (<i>+</i>). An optional type string can follow which controls variant types
  which should be split or merged together: If only SNP records should be split
  or merged, specify <i>snps</i>; if both SNPs and indels should be merged
  separately into two records, specify <i>both</i>; if SNPs and indels should be
  merged into a single record, specify <i>any</i>.</div>
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-N, --do-not-normalize</b>
<div style="margin-left: 4.00ex;">the <i>-c s</i> option can be used to fix or
  set the REF allele from the reference <i>-f</i>. The <i>-N</i> option will not
  turn on indel normalisation as the <i>-f</i> option normally implies</div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --strict-filter</b>
<div style="margin-left: 4.00ex;">when merging (<i>-m+</i>), merged site is PASS
  only if all sites being merged PASS</div>
<div class="Pp"></div>
<b>-t, --targets</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-w, --site-win</b> <i>INT</i>
<div style="margin-left: 4.00ex;">maximum distance between two records to
  consider when locally sorting variants which changed position during the
  realignment</div>
<h2 class="Ss" title="Ss" id="bcftools_[plugin_NAME|+NAME]_[OPTIONS]_FILE_&#x2014;_[PLUGIN_OPTIONS]"><a class="selflink" href="#bcftools_[plugin_NAME|+NAME]_[OPTIONS]_FILE_&#x2014;_[PLUGIN_OPTIONS]">bcftools
  [plugin <i>NAME</i>|+<i>NAME</i>] <i>[OPTIONS]</i> <i>FILE</i> &#x2014;
  <i>[PLUGIN OPTIONS]</i></a></h2>
<b>VCF input options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-e, --exclude</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">exclude sites for which <i>EXPRESSION</i> is
  true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-i, --include</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">include only sites for which <i>EXPRESSION</i>
  is true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-t, --targets</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>VCF output options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Plugin options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-h, --help</b>
<div style="margin-left: 4.00ex;">list plugin&#x2019;s options</div>
<div class="Pp"></div>
<b>-l, --list-plugins</b>
<div style="margin-left: 4.00ex;">List all available plugins.
<div style="height: 1.00em;">&#x00A0;</div>
By default, appropriate system directories are searched for installed plugins.
  You can override this by setting the BCFTOOLS_PLUGINS environment variable to
  a colon-separated list of directories to search. If BCFTOOLS_PLUGINS begins
  with a colon, ends with a colon, or contains adjacent colons, the system
  directories are also searched at that position in the list of directories.
<div style="height: 1.00em;">&#x00A0;</div>
If htslib is not installed systemwide, set the environment variable
  LD_LIBRARY_PATH (linux) or DYLD_LIBRARY_PATH (Mac OS X) to include the
  directory where <b>libhts.so.1</b> is located.</div>
<div class="Pp"></div>
<b>-v, --verbose</b>
<div style="margin-left: 4.00ex;">print debugging information to debug plugin
  failure</div>
<div class="Pp"></div>
<b>-V, --version</b>
<div style="margin-left: 4.00ex;">print version string and exit</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>List of plugins coming with the distribution:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>counts</b>
<div style="margin-left: 4.00ex;">a minimal plugin which counts number of SNPs,
  Indels, and total number of sites.</div>
<div class="Pp"></div>
<b>dosage</b>
<div style="margin-left: 4.00ex;">print genotype dosage. By default the plugin
  searches for PL, GL and GT, in that order.</div>
<div class="Pp"></div>
<b>fill-AN-AC</b>
<div style="margin-left: 4.00ex;">fill INFO fields AN and AC.</div>
<div class="Pp"></div>
<b>fix-ploidy</b>
<div style="margin-left: 4.00ex;">sets correct ploidy</div>
<div class="Pp"></div>
<b>frameshifts</b>
<div style="margin-left: 4.00ex;">annotate frameshift indels</div>
<div class="Pp"></div>
<b>missing2ref</b>
<div style="margin-left: 4.00ex;">sets missing genotypes (&quot;./.&quot;) to
  ref allele (&quot;0/0&quot; or &quot;0|0&quot;)</div>
<div class="Pp"></div>
<b>tag2tag</b>
<div style="margin-left: 4.00ex;">Convert between similar tags, such as GL and
  GP.</div>
<div class="Pp"></div>
<b>vcf2sex</b>
<div style="margin-left: 4.00ex;">determine sample sex by checking genotypes in
  haploid regions</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Examples:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
# List options common to all plugins
bcftools plugin
<div class="Pp"></div>
# List available plugins
bcftools plugin -l
<div class="Pp"></div>
# Run a plugin
bcftools plugin counts in.vcf
<div class="Pp"></div>
# Run a plugin using the abbreviated &quot;+&quot; notation
bcftools +counts in.vcf
<div class="Pp"></div>
# The input VCF can be streamed just like in other commands
cat in.vcf | bcftools +counts
<div class="Pp"></div>
# Print usage information of plugin &quot;dosage&quot;
bcftools +dosage -h
<div class="Pp"></div>
# Replace missing genotypes with 0/0
bcftools +missing2ref in.vcf
<div class="Pp"></div>
# Replace missing genotypes with 0|0
bcftools +missing2ref in.vcf -- -p
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Plugins troubleshooting:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
Things to check if your plugin does not show up in the <b>bcftools plugin -l</b>
  output:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Run with the <b>-v</b> option for
  verbose output: <b>bcftools plugin -lv</b></div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Does the environment variable
  BCFTOOLS_PLUGINS include the correct path?</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Are all shared libraries, namely
  libhts.so, accessible? Verify with
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;on Mac OS X: <b>otool -L
  your/plugin.so</b> and set DYLD_LIBRARY_PATH if they are not</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;on Linux: <b>ldd your/plugin.so</b>
  and set LD_LIBRARY_PATH if they are not</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;If not installed systemwide, set the
  environment variable LD_LIBRARY_PATH (linux) or DYLD_LIBRARY_PATH (mac) to
  include directory where <b>libhts.so</b> is located.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Plugins API:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
// Short description used by 'bcftools plugin -l'
const char *about(void);
<div class="Pp"></div>
// Longer description used by 'bcftools +name -h'
const char *usage(void);
<div class="Pp"></div>
// Called once at startup, allows initialization of local variables.
// Return 1 to suppress normal VCF/BCF header output, -1 on critical
// errors, 0 otherwise.
int init(int argc, char **argv, bcf_hdr_t *in_hdr, bcf_hdr_t *out_hdr);
<div class="Pp"></div>
// Called for each VCF record, return NULL to suppress the output
bcf1_t *process(bcf1_t *rec);
<div class="Pp"></div>
// Called after all lines have been processed to clean up
void destroy(void);
</pre>
</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_polysomy_[OPTIONS]_file.vcf.gz"><a class="selflink" href="#bcftools_polysomy_[OPTIONS]_file.vcf.gz">bcftools
  polysomy [ <i>OPTIONS</i>] <i>file.vcf.gz</i></a></h2>
Detect number of chromosomal copies in VCFs annotates with the Illumina&#x2019;s
  B-allele frequency (BAF) values. Note that this command is not compiled in by
  default, see the section <b>Optional Compilation with GSL</b> in the INSTALL
  file for help.
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>General options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-o, --output-dir</b> <i>path</i>
<div style="margin-left: 4.00ex;">output directory</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --sample</b> <i>string</i>
<div style="margin-left: 4.00ex;">sample name</div>
<div class="Pp"></div>
<b>-t, --targets</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-v, --verbose</b>
<div style="margin-left: 4.00ex;">verbose debugging output which gives hints
  about the thresholds and decisions made by the program. Note that the exact
  output can change between versions.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Algorithm options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-b, --peak-size</b> <i>float</i>
<div style="margin-left: 4.00ex;">the minimum peak size considered as a good
  match can be from the interval [0,1] where larger is stricter</div>
<div class="Pp"></div>
<b>-c, --cn-penalty</b> <i>float</i>
<div style="margin-left: 4.00ex;">a penalty for increasing copy number state.
  How this works: multiple peaks are always a better fit than a single peak,
  therefore the program prefers a single peak (normal copy number) unless the
  absolute deviation of the multiple peaks fit is significantly smaller. Here
  the meaning of &quot;significant&quot; is given by the <i>float</i> from the
  interval [0,1] where larger is stricter.</div>
<div class="Pp"></div>
<b>-f, --fit-th</b> <i>float</i>
<div style="margin-left: 4.00ex;">threshold for goodness of fit (normalized
  absolute deviation), smaller is stricter</div>
<div class="Pp"></div>
<b>-i, --include-aa</b>
<div style="margin-left: 4.00ex;">include also the AA peak in CN2 and CN3
  evaluation. This usually requires increasing <b>-f</b>.</div>
<div class="Pp"></div>
<b>-m, --min-fraction</b> <i>float</i>
<div style="margin-left: 4.00ex;">minimum distinguishable fraction of aberrant
  cells. The experience shows that trustworthy are estimates of 20% and
  more.</div>
<div class="Pp"></div>
<b>-p, --peak-symmetry</b> <i>float</i>
<div style="margin-left: 4.00ex;">a heuristics to filter failed fits where the
  expected peak symmetry is violated. The <i>float</i> is from the interval
  [0,1] and larger is stricter</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_query_[OPTIONS]_file.vcf.gz_[file.vcf.gz_[...]]"><a class="selflink" href="#bcftools_query_[OPTIONS]_file.vcf.gz_[file.vcf.gz_[...]]">bcftools
  query [ <i>OPTIONS</i>] <i>file.vcf.gz</i> [<i>file.vcf.gz</i> [...]]</a></h2>
Extracts fields from VCF or BCF files and outputs them in user-defined format.
<div class="Pp"></div>
<b>-c, --collapse</b>
  <i>snps</i>|<i>indels</i>|<i>both</i>|<i>all</i>|<i>some</i>| <i>none</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-e, --exclude</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">exclude sites for which <i>EXPRESSION</i> is
  true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-f, --format</b> <i>FORMAT</i>
<div style="margin-left: 4.00ex;">learn by example, see below</div>
<div class="Pp"></div>
<b>-H, --print-header</b>
<div style="margin-left: 4.00ex;">print header</div>
<div class="Pp"></div>
<b>-i, --include</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">include only sites for which <i>EXPRESSION</i>
  is true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-l, --list-samples</b>
<div style="margin-left: 4.00ex;">list sample names and exit</div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --samples</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-S, --samples-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-t, --targets</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-u, --allow-undef-tags</b>
<div style="margin-left: 4.00ex;">do not throw an error if there are undefined
  tags in the format string, print &quot;.&quot; instead</div>
<div class="Pp"></div>
<b>-v, --vcf-list</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">process multiple VCFs listed in the file</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Format:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
%CHROM          The CHROM column (similarly also other columns: POS, ID, REF, ALT, QUAL, FILTER)
%INFO/TAG       Any tag in the INFO column
%TYPE           Variant type (REF, SNP, MNP, INDEL, OTHER)
%MASK           Indicates presence of the site in other files (with multiple files)
%TAG{INT}       Curly brackets to subscript vectors (0-based)
%FIRST_ALT      Alias for %ALT{0}
[]              The brackets loop over all samples
%GT             Genotype (e.g. 0/1)
%TGT            Translated genotype (e.g. C/A)
%IUPACGT        Genotype translated to IUPAC ambiguity codes (e.g. M instead of C/A)
%LINE           Prints the whole line
%SAMPLE         Sample name
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Examples:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
bcftools query -f '%CHROM  %POS  %REF  %ALT{0}\n' file.vcf.gz
bcftools query -f '%CHROM\t%POS\t%REF\t%ALT[\t%SAMPLE=%GT]\n' file.vcf.gz
</pre>
</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_reheader_[OPTIONS]_file.vcf.gz"><a class="selflink" href="#bcftools_reheader_[OPTIONS]_file.vcf.gz">bcftools
  reheader [ <i>OPTIONS</i>] <i>file.vcf.gz</i></a></h2>
Modify header of VCF/BCF files, change sample names.
<div class="Pp"></div>
<b>-h, --header</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">new VCF header</div>
<div class="Pp"></div>
<b>-o, --output</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --samples</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">new sample names, one name per line, in the
  same order as they appear in the VCF file. Alternatively, only samples which
  need to be renamed can be listed as &quot;old_name new_name\n&quot; pairs
  separated by whitespaces, each on a separate line. If a sample name contains
  spaces, the spaces can be escaped using the backslash character, for example
  &quot;Not\ a\ good\ sample\ name&quot;.</div>
<h2 class="Ss" title="Ss" id="bcftools_roh_[OPTIONS]_file.vcf.gz"><a class="selflink" href="#bcftools_roh_[OPTIONS]_file.vcf.gz">bcftools
  roh [ <i>OPTIONS</i>] <i>file.vcf.gz</i></a></h2>
A program for detecting runs of homo/autozygosity. Only bi-allelic sites are
  considered.
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>The HMM model:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
Notation:
  D  = Data, AZ = autozygosity, HW = Hardy-Weinberg (non-autozygosity),
  f  = non-ref allele frequency
<div class="Pp"></div>
Emission probabilities:
  oAZ = P_i(D|AZ) = (1-f)*P(D|RR) + f*P(D|AA)
  oHW = P_i(D|HW) = (1-f)^2 * P(D|RR) + f^2 * P(D|AA) + 2*f*(1-f)*P(D|RA)
<div class="Pp"></div>
Transition probabilities:
  tAZ = P(AZ|HW)  .. from HW to AZ, the -a parameter
  tHW = P(HW|AZ)  .. from AZ to HW, the -H parameter
<div class="Pp"></div>
  ci  = P_i(C)  .. probability of cross-over at site i, from genetic map
  AZi = P_i(AZ) .. probability of site i being AZ/non-AZ, scaled so that AZi+HWi = 1
  HWi = P_i(HW)
<div class="Pp"></div>
  P_{i+1}(AZ) = oAZ * max[(1 - tAZ * ci) * AZ{i-1} , tAZ * ci * (1-AZ{i-1})]
  P_{i+1}(HW) = oHW * max[(1 - tHW * ci) * (1-AZ{i-1}) , tHW * ci * AZ{i-1}]
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>General Options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>--AF-dflt</b> <i>FLOAT</i>
<div style="margin-left: 4.00ex;">in case allele frequency is not known, use the
  <i>FLOAT</i>. By default, sites where allele frequency cannot be determined,
  or is 0, are skipped.</div>
<div class="Pp"></div>
<b>--AF-tag</b> <i>TAG</i>
<div style="margin-left: 4.00ex;">use the specified INFO tag <i>TAG</i> as an
  allele frequency estimate instead of the default AC and AN tags. Sites which
  do not have <i>TAG</i> will be skipped.</div>
<div class="Pp"></div>
<b>--AF-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">Read allele frequencies from a tab-delimited
  file containing the columns: CHROM\tPOS\tREF,ALT\tAF. The file can be
  compressed with <b>bgzip</b> and indexed with tabix -s1 -b2 -e2. Sites which
  are not present in the <i>FILE</i> or have different reference or alternate
  allele will be skipped. Note that such a file can be easily created from a VCF
  using:</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    bcftools query -f'%CHROM\t%POS\t%REF,%ALT\t%INFO/TAG\n' file.vcf | bgzip -c &gt; freqs.tab.gz
</pre>
</div>
<div class="Pp"></div>
<b>-e, --estimate-AF</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">recalculate INFO/AC and INFO/AN on the fly,
  using either all samples (&quot;-&quot;) or samples listed in <i>FILE</i>. By
  default, allele frequency is estimated from AC and AN counts which are already
  present in the INFO field.</div>
<div class="Pp"></div>
<b>-G, --GTs-only</b> <i>FLOAT</i>
<div style="margin-left: 4.00ex;">use genotypes (FORMAT/GT fields) ignoring
  genotype likelihoods (FORMAT/PL), setting PL of unseen genotypes to
  <i>FLOAT</i>. Safe value to use is 30 to account for GT errors.</div>
<div class="Pp"></div>
<b>-I, --skip-indels</b>
<div style="margin-left: 4.00ex;">skip indels as their genotypes are usually
  enriched for errors</div>
<div class="Pp"></div>
<b>-m, --genetic-map</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">genetic map in the format required also by
  IMPUTE2. Only the first and third column are used (position and
  Genetic_Map(cM)). The <i>FILE</i> can chromosome name.</div>
<div class="Pp"></div>
<b>-M, --rec-rate</b> <i>FLOAT</i>
<div style="margin-left: 4.00ex;">constant recombination rate per bp</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --sample</b> <i>name</i>
<div style="margin-left: 4.00ex;">the name of sample to analyze</div>
<div class="Pp"></div>
<b>-t, --targets</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>HMM Options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-a, --hw-to-az</b> <i>FLOAT</i>
<div style="margin-left: 4.00ex;">P(AZ|HW) transition probability from AZ
  (autozygous) to HW (Hardy-Weinberg) state</div>
<div class="Pp"></div>
<b>-H, --az-to-hw</b> <i>FLOAT</i>
<div style="margin-left: 4.00ex;">P(HW|AZ) transition probability from HW to AZ
  state</div>
<div class="Pp"></div>
<b>-V, --viterbi-training</b>
<div style="margin-left: 4.00ex;">perform Viterbi training to estimate
  transition probabilities</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_stats_[OPTIONS]_A.vcf.gz_[B.vcf.gz]"><a class="selflink" href="#bcftools_stats_[OPTIONS]_A.vcf.gz_[B.vcf.gz]">bcftools
  stats [ <i>OPTIONS</i>] <i>A.vcf.gz</i> [<i>B.vcf.gz</i>]</a></h2>
Parses VCF or BCF and produces text file stats which is suitable for machine
  processing and can be plotted using <b>plot-vcfstats</b>. When two files are
  given, the program generates separate stats for intersection and the
  complements. By default only sites are compared, <b>-s</b>/<b>-S</b> must
  given to include also sample columns.
<div class="Pp"></div>
<b>-1, --1st-allele-only</b>
<div style="margin-left: 4.00ex;">consider only the 1st alternate allele at
  multiallelic sites</div>
<div class="Pp"></div>
<b>-c, --collapse</b>
  <i>snps</i>|<i>indels</i>|<i>both</i>|<i>all</i>|<i>some</i>| <i>none</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-d, --depth</b> <i>INT</i>,<i>INT</i>,<i>INT</i>
<div style="margin-left: 4.00ex;">ranges of depth distribution: min, max, and
  size of the bin</div>
<div class="Pp"></div>
<b>--debug</b>
<div style="margin-left: 4.00ex;">produce verbose per-site and per-sample
  output</div>
<div class="Pp"></div>
<b>-e, --exclude</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">exclude sites for which <i>EXPRESSION</i> is
  true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-E, --exons</b> <i>file.gz</i>
<div style="margin-left: 4.00ex;">tab-delimited file with exons for indel
  frameshifts statistics. The columns of the file are CHR, FROM, TO, with
  1-based, inclusive, positions. The file is BGZF-compressed and indexed with
  tabix</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
    tabix -s1 -b2 -e3 file.gz
</pre>
</div>
<div class="Pp"></div>
<b>-f, --apply-filters</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-F, --fasta-ref</b> <i>ref.fa</i>
<div style="margin-left: 4.00ex;">faidx indexed reference sequence file to
  determine INDEL context</div>
<div class="Pp"></div>
<b>-i, --include</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">include only sites for which <i>EXPRESSION</i>
  is true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-I, --split-by-ID</b>
<div style="margin-left: 4.00ex;">collect stats separately for sites which have
  the ID column set (&quot;known sites&quot;) or which do not have the ID column
  set (&quot;novel sites&quot;).</div>
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-s, --samples</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-S, --samples-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-t, --targets</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-u, --user-tstv</b> <i>&lt;TAG[:min:max:n]&gt;</i>
<div style="margin-left: 4.00ex;">collect Ts/Tv stats for any tag using the
  given binning [0:1:100]</div>
<div class="Pp"></div>
<b>-v, --verbose</b>
<div style="margin-left: 4.00ex;">produce verbose per-site and per-sample
  output</div>
<h2 class="Ss" title="Ss" id="bcftools_view_[OPTIONS]_file.vcf.gz_[REGION_[...]]"><a class="selflink" href="#bcftools_view_[OPTIONS]_file.vcf.gz_[REGION_[...]]">bcftools
  view [ <i>OPTIONS</i>] <i>file.vcf.gz</i> [<i>REGION</i> [...]]</a></h2>
View, subset and filter VCF or BCF files by position and filtering expression.
  Convert between VCF and BCF. Former <b>bcftools subset</b>.
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Output options</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-G, --drop-genotypes</b>
<div style="margin-left: 4.00ex;">drop individual genotype information (after
  subsetting if <b>-s</b> option is set)</div>
<div class="Pp"></div>
<b>-h, --header-only</b>
<div style="margin-left: 4.00ex;">output the VCF header only</div>
<div class="Pp"></div>
<b>-H, --no-header</b>
<div style="margin-left: 4.00ex;">suppress the header in VCF output</div>
<div class="Pp"></div>
<b>-l, --compression-level</b> [<i>0-9</i>]
<div style="margin-left: 4.00ex;">compression level. 0 stands for uncompressed,
  1 for best speed and 9 for best compression.</div>
<div class="Pp"></div>
<b>--no-version</b>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-O, --output-type</b> <i>b</i>|<i>u</i>|<i>z</i>|<i>v</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>-o, --output-file</b> <i>FILE</i>: output file name. If not present, the
  default is to print to standard output (stdout).
<div class="Pp"></div>
<b>-r, --regions</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-R, --regions-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-t, --targets</b>
  <i>chr</i>|<i>chr:pos</i>|<i>chr:from-to</i>|<i>chr:from-</i>[,...]
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-T, --targets-file</b> <i>file</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>--threads</b> <i>INT</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Subset options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-a, --trim-alt-alleles</b>
<div style="margin-left: 4.00ex;">trim alternate alleles not seen in subset.
  Type A, G and R INFO and FORMAT fields will also be trimmed</div>
<div class="Pp"></div>
<b>--force-samples</b>
<div style="margin-left: 4.00ex;">only warn about unknown subset samples</div>
<div class="Pp"></div>
<b>-I, --no-update</b>
<div style="margin-left: 4.00ex;">do not (re)calculate INFO fields for the
  subset (currently INFO/AC and INFO/AN)</div>
<div class="Pp"></div>
<b>-s, --samples</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-S, --samples-file</b> <i>FILE</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Filter options:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
Note that filter options below dealing with counting the number of alleles will,
  for speed, first check for the values of AC and AN in the INFO column to avoid
  parsing all the genotype (FORMAT/GT) fields in the VCF. This means that a
  filter like <i>--min-af 0.1</i> will be based &#x2018;AC/AN&#x2019; where AC
  and AN come from either INFO/AC and INFO/AN if available or FORMAT/GT if not.
  It will not filter on another field like INFO/AF. The <i>--include</i> and
  <i>--exclude</i> filter expressions should instead be used to explicitly
  filter based on fields in the INFO column, e.g. <i>--exclude AF&lt;0.1</i>.
<div class="Pp"></div>
<b>-c, --min-ac</b>
  <i>INT</i>[<i>:nref</i>|<i>:alt1</i>|<i>:minor</i>|<i>:major</i>|:'nonmajor']
<div style="margin-left: 4.00ex;">minimum allele count (INFO/AC) of sites to be
  printed. Specifying the type of allele is optional and can be set to
  non-reference ( <i>nref</i>, the default), 1st alternate (<i>alt1</i>), the
  least frequent ( <i>minor</i>), the most frequent (<i>major</i>) or sum of all
  but the most frequent ( <i>nonmajor</i>) alleles.</div>
<div class="Pp"></div>
<b>-C, --max-ac</b>
  <i>INT</i>[<i>:nref</i>|<i>:alt1</i>|<i>:minor</i>|:'major'|:'nonmajor']
<div style="margin-left: 4.00ex;">maximum allele count (INFO/AC) of sites to be
  printed. Specifying the type of allele is optional and can be set to
  non-reference ( <i>nref</i>, the default), 1st alternate (<i>alt1</i>), the
  least frequent ( <i>minor</i>), the most frequent (<i>major</i>) or sum of all
  but the most frequent ( <i>nonmajor</i>) alleles.</div>
<div class="Pp"></div>
<b>-e, --exclude</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">exclude sites for which <i>EXPRESSION</i> is
  true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-f, --apply-filters</b> <i>LIST</i>
<div style="margin-left: 4.00ex;">see <b>Common Options</b></div>
<div class="Pp"></div>
<b>-g, --genotype</b> [^][<i>hom</i>|<i>het</i>|<i>miss</i>]
<div style="margin-left: 4.00ex;">include only sites with one or more homozygous
  ( <i>hom</i>), heterozygous (<i>het</i>) or missing (<i>miss</i>) genotypes.
  When prefixed with <i>^</i>, the logic is reversed; thus <i>^het</i> excludes
  sites with heterozygous genotypes.</div>
<div class="Pp"></div>
<b>-i, --include</b> <i>EXPRESSION</i>
<div style="margin-left: 4.00ex;">include sites for which <i>EXPRESSION</i> is
  true. For valid expressions see <b>EXPRESSIONS</b>.</div>
<div class="Pp"></div>
<b>-k, --known</b>
<div style="margin-left: 4.00ex;">print known sites only (ID column is not
  &quot;.&quot;)</div>
<div class="Pp"></div>
<b>-m, --min-alleles</b> <i>INT</i>
<div style="margin-left: 4.00ex;">print sites with at least <i>INT</i> alleles
  listed in REF and ALT columns</div>
<div class="Pp"></div>
<b>-M, --max-alleles</b> <i>INT</i>
<div style="margin-left: 4.00ex;">print sites with at most <i>INT</i> alleles
  listed in REF and ALT columns. Use <b>-m2 -M2 -v snps</b> to only view
  biallelic SNPs.</div>
<div class="Pp"></div>
<b>-n, --novel</b>
<div style="margin-left: 4.00ex;">print novel sites only (ID column is
  &quot;.&quot;)</div>
<div class="Pp"></div>
<b>-p, --phased</b>
<div style="margin-left: 4.00ex;">print sites where all samples are phased.
  Haploid genotypes are considered phased. Missing genotypes considered unphased
  unless the phased bit is set.</div>
<div class="Pp"></div>
<b>-P, --exclude-phased</b>
<div style="margin-left: 4.00ex;">exclude sites where all samples are
  phased</div>
<div class="Pp"></div>
<b>-q, --min-af</b>
  <i>FLOAT</i>[<i>:nref</i>|<i>:alt1</i>|<i>:minor</i>|<i>:major</i>|
  <i>:nonmajor</i>]
<div style="margin-left: 4.00ex;">minimum allele frequency (INFO/AC / INFO/AN)
  of sites to be printed. Specifying the type of allele is optional and can be
  set to non-reference ( <i>nref</i>, the default), 1st alternate (<i>alt1</i>),
  the least frequent ( <i>minor</i>), the most frequent (<i>major</i>) or sum of
  all but the most frequent ( <i>nonmajor</i>) alleles.</div>
<div class="Pp"></div>
<b>-Q, --max-af</b>
  <i>FLOAT</i>[<i>:nref</i>|<i>:alt1</i>|<i>:minor</i>|<i>:major</i>|
  <i>:nonmajor</i>]
<div style="margin-left: 4.00ex;">maximum allele frequency (INFO/AC / INFO/AN)
  of sites to be printed. Specifying the type of allele is optional and can be
  set to non-reference ( <i>nref</i>, the default), 1st alternate (<i>alt1</i>),
  the least frequent ( <i>minor</i>), the most frequent (<i>major</i>) or sum of
  all but the most frequent ( <i>nonmajor</i>) alleles.</div>
<div class="Pp"></div>
<b>-u, --uncalled</b>
<div style="margin-left: 4.00ex;">print sites without a called genotype</div>
<div class="Pp"></div>
<b>-U, --exclude-uncalled</b>
<div style="margin-left: 4.00ex;">exclude sites without a called genotype</div>
<div class="Pp"></div>
<b>-v, --types</b> <i>snps</i>|<i>indels</i>|<i>mnps</i>|<i>other</i>
<div style="margin-left: 4.00ex;">comma-separated list of variant types to
  select. Site is selected if any of the ALT alleles is of the type requested.
  Types are determined by comparing the REF and ALT alleles in the VCF record
  not INFO tags like INFO/INDEL or INFO/VT. Use <b>--include</b> to select based
  on INFO tags.</div>
<div class="Pp"></div>
<b>-V, --exclude-types</b> <i>snps</i>|<i>indels</i>|<i>mnps</i>|<i>other</i>
<div style="margin-left: 4.00ex;">comma-separated list of variant types to
  exclude. Site is excluded if any of the ALT alleles is of the type requested.
  Types are determined by comparing the REF and ALT alleles in the VCF record
  not INFO tags like INFO/INDEL or INFO/VT. Use <b>--exclude</b> to exclude
  based on INFO tags.</div>
<div class="Pp"></div>
<b>-x, --private</b>
<div style="margin-left: 4.00ex;">print sites where only the subset samples
  carry an non-reference allele. Requires <b>--samples</b> or
  <b>--samples-file</b>.</div>
<div class="Pp"></div>
<b>-X, --exclude-private</b>
<div style="margin-left: 4.00ex;">exclude sites where only the subset samples
  carry an non-reference allele</div>
</div>
<h2 class="Ss" title="Ss" id="bcftools_help_[COMMAND]_|_bcftools_--help_[COMMAND]"><a class="selflink" href="#bcftools_help_[COMMAND]_|_bcftools_--help_[COMMAND]">bcftools
  help [ <i>COMMAND</i>] | bcftools --help [<i>COMMAND</i>]</a></h2>
Display a brief usage message listing the bcftools commands available. If the
  name of a command is also given, e.g., bcftools help view, the detailed usage
  message for that particular command is displayed.
<h2 class="Ss" title="Ss" id="bcftools_[--version|-v]"><a class="selflink" href="#bcftools_[--version|-v]">bcftools
  [ <i>--version</i>|<i>-v</i>]</a></h2>
Display the version numbers and copyright information for bcftools and the
  important libraries used by bcftools.
<h2 class="Ss" title="Ss" id="bcftools_[--version-only]"><a class="selflink" href="#bcftools_[--version-only]">bcftools
  [ <i>--version-only</i>]</a></h2>
Display the full bcftools version number in a machine-readable format.
<h1 class="Sh" title="Sh" id="EXPRESSIONS"><a class="selflink" href="#EXPRESSIONS">EXPRESSIONS</a></h1>
These filtering expressions are accepted by <b>annotate</b>, <b>filter</b>,
  <b>query</b> and <b>view</b> commands.
<div class="Pp"></div>
<b>Valid expressions may contain:</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;numerical constants, string constants,
  file names
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
1, 1.0, 1e-4
&quot;String&quot;
@file_name
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;arithmetic operators
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
+,*,-,/
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;comparison operators
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
== (same as =), &gt;, &gt;=, &lt;=, &lt;, !=
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;regex operators &quot;~&quot; and its
  negation &quot;!~&quot;
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
INFO/HAYSTACK ~ &quot;needle&quot;
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;parentheses
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
(, )
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;logical operators
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
&amp;&amp; (same as &amp;), ||,  |
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;INFO tags, FORMAT tags, column names
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
INFO/DP or DP
FORMAT/DV, FMT/DV, or DV
FILTER, QUAL, ID, POS, REF, ALT[0]
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;1 (or 0) to test the presence (or
  absence) of a flag
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
FlagA=1 &amp;&amp; FlagB=0
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;&quot;.&quot; to test missing values
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
DP=&quot;.&quot;, DP!=&quot;.&quot;, ALT=&quot;.&quot;
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;missing genotypes can be matched
  regardless of phase and ploidy (&quot;.|.&quot;, &quot;./.&quot;,
  &quot;.&quot;) using this expression
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
GT=&quot;.&quot;
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;TYPE for variant type in REF,ALT
  columns (indel,snp,mnp,ref,other)
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
TYPE=&quot;indel&quot; | TYPE=&quot;snp&quot;
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;array subscripts, &quot;*&quot; for
  any field
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
(DP4[0]+DP4[1])/(DP4[2]+DP4[3]) &gt; 0.3
DP4[*] == 0
CSQ[*] ~ &quot;missense_variant.*deleterious&quot;
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;function on FORMAT tags (over samples)
  and INFO tags (over vector fields)
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
MAX, MIN, AVG, SUM, STRLEN, ABS
</pre>
</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;variables calculated on the fly if not
  present: number of alternate alleles; number of samples; count of alternate
  alleles; minor allele count (similar to AC but is always smaller than 0.5);
  frequency of alternate alleles (AF=AC/AN); frequency of minor alleles
  (MAF=MAC/AN); number of alleles in called genotypes
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
N_ALT, N_SAMPLES, AC, MAC, AF, MAF, AN
</pre>
</div>
</div>
<div class="Pp"></div>
<b>Notes:</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;String comparisons and regular
  expressions are case-insensitive</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;If the subscript &quot;*&quot; is used
  in regular expression search, the whole field is treated as one string. For
  example, the regex STR[*]~&quot;B,C&quot; will be true for the string vector
  INFO/STR=AB,CD.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Variables and function names are
  case-insensitive, but not tag names. For example, &quot;qual&quot; can be used
  instead of &quot;QUAL&quot;, &quot;strlen()&quot; instead of
  &quot;STRLEN()&quot; , but not &quot;dp&quot; instead of &quot;DP&quot;.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>Examples:</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
MIN(DV)&gt;5
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
MIN(DV/DP)&gt;0.3
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
MIN(DP)&gt;10 &amp; MIN(DV)&gt;3
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
FMT/DP&gt;10  &amp; FMT/GQ&gt;10 .. both conditions must be satisfied within one sample
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
FMT/DP&gt;10 &amp;&amp; FMT/GQ&gt;10 .. the conditions can be satisfied in different samples
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
QUAL&gt;10 |  FMT/GQ&gt;10   .. selects only GQ&gt;10 samples
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
QUAL&gt;10 || FMT/GQ&gt;10   .. selects all samples at QUAL&gt;10 sites
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
TYPE=&quot;snp&quot; &amp;&amp; QUAL&gt;=10 &amp;&amp; (DP4[2]+DP4[3] &gt; 2)
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
MIN(DP)&gt;35 &amp;&amp; AVG(GQ)&gt;50
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
ID=@file       .. selects lines with ID present in the file
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
ID!=@~/file    .. skip lines with ID present in the ~/file
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
MAF[0]&lt;0.05    .. select rare variants at 5% cutoff
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
POS&gt;=100   .. restrict your range query, e.g. 20:100-200 to strictly sites with POS in that range.
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>Shell expansion:</b>
<div style="height: 1.00em;">&#x00A0;</div>
Note that expressions must often be quoted because some characters have special
  meaning in the shell. An example of expression enclosed in single quotes which
  cause that the whole expression is passed to the program as intended:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
bcftools view -i '%ID!=&quot;.&quot; &amp; MAF[0]&lt;0.01'
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
Please refer to the documentation of your shell for details.
<h1 class="Sh" title="Sh" id="SCRIPTS_AND_OPTIONS"><a class="selflink" href="#SCRIPTS_AND_OPTIONS">SCRIPTS
  AND OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="plot-vcfstats_[OPTIONS]_file.vchk_[...]"><a class="selflink" href="#plot-vcfstats_[OPTIONS]_file.vchk_[...]">plot-vcfstats
  [ <i>OPTIONS</i>] <i>file.vchk</i> [...]</a></h2>
Script for processing output of <b>bcftools stats</b>. It can merge results from
  multiple outputs (useful when running the stats for each chromosome
  separately), plots graphs and creates a PDF presentation.
<div class="Pp"></div>
<b>-m, --merge</b>
<div style="margin-left: 4.00ex;">Merge vcfstats files to STDOUT, skip
  plotting.</div>
<div class="Pp"></div>
<b>-p, --prefix</b> <i>PATH</i>
<div style="margin-left: 4.00ex;">The output files prefix, add a slash to create
  new directory.</div>
<div class="Pp"></div>
<b>-P, --no-PDF</b>
<div style="margin-left: 4.00ex;">Skip the PDF creation step.</div>
<div class="Pp"></div>
<b>-r, --rasterize</b>
<div style="margin-left: 4.00ex;">Rasterize PDF images for faster
  rendering.</div>
<div class="Pp"></div>
<b>-s, --sample-names</b>
<div style="margin-left: 4.00ex;">Use sample names for xticks rather than
  numeric IDs.</div>
<div class="Pp"></div>
<b>-t, --title</b> <i>STRING</i>
<div style="margin-left: 4.00ex;">Identify files by these titles in plots. The
  option can be given multiple times, for each ID in the <b>bcftools stats</b>
  output. If not present, the script will use abbreviated source file names for
  the titles.</div>
<div class="Pp"></div>
<b>-T, --main-title</b> <i>STRING</i>
<div style="margin-left: 4.00ex;">Main title for the PDF.</div>
<h1 class="Sh" title="Sh" id="PERFORMANCE"><a class="selflink" href="#PERFORMANCE">PERFORMANCE</a></h1>
HTSlib was designed with BCF format in mind. When parsing VCF files, all records
  are internally converted into BCF representation. Simple operations, like
  removing a single column from a VCF file, can be therefore done much faster
  with standard UNIX commands, such as <b>awk</b> or <b>cut</b>. Therefore it is
  recommended to use BCF as input/output format whenever possible to avoid large
  overhead of the VCF &#x2192; BCF &#x2192; VCF conversion.
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Please report any bugs you encounter on the github website:
  http://github.com/samtools/bcftools
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
Heng Li from the Sanger Institute wrote the original C version of htslib,
  samtools and bcftools. Bob Handsaker from the Broad Institute implemented the
  BGZF library. Petr Danecek, Shane McCarthy and John Marshall are maintaining
  and further developing bcftools. Many other people contributed to the program
  and to the file format specifications, both directly and indirectly by
  providing patches, testing and reporting bugs. We thank them all.
<h1 class="Sh" title="Sh" id="RESOURCES"><a class="selflink" href="#RESOURCES">RESOURCES</a></h1>
BCFtools GitHub website: http://github.com/samtools/bcftools
<div style="height: 1.00em;">&#x00A0;</div>
Samtools GitHub website: http://github.com/samtools/samtools
<div style="height: 1.00em;">&#x00A0;</div>
HTSlib GitHub website: http://github.com/samtools/htslib
<div style="height: 1.00em;">&#x00A0;</div>
File format specifications: http://samtools.github.io/hts-specs
<div style="height: 1.00em;">&#x00A0;</div>
BCFtools documentation: http://samtools.github.io/bcftools
<div style="height: 1.00em;">&#x00A0;</div>
BCFtools wiki page: https://github.com/samtools/bcftools/wiki
<h1 class="Sh" title="Sh" id="COPYING"><a class="selflink" href="#COPYING">COPYING</a></h1>
The MIT/Expat License or GPL License, see the LICENSE document for details.
  Copyright (c) Genome Research Ltd.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-04-18 14:18 BST</td>
    <td class="foot-os">&#x00A0;</td>
  </tr>
</table>
</body>
</html>
