<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:54:28 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BCFTOOLS(1) BCFTOOLS(1)</p>

<p style="margin-top: 1em">NAME <br>
bcftools - utilities for variant calling and manipulating
VCFs and BCFs.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bcftools [--version|--version-only] [--help] [COMMAND]
[OPTIONS]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
BCFtools is a set of utilities that manipulate variant calls
in the Variant Call Format (VCF) and its binary counterpart
BCF. All commands work transparently with both VCFs and <br>
BCFs, both uncompressed and BGZF-compressed.</p>

<p style="margin-top: 1em">Most commands accept VCF,
bgzipped VCF and BCF with filetype detected automatically
even when streaming from a pipe. Indexed VCF and BCF will
work in all situations. Un-indexed <br>
VCF and BCF and streams will work in most, but not all
situations. In general, whenever multiple VCFs are read
simultaneously, they must be indexed and therefore also
compressed.</p>

<p style="margin-top: 1em">BCFtools is designed to work on
a stream. It regards an input file &quot;-&quot; as the
standard input (stdin) and outputs to the standard output
(stdout). Several commands can thus be <br>
combined with Unix pipes.</p>

<p style="margin-top: 1em">VERSION <br>
This manual page was last updated 2016-04-18 14:18 BST and
refers to bcftools git version 1.3-36-g47e811c+.</p>

<p style="margin-top: 1em">BCF1 <br>
The BCF1 format output by versions of samtools &lt;= 0.1.19
is not compatible with this version of bcftools. To read
BCF1 files one can use the view command from old versions of
<br>
bcftools packaged with samtools versions &lt;= 0.1.19 to
convert to VCF, which can then be read by this version of
bcftools.</p>


<p style="margin-top: 1em">samtools-0.1.19/bcftools/bcftools
view file.bcf1 | bcftools view</p>

<p style="margin-top: 1em">VARIANT CALLING <br>
See bcftools call for variant calling from the output of the
samtools mpileup command. In versions of samtools &lt;=
0.1.19 calling was done with bcftools view. Users are now
<br>
required to choose between the old samtools calling model
(-c/--consensus-caller) and the new multiallelic calling
model (-m/--multiallelic-caller). The multiallelic calling
<br>
model is recommended for most tasks.</p>

<p style="margin-top: 1em">LIST OF COMMANDS <br>
For a full list of available commands, run bcftools without
arguments. For a full list of available options, run
bcftools COMMAND without arguments.</p>

<p style="margin-top: 1em">&Acirc;&middot; annotate .. edit
VCF files, add or remove annotations</p>

<p style="margin-top: 1em">&Acirc;&middot; call ..
SNP/indel calling (former &quot;view&quot;)</p>

<p style="margin-top: 1em">&Acirc;&middot; cnv .. Copy
Number Variation caller</p>

<p style="margin-top: 1em">&Acirc;&middot; concat ..
concatenate VCF/BCF files from the same set of samples</p>

<p style="margin-top: 1em">&Acirc;&middot; consensus ..
create consensus sequence by applying VCF variants</p>

<p style="margin-top: 1em">&Acirc;&middot; convert ..
convert VCF/BCF to other formats and back</p>

<p style="margin-top: 1em">&Acirc;&middot; filter .. filter
VCF/BCF files using fixed thresholds</p>

<p style="margin-top: 1em">&Acirc;&middot; gtcheck .. check
sample concordance, detect sample swaps and
contamination</p>

<p style="margin-top: 1em">&Acirc;&middot; index .. index
VCF/BCF</p>

<p style="margin-top: 1em">&Acirc;&middot; isec ..
intersections of VCF/BCF files</p>

<p style="margin-top: 1em">&Acirc;&middot; merge .. merge
VCF/BCF files files from non-overlapping sample sets</p>

<p style="margin-top: 1em">&Acirc;&middot; norm ..
normalize indels</p>

<p style="margin-top: 1em">&Acirc;&middot; plugin .. run
user-defined plugin</p>

<p style="margin-top: 1em">&Acirc;&middot; polysomy ..
detect contaminations and whole-chromosome aberrations</p>

<p style="margin-top: 1em">&Acirc;&middot; query ..
transform VCF/BCF into user-defined formats</p>

<p style="margin-top: 1em">&Acirc;&middot; reheader ..
modify VCF/BCF header, change sample names</p>

<p style="margin-top: 1em">&Acirc;&middot; roh .. identify
runs of homo/auto-zygosity</p>

<p style="margin-top: 1em">&Acirc;&middot; stats .. produce
VCF/BCF stats (former vcfcheck)</p>

<p style="margin-top: 1em">&Acirc;&middot; view .. subset,
filter and convert VCF and BCF files</p>

<p style="margin-top: 1em">LIST OF SCRIPTS <br>
Some helper scripts are bundled with the bcftools code.</p>

<p style="margin-top: 1em">&Acirc;&middot; plot-vcfstats ..
plots the output of stats</p>

<p style="margin-top: 1em">COMMANDS AND OPTIONS <br>
Common Options <br>
The following options are common to many bcftools commands.
See usage for specific commands to see if they apply.</p>

<p style="margin-top: 1em">FILE <br>
Files can be both VCF or BCF, uncompressed or
BGZF-compressed. The file &quot;-&quot; is interpreted as
standard input. Some tools may require tabix- or CSI-indexed
files.</p>

<p style="margin-top: 1em">-c, --collapse
snps|indels|both|all|some|none|id <br>
Controls how to treat records with duplicate positions and
defines compatible records across multiple input files. Here
by &quot;compatible&quot; we mean records which should be
<br>
considered as identical by the tools. For example, when
performing line intersections, the desire may be to consider
as identical all sites with matching positions (bcftools
<br>
isec -c all), or only sites with matching variant type
(bcftools isec -c snps -c indels), or only sites with all
alleles identical (bcftools isec -c none).</p>

<p style="margin-top: 1em">none <br>
only records with identical REF and ALT alleles are
compatible</p>

<p style="margin-top: 1em">some <br>
only records where some subset of ALT alleles match are
compatible</p>

<p style="margin-top: 1em">all <br>
all records are compatible, regardless of whether the ALT
alleles match or not. In the case of records with the same
position, only the first will be considered and <br>
appear on output.</p>

<p style="margin-top: 1em">snps <br>
any SNP records are compatible, regardless of whether the
ALT alleles match or not. For duplicate positions, only the
first SNP record will be considered and appear on <br>
output.</p>

<p style="margin-top: 1em">indels <br>
all indel records are compatible, regardless of whether the
REF and ALT alleles match or not. For duplicate positions,
only the first indel record will be considered and <br>
appear on output.</p>

<p style="margin-top: 1em">both <br>
abbreviation of &quot;-c indels -c snps&quot;</p>

<p style="margin-top: 1em">id <br>
only records with identical ID column are compatible.
Supported by bcftools merge only.</p>

<p style="margin-top: 1em">-f, --apply-filters LIST <br>
Skip sites where FILTER column does not contain any of the
strings listed in LIST. For example, to include only sites
which have no filters set, use -f .,PASS.</p>

<p style="margin-top: 1em">--no-version <br>
Do not append version and command line information to the
output VCF header.</p>

<p style="margin-top: 1em">-o, --output FILE <br>
When output consists of a single stream, write it to FILE
rather than to standard output, where it is written by
default.</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
Output compressed BCF (b), uncompressed BCF (u), compressed
VCF (z), uncompressed VCF (v). Use the -Ou option when
piping between bcftools subcommands to speed up performance
<br>
by removing unnecessary compression/decompression and
VCF&acirc;&acirc;BCF conversion.</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
Comma-separated list of regions, see also -R,
--regions-file. Note that -r cannot be used in combination
with -R.</p>

<p style="margin-top: 1em">-R, --regions-file FILE <br>
Regions can be specified either on command line or in a VCF,
BED, or tab-delimited file (the default). The columns of the
tab-delimited file are: CHROM, POS, and, optionally, <br>
POS_TO, where positions are 1-based and inclusive. The
columns of the tab-delimited BED file are also CHROM, POS
and POS_TO (trailing columns are ignored), but coordinates
<br>
are 0-based, half-open. To indicate that a file be treated
as BED rather than the 1-based tab-delimited file, the file
must have the &quot;.bed&quot; or &quot;.bed.gz&quot; suffix
<br>
(case-insensitive). Uncompressed files are stored in memory,
while bgzip-compressed and tabix-indexed region files are
streamed. Note that sequence names must match exactly, <br>
&quot;chr20&quot; is not the same as &quot;20&quot;. Also
note that chromosome ordering in FILE will be respected, the
VCF will be processed in the order in which chromosomes
first appear in <br>
FILE. However, within chromosomes, the VCF will always be
processed in ascending genomic coordinate order no matter
what order they appear in FILE. Note that overlapping <br>
regions in FILE can result in duplicated out of order
positions in the output. This option requires indexed
VCF/BCF files. Note that -R cannot be used in combination
with -r.</p>

<p style="margin-top: 1em">-s, --samples [^]LIST <br>
Comma-separated list of samples to include or exclude if
prefixed with &quot;^&quot;. Note that in general tags such
as INFO/AC, INFO/AN, etc are not updated to correspond to
the <br>
subset samples. bcftools view is the exception where some
tags will be updated (unless the -I, --no-update option is
used; see bcftools view documentation). To use updated <br>
tags for the subset in another command one can pipe from
view into that command. For example:</p>

<p style="margin-top: 1em">bcftools view -Ou -s
sample1,sample2 file.vcf | bcftools query -f
%INFO/AC%INFO/AN</p>

<p style="margin-top: 1em">-S, --samples-file FILE <br>
File of sample names to include or exclude if prefixed with
&quot;^&quot;. One sample per line. See also the note above
for the -s, --samples option. The command bcftools call
accepts <br>
an optional second column indicating ploidy (0, 1 or 2) or
sex (as defined by --ploidy, for example &quot;F&quot; or
&quot;M&quot;), and can parse also PED files. If the second
column is not <br>
present, the sex &quot;F&quot; is assumed. With bcftools
call -C trio, PED file is expected. File formats
examples:</p>

<p style="margin-top: 1em">sample1 1 <br>
sample2 2 <br>
sample3 2</p>

<p style="margin-top: 1em">or</p>

<p style="margin-top: 1em">sample1 M <br>
sample2 F <br>
sample3 F</p>

<p style="margin-top: 1em">or a .ped file (here is shown a
minimum working example, the first column is <br>
ignored and the last indicates sex: 1=male, 2=female)</p>

<p style="margin-top: 1em">ignored daughterA fatherA
motherA 2 <br>
ignored sonB fatherB motherB 1</p>

<p style="margin-top: 1em">-t, --targets
[^]chr|chr:pos|chr:from-to|chr:from-[,...] <br>
Similar as -r, --regions, but the next position is accessed
by streaming the whole VCF/BCF rather than using the tbi/csi
index. Both -r and -t options can be applied <br>
simultaneously: -r uses the index to jump to a region and -t
discards positions which are not in the targets. Unlike -r,
targets can be prefixed with &quot;^&quot; to request
logical <br>
complement. For example, &quot;^X,Y,MT&quot; indicates that
sequences X, Y and MT should be skipped. Yet another
difference between the two is that -r checks both start and
end <br>
positions of indels, whereas -t checks start positions only.
Note that -t cannot be used in combination with -T.</p>

<p style="margin-top: 1em">-T, --targets-file [^]FILE <br>
Same -t, --targets, but reads regions from a file. Note that
-T cannot be used in combination with -t.</p>

<p style="margin-top: 1em">With the call -C alleles
command, third column of the targets file must be
comma-separated list of alleles, starting with the reference
allele. Note that the file must be <br>
compressed and index. Such a file can be easily created from
a VCF using:</p>

<p style="margin-top: 1em">bcftools query
-f&rsquo;%CHROM%POS%REF,%ALT0 file.vcf | bgzip -c &gt;
als.tsv.gz &amp;&amp; tabix -s1 -b2 -e2 als.tsv.gz</p>

<p style="margin-top: 1em">--threads INT <br>
Number of output compression threads to use in addition to
main thread. Only used when --output-type is b or z.
Default: 0.</p>

<p style="margin-top: 1em">bcftools annotate [OPTIONS] FILE
<br>
Add or remove annotations.</p>

<p style="margin-top: 1em">-a, --annotations file <br>
Bgzip-compressed and tabix-indexed file with annotations.
The file can be VCF, BED, or a tab-delimited file with
mandatory columns CHROM, POS (or, alternatively, FROM and
<br>
TO), optional columns REF and ALT, and arbitrary number of
annotation columns. BED files are expected to have the
&quot;.bed&quot; or &quot;.bed.gz&quot; suffix
(case-insensitive), otherwise a <br>
tab-delimited file is assumed. Note that in case of
tab-delimited file, the coordinates POS, FROM and TO are
one-based and inclusive. When REF and ALT are present, only
<br>
matching VCF records will be annotated. When multiple ALT
alleles are present in the annotation file (given as
comma-separated list of alleles), at least one must match
one <br>
of the alleles in the corresponding VCF record. Similarly,
at least one alternate allele from a multi-allelic VCF
record must be present in the annotation file. Note that
<br>
flag types, such as &quot;INFO/FLAG&quot;, can be annotated
by including a field with the value &quot;1&quot; to set the
flag, &quot;0&quot; to remove it, or &quot;.&quot; to keep
existing flags. See also -c, <br>
--columns and -h, --header-lines.</p>

<p style="margin-top: 1em"># Sample annotation file with
columns CHROM, POS, STRING_TAG, NUMERIC_TAG <br>
1 752566 SomeString 5 <br>
1 798959 SomeOtherString 6 <br>
# etc.</p>

<p style="margin-top: 1em">-c, --columns list <br>
Comma-separated list of columns or tags to carry over from
the annotation file (see also -a, --annotations). If the
annotation file is not a VCF/BCF, list describes the <br>
columns of the annotation file and must include CHROM, POS
(or, alternatively, FROM and TO), and optionally REF and
ALT. Unused columns which should be ignored can be <br>
indicated by &quot;-&quot;. If the annotation file is a
VCF/BCF, only the edited columns/tags must be present and
their order does not matter. The columns ID, QUAL, FILTER,
INFO and <br>
FORMAT can be edited, where INFO tags can be written both as
&quot;INFO/TAG&quot; or simply &quot;TAG&quot;, and FORMAT
tags can be written as &quot;FORMAT/TAG&quot; or
&quot;FMT/TAG&quot;. To carry over all INFO <br>
annotations, use &quot;INFO&quot;. To add all INFO
annotations except &quot;TAG&quot;, use
&quot;^INFO/TAG&quot;. By default, existing values are
replaced. To add annotations without overwriting existing
<br>
values (that is, to add missing tags or add values to
existing tags with missing values), use &quot;+TAG&quot;
instead of &quot;TAG&quot;. To append to existing values
(rather than replacing or <br>
leaving untouched), use &quot;=TAG&quot; (instead of
&quot;TAG&quot; or &quot;+TAG&quot;). To replace only
existing values without modifying missing annotations, use
&quot;-TAG&quot;. If the annotation file is not <br>
a VCF/BCF, all new annotations must be defined via -h,
--header-lines.</p>

<p style="margin-top: 1em">-e, --exclude EXPRESSION <br>
exclude sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-h, --header-lines file <br>
Lines to append to the VCF header, see also -c, --columns
and -a, --annotations. For example:</p>


<p style="margin-top: 1em">##INFO=&lt;ID=NUMERIC_TAG,Number=1,Type=Integer,Description=&quot;Example
header line&quot;&gt; <br>

##INFO=&lt;ID=STRING_TAG,Number=1,Type=String,Description=&quot;Yet
another header line&quot;&gt;</p>

<p style="margin-top: 1em">-I, --set-id [+]FORMAT <br>
assign ID on the fly. The format is the same as in the query
command (see below). By default all existing IDs are
replaced. If the format string is preceded by &quot;+&quot;,
only <br>
missing IDs will be set. For example, one can use</p>

<p style="margin-top: 1em">bcftools annotate --set-id
+&rsquo;%CHROM_%POS_%REF_%FIRST_ALT&rsquo; file.vcf</p>

<p style="margin-top: 1em">-i, --include EXPRESSION <br>
include only sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-m, --mark-sites TAG <br>
annotate sites which are present (&quot;+&quot;) or absent
(&quot;-&quot;) in the -a file with a new INFO/TAG flag</p>

<p style="margin-top: 1em">--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">--rename-chrs file <br>
rename chromosomes according to the map in file, with
&quot;old_name new_name0 pairs separated by whitespaces,
each on a separate line.</p>

<p style="margin-top: 1em">-s, --samples [^]LIST <br>
subset of samples to annotate, see also Common Options</p>

<p style="margin-top: 1em">-S, --samples-file FILE <br>
subset of samples to annotate. If the samples are named
differently in the target VCF and the -a, --annotations VCF,
the name mapping can be given as &quot;src_name dst_name0,
<br>
separated by whitespaces, each pair on a separate line.</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">-x, --remove list <br>
List of annotations to remove. Use &quot;FILTER&quot; to
remove all filters or &quot;FILTER/SomeFilter&quot; to
remove a specific filter. Similarly, &quot;INFO&quot; can be
used to remove all INFO tags <br>
and &quot;FORMAT&quot; to remove all FORMAT tags except GT.
To remove all INFO tags except &quot;FOO&quot; and
&quot;BAR&quot;, use &quot;^INFO/FOO,INFO/BAR&quot; (and
similarly for FORMAT and FILTER). &quot;INFO&quot; can <br>
be abbreviated to &quot;INF&quot; and &quot;FORMAT&quot; to
&quot;FMT&quot;.</p>

<p style="margin-top: 1em">Examples:</p>

<p style="margin-top: 1em"># Remove three fields <br>
bcftools annotate -x ID,INFO/DP,FORMAT/DP file.vcf.gz</p>

<p style="margin-top: 1em"># Remove all INFO fields and all
FORMAT fields except for GT and PL <br>
bcftools annotate -x INFO,^FORMAT/GT,FORMAT/PL file.vcf</p>

<p style="margin-top: 1em"># Add ID, QUAL and INFO/TAG, not
replacing TAG if already present <br>
bcftools annotate -a src.bcf -c ID,QUAL,+TAG dst.bcf</p>

<p style="margin-top: 1em"># Carry over all INFO and FORMAT
annotations except FORMAT/GT <br>
bcftools annotate -a src.bcf -c INFO,^FORMAT/GT dst.bcf</p>

<p style="margin-top: 1em"># Annotate from a tab-delimited
file with six columns (the fifth is ignored), <br>
# first indexing with tabix. The coordinates are 1-based.
<br>
tabix -s1 -b2 -e2 annots.tab.gz <br>
bcftools annotate -a annots.tab.gz -h annots.hdr -c
CHROM,POS,REF,ALT,-,TAG file.vcf</p>

<p style="margin-top: 1em"># Annotate from a tab-delimited
file with regions (1-based coordinates, inclusive) <br>
tabix -s1 -b2 -e3 annots.tab.gz <br>
bcftools annotate -a annots.tab.gz -h annots.hdr -c
CHROM,FROM,TO,TAG inut.vcf</p>

<p style="margin-top: 1em"># Annotate from a bed file
(0-based coordinates, half-closed, half-open intervals) <br>
bcftools annotate -a annots.bed.gz -h annots.hdr -c
CHROM,FROM,TO,TAG input.vcf</p>

<p style="margin-top: 1em">bcftools cnv [OPTIONS] FILE <br>
Copy number variation caller, requires a VCF annotated with
the Illumina&acirc;s B-allele frequency (BAF) and Log R
Ratio intensity (LRR) values. The HMM considers the
following copy <br>
number states: CN 2 (normal), 1 (single-copy loss), 0
(complete loss), 3 (single-copy gain).</p>

<p style="margin-top: 1em">General Options: <br>
-c, --control-sample string <br>
optional control sample name. If given, pairwise calling is
performed and the -P option can be used</p>

<p style="margin-top: 1em">-f, --AF-file file <br>
read allele frequencies from a tab-delimited file with the
columns CHR,POS,REF,ALT,AF</p>

<p style="margin-top: 1em">*-o, --output-dir path <br>
output directory</p>

<p style="margin-top: 1em">*-p, --plot-threshold float <br>
call matplotlib to produce plots for chromosomes with
quality at least float, useful for visual inspection of the
calls. With -p 0, plots for all chromosomes will be <br>
generated. If not given, a matplotlib script will be created
but not called.</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --query-sample string <br>
query samply name</p>

<p style="margin-top: 1em">-t, --targets LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">HMM Options: <br>
-a, --aberrant float[,float] <br>
fraction of aberrant cells in query and control. The
hallmark of duplications and contaminations is the BAF value
of heterozygous markers which is dependent on the <br>
fraction of aberrant cells. Sensitivity to smaller fractions
of cells can be increased by setting -a to a lower value.
Note however, that this comes at the cost of <br>
increased false discovery rate.</p>

<p style="margin-top: 1em">-b, --BAF-weight float <br>
relative contribution from BAF</p>

<p style="margin-top: 1em">d, --BAF-dev float[,float] <br>
expected BAF deviation in query and control, i.e. the noise
observed in the data.</p>

<p style="margin-top: 1em">-e, --err-prob float <br>
uniform error probability</p>

<p style="margin-top: 1em">-l, --LRR-weight float <br>
relative contribution from LRR. With noisy data, this option
can have big effect on the number of calls produced. In
truly random noise (such as in simulated data), the <br>
value should be set high (1.0), but in the presence of
systematic noise when LRR are not informative, lower values
result in cleaner calls (0.2).</p>

<p style="margin-top: 1em">-L, --LRR-smooth-win int <br>
reduce LRR noise by applying moving average given this
window size</p>

<p style="margin-top: 1em">-O, --optimize float <br>
iteratively estimate the fraction of aberrant cells, down to
the given fraction. Lowering this value from the default 1.0
to say, 0.3, can help discover more events but <br>
also increases noise</p>

<p style="margin-top: 1em">-P, --same-prob float <br>
the prior probability of the query and the control sample
being the same. Setting to 0 calls both independently,
setting to 1 forces the same copy number state in both.</p>

<p style="margin-top: 1em">-x, --xy-prob float <br>
the HMM probability of transition to another copy number
state. Increasing this values leads to smaller and more
frequent calls.</p>

<p style="margin-top: 1em">bcftools call [OPTIONS] FILE
<br>
This command replaces the former bcftools view caller. Some
of the original functionality has been temporarily lost in
the process of transition under htslib, but will be added
<br>
back on popular demand. The original calling model can be
invoked with the -c option.</p>

<p style="margin-top: 1em">File format options: <br>
--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">--ploidy ASSEMBLY[?] <br>
predefined ploidy, use list (or any other unused word) to
print a list of all predefined assemblies. Append a question
mark to print the actual definition. See also <br>
--ploidy-file.</p>

<p style="margin-top: 1em">--ploidy-file FILE <br>
ploidy definition given as a space/tab-delimited list of
CHROM, FROM, TO, SEX, PLOIDY. The SEX codes are arbitrary
and correspond to the ones used by --samples-file. The <br>
default ploidy can be given using the starred records (see
below), unlisted regions have ploidy 2. The default ploidy
definition is</p>

<p style="margin-top: 1em">X 1 60000 M 1 <br>
X 2699521 154931043 M 1 <br>
Y 1 59373566 M 1 <br>
Y 1 59373566 F 0 <br>
MT 1 16569 M 1 <br>
MT 1 16569 F 1 <br>
* * * M 2 <br>
* * * F 2</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --samples LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-S, --samples-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-t, --targets LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">Input/output options: <br>
-A, --keep-alts <br>
output all alternate alleles present in the alignments even
if they do not appear in any of the genotypes</p>

<p style="margin-top: 1em">-f, --format-fields list <br>
comma-separated list of FORMAT fields to output for each
sample. Currently GQ and GP fields are supported. For
convenience, the fields can be given as lower case
letters.</p>

<p style="margin-top: 1em">-g, --gvcf INT <br>
output also gVCF blocks of homozygous REF calls. The
parameter INT is the minimum per-sample depth required to
include a site in the non-variant block.</p>

<p style="margin-top: 1em">-i, --insert-missed INT <br>
output also sites missed by mpileup but present in -T,
--targets-file.</p>

<p style="margin-top: 1em">-M, --keep-masked-ref <br>
output sites where REF allele is N</p>

<p style="margin-top: 1em">-V, --skip-variants snps|indels
<br>
skip indel/SNP sites</p>

<p style="margin-top: 1em">-v, --variants-only <br>
output variant sites only</p>

<p style="margin-top: 1em">Consensus/variant calling
options: <br>
-c, --consensus-caller <br>
the original samtools/bcftools calling method (conflicts
with -m)</p>

<p style="margin-top: 1em">-C, --constrain alleles|trio</p>

<p style="margin-top: 1em">alleles <br>
call genotypes given alleles. See also -T,
--targets-file.</p>

<p style="margin-top: 1em">trio <br>
call genotypes given the father-mother-child constraint. See
also -s, --samples and -n, --novel-rate.</p>

<p style="margin-top: 1em">-m, --multiallelic-caller <br>
alternative modelfor multiallelic and rare-variant calling
designed to overcome known limitations in -c calling model
(conflicts with -c)</p>

<p style="margin-top: 1em">-n, --novel-rate float[,...]
<br>
likelihood of novel mutation for constrained -C trio
calling. The trio genotype calling maximizes likelihood of a
particular combination of genotypes for father, mother <br>
and the child P(F=i,M=j,C=k) = P(unconstrained) * Pn +
P(constrained) * (1-Pn). By providing three values, the
mutation rate Pn is set explicitly for SNPs, deletions and
<br>
insertions, respectively. If two values are given, the first
is interpreted as the mutation rate of SNPs and the second
is used to calculate the mutation rate of indels <br>
according to their length as Pn=float*exp(-a-b*len), where
a=22.8689, b=0.2994 for insertions and a=21.9313, b=0.2856
for deletions [pubmed:23975140]. If only one value <br>
is given, the same mutation rate Pn is used for SNPs and
indels.</p>

<p style="margin-top: 1em">-p, --pval-threshold float <br>
with -c, accept variant if P(ref|D) &lt; float.</p>

<p style="margin-top: 1em">-P, --prior float <br>
expected substitution rate, or 0 to disable the prior.</p>

<p style="margin-top: 1em">-t, --targets
file|chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-X, --chromosome-X <br>
haploid output for male samples (requires PED file with
-s)</p>

<p style="margin-top: 1em">-Y, --chromosome-Y <br>
haploid output for males and skips females (requires PED
file with -s)</p>

<p style="margin-top: 1em">bcftools concat [OPTIONS] FILE1
FILE2 [...] <br>
Concatenate or combine VCF/BCF files. All source files must
have the same sample columns appearing in the same order.
Can be used, for example, to concatenate chromosome VCFs
<br>
into one VCF, or combine a SNP VCF and an indel VCF into
one. The input files must be sorted by chr and position. The
files must be given in the correct order to produce sorted
<br>
VCF on output unless the -a, --allow-overlaps option is
specified. With the --naive option, the files are
concatenated without being recompressed, which is very fast
but <br>
dangerous if the BCF headers differ.</p>

<p style="margin-top: 1em">-a, --allow-overlaps <br>
First coordinate of the next file can precede last record of
the current file.</p>

<p style="margin-top: 1em">-c, --compact-PS <br>
Do not output PS tag at each site, only at the start of a
new phase set block.</p>

<p style="margin-top: 1em">-d, --rm-dups
snps|indels|both|all|none <br>
Output duplicate records of specified type present in
multiple files only once. Requires -a, --allow-overlaps.</p>

<p style="margin-top: 1em">-D, --remove-duplicates <br>
Alias for -d none</p>

<p style="margin-top: 1em">-f, --file-list FILE <br>
Read the list of files from a file.</p>

<p style="margin-top: 1em">-l, --ligate <br>
Ligate phased VCFs by matching phase at overlapping
haplotypes</p>

<p style="margin-top: 1em">--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-n, --naive <br>
Concatenate BCF files without recompression. This is very
fast but requires that all files have the same headers. This
is because all tags and chromosome names in the BCF <br>
body rely on the implicit order of the contig and tag
definitions in the header. Currently no sanity checks are in
place and only works for compressed BCF files. Dangerous,
<br>
use with caution.</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">-q, --min-PQ INT <br>
Break phase set if phasing quality is lower than INT</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options. Requires -a, --allow-overlaps.</p>

<p style="margin-top: 1em">-R, --regions-file FILE <br>
see Common Options. Requires -a, --allow-overlaps.</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">bcftools consensus [OPTIONS]
FILE <br>
Create consensus sequence by applying VCF variants to a
reference fasta file.</p>

<p style="margin-top: 1em">-f, --fasta-ref FILE <br>
reference sequence in fasta format</p>

<p style="margin-top: 1em">-H, --haplotype 1|2 <br>
apply variants for the given haplotype. This option requires
-s, unless exactly one sample is present in the VCF</p>

<p style="margin-top: 1em">-i, --iupac-codes <br>
output variants in the form of IUPAC ambiguity codes</p>

<p style="margin-top: 1em">-m, --mask FILE <br>
BED file or TAB file with regions to be replaced with N. See
discussion of --regions-file in Common Options for file
format details.</p>

<p style="margin-top: 1em">-o, --output FILE <br>
write output to a file</p>

<p style="margin-top: 1em">-s, --sample NAME <br>
apply variants of the given sample</p>

<p style="margin-top: 1em">Examples:</p>

<p style="margin-top: 1em"># Apply variants present in
sample &quot;NA001&quot;, output IUPAC codes for hets <br>
bcftools consensus -i -s NA001 -f in.fa in.vcf.gz &gt;
out.fa</p>

<p style="margin-top: 1em"># Create consensus for one
region. The fasta header lines are then expected <br>
# in the form &quot;&gt;chr:from-to&quot;. <br>
samtools faidx ref.fa 8:11870-11890 | bcftools consensus
in.vcf.gz -o out.fa</p>

<p style="margin-top: 1em">bcftools convert [OPTIONS] FILE
<br>
VCF input options: <br>
-e, --exclude EXPRESSION <br>
exclude sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-i, --include EXPRESSION <br>
include only sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --samples LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-S, --samples-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-t, --targets LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">VCF output options: <br>
--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">GEN/SAMPLE conversion: <br>
-G, --gensample2vcf prefix or gen-file,sample-file <br>
convert IMPUTE2 output to VCF. The second column must be of
the form &quot;CHROM:POS_REF_ALT&quot; to detect possible
strand swaps; IMPUTE2 leaves the first one empty
(&quot;--&quot;) when <br>
sites from reference panel are filled in. See also -g
below.</p>

<p style="margin-top: 1em">-g, --gensample prefix or
gen-file,sample-file <br>
convert from VCF to gen/sample format used by IMPUTE2 and
SHAPEIT. The columns of .gen file format are ID1,ID2,POS,A,B
followed by three genotype probabilities P(AA), <br>
P(AB), P(BB) for each sample. In order to prevent strand
swaps, the program uses IDs of the form
&quot;CHROM:POS_REF_ALT&quot;. For example:</p>

<p style="margin-top: 1em">.gen <br>
---- <br>
1:111485207_G_A 1:111485207_G_A 111485207 G A 0 1 0 0 1 0
<br>
1:111494194_C_T 1:111494194_C_T 111494194 C T 0 1 0 0 0
1</p>

<p style="margin-top: 1em">.sample <br>
------- <br>
ID_1 ID_2 missing <br>
0 0 0 <br>
sample1 sample1 0 <br>
sample2 sample2 0</p>

<p style="margin-top: 1em">--tag STRING <br>
tag to take values for .gen file: GT,PL,GL,GP</p>

<p style="margin-top: 1em">gVCF conversion: <br>
--gvcf2vcf <br>
convert gVCF to VCF, expanding REF blocks into sites. Only
sites with FILTER set to &quot;PASS&quot; or &quot;.&quot;
will be expanded.</p>

<p style="margin-top: 1em">-f, --fasta-ref file <br>
reference sequence in fasta format. Must be indexed with
samtools faidx</p>

<p style="margin-top: 1em">HAPS/SAMPLE conversion: <br>
--hapsample2vcf prefix or haps-file,sample-file <br>
convert from haps/sample format to VCF. The columns of .haps
file are similar to .gen file above, but there are only two
haplotype columns per sample. Note that the first <br>
column of the haps file is expected to be in the form
&quot;CHR:POS_REF_ALT(_END)?&quot;, with the _END being
optional for defining the INFO/END tag when ALT is a
symbolic allele, <br>
for example:</p>

<p style="margin-top: 1em">.haps <br>
---- <br>
1:111485207_G_A rsID1 111485207 G A 0 1 0 0 <br>
1:111494194_C_T rsID2 111494194 C T 0 1 0 0 <br>
1:111495231_A_&lt;DEL&gt;_111495784 rsID3 111495231 A
&lt;DEL&gt; 0 0 1 0</p>

<p style="margin-top: 1em">--hapsample prefix or
haps-file,sample-file <br>
convert from VCF to haps/sample format used by IMPUTE2 and
SHAPEIT. The columns of .haps file begin with
ID,RSID,POS,REF,ALT. In order to prevent strand swaps, the
<br>
program uses IDs of the form
&quot;CHROM:POS_REF_ALT&quot;.</p>

<p style="margin-top: 1em">--haploid2diploid <br>
with -h option converts haploid genotypes to homozygous
diploid genotypes. For example, the program will print 0 0
instead of the default 0 -. This is useful for programs <br>
which do not handle haploid genotypes correctly.</p>

<p style="margin-top: 1em">--vcf-ids <br>
output VCF IDs instead of &quot;CHROM:POS_REF_ALT&quot;
IDs</p>

<p style="margin-top: 1em">HAPS/LEGEND/SAMPLE conversion:
<br>
-H, --haplegendsample2vcf prefix or
haps-file,legend-file,sample-file <br>
convert from haps/legend/sample format used by IMPUTE2 to
VCF, see also -h, --hapslegendsample below.</p>

<p style="margin-top: 1em">-h, --haplegendsample prefix or
haps-file,legend-file,sample-file <br>
convert from VCF to haps/legend/sample format used by
IMPUTE2 and SHAPEIT. The columns of .legend file
ID,POS,REF,ALT. In order to prevent strand swaps, the
program uses <br>
IDs of the form &quot;CHROM:POS_REF_ALT&quot;. The .sample
file is quite basic at the moment with columns for
population, group and sex expected to be edited by the user.
For <br>
example:</p>

<p style="margin-top: 1em">.haps <br>
----- <br>
0 1 0 0 1 0 <br>
0 1 0 0 0 1</p>

<p style="margin-top: 1em">.legend <br>
------- <br>
id position a0 a1 <br>
1:111485207_G_A 111485207 G A <br>
1:111494194_C_T 111494194 C T</p>

<p style="margin-top: 1em">.sample <br>
------- <br>
sample population group sex <br>
sample1 sample1 sample1 2 <br>
sample2 sample2 sample2 2</p>

<p style="margin-top: 1em">--haploid2diploid <br>
with -h option converts haploid genotypes to homozygous
diploid genotypes. For example, the program will print 0 0
instead of the default 0 -. This is useful for programs <br>
which do not handle haploid genotypes correctly.</p>

<p style="margin-top: 1em">--vcf-ids <br>
output VCF IDs instead of &quot;CHROM:POS_REF_ALT&quot;
IDs</p>

<p style="margin-top: 1em">TSV conversion: <br>
--tsv2vcf file <br>
convert from TSV (tab-separated values) format (such as
generated by 23andMe) to VCF. The input file fields can be
tab- or space- delimited</p>

<p style="margin-top: 1em">-c, --columns list <br>
comma-separated list of fields in the input file. In the
current version, the fields CHROM, POS, ID, and AA are
expected and can appear in arbitrary order, columns which
<br>
should be ignored in the input file can be indicated by
&quot;-&quot;. The AA field lists alleles on the forward
reference strand, for example &quot;CC&quot; or
&quot;CT&quot; for diploid genotypes or <br>
&quot;C&quot; for haploid genotypes (sex chromosomes).
Insertions and deletions are not supported yet, missing data
can be indicated with &quot;--&quot;.</p>

<p style="margin-top: 1em">-f, --fasta-ref file <br>
reference sequence in fasta format. Must be indexed with
samtools faidx</p>

<p style="margin-top: 1em">-s, --samples LIST <br>
list of sample names. See Common Options</p>

<p style="margin-top: 1em">-S, --samples-file FILE <br>
file of sample names. See Common Options</p>

<p style="margin-top: 1em">Example:</p>

<p style="margin-top: 1em"># Convert 23andme results into
VCF <br>
bcftools convert -c ID,CHROM,POS,AA -s SampleName -f
23andme-ref.fa --tsv2vcf 23andme.txt -Oz -o out.vcf.gz</p>

<p style="margin-top: 1em">bcftools filter [OPTIONS] FILE
<br>
Apply fixed-threshold filters.</p>

<p style="margin-top: 1em">-e, --exclude EXPRESSION <br>
exclude sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-g, --SnpGap INT <br>
filter SNPs within INT base pairs of an indel. The following
example demonstrates the logic of --SnpGap 3 applied on a
deletion and an insertion:</p>

<p style="margin-top: 1em">The SNPs at positions 1 and 7
are filtered, positions 0 and 8 are not: <br>
0123456789 <br>
ref .G.GT..G.. <br>
del .A.G-..A.. <br>
Here the positions 1 and 6 are filtered, 0 and 7 are not:
<br>
0123-456789 <br>
ref .G.G-..G.. <br>
ins .A.GT..A..</p>

<p style="margin-top: 1em">-G, --IndelGap INT <br>
filter clusters of indels separated by INT or fewer base
pairs allowing only one to pass. The following example
demonstrates the logic of --IndelGap 2 applied on a deletion
<br>
and an insertion:</p>

<p style="margin-top: 1em">The second indel is filtered:
<br>
012345678901 <br>
ref .GT.GT..GT.. <br>
del .G-.G-..G-.. <br>
And similarly here, the second is filtered: <br>
01 23 456 78 <br>
ref .A-.A-..A-.. <br>
ins .AT.AT..AT..</p>

<p style="margin-top: 1em">-i, --include EXPRESSION <br>
include only sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-m, --mode [+x] <br>
define behaviour at sites with existing FILTER annotations.
The default mode replaces existing filters of failed sites
with a new FILTER string while leaving sites which pass <br>
untouched when non-empty and setting to &quot;PASS&quot;
when the FILTER string is absent. The &quot;+&quot; mode
appends new FILTER strings of failed sites instead of
replacing them. The &quot;x&quot; <br>
mode resets filters of sites which pass to &quot;PASS&quot;.
Modes &quot;+&quot; and &quot;x&quot; can both be set.</p>

<p style="margin-top: 1em">--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --soft-filter STRING|+ <br>
annotate FILTER column with STRING or, with +, a unique
filter name generated by the program
(&quot;Filter%d&quot;).</p>

<p style="margin-top: 1em">-S, --set-GTs .|0 <br>
set genotypes of failed samples to missing value (.) or
reference allele (0)</p>

<p style="margin-top: 1em">-t, --targets
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file file <br>
see Common Options</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">bcftools gtcheck [OPTIONS] [-g
genotypes.vcf.gz] query.vcf.gz <br>
Checks sample identity or, without -g, multi-sample
cross-check is performed.</p>

<p style="margin-top: 1em">-a, --all-sites <br>
output for all sites</p>

<p style="margin-top: 1em">-g, --genotypes genotypes.vcf.gz
<br>
reference genotypes to compare against</p>

<p style="margin-top: 1em">-G, --GTs-only INT <br>
use genotypes (GT) instead of genotype likelihoods (PL).
When set to 1, reported discordance is the number of
non-matching GTs, otherwise the number INT is interpreted as
<br>
phred-scaled likelihood of unobserved genotypes.</p>

<p style="margin-top: 1em">-H, --homs-only <br>
consider only genotypes which are homozygous in both
genotypes and query VCF. This may be useful with low
coverage data.</p>

<p style="margin-top: 1em">-p, --plot PREFIX <br>
produce plots</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --query-sample STRING <br>
query sample in query.vcf.gz. By default, the first sample
is checked.</p>

<p style="margin-top: 1em">-S, --target-sample STRING <br>
target sample in the -g file, used only for plotting, not
for analysis</p>

<p style="margin-top: 1em">-t, --targets file <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file file <br>
see Common Options</p>

<p style="margin-top: 1em">Output files format: <br>
CN, Discordance <br>
Pairwise discordance for all sample pairs is calculated
as</p>

<p style="margin-top: 1em">_s { min_G { PL_a(G) + PL_b(G) }
},</p>

<p style="margin-top: 1em">where the sum runs over all
sites s and G is the the most likely genotype shared by both
samples a and b. When PL field is not present, a constant
value 99 is used for <br>
the unseen genotypes. With -G, the value 1 can be used
instead; the discordance value then gives exactly the number
of differing genotypes.</p>

<p style="margin-top: 1em">SM, Average Discordance <br>
Average discordance between sample a and all other
samples.</p>

<p style="margin-top: 1em">SM, Average Depth <br>
Average depth at evaluated sites, or 1 if FORMAT/DP field is
not present.</p>

<p style="margin-top: 1em">SM, Average Number of sites <br>
The average number of sites used to calculate the
discordance. In other words, the average number of
non-missing PLs/genotypes seen both samples.</p>

<p style="margin-top: 1em">bcftools index [OPTIONS]
&lt;in.bcf&gt;|&lt;in.vcf.gz&gt; <br>
Creates index for bgzip compressed VCF/BCF files for random
access. CSI (coordinate-sorted index) is created by default.
The CSI format supports indexing of chromosomes up to <br>
length 2^31. TBI (tabix index) index files, which support
chromosome lengths up to 2^29, can be created by using the
-t/--tbi option or using the tabix program packaged with
<br>
htslib. When loading an index file, bcftools will try the
CSI first and then the TBI.</p>

<p style="margin-top: 1em">Indexing options: <br>
-c, --csi <br>
generate CSI-format index for VCF/BCF files [default]</p>

<p style="margin-top: 1em">-f, --force <br>
overwrite index if it already exists</p>

<p style="margin-top: 1em">-m, --min-shift INT <br>
set minimal interval size for CSI indices to 2^INT; default:
14</p>

<p style="margin-top: 1em">-t, --tbi <br>
generate TBI-format index for VCF files</p>

<p style="margin-top: 1em">Stats options: <br>
-n, --nrecords <br>
print the number of records based on the CSI or TBI index
files</p>

<p style="margin-top: 1em">-s, --stats <br>
Print per contig stats based on the CSI or TBI index files.
Output format is three tab-delimited columns listing the
contig name, contig length (. if unknown) and number <br>
of records for the contig. Contigs with zero records are not
printed.</p>

<p style="margin-top: 1em">bcftools isec [OPTIONS] A.vcf.gz
B.vcf.gz [...] <br>
Creates intersections, unions and complements of VCF files.
Depending on the options, the program can output records
from one (or more) files which have (or do not have) <br>
corresponding records with the same position in the other
files.</p>

<p style="margin-top: 1em">-c, --collapse
snps|indels|both|all|some|none <br>
see Common Options</p>

<p style="margin-top: 1em">-C, --complement <br>
output positions present only in the first file but missing
in the others</p>

<p style="margin-top: 1em">-e, --exclude -|EXPRESSION <br>
exclude sites for which EXPRESSION is true. If -e (or -i)
appears only once, the same filtering expression will be
applied to all input files. Otherwise, -e or -i must be <br>
given for each input file. To indicate that no filtering
should be performed on a file, use &quot;-&quot; in place of
EXPRESSION, as shown in the example below. For valid
expressions <br>
see EXPRESSIONS.</p>

<p style="margin-top: 1em">-f, --apply-filters LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-i, --include EXPRESSION <br>
include only sites for which EXPRESSION is true. See
discussion of -e, --exclude above.</p>

<p style="margin-top: 1em">-n, --nfiles [+-=]INT|~BITMAP
<br>
output positions present in this many (=), this many or more
(+), this many or fewer (-), or the exact same (~) files</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options. When several files are being output,
their names are controlled via -p instead.</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">-p, --prefix DIR <br>
if given, subset each of the input files accordingly. See
also -w.</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-t, --targets
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-w, --write LIST <br>
list of input files to output given as 1-based indices. With
-p and no -w, all files are written.</p>

<p style="margin-top: 1em">Examples: <br>
Create intersection and complements of two sets saving the
output in dir/*</p>

<p style="margin-top: 1em">bcftools isec -p dir A.vcf.gz
B.vcf.gz</p>

<p style="margin-top: 1em">Filter sites in A and B (but not
in C) and create intersection</p>

<p style="margin-top: 1em">bcftools isec
-e&rsquo;MAF&lt;0.01&rsquo; -i&rsquo;dbSNP=1&rsquo; -e-
A.vcf.gz B.vcf.gz C.vcf.gz -p dir</p>

<p style="margin-top: 1em">Extract and write records from A
shared by both A and B using exact allele match</p>

<p style="margin-top: 1em">bcftools isec -p dir -n=2 -w1
A.vcf.gz B.vcf.gz</p>

<p style="margin-top: 1em">Extract records private to A or
B comparing by position only</p>

<p style="margin-top: 1em">bcftools isec -p dir -n-1 -c all
A.vcf.gz B.vcf.gz</p>

<p style="margin-top: 1em">Print a list of records which
are present in A and B but not in C and D</p>

<p style="margin-top: 1em">bcftools isec -n~1100 -c all
A.vcf.gz B.vcf.gz C.vcf.gz D.vcf.gz</p>

<p style="margin-top: 1em">bcftools merge [OPTIONS]
A.vcf.gz B.vcf.gz [...] <br>
Merge multiple VCF/BCF files from non-overlapping sample
sets to create one multi-sample file. For example, when
merging file A.vcf.gz containing samples S1, S2 and S3 and
file <br>
B.vcf.gz containing samples S3 and S4, the output file will
contain four samples named S1, S2, S3, 2:S3 and S4.</p>

<p style="margin-top: 1em">Note that it is responsibility
of the user to ensure that the sample names are unique
across all files. If they are not, the program will exit
with an error unless the option <br>
--force-samples is given. The sample names can be also given
explicitly using the --print-header and --use-header
options.</p>

<p style="margin-top: 1em">Note that only records from
different files can be merged, never from the same file. For
&quot;vertical&quot; merge take a look at bcftools norm
instead.</p>

<p style="margin-top: 1em">--force-samples <br>
if the merged files contain duplicate samples names, proceed
anyway. Duplicate sample names will be resolved by
prepending index of the file as it appeared on the command
<br>
line to the conflicting sample name (see 2:S3 in the above
example).</p>

<p style="margin-top: 1em">--print-header <br>
print only merged header and exit</p>

<p style="margin-top: 1em">--use-header FILE <br>
use the VCF header in the provided text FILE</p>

<p style="margin-top: 1em">-f, --apply-filters LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-i, --info-rules
-|TAG:METHOD[,...] <br>
Rules for merging INFO fields (scalars or vectors) or - to
disable the default rules. METHOD is one of sum, avg, min,
max, join. Default is DP:sum,DP4:sum if these fields <br>
exist in the input files. Fields with no specified rule will
take the value from the first input file. The merged QUAL
value is currently set to the maximum. This behaviour <br>
is not user controllable at the moment.</p>

<p style="margin-top: 1em">-l, --file-list FILE <br>
read file names from FILE</p>

<p style="margin-top: 1em">-m, --merge
snps|indels|both|all|none|id <br>
The option controls what types of multiallelic records can
be created:</p>

<p style="margin-top: 1em">-m none .. no new multiallelics,
output multiple records instead <br>
-m snps .. allow multiallelic SNP records <br>
-m indels .. allow multiallelic indel records <br>
-m both .. both SNP and indel records can be multiallelic
<br>
-m all .. SNP records can be merged with indel records <br>
-m id .. merge by ID</p>

<p style="margin-top: 1em">--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">bcftools norm [OPTIONS]
file.vcf.gz <br>
Left-align and normalize indels, check if REF alleles match
the reference, split multiallelic sites into multiple rows;
recover multiallelics from multiple rows. Left-alignment
<br>
and normalization will only be applied if the --fasta-ref
option is supplied.</p>

<p style="margin-top: 1em">-c, --check-ref e|w|x|s <br>
what to do when incorrect or missing REF allele is
encountered: exit (e), warn (w), exclude (x), or set/fix (s)
bad sites. The w option can be combined with x and s. Note
<br>
that s can swap alleles and will update genotypes (GT) and
AC counts, but will not attempt to fix PL or other
fields.</p>

<p style="margin-top: 1em">-d, --rm-dup
snps|indels|both|all|none <br>
If a record is present in multiple files, output only the
first instance, see --collapse in Common Options. Requires
-a, --allow-overlaps.</p>

<p style="margin-top: 1em">-D, --remove-duplicates <br>
If a record is present in multiple files, output only the
first instance. Alias for -d none. Requires -a,
--allow-overlaps.</p>

<p style="margin-top: 1em">-f, --fasta-ref FILE <br>
reference sequence. Supplying this option will turn on
left-alignment and normalization, however, see also the
--do-not-normalize option below.</p>

<p style="margin-top: 1em">-m, --multiallelics
&acirc;|+&gt;[snps|indels|both|any] <br>
split multiallelic sites into biallelic records (-) or join
biallelic sites into multiallelic records (+). An optional
type string can follow which controls variant types <br>
which should be split or merged together: If only SNP
records should be split or merged, specify snps; if both
SNPs and indels should be merged separately into two
records, <br>
specify both; if SNPs and indels should be merged into a
single record, specify any.</p>

<p style="margin-top: 1em">--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-N, --do-not-normalize <br>
the -c s option can be used to fix or set the REF allele
from the reference -f. The -N option will not turn on indel
normalisation as the -f option normally implies</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --strict-filter <br>
when merging (-m+), merged site is PASS only if all sites
being merged PASS</p>

<p style="margin-top: 1em">-t, --targets LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">-w, --site-win INT <br>
maximum distance between two records to consider when
locally sorting variants which changed position during the
realignment</p>

<p style="margin-top: 1em">bcftools [plugin NAME|+NAME]
[OPTIONS] FILE &acirc; [PLUGIN OPTIONS] <br>
VCF input options: <br>
-e, --exclude EXPRESSION <br>
exclude sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-i, --include EXPRESSION <br>
include only sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-t, --targets
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file file <br>
see Common Options</p>

<p style="margin-top: 1em">VCF output options: <br>
--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">Plugin options: <br>
-h, --help <br>
list plugin&acirc;s options</p>

<p style="margin-top: 1em">-l, --list-plugins <br>
List all available plugins.</p>

<p style="margin-top: 1em">By default, appropriate system
directories are searched for installed plugins. You can
override this by setting the BCFTOOLS_PLUGINS environment
variable to a <br>
colon-separated list of directories to search. If
BCFTOOLS_PLUGINS begins with a colon, ends with a colon, or
contains adjacent colons, the system directories are also
<br>
searched at that position in the list of directories.</p>

<p style="margin-top: 1em">If htslib is not installed
systemwide, set the environment variable LD_LIBRARY_PATH
(linux) or DYLD_LIBRARY_PATH (Mac OS X) to include the
directory where libhts.so.1 is <br>
located.</p>

<p style="margin-top: 1em">-v, --verbose <br>
print debugging information to debug plugin failure</p>

<p style="margin-top: 1em">-V, --version <br>
print version string and exit</p>

<p style="margin-top: 1em">List of plugins coming with the
distribution: <br>
counts <br>
a minimal plugin which counts number of SNPs, Indels, and
total number of sites.</p>

<p style="margin-top: 1em">dosage <br>
print genotype dosage. By default the plugin searches for
PL, GL and GT, in that order.</p>

<p style="margin-top: 1em">fill-AN-AC <br>
fill INFO fields AN and AC.</p>

<p style="margin-top: 1em">fix-ploidy <br>
sets correct ploidy</p>

<p style="margin-top: 1em">frameshifts <br>
annotate frameshift indels</p>

<p style="margin-top: 1em">missing2ref <br>
sets missing genotypes (&quot;./.&quot;) to ref allele
(&quot;0/0&quot; or &quot;0|0&quot;)</p>

<p style="margin-top: 1em">tag2tag <br>
Convert between similar tags, such as GL and GP.</p>

<p style="margin-top: 1em">vcf2sex <br>
determine sample sex by checking genotypes in haploid
regions</p>

<p style="margin-top: 1em">Examples: <br>
# List options common to all plugins <br>
bcftools plugin</p>

<p style="margin-top: 1em"># List available plugins <br>
bcftools plugin -l</p>

<p style="margin-top: 1em"># Run a plugin <br>
bcftools plugin counts in.vcf</p>

<p style="margin-top: 1em"># Run a plugin using the
abbreviated &quot;+&quot; notation <br>
bcftools +counts in.vcf</p>

<p style="margin-top: 1em"># The input VCF can be streamed
just like in other commands <br>
cat in.vcf | bcftools +counts</p>

<p style="margin-top: 1em"># Print usage information of
plugin &quot;dosage&quot; <br>
bcftools +dosage -h</p>

<p style="margin-top: 1em"># Replace missing genotypes with
0/0 <br>
bcftools +missing2ref in.vcf</p>

<p style="margin-top: 1em"># Replace missing genotypes with
0|0 <br>
bcftools +missing2ref in.vcf -- -p</p>

<p style="margin-top: 1em">Plugins troubleshooting: <br>
Things to check if your plugin does not show up in the
bcftools plugin -l output:</p>

<p style="margin-top: 1em">&Acirc;&middot; Run with the -v
option for verbose output: bcftools plugin -lv</p>

<p style="margin-top: 1em">&Acirc;&middot; Does the
environment variable BCFTOOLS_PLUGINS include the correct
path?</p>

<p style="margin-top: 1em">&Acirc;&middot; Are all shared
libraries, namely libhts.so, accessible? Verify with</p>

<p style="margin-top: 1em">&Acirc;&middot; on Mac OS X:
otool -L your/plugin.so and set DYLD_LIBRARY_PATH if they
are not</p>

<p style="margin-top: 1em">&Acirc;&middot; on Linux: ldd
your/plugin.so and set LD_LIBRARY_PATH if they are not</p>

<p style="margin-top: 1em">&Acirc;&middot; If not installed
systemwide, set the environment variable LD_LIBRARY_PATH
(linux) or DYLD_LIBRARY_PATH (mac) to include directory
where libhts.so is located.</p>

<p style="margin-top: 1em">Plugins API: <br>
// Short description used by &rsquo;bcftools plugin
-l&rsquo; <br>
const char *about(void);</p>

<p style="margin-top: 1em">// Longer description used by
&rsquo;bcftools +name -h&rsquo; <br>
const char *usage(void);</p>

<p style="margin-top: 1em">// Called once at startup,
allows initialization of local variables. <br>
// Return 1 to suppress normal VCF/BCF header output, -1 on
critical <br>
// errors, 0 otherwise. <br>
int init(int argc, char **argv, bcf_hdr_t *in_hdr, bcf_hdr_t
*out_hdr);</p>

<p style="margin-top: 1em">// Called for each VCF record,
return NULL to suppress the output <br>
bcf1_t *process(bcf1_t *rec);</p>

<p style="margin-top: 1em">// Called after all lines have
been processed to clean up <br>
void destroy(void);</p>

<p style="margin-top: 1em">bcftools polysomy [OPTIONS]
file.vcf.gz <br>
Detect number of chromosomal copies in VCFs annotates with
the Illumina&acirc;s B-allele frequency (BAF) values. Note
that this command is not compiled in by default, see the
section <br>
Optional Compilation with GSL in the INSTALL file for
help.</p>

<p style="margin-top: 1em">General options: <br>
-o, --output-dir path <br>
output directory</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --sample string <br>
sample name</p>

<p style="margin-top: 1em">-t, --targets LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-v, --verbose <br>
verbose debugging output which gives hints about the
thresholds and decisions made by the program. Note that the
exact output can change between versions.</p>

<p style="margin-top: 1em">Algorithm options: <br>
-b, --peak-size float <br>
the minimum peak size considered as a good match can be from
the interval [0,1] where larger is stricter</p>

<p style="margin-top: 1em">-c, --cn-penalty float <br>
a penalty for increasing copy number state. How this works:
multiple peaks are always a better fit than a single peak,
therefore the program prefers a single peak (normal <br>
copy number) unless the absolute deviation of the multiple
peaks fit is significantly smaller. Here the meaning of
&quot;significant&quot; is given by the float from the
interval <br>
[0,1] where larger is stricter.</p>

<p style="margin-top: 1em">-f, --fit-th float <br>
threshold for goodness of fit (normalized absolute
deviation), smaller is stricter</p>

<p style="margin-top: 1em">-i, --include-aa <br>
include also the AA peak in CN2 and CN3 evaluation. This
usually requires increasing -f.</p>

<p style="margin-top: 1em">-m, --min-fraction float <br>
minimum distinguishable fraction of aberrant cells. The
experience shows that trustworthy are estimates of 20% and
more.</p>

<p style="margin-top: 1em">-p, --peak-symmetry float <br>
a heuristics to filter failed fits where the expected peak
symmetry is violated. The float is from the interval [0,1]
and larger is stricter</p>

<p style="margin-top: 1em">bcftools query [OPTIONS]
file.vcf.gz [file.vcf.gz [...]] <br>
Extracts fields from VCF or BCF files and outputs them in
user-defined format.</p>

<p style="margin-top: 1em">-c, --collapse
snps|indels|both|all|some|none <br>
see Common Options</p>

<p style="margin-top: 1em">-e, --exclude EXPRESSION <br>
exclude sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-f, --format FORMAT <br>
learn by example, see below</p>

<p style="margin-top: 1em">-H, --print-header <br>
print header</p>

<p style="margin-top: 1em">-i, --include EXPRESSION <br>
include only sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-l, --list-samples <br>
list sample names and exit</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --samples LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-S, --samples-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-t, --targets
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-u, --allow-undef-tags <br>
do not throw an error if there are undefined tags in the
format string, print &quot;.&quot; instead</p>

<p style="margin-top: 1em">-v, --vcf-list FILE <br>
process multiple VCFs listed in the file</p>

<p style="margin-top: 1em">Format: <br>
%CHROM The CHROM column (similarly also other columns: POS,
ID, REF, ALT, QUAL, FILTER) <br>
%INFO/TAG Any tag in the INFO column <br>
%TYPE Variant type (REF, SNP, MNP, INDEL, OTHER) <br>
%MASK Indicates presence of the site in other files (with
multiple files) <br>
%TAG{INT} Curly brackets to subscript vectors (0-based) <br>
%FIRST_ALT Alias for %ALT{0} <br>
[] The brackets loop over all samples <br>
%GT Genotype (e.g. 0/1) <br>
%TGT Translated genotype (e.g. C/A) <br>
%IUPACGT Genotype translated to IUPAC ambiguity codes (e.g.
M instead of C/A) <br>
%LINE Prints the whole line <br>
%SAMPLE Sample name</p>

<p style="margin-top: 1em">Examples: <br>
bcftools query -f &rsquo;%CHROM %POS %REF %ALT{0}0
file.vcf.gz <br>
bcftools query -f &rsquo;%CHROM%POS%REF%ALT[%SAMPLE=%GT]0
file.vcf.gz</p>

<p style="margin-top: 1em">bcftools reheader [OPTIONS]
file.vcf.gz <br>
Modify header of VCF/BCF files, change sample names.</p>

<p style="margin-top: 1em">-h, --header FILE <br>
new VCF header</p>

<p style="margin-top: 1em">-o, --output FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --samples FILE <br>
new sample names, one name per line, in the same order as
they appear in the VCF file. Alternatively, only samples
which need to be renamed can be listed as &quot;old_name
<br>
new_name0 pairs separated by whitespaces, each on a separate
line. If a sample name contains spaces, the spaces can be
escaped using the backslash character, for example <br>
&quot;Not&nbsp;a&nbsp;good&nbsp;sample&nbsp;name&quot;.</p>

<p style="margin-top: 1em">bcftools roh [OPTIONS]
file.vcf.gz <br>
A program for detecting runs of homo/autozygosity. Only
bi-allelic sites are considered.</p>

<p style="margin-top: 1em">The HMM model: <br>
Notation: <br>
D = Data, AZ = autozygosity, HW = Hardy-Weinberg
(non-autozygosity), <br>
f = non-ref allele frequency</p>

<p style="margin-top: 1em">Emission probabilities: <br>
oAZ = P_i(D|AZ) = (1-f)*P(D|RR) + f*P(D|AA) <br>
oHW = P_i(D|HW) = (1-f)^2 * P(D|RR) + f^2 * P(D|AA) +
2*f*(1-f)*P(D|RA)</p>

<p style="margin-top: 1em">Transition probabilities: <br>
tAZ = P(AZ|HW) .. from HW to AZ, the -a parameter <br>
tHW = P(HW|AZ) .. from AZ to HW, the -H parameter</p>

<p style="margin-top: 1em">ci = P_i(C) .. probability of
cross-over at site i, from genetic map <br>
AZi = P_i(AZ) .. probability of site i being AZ/non-AZ,
scaled so that AZi+HWi = 1 <br>
HWi = P_i(HW)</p>

<p style="margin-top: 1em">P_{i+1}(AZ) = oAZ * max[(1 - tAZ
* ci) * AZ{i-1} , tAZ * ci * (1-AZ{i-1})] <br>
P_{i+1}(HW) = oHW * max[(1 - tHW * ci) * (1-AZ{i-1}) , tHW *
ci * AZ{i-1}]</p>

<p style="margin-top: 1em">General Options: <br>
--AF-dflt FLOAT <br>
in case allele frequency is not known, use the FLOAT. By
default, sites where allele frequency cannot be determined,
or is 0, are skipped.</p>

<p style="margin-top: 1em">--AF-tag TAG <br>
use the specified INFO tag TAG as an allele frequency
estimate instead of the default AC and AN tags. Sites which
do not have TAG will be skipped.</p>

<p style="margin-top: 1em">--AF-file FILE <br>
Read allele frequencies from a tab-delimited file containing
the columns: CHROMPOSREF,ALTAF. The file can be compressed
with bgzip and indexed with tabix -s1 -b2 <br>
-e2. Sites which are not present in the FILE or have
different reference or alternate allele will be skipped.
Note that such a file can be easily created from a VCF <br>
using:</p>

<p style="margin-top: 1em">bcftools query
-f&rsquo;%CHROM%POS%REF,%ALT%INFO/TAG0 file.vcf | bgzip -c
&gt; freqs.tab.gz</p>

<p style="margin-top: 1em">-e, --estimate-AF FILE <br>
recalculate INFO/AC and INFO/AN on the fly, using either all
samples (&quot;-&quot;) or samples listed in FILE. By
default, allele frequency is estimated from AC and AN counts
<br>
which are already present in the INFO field.</p>

<p style="margin-top: 1em">-G, --GTs-only FLOAT <br>
use genotypes (FORMAT/GT fields) ignoring genotype
likelihoods (FORMAT/PL), setting PL of unseen genotypes to
FLOAT. Safe value to use is 30 to account for GT errors.</p>

<p style="margin-top: 1em">-I, --skip-indels <br>
skip indels as their genotypes are usually enriched for
errors</p>

<p style="margin-top: 1em">-m, --genetic-map FILE <br>
genetic map in the format required also by IMPUTE2. Only the
first and third column are used (position and
Genetic_Map(cM)). The FILE can chromosome name.</p>

<p style="margin-top: 1em">-M, --rec-rate FLOAT <br>
constant recombination rate per bp</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --sample name <br>
the name of sample to analyze</p>

<p style="margin-top: 1em">-t, --targets
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file file <br>
see Common Options</p>

<p style="margin-top: 1em">HMM Options: <br>
-a, --hw-to-az FLOAT <br>
P(AZ|HW) transition probability from AZ (autozygous) to HW
(Hardy-Weinberg) state</p>

<p style="margin-top: 1em">-H, --az-to-hw FLOAT <br>
P(HW|AZ) transition probability from HW to AZ state</p>

<p style="margin-top: 1em">-V, --viterbi-training <br>
perform Viterbi training to estimate transition
probabilities</p>

<p style="margin-top: 1em">bcftools stats [OPTIONS]
A.vcf.gz [B.vcf.gz] <br>
Parses VCF or BCF and produces text file stats which is
suitable for machine processing and can be plotted using
plot-vcfstats. When two files are given, the program
generates <br>
separate stats for intersection and the complements. By
default only sites are compared, -s/-S must given to include
also sample columns.</p>

<p style="margin-top: 1em">-1, --1st-allele-only <br>
consider only the 1st alternate allele at multiallelic
sites</p>

<p style="margin-top: 1em">-c, --collapse
snps|indels|both|all|some|none <br>
see Common Options</p>

<p style="margin-top: 1em">-d, --depth INT,INT,INT <br>
ranges of depth distribution: min, max, and size of the
bin</p>

<p style="margin-top: 1em">--debug <br>
produce verbose per-site and per-sample output</p>

<p style="margin-top: 1em">-e, --exclude EXPRESSION <br>
exclude sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-E, --exons file.gz <br>
tab-delimited file with exons for indel frameshifts
statistics. The columns of the file are CHR, FROM, TO, with
1-based, inclusive, positions. The file is BGZF-compressed
and <br>
indexed with tabix</p>

<p style="margin-top: 1em">tabix -s1 -b2 -e3 file.gz</p>

<p style="margin-top: 1em">-f, --apply-filters LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-F, --fasta-ref ref.fa <br>
faidx indexed reference sequence file to determine INDEL
context</p>

<p style="margin-top: 1em">-i, --include EXPRESSION <br>
include only sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-I, --split-by-ID <br>
collect stats separately for sites which have the ID column
set (&quot;known sites&quot;) or which do not have the ID
column set (&quot;novel sites&quot;).</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-s, --samples LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-S, --samples-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">-t, --targets
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-u, --user-tstv
&lt;TAG[:min:max:n]&gt; <br>
collect Ts/Tv stats for any tag using the given binning
[0:1:100]</p>

<p style="margin-top: 1em">-v, --verbose <br>
produce verbose per-site and per-sample output</p>

<p style="margin-top: 1em">bcftools view [OPTIONS]
file.vcf.gz [REGION [...]] <br>
View, subset and filter VCF or BCF files by position and
filtering expression. Convert between VCF and BCF. Former
bcftools subset.</p>

<p style="margin-top: 1em">Output options <br>
-G, --drop-genotypes <br>
drop individual genotype information (after subsetting if -s
option is set)</p>

<p style="margin-top: 1em">-h, --header-only <br>
output the VCF header only</p>

<p style="margin-top: 1em">-H, --no-header <br>
suppress the header in VCF output</p>

<p style="margin-top: 1em">-l, --compression-level [0-9]
<br>
compression level. 0 stands for uncompressed, 1 for best
speed and 9 for best compression.</p>

<p style="margin-top: 1em">--no-version <br>
see Common Options</p>

<p style="margin-top: 1em">-O, --output-type b|u|z|v <br>
see Common Options</p>

<p style="margin-top: 1em">-o, --output-file FILE: output
file name. If not present, the default is to print to
standard output (stdout).</p>

<p style="margin-top: 1em">-r, --regions
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-R, --regions-file file <br>
see Common Options</p>

<p style="margin-top: 1em">-t, --targets
chr|chr:pos|chr:from-to|chr:from-[,...] <br>
see Common Options</p>

<p style="margin-top: 1em">-T, --targets-file file <br>
see Common Options</p>

<p style="margin-top: 1em">--threads INT <br>
see Common Options</p>

<p style="margin-top: 1em">Subset options: <br>
-a, --trim-alt-alleles <br>
trim alternate alleles not seen in subset. Type A, G and R
INFO and FORMAT fields will also be trimmed</p>

<p style="margin-top: 1em">--force-samples <br>
only warn about unknown subset samples</p>

<p style="margin-top: 1em">-I, --no-update <br>
do not (re)calculate INFO fields for the subset (currently
INFO/AC and INFO/AN)</p>

<p style="margin-top: 1em">-s, --samples LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-S, --samples-file FILE <br>
see Common Options</p>

<p style="margin-top: 1em">Filter options: <br>
Note that filter options below dealing with counting the
number of alleles will, for speed, first check for the
values of AC and AN in the INFO column to avoid parsing all
<br>
the genotype (FORMAT/GT) fields in the VCF. This means that
a filter like --min-af 0.1 will be based &acirc;AC/AN&acirc;
where AC and AN come from either INFO/AC and INFO/AN if
available <br>
or FORMAT/GT if not. It will not filter on another field
like INFO/AF. The --include and --exclude filter expressions
should instead be used to explicitly filter based on <br>
fields in the INFO column, e.g. --exclude AF&lt;0.1.</p>

<p style="margin-top: 1em">-c, --min-ac
INT[:nref|:alt1|:minor|:major|:&rsquo;nonmajor&rsquo;] <br>
minimum allele count (INFO/AC) of sites to be printed.
Specifying the type of allele is optional and can be set to
non-reference (nref, the default), 1st alternate <br>
(alt1), the least frequent (minor), the most frequent
(major) or sum of all but the most frequent (nonmajor)
alleles.</p>

<p style="margin-top: 1em">-C, --max-ac
INT[:nref|:alt1|:minor|:&rsquo;major&rsquo;|:&rsquo;nonmajor&rsquo;]
<br>
maximum allele count (INFO/AC) of sites to be printed.
Specifying the type of allele is optional and can be set to
non-reference (nref, the default), 1st alternate <br>
(alt1), the least frequent (minor), the most frequent
(major) or sum of all but the most frequent (nonmajor)
alleles.</p>

<p style="margin-top: 1em">-e, --exclude EXPRESSION <br>
exclude sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-f, --apply-filters LIST <br>
see Common Options</p>

<p style="margin-top: 1em">-g, --genotype [^][hom|het|miss]
<br>
include only sites with one or more homozygous (hom),
heterozygous (het) or missing (miss) genotypes. When
prefixed with ^, the logic is reversed; thus ^het excludes
<br>
sites with heterozygous genotypes.</p>

<p style="margin-top: 1em">-i, --include EXPRESSION <br>
include sites for which EXPRESSION is true. For valid
expressions see EXPRESSIONS.</p>

<p style="margin-top: 1em">-k, --known <br>
print known sites only (ID column is not &quot;.&quot;)</p>

<p style="margin-top: 1em">-m, --min-alleles INT <br>
print sites with at least INT alleles listed in REF and ALT
columns</p>

<p style="margin-top: 1em">-M, --max-alleles INT <br>
print sites with at most INT alleles listed in REF and ALT
columns. Use -m2 -M2 -v snps to only view biallelic
SNPs.</p>

<p style="margin-top: 1em">-n, --novel <br>
print novel sites only (ID column is &quot;.&quot;)</p>

<p style="margin-top: 1em">-p, --phased <br>
print sites where all samples are phased. Haploid genotypes
are considered phased. Missing genotypes considered unphased
unless the phased bit is set.</p>

<p style="margin-top: 1em">-P, --exclude-phased <br>
exclude sites where all samples are phased</p>

<p style="margin-top: 1em">-q, --min-af
FLOAT[:nref|:alt1|:minor|:major|:nonmajor] <br>
minimum allele frequency (INFO/AC / INFO/AN) of sites to be
printed. Specifying the type of allele is optional and can
be set to non-reference (nref, the default), 1st <br>
alternate (alt1), the least frequent (minor), the most
frequent (major) or sum of all but the most frequent
(nonmajor) alleles.</p>

<p style="margin-top: 1em">-Q, --max-af
FLOAT[:nref|:alt1|:minor|:major|:nonmajor] <br>
maximum allele frequency (INFO/AC / INFO/AN) of sites to be
printed. Specifying the type of allele is optional and can
be set to non-reference (nref, the default), 1st <br>
alternate (alt1), the least frequent (minor), the most
frequent (major) or sum of all but the most frequent
(nonmajor) alleles.</p>

<p style="margin-top: 1em">-u, --uncalled <br>
print sites without a called genotype</p>

<p style="margin-top: 1em">-U, --exclude-uncalled <br>
exclude sites without a called genotype</p>

<p style="margin-top: 1em">-v, --types
snps|indels|mnps|other <br>
comma-separated list of variant types to select. Site is
selected if any of the ALT alleles is of the type requested.
Types are determined by comparing the REF and ALT <br>
alleles in the VCF record not INFO tags like INFO/INDEL or
INFO/VT. Use --include to select based on INFO tags.</p>

<p style="margin-top: 1em">-V, --exclude-types
snps|indels|mnps|other <br>
comma-separated list of variant types to exclude. Site is
excluded if any of the ALT alleles is of the type requested.
Types are determined by comparing the REF and ALT <br>
alleles in the VCF record not INFO tags like INFO/INDEL or
INFO/VT. Use --exclude to exclude based on INFO tags.</p>

<p style="margin-top: 1em">-x, --private <br>
print sites where only the subset samples carry an
non-reference allele. Requires --samples or
--samples-file.</p>

<p style="margin-top: 1em">-X, --exclude-private <br>
exclude sites where only the subset samples carry an
non-reference allele</p>

<p style="margin-top: 1em">bcftools help [COMMAND] |
bcftools --help [COMMAND] <br>
Display a brief usage message listing the bcftools commands
available. If the name of a command is also given, e.g.,
bcftools help view, the detailed usage message for that <br>
particular command is displayed.</p>

<p style="margin-top: 1em">bcftools [--version|-v] <br>
Display the version numbers and copyright information for
bcftools and the important libraries used by bcftools.</p>

<p style="margin-top: 1em">bcftools [--version-only] <br>
Display the full bcftools version number in a
machine-readable format.</p>

<p style="margin-top: 1em">EXPRESSIONS <br>
These filtering expressions are accepted by annotate,
filter, query and view commands.</p>

<p style="margin-top: 1em">Valid expressions may
contain:</p>

<p style="margin-top: 1em">&Acirc;&middot; numerical
constants, string constants, file names</p>

<p style="margin-top: 1em">1, 1.0, 1e-4 <br>
&quot;String&quot; <br>
@file_name</p>

<p style="margin-top: 1em">&Acirc;&middot; arithmetic
operators</p>

<p style="margin-top: 1em">+,*,-,/</p>

<p style="margin-top: 1em">&Acirc;&middot; comparison
operators</p>

<p style="margin-top: 1em">== (same as =), &gt;, &gt;=,
&lt;=, &lt;, !=</p>

<p style="margin-top: 1em">&Acirc;&middot; regex operators
&quot;~&quot; and its negation &quot;!~&quot;</p>

<p style="margin-top: 1em">INFO/HAYSTACK ~
&quot;needle&quot;</p>

<p style="margin-top: 1em">&Acirc;&middot; parentheses</p>

<p style="margin-top: 1em">(, )</p>

<p style="margin-top: 1em">&Acirc;&middot; logical
operators</p>

<p style="margin-top: 1em">&amp;&amp; (same as &amp;), ||,
|</p>

<p style="margin-top: 1em">&Acirc;&middot; INFO tags,
FORMAT tags, column names</p>

<p style="margin-top: 1em">INFO/DP or DP <br>
FORMAT/DV, FMT/DV, or DV <br>
FILTER, QUAL, ID, POS, REF, ALT[0]</p>

<p style="margin-top: 1em">&Acirc;&middot; 1 (or 0) to test
the presence (or absence) of a flag</p>

<p style="margin-top: 1em">FlagA=1 &amp;&amp; FlagB=0</p>

<p style="margin-top: 1em">&Acirc;&middot; &quot;.&quot; to
test missing values</p>

<p style="margin-top: 1em">DP=&quot;.&quot;,
DP!=&quot;.&quot;, ALT=&quot;.&quot;</p>

<p style="margin-top: 1em">&Acirc;&middot; missing
genotypes can be matched regardless of phase and ploidy
(&quot;.|.&quot;, &quot;./.&quot;, &quot;.&quot;) using this
expression</p>

<p style="margin-top: 1em">GT=&quot;.&quot;</p>

<p style="margin-top: 1em">&Acirc;&middot; TYPE for variant
type in REF,ALT columns (indel,snp,mnp,ref,other)</p>

<p style="margin-top: 1em">TYPE=&quot;indel&quot; |
TYPE=&quot;snp&quot;</p>

<p style="margin-top: 1em">&Acirc;&middot; array
subscripts, &quot;*&quot; for any field</p>

<p style="margin-top: 1em">(DP4[0]+DP4[1])/(DP4[2]+DP4[3])
&gt; 0.3 <br>
DP4[*] == 0 <br>
CSQ[*] ~ &quot;missense_variant.*deleterious&quot;</p>

<p style="margin-top: 1em">&Acirc;&middot; function on
FORMAT tags (over samples) and INFO tags (over vector
fields)</p>

<p style="margin-top: 1em">MAX, MIN, AVG, SUM, STRLEN,
ABS</p>

<p style="margin-top: 1em">&Acirc;&middot; variables
calculated on the fly if not present: number of alternate
alleles; number of samples; count of alternate alleles;
minor allele count (similar to AC but is always <br>
smaller than 0.5); frequency of alternate alleles
(AF=AC/AN); frequency of minor alleles (MAF=MAC/AN); number
of alleles in called genotypes</p>

<p style="margin-top: 1em">N_ALT, N_SAMPLES, AC, MAC, AF,
MAF, AN</p>

<p style="margin-top: 1em">Notes:</p>

<p style="margin-top: 1em">&Acirc;&middot; String
comparisons and regular expressions are case-insensitive</p>

<p style="margin-top: 1em">&Acirc;&middot; If the subscript
&quot;*&quot; is used in regular expression search, the
whole field is treated as one string. For example, the regex
STR[*]~&quot;B,C&quot; will be true for the string vector
<br>
INFO/STR=AB,CD.</p>

<p style="margin-top: 1em">&Acirc;&middot; Variables and
function names are case-insensitive, but not tag names. For
example, &quot;qual&quot; can be used instead of
&quot;QUAL&quot;, &quot;strlen()&quot; instead of
&quot;STRLEN()&quot; , but not &quot;dp&quot; <br>
instead of &quot;DP&quot;.</p>

<p style="margin-top: 1em">Examples:</p>

<p style="margin-top: 1em">MIN(DV)&gt;5</p>

<p style="margin-top: 1em">MIN(DV/DP)&gt;0.3</p>

<p style="margin-top: 1em">MIN(DP)&gt;10 &amp;
MIN(DV)&gt;3</p>

<p style="margin-top: 1em">FMT/DP&gt;10 &amp; FMT/GQ&gt;10
.. both conditions must be satisfied within one sample</p>

<p style="margin-top: 1em">FMT/DP&gt;10 &amp;&amp;
FMT/GQ&gt;10 .. the conditions can be satisfied in different
samples</p>

<p style="margin-top: 1em">QUAL&gt;10 | FMT/GQ&gt;10 ..
selects only GQ&gt;10 samples</p>

<p style="margin-top: 1em">QUAL&gt;10 || FMT/GQ&gt;10 ..
selects all samples at QUAL&gt;10 sites</p>

<p style="margin-top: 1em">TYPE=&quot;snp&quot; &amp;&amp;
QUAL&gt;=10 &amp;&amp; (DP4[2]+DP4[3] &gt; 2)</p>

<p style="margin-top: 1em">MIN(DP)&gt;35 &amp;&amp;
AVG(GQ)&gt;50</p>

<p style="margin-top: 1em">ID=@file .. selects lines with
ID present in the file</p>

<p style="margin-top: 1em">ID!=@~/file .. skip lines with
ID present in the ~/file</p>

<p style="margin-top: 1em">MAF[0]&lt;0.05 .. select rare
variants at 5% cutoff</p>

<p style="margin-top: 1em">POS&gt;=100 .. restrict your
range query, e.g. 20:100-200 to strictly sites with POS in
that range.</p>

<p style="margin-top: 1em">Shell expansion:</p>

<p style="margin-top: 1em">Note that expressions must often
be quoted because some characters have special meaning in
the shell. An example of expression enclosed in single
quotes which cause that the <br>
whole expression is passed to the program as intended:</p>

<p style="margin-top: 1em">bcftools view -i
&rsquo;%ID!=&quot;.&quot; &amp; MAF[0]&lt;0.01&rsquo;</p>

<p style="margin-top: 1em">Please refer to the
documentation of your shell for details.</p>

<p style="margin-top: 1em">SCRIPTS AND OPTIONS <br>
plot-vcfstats [OPTIONS] file.vchk [...] <br>
Script for processing output of bcftools stats. It can merge
results from multiple outputs (useful when running the stats
for each chromosome separately), plots graphs and <br>
creates a PDF presentation.</p>

<p style="margin-top: 1em">-m, --merge <br>
Merge vcfstats files to STDOUT, skip plotting.</p>

<p style="margin-top: 1em">-p, --prefix PATH <br>
The output files prefix, add a slash to create new
directory.</p>

<p style="margin-top: 1em">-P, --no-PDF <br>
Skip the PDF creation step.</p>

<p style="margin-top: 1em">-r, --rasterize <br>
Rasterize PDF images for faster rendering.</p>

<p style="margin-top: 1em">-s, --sample-names <br>
Use sample names for xticks rather than numeric IDs.</p>

<p style="margin-top: 1em">-t, --title STRING <br>
Identify files by these titles in plots. The option can be
given multiple times, for each ID in the bcftools stats
output. If not present, the script will use abbreviated <br>
source file names for the titles.</p>

<p style="margin-top: 1em">-T, --main-title STRING <br>
Main title for the PDF.</p>

<p style="margin-top: 1em">PERFORMANCE <br>
HTSlib was designed with BCF format in mind. When parsing
VCF files, all records are internally converted into BCF
representation. Simple operations, like removing a single
<br>
column from a VCF file, can be therefore done much faster
with standard UNIX commands, such as awk or cut. Therefore
it is recommended to use BCF as input/output format whenever
<br>
possible to avoid large overhead of the VCF &acirc; BCF
&acirc; VCF conversion.</p>

<p style="margin-top: 1em">BUGS <br>
Please report any bugs you encounter on the github website:
http://github.com/samtools/bcftools</p>

<p style="margin-top: 1em">AUTHORS <br>
Heng Li from the Sanger Institute wrote the original C
version of htslib, samtools and bcftools. Bob Handsaker from
the Broad Institute implemented the BGZF library. Petr <br>
Danecek, Shane McCarthy and John Marshall are maintaining
and further developing bcftools. Many other people
contributed to the program and to the file format
specifications, <br>
both directly and indirectly by providing patches, testing
and reporting bugs. We thank them all.</p>

<p style="margin-top: 1em">RESOURCES <br>
BCFtools GitHub website:
http://github.com/samtools/bcftools</p>

<p style="margin-top: 1em">Samtools GitHub website:
http://github.com/samtools/samtools</p>

<p style="margin-top: 1em">HTSlib GitHub website:
http://github.com/samtools/htslib</p>

<p style="margin-top: 1em">File format specifications:
http://samtools.github.io/hts-specs</p>

<p style="margin-top: 1em">BCFtools documentation:
http://samtools.github.io/bcftools</p>

<p style="margin-top: 1em">BCFtools wiki page:
https://github.com/samtools/bcftools/wiki</p>

<p style="margin-top: 1em">COPYING <br>
The MIT/Expat License or GPL License, see the LICENSE
document for details. Copyright (c) Genome Research Ltd.</p>

<p style="margin-top: 1em">2016-04-18 14:18 BST
BCFTOOLS(1)</p>
<hr>
</body>
</html>
