<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:54:34 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>Beagle(1) General Commands Manual Beagle(1)</p>

<p style="margin-top: 1em">NAME <br>
Beagle - Genotype calling, genotype phasing and imputation
of ungenotyped markers</p>

<p style="margin-top: 1em">SYNOPSIS <br>
java -Xmx[GB]g -jar /usr/share/beagle/beagle.jar
[options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Beagle performs genotype calling, genotype phasing,
imputation of ungenotyped markers, and identity-by-descent
segment detection. Genotypic imputation works on phased
haplotypes <br>
using a Li and Stephens haplotype frequency model. Beagle
also implements the Refined IBD algorithm for detecting
homozygosity-by-descent (HBD) and identity-by-descent (IBD)
<br>
segments.</p>

<p style="margin-top: 1em">OPTIONS <br>
Data input/output parameters <br>
gt=filename <br>
Optional <br>
Specifies a VCF file containing a GT (genotype) format field
for each marker. If a genotype contains the phased allele
separator, &quot;|&quot;, then Beagle will preserve the
phase <br>
of the genotype during the analysis. If you use the gt
argument, all genotypes in the output file will be phased
and non-missing.</p>

<p style="margin-top: 1em">gl=filename <br>
Optional <br>
Specifies a VCF file containing a GL or PL (genotype
likelihood) format field for each marker. Any data in the GT
format field will be ignored. If both GL and PL format <br>
fields are present for a marker, the GL format will be
used.</p>

<p style="margin-top: 1em">gtgl=filename <br>
Optional <br>
Specifies a VCF file containing a GT, GL or PL format field
for each marker. If a genotype is non-missing, Beagle will
ignore the genotype likelihood. If both GL and PL <br>
format fields are present for a marker, the GL field will be
used.</p>

<p style="margin-top: 1em">ref=filename <br>
Optional <br>
Specifies a VCF file containing phased reference genotypes.
See the impute parameter.</p>

<p style="margin-top: 1em">out=prefix <br>
Required <br>
Specifies the output filename prefix. The prefix may be an
absolute or relative filename, but it cannot be a directory
name.</p>

<p style="margin-top: 1em">excludesamples=filename <br>
Optional <br>
Specifies a file containing non-reference samples (one
sample per line) to be excluded from the analysis and output
files.</p>

<p style="margin-top: 1em">excludemarkers=filename <br>
Optional <br>
Specifies a file containing markers (one marker per line) to
be excluded from the analysis and the output files. An
excluded marker identifier can either be an identifier <br>
from the VCF record&acirc;s ID field or a genomic coordinate
in the format: CHROM:POS.</p>

<p style="margin-top: 1em">map=filename <br>
Optional <br>
Specifies a PLINK format genetic map on the cM scale. HapMap
GrCh36 and GrCh37 genetic maps in PLINK format are available
for download from the Beagle website. Use of a <br>
genetic map is recommended if you are imputing ungenotyped
markers. If no genetic map is specified, Beagle will assume
a constant recombination rate of 1 cM / Mb.</p>

<p style="margin-top: 1em">chrom=chrom:start-end <br>
Optional <br>
Specifies a chromosome or chromosome interval using a
chromosome identifier in the VCF file and the starting and
ending positions of the interval. The entire chromosome,
<br>
the beginning of the chromosome, and the end of a chromosome
can be specified by chrom=[chrom], chrom=[chrom:-end], and
chrom=[chrom:start-] respectively.</p>

<p style="margin-top: 1em">maxlr=number_&acirc;&yen;_1 <br>
Default = 5000 <br>
Specifies the maximum likelihood ratio at a genotype. If M
is the maximum of the likelihoods of each possible genotype,
any likelihood that is less than (M &acirc; maxlr) is <br>
set to 0.0 to improve computational efficiency.</p>

<p style="margin-top: 1em">General parameters <br>
nthreads=positive_integer <br>
Default: machine-dependent <br>
Specifies the number of threads of execution. If no nthreads
parameter is specified, the nthreads parameter will be set
equal to the number of CPU cores on the host <br>
machine.</p>

<p style="margin-top: 1em">lowmem=true/false <br>
Default = false <br>
Specifies whether a memory efficient algorithm should be
used. The memory efficient algorithm increases run-time by a
factor less than 2.0.</p>

<p style="margin-top: 1em">window=positive_integer <br>
Default = 50000 <br>
Specifies the number of markers to include in each sliding
window. The window parameter must be at least twice as large
as the overlap parameter. The window parameter <br>
controls the amount of memory used in the analysis. For
human data, it is recommended that the window parameter be
greater than or equal to the typical number of markers <br>
in 5 cM.</p>

<p style="margin-top: 1em">overlap=positive_integer <br>
Default = 3000 <br>
Specifies the number of markers of overlap between sliding
windows. For human data, it is recommended that the overlap
be set to the typical number of markers in 0.5 cM <br>
(when ibd=false) or 2.0 cM (when ibd=true).</p>

<p style="margin-top: 1em">seed=integer <br>
Default = -99999 <br>
Specifies the seed for the random number generator.</p>

<p style="margin-top: 1em">Phasing and imputation
parameters <br>
niterations=non-negative_integer <br>
Default = 5 <br>
Specifies the number of phasing iterations. The phasing
iterations are preceded by 10 burn-in iterations which carry
out the Beagle version 4.0 phasing algorithm. If you <br>
want to phase your data with the Beagle 4.0 phasing
algorithm, use niterations=0. Accuracy and compute time
increase with the number of iterations.</p>

<p style="margin-top: 1em">impute=true/false <br>
Default = true <br>
Specifies whether markers that are present in the reference
panel but absent in your data will be imputed. This option
has no effect if the ref and gt arguments are not <br>
used.</p>

<p style="margin-top: 1em">gprobs=true/false <br>
Default = false <br>
Specifies whether a GP (genotype probability) format field
will be included in the output VCF file when imputing
ungenotyped markers. By default, a GP fields is not <br>
printed because a DS (alternate allele dose) format field is
always printed when imputing ungenotyped markers.</p>

<p style="margin-top: 1em">ne=integer <br>
Default = 1000000 <br>
Specifies the effective population size when imputing
ungenotyped markers. The default value is suitable for a
large outbred human population. Smaller values in the
hun&acirc; <br>
dreds or thousands for the ne parameter are suggested for
inbred human and animal populations.</p>

<p style="margin-top: 1em">err=non-negative_number <br>
Default = 0.0001 <br>
Specifies the allele miscall rate. The default value should
give good results for most sequence and SNP array data.</p>

<p style="margin-top: 1em">cluster=non-negative_number <br>
Default = 0.005 <br>
Specifies the maximum cM distance between individual markers
that are combined into an aggregate marker when imputing
ungenotyped markers.</p>

<p style="margin-top: 1em">IBD parameters <br>
ibd=true/false <br>
Default = false <br>
Specifies whether IBD analysis will be performed when the gt
argument is used.</p>

<p style="margin-top: 1em">ibdlod=non-negative_integer <br>
Default = 3.0 <br>
Specifies the minimum LOD score for reported IBD.</p>

<p style="margin-top: 1em">ibdscale=non-negative_number
<br>
Default: data-dependent <br>
Specifies the scale parameter used to build the haplotype
frequency model for IBD analysis. If no ibdscale parameter
is specified the scale parameter for the IBD analysis <br>
will be set to max{2, sqrt[sample size]/100}, which we have
found to work well for outbred populations.</p>

<p style="margin-top: 1em">ibdtrim=non-negative_integer
<br>
Default = 40 <br>
Specifies the number of markers trimmed from the end of a
shared haplotype when testing for IBD. Note: The default
ibdtrim parameter is designed for European samples <br>
genotyped with a 1M SNP array (~ 1 marker per 3 kb). For
human SNP array data, it is recommended to set the ibdtrim
parameter to the typical number of markers in a 0.15 <br>
cM region. Pilot studies of randomly selected genomic
regions can be used to fine-tune the values of the ibdtrim
parameter.</p>

<p style="margin-top: 1em">SEE ALSO <br>

https://faculty.washington.edu/browning/beagle/beagle.html</p>

<p style="margin-top: 1em">AUTHOR <br>
Beagle was written by Brian L. Browning.</p>

<p style="margin-top: 1em">This manual page was written by
Dylan A&Atilde;&macr;ssi &lt;bob.dybian@gmail.com&gt;, for
the Debian project (but may be used by others).</p>

<p style="margin-top: 1em">4.1 February 2016 Beagle(1)</p>
<hr>
</body>
</html>
