<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>Beagle(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">Beagle(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">Beagle(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
Beagle - Genotype calling, genotype phasing and imputation of ungenotyped
  markers
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>java -Xmx[GB]g -jar /usr/share/beagle/beagle.jar</b> [<i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Beagle performs genotype calling, genotype phasing, imputation of ungenotyped
  markers, and identity-by-descent segment detection. Genotypic imputation works
  on phased haplotypes using a Li and Stephens haplotype frequency model. Beagle
  also implements the Refined IBD algorithm for detecting
  homozygosity-by-descent (HBD) and identity-by-descent (IBD) segments.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Data_input/output_parameters"><a class="selflink" href="#Data_input/output_parameters">Data
  input/output parameters</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>gt</b>=<i>filename</i></dt>
  <dd class="It-tag">Optional
    <div>&#x00A0;</div>
    Specifies a VCF file containing a GT (genotype) format field for each
      marker. If a genotype contains the phased allele separator, &quot;|&quot;,
      then Beagle will preserve the phase of the genotype during the analysis.
      If you use the gt argument, all genotypes in the output file will be
      phased and non-missing.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>gl</b>=<i>filename</i></dt>
  <dd class="It-tag">Optional
    <div>&#x00A0;</div>
    Specifies a VCF file containing a GL or PL (genotype likelihood) format
      field for each marker. Any data in the GT format field will be ignored. If
      both GL and PL format fields are present for a marker, the GL format will
      be used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>gtgl</b>=<i>filename</i></dt>
  <dd class="It-tag">Optional
    <div>&#x00A0;</div>
    Specifies a VCF file containing a GT, GL or PL format field for each marker.
      If a genotype is non-missing, Beagle will ignore the genotype likelihood.
      If both GL and PL format fields are present for a marker, the GL field
      will be used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ref</b>=<i>filename</i></dt>
  <dd class="It-tag">Optional
    <div>&#x00A0;</div>
    Specifies a VCF file containing phased reference genotypes. See the impute
      parameter.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>out</b>=<i>prefix</i></dt>
  <dd class="It-tag">Required
    <div>&#x00A0;</div>
    Specifies the output filename prefix. The prefix may be an absolute or
      relative filename, but it cannot be a directory name.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>excludesamples</b>=<i>filename</i></dt>
  <dd class="It-tag">Optional
    <div>&#x00A0;</div>
    Specifies a file containing non-reference samples (one sample per line) to
      be excluded from the analysis and output files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>excludemarkers</b>=<i>filename</i></dt>
  <dd class="It-tag">Optional
    <div>&#x00A0;</div>
    Specifies a file containing markers (one marker per line) to be excluded
      from the analysis and the output files. An excluded marker identifier can
      either be an identifier from the VCF record&#x2019;s ID field or a genomic
      coordinate in the format: CHROM:POS.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>map</b>=<i>filename</i></dt>
  <dd class="It-tag">Optional
    <div>&#x00A0;</div>
    Specifies a PLINK format genetic map on the cM scale. HapMap GrCh36 and
      GrCh37 genetic maps in PLINK format are available for download from the
      Beagle website. Use of a genetic map is recommended if you are imputing
      ungenotyped markers. If no genetic map is specified, Beagle will assume a
      constant recombination rate of 1 cM / Mb.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>chrom</b>=<i>chrom:start-end</i></dt>
  <dd class="It-tag">Optional
    <div>&#x00A0;</div>
    Specifies a chromosome or chromosome interval using a chromosome identifier
      in the VCF file and the starting and ending positions of the interval. The
      entire chromosome, the beginning of the chromosome, and the end of a
      chromosome can be specified by <b>chrom</b>=[chrom],
      <b>chrom</b>=[chrom:-end], and <b>chrom</b>=[chrom:start-]
    respectively.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>maxlr</b>=<i>number_&#x2265;_1</i></dt>
  <dd class="It-tag">Default = 5000
    <div>&#x00A0;</div>
    Specifies the maximum likelihood ratio at a genotype. If M is the maximum of
      the likelihoods of each possible genotype, any likelihood that is less
      than (M &#x2044; maxlr) is set to 0.0 to improve computational efficiency.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="General_parameters"><a class="selflink" href="#General_parameters">General
  parameters</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>nthreads</b>=<i>positive_integer</i></dt>
  <dd class="It-tag">Default: machine-dependent
    <div>&#x00A0;</div>
    Specifies the number of threads of execution. If no <b>nthreads</b>
      parameter is specified, the <b>nthreads</b> parameter will be set equal to
      the number of CPU cores on the host machine.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>lowmem</b>=<i>true/false</i></dt>
  <dd class="It-tag">Default = false
    <div>&#x00A0;</div>
    Specifies whether a memory efficient algorithm should be used. The memory
      efficient algorithm increases run-time by a factor less than 2.0.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>window</b>=<i>positive_integer</i></dt>
  <dd class="It-tag">Default = 50000
    <div>&#x00A0;</div>
    Specifies the number of markers to include in each sliding window. The
      <b>window</b> parameter must be at least twice as large as the overlap
      parameter. The <b>window</b> parameter controls the amount of memory used
      in the analysis. For human data, it is recommended that the <b>window</b>
      parameter be greater than or equal to the typical number of markers in 5
      cM.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>overlap</b>=<i>positive_integer</i></dt>
  <dd class="It-tag">Default = 3000
    <div>&#x00A0;</div>
    Specifies the number of markers of overlap between sliding windows. For
      human data, it is recommended that the overlap be set to the typical
      number of markers in 0.5 cM (when <b>ibd</b>=false) or 2.0 cM (when
      <b>ibd</b>=true).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>seed</b>=<i>integer</i></dt>
  <dd class="It-tag">Default = -99999
    <div>&#x00A0;</div>
    Specifies the seed for the random number generator.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Phasing_and_imputation_parameters"><a class="selflink" href="#Phasing_and_imputation_parameters">Phasing
  and imputation parameters</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>niterations</b>=<i>non-negative_integer</i></dt>
  <dd class="It-tag">Default = 5
    <div>&#x00A0;</div>
    Specifies the number of phasing iterations. The phasing iterations are
      preceded by 10 burn-in iterations which carry out the Beagle version 4.0
      phasing algorithm. If you want to phase your data with the Beagle 4.0
      phasing algorithm, use <b>niterations</b>=0. Accuracy and compute time
      increase with the number of iterations.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>impute</b>=<i>true/false</i></dt>
  <dd class="It-tag">Default = true
    <div>&#x00A0;</div>
    Specifies whether markers that are present in the reference panel but absent
      in your data will be imputed. This option has no effect if the <b>ref</b>
      and <b>gt</b> arguments are not used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>gprobs</b>=<i>true/false</i></dt>
  <dd class="It-tag">Default = false
    <div>&#x00A0;</div>
    Specifies whether a GP (genotype probability) format field will be included
      in the output VCF file when imputing ungenotyped markers. By default, a GP
      fields is not printed because a DS (alternate allele dose) format field is
      always printed when imputing ungenotyped markers.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ne</b>=<i>integer</i></dt>
  <dd class="It-tag">Default = 1000000
    <div>&#x00A0;</div>
    Specifies the effective population size when imputing ungenotyped markers.
      The default value is suitable for a large outbred human population.
      Smaller values in the hundreds or thousands for the <b>ne</b> parameter
      are suggested for inbred human and animal populations.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>err</b>=<i>non-negative_number</i></dt>
  <dd class="It-tag">Default = 0.0001
    <div>&#x00A0;</div>
    Specifies the allele miscall rate. The default value should give good
      results for most sequence and SNP array data.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cluster</b>=<i>non-negative_number</i></dt>
  <dd class="It-tag">Default = 0.005
    <div>&#x00A0;</div>
    Specifies the maximum cM distance between individual markers that are
      combined into an aggregate marker when imputing ungenotyped markers.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="IBD_parameters"><a class="selflink" href="#IBD_parameters">IBD
  parameters</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ibd</b>=<i>true/false</i></dt>
  <dd class="It-tag">Default = false
    <div>&#x00A0;</div>
    Specifies whether IBD analysis will be performed when the <b>gt</b> argument
      is used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ibdlod</b>=<i>non-negative_integer</i></dt>
  <dd class="It-tag">Default = 3.0
    <div>&#x00A0;</div>
    Specifies the minimum LOD score for reported IBD.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ibdscale</b>=<i>non-negative_number</i></dt>
  <dd class="It-tag">Default: data-dependent
    <div>&#x00A0;</div>
    Specifies the scale parameter used to build the haplotype frequency model
      for IBD analysis. If no <b>ibdscale</b> parameter is specified the scale
      parameter for the IBD analysis will be set to max{2, sqrt[sample
      size]/100}, which we have found to work well for outbred populations.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ibdtrim</b>=<i>non-negative_integer</i></dt>
  <dd class="It-tag">Default = 40
    <div>&#x00A0;</div>
    Specifies the number of markers trimmed from the end of a shared haplotype
      when testing for IBD. Note: The default <b>ibdtrim</b> parameter is
      designed for European samples genotyped with a 1M SNP array (~ 1 marker
      per 3 kb). For human SNP array data, it is recommended to set the
      <b>ibdtrim</b> parameter to the typical number of markers in a 0.15 cM
      region. Pilot studies of randomly selected genomic regions can be used to
      fine-tune the values of the <b>ibdtrim</b> parameter.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>https://faculty.washington.edu/browning/beagle/beagle.html</b>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Beagle was written by Brian L. Browning.
<div class="Pp"></div>
This manual page was written by Dylan A&#x00EF;ssi &lt;bob.dybian@gmail.com&gt;,
  for the Debian project (but may be used by others).
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2016</td>
    <td class="foot-os">4.1</td>
  </tr>
</table>
</body>
</html>
