<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BED2GFF3(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BED2GFF3(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BED2GFF3(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
ucsc_genes2gff.pl - Convert UCSC Genome Browser-format gene files into GFF files
  suitable for loading into gbrowse
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  % uscsc_genes2gff.pl [options] ucsc_file1 ucsc_file2...
</pre>
<div class="Pp"></div>
Options:
<div class="Pp"></div>
<pre>
    -src    &lt;string&gt;   Choose a source for the gene, default &quot;UCSC&quot;
    -origin &lt;integer&gt;  Choose a relative position to number from, default
is &quot;1&quot;
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script massages the gene files available from the &quot;tables&quot; link
  of the UCSC genome browser (genome.ucsc.edu) into a form suitable for loading
  of gbrowse. Warning: it only works with the gene tables. Other tables, such as
  EST alignments, contours and repeats, have their own formats which will
  require other scripts to parse.
<div class="Pp"></div>
To use this script, get one or more UCSC tables, either from the
  &quot;Tables&quot; link on the browser, or from the UCSC Genome Browser FTP
  site. Give the table file as the argument to this script. You may want to
  provide an alternative &quot;source&quot; field. Otherwise this script
  defaults to &quot;UCSC&quot;.
<div class="Pp"></div>
<pre>
  % pucsc_genes2gff.pl -src RefSeq refseq_data.ucsc &gt; refseq.gff
</pre>
<div class="Pp"></div>
The resulting GFF file can then be loaded into a Bio::DB::GFF database using the
  following command:
<div class="Pp"></div>
<pre>
  % bulk_load_gff.pl -d &lt;databasename&gt; refseq.gff
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Bio::DB::GFF, bulk_load_gff.pl, load_gff.pl
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Lincoln Stein &lt;lstein@cshl.org&gt;.
<div class="Pp"></div>
Copyright (c) 2003 Cold Spring Harbor Laboratory
<div class="Pp"></div>
This library is free software; you can redistribute it and/or modify it under
  the same terms as Perl itself. See DISCLAIMER.txt for disclaimers of
  warranty.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-04-12</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
