<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:54:42 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>EXPRESS(1) User Commands EXPRESS(1)</p>

<p style="margin-top: 1em">NAME <br>
berkeley-express - Streaming quantification for
high-throughput sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
File Usage: express [options] &lt;target_seqs.fa&gt;
&lt;hits.(sam/bam)&gt; Piped Usage: bowtie [options] -S
&lt;index&gt; &lt;reads.fq&gt; | express [options]
&lt;target_seqs.fa&gt;</p>

<p style="margin-top: 1em">Required arguments: <br>
&lt;target_seqs.fa&gt; <br>
target sequence file in fasta format</p>

<p style="margin-top: 1em">&lt;hits.(sam/bam)&gt; <br>
read alignment file in SAM or BAM format</p>

<p style="margin-top: 1em">Standard Options: <br>
-h [ --help ] <br>
produce help message</p>

<p style="margin-top: 1em">-o [ --output-dir ] arg (=.)
<br>
write all output files to this directory</p>

<p style="margin-top: 1em">-D [ --preprocess ] <br>
run preprocess script for eXpressD</p>

<p style="margin-top: 1em">-m [ --frag-len-mean ] arg
(=200) <br>
prior estimate for average fragment length</p>

<p style="margin-top: 1em">-s [ --frag-len-stddev ] arg
(=80) <br>
prior estimate for fragment length std deviation</p>

<p style="margin-top: 1em">-H [ --haplotype-file ] arg <br>
path to a file containing haplotype pairs</p>

<p style="margin-top: 1em">-B [ --additional-batch ] arg
(=0) <br>
number of additional batch EM rounds after initial online
round</p>

<p style="margin-top: 1em">-O [ --additional-online ] arg
(=0) number of additional online EM rounds <br>
after initial online round</p>

<p style="margin-top: 1em">-L [ --max-read-len ] arg (=250)
<br>
maximum allowed length of a read</p>

<p style="margin-top: 1em">--output-align-prob <br>
output alignments (sam/bam) with probabilistic
assignments</p>

<p style="margin-top: 1em">--output-align-samp <br>
output alignments (sam/bam) with sampled assignments</p>

<p style="margin-top: 1em">--fr-stranded <br>
accept only forward-&gt;reverse alignments (second-stranded
protocols)</p>

<p style="margin-top: 1em">--rf-stranded <br>
accept only reverse-&gt;forward alignments (first-stranded
protocols)</p>

<p style="margin-top: 1em">--f-stranded <br>
accept only forward single-end alignments (second-stranded
protocols)</p>

<p style="margin-top: 1em">--r-stranded <br>
accept only reverse single-end alignments (first-stranded
protocols)</p>

<p style="margin-top: 1em">--no-update-check <br>
disables automatic check for update via web</p>

<p style="margin-top: 1em">--logtostderr <br>
prints all logging messages to stderr</p>

<p style="margin-top: 1em">Advanced Options: <br>
-f [ --forget-param ] arg (=0.84999999999999998) <br>
sets the &rsquo;forgetting factor&rsquo; parameter (0.5 &lt;
c &lt;= 1)</p>

<p style="margin-top: 1em">--library-size arg <br>
specifies library size for FPKM instead of calculating from
alignments</p>

<p style="margin-top: 1em">--max-indel-size arg (=10) <br>
sets the maximum allowed indel size, affecting geometric
indel prior</p>

<p style="margin-top: 1em">--calc-covar <br>
calculate and output covariance matrix</p>

<p style="margin-top: 1em">--expr-alpha arg
(=0.0050000000000000001) <br>
sets the strength of the prior, per bp</p>

<p style="margin-top: 1em">--stop-at arg (=0) <br>
sets the number of fragments to process, disabled with 0</p>

<p style="margin-top: 1em">--burn-out arg (=5000000) <br>
sets number of fragments after which to stop updating
auxiliary parameters</p>

<p style="margin-top: 1em">--no-bias-correct <br>
disables bias correction</p>

<p style="margin-top: 1em">--no-error-model <br>
disables error modelling</p>

<p style="margin-top: 1em">--aux-param-file arg <br>
path to file containing auxiliary parameters to use instead
of learning</p>

<p style="margin-top: 1em">express 1.5.1 February 2015
EXPRESS(1)</p>
<hr>
</body>
</html>
