<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>EAGLE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">EAGLE(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">EAGLE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
eagle - Haplotype phasing within a genotyped cohort or using a phased reference
  panel
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>eagle</b> [ <i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Eagle estimates haplotype phase either within a genotyped cohort or using a
  phased reference panel. The basic idea of the Eagle1 algorithm is to harness
  identity-by-descent among distant relatives&#x2014;which is pervasive at very
  large sample sizes but rare among smaller numbers of samples&#x2014;to rapidly
  call phase using a fast scoring approach. In contrast, the Eagle2 algorithm
  analyzes a full probabilistic model similar to the diploid Li-Stephens model
  used by previous HMM-based methods.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">Print this help</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--geneticMapFile</b> arg</dt>
  <dd class="It-tag">HapMap genetic map provided with download:
      tables/genetic_map_hg##.txt.gz</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outPrefix</b> arg</dt>
  <dd class="It-tag">prefix for output files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--numThreads</b> arg (=1)</dt>
  <dd class="It-tag">number of computational threads</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_options_for_phasing_without_a_reference:"><a class="selflink" href="#Input_options_for_phasing_without_a_reference:">Input
  options for phasing without a reference:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bfile</b> arg</dt>
  <dd class="It-tag">prefix of PLINK .fam, .bim, .bed files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bfilegz</b> arg</dt>
  <dd class="It-tag">prefix of PLINK .fam.gz, .bim.gz, .bed.gz files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fam</b> arg</dt>
  <dd class="It-tag">PLINK .fam file (note: file names ending in .gz are
      auto-decompressed)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bim</b> arg</dt>
  <dd class="It-tag">PLINK .bim file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bed</b> arg</dt>
  <dd class="It-tag">PLINK .bed file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--vcf</b> arg</dt>
  <dd class="It-tag">[compressed] VCF/BCF file containing input genotypes</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--remove</b> arg</dt>
  <dd class="It-tag">file(s) listing individuals to ignore (no header; FID IID
      must be first two columns)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--exclude</b> arg</dt>
  <dd class="It-tag">file(s) listing SNPs to ignore (no header; SNP ID must be
      first column)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxMissingPerSnp</b> arg (=0.1)</dt>
  <dd class="It-tag">QC filter: max missing rate per SNP</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxMissingPerIndiv</b> arg (=0.1)</dt>
  <dd class="It-tag">QC filter: max missing rate per person</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input/output_options_for_phasing_using_a_reference_panel:"><a class="selflink" href="#Input/output_options_for_phasing_using_a_reference_panel:">Input/output
  options for phasing using a reference panel:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--vcfRef</b> arg</dt>
  <dd class="It-tag">tabix-indexed [compressed] VCF/BCF file for reference
      haplotypes</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--vcfTarget</b> arg</dt>
  <dd class="It-tag">tabix-indexed [compressed] VCF/BCF file for target
      genotypes</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--vcfOutFormat</b> arg (=z)</dt>
  <dd class="It-tag">b|u|z|v: compressed BCF (b), uncomp BCF (u), compressed VCF
      (z), uncomp VCF (v)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noImpMissing</b></dt>
  <dd class="It-tag">disable imputation of missing ./. target genotypes</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--allowRefAltSwap</b></dt>
  <dd class="It-tag">allow swapping of REF/ALT in target vs. ref VCF</dd>
</dl>
<h2 class="Ss" title="Ss" id="Region_selection_options:"><a class="selflink" href="#Region_selection_options:">Region
  selection options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--chrom</b> arg (=0)</dt>
  <dd class="It-tag">chromosome to analyze (if input has many)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bpStart</b> arg (=0)</dt>
  <dd class="It-tag">minimum base pair position to analyze</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bpEnd</b> arg (=1e9)</dt>
  <dd class="It-tag">maximum base pair position to analyze</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bpFlanking</b> arg (=0)</dt>
  <dd class="It-tag">(ref-mode only) flanking region to use during phasing but
      discard in output</dd>
</dl>
<h2 class="Ss" title="Ss" id="Algorithm_options:"><a class="selflink" href="#Algorithm_options:">Algorithm
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--Kpbwt</b> arg (=10000)</dt>
  <dd class="It-tag">number of conditioning haplotypes</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pbwtIters</b> arg (=0)</dt>
  <dd class="It-tag">number of PBWT phasing iterations (0=auto)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--expectIBDcM</b> arg (=2.0)</dt>
  <dd class="It-tag">expected length of haplotype copying (cM)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--histFactor</b> arg (=0)</dt>
  <dd class="It-tag">history length multiplier (0=auto)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--genoErrProb</b> arg (=0.003)</dt>
  <dd class="It-tag">estimated genotype error probability</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pbwtOnly</b></dt>
  <dd class="It-tag">in non-ref mode, use only PBWT iters (automatic for
      sequence data)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--v1</b></dt>
  <dd class="It-tag">use Eagle1 phasing algorithm (instead of default Eagle2
      algorithm)</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>https://data.broadinstitute.org/alkesgroup/Eagle/</b>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright &#x00A9; 2015-2016 Harvard University. Distributed under the GNU
  GPLv3+ open source license.</div>
<table class="foot">
  <tr>
    <td class="foot-date">September 2016</td>
    <td class="foot-os">2.3</td>
  </tr>
</table>
</body>
</html>
