<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:54:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>EAGLE(1) General Commands Manual EAGLE(1)</p>

<p style="margin-top: 1em">NAME <br>
eagle - Haplotype phasing within a genotyped cohort or using
a phased reference panel</p>

<p style="margin-top: 1em">SYNOPSIS <br>
eagle [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Eagle estimates haplotype phase either within a genotyped
cohort or using a phased reference panel. The basic idea of
the Eagle1 algorithm is to harness identity-by-descent among
<br>
distant relatives&acirc;which is pervasive at very large
sample sizes but rare among smaller numbers of
samples&acirc;to rapidly call phase using a fast scoring
approach. In contrast, the <br>
Eagle2 algorithm analyzes a full probabilistic model similar
to the diploid Li-Stephens model used by previous HMM-based
methods.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h, --help <br>
Print this help</p>

<p style="margin-top: 1em">--geneticMapFile arg <br>
HapMap genetic map provided with download:
tables/genetic_map_hg##.txt.gz</p>

<p style="margin-top: 1em">--outPrefix arg <br>
prefix for output files</p>

<p style="margin-top: 1em">--numThreads arg (=1) <br>
number of computational threads</p>

<p style="margin-top: 1em">Input options for phasing
without a reference: <br>
--bfile arg <br>
prefix of PLINK .fam, .bim, .bed files</p>

<p style="margin-top: 1em">--bfilegz arg <br>
prefix of PLINK .fam.gz, .bim.gz, .bed.gz files</p>

<p style="margin-top: 1em">--fam arg <br>
PLINK .fam file (note: file names ending in .gz are
auto-decompressed)</p>

<p style="margin-top: 1em">--bim arg <br>
PLINK .bim file</p>

<p style="margin-top: 1em">--bed arg <br>
PLINK .bed file</p>

<p style="margin-top: 1em">--vcf arg <br>
[compressed] VCF/BCF file containing input genotypes</p>

<p style="margin-top: 1em">--remove arg <br>
file(s) listing individuals to ignore (no header; FID IID
must be first two columns)</p>

<p style="margin-top: 1em">--exclude arg <br>
file(s) listing SNPs to ignore (no header; SNP ID must be
first column)</p>

<p style="margin-top: 1em">--maxMissingPerSnp arg (=0.1)
<br>
QC filter: max missing rate per SNP</p>

<p style="margin-top: 1em">--maxMissingPerIndiv arg (=0.1)
<br>
QC filter: max missing rate per person</p>

<p style="margin-top: 1em">Input/output options for phasing
using a reference panel: <br>
--vcfRef arg <br>
tabix-indexed [compressed] VCF/BCF file for reference
haplotypes</p>

<p style="margin-top: 1em">--vcfTarget arg <br>
tabix-indexed [compressed] VCF/BCF file for target
genotypes</p>

<p style="margin-top: 1em">--vcfOutFormat arg (=z) <br>
b|u|z|v: compressed BCF (b), uncomp BCF (u), compressed VCF
(z), uncomp VCF (v)</p>

<p style="margin-top: 1em">--noImpMissing <br>
disable imputation of missing ./. target genotypes</p>

<p style="margin-top: 1em">--allowRefAltSwap <br>
allow swapping of REF/ALT in target vs. ref VCF</p>

<p style="margin-top: 1em">Region selection options: <br>
--chrom arg (=0) <br>
chromosome to analyze (if input has many)</p>

<p style="margin-top: 1em">--bpStart arg (=0) <br>
minimum base pair position to analyze</p>

<p style="margin-top: 1em">--bpEnd arg (=1e9) <br>
maximum base pair position to analyze</p>

<p style="margin-top: 1em">--bpFlanking arg (=0) <br>
(ref-mode only) flanking region to use during phasing but
discard in output</p>

<p style="margin-top: 1em">Algorithm options: <br>
--Kpbwt arg (=10000) <br>
number of conditioning haplotypes</p>

<p style="margin-top: 1em">--pbwtIters arg (=0) <br>
number of PBWT phasing iterations (0=auto)</p>

<p style="margin-top: 1em">--expectIBDcM arg (=2.0) <br>
expected length of haplotype copying (cM)</p>

<p style="margin-top: 1em">--histFactor arg (=0) <br>
history length multiplier (0=auto)</p>

<p style="margin-top: 1em">--genoErrProb arg (=0.003) <br>
estimated genotype error probability</p>

<p style="margin-top: 1em">--pbwtOnly <br>
in non-ref mode, use only PBWT iters (automatic for sequence
data)</p>

<p style="margin-top: 1em">--v1 use Eagle1 phasing
algorithm (instead of default Eagle2 algorithm)</p>

<p style="margin-top: 1em">SEE ALSO <br>
https://data.broadinstitute.org/alkesgroup/Eagle/</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 2015-2016 Harvard University.
Distributed under the GNU GPLv3+ open source license.</p>

<p style="margin-top: 1em">2.3 September 2016 EAGLE(1)</p>
<hr>
</body>
</html>
