<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:54:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>RAINBOW(1) User Commands RAINBOW(1)</p>

<p style="margin-top: 1em">NAME <br>
rainbow - manual page for rainbow 2.0.4 --
&lt;ruanjue@gmail.com, chongzechen@gmail.com&gt;</p>

<p style="margin-top: 1em">SYNOPSIS <br>
rainbow &lt;cmd&gt; [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
rainbow 2.0.4 -- &lt;ruanjue@gmail.com,
chongzechen@gmail.com&gt;</p>

<p style="margin-top: 1em">cluster</p>

<p style="margin-top: 1em">Input File Format: paired
fasta/fastq file(s) Output File Format:
&lt;seqid:int&gt;&lt;cluster_id:int&gt;&lt;read1:string&gt;&lt;read2:string&gt;</p>

<p style="margin-top: 1em">-1 &lt;string&gt; Input
fasta/fastq file, supports multiple &rsquo;-1&rsquo;</p>

<p style="margin-top: 1em">-2 &lt;string&gt; Input
fasta/fastq file, supports multiple &rsquo;-2&rsquo;
[null]</p>

<p style="margin-top: 1em">-l &lt;int&gt; <br>
Read length, default: 0 variable</p>

<p style="margin-top: 1em">-m &lt;int&gt; <br>
Maximum mismatches [4]</p>

<p style="margin-top: 1em">-e &lt;int&gt; <br>
Exactly matching threshold [2000]</p>

<p style="margin-top: 1em">-L Low level of polymorphism</p>

<p style="margin-top: 1em">div</p>

<p style="margin-top: 1em">Input File Format:
&lt;seqid:int&gt;&lt;cluster_id:int&gt;&lt;read1:string&gt;&lt;read2:string&gt;
Output File Format:
&lt;seqid:int&gt;&lt;cluster_id:int&gt;&lt;read1:string&gt;&lt;read2:string&gt;[&lt;pre_clus&acirc;
<br>
ter_id:int&gt;]</p>

<p style="margin-top: 1em">-i &lt;string&gt; Input file
[stdin]</p>

<p style="margin-top: 1em">-o &lt;string&gt; Output file
[stdout]</p>

<p style="margin-top: 1em">-k &lt;int&gt; <br>
K_allele, min variants to create a new group [2]</p>

<p style="margin-top: 1em">-K &lt;int&gt; <br>
K_allele, divide regardless of frequency when num of
variants exceed this value [50]</p>

<p style="margin-top: 1em">-f &lt;float&gt; <br>
Frequency, min variant frequency to create a new group
[0.2]</p>

<p style="margin-top: 1em">merge</p>

<p style="margin-top: 1em">Input File Format:
&lt;seqid:int&gt;&lt;cluster_id:int&gt;&lt;read1:string&gt;&lt;read2:string&gt;[&lt;pre_cluster_id:int&gt;]</p>

<p style="margin-top: 1em">-i &lt;string&gt; Input rbasm
output file [stdin]</p>

<p style="margin-top: 1em">-a output assembly</p>

<p style="margin-top: 1em">-o &lt;string&gt; Output file
for merged contigs, one line per cluster [stdout]</p>

<p style="margin-top: 1em">-N &lt;int&gt; <br>
Maximum number of divided clusters to merge [300]</p>

<p style="margin-top: 1em">-l &lt;int&gt; <br>
Minimum overlap when assemble two reads (valid only when
&rsquo;-a&rsquo; is opened) [5]</p>

<p style="margin-top: 1em">-f &lt;float&gt; <br>
Minimum fraction of similarity when assembly (valid only
when &rsquo;-a&rsquo; is opened) [0.90]</p>

<p style="margin-top: 1em">-r &lt;int&gt; <br>
Minimum number of reads to assemble (valid only when
&rsquo;-a&rsquo; is opened) [5]</p>

<p style="margin-top: 1em">-R &lt;int&gt; <br>
Maximum number of reads to assemble (valid only when
&rsquo;-a&rsquo; is opened) [300]</p>

<p style="margin-top: 1em">Usage: rainbow &lt;cmd&gt;
[options]</p>

<p style="margin-top: 1em">cluster</p>

<p style="margin-top: 1em">Input File Format: paired
fasta/fastq file(s) Output File Format:
&lt;seqid:int&gt;&lt;cluster_id:int&gt;&lt;read1:string&gt;&lt;read2:string&gt;</p>

<p style="margin-top: 1em">-1 &lt;string&gt; Input
fasta/fastq file, supports multiple &rsquo;-1&rsquo;</p>

<p style="margin-top: 1em">-2 &lt;string&gt; Input
fasta/fastq file, supports multiple &rsquo;-2&rsquo;
[null]</p>

<p style="margin-top: 1em">-l &lt;int&gt; <br>
Read length, default: 0 variable</p>

<p style="margin-top: 1em">-m &lt;int&gt; <br>
Maximum mismatches [4]</p>

<p style="margin-top: 1em">-e &lt;int&gt; <br>
Exactly matching threshold [2000]</p>

<p style="margin-top: 1em">-L Low level of polymorphism</p>

<p style="margin-top: 1em">div</p>

<p style="margin-top: 1em">Input File Format:
&lt;seqid:int&gt;&lt;cluster_id:int&gt;&lt;read1:string&gt;&lt;read2:string&gt;
Output File Format:
&lt;seqid:int&gt;&lt;cluster_id:int&gt;&lt;read1:string&gt;&lt;read2:string&gt;[&lt;pre_clus&acirc;
<br>
ter_id:int&gt;]</p>

<p style="margin-top: 1em">-i &lt;string&gt; Input file
[stdin]</p>

<p style="margin-top: 1em">-o &lt;string&gt; Output file
[stdout]</p>

<p style="margin-top: 1em">-k &lt;int&gt; <br>
K_allele, min variants to create a new group [2]</p>

<p style="margin-top: 1em">-K &lt;int&gt; <br>
K_allele, divide regardless of frequency when num of
variants exceed this value [50]</p>

<p style="margin-top: 1em">-f &lt;float&gt; <br>
Frequency, min variant frequency to create a new group
[0.2]</p>

<p style="margin-top: 1em">merge</p>

<p style="margin-top: 1em">Input File Format:
&lt;seqid:int&gt;&lt;cluster_id:int&gt;&lt;read1:string&gt;&lt;read2:string&gt;[&lt;pre_cluster_id:int&gt;]</p>

<p style="margin-top: 1em">-i &lt;string&gt; Input rbasm
output file [stdin]</p>

<p style="margin-top: 1em">-a output assembly</p>

<p style="margin-top: 1em">-o &lt;string&gt; Output file
for merged contigs, one line per cluster [stdout]</p>

<p style="margin-top: 1em">-N &lt;int&gt; <br>
Maximum number of divided clusters to merge [300]</p>

<p style="margin-top: 1em">-l &lt;int&gt; <br>
Minimum overlap when assemble two reads (valid only when
&rsquo;-a&rsquo; is opened) [5]</p>

<p style="margin-top: 1em">-f &lt;float&gt; <br>
Minimum fraction of similarity when assembly (valid only
when &rsquo;-a&rsquo; is opened) [0.90]</p>

<p style="margin-top: 1em">-r &lt;int&gt; <br>
Minimum number of reads to assemble (valid only when
&rsquo;-a&rsquo; is opened) [5]</p>

<p style="margin-top: 1em">-R &lt;int&gt; <br>
Maximum number of reads to assemble (valid only when
&rsquo;-a&rsquo; is opened) [300]</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation for rainbow is maintained as a
Texinfo manual. If the info and rainbow programs are
properly installed at your site, the command</p>

<p style="margin-top: 1em">info rainbow</p>

<p style="margin-top: 1em">should give you access to the
complete manual.</p>

<p style="margin-top: 1em">rainbow 2.0.4 --
&lt;ruanjue@gmail.com, chongzechen@gmail.com&gt; August 2015
RAINBOW(1)</p>
<hr>
</body>
</html>
