<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>RAINBOW(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">RAINBOW(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">RAINBOW(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
rainbow - manual page for rainbow 2.0.4 -- &lt;ruanjue@gmail.com,
  chongzechen@gmail.com&gt;
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>rainbow</b> <i>&lt;cmd&gt; </i>[<i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
rainbow 2.0.4 <b>--</b> &lt;ruanjue@gmail.com, chongzechen@gmail.com&gt;
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">cluster</dd>
</dl>
<div class="Pp"></div>
Input File Format: paired fasta/fastq file(s) Output File Format:
  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-1</b>
  &lt;string&gt; Input fasta/fastq file, supports multiple '-1'</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-2</b>
  &lt;string&gt; Input fasta/fastq file, supports multiple '-2' [null]</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> &lt;int&gt;</dt>
  <dd class="It-tag">Read length, default: 0 variable</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &lt;int&gt;</dt>
  <dd class="It-tag">Maximum mismatches [4]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b> &lt;int&gt;</dt>
  <dd class="It-tag">Exactly matching threshold [2000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b></dt>
  <dd class="It-tag">Low level of polymorphism</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">div</dd>
</dl>
<div class="Pp"></div>
Input File Format:
  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;
  Output File Format:
  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;[\t&lt;pre_cluster_id:int&gt;]
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-i</b>
  &lt;string&gt; Input file [stdin]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  &lt;string&gt; Output file [stdout]</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> &lt;int&gt;</dt>
  <dd class="It-tag">K_allele, min variants to create a new group [2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-K</b> &lt;int&gt;</dt>
  <dd class="It-tag">K_allele, divide regardless of frequency when num of
      variants exceed this value [50]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> &lt;float&gt;</dt>
  <dd class="It-tag">Frequency, min variant frequency to create a new group
      [0.2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">merge</dd>
</dl>
<div class="Pp"></div>
Input File Format:
  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;[\t&lt;pre_cluster_id:int&gt;]
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-i</b>
  &lt;string&gt; Input rbasm output file [stdin]</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b></dt>
  <dd class="It-tag">output assembly</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  &lt;string&gt; Output file for merged contigs, one line per cluster
  [stdout]</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b> &lt;int&gt;</dt>
  <dd class="It-tag">Maximum number of divided clusters to merge [300]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> &lt;int&gt;</dt>
  <dd class="It-tag">Minimum overlap when assemble two reads (valid only when
      '-a' is opened) [5]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> &lt;float&gt;</dt>
  <dd class="It-tag">Minimum fraction of similarity when assembly (valid only
      when '-a' is opened) [0.90]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> &lt;int&gt;</dt>
  <dd class="It-tag">Minimum number of reads to assemble (valid only when '-a'
      is opened) [5]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b> &lt;int&gt;</dt>
  <dd class="It-tag">Maximum number of reads to assemble (valid only when '-a'
      is opened) [300]</dd>
</dl>
<div class="Pp"></div>
Usage: rainbow &lt;cmd&gt; [options]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">cluster</dd>
</dl>
<div class="Pp"></div>
Input File Format: paired fasta/fastq file(s) Output File Format:
  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-1</b>
  &lt;string&gt; Input fasta/fastq file, supports multiple '-1'</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-2</b>
  &lt;string&gt; Input fasta/fastq file, supports multiple '-2' [null]</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> &lt;int&gt;</dt>
  <dd class="It-tag">Read length, default: 0 variable</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &lt;int&gt;</dt>
  <dd class="It-tag">Maximum mismatches [4]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b> &lt;int&gt;</dt>
  <dd class="It-tag">Exactly matching threshold [2000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b></dt>
  <dd class="It-tag">Low level of polymorphism</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">div</dd>
</dl>
<div class="Pp"></div>
Input File Format:
  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;
  Output File Format:
  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;[\t&lt;pre_cluster_id:int&gt;]
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-i</b>
  &lt;string&gt; Input file [stdin]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  &lt;string&gt; Output file [stdout]</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> &lt;int&gt;</dt>
  <dd class="It-tag">K_allele, min variants to create a new group [2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-K</b> &lt;int&gt;</dt>
  <dd class="It-tag">K_allele, divide regardless of frequency when num of
      variants exceed this value [50]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> &lt;float&gt;</dt>
  <dd class="It-tag">Frequency, min variant frequency to create a new group
      [0.2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">merge</dd>
</dl>
<div class="Pp"></div>
Input File Format:
  &lt;seqid:int&gt;\t&lt;cluster_id:int&gt;\t&lt;read1:string&gt;\t&lt;read2:string&gt;[\t&lt;pre_cluster_id:int&gt;]
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-i</b>
  &lt;string&gt; Input rbasm output file [stdin]</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b></dt>
  <dd class="It-tag">output assembly</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  &lt;string&gt; Output file for merged contigs, one line per cluster
  [stdout]</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b> &lt;int&gt;</dt>
  <dd class="It-tag">Maximum number of divided clusters to merge [300]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> &lt;int&gt;</dt>
  <dd class="It-tag">Minimum overlap when assemble two reads (valid only when
      '-a' is opened) [5]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> &lt;float&gt;</dt>
  <dd class="It-tag">Minimum fraction of similarity when assembly (valid only
      when '-a' is opened) [0.90]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> &lt;int&gt;</dt>
  <dd class="It-tag">Minimum number of reads to assemble (valid only when '-a'
      is opened) [5]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b> &lt;int&gt;</dt>
  <dd class="It-tag">Maximum number of reads to assemble (valid only when '-a'
      is opened) [300]</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The full documentation for <b>rainbow</b> is maintained as a Texinfo manual. If
  the <b>info</b> and <b>rainbow</b> programs are properly installed at your
  site, the command
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>info rainbow</b></dd>
</dl>
<div class="Pp"></div>
should give you access to the complete manual.</div>
<table class="foot">
  <tr>
    <td class="foot-date">August 2015</td>
    <td class="foot-os">rainbow 2.0.4 -- &lt;ruanjue@gmail.com,
      chongzechen@gmail.com&gt;</td>
  </tr>
</table>
</body>
</html>
