<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:55:51 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BOWTIE2-ALIGN-L(1) User Commands BOWTIE2-ALIGN-L(1)</p>

<p style="margin-top: 1em">NAME <br>
bowtie2-align-l - ultrafast and memory-efficient backend
tool for aligning sequencing reads to long reference
sequences</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Bowtie 2 version 2.3.0 by Ben Langmead (langmea@cs.jhu.edu,
www.cs.jhu.edu/~langmea)</p>

<p style="margin-top: 1em">USAGE <br>
bowtie2-align [options]* -x &lt;bt2-idx&gt; {-1 &lt;m1&gt;
-2 &lt;m2&gt; | -U &lt;r&gt;} [-S &lt;sam&gt;]</p>

<p style="margin-top: 1em">&lt;bt2-idx&gt; <br>
Index filename prefix (minus trailing .X.bt2l). NOTE: Bowtie
1 and Bowtie 2 indexes are not compatible.</p>

<p style="margin-top: 1em">&lt;m1&gt; Files with #1 mates,
paired with files in &lt;m2&gt;.</p>

<p style="margin-top: 1em">&lt;m2&gt; Files with #2 mates,
paired with files in &lt;m1&gt;.</p>

<p style="margin-top: 1em">&lt;r&gt; Files with unpaired
reads.</p>

<p style="margin-top: 1em">&lt;sam&gt; File for SAM output
(default: stdout)</p>

<p style="margin-top: 1em">&lt;m1&gt;, &lt;m2&gt;,
&lt;r&gt; can be comma-separated lists (no whitespace) and
can be specified many times. E.g. &rsquo;-U
file1.fq,file2.fq -U file3.fq&rsquo;.</p>

<p style="margin-top: 1em">OPTIONS (defaults in
parentheses) <br>
Input: <br>
-q query input files are FASTQ .fq/.fastq (default)</p>

<p style="margin-top: 1em">--qseq query input files are in
Illumina&rsquo;s qseq format</p>

<p style="margin-top: 1em">-f query input files are
(multi-)FASTA .fa/.mfa</p>

<p style="margin-top: 1em">-r query input files are raw
one-sequence-per-line</p>

<p style="margin-top: 1em">-c &lt;m1&gt;, &lt;m2&gt;,
&lt;r&gt; are sequences themselves, not files</p>

<p style="margin-top: 1em">-s/--skip &lt;int&gt; <br>
skip the first &lt;int&gt; reads/pairs in the input
(none)</p>

<p style="margin-top: 1em">-u/--upto &lt;int&gt; <br>
stop after first &lt;int&gt; reads/pairs (no limit)</p>

<p style="margin-top: 1em">-5/--trim5 &lt;int&gt; <br>
trim &lt;int&gt; bases from 5&rsquo;/left end of reads
(0)</p>

<p style="margin-top: 1em">-3/--trim3 &lt;int&gt; <br>
trim &lt;int&gt; bases from 3&rsquo;/right end of reads
(0)</p>

<p style="margin-top: 1em">--phred33 <br>
qualities are Phred+33 (default)</p>

<p style="margin-top: 1em">--phred64 <br>
qualities are Phred+64</p>

<p style="margin-top: 1em">--int-quals <br>
qualities encoded as space-delimited integers</p>

<p style="margin-top: 1em">Presets: <br>
Same as:</p>

<p style="margin-top: 1em">For --end-to-end:</p>

<p style="margin-top: 1em">--very-fast -D 5 -R 1 -N 0 -L 22
-i S,0,2.50</p>

<p style="margin-top: 1em">--fast -D 10 -R 2 -N 0 -L 22 -i
S,0,2.50</p>

<p style="margin-top: 1em">--sensitive -D 15 -R 2 -N 0 -L
22 -i S,1,1.15 (default)</p>

<p style="margin-top: 1em">--very-sensitive -D 20 -R 3 -N 0
-L 20 -i S,1,0.50</p>

<p style="margin-top: 1em">For --local:</p>

<p style="margin-top: 1em">--very-fast-local -D 5 -R 1 -N 0
-L 25 -i S,1,2.00</p>

<p style="margin-top: 1em">--fast-local -D 10 -R 2 -N 0 -L
22 -i S,1,1.75</p>

<p style="margin-top: 1em">--sensitive-local -D 15 -R 2 -N
0 -L 20 -i S,1,0.75 (default)</p>

<p style="margin-top: 1em">--very-sensitive-local -D 20 -R
3 -N 0 -L 20 -i S,1,0.50</p>

<p style="margin-top: 1em">Alignment: <br>
-N &lt;int&gt; <br>
max # mismatches in seed alignment; can be 0 or 1 (0)</p>

<p style="margin-top: 1em">-L &lt;int&gt; <br>
length of seed substrings; must be &gt;3, &lt;32 (22)</p>

<p style="margin-top: 1em">-i &lt;func&gt; <br>
interval between seed substrings w/r/t read len
(S,1,1.15)</p>

<p style="margin-top: 1em">--n-ceil &lt;func&gt; <br>
func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)</p>

<p style="margin-top: 1em">--dpad &lt;int&gt; <br>
include &lt;int&gt; extra ref chars on sides of DP table
(15)</p>

<p style="margin-top: 1em">--gbar &lt;int&gt; <br>
disallow gaps within &lt;int&gt; nucs of read extremes
(4)</p>

<p style="margin-top: 1em">--ignore-quals <br>
treat all quality values as 30 on Phred scale (off)</p>

<p style="margin-top: 1em">--nofw do not align forward
(original) version of read (off)</p>

<p style="margin-top: 1em">--norc do not align
reverse-complement version of read (off)</p>

<p style="margin-top: 1em">--no-1mm-upfront <br>
do not allow 1 mismatch alignments before attempting to scan
for the optimal seeded alignments</p>

<p style="margin-top: 1em">--end-to-end <br>
entire read must align; no clipping (on)</p>

<p style="margin-top: 1em">OR</p>

<p style="margin-top: 1em">--local <br>
local alignment; ends might be soft clipped (off)</p>

<p style="margin-top: 1em">Scoring: <br>
--ma &lt;int&gt; <br>
match bonus (0 for --end-to-end, 2 for --local)</p>

<p style="margin-top: 1em">--mp &lt;int&gt; <br>
max penalty for mismatch; lower qual = lower penalty (6)</p>

<p style="margin-top: 1em">--np &lt;int&gt; <br>
penalty for non-A/C/G/Ts in read/ref (1)</p>

<p style="margin-top: 1em">--rdg &lt;int&gt;,&lt;int&gt;
<br>
read gap open, extend penalties (5,3)</p>

<p style="margin-top: 1em">--rfg &lt;int&gt;,&lt;int&gt;
<br>
reference gap open, extend penalties (5,3)</p>

<p style="margin-top: 1em">--score-min &lt;func&gt; min
acceptable alignment score w/r/t read length <br>
(G,20,8 for local, L,-0.6,-0.6 for end-to-end)</p>

<p style="margin-top: 1em">Reporting: <br>
(default) <br>
look for multiple alignments, report best, with MAPQ</p>

<p style="margin-top: 1em">OR</p>

<p style="margin-top: 1em">-k &lt;int&gt; <br>
report up to &lt;int&gt; alns per read; MAPQ not
meaningful</p>

<p style="margin-top: 1em">OR</p>

<p style="margin-top: 1em">-a/--all <br>
report all alignments; very slow, MAPQ not meaningful</p>

<p style="margin-top: 1em">Effort: <br>
-D &lt;int&gt; <br>
give up extending after &lt;int&gt; failed extends in a row
(15)</p>

<p style="margin-top: 1em">-R &lt;int&gt; <br>
for reads w/ repetitive seeds, try &lt;int&gt; sets of seeds
(2)</p>

<p style="margin-top: 1em">Paired-end:</p>

<p style="margin-top: 1em">-I/--minins &lt;int&gt; <br>
minimum fragment length (0)</p>

<p style="margin-top: 1em">-X/--maxins &lt;int&gt; <br>
maximum fragment length (500)</p>

<p style="margin-top: 1em">--fr/--rf/--ff -1, -2 mates
align fw/rev, rev/fw, fw/fw (--fr)</p>

<p style="margin-top: 1em">--no-mixed <br>
suppress unpaired alignments for paired reads</p>

<p style="margin-top: 1em">--no-discordant <br>
suppress discordant alignments for paired reads</p>

<p style="margin-top: 1em">--no-dovetail <br>
not concordant when mates extend past each other</p>

<p style="margin-top: 1em">--no-contain <br>
not concordant when one mate alignment contains other</p>

<p style="margin-top: 1em">--no-overlap <br>
not concordant when mates overlap at all</p>

<p style="margin-top: 1em">Output: <br>
-t/--time <br>
print wall-clock time taken by search phases</p>

<p style="margin-top: 1em">--quiet <br>
print nothing to stderr except serious errors</p>

<p style="margin-top: 1em">--met-file &lt;path&gt; <br>
send metrics to file at &lt;path&gt; (off)</p>

<p style="margin-top: 1em">--met-stderr <br>
send metrics to stderr (off)</p>

<p style="margin-top: 1em">--met &lt;int&gt; <br>
report internal counters &amp; metrics every &lt;int&gt;
secs (1)</p>

<p style="margin-top: 1em">--no-unal <br>
suppress SAM records for unaligned reads</p>

<p style="margin-top: 1em">--no-head <br>
suppress header lines, i.e. lines starting with @</p>

<p style="margin-top: 1em">--no-sq <br>
suppress @SQ header lines</p>

<p style="margin-top: 1em">--rg-id &lt;text&gt; <br>
set read group id, reflected in @RG line and RG:Z: opt
field</p>

<p style="margin-top: 1em">--rg &lt;text&gt; <br>
add &lt;text&gt; (&quot;lab:value&quot;) to @RG line of SAM
header. Note: @RG line only printed when --rg-id is set.</p>

<p style="margin-top: 1em">--omit-sec-seq <br>
put &rsquo;*&rsquo; in SEQ and QUAL fields for secondary
alignments.</p>

<p style="margin-top: 1em">Performance: <br>
-p/--threads &lt;int&gt; number of alignment threads to
launch (1)</p>

<p style="margin-top: 1em">--reorder <br>
force SAM output order to match order of input reads</p>

<p style="margin-top: 1em">--mm use memory-mapped I/O for
index; many &rsquo;bowtie&rsquo;s can share</p>

<p style="margin-top: 1em">Other: <br>
--qc-filter <br>
filter out reads that are bad according to QSEQ filter</p>

<p style="margin-top: 1em">--seed &lt;int&gt; <br>
seed for random number generator (0)</p>

<p style="margin-top: 1em">--non-deterministic seed rand.
gen. arbitrarily instead of using read attributes</p>

<p style="margin-top: 1em">--version <br>
print version information and quit</p>

<p style="margin-top: 1em">-h/--help <br>
print this usage message</p>

<p style="margin-top: 1em">bowtie2-align-l 2.3.0 February
2017 BOWTIE2-ALIGN-L(1)</p>
<hr>
</body>
</html>
