<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BOWTIE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BOWTIE(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">BOWTIE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bowtie - ultrafast memory-efficient short read aligner
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Usage: bowtie [options]* &lt;ebwt&gt; {-1 &lt;m1&gt; <b>-2</b> &lt;m2&gt; |
  <b>--12</b> &lt;r&gt; | &lt;s&gt;} [&lt;hit&gt;]
<dl class="Bl-tag">
  <dt class="It-tag">&lt;m1&gt;</dt>
  <dd class="It-tag">Comma-separated list of files containing upstream mates (or
      the sequences themselves, if <b>-c</b> is set) paired with mates in
      &lt;m2&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;m2&gt;</dt>
  <dd class="It-tag">Comma-separated list of files containing downstream mates
      (or the sequences themselves if <b>-c</b> is set) paired with mates in
      &lt;m1&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;r&gt;</dt>
  <dd class="It-tag">Comma-separated list of files containing Crossbow-style
      reads. Can be a mixture of paired and unpaired. Specify &quot;-&quot; for
      stdin.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;s&gt;</dt>
  <dd class="It-tag">Comma-separated list of files containing unpaired reads, or
      the sequences themselves, if <b>-c</b> is set. Specify &quot;-&quot; for
      stdin.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;hit&gt;</dt>
  <dd class="It-tag">File to write hits to (default: stdout)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input:"><a class="selflink" href="#Input:">Input:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b></dt>
  <dd class="It-tag">query input files are FASTQ .fq/.fastq (default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b></dt>
  <dd class="It-tag">query input files are (multi-)FASTA .fa/.mfa</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b></dt>
  <dd class="It-tag">query input files are raw one-sequence-per-line</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">query sequences given on cmd line (as &lt;mates&gt;,
      &lt;singles&gt;)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b></dt>
  <dd class="It-tag">reads and index are in colorspace</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b>/--quals &lt;file&gt;</dt>
  <dd class="It-tag">QV file(s) corresponding to CSFASTA inputs; use with
      <b>-f</b> <b>-C</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--Q1</b>/--Q2 &lt;file&gt;</dt>
  <dd class="It-tag">same as <b>-Q</b>, but for mate files 1 and 2
    respectively</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>/--skip &lt;int&gt;</dt>
  <dd class="It-tag">skip the first &lt;int&gt; reads/pairs in the input</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b>/--qupto &lt;int&gt;</dt>
  <dd class="It-tag">stop after first &lt;int&gt; reads/pairs (excl. skipped
      reads)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-5</b>/--trim5 &lt;int&gt;</dt>
  <dd class="It-tag">trim &lt;int&gt; bases from 5' (left) end of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-3</b>/--trim3 &lt;int&gt;</dt>
  <dd class="It-tag">trim &lt;int&gt; bases from 3' (right) end of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--phred33-quals</b></dt>
  <dd class="It-tag">input quals are Phred+33 (default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--phred64-quals</b></dt>
  <dd class="It-tag">input quals are Phred+64 (same as
    <b>--solexa1</b>.3-quals)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--solexa-quals</b></dt>
  <dd class="It-tag">input quals are from GA Pipeline ver. &lt; 1.3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--solexa1</b>.3-quals</dt>
  <dd class="It-tag">input quals are from GA Pipeline ver. &gt;= 1.3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--integer-quals</b></dt>
  <dd class="It-tag">qualities are given as space-separated integers (not
    ASCII)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--large-index</b></dt>
  <dd class="It-tag">force usage of a 'large' index, even if a small one is
      present</dd>
</dl>
<h2 class="Ss" title="Ss" id="Alignment:"><a class="selflink" href="#Alignment:">Alignment:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b> &lt;int&gt;</dt>
  <dd class="It-tag">report end-to-end hits w/ &lt;=v mismatches; ignore
      qualities</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">or</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-n</b>/--seedmms
  &lt;int&gt; max mismatches in seed (can be 0-3, default: <b>-n</b> 2)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>/--maqerr &lt;int&gt;</dt>
  <dd class="It-tag">max sum of mismatch quals across alignment for <b>-n</b>
      (def: 70)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-l</b>/--seedlen
  &lt;int&gt; seed length for <b>-n</b> (default: 28)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nomaqround</b></dt>
  <dd class="It-tag">disable Maq-like quality rounding for <b>-n</b> (nearest 10
      &lt;= 30)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-I</b>/--minins &lt;int&gt;</dt>
  <dd class="It-tag">minimum insert size for paired-end alignment (default:
    0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-X</b>/--maxins &lt;int&gt;</dt>
  <dd class="It-tag">maximum insert size for paired-end alignment (default:
    250)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--fr</b>/--rf/--ff
  <b>-1</b>, <b>-2</b> mates align fw/rev, rev/fw, fw/fw (default:
  <b>--fr</b>)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nofw</b>/--norc</dt>
  <dd class="It-tag">do not align to forward/reverse-complement reference
    strand</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxbts</b> &lt;int&gt;</dt>
  <dd class="It-tag">max # backtracks for <b>-n</b> 2/3 (default: 125, 800 for
      <b>--best</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pairtries</b> &lt;int&gt;</dt>
  <dd class="It-tag">max # attempts to find mate for anchor hit (default:
    100)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-y</b>/--tryhard</dt>
  <dd class="It-tag">try hard to find valid alignments, at the expense of
    speed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--chunkmbs</b> &lt;int&gt;</dt>
  <dd class="It-tag">max megabytes of RAM for best-first search frames (def:
    64)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Reporting:"><a class="selflink" href="#Reporting:">Reporting:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> &lt;int&gt;</dt>
  <dd class="It-tag">report up to &lt;int&gt; good alignments per read (default:
      1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b>/--all</dt>
  <dd class="It-tag">report all alignments per read (much slower than low
      <b>-k</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &lt;int&gt;</dt>
  <dd class="It-tag">suppress all alignments if &gt; &lt;int&gt; exist (def: no
      limit)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> &lt;int&gt;</dt>
  <dd class="It-tag">like <b>-m</b>, but reports 1 random hit (MAPQ=0); requires
      <b>--best</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--best</b></dt>
  <dd class="It-tag">hits guaranteed best stratum; ties broken by quality</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--strata</b></dt>
  <dd class="It-tag">hits in sub-optimal strata aren't reported (requires
      <b>--best</b>)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Output:"><a class="selflink" href="#Output:">Output:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>/--time</dt>
  <dd class="It-tag">print wall-clock time taken by search phases</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-B</b>/--offbase
  &lt;int&gt; leftmost ref offset = &lt;int&gt; in bowtie output (default:
  0)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quiet</b></dt>
  <dd class="It-tag">print nothing but the alignments</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--refout</b></dt>
  <dd class="It-tag">write alignments to files refXXXXX.map, 1 map per
    reference</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--refidx</b></dt>
  <dd class="It-tag">refer to ref. seqs by 0-based index rather than name</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--al</b> &lt;fname&gt;</dt>
  <dd class="It-tag">write aligned reads/pairs to file(s) &lt;fname&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--un</b> &lt;fname&gt;</dt>
  <dd class="It-tag">write unaligned reads/pairs to file(s) &lt;fname&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max</b> &lt;fname&gt;</dt>
  <dd class="It-tag">write reads/pairs over <b>-m</b> limit to file(s)
      &lt;fname&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--suppress</b> &lt;cols&gt;</dt>
  <dd class="It-tag">suppresses given columns (comma-delim'ed) in default
    output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fullref</b></dt>
  <dd class="It-tag">write entire ref name (default: only up to 1st space)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Colorspace:"><a class="selflink" href="#Colorspace:">Colorspace:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--snpphred</b> &lt;int&gt;</dt>
  <dd class="It-tag">Phred penalty for SNP when decoding colorspace (def:
    30)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">or</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--snpfrac</b> &lt;dec&gt;</dt>
  <dd class="It-tag">approx. fraction of SNP bases (e.g. 0.001); sets
      <b>--snpphred</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--col-cseq</b></dt>
  <dd class="It-tag">print aligned colorspace seqs as colors, not decoded
    bases</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--col-cqual</b></dt>
  <dd class="It-tag">print original colorspace quals, not decoded quals</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--col-keepends</b></dt>
  <dd class="It-tag">keep nucleotides at extreme ends of decoded alignment</dd>
</dl>
<h2 class="Ss" title="Ss" id="SAM:"><a class="selflink" href="#SAM:">SAM:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b>/--sam</dt>
  <dd class="It-tag">write hits in SAM format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mapq</b> &lt;int&gt;</dt>
  <dd class="It-tag">default mapping quality (MAPQ) to print for SAM
    alignments</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-nohead</b></dt>
  <dd class="It-tag">supppress header lines (starting with @) for SAM
    output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-nosq</b></dt>
  <dd class="It-tag">supppress @SQ header lines for SAM output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-RG</b> &lt;text&gt;</dt>
  <dd class="It-tag">add &lt;text&gt; (usually &quot;lab=value&quot;) to @RG
      line of SAM header</dd>
</dl>
<h2 class="Ss" title="Ss" id="Performance:"><a class="selflink" href="#Performance:">Performance:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>/--offrate
  &lt;int&gt; override offrate of index; must be &gt;= index's offrate</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-p</b>/--threads
  &lt;int&gt; number of alignment threads to launch (default: 1)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mm</b></dt>
  <dd class="It-tag">use memory-mapped I/O for index; many 'bowtie's can
    share</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--shmem</b></dt>
  <dd class="It-tag">use shared mem for index; many 'bowtie's can share</dd>
</dl>
<h2 class="Ss" title="Ss" id="Other:"><a class="selflink" href="#Other:">Other:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b> &lt;int&gt;</dt>
  <dd class="It-tag">seed for random number generator</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--verbose</b></dt>
  <dd class="It-tag">verbose output (for debugging)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">print version information and quit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>/--help</dt>
  <dd class="It-tag">print this usage message</dd>
</dl>
<div class="Pp"></div>
64-bit Built on Debian-reproducible Fri, 14 Oct 2016 20:53:17 +0200 Compiler:
  gcc version 6.2.1 20161215 Options: <b>-O3</b> <b>-Wl</b>,--hash-style=both
  <b>-Wdate-time</b> <b>-D_FORTIFY_SOURCE</b>=<i>2</i> <b>-g</b> <b>-O2</b>
  <b>-fdebug-prefix-map=</b>.=. <b>-fPIE</b> <b>-fstack-protector-strong</b>
  <b>-Wformat</b> <b>-Werror</b>=<i>format-security</i> <b>-g</b> <b>-O2</b>
  <b>-fdebug-prefix-map=</b>.=. <b>-fPIE</b> <b>-fstack-protector-strong</b>
  <b>-Wformat</b> <b>-Werror</b>=<i>format-security</i>
  <b>-Wl</b>,-Bsymbolic-functions <b>-fPIE</b> <b>-pie</b> <b>-Wl</b>,-z,relro
  <b>-Wl</b>,-z,now Sizeof {int, long, long long, void*, size_t, off_t}: {4, 8,
  8, 8, 8, 8}</div>
<table class="foot">
  <tr>
    <td class="foot-date">October 2016</td>
    <td class="foot-os">bowtie version 1.1.2</td>
  </tr>
</table>
</body>
</html>
