<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:55:50 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BOWTIE(1) User Commands BOWTIE(1)</p>

<p style="margin-top: 1em">NAME <br>
bowtie - ultrafast memory-efficient short read aligner</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Usage: bowtie [options]* &lt;ebwt&gt; {-1 &lt;m1&gt; -2
&lt;m2&gt; | --12 &lt;r&gt; | &lt;s&gt;} [&lt;hit&gt;]</p>

<p style="margin-top: 1em">&lt;m1&gt; Comma-separated list
of files containing upstream mates (or the sequences
themselves, if -c is set) paired with mates in
&lt;m2&gt;</p>

<p style="margin-top: 1em">&lt;m2&gt; Comma-separated list
of files containing downstream mates (or the sequences
themselves if -c is set) paired with mates in &lt;m1&gt;</p>

<p style="margin-top: 1em">&lt;r&gt; Comma-separated list
of files containing Crossbow-style reads. Can be a mixture
of paired and unpaired. Specify &quot;-&quot; for stdin.</p>

<p style="margin-top: 1em">&lt;s&gt; Comma-separated list
of files containing unpaired reads, or the sequences
themselves, if -c is set. Specify &quot;-&quot; for
stdin.</p>

<p style="margin-top: 1em">&lt;hit&gt; File to write hits
to (default: stdout)</p>

<p style="margin-top: 1em">Input: <br>
-q query input files are FASTQ .fq/.fastq (default)</p>

<p style="margin-top: 1em">-f query input files are
(multi-)FASTA .fa/.mfa</p>

<p style="margin-top: 1em">-r query input files are raw
one-sequence-per-line</p>

<p style="margin-top: 1em">-c query sequences given on cmd
line (as &lt;mates&gt;, &lt;singles&gt;)</p>

<p style="margin-top: 1em">-C reads and index are in
colorspace</p>

<p style="margin-top: 1em">-Q/--quals &lt;file&gt; <br>
QV file(s) corresponding to CSFASTA inputs; use with -f
-C</p>

<p style="margin-top: 1em">--Q1/--Q2 &lt;file&gt; <br>
same as -Q, but for mate files 1 and 2 respectively</p>

<p style="margin-top: 1em">-s/--skip &lt;int&gt; <br>
skip the first &lt;int&gt; reads/pairs in the input</p>

<p style="margin-top: 1em">-u/--qupto &lt;int&gt; <br>
stop after first &lt;int&gt; reads/pairs (excl. skipped
reads)</p>

<p style="margin-top: 1em">-5/--trim5 &lt;int&gt; <br>
trim &lt;int&gt; bases from 5&rsquo; (left) end of reads</p>

<p style="margin-top: 1em">-3/--trim3 &lt;int&gt; <br>
trim &lt;int&gt; bases from 3&rsquo; (right) end of
reads</p>

<p style="margin-top: 1em">--phred33-quals <br>
input quals are Phred+33 (default)</p>

<p style="margin-top: 1em">--phred64-quals <br>
input quals are Phred+64 (same as --solexa1.3-quals)</p>

<p style="margin-top: 1em">--solexa-quals <br>
input quals are from GA Pipeline ver. &lt; 1.3</p>

<p style="margin-top: 1em">--solexa1.3-quals <br>
input quals are from GA Pipeline ver. &gt;= 1.3</p>

<p style="margin-top: 1em">--integer-quals <br>
qualities are given as space-separated integers (not
ASCII)</p>

<p style="margin-top: 1em">--large-index <br>
force usage of a &rsquo;large&rsquo; index, even if a small
one is present</p>

<p style="margin-top: 1em">Alignment: <br>
-v &lt;int&gt; <br>
report end-to-end hits w/ &lt;=v mismatches; ignore
qualities</p>

<p style="margin-top: 1em">or</p>

<p style="margin-top: 1em">-n/--seedmms &lt;int&gt; max
mismatches in seed (can be 0-3, default: -n 2)</p>

<p style="margin-top: 1em">-e/--maqerr &lt;int&gt; <br>
max sum of mismatch quals across alignment for -n (def:
70)</p>

<p style="margin-top: 1em">-l/--seedlen &lt;int&gt; seed
length for -n (default: 28)</p>

<p style="margin-top: 1em">--nomaqround <br>
disable Maq-like quality rounding for -n (nearest 10 &lt;=
30)</p>

<p style="margin-top: 1em">-I/--minins &lt;int&gt; <br>
minimum insert size for paired-end alignment (default:
0)</p>

<p style="margin-top: 1em">-X/--maxins &lt;int&gt; <br>
maximum insert size for paired-end alignment (default:
250)</p>

<p style="margin-top: 1em">--fr/--rf/--ff -1, -2 mates
align fw/rev, rev/fw, fw/fw (default: --fr)</p>

<p style="margin-top: 1em">--nofw/--norc <br>
do not align to forward/reverse-complement reference
strand</p>

<p style="margin-top: 1em">--maxbts &lt;int&gt; <br>
max # backtracks for -n 2/3 (default: 125, 800 for
--best)</p>

<p style="margin-top: 1em">--pairtries &lt;int&gt; <br>
max # attempts to find mate for anchor hit (default:
100)</p>

<p style="margin-top: 1em">-y/--tryhard <br>
try hard to find valid alignments, at the expense of
speed</p>

<p style="margin-top: 1em">--chunkmbs &lt;int&gt; <br>
max megabytes of RAM for best-first search frames (def:
64)</p>

<p style="margin-top: 1em">Reporting: <br>
-k &lt;int&gt; <br>
report up to &lt;int&gt; good alignments per read (default:
1)</p>

<p style="margin-top: 1em">-a/--all <br>
report all alignments per read (much slower than low -k)</p>

<p style="margin-top: 1em">-m &lt;int&gt; <br>
suppress all alignments if &gt; &lt;int&gt; exist (def: no
limit)</p>

<p style="margin-top: 1em">-M &lt;int&gt; <br>
like -m, but reports 1 random hit (MAPQ=0); requires
--best</p>

<p style="margin-top: 1em">--best hits guaranteed best
stratum; ties broken by quality</p>

<p style="margin-top: 1em">--strata <br>
hits in sub-optimal strata aren&rsquo;t reported (requires
--best)</p>

<p style="margin-top: 1em">Output: <br>
-t/--time <br>
print wall-clock time taken by search phases</p>

<p style="margin-top: 1em">-B/--offbase &lt;int&gt;
leftmost ref offset = &lt;int&gt; in bowtie output (default:
0)</p>

<p style="margin-top: 1em">--quiet <br>
print nothing but the alignments</p>

<p style="margin-top: 1em">--refout <br>
write alignments to files refXXXXX.map, 1 map per
reference</p>

<p style="margin-top: 1em">--refidx <br>
refer to ref. seqs by 0-based index rather than name</p>

<p style="margin-top: 1em">--al &lt;fname&gt; <br>
write aligned reads/pairs to file(s) &lt;fname&gt;</p>

<p style="margin-top: 1em">--un &lt;fname&gt; <br>
write unaligned reads/pairs to file(s) &lt;fname&gt;</p>

<p style="margin-top: 1em">--max &lt;fname&gt; <br>
write reads/pairs over -m limit to file(s) &lt;fname&gt;</p>

<p style="margin-top: 1em">--suppress &lt;cols&gt; <br>
suppresses given columns (comma-delim&rsquo;ed) in default
output</p>

<p style="margin-top: 1em">--fullref <br>
write entire ref name (default: only up to 1st space)</p>

<p style="margin-top: 1em">Colorspace: <br>
--snpphred &lt;int&gt; <br>
Phred penalty for SNP when decoding colorspace (def: 30)</p>

<p style="margin-top: 1em">or</p>

<p style="margin-top: 1em">--snpfrac &lt;dec&gt; <br>
approx. fraction of SNP bases (e.g. 0.001); sets
--snpphred</p>

<p style="margin-top: 1em">--col-cseq <br>
print aligned colorspace seqs as colors, not decoded
bases</p>

<p style="margin-top: 1em">--col-cqual <br>
print original colorspace quals, not decoded quals</p>

<p style="margin-top: 1em">--col-keepends <br>
keep nucleotides at extreme ends of decoded alignment</p>

<p style="margin-top: 1em">SAM: <br>
-S/--sam <br>
write hits in SAM format</p>

<p style="margin-top: 1em">--mapq &lt;int&gt; <br>
default mapping quality (MAPQ) to print for SAM
alignments</p>

<p style="margin-top: 1em">--sam-nohead <br>
supppress header lines (starting with @) for SAM output</p>

<p style="margin-top: 1em">--sam-nosq <br>
supppress @SQ header lines for SAM output</p>

<p style="margin-top: 1em">--sam-RG &lt;text&gt; <br>
add &lt;text&gt; (usually &quot;lab=value&quot;) to @RG line
of SAM header</p>

<p style="margin-top: 1em">Performance: <br>
-o/--offrate &lt;int&gt; override offrate of index; must be
&gt;= index&rsquo;s offrate</p>

<p style="margin-top: 1em">-p/--threads &lt;int&gt; number
of alignment threads to launch (default: 1)</p>

<p style="margin-top: 1em">--mm use memory-mapped I/O for
index; many &rsquo;bowtie&rsquo;s can share</p>

<p style="margin-top: 1em">--shmem <br>
use shared mem for index; many &rsquo;bowtie&rsquo;s can
share</p>

<p style="margin-top: 1em">Other: <br>
--seed &lt;int&gt; <br>
seed for random number generator</p>

<p style="margin-top: 1em">--verbose <br>
verbose output (for debugging)</p>

<p style="margin-top: 1em">--version <br>
print version information and quit</p>

<p style="margin-top: 1em">-h/--help <br>
print this usage message</p>

<p style="margin-top: 1em">64-bit Built on
Debian-reproducible Fri, 14 Oct 2016 20:53:17 +0200
Compiler: gcc version 6.2.1 20161215 Options: -O3
-Wl,--hash-style=both -Wdate-time -D_FORTIFY_SOURCE=2 -g
<br>
-O2 -fdebug-prefix-map=.=. -fPIE -fstack-protector-strong
-Wformat -Werror=format-security -g -O2
-fdebug-prefix-map=.=. -fPIE -fstack-protector-strong
-Wformat -Werror=for&acirc; <br>
mat-security -Wl,-Bsymbolic-functions -fPIE -pie
-Wl,-z,relro -Wl,-z,now Sizeof {int, long, long long, void*,
size_t, off_t}: {4, 8, 8, 8, 8, 8}</p>

<p style="margin-top: 1em">bowtie version 1.1.2 October
2016 BOWTIE(1)</p>
<hr>
</body>
</html>
