<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:55:53 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BOXSHADE(1) Boxshade Manual BOXSHADE(1)</p>

<p style="margin-top: 1em">NAME <br>
boxshade - Pretty-printing of multiple sequence
alignments</p>

<p style="margin-top: 1em">SYNOPSIS <br>
boxshade</p>

<p style="margin-top: 1em">DESCRIPTION <br>
BOXSHADE is a program for pretty-printing multiple alignment
output. The program itself doesn&rsquo;t do any alignment,
you have to use a multiple alignment program like ClustalW
or <br>
Pileup and use the output of these programs as input for
BOXSHADE.</p>

<p style="margin-top: 1em">-help <br>
Show the help.</p>

<p style="margin-top: 1em">-check <br>
Show the help and extend command line.</p>

<p style="margin-top: 1em">-def <br>
Use defaults, no unnecessary questions.</p>

<p style="margin-top: 1em">-numdef <br>
Use default numbering.</p>

<p style="margin-top: 1em">-dna <br>
Assume DNA sequences, use box_dna.par.</p>

<p style="margin-top: 1em">-split <br>
Create separate files for multiple pages.</p>

<p style="margin-top: 1em">-toseq=xxx <br>
Shading according to sequence No. xxx.</p>

<p style="margin-top: 1em">-in=xxxxx</p>

<p style="margin-top: 1em">xxxxx is input file name.</p>

<p style="margin-top: 1em">-out=xxxxx</p>

<p style="margin-top: 1em">xxxxx is output file name.</p>

<p style="margin-top: 1em">-par=xxxxx</p>

<p style="margin-top: 1em">xxxxx is parameter file
name.</p>

<p style="margin-top: 1em">-sim=xxxxx</p>

<p style="margin-top: 1em">xxxxx is file name for similar
residues def.</p>

<p style="margin-top: 1em">-grp=xxxxx</p>

<p style="margin-top: 1em">xxxxx is file name for grouping
residues def.</p>

<p style="margin-top: 1em">-thr=x</p>

<p style="margin-top: 1em">x is the fraction of sequences
that must agree for a consensus.</p>

<p style="margin-top: 1em">-dev=x</p>

<p style="margin-top: 1em">x is output device class (see
below).</p>

<p style="margin-top: 1em">-type=x</p>

<p style="margin-top: 1em">x is input file format (see
below).</p>

<p style="margin-top: 1em">-ruler <br>
Print ruler line.</p>

<p style="margin-top: 1em">-cons <br>
Create consensus line.</p>

<p style="margin-top: 1em">-symbcons=xyz</p>

<p style="margin-top: 1em">xyz are consensus symbols.</p>

<p style="margin-top: 1em">-symbcons=&quot;xyz&quot; <br>
If the one above does not work, try this one.</p>

<p style="margin-top: 1em">-unix <br>
Output files lines are terminated with LF only.</p>

<p style="margin-top: 1em">-mac <br>
Output files lines are terminated with CR only.</p>

<p style="margin-top: 1em">-dos <br>
Output files lines are terminated with CRLF.</p>

<p style="margin-top: 1em">This manual page was written for
the Debian(TM) distribution because the original program
does not have a manual page. The presented information comes
from the documentation of <br>
the Web Service of the 3.21 version that is not available as
a Debian package.</p>

<p style="margin-top: 1em">BOXSHADE is a program for
creating good looking printouts from multiple-aligned
protein or DNA sequences. The program does no alignment by
itself, it has to take as input a file <br>
preprocessed by a multiple alignment program or a multiple
file editor. See below for a list of supported input formats
and output devices. In the standard BOXSHADE output, <br>
identical and similar residues in the multiple-alignment
chart are represented by different colors or shadings. There
are some more options concerning the kind of shading to be
<br>
applied, sequence numbering, consensus output and so on. The
user interface is a bit clumsy at the moment, one has to
answer a lot of questions in order to get the desired <br>
output. There is, however, the possibility to use default
parameters from a standard parameter file or to supply the
program with parameters from the command line. At the
moment, <br>
the VMS and DOS versions of BOXSHADE have identical user
interfaces.</p>

<p style="margin-top: 1em">Input formats <br>
BOXSHADE 3.2 knows about the following input file formats:
(some of the are generally used only for MSDOS or VMS
systems) + CLUSTAL and CLUSTALV, multiple alignment program,
<br>
DOS/VMS/MAC default extension .ALN + ESEE, multiple sequence
editor, DOS default extension .ESE + PHYLIP, phylogenetic
analysis package, DOS, VMS, UNIX default extension .PHY +
<br>
PILEUP and PRETTY of the GCG sequence analysis package
VMS/UNIX default extensions .MSF and .PRE NB!! you are
strongly encouraged NOT to use the PRETTY format as input,
it may be <br>
incompatible with the revised version of .MSF input. We
can&rsquo;t actually think why anyone would use this format
now, .MSF files are more useful generally. + MALIGNED,
multiple <br>
sequence editor, VMS only default extension .MAL BOXSHADE
tries to determine the file type from the extension but will
work also if different extensions are used.</p>

<p style="margin-top: 1em">Output devices <br>
POSTSCRIPT/EPS creates POSTSCRIPT(TM) files for printing on
a Laserprinter or for further conversion with a POSTSCRIPT
interpreter (like GHOSTSCRIPT) + HPGL for export to various
<br>
graphics programs or for conversion/printing with the
shareware program PRINTGL. Plotting BOXSHADE output on a
plotter is generally not recommended + RTF for export to
various <br>
word-processing and graphics programs + CRT, uses direct
screen writes to the PC-monitor. Possible options depend on
the graphics adapter used. This output device is supported
<br>
only in the MSDOS version. + ANSI. On a PC, this option uses
an ANSI device driver (ANSI.SYS) that has to be loaded in
CONFIG.SYS previously. Possible character renditions are
<br>
reverse, bold,underlined, blinking etc. On non-DOS systems,
this option behaves more or less like the VT100 output mode.
+ VT100 for display on a VT100 compatible terminal or <br>
emulator. + ReGISterm for display on a ReGIS compatible
graphics terminal or emulator. + ReGISfile for later
conversion by the program RETOS (copyright DEC) in order to
print on <br>
DIGITALs printer series. + LJ250 for printing on DIGITALS
LJ250 color printer. + ASCII output showing either the
conserved residues or the varying ones (others as
&rsquo;-&rsquo;). + FIG <br>
file for xfig 2.1. + PICT files for import to Mac and PC
graphics progs. Some of the formats above offer the
possibility of scaling the characters and of rotating the
plot. <br>
Character size has to be entered in &rsquo;point&rsquo;
units. Normal output orientation is in portrait mode
(PS/EPS/HPGL/PICT only), to obtain output in landscape
orientation, &rsquo;rotate plot <br>
= y&rsquo; has to be chosen. When creating multi-page
output, all pages are contained in a single output file. If
one page per file is desired, one has to use the command
line <br>
parameter /SPLIT. This is enforced when requesting EPSF or
PICT file output, as multi-page EPSFs are a contradiction of
the purpose of an EPSF and large PICT files would probably
<br>
be too big for most personal computers. While using the
terminal as output device, the &rsquo;RETURN&rsquo; key has
to be pressed to obtain the next page of output.</p>

<p style="margin-top: 1em">Sequence numbering <br>
Starting with version 2.2 there is the possibility to add
numbering to the output files. The numbers are printed
between the sequence names and the sequence itself. Since
most of <br>
the input-files either use no numbering or number the first
position in the alignment always with a &quot;1&quot; (and
that does not necessarily reflect the numbers within the
original <br>
sequence), the user is asked to enter the starting position
for each sequence. The command line flag /DEFNUM suppressed
that question, a starting position of 1 is assumed for all
<br>
sequences. Boxshade starts with the value entered for the
leftmost position and continues numbering every valid
symbol, skipping blanks, &rsquo;-&rsquo;,&rsquo;.&rsquo; and
stuff like that.</p>

<p style="margin-top: 1em">Default parameters <br>
Several people using previous releases of BOXSHADE pointed
me to the need of having default parameters for the various
questions asked by the program. They argued that most sites
<br>
only use one type of input files, one output device and one
choice of colors for the output. I therefore added a
management of default parameters allowing two levels of <br>
assistance to the user. 1) all default parameters are
contained in an ASCII file that can be modified easily to
accommodate the users taste. The format is roughly
documented <br>
within the file-header, it resembles the keyboard input one
has to make if using the program interactively. There are
two such files supplied with this release of BOXSHADE, <br>
BOX_DNA.PAR and BOX_PEP.PAR , holding some example
parameters for peptide and dna-comparisons. there are no big
differences between these two, the major one is that when
shading <br>
DNA-comparisons one doesn&rsquo;t care of
&quot;similar&quot; residues. 2) to run the program with
minimal user interaction, I have added the possibility to
use command line parameters. At the <br>
moment, you can use: /check : list all allowed command line
parameters (this list) and allows parameters to be added.
/def : program runs without questions, BOX_PEP.PAR is used
<br>
as default /dna : makes the program use BOX_DNA.PAR as
parameter file /pep : makes the program use BOX_PEP.PAR as
parameter file /in=xxx : makes the program take xxx as input
<br>
file /out=yyy : makes the program take yyy as output file
(note1) /par=zzz : makes the program use zzz as a default
parameter file /type=1 : makes the program assume an input
<br>
file of type 1 (PRETTY/MSF) /dev=1 : makes the program
assume and output device of type 1 (CRT) /numdef : use
default numbering (all sequences starting with
&quot;1&quot;) /thr : threshold <br>
fraction of residues that must agree for a consensus /split
: forces one page per file output, creates multiple output
files. /cons : makes the program create an additional <br>
consensus line (see below) /symbcons=: influences the way
the consensus line is displayed. (see below) /unix : writes
output files in unix style (LF only) (note2) /dos : writes
<br>
output files in DOS style (CR/LF) (note2) note1: on unix
machines, use out=OUTPUT for terminal output on DOS
machines, use out=con: on VMS machines, use out=tt: note2:
if no mode <br>
is specified, the native style of the machine is used.</p>

<p style="margin-top: 1em">ATTENTION <br>
on unix systems, the dash (-) instead of the slash (/) has
to be used as separation character for command line
parameters. For example, a valid unix command line is:
boxshade <br>
-def -numdef -cons -symbcons=&quot; .*&quot;</p>

<p style="margin-top: 1em">Shading strategies (similarity
to consensus or single sequence) <br>
Starting with version 3, BOXSHADE has a new shading system.
The first difference is the introduction of a threshold
fraction of residues that must agree for there to be a <br>
consensus. Previously, the program assumed that SOME residue
was always the consensus. If no two residues were the same,
the first sequence provided the consensus residue. This <br>
threshold fraction can be any number between 0.0 and 1.0.
The number of sequences that must agree for there to be a
consensus is, as you might expect, this fraction times the
<br>
total number of sequences in the alignment (fractions of a
sequence count as one, e.g. 3.2 becomes 4). The second
difference is the idea of &rsquo;consensus by
similarity&rsquo;; this tries <br>
to take account of the situations where all the sequences
may have (for example) R or K at a position, but neither in
a majority. It would not be logical to shade one type of
<br>
residue as &rsquo;identical&rsquo; and the other as
&rsquo;similar&rsquo;; the threshold function might also
eliminate both as being in too small numbers. Therefore, if
there is not a single residue <br>
that is conserved (greater than the threshold) at a
position, the program looks for a &rsquo;group&rsquo; of
amino acids that fulfills the requirements.
&rsquo;Groups&rsquo; are defined in the .grp <br>
files. Users can tailor these to their personal prejudices.
Any amino acid not listed is assumed not to be in a group.
All members of a group are considered to be mutually <br>
similar, unlike the .sim files, described below. If
consensus by similarity is found, all the residues in the
consensus are shaded using the &rsquo;similar&rsquo; shading
defined by the <br>
user. If the user does not select &rsquo;shading by
similarity&rsquo;, only identity-type consensus is looked
at. If an identity-type consensus is found, and similarity
shading is in <br>
operation, the program looks to see if the remaining
residues are similar to the consensus residue. Here the
box_xxx.sim files are used. The main difference between
relationships <br>
in these files and those in the .grp files is that, e.g. in
a .grp file the line STA means that all three a.a.s are
mutually similar. In a .sim file S TA means that both T and
A <br>
are considered similar to S, where there is a conserved S
residue in more than threshold number of sequences. However,
it does NOT mean that T and A are similar to each other.
<br>
Note that cases where two residues, or groups of residues,
fulfill the threshold requirements (as could happen with
values of the thr. fraction less than or equal to 0.5) are
<br>
treated as having no consensus. This describes the main
shading model &rsquo;shading according to a
consensus&rsquo;. The alternative model is called
&rsquo;shading according to a master <br>
sequence&rsquo;. In this case the user is prompted for a
sequence of the alignment and consecutively that sequence is
taken to be the &rsquo;consensus&rsquo;. Only those residues
become shaded <br>
that are identical or similar to the chosen sequence. Output
obtained with this option tends to be less shaded and
neglects similarities between the other (non-chosen)
sequences. <br>
Starting in V2.7, this &rsquo;master sequence&rsquo; can be
hidden. Thus, it only influences the shading of the other
sequences without being shown itself.</p>

<p style="margin-top: 1em">Consensus display <br>
Starting with version 2.5, BOXSHADE offers the possibility
to create an additional line holding a consensus symbol.
This line can either be obtained by using the command line
<br>
qualifier /CONS or interactively by answering the question
&rsquo; create consensus? &rsquo;. The way this consensus
line is displayed can be modified by the command line
parameter <br>
SYMBCONS=xyz, by editing the respective entry in the .PAR
file or interactively. Since the SYMBCONS syntax is not
intuitive, here a brief description: The SYMBCONS parameter
<br>
consist of exactly three symbols: + the first one stands for
&rsquo;normal&rsquo; sequence residues that are not involved
in any similar/identical relationship. + the second symbol
<br>
represents positions that are similar in all sequences of
the alignment. See the files BOX_PEP.SIM and BOX_DNA.SIM to
see what residues are considered similar. + the third symbol
<br>
represents positions that are identical in all sequences of
the alignment. A SYMBCONS parameter string &quot; .*&quot;
(blank/point/asterisk) means: label all positions in the
alignment <br>
with totally identical residues by an asterisk, all
positions with all similar residues by a point and do not
mark the other positions. The letter &rsquo;B&rsquo; can be
used instead of the <br>
blank, this is necessary e.g. when using the command line
option /SYMBCONS=B.* which gives the same result as the
above example. The option /SYMBCONS= .* would result in an
<br>
unexpected behaviour because MSDOS squeezes blanks out of
the command line. Besides points, asterisks and other
symbols, there are two special characters when they appear
in the <br>
SYMBCONS string: &rsquo;L&rsquo; and &rsquo;U&rsquo;. An
&rsquo;L&rsquo; means, that a lowercase representation of
the most abundant residue at that position is to be used
instead of a fixed consensus symbol while <br>
an &rsquo;U&rsquo; means an uppercase character
representation of that residue. A possible application would
be the SYMBCONS string &quot; LU&quot; where similar
residues are represented by lowercase <br>
characters and identical by uppercase characters.</p>

<p style="margin-top: 1em">Shareware/PD programs useful in
conjunction with BOXSHADE <br>
multiple alignment files that to be used by BOXSHADE can be
created, amongst others, by the following PD/freeware
programs: + PHYLIP by Joe Felsenstein, available by ftp from
<br>
anthro.utah.edu + ESEE by Eric Cabot, available from the
same sources as BOXSHADE (see above) + CLUSTAL by Des
Higgins, ditto for preview/conversion of POSTSCRIPT files,
the <br>
program GHOSTSCRIPT from GNU software foundation is highly
recommended. It is available from all major MSDOS ftp-sites
(e.g. SIMTEL or ftp.uni-koeln.de) There is also a version
<br>
tested for use with boxshade available at
vax0.biomed.uni-koeln.de although this might be not the most
recent release. for Mac users, there is MacGhostscript, also
available from <br>
the main archives (info-mac, umich and their mirrors). A
*very* good tool for putting a preview image into an EPSF
file, often a prerequisite for incorporating into a drawing
<br>
package, is PS2EPS, by Peter Lerup. This can be found on
info-mac. for preview/conversion of HPGL files, the
shareware program PRINTGL 1.18 by Cary Ravitz is highly
recommended. <br>
It is available from many MSDOS ftp sites and from
netserv@embl-heidelberg.de - output on dot printers - Since
PRINTGL offers a broad choice of printer types and is a nice
<br>
program, I recommend its use for printing BOXSHADE output on
non-POSTSCRIPT printers. Use HPGL output with options 0F1N
for normal residues 2F1N for identical residues 3F1N for
<br>
similar residues 2F4N for conserved residues 8 for character
size not rotated (these are the standard parameters in
BOX_PEP.PAR) for creating a HPGL files. (lets call it <br>
TEST.PLT) Now use PRINTGL either interactively by calling
PMI or use a command line like: PRINTGL
/Fx/S0340/Waaac/Ptest.plt where test.plt is to be replaced
by the filename to <br>
convert and the x in the expression /Fx is to be replaced by
the letter of the printer you use. (See the PRINTGL
documentation for further details)</p>

<p style="margin-top: 1em">RESTRICTIONS <br>
The RTF output and PHYLIP input implementations are still
experimental. Please tell me of your experiences with the
program. + the current DOS version supports only 13
sequences <br>
with 2000 residues each. This parameters can be easily
changed in the source code. If you cannot compile the
sources because you are lacking a pascal compiler, contact
the author <br>
for precompiled versions</p>

<p style="margin-top: 1em">CITING BOXSHADE <br>
There is no publication on BOXSHADE and none is planned.
Most people just use it for figures in publications and
don&rsquo;t mention anything, this is ok for the authors of
BOXSHADE. <br>
If you really feel like mentioning BOXSHADE, you could
either acknowledge it in the figure legend or in the
Mat&amp;Meth part on sequence analysis.</p>

<p style="margin-top: 1em">SEE ALSO <br>
/etc/boxshade/*.par</p>

<p style="margin-top: 1em">seaview(1) kalign(1)</p>

<p style="margin-top: 1em">AUTHORS <br>
Kay Hofmann &lt;kay.hofmann@memorec.com&gt; <br>
ISREC, Bioinformatics Group, <br>
CH-1066 <br>
Epalinges s/Lausanne Switzerland</p>

<p style="margin-top: 1em">Wrote Boxshade.</p>

<p style="margin-top: 1em">Michael Baron
&lt;michael.baron@bbsrc.ac.uk&gt; <br>
BBSRC Institute for Animal Health, <br>
Pirbright, <br>
Surrey <br>
GU24 0NF <br>
U.K.</p>

<p style="margin-top: 1em">Wrote Boxshade.</p>

<p style="margin-top: 1em">Harmut Schirmer
&lt;hsc@techfak.uni-kiel.de&gt; <br>
Technische Fakultaet, <br>
Kaiserstr. 2 <br>
D-24143 <br>
Kiel <br>
Germany</p>

<p style="margin-top: 1em">C port of Boxshade. (don&rsquo;t
send Kay or Michael any questions concerning the
&rsquo;C&rsquo; version of boxshade)</p>

<p style="margin-top: 1em">Steffen M&Atilde;&para;ller
&lt;moeller@debian.org&gt; <br>
Wrote the manpage.</p>

<p style="margin-top: 1em">Charles Plessy
&lt;plessy@debian.org&gt; <br>
Updated the manpage</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 1997 Kay Hofmann, Michael Baron and
Harmut Schirmer <br>
Copyright &Acirc;&copy; 2003, 2007 Steffen Moeller, Charles
Plessy</p>

<p style="margin-top: 1em">The above copyright notices
refer to the program and its manpage respectively.</p>

<p style="margin-top: 1em">BOXSHADE is completely
public-domain and may be passed around and modified without
any notice to the authors.</p>

<p style="margin-top: 1em">This manual page was written for
the Debian(TM) system but may be used by others. Permission
is granted to copy, distribute and/or modify this document
under same terms as <br>
boxshade itself.</p>

<p style="margin-top: 1em">boxshade 3.3.1 February 09, 2013
BOXSHADE(1)</p>
<hr>
</body>
</html>
