<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_BULK_LOAD_GFF(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_BULK_LOAD_GFF(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_BULK_LOAD_GFF(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_bulk_load_gff.pl - Bulk-load a Bio::DB::GFF database from GFF files.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  % bp_bulk_load_gff.pl -d testdb dna1.fa dna2.fa features1.gff features2.gff ...
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script loads a Bio::DB::GFF database with the features contained in a list
  of GFF files and/or FASTA sequence files. You must use the exact variant of
  GFF described in Bio::DB::GFF. Various command-line options allow you to
  control which database to load and whether to allow an existing database to be
  overwritten.
<div class="Pp"></div>
This script differs from bp_load_gff.pl in that it is hard-coded to use MySQL
  and cannot perform incremental loads. See bp_load_gff.pl for an incremental
  loader that works with all databases supported by Bio::DB::GFF, and
  bp_fast_load_gff.pl for a MySQL loader that supports fast incremental loads.
<h2 class="Ss" title="Ss" id="NOTES"><a class="selflink" href="#NOTES">NOTES</a></h2>
If the filename is given as &quot;-&quot; then the input is taken from standard
  input. Compressed files (.gz, .Z, .bz2) are automatically uncompressed.
<div class="Pp"></div>
FASTA format files are distinguished from GFF files by their filename
  extensions. Files ending in .fa, .fasta, .fast, .seq, .dna and their uppercase
  variants are treated as FASTA files. Everything else is treated as a GFF file.
  If you wish to load -fasta files from STDIN, then use the -f command-line
  swith with an argument of '-', as in
<div class="Pp"></div>
<pre>
    gunzip my_data.fa.gz | bp_fast_load_gff.pl -d test -f -
</pre>
<div class="Pp"></div>
The nature of the bulk load requires that the database be on the local machine
  and that the indicated user have the &quot;file&quot; privilege to load the
  tables and have enough room in /usr/tmp (or whatever is specified by the
  \$TMPDIR environment variable), to hold the tables transiently.
<div class="Pp"></div>
Local data may now be uploaded to a remote server via the --local option with
  the database host specified in the dsn, e.g. dbi:mysql:test:db_host
<div class="Pp"></div>
The adaptor used is dbi::mysqlopt. There is currently no way to change this.
<div class="Pp"></div>
About maxfeature: the default value is 100,000,000 bases. If you have features
  that are close to or greater that 100Mb in length, then the value of
  maxfeature should be increased to 1,000,000,000. This value must be a power of
  10.
<div class="Pp"></div>
Note that Windows users must use the --create option.
<div class="Pp"></div>
If the list of GFF or fasta files exceeds the kernel limit for the maximum
  number of command-line arguments, use the --long_list /path/to/files option.
<h1 class="Sh" title="Sh" id="COMMAND-LINE_OPTIONS"><a class="selflink" href="#COMMAND-LINE_OPTIONS">COMMAND-LINE
  OPTIONS</a></h1>
Command-line options can be abbreviated to single-letter options. e.g. -d
  instead of --database.
<div class="Pp"></div>
<pre>
   --database &lt;dsn&gt;      Database name (default dbi:mysql:test)
   --adaptor             Adaptor name (default mysql)
   --create              Reinitialize/create data tables without asking
   --user                Username to log in as
   --fasta               File or directory containing fasta files to load
   --long_list           Directory containing a very large number of 
                         GFF and/or FASTA files
   --password            Password to use for authentication
                           (Does not work with Postgres, password must be
                           supplied interactively or be left empty for
                           ident authentication)
   --maxbin              Set the value of the maximum bin size
   --local               Flag to indicate that the data source is local
   --maxfeature          Set the value of the maximum feature size (power of 10)
   --group               A list of one or more tag names (comma or space separated)
                         to be used for grouping in the 9th column.
   --gff3_munge          Activate GFF3 name munging (see Bio::DB::GFF)
   --summary             Generate summary statistics for drawing coverage histograms.
                           This can be run on a previously loaded database or during
                           the load.
   --Temporary           Location of a writable scratch directory
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Bio::DB::GFF, fast_load_gff.pl, load_gff.pl
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Lincoln Stein, lstein@cshl.org
<div class="Pp"></div>
Copyright (c) 2002 Cold Spring Harbor Laboratory
<div class="Pp"></div>
This library is free software; you can redistribute it and/or modify it under
  the same terms as Perl itself. See DISCLAIMER.txt for disclaimers of
  warranty.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
