<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:55:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_BULK_LOAD_GFF(1p) User Contributed Perl Documentation
BP_BULK_LOAD_GFF(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_bulk_load_gff.pl - Bulk-load a Bio::DB::GFF database from
GFF files.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
% bp_bulk_load_gff.pl -d testdb dna1.fa dna2.fa
features1.gff features2.gff ...</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script loads a Bio::DB::GFF database with the features
contained in a list of GFF files and/or FASTA sequence
files. You must use the exact variant of GFF described in
<br>
Bio::DB::GFF. Various command-line options allow you to
control which database to load and whether to allow an
existing database to be overwritten.</p>

<p style="margin-top: 1em">This script differs from
bp_load_gff.pl in that it is hard-coded to use MySQL and
cannot perform incremental loads. See bp_load_gff.pl for an
incremental loader that works with <br>
all databases supported by Bio::DB::GFF, and
bp_fast_load_gff.pl for a MySQL loader that supports fast
incremental loads.</p>

<p style="margin-top: 1em">NOTES <br>
If the filename is given as &quot;-&quot; then the input is
taken from standard input. Compressed files (.gz, .Z, .bz2)
are automatically uncompressed.</p>

<p style="margin-top: 1em">FASTA format files are
distinguished from GFF files by their filename extensions.
Files ending in .fa, .fasta, .fast, .seq, .dna and their
uppercase variants are treated as <br>
FASTA files. Everything else is treated as a GFF file. If
you wish to load -fasta files from STDIN, then use the -f
command-line swith with an argument of &rsquo;-&rsquo;, as
in</p>

<p style="margin-top: 1em">gunzip my_data.fa.gz |
bp_fast_load_gff.pl -d test -f -</p>

<p style="margin-top: 1em">The nature of the bulk load
requires that the database be on the local machine and that
the indicated user have the &quot;file&quot; privilege to
load the tables and have enough room in <br>
/usr/tmp (or whatever is specified by the MPDIR environment
variable), to hold the tables transiently.</p>

<p style="margin-top: 1em">Local data may now be uploaded
to a remote server via the --local option with the database
host specified in the dsn, e.g. dbi:mysql:test:db_host</p>

<p style="margin-top: 1em">The adaptor used is
dbi::mysqlopt. There is currently no way to change this.</p>

<p style="margin-top: 1em">About maxfeature: the default
value is 100,000,000 bases. If you have features that are
close to or greater that 100Mb in length, then the value of
maxfeature should be <br>
increased to 1,000,000,000. This value must be a power of
10.</p>

<p style="margin-top: 1em">Note that Windows users must use
the --create option.</p>

<p style="margin-top: 1em">If the list of GFF or fasta
files exceeds the kernel limit for the maximum number of
command-line arguments, use the --long_list /path/to/files
option.</p>

<p style="margin-top: 1em">COMMAND-LINE OPTIONS <br>
Command-line options can be abbreviated to single-letter
options. e.g. -d instead of --database.</p>

<p style="margin-top: 1em">--database &lt;dsn&gt; Database
name (default dbi:mysql:test) <br>
--adaptor Adaptor name (default mysql) <br>
--create Reinitialize/create data tables without asking <br>
--user Username to log in as <br>
--fasta File or directory containing fasta files to load
<br>
--long_list Directory containing a very large number of <br>
GFF and/or FASTA files <br>
--password Password to use for authentication <br>
(Does not work with Postgres, password must be <br>
supplied interactively or be left empty for <br>
ident authentication) <br>
--maxbin Set the value of the maximum bin size <br>
--local Flag to indicate that the data source is local <br>
--maxfeature Set the value of the maximum feature size
(power of 10) <br>
--group A list of one or more tag names (comma or space
separated) <br>
to be used for grouping in the 9th column. <br>
--gff3_munge Activate GFF3 name munging (see Bio::DB::GFF)
<br>
--summary Generate summary statistics for drawing coverage
histograms. <br>
This can be run on a previously loaded database or during
<br>
the load. <br>
--Temporary Location of a writable scratch directory</p>

<p style="margin-top: 1em">SEE ALSO <br>
Bio::DB::GFF, fast_load_gff.pl, load_gff.pl</p>

<p style="margin-top: 1em">AUTHOR <br>
Lincoln Stein, lstein@cshl.org</p>

<p style="margin-top: 1em">Copyright (c) 2002 Cold Spring
Harbor Laboratory</p>

<p style="margin-top: 1em">This library is free software;
you can redistribute it and/or modify it under the same
terms as Perl itself. See DISCLAIMER.txt for disclaimers of
warranty.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_BULK_LOAD_GFF(1p)</p>
<hr>
</body>
</html>
