<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:55:57 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_FLANKS(1p) User Contributed Perl Documentation
BP_FLANKS(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_flanks - finding flanking sequences for a variant in a
sequence position</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bp_flanks --position POS [-p POS ...] [--flanklen INT] <br>
accession | filename</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script allows you to extract a subsequence around a
region of interest from an existing sequence. The output if
fasta formatted sequence entry where the header line
contains <br>
additional information about the location.</p>

<p style="margin-top: 1em">OPTIONS <br>
The script takes one unnamed argument which be either a file
name in the local file system or a nucleotide sequence
accession number.</p>

<p style="margin-top: 1em">-p Position uses simple
nucleotide sequence feature table <br>
--position notation to define the region of interest,
typically a <br>
SNP or microsatellite repeat around which the flanks are
<br>
defined.</p>

<p style="margin-top: 1em">There can be more than one
position option or you can <br>
give a comma separated list to one position option.</p>

<p style="margin-top: 1em">The format of a position is:</p>

<p style="margin-top: 1em">[id:] int | range | in-between
[-]</p>

<p style="margin-top: 1em">The optional id is the name you
want to call the new <br>
sequence. If it not given in joins running number to the
<br>
entry name with an underscore.</p>

<p style="margin-top: 1em">The position is either a point
(e.g. 234), a range (e.g <br>
250..300) or insertion point between nucleotides <br>
(e.g. 234^235)</p>

<p style="margin-top: 1em">If the position is not
completely within the source <br>
sequence the output sequence will be truncated and it <br>
will print a warning.</p>

<p style="margin-top: 1em">The optional hyphen [-] at the
end of the position <br>
indicates that that you want the retrieved sequence to be
<br>
in the opposite strand.</p>

<p style="margin-top: 1em">-f Defaults to 100. This is the
length of the nucleotides <br>
--flanklen sequence retrieved on both sides of the given
position.</p>

<p style="margin-top: 1em">If the source file does not
contain</p>

<p style="margin-top: 1em">OUTPUT FORMAT <br>
The output is a fasta formatted entry where the description
file contains tag=value pairs for information about where in
the original sequence the subsequence was taken.</p>

<p style="margin-top: 1em">The ID of the fasta entry is the
name given at the command line joined by hyphen to the
filename or accesion of the source sequence. If no id is
given a series of consequtive <br>
integers is used.</p>

<p style="margin-top: 1em">The tag=value pairs are:</p>

<p style="margin-top: 1em">oripos=int <br>
position in the source file</p>

<p style="margin-top: 1em">strand=1|-1 <br>
strand of the sequence compared to the source sequence</p>

<p style="margin-top: 1em">allelepos=int <br>
position of the region of interest in the current entry. The
tag is the same as used by dbSNP@NCBI</p>

<p style="margin-top: 1em">The sequence highlights the
allele variant position by showing it in upper case and rest
of the sequence in lower case characters.</p>

<p style="margin-top: 1em">EXAMPLE <br>
% bp_flanks ~/seq/ar.embl</p>

<p style="margin-top: 1em">&gt;1_/HOME/HEIKKI/SEQ/AR.EMBL
oripos=100 strand=1 allelepos=100 <br>

taataactcagttcttatttgcacctacttcagtggacactgaatttggaaggtggagga
<br>

ttttgtttttttcttttaagatctgggcatcttttgaatCtacccttcaagtattaagag
<br>

acagactgtgagcctagcagggcagatcttgtccaccgtgtgtcttcttctgcacgagac
<br>
tttgaggctgtcagagcgct</p>

<p style="margin-top: 1em">TODO <br>
The input files are assumed to be in EMBL format and the
sequences are retrieved only from the EMB database. Make
this more generic and use the registry.</p>

<p style="margin-top: 1em">head1 FEEDBACK</p>

<p style="margin-top: 1em">Mailing Lists <br>
User feedback is an integral part of the evolution of this
and other Bioperl modules. Send your comments and
suggestions preferably to the Bioperl mailing lists Your
<br>
participation is much appreciated.</p>

<p style="margin-top: 1em">bioperl-l@bioperl.org - General
discussion <br>
http://bioperl.org/wiki/Mailing_lists - About the mailing
lists</p>

<p style="margin-top: 1em">Reporting Bugs <br>
Report bugs to the Bioperl bug tracking system to help us
keep track the bugs and their resolution. Bug reports can be
submitted via the web:</p>


<p style="margin-top: 1em">https://github.com/bioperl/bioperl-live/issues</p>

<p style="margin-top: 1em">AUTHOR - Heikki Lehvaslaiho <br>
Email: &lt;heikki-at-bioperl-dot-org&gt;</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_FLANKS(1p)</p>
<hr>
</body>
</html>
