<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_FLANKS(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_FLANKS(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_FLANKS(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_flanks - finding flanking sequences for a variant in a sequence position
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  bp_flanks --position POS [-p POS ...]  [--flanklen INT]
         accession | filename
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script allows you to extract a subsequence around a region of interest from
  an existing sequence. The output if fasta formatted sequence entry where the
  header line contains additional information about the location.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
The script takes one unnamed argument which be either a file name in the local
  file system or a nucleotide sequence accession number.
<div class="Pp"></div>
<pre>
  -p         Position uses simple nucleotide sequence feature table
  --position notation to define the region of interest, typically a
             SNP or microsatellite repeat around which the flanks are
             defined.
             There can be more than one position option or you can
             give a comma separated list to one position option.
             The format of a position is:
                 [id:] int | range | in-between [-]
             The optional id is the name you want to call the new
             sequence. If it not given in joins running number to the
             entry name with an underscore.
             The position is either a point (e.g. 234), a range (e.g
             250..300) or insertion point between nucleotides
             (e.g. 234^235)
             If the position is not completely within the source
             sequence the output sequence will be truncated and it
             will print a warning.
             The optional hyphen [-] at the end of the position
             indicates that that you want the retrieved sequence to be
             in the opposite strand.
  -f         Defaults to 100. This is the length of the nucleotides
  --flanklen sequence retrieved on both sides of the given position.
             If the source file does not contain
</pre>
<h1 class="Sh" title="Sh" id="OUTPUT_FORMAT"><a class="selflink" href="#OUTPUT_FORMAT">OUTPUT
  FORMAT</a></h1>
The output is a fasta formatted entry where the description file contains
  tag=value pairs for information about where in the original sequence the
  subsequence was taken.
<div class="Pp"></div>
The ID of the fasta entry is the name given at the command line joined by hyphen
  to the filename or accesion of the source sequence. If no id is given a series
  of consequtive integers is used.
<div class="Pp"></div>
The tag=value pairs are:
<dl class="Bl-tag">
  <dt class="It-tag">oripos=int</dt>
  <dd class="It-tag">position in the source file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">strand=1|-1</dt>
  <dd class="It-tag">strand of the sequence compared to the source sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">allelepos=int</dt>
  <dd class="It-tag">position of the region of interest in the current entry.
      The tag is the same as used by dbSNP@NCBI</dd>
</dl>
<div class="Pp"></div>
The sequence highlights the allele variant position by showing it in upper case
  and rest of the sequence in lower case characters.
<h1 class="Sh" title="Sh" id="EXAMPLE"><a class="selflink" href="#EXAMPLE">EXAMPLE</a></h1>
<pre>
  % bp_flanks ~/seq/ar.embl
  &gt;1_/HOME/HEIKKI/SEQ/AR.EMBL oripos=100 strand=1 allelepos=100
  taataactcagttcttatttgcacctacttcagtggacactgaatttggaaggtggagga
  ttttgtttttttcttttaagatctgggcatcttttgaatCtacccttcaagtattaagag
  acagactgtgagcctagcagggcagatcttgtccaccgtgtgtcttcttctgcacgagac
  tttgaggctgtcagagcgct
</pre>
<h1 class="Sh" title="Sh" id="TODO"><a class="selflink" href="#TODO">TODO</a></h1>
The input files are assumed to be in EMBL format and the sequences are retrieved
  only from the EMB database. Make this more generic and use the registry.
<div class="Pp"></div>
head1 FEEDBACK
<h2 class="Ss" title="Ss" id="Mailing_Lists"><a class="selflink" href="#Mailing_Lists">Mailing
  Lists</a></h2>
User feedback is an integral part of the evolution of this and other Bioperl
  modules. Send your comments and suggestions preferably to the Bioperl mailing
  lists Your participation is much appreciated.
<div class="Pp"></div>
<pre>
  bioperl-l@bioperl.org                  - General discussion
  http://bioperl.org/wiki/Mailing_lists  - About the mailing lists
</pre>
<h2 class="Ss" title="Ss" id="Reporting_Bugs"><a class="selflink" href="#Reporting_Bugs">Reporting
  Bugs</a></h2>
Report bugs to the Bioperl bug tracking system to help us keep track the bugs
  and their resolution. Bug reports can be submitted via the web:
<div class="Pp"></div>
<pre>
  https://github.com/bioperl/bioperl-live/issues
</pre>
<h1 class="Sh" title="Sh" id="AUTHOR_-_Heikki_Lehvaslaiho"><a class="selflink" href="#AUTHOR_-_Heikki_Lehvaslaiho">AUTHOR
  - Heikki Lehvaslaiho</a></h1>
Email: &lt;heikki-at-bioperl-dot-org&gt;</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
