<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:55:58 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_GENBANK2GFF3(1p) User Contributed Perl Documentation
BP_GENBANK2GFF3(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_genbank2gff3.pl -- Genbank-&gt;gbrowse-friendly GFF3</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bp_genbank2gff3.pl [options] filename(s)</p>

<p style="margin-top: 1em"># process a directory containing
GenBank flatfiles <br>
perl bp_genbank2gff3.pl --dir path_to_files --zip</p>

<p style="margin-top: 1em"># process a single file, ignore
explicit exons and introns <br>
perl bp_genbank2gff3.pl --filter exon --filter intron
file.gbk.gz</p>

<p style="margin-top: 1em"># process a list of files <br>
perl bp_genbank2gff3.pl *gbk.gz</p>

<p style="margin-top: 1em"># process data from URL, with
Chado GFF model (-noCDS), and pipe to database loader <br>
curl
ftp://ftp.ncbi.nih.gov/genomes/Saccharomyces_cerevisiae/CHR_X/NC_001142.gbk
| perl bp_genbank2gff3.pl -noCDS -in stdin -out stdout |
perl gmod_bulk_load_gff3.pl -dbname mychado -organism
fromdata</p>

<p style="margin-top: 1em">Options: <br>
--noinfer -r don&rsquo;t infer exon/mRNA subfeatures <br>
--conf -i path to the curation configuration file that
contains user preferences <br>
for Genbank entries (must be YAML format) <br>
(if --manual is passed without --ini, user will be prompted
to <br>
create the file if any manual input is saved) <br>
--sofile -l path to to the so.obo file to use for feature
type mapping <br>
(--sofile live will download the latest online revision)
<br>
--manual -m when trying to guess the proper SO term, if more
than <br>
one option matches the primary tag, the converter will <br>
wait for user input to choose the correct one <br>
(only works with --sofile) <br>
--dir -d path to a list of genbank flatfiles <br>
--outdir -o location to write GFF files (can be
&rsquo;stdout&rsquo; or &rsquo;-&rsquo; for pipe) <br>
--zip -z compress GFF3 output files with gzip <br>
--summary -s print a summary of the features in each contig
<br>
--filter -x genbank feature type(s) to ignore <br>
--split -y split output to separate GFF and fasta files for
<br>
each genbank record <br>
--nolump -n separate file for each reference sequence <br>
(default is to lump all records together into one <br>
output file for each input file) <br>
--ethresh -e error threshold for unflattener <br>
set this high (&gt;2) to ignore all unflattener errors <br>
--[no]CDS -c Keep CDS-exons, or convert to alternate
gene-RNA-protein-exon <br>
model. --CDS is default. Use --CDS to keep default GFF gene
model, <br>
use --noCDS to convert to g-r-p-e. <br>
--format -f Input format (SeqIO types): GenBank, Swiss or
Uniprot, EMBL work <br>
(GenBank is default) <br>
--GFF_VERSION 3 is default, 2 and 2.5 and other
Bio::Tools::GFF versions available <br>
--quiet don&rsquo;t talk about what is being processed <br>
--typesource SO sequence type for source (e.g. chromosome;
region; contig) <br>
--help -h display this message</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script uses Bio::SeqFeature::Tools::Unflattener and
Bio::Tools::GFF to convert GenBank flatfiles to GFF3 with
gene containment hierarchies mapped for optimal display in
<br>
gbrowse.</p>

<p style="margin-top: 1em">The input files are assumed to
be gzipped GenBank flatfiles for refseq contigs. The files
may contain multiple GenBank records. Either a single file
or an entire directory can <br>
be processed. By default, the DNA sequence is embedded in
the GFF but it can be saved into separate fasta file with
the --split(-y) option.</p>

<p style="margin-top: 1em">If an input file contains
multiple records, the default behaviour is to dump all GFF
and sequence to a file of the same name (with .gff
appended). Using the &rsquo;nolump&rsquo; option will <br>
create a separate file for each genbank record. Using the
&rsquo;split&rsquo; option will create separate GFF and
Fasta files for each genbank record.</p>

<p style="margin-top: 1em">Notes <br>
&rsquo;split&rsquo; and &rsquo;nolump&rsquo; produce many
files</p>

<p style="margin-top: 1em">In cases where the input files
contain many GenBank records (for example, the chromosome
files for the mouse genome build), a very large number of
output files will be produced <br>
if the &rsquo;split&rsquo; or &rsquo;nolump&rsquo; options
are selected. If you do have lists of files &gt; 6000, use
the --long_list option in bp_bulk_load_gff.pl or
bp_fast_load_gff.pl to load the gff <br>
and/ or fasta files.</p>

<p style="margin-top: 1em">Designed for RefSeq</p>

<p style="margin-top: 1em">This script is designed for
RefSeq genomic sequence entries. It may work for third party
annotations but this has not been tested. But see below,
Uniprot/Swissprot works, EMBL <br>
and possibly EMBL/Ensembl if you don&rsquo;t mind some gene
model unflattener errors (dgg).</p>

<p style="margin-top: 1em">G-R-P-E Gene Model</p>

<p style="margin-top: 1em">Don Gilbert worked this over
with needs to produce GFF3 suited to loading to GMOD Chado
databases. Most of the changes I believe are suited for
general use. One main chado- <br>
specific addition is the <br>
--[no]cds2protein flag</p>

<p style="margin-top: 1em">My favorite GFF is to set the
above as ON by default (disable with --nocds2prot) For
general use it probably should be OFF, enabled with
--cds2prot.</p>

<p style="margin-top: 1em">This writes GFF with an
alternate, but useful Gene model, instead of the consensus
model for GFF3</p>

<p style="margin-top: 1em">[ gene &gt; mRNA&gt;
(exon,CDS,UTR) ]</p>

<p style="margin-top: 1em">This alternate is</p>

<p style="margin-top: 1em">gene &gt; mRNA &gt; polypeptide
&gt; exon</p>

<p style="margin-top: 1em">means the only feature with dna
bases is the exon. The others specify only location ranges
on a genome. Exon of course is a child of mRNA and
protein/peptide.</p>

<p style="margin-top: 1em">The protein/polypeptide feature
is an important one, having all the annotations of the
GenBank CDS feature, protein ID, translation, GO terms,
Dbxrefs to other proteins.</p>

<p style="margin-top: 1em">UTRs, introns, CDS-exons are all
inferred from the primary exon bases inside/outside
appropriate higher feature ranges. Other special gene model
features remain the same.</p>

<p style="margin-top: 1em">Several other improvements and
bugfixes, minor but useful are included</p>

<p style="margin-top: 1em">* IO pipes now work: <br>
curl ftp://ncbigenomes/... | bp_genbank2gff3 --in stdin
--out stdout | gff2chado ...</p>

<p style="margin-top: 1em">* GenBank main record fields are
added to source feature, e.g. organism, date, <br>
and the sourcetype, commonly chromosome for genomes, is
used.</p>

<p style="margin-top: 1em">* Gene Model handling for ncRNA,
pseudogenes are added.</p>

<p style="margin-top: 1em">* GFF header is cleaner, more
informative. <br>
--GFF_VERSION flag allows choice of v2 as well as default
v3</p>

<p style="margin-top: 1em">* GFF ##FASTA inclusion is
improved, and <br>
CDS translation sequence is moved to FASTA records.</p>

<p style="margin-top: 1em">* FT -&gt; GFF attribute mapping
is improved.</p>

<p style="margin-top: 1em">* --format choice of SeqIO input
formats (GenBank default). <br>
Uniprot/Swissprot and EMBL work and produce useful GFF.</p>

<p style="margin-top: 1em">* SeqFeature::Tools::TypeMapper
has a few FT -&gt; SOFA additions <br>
and more flexible usage.</p>

<p style="margin-top: 1em">TODO <br>
Are these additions desired? <br>
* filter input records by taxon (e.g. keep only organism=xxx
or taxa level = classYYY <br>
* handle Entrezgene, other non-sequence SeqIO structures
(really should change <br>
those parsers to produce consistent annotation tags).</p>

<p style="margin-top: 1em">Related bugfixes/tests <br>
These items from Bioperl mail were tested (sample data
generating errors), and found corrected:</p>

<p style="margin-top: 1em">From: Ed Green &lt;green
&lt;at&gt; eva.mpg.de&gt; <br>
Subject: genbank2gff3.pl on new human RefSeq <br>
Date: 2006-03-13 21:22:26 GMT <br>
-- unspecified errors (sample data works now).</p>

<p style="margin-top: 1em">From: Eric Just &lt;e-just
&lt;at&gt; northwestern.edu&gt; <br>
Subject: genbank2gff3.pl <br>
Date: 2007-01-26 17:08:49 GMT <br>
-- bug fixed in genbank2gff3 for multi-record handling</p>

<p style="margin-top: 1em">This error is for a
/trans_splice gene that is hard to handle, and
unflattner/genbank2 doesn&rsquo;t</p>

<p style="margin-top: 1em">From: Chad Matsalla &lt;chad
&lt;at&gt; dieselwurks.com&gt; <br>
Subject: genbank2gff3.PLS and the unflatenner - Inconsistent
order? <br>
Date: 2005-07-15 19:51:48 GMT</p>

<p style="margin-top: 1em">AUTHOR <br>
Sheldon McKay (mckays@cshl.edu)</p>

<p style="margin-top: 1em">Copyright (c) 2004 Cold Spring
Harbor Laboratory.</p>

<p style="margin-top: 1em">AUTHOR of hacks for GFF2Chado
loading <br>
Don Gilbert (gilbertd@indiana.edu)</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_GENBANK2GFF3(1p)</p>
<hr>
</body>
</html>
