<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:55:58 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_GENBANK_REF_EXTRACTOR(1p) User Contributed Perl
Documentation BP_GENBANK_REF_EXTRACTOR(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_genbank_ref_extractor - Retrieves all related sequences
for a list of searches on Entrez gene</p>

<p style="margin-top: 1em">VERSION <br>
version 1.75</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bp_genbank_ref_extractor [options] [Entrez Gene Queries]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script searches on Entrez Gene database and retrieves
not only the gene sequence but also the related transcript
and protein sequences.</p>

<p style="margin-top: 1em">The gene UIDs of multiple
searches are collected before attempting to retrieve them so
each gene will only be analyzed once even if appearing as
result on more than one search.</p>

<p style="margin-top: 1em">Note that by default no
sequences are saved (see options and examples).</p>

<p style="margin-top: 1em">OPTIONS <br>
Several options can be used to fine tune the script
behaviour. It is possible to obtain extra base pairs
upstream and downstream of the gene, control the naming of
files and <br>
genome assembly to use.</p>

<p style="margin-top: 1em">See the section bugs for
problems when using default values of options.</p>

<p style="margin-top: 1em">--assembly <br>
When retrieving the sequence, a specific assemly can be
defined. The value expected is a regex that will be
case-insensitive. If it matches more than one assembly, it
will <br>
use the first match. It defauls to &quot;(primary|reference)
assembly&quot;.</p>

<p style="margin-top: 1em">--debug <br>
If set, even more output will be printed that may help on
debugging. Unlike the messages from --verbose and
--very-verbose, these will not appear on the log file unless
this <br>
option is selected. This option also sets
--very-verbose.</p>

<p style="margin-top: 1em">--downstream, --down <br>
Specifies the number of extra base pairs to be retrieved
downstream of the gene. This extra base pairs will only
affect the gene sequence, not the transcript or
proteins.</p>

<p style="margin-top: 1em">--email <br>
A valid email used to connect to the NCBI servers. This may
be used by NCBI to contact users in case of problems and
before blocking access in case of heavy usage.</p>

<p style="margin-top: 1em">--format <br>
Specifies the format that the sequences will be saved.
Defaults to genbank format. Valid formats are
&rsquo;genbank&rsquo; or &rsquo;fasta&rsquo;.</p>

<p style="margin-top: 1em">--genes <br>
Specifies the name for gene file. By default, they are not
saved. If no value is given defaults to its UID. Possible
values are &rsquo;uid&rsquo;, &rsquo;name&rsquo;,
&rsquo;symbol&rsquo; (the official symbol <br>
or nomenclature).</p>

<p style="margin-top: 1em">--help <br>
Display the documentation (this text).</p>

<p style="margin-top: 1em">--limit <br>
When making a query, limit the result to these first
specific results. This is to prevent the use of specially
unspecific queries and a warning will be given if a query
<br>
returns more results than the limit. The default value is
200. Note that this limit is for each search.</p>

<p style="margin-top: 1em">--non-coding, --nonon-coding
<br>
Some protein coding genes have transcripts that are
non-coding. By default, these sequences are saved as well.
--nonon-coding can be used to ignore those transcripts.</p>

<p style="margin-top: 1em">--proteins <br>
Specifies the name for proteins file. By default, they are
not saved. If no value is given defaults to its accession.
Possible values are &rsquo;accession&rsquo;,
&rsquo;description&rsquo;, &rsquo;gene&rsquo; <br>
(the corresponding gene ID) and &rsquo;transcript&rsquo;
(the corresponding transcript accesion).</p>

<p style="margin-top: 1em">Note that if not using
&rsquo;accession&rsquo; is possible for files to be
overwritten. It is possible for the same gene to encode more
than one protein or different proteins to have the <br>
same description.</p>

<p style="margin-top: 1em">--pseudo, --nopseudo <br>
By default, sequences of pseudo genes will be saved.
--nopseudo can be used to ignore those genes.</p>

<p style="margin-top: 1em">--save <br>
Specifies the path for the directory where the sequence and
log files will be saved. If the directory does not exist it
will be created altough the path to it must exist. <br>
Files on the directory may be rewritten if necessary. If
unspecified, a directory named extracted sequences on the
current directory will be used.</p>

<p style="margin-top: 1em">--save-data <br>
This options saves the data (gene UIDs, description, product
accessions, etc) to a file. As an optional value, the file
format can be specified. Defaults to CSV.</p>

<p style="margin-top: 1em">Currently only CSV is
supported.</p>

<p style="margin-top: 1em">Saving the data structure as a
CSV file, requires the installation of the Text::CSV
module.</p>

<p style="margin-top: 1em">--transcripts, --mrna <br>
Specifies the name for transcripts file. By default, they
are not saved. If no value is given defaults to its
accession. Possible values are &rsquo;accession&rsquo;,
&rsquo;description&rsquo;, <br>
&rsquo;gene&rsquo; (the corresponding gene ID) and
&rsquo;protein&rsquo; (the protein the transcript
encodes).</p>

<p style="margin-top: 1em">Note that if not using
&rsquo;accession&rsquo; is possible for files to be
overwritten. It is possible for the same gene to have more
than one transcript or different transcripts to have <br>
the same description. Also, non-coding transcripts will
create problems if using &rsquo;protein&rsquo;.</p>

<p style="margin-top: 1em">--upstream, --up <br>
Specifies the number of extra base pairs to be extracted
upstream of the gene. This extra base pairs will only affect
the gene sequence, not the transcript or proteins.</p>

<p style="margin-top: 1em">--verbose, --v <br>
If set, program becomes verbose. For an extremely verbose
program, use --very-verbose instead.</p>

<p style="margin-top: 1em">--very-verbose, --vv <br>
If set, program becomes extremely verbose. Setting this
option, automatically sets --verbose as well. For help in
debugging, consider using --debug</p>

<p style="margin-top: 1em">EXAMPLES <br>
&quot;bp_genbank_ref_extractor --transcripts=accession
&rsquo;&quot;homo sapiens&quot;[organism] AND
H2B&rsquo;&quot; <br>
Search Entrez gene with the query &rsquo;&quot;homo
sapiens&quot;[organism] AND H2B&rsquo;, and save their
transcripts sequences. Note that default value of --limit
may only extract some of the <br>
hits.</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--transcripts=accession --proteins=accession --format=fasta
&rsquo;&quot;homo sapiens&quot;[organism] AND H2B&rsquo;
&rsquo;&quot;homo sapiens&quot;[organism] AND
MCPH1&rsquo;&quot; <br>
Same as first example but also searches for
&rsquo;&quot;homo sapiens&quot;[organism] AND MCPH1&rsquo;,
proteins sequences, and saves them in the fasta format.</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--genes --up=100 --down=500 &rsquo;&quot;homo
sapiens&quot;[organism] AND H2B&rsquo;&quot; <br>
Same search as first example but saves the genomic sequences
instead including 100 and 500 bp upstream and
downstream.</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--genes --asembly=&rsquo;Alternate HuRef&rsquo;
&rsquo;&quot;homo sapiens&quot;[organism] AND
H2B&rsquo;&quot; <br>
Same search as first example but saves genomic sequences and
from the Alternate HuRef genome assembly instead.</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--save-data=CSV &rsquo;&quot;homo sapiens&quot;[organism]
AND H2B&rsquo;&quot; <br>
Same search as first example but does not save any sequence
but saves all the results in a CSV file.</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--save=&rsquo;search results&rsquo; --genes=name
--upstream=200 downstream=500 --nopseudo --nonnon-coding
--transcripts --proteins --format=fasta --save-data=CSV <br>
&rsquo;&quot;homo sapiens&quot;[organism] AND H2B&rsquo;
&rsquo;&quot;homo sapiens&quot;[organism] AND
MCPH1&rsquo;&quot; <br>
Searches on Entrez gene for both &rsquo;&quot;homo
sapiens&quot;[organism] AND H2B&rsquo; and &rsquo;&quot;homo
sapiens&quot;[organism] AND MCPH1&rsquo; and saves the gene
sequences of all hits (not passing the default <br>
limit and ignoring pseudogenes) plus 200 and 500bp upstream
and downstream of them. It will also save the sequences of
all transcripts and proteins of each gene (but ignoring <br>
non-coding transcripts). It will save the sequences in the
fasta format, inside a directory &quot;search results&quot;,
and save the results in a CSV file</p>

<p style="margin-top: 1em">KNOWN BUGS <br>
&Acirc;&middot; When supplying options, it&rsquo;s possible
to not supply a value and use their default. However, when
the expected value is a string, the next argument may be
confused as value <br>
for the option. For example, when using the following
command:</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--transcripts &rsquo;H2A AND homo sapiens&rsquo;&quot;</p>

<p style="margin-top: 1em">we mean to search for &rsquo;H2A
AND homo sapiens&rsquo; saving only the transcripts and
using the default as base for the filename. However, the
search terms will be interpreted as the <br>
base for the filenames (but since it&rsquo;s not a valid
identifier, it will return an error). To prevent this, you
can either specify the values:</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--transcripts &rsquo;accession&rsquo; &rsquo;H2A AND homo
sapiens&rsquo;&quot;</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--transcripts=&rsquo;accession&rsquo; &rsquo;H2A AND homo
sapiens&rsquo;&quot;</p>

<p style="margin-top: 1em">or you can use the double hash
to stop processing options. Note that this should only be
used after the last option. All arguments supplied after the
double dash will be <br>
interpreted as search terms</p>

<p style="margin-top: 1em">&quot;bp_genbank_ref_extractor
--transcripts -- &rsquo;H2A AND homo
sapiens&rsquo;&quot;</p>

<p style="margin-top: 1em">NOTES ON USAGE <br>
&Acirc;&middot; Genes that are marked as &rsquo;live&rsquo;
and &rsquo;protein-coding&rsquo; should have at least one
transcript. However, This is not always true due to mistakes
on annotation. Such cases will <br>
throw a warning. When faced with this, be nice and write to
the entrez RefSeq maintainers
&lt;http://www.ncbi.nlm.nih.gov/RefSeq/update.cgi&gt;.</p>

<p style="margin-top: 1em">&Acirc;&middot; When creating
the directories to save the files, if the directory already
exists it will be used and no error or warning will be
issued unless --debug as been set. If a non- <br>
directory file already exists with that name
bp_genbank_ref_extractor exits with an error.</p>

<p style="margin-top: 1em">&Acirc;&middot; On the subject
of verbosity, all messages are saved on the log file. The
options --verbose and --very-verbose only affect their
printing to standard output. Debug messages <br>
are different as they will only show up (and be logged) if
requested with --debug.</p>

<p style="margin-top: 1em">&Acirc;&middot; When saving a
file, to avoid problems with limited filesystems such as
NTFS or FAT, only some characters are allowed. All other
characters will be replaced by an underscore. <br>
Allowed characters are:</p>

<p style="margin-top: 1em">a-z 0-9 - + . , () {}
[]&rsquo;</p>

<p style="margin-top: 1em">&Acirc;&middot;
bp_genbank_ref_extractor tries to use the same file
extensions that bioperl would expect when saving the file.
If unable it will use the &rsquo;.seq&rsquo; extension.</p>

<p style="margin-top: 1em">FEEDBACK <br>
Mailing lists <br>
User feedback is an integral part of the evolution of this
and other Bioperl modules. Send your comments and
suggestions preferably to the Bioperl mailing list. Your
<br>
participation is much appreciated.</p>

<p style="margin-top: 1em">bioperl-l@bioperl.org - General
discussion <br>
http://bioperl.org/wiki/Mailing_lists - About the mailing
lists</p>

<p style="margin-top: 1em">Support <br>
Please direct usage questions or support issues to the
mailing list: bioperl-l@bioperl.org</p>

<p style="margin-top: 1em">rather than to the module
maintainer directly. Many experienced and reponsive experts
will be able look at the problem and quickly address it.
Please include a thorough <br>
description of the problem with code and data examples if at
all possible.</p>

<p style="margin-top: 1em">Reporting bugs <br>
Report bugs to the Bioperl bug tracking system to help us
keep track of the bugs and their resolution. Bug reports can
be submitted via the web:</p>


<p style="margin-top: 1em">https://github.com/bioperl/%%7Bdist%7D</p>

<p style="margin-top: 1em">AUTHOR <br>
Carn&Atilde;&laquo; Draug
&lt;carandraug+dev@gmail.com&gt;</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This software is copyright (c) 2011-2015 by
Carn&Atilde;&laquo; Draug.</p>

<p style="margin-top: 1em">This software is available under
the GNU General Public License, Version 3, June 2007.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-02-06
BP_GENBANK_REF_EXTRACTOR(1p)</p>
<hr>
</body>
</html>
