<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_GENBANK2GFF3(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_GENBANK2GFF3(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_GENBANK2GFF3(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_genbank2gff3.pl -- Genbank-&gt;gbrowse-friendly GFF3
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  bp_genbank2gff3.pl [options] filename(s)
  # process a directory containing GenBank flatfiles
  perl bp_genbank2gff3.pl --dir path_to_files --zip
  # process a single file, ignore explicit exons and introns
  perl bp_genbank2gff3.pl --filter exon --filter intron file.gbk.gz
  # process a list of files 
  perl bp_genbank2gff3.pl *gbk.gz
  # process data from URL, with Chado GFF model (-noCDS), and pipe to database loader
  curl ftp://ftp.ncbi.nih.gov/genomes/Saccharomyces_cerevisiae/CHR_X/NC_001142.gbk \
  | perl bp_genbank2gff3.pl -noCDS -in stdin -out stdout \
  | perl gmod_bulk_load_gff3.pl -dbname mychado -organism fromdata
    Options:
        --noinfer  -r  don't infer exon/mRNA subfeatures
        --conf     -i  path to the curation configuration file that contains user preferences
                       for Genbank entries (must be YAML format)
                       (if --manual is passed without --ini, user will be prompted to 
                        create the file if any manual input is saved)
        --sofile  -l  path to to the so.obo file to use for feature type mapping
                       (--sofile live will download the latest online revision)
        --manual   -m  when trying to guess the proper SO term, if more than
                       one option matches the primary tag, the converter will 
                       wait for user input to choose the correct one
                       (only works with --sofile)
        --dir      -d  path to a list of genbank flatfiles
        --outdir   -o  location to write GFF files (can be 'stdout' or '-' for pipe)
        --zip      -z  compress GFF3 output files with gzip
        --summary  -s  print a summary of the features in each contig
        --filter   -x  genbank feature type(s) to ignore
        --split    -y  split output to separate GFF and fasta files for
                       each genbank record
        --nolump   -n  separate file for each reference sequence
                       (default is to lump all records together into one 
                       output file for each input file)
        --ethresh  -e  error threshold for unflattener
                       set this high (&gt;2) to ignore all unflattener errors
        --[no]CDS  -c  Keep CDS-exons, or convert to alternate gene-RNA-protein-exon 
                       model. --CDS is default. Use --CDS to keep default GFF gene model, 
                       use --noCDS to convert to g-r-p-e.
        --format   -f  Input format (SeqIO types): GenBank, Swiss or Uniprot, EMBL work
                       (GenBank is default)
        --GFF_VERSION  3 is default, 2 and 2.5 and other Bio::Tools::GFF versions available
        --quiet        don't talk about what is being processed 
        --typesource   SO sequence type for source (e.g. chromosome; region; contig)
        --help     -h  display this message
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script uses Bio::SeqFeature::Tools::Unflattener and Bio::Tools::GFF to
  convert GenBank flatfiles to GFF3 with gene containment hierarchies mapped for
  optimal display in gbrowse.
<div class="Pp"></div>
The input files are assumed to be gzipped GenBank flatfiles for refseq contigs.
  The files may contain multiple GenBank records. Either a single file or an
  entire directory can be processed. By default, the DNA sequence is embedded in
  the GFF but it can be saved into separate fasta file with the --split(-y)
  option.
<div class="Pp"></div>
If an input file contains multiple records, the default behaviour is to dump all
  GFF and sequence to a file of the same name (with .gff appended). Using the
  'nolump' option will create a separate file for each genbank record. Using the
  'split' option will create separate GFF and Fasta files for each genbank
  record.
<h2 class="Ss" title="Ss" id="Notes"><a class="selflink" href="#Notes">Notes</a></h2>
<i>'split' and 'nolump' produce many files</i>
<div class="Pp"></div>
In cases where the input files contain many GenBank records (for example, the
  chromosome files for the mouse genome build), a very large number of output
  files will be produced if the 'split' or 'nolump' options are selected. If you
  do have lists of files &gt; 6000, use the --long_list option in
  bp_bulk_load_gff.pl or bp_fast_load_gff.pl to load the gff and/ or fasta
  files.
<div class="Pp"></div>
<i>Designed for RefSeq</i>
<div class="Pp"></div>
This script is designed for RefSeq genomic sequence entries. It may work for
  third party annotations but this has not been tested. But see below,
  Uniprot/Swissprot works, EMBL and possibly EMBL/Ensembl if you don't mind some
  gene model unflattener errors (dgg).
<div class="Pp"></div>
<i>G-R-P-E Gene Model</i>
<div class="Pp"></div>
Don Gilbert worked this over with needs to produce GFF3 suited to loading to
  GMOD Chado databases. Most of the changes I believe are suited for general
  use. One main chado-specific addition is the
<br/>
 --[no]cds2protein flag
<div class="Pp"></div>
My favorite GFF is to set the above as ON by default (disable with --nocds2prot)
  For general use it probably should be OFF, enabled with --cds2prot.
<div class="Pp"></div>
This writes GFF with an alternate, but useful Gene model, instead of the
  consensus model for GFF3
<div class="Pp"></div>
<pre>
  [ gene &gt; mRNA&gt; (exon,CDS,UTR) ]
</pre>
<div class="Pp"></div>
This alternate is
<div class="Pp"></div>
<pre>
  gene &gt; mRNA &gt; polypeptide &gt; exon
</pre>
<div class="Pp"></div>
means the only feature with dna bases is the exon. The others specify only
  location ranges on a genome. Exon of course is a child of mRNA and
  protein/peptide.
<div class="Pp"></div>
The protein/polypeptide feature is an important one, having all the annotations
  of the GenBank CDS feature, protein ID, translation, GO terms, Dbxrefs to
  other proteins.
<div class="Pp"></div>
UTRs, introns, CDS-exons are all inferred from the primary exon bases
  inside/outside appropriate higher feature ranges. Other special gene model
  features remain the same.
<div class="Pp"></div>
Several other improvements and bugfixes, minor but useful are included
<div class="Pp"></div>
<pre>
  * IO pipes now work:
    curl ftp://ncbigenomes/... | bp_genbank2gff3 --in stdin --out stdout | gff2chado ...
  * GenBank main record fields are added to source feature, e.g. organism, date,
    and the sourcetype, commonly chromosome for  genomes, is used.
  * Gene Model handling for ncRNA, pseudogenes are added.
  * GFF header is cleaner, more informative.
    --GFF_VERSION flag allows choice of v2 as well as default v3
  * GFF ##FASTA inclusion is improved, and
    CDS translation sequence is moved to FASTA records.
  * FT -&gt; GFF attribute mapping is improved.
  * --format choice of SeqIO input formats (GenBank default). 
    Uniprot/Swissprot and EMBL work and produce useful GFF.
  * SeqFeature::Tools::TypeMapper has a few FT -&gt; SOFA additions
      and more flexible usage.
</pre>
<h1 class="Sh" title="Sh" id="TODO"><a class="selflink" href="#TODO">TODO</a></h1>
<h2 class="Ss" title="Ss" id="Are_these_additions_desired?"><a class="selflink" href="#Are_these_additions_desired?">Are
  these additions desired?</a></h2>
<pre>
 * filter input records by taxon (e.g. keep only organism=xxx or taxa level = classYYY
 * handle Entrezgene, other non-sequence SeqIO structures (really should change
    those parsers to produce consistent annotation tags).
</pre>
<h2 class="Ss" title="Ss" id="Related_bugfixes/tests"><a class="selflink" href="#Related_bugfixes/tests">Related
  bugfixes/tests</a></h2>
These items from Bioperl mail were tested (sample data generating errors), and
  found corrected:
<div class="Pp"></div>
<pre>
 From: Ed Green &lt;green &lt;at&gt; eva.mpg.de&gt;
 Subject: genbank2gff3.pl on new human RefSeq
 Date: 2006-03-13 21:22:26 GMT 
   -- unspecified errors (sample data works now).
 From: Eric Just &lt;e-just &lt;at&gt; northwestern.edu&gt;
 Subject: genbank2gff3.pl
 Date: 2007-01-26 17:08:49 GMT
   -- bug fixed in genbank2gff3 for multi-record handling
</pre>
<div class="Pp"></div>
This error is for a /trans_splice gene that is hard to handle, and
  unflattner/genbank2 doesn't
<div class="Pp"></div>
<pre>
 From: Chad Matsalla &lt;chad &lt;at&gt; dieselwurks.com&gt; 
 Subject: genbank2gff3.PLS and the unflatenner - Inconsistent order?
 Date: 2005-07-15 19:51:48 GMT
</pre>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Sheldon McKay (mckays@cshl.edu)
<div class="Pp"></div>
Copyright (c) 2004 Cold Spring Harbor Laboratory.
<h2 class="Ss" title="Ss" id="AUTHOR_of_hacks_for_GFF2Chado_loading"><a class="selflink" href="#AUTHOR_of_hacks_for_GFF2Chado_loading">AUTHOR
  of hacks for GFF2Chado loading</a></h2>
Don Gilbert (gilbertd@indiana.edu)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
