<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_GENBANK_REF_EXTRACTOR(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_GENBANK_REF_EXTRACTOR(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_GENBANK_REF_EXTRACTOR(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_genbank_ref_extractor - Retrieves all related sequences for a list of
  searches on Entrez gene
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
version 1.75
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>bp_genbank_ref_extractor</b> [options] [Entrez Gene Queries]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script searches on <i>Entrez Gene</i> database and retrieves not only the
  gene sequence but also the related transcript and protein sequences.
<div class="Pp"></div>
The gene UIDs of multiple searches are collected before attempting to retrieve
  them so each gene will only be analyzed once even if appearing as result on
  more than one search.
<div class="Pp"></div>
Note that <i>by default no sequences are saved</i> (see options and examples).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
Several options can be used to fine tune the script behaviour. It is possible to
  obtain extra base pairs upstream and downstream of the gene, control the
  naming of files and genome assembly to use.
<div class="Pp"></div>
See the section bugs for problems when using default values of options.
<dl class="Bl-tag">
  <dt class="It-tag"><b>--assembly</b></dt>
  <dd class="It-tag">When retrieving the sequence, a specific assemly can be
      defined. The value expected is a regex that will be case-insensitive. If
      it matches more than one assembly, it will use the first match. It defauls
      to &quot;(primary|reference) assembly&quot;.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--debug</b></dt>
  <dd class="It-tag">If set, even more output will be printed that may help on
      debugging. Unlike the messages from <b>--verbose</b> and
      <b>--very-verbose</b>, these will not appear on the log file unless this
      option is selected. This option also sets <b>--very-verbose</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--downstream</b>, <b>--down</b></dt>
  <dd class="It-tag">Specifies the number of extra base pairs to be retrieved
      downstream of the gene. This extra base pairs will only affect the gene
      sequence, not the transcript or proteins.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--email</b></dt>
  <dd class="It-tag">A valid email used to connect to the NCBI servers. This may
      be used by NCBI to contact users in case of problems and before blocking
      access in case of heavy usage.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--format</b></dt>
  <dd class="It-tag">Specifies the format that the sequences will be saved.
      Defaults to <i>genbank</i> format. Valid formats are 'genbank' or
    'fasta'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--genes</b></dt>
  <dd class="It-tag">Specifies the name for gene file. By default, they are not
      saved. If no value is given defaults to its UID. Possible values are
      'uid', 'name', 'symbol' (the official symbol or nomenclature).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">Display the documentation (this text).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--limit</b></dt>
  <dd class="It-tag">When making a query, limit the result to these first
      specific results. This is to prevent the use of specially unspecific
      queries and a warning will be given if a query returns more results than
      the limit. The default value is 200. Note that this limit is for
      <i>each</i> search.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--non-coding</b>, <b>--nonon-coding</b></dt>
  <dd class="It-tag">Some protein coding genes have transcripts that are
      non-coding. By default, these sequences are saved as well.
      <b>--nonon-coding</b> can be used to ignore those transcripts.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--proteins</b></dt>
  <dd class="It-tag">Specifies the name for proteins file. By default, they are
      not saved. If no value is given defaults to its accession. Possible values
      are 'accession', 'description', 'gene' (the corresponding gene ID) and
      'transcript' (the corresponding transcript accesion).
    <div style="height: 1.00em;">&#x00A0;</div>
    Note that if not using 'accession' is possible for files to be overwritten.
      It is possible for the same gene to encode more than one protein or
      different proteins to have the same description.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pseudo</b>, <b>--nopseudo</b></dt>
  <dd class="It-tag">By default, sequences of pseudo genes will be saved.
      <b>--nopseudo</b> can be used to ignore those genes.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--save</b></dt>
  <dd class="It-tag">Specifies the path for the directory where the sequence and
      log files will be saved. If the directory does not exist it will be
      created altough the path to it must exist. Files on the directory may be
      rewritten if necessary. If unspecified, a directory named <i>extracted
      sequences</i> on the current directory will be used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--save-data</b></dt>
  <dd class="It-tag">This options saves the data (gene UIDs, description,
      product accessions, etc) to a file. As an optional value, the file format
      can be specified. Defaults to CSV.
    <div style="height: 1.00em;">&#x00A0;</div>
    Currently only CSV is supported.
    <div style="height: 1.00em;">&#x00A0;</div>
    Saving the data structure as a CSV file, requires the installation of the
      Text::CSV module.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--transcripts</b>, <b>--mrna</b></dt>
  <dd class="It-tag">Specifies the name for transcripts file. By default, they
      are not saved. If no value is given defaults to its accession. Possible
      values are 'accession', 'description', 'gene' (the corresponding gene ID)
      and 'protein' (the protein the transcript encodes).
    <div style="height: 1.00em;">&#x00A0;</div>
    Note that if not using 'accession' is possible for files to be overwritten.
      It is possible for the same gene to have more than one transcript or
      different transcripts to have the same description. Also, non-coding
      transcripts will create problems if using 'protein'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--upstream</b>, <b>--up</b></dt>
  <dd class="It-tag">Specifies the number of extra base pairs to be extracted
      upstream of the gene. This extra base pairs will only affect the gene
      sequence, not the transcript or proteins.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--verbose</b>, <b>--v</b></dt>
  <dd class="It-tag">If set, program becomes verbose. For an extremely verbose
      program, use <b>--very-verbose</b> instead.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--very-verbose</b>, <b>--vv</b></dt>
  <dd class="It-tag">If set, program becomes extremely verbose. Setting this
      option, automatically sets <b>--verbose</b> as well. For help in
      debugging, consider using <b>--debug</b></dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;bp_genbank_ref_extractor --transcripts=accession
    '&quot;homo sapiens&quot;[organism] AND H2B'&quot;</dt>
  <dd class="It-tag">Search Entrez gene with the query '&quot;homo
      sapiens&quot;[organism] AND H2B', and save their transcripts sequences.
      Note that default value of <b>--limit</b> may only extract some of the
      hits.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;bp_genbank_ref_extractor --transcripts=accession
    --proteins=accession --format=fasta '&quot;homo sapiens&quot;[organism] AND
    H2B' '&quot;homo sapiens&quot;[organism] AND MCPH1'&quot;</dt>
  <dd class="It-tag">Same as first example but also searches for '&quot;homo
      sapiens&quot;[organism] AND MCPH1', proteins sequences, and saves them in
      the fasta format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;bp_genbank_ref_extractor --genes --up=100 --down=500
    '&quot;homo sapiens&quot;[organism] AND H2B'&quot;</dt>
  <dd class="It-tag">Same search as first example but saves the genomic
      sequences instead including 100 and 500 bp upstream and downstream.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;bp_genbank_ref_extractor --genes --asembly='Alternate
    HuRef' '&quot;homo sapiens&quot;[organism] AND H2B'&quot;</dt>
  <dd class="It-tag">Same search as first example but saves genomic sequences
      and from the Alternate HuRef genome assembly instead.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;bp_genbank_ref_extractor --save-data=CSV '&quot;homo
    sapiens&quot;[organism] AND H2B'&quot;</dt>
  <dd class="It-tag">Same search as first example but does not save any sequence
      but saves all the results in a CSV file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;bp_genbank_ref_extractor --save='search results'
    --genes=name --upstream=200 downstream=500 --nopseudo --nonnon-coding
    --transcripts --proteins --format=fasta --save-data=CSV '&quot;homo
    sapiens&quot;[organism] AND H2B' '&quot;homo sapiens&quot;[organism] AND
    MCPH1'&quot;</dt>
  <dd class="It-tag">Searches on Entrez gene for both '&quot;homo
      sapiens&quot;[organism] AND H2B' and '&quot;homo sapiens&quot;[organism]
      AND MCPH1' and saves the gene sequences of all hits (not passing the
      default limit and ignoring pseudogenes) plus 200 and 500bp upstream and
      downstream of them. It will also save the sequences of all transcripts and
      proteins of each gene (but ignoring non-coding transcripts). It will save
      the sequences in the fasta format, inside a directory &quot;search
      results&quot;, and save the results in a CSV file</dd>
</dl>
<h1 class="Sh" title="Sh" id="KNOWN_BUGS"><a class="selflink" href="#KNOWN_BUGS">KNOWN
  BUGS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">When supplying options, it's possible to not supply a value
      and use their default. However, when the expected value is a string, the
      next argument may be confused as value for the option. For example, when
      using the following command:
    <div style="height: 1.00em;">&#x00A0;</div>
    &quot;bp_genbank_ref_extractor --transcripts 'H2A AND homo sapiens'&quot;
    <div style="height: 1.00em;">&#x00A0;</div>
    we mean to search for 'H2A AND homo sapiens' saving only the transcripts and
      using the default as base for the filename. However, the search terms will
      be interpreted as the base for the filenames (but since it's not a valid
      identifier, it will return an error). To prevent this, you can either
      specify the values:
    <div style="height: 1.00em;">&#x00A0;</div>
    &quot;bp_genbank_ref_extractor --transcripts 'accession' 'H2A AND homo
      sapiens'&quot;
    <div style="height: 1.00em;">&#x00A0;</div>
    &quot;bp_genbank_ref_extractor --transcripts='accession' 'H2A AND homo
      sapiens'&quot;
    <div style="height: 1.00em;">&#x00A0;</div>
    or you can use the double hash to stop processing options. Note that this
      should only be used after the last option. All arguments supplied after
      the double dash will be interpreted as search terms
    <div style="height: 1.00em;">&#x00A0;</div>
    &quot;bp_genbank_ref_extractor --transcripts -- 'H2A AND homo
    sapiens'&quot;</dd>
</dl>
<h1 class="Sh" title="Sh" id="NOTES_ON_USAGE"><a class="selflink" href="#NOTES_ON_USAGE">NOTES
  ON USAGE</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Genes that are marked as 'live' and 'protein-coding' should
      have at least one transcript. However, This is not always true due to
      mistakes on annotation. Such cases will throw a warning. When faced with
      this, be nice and write to the entrez RefSeq maintainers
      &lt;http://www.ncbi.nlm.nih.gov/RefSeq/update.cgi&gt;.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">When creating the directories to save the files, if the
      directory already exists it will be used and no error or warning will be
      issued unless <b>--debug</b> as been set. If a non-directory file already
      exists with that name bp_genbank_ref_extractor exits with an error.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">On the subject of verbosity, all messages are saved on the
      log file. The options <b>--verbose</b> and <b>--very-verbose</b> only
      affect their printing to standard output. Debug messages are different as
      they will only show up (and be logged) if requested with
    <b>--debug</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">When saving a file, to avoid problems with limited
      filesystems such as NTFS or FAT, only some characters are allowed. All
      other characters will be replaced by an underscore. Allowed characters
      are:
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>a-z 0-9 - + . , () {} []'</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag"><b>bp_genbank_ref_extractor</b> tries to use the same file
      extensions that bioperl would expect when saving the file. If unable it
      will use the '.seq' extension.</dd>
</dl>
<h1 class="Sh" title="Sh" id="FEEDBACK"><a class="selflink" href="#FEEDBACK">FEEDBACK</a></h1>
<h2 class="Ss" title="Ss" id="Mailing_lists"><a class="selflink" href="#Mailing_lists">Mailing
  lists</a></h2>
User feedback is an integral part of the evolution of this and other Bioperl
  modules. Send your comments and suggestions preferably to the Bioperl mailing
  list. Your participation is much appreciated.
<div class="Pp"></div>
<pre>
  bioperl-l@bioperl.org                  - General discussion
  http://bioperl.org/wiki/Mailing_lists  - About the mailing lists
</pre>
<h2 class="Ss" title="Ss" id="Support"><a class="selflink" href="#Support">Support</a></h2>
Please direct usage questions or support issues to the mailing list:
  <i>bioperl-l@bioperl.org</i>
<div class="Pp"></div>
rather than to the module maintainer directly. Many experienced and reponsive
  experts will be able look at the problem and quickly address it. Please
  include a thorough description of the problem with code and data examples if
  at all possible.
<h2 class="Ss" title="Ss" id="Reporting_bugs"><a class="selflink" href="#Reporting_bugs">Reporting
  bugs</a></h2>
Report bugs to the Bioperl bug tracking system to help us keep track of the bugs
  and their resolution. Bug reports can be submitted via the web:
<div class="Pp"></div>
<pre>
  https://github.com/bioperl/%%7Bdist%7D
</pre>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Carn&#x00EB; Draug &lt;carandraug+dev@gmail.com&gt;
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This software is copyright (c) 2011-2015 by Carn&#x00EB; Draug.
<div class="Pp"></div>
This software is available under the GNU General Public License, Version 3, June
  2007.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-02-06</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
