<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:55:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_LOAD_GFF(1p) User Contributed Perl Documentation
BP_LOAD_GFF(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_load_gff.pl - Load a Bio::DB::GFF database from GFF
files.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
% bp_load_gff.pl -d testdb -u user -p pw <br>
--dsn
&rsquo;dbi:mysql:database=dmel_r5_1;host=myhost;port=myport&rsquo;
<br>
dna1.fa dna2.fa features1.gff features2.gff ...</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script loads a Bio::DB::GFF database with the features
contained in a list of GFF files and/or FASTA sequence
files. You must use the exact variant of GFF described in
<br>
Bio::DB::GFF. Various command-line options allow you to
control which database to load and whether to allow an
existing database to be overwritten.</p>

<p style="margin-top: 1em">This script uses the
Bio::DB::GFF interface, and so works with all database
adaptors currently supported by that module (MySQL, Oracle,
PostgreSQL soon). However, it is slow. <br>
For faster loading, see the MySQL-specific
bp_bulk_load_gff.pl and bp_fast_load_gff.pl scripts.</p>

<p style="margin-top: 1em">NOTES <br>
If the filename is given as &quot;-&quot; then the input is
taken from standard input. Compressed files (.gz, .Z, .bz2)
are automatically uncompressed.</p>

<p style="margin-top: 1em">FASTA format files are
distinguished from GFF files by their filename extensions.
Files ending in .fa, .fasta, .fast, .seq, .dna and their
uppercase variants are treated as <br>
FASTA files. Everything else is treated as a GFF file. If
you wish to load -fasta files from STDIN, then use the -f
command-line swith with an argument of &rsquo;-&rsquo;, as
in</p>

<p style="margin-top: 1em">gunzip my_data.fa.gz |
bp_fast_load_gff.pl -d test -f -</p>

<p style="margin-top: 1em">On the first load of a database,
you will see a number of &quot;unknown table&quot; errors.
This is normal.</p>

<p style="margin-top: 1em">About maxfeature: the default
value is 100,000,000 bases. If you have features that are
close to or greater that 100Mb in length, then the value of
maxfeature should be <br>
increased to 1,000,000,000, or another power of 10.</p>

<p style="margin-top: 1em">COMMAND-LINE OPTIONS <br>
Command-line options can be abbreviated to single-letter
options. e.g. -d instead of --database.</p>

<p style="margin-top: 1em">--dsn &lt;dsn&gt; Data source
(default dbi:mysql:test) <br>
--adaptor &lt;adaptor&gt; Schema adaptor (default
dbi::mysqlopt) <br>
--user &lt;user&gt; Username for mysql authentication <br>
--pass &lt;password&gt; Password for mysql authentication
<br>
--fasta &lt;path&gt; Fasta file or directory containing
fasta files for the DNA <br>
--create Force creation and initialization of database <br>
--maxfeature Set the value of the maximum feature size
(default 100 Mb; must be a power of 10) <br>
--group A list of one or more tag names (comma or space
separated) <br>
to be used for grouping in the 9th column. <br>
--upgrade Upgrade existing database to current schema <br>
--gff3_munge Activate GFF3 name munging (see Bio::DB::GFF)
<br>
--quiet No progress reports <br>
--summary Generate summary statistics for drawing coverage
histograms. <br>
This can be run on a previously loaded database or during
<br>
the load.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Bio::DB::GFF, bulk_load_gff.pl, bp_load_gff.pl</p>

<p style="margin-top: 1em">AUTHOR <br>
Lincoln Stein, lstein@cshl.org</p>

<p style="margin-top: 1em">Copyright (c) 2002 Cold Spring
Harbor Laboratory</p>

<p style="margin-top: 1em">This library is free software;
you can redistribute it and/or modify it under the same
terms as Perl itself. See DISCLAIMER.txt for disclaimers of
warranty.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_LOAD_GFF(1p)</p>
<hr>
</body>
</html>
