<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_MASK_BY_SEARCH(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_MASK_BY_SEARCH(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_MASK_BY_SEARCH(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_mask_by_search - mask sequence(s) based on its alignment results
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  bp_mask_by_search.pl -f blast genomefile blastfile.bls &gt; maskedgenome.fa
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Mask sequence based on significant alignments of another sequence. You need to
  provide the report file and the entire sequence data which you want to mask.
  By default this will assume you have done a TBLASTN (or TFASTY) and try and
  mask the hit sequence assuming you've provided the sequence file for the hit
  database. If you would like to do the reverse and mask the query sequence
  specify the -t/--type query flag.
<div class="Pp"></div>
This is going to read in the whole sequence file into memory so for large
  genomes this may fall over. I'm using DB_File to prevent keeping everything in
  memory, one solution is to split the genome into pieces (BEFORE you run the DB
  search though, you want to use the exact file you BLASTed with as input to
  this program).
<div class="Pp"></div>
Below the double dash (--) options are of the form --format=fasta or --format
  fasta or you can just say -f fasta
<div class="Pp"></div>
By -f/--format I mean either are acceptable options. The =s or =n or =c specify
  these arguments expect a 'string'
<div class="Pp"></div>
Options:
<br/>
 -f/--format=s Search report format (fasta,blast,axt,hmmer,etc)
<br/>
 -sf/--sformat=s Sequence format (fasta,genbank,embl,swissprot)
<br/>
 --hardmask (booelean) Hard mask the sequence
<br/>
 with the maskchar [default is lowercase mask]
<br/>
 --maskchar=c Character to mask with [default is N], change
<br/>
 to 'X' for protein sequences
<br/>
 -e/--evalue=n Evalue cutoff for HSPs and Hits, only
<br/>
 mask sequence if alignment has specified evalue
<br/>
 or better
<br/>
 -o/--out/
<br/>
 --outfile=file Output file to save the masked sequence to.
<br/>
 -t/--type=s Alignment seq type you want to mask, the
<br/>
 'hit' or the 'query' sequence. [default is 'hit']
<br/>
 --minlen=n Minimum length of an HSP for it to be used
<br/>
 in masking [default 0]
<br/>
 -h/--help See this help information
<h1 class="Sh" title="Sh" id="AUTHOR_-_Jason_Stajich"><a class="selflink" href="#AUTHOR_-_Jason_Stajich">AUTHOR
  - Jason Stajich</a></h1>
Jason Stajich, jason-at-bioperl-dot-org.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
