<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:00 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_MASK_BY_SEARCH(1p) User Contributed Perl
Documentation BP_MASK_BY_SEARCH(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_mask_by_search - mask sequence(s) based on its alignment
results</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bp_mask_by_search.pl -f blast genomefile blastfile.bls &gt;
maskedgenome.fa</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Mask sequence based on significant alignments of another
sequence. You need to provide the report file and the entire
sequence data which you want to mask. By default this will
<br>
assume you have done a TBLASTN (or TFASTY) and try and mask
the hit sequence assuming you&rsquo;ve provided the sequence
file for the hit database. If you would like to do the
reverse <br>
and mask the query sequence specify the -t/--type query
flag.</p>

<p style="margin-top: 1em">This is going to read in the
whole sequence file into memory so for large genomes this
may fall over. I&rsquo;m using DB_File to prevent keeping
everything in memory, one solution is <br>
to split the genome into pieces (BEFORE you run the DB
search though, you want to use the exact file you BLASTed
with as input to this program).</p>

<p style="margin-top: 1em">Below the double dash (--)
options are of the form --format=fasta or --format fasta or
you can just say -f fasta</p>

<p style="margin-top: 1em">By -f/--format I mean either are
acceptable options. The =s or =n or =c specify these
arguments expect a &rsquo;string&rsquo;</p>

<p style="margin-top: 1em">Options: <br>
-f/--format=s Search report format
(fasta,blast,axt,hmmer,etc) <br>
-sf/--sformat=s Sequence format
(fasta,genbank,embl,swissprot) <br>
--hardmask (booelean) Hard mask the sequence <br>
with the maskchar [default is lowercase mask] <br>
--maskchar=c Character to mask with [default is N], change
<br>
to &rsquo;X&rsquo; for protein sequences <br>
-e/--evalue=n Evalue cutoff for HSPs and Hits, only <br>
mask sequence if alignment has specified evalue <br>
or better <br>
-o/--out/ <br>
--outfile=file Output file to save the masked sequence to.
<br>
-t/--type=s Alignment seq type you want to mask, the <br>
&rsquo;hit&rsquo; or the &rsquo;query&rsquo; sequence.
[default is &rsquo;hit&rsquo;] <br>
--minlen=n Minimum length of an HSP for it to be used <br>
in masking [default 0] <br>
-h/--help See this help information</p>

<p style="margin-top: 1em">AUTHOR - Jason Stajich <br>
Jason Stajich, jason-at-bioperl-dot-org.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_MASK_BY_SEARCH(1p)</p>
<hr>
</body>
</html>
