<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_OLIGO_COUNT(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_OLIGO_COUNT(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_OLIGO_COUNT(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_oligo_count - oligo count and frequency
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  Usage:  bp_oligo_count [-h/--help] [-l/--length OLIGOLENGTH]
          [-f/--format SEQFORMAT] [-i/--in/-s/--sequence SEQFILE]
          [-o/--out OUTFILE]
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This scripts counts occurrence and frequency for all oligonucleotides of given
  length.
<div class="Pp"></div>
It can be used to determine what primers are useful for frequent priming of
  nucleic acid for random labeling.
<div class="Pp"></div>
Note that this script could be run by utilizing the compseq program which is
  part of EMBOSS.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
The default sequence format is fasta. If no outfile is given, the results will
  be printed to standard out. All other options can entered interactively.
<h1 class="Sh" title="Sh" id="FEEDBACK"><a class="selflink" href="#FEEDBACK">FEEDBACK</a></h1>
<h2 class="Ss" title="Ss" id="Mailing_Lists"><a class="selflink" href="#Mailing_Lists">Mailing
  Lists</a></h2>
User feedback is an integral part of the evolution of this and other Bioperl
  modules. Send your comments and suggestions preferably to the Bioperl mailing
  list. Your participation is much appreciated.
<div class="Pp"></div>
<pre>
  bioperl-l@bioperl.org                  - General discussion
  http://bioperl.org/wiki/Mailing_lists  - About the mailing lists
</pre>
<h2 class="Ss" title="Ss" id="Reporting_Bugs"><a class="selflink" href="#Reporting_Bugs">Reporting
  Bugs</a></h2>
Report bugs to the Bioperl bug tracking system to help us keep track of the bugs
  and their resolution. Bug reports can be submitted via the web:
<div class="Pp"></div>
<pre>
  https://github.com/bioperl/bioperl-live/issues
</pre>
<h1 class="Sh" title="Sh" id="AUTHOR_-_Charles_C._Kim"><a class="selflink" href="#AUTHOR_-_Charles_C._Kim">AUTHOR
  - Charles C. Kim</a></h1>
Email cckim@stanford.edu
<h1 class="Sh" title="Sh" id="HISTORY"><a class="selflink" href="#HISTORY">HISTORY</a></h1>
Written July 2, 2001
<div class="Pp"></div>
Submitted to bioperl scripts project 2001/08/06
<div class="Pp"></div>
&gt;&gt; 100 x speed optimization by Heikki Lehvaslaiho</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
