<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:01 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_OLIGO_COUNT(1p) User Contributed Perl Documentation
BP_OLIGO_COUNT(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_oligo_count - oligo count and frequency</p>

<p style="margin-top: 1em">SYNOPSIS <br>
Usage: bp_oligo_count [-h/--help] [-l/--length OLIGOLENGTH]
<br>
[-f/--format SEQFORMAT] [-i/--in/-s/--sequence SEQFILE] <br>
[-o/--out OUTFILE]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This scripts counts occurrence and frequency for all
oligonucleotides of given length.</p>

<p style="margin-top: 1em">It can be used to determine what
primers are useful for frequent priming of nucleic acid for
random labeling.</p>

<p style="margin-top: 1em">Note that this script could be
run by utilizing the compseq program which is part of
EMBOSS.</p>

<p style="margin-top: 1em">OPTIONS <br>
The default sequence format is fasta. If no outfile is
given, the results will be printed to standard out. All
other options can entered interactively.</p>

<p style="margin-top: 1em">FEEDBACK <br>
Mailing Lists <br>
User feedback is an integral part of the evolution of this
and other Bioperl modules. Send your comments and
suggestions preferably to the Bioperl mailing list. Your
<br>
participation is much appreciated.</p>

<p style="margin-top: 1em">bioperl-l@bioperl.org - General
discussion <br>
http://bioperl.org/wiki/Mailing_lists - About the mailing
lists</p>

<p style="margin-top: 1em">Reporting Bugs <br>
Report bugs to the Bioperl bug tracking system to help us
keep track of the bugs and their resolution. Bug reports can
be submitted via the web:</p>


<p style="margin-top: 1em">https://github.com/bioperl/bioperl-live/issues</p>

<p style="margin-top: 1em">AUTHOR - Charles C. Kim <br>
Email cckim@stanford.edu</p>

<p style="margin-top: 1em">HISTORY <br>
Written July 2, 2001</p>

<p style="margin-top: 1em">Submitted to bioperl scripts
project 2001/08/06</p>

<p style="margin-top: 1em">&gt;&gt; 100 x speed
optimization by Heikki Lehvaslaiho</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_OLIGO_COUNT(1p)</p>
<hr>
</body>
</html>
