<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:03 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_PROCESS_WORMBASE(1p) User Contributed Perl
Documentation BP_PROCESS_WORMBASE(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_process_wormbase.pl - Massage WormBase GFF files into a
version suitable for the Generic Genome Browser</p>

<p style="margin-top: 1em">SYNOPSIS <br>
% bp_process_wormbase.pl ./WS61 &gt; wormbase.gff</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script massages the Wormbase GFF files located at
ftp://www.wormbase.org/pub/wormbase/GENE_DUMPS into a
version of the GFF format suitable for display by the
generic genome <br>
browser. It mainly adds comments to the annotations and
designates certain well-spaced genetic loci as framework
landmarks.</p>

<p style="margin-top: 1em">This script requires the AcePerl
distribution, which is available on CPAN (look for the
&quot;Ace&quot; module).</p>

<p style="margin-top: 1em">To use this script, get the
WormBase GFF files from the FTP site listed above and place
them in a directory. It might be a good idea to name the
directory after the current <br>
release, such as WS61. You do not need to uncompress the
files.</p>

<p style="margin-top: 1em">Then give that directory as the
argument to this script and capture the script&rsquo;s
output to a file:</p>

<p style="margin-top: 1em">% bp_process_wormbase.pl ./WS61
&gt; wormbase.gff</p>

<p style="margin-top: 1em">It may take a while before you
see output from this script, since it must first fetch gene
and protein database from the remote AceDB running at
www.wormbase.org. The <br>
wormbase.gff file can then be loaded into a Bio::DB::GFF
database using the following command:</p>

<p style="margin-top: 1em">% bulk_load_gff.pl -d
&lt;databasename&gt; wormbase.gff</p>

<p style="margin-top: 1em">SEE ALSO <br>
Bio::DB::GFF, bulk_load_gff.pl, load_gff.pl</p>

<p style="margin-top: 1em">AUTHOR <br>
Lincoln Stein &lt;lstein@cshl.org&gt;</p>

<p style="margin-top: 1em">Copyright (c) 2002 Cold Spring
Harbor Laboratory</p>

<p style="margin-top: 1em">This library is free software;
you can redistribute it and/or modify it under the same
terms as Perl itself. See DISCLAIMER.txt for disclaimers of
warranty.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_PROCESS_WORMBASE(1p)</p>
<hr>
</body>
</html>
