<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:01 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_PAIRWISE_KAKS(1p) User Contributed Perl Documentation
BP_PAIRWISE_KAKS(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_pairwise_kaks - script to calculate pairwise Ka,Ks for a
set of sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bp_pairwise_kaks.PLS -i t/data/worm_fam_2785.cdna [-f
fasta/genbank/embl...] [-msa tcoffee/clustal] [-kaks
yn00/codeml]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script will take as input a dataset of cDNA sequences
verify <br>
that they contain no stop codons, align them in protein
space, <br>
project the alignment back into cDNA and estimate the Ka
<br>
(non-synonymous) and Ks (synonymous) substitutions based on
the ML <br>
method of Yang with the PAML package.</p>

<p style="margin-top: 1em">Requires: <br>
* bioperl-run package <br>
* PAML program codeml or yn00 <br>
* Multiple sequence alignment programs Clustalw OR
T-Coffee</p>

<p style="margin-top: 1em">Often there are specific
specific parameters you want to run when you <br>
a computing Ka/Ks ratios so consider this script a starting
point and <br>
do not rely it on for every situation.</p>

<p style="margin-top: 1em">FEEDBACK <br>
Mailing Lists <br>
User feedback is an integral part of the evolution of this
and other Bioperl modules. Send your comments and
suggestions preferably to the Bioperl mailing list. Your
<br>
participation is much appreciated.</p>

<p style="margin-top: 1em">bioperl-l@bioperl.org - General
discussion <br>
http://bioperl.org/wiki/Mailing_lists - About the mailing
lists</p>

<p style="margin-top: 1em">Reporting Bugs <br>
Report bugs to the Bioperl bug tracking system to help us
keep track of the bugs and their resolution. Bug reports can
be submitted via the web:</p>


<p style="margin-top: 1em">https://github.com/bioperl/bioperl-live/issues</p>

<p style="margin-top: 1em">AUTHOR <br>
Jason Stajich jason-at-bioperl-dot-org</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_PAIRWISE_KAKS(1p)</p>
<hr>
</body>
</html>
