<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_PROCESS_GADFLY(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_PROCESS_GADFLY(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_PROCESS_GADFLY(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_process_gadfly.pl - Massage Gadfly/FlyBase GFF files into a version suitable
  for the Generic Genome Browser
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  % bp_process_gadfly.pl ./RELEASE2 &gt; gadfly.gff
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script massages the RELEASE 3 Flybase/Gadfly GFF files located at
  http://www.fruitfly.org/sequence/release3download.shtml into the
  &quot;correct&quot; version of the GFF format.
<div class="Pp"></div>
To use this script, download the whole genome FASTA file and save it to disk.
  (The downloaded file will be called something like
  &quot;na_whole-genome_genomic_dmel_RELEASE3.FASTA&quot;, but the link on the
  HTML page doesn't give the filename.) Do the same for the whole genome GFF
  annotation file (the saved file will be called something like
  &quot;whole-genome_annotation-feature-region_dmel_RELEASE3.GFF&quot;.) If you
  wish you can download the ZIP compressed versions of these files.
<div class="Pp"></div>
Next run this script on the two files, indicating the name of the downloaded
  FASTA file first, followed by the gff file:
<div class="Pp"></div>
<pre>
 % bp_process_gadfly.pl na_whole-genome_genomic_dmel_RELEASE3.FASTA whole-genome_annotation-feature-region_dmel_RELEASE3.GFF &gt; fly.gff
</pre>
<div class="Pp"></div>
The gadfly.gff file and the fasta file can now be loaded into a Bio::DB::GFF
  database using the following command:
<div class="Pp"></div>
<pre>
  % bulk_load_gff.pl -d fly -fasta na_whole-genome_genomic_dmel_RELEASE3.FASTA fly.gff
</pre>
<div class="Pp"></div>
(Where &quot;fly&quot; is the name of the database. Change it as appropriate.
  The database must already exist and be writable by you!)
<div class="Pp"></div>
The resulting database will have the following feature types (represented as
  &quot;method:source&quot;):
<div class="Pp"></div>
<pre>
  Component:arm              A chromosome arm
  Component:scaffold         A chromosome scaffold (accession #)
  Component:gap              A gap in the assembly
  clone:clonelocator         A BAC clone
  gene:gadfly                A gene accession number
  transcript:gadfly          A transcript accession number
  translation:gadfly         A translation
  codon:gadfly               Significance unknown
  exon:gadfly                An exon
  symbol:gadfly              A classical gene symbol
  similarity:blastn          A BLASTN hit
  similarity:blastx          A BLASTX hit
  similarity:sim4            EST-&gt;genome using SIM4
  similarity:groupest        EST-&gt;genome using GROUPEST
  similarity:repeatmasker    A repeat
</pre>
<div class="Pp"></div>
IMPORTANT NOTE: This script will *only* work with the RELEASE3 gadfly files and
  will not work with earlier releases.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Bio::DB::GFF, bulk_load_gff.pl, load_gff.pl
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Lincoln Stein, lstein@cshl.org
<div class="Pp"></div>
Copyright (c) 2002 Cold Spring Harbor Laboratory
<div class="Pp"></div>
This library is free software; you can redistribute it and/or modify it under
  the same terms as Perl itself. See DISCLAIMER.txt for disclaimers of
  warranty.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
