<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_PROCESS_WORMBASE(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_PROCESS_WORMBASE(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_PROCESS_WORMBASE(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_process_wormbase.pl - Massage WormBase GFF files into a version suitable for
  the Generic Genome Browser
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  % bp_process_wormbase.pl ./WS61 &gt; wormbase.gff
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script massages the Wormbase GFF files located at
  ftp://www.wormbase.org/pub/wormbase/GENE_DUMPS into a version of the GFF
  format suitable for display by the generic genome browser. It mainly adds
  comments to the annotations and designates certain well-spaced genetic loci as
  framework landmarks.
<div class="Pp"></div>
This script requires the AcePerl distribution, which is available on CPAN (look
  for the &quot;Ace&quot; module).
<div class="Pp"></div>
To use this script, get the WormBase GFF files from the FTP site listed above
  and place them in a directory. It might be a good idea to name the directory
  after the current release, such as WS61. You do not need to uncompress the
  files.
<div class="Pp"></div>
Then give that directory as the argument to this script and capture the script's
  output to a file:
<div class="Pp"></div>
<pre>
  % bp_process_wormbase.pl ./WS61 &gt; wormbase.gff
</pre>
<div class="Pp"></div>
It may take a while before you see output from this script, since it must first
  fetch gene and protein database from the remote AceDB running at
  www.wormbase.org. The wormbase.gff file can then be loaded into a Bio::DB::GFF
  database using the following command:
<div class="Pp"></div>
<pre>
  % bulk_load_gff.pl -d &lt;databasename&gt; wormbase.gff
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Bio::DB::GFF, bulk_load_gff.pl, load_gff.pl
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Lincoln Stein &lt;lstein@cshl.org&gt;
<div class="Pp"></div>
Copyright (c) 2002 Cold Spring Harbor Laboratory
<div class="Pp"></div>
This library is free software; you can redistribute it and/or modify it under
  the same terms as Perl itself. See DISCLAIMER.txt for disclaimers of
  warranty.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
