<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:03 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_PROCESS_GADFLY(1p) User Contributed Perl
Documentation BP_PROCESS_GADFLY(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_process_gadfly.pl - Massage Gadfly/FlyBase GFF files into
a version suitable for the Generic Genome Browser</p>

<p style="margin-top: 1em">SYNOPSIS <br>
% bp_process_gadfly.pl ./RELEASE2 &gt; gadfly.gff</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script massages the RELEASE 3 Flybase/Gadfly GFF files
located at
http://www.fruitfly.org/sequence/release3download.shtml into
the &quot;correct&quot; version of the GFF format.</p>

<p style="margin-top: 1em">To use this script, download the
whole genome FASTA file and save it to disk. (The downloaded
file will be called something like
&quot;na_whole-genome_genomic_dmel_RELEASE3.FASTA&quot;,
<br>
but the link on the HTML page doesn&rsquo;t give the
filename.) Do the same for the whole genome GFF annotation
file (the saved file will be called something like <br>

&quot;whole-genome_annotation-feature-region_dmel_RELEASE3.GFF&quot;.)
If you wish you can download the ZIP compressed versions of
these files.</p>

<p style="margin-top: 1em">Next run this script on the two
files, indicating the name of the downloaded FASTA file
first, followed by the gff file:</p>

<p style="margin-top: 1em">% bp_process_gadfly.pl
na_whole-genome_genomic_dmel_RELEASE3.FASTA
whole-genome_annotation-feature-region_dmel_RELEASE3.GFF
&gt; fly.gff</p>

<p style="margin-top: 1em">The gadfly.gff file and the
fasta file can now be loaded into a Bio::DB::GFF database
using the following command:</p>

<p style="margin-top: 1em">% bulk_load_gff.pl -d fly -fasta
na_whole-genome_genomic_dmel_RELEASE3.FASTA fly.gff</p>

<p style="margin-top: 1em">(Where &quot;fly&quot; is the
name of the database. Change it as appropriate. The database
must already exist and be writable by you!)</p>

<p style="margin-top: 1em">The resulting database will have
the following feature types (represented as
&quot;method:source&quot;):</p>

<p style="margin-top: 1em">Component:arm A chromosome arm
<br>
Component:scaffold A chromosome scaffold (accession #) <br>
Component:gap A gap in the assembly <br>
clone:clonelocator A BAC clone <br>
gene:gadfly A gene accession number <br>
transcript:gadfly A transcript accession number <br>
translation:gadfly A translation <br>
codon:gadfly Significance unknown <br>
exon:gadfly An exon <br>
symbol:gadfly A classical gene symbol <br>
similarity:blastn A BLASTN hit <br>
similarity:blastx A BLASTX hit <br>
similarity:sim4 EST-&gt;genome using SIM4 <br>
similarity:groupest EST-&gt;genome using GROUPEST <br>
similarity:repeatmasker A repeat</p>

<p style="margin-top: 1em">IMPORTANT NOTE: This script will
*only* work with the RELEASE3 gadfly files and will not work
with earlier releases.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Bio::DB::GFF, bulk_load_gff.pl, load_gff.pl</p>

<p style="margin-top: 1em">AUTHOR <br>
Lincoln Stein, lstein@cshl.org</p>

<p style="margin-top: 1em">Copyright (c) 2002 Cold Spring
Harbor Laboratory</p>

<p style="margin-top: 1em">This library is free software;
you can redistribute it and/or modify it under the same
terms as Perl itself. See DISCLAIMER.txt for disclaimers of
warranty.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_PROCESS_GADFLY(1p)</p>
<hr>
</body>
</html>
