<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_REVTRANS-MOTIF(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_REVTRANS-MOTIF(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_REVTRANS-MOTIF(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_revtrans-motif - Reverse translate a Profam-like protein motif
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
Version 0.01
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
From a file:
<div class="Pp"></div>
<pre>
    bp_revtrans-motif.pl -i motifs.txt
</pre>
<div class="Pp"></div>
Using pipes:
<div class="Pp"></div>
<pre>
   bp_revtrans-motif.pl &lt; motifs.txt &gt; output.txt
</pre>
<div class="Pp"></div>
Using interactively at the command prompt:
<div class="Pp"></div>
<pre>
   $ bp_revtrans-motif.pl
   MAAEEL[VIKP]
   1.   ATGGCNGCNGARGARYTNVHN
   [^P]H(IW){2,3}
   2.   NDNCAY(ATHTGG){2,3}
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script takes a protein motif as input and returns a degenerate
  oligonucleotide sequence corresponding to it. The main reason for doing this
  is to design degenerate primers that amplify a given sequence pattern.
<div class="Pp"></div>
The input motif consists of a string of one-letter residues, with any of the
  following syntactic elements:
<dl class="Bl-tag">
  <dt class="It-tag"><b>[...]</b> : Redundant position.</dt>
  <dd class="It-tag">A position in which more than one residue is allowed.
      Example:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
    [TS]YW[RKSD]
     ^^    ^^^^
    </pre>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>[^...]</b> : Negated position.</dt>
  <dd class="It-tag">A position in which any residue is allowed, saved for those
      between brackets. Example:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
    [^PW]MK[LAE]
      ^^
    </pre>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>(...){n,m,...}</b> : Repeated motif.</dt>
  <dd class="It-tag">A motif that is repeated <b>n</b> or <b>m</b> times. It can
      have any of the previous syntactic elements. Example:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
    A[SN]C(TXX){2,4,8}
           ^^^
    </pre>
  </dd>
</dl>
<div class="Pp"></div>
The allowed letters are those that correspond to the 20 natural aminoacids,
  plus:
<div class="Pp"></div>
<pre>
    B = N + D
    Z = Q + E
    X = All
</pre>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="-i_input-file:"><a class="selflink" href="#-i_input-file:"><b>-i
  input-file</b>:</a></h2>
A file with a list of motifs to reverse translate.
<h2 class="Ss" title="Ss" id="-h"><a class="selflink" href="#-h"><b>-h</b></a></h2>
Display this help message.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Bruno Vecchi, &quot;vecchi.b at gmail.com&quot;
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Please report any bugs or feature requests to &quot;vecchi.b at gmail.com&quot;
<h1 class="Sh" title="Sh" id="COPYRIGHT_&amp;_LICENSE"><a class="selflink" href="#COPYRIGHT_&amp;_LICENSE">COPYRIGHT
  &amp; LICENSE</a></h1>
Copyright 2009 Bruno Vecchi, all rights reserved.
<div class="Pp"></div>
This program is free software; you can redistribute it and/or modify it under
  the same terms as Perl itself.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
