<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_SEQFEATURE_LOAD(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_SEQFEATURE_LOAD(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_SEQFEATURE_LOAD(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_seqfeature_load.pl - Load GFF into a SeqFeature database
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Pass any number of GFF or fasta format files (or GFF with embedded fasta) to
  load the features and sequences into a SeqFeature database. The database (and
  adaptor) to use is specified on the command line. Use the --create flag to
  create a new SeqFeature database.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
 bp_seqfeature_load.pl [options] gff_or_fasta_file1 [gff_or_fasta_file2 [...]]
</pre>
<div class="Pp"></div>
Try 'bp_seqfeature_load.pl --help' or '--man' for more information.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">-d, --dsn</dt>
  <dd class="It-tag">DBI data source (default dbi:mysql:test)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-n, --namespace</dt>
  <dd class="It-tag">The table prefix to use (default undef) Allows several
      independent sequence feature databases to be stored in a single
    database</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-s, --seqfeature</dt>
  <dd class="It-tag">The type of SeqFeature to create... RTSC (default
      Bio::DB::SeqFeature)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-a, --adaptor</dt>
  <dd class="It-tag">The storage adaptor (class) to use (default
    DBI::mysql)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-v, --verbose</dt>
  <dd class="It-tag">Turn on verbose progress reporting (default true) Use
      --noverbose to switch this off.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-f, --fast</dt>
  <dd class="It-tag">Activate fast loading. (default 0) Only available for some
      adaptors.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-T, --temporary-directory</dt>
  <dd class="It-tag">Specify temporary directory for fast loading (default
      File::Spec-&gt; <i>tmpdir()</i>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-i, --ignore-seqregion</dt>
  <dd class="It-tag">If true, then ignore ##sequence-region directives in the
      GFF3 file (default, create a feature for each region)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-c, --create</dt>
  <dd class="It-tag">Create the database and reinitialize it (default false)
      Note, this will erase previous database contents, if any.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-u, --user</dt>
  <dd class="It-tag">User to connect to database as</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-p, --password</dt>
  <dd class="It-tag">Password to use to connect to database</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-z, --zip</dt>
  <dd class="It-tag">Compress database tables to save space (default false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-S, --subfeatures</dt>
  <dd class="It-tag">Turn on indexing of subfeatures (default true) Use
      --nosubfeatures to switch this off.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--fts</dt>
  <dd class="It-tag">Index the attribute table for full-text search (default
      false). Applicable only when --create is specified. Currently applicable
      to the DBI::SQLite storage adaptor only (using the most recent supported
      FTS indexing method, which may not be portable to older DBI::SQLite
      versions).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--summary</dt>
  <dd class="It-tag">Generate summary statistics for coverage graphs (default
      false) This can be run on a previously loaded database or during the load.
      It will default to true if --create is used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-N, --nosummary</dt>
  <dd class="It-tag">Do not generate summary statistics to save some space and
      load time (default if --create is not specified, use this option to
      explicitly turn off summary statistics when --create is specified)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--noalias-target</dt>
  <dd class="It-tag">Don't create an Alias attribute whose value is the
      target_id in a Target attribute (if the feature contains a Target
      attribute, the default is to create an Alias attribute whose value is the
      target_id in the Target attribute)</dd>
</dl>
<div class="Pp"></div>
Please see http://www.sequenceontology.org/gff3.shtml for information about the
  GFF3 format. BioPerl extends the format slightly by adding a
  ##index-subfeatures directive. Set this to a true value if you wish the
  database to be able to retrieve a feature's individual parts (such as the
  exons of a transcript) independently of the top level feature:
<div class="Pp"></div>
<pre>
  ##index-subfeatures 1
</pre>
<div class="Pp"></div>
It is also possible to control the indexing of subfeatures on a case-by-case
  basis by adding &quot;index=1&quot; or &quot;index=0&quot; to the feature's
  attribute list. This should only be used for subfeatures.
<div class="Pp"></div>
Subfeature indexing is true by default. Set to false (0) to save lots of
  database space and speed performance. You may use --nosubfeatures to force
  this.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
