<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:05 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_SEARCH2ALNBLOCKS(1p) User Contributed Perl
Documentation BP_SEARCH2ALNBLOCKS(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_search2alnblocks - Turn SearchIO parseable reports(s)
into a set of aligned blocks</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bp_search2alnblocks --minid PERCENTID --minlen LEN
--minevalue EVALUE file1. <br>
blast file2.blast ...&gt; out.fas</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script will parse and filter BLAST (or other formats
Bio::SearchIO can parse) output and format the alignment as
blocks of alignments based on the HSPs. Note this can only
<br>
work if the input file parsed contains the necessary.</p>

<p style="margin-top: 1em">Typically this can be used to
turn BLAST output into a FASTA alignment format for input
into the QRNA comparative gene finder for RNA genes
(E.Rivas).</p>

<p style="margin-top: 1em">OPTIONS <br>
--maxevalue Maximum E-value for an HSP <br>
--minevalue Minimum E-value for an HSP <br>
--minlen Minimum length of an HSP [default 0] <br>
--maxid Maximum Percent Id [default 100] <br>
(to help remove sequences which are really close) <br>
--minid Minimum Percent Identity for an HSP [default 0] <br>
-i/--input An optional input filename (expects input on
STDIN by default) <br>
-o/--output An optional output filename (exports to STDOUT
by default) <br>
-f/--format Specify a different Search Alignment format-
<br>
{fasta, axt, waba, blast, blastxml} are all permitted <br>
although the format must have actual alignment <br>
sequence for this script to work <br>
See L&lt;Bio::SearchIO&gt; for more information. <br>
-of/--outformat Output format for the alignment blocks,
anything <br>
L&lt;Bio::AlignIO&gt; supports. <br>
-v/--verbose Turn on debugging</p>

<p style="margin-top: 1em">AUTHOR - Jason Stajich <br>
Jason Stajich, jason-at-bioperl-dot-org.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_SEARCH2ALNBLOCKS(1p)</p>
<hr>
</body>
</html>
