<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:06 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_SEQFEATURE_LOAD(1p) User Contributed Perl
Documentation BP_SEQFEATURE_LOAD(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_seqfeature_load.pl - Load GFF into a SeqFeature
database</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Pass any number of GFF or fasta format files (or GFF with
embedded fasta) to load the features and sequences into a
SeqFeature database. The database (and adaptor) to use is
<br>
specified on the command line. Use the --create flag to
create a new SeqFeature database.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bp_seqfeature_load.pl [options] gff_or_fasta_file1
[gff_or_fasta_file2 [...]]</p>

<p style="margin-top: 1em">Try &rsquo;bp_seqfeature_load.pl
--help&rsquo; or &rsquo;--man&rsquo; for more
information.</p>

<p style="margin-top: 1em">OPTIONS <br>
-d, --dsn <br>
DBI data source (default dbi:mysql:test)</p>

<p style="margin-top: 1em">-n, --namespace <br>
The table prefix to use (default undef) Allows several
independent sequence feature databases to be stored in a
single database</p>

<p style="margin-top: 1em">-s, --seqfeature <br>
The type of SeqFeature to create... RTSC (default
Bio::DB::SeqFeature)</p>

<p style="margin-top: 1em">-a, --adaptor <br>
The storage adaptor (class) to use (default DBI::mysql)</p>

<p style="margin-top: 1em">-v, --verbose <br>
Turn on verbose progress reporting (default true) Use
--noverbose to switch this off.</p>

<p style="margin-top: 1em">-f, --fast <br>
Activate fast loading. (default 0) Only available for some
adaptors.</p>

<p style="margin-top: 1em">-T, --temporary-directory <br>
Specify temporary directory for fast loading (default
File::Spec-&gt;tmpdir())</p>

<p style="margin-top: 1em">-i, --ignore-seqregion <br>
If true, then ignore ##sequence-region directives in the
GFF3 file (default, create a feature for each region)</p>

<p style="margin-top: 1em">-c, --create <br>
Create the database and reinitialize it (default false)
Note, this will erase previous database contents, if
any.</p>

<p style="margin-top: 1em">-u, --user <br>
User to connect to database as</p>

<p style="margin-top: 1em">-p, --password <br>
Password to use to connect to database</p>

<p style="margin-top: 1em">-z, --zip <br>
Compress database tables to save space (default false)</p>

<p style="margin-top: 1em">-S, --subfeatures <br>
Turn on indexing of subfeatures (default true) Use
--nosubfeatures to switch this off.</p>

<p style="margin-top: 1em">--fts <br>
Index the attribute table for full-text search (default
false). Applicable only when --create is specified.
Currently applicable to the DBI::SQLite storage adaptor only
<br>
(using the most recent supported FTS indexing method, which
may not be portable to older DBI::SQLite versions).</p>

<p style="margin-top: 1em">--summary <br>
Generate summary statistics for coverage graphs (default
false) This can be run on a previously loaded database or
during the load. It will default to true if --create is <br>
used.</p>

<p style="margin-top: 1em">-N, --nosummary <br>
Do not generate summary statistics to save some space and
load time (default if --create is not specified, use this
option to explicitly turn off summary statistics when <br>
--create is specified)</p>

<p style="margin-top: 1em">--noalias-target <br>
Don&rsquo;t create an Alias attribute whose value is the
target_id in a Target attribute (if the feature contains a
Target attribute, the default is to create an Alias
attribute <br>
whose value is the target_id in the Target attribute)</p>

<p style="margin-top: 1em">Please see
http://www.sequenceontology.org/gff3.shtml for information
about the GFF3 format. BioPerl extends the format slightly
by adding a ##index-subfeatures directive. Set <br>
this to a true value if you wish the database to be able to
retrieve a feature&rsquo;s individual parts (such as the
exons of a transcript) independently of the top level
feature:</p>

<p style="margin-top: 1em">##index-subfeatures 1</p>

<p style="margin-top: 1em">It is also possible to control
the indexing of subfeatures on a case-by-case basis by
adding &quot;index=1&quot; or &quot;index=0&quot; to the
feature&rsquo;s attribute list. This should only be used for
<br>
subfeatures.</p>

<p style="margin-top: 1em">Subfeature indexing is true by
default. Set to false (0) to save lots of database space and
speed performance. You may use --nosubfeatures to force
this.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_SEQFEATURE_LOAD(1p)</p>
<hr>
</body>
</html>
