<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_SEARCH2GFF(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_SEARCH2GFF(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_SEARCH2GFF(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_search2gff
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
Usage:
<div class="Pp"></div>
<pre>
  bp_search2gff [-o outputfile] [-f reportformat] [-i inputfilename]  OR file1 file2 ..
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script will turn a SearchIO report (BLAST, FASTP, SSEARCH, AXT, WABA) into
  GFF.
<div class="Pp"></div>
The options are:
<div class="Pp"></div>
<pre>
   -i infilename      - (optional) inputfilename, will read
                        either ARGV files or from STDIN
   -o filename        - the output filename [default STDOUT]
   -f format          - search result format (blast, fasta,waba,axt)
                        (ssearch is fasta format). default is blast.
   -t/--type seqtype  - if you want to see query or hit information
                        in the GFF report
   -s/--source        - specify the source (will be algorithm name
                        otherwise like BLASTN)
   --method           - the method tag (primary_tag) of the features
                        (default is similarity)
   --scorefunc        - a string or a file that when parsed evaluates
                        to a closure which will be passed a feature
                        object and that returns the score to be printed
   --locfunc          - a string or a file that when parsed evaluates
                        to a closure which will be passed two
                        features, query and hit, and returns the
                        location (Bio::LocationI compliant) for the
                        GFF3 feature created for each HSP; the closure
                        may use the clone_loc() and create_loc()
                        functions for convenience, see their PODs
   --onehsp           - only print the first HSP feature for each hit
   -p/--parent        - the parent to which HSP features should refer
                        if not the name of the hit or query (depending
                        on --type)
   --target/--notarget - whether to always add the Target tag or not
   -h                 - this help menu
   --version          - GFF version to use (put a 3 here to use gff 3)
   --component        - generate GFF component fields (chromosome)
   -m/--match         - generate a 'match' line which is a container
                        of all the similarity HSPs
   --addid            - add ID tag in the absence of --match
   -c/--cutoff        - specify an evalue cutoff
</pre>
<div class="Pp"></div>
Additionally specify the filenames you want to process on the command-line. If
  no files are specified then STDIN input is assumed. You specify this by doing:
  bp_search2gff &lt; file1 file2 file3
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Jason Stajich, jason-at-bioperl-dot-org
<h1 class="Sh" title="Sh" id="Contributors"><a class="selflink" href="#Contributors">Contributors</a></h1>
Hilmar Lapp, hlapp-at-gmx-dot-net
<h2 class="Ss" title="Ss" id="clone_loc"><a class="selflink" href="#clone_loc">clone_loc</a></h2>
Title : clone_loc Usage : my $l = clone_loc($feature-&gt;location); Function:
  Helper function to simplify the task of cloning locations
<br/>
 for --locfunc closures.
<div class="Pp"></div>
<pre>
          Presently simply implemented using Storable::dclone().
Example :
Returns : A L&lt;Bio::LocationI&gt; object of the same type and with the
          same properties as the argument, but physically different.
          All structured properties will be cloned as well.
Args    : A L&lt;Bio::LocationI&gt; compliant object
</pre>
<h2 class="Ss" title="Ss" id="create_loc"><a class="selflink" href="#create_loc">create_loc</a></h2>
Title : create_loc Usage : my $l = create_loc(&quot;10..12&quot;); Function:
  Helper function to simplify the task of creating locations
<br/>
 for --locfunc closures. Creates a location from a feature-
<br/>
 table formatted string.
<div class="Pp"></div>
Example : Returns : A Bio::LocationI object representing the location given
<br/>
 as formatted string. Args : A GenBank feature-table formatted string.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
