<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:05 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_SEARCH2GFF(1p) User Contributed Perl Documentation
BP_SEARCH2GFF(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_search2gff</p>

<p style="margin-top: 1em">SYNOPSIS <br>
Usage:</p>

<p style="margin-top: 1em">bp_search2gff [-o outputfile]
[-f reportformat] [-i inputfilename] OR file1 file2 ..</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script will turn a SearchIO report (BLAST, FASTP,
SSEARCH, AXT, WABA) into GFF.</p>

<p style="margin-top: 1em">The options are:</p>

<p style="margin-top: 1em">-i infilename - (optional)
inputfilename, will read <br>
either ARGV files or from STDIN <br>
-o filename - the output filename [default STDOUT] <br>
-f format - search result format (blast, fasta,waba,axt)
<br>
(ssearch is fasta format). default is blast. <br>
-t/--type seqtype - if you want to see query or hit
information <br>
in the GFF report <br>
-s/--source - specify the source (will be algorithm name
<br>
otherwise like BLASTN) <br>
--method - the method tag (primary_tag) of the features <br>
(default is similarity) <br>
--scorefunc - a string or a file that when parsed evaluates
<br>
to a closure which will be passed a feature <br>
object and that returns the score to be printed <br>
--locfunc - a string or a file that when parsed evaluates
<br>
to a closure which will be passed two <br>
features, query and hit, and returns the <br>
location (Bio::LocationI compliant) for the <br>
GFF3 feature created for each HSP; the closure <br>
may use the clone_loc() and create_loc() <br>
functions for convenience, see their PODs <br>
--onehsp - only print the first HSP feature for each hit
<br>
-p/--parent - the parent to which HSP features should refer
<br>
if not the name of the hit or query (depending <br>
on --type) <br>
--target/--notarget - whether to always add the Target tag
or not <br>
-h - this help menu <br>
--version - GFF version to use (put a 3 here to use gff 3)
<br>
--component - generate GFF component fields (chromosome)
<br>
-m/--match - generate a &rsquo;match&rsquo; line which is a
container <br>
of all the similarity HSPs <br>
--addid - add ID tag in the absence of --match <br>
-c/--cutoff - specify an evalue cutoff</p>

<p style="margin-top: 1em">Additionally specify the
filenames you want to process on the command-line. If no
files are specified then STDIN input is assumed. You specify
this by doing: bp_search2gff &lt; <br>
file1 file2 file3</p>

<p style="margin-top: 1em">AUTHOR <br>
Jason Stajich, jason-at-bioperl-dot-org</p>

<p style="margin-top: 1em">Contributors <br>
Hilmar Lapp, hlapp-at-gmx-dot-net</p>

<p style="margin-top: 1em">clone_loc <br>
Title : clone_loc Usage : my $l =
clone_loc($feature-&gt;location); Function: Helper function
to simplify the task of cloning locations <br>
for --locfunc closures.</p>

<p style="margin-top: 1em">Presently simply implemented
using Storable::dclone(). <br>
Example : <br>
Returns : A L&lt;Bio::LocationI&gt; object of the same type
and with the <br>
same properties as the argument, but physically different.
<br>
All structured properties will be cloned as well. <br>
Args : A L&lt;Bio::LocationI&gt; compliant object</p>

<p style="margin-top: 1em">create_loc <br>
Title : create_loc Usage : my $l =
create_loc(&quot;10..12&quot;); Function: Helper function to
simplify the task of creating locations <br>
for --locfunc closures. Creates a location from a feature-
<br>
table formatted string.</p>

<p style="margin-top: 1em">Example : Returns : A
Bio::LocationI object representing the location given <br>
as formatted string. Args : A GenBank feature-table
formatted string.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_SEARCH2GFF(1p)</p>
<hr>
</body>
</html>
