<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_SEARCH2ALNBLOCKS(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_SEARCH2ALNBLOCKS(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_SEARCH2ALNBLOCKS(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_search2alnblocks - Turn SearchIO parseable reports(s) into a set of aligned
  blocks
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  bp_search2alnblocks --minid PERCENTID --minlen LEN --minevalue EVALUE file1.
  blast file2.blast ...&gt; out.fas
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script will parse and filter BLAST (or other formats Bio::SearchIO can
  parse) output and format the alignment as blocks of alignments based on the
  HSPs. Note this can only work if the input file parsed contains the necessary.
<div class="Pp"></div>
Typically this can be used to turn BLAST output into a FASTA alignment format
  for input into the QRNA comparative gene finder for RNA genes (E.Rivas).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<pre>
 --maxevalue   Maximum E-value for an HSP
 --minevalue   Minimum E-value for an HSP 
 --minlen      Minimum length of an HSP [default 0] 
 --maxid       Maximum Percent Id [default 100]
               (to help remove sequences which are really close)
 --minid       Minimum Percent Identity for an HSP [default 0]
 -i/--input    An optional input filename (expects input on STDIN by default)
 -o/--output   An optional output filename (exports to STDOUT by default)
 -f/--format   Specify a different Search Alignment format- 
               {fasta, axt, waba, blast, blastxml} are all permitted
               although the format must have actual alignment 
               sequence for this script to work
               See L&lt;Bio::SearchIO&gt; for more information.
 -of/--outformat Output format for the alignment blocks, anything
               L&lt;Bio::AlignIO&gt; supports.
 -v/--verbose  Turn on debugging
</pre>
<h1 class="Sh" title="Sh" id="AUTHOR_-_Jason_Stajich"><a class="selflink" href="#AUTHOR_-_Jason_Stajich">AUTHOR
  - Jason Stajich</a></h1>
Jason Stajich, jason-at-bioperl-dot-org.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
