<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:08 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>BP_UNFLATTEN_SEQ(1p) User Contributed Perl Documentation
BP_UNFLATTEN_SEQ(1p)</p>

<p style="margin-top: 1em">NAME <br>
bp_unflatten_seq - unflatten a genbank or genbank-style
feature file into a nested SeqFeature hierarchy</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bp_unflatten_seq.PLS -e 3 -gff
~/cvs/bioperl-live/t/data/AE003644_Adh-genomic.gb</p>

<p style="margin-top: 1em">bp_unflatten_seq.PLS --detail
~/cvs/bioperl-live/t/data/AE003644_Adh-genomic.gb</p>

<p style="margin-top: 1em">bp_unflatten_seq.PLS -i foo.embl
--from embl --to chadoxml -o out.chado.xml</p>

<p style="margin-top: 1em">bp_unflatten_seq.PLS --notypemap
--detail --to asciitree -ethresh 2
AE003644_Adh-genomic.gb</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script will unflatten a genbank or genbank-style file
of SeqFeatures into a nested hierarchy.</p>

<p style="margin-top: 1em">See
Bio::SeqFeature::Tools::Unflattener</p>

<p style="margin-top: 1em">In a GenBank/EMBL
representation, features are &rsquo;flat&rsquo; - for
example, there is no link between an mRNA and a CDS, other
than implicit links (eg via tags or via splice site <br>
coordinates) which may be hard to code for.</p>

<p style="margin-top: 1em">This is most easily illustrated
with the default output format, asciitree</p>

<p style="margin-top: 1em">An unflattened genbank feature
set may look like this (AB077698)</p>

<p style="margin-top: 1em">Seq: AB077698 <br>
databank_entry 1..2701[+] <br>
gene <br>
mRNA <br>
CDS hCHCR-G 80..1144[+] <br>
exon 80..1144[+] <br>
five_prime_UTR 1..79[+] <br>
located_sequence_feature 137..196[+] <br>
located_sequence_feature 239..292[+] <br>
located_sequence_feature 617..676[+] <br>
located_sequence_feature 725..778[+] <br>
three_prime_UTR 1145..2659[+] <br>
polyA_site 1606..1606[+] <br>
polyA_site 2660..2660[+]</p>

<p style="margin-top: 1em">Or like this (portion of
AE003734)</p>

<p style="margin-top: 1em">gene <br>
mRNA CG3320-RA <br>
CDS CG3320-PA 53126..54971[-] <br>
exon 52204..53323[-] <br>
exon 53404..53631[-] <br>
exon 53688..53735[-] <br>
exon 53798..53918[-] <br>
exon 54949..55287[-] <br>
mRNA CG3320-RB <br>
CDS CG3320-PB 53383..54971[-] <br>
exon 52204..53631[-] <br>
exon 53688..53735[-] <br>
exon 53798..53918[-] <br>
exon 54949..55287[-]</p>

<p style="margin-top: 1em">The unflattening will also
&rsquo;normalize&rsquo; the containment hierarchy (in the
sense of standardising it - e.g. making sure there is always
a transcript record, even if genbank just <br>
specifies CDS and gene)</p>

<p style="margin-top: 1em">By default, the GenBank types
will be mapped to SO types</p>

<p style="margin-top: 1em">See
Bio::SeqFeature::Tools::TypeMapper</p>

<p style="margin-top: 1em">COMMAND LINE ARGUMENTS <br>
-i|input FILE <br>
input file (can also be specified as last argument)</p>

<p style="margin-top: 1em">-from FORMAT <br>
input format (defaults to genbank)</p>

<p style="margin-top: 1em">probably doesnt make so much
sense to use this for non-flat formats; ie other than
embl/genbank</p>

<p style="margin-top: 1em">-to FORMAT <br>
output format (defaults to asciitree)</p>

<p style="margin-top: 1em">should really be a format that
is nested SeqFeature aware; I think this is only asciitree,
chadoxml and gff3</p>

<p style="margin-top: 1em">-gff <br>
with export to GFF3 format (pre-3 GFFs make no sense with
unflattened sequences, as they have no set way of
representing feature graphs)</p>

<p style="margin-top: 1em">-o|output FILE <br>
outfile defaults to STDOUT</p>

<p style="margin-top: 1em">-detail <br>
show extra detail on features (asciitree mode only)</p>

<p style="margin-top: 1em">-e|ethresh INT <br>
sets the error threshold on unflattening</p>

<p style="margin-top: 1em">by default this script will
throw a wobbly if it encounters weird stuff in the genbank
file - raise the error threshold to signal these to be
ignored (and reported on STDERR)</p>

<p style="margin-top: 1em">-nomagic <br>
suppress use_magic in unflattening (see
Bio::SeqFeature::Tools::Unflattener</p>

<p style="margin-top: 1em">-notypemap <br>
suppress type mapping (see
Bio::SeqFeature::Tools::TypeMapper</p>

<p style="margin-top: 1em">TODO <br>
Bio::SeqFeature::Tools::Unflattener allows fine-grained
control over the unflattening process - need to add more
options to allow this control at the command line</p>

<p style="margin-top: 1em">FEEDBACK <br>
Mailing Lists <br>
User feedback is an integral part of the evolution of this
and other Bioperl modules. Send your comments and
suggestions preferably to the Bioperl mailing list. Your
<br>
participation is much appreciated.</p>

<p style="margin-top: 1em">bioperl-l@bioperl.org - General
discussion <br>
http://bioperl.org/wiki/Mailing_lists - About the mailing
lists</p>

<p style="margin-top: 1em">Reporting Bugs <br>
Report bugs to the Bioperl bug tracking system to help us
keep track of the bugs and their resolution. Bug reports can
be submitted via email or the web:</p>


<p style="margin-top: 1em">https://github.com/bioperl/bioperl-live/issues</p>

<p style="margin-top: 1em">AUTHOR <br>
Chris Mungall E&lt;lt&gt;cjm-at-bioperl.orgE&lt;gt&gt;</p>

<p style="margin-top: 1em">perl v5.24.1 2017-01-15
BP_UNFLATTEN_SEQ(1p)</p>
<hr>
</body>
</html>
