<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>BP_UNFLATTEN_SEQ(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">BP_UNFLATTEN_SEQ(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">BP_UNFLATTEN_SEQ(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
bp_unflatten_seq - unflatten a genbank or genbank-style feature file into a
  nested SeqFeature hierarchy
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  bp_unflatten_seq.PLS -e 3 -gff ~/cvs/bioperl-live/t/data/AE003644_Adh-genomic.gb
  bp_unflatten_seq.PLS --detail ~/cvs/bioperl-live/t/data/AE003644_Adh-genomic.gb
  bp_unflatten_seq.PLS -i foo.embl --from embl --to chadoxml -o out.chado.xml
  bp_unflatten_seq.PLS --notypemap --detail --to asciitree -ethresh 2 AE003644_Adh-genomic.gb
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script will <b>unflatten</b> a genbank or genbank-style file of SeqFeatures
  into a nested hierarchy.
<div class="Pp"></div>
See Bio::SeqFeature::Tools::Unflattener
<div class="Pp"></div>
In a GenBank/EMBL representation, features are 'flat' - for example, there is no
  link between an mRNA and a CDS, other than implicit links (eg via tags or via
  splice site coordinates) which may be hard to code for.
<div class="Pp"></div>
This is most easily illustrated with the default output format, <b>asciitree</b>
<div class="Pp"></div>
An unflattened genbank feature set may look like this (AB077698)
<div class="Pp"></div>
<pre>
  Seq: AB077698
    databank_entry                                   1..2701[+]
    gene                                             
      mRNA                                           
        CDS hCHCR-G                                  80..1144[+]
        exon                                         80..1144[+]
      five_prime_UTR                                 1..79[+]
      located_sequence_feature                       137..196[+]
      located_sequence_feature                       239..292[+]
      located_sequence_feature                       617..676[+]
      located_sequence_feature                       725..778[+]
      three_prime_UTR                                1145..2659[+]
      polyA_site                                     1606..1606[+]
      polyA_site                                     2660..2660[+]
</pre>
<div class="Pp"></div>
Or like this (portion of AE003734)
<div class="Pp"></div>
<pre>
  gene                                             
    mRNA CG3320-RA                                 
      CDS CG3320-PA                                53126..54971[-]
      exon                                         52204..53323[-]
      exon                                         53404..53631[-]
      exon                                         53688..53735[-]
      exon                                         53798..53918[-]
      exon                                         54949..55287[-]
    mRNA CG3320-RB                                 
      CDS CG3320-PB                                53383..54971[-]
      exon                                         52204..53631[-]
      exon                                         53688..53735[-]
      exon                                         53798..53918[-]
      exon                                         54949..55287[-]
</pre>
<div class="Pp"></div>
The unflattening will also 'normalize' the containment hierarchy (in the sense
  of standardising it - e.g. making sure there is always a transcript record,
  even if genbank just specifies CDS and gene)
<div class="Pp"></div>
By default, the GenBank types will be mapped to SO types
<div class="Pp"></div>
See Bio::SeqFeature::Tools::TypeMapper
<h1 class="Sh" title="Sh" id="COMMAND_LINE_ARGUMENTS"><a class="selflink" href="#COMMAND_LINE_ARGUMENTS">COMMAND
  LINE ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">-i|input FILE</dt>
  <dd class="It-tag">input file (can also be specified as last argument)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-from FORMAT</dt>
  <dd class="It-tag">input format (defaults to genbank)
    <div style="height: 1.00em;">&#x00A0;</div>
    probably doesnt make so much sense to use this for non-flat formats; ie
      other than embl/genbank</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-to FORMAT</dt>
  <dd class="It-tag">output format (defaults to asciitree)
    <div style="height: 1.00em;">&#x00A0;</div>
    should really be a format that is nested SeqFeature aware; I think this is
      only asciitree, chadoxml and gff3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-gff</dt>
  <dd class="It-tag">with export to GFF3 format (pre-3 GFFs make no sense with
      unflattened sequences, as they have no set way of representing feature
      graphs)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-o|output FILE</dt>
  <dd class="It-tag">outfile defaults to STDOUT</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-detail</dt>
  <dd class="It-tag">show extra detail on features (asciitree mode only)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-e|ethresh INT</dt>
  <dd class="It-tag">sets the error threshold on unflattening
    <div style="height: 1.00em;">&#x00A0;</div>
    by default this script will throw a wobbly if it encounters weird stuff in
      the genbank file - raise the error threshold to signal these to be ignored
      (and reported on STDERR)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-nomagic</dt>
  <dd class="It-tag">suppress use_magic in unflattening (see
      Bio::SeqFeature::Tools::Unflattener</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-notypemap</dt>
  <dd class="It-tag">suppress type mapping (see
      Bio::SeqFeature::Tools::TypeMapper</dd>
</dl>
<h1 class="Sh" title="Sh" id="TODO"><a class="selflink" href="#TODO">TODO</a></h1>
Bio::SeqFeature::Tools::Unflattener allows fine-grained control over the
  unflattening process - need to add more options to allow this control at the
  command line
<h1 class="Sh" title="Sh" id="FEEDBACK"><a class="selflink" href="#FEEDBACK">FEEDBACK</a></h1>
<h2 class="Ss" title="Ss" id="Mailing_Lists"><a class="selflink" href="#Mailing_Lists">Mailing
  Lists</a></h2>
User feedback is an integral part of the evolution of this and other Bioperl
  modules. Send your comments and suggestions preferably to the Bioperl mailing
  list. Your participation is much appreciated.
<div class="Pp"></div>
<pre>
  bioperl-l@bioperl.org                  - General discussion
  http://bioperl.org/wiki/Mailing_lists  - About the mailing lists
</pre>
<h2 class="Ss" title="Ss" id="Reporting_Bugs"><a class="selflink" href="#Reporting_Bugs">Reporting
  Bugs</a></h2>
Report bugs to the Bioperl bug tracking system to help us keep track of the bugs
  and their resolution. Bug reports can be submitted via email or the web:
<div class="Pp"></div>
<pre>
  https://github.com/bioperl/bioperl-live/issues
</pre>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
 Chris Mungall E&lt;lt&gt;cjm-at-bioperl.orgE&lt;gt&gt;
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-01-15</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
