<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:56:56 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>bwa(1) Bioinformatics tools bwa(1)</p>

<p style="margin-top: 1em">NAME <br>
bwa - Burrows-Wheeler Alignment Tool</p>

<p style="margin-top: 1em">SYNOPSIS <br>
bwa index ref.fa</p>

<p style="margin-top: 1em">bwa mem ref.fa reads.fq &gt;
aln-se.sam</p>

<p style="margin-top: 1em">bwa mem ref.fa read1.fq read2.fq
&gt; aln-pe.sam</p>

<p style="margin-top: 1em">bwa aln ref.fa short_read.fq
&gt; aln_sa.sai</p>

<p style="margin-top: 1em">bwa samse ref.fa aln_sa.sai
short_read.fq &gt; aln-se.sam</p>

<p style="margin-top: 1em">bwa sampe ref.fa aln_sa1.sai
aln_sa2.sai read1.fq read2.fq &gt; aln-pe.sam</p>

<p style="margin-top: 1em">bwa bwasw ref.fa long_read.fq
&gt; aln.sam</p>

<p style="margin-top: 1em">DESCRIPTION <br>
BWA is a software package for mapping low-divergent
sequences against a large reference genome, such as the
human genome. It consists of three algorithms:
BWA-backtrack, BWA-SW <br>
and BWA-MEM. The first algorithm is designed for Illumina
sequence reads up to 100bp, while the rest two for longer
sequences ranged from 70bp to 1Mbp. BWA-MEM and BWA-SW share
<br>
similar features such as long-read support and split
alignment, but BWA-MEM, which is the latest, is generally
recommended for high-quality queries as it is faster and
more accu&acirc; <br>
rate. BWA-MEM also has better performance than BWA-backtrack
for 70-100bp Illumina reads.</p>

<p style="margin-top: 1em">For all the algorithms, BWA
first needs to construct the FM-index for the reference
genome (the index command). Alignment algorithms are invoked
with different sub-commands: <br>
aln/samse/sampe for BWA-backtrack, bwasw for BWA-SW and mem
for the BWA-MEM algorithm.</p>

<p style="margin-top: 1em">COMMANDS AND OPTIONS <br>
index bwa index [-p prefix] [-a algoType] db.fa</p>

<p style="margin-top: 1em">Index database sequences in the
FASTA format.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-p STR Prefix of the output
database [same as db filename]</p>

<p style="margin-top: 1em">-a STR Algorithm for
constructing BWT index. BWA implements three algorithms for
BWT construction: is, bwtsw and rb2. The first algorithm is
a little faster for small <br>
database but requires large RAM and does not work for
databases with total length longer than 2GB. The second
algorithm is adapted from the BWT-SW source code. <br>
It in theory works with database with trillions of bases.
When this option is not specified, the appropriate algorithm
will be chosen automatically.</p>

<p style="margin-top: 1em">mem bwa mem [-aCHjMpP] [-t
nThreads] [-k minSeedLen] [-w bandWidth] [-d zDropoff] [-r
seedSplitRatio] [-c maxOcc] [-D chainShadow] [-m maxMateSW]
[-W minSeedMatch] [-A match&acirc; <br>
Score] [-B mmPenalty] [-O gapOpenPen] [-E gapExtPen] [-L
clipPen] [-U unpairPen] [-R RGline] [-H HDlines] [-v
verboseLevel] db.prefix reads.fq [mates.fq]</p>

<p style="margin-top: 1em">Align 70bp-1Mbp query sequences
with the BWA-MEM algorithm. Briefly, the algorithm works by
seeding alignments with maximal exact matches (MEMs) and
then extending seeds <br>
with the affine-gap Smith-Waterman algorithm (SW).</p>

<p style="margin-top: 1em">If mates.fq file is absent and
option -p is not set, this command regards input reads are
single-end. If mates.fq is present, this command assumes the
i-th read in <br>
reads.fq and the i-th read in mates.fq constitute a read
pair. If -p is used, the command assumes the 2i-th and the
(2i+1)-th read in reads.fq constitute a read pair (such <br>
input file is said to be interleaved). In this case,
mates.fq is ignored. In the paired-end mode, the mem command
will infer the read orientation and the insert size
dis&acirc; <br>
tribution from a batch of reads.</p>

<p style="margin-top: 1em">The BWA-MEM algorithm performs
local alignment. It may produce multiple primary alignments
for different part of a query sequence. This is a crucial
feature for long <br>
sequences. However, some tools such as Picard&rsquo;s
markDuplicates does not work with split alignments. One may
consider to use option -M to flag shorter split hits as
sec&acirc; <br>
ondary.</p>

<p style="margin-top: 1em">ALGORITHM OPTIONS:</p>

<p style="margin-top: 1em">-t INT Number of threads [1]</p>

<p style="margin-top: 1em">-k INT Minimum seed length.
Matches shorter than INT will be missed. The alignment speed
is usually insensitive to this value unless it significantly
deviates from 20. <br>
[19]</p>

<p style="margin-top: 1em">-w INT Band width. Essentially,
gaps longer than INT will not be found. Note that the
maximum gap length is also affected by the scoring matrix
and the hit length, not <br>
solely determined by this option. [100]</p>

<p style="margin-top: 1em">-d INT Off-diagonal X-dropoff
(Z-dropoff). Stop extension when the difference between the
best and the current extension score is above |i-j|*A+INT,
where i and j are <br>
the current positions of the query and reference,
respectively, and A is the matching score. Z-dropoff is
similar to BLAST&rsquo;s X-dropoff except that it
doesn&rsquo;t <br>
penalize gaps in one of the sequences in the alignment.
Z-dropoff not only avoids unnecessary extension, but also
reduces poor alignments inside a long good <br>
alignment. [100]</p>

<p style="margin-top: 1em">-r FLOAT Trigger re-seeding for
a MEM longer than minSeedLen*FLOAT. This is a key heuristic
parameter for tuning the performance. Larger value yields
fewer seeds, which <br>
leads to faster alignment speed but lower accuracy.
[1.5]</p>

<p style="margin-top: 1em">-c INT Discard a MEM if it has
more than INT occurence in the genome. This is an
insensitive parameter. [500]</p>

<p style="margin-top: 1em">-D FLOAT Drop chains shorter
than FLOAT fraction of the longest overlapping chain
[0.5]</p>

<p style="margin-top: 1em">-m INT Perform at most INT
rounds of mate-SW [50]</p>

<p style="margin-top: 1em">-W INT Drop a chain if the
number of bases in seeds is smaller than INT. This option is
primarily used for longer contigs/reads. When positive, it
also affects seed <br>
filtering. [0]</p>

<p style="margin-top: 1em">-P In the paired-end mode,
perform SW to rescue missing hits only but do not try to
find hits that fit a proper pair.</p>

<p style="margin-top: 1em">SCORING OPTIONS:</p>

<p style="margin-top: 1em">-A INT Matching score. [1]</p>

<p style="margin-top: 1em">-B INT Mismatch penalty. The
sequence error rate is approximately: {.75 * exp[-log(4) *
B/A]}. [4]</p>

<p style="margin-top: 1em">-O INT[,INT] <br>
Gap open penalty. If two numbers are specified, the first is
the penalty of openning a deletion and the second for
openning an insertion. [6]</p>

<p style="margin-top: 1em">-E INT[,INT] <br>
Gap extension penalty. If two numbers are specified, the
first is the penalty of extending a deletion and second for
extending an insertion. A gap of length k <br>
costs O + k*E (i.e. -O is for opening a zero-length gap).
[1]</p>

<p style="margin-top: 1em">-L INT[,INT] <br>
Clipping penalty. When performing SW extension, BWA-MEM
keeps track of the best score reaching the end of query. If
this score is larger than the best SW score <br>
minus the clipping penalty, clipping will not be applied.
Note that in this case, the SAM AS tag reports the best SW
score; clipping penalty is not deduced. If <br>
two numbers are provided, the first is for 5&rsquo;-end
clipping and second for 3&rsquo;-end clipping. [5]</p>

<p style="margin-top: 1em">-U INT Penalty for an unpaired
read pair. BWA-MEM scores an unpaired read pair as
scoreRead1+scoreRead2-INT and scores a paired as
scoreRead1+scoreRead2-insertPenalty. <br>
It compares these two scores to determine whether we should
force pairing. A larger value leads to more aggressive read
pair. [17]</p>

<p style="margin-top: 1em">INPUT/OUTPUT OPTIONS:</p>

<p style="margin-top: 1em">-p Smart pairing. If two
adjacent reads have the same name, they are considered to
form a read pair. This way, paired-end and single-end reads
can be mixed in a <br>
single FASTA/Q stream.</p>

<p style="margin-top: 1em">-R STR Complete read group
header line. &rsquo;&rsquo; can be used in STR and will be
converted to a TAB in the output SAM. The read group ID will
be attached to every read in <br>
the output. An example is &rsquo;@RGID:fooSM:bar&rsquo;.
[null]</p>

<p style="margin-top: 1em">-H ARG If ARG starts with @, it
is interpreted as a string and gets inserted into the output
SAM header; otherwise, ARG is interpreted as a file with all
lines starting <br>
with @ in the file inserted into the SAM header. [null]</p>

<p style="margin-top: 1em">-T INT Don&rsquo;t output
alignment with score lower than INT. This option affects
output and occasionally SAM flag 2. [30]</p>

<p style="margin-top: 1em">-j Treat ALT contigs as part of
the primary assembly (i.e. ignore the db.prefix.alt
file).</p>

<p style="margin-top: 1em">-h INT[,INT2] <br>
If a query has not more than INT hits with score higher than
80% of the best hit, output them all in the XA tag. If INT2
is specified, BWA-MEM outputs up to <br>
INT2 hits if the list contains a hit to an ALT contig.
[5,200]</p>

<p style="margin-top: 1em">-a Output all found alignments
for single-end or unpaired paired-end reads. These
alignments will be flagged as secondary alignments.</p>

<p style="margin-top: 1em">-C Append append FASTA/Q comment
to SAM output. This option can be used to transfer read meta
information (e.g. barcode) to the SAM output. Note that the
FASTA/Q <br>
comment (the string after a space in the header line) must
conform the SAM spec (e.g. BC:Z:CGTAC). Malformated comments
lead to incorrect SAM output.</p>

<p style="margin-top: 1em">-Y Use soft clipping CIGAR
operation for supplementary alignments. By default, BWA-MEM
uses soft clipping for the primary alignment and hard
clipping for supplemen&acirc; <br>
tary alignments.</p>

<p style="margin-top: 1em">-M Mark shorter split hits as
secondary (for Picard compatibility).</p>

<p style="margin-top: 1em">-v INT Control the verbose level
of the output. This option has not been fully supported
throughout BWA. Ideally, a value 0 for disabling all the
output to stderr; 1 <br>
for outputting errors only; 2 for warnings and errors; 3 for
all normal messages; 4 or higher for debugging. When this
option takes value 4, the output is not <br>
SAM. [3]</p>

<p style="margin-top: 1em">-I FLOAT[,FLOAT[,INT[,INT]]]
<br>
Specify the mean, standard deviation (10% of the mean if
absent), max (4 sigma from the mean if absent) and min (4
sigma if absent) of the insert size distribu&acirc; <br>
tion. Only applicable to the FR orientation. By default,
BWA-MEM infers these numbers and the pair orientations given
enough reads. [inferred]</p>

<p style="margin-top: 1em">aln bwa aln [-n maxDiff] [-o
maxGapO] [-e maxGapE] [-d nDelTail] [-i nIndelEnd] [-k
maxSeedDiff] [-l seedLen] [-t nThrds] [-cRN] [-M misMsc] [-O
gapOsc] [-E gapEsc] [-q <br>
trimQual] &lt;in.db.fasta&gt; &lt;in.query.fq&gt; &gt;
&lt;out.sai&gt;</p>

<p style="margin-top: 1em">Find the SA coordinates of the
input reads. Maximum maxSeedDiff differences are allowed in
the first seedLen subsequence and maximum maxDiff
differences are allowed in the <br>
whole sequence.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-n NUM Maximum edit distance if
the value is INT, or the fraction of missing alignments
given 2% uniform base error rate if FLOAT. In the latter
case, the maximum edit <br>
distance is automatically chosen for different read lengths.
[0.04]</p>

<p style="margin-top: 1em">-o INT Maximum number of gap
opens [1]</p>

<p style="margin-top: 1em">-e INT Maximum number of gap
extensions, -1 for k-difference mode (disallowing long gaps)
[-1]</p>

<p style="margin-top: 1em">-d INT Disallow a long deletion
within INT bp towards the 3&rsquo;-end [16]</p>

<p style="margin-top: 1em">-i INT Disallow an indel within
INT bp towards the ends [5]</p>

<p style="margin-top: 1em">-l INT Take the first INT
subsequence as seed. If INT is larger than the query
sequence, seeding will be disabled. For long reads, this
option is typically ranged from <br>
25 to 35 for &lsquo;-k 2&rsquo;. [inf]</p>

<p style="margin-top: 1em">-k INT Maximum edit distance in
the seed [2]</p>

<p style="margin-top: 1em">-t INT Number of threads
(multi-threading mode) [1]</p>

<p style="margin-top: 1em">-M INT Mismatch penalty. BWA
will not search for suboptimal hits with a score lower than
(bestScore-misMsc). [3]</p>

<p style="margin-top: 1em">-O INT Gap open penalty [11]</p>

<p style="margin-top: 1em">-E INT Gap extension penalty
[4]</p>

<p style="margin-top: 1em">-R INT Proceed with suboptimal
alignments if there are no more than INT equally best hits.
This option only affects paired-end mapping. Increasing this
threshold helps <br>
to improve the pairing accuracy at the cost of speed,
especially for short reads (~32bp).</p>

<p style="margin-top: 1em">-c Reverse query but not
complement it, which is required for alignment in the color
space. (Disabled since 0.6.x)</p>

<p style="margin-top: 1em">-N Disable iterative search. All
hits with no more than maxDiff differences will be found.
This mode is much slower than the default.</p>

<p style="margin-top: 1em">-q INT Parameter for read
trimming. BWA trims a read down to
argmax_x{_{i=x+1}^l(INT-q_i)} if q_l&lt;INT where l is the
original read length. [0]</p>

<p style="margin-top: 1em">-I The input is in the Illumina
1.3+ read format (quality equals ASCII-64).</p>

<p style="margin-top: 1em">-B INT Length of barcode
starting from the 5&rsquo;-end. When INT is positive, the
barcode of each read will be trimmed before mapping and will
be written at the BC SAM tag. <br>
For paired-end reads, the barcode from both ends are
concatenated. [0]</p>

<p style="margin-top: 1em">-b Specify the input read
sequence file is the BAM format. For paired-end data, two
ends in a pair must be grouped together and options -1 or -2
are usually applied <br>
to specify which end should be mapped. Typical command lines
for mapping pair-end data in the BAM format are:</p>

<p style="margin-top: 1em">bwa aln ref.fa -b1 reads.bam
&gt; 1.sai <br>
bwa aln ref.fa -b2 reads.bam &gt; 2.sai <br>
bwa sampe ref.fa 1.sai 2.sai reads.bam reads.bam &gt;
aln.sam</p>

<p style="margin-top: 1em">-0 When -b is specified, only
use single-end reads in mapping.</p>

<p style="margin-top: 1em">-1 When -b is specified, only
use the first read in a read pair in mapping (skip
single-end reads and the second reads).</p>

<p style="margin-top: 1em">-2 When -b is specified, only
use the second read in a read pair in mapping.</p>

<p style="margin-top: 1em">samse bwa samse [-n maxOcc]
&lt;in.db.fasta&gt; &lt;in.sai&gt; &lt;in.fq&gt; &gt;
&lt;out.sam&gt;</p>

<p style="margin-top: 1em">Generate alignments in the SAM
format given single-end reads. Repetitive hits will be
randomly chosen.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-n INT Maximum number of
alignments to output in the XA tag for reads paired
properly. If a read has more than INT hits, the XA tag will
not be written. [3]</p>

<p style="margin-top: 1em">-r STR Specify the read group in
a format like &lsquo;@RGID:fooSM:bar&rsquo;. [null]</p>

<p style="margin-top: 1em">sampe bwa sampe [-a maxInsSize]
[-o maxOcc] [-n maxHitPaired] [-N maxHitDis] [-P]
&lt;in.db.fasta&gt; &lt;in1.sai&gt; &lt;in2.sai&gt;
&lt;in1.fq&gt; &lt;in2.fq&gt; &gt; &lt;out.sam&gt;</p>

<p style="margin-top: 1em">Generate alignments in the SAM
format given paired-end reads. Repetitive read pairs will be
placed randomly.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-a INT Maximum insert size for a
read pair to be considered being mapped properly. Since
0.4.5, this option is only used when there are not enough
good alignment to infer <br>
the distribution of insert sizes. [500]</p>

<p style="margin-top: 1em">-o INT Maximum occurrences of a
read for pairing. A read with more occurrneces will be
treated as a single-end read. Reducing this parameter helps
faster pairing. <br>
[100000]</p>

<p style="margin-top: 1em">-P Load the entire FM-index into
memory to reduce disk operations (base-space reads only).
With this option, at least 1.25N bytes of memory are
required, where N is <br>
the length of the genome.</p>

<p style="margin-top: 1em">-n INT Maximum number of
alignments to output in the XA tag for reads paired
properly. If a read has more than INT hits, the XA tag will
not be written. [3]</p>

<p style="margin-top: 1em">-N INT Maximum number of
alignments to output in the XA tag for disconcordant read
pairs (excluding singletons). If a read has more than INT
hits, the XA tag will not be <br>
written. [10]</p>

<p style="margin-top: 1em">-r STR Specify the read group in
a format like &lsquo;@RGID:fooSM:bar&rsquo;. [null]</p>

<p style="margin-top: 1em">bwasw bwa bwasw [-a matchScore]
[-b mmPen] [-q gapOpenPen] [-r gapExtPen] [-t nThreads] [-w
bandWidth] [-T thres] [-s hspIntv] [-z zBest] [-N nHspRev]
[-c thresCoef] <br>
&lt;in.db.fasta&gt; &lt;in.fq&gt; [mate.fq]</p>

<p style="margin-top: 1em">Align query sequences in the
in.fq file. When mate.fq is present, perform paired-end
alignment. The paired-end mode only works for reads Illumina
short-insert libraries. <br>
In the paired-end mode, BWA-SW may still output split
alignments but they are all marked as not properly paired;
the mate positions will not be written if the mate has <br>
multiple local hits.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-a INT Score of a match [1]</p>

<p style="margin-top: 1em">-b INT Mismatch penalty [3]</p>

<p style="margin-top: 1em">-q INT Gap open penalty [5]</p>

<p style="margin-top: 1em">-r INT Gap extension penalty.
The penalty for a contiguous gap of size k is q+k*r. [2]</p>

<p style="margin-top: 1em">-t INT Number of threads in the
multi-threading mode [1]</p>

<p style="margin-top: 1em">-w INT Band width in the banded
alignment [33]</p>

<p style="margin-top: 1em">-T INT Minimum score threshold
divided by a [37]</p>

<p style="margin-top: 1em">-c FLOAT Coefficient for
threshold adjustment according to query length. Given an
l-long query, the threshold for a hit to be retained is
a*max{T,c*log(l)}. [5.5]</p>

<p style="margin-top: 1em">-z INT Z-best heuristics. Higher
-z increases accuracy at the cost of speed. [1]</p>

<p style="margin-top: 1em">-s INT Maximum SA interval size
for initiating a seed. Higher -s increases accuracy at the
cost of speed. [3]</p>

<p style="margin-top: 1em">-N INT Minimum number of seeds
supporting the resultant alignment to skip reverse
alignment. [5]</p>

<p style="margin-top: 1em">SAM ALIGNMENT FORMAT <br>
The output of the &lsquo;aln&rsquo; command is binary and
designed for BWA use only. BWA outputs the final alignment
in the SAM (Sequence Alignment/Map) format. Each line
consists of:</p>


<p style="margin-top: 1em">&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;
<br>
&acirc;Col &acirc; Field &acirc; Description &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc; 1 &acirc; QNAME &acirc; Query (pair) NAME &acirc;
<br>
&acirc; 2 &acirc; FLAG &acirc; bitwise FLAG &acirc; <br>
&acirc; 3 &acirc; RNAME &acirc; Reference sequence NAME
&acirc; <br>
&acirc; 4 &acirc; POS &acirc; 1-based leftmost
POSition/coordinate of clipped sequence &acirc; <br>
&acirc; 5 &acirc; MAPQ &acirc; MAPping Quality
(Phred-scaled) &acirc; <br>
&acirc; 6 &acirc; CIAGR &acirc; extended CIGAR string
&acirc; <br>
&acirc; 7 &acirc; MRNM &acirc; Mate Reference sequence NaMe
(&lsquo;=&rsquo; if same as RNAME) &acirc; <br>
&acirc; 8 &acirc; MPOS &acirc; 1-based Mate POSistion
&acirc; <br>
&acirc; 9 &acirc; ISIZE &acirc; Inferred insert SIZE &acirc;
<br>
&acirc;10 &acirc; SEQ &acirc; query SEQuence on the same
strand as the reference &acirc; <br>
&acirc;11 &acirc; QUAL &acirc; query QUALity (ASCII-33 gives
the Phred base quality) &acirc; <br>
&acirc;12 &acirc; OPT &acirc; variable OPTional fields in
the format TAG:VTYPE:VALUE &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;</p>

<p style="margin-top: 1em">Each bit in the FLAG field is
defined as:</p>


<p style="margin-top: 1em">&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;
<br>
&acirc;Chr &acirc; Flag &acirc; Description &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc; p &acirc; 0x0001 &acirc; the read is paired in
sequencing &acirc; <br>
&acirc; P &acirc; 0x0002 &acirc; the read is mapped in a
proper pair &acirc; <br>
&acirc; u &acirc; 0x0004 &acirc; the query sequence itself
is unmapped &acirc; <br>
&acirc; U &acirc; 0x0008 &acirc; the mate is unmapped
&acirc; <br>
&acirc; r &acirc; 0x0010 &acirc; strand of the query (1 for
reverse) &acirc; <br>
&acirc; R &acirc; 0x0020 &acirc; strand of the mate &acirc;
<br>
&acirc; 1 &acirc; 0x0040 &acirc; the read is the first read
in a pair &acirc; <br>
&acirc; 2 &acirc; 0x0080 &acirc; the read is the second read
in a pair &acirc; <br>
&acirc; s &acirc; 0x0100 &acirc; the alignment is not
primary &acirc; <br>
&acirc; f &acirc; 0x0200 &acirc; QC failure &acirc; <br>
&acirc; d &acirc; 0x0400 &acirc; optical or PCR duplicate
&acirc; <br>
&acirc; S &acirc; 0x0800 &acirc; supplementary alignment
&acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;</p>

<p style="margin-top: 1em">The Please check
&lt;http://samtools.sourceforge.net&gt; for the format
specification and the tools for post-processing the
alignment.</p>

<p style="margin-top: 1em">BWA generates the following
optional fields. Tags starting with &lsquo;X&rsquo; are
specific to BWA.</p>


<p style="margin-top: 1em">&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;
<br>
&acirc;Tag &acirc; Meaning &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc;NM &acirc; Edit distance &acirc; <br>
&acirc;MD &acirc; Mismatching positions/bases &acirc; <br>
&acirc;AS &acirc; Alignment score &acirc; <br>
&acirc;BC &acirc; Barcode sequence &acirc; <br>
&acirc;SA &acirc; Supplementary alignments &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc;X0 &acirc; Number of best hits &acirc; <br>
&acirc;X1 &acirc; Number of suboptimal hits found by BWA
&acirc; <br>
&acirc;XN &acirc; Number of ambiguous bases in the
referenece &acirc; <br>
&acirc;XM &acirc; Number of mismatches in the alignment
&acirc; <br>
&acirc;XO &acirc; Number of gap opens &acirc; <br>
&acirc;XG &acirc; Number of gap extentions &acirc; <br>
&acirc;XT &acirc; Type: Unique/Repeat/N/Mate-sw &acirc; <br>
&acirc;XA &acirc; Alternative hits; format:
/(chr,pos,CIGAR,NM;)*/ &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc;XS &acirc; Suboptimal alignment score &acirc; <br>
&acirc;XF &acirc; Support from forward/reverse alignment
&acirc; <br>
&acirc;XE &acirc; Number of supporting seeds &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;</p>

<p style="margin-top: 1em">Note that XO and XG are
generated by BWT search while the CIGAR string by
Smith-Waterman alignment. These two tags may be inconsistent
with the CIGAR string. This is not a bug.</p>

<p style="margin-top: 1em">NOTES ON SHORT-READ ALIGNMENT
<br>
Alignment Accuracy <br>
When seeding is disabled, BWA guarantees to find an
alignment containing maximum maxDiff differences including
maxGapO gap opens which do not occur within nIndelEnd bp
towards <br>
either end of the query. Longer gaps may be found if maxGapE
is positive, but it is not guaranteed to find all hits. When
seeding is enabled, BWA further requires that the first <br>
seedLen subsequence contains no more than maxSeedDiff
differences.</p>

<p style="margin-top: 1em">When gapped alignment is
disabled, BWA is expected to generate the same alignment as
Eland version 1, the Illumina alignment program. However, as
BWA change &lsquo;N&rsquo; in the database <br>
sequence to random nucleotides, hits to these random
sequences will also be counted. As a consequence, BWA may
mark a unique hit as a repeat, if the random sequences
happen to be <br>
identical to the sequences which should be unqiue in the
database.</p>

<p style="margin-top: 1em">By default, if the best hit is
not highly repetitive (controlled by -R), BWA also finds all
hits contains one more mismatch; otherwise, BWA finds all
equally best hits only. Base <br>
quality is NOT considered in evaluating hits. In the
paired-end mode, BWA pairs all hits it found. It further
performs Smith-Waterman alignment for unmapped reads to
rescue reads <br>
with a high erro rate, and for high-quality anomalous pairs
to fix potential alignment errors.</p>

<p style="margin-top: 1em">Estimating Insert Size
Distribution <br>
BWA estimates the insert size distribution per 256*1024 read
pairs. It first collects pairs of reads with both ends
mapped with a single-end quality 20 or higher and then
calcu&acirc; <br>
lates median (Q2), lower and higher quartile (Q1 and Q3). It
estimates the mean and the variance of the insert size
distribution from pairs whose insert sizes are within
interval <br>
[Q1-2(Q3-Q1), Q3+2(Q3-Q1)]. The maximum distance x for a
pair considered to be properly paired (SAM flag 0x2) is
calculated by solving equation Phi((x-mu)/sigma)=x/L*p0,
where mu <br>
is the mean, sigma is the standard error of the insert size
distribution, L is the length of the genome, p0 is prior of
anomalous pair and Phi() is the standard cumulative
dis&acirc; <br>
tribution function. For mapping Illumina short-insert reads
to the human genome, x is about 6-7 sigma away from the
mean. Quartiles, mean, variance and x will be printed to the
<br>
standard error output.</p>

<p style="margin-top: 1em">Memory Requirement <br>
With bwtsw algorithm, 5GB memory is required for indexing
the complete human genome sequences. For short reads, the
aln command uses ~3.2GB memory and the sampe command uses
<br>
~5.4GB.</p>

<p style="margin-top: 1em">Speed <br>
Indexing the human genome sequences takes 3 hours with bwtsw
algorithm. Indexing smaller genomes with IS algorithms is
faster, but requires more memory.</p>

<p style="margin-top: 1em">The speed of alignment is
largely determined by the error rate of the query sequences
(r). Firstly, BWA runs much faster for near perfect hits
than for hits with many differ&acirc; <br>
ences, and it stops searching for a hit with l+2 differences
if a l-difference hit is found. This means BWA will be very
slow if r is high because in this case BWA has to visit <br>
hits with many differences and looking for these hits is
expensive. Secondly, the alignment algorithm behind makes
the speed sensitive to [k log(N)/m], where k is the maximum
<br>
allowed differences, N the size of database and m the length
of a query. In practice, we choose k w.r.t. r and therefore
r is the leading factor. I would not recommend to use BWA
<br>
on data with r&gt;0.02.</p>

<p style="margin-top: 1em">Pairing is slower for shorter
reads. This is mainly because shorter reads have more
spurious hits and converting SA coordinates to chromosomal
coordinates are very costly.</p>

<p style="margin-top: 1em">CHANGES IN BWA-0.6 <br>
Since version 0.6, BWA has been able to work with a
reference genome longer than 4GB. This feature makes it
possible to integrate the forward and reverse complemented
genome in <br>
one FM-index, which speeds up both BWA-short and BWA-SW. As
a tradeoff, BWA uses more memory because it has to keep all
positions and ranks in 64-bit integers, twice larger than
<br>
32-bit integers used in the previous versions.</p>

<p style="margin-top: 1em">The latest BWA-SW also works for
paired-end reads longer than 100bp. In comparison to
BWA-short, BWA-SW tends to be more accurate for highly
unique reads and more robust to rela&acirc; <br>
tive long INDELs and structural variants. Nonetheless,
BWA-short usually has higher power to distinguish the
optimal hit from many suboptimal hits. The choice of the
mapping <br>
algorithm may depend on the application.</p>

<p style="margin-top: 1em">SEE ALSO <br>
BWA website &lt;http://bio-bwa.sourceforge.net&gt;, Samtools
website &lt;http://samtools.sourceforge.net&gt;</p>

<p style="margin-top: 1em">AUTHOR <br>
Heng Li at the Sanger Institute wrote the key source codes
and integrated the following codes for BWT construction:
bwtsw &lt;http://i.cs.hku.hk/~ckwong3/bwtsw/&gt;,
implemented by <br>
Chi-Kwong Wong at the University of Hong Kong and IS
&lt;http://yuta.256.googlepages.com/sais&gt; originally
proposed by Nong Ge &lt;http://www.cs.sysu.edu.cn/nong/&gt;
at the Sun Yat-Sen <br>
University and implemented by Yuta Mori.</p>

<p style="margin-top: 1em">LICENSE AND CITATION <br>
The full BWA package is distributed under GPLv3 as it uses
source codes from BWT-SW which is covered by GPL. Sorting,
hash table, BWT and IS libraries are distributed under the
<br>
MIT license.</p>

<p style="margin-top: 1em">If you use the BWA-backtrack
algorithm, please cite the following paper:</p>

<p style="margin-top: 1em">Li H. and Durbin R. (2009) Fast
and accurate short read alignment with Burrows-Wheeler
transform. Bioinformatics, 25, 1754-1760. [PMID:
19451168]</p>

<p style="margin-top: 1em">If you use the BWA-SW algorithm,
please cite:</p>

<p style="margin-top: 1em">Li H. and Durbin R. (2010) Fast
and accurate long-read alignment with Burrows-Wheeler
transform. Bioinformatics, 26, 589-595. [PMID: 20080505]</p>

<p style="margin-top: 1em">If you use BWA-MEM or the
fastmap component of BWA, please cite:</p>

<p style="margin-top: 1em">Li H. (2013) Aligning sequence
reads, clone sequences and assembly contigs with BWA-MEM.
arXiv:1303.3997v1 [q-bio.GN].</p>

<p style="margin-top: 1em">It is likely that the BWA-MEM
manuscript will not appear in a peer-reviewed journal.</p>

<p style="margin-top: 1em">HISTORY <br>
BWA is largely influenced by BWT-SW. It uses source codes
from BWT-SW and mimics its binary file formats; BWA-SW
resembles BWT-SW in several ways. The initial idea about
BWT- <br>
based alignment also came from the group who developed
BWT-SW. At the same time, BWA is different enough from
BWT-SW. The short-read alignment algorithm bears no
similarity to <br>
Smith-Waterman algorithm any more. While BWA-SW learns from
BWT-SW, it introduces heuristics that can hardly be applied
to the original algorithm. In all, BWA does not guarantee
<br>
to find all local hits as what BWT-SW is designed to do, but
it is much faster than BWT-SW on both short and long query
sequences.</p>

<p style="margin-top: 1em">I started to write the first
piece of codes on 24 May 2008 and got the initial stable
version on 02 June 2008. During this period, I was
acquainted that Professor Tak-Wah Lam, <br>
the first author of BWT-SW paper, was collaborating with
Beijing Genomics Institute on SOAP2, the successor to SOAP
(Short Oligonucleotide Analysis Package). SOAP2 has come out
<br>
in November 2008. According to the SourceForge download
page, the third BWT-based short read aligner, bowtie, was
first released in August 2008. At the time of writing this
man&acirc; <br>
ual, at least three more BWT-based short-read aligners are
being implemented.</p>

<p style="margin-top: 1em">The BWA-SW algorithm is a new
component of BWA. It was conceived in November 2008 and
implemented ten months later.</p>

<p style="margin-top: 1em">The BWA-MEM algorithm is based
on an algorithm finding super-maximal exact matches (SMEMs),
which was first published with the fermi assembler paper in
2012. I first implemented <br>
the basic SMEM algorithm in the fastmap command for an
experiment and then extended the basic algorithm and added
the extension part in Feburary 2013 to make BWA-MEM a fully
fea&acirc; <br>
tured mapper.</p>

<p style="margin-top: 1em">bwa-0.7.15-r1140 31 May 2016
bwa(1)</p>
<hr>
</body>
</html>
