<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:57:49 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CD-HIT-2D(1) User Commands CD-HIT-2D(1)</p>

<p style="margin-top: 1em">NAME <br>
cd-hit-2d - quickly group sequences in db1 or db2 format</p>

<p style="margin-top: 1em">SYNOPSIS <br>
cd-hit-2d [Options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
====== CD-HIT version 4.6 (built on Aug 14 2016) ======</p>

<p style="margin-top: 1em">Options</p>

<p style="margin-top: 1em">-i input filename for db1 in
fasta format, required</p>

<p style="margin-top: 1em">-i2 input filename for db2 in
fasta format, required</p>

<p style="margin-top: 1em">-o output filename, required</p>

<p style="margin-top: 1em">-c sequence identity threshold,
default 0.9 this is the default cd-hit&rsquo;s &quot;global
sequence identity&quot; calculated as: number of identical
amino acids in alignment divided by the <br>
full length of the shorter sequence</p>

<p style="margin-top: 1em">-G use global sequence identity,
default 1 if set to 0, then use local sequence identity,
calculated as : number of identical amino acids in alignment
divided by the length <br>
of the alignment NOTE!!! don&rsquo;t use -G 0 unless you use
alignment coverage controls see options -aL, -AL, -aS,
-AS</p>

<p style="margin-top: 1em">-b band_width of alignment,
default 20</p>

<p style="margin-top: 1em">-M memory limit (in MB) for the
program, default 800; 0 for unlimitted;</p>

<p style="margin-top: 1em">-T number of threads, default 1;
with 0, all CPUs will be used</p>

<p style="margin-top: 1em">-n word_length, default 5, see
user&rsquo;s guide for choosing it</p>

<p style="margin-top: 1em">-l length of
throw_away_sequences, default 10</p>

<p style="margin-top: 1em">-t tolerance for redundance,
default 2</p>

<p style="margin-top: 1em">-d length of description in
.clstr file, default 20 if set to 0, it takes the fasta
defline and stops at first space</p>

<p style="margin-top: 1em">-s length difference cutoff,
default 0.0 if set to 0.9, the shorter sequences need to be
at least 90% length of the representative of the cluster</p>

<p style="margin-top: 1em">-S length difference cutoff in
amino acid, default 999999 if set to 60, the length
difference between the shorter sequences and the
representative of the cluster can not be <br>
bigger than 60</p>

<p style="margin-top: 1em">-s2 length difference cutoff for
db1, default 1.0 by default, seqs in db1 &gt;= seqs in db2
in a same cluster if set to 0.9, seqs in db1 may just &gt;=
90% seqs in db2</p>

<p style="margin-top: 1em">-S2 length difference cutoff,
default 0 by default, seqs in db1 &gt;= seqs in db2 in a
same cluster if set to 60, seqs in db2 may 60aa longer than
seqs in db1</p>

<p style="margin-top: 1em">-aL alignment coverage for the
longer sequence, default 0.0 if set to 0.9, the alignment
must covers 90% of the sequence</p>

<p style="margin-top: 1em">-AL alignment coverage control
for the longer sequence, default 99999999 if set to 60, and
the length of the sequence is 400, then the alignment must
be &gt;= 340 (400-60) <br>
residues</p>

<p style="margin-top: 1em">-aS alignment coverage for the
shorter sequence, default 0.0 if set to 0.9, the alignment
must covers 90% of the sequence</p>

<p style="margin-top: 1em">-AS alignment coverage control
for the shorter sequence, default 99999999 if set to 60, and
the length of the sequence is 400, then the alignment must
be &gt;= 340 (400-60) <br>
residues</p>

<p style="margin-top: 1em">-A minimal alignment coverage
control for the both sequences, default 0 alignment must
cover &gt;= this value for both sequences</p>

<p style="margin-top: 1em">-uL maximum unmatched percentage
for the longer sequence, default 1.0 if set to 0.1, the
unmatched region (excluding leading and tailing gaps) must
not be more than 10% of the <br>
sequence</p>

<p style="margin-top: 1em">-uS maximum unmatched percentage
for the shorter sequence, default 1.0 if set to 0.1, the
unmatched region (excluding leading and tailing gaps) must
not be more than 10% of <br>
the sequence</p>

<p style="margin-top: 1em">-U maximum unmatched length,
default 99999999 if set to 10, the unmatched region
(excluding leading and tailing gaps) must not be more than
10 bases</p>

<p style="margin-top: 1em">-B 1 or 0, default 0, by
default, sequences are stored in RAM if set to 1, sequence
are stored on hard drive it is recommended to use -B 1 for
huge databases</p>

<p style="margin-top: 1em">-p 1 or 0, default 0 if set to
1, print alignment overlap in .clstr file</p>

<p style="margin-top: 1em">-g 1 or 0, default 0 by
cd-hit&rsquo;s default algorithm, a sequence is clustered to
the first cluster that meet the threshold (fast cluster). If
set to 1, the program will cluster <br>
it into the most similar cluster that meet the threshold
(accurate but slow mode) but either 1 or 0 won&rsquo;t
change the representatives of final clusters</p>

<p style="margin-top: 1em">-bak write backup cluster file
(1 or 0, default 0)</p>

<p style="margin-top: 1em">-h print this help</p>

<p style="margin-top: 1em">Questions, bugs, contact
Weizhong Li at liwz@sdsc.edu</p>

<p style="margin-top: 1em">If you find cd-hit useful,
please kindly cite:</p>

<p style="margin-top: 1em">&quot;Clustering of highly
homologous sequences to reduce thesize of large protein
database&quot;, Weizhong Li, Lukasz Jaroszewski &amp; Adam
Godzik. Bioinformatics, (2001) 17:282-283 <br>
&quot;Cd-hit: a fast program for clustering and comparing
large sets of protein or nucleotide sequences&quot;,
Weizhong Li &amp; Adam Godzik. Bioinformatics, (2006)
22:1658-1659</p>

<p style="margin-top: 1em">cd-hit-2d 4.6.6 August 2016
CD-HIT-2D(1)</p>
<hr>
</body>
</html>
