<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CD-HIT-2D(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CD-HIT-2D(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">CD-HIT-2D(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cd-hit-2d - quickly group sequences in db1 or db2 format
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>cd-hit-2d</b> [ <i>Options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">====== CD-HIT version 4.6 (built on Aug 14 2016)
    ======</dd>
</dl>
<div class="Pp"></div>
Options
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b></dt>
  <dd class="It-tag">input filename for db1 in fasta format, required</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i2</b></dt>
  <dd class="It-tag">input filename for db2 in fasta format, required</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b></dt>
  <dd class="It-tag">output filename, required</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">sequence identity threshold, default 0.9 this is the
      default cd-hit's &quot;global sequence identity&quot; calculated as:
      number of identical amino acids in alignment divided by the full length of
      the shorter sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-G</b></dt>
  <dd class="It-tag">use global sequence identity, default 1 if set to 0, then
      use local sequence identity, calculated as : number of identical amino
      acids in alignment divided by the length of the alignment NOTE!!! don't
      use <b>-G</b> 0 unless you use alignment coverage controls see options
      <b>-aL</b>, <b>-AL</b>, <b>-aS</b>, <b>-AS</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b></dt>
  <dd class="It-tag">band_width of alignment, default 20</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b></dt>
  <dd class="It-tag">memory limit (in MB) for the program, default 800; 0 for
      unlimitted;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b></dt>
  <dd class="It-tag">number of threads, default 1; with 0, all CPUs will be
    used</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">word_length, default 5, see user's guide for choosing
    it</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b></dt>
  <dd class="It-tag">length of throw_away_sequences, default 10</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b></dt>
  <dd class="It-tag">tolerance for redundance, default 2</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b></dt>
  <dd class="It-tag">length of description in .clstr file, default 20 if set to
      0, it takes the fasta defline and stops at first space</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">length difference cutoff, default 0.0 if set to 0.9, the
      shorter sequences need to be at least 90% length of the representative of
      the cluster</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">length difference cutoff in amino acid, default 999999 if
      set to 60, the length difference between the shorter sequences and the
      representative of the cluster can not be bigger than 60</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s2</b></dt>
  <dd class="It-tag">length difference cutoff for db1, default 1.0 by default,
      seqs in db1 &gt;= seqs in db2 in a same cluster if set to 0.9, seqs in db1
      may just &gt;= 90% seqs in db2</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S2</b></dt>
  <dd class="It-tag">length difference cutoff, default 0 by default, seqs in db1
      &gt;= seqs in db2 in a same cluster if set to 60, seqs in db2 may 60aa
      longer than seqs in db1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-aL</b></dt>
  <dd class="It-tag">alignment coverage for the longer sequence, default 0.0 if
      set to 0.9, the alignment must covers 90% of the sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-AL</b></dt>
  <dd class="It-tag">alignment coverage control for the longer sequence, default
      99999999 if set to 60, and the length of the sequence is 400, then the
      alignment must be &gt;= 340 (400-60) residues</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-aS</b></dt>
  <dd class="It-tag">alignment coverage for the shorter sequence, default 0.0 if
      set to 0.9, the alignment must covers 90% of the sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-AS</b></dt>
  <dd class="It-tag">alignment coverage control for the shorter sequence,
      default 99999999 if set to 60, and the length of the sequence is 400, then
      the alignment must be &gt;= 340 (400-60) residues</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b></dt>
  <dd class="It-tag">minimal alignment coverage control for the both sequences,
      default 0 alignment must cover &gt;= this value for both sequences</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-uL</b></dt>
  <dd class="It-tag">maximum unmatched percentage for the longer sequence,
      default 1.0 if set to 0.1, the unmatched region (excluding leading and
      tailing gaps) must not be more than 10% of the sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-uS</b></dt>
  <dd class="It-tag">maximum unmatched percentage for the shorter sequence,
      default 1.0 if set to 0.1, the unmatched region (excluding leading and
      tailing gaps) must not be more than 10% of the sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b></dt>
  <dd class="It-tag">maximum unmatched length, default 99999999 if set to 10,
      the unmatched region (excluding leading and tailing gaps) must not be more
      than 10 bases</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b></dt>
  <dd class="It-tag">1 or 0, default 0, by default, sequences are stored in RAM
      if set to 1, sequence are stored on hard drive it is recommended to use
      <b>-B</b> 1 for huge databases</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b></dt>
  <dd class="It-tag">1 or 0, default 0 if set to 1, print alignment overlap in
      .clstr file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">1 or 0, default 0 by cd-hit's default algorithm, a sequence
      is clustered to the first cluster that meet the threshold (fast cluster).
      If set to 1, the program will cluster it into the most similar cluster
      that meet the threshold (accurate but slow mode) but either 1 or 0 won't
      change the representatives of final clusters</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-bak</b>
  write backup cluster file (1 or 0, default 0)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">print this help</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Questions, bugs, contact Weizhong Li at liwz@sdsc.edu</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">If you find cd-hit useful, please kindly cite:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&quot;Clustering of highly homologous sequences to reduce
      thesize of large protein database&quot;, Weizhong Li, Lukasz Jaroszewski
      &amp; Adam Godzik. Bioinformatics, (2001) 17:282-283 &quot;Cd-hit: a fast
      program for clustering and comparing large sets of protein or nucleotide
      sequences&quot;, Weizhong Li &amp; Adam Godzik. Bioinformatics, (2006)
      22:1658-1659</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">August 2016</td>
    <td class="foot-os">cd-hit-2d 4.6.6</td>
  </tr>
</table>
</body>
</html>
