<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CD-HIT-454(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CD-HIT-454(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">CD-HIT-454(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cd-hit-454 - quickly group sequences, optimised for 454 data
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>cd-hit-454</b> [ <i>Options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">====== CD-HIT version 4.6 (built on Aug 14 2016)
    ======</dd>
</dl>
<div class="Pp"></div>
Options
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b></dt>
  <dd class="It-tag">input filename in fasta format, required</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b></dt>
  <dd class="It-tag">output filename, required</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">sequence identity threshold, default 0.98 this is a
      &quot;global sequence identity&quot; calculated as : number of identical
      amino acids in alignment divided by the full length of the shorter
      sequence + gaps</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b></dt>
  <dd class="It-tag">band_width of alignment, default 10</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b></dt>
  <dd class="It-tag">memory limit (in MB) for the program, default 800; 0 for
      unlimitted;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b></dt>
  <dd class="It-tag">number of threads, default 1; with 0, all CPUs will be
    used</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">word_length, default 10, see user's guide for choosing
    it</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-aL</b></dt>
  <dd class="It-tag">alignment coverage for the longer sequence, default 0.0 if
      set to 0.9, the alignment must covers 90% of the sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-AL</b></dt>
  <dd class="It-tag">alignment coverage control for the longer sequence, default
      99999999 if set to 60, and the length of the sequence is 400, then the
      alignment must be &gt;= 340 (400-60) residues</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-aS</b></dt>
  <dd class="It-tag">alignment coverage for the shorter sequence, default 0.0 if
      set to 0.9, the alignment must covers 90% of the sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-AS</b></dt>
  <dd class="It-tag">alignment coverage control for the shorter sequence,
      default 99999999 if set to 60, and the length of the sequence is 400, then
      the alignment must be &gt;= 340 (400-60) residues</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b></dt>
  <dd class="It-tag">1 or 0, default 0, by default, sequences are stored in RAM
      if set to 1, sequence are stored on hard drive it is recommended to use
      <b>-B</b> 1 for huge databases</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">1 or 0, default 0 by cd-hit's default algorithm, a sequence
      is clustered to the first cluster that meet the threshold (fast cluster).
      If set to 1, the program will cluster it into the most similar cluster
      that meet the threshold (accurate but slow mode) but either 1 or 0 won't
      change the representatives of final clusters</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b></dt>
  <dd class="It-tag">max size per indel, default 1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-match</b></dt>
  <dd class="It-tag">matching score, default 2</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mismatch</b></dt>
  <dd class="It-tag">mismatching score, default <b>-1</b></dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-gap</b>
  gap opening score, default <b>-3</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gap-ext</b></dt>
  <dd class="It-tag">gap extension score, default <b>-1</b></dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-bak</b>
  write backup cluster file (1 or 0, default 0)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">print this help</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Questions, bugs, contact Weizhong Li at liwz@sdsc.edu</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">If you find cd-hit useful, please kindly cite:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&quot;Clustering of highly homologous sequences to reduce
      thesize of large protein database&quot;, Weizhong Li, Lukasz Jaroszewski
      &amp; Adam Godzik. Bioinformatics, (2001) 17:282-283 &quot;Cd-hit: a fast
      program for clustering and comparing large sets of protein or nucleotide
      sequences&quot;, Weizhong Li &amp; Adam Godzik. Bioinformatics, (2006)
      22:1658-1659 &quot;Beifang Niu, Limin Fu, Shulei Sun and Weizhong Li.
      Artificial and natural duplicates in pyrosequencing reads of metagenomic
      data. BMC Bioinformatics (2010) 11:187</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">August 2016</td>
    <td class="foot-os">cd-hit-454 4.6.6</td>
  </tr>
</table>
</body>
</html>
