<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:58:21 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CHIMERASLAYER(1) User Commands CHIMERASLAYER(1)</p>

<p style="margin-top: 1em">NAME <br>
chimeraslayer - detects likely chimeras in PCR amplified
DNA</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ChimeraSlayer is a chimeric sequence detection utility,
compatible with near-full length Sanger sequences and
shorter 454-FLX sequences (~500bp).</p>

<p style="margin-top: 1em">Chimera Slayer involves the
following series of steps that operate to flag chimeric 16S
rRNA sequences:</p>

<p style="margin-top: 1em">1. the ends of a query sequence
are searched against an included database of reference
chimera-free 16S sequences to identify potential parents of
a chimera</p>

<p style="margin-top: 1em">2. candidate parents of a
chimera are selected as those that form a branched best
scoring alignment to the NAST-formatted query sequence</p>

<p style="margin-top: 1em">3. the NAST alignment of the
query sequence is improved in a &acirc;chimera-aware&acirc;
profile-based NAST realignment to the selected reference
parent sequences</p>

<p style="margin-top: 1em">4. an evolutionary framework is
used to flag query sequences found to exhibit greater
sequence homology to an in silico chimera formed between any
two of the selected refer&acirc; <br>
ence parent sequences.</p>

<p style="margin-top: 1em">To run Chimera Slayer, you need
NAST-formatted sequences generated by the nast-ier
utility.</p>

<p style="margin-top: 1em">ChimeraSlayer is part of the
microbiomeutil suite.</p>

<p style="margin-top: 1em">OPTIONS <br>
Required <br>
--query_NAST <br>
multi-fasta file containing query sequences in alignment
format</p>

<p style="margin-top: 1em">Common options <br>
--db_NAST <br>
db in NAST format (default:
/usr/share/microbiomeutil-data/RESOURCES/rRNA16S.gold.NAST_ALIGNED.fasta)</p>

<p style="margin-top: 1em">--db_FASTA <br>
db in fasta format (megablast formatted) (default:
/usr/share/microbiomeutil-data/RESOURCES/rRNA16S.gold.fasta)</p>

<p style="margin-top: 1em">-n number of top matching
database sequences to compare to (default 15)</p>

<p style="margin-top: 1em">-R min divergence ratio default:
1.007</p>

<p style="margin-top: 1em">-P min percent identity among
matching sequences (default: 90)</p>

<p style="margin-top: 1em">Parameters to tune
ChimeraParentSelector: <br>
Scoring parameters:</p>

<p style="margin-top: 1em">-M match score (default: +5)</p>

<p style="margin-top: 1em">-N mismatch penalty (default:
-4)</p>

<p style="margin-top: 1em">-Q min query coverage by
matching database sequence (default: 70)</p>

<p style="margin-top: 1em">-T maximum traverses of the
multiple alignment (default: 1)</p>

<p style="margin-top: 1em">Parameters to tune
ChimeraPhyloChecker <br>
--windowSize <br>
default 50</p>

<p style="margin-top: 1em">--windowStep <br>
default 5</p>

<p style="margin-top: 1em">--minBS <br>
minimum bootstrap support for calling chimera (default:
90)</p>

<p style="margin-top: 1em">--num_BS_replicates <br>
default: 100</p>

<p style="margin-top: 1em">--low_range_finer_BS <br>
(default: 10) If computed BS is between minBS and (minBS -
low_range_finer_BS), then num_finer_BS_replicates
computed.</p>

<p style="margin-top: 1em">--num_finer_BS_replicates <br>
(default: 1000)</p>

<p style="margin-top: 1em">-S percent of SNPs to sample on
each side of breakpoint for computing bootstrap support
(default: 10)</p>

<p style="margin-top: 1em">--num_parents_test <br>
number of potential parents to test for chimeras (default:
3)</p>

<p style="margin-top: 1em">--MAX_CHIMERA_PARENT_PER_ID <br>
Chimera/Parent alignments with perID above this are
considered non-chimeras (default 100; turned off)</p>

<p style="margin-top: 1em">Misc options <br>
--printFinalAlignments <br>
shows alignment between query sequence and pair of candidate
chimera parents</p>

<p style="margin-top: 1em">--printCSalignments <br>
print ChimeraSlayer alignments in ChimeraSlayer output</p>

<p style="margin-top: 1em">--exec_dir <br>
chdir to here before running</p>

<p style="margin-top: 1em">SEE ALSO <br>
http://microbiomeutil.sourceforge.net/#A_CS</p>

<p style="margin-top: 1em">AUTHOR <br>
This manual page was written by Andreas Tille
&lt;tille@debian.org&gt; but can be freely used for any
other distribution.</p>

<p style="margin-top: 1em">chimeraslayer June 2013
CHIMERASLAYER(1)</p>
<hr>
</body>
</html>
