<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CHIMERASLAYER(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CHIMERASLAYER(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">CHIMERASLAYER(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
chimeraslayer - detects likely chimeras in PCR amplified DNA
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
ChimeraSlayer is a chimeric sequence detection utility, compatible with
  near-full length Sanger sequences and shorter 454-FLX sequences (~500bp).
<div class="Pp"></div>
Chimera Slayer involves the following series of steps that operate to flag
  chimeric 16S rRNA sequences:
<dl class="Bl-tag">
  <dt class="It-tag">1.</dt>
  <dd class="It-tag">the ends of a query sequence are searched against an
      included database of reference chimera-free 16S sequences to identify
      potential parents of a chimera</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">2.</dt>
  <dd class="It-tag">candidate parents of a chimera are selected as those that
      form a branched best scoring alignment to the NAST-formatted query
      sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">3.</dt>
  <dd class="It-tag">the NAST alignment of the query sequence is improved in a
      &#x2018;chimera-aware&#x2019; profile-based NAST realignment to the
      selected reference parent sequences</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">4.</dt>
  <dd class="It-tag">an evolutionary framework is used to flag query sequences
      found to exhibit greater sequence homology to an in silico chimera formed
      between any two of the selected reference parent sequences.</dd>
</dl>
<div class="Pp"></div>
To run Chimera Slayer, you need NAST-formatted sequences generated by the
  nast-ier utility.
<div class="Pp"></div>
ChimeraSlayer is part of the microbiomeutil suite.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Required"><a class="selflink" href="#Required">Required</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--query_NAST</b></dt>
  <dd class="It-tag">multi-fasta file containing query sequences in alignment
      format</dd>
</dl>
<h2 class="Ss" title="Ss" id="Common_options"><a class="selflink" href="#Common_options">Common
  options</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--db_NAST</b></dt>
  <dd class="It-tag">db in NAST format (default:
      <i>/usr/share/microbiomeutil-data/RESOURCES/rRNA16S.gold.NAST_ALIGNED.fasta</i>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--db_FASTA</b></dt>
  <dd class="It-tag">db in fasta format (megablast formatted) (default:
      <i>/usr/share/microbiomeutil-data/RESOURCES/rRNA16S.gold.fasta</i>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">number of top matching database sequences to compare to
      (default 15)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b></dt>
  <dd class="It-tag">min divergence ratio default: 1.007</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P</b></dt>
  <dd class="It-tag">min percent identity among matching sequences (default:
    90)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Parameters_to_tune_ChimeraParentSelector:"><a class="selflink" href="#Parameters_to_tune_ChimeraParentSelector:">Parameters
  to tune ChimeraParentSelector:</a></h2>
Scoring parameters:
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b></dt>
  <dd class="It-tag">match score (default: +5)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b></dt>
  <dd class="It-tag">mismatch penalty (default: <b>-4</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b></dt>
  <dd class="It-tag">min query coverage by matching database sequence (default:
      70)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b></dt>
  <dd class="It-tag">maximum traverses of the multiple alignment (default:
    1)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Parameters_to_tune_ChimeraPhyloChecker"><a class="selflink" href="#Parameters_to_tune_ChimeraPhyloChecker">Parameters
  to tune ChimeraPhyloChecker</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--windowSize</b></dt>
  <dd class="It-tag">default 50</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--windowStep</b></dt>
  <dd class="It-tag">default 5</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minBS</b></dt>
  <dd class="It-tag">minimum bootstrap support for calling chimera (default:
    90)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--num_BS_replicates</b></dt>
  <dd class="It-tag">default: 100</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--low_range_finer_BS</b></dt>
  <dd class="It-tag">(default: 10) If computed BS is between minBS and (minBS -
      low_range_finer_BS), then num_finer_BS_replicates computed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--num_finer_BS_replicates</b></dt>
  <dd class="It-tag">(default: 1000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">percent of SNPs to sample on each side of breakpoint for
      computing bootstrap support (default: 10)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--num_parents_test</b></dt>
  <dd class="It-tag">number of potential parents to test for chimeras (default:
      3)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--MAX_CHIMERA_PARENT_PER_ID</b></dt>
  <dd class="It-tag">Chimera/Parent alignments with perID above this are
      considered non-chimeras (default 100; turned off)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Misc_options"><a class="selflink" href="#Misc_options">Misc
  options</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--printFinalAlignments</b></dt>
  <dd class="It-tag">shows alignment between query sequence and pair of
      candidate chimera parents</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--printCSalignments</b></dt>
  <dd class="It-tag">print ChimeraSlayer alignments in ChimeraSlayer output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--exec_dir</b></dt>
  <dd class="It-tag">chdir to here before running</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
http://microbiomeutil.sourceforge.net/#A_CS
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manual page was written by Andreas Tille &lt;tille@debian.org&gt; but can
  be freely used for any other distribution.</div>
<table class="foot">
  <tr>
    <td class="foot-date">June 2013</td>
    <td class="foot-os">chimeraslayer </td>
  </tr>
</table>
</body>
</html>
