<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:59:05 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>clustalo(1) USER COMMANDS clustalo(1)</p>

<p style="margin-top: 1em">NAME <br>
clustalo - General purpose multiple sequence alignment
program for proteins</p>

<p style="margin-top: 1em">SYNOPSIS <br>
clustalo [-h]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Clustal-Omega is a general purpose multiple sequence
alignment (MSA) program for proteins. It produces high
quality MSAs and is capable of handling data-sets of
hundreds of thou&acirc; <br>
sands of sequences in reasonable time.</p>

<p style="margin-top: 1em">In default mode, users give a
file of sequences to be aligned and these are clustered to
produce a guide tree and this is used to guide a
&quot;progressive alignment&quot; of the <br>
sequences. There are also facilities for aligning existing
alignments to each other, aligning a sequence to an
alignment and for using a hidden Markov model (HMM) to help
guide <br>
an alignment of new sequences that are homologous to the
sequences used to make the HMM. This latter procedure is
referred to as &quot;external profile alignment&quot; or
EPA.</p>

<p style="margin-top: 1em">Clustal-Omega uses HMMs for the
alignment engine, based on the HHalign package from Johannes
Soeding [1]. Guide trees are made using an enhanced version
of mBed [2] which can <br>
cluster very large numbers of sequences in O(N*log(N)) time.
Multiple alignment then proceeds by aligning larger and
larger alignments using HHalign, following the clustering
<br>
given by the guide tree.</p>

<p style="margin-top: 1em">In its current form
Clustal-Omega can only align protein sequences but not
DNA/RNA sequences. It is envisioned that DNA/RNA will become
available in a future version.</p>

<p style="margin-top: 1em">USAGE <br>
Tool usage is available in
/usr/share/doc/clustalo/README.</p>

<p style="margin-top: 1em">DEVELOPMENT <br>
Headers and libraries are available in libclustalo-dev
package.</p>

<p style="margin-top: 1em">CITING <br>
Sievers F, Wilm A, Dineen DG, Gibson TJ, Karplus K, Li W,
Lopez R, McWilliam H, <br>
Remmert M, S&Atilde;&para;ding J, Thompson JD, Higgins DG
(2011). Fast, scalable generation of high-quality protein
multiple sequence alignments <br>
using Clustal Omega. Mol Syst Biol 7.</p>

<p style="margin-top: 1em">AUTHOR <br>
Olivier Sallou (olivier.sallou (at) irisa.fr) - Man page and
packaging</p>

<p style="margin-top: 1em">Conway Institute UCD Dublin
(clustalw (at) ucd.ie) - clustalo</p>

<p style="margin-top: 1em">version 1.0.3 December 14, 2011
clustalo(1)</p>
<hr>
</body>
</html>
