<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CLUSTALW(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CLUSTALW(1)</td>
    <td class="head-vol">Clustal Manual</td>
    <td class="head-rtitle">CLUSTALW(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
clustalw - Multiple alignment of nucleic acid and protein sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>clustalw</b>
  [ <b>-infile</b>] <i>file.ext</i> [<b>OPTIONS</b>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>clustalw</b>
  [ <b>-help</b> | <b>-fullhelp</b>]</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Clustal&#x00A0;W is a general purpose multiple alignment program for DNA or
  proteins.
<div class="Pp"></div>
The program performs simultaneous alignment of many nucleotide or amino acid
  sequences. It is typically run interactively, providing a menu and an online
  help. If you prefer to use it in command-line (batch) mode, you will have to
  give several options, the minimum being <b>-infile</b>.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="DATA_(sequences)"><a class="selflink" href="#DATA_(sequences)">DATA
  (sequences)</a></h2>
<b> -infile=</b><b></b><i>file.ext</i><b> </b>
<div style="margin-left: 4.00ex;">Input sequences.</div>
<div class="Pp"></div>
<b> -profile1=</b><b></b><i>file.ext</i><b> and
  -profile2=</b><b></b><i>file.ext</i> <b> </b>
<div style="margin-left: 4.00ex;">Profiles (old alignment)</div>
<h2 class="Ss" title="Ss" id="VERBS_(do_things)"><a class="selflink" href="#VERBS_(do_things)">VERBS
  (do things)</a></h2>
<b>-options</b>
<div style="margin-left: 4.00ex;">List the command line parameters.</div>
<div class="Pp"></div>
<b>-help or -check</b>
<div style="margin-left: 4.00ex;">Outline the command line params.</div>
<div class="Pp"></div>
<b>-fullhelp</b>
<div style="margin-left: 4.00ex;">Output full help content.</div>
<div class="Pp"></div>
<b>-align</b>
<div style="margin-left: 4.00ex;">Do full multiple alignment.</div>
<div class="Pp"></div>
<b>-tree</b>
<div style="margin-left: 4.00ex;">Calculate NJ tree.</div>
<div class="Pp"></div>
<b>-pim</b>
<div style="margin-left: 4.00ex;">Output percent identity matrix (while
  calculating the tree).</div>
<div class="Pp"></div>
<b> -bootstrap</b><b></b><i>=n</i><b> </b>
<div style="margin-left: 4.00ex;">Bootstrap a NJ tree (<i>n</i>= number of
  bootstraps; def. = 1000).</div>
<div class="Pp"></div>
<b>-convert</b>
<div style="margin-left: 4.00ex;">Output the input sequences in a different file
  format.</div>
<h2 class="Ss" title="Ss" id="PARAMETERS_(set_things)"><a class="selflink" href="#PARAMETERS_(set_things)">PARAMETERS
  (set things)</a></h2>
<b>General settings:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-interactive</b>
<div style="margin-left: 4.00ex;">Read command line, then enter normal
  interactive menus.</div>
<div class="Pp"></div>
<b>-quicktree</b>
<div style="margin-left: 4.00ex;">Use FAST algorithm for the alignment guide
  tree.</div>
<div class="Pp"></div>
<b>-type=</b>
<div style="margin-left: 4.00ex;"><i>PROTEIN</i> or <i>DNA</i> sequences.</div>
<div class="Pp"></div>
<b>-negative</b>
<div style="margin-left: 4.00ex;">Protein alignment with negative values in
  matrix.</div>
<div class="Pp"></div>
<b>-outfile=</b>
<div style="margin-left: 4.00ex;">Sequence alignment file name.</div>
<div class="Pp"></div>
<b>-output=</b>
<div style="margin-left: 4.00ex;"><i>GCG</i>, <i>GDE</i>, <i>PHYLIP</i>,
  <i>PIR</i> or <i>NEXUS</i>.</div>
<div class="Pp"></div>
<b>-outputorder=</b>
<div style="margin-left: 4.00ex;"><i>INPUT</i> or <i>ALIGNED</i></div>
<div class="Pp"></div>
<b>-case</b>
<div style="margin-left: 4.00ex;"><i>LOWER</i> or <i>UPPER</i> (for GDE output
  only).</div>
<div class="Pp"></div>
<b>-seqnos=</b>
<div style="margin-left: 4.00ex;"><i>OFF</i> or <i>ON</i> (for Clustal output
  only).</div>
<div class="Pp"></div>
<b>-seqnos_range=</b>
<div style="margin-left: 4.00ex;"><i>OFF</i> or <i>ON</i> (NEW: for all output
  formats).</div>
<div class="Pp"></div>
<b>-range=</b><b></b><i>m</i><b>,</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Sequence range to write starting <i>m</i> to
  <i>m</i>+<i>n</i>.</div>
<div class="Pp"></div>
<b>-maxseqlen=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Maximum allowed input sequence length.</div>
<div class="Pp"></div>
<b>-quiet</b>
<div style="margin-left: 4.00ex;">Reduce console output to minimum.</div>
<div class="Pp"></div>
<b>-stats=</b><b></b><i>file</i>
<div style="margin-left: 4.00ex;">Log some alignments statistics to
  <i>file</i>.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Fast Pairwise Alignments:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-ktuple=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Word size.</div>
<div class="Pp"></div>
<b>-topdiags=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Number of best diags.</div>
<div class="Pp"></div>
<b>-window=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Window around best diags.</div>
<div class="Pp"></div>
<b>-pairgap=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Gap penalty.</div>
<div class="Pp"></div>
<b>-score</b>
<div style="margin-left: 4.00ex;"><i>PERCENT</i> or <i>ABSOLUTE</i>.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Slow Pairwise Alignments:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-pwmatrix=</b>
<div style="margin-left: 4.00ex;">:Protein weight matrix=<i>BLOSUM</i>,
  <i>PAM</i>, <i>GONNET</i>, <i>ID</i> or <i>filename</i></div>
<div class="Pp"></div>
<b>-pwdnamatrix=</b>
<div style="margin-left: 4.00ex;">DNA weight matrix=<i>BLOSUM</i>IUB,
  <i>BLOSUM</i>CLUSTALW or <i>BLOSUM</i>filename.</div>
<div class="Pp"></div>
<b>-pwgapopen=</b><b></b><i>f</i>
<div style="margin-left: 4.00ex;">Gap opening penalty.</div>
<div class="Pp"></div>
<b>-pwgapext=</b><b></b><i>f</i>
<div style="margin-left: 4.00ex;">Gap extension penalty.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Multiple Alignments:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-newtree=</b>
<div style="margin-left: 4.00ex;">File for new guide tree.</div>
<div class="Pp"></div>
<b>-usetree=</b>
<div style="margin-left: 4.00ex;">File for old guide tree.</div>
<div class="Pp"></div>
<b>-matrix=</b>
<div style="margin-left: 4.00ex;">Protein weight matrix=<i>BLOSUM</i>,
  <i>PAM</i>, <i>GONNET</i>, <i>ID</i> or <i>filename</i>.</div>
<div class="Pp"></div>
<b>-dnamatrix=</b>
<div style="margin-left: 4.00ex;">DNA weight matrix=<i>IUB</i>, <i>CLUSTALW</i>
  or <i>filename</i>.</div>
<div class="Pp"></div>
<b>-gapopen=</b><b></b><i>f</i>
<div style="margin-left: 4.00ex;">Gap opening penalty.</div>
<div class="Pp"></div>
<b>-gapext=</b><b></b><i>f</i>
<div style="margin-left: 4.00ex;">Gap extension penalty.</div>
<div class="Pp"></div>
<b>-engaps</b>
<div style="margin-left: 4.00ex;">No end gap separation pen.</div>
<div class="Pp"></div>
<b>-gapdist=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Gap separation pen. range.</div>
<div class="Pp"></div>
<b>-nogap</b>
<div style="margin-left: 4.00ex;">Residue-specific gaps off.</div>
<div class="Pp"></div>
<b>-nohgap</b>
<div style="margin-left: 4.00ex;">Hydrophilic gaps off.</div>
<div class="Pp"></div>
<b>-hgapresidues=</b>
<div style="margin-left: 4.00ex;">List hydrophilic res.</div>
<div class="Pp"></div>
<b>-maxdiv=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Percent identity for delay.</div>
<div class="Pp"></div>
<b>-type=</b>
<div style="margin-left: 4.00ex;"><i>PROTEIN</i> or <i>DNA</i></div>
<div class="Pp"></div>
<b>-transweight=</b><b></b><i>f</i>
<div style="margin-left: 4.00ex;">Transitions weighting.</div>
<div class="Pp"></div>
<b>-iteration=</b>
<div style="margin-left: 4.00ex;"><i>NONE</i> or <i>TREE</i> or
  <i>ALIGNMENT</i>.</div>
<div class="Pp"></div>
<b>-numiter=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Maximum number of iterations to perform.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Profile Alignments:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-profile</b>
<div style="margin-left: 4.00ex;">Merge two alignments by profile
  alignment.</div>
<div class="Pp"></div>
<b>-newtree1=</b>
<div style="margin-left: 4.00ex;">File for new guide tree for profile1.</div>
<div class="Pp"></div>
<b>-newtree2=</b>
<div style="margin-left: 4.00ex;">File for new guide tree for profile2.</div>
<div class="Pp"></div>
<b>-usetree1=</b>
<div style="margin-left: 4.00ex;">File for old guide tree for profile1.</div>
<div class="Pp"></div>
<b>-usetree2=</b>
<div style="margin-left: 4.00ex;">File for old guide tree for profile2.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Sequence to Profile Alignments:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-sequences</b>
<div style="margin-left: 4.00ex;">Sequentially add profile2 sequences to
  profile1 alignment.</div>
<div class="Pp"></div>
<b>-newtree=</b>
<div style="margin-left: 4.00ex;">File for new guide tree.</div>
<div class="Pp"></div>
<b>-usetree=</b>
<div style="margin-left: 4.00ex;">File for old guide tree.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Structure Alignments:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-nosecstr1</b>
<div style="margin-left: 4.00ex;">Do not use secondary structure-gap penalty
  mask for profile 1.</div>
<div class="Pp"></div>
<b>-nosecstr2</b>
<div style="margin-left: 4.00ex;">Do not use secondary structure-gap penalty
  mask for profile 2.</div>
<div class="Pp"></div>
<b>-secstrout=</b><b></b><i>STRUCTURE</i><b> or </b><b></b><i>MASK</i><b> or
  </b> <b></b><i>BOTH</i><b> or </b><b></b><i>NONE</i>
<div style="margin-left: 4.00ex;">Output in alignment file.</div>
<div class="Pp"></div>
<b>-helixgap=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Gap penalty for helix core residues.</div>
<div class="Pp"></div>
<b>-strandgap=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Gap penalty for strand core residues.</div>
<div class="Pp"></div>
<b>loopgap=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Gap penalty for loop regions.</div>
<div class="Pp"></div>
<b>-terminalgap=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Gap penalty for structure termini.</div>
<div class="Pp"></div>
<b>-helixendin=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Number of residues inside helix to be treated
  as terminal.</div>
<div class="Pp"></div>
<b>-helixendout=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Number of residues outside helix to be treated
  as terminal.</div>
<div class="Pp"></div>
<b>-strandendin=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Number of residues inside strand to be treated
  as terminal.</div>
<div class="Pp"></div>
<b>-strandendout=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Number of residues outside strand to be
  treated as terminal.</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
<b>Trees:</b>
<div>&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<div class="Pp"></div>
<b>-outputtree=</b><b></b><i>nj</i><b> OR </b><b></b><i>phylip</i><b> OR
  </b><b></b> <i>dist</i><b> OR </b><b></b><i>nexus</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-seed=</b><b></b><i>n</i>
<div style="margin-left: 4.00ex;">Seed number for bootstraps.</div>
<div class="Pp"></div>
<b>-kimura</b>
<div style="margin-left: 4.00ex;">Use Kimura's correction.</div>
<div class="Pp"></div>
<b>-tossgaps</b>
<div style="margin-left: 4.00ex;">Ignore positions with gaps.</div>
<div class="Pp"></div>
<b>-bootlabels=</b><b></b><i>node</i>
<div style="margin-left: 4.00ex;">Position of bootstrap values in tree
  display.</div>
<div class="Pp"></div>
<b>-clustering=</b>
<div style="margin-left: 4.00ex;">NJ or UPGMA.</div>
</div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
The Clustal bug tracking system can be found at
  <b>http://bioinf.ucd.ie/bugzilla/buglist.cgi?quicksearch=clustal</b>.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>clustalx</b>(1).
<h1 class="Sh" title="Sh" id="REFERENCES"><a class="selflink" href="#REFERENCES">REFERENCES</a></h1>
<div style="margin-left: 4.00ex;">&#x2022;Larkin MA, Blackshields G, Brown NP,
  Chenna R, McGettigan PA, McWilliam H, Valentin F, Wallace IM, Wilm A, Lopez R,
  Thompson JD, Gibson TJ, Higgins DG. (2007). <b>Clustal W and Clustal X version
  2.0.</b>[1] Bioinformatics, 23, 2947-2948.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Chenna R, Sugawara H, Koike T, Lopez
  R, Gibson TJ, Higgins DG, Thompson JD. (2003). <b>Multiple sequence alignment
  with the Clustal series of programs.</b>[2] Nucleic Acids Res., 31,
  3497-3500.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Jeanmougin F, Thompson JD, Gouy M,
  Higgins DG, Gibson TJ. (1998). <b>Multiple sequence alignment with Clustal
  X</b>[3]. Trends Biochem Sci., 23, 403-405.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Thompson JD, Gibson TJ, Plewniak F,
  Jeanmougin F, Higgins DG. (1997). <b>The CLUSTAL_X windows interface: flexible
  strategies for multiple sequence alignment aided by quality analysis
  tools.</b>[4] Nucleic Acids Res., 25, 4876-4882.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Higgins DG, Thompson JD, Gibson TJ.
  (1996). <b>Using CLUSTAL for multiple sequence alignments.</b>[5] Methods
  Enzymol., 266, 383-402.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Thompson JD, Higgins DG, Gibson TJ.
  (1994). <b>CLUSTAL W: improving the sensitivity of progressive multiple
  sequence alignment through sequence weighting, position-specific gap penalties
  and weight matrix choice.</b>[6] Nucleic Acids Res., 22, 4673-4680.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Higgins DG. (1994). <b>CLUSTAL V:
  multiple alignment of DNA and protein sequences.</b>[7] Methods Mol Biol., 25,
  307-318</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Higgins DG, Bleasby AJ, Fuchs R.
  (1992). <b>CLUSTAL V: improved software for multiple sequence
  alignment.</b>[8] Comput. Appl. Biosci., 8, 189-191.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Higgins,D.G. and Sharp,P.M. (1989).
  <b>Fast and sensitive multiple sequence alignments on a microcomputer.</b>[9]
  Comput. Appl. Biosci., 5, 151-153.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Higgins,D.G. and Sharp,P.M. (1988).
  <b>CLUSTAL: a package for performing multiple sequence alignment on a
  microcomputer.</b>[10] Gene, 73, 237-244.</div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<b>Des Higgins</b>
<div style="margin-left: 4.00ex;">Copyright holder for Clustal.</div>
<div class="Pp"></div>
<b>Julie Thompson</b>
<div style="margin-left: 4.00ex;">Copyright holder for Clustal.</div>
<div class="Pp"></div>
<b>Toby Gibson</b>
<div style="margin-left: 4.00ex;">Copyright holder for Clustal.</div>
<div class="Pp"></div>
<b>Charles Plessy</b> &lt;plessy@debian.org&gt;
<div style="margin-left: 4.00ex;">Prepared this manpage in DocBook XML for the
  Debian distribution.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright &#x00A9; 1988&#x2013;2010 Des Higgins, Julie Thompson &amp; Toby
  Giboson (Clustal)
<div>&#x00A0;</div>
Copyright &#x00A9; 2008&#x2013;2010 Charles Plessy (This manpage)
<div>&#x00A0;</div>
<div class="Pp"></div>
This program is free software: you can redistribute it and/or modify it under
  the terms of the GNU Lesser General Public License as published by the Free
  Software Foundation, either version 3 of the License, or (at your option) any
  later version.
<div class="Pp"></div>
This program is distributed in the hope that it will be useful, but WITHOUT ANY
  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
  A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
  details.
<div class="Pp"></div>
You should have received a copy of the GNU Lesser General Public License along
  with this program. If not, see http://www.gnu.org/licenses/, or on Debian
  systems, /usr/share/common-licenses/LGPL-3.
<div class="Pp"></div>
This manual page and its XML source can be used, modified, and redistributed as
  if it were in public domain.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="NOTES"><a class="selflink" href="#NOTES">NOTES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"> 1.</dt>
  <dd class="It-tag">Clustal W and Clustal X version 2.0.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/17846036</div>
<dl class="Bl-tag">
  <dt class="It-tag"> 2.</dt>
  <dd class="It-tag">Multiple sequence alignment with the Clustal series of
      programs.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/12824352</div>
<dl class="Bl-tag">
  <dt class="It-tag"> 3.</dt>
  <dd class="It-tag">Multiple sequence alignment with Clustal X</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/9810230</div>
<dl class="Bl-tag">
  <dt class="It-tag"> 4.</dt>
  <dd class="It-tag">The CLUSTAL_X windows interface: flexible strategies for
      multiple sequence alignment aided by quality analysis tools.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/9396791</div>
<dl class="Bl-tag">
  <dt class="It-tag"> 5.</dt>
  <dd class="It-tag">Using CLUSTAL for multiple sequence alignments.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/8743695</div>
<dl class="Bl-tag">
  <dt class="It-tag"> 6.</dt>
  <dd class="It-tag">CLUSTAL W: improving the sensitivity of progressive
      multiple sequence alignment through sequence weighting, position-specific
      gap penalties and weight matrix choice.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/7984417</div>
<dl class="Bl-tag">
  <dt class="It-tag"> 7.</dt>
  <dd class="It-tag">CLUSTAL V: multiple alignment of DNA and protein
    sequences.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/8004173</div>
<dl class="Bl-tag">
  <dt class="It-tag"> 8.</dt>
  <dd class="It-tag">CLUSTAL V: improved software for multiple sequence
      alignment.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/1591615</div>
<dl class="Bl-tag">
  <dt class="It-tag"> 9.</dt>
  <dd class="It-tag">Fast and sensitive multiple sequence alignments on a
      microcomputer.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/2720464</div>
<dl class="Bl-tag">
  <dt class="It-tag">10.</dt>
  <dd class="It-tag">CLUSTAL: a package for performing multiple sequence
      alignment on a microcomputer.</dd>
</dl>
<div style="margin-left: 4.00ex;">http://www.ncbi.nlm.nih.gov/pubmed/3243435</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">12/28/2010</td>
    <td class="foot-os">Clustal 2.1</td>
  </tr>
</table>
</body>
</html>
