<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:59:05 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CLUSTALW(1) Clustal Manual CLUSTALW(1)</p>

<p style="margin-top: 1em">NAME <br>
clustalw - Multiple alignment of nucleic acid and protein
sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
clustalw [-infile] file.ext [OPTIONS]</p>

<p style="margin-top: 1em">clustalw [-help | -fullhelp]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Clustal W is a general purpose multiple alignment program
for DNA or proteins.</p>

<p style="margin-top: 1em">The program performs
simultaneous alignment of many nucleotide or amino acid
sequences. It is typically run interactively, providing a
menu and an online help. If you prefer to <br>
use it in command-line (batch) mode, you will have to give
several options, the minimum being -infile.</p>

<p style="margin-top: 1em">OPTIONS <br>
DATA (sequences) <br>
-infile=file.ext <br>
Input sequences.</p>

<p style="margin-top: 1em">-profile1=file.ext and
-profile2=file.ext <br>
Profiles (old alignment)</p>

<p style="margin-top: 1em">VERBS (do things) <br>
-options <br>
List the command line parameters.</p>

<p style="margin-top: 1em">-help or -check <br>
Outline the command line params.</p>

<p style="margin-top: 1em">-fullhelp <br>
Output full help content.</p>

<p style="margin-top: 1em">-align <br>
Do full multiple alignment.</p>

<p style="margin-top: 1em">-tree <br>
Calculate NJ tree.</p>

<p style="margin-top: 1em">-pim <br>
Output percent identity matrix (while calculating the
tree).</p>

<p style="margin-top: 1em">-bootstrap=n <br>
Bootstrap a NJ tree (n= number of bootstraps; def. =
1000).</p>

<p style="margin-top: 1em">-convert <br>
Output the input sequences in a different file format.</p>

<p style="margin-top: 1em">PARAMETERS (set things) <br>
General settings: <br>
-interactive <br>
Read command line, then enter normal interactive menus.</p>

<p style="margin-top: 1em">-quicktree <br>
Use FAST algorithm for the alignment guide tree.</p>

<p style="margin-top: 1em">-type= <br>
PROTEIN or DNA sequences.</p>

<p style="margin-top: 1em">-negative <br>
Protein alignment with negative values in matrix.</p>

<p style="margin-top: 1em">-outfile= <br>
Sequence alignment file name.</p>

<p style="margin-top: 1em">-output= <br>
GCG, GDE, PHYLIP, PIR or NEXUS.</p>

<p style="margin-top: 1em">-outputorder= <br>
INPUT or ALIGNED</p>

<p style="margin-top: 1em">-case <br>
LOWER or UPPER (for GDE output only).</p>

<p style="margin-top: 1em">-seqnos= <br>
OFF or ON (for Clustal output only).</p>

<p style="margin-top: 1em">-seqnos_range= <br>
OFF or ON (NEW: for all output formats).</p>

<p style="margin-top: 1em">-range=m,n <br>
Sequence range to write starting m to m+n.</p>

<p style="margin-top: 1em">-maxseqlen=n <br>
Maximum allowed input sequence length.</p>

<p style="margin-top: 1em">-quiet <br>
Reduce console output to minimum.</p>

<p style="margin-top: 1em">-stats=file <br>
Log some alignments statistics to file.</p>

<p style="margin-top: 1em">Fast Pairwise Alignments: <br>
-ktuple=n <br>
Word size.</p>

<p style="margin-top: 1em">-topdiags=n <br>
Number of best diags.</p>

<p style="margin-top: 1em">-window=n <br>
Window around best diags.</p>

<p style="margin-top: 1em">-pairgap=n <br>
Gap penalty.</p>

<p style="margin-top: 1em">-score <br>
PERCENT or ABSOLUTE.</p>

<p style="margin-top: 1em">Slow Pairwise Alignments: <br>
-pwmatrix= <br>
:Protein weight matrix=BLOSUM, PAM, GONNET, ID or
filename</p>

<p style="margin-top: 1em">-pwdnamatrix= <br>
DNA weight matrix=BLOSUMIUB, BLOSUMCLUSTALW or
BLOSUMfilename.</p>

<p style="margin-top: 1em">-pwgapopen=f <br>
Gap opening penalty.</p>

<p style="margin-top: 1em">-pwgapext=f <br>
Gap extension penalty.</p>

<p style="margin-top: 1em">Multiple Alignments: <br>
-newtree= <br>
File for new guide tree.</p>

<p style="margin-top: 1em">-usetree= <br>
File for old guide tree.</p>

<p style="margin-top: 1em">-matrix= <br>
Protein weight matrix=BLOSUM, PAM, GONNET, ID or
filename.</p>

<p style="margin-top: 1em">-dnamatrix= <br>
DNA weight matrix=IUB, CLUSTALW or filename.</p>

<p style="margin-top: 1em">-gapopen=f <br>
Gap opening penalty.</p>

<p style="margin-top: 1em">-gapext=f <br>
Gap extension penalty.</p>

<p style="margin-top: 1em">-engaps <br>
No end gap separation pen.</p>

<p style="margin-top: 1em">-gapdist=n <br>
Gap separation pen. range.</p>

<p style="margin-top: 1em">-nogap <br>
Residue-specific gaps off.</p>

<p style="margin-top: 1em">-nohgap <br>
Hydrophilic gaps off.</p>

<p style="margin-top: 1em">-hgapresidues= <br>
List hydrophilic res.</p>

<p style="margin-top: 1em">-maxdiv=n <br>
Percent identity for delay.</p>

<p style="margin-top: 1em">-type= <br>
PROTEIN or DNA</p>

<p style="margin-top: 1em">-transweight=f <br>
Transitions weighting.</p>

<p style="margin-top: 1em">-iteration= <br>
NONE or TREE or ALIGNMENT.</p>

<p style="margin-top: 1em">-numiter=n <br>
Maximum number of iterations to perform.</p>

<p style="margin-top: 1em">Profile Alignments: <br>
-profile <br>
Merge two alignments by profile alignment.</p>

<p style="margin-top: 1em">-newtree1= <br>
File for new guide tree for profile1.</p>

<p style="margin-top: 1em">-newtree2= <br>
File for new guide tree for profile2.</p>

<p style="margin-top: 1em">-usetree1= <br>
File for old guide tree for profile1.</p>

<p style="margin-top: 1em">-usetree2= <br>
File for old guide tree for profile2.</p>

<p style="margin-top: 1em">Sequence to Profile Alignments:
<br>
-sequences <br>
Sequentially add profile2 sequences to profile1
alignment.</p>

<p style="margin-top: 1em">-newtree= <br>
File for new guide tree.</p>

<p style="margin-top: 1em">-usetree= <br>
File for old guide tree.</p>

<p style="margin-top: 1em">Structure Alignments: <br>
-nosecstr1 <br>
Do not use secondary structure-gap penalty mask for profile
1.</p>

<p style="margin-top: 1em">-nosecstr2 <br>
Do not use secondary structure-gap penalty mask for profile
2.</p>

<p style="margin-top: 1em">-secstrout=STRUCTURE or MASK or
BOTH or NONE <br>
Output in alignment file.</p>

<p style="margin-top: 1em">-helixgap=n <br>
Gap penalty for helix core residues.</p>

<p style="margin-top: 1em">-strandgap=n <br>
Gap penalty for strand core residues.</p>

<p style="margin-top: 1em">loopgap=n <br>
Gap penalty for loop regions.</p>

<p style="margin-top: 1em">-terminalgap=n <br>
Gap penalty for structure termini.</p>

<p style="margin-top: 1em">-helixendin=n <br>
Number of residues inside helix to be treated as
terminal.</p>

<p style="margin-top: 1em">-helixendout=n <br>
Number of residues outside helix to be treated as
terminal.</p>

<p style="margin-top: 1em">-strandendin=n <br>
Number of residues inside strand to be treated as
terminal.</p>

<p style="margin-top: 1em">-strandendout=n <br>
Number of residues outside strand to be treated as
terminal.</p>

<p style="margin-top: 1em">Trees: <br>
-outputtree=nj OR phylip OR dist OR nexus</p>

<p style="margin-top: 1em">-seed=n <br>
Seed number for bootstraps.</p>

<p style="margin-top: 1em">-kimura <br>
Use Kimura&rsquo;s correction.</p>

<p style="margin-top: 1em">-tossgaps <br>
Ignore positions with gaps.</p>

<p style="margin-top: 1em">-bootlabels=node <br>
Position of bootstrap values in tree display.</p>

<p style="margin-top: 1em">-clustering= <br>
NJ or UPGMA.</p>

<p style="margin-top: 1em">BUGS <br>
The Clustal bug tracking system can be found at
http://bioinf.ucd.ie/bugzilla/buglist.cgi?quicksearch=clustal.</p>

<p style="margin-top: 1em">SEE ALSO <br>
clustalx(1).</p>

<p style="margin-top: 1em">REFERENCES <br>
&Acirc;&middot; Larkin MA, Blackshields G, Brown NP, Chenna
R, McGettigan PA, McWilliam H, Valentin F, Wallace IM, Wilm
A, Lopez R, Thompson JD, Gibson TJ, Higgins DG. (2007).
Clustal W and <br>
Clustal X version 2.0.[1] Bioinformatics, 23, 2947-2948.</p>

<p style="margin-top: 1em">&Acirc;&middot; Chenna R,
Sugawara H, Koike T, Lopez R, Gibson TJ, Higgins DG,
Thompson JD. (2003). Multiple sequence alignment with the
Clustal series of programs.[2] Nucleic Acids Res., <br>
31, 3497-3500.</p>

<p style="margin-top: 1em">&Acirc;&middot; Jeanmougin F,
Thompson JD, Gouy M, Higgins DG, Gibson TJ. (1998). Multiple
sequence alignment with Clustal X[3]. Trends Biochem Sci.,
23, 403-405.</p>

<p style="margin-top: 1em">&Acirc;&middot; Thompson JD,
Gibson TJ, Plewniak F, Jeanmougin F, Higgins DG. (1997). The
CLUSTAL_X windows interface: flexible strategies for
multiple sequence alignment aided by quality <br>
analysis tools.[4] Nucleic Acids Res., 25, 4876-4882.</p>

<p style="margin-top: 1em">&Acirc;&middot; Higgins DG,
Thompson JD, Gibson TJ. (1996). Using CLUSTAL for multiple
sequence alignments.[5] Methods Enzymol., 266, 383-402.</p>

<p style="margin-top: 1em">&Acirc;&middot; Thompson JD,
Higgins DG, Gibson TJ. (1994). CLUSTAL W: improving the
sensitivity of progressive multiple sequence alignment
through sequence weighting, position-specific gap <br>
penalties and weight matrix choice.[6] Nucleic Acids Res.,
22, 4673-4680.</p>

<p style="margin-top: 1em">&Acirc;&middot; Higgins DG.
(1994). CLUSTAL V: multiple alignment of DNA and protein
sequences.[7] Methods Mol Biol., 25, 307-318</p>

<p style="margin-top: 1em">&Acirc;&middot; Higgins DG,
Bleasby AJ, Fuchs R. (1992). CLUSTAL V: improved software
for multiple sequence alignment.[8] Comput. Appl. Biosci.,
8, 189-191.</p>

<p style="margin-top: 1em">&Acirc;&middot; Higgins,D.G. and
Sharp,P.M. (1989). Fast and sensitive multiple sequence
alignments on a microcomputer.[9] Comput. Appl. Biosci., 5,
151-153.</p>

<p style="margin-top: 1em">&Acirc;&middot; Higgins,D.G. and
Sharp,P.M. (1988). CLUSTAL: a package for performing
multiple sequence alignment on a microcomputer.[10] Gene,
73, 237-244.</p>

<p style="margin-top: 1em">AUTHORS <br>
Des Higgins <br>
Copyright holder for Clustal.</p>

<p style="margin-top: 1em">Julie Thompson <br>
Copyright holder for Clustal.</p>

<p style="margin-top: 1em">Toby Gibson <br>
Copyright holder for Clustal.</p>

<p style="margin-top: 1em">Charles Plessy
&lt;plessy@debian.org&gt; <br>
Prepared this manpage in DocBook XML for the Debian
distribution.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 1988&acirc;2010 Des Higgins, Julie
Thompson &amp; Toby Giboson (Clustal) <br>
Copyright &Acirc;&copy; 2008&acirc;2010 Charles Plessy (This
manpage)</p>

<p style="margin-top: 1em">This program is free software:
you can redistribute it and/or modify it under the terms of
the GNU Lesser General Public License as published by the
Free Software Foundation, <br>
either version 3 of the License, or (at your option) any
later version.</p>

<p style="margin-top: 1em">This program is distributed in
the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR <br>
PURPOSE. See the GNU Lesser General Public License for more
details.</p>

<p style="margin-top: 1em">You should have received a copy
of the GNU Lesser General Public License along with this
program. If not, see http://www.gnu.org/licenses/, or on
Debian systems, <br>
/usr/share/common-licenses/LGPL-3.</p>

<p style="margin-top: 1em">This manual page and its XML
source can be used, modified, and redistributed as if it
were in public domain.</p>

<p style="margin-top: 1em">NOTES <br>
1. Clustal W and Clustal X version 2.0. <br>
http://www.ncbi.nlm.nih.gov/pubmed/17846036</p>

<p style="margin-top: 1em">2. Multiple sequence alignment
with the Clustal series of programs. <br>
http://www.ncbi.nlm.nih.gov/pubmed/12824352</p>

<p style="margin-top: 1em">3. Multiple sequence alignment
with Clustal X <br>
http://www.ncbi.nlm.nih.gov/pubmed/9810230</p>

<p style="margin-top: 1em">4. The CLUSTAL_X windows
interface: flexible strategies for multiple sequence
alignment aided by quality analysis tools. <br>
http://www.ncbi.nlm.nih.gov/pubmed/9396791</p>

<p style="margin-top: 1em">5. Using CLUSTAL for multiple
sequence alignments. <br>
http://www.ncbi.nlm.nih.gov/pubmed/8743695</p>

<p style="margin-top: 1em">6. CLUSTAL W: improving the
sensitivity of progressive multiple sequence alignment
through sequence weighting, position-specific gap penalties
and weight matrix choice. <br>
http://www.ncbi.nlm.nih.gov/pubmed/7984417</p>

<p style="margin-top: 1em">7. CLUSTAL V: multiple alignment
of DNA and protein sequences. <br>
http://www.ncbi.nlm.nih.gov/pubmed/8004173</p>

<p style="margin-top: 1em">8. CLUSTAL V: improved software
for multiple sequence alignment. <br>
http://www.ncbi.nlm.nih.gov/pubmed/1591615</p>

<p style="margin-top: 1em">9. Fast and sensitive multiple
sequence alignments on a microcomputer. <br>
http://www.ncbi.nlm.nih.gov/pubmed/2720464</p>

<p style="margin-top: 1em">10. CLUSTAL: a package for
performing multiple sequence alignment on a microcomputer.
<br>
http://www.ncbi.nlm.nih.gov/pubmed/3243435</p>

<p style="margin-top: 1em">Clustal 2.1 12/28/2010
CLUSTALW(1)</p>
<hr>
</body>
</html>
