<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>clustalo(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">clustalo(1)</td>
    <td class="head-vol">USER COMMANDS</td>
    <td class="head-rtitle">clustalo(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
clustalo - General purpose multiple sequence alignment program for proteins
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>clustalo</b> [-h]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Clustal-Omega is a general purpose multiple sequence alignment (MSA) program for
  proteins. It produces high quality MSAs and is capable of handling data-sets
  of hundreds of thousands of sequences in reasonable time.
<div style="height: 1.00em;">&#x00A0;</div>
In default mode, users give a file of sequences to be aligned and these are
  clustered to produce a guide tree and this is used to guide a
  &quot;progressive alignment&quot; of the sequences. There are also facilities
  for aligning existing alignments to each other, aligning a sequence to an
  alignment and for using a hidden Markov model (HMM) to help guide an alignment
  of new sequences that are homologous to the sequences used to make the HMM.
  This latter procedure is referred to as &quot;external profile alignment&quot;
  or EPA.
<div style="height: 1.00em;">&#x00A0;</div>
Clustal-Omega uses HMMs for the alignment engine, based on the HHalign package
  from Johannes Soeding [1]. Guide trees are made using an enhanced version of
  mBed [2] which can cluster very large numbers of sequences in O(N*log(N))
  time. Multiple alignment then proceeds by aligning larger and larger
  alignments using HHalign, following the clustering given by the guide tree.
<div style="height: 1.00em;">&#x00A0;</div>
In its current form Clustal-Omega can only align protein sequences but not
  DNA/RNA sequences. It is envisioned that DNA/RNA will become available in a
  future version.
<h1 class="Sh" title="Sh" id="USAGE"><a class="selflink" href="#USAGE">USAGE</a></h1>
Tool usage is available in /usr/share/doc/clustalo/README.
<h1 class="Sh" title="Sh" id="DEVELOPMENT"><a class="selflink" href="#DEVELOPMENT">DEVELOPMENT</a></h1>
Headers and libraries are available in libclustalo-dev package.
<h1 class="Sh" title="Sh" id="CITING"><a class="selflink" href="#CITING">CITING</a></h1>
Sievers F, Wilm A, Dineen DG, Gibson TJ, Karplus K, Li W, Lopez R, McWilliam H,
<br/>
 Remmert M, S&#x00F6;ding J, Thompson JD, Higgins DG (2011). Fast, scalable
  generation of high-quality protein multiple sequence alignments
<br/>
 using Clustal Omega. Mol Syst Biol 7.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Olivier Sallou (olivier.sallou (at) irisa.fr) - Man page and packaging
<div style="height: 1.00em;">&#x00A0;</div>
Conway Institute UCD Dublin (clustalw (at) ucd.ie) - clustalo
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">December 14, 2011</td>
    <td class="foot-os">version 1.0.3</td>
  </tr>
</table>
</body>
</html>
