<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:59:20 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>cmalign(1) Infernal Manual cmalign(1)</p>

<p style="margin-top: 1em">NAME <br>
cmalign - align sequences to a covariance model</p>

<p style="margin-top: 1em">SYNOPSIS <br>
cmalign <br>
[options] &lt;cmfile&gt; &lt;seqfile&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
cmalign aligns the RNA sequences in &lt;seqfile&gt; to the
covariance model (CM) in &lt;cmfile&gt;. The new alignment
is output to stdout in Stockholm format, but can be
redirected to a <br>
file &lt;f&gt; with the -o &lt;f&gt; option.</p>

<p style="margin-top: 1em">Either &lt;cmfile&gt; or
&lt;seqfile&gt; (but not both) may be &rsquo;-&rsquo;
(dash), which means reading this input from stdin rather
than a file.</p>

<p style="margin-top: 1em">The sequence file
&lt;seqfile&gt; must be in FASTA or Genbank format.</p>

<p style="margin-top: 1em">cmalign uses an HMM banding
technique to accelerate alignment by default as described
below for the --hbanded option. HMM banding can be turned
off with the --nonbanded option.</p>

<p style="margin-top: 1em">By default, cmalign computes the
alignment with maximum expected accuracy that is consistent
with constraints (bands) derived from an HMM, using a banded
version of the <br>
Durbin/Holmes optimal accuracy algorithm. This behavior can
be changed with the --cyk or --sample options.</p>

<p style="margin-top: 1em">cmalign takes special care to
correctly align truncated sequences, where some nucleotides
from the beginning (5&rsquo;) and/or end (3&rsquo;) of the
actual full length biological sequence <br>
are not present in the input sequence (see DL Kolbe and SR
Eddy, Bioinformatics, 25:1236-1243, 2009). This behavior is
on by default, but can be turned off with --notrunc. In <br>
previous versions of cmalign the --sub option was required
to appropriately handle truncated sequences. The --sub
option is still available in this version, but the new
default <br>
method for handling truncated sequences should be as good or
superior to the sub method in nearly all cases.</p>

<p style="margin-top: 1em">The --mapali &lt;s&gt; option
allows inclusion of the fixed training alignment used to
build the CM from file &lt;s&gt; within the output alignment
of cmalign.</p>

<p style="margin-top: 1em">It is possible to merge two or
more alignments created by the same CM using the Easel
miniapp esl-alimerge (included in the easel/miniapps/
subdirectory of Infernal). Previous <br>
versions of cmalign included options to merge alignments but
they were deprecated upon development of esl-alimerge, which
is significantly more memory efficient.</p>

<p style="margin-top: 1em">By default, cmalign will output
the alignment to stdout. The alignment can be redirected to
an output file &lt;f&gt; with the -o &lt;f&gt; option. With
-o, information on each aligned <br>
sequence, including score and model alignment boundaries
will be printed to stdout (more on this below).</p>

<p style="margin-top: 1em">The output alignment will be in
Stockholm format by default. This can be changed to Pfam,
aligned FASTA (AFA), A2M, Clustal, or Phylip format using
the --outformat &lt;s&gt; option, <br>
where &lt;s&gt; is the name of the desired format. As a
special case, if the output alignment is large (more than
10,000 sequences or more than 10,000,000 total nucleotides)
than the <br>
output format will be Pfam format, with each sequence
appearing on a single line, for reasons of memory
efficiency. For alignments larger than this, using --ileaved
will force <br>
interleaved Stockholm format, but the user should be aware
that this may require a lot of memory. --ileaved will only
work for alignments up to 100,000 sequences or 100,000,000
<br>
total nucleotides.</p>

<p style="margin-top: 1em">If the output alignment format
is Stockholm or Pfam, the output alignment will be annotated
with posterior probabilities which estimate the confidence
level of each aligned nu&acirc; <br>
cleotide. This annotation appears as lines beginning with
&quot;#=GR &lt;seq name&gt; PP&quot;, one per sequence, each
immediately below the corresponding aligned sequence
&quot;&lt;seq name&gt;&quot;. Charac&acirc; <br>
ters in PP lines have 12 possible values: &quot;0-9&quot;,
&quot;*&quot;, or &quot;.&quot;. If &quot;.&quot;, the
position corresponds to a gap in the sequence. A value of
&quot;0&quot; indicates a posterior probability of <br>
between 0.0 and 0.05, &quot;1&quot; indicates between 0.05
and 0.15, &quot;2&quot; indicates between 0.15 and 0.25 and
so on up to &quot;9&quot; which indicates between 0.85 and
0.95. A value of &quot;*&quot; indicates <br>
a posterior probability of between 0.95 and 1.0. Higher
posterior probabilities correspond to greater confidence
that the aligned nucleotide belongs where it appears in the
<br>
alignment. With --nonbanded, the calculation of the
posterior probabilities considers all possible alignments of
the target sequence to the CM. Without --nonbanded (i.e. in
<br>
default mode), the calculation considers only possible
alignments within the HMM bands. Further, the posterior
probabilities are conditional on the truncation mode of the
align&acirc; <br>
ment. For example, if the sequence alignment is truncated
5&rsquo;, a PP value of &quot;9&quot; indicates between 0.85
and 0.95 of all 5&rsquo; truncated alignments include the
given nucleotide at the <br>
given position. The posterior annotation can be turned off
with the --noprob option. If --small is enabled, posterior
annotation must also be turned off using --noprob.</p>

<p style="margin-top: 1em">The tabular output that is
printed to stdout if the -o option is used includes one line
per sequence and twelve fields per line: &quot;idx&quot;:
the index of the sequence in the input <br>
file, &quot;seq name&quot;: the sequence name;
&quot;length&quot;: the length of the sequence; &quot;cm
from&quot; and &quot;cm to&quot;: the model start and end
positions of the alignment; &quot;trunc&quot;:
&quot;no&quot; if the <br>
sequence is not truncated, &quot;5&rsquo;&quot; if the
beginning of the sequence truncated 5&rsquo;,
&quot;3&rsquo;&quot; if the end of the sequence is
truncated, and &quot;5&rsquo;&amp;3&rsquo;&quot; if both the
beginning and the end are <br>
truncated; &quot;bit sc&quot;: the bit score of the
alignment, &quot;avg pp&quot; the average posterior
probability of all aligned nucleotides in the alignment;
&quot;band calc&quot;, &quot;alignment&quot; and
&quot;total&quot;: <br>
the time in seconds required for calculating HMM bands,
computing the alignment, and complete processing of the
sequence, respectively; &quot;mem (Mb)&quot;: the size in Mb
of all dynamic <br>
programming matrices required for aligning the sequence.
This tabular data can be saved to file &lt;f&gt; with the
--sfile &lt;f&gt; option.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
available options.</p>

<p style="margin-top: 1em">-o &lt;f&gt; Save the alignment
in Stockholm format to a file &lt;f&gt;. The default is to
write it to standard output.</p>

<p style="margin-top: 1em">-g Configure the model for
global alignment of the query model to the target sequences.
By default, the model is configured for local alignment.
Local alignments can contain <br>
large insertions and deletions called &quot;local ends&quot;
in the structure to be penalized differently than normal
indels. These are annotated as &quot;~&quot; columns in the
RF line of <br>
the output alignment. The -g option can be used to disallow
these local ends. The -g option is required if the --sub
option is also used.</p>

<p style="margin-top: 1em">OPTIONS FOR CONTROLLING THE
ALIGNMENT ALGORITHM <br>
--optacc <br>
Align sequences using the Durbin/Holmes optimal accuracy
algorithm. This is the default. The optimal accuracy
alignment will be constrained by HMM bands for acceleration
<br>
unless the --nonbanded option is enabled. The optimal
accuracy algorithm determines the alignment that maximizes
the posterior probabilities of the aligned nucleotides <br>
within it. The posterior probabilites are determined using
(possibly HMM banded) variants of the Inside and Outside
algorithms.</p>

<p style="margin-top: 1em">--cyk Do not use the
Durbin/Holmes optimal accuracy alignment to align the
sequences, instead use the CYK algorithm which determines
the optimally scoring (maximum likelihood) <br>
alignment of the sequence to the model, given the HMM bands
(unless --nonbanded is also enabled).</p>

<p style="margin-top: 1em">--sample <br>
Sample an alignment from the posterior distribution of
alignments. The posterior distribution is determined using
an HMM banded (unless --nonbanded) variant of the Inside
<br>
algorithm.</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Seed the random number generator with &lt;n&gt;, an integer
&gt;= 0. This option can only be used in combination with
--sample. If &lt;n&gt; is nonzero, stochastic sampling of
align&acirc; <br>
ments will be reproducible; the same command will give the
same results. If &lt;n&gt; is 0, the random number generator
is seeded arbitrarily, and stochastic samplings may vary
<br>
from run to run of the same command. The default seed is
181.</p>

<p style="margin-top: 1em">--notrunc <br>
Turn off truncated alignment algorithms. All sequences in
the input file will be assumed to be full length, unless
--sub is also used, in which case the program can still <br>
handle truncated sequences but will use an alternative
strategy for their alignment.</p>

<p style="margin-top: 1em">--sub Turn on the sub model
construction and alignment procedure. For each sequence, an
HMM is first used to predict the model start and end
consensus columns, and a new sub CM <br>
is constructed that only models consensus columns from start
to end. The sequence is then aligned to this sub CM. Sub
alignment is an older method than the default one <br>
for aligning sequences that are possibly truncated. By
default, cmalign uses special DP algorithms to handle
truncated sequences which should be more accurate than the
sub <br>
method in most cases. --sub is still included as an option
mainly for testing against this default truncated sequence
handling. This &quot;sub CM&quot; procedure is not the same
<br>
as the &quot;sub CMs&quot; described by Weinberg and
Ruzzo.</p>

<p style="margin-top: 1em">OPTIONS FOR CONTROLLING SPEED
AND MEMORY REQUIREMENTS <br>
--hbanded <br>
This option is turned on by default. Accelerate alignment by
pruning away regions of the CM DP matrix that are deemed
negligible by an HMM. First, each sequence is scored <br>
with a CM plan 9 HMM derived from the CM using the Forward
and Backward HMM algorithms to calculate posterior
probabilities that each nucleotide aligns to each state of
<br>
the HMM. These posterior probabilities are used to derive
constraints (bands) on the CM DP matrix. Finally, the target
sequence is aligned to the CM using the banded DP <br>
matrix, during which cells outside the bands are ignored.
Usually most of the full DP matrix lies outside the bands
(often more than 95%), making this technique faster <br>
because fewer DP calculations are required, and more memory
efficient because only cells within the bands need be
allocated.</p>

<p style="margin-top: 1em">Importantly, HMM banding
sacrifices the guarantee of determining the optimally
accurarte or optimal alignment, which will be missed if it
lies outside the bands. The tau <br>
paramater is the amount of probability mass considered
negligible during HMM band calculation; lower values of tau
yield greater speedups but also a greater chance of <br>
missing the optimal alignment. The default tau is 1E-7,
determined empirically as a good tradeoff between
sensitivity and speed, though this value can be changed with
the <br>
--tau &lt;x&gt; option. The level of acceleration increases
with both the length and primary sequence conservation level
of the family. For example, with the default tau of <br>
1E-7, tRNA models (low primary sequence conservation with
length of about 75 nucleotides) show about 10X acceleration,
and SSU bacterial rRNA models (high primary sequence <br>
conservation with length of about 1500 nucleotides) show
about 700X. HMM banding can be turned off with the
--nonbanded option.</p>

<p style="margin-top: 1em">--tau &lt;x&gt; <br>
Set the tail loss probability used during HMM band
calculation to &lt;x&gt;. This is the amount of probability
mass within the HMM posterior probabilities that is
considered <br>
negligible. The default value is 1E-7. In general, higher
values will result in greater acceleration, but increase the
chance of missing the optimal alignment due to the <br>
HMM bands.</p>

<p style="margin-top: 1em">--mxsize &lt;x&gt; <br>
Set the maximum allowable total DP matrix size to &lt;x&gt;
megabytes. By default this size is 1028 Mb. This should be
large enough for the vast majority of alignments, however
<br>
if it is not cmalign will attempt to iteratively tighten the
HMM bands it uses to constrain the alignment by raising the
tau parameter and recalculating the bands until <br>
the total matrix size needed falls below &lt;x&gt; megabytes
or the maximum allowable tau value (0.05 by default, but
changeable with --maxtau) is reached. At each iteration of
<br>
band tightening, tau is multiplied by a 2.0. The band
tightening strategy can be turned off with the --fixedtau
option. If the maximum tau is reached and the required <br>
matrix size still exceeds &lt;x&gt; or if HMM banding is not
being used and the required matrix size exceeds &lt;x&gt;
then cmalign will exit prematurely and report an error
message <br>
that the matrix exceeded its maximum allowable size. In this
case, the --mxsize can be used to raise the size limit or
the maximum tau can be raised with --maxtau. The <br>
limit will commonly be exceeded when the --nonbanded option
is used without the --small option, but can still occur when
--nonbanded is not used. Note that if cmalign is <br>
being run in &lt;n&gt; multiple threads on a multicore
machine then each thread may have an allocated matrix of up
to size &lt;x&gt; Mb at any given time.</p>

<p style="margin-top: 1em">--fixedtau <br>
Turn off the HMM band tightening strategy described in the
explanation of the --mxsize option above.</p>

<p style="margin-top: 1em">--maxtau &lt;x&gt; <br>
Set the maximum allowed value for tau during band
tightening, described in the explanation of --mxsize above,
to &lt;x&gt;. By default this value is 0.05.</p>

<p style="margin-top: 1em">--nonbanded <br>
Turns off HMM banding. The returned alignment is guaranteed
to be the globally optimally accurate one (by default) or
the globally optimally scoring one (if --cyk is <br>
enabled). The --small option is recommended in combination
with this option, because standard alignment without HMM
banding requires a lot of memory (see --small ).</p>

<p style="margin-top: 1em">--small <br>
Use the divide and conquer CYK alignment algorithm described
in SR Eddy, BMC Bioinformatics 3:18, 2002. The --nonbanded
option must be used in combination with this <br>
options. Also, it is recommended whenever --nonbanded is
used that --small is also used because standard CM alignment
without HMM banding requires a lot of memory, espe&acirc;
<br>
cially for large RNAs. --small allows CM alignment within
practical memory limits, reducing the memory required for
alignment LSU rRNA, the largest known RNAs, from 150 <br>
Gb to less than 300 Mb. This option can only be used in
combination with --nonbanded, --notrunc, and --cyk.</p>

<p style="margin-top: 1em">OPTIONAL OUTPUT FILES <br>
--sfile &lt;f&gt; <br>
Dump per-sequence alignment score and timig information to
file &lt;f&gt;. The format of this file is described above
(it&rsquo;s the same data in the same format as the tabular
std&acirc; <br>
out output when the -o option is used).</p>

<p style="margin-top: 1em">--tfile &lt;f&gt; <br>
Dump tabular sequence tracebacks for each individual
sequence to a file &lt;f&gt;. Primarily useful for
debugging.</p>

<p style="margin-top: 1em">--ifile &lt;f&gt; <br>
Dump per-sequence insert information to file &lt;f&gt;. The
format of the file is described by &quot;#&quot;-prefixed
comment lines included at the top of the file &lt;f&gt;. The
insert <br>
information is valid even when the --matchonly option is
used.</p>

<p style="margin-top: 1em">--elfile &lt;f&gt; <br>
Dump per-sequence EL state (local end) insert information to
file &lt;f&gt;. The format of the file is described by
&quot;#&quot;-prefixed comment lines included at the top of
the file <br>
&lt;f&gt;. The EL insert information is valid even when the
--matchonly option is used.</p>

<p style="margin-top: 1em">OTHER OPTIONS <br>
--mapali &lt;f&gt; <br>
Reads the alignment from file &lt;f&gt; used to build the
model aligns it as a single object to the CM; e.g. the
alignment in &lt;f&gt; is held fixed. This allows you to
align <br>
sequences to a model with cmalign and view them in the
context of an existing trusted multiple alignment. &lt;f&gt;
must be the alignment file that the CM was built from. The
<br>
program verifies that the checksum of the file matches that
of the file used to construct the CM. A similar option to
this one was called --withali in previous versions of <br>
cmalign.</p>

<p style="margin-top: 1em">--mapstr <br>
Must be used in combination with --mapali &lt;f&gt;.
Propogate structural information for any pseudoknots that
exist in &lt;f&gt; to the output alignment. A similar option
to this <br>
one was called --withstr in previous versions of
cmalign.</p>

<p style="margin-top: 1em">--informat &lt;s&gt; <br>
Assert that the input &lt;seqfile&gt; is in format
&lt;s&gt;. Do not run Babelfish format autodection. This
increases the reliability of the program somewhat, because
the Babelfish <br>
can make mistakes; particularly recommended for unattended,
high-throughput runs of Infernal. Acceptable formats are:
FASTA, GENBANK, and DDBJ. &lt;s&gt; is case-insensitive.</p>

<p style="margin-top: 1em">--outformat &lt;s&gt; <br>
Specify the output alignment format as &lt;s&gt;. Acceptable
formats are: Pfam, AFA, A2M, Clustal, and Phylip. AFA is
aligned fasta. Only Pfam and Stockholm alignment formats
<br>
will include consensus structure annotation and posterior
probability annotation of aligned residues.</p>

<p style="margin-top: 1em">--dnaout <br>
Output the alignments as DNA sequence alignments, instead of
RNA ones.</p>

<p style="margin-top: 1em">--noprob <br>
Do not annotate the output alignment with posterior
probabilities.</p>

<p style="margin-top: 1em">--matchonly <br>
Only include match columns in the output alignment, do not
include any insertions relative to the consensus model. This
option may be useful when creating very large <br>
alignments that require a lot of memory and disk space, most
of which is necessary only to deal with insert columns that
are gaps in most sequences.</p>

<p style="margin-top: 1em">--ileaved <br>
Output the alignment in interleaved Stockholm format of a
fixed width that may be more convenient for examination.
This was the default output alignment format of previous
<br>
versions of cmalign. Note that cmalign requires more memory
when this option is used. For this reason, --ileaved will
only work for alignments of up to 100,000 sequences <br>
or a total of 100,000,000 aligned nucleotides.</p>

<p style="margin-top: 1em">--regress &lt;s&gt; <br>
Save an additional copy of the output alignment with no
author information to file &lt;s&gt;.</p>

<p style="margin-top: 1em">--verbose <br>
Output additional information in the tabular scores output
(output to stdout if -o is used, or to &lt;f&gt; if --sfile
&lt;f&gt; is used). These are mainly useful for testing and
<br>
debugging.</p>

<p style="margin-top: 1em">--cpu &lt;n&gt; <br>
Specify that &lt;n&gt; parallel CPU workers be used. If
&lt;n&gt; is set as &quot;0&quot;, then the program will be
run in serial mode, without using threads. You can also
control this number <br>
by setting an environment variable, INFERNAL_NCPU. This
option will only be available if the machine on which
Infernal was built is capable of using POSIX threading (see
<br>
the Installation section of the user guide for more
information).</p>

<p style="margin-top: 1em">--mpi Run as an MPI parallel
program. This option will only be available if Infernal has
been configured and built with the &quot;--enable-mpi&quot;
flag (see the Installation section of <br>
the user guide for more information).</p>

<p style="margin-top: 1em">SEE ALSO <br>
See infernal(1) for a master man page with a list of all the
individual man pages for programs in the Infernal
package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your Infernal distribution
(Userguide.pdf); or see the Infernal web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2016 Howard Hughes Medical Institute. <br>
Freely distributed under a BSD open source license.</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your Infernal source distribution, or see the Infernal web
page ().</p>

<p style="margin-top: 1em">AUTHOR <br>
The Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">Infernal 1.1.2 July 2016
cmalign(1)</p>
<hr>
</body>
</html>
