<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>cmalign(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">cmalign(1)</td>
    <td class="head-vol">Infernal Manual</td>
    <td class="head-rtitle">cmalign(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cmalign - align sequences to a covariance model
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cmalign</b></dt>
  <dd class="It-tag"><i>[options]</i> <i>&lt;cmfile&gt;</i>
      <i>&lt;seqfile&gt;</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>cmalign</b> aligns the RNA sequences in <i>&lt;seqfile&gt;</i> to the
  covariance model (CM) in <i>&lt;cmfile&gt;.</i> The new alignment is output to
  <i>stdout</i> in Stockholm format, but can be redirected to a file
  <i>&lt;f&gt;</i> with the <b>-o</b><i> &lt;f&gt;</i> option.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Either <i>&lt;cmfile&gt;</i> or <i>&lt;seqfile&gt;</i> (but not both) may be '-'
  (dash), which means reading this input from <i>stdin</i> rather than a file.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The sequence file <i>&lt;seqfile&gt;</i> must be in FASTA or Genbank format.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<b>cmalign</b> uses an HMM banding technique to accelerate alignment by default
  as described below for the <b>--hbanded</b> option. HMM banding can be turned
  off with the <b>--nonbanded</b> option.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
By default, <b>cmalign</b> computes the alignment with maximum expected accuracy
  that is consistent with constraints (bands) derived from an HMM, using a
  banded version of the Durbin/Holmes optimal accuracy algorithm. This behavior
  can be changed with the <b>--cyk</b> or <b>--sample</b> options.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<b>cmalign</b> takes special care to correctly align truncated sequences, where
  some nucleotides from the beginning (5') and/or end (3') of the actual full
  length biological sequence are not present in the input sequence (see DL Kolbe
  and SR Eddy, Bioinformatics, 25:1236-1243, 2009). This behavior is on by
  default, but can be turned off with <b>--notrunc.</b> In previous versions of
  <b>cmalign</b> the <b>--sub</b> option was required to appropriately handle
  truncated sequences. The <b>--sub</b> option is still available in this
  version, but the new default method for handling truncated sequences should be
  as good or superior to the sub method in nearly all cases.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The <b>--mapali</b><i> &lt;s&gt;</i> option allows inclusion of the fixed
  training alignment used to build the CM from file <i>&lt;s&gt;</i> within the
  output alignment of <b>cmalign.</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
It is possible to merge two or more alignments created by the same CM using the
  Easel miniapp <b>esl-alimerge</b> (included in the easel/miniapps/
  subdirectory of Infernal). Previous versions of <b>cmalign</b> included
  options to merge alignments but they were deprecated upon development of
  <b>esl-alimerge,</b> which is significantly more memory efficient.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
By default, <b>cmalign</b> will output the alignment to stdout. The alignment
  can be redirected to an output file <i>&lt;f&gt;</i> with the <b>-o</b><i>
  &lt;f&gt;</i> option. With <b>-o,</b> information on each aligned sequence,
  including score and model alignment boundaries will be printed to stdout (more
  on this below).
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The output alignment will be in Stockholm format by default. This can be changed
  to Pfam, aligned FASTA (AFA), A2M, Clustal, or Phylip format using the
  <b>--outformat</b><i> &lt;s&gt;</i> option, where <i>&lt;s&gt;</i> is the name
  of the desired format. As a special case, if the output alignment is large
  (more than 10,000 sequences or more than 10,000,000 total nucleotides) than
  the output format will be Pfam format, with each sequence appearing on a
  single line, for reasons of memory efficiency. For alignments larger than
  this, using <b>--ileaved</b> will force interleaved Stockholm format, but the
  user should be aware that this may require a lot of memory. <b>--ileaved</b>
  will only work for alignments up to 100,000 sequences or 100,000,000 total
  nucleotides.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
If the output alignment format is Stockholm or Pfam, the output alignment will
  be annotated with posterior probabilities which estimate the confidence level
  of each aligned nucleotide. This annotation appears as lines beginning with
  &quot;#=GR &lt;seq name&gt; PP&quot;, one per sequence, each immediately below
  the corresponding aligned sequence &quot;&lt;seq name&gt;&quot;. Characters in
  PP lines have 12 possible values: &quot;0-9&quot;, &quot;*&quot;, or
  &quot;.&quot;. If &quot;.&quot;, the position corresponds to a gap in the
  sequence. A value of &quot;0&quot; indicates a posterior probability of
  between 0.0 and 0.05, &quot;1&quot; indicates between 0.05 and 0.15,
  &quot;2&quot; indicates between 0.15 and 0.25 and so on up to &quot;9&quot;
  which indicates between 0.85 and 0.95. A value of &quot;*&quot; indicates a
  posterior probability of between 0.95 and 1.0. Higher posterior probabilities
  correspond to greater confidence that the aligned nucleotide belongs where it
  appears in the alignment. With <b>--nonbanded,</b> the calculation of the
  posterior probabilities considers all possible alignments of the target
  sequence to the CM. Without <b>--nonbanded</b> (i.e. in default mode), the
  calculation considers only possible alignments within the HMM bands. Further,
  the posterior probabilities are conditional on the truncation mode of the
  alignment. For example, if the sequence alignment is truncated 5', a PP value
  of &quot;9&quot; indicates between 0.85 and 0.95 of all 5' truncated
  alignments include the given nucleotide at the given position. The posterior
  annotation can be turned off with the <b>--noprob</b> option. If
  <b>--small</b> is enabled, posterior annotation must also be turned off using
  <b>--noprob.</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The tabular output that is printed to stdout if the <b>-o</b> option is used
  includes one line per sequence and twelve fields per line: &quot;idx&quot;:
  the index of the sequence in the input file, &quot;seq name&quot;: the
  sequence name; &quot;length&quot;: the length of the sequence; &quot;cm
  from&quot; and &quot;cm to&quot;: the model start and end positions of the
  alignment; &quot;trunc&quot;: &quot;no&quot; if the sequence is not truncated,
  &quot;5'&quot; if the beginning of the sequence truncated 5', &quot;3'&quot;
  if the end of the sequence is truncated, and &quot;5'&amp;3'&quot; if both the
  beginning and the end are truncated; &quot;bit sc&quot;: the bit score of the
  alignment, &quot;avg pp&quot; the average posterior probability of all aligned
  nucleotides in the alignment; &quot;band calc&quot;, &quot;alignment&quot; and
  &quot;total&quot;: the time in seconds required for calculating HMM bands,
  computing the alignment, and complete processing of the sequence,
  respectively; &quot;mem (Mb)&quot;: the size in Mb of all dynamic programming
  matrices required for aligning the sequence. This tabular data can be saved to
  file <i>&lt;f&gt;</i> with the <b>--sfile</b><i> &lt;f&gt;</i> option.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save the alignment in Stockholm format to a file
      <i>&lt;f&gt;.</i> The default is to write it to standard output.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">Configure the model for global alignment of the query model
      to the target sequences. By default, the model is configured for local
      alignment. Local alignments can contain large insertions and deletions
      called &quot;local ends&quot; in the structure to be penalized differently
      than normal indels. These are annotated as &quot;~&quot; columns in the RF
      line of the output alignment. The <b>-g</b> option can be used to disallow
      these local ends. The <b>-g</b> option is required if the <b>--sub</b>
      option is also used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_CONTROLLING_THE_ALIGNMENT_ALGORITHM"><a class="selflink" href="#OPTIONS_FOR_CONTROLLING_THE_ALIGNMENT_ALGORITHM">OPTIONS
  FOR CONTROLLING THE ALIGNMENT ALGORITHM</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--optacc</b></dt>
  <dd class="It-tag">Align sequences using the Durbin/Holmes optimal accuracy
      algorithm. This is the default. The optimal accuracy alignment will be
      constrained by HMM bands for acceleration unless the <b>--nonbanded</b>
      option is enabled. The optimal accuracy algorithm determines the alignment
      that maximizes the posterior probabilities of the aligned nucleotides
      within it. The posterior probabilites are determined using (possibly HMM
      banded) variants of the Inside and Outside algorithms.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cyk</b></dt>
  <dd class="It-tag">Do not use the Durbin/Holmes optimal accuracy alignment to
      align the sequences, instead use the CYK algorithm which determines the
      optimally scoring (maximum likelihood) alignment of the sequence to the
      model, given the HMM bands (unless <b>--nonbanded</b> is also enabled).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sample</b></dt>
  <dd class="It-tag">Sample an alignment from the posterior distribution of
      alignments. The posterior distribution is determined using an HMM banded
      (unless <b>--nonbanded)</b> variant of the Inside algorithm.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Seed the random number generator with <i>&lt;n&gt;,</i> an
      integer &gt;= 0. This option can only be used in combination with
      <b>--sample.</b> If <i>&lt;n&gt;</i> is nonzero, stochastic sampling of
      alignments will be reproducible; the same command will give the same
      results. If <i>&lt;n&gt;</i> is 0, the random number generator is seeded
      arbitrarily, and stochastic samplings may vary from run to run of the same
      command. The default seed is 181.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--notrunc</b></dt>
  <dd class="It-tag">Turn off truncated alignment algorithms. All sequences in
      the input file will be assumed to be full length, unless <b>--sub</b> is
      also used, in which case the program can still handle truncated sequences
      but will use an alternative strategy for their alignment.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sub</b></dt>
  <dd class="It-tag">Turn on the sub model construction and alignment procedure.
      For each sequence, an HMM is first used to predict the model start and end
      consensus columns, and a new sub CM is constructed that only models
      consensus columns from start to end. The sequence is then aligned to this
      sub CM. Sub alignment is an older method than the default one for aligning
      sequences that are possibly truncated. By default, <b>cmalign</b> uses
      special DP algorithms to handle truncated sequences which should be more
      accurate than the sub method in most cases. <b>--sub</b> is still included
      as an option mainly for testing against this default truncated sequence
      handling. This &quot;sub CM&quot; procedure is not the same as the
      &quot;sub CMs&quot; described by Weinberg and Ruzzo.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_CONTROLLING_SPEED_AND_MEMORY_REQUIREMENTS"><a class="selflink" href="#OPTIONS_FOR_CONTROLLING_SPEED_AND_MEMORY_REQUIREMENTS">OPTIONS
  FOR CONTROLLING SPEED AND MEMORY REQUIREMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hbanded</b></dt>
  <dd class="It-tag">This option is turned on by default. Accelerate alignment
      by pruning away regions of the CM DP matrix that are deemed negligible by
      an HMM. First, each sequence is scored with a CM plan 9 HMM derived from
      the CM using the Forward and Backward HMM algorithms to calculate
      posterior probabilities that each nucleotide aligns to each state of the
      HMM. These posterior probabilities are used to derive constraints (bands)
      on the CM DP matrix. Finally, the target sequence is aligned to the CM
      using the banded DP matrix, during which cells outside the bands are
      ignored. Usually most of the full DP matrix lies outside the bands (often
      more than 95%), making this technique faster because fewer DP calculations
      are required, and more memory efficient because only cells within the
      bands need be allocated.
    <div style="height: 1.00em;">&#x00A0;</div>
    Importantly, HMM banding sacrifices the guarantee of determining the
      optimally accurarte or optimal alignment, which will be missed if it lies
      outside the bands. The tau paramater is the amount of probability mass
      considered negligible during HMM band calculation; lower values of tau
      yield greater speedups but also a greater chance of missing the optimal
      alignment. The default tau is 1E-7, determined empirically as a good
      tradeoff between sensitivity and speed, though this value can be changed
      with the <b>--tau &lt;x&gt;</b> option. The level of acceleration
      increases with both the length and primary sequence conservation level of
      the family. For example, with the default tau of 1E-7, tRNA models (low
      primary sequence conservation with length of about 75 nucleotides) show
      about 10X acceleration, and SSU bacterial rRNA models (high primary
      sequence conservation with length of about 1500 nucleotides) show about
      700X. HMM banding can be turned off with the <b>--nonbanded</b> option.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tau</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the tail loss probability used during HMM band
      calculation to <i>&lt;x&gt;.</i> This is the amount of probability mass
      within the HMM posterior probabilities that is considered negligible. The
      default value is 1E-7. In general, higher values will result in greater
      acceleration, but increase the chance of missing the optimal alignment due
      to the HMM bands.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mxsize</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the maximum allowable total DP matrix size to
      <i>&lt;x&gt;</i> megabytes. By default this size is 1028 Mb. This should
      be large enough for the vast majority of alignments, however if it is not
      <b>cmalign</b> will attempt to iteratively tighten the HMM bands it uses
      to constrain the alignment by raising the tau parameter and recalculating
      the bands until the total matrix size needed falls below <i>&lt;x&gt;</i>
      megabytes or the maximum allowable tau value (0.05 by default, but
      changeable with <b>--maxtau)</b> is reached. At each iteration of band
      tightening, tau is multiplied by a 2.0. The band tightening strategy can
      be turned off with the <b>--fixedtau</b> option. If the maximum tau is
      reached and the required matrix size still exceeds <i>&lt;x&gt;</i> or if
      HMM banding is not being used and the required matrix size exceeds
      <i>&lt;x&gt;</i> then <b>cmalign</b> will exit prematurely and report an
      error message that the matrix exceeded its maximum allowable size. In this
      case, the <b>--mxsize</b> can be used to raise the size limit or the
      maximum tau can be raised with <b>--maxtau.</b> The limit will commonly be
      exceeded when the <b>--nonbanded</b> option is used without the
      <b>--small</b> option, but can still occur when <b>--nonbanded</b> is not
      used. Note that if <b>cmalign</b> is being run in <i>&lt;n&gt;</i>
      multiple threads on a multicore machine then each thread may have an
      allocated matrix of up to size <i>&lt;x&gt;</i> Mb at any given time.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fixedtau</b></dt>
  <dd class="It-tag">Turn off the HMM band tightening strategy described in the
      explanation of the <b>--mxsize</b> option above.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxtau</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the maximum allowed value for tau during band
      tightening, described in the explanation of <b>--mxsize</b> above, to
      <i>&lt;x&gt;.</i> By default this value is 0.05.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nonbanded</b></dt>
  <dd class="It-tag">Turns off HMM banding. The returned alignment is guaranteed
      to be the globally optimally accurate one (by default) or the globally
      optimally scoring one (if <b>--cyk</b> is enabled). The <b>--small</b>
      option is recommended in combination with this option, because standard
      alignment without HMM banding requires a lot of memory (see <b>--small</b>
      ).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--small</b></dt>
  <dd class="It-tag">Use the divide and conquer CYK alignment algorithm
      described in SR Eddy, BMC Bioinformatics 3:18, 2002. The
      <b>--nonbanded</b> option must be used in combination with this options.
      Also, it is recommended whenever <b>--nonbanded</b> is used that
      <b>--small</b> is also used because standard CM alignment without HMM
      banding requires a lot of memory, especially for large RNAs.
      <b>--small</b> allows CM alignment within practical memory limits,
      reducing the memory required for alignment LSU rRNA, the largest known
      RNAs, from 150 Gb to less than 300 Mb. This option can only be used in
      combination with <b>--nonbanded,</b> <b>--notrunc,</b> and <b>--cyk.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_OUTPUT_FILES"><a class="selflink" href="#OPTIONAL_OUTPUT_FILES">OPTIONAL
  OUTPUT FILES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sfile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Dump per-sequence alignment score and timig information to
      file <i>&lt;f&gt;.</i> The format of this file is described above (it's
      the same data in the same format as the tabular stdout output when the
      <b>-o</b> option is used).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tfile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Dump tabular sequence tracebacks for each individual
      sequence to a file <i>&lt;f&gt;.</i> Primarily useful for debugging.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ifile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Dump per-sequence insert information to file
      <i>&lt;f&gt;.</i> The format of the file is described by
      &quot;#&quot;-prefixed comment lines included at the top of the file
      <i>&lt;f&gt;.</i> The insert information is valid even when the
      <b>--matchonly</b> option is used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--elfile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Dump per-sequence EL state (local end) insert information
      to file <i>&lt;f&gt;.</i> The format of the file is described by
      &quot;#&quot;-prefixed comment lines included at the top of the file
      <i>&lt;f&gt;.</i> The EL insert information is valid even when the
      <b>--matchonly</b> option is used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_OPTIONS"><a class="selflink" href="#OTHER_OPTIONS">OTHER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mapali</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Reads the alignment from file <i>&lt;f&gt;</i> used to
      build the model aligns it as a single object to the CM; e.g. the alignment
      in <i>&lt;f&gt;</i> is held fixed. This allows you to align sequences to a
      model with <b>cmalign</b> and view them in the context of an existing
      trusted multiple alignment. <i>&lt;f&gt;</i> must be the alignment file
      that the CM was built from. The program verifies that the checksum of the
      file matches that of the file used to construct the CM. A similar option
      to this one was called <b>--withali</b> in previous versions of
      <b>cmalign.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mapstr</b></dt>
  <dd class="It-tag">Must be used in combination with <b>--mapali</b><i>
      &lt;f&gt;</i><b>.</b> Propogate structural information for any pseudoknots
      that exist in <i>&lt;f&gt;</i> to the output alignment. A similar option
      to this one was called <b>--withstr</b> in previous versions of
      <b>cmalign.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Assert that the input <i>&lt;seqfile&gt;</i> is in format
      <i>&lt;s&gt;.</i> Do not run Babelfish format autodection. This increases
      the reliability of the program somewhat, because the Babelfish can make
      mistakes; particularly recommended for unattended, high-throughput runs of
      Infernal. Acceptable formats are: FASTA, GENBANK, and DDBJ.
      <i>&lt;s&gt;</i> is case-insensitive.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Specify the output alignment format as <i>&lt;s&gt;.</i>
      Acceptable formats are: Pfam, AFA, A2M, Clustal, and Phylip. AFA is
      aligned fasta. Only Pfam and Stockholm alignment formats will include
      consensus structure annotation and posterior probability annotation of
      aligned residues.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dnaout</b></dt>
  <dd class="It-tag">Output the alignments as DNA sequence alignments, instead
      of RNA ones.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noprob</b></dt>
  <dd class="It-tag">Do not annotate the output alignment with posterior
      probabilities.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--matchonly</b></dt>
  <dd class="It-tag">Only include match columns in the output alignment, do not
      include any insertions relative to the consensus model. This option may be
      useful when creating very large alignments that require a lot of memory
      and disk space, most of which is necessary only to deal with insert
      columns that are gaps in most sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ileaved</b></dt>
  <dd class="It-tag">Output the alignment in interleaved Stockholm format of a
      fixed width that may be more convenient for examination. This was the
      default output alignment format of previous versions of <b>cmalign.</b>
      Note that <b>cmalign</b> requires more memory when this option is used.
      For this reason, <b>--ileaved</b> will only work for alignments of up to
      100,000 sequences or a total of 100,000,000 aligned nucleotides.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--regress</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Save an additional copy of the output alignment with no
      author information to file <i>&lt;s&gt;.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--verbose</b></dt>
  <dd class="It-tag">Output additional information in the tabular scores output
      (output to stdout if <b>-o</b> is used, or to <i>&lt;f&gt;</i> if
      <b>--sfile</b><i> &lt;f&gt;</i> is used). These are mainly useful for
      testing and debugging.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cpu</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Specify that <i>&lt;n&gt;</i> parallel CPU workers be used.
      If <i>&lt;n&gt;</i> is set as &quot;0&quot;, then the program will be run
      in serial mode, without using threads. You can also control this number by
      setting an environment variable, <i>INFERNAL_NCPU.</i> This option will
      only be available if the machine on which Infernal was built is capable of
      using POSIX threading (see the Installation section of the user guide for
      more information).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mpi</b></dt>
  <dd class="It-tag">Run as an MPI parallel program. This option will only be
      available if Infernal has been configured and built with the
      &quot;--enable-mpi&quot; flag (see the Installation section of the user
      guide for more information).
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>infernal(1)</b> for a master man page with a list of all the individual
  man pages for programs in the Infernal package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your Infernal
  distribution (Userguide.pdf); or see the Infernal web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2016 Howard Hughes Medical Institute.
Freely distributed under a BSD open source license.
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your Infernal source distribution, or see the Infernal web page
  ().
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2016</td>
    <td class="foot-os">Infernal 1.1.2</td>
  </tr>
</table>
</body>
</html>
