<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>cmbuild(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">cmbuild(1)</td>
    <td class="head-vol">Infernal Manual</td>
    <td class="head-rtitle">cmbuild(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cmbuild - construct covariance model(s) from structurally annotated RNA multiple
  sequence alignment(s)
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>cmbuild</b> <i>[options]</i> <i>&lt;cmfile_out&gt;</i> <i>&lt;msafile&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
For each multiple sequence alignment in <i>&lt;msafile&gt;</i> build a
  covariance model and save it to a new file <i>&lt;cmfile_out&gt;.</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The alignment file must be in Stockholm or SELEX format, and must contain
  consensus secondary structure annotation. <b>cmbuild</b> uses the consensus
  structure to determine the architecture of the CM.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<i>&lt;msafile&gt;</i> may be '-' (dash), which means reading this input from
  <i>stdin</i> rather than a file. To use '-', you must also specify the
  alignment file format with <b>--informat</b><i> &lt;s&gt;</i><b>,</b> as in
  <b>--informat stockholm</b> (because of a current limitation in our
  implementation, MSA file formats cannot be autodetected in a nonrewindable
  input stream.)
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<i>&lt;cmfile_out&gt;</i> may not be '-' <i>(stdout),</i> because sending the CM
  file to <i>stdout</i> would conflict with the other text output of the
  program.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
In addition to writing CM(s) to <i>&lt;cmfile_out&gt;,</i> <b>cmbuild</b> also
  outputs a single line for each model created to stdout. Each line has the
  following fields: &quot;aln&quot;: the index of the alignment used to build
  the CM; &quot;idx&quot;: the index of the CM in the <i>&lt;cmfile_out&gt;;</i>
  &quot;name&quot;: the name of the CM; &quot;nseq&quot;: the number of
  sequences in the alignment used to build the CM; &quot;eff_nseq&quot;: the
  effective number of sequences used to build the model; &quot;alen&quot;: the
  length of the alignment used to build the CM; &quot;clen&quot;: the number of
  columns from the alignment defined as consensus (match) columns;
  &quot;bps&quot;: the number of basepairs in the CM; &quot;bifs&quot;: the
  number of bifurcations in the CM; &quot;rel entropy: CM&quot;: the total
  relative entropy of the model divided by the number of consensus columns;
  &quot;rel entropy: HMM&quot;: the total relative entropy of the model ignoring
  secondary structure divided by the number of consensus columns.
  &quot;description&quot;: description of the model/alignment.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Name the new CM <i>&lt;s&gt;.</i> The default is to use the
      name of the alignment (if one is present in the <i>&lt;msafile&gt;),</i>
      or, failing that, the name of the <i>&lt;msafile&gt;.</i> If
      <i>&lt;msafile&gt;</i> contains more than one alignment, <i>-n</i> doesn't
      work, and every alignment must have a name annotated in the
      <i>&lt;msafile&gt;</i> (as in Stockholm #=GF ID annotation).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b></dt>
  <dd class="It-tag">Allow <i>&lt;cmfile_out&gt;</i> to be overwritten. Without
      this option, if <i>&lt;cmfile_out&gt;</i> already exists, <b>cmbuild</b>
      exits with an error.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Direct the summary output to file <i>&lt;f&gt;,</i> rather
      than to <i>stdout.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-O</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">After each model is constructed, resave annotated source
      alignments to a file <i>&lt;f&gt;</i> in Stockholm format. Sequences are
      annoted with what relative sequence weights were assigned. The alignments
      are also annotated with a reference annotation line indicating which
      columns were assigned as consensus. If the source alignment had reference
      annotation (&quot;#=GC RF&quot;) it will be replaced with the consensus
      residue of the model for consensus columns and '.' for insert columns,
      unless the <b>--hand</b> option was used for specifying consensus
      positions, in which case it will be unchanged.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>--devhelp</b> Print help, as with <b>-h ,</b> but also include expert
      options that are not displayed with <b>-h .</b> These expert options are
      not expected to be relevant for the vast majority of users and so are not
      described in the manual page. The only resources for understanding what
      they actually do are the brief one-line descriptions output when
      <b>--devhelp</b> is enabled, and the source code.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_MODEL_CONSTRUCTION"><a class="selflink" href="#OPTIONS_CONTROLLING_MODEL_CONSTRUCTION">OPTIONS
  CONTROLLING MODEL CONSTRUCTION</a></h1>
These options control how consensus columns are defined in an alignment.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fast</b></dt>
  <dd class="It-tag">Define consensus columns automatically as those that have a
      fraction &gt;= <b>symfrac</b> of residues as opposed to gaps. (See below
      for the <b>--symfrac</b> option.) This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hand</b></dt>
  <dd class="It-tag">Use reference coordinate annotation (#=GC RF line, in
      Stockholm) to determine which columns are consensus, and which are
      inserts. Any non-gap character indicates a consensus column. (For example,
      mark consensus columns with &quot;x&quot;, and insert columns with
      &quot;.&quot;.) This option was called <b>--rf</b> in previous versions of
      Infernal (0.1 through 1.0.2).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--symfrac</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Define the residue fraction threshold necessary to define a
      consensus column when not using <b>--hand.</b> The default is 0.5. The
      symbol fraction in each column is calculated after taking relative
      sequence weighting into account. Setting this to 0.0 means that every
      alignment column will be assigned as consensus, which may be useful in
      some cases. Setting it to 1.0 means that only columns that include 0 gaps
      will be assigned as consensus. This option replaces the
      <b>--gapthresh</b><i> &lt;y&gt;</i> option from previous versions of
      Infernal (0.1 through 1.0.2), with <i>&lt;x&gt;</i> equal to (1.0 -
      <i>&lt;y&gt;).</i> For example to reproduce behavior for a command of
      <b>cmbuild --gapthresh 0.8</b> in a previous version, use <b>cmbuild
      --symfrac 0.2</b> with this version.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noss</b></dt>
  <dd class="It-tag">Ignore the secondary structure annotation, if any, in
      <i>&lt;msafile&gt;</i> and build a CM with zero basepairs. This model will
      be similar to a profile HMM and the <b>cmsearch</b> and <b>cmscan</b>
      programs will use HMM algorithms which are faster than CM ones for this
      model. Additionally, a zero basepair model need not be calibrated with
      <b>cmcalibrate</b> prior to running <b>cmsearch</b> with it. The
      <b>--noss</b> option must be used if there is no secondary structure
      annotation in <b>&lt;msafile&gt;.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rsearch</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Parameterize emission scores a la RSEARCH, using the
      RIBOSUM matrix in file <i>&lt;f&gt;.</i> With <b>--rsearch</b> enabled,
      all alignments in <i>&lt;msafile&gt;</i> must contain exactly one sequence
      or the <b>--call</b> option must also be enabled. All positions in each
      sequence will be considered consensus &quot;columns&quot;. Actually, the
      emission scores for these models will not be identical to RIBOSUM scores
      due of differences in the modelling strategy between Infernal and RSEARCH,
      but they will be as similar as possible. RIBOSUM matrix files are included
      with Infernal in the &quot;matrices/&quot; subdirectory of the top-level
      &quot;infernal-xxx&quot; directory. RIBOSUM matrices are substitution
      score matrices trained specifically for structural RNAs with separate
      single stranded residue and base pair substitution scores. For more
      information see the RSEARCH publication (Klein and Eddy, BMC
      Bioinformatics 4:44, 2003).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_MODEL_CONSTRUCTION_OPTIONS"><a class="selflink" href="#OTHER_MODEL_CONSTRUCTION_OPTIONS">OTHER
  MODEL CONSTRUCTION OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--null</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Read a null model from <i>&lt;f&gt;.</i> The null model
      defines the probability of each RNA nucleotide in background sequence, the
      default is to use 0.25 for each nucleotide. The format of null files is
      specified in the user guide.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--prior</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Read a Dirichlet prior from <i>&lt;f&gt;,</i> replacing the
      default mixture Dirichlet. The format of prior files is specified in the
      user guide.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
Use <b>--devhelp</b> to see additional, otherwise undocumented, model
  construction options.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_RELATIVE_WEIGHTS"><a class="selflink" href="#OPTIONS_CONTROLLING_RELATIVE_WEIGHTS">OPTIONS
  CONTROLLING RELATIVE WEIGHTS</a></h1>
<b>cmbuild</b> uses an ad hoc sequence weighting algorithm to downweight closely
  related sequences and upweight distantly related ones. This has the effect of
  making models less biased by uneven phylogenetic representation. For example,
  two identical sequences would typically each receive half the weight that one
  sequence would. These options control which algorithm gets used.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wpb</b></dt>
  <dd class="It-tag">Use the Henikoff position-based sequence weighting scheme
      [Henikoff and Henikoff, J. Mol. Biol. 243:574, 1994]. This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wgsc</b></dt>
  <dd class="It-tag">Use the Gerstein/Sonnhammer/Chothia weighting algorithm
      [Gerstein et al, J. Mol. Biol. 235:1067, 1994].
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wnone</b></dt>
  <dd class="It-tag">Turn sequence weighting off; e.g. explicitly set all
      sequence weights to 1.0.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wgiven</b></dt>
  <dd class="It-tag">Use sequence weights as given in annotation in the input
      alignment file. If no weights were given, assume they are all 1.0. The
      default is to determine new sequence weights by the
      Gerstein/Sonnhammer/Chothia algorithm, ignoring any annotated weights.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wblosum</b></dt>
  <dd class="It-tag">Use the BLOSUM filtering algorithm to weight the sequences,
      instead of the default GSC weighting. Cluster the sequences at a given
      percentage identity (see <b>--wid);</b> assign each cluster a total weight
      of 1.0, distributed equally amongst the members of that cluster.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wid</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Controls the behavior of the <i>--wblosum</i> weighting
      option by setting the percent identity for clustering the alignment to
      <i>&lt;x&gt;.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_EFFECTIVE_SEQUENCE_NUMBER"><a class="selflink" href="#OPTIONS_CONTROLLING_EFFECTIVE_SEQUENCE_NUMBER">OPTIONS
  CONTROLLING EFFECTIVE SEQUENCE NUMBER</a></h1>
After relative weights are determined, they are normalized to sum to a total
  effective sequence number, <i>eff_nseq.</i> This number may be the actual
  number of sequences in the alignment, but it is almost always smaller than
  that. The default entropy weighting method <i>(--eent)</i> reduces the
  effective sequence number to reduce the information content (relative entropy,
  or average expected score on true homologs) per consensus position. The target
  relative entropy is controlled by a two-parameter function, where the two
  parameters are settable with <i>--ere</i> and <i>--esigma.</i>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eent</b></dt>
  <dd class="It-tag">Use the entropy weighting strategy to determine the
      effective sequence number that gives a target mean match state relative
      entropy. This option is the default, and can be turned off with
      <b>--enone.</b> The default target mean match state relative entropy is
      0.59 bits for models with at least 1 basepair and 0.38 bits for models
      with zero basepairs, but changed with <b>--ere.</b> The default of 0.59 or
      0.38 bits is automatically changed if the total relative entropy of the
      model (summed match state relative entropy) is less than a cutoff, which
      is is 6.0 bits by default, but can be changed with the expert,
      undocumented <b>--eX</b> option. If you really want to play with that
      option, consult the source code.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--enone</b></dt>
  <dd class="It-tag">Turn off the entropy weighting strategy. The effective
      sequence number is just the number of sequences in the alignment.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ere</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the target mean match state relative entropy as
      <i>&lt;x&gt;.</i> By default the target relative entropy per match
      position is 0.59 bits for models with at least 1 basepair and 0.38 for
      models with zero basepairs.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eminseq</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Define the minimum allowed effective sequence number as
      <i>&lt;x&gt;.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ehmmre</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the target HMM mean match state relative entropy as
      <i>&lt;x&gt;.</i> Entropy for basepairing match states is calculated using
      marginalized basepair emission probabilities.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eset</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the effective sequence number for entropy weighting as
      <i>&lt;x&gt;.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_FILTER_P7_HMM_CONSTRUCTION"><a class="selflink" href="#OPTIONS_CONTROLLING_FILTER_P7_HMM_CONSTRUCTION">OPTIONS
  CONTROLLING FILTER P7 HMM CONSTRUCTION</a></h1>
For each CM that <b>cmbuild</b> constructs, an accompanying filter p7 HMM is
  built from the input alignment as well. These options control filter HMM
  construction:
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--p7ere</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the target mean match state relative entropy for the
      filter p7 HMM as <i>&lt;x&gt;.</i> By default the target relative entropy
      per match position is 0.38 bits.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--p7ml</b></dt>
  <dd class="It-tag">Use a maximum likelihood p7 HMM built from the CM as the
      filter HMM. This HMM will be as similar as possible to the CM (while
      necessarily ignorant of secondary structure).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
Use <b>--devhelp</b> to see additional, otherwise undocumented, filter HMM
  construction options.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_FILTER_P7_HMM_CALIBRATION"><a class="selflink" href="#OPTIONS_CONTROLLING_FILTER_P7_HMM_CALIBRATION">OPTIONS
  CONTROLLING FILTER P7 HMM CALIBRATION</a></h1>
After building each filter HMM, <b>cmbuild</b> determines appropriate E-value
  parameters to use during filtering in <b>cmsearch</b> and <b>cmscan</b> by
  sampling a set of sequences and searching them with each HMM filter
  configuration and algorithm.
<div style="height: 1.00em;">&#x00A0;</div>
<b>--EmN</b><i> &lt;n&gt;</i> Set the number of sampled sequences for local MSV
  filter HMM calibration to <i>&lt;n&gt;.</i> 200 by default.
<div style="height: 1.00em;">&#x00A0;</div>
<b>--EvN</b><i> &lt;n&gt;</i> Set the number of sampled sequences for local
  Viterbi filter HMM calibration to <i>&lt;n&gt;.</i> 200 by default.
<div style="height: 1.00em;">&#x00A0;</div>
<b>--ElfN</b><i> &lt;n&gt;</i> Set the number of sampled sequences for local
  Forward filter HMM calibration to <i>&lt;n&gt;.</i> 200 by default.
<div style="height: 1.00em;">&#x00A0;</div>
<b>--EgfN</b><i> &lt;n&gt;</i> Set the number of sampled sequences for glocal
  Forward filter HMM calibration to <i>&lt;n&gt;.</i> 200 by default.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Use <b>--devhelp</b> to see additional, otherwise undocumented, filter HMM
  calibration options.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_REFINING_THE_INPUT_ALIGNMENT"><a class="selflink" href="#OPTIONS_FOR_REFINING_THE_INPUT_ALIGNMENT">OPTIONS
  FOR REFINING THE INPUT ALIGNMENT</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--refine</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Attempt to refine the alignment before building the CM
      using expectation-maximization (EM). A CM is first built from the initial
      alignment as usual. Then, the sequences in the alignment are realigned
      optimally (with the HMM banded CYK algorithm, optimal means optimal given
      the bands) to the CM, and a new CM is built from the resulting alignment.
      The sequences are then realigned to the new CM, and a new CM is built from
      that alignment. This is continued until convergence, specifically when the
      alignments for two successive iterations are not significantly different
      (the summed bit scores of all the sequences in the alignment changes less
      than 1% between two successive iterations). The final alignment (the
      alignment used to build the CM that gets written to
      <i>&lt;cmfile_out&gt;)</i> is written to <i>&lt;f&gt;.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b></dt>
  <dd class="It-tag">With <b>--refine,</b> turn on the local alignment
      algorithm, which allows the alignment to span two or more subsequences if
      necessary (e.g. if the structures of the query model and target sequence
      are only partially shared), allowing certain large insertions and
      deletions in the structure to be penalized differently than normal indels.
      The default is to globally align the query model to the target sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gibbs</b></dt>
  <dd class="It-tag">Modifies the behavior of <b>--refine</b> so Gibbs sampling
      is used instead of EM. The difference is that during the alignment stage
      the alignment is not necessarily optimal, instead an alignment (parsetree)
      for each sequences is sampled from the posterior distribution of
      alignments as determined by the Inside algorithm. Due to this sampling
      step <b>--gibbs</b> is non-deterministic, so different runs with the same
      alignment may yield different results. This is not true when
      <b>--refine</b> is used without the <b>--gibbs</b> option, in which case
      the final alignment and CM will always be the same. When <b>--gibbs</b> is
      enabled, the <b>--seed &lt;n&gt;</b> option can be used to seed the random
      number generator predictably, making the results reproducible. The goal of
      the <b>--gibbs</b> option is to help expert RNA alignment curators refine
      structural alignments by allowing them to observe alternative high scoring
      alignments.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Seed the random number generator with <i>&lt;n&gt;,</i> an
      integer &gt;= 0. This option can only be used in combination with
      <b>--gibbs.</b> If <i>&lt;n&gt;</i> is nonzero, stochastic sampling of
      alignments will be reproducible; the same command will give the same
      results. If <i>&lt;n&gt;</i> is 0, the random number generator is seeded
      arbitrarily, and stochastic samplings may vary from run to run of the same
      command. The default seed is 0.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cyk</b></dt>
  <dd class="It-tag">With <b>--refine,</b> align with the CYK algorithm. By
      default the optimal accuracy algorithm is used. There is more information
      on this in the <b>cmalign</b> manual page.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--notrunc</b></dt>
  <dd class="It-tag">With <b>--refine,</b> turn off the the truncated alignment
      algorithm. There is more information on this in the <b>cmalign</b> manual
      page.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
Use <b>--devhelp</b> to see additional, otherwise undocumented, alignment
  refinement options as well as other output file options and options for
  building multiple models for a single alignment.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>infernal(1)</b> for a master man page with a list of all the individual
  man pages for programs in the Infernal package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your Infernal
  distribution (Userguide.pdf); or see the Infernal web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2016 Howard Hughes Medical Institute.
Freely distributed under a BSD open source license.
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your Infernal source distribution, or see the Infernal web page
  ().
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2016</td>
    <td class="foot-os">Infernal 1.1.2</td>
  </tr>
</table>
</body>
</html>
