<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:59:21 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>cmbuild(1) Infernal Manual cmbuild(1)</p>

<p style="margin-top: 1em">NAME <br>
cmbuild - construct covariance model(s) from structurally
annotated RNA multiple sequence alignment(s)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
cmbuild [options] &lt;cmfile_out&gt; &lt;msafile&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
For each multiple sequence alignment in &lt;msafile&gt;
build a covariance model and save it to a new file
&lt;cmfile_out&gt;.</p>

<p style="margin-top: 1em">The alignment file must be in
Stockholm or SELEX format, and must contain consensus
secondary structure annotation. cmbuild uses the consensus
structure to determine the archi&acirc; <br>
tecture of the CM.</p>

<p style="margin-top: 1em">&lt;msafile&gt; may be
&rsquo;-&rsquo; (dash), which means reading this input from
stdin rather than a file. To use &rsquo;-&rsquo;, you must
also specify the alignment file format with --informat
&lt;s&gt;, as in <br>
--informat stockholm (because of a current limitation in our
implementation, MSA file formats cannot be autodetected in a
nonrewindable input stream.)</p>

<p style="margin-top: 1em">&lt;cmfile_out&gt; may not be
&rsquo;-&rsquo; (stdout), because sending the CM file to
stdout would conflict with the other text output of the
program.</p>

<p style="margin-top: 1em">In addition to writing CM(s) to
&lt;cmfile_out&gt;, cmbuild also outputs a single line for
each model created to stdout. Each line has the following
fields: &quot;aln&quot;: the index of the <br>
alignment used to build the CM; &quot;idx&quot;: the index
of the CM in the &lt;cmfile_out&gt;; &quot;name&quot;: the
name of the CM; &quot;nseq&quot;: the number of sequences in
the alignment used to build the CM; <br>
&quot;eff_nseq&quot;: the effective number of sequences used
to build the model; &quot;alen&quot;: the length of the
alignment used to build the CM; &quot;clen&quot;: the number
of columns from the alignment <br>
defined as consensus (match) columns; &quot;bps&quot;: the
number of basepairs in the CM; &quot;bifs&quot;: the number
of bifurcations in the CM; &quot;rel entropy: CM&quot;: the
total relative entropy of the <br>
model divided by the number of consensus columns; &quot;rel
entropy: HMM&quot;: the total relative entropy of the model
ignoring secondary structure divided by the number of
consensus col&acirc; <br>
umns. &quot;description&quot;: description of the
model/alignment.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
available options.</p>

<p style="margin-top: 1em">-n &lt;s&gt; Name the new CM
&lt;s&gt;. The default is to use the name of the alignment
(if one is present in the &lt;msafile&gt;), or, failing
that, the name of the &lt;msafile&gt;. If &lt;msafile&gt;
con&acirc; <br>
tains more than one alignment, -n doesn&rsquo;t work, and
every alignment must have a name annotated in the
&lt;msafile&gt; (as in Stockholm #=GF ID annotation).</p>

<p style="margin-top: 1em">-F Allow &lt;cmfile_out&gt; to
be overwritten. Without this option, if &lt;cmfile_out&gt;
already exists, cmbuild exits with an error.</p>

<p style="margin-top: 1em">-o &lt;f&gt; Direct the summary
output to file &lt;f&gt;, rather than to stdout.</p>

<p style="margin-top: 1em">-O &lt;f&gt; After each model is
constructed, resave annotated source alignments to a file
&lt;f&gt; in Stockholm format. Sequences are annoted with
what relative sequence weights were <br>
assigned. The alignments are also annotated with a reference
annotation line indicating which columns were assigned as
consensus. If the source alignment had reference <br>
annotation (&quot;#=GC RF&quot;) it will be replaced with
the consensus residue of the model for consensus columns and
&rsquo;.&rsquo; for insert columns, unless the --hand option
was used for <br>
specifying consensus positions, in which case it will be
unchanged.</p>

<p style="margin-top: 1em">--devhelp Print help, as with -h
, but also include expert options that are not displayed
with -h . These expert options are not expected to be
relevant for the vast <br>
majority of users and so are not described in the manual
page. The only resources for understanding what they
actually do are the brief one-line descriptions output when
<br>
--devhelp is enabled, and the source code.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING MODEL
CONSTRUCTION <br>
These options control how consensus columns are defined in
an alignment.</p>

<p style="margin-top: 1em">--fast Define consensus columns
automatically as those that have a fraction &gt;= symfrac of
residues as opposed to gaps. (See below for the --symfrac
option.) This is the default.</p>

<p style="margin-top: 1em">--hand Use reference coordinate
annotation (#=GC RF line, in Stockholm) to determine which
columns are consensus, and which are inserts. Any non-gap
character indicates a con&acirc; <br>
sensus column. (For example, mark consensus columns with
&quot;x&quot;, and insert columns with &quot;.&quot;.) This
option was called --rf in previous versions of Infernal (0.1
through <br>
1.0.2).</p>

<p style="margin-top: 1em">--symfrac &lt;x&gt; <br>
Define the residue fraction threshold necessary to define a
consensus column when not using --hand. The default is 0.5.
The symbol fraction in each column is calculated <br>
after taking relative sequence weighting into account.
Setting this to 0.0 means that every alignment column will
be assigned as consensus, which may be useful in some <br>
cases. Setting it to 1.0 means that only columns that
include 0 gaps will be assigned as consensus. This option
replaces the --gapthresh &lt;y&gt; option from previous
versions <br>
of Infernal (0.1 through 1.0.2), with &lt;x&gt; equal to
(1.0 - &lt;y&gt;). For example to reproduce behavior for a
command of cmbuild --gapthresh 0.8 in a previous version,
use <br>
cmbuild --symfrac 0.2 with this version.</p>

<p style="margin-top: 1em">--noss Ignore the secondary
structure annotation, if any, in &lt;msafile&gt; and build a
CM with zero basepairs. This model will be similar to a
profile HMM and the cmsearch and cmscan <br>
programs will use HMM algorithms which are faster than CM
ones for this model. Additionally, a zero basepair model
need not be calibrated with cmcalibrate prior to running
<br>
cmsearch with it. The --noss option must be used if there is
no secondary structure annotation in &lt;msafile&gt;.</p>

<p style="margin-top: 1em">--rsearch &lt;f&gt; <br>
Parameterize emission scores a la RSEARCH, using the RIBOSUM
matrix in file &lt;f&gt;. With --rsearch enabled, all
alignments in &lt;msafile&gt; must contain exactly one
sequence or <br>
the --call option must also be enabled. All positions in
each sequence will be considered consensus
&quot;columns&quot;. Actually, the emission scores for these
models will not be <br>
identical to RIBOSUM scores due of differences in the
modelling strategy between Infernal and RSEARCH, but they
will be as similar as possible. RIBOSUM matrix files are
<br>
included with Infernal in the &quot;matrices/&quot;
subdirectory of the top-level &quot;infernal-xxx&quot;
directory. RIBOSUM matrices are substitution score matrices
trained specifically <br>
for structural RNAs with separate single stranded residue
and base pair substitution scores. For more information see
the RSEARCH publication (Klein and Eddy, BMC Bioin&acirc;
<br>
formatics 4:44, 2003).</p>

<p style="margin-top: 1em">OTHER MODEL CONSTRUCTION OPTIONS
<br>
--null &lt;f&gt; <br>
Read a null model from &lt;f&gt;. The null model defines the
probability of each RNA nucleotide in background sequence,
the default is to use 0.25 for each nucleotide. The <br>
format of null files is specified in the user guide.</p>

<p style="margin-top: 1em">--prior &lt;f&gt; <br>
Read a Dirichlet prior from &lt;f&gt;, replacing the default
mixture Dirichlet. The format of prior files is specified in
the user guide.</p>

<p style="margin-top: 1em">Use --devhelp to see additional,
otherwise undocumented, model construction options.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING RELATIVE
WEIGHTS <br>
cmbuild uses an ad hoc sequence weighting algorithm to
downweight closely related sequences and upweight distantly
related ones. This has the effect of making models less
biased <br>
by uneven phylogenetic representation. For example, two
identical sequences would typically each receive half the
weight that one sequence would. These options control which
<br>
algorithm gets used.</p>

<p style="margin-top: 1em">--wpb Use the Henikoff
position-based sequence weighting scheme [Henikoff and
Henikoff, J. Mol. Biol. 243:574, 1994]. This is the
default.</p>

<p style="margin-top: 1em">--wgsc Use the
Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et
al, J. Mol. Biol. 235:1067, 1994].</p>

<p style="margin-top: 1em">--wnone <br>
Turn sequence weighting off; e.g. explicitly set all
sequence weights to 1.0.</p>

<p style="margin-top: 1em">--wgiven <br>
Use sequence weights as given in annotation in the input
alignment file. If no weights were given, assume they are
all 1.0. The default is to determine new sequence <br>
weights by the Gerstein/Sonnhammer/Chothia algorithm,
ignoring any annotated weights.</p>

<p style="margin-top: 1em">--wblosum <br>
Use the BLOSUM filtering algorithm to weight the sequences,
instead of the default GSC weighting. Cluster the sequences
at a given percentage identity (see --wid); assign <br>
each cluster a total weight of 1.0, distributed equally
amongst the members of that cluster.</p>

<p style="margin-top: 1em">--wid &lt;x&gt; <br>
Controls the behavior of the --wblosum weighting option by
setting the percent identity for clustering the alignment to
&lt;x&gt;.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING EFFECTIVE
SEQUENCE NUMBER <br>
After relative weights are determined, they are normalized
to sum to a total effective sequence number, eff_nseq. This
number may be the actual number of sequences in the
align&acirc; <br>
ment, but it is almost always smaller than that. The default
entropy weighting method (--eent) reduces the effective
sequence number to reduce the information content (relative
<br>
entropy, or average expected score on true homologs) per
consensus position. The target relative entropy is
controlled by a two-parameter function, where the two
parameters are <br>
settable with --ere and --esigma.</p>

<p style="margin-top: 1em">--eent Use the entropy weighting
strategy to determine the effective sequence number that
gives a target mean match state relative entropy. This
option is the default, and can be <br>
turned off with --enone. The default target mean match state
relative entropy is 0.59 bits for models with at least 1
basepair and 0.38 bits for models with zero base&acirc; <br>
pairs, but changed with --ere. The default of 0.59 or 0.38
bits is automatically changed if the total relative entropy
of the model (summed match state relative entropy) <br>
is less than a cutoff, which is is 6.0 bits by default, but
can be changed with the expert, undocumented --eX option. If
you really want to play with that option, consult <br>
the source code.</p>

<p style="margin-top: 1em">--enone <br>
Turn off the entropy weighting strategy. The effective
sequence number is just the number of sequences in the
alignment.</p>

<p style="margin-top: 1em">--ere &lt;x&gt; <br>
Set the target mean match state relative entropy as
&lt;x&gt;. By default the target relative entropy per match
position is 0.59 bits for models with at least 1 basepair
and <br>
0.38 for models with zero basepairs.</p>

<p style="margin-top: 1em">--eminseq &lt;x&gt; <br>
Define the minimum allowed effective sequence number as
&lt;x&gt;.</p>

<p style="margin-top: 1em">--ehmmre &lt;x&gt; <br>
Set the target HMM mean match state relative entropy as
&lt;x&gt;. Entropy for basepairing match states is
calculated using marginalized basepair emission
probabilities.</p>

<p style="margin-top: 1em">--eset &lt;x&gt; <br>
Set the effective sequence number for entropy weighting as
&lt;x&gt;.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING FILTER P7
HMM CONSTRUCTION <br>
For each CM that cmbuild constructs, an accompanying filter
p7 HMM is built from the input alignment as well. These
options control filter HMM construction:</p>

<p style="margin-top: 1em">--p7ere &lt;x&gt; <br>
Set the target mean match state relative entropy for the
filter p7 HMM as &lt;x&gt;. By default the target relative
entropy per match position is 0.38 bits.</p>

<p style="margin-top: 1em">--p7ml Use a maximum likelihood
p7 HMM built from the CM as the filter HMM. This HMM will be
as similar as possible to the CM (while necessarily ignorant
of secondary structure).</p>

<p style="margin-top: 1em">Use --devhelp to see additional,
otherwise undocumented, filter HMM construction options.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING FILTER P7
HMM CALIBRATION <br>
After building each filter HMM, cmbuild determines
appropriate E-value parameters to use during filtering in
cmsearch and cmscan by sampling a set of sequences and
searching them <br>
with each HMM filter configuration and algorithm.</p>

<p style="margin-top: 1em">--EmN &lt;n&gt; Set the number
of sampled sequences for local MSV filter HMM calibration to
&lt;n&gt;. 200 by default.</p>

<p style="margin-top: 1em">--EvN &lt;n&gt; Set the number
of sampled sequences for local Viterbi filter HMM
calibration to &lt;n&gt;. 200 by default.</p>

<p style="margin-top: 1em">--ElfN &lt;n&gt; Set the number
of sampled sequences for local Forward filter HMM
calibration to &lt;n&gt;. 200 by default.</p>

<p style="margin-top: 1em">--EgfN &lt;n&gt; Set the number
of sampled sequences for glocal Forward filter HMM
calibration to &lt;n&gt;. 200 by default.</p>

<p style="margin-top: 1em">Use --devhelp to see additional,
otherwise undocumented, filter HMM calibration options.</p>

<p style="margin-top: 1em">OPTIONS FOR REFINING THE INPUT
ALIGNMENT <br>
--refine &lt;f&gt; <br>
Attempt to refine the alignment before building the CM using
expectation-maximization (EM). A CM is first built from the
initial alignment as usual. Then, the sequences in <br>
the alignment are realigned optimally (with the HMM banded
CYK algorithm, optimal means optimal given the bands) to the
CM, and a new CM is built from the resulting align&acirc;
<br>
ment. The sequences are then realigned to the new CM, and a
new CM is built from that alignment. This is continued until
convergence, specifically when the alignments for <br>
two successive iterations are not significantly different
(the summed bit scores of all the sequences in the alignment
changes less than 1% between two successive itera&acirc;
<br>
tions). The final alignment (the alignment used to build the
CM that gets written to &lt;cmfile_out&gt;) is written to
&lt;f&gt;.</p>

<p style="margin-top: 1em">-l With --refine, turn on the
local alignment algorithm, which allows the alignment to
span two or more subsequences if necessary (e.g. if the
structures of the query model <br>
and target sequence are only partially shared), allowing
certain large insertions and deletions in the structure to
be penalized differently than normal indels. The <br>
default is to globally align the query model to the target
sequences.</p>

<p style="margin-top: 1em">--gibbs <br>
Modifies the behavior of --refine so Gibbs sampling is used
instead of EM. The difference is that during the alignment
stage the alignment is not necessarily optimal, <br>
instead an alignment (parsetree) for each sequences is
sampled from the posterior distribution of alignments as
determined by the Inside algorithm. Due to this sampling
<br>
step --gibbs is non-deterministic, so different runs with
the same alignment may yield different results. This is not
true when --refine is used without the --gibbs <br>
option, in which case the final alignment and CM will always
be the same. When --gibbs is enabled, the --seed &lt;n&gt;
option can be used to seed the random number generator <br>
predictably, making the results reproducible. The goal of
the --gibbs option is to help expert RNA alignment curators
refine structural alignments by allowing them to <br>
observe alternative high scoring alignments.</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Seed the random number generator with &lt;n&gt;, an integer
&gt;= 0. This option can only be used in combination with
--gibbs. If &lt;n&gt; is nonzero, stochastic sampling of
align&acirc; <br>
ments will be reproducible; the same command will give the
same results. If &lt;n&gt; is 0, the random number generator
is seeded arbitrarily, and stochastic samplings may vary
<br>
from run to run of the same command. The default seed is
0.</p>

<p style="margin-top: 1em">--cyk With --refine, align with
the CYK algorithm. By default the optimal accuracy algorithm
is used. There is more information on this in the cmalign
manual page.</p>

<p style="margin-top: 1em">--notrunc <br>
With --refine, turn off the the truncated alignment
algorithm. There is more information on this in the cmalign
manual page.</p>

<p style="margin-top: 1em">Use --devhelp to see additional,
otherwise undocumented, alignment refinement options as well
as other output file options and options for building
multiple models for a single <br>
alignment.</p>

<p style="margin-top: 1em">SEE ALSO <br>
See infernal(1) for a master man page with a list of all the
individual man pages for programs in the Infernal
package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your Infernal distribution
(Userguide.pdf); or see the Infernal web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2016 Howard Hughes Medical Institute. <br>
Freely distributed under a BSD open source license.</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your Infernal source distribution, or see the Infernal web
page ().</p>

<p style="margin-top: 1em">AUTHOR <br>
The Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">Infernal 1.1.2 July 2016
cmbuild(1)</p>
<hr>
</body>
</html>
