<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>cmemit(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">cmemit(1)</td>
    <td class="head-vol">Infernal Manual</td>
    <td class="head-rtitle">cmemit(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cmemit - sample sequences from a covariance model
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>cmemit</b> <i>[options]</i> <i>&lt;cmfile&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
The <b>cmemit</b> program samples (emits) sequences from the covariance model(s)
  in <i>&lt;cmfile&gt;,</i> and writes them to output. Sampling sequences may be
  useful for a variety of purposes, including creating synthetic true positives
  for benchmarks or tests.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The default is to sample ten unaligned sequence from each CM. Alternatively,
  with the <b>-c</b> option, you can emit a single majority-rule consensus
  sequence; or with the <b>-a</b> option, you can emit an alignment.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The <i>&lt;cmfile&gt;</i> may contain a library of CMs, in which case each CM
  will be used in turn.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<i>&lt;cmfile&gt;</i> may be '-' (dash), which means reading this input from
  <i>stdin</i> rather than a file.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For models with zero basepairs, sequences are sampled from the profile HMM
  filter instead of the CM. However, since these models will be nearly identical
  (unless special options were used in <b>cmbuild</b> to prevent this), using
  the HMM instead of the CM will not change the output in a significant way,
  unless the <b>-l</b> option is used. With <b>-l,</b> the HMM will be
  configured for equiprobable model begin and end positions, while the CM will
  not. You can force <b>cmemit</b> to always sample from the CM with the
  <b>--nohmmonly</b> option.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save the synthetic sequences to file <i>&lt;f&gt;</i>
      rather than writing them to stdout.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Generate <i>&lt;n&gt;</i> sequences. The default value for
      <i>&lt;n&gt;</i> is 10.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b></dt>
  <dd class="It-tag">Write the generated sequences in unaligned format (FASTA).
      This is the default behavior.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b></dt>
  <dd class="It-tag">Write the generated sequences in an aligned format
      (STOCKHOLM) with consensus structure annotation rather than FASTA. Other
      output formats are possible with the <b>--outformat</b> option.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">Predict a single majority-rule consensus sequence instead
      of sampling sequences from the CM&#x00B4;s probability distribution.
      Highly conserved residues (base paired residues that score higher than 3.0
      bits, or single stranded residues that score higher than 1.0 bits) are
      shown in upper case; others are shown in lower case.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Embed the CM emitted sequences in a larger randomly
      generated sequence of length <i>&lt;n&gt;</i> generated from an HMM that
      was trained on real genomic sequences with various GC contents (the same
      HMM used by <b>cmcalibrate).</b> You can use the <b>--iid</b> option to
      generate 25% A, C, G, and U sequence instead. The CM emitted sequence will
      begin at a random position within the larger sequence and will be included
      in its entirety unless the <b>--u5p</b> or <b>--u3p</b> options are used.
      When <b>-e</b> is used in combination with <b>--u5p,</b> the CM emitted
      sequence will always begin at position 1 of the larger sequence and will
      be truncated 5'. When used in combination <b>--u3p</b> the CM emitted
      sequence will always end at position <i>&lt;n&gt;</i> of the larger
      sequence and will be truncated 3'.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b></dt>
  <dd class="It-tag">Configure the CMs into local mode before emitting
      sequences. By default the model will be in global mode. In local mode,
      large insertions and deletions are more common than in global mode.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_TRUNCATING_EMITTED_SEQUENCES"><a class="selflink" href="#OPTIONS_FOR_TRUNCATING_EMITTED_SEQUENCES">OPTIONS
  FOR TRUNCATING EMITTED SEQUENCES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--u5p</b></dt>
  <dd class="It-tag">Truncate all emitted sequences at a randomly chosen start
      position <b>&lt;n&gt;,</b> by only outputting residues beginning at
      <b>&lt;n&gt;.</b> A different start point is randomly chosen for each
      sequence.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--u3p</b></dt>
  <dd class="It-tag">Truncate all emitted sequences at a randomly chosen end
      position <b>&lt;n&gt;,</b> by only outputting residues up to position
      <b>&lt;n&gt;.</b> A different end point is randomly chosen for each
      sequence.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--a5p</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">In combination with the <b>-a</b> option, truncate the
      emitted alignment at a randomly chosen start match position
      <i>&lt;n&gt;,</i> by only outputting alignment columns for positions after
      match state <i>&lt;n&gt;</i> - 1. <i>&lt;n&gt;</i> must be an integer
      between 0 and the consensus length of the model (which can be determined
      using the <b>cmstat</b> program. As a special case, using 0 as
      <i>&lt;n&gt;</i> will result in a randomly chosen start position.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--a3p</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">In combination with the <b>-a</b> option, truncate the
      emitted alignment at a randomly chosen end match position
      <i>&lt;n&gt;,</i> by only outputting alignment columns for positions
      before match state <i>&lt;n&gt;</i> + 1. <i>&lt;n&gt;</i> must be an
      integer between 1 and the consensus length of the model (which can be
      determined using the <b>cmstat</b> program). As a special case, using 0 as
      <i>&lt;n&gt;</i> will result in a randomly chosen end position.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_OPTIONS"><a class="selflink" href="#OTHER_OPTIONS">OTHER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Seed the random number generator with <i>&lt;n&gt;,</i> an
      integer &gt;= 0. If <i>&lt;n&gt;</i> is nonzero, stochastic sampling of
      sequences will be reproducible; the same command will give the same
      results. If <i>&lt;n&gt;</i> is 0, the random number generator is seeded
      arbitrarily, and stochastic samplings will vary from run to run of the
      same command. The default seed is 0.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--iid</b></dt>
  <dd class="It-tag">With <b>-e,</b> generate the larger sequences as 25% each
      A, C, G and U.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rna</b></dt>
  <dd class="It-tag">Specify that the emitted sequences be output as RNA
      sequences. This is true by default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dna</b></dt>
  <dd class="It-tag">Specify that the emitted sequences be output as DNA
      sequences. By default, the output alphabet is RNA.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--idx</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Specify that the emitted sequences be named starting with
      <i>&lt;modelname&gt;.&lt;n&gt;.</i> By default <i>&lt;n&gt;</i> is 1.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">With <b>-a,</b> specify the output alignment format as
      <i>&lt;s&gt;.</i> Acceptable formats are: Pfam, AFA, A2M, Clustal, and
      Phylip. AFA is aligned fasta. Only Pfam and Stockholm alignment formats
      will include consensus structure annotation.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tfile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Dump tabular sequence parsetrees (tracebacks) for each
      emitted sequence to file <i>&lt;f&gt;.</i> Primarily useful for debugging.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--exp</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Exponentiate the emission and transition probabilities of
      the CM by <i>&lt;x&gt;</i> and then renormalize those distributions before
      emitting sequences. This option changes the CM probability distribution of
      parsetrees relative to default. With <i>&lt;x&gt;</i> less than 1.0 the
      emitted sequences will tend to have lower bit scores upon alignment to the
      CM. With &lt;x&gt; greater than 1.0, the emitted sequences will tend to
      have higher bit scores upon alignment to the CM. This bit score difference
      will increase as &lt;x&gt; moves further away from 1.0 in either
      direction. If &lt;x&gt; equals 1.0, this option has no effect relative to
      default. This option is useful for generating sequences that are either
      more difficult ( <i>&lt;x&gt;</i> &lt; 1.0) or easier ( <i>&lt;x&gt;</i>
      &gt; 1.0) for the CM to distinguish as homologous from background, random
      sequence.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hmmonly</b></dt>
  <dd class="It-tag">Emit from the filter profile HMM instead of the CM.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nohmmonly</b></dt>
  <dd class="It-tag">Never emit from the filter profile HMM, always use the CM,
      even for models with zero basepairs.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>infernal(1)</b> for a master man page with a list of all the individual
  man pages for programs in the Infernal package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your Infernal
  distribution (Userguide.pdf); or see the Infernal web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2016 Howard Hughes Medical Institute.
Freely distributed under a BSD open source license.
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your Infernal source distribution, or see the Infernal web page
  ().
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2016</td>
    <td class="foot-os">Infernal 1.1.2</td>
  </tr>
</table>
</body>
</html>
