<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:59:22 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>cmemit(1) Infernal Manual cmemit(1)</p>

<p style="margin-top: 1em">NAME <br>
cmemit - sample sequences from a covariance model</p>

<p style="margin-top: 1em">SYNOPSIS <br>
cmemit [options] &lt;cmfile&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The cmemit program samples (emits) sequences from the
covariance model(s) in &lt;cmfile&gt;, and writes them to
output. Sampling sequences may be useful for a variety of
purposes, <br>
including creating synthetic true positives for benchmarks
or tests.</p>

<p style="margin-top: 1em">The default is to sample ten
unaligned sequence from each CM. Alternatively, with the -c
option, you can emit a single majority-rule consensus
sequence; or with the -a option, <br>
you can emit an alignment.</p>

<p style="margin-top: 1em">The &lt;cmfile&gt; may contain a
library of CMs, in which case each CM will be used in
turn.</p>

<p style="margin-top: 1em">&lt;cmfile&gt; may be
&rsquo;-&rsquo; (dash), which means reading this input from
stdin rather than a file.</p>

<p style="margin-top: 1em">For models with zero basepairs,
sequences are sampled from the profile HMM filter instead of
the CM. However, since these models will be nearly identical
(unless special options <br>
were used in cmbuild to prevent this), using the HMM instead
of the CM will not change the output in a significant way,
unless the -l option is used. With -l, the HMM will be <br>
configured for equiprobable model begin and end positions,
while the CM will not. You can force cmemit to always sample
from the CM with the --nohmmonly option.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
available options.</p>

<p style="margin-top: 1em">-o &lt;f&gt; Save the synthetic
sequences to file &lt;f&gt; rather than writing them to
stdout.</p>

<p style="margin-top: 1em">-N &lt;n&gt; Generate &lt;n&gt;
sequences. The default value for &lt;n&gt; is 10.</p>

<p style="margin-top: 1em">-u Write the generated sequences
in unaligned format (FASTA). This is the default
behavior.</p>

<p style="margin-top: 1em">-a Write the generated sequences
in an aligned format (STOCKHOLM) with consensus structure
annotation rather than FASTA. Other output formats are
possible with the --outfor&acirc; <br>
mat option.</p>

<p style="margin-top: 1em">-c Predict a single
majority-rule consensus sequence instead of sampling
sequences from the CM&Acirc;&acute;s probability
distribution. Highly conserved residues (base paired
residues <br>
that score higher than 3.0 bits, or single stranded residues
that score higher than 1.0 bits) are shown in upper case;
others are shown in lower case.</p>

<p style="margin-top: 1em">-e &lt;n&gt; Embed the CM
emitted sequences in a larger randomly generated sequence of
length &lt;n&gt; generated from an HMM that was trained on
real genomic sequences with various GC con&acirc; <br>
tents (the same HMM used by cmcalibrate). You can use the
--iid option to generate 25% A, C, G, and U sequence
instead. The CM emitted sequence will begin at a random <br>
position within the larger sequence and will be included in
its entirety unless the --u5p or --u3p options are used.
When -e is used in combination with --u5p, the CM <br>
emitted sequence will always begin at position 1 of the
larger sequence and will be truncated 5&rsquo;. When used in
combination --u3p the CM emitted sequence will always end at
<br>
position &lt;n&gt; of the larger sequence and will be
truncated 3&rsquo;.</p>

<p style="margin-top: 1em">-l Configure the CMs into local
mode before emitting sequences. By default the model will be
in global mode. In local mode, large insertions and
deletions are more common <br>
than in global mode.</p>

<p style="margin-top: 1em">OPTIONS FOR TRUNCATING EMITTED
SEQUENCES <br>
--u5p Truncate all emitted sequences at a randomly chosen
start position &lt;n&gt;, by only outputting residues
beginning at &lt;n&gt;. A different start point is randomly
chosen for each <br>
sequence.</p>

<p style="margin-top: 1em">--u3p Truncate all emitted
sequences at a randomly chosen end position &lt;n&gt;, by
only outputting residues up to position &lt;n&gt;. A
different end point is randomly chosen for each <br>
sequence.</p>

<p style="margin-top: 1em">--a5p &lt;n&gt; <br>
In combination with the -a option, truncate the emitted
alignment at a randomly chosen start match position
&lt;n&gt;, by only outputting alignment columns for
positions after <br>
match state &lt;n&gt; - 1. &lt;n&gt; must be an integer
between 0 and the consensus length of the model (which can
be determined using the cmstat program. As a special case,
using 0 <br>
as &lt;n&gt; will result in a randomly chosen start
position.</p>

<p style="margin-top: 1em">--a3p &lt;n&gt; <br>
In combination with the -a option, truncate the emitted
alignment at a randomly chosen end match position &lt;n&gt;,
by only outputting alignment columns for positions before
<br>
match state &lt;n&gt; + 1. &lt;n&gt; must be an integer
between 1 and the consensus length of the model (which can
be determined using the cmstat program). As a special case,
using 0 <br>
as &lt;n&gt; will result in a randomly chosen end
position.</p>

<p style="margin-top: 1em">OTHER OPTIONS <br>
--seed &lt;n&gt; <br>
Seed the random number generator with &lt;n&gt;, an integer
&gt;= 0. If &lt;n&gt; is nonzero, stochastic sampling of
sequences will be reproducible; the same command will give
the same <br>
results. If &lt;n&gt; is 0, the random number generator is
seeded arbitrarily, and stochastic samplings will vary from
run to run of the same command. The default seed is 0.</p>

<p style="margin-top: 1em">--iid With -e, generate the
larger sequences as 25% each A, C, G and U.</p>

<p style="margin-top: 1em">--rna Specify that the emitted
sequences be output as RNA sequences. This is true by
default.</p>

<p style="margin-top: 1em">--dna Specify that the emitted
sequences be output as DNA sequences. By default, the output
alphabet is RNA.</p>

<p style="margin-top: 1em">--idx &lt;n&gt; <br>
Specify that the emitted sequences be named starting with
&lt;modelname&gt;.&lt;n&gt;. By default &lt;n&gt; is 1.</p>

<p style="margin-top: 1em">--outformat &lt;s&gt; <br>
With -a, specify the output alignment format as &lt;s&gt;.
Acceptable formats are: Pfam, AFA, A2M, Clustal, and Phylip.
AFA is aligned fasta. Only Pfam and Stockholm alignment <br>
formats will include consensus structure annotation.</p>

<p style="margin-top: 1em">--tfile &lt;f&gt; <br>
Dump tabular sequence parsetrees (tracebacks) for each
emitted sequence to file &lt;f&gt;. Primarily useful for
debugging.</p>

<p style="margin-top: 1em">--exp &lt;x&gt; <br>
Exponentiate the emission and transition probabilities of
the CM by &lt;x&gt; and then renormalize those distributions
before emitting sequences. This option changes the CM <br>
probability distribution of parsetrees relative to default.
With &lt;x&gt; less than 1.0 the emitted sequences will tend
to have lower bit scores upon alignment to the CM. With <br>
&lt;x&gt; greater than 1.0, the emitted sequences will tend
to have higher bit scores upon alignment to the CM. This bit
score difference will increase as &lt;x&gt; moves further
away <br>
from 1.0 in either direction. If &lt;x&gt; equals 1.0, this
option has no effect relative to default. This option is
useful for generating sequences that are either more
dif&acirc; <br>
ficult ( &lt;x&gt; &lt; 1.0) or easier ( &lt;x&gt; &gt; 1.0)
for the CM to distinguish as homologous from background,
random sequence.</p>

<p style="margin-top: 1em">--hmmonly <br>
Emit from the filter profile HMM instead of the CM.</p>

<p style="margin-top: 1em">--nohmmonly <br>
Never emit from the filter profile HMM, always use the CM,
even for models with zero basepairs.</p>

<p style="margin-top: 1em">SEE ALSO <br>
See infernal(1) for a master man page with a list of all the
individual man pages for programs in the Infernal
package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your Infernal distribution
(Userguide.pdf); or see the Infernal web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2016 Howard Hughes Medical Institute. <br>
Freely distributed under a BSD open source license.</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your Infernal source distribution, or see the Infernal web
page ().</p>

<p style="margin-top: 1em">AUTHOR <br>
The Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">Infernal 1.1.2 July 2016
cmemit(1)</p>
<hr>
</body>
</html>
