<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>cmsearch(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">cmsearch(1)</td>
    <td class="head-vol">Infernal Manual</td>
    <td class="head-rtitle">cmsearch(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cmsearch - search covariance model(s) against a sequence database
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>cmsearch</b> <i>[options]</i> <i>&lt;cmfile&gt;</i> <i>&lt;seqdb&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>cmsearch</b> is used to search one or more covariance models (CMs) against a
  sequence database. For each CM in <i>&lt;cmfile&gt;,</i> use that query CM to
  search the target database of sequences in <i>&lt;seqdb&gt;,</i> and output
  ranked lists of the sequences with the most significant matches to the CM. To
  build CMs from multiple alignments, see <b>cmbuild.</b>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The query <i>&lt;cmfile&gt;</i> must have been calibrated for E-values with
  <b>cmcalibrate.</b> As a special exception, any models in
  <i>&lt;cmfile&gt;</i> that have zero basepairs need not be calibrated. For
  these models, profile HMM search algorithms will be used instead of CM ones,
  as discussed further below.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The query <i>&lt;cmfile&gt;</i> may be '-' (a dash character), in which case the
  query CM input will be read from a &lt;stdin&gt; pipe instead of from a file.
  The <i>&lt;seqdb&gt;</i> may not be '-' because the current implementation
  needs to be able to rewind the database, which is not possible with
  <i>stdin</i> input.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The output format is designed to be human-readable, but is often so voluminous
  that reading it is impractical, and parsing it is a pain. The <b>--tblout</b>
  option saves output in a simple tabular format that is concise and easier to
  parse. The <b>-o</b> option allows redirecting the main output, including
  throwing it away in /dev/null.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<b>cmsearch</b> reexamines the 5' and 3' termini of target sequences using
  specialized algorithms for detection of <i>truncated</i> hits, in which part
  of the 5' and/or 3' end of the actual full length homologous sequence is
  missing in the target sequence file. These types of hits will be most common
  in sequence files consisting of unassembled sequencing reads. By default, any
  5' truncated hit is required to include the first residue of the target
  sequence it derives from in <i>&lt;seqdb&gt;,</i> and any 3' truncated hit is
  required to include the final residue of the target sequence it derives from.
  Any 5' and 3' truncated hit must include the first and final residue of the
  target sequence it derives from. The <b>--anytrunc</b> option will relax the
  requirements for hit inclusion of sequence endpoints, and truncated hits are
  allowed to start and stop at any positions of target sequences. Importantly
  though, with <b>--anytrunc,</b> hit E-values will be less accurate because
  model calibration does not consider the possibility of truncated hits, so use
  it with caution. The <b>--notrunc</b> option can be used to turn off truncated
  hit detection. <b>--notrunc</b> will reduce the running time of
  <b>cmsearch,</b> most significantly for target <i>&lt;seqdb&gt;</i> files that
  include many short sequences.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Truncated hit detection is automatically turned off when the <b>--max,</b>
  <b>--nohmm,</b> <b>--qdb,</b> or <b>--nonbanded</b> options are used because
  it relies on the use of an accelerated HMM banded alignment strategy that is
  turned off by any of those options.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and all
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">Turn on the <i>glocal</i> alignment algorithm, global with
      respect to the query model and local with respect to the target database.
      By default, the local alignment algorithm is used which is local with
      respect to both the target sequence and the model. In local mode, the
      alignment to span two or more subsequences if necessary (e.g. if the
      structures of the query model and target sequence are only partially
      shared), allowing certain large insertions and deletions in the structure
      to be penalized differently than normal indels. Local mode performs better
      on empirical benchmarks and is significantly more sensitive for remote
      homology detection. Empirically, glocal searches return many fewer hits
      than local searches, so glocal may be desired for some applications. With
      <b>-g,</b> all models must be calibrated, even those with zero basepairs.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Calculate E-values as if the search space size was
      <i>&lt;x&gt;</i> megabases (Mb). Without the use of this option, the
      search space size is defined as the total number of nucleotides in
      <i>&lt;seqdb&gt;</i> times 2, because both strands of each target sequence
      will be searched.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--devhelp</b></dt>
  <dd class="It-tag">Print help, as with <b>-h ,</b> but also include expert
      options that are not displayed with <b>-h .</b> These expert options are
      not expected to be relevant for the vast majority of users and so are not
      described in the manual page. The only resources for understanding what
      they actually do are the brief one-line descriptions output when
      <b>--devhelp</b> is enabled, and the source code.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_CONTROLLING_OUTPUT"><a class="selflink" href="#OPTIONS_FOR_CONTROLLING_OUTPUT">OPTIONS
  FOR CONTROLLING OUTPUT</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Direct the main human-readable output to a file
      <i>&lt;f&gt;</i> instead of the default stdout.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save a multiple alignment of all significant hits (those
      satisfying <i>inclusion thresholds)</i> to the file <i>&lt;f&gt;.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tblout</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save a simple tabular (space-delimited) file summarizing
      the hits found, with one data line per hit. The format of this file is
      described in section 6 of the Infernal user guide.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--acc</b></dt>
  <dd class="It-tag">Use accessions instead of names in the main output, where
      available for profiles and/or sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noali</b></dt>
  <dd class="It-tag">Omit the alignment section from the main output. This can
      greatly reduce the output volume.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--notextw</b></dt>
  <dd class="It-tag">Unlimit the length of each line in the main output. The
      default is a limit of 120 characters per line, which helps in displaying
      the output cleanly on terminals and in editors, but can truncate target
      profile description lines.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--textw</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the main output's line length limit to <i>&lt;n&gt;</i>
      characters per line. The default is 120.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--verbose</b></dt>
  <dd class="It-tag">Include extra search pipeline statistics in the main
      output, including filter survival statistics for truncated hit detection
      and number of envelopes discarded due to matrix size overflows.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_REPORTING_THRESHOLDS"><a class="selflink" href="#OPTIONS_CONTROLLING_REPORTING_THRESHOLDS">OPTIONS
  CONTROLLING REPORTING THRESHOLDS</a></h1>
Reporting thresholds control which hits are reported in output files (the main
  output and <i>--tblout)</i> Hits are ranked by statistical significance
  (E-value). By default, all hits with an E-value &lt;= 10 are reported. The
  following options allow you to change the default E-value reporting
  thresholds, or to use bit score thresholds instead.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">In the per-target output, report target sequences with an
      E-value of &lt;= <i>&lt;x&gt;.</i> The default is 10.0, meaning that on
      average, about 10 false positives will be reported per query, so you can
      see the top of the noise and decide for yourself if it's really noise.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Instead of thresholding per-CM output on E-value, report
      target sequences with a bit score of &gt;= <i>&lt;x&gt;.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_INCLUSION_THRESHOLDS"><a class="selflink" href="#OPTIONS_FOR_INCLUSION_THRESHOLDS">OPTIONS
  FOR INCLUSION THRESHOLDS</a></h1>
Inclusion thresholds are stricter than reporting thresholds. Inclusion
  thresholds control which hits are considered to be reliable enough to be
  included in an output alignment or in a possible subsequent search round, or
  marked as significant (&quot;!&quot;) as opposed to questionable
  (&quot;?&quot;) in hit output.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Use an E-value of &lt;= <i>&lt;x&gt;</i> as the hit
      inclusion threshold. The default is 0.01, meaning that on average, about 1
      false positive would be expected in every 100 searches with different
      query sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Instead of using E-values for setting the inclusion
      threshold, instead use a bit score of &gt;= <i>&lt;x&gt;</i> as the hit
      inclusion threshold. By default this option is unset.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_MODEL-SPECIFIC_SCORE_THRESHOLDING"><a class="selflink" href="#OPTIONS_FOR_MODEL-SPECIFIC_SCORE_THRESHOLDING">OPTIONS
  FOR MODEL-SPECIFIC SCORE THRESHOLDING</a></h1>
Curated CM databases may define specific bit score thresholds for each CM,
  superseding any thresholding based on statistical significance alone.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
To use these options, the profile must contain the appropriate (GA, TC, and/or
  NC) optional score threshold annotation; this is picked up by <b>cmbuild</b>
  from Stockholm format alignment files. Each thresholding option has a score of
  <i>&lt;x&gt;</i> bits, and acts as if <b>-T</b><i> &lt;x&gt;</i>
  <b>--incT</b><i> &lt;x&gt;</i> has been applied specifically using each
  model's curated thresholds.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_ga</b></dt>
  <dd class="It-tag">Use the GA (gathering) bit scores in the model to set hit
      reporting and inclusion thresholds. GA thresholds are generally considered
      to be the reliable curated thresholds defining family membership; for
      example, in Rfam, these thresholds define what gets included in Rfam Full
      alignments based on searches with Rfam Seed models.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_nc</b></dt>
  <dd class="It-tag">Use the NC (noise cutoff) bit score thresholds in the model
      to set hit reporting and inclusion thresholds. NC thresholds are generally
      considered to be the score of the highest-scoring known false positive.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_tc</b></dt>
  <dd class="It-tag">Use the TC (trusted cutoff) bit score thresholds in the
      model to set hit reporting and inclusion thresholds. TC thresholds are
      generally considered to be the score of the lowest-scoring known true
      positive that is above all known false positives.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_THE_ACCELERATION_PIPELINE"><a class="selflink" href="#OPTIONS_CONTROLLING_THE_ACCELERATION_PIPELINE">OPTIONS
  CONTROLLING THE ACCELERATION PIPELINE</a></h1>
Infernal 1.1 searches are accelerated in a six-stage filter pipeline. The first
  five stages use a profile HMM to define envelopes that are passed to the stage
  six CM CYK filter. Any envelopes that survive all filters are assigned final
  scores using the the CM Inside algorithm. (See the user guide for more
  information.)
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The profile HMM filter is built by the <b>cmbuild</b> program and is stored in
  <i>&lt;cmfile&gt;.</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Each successive filter is slower than the previous one, but better than it at
  disciminating between subsequences that may contain high-scoring CM hits and
  those that do not. The first three HMM filter stages are the same as those
  used in HMMER3. Stage 1 (F1) is the local HMM SSV filter modified for long
  sequences. Stage 2 (F2) is the local HMM Viterbi filter. Stage 3 (F3) is the
  local HMM Forward filter. Each of the first three stages uses the profile HMM
  in local mode, which allows a target subsequence to align to any region of the
  HMM. Stage 4 (F4) is a glocal HMM filter, which requires a target subsequence
  to align to the full-length profile HMM. Stage 5 (F5) is the glocal HMM
  envelope definition filter, which uses HMMER3's domain identification
  heursitics to define envelope boundaries. After each stage from 2 to 5 a bias
  filter step (F2b, F3b, F4b, and F5b) is used to remove sequences that appear
  to have passed the filter due to biased composition alone. Any envelopes that
  survive stages F1 through F5b are then passed with the local CM CYK filter.
  The CYK filter uses constraints (bands) derived from an HMM alignment of the
  envelope to reduce the number of required calculations and save time. Any
  envelopes that pass CYK are scored with the local CM Inside algorithm, again
  using HMM bands for acceleration.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The default filter thresholds that define the minimum score required for a
  subsequence to survive each stage are defined based on the size of the
  database in <i>&lt;seqdb&gt;</i> (or the size <i>&lt;x&gt;</i> in megabases
  (Mb) specified by the <b>-Z</b><i> &lt;x&gt;</i> or <b>--FZ</b><i>
  &lt;x&gt;</i> options). For larger databases, the filters are more strict
  leading to more acceleration but potentially a greater loss of sensitivity.
  The rationale is that for larger databases, hits must have higher scores to
  achieve statistical significance, so stricter filtering that removes lower
  scoring insignificant hits is acceptable.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The P-value thresholds for all possible search space sizes and all filter stages
  are listed next. (A P-value threshold of 0.01 means that roughly 1% of the
  highest scoring nonhomologous subsequence are expected to pass the filter.) Z
  is defined as the number of nucleotides in the complete target sequence file
  times 2 because both strands will be searched with each model.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
If Z is less than 2 Mb: F1 is 0.35; F2 and F2b are off; F3, F3b, F4, F4b and F5
  are 0.02; F6 is 0.0001.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
If Z is between 2 Mb and 20 Mb: F1 is 0.35; F2 and F2b are off; F3, F3b, F4, F4b
  and F5 are 0.005; F6 is 0.0001.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
If Z is between 20 Mb and 200 Mb: F1 is 0.35; F2 and F2b are 0.15; F3, F3b, F4,
  F4b and F5 are 0.003; F6 is 0.0001.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
If Z is between 200 Mb and 2 Gb: F1 is 0.15; F2 and F2b are 0.15; F3, F3b, F4,
  F4b, F5, and F5b are 0.0008; and F6 is 0.0001.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
If Z is between 2 Gb and 20 Gb: F1 is 0.15; F2 and F2b are 0.15; F3, F3b, F4,
  F4b, F5, and F5b are 0.0002; and F6 is 0.0001.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
If Z is more than 20 Gb: F1 is 0.06; F2 and F2b are 0.02; F3, F3b, F4, F4b, F5,
  and F5b are 0.0002; and F6 is 0.0001.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
These thresholds were chosen based on performance on an internal benchmark
  testing many different possible settings.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
There are five options for controlling the general filtering level. These
  options are, in order from least strict (slowest but most sensitive) to most
  strict (fastest but least sensitive): <b>--max,</b> <b>--nohmm,</b>
  <b>--mid,</b> <b>--default,</b> (this is the default setting), <b>--rfam.</b>
  and <b>--hmmonly.</b> With <b>--default</b> the filter thresholds will be
  database-size dependent. See the explanation of each of these individual
  options below for more information.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Additionally, an expert user can precisely control each filter stage score
  threshold with the <b>--F1,</b> <b>--F1b,</b> <b>--F2,</b> <b>--F2b,</b>
  <b>--F3,</b> <b>--F3b,</b> <b>--F4,</b> <b>--F4b,</b> <b>--F5,</b>
  <b>--F5b,</b> and <b>--F6</b> options. As well as turn each stage on or off
  with the <b>--noF1,</b> <b>--doF1b,</b> <b>--noF2,</b> <b>--noF2b,</b>
  <b>--noF3,</b> <b>--noF3b,</b> <b>--noF4,</b> <b>--noF4b,</b> <b>--noF5,</b>
  and <b>--noF6.</b> options. These options are only displayed if the
  <b>--devhelp</b> option is used to keep the number of displayed options with
  <b>-h</b> reasonable, and because they are only expected to be useful to a
  small minority of users.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
As a special case, for any models in <i>&lt;cmfile&gt;</i> which have zero
  basepairs, profile HMM searches are run instead of CM searches. HMM algorithms
  are more efficient than CM algorithms, and the benefit of CM algorithms is
  lost for models with no secondary structure (zero basepairs). These profile
  HMM searches will run significantly faster than the CM searches. You can force
  HMM-only searches with the <b>--hmmonly</b> option. For more information on
  HMM-only searches see the description of the <b>--hmmonly</b> option below,
  and the user guide.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max</b></dt>
  <dd class="It-tag">Turn off all filters, and run non-banded Inside on every
      full-length target sequence. This increases sensitivity somewhat, at an
      extremely large cost in speed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nohmm</b></dt>
  <dd class="It-tag">Turn off all HMM filter stages (F1 through F5b). The CYK
      filter, using QDBs, will be run on every full-length target sequence and
      will enforce a P-value threshold of 0.0001. Each subsequence that survives
      CYK will be passed to Inside, which will also use QDBs (but a looser set).
      This increases sensitivity somewhat, at a very large cost in speed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mid</b></dt>
  <dd class="It-tag">Turn off the HMM SSV and Viterbi filter stages (F1 through
      F2b). Set remaining HMM filter thresholds (F3 through F5b) to 0.02 by
      default, but changeable to <i>&lt;x&gt;</i> with <b>--Fmid</b><i>
      &lt;x&gt;</i> sequence. This may increase sensitivity, at a significant
      cost in speed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--default</b></dt>
  <dd class="It-tag">Use the default filtering strategy. This option is on by
      default. The filter thresholds are determined based on the database size.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rfam</b></dt>
  <dd class="It-tag">Use a strict filtering strategy devised for large databases
      (more than 20 Gb). This will accelerate the search at a potential cost to
      sensitivity. It will have no effect if the database is larger than 20 Gb.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hmmonly</b></dt>
  <dd class="It-tag">Only use the filter profile HMM for searches, do not use
      the CM. Only filter stages F1 through F3 will be executed, using strict
      P-value thresholds (0.02 for F1, 0.001 for F2 and 0.00001 for F3).
      Additionally a bias composition filter is used after the F1 stage (with
      P=0.02 survival threshold). Any hit that survives all stages and has an
      HMM E-value or bit score above the reporting threshold will be output. The
      user can change the HMM-only filter thresholds and options with
      <b>--hmmF1,</b> <b>--hmmF2,</b> <b>--hmmF3,</b> <b>--hmmnobias,</b>
      <b>--hmmnonull2,</b> and <b>--hmmmax.</b> By default, searches for any
      model with zero basepairs will be run in HMM-only mode. This can be turned
      off, forcing CM searches for these models with the <b>--nohmmonly</b>
      option. These options are only displayed if the <b>--devhelp</b> option is
      used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--FZ</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set filter thresholds as the defaults used if the database
      were <b>&lt;x&gt;</b> megabases (Mb). If used with <b>&lt;x&gt;</b>
      greater than 20000 (20 Gb) this option has the same effect as
      <b>--rfam.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--Fmid</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">With the <b>--mid</b> option set the HMM filter thresholds
      (F3 through F5b) to <i>&lt;x&gt;.</i> By default, <i>&lt;x&gt;</i> is
      0.02.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_OPTIONS"><a class="selflink" href="#OTHER_OPTIONS">OTHER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--notrunc</b></dt>
  <dd class="It-tag">Turn off truncated hit detection.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--anytrunc</b></dt>
  <dd class="It-tag">Allow truncated hits to begin and end at any position in a
      target sequence. By default, 5' truncated hits must include the first
      residue of their target sequence and 3' truncated hits must include the
      final residue of their target sequence. With this option you may observe
      fewer full length hits that extend to the beginning and end of the query
      CM.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nonull3</b></dt>
  <dd class="It-tag">Turn off the null3 CM score corrections for biased
      composition. This correction is not used during the HMM filter stages.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mxsize</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the maximum allowable CM DP matrix size to
      <i>&lt;x&gt;</i> megabytes. By default this size is 128 Mb. This should be
      large enough for the vast majority of searches, especially with smaller
      models. If <b>cmsearch</b> encounters an envelope in the CYK or Inside
      stage that requires a larger matrix, the envelope will be discounted from
      consideration. This behavior is like an additional filter that prevents
      expensive (slow) CM DP calculations, but at a potential cost to
      sensitivity. Note that if <b>cmsearch</b> is being run in <i>&lt;n&gt;</i>
      multiple threads on a multicore machine then each thread may have an
      allocated matrix of up to size <i>&lt;x&gt;</i> Mb at any given time.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--smxsize</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the maximum allowable CM search DP matrix size to
      <i>&lt;x&gt;</i> megabytes. By default this size is 128 Mb. This option is
      only relevant if the CM will not use HMM banded matrices, i.e. if the
      <b>--max,</b> <b>--nohmm,</b> <b>--qdb,</b> <b>--fqdb,</b>
      <b>--nonbanded,</b> or <b>--fnonbanded</b> options are also used. Note
      that if <b>cmsearch</b> is being run in <i>&lt;n&gt;</i> multiple threads
      on a multicore machine then each thread may have an allocated matrix of up
      to size <i>&lt;x&gt;</i> Mb at any given time.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cyk</b></dt>
  <dd class="It-tag">Use the CYK algorithm, not Inside, to determine the final
      score of all hits.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--acyk</b></dt>
  <dd class="It-tag">Use the CYK algorithm to align hits. By default, the
      Durbin/Holmes optimal accuracy algorithm is used, which finds the
      alignment that maximizes the expected accuracy of all aligned residues.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wcx</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">For each CM, set the W parameter, the expected maximum
      length of a hit, to <i>&lt;x&gt;</i> times the consensus length of the
      model. By default, the W parameter is read from the CM file and was
      calculated based on the transition probabilities of the model by
      <b>cmbuild.</b> You can find out what the default W is for a model using
      <b>cmstat.</b> This option should be used with caution as it impacts the
      filtering pipeline at several different stages in nonobvious ways. It is
      only recommended for expert users searching for hits that are much longer
      than any of the homologs used to build the model in <b>cmbuild,</b> e.g.
      ones with large introns or other large insertions. This option cannot be
      used in combination with the <b>--nohmm,</b> <b>--fqdb</b> or <b>--qdb</b>
      options because in those cases W is limited by query-dependent bands.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--toponly</b></dt>
  <dd class="It-tag">Only search the top (Watson) strand of target sequences in
      <i>&lt;seqdb&gt;.</i> By default, both strands are searched. This will
      halve the database size (Z).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bottomonly</b></dt>
  <dd class="It-tag">Only search the bottom (Crick) strand of target sequences
      in <i>&lt;seqdb&gt;.</i> By default, both strands are searched. This will
      halve the database size (Z).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Assert that the target sequence database file is in format
      <i>&lt;s&gt;.</i> Accepted formats include <i>fasta,</i> <i>embl,</i>
      <i>genbank,</i> <i>ddbj,</i> <i>stockholm,</i> <i>pfam,</i> <i>a2m,</i>
      <i>afa,</i> <i>clustal,</i> and <i>phylip</i> The default is to autodetect
      the format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cpu</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the number of parallel worker threads to
      <i>&lt;n&gt;.</i> By default, Infernal sets this to the number of CPU
      cores it detects in your machine - that is, it tries to maximize the use
      of your available processor cores. Setting <i>&lt;n&gt;</i> higher than
      the number of available cores is of little if any value, but you may want
      to set it to something less. You can also control this number by setting
      an environment variable, <i>INFERNAL_NCPU.</i> This option is only
      available if Infernal was compiled with POSIX threads support. This is the
      default, but it may have been turned off at compile-time for your site or
      machine for some reason.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stall</b></dt>
  <dd class="It-tag">For debugging the MPI master/worker version: pause after
      start, to enable the developer to attach debuggers to the running master
      and worker(s) processes. Send SIGCONT signal to release the pause. (Under
      gdb: <i>(gdb) signal SIGCONT)</i> (Only available if optional MPI support
      was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mpi</b></dt>
  <dd class="It-tag">Run in MPI master/worker mode, using <i>mpirun.</i> To use
      <b>--mpi,</b> the sequence file must have first been 'indexed' using the
      <b>esl-sfetch</b> program, which is included with Infernal, in the
      <i>easel/miniapps/</i> subdirectory. (Only available if optional MPI
      support was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>infernal(1)</b> for a master man page with a list of all the individual
  man pages for programs in the Infernal package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your Infernal
  distribution (Userguide.pdf); or see the Infernal web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2016 Howard Hughes Medical Institute.
Freely distributed under a BSD open source license.
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your Infernal source distribution, or see the Infernal web page
  ().
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2016</td>
    <td class="foot-os">Infernal 1.1.2</td>
  </tr>
</table>
</body>
</html>
