<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 15:59:43 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CODONW(1) User Commands CODONW(1)</p>

<p style="margin-top: 1em">NAME <br>
codonw - Correspondence Analysis of Codon Usage</p>

<p style="margin-top: 1em">DESCRIPTION <br>
codonW [inputfile] [outputfile] [bulkoutfile] [options]
General options and defaults:</p>

<p style="margin-top: 1em">-h(elp) <br>
This help message</p>

<p style="margin-top: 1em">-nomenu <br>
Prevent the menu interface being displayed</p>

<p style="margin-top: 1em">-nowarn <br>
Prevent warnings about sequences being displayed</p>

<p style="margin-top: 1em">-silent <br>
Overwrite files silently</p>

<p style="margin-top: 1em">-totals <br>
Concatenate all genes in inputfile</p>

<p style="margin-top: 1em">-machine <br>
Machine readable output</p>

<p style="margin-top: 1em">-human Human readable output</p>

<p style="margin-top: 1em">-code N <br>
Genetic code as defined under menu 3 option 5</p>

<p style="margin-top: 1em">-f_type N <br>
Fop/CBI codons as defined by menu 3 option 6</p>

<p style="margin-top: 1em">-c_type N <br>
Cai fitness values as defined by menu 3 option 7</p>

<p style="margin-top: 1em">-t (char) <br>
Column separator to be used in output files
(comma,tab,space)</p>

<p style="margin-top: 1em">Codon usage indices and Amino
acid indices</p>

<p style="margin-top: 1em">-cai calculate Codon Adaptation
Index (CAI)</p>

<p style="margin-top: 1em">-fop calculate Frequency of
OPtimal codons index (FOP)</p>

<p style="margin-top: 1em">-cbi calculate Codon Bias Index
(CBI)</p>

<p style="margin-top: 1em">-enc Effective Number of Codons
(ENc)</p>

<p style="margin-top: 1em">-gc G+C content of gene (all 3
codon positions)</p>

<p style="margin-top: 1em">-gcs3 GC of synonymous codons
3rd positions</p>

<p style="margin-top: 1em">-sil_base <br>
Base composition at synonymous third codon positions</p>

<p style="margin-top: 1em">-L_sym Number of synonymous
codons</p>

<p style="margin-top: 1em">-L_aa Total number of synonymous
and non-synonymous codons</p>

<p style="margin-top: 1em">-all_indices <br>
All the above indices</p>

<p style="margin-top: 1em">-aro Calculate aromaticity of
protein</p>

<p style="margin-top: 1em">-hyd Calculate hydropathicity of
protein</p>

<p style="margin-top: 1em">-cai_file <br>
{file} User input file of CAI values</p>

<p style="margin-top: 1em">-cbi_file <br>
{file} User input file of CBI values</p>

<p style="margin-top: 1em">-fop_file <br>
{file} User input file of Fop values</p>

<p style="margin-top: 1em">Correspondence analysis (COA)
options</p>

<p style="margin-top: 1em">-coa_cu <br>
COA of codon usage frequencies</p>

<p style="margin-top: 1em">-coa_rscu <br>
COA of Relative Synonymous Codon Usage</p>

<p style="margin-top: 1em">-coa_aa <br>
COA of amino acid usage frequencies</p>

<p style="margin-top: 1em">-coa_expert <br>
Generate detailed(expert) statistics on COA</p>

<p style="margin-top: 1em">-coa_axes N <br>
Select number of axis to record</p>

<p style="margin-top: 1em">-coa_num N <br>
Select number of genes to use to identify optimal codons
values can be whole numbers or a percentage (5 or 10%)</p>

<p style="margin-top: 1em">Bulk output options | only one
can be selected per analysis</p>

<p style="margin-top: 1em">-aau Amino Acid Usage (AAU)</p>

<p style="margin-top: 1em">-raau Relative Amino Acid Usage
(RAAU)</p>

<p style="margin-top: 1em">-cu Codon Usage (CU)
(default)</p>

<p style="margin-top: 1em">-cutab Tabulation of codon
usage</p>

<p style="margin-top: 1em">-cutot Tabulation of
dataset&rsquo;s codon usage</p>

<p style="margin-top: 1em">-rscu Relative Synonymous Codon
Usage (RSCU)</p>

<p style="margin-top: 1em">-fasta fasta format</p>

<p style="margin-top: 1em">-tidy fasta format</p>

<p style="margin-top: 1em">-reader <br>
Reader format (codons are separated by spaces)</p>

<p style="margin-top: 1em">-transl <br>
Conceptual translation of DNA to amino acid</p>

<p style="margin-top: 1em">-base Detailed report of codon
G+C composition</p>

<p style="margin-top: 1em">-dinuc Dinucleotide usage of the
three codon pos.</p>

<p style="margin-top: 1em">-noblk No bulk output to be
written to file</p>

<p style="margin-top: 1em">Where {file} represents an input
filename, and N an integer value Controlled exit
&lt;&gt;</p>

<p style="margin-top: 1em">codonw 1.4.4 February 2016
CODONW(1)</p>
<hr>
</body>
</html>
