<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CODONW(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CODONW(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">CODONW(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
codonw - Correspondence Analysis of Codon Usage
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
codonW [inputfile] [outputfile] [bulkoutfile] [options] General options and
  defaults:
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>(elp)</dt>
  <dd class="It-tag">This help message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nomenu</b></dt>
  <dd class="It-tag">Prevent the menu interface being displayed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nowarn</b></dt>
  <dd class="It-tag">Prevent warnings about sequences being displayed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-silent</b></dt>
  <dd class="It-tag">Overwrite files silently</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-totals</b></dt>
  <dd class="It-tag">Concatenate all genes in inputfile</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-machine</b></dt>
  <dd class="It-tag">Machine readable output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-human</b></dt>
  <dd class="It-tag">Human readable output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-code</b> N</dt>
  <dd class="It-tag">Genetic code as defined under menu 3 option 5</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f_type</b> N</dt>
  <dd class="It-tag">Fop/CBI codons as defined by menu 3 option 6</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c_type</b> N</dt>
  <dd class="It-tag">Cai fitness values as defined by menu 3 option 7</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> (char)</dt>
  <dd class="It-tag">Column separator to be used in output files
      (comma,tab,space)</dd>
</dl>
<div class="Pp"></div>
Codon usage indices and Amino acid indices
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cai</b></dt>
  <dd class="It-tag">calculate Codon Adaptation Index (CAI)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-fop</b></dt>
  <dd class="It-tag">calculate Frequency of OPtimal codons index (FOP)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cbi</b></dt>
  <dd class="It-tag">calculate Codon Bias Index (CBI)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-enc</b></dt>
  <dd class="It-tag">Effective Number of Codons (ENc)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gc</b></dt>
  <dd class="It-tag">G+C content of gene (all 3 codon positions)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gcs3</b></dt>
  <dd class="It-tag">GC of synonymous codons 3rd positions</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sil_base</b></dt>
  <dd class="It-tag">Base composition at synonymous third codon positions</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L_sym</b></dt>
  <dd class="It-tag">Number of synonymous codons</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L_aa</b></dt>
  <dd class="It-tag">Total number of synonymous and non-synonymous codons</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-all_indices</b></dt>
  <dd class="It-tag">All the above indices</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-aro</b></dt>
  <dd class="It-tag">Calculate aromaticity of protein</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-hyd</b></dt>
  <dd class="It-tag">Calculate hydropathicity of protein</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cai_file</b></dt>
  <dd class="It-tag">{file} User input file of CAI values</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cbi_file</b></dt>
  <dd class="It-tag">{file} User input file of CBI values</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-fop_file</b></dt>
  <dd class="It-tag">{file} User input file of Fop values</dd>
</dl>
<div class="Pp"></div>
Correspondence analysis (COA) options
<dl class="Bl-tag">
  <dt class="It-tag"><b>-coa_cu</b></dt>
  <dd class="It-tag">COA of codon usage frequencies</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-coa_rscu</b></dt>
  <dd class="It-tag">COA of Relative Synonymous Codon Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-coa_aa</b></dt>
  <dd class="It-tag">COA of amino acid usage frequencies</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-coa_expert</b></dt>
  <dd class="It-tag">Generate detailed(expert) statistics on COA</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-coa_axes</b> N</dt>
  <dd class="It-tag">Select number of axis to record</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-coa_num</b> N</dt>
  <dd class="It-tag">Select number of genes to use to identify optimal codons
      values can be whole numbers or a percentage (5 or 10%)</dd>
</dl>
<div class="Pp"></div>
Bulk output options | only one can be selected per analysis
<dl class="Bl-tag">
  <dt class="It-tag"><b>-aau</b></dt>
  <dd class="It-tag">Amino Acid Usage (AAU)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-raau</b></dt>
  <dd class="It-tag">Relative Amino Acid Usage (RAAU)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cu</b></dt>
  <dd class="It-tag">Codon Usage (CU) (default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cutab</b></dt>
  <dd class="It-tag">Tabulation of codon usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cutot</b></dt>
  <dd class="It-tag">Tabulation of dataset's codon usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rscu</b></dt>
  <dd class="It-tag">Relative Synonymous Codon Usage (RSCU)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-fasta</b></dt>
  <dd class="It-tag">fasta format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-tidy</b></dt>
  <dd class="It-tag">fasta format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-reader</b></dt>
  <dd class="It-tag">Reader format (codons are separated by spaces)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-transl</b></dt>
  <dd class="It-tag">Conceptual translation of DNA to amino acid</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-base</b></dt>
  <dd class="It-tag">Detailed report of codon G+C composition</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-dinuc</b></dt>
  <dd class="It-tag">Dinucleotide usage of the three codon pos.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-noblk</b></dt>
  <dd class="It-tag">No bulk output to be written to file</dd>
</dl>
<div class="Pp"></div>
Where {file} represents an input filename, and N an integer value Controlled
  exit &lt;&gt;</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2016</td>
    <td class="foot-os">codonw 1.4.4</td>
  </tr>
</table>
</body>
</html>
