<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:00:02 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>COMPSEQ(1e) EMBOSS Manual for Debian COMPSEQ(1e)</p>

<p style="margin-top: 1em">NAME <br>
compseq - Calculate the composition of unique words in
sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
compseq -sequence seqall [-infile infile] -word integer
[-frame integer] -ignorebz boolean -reverse boolean
[-calcfreq boolean] -outfile outfile [-zerocount
boolean]</p>

<p style="margin-top: 1em">compseq -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
compseq is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the
&quot;Nucleic:Composition,Protein:Composition&quot; command
<br>
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall</p>

<p style="margin-top: 1em">-infile infile <br>
This is a file previously produced by &rsquo;compseq&rsquo;
that can be used to set the expected frequencies of words in
this analysis. The word size in the current run must be the
same <br>
as the one in this results file. Obviously, you should use a
file produced from protein sequences if you are counting
protein sequence word frequencies, and you must use one <br>
made from nucleotide frequencies if you are analysing a
nucleotide sequence.</p>

<p style="margin-top: 1em">Required section <br>
-word integer <br>
This is the size of word (n-mer) to count. Thus if you want
to count codon frequencies for a nucleotide sequence, you
should enter 3 here. Default value: 2</p>

<p style="margin-top: 1em">Additional section <br>
-frame integer <br>
The normal behaviour of &rsquo;compseq&rsquo; is to count
the frequencies of all words that occur by moving a window
of length &rsquo;word&rsquo; up by one each time. This
option allows you to move <br>
the window up by the length of the word each time, skipping
over the intervening words. You can count only those words
that occur in a single frame of the word by setting <br>
this value to a number other than zero. If you set it to 1
it will only count the words in frame 1, 2 will only count
the words in frame 2 and so on.</p>

<p style="margin-top: 1em">-ignorebz boolean <br>
The amino acid code B represents Asparagine or Aspartic acid
and the code Z represents Glutamine or Glutamic acid. These
are not commonly used codes and you may wish not to <br>
count words containing them, just noting them in the count
of &rsquo;Other&rsquo; words. Default value: Y</p>

<p style="margin-top: 1em">-reverse boolean <br>
Set this to be true if you also wish to also count words in
the reverse complement of a nucleic sequence. Default value:
N</p>

<p style="margin-top: 1em">-calcfreq boolean <br>
If this is set true then the expected frequencies of words
are calculated from the observed frequency of single bases
or residues in the sequences. If you are reporting a <br>
word size of 1 (single bases or residues) then there is no
point in using this option because the calculated expected
frequency will be equal to the observed frequency. <br>
Calculating the expected frequencies like this will give an
approximation of the expected frequencies that you might get
by using an input file of frequencies produced by a <br>
previous run of this program. If an input file of expected
word frequencies has been specified then the values from
that file will be used instead of this calculation of <br>
expected frequency from the sequence, even if
&rsquo;calcfreq&rsquo; is set to be true. Default value:
N</p>

<p style="margin-top: 1em">Output section <br>
-outfile outfile <br>
This is the results file.</p>

<p style="margin-top: 1em">-zerocount boolean <br>
You can make the output results file much smaller if you do
not display the words with a zero count. Default value:
Y</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
compseq is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
COMPSEQ(1e)</p>
<hr>
</body>
</html>
