<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>crac(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">crac(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">crac(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
crac v. 2.5.0 - Crac is a tool to analyse RNA-Seq data provided by NGS.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>crac [ options ] <b>-i</b> &lt;index_file&gt; <b>-r</b> &lt;reads_file1&gt;
  [reads_file2]</b> <b>-k</b> &lt;int&gt; <b>-o</b> &lt;output_file&gt;
<div class="Pp"></div>
<b>crac <b>-h</b>|<b>--help</b></b>
<div>&#x00A0;</div>
<b>crac <b>-f</b>|<b>--full-help</b></b>
<div>&#x00A0;</div>
<b>crac <b>-v</b>|<b>--version</b></b> 		
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>crac</b> CRAC: an integrated approach to the analysis of RNA-seq reads
<div class="Pp"></div>
Whatever the biological questions it addresses, each RNA-seq analysis requires a
  computational prediction of either small scale mutations, indels, splice
  junctions or fusion RNAs. This prediction is currently performed using complex
  pipelines involving multiple tools for mapping, coverage computation, and
  prediction at distinct steps. We propose a novel way of analyzing reads that
  integrates genomic locations and local coverage, and delivers all above
  mentioned predictions in a single step. Our program, CRAC, uses a double k-mer
  profiling approach to detect candidate mutations, indels, splice or fusion
  junctions in each single read. Compared to existing tools, CRAC provides state
  of the art sensitivity and improved precision for all types of predictions,
  yielding high rates of true positive candidates (99.5% for splice junctions).
  When applied to four breast cancer libraries, CRAC recovered 74% of validated
  fusion RNAs and predicted reccurrent fusion junctions that were overseen in
  previous studies. Importantly, CRAC improves its predictive performance when
  supplied with e.g. 200 nt reads and should fit future needs of read analyses.
<h1 class="Sh" title="Sh" id="SPECIAL_OPTIONS"><a class="selflink" href="#SPECIAL_OPTIONS">SPECIAL
  OPTIONS</a></h1>
<b>crac</b> As a lot of softwares, there are many optional parameters in CRAC
  but only three are mandatory. This document is intended to guide users of CRAC
  to choose the more appropriate parameters according to their needs:
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">to print the principal help page of CRAC</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>, <b>--full-help</b></dt>
  <dd class="It-tag">to print the complete help page of CRAC</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--version</b></dt>
  <dd class="It-tag">to print version of CRAC</dd>
</dl>
<h1 class="Sh" title="Sh" id="USUAL_ARGUMENTS"><a class="selflink" href="#USUAL_ARGUMENTS">USUAL
  ARGUMENTS</a></h1>
<h2 class="Ss" title="Ss" id="Mandatory_options"><a class="selflink" href="#Mandatory_options">Mandatory
  options</a></h2>
All these flags must be set.
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> <i>&lt;index_file&gt;</i></dt>
  <dd class="It-tag">is the name of the index previously built with the
      crac-index binary file. Note that crac-index construct the structure
      &lt;index_file.ssa&gt; with its configuration &lt;index_file.conf&gt; so
      only the prefix &lt;index_file&gt; must be specified (without extension)
      to consider the structure and the configuration files both in CRAC</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> <i>&lt;reads_file1&gt; [reads_file2]</i></dt>
  <dd class="It-tag">is the source(s) of the FASTA or FASTQ file(s) containing
      the reads. Note that the number of files depends if single or
      paired-reads. The input file may also be compressed using gzip</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the length of the k-mer to be used to map the reads on
      the reference &lt;index_file&gt;. Note that the condition (k &lt; m) is
      necessary and reads (or both paired reads) are ignored if m &lt; k. It
      must be chosen to ensure (as much as possible) that a k-mer has a very
      high probability to occur a single time on the genome</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>, <b>--sam</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">is the output file in SAM format (see the Documentation of
      SAM format in CRAC for more details) or print on STDOUT with &quot;-o
      -&quot; argument</dd>
</dl>
<h2 class="Ss" title="Ss" id="Optional_parameters"><a class="selflink" href="#Optional_parameters">Optional
  parameters</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stranded</b></dt>
  <dd class="It-tag">must be specificied if reads are produced by a stranded
      protocol of RNA-Seq (not stranded by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fr/--rf/--ff</b></dt>
  <dd class="It-tag">set the mates alignement orientation (--rf by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>, <b>--reads-length</b>, <b>-m</b>
    <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">must be specified for reads of fixed length. If the read
      length is fixed, we deeply recommend you to specify the read length, by
      using the -m parameter. CRAC will therefore be much faster. --reads-length
      &lt;int&gt; is specified for variable or longer reads, reads shorter are
      ignored and reads longer are trimmed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--treat-multiple</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">display alignments with multiple locations (with a fixed
      limit) rather than a single alignment per read in the SAM file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nb-threads</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the number of threads to run crac, computational time is
      almost divided by the number of threads (one thread by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-locs</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">corresponds to the max number of occurrences retrieved in
      the index for a given k-mer: smaller is faster, but with a small value,
      you may miss some locations that would help CRAC detecting the right
    cause</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-ambiguity</b> <i>&lt;none&gt;</i></dt>
  <dd class="It-tag">discard biological events (splice, svn, indel, chimera)
      which have several matches on the reference index. Indeed, if crac has
      identified a biological cause in the read that can match in differents
      places of the genome we classify this cause as a biological undetermined
      event.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Optional_output_arguments"><a class="selflink" href="#Optional_output_arguments">Optional
  output arguments</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gz</b> <i>&lt;none&gt;</i></dt>
  <dd class="It-tag">all output files specified after this argument are gzipped
      (included for the sam file if -o/--sam argument is specified after)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bam</b> <i>&lt;none&gt;</i></dt>
  <dd class="It-tag">sam output is encode in binary format(BAM)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--summary</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save some statistics about mapping and classification</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--show-progressbar</b> <i>&lt;none&gt;</i></dt>
  <dd class="It-tag">show a progress bar for the process times on STDERR</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--use-x-in-cigar</b> <i>&lt;none&gt;</i></dt>
  <dd class="It-tag">use X cigar operator when CRAC identifies a mismatch</dd>
</dl>
<h2 class="Ss" title="Ss" id="Optional_output_homemade_file_formats"><a class="selflink" href="#Optional_output_homemade_file_formats">Optional
  output homemade file formats</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--all</b> <i>&lt;base_filename&gt;</i></dt>
  <dd class="It-tag">set output base filename for all causes following. Note
      that only a base_filename must be specified. Then, the appropriate file
      extension is added for each cause (SNP, chimera, splice, etc) set output
      base filename for all causes following</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--normal</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that do not contain any break</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--almost-normal</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that do not contain any break but with a
      variable support</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--single</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads which are located in this way: at least
      --min-percent-single-loc &lt;float&gt; of k-mers are once located on the
      reference index</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--duplicate</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads which are located in this way: at least
      --min-percent-duplication-loc &lt;float&gt; of k-mers are a few times on
      the reference index (ie. between --min-duplication &lt;int&gt; and
      --max-duplication &lt;int&gt; of locations)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--multiple</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads which are located in this way: at least
      --min-percent-multiple-loc &lt;float&gt; of k-mers are a many times on the
      reference index (ie. more than --max-duplication &lt;int&gt; of
    locations)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--none</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">
    <br/>
     save reads which are not located on the reference index</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--snv</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that contain at least a snv</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--indel</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that contain at least a biological indel</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--splice</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that contain at least a splicing junction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--weak-splice</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that contain at least a low coverage splicing
      junction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--chimera</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that contain at least a chimera junction
      (junction on different chromosomes, strands or genes)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--paired-end-chimera</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">paired-end-chimera &lt;output_file&gt;= save paired-end
      reads that contains a chimera in the non-sequenced part of the original
      fragment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--biological</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that contain a biological cause but for which
      there is not enough informations to be more specific. Note that the
      biological cause is described for each read</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--errors</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that contain at least a sequence error</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--repeat</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">
    <br/>
     save reads that contain a repeated sequence: at least
    <br/>
     --min-percent-repetition-loc &lt;float&gt; percent of k-mers of a given
    <br/>
     read are located at least --min-repetition &lt;int&gt; occurrences on the
    <br/>
     reference index</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--undetermined</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that contain an undetermined error: some k-mers
      are not located on the genome, but the reason for that could not be
      determined. Note that the error is described for each read</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nothing</b> <i>&lt;output_file&gt;</i></dt>
  <dd class="It-tag">save reads that are unclassified</dd>
</dl>
<h2 class="Ss" title="Ss" id="Optional_process_for_specific_research"><a class="selflink" href="#Optional_process_for_specific_research">Optional
  process for specific research</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--deep-snv</b></dt>
  <dd class="It-tag">must be specified to increase sensitivity to find SNVs at
      the cost of more computations (only substitution, no indels YET). That
      process searches for SNV in border cases reads. Those reads would
      otherwise be classified in bioundetermined</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stringent-chimera</b></dt>
  <dd class="It-tag">must be specified to increase accuracy to find chimera
      junctions in exchange of sensitivity and computational times</dd>
</dl>
<h2 class="Ss" title="Ss" id="Optional_process_launcher_(once_must_be_selected)"><a class="selflink" href="#Optional_process_launcher_(once_must_be_selected)">Optional
  process launcher (once must be selected)</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--emt</b></dt>
  <dd class="It-tag">launch an exact matching processing of reads on the index.
      Either the argument specified with -k is equal to 0 which means that the
      entire read is perfectly mapped on the genome or only a factor of length k
      per read is mapped (the first one with a location) and the rest is
      sofclipped. With this process, reads are not indexed and it provides a low
      memory consumption. Note this kind of method is very useful for DGE reads
      mapping.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--server</b></dt>
  <dd class="It-tag">launch a server to query a given read more precisely. That
      process is useful for debugging. Note that the output arguments will not
      be taken into account. Give an --input-name-server &lt;string&gt; to set
      the input fifo name (classify.fifo by default) and give an
      --output-name-server &lt;string&gt; to set the output fifo name
      (classify.out.fifo by default). The server can then be used through a
      client crac-client</dd>
</dl>
<h2 class="Ss" title="Ss" id="Additional_settings_for_users"><a class="selflink" href="#Additional_settings_for_users">Additional
  settings for users</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--detailed-sam</b></dt>
  <dd class="It-tag">more informations are added in SAM output file. See the
      Documentation of SAM format in CRAC for more details</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-percent-single-loc</b> <i>&lt;float&gt;</i></dt>
  <dd class="It-tag">is, to consider a given read as uniquely mapped, the
      minimum proportion of k-mers that are uniquely mapped on the index (0.15
      by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-duplication</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the minimum number of location to consider a duplicated
      k-mer (2 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-duplication</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the maximum number of location to consider a duplicated
      k-mer (9 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-percent-duplication-loc</b>
    <i>&lt;float&gt;</i></dt>
  <dd class="It-tag">is, to consider a given read as duplicated, the minimum
      proportion of k-mers that are duplicated on the index (0.15 by
    default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-percent-multiple-loc</b> <i>&lt;float&gt;</i></dt>
  <dd class="It-tag">is, to consider a given read as &#x201C;multiple&#x201D;,
      the minimum proportion of k-mers that are multiple mapped on the index
      (0.50 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-repetition</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the minimum number of locations to consider a repeated
      k-mer (20 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-percent-repetition-loc</b>
    <i>&lt;float&gt;</i></dt>
  <dd class="It-tag">is, for a given read, the minimum proportion of k-mers that
      are repeated on the index to consider a repetition (0.20 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-splice-length</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the threshold to consider a splice, ie. a splice is
      reported if the junction length is below max-splice-length &lt;int&gt;, a
      chimera is considered otherwise (distance by default is 300Kb)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-bio-indel</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the threshold to consider a biological indel, ie. an
      indel is reported if the gap length is below max-bio-indel, a splice is
      considered otherwise (distance by default is 15)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-bases-retrieved</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the number of nucleotides to display in outputfile in
      case of insertion (15 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-support-no-cover</b> <i>&lt;float&gt;</i></dt>
  <dd class="It-tag">is the minimum coverage to be able to report a biological
      cause. Note that if a single read contains a given substitution, it is
      difficult (if not impossible) to distinguish a sequence error and a
      biological cause (1.30 by default)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Additional_settings_for_advanced_users"><a class="selflink" href="#Additional_settings_for_advanced_users">Additional
  settings for advanced users</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-break-length</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the minimal break length (as the percentage of k, the
      k-mer length) so that a cause can be reported. Theoretically, for a given
      cause, the break length is always &gt;= (kmer_length - 1). Otherwise, the
      break may be merged with a close enough break, or the break will be
      considered as undetermined. (0.5 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-bases-randomly-matched</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">A k-mer overlapping an exon-exon junction, for example, may
      still match on the genome if the overlap is at the end of the read
      (without loss of generality). This is due to the fact that the nucleotides
      starting the second exon may be the same as the nucleotides starting the
      intron. Theoretically, there is a 0.25 probability that we have the same
      nucleotide at the first position of the intron and the exon. This option
      specifies how many nucleotides may be matched randomly at most</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-extension-length</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is the maximum number of k-mers extended at each side of a
      read break. In fact, for a given break, k-mers with false locations can
      generate false biological causes, so the consistency is checked for each
      side of the break to discard false k-mers and readjust the good boundaries
      of the break (10 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nb-tags-info-stored</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">is a buffer to store informations for each thread during
      the computing phase (1000 by default). This value must be increased if
      threads work below their real capabilities. With --nb-threads 15, CPU
      usage must be about 1400%</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--reads-index</b> <i>&lt;string&gt;</i></dt>
  <dd class="It-tag">the reads index data-structure uses by CRAC. Available
      reads index are: JELLYFISH and GKARRAYS. (JELLYFISH by default).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nb-nucleotides-snp-comparison</b>
    <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">
    <br/>
     is the minimum k-mer length tolerated for the deep SNVs search (8 by
    <br/>
     default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-number-of-merges</b> <i>&lt;int&gt;</i></dt>
  <dd class="It-tag">
    <br/>
     is the maximum number of merges tolerated during the break merge process
    <br/>
     for the chimera detection (4 by default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-score-chimera-stringent</b>
    <i>&lt;float&gt;</i></dt>
  <dd class="It-tag">
    <br/>
     is the mimimal score to consider a chimera event
    <br/>
     otherwise it is classify as a bioundetermined event (0.6 by default)
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The full documentation for <b>crac</b> is maintained as a org manual. If the
  <b>info</b> and <b>crac</b> programs are properly installed at your site, the
  command
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>info crac</b></dd>
</dl>
<div class="Pp"></div>
should give you access to the complete manual.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<h2 class="Ss" title="Ss" id="About_the_crac_package."><a class="selflink" href="#About_the_crac_package.">About
  the crac package.</a></h2>
You can contact Nicolas PHILIPPE, Mikael SALSON, Jerome AUDOUX and Alban
  MANCHERON by sending an e-mail to &lt;crac-bugs@lists.gforge.inria.fr&gt;.
<dl class="Bl-tag">
  <dt class="It-tag">Programming:</dt>
  <dd class="It-tag">
    <br/>
     Nicolas PHILIPPE &lt;nphilippe.research@gmail.com&gt;
    <br/>
     Mika&#x00EB;l SALSON &lt;mikael.salson@lifl.fr&gt; 	J&#x00E9;rome AUDOUX
      &lt;jerome.audoux@gmail.com&gt;
    <div>&#x00A0;</div>
    with additional contribution for the packaging of:
    <br/>
     Alban MANCHERON &lt;alban.mancheron@lirmm.fr&gt;</dd>
</dl>
<h2 class="Ss" title="Ss" id="About_the_crac_publication."><a class="selflink" href="#About_the_crac_publication.">About
  the crac publication.</a></h2>
You may cite the following paper if you use our tool:
<dl class="Bl-tag">
  <dt class="It-tag">Gk-arrays: Querying large read collections in main memory:
    a versatile</dt>
  <dd class="It-tag">data structure
    <div>&#x00A0;</div>
    <b>Philippe N., Salson M., Lecroq T., Leonard M., Commes T., Rivals E.</b>
    <div>&#x00A0;</div>
    BMC Bioinformatics 2011, 12:242.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Crac: An integrated RNA-Seq read analysis</dt>
  <dd class="It-tag">
    <div>&#x00A0;</div>
    <b>Philippe N., Salson M., Commes T., Rivals E.</b>
    <div>&#x00A0;</div>
    Genome Biology 2013; 14:R30.</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-11-09</td>
    <td class="foot-os"></td>
  </tr>
</table>
</body>
</html>
