<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:00:52 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p style="margin-top: 1em">crac(1) General Commands Manual
crac(1)</p>

<p style="margin-top: 1em">NAME <br>
crac v. 2.5.0 - Crac is a tool to analyse RNA-Seq data
provided by NGS.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
crac [ options ] -i &lt;index_file&gt; -r
&lt;reads_file1&gt; [reads_file2] -k &lt;int&gt; -o
&lt;output_file&gt;</p>

<p style="margin-top: 1em">crac -h|--help <br>
crac -f|--full-help <br>
crac -v|--version</p>

<p style="margin-top: 1em">DESCRIPTION <br>
crac CRAC: an integrated approach to the analysis of RNA-seq
reads</p>

<p style="margin-top: 1em">Whatever the biological
questions it addresses, each RNA-seq analysis requires a
computational prediction of either small scale mutations,
indels, splice junctions or fusion <br>
RNAs. This prediction is currently performed using complex
pipelines involving multiple tools for mapping, coverage
computation, and prediction at distinct steps. We propose a
<br>
novel way of analyzing reads that integrates genomic
locations and local coverage, and delivers all above
mentioned predictions in a single step. Our program, CRAC,
uses a double <br>
k-mer profiling approach to detect candidate mutations,
indels, splice or fusion junctions in each single read.
Compared to existing tools, CRAC provides state of the art
sensi&acirc; <br>
tivity and improved precision for all types of predictions,
yielding high rates of true positive candidates (99.5% for
splice junctions). When applied to four breast cancer <br>
libraries, CRAC recovered 74% of validated fusion RNAs and
predicted reccurrent fusion junctions that were overseen in
previous studies. Importantly, CRAC improves its predictive
<br>
performance when supplied with e.g. 200 nt reads and should
fit future needs of read analyses.</p>

<p style="margin-top: 1em">SPECIAL OPTIONS <br>
crac As a lot of softwares, there are many optional
parameters in CRAC but only three are mandatory. This
document is intended to guide users of CRAC to choose the
more appropri&acirc; <br>
ate parameters according to their needs:</p>

<p style="margin-top: 1em">-h, --help <br>
to print the principal help page of CRAC</p>

<p style="margin-top: 1em">-f, --full-help <br>
to print the complete help page of CRAC</p>

<p style="margin-top: 1em">-v, --version <br>
to print version of CRAC</p>

<p style="margin-top: 1em">USUAL ARGUMENTS <br>
Mandatory options <br>
All these flags must be set.</p>

<p style="margin-top: 1em">-i &lt;index_file&gt; <br>
is the name of the index previously built with the
crac-index binary file. Note that crac-index construct the
structure &lt;index_file.ssa&gt; with its configuration <br>
&lt;index_file.conf&gt; so only the prefix
&lt;index_file&gt; must be specified (without extension) to
consider the structure and the configuration files both in
CRAC</p>

<p style="margin-top: 1em">-r &lt;reads_file1&gt;
[reads_file2] <br>
is the source(s) of the FASTA or FASTQ file(s) containing
the reads. Note that the number of files depends if single
or paired-reads. The input file may also be compressed <br>
using gzip</p>

<p style="margin-top: 1em">-k &lt;int&gt; <br>
is the length of the k-mer to be used to map the reads on
the reference &lt;index_file&gt;. Note that the condition (k
&lt; m) is necessary and reads (or both paired reads) are
<br>
ignored if m &lt; k. It must be chosen to ensure (as much as
possible) that a k-mer has a very high probability to occur
a single time on the genome</p>

<p style="margin-top: 1em">-o, --sam &lt;output_file&gt;
<br>
is the output file in SAM format (see the Documentation of
SAM format in CRAC for more details) or print on STDOUT with
&quot;-o -&quot; argument</p>

<p style="margin-top: 1em">Optional parameters <br>
--stranded <br>
must be specificied if reads are produced by a stranded
protocol of RNA-Seq (not stranded by default)</p>

<p style="margin-top: 1em">--fr/--rf/--ff <br>
set the mates alignement orientation (--rf by default)</p>

<p style="margin-top: 1em">-m, --reads-length, -m
&lt;int&gt; <br>
must be specified for reads of fixed length. If the read
length is fixed, we deeply recommend you to specify the read
length, by using the -m parameter. CRAC will there&acirc;
<br>
fore be much faster. --reads-length &lt;int&gt; is specified
for variable or longer reads, reads shorter are ignored and
reads longer are trimmed</p>

<p style="margin-top: 1em">--treat-multiple &lt;int&gt;
<br>
display alignments with multiple locations (with a fixed
limit) rather than a single alignment per read in the SAM
file</p>

<p style="margin-top: 1em">--nb-threads &lt;int&gt; <br>
is the number of threads to run crac, computational time is
almost divided by the number of threads (one thread by
default)</p>

<p style="margin-top: 1em">--max-locs &lt;int&gt; <br>
corresponds to the max number of occurrences retrieved in
the index for a given k-mer: smaller is faster, but with a
small value, you may miss some locations that would <br>
help CRAC detecting the right cause</p>

<p style="margin-top: 1em">--no-ambiguity &lt;none&gt; <br>
discard biological events (splice, svn, indel, chimera)
which have several matches on the reference index. Indeed,
if crac has identified a biological cause in the read <br>
that can match in differents places of the genome we
classify this cause as a biological undetermined event.</p>

<p style="margin-top: 1em">Optional output arguments <br>
--gz &lt;none&gt; <br>
all output files specified after this argument are gzipped
(included for the sam file if -o/--sam argument is specified
after)</p>

<p style="margin-top: 1em">--bam &lt;none&gt; <br>
sam output is encode in binary format(BAM)</p>

<p style="margin-top: 1em">--summary &lt;output_file&gt;
<br>
save some statistics about mapping and classification</p>

<p style="margin-top: 1em">--show-progressbar &lt;none&gt;
<br>
show a progress bar for the process times on STDERR</p>

<p style="margin-top: 1em">--use-x-in-cigar &lt;none&gt;
<br>
use X cigar operator when CRAC identifies a mismatch</p>

<p style="margin-top: 1em">Optional output homemade file
formats <br>
--all &lt;base_filename&gt; <br>
set output base filename for all causes following. Note that
only a base_filename must be specified. Then, the
appropriate file extension is added for each cause (SNP,
<br>
chimera, splice, etc) set output base filename for all
causes following</p>

<p style="margin-top: 1em">--normal &lt;output_file&gt;
<br>
save reads that do not contain any break</p>

<p style="margin-top: 1em">--almost-normal
&lt;output_file&gt; <br>
save reads that do not contain any break but with a variable
support</p>

<p style="margin-top: 1em">--single &lt;output_file&gt;
<br>
save reads which are located in this way: at least
--min-percent-single-loc &lt;float&gt; of k-mers are once
located on the reference index</p>

<p style="margin-top: 1em">--duplicate &lt;output_file&gt;
<br>
save reads which are located in this way: at least
--min-percent-duplication-loc &lt;float&gt; of k-mers are a
few times on the reference index (ie. between
--min-duplication <br>
&lt;int&gt; and --max-duplication &lt;int&gt; of
locations)</p>

<p style="margin-top: 1em">--multiple &lt;output_file&gt;
<br>
save reads which are located in this way: at least
--min-percent-multiple-loc &lt;float&gt; of k-mers are a
many times on the reference index (ie. more than
--max-duplication <br>
&lt;int&gt; of locations)</p>

<p style="margin-top: 1em">--none &lt;output_file&gt; <br>
save reads which are not located on the reference index</p>

<p style="margin-top: 1em">--snv &lt;output_file&gt; <br>
save reads that contain at least a snv</p>

<p style="margin-top: 1em">--indel &lt;output_file&gt; <br>
save reads that contain at least a biological indel</p>

<p style="margin-top: 1em">--splice &lt;output_file&gt;
<br>
save reads that contain at least a splicing junction</p>

<p style="margin-top: 1em">--weak-splice
&lt;output_file&gt; <br>
save reads that contain at least a low coverage splicing
junction</p>

<p style="margin-top: 1em">--chimera &lt;output_file&gt;
<br>
save reads that contain at least a chimera junction
(junction on different chromosomes, strands or genes)</p>

<p style="margin-top: 1em">--paired-end-chimera
&lt;output_file&gt; <br>
paired-end-chimera &lt;output_file&gt;= save paired-end
reads that contains a chimera in the non-sequenced part of
the original fragment.</p>

<p style="margin-top: 1em">--biological &lt;output_file&gt;
<br>
save reads that contain a biological cause but for which
there is not enough informations to be more specific. Note
that the biological cause is described for each read</p>

<p style="margin-top: 1em">--errors &lt;output_file&gt;
<br>
save reads that contain at least a sequence error</p>

<p style="margin-top: 1em">--repeat &lt;output_file&gt;
<br>
save reads that contain a repeated sequence: at least <br>
--min-percent-repetition-loc &lt;float&gt; percent of k-mers
of a given <br>
read are located at least --min-repetition &lt;int&gt;
occurrences on the <br>
reference index</p>

<p style="margin-top: 1em">--undetermined
&lt;output_file&gt; <br>
save reads that contain an undetermined error: some k-mers
are not located on the genome, but the reason for that could
not be determined. Note that the error is described <br>
for each read</p>

<p style="margin-top: 1em">--nothing &lt;output_file&gt;
<br>
save reads that are unclassified</p>

<p style="margin-top: 1em">Optional process for specific
research <br>
--deep-snv <br>
must be specified to increase sensitivity to find SNVs at
the cost of more computations (only substitution, no indels
YET). That process searches for SNV in border cases <br>
reads. Those reads would otherwise be classified in
bioundetermined</p>

<p style="margin-top: 1em">--stringent-chimera <br>
must be specified to increase accuracy to find chimera
junctions in exchange of sensitivity and computational
times</p>

<p style="margin-top: 1em">Optional process launcher (once
must be selected) <br>
--emt launch an exact matching processing of reads on the
index. Either the argument specified with -k is equal to 0
which means that the entire read is perfectly mapped on the
<br>
genome or only a factor of length k per read is mapped (the
first one with a location) and the rest is sofclipped. With
this process, reads are not indexed and it provides <br>
a low memory consumption. Note this kind of method is very
useful for DGE reads mapping.</p>

<p style="margin-top: 1em">--server <br>
launch a server to query a given read more precisely. That
process is useful for debugging. Note that the output
arguments will not be taken into account. Give an <br>
--input-name-server &lt;string&gt; to set the input fifo
name (classify.fifo by default) and give an
--output-name-server &lt;string&gt; to set the output fifo
name (classify.out.fifo <br>
by default). The server can then be used through a client
crac-client</p>

<p style="margin-top: 1em">Additional settings for users
<br>
--detailed-sam <br>
more informations are added in SAM output file. See the
Documentation of SAM format in CRAC for more details</p>

<p style="margin-top: 1em">--min-percent-single-loc
&lt;float&gt; <br>
is, to consider a given read as uniquely mapped, the minimum
proportion of k-mers that are uniquely mapped on the index
(0.15 by default)</p>

<p style="margin-top: 1em">--min-duplication &lt;int&gt;
<br>
is the minimum number of location to consider a duplicated
k-mer (2 by default)</p>

<p style="margin-top: 1em">--max-duplication &lt;int&gt;
<br>
is the maximum number of location to consider a duplicated
k-mer (9 by default)</p>

<p style="margin-top: 1em">--min-percent-duplication-loc
&lt;float&gt; <br>
is, to consider a given read as duplicated, the minimum
proportion of k-mers that are duplicated on the index (0.15
by default)</p>

<p style="margin-top: 1em">--min-percent-multiple-loc
&lt;float&gt; <br>
is, to consider a given read as &acirc;multiple&acirc;, the
minimum proportion of k-mers that are multiple mapped on the
index (0.50 by default)</p>

<p style="margin-top: 1em">--min-repetition &lt;int&gt;
<br>
is the minimum number of locations to consider a repeated
k-mer (20 by default)</p>

<p style="margin-top: 1em">--max-percent-repetition-loc
&lt;float&gt; <br>
is, for a given read, the minimum proportion of k-mers that
are repeated on the index to consider a repetition (0.20 by
default)</p>

<p style="margin-top: 1em">--max-splice-length &lt;int&gt;
<br>
is the threshold to consider a splice, ie. a splice is
reported if the junction length is below max-splice-length
&lt;int&gt;, a chimera is considered otherwise (distance by
<br>
default is 300Kb)</p>

<p style="margin-top: 1em">--max-bio-indel &lt;int&gt; <br>
is the threshold to consider a biological indel, ie. an
indel is reported if the gap length is below max-bio-indel,
a splice is considered otherwise (distance by default <br>
is 15)</p>

<p style="margin-top: 1em">--max-bases-retrieved
&lt;int&gt; <br>
is the number of nucleotides to display in outputfile in
case of insertion (15 by default)</p>

<p style="margin-top: 1em">--min-support-no-cover
&lt;float&gt; <br>
is the minimum coverage to be able to report a biological
cause. Note that if a single read contains a given
substitution, it is difficult (if not impossible) to
distin&acirc; <br>
guish a sequence error and a biological cause (1.30 by
default)</p>

<p style="margin-top: 1em">Additional settings for advanced
users <br>
--min-break-length &lt;int&gt; <br>
is the minimal break length (as the percentage of k, the
k-mer length) so that a cause can be reported.
Theoretically, for a given cause, the break length is always
&gt;= <br>
(kmer_length - 1). Otherwise, the break may be merged with a
close enough break, or the break will be considered as
undetermined. (0.5 by default)</p>

<p style="margin-top: 1em">--max-bases-randomly-matched
&lt;int&gt; <br>
A k-mer overlapping an exon-exon junction, for example, may
still match on the genome if the overlap is at the end of
the read (without loss of generality). This is due to <br>
the fact that the nucleotides starting the second exon may
be the same as the nucleotides starting the intron.
Theoretically, there is a 0.25 probability that we have the
<br>
same nucleotide at the first position of the intron and the
exon. This option specifies how many nucleotides may be
matched randomly at most</p>

<p style="margin-top: 1em">--max-extension-length
&lt;int&gt; <br>
is the maximum number of k-mers extended at each side of a
read break. In fact, for a given break, k-mers with false
locations can generate false biological causes, so the <br>
consistency is checked for each side of the break to discard
false k-mers and readjust the good boundaries of the break
(10 by default)</p>

<p style="margin-top: 1em">--nb-tags-info-stored
&lt;int&gt; <br>
is a buffer to store informations for each thread during the
computing phase (1000 by default). This value must be
increased if threads work below their real capabilities.
<br>
With --nb-threads 15, CPU usage must be about 1400%</p>

<p style="margin-top: 1em">--reads-index &lt;string&gt;
<br>
the reads index data-structure uses by CRAC. Available reads
index are: JELLYFISH and GKARRAYS. (JELLYFISH by
default).</p>

<p style="margin-top: 1em">--nb-nucleotides-snp-comparison
&lt;int&gt; <br>
is the minimum k-mer length tolerated for the deep SNVs
search (8 by <br>
default)</p>

<p style="margin-top: 1em">--max-number-of-merges
&lt;int&gt; <br>
is the maximum number of merges tolerated during the break
merge process <br>
for the chimera detection (4 by default)</p>

<p style="margin-top: 1em">--min-score-chimera-stringent
&lt;float&gt; <br>
is the mimimal score to consider a chimera event <br>
otherwise it is classify as a bioundetermined event (0.6 by
default)</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation for crac is maintained as a org
manual. If the info and crac programs are properly installed
at your site, the command</p>

<p style="margin-top: 1em">info crac</p>

<p style="margin-top: 1em">should give you access to the
complete manual.</p>

<p style="margin-top: 1em">AUTHOR <br>
About the crac package. <br>
You can contact Nicolas PHILIPPE, Mikael SALSON, Jerome
AUDOUX and Alban MANCHERON by sending an e-mail to
&lt;crac-bugs@lists.gforge.inria.fr&gt;.</p>

<p style="margin-top: 1em">Programming: <br>
Nicolas PHILIPPE &lt;nphilippe.research@gmail.com&gt; <br>
Mika&Atilde;&laquo;l SALSON &lt;mikael.salson@lifl.fr&gt;
J&Atilde;&copy;rome AUDOUX &lt;jerome.audoux@gmail.com&gt;
<br>
with additional contribution for the packaging of: <br>
Alban MANCHERON &lt;alban.mancheron@lirmm.fr&gt;</p>

<p style="margin-top: 1em">About the crac publication. <br>
You may cite the following paper if you use our tool:</p>

<p style="margin-top: 1em">Gk-arrays: Querying large read
collections in main memory: a versatile <br>
data structure <br>
Philippe N., Salson M., Lecroq T., Leonard M., Commes T.,
Rivals E. <br>
BMC Bioinformatics 2011, 12:242.</p>

<p style="margin-top: 1em">Crac: An integrated RNA-Seq read
analysis <br>
Philippe N., Salson M., Commes T., Rivals E. <br>
Genome Biology 2013; 14:R30.</p>

<p style="margin-top: 1em">2016-11-09 crac(1)</p>
<hr>
</body>
</html>
