<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CREATE_MATRIX(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CREATE_MATRIX(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">CREATE_MATRIX(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
create_matrix - calculate the genome abundance similarity matrix
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>create_matrix</b> [ <i>options</i>] <i>NAMES</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Calculate the similarity matrix.
<div class="Pp"></div>
First, a set of reads is simulated for every reference genome using a read
  simulator from core/tools.py specified via <b>-s</b>. Second, the simulated
  reads of each species are mapped against all reference genomes using the
  mapper specified with <b>-m</b>. Third, the resulting SAM-files are analyzed
  to calculate the similarity matrix. The similarity matrix is stored as a numpy
  file ( <b>-o</b>).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>NAMES</b></dt>
  <dd class="It-tag">Filename of the names file; the plain text names file
      should contain one name per line. The name is used as identifier in the
      whole algorithm.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> SIMULATOR,
    <b>--simulator</b>=<i>SIMULATOR</i></dt>
  <dd class="It-tag">Identifier of read simulator defined in core/tools.py
      [default: none]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> REF, <b>--reference</b>=<i>REF</i></dt>
  <dd class="It-tag">Reference sequence file pattern for the read simulator.
      Placeholder for the name is &quot;%s&quot;. [default: ./ref/%s.fasta]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> MAPPER, <b>--mapper</b>=<i>MAPPER</i></dt>
  <dd class="It-tag">Identifier of mapper defined in core/tools.py [default:
      none]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> INDEX, <b>--index</b>=<i>INDEX</i></dt>
  <dd class="It-tag">Reference index files for the read mapper. Placeholder for
      the name is &quot;%s&quot;. [default: ./ref/%s.fasta]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> TEMP, <b>--temp</b>=<i>TEMP</i></dt>
  <dd class="It-tag">Directory to store temporary simulated datasets and SAM
      files. [default: ./temp]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> OUT, <b>--output</b>=<i>OUT</i></dt>
  <dd class="It-tag">Output similarity matrix file. [default:
      ./similarity_matrix.npy]</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2014</td>
    <td class="foot-os">create_matrix SVNr18</td>
  </tr>
</table>
</body>
</html>
