<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:00:57 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CREATE_MATRIX(1) User Commands CREATE_MATRIX(1)</p>

<p style="margin-top: 1em">NAME <br>
create_matrix - calculate the genome abundance similarity
matrix</p>

<p style="margin-top: 1em">SYNOPSIS <br>
create_matrix [options] NAMES</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Calculate the similarity matrix.</p>

<p style="margin-top: 1em">First, a set of reads is
simulated for every reference genome using a read simulator
from core/tools.py specified via -s. Second, the simulated
reads of each species are mapped <br>
against all reference genomes using the mapper specified
with -m. Third, the resulting SAM-files are analyzed to
calculate the similarity matrix. The similarity matrix is
stored <br>
as a numpy file (-o).</p>

<p style="margin-top: 1em">OPTIONS <br>
NAMES Filename of the names file; the plain text names file
should contain one name per line. The name is used as
identifier in the whole algorithm.</p>

<p style="margin-top: 1em">-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">-s SIMULATOR,
--simulator=SIMULATOR <br>
Identifier of read simulator defined in core/tools.py
[default: none]</p>

<p style="margin-top: 1em">-r REF, --reference=REF <br>
Reference sequence file pattern for the read simulator.
Placeholder for the name is &quot;%s&quot;. [default:
./ref/%s.fasta]</p>

<p style="margin-top: 1em">-m MAPPER, --mapper=MAPPER <br>
Identifier of mapper defined in core/tools.py [default:
none]</p>

<p style="margin-top: 1em">-i INDEX, --index=INDEX <br>
Reference index files for the read mapper. Placeholder for
the name is &quot;%s&quot;. [default: ./ref/%s.fasta]</p>

<p style="margin-top: 1em">-t TEMP, --temp=TEMP <br>
Directory to store temporary simulated datasets and SAM
files. [default: ./temp]</p>

<p style="margin-top: 1em">-o OUT, --output=OUT <br>
Output similarity matrix file. [default:
./similarity_matrix.npy]</p>

<p style="margin-top: 1em">create_matrix SVNr18 February
2014 CREATE_MATRIX(1)</p>
<hr>
</body>
</html>
