<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:01:40 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CUFFLINKS(1) User Commands CUFFLINKS(1)</p>

<p style="margin-top: 1em">NAME <br>
cufflinks - component of cufflinks suite</p>

<p style="margin-top: 1em">SYNOPSIS <br>
cufflinks [options] &lt;hits.sam&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
cufflinks v2.2.1 linked against Boost version 106200
-----------------------------</p>

<p style="margin-top: 1em">General Options: <br>
-o/--output-dir <br>
write all output files to this directory [ default: ./ ]</p>

<p style="margin-top: 1em">-p/--num-threads <br>
number of threads used during analysis [ default: 1 ]</p>

<p style="margin-top: 1em">--seed value of random number
generator seed [ default: 0 ]</p>

<p style="margin-top: 1em">-G/--GTF <br>
quantitate against reference transcript annotations</p>

<p style="margin-top: 1em">-g/--GTF-guide <br>
use reference transcript annotation to guide assembly</p>

<p style="margin-top: 1em">-M/--mask-file <br>
ignore all alignment within transcripts in this file</p>

<p style="margin-top: 1em">-b/--frag-bias-correct <br>
use bias correction - reference fasta required [ default:
NULL ]</p>

<p style="margin-top: 1em">-u/--multi-read-correct <br>
use &rsquo;rescue method&rsquo; for multi-reads (more
accurate) [ default: FALSE ]</p>

<p style="margin-top: 1em">--library-type <br>
library prep used for input reads [ default: below ]</p>

<p style="margin-top: 1em">--library-norm-method <br>
Method used to normalize library sizes [ default: below
]</p>

<p style="margin-top: 1em">Advanced Abundance Estimation
Options: <br>
-m/--frag-len-mean <br>
average fragment length (unpaired reads only) [ default: 200
]</p>

<p style="margin-top: 1em">-s/--frag-len-std-dev <br>
fragment length std deviation (unpaired reads only) [
default: 80 ]</p>

<p style="margin-top: 1em">--max-mle-iterations <br>
maximum iterations allowed for MLE calculation [ default:
5000 ]</p>

<p style="margin-top: 1em">--compatible-hits-norm <br>
count hits compatible with reference RNAs only [ default:
FALSE ]</p>

<p style="margin-top: 1em">--total-hits-norm <br>
count all hits for normalization [ default: TRUE ]</p>

<p style="margin-top: 1em">--num-frag-count-draws <br>
Number of fragment generation samples [ default: 100 ]</p>

<p style="margin-top: 1em">--num-frag-assign-draws <br>
Number of fragment assignment samples per generation [
default: 50 ]</p>

<p style="margin-top: 1em">--max-frag-multihits <br>
Maximum number of alignments allowed per fragment [ default:
unlim ]</p>


<p style="margin-top: 1em">--no-effective-length-correction
<br>
No effective length correction [ default: FALSE ]</p>

<p style="margin-top: 1em">--no-length-correction <br>
No length correction [ default: FALSE ]</p>

<p style="margin-top: 1em">-N/--upper-quartile-norm <br>
Deprecated, use --library-norm-method [ DEPRECATED ]</p>

<p style="margin-top: 1em">--raw-mapped-norm <br>
Deprecated, use --library-norm-method [ DEPRECATED ]</p>

<p style="margin-top: 1em">Advanced Assembly Options: <br>
-L/--label <br>
assembled transcripts have this ID prefix [ default: CUFF
]</p>

<p style="margin-top: 1em">-F/--min-isoform-fraction <br>
suppress transcripts below this abundance level [ default:
0.10 ]</p>

<p style="margin-top: 1em">-j/--pre-mrna-fraction <br>
suppress intra-intronic transcripts below this level [
default: 0.15 ]</p>

<p style="margin-top: 1em">-I/--max-intron-length <br>
ignore alignments with gaps longer than this [ default:
300000 ]</p>

<p style="margin-top: 1em">-a/--junc-alpha <br>
alpha for junction binomial test filter [ default: 0.001
]</p>

<p style="margin-top: 1em">-A/--small-anchor-fraction <br>
percent read overhang taken as &rsquo;suspiciously
small&rsquo; [ default: 0.09 ]</p>

<p style="margin-top: 1em">--min-frags-per-transfrag <br>
minimum number of fragments needed for new transfrags [
default: 10 ]</p>

<p style="margin-top: 1em">--overhang-tolerance <br>
number of terminal exon bp to tolerate in introns [ default:
8 ]</p>

<p style="margin-top: 1em">--max-bundle-length <br>
maximum genomic length allowed for a given bundle [
default:3500000 ]</p>

<p style="margin-top: 1em">--max-bundle-frags <br>
maximum fragments allowed in a bundle before skipping [
default: 500000 ]</p>

<p style="margin-top: 1em">--min-intron-length <br>
minimum intron size allowed in genome [ default: 50 ]</p>

<p style="margin-top: 1em">--trim-3-avgcov-thresh <br>
minimum avg coverage required to attempt 3&rsquo; trimming [
default: 10 ]</p>

<p style="margin-top: 1em">--trim-3-dropoff-frac <br>
fraction of avg coverage below which to trim 3&rsquo; end [
default: 0.1 ]</p>

<p style="margin-top: 1em">--max-multiread-fraction <br>
maximum fraction of allowed multireads per transcript [
default: 0.75 ]</p>

<p style="margin-top: 1em">--overlap-radius <br>
maximum gap size to fill between transfrags (in bp) [
default: 50 ]</p>

<p style="margin-top: 1em">Advanced Reference Annotation
Guided Assembly Options: <br>
--no-faux-reads <br>
disable tiling by faux reads [ default: FALSE ]</p>

<p style="margin-top: 1em">--3-overhang-tolerance <br>
overhang allowed on 3&rsquo; end when merging with
reference[ default: 600 ]</p>

<p style="margin-top: 1em">--intron-overhang-tolerance <br>
overhang allowed inside reference intron when merging [
default: 30 ]</p>

<p style="margin-top: 1em">Advanced Program Behavior
Options: <br>
-v/--verbose <br>
log-friendly verbose processing (no progress bar) [ default:
FALSE ]</p>

<p style="margin-top: 1em">-q/--quiet <br>
log-friendly quiet processing (no progress bar) [ default:
FALSE ]</p>

<p style="margin-top: 1em">--no-update-check <br>
do not contact server to check for update availability[
default: FALSE ]</p>

<p style="margin-top: 1em">Supported library types: <br>
ff-firststrand ff-secondstrand ff-unstranded fr-firststrand
fr-secondstrand fr-unstranded (default) transfrags</p>

<p style="margin-top: 1em">Supported library normalization
methods: <br>
classic-fpkm</p>

<p style="margin-top: 1em">cufflinks 2.2.1 November 2016
CUFFLINKS(1)</p>
<hr>
</body>
</html>
