<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CUFFCOMPARE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CUFFCOMPARE(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">CUFFCOMPARE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cuffcompare - component of cufflinks suite
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
cuffcompare v2.2.1 () <b>-----------------------------</b> Usage: cuffcompare
  [-r &lt;reference_mrna.gtf&gt;] [-R] [-T] [-V] [-s &lt;seq_path&gt;]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[-o &lt;outprefix&gt;] [-p &lt;cprefix&gt;] {-i
      &lt;input_gtf_list&gt; | &lt;input1.gtf&gt; [&lt;input2.gtf&gt; ..
      &lt;inputN.gtf&gt;]}</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Cuffcompare provides classification, reference annotation
      mapping and various statistics for Cufflinks transfrags. Cuffcompare
      clusters and tracks transfrags across multiple samples, writing matching
      transcripts (intron chains) into &lt;outprefix&gt;.tracking, and a GTF
      file &lt;outprefix&gt;.combined.gtf containing a nonredundant set of
      transcripts across all input files (with a single representative transfrag
      chosen for each clique of matching transfrags across samples).</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<b>-i</b> provide a text file with a list of Cufflinks GTF files to process
  instead
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">of expecting them as command line arguments (useful when a
      large number of GTF files should be processed)</dd>
</dl>
<div class="Pp"></div>
<b>-r</b> a set of known mRNAs to use as a reference for assessing
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">the accuracy of mRNAs or gene models given in
      &lt;input.gtf&gt;</dd>
</dl>
<div class="Pp"></div>
<b>-R</b> for <b>-r</b> option, consider only the reference transcripts that
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">overlap any of the input transfrags (Sn correction)</dd>
</dl>
<div class="Pp"></div>
<b>-Q</b> for <b>-r</b> option, consider only the input transcripts that
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">overlap any of the reference transcripts (Sp correction);
      (Warning: this will discard all &quot;novel&quot; loci!)</dd>
</dl>
<div class="Pp"></div>
<b>-M</b> discard (ignore) single-exon transfrags and reference transcripts
  <b>-N</b> discard (ignore) single-exon reference transcripts
<div class="Pp"></div>
<b>-s</b> &lt;seq_path&gt; can be a multi-fasta file with all the genomic
  sequences or
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">a directory containing multiple single-fasta files (one
      file per contig); lower case bases will be used to classify input
      transcripts as repeats</dd>
</dl>
<div class="Pp"></div>
<b>-e</b> max. distance (range) allowed from free ends of terminal exons of
  reference
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">transcripts when assessing exon accuracy (100)</dd>
</dl>
<div class="Pp"></div>
<b>-d</b> max. distance (range) for grouping transcript start sites (100)
  <b>-p</b> the name prefix to use for consensus transcripts in the
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&lt;outprefix&gt;.combined.gtf file (default: 'TCONS')</dd>
</dl>
<div class="Pp"></div>
<b>-C</b> include the &quot;contained&quot; transcripts in the .combined.gtf
  file <b>-F</b> do not discard intron-redundant transfrags if they share the 5'
  end
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(if they differ only at the 3' end))</dd>
</dl>
<div class="Pp"></div>
<b>-G</b> generic GFF input file(s): do not assume Cufflinks GTF, do not
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">discard any intron-redundant transfrags)</dd>
</dl>
<div class="Pp"></div>
<b>-T</b> do not generate .tmap and .refmap files for each input file <b>-V</b>
  verbose processing mode (showing all GFF parsing warnings)</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2016</td>
    <td class="foot-os">cuffcompare 2.2.1</td>
  </tr>
</table>
</body>
</html>
