<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:01:39 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CUFFCOMPARE(1) User Commands CUFFCOMPARE(1)</p>

<p style="margin-top: 1em">NAME <br>
cuffcompare - component of cufflinks suite</p>

<p style="margin-top: 1em">DESCRIPTION <br>
cuffcompare v2.2.1 () ----------------------------- Usage:
cuffcompare [-r &lt;reference_mrna.gtf&gt;] [-R] [-T] [-V]
[-s &lt;seq_path&gt;]</p>

<p style="margin-top: 1em">[-o &lt;outprefix&gt;] [-p
&lt;cprefix&gt;] {-i &lt;input_gtf_list&gt; |
&lt;input1.gtf&gt; [&lt;input2.gtf&gt; ..
&lt;inputN.gtf&gt;]}</p>

<p style="margin-top: 1em">Cuffcompare provides
classification, reference annotation mapping and various
statistics for Cufflinks transfrags. Cuffcompare clusters
and tracks transfrags across mul&acirc; <br>
tiple samples, writing matching transcripts (intron chains)
into &lt;outprefix&gt;.tracking, and a GTF file
&lt;outprefix&gt;.combined.gtf containing a nonredundant set
of transcripts <br>
across all input files (with a single representative
transfrag chosen for each clique of matching transfrags
across samples).</p>

<p style="margin-top: 1em">OPTIONS <br>
-i provide a text file with a list of Cufflinks GTF files to
process instead</p>

<p style="margin-top: 1em">of expecting them as command
line arguments (useful when a large number of GTF files
should be processed)</p>

<p style="margin-top: 1em">-r a set of known mRNAs to use
as a reference for assessing</p>

<p style="margin-top: 1em">the accuracy of mRNAs or gene
models given in &lt;input.gtf&gt;</p>

<p style="margin-top: 1em">-R for -r option, consider only
the reference transcripts that</p>

<p style="margin-top: 1em">overlap any of the input
transfrags (Sn correction)</p>

<p style="margin-top: 1em">-Q for -r option, consider only
the input transcripts that</p>

<p style="margin-top: 1em">overlap any of the reference
transcripts (Sp correction); (Warning: this will discard all
&quot;novel&quot; loci!)</p>

<p style="margin-top: 1em">-M discard (ignore) single-exon
transfrags and reference transcripts -N discard (ignore)
single-exon reference transcripts</p>

<p style="margin-top: 1em">-s &lt;seq_path&gt; can be a
multi-fasta file with all the genomic sequences or</p>

<p style="margin-top: 1em">a directory containing multiple
single-fasta files (one file per contig); lower case bases
will be used to classify input transcripts as repeats</p>

<p style="margin-top: 1em">-e max. distance (range) allowed
from free ends of terminal exons of reference</p>

<p style="margin-top: 1em">transcripts when assessing exon
accuracy (100)</p>

<p style="margin-top: 1em">-d max. distance (range) for
grouping transcript start sites (100) -p the name prefix to
use for consensus transcripts in the</p>

<p style="margin-top: 1em">&lt;outprefix&gt;.combined.gtf
file (default: &rsquo;TCONS&rsquo;)</p>

<p style="margin-top: 1em">-C include the
&quot;contained&quot; transcripts in the .combined.gtf file
-F do not discard intron-redundant transfrags if they share
the 5&rsquo; end</p>

<p style="margin-top: 1em">(if they differ only at the
3&rsquo; end))</p>

<p style="margin-top: 1em">-G generic GFF input file(s): do
not assume Cufflinks GTF, do not</p>

<p style="margin-top: 1em">discard any intron-redundant
transfrags)</p>

<p style="margin-top: 1em">-T do not generate .tmap and
.refmap files for each input file -V verbose processing mode
(showing all GFF parsing warnings)</p>

<p style="margin-top: 1em">cuffcompare 2.2.1 November 2016
CUFFCOMPARE(1)</p>
<hr>
</body>
</html>
