<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CUFFLINKS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CUFFLINKS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">CUFFLINKS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cufflinks - component of cufflinks suite
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>cufflinks</b> [ <i>options</i>] <i>&lt;hits.sam&gt;</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
cufflinks v2.2.1 linked against Boost version 106200
  <b>-----------------------------</b>
<h2 class="Ss" title="Ss" id="General_Options:"><a class="selflink" href="#General_Options:">General
  Options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>/--output-dir</dt>
  <dd class="It-tag">write all output files to this directory [ default: ./
    ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b>/--num-threads</dt>
  <dd class="It-tag">number of threads used during analysis [ default: 1 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b></dt>
  <dd class="It-tag">value of random number generator seed [ default: 0 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-G</b>/--GTF</dt>
  <dd class="It-tag">quantitate against reference transcript annotations</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b>/--GTF-guide</dt>
  <dd class="It-tag">use reference transcript annotation to guide assembly</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b>/--mask-file</dt>
  <dd class="It-tag">ignore all alignment within transcripts in this file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b>/--frag-bias-correct</dt>
  <dd class="It-tag">use bias correction - reference fasta required [ default:
      NULL ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b>/--multi-read-correct</dt>
  <dd class="It-tag">use 'rescue method' for multi-reads (more accurate) [
      default: FALSE ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--library-type</b></dt>
  <dd class="It-tag">library prep used for input reads [ default: below ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--library-norm-method</b></dt>
  <dd class="It-tag">Method used to normalize library sizes [ default: below
    ]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Advanced_Abundance_Estimation_Options:"><a class="selflink" href="#Advanced_Abundance_Estimation_Options:">Advanced
  Abundance Estimation Options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>/--frag-len-mean</dt>
  <dd class="It-tag">average fragment length (unpaired reads only) [ default:
      200 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>/--frag-len-std-dev</dt>
  <dd class="It-tag">fragment length std deviation (unpaired reads only) [
      default: 80 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-mle-iterations</b></dt>
  <dd class="It-tag">maximum iterations allowed for MLE calculation [ default:
      5000 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--compatible-hits-norm</b></dt>
  <dd class="It-tag">count hits compatible with reference RNAs only [ default:
      FALSE ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--total-hits-norm</b></dt>
  <dd class="It-tag">count all hits for normalization [ default: TRUE ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--num-frag-count-draws</b></dt>
  <dd class="It-tag">Number of fragment generation samples [ default: 100 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--num-frag-assign-draws</b></dt>
  <dd class="It-tag">Number of fragment assignment samples per generation [
      default: 50 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-frag-multihits</b></dt>
  <dd class="It-tag">Maximum number of alignments allowed per fragment [
      default: unlim ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-effective-length-correction</b></dt>
  <dd class="It-tag">No effective length correction [ default: FALSE ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-length-correction</b></dt>
  <dd class="It-tag">No length correction [ default: FALSE ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b>/--upper-quartile-norm</dt>
  <dd class="It-tag">Deprecated, use <b>--library-norm-method</b> [ DEPRECATED
    ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--raw-mapped-norm</b></dt>
  <dd class="It-tag">Deprecated, use <b>--library-norm-method</b> [ DEPRECATED
    ]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Advanced_Assembly_Options:"><a class="selflink" href="#Advanced_Assembly_Options:">Advanced
  Assembly Options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b>/--label</dt>
  <dd class="It-tag">assembled transcripts have this ID prefix [ default: CUFF
    ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b>/--min-isoform-fraction</dt>
  <dd class="It-tag">suppress transcripts below this abundance level [ default:
      0.10 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-j</b>/--pre-mrna-fraction</dt>
  <dd class="It-tag">suppress intra-intronic transcripts below this level [
      default: 0.15 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-I</b>/--max-intron-length</dt>
  <dd class="It-tag">ignore alignments with gaps longer than this [ default:
      300000 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b>/--junc-alpha</dt>
  <dd class="It-tag">alpha for junction binomial test filter [ default: 0.001
    ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b>/--small-anchor-fraction</dt>
  <dd class="It-tag">percent read overhang taken as 'suspiciously small' [
      default: 0.09 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-frags-per-transfrag</b></dt>
  <dd class="It-tag">minimum number of fragments needed for new transfrags [
      default: 10 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--overhang-tolerance</b></dt>
  <dd class="It-tag">number of terminal exon bp to tolerate in introns [
      default: 8 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-bundle-length</b></dt>
  <dd class="It-tag">maximum genomic length allowed for a given bundle [
      default:3500000 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-bundle-frags</b></dt>
  <dd class="It-tag">maximum fragments allowed in a bundle before skipping [
      default: 500000 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-intron-length</b></dt>
  <dd class="It-tag">minimum intron size allowed in genome [ default: 50 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trim-3-avgcov-thresh</b></dt>
  <dd class="It-tag">minimum avg coverage required to attempt 3' trimming [
      default: 10 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trim-3-dropoff-frac</b></dt>
  <dd class="It-tag">fraction of avg coverage below which to trim 3' end [
      default: 0.1 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-multiread-fraction</b></dt>
  <dd class="It-tag">maximum fraction of allowed multireads per transcript [
      default: 0.75 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--overlap-radius</b></dt>
  <dd class="It-tag">maximum gap size to fill between transfrags (in bp) [
      default: 50 ]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Advanced_Reference_Annotation_Guided_Assembly_Options:"><a class="selflink" href="#Advanced_Reference_Annotation_Guided_Assembly_Options:">Advanced
  Reference Annotation Guided Assembly Options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-faux-reads</b></dt>
  <dd class="It-tag">disable tiling by faux reads [ default: FALSE ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--3-overhang-tolerance</b></dt>
  <dd class="It-tag">overhang allowed on 3' end when merging with reference[
      default: 600 ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--intron-overhang-tolerance</b></dt>
  <dd class="It-tag">overhang allowed inside reference intron when merging [
      default: 30 ]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Advanced_Program_Behavior_Options:"><a class="selflink" href="#Advanced_Program_Behavior_Options:">Advanced
  Program Behavior Options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>/--verbose</dt>
  <dd class="It-tag">log-friendly verbose processing (no progress bar) [
      default: FALSE ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b>/--quiet</dt>
  <dd class="It-tag">log-friendly quiet processing (no progress bar) [ default:
      FALSE ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-update-check</b></dt>
  <dd class="It-tag">do not contact server to check for update availability[
      default: FALSE ]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Supported_library_types:"><a class="selflink" href="#Supported_library_types:">Supported
  library types:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">ff-firststrand ff-secondstrand ff-unstranded fr-firststrand
      fr-secondstrand fr-unstranded (default) transfrags</dd>
</dl>
<h2 class="Ss" title="Ss" id="Supported_library_normalization_methods:"><a class="selflink" href="#Supported_library_normalization_methods:">Supported
  library normalization methods:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">classic-fpkm</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2016</td>
    <td class="foot-os">cufflinks 2.2.1</td>
  </tr>
</table>
</body>
</html>
