<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:01:40 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CUFFDIFF(1) User Commands CUFFDIFF(1)</p>

<p style="margin-top: 1em">NAME <br>
cuffdiff - component of cufflinks suite</p>

<p style="margin-top: 1em">SYNOPSIS <br>
cuffdiff [options] &lt;transcripts.gtf&gt;
&lt;sample1_hits.sam&gt; &lt;sample2_hits.sam&gt; [...
sampleN_hits.sam]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
cuffdiff v2.2.1 () -----------------------------</p>

<p style="margin-top: 1em">Supply replicate SAMs as comma
separated lists for each condition:
sample1_rep1.sam,sample1_rep2.sam,...sample1_repM.sam</p>

<p style="margin-top: 1em">General Options: <br>
-o/--output-dir <br>
write all output files to this directory [ default: ./ ]</p>

<p style="margin-top: 1em">-L/--labels <br>
comma-separated list of condition labels</p>

<p style="margin-top: 1em">--FDR False discovery rate used
in testing [ default: 0.05 ]</p>

<p style="margin-top: 1em">-M/--mask-file <br>
ignore all alignment within transcripts in this file [
default: NULL ]</p>

<p style="margin-top: 1em">-C/--contrast-file <br>
Perform the constrasts specified in this file [ default:
NULL ]</p>

<p style="margin-top: 1em">-b/--frag-bias-correct <br>
use bias correction - reference fasta required [ default:
NULL ]</p>

<p style="margin-top: 1em">-u/--multi-read-correct <br>
use &rsquo;rescue method&rsquo; for multi-reads [ default:
FALSE ]</p>

<p style="margin-top: 1em">-p/--num-threads <br>
number of threads used during quantification [ default: 1
]</p>

<p style="margin-top: 1em">--no-diff <br>
Don&rsquo;t generate differential analysis files [ default:
FALSE ]</p>

<p style="margin-top: 1em">--no-js-tests <br>
Don&rsquo;t perform isoform switching tests [ default: FALSE
]</p>

<p style="margin-top: 1em">-T/--time-series <br>
treat samples as a time-series [ default: FALSE ]</p>

<p style="margin-top: 1em">--library-type <br>
Library prep used for input reads [ default: below ]</p>

<p style="margin-top: 1em">--dispersion-method <br>
Method used to estimate dispersion models [ default: below
]</p>

<p style="margin-top: 1em">--library-norm-method <br>
Method used to normalize library sizes [ default: below
]</p>

<p style="margin-top: 1em">Advanced Options: <br>
-m/--frag-len-mean <br>
average fragment length (unpaired reads only) [ default: 200
]</p>

<p style="margin-top: 1em">-s/--frag-len-std-dev <br>
fragment length std deviation (unpaired reads only) [
default: 80 ]</p>

<p style="margin-top: 1em">-c/--min-alignment-count <br>
minimum number of alignments in a locus for testing [
default: 10 ]</p>

<p style="margin-top: 1em">--max-mle-iterations <br>
maximum iterations allowed for MLE calculation [ default:
5000 ]</p>

<p style="margin-top: 1em">--compatible-hits-norm <br>
count hits compatible with reference RNAs only [ default:
TRUE ]</p>

<p style="margin-top: 1em">--total-hits-norm <br>
count all hits for normalization [ default: FALSE ]</p>

<p style="margin-top: 1em">-v/--verbose <br>
log-friendly verbose processing (no progress bar) [ default:
FALSE ]</p>

<p style="margin-top: 1em">-q/--quiet <br>
log-friendly quiet processing (no progress bar) [ default:
FALSE ]</p>

<p style="margin-top: 1em">--seed value of random number
generator seed [ default: 0 ]</p>

<p style="margin-top: 1em">--no-update-check <br>
do not contact server to check for update availability[
default: FALSE ]</p>

<p style="margin-top: 1em">--emit-count-tables <br>
print count tables used to fit overdispersion [ DEPRECATED
]</p>

<p style="margin-top: 1em">--max-bundle-frags <br>
maximum fragments allowed in a bundle before skipping [
default: 500000 ]</p>

<p style="margin-top: 1em">--num-frag-count-draws <br>
Number of fragment generation samples [ default: 100 ]</p>

<p style="margin-top: 1em">--num-frag-assign-draws <br>
Number of fragment assignment samples per generation [
default: 50 ]</p>

<p style="margin-top: 1em">--max-frag-multihits <br>
Maximum number of alignments allowed per fragment [ default:
unlim ]</p>

<p style="margin-top: 1em">--min-outlier-p <br>
Min replicate p value to admit for testing [ DEPRECATED
]</p>

<p style="margin-top: 1em">--min-reps-for-js-test <br>
Replicates needed for relative isoform shift testing [
default: 3 ]</p>


<p style="margin-top: 1em">--no-effective-length-correction
<br>
No effective length correction [ default: FALSE ]</p>

<p style="margin-top: 1em">--no-length-correction <br>
No length correction [ default: FALSE ]</p>

<p style="margin-top: 1em">-N/--upper-quartile-norm <br>
Deprecated, use --library-norm-method [ DEPRECATED ]</p>

<p style="margin-top: 1em">--geometric-norm <br>
Deprecated, use --library-norm-method [ DEPRECATED ]</p>

<p style="margin-top: 1em">--raw-mapped-norm <br>
Deprecated, use --library-norm-method [ DEPRECATED ]</p>

<p style="margin-top: 1em">--poisson-dispersion <br>
Deprecated, use --dispersion-method [ DEPRECATED ]</p>

<p style="margin-top: 1em">Debugging use only: <br>
--read-skip-fraction <br>
Skip a random subset of reads this size [ default: 0.0 ]</p>

<p style="margin-top: 1em">--no-read-pairs <br>
Break all read pairs [ default: FALSE ]</p>

<p style="margin-top: 1em">--trim-read-length <br>
Trim reads to be this long (keep 5&rsquo; end) [ default:
none ]</p>

<p style="margin-top: 1em">--no-scv-correction <br>
Disable SCV correction [ default: FALSE ]</p>

<p style="margin-top: 1em">Supported library types: <br>
ff-firststrand ff-secondstrand ff-unstranded fr-firststrand
fr-secondstrand fr-unstranded (default) transfrags</p>

<p style="margin-top: 1em">Supported dispersion methods:
<br>
blind per-condition poisson pooled (default)</p>

<p style="margin-top: 1em">Supported library normalization
methods: <br>
classic-fpkm geometric (default) geometric quartile</p>

<p style="margin-top: 1em">cuffdiff 2.2.1 November 2016
CUFFDIFF(1)</p>
<hr>
</body>
</html>
