<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:01:45 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>CUTADAPT(1) User Commands CUTADAPT(1)</p>

<p style="margin-top: 1em">NAME <br>
cutadapt - remove adapter sequences from high-throughput
sequencing reads</p>

<p style="margin-top: 1em">SYNOPSIS <br>
cutadapt -a ADAPTER [options] [-o output.fastq]
input.fastq</p>

<p style="margin-top: 1em">For paired-end reads:</p>

<p style="margin-top: 1em">cutadapt -a ADAPT1 -A ADAPT2
[options] -o out1.fastq -p out2.fastq in1.fastq
in2.fastq</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Replace &quot;ADAPTER&quot; with the actual sequence of your
3&rsquo; adapter. IUPAC wildcard characters are supported.
The reverse complement is *not* automatically searched. All
reads from <br>
input.fastq will be written to output.fastq with the adapter
sequence removed. Adapter matching is error-tolerant.
Multiple adapter sequences can be given (use further -a <br>
options), but only the best-matching adapter will be
removed.</p>

<p style="margin-top: 1em">Input may also be in FASTA
format. Compressed input and output is supported and
auto-detected from the file name (.gz, .xz, .bz2). Use the
file name &rsquo;-&rsquo; for standard input/out&acirc; <br>
put. Without the -o option, output is sent to standard
output.</p>

<p style="margin-top: 1em">OPTIONS <br>
--help show all command-line options</p>

<p style="margin-top: 1em">--version <br>
show program&rsquo;s version number and exit</p>

<p style="margin-top: 1em">-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">--debug <br>
Print debugging information.</p>

<p style="margin-top: 1em">-f FORMAT, --format=FORMAT <br>
Input file format; can be either &rsquo;fasta&rsquo;,
&rsquo;fastq&rsquo; or &rsquo;sra-fastq&rsquo;. Ignored when
reading csfasta/qual files. Default: auto-detect from file
name extension.</p>

<p style="margin-top: 1em">Finding adapters::</p>

<p style="margin-top: 1em">Parameters -a, -g, -b specify
adapters to be removed from each read (or from the first
read in a pair if data is paired). If specified multiple
times, only the best match&acirc; <br>
ing adapter is trimmed (but see the --times option). When
the special notation &rsquo;file:FILE&rsquo; is used,
adapter sequences are read from the given FASTA file.</p>

<p style="margin-top: 1em">-a ADAPTER, --adapter=ADAPTER
<br>
Sequence of an adapter ligated to the 3&rsquo; end (paired
data: of the first read). The adapter and subsequent bases
are trimmed. If a &rsquo;$&rsquo; character is appended
(&rsquo;anchoring&rsquo;), <br>
the adapter is only found if it is a suffix of the read.</p>

<p style="margin-top: 1em">-g ADAPTER, --front=ADAPTER <br>
Sequence of an adapter ligated to the 5&rsquo; end (paired
data: of the first read). The adapter and any preceding
bases are trimmed. Partial matches at the 5&rsquo; end are
allowed. <br>
If a &rsquo;^&rsquo; character is prepended
(&rsquo;anchoring&rsquo;), the adapter is only found if it
is a prefix of the read.</p>

<p style="margin-top: 1em">-b ADAPTER, --anywhere=ADAPTER
<br>
Sequence of an adapter that may be ligated to the 5&rsquo;
or 3&rsquo; end (paired data: of the first read). Both types
of matches as described under -a und -g are allowed. If the
<br>
first base of the read is part of the match, the behavior is
as with -g, otherwise as with -a. This option is mostly for
rescuing failed library preparations - do not use <br>
if you know which end your adapter was ligated to!</p>

<p style="margin-top: 1em">-e ERROR_RATE,
--error-rate=ERROR_RATE <br>
Maximum allowed error rate (no. of errors divided by the
length of the matching region). Default: 0.1</p>

<p style="margin-top: 1em">--no-indels <br>
Allow only mismatches in alignments. Default: allow both
mismatches and indels</p>

<p style="margin-top: 1em">-n COUNT, --times=COUNT <br>
Remove up to COUNT adapters from each read. Default: 1</p>

<p style="margin-top: 1em">-O MINLENGTH,
--overlap=MINLENGTH <br>
If the overlap between the read and the adapter is shorter
than MINLENGTH, the read is not modified. Reduces the no. of
bases trimmed due to random adapter matches. <br>
Default: 3</p>

<p style="margin-top: 1em">--match-read-wildcards <br>
Interpret IUPAC wildcards in reads. Default: False</p>

<p style="margin-top: 1em">-N, --no-match-adapter-wildcards
<br>
Do not interpret IUPAC wildcards in adapters.</p>

<p style="margin-top: 1em">--no-trim <br>
Match and redirect reads to output/untrimmed-output as
usual, but do not remove adapters.</p>

<p style="margin-top: 1em">--mask-adapter <br>
Mask adapters with &rsquo;N&rsquo; characters instead of
trimming them.</p>

<p style="margin-top: 1em">Additional read
modifications:</p>

<p style="margin-top: 1em">-u LENGTH, --cut=LENGTH <br>
Remove bases from each read (first read only if paired). If
LENGTH is positive, remove bases from the beginning. If
LENGTH is negative, remove bases from the end. Can be <br>
used twice if LENGTHs have different signs.</p>

<p style="margin-top: 1em">-q
[5&rsquo;CUTOFF,]3&rsquo;CUTOFF,
--quality-cutoff=[5&rsquo;CUTOFF,]3&rsquo;CUTOFF <br>
Trim low-quality bases from 5&rsquo; and/or 3&rsquo; ends of
each read before adapter removal. Applied to both reads if
data is paired. If one value is given, only the 3&rsquo; end
is <br>
trimmed. If two comma-separated cutoffs are given, the
5&rsquo; end is trimmed with the first cutoff, the 3&rsquo;
end with the second.</p>

<p style="margin-top: 1em">--nextseq-trim=3&rsquo;CUTOFF
<br>
NextSeq-specific quality trimming (each read). Trims also
dark cycles appearing as high-quality G bases
(EXPERIMENTAL).</p>

<p style="margin-top: 1em">--quality-base=QUALITY_BASE <br>
Assume that quality values in FASTQ are encoded as
ascii(quality + QUALITY_BASE). This needs to be set to 64
for some old Illumina FASTQ files. Default: 33</p>

<p style="margin-top: 1em">--trim-n <br>
Trim N&rsquo;s on ends of reads.</p>

<p style="margin-top: 1em">-x PREFIX, --prefix=PREFIX <br>
Add this prefix to read names. Use {name} to insert the name
of the matching adapter.</p>

<p style="margin-top: 1em">-y SUFFIX, --suffix=SUFFIX <br>
Add this suffix to read names; can also include {name}</p>

<p style="margin-top: 1em">--strip-suffix=STRIP_SUFFIX <br>
Remove this suffix from read names if present. Can be given
multiple times.</p>

<p style="margin-top: 1em">--length-tag=TAG <br>
Search for TAG followed by a decimal number in the
description field of the read. Replace the decimal number
with the correct length of the trimmed read. For example,
use <br>
--length-tag &rsquo;length=&rsquo; to correct fields like
&rsquo;length=123&rsquo;.</p>

<p style="margin-top: 1em">Filtering of processed
reads:</p>

<p style="margin-top: 1em">--discard-trimmed, --discard
<br>
Discard reads that contain an adapter. Also use -O to avoid
discarding too many randomly matching reads!</p>

<p style="margin-top: 1em">--discard-untrimmed,
--trimmed-only <br>
Discard reads that do not contain the adapter.</p>

<p style="margin-top: 1em">-m LENGTH,
--minimum-length=LENGTH <br>
Discard trimmed reads that are shorter than LENGTH. Reads
that are too short even before adapter removal are also
discarded. In colorspace, an initial primer is not <br>
counted. Default: 0</p>

<p style="margin-top: 1em">-M LENGTH,
--maximum-length=LENGTH <br>
Discard trimmed reads that are longer than LENGTH. Reads
that are too long even before adapter removal are also
discarded. In colorspace, an initial primer is not <br>
counted. Default: no limit</p>

<p style="margin-top: 1em">--max-n=COUNT <br>
Discard reads with too many N bases. If COUNT is an integer,
it is treated as the absolute number of N bases. If it is
between 0 and 1, it is treated as the proportion of <br>
N&rsquo;s allowed in a read.</p>

<p style="margin-top: 1em">Output:</p>

<p style="margin-top: 1em">--quiet <br>
Print only error messages.</p>

<p style="margin-top: 1em">-o FILE, --output=FILE <br>
Write trimmed reads to FILE. FASTQ or FASTA format is chosen
depending on input. The summary report is sent to standard
output. Use &rsquo;{name}&rsquo; in FILE to demultiplex
reads <br>
into multiple files. Default: write to standard output</p>

<p style="margin-top: 1em">--info-file=FILE <br>
Write information about each read and its adapter matches
into FILE. See the documentation for the file format.</p>

<p style="margin-top: 1em">-r FILE, --rest-file=FILE <br>
When the adapter matches in the middle of a read, write the
rest (after the adapter) into FILE.</p>

<p style="margin-top: 1em">--wildcard-file=FILE <br>
When the adapter has N bases (wildcards), write adapter
bases matching wildcard positions to FILE. When there are
indels in the alignment, this will often not be accu&acirc;
<br>
rate.</p>

<p style="margin-top: 1em">--too-short-output=FILE <br>
Write reads that are too short (according to length
specified by -m) to FILE. Default: discard reads</p>

<p style="margin-top: 1em">--too-long-output=FILE <br>
Write reads that are too long (according to length specified
by -M) to FILE. Default: discard reads</p>

<p style="margin-top: 1em">--untrimmed-output=FILE <br>
Write reads that do not contain the adapter to FILE.
Default: output to same file as trimmed reads</p>

<p style="margin-top: 1em">Colorspace options:</p>

<p style="margin-top: 1em">-c, --colorspace <br>
Enable colorspace mode: Also trim the color that is adjacent
to the found adapter.</p>

<p style="margin-top: 1em">-d, --double-encode <br>
Double-encode colors (map 0,1,2,3,4 to A,C,G,T,N).</p>

<p style="margin-top: 1em">-t, --trim-primer <br>
Trim primer base and the first color (which is the
transition to the first nucleotide)</p>

<p style="margin-top: 1em">--strip-f3 <br>
Strip the _F3 suffix of read names</p>

<p style="margin-top: 1em">--maq, --bwa <br>
MAQ- and BWA-compatible colorspace output. This enables -c,
-d, -t, --strip-f3 and -y &rsquo;/1&rsquo;.</p>

<p style="margin-top: 1em">--no-zero-cap <br>
Do not change negative quality values to zero in colorspace
data. By default, they are since many tools have problems
with negative qualities.</p>

<p style="margin-top: 1em">-z, --zero-cap <br>
Change negative quality values to zero. This is enabled by
default when -c/--colorspace is also enabled. Use the above
option to disable it.</p>

<p style="margin-top: 1em">Paired-end options:</p>

<p style="margin-top: 1em">The -A/-G/-B/-U options work
like their -a/-b/-g/-u counterparts, but are applied to the
second read in each pair.</p>

<p style="margin-top: 1em">-A ADAPTER <br>
3&rsquo; adapter to be removed from second read in a
pair.</p>

<p style="margin-top: 1em">-G ADAPTER <br>
5&rsquo; adapter to be removed from second read in a
pair.</p>

<p style="margin-top: 1em">-B ADAPTER <br>
5&rsquo;/3 adapter to be removed from second read in a
pair.</p>

<p style="margin-top: 1em">-U LENGTH <br>
Remove LENGTH bases from second read in a pair (see
--cut).</p>

<p style="margin-top: 1em">-p FILE, --paired-output=FILE
<br>
Write second read in a pair to FILE.</p>

<p style="margin-top: 1em">--pair-filter=(any|both) <br>
Which of the reads in a paired-end read have to match the
filtering criterion in order for it to be filtered. Default:
any</p>

<p style="margin-top: 1em">--interleaved <br>
Read and write interleaved paired-end reads.</p>

<p style="margin-top: 1em">--untrimmed-paired-output=FILE
<br>
Write second read in a pair to this FILE when no adapter was
found in the first read. Use this option together with
--untrimmed-output when trimming pairedend reads. <br>
Default: output to same file as trimmed reads</p>

<p style="margin-top: 1em">--too-short-paired-output=FILE
<br>
Write second read in a pair to this file if pair is too
short. Use together with --too-short-output.</p>

<p style="margin-top: 1em">--too-long-paired-output=FILE
<br>
Write second read in a pair to this file if pair is too
long. Use together with --too-long-output.</p>

<p style="margin-top: 1em">SEE ALSO <br>
See http://cutadapt.readthedocs.org/ for full
documentation.</p>

<p style="margin-top: 1em">Citation <br>
Marcel Martin. Cutadapt removes adapter sequences from
high-throughput sequencing reads. EMBnet.Journal,
17(1):10-12, May 2011.
http://dx.doi.org/10.14806/ej.17.1.200</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">cutadapt 1.10 June 2016
CUTADAPT(1)</p>
<hr>
</body>
</html>
