<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>CUTADAPT(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">CUTADAPT(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">CUTADAPT(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
cutadapt - remove adapter sequences from high-throughput sequencing reads
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>cutadapt</b> <b>-a</b> ADAPTER [options] [-o
      output.fastq] input.fastq</dd>
</dl>
<div class="Pp"></div>
For paired-end reads:
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>cutadapt</b> <b>-a</b> ADAPT1 <b>-A</b> ADAPT2 [options]
      <b>-o</b> out1.fastq <b>-p</b> out2.fastq in1.fastq in2.fastq</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Replace &quot;ADAPTER&quot; with the actual sequence of your 3' adapter. IUPAC
  wildcard characters are supported. The reverse complement is *not*
  automatically searched. All reads from input.fastq will be written to
  output.fastq with the adapter sequence removed. Adapter matching is
  error-tolerant. Multiple adapter sequences can be given (use further <b>-a</b>
  options), but only the best-matching adapter will be removed.
<div class="Pp"></div>
Input may also be in FASTA format. Compressed input and output is supported and
  auto-detected from the file name (.gz, .xz, .bz2). Use the file name '-' for
  standard input/output. Without the <b>-o</b> option, output is sent to
  standard output.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">show all command-line options</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">show program's version number and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--debug</b></dt>
  <dd class="It-tag">Print debugging information.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> FORMAT, <b>--format</b>=<i>FORMAT</i></dt>
  <dd class="It-tag">Input file format; can be either 'fasta', 'fastq' or
      'sra-fastq'. Ignored when reading csfasta/qual files. Default: auto-detect
      from file name extension.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Finding adapters::</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Parameters <b>-a</b>, <b>-g</b>, <b>-b</b> specify adapters
      to be removed from each read (or from the first read in a pair if data is
      paired). If specified multiple times, only the best matching adapter is
      trimmed (but see the <b>--times</b> option). When the special notation
      'file:FILE' is used, adapter sequences are read from the given FASTA
    file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> ADAPTER, <b>--adapter</b>=<i>ADAPTER</i></dt>
  <dd class="It-tag">Sequence of an adapter ligated to the 3' end (paired data:
      of the first read). The adapter and subsequent bases are trimmed. If a '$'
      character is appended ('anchoring'), the adapter is only found if it is a
      suffix of the read.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b> ADAPTER, <b>--front</b>=<i>ADAPTER</i></dt>
  <dd class="It-tag">Sequence of an adapter ligated to the 5' end (paired data:
      of the first read). The adapter and any preceding bases are trimmed.
      Partial matches at the 5' end are allowed. If a '^' character is prepended
      ('anchoring'), the adapter is only found if it is a prefix of the
    read.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b> ADAPTER, <b>--anywhere</b>=<i>ADAPTER</i></dt>
  <dd class="It-tag">Sequence of an adapter that may be ligated to the 5' or 3'
      end (paired data: of the first read). Both types of matches as described
      under <b>-a</b> und <b>-g</b> are allowed. If the first base of the read
      is part of the match, the behavior is as with <b>-g</b>, otherwise as with
      <b>-a</b>. This option is mostly for rescuing failed library preparations
      - do not use if you know which end your adapter was ligated to!</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b> ERROR_RATE,
    <b>--error-rate</b>=<i>ERROR_RATE</i></dt>
  <dd class="It-tag">Maximum allowed error rate (no. of errors divided by the
      length of the matching region). Default: 0.1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-indels</b></dt>
  <dd class="It-tag">Allow only mismatches in alignments. Default: allow both
      mismatches and indels</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> COUNT, <b>--times</b>=<i>COUNT</i></dt>
  <dd class="It-tag">Remove up to COUNT adapters from each read. Default: 1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-O</b> MINLENGTH, <b>--overlap</b>=<i>MINLENGTH</i></dt>
  <dd class="It-tag">If the overlap between the read and the adapter is shorter
      than MINLENGTH, the read is not modified. Reduces the no. of bases trimmed
      due to random adapter matches. Default: 3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--match-read-wildcards</b></dt>
  <dd class="It-tag">Interpret IUPAC wildcards in reads. Default: False</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b>, <b>--no-match-adapter-wildcards</b></dt>
  <dd class="It-tag">Do not interpret IUPAC wildcards in adapters.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-trim</b></dt>
  <dd class="It-tag">Match and redirect reads to output/untrimmed-output as
      usual, but do not remove adapters.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mask-adapter</b></dt>
  <dd class="It-tag">Mask adapters with 'N' characters instead of trimming
    them.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Additional read modifications:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b> LENGTH, <b>--cut</b>=<i>LENGTH</i></dt>
  <dd class="It-tag">Remove bases from each read (first read only if paired). If
      LENGTH is positive, remove bases from the beginning. If LENGTH is
      negative, remove bases from the end. Can be used twice if LENGTHs have
      different signs.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> [5'CUTOFF,]3'CUTOFF,
    <b>--quality-cutoff</b>=<i>[5</i>'CUTOFF,]3'CUTOFF</dt>
  <dd class="It-tag">Trim low-quality bases from 5' and/or 3' ends of each read
      before adapter removal. Applied to both reads if data is paired. If one
      value is given, only the 3' end is trimmed. If two comma-separated cutoffs
      are given, the 5' end is trimmed with the first cutoff, the 3' end with
      the second.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nextseq-trim</b>=<i>3</i>'CUTOFF</dt>
  <dd class="It-tag">NextSeq-specific quality trimming (each read). Trims also
      dark cycles appearing as high-quality G bases (EXPERIMENTAL).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quality-base</b>=<i>QUALITY_BASE</i></dt>
  <dd class="It-tag">Assume that quality values in FASTQ are encoded as
      ascii(quality + QUALITY_BASE). This needs to be set to 64 for some old
      Illumina FASTQ files. Default: 33</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trim-n</b></dt>
  <dd class="It-tag">Trim N's on ends of reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-x</b> PREFIX, <b>--prefix</b>=<i>PREFIX</i></dt>
  <dd class="It-tag">Add this prefix to read names. Use {name} to insert the
      name of the matching adapter.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-y</b> SUFFIX, <b>--suffix</b>=<i>SUFFIX</i></dt>
  <dd class="It-tag">Add this suffix to read names; can also include {name}</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--strip-suffix</b>=<i>STRIP_SUFFIX</i></dt>
  <dd class="It-tag">Remove this suffix from read names if present. Can be given
      multiple times.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--length-tag</b>=<i>TAG</i></dt>
  <dd class="It-tag">Search for TAG followed by a decimal number in the
      description field of the read. Replace the decimal number with the correct
      length of the trimmed read. For example, use <b>--length-tag</b> 'length='
      to correct fields like 'length=123'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Filtering of processed reads:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--discard-trimmed</b>, <b>--discard</b></dt>
  <dd class="It-tag">Discard reads that contain an adapter. Also use <b>-O</b>
      to avoid discarding too many randomly matching reads!</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--discard-untrimmed</b>, <b>--trimmed-only</b></dt>
  <dd class="It-tag">Discard reads that do not contain the adapter.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> LENGTH,
    <b>--minimum-length</b>=<i>LENGTH</i></dt>
  <dd class="It-tag">Discard trimmed reads that are shorter than LENGTH. Reads
      that are too short even before adapter removal are also discarded. In
      colorspace, an initial primer is not counted. Default: 0</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> LENGTH,
    <b>--maximum-length</b>=<i>LENGTH</i></dt>
  <dd class="It-tag">Discard trimmed reads that are longer than LENGTH. Reads
      that are too long even before adapter removal are also discarded. In
      colorspace, an initial primer is not counted. Default: no limit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-n</b>=<i>COUNT</i></dt>
  <dd class="It-tag">Discard reads with too many N bases. If COUNT is an
      integer, it is treated as the absolute number of N bases. If it is between
      0 and 1, it is treated as the proportion of N's allowed in a read.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Output:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quiet</b></dt>
  <dd class="It-tag">Print only error messages.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> FILE, <b>--output</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write trimmed reads to FILE. FASTQ or FASTA format is
      chosen depending on input. The summary report is sent to standard output.
      Use '{name}' in FILE to demultiplex reads into multiple files. Default:
      write to standard output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--info-file</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write information about each read and its adapter matches
      into FILE. See the documentation for the file format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> FILE, <b>--rest-file</b>=<i>FILE</i></dt>
  <dd class="It-tag">When the adapter matches in the middle of a read, write the
      rest (after the adapter) into FILE.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wildcard-file</b>=<i>FILE</i></dt>
  <dd class="It-tag">When the adapter has N bases (wildcards), write adapter
      bases matching wildcard positions to FILE. When there are indels in the
      alignment, this will often not be accurate.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--too-short-output</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write reads that are too short (according to length
      specified by <b>-m</b>) to FILE. Default: discard reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--too-long-output</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write reads that are too long (according to length
      specified by <b>-M</b>) to FILE. Default: discard reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--untrimmed-output</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write reads that do not contain the adapter to FILE.
      Default: output to same file as trimmed reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Colorspace options:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>, <b>--colorspace</b></dt>
  <dd class="It-tag">Enable colorspace mode: Also trim the color that is
      adjacent to the found adapter.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b>, <b>--double-encode</b></dt>
  <dd class="It-tag">Double-encode colors (map 0,1,2,3,4 to A,C,G,T,N).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>, <b>--trim-primer</b></dt>
  <dd class="It-tag">Trim primer base and the first color (which is the
      transition to the first nucleotide)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--strip-f3</b></dt>
  <dd class="It-tag">Strip the _F3 suffix of read names</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maq</b>, <b>--bwa</b></dt>
  <dd class="It-tag">MAQ- and BWA-compatible colorspace output. This enables
      <b>-c</b>, <b>-d</b>, <b>-t</b>, <b>--strip-f3</b> and <b>-y</b>
    '/1'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-zero-cap</b></dt>
  <dd class="It-tag">Do not change negative quality values to zero in colorspace
      data. By default, they are since many tools have problems with negative
      qualities.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-z</b>, <b>--zero-cap</b></dt>
  <dd class="It-tag">Change negative quality values to zero. This is enabled by
      default when <b>-c</b>/--colorspace is also enabled. Use the above option
      to disable it.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Paired-end options:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">The <b>-A</b>/-G/-B/-U options work like their
      <b>-a</b>/-b/-g/-u counterparts, but are applied to the second read in
      each pair.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b> ADAPTER</dt>
  <dd class="It-tag">3' adapter to be removed from second read in a pair.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-G</b> ADAPTER</dt>
  <dd class="It-tag">5' adapter to be removed from second read in a pair.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b> ADAPTER</dt>
  <dd class="It-tag">5'/3 adapter to be removed from second read in a pair.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b> LENGTH</dt>
  <dd class="It-tag">Remove LENGTH bases from second read in a pair (see
      <b>--cut</b>).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> FILE, <b>--paired-output</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write second read in a pair to FILE.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pair-filter=</b>(any|both)</dt>
  <dd class="It-tag">Which of the reads in a paired-end read have to match the
      filtering criterion in order for it to be filtered. Default: any</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--interleaved</b></dt>
  <dd class="It-tag">Read and write interleaved paired-end reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--untrimmed-paired-output</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write second read in a pair to this FILE when no adapter
      was found in the first read. Use this option together with
      <b>--untrimmed-output</b> when trimming pairedend reads. Default: output
      to same file as trimmed reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--too-short-paired-output</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write second read in a pair to this file if pair is too
      short. Use together with <b>--too-short-output</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--too-long-paired-output</b>=<i>FILE</i></dt>
  <dd class="It-tag">Write second read in a pair to this file if pair is too
      long. Use together with <b>--too-long-output</b>.</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See http://cutadapt.readthedocs.org/ for full documentation.
<h2 class="Ss" title="Ss" id="Citation"><a class="selflink" href="#Citation">Citation</a></h2>
Marcel Martin. Cutadapt removes adapter sequences from high-throughput
  sequencing reads. EMBnet.Journal, 17(1):10-12, May 2011.
  http://dx.doi.org/10.14806/ej.17.1.200
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">June 2016</td>
    <td class="foot-os">cutadapt 1.10</td>
  </tr>
</table>
</body>
</html>
