<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:02:07 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DALIGNER(1) General Commands Manual DALIGNER(1)</p>

<p style="margin-top: 1em">NAME <br>
daligner - local alignment discovery between long nucleotide
sequencing reads</p>

<p style="margin-top: 1em">SYNOPSIS <br>
daligner [options] files...</p>

<p style="margin-top: 1em">DESCRIPTION <br>
These tools permit one to find all significant local
alignments between reads encoded in a Dazzler database. The
assumption is that the reads are from a Pacific Biosciences
RS <br>
II long read sequencer. That is, the reads are long and
noisy, up to 15% on average.</p>

<p style="margin-top: 1em">OPTIONS <br>
For a complete description of available commands and their
options, see /usr/share/doc/daligner/README.md</p>

<p style="margin-top: 1em">October 2016 DALIGNER(1)</p>
<hr>
</body>
</html>
