<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DALIGNER(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DALIGNER(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">DALIGNER(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
daligner - local alignment discovery between long nucleotide sequencing reads
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>daligner</b> [<i>options</i>]<i> files</i>...
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
These tools permit one to find all significant local alignments between reads
  encoded in a Dazzler database. The assumption is that the reads are from a
  Pacific Biosciences RS II long read sequencer. That is, the reads are long and
  noisy, up to 15% on average.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
For a complete description of available commands and their options, see
  /usr/share/doc/daligner/README.md</div>
<table class="foot">
  <tr>
    <td class="foot-date">October 2016</td>
    <td class="foot-os"></td>
  </tr>
</table>
</body>
</html>
