<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DAWG(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DAWG(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">DAWG(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
dawg - DNA assembly with gaps, a DNA sequence simulator
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
dawg 1.2-release DNA Assembly With Gaps Copyright &#x00A9; 2004-2009 Reed A.
  Cartwright
<div class="Pp"></div>
dawg <b>-[scubvhqew</b>?] [-o outputfile] file1 [file2...]
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-s</b>:
  process files serially [default]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-c</b>:
  process files combined together</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-u</b>:
  unbuffered output</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-b</b>:
  buffered output [default]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-q</b>:
  disable error and warning reports (quiet)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-e</b>:
  enable error reports [default]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-w</b>:
  enable warning reports [default]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-v</b>:
  display version information</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-h</b>:
  display help information</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;">-?: same as
  <b>-h</b></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  outputfile: override output filename in the configuration file</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Dawg will read stdin if filename is &quot;-&quot;.</dd>
</dl>
<div class="Pp"></div>
FILE FORMAT
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">The file format takes a series of statements in the form of
      &quot;name = value,&quot; where &quot;name&quot; is alphanumeric and value
      can be a string, number, boolean, tree, or vector of values. A single
      variable is equivalent to a vector of a single entry.</dd>
</dl>
<div class="Pp"></div>
string: &quot;[char-sequence]&quot;
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">'[char-sequence]' &quot;&quot;&quot;[multi-line
      char-sequence]&quot;&quot;&quot; (rm initial and final newlines)
      '''[multi-line char-sequence]''' (kp initial and final newlines)</dd>
</dl>
<div class="Pp"></div>
number: [sign]digits[.digits][(e|E)[sign]digits] boolean: true|false tree:
  Newick Format vector: { value, value, ...}
<div class="Pp"></div>
OPTIONS
<dl class="Bl-tag">
  <dt class="It-tag">Name</dt>
  <dd class="It-tag">Type Description</dd>
</dl>
<div class="Pp"></div>
<b>---------------------------------</b>
<dl class="Bl-tag">
  <dt class="It-tag">Tree</dt>
  <dd class="It-tag">VT phylogeny</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">TreeScale</dt>
  <dd class="It-tag">N coefficient to scale branch lengths by</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Sequence</dt>
  <dd class="It-tag">VS root sequences</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Length</dt>
  <dd class="It-tag">VN length of generated root sequences</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Rates</dt>
  <dd class="It-tag">VVN rate of evolution of each root nucleotide</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Model</dt>
  <dd class="It-tag">S model of evolution: GTR|JC|K2P|K3P|HKY|F81|F84|TN</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Freqs</dt>
  <dd class="It-tag">VN nucleotide (ACGT) frequencies</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Params</dt>
  <dd class="It-tag">VN parameters for the model of evolution</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Width</dt>
  <dd class="It-tag">N block width for indels and recombination</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Scale</dt>
  <dd class="It-tag">VN block position scales</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Gamma</dt>
  <dd class="It-tag">VN coefficients of variance for rate heterogenity</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Alpha</dt>
  <dd class="It-tag">VN shape parameters</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Iota</dt>
  <dd class="It-tag">VN proportions of invariant sites</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">GapModel</dt>
  <dd class="It-tag">VS models of indel formation: NB|PL|US</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Lambda</dt>
  <dd class="It-tag">VN rates of indel formation</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">GapParams</dt>
  <dd class="It-tag">VVN parameter for the indel model</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Reps</dt>
  <dd class="It-tag">N number of data sets to output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">File</dt>
  <dd class="It-tag">S output file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Format</dt>
  <dd class="It-tag">S output format: Fasta|Nexus|Phylip|Clustal</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">GapSingleChar</dt>
  <dd class="It-tag">B output gaps as a single character</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">GapPlus</dt>
  <dd class="It-tag">B distinguish insertions from deletions in alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">KeepFlank</dt>
  <dd class="It-tag">N undeletable flanking regions N nucs from sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">KeepEmpty</dt>
  <dd class="It-tag">B preserve empty columns in final alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">LowerCase</dt>
  <dd class="It-tag">B output sequences in lowercase</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Translate</dt>
  <dd class="It-tag">B translate outputed sequences to amino acids</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Seed</dt>
  <dd class="It-tag">VN pseudo-random-number-generator seed (integers)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Out.Block.Head</dt>
  <dd class="It-tag">S string to insert at the start of the output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Out.Block.Tail</dt>
  <dd class="It-tag">S string to insert at the end of the output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Out.Block.Before S</dt>
  <dd class="It-tag">string to insert before a sequence set in the output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Out.Block.After</dt>
  <dd class="It-tag">S string to insert after a sequence set in the output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Out.Subst</dt>
  <dd class="It-tag">B do variable subsitution in Out.Block.*</dd>
</dl>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
This program is free software; you can redistribute it and/or modify it under
  the terms of the GNU General Public License as published by the Free Software
  Foundation; either version 2 of the License, or (at your option) any later
  version.
<div class="Pp"></div>
This program is distributed in the hope that it will be useful, but WITHOUT ANY
  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
  A PARTICULAR PURPOSE. See the GNU General Public License for more details.
<div class="Pp"></div>
You should have received a copy of the GNU General Public License along with
  this program; if not, write to the Free Software Foundation, Inc., 59 Temple
  Place, Suite 330, Boston, MA 02111-1307 USA
<div class="Pp"></div>
Send Bug Reports to reed@scit.us.</div>
<table class="foot">
  <tr>
    <td class="foot-date">December 2015</td>
    <td class="foot-os">dawg 1.2-release</td>
  </tr>
</table>
</body>
</html>
