<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:02:25 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DAWG(1) User Commands DAWG(1)</p>

<p style="margin-top: 1em">NAME <br>
dawg - DNA assembly with gaps, a DNA sequence simulator</p>

<p style="margin-top: 1em">DESCRIPTION <br>
dawg 1.2-release DNA Assembly With Gaps Copyright
&Acirc;&copy; 2004-2009 Reed A. Cartwright</p>

<p style="margin-top: 1em">dawg -[scubvhqew?] [-o
outputfile] file1 [file2...]</p>

<p style="margin-top: 1em">-s: process files serially
[default]</p>

<p style="margin-top: 1em">-c: process files combined
together</p>

<p style="margin-top: 1em">-u: unbuffered output</p>

<p style="margin-top: 1em">-b: buffered output
[default]</p>

<p style="margin-top: 1em">-q: disable error and warning
reports (quiet)</p>

<p style="margin-top: 1em">-e: enable error reports
[default]</p>

<p style="margin-top: 1em">-w: enable warning reports
[default]</p>

<p style="margin-top: 1em">-v: display version
information</p>

<p style="margin-top: 1em">-h: display help information</p>

<p style="margin-top: 1em">-?: same as -h</p>

<p style="margin-top: 1em">-o outputfile: override output
filename in the configuration file</p>

<p style="margin-top: 1em">Dawg will read stdin if filename
is &quot;-&quot;.</p>

<p style="margin-top: 1em">FILE FORMAT</p>

<p style="margin-top: 1em">The file format takes a series
of statements in the form of &quot;name = value,&quot; where
&quot;name&quot; is alphanumeric and value can be a string,
number, boolean, tree, or vector of <br>
values. A single variable is equivalent to a vector of a
single entry.</p>

<p style="margin-top: 1em">string:
&quot;[char-sequence]&quot;</p>

<p style="margin-top: 1em">&rsquo;[char-sequence]&rsquo;
&quot;&quot;&quot;[multi-line
char-sequence]&quot;&quot;&quot; (rm initial and final
newlines) &rsquo;&rsquo;&rsquo;[multi-line
char-sequence]&rsquo;&rsquo;&rsquo; (kp initial and final
newlines)</p>

<p style="margin-top: 1em">number:
[sign]digits[.digits][(e|E)[sign]digits] boolean: true|false
tree: Newick Format vector: { value, value, ...}</p>

<p style="margin-top: 1em">OPTIONS</p>

<p style="margin-top: 1em">Name Type Description</p>


<p style="margin-top: 1em">---------------------------------</p>

<p style="margin-top: 1em">Tree VT phylogeny</p>

<p style="margin-top: 1em">TreeScale <br>
N coefficient to scale branch lengths by</p>

<p style="margin-top: 1em">Sequence <br>
VS root sequences</p>

<p style="margin-top: 1em">Length VN length of generated
root sequences</p>

<p style="margin-top: 1em">Rates VVN rate of evolution of
each root nucleotide</p>

<p style="margin-top: 1em">Model S model of evolution:
GTR|JC|K2P|K3P|HKY|F81|F84|TN</p>

<p style="margin-top: 1em">Freqs VN nucleotide (ACGT)
frequencies</p>

<p style="margin-top: 1em">Params VN parameters for the
model of evolution</p>

<p style="margin-top: 1em">Width N block width for indels
and recombination</p>

<p style="margin-top: 1em">Scale VN block position
scales</p>

<p style="margin-top: 1em">Gamma VN coefficients of
variance for rate heterogenity</p>

<p style="margin-top: 1em">Alpha VN shape parameters</p>

<p style="margin-top: 1em">Iota VN proportions of invariant
sites</p>

<p style="margin-top: 1em">GapModel <br>
VS models of indel formation: NB|PL|US</p>

<p style="margin-top: 1em">Lambda VN rates of indel
formation</p>

<p style="margin-top: 1em">GapParams <br>
VVN parameter for the indel model</p>

<p style="margin-top: 1em">Reps N number of data sets to
output</p>

<p style="margin-top: 1em">File S output file</p>

<p style="margin-top: 1em">Format S output format:
Fasta|Nexus|Phylip|Clustal</p>

<p style="margin-top: 1em">GapSingleChar <br>
B output gaps as a single character</p>

<p style="margin-top: 1em">GapPlus <br>
B distinguish insertions from deletions in alignment</p>

<p style="margin-top: 1em">KeepFlank <br>
N undeletable flanking regions N nucs from sequence</p>

<p style="margin-top: 1em">KeepEmpty <br>
B preserve empty columns in final alignment</p>

<p style="margin-top: 1em">LowerCase <br>
B output sequences in lowercase</p>

<p style="margin-top: 1em">Translate <br>
B translate outputed sequences to amino acids</p>

<p style="margin-top: 1em">Seed VN
pseudo-random-number-generator seed (integers)</p>

<p style="margin-top: 1em">Out.Block.Head <br>
S string to insert at the start of the output</p>

<p style="margin-top: 1em">Out.Block.Tail <br>
S string to insert at the end of the output</p>

<p style="margin-top: 1em">Out.Block.Before S <br>
string to insert before a sequence set in the output</p>

<p style="margin-top: 1em">Out.Block.After <br>
S string to insert after a sequence set in the output</p>

<p style="margin-top: 1em">Out.Subst <br>
B do variable subsitution in Out.Block.*</p>

<p style="margin-top: 1em">LICENSE <br>
This program is free software; you can redistribute it
and/or modify it under the terms of the GNU General Public
License as published by the Free Software Foundation; either
<br>
version 2 of the License, or (at your option) any later
version.</p>

<p style="margin-top: 1em">This program is distributed in
the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PUR&acirc; <br>
POSE. See the GNU General Public License for more
details.</p>

<p style="margin-top: 1em">You should have received a copy
of the GNU General Public License along with this program;
if not, write to the Free Software Foundation, Inc., 59
Temple Place, Suite 330, Bos&acirc; <br>
ton, MA 02111-1307 USA</p>

<p style="margin-top: 1em">Send Bug Reports to
reed@scit.us.</p>

<p style="margin-top: 1em">dawg 1.2-release December 2015
DAWG(1)</p>
<hr>
</body>
</html>
