<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:04:00 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DIALIGN(1) User Manual DIALIGN(1)</p>

<p style="margin-top: 1em">NAME <br>
dialign2-2 - Multiple alignment program using the
segment-to-segment approach</p>

<p style="margin-top: 1em">SYNOPSIS <br>
dialign2-2 [options] [seq_file]</p>

<p style="margin-top: 1em">seq_file is the name of the
input sequence file; this must be a multiple FASTA file (all
sequences in one file).</p>

<p style="margin-top: 1em">DESCRIPTION <br>
dialign2-2 is a program that constructs alignments from
gapfree pairs of similar segments of the sequences. If
(possibly) coding nucleic acid sequences are to be aligned,
DIALIGN <br>
optionally translates the compared &lsquo;nucleic acid
segments&Acirc;&acute; to &lsquo;peptide
segments&Acirc;&acute; according to the genetic code --
without presupposing any of the three possible reading
frames, <br>
so all combinations of reading frames get checked for
significant similarity.</p>

<p style="margin-top: 1em">By default, DIALIGN creates a
single file containing</p>

<p style="margin-top: 1em">&Acirc;&middot; An alignment of
the input sequences in DIALIGN format.</p>

<p style="margin-top: 1em">&Acirc;&middot; The same
alignment in FASTA format.</p>

<p style="margin-top: 1em">&Acirc;&middot; A sequence tree
in PHYLIP format. This tree is constructed by applying the
UPGMA clustering method to the DIALIGN similarity scores. It
roughly reflects the different degrees <br>
of similarity among sequences. For detailed phylogenetic
analysis, we recommend the usual methods for phylogenetic
reconstruction.</p>

<p style="margin-top: 1em">The format of the output files
is documented in /usr/share/doc/dialign/USER_GUIDE.gz. The
FASTA, CLUSTALW and MSF output formats are optionally
available (see OPTIONS).</p>

<p style="margin-top: 1em">OPTIONS <br>
-afc <br>
Creates additional output file &quot;*.afc&quot; containing
data of all fragments considered for alignment. WARNING:
this file can be HUGE!</p>

<p style="margin-top: 1em">-afc_v <br>
Like &quot;-afc&quot; but verbose: fragments are explicitly
printed. WARNING: this file can be EVEN BIGGER!</p>

<p style="margin-top: 1em">-anc <br>
Anchored alignment. Requires a file seq_file.anc containing
anchor points.</p>

<p style="margin-top: 1em">-cs <br>
If segments are translated, not only the &lsquo;Watson
strand&Acirc;&acute; but also the &lsquo;Crick
strand&Acirc;&acute; is looked at.</p>

<p style="margin-top: 1em">-cw <br>
Additional output file in CLUSTAL W format.</p>

<p style="margin-top: 1em">-ds <br>
&lsquo;DNA alignment speed up&Acirc;&acute;. Non-translated
nucleic acid fragments are taken into account only if they
start with at least two matches. Speeds up DNA alignment at
the expense of <br>
sensitivity.</p>

<p style="margin-top: 1em">-fa <br>
Additional output file in FASTA format.</p>

<p style="margin-top: 1em">-ff <br>
Creates file *.frg containing information about all
fragments that are part of the respective optimal pairwise
alignmnets plus information about consistency in the
multiple <br>
alignment.</p>

<p style="margin-top: 1em">-fn out_file <br>
Output files are named out_file.extension.</p>

<p style="margin-top: 1em">-fop <br>
Creates file *.fop containing coordinates of all fragments
that are part of the respective pairwise alignments.</p>

<p style="margin-top: 1em">-fsm <br>
Creates file *.fsm containing coordinates of all fragments
that are part of the final alignment</p>

<p style="margin-top: 1em">-iw <br>
Overlap weights switched off (by default, overlap weights
are used if up to 35 sequences are aligned). This option
speeds up the alignment but may lead to reduced alignment
<br>
quality.</p>

<p style="margin-top: 1em">-lgs <br>
&lsquo;Long genomic sequences&Acirc;&acute; - combines the
following options: -ma, -thr 2, -lmax 30, -smin 8, -nta,
-ff, -fop, -ff, -cs, -ds, -pst.</p>

<p style="margin-top: 1em">-lgs_t <br>
Like &quot;-lgs&quot; but with all segment pairs assessed at
the peptide level (rather than &Acirc;&acute;mixed
alignments&Acirc;&acute; as with the &quot;-lgs&quot;
option). Therefore faster than -lgs but not very <br>
sensitive for non-coding regions.</p>

<p style="margin-top: 1em">-lmax x <br>
Maximum fragment length = x (default: x = 40 or x = 120 for
&lsquo;translated&Acirc;&acute; fragments). Shorter x speeds
up the program but may affect alignment quality.</p>

<p style="margin-top: 1em">-lo <br>
(Long Output) Additional file *.log with information abut
fragments selected for pairwise alignment and about
consistency in multi-alignment proceedure.</p>

<p style="margin-top: 1em">-ma <br>
&lsquo;mixed alignments&Acirc;&acute; consisting of
P-fragments and N-fragments if nucleic acid sequences are
aligned.</p>

<p style="margin-top: 1em">-mask <br>
Residues not belonging to selected fragments are replaced by
&lsquo;*&Acirc;&acute; characters in output alignment
(rather than being printed in lower-case characters)</p>

<p style="margin-top: 1em">-mat <br>
Creates file *mat with substitution counts derived from the
fragments that have been selected for alignment.</p>

<p style="margin-top: 1em">-mat_thr t <br>
Like &quot;-mat&quot; but only fragments with weight score
&gt; t are considered.</p>

<p style="margin-top: 1em">-max_link <br>
&quot;Maximum linkage&quot; clustering used to construct
sequence tree (instead of UPGMA).</p>

<p style="margin-top: 1em">-min_link <br>
&quot;Minimum linkage&quot; clustering used.</p>

<p style="margin-top: 1em">-mot <br>
&quot;Motif&quot; option.</p>

<p style="margin-top: 1em">-msf <br>
Separate output file in MSF format.</p>

<p style="margin-top: 1em">-n <br>
Input sequences are nucleic acid sequences. No translation
of fragments.</p>

<p style="margin-top: 1em">-nt <br>
Input sequences are nucleic acid sequences and
&lsquo;nucleic acid segments&Acirc;&acute; are translated to
&lsquo;peptide segments&Acirc;&acute;.</p>

<p style="margin-top: 1em">-nta <br>
&lsquo;No textual alignment&Acirc;&acute;. Textual alignment
suppressed. This option makes sense if other output files
are of intrest -- e.g. the fragment files created with -ff,
-fop, -fsm or <br>
-lo.</p>

<p style="margin-top: 1em">-o <br>
Fast version, resulting alignments may be slightly
different.</p>

<p style="margin-top: 1em">-ow <br>
Overlap weights enforced (By default, overlap weights are
used only if up to 35 sequences are aligned since
calculating overlap weights is time consuming). Warning:
overlap <br>
weights generally improve alignment quality but the running
time increases in the order O(n^4) with the number of
sequences. This is why, by default, overlap weights are used
<br>
only for sequence sets with &lt; 36 sequences.</p>

<p style="margin-top: 1em">-pst <br>
&quot;Print status&quot;. Creates and updates a file *.sta
with information about the current status of the program
run. This option is recommended if large data sets are
aligned since <br>
it allows the user to estimate the remaining running
time.</p>

<p style="margin-top: 1em">-smin x <br>
Minimum similarity value for first residue pair (or codon
pair) in fragments. Speeds up protein alignment or alignment
of translated DNA fragments at the expense of <br>
sensitivity.</p>

<p style="margin-top: 1em">-stars x <br>
Maximum number of &lsquo;*&Acirc;&acute; characters
indicating degree of local similarity among sequences. By
default, no stars are used but numbers between 0 and 9,
instead.</p>

<p style="margin-top: 1em">-stdo <br>
Results written to standard output.</p>

<p style="margin-top: 1em">-ta <br>
Standard textual alignment printed (overrides suppression of
textual alignments in special options, e.g. -lgs).</p>

<p style="margin-top: 1em">-thr x <br>
Threshold T = x.</p>

<p style="margin-top: 1em">-xfr <br>
&quot;Exclude fragments&quot;. List of fragments can be
specified that are NOT considered for pairwise
alignment.</p>

<p style="margin-top: 1em">General remark: If contradictory
options are used, subsequent options override previous ones,
e.g.: dialign2-2 -nt -n seq_file runs the program with the
&quot;-n&quot; option (no <br>
translation!), while dialign2-2 -n -nt seq_file runs it with
the &quot;-nt&quot; option (translation!).</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation is in /usr/share/doc/dialign/.</p>

<p style="margin-top: 1em">The website of dialign:
http://dialign.gobics.de/</p>

<p style="margin-top: 1em">DIALIGN2 has been re-implemented
in dialign-tx(1). See http://dialign-tx.gobics.de/</p>

<p style="margin-top: 1em">ENVIRONMENT VARIABLES <br>
You can create an environment variable
&lsquo;DIALIGN2_DIR&Acirc;&acute; pointing to a directory
where the substitution matrices are (see FILES). When
installed from the Debian package, it is not <br>
necessary to set this environnement variable to run
DIALIGN.</p>

<p style="margin-top: 1em">FILES <br>
DIALIGN2 needs the files tp400_dna, tp400_prot, tp400_trans
and BLOSUM. When DIALIGN is installed from the Debian
package, they are stored in /usr/share/dialign/.</p>

<p style="margin-top: 1em">DIALIGN 2 uses the BLOSUM62
amino acid substitution matrix. In the current version, it
is NOT possible to replace BLOSUM62 by other similarity
matrices.</p>

<p style="margin-top: 1em">REFERENCE <br>
B. Morgenstern (1999). DIALIGN 2: improvement of the
segment-to-segment approach to multiple sequence alignment.
Bioinformatics 15, 211 - 218. Public research assisted by
DIALIGN <br>
should cite this article.</p>

<p style="margin-top: 1em">AUTHORS <br>
Burkhard Morgenstern &lt;bmorgen@gwdg.de&gt; <br>
Author of DIALIGN</p>

<p style="margin-top: 1em">Said Abdeddaim <br>
Author of DIALIGN</p>

<p style="margin-top: 1em">Charles Plessy
&lt;plessy@debian.org&gt; <br>
Wrote this manpage</p>

<p style="margin-top: 1em">COPYRIGHT <br>
DIALIGN was written by Burkhard Morgenstern and Said
Abdeddaim at University of Bielefeld (FSPM and International
Graduate School in Bioinformatics and Genome Research), GSF
<br>
(ISG, IBB, MIPS/IBI), North Carolina State University,
Universite de Rouen, MPI fuer Biochemie (Martinsried),
University of Goettingen, Institute of Microbiology and
Genetics.</p>

<p style="margin-top: 1em">This manual page was adapted
from the DIALIGN manual by Charles Plessy
&lt;plessy@debian.org&gt; for the Debian system (but may be
used by others). Permission is granted to copy, <br>
distribute and/or modify this document under the terms of
the GNU Lesser General Public License, Version 2.1 any later
version published by the Free Software Foundation.</p>

<p style="margin-top: 1em">On Debian systems, the complete
text of the GNU Lesser General Public License can be found
in /usr/share/common-licenses/LGPL.</p>

<p style="margin-top: 1em">Copyright &Acirc;&copy; 1999
Burkhard Morgenstern (for DIALIGN) <br>
Copyright &Acirc;&copy; 2006, 2007, 2008 Charles Plessy (for
this manpage)</p>

<p style="margin-top: 1em">dialign2-2 2.2.1 avril 9, 2006
DIALIGN(1)</p>
<hr>
</body>
</html>
