<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DIALIGN-TX(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DIALIGN-TX(1)</td>
    <td class="head-vol">DIALIGN-TX: Parameters</td>
    <td class="head-rtitle">DIALIGN-TX(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
dialign-tx - Segment-based multiple sequence alignment
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 11.00ex; text-indent: -11.00ex;"><b>dialign-tx</b>
  [ <b>OPTIONS</b>] {<i>conf-directory</i>} {<i>fasta-file</i>}
  [<i>fasta-out-file</i>]</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
DIALIGN-TX is an improved algorithm for segment-based multiple protein
  alignments. DIALIGN-TX is a complete reimplementation of the segment-base
  approach including several new improvements and heuristics that significantly
  enhance the quality of the output alignments compared to DIALIGN 2.2. This
  significant superiority has been observed on local as well on global alignment
  benchmarks.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<b>-d</b>
<div style="margin-left: 4.00ex;">Debug-Mode [DEFAULT 0]
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">0 no debug statements</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">1 debugs the current phase of the processing</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">2 very loquacious debugging</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">5 hardcore debugging</dd>
</dl>
</div>
<div class="Pp"></div>
<b>-s</b>
<div style="margin-left: 4.00ex;">Maximum amount of input sequences [DEFAULT
  5000].</div>
<div class="Pp"></div>
<b>-a</b>
<div style="margin-left: 4.00ex;">Maximum number of characters per line in a
  FASTA file [DEFAULT 100].</div>
<div class="Pp"></div>
<b>-c</b>
<div style="margin-left: 4.00ex;">Maximum amount of characters per line when
  printing a sequence [DEFAULT 80].</div>
<div class="Pp"></div>
<b>-l</b>
<div style="margin-left: 4.00ex;">sensitivity mode, the higher the level the
  less likely spurious random fragments are aligned in local alignments [DEFAULT
  0]
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">0 switched off</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">1 level-1, reduced sensitivity</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">2 level-2, strongly reduced sensitivity</dd>
</dl>
</div>
<div class="Pp"></div>
<b>-m</b>
<div style="margin-left: 4.00ex;">Score matrix file name (in the configuration
  directory) [DEFAULT PROTEIN: BLOSUM.scr] / [DEFAULT DNA:
  dna_matrix.scr].</div>
<div class="Pp"></div>
<b>-w</b>
<div style="margin-left: 4.00ex;">Defines the minimum weight when the weight
  formula is changed to 1-pow(1-prob, factor) [DEFAULT 0.000000065].</div>
<div class="Pp"></div>
<b>-p</b>
<div style="margin-left: 4.00ex;">Probability distribution file name (in the
  configuration directory) [DEFAULT PROTEIN: BLOSUM.diag_prob_t10] / [DEFAULT
  DNA: dna_diag_prob_100_exp_550000].</div>
<div class="Pp"></div>
<b>-v</b>
<div style="margin-left: 4.00ex;">Add to each score (to prevent negative values)
  [DEFAULT 0].</div>
<div class="Pp"></div>
<b>-t</b>
<div style="margin-left: 4.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
&#x201C;Even&#x201D; threshold for low score for sequences alignment [DEFAULT
  PROTEIN: 4] / [DEFAULT DNA: 0].</div>
<div class="Pp"></div>
<b>-n</b>
<div style="margin-left: 4.00ex;">Maximum number of consecutive positions for
  window containing low scoring positions [DEFAULT PROTEIN: 4] / [DEFAULT DNA:
  1].</div>
<div class="Pp"></div>
<b>-g</b>
<div style="margin-left: 4.00ex;">Global minimum fragment length for stop
  criterion [DEFAULT PROTEIN: 40] / [DEFAULT DNA: 1].</div>
<div class="Pp"></div>
<b>-m</b>
<div style="margin-left: 4.00ex;">Minimal allowed average score in frag window
  containing low scoring positions [DEFAULT PROTEIN: 4.0] / [DEFAULT DNA:
  0.9].</div>
<div class="Pp"></div>
<b>-o</b>
<div style="margin-left: 4.00ex;">Whether overlap weights are calculated or not
  [DEFAULT 0].</div>
<div class="Pp"></div>
<b>-f</b>
<div style="margin-left: 4.00ex;">Minimum fragment length [DEFAULT 1].</div>
<div class="Pp"></div>
<b>-r</b>
<div style="margin-left: 4.00ex;">Threshold weight to consider the fragment at
  all [DEFAULT 0.0].</div>
<div class="Pp"></div>
<b>-u</b>
<div style="margin-left: 4.00ex;">[DEFAULT 0]
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     1: Only use a sqrt(amount_of_seqs) stripe of neighbour sequences to
      calculate pairwise alignments (increase performance).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     0: All pairwise alignments will be calculated.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
</div>
<div class="Pp"></div>
<b>-A</b>
<div style="margin-left: 4.00ex;">Optional anchor file. [DEFAULT none]</div>
<div class="Pp"></div>
<b>-D</b>
<div style="margin-left: 4.00ex;">Input is DNA-sequence.</div>
<div class="Pp"></div>
<b>-T</b>
<div style="margin-left: 4.00ex;">Translate DNA into aminoacids from begin to
  end (length will be cut to mod 3 = 0).
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
Warning
<div>&#x00A0;</div>
Do not use <b>-D</b> with this option (Default values for PROTEIN input will be
  loaded).
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<div class="Pp"></div>
<b>-L</b>
<div style="margin-left: 4.00ex;">Compare only longest Open Reading Frame.
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
Warning
<div>&#x00A0;</div>
Do not use <b>-D</b> with this option (Default values for PROTEIN input will be
  loaded).
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<div class="Pp"></div>
<b>-O</b>
<div style="margin-left: 4.00ex;">Translate DNA to aminoacids, reading frame for
  each sequence calculated due to its longest ORF.
<div style="height: 1.00em;">&#x00A0;</div>
<div>&#x00A0;</div>
Warning
<div>&#x00A0;</div>
Do not use <b>-D</b> with this option (Default values for PROTEIN input will be
  loaded).
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<div class="Pp"></div>
<b>-P</b>
<div style="margin-left: 4.00ex;">Output in aminoacids, no retranslation of DNA
  sequences [DEFAULT: input = output].</div>
<div class="Pp"></div>
<b>-F</b>
<div style="margin-left: 4.00ex;">Fast mode (implies -l0, since it already
  significantly reduces sensitivity).</div>
<div class="Pp"></div>
<b>-C</b>
<div style="margin-left: 4.00ex;">Generate probability table saved in
  /usr/share/dialign-tx/prob_table and exit.</div>
<div class="Pp"></div>
<b>-H</b>, <b>-h</b>
<div style="margin-left: 4.00ex;">Print this message.</div>
<h1 class="Sh" title="Sh" id="FILES"><a class="selflink" href="#FILES">FILES</a></h1>
<i>/usr/share/dialign-tx</i>
<div style="margin-left: 4.00ex;">This is the default conf-directory that
  <b>dialign-tx</b> expects as its first argument, as supplied in the upstream
  sources.</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
DIALIGN-TX is a re-implementation of <b>dialign2-2</b>(1). (See
  http://dialign.gobics.de/ for more information about DIALIGN2).
<div class="Pp"></div>
The website of DIALIGN-TX is http://dialign-tx.gobics.de/
<h1 class="Sh" title="Sh" id="REFERENCES"><a class="selflink" href="#REFERENCES">REFERENCES</a></h1>
Amarendran R. Subramanian, Michael Kaufmann, Burkhard Morgenstern: DIALIGN-TX:
  improvement of the segment-based approach for multiple sequence alignment by
  combining greedy and progressive alignment strategies, Algorithms for
  Molecular Biology 3:6, 2008.
<div class="Pp"></div>
Amarendran R. Subramanian, Jan Weyer-Menkhoff, Michael Kaufmann, Burkhard
  Morgenstern: DIALIGN-T: An improved algorithm for segment-based multiple
  sequence alignment. BMC Bioinformatics 2005, 6:66.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<b>Amarendran R. Subramanian</b> &lt;subraman@informatik.uni-tuebingen.de&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Author of dialign-tx</dd>
</dl>
<div class="Pp"></div>
<b>Volker Menrad</b>
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Co-author of dialign-tx</dd>
</dl>
<div class="Pp"></div>
<b>Dorothea Emig</b>
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Co-author of dialign-tx</dd>
</dl>
<div class="Pp"></div>
<b>Charles Plessy</b> &lt;plessy@debian.org&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Converted this guide in DocBook XML for the Debian
      distribution.</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright &#x00A9; 2004, 2005, 2006, 2007, 2008 Amarendran R. Subramanian
  (DIALIGN-TX)
<div>&#x00A0;</div>
Copyright &#x00A9; 2004 Volker Menrad (DIALIGN-TX)
<div>&#x00A0;</div>
Copyright &#x00A9; 2004 Dorothea Emig (DIALIGN-TX)
<div>&#x00A0;</div>
Copyright &#x00A9; 2007, 2008 Charles Plessy (This document and its XML source.)
<div>&#x00A0;</div>
<div class="Pp"></div>
DIALIGN-TX is free software; you can redistribute it and/or modify it under the
  terms of the GNU Lesser General Public License as published by the Free
  Software Foundation; either version 2.1 of the License, or (at your option)
  any later version.
<div class="Pp"></div>
DIALIGN-TX is distributed in the hope that it will be useful, but WITHOUT ANY
  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
  A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
  details.
<div class="Pp"></div>
You should have received a copy of the GNU Lesser General Public License along
  with this library; if not, write to the Free Software Foundation, Inc., 51
  Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
<div class="Pp"></div>
On Debian system, a copy of the GNU Lesser General Public License is available
  in <i>/usr/share/common-licences</i>.
<div class="Pp"></div>
This documentation and its XML source file can be used, modified and
  redistributed under the same terms as DIALIGN-TX itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">12/15/2008</td>
    <td class="foot-os">dialign-tx 1.0.2</td>
  </tr>
</table>
</body>
</html>
