<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DIALIGN(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DIALIGN(1)</td>
    <td class="head-vol">User Manual</td>
    <td class="head-rtitle">DIALIGN(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
dialign2-2 - Multiple alignment program using the segment-to-segment approach
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 11.00ex; text-indent: -11.00ex;"><b>dialign2-2</b>
  [options] [seq_file]</div>
<div class="Pp"></div>
<i>seq_file</i> is the name of the input sequence file; this must be a multiple
  FASTA file (all sequences in one file).
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>dialign2-2</b> is a program that constructs alignments from gapfree pairs of
  similar segments of the sequences. If (possibly) coding nucleic acid sequences
  are to be aligned, DIALIGN optionally translates the compared `nucleic acid
  segments&#x00B4; to `peptide segments&#x00B4; according to the genetic code --
  without presupposing any of the three possible reading frames, so all
  combinations of reading frames get checked for significant similarity.
<div class="Pp"></div>
By default, DIALIGN creates a single file containing
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;An alignment of the input sequences in
  DIALIGN format.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;The same alignment in FASTA
  format.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;A sequence tree in PHYLIP format. This
  tree is constructed by applying the UPGMA clustering method to the DIALIGN
  similarity scores. It roughly reflects the different degrees of similarity
  among sequences. For detailed phylogenetic analysis, we recommend the usual
  methods for phylogenetic reconstruction.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The format of the output files is documented in
  <i>/usr/share/doc/dialign/USER_GUIDE.gz</i>. The FASTA, CLUSTALW and MSF
  output formats are optionally available (see OPTIONS).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<b>-afc</b>
<div style="margin-left: 4.00ex;">Creates additional output file
  &quot;*.afc&quot; containing data of all fragments considered for alignment.
  WARNING: this file can be HUGE!</div>
<div class="Pp"></div>
<b>-afc_v</b>
<div style="margin-left: 4.00ex;">Like &quot;<b>-afc</b>&quot; but verbose:
  fragments are explicitly printed. WARNING: this file can be EVEN BIGGER!</div>
<div class="Pp"></div>
<b>-anc</b>
<div style="margin-left: 4.00ex;">Anchored alignment. Requires a file
  <i>seq_file.anc</i> containing anchor points.</div>
<div class="Pp"></div>
<b>-cs</b>
<div style="margin-left: 4.00ex;">If segments are translated, not only the
  `Watson strand&#x00B4; but also the `Crick strand&#x00B4; is looked at.</div>
<div class="Pp"></div>
<b>-cw</b>
<div style="margin-left: 4.00ex;">Additional output file in CLUSTAL W
  format.</div>
<div class="Pp"></div>
<b>-ds</b>
<div style="margin-left: 4.00ex;">`DNA alignment speed up&#x00B4;.
  Non-translated nucleic acid fragments are taken into account only if they
  start with at least two matches. Speeds up DNA alignment at the expense of
  sensitivity.</div>
<div class="Pp"></div>
<b>-fa</b>
<div style="margin-left: 4.00ex;">Additional output file in FASTA format.</div>
<div class="Pp"></div>
<b>-ff</b>
<div style="margin-left: 4.00ex;">Creates file <i>*.frg</i> containing
  information about all fragments that are part of the respective optimal
  pairwise alignmnets plus information about consistency in the multiple
  alignment.</div>
<div class="Pp"></div>
<b>-fn </b><b></b><i>out_file</i>
<div style="margin-left: 4.00ex;">Output files are named
  <i>out_file.extension</i>.</div>
<div class="Pp"></div>
<b>-fop</b>
<div style="margin-left: 4.00ex;">Creates file <i>*.fop</i> containing
  coordinates of all fragments that are part of the respective pairwise
  alignments.</div>
<div class="Pp"></div>
<b>-fsm</b>
<div style="margin-left: 4.00ex;">Creates file <i>*.fsm</i> containing
  coordinates of all fragments that are part of the final alignment</div>
<div class="Pp"></div>
<b>-iw</b>
<div style="margin-left: 4.00ex;">Overlap weights switched off (by default,
  overlap weights are used if up to 35 sequences are aligned). This option
  speeds up the alignment but may lead to reduced alignment quality.</div>
<div class="Pp"></div>
<b>-lgs</b>
<div style="margin-left: 4.00ex;">`Long genomic sequences&#x00B4; - combines the
  following options: <b>-ma</b>, <b>-thr </b><b></b><i>2</i>, <b>-lmax
  </b><b></b> <i>30</i>, <b>-smin </b><b></b><i>8</i>, <b>-nta</b>, <b>-ff</b>,
  <b>-fop</b>, <b>-ff</b>, <b>-cs</b>, <b>-ds</b>, <b>-pst</b>.</div>
<div class="Pp"></div>
<b>-lgs_t</b>
<div style="margin-left: 4.00ex;">Like &quot;<b>-lgs</b>&quot; but with all
  segment pairs assessed at the peptide level (rather than &#x00B4;mixed
  alignments&#x00B4; as with the &quot;-lgs&quot; option). Therefore faster than
  -lgs but not very sensitive for non-coding regions.</div>
<div class="Pp"></div>
<b>-lmax </b><b></b><i>x</i>
<div style="margin-left: 4.00ex;">Maximum fragment length = <i>x</i> (default:
  <i>x</i> = 40 or <i>x</i> = 120 for `translated&#x00B4; fragments). Shorter
  <i>x</i> speeds up the program but may affect alignment quality.</div>
<div class="Pp"></div>
<b>-lo</b>
<div style="margin-left: 4.00ex;">(Long Output) Additional file <i>*.log</i>
  with information abut fragments selected for pairwise alignment and about
  consistency in multi-alignment proceedure.</div>
<div class="Pp"></div>
<b>-ma</b>
<div style="margin-left: 4.00ex;">`mixed alignments&#x00B4; consisting of
  P-fragments and N-fragments if nucleic acid sequences are aligned.</div>
<div class="Pp"></div>
<b>-mask</b>
<div style="margin-left: 4.00ex;">Residues not belonging to selected fragments
  are replaced by `*&#x00B4; characters in output alignment (rather than being
  printed in lower-case characters)</div>
<div class="Pp"></div>
<b>-mat</b>
<div style="margin-left: 4.00ex;">Creates file <i>*mat</i> with substitution
  counts derived from the fragments that have been selected for alignment.</div>
<div class="Pp"></div>
<b>-mat_thr </b><b></b><i>t</i>
<div style="margin-left: 4.00ex;">Like &quot;-mat&quot; but only fragments with
  weight score &gt; <i>t</i> are considered.</div>
<div class="Pp"></div>
<b>-max_link</b>
<div style="margin-left: 4.00ex;">&quot;Maximum linkage&quot; clustering used to
  construct sequence tree (instead of UPGMA).</div>
<div class="Pp"></div>
<b>-min_link</b>
<div style="margin-left: 4.00ex;">&quot;Minimum linkage&quot; clustering
  used.</div>
<div class="Pp"></div>
<b>-mot</b>
<div style="margin-left: 4.00ex;">&quot;Motif&quot; option.</div>
<div class="Pp"></div>
<b>-msf</b>
<div style="margin-left: 4.00ex;">Separate output file in MSF format.</div>
<div class="Pp"></div>
<b>-n</b>
<div style="margin-left: 4.00ex;">Input sequences are nucleic acid sequences. No
  translation of fragments.</div>
<div class="Pp"></div>
<b>-nt</b>
<div style="margin-left: 4.00ex;">Input sequences are nucleic acid sequences and
  `nucleic acid segments&#x00B4; are translated to `peptide
  segments&#x00B4;.</div>
<div class="Pp"></div>
<b>-nta</b>
<div style="margin-left: 4.00ex;">`No textual alignment&#x00B4;. Textual
  alignment suppressed. This option makes sense if other output files are of
  intrest -- e.g. the fragment files created with <b>-ff</b>, <b>-fop</b>,
  <b>-fsm</b> or <b>-lo</b>.</div>
<div class="Pp"></div>
<b>-o</b>
<div style="margin-left: 4.00ex;">Fast version, resulting alignments may be
  slightly different.</div>
<div class="Pp"></div>
<b>-ow</b>
<div style="margin-left: 4.00ex;">Overlap weights enforced (By default, overlap
  weights are used only if up to 35 sequences are aligned since calculating
  overlap weights is time consuming). Warning: overlap weights generally improve
  alignment quality but the running time increases in the order O(n^4) with the
  number of sequences. This is why, by default, overlap weights are used only
  for sequence sets with &lt; 36 sequences.</div>
<div class="Pp"></div>
<b>-pst</b>
<div style="margin-left: 4.00ex;">&quot;Print status&quot;. Creates and updates
  a file <i>*.sta</i> with information about the current status of the program
  run. This option is recommended if large data sets are aligned since it allows
  the user to estimate the remaining running time.</div>
<div class="Pp"></div>
<b>-smin </b><b></b><i>x</i>
<div style="margin-left: 4.00ex;">Minimum similarity value for first residue
  pair (or codon pair) in fragments. Speeds up protein alignment or alignment of
  translated DNA fragments at the expense of sensitivity.</div>
<div class="Pp"></div>
<b>-stars </b><b></b><i>x</i>
<div style="margin-left: 4.00ex;">Maximum number of `*&#x00B4; characters
  indicating degree of local similarity among sequences. By default, no stars
  are used but numbers between 0 and 9, instead.</div>
<div class="Pp"></div>
<b>-stdo</b>
<div style="margin-left: 4.00ex;">Results written to standard output.</div>
<div class="Pp"></div>
<b>-ta</b>
<div style="margin-left: 4.00ex;">Standard textual alignment printed (overrides
  suppression of textual alignments in special options, e.g. -lgs).</div>
<div class="Pp"></div>
<b>-thr </b><b></b><i>x</i>
<div style="margin-left: 4.00ex;">Threshold T = x.</div>
<div class="Pp"></div>
<b>-xfr</b>
<div style="margin-left: 4.00ex;">&quot;Exclude fragments&quot;. List of
  fragments can be specified that are NOT considered for pairwise
  alignment.</div>
<div class="Pp"></div>
General remark: If contradictory options are used, subsequent options override
  previous ones, e.g.: dialign2-2 -nt -n <i>seq_file</i> runs the program with
  the &quot;-n&quot; option (no translation!), while dialign2-2 -n -nt
  <i>seq_file</i> runs it with the &quot;-nt&quot; option (translation!).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The full documentation is in <i>/usr/share/doc/dialign/</i>.
<div class="Pp"></div>
The website of dialign: http://dialign.gobics.de/
<div class="Pp"></div>
DIALIGN2 has been re-implemented in <b>dialign-tx</b>(1). See
  http://dialign-tx.gobics.de/
<h1 class="Sh" title="Sh" id="ENVIRONMENT_VARIABLES"><a class="selflink" href="#ENVIRONMENT_VARIABLES">ENVIRONMENT
  VARIABLES</a></h1>
You can create an environment variable ` <b>DIALIGN2_DIR</b>&#x00B4; pointing to
  a directory where the substitution matrices are (see FILES). When installed
  from the Debian package, it is not necessary to set this environnement
  variable to run DIALIGN.
<h1 class="Sh" title="Sh" id="FILES"><a class="selflink" href="#FILES">FILES</a></h1>
DIALIGN2 needs the files <i>tp400_dna</i>, <i>tp400_prot</i>, <i>tp400_trans</i>
  and <i>BLOSUM</i>. When DIALIGN is installed from the Debian package, they are
  stored in <i>/usr/share/dialign/</i>.
<div class="Pp"></div>
DIALIGN 2 uses the BLOSUM62 amino acid substitution matrix. In the current
  version, it is NOT possible to replace BLOSUM62 by other similarity matrices.
<h1 class="Sh" title="Sh" id="REFERENCE"><a class="selflink" href="#REFERENCE">REFERENCE</a></h1>
B. Morgenstern (1999). DIALIGN 2: improvement of the segment-to-segment approach
  to multiple sequence alignment. Bioinformatics 15, 211 - 218. Public research
  assisted by DIALIGN should cite this article.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<b>Burkhard Morgenstern</b> &lt;bmorgen@gwdg.de&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Author of DIALIGN</dd>
</dl>
<div class="Pp"></div>
<b>Said Abdeddaim</b>
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Author of DIALIGN</dd>
</dl>
<div class="Pp"></div>
<b>Charles Plessy</b> &lt;plessy@debian.org&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Wrote this manpage</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
DIALIGN was written by Burkhard Morgenstern and Said Abdeddaim at University of
  Bielefeld (FSPM and International Graduate School in Bioinformatics and Genome
  Research), GSF (ISG, IBB, MIPS/IBI), North Carolina State University,
  Universite de Rouen, MPI fuer Biochemie (Martinsried), University of
  Goettingen, Institute of Microbiology and Genetics.
<div class="Pp"></div>
This manual page was adapted from the DIALIGN manual by Charles Plessy
  &lt;plessy@debian.org&gt; for the Debian system (but may be used by others).
  Permission is granted to copy, distribute and/or modify this document under
  the terms of the GNU Lesser General Public License, Version 2.1 any later
  version published by the Free Software Foundation.
<div class="Pp"></div>
On Debian systems, the complete text of the GNU Lesser General Public License
  can be found in /usr/share/common-licenses/LGPL.
<div style="height: 1.00em;">&#x00A0;</div>
Copyright &#x00A9; 1999 Burkhard Morgenstern (for DIALIGN)
<div>&#x00A0;</div>
Copyright &#x00A9; 2006, 2007, 2008 Charles Plessy (for this manpage)
<div>&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">avril 9, 2006</td>
    <td class="foot-os">dialign2-2 2.2.1</td>
  </tr>
</table>
</body>
</html>
