<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:04:00 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DIALIGN-TX(1) DIALIGN-TX: Parameters DIALIGN-TX(1)</p>

<p style="margin-top: 1em">NAME <br>
dialign-tx - Segment-based multiple sequence alignment</p>

<p style="margin-top: 1em">SYNOPSIS <br>
dialign-tx [OPTIONS] {conf-directory} {fasta-file}
[fasta-out-file]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
DIALIGN-TX is an improved algorithm for segment-based
multiple protein alignments. DIALIGN-TX is a complete
reimplementation of the segment-base approach including
several new <br>
improvements and heuristics that significantly enhance the
quality of the output alignments compared to DIALIGN 2.2.
This significant superiority has been observed on local as
<br>
well on global alignment benchmarks.</p>

<p style="margin-top: 1em">OPTIONS <br>
-d <br>
Debug-Mode [DEFAULT 0]</p>

<p style="margin-top: 1em">0 no debug statements</p>

<p style="margin-top: 1em">1 debugs the current phase of
the processing</p>

<p style="margin-top: 1em">2 very loquacious debugging</p>

<p style="margin-top: 1em">5 hardcore debugging</p>

<p style="margin-top: 1em">-s <br>
Maximum amount of input sequences [DEFAULT 5000].</p>

<p style="margin-top: 1em">-a <br>
Maximum number of characters per line in a FASTA file
[DEFAULT 100].</p>

<p style="margin-top: 1em">-c <br>
Maximum amount of characters per line when printing a
sequence [DEFAULT 80].</p>

<p style="margin-top: 1em">-l <br>
sensitivity mode, the higher the level the less likely
spurious random fragments are aligned in local alignments
[DEFAULT 0]</p>

<p style="margin-top: 1em">0 switched off</p>

<p style="margin-top: 1em">1 level-1, reduced
sensitivity</p>

<p style="margin-top: 1em">2 level-2, strongly reduced
sensitivity</p>

<p style="margin-top: 1em">-m <br>
Score matrix file name (in the configuration directory)
[DEFAULT PROTEIN: BLOSUM.scr] / [DEFAULT DNA:
dna_matrix.scr].</p>

<p style="margin-top: 1em">-w <br>
Defines the minimum weight when the weight formula is
changed to 1-pow(1-prob, factor) [DEFAULT 0.000000065].</p>

<p style="margin-top: 1em">-p <br>
Probability distribution file name (in the configuration
directory) [DEFAULT PROTEIN: BLOSUM.diag_prob_t10] /
[DEFAULT DNA: dna_diag_prob_100_exp_550000].</p>

<p style="margin-top: 1em">-v <br>
Add to each score (to prevent negative values) [DEFAULT
0].</p>

<p style="margin-top: 1em">-t</p>

<p style="margin-top: 1em">&acirc;Even&acirc; threshold for
low score for sequences alignment [DEFAULT PROTEIN: 4] /
[DEFAULT DNA: 0].</p>

<p style="margin-top: 1em">-n <br>
Maximum number of consecutive positions for window
containing low scoring positions [DEFAULT PROTEIN: 4] /
[DEFAULT DNA: 1].</p>

<p style="margin-top: 1em">-g <br>
Global minimum fragment length for stop criterion [DEFAULT
PROTEIN: 40] / [DEFAULT DNA: 1].</p>

<p style="margin-top: 1em">-m <br>
Minimal allowed average score in frag window containing low
scoring positions [DEFAULT PROTEIN: 4.0] / [DEFAULT DNA:
0.9].</p>

<p style="margin-top: 1em">-o <br>
Whether overlap weights are calculated or not [DEFAULT
0].</p>

<p style="margin-top: 1em">-f <br>
Minimum fragment length [DEFAULT 1].</p>

<p style="margin-top: 1em">-r <br>
Threshold weight to consider the fragment at all [DEFAULT
0.0].</p>

<p style="margin-top: 1em">-u <br>
[DEFAULT 0]</p>

<p style="margin-top: 1em">1: Only use a
sqrt(amount_of_seqs) stripe of neighbour sequences to
calculate pairwise alignments (increase performance).</p>

<p style="margin-top: 1em">0: All pairwise alignments will
be calculated.</p>

<p style="margin-top: 1em">-A <br>
Optional anchor file. [DEFAULT none]</p>

<p style="margin-top: 1em">-D <br>
Input is DNA-sequence.</p>

<p style="margin-top: 1em">-T <br>
Translate DNA into aminoacids from begin to end (length will
be cut to mod 3 = 0).</p>

<p style="margin-top: 1em">Warning <br>
Do not use -D with this option (Default values for PROTEIN
input will be loaded).</p>

<p style="margin-top: 1em">-L <br>
Compare only longest Open Reading Frame.</p>

<p style="margin-top: 1em">Warning <br>
Do not use -D with this option (Default values for PROTEIN
input will be loaded).</p>

<p style="margin-top: 1em">-O <br>
Translate DNA to aminoacids, reading frame for each sequence
calculated due to its longest ORF.</p>

<p style="margin-top: 1em">Warning <br>
Do not use -D with this option (Default values for PROTEIN
input will be loaded).</p>

<p style="margin-top: 1em">-P <br>
Output in aminoacids, no retranslation of DNA sequences
[DEFAULT: input = output].</p>

<p style="margin-top: 1em">-F <br>
Fast mode (implies -l0, since it already significantly
reduces sensitivity).</p>

<p style="margin-top: 1em">-C <br>
Generate probability table saved in
/usr/share/dialign-tx/prob_table and exit.</p>

<p style="margin-top: 1em">-H, -h <br>
Print this message.</p>

<p style="margin-top: 1em">FILES <br>
/usr/share/dialign-tx <br>
This is the default conf-directory that dialign-tx expects
as its first argument, as supplied in the upstream
sources.</p>

<p style="margin-top: 1em">SEE ALSO <br>
DIALIGN-TX is a re-implementation of dialign2-2(1). (See
http://dialign.gobics.de/ for more information about
DIALIGN2).</p>

<p style="margin-top: 1em">The website of DIALIGN-TX is
http://dialign-tx.gobics.de/</p>

<p style="margin-top: 1em">REFERENCES <br>
Amarendran R. Subramanian, Michael Kaufmann, Burkhard
Morgenstern: DIALIGN-TX: improvement of the segment-based
approach for multiple sequence alignment by combining greedy
and <br>
progressive alignment strategies, Algorithms for Molecular
Biology 3:6, 2008.</p>

<p style="margin-top: 1em">Amarendran R. Subramanian, Jan
Weyer-Menkhoff, Michael Kaufmann, Burkhard Morgenstern:
DIALIGN-T: An improved algorithm for segment-based multiple
sequence alignment. BMC <br>
Bioinformatics 2005, 6:66.</p>

<p style="margin-top: 1em">AUTHORS <br>
Amarendran R. Subramanian
&lt;subraman@informatik.uni-tuebingen.de&gt; <br>
Author of dialign-tx</p>

<p style="margin-top: 1em">Volker Menrad <br>
Co-author of dialign-tx</p>

<p style="margin-top: 1em">Dorothea Emig <br>
Co-author of dialign-tx</p>

<p style="margin-top: 1em">Charles Plessy
&lt;plessy@debian.org&gt; <br>
Converted this guide in DocBook XML for the Debian
distribution.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 2004, 2005, 2006, 2007, 2008
Amarendran R. Subramanian (DIALIGN-TX) <br>
Copyright &Acirc;&copy; 2004 Volker Menrad (DIALIGN-TX) <br>
Copyright &Acirc;&copy; 2004 Dorothea Emig (DIALIGN-TX) <br>
Copyright &Acirc;&copy; 2007, 2008 Charles Plessy (This
document and its XML source.)</p>

<p style="margin-top: 1em">DIALIGN-TX is free software; you
can redistribute it and/or modify it under the terms of the
GNU Lesser General Public License as published by the Free
Software Foundation; <br>
either version 2.1 of the License, or (at your option) any
later version.</p>

<p style="margin-top: 1em">DIALIGN-TX is distributed in the
hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. <br>
See the GNU Lesser General Public License for more
details.</p>

<p style="margin-top: 1em">You should have received a copy
of the GNU Lesser General Public License along with this
library; if not, write to the Free Software Foundation,
Inc., 51 Franklin St, Fifth <br>
Floor, Boston, MA 02110-1301 USA</p>

<p style="margin-top: 1em">On Debian system, a copy of the
GNU Lesser General Public License is available in
/usr/share/common-licences.</p>

<p style="margin-top: 1em">This documentation and its XML
source file can be used, modified and redistributed under
the same terms as DIALIGN-TX itself.</p>

<p style="margin-top: 1em">dialign-tx 1.0.2 12/15/2008
DIALIGN-TX(1)</p>
<hr>
</body>
</html>
