<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:04:09 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DINDEL(1) User Commands DINDEL(1)</p>

<p style="margin-top: 1em">NAME <br>
dindel - finds of insertions and deletions from short
nucleotide sequences</p>

<p style="margin-top: 1em">DESCRIPTION <br>
[Required] : <br>
--ref arg <br>
fasta reference sequence (should be indexed with .fai
file)</p>

<p style="margin-top: 1em">--outputFile arg <br>
file-prefix for output results</p>

<p style="margin-top: 1em">[Required] Program option: <br>
--analysis arg (=indels) Analysis type: <br>
getCIGARindels: Extract indels from CIGARs of mapped reads,
and infer library insert size distributions indels: infer
indels realignCandidates: Realign/reposition candi&acirc;
<br>
dates in candidate file</p>

<p style="margin-top: 1em">[Required] BAM input. Choose one
of the following: <br>
--bamFile arg <br>
read alignment file (should be indexed)</p>

<p style="margin-top: 1em">--bamFiles arg <br>
file containing filepaths for BAMs to be jointly analysed
(not possible for --analysis==indels</p>

<p style="margin-top: 1em">[Required for analysis ==
getCIGARindels]: Region to be considered for extraction of
candidate indels.:</p>

<p style="margin-top: 1em">--region arg <br>
region to be analysed in format start-end, eg. 1000-2000</p>

<p style="margin-top: 1em">--tid arg <br>
target sequence (eg &rsquo;X&rsquo;)</p>

<p style="margin-top: 1em">[Required for analysis ==
indels]: <br>
--varFile arg <br>
file with candidate variants to be tested.</p>

<p style="margin-top: 1em">--varFileIsOneBased <br>
coordinates in varFile are one-based</p>

<p style="margin-top: 1em">Output options: <br>
--outputRealignedBAM <br>
output BAM file with realigned reads</p>

<p style="margin-top: 1em">--processRealignedBAM arg
ABSOLUTE path to script to process realigned BAM <br>
file</p>

<p style="margin-top: 1em">--quiet <br>
quiet output</p>

<p style="margin-top: 1em">parameters for analysis==indels
option: <br>
--doDiploid <br>
analyze data assuming a diploid sequence</p>

<p style="margin-top: 1em">--doPooled <br>
estimate haplotype frequencies using Bayesian EM algorithm.
May be applied to single individual and pools.</p>

<p style="margin-top: 1em">General algorithm parameters:
<br>
--faster <br>
use faster but less accurate ungapped read-haplotype
alignment model</p>

<p style="margin-top: 1em">--filterHaplotypes <br>
prefilter haplotypes based on coverage</p>

<p style="margin-top: 1em">--flankRefSeq arg (=2) <br>
#bases of reference sequence of indel region</p>

<p style="margin-top: 1em">--flankMaxMismatch arg (=2) <br>
max number of mismatches in indel region</p>

<p style="margin-top: 1em">--priorSNP arg (=0.001) <br>
prior probability of a SNP site</p>

<p style="margin-top: 1em">--priorIndel arg (=0.0001) <br>
prior probability of a detected indel not being a sequencing
error</p>

<p style="margin-top: 1em">--width arg (=60) <br>
number of bases to left and right of indel</p>

<p style="margin-top: 1em">--maxHap arg (=8) <br>
maximum number of haplotypes in likelihood computation</p>

<p style="margin-top: 1em">--maxRead arg (=10000) <br>
maximum number of reads in likelihood computation</p>

<p style="margin-top: 1em">--mapQualThreshold arg
(=0.98999999999999999) <br>
lower limit for read mapping quality</p>

<p style="margin-top: 1em">--capMapQualThreshold arg (=100)
<br>
upper limit for read mapping quality in observationmodel_old
(phred units)</p>

<p style="margin-top: 1em">--capMapQualFast arg (=45) <br>
cap mapping quality in alignment using fast ungapped
method</p>

<p style="margin-top: 1em">(WARNING: setting it too high
(&gt;50)</p>

<p style="margin-top: 1em">might result in significant
overcalling!)</p>

<p style="margin-top: 1em">--skipMaxHap arg (=200) <br>
skip computation if number of haplotypes exceeds this
number</p>

<p style="margin-top: 1em">--minReadOverlap arg (=20) <br>
minimum overlap between read and haplotype</p>

<p style="margin-top: 1em">--maxReadLength arg (=500) <br>
maximum length of reads</p>

<p style="margin-top: 1em">--minCount arg (=1) <br>
minimum number of WS observations of indel</p>

<p style="margin-top: 1em">--maxHapReadProd arg (=10000000)
<br>
skip if product of number of reads and haplotypes exceeds
this value</p>

<p style="margin-top: 1em">--changeINStoN <br>
change sequence of inserted sequence to &rsquo;N&rsquo;, so
that no penalty is incurred if a read mismatches the
inserted sequence</p>

<p style="margin-top: 1em">parameters for --pooled option:
<br>
--bayesa0 arg (=0.001) <br>
Dirichlet a0 parameter haplotype frequency prior</p>

<p style="margin-top: 1em">--bayesType arg (=singlevariant)
Bayesian EM program type (all or <br>
singlevariant or priorpersite)</p>

<p style="margin-top: 1em">General algorithm filtering
options: <br>
--checkAllCIGARs arg (=1) include all indels at the position
of the call site</p>

<p style="margin-top: 1em">--filterReadAux arg <br>
match string for exclusion of reads based on auxilary
information</p>

<p style="margin-top: 1em">Observation model parameters:
<br>
--pError arg (=0.00050000000000000001) <br>
probability of a read indel</p>

<p style="margin-top: 1em">--pMut arg
(=1.0000000000000001e-05) <br>
probability of a mutation in the read</p>

<p style="margin-top: 1em">--maxLengthIndel arg (=5) <br>
maximum length of a _sequencing error_ indel in read [not
for --faster option]</p>

<p style="margin-top: 1em">Library options: <br>
--libFile arg <br>
file with library insert histograms (as generated by
--analysis getCIGARindels)</p>

<p style="margin-top: 1em">Misc results analysis options:
<br>
--compareReadHap <br>
compare likelihood differences in reads against
haplotypes</p>

<p style="margin-top: 1em">--compareReadHapThreshold arg
(=0.5) difference threshold for viewing</p>

<p style="margin-top: 1em">--showEmpirical <br>
show empirical distribution over nucleotides</p>

<p style="margin-top: 1em">--showCandHap <br>
show candidate haplotypes for fast method</p>

<p style="margin-top: 1em">--showHapAlignments <br>
show for each haplotype which reads map to it</p>

<p style="margin-top: 1em">--showReads <br>
show reads</p>

<p style="margin-top: 1em">--inferenceMethod arg
(=empirical) <br>
inference method</p>

<p style="margin-top: 1em">--opl output likelihoods for
every read and haplotype</p>

<p style="margin-top: 1em">[Required] : <br>
--ref arg <br>
fasta reference sequence (should be indexed with .fai
file)</p>

<p style="margin-top: 1em">--outputFile arg <br>
file-prefix for output results</p>

<p style="margin-top: 1em">[Required] Program option: <br>
--analysis arg (=indels) Analysis type: <br>
getCIGARindels: Extract indels from CIGARs of mapped reads,
and infer library insert size distributions indels: infer
indels realignCandidates: Realign/reposition candi&acirc;
<br>
dates in candidate file</p>

<p style="margin-top: 1em">[Required] BAM input. Choose one
of the following: <br>
--bamFile arg <br>
read alignment file (should be indexed)</p>

<p style="margin-top: 1em">--bamFiles arg <br>
file containing filepaths for BAMs to be jointly analysed
(not possible for --analysis==indels</p>

<p style="margin-top: 1em">[Required for analysis ==
getCIGARindels]: Region to be considered for extraction of
candidate indels.:</p>

<p style="margin-top: 1em">--region arg <br>
region to be analysed in format start-end, eg. 1000-2000</p>

<p style="margin-top: 1em">--tid arg <br>
target sequence (eg &rsquo;X&rsquo;)</p>

<p style="margin-top: 1em">[Required for analysis ==
indels]: <br>
--varFile arg <br>
file with candidate variants to be tested.</p>

<p style="margin-top: 1em">--varFileIsOneBased <br>
coordinates in varFile are one-based</p>

<p style="margin-top: 1em">Output options: <br>
--outputRealignedBAM <br>
output BAM file with realigned reads</p>

<p style="margin-top: 1em">--processRealignedBAM arg
ABSOLUTE path to script to process realigned BAM <br>
file</p>

<p style="margin-top: 1em">--quiet <br>
quiet output</p>

<p style="margin-top: 1em">parameters for analysis==indels
option: <br>
--doDiploid <br>
analyze data assuming a diploid sequence</p>

<p style="margin-top: 1em">--doPooled <br>
estimate haplotype frequencies using Bayesian EM algorithm.
May be applied to single individual and pools.</p>

<p style="margin-top: 1em">General algorithm parameters:
<br>
--faster <br>
use faster but less accurate ungapped read-haplotype
alignment model</p>

<p style="margin-top: 1em">--filterHaplotypes <br>
prefilter haplotypes based on coverage</p>

<p style="margin-top: 1em">--flankRefSeq arg (=2) <br>
#bases of reference sequence of indel region</p>

<p style="margin-top: 1em">--flankMaxMismatch arg (=2) <br>
max number of mismatches in indel region</p>

<p style="margin-top: 1em">--priorSNP arg (=0.001) <br>
prior probability of a SNP site</p>

<p style="margin-top: 1em">--priorIndel arg (=0.0001) <br>
prior probability of a detected indel not being a sequencing
error</p>

<p style="margin-top: 1em">--width arg (=60) <br>
number of bases to left and right of indel</p>

<p style="margin-top: 1em">--maxHap arg (=8) <br>
maximum number of haplotypes in likelihood computation</p>

<p style="margin-top: 1em">--maxRead arg (=10000) <br>
maximum number of reads in likelihood computation</p>

<p style="margin-top: 1em">--mapQualThreshold arg
(=0.98999999999999999) <br>
lower limit for read mapping quality</p>

<p style="margin-top: 1em">--capMapQualThreshold arg (=100)
<br>
upper limit for read mapping quality in observationmodel_old
(phred units)</p>

<p style="margin-top: 1em">--capMapQualFast arg (=45) <br>
cap mapping quality in alignment using fast ungapped
method</p>

<p style="margin-top: 1em">(WARNING: setting it too high
(&gt;50)</p>

<p style="margin-top: 1em">might result in significant
overcalling!)</p>

<p style="margin-top: 1em">--skipMaxHap arg (=200) <br>
skip computation if number of haplotypes exceeds this
number</p>

<p style="margin-top: 1em">--minReadOverlap arg (=20) <br>
minimum overlap between read and haplotype</p>

<p style="margin-top: 1em">--maxReadLength arg (=500) <br>
maximum length of reads</p>

<p style="margin-top: 1em">--minCount arg (=1) <br>
minimum number of WS observations of indel</p>

<p style="margin-top: 1em">--maxHapReadProd arg (=10000000)
<br>
skip if product of number of reads and haplotypes exceeds
this value</p>

<p style="margin-top: 1em">--changeINStoN <br>
change sequence of inserted sequence to &rsquo;N&rsquo;, so
that no penalty is incurred if a read mismatches the
inserted sequence</p>

<p style="margin-top: 1em">parameters for --pooled option:
<br>
--bayesa0 arg (=0.001) <br>
Dirichlet a0 parameter haplotype frequency prior</p>

<p style="margin-top: 1em">--bayesType arg (=singlevariant)
Bayesian EM program type (all or <br>
singlevariant or priorpersite)</p>

<p style="margin-top: 1em">General algorithm filtering
options: <br>
--checkAllCIGARs arg (=1) include all indels at the position
of the call site</p>

<p style="margin-top: 1em">--filterReadAux arg <br>
match string for exclusion of reads based on auxilary
information</p>

<p style="margin-top: 1em">Observation model parameters:
<br>
--pError arg (=0.00050000000000000001) <br>
probability of a read indel</p>

<p style="margin-top: 1em">--pMut arg
(=1.0000000000000001e-05) <br>
probability of a mutation in the read</p>

<p style="margin-top: 1em">--maxLengthIndel arg (=5) <br>
maximum length of a _sequencing error_ indel in read [not
for --faster option]</p>

<p style="margin-top: 1em">Library options: <br>
--libFile arg <br>
file with library insert histograms (as generated by
--analysis getCIGARindels)</p>

<p style="margin-top: 1em">Misc results analysis options:
<br>
--compareReadHap <br>
compare likelihood differences in reads against
haplotypes</p>

<p style="margin-top: 1em">--compareReadHapThreshold arg
(=0.5) difference threshold for viewing</p>

<p style="margin-top: 1em">--showEmpirical <br>
show empirical distribution over nucleotides</p>

<p style="margin-top: 1em">--showCandHap <br>
show candidate haplotypes for fast method</p>

<p style="margin-top: 1em">--showHapAlignments <br>
show for each haplotype which reads map to it</p>

<p style="margin-top: 1em">--showReads <br>
show reads</p>

<p style="margin-top: 1em">--inferenceMethod arg
(=empirical) <br>
inference method</p>

<p style="margin-top: 1em">--opl output likelihoods for
every read and haplotype</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation for dindel you find referenced on
https://sites.google.com/site/keesalbers/soft/dindel</p>

<p style="margin-top: 1em">March 2016 DINDEL(1)</p>
<hr>
</body>
</html>
