<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DINDEL(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DINDEL(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">DINDEL(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
dindel - finds of insertions and deletions from short nucleotide sequences
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<h2 class="Ss" title="Ss" id="[Required]_:"><a class="selflink" href="#[Required]_:">[Required]
  :</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ref</b> arg</dt>
  <dd class="It-tag">fasta reference sequence (should be indexed with .fai
    file)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outputFile</b> arg</dt>
  <dd class="It-tag">file-prefix for output results</dd>
</dl>
<h2 class="Ss" title="Ss" id="[Required]_Program_option:"><a class="selflink" href="#[Required]_Program_option:">[Required]
  Program option:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--analysis</b> arg (=indels) Analysis type:</dt>
  <dd class="It-tag">getCIGARindels: Extract indels from CIGARs of mapped reads,
      and infer library insert size distributions indels: infer indels
      realignCandidates: Realign/reposition candidates in candidate file</dd>
</dl>
<h2 class="Ss" title="Ss" id="[Required]_BAM_input._Choose_one_of_the_following:"><a class="selflink" href="#[Required]_BAM_input._Choose_one_of_the_following:">[Required]
  BAM input. Choose one of the following:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bamFile</b> arg</dt>
  <dd class="It-tag">read alignment file (should be indexed)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bamFiles</b> arg</dt>
  <dd class="It-tag">file containing filepaths for BAMs to be jointly analysed
      (not possible for <b>--analysis</b>=<i>=indels</i></dd>
</dl>
<div class="Pp"></div>
[Required for analysis == getCIGARindels]: Region to be considered for
  extraction of candidate indels.:
<dl class="Bl-tag">
  <dt class="It-tag"><b>--region</b> arg</dt>
  <dd class="It-tag">region to be analysed in format start-end, eg.
    1000-2000</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tid</b> arg</dt>
  <dd class="It-tag">target sequence (eg 'X')</dd>
</dl>
<h2 class="Ss" title="Ss" id="[Required_for_analysis_==_indels]:"><a class="selflink" href="#[Required_for_analysis_==_indels]:">[Required
  for analysis == indels]:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--varFile</b> arg</dt>
  <dd class="It-tag">file with candidate variants to be tested.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--varFileIsOneBased</b></dt>
  <dd class="It-tag">coordinates in varFile are one-based</dd>
</dl>
<h2 class="Ss" title="Ss" id="Output_options:"><a class="selflink" href="#Output_options:">Output
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outputRealignedBAM</b></dt>
  <dd class="It-tag">output BAM file with realigned reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--processRealignedBAM</b> arg ABSOLUTE path to script to
    process realigned BAM</dt>
  <dd class="It-tag">file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quiet</b></dt>
  <dd class="It-tag">quiet output</dd>
</dl>
<h2 class="Ss" title="Ss" id="parameters_for_analysis==indels_option:"><a class="selflink" href="#parameters_for_analysis==indels_option:">parameters
  for analysis==indels option:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--doDiploid</b></dt>
  <dd class="It-tag">analyze data assuming a diploid sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--doPooled</b></dt>
  <dd class="It-tag">estimate haplotype frequencies using Bayesian EM algorithm.
      May be applied to single individual and pools.</dd>
</dl>
<h2 class="Ss" title="Ss" id="General_algorithm_parameters:"><a class="selflink" href="#General_algorithm_parameters:">General
  algorithm parameters:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--faster</b></dt>
  <dd class="It-tag">use faster but less accurate ungapped read-haplotype
      alignment model</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--filterHaplotypes</b></dt>
  <dd class="It-tag">prefilter haplotypes based on coverage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--flankRefSeq</b> arg (=2)</dt>
  <dd class="It-tag">#bases of reference sequence of indel region</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--flankMaxMismatch</b> arg (=2)</dt>
  <dd class="It-tag">max number of mismatches in indel region</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--priorSNP</b> arg (=0.001)</dt>
  <dd class="It-tag">prior probability of a SNP site</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--priorIndel</b> arg (=0.0001)</dt>
  <dd class="It-tag">prior probability of a detected indel not being a
      sequencing error</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--width</b> arg (=60)</dt>
  <dd class="It-tag">number of bases to left and right of indel</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxHap</b> arg (=8)</dt>
  <dd class="It-tag">maximum number of haplotypes in likelihood computation</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxRead</b> arg (=10000)</dt>
  <dd class="It-tag">maximum number of reads in likelihood computation</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mapQualThreshold</b> arg (=0.98999999999999999)</dt>
  <dd class="It-tag">lower limit for read mapping quality</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--capMapQualThreshold</b> arg (=100)</dt>
  <dd class="It-tag">upper limit for read mapping quality in
      observationmodel_old (phred units)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--capMapQualFast</b> arg (=45)</dt>
  <dd class="It-tag">cap mapping quality in alignment using fast ungapped
    method</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(WARNING: setting it too high (&gt;50)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">might result in significant overcalling!)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--skipMaxHap</b> arg (=200)</dt>
  <dd class="It-tag">skip computation if number of haplotypes exceeds this
      number</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minReadOverlap</b> arg (=20)</dt>
  <dd class="It-tag">minimum overlap between read and haplotype</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxReadLength</b> arg (=500)</dt>
  <dd class="It-tag">maximum length of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minCount</b> arg (=1)</dt>
  <dd class="It-tag">minimum number of WS observations of indel</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxHapReadProd</b> arg (=10000000)</dt>
  <dd class="It-tag">skip if product of number of reads and haplotypes exceeds
      this value</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--changeINStoN</b></dt>
  <dd class="It-tag">change sequence of inserted sequence to 'N', so that no
      penalty is incurred if a read mismatches the inserted sequence</dd>
</dl>
<h2 class="Ss" title="Ss" id="parameters_for_--pooled_option:"><a class="selflink" href="#parameters_for_--pooled_option:">parameters
  for --pooled option:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bayesa0</b> arg (=0.001)</dt>
  <dd class="It-tag">Dirichlet a0 parameter haplotype frequency prior</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bayesType</b> arg (=singlevariant) Bayesian EM program
    type (all or</dt>
  <dd class="It-tag">singlevariant or priorpersite)</dd>
</dl>
<h2 class="Ss" title="Ss" id="General_algorithm_filtering_options:"><a class="selflink" href="#General_algorithm_filtering_options:">General
  algorithm filtering options:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--checkAllCIGARs</b>
  arg (=1) include all indels at the position of the call site</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--filterReadAux</b> arg</dt>
  <dd class="It-tag">match string for exclusion of reads based on auxilary
      information</dd>
</dl>
<h2 class="Ss" title="Ss" id="Observation_model_parameters:"><a class="selflink" href="#Observation_model_parameters:">Observation
  model parameters:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pError</b> arg (=0.00050000000000000001)</dt>
  <dd class="It-tag">probability of a read indel</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pMut</b> arg (=1.0000000000000001e-05)</dt>
  <dd class="It-tag">probability of a mutation in the read</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxLengthIndel</b> arg (=5)</dt>
  <dd class="It-tag">maximum length of a _sequencing error_ indel in read [not
      for <b>--faster</b> option]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Library_options:"><a class="selflink" href="#Library_options:">Library
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--libFile</b> arg</dt>
  <dd class="It-tag">file with library insert histograms (as generated by
      <b>--analysis</b> getCIGARindels)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Misc_results_analysis_options:"><a class="selflink" href="#Misc_results_analysis_options:">Misc
  results analysis options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--compareReadHap</b></dt>
  <dd class="It-tag">compare likelihood differences in reads against
    haplotypes</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--compareReadHapThreshold</b>
  arg (=0.5) difference threshold for viewing</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showEmpirical</b></dt>
  <dd class="It-tag">show empirical distribution over nucleotides</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showCandHap</b></dt>
  <dd class="It-tag">show candidate haplotypes for fast method</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showHapAlignments</b></dt>
  <dd class="It-tag">show for each haplotype which reads map to it</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showReads</b></dt>
  <dd class="It-tag">show reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--inferenceMethod</b> arg (=empirical)</dt>
  <dd class="It-tag">inference method</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--opl</b></dt>
  <dd class="It-tag">output likelihoods for every read and haplotype</dd>
</dl>
<h2 class="Ss" title="Ss" id="[Required]_:"><a class="selflink" href="#[Required]_:">[Required]
  :</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ref</b> arg</dt>
  <dd class="It-tag">fasta reference sequence (should be indexed with .fai
    file)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outputFile</b> arg</dt>
  <dd class="It-tag">file-prefix for output results</dd>
</dl>
<h2 class="Ss" title="Ss" id="[Required]_Program_option:"><a class="selflink" href="#[Required]_Program_option:">[Required]
  Program option:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--analysis</b> arg (=indels) Analysis type:</dt>
  <dd class="It-tag">getCIGARindels: Extract indels from CIGARs of mapped reads,
      and infer library insert size distributions indels: infer indels
      realignCandidates: Realign/reposition candidates in candidate file</dd>
</dl>
<h2 class="Ss" title="Ss" id="[Required]_BAM_input._Choose_one_of_the_following:"><a class="selflink" href="#[Required]_BAM_input._Choose_one_of_the_following:">[Required]
  BAM input. Choose one of the following:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bamFile</b> arg</dt>
  <dd class="It-tag">read alignment file (should be indexed)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bamFiles</b> arg</dt>
  <dd class="It-tag">file containing filepaths for BAMs to be jointly analysed
      (not possible for <b>--analysis</b>=<i>=indels</i></dd>
</dl>
<div class="Pp"></div>
[Required for analysis == getCIGARindels]: Region to be considered for
  extraction of candidate indels.:
<dl class="Bl-tag">
  <dt class="It-tag"><b>--region</b> arg</dt>
  <dd class="It-tag">region to be analysed in format start-end, eg.
    1000-2000</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tid</b> arg</dt>
  <dd class="It-tag">target sequence (eg 'X')</dd>
</dl>
<h2 class="Ss" title="Ss" id="[Required_for_analysis_==_indels]:"><a class="selflink" href="#[Required_for_analysis_==_indels]:">[Required
  for analysis == indels]:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--varFile</b> arg</dt>
  <dd class="It-tag">file with candidate variants to be tested.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--varFileIsOneBased</b></dt>
  <dd class="It-tag">coordinates in varFile are one-based</dd>
</dl>
<h2 class="Ss" title="Ss" id="Output_options:"><a class="selflink" href="#Output_options:">Output
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outputRealignedBAM</b></dt>
  <dd class="It-tag">output BAM file with realigned reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--processRealignedBAM</b> arg ABSOLUTE path to script to
    process realigned BAM</dt>
  <dd class="It-tag">file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quiet</b></dt>
  <dd class="It-tag">quiet output</dd>
</dl>
<h2 class="Ss" title="Ss" id="parameters_for_analysis==indels_option:"><a class="selflink" href="#parameters_for_analysis==indels_option:">parameters
  for analysis==indels option:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--doDiploid</b></dt>
  <dd class="It-tag">analyze data assuming a diploid sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--doPooled</b></dt>
  <dd class="It-tag">estimate haplotype frequencies using Bayesian EM algorithm.
      May be applied to single individual and pools.</dd>
</dl>
<h2 class="Ss" title="Ss" id="General_algorithm_parameters:"><a class="selflink" href="#General_algorithm_parameters:">General
  algorithm parameters:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--faster</b></dt>
  <dd class="It-tag">use faster but less accurate ungapped read-haplotype
      alignment model</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--filterHaplotypes</b></dt>
  <dd class="It-tag">prefilter haplotypes based on coverage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--flankRefSeq</b> arg (=2)</dt>
  <dd class="It-tag">#bases of reference sequence of indel region</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--flankMaxMismatch</b> arg (=2)</dt>
  <dd class="It-tag">max number of mismatches in indel region</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--priorSNP</b> arg (=0.001)</dt>
  <dd class="It-tag">prior probability of a SNP site</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--priorIndel</b> arg (=0.0001)</dt>
  <dd class="It-tag">prior probability of a detected indel not being a
      sequencing error</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--width</b> arg (=60)</dt>
  <dd class="It-tag">number of bases to left and right of indel</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxHap</b> arg (=8)</dt>
  <dd class="It-tag">maximum number of haplotypes in likelihood computation</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxRead</b> arg (=10000)</dt>
  <dd class="It-tag">maximum number of reads in likelihood computation</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mapQualThreshold</b> arg (=0.98999999999999999)</dt>
  <dd class="It-tag">lower limit for read mapping quality</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--capMapQualThreshold</b> arg (=100)</dt>
  <dd class="It-tag">upper limit for read mapping quality in
      observationmodel_old (phred units)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--capMapQualFast</b> arg (=45)</dt>
  <dd class="It-tag">cap mapping quality in alignment using fast ungapped
    method</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(WARNING: setting it too high (&gt;50)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">might result in significant overcalling!)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--skipMaxHap</b> arg (=200)</dt>
  <dd class="It-tag">skip computation if number of haplotypes exceeds this
      number</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minReadOverlap</b> arg (=20)</dt>
  <dd class="It-tag">minimum overlap between read and haplotype</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxReadLength</b> arg (=500)</dt>
  <dd class="It-tag">maximum length of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minCount</b> arg (=1)</dt>
  <dd class="It-tag">minimum number of WS observations of indel</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxHapReadProd</b> arg (=10000000)</dt>
  <dd class="It-tag">skip if product of number of reads and haplotypes exceeds
      this value</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--changeINStoN</b></dt>
  <dd class="It-tag">change sequence of inserted sequence to 'N', so that no
      penalty is incurred if a read mismatches the inserted sequence</dd>
</dl>
<h2 class="Ss" title="Ss" id="parameters_for_--pooled_option:"><a class="selflink" href="#parameters_for_--pooled_option:">parameters
  for --pooled option:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bayesa0</b> arg (=0.001)</dt>
  <dd class="It-tag">Dirichlet a0 parameter haplotype frequency prior</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bayesType</b> arg (=singlevariant) Bayesian EM program
    type (all or</dt>
  <dd class="It-tag">singlevariant or priorpersite)</dd>
</dl>
<h2 class="Ss" title="Ss" id="General_algorithm_filtering_options:"><a class="selflink" href="#General_algorithm_filtering_options:">General
  algorithm filtering options:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--checkAllCIGARs</b>
  arg (=1) include all indels at the position of the call site</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--filterReadAux</b> arg</dt>
  <dd class="It-tag">match string for exclusion of reads based on auxilary
      information</dd>
</dl>
<h2 class="Ss" title="Ss" id="Observation_model_parameters:"><a class="selflink" href="#Observation_model_parameters:">Observation
  model parameters:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pError</b> arg (=0.00050000000000000001)</dt>
  <dd class="It-tag">probability of a read indel</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pMut</b> arg (=1.0000000000000001e-05)</dt>
  <dd class="It-tag">probability of a mutation in the read</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxLengthIndel</b> arg (=5)</dt>
  <dd class="It-tag">maximum length of a _sequencing error_ indel in read [not
      for <b>--faster</b> option]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Library_options:"><a class="selflink" href="#Library_options:">Library
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--libFile</b> arg</dt>
  <dd class="It-tag">file with library insert histograms (as generated by
      <b>--analysis</b> getCIGARindels)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Misc_results_analysis_options:"><a class="selflink" href="#Misc_results_analysis_options:">Misc
  results analysis options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--compareReadHap</b></dt>
  <dd class="It-tag">compare likelihood differences in reads against
    haplotypes</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--compareReadHapThreshold</b>
  arg (=0.5) difference threshold for viewing</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showEmpirical</b></dt>
  <dd class="It-tag">show empirical distribution over nucleotides</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showCandHap</b></dt>
  <dd class="It-tag">show candidate haplotypes for fast method</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showHapAlignments</b></dt>
  <dd class="It-tag">show for each haplotype which reads map to it</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showReads</b></dt>
  <dd class="It-tag">show reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--inferenceMethod</b> arg (=empirical)</dt>
  <dd class="It-tag">inference method</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--opl</b></dt>
  <dd class="It-tag">output likelihoods for every read and haplotype</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The full documentation for <b>dindel</b> you find referenced on
  https://sites.google.com/site/keesalbers/soft/dindel</div>
<table class="foot">
  <tr>
    <td class="foot-date">March 2016</td>
    <td class="foot-os"></td>
  </tr>
</table>
</body>
</html>
