<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:04:21 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DISTMAT(1e) EMBOSS Manual for Debian DISTMAT(1e)</p>

<p style="margin-top: 1em">NAME <br>
distmat - Create a distance matrix from a multiple sequence
alignment</p>

<p style="margin-top: 1em">SYNOPSIS <br>
distmat -sequence seqset -nucmethod list -protmethod list
-ambiguous boolean -gapweight float -position integer
-calculatea boolean -parametera float -outfile outfile</p>

<p style="margin-top: 1em">distmat -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
distmat is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Phylogeny:Molecular sequence&quot;
command group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqset <br>
File containing a sequence alignment.</p>

<p style="margin-top: 1em">Required section <br>
-nucmethod list <br>
Multiple substitution correction methods for
nucleotides.</p>

<p style="margin-top: 1em">-protmethod list <br>
Multiple substitution correction methods for proteins.</p>

<p style="margin-top: 1em">Additional section <br>
-ambiguous boolean <br>
Option to use the ambiguous codes in the calculation of the
Jukes-Cantor method or if the sequences are proteins.
Default value: N</p>

<p style="margin-top: 1em">-gapweight float <br>
Option to weight gaps in the uncorrected (nucleotide) and
Jukes-Cantor distance methods. Default value: 0.</p>

<p style="margin-top: 1em">-position integer <br>
Choose base positions to analyse in each codon i.e. 123 (all
bases), 12 (the first two bases), 1, 2, or 3 individual
bases. Default value: 123</p>

<p style="margin-top: 1em">-calculatea boolean <br>
This will force the calculation of parameter &rsquo;a&rsquo;
in the Jin-Nei Gamma distance calculation, otherwise the
default is 1.0 (see -parametera option). Default value:
N</p>

<p style="margin-top: 1em">-parametera float <br>
User defined parameter &rsquo;a&rsquo; to be use in the
Jin-Nei Gamma distance calculation. The suggested value to
be used is 1.0 (Jin et al.) and this is the default. Default
value: 1.0</p>

<p style="margin-top: 1em">Output section <br>
-outfile outfile</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
distmat is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
DISTMAT(1e)</p>
<hr>
</body>
</html>
