<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DISTMAT(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DISTMAT(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">DISTMAT(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
distmat - Create a distance matrix from a multiple sequence alignment
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 8.00ex; text-indent: -8.00ex;"><b>distmat</b>
  <b>-sequence&#x00A0;</b><b></b><i>seqset</i>
  <b>-nucmethod&#x00A0;</b><b></b><i>list</i>
  <b>-protmethod&#x00A0;</b><b></b><i>list</i>
  <b>-ambiguous&#x00A0;</b><b></b><i>boolean</i>
  <b>-gapweight&#x00A0;</b><b></b> <i>float</i>
  <b>-position&#x00A0;</b><b></b><i>integer</i>
  <b>-calculatea&#x00A0;</b><b></b> <i>boolean</i>
  <b>-parametera&#x00A0;</b><b></b> <i>float</i>
  <b>-outfile&#x00A0;</b><b></b><i>outfile</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 8.00ex; text-indent: -8.00ex;"><b>distmat</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>distmat</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Phylogeny:Molecular sequence&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-sequence</b> <i>seqset</i>
<div style="margin-left: 4.00ex;">File containing a sequence alignment.</div>
<h2 class="Ss" title="Ss" id="Required_section"><a class="selflink" href="#Required_section">Required
  section</a></h2>
<b>-nucmethod</b> <i>list</i>
<div style="margin-left: 4.00ex;">Multiple substitution correction methods for
  nucleotides.</div>
<div class="Pp"></div>
<b>-protmethod</b> <i>list</i>
<div style="margin-left: 4.00ex;">Multiple substitution correction methods for
  proteins.</div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-ambiguous</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Option to use the ambiguous codes in the
  calculation of the Jukes-Cantor method or if the sequences are proteins.
  Default value: N</div>
<div class="Pp"></div>
<b>-gapweight</b> <i>float</i>
<div style="margin-left: 4.00ex;">Option to weight gaps in the uncorrected
  (nucleotide) and Jukes-Cantor distance methods. Default value: 0.</div>
<div class="Pp"></div>
<b>-position</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Choose base positions to analyse in each codon
  i.e. 123 (all bases), 12 (the first two bases), 1, 2, or 3 individual bases.
  Default value: 123</div>
<div class="Pp"></div>
<b>-calculatea</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This will force the calculation of parameter
  'a' in the Jin-Nei Gamma distance calculation, otherwise the default is 1.0
  (see -parametera option). Default value: N</div>
<div class="Pp"></div>
<b>-parametera</b> <i>float</i>
<div style="margin-left: 4.00ex;">User defined parameter 'a' to be use in the
  Jin-Nei Gamma distance calculation. The suggested value to be used is 1.0 (Jin
  et al.) and this is the default. Default value: 1.0</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
distmat is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
