<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:04:31 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DNACLUST(1) Bioinformatics Tools DNACLUST(1)</p>

<p style="margin-top: 1em">NAME <br>
dnaclust - program to cluster large number of short DNA
sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
dnaclust {-i | --input} infile [{-s | --similarity}
threshold] [{-m | --multiple-alignment}] [{-d | --header}]
[{-l | --left-gaps-allowed}] [{-k | --k-mer-length} length]
<br>
[{-a | --approximate-filter}] [--no-k-mer-filter]</p>

<p style="margin-top: 1em">dnaclust [{-h | --help} | {-v |
--version}]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This manual page documents briefly the dnaclust program.</p>

<p style="margin-top: 1em">dnaclust is a tool for
clustering large number of short DNA sequences. The clusters
are created in such a way that the &quot;radius&quot; of
each clusters is no more than the specified <br>
threshold.</p>

<p style="margin-top: 1em">The input sequences to be
clustered should be in Fasta format. The id of each sequence
is based on the first word of the seqeunce in the Fasta
format. The first word is the <br>
prefix of the header up to the first occurance of white
space characters in the header. The output is written to
STDOUT. If you want the output to be written to a file, just
<br>
redirect the output (See Examples).</p>

<p style="margin-top: 1em">The output has two modes: the
default clustering mode, and clustering with multiple
sequence alignment. In the clustering mode (without multiple
alignment), each cluster will be <br>
printed on a separate line. The line will contain the ids of
the sequences in the cluster. The first id in each line is
the cluster center sequence id. Because of the way our <br>
clusters are constructed, the length of the cluster center
sequence is always greater than or equal to the length of
any of the sequences in the cluster. Please note that since
<br>
usually some clusters contain many sequences, the lines of
the output may be very long. If you want to visually inspect
the output, please use the &rsquo;less -S&rsquo;, or an
editor that <br>
does not wrap long lines. The number of clusters can be
found using &rsquo;wc -l&rsquo;.</p>

<p style="margin-top: 1em">For more information about the
multiple sequence alignment mode see the description of
--multiple-alignment option.</p>

<p style="margin-top: 1em">OPTIONS <br>
The program follows the usual GNU command line syntax, with
long options starting with two dashes (&rsquo;-&rsquo;). A
summary of options is included below.</p>

<p style="margin-top: 1em">--similarity threshold, -s
threshold <br>
The similarity threshold specifies the radius of the
clusters created. This parameter is a floating point number
between 0 and 1. It is calculated based on semi-global <br>
alignment of a sequence to the cluster center sequence.
Namely similarity = 1 - (edit distance) / (length of the
shorter sequence). The edit distance is the minimum number
of <br>
insertions, deletions or substitutions necessary to aling a
sequence to the cluster center sequence. Our algorithms are
faster when the similarity is higher.</p>

<p style="margin-top: 1em">--k-mer-length length, -k length
<br>
When you use the k-mer filter (which is enabled by default)
you can specify the maximum length of the k-mers used for
filtering.</p>

<p style="margin-top: 1em">The longer k-mer lengths require
more memory to store k-mer counts and the filtering will be
slower. However with the longer k-mer length, the filter
will be more specific <br>
and therefore the sequence alignment search may be
faster.</p>

<p style="margin-top: 1em">There is a tradeoff between
filtering and searching time. If you do not specify the
k-mer length a value of log4(median of the lengths of the
input sequences) is picked <br>
automatically. By using this option you can override the
default value.</p>

<p style="margin-top: 1em">Keep in mind, however, that
longer k-mer lengths would require more memory to store the
filtering data structures.</p>

<p style="margin-top: 1em">--approximate-filter , -a <br>
By default the k-mer filter is 100 percent sensitive. This
means that in the output clustering, no two cluster centers
are within the threshold distance from each other. The <br>
exact filter, however, is somewhat slow. This option speeds
up the filter by using a heuristic. The use of the
approximate filter may result in cluster centers that are
<br>
close, and a larger number clusters overall. However the
approximate filter is usually several times faster than the
exact sensitive filter. Use this option if you are <br>
clustering primarily to reduce the redundancy in the data,
and do not care about the quality of the clustering.</p>

<p style="margin-top: 1em">--allow-left-gaps , -l <br>
With this option the distances are measured based on
semi-global alignment. The semi-global alignment allows for
gaps without penalty on both ends of the shorter
sequence.</p>

<p style="margin-top: 1em">The default alignment is a one
sided semi-global alignment. i.e. gaps are only allowed at
the right end of the shorter sequence without penalty. This
behavior corresponds to <br>
the data from targeted sequening of a region (e.g. of 16S
ribosomal RNA gene).</p>

<p style="margin-top: 1em">--multiple-alignment, -m <br>
Set the output format to show the multiple sequence
alignment of each cluster. The gaps in the alignments are
represented with the dash &rsquo;-&rsquo; charachter.</p>

<p style="margin-top: 1em">The format of the MSA output is
as follows: The MSA of each cluster spans several lines. The
MSA starts with a line containing charachter &rsquo;#&rsquo;
followed by the number of <br>
sequences in that cluster. The aligned sequences (which may
contain gaps) follow in the Fasta format. Each Fasta record
will be composed of two lines. The header line and the <br>
sequence line. Since each aligned sequence is output on a
single line, the output may contain very long lines. Please
use &rsquo;less -S&rsquo;, or an editor that does not wrap
long <br>
lines for inspecting the MSA also.</p>

<p style="margin-top: 1em">--no-k-mer-filter <br>
Disables the k-mer filter. Suitable for clustering very
short sequences at a high similarity threshold.</p>

<p style="margin-top: 1em">-d, --header <br>
Write program options to output.</p>

<p style="margin-top: 1em">-h, --help <br>
Show summary of options.</p>

<p style="margin-top: 1em">-v, --version <br>
Show version of program.</p>

<p style="margin-top: 1em">EXAMPLES <br>
./dnaclust file.fasta -l -s 0.98 -k 3 &gt; clusters</p>

<p style="margin-top: 1em">BUGS <br>
The program is currently limited to only work with the boost
library.</p>

<p style="margin-top: 1em">SEE ALSO <br>
awk(1), less(1), wc(1)</p>

<p style="margin-top: 1em">AUTHOR <br>
Mohammadreza Ghodsi &lt;ghodsi@cs.umd.edu&gt; <br>
Designed and developed and maintains this package. Wrote
this manpage.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 2010 Mohammadreza Ghodsi</p>

<p style="margin-top: 1em">This manual page was written for
the Debian system (but may be used by others).</p>

<p style="margin-top: 1em">Permission is granted to copy,
distribute and/or modify this document under the terms of
the GNU General Public License, Version 2 or (at your
option) any later version published <br>
by the Free Software Foundation.</p>

<p style="margin-top: 1em">On Debian systems, the complete
text of the GNU General Public License can be found in
/usr/share/common-licenses/GPL.</p>

<p style="margin-top: 1em">dnaclust 02/01/2014
DNACLUST(1)</p>
<hr>
</body>
</html>
