<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DNACLUST(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DNACLUST(1)</td>
    <td class="head-vol">Bioinformatics Tools</td>
    <td class="head-rtitle">DNACLUST(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
dnaclust - program to cluster large number of short DNA sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>dnaclust</b>
  { <b>-i</b>&#x00A0;|&#x00A0;<b>--input</b>}&#x00A0;<i>infile</i>
  [{<b>-s</b>&#x00A0;|&#x00A0; <b>--similarity</b>}&#x00A0;<i>threshold</i>]
  [{<b>-m</b>&#x00A0;|&#x00A0;<b>--multiple-alignment</b>}]
  [{<b>-d</b>&#x00A0;|&#x00A0; <b>--header</b>}]
  [{<b>-l</b>&#x00A0;|&#x00A0;<b>--left-gaps-allowed</b>}] [{
  <b>-k</b>&#x00A0;|&#x00A0; <b>--k-mer-length</b>}&#x00A0;<i>length</i>]
  [{<b>-a</b>&#x00A0;|&#x00A0; <b>--approximate-filter</b>}]
  [<b>--no-k-mer-filter</b>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>dnaclust</b>
  [{ <b>-h</b>&#x00A0;|&#x00A0;<b>--help</b>} |
  {<b>-v</b>&#x00A0;|&#x00A0;<b>--version</b>}]</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This manual page documents briefly the <b>dnaclust</b> program.
<div class="Pp"></div>
<b>dnaclust</b> is a tool for clustering large number of short DNA sequences.
  The clusters are created in such a way that the &quot;radius&quot; of each
  clusters is no more than the specified threshold.
<div class="Pp"></div>
The input sequences to be clustered should be in Fasta format. The id of each
  sequence is based on the first word of the seqeunce in the Fasta format. The
  first word is the prefix of the header up to the first occurance of white
  space characters in the header. The output is written to STDOUT. If you want
  the output to be written to a file, just redirect the output (See Examples).
<div class="Pp"></div>
The output has two modes: the default clustering mode, and clustering with
  multiple sequence alignment. In the clustering mode (without multiple
  alignment), each cluster will be printed on a separate line. The line will
  contain the ids of the sequences in the cluster. The first id in each line is
  the cluster center sequence id. Because of the way our clusters are
  constructed, the length of the cluster center sequence is always greater than
  or equal to the length of any of the sequences in the cluster. Please note
  that since usually some clusters contain many sequences, the lines of the
  output may be very long. If you want to visually inspect the output, please
  use the 'less -S', or an editor that does not wrap long lines. The number of
  clusters can be found using 'wc -l'.
<div class="Pp"></div>
For more information about the multiple sequence alignment mode see the
  description of <b>--multiple-alignment</b> option.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
The program follows the usual GNU command line syntax, with long options
  starting with two dashes ('-'). A summary of options is included below.
<div class="Pp"></div>
<b>--similarity </b><b></b><i>threshold</i>, <b>-s </b><b></b><i>threshold</i>
<div style="margin-left: 4.00ex;">The similarity threshold specifies the radius
  of the clusters created. This parameter is a floating point number between 0
  and 1. It is calculated based on semi-global alignment of a sequence to the
  cluster center sequence. Namely similarity = 1 - (edit distance) / (length of
  the shorter sequence). The edit distance is the minimum number of insertions,
  deletions or substitutions necessary to aling a sequence to the cluster center
  sequence. Our algorithms are faster when the similarity is higher.</div>
<div class="Pp"></div>
<b>--k-mer-length </b><b></b><i>length</i>, <b>-k </b><b></b><i>length</i>
<div style="margin-left: 4.00ex;">When you use the k-mer filter (which is
  enabled by default) you can specify the maximum length of the k-mers used for
  filtering.
<div style="height: 1.00em;">&#x00A0;</div>
The longer k-mer lengths require more memory to store k-mer counts and the
  filtering will be slower. However with the longer k-mer length, the filter
  will be more specific and therefore the sequence alignment search may be
  faster.
<div style="height: 1.00em;">&#x00A0;</div>
There is a tradeoff between filtering and searching time. If you do not specify
  the k-mer length a value of log4(median of the lengths of the input sequences)
  is picked automatically. By using this option you can override the default
  value.
<div style="height: 1.00em;">&#x00A0;</div>
Keep in mind, however, that longer k-mer lengths would require more memory to
  store the filtering data structures.</div>
<div class="Pp"></div>
<b>--approximate-filter </b>, <b>-a </b>
<div style="margin-left: 4.00ex;">By default the k-mer filter is 100 percent
  sensitive. This means that in the output clustering, no two cluster centers
  are within the threshold distance from each other. The exact filter, however,
  is somewhat slow. This option speeds up the filter by using a heuristic. The
  use of the approximate filter may result in cluster centers that are close,
  and a larger number clusters overall. However the approximate filter is
  usually several times faster than the exact sensitive filter. Use this option
  if you are clustering primarily to reduce the redundancy in the data, and do
  not care about the quality of the clustering.</div>
<div class="Pp"></div>
<b>--allow-left-gaps </b>, <b>-l </b>
<div style="margin-left: 4.00ex;">With this option the distances are measured
  based on semi-global alignment. The semi-global alignment allows for gaps
  without penalty on both ends of the shorter sequence.
<div style="height: 1.00em;">&#x00A0;</div>
The default alignment is a one sided semi-global alignment. i.e. gaps are only
  allowed at the right end of the shorter sequence without penalty. This
  behavior corresponds to the data from targeted sequening of a region (e.g. of
  16S ribosomal RNA gene).</div>
<div class="Pp"></div>
<b>--multiple-alignment</b>, <b>-m</b>
<div style="margin-left: 4.00ex;">Set the output format to show the multiple
  sequence alignment of each cluster. The gaps in the alignments are represented
  with the dash '-' charachter.
<div style="height: 1.00em;">&#x00A0;</div>
The format of the MSA output is as follows: The MSA of each cluster spans
  several lines. The MSA starts with a line containing charachter '#' followed
  by the number of sequences in that cluster. The aligned sequences (which may
  contain gaps) follow in the Fasta format. Each Fasta record will be composed
  of two lines. The header line and the sequence line. Since each aligned
  sequence is output on a single line, the output may contain very long lines.
  Please use 'less -S', or an editor that does not wrap long lines for
  inspecting the MSA also.</div>
<div class="Pp"></div>
<b>--no-k-mer-filter</b>
<div style="margin-left: 4.00ex;">Disables the k-mer filter. Suitable for
  clustering very short sequences at a high similarity threshold.</div>
<div class="Pp"></div>
<b>-d</b>, <b>--header</b>
<div style="margin-left: 4.00ex;">Write program options to output.</div>
<div class="Pp"></div>
<b>-h</b>, <b>--help</b>
<div style="margin-left: 4.00ex;">Show summary of options.</div>
<div class="Pp"></div>
<b>-v</b>, <b>--version</b>
<div style="margin-left: 4.00ex;">Show version of program.</div>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<div style="margin-left: 4.00ex;">
<pre>
./dnaclust file.fasta -l -s 0.98 -k 3 &gt; clusters
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
The program is currently limited to only work with the boost library.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>awk</b>(1), <b>less</b>(1), <b>wc</b>(1)
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Mohammadreza Ghodsi</b> &lt;ghodsi@cs.umd.edu&gt;
<div style="margin-left: 4.00ex;">Designed and developed and maintains this
  package. Wrote this manpage.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright &#x00A9; 2010 Mohammadreza Ghodsi
<div>&#x00A0;</div>
<div class="Pp"></div>
This manual page was written for the Debian system (but may be used by others).
<div class="Pp"></div>
Permission is granted to copy, distribute and/or modify this document under the
  terms of the GNU General Public License, Version 2 or (at your option) any
  later version published by the Free Software Foundation.
<div class="Pp"></div>
On Debian systems, the complete text of the GNU General Public License can be
  found in /usr/share/common-licenses/GPL.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">02/01/2014</td>
    <td class="foot-os">dnaclust</td>
  </tr>
</table>
</body>
</html>
