<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DOMAINNR(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DOMAINNR(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">DOMAINNR(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
domainnr - Removes redundant domains from a DCF file.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>domainnr</b>
  <b>-dcfinfile&#x00A0;</b><b></b><i>infile</i>
  [<b>-datafile&#x00A0;</b><b></b><i>matrixf</i>]
  <b>-retain&#x00A0;</b><b></b><i>toggle</i>
  <b>-node&#x00A0;</b><b></b><i>list</i> <b>-mode&#x00A0;</b><b></b><i>list</i>
  <b>-threshold&#x00A0;</b><b></b><i>float</i>
  <b>-threshlow&#x00A0;</b><b></b><i>float</i>
  <b>-threshup&#x00A0;</b><b></b><i>float</i>
  [<b>-gapopen&#x00A0;</b><b></b><i>float</i>] [
  <b>-gapextend&#x00A0;</b><b></b> <i>float</i>]
  <b>-dcfoutfile&#x00A0;</b><b></b> <i>outfile</i>
  <b>-redoutfile&#x00A0;</b><b></b> <i>outfile</i>
  <b>-logfile&#x00A0;</b><b></b> <i>outfile</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>domainnr</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>domainnr</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Utils:Database creation&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-dcfinfile</b> <i>infile</i>
<div style="margin-left: 4.00ex;">This option specifies name of DCF file (domain
  classification file) (input). A 'domain classification file' contains
  classification and other data for domains from SCOP or CATH, in DCF format
  (EMBL-like). The files are generated by using SCOPPARSE and CATHPARSE. Domain
  sequence information can be added to the file by using DOMAINSEQS.</div>
<div class="Pp"></div>
<b>-datafile</b> <i>matrixf</i>
<div style="margin-left: 4.00ex;">This option specifies the residue substitution
  matrix. This is used for sequence comparison. Default value: EBLOSUM62</div>
<div class="Pp"></div>
<b>-retain</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">This option specifies whether to write
  redundant domains to a separate file. If this option is selected, redundant
  domains are written to a separate output file. Default value: N</div>
<h2 class="Ss" title="Ss" id="Required_section"><a class="selflink" href="#Required_section">Required
  section</a></h2>
<b>-node</b> <i>list</i>
<div style="margin-left: 4.00ex;">This option specifies the node for redundancy
  removal. Redundancy can be removed at any specified node in the SCOP or CATH
  hierarchies. For example by selecting 'Class' entries belonging to the same
  Class will be non-redundant. Default value: 1</div>
<div class="Pp"></div>
<b>-mode</b> <i>list</i>
<div style="margin-left: 4.00ex;">This option specifies whether to remove
  redundancy at a single threshold % sequence similarity or remove redundancy
  outside a range of acceptable threshold % similarity. All permutations of
  pair-wise sequence alignments are calculated for each domain family in turn
  using the EMBOSS implementation of the Needleman and Wunsch global alignment
  algorithm. Redundant sequences are removed in one of two modes as follows: (i)
  If a pair of proteins achieve greater than a threshold percentage sequence
  similarity (specified by the user) the shortest sequence is discarded. (ii) If
  a pair of proteins have a percentage sequence similarity that lies outside an
  acceptable range (specified by the user) the shortest sequence is discarded.
  Default value: 1</div>
<div class="Pp"></div>
<b>-threshold</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the % sequence identity
  redundancy threshold, which determines the redundancy calculation. If a pair
  of proteins achieve greater than this threshold the shortest sequence is
  discarded. Default value: 95.0</div>
<div class="Pp"></div>
<b>-threshlow</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the % sequence identity
  redundancy threshold, which determines the redundancy calculation. If a pair
  of proteins have a percentage sequence similarity that lies outside an
  acceptable range the shortest sequence is discarded. Default value: 30.0</div>
<div class="Pp"></div>
<b>-threshup</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the % sequence identity
  redundancy threshold, which determines the redundancy calculation. If a pair
  of proteins have a percentage sequence similarity that lies outside an
  acceptable range the shortest sequence is discarded. Default value: 90.0</div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-gapopen</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the gap insertion
  penalty. This is the score taken away when a gap is created. The best value
  depends on the choice of comparison matrix. The default value assumes you are
  using the EBLOSUM62 matrix for protein sequences, and the EDNAFULL matrix for
  nucleotide sequences. Default value: 10</div>
<div class="Pp"></div>
<b>-gapextend</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the gap extension
  penalty. This is added to the standard gap penalty for each base or residue in
  the gap. This is how long gaps are penalized. Usually you will expect a few
  long gaps rather than many short gaps, so the gap extension penalty should be
  lower than the gap penalty. Default value: 0.5</div>
<h2 class="Ss" title="Ss" id="Advanced_section"><a class="selflink" href="#Advanced_section">Advanced
  section</a></h2>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-dcfoutfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of
  non-redundant DCF file (domain classification file) (output). A 'domain
  classification file' contains classification and other data for domains from
  SCOP or CATH, in DCF format (EMBL-like). The files are generated by using
  SCOPPARSE and CATHPARSE. Domain sequence information can be added to the file
  by using DOMAINSEQS. Default value: test.scop</div>
<div class="Pp"></div>
<b>-redoutfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of DCF file
  (domain classification file) for redundant sequences (output). A 'domain
  classification file' contains classification and other data for domains from
  SCOP or CATH, in DCF format (EMBL-like). The files are generated by using
  SCOPPARSE and CATHPARSE. Domain sequence information can be added to the file
  by using DOMAINSEQS.</div>
<div class="Pp"></div>
<b>-logfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of log file for
  the build. The log file contains messages about any errors arising while
  domainnr ran. Default value: domainnr.log</div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
domainnr is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">08/11/2010</td>
    <td class="foot-os">DOMAINATRIX 0.1.0+20100721</td>
  </tr>
</table>
</body>
</html>
