<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:05:04 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DOTTER:(1) General Commands Manual DOTTER:(1)</p>

<p style="margin-top: 1em">NAME <br>
dotter - Sequence dotplots with image enhancement tools.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
dotter [options] &lt;horizontal_sequence&gt;
&lt;vertical_sequence&gt; [X options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Reference: Sonnhammer ELL &amp; Durbin R (1995). A
dot-matrix program with dynamic threshold control suited for
genomic DNA and protein sequence analysis. Gene
167(2):GC1-10.</p>

<p style="margin-top: 1em">Allowed types: <br>
Protein - Protein DNA - DNA DNA - Protein</p>

<p style="margin-top: 1em">Options:</p>

<p style="margin-top: 1em">-b &lt;file&gt; <br>
Batch mode, write dotplot to &lt;file&gt;</p>

<p style="margin-top: 1em">-l &lt;file&gt; <br>
Load dotplot from &lt;file&gt;</p>

<p style="margin-top: 1em">-m &lt;float&gt; <br>
Memory usage limit in Mb (default 0.5)</p>

<p style="margin-top: 1em">-z &lt;int&gt; <br>
Set zoom (compression) factor</p>

<p style="margin-top: 1em">-p &lt;int&gt; <br>
Set pixel factor manually (ratio pixelvalue/score)</p>

<p style="margin-top: 1em">-W &lt;int&gt; <br>
Set sliding window size. (K =&gt; Karlin/Altschul
estimate)</p>

<p style="margin-top: 1em">-M &lt;file&gt; <br>
Read in score matrix from &lt;file&gt; (Blast format;
Default: Blosum62).</p>

<p style="margin-top: 1em">-F &lt;file&gt; <br>
Read in sequences and data from &lt;file&gt; (replaces
sequencefiles).</p>

<p style="margin-top: 1em">-f &lt;file&gt; <br>
Read feature segments from &lt;file&gt;</p>

<p style="margin-top: 1em">-H Do not calculate dotplot at
startup.</p>

<p style="margin-top: 1em">-R Reversed Greyramp tool at
start.</p>

<p style="margin-top: 1em">-r Reverse and complement
horizontal_sequence (DNA vs Protein)</p>

<p style="margin-top: 1em">-D Don&rsquo;t display mirror
image in self comparisons</p>

<p style="margin-top: 1em">-w For DNA: horizontal_sequence
top strand only (Watson)</p>

<p style="margin-top: 1em">-c For DNA: horizontal_sequence
bottom strand only (Crick)</p>

<p style="margin-top: 1em">-q &lt;int&gt; <br>
Horizontal_sequence offset</p>

<p style="margin-top: 1em">-s &lt;int&gt; <br>
Vertical_sequence offset</p>

<p style="margin-top: 1em">Some X options: -acefont
&lt;font&gt; Main font. -font &lt;font&gt; Menu font.</p>

<p style="margin-top: 1em">SEE ALSO <br>
http://www.cgb.ki.se/cgb/groups/sonnhammer/Dotter.html for
more info.</p>

<p style="margin-top: 1em">AUTHOR <br>
Erik.Sonnhammer@cgb.ki.se Version 3.1, compiled Jul 10
2010</p>

<p style="margin-top: 1em">July 2010 DOTTER:(1)</p>
<hr>
</body>
</html>
