<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:06:05 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DWGSIM:(1) User Commands DWGSIM:(1)</p>

<p style="margin-top: 1em">NAME <br>
dwgsim: - a short read simulator</p>

<p style="margin-top: 1em">SYNOPSIS <br>
dwgsim [options] &lt;in.ref.fa&gt; &lt;out.prefix&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
DWGSIM is a short read simulator which simulates reads from
modern sequencing platforms.</p>

<p style="margin-top: 1em">OPTIONS <br>
-e FLOAT <br>
per base/color/flow error rate of the first read [from 0.020
to 0.020 by 0.000]</p>

<p style="margin-top: 1em">-E FLOAT <br>
per base/color/flow error rate of the second read [from
0.020 to 0.020 by 0.000]</p>

<p style="margin-top: 1em">-i use the inner distance
instead of the outer distance for pairs [False]</p>

<p style="margin-top: 1em">-d INT outer distance between
the two ends for pairs [500]</p>

<p style="margin-top: 1em">-s INT standard deviation of the
distance for pairs [50.000]</p>

<p style="margin-top: 1em">-N INT number of read pairs (-1
to disable) [-1]</p>

<p style="margin-top: 1em">-C FLOAT <br>
mean coverage across available positions (-1 to disable)
[100.00]</p>

<p style="margin-top: 1em">-1 INT length of the first read
[70]</p>

<p style="margin-top: 1em">-2 INT length of the second read
[70]</p>

<p style="margin-top: 1em">-r FLOAT <br>
rate of mutations [0.0010]</p>

<p style="margin-top: 1em">-F FLOAT <br>
frequency of given mutation to simulate low fequency somatic
mutations [0.5000] NB: freqeuncy F refers to the first
strand of mutation, therefore mutations on the second <br>
strand occour with a frequency of 1-F</p>

<p style="margin-top: 1em">-R FLOAT <br>
fraction of mutations that are indels [0.10]</p>

<p style="margin-top: 1em">-X FLOAT <br>
probability an indel is extended [0.30]</p>

<p style="margin-top: 1em">-I INT the minimum length indel
[1]</p>

<p style="margin-top: 1em">-y FLOAT <br>
probability of a random DNA read [0.05]</p>

<p style="margin-top: 1em">-n INT maximum number of Ns
allowed in a given read [0]</p>

<p style="margin-top: 1em">-c INT generate reads for [0]:
0: Illumina 1: SOLiD 2: Ion Torrent</p>

<p style="margin-top: 1em">-S INT generate reads [0]: 0:
default (opposite strand for Illumina, same strand for
SOLiD/Ion Torrent) 1: same strand (mate pair) 2: opposite
strand (paired end)</p>

<p style="margin-top: 1em">-f STRING <br>
the flow order for Ion Torrent data [(null)]</p>

<p style="margin-top: 1em">-B use a per-base error rate for
Ion Torrent data [False]</p>

<p style="margin-top: 1em">-H haploid mode [False]</p>

<p style="margin-top: 1em">-z INT random seed (-1 uses the
current time) [-1]</p>

<p style="margin-top: 1em">-M generate a mutations file
only [False]</p>

<p style="margin-top: 1em">-m FILE <br>
the mutations txt file to re-create [not using]</p>

<p style="margin-top: 1em">-b FILE <br>
the bed-like file set of candidate mutations [(null)]</p>

<p style="margin-top: 1em">-v FILE <br>
the vcf file set of candidate mutations (use pl tag for
strand) [(null)]</p>

<p style="margin-top: 1em">-x FILE <br>
the bed of regions to cover [not using]</p>

<p style="margin-top: 1em">-P STRING <br>
a read prefix to prepend to each read name [not using]</p>

<p style="margin-top: 1em">-q STRING <br>
a fixed base quality to apply (single character) [not
using]</p>

<p style="margin-top: 1em">-Q FLOAT <br>
standard deviation of the base quality scores [2.00]</p>

<p style="margin-top: 1em">-s INT standard deviation of the
distance for pairs [50.000]</p>

<p style="margin-top: 1em">-h print this message</p>

<p style="margin-top: 1em">Note: For SOLiD mate pair reads
and BFAST, the first read is F3 and the second is R3. For
SOLiD mate pair reads and BWA, the reads in the first file
are R3 the reads annotated as <br>
the first read etc.</p>

<p style="margin-top: 1em">Note: The longest supported
insertion is 4294967295.</p>

<p style="margin-top: 1em">dwgsim: 0.1.11 September 2015
DWGSIM:(1)</p>
<hr>
</body>
</html>
