<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ECOPCR(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ECOPCR(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">ECOPCR(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
ecoPCR - searching for sequence and taxonomy hybriding with given primers
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>ecoPCR</b> <b>[options]</b> <i>&lt;nucleotidic patterns&gt;</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
ecoPCR is an electronic PCR software developed by LECA and Helix-Project. It
  helps to estimate Barcode primers quality.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> : Salt concentration in M for Tm computation
    (default 0.05 M)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b> : Consider that the database sequences are
    [c]ircular</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> : [D]atabase : to match the expected format, the
    database</dt>
  <dd class="It-tag">has to be formatted first by the ecoPCRFormat(1) program.
      ecoPCRFormat(1) creates three file types:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">.sdx : contains the sequences</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">.tdx : contains information concerning the taxonomy</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">.rdx : contains the taxonomy rank</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">ecoPCR needs all the file type. As a result, you have to
      write the database radical without any extension. For example
      <i>/ecoPCRDB/gbmam</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b> : Keeps the specified number of nucleotides on
    each side of the in silico</dt>
  <dd class="It-tag">amplified sequences (including the amplified DNA fragment
      plus the two target sequences of the primers).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b> : [E]rror : max errors allowed by oligonucleotide
    (0 by default)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b> : [H]elp - print &lt;this&gt; help</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> : [I]gnore the given taxonomy id.</dt>
  <dd class="It-tag">Taxonomy id are available using the ecofind program. see
      its help typing ecofind <b>-h</b> for more information.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> : [K]ingdom mode : set the kingdom mode</dt>
  <dd class="It-tag">super kingdom mode by default.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> : minimum [L]ength : define the minimum
    amplication length.</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b> : maximum [L]ength : define the maximum
    amplicationlength.</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> : Salt correction method for Tm computation
    (SANTALUCIA : 1</dt>
  <dd class="It-tag">or OWCZARZY:2, default=1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> : [R]estricts the search to the given taxonomic
    id.</dt>
  <dd class="It-tag">Taxonomy id are available using the ecofind program. see
      its help typing ecofind <b>-h</b> for more information.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">first argument : oligonucleotide for direct strand</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">second argument : oligonucleotide for reverse strand</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Table result description :</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 1 : accession number</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 2 : sequence length</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 3 : taxonomic id</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 4 : rank</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 5 : species taxonomic id</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 6 : scientific name</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 7 : genus taxonomic id</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 8 : genus name</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 9 : family taxonomic id</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 10 : family name</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 11 : super kingdom taxonomic id</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 12 : super kingdom name</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 13 : strand (direct or reverse)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 14 : first oligonucleotide</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 15 : number of errors for the first strand</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 16 : Tm for hybridization of primer 1 at this
    site</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 17 : second oligonucleotide</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 18 : number of errors for the second strand</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 19 : Tm for hybridization of primer 1 at this
    site</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 20 : amplification length</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 21 : sequence</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">column 22 : definition</dt>
  <dd class="It-tag"></dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
http://www.grenoble.prabi.fr/trac/ecoPCR/</div>
<table class="foot">
  <tr>
    <td class="foot-date">June 2015</td>
    <td class="foot-os">ecoPCR 0.5.0</td>
  </tr>
</table>
</body>
</html>
