<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:06:49 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ECOPCR(1) User Commands ECOPCR(1)</p>

<p style="margin-top: 1em">NAME <br>
ecoPCR - searching for sequence and taxonomy hybriding with
given primers</p>

<p style="margin-top: 1em">SYNOPSIS <br>
ecoPCR [options] &lt;nucleotidic patterns&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ecoPCR is an electronic PCR software developed by LECA and
Helix-Project. It helps to estimate Barcode primers
quality.</p>

<p style="margin-top: 1em">OPTIONS <br>
-a : Salt concentration in M for Tm computation (default
0.05 M)</p>

<p style="margin-top: 1em">-c : Consider that the database
sequences are [c]ircular</p>

<p style="margin-top: 1em">-d : [D]atabase : to match the
expected format, the database <br>
has to be formatted first by the ecoPCRFormat(1) program.
ecoPCRFormat(1) creates three file types:</p>

<p style="margin-top: 1em">.sdx : contains the
sequences</p>

<p style="margin-top: 1em">.tdx : contains information
concerning the taxonomy</p>

<p style="margin-top: 1em">.rdx : contains the taxonomy
rank</p>

<p style="margin-top: 1em">ecoPCR needs all the file type.
As a result, you have to write the database radical without
any extension. For example /ecoPCRDB/gbmam</p>

<p style="margin-top: 1em">-D : Keeps the specified number
of nucleotides on each side of the in silico <br>
amplified sequences (including the amplified DNA fragment
plus the two target sequences of the primers).</p>

<p style="margin-top: 1em">-e : [E]rror : max errors
allowed by oligonucleotide (0 by default)</p>

<p style="margin-top: 1em">-h : [H]elp - print &lt;this&gt;
help</p>

<p style="margin-top: 1em">-i : [I]gnore the given taxonomy
id. <br>
Taxonomy id are available using the ecofind program. see its
help typing ecofind -h for more information.</p>

<p style="margin-top: 1em">-k : [K]ingdom mode : set the
kingdom mode <br>
super kingdom mode by default.</p>

<p style="margin-top: 1em">-l : minimum [L]ength : define
the minimum amplication length.</p>

<p style="margin-top: 1em">-L : maximum [L]ength : define
the maximum amplicationlength.</p>

<p style="margin-top: 1em">-m : Salt correction method for
Tm computation (SANTALUCIA : 1 <br>
or OWCZARZY:2, default=1)</p>

<p style="margin-top: 1em">-r : [R]estricts the search to
the given taxonomic id. <br>
Taxonomy id are available using the ecofind program. see its
help typing ecofind -h for more information.</p>

<p style="margin-top: 1em">first argument : oligonucleotide
for direct strand</p>

<p style="margin-top: 1em">second argument :
oligonucleotide for reverse strand</p>

<p style="margin-top: 1em">Table result description :</p>

<p style="margin-top: 1em">column 1 : accession number</p>

<p style="margin-top: 1em">column 2 : sequence length</p>

<p style="margin-top: 1em">column 3 : taxonomic id</p>

<p style="margin-top: 1em">column 4 : rank</p>

<p style="margin-top: 1em">column 5 : species taxonomic
id</p>

<p style="margin-top: 1em">column 6 : scientific name</p>

<p style="margin-top: 1em">column 7 : genus taxonomic
id</p>

<p style="margin-top: 1em">column 8 : genus name</p>

<p style="margin-top: 1em">column 9 : family taxonomic
id</p>

<p style="margin-top: 1em">column 10 : family name</p>

<p style="margin-top: 1em">column 11 : super kingdom
taxonomic id</p>

<p style="margin-top: 1em">column 12 : super kingdom
name</p>

<p style="margin-top: 1em">column 13 : strand (direct or
reverse)</p>

<p style="margin-top: 1em">column 14 : first
oligonucleotide</p>

<p style="margin-top: 1em">column 15 : number of errors for
the first strand</p>

<p style="margin-top: 1em">column 16 : Tm for hybridization
of primer 1 at this site</p>

<p style="margin-top: 1em">column 17 : second
oligonucleotide</p>

<p style="margin-top: 1em">column 18 : number of errors for
the second strand</p>

<p style="margin-top: 1em">column 19 : Tm for hybridization
of primer 1 at this site</p>

<p style="margin-top: 1em">column 20 : amplification
length</p>

<p style="margin-top: 1em">column 21 : sequence</p>

<p style="margin-top: 1em">column 22 : definition</p>

<p style="margin-top: 1em">SEE ALSO <br>
http://www.grenoble.prabi.fr/trac/ecoPCR/</p>

<p style="margin-top: 1em">ecoPCR 0.5.0 June 2015
ECOPCR(1)</p>
<hr>
</body>
</html>
