<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>EMMA(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">EMMA(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">EMMA(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
emma - Multiple sequence alignment (ClustalW wrapper)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>emma</b>
  <b>-sequence&#x00A0;</b><b></b><i>seqall</i>
  [<b>-onlydend&#x00A0;</b><b></b><i>toggle</i>]
  <b>-dend&#x00A0;</b><b></b><i>toggle</i>
  <b>-dendfile&#x00A0;</b><b></b><i>infile</i> [
  <b>-slow&#x00A0;</b><b></b><i>toggle</i>]
  <b>-pwmatrix&#x00A0;</b><b></b><i>list</i>
  <b>-pwdnamatrix&#x00A0;</b><b></b><i>list</i>
  <b>-usermatrix&#x00A0;</b><b></b> <i>variable</i>
  <b>-pairwisedatafile&#x00A0;</b> <b></b><i>infile</i>
  <b>-matrix&#x00A0;</b><b></b><i>list</i>
  <b>-usermamatrix&#x00A0;</b><b></b><i>variable</i>
  <b>-dnamatrix&#x00A0;</b><b></b><i>list</i>
  <b>-umamatrix&#x00A0;</b><b></b><i>variable</i>
  <b>-mamatrixfile&#x00A0;</b><b></b><i>infile</i>
  <b>-pwgapopen&#x00A0;</b><b></b> <i>float</i>
  <b>-pwgapextend&#x00A0;</b><b></b> <i>float</i>
  <b>-ktup&#x00A0;</b><b></b><i>integer</i>
  <b>-gapw&#x00A0;</b><b></b><i>integer</i>
  <b>-topdiags&#x00A0;</b><b></b><i>integer</i>
  <b>-window&#x00A0;</b><b></b><i>integer</i>
  <b>-nopercent&#x00A0;</b><b></b><i>boolean</i>
  [<b>-gapopen&#x00A0;</b><b></b><i>float</i>]
  [<b>-gapextend&#x00A0;</b><b></b><i>float</i>]
  [<b>-endgaps&#x00A0;</b><b></b><i>boolean</i>]
  [<b>-gapdist&#x00A0;</b><b></b><i>integer</i>]
  <b>-norgap&#x00A0;</b><b></b><i>boolean</i>
  <b>-hgapres&#x00A0;</b><b></b><i>string</i>
  <b>-nohgap&#x00A0;</b><b></b><i>boolean</i> [
  <b>-maxdiv&#x00A0;</b><b></b><i>integer</i>]
  <b>-outseq&#x00A0;</b><b></b><i>seqoutset</i>
  <b>-dendoutfile&#x00A0;</b><b></b> <i>outfile</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>emma</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>emma</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Alignment:Multiple&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-sequence</b> <i>seqall</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-onlydend</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<div class="Pp"></div>
<b>-dend</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<div class="Pp"></div>
<b>-dendfile</b> <i>infile</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-slow</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">A distance is calculated between every pair of
  sequences and these are used to construct the dendrogram which guides the
  final multiple alignment. The scores are calculated from separate pairwise
  alignments. These can be calculated using 2 methods: dynamic programming (slow
  but accurate) or by the method of Wilbur and Lipman (extremely fast but
  approximate). The slow-accurate method is fine for short sequences but will be
  VERY SLOW for many (e.g. &gt;100) long (e.g. &gt;1000 residue) sequences.
  Default value: Y</div>
<h2 class="Ss" title="Ss" id="Pairwise_align_options"><a class="selflink" href="#Pairwise_align_options">Pairwise
  align options</a></h2>
<b>-pwmatrix</b> <i>list</i>
<div style="margin-left: 4.00ex;">The scoring table which describes the
  similarity of each amino acid to each other. There are three 'in-built' series
  of weight matrices offered. Each consists of several matrices which work
  differently at different evolutionary distances. To see the exact details,
  read the documentation. Crudely, we store several matrices in memory, spanning
  the full range of amino acid distance (from almost identical sequences to
  highly divergent ones). For very similar sequences, it is best to use a strict
  weight matrix which only gives a high score to identities and the most
  favoured conservative substitutions. For more divergent sequences, it is
  appropriate to use 'softer' matrices which give a high score to many other
  frequent substitutions. 1) BLOSUM (Henikoff). These matrices appear to be the
  best available for carrying out data base similarity (homology searches). The
  matrices used are: Blosum80, 62, 45 and 30. 2) PAM (Dayhoff). These have been
  extremely widely used since the late '70s. We use the PAM 120, 160, 250 and
  350 matrices. 3) GONNET . These matrices were derived using almost the same
  procedure as the Dayhoff one (above) but are much more up to date and are
  based on a far larger data set. They appear to be more sensitive than the
  Dayhoff series. We use the GONNET 40, 80, 120, 160, 250 and 350 matrices. We
  also supply an identity matrix which gives a score of 1.0 to two identical
  amino acids and a score of zero otherwise. This matrix is not very useful.
  Default value: b</div>
<div class="Pp"></div>
<b>-pwdnamatrix</b> <i>list</i>
<div style="margin-left: 4.00ex;">The scoring table which describes the scores
  assigned to matches and mismatches (including IUB ambiguity codes). Default
  value: i</div>
<div class="Pp"></div>
<b>-usermatrix</b> <i>variable</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-pairwisedatafile</b> <i>infile</i>
<div style="margin-left: 4.00ex;"></div>
<h2 class="Ss" title="Ss" id="Matrix_options"><a class="selflink" href="#Matrix_options">Matrix
  options</a></h2>
<b>-matrix</b> <i>list</i>
<div style="margin-left: 4.00ex;">This gives a menu where you are offered a
  choice of weight matrices. The default for proteins is the PAM series derived
  by Gonnet and colleagues. Note, a series is used! The actual matrix that is
  used depends on how similar the sequences to be aligned at this alignment step
  are. Different matrices work differently at each evolutionary distance. There
  are three 'in-built' series of weight matrices offered. Each consists of
  several matrices which work differently at different evolutionary distances.
  To see the exact details, read the documentation. Crudely, we store several
  matrices in memory, spanning the full range of amino acid distance (from
  almost identical sequences to highly divergent ones). For very similar
  sequences, it is best to use a strict weight matrix which only gives a high
  score to identities and the most favoured conservative substitutions. For more
  divergent sequences, it is appropriate to use 'softer' matrices which give a
  high score to many other frequent substitutions. 1) BLOSUM (Henikoff). These
  matrices appear to be the best available for carrying out data base similarity
  (homology searches). The matrices used are: Blosum80, 62, 45 and 30. 2) PAM
  (Dayhoff). These have been extremely widely used since the late '70s. We use
  the PAM 120, 160, 250 and 350 matrices. 3) GONNET . These matrices were
  derived using almost the same procedure as the Dayhoff one (above) but are
  much more up to date and are based on a far larger data set. They appear to be
  more sensitive than the Dayhoff series. We use the GONNET 40, 80, 120, 160,
  250 and 350 matrices. We also supply an identity matrix which gives a score of
  1.0 to two identical amino acids and a score of zero otherwise. This matrix is
  not very useful. Alternatively, you can read in your own (just one matrix, not
  a series). Default value: b</div>
<div class="Pp"></div>
<b>-usermamatrix</b> <i>variable</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-dnamatrix</b> <i>list</i>
<div style="margin-left: 4.00ex;">This gives a menu where a single matrix (not a
  series) can be selected. Default value: i</div>
<div class="Pp"></div>
<b>-umamatrix</b> <i>variable</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-mamatrixfile</b> <i>infile</i>
<div style="margin-left: 4.00ex;"></div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<h2 class="Ss" title="Ss" id="Slow_align_options"><a class="selflink" href="#Slow_align_options">Slow
  align options</a></h2>
<b>-pwgapopen</b> <i>float</i>
<div style="margin-left: 4.00ex;">The penalty for opening a gap in the pairwise
  alignments. Default value: 10.0</div>
<div class="Pp"></div>
<b>-pwgapextend</b> <i>float</i>
<div style="margin-left: 4.00ex;">The penalty for extending a gap by 1 residue
  in the pairwise alignments. Default value: 0.1</div>
<h2 class="Ss" title="Ss" id="Fast_align_options"><a class="selflink" href="#Fast_align_options">Fast
  align options</a></h2>
<b>-ktup</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This is the size of exactly matching fragment
  that is used. INCREASE for speed (max= 2 for proteins; 4 for DNA), DECREASE
  for sensitivity. For longer sequences (e.g. &gt;1000 residues) you may need to
  increase the default. Default value: @($(acdprotein)?1:2)</div>
<div class="Pp"></div>
<b>-gapw</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This is a penalty for each gap in the fast
  alignments. It has little affect on the speed or sensitivity except for
  extreme values. Default value: @($(acdprotein)?3:5)</div>
<div class="Pp"></div>
<b>-topdiags</b> <i>integer</i>
<div style="margin-left: 4.00ex;">The number of k-tuple matches on each diagonal
  (in an imaginary dot-matrix plot) is calculated. Only the best ones (with most
  matches) are used in the alignment. This parameter specifies how many.
  Decrease for speed; increase for sensitivity. Default value:
  @($(acdprotein)?5:4)</div>
<div class="Pp"></div>
<b>-window</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This is the number of diagonals around each of
  the 'best' diagonals that will be used. Decrease for speed; increase for
  sensitivity. Default value: @($(acdprotein)?5:4)</div>
<div class="Pp"></div>
<b>-nopercent</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<h2 class="Ss" title="Ss" id="Gap_options"><a class="selflink" href="#Gap_options">Gap
  options</a></h2>
<b>-gapopen</b> <i>float</i>
<div style="margin-left: 4.00ex;">The penalty for opening a gap in the
  alignment. Increasing the gap opening penalty will make gaps less frequent.
  Default value: 10.0</div>
<div class="Pp"></div>
<b>-gapextend</b> <i>float</i>
<div style="margin-left: 4.00ex;">The penalty for extending a gap by 1 residue.
  Increasing the gap extension penalty will make gaps shorter. Terminal gaps are
  not penalised. Default value: 5.0</div>
<div class="Pp"></div>
<b>-endgaps</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">End gap separation: treats end gaps just like
  internal gaps for the purposes of avoiding gaps that are too close (set by
  'gap separation distance'). If you turn this off, end gaps will be ignored for
  this purpose. This is useful when you wish to align fragments where the end
  gaps are not biologically meaningful. Default value: Y</div>
<div class="Pp"></div>
<b>-gapdist</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Gap separation distance: tries to decrease the
  chances of gaps being too close to each other. Gaps that are less than this
  distance apart are penalised more than other gaps. This does not prevent close
  gaps; it makes them less frequent, promoting a block-like appearance of the
  alignment. Default value: 8</div>
<div class="Pp"></div>
<b>-norgap</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Residue specific penalties: amino acid
  specific gap penalties that reduce or increase the gap opening penalties at
  each position in the alignment or sequence. As an example, positions that are
  rich in glycine are more likely to have an adjacent gap than positions that
  are rich in valine. Default value: N</div>
<div class="Pp"></div>
<b>-hgapres</b> <i>string</i>
<div style="margin-left: 4.00ex;">This is a set of the residues 'considered' to
  be hydrophilic. It is used when introducing Hydrophilic gap penalties. Default
  value: GPSNDQEKR</div>
<div class="Pp"></div>
<b>-nohgap</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Hydrophilic gap penalties: used to increase
  the chances of a gap within a run (5 or more residues) of hydrophilic amino
  acids; these are likely to be loop or random coil regions where gaps are more
  common. The residues that are 'considered' to be hydrophilic are set by
  '-hgapres'. Default value: N</div>
<div class="Pp"></div>
<b>-maxdiv</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This switch, delays the alignment of the most
  distantly related sequences until after the most closely related sequences
  have been aligned. The setting shows the percent identity level required to
  delay the addition of a sequence; sequences that are less identical than this
  level to any other sequences will be aligned later. Default value: 30</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outseq</b> <i>seqoutset</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-dendoutfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
emma is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
