<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:07:16 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>EMMA(1e) EMBOSS Manual for Debian EMMA(1e)</p>

<p style="margin-top: 1em">NAME <br>
emma - Multiple sequence alignment (ClustalW wrapper)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
emma -sequence seqall [-onlydend toggle] -dend toggle
-dendfile infile [-slow toggle] -pwmatrix list -pwdnamatrix
list -usermatrix variable -pairwisedatafile infile -matrix
list <br>
-usermamatrix variable -dnamatrix list -umamatrix variable
-mamatrixfile infile -pwgapopen float -pwgapextend float
-ktup integer -gapw integer -topdiags integer <br>
-window integer -nopercent boolean [-gapopen float]
[-gapextend float] [-endgaps boolean] [-gapdist integer]
-norgap boolean -hgapres string -nohgap boolean <br>
[-maxdiv integer] -outseq seqoutset -dendoutfile outfile</p>

<p style="margin-top: 1em">emma -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
emma is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Alignment:Multiple&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall</p>

<p style="margin-top: 1em">-onlydend toggle <br>
Default value: N</p>

<p style="margin-top: 1em">-dend toggle <br>
Default value: N</p>

<p style="margin-top: 1em">-dendfile infile</p>

<p style="margin-top: 1em">-slow toggle <br>
A distance is calculated between every pair of sequences and
these are used to construct the dendrogram which guides the
final multiple alignment. The scores are calculated <br>
from separate pairwise alignments. These can be calculated
using 2 methods: dynamic programming (slow but accurate) or
by the method of Wilbur and Lipman (extremely fast but <br>
approximate). The slow-accurate method is fine for short
sequences but will be VERY SLOW for many (e.g. &gt;100) long
(e.g. &gt;1000 residue) sequences. Default value: Y</p>

<p style="margin-top: 1em">Pairwise align options <br>
-pwmatrix list <br>
The scoring table which describes the similarity of each
amino acid to each other. There are three
&rsquo;in-built&rsquo; series of weight matrices offered.
Each consists of several <br>
matrices which work differently at different evolutionary
distances. To see the exact details, read the documentation.
Crudely, we store several matrices in memory, spanning <br>
the full range of amino acid distance (from almost identical
sequences to highly divergent ones). For very similar
sequences, it is best to use a strict weight matrix which
<br>
only gives a high score to identities and the most favoured
conservative substitutions. For more divergent sequences, it
is appropriate to use &rsquo;softer&rsquo; matrices which
give a <br>
high score to many other frequent substitutions. 1) BLOSUM
(Henikoff). These matrices appear to be the best available
for carrying out data base similarity (homology <br>
searches). The matrices used are: Blosum80, 62, 45 and 30.
2) PAM (Dayhoff). These have been extremely widely used
since the late &rsquo;70s. We use the PAM 120, 160, 250 and
350 <br>
matrices. 3) GONNET . These matrices were derived using
almost the same procedure as the Dayhoff one (above) but are
much more up to date and are based on a far larger data <br>
set. They appear to be more sensitive than the Dayhoff
series. We use the GONNET 40, 80, 120, 160, 250 and 350
matrices. We also supply an identity matrix which gives a
score <br>
of 1.0 to two identical amino acids and a score of zero
otherwise. This matrix is not very useful. Default value:
b</p>

<p style="margin-top: 1em">-pwdnamatrix list <br>
The scoring table which describes the scores assigned to
matches and mismatches (including IUB ambiguity codes).
Default value: i</p>

<p style="margin-top: 1em">-usermatrix variable</p>

<p style="margin-top: 1em">-pairwisedatafile infile</p>

<p style="margin-top: 1em">Matrix options <br>
-matrix list <br>
This gives a menu where you are offered a choice of weight
matrices. The default for proteins is the PAM series derived
by Gonnet and colleagues. Note, a series is used! The <br>
actual matrix that is used depends on how similar the
sequences to be aligned at this alignment step are.
Different matrices work differently at each evolutionary
distance. <br>
There are three &rsquo;in-built&rsquo; series of weight
matrices offered. Each consists of several matrices which
work differently at different evolutionary distances. To see
the exact <br>
details, read the documentation. Crudely, we store several
matrices in memory, spanning the full range of amino acid
distance (from almost identical sequences to highly <br>
divergent ones). For very similar sequences, it is best to
use a strict weight matrix which only gives a high score to
identities and the most favoured conservative <br>
substitutions. For more divergent sequences, it is
appropriate to use &rsquo;softer&rsquo; matrices which give
a high score to many other frequent substitutions. 1) BLOSUM
(Henikoff). <br>
These matrices appear to be the best available for carrying
out data base similarity (homology searches). The matrices
used are: Blosum80, 62, 45 and 30. 2) PAM (Dayhoff). <br>
These have been extremely widely used since the late
&rsquo;70s. We use the PAM 120, 160, 250 and 350 matrices.
3) GONNET . These matrices were derived using almost the
same <br>
procedure as the Dayhoff one (above) but are much more up to
date and are based on a far larger data set. They appear to
be more sensitive than the Dayhoff series. We use the <br>
GONNET 40, 80, 120, 160, 250 and 350 matrices. We also
supply an identity matrix which gives a score of 1.0 to two
identical amino acids and a score of zero otherwise. This
<br>
matrix is not very useful. Alternatively, you can read in
your own (just one matrix, not a series). Default value:
b</p>

<p style="margin-top: 1em">-usermamatrix variable</p>

<p style="margin-top: 1em">-dnamatrix list <br>
This gives a menu where a single matrix (not a series) can
be selected. Default value: i</p>

<p style="margin-top: 1em">-umamatrix variable</p>

<p style="margin-top: 1em">-mamatrixfile infile</p>

<p style="margin-top: 1em">Additional section <br>
Slow align options <br>
-pwgapopen float <br>
The penalty for opening a gap in the pairwise alignments.
Default value: 10.0</p>

<p style="margin-top: 1em">-pwgapextend float <br>
The penalty for extending a gap by 1 residue in the pairwise
alignments. Default value: 0.1</p>

<p style="margin-top: 1em">Fast align options <br>
-ktup integer <br>
This is the size of exactly matching fragment that is used.
INCREASE for speed (max= 2 for proteins; 4 for DNA),
DECREASE for sensitivity. For longer sequences (e.g.
&gt;1000 <br>
residues) you may need to increase the default. Default
value: @($(acdprotein)?1:2)</p>

<p style="margin-top: 1em">-gapw integer <br>
This is a penalty for each gap in the fast alignments. It
has little affect on the speed or sensitivity except for
extreme values. Default value: @($(acdprotein)?3:5)</p>

<p style="margin-top: 1em">-topdiags integer <br>
The number of k-tuple matches on each diagonal (in an
imaginary dot-matrix plot) is calculated. Only the best ones
(with most matches) are used in the alignment. This <br>
parameter specifies how many. Decrease for speed; increase
for sensitivity. Default value: @($(acdprotein)?5:4)</p>

<p style="margin-top: 1em">-window integer <br>
This is the number of diagonals around each of the
&rsquo;best&rsquo; diagonals that will be used. Decrease for
speed; increase for sensitivity. Default value:
@($(acdprotein)?5:4)</p>

<p style="margin-top: 1em">-nopercent boolean <br>
Default value: N</p>

<p style="margin-top: 1em">Gap options <br>
-gapopen float <br>
The penalty for opening a gap in the alignment. Increasing
the gap opening penalty will make gaps less frequent.
Default value: 10.0</p>

<p style="margin-top: 1em">-gapextend float <br>
The penalty for extending a gap by 1 residue. Increasing the
gap extension penalty will make gaps shorter. Terminal gaps
are not penalised. Default value: 5.0</p>

<p style="margin-top: 1em">-endgaps boolean <br>
End gap separation: treats end gaps just like internal gaps
for the purposes of avoiding gaps that are too close (set by
&rsquo;gap separation distance&rsquo;). If you turn this
off, end <br>
gaps will be ignored for this purpose. This is useful when
you wish to align fragments where the end gaps are not
biologically meaningful. Default value: Y</p>

<p style="margin-top: 1em">-gapdist integer <br>
Gap separation distance: tries to decrease the chances of
gaps being too close to each other. Gaps that are less than
this distance apart are penalised more than other gaps. <br>
This does not prevent close gaps; it makes them less
frequent, promoting a block-like appearance of the
alignment. Default value: 8</p>

<p style="margin-top: 1em">-norgap boolean <br>
Residue specific penalties: amino acid specific gap
penalties that reduce or increase the gap opening penalties
at each position in the alignment or sequence. As an
example, <br>
positions that are rich in glycine are more likely to have
an adjacent gap than positions that are rich in valine.
Default value: N</p>

<p style="margin-top: 1em">-hgapres string <br>
This is a set of the residues &rsquo;considered&rsquo; to be
hydrophilic. It is used when introducing Hydrophilic gap
penalties. Default value: GPSNDQEKR</p>

<p style="margin-top: 1em">-nohgap boolean <br>
Hydrophilic gap penalties: used to increase the chances of a
gap within a run (5 or more residues) of hydrophilic amino
acids; these are likely to be loop or random coil <br>
regions where gaps are more common. The residues that are
&rsquo;considered&rsquo; to be hydrophilic are set by
&rsquo;-hgapres&rsquo;. Default value: N</p>

<p style="margin-top: 1em">-maxdiv integer <br>
This switch, delays the alignment of the most distantly
related sequences until after the most closely related
sequences have been aligned. The setting shows the percent
<br>
identity level required to delay the addition of a sequence;
sequences that are less identical than this level to any
other sequences will be aligned later. Default value: 30</p>

<p style="margin-top: 1em">Output section <br>
-outseq seqoutset</p>

<p style="margin-top: 1em">-dendoutfile outfile</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
emma is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
EMMA(1e)</p>
<hr>
</body>
</html>
