<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:07:26 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>EPESTFIND(1e) EMBOSS Manual for Debian EPESTFIND(1e)</p>

<p style="margin-top: 1em">NAME <br>
epestfind - Finds PEST motifs as potential proteolytic
cleavage sites</p>

<p style="margin-top: 1em">SYNOPSIS <br>
epestfind -sequence sequence [-mwdata datafile] -window
integer -order selection [-threshold float] -mono boolean
-potential boolean -poor boolean -invalid boolean -map
boolean <br>
-outfile outfile -graph xygraph</p>

<p style="margin-top: 1em">epestfind -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
epestfind is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Protein:Motifs&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence sequence <br>
Protein sequence USA to be analysed.</p>

<p style="margin-top: 1em">-mwdata datafile <br>
Default value: Emolwt.dat</p>

<p style="margin-top: 1em">Required section <br>
-window integer <br>
Minimal distance between positively charged amino acids.
Default value: 10</p>

<p style="margin-top: 1em">-order selection <br>
Name of the output file which holds the results of the
analysis. Results may be sorted by length, position and
score. Default value: score</p>

<p style="margin-top: 1em">Additional section <br>
-threshold float <br>
Threshold value to discriminate weak from potential PEST
motifs. Valid PEST motifs are discriminated into
&rsquo;poor&rsquo; and &rsquo;potential&rsquo; motifs
depending on this threshold score. By <br>
default, the default value is set to +5.0 based on
experimental data. Alterations are not recommended since
significance is a matter of biology, not mathematics.
Default <br>
value: +5.0</p>

<p style="margin-top: 1em">Advanced section <br>
-mono boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-potential boolean <br>
Decide whether potential PEST motifs should be printed.
Default value: Y</p>

<p style="margin-top: 1em">-poor boolean <br>
Decide whether poor PEST motifs should be printed. Default
value: Y</p>

<p style="margin-top: 1em">-invalid boolean <br>
Decide whether invalid PEST motifs should be printed.
Default value: N</p>

<p style="margin-top: 1em">-map boolean <br>
Decide whether PEST motifs should be mapped to sequence.
Default value: Y</p>

<p style="margin-top: 1em">Output section <br>
-outfile outfile <br>
Name of file to which results will be written.</p>

<p style="margin-top: 1em">-graph xygraph</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
epestfind is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
EPESTFIND(1e)</p>
<hr>
</body>
</html>
