<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>EPESTFIND(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">EPESTFIND(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">EPESTFIND(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
epestfind - Finds PEST motifs as potential proteolytic cleavage sites
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 10.00ex; text-indent: -10.00ex;"><b>epestfind</b>
  <b>-sequence&#x00A0;</b><b></b><i>sequence</i>
  [<b>-mwdata&#x00A0;</b><b></b><i>datafile</i>]
  <b>-window&#x00A0;</b><b></b><i>integer</i>
  <b>-order&#x00A0;</b><b></b><i>selection</i> [
  <b>-threshold&#x00A0;</b><b></b> <i>float</i>]
  <b>-mono&#x00A0;</b><b></b><i>boolean</i>
  <b>-potential&#x00A0;</b><b></b><i>boolean</i>
  <b>-poor&#x00A0;</b><b></b><i>boolean</i>
  <b>-invalid&#x00A0;</b><b></b><i>boolean</i>
  <b>-map&#x00A0;</b><b></b><i>boolean</i>
  <b>-outfile&#x00A0;</b><b></b><i>outfile</i>
  <b>-graph&#x00A0;</b><b></b><i>xygraph</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 10.00ex; text-indent: -10.00ex;"><b>epestfind</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>epestfind</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Protein:Motifs&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-sequence</b> <i>sequence</i>
<div style="margin-left: 4.00ex;">Protein sequence USA to be analysed.</div>
<div class="Pp"></div>
<b>-mwdata</b> <i>datafile</i>
<div style="margin-left: 4.00ex;">Default value: Emolwt.dat</div>
<h2 class="Ss" title="Ss" id="Required_section"><a class="selflink" href="#Required_section">Required
  section</a></h2>
<b>-window</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Minimal distance between positively charged
  amino acids. Default value: 10</div>
<div class="Pp"></div>
<b>-order</b> <i>selection</i>
<div style="margin-left: 4.00ex;">Name of the output file which holds the
  results of the analysis. Results may be sorted by length, position and score.
  Default value: score</div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-threshold</b> <i>float</i>
<div style="margin-left: 4.00ex;">Threshold value to discriminate weak from
  potential PEST motifs. Valid PEST motifs are discriminated into 'poor' and
  'potential' motifs depending on this threshold score. By default, the default
  value is set to +5.0 based on experimental data. Alterations are not
  recommended since significance is a matter of biology, not mathematics.
  Default value: +5.0</div>
<h2 class="Ss" title="Ss" id="Advanced_section"><a class="selflink" href="#Advanced_section">Advanced
  section</a></h2>
<b>-mono</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<div class="Pp"></div>
<b>-potential</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Decide whether potential PEST motifs should be
  printed. Default value: Y</div>
<div class="Pp"></div>
<b>-poor</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Decide whether poor PEST motifs should be
  printed. Default value: Y</div>
<div class="Pp"></div>
<b>-invalid</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Decide whether invalid PEST motifs should be
  printed. Default value: N</div>
<div class="Pp"></div>
<b>-map</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Decide whether PEST motifs should be mapped to
  sequence. Default value: Y</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;">Name of file to which results will be
  written.</div>
<div class="Pp"></div>
<b>-graph</b> <i>xygraph</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
epestfind is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
