<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:07:28 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>EPRIMER3(1e) EMBOSS Manual for Debian EPRIMER3(1e)</p>

<p style="margin-top: 1em">NAME <br>
eprimer3 - Picks PCR primers and hybridization oligos</p>

<p style="margin-top: 1em">SYNOPSIS <br>
eprimer3 -sequence seqall [-primer toggle] -task list
[-hybridprobe toggle] -mishyblibraryfile infile
-mispriminglibraryfile infile [-numreturn integer]
[-includedregion range] <br>
[-targetregion range] [-excludedregion range] [-forwardinput
string] [-reverseinput string] -gcclamp integer -osize
integer -minsize integer -maxsize integer -otm float <br>
-mintm float -maxtm float -maxdifftm float -ogcpercent float
-mingc float -maxgc float -saltconc float -dnaconc float
-maxpolyx integer -psizeopt integer -prange range <br>
-ptmopt float -ptmmin float -ptmmax float -oexcludedregion
range -oligoinput string -osizeopt integer -ominsize integer
-omaxsize integer -otmopt float -otmmin float <br>
-otmmax float -ogcopt float -ogcmin float -ogcmax float
-osaltconc float -odnaconc float -oanyself float -oendself
float -opolyxmax integer -omishybmax float <br>
-explainflag boolean -fileflag boolean -firstbaseindex
integer -pickanyway boolean -maxmispriming float
-pairmaxmispriming float -numnsaccepted integer -selfany
float <br>
-selfend float -maxendstability float -outfile outfile</p>

<p style="margin-top: 1em">eprimer3 -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
eprimer3 is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Nucleic:Primers&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall <br>
The sequence from which to choose primers. The sequence must
be presented 5&rsquo; to 3&rsquo;</p>

<p style="margin-top: 1em">-primer toggle <br>
Tell EPrimer3 to pick primer(s) Default value: Y</p>

<p style="margin-top: 1em">-task list <br>
Tell EPrimer3 what task to perform. Legal values are 1:
&rsquo;Pick PCR primers&rsquo;, 2: &rsquo;Pick forward
primer only&rsquo;, 3: &rsquo;Pick reverse primer
only&rsquo;, 4: &rsquo;No primers needed&rsquo;. Default
<br>
value: 1</p>

<p style="margin-top: 1em">-hybridprobe toggle <br>
An &rsquo;internal oligo&rsquo; is intended to be used as a
hybridization probe (hyb probe) to detect the PCR product
after amplification. Default value: N</p>

<p style="margin-top: 1em">-mishyblibraryfile infile <br>
Similar to MISPRIMING-LIBRARY, except that the event we seek
to avoid is hybridization of the internal oligo to sequences
in this library rather than priming from them. The <br>
file must be in (a slightly restricted) FASTA format (W. B.
Pearson and D.J. Lipman, PNAS 85:8 pp 2444-2448 [1988]); we
briefly discuss the organization of this file below. <br>
If this parameter is specified then EPrimer3 locally aligns
each candidate oligo against each library sequence and
rejects those primers for which the local alignment score
<br>
times a specified weight (see below) exceeds
INTERNAL-OLIGO-MAX-MISHYB. (The maximum value of the weight
is arbitrarily set to 12.0.) Each sequence entry in the
FASTA-format <br>
file must begin with an &rsquo;id line&rsquo; that starts
with &rsquo;&gt;&rsquo;. The contents of the id line is
&rsquo;slightly restricted&rsquo; in that EPrimer3 parses
everything after any optional asterisk <br>
(&rsquo;*&rsquo;) as a floating point number to use as the
weight mentioned above. If the id line contains no asterisk
then the weight defaults to 1.0. The alignment scoring
system used <br>
is the same as for calculating complementarity among oligos
(e.g. SELF-ANY). The remainder of an entry contains the
sequence as lines following the id line up until a line <br>
starting with &rsquo;&gt;&rsquo; or the end of the file.
Whitespace and newlines are ignored. Characters
&rsquo;A&rsquo;, &rsquo;T&rsquo;, &rsquo;G&rsquo;,
&rsquo;C&rsquo;, &rsquo;a&rsquo;, &rsquo;t&rsquo;,
&rsquo;g&rsquo;, &rsquo;c&rsquo; are retained and any other
character is <br>
converted to &rsquo;N&rsquo; (with the consequence that any
IUB / IUPAC codes for ambiguous bases are converted to
&rsquo;N&rsquo;). There are no restrictions on line length.
An empty value for this <br>
parameter indicates that no library should be used.</p>

<p style="margin-top: 1em">-mispriminglibraryfile infile
<br>
The name of a file containing a nucleotide sequence library
of sequences to avoid amplifying (for example repetitive
sequences, or possibly the sequences of genes in a gene <br>
family that should not be amplified.) The file must be in (a
slightly restricted) FASTA format (W. B. Pearson and D.J.
Lipman, PNAS 85:8 pp 2444-2448 [1988]); we briefly <br>
discuss the organization of this file below. If this
parameter is specified then EPrimer3 locally aligns each
candidate primer against each library sequence and rejects
those <br>
primers for which the local alignment score times a
specified weight (see below) exceeds MAX-MISPRIMING. (The
maximum value of the weight is arbitrarily set to 100.0.)
Each <br>
sequence entry in the FASTA-format file must begin with an
&rsquo;id line&rsquo; that starts with &rsquo;&gt;&rsquo;.
The contents of the id line is &rsquo;slightly
restricted&rsquo; in that EPrimer3 parses <br>
everything after any optional asterisk (&rsquo;*&rsquo;) as
a floating point number to use as the weight mentioned
above. If the id line contains no asterisk then the weight
defaults to <br>
1.0. The alignment scoring system used is the same as for
calculating complementarity among oligos (e.g. SELF-ANY).
The remainder of an entry contains the sequence as lines
<br>
following the id line up until a line starting with
&rsquo;&gt;&rsquo; or the end of the file. Whitespace and
newlines are ignored. Characters &rsquo;A&rsquo;,
&rsquo;T&rsquo;, &rsquo;G&rsquo;, &rsquo;C&rsquo;,
&rsquo;a&rsquo;, &rsquo;t&rsquo;, &rsquo;g&rsquo;,
&rsquo;c&rsquo; are <br>
retained and any other character is converted to
&rsquo;N&rsquo; (with the consequence that any IUB / IUPAC
codes for ambiguous bases are converted to &rsquo;N&rsquo;).
There are no restrictions on <br>
line length. An empty value for this parameter indicates
that no repeat library should be used.</p>

<p style="margin-top: 1em">Additional section <br>
Program options <br>
-numreturn integer <br>
The maximum number of primer pairs to return. Primer pairs
returned are sorted by their &rsquo;quality&rsquo;, in other
words by the value of the objective function (where a lower
number <br>
indicates a better primer pair). Caution: setting this
parameter to a large value will increase running time.
Default value: 5</p>

<p style="margin-top: 1em">Sequence options <br>
-includedregion range <br>
A sub-region of the given sequence in which to pick primers.
For example, often the first dozen or so bases of a sequence
are vector, and should be excluded from <br>
consideration. The value for this parameter has the form
(start),(end) where (start) is the index of the first base
to consider, and (end) is the last in the primer-picking
<br>
region.</p>

<p style="margin-top: 1em">-targetregion range <br>
If one or more Targets is specified then a legal primer pair
must flank at least one of them. A Target might be a simple
sequence repeat site (for example a CA repeat) or a <br>
single-base-pair polymorphism. The value should be a
space-separated list of (start),(end) pairs where (start) is
the index of the first base of a Target, and (end) is the
<br>
last E.g. 50,51 requires primers to surround the 2 bases at
positions 50 and 51.</p>

<p style="margin-top: 1em">-excludedregion range <br>
Primer oligos may not overlap any region specified in this
tag. The associated value must be a space-separated list of
(start),(end) pairs where (start) is the index of the <br>
first base of the excluded region, and and (end) is the
last. This tag is useful for tasks such as excluding regions
of low sequence quality or for excluding regions <br>
containing repetitive elements such as ALUs or LINEs. E.g.
401,407 68,70 forbids selection of primers in the 7 bases
starting at 401 and the 3 bases at 68.</p>

<p style="margin-top: 1em">-forwardinput string <br>
The sequence of a forward primer to check and around which
to design reverse primers and optional internal oligos. Must
be a substring of SEQUENCE.</p>

<p style="margin-top: 1em">-reverseinput string <br>
The sequence of a reverse primer to check and around which
to design forward primers and optional internal oligos. Must
be a substring of the reverse strand of SEQUENCE.</p>

<p style="margin-top: 1em">Primer options <br>
-gcclamp integer <br>
Require the specified number of consecutive Gs and Cs at the
3&rsquo; end of both the forward and reverse primer. (This
parameter has no effect on the internal oligo if one is <br>
requested.)</p>

<p style="margin-top: 1em">-osize integer <br>
Optimum length (in bases) of a primer oligo. EPrimer3 will
attempt to pick primers close to this length. Default value:
20</p>

<p style="margin-top: 1em">-minsize integer <br>
Minimum acceptable length of a primer. Must be greater than
0 and less than or equal to MAX-SIZE. Default value: 18</p>

<p style="margin-top: 1em">-maxsize integer <br>
Maximum acceptable length (in bases) of a primer. Currently
this parameter cannot be larger than 35. This limit is
governed by the maximum oligo size for which
EPrimer3&rsquo;s <br>
melting-temperature is valid. Default value: 27</p>

<p style="margin-top: 1em">-otm float <br>
Optimum melting temperature(Celsius) for a primer oligo.
EPrimer3 will try to pick primers with melting temperatures
are close to this temperature. The oligo melting <br>
temperature formula in EPrimer3 is that given in Rychlik,
Spencer and Rhoads, Nucleic Acids Research, vol 18, num 21,
pp 6409-6412 and Breslauer, Frank, Bloecker and Marky, <br>
Proc. Natl. Acad. Sci. USA, vol 83, pp 3746-3750. Please
refer to the former paper for background discussion. Default
value: 60.0</p>

<p style="margin-top: 1em">-mintm float <br>
Minimum acceptable melting temperature(Celsius) for a primer
oligo. Default value: 57.0</p>

<p style="margin-top: 1em">-maxtm float <br>
Maximum acceptable melting temperature(Celsius) for a primer
oligo. Default value: 63.0</p>

<p style="margin-top: 1em">-maxdifftm float <br>
Maximum acceptable (unsigned) difference between the melting
temperatures of the forward and reverse primers. Default
value: 100.0</p>

<p style="margin-top: 1em">-ogcpercent float <br>
Primer optimum GC percent. Default value: 50.0</p>

<p style="margin-top: 1em">-mingc float <br>
Minimum allowable percentage of Gs and Cs in any primer.
Default value: 20.0</p>

<p style="margin-top: 1em">-maxgc float <br>
Maximum allowable percentage of Gs and Cs in any primer
generated by Primer. Default value: 80.0</p>

<p style="margin-top: 1em">-saltconc float <br>
The millimolar concentration of salt (usually KCl) in the
PCR. EPrimer3 uses this argument to calculate oligo melting
temperatures. Default value: 50.0</p>

<p style="margin-top: 1em">-dnaconc float <br>
The nanomolar concentration of annealing oligos in the PCR.
EPrimer3 uses this argument to calculate oligo melting
temperatures. The default (50nM) works well with the <br>
standard protocol used at the Whitehead/MIT Center for
Genome Research--0.5 microliters of 20 micromolar
concentration for each primer oligo in a 20 microliter
reaction with <br>
10 nanograms template, 0.025 units/microliter Taq polymerase
in 0.1 mM each dNTP, 1.5mM MgCl2, 50mM KCl, 10mM Tris-HCL
(pH 9.3) using 35 cycles with an annealing temperature <br>
of 56 degrees Celsius. This parameter corresponds to
&rsquo;c&rsquo; in Rychlik, Spencer and Rhoads&rsquo;
equation (ii) (Nucleic Acids Research, vol 18, num 21) where
a suitable value (for a <br>
lower initial concentration of template) is
&rsquo;empirically determined&rsquo;. The value of this
parameter is less than the actual concentration of oligos in
the reaction because it <br>
is the concentration of annealing oligos, which in turn
depends on the amount of template (including PCR product) in
a given cycle. This concentration increases a great deal
<br>
during a PCR; fortunately PCR seems quite robust for a
variety of oligo melting temperatures. See ADVICE FOR
PICKING PRIMERS. Default value: 50.0</p>

<p style="margin-top: 1em">-maxpolyx integer <br>
The maximum allowable length of a mononucleotide repeat in a
primer, for example AAAAAA. Default value: 5</p>

<p style="margin-top: 1em">Product options <br>
-psizeopt integer <br>
The optimum size for the PCR product. 0 indicates that there
is no optimum product size. Default value: 200</p>

<p style="margin-top: 1em">-prange range <br>
The associated values specify the lengths of the product
that the user wants the primers to create, and is a space
separated list of elements of the form (x)-(y) where an <br>
(x)-(y) pair is a legal range of lengths for the product.
For example, if one wants PCR products to be between 100 to
150 bases (inclusive) then one would set this parameter <br>
to 100-150. If one desires PCR products in either the range
from 100 to 150 bases or in the range from 200 to 250 bases
then one would set this parameter to 100-150 200-250. <br>
EPrimer3 favors ranges to the left side of the parameter
string. EPrimer3 will return legal primers pairs in the
first range regardless the value of the objective function
<br>
for these pairs. Only if there are an insufficient number of
primers in the first range will EPrimer3 return primers in a
subsequent range. Default value: 100-300</p>

<p style="margin-top: 1em">-ptmopt float <br>
The optimum melting temperature for the PCR product. 0
indicates that there is no optimum temperature. Default
value: 0.0</p>

<p style="margin-top: 1em">-ptmmin float <br>
The minimum allowed melting temperature of the amplicon.
Please see the documentation on the maximum melting
temperature of the product for details. Default value:
-1000000.0</p>

<p style="margin-top: 1em">-ptmmax float <br>
The maximum allowed melting temperature of the amplicon.
Product Tm is calculated using the formula from Bolton and
McCarthy, PNAS 84:1390 (1962) as presented in Sambrook, <br>
Fritsch and Maniatis, Molecular Cloning, p 11.46 (1989, CSHL
Press). Tm = 81.5 + 16.6(log10([Na+])) + .41*(%GC) -
600/length Where [Na+} is the molar sodium concentration,
<br>
(%GC) is the percent of Gs and Cs in the sequence, and
length is the length of the sequence. A similar formula is
used by the prime primer selection program in GCG, which
<br>
instead uses 675.0/length in the last term (after F.
Baldino, Jr, M.-F. Chesselet, and M.E. Lewis, Methods in
Enzymology 168:766 (1989) eqn (1) on page 766 without the
<br>
mismatch and formamide terms). The formulas here and in
Baldino et al. assume Na+ rather than K+. According to J.G.
Wetmur, Critical Reviews in BioChem. and Mol. Bio. 26:227
<br>
(1991) 50 mM K+ should be equivalent in these formulae to .2
M Na+. EPrimer3 uses the same salt concentration value for
calculating both the primer melting temperature and <br>
the oligo melting temperature. If you are planning to use
the PCR product for hybridization later this behavior will
not give you the Tm under hybridization conditions. <br>
Default value: 1000000.0</p>

<p style="margin-top: 1em">Internal oligo input <br>
-oexcludedregion range <br>
Middle oligos may not overlap any region specified by this
tag. The associated value must be a space-separated list of
(start),(end) pairs, where (start) is the index of the <br>
first base of an excluded region, and (end) is the last.
Often one would make Target regions excluded regions for
internal oligos.</p>

<p style="margin-top: 1em">-oligoinput string <br>
The sequence of an internal oligo to check and around which
to design forward and reverse primers. Must be a substring
of SEQUENCE.</p>

<p style="margin-top: 1em">Internal oligo options <br>
-osizeopt integer <br>
Optimum length (in bases) of an internal oligo. EPrimer3
will attempt to pick primers close to this length. Default
value: 20</p>

<p style="margin-top: 1em">-ominsize integer <br>
Minimum acceptable length of an internal oligo. Must be
greater than 0 and less than or equal to
INTERNAL-OLIGO-MAX-SIZE. Default value: 18</p>

<p style="margin-top: 1em">-omaxsize integer <br>
Maximum acceptable length (in bases) of an internal oligo.
Currently this parameter cannot be larger than 35. This
limit is governed by maximum oligo size for which <br>
EPrimer3&rsquo;s melting-temperature is valid. Default
value: 27</p>

<p style="margin-top: 1em">-otmopt float <br>
Optimum melting temperature (Celsius) for an internal oligo.
EPrimer3 will try to pick oligos with melting temperatures
that are close to this temperature. The oligo melting <br>
temperature formula in EPrimer3 is that given in Rychlik,
Spencer and Rhoads, Nucleic Acids Research, vol 18, num 21,
pp 6409-6412 and Breslauer, Frank, Bloecker and Marky, <br>
Proc. Natl. Acad. Sci. USA, vol 83, pp 3746-3750. Please
refer to the former paper for background discussion. Default
value: 60.0</p>

<p style="margin-top: 1em">-otmmin float <br>
Minimum acceptable melting temperature(Celsius) for an
internal oligo. Default value: 57.0</p>

<p style="margin-top: 1em">-otmmax float <br>
Maximum acceptable melting temperature (Celsius) for an
internal oligo. Default value: 63.0</p>

<p style="margin-top: 1em">-ogcopt float <br>
Internal oligo optimum GC percent. Default value: 50.0</p>

<p style="margin-top: 1em">-ogcmin float <br>
Minimum allowable percentage of Gs and Cs in an internal
oligo. Default value: 20.0</p>

<p style="margin-top: 1em">-ogcmax float <br>
Maximum allowable percentage of Gs and Cs in any internal
oligo generated by Primer. Default value: 80.0</p>

<p style="margin-top: 1em">-osaltconc float <br>
The millimolar concentration of salt (usually KCl) in the
hybridization. EPrimer3 uses this argument to calculate
internal oligo melting temperatures. Default value: 50.0</p>

<p style="margin-top: 1em">-odnaconc float <br>
The nanomolar concentration of annealing internal oligo in
the hybridization. Default value: 50.0</p>

<p style="margin-top: 1em">-oanyself float <br>
The maximum allowable local alignment score when testing an
internal oligo for (local) self-complementarity. Local
self-complementarity is taken to predict the tendency of
<br>
oligos to anneal to themselves The scoring system gives 1.00
for complementary bases, -0.25 for a match of any base (or
N) with an N, -1.00 for a mismatch, and -2.00 for a <br>
gap. Only single-base-pair gaps are allowed. For example,
the alignment 5&rsquo; ATCGNA 3&rsquo; || | | 3&rsquo;
TA-CGT 5&rsquo; is allowed (and yields a score of 1.75), but
the alignment 5&rsquo; ATCCGNA <br>
3&rsquo; || | | 3&rsquo; TA--CGT 5&rsquo; is not considered.
Scores are non-negative, and a score of 0.00 indicates that
there is no reasonable local alignment between two oligos.
Default <br>
value: 12.00</p>

<p style="margin-top: 1em">-oendself float <br>
The maximum allowable 3&rsquo;-anchored global alignment
score when testing a single oligo for self-complementarity.
The scoring system is as for the Maximum Complementarity
<br>
argument. In the examples above the scores are 7.00 and 6.00
respectively. Scores are non-negative, and a score of 0.00
indicates that there is no reasonable 3&rsquo;-anchored <br>
global alignment between two oligos. In order to estimate
3&rsquo;-anchored global alignments for candidate oligos,
Primer assumes that the sequence from which to choose oligos
is <br>
presented 5&rsquo; to 3&rsquo;. INTERNAL-OLIGO-SELF-END is
meaningless when applied to internal oligos used for
hybridization-based detection, since primer-dimer will not
occur. We <br>
recommend that INTERNAL-OLIGO-SELF-END be set at least as
high as INTERNAL-OLIGO-SELF-ANY. Default value: 12.00</p>

<p style="margin-top: 1em">-opolyxmax integer <br>
The maximum allowable length of an internal oligo
mononucleotide repeat, for example AAAAAA. Default value:
5</p>

<p style="margin-top: 1em">-omishybmax float <br>
Similar to MAX-MISPRIMING except that this parameter applies
to the similarity of candidate internal oligos to the
library specified in INTERNAL-OLIGO-MISHYB-LIBRARY. Default
<br>
value: 12.0</p>

<p style="margin-top: 1em">Advanced section <br>
-explainflag boolean <br>
If this flag is true, produce LEFT-EXPLAIN, RIGHT-EXPLAIN,
and INTERNAL-OLIGO-EXPLAIN output tags, which are intended
to provide information on the number of oligos and <br>
primer pairs that EPrimer3 examined, and statistics on the
number discarded for various reasons. Default value: N</p>

<p style="margin-top: 1em">-fileflag boolean <br>
If the associated value is true, then EPrimer3 creates two
output files for each input SEQUENCE. File (sequence-id).for
lists all acceptable forward primers for <br>
(sequence-id), and (sequence-id).rev lists all acceptable
reverse primers for (sequence-id), where (sequence-id) is
the value of the SEQUENCE-ID tag (which must be supplied).
<br>
In addition, if the input tag TASK is 1 or 4, EPrimer3
produces a file (sequence-id).int, which lists all
acceptable internal oligos. Default value: N</p>

<p style="margin-top: 1em">-firstbaseindex integer <br>
This parameter is the index of the first base in the input
sequence. For input and output using 1-based indexing (such
as that used in GenBank and to which many users are <br>
accustomed) set this parameter to 1. For input and output
using 0-based indexing set this parameter to 0. (This
parameter also affects the indexes in the contents of the
<br>
files produced when the primer file flag is set.) Default
value: 1</p>

<p style="margin-top: 1em">-pickanyway boolean <br>
If true pick a primer pair even if LEFT-INPUT, RIGHT-INPUT,
or INTERNAL-OLIGO-INPUT violates specific constraints.
Default value: N</p>

<p style="margin-top: 1em">-maxmispriming float <br>
The maximum allowed weighted similarity with any sequence in
MISPRIMING-LIBRARY. Default value: 12.00</p>

<p style="margin-top: 1em">-pairmaxmispriming float <br>
The maximum allowed sum of weighted similarities of a primer
pair (one similarity for each primer) with any single
sequence in MISPRIMING-LIBRARY. Default value: 24.00</p>

<p style="margin-top: 1em">-numnsaccepted integer <br>
Maximum number of unknown bases (N) allowable in any
primer.</p>

<p style="margin-top: 1em">-selfany float <br>
The maximum allowable local alignment score when testing a
single primer for (local) self-complementarity and the
maximum allowable local alignment score when testing for
<br>
complementarity between forward and reverse primers. Local
self-complementarity is taken to predict the tendency of
primers to anneal to each other without necessarily <br>
causing self-priming in the PCR. The scoring system gives
1.00 for complementary bases, -0.25 for a match of any base
(or N) with an N, -1.00 for a mismatch, and -2.00 for a <br>
gap. Only single-base-pair gaps are allowed. For example,
the alignment 5&rsquo; ATCGNA 3&rsquo; ...|| | | 3&rsquo;
TA-CGT 5&rsquo; is allowed (and yields a score of 1.75), but
the alignment 5&rsquo; <br>
ATCCGNA 3&rsquo; ...|| | | 3&rsquo; TA--CGT 5&rsquo; is not
considered. Scores are non-negative, and a score of 0.00
indicates that there is no reasonable local alignment
between two oligos. <br>
Default value: 8.00</p>

<p style="margin-top: 1em">-selfend float <br>
The maximum allowable 3&rsquo;-anchored global alignment
score when testing a single primer for self-complementarity,
and the maximum allowable 3&rsquo;-anchored global alignment
score <br>
when testing for complementarity between forward and reverse
primers. The 3&rsquo;-anchored global alignment score is
taken to predict the likelihood of PCR-priming
primer-dimers, <br>
for example 5&rsquo; ATGCCCTAGCTTCCGGATG 3&rsquo;
.............||| ||||| ..........3&rsquo;
AAGTCCTACATTTAGCCTAGT 5&rsquo; or 5&rsquo;
AGGCTATGGGCCTCGCGA 3&rsquo; ...............||||||
............3&rsquo; <br>
AGCGCTCCGGGTATCGGA 5&rsquo; The scoring system is as for the
Maximum Complementarity argument. In the examples above the
scores are 7.00 and 6.00 respectively. Scores are <br>
non-negative, and a score of 0.00 indicates that there is no
reasonable 3&rsquo;-anchored global alignment between two
oligos. In order to estimate 3&rsquo;-anchored global
alignments <br>
for candidate primers and primer pairs, Primer assumes that
the sequence from which to choose primers is presented
5&rsquo; to 3&rsquo;. It is nonsensical to provide a larger
value for <br>
this parameter than for the Maximum (local) Complementarity
parameter because the score of a local alignment will always
be at least as great as the score of a global <br>
alignment. Default value: 3.00</p>

<p style="margin-top: 1em">Primer penalty weights <br>
-maxendstability float <br>
The maximum stability for the five 3&rsquo; bases of a
forward or reverse primer. Bigger numbers mean more stable
3&rsquo; ends. The value is the maximum delta G for duplex
disruption <br>
for the five 3&rsquo; bases as calculated using the nearest
neighbor parameters published in Breslauer, Frank, Bloecker
and Marky, Proc. Natl. Acad. Sci. USA, vol 83, pp 3746-3750.
<br>
EPrimer3 uses a completely permissive default value for
backward compatibility (which we may change in the next
release). Rychlik recommends a maximum value of 9 (Wojciech
<br>
Rychlik, &rsquo;Selection of Primers for Polymerase Chain
Reaction&rsquo; in BA White, Ed., &rsquo;Methods in
Molecular Biology, Vol. 15: PCR Protocols: Current Methods
and Applications&rsquo;, <br>
1993, pp 31-40, Humana Press, Totowa NJ). Default value:
9.0</p>

<p style="margin-top: 1em">Output section <br>
-outfile outfile</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
eprimer3 is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
EPRIMER3(1e)</p>
<hr>
</body>
</html>
