<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>EPRIMER3(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">EPRIMER3(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">EPRIMER3(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
eprimer3 - Picks PCR primers and hybridization oligos
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>eprimer3</b>
  <b>-sequence&#x00A0;</b><b></b><i>seqall</i>
  [<b>-primer&#x00A0;</b><b></b><i>toggle</i>]
  <b>-task&#x00A0;</b><b></b><i>list</i> [
  <b>-hybridprobe&#x00A0;</b><b></b><i>toggle</i>]
  <b>-mishyblibraryfile&#x00A0;</b> <b></b><i>infile</i>
  <b>-mispriminglibraryfile&#x00A0;</b> <b></b><i>infile</i>
  [<b>-numreturn&#x00A0;</b> <b></b><i>integer</i>]
  [<b>-includedregion&#x00A0;</b> <b></b><i>range</i>]
  [<b>-targetregion&#x00A0;</b> <b></b><i>range</i>]
  [<b>-excludedregion&#x00A0;</b> <b></b><i>range</i>]
  [<b>-forwardinput&#x00A0;</b> <b></b><i>string</i>]
  [<b>-reverseinput&#x00A0;</b> <b></b><i>string</i>]
  <b>-gcclamp&#x00A0;</b><b></b> <i>integer</i>
  <b>-osize&#x00A0;</b><b></b><i>integer</i>
  <b>-minsize&#x00A0;</b><b></b><i>integer</i>
  <b>-maxsize&#x00A0;</b><b></b><i>integer</i>
  <b>-otm&#x00A0;</b><b></b><i>float</i>
  <b>-mintm&#x00A0;</b><b></b><i>float</i>
  <b>-maxtm&#x00A0;</b><b></b><i>float</i>
  <b>-maxdifftm&#x00A0;</b><b></b><i>float</i>
  <b>-ogcpercent&#x00A0;</b><b></b><i>float</i>
  <b>-mingc&#x00A0;</b><b></b><i>float</i>
  <b>-maxgc&#x00A0;</b><b></b><i>float</i>
  <b>-saltconc&#x00A0;</b><b></b><i>float</i>
  <b>-dnaconc&#x00A0;</b><b></b><i>float</i>
  <b>-maxpolyx&#x00A0;</b><b></b><i>integer</i>
  <b>-psizeopt&#x00A0;</b><b></b><i>integer</i>
  <b>-prange&#x00A0;</b><b></b><i>range</i>
  <b>-ptmopt&#x00A0;</b><b></b><i>float</i>
  <b>-ptmmin&#x00A0;</b><b></b><i>float</i>
  <b>-ptmmax&#x00A0;</b><b></b><i>float</i>
  <b>-oexcludedregion&#x00A0;</b><b></b> <i>range</i>
  <b>-oligoinput&#x00A0;</b><b></b><i>string</i>
  <b>-osizeopt&#x00A0;</b><b></b><i>integer</i>
  <b>-ominsize&#x00A0;</b><b></b><i>integer</i>
  <b>-omaxsize&#x00A0;</b><b></b><i>integer</i>
  <b>-otmopt&#x00A0;</b><b></b><i>float</i>
  <b>-otmmin&#x00A0;</b><b></b><i>float</i>
  <b>-otmmax&#x00A0;</b><b></b><i>float</i>
  <b>-ogcopt&#x00A0;</b><b></b><i>float</i>
  <b>-ogcmin&#x00A0;</b><b></b><i>float</i>
  <b>-ogcmax&#x00A0;</b><b></b><i>float</i>
  <b>-osaltconc&#x00A0;</b><b></b><i>float</i>
  <b>-odnaconc&#x00A0;</b><b></b><i>float</i>
  <b>-oanyself&#x00A0;</b><b></b><i>float</i>
  <b>-oendself&#x00A0;</b><b></b><i>float</i>
  <b>-opolyxmax&#x00A0;</b><b></b><i>integer</i>
  <b>-omishybmax&#x00A0;</b><b></b> <i>float</i>
  <b>-explainflag&#x00A0;</b><b></b> <i>boolean</i>
  <b>-fileflag&#x00A0;</b><b></b> <i>boolean</i>
  <b>-firstbaseindex&#x00A0;</b><b></b><i>integer</i>
  <b>-pickanyway&#x00A0;</b><b></b><i>boolean</i>
  <b>-maxmispriming&#x00A0;</b><b></b><i>float</i>
  <b>-pairmaxmispriming&#x00A0;</b> <b></b><i>float</i>
  <b>-numnsaccepted&#x00A0;</b> <b></b><i>integer</i>
  <b>-selfany&#x00A0;</b><b></b> <i>float</i>
  <b>-selfend&#x00A0;</b><b></b><i>float</i>
  <b>-maxendstability&#x00A0;</b><b></b> <i>float</i>
  <b>-outfile&#x00A0;</b><b></b> <i>outfile</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>eprimer3</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>eprimer3</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Nucleic:Primers&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-sequence</b> <i>seqall</i>
<div style="margin-left: 4.00ex;">The sequence from which to choose primers. The
  sequence must be presented 5' to 3'</div>
<div class="Pp"></div>
<b>-primer</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">Tell EPrimer3 to pick primer(s) Default value:
  Y</div>
<div class="Pp"></div>
<b>-task</b> <i>list</i>
<div style="margin-left: 4.00ex;">Tell EPrimer3 what task to perform. Legal
  values are 1: 'Pick PCR primers', 2: 'Pick forward primer only', 3: 'Pick
  reverse primer only', 4: 'No primers needed'. Default value: 1</div>
<div class="Pp"></div>
<b>-hybridprobe</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">An 'internal oligo' is intended to be used as
  a hybridization probe (hyb probe) to detect the PCR product after
  amplification. Default value: N</div>
<div class="Pp"></div>
<b>-mishyblibraryfile</b> <i>infile</i>
<div style="margin-left: 4.00ex;">Similar to MISPRIMING-LIBRARY, except that the
  event we seek to avoid is hybridization of the internal oligo to sequences in
  this library rather than priming from them. The file must be in (a slightly
  restricted) FASTA format (W. B. Pearson and D.J. Lipman, PNAS 85:8 pp
  2444-2448 [1988]); we briefly discuss the organization of this file below. If
  this parameter is specified then EPrimer3 locally aligns each candidate oligo
  against each library sequence and rejects those primers for which the local
  alignment score times a specified weight (see below) exceeds
  INTERNAL-OLIGO-MAX-MISHYB. (The maximum value of the weight is arbitrarily set
  to 12.0.) Each sequence entry in the FASTA-format file must begin with an 'id
  line' that starts with '&gt;'. The contents of the id line is 'slightly
  restricted' in that EPrimer3 parses everything after any optional asterisk
  ('*') as a floating point number to use as the weight mentioned above. If the
  id line contains no asterisk then the weight defaults to 1.0. The alignment
  scoring system used is the same as for calculating complementarity among
  oligos (e.g. SELF-ANY). The remainder of an entry contains the sequence as
  lines following the id line up until a line starting with '&gt;' or the end of
  the file. Whitespace and newlines are ignored. Characters 'A', 'T', 'G', 'C',
  'a', 't', 'g', 'c' are retained and any other character is converted to 'N'
  (with the consequence that any IUB / IUPAC codes for ambiguous bases are
  converted to 'N'). There are no restrictions on line length. An empty value
  for this parameter indicates that no library should be used.</div>
<div class="Pp"></div>
<b>-mispriminglibraryfile</b> <i>infile</i>
<div style="margin-left: 4.00ex;">The name of a file containing a nucleotide
  sequence library of sequences to avoid amplifying (for example repetitive
  sequences, or possibly the sequences of genes in a gene family that should not
  be amplified.) The file must be in (a slightly restricted) FASTA format (W. B.
  Pearson and D.J. Lipman, PNAS 85:8 pp 2444-2448 [1988]); we briefly discuss
  the organization of this file below. If this parameter is specified then
  EPrimer3 locally aligns each candidate primer against each library sequence
  and rejects those primers for which the local alignment score times a
  specified weight (see below) exceeds MAX-MISPRIMING. (The maximum value of the
  weight is arbitrarily set to 100.0.) Each sequence entry in the FASTA-format
  file must begin with an 'id line' that starts with '&gt;'. The contents of the
  id line is 'slightly restricted' in that EPrimer3 parses everything after any
  optional asterisk ('*') as a floating point number to use as the weight
  mentioned above. If the id line contains no asterisk then the weight defaults
  to 1.0. The alignment scoring system used is the same as for calculating
  complementarity among oligos (e.g. SELF-ANY). The remainder of an entry
  contains the sequence as lines following the id line up until a line starting
  with '&gt;' or the end of the file. Whitespace and newlines are ignored.
  Characters 'A', 'T', 'G', 'C', 'a', 't', 'g', 'c' are retained and any other
  character is converted to 'N' (with the consequence that any IUB / IUPAC codes
  for ambiguous bases are converted to 'N'). There are no restrictions on line
  length. An empty value for this parameter indicates that no repeat library
  should be used.</div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<h2 class="Ss" title="Ss" id="Program_options"><a class="selflink" href="#Program_options">Program
  options</a></h2>
<b>-numreturn</b> <i>integer</i>
<div style="margin-left: 4.00ex;">The maximum number of primer pairs to return.
  Primer pairs returned are sorted by their 'quality', in other words by the
  value of the objective function (where a lower number indicates a better
  primer pair). Caution: setting this parameter to a large value will increase
  running time. Default value: 5</div>
<h2 class="Ss" title="Ss" id="Sequence_options"><a class="selflink" href="#Sequence_options">Sequence
  options</a></h2>
<b>-includedregion</b> <i>range</i>
<div style="margin-left: 4.00ex;">A sub-region of the given sequence in which to
  pick primers. For example, often the first dozen or so bases of a sequence are
  vector, and should be excluded from consideration. The value for this
  parameter has the form (start),(end) where (start) is the index of the first
  base to consider, and (end) is the last in the primer-picking region.</div>
<div class="Pp"></div>
<b>-targetregion</b> <i>range</i>
<div style="margin-left: 4.00ex;">If one or more Targets is specified then a
  legal primer pair must flank at least one of them. A Target might be a simple
  sequence repeat site (for example a CA repeat) or a single-base-pair
  polymorphism. The value should be a space-separated list of (start),(end)
  pairs where (start) is the index of the first base of a Target, and (end) is
  the last E.g. 50,51 requires primers to surround the 2 bases at positions 50
  and 51.</div>
<div class="Pp"></div>
<b>-excludedregion</b> <i>range</i>
<div style="margin-left: 4.00ex;">Primer oligos may not overlap any region
  specified in this tag. The associated value must be a space-separated list of
  (start),(end) pairs where (start) is the index of the first base of the
  excluded region, and and (end) is the last. This tag is useful for tasks such
  as excluding regions of low sequence quality or for excluding regions
  containing repetitive elements such as ALUs or LINEs. E.g. 401,407 68,70
  forbids selection of primers in the 7 bases starting at 401 and the 3 bases at
  68.</div>
<div class="Pp"></div>
<b>-forwardinput</b> <i>string</i>
<div style="margin-left: 4.00ex;">The sequence of a forward primer to check and
  around which to design reverse primers and optional internal oligos. Must be a
  substring of SEQUENCE.</div>
<div class="Pp"></div>
<b>-reverseinput</b> <i>string</i>
<div style="margin-left: 4.00ex;">The sequence of a reverse primer to check and
  around which to design forward primers and optional internal oligos. Must be a
  substring of the reverse strand of SEQUENCE.</div>
<h2 class="Ss" title="Ss" id="Primer_options"><a class="selflink" href="#Primer_options">Primer
  options</a></h2>
<b>-gcclamp</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Require the specified number of consecutive Gs
  and Cs at the 3' end of both the forward and reverse primer. (This parameter
  has no effect on the internal oligo if one is requested.)</div>
<div class="Pp"></div>
<b>-osize</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Optimum length (in bases) of a primer oligo.
  EPrimer3 will attempt to pick primers close to this length. Default value:
  20</div>
<div class="Pp"></div>
<b>-minsize</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Minimum acceptable length of a primer. Must be
  greater than 0 and less than or equal to MAX-SIZE. Default value: 18</div>
<div class="Pp"></div>
<b>-maxsize</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Maximum acceptable length (in bases) of a
  primer. Currently this parameter cannot be larger than 35. This limit is
  governed by the maximum oligo size for which EPrimer3's melting-temperature is
  valid. Default value: 27</div>
<div class="Pp"></div>
<b>-otm</b> <i>float</i>
<div style="margin-left: 4.00ex;">Optimum melting temperature(Celsius) for a
  primer oligo. EPrimer3 will try to pick primers with melting temperatures are
  close to this temperature. The oligo melting temperature formula in EPrimer3
  is that given in Rychlik, Spencer and Rhoads, Nucleic Acids Research, vol 18,
  num 21, pp 6409-6412 and Breslauer, Frank, Bloecker and Marky, Proc. Natl.
  Acad. Sci. USA, vol 83, pp 3746-3750. Please refer to the former paper for
  background discussion. Default value: 60.0</div>
<div class="Pp"></div>
<b>-mintm</b> <i>float</i>
<div style="margin-left: 4.00ex;">Minimum acceptable melting
  temperature(Celsius) for a primer oligo. Default value: 57.0</div>
<div class="Pp"></div>
<b>-maxtm</b> <i>float</i>
<div style="margin-left: 4.00ex;">Maximum acceptable melting
  temperature(Celsius) for a primer oligo. Default value: 63.0</div>
<div class="Pp"></div>
<b>-maxdifftm</b> <i>float</i>
<div style="margin-left: 4.00ex;">Maximum acceptable (unsigned) difference
  between the melting temperatures of the forward and reverse primers. Default
  value: 100.0</div>
<div class="Pp"></div>
<b>-ogcpercent</b> <i>float</i>
<div style="margin-left: 4.00ex;">Primer optimum GC percent. Default value:
  50.0</div>
<div class="Pp"></div>
<b>-mingc</b> <i>float</i>
<div style="margin-left: 4.00ex;">Minimum allowable percentage of Gs and Cs in
  any primer. Default value: 20.0</div>
<div class="Pp"></div>
<b>-maxgc</b> <i>float</i>
<div style="margin-left: 4.00ex;">Maximum allowable percentage of Gs and Cs in
  any primer generated by Primer. Default value: 80.0</div>
<div class="Pp"></div>
<b>-saltconc</b> <i>float</i>
<div style="margin-left: 4.00ex;">The millimolar concentration of salt (usually
  KCl) in the PCR. EPrimer3 uses this argument to calculate oligo melting
  temperatures. Default value: 50.0</div>
<div class="Pp"></div>
<b>-dnaconc</b> <i>float</i>
<div style="margin-left: 4.00ex;">The nanomolar concentration of annealing
  oligos in the PCR. EPrimer3 uses this argument to calculate oligo melting
  temperatures. The default (50nM) works well with the standard protocol used at
  the Whitehead/MIT Center for Genome Research--0.5 microliters of 20 micromolar
  concentration for each primer oligo in a 20 microliter reaction with 10
  nanograms template, 0.025 units/microliter Taq polymerase in 0.1 mM each dNTP,
  1.5mM MgCl2, 50mM KCl, 10mM Tris-HCL (pH 9.3) using 35 cycles with an
  annealing temperature of 56 degrees Celsius. This parameter corresponds to 'c'
  in Rychlik, Spencer and Rhoads' equation (ii) (Nucleic Acids Research, vol 18,
  num 21) where a suitable value (for a lower initial concentration of template)
  is 'empirically determined'. The value of this parameter is less than the
  actual concentration of oligos in the reaction because it is the concentration
  of annealing oligos, which in turn depends on the amount of template
  (including PCR product) in a given cycle. This concentration increases a great
  deal during a PCR; fortunately PCR seems quite robust for a variety of oligo
  melting temperatures. See ADVICE FOR PICKING PRIMERS. Default value:
  50.0</div>
<div class="Pp"></div>
<b>-maxpolyx</b> <i>integer</i>
<div style="margin-left: 4.00ex;">The maximum allowable length of a
  mononucleotide repeat in a primer, for example AAAAAA. Default value: 5</div>
<h2 class="Ss" title="Ss" id="Product_options"><a class="selflink" href="#Product_options">Product
  options</a></h2>
<b>-psizeopt</b> <i>integer</i>
<div style="margin-left: 4.00ex;">The optimum size for the PCR product. 0
  indicates that there is no optimum product size. Default value: 200</div>
<div class="Pp"></div>
<b>-prange</b> <i>range</i>
<div style="margin-left: 4.00ex;">The associated values specify the lengths of
  the product that the user wants the primers to create, and is a space
  separated list of elements of the form (x)-(y) where an (x)-(y) pair is a
  legal range of lengths for the product. For example, if one wants PCR products
  to be between 100 to 150 bases (inclusive) then one would set this parameter
  to 100-150. If one desires PCR products in either the range from 100 to 150
  bases or in the range from 200 to 250 bases then one would set this parameter
  to 100-150 200-250. EPrimer3 favors ranges to the left side of the parameter
  string. EPrimer3 will return legal primers pairs in the first range regardless
  the value of the objective function for these pairs. Only if there are an
  insufficient number of primers in the first range will EPrimer3 return primers
  in a subsequent range. Default value: 100-300</div>
<div class="Pp"></div>
<b>-ptmopt</b> <i>float</i>
<div style="margin-left: 4.00ex;">The optimum melting temperature for the PCR
  product. 0 indicates that there is no optimum temperature. Default value:
  0.0</div>
<div class="Pp"></div>
<b>-ptmmin</b> <i>float</i>
<div style="margin-left: 4.00ex;">The minimum allowed melting temperature of the
  amplicon. Please see the documentation on the maximum melting temperature of
  the product for details. Default value: -1000000.0</div>
<div class="Pp"></div>
<b>-ptmmax</b> <i>float</i>
<div style="margin-left: 4.00ex;">The maximum allowed melting temperature of the
  amplicon. Product Tm is calculated using the formula from Bolton and McCarthy,
  PNAS 84:1390 (1962) as presented in Sambrook, Fritsch and Maniatis, Molecular
  Cloning, p 11.46 (1989, CSHL Press). Tm = 81.5 + 16.6(log10([Na+])) +
  .41*(%GC) - 600/length Where [Na+} is the molar sodium concentration, (%GC) is
  the percent of Gs and Cs in the sequence, and length is the length of the
  sequence. A similar formula is used by the prime primer selection program in
  GCG, which instead uses 675.0/length in the last term (after F. Baldino, Jr,
  M.-F. Chesselet, and M.E. Lewis, Methods in Enzymology 168:766 (1989) eqn (1)
  on page 766 without the mismatch and formamide terms). The formulas here and
  in Baldino et al. assume Na+ rather than K+. According to J.G. Wetmur,
  Critical Reviews in BioChem. and Mol. Bio. 26:227 (1991) 50 mM K+ should be
  equivalent in these formulae to .2 M Na+. EPrimer3 uses the same salt
  concentration value for calculating both the primer melting temperature and
  the oligo melting temperature. If you are planning to use the PCR product for
  hybridization later this behavior will not give you the Tm under hybridization
  conditions. Default value: 1000000.0</div>
<h2 class="Ss" title="Ss" id="Internal_oligo_input"><a class="selflink" href="#Internal_oligo_input">Internal
  oligo input</a></h2>
<b>-oexcludedregion</b> <i>range</i>
<div style="margin-left: 4.00ex;">Middle oligos may not overlap any region
  specified by this tag. The associated value must be a space-separated list of
  (start),(end) pairs, where (start) is the index of the first base of an
  excluded region, and (end) is the last. Often one would make Target regions
  excluded regions for internal oligos.</div>
<div class="Pp"></div>
<b>-oligoinput</b> <i>string</i>
<div style="margin-left: 4.00ex;">The sequence of an internal oligo to check and
  around which to design forward and reverse primers. Must be a substring of
  SEQUENCE.</div>
<h2 class="Ss" title="Ss" id="Internal_oligo_options"><a class="selflink" href="#Internal_oligo_options">Internal
  oligo options</a></h2>
<b>-osizeopt</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Optimum length (in bases) of an internal
  oligo. EPrimer3 will attempt to pick primers close to this length. Default
  value: 20</div>
<div class="Pp"></div>
<b>-ominsize</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Minimum acceptable length of an internal
  oligo. Must be greater than 0 and less than or equal to
  INTERNAL-OLIGO-MAX-SIZE. Default value: 18</div>
<div class="Pp"></div>
<b>-omaxsize</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Maximum acceptable length (in bases) of an
  internal oligo. Currently this parameter cannot be larger than 35. This limit
  is governed by maximum oligo size for which EPrimer3's melting-temperature is
  valid. Default value: 27</div>
<div class="Pp"></div>
<b>-otmopt</b> <i>float</i>
<div style="margin-left: 4.00ex;">Optimum melting temperature (Celsius) for an
  internal oligo. EPrimer3 will try to pick oligos with melting temperatures
  that are close to this temperature. The oligo melting temperature formula in
  EPrimer3 is that given in Rychlik, Spencer and Rhoads, Nucleic Acids Research,
  vol 18, num 21, pp 6409-6412 and Breslauer, Frank, Bloecker and Marky, Proc.
  Natl. Acad. Sci. USA, vol 83, pp 3746-3750. Please refer to the former paper
  for background discussion. Default value: 60.0</div>
<div class="Pp"></div>
<b>-otmmin</b> <i>float</i>
<div style="margin-left: 4.00ex;">Minimum acceptable melting
  temperature(Celsius) for an internal oligo. Default value: 57.0</div>
<div class="Pp"></div>
<b>-otmmax</b> <i>float</i>
<div style="margin-left: 4.00ex;">Maximum acceptable melting temperature
  (Celsius) for an internal oligo. Default value: 63.0</div>
<div class="Pp"></div>
<b>-ogcopt</b> <i>float</i>
<div style="margin-left: 4.00ex;">Internal oligo optimum GC percent. Default
  value: 50.0</div>
<div class="Pp"></div>
<b>-ogcmin</b> <i>float</i>
<div style="margin-left: 4.00ex;">Minimum allowable percentage of Gs and Cs in
  an internal oligo. Default value: 20.0</div>
<div class="Pp"></div>
<b>-ogcmax</b> <i>float</i>
<div style="margin-left: 4.00ex;">Maximum allowable percentage of Gs and Cs in
  any internal oligo generated by Primer. Default value: 80.0</div>
<div class="Pp"></div>
<b>-osaltconc</b> <i>float</i>
<div style="margin-left: 4.00ex;">The millimolar concentration of salt (usually
  KCl) in the hybridization. EPrimer3 uses this argument to calculate internal
  oligo melting temperatures. Default value: 50.0</div>
<div class="Pp"></div>
<b>-odnaconc</b> <i>float</i>
<div style="margin-left: 4.00ex;">The nanomolar concentration of annealing
  internal oligo in the hybridization. Default value: 50.0</div>
<div class="Pp"></div>
<b>-oanyself</b> <i>float</i>
<div style="margin-left: 4.00ex;">The maximum allowable local alignment score
  when testing an internal oligo for (local) self-complementarity. Local
  self-complementarity is taken to predict the tendency of oligos to anneal to
  themselves The scoring system gives 1.00 for complementary bases, -0.25 for a
  match of any base (or N) with an N, -1.00 for a mismatch, and -2.00 for a gap.
  Only single-base-pair gaps are allowed. For example, the alignment 5' ATCGNA
  3' || | | 3' TA-CGT 5' is allowed (and yields a score of 1.75), but the
  alignment 5' ATCCGNA 3' || | | 3' TA--CGT 5' is not considered. Scores are
  non-negative, and a score of 0.00 indicates that there is no reasonable local
  alignment between two oligos. Default value: 12.00</div>
<div class="Pp"></div>
<b>-oendself</b> <i>float</i>
<div style="margin-left: 4.00ex;">The maximum allowable 3'-anchored global
  alignment score when testing a single oligo for self-complementarity. The
  scoring system is as for the Maximum Complementarity argument. In the examples
  above the scores are 7.00 and 6.00 respectively. Scores are non-negative, and
  a score of 0.00 indicates that there is no reasonable 3'-anchored global
  alignment between two oligos. In order to estimate 3'-anchored global
  alignments for candidate oligos, Primer assumes that the sequence from which
  to choose oligos is presented 5' to 3'. INTERNAL-OLIGO-SELF-END is meaningless
  when applied to internal oligos used for hybridization-based detection, since
  primer-dimer will not occur. We recommend that INTERNAL-OLIGO-SELF-END be set
  at least as high as INTERNAL-OLIGO-SELF-ANY. Default value: 12.00</div>
<div class="Pp"></div>
<b>-opolyxmax</b> <i>integer</i>
<div style="margin-left: 4.00ex;">The maximum allowable length of an internal
  oligo mononucleotide repeat, for example AAAAAA. Default value: 5</div>
<div class="Pp"></div>
<b>-omishybmax</b> <i>float</i>
<div style="margin-left: 4.00ex;">Similar to MAX-MISPRIMING except that this
  parameter applies to the similarity of candidate internal oligos to the
  library specified in INTERNAL-OLIGO-MISHYB-LIBRARY. Default value: 12.0</div>
<h2 class="Ss" title="Ss" id="Advanced_section"><a class="selflink" href="#Advanced_section">Advanced
  section</a></h2>
<b>-explainflag</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this flag is true, produce LEFT-EXPLAIN,
  RIGHT-EXPLAIN, and INTERNAL-OLIGO-EXPLAIN output tags, which are intended to
  provide information on the number of oligos and primer pairs that EPrimer3
  examined, and statistics on the number discarded for various reasons. Default
  value: N</div>
<div class="Pp"></div>
<b>-fileflag</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If the associated value is true, then EPrimer3
  creates two output files for each input SEQUENCE. File (sequence-id).for lists
  all acceptable forward primers for (sequence-id), and (sequence-id).rev lists
  all acceptable reverse primers for (sequence-id), where (sequence-id) is the
  value of the SEQUENCE-ID tag (which must be supplied). In addition, if the
  input tag TASK is 1 or 4, EPrimer3 produces a file (sequence-id).int, which
  lists all acceptable internal oligos. Default value: N</div>
<div class="Pp"></div>
<b>-firstbaseindex</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This parameter is the index of the first base
  in the input sequence. For input and output using 1-based indexing (such as
  that used in GenBank and to which many users are accustomed) set this
  parameter to 1. For input and output using 0-based indexing set this parameter
  to 0. (This parameter also affects the indexes in the contents of the files
  produced when the primer file flag is set.) Default value: 1</div>
<div class="Pp"></div>
<b>-pickanyway</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If true pick a primer pair even if LEFT-INPUT,
  RIGHT-INPUT, or INTERNAL-OLIGO-INPUT violates specific constraints. Default
  value: N</div>
<div class="Pp"></div>
<b>-maxmispriming</b> <i>float</i>
<div style="margin-left: 4.00ex;">The maximum allowed weighted similarity with
  any sequence in MISPRIMING-LIBRARY. Default value: 12.00</div>
<div class="Pp"></div>
<b>-pairmaxmispriming</b> <i>float</i>
<div style="margin-left: 4.00ex;">The maximum allowed sum of weighted
  similarities of a primer pair (one similarity for each primer) with any single
  sequence in MISPRIMING-LIBRARY. Default value: 24.00</div>
<div class="Pp"></div>
<b>-numnsaccepted</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Maximum number of unknown bases (N) allowable
  in any primer.</div>
<div class="Pp"></div>
<b>-selfany</b> <i>float</i>
<div style="margin-left: 4.00ex;">The maximum allowable local alignment score
  when testing a single primer for (local) self-complementarity and the maximum
  allowable local alignment score when testing for complementarity between
  forward and reverse primers. Local self-complementarity is taken to predict
  the tendency of primers to anneal to each other without necessarily causing
  self-priming in the PCR. The scoring system gives 1.00 for complementary
  bases, -0.25 for a match of any base (or N) with an N, -1.00 for a mismatch,
  and -2.00 for a gap. Only single-base-pair gaps are allowed. For example, the
  alignment 5' ATCGNA 3' ...|| | | 3' TA-CGT 5' is allowed (and yields a score
  of 1.75), but the alignment 5' ATCCGNA 3' ...|| | | 3' TA--CGT 5' is not
  considered. Scores are non-negative, and a score of 0.00 indicates that there
  is no reasonable local alignment between two oligos. Default value: 8.00</div>
<div class="Pp"></div>
<b>-selfend</b> <i>float</i>
<div style="margin-left: 4.00ex;">The maximum allowable 3'-anchored global
  alignment score when testing a single primer for self-complementarity, and the
  maximum allowable 3'-anchored global alignment score when testing for
  complementarity between forward and reverse primers. The 3'-anchored global
  alignment score is taken to predict the likelihood of PCR-priming
  primer-dimers, for example 5' ATGCCCTAGCTTCCGGATG 3' .............||| |||||
  ..........3' AAGTCCTACATTTAGCCTAGT 5' or 5' AGGCTATGGGCCTCGCGA 3'
  ...............|||||| ............3' AGCGCTCCGGGTATCGGA 5' The scoring system
  is as for the Maximum Complementarity argument. In the examples above the
  scores are 7.00 and 6.00 respectively. Scores are non-negative, and a score of
  0.00 indicates that there is no reasonable 3'-anchored global alignment
  between two oligos. In order to estimate 3'-anchored global alignments for
  candidate primers and primer pairs, Primer assumes that the sequence from
  which to choose primers is presented 5' to 3'. It is nonsensical to provide a
  larger value for this parameter than for the Maximum (local) Complementarity
  parameter because the score of a local alignment will always be at least as
  great as the score of a global alignment. Default value: 3.00</div>
<h2 class="Ss" title="Ss" id="Primer_penalty_weights"><a class="selflink" href="#Primer_penalty_weights">Primer
  penalty weights</a></h2>
<b>-maxendstability</b> <i>float</i>
<div style="margin-left: 4.00ex;">The maximum stability for the five 3' bases of
  a forward or reverse primer. Bigger numbers mean more stable 3' ends. The
  value is the maximum delta G for duplex disruption for the five 3' bases as
  calculated using the nearest neighbor parameters published in Breslauer,
  Frank, Bloecker and Marky, Proc. Natl. Acad. Sci. USA, vol 83, pp 3746-3750.
  EPrimer3 uses a completely permissive default value for backward compatibility
  (which we may change in the next release). Rychlik recommends a maximum value
  of 9 (Wojciech Rychlik, 'Selection of Primers for Polymerase Chain Reaction'
  in BA White, Ed., 'Methods in Molecular Biology, Vol. 15: PCR Protocols:
  Current Methods and Applications', 1993, pp 31-40, Humana Press, Totowa NJ).
  Default value: 9.0</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
eprimer3 is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
