<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:07:41 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>EST2GENOME(1e) EMBOSS Manual for Debian
EST2GENOME(1e)</p>

<p style="margin-top: 1em">NAME <br>
est2genome - Align EST sequences to genomic DNA sequence</p>

<p style="margin-top: 1em">SYNOPSIS <br>
est2genome -estsequence seqall -genomesequence sequence
[-match integer] [-mismatch integer] [-gappenalty integer]
[-intronpenalty integer] [-splicepenalty integer] <br>
[-minscore integer] -reverse boolean -usesplice boolean
-mode list -best boolean -space float -shuffle integer -seed
integer -outfile outfile -align boolean <br>
-width integer</p>

<p style="margin-top: 1em">est2genome -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
est2genome is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Alignment:Global&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-estsequence seqall</p>

<p style="margin-top: 1em">-genomesequence sequence</p>

<p style="margin-top: 1em">Additional section <br>
-match integer <br>
Default value: 1</p>

<p style="margin-top: 1em">-mismatch integer <br>
Default value: 1</p>

<p style="margin-top: 1em">-gappenalty integer <br>
Cost for deleting a single base in either sequence,
excluding introns Default value: 2</p>

<p style="margin-top: 1em">-intronpenalty integer <br>
Cost for an intron, independent of length. Default value:
40</p>

<p style="margin-top: 1em">-splicepenalty integer <br>
Cost for an intron, independent of length and
starting/ending on donor-acceptor sites Default value:
20</p>

<p style="margin-top: 1em">-minscore integer <br>
Exclude alignments with scores below this threshold score.
Default value: 30</p>

<p style="margin-top: 1em">Advanced section <br>
-reverse boolean <br>
Reverse the orientation of the EST sequence</p>

<p style="margin-top: 1em">-usesplice boolean <br>
Use donor and acceptor splice sites. If you want to ignore
donor-acceptor sites then set this to be false. Default
value: Y</p>

<p style="margin-top: 1em">-mode list <br>
This determines the comparison mode. The default value is
&rsquo;both&rsquo;, in which case both strands of the est
are compared assuming a forward gene direction (ie GT/AG
splice <br>
sites), and the best comparison redone assuming a reversed
(CT/AC) gene splicing direction. The other allowed modes are
&rsquo;forward&rsquo;, when just the forward strand is
searched, <br>
and &rsquo;reverse&rsquo;, ditto for the reverse strand.
Default value: both</p>

<p style="margin-top: 1em">-best boolean <br>
You can print out all comparisons instead of just the best
one by setting this to be false. Default value: Y</p>

<p style="margin-top: 1em">-space float <br>
For linear-space recursion. If product of sequence lengths
divided by 4 exceeds this then a divide-and-conquer strategy
is used to control the memory requirements. In this <br>
way very long sequences can be aligned. If you have a
machine with plenty of memory you can raise this parameter
(but do not exceed the machine&rsquo;s physical RAM) Default
value: <br>
10.0</p>

<p style="margin-top: 1em">-shuffle integer</p>

<p style="margin-top: 1em">-seed integer <br>
Default value: 20825</p>

<p style="margin-top: 1em">Output section <br>
-outfile outfile</p>

<p style="margin-top: 1em">-align boolean <br>
Show the alignment. The alignment includes the first and
last 5 bases of each intron, together with the intron width.
The direction of splicing is indicated by angle brackets
<br>
(forward or reverse) or ???? (unknown).</p>

<p style="margin-top: 1em">-width integer <br>
Default value: 50</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
est2genome is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
EST2GENOME(1e)</p>
<hr>
</body>
</html>
