<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>EST2GENOME(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">EST2GENOME(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">EST2GENOME(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
est2genome - Align EST sequences to genomic DNA sequence
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 11.00ex; text-indent: -11.00ex;"><b>est2genome</b>
  <b>-estsequence&#x00A0;</b><b></b><i>seqall</i>
  <b>-genomesequence&#x00A0;</b><b></b><i>sequence</i>
  [<b>-match&#x00A0;</b><b></b> <i>integer</i>]
  [<b>-mismatch&#x00A0;</b><b></b><i>integer</i>]
  [<b>-gappenalty&#x00A0;</b><b></b> <i>integer</i>]
  [<b>-intronpenalty&#x00A0;</b> <b></b><i>integer</i>]
  [<b>-splicepenalty&#x00A0;</b> <b></b><i>integer</i>]
  [<b>-minscore&#x00A0;</b> <b></b><i>integer</i>]
  <b>-reverse&#x00A0;</b><b></b> <i>boolean</i>
  <b>-usesplice&#x00A0;</b><b></b><i>boolean</i>
  <b>-mode&#x00A0;</b><b></b><i>list</i>
  <b>-best&#x00A0;</b><b></b><i>boolean</i>
  <b>-space&#x00A0;</b><b></b><i>float</i>
  <b>-shuffle&#x00A0;</b><b></b><i>integer</i>
  <b>-seed&#x00A0;</b><b></b><i>integer</i>
  <b>-outfile&#x00A0;</b><b></b><i>outfile</i>
  <b>-align&#x00A0;</b><b></b><i>boolean</i>
  <b>-width&#x00A0;</b><b></b><i>integer</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 11.00ex; text-indent: -11.00ex;"><b>est2genome</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>est2genome</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Alignment:Global&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-estsequence</b> <i>seqall</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-genomesequence</b> <i>sequence</i>
<div style="margin-left: 4.00ex;"></div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-match</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Default value: 1</div>
<div class="Pp"></div>
<b>-mismatch</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Default value: 1</div>
<div class="Pp"></div>
<b>-gappenalty</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Cost for deleting a single base in either
  sequence, excluding introns Default value: 2</div>
<div class="Pp"></div>
<b>-intronpenalty</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Cost for an intron, independent of length.
  Default value: 40</div>
<div class="Pp"></div>
<b>-splicepenalty</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Cost for an intron, independent of length and
  starting/ending on donor-acceptor sites Default value: 20</div>
<div class="Pp"></div>
<b>-minscore</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Exclude alignments with scores below this
  threshold score. Default value: 30</div>
<h2 class="Ss" title="Ss" id="Advanced_section"><a class="selflink" href="#Advanced_section">Advanced
  section</a></h2>
<b>-reverse</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Reverse the orientation of the EST
  sequence</div>
<div class="Pp"></div>
<b>-usesplice</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Use donor and acceptor splice sites. If you
  want to ignore donor-acceptor sites then set this to be false. Default value:
  Y</div>
<div class="Pp"></div>
<b>-mode</b> <i>list</i>
<div style="margin-left: 4.00ex;">This determines the comparison mode. The
  default value is 'both', in which case both strands of the est are compared
  assuming a forward gene direction (ie GT/AG splice sites), and the best
  comparison redone assuming a reversed (CT/AC) gene splicing direction. The
  other allowed modes are 'forward', when just the forward strand is searched,
  and 'reverse', ditto for the reverse strand. Default value: both</div>
<div class="Pp"></div>
<b>-best</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">You can print out all comparisons instead of
  just the best one by setting this to be false. Default value: Y</div>
<div class="Pp"></div>
<b>-space</b> <i>float</i>
<div style="margin-left: 4.00ex;">For linear-space recursion. If product of
  sequence lengths divided by 4 exceeds this then a divide-and-conquer strategy
  is used to control the memory requirements. In this way very long sequences
  can be aligned. If you have a machine with plenty of memory you can raise this
  parameter (but do not exceed the machine's physical RAM) Default value:
  10.0</div>
<div class="Pp"></div>
<b>-shuffle</b> <i>integer</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-seed</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Default value: 20825</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-align</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Show the alignment. The alignment includes the
  first and last 5 bases of each intron, together with the intron width. The
  direction of splicing is indicated by angle brackets (forward or reverse) or
  ???? (unknown).</div>
<div class="Pp"></div>
<b>-width</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Default value: 50</div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
est2genome is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
