<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>exonerate(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">exonerate(1)</td>
    <td class="head-vol">sequence comparison tool</td>
    <td class="head-rtitle">exonerate(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
exonerate - a generic tool for sequence comparison
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>exonerate [ options ] &lt;query path&gt; &lt;target path&gt;</b>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>exonerate</b> is a general tool for sequence comparison.
<div style="height: 1.00em;">&#x00A0;</div>
It uses the <b>C4</b> dynamic programming library. It is designed to be both
  general and fast. It can produce either gapped or ungapped alignments,
  according to a variety of different alignment models. The C4 library allows
  sequence alignment using a reduced space full dynamic programming
  implementation, but also allows automated generation of heuristics from the
  alignment models, using bounded sparse dynamic programming, so that these
  alignments may also be rapidly generated. Alignments generated using these
  heuristics will represent a valid path through the alignment model, yet
  (unlike the exhaustive alignments), the results are not guaranteed to be
  optimal.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="CONVENTIONS"><a class="selflink" href="#CONVENTIONS">CONVENTIONS</a></h1>
A number of conventions (and idiosyncracies) are used within exonerate. An
  understanding of them facilitates interpretation of the output.
<dl class="Bl-tag">
  <dt class="It-tag"><b>Coordinates</b></dt>
  <dd class="It-tag">An in-between coordinate system is used, where the
      positions are counted between the symbols, rather than on the symbols.
      This numbering scheme starts from zero. This numbering is shown below for
      the sequence &quot;ACGT&quot;:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
<div class="Pp"></div>
 A C G T
0 1 2 3 4
<div class="Pp"></div>
<div style="height: 1.00em;">&#x00A0;</div>
    </pre>
    Hence the subsequence &quot;CG&quot; would have start=1, end=3, and
      length=2. This coordinate system is used internally in exonerate, and for
      all the output formats produced with the exception of the &quot;human
      readable&quot; alignment display and the GFF output where convention and
      standards dictate otherwise.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Reverse Complements</b></dt>
  <dd class="It-tag">When an alignment is reported on the reverse complement of
      a sequence, the coordinates are simply given on the reverse complement
      copy of the sequence. Hence positions on the sequences are never negative.
      Generally, the forward strand is indicated by '+', the reverse strand by
      '-', and an unknown or not-applicable strand (as in the case of a protein
      sequence) is indicated by '.'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Alignment Scores</b></dt>
  <dd class="It-tag">Currently, only the raw alignment scores are displayed.
      This score just is the sum of transistion scores used in the dynamic
      programming. For example, in the case of a Smith-Waterman alignment, this
      will be the sum of the substitution matrix scores and the gap penalties.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="GENERAL_OPTIONS"><a class="selflink" href="#GENERAL_OPTIONS">GENERAL
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">Most arguments have short and long forms. The long
    forms</dt>
  <dd class="It-tag">are more likely to be stable over time, and hence should be
      used in scripts which call exonerate.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h | --shorthelp &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Show help. This will display a concise summary of the
      available options, defaults and values currently set.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help &lt;boolean&gt;</b></dt>
  <dd class="It-tag">This shows all the help options including the defaults, the
      value currently set, and the environment variable which may be used to set
      each parameter. There will be an indication of which options are
      mandatory. Mandatory options have no default, and must have a value
      supplied for exonerate to run. If mandatory options are used in order,
      their flags may be skipped from the command line (see examples below).
      Unlike this man page, the information from this option will always be up
      to date with the latest version of the program.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v | --version &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Display the version number. Also displays other information
      such as the build date and glib version used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEQUENCE_INPUT_OPTIONS"><a class="selflink" href="#SEQUENCE_INPUT_OPTIONS">SEQUENCE
  INPUT OPTIONS</a></h1>
Pairwise comparisons will be performed between all query sequences and all
  target sequences. Generally, for the best performance, shorter sequences (eg.
  ESTs, shotgun reads, proteins) should be used as the query sequences, and
  longer sequences (eg. genomic sequences) should be used as the target
  sequences.
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q | --query &lt;paths&gt;</b></dt>
  <dd class="It-tag">Specify the query sequences required. These must be in a
      FASTA format file. Single or muiltiple query sequences may be supplied.
      Additionally multiple copies of the fasta file may be supplied following a
      --query flag, or by using with multiple --query flags.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t | --target &lt;paths&gt;</b></dt>
  <dd class="It-tag">Specify the target sequences required. Also, must be in a
      FASTA format file. As with the query sequences, single or multiple target
      sequences and files may be supplied. The target filename may by replace by
      a server name and port number in the form of <b>hostname:port</b> when
      using <b>exonerate-server.</b> See the man page for
      <b>exonerate-server</b> for more information on running exonerate in
      client:server mode. <b>NEW(v2.4.0):</b> multiple servers may now be used.
      These will be queried in parallel if you have set the <b>--cores</b>
      option. <b>NEW(v2.4.0):</b> If an input file is not a FASTA format file,
      it is assumed to contain a list of other fasta files, directories or
      servers (one per line).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q | --querytype &lt;dna | protein&gt;</b></dt>
  <dd class="It-tag">Specify the query type to use. If this is not supplied, the
      query type is assumed to be DNA when the first sequence in the file
      contains more than 85% [ACGTN] bases. Otherwise, it is assumed to be
      peptide. This option forces the query type as some nucleotide and peptide
      sequences can fall either side of this threshold.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T | --targettype &lt;dna | protein&gt;</b></dt>
  <dd class="It-tag">Specify the target type to use. The same as
      <b>--querytype</b> (above), except that it applies to the target.
      Specifying the sequence type will avoid the overhead of having to read the
      first sequence in the database twice (which may be significant with
      chromosome-sized sequences)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--querychunkid &lt;id&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--querychunktotal &lt;total&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--targetchunkid &lt;id&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--targetchunktotal &lt;total&gt;</b></dt>
  <dd class="It-tag">These options to facilitate running exonerate on compute
      farms, and avoid having to split up sequence databases into small chunks
      to run on different nodes. If, for example, you wished to split the target
      database into three parts, you would run three exonerate jobs on different
      nodes including the options:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">--targetchunkid 1 --targetchunktotal 3</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--targetchunkid 2 --targetchunktotal 3</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--targetchunkid 3 --targetchunktotal 3</dt>
  <dd class="It-tag"></dd>
</dl>
<div class="Pp"></div>
NB. The granularity offered by this option only goes down to a single sequence,
  so when there are more chunks than sequences in the database, some processes
  will do nothing.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V | --verbose &lt;int&gt;</b></dt>
  <dd class="It-tag">Be verbose - show information about what is going on during
      the analysis. The default is 1 (little information), the higher the number
      given, the more information is printed. To silence all the default output
      from exonerate, use --verbose 0 --showalignment no --showvulgar no
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="ANALYSIS_OPTIONS"><a class="selflink" href="#ANALYSIS_OPTIONS">ANALYSIS
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E | --exhaustive &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Specify whether or not exhaustive alignment should be used.
      By default, this is FALSE, and alignment heuristics will be used. If it is
      set to TRUE, an exhaustive alignment will be calculated. This requires
      quadratic time, and will be much, much slower, but will provide the
      optimal result for the given model.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B | --bigseq &lt;int&gt;</b></dt>
  <dd class="It-tag">Perform alignment of large (multi-megabase) sequences. This
      is very memory efficient and fast when both sequences are
      chromosome-sized, but currently does not currently permit the use of a
      word neighbourhood (ie. exactly matching seeds only).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--revcomp &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Include comparison of the reverse complement of the query
      and target where possible. By default, this option is enabled, but when
      you know the gene is definitely on the forward strand of the query and
      target, this option can halve the time taken to compute alignments.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--forcescan &lt;none | query | target&gt;</b></dt>
  <dd class="It-tag">Force the FSM to scan the query sequence rather than the
      target. This option is useful, for example, if you have a single piece of
      genomic sequence and you with to compare it to the whole of dbEST. By
      scanning the database, rather than the query, the analysis will be
      completed much more quickly, as the overheads of multiple query FSM
      construction, multiple target reading and splice site predictions will be
      removed. By default, exonerate will guess the optimal strategy based on
      database sequence sizes.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--saturatethreshold &lt;number&gt;</b></dt>
  <dd class="It-tag">When set to zero, this option does nothing. Otherwise, once
      more than this number of words (in addition to the expected number of
      words by chance) have matched a position on the query, the position on the
      query will be 'numbed' (ignore further matches) for the current pairwise
      comparison.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--customserver &lt;command&gt;</b></dt>
  <dd class="It-tag">When using exonerate in client:server mode with a
      non-standard server, this command allows you to send a custom command to
      the server. This command is sent by the client (exonerate) before any
      other commands, and is provided as a way of passing parameters or other
      commands specific to the custom server. See the <b>exonerate-server</b>
      man page for more information on running exonerate in client:server
    mode.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cores &lt;number&gt;</b></dt>
  <dd class="It-tag">The number of cores/CPUs/threads that should be used. On a
      multi-core or multi-CPU machine, increasing this ammount allows alignment
      computations to run in parallel on separate CPUs/cores. NB. Generally, it
      is better to parallelise the analysis by splitting it up into separate
      jobs, but this option may prove useful for problems such as interactive
      single-gene queries.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="FASTA_DATABASE_OPTIONS"><a class="selflink" href="#FASTA_DATABASE_OPTIONS">FASTA
  DATABASE OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fastasuffix &lt;extension&gt;</b></dt>
  <dd class="It-tag">If any of the inputs given with <b>--query</b> or
      <b>--target</b> are directories, then exonerate will recursively descent
      these directories, reading all files ending with this suffix as fasta
      format input.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="GAPPED_ALIGNMENT_OPTIONS"><a class="selflink" href="#GAPPED_ALIGNMENT_OPTIONS">GAPPED
  ALIGNMENT OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m | --model &lt;alignment model&gt;</b></dt>
  <dd class="It-tag">Specify the alignment model to use. The models currently
      supported are:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>ungapped</b></dt>
  <dd class="It-tag">The simplest type of model, used by default. An appropriate
      model with be selected automatically for the type of input sequences
      provided.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ungapped:trans</b></dt>
  <dd class="It-tag">This ungapped model includes translation of all frames of
      both the query and target sequences. This is similar to an ungapped
      tblastx type search.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>affine:global</b></dt>
  <dd class="It-tag">This performs gapped global alignment, similar to the
      Needleman-Wunsch algorithm, except with affine gaps. Global alignment
      requires that both the sequences in their entirety are included in the
      alignment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>affine:bestfit</b></dt>
  <dd class="It-tag">This performs a best fit or best location alignment of the
      query onto the target sequence. The entire query sequence will be included
      in the alignment, but only the best location for its alignment on the
      target sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>affine:local</b></dt>
  <dd class="It-tag">This is local alignment with affine gaps, similar to the
      Smith-Waterman-Gotoh algorithm. A general-purpose alignment algorithm. As
      this is local alignment, any subsequence of the query and target sequence
      may appear in the alignment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>affine:overlap</b></dt>
  <dd class="It-tag">This type of alignment finds the best overlap between the
      query and target. The overlap alignment must include the start of the
      query or target and the end of the query or the target sequence, to align
      sequences which overlap at the ends, or in the mid-section of a longer
      sequence.. This is the type of alignment frequently used in assembly
      algorithms.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>est2genome</b></dt>
  <dd class="It-tag">This model is similar to the affine:local model, but it
      also includes intron modelling on the target sequence to allow alignment
      of spliced to unspliced coding sequences for both forward and reversed
      genes. This is similar to the alignment models used in programs such as
      EST_GENOME and sim4.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ner</b></dt>
  <dd class="It-tag">NERs are non-equivalenced regions - large regions in both
      the query and target which are not aligned. This model can be used for
      protein alignments where strongly conserved helix regions will be aligned,
      but weakly conserved loop regions are not. Similarly, this model could be
      used to look for co-linearly conserved regions in comparison of genomic
      sequences.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>protein2dna</b></dt>
  <dd class="It-tag">This model compares a protein sequence to a DNA sequence,
      incorporating all the appropriate gaps and frameshifts.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>protein2dna:bestfit</b></dt>
  <dd class="It-tag">This is a bestfit version of the protein2dna model, with
      which the entire protein is included in the alignment. It is currently
      only available when using exhaustive alignment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>protein2genome</b></dt>
  <dd class="It-tag">This model allows alignment of a protein sequence to
      genomic DNA. This is similar to the protein2dna model, with the addition
      of modelling of introns and intron phases. This model is similar to those
      used by genewise.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>protein2genome:bestfit</b></dt>
  <dd class="It-tag">This is a bestfit version of the protein2genome model, with
      which the entire protein is included in the alignment. It is currently
      only available when using exhaustive alignment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>coding2coding</b></dt>
  <dd class="It-tag">This model is similar to the ungapped:trans model, except
      that gaps and frameshifts are allowed. It is similar to a gapped tblastx
      search.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>coding2genome</b></dt>
  <dd class="It-tag">This is similar to the est2genome model, except that the
      query sequence is translated during comparison, allowing a more sensitive
      comparison.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cdna2genome</b></dt>
  <dd class="It-tag">This combines properties of the est2genome and
      coding2genome models, to allow modeling of an whole cDNA where a central
      coding region can be flanked by non-coding UTRs. When the CDS start and
      end is known it may be specified using the --annotation option (see below)
      to permit only the correct coding region to appear in the alignemnt.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>genome2genome</b></dt>
  <dd class="It-tag">This model is similar to the coding2coding model, except
      introns are modelled on both sequences. (not working well yet)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
</div>
<dl class="Bl-tag">
  <dt class="It-tag">The short names u, u:t, a:g, a:b, a:l, a:o, e2g, ner,</dt>
  <dd class="It-tag">p2d, p2d:b p2g, p2g:b, c2c, c2g cd2g and g2g can also be
      used for specifying models.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s | --score &lt;threshold&gt;</b></dt>
  <dd class="It-tag">This is the overall score threshold. Alignments will not be
      reported below this threshold. For heuristic alignments, the higher this
      threshold, the less time the analysis will take.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--percent &lt;percentage&gt;</b></dt>
  <dd class="It-tag">Report only alignments scoring at least this percentage of
      the maximal score for each query. eg. use <b>--percent 90</b> to report
      alignments with 90% of the maximal score optainable for that query. This
      option is useful not only because it reduces the spurious matches in the
      output, but because it generates query-specific thresholds (unlike
      <b>--score</b> ) for a set of queries of differing lengths, and will also
      speed up the search considerably. <b>NB.</b> with this option, it is
      possible to have a cDNA match its corresponding gene exactly, yet still
      score less than 100%, due to the addition of the intron penalty scores,
      hence this option must be used with caution.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showalignment &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Show the alignments in an human readable form.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showsugar &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Display &quot;sugar&quot; output for ungapped alignments.
      Sugar is Simple UnGapped Alignment Report, which displays ungapped
      alignments one-per-line. The sugar line starts with the string
      &quot;sugar:&quot; for easy extraction from the output, and is followed by
      the the following 9 fields in the order below:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>query_id</b></dt>
  <dd class="It-tag">Query identifier</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>query_start</b></dt>
  <dd class="It-tag">Query position at alignment start</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>query_end</b></dt>
  <dd class="It-tag">Query position alignment end</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>query_strand</b></dt>
  <dd class="It-tag">Strand of query matched</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>target_id</b></dt>
  <dd class="It-tag">|</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>target_start</b></dt>
  <dd class="It-tag">| the same 4 fields</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>target_end</b></dt>
  <dd class="It-tag">| for the target sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>target_strand</b></dt>
  <dd class="It-tag">|</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>score</b></dt>
  <dd class="It-tag">The raw alignment score</dd>
</dl>
</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showcigar &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Show the alignments in &quot;cigar&quot; format. Cigar is a
      Compact Idiosyncratic Gapped Alignment Report, which displays gapped
      alignments one-per-line. The format starts with the same 9 fields as sugar
      output (see above), and is followed by a series of &lt;operation,
      length&gt; pairs where operation is one of match, insert or delete, and
      the length describes the number of times this operation is repeated.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showvulgar &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Shows the alignments in &quot;vulgar&quot; format. Vulgar
      is Verbose Useful Labelled Gapped Alignment Report, This format also
      starts with the same 9 fields as sugar output (see above), and is followed
      by a series of &lt;label, query_length, target_length&gt; triplets. The
      label may be one of the following:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>M</b></dt>
  <dd class="It-tag">Match</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>C</b></dt>
  <dd class="It-tag">Codon</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>G</b></dt>
  <dd class="It-tag">Gap</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>N</b></dt>
  <dd class="It-tag">Non-equivalenced region</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>5</b></dt>
  <dd class="It-tag">5' splice site</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>3</b></dt>
  <dd class="It-tag">3' splice site</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>I</b></dt>
  <dd class="It-tag">Intron</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>S</b></dt>
  <dd class="It-tag">Split codon</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>F</b></dt>
  <dd class="It-tag">Frameshift</dd>
</dl>
</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showquerygff &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Report GFF output for features on the query sequence. See
      http://www.sanger.ac.uk/Software/formats/GFF for more information.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--showtargetgff &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Report GFF output for features on the target sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ryo &lt;format&gt;</b></dt>
  <dd class="It-tag">Roll-your-own output format. This allows specification of a
      printf-esque format line which is used to specify which information to
      include in the output, and how it is to be shown. The format field may
      contain the following fields:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>%[qt][idlsSt]</b></dt>
  <dd class="It-tag">For either {query,target}, report the
      {id,definition,length,sequence,Strand,type} Sequences are reported in a
      fasta-format like block (no headers).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%[qt]a[bels]</b></dt>
  <dd class="It-tag">For either {query,target} region which occurs <b>in the
      alignment,</b> report the {begin,end,length,sequence}</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%[qt]c[bels]</b></dt>
  <dd class="It-tag">For either {query,target} region which occurs in the
      <b>coding sequence</b> in the alignment, report the
      {begin,end,length,sequence}</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%s</b></dt>
  <dd class="It-tag">The raw score</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%r</b></dt>
  <dd class="It-tag">The rank (in results from a bestn search)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%m</b></dt>
  <dd class="It-tag">Model name</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%e[tism]</b></dt>
  <dd class="It-tag">Equivalenced {total,id,similarity,mismatches} (ie. %em ==
      (%et - %ei))</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%p[isS]</b></dt>
  <dd class="It-tag">Percent {id,similarity,Self} over the equivalenced portions
      of the alignment. (ie. %pi == 100*(%ei / %et)). Percent Self is the score
      over the equivalenced portions of the alignment as a percentage of the
      self comparison score of the query sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%g</b></dt>
  <dd class="It-tag">Gene orientation ('+' = forward, '-' = reverse, '.' =
      unknown)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%S</b></dt>
  <dd class="It-tag">Sugar block (the 9 fields used in sugar output (see
    above)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%C</b></dt>
  <dd class="It-tag">Cigar block (the fields of a cigar line after the sugar
      portion)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%V</b></dt>
  <dd class="It-tag">Vulgar block (the fields of a vulgar line after the sugar
      portion)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%%</b></dt>
  <dd class="It-tag">Expands to a percentage sign (%)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>\n</b></dt>
  <dd class="It-tag">Newline</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>\t</b></dt>
  <dd class="It-tag">Tab</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>\\</b></dt>
  <dd class="It-tag">Expands to a backslash (\)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>\{</b></dt>
  <dd class="It-tag">Open curly brace</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>\}</b></dt>
  <dd class="It-tag">Close curly brace</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>{</b></dt>
  <dd class="It-tag">Begin per-transition output section</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>}</b></dt>
  <dd class="It-tag">End per-transition output section</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%P[qt][sabe]</b></dt>
  <dd class="It-tag">Per-transition output for {query,target}
      {sequence,advance,begin,end}</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>%P[nsl]</b></dt>
  <dd class="It-tag">Per-transition output for {name,score,label}</dd>
</dl>
</div>
<div class="Pp"></div>
This option is very useful and flexible. For example, to report all the sections
  of query sequences which feature in alignments in fasta format, use:
<div class="Pp"></div>
--ryo <b>&quot;&gt;%qi %qd\n%qas\n&quot;</b>
<div class="Pp"></div>
To output all the symbols and scores in an alignment, try something like:
<div class="Pp"></div>
--ryo <b>&quot;%V{%Pqs %Pts %Ps\n}&quot;</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n | --bestn &lt;number&gt;</b></dt>
  <dd class="It-tag">Report the best N results for each query. (Only results
      scoring better than the score threshold
    <br/>
     will be reported). The option reduces the amount of output generated, and
      also allows exonerate to speed up the search.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S | --subopt &lt;boolean&gt;</b></dt>
  <dd class="It-tag">This option allows for the reporting of (Waterman-Eggert
      style) suboptimal alignments. (It is on by default.) All suboptimal (ie.
      non-intersecting) alignments will be reported for each pair of sequences
      scoring at least the threshold provided by <b>--score.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    When this option is used with exhaustive alignments, several full quadratic
      time passes will be required, so the running time will be considerably
      increased.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g | --gappedextension &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Causes a gapped extension stage to be performed ie. dynamic
      programming is applied in arbitrarily shaped and dynamically sized regions
      surrounding HSP seeds. The extension threshold is controlled by the
      --extensionthreshold option.
    <div style="height: 1.00em;">&#x00A0;</div>
    Although sometimes slower than BSDP, gapped extension improves sensitivity
      with weak, gap-rich alignments such as during cross-species comparison.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>NB. This option is now the default. Set it to false</b> to reverse to the
      old BSDP type alignments. This option may be slower than BSDP for some
      large scale analyses with simple alignment models.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--refine &lt;strategy&gt;</b></dt>
  <dd class="It-tag">Force exonerate to refine alignments generated by
      heuristics using dynamic programming over larger regions. This takes more
      time, but improves the quality of the final alignments.
    <div style="height: 1.00em;">&#x00A0;</div>
    The strategies available for refinement are:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>none</b></dt>
  <dd class="It-tag">The default - no refinement is used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>full</b></dt>
  <dd class="It-tag">An exhaustive alignment is calculated from the pair of
      sequences in their entirety.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>region</b></dt>
  <dd class="It-tag">DP is applied just to the region of the sequences covered
      by the heuristic alignment.</dd>
</dl>
</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--refineboundary &lt;size&gt;</b></dt>
  <dd class="It-tag">Specify an extra boundary to be included in the region
      subject to alignment during refinement by region.</dd>
</dl>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="VITERBI_ALGORITHM_OPTIONS"><a class="selflink" href="#VITERBI_ALGORITHM_OPTIONS">VITERBI
  ALGORITHM OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D | --dpmemory &lt;Mb&gt;</b></dt>
  <dd class="It-tag">The exhaustive alignment traceback routines use a
      Hughey-style reduced memory technique. This option specifies how much
      memory will be used for this. Generally, the more memory is permitted
      here, the faster the alignments will be produced.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="CODE_GENERATION_OPTIONS"><a class="selflink" href="#CODE_GENERATION_OPTIONS">CODE
  GENERATION OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C | --compiled &lt;boolean&gt;</b></dt>
  <dd class="It-tag">This option allows disabling of generated code for dynamic
      programming. It is mainly used during development of exonerate. When set
      to FALSE, an &quot;interpreted&quot; version of the dynamic programming
      implementation is used, which is much slower.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="HEURISTIC_OPTIONS"><a class="selflink" href="#HEURISTIC_OPTIONS">HEURISTIC
  OPTIONS</a></h1>
<div class="Pp"></div>
<b>--terminalrangeint</b>
<div class="Pp"></div>
<b>--terminalrangeext</b>
<div class="Pp"></div>
<b>--joinrangeint</b>
<div class="Pp"></div>
<b>--joinrangeext</b>
<div class="Pp"></div>
<b>--spanrangeint</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--spanrangeext</b></dt>
  <dd class="It-tag">These options are used to specify the size of the
      sub-alignment regions to which DP is applied around the ends of the HSPs.
      This can be at the HSP ends (terminal range), between HSPs (join range),
      or between HSPs which may be connected by a large region such as an intron
      or non-equivalenced region (span range). These ranges can be specified for
      a number of matches back onto the HSP (internal range) or out from the HSP
      (external range).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEEDED_DYNAMIC_PROGRAMMING_OPTIONS"><a class="selflink" href="#SEEDED_DYNAMIC_PROGRAMMING_OPTIONS">SEEDED
  DYNAMIC PROGRAMMING OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-x | --extensionthreshold &lt;score&gt;</b></dt>
  <dd class="It-tag">This is the amount by which the score will be allowed to
      degrade during SDP. This is the equivalent of the hspdropoff penalties,
      except it is applied during dynamic programming, not HSP extension.
      Decreasing this parameter will increase the speed of the SDP, and
      increasing it will increase the sensitivity.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--singlepass &lt;boolean&gt;</b></dt>
  <dd class="It-tag">By default the suboptimal SDP alignments are reported by a
      singlepass algorithm, but may miss some suboptimal alignments that are
      close together. This option can be used to force the use of a multipass
      suboptimal alignment algorithm for SDP, resulting in higher quality
      suboptimal alignments.</dd>
</dl>
<h1 class="Sh" title="Sh" id="BSDP_OPTIONS"><a class="selflink" href="#BSDP_OPTIONS">BSDP
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--joinfilter &lt;limit&gt;</b></dt>
  <dd class="It-tag">(experimental)
    <div style="height: 1.00em;">&#x00A0;</div>
    Only allow consider this number of SARs for joining HSPs together. The SARs
      with the highest potential for appearing in a high-scoring alignment are
      considered. This option useful for limiting time and memory usage when
      searching unmasked data with repetitive sequences, but should not be set
      too low, as valid matches may be ignored. Something like <b>--joinfilter
      32</b> seems to work well.</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEQUENCE_OPTIONS"><a class="selflink" href="#SEQUENCE_OPTIONS">SEQUENCE
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--annotation &lt;path&gt;</b></dt>
  <dd class="It-tag">Specify basic sequence annotation information. This is most
      useful with the cdna2genome model, but will work with other models. The
      annotation file contains four fields per line:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">&lt;id&gt; &lt;strand&gt; &lt;cds_start&gt;
    &lt;cds_length&gt;</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Here is a simple example of such a file for 4 cDNAs:</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">dhh.human.cdna + 308 1191</dt>
  <dd class="It-tag"></dd>
</dl>
<div class="Pp"></div>
dhh.mouse.cdna + 250 1191
<div class="Pp"></div>
csn7a.human.cdna + 178 828
<div class="Pp"></div>
csn7a.mouse.cdna + 126 828
<div class="Pp"></div>
These annotation lines will also work when only the first two fields are used.
  This can be used when specifying which strand of a specific sequence should be
  included in a comparison.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYMBOL_COMPARISON_OPTIONS"><a class="selflink" href="#SYMBOL_COMPARISON_OPTIONS">SYMBOL
  COMPARISON OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--softmaskquery &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Indicate that the query is softmasked. See description
      below for <b>--softmasktarget</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--softmasktarget &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Indicate that the target is softmasked. In a softmasked
      sequence file, instead of masking regions by Ns or Xs they are masked by
      putting those regions in lower case (and with unmasked regions in upper
      case). This option allows the masking to be ignored by some parts of the
      program, combining the speed of searching masked data with sensitivity of
      searching unmasked data. The utility <b>fastasoftmask</b> supplied which
      is supplied with exonerate can be used for producing softmasked sequence
      from conventionally masked sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d | --dnasubmat &lt;name&gt;</b></dt>
  <dd class="It-tag">Specify the the substitution matrix to be used for DNA
      comparison. This should be a path to a substitution matrix in same format
      as that which is used by blast.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p | --proteinsubmat &lt;name&gt;</b></dt>
  <dd class="It-tag">Specify the the substitution matrix to be used for protein
      comparison. (Both DNA and protein substitution matrices are required for
      some types of analysis). The use of the special names, <b>nucleic,
      blosum62, pam250, edit</b> or <b>identity</b> will cause built-in
      substitution matrices to be used.</dd>
</dl>
<h1 class="Sh" title="Sh" id="ALIGNMENT_SEEDING_OPTIONS"><a class="selflink" href="#ALIGNMENT_SEEDING_OPTIONS">ALIGNMENT
  SEEDING OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M | --fsmmemory &lt;Mb&gt;</b></dt>
  <dd class="It-tag">Specify the amount of memory to use for the FSM in
      heuristic analyses. exonerate multiplexes the query to accelerate
      large-throughput database queries. This figure should always be less than
      the physical memory on the machine, but when searching large databases,
      generally, the more memory it is allowed to use, the faster it will
    go.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--forcefsm &lt;none | normal | compact&gt;</b></dt>
  <dd class="It-tag">Force the use of more compact finite state machines for
      analyses involving big sequences and large word neighbourhoods. By
      default, exonerate will pick a sensible strategy, so this option will
      rarely need to be set.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wordjump &lt;int&gt;</b></dt>
  <dd class="It-tag">The jump between query words used to yield the word
      neighbourhood. If set to 1, every word is used, if set to 2, every other
      word is used, and if set to the wordlength, only non-overlapping words
      will be used. This option reduces the memory requirements when using very
      large query sequences, and makes the search run faster, but it also
      damages search sensitivity when high values are set.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wordambiguity &lt;limit&gt;</b></dt>
  <dd class="It-tag">This option may be used to allow alignment seeds containing
      IUPAC ambiguity symbols. The limit is the maximum number of ambiguous
      words allowed at a single position. If this limit is reached then the
      position is not used for alignment seeding. Using this option may slow
      down a search. For large datasets, it is recommended to use <b>esd2esi
      --wordambiguity</b> instead, as then the speed overhead is only incurred
      during indexing, rather than during the database searching itself. NB.
      This option only works for IUPAC symbols in the target sequence. Query
      words containing IUPAC symbols are (currently) excluded from seeding.</dd>
</dl>
<h1 class="Sh" title="Sh" id="AFFINE_MODEL_OPTIONS"><a class="selflink" href="#AFFINE_MODEL_OPTIONS">AFFINE
  MODEL OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o | --gapopen &lt;penalty&gt;</b></dt>
  <dd class="It-tag">This is the gap open penalty.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e | --gapextend &lt;penalty&gt;</b></dt>
  <dd class="It-tag">This is the gap extension penalty.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--codongapopen &lt;penalty&gt;</b></dt>
  <dd class="It-tag">This is the codon gap open penalty.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--codongapextend &lt;penalty&gt;</b></dt>
  <dd class="It-tag">This is the codon gap extension penalty.</dd>
</dl>
<h1 class="Sh" title="Sh" id="NER_OPTIONS"><a class="selflink" href="#NER_OPTIONS">NER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minner &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Minimum NER length allowed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxner &lt;length&gt;</b></dt>
  <dd class="It-tag">Maximum NER length allowed. NB. this option only affects
      heuristic alignments.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--neropen &lt;penalty&gt;</b></dt>
  <dd class="It-tag">Penalty for opening a non-equivalenced region.</dd>
</dl>
<h1 class="Sh" title="Sh" id="INTRON_MODELLING_OPTIONS"><a class="selflink" href="#INTRON_MODELLING_OPTIONS">INTRON
  MODELLING OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--minintron &lt;length&gt;</b></dt>
  <dd class="It-tag">Minimum intron length limit. NB. this option only affects
      heuristic alignments. This is not a hard limit - it only affects size of
      introns which are sought during heuristic alignment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxintron &lt;length&gt;</b></dt>
  <dd class="It-tag">Maximum intron length limit. See notes above for
      <b>--minintron</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i | --intronpenalty &lt;penalty&gt;</b></dt>
  <dd class="It-tag">Penalty for introduction of an intron.</dd>
</dl>
<h1 class="Sh" title="Sh" id="FRAMESHIFT_MODELLING_OPTIONS"><a class="selflink" href="#FRAMESHIFT_MODELLING_OPTIONS">FRAMESHIFT
  MODELLING OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f | --frameshift &lt;penalty&gt;</b></dt>
  <dd class="It-tag">The penalty for the inclusion of a frameshift in an
      alignment.</dd>
</dl>
<h1 class="Sh" title="Sh" id="ALPHABET_OPTIONS"><a class="selflink" href="#ALPHABET_OPTIONS">ALPHABET
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--useaatla &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Use three-letter abbreviations for AA names. ie. when
      displaying alignment &quot;Met&quot; is used instead of &quot; M
    &quot;</dd>
</dl>
<h1 class="Sh" title="Sh" id="TRANSLATION_OPTIONS"><a class="selflink" href="#TRANSLATION_OPTIONS">TRANSLATION
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--geneticcode &lt;code&gt;</b></dt>
  <dd class="It-tag">Specify an alternative genetic code. The default code (1)
      is the standard genetic code. Other genetic codes may be specified by in
      shorthand or longhand form.
    <div style="height: 1.00em;">&#x00A0;</div>
    In shorthand form, a number between 1 and 23 is used to specify one of 17
      built-in genetic code variants. These are genetic code variants taken
      from:
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    These are:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>1</b></dt>
  <dd class="It-tag">The Standard Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>2</b></dt>
  <dd class="It-tag">The Vertebrate Mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>3</b></dt>
  <dd class="It-tag">The Yeast Mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>4</b></dt>
  <dd class="It-tag">The Mold, Protozoan, and Coelenterate Mitochondrial Code
      and the Mycoplasma/Spiroplasma Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>5</b></dt>
  <dd class="It-tag">The Invertebrate Mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>6</b></dt>
  <dd class="It-tag">The Ciliate, Dasycladacean and Hexamita Nuclear Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>9</b></dt>
  <dd class="It-tag">The Echinoderm and Flatworm Mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>10</b></dt>
  <dd class="It-tag">The Euplotid Nuclear Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>11</b></dt>
  <dd class="It-tag">The Bacterial and Plant Plastid Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>12</b></dt>
  <dd class="It-tag">The Alternative Yeast Nuclear Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>13</b></dt>
  <dd class="It-tag">The Ascidian Mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>14</b></dt>
  <dd class="It-tag">The Alternative Flatworm Mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>15</b></dt>
  <dd class="It-tag">Blepharisma Nuclear Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>16</b></dt>
  <dd class="It-tag">Chlorophycean Mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>21</b></dt>
  <dd class="It-tag">Trematode Mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>22</b></dt>
  <dd class="It-tag">Scenedesmus obliquus mitochondrial Code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>23</b></dt>
  <dd class="It-tag">Thraustochytrium Mitochondrial Code&quot;,</dd>
</dl>
<div class="Pp"></div>
In longhand form, a genetic code variant may be provided as a 64 byte string in
  TCAG order, eg. the standard genetic code in this form would be:
<div style="height: 1.00em;">&#x00A0;</div>
FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<h1 class="Sh" title="Sh" id="HSP_CREATION_OPTIONS"><a class="selflink" href="#HSP_CREATION_OPTIONS">HSP
  CREATION OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hspfilter &lt;threshold&gt;</b></dt>
  <dd class="It-tag">Use aggressive HSP filtering to speed up heuristic
      searches. The threshold specifies the number of HSPs centred about a point
      in the query which will be stored. Any lower scoring HSPs will be
      discarded. This is an experimental option to handle speed problems caused
      by some sequences. A value of about 100 seems to work well.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--useworddropoff &lt;boolean&gt;</b></dt>
  <dd class="It-tag">When this is TRUE, the score threshold for admitting words
      into the word neighbourhood is set to be the initial word score minus the
      word threshold (see below). This strategy is designed to prevent
      restricting the word
      SSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG When this is
      FALSE, the word threshold is taken to be an absolute value.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seedrepeat &lt;count&gt;</b></dt>
  <dd class="It-tag">The seedrepeat parameter sets the number of seeds which
      must be found on the same diagonal or reading frame before HSP extension
      will occur. Increasing the value for <b>--seedrepeat</b> will speed up
      searches, and is usually a better option than using longer word lengths,
      particularly when using the <b>exonerate-server</b> where increasing word
      lengths requires recomputing the index, and greater increases memory
      requirements.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w --dnawordlen &lt;bases&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-W --proteinwordlen &lt;residues&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-W --codonnwordlen &lt;bases&gt;</b></dt>
  <dd class="It-tag">The word length used for DNA, protein or codon words. When
      performing DNA vs protein comparisons, a the DNA wordlength will always
      (automatically) be triple the protein wordlength.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dnahspdropoff &lt;score&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--proteinhspdropoff &lt;score&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--codonhspdropoff &lt;score&gt;</b></dt>
  <dd class="It-tag">The amount by which an HSP score will be allowed to degrade
      during HSP extension. Separate threshold can be set for dna or protein
      comparisons.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dnahspthreshold &lt;score&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--proteinhspthreshold &lt;score&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--codonhspthreshold &lt;score&gt;</b></dt>
  <dd class="It-tag">The HSP score thresholds. An HSP must score at least this
      much before it will be reported or be used in preparation of a heuristic
      alignment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dnawordlimit &lt;score&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--proteinwordlimit &lt;score&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--codonwordlimit &lt;score&gt;</b></dt>
  <dd class="It-tag">The threshold for admitting DNA or protein words into the
      word neighbourhood. The behaviour of this option is altered by the
      <b>--useworddropoff</b> option (see above).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--geneseed &lt;threshold&gt;</b></dt>
  <dd class="It-tag">Exclude HSPs from gapped alignment computation which cannot
      feature in a alignment containing at least one HSP scoring at least this
      threshold.
    <div style="height: 1.00em;">&#x00A0;</div>
    This option provides considerable speed up for gapped alignment computation,
      but may cause some very gap-rich alignments to be missed.
    <div style="height: 1.00em;">&#x00A0;</div>
    It is useful when aligning similar sequences back onto genome quickly, eg.
      try --geneseed 250</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--geneseedrepeat &lt;count&gt;</b></dt>
  <dd class="It-tag">The geneseedrepeat parameter is like the seedrepeat
      parameter, but is only applied when looking for the geneseed hsps. Using a
      larger value for <b>--geneseedrepeat</b> will speed up searches when the
      <b>--geneseed</b> parameter is also used. (experimental, implementation
      incomplete)</dd>
</dl>
<h1 class="Sh" title="Sh" id="ALIGNMENT_OPTIONS"><a class="selflink" href="#ALIGNMENT_OPTIONS">ALIGNMENT
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--alignmentwidth &lt;width&gt;</b></dt>
  <dd class="It-tag">Width of alignment display. The default is 80.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--forwardcoordinates &lt;boolean&gt;</b></dt>
  <dd class="It-tag">By default, all coordinates are reported on the forward
      strand. Setting this option to false reverts to the old behaviour
      (pre-0.8.3) whereby alignments on the reverse complement of a sequence are
      reported using coordinates on the reverse complement.</dd>
</dl>
<h1 class="Sh" title="Sh" id="SUB-ALIGNMENT_REGION_OPTIONS"><a class="selflink" href="#SUB-ALIGNMENT_REGION_OPTIONS">SUB-ALIGNMENT
  REGION OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quality &lt;percent&gt;</b></dt>
  <dd class="It-tag">This option excludes HSPs from BSDP when their components
      outside of the SARs fall below this quality threshold.</dd>
</dl>
<h1 class="Sh" title="Sh" id="SPLICE_SITE_PREDICTION_OPTIONS"><a class="selflink" href="#SPLICE_SITE_PREDICTION_OPTIONS">SPLICE
  SITE PREDICTION OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--splice3 &lt;path&gt;</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--splice5 &lt;path&gt;</b></dt>
  <dd class="It-tag">Provide a file containing a custom PSSM (position specific
      score matrix) for prediction of the intron splice sites.
    <div style="height: 1.00em;">&#x00A0;</div>
    The file format for splice data is simple: lines beginning with
      &#x00B4;#&#x00B4; are comments, a line containing just the word
      &#x00B4;splice&#x00B4; denotes the position of the splice site, and the
      other lines show the observed relative frequencies of the bases flanking
      the splice sites in the chosen organism (in ACGT order).
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Example 5' splice data file:</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
<div class="Pp"></div>
 # start of example 5' splice data
 # A C G T
 28 40  17  14
 59 14  13  14
  8  5  81   6
 splice
  0  0 100   0
  0  0   0 100
 54  2  42   2
 74  8  11   8
  5  6  85   4
 16 18  21  45
 # end of test 5' splice data
<div style="height: 1.00em;">&#x00A0;</div>
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Example 3' splice data file:</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
<div class="Pp"></div>
 # start of example 3' splice data
 # A C G T
  10  31  14  44
   8  36  14  43
   6  34  12  48
   6  34   8  52
   9  37   9  45
   9  38  10  44
   8  44   9  40
   9  41   8  41
   6  44   6  45
   6  40   6  48
  23  28  26  23
   2  79   1  18
 100   0   0   0
   0   0 100   0
 splice
  28  14  47  11
 # end of example 3' splice data
<div style="height: 1.00em;">&#x00A0;</div>
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--forcegtag &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Only allow splice sites at gt....ag sites (or ct....ac
      sites when the gene is reversed) With this restriction in place, the
      splice site prediction scores are still used and allow tie breaking when
      there is more than one possible splice site.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="STRATEGIES_FOR_SPEED"><a class="selflink" href="#STRATEGIES_FOR_SPEED">STRATEGIES
  FOR SPEED</a></h1>
Keep all data on local disks.
<div style="height: 1.00em;">&#x00A0;</div>
Apply the highest acceptable score thresholds using a combination of --score,
  --percent and --bestn.
<div style="height: 1.00em;">&#x00A0;</div>
Repeat mask and dust the genomic (target) sequence. (Softmask these sequences
  and use --softmasktarget).
<div style="height: 1.00em;">&#x00A0;</div>
Increase the --fsmmemory option to allow more query multiplexing.
<div style="height: 1.00em;">&#x00A0;</div>
Increase the value for --seedrepeat
<div style="height: 1.00em;">&#x00A0;</div>
When using an alignment model containing introns, set --geneseed as high as
  possible.
<div style="height: 1.00em;">&#x00A0;</div>
If you are compiling exonerate yourself, see the README file supplied with the
  source code for details of compile-time optimisations.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="STRATEGIES_FOR_SENSITIVITY"><a class="selflink" href="#STRATEGIES_FOR_SENSITIVITY">STRATEGIES
  FOR SENSITIVITY</a></h1>
Not documented yet.
<div style="height: 1.00em;">&#x00A0;</div>
Increase the word neighbourhood. Decrease the HSP threshold. Increase the SAR
  ranges. Run exhaustively.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="ENVIRONMENT"><a class="selflink" href="#ENVIRONMENT">ENVIRONMENT</a></h1>
Not documented yet.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<b>exonerate cdna.fasta genomic.fasta</b>
<div style="margin-left: 5.00ex;">This simplest way in which exonerate may be
  used. By default, an ungapped alignment model will be used.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>exonerate --exhaustive y --model est2genome cdna.fasta
  genomic.masked.fasta</b>
<div style="margin-left: 5.00ex;">Exhaustively align cdnas to genomic sequence.
  This will be much, much slower, but more accurate. This option causes
  exonerate to behave like EST_GENOME.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>exonerate --exhaustive --model affine:local</b> <b>query.fasta
  target.fasta</b>
<div style="margin-left: 5.00ex;">If the affine:local model is used with
  exhaustive alignment, you have the Smith-Waterman algorithm.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>exonerate --exhaustive --model affine:global</b> <b>protein.fasta
  protein.fasta</b>
<div style="margin-left: 5.00ex;">Switch to a global model, and you have
  Needleman-Wunsch.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>exonerate --wordthreshold 1 --gapped no --showhsp yes protein.fasta
  genome.fasta</b>
<div style="margin-left: 5.00ex;">Generate ungapped Protein:DNA alignments</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>exonerate --model coding2coding --score 1000 --bigseq yes
  --proteinhspthreshold 90 chr21.fa chr22.fa</b>
<div style="margin-left: 5.00ex;">Perform quick-and-dirty translated pairwise
  alignment of two very large DNA sequences.</div>
<div style="height: 1.00em;">&#x00A0;</div>
Many similar combinations should work. Try them out.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This documentation accompanies version 2.2.0 of the exonerate package.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Guy St.C. Slater. &lt;guy@ebi.ac.uk&gt;.
<div class="Pp"></div>
See the AUTHORS file accompanying the source code for a list of contributors.
<h1 class="Sh" title="Sh" id="AVAILABILITY"><a class="selflink" href="#AVAILABILITY">AVAILABILITY</a></h1>
This source code for the exonerate package is available under the terms of the
  GNU <i>general</i> public licence.
<div style="height: 1.00em;">&#x00A0;</div>
Please see the file COPYING which was distrubuted with this package, or
  http://www.gnu.org/licenses/gpl.txt for details.
<div style="height: 1.00em;">&#x00A0;</div>
This package has been developed as part of the ensembl project. Please see
  http://www.ensembl.org/ for more information.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>exonerate-server</b>(1), <b>ipcress</b>(1), <b>blast</b>(1L).</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2002</td>
    <td class="foot-os">exonerate</td>
  </tr>
</table>
</body>
</html>
