<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>EXTRACTFEAT(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">EXTRACTFEAT(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">EXTRACTFEAT(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
extractfeat - Extract features from sequence(s)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 12.00ex; text-indent: -12.00ex;"><b>extractfeat</b>
  <b>-sequence&#x00A0;</b><b></b><i>seqall</i>
  [<b>-before&#x00A0;</b><b></b><i>integer</i>] [
  <b>-after&#x00A0;</b><b></b><i>integer</i>]
  [<b>-source&#x00A0;</b><b></b><i>string</i>] [
  <b>-type&#x00A0;</b><b></b><i>string</i>] [
  <b>-sense&#x00A0;</b><b></b><i>integer</i>] [
  <b>-minscore&#x00A0;</b><b></b><i>float</i>]
  [<b>-maxscore&#x00A0;</b><b></b><i>float</i>]
  [<b>-tag&#x00A0;</b><b></b><i>string</i>] [
  <b>-value&#x00A0;</b><b></b><i>string</i>] [
  <b>-join&#x00A0;</b><b></b><i>boolean</i>] [
  <b>-featinname&#x00A0;</b><b></b><i>boolean</i>]
  [<b>-describe&#x00A0;</b><b></b> <i>string</i>]
  <b>-outseq&#x00A0;</b><b></b><i>seqout</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 12.00ex; text-indent: -12.00ex;"><b>extractfeat</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>extractfeat</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Edit,Feature tables&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-sequence</b> <i>seqall</i>
<div style="margin-left: 4.00ex;"></div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-before</b> <i>integer</i>
<div style="margin-left: 4.00ex;">If this value is greater than 0 then that
  number of bases or residues before the feature are included in the extracted
  sequence. This allows you to get the context of the feature. If this value is
  negative then the start of the extracted sequence will be this number of
  bases/residues before the end of the feature. So a value of '10' will start
  the extraction 10 bases/residues before the start of the sequence, and a value
  of '-10' will start the extraction 10 bases/residues before the end of the
  feature. The output sequence will be padded with 'N' or 'X' characters if the
  sequence starts after the required start of the extraction.</div>
<div class="Pp"></div>
<b>-after</b> <i>integer</i>
<div style="margin-left: 4.00ex;">If this value is greater than 0 then that
  number of bases or residues after the feature are included in the extracted
  sequence. This allows you to get the context of the feature. If this value is
  negative then the end of the extracted sequence will be this number of
  bases/residues after the start of the feature. So a value of '10' will end the
  extraction 10 bases/residues after the end of the sequence, and a value of
  '-10' will end the extraction 10 bases/residues after the start of the
  feature. The output sequence will be padded with 'N' or 'X' characters if the
  sequence ends before the required end of the extraction.</div>
<div class="Pp"></div>
<b>-source</b> <i>string</i>
<div style="margin-left: 4.00ex;">By default any feature source in the feature
  table is shown. You can set this to match any feature source you wish to show.
  The source name is usually either the name of the program that detected the
  feature or it is the feature table (eg: EMBL) that the feature came from. The
  source may be wildcarded by using '*'. If you wish to show more than one
  source, separate their names with the character '|', eg: gene* | embl Default
  value: *</div>
<div class="Pp"></div>
<b>-type</b> <i>string</i>
<div style="margin-left: 4.00ex;">By default every feature in the feature table
  is extracted. You can set this to be any feature type you wish to extract. See
  http://www.ebi.ac.uk/embl/WebFeat/ for a list of the EMBL feature types and
  see the Uniprot user manual in
  http://www.uniprot.org/manual/sequence_annotation for a list of the Uniprot
  feature types. The type may be wildcarded by using '*'. If you wish to extract
  more than one type, separate their names with the character '|', eg: *UTR |
  intron Default value: *</div>
<div class="Pp"></div>
<b>-sense</b> <i>integer</i>
<div style="margin-left: 4.00ex;">By default any feature type in the feature
  table is extracted. You can set this to match any feature sense you wish. 0 -
  any sense, 1 - forward sense, -1 - reverse sense</div>
<div class="Pp"></div>
<b>-minscore</b> <i>float</i>
<div style="margin-left: 4.00ex;">Minimum score of feature to extract (see also
  maxscore) Default value: 0.0</div>
<div class="Pp"></div>
<b>-maxscore</b> <i>float</i>
<div style="margin-left: 4.00ex;">Maximum score of feature to extract. If both
  minscore and maxscore are zero (the default), then any score is ignored
  Default value: 0.0</div>
<div class="Pp"></div>
<b>-tag</b> <i>string</i>
<div style="margin-left: 4.00ex;">Tags are the types of extra values that a
  feature may have. For example in the EMBL feature table, a 'CDS' type of
  feature may have the tags '/codon', '/codon_start', '/db_xref', '/EC_number',
  '/evidence', '/exception', '/function', '/gene', '/label', '/map', '/note',
  '/number', '/partial', '/product', '/protein_id', '/pseudo', '/standard_name',
  '/translation', '/transl_except', '/transl_table', or '/usedin'. Some of these
  tags also have values, for example '/gene' can have the value of the gene
  name. By default any feature tag in the feature table is extracted. You can
  set this to match any feature tag you wish to show. The tag may be wildcarded
  by using '*'. If you wish to extract more than one tag, separate their names
  with the character '|', eg: gene | label Default value: *</div>
<div class="Pp"></div>
<b>-value</b> <i>string</i>
<div style="margin-left: 4.00ex;">Tag values are the values associated with a
  feature tag. Tags are the types of extra values that a feature may have. For
  example in the EMBL feature table, a 'CDS' type of feature may have the tags
  '/codon', '/codon_start', '/db_xref', '/EC_number', '/evidence', '/exception',
  '/function', '/gene', '/label', '/map', '/note', '/number', '/partial',
  '/product', '/protein_id', '/pseudo', '/standard_name', '/translation',
  '/transl_except', '/transl_table', or '/usedin'. Only some of these tags can
  have values, for example '/gene' can have the value of the gene name. By
  default any feature tag value in the feature table is shown. You can set this
  to match any feature tag value you wish to show. The tag value may be
  wildcarded by using '*'. If you wish to show more than one tag value, separate
  their names with a space or the character '|', eg: pax* | 10 Default value:
  *</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-join</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Some features, such as CDS (coding sequence)
  and mRNA are composed of introns concatenated together. There may be other
  forms of 'joined' sequence, depending on the feature table. If this option is
  set TRUE, then any group of these features will be output as a single
  sequence. If the 'before' and 'after' qualifiers have been set, then only the
  sequence before the first feature and after the last feature are added.
  Default value: N</div>
<div class="Pp"></div>
<b>-featinname</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">To aid you in identifying the type of feature
  that has been output, the type of feature is added to the start of the
  description of the output sequence. Sometimes the description of a sequence is
  lost in subsequent processing of the sequences file, so it is useful for the
  type to be a part of the sequence ID name. If you set this to be TRUE then the
  name is added to the ID name of the output sequence. Default value: N</div>
<div class="Pp"></div>
<b>-describe</b> <i>string</i>
<div style="margin-left: 4.00ex;">To aid you in identifying some further
  properties of a feature that has been output, this lets you specify one or
  more tag names that should be added to the output sequence Description text,
  together with their values (if any). For example, if this is set to be 'gene',
  then if any output feature has the tag (for example) '/gene=BRCA1' associated
  with it, then the text '(gene=BRCA1)' will be added to the Description line.
  Tags are the types of extra values that a feature may have. For example in the
  EMBL feature table, a 'CDS' type of feature may have the tags '/codon',
  '/codon_start', '/db_xref', '/EC_number', '/evidence', '/exception',
  '/function', '/gene', '/label', '/map', '/note', '/number', '/partial',
  '/product', '/protein_id', '/pseudo', '/standard_name', '/translation',
  '/transl_except', '/transl_table', or '/usedin'. Some of these tags also have
  values, for example '/gene' can have the value of the gene name. By default no
  feature tag is displayed. You can set this to match any feature tag you wish
  to show. The tag may be wildcarded by using '*'. If you wish to extract more
  than one tag, separate their names with the character '|', eg: gene |
  label</div>
<div class="Pp"></div>
<b>-outseq</b> <i>seqout</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
extractfeat is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
