<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:44 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>EXTRACTFEAT(1e) EMBOSS Manual for Debian
EXTRACTFEAT(1e)</p>

<p style="margin-top: 1em">NAME <br>
extractfeat - Extract features from sequence(s)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
extractfeat -sequence seqall [-before integer] [-after
integer] [-source string] [-type string] [-sense integer]
[-minscore float] [-maxscore float] [-tag string] [-value
string] <br>
[-join boolean] [-featinname boolean] [-describe string]
-outseq seqout</p>

<p style="margin-top: 1em">extractfeat -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
extractfeat is a command line program from EMBOSS
(&acirc;the European Molecular Biology Open Software
Suite&acirc;). It is part of the &quot;Edit,Feature
tables&quot; command group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall</p>

<p style="margin-top: 1em">Additional section <br>
-before integer <br>
If this value is greater than 0 then that number of bases or
residues before the feature are included in the extracted
sequence. This allows you to get the context of the <br>
feature. If this value is negative then the start of the
extracted sequence will be this number of bases/residues
before the end of the feature. So a value of
&rsquo;10&rsquo; will start <br>
the extraction 10 bases/residues before the start of the
sequence, and a value of &rsquo;-10&rsquo; will start the
extraction 10 bases/residues before the end of the feature.
The output <br>
sequence will be padded with &rsquo;N&rsquo; or
&rsquo;X&rsquo; characters if the sequence starts after the
required start of the extraction.</p>

<p style="margin-top: 1em">-after integer <br>
If this value is greater than 0 then that number of bases or
residues after the feature are included in the extracted
sequence. This allows you to get the context of the <br>
feature. If this value is negative then the end of the
extracted sequence will be this number of bases/residues
after the start of the feature. So a value of
&rsquo;10&rsquo; will end <br>
the extraction 10 bases/residues after the end of the
sequence, and a value of &rsquo;-10&rsquo; will end the
extraction 10 bases/residues after the start of the feature.
The output <br>
sequence will be padded with &rsquo;N&rsquo; or
&rsquo;X&rsquo; characters if the sequence ends before the
required end of the extraction.</p>

<p style="margin-top: 1em">-source string <br>
By default any feature source in the feature table is shown.
You can set this to match any feature source you wish to
show. The source name is usually either the name of the <br>
program that detected the feature or it is the feature table
(eg: EMBL) that the feature came from. The source may be
wildcarded by using &rsquo;*&rsquo;. If you wish to show
more than <br>
one source, separate their names with the character
&rsquo;|&rsquo;, eg: gene* | embl Default value: *</p>

<p style="margin-top: 1em">-type string <br>
By default every feature in the feature table is extracted.
You can set this to be any feature type you wish to extract.
See http://www.ebi.ac.uk/embl/WebFeat/ for a list of <br>
the EMBL feature types and see the Uniprot user manual in
http://www.uniprot.org/manual/sequence_annotation for a list
of the Uniprot feature types. The type may be <br>
wildcarded by using &rsquo;*&rsquo;. If you wish to extract
more than one type, separate their names with the character
&rsquo;|&rsquo;, eg: *UTR | intron Default value: *</p>

<p style="margin-top: 1em">-sense integer <br>
By default any feature type in the feature table is
extracted. You can set this to match any feature sense you
wish. 0 - any sense, 1 - forward sense, -1 - reverse
sense</p>

<p style="margin-top: 1em">-minscore float <br>
Minimum score of feature to extract (see also maxscore)
Default value: 0.0</p>

<p style="margin-top: 1em">-maxscore float <br>
Maximum score of feature to extract. If both minscore and
maxscore are zero (the default), then any score is ignored
Default value: 0.0</p>

<p style="margin-top: 1em">-tag string <br>
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, <br>
&rsquo;/db_xref&rsquo;, &rsquo;/EC_number&rsquo;,
&rsquo;/evidence&rsquo;, &rsquo;/exception&rsquo;,
&rsquo;/function&rsquo;, &rsquo;/gene&rsquo;,
&rsquo;/label&rsquo;, &rsquo;/map&rsquo;,
&rsquo;/note&rsquo;, &rsquo;/number&rsquo;,
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;, <br>
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Some of these tags also have
values, for example &rsquo;/gene&rsquo; can have the value
of the gene <br>
name. By default any feature tag in the feature table is
extracted. You can set this to match any feature tag you
wish to show. The tag may be wildcarded by using
&rsquo;*&rsquo;. If you <br>
wish to extract more than one tag, separate their names with
the character &rsquo;|&rsquo;, eg: gene | label Default
value: *</p>

<p style="margin-top: 1em">-value string <br>
Tag values are the values associated with a feature tag.
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of <br>
feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, &rsquo;/db_xref&rsquo;,
&rsquo;/EC_number&rsquo;, &rsquo;/evidence&rsquo;,
&rsquo;/exception&rsquo;, &rsquo;/function&rsquo;,
&rsquo;/gene&rsquo;, &rsquo;/label&rsquo;,
&rsquo;/map&rsquo;, &rsquo;/note&rsquo;,
&rsquo;/number&rsquo;, <br>
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;,
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Only some of these tags can have
values, <br>
for example &rsquo;/gene&rsquo; can have the value of the
gene name. By default any feature tag value in the feature
table is shown. You can set this to match any feature tag
value you <br>
wish to show. The tag value may be wildcarded by using
&rsquo;*&rsquo;. If you wish to show more than one tag
value, separate their names with a space or the character
&rsquo;|&rsquo;, eg: pax* | 10 <br>
Default value: *</p>

<p style="margin-top: 1em">Output section <br>
-join boolean <br>
Some features, such as CDS (coding sequence) and mRNA are
composed of introns concatenated together. There may be
other forms of &rsquo;joined&rsquo; sequence, depending on
the feature <br>
table. If this option is set TRUE, then any group of these
features will be output as a single sequence. If the
&rsquo;before&rsquo; and &rsquo;after&rsquo; qualifiers have
been set, then only the <br>
sequence before the first feature and after the last feature
are added. Default value: N</p>

<p style="margin-top: 1em">-featinname boolean <br>
To aid you in identifying the type of feature that has been
output, the type of feature is added to the start of the
description of the output sequence. Sometimes the <br>
description of a sequence is lost in subsequent processing
of the sequences file, so it is useful for the type to be a
part of the sequence ID name. If you set this to be <br>
TRUE then the name is added to the ID name of the output
sequence. Default value: N</p>

<p style="margin-top: 1em">-describe string <br>
To aid you in identifying some further properties of a
feature that has been output, this lets you specify one or
more tag names that should be added to the output sequence
<br>
Description text, together with their values (if any). For
example, if this is set to be &rsquo;gene&rsquo;, then if
any output feature has the tag (for example)
&rsquo;/gene=BRCA1&rsquo; associated <br>
with it, then the text &rsquo;(gene=BRCA1)&rsquo; will be
added to the Description line. Tags are the types of extra
values that a feature may have. For example in the EMBL
feature <br>
table, a &rsquo;CDS&rsquo; type of feature may have the tags
&rsquo;/codon&rsquo;, &rsquo;/codon_start&rsquo;,
&rsquo;/db_xref&rsquo;, &rsquo;/EC_number&rsquo;,
&rsquo;/evidence&rsquo;, &rsquo;/exception&rsquo;,
&rsquo;/function&rsquo;, &rsquo;/gene&rsquo;,
&rsquo;/label&rsquo;, &rsquo;/map&rsquo;, <br>
&rsquo;/note&rsquo;, &rsquo;/number&rsquo;,
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;,
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Some of these tags <br>
also have values, for example &rsquo;/gene&rsquo; can have
the value of the gene name. By default no feature tag is
displayed. You can set this to match any feature tag you
wish to show. <br>
The tag may be wildcarded by using &rsquo;*&rsquo;. If you
wish to extract more than one tag, separate their names with
the character &rsquo;|&rsquo;, eg: gene | label</p>

<p style="margin-top: 1em">-outseq seqout</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
extractfeat is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
EXTRACTFEAT(1e)</p>
<hr>
</body>
</html>
