<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p style="margin-top: 1em">FA2DNA(1) User Manuals
FA2DNA(1)</p>

<p style="margin-top: 1em">NAME <br>
fa2dna - format fasta database for use with ANFO</p>

<p style="margin-top: 1em">SYNOPSIS <br>
fa2dna [ option | file ... ]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
fa2dna reads one or more files in fasta format and reformats
them into a database suitable for anfo(1). The input file(s)
can contain more than one sequence, and each sequence <br>
can be broken up into multiple contigs by a stretch of Ns.
All IUPAC ambiguity codes are fully supported.</p>

<p style="margin-top: 1em">OPTIONS <br>
-V, --version <br>
Print version number and exit.</p>

<p style="margin-top: 1em">-o file, --output file <br>
Write output to file.file should end in .dna, because anfo
and some downstream tools might stumble over a different
extension. Default is the genome name with the .dna <br>
extension.</p>

<p style="margin-top: 1em">-m N, --maxn N <br>
Sets the maximum number of Ns to be interpreted as IUPAC
ambiguity codes to N; any longer stretch is interpreted as
separing contigs. Default is 2.</p>

<p style="margin-top: 1em">-g name, --genome name <br>
Sets the genome name to name. This name is stored in the
output file and will be used by anfo to identify the matched
genome in its output. The genome should be a prefix <br>
of the output file name to allow downstream tools to find
it. The name should be short, but unique, something like
&quot;hg18&quot; or &quot;pt2&quot; for human or chimpanzee
genomes would <br>
work fine.</p>

<p style="margin-top: 1em">-d text, --description text <br>
Adds text as description to the metadata. This is purely
informational.</p>

<p style="margin-top: 1em">-v, --verbose <br>
Causes a progress indicator to be printed.</p>

<p style="margin-top: 1em">FILES <br>
/etc/popt <br>
The system wide configuration file for popt(3). fa2dna
identifies itself as &quot;fa2dna&quot; to popt.</p>

<p style="margin-top: 1em">~/.popt <br>
Per user configuration file for popt(3).</p>

<p style="margin-top: 1em">BUGS <br>
None known.</p>

<p style="margin-top: 1em">AUTHOR <br>
Udo Stenzel &lt;udo_stenzel@eva.mpg.de&gt;</p>

<p style="margin-top: 1em">SEE ALSO <br>
anfo(1), fa2dna(1) popt(3), fasta(5)</p>

<p style="margin-top: 1em">Applications OCTOBER 2009
FA2DNA(1)</p>
<hr>
</body>
</html>
