<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:09:00 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FastQTL(1) General Commands Manual FastQTL(1)</p>

<p style="margin-top: 1em">NAME <br>
FastQTL - Quantitative Trait Loci (QTL) mapper in cis for
molecular phenotypes</p>

<p style="margin-top: 1em">SYNOPSIS <br>
fastQTL [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The goal of FastQTL is to identify single-nucleotide
polymorphisms (SNPs) which are significantly associated with
various molecular phenotypes (i.e. expression of known
genes, <br>
cytosine methylation levels, etc). It performs scans for all
possible phenotype-variant pairs in cis (i.e. variants
located within a specific window around a phenotype). <br>
FastQTL implements a new permutation scheme (Beta
approximation) to accurately and rapidly correct for
multiple-testing at both the genotype and phenotype
levels.</p>

<p style="margin-top: 1em">OPTIONS <br>
Basic options <br>
--help <br>
Print help about options.</p>

<p style="margin-top: 1em">--silent <br>
Silent mode on terminal.</p>

<p style="margin-top: 1em">--seed arg (=1459232241) <br>
Random number generator seed. Useful to replicate runs of
the software.</p>

<p style="margin-top: 1em">Input and Output files options
<br>
-L, --log <br>
Screen output is copied in this file.</p>

<p style="margin-top: 1em">-V, --vcf <br>
Genotypes in VCF format.</p>

<p style="margin-top: 1em">-B, --bed <br>
Phenotypes in BED format.</p>

<p style="margin-top: 1em">-C, --cov <br>
Covariates in TXT format.</p>

<p style="margin-top: 1em">-G, --grp <br>
Phenotype groups in TXT format.</p>

<p style="margin-top: 1em">-O, --out <br>
Output file.</p>

<p style="margin-top: 1em">Exclusion and Inclusion files
options <br>
--exclude-samples <br>
List of samples to exclude.</p>

<p style="margin-top: 1em">--include-samples <br>
List of samples to include.</p>

<p style="margin-top: 1em">--exclude-sites <br>
List of sites to exclude.</p>

<p style="margin-top: 1em">--include-sites <br>
List of sites to include.</p>

<p style="margin-top: 1em">--exclude-phenotypes <br>
List of phenotypes to exclude.</p>

<p style="margin-top: 1em">--include-phenotypes <br>
List of phenotypes to include.</p>

<p style="margin-top: 1em">--exclude-covariates <br>
List of covariates to exclude.</p>

<p style="margin-top: 1em">--include-covariates <br>
List of covariates to include.</p>

<p style="margin-top: 1em">Parameters <br>
--normal <br>
To perform quantile normalization on the phenotype
quantifications to make them normally distributed.
Implemented as the rntransform function of the GenABEL
package.</p>

<p style="margin-top: 1em">-W, --window <br>
Cis-window size. Default values is 1Mb (1e6). It means that
all variants within 1e6 bp of the phenotype location (e.g.
TSS) is analyzed.</p>

<p style="margin-top: 1em">-T, --threshold <br>
To filter out all phenotype-variant pairs with a p-value
above the specified threshold in the output of a nominal
pass.</p>

<p style="margin-top: 1em">Modes <br>
-P, --permute <br>
Permutation pass to calculate corrected p-values for
molecular phenotypes.</p>

<p style="margin-top: 1em">--psequence <br>
Permutation sequence.</p>

<p style="margin-top: 1em">--map Map best QTL candidates
per molecular phenotype.</p>

<p style="margin-top: 1em">--map-full <br>
Scan full cis-window to discover independent signals.</p>

<p style="margin-top: 1em">--interaction <br>
Test for interactions with variable specified in file.</p>

<p style="margin-top: 1em">Parallelization <br>
-K, --chunk <br>
Specify which chunk needs to be processed.</p>

<p style="margin-top: 1em">--commands <br>
Generates all commands.</p>

<p style="margin-top: 1em">-R, --region <br>
Region of interest.</p>

<p style="margin-top: 1em">EXAMPLE <br>
for c in $(seq 1 256); do <br>
fastQTL --vcf genotypes.vcf.gz <br>
--bed phenotypes.bed.gz <br>
--chunk $c 256 <br>
--permute 1000 <br>
--out results.$c.txt.gz <br>
done <br>
zcat results.*.txt.gz | gzip -c &gt;
fastqtl_expected_output.txt.gz</p>

<p style="margin-top: 1em">SEE ALSO <br>
http://fastqtl.sourceforge.net/</p>

<p style="margin-top: 1em">AUTHOR <br>
FastQTL was written by Olivier Delaneau, Halit Ongen,
Alfonso Buil and Manolis Dermitzakis.</p>

<p style="margin-top: 1em">This manual page was written by
Dylan A&Atilde;&macr;ssi &lt;bob.dybian@gmail.com&gt;, for
the Debian project (but may be used by others).</p>

<p style="margin-top: 1em">2.184 March 2016 FastQTL(1)</p>
<hr>
</body>
</html>
