<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTAQ(1) User Commands FASTAQ(1)</p>

<p style="margin-top: 1em">NAME <br>
fastaq - FASTA and FASTQ file manipulation tools</p>

<p style="margin-top: 1em">SYNOPSIS <br>
fastaq &lt;command&gt; [options]</p>

<p style="margin-top: 1em">DESCRIPTION0nf <br>
To get minimal usage for a command use: fastaq command</p>

<p style="margin-top: 1em">To get full help for a command
use one of: fastaq command -h fastaq command --help</p>

<p style="margin-top: 1em">Available commands:</p>

<p style="margin-top: 1em">acgtn_only Replace every non
acgtnACGTN with an N add_indels Deletes or inserts bases at
given position(s) caf_to_fastq Converts a CAF file to <br>
FASTQ format capillary_to_pairs Converts file of capillary
reads to paired and unpaired files chunker Splits sequences
into equal sized chunks count_sequences <br>
Counts the sequences in input file deinterleave Splits
interleaved paired file into two separate files
enumerate_names Renames sequences in a file, calling them
<br>
1,2,3... etc expand_nucleotides Makes every combination of
degenerate nucleotides fasta_to_fastq Convert FASTA and
.qual to FASTQ filter Filter <br>
sequences to get a subset of them get_ids Get the ID of each
sequence get_seq_flanking_gaps Gets the sequences flanking
gaps interleave Interleaves <br>
two files, output is alternating between fwd/rev reads
make_random_contigs Make contigs of random sequence merge
Converts multi sequence file to a single <br>
sequence replace_bases Replaces all occurrences of one
letter with another reverse_complement Reverse complement
all sequences scaffolds_to_contigs Creates a file <br>
of contigs from a file of scaffolds search_for_seq Find all
exact matches to a string (and its reverse complement)
sequence_trim Trim exact matches to a given <br>
string off the start of every sequence sort_by_size Sorts
sequences in length order split_by_base_count Split multi
sequence file into separate files strip_illu&acirc; <br>
mina_suffix Strips /1 or /2 off the end of every read name
to_boulderio Converts to Boulder-IO format, used by primer3
to_fake_qual Make fake quality scores <br>
file to_fasta Converts a variety of input formats to nicely
formatted FASTA format to_mira_xml Create an xml file from a
file of reads, for use with Mira <br>
assembler to_orfs_gff Writes a GFF file of open reading
frames to_perfect_reads Make perfect paired reads from
reference to_random_subset Make a random <br>
sample of sequences (and optionally mates as well)
to_tiling_bam Make a BAM file of reads uniformly spread
across the input reference to_unique_by_id Remove <br>
duplicate sequences, based on their names. Keep longest seqs
translate Translate all sequences in input nucleotide
sequences trim_Ns_at_end Trims all Ns at <br>
the start/end of all sequences trim_contigs Trims a set
number of bases off the end of every contig trim_ends Trim
fixed number of bases of start and/or <br>
end of every sequence version Print version number and
exit</p>

<p style="margin-top: 1em">fastaq 3.14.0 September 2016
FASTAQ(1)</p>
<hr>
</body>
</html>
