<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:09:00 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTX_CLIPPER(1) User Commands FASTX_CLIPPER(1)</p>

<p style="margin-top: 1em">NAME <br>
fastx_clipper - FASTA/Q Clipper</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: fastx_clipper [-h] [-a ADAPTER] [-D] [-l N] [-n] [-d
N] [-c] [-C] [-o] [-v] [-z] [-i INFILE] [-o OUTFILE] Part of
FASTX Toolkit 0.0.14 by A. Gordon
(assafgordon@gmail.com)</p>

<p style="margin-top: 1em">[-h] = This helpful help
screen.</p>

<p style="margin-top: 1em">[-a ADAPTER] = ADAPTER string.
default is CCTTAAGG (dummy adapter). [-l N] = discard
sequences shorter than N nucleotides. default is 5. [-d N] =
Keep the <br>
adapter and N bases after it.</p>

<p style="margin-top: 1em">(using &rsquo;-d 0&rsquo; is the
same as not using &rsquo;-d&rsquo; at all. which is the
default).</p>

<p style="margin-top: 1em">[-c] = Discard non-clipped
sequences (i.e. - keep only sequences which contained the
adapter).</p>

<p style="margin-top: 1em">[-C] = Discard clipped sequences
(i.e. - keep only sequences which did not contained the
adapter).</p>

<p style="margin-top: 1em">[-k] = Report Adapter-Only
sequences.</p>

<p style="margin-top: 1em">[-n] = keep sequences with
unknown (N) nucleotides. default is to discard such
sequences.</p>

<p style="margin-top: 1em">[-v] = Verbose - report number
of sequences.</p>

<p style="margin-top: 1em">If [-o] is specified, <br>
report will be printed to STDOUT.</p>

<p style="margin-top: 1em">If [-o] is not specified (and
output goes to STDOUT), report will be printed to
STDERR.</p>

<p style="margin-top: 1em">[-z] = Compress output with
GZIP.</p>

<p style="margin-top: 1em">[-D] = DEBUG output.</p>

<p style="margin-top: 1em">[-M N] = require minimum adapter
alignment length of N.</p>

<p style="margin-top: 1em">If less than N nucleotides
aligned with the adapter - don&rsquo;t clip it. <br>
[-i INFILE] = FASTA/Q input file. default is STDIN.</p>

<p style="margin-top: 1em">[-o OUTFILE] = FASTA/Q output
file. default is STDOUT.</p>

<p style="margin-top: 1em">SEE ALSO <br>
The quality of this automatically generated manpage might be
insufficient. It is suggested to visit</p>


<p style="margin-top: 1em">http://hannonlab.cshl.edu/fastx_toolkit/commandline.html</p>

<p style="margin-top: 1em">to get a better layout as well
as an overview about connected FASTX tools.</p>

<p style="margin-top: 1em">fastx_clipper 0.0.14 December
2016 FASTX_CLIPPER(1)</p>
<hr>
</body>
</html>
