<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:09:00 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>fasttreeMP(1) User Commands fasttreeMP(1)</p>

<p style="margin-top: 1em">NAME <br>
fasttreeMP - create phylogenetic trees from alignments of
nucleotide or protein sequences (openMP version)</p>

<p style="margin-top: 1em">DESCRIPTION <br>
fasttreeMP infers approximately-maximum-likelihood
phylogenetic trees from alignments of nucleotide or protein
sequences. It handles alignments with up to a million of
sequences <br>
in a reasonable amount of time and memory.</p>

<p style="margin-top: 1em">fasttreeMP is more accurate than
PhyML 3 with default settings, and much more accurate than
the distance-matrix methods that are traditionally used for
large alignments. fast&acirc; <br>
treeMP uses the Jukes-Cantor or generalized time-reversible
(GTR) models of nucleotide evolution and the JTT
(Jones-Taylor-Thornton 1992) model of amino acid evolution.
To <br>
account for the varying rates of evolution across sites,
fasttreeMP uses a single rate for each site (the
&quot;CAT&quot; approximation). To quickly estimate the
reliability of each split <br>
in the tree, fasttreeMP computes local support values with
the Shimodaira-Hasegawa test (these are the same as PhyML
3&rsquo;s &quot;SH-like local supports&quot;).</p>

<p style="margin-top: 1em">SYNOPSIS <br>
fasttreeMP protein_alignment &gt; tree</p>

<p style="margin-top: 1em">fasttreeMP -nt
nucleotide_alignment &gt; tree</p>

<p style="margin-top: 1em">fasttreeMP -nt -gtr &lt;
nucleotide_alignment &gt; tree</p>

<p style="margin-top: 1em">fasttreeMP accepts alignments in
fasta or phylip interleaved formats</p>

<p style="margin-top: 1em">Common options (must be before
the alignment file): <br>
-quiet to suppress reporting information</p>

<p style="margin-top: 1em">-nopr to suppress progress
indicator</p>

<p style="margin-top: 1em">-log logfile -- save
intermediate trees, settings, and model details</p>

<p style="margin-top: 1em">-fastest -- speed up the
neighbor joining phase &amp; reduce memory usage</p>

<p style="margin-top: 1em">(recommended for &gt;50,000
sequences)</p>

<p style="margin-top: 1em">-n &lt;number&gt; to analyze
multiple alignments (phylip format only)</p>

<p style="margin-top: 1em">(use for global bootstrap, with
seqboot and CompareToBootstrap.pl)</p>

<p style="margin-top: 1em">-nosupport to not compute
support values</p>

<p style="margin-top: 1em">-intree newick_file to set the
starting tree(s)</p>

<p style="margin-top: 1em">-intree1 newick_file to use this
starting tree for all the alignments</p>

<p style="margin-top: 1em">(for faster global bootstrap on
huge alignments)</p>

<p style="margin-top: 1em">-pseudo to use pseudocounts
(recommended for highly gapped sequences)</p>

<p style="margin-top: 1em">-gtr -- generalized
time-reversible model (nucleotide alignments only)</p>

<p style="margin-top: 1em">-wag -- Whelan-And-Goldman 2001
model (amino acid alignments only)</p>

<p style="margin-top: 1em">-quote -- allow spaces and other
restricted characters (but not &rsquo; characters) in</p>

<p style="margin-top: 1em">sequence names and quote names
in the output tree (fasta input only; fasttreeMP will not be
able to read these trees back in</p>

<p style="margin-top: 1em">-noml to turn off
maximum-likelihood</p>

<p style="margin-top: 1em">-nome to turn off
minimum-evolution NNIs and SPRs</p>

<p style="margin-top: 1em">(recommended if running
additional ML NNIs with -intree)</p>

<p style="margin-top: 1em">-nome -mllen with -intree to
optimize branch lengths for a fixed topology</p>

<p style="margin-top: 1em">-cat # to specify the number of
rate categories of sites (default 20)</p>

<p style="margin-top: 1em">or -nocat to use constant
rates</p>

<p style="margin-top: 1em">-gamma -- after optimizing the
tree under the CAT approximation,</p>

<p style="margin-top: 1em">rescale the lengths to optimize
the Gamma20 likelihood</p>

<p style="margin-top: 1em">-constraints constraintAlignment
to constrain the topology search</p>

<p style="margin-top: 1em">constraintAlignment should have
1s or 0s to indicates splits</p>

<p style="margin-top: 1em">-expert -- see more options</p>

<p style="margin-top: 1em">Detailed usage for fasttreeMP
2.1.4 SSE3: <br>
fasttreeMP [-nt] [-n 100] [-quote] [-pseudo | -pseudo
1.0]</p>

<p style="margin-top: 1em">[-boot 1000 | -nosupport]
[-intree starting_trees_file | -intree1 starting_tree_file]
[-quiet | -nopr] [-nni 10] [-spr 2] [-noml | -mllen | -mlnni
10] [-mlacc 2] [-cat 20 <br>
| -nocat] [-gamma] [-slow | -fastest] [-2nd | -no2nd]
[-slownni] [-seed 1253] [-top | -notop] [-topm 1.0 [-close
0.75] [-refresh 0.8]] [-matrix Matrix | -nomatrix] [-nj |
<br>
-bionj] [-wag] [-nt] [-gtr] [-gtrrates ac ag at cg ct gt]
[-gtrfreq A C G T] [ -constraints constraintAlignment [
-constraintWeight 100.0 ] ] [-log logfile]</p>

<p style="margin-top: 1em">[ alignment_file ]</p>

<p style="margin-top: 1em">&gt; newick_tree</p>

<p style="margin-top: 1em">or</p>

<p style="margin-top: 1em">fasttreeMP [-nt] [-matrix Matrix
| -nomatrix] [-rawdist] -makematrix [alignment]</p>

<p style="margin-top: 1em">[-n 100] &gt;
phylip_distance_matrix</p>

<p style="margin-top: 1em">fasttreeMP supports fasta or
phylip interleaved alignments By default fasttreeMP expects
protein alignments, use -nt for nucleotides fasttreeMP reads
standard input if no <br>
alignment file is given</p>

<p style="margin-top: 1em">Input/output options: <br>
-n -- read in multiple alignments in. This only</p>

<p style="margin-top: 1em">works with phylip interleaved
format. For example, you can use it with the output from
phylip&rsquo;s seqboot. If you use -n, fasttreeMP will write
1 tree per line to standard <br>
output.</p>

<p style="margin-top: 1em">-intree newickfile -- read the
starting tree in from newickfile.</p>

<p style="margin-top: 1em">Any branch lengths in the
starting trees are ignored.</p>

<p style="margin-top: 1em">-intree with -n will read a
separate starting tree for each alignment.</p>

<p style="margin-top: 1em">-intree1 newickfile -- read the
same starting tree for each alignment</p>

<p style="margin-top: 1em">-quiet -- do not write to
standard error during normal operation (no progress</p>

<p style="margin-top: 1em">indicator, no options summary,
no likelihood values, etc.)</p>

<p style="margin-top: 1em">-nopr -- do not write the
progress indicator to stderr</p>

<p style="margin-top: 1em">-log logfile -- save
intermediate trees so you can extract</p>

<p style="margin-top: 1em">the trees and restart
long-running jobs if they crash -log also reports the
per-site rates (1 means slowest category)</p>

<p style="margin-top: 1em">-quote -- quote sequence names
in the output and allow spaces, commas,</p>

<p style="margin-top: 1em">parentheses, and colons in them
but not &rsquo; characters (fasta files only)</p>

<p style="margin-top: 1em">Distances: <br>
Default: For protein sequences, log-corrected distances and
an</p>

<p style="margin-top: 1em">amino acid dissimilarity matrix
derived from BLOSUM45</p>

<p style="margin-top: 1em">or for nucleotide sequences,
Jukes-Cantor distances To specify a different matrix, use
-matrix FilePrefix or -nomatrix Use -rawdist to turn the
log-correction off or to <br>
use %different instead of Jukes-Cantor</p>

<p style="margin-top: 1em">-pseudo [weight] -- Use
pseudocounts to estimate distances between</p>

<p style="margin-top: 1em">sequences with little or no
overlap. (Off by default.) Recommended if analyzing the
alignment has sequences with little or no overlap. If the
weight is not specified, it <br>
is 1.0</p>

<p style="margin-top: 1em">Topology refinement: <br>
By default, fasttreeMP tries to improve the tree with up to
4*log2(N) rounds of minimum-evolution nearest-neighbor
interchanges (NNI), where N is the number of unique <br>
sequences, 2 rounds of subtree-prune-regraft (SPR) moves
(also min. evo.), and up to 2*log(N) rounds of
maximum-likelihood NNIs. Use -nni to set the number of
rounds of <br>
min. evo. NNIs, and -spr to set the rounds of SPRs. Use
-noml to turn off both min-evo NNIs and SPRs (useful if
refining</p>

<p style="margin-top: 1em">an approximately
maximum-likelihood tree with further NNIs)</p>

<p style="margin-top: 1em">Use -sprlength set the maximum
length of a SPR move (default 10) Use -mlnni to set the
number of rounds of maximum-likelihood NNIs Use -mlacc 2 or
-mlacc 3 to always opti&acirc; <br>
mize all 5 branches at each NNI,</p>

<p style="margin-top: 1em">and to optimize all 5 branches
in 2 or 3 rounds</p>

<p style="margin-top: 1em">Use -mllen to optimize branch
lengths without ML NNIs Use -mllen -nome with -intree to
optimize branch lengths on a fixed topology Use -slownni to
turn off heuristics to <br>
avoid constant subtrees (affects both</p>

<p style="margin-top: 1em">ML and ME NNIs)</p>

<p style="margin-top: 1em">Maximum likelihood model
options: <br>
-wag -- Whelan-And-Goldman 2001 model instead of (default)
Jones-Taylor-Thorton 1992 model (a.a. only)</p>

<p style="margin-top: 1em">-gtr -- generalized
time-reversible instead of (default) Jukes-Cantor (nt
only)</p>

<p style="margin-top: 1em">-cat # -- specify the number of
rate categories of sites (default 20)</p>

<p style="margin-top: 1em">-nocat -- no CAT model (just 1
category)</p>

<p style="margin-top: 1em">-gamma -- after the final round
of optimizing branch lengths with the CAT model,</p>

<p style="margin-top: 1em">report the likelihood under the
discrete gamma model with the same number of categories.
fasttreeMP uses the same branch lengths but optimizes the
gamma shape parameter <br>
and the scale of the lengths. The final tree will have
rescaled lengths. Used with -log, this also generates
per-site likelihoods for use with CONSEL, see
GammaLog&acirc; <br>
ToPaup.pl and documentation on the fasttreeMP web site.</p>

<p style="margin-top: 1em">Support value options: <br>
By default, fasttreeMP computes local support values by
resampling the site likelihoods 1,000 times and the
Shimodaira Hasegawa test. If you specify -nome, it will
compute <br>
minimum-evolution bootstrap supports instead In either case,
the support values are proportions ranging from 0 to 1</p>

<p style="margin-top: 1em">Use -nosupport to turn off
support values or -boot 100 to use just 100 resamples Use
-seed to initialize the random number generator</p>

<p style="margin-top: 1em">Searching for the best join:
<br>
By default, fasttreeMP combines the &rsquo;visible
set&rsquo; of fast neighbor-joining with</p>

<p style="margin-top: 1em">local hill-climbing as in
relaxed neighbor-joining</p>

<p style="margin-top: 1em">-slow -- exhaustive search (like
NJ or BIONJ, but different gap handling)</p>

<p style="margin-top: 1em">-slow takes half an hour instead
of 8 seconds for 1,250 proteins</p>

<p style="margin-top: 1em">-fastest -- search the visible
set (the top hit for each node) only</p>

<p style="margin-top: 1em">Unlike the original fast
neighbor-joining, -fastest updates visible(C) after joining
A and B if join(AB,C) is better than join(C,visible(C))
-fastest also updates out-dis&acirc; <br>
tances in a very lazy way, -fastest sets -2nd on as well,
use -fastest -no2nd to avoid this</p>

<p style="margin-top: 1em">Top-hit heuristics: <br>
By default, fasttreeMP uses a top-hit list to speed up
search Use -notop (or -slow) to turn this feature off</p>

<p style="margin-top: 1em">and compare all leaves to each
other, and all new joined nodes to each other</p>

<p style="margin-top: 1em">-topm 1.0 -- set the top-hit
list size to parameter*sqrt(N)</p>

<p style="margin-top: 1em">fasttreeMP estimates the top m
hits of a leaf from the top 2*m hits of a
&rsquo;close&rsquo; neighbor, where close is defined as
d(seed,close) &lt; 0.75 * d(seed, hit of rank 2*m), and <br>
updates the top-hits as joins proceed</p>

<p style="margin-top: 1em">-close 0.75 -- modify the close
heuristic, lower is more conservative</p>

<p style="margin-top: 1em">-refresh 0.8 -- compare a joined
node to all other nodes if its</p>

<p style="margin-top: 1em">top-hit list is less than 80% of
the desired length, or if the age of the top-hit list is
log2(m) or greater</p>

<p style="margin-top: 1em">-2nd or -no2nd to turn 2nd-level
top hits heuristic on or off</p>

<p style="margin-top: 1em">This reduces memory usage and
running time but may lead to marginal reductions in tree
quality. (By default, -fastest turns on -2nd.)</p>

<p style="margin-top: 1em">Join options: <br>
-nj: regular (unweighted) neighbor-joining (default)</p>

<p style="margin-top: 1em">-bionj: weighted joins as in
BIONJ</p>

<p style="margin-top: 1em">fasttreeMP will also weight
joins during NNIs</p>

<p style="margin-top: 1em">Constrained topology search
options: <br>
-constraints alignmentfile -- an alignment with values of 0,
1, and -</p>

<p style="margin-top: 1em">Not all sequences need be
present. A column of 0s and 1s defines a constrained split.
Some constraints may be violated (see &rsquo;violating
constraints:&rsquo; in standard error).</p>

<p style="margin-top: 1em">-constraintWeight -- how
strongly to weight the constraints. A value of 1</p>

<p style="margin-top: 1em">means a penalty of 1 in tree
length for violating a constraint Default: 100.0</p>

<p style="margin-top: 1em">For more information, see
http://www.microbesonline.org/fasttree/</p>

<p style="margin-top: 1em">or the comments in the source
code</p>

<p style="margin-top: 1em">fasttreeMP protein_alignment
&gt; tree fasttreeMP -nt nucleotide_alignment &gt; tree
fasttreeMP -nt -gtr &lt; nucleotide_alignment &gt; tree</p>

<p style="margin-top: 1em">fasttreeMP accepts alignments in
fasta or phylip interleaved formats</p>

<p style="margin-top: 1em">Common options (must be before
the alignment file): <br>
-quiet to suppress reporting information</p>

<p style="margin-top: 1em">-nopr to suppress progress
indicator</p>

<p style="margin-top: 1em">-log logfile -- save
intermediate trees, settings, and model details</p>

<p style="margin-top: 1em">-fastest -- speed up the
neighbor joining phase &amp; reduce memory usage</p>

<p style="margin-top: 1em">(recommended for &gt;50,000
sequences)</p>

<p style="margin-top: 1em">-n &lt;number&gt; to analyze
multiple alignments (phylip format only)</p>

<p style="margin-top: 1em">(use for global bootstrap, with
seqboot and CompareToBootstrap.pl)</p>

<p style="margin-top: 1em">-nosupport to not compute
support values</p>

<p style="margin-top: 1em">-intree newick_file to set the
starting tree(s)</p>

<p style="margin-top: 1em">-intree1 newick_file to use this
starting tree for all the alignments</p>

<p style="margin-top: 1em">(for faster global bootstrap on
huge alignments)</p>

<p style="margin-top: 1em">-pseudo to use pseudocounts
(recommended for highly gapped sequences)</p>

<p style="margin-top: 1em">-gtr -- generalized
time-reversible model (nucleotide alignments only)</p>

<p style="margin-top: 1em">-wag -- Whelan-And-Goldman 2001
model (amino acid alignments only)</p>

<p style="margin-top: 1em">-quote -- allow spaces and other
restricted characters (but not &rsquo; characters) in</p>

<p style="margin-top: 1em">sequence names and quote names
in the output tree (fasta input only; fasttreeMP will not be
able to read these trees back in</p>

<p style="margin-top: 1em">-noml to turn off
maximum-likelihood</p>

<p style="margin-top: 1em">-nome to turn off
minimum-evolution NNIs and SPRs</p>

<p style="margin-top: 1em">(recommended if running
additional ML NNIs with -intree)</p>

<p style="margin-top: 1em">-nome -mllen with -intree to
optimize branch lengths for a fixed topology</p>

<p style="margin-top: 1em">-cat # to specify the number of
rate categories of sites (default 20)</p>

<p style="margin-top: 1em">or -nocat to use constant
rates</p>

<p style="margin-top: 1em">-gamma -- after optimizing the
tree under the CAT approximation,</p>

<p style="margin-top: 1em">rescale the lengths to optimize
the Gamma20 likelihood</p>

<p style="margin-top: 1em">-constraints constraintAlignment
to constrain the topology search</p>

<p style="margin-top: 1em">constraintAlignment should have
1s or 0s to indicates splits</p>

<p style="margin-top: 1em">-expert -- see more options</p>

<p style="margin-top: 1em">For more information, see
http://www.microbesonline.org/fasttree/</p>

<p style="margin-top: 1em">Lawrence Berkeley National Lab
June 2012 fasttreeMP(1)</p>
<hr>
</body>
</html>
