<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FASTQ-MCF(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FASTQ-MCF(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">FASTQ-MCF(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
fastq-mcf - ea-utils: detect levels of adapter presence, compute likelihoods and
  locations of the adapters
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>fastq-mcf</b> [ <i>options</i>] <i>&lt;adapters.fa&gt; &lt;reads.fq&gt;
  </i>[<i>mates1.fq </i>...]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Version: 1.04.676
<div class="Pp"></div>
Detects levels of adapter presence, computes likelihoods and locations (start,
  end) of the adapters. Removes the adapter sequences from the fastq file(s).
<div class="Pp"></div>
Stats go to stderr, unless <b>-o</b> is specified.
<div class="Pp"></div>
Specify <b>-0</b> to turn off all default settings
<div class="Pp"></div>
If you specify multiple 'paired-end' inputs, then a <b>-o</b> option is required
  for each. IE: <b>-o</b> read1.clip.q <b>-o</b> read2.clip.fq
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">This help</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> FIL</dt>
  <dd class="It-tag">Output file (stats to stdout)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> N.N</dt>
  <dd class="It-tag">Log scale for adapter minimum-length-match (2.2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> N</dt>
  <dd class="It-tag">% occurance threshold before adapter clipping (0.25)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> N</dt>
  <dd class="It-tag">Minimum clip length, overrides scaled auto (1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> N</dt>
  <dd class="It-tag">Maximum adapter difference percentage (10)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> N</dt>
  <dd class="It-tag">Minimum remaining sequence length (19)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b> N</dt>
  <dd class="It-tag">Maximum remaining sequence length (none)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b> N</dt>
  <dd class="It-tag">Remove duplicate reads : Read_1 has an identical N bases
      (0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> N</dt>
  <dd class="It-tag">sKew percentage-less-than causing cycle removal (2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-x</b> N</dt>
  <dd class="It-tag">'N' (Bad read) percentage causing cycle removal (20)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> N</dt>
  <dd class="It-tag">quality threshold causing base removal (10)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b> N</dt>
  <dd class="It-tag">window-size for quality trimming (1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-H</b></dt>
  <dd class="It-tag">remove &gt;95% homopolymer reads (no)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-X</b></dt>
  <dd class="It-tag">remove low complexity reads (no)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-0</b></dt>
  <dd class="It-tag">Set all default parameters to zero/do nothing</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b>|u</dt>
  <dd class="It-tag">Force disable/enable Illumina PF filtering (auto)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P</b> N</dt>
  <dd class="It-tag">Phred-scale (auto)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b></dt>
  <dd class="It-tag">Don't remove N's from the fronts/ends of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">Don't clip, just output what would be done</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b> N</dt>
  <dd class="It-tag">Number of reads to use for subsampling (300k)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">Save all discarded reads to '.skip' files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b></dt>
  <dd class="It-tag">Output lots of random debugging stuff</dd>
</dl>
<h2 class="Ss" title="Ss" id="Quality_adjustment_options:"><a class="selflink" href="#Quality_adjustment_options:">Quality
  adjustment options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cycle-adjust</b></dt>
  <dd class="It-tag">CYC,AMT Adjust cycle CYC (negative = offset from end) by
      amount AMT</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--phred-adjust</b></dt>
  <dd class="It-tag">SCORE,AMT Adjust score SCORE by amount AMT</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--phred-adjust-max</b></dt>
  <dd class="It-tag">SCORE Adjust scores &gt; SCORE to SCOTE</dd>
</dl>
<h2 class="Ss" title="Ss" id="Filtering_options*:"><a class="selflink" href="#Filtering_options*:">Filtering
  options*:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--[mate-]qual-mean</b></dt>
  <dd class="It-tag">NUM Minimum mean quality score</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--[mate-]qual-gt</b></dt>
  <dd class="It-tag">NUM,THR At least NUM quals &gt; THR</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--[mate-]max-ns</b></dt>
  <dd class="It-tag">NUM Maxmium N-calls in a read (can be a %)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--[mate-]min-len</b></dt>
  <dd class="It-tag">NUM Minimum remaining length (same as <b>-l</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--homopolymer-pct</b></dt>
  <dd class="It-tag">PCT Homopolymer filter percent (95)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--lowcomplex-pct</b></dt>
  <dd class="It-tag">PCT Complexity filter percent (95)</dd>
</dl>
<div class="Pp"></div>
If mate- prefix is used, then applies to second non-barcode read only
<div class="Pp"></div>
Adapter files are 'fasta' formatted:
<div class="Pp"></div>
Specify n/a to turn off adapter clipping, and just use filters
<div class="Pp"></div>
Increasing the scale makes recognition-lengths longer, a scale of 100 will force
  full-length recognition of adapters.
<div class="Pp"></div>
Adapter sequences with _5p in their label will match 'end's, and sequences with
  _3p in their label will match 'start's, otherwise the 'end' is
  auto-determined.
<div class="Pp"></div>
Skew is when one cycle is poor, 'skewed' toward a particular base. If any
  nucleotide is less than the skew percentage, then the whole cycle is removed.
  Disable for methyl-seq, etc.
<div class="Pp"></div>
Set the skew ( <b>-k</b>) or N-pct (<b>-x</b>) to 0 to turn it off (should be
  done for miRNA, amplicon and other low-complexity situations!)
<div class="Pp"></div>
Duplicate read filtering is appropriate for assembly tasks, and never when read
  length &lt; expected coverage. <b>-D</b> 50 will use 4.5GB RAM on 100m DNA
  reads - be careful. Great for RNA assembly.
<div class="Pp"></div>
*Quality filters are evaluated after clipping/trimming
<div class="Pp"></div>
Homopolymer filtering is a subset of low-complexity, but will not be separately
  tracked unless both are turned on.</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2015</td>
    <td class="foot-os">fastq-mcf 1.1.2</td>
  </tr>
</table>
</body>
</html>
