<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTQ-MCF(1) User Commands FASTQ-MCF(1)</p>

<p style="margin-top: 1em">NAME <br>
fastq-mcf - ea-utils: detect levels of adapter presence,
compute likelihoods and locations of the adapters</p>

<p style="margin-top: 1em">SYNOPSIS <br>
fastq-mcf [options] &lt;adapters.fa&gt; &lt;reads.fq&gt;
[mates1.fq ...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Version: 1.04.676</p>

<p style="margin-top: 1em">Detects levels of adapter
presence, computes likelihoods and locations (start, end) of
the adapters. Removes the adapter sequences from the fastq
file(s).</p>

<p style="margin-top: 1em">Stats go to stderr, unless -o is
specified.</p>

<p style="margin-top: 1em">Specify -0 to turn off all
default settings</p>

<p style="margin-top: 1em">If you specify multiple
&rsquo;paired-end&rsquo; inputs, then a -o option is
required for each. IE: -o read1.clip.q -o read2.clip.fq</p>

<p style="margin-top: 1em">OPTIONS <br>
-h This help</p>

<p style="margin-top: 1em">-o FIL Output file (stats to
stdout)</p>

<p style="margin-top: 1em">-s N.N Log scale for adapter
minimum-length-match (2.2)</p>

<p style="margin-top: 1em">-t N % occurance threshold
before adapter clipping (0.25)</p>

<p style="margin-top: 1em">-m N Minimum clip length,
overrides scaled auto (1)</p>

<p style="margin-top: 1em">-p N Maximum adapter difference
percentage (10)</p>

<p style="margin-top: 1em">-l N Minimum remaining sequence
length (19)</p>

<p style="margin-top: 1em">-L N Maximum remaining sequence
length (none)</p>

<p style="margin-top: 1em">-D N Remove duplicate reads :
Read_1 has an identical N bases (0)</p>

<p style="margin-top: 1em">-k N sKew percentage-less-than
causing cycle removal (2)</p>

<p style="margin-top: 1em">-x N &rsquo;N&rsquo; (Bad read)
percentage causing cycle removal (20)</p>

<p style="margin-top: 1em">-q N quality threshold causing
base removal (10)</p>

<p style="margin-top: 1em">-w N window-size for quality
trimming (1)</p>

<p style="margin-top: 1em">-H remove &gt;95% homopolymer
reads (no)</p>

<p style="margin-top: 1em">-X remove low complexity reads
(no)</p>

<p style="margin-top: 1em">-0 Set all default parameters to
zero/do nothing</p>

<p style="margin-top: 1em">-U|u Force disable/enable
Illumina PF filtering (auto)</p>

<p style="margin-top: 1em">-P N Phred-scale (auto)</p>

<p style="margin-top: 1em">-R Don&rsquo;t remove N&rsquo;s
from the fronts/ends of reads</p>

<p style="margin-top: 1em">-n Don&rsquo;t clip, just output
what would be done</p>

<p style="margin-top: 1em">-C N Number of reads to use for
subsampling (300k)</p>

<p style="margin-top: 1em">-S Save all discarded reads to
&rsquo;.skip&rsquo; files</p>

<p style="margin-top: 1em">-d Output lots of random
debugging stuff</p>

<p style="margin-top: 1em">Quality adjustment options: <br>
--cycle-adjust <br>
CYC,AMT Adjust cycle CYC (negative = offset from end) by
amount AMT</p>

<p style="margin-top: 1em">--phred-adjust <br>
SCORE,AMT Adjust score SCORE by amount AMT</p>

<p style="margin-top: 1em">--phred-adjust-max <br>
SCORE Adjust scores &gt; SCORE to SCOTE</p>

<p style="margin-top: 1em">Filtering options*: <br>
--[mate-]qual-mean <br>
NUM Minimum mean quality score</p>

<p style="margin-top: 1em">--[mate-]qual-gt <br>
NUM,THR At least NUM quals &gt; THR</p>

<p style="margin-top: 1em">--[mate-]max-ns <br>
NUM Maxmium N-calls in a read (can be a %)</p>

<p style="margin-top: 1em">--[mate-]min-len <br>
NUM Minimum remaining length (same as -l)</p>

<p style="margin-top: 1em">--homopolymer-pct <br>
PCT Homopolymer filter percent (95)</p>

<p style="margin-top: 1em">--lowcomplex-pct <br>
PCT Complexity filter percent (95)</p>

<p style="margin-top: 1em">If mate- prefix is used, then
applies to second non-barcode read only</p>

<p style="margin-top: 1em">Adapter files are
&rsquo;fasta&rsquo; formatted:</p>

<p style="margin-top: 1em">Specify n/a to turn off adapter
clipping, and just use filters</p>

<p style="margin-top: 1em">Increasing the scale makes
recognition-lengths longer, a scale of 100 will force
full-length recognition of adapters.</p>

<p style="margin-top: 1em">Adapter sequences with _5p in
their label will match &rsquo;end&rsquo;s, and sequences
with _3p in their label will match &rsquo;start&rsquo;s,
otherwise the &rsquo;end&rsquo; is auto-determined.</p>

<p style="margin-top: 1em">Skew is when one cycle is poor,
&rsquo;skewed&rsquo; toward a particular base. If any
nucleotide is less than the skew percentage, then the whole
cycle is removed. Disable for methyl-seq, <br>
etc.</p>

<p style="margin-top: 1em">Set the skew (-k) or N-pct (-x)
to 0 to turn it off (should be done for miRNA, amplicon and
other low-complexity situations!)</p>

<p style="margin-top: 1em">Duplicate read filtering is
appropriate for assembly tasks, and never when read length
&lt; expected coverage. -D 50 will use 4.5GB RAM on 100m DNA
reads - be careful. Great for <br>
RNA assembly.</p>

<p style="margin-top: 1em">*Quality filters are evaluated
after clipping/trimming</p>

<p style="margin-top: 1em">Homopolymer filtering is a
subset of low-complexity, but will not be separately tracked
unless both are turned on.</p>

<p style="margin-top: 1em">fastq-mcf 1.1.2 July 2015
FASTQ-MCF(1)</p>
<hr>
</body>
</html>
