<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FASTX_BARCODE_SPLITTER.PL(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FASTX_BARCODE_SPLITTER.PL(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">FASTX_BARCODE_SPLITTER.PL(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
fastx_barcode_splitter.pl - FASTX Barcode Splitter
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Barcode Splitter, by Assaf Gordon (gordon@cshl.edu), 11sep2008
<div class="Pp"></div>
This program reads FASTA/FASTQ file and splits it into several smaller files,
  Based on barcode matching. FASTA/FASTQ data is read from STDIN (format is
  auto-detected.) Output files will be writen to disk. Summary will be printed
  to STDOUT.
<div class="Pp"></div>
usage: r.pl <b>--bcfile</b> FILE <b>--prefix</b> PREFIX [--suffix SUFFIX]
  [--bol|--eol]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[--mismatches N] [--exact] [--partial N] [--help] [--quiet]
      [--debug]</dd>
</dl>
<div class="Pp"></div>
Arguments:
<div class="Pp"></div>
<b>--bcfile</b> FILE - Barcodes file name. (see explanation below.)
  <b>--prefix</b> PREFIX - File prefix. will be added to the output files. Can
  be used
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">to specify output directories.</dd>
</dl>
<div class="Pp"></div>
<b>--suffix</b> SUFFIX - File suffix (optional). Can be used to specify file
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">extensions.</dd>
</dl>
<div class="Pp"></div>
<b>--bol</b> - Try to match barcodes at the BEGINNING of sequences.
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(What biologists would call the 5' end, and programmers
      would call index 0.)</dd>
</dl>
<div class="Pp"></div>
<b>--eol</b> - Try to match barcodes at the END of sequences.
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(What biologists would call the 3' end, and programmers
      would call the end of the string.) NOTE: one of <b>--bol</b>, <b>--eol</b>
      must be specified, but not both.</dd>
</dl>
<div class="Pp"></div>
<b>--mismatches</b> N - Max. number of mismatches allowed. default is 1.
  <b>--exact</b> - Same as '--mismatches 0'. If both <b>--exact</b> and
  <b>--mismatches</b>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">are specified, '--exact' takes precedence.</dd>
</dl>
<div class="Pp"></div>
<b>--partial</b> N - Allow partial overlap of barcodes. (see explanation below.)
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(Default is not partial matching)</dd>
</dl>
<div class="Pp"></div>
<b>--quiet</b> - Don't print counts and summary at the end of the run.
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(Default is to print.)</dd>
</dl>
<div class="Pp"></div>
<b>--debug</b> - Print lots of useless debug information to STDERR.
  <b>--help</b> - This helpful help screen.
<div class="Pp"></div>
Example (Assuming 's_2_100.txt' is a FASTQ file, 'mybarcodes.txt' is the
  barcodes file):
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">$ cat s_2_100.txt | fastx_barcode_splitter.pl --bcfile
      mybarcodes.txt --bol --mismatches 2 \</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--prefix</b>
  <i>/tmp/bla_</i> <b>--suffix</b> &quot;.txt&quot;</div>
<div class="Pp"></div>
Barcode file format <b>-------------------</b> Barcode files are simple text
  files. Each line should contain an identifier (descriptive name for the
  barcode), and the barcode itself (A/C/G/T), separated by a TAB character.
  Example:
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">#This line is a comment (starts with a 'number' sign) BC1
      GATCT BC2 ATCGT BC3 GTGAT BC4 TGTCT</dd>
</dl>
<div class="Pp"></div>
For each barcode, a new FASTQ file will be created (with the barcode's
  identifier as part of the file name). Sequences matching the barcode will be
  stored in the appropriate file.
<div class="Pp"></div>
Running the above example (assuming &quot;mybarcodes.txt&quot; contains the
  above barcodes), will create the following files:
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><i>/tmp/bla_BC1.txt</i> <i>/tmp/bla_BC2.txt</i>
      <i>/tmp/bla_BC3.txt</i> <i>/tmp/bla_BC4.txt</i>
      <i>/tmp/bla_unmatched.txt</i></dd>
</dl>
<div class="Pp"></div>
The 'unmatched' file will contain all sequences that didn't match any barcode.
<div class="Pp"></div>
Barcode matching <b>----------------</b>
<div class="Pp"></div>
** Without partial matching:
<div class="Pp"></div>
Count mismatches between the FASTA/Q sequences and the barcodes. The barcode
  which matched with the lowest mismatches count (providing the count is small
  or equal to '--mismatches N') 'gets' the sequences.
<div class="Pp"></div>
Example (using the above barcodes): Input Sequence:
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">GATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG</dd>
</dl>
<h2 class="Ss" title="Ss" id="Matching_with_'--bol_--mismatches_1':"><a class="selflink" href="#Matching_with_'--bol_--mismatches_1':">Matching
  with '--bol --mismatches 1':</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">GATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG GATCT (1 mismatch,
      BC1) ATCGT (4 mismatches, BC2) GTGAT (3 mismatches, BC3) TGTCT (3
      mismatches, BC4)</dd>
</dl>
<div class="Pp"></div>
This sequence will be classified as 'BC1' (it has the lowest mismatch count). If
  '--exact' or '--mismatches 0' were specified, this sequence would be
  classified as 'unmatched' (because, although BC1 had the lowest mismatch
  count, it is above the maximum allowed mismatches).
<div class="Pp"></div>
Matching with '--eol' (end of line) does the same, but from the other side of
  the sequence.
<div class="Pp"></div>
** With partial matching (very similar to indels):
<div class="Pp"></div>
Same as above, with the following addition: barcodes are also checked for
  partial overlap (number of allowed non-overlapping bases is '--partial N').
<div class="Pp"></div>
Example: Input sequence is ATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG (Same as above,
  but note the missing 'G' at the beginning.)
<h2 class="Ss" title="Ss" id="Matching_(without_partial_overlapping)_against_BC1_yields_4_mismatches:"><a class="selflink" href="#Matching_(without_partial_overlapping)_against_BC1_yields_4_mismatches:">Matching
  (without partial overlapping) against BC1 yields 4 mismatches:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">ATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG GATCT (4
    mismatches)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Partial_overlapping_would_also_try_the_following_match:"><a class="selflink" href="#Partial_overlapping_would_also_try_the_following_match:">Partial
  overlapping would also try the following match:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-ATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">GATCT (1 mismatch)</dd>
</dl>
<div class="Pp"></div>
Note: scoring counts a missing base as a mismatch, so the final mismatch count
  is 2 (1 'real' mismatch, 1 'missing base' mismatch). If running with
  '--mismatches 2' (meaning allowing upto 2 mismatches) - this seqeunce will be
  classified as BC1.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The quality of this automatically generated manpage might be insufficient. It is
  suggested to visit
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">http://hannonlab.cshl.edu/fastx_toolkit/commandline.html</dd>
</dl>
<div class="Pp"></div>
to get a better layout as well as an overview about connected FASTX tools.</div>
<table class="foot">
  <tr>
    <td class="foot-date">December 2016</td>
    <td class="foot-os">fastx_barcode_splitter.pl 0.0.14</td>
  </tr>
</table>
</body>
</html>
