<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FastQTL(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FastQTL(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">FastQTL(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
FastQTL - Quantitative Trait Loci (QTL) mapper in cis for molecular phenotypes
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>fastQTL [<i>options</i>]</b>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
The goal of FastQTL is to identify single-nucleotide polymorphisms (SNPs) which
  are significantly associated with various molecular phenotypes (i.e.
  expression of known genes, cytosine methylation levels, etc). It performs
  scans for all possible phenotype-variant pairs in cis (i.e. variants located
  within a specific window around a phenotype). FastQTL implements a new
  permutation scheme (Beta approximation) to accurately and rapidly correct for
  multiple-testing at both the genotype and phenotype levels.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Basic_options"><a class="selflink" href="#Basic_options">Basic
  options</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">
    <div>&#x00A0;</div>
    Print help about options.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--silent</b></dt>
  <dd class="It-tag">Silent mode on terminal.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b> arg (=1459232241)</dt>
  <dd class="It-tag">Random number generator seed. Useful to replicate runs of
      the software.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_and_Output_files_options"><a class="selflink" href="#Input_and_Output_files_options">Input
  and Output files options</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L, --log</b></dt>
  <dd class="It-tag">Screen output is copied in this file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V, --vcf</b></dt>
  <dd class="It-tag">Genotypes in VCF format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B, --bed</b></dt>
  <dd class="It-tag">Phenotypes in BED format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C, --cov</b></dt>
  <dd class="It-tag">Covariates in TXT format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-G, --grp</b></dt>
  <dd class="It-tag">Phenotype groups in TXT format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-O, --out</b></dt>
  <dd class="It-tag">Output file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Exclusion_and_Inclusion_files_options"><a class="selflink" href="#Exclusion_and_Inclusion_files_options">Exclusion
  and Inclusion files options</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--exclude-samples</b></dt>
  <dd class="It-tag">List of samples to exclude.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--include-samples</b></dt>
  <dd class="It-tag">List of samples to include.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--exclude-sites</b></dt>
  <dd class="It-tag">List of sites to exclude.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--include-sites</b></dt>
  <dd class="It-tag">List of sites to include.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--exclude-phenotypes</b></dt>
  <dd class="It-tag">List of phenotypes to exclude.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--include-phenotypes</b></dt>
  <dd class="It-tag">List of phenotypes to include.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--exclude-covariates</b></dt>
  <dd class="It-tag">List of covariates to exclude.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--include-covariates</b></dt>
  <dd class="It-tag">List of covariates to include.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Parameters"><a class="selflink" href="#Parameters">Parameters</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--normal</b></dt>
  <dd class="It-tag">To perform quantile normalization on the phenotype
      quantifications to make them normally distributed. Implemented as the
      rntransform function of the GenABEL package.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-W, --window</b></dt>
  <dd class="It-tag">Cis-window size. Default values is 1Mb (1e6). It means that
      all variants within 1e6 bp of the phenotype location (e.g. TSS) is
      analyzed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T, --threshold</b></dt>
  <dd class="It-tag">To filter out all phenotype-variant pairs with a p-value
      above the specified threshold in the output of a nominal pass.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Modes"><a class="selflink" href="#Modes">Modes</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P, --permute</b></dt>
  <dd class="It-tag">Permutation pass to calculate corrected p-values for
      molecular phenotypes.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--psequence</b></dt>
  <dd class="It-tag">Permutation sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--map</b></dt>
  <dd class="It-tag">Map best QTL candidates per molecular phenotype.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--map-full</b></dt>
  <dd class="It-tag">Scan full cis-window to discover independent signals.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--interaction</b></dt>
  <dd class="It-tag">Test for interactions with variable specified in file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h2 class="Ss" title="Ss" id="Parallelization"><a class="selflink" href="#Parallelization">Parallelization</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-K, --chunk</b></dt>
  <dd class="It-tag">Specify which chunk needs to be processed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--commands</b></dt>
  <dd class="It-tag">Generates all commands.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R, --region</b></dt>
  <dd class="It-tag">Region of interest.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLE"><a class="selflink" href="#EXAMPLE">EXAMPLE</a></h1>
for c in $(seq 1 256); do
<div>&#x00A0;</div>
<br/>
 <b>fastQTL</b> <b>--vcf</b> genotypes.vcf.gz
<div>&#x00A0;</div>
<br/>
 <b>--bed</b> phenotypes.bed.gz
<div>&#x00A0;</div>
<br/>
 <b>--chunk</b> $c 256
<div>&#x00A0;</div>
<br/>
 <b>--permute</b> 1000
<div>&#x00A0;</div>
<br/>
 <b>--out</b> results.$c.txt.gz
<div>&#x00A0;</div>
done
<div>&#x00A0;</div>
zcat results.*.txt.gz | gzip -c &gt; fastqtl_expected_output.txt.gz
<div>&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>http://fastqtl.sourceforge.net/</b>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
FastQTL was written by Olivier Delaneau, Halit Ongen, Alfonso Buil and Manolis
  Dermitzakis.
<div class="Pp"></div>
This manual page was written by Dylan A&#x00EF;ssi &lt;bob.dybian@gmail.com&gt;,
  for the Debian project (but may be used by others).</div>
<table class="foot">
  <tr>
    <td class="foot-date">March 2016</td>
    <td class="foot-os">2.184</td>
  </tr>
</table>
</body>
</html>
