<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:56 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTAQ-SEQUENCE_TRIM(1) User Commands
FASTAQ-SEQUENCE_TRIM(1)</p>

<p style="margin-top: 1em">NAME <br>
fastaq_sequence_trim - Trim exact matches to a given string
off the start of every sequence</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: fastaq_sequence_trim [options] &lt;infile_1&gt;
&lt;infile_2&gt; &lt;outfile_1&gt; &lt;outfile_2&gt;
&lt;trim_seqs&gt;</p>

<p style="margin-top: 1em">Trims sequences off the start of
all sequences in a pair of sequence files, whenever there is
a perfect match. Only keeps a read pair if both reads of the
pair are at least a <br>
minimum length after any trimming</p>

<p style="margin-top: 1em">positional arguments: <br>
infile_1 <br>
Name of forward fasta/q file to be trimmed</p>

<p style="margin-top: 1em">infile_2 <br>
Name of reverse fasta/q file to be trimmed</p>

<p style="margin-top: 1em">outfile_1 <br>
Name of output forward fasta/q file</p>

<p style="margin-top: 1em">outfile_2 <br>
Name of output reverse fasta/q file</p>

<p style="margin-top: 1em">trim_seqs <br>
Name of file of sequences to search for at the start of each
input sequence</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">--min_length INT <br>
Minimum length of output sequences [50]</p>

<p style="margin-top: 1em">--revcomp <br>
Trim the end of each sequence if it matches the reverse
complement. This option is intended for PCR primer
trimming</p>

<p style="margin-top: 1em">fastaq 3.14.0 September 2016
FASTAQ-SEQUENCE_TRIM(1)</p>
<hr>
</body>
</html>
