<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FASTML.SH(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FASTML.SH(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">FASTML.SH(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
fastml - maximum likelihood ancestral amino-acid sequence reconstruction
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>fastml</b> <i>[options]</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
FastML is a bioinformatics tool for the reconstruction of ancestral sequences
  based on the phylogenetic relations between homologous sequences. FastML runs
  several algorithms that reconstruct the ancestral sequences with emphasis on
  an accurate reconstruction of both indels and characters. For character
  reconstruction the previously described FastML algorithms are used to
  efficiently infer the most likely ancestral sequences for each internal node
  of the tree. Both joint and the marginal reconstructions are provided. For
  indels reconstruction the sequences are first coded according to the indel
  events detected within the multiple sequence alignment (MSA) and then a
  state-of-the-art likelihood model is used to reconstruct ancestral indels
  states. The results are the most probable sequences, together with posterior
  probabilities for each character and indel at each sequence position for each
  internal node of the tree. FastML is generic and is applicable for any type of
  molecular sequences (nucleotide, protein, or codon sequences).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b> help</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> sequence input file (for example use <b>-s</b>
    emySequences/eseq.txt)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> tree input file (if tree is not given, a neighbor
    joining tree is computed).</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b> Assume among site rate variation model (Gamma)
    [By default the program</dt>
  <dd class="It-tag">will assume an homogeneous model. very fast, but less
      accurate!]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> model name</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mj</b> [JTT]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ml</b> LG</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mr</b> mtREV (for mitochondrial genomes)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-md</b> DAY</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mw</b> WAG</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mc</b> cpREV (for chloroplasts genomes)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ma</b> Jukes and Cantor (JC) for amino acids</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mn</b> Jukes and Cantor (JC) for nucleotides</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mh</b> HKY Model for nucleotides</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mg</b> nucgtr Model for nucleotides</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mt</b> tamura92 Model for nucleotides</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-my</b> yang M5 codons model</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-me</b> empirical codon matrix</dt>
  <dd class="It-tag"></dd>
</dl>
<div class="Pp"></div>
Controling the output options:
<dl class="Bl-tag">
  <dt class="It-tag"><b>-x</b> tree file output in Newick format
    [tree.newick.txt]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-y</b> tree file output in ANCESTOR format
    [tree.ancestor.txt]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-j</b> joint sequences output file [seq.joint.txt]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> marginal sequences output file
    [seq.marginal.txt]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> joint probabilities output file
    [prob.joint.txt]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b> marginal probabilities output file
    [prob.marginal.txt]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> ancestral sequences output format. (<b>-qc</b> =
    [CLUSTAL], <b>-qf</b> = FASTA,</dt>
  <dd class="It-tag"><b>-qm</b> = MOLPHY, <b>-qs</b> = MASE, <b>-qp</b> =
      PHLIYP, <b>-qn</b> = Nexus)</dd>
</dl>
<div class="Pp"></div>
Advanced options:
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> Threshold for computing again marginal
    probabilities [0.9]</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b> Do not optimize branch lengths on starting
    tree</dt>
  <dd class="It-tag">[by default branches and alpha are ML optimized from the
      data]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">number of discrete Gamma categories for the gamma
      distribution [8]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b></dt>
  <dd class="It-tag">don't compute Joint reconstruction (good if the branch and
      bound algorithm takes too much time, and the goal is to compute the
      marginal reconstruction with Gamma).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-z</b></dt>
  <dd class="It-tag">The bound used. <b>-zs</b> - bound based on sum. <b>-zm</b>
      based on max. <b>-zb</b> [both]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b></dt>
  <dd class="It-tag">user alpha parameter of the gamma distribution [if alpha is
      not given, alpha and branches will be evaluated from the data (override
      <b>-b</b>)</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2015</td>
    <td class="foot-os">fastml.sh 3.1</td>
  </tr>
</table>
</body>
</html>
