<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:55 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTAQ-EXPAND_NUCLEOTIDES(1) User Commands
FASTAQ-EXPAND_NUCLEOTIDES(1)</p>

<p style="margin-top: 1em">NAME <br>
fastaq_expand_nucleotides - Makes every combination of
degenerate nucleotides</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: fastaq_expand_nucleotides &lt;infile&gt;
&lt;outfile&gt;</p>

<p style="margin-top: 1em">Makes all combinations of
sequences in input file by using all possibilities of
redundant bases. e.g. ART could be AAT or AGT. Assumes input
is nucleotides, not amino acids</p>

<p style="margin-top: 1em">positional arguments: <br>
infile Name of input file</p>

<p style="margin-top: 1em">outfile <br>
Name of output file</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">fastaq 3.14.0 September 2016
FASTAQ-EXPAND_NUCLEOTIDES(1)</p>
<hr>
</body>
</html>
