<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FASTAQ(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FASTAQ(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">FASTAQ(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
fastaq - FASTA and FASTQ file manipulation tools
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>fastaq</b> <i>&lt;command&gt; </i>[<i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION0nf"><a class="selflink" href="#DESCRIPTION0nf">DESCRIPTION0nf</a></h1>
To get minimal usage for a command use: fastaq command
<div class="Pp"></div>
To get full help for a command use one of: fastaq command <b>-h</b> fastaq
  command <b>--help</b>
<div class="Pp"></div>
Available commands:
<div class="Pp"></div>
acgtn_only Replace every non acgtnACGTN with an N add_indels Deletes or inserts
  bases at given position(s) caf_to_fastq Converts a CAF file to FASTQ format
  capillary_to_pairs Converts file of capillary reads to paired and unpaired
  files chunker Splits sequences into equal sized chunks count_sequences Counts
  the sequences in input file deinterleave Splits interleaved paired file into
  two separate files enumerate_names Renames sequences in a file, calling them
  1,2,3... etc expand_nucleotides Makes every combination of degenerate
  nucleotides fasta_to_fastq Convert FASTA and .qual to FASTQ filter Filter
  sequences to get a subset of them get_ids Get the ID of each sequence
  get_seq_flanking_gaps Gets the sequences flanking gaps interleave Interleaves
  two files, output is alternating between fwd/rev reads make_random_contigs
  Make contigs of random sequence merge Converts multi sequence file to a single
  sequence replace_bases Replaces all occurrences of one letter with another
  reverse_complement Reverse complement all sequences scaffolds_to_contigs
  Creates a file of contigs from a file of scaffolds search_for_seq Find all
  exact matches to a string (and its reverse complement) sequence_trim Trim
  exact matches to a given string off the start of every sequence sort_by_size
  Sorts sequences in length order split_by_base_count Split multi sequence file
  into separate files strip_illumina_suffix Strips <i>/1</i> or <i>/2</i> off
  the end of every read name to_boulderio Converts to Boulder-IO format, used by
  primer3 to_fake_qual Make fake quality scores file to_fasta Converts a variety
  of input formats to nicely formatted FASTA format to_mira_xml Create an xml
  file from a file of reads, for use with Mira assembler to_orfs_gff Writes a
  GFF file of open reading frames to_perfect_reads Make perfect paired reads
  from reference to_random_subset Make a random sample of sequences (and
  optionally mates as well) to_tiling_bam Make a BAM file of reads uniformly
  spread across the input reference to_unique_by_id Remove duplicate sequences,
  based on their names. Keep longest seqs translate Translate all sequences in
  input nucleotide sequences trim_Ns_at_end Trims all Ns at the start/end of all
  sequences trim_contigs Trims a set number of bases off the end of every contig
  trim_ends Trim fixed number of bases of start and/or end of every sequence
  version Print version number and exit</div>
<table class="foot">
  <tr>
    <td class="foot-date">September 2016</td>
    <td class="foot-os">fastaq 3.14.0</td>
  </tr>
</table>
</body>
</html>
