<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>fasttree(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">fasttree(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">fasttree(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
fasttree - create phylogenetic trees from alignments of nucleotide or protein
  sequences
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
fasttree infers approximately-maximum-likelihood phylogenetic trees from
  alignments of nucleotide or protein sequences. It handles alignments with up
  to a million of sequences in a reasonable amount of time and memory.
<div style="height: 1.00em;">&#x00A0;</div>
fasttree is more accurate than PhyML 3 with default settings, and much more
  accurate than the distance-matrix methods that are traditionally used for
  large alignments. fasttree uses the Jukes-Cantor or generalized
  time-reversible (GTR) models of nucleotide evolution and the JTT
  (Jones-Taylor-Thornton 1992) model of amino acid evolution. To account for the
  varying rates of evolution across sites, fasttree uses a single rate for each
  site (the &quot;CAT&quot; approximation). To quickly estimate the reliability
  of each split in the tree, fasttree computes local support values with the
  Shimodaira-Hasegawa test (these are the same as PhyML 3's &quot;SH-like local
  supports&quot;).
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>fasttree protein_alignment &gt; tree</b>
<div class="Pp"></div>
<b>fasttree <b>-nt</b> nucleotide_alignment &gt; tree</b>
<div class="Pp"></div>
<b>fasttree <b>-nt</b> <b>-gtr</b> &lt; nucleotide_alignment &gt; tree</b>
<div class="Pp"></div>
fasttree accepts alignments in fasta or phylip interleaved formats
<h2 class="Ss" title="Ss" id="Common_options_(must_be_before_the_alignment_file):"><a class="selflink" href="#Common_options_(must_be_before_the_alignment_file):">Common
  options (must be before the alignment file):</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-quiet</b>
  to suppress reporting information</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nopr</b>
  to suppress progress indicator</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-log</b>
  logfile <b>--</b> save intermediate trees, settings, and model details</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-fastest</b>
  <b>--</b> speed up the neighbor joining phase &amp; reduce memory usage</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(recommended for &gt;50,000 sequences)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-n</b>
  &lt;number&gt; to analyze multiple alignments (phylip format only)</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(use for global bootstrap, with seqboot and
      CompareToBootstrap.pl)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nosupport</b>
  to not compute support values</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-intree</b>
  newick_file to set the starting tree(s)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-intree1</b>
  newick_file to use this starting tree for all the alignments</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(for faster global bootstrap on huge alignments)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-pseudo</b>
  to use pseudocounts (recommended for highly gapped sequences)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-gtr</b>
  <b>--</b> generalized time-reversible model (nucleotide alignments only)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-wag</b>
  <b>--</b> Whelan-And-Goldman 2001 model (amino acid alignments only)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-quote</b>
  <b>--</b> allow spaces and other restricted characters (but not ' characters)
  in</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">sequence names and quote names in the output tree (fasta
      input only; fasttree will not be able to read these trees back in</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-noml</b>
  to turn off maximum-likelihood</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nome</b>
  to turn off minimum-evolution NNIs and SPRs</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(recommended if running additional ML NNIs with
      <b>-intree</b>)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nome</b>
  <b>-mllen</b> with <b>-intree</b> to optimize branch lengths for a fixed
  topology</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-cat</b> #
  to specify the number of rate categories of sites (default 20)</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">or <b>-nocat</b> to use constant rates</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-gamma</b>
  <b>--</b> after optimizing the tree under the CAT approximation,</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">rescale the lengths to optimize the Gamma20 likelihood</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-constraints</b>
  constraintAlignment to constrain the topology search</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">constraintAlignment should have 1s or 0s to indicates
      splits</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-expert</b>
  <b>--</b> see more options</div>
<h2 class="Ss" title="Ss" id="Detailed_usage_for_fasttree_2.1.4_SSE3:"><a class="selflink" href="#Detailed_usage_for_fasttree_2.1.4_SSE3:">Detailed
  usage for fasttree 2.1.4 SSE3:</a></h2>
fasttree [-nt] [-n 100] [-quote] [-pseudo | <b>-pseudo</b> 1.0]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[-boot 1000 | <b>-nosupport]</b> [-intree
      starting_trees_file | <b>-intree1</b> starting_tree_file] [-quiet |
      <b>-nopr]</b> [-nni 10] [-spr 2] [-noml | <b>-mllen</b> | <b>-mlnni</b>
      10] [-mlacc 2] [-cat 20 | <b>-nocat]</b> [-gamma] [-slow |
      <b>-fastest]</b> [-2nd | <b>-no2nd]</b> [-slownni] [-seed 1253] [-top |
      <b>-notop]</b> [-topm 1.0 [-close 0.75] [-refresh 0.8]] [-matrix Matrix |
      <b>-nomatrix]</b> [-nj | <b>-bionj]</b> [-wag] [-nt] [-gtr] [-gtrrates ac
      ag at cg ct gt] [-gtrfreq A C G T] [ <b>-constraints</b>
      constraintAlignment [ <b>-constraintWeight</b> 100.0 ] ] [-log
    logfile]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[ alignment_file ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&gt; newick_tree</dd>
</dl>
<div class="Pp"></div>
or
<div class="Pp"></div>
fasttree [-nt] [-matrix Matrix | <b>-nomatrix]</b> [-rawdist] <b>-makematrix</b>
  [alignment]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[-n 100] &gt; phylip_distance_matrix</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">fasttree supports fasta or phylip interleaved alignments By
      default fasttree expects protein alignments, use <b>-nt</b> for
      nucleotides fasttree reads standard input if no alignment file is
    given</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input/output_options:"><a class="selflink" href="#Input/output_options:">Input/output
  options:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-n</b>
  <b>--</b> read in multiple alignments in. This only</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">works with phylip interleaved format. For example, you can
      use it with the output from phylip's seqboot. If you use <b>-n</b>,
      fasttree will write 1 tree per line to standard output.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-intree</b>
  newickfile <b>--</b> read the starting tree in from newickfile.</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Any branch lengths in the starting trees are ignored.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-intree</b>
  with <b>-n</b> will read a separate starting tree for each alignment.</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-intree1</b>
  newickfile <b>--</b> read the same starting tree for each alignment</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-quiet</b>
  <b>--</b> do not write to standard error during normal operation (no
  progress</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">indicator, no options summary, no likelihood values,
    etc.)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nopr</b>
  <b>--</b> do not write the progress indicator to stderr</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-log</b>
  logfile <b>--</b> save intermediate trees so you can extract</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">the trees and restart long-running jobs if they crash
      <b>-log</b> also reports the per-site rates (1 means slowest
    category)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-quote</b>
  <b>--</b> quote sequence names in the output and allow spaces, commas,</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">parentheses, and colons in them but not ' characters (fasta
      files only)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Distances:"><a class="selflink" href="#Distances:">Distances:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Default: For protein sequences, log-corrected distances and
      an</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">amino acid dissimilarity matrix derived from BLOSUM45</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">or for nucleotide sequences, Jukes-Cantor distances To
      specify a different matrix, use <b>-matrix</b> FilePrefix or
      <b>-nomatrix</b> Use <b>-rawdist</b> to turn the log-correction off or to
      use %different instead of Jukes-Cantor</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-pseudo</b>
  [weight] <b>--</b> Use pseudocounts to estimate distances between</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">sequences with little or no overlap. (Off by default.)
      Recommended if analyzing the alignment has sequences with little or no
      overlap. If the weight is not specified, it is 1.0</dd>
</dl>
<h2 class="Ss" title="Ss" id="Topology_refinement:"><a class="selflink" href="#Topology_refinement:">Topology
  refinement:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">By default, fasttree tries to improve the tree with up to
      4*log2(N) rounds of minimum-evolution nearest-neighbor interchanges (NNI),
      where N is the number of unique sequences, 2 rounds of
      subtree-prune-regraft (SPR) moves (also min. evo.), and up to 2*log(N)
      rounds of maximum-likelihood NNIs. Use <b>-nni</b> to set the number of
      rounds of min. evo. NNIs, and <b>-spr</b> to set the rounds of SPRs. Use
      <b>-noml</b> to turn off both min-evo NNIs and SPRs (useful if
    refining</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">an approximately maximum-likelihood tree with further
    NNIs)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Use <b>-sprlength</b> set the maximum length of a SPR move
      (default 10) Use <b>-mlnni</b> to set the number of rounds of
      maximum-likelihood NNIs Use <b>-mlacc</b> 2 or <b>-mlacc</b> 3 to always
      optimize all 5 branches at each NNI,</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">and to optimize all 5 branches in 2 or 3 rounds</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Use <b>-mllen</b> to optimize branch lengths without ML
      NNIs Use <b>-mllen</b> <b>-nome</b> with <b>-intree</b> to optimize branch
      lengths on a fixed topology Use <b>-slownni</b> to turn off heuristics to
      avoid constant subtrees (affects both</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">ML and ME NNIs)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Maximum_likelihood_model_options:"><a class="selflink" href="#Maximum_likelihood_model_options:">Maximum
  likelihood model options:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-wag</b>
  <b>--</b> Whelan-And-Goldman 2001 model instead of (default)
  Jones-Taylor-Thorton 1992 model (a.a. only)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-gtr</b>
  <b>--</b> generalized time-reversible instead of (default) Jukes-Cantor (nt
  only)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-cat</b> #
  <b>--</b> specify the number of rate categories of sites (default 20)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nocat</b>
  <b>--</b> no CAT model (just 1 category)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-gamma</b>
  <b>--</b> after the final round of optimizing branch lengths with the CAT
  model,</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">report the likelihood under the discrete gamma model with
      the same number of categories. fasttree uses the same branch lengths but
      optimizes the gamma shape parameter and the scale of the lengths. The
      final tree will have rescaled lengths. Used with <b>-log</b>, this also
      generates per-site likelihoods for use with CONSEL, see GammaLogToPaup.pl
      and documentation on the fasttree web site.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Support_value_options:"><a class="selflink" href="#Support_value_options:">Support
  value options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">By default, fasttree computes local support values by
      resampling the site likelihoods 1,000 times and the Shimodaira Hasegawa
      test. If you specify <b>-nome</b>, it will compute minimum-evolution
      bootstrap supports instead In either case, the support values are
      proportions ranging from 0 to 1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Use <b>-nosupport</b> to turn off support values or
      <b>-boot</b> 100 to use just 100 resamples Use <b>-seed</b> to initialize
      the random number generator</dd>
</dl>
<h2 class="Ss" title="Ss" id="Searching_for_the_best_join:"><a class="selflink" href="#Searching_for_the_best_join:">Searching
  for the best join:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">By default, fasttree combines the 'visible set' of fast
      neighbor-joining with</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">local hill-climbing as in relaxed neighbor-joining</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-slow</b>
  <b>--</b> exhaustive search (like NJ or BIONJ, but different gap
  handling)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-slow</b>
  takes half an hour instead of 8 seconds for 1,250 proteins</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-fastest</b>
  <b>--</b> search the visible set (the top hit for each node) only</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Unlike the original fast neighbor-joining, <b>-fastest</b>
      updates visible(C) after joining A and B if join(AB,C) is better than
      join(C,visible(C)) <b>-fastest</b> also updates out-distances in a very
      lazy way, <b>-fastest</b> sets <b>-2nd</b> on as well, use <b>-fastest</b>
      <b>-no2nd</b> to avoid this</dd>
</dl>
<h2 class="Ss" title="Ss" id="Top-hit_heuristics:"><a class="selflink" href="#Top-hit_heuristics:">Top-hit
  heuristics:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">By default, fasttree uses a top-hit list to speed up search
      Use <b>-notop</b> (or <b>-slow</b>) to turn this feature off</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">and compare all leaves to each other, and all new joined
      nodes to each other</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-topm</b>
  1.0 <b>--</b> set the top-hit list size to parameter*sqrt(N)</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">fasttree estimates the top m hits of a leaf from the top
      2*m hits of a 'close' neighbor, where close is defined as d(seed,close)
      &lt; 0.75 * d(seed, hit of rank 2*m), and updates the top-hits as joins
      proceed</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-close</b>
  0.75 <b>--</b> modify the close heuristic, lower is more conservative</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-refresh</b>
  0.8 <b>--</b> compare a joined node to all other nodes if its</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">top-hit list is less than 80% of the desired length, or if
      the age of the top-hit list is log2(m) or greater</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-2nd</b>
  or <b>-no2nd</b> to turn 2nd-level top hits heuristic on or off</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">This reduces memory usage and running time but may lead to
      marginal reductions in tree quality. (By default, <b>-fastest</b> turns on
      <b>-2nd</b>.)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Join_options:"><a class="selflink" href="#Join_options:">Join
  options:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nj</b>:
  regular (unweighted) neighbor-joining (default)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-bionj</b>:
  weighted joins as in BIONJ</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">fasttree will also weight joins during NNIs</dd>
</dl>
<h2 class="Ss" title="Ss" id="Constrained_topology_search_options:"><a class="selflink" href="#Constrained_topology_search_options:">Constrained
  topology search options:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-constraints</b>
  alignmentfile <b>--</b> an alignment with values of 0, 1, and -</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Not all sequences need be present. A column of 0s and 1s
      defines a constrained split. Some constraints may be violated (see
      'violating constraints:' in standard error).</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-constraintWeight</b>
  <b>--</b> how strongly to weight the constraints. A value of 1</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">means a penalty of 1 in tree length for violating a
      constraint Default: 100.0</dd>
</dl>
<div class="Pp"></div>
For more information, see http://www.microbesonline.org/fasttree/
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">or the comments in the source code</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">fasttree protein_alignment &gt; tree fasttree <b>-nt</b>
      nucleotide_alignment &gt; tree fasttree <b>-nt</b> <b>-gtr</b> &lt;
      nucleotide_alignment &gt; tree</dd>
</dl>
<div class="Pp"></div>
fasttree accepts alignments in fasta or phylip interleaved formats
<h2 class="Ss" title="Ss" id="Common_options_(must_be_before_the_alignment_file):"><a class="selflink" href="#Common_options_(must_be_before_the_alignment_file):">Common
  options (must be before the alignment file):</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-quiet</b>
  to suppress reporting information</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nopr</b>
  to suppress progress indicator</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-log</b>
  logfile <b>--</b> save intermediate trees, settings, and model details</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-fastest</b>
  <b>--</b> speed up the neighbor joining phase &amp; reduce memory usage</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(recommended for &gt;50,000 sequences)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-n</b>
  &lt;number&gt; to analyze multiple alignments (phylip format only)</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(use for global bootstrap, with seqboot and
      CompareToBootstrap.pl)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nosupport</b>
  to not compute support values</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-intree</b>
  newick_file to set the starting tree(s)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-intree1</b>
  newick_file to use this starting tree for all the alignments</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(for faster global bootstrap on huge alignments)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-pseudo</b>
  to use pseudocounts (recommended for highly gapped sequences)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-gtr</b>
  <b>--</b> generalized time-reversible model (nucleotide alignments only)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-wag</b>
  <b>--</b> Whelan-And-Goldman 2001 model (amino acid alignments only)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-quote</b>
  <b>--</b> allow spaces and other restricted characters (but not ' characters)
  in</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">sequence names and quote names in the output tree (fasta
      input only; fasttree will not be able to read these trees back in</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-noml</b>
  to turn off maximum-likelihood</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nome</b>
  to turn off minimum-evolution NNIs and SPRs</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(recommended if running additional ML NNIs with
      <b>-intree</b>)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nome</b>
  <b>-mllen</b> with <b>-intree</b> to optimize branch lengths for a fixed
  topology</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-cat</b> #
  to specify the number of rate categories of sites (default 20)</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">or <b>-nocat</b> to use constant rates</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-gamma</b>
  <b>--</b> after optimizing the tree under the CAT approximation,</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">rescale the lengths to optimize the Gamma20 likelihood</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-constraints</b>
  constraintAlignment to constrain the topology search</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">constraintAlignment should have 1s or 0s to indicates
      splits</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-expert</b>
  <b>--</b> see more options</div>
<div class="Pp"></div>
For more information, see http://www.microbesonline.org/fasttree/</div>
<table class="foot">
  <tr>
    <td class="foot-date">June 2012</td>
    <td class="foot-os">Lawrence Berkeley National Lab</td>
  </tr>
</table>
</body>
</html>
