<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTQC(1) User Commands FASTQC(1)</p>

<p style="margin-top: 1em">NAME <br>
FastQC - high throughput sequence QC analysis tool</p>

<p style="margin-top: 1em">SYNOPSIS</p>

<p style="margin-top: 1em">fastqc seqfile1 seqfile2 ..
seqfileN</p>

<p style="margin-top: 1em">fastqc [-o output dir]
[--(no)extract] [-f fastq|bam|sam]</p>

<p style="margin-top: 1em">[-c contaminant file] seqfile1
.. seqfileN</p>

<p style="margin-top: 1em">DESCRIPTION <br>
FastQC reads a set of sequence files and produces from each
one a quality control report consisting of a number of
different modules, each one of which will help to
iden&acirc; <br>
tify a different potential type of problem in your data.</p>

<p style="margin-top: 1em">If no files to process are
specified on the command line then the program will start as
an interactive graphical application. If files are provided
on the command line <br>
then the program will run with no user interaction required.
In this mode it is suitable for inclusion into a
standardised analysis pipeline.</p>

<p style="margin-top: 1em">The options for the program as
as follows:</p>

<p style="margin-top: 1em">-h --help <br>
Print this help file and exit</p>

<p style="margin-top: 1em">-v --version <br>
Print the version of the program and exit</p>

<p style="margin-top: 1em">-o --outdir <br>
Create all output files in the specified output directory.
Please note that this directory must exist as the program
will not create it. If this option is not set then <br>
the output file for each sequence file is created in the
same directory as the sequence file which was processed.</p>

<p style="margin-top: 1em">--casava <br>
Files come from raw casava output. Files in the same sample
group (differing only by the group number) will be analysed
as a set rather than individually. Sequences with <br>
the filter flag set in the header will be excluded from the
analysis. Files must have the same names given to them by
casava (including being gzipped and ending with .gz) <br>
otherwise they won&rsquo;t be grouped together
correctly.</p>

<p style="margin-top: 1em">--extract <br>
If set then the zipped output file will be uncompressed in
the same directory after it has been created. By default
this option will be set if fastqc is run in non-inter&acirc;
<br>
active mode.</p>

<p style="margin-top: 1em">-j --java <br>
Provides the full path to the java binary you want to use to
launch fastqc. If not supplied then java is assumed to be in
your path.</p>

<p style="margin-top: 1em">--noextract <br>
Do not uncompress the output file after creating it. You
should set this option if you do not wish to uncompress the
output when running in non-interactive mode.</p>

<p style="margin-top: 1em">--nogroup <br>
Disable grouping of bases for reads &gt;50bp. All reports
will show data for every base in the read. WARNING: Using
this option will cause fastqc to crash and burn if you <br>
use it on really long reads, and your plots may end up a
ridiculous size. You have been warned!</p>

<p style="margin-top: 1em">-f --format <br>
Bypasses the normal sequence file format detection and
forces the program to use the specified format. Valid
formats are bam,sam,bam_mapped,sam_mapped and fastq</p>

<p style="margin-top: 1em">-t --threads <br>
Specifies the number of files which can be processed
simultaneously. Each thread will be allocated 250MB of
memory so you shouldn&rsquo;t run more threads than your
available <br>
memory will cope with, and not more than 6 threads on a 32
bit machine</p>

<p style="margin-top: 1em">-c Specifies a non-default file
which contains the list of</p>

<p style="margin-top: 1em">--contaminants <br>
contaminants to screen overrepresented sequences against.
The file must contain sets of named contaminants in the form
name[tab]sequence. Lines prefixed with a hash will <br>
be ignored.</p>

<p style="margin-top: 1em">-k --kmers <br>
Specifies the length of Kmer to look for in the Kmer content
module. Specified Kmer length must be between 2 and 10.
Default length is 5 if not specified.</p>

<p style="margin-top: 1em">-q --quiet <br>
Suppress all progress messages on stdout and only report
errors.</p>

<p style="margin-top: 1em">BUGS <br>
Any bugs in fastqc should be reported either to
simon.andrews@babraham.ac.uk or in
www.bioinformatics.babraham.ac.uk/bugzilla/</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was created using help2man by Andreas Tille
&lt;tille@debian.org&gt; for the Debian distribution but can
be used by others as well.</p>

<p style="margin-top: 1em">FastQC v0.10.1 November 2012
FASTQC(1)</p>
<hr>
</body>
</html>
