<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FASTQC(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FASTQC(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">FASTQC(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
FastQC - high throughput sequence QC analysis tool
<div class="Pp"></div>
SYNOPSIS
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">fastqc seqfile1 seqfile2 .. seqfileN</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">fastqc [-o output dir] [--(no)extract] [-f
    fastq|bam|sam]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[-c contaminant file] seqfile1 .. seqfileN</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">FastQC reads a set of sequence files and produces from each
      one a quality control report consisting of a number of different modules,
      each one of which will help to identify a different potential type of
      problem in your data.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">If no files to process are specified on the command line
      then the program will start as an interactive graphical application. If
      files are provided on the command line then the program will run with no
      user interaction required. In this mode it is suitable for inclusion into
      a standardised analysis pipeline.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">The options for the program as as follows:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b> <b>--help</b></dt>
  <dd class="It-tag">Print this help file and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b> <b>--version</b></dt>
  <dd class="It-tag">Print the version of the program and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> <b>--outdir</b></dt>
  <dd class="It-tag">Create all output files in the specified output directory.
      Please note that this directory must exist as the program will not create
      it. If this option is not set then the output file for each sequence file
      is created in the same directory as the sequence file which was
    processed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--casava</b></dt>
  <dd class="It-tag">Files come from raw casava output. Files in the same sample
      group (differing only by the group number) will be analysed as a set
      rather than individually. Sequences with the filter flag set in the header
      will be excluded from the analysis. Files must have the same names given
      to them by casava (including being gzipped and ending with .gz) otherwise
      they won't be grouped together correctly.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--extract</b></dt>
  <dd class="It-tag">If set then the zipped output file will be uncompressed in
      the same directory after it has been created. By default this option will
      be set if fastqc is run in non-interactive mode.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-j</b> <b>--java</b></dt>
  <dd class="It-tag">Provides the full path to the java binary you want to use
      to launch fastqc. If not supplied then java is assumed to be in your
    path.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noextract</b></dt>
  <dd class="It-tag">Do not uncompress the output file after creating it. You
      should set this option if you do not wish to uncompress the output when
      running in non-interactive mode.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nogroup</b></dt>
  <dd class="It-tag">Disable grouping of bases for reads &gt;50bp. All reports
      will show data for every base in the read. WARNING: Using this option will
      cause fastqc to crash and burn if you use it on really long reads, and
      your plots may end up a ridiculous size. You have been warned!</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> <b>--format</b></dt>
  <dd class="It-tag">Bypasses the normal sequence file format detection and
      forces the program to use the specified format. Valid formats are
      bam,sam,bam_mapped,sam_mapped and fastq</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> <b>--threads</b></dt>
  <dd class="It-tag">Specifies the number of files which can be processed
      simultaneously. Each thread will be allocated 250MB of memory so you
      shouldn't run more threads than your available memory will cope with, and
      not more than 6 threads on a 32 bit machine</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">Specifies a non-default file which contains the list
    of</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--contaminants</b></dt>
  <dd class="It-tag">contaminants to screen overrepresented sequences against.
      The file must contain sets of named contaminants in the form
      name[tab]sequence. Lines prefixed with a hash will be ignored.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> <b>--kmers</b></dt>
  <dd class="It-tag">Specifies the length of Kmer to look for in the Kmer
      content module. Specified Kmer length must be between 2 and 10. Default
      length is 5 if not specified.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> <b>--quiet</b></dt>
  <dd class="It-tag">Suppress all progress messages on stdout and only report
      errors.</dd>
</dl>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Any bugs in fastqc should be reported either to
      simon.andrews@babraham.ac.uk or in
      www.bioinformatics.babraham.ac.uk/bugzilla/</dd>
</dl>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">This manpage was created using <b>help2man</b> by Andreas
      Tille &lt;tille@debian.org&gt; for the Debian distribution but can be used
      by others as well.</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2012</td>
    <td class="foot-os">FastQC v0.10.1</td>
  </tr>
</table>
</body>
</html>
