<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:09:00 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTX_BARCODE_SPLITTER.PL(1) User Commands
FASTX_BARCODE_SPLITTER.PL(1)</p>

<p style="margin-top: 1em">NAME <br>
fastx_barcode_splitter.pl - FASTX Barcode Splitter</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Barcode Splitter, by Assaf Gordon (gordon@cshl.edu),
11sep2008</p>

<p style="margin-top: 1em">This program reads FASTA/FASTQ
file and splits it into several smaller files, Based on
barcode matching. FASTA/FASTQ data is read from STDIN
(format is auto-detected.) Output <br>
files will be writen to disk. Summary will be printed to
STDOUT.</p>

<p style="margin-top: 1em">usage: r.pl --bcfile FILE
--prefix PREFIX [--suffix SUFFIX] [--bol|--eol]</p>

<p style="margin-top: 1em">[--mismatches N] [--exact]
[--partial N] [--help] [--quiet] [--debug]</p>

<p style="margin-top: 1em">Arguments:</p>

<p style="margin-top: 1em">--bcfile FILE - Barcodes file
name. (see explanation below.) --prefix PREFIX - File
prefix. will be added to the output files. Can be used</p>

<p style="margin-top: 1em">to specify output
directories.</p>

<p style="margin-top: 1em">--suffix SUFFIX - File suffix
(optional). Can be used to specify file</p>

<p style="margin-top: 1em">extensions.</p>

<p style="margin-top: 1em">--bol - Try to match barcodes at
the BEGINNING of sequences.</p>

<p style="margin-top: 1em">(What biologists would call the
5&rsquo; end, and programmers would call index 0.)</p>

<p style="margin-top: 1em">--eol - Try to match barcodes at
the END of sequences.</p>

<p style="margin-top: 1em">(What biologists would call the
3&rsquo; end, and programmers would call the end of the
string.) NOTE: one of --bol, --eol must be specified, but
not both.</p>

<p style="margin-top: 1em">--mismatches N - Max. number of
mismatches allowed. default is 1. --exact - Same as
&rsquo;--mismatches 0&rsquo;. If both --exact and
--mismatches</p>

<p style="margin-top: 1em">are specified,
&rsquo;--exact&rsquo; takes precedence.</p>

<p style="margin-top: 1em">--partial N - Allow partial
overlap of barcodes. (see explanation below.)</p>

<p style="margin-top: 1em">(Default is not partial
matching)</p>

<p style="margin-top: 1em">--quiet - Don&rsquo;t print
counts and summary at the end of the run.</p>

<p style="margin-top: 1em">(Default is to print.)</p>

<p style="margin-top: 1em">--debug - Print lots of useless
debug information to STDERR. --help - This helpful help
screen.</p>

<p style="margin-top: 1em">Example (Assuming
&rsquo;s_2_100.txt&rsquo; is a FASTQ file,
&rsquo;mybarcodes.txt&rsquo; is the barcodes file):</p>

<p style="margin-top: 1em">$ cat s_2_100.txt |
fastx_barcode_splitter.pl --bcfile mybarcodes.txt --bol
--mismatches 2 <br>
--prefix /tmp/bla_ --suffix &quot;.txt&quot;</p>

<p style="margin-top: 1em">Barcode file format
------------------- Barcode files are simple text files.
Each line should contain an identifier (descriptive name for
the barcode), and the barcode itself <br>
(A/C/G/T), separated by a TAB character. Example:</p>

<p style="margin-top: 1em">#This line is a comment (starts
with a &rsquo;number&rsquo; sign) BC1 GATCT BC2 ATCGT BC3
GTGAT BC4 TGTCT</p>

<p style="margin-top: 1em">For each barcode, a new FASTQ
file will be created (with the barcode&rsquo;s identifier as
part of the file name). Sequences matching the barcode will
be stored in the appropriate <br>
file.</p>

<p style="margin-top: 1em">Running the above example
(assuming &quot;mybarcodes.txt&quot; contains the above
barcodes), will create the following files:</p>

<p style="margin-top: 1em">/tmp/bla_BC1.txt
/tmp/bla_BC2.txt /tmp/bla_BC3.txt /tmp/bla_BC4.txt
/tmp/bla_unmatched.txt</p>

<p style="margin-top: 1em">The &rsquo;unmatched&rsquo; file
will contain all sequences that didn&rsquo;t match any
barcode.</p>

<p style="margin-top: 1em">Barcode matching
----------------</p>

<p style="margin-top: 1em">** Without partial matching:</p>

<p style="margin-top: 1em">Count mismatches between the
FASTA/Q sequences and the barcodes. The barcode which
matched with the lowest mismatches count (providing the
count is small or equal to &rsquo;--mis&acirc; <br>
matches N&rsquo;) &rsquo;gets&rsquo; the sequences.</p>

<p style="margin-top: 1em">Example (using the above
barcodes): Input Sequence:</p>


<p style="margin-top: 1em">GATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG</p>

<p style="margin-top: 1em">Matching with &rsquo;--bol
--mismatches 1&rsquo;: <br>
GATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG GATCT (1 mismatch, BC1)
ATCGT (4 mismatches, BC2) GTGAT (3 mismatches, BC3) TGTCT (3
mismatches, BC4)</p>

<p style="margin-top: 1em">This sequence will be classified
as &rsquo;BC1&rsquo; (it has the lowest mismatch count). If
&rsquo;--exact&rsquo; or &rsquo;--mismatches 0&rsquo; were
specified, this sequence would be classified as
&rsquo;unmatched&rsquo; <br>
(because, although BC1 had the lowest mismatch count, it is
above the maximum allowed mismatches).</p>

<p style="margin-top: 1em">Matching with
&rsquo;--eol&rsquo; (end of line) does the same, but from
the other side of the sequence.</p>

<p style="margin-top: 1em">** With partial matching (very
similar to indels):</p>

<p style="margin-top: 1em">Same as above, with the
following addition: barcodes are also checked for partial
overlap (number of allowed non-overlapping bases is
&rsquo;--partial N&rsquo;).</p>

<p style="margin-top: 1em">Example: Input sequence is
ATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG (Same as above, but note
the missing &rsquo;G&rsquo; at the beginning.)</p>

<p style="margin-top: 1em">Matching (without partial
overlapping) against BC1 yields 4 mismatches: <br>
ATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG GATCT (4 mismatches)</p>

<p style="margin-top: 1em">Partial overlapping would also
try the following match: <br>
-ATTTACTATGTAAAGATAGAAGGAATAAGGTGAAG</p>

<p style="margin-top: 1em">GATCT (1 mismatch)</p>

<p style="margin-top: 1em">Note: scoring counts a missing
base as a mismatch, so the final mismatch count is 2 (1
&rsquo;real&rsquo; mismatch, 1 &rsquo;missing base&rsquo;
mismatch). If running with &rsquo;--mismatches 2&rsquo;
(meaning <br>
allowing upto 2 mismatches) - this seqeunce will be
classified as BC1.</p>

<p style="margin-top: 1em">SEE ALSO <br>
The quality of this automatically generated manpage might be
insufficient. It is suggested to visit</p>


<p style="margin-top: 1em">http://hannonlab.cshl.edu/fastx_toolkit/commandline.html</p>

<p style="margin-top: 1em">to get a better layout as well
as an overview about connected FASTX tools.</p>

<p style="margin-top: 1em">fastx_barcode_splitter.pl 0.0.14
December 2016 FASTX_BARCODE_SPLITTER.PL(1)</p>
<hr>
</body>
</html>
