<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:58 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTBLOCKSEARCH(1) FASTBLOCKSEARCH(1)</p>

<p style="margin-top: 1em">NAME <br>
fastblocksearch - search loci matching protein block
profiles</p>

<p style="margin-top: 1em">SYNOPSIS <br>
fastBlockSearch [options] seqs.fa fam.prfl</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Searches hits (matches) of the blocks in the profile given
by fam.prfl within the genomic sequences in the file
seqs.fa. <br>
Hits are sorted increasingly by score, so the last displayed
hit is the best one found in the region. The format is
similar <br>
to that of the blockscore file (which is optionally
generated by msa2prfl.pl): It shows coordinate, strand, <br>
mean odds-ratio score, and specificity of score, and the
motif. <br>
From the output users can chose regions with matching blocks
to perform gene prediction with AUGUSTUS-PPX using the same
block profile.</p>

<p style="margin-top: 1em">EXAMPLE <br>
&gt; fastBlockSearch --cutoff=1.1 chr4.103M.fa
PF00225_seed.prfl</p>

<p style="margin-top: 1em">Hits found in chr4 103000000
105000000 <br>
Score:207.987 <br>
Mult. score:4.83391 <br>
1081586 unknown_M[5,13] - 2.32574 5.04633 .....YATRLKNI <br>
1103952 unknown_L - 4.85363 6.75245 NAKTRIICTITP <br>
1103991 unknown_K - 8.38065 9.92928 YRDSKLTRILQNSLG <br>
1104375 unknown_J - 3.96065 6.79408 RSLFILGQVIKKL <br>
1106992 unknown_I - 9.22487 7.64306 LVDLAGSE <br>
1115567 unknown_H[5,16] - 2.31869 5.58986 .....ESRHYGETKMN
<br>
1116319 unknown_G - 7.34282 8.29425 EIYNETITDLL <br>
1117092 unknown_F - 5.10694 6.10274 VIPRAIHDIF <br>
1117146 unknown_E - 9.43596 9.18891 QTASGKTYTM <br>
1117176 unknown_D[1,8] - 5.73796 6.31532 .GTIFAYG <br>
1117399 unknown_B[1,7] - 3.59083 5.03059 .CLDRVF <br>
1119420 unknown_A[0,8] - 4.64107 6.44285 RVRPLNSR.</p>

<p style="margin-top: 1em">OPTIONS <br>
--cutoff=c <br>
This minimum for the average log score of the motifs found
can be used to adjust the sensitivity of the block search.
The standard cutoff is 0.7, which is very sensitive but <br>
can give many false positive hits for smaller profiles.</p>

<p style="margin-top: 1em">AUTHORS <br>
Oliver Keller</p>
 
<p style="margin-top: 1em">FASTBLOCKSEARCH(1)</p>
<hr>
</body>
</html>
