<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FASTAQ-SEQUENCE_TRIM(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FASTAQ-SEQUENCE_TRIM(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">FASTAQ-SEQUENCE_TRIM(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
fastaq_sequence_trim - Trim exact matches to a given string off the start of
  every sequence
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
usage: fastaq_sequence_trim [options] &lt;infile_1&gt; &lt;infile_2&gt;
  &lt;outfile_1&gt; &lt;outfile_2&gt; &lt;trim_seqs&gt;
<div class="Pp"></div>
Trims sequences off the start of all sequences in a pair of sequence files,
  whenever there is a perfect match. Only keeps a read pair if both reads of the
  pair are at least a minimum length after any trimming
<h2 class="Ss" title="Ss" id="positional_arguments:"><a class="selflink" href="#positional_arguments:">positional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">infile_1</dt>
  <dd class="It-tag">Name of forward fasta/q file to be trimmed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">infile_2</dt>
  <dd class="It-tag">Name of reverse fasta/q file to be trimmed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">outfile_1</dt>
  <dd class="It-tag">Name of output forward fasta/q file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">outfile_2</dt>
  <dd class="It-tag">Name of output reverse fasta/q file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">trim_seqs</dt>
  <dd class="It-tag">Name of file of sequences to search for at the start of
      each input sequence</dd>
</dl>
<h2 class="Ss" title="Ss" id="optional_arguments:"><a class="selflink" href="#optional_arguments:">optional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_length</b> INT</dt>
  <dd class="It-tag">Minimum length of output sequences [50]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--revcomp</b></dt>
  <dd class="It-tag">Trim the end of each sequence if it matches the reverse
      complement. This option is intended for PCR primer trimming</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">September 2016</td>
    <td class="foot-os">fastaq 3.14.0</td>
  </tr>
</table>
</body>
</html>
