<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:08:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FASTML.SH(1) User Commands FASTML.SH(1)</p>

<p style="margin-top: 1em">NAME <br>
fastml - maximum likelihood ancestral amino-acid sequence
reconstruction</p>

<p style="margin-top: 1em">SYNOPSIS <br>
fastml [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
FastML is a bioinformatics tool for the reconstruction of
ancestral sequences based on the phylogenetic relations
between homologous sequences. FastML runs several algorithms
<br>
that reconstruct the ancestral sequences with emphasis on an
accurate reconstruction of both indels and characters. For
character reconstruction the previously described FastML
<br>
algorithms are used to efficiently infer the most likely
ancestral sequences for each internal node of the tree. Both
joint and the marginal reconstructions are provided. For
<br>
indels reconstruction the sequences are first coded
according to the indel events detected within the multiple
sequence alignment (MSA) and then a state-of-the-art
likelihood <br>
model is used to reconstruct ancestral indels states. The
results are the most probable sequences, together with
posterior probabilities for each character and indel at each
<br>
sequence position for each internal node of the tree. FastML
is generic and is applicable for any type of molecular
sequences (nucleotide, protein, or codon sequences).</p>

<p style="margin-top: 1em">OPTIONS <br>
-h help</p>

<p style="margin-top: 1em">-s sequence input file (for
example use -s emySequences/eseq.txt)</p>

<p style="margin-top: 1em">-t tree input file (if tree is
not given, a neighbor joining tree is computed).</p>

<p style="margin-top: 1em">-g Assume among site rate
variation model (Gamma) [By default the program <br>
will assume an homogeneous model. very fast, but less
accurate!]</p>

<p style="margin-top: 1em">-m model name</p>

<p style="margin-top: 1em">-mj [JTT]</p>

<p style="margin-top: 1em">-ml LG</p>

<p style="margin-top: 1em">-mr mtREV (for mitochondrial
genomes)</p>

<p style="margin-top: 1em">-md DAY</p>

<p style="margin-top: 1em">-mw WAG</p>

<p style="margin-top: 1em">-mc cpREV (for chloroplasts
genomes)</p>

<p style="margin-top: 1em">-ma Jukes and Cantor (JC) for
amino acids</p>

<p style="margin-top: 1em">-mn Jukes and Cantor (JC) for
nucleotides</p>

<p style="margin-top: 1em">-mh HKY Model for
nucleotides</p>

<p style="margin-top: 1em">-mg nucgtr Model for
nucleotides</p>

<p style="margin-top: 1em">-mt tamura92 Model for
nucleotides</p>

<p style="margin-top: 1em">-my yang M5 codons model</p>

<p style="margin-top: 1em">-me empirical codon matrix</p>

<p style="margin-top: 1em">Controling the output
options:</p>

<p style="margin-top: 1em">-x tree file output in Newick
format [tree.newick.txt]</p>

<p style="margin-top: 1em">-y tree file output in ANCESTOR
format [tree.ancestor.txt]</p>

<p style="margin-top: 1em">-j joint sequences output file
[seq.joint.txt]</p>

<p style="margin-top: 1em">-k marginal sequences output
file [seq.marginal.txt]</p>

<p style="margin-top: 1em">-d joint probabilities output
file [prob.joint.txt]</p>

<p style="margin-top: 1em">-e marginal probabilities output
file [prob.marginal.txt]</p>

<p style="margin-top: 1em">-q ancestral sequences output
format. (-qc = [CLUSTAL], -qf = FASTA, <br>
-qm = MOLPHY, -qs = MASE, -qp = PHLIYP, -qn = Nexus)</p>

<p style="margin-top: 1em">Advanced options:</p>

<p style="margin-top: 1em">-a Threshold for computing again
marginal probabilities [0.9]</p>

<p style="margin-top: 1em">-b Do not optimize branch
lengths on starting tree <br>
[by default branches and alpha are ML optimized from the
data]</p>

<p style="margin-top: 1em">-c number of discrete Gamma
categories for the gamma distribution [8]</p>

<p style="margin-top: 1em">-f don&rsquo;t compute Joint
reconstruction (good if the branch and bound algorithm takes
too much time, and the goal is to compute the marginal
reconstruction with Gamma).</p>

<p style="margin-top: 1em">-z The bound used. -zs - bound
based on sum. -zm based on max. -zb [both]</p>

<p style="margin-top: 1em">-p user alpha parameter of the
gamma distribution [if alpha is not given, alpha and
branches will be evaluated from the data (override -b)</p>

<p style="margin-top: 1em">fastml.sh 3.1 November 2015
FASTML.SH(1)</p>
<hr>
</body>
</html>
