<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FASTX_CLIPPER(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FASTX_CLIPPER(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">FASTX_CLIPPER(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
fastx_clipper - FASTA/Q Clipper
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
usage: fastx_clipper [-h] [-a ADAPTER] [-D] [-l N] [-n] [-d N] [-c] [-C] [-o]
  [-v] [-z] [-i INFILE] [-o OUTFILE] Part of FASTX Toolkit 0.0.14 by A. Gordon
  (assafgordon@gmail.com)
<dl class="Bl-tag">
  <dt class="It-tag">[-h]</dt>
  <dd class="It-tag">= This helpful help screen.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[-a ADAPTER] = ADAPTER string. default is CCTTAAGG (dummy
      adapter). [-l N] = discard sequences shorter than N nucleotides. default
      is 5. [-d N] = Keep the adapter and N bases after it.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(using '-d 0' is the same as not using '-d' at all. which
      is the default).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">[-c]</dt>
  <dd class="It-tag">= Discard non-clipped sequences (i.e. - keep only sequences
      which contained the adapter).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">[-C]</dt>
  <dd class="It-tag">= Discard clipped sequences (i.e. - keep only sequences
      which did not contained the adapter).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">[-k]</dt>
  <dd class="It-tag">= Report Adapter-Only sequences.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">[-n]</dt>
  <dd class="It-tag">= keep sequences with unknown (N) nucleotides. default is
      to discard such sequences.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">[-v]</dt>
  <dd class="It-tag">= Verbose - report number of sequences.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">If [-o] is specified,</dt>
  <dd class="It-tag">report will be printed to STDOUT.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">If [-o] is not specified (and output goes to STDOUT),
      report will be printed to STDERR.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">[-z]</dt>
  <dd class="It-tag">= Compress output with GZIP.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">[-D]</dt>
  <dd class="It-tag">= DEBUG output.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">[-M N]</dt>
  <dd class="It-tag">= require minimum adapter alignment length of N.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">If less than N nucleotides aligned with the adapter - don't
    clip it.</dt>
  <dd class="It-tag">[-i INFILE] = FASTA/Q input file. default is STDIN.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[-o OUTFILE] = FASTA/Q output file. default is STDOUT.</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The quality of this automatically generated manpage might be insufficient. It is
  suggested to visit
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">http://hannonlab.cshl.edu/fastx_toolkit/commandline.html</dd>
</dl>
<div class="Pp"></div>
to get a better layout as well as an overview about connected FASTX tools.</div>
<table class="foot">
  <tr>
    <td class="foot-date">December 2016</td>
    <td class="foot-os">fastx_clipper 0.0.14</td>
  </tr>
</table>
</body>
</html>
