<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:09:20 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FEATURE_DRAW(1p) User Contributed Perl Documentation
FEATURE_DRAW(1p)</p>

<p style="margin-top: 1em">NAME <br>
feature_draw.pl -- Render a Bio::Graphics Feature File</p>

<p style="margin-top: 1em">SYNOPSIS <br>
feature_draw.pl [options] file.txt [file2.txt...] &gt;
rendering.png <br>
feature_draw.pl [options] file.txt [file2.txt...] | display
-</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The feature_draw.pl script is a thin front end around the
Bio::Graphics module. It accepts a list of files containing
sequence (protein, nucleotide) feature coordinates from the
<br>
file(s) listed on the command line or on standard input,
renders them, and produces a PNG file on standard
output.</p>

<p style="margin-top: 1em">Options <br>
This script uses GNU-style long options. This allows you to
specify the image width option, for example, with any of the
following alternative forms:</p>

<p style="margin-top: 1em">--width=800 <br>
--width 800 <br>
-width 800 <br>
-w 800</p>

<p style="margin-top: 1em">--width <br>
This sets the width of the image, in pixels. The default is
800 pixels.</p>

<p style="margin-top: 1em">--range <br>
This sets the range of the region displayed, in base pairs
from start to stop. Any of the following formats are
accepted:</p>

<p style="margin-top: 1em">--range 1..1000 <br>
--range 1,1000 <br>
--range 1-1000</p>

<p style="margin-top: 1em">Negative ranges are allowed.</p>

<p style="margin-top: 1em">Feature Files Format <br>
This script accepts and processes sequence annotations in a
simple tab-delimited format or in GFF format.</p>

<p style="margin-top: 1em">The feature file format has a
configuration section and a data section. The configuration
section sets up the size and overall properties of the
image, and the data section gives <br>
the feature data itself.</p>

<p style="margin-top: 1em">Configuration Section <br>
If not provided, this scripts generates a reasonable default
configuration section for you, so you do not need to provide
a configuration section to get a reasonable image. <br>
However, to tune the appearance of the image, you will
probably want to tweak the configuration. Here is an excerpt
from the configuration section:</p>

<p style="margin-top: 1em"># example file <br>
[general] <br>
bases = -1000..21000 <br>
height = 12</p>

<p style="margin-top: 1em">[EST] <br>
glyph = segments <br>
bgcolor= yellow <br>
connector = dashed <br>
height = 5</p>

<p style="margin-top: 1em">[FGENES] <br>
glyph = transcript2 <br>
bgcolor = green <br>
description = 1</p>

<p style="margin-top: 1em">The configuration section is
divided into a set of sections, each one labeled with a
[section title]. The [general] section specifies global
options for the entire image. Other <br>
sections apply to particular feature types. In the example
above, the configuration in the [EST] section applies to
features labeled as ESTs, while the configuration in the
<br>
[FGENES] section applies to features labeled as predictions
from the FGENES gene prediction program.</p>

<p style="margin-top: 1em">Inside each section is a series
of name=value pairs, where the name is the name of an option
to set. You can put whitespace around the = sign to make it
more readable, or even <br>
use a colon (:) if you prefer. The following option names
are recognized:</p>

<p style="margin-top: 1em">Option Value Example <br>
------ ----- -------</p>

<p style="margin-top: 1em">bases Min &amp; max of the
sequence range (bp) 1200..60000 <br>
width width of the image (pixels) 600 <br>
height Height of each graphical element (pixels) 10 <br>
glyph Style of each graphical element (see below) transcript
<br>
fgcolor Foreground color of each element yellow <br>
bgcolor Background color of each element blue <br>
linewidth Width of lines 3 <br>
label Print the feature&rsquo;s name 1 <br>
description Whether to print the feature&rsquo;s description
0 <br>
bump Elements are not allowed to collide 1 <br>
ticks Print tick marks on arrows 1 <br>
connector Type of group connector (dashed, hat or solid)
dashed</p>

<p style="margin-top: 1em">The &quot;bases&quot; and
&quot;width&quot; options are only relevant in the [general]
section. They are overridden by the like-named command-line
options.</p>

<p style="margin-top: 1em">The remainder of the options can
be located in any section, but if present in the [general]
section will set defaults for the others.</p>

<p style="margin-top: 1em">Colors are English-language
color names or Web-style #RRGGBB colors (see a book on HTML
for an explanation). True/false values are 1 for true, and 0
for false. Numeric ranges can <br>
be expressed in start..end fashion with two dots, or as
start-end with a hyphen.</p>

<p style="margin-top: 1em">The &quot;glyph&quot; option
controls how the features are rendered. The following glyphs
are implemented:</p>

<p style="margin-top: 1em">Name Description <br>
---- -----------</p>

<p style="margin-top: 1em">box A filled rectangle,
nondirectional. <br>
ellipse An oval. <br>
arrow An arrow; can be unidirectional or <br>
bidirectional. It is also capable of displaying <br>
a scale with major and minor tickmarks, and can <br>
be oriented horizontally or vertically. <br>
segments A set of filled rectangles connected by solid <br>
lines. Used for interrupted features, such as <br>
gapped alignments and exon groups. <br>
transcript Similar to segments, but the connecting line is
<br>
a &quot;hat&quot; shape, and the direction of <br>
transcription is indicated by a small arrow. <br>
transcript2 Similar to transcript, but the direction of <br>
transcription is indicated by a terminal segment <br>
in the shape of an arrow. <br>
primers Two inward pointing arrows connected by a line. Used
for STSs.</p>

<p style="margin-top: 1em">The bump option is the most
important option for controlling the look of the image. If
set to false (the number 0), then the features are allowed
to overlap. If set to true (the <br>
number 1), then the features will move vertically to avoid
colliding. If not specified, bump is turned on if the number
of any given type of sequence feature is greater than
50.</p>

<p style="margin-top: 1em">Data Section <br>
The data section can follow or proceed the configuration
section. The two sections can also be intermixed. The data
section is a tab or whitespace-delimited file which you can
<br>
export from a spreadsheet application or word processor file
(be sure to save as text only!)</p>

<p style="margin-top: 1em">Here is an example data
section:</p>

<p style="margin-top: 1em">Cosmid B0511 . 516-619 Cosmid
B0511 . 3185-3294 Cosmid B0511 . 10946-11208 Cosmid B0511 .
13126-13511 Cosmid <br>
B0511 . 66-208 Cosmid B0511 . 6354-6499 Cosmid B0511 .
13955-14115 EST yk595e6.5 + 3187-3294 EST yk846e07.3 <br>
- 11015-11208 EST yk53c10 <br>
yk53c10.5 + 18892-19154 <br>
yk53c10.3 - 15000-15500,15700-15800 EST yk53c10.5 +
16032-16105 SwissProt PECANEX + 13153-13656 Swedish fish
FGENESH &quot;Gene <br>
1&quot; - 1-205,518-616,661-735,3187-3365,3436-3846
Transmembrane domain FGENESH &quot;Gene 2&quot; -
16626-17396,17451-17597 Kinase and sushi domains</p>

<p style="margin-top: 1em">Each line of the file contains
five columns. The columns are:</p>

<p style="margin-top: 1em">Column # Description <br>
-------- -----------</p>

<p style="margin-top: 1em">1 feature type <br>
2 feature name <br>
3 strand <br>
4 coordinates <br>
5 description</p>

<p style="margin-top: 1em">Feature type <br>
The feature type should correspond to one of the [feature
type] headings in the configuration section. If it
doesn&rsquo;t, the [general] options will be applied to the
feature <br>
when rendering it. The feature name is a name for the
feature. Use a &quot;.&quot; or &quot;-&quot; if this is not
relevant. If the name contains whitespace, put single or
double quotes (&quot;&quot;) <br>
around the name.</p>

<p style="margin-top: 1em">Strand <br>
The strand indicates which strand the feature is on. It is
one of &quot;+&quot; for the forward strand, &quot;-&quot;
for the reverse strand, or &quot;.&quot; for features that
are not stranded.</p>

<p style="margin-top: 1em">Coordinates <br>
The coordinates column is a set of one or more ranges that
the feature occupies. Ranges are written using
&quot;..&quot; as in start..stop, or with hyphens, as in
start-stop. For <br>
features that are composed of multiple ranges &amp;em; for
example transcripts that have multiple exons &amp;em; you
can either put the ranges on the same line separated by
commas or <br>
spaces, or put the ranges on individual lines and just use
the same feature name and type to group them. In the example
above, the Cosmid B0511 features use the individual <br>
line style, while the FGENESH features use the
all-ranges-on-one-line style.</p>

<p style="margin-top: 1em">Description <br>
The last column contains some descriptive text. If the
description option is set to true, this text will be printed
underneath the feature in the rendering.</p>

<p style="margin-top: 1em">Finally, it is possible to group
related features together. An example is the ESTs yk53c10.5
and yk53c10.3, which are related by being reads from the two
ends of the clone <br>
yk53c10. To indicate this relationship, generate a section
that looks like this:</p>

<p style="margin-top: 1em">EST yk53c10 <br>
yk53c10.5 + 18892-19154 <br>
yk53c10.3 - 15000-15500,15700-15800</p>

<p style="margin-top: 1em">The group is indicated by a line
that contains just two columns containing the feature type
and a unique name for the group. Follow this line with all
the features that form the <br>
group, but leave the first column (the feature type) blank.
The group will be rendered by drawing a dashed line between
all the members of the group. You can change this by <br>
specifying a different connector option in the configuration
section for this feature type.</p>

<p style="margin-top: 1em">BUGS <br>
Please report them to the author.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Bio::Graphics</p>

<p style="margin-top: 1em">AUTHOR <br>
Lincoln Stein, lstein@cshl.org</p>

<p style="margin-top: 1em">perl v5.24.1 2016-12-20
FEATURE_DRAW(1p)</p>
<hr>
</body>
</html>
