<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FILTERBAM(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FILTERBAM(1)</td>
    <td class="head-vol">&#x00A0;</td>
    <td class="head-rtitle">FILTERBAM(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
filterbam - filter BAM file for use with AUGUSTUS tools
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>filterBam</b> --in in.bam --out out.bam [options]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
The input file must be sorted lexicographically by 'queryname', with e.g.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;<b>sort -k 1,1</b> [be aware: 'export
  LC_ALL=C' might be used because sort ignores characters like ':'] Also, please
  bear in mind that this will require converting your BAM file into SAM.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;<b>samtools</b> and <b>bamtools</b>
  provide facilities to do the sorting, but they are not guaranteed to work
  because of the problem mentioned above.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;In the case of <b>samtools</b>, the
  command is: 'samtools sort [-n] file.bam'. The option [-n] should sort by
  query name, just as 'sort -k 10,10' would do in a PSL file. Without options,
  the sorting will be done by reference name and target coordinate, just as a
  'sort -n -k 16,16 | sort -k 14,14' would do with PSL. For more information
  check the man page included in samtools distribution.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;bamtools can also sort bam files:
  <b>bamtools sort -queryname -in file.bam</b>, but only provides the option to
  do it by queryname.</div>
<div style="height: 1.00em;">&#x00A0;</div>
If the option 'paired' is used, then alignment names must include suffixes /1,/2
  or /f,/r.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<b>--best</b>
<div style="margin-left: 4.00ex;">output all best matches that satisfy minId and
  minCover (default 0)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--noIntrons</b>
<div style="margin-left: 4.00ex;">do not allow longer gaps -for RNA-RNA
  alignments- (default 0)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--paired</b>
<div style="margin-left: 4.00ex;">require that paired reads are on opposite
  strands of same target (default 0). NOTE: see prerequisite section
  above.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--uniq</b>
<div style="margin-left: 4.00ex;">take only best match, iff, second best is much
  worse (default 0)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--verbose</b>
<div style="margin-left: 4.00ex;">output debugging info (default 0)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--insertLimit n</b>
<div style="margin-left: 4.00ex;">maximum assumed size of inserts (default
  10)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--maxIntronLen n</b>
<div style="margin-left: 4.00ex;">maximal separation of paired reads (default
  500000)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--maxSortesTest n</b>
<div style="margin-left: 4.00ex;">maximal sortedness (default 100000)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--minCover n</b>
<div style="margin-left: 4.00ex;">minimal percentage of coverage of the query
  read (default 80)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--minId n</b>
<div style="margin-left: 4.00ex;">minimal percentage of identity (default
  92)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--minIntronLen n</b>
<div style="margin-left: 4.00ex;">minimal intron length (default 35)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--uniqThresh n</b>
<div style="margin-left: 4.00ex;">threshold % for uniq, second best must be at
  most this fraction of best (default 0.96)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--commonGeneFile s</b>
<div style="margin-left: 4.00ex;">file name in which to write cases where one
  read maps several different genes</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--pairBedFile s</b>
<div style="margin-left: 4.00ex;">file name of pairedness coverage: a BED format
  file in which for each position the number of filtered read pairs is reported
  that contain the position in or between the reads</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--pairwiseAlignments</b>
<div style="margin-left: 4.00ex;">use in case alignments were done in pairwise
  fashion (default: 0)</div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
AUGUSTUS was written by M. Stanke, O. Keller, S. K&#x00F6;nig, L. Gerischer and
  L. Romoth.
<h1 class="Sh" title="Sh" id="ADDITIONAL_DOCUMENTATION"><a class="selflink" href="#ADDITIONAL_DOCUMENTATION">ADDITIONAL
  DOCUMENTATION</a></h1>
An exhaustive documentation can be found in the file
  /usr/share/augustus/README.TXT.</div>
<table class="foot">
  <tr>
    <td class="foot-date"></td>
    <td class="foot-os">&#x00A0;</td>
  </tr>
</table>
</body>
</html>
