<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:09:57 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FILTERBAM(1) FILTERBAM(1)</p>

<p style="margin-top: 1em">NAME <br>
filterbam - filter BAM file for use with AUGUSTUS tools</p>

<p style="margin-top: 1em">SYNOPSIS <br>
filterBam --in in.bam --out out.bam [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The input file must be sorted lexicographically by
&rsquo;queryname&rsquo;, with e.g.</p>

<p style="margin-top: 1em">&Acirc;&middot; sort -k 1,1 [be
aware: &rsquo;export LC_ALL=C&rsquo; might be used because
sort ignores characters like &rsquo;:&rsquo;] Also, please
bear in mind that this will require converting your BAM file
<br>
into SAM.</p>

<p style="margin-top: 1em">&Acirc;&middot; samtools and
bamtools provide facilities to do the sorting, but they are
not guaranteed to work because of the problem mentioned
above.</p>

<p style="margin-top: 1em">&Acirc;&middot; In the case of
samtools, the command is: &rsquo;samtools sort [-n]
file.bam&rsquo;. The option [-n] should sort by query name,
just as &rsquo;sort -k 10,10&rsquo; would do in a PSL file.
Without <br>
options, the sorting will be done by reference name and
target coordinate, just as a &rsquo;sort -n -k 16,16 | sort
-k 14,14&rsquo; would do with PSL. For more information
check the man <br>
page included in samtools distribution.</p>

<p style="margin-top: 1em">&Acirc;&middot; bamtools can
also sort bam files: bamtools sort -queryname -in file.bam,
but only provides the option to do it by queryname.</p>

<p style="margin-top: 1em">If the option
&rsquo;paired&rsquo; is used, then alignment names must
include suffixes /1,/2 or /f,/r.</p>

<p style="margin-top: 1em">OPTIONS <br>
--best <br>
output all best matches that satisfy minId and minCover
(default 0)</p>

<p style="margin-top: 1em">--noIntrons <br>
do not allow longer gaps -for RNA-RNA alignments- (default
0)</p>

<p style="margin-top: 1em">--paired <br>
require that paired reads are on opposite strands of same
target (default 0). NOTE: see prerequisite section
above.</p>

<p style="margin-top: 1em">--uniq <br>
take only best match, iff, second best is much worse
(default 0)</p>

<p style="margin-top: 1em">--verbose <br>
output debugging info (default 0)</p>

<p style="margin-top: 1em">--insertLimit n <br>
maximum assumed size of inserts (default 10)</p>

<p style="margin-top: 1em">--maxIntronLen n <br>
maximal separation of paired reads (default 500000)</p>

<p style="margin-top: 1em">--maxSortesTest n <br>
maximal sortedness (default 100000)</p>

<p style="margin-top: 1em">--minCover n <br>
minimal percentage of coverage of the query read (default
80)</p>

<p style="margin-top: 1em">--minId n <br>
minimal percentage of identity (default 92)</p>

<p style="margin-top: 1em">--minIntronLen n <br>
minimal intron length (default 35)</p>

<p style="margin-top: 1em">--uniqThresh n <br>
threshold % for uniq, second best must be at most this
fraction of best (default 0.96)</p>

<p style="margin-top: 1em">--commonGeneFile s <br>
file name in which to write cases where one read maps
several different genes</p>

<p style="margin-top: 1em">--pairBedFile s <br>
file name of pairedness coverage: a BED format file in which
for each position the number of filtered read pairs is
reported that contain the position in or between the
reads</p>

<p style="margin-top: 1em">--pairwiseAlignments <br>
use in case alignments were done in pairwise fashion
(default: 0)</p>

<p style="margin-top: 1em">AUTHORS <br>
AUGUSTUS was written by M. Stanke, O. Keller, S.
K&Atilde;&para;nig, L. Gerischer and L. Romoth.</p>

<p style="margin-top: 1em">ADDITIONAL DOCUMENTATION <br>
An exhaustive documentation can be found in the file
/usr/share/augustus/README.TXT.</p>
 
<p style="margin-top: 1em">FILTERBAM(1)</p>
<hr>
</body>
</html>
