<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:10:17 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FLEXBAR(1) User Commands FLEXBAR(1)</p>

<p style="margin-top: 1em">NAME <br>
flexbar - flexible barcode and adapter removal for
sequencing platforms</p>

<p style="margin-top: 1em">SYNOPSIS <br>
flexbar -r reads [-t target] [-b barcodes] [-a adapters]
[options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Flexbar preprocesses high-throughput sequencing data
efficiently. It demultiplexes barcoded runs and removes
adapter sequences. Moreover, trimming and filtering features
are pro&acirc; <br>
vided. Flexbar increases mapping rates and improves genome
and transcriptome assemblies. It supports next-generation
sequencing data in fasta/q and csfasta/q format from
Illu&acirc; <br>
mina, Roche 454, and the SOLiD platform.</p>

<p style="margin-top: 1em">Parameter names changed in
Flexbar. Please review scripts. The recent months, default
settings were optimised, several bugs were fixed and various
improvements were made, e.g. <br>
revamped command-line interface, new trimming modes as well
as lower time and memory requirements.</p>

<p style="margin-top: 1em">-h, --help</p>

<p style="margin-top: 1em">Displays this help message.</p>

<p style="margin-top: 1em">-w, --advanced</p>

<p style="margin-top: 1em">Print advanced options help
screen.</p>

<p style="margin-top: 1em">-i, --cite</p>

<p style="margin-top: 1em">Show program reference for
citation.</p>

<p style="margin-top: 1em">Basic options:</p>

<p style="margin-top: 1em">-n, --threads NUM</p>

<p style="margin-top: 1em">Number of threads to employ.
Default: 1.</p>

<p style="margin-top: 1em">-t, --target STR</p>

<p style="margin-top: 1em">Prefix for output file names or
paths. Default: flexbar.</p>

<p style="margin-top: 1em">-r, --reads FILE</p>

<p style="margin-top: 1em">Fasta/q file or stdin (-) with
reads that may contain barcodes.</p>

<p style="margin-top: 1em">-p, --reads2 FILE</p>

<p style="margin-top: 1em">Second input file of paired
reads, gz and bz2 files supported.</p>

<p style="margin-top: 1em">-f, --format STR</p>

<p style="margin-top: 1em">Quality format: sanger, solexa,
i1.3, i1.5, i1.8 (illumina 1.8+).</p>

<p style="margin-top: 1em">-c, --color-space</p>

<p style="margin-top: 1em">Input in color-space format
csfasta or csfastq in sanger scaling.</p>

<p style="margin-top: 1em">Barcode detection:</p>

<p style="margin-top: 1em">-b, --barcodes FILE</p>

<p style="margin-top: 1em">Fasta file with barcodes for
demultiplexing that may contain N.</p>

<p style="margin-top: 1em">-br, --barcode-reads FILE</p>

<p style="margin-top: 1em">Fasta/q file composed of
separate barcode reads for detection.</p>

<p style="margin-top: 1em">-be, --barcode-trim-end STR</p>

<p style="margin-top: 1em">Type of detection, see section
trim-end modes. Default: ANY.</p>

<p style="margin-top: 1em">-bo, --barcode-min-overlap
NUM</p>

<p style="margin-top: 1em">Minimum overlap of barcode and
read. Default: barcode length.</p>

<p style="margin-top: 1em">-bt, --barcode-threshold NUM</p>

<p style="margin-top: 1em">Allowed mismatches and gaps per
10 bases overlap. Default: 1.0.</p>

<p style="margin-top: 1em">-bu, --barcode-unassigned</p>

<p style="margin-top: 1em">Include unassigned reads in
output generation.</p>

<p style="margin-top: 1em">Adapter removal:</p>

<p style="margin-top: 1em">-a, --adapters FILE</p>

<p style="margin-top: 1em">Fasta file with adapters for
removal that may contain N.</p>

<p style="margin-top: 1em">-as, --adapter-seq STR</p>

<p style="margin-top: 1em">Single adapter sequence as
alternative to adapters option.</p>

<p style="margin-top: 1em">-ae, --adapter-trim-end STR</p>

<p style="margin-top: 1em">Type of removal, see section
trim-end modes. Default: RIGHT.</p>

<p style="margin-top: 1em">-ao, --adapter-min-overlap
NUM</p>

<p style="margin-top: 1em">Minimum overlap of adapter and
read sequence. Default: 1.</p>

<p style="margin-top: 1em">-at, --adapter-threshold NUM</p>

<p style="margin-top: 1em">Allowed mismatches and gaps per
10 bases overlap. Default: 3.0.</p>

<p style="margin-top: 1em">Filtering and trimming:</p>

<p style="margin-top: 1em">-u, --max-uncalled NUM</p>

<p style="margin-top: 1em">Allowed uncalled bases (N or .)
for each read. Default: 0.</p>

<p style="margin-top: 1em">-x, --pre-trim-left NUM</p>

<p style="margin-top: 1em">Trim given number of bases on
5&rsquo; read end before detection.</p>

<p style="margin-top: 1em">-y, --pre-trim-right NUM</p>

<p style="margin-top: 1em">Trim specified number of bases
on 3&rsquo; end prior to detection.</p>

<p style="margin-top: 1em">-q, --pre-trim-phred NUM</p>

<p style="margin-top: 1em">Trim 3&rsquo; end until
specified or higher quality reached.</p>

<p style="margin-top: 1em">-k, --post-trim-length NUM</p>

<p style="margin-top: 1em">Trim to specified read length
from 3&rsquo; end after removal.</p>

<p style="margin-top: 1em">-m, --min-read-length NUM</p>

<p style="margin-top: 1em">Minimum read length to remain
after removal. Default: 18.</p>

<p style="margin-top: 1em">Output selection:</p>

<p style="margin-top: 1em">-o, --fasta-output</p>

<p style="margin-top: 1em">Prefer non-quality formats fasta
and csfasta for output.</p>

<p style="margin-top: 1em">-z, --zip-output STR</p>

<p style="margin-top: 1em">Enable direct compression of
output files. One of GZ and BZ2.</p>

<p style="margin-top: 1em">-s, --single-reads</p>

<p style="margin-top: 1em">Write single paired reads for
too short counterparts.</p>

<p style="margin-top: 1em">Logging and tagging:</p>

<p style="margin-top: 1em">-l, --log-level STR</p>

<p style="margin-top: 1em">Print valid optimal read
alignment. One of ALL, MOD, and TAB.</p>

<p style="margin-top: 1em">-g, --removal-tags</p>

<p style="margin-top: 1em">Tag reads that are subject to
adapter or barcode removal.</p>

<p style="margin-top: 1em">Trim-end modes:</p>

<p style="margin-top: 1em">ANY: longer side of read remains
after removal of overlap LEFT: right side remains after
removal, align &lt;= read end RIGHT: left part remains after
removal, align &gt;= read <br>
start LEFT_TAIL: consider first n bases of reads in
alignment RIGHT_TAIL: use only last n bases, see tail-length
options</p>

<p style="margin-top: 1em">EXAMPLES <br>
flexbar -r reads.fq -f i1.8 -t target -b brc.fa -a adap.fa
flexbar -r reads.csfastq.gz -a adap.fa -ao 5 -ae LEFT -c</p>

<p style="margin-top: 1em">VERSION <br>
flexbar version: 2.4 Last update July 29, 2013</p>

<p style="margin-top: 1em">Advanced options: flexbar -w</p>

<p style="margin-top: 1em">flexbar version 2.4</p>

<p style="margin-top: 1em">2.4 January 2014 FLEXBAR(1)</p>
<hr>
</body>
</html>
