<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:10:28 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FML-ASM(1) User Commands FML-ASM(1)</p>

<p style="margin-top: 1em">NAME <br>
fml-asm - assemble Illumina short reads in small regions</p>

<p style="margin-top: 1em">SYNOPSIS <br>
fml-asm [options] &lt;in.fq&gt;</p>

<p style="margin-top: 1em">OPTIONS <br>
-e INT k-mer length for error correction (0 for auto; -1 to
disable) [0]</p>

<p style="margin-top: 1em">-c INT1[,INT2] <br>
range of k-mer &amp; read count thresholds for ec and graph
cleaning [4,8]</p>

<p style="margin-top: 1em">-l INT min overlap length during
initial assembly [33]</p>

<p style="margin-top: 1em">-r FLOAT <br>
drop an overlap if its length is below maxOvlpLen*FLOAT
[0.7]</p>

<p style="margin-top: 1em">-t INT number of threads
(don&rsquo;t use multi-threading for small data sets)
[1]</p>

<p style="margin-top: 1em">-A discard heterozygotes (apply
this to assemble bacterial genomes)</p>

<p style="margin-top: 1em">fml-asm 0.1 August 2016
FML-ASM(1)</p>
<hr>
</body>
</html>
