<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>FREECONTACT(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">FREECONTACT(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">FREECONTACT(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
freecontact - fast protein contact predictor
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
freecontact [OPTION]
<div class="Pp"></div>
freecontact --parprof [evfold|psicov|psicov-sd] &lt; alignment.aln &gt;
  contacts.out
<div class="Pp"></div>
/usr/share/freecontact/a2m2aln --query '^RASH_HUMAN/(\d+)' &lt; alignment.fa |
  freecontact --parprof evfold &gt; contacts.out
<div class="Pp"></div>
freecontact --ali= <i>ALIFILE</i> --apply-gapth=<i>BOOL</i> --clustpc=<i>NUM</i>
  --density= <i>NUM</i> --cov20=<i>BOOL</i> --estimate-ivcov=<i>BOOL</i>
  --gapth= <i>NUM</i> --icme-timeout=<i>NUM</i> --input-format=<i>[flat|xml]</i>
  --mincontsep= <i>NUM</i> --output-format=<i>[evfold|pfrmat_rr|bioxsd]</i>
  --pseudocnt= <i>NUM</i> --pscount-weight=<i>NUM</i> --rho=<i>NUM</i>
  --threads= <i>NUM</i> --veczw=<i>BOOL</i>
<div class="Pp"></div>
freecontact --help --debug --quiet --version
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
FreeContact is a protein residue contact predictor optimized for speed.
  FreeContact can function as an accelerated drop-in for the published contact
  predictors EVfold-mfDCA of DS. Marks et al. (2011) [1], and PSICOV of D. Jones
  et al. (2011) [2].
<div class="Pp"></div>
FreeContact is accelerated by a combination of vector instructions, multiple
  threads, and faster implementation of key parts. Depending on the alignment,
  8-fold or higher speedups are possible.
<div class="Pp"></div>
A sufficiently large alignment is required for meaningful results. As a minimum,
  an alignment with an effective (after-weighting) sequence count bigger than
  the length of the query sequence should be used. Alignments with tens of
  thousands of (effective) sequences are considered good input.
<div class="Pp"></div>
<i>jackhmmer</i>(1) or <i>hhblits</i>(1) can be used to generate the alignments,
  for example.
<div class="Pp"></div>
[1] PLoS One. 2011;6(12):e28766. doi: 10.1371/journal.pone.0028766. Epub 2011
  Dec 7. Protein 3D structure computed from evolutionary sequence variation.
  Marks DS, Colwell LJ, Sheridan R, Hopf TA, Pagnani A, Zecchina R, Sander C.
<div class="Pp"></div>
[2] Bioinformatics. 2012 Jan 15;28(2):184-90. Epub 2011 Nov 17. PSICOV: precise
  structural contact prediction using sparse inverse covariance estimation on
  large multiple sequence alignments. Jones DT, Buchan DW, Cozzetto D, Pontil M.
<h2 class="Ss" title="Ss" id="Input"><a class="selflink" href="#Input">Input</a></h2>
The following formats are supported:
<dl class="Bl-tag">
  <dt class="It-tag"><i>flat</i></dt>
  <dd class="It-tag">The following simple input file format is used:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
 # querystart=5
 # query=QUERYwithinsertionSEQUENCEWITHNOGAPSORINSERTIONS
 QUERYSEQUENCEWITHNOGAPSORINSERTIONS
 -ALIGNED---SEQUENCE--WITH-GAPS-----
 ANOTHER-ALIGNED------------SEQUENCE
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    The '#' header lines are optional. Header lines are used to calculate
      contact residue numbers and to look up respective query residues for
      certain output formats.
    <div style="height: 1.00em;">&#x00A0;</div>
    If no query is defined, the first sequence in the alignment is used as the
      query sequence. The query sequence must not contain gaps in the alignment.
    <div style="height: 1.00em;">&#x00A0;</div>
    All alignment rows must be the same length, and may contain only
      [ABCDEFGHIJKLMNOPQRSTUVWXYZ-]. [B] is mapped to [D], [Z] is mapped to [E],
      [JOUX] are mapped to [X]. [X] matches only itself for the entire program.
    <div style="height: 1.00em;">&#x00A0;</div>
    A2M input alignments can be converted to the above format using
      <i>/usr/share/freecontact/a2m2aln</i>. a2m2aln can be used to pipe the
      alignment directly into freecontact.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>xml</i></dt>
  <dd class="It-tag">XML document with one &lt;fc:alignment
      xmlns:fc=&quot;http://rostlab.org/freecontact/xsd&quot;/&gt; element,
      defined in the FreeContact schema [4] derived from BioXSD [5].
    <div style="height: 1.00em;">&#x00A0;</div>
    Example:
    <i>/usr/share/doc/freecontact/examples/PF00071_v25_1000.xml</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Output"><a class="selflink" href="#Output">Output</a></h2>
The original EVfold-mfDCA or PSICOV output format is used by default when the
  respective parameter profile is selected.
<dl class="Bl-tag">
  <dt class="It-tag"><i>evfold</i> (EVfold-mfDCA)</dt>
  <dd class="It-tag">
    <pre>
 5 K 6 L 0.332129 3.59798
 | | | | |        + corrected norm (CN) contact score
 | | | | + mutual information (MI) score
 | | | + contact amino acid residue code
 | | + contact residue number
 | + contact amino acid residue code
 + contact residue number
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    Contacts are sorted on residue number.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>pfrmat_rr</i> (PSICOV)</dt>
  <dd class="It-tag">CASP residue-residue separation prediction (PFRMAT RR)
      format [3]:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
 55 67 0 8 10.840280
 |  |  | | + contact score
 |  |  +-+ range [&#x212B;] of Cb-Cb distance predicted for the residue pair
 |  |      (C-alpha for glycines)
 |  |      These two fields are invariant in the output.
 |  + contact residue number
 + contact residue number
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    Contacts are sorted on score, descending.
    <div style="height: 1.00em;">&#x00A0;</div>
    [3] &lt;http://predictioncenter.org/casp10/index.cgi?page=format&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>bioxsd</i></dt>
  <dd class="It-tag">XML document with one &lt;fc:contactMap
      xmlns:fc=&quot;http://rostlab.org/freecontact/xsd&quot;/&gt; element,
      defined in the FreeContact schema [4] derived from BioXSD [5].
    <div style="height: 1.00em;">&#x00A0;</div>
    Example:
      <i>/usr/share/doc/freecontact/examples/PF00071_v25_1000.evfold.50.xml</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    Note: as BioXSD is under active development in collaboration with
      FreeContact, the FreeContact schema may actually be derived from a version
      not yet available at [5].
    <div style="height: 1.00em;">&#x00A0;</div>
    [4] &lt;file:///usr/share/freecontact/freecontact.xsd&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    [5] &lt;http://bioxsd.org&gt;</dd>
</dl>
<div class="Pp"></div>
The output may not list all possible contacts.
<h1 class="Sh" title="Sh" id="REFERENCES"><a class="selflink" href="#REFERENCES">REFERENCES</a></h1>
<div style="margin-left: 4.00ex;">Submitted. FreeContact: fast and free direct
  residue-residue contact prediction. Kaj&#x00E1;n L, Sustik MA, Marks DS, Hopf
  TA, Kala&#x0161; M, Rost B.</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">-a [ --threads ] arg</dt>
  <dd class="It-tag">Threads to use [0-). 0 means as many as cores.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--apply-gapth arg</dt>
  <dd class="It-tag">When true, exclude residue columns and rows with a weighted
      gap frequency &gt; <b>--gapth</b> from the covariance matrix
    [Boolean].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-c [ --clustpc ] arg</dt>
  <dd class="It-tag">BLOSUM clustering percentage [0-100].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--cov20 arg</dt>
  <dd class="It-tag">If true, leave one amino acid off the covariance matrix,
      making it non-overdetermined [Boolean].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-d [ --density ] arg</dt>
  <dd class="It-tag">Target precision matrix density [0-1]. Set <i>0</i> to not
      control density.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--debug</dt>
  <dd class="It-tag">Turn on debugging.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--estimate-ivcov arg</dt>
  <dd class="It-tag">Use inverse covariance matrix estimation instead of matrix
      inversion [Boolean].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-f [ --ali ] arg (=-)</dt>
  <dd class="It-tag">Alignment file [path]. If '-', standard input. Default:
      '-'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-g [ --gapth ] arg</dt>
  <dd class="It-tag">Weighted gap frequency threshold (0-1].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-h [ --help ]</dt>
  <dd class="It-tag">Produce this help message.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-i [ --input-format ] arg (=flat)</dt>
  <dd class="It-tag">Input format [flat|xml].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--icme-timeout arg (=1800)</dt>
  <dd class="It-tag">Inverse covariance matrix estimation timeout in seconds
      [0-). Applied to each iversion call independently. If a timeout occurs,
      the program exits with status <i>2</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--mincontsep arg</dt>
  <dd class="It-tag">Minimum sequence-wise contacting residue pair separation
      given in amino acids as (j-i&gt;=arg). <i>1</i> for adjacent residues.
      [1-).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-o [ --output-format ] arg</dt>
  <dd class="It-tag">Output format [evfold|pfrmat_rr|bioxsd].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--parprof arg (=default)</dt>
  <dd class="It-tag">Parameter profile (optional) [default|evfold|psicov]. The
      default profile is <i>evfold</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    Command line arguments can be used to override profile values.</dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">evfold</dt>
  <dd class="It-tag">Triggers EVfold-mfDCA [1] compatibility mode.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">psicov</dt>
  <dd class="It-tag">Triggers PSICOV [2] compatibility mode.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">psicov-sd</dt>
  <dd class="It-tag">Triggers PSICOV [2] sensible default mode: fixed default
      rho, no density control.</dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">-w [ --pscount-weight ] arg</dt>
  <dd class="It-tag">Pseudocount weight [0-1].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-p [ --pseudocnt ] arg</dt>
  <dd class="It-tag">Pseudocount [0-).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--pep</dt>
  <dd class="It-tag">Print effective parameters on standard error. Use this
      option to see what parameters <i>freecontact</i>(1) is run with in detail.
      This is especially useful when the <b>--parprof</b> option is used in
      combination with other options.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--rho arg</dt>
  <dd class="It-tag">Initial value of Glasso regularization parameter [0-). If
      negative, choose value automatically.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--quiet arg (=0)</dt>
  <dd class="It-tag">Print nothing but error messages on standard error. Does
      not affect <b>--debug</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--veczw arg</dt>
  <dd class="It-tag">Use vectorized sequence weighting when available
    [Boolean].</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--version</dt>
  <dd class="It-tag">Print version.</dd>
</dl>
<h1 class="Sh" title="Sh" id="EXIT_STATUS"><a class="selflink" href="#EXIT_STATUS">EXIT
  STATUS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><i>0</i></dt>
  <dd class="It-tag">No error - success.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>1</i></dt>
  <dd class="It-tag">Unspecified error.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>2</i></dt>
  <dd class="It-tag">A timeout (see <b>--icme-timeout</b>) occurred.</dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<pre>
 /usr/share/freecontact/a2m2aln --query '^RASH_HUMAN/(\d+)' &lt; '/usr/share/doc/freecontact/examples/PF00071_v25_1000.fa' | \
  freecontact --parprof evfold &gt; PF00071_v25_1000.evfold
 freecontact --parprof evfold -i xml -o bioxsd &lt; '/usr/share/doc/freecontact/examples/PF00071_v25_1000.xml' &gt; PF00071_v25_1000.evfold.xml
 freecontact --parprof psicov &lt; /usr/share/doc/freecontact/examples/demo_1000.aln &gt; demo_1000.psicov
</pre>
<h1 class="Sh" title="Sh" id="NOTES"><a class="selflink" href="#NOTES">NOTES</a></h1>
For optimal performance, use the Automatically Tuned Linear Algebra Software
  (ATLAS) library <i>compiled on the machine</i> where freecontact is run.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
L&#x00E1;szl&#x00F3; Kaj&#x00E1;n &lt;lkajan@rostlab.org&gt;
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<i>jackhmmer</i>(1), <i>hhblits</i>(1), <i>blastpgp</i>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-11-01</td>
    <td class="foot-os">1.0.21</td>
  </tr>
</table>
</body>
</html>
