<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:10:45 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>FREECONTACT(1) User Commands FREECONTACT(1)</p>

<p style="margin-top: 1em">NAME <br>
freecontact - fast protein contact predictor</p>

<p style="margin-top: 1em">SYNOPSIS <br>
freecontact [OPTION]</p>

<p style="margin-top: 1em">freecontact --parprof
[evfold|psicov|psicov-sd] &lt; alignment.aln &gt;
contacts.out</p>

<p style="margin-top: 1em">/usr/share/freecontact/a2m2aln
--query &rsquo;^RASH_HUMAN/(+)&rsquo; &lt; alignment.fa |
freecontact --parprof evfold &gt; contacts.out</p>

<p style="margin-top: 1em">freecontact --ali=ALIFILE
--apply-gapth=BOOL --clustpc=NUM --density=NUM --cov20=BOOL
--estimate-ivcov=BOOL --gapth=NUM --icme-timeout=NUM
--input-format=[flat|xml] <br>
--mincontsep=NUM --output-format=[evfold|pfrmat_rr|bioxsd]
--pseudocnt=NUM --pscount-weight=NUM --rho=NUM --threads=NUM
--veczw=BOOL</p>

<p style="margin-top: 1em">freecontact --help --debug
--quiet --version</p>

<p style="margin-top: 1em">DESCRIPTION <br>
FreeContact is a protein residue contact predictor optimized
for speed. FreeContact can function as an accelerated
drop-in for the published contact predictors EVfold-mfDCA of
<br>
DS. Marks et al. (2011) [1], and PSICOV of D. Jones et al.
(2011) [2].</p>

<p style="margin-top: 1em">FreeContact is accelerated by a
combination of vector instructions, multiple threads, and
faster implementation of key parts. Depending on the
alignment, 8-fold or higher <br>
speedups are possible.</p>

<p style="margin-top: 1em">A sufficiently large alignment
is required for meaningful results. As a minimum, an
alignment with an effective (after-weighting) sequence count
bigger than the length of the <br>
query sequence should be used. Alignments with tens of
thousands of (effective) sequences are considered good
input.</p>

<p style="margin-top: 1em">jackhmmer(1) or hhblits(1) can
be used to generate the alignments, for example.</p>

<p style="margin-top: 1em">[1] PLoS One. 2011;6(12):e28766.
doi: 10.1371/journal.pone.0028766. Epub 2011 Dec 7. Protein
3D structure computed from evolutionary sequence variation.
Marks DS, Colwell LJ, <br>
Sheridan R, Hopf TA, Pagnani A, Zecchina R, Sander C.</p>

<p style="margin-top: 1em">[2] Bioinformatics. 2012 Jan
15;28(2):184-90. Epub 2011 Nov 17. PSICOV: precise
structural contact prediction using sparse inverse
covariance estimation on large multiple <br>
sequence alignments. Jones DT, Buchan DW, Cozzetto D, Pontil
M.</p>

<p style="margin-top: 1em">Input <br>
The following formats are supported:</p>

<p style="margin-top: 1em">flat <br>
The following simple input file format is used:</p>

<p style="margin-top: 1em"># querystart=5 <br>
# query=QUERYwithinsertionSEQUENCEWITHNOGAPSORINSERTIONS
<br>
QUERYSEQUENCEWITHNOGAPSORINSERTIONS <br>
-ALIGNED---SEQUENCE--WITH-GAPS----- <br>
ANOTHER-ALIGNED------------SEQUENCE</p>

<p style="margin-top: 1em">The &rsquo;#&rsquo; header lines
are optional. Header lines are used to calculate contact
residue numbers and to look up respective query residues for
certain output formats.</p>

<p style="margin-top: 1em">If no query is defined, the
first sequence in the alignment is used as the query
sequence. The query sequence must not contain gaps in the
alignment.</p>

<p style="margin-top: 1em">All alignment rows must be the
same length, and may contain only
[ABCDEFGHIJKLMNOPQRSTUVWXYZ-]. [B] is mapped to [D], [Z] is
mapped to [E], [JOUX] are mapped to [X]. [X] <br>
matches only itself for the entire program.</p>

<p style="margin-top: 1em">A2M input alignments can be
converted to the above format using
/usr/share/freecontact/a2m2aln. a2m2aln can be used to pipe
the alignment directly into freecontact.</p>

<p style="margin-top: 1em">xml XML document with one
&lt;fc:alignment
xmlns:fc=&quot;http://rostlab.org/freecontact/xsd&quot;/&gt;
element, defined in the FreeContact schema [4] derived from
BioXSD [5].</p>

<p style="margin-top: 1em">Example:
/usr/share/doc/freecontact/examples/PF00071_v25_1000.xml.</p>

<p style="margin-top: 1em">Output <br>
The original EVfold-mfDCA or PSICOV output format is used by
default when the respective parameter profile is
selected.</p>

<p style="margin-top: 1em">evfold (EVfold-mfDCA) <br>
5 K 6 L 0.332129 3.59798 <br>
| | | | | + corrected norm (CN) contact score <br>
| | | | + mutual information (MI) score <br>
| | | + contact amino acid residue code <br>
| | + contact residue number <br>
| + contact amino acid residue code <br>
+ contact residue number</p>

<p style="margin-top: 1em">Contacts are sorted on residue
number.</p>

<p style="margin-top: 1em">pfrmat_rr (PSICOV) <br>
CASP residue-residue separation prediction (PFRMAT RR)
format [3]:</p>

<p style="margin-top: 1em">55 67 0 8 10.840280 <br>
| | | | + contact score <br>
| | +-+ range [&Atilde;] of Cb-Cb distance predicted for the
residue pair <br>
| | (C-alpha for glycines) <br>
| | These two fields are invariant in the output. <br>
| + contact residue number <br>
+ contact residue number</p>

<p style="margin-top: 1em">Contacts are sorted on score,
descending.</p>

<p style="margin-top: 1em">[3]
&lt;http://predictioncenter.org/casp10/index.cgi?page=format&gt;</p>

<p style="margin-top: 1em">bioxsd <br>
XML document with one &lt;fc:contactMap
xmlns:fc=&quot;http://rostlab.org/freecontact/xsd&quot;/&gt;
element, defined in the FreeContact schema [4] derived from
BioXSD [5].</p>

<p style="margin-top: 1em">Example:
/usr/share/doc/freecontact/examples/PF00071_v25_1000.evfold.50.xml.</p>

<p style="margin-top: 1em">Note: as BioXSD is under active
development in collaboration with FreeContact, the
FreeContact schema may actually be derived from a version
not yet available at [5].</p>

<p style="margin-top: 1em">[4]
&lt;file:///usr/share/freecontact/freecontact.xsd&gt;</p>

<p style="margin-top: 1em">[5]
&lt;http://bioxsd.org&gt;</p>

<p style="margin-top: 1em">The output may not list all
possible contacts.</p>

<p style="margin-top: 1em">REFERENCES <br>
Submitted. FreeContact: fast and free direct residue-residue
contact prediction. Kaj&Atilde;&iexcl;n L, Sustik MA, Marks
DS, Hopf TA, Kala&Aring;&iexcl; M, Rost B.</p>

<p style="margin-top: 1em">OPTIONS <br>
-a [ --threads ] arg <br>
Threads to use [0-). 0 means as many as cores.</p>

<p style="margin-top: 1em">--apply-gapth arg <br>
When true, exclude residue columns and rows with a weighted
gap frequency &gt; --gapth from the covariance matrix
[Boolean].</p>

<p style="margin-top: 1em">-c [ --clustpc ] arg <br>
BLOSUM clustering percentage [0-100].</p>

<p style="margin-top: 1em">--cov20 arg <br>
If true, leave one amino acid off the covariance matrix,
making it non-overdetermined [Boolean].</p>

<p style="margin-top: 1em">-d [ --density ] arg <br>
Target precision matrix density [0-1]. Set 0 to not control
density.</p>

<p style="margin-top: 1em">--debug <br>
Turn on debugging.</p>

<p style="margin-top: 1em">--estimate-ivcov arg <br>
Use inverse covariance matrix estimation instead of matrix
inversion [Boolean].</p>

<p style="margin-top: 1em">-f [ --ali ] arg (=-) <br>
Alignment file [path]. If &rsquo;-&rsquo;, standard input.
Default: &rsquo;-&rsquo;.</p>

<p style="margin-top: 1em">-g [ --gapth ] arg <br>
Weighted gap frequency threshold (0-1].</p>

<p style="margin-top: 1em">-h [ --help ] <br>
Produce this help message.</p>

<p style="margin-top: 1em">-i [ --input-format ] arg
(=flat) <br>
Input format [flat|xml].</p>

<p style="margin-top: 1em">--icme-timeout arg (=1800) <br>
Inverse covariance matrix estimation timeout in seconds
[0-). Applied to each iversion call independently. If a
timeout occurs, the program exits with status 2.</p>

<p style="margin-top: 1em">--mincontsep arg <br>
Minimum sequence-wise contacting residue pair separation
given in amino acids as (j-i&gt;=arg). 1 for adjacent
residues. [1-).</p>

<p style="margin-top: 1em">-o [ --output-format ] arg <br>
Output format [evfold|pfrmat_rr|bioxsd].</p>

<p style="margin-top: 1em">--parprof arg (=default) <br>
Parameter profile (optional) [default|evfold|psicov]. The
default profile is evfold.</p>

<p style="margin-top: 1em">Command line arguments can be
used to override profile values.</p>

<p style="margin-top: 1em">evfold <br>
Triggers EVfold-mfDCA [1] compatibility mode.</p>

<p style="margin-top: 1em">psicov <br>
Triggers PSICOV [2] compatibility mode.</p>

<p style="margin-top: 1em">psicov-sd <br>
Triggers PSICOV [2] sensible default mode: fixed default
rho, no density control.</p>

<p style="margin-top: 1em">-w [ --pscount-weight ] arg <br>
Pseudocount weight [0-1].</p>

<p style="margin-top: 1em">-p [ --pseudocnt ] arg <br>
Pseudocount [0-).</p>

<p style="margin-top: 1em">--pep <br>
Print effective parameters on standard error. Use this
option to see what parameters freecontact(1) is run with in
detail. This is especially useful when the --parprof <br>
option is used in combination with other options.</p>

<p style="margin-top: 1em">--rho arg <br>
Initial value of Glasso regularization parameter [0-). If
negative, choose value automatically.</p>

<p style="margin-top: 1em">--quiet arg (=0) <br>
Print nothing but error messages on standard error. Does not
affect --debug.</p>

<p style="margin-top: 1em">--veczw arg <br>
Use vectorized sequence weighting when available
[Boolean].</p>

<p style="margin-top: 1em">--version <br>
Print version.</p>

<p style="margin-top: 1em">EXIT STATUS <br>
0 No error - success.</p>

<p style="margin-top: 1em">1 Unspecified error.</p>

<p style="margin-top: 1em">2 A timeout (see --icme-timeout)
occurred.</p>

<p style="margin-top: 1em">EXAMPLES <br>
/usr/share/freecontact/a2m2aln --query
&rsquo;^RASH_HUMAN/(+)&rsquo; &lt;
&rsquo;/usr/share/doc/freecontact/examples/PF00071_v25_1000.fa&rsquo;
| freecontact --parprof evfold &gt;
PF00071_v25_1000.evfold</p>

<p style="margin-top: 1em">freecontact --parprof evfold -i
xml -o bioxsd &lt;
&rsquo;/usr/share/doc/freecontact/examples/PF00071_v25_1000.xml&rsquo;
&gt; PF00071_v25_1000.evfold.xml</p>

<p style="margin-top: 1em">freecontact --parprof psicov
&lt; /usr/share/doc/freecontact/examples/demo_1000.aln &gt;
demo_1000.psicov</p>

<p style="margin-top: 1em">NOTES <br>
For optimal performance, use the Automatically Tuned Linear
Algebra Software (ATLAS) library compiled on the machine
where freecontact is run.</p>

<p style="margin-top: 1em">AUTHOR <br>
L&Atilde;&iexcl;szl&Atilde;&sup3; Kaj&Atilde;&iexcl;n
&lt;lkajan@rostlab.org&gt;</p>

<p style="margin-top: 1em">SEE ALSO <br>
jackhmmer(1), hhblits(1), blastpgp(1)</p>

<p style="margin-top: 1em">1.0.21 2016-11-01
FREECONTACT(1)</p>
<hr>
</body>
</html>
