<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GBrowse2(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GBrowse2(1)</td>
    <td class="head-vol">USER COMMANDS</td>
    <td class="head-rtitle">GBrowse2(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
GBrowse2 - The Generic Genome Browser
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>libgbrowse-perl</b> [-h]
<h1 class="Sh" title="Sh" id="INSTALLATION"><a class="selflink" href="#INSTALLATION">INSTALLATION</a></h1>
If Apache2 is installed on the system, the software has configured it to link
  URLs to GBrowse directories. A restart of Apache is required to take it into
  account. Configuration is available in /etc/gbrowse2/apache2.conf file.
<br/>
 If another web server is used, one should refer to the apache2 template to map
  urls in the same way.
<h1 class="Sh" title="Sh" id="DEPENDENCIES"><a class="selflink" href="#DEPENDENCIES">DEPENDENCIES</a></h1>
Additional optional perl dependencies exist to provide additional features.
  Please refer to Optional modules in INSTALL file for more information.
<h1 class="Sh" title="Sh" id="CONFIGURATION"><a class="selflink" href="#CONFIGURATION">CONFIGURATION</a></h1>
<b>gbrowse2</b> Configuration files are located in /etc/gbrowse. Main
  configuration file is GBrowse.conf. It contains the global configuration
  elements. Specific configuration files are located in the same directory per
  data bank. For more details, one should refer to the GBrowse documentation.
<br/>
 Installation is provided with a Yeast genome sample.
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
GBrowse is a simple but highly
<br/>
 configurable web-based genome browser. It is a component of the
<br/>
 Generic Model Organism Systems Database project (GMOD).
<br/>
 Some of its features:
<br/>
 * Simultaneous bird's eye and detailed views of the genome;
<br/>
 * Scroll, zoom, center;
<br/>
 * Attach arbitrary URLs to any annotation;
<br/>
 * Order and appearance of tracks are customizable by administrator and
<br/>
 end-user;
<br/>
 * Search by annotation ID, name, or comment;
<br/>
 * Supports third party annotation using GFF formats;
<br/>
 * Settings persist across sessions;
<br/>
 * DNA and GFF dumps;
<br/>
 * Connectivity to different databases, including BioSQL and Chado;
<br/>
 * Multi-language support;
<br/>
 * Third-party feature loading;
<br/>
 * Customizable plug-in architecture (e.g. run BLAST, dump &amp; import many
<br/>
 formats, find oligonucleotides, design primers, create restriction maps,
<br/>
 edit features).
<h1 class="Sh" title="Sh" id="WEB_ACCESS"><a class="selflink" href="#WEB_ACCESS">WEB
  ACCESS</a></h1>
GBrowse is accessible at URL http://localhost/gbrowse2
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Olivier Sallou (olivier.sallou (at) irisa.fr) - Man page and packaging Lincoln
  Stein - GBrowse</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 7, 2011</td>
    <td class="foot-os">version 1.1.0</td>
  </tr>
</table>
</body>
</html>
