<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:11:46 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GBrowse2(1) USER COMMANDS GBrowse2(1)</p>

<p style="margin-top: 1em">NAME <br>
GBrowse2 - The Generic Genome Browser</p>

<p style="margin-top: 1em">SYNOPSIS <br>
libgbrowse-perl [-h]</p>

<p style="margin-top: 1em">INSTALLATION <br>
If Apache2 is installed on the system, the software has
configured it to link URLs to GBrowse directories. A restart
of Apache is required to take it into account. Configuration
<br>
is available in /etc/gbrowse2/apache2.conf file. <br>
If another web server is used, one should refer to the
apache2 template to map urls in the same way.</p>

<p style="margin-top: 1em">DEPENDENCIES <br>
Additional optional perl dependencies exist to provide
additional features. Please refer to Optional modules in
INSTALL file for more information.</p>

<p style="margin-top: 1em">CONFIGURATION <br>
gbrowse2 Configuration files are located in /etc/gbrowse.
Main configuration file is GBrowse.conf. It contains the
global configuration elements. Specific configuration files
are <br>
located in the same directory per data bank. For more
details, one should refer to the GBrowse documentation. <br>
Installation is provided with a Yeast genome sample.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
GBrowse is a simple but highly <br>
configurable web-based genome browser. It is a component of
the <br>
Generic Model Organism Systems Database project (GMOD). <br>
Some of its features: <br>
* Simultaneous bird&rsquo;s eye and detailed views of the
genome; <br>
* Scroll, zoom, center; <br>
* Attach arbitrary URLs to any annotation; <br>
* Order and appearance of tracks are customizable by
administrator and <br>
end-user; <br>
* Search by annotation ID, name, or comment; <br>
* Supports third party annotation using GFF formats; <br>
* Settings persist across sessions; <br>
* DNA and GFF dumps; <br>
* Connectivity to different databases, including BioSQL and
Chado; <br>
* Multi-language support; <br>
* Third-party feature loading; <br>
* Customizable plug-in architecture (e.g. run BLAST, dump
&amp; import many <br>
formats, find oligonucleotides, design primers, create
restriction maps, <br>
edit features).</p>

<p style="margin-top: 1em">WEB ACCESS <br>
GBrowse is accessible at URL http://localhost/gbrowse2</p>

<p style="margin-top: 1em">AUTHOR <br>
Olivier Sallou (olivier.sallou (at) irisa.fr) - Man page and
packaging Lincoln Stein - GBrowse</p>

<p style="margin-top: 1em">version 1.1.0 February 07, 2011
GBrowse2(1)</p>
<hr>
</body>
</html>
