<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-BMR-CALC-WIG-COVG(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-BMR-CALC-WIG-COVG(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-BMR-CALC-WIG-COVG(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_bmr_calc-wig-covg"><a class="selflink" href="#genome_music_bmr_calc-wig-covg">genome
  music bmr calc-wig-covg</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music bmr calc-wig-covg - Count covered bases per-gene for each given
  wiggle track format file.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music bmr calc-wig-covg version 0.04 (2016-01-01
  at 23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music bmr calc-wig-covg --roi-file=? --reference-sequence=? --wig-list=?
  --output-dir=?
<div class="Pp"></div>
General usage:
<div class="Pp"></div>
<pre>
 ... music bmr calc-wig-covg \
    --wig-list input_dir/wig_list \
    --output-dir output_dir/ \
    --reference-sequence input_dir/all_sequences.fa \
    --roi-file input_dir/all_coding_exons.tsv
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">roi-file <i>Text</i></dt>
  <dd class="It-tag">Tab-delimited list of ROIs [chr start stop gene_name] (See
      Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reference-sequence <i>Text</i></dt>
  <dd class="It-tag">Path to reference sequence in FASTA format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">wig-list <i>Text</i></dt>
  <dd class="It-tag">Tab-delimited list of WIG files [sample_name wig_file] (See
      Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-dir <i>Text</i></dt>
  <dd class="It-tag">Directory where output files and subdirectories will be
      written</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script counts bases with sufficient coverage in the ROIs of each gene from
  given wiggle track format files, and categorizes them into - AT, CG (non-CpG),
  and CpG counts. It also adds up these base-counts across all ROIs of each gene
  for each sample, but covered bases that lie within overlapping ROIs are not
  counted more than once towards these total counts.
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--roi-file</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The regions of interest (ROIs) of each gene are typically
    regions targeted for sequencing or are merged exon loci (from multiple
    transcripts) of genes with 2-bp flanks (splice junctions). For per-gene base
    counts, an overlapping base will be counted each time it appears in an ROI
    of the same gene. To avoid this, be sure to merge together overlapping ROIs
    of the same gene. BEDtools' mergeBed can help if used per gene.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--reference-sequence</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The reference sequence in FASTA format. If a reference
    sequence index is not found next to this file (a .fai file), it will be
    created.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--wig-list</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Provide a file containing sample names and the wiggle track
    format file locations for each. Use the tab-delimited format [sample_name
    wig_file] per line. Additional columns like clinical data are allowed, but
    ignored. The sample_name must be the same as the tumor sample names used in
    the MAF file (16th column, with the header Tumor_Sample_Barcode).</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--output-dir</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Specify an output directory where the following will be
    created/written: roi_covgs: Subdirectory containing per-ROI covered base
    counts for each sample. gene_covgs: Subdirectory containing per-gene covered
    base counts for each sample. total_covgs: File containing the overall
    non-overlapping coverages per sample.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2010-2011 Washington University in St. Louis.
<div class="Pp"></div>
It is released under the Lesser GNU Public License (LGPL) version 3. See the
  associated LICENSE file in this distribution.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Cyriac Kandoth, Ph.D.
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome-music-bmr</b>(1), <b>genome-music</b>(1), <b>genome</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
