<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-SMG(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-SMG(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-SMG(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_smg"><a class="selflink" href="#genome_music_smg">genome
  music smg</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music smg - Identify significantly mutated genes.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music smg version 0.04 (2016-01-01 at 23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music smg --gene-mr-file=? --output-file=? [--max-fdr=?]
  [--skip-low-mr-genes] [--bmr-modifier-file=?] [--processors=?]
<div class="Pp"></div>
<pre>
 ... music smg \
      --gene-mr-file output_dir/gene_mrs \
      --output-file output_dir/smgs
</pre>
<div class="Pp"></div>
(A &quot;gene-mr-file&quot; can be generated using the tool &quot;music bmr
  calc-bmr&quot;.)
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">gene-mr-file <i>Text</i></dt>
  <dd class="It-tag">File with per-gene mutation rates (Created using
      &quot;music bmr calc-bmr&quot;)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-file <i>Text</i></dt>
  <dd class="It-tag">Output file that will list significantly mutated genes and
      their p-values</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">max-fdr <i>Number</i></dt>
  <dd class="It-tag">The maximum allowed false discovery rate for a gene to be
      considered an SMG
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '0.2' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-low-mr-genes <i>Boolean</i></dt>
  <dd class="It-tag">Skip testing genes with MRs lower than the background MR
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-low-mr-genes <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-low-mr-genes 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bmr-modifier-file <i>Text</i></dt>
  <dd class="It-tag">Tab delimited multipliers per gene that modify BMR before
      testing [gene_name bmr_modifier]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">processors <i>Integer</i></dt>
  <dd class="It-tag">Number of processors to use (requires 'foreach' and 'doMC'
      R packages)
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '1' if not specified</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script runs R-based statistical tools to identify Significantly Mutated
  Genes (SMGs), when given per-gene mutation rates categorized by mutation type,
  and the overall background mutation rates (BMRs) for each of those categories
  (gene_mr_file, created using &quot;music bmr calc-bmr&quot;).
<div class="Pp"></div>
P-values and false discovery rates (FDRs) for each gene in gene_mr_file is
  calculated using three tests: Fisher's Combined P-value test (FCPT),
  Likelihood Ratio test (LRT), and the Convolution test (CT). For a gene, if its
  FDR for at least 2 of these tests is &lt;= max_fdr, it will be output as an
  SMG. Another output file with prefix &quot;_detailed&quot; will have p-values
  and FDRs for all genes.
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--bmr-modifier-file</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The user can provide a BMR modifier for each gene in the
    ROI file, which is a multiplier for the categorized background mutation
    rates, before testing them against the gene's categorized mutation rates.
    Such a file can be used to correct for regional or systematic bias in
    mutation rates across the genome that may be correlated to CpG deamination
    or DNA repair processes like transcription-coupled repair or mismatch
    repair. Mutation rates have also been associated with DNA replication
    timing, where higher mutation rates are seen in late replicating regions.
    Note that the same per-gene multiplier is used on each mutation category of
    BMR. Any genes from the ROI file that are not in the BMR modifier file will
    be tested against unmodified overall BMRs per mutation category. BMR
    modifiers of &lt;=0 are not permitted, because that's just silly.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--skip-low-mr-genes</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Genes with consistently lower MRs than the BMRs across
    mutation categories, may show up in the results as an SMG (by CT or LRT). If
    such genes are not of interest, they may be assigned a p-value of 1. This
    should also speed things up. Genes with higher Indel or Truncation rates
    than the background will not be skipped even if the gene's overall MR is
    lower than the BMR. If bmr-modifiers are applied, this step uses the
    modified BMRs instead.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Qunyuan Zhang, Ph.D.
 Cyriac Kandoth, Ph.D.
 Nathan D. Dees, Ph.D.
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
