<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-SURVIVAL(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-SURVIVAL(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-SURVIVAL(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_survival"><a class="selflink" href="#genome_music_survival">genome
  music survival</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music survival - Create survival plots and P-values for clinical and
  mutational phenotypes.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music survival version 0.04 (2016-01-01 at
  23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music survival --bam-list=? --output-dir=? [--maf-file=?] [--skip-silent]
  [--genetic-data-type=?] [--numeric-clinical-data-file=?]
  [--categorical-clinical-data-file=?] [--glm-clinical-data-file=?]
  [--phenotypes-to-include=?] [--legend-placement=?] [--skip-non-coding]
<div class="Pp"></div>
<pre>
 ... music survival \
        --bam-list /path/myBamList.tsv \
        --maf-file /path/myMAF.tsv \
        --numeric-clinical-data-file /path/myNumericData.tsv \
        --categorical-clinical-data-file /path/myClassData.tsv \
        --output-dir /path/output_directory
 ... music survival \
        --bam-list /path/myBamList.tsv \
        --maf-file /path/myMAF.tsv \
        --glm-clinical-data-file /path/myGLMClinicalData.tsv \
        --output-dir /path/output_directory
 ... music survival \
        --bam-list /path/myBamList.tsv \
        --maf-file /path/myMAF.tsv \
        --genetic-data-type 'gene' \
        --glm-clinical-data-file /path/myGlmClinicalData.tsv \
        --phenotypes-to-include 'Race,Gender,TP53' \
        --output-dir /path/output_directory
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">bam-list <i>Text</i></dt>
  <dd class="It-tag">List of sample names to be included in the analysis. (See
      Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-dir <i>Text</i></dt>
  <dd class="It-tag">Directory where output files will be written</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">maf-file <i>Text</i></dt>
  <dd class="It-tag">List of mutations in MAF format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Skip silent mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-silent 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">genetic-data-type <i>Text</i></dt>
  <dd class="It-tag">Correlate clinical data to &quot;gene&quot; or
      &quot;variant&quot; level data
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'gene' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">numeric-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Table of samples (y) vs. numeric clinical data category
    (x)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">categorical-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Table of samples (y) vs. categorical clinical data category
      (x)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">glm-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Clinical traits, mutational profiles, other mixed clinical
      data (See DESCRIPTION).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">phenotypes-to-include <i>Text</i></dt>
  <dd class="It-tag">Include only these genes and/or phenotypes in the anlaysis.
      (COMMA-DELIMITED)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">legend-placement <i>Text</i></dt>
  <dd class="It-tag">Choose one of 'bottomleft', 'topleft', 'topright', or
      'bottomright'.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'bottomleft' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Skip non-coding mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-non-coding 'false'</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This command performs survival analysis and plots survival curves for mutational
  data, as well as any clinical traits of interest as specified via the
  --phenotypes-to-include input parameter. The analyses performed include the
  Kaplan-Meier estimator followed by the Cox Proportional Hazards model. Outputs
  for each gene/clinical trait analyzed include survival curves, a hazard ratio
  (with confidence intervals), and P-values and FDRs describing the significance
  of the difference between survivors and non-survivors.
<div class="Pp"></div>
All clinical data files are searched for the required (case insensitive)
  &quot;vital_status&quot; and &quot;days_to_last_followup&quot; columns which
  are paired to phenotypes via sample IDs for the survival analysis. The first
  column of all clinical data files MUST contain the sample IDs, same as in
  other MuSiC tools. By default, analysis is performed on every gene present in
  the MAF. Optionally, the analysis may be limited to only specific genes by
  listing them (comma delimited) after the --phenotypes-to-include input
  parameter. Survival analysis may also be performed on other columns in the
  clinical data file by adding the column headers to the list of entries
  specified after the --phenotypes-to-include input parameter.
<div class="Pp"></div>
Here are some general guildelines for creating clinical data input files:
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Headers are required.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The first column of each clinical data file must contain
      sample IDs which match those in both the --bam-list and the MAF variant
      list (in the MAF, this is the Tumor_Sample_Barcode column,
    specifically).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">In at least one of the clinical data files input, columns
      with headers &quot;vital_status&quot; and
      &quot;days_to_last_followup&quot; (case insensitive) must exist.
      &quot;vital_status&quot; must be delineated by 1's and 0's, where 0
      denotes 'living', and 1 denotes 'deceased'.</dd>
</dl>
<div class="Pp"></div>
Note that all input files must be tab-separated.
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--bam-list</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Provide a file containing sample names and normal/tumor BAM
    locations for each. Use the tab- delimited format [sample_name normal_bam
    tumor_bam] per line. This tool only needs sample_name, so all other columns
    can be skipped. The sample_name must be the same as the tumor sample names
    used in the MAF file (16th column, with the header
    Tumor_Sample_Barcode).</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2010-2011 Washington University in St. Louis.
<div class="Pp"></div>
It is released under the Lesser GNU Public License (LGPL) version 3. See the
  associated LICENSE file in this distribution.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Nathan D. Dees, Ph.D.
 Qunyuan Zhang, Ph.D.
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome-music</b>(1), <b>genome</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
