<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-BMR-CALC-COVG(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-BMR-CALC-COVG(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-BMR-CALC-COVG(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_bmr_calc-covg"><a class="selflink" href="#genome_music_bmr_calc-covg">genome
  music bmr calc-covg</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music bmr calc-covg - Uses calcRoiCovg.c to count covered bases per-gene
  for each given tumor-normal pair of BAMs.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music bmr calc-covg version 0.04 (2016-01-01 at
  23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music bmr calc-covg --gene-covg-dir=? --roi-file=? --reference-sequence=?
  --bam-list=? --output-dir=? [--cmd-list-file=?] [--cmd-prefix=?]
  [--normal-min-depth=?] [--tumor-min-depth=?] [--min-mapq=?]
<div class="Pp"></div>
General usage:
<div class="Pp"></div>
<pre>
 ... music bmr calc-covg \
    --bam-list input_dir/bam_list \
    --output-dir output_dir/ \
    --reference-sequence input_dir/all_sequences.fa \
    --roi-file input_dir/all_coding_exons.tsv
</pre>
<div class="Pp"></div>
To create a list of commands that will allow the processing of each tumor-normal
  pair in parallel with an LSF job scheduler:
<div class="Pp"></div>
<pre>
 ... music bmr calc-covg \
    --bam-list input_dir/bam_list \
    --output-dir output_dir/ \
    --reference-sequence input_dir/all_sequences.fa \
    --roi-file input_dir/all_coding_exons.tsv \
    --cmd_list_file parallelizable_commands \
    --cmd_prefix bsub
</pre>
<div class="Pp"></div>
In the above case, the commands printed into the output file
  &quot;parallelizable_commands&quot; can be run in parallel. After they
  complete, rerun this script as printed directly below (--cmd_list_file and
  --cmd_prefix have been removed) to merge the parallelized calculations:
<div class="Pp"></div>
<pre>
 ... music bmr calc-covg \
    --bam-list input_dir/bam_list \
    --output-dir output_dir/ \
    --reference-sequence input_dir/all_sequences.fa \
    --roi-file input_dir/all_coding_exons.tsv
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">gene-covg-dir <i>Text</i></dt>
  <dd class="It-tag">Directory where per-sample gene coverage files are
    located</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">roi-file <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of ROIs [chr start stop gene_name] (See
      Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reference-sequence <i>Text</i></dt>
  <dd class="It-tag">Path to reference sequence in FASTA format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bam-list <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of BAM files [sample_name normal_bam
      tumor_bam] (See Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-dir <i>Text</i></dt>
  <dd class="It-tag">Directory where output files and subdirectories will be
      written</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">cmd-list-file <i>Text</i></dt>
  <dd class="It-tag">A file to write calcRoiCovg commands to (See
    Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">cmd-prefix <i>Text</i></dt>
  <dd class="It-tag">A command that submits a job to your cluster (See
      Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">normal-min-depth <i>Integer</i></dt>
  <dd class="It-tag">The minimum read depth to consider a Normal BAM base as
      covered</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">tumor-min-depth <i>Integer</i></dt>
  <dd class="It-tag">The minimum read depth to consider a Tumor BAM base as
      covered</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">min-mapq <i>Integer</i></dt>
  <dd class="It-tag">The minimum mapping quality of reads to consider towards
      read depth counts</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script counts bases with sufficient coverage in the ROIs of each gene in
  the given pairs of tumor-normal BAM files and categorizes them into - AT, CG
  (non-CpG), and CpG counts. It also adds up these base-counts across all ROIs
  of each gene for each sample, but covered bases that lie within overlapping
  ROIs are not counted more than once towards these total counts.
<div class="Pp"></div>
By default, this script runs a C-based tool named calcRoiCovg for each sample
  one after another, taking ~30 mins per sample to generate per-ROI covered base
  counts. If the results of calcRoiCovg for a sample already exists in the
  output subdirectory roi_covgs, re-calculation is skipped. This allows you to
  run your own calcRoiCovg jobs in parallel or on multiple machines (Keep
  reading).
<div class="Pp"></div>
Speed things up by running calcRoiCovg jobs in parallel: If a compute cluster or
  multiple machines are available, run this script twice as follows:
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Define cmd-list-file and cmd-prefix to generate a file with
      commands that can be submitted to a cluster or run manually. These jobs
      will write per-ROI base counts in a subdirectory roi_covgs.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">After all the parallelized calcRoiCovg jobs are completed,
      run this script again to add them up and generate the final per-gene base
      counts in a subdirectory gene_covgs. Remember to remove the cmd-list-file
      and cmd-prefix arguments or you will just be re-creating a list of
      commands.</dd>
</dl>
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--roi-file</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The regions of interest (ROIs) of each gene are typically
    regions targeted for sequencing or are merged exon loci (from multiple
    transcripts) of genes with 2-bp flanks (splice junctions). ROIs from the
    same chromosome must be listed adjacent to each other in this file. This
    allows the underlying C-based code to run much more efficiently and avoid
    re-counting bases seen in overlapping ROIs (for overall covered base
    counts). For per-gene base counts, an overlapping base will be counted each
    time it appears in an ROI of the same gene. To avoid this, be sure to merge
    together overlapping ROIs of the same gene. BEDtools' mergeBed can help if
    used per gene.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--reference-sequence</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The reference sequence in FASTA format. If a reference
    sequence index is not found next to this file (a .fai file), it will be
    created.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--bam-list</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Provide a file containing sample names and normal/tumor BAM
    locations for each. Use the tab- delimited format [sample_name normal_bam
    tumor_bam] per line. Additional columns like clinical data are allowed, but
    ignored. The sample_name must be the same as the tumor sample names used in
    the MAF file (16th column, with the header Tumor_Sample_Barcode).</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--output-dir</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Specify an output directory where the following will be
    created/written: roi_covgs: Subdirectory containing per-ROI covered base
    counts for each sample. gene_covgs: Subdirectory containing per-gene covered
    base counts for each sample. total_covgs: File containing the overall
    non-overlapping coverages per sample.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--cmd-list-file</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Specify a file into which a list of calcRoiCovg jobs will
    be written to. These can be scheduled in parallel, and will write per-ROI
    covered base-counts into the output subdirectory roi_covgs. If cmd-list-file
    is left unspecified, this script runs calcRoiCovg per sample one after
    another, taking ~30 mins per sample, but it skips samples whose output is
    already in roi_covgs.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--cmd-prefix</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Specify a job submission command that will be prefixed to
    each command in cmd-list-file. This makes batch submission easier. Just run
    the cmd-list-file file as a shell script to submit jobs. cmd-prefix is
    &quot;bsub&quot; if your cluster uses the LSF job scheduler, or
    &quot;qsub&quot; in Torque. Add arguments as necessary. For example,
    &quot;bsub -M 4GB&quot; sets a soft memory limit of 4GB.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2010-2011 Washington University in St. Louis.
<div class="Pp"></div>
It is released under the Lesser GNU Public License (LGPL) version 3. See the
  associated LICENSE file in this distribution.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Cyriac Kandoth, Ph.D.
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome-music-bmr</b>(1), <b>genome-music</b>(1), <b>genome</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
