<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:12:30 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GENOME-MUSIC-BMR-CALC-WIG-COVG(1p) User Contributed Perl
Documentation GENOME-MUSIC-BMR-CALC-WIG-COVG(1p)</p>

<p style="margin-top: 1em">genome music bmr calc-wig-covg
NAME <br>
genome music bmr calc-wig-covg - Count covered bases
per-gene for each given wiggle track format file.</p>

<p style="margin-top: 1em">VERSION <br>
This document describes genome music bmr calc-wig-covg
version 0.04 (2016-01-01 at 23:10:18)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
genome music bmr calc-wig-covg --roi-file=?
--reference-sequence=? --wig-list=? --output-dir=?</p>

<p style="margin-top: 1em">General usage:</p>

<p style="margin-top: 1em">... music bmr calc-wig-covg
--wig-list input_dir/wig_list --output-dir output_dir/
--reference-sequence input_dir/all_sequences.fa --roi-file
input_dir/all_coding_exons.tsv</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
roi-file Text <br>
Tab-delimited list of ROIs [chr start stop gene_name] (See
Description)</p>

<p style="margin-top: 1em">reference-sequence Text <br>
Path to reference sequence in FASTA format</p>

<p style="margin-top: 1em">wig-list Text <br>
Tab-delimited list of WIG files [sample_name wig_file] (See
Description)</p>

<p style="margin-top: 1em">output-dir Text <br>
Directory where output files and subdirectories will be
written</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script counts bases with sufficient coverage in the
ROIs of each gene from given wiggle track format files, and
categorizes them into - AT, CG (non-CpG), and CpG counts. It
<br>
also adds up these base-counts across all ROIs of each gene
for each sample, but covered bases that lie within
overlapping ROIs are not counted more than once towards
these total <br>
counts.</p>

<p style="margin-top: 1em">ARGUMENTS <br>
--roi-file <br>
The regions of interest (ROIs) of each gene are typically
regions targeted for sequencing or are merged exon loci
(from multiple transcripts) of genes with 2-bp flanks <br>
(splice junctions). For per-gene base counts, an overlapping
base will be counted each time it appears in an ROI of the
same gene. To avoid this, be sure to merge together <br>
overlapping ROIs of the same gene. BEDtools&rsquo; mergeBed
can help if used per gene. <br>
--reference-sequence <br>
The reference sequence in FASTA format. If a reference
sequence index is not found next to this file (a .fai file),
it will be created. <br>
--wig-list <br>
Provide a file containing sample names and the wiggle track
format file locations for each. Use the tab-delimited format
[sample_name wig_file] per line. Additional columns <br>
like clinical data are allowed, but ignored. The sample_name
must be the same as the tumor sample names used in the MAF
file (16th column, with the header <br>
Tumor_Sample_Barcode). <br>
--output-dir <br>
Specify an output directory where the following will be
created/written: roi_covgs: Subdirectory containing per-ROI
covered base counts for each sample. gene_covgs: <br>
Subdirectory containing per-gene covered base counts for
each sample. total_covgs: File containing the overall
non-overlapping coverages per sample.</p>

<p style="margin-top: 1em">LICENSE <br>
Copyright (C) 2010-2011 Washington University in St.
Louis.</p>

<p style="margin-top: 1em">It is released under the Lesser
GNU Public License (LGPL) version 3. See the associated
LICENSE file in this distribution.</p>

<p style="margin-top: 1em">AUTHORS <br>
Cyriac Kandoth, Ph.D.</p>

<p style="margin-top: 1em">SEE ALSO <br>
genome-music-bmr(1), genome-music(1), genome(1)</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GENOME-MUSIC-BMR-CALC-WIG-COVG(1p)</p>
<hr>
</body>
</html>
