<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:12:30 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GENOME-MUSIC-CLINICAL-CORRELATION(1p) User Contributed
Perl Documentation GENOME-MUSIC-CLINICAL-CORRELATION(1p)</p>

<p style="margin-top: 1em">genome music
clinical-correlation NAME <br>
genome music clinical-correlation - Correlate phenotypic
traits against mutated genes, or against individual
variants</p>

<p style="margin-top: 1em">VERSION <br>
This document describes genome music clinical-correlation
version 0.04 (2016-01-01 at 23:10:18)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
genome music clinical-correlation --bam-list=?
--output-file=? [--maf-file=?] [--glm-clinical-data-file=?]
[--use-maf-in-glm] [--skip-non-coding] [--skip-silent] <br>
[--clinical-correlation-matrix-file=?]
[--input-clinical-correlation-matrix-file=?]
[--genetic-data-type=?] [--numeric-clinical-data-file=?]
[--numerical-data-test-method=?] <br>
[--categorical-clinical-data-file=?]
[--glm-model-file=?]</p>

<p style="margin-top: 1em">... music clinical-correlation
--bam-list /path/myBamList.tsv --maf-file /path/myMAF.tsv
--numeric-clinical-data-file /path/myNumericData.tsv
--genetic-data-type &rsquo;gene&rsquo; --output-file
/path/output_file</p>

<p style="margin-top: 1em">... music clinical-correlation
--maf-file /path/myMAF.tsv --bam-list /path/myBamList.tsv
--numeric-clinical-data-file /path/myNumericData.tsv
--categorical-clinical-data-file /path/myClassData.tsv
--genetic-data-type &rsquo;gene&rsquo; --output-file
/path/output_file</p>

<p style="margin-top: 1em">... music clinical-correlation
--maf-file /path/myMAF.tsv --bam-list /path/myBamList.tsv
--output-file /path/output_file --glm-model-file
/path/model.tsv --glm-clinical-data-file
/path/glm_clinical_data.tsv --use-maf-in-glm</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
bam-list Text <br>
Tab delimited list of BAM files [sample_name, normal_bam,
tumor_bam] (See Description)</p>

<p style="margin-top: 1em">output-file Text <br>
Results of clinical-correlation tool. Will have suffix added
for data type</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
maf-file Text <br>
List of mutations using TCGA MAF specification v2.3</p>

<p style="margin-top: 1em">glm-clinical-data-file Text <br>
Clinical traits, mutational profiles, other mixed clinical
data (See DESCRIPTION)</p>

<p style="margin-top: 1em">use-maf-in-glm Boolean <br>
Create a variant matrix from the MAF file as variant input
to GLM analysis.</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--nouse-maf-in-glm) if not
specified</p>

<p style="margin-top: 1em">nouse-maf-in-glm Boolean <br>
Make use-maf-in-glm &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">skip-non-coding Boolean <br>
Skip non-coding mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-non-coding Boolean <br>
Make skip-non-coding &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">skip-silent Boolean <br>
Skip silent mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-silent Boolean <br>
Make skip-silent &rsquo;false&rsquo;</p>


<p style="margin-top: 1em">clinical-correlation-matrix-file
Text <br>
Specify a file to store the sample-vs-gene matrix created
during calculations</p>


<p style="margin-top: 1em">input-clinical-correlation-matrix-file
Text <br>
Instead of creating this from the MAF, input the
sample-vs-gene matrix for calculations</p>

<p style="margin-top: 1em">genetic-data-type Text <br>
Correlate clinical data to &quot;gene&quot; or
&quot;variant&quot; level data</p>

<p style="margin-top: 1em">Default value &rsquo;gene&rsquo;
if not specified</p>

<p style="margin-top: 1em">numeric-clinical-data-file Text
<br>
Table of samples (y) vs. numeric clinical data category
(x)</p>

<p style="margin-top: 1em">numerical-data-test-method Text
<br>
Either &rsquo;cor&rsquo; for Pearson Correlation or
&rsquo;wilcox&rsquo; for the Wilcoxon Rank-Sum Test for
numerical clinical data</p>

<p style="margin-top: 1em">Default value &rsquo;cor&rsquo;
if not specified</p>

<p style="margin-top: 1em">categorical-clinical-data-file
Text <br>
Table of samples (y) vs. categorical clinical data category
(x)</p>

<p style="margin-top: 1em">glm-model-file Text <br>
File outlining the type of model, response variable,
covariants, etc. for the GLM analysis. (See DESCRIPTION)</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This command relates clinical traits and mutational data.
Either one can perform correlation analysis between
mutations recorded in a MAF and the particular phenotypic
traits <br>
recorded in clinical data files for the same samples, or one
can run a generalized linear model (GLM) analysis on the
same types of data.</p>

<p style="margin-top: 1em">The clinical data files for
correlation must be separated between numeric and categoric
data and must follow these conventions:</p>

<p style="margin-top: 1em">&Acirc;&middot; Headers are
required</p>

<p style="margin-top: 1em">&Acirc;&middot; Each file must
include at least 1 sample_id column and 1 attribute column,
with the format being [sample_id clinical_data_attribute_1
clinical_data_attribute_2 ...]</p>

<p style="margin-top: 1em">&Acirc;&middot; The sample ID
must match the sample ID listed in the MAF under
&quot;Tumor_Sample_Barcode&quot; for relating the mutations
of this sample.</p>

<p style="margin-top: 1em">Note the importance of the
headers: the header for each clinical_data_attribute will
appear in the output file to denote relationships with the
mutation data from the MAF.</p>

<p style="margin-top: 1em">Internally, the input data is
fed into an R script which calculates a P-value representing
the probability that the correlation seen between the
mutations in each gene (or <br>
variant) and each phenotype trait are random. Lower P-values
indicate lower randomness, or likely true correlations.</p>

<p style="margin-top: 1em">The results are saved to the
output filename given with a suffix appended;
&quot;.numeric.csv&quot; will be appended for results
derived from numeric clinical data, and
&quot;.categorical.csv&quot; <br>
will be appended for results derived from categorical
clinical data. Also, &quot;.glm.csv&quot; will be appended
to the output filename for GLM results.</p>

<p style="margin-top: 1em">The GLM analysis accepts a mixed
numeric and categoric clinical data file, input using the
parameter --glm-clinical-data-file. GLM clinical data must
adhere to the formats <br>
described above for the correlation clinical data files. GLM
also requires the user to input a --glm-model-file. This
file requires specific headers and defines the analysis to
<br>
be performed rather exactly. Here are the conventions
required for this file:</p>

<p style="margin-top: 1em">&Acirc;&middot; Columns must be
ordered as such:</p>

<p style="margin-top: 1em">&Acirc;&middot; [ analysis_type
clinical_data_trait_name variant/gene_name covariates memo
]</p>

<p style="margin-top: 1em">&Acirc;&middot; The
&rsquo;analysis_type&rsquo; column must contain either
&quot;Q&quot;, indicating a quantative trait, or
&quot;B&quot;, indicating a binary trait will be
examined.</p>

<p style="margin-top: 1em">&Acirc;&middot; The
&rsquo;clinical_data_trait_name&rsquo; is the name of a
clinical data trait defined by being a header in the
--glm-clinical-data-file.</p>

<p style="margin-top: 1em">&Acirc;&middot; The
&rsquo;variant/gene_name&rsquo; can either be the name of
one or more columns from the --glm-clinical-data-file, or
the name of one or more mutated gene names from the MAF,
separated <br>
by &quot;|&quot;. If this column is left blank, or instead
contains &quot;NA&quot;, then each column from either the
variant mutation matrix (--use-maf-in-glm) or alternatively
the <br>
--glm-clinical-data-file is used consecutively as the
variant column in independent analyses.</p>

<p style="margin-top: 1em">&Acirc;&middot;
&rsquo;covariates&rsquo; are the names of one or more
columns from the --glm-clinical-data-file, separated by
&quot;+&quot;.</p>

<p style="margin-top: 1em">&Acirc;&middot;
&rsquo;memo&rsquo; is any note deemed useful to the user. It
will be printed in the output data file for reference.</p>

<p style="margin-top: 1em">GLM analysis may be performed
using solely the data input into --glm-clinical-data-file,
as described above, or alternatively, mutational data from
the MAF may be included as <br>
variants in the GLM analysis, as also described above. Use
the --use-maf-in-glm flag to include the mutation matrix
derived from the maf as variant data.</p>

<p style="margin-top: 1em">Note that all input files for
both correlation and GLM analysis must be tab-separated.</p>

<p style="margin-top: 1em">ARGUMENTS <br>
--bam-list <br>
Provide a file containing sample names and normal/tumor BAM
locations for each. Use the tab- delimited format
[sample_name normal_bam tumor_bam] per line. This tool only
<br>
needs sample_name, so all other columns can be skipped. The
sample_name must be the same as the tumor sample names used
in the MAF file (16th column, with the header <br>
Tumor_Sample_Barcode).</p>

<p style="margin-top: 1em">LICENSE <br>
Copyright (C) 2010-2011 Washington University in St.
Louis.</p>

<p style="margin-top: 1em">It is released under the Lesser
GNU Public License (LGPL) version 3. See the associated
LICENSE file in this distribution.</p>

<p style="margin-top: 1em">AUTHORS <br>
Nathan D. Dees, Ph.D. <br>
Qunyuan Zhang, Ph.D. <br>
William Schierding, M.S.</p>

<p style="margin-top: 1em">SEE ALSO <br>
genome-music(1), genome(1)</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GENOME-MUSIC-CLINICAL-CORRELATION(1p)</p>
<hr>
</body>
</html>
