<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-COSMIC-OMIM(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-COSMIC-OMIM(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-COSMIC-OMIM(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_cosmic-omim"><a class="selflink" href="#genome_music_cosmic-omim">genome
  music cosmic-omim</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music cosmic-omim - Compare the amino acid changes of supplied mutations
  to COSMIC and OMIM databases.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music cosmic-omim (2016-01-01 at 23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music cosmic-omim --maf-file=? --output-file=? --reference-build=?
  [--omimaa-dir=?] [--cosmic-dir=?] [--verbose] [--wu-annotation-headers]
  [--aa-range=?] [--nuc-range=?] [--show-known-hits]
<div class="Pp"></div>
<pre>
 ... music cosmic-omim \
        --maf-file input_dir/myMAF.tsv \
        --output-file output_dir/myMAF_output.tsv \
        --no-verbose
 ... music cosmic-omim \
        --maf-file input_dir/myMAF.tsv \
        --output-file output_dir/myMAF_output.tsv \
        --omimaa-dir omim_dir/ \
        --cosmic-dir cosmic_dir/ \
        --no-verbose
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">maf-file <i>Path</i></dt>
  <dd class="It-tag">list of annotated mutations in MAF format (or any file with
      MAF+annotation headers)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-file <i>Path</i></dt>
  <dd class="It-tag">Output file contains the input file with two columns
      appended to the end, corresponding to cosmic and omim mutation
      comparisons, respectively</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reference-build <i>Text</i></dt>
  <dd class="It-tag">Put either &quot;Build36&quot; or &quot;Build37&quot;
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'Build37' if not specified</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">omimaa-dir <i>Path</i></dt>
  <dd class="It-tag">omim amino acid mutation database folder</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">cosmic-dir <i>Path</i></dt>
  <dd class="It-tag">cosmic amino acid mutation database folder</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">verbose <i>Boolean</i></dt>
  <dd class="It-tag">Use this to display the larger working output
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--noverbose) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noverbose <i>Boolean</i></dt>
  <dd class="It-tag">Make verbose 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">wu-annotation-headers <i>Boolean</i></dt>
  <dd class="It-tag">Use this if input MAF contains WUSTL annotation format
      headers
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--nowu-annotation-headers) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">nowu-annotation-headers <i>Boolean</i></dt>
  <dd class="It-tag">Make wu-annotation-headers 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">aa-range <i>Integer</i></dt>
  <dd class="It-tag">Set how close a 'near' match is when searching for amino
      acid near hits
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '2' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">nuc-range <i>Integer</i></dt>
  <dd class="It-tag">Set how close a 'near' match is when searching for
      nucleotide position near hits
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '5' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">show-known-hits <i>Boolean</i></dt>
  <dd class="It-tag">When a finding is novel, show known AA in that gene
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noshow-known-hits <i>Boolean</i></dt>
  <dd class="It-tag">Make show-known-hits 'false'</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This tool looks at the amino acid changes for the given set of mutations and
  compares the genomic coordinates as well as the affected amino acid to the
  coordinates and amino acids of all cancer-specific mutations listed in the
  Cosmic and OMIM databases. The database files are specially prepared for this
  task and provided with the MuSiC suite. The tool reports various types of
  matches, including matches within &quot;near proximity&quot;, where &quot;near
  proximity&quot; is currently defined as a linear DNA distance of 5 bases or 2
  amino acids. (This type of matching helps to account for the possibility of
  subtle differences in reported positions for variants due to differences in
  transcript definitions or other things of this nature.) Any site without a
  match in a particular databases is reported as &quot;novel&quot; with respect
  to that database.
<div class="Pp"></div>
The output of this script returns each row the original input MAF file with two
  columns appended to the end of each, one column for each of the databases.
  Also included is a STDOUT printout of a summary of what was found in the input
  MAF. Neither output can be suppressed in the current version. The --verbose
  option is used to display working notes that are useful for various purposes
  in debugging potential MAF problems. The Omim and Cosmic directories must
  point to the output of the downloader, named appropriately, as they don't
  recognize the OMIM database in the raw download format.
<div class="Pp"></div>
This tool only compares build 36 or build 37 coordinates that are specified in
  Cosmic to the coordinates in your maf file. This is a weakness of the Cosmic
  database (not all position entries are currently available for both builds)
  that is out of our control.
<div class="Pp"></div>
In addition to the standard version 2.3 MAF headers, there needs to be 3 columns
  appended. These column headers in the MAF must have these names in the header
  in order for the tool to find them:
<br/>
 transcript_name - the transcript name, such as NM_000028 amino_acid_change -
  the amino acid change, such as p.R290H
<br/>
 c_position - the nucleotide position changed, such as c.869
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2010-2011 Washington University in St. Louis.
<div class="Pp"></div>
It is released under the Lesser GNU Public License (LGPL) version 3. See the
  associated LICENSE file in this distribution.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Brian Dunford-Shore
 David Larson, Ph.D.
 Michael C. Wendl, Ph.D.
 William Schierding, M.S.
</pre>
<h1 class="Sh" title="Sh" id="CREDITS"><a class="selflink" href="#CREDITS">CREDITS</a></h1>
This tool depends on copies of data from the following databases, packaged in a
  form useable for quick analysis:
<div class="Pp"></div>
<pre>
 * COSMIC - http://www.sanger.ac.uk/genetics/CGP/cosmic/
 * OMIM - http://www.ncbi.nlm.nih.gov/omim
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome-music</b>(1), <b>genome</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
