<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-BMR-CALC-COVG-HELPER(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-BMR-CALC-COVG-HELPER(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-BMR-CALC-COVG-HELPER(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_bmr_calc-covg-helper"><a class="selflink" href="#genome_music_bmr_calc-covg-helper">genome
  music bmr calc-covg-helper</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music bmr calc-covg-helper - Uses calcRoiCovg.c to count covered bases
  per-gene for a tumor-normal pair of BAMs.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music bmr calc-covg-helper version 0.04
  (2016-01-01 at 23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music bmr calc-covg-helper --roi-file=? --reference-sequence=?
  --normal-tumor-bam-pair=? [--output-file=?] [--output-dir=?]
  [--normal-min-depth=?] [--tumor-min-depth=?] [--min-mapq=?]
<div class="Pp"></div>
General usage:
<div class="Pp"></div>
<pre>
 ... music bmr calc-covg-helper \
    --normal-tumor-bam-pair &quot;sample-name path/to/normal_bam path/to/tumor_bam&quot; \
    --reference-sequence input_dir/all_sequences.fa \
    --output-file output_file \
    --roi-file input_dir/all_coding_exons.tsv
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">roi-file <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of ROIs [chr start stop gene_name] (See
      Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reference-sequence <i>Text</i></dt>
  <dd class="It-tag">Path to reference sequence in FASTA format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">normal-tumor-bam-pair <i>Text</i></dt>
  <dd class="It-tag">Tab delimited line with sample name, path to normal bam
      file, and path to tumor bam file (See Description)</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">output-file <i>Text</i></dt>
  <dd class="It-tag">Output file path. Specify either output-file or
      output-directory.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-dir <i>Text</i></dt>
  <dd class="It-tag">Output directory path. Specify either output-file or
      output-directory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">normal-min-depth <i>Integer</i></dt>
  <dd class="It-tag">The minimum read depth to consider a Normal BAM base as
      covered
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '6' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">tumor-min-depth <i>Integer</i></dt>
  <dd class="It-tag">The minimum read depth to consider a Tumor BAM base as
      covered
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '8' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">min-mapq <i>Integer</i></dt>
  <dd class="It-tag">The minimum mapping quality of reads to consider towards
      read depth counts
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '20' if not specified</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script counts bases with sufficient coverage in the ROIs of each gene in
  the given pair of tumor-normal BAM files and categorizes them into - AT, CG
  (non-CpG), and CpG counts. It also adds up these base-counts across all ROIs
  of each gene in the sample, but covered bases that lie within overlapping ROIs
  are not counted more than once towards these total counts.
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--roi-file</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The regions of interest (ROIs) of each gene are typically
    regions targeted for sequencing or are merged exon loci (from multiple
    transcripts) of genes with 2-bp flanks (splice junctions). ROIs from the
    same chromosome must be listed adjacent to each other in this file. This
    allows the underlying C-based code to run much more efficiently and avoid
    re-counting bases seen in overlapping ROIs (for overall covered base
    counts). For per-gene base counts, an overlapping base will be counted each
    time it appears in an ROI of the same gene. To avoid this, be sure to merge
    together overlapping ROIs of the same gene. BEDtools' mergeBed can help if
    used per gene.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--reference-sequence</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The reference sequence in FASTA format. If a reference
    sequence index is not found next to this file (a .fai file), it will be
    created.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--normal-tumor-bam-pair</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">&quot;sample-name path/to/normal_bam
    path/to/tumor_bam&quot;</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--output-file</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Specify an output file where the per-ROI covered base
    counts will be written</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2010-2011 Washington University in St. Louis.
<div class="Pp"></div>
It is released under the Lesser GNU Public License (LGPL) version 3. See the
  associated LICENSE file in this distribution.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Cyriac Kandoth, Ph.D.
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome-music-bmr</b>(1), <b>genome-music</b>(1), <b>genome</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
