<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-GALAXY(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-GALAXY(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-GALAXY(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_galaxy"><a class="selflink" href="#genome_music_galaxy">genome
  music galaxy</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music galaxy - Run the full suite of MuSiC tools sequentially.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music galaxy version 0.04 (2016-01-01 at
  23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music galaxy [--output-dir=?] --bam-list=? --output-bundle=? --roi-file=?
  --reference-sequence=? --maf-file=? --pathway-file=?
  [--numeric-clinical-data-file=?] [--categorical-clinical-data-file=?]
  [--mutation-matrix-file=?] [--permutations=?] [--normal-min-depth=?]
  [--tumor-min-depth=?] [--min-mapq=?] [--show-skipped] [--genes-to-ignore=?]
  [--bmr=?] [--max-proximity=?] [--bmr-modifier-file=?]
  [--numerical-data-test-method=?] [--skip-low-mr-genes] [--max-fdr=?]
  [--genetic-data-type=?] [--wu-annotation-headers] [--bmr-groups=?]
  [--separate-truncations] [--merge-concurrent-muts] [--skip-non-coding]
  [--skip-silent] [--min-mut-genes-per-path=?] [--glm-model-file=?]
  [--processors=?] [--aa-range=?] [--nuc-range=?] [--reference-build=?]
  [--show-known-hits] [--glm-clinical-data-file=?] [--use-maf-in-glm]
  [--omimaa-dir=?] [--cosmic-dir=?] [--verbose]
  [--clinical-correlation-matrix-file=?]
<div class="Pp"></div>
This tool takes as parameters all the information required to run the individual
  tools. An example usage is:
<div class="Pp"></div>
<pre>
 ... music play \
        --bam-list input/bams_to_analyze.txt \
        --numeric-clinical-data-file input/numeric_clinical_data.csv \
        --maf-file input/myMAF.tsv \
        --output-dir play_output_dir \
        --pathway-file input/pathway_db \
        --reference-sequence input/refseq/all_sequences.fa \
        --roi-file input/all_coding_regions.bed \
        --genetic-data-type gene
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">bam-list <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of BAM files [sample_name normal_bam
      tumor_bam]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-bundle <i>Text</i></dt>
  <dd class="It-tag">Location where Galaxy would like the bundle of Music
      outputs to be saved</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">roi-file <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of ROIs [chr start stop gene_name]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reference-sequence <i>Text</i></dt>
  <dd class="It-tag">Path to reference sequence in FASTA format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">maf-file <i>Text</i></dt>
  <dd class="It-tag">List of mutations using TCGA MAF specifications v2.3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">pathway-file <i>Text</i></dt>
  <dd class="It-tag">Tab-delimited file of pathway information</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">output-dir</dt>
  <dd class="It-tag">Directory where output files and subdirectories will be
      written
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">numeric-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Table of samples (y) vs. numeric clinical data category
    (x)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">categorical-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Table of samples (y) vs. categorical clinical data category
      (x)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">mutation-matrix-file <i>Text</i></dt>
  <dd class="It-tag">Optionally store the sample-vs-gene matrix used during
      calculations.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">permutations <i>Number</i></dt>
  <dd class="It-tag">Number of permutations used to determine P-values</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">normal-min-depth <i>Integer</i></dt>
  <dd class="It-tag">The minimum read depth to consider a Normal BAM base as
      covered</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">tumor-min-depth <i>Integer</i></dt>
  <dd class="It-tag">The minimum read depth to consider a Tumor BAM base as
      covered</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">min-mapq <i>Integer</i></dt>
  <dd class="It-tag">The minimum mapping quality of reads to consider towards
      read depth counts</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">show-skipped <i>Boolean</i></dt>
  <dd class="It-tag">Report each skipped mutation, not just how many
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--noshow-skipped) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noshow-skipped <i>Boolean</i></dt>
  <dd class="It-tag">Make show-skipped 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">genes-to-ignore <i>Text</i></dt>
  <dd class="It-tag">Comma-delimited list of genes to ignore for background
      mutation rates</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bmr <i>Number</i></dt>
  <dd class="It-tag">Background mutation rate in the targeted regions</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">max-proximity <i>Text</i></dt>
  <dd class="It-tag">Maximum AA distance between 2 mutations</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bmr-modifier-file <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of values per gene that modify BMR
      before testing [gene_name bmr_modifier]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">numerical-data-test-method <i>Text</i></dt>
  <dd class="It-tag">Either 'cor' for Pearson Correlation or 'wilcox' for the
      Wilcoxon Rank-Sum Test for numerical clinical data.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'cor' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-low-mr-genes <i>Boolean</i></dt>
  <dd class="It-tag">Skip testing genes with MRs lower than the background MR
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-low-mr-genes <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-low-mr-genes 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">max-fdr <i>Number</i></dt>
  <dd class="It-tag">The maximum allowed false discovery rate for a gene to be
      considered an SMG
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '0.2' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">genetic-data-type <i>Text</i></dt>
  <dd class="It-tag">Data in matrix file must be either &quot;gene&quot; or
      &quot;variant&quot; type data</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">wu-annotation-headers <i>Boolean</i></dt>
  <dd class="It-tag">Use this to default to wustl annotation format headers</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">nowu-annotation-headers <i>Boolean</i></dt>
  <dd class="It-tag">Make wu-annotation-headers 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bmr-groups <i>Integer</i></dt>
  <dd class="It-tag">Number of clusters of samples with comparable BMRs
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '1' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">separate-truncations <i>Boolean</i></dt>
  <dd class="It-tag">Group truncational mutations as a separate category
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--noseparate-truncations) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noseparate-truncations <i>Boolean</i></dt>
  <dd class="It-tag">Make separate-truncations 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">merge-concurrent-muts <i>Boolean</i></dt>
  <dd class="It-tag">Multiple mutations of a gene in the same sample are treated
      as 1
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--nomerge-concurrent-muts) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">nomerge-concurrent-muts <i>Boolean</i></dt>
  <dd class="It-tag">Make merge-concurrent-muts 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Skip non-coding mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-non-coding 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Skip silent mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-silent 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">min-mut-genes-per-path <i>Integer</i></dt>
  <dd class="It-tag">Pathways with fewer mutated genes than this will be ignored
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '1' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">glm-model-file <i>Text</i></dt>
  <dd class="It-tag">File outlining the type of model, response variable,
      covariants, etc. for the GLM analysis. (See DESCRIPTION).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">processors <i>Integer</i></dt>
  <dd class="It-tag">Number of processors to use in SMG (requires 'foreach' and
      'doMC' R packages)
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '1' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">aa-range <i>Integer</i></dt>
  <dd class="It-tag">Set how close a 'near' match is when searching for amino
      acid near hits
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '2' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">nuc-range <i>Integer</i></dt>
  <dd class="It-tag">Set how close a 'near' match is when searching for
      nucleotide position near hits
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '5' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reference-build <i>Text</i></dt>
  <dd class="It-tag">Put either &quot;Build36&quot; or &quot;Build37&quot;
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'Build37' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">show-known-hits <i>Boolean</i></dt>
  <dd class="It-tag">When a finding is novel, show known AA in that gene
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noshow-known-hits <i>Boolean</i></dt>
  <dd class="It-tag">Make show-known-hits 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">glm-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Clinical traits, mutational profiles, other mixed clinical
      data (See DESCRIPTION).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">use-maf-in-glm <i>Boolean</i></dt>
  <dd class="It-tag">Set this flag to use the variant matrix created from the
      MAF file as variant input to GLM analysis.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--nouse-maf-in-glm) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">nouse-maf-in-glm <i>Boolean</i></dt>
  <dd class="It-tag">Make use-maf-in-glm 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">omimaa-dir <i>Path</i></dt>
  <dd class="It-tag">omim amino acid mutation database folder</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">cosmic-dir <i>Path</i></dt>
  <dd class="It-tag">cosmic amino acid mutation database folder</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">verbose <i>Boolean</i></dt>
  <dd class="It-tag">turn on to display larger working output
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noverbose <i>Boolean</i></dt>
  <dd class="It-tag">Make verbose 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">clinical-correlation-matrix-file <i>Text</i></dt>
  <dd class="It-tag">Optionally store the sample-vs-gene matrix used internally
      during calculations.</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This command can be used to run all of the MuSiC analysis tools on a set of
  data. Please see the individual tools for further description of the
  parameters.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Thomas B. Mooney, M.S.
</pre>
<h1 class="Sh" title="Sh" id="CREDITS"><a class="selflink" href="#CREDITS">CREDITS</a></h1>
Please see the credits for <b>genome-music</b>(1).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome-music</b>(1), <b>genome-music-path-scan</b>(1),
  <b>genome-music-smg</b>(1), <b>genome-music-clinical-correlation</b>(1),
  <b>genome-music-mutation-relation</b>(1), <b>genome-music-cosmic-omim</b>(1),
  <b>genome-music-proximity</b>(1), <b>genome-music-pfam</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
