<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-MUTATION-RELATION(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-MUTATION-RELATION(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-MUTATION-RELATION(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_mutation-relation"><a class="selflink" href="#genome_music_mutation-relation">genome
  music mutation-relation</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music mutation-relation - Identify relationships of mutation concurrency
  or mutual exclusivity in genes across cases.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music mutation-relation version 0.04 (2016-01-01
  at 23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music mutation-relation --bam-list=? --maf-file=? --output-file=?
  [--mutation-matrix-file=?] [--permutations=?] [--gene-list=?]
  [--skip-non-coding] [--skip-silent]
<div class="Pp"></div>
<pre>
 ... music mutation-relation \
        --maf-file /path/myMAF.tsv \
        --permutations 1000 \
        --output-file /path/mutation_relation.csv
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">bam-list <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of BAM files [sample_name, normal_bam,
      tumor_bam] (See Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">maf-file <i>Text</i></dt>
  <dd class="It-tag">List of mutations in MAF format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-file <i>Text</i></dt>
  <dd class="It-tag">Results of mutation-relation tool</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">mutation-matrix-file <i>Text</i></dt>
  <dd class="It-tag">Optionally store the sample-vs-gene matrix used during
      calculations.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">permutations <i>Number</i></dt>
  <dd class="It-tag">Number of permutations used to determine P-values
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '100' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">gene-list <i>Text</i></dt>
  <dd class="It-tag">List of genes to test, typically SMGs. If unspecified, all
      genes in MAF are tested.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Skip non-coding mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-non-coding 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Skip silent mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-silent 'false'</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This module parses a list of mutations in MAF format and attempts to determine
  relationships among mutated genes. It employs a correlation test to see
  whether or not any two genes are mutated concurrently (positive correlation)
  or mutually exclusively (negative correlation). Because of the possibility of
  largely varying numbers of mutations present in different genes, P-values are
  calculated using restricted permutations that take into account the
  distribution of mutation counts among the samples. In the output file, 'pand'
  is the P-value for concurrent mutation events, and 'pexc' is the P-value for
  mutually exclusive mutation events.
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--bam-list</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Provide a file containing sample names and normal/tumor BAM
    locations for each. Use the tab- delimited format [sample_name normal_bam
    tumor_bam] per line. This tool only needs sample_name, so all other columns
    can be skipped. The sample_name must be the same as the tumor sample names
    used in the MAF file (16th column, with the header
    Tumor_Sample_Barcode).</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Nathan D. Dees, Ph.D.
 Qunyuan Zhang, Ph.D.
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
