<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC-PROXIMITY(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC-PROXIMITY(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC-PROXIMITY(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music_proximity"><a class="selflink" href="#genome_music_proximity">genome
  music proximity</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music proximity - Perform a proximity analysis on a list of mutations.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music proximity version 0.04 (2016-01-01 at
  23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music proximity --maf-file=? --output-file=? --output-dir=?
  [--max-proximity=?] [--skip-non-coding] [--skip-silent]
<div class="Pp"></div>
<pre>
 ... music proximity \
        --maf-file input_dir/myMAF.tsv \
        --output-dir output_dir/ \
        --max-proximity 15
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">maf-file <i>Text</i></dt>
  <dd class="It-tag">List of mutations using TCGA MAF specifications v2.3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-file <i>Text</i></dt>
  <dd class="It-tag">TODO</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-dir <i>Text</i></dt>
  <dd class="It-tag">Directory where output files will be written</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">max-proximity <i>Text</i></dt>
  <dd class="It-tag">Maximum allowed AA distance between 2 mutations
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '7' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Skip non-coding mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-non-coding 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Skip silent mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-silent 'false'</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This module first calculates the amino acid position of each mutation in the MAF
  file within its respective transcript. Then, for each mutation, two values are
  calculated: 1) the number of other mutations on the same transcript within the
  proximity limit set by the max-proximity input parameter, and 2) the distance
  to the closest other mutation in this nearby set. Only mutations which have
  another mutation within close proximity are reported in the output-file.
<div class="Pp"></div>
In addition to the standard version 2.3 MAF headers, there needs to be 3 columns
  appended. These column headers in the MAF must have these names in the header
  in order for the tool to find them:
<br/>
 transcript_name - the transcript name, such as NM_000028 amino_acid_change -
  the amino acid change, such as p.R290H
<br/>
 c_position - the nucleotide position changed, such as c.869
<div class="Pp"></div>
The output is generated with the folowing column headers:
  Mutations_Within_Proximity, Nearest_Mutation, Gene, Transcript,
  Affected_Amino_Acid(s), Chr, Start, Stop, Ref_Allele, Var_Allele, Sample
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Nathan D. Dees, Ph.D.
 Dan Koboldt, M.S.
 Cyriac Kandoth, Ph.D.
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
