<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:12:29 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GENOME-MUSIC-BMR-CALC-BMR(1p) User Contributed Perl
Documentation GENOME-MUSIC-BMR-CALC-BMR(1p)</p>

<p style="margin-top: 1em">genome music bmr calc-bmr NAME
<br>
genome music bmr calc-bmr - Calculates mutation rates given
per-gene coverage (from &quot;music bmr calc-covg&quot;),
and a mutation list</p>

<p style="margin-top: 1em">VERSION <br>
This document describes genome music bmr calc-bmr version
0.04 (2016-01-01 at 23:10:18)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
genome music bmr calc-bmr --bmr-output=? --roi-file=?
--gene-mr-file=? --reference-sequence=? --bam-list=?
--output-dir=? --maf-file=? [--skip-non-coding]
[--skip-silent] <br>
[--bmr-groups=?] [--show-skipped] [--separate-truncations]
[--merge-concurrent-muts] [--genes-to-ignore=?]</p>

<p style="margin-top: 1em">... music bmr calc-bmr
--bam-list input_dir/bam_list --maf-file input_dir/myMAF.tsv
--output-dir output_dir/ --reference-sequence
input_dir/all_sequences.fa --roi-file
input_dir/all_coding_exons.tsv</p>

<p style="margin-top: 1em">... music bmr calc-bmr
--bam-list input_dir/bam_list --maf-file input_dir/myMAF.tsv
--output-dir output_dir/ --reference-sequence
input_dir/all_sequences.fa --roi-file
input_dir/all_coding_exons.tsv --genes-to-ignore
GENE1,GENE2</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
bmr-output Number <br>
TODO</p>

<p style="margin-top: 1em">roi-file Text <br>
Tab delimited list of ROIs [chr start stop gene_name] (See
DESCRIPTION)</p>

<p style="margin-top: 1em">gene-mr-file Text <br>
TODO</p>

<p style="margin-top: 1em">reference-sequence Text <br>
Path to reference sequence in FASTA format</p>

<p style="margin-top: 1em">bam-list Text <br>
Tab delimited list of BAM files [sample_name normal_bam
tumor_bam] (See DESCRIPTION)</p>

<p style="margin-top: 1em">output-dir Text <br>
Directory where output files will be written (Use the same
one used with calc-covg)</p>

<p style="margin-top: 1em">maf-file Text <br>
List of mutations using TCGA MAF specification v2.3</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
skip-non-coding Boolean <br>
Skip non-coding mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-non-coding Boolean <br>
Make skip-non-coding &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">skip-silent Boolean <br>
Skip silent mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-silent Boolean <br>
Make skip-silent &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">bmr-groups Integer <br>
Number of clusters of samples with comparable BMRs (See
DESCRIPTION)</p>

<p style="margin-top: 1em">Default value &rsquo;1&rsquo; if
not specified</p>

<p style="margin-top: 1em">show-skipped Boolean <br>
Report each skipped mutation, not just how many</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--noshow-skipped) if not specified</p>

<p style="margin-top: 1em">noshow-skipped Boolean <br>
Make show-skipped &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">separate-truncations Boolean
<br>
Group truncational mutations as a separate category</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--noseparate-truncations) if not
specified</p>

<p style="margin-top: 1em">noseparate-truncations Boolean
<br>
Make separate-truncations &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">merge-concurrent-muts Boolean
<br>
Multiple mutations of a gene in the same sample are treated
as 1</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--nomerge-concurrent-muts) if not
specified</p>

<p style="margin-top: 1em">nomerge-concurrent-muts Boolean
<br>
Make merge-concurrent-muts &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">genes-to-ignore Text <br>
Comma-delimited list of genes to ignore for background
mutation rates</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Given a mutation list (MAF), and per-gene coverage data
calculated using &quot;music bmr calc-covg&quot;), this
script calculates overall Background Mutation Rate (BMR) and
BMRs in the <br>
categories of AT/CG/CpG Transitions, AT/CG/CpG
Transversions, and Indels. An optional category for
truncational mutations can also be specified. The script
generates a file with <br>
per-gene mutation rates that can be used with the tool that
tests for significantly mutated genes (music smg).</p>

<p style="margin-top: 1em">ARGUMENTS <br>
--roi-file <br>
The regions of interest (ROIs) of each gene are typically
regions targeted for sequencing or are merged exon loci
(from multiple transcripts) of genes with 2-bp flanks <br>
(splice junctions). ROIs from the same chromosome must be
listed adjacent to each other in this file. This allows the
underlying C-based code to run much more efficiently and
<br>
avoid re-counting bases seen in overlapping ROIs (for
overall covered base counts). For per-gene base counts, an
overlapping base will be counted each time it appears in an
<br>
ROI of the same gene. To avoid this, be sure to merge
together overlapping ROIs of the same gene. BEDtools&rsquo;
mergeBed can help if used per gene. <br>
--reference-sequence <br>
The reference sequence in FASTA format. If a reference
sequence index is not found next to this file (a .fai file),
it will be created. <br>
--bam-list <br>
Provide a file containing sample names and normal/tumor BAM
locations for each. Use the tab- delimited format
[sample_name normal_bam tumor_bam] per line. Additional
columns <br>
like clinical data are allowed, but ignored. The sample_name
must be the same as the tumor sample names used in the MAF
file (16th column, with the header <br>
Tumor_Sample_Barcode). <br>
--bmr-groups <br>
Ideally, we want to test the mutation rate (MR) of a gene in
a sample against the background mutation rate (BMR) across
that sample. But if the BMRs of some samples are <br>
comparable, we can instead test the MR of a gene across a
group of samples with comparable BMR, against the overall
BMR of that group. This argument specifies how many such
<br>
groups you want to cluster samples into. By default, it is
assumed that all samples have comparable BMRs (bmr-groups =
1). <br>
--output-dir <br>
This should be the same output directory used when running
&quot;music bmr calc-covg&quot;. The following outputs of
this script will also be created/written: overall_bmrs: File
<br>
containing categorized overall background mutation rates.
gene_mrs: File containing categorized per-gene mutation
rates. <br>
--genes-to-ignore <br>
A comma-delimited list of genes to ignore for overall BMR
calculations. List genes that are known factors in this
disease and whose mutations should not be classified as <br>
background.</p>

<p style="margin-top: 1em">LICENSE <br>
Copyright (C) 2010-2011 Washington University in St.
Louis.</p>

<p style="margin-top: 1em">It is released under the Lesser
GNU Public License (LGPL) version 3. See the associated
LICENSE file in this distribution.</p>

<p style="margin-top: 1em">AUTHORS <br>
Cyriac Kandoth, Ph.D.</p>

<p style="margin-top: 1em">SEE ALSO <br>
genome-music-bmr(1), genome-music(1), genome(1)</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GENOME-MUSIC-BMR-CALC-BMR(1p)</p>
<hr>
</body>
</html>
