<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GENOME-MUSIC(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GENOME-MUSIC(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GENOME-MUSIC(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="genome_music"><a class="selflink" href="#genome_music">genome
  music</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
genome music - Mutational Significance in Cancer (Cancer Mutation Analysis)
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes genome music version 0.04 (2016-01-01 at 23:10:18)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
genome music ...
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
The MuSiC suite is a set of tools aimed at discovering the significance of
  somatic mutations found within a given cohort of cancer samples, and with
  respect to a variety of external data sources. The standard inputs required
  are:
<dl class="Bl-tag">
  <dt class="It-tag">1. mapped reads in BAM format</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">2. predicted or validated SNVs or indels in mutation
    annotation format (MAF)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">3. a list of regions of interest (typically the boundaries
    of coding exons)</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">4. any relevant numeric or categorical clinical data.</dt>
  <dd class="It-tag"></dd>
</dl>
<div class="Pp"></div>
The formats for inputs 3. and 4. are:
<dl class="Bl-tag">
  <dt class="It-tag">3. Regions of Interest File:</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Do not use headers</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">4 columns, which are [chromosome start-position(1-based)
      stop-position(1-based) gene_name]</dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">4. Clinical Data Files:</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Headers are required</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">At least 1 sample_id column and 1 attribute column, with
      the format being [sample_id clinical_data_attribute
      clinical_data_attribute ...]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The sample_id must match the sample_id listed in the MAF
      under &quot;Tumor_Sample_Barcode&quot; for relating the mutations of this
      sample.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The header for each clinical_data_attribute will appear in
      the output file to denote relationships with the mutation data from the
      MAF.</dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
Descriptions for the usage of each tool (each sub-command) can be found
  separately.
<div class="Pp"></div>
The <b>play</b> command runs all of the sub-commands serially on a selected
  input set.
<h1 class="Sh" title="Sh" id="SUB-COMMANDS"><a class="selflink" href="#SUB-COMMANDS">SUB-COMMANDS</a></h1>
<h2 class="Ss" title="Ss" id="GENERAL"><a class="selflink" href="#GENERAL">GENERAL</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>bmr</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Calculate gene coverages and background mutation
    rates.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>clinical-correlation</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Correlate phenotypic traits against mutated genes, or
    against individual variants</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cosmic-omim</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Compare the amino acid changes of supplied mutations to
    COSMIC and OMIM databases.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>galaxy</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Run the full suite of MuSiC tools sequentially.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mutation-relation</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Identify relationships of mutation concurrency or mutual
    exclusivity in genes across cases.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>path-scan</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Find signifcantly mutated pathways in a cohort given a list
    of somatic mutations.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pfam</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Add Pfam annotation to a MAF file</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>play</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Run the full suite of MuSiC tools sequentially.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>plot</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Generate relevant plots and visualizations for MuSiC.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>proximity</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Perform a proximity analysis on a list of mutations.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>smg</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Identify significantly mutated genes.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>survival</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Create survival plots and P-values for clinical and
    mutational phenotypes.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2007-2011 Washington University in St. Louis.
<div class="Pp"></div>
It is released under the Lesser GNU Public License (LGPL) version 3. See the
  associated LICENSE file in this distribution.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
This software is developed by the analysis and engineering teams at The Genome
  Institute at Washington University School of Medicine in St. Louis, with
  funding from the National Human Genome Research Institute. Richard K. Wilson,
  P.I.
<div class="Pp"></div>
The primary authors of the MuSiC suite are:
<div class="Pp"></div>
<pre>
 Nathan D. Dees, Ph.D.
 Cyriac Kandoth, Ph.D.
 Dan Koboldt, M.S.
 William Schierding, M.S.
 Michael Wendl, Ph.D.
 Qunyuan Zhang, Ph.D.
 Thomas B. Mooney, M.S.
</pre>
<h1 class="Sh" title="Sh" id="CREDITS"><a class="selflink" href="#CREDITS">CREDITS</a></h1>
The MuSiC suite uses tabix, by Heng Li. See
  http://samtools.sourceforge.net/tabix.shtml.
<div class="Pp"></div>
MuSiC depends on copies of data from the following databases, packaged in a form
  useable for quick analysis:
<div class="Pp"></div>
<pre>
 * KEGG - http://www.genome.jp/kegg/
 * COSMIC - http://www.sanger.ac.uk/genetics/CGP/cosmic/
 * OMIM - http://www.ncbi.nlm.nih.gov/omim
 * Pfam - http://pfam.sanger.ac.uk/
 * SMART - http://smart.embl-heidelberg.de/
 * SUPERFAMILY - http://supfam.cs.bris.ac.uk/SUPERFAMILY/
 * PatternScan - http://www.expasy.ch/prosite/
</pre>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
For defects with any software in the genome namespace, contact
<br/>
 genome-dev ~at~ genome.wustl.edu.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
