<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:12:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GFF2APLOT(1) General Commands Manual GFF2APLOT(1)</p>

<p style="margin-top: 1em">NAME <br>
gff2aplot - Converting GFF files for pairwise alignments to
PostScript.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gff2aplot [options] &lt;GFF_files|STDIN&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This manual page documents briefly the gff2aplot
command.</p>

<p style="margin-top: 1em">gff2aplot is a program that
draws color-filled alignment plots from GFF files for that
alignment and two sequences annotations.</p>

<p style="margin-top: 1em">OPTIONS <br>
A double dash on itself &quot;--&quot; signals end of the
options and start of file names (if present). You can use a
single dash &quot;-&quot; as STDIN placeholder. Available
options and a short <br>
description are listed here:</p>

<p style="margin-top: 1em">-h, --help <br>
Show summary of options.</p>

<p style="margin-top: 1em">--version <br>
Shows current version and exits.</p>

<p style="margin-top: 1em">-v, --verbose <br>
Verbose mode, a full report is sent to standard error
(default is set to showing only WARNINGS).</p>

<p style="margin-top: 1em">-V, --logs-filename
&lt;filename&gt; <br>
Report is written to a log file.</p>

<p style="margin-top: 1em">-q, --quiet <br>
Quiet mode, do not show any message/warning to standard
error (reporting only ERRORS).</p>

<p style="margin-top: 1em">-p, --page-size
&lt;format_name&gt; <br>
Setting a page size among pre-defined ones see below for a
list of available page formats).</p>

<p style="margin-top: 1em">--margin-left &lt;length&gt;</p>

<p style="margin-top: 1em">--margin-right
&lt;length&gt;</p>

<p style="margin-top: 1em">--margin-top &lt;length&gt;</p>

<p style="margin-top: 1em">--margin-bottom &lt;length&gt;
<br>
Setting each page margin to &lt;length&gt;. If no units are
provided, points are assumed, you can use points,
milimeters, centimeters or inches (pt, mm, cm or in,
respectively).</p>

<p style="margin-top: 1em">-B, --background-color
&lt;color&gt; <br>
Background color.</p>

<p style="margin-top: 1em">-F, --foreground-color
&lt;color&gt; <br>
Foreground color.</p>

<p style="margin-top: 1em">-T, --title &lt;string&gt; <br>
Definning plot title.</p>

<p style="margin-top: 1em">-t, --subtitle &lt;string&gt;
<br>
Definning plot subtitle.</p>

<p style="margin-top: 1em">-X, --x-label &lt;string&gt;
<br>
Defining X-axis label.</p>

<p style="margin-top: 1em">-Y, --y-label &lt;Y-Label&gt;
<br>
Defining Y-axis label.</p>

<p style="margin-top: 1em">-L, --percent-box-label
&lt;string&gt; <br>
Defining percent-box label.</p>

<p style="margin-top: 1em">-l, --extra-box-label
&lt;XBox-Label&gt; <br>
Definning Extra-Box Label.</p>

<p style="margin-top: 1em">-x, --x-sequence-coords
&lt;pos..pos&gt;</p>

<p style="margin-top: 1em">-S, --start-x-sequence
&lt;pos&gt; <br>
Sets X-sequence first nucleotide.</p>

<p style="margin-top: 1em">-E, --end-x-sequence &lt;pos&gt;
<br>
Sets X-sequence last nucleotide.</p>

<p style="margin-top: 1em">-y, ---y-sequence-coords
&lt;pos..pos&gt;</p>

<p style="margin-top: 1em">-s, --start-y-sequence
&lt;pos&gt; <br>
Sets Y-sequence first nucleotide.</p>

<p style="margin-top: 1em">-e, --end-y-sequence &lt;pos&gt;
<br>
Sets Y-sequence last nucleotide.</p>

<p style="margin-top: 1em">--x-sequence-zoom
&lt;pos..pos&gt;</p>

<p style="margin-top: 1em">--y-sequence-zoom
&lt;pos..pos&gt;</p>

<p style="margin-top: 1em">-Z, --zoom [ [-S &lt;pos&gt;]
[-E &lt;pos&gt;] [-s &lt;pos&gt;] [-e &lt;pos&gt;] ] <br>
This option zooms an area you have selected with -S,-E,-s,-e
(all 4 are optional).</p>

<p style="margin-top: 1em">-z, --zoom-area [ [-S
&lt;pos&gt;] [-E &lt;pos&gt;] [-s &lt;pos&gt;] [-e
&lt;pos&gt;] ] <br>
This option marks a zoom area on your plot, but does not
make a zoom.</p>

<p style="margin-top: 1em">-A, --alignment-name
&lt;SeqXName:SeqYName&gt; <br>
Defining which alignment is going to be plotted if you have
more than one alignment in your gff input.</p>

<p style="margin-top: 1em">-N, --x-sequence-name
&lt;SeqXName&gt; <br>
Defining which sequence is going to be plotted at
X-axis.</p>

<p style="margin-top: 1em">-n, --y-sequence-name
&lt;SeqYName&gt; <br>
Defining which sequence is going to be plotted at
Y-axis.</p>

<p style="margin-top: 1em">-r, --aplot-xy-noteq <br>
By default X and Y axes have same length, this option
disables such behaviour, so X and Y sequence will have
axes-lengths proportional to their nucleotide lengths.</p>

<p style="margin-top: 1em">-R, --xy-axes-scale &lt;X/Y
ratio&gt; <br>
This option allows one to set a different scale between X
and Y axes lengths (by default is &rsquo;1&rsquo;). Below 1
values make Y larger than X, and larger than 1 result in
getting <br>
X larger than Y. # Must be explained better.</p>

<p style="margin-top: 1em">-W, --aln-scale-width <br>
Scaling score on width for Aplot lines.</p>

<p style="margin-top: 1em">-w, --aln-scale-color <br>
Scaling score on color for Aplot lines.</p>

<p style="margin-top: 1em">-K, --show-ribbons
&lt;ribbon_type&gt; <br>
Force Ribbons for all features on axes to be: (N)one,
(L)ines, (R)ibbons, (B)oth.</p>

<p style="margin-top: 1em">-G, --show-grid <br>
Switches &rsquo;on&rsquo; grid (default is
&rsquo;off&rsquo;).</p>

<p style="margin-top: 1em">-g, --hide-grid <br>
Switches &rsquo;off&rsquo; grid (if switched on from
customization files).</p>

<p style="margin-top: 1em">-I, --show-percent-box <br>
Switches &rsquo;on&rsquo; Percent box (default is
&rsquo;off&rsquo;).</p>

<p style="margin-top: 1em">-i, --hide-percent-box <br>
Switches &rsquo;off&rsquo; Percent box (if set to
&rsquo;on&rsquo; on custom files).</p>

<p style="margin-top: 1em">-O, --show-extra-box <br>
Switches &rsquo;on&rsquo; Extra box (default is
&rsquo;off&rsquo;).</p>

<p style="margin-top: 1em">-o, --hide-extra-box <br>
Switches &rsquo;off&rsquo; Extra box (if set to
&rsquo;on&rsquo; on custom files).</p>

<p style="margin-top: 1em">-D, --aplot-box-color
&lt;color&gt; <br>
Aplot main box background color.</p>

<p style="margin-top: 1em">-d, --percent-box-color
&lt;color&gt; <br>
Percent box background color.</p>

<p style="margin-top: 1em">-b, --extra-box-color
&lt;color&gt; <br>
Extra box background color.</p>

<p style="margin-top: 1em">--nopswarnings <br>
Switch off warnings that may appear on the final PostScript
figure when X sequence, Y sequence and/or alignment input
data is missing.</p>

<p style="margin-top: 1em">-a, --hide-credits <br>
Switch off gff2aplot credits line on plot.</p>

<p style="margin-top: 1em">--debug <br>
Reporting variable contents when testing the program.
Requires that log report file option was also activated.</p>

<p style="margin-top: 1em">--layout-var
&rsquo;&lt;variable=value&gt;&rsquo;</p>

<p style="margin-top: 1em">--sequence-var
&rsquo;&lt;sequence::variable=value&gt;&rsquo;</p>

<p style="margin-top: 1em">--source-var
&rsquo;&lt;source::variable=value&gt;&rsquo;</p>

<p style="margin-top: 1em">--strand-var
&rsquo;&lt;strand::variable=value&gt;&rsquo;</p>

<p style="margin-top: 1em">--group-var
&rsquo;&lt;group::variable=value&gt;&rsquo;</p>

<p style="margin-top: 1em">--feature-var
&rsquo;&lt;feature::variable=value&gt;&rsquo; <br>
Loading a feature/group/strand/source/sequence/layout
customization variable from command-line. You can set
several variables by repeating any of these options, i.e.:
<br>
... --feature-var &rsquo;cds::feature_shape=box&rsquo; <br>
--feature-var &rsquo;cds::feature_color=blue&rsquo; ...</p>

<p style="margin-top: 1em">-C, --custom-filename
&lt;filename&gt; <br>
Loading customization parameters from a given file (if
default &quot;.gff2aplotrc&quot; exists is loaded before
it).</p>

<p style="margin-top: 1em">Those are the colors defined in
gff2aplot: <br>
+ Default Colors: black white. <br>
+ Color Palette: <br>
grey pink magenta violet blue skyblue cyan seagreen <br>
green limegreen yellow orange red brown <br>
You can get up to nine color shades from Variable Colors
with &quot;verydeep&quot;, &quot;deep&quot;,
&quot;verydark&quot;, &quot;dark&quot;, &quot;light&quot;,
&quot;verylight&quot;, &quot;pale&quot; and
&quot;verypale&quot; prefixes, as example: <br>
verydeepblue, deepblue, verydarkblue, darkblue, blue,
lightblue, verylightblue, paleblue and verypaleblue. <br>
+ Dynamic colors: gradient rainbow.</p>

<p style="margin-top: 1em">The following page sizes are
available: from A0 to A10, from B0 to B10, 10x14, executive,
folio, ledger, legal, letter, quarto, statement and
tabloid.</p>

<p style="margin-top: 1em">ENVIRONMENT VARIABLES <br>
There are two environmental variables that can be set by
users to their preferences: <br>
+ You can specify the path where gff2aplot can find the
default files with the shell variable
&quot;GFF2APLOT_CUSTOMPATH&quot;. Default value is the path
where you are running gff2aplot. <br>
+ You can also define the default custom filename you will
like with the variable &quot;GFF2APLOT_CUSTOMFILE&quot;,
program default filename for custom file is
&quot;.gff2aplotrc&quot;. <br>
+ Now gff2aplot does not need to write any temporary file,
so that previous versions default temporary directory path
variable (&quot;GFF2APLOT_TMP&quot;) is no longer used. <br>
+ Setting those vars in Bourne-shell and C-shell: <br>
o Using a Bourne-Shell (e.g. bash): <br>
export GFF2APLOT_CUSTOMPATH=&quot;path&quot; export
GFF2APLOT_CUSTOMFILE=&quot;file_name&quot; <br>
o Using a C-Shell: <br>
setenv GFF2APLOT_CUSTOMPATH &quot;path&quot; setenv
GFF2APLOT_CUSTOMFILE &quot;file_name&quot;</p>

<p style="margin-top: 1em">SEE ALSO <br>
ali2gff(1), blat2gff(1), parseblast(1), sim2gff(1).</p>

<p style="margin-top: 1em">AUTHOR <br>
gff2aplot was written by Josep Francesc ABRIL FERRANDO
&lt;jabril@imim.es&gt;, Thomas WIEHE
&lt;twiehe@zedat.fu-berlin.de&gt; and Roderic GUIGO SERRA
&lt;rguigo@imim.es&gt;.</p>

<p style="margin-top: 1em">This manual page was written by
Nelson A. de Oliveira &lt;naoliv@gmail.com&gt;, for the
Debian project (but may be used by others).</p>

<p style="margin-top: 1em">Mon, 21 Mar 2005 21:44:15 -0300
GFF2APLOT(1)</p>
<hr>
</body>
</html>
