<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GFFREAD(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GFFREAD(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">GFFREAD(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gffread - component of cufflinks suite
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Usage: gffread &lt;input_gff&gt; [-g &lt;genomic_seqs_fasta&gt; |
  &lt;dir&gt;][-s &lt;seq_info.fsize&gt;]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">[-o &lt;outfile.gff&gt;] [-t &lt;tname&gt;] [-r
      [[&lt;strand&gt;]&lt;chr&gt;:]&lt;start&gt;..&lt;end&gt; [-R]]
      [-CTVNJMKQAFGUBHZWTOLE] [-w &lt;exons.fa&gt;] [-x &lt;cds.fa&gt;] [-y
      &lt;tr_cds.fa&gt;] [-i &lt;maxintron&gt;]</dd>
</dl>
<div class="Pp"></div>
Filters and/or converts GFF3/GTF2 records. &lt;input_gff&gt; is a GFF file, use
  '-' if the GFF records will be given at stdin
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">full path to a multi-fasta file with the genomic sequences
      for all input mappings, OR a directory with single-fasta files (one per
      genomic sequence, with file names matching sequence names)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">&lt;seq_info.fsize&gt; is a tab-delimited file providing
      this info for each of the mapped sequences: &lt;seq-name&gt;
      &lt;seq-length&gt; &lt;seq-description&gt; (useful for <b>-A</b> option
      with mRNA/EST/protein mappings)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b></dt>
  <dd class="It-tag">discard transcripts having an intron larger than
      &lt;maxintron&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b></dt>
  <dd class="It-tag">only show transcripts overlapping coordinate range
      &lt;start&gt;..&lt;end&gt; (on chromosome/contig &lt;chr&gt;, strand
      &lt;strand&gt; if provided)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b></dt>
  <dd class="It-tag">for <b>-r</b> option, discard all transcripts that are not
      fully contained within the given range</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b></dt>
  <dd class="It-tag">discard single-exon transcripts</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b></dt>
  <dd class="It-tag">coding only: discard mRNAs that have no CDS feature</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b></dt>
  <dd class="It-tag">full GFF attribute preservation (all attributes are
    shown)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-G</b></dt>
  <dd class="It-tag">only parse additional exon attributes from the first exon
      and move them to the mRNA level (useful for GTF input)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b></dt>
  <dd class="It-tag">use the description field from &lt;seq_info.fsize&gt; and
      add it as the value for a 'descr' attribute to the GFF record</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-O</b></dt>
  <dd class="It-tag">process also non-transcript GFF records (by default
      non-transcript records are ignored)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b></dt>
  <dd class="It-tag">discard any mRNAs with CDS having in-frame stop codons</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-H</b></dt>
  <dd class="It-tag">for <b>-V</b> option, check and adjust the starting CDS
      phase if the original phase leads to a translation with an in-frame stop
      codon</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b></dt>
  <dd class="It-tag">for <b>-V</b> option, single-exon transcripts are also
      checked on the opposite strand</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b></dt>
  <dd class="It-tag">discard multi-exon mRNAs that have any intron with a
      non-canonical splice site consensus (i.e. not GT-AG, GC-AG or AT-AC)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-J</b></dt>
  <dd class="It-tag">discard any mRNAs that either lack initial START codon or
      the terminal STOP codon, or have an in-frame stop codon (only print mRNAs
      with a fulll, valid CDS)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--no-pseudo</b>:
  filter out records matching the 'pseudo' keyword</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-M</b>/--merge
  : cluster the input transcripts into loci, collapsing matching</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">transcripts (those with the same exact introns and fully
      contained)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-d</b>
  &lt;dupinfo&gt; : for <b>-M</b> option, write collapsing info to file
  &lt;dupinfo&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--cluster-only</b>:
  same as <b>--merge</b> but without collapsing matching transcripts</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-K</b></dt>
  <dd class="It-tag">for <b>-M</b> option: also collapse shorter, fully
      contained transcripts with fewer introns than the container</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b></dt>
  <dd class="It-tag">for <b>-M</b> option, remove the containment restriction:
      (multi-exon transcripts will be collapsed if just their introns match,
      while single-exon transcripts can partially overlap (80%))</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--force-exons</b>:
  make sure that the lowest level GFF features are printed as</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&quot;exon&quot; features</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b></dt>
  <dd class="It-tag">expose (warn about) duplicate transcript IDs and other
      potential problems with the given GFF/GTF records</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b></dt>
  <dd class="It-tag">decode url encoded characters within attributes</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b></dt>
  <dd class="It-tag">merge close exons into a single exon (for intron
    size&lt;4)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b></dt>
  <dd class="It-tag">write a fasta file with spliced exons for each GFF
      transcript</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-x</b></dt>
  <dd class="It-tag">write a fasta file with spliced CDS for each GFF
    transcript</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-W</b></dt>
  <dd class="It-tag">for <b>-w</b> and <b>-x</b> options, also write for each
      fasta record the exon coordinates projected onto the spliced sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-y</b></dt>
  <dd class="It-tag">write a protein fasta file with the translation of CDS for
      each record</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b></dt>
  <dd class="It-tag">Ensembl GTF to GFF3 conversion (implies <b>-F</b>; should
      be used with <b>-m</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b></dt>
  <dd class="It-tag">&lt;chr_replace&gt; is a reference (genomic) sequence
      replacement table with this format: &lt;original_ref_ID&gt;
      &lt;new_ref_ID&gt; GFF records on reference sequences that are not found
      among the &lt;original_ref_ID&gt; entries in this file will be filtered
      out</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b></dt>
  <dd class="It-tag">the &quot;filtered&quot; GFF records will be written to
      &lt;outfile.gff&gt; (use <b>-o-</b> for printing to stdout)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b></dt>
  <dd class="It-tag">use &lt;trackname&gt; in the second column of each GFF
      output line</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-T</b>
  <b>-o</b> option will output GTF format instead of GFF3</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2016</td>
    <td class="foot-os">gffread 2.2.1</td>
  </tr>
</table>
</body>
</html>
