<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:12:49 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GFFREAD(1) User Commands GFFREAD(1)</p>

<p style="margin-top: 1em">NAME <br>
gffread - component of cufflinks suite</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Usage: gffread &lt;input_gff&gt; [-g
&lt;genomic_seqs_fasta&gt; | &lt;dir&gt;][-s
&lt;seq_info.fsize&gt;]</p>

<p style="margin-top: 1em">[-o &lt;outfile.gff&gt;] [-t
&lt;tname&gt;] [-r
[[&lt;strand&gt;]&lt;chr&gt;:]&lt;start&gt;..&lt;end&gt;
[-R]] [-CTVNJMKQAFGUBHZWTOLE] [-w &lt;exons.fa&gt;] [-x
&lt;cds.fa&gt;] [-y &lt;tr_cds.fa&gt;] [-i
&lt;maxintron&gt;]</p>

<p style="margin-top: 1em">Filters and/or converts
GFF3/GTF2 records. &lt;input_gff&gt; is a GFF file, use
&rsquo;-&rsquo; if the GFF records will be given at
stdin</p>

<p style="margin-top: 1em">OPTIONS <br>
-g full path to a multi-fasta file with the genomic
sequences for all input mappings, OR a directory with
single-fasta files (one per genomic sequence, with file
names match&acirc; <br>
ing sequence names)</p>

<p style="margin-top: 1em">-s &lt;seq_info.fsize&gt; is a
tab-delimited file providing this info for each of the
mapped sequences: &lt;seq-name&gt; &lt;seq-length&gt;
&lt;seq-description&gt; (useful for -A option with <br>
mRNA/EST/protein mappings)</p>

<p style="margin-top: 1em">-i discard transcripts having an
intron larger than &lt;maxintron&gt;</p>

<p style="margin-top: 1em">-r only show transcripts
overlapping coordinate range &lt;start&gt;..&lt;end&gt; (on
chromosome/contig &lt;chr&gt;, strand &lt;strand&gt; if
provided)</p>

<p style="margin-top: 1em">-R for -r option, discard all
transcripts that are not fully contained within the given
range</p>

<p style="margin-top: 1em">-U discard single-exon
transcripts</p>

<p style="margin-top: 1em">-C coding only: discard mRNAs
that have no CDS feature</p>

<p style="margin-top: 1em">-F full GFF attribute
preservation (all attributes are shown)</p>

<p style="margin-top: 1em">-G only parse additional exon
attributes from the first exon and move them to the mRNA
level (useful for GTF input)</p>

<p style="margin-top: 1em">-A use the description field
from &lt;seq_info.fsize&gt; and add it as the value for a
&rsquo;descr&rsquo; attribute to the GFF record</p>

<p style="margin-top: 1em">-O process also non-transcript
GFF records (by default non-transcript records are
ignored)</p>

<p style="margin-top: 1em">-V discard any mRNAs with CDS
having in-frame stop codons</p>

<p style="margin-top: 1em">-H for -V option, check and
adjust the starting CDS phase if the original phase leads to
a translation with an in-frame stop codon</p>

<p style="margin-top: 1em">-B for -V option, single-exon
transcripts are also checked on the opposite strand</p>

<p style="margin-top: 1em">-N discard multi-exon mRNAs that
have any intron with a non-canonical splice site consensus
(i.e. not GT-AG, GC-AG or AT-AC)</p>

<p style="margin-top: 1em">-J discard any mRNAs that either
lack initial START codon or the terminal STOP codon, or have
an in-frame stop codon (only print mRNAs with a fulll, valid
CDS)</p>

<p style="margin-top: 1em">--no-pseudo: filter out records
matching the &rsquo;pseudo&rsquo; keyword</p>

<p style="margin-top: 1em">-M/--merge : cluster the input
transcripts into loci, collapsing matching</p>

<p style="margin-top: 1em">transcripts (those with the same
exact introns and fully contained)</p>

<p style="margin-top: 1em">-d &lt;dupinfo&gt; : for -M
option, write collapsing info to file &lt;dupinfo&gt;</p>

<p style="margin-top: 1em">--cluster-only: same as --merge
but without collapsing matching transcripts</p>

<p style="margin-top: 1em">-K for -M option: also collapse
shorter, fully contained transcripts with fewer introns than
the container</p>

<p style="margin-top: 1em">-Q for -M option, remove the
containment restriction: (multi-exon transcripts will be
collapsed if just their introns match, while single-exon
transcripts can partially over&acirc; <br>
lap (80%))</p>

<p style="margin-top: 1em">--force-exons: make sure that
the lowest level GFF features are printed as</p>

<p style="margin-top: 1em">&quot;exon&quot; features</p>

<p style="margin-top: 1em">-E expose (warn about) duplicate
transcript IDs and other potential problems with the given
GFF/GTF records</p>

<p style="margin-top: 1em">-D decode url encoded characters
within attributes</p>

<p style="margin-top: 1em">-Z merge close exons into a
single exon (for intron size&lt;4)</p>

<p style="margin-top: 1em">-w write a fasta file with
spliced exons for each GFF transcript</p>

<p style="margin-top: 1em">-x write a fasta file with
spliced CDS for each GFF transcript</p>

<p style="margin-top: 1em">-W for -w and -x options, also
write for each fasta record the exon coordinates projected
onto the spliced sequence</p>

<p style="margin-top: 1em">-y write a protein fasta file
with the translation of CDS for each record</p>

<p style="margin-top: 1em">-L Ensembl GTF to GFF3
conversion (implies -F; should be used with -m)</p>

<p style="margin-top: 1em">-m &lt;chr_replace&gt; is a
reference (genomic) sequence replacement table with this
format: &lt;original_ref_ID&gt; &lt;new_ref_ID&gt; GFF
records on reference sequences that are not found <br>
among the &lt;original_ref_ID&gt; entries in this file will
be filtered out</p>

<p style="margin-top: 1em">-o the &quot;filtered&quot; GFF
records will be written to &lt;outfile.gff&gt; (use -o- for
printing to stdout)</p>

<p style="margin-top: 1em">-t use &lt;trackname&gt; in the
second column of each GFF output line</p>

<p style="margin-top: 1em">-T -o option will output GTF
format instead of GFF3</p>

<p style="margin-top: 1em">gffread 2.2.1 November 2016
GFFREAD(1)</p>
<hr>
</body>
</html>
