<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:13:02 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GIIRA(1) User Commands GIIRA(1)</p>

<p style="margin-top: 1em">NAME <br>
giira - Gene Identification Incorporating RNA-Seq data and
Ambiguous reads</p>

<p style="margin-top: 1em">SYNOPSIS <br>
giira -iG genomeFile.fasta -iR rnaFile.fastq -libPath</p>

<p style="margin-top: 1em">DESCRIPTION <br>
GIIRA (Gene Identification Incorporating RNA-Seq data and
Ambiguous reads) is a method to identify potential gene
regions in a genome based on a RNA-Seq mapping and
incorporating <br>
ambiguously mapped reads.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h : help text and exit</p>

<p style="margin-top: 1em">-iG [pathToGenomes] : specify
path to directory with genome files in fasta format</p>

<p style="margin-top: 1em">-iR [pathToRna] : specify path
to directory with rna read files in fastq format</p>

<p style="margin-top: 1em">-scripts [absolutePath] :
specify the absolute path to the directory containing the
required helper scripts, DEFAULT: directory of GIIRA.jar</p>

<p style="margin-top: 1em">-out [pathToResults] : specify
the directory that shall contain the results files</p>

<p style="margin-top: 1em">-outName [outputName] : specify
desired name for output files, DEFAULT: genes</p>

<p style="margin-top: 1em">-haveSam [samfileName]: if a sam
file already exists, provide the name, else a mapping is
performed. NOTE: the sam file has to be sorted according to
read names!</p>

<p style="margin-top: 1em">-nT [numberThreads] : specify
the maximal number of threads that are allowed to be used,
DEFAULT: 1</p>

<p style="margin-top: 1em">-mT [tophat/bwa/bwasw] : specify
desired tool for the read mapping, DEFAULT: tophat</p>

<p style="margin-top: 1em">-mem [int] : specify the amount
of memory that cplex is allowed to use</p>

<p style="margin-top: 1em">-maxReportedHits [int] : if
using BWA as mapping tool, specify the maximal number of
reported hits, DEFAULT: 2</p>

<p style="margin-top: 1em">-prokaryote : if specified,
genome is treated as prokaryotic, no spliced reads are
accepted, and structural genes are resolved. DEFAULT: n</p>

<p style="margin-top: 1em">-minCov [double] : specify the
minimum required coverage of the gene candidate extraction,
DEFAULT: -1 (is estimated from mapping)</p>

<p style="margin-top: 1em">-maxCov [double] : optional
maximal coverage threshold, can also be estimated from
mapping (DEFAULT)</p>

<p style="margin-top: 1em">-endCov [double] : if the
coverage falls below this value, the currently open
candidate gene is closed. This value can be estimated from
the minimum coverage (-1); DEFAULT: -1</p>

<p style="margin-top: 1em">-dispCov [0/1] : estimate (1)
the coverage histogram for the read mapping, DEFAULT: 0</p>

<p style="margin-top: 1em">-interval [int] : specify the
minimal size of an interval between near candidate genes, if
&quot;-1&quot; it equals the read length. DEFAULT: -1</p>

<p style="margin-top: 1em">-splLim [double] : specify the
minimal coverage that is required to accept a splice site,
if (-1) the threshold is equal to minCov, DEFAULT: -1</p>

<p style="margin-top: 1em">-rL [int] : specify read length,
otherwise this information is extracted from SAM file
(DEFAULT)</p>

<p style="margin-top: 1em">-samForSequential
[pathToSamFile] : if it is desired to analyse chromosomes in
a sequential manner, provide a chromosome sorted sam file in
addition to the one sorted by read <br>
names, DEFAULT: noSequential</p>

<p style="margin-top: 1em">-noAmbiOpti : if specified,
ambiguous hits are not included in the analysis</p>

<p style="margin-top: 1em">-settingMapper [(list of
parameters)] : A comma-separated list of the desired
parameters for TopHat or BWA. Please provide</p>

<p style="margin-top: 1em">for each parameter a pair of
indicator and value, separated by an equality sign. Note
that parameters intended for the 3 different parts
(indexing, aln, sam) of BWA have <br>
to be separated by a lowercase bar</p>

<p style="margin-top: 1em">Example: -settingMapper
[-a=is_-t=5,-N_-n=5]</p>

<p style="margin-top: 1em">giira 2014-02-10 February 2014
GIIRA(1)</p>
<hr>
</body>
</html>
