<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GIIRA(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GIIRA(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">GIIRA(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
giira - Gene Identification Incorporating RNA-Seq data and Ambiguous reads
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>giira</b> <b>-iG</b> genomeFile.fasta <b>-iR</b> rnaFile.fastq
  <b>-libPath</b>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
GIIRA (Gene Identification Incorporating RNA-Seq data and Ambiguous reads) is a
  method to identify potential gene regions in a genome based on a RNA-Seq
  mapping and incorporating ambiguously mapped reads.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-h</b> :
  help text and exit</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-iG</b>
  [pathToGenomes] : specify path to directory with genome files in fasta
  format</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-iR</b>
  [pathToRna] : specify path to directory with rna read files in fastq
  format</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-scripts</b>
  [absolutePath] : specify the absolute path to the directory containing the
  required helper scripts, DEFAULT: directory of GIIRA.jar</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-out</b>
  [pathToResults] : specify the directory that shall contain the results
  files</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-outName</b>
  [outputName] : specify desired name for output files, DEFAULT: genes</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-haveSam</b>
  [samfileName]: if a sam file already exists, provide the name, else a mapping
  is performed. NOTE: the sam file has to be sorted according to read
  names!</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nT</b>
  [numberThreads] : specify the maximal number of threads that are allowed to be
  used, DEFAULT: 1</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-mT</b>
  [tophat/bwa/bwasw] : specify desired tool for the read mapping, DEFAULT:
  tophat</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-mem</b>
  [int] : specify the amount of memory that cplex is allowed to use</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-maxReportedHits</b>
  [int] : if using BWA as mapping tool, specify the maximal number of reported
  hits, DEFAULT: 2</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-prokaryote</b>
  : if specified, genome is treated as prokaryotic, no spliced reads are
  accepted, and structural genes are resolved. DEFAULT: n</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-minCov</b>
  [double] : specify the minimum required coverage of the gene candidate
  extraction, DEFAULT: <b>-1</b> (is estimated from mapping)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-maxCov</b>
  [double] : optional maximal coverage threshold, can also be estimated from
  mapping (DEFAULT)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-endCov</b>
  [double] : if the coverage falls below this value, the currently open
  candidate gene is closed. This value can be estimated from the minimum
  coverage ( <b>-1</b>); DEFAULT: <b>-1</b></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-dispCov</b>
  [0/1] : estimate (1) the coverage histogram for the read mapping, DEFAULT:
  0</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-interval</b>
  [int] : specify the minimal size of an interval between near candidate genes,
  if &quot;-1&quot; it equals the read length. DEFAULT: <b>-1</b></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-splLim</b>
  [double] : specify the minimal coverage that is required to accept a splice
  site, if ( <b>-1</b>) the threshold is equal to minCov, DEFAULT:
  <b>-1</b></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-rL</b>
  [int] : specify read length, otherwise this information is extracted from SAM
  file (DEFAULT)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-samForSequential</b>
  [pathToSamFile] : if it is desired to analyse chromosomes in a sequential
  manner, provide a chromosome sorted sam file in addition to the one sorted by
  read names, DEFAULT: noSequential</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-noAmbiOpti</b>
  : if specified, ambiguous hits are not included in the analysis</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-settingMapper</b>
  [(list of parameters)] : A comma-separated list of the desired parameters for
  TopHat or BWA. Please provide</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">for each parameter a pair of indicator and value, separated
      by an equality sign. Note that parameters intended for the 3 different
      parts (indexing, aln, sam) of BWA have to be separated by a lowercase
    bar</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Example: <b>-settingMapper</b> [-a=is_-t=5,-N_-n=5]</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2014</td>
    <td class="foot-os">giira 2014-02-10</td>
  </tr>
</table>
</body>
</html>
