<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:13:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GLAM2SCAN(1) glam2 Manual GLAM2SCAN(1)</p>

<p style="margin-top: 1em">NAME <br>
glam2scan - finds a GLAM2 motif in a database</p>

<p style="margin-top: 1em">SYNOPSIS <br>
glam2scan [options] alphabet my_motif.glam2 my_seqs.fa</p>

<p style="margin-top: 1em">An alphabet other than p or n is
interpreted as the name of an alphabet file.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
glam2scan finds matches, in a sequence database, to a motif
discovered by glam2. Each match receives a score, indicating
how well it fits the motif.</p>

<p style="margin-top: 1em">OPTIONS (DEFAULT SETTINGS) <br>
-h <br>
Show all options and their default settings.</p>

<p style="margin-top: 1em">-o <br>
Output file (stdout).</p>

<p style="margin-top: 1em">-n <br>
Number of alignments to report (25).</p>

<p style="margin-top: 1em">-2 <br>
Examine both strands - forward and reverse complement.</p>

<p style="margin-top: 1em">-D <br>
Deletion pseudocount (0.1).</p>

<p style="margin-top: 1em">-E <br>
No-deletion pseudocount (2.0).</p>

<p style="margin-top: 1em">-I <br>
Insertion pseudocount (0.02).</p>

<p style="margin-top: 1em">-J <br>
No-insertion pseudocount (1.0).</p>

<p style="margin-top: 1em">-d <br>
Dirichlet mixture file.</p>

<p style="margin-top: 1em">SEE ALSO <br>
glam2format(1), glam2mask(1), glam2-purge(1), glam2(1)</p>

<p style="margin-top: 1em">The full Hypertext documentation
of GLAM2 is available online at
http://bioinformatics.org.au/glam2/ or on this computer in
/usr/share/doc/glam2/.</p>

<p style="margin-top: 1em">REFERENCE <br>
If you use GLAM2, please cite: MC Frith, NFW Saunders, B
Kobe, TL Bailey (2008) Discovering sequence motifs with
arbitrary insertions and deletions, PLoS Computational
Biology <br>
(in press).</p>

<p style="margin-top: 1em">AUTHORS <br>
Martin Frith <br>
Author of GLAM2.</p>

<p style="margin-top: 1em">Timothy Bailey <br>
Author of GLAM2.</p>

<p style="margin-top: 1em">Charles Plessy
&lt;plessy@debian.org&gt; <br>
Formatted this manpage in DocBook XML for the Debian
distribution.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
The source code and the documentation of GLAM2 are released
in the public domain.</p>

<p style="margin-top: 1em">GLAM2 1056 05/19/2008
GLAM2SCAN(1)</p>
<hr>
</body>
</html>
