<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GLAM2MASK(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GLAM2MASK(1)</td>
    <td class="head-vol">glam2 Manual</td>
    <td class="head-rtitle">GLAM2MASK(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
glam2mask - masks a GLAM2 motif in sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 10.00ex; text-indent: -10.00ex;"><b>glam2mask</b>
  [ <b>options</b>] <i>my_motif.glam2</i> <i>my_seqs.fa</i></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>glam2mask</b> masks a <b>glam2</b> motif out of sequences, so that weaker
  motifs can be found. Masking replaces residues aligned to key positions with
  the symbol &#x00B4;x&#x00B4;. By alternately applying <b>glam2</b> and
  <b>glam2mask</b> several times, it is possible to find the strongest,
  second-strongest, third-strongest, etc. motifs in a set of sequences.
<h1 class="Sh" title="Sh" id="OPTIONS_(DEFAULT_SETTINGS)"><a class="selflink" href="#OPTIONS_(DEFAULT_SETTINGS)">OPTIONS
  (DEFAULT SETTINGS)</a></h1>
<b>-o</b>
<div style="margin-left: 4.00ex;">Output file (<i>stdout</i>).</div>
<div class="Pp"></div>
<b>-x</b>
<div style="margin-left: 4.00ex;">Mask character (<i>x</i>).</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>glam2format</b>(1), <b>glam2</b>(1), <b>glam2-purge</b>(1),
  <b>glam2scan</b>(1)
<div class="Pp"></div>
The full Hypertext documentation of GLAM2 is available online at
  <i>http://bioinformatics.org.au/glam2/</i> or on this computer in
  <i>/usr/share/doc/glam2/</i>.
<h1 class="Sh" title="Sh" id="REFERENCE"><a class="selflink" href="#REFERENCE">REFERENCE</a></h1>
If you use GLAM2, please cite: MC Frith, NFW Saunders, B Kobe, TL Bailey (2008)
  Discovering sequence motifs with arbitrary insertions and deletions, PLoS
  Computational Biology (in press).
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<b>Martin Frith</b>
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Author of GLAM2.</dd>
</dl>
<div class="Pp"></div>
<b>Timothy Bailey</b>
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Author of GLAM2.</dd>
</dl>
<div class="Pp"></div>
<b>Charles Plessy</b> &lt;plessy@debian.org&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Formatted this manpage in DocBook XML for the Debian
      distribution.</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
The source code and the documentation of GLAM2 are released in the public
  domain.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/19/2008</td>
    <td class="foot-os">GLAM2 1056</td>
  </tr>
</table>
</body>
</html>
