<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GLAM2(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GLAM2(1)</td>
    <td class="head-vol">glam2 Manual</td>
    <td class="head-rtitle">GLAM2(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
glam2 - Gapped Local Alignment of Motifs
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>glam2</b>
  [ <b>options</b>] <b>alphabet</b> <i>my_seqs.fa</i></div>
<div class="Pp"></div>
An alphabet other than <i>p</i> or <i>n</i> is interpreted as the name of an
  alphabet file.
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
GLAM2 is a software package for finding motifs in sequences, typically
  amino-acid or nucleotide sequences. A motif is a re-occurring sequence
  pattern: typical examples are the TATA box and the CAAX prenylation motif. The
  main innovation of GLAM2 is that it allows insertions and deletions in motifs.
<h1 class="Sh" title="Sh" id="OPTIONS_(DEFAULT_SETTINGS)"><a class="selflink" href="#OPTIONS_(DEFAULT_SETTINGS)">OPTIONS
  (DEFAULT SETTINGS)</a></h1>
<b>-h</b>
<div style="margin-left: 4.00ex;">Show all options and their default
  settings.</div>
<div class="Pp"></div>
<b>-o</b>
<div style="margin-left: 4.00ex;">Output file (<i>stdout</i>).</div>
<div class="Pp"></div>
<b>-r</b>
<div style="margin-left: 4.00ex;">Number of alignment runs (<i>10</i>).</div>
<div class="Pp"></div>
<b>-n</b>
<div style="margin-left: 4.00ex;">End each run after this many iterations
  without improvement ( <i>10000</i>).</div>
<div class="Pp"></div>
<b>-2</b>
<div style="margin-left: 4.00ex;">Examine both strands - forward and reverse
  complement.</div>
<div class="Pp"></div>
<b>-z</b>
<div style="margin-left: 4.00ex;">Minimum number of sequences in the alignment
  (<i>2</i>).</div>
<div class="Pp"></div>
<b>-a</b>
<div style="margin-left: 4.00ex;">Minimum number of aligned columns
  (<i>2</i>).</div>
<div class="Pp"></div>
<b>-b</b>
<div style="margin-left: 4.00ex;">Maximum number of aligned columns
  (<i>50</i>).</div>
<div class="Pp"></div>
<b>-w</b>
<div style="margin-left: 4.00ex;">Initial number of aligned columns
  (<i>20</i>).</div>
<div class="Pp"></div>
<b>-d</b>
<div style="margin-left: 4.00ex;">Dirichlet mixture file.</div>
<div class="Pp"></div>
<b>-D</b>
<div style="margin-left: 4.00ex;">Deletion pseudocount (<i>0.1</i>).</div>
<div class="Pp"></div>
<b>-E</b>
<div style="margin-left: 4.00ex;">No-deletion pseudocount (<i>2.0</i>).</div>
<div class="Pp"></div>
<b>-I</b>
<div style="margin-left: 4.00ex;">Insertion pseudocount (<i>0.02</i>).</div>
<div class="Pp"></div>
<b>-J</b>
<div style="margin-left: 4.00ex;">No-insertion pseudocount (<i>1.0</i>).</div>
<div class="Pp"></div>
<b>-q</b>
<div style="margin-left: 4.00ex;">Weight for generic versus
  sequence-set-specific residue abundances ( <i>1e+99</i>).</div>
<div class="Pp"></div>
<b>-t</b>
<div style="margin-left: 4.00ex;">Initial temperature (<i>1.2</i>).</div>
<div class="Pp"></div>
<b>-c</b>
<div style="margin-left: 4.00ex;">Cooling factor per n iterations
  (<i>1.44</i>).</div>
<div class="Pp"></div>
<b>-u</b>
<div style="margin-left: 4.00ex;">Temperature lower bound (<i>0.1</i>).</div>
<div class="Pp"></div>
<b>-p</b>
<div style="margin-left: 4.00ex;">Print progress information at each
  iteration.</div>
<div class="Pp"></div>
<b>-m</b>
<div style="margin-left: 4.00ex;">Column-sampling moves per site-sampling move
  (<i>1.0</i>).</div>
<div class="Pp"></div>
<b>-x</b>
<div style="margin-left: 4.00ex;">Site sampling algorithm: <i>0</i>=FAST
  <i>1</i>=SLOW <i>2</i>=FFT (<i>0</i>).</div>
<div class="Pp"></div>
<b>-s</b>
<div style="margin-left: 4.00ex;">Seed for pseudo-random numbers
  (<i>1</i>).</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>glam2format</b>(1), <b>glam2mask</b>(1), <b>glam2-purge</b>(1),
  <b>glam2scan</b>(1)
<div class="Pp"></div>
The full Hypertext documentation of GLAM2 is available online at
  <i>http://bioinformatics.org.au/glam2/</i> or on this computer in
  <i>/usr/share/doc/glam2/</i>.
<h1 class="Sh" title="Sh" id="REFERENCE"><a class="selflink" href="#REFERENCE">REFERENCE</a></h1>
If you use GLAM2, please cite: MC Frith, NFW Saunders, B Kobe, TL Bailey (2008)
  Discovering sequence motifs with arbitrary insertions and deletions, PLoS
  Computational Biology (in press).
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<b>Martin Frith</b>
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Author of GLAM2.</dd>
</dl>
<div class="Pp"></div>
<b>Timothy Bailey</b>
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Author of GLAM2.</dd>
</dl>
<div class="Pp"></div>
<b>Charles Plessy</b> &lt;plessy@debian.org&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Formatted this manpage in DocBook XML for the Debian
      distribution.</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
The source code and the documentation of GLAM2 are released in the public
  domain.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/19/2008</td>
    <td class="foot-os">GLAM2 1056</td>
  </tr>
</table>
</body>
</html>
