<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:13:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GLAM2FORMAT(1) glam2 Manual GLAM2FORMAT(1)</p>

<p style="margin-top: 1em">NAME <br>
glam2format - converts GLAM2 motifs to FASTA or MSF
format</p>

<p style="margin-top: 1em">SYNOPSIS <br>
glam2format [options] my_format my_motif.glam2</p>

<p style="margin-top: 1em">Formats: fasta, msf.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
glam2format reads in a motif found by glam2, and writes it
in a standard alignment format (FASTA-with-gaps or MSF).
This enables the alignment to be passed to third-party <br>
software, including graphical visualization tools such as
Kalignvu, Boxshade, and WebLogo. On the other hand, not all
the motif information is preserved: in particular, the key
<br>
positions are lost. Only the top motif in glam2 output is
converted.</p>

<p style="margin-top: 1em">OPTIONS (DEFAULT SETTINGS) <br>
-o <br>
Output file (stdout).</p>

<p style="margin-top: 1em">-c <br>
Make a compact alignment. By default, residues that are
inserted between key positions are written as unaligned with
each other. This best reflects glam2&Acirc;&acute;s
intention, but it <br>
can make the alignment large and full of gaps. With -c,
inserted residues are written as arbitrarily aligned with
each other, just as they appear in the glam2 output.</p>

<p style="margin-top: 1em">-f <br>
Sequence file to make a &acirc;global&acirc; alignment by
adding flanking sequences from the original FASTA-format
sequence file. The flanking sequences will be written as
either <br>
unaligned with each other or arbitrarily aligned, depending
on the -c option. The sequences should have unique names and
their order should be unchanged.</p>

<p style="margin-top: 1em">SEE ALSO <br>
boxshade(1), glam2(1), glam2mask(1), glam2-purge(1),
glam2scan(1)</p>

<p style="margin-top: 1em">The full Hypertext documentation
of GLAM2 is available online at
http://bioinformatics.org.au/glam2/ or on this computer in
/usr/share/doc/glam2/.</p>

<p style="margin-top: 1em">REFERENCE <br>
If you use GLAM2, please cite: MC Frith, NFW Saunders, B
Kobe, TL Bailey (2008) Discovering sequence motifs with
arbitrary insertions and deletions, PLoS Computational
Biology <br>
(in press).</p>

<p style="margin-top: 1em">AUTHORS <br>
Martin Frith <br>
Author of GLAM2.</p>

<p style="margin-top: 1em">Timothy Bailey <br>
Author of GLAM2.</p>

<p style="margin-top: 1em">Charles Plessy
&lt;plessy@debian.org&gt; <br>
Formatted this manpage in DocBook XML for the Debian
distribution.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
The source code and the documentation of GLAM2 are released
in the public domain.</p>

<p style="margin-top: 1em">GLAM2 1056 05/19/2008
GLAM2FORMAT(1)</p>
<hr>
</body>
</html>
