<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:13:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GLAM2-PURGE(1) glam2 Manual GLAM2-PURGE(1)</p>

<p style="margin-top: 1em">NAME <br>
glam2-purge - Removes redundant sequences from a FASTA
file</p>

<p style="margin-top: 1em">SYNOPSIS <br>
glam2-purge file score [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
glam2-purge is a modified version of Andrew
Neuwald&Acirc;&acute;s purge program that removes redundant
sequences from a FASTA file. This is recommended in order to
prevent highly similar <br>
sequences distorting the search for motifs. Purge works with
either DNA or protein sequences and creates an output file
such that no two sequences have a (gapless) local <br>
alignment score greater than a threshold specified by the
user. The output file is named &lt;file&gt;.&lt;score&gt;.
The alignment score is based on the BLOSUM62 matrix for
proteins, and on <br>
a +5/-1 scoring scheme for DNA. Purge can also be used to
mask tandem repeats. It uses the XNU program for this
purpose.</p>

<p style="margin-top: 1em">OPTIONS <br>
-n <br>
Sequences are DNA (default: protein).</p>

<p style="margin-top: 1em">-b <br>
Use blast heuristic method (default for protein).</p>

<p style="margin-top: 1em">-e <br>
Use an exhaustive method (default for DNA).</p>

<p style="margin-top: 1em">-q <br>
Keep first sequence in the set.</p>

<p style="margin-top: 1em">-x <br>
Use xnu to mask protein tandem repeats.</p>

<p style="margin-top: 1em">SEE ALSO <br>
glam2(1), glam2format(1), glam2mask(1), glam2scan(1),
xnu(1)</p>

<p style="margin-top: 1em">The full Hypertext documentation
of GLAM2 is available online at
http://bioinformatics.org.au/glam2/ or on this computer in
/usr/share/doc/glam2/.</p>

<p style="margin-top: 1em">REFERENCES <br>
Purge was written by Andy Neuwald and is described in more
detail in Neuwald et al., &quot;Gibbs motif sampling:
detection of bacterial outer membrane protein repeats&quot;,
Protein <br>
Science, 4:1618&acirc;1632, 1995. Please cite it if you use
Purge.</p>

<p style="margin-top: 1em">If you use GLAM2, please cite:
MC Frith, NFW Saunders, B Kobe, TL Bailey (2008) Discovering
sequence motifs with arbitrary insertions and deletions,
PLoS Computational Biology <br>
(in press).</p>

<p style="margin-top: 1em">AUTHORS <br>
Andrew Neuwald <br>
Author of purge, renamed glam2-purge in Debian.</p>

<p style="margin-top: 1em">Martin Frith <br>
Modified purge to be ANSI standard C and improved the user
interface.</p>

<p style="margin-top: 1em">Timothy Bailey <br>
Modified purge to be ANSI standard C and improved the user
interface.</p>

<p style="margin-top: 1em">Charles Plessy
&lt;plessy@debian.org&gt; <br>
Formatted this manpage in DocBook XML for the Debian
distribution.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
The source code and the documentation of Purge and GLAM2 are
released in the public domain.</p>

<p style="margin-top: 1em">GLAM2 1056 05/19/2008
GLAM2-PURGE(1)</p>
<hr>
</body>
</html>
