<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GMAP(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GMAP(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">GMAP(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gmap - Genomic Mapping and Alignment Program
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gmap</b> [ <i>OPTIONS</i>...] <i>&lt;FASTA files</i>...<i>&gt;, or</i> cat
  &lt;FASTA files...&gt; | gmap [OPTIONS...]
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_options_(must_include_-d_or_-g)"><a class="selflink" href="#Input_options_(must_include_-d_or_-g)">Input
  options (must include <b>-d</b> or <b>-g</b>)</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b>, <b>--dir</b>=<i>directory</i></dt>
  <dd class="It-tag">Genome directory. Default (as specified by
      <b>--with-gmapdb</b> to the configure program) is
    <i>/var/cache/gmap</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b>, <b>--db</b>=<i>STRING</i></dt>
  <dd class="It-tag">Genome database. If argument is '?' (with the quotes), this
      command lists available databases.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b>, <b>--kmer</b>=<i>INT</i></dt>
  <dd class="It-tag">kmer size to use in genome database (allowed values: 16 or
      less). If not specified, the program will find the highest available kmer
      size in the genome database</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sampling</b>=<i>INT</i></dt>
  <dd class="It-tag">Sampling to use in genome database. If not specified, the
      program will find the smallest available sampling value in the genome
      database within selected k-mer size</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-G</b>, <b>--genomefull</b></dt>
  <dd class="It-tag">Use full genome (all ASCII chars allowed; built explicitly
      during setup), not compressed version</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b>, <b>--gseg</b>=<i>filename</i></dt>
  <dd class="It-tag">User-supplied genomic segment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b>, <b>--selfalign</b></dt>
  <dd class="It-tag">Align one sequence against itself in FASTA format via stdin
      (Useful for getting protein translation of a nucleotide sequence)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-2</b>, <b>--pairalign</b></dt>
  <dd class="It-tag">Align two sequences in FASTA format via stdin, first one
      being genomic and second one being cDNA</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cmdline</b>=<i>STRING</i>,STRING</dt>
  <dd class="It-tag">Align these two sequences provided on the command line,
      first one being genomic and second one being cDNA</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b>, <b>--part</b>=<i>INT</i>/INT</dt>
  <dd class="It-tag">Process only the i-th out of every n sequences e.g., 0/100
      or 99/100 (useful for distributing jobs to a computer farm).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--input-buffer-size</b>=<i>INT</i></dt>
  <dd class="It-tag">Size of input buffer (program reads this many sequences at
      a time for efficiency) (default 1000)</dd>
</dl>
<div class="Pp"></div>
Computation options
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b>, <b>--batch</b>=<i>INT</i></dt>
  <dd class="It-tag">Batch mode (default = 2)
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     Mode Offsets Positions Genome
    <br/>
     0 see note mmap mmap
    <br/>
     1 see note mmap &amp; preload mmap
    <br/>
     (default) 2 see note mmap &amp; preload mmap &amp; preload
    <br/>
     3 see note allocate mmap &amp; preload
    <br/>
     4 see note allocate allocate
    <br/>
     5 expand allocate allocate</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Note: For a single sequence, all data structures use
    mmap</dt>
  <dd class="It-tag">If mmap not available and allocate not chosen, then will
      use fileio (very slow)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Note about <b>--batch</b> and offsets: Expansion of offsets
    can be controlled</dt>
  <dd class="It-tag">independently by the <b>--expand-offsets</b> flag. The
      <b>--batch</b>= <i>5</i> option is equivalent to <b>--batch</b>=<i>4</i>
      plus <b>--expand-offsets</b>=<i>1</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--expand-offsets</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether to expand the genomic offsets index Values: 0 (no,
      default), or 1 (yes). Expansion gives faster alignment, but requires more
      memory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nosplicing</b></dt>
  <dd class="It-tag">Turns off splicing (useful for aligning genomic sequences
      onto a genome)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-intronlength</b>=<i>INT</i></dt>
  <dd class="It-tag">Min length for one internal intron (default 9). Below this
      size, a genomic gap will be considered a deletion rather than an
    intron.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-intronlength-middle</b>=<i>INT</i></dt>
  <dd class="It-tag">Max length for one internal intron (default 200000). Note:
      for backward compatibility, the <b>-K</b> flag will set both
      <b>--max-intronlength-middle</b> and <b>--max-intronlength-ends</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-intronlength-ends</b>=<i>INT</i></dt>
  <dd class="It-tag">Max length for first or last intron (default 10000). Note:
      for backward compatibility, the <b>-K</b> flag will set both
      <b>--max-intronlength-middle</b> and <b>--max-intronlength-ends</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trim-end-exons</b>=<i>INT</i></dt>
  <dd class="It-tag">Trim end exons with fewer than given number of matches (in
      nt, default 12)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b>, <b>--localsplicedist</b>=<i>INT</i></dt>
  <dd class="It-tag">Max length for known splice sites at ends of sequence
      (default 2000000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b>, <b>--totallength</b>=<i>INT</i></dt>
  <dd class="It-tag">Max total intron length (default 2400000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-x</b>, <b>--chimera-margin</b>=<i>INT</i></dt>
  <dd class="It-tag">Amount of unaligned sequence that triggers search for the
      remaining sequence (default 30). Enables alignment of chimeric reads, and
      may help with some non-chimeric reads. To turn off, set to zero.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-chimeras</b></dt>
  <dd class="It-tag">Turns off finding of chimeras. Same effect as
      <b>--chimera-margin</b>= <i>0</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>, <b>--nthreads</b>=<i>INT</i></dt>
  <dd class="It-tag">Number of worker threads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>, <b>--chrsubset</b>=<i>string</i></dt>
  <dd class="It-tag">Limit search to given chromosome</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-z</b>, <b>--direction</b>=<i>STRING</i></dt>
  <dd class="It-tag">cDNA direction (sense_force, antisense_force, sense_filter,
      antisense_filter,or auto (default))</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--canonical-mode</b>=<i>INT</i></dt>
  <dd class="It-tag">Reward for canonical and semi-canonical introns 0=low
      reward, 1=high reward (default), 2=low reward for high-identity sequences
      and high reward otherwise</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cross-species</b></dt>
  <dd class="It-tag">Use a more sensitive search for canonical splicing, which
      helps especially for cross-species alignments and other difficult
    cases</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--allow-close-indels</b>=<i>INT</i></dt>
  <dd class="It-tag">Allow an insertion and deletion close to each other (0=no,
      1=yes (default), 2=only for high-quality alignments)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--microexon-spliceprob</b>=<i>FLOAT</i></dt>
  <dd class="It-tag">Allow microexons only if one of the splice site
      probabilities is greater than this value (default 0.95)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cmetdir</b>=<i>STRING</i></dt>
  <dd class="It-tag">Directory for methylcytosine index files (created using
      cmetindex) (default is location of genome index files specified using
      <b>-D</b>, <b>-V</b>, and <b>-d</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--atoidir</b>=<i>STRING</i></dt>
  <dd class="It-tag">Directory for A-to-I RNA editing index files (created using
      atoiindex) (default is location of genome index files specified using
      <b>-D</b>, <b>-V</b>, and <b>-d</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mode</b>=<i>STRING</i></dt>
  <dd class="It-tag">Alignment mode: standard (default), cmet-stranded,
      cmet-nonstranded, atoi-stranded, atoi-nonstranded, ttoc-stranded, or
      ttoc-nonstranded. Non-standard modes requires you to have previously run
      the cmetindex or atoiindex programs (which also cover the ttoc modes) on
      the genome</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b>, <b>--prunelevel</b></dt>
  <dd class="It-tag">Pruning level: 0=no pruning (default), 1=poor seqs,
      2=repetitive seqs, 3=poor and repetitive</dd>
</dl>
<div class="Pp"></div>
Output types
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b>, <b>--summary</b></dt>
  <dd class="It-tag">Show summary of alignments only</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b>, <b>--align</b></dt>
  <dd class="It-tag">Show alignments</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-3</b>, <b>--continuous</b></dt>
  <dd class="It-tag">Show alignment in three continuous lines</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-4</b>, <b>--continuous-by-exon</b></dt>
  <dd class="It-tag">Show alignment in three lines per exon</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b>, <b>--compress</b></dt>
  <dd class="It-tag">Print output in compressed format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b>, <b>--exons</b>=<i>STRING</i></dt>
  <dd class="It-tag">Print exons (&quot;cdna&quot; or &quot;genomic&quot;)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P</b>, <b>--protein_dna</b></dt>
  <dd class="It-tag">Print protein sequence (cDNA)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b>, <b>--protein_gen</b></dt>
  <dd class="It-tag">Print protein sequence (genomic)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>, <b>--format</b>=<i>INT</i></dt>
  <dd class="It-tag">Other format for output (also note the <b>-A</b> and
      <b>-S</b> options and other options listed under Output types):
    <br/>
     psl (or 1) = PSL (BLAT) format,
    <br/>
     gff3_gene (or 2) = GFF3 gene format,
    <br/>
     gff3_match_cdna (or 3) = GFF3 cDNA_match format,
    <br/>
     gff3_match_est (or 4) = GFF3 EST_match format,
    <br/>
     splicesites (or 6) = splicesites output (for GSNAP splicing file),
    <br/>
     introns = introns output (for GSNAP splicing file),
    <br/>
     map_exons (or 7) = IIT FASTA exon map format,
    <br/>
     map_ranges (or 8) = IIT FASTA range map format,
    <br/>
     coords (or 9) = coords in table format,
    <br/>
     sampe = SAM format (setting paired_read bit in flag),
    <br/>
     samse = SAM format (without setting paired_read bit)</dd>
</dl>
<div class="Pp"></div>
Output options
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b>, <b>--npaths</b>=<i>INT</i></dt>
  <dd class="It-tag">Maximum number of paths to show (default 5). If set to 1,
      GMAP will not report chimeric alignments, since those imply two paths. If
      you want a single alignment plus chimeric alignments, then set this to be
      0.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--suboptimal-score</b>=<i>INT</i></dt>
  <dd class="It-tag">Report only paths whose score is within this value of the
      best path. By default, if this option is not provided,
    <br/>
     the program prints all paths found.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-O</b>, <b>--ordered</b></dt>
  <dd class="It-tag">Print output in same order as input (relevant only if there
      is more than one worker thread)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-5</b>, <b>--md5</b></dt>
  <dd class="It-tag">Print MD5 checksum for each query sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>, <b>--chimera-overlap</b></dt>
  <dd class="It-tag">Overlap to show, if any, at chimera breakpoint</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--failsonly</b></dt>
  <dd class="It-tag">Print only failed alignments, those with no results</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nofails</b></dt>
  <dd class="It-tag">Exclude printing of failed alignments</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>, <b>--snpsdir</b>=<i>STRING</i></dt>
  <dd class="It-tag">Directory for SNPs index files (created using snpindex)
      (default is location of genome index files specified using <b>-D</b> and
      <b>-d</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--use-snps</b>=<i>STRING</i></dt>
  <dd class="It-tag">Use database containing known SNPs (in &lt;STRING&gt;.iit,
      built previously using snpindex) for tolerance to SNPs</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--split-output</b>=<i>STRING</i></dt>
  <dd class="It-tag">Basename for multiple-file output, separately for
      nomapping,
    <br/>
     uniq, mult, (and chimera, if <b>--chimera-margin</b> is selected)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--failed-input</b>=<i>STRING</i></dt>
  <dd class="It-tag">Print completely failed alignments as input FASTA or FASTQ
      format to the given file. If the <b>--split-output</b> flag is also given,
      this file is generated in addition to the output in the .nomapping
    file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--append-output</b></dt>
  <dd class="It-tag">When <b>--split-output</b> or <b>--failedinput</b> is
      given, this flag will append output to the existing files. Otherwise, the
      default is to create new files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--output-buffer-size</b>=<i>INT</i></dt>
  <dd class="It-tag">Buffer size, in queries, for output thread (default 1000).
      When the number of results to be printed exceeds this size, the worker
      threads are halted until the backlog is cleared</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--translation-code</b>=<i>INT</i></dt>
  <dd class="It-tag">Genetic code used for translating codons to amino acids and
      computing CDS Integer value (default=1) corresponds to an available code
      at http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b>, <b>--fulllength</b></dt>
  <dd class="It-tag">Assume full-length protein, starting with Met</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b>, <b>--cdsstart</b>=<i>INT</i></dt>
  <dd class="It-tag">Translate codons from given nucleotide (1-based)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b>, <b>--truncate</b></dt>
  <dd class="It-tag">Truncate alignment around full-length protein, Met to Stop
      Implies <b>-F</b> flag.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Y</b>, <b>--tolerant</b></dt>
  <dd class="It-tag">Translates cDNA with corrections for frameshifts</dd>
</dl>
<div class="Pp"></div>
Options for GFF3 output
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gff3-add-separators</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether to add a ### separator after each query sequence
      Values: 0 (no), 1 (yes, default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gff3-swap-phase</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether to swap phase (0 =&gt; 0, 1 =&gt; 2, 2 =&gt; 1) in
      gff3_gene format Needed by some analysis programs, but deviates from GFF3
      specification Values: 0 (no, default), 1 (yes)</dd>
</dl>
<div class="Pp"></div>
Options for SAM output
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-sam-headers</b></dt>
  <dd class="It-tag">Do not print headers beginning with '@'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-use-0M</b></dt>
  <dd class="It-tag">Insert 0M in CIGAR between adjacent insertions and
      deletions Required by Picard, but can cause errors in other tools</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-extended-cigar</b></dt>
  <dd class="It-tag">Use extended CIGAR format (using X and = symbols instead of
      M,
    <br/>
     to indicate matches and mismatches, respectively</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--force-xs-dir</b></dt>
  <dd class="It-tag">For RNA-Seq alignments, disallows XS:A:? when the sense
      direction is unclear, and replaces this value arbitrarily with XS:A:+. May
      be useful for some programs, such as Cufflinks, that cannot handle XS:A:?.
      However, if you use this flag, the reported value of XS:A:+ in these cases
      will not be meaningful.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--md-lowercase-snp</b></dt>
  <dd class="It-tag">In MD string, when known SNPs are given by the <b>-v</b>
      flag,
    <br/>
     prints difference nucleotides as lower-case when they,
    <br/>
     differ from reference but match a known alternate allele</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--action-if-cigar-error</b></dt>
  <dd class="It-tag">Action to take if there is a disagreement between CIGAR
      length and sequence length Allowed values: ignore, warning (default),
      abort</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read-group-id</b>=<i>STRING</i></dt>
  <dd class="It-tag">Value to put into read-group id (RG-ID) field</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read-group-name</b>=<i>STRING</i></dt>
  <dd class="It-tag">Value to put into read-group name (RG-SM) field</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read-group-library</b>=<i>STRING</i></dt>
  <dd class="It-tag">Value to put into read-group library (RG-LB) field</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read-group-platform</b>=<i>STRING</i></dt>
  <dd class="It-tag">Value to put into read-group library (RG-PL) field</dd>
</dl>
<div class="Pp"></div>
Options for quality scores
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quality-protocol</b>=<i>STRING</i></dt>
  <dd class="It-tag">Protocol for input quality scores. Allowed values: illumina
      (ASCII 64-126) (equivalent to <b>-J</b> 64 <b>-j</b> <b>-31</b>) sanger
      (ASCII 33-126) (equivalent to <b>-J</b> 33 <b>-j</b> 0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Default is sanger (no quality print shift)</dt>
  <dd class="It-tag">SAM output files should have quality scores in sanger
      protocol</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Or you can specify the print shift with this flag:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-j</b>, <b>--quality-print-shift</b>=<i>INT</i></dt>
  <dd class="It-tag">Shift FASTQ quality scores by this amount in output
      (default is 0 for sanger protocol; to change Illumina input to Sanger
      output, select <b>-31</b>)</dd>
</dl>
<div class="Pp"></div>
External map file options
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b>, <b>--mapdir</b>=<i>directory</i></dt>
  <dd class="It-tag">Map directory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>, <b>--map</b>=<i>iitfile</i></dt>
  <dd class="It-tag">Map file. If argument is '?' (with the quotes),
    <br/>
     this lists available map files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>, <b>--mapexons</b></dt>
  <dd class="It-tag">Map each exon separately</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b>, <b>--mapboth</b></dt>
  <dd class="It-tag">Report hits from both strands of genome</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b>, <b>--flanking</b>=<i>INT</i></dt>
  <dd class="It-tag">Show flanking hits (default 0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--print-comment</b></dt>
  <dd class="It-tag">Show comment line for each hit</dd>
</dl>
<div class="Pp"></div>
Alignment output options
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b>, <b>--nolengths</b></dt>
  <dd class="It-tag">No intron lengths in alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-I</b>, <b>--invertmode</b>=<i>INT</i></dt>
  <dd class="It-tag">Mode for alignments to genomic (-) strand: 0=Don't invert
      the cDNA (default) 1=Invert cDNA and print genomic (-) strand 2=Invert
      cDNA and print genomic (+) strand</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b>, <b>--introngap</b>=<i>INT</i></dt>
  <dd class="It-tag">Nucleotides to show on each end of intron (default 3)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b>, <b>--wraplength</b>=<i>INT</i></dt>
  <dd class="It-tag">Wrap length for alignment (default 50)</dd>
</dl>
<div class="Pp"></div>
Filtering output options
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-trimmed-coverage</b>=<i>FLOAT</i></dt>
  <dd class="It-tag">Do not print alignments with trimmed coverage less this
      value (default=0.0, which means no filtering) Note that chimeric
      alignments will be output regardless of this filter</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-identity</b>=<i>FLOAT</i></dt>
  <dd class="It-tag">Do not print alignments with identity less this value
      (default=0.0, which means no filtering) Note that chimeric alignments will
      be output regardless of this filter Help options</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--check</b></dt>
  <dd class="It-tag">Check compiler assumptions</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">Show version</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">Show this help message
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Other tools of GMAP suite are located in /usr/lib/gmap</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">gmap 2017-01-14-1</td>
  </tr>
</table>
</body>
</html>
