<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GMAP_BUILD(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GMAP_BUILD(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">GMAP_BUILD(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gmap_build - Tool for genome database creation for GMAP or GSNAP
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gmap_build</b> [ <i>options</i>...] <i>-d &lt;genomename&gt;
  &lt;fasta_files&gt;</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
gmap_build: Builds a gmap database for a genome to be used by GMAP or GSNAP.
  Part of GMAP package, version 2017-01-14.
<div class="Pp"></div>
A simplified alternative to using the program gmap_setup, which creates a
  Makefile.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b>, <b>--dir</b>=<i>STRING</i></dt>
  <dd class="It-tag">Destination directory for installation (defaults to gmapdb
      directory specified at configure time)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b>, <b>--db</b>=<i>STRING</i></dt>
  <dd class="It-tag">Genome name</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b>, <b>--names</b>=<i>STRING</i></dt>
  <dd class="It-tag">Substitute names for chromosomes, provided in a file. The
      file should have one line for each chromosome name to be changed, with the
      original FASTA name in column 1 and the desired chromosome name in column
      2. This provides an easy way to change the names of chromosomes, for
      example, to add or remove the &quot;chr&quot; prefix. Column 2 may be
      blank, which indicates no name change. This file can also be combined with
      <b>--sort</b>=<i>names</i> to provide a particular order for the
      chromosomes in the genome index.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b>, <b>--mdflag</b>=<i>STRING</i></dt>
  <dd class="It-tag">Use MD file from NCBI for mapping contigs to chromosomal
      coordinates</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b>, <b>--contigs-are-mapped</b></dt>
  <dd class="It-tag">Find a chromosomal region in each FASTA header line. Useful
      for contigs that have been mapped to chromosomal coordinates. Ignored if
      the <b>--mdflag</b> is provided.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-z</b>, <b>--compression</b>=<i>STRING</i></dt>
  <dd class="It-tag">Use given compression types (separated by commas; default
      is bitpack64) bitpack64 - optimized for modern computers with SIMD
      instructions (recommended) all - create all available compression types,
      currently bitpack64 none - do not compress offset files (I believe this is
      no longer supported)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b>, <b>--kmer</b>=<i>INT</i></dt>
  <dd class="It-tag">k-mer value for genomic index (allowed: 15 or less, default
      is 15)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> INT</dt>
  <dd class="It-tag">sampling interval for genomoe (allowed: 1-3, default
    3)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>, <b>--sort</b>=<i>STRING</i></dt>
  <dd class="It-tag">Sort chromosomes using given method: none - use chromosomes
      as found in FASTA file(s) alpha - sort chromosomes alphabetically (chr10
      before chr 1) numeric-alpha - chr1, chr1U, chr2, chrM, chrU, chrX, chrY
      chrom - chr1, chr2, chrM, chrX, chrY, chr1U, chrU names - sort chromosomes
      based on file provided to <b>--names</b> flag</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b>, <b>--gunzip</b></dt>
  <dd class="It-tag">Files are gzipped, so need to gunzip each file first</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b>, <b>--fasta-pipe</b>=<i>STRING</i></dt>
  <dd class="It-tag">Interpret argument as a command, instead of a list of FASTA
      files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b> INT</dt>
  <dd class="It-tag">Wait (sleep) this many seconds after each step (default
    2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>, <b>--circular</b>=<i>STRING</i></dt>
  <dd class="It-tag">Circular chromosomes (either a list of chromosomes
      separated by a comma, or a filename containing circular chromosomes, one
      per line). If you use the <b>--names</b> feature, then you should use the
      original name of the chromosome, not the substitute name, for this
    option.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-2</b>, <b>--altscaffold</b>=<i>STRING</i></dt>
  <dd class="It-tag">File with alt scaffold info, listing alternate scaffolds,
      one per line, tab-delimited, with the following fields: (1) alt_scaf_acc,
      (2) parent_name, (3) orientation, (4) alt_scaf_start, (5) alt_scaf_stop,
      (6) parent_start, (7) parent_end.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>, <b>--nmessages</b>=<i>INT</i></dt>
  <dd class="It-tag">Maximum number of messages (warnings, contig reports) to
      report (default 50)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--build-sarray</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether to build suffix array: 0=no, 1=yes (default)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Obsolete_options:"><a class="selflink" href="#Obsolete_options:">Obsolete
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b> STRING</dt>
  <dd class="It-tag">Temporary build directory (may need to specify if you run
      out of space in your current directory) This is no longer necessary, since
      gmap_build now builds directly in the destination (or <b>-D</b>)
      directory.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Other tools of GMAP suite are located in /usr/lib/gmap</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">gmap_build 2017-01-14-1</td>
  </tr>
</table>
</body>
</html>
