<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:13:58 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMAP(1) User Commands GMAP(1)</p>

<p style="margin-top: 1em">NAME <br>
gmap - Genomic Mapping and Alignment Program</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmap [OPTIONS...] &lt;FASTA files...&gt;, or cat &lt;FASTA
files...&gt; | gmap [OPTIONS...]</p>

<p style="margin-top: 1em">OPTIONS <br>
Input options (must include -d or -g) <br>
-D, --dir=directory <br>
Genome directory. Default (as specified by --with-gmapdb to
the configure program) is /var/cache/gmap</p>

<p style="margin-top: 1em">-d, --db=STRING <br>
Genome database. If argument is &rsquo;?&rsquo; (with the
quotes), this command lists available databases.</p>

<p style="margin-top: 1em">-k, --kmer=INT <br>
kmer size to use in genome database (allowed values: 16 or
less). If not specified, the program will find the highest
available kmer size in the genome database</p>

<p style="margin-top: 1em">--sampling=INT <br>
Sampling to use in genome database. If not specified, the
program will find the smallest available sampling value in
the genome database within selected k-mer size</p>

<p style="margin-top: 1em">-G, --genomefull <br>
Use full genome (all ASCII chars allowed; built explicitly
during setup), not compressed version</p>

<p style="margin-top: 1em">-g, --gseg=filename <br>
User-supplied genomic segment</p>

<p style="margin-top: 1em">-1, --selfalign <br>
Align one sequence against itself in FASTA format via stdin
(Useful for getting protein translation of a nucleotide
sequence)</p>

<p style="margin-top: 1em">-2, --pairalign <br>
Align two sequences in FASTA format via stdin, first one
being genomic and second one being cDNA</p>

<p style="margin-top: 1em">--cmdline=STRING,STRING <br>
Align these two sequences provided on the command line,
first one being genomic and second one being cDNA</p>

<p style="margin-top: 1em">-q, --part=INT/INT <br>
Process only the i-th out of every n sequences e.g., 0/100
or 99/100 (useful for distributing jobs to a computer
farm).</p>

<p style="margin-top: 1em">--input-buffer-size=INT <br>
Size of input buffer (program reads this many sequences at a
time for efficiency) (default 1000)</p>

<p style="margin-top: 1em">Computation options</p>

<p style="margin-top: 1em">-B, --batch=INT <br>
Batch mode (default = 2)</p>

<p style="margin-top: 1em">Mode Offsets Positions Genome
<br>
0 see note mmap mmap <br>
1 see note mmap &amp; preload mmap <br>
(default) 2 see note mmap &amp; preload mmap &amp; preload
<br>
3 see note allocate mmap &amp; preload <br>
4 see note allocate allocate <br>
5 expand allocate allocate</p>

<p style="margin-top: 1em">Note: For a single sequence, all
data structures use mmap <br>
If mmap not available and allocate not chosen, then will use
fileio (very slow)</p>

<p style="margin-top: 1em">Note about --batch and offsets:
Expansion of offsets can be controlled <br>
independently by the --expand-offsets flag. The --batch=5
option is equivalent to --batch=4 plus
--expand-offsets=1</p>

<p style="margin-top: 1em">--expand-offsets=INT <br>
Whether to expand the genomic offsets index Values: 0 (no,
default), or 1 (yes). Expansion gives faster alignment, but
requires more memory</p>

<p style="margin-top: 1em">--nosplicing <br>
Turns off splicing (useful for aligning genomic sequences
onto a genome)</p>

<p style="margin-top: 1em">--min-intronlength=INT <br>
Min length for one internal intron (default 9). Below this
size, a genomic gap will be considered a deletion rather
than an intron.</p>

<p style="margin-top: 1em">--max-intronlength-middle=INT
<br>
Max length for one internal intron (default 200000). Note:
for backward compatibility, the -K flag will set both
--max-intronlength-middle and --max-intronlength-ends.</p>

<p style="margin-top: 1em">--max-intronlength-ends=INT <br>
Max length for first or last intron (default 10000). Note:
for backward compatibility, the -K flag will set both
--max-intronlength-middle and --max-intronlength-ends.</p>

<p style="margin-top: 1em">--trim-end-exons=INT <br>
Trim end exons with fewer than given number of matches (in
nt, default 12)</p>

<p style="margin-top: 1em">-w, --localsplicedist=INT <br>
Max length for known splice sites at ends of sequence
(default 2000000)</p>

<p style="margin-top: 1em">-L, --totallength=INT <br>
Max total intron length (default 2400000)</p>

<p style="margin-top: 1em">-x, --chimera-margin=INT <br>
Amount of unaligned sequence that triggers search for the
remaining sequence (default 30). Enables alignment of
chimeric reads, and may help with some non-chimeric reads.
<br>
To turn off, set to zero.</p>

<p style="margin-top: 1em">--no-chimeras <br>
Turns off finding of chimeras. Same effect as
--chimera-margin=0</p>

<p style="margin-top: 1em">-t, --nthreads=INT <br>
Number of worker threads</p>

<p style="margin-top: 1em">-c, --chrsubset=string <br>
Limit search to given chromosome</p>

<p style="margin-top: 1em">-z, --direction=STRING <br>
cDNA direction (sense_force, antisense_force, sense_filter,
antisense_filter,or auto (default))</p>

<p style="margin-top: 1em">--canonical-mode=INT <br>
Reward for canonical and semi-canonical introns 0=low
reward, 1=high reward (default), 2=low reward for
high-identity sequences and high reward otherwise</p>

<p style="margin-top: 1em">--cross-species <br>
Use a more sensitive search for canonical splicing, which
helps especially for cross-species alignments and other
difficult cases</p>

<p style="margin-top: 1em">--allow-close-indels=INT <br>
Allow an insertion and deletion close to each other (0=no,
1=yes (default), 2=only for high-quality alignments)</p>

<p style="margin-top: 1em">--microexon-spliceprob=FLOAT
<br>
Allow microexons only if one of the splice site
probabilities is greater than this value (default 0.95)</p>

<p style="margin-top: 1em">--cmetdir=STRING <br>
Directory for methylcytosine index files (created using
cmetindex) (default is location of genome index files
specified using -D, -V, and -d)</p>

<p style="margin-top: 1em">--atoidir=STRING <br>
Directory for A-to-I RNA editing index files (created using
atoiindex) (default is location of genome index files
specified using -D, -V, and -d)</p>

<p style="margin-top: 1em">--mode=STRING <br>
Alignment mode: standard (default), cmet-stranded,
cmet-nonstranded, atoi-stranded, atoi-nonstranded,
ttoc-stranded, or ttoc-nonstranded. Non-standard modes
requires you <br>
to have previously run the cmetindex or atoiindex programs
(which also cover the ttoc modes) on the genome</p>

<p style="margin-top: 1em">-p, --prunelevel <br>
Pruning level: 0=no pruning (default), 1=poor seqs,
2=repetitive seqs, 3=poor and repetitive</p>

<p style="margin-top: 1em">Output types</p>

<p style="margin-top: 1em">-S, --summary <br>
Show summary of alignments only</p>

<p style="margin-top: 1em">-A, --align <br>
Show alignments</p>

<p style="margin-top: 1em">-3, --continuous <br>
Show alignment in three continuous lines</p>

<p style="margin-top: 1em">-4, --continuous-by-exon <br>
Show alignment in three lines per exon</p>

<p style="margin-top: 1em">-Z, --compress <br>
Print output in compressed format</p>

<p style="margin-top: 1em">-E, --exons=STRING <br>
Print exons (&quot;cdna&quot; or &quot;genomic&quot;)</p>

<p style="margin-top: 1em">-P, --protein_dna <br>
Print protein sequence (cDNA)</p>

<p style="margin-top: 1em">-Q, --protein_gen <br>
Print protein sequence (genomic)</p>

<p style="margin-top: 1em">-f, --format=INT <br>
Other format for output (also note the -A and -S options and
other options listed under Output types): <br>
psl (or 1) = PSL (BLAT) format, <br>
gff3_gene (or 2) = GFF3 gene format, <br>
gff3_match_cdna (or 3) = GFF3 cDNA_match format, <br>
gff3_match_est (or 4) = GFF3 EST_match format, <br>
splicesites (or 6) = splicesites output (for GSNAP splicing
file), <br>
introns = introns output (for GSNAP splicing file), <br>
map_exons (or 7) = IIT FASTA exon map format, <br>
map_ranges (or 8) = IIT FASTA range map format, <br>
coords (or 9) = coords in table format, <br>
sampe = SAM format (setting paired_read bit in flag), <br>
samse = SAM format (without setting paired_read bit)</p>

<p style="margin-top: 1em">Output options</p>

<p style="margin-top: 1em">-n, --npaths=INT <br>
Maximum number of paths to show (default 5). If set to 1,
GMAP will not report chimeric alignments, since those imply
two paths. If you want a single alignment plus <br>
chimeric alignments, then set this to be 0.</p>

<p style="margin-top: 1em">--suboptimal-score=INT <br>
Report only paths whose score is within this value of the
best path. By default, if this option is not provided, <br>
the program prints all paths found.</p>

<p style="margin-top: 1em">-O, --ordered <br>
Print output in same order as input (relevant only if there
is more than one worker thread)</p>

<p style="margin-top: 1em">-5, --md5 <br>
Print MD5 checksum for each query sequence</p>

<p style="margin-top: 1em">-o, --chimera-overlap <br>
Overlap to show, if any, at chimera breakpoint</p>

<p style="margin-top: 1em">--failsonly <br>
Print only failed alignments, those with no results</p>

<p style="margin-top: 1em">--nofails <br>
Exclude printing of failed alignments</p>

<p style="margin-top: 1em">-V, --snpsdir=STRING <br>
Directory for SNPs index files (created using snpindex)
(default is location of genome index files specified using
-D and -d)</p>

<p style="margin-top: 1em">-v, --use-snps=STRING <br>
Use database containing known SNPs (in &lt;STRING&gt;.iit,
built previously using snpindex) for tolerance to SNPs</p>

<p style="margin-top: 1em">--split-output=STRING <br>
Basename for multiple-file output, separately for nomapping,
<br>
uniq, mult, (and chimera, if --chimera-margin is
selected)</p>

<p style="margin-top: 1em">--failed-input=STRING <br>
Print completely failed alignments as input FASTA or FASTQ
format to the given file. If the --split-output flag is also
given, this file is generated in addition to the <br>
output in the .nomapping file.</p>

<p style="margin-top: 1em">--append-output <br>
When --split-output or --failedinput is given, this flag
will append output to the existing files. Otherwise, the
default is to create new files.</p>

<p style="margin-top: 1em">--output-buffer-size=INT <br>
Buffer size, in queries, for output thread (default 1000).
When the number of results to be printed exceeds this size,
the worker threads are halted until the backlog is <br>
cleared</p>

<p style="margin-top: 1em">--translation-code=INT <br>
Genetic code used for translating codons to amino acids and
computing CDS Integer value (default=1) corresponds to an
available code at http://www.ncbi.nlm.nih.gov/Taxon&acirc;
<br>
omy/Utils/wprintgc.cgi</p>

<p style="margin-top: 1em">-F, --fulllength <br>
Assume full-length protein, starting with Met</p>

<p style="margin-top: 1em">-a, --cdsstart=INT <br>
Translate codons from given nucleotide (1-based)</p>

<p style="margin-top: 1em">-T, --truncate <br>
Truncate alignment around full-length protein, Met to Stop
Implies -F flag.</p>

<p style="margin-top: 1em">-Y, --tolerant <br>
Translates cDNA with corrections for frameshifts</p>

<p style="margin-top: 1em">Options for GFF3 output</p>

<p style="margin-top: 1em">--gff3-add-separators=INT <br>
Whether to add a ### separator after each query sequence
Values: 0 (no), 1 (yes, default)</p>

<p style="margin-top: 1em">--gff3-swap-phase=INT <br>
Whether to swap phase (0 =&gt; 0, 1 =&gt; 2, 2 =&gt; 1) in
gff3_gene format Needed by some analysis programs, but
deviates from GFF3 specification Values: 0 (no, default), 1
(yes)</p>

<p style="margin-top: 1em">Options for SAM output</p>

<p style="margin-top: 1em">--no-sam-headers <br>
Do not print headers beginning with &rsquo;@&rsquo;</p>

<p style="margin-top: 1em">--sam-use-0M <br>
Insert 0M in CIGAR between adjacent insertions and deletions
Required by Picard, but can cause errors in other tools</p>

<p style="margin-top: 1em">--sam-extended-cigar <br>
Use extended CIGAR format (using X and = symbols instead of
M, <br>
to indicate matches and mismatches, respectively</p>

<p style="margin-top: 1em">--force-xs-dir <br>
For RNA-Seq alignments, disallows XS:A:? when the sense
direction is unclear, and replaces this value arbitrarily
with XS:A:+. May be useful for some programs, such as <br>
Cufflinks, that cannot handle XS:A:?. However, if you use
this flag, the reported value of XS:A:+ in these cases will
not be meaningful.</p>

<p style="margin-top: 1em">--md-lowercase-snp <br>
In MD string, when known SNPs are given by the -v flag, <br>
prints difference nucleotides as lower-case when they, <br>
differ from reference but match a known alternate allele</p>

<p style="margin-top: 1em">--action-if-cigar-error <br>
Action to take if there is a disagreement between CIGAR
length and sequence length Allowed values: ignore, warning
(default), abort</p>

<p style="margin-top: 1em">--read-group-id=STRING <br>
Value to put into read-group id (RG-ID) field</p>

<p style="margin-top: 1em">--read-group-name=STRING <br>
Value to put into read-group name (RG-SM) field</p>

<p style="margin-top: 1em">--read-group-library=STRING <br>
Value to put into read-group library (RG-LB) field</p>

<p style="margin-top: 1em">--read-group-platform=STRING
<br>
Value to put into read-group library (RG-PL) field</p>

<p style="margin-top: 1em">Options for quality scores</p>

<p style="margin-top: 1em">--quality-protocol=STRING <br>
Protocol for input quality scores. Allowed values: illumina
(ASCII 64-126) (equivalent to -J 64 -j -31) sanger (ASCII
33-126) (equivalent to -J 33 -j 0)</p>

<p style="margin-top: 1em">Default is sanger (no quality
print shift) <br>
SAM output files should have quality scores in sanger
protocol</p>

<p style="margin-top: 1em">Or you can specify the print
shift with this flag:</p>

<p style="margin-top: 1em">-j, --quality-print-shift=INT
<br>
Shift FASTQ quality scores by this amount in output (default
is 0 for sanger protocol; to change Illumina input to Sanger
output, select -31)</p>

<p style="margin-top: 1em">External map file options</p>

<p style="margin-top: 1em">-M, --mapdir=directory <br>
Map directory</p>

<p style="margin-top: 1em">-m, --map=iitfile <br>
Map file. If argument is &rsquo;?&rsquo; (with the quotes),
<br>
this lists available map files.</p>

<p style="margin-top: 1em">-e, --mapexons <br>
Map each exon separately</p>

<p style="margin-top: 1em">-b, --mapboth <br>
Report hits from both strands of genome</p>

<p style="margin-top: 1em">-u, --flanking=INT <br>
Show flanking hits (default 0)</p>

<p style="margin-top: 1em">--print-comment <br>
Show comment line for each hit</p>

<p style="margin-top: 1em">Alignment output options</p>

<p style="margin-top: 1em">-N, --nolengths <br>
No intron lengths in alignment</p>

<p style="margin-top: 1em">-I, --invertmode=INT <br>
Mode for alignments to genomic (-) strand: 0=Don&rsquo;t
invert the cDNA (default) 1=Invert cDNA and print genomic
(-) strand 2=Invert cDNA and print genomic (+) strand</p>

<p style="margin-top: 1em">-i, --introngap=INT <br>
Nucleotides to show on each end of intron (default 3)</p>

<p style="margin-top: 1em">-l, --wraplength=INT <br>
Wrap length for alignment (default 50)</p>

<p style="margin-top: 1em">Filtering output options</p>

<p style="margin-top: 1em">--min-trimmed-coverage=FLOAT
<br>
Do not print alignments with trimmed coverage less this
value (default=0.0, which means no filtering) Note that
chimeric alignments will be output regardless of this
fil&acirc; <br>
ter</p>

<p style="margin-top: 1em">--min-identity=FLOAT <br>
Do not print alignments with identity less this value
(default=0.0, which means no filtering) Note that chimeric
alignments will be output regardless of this filter Help
<br>
options</p>

<p style="margin-top: 1em">--check <br>
Check compiler assumptions</p>

<p style="margin-top: 1em">--version <br>
Show version</p>

<p style="margin-top: 1em">--help Show this help
message</p>

<p style="margin-top: 1em">Other tools of GMAP suite are
located in /usr/lib/gmap</p>

<p style="margin-top: 1em">gmap 2017-01-14-1 January 2017
GMAP(1)</p>
<hr>
</body>
</html>
