<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:13:58 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMAP_BUILD(1) User Commands GMAP_BUILD(1)</p>

<p style="margin-top: 1em">NAME <br>
gmap_build - Tool for genome database creation for GMAP or
GSNAP</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmap_build [options...] -d &lt;genomename&gt;
&lt;fasta_files&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
gmap_build: Builds a gmap database for a genome to be used
by GMAP or GSNAP. Part of GMAP package, version
2017-01-14.</p>

<p style="margin-top: 1em">A simplified alternative to
using the program gmap_setup, which creates a Makefile.</p>

<p style="margin-top: 1em">OPTIONS <br>
-D, --dir=STRING <br>
Destination directory for installation (defaults to gmapdb
directory specified at configure time)</p>

<p style="margin-top: 1em">-d, --db=STRING <br>
Genome name</p>

<p style="margin-top: 1em">-n, --names=STRING <br>
Substitute names for chromosomes, provided in a file. The
file should have one line for each chromosome name to be
changed, with the original FASTA name in column 1 and <br>
the desired chromosome name in column 2. This provides an
easy way to change the names of chromosomes, for example, to
add or remove the &quot;chr&quot; prefix. Column 2 may be
<br>
blank, which indicates no name change. This file can also be
combined with --sort=names to provide a particular order for
the chromosomes in the genome index.</p>

<p style="margin-top: 1em">-M, --mdflag=STRING <br>
Use MD file from NCBI for mapping contigs to chromosomal
coordinates</p>

<p style="margin-top: 1em">-C, --contigs-are-mapped <br>
Find a chromosomal region in each FASTA header line. Useful
for contigs that have been mapped to chromosomal
coordinates. Ignored if the --mdflag is provided.</p>

<p style="margin-top: 1em">-z, --compression=STRING <br>
Use given compression types (separated by commas; default is
bitpack64) bitpack64 - optimized for modern computers with
SIMD instructions (recommended) all - create all <br>
available compression types, currently bitpack64 none - do
not compress offset files (I believe this is no longer
supported)</p>

<p style="margin-top: 1em">-k, --kmer=INT <br>
k-mer value for genomic index (allowed: 15 or less, default
is 15)</p>

<p style="margin-top: 1em">-q INT sampling interval for
genomoe (allowed: 1-3, default 3)</p>

<p style="margin-top: 1em">-s, --sort=STRING <br>
Sort chromosomes using given method: none - use chromosomes
as found in FASTA file(s) alpha - sort chromosomes
alphabetically (chr10 before chr 1) numeric-alpha - chr1,
<br>
chr1U, chr2, chrM, chrU, chrX, chrY chrom - chr1, chr2,
chrM, chrX, chrY, chr1U, chrU names - sort chromosomes based
on file provided to --names flag</p>

<p style="margin-top: 1em">-g, --gunzip <br>
Files are gzipped, so need to gunzip each file first</p>

<p style="margin-top: 1em">-E, --fasta-pipe=STRING <br>
Interpret argument as a command, instead of a list of FASTA
files</p>

<p style="margin-top: 1em">-w INT Wait (sleep) this many
seconds after each step (default 2)</p>

<p style="margin-top: 1em">-c, --circular=STRING <br>
Circular chromosomes (either a list of chromosomes separated
by a comma, or a filename containing circular chromosomes,
one per line). If you use the --names feature, <br>
then you should use the original name of the chromosome, not
the substitute name, for this option.</p>

<p style="margin-top: 1em">-2, --altscaffold=STRING <br>
File with alt scaffold info, listing alternate scaffolds,
one per line, tab-delimited, with the following fields: (1)
alt_scaf_acc, (2) parent_name, (3) orientation, (4) <br>
alt_scaf_start, (5) alt_scaf_stop, (6) parent_start, (7)
parent_end.</p>

<p style="margin-top: 1em">-e, --nmessages=INT <br>
Maximum number of messages (warnings, contig reports) to
report (default 50)</p>

<p style="margin-top: 1em">--build-sarray=INT <br>
Whether to build suffix array: 0=no, 1=yes (default)</p>

<p style="margin-top: 1em">Obsolete options: <br>
-T STRING <br>
Temporary build directory (may need to specify if you run
out of space in your current directory) This is no longer
necessary, since gmap_build now builds directly in the <br>
destination (or -D) directory.</p>

<p style="margin-top: 1em">Other tools of GMAP suite are
located in /usr/lib/gmap</p>

<p style="margin-top: 1em">gmap_build 2017-01-14-1 January
2017 GMAP_BUILD(1)</p>
<hr>
</body>
</html>
