<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:04 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMOD_BULK_LOAD_GFF3(1p) User Contributed Perl
Documentation GMOD_BULK_LOAD_GFF3(1p)</p>

<p style="margin-top: 1em">NAME <br>
$0 - Bulk loads gff3 files into a chado database.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
% $0 [options] <br>
% cat &lt;gff-file&gt; | $0 [options]</p>

<p style="margin-top: 1em">OPTIONS <br>
--gfffile The file containing GFF3 (optional, can read <br>
from stdin) <br>
--fastafile Fasta file to load sequence from <br>
--organism The organism for the data <br>
(use the value &rsquo;fromdata&rsquo; to read from GFF
organism=xxx) <br>
--dbprofile Database config profile name <br>
--dbname Database name <br>
--dbuser Database user name <br>
--dbpass Database password <br>
--dbhost Database host <br>
--dbport Database port <br>
--analysis The GFF data is from computational analysis <br>
--noload Create bulk load files, but don&rsquo;t actually
load them. <br>
--nosequence Don&rsquo;t load sequence even if it is in the
file <br>
--notransact Don&rsquo;t use a single transaction to load
the database <br>
--drop_indexes Drop indexes of affected tables before
starting load <br>
and recreate after load is finished; generally <br>
does not help performance. <br>
--validate Validate SOFA terms before attempting insert (can
<br>
cause script startup to be slow, off by default) <br>
--ontology Give directions for handling misc Ontology_terms
<br>
--skip_vacuum Skip vacuuming the tables after the inserts
(default) <br>
--no_skip_vaccum Don&rsquo;t skip vacuuming the tables <br>
--inserts Print INSERT statements instead of COPY FROM STDIN
<br>
--noexon Don&rsquo;t convert CDS features to exons (but
still create <br>
polypeptide features) <br>
--recreate_cache Causes the uniquename cache to be recreated
<br>
--remove_lock Remove the lock to allow a new process to run
<br>
--save_tmpfiles Save the temp files used for loading the
database <br>
--random_tmp_dir Use a randomly generated tmp dir (the
default is <br>
to use the current directory) <br>
--no_target_syn By default, the loader adds the targetId in
<br>
the synonyms list of the feature. This flag <br>
deactivate this. <br>
--unique_target Trust the unicity of the target IDs. IDs are
case <br>
sensitive. By default, the uniquename of a new target <br>
will be &rsquo;TargetId_PrimaryKey&rsquo;. With this flag,
<br>
it will be &rsquo;TargetId&rsquo;. Furthermore, the Name of
the <br>
created target will be its TargetId, instead of the <br>
feature&rsquo;s Name. <br>
--dbxref Use either the first Dbxref annotation as the <br>
primary dbxref (that goes into feature.dbxref_id), <br>
or if an optional argument is supplied, the first <br>
dbxref that has a database part (ie, before the
&rsquo;:&rsquo;) <br>
that matches the supplied pattern is used. <br>
--delete Instead of inserting features into the database,
<br>
use the GFF lines to delete features as though <br>
the CRUD=delete-all option were set on all lines <br>
(see &rsquo;Deletes and updates via GFF below&rsquo;). The
<br>
loader will ask for confirmation before continuing. <br>
--delete_i_really_mean_it <br>
Works like --delete except that it does not ask <br>
for confirmation. <br>
--fp_cv Name of the feature property controlled vocabulary
<br>
(defaults to &rsquo;feature_property&rsquo;). <br>
--noaddfpcv By default, the loader adds GFF attribute types
as <br>
new feature_property cv terms when missing. This flag <br>
deactivates it. <br>
** dgg note: should rename this flag: --[no]autoupdate <br>
for Chado tables cvterm, cv, db, organism, analysis ...</p>

<p style="margin-top: 1em">--manual Detailed manual pages
<br>
--custom_adapter Use a custom subclass adaptor for
Bio::GMOD::DB::Adapter <br>
Provide the path to the adapter as an argument <br>
--private_schema Load the data into a non-public schema.
<br>
--use_public_cv When loading into a non-public schema, load
any cv and <br>
cvterm data into the public schema <br>
--end_sql SQL code to execute after the data load is
complete <br>
--allow_external_parent <br>
Allow Parent tags to refer to IDs outside the current <br>
GFF file</p>

<p style="margin-top: 1em">Note that all of the arguments
that begin &rsquo;db&rsquo; as well as organism can be
provided by default by Bio::GMOD::Config, which was
installed when &rsquo;make install&rsquo; was run. Also note
<br>
the the option dbprofile and all other db* options are
mutually exclusive--if you supply dbprofile, do not supply
any other db* options, as they will not be used.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The GFF in the datafile must be version 3 due to its tighter
control of the specification and use of controlled
vocabulary. Accordingly, the names of feature types must be
<br>
exactly those in the Sequence Ontology Feature Annotation
(SOFA), not the synonyms and not the accession numbers (SO
accession numbers may be supported in future versions of
this <br>
script).</p>

<p style="margin-top: 1em">Note that the ##sequence-region
directive is not supported as a way of declaring a reference
sequence for a GFF3 file. The ##sequence-region directive is
not expressive enough <br>
to define what type of thing the sequence is (ie, is it a
chromosome, a contig, an arm, etc?). If your GFF file uses a
##sequence-region directive in this way, you must convert
<br>
it to a full GFF3 line. For example, if you have this
line:</p>

<p style="margin-top: 1em">##sequence-region chrI 1
9999999</p>

<p style="margin-top: 1em">Then is should be converted to a
GFF3 line like this:</p>

<p style="margin-top: 1em">chrI . chromosome 1 9999999 . .
. ID=chrI</p>

<p style="margin-top: 1em">How GFF3 is stored in chado <br>
Here is summary of how GFF3 data is stored in chado:</p>

<p style="margin-top: 1em">Column 1 (reference sequence)
<br>
The reference sequence for the feature becomes the
srcfeature_id of the feature in the featureloc table for
that feature. That featureloc generally assigned a rank of
zero <br>
if there are other locations associated with this feature
(for instance, for a match feature), the other locations
will be assigned featureloc.rank values greater than
zero.</p>

<p style="margin-top: 1em">Column 2 (source) <br>
The source is stored as a dbxref. The chado instance must of
an entry in the db table named &rsquo;GFF_source&rsquo;. The
script will then create a dbxref entry for the
feature&rsquo;s <br>
source and associate it to the feature via the
feature_dbxref table.</p>

<p style="margin-top: 1em">Column 3 (type) <br>
The cvterm.cvterm_id of the SOFA type is stored in
feature.type_id.</p>

<p style="margin-top: 1em">Column 4 (start) <br>
The value of start minus 1 is stored in featureloc.fmin (one
is subtracted because chado uses interbase coordinates,
whereas GFF uses base coordinates).</p>

<p style="margin-top: 1em">Column 5 (end) <br>
The value of end is stored in featureloc.fmax.</p>

<p style="margin-top: 1em">Column 6 (score) <br>
The score is stored in one of the score columns in the
analysisfeature table. The default is
analysisfeature.significance. See the section below on
analysis results for <br>
more information.</p>

<p style="margin-top: 1em">Column 7 (strand) <br>
The strand is stored in featureloc.strand.</p>

<p style="margin-top: 1em">Column 8 (phase) <br>
The phase is stored in featureloc.phase. Note that there is
currently a problem with the chado schema for the case of
single exons having different phases in different <br>
transcripts. If your data has just such a case, complain to
gmod-schema@lists.sourceforge.net to find ways to address
this problem.</p>

<p style="margin-top: 1em">Column 9 (group) <br>
Here is where the magic happens.</p>

<p style="margin-top: 1em">Assigning feature.name,
feature.uniquename <br>
The values of feature.name and feature.uniquename are
assigned according to these simple rules:</p>

<p style="margin-top: 1em">If there is an ID tag, that is
used as feature.uniquename <br>
otherwise, it is assigned a uniquename that is equal to
&rsquo;auto&rsquo; concatenated with the feature_id.</p>

<p style="margin-top: 1em">If there is a Name tag,
it&rsquo;s value is set to feature.name; <br>
otherwise it is null.</p>

<p style="margin-top: 1em">Note that these rules are much
more simple than that those that Bio::DB::GFF uses, and may
need to be revisited.</p>

<p style="margin-top: 1em">Assigning feature_relationship
entries <br>
All Parent tagged features are assigned feature_relationship
entries of &rsquo;part_of&rsquo; to their parent features.
Derived_from tags are assigned &rsquo;derived_from&rsquo;
relationships. <br>
Note that parent features must appear in the file before any
features use a Parent or Derived_from tags referring to that
feature.</p>

<p style="margin-top: 1em">Alias tags <br>
Alias values are stored in the synonym table, under both
synonym.name and synonym.synonym_sgml, and are linked to the
feature via the feature_synonym table.</p>

<p style="margin-top: 1em">Dbxref tags <br>
Dbxref values must be of the form
&rsquo;db_name:accession&rsquo;, where db_name must have an
entry in the db table, with a value of db.name equal to
&rsquo;DB:db_name&rsquo;; several database <br>
names were preinstalled with the database when &rsquo;make
prepdb&rsquo; was run. Execute &rsquo;SELECT name FROM
db&rsquo; to find out what databases are already available.
New dbxref entries <br>
are created in the dbxref table, and dbxrefs are linked to
features via the feature_dbxref table.</p>

<p style="margin-top: 1em">Gap tags <br>
Currently is mostly ignored--the value is stored as a
featureprop, but otherwise is not used yet.</p>

<p style="margin-top: 1em">Note tags <br>
The values are stored as featureprop entries for the
feature.</p>

<p style="margin-top: 1em">Any custom (ie, lowercase-first)
tags <br>
Custom tags are supported. If the tag doesn&rsquo;t already
exist in the cvterm table, it will be created. The value
will stored with its associated cvterm in the featureprop
<br>
table.</p>

<p style="margin-top: 1em">Ontology_term <br>
When the Ontology_term tags are used, items from the Gene
Ontology and Sequence Ontology will be processed
automatically when the standard DB:accession format is used
<br>
(e.g. GO:0001234). To use other ontology terms, you must
specify that mapping of the DB indentifiers in the GFF file
and the name of the ontologies in the cv table as a <br>
comma separated tag=value pairs. For example, to use plant
and cell ontology terms, you would supply on the command
line:</p>

<p style="margin-top: 1em">--ontology &rsquo;PO=plant
ontology,CL=cell ontology&rsquo;</p>

<p style="margin-top: 1em">where &rsquo;plant
ontology&rsquo; and &rsquo;cell ontology&rsquo; are the
names in the cv table exactly as they appear.</p>

<p style="margin-top: 1em">Target tags <br>
Proper processing of Target tags requires that there be two
source features already available in the database, the
&rsquo;primary&rsquo; source feature (the chromosome or
contig) and <br>
the &rsquo;subject&rsquo; from the similarity analysis, like
an EST, cDNA or syntenic chromosome. If the subject feature
is not present, the loader will attempt to create a <br>
placeholder feature object in its place. If you have a fasta
file the contains the subject, you can use the perl script,
gmod_fasta2gff3.pl, that comes with this <br>
distribution to make a GFF3 file suitable for loading into
chado before loading your analysis results.</p>

<p style="margin-top: 1em">CDS and UTR features <br>
The way CDS features are represented in Chado is as an
intersection of a transcript&rsquo;s exons and the
transcripts polypeptide feature. To allow proper translation
of a <br>
GFF3 file&rsquo;s CDS features, this loader will convert CDS
and UTR feature lines to corresponding exon features (and
add a featureprop note that the exon was inferred from a
<br>
GFF3 CDS and/or UTR line), and create a polypeptide feature
that spans the genomic region from the start of translation
to the stop.</p>

<p style="margin-top: 1em">If your GFF3 file contains both
exon and CDS/UTR features, then you will want to suppress
the creation of the exon features and instead will only want
a polypeptide <br>
feature to be created. To do this, use the --noexon option.
In this case, the CDS and UTR features will still be
converted to exon features as described above.</p>

<p style="margin-top: 1em">Note that in the case where your
GFF file contains CDS and/or UTR features that do not belong
to &rsquo;central dogma&rsquo; genes (that is, that have a
gene, transcript and CDS/exon <br>
features), none of the above will happen and the features
will be stored as is.</p>

<p style="margin-top: 1em">NOTES <br>
Loading fasta file <br>
When the --fastafile is provided with an argument that is
the path to a file containing fasta sequence, the loader
will attempt to update the feature table with the sequence
<br>
provided. Note that the ID provided in the fasta description
line must exactly match what is in the feature table
uniquename field. Be careful if it is possible that the <br>
uniquename of the feature was changed to ensure uniqueness
when it was loaded from the original GFF. Also note that
when loading sequence from a fasta file, loading GFF from
<br>
standard in is disabled. Sorry for any inconvenience.</p>

<p style="margin-top: 1em">##sequence-region <br>
This script does not use sequence-region directives for
anything. If it represents a feature that needs to be
inserted into the database, it should be represented with a
<br>
full GFF line. This includes the reference sequence for the
features if it is not already in the database, like a
chromosome. For example, this:</p>

<p style="margin-top: 1em">##sequence-region chr1 1
213456789</p>

<p style="margin-top: 1em">should change to this:</p>

<p style="margin-top: 1em">chr1 UCSC chromosome 1 213456789
. . . ID=chr1</p>

<p style="margin-top: 1em">Transactions <br>
This application will, by default, try to load all of the
data at once as a single transcation. This is safer from the
database&rsquo;s point of view, since if anything bad <br>
happens during the load, the transaction will be rolled back
and the database will be untouched. The problem occurs if
there are many (say, greater than a 2-300,000) rows in <br>
the GFF file. When that is the case, doing the load as a
single transcation can result in the machine running out of
memory and killing processes. If --notranscat is <br>
provided on the commandline, each table will be loaded as a
separate transaction.</p>

<p style="margin-top: 1em">SQL INSERTs versus COPY FROM
<br>
This bulk loader was originally designed to use the
PostgreSQL COPY FROM syntax for bulk loading of data.
However, as mentioned in the &rsquo;Transactions&rsquo;
section, memory issues <br>
can sometimes interfere with such bulk loads. In another
effort to circumvent this issue, the bulk loader has been
modified to optionally create INSERT statements instead of
<br>
the COPY FROM statements. INSERT statements will load much
more slowly than COPY FROM statements, but as they load and
commit individually, they are more more likely to <br>
complete successfully. As an indication of the speed
differences involved, loading yeast GFF3 annotations (about
16K rows), it takes about 5 times longer using INSERTs <br>
versus COPY on my laptop.</p>

<p style="margin-top: 1em">Deletes and updates via GFF <br>
There is rudimentary support for modifying the features in
an existing database via GFF. Currently, there is only
support for deleting. In order to delete, the GFF line <br>
must have a custom tag in the ninth column,
&rsquo;CRUD&rsquo; (for Create, Replace, Update and Delete)
and have a recognized value. Currently the two recognized
values are CRUD=delete <br>
and CRUD=delete-all.</p>

<p style="margin-top: 1em">IMPORTANT NOTE: Using the delete
operations have the potential of creating orphan features
(eg, exons whose gene has been deleted). You should be
careful to make sure that <br>
doesn&rsquo;t happen. Included in this distribution is a
PostgreSQL trigger (written in plpgsql) that will delete all
orphan children recursively, so if a gene is deleted, all
<br>
transcripts, exons and polypeptides that belong to that gene
will be deleted too. See the file
modules/sequence/functions/delete-trigger.plpgsql for more
information.</p>

<p style="margin-top: 1em">delete <br>
The delete option will delete one and only one feature for
which the name, type and organism match what is in the GFF
line with what is in the database. Note that <br>
feature.uniquename are not considered, nor are the
coordinates presented in the GFF file. This is so that
updates via GFF can be done on the coordinants. If there is
<br>
more than one feature for which the name, type and organism
match, the loader will print an error message and stop. If
there are no features that match the name, type <br>
and organism, the loader will print a warning message and
continue.</p>

<p style="margin-top: 1em">delete-all <br>
The delete-all option works similarly to the delete option,
except that it will delete all features that match the name,
type and organism in the GFF line (as opposed to <br>
allowing only one feature to be deleted). If there are no
features that match, the loader will print a warning message
and continue.</p>

<p style="margin-top: 1em">The run lock <br>
The bulk loader is not a multiuser application. If two
separate bulk load processes try to load data into the
database at the same time, at least one and possibly all
loads <br>
will fail. To keep this from happening, the bulk loader
places a lock in the database to prevent other
gmod_bulk_load_gff3.pl processes from running at the same
time. When <br>
the application exits normally, this lock will be removed,
but if it crashes for some reason, the lock will not be
removed. To remove the lock from the command line, provide
<br>
the flag --remove_lock. Note that if the loader crashed
necessitating the removal of the lock, you also may need to
rebuild the uniquename cache (see the next section).</p>

<p style="margin-top: 1em">The uniquename cache <br>
The loader uses the chado database to create a table that
caches feature_ids, uniquenames, type_ids, and organism_ids
of the features that exist in the database at the time <br>
the load starts and the features that will be added when the
load is complete. If it is possilbe that new features have
been added via some method that is not this loader <br>
(eg, Apollo edits or loads with XORT) or if a previous load
using this loader was aborted, then you should supply the
--recreate_cache option to make sure the cache is fresh.</p>

<p style="margin-top: 1em">Sequence <br>
By default, if there is sequence in the GFF file, it will be
loaded into the residues column in the feature table row
that corresponds to that feature. By supplying the <br>
--nosequence option, the sequence will be skipped. You might
want to do this if you have very large sequences, which can
be difficult to load. In this context, &quot;very
large&quot; <br>
means more than 200MB.</p>

<p style="margin-top: 1em">Also note that for sequences to
load properly, the GFF file must have the ##FASTA directive
(it is required for proper parsing by Bio::FeatureIO), and
the ID of the feature <br>
must be exactly the same as the name of the sequence
following the &gt; in the fasta section.</p>

<p style="margin-top: 1em">The ORGANISM table <br>
This script assumes that the organism table is populated
with information about your organism. If you are unsure if
that is the case, you can execute this command from the <br>
psql command-line:</p>

<p style="margin-top: 1em">select * from organism;</p>

<p style="margin-top: 1em">If you do not see your organism
listed, execute this command to insert it:</p>

<p style="margin-top: 1em">insert into organism
(abbreviation, genus, species, common_name) <br>
values (&rsquo;H.sapiens&rsquo;,
&rsquo;Homo&rsquo;,&rsquo;sapiens&rsquo;,&rsquo;Human&rsquo;);</p>

<p style="margin-top: 1em">substituting in the appropriate
values for your organism.</p>

<p style="margin-top: 1em">Parents/children order <br>
Parents must come before children in the GFF file.</p>

<p style="margin-top: 1em">Analysis <br>
If you are loading analysis results (ie, blat results, gene
predictions), you should specify the -a flag. If no
arguments are supplied with the -a, then the loader will
<br>
assume that the results belong to an analysis set with a
name that is the concatenation of the source (column 2) and
the method (column 3) with an underscore in between. <br>
Otherwise, the argument provided with -a will be taken as
the name of the analysis set. Either way, the analysis set
must already be in the analysis table. The easist way <br>
to do this is to insert it directly in the psql shell:</p>

<p style="margin-top: 1em">INSERT INTO analysis (name,
program, programversion) <br>
VALUES (&rsquo;genscan
2005-2-28&rsquo;,&rsquo;genscan&rsquo;,&rsquo;5.4&rsquo;);</p>

<p style="margin-top: 1em">There are other columns in the
analysis table that are optional; see the schema
documentation and &rsquo; analysis&rsquo; in psql for more
information.</p>

<p style="margin-top: 1em">Chado has four possible columns
for storing the score in the GFF score column; please use
whichever is most appropriate and identifiy it with
--score_col flag (significance <br>
is the default). Note that the name of the column can be
shortened to one letter. If you have more than one score
associated with each feature, you can put the other scores
<br>
in the ninth column as a tag=value pair, like
&rsquo;identity=99&rsquo;, and the bulk loader will put it
in the featureprop table (provided there is a cvterm for
identity; see the <br>
section above concerning custom tags). Available options
are:</p>

<p style="margin-top: 1em">significance (default) <br>
identity <br>
normscore <br>
rawscore</p>

<p style="margin-top: 1em">A planned addtion to the
functionality of handling analysis results is to allow
&quot;mixed&quot; GFF files, where some lines are analysis
results and some are not. Additionally, one <br>
will be able to supply lists of types (optionally with
sources) and their associated entry in the analysis table.
The format will probably be tag value pairs:</p>

<p style="margin-top: 1em">--analysis
match:Rice_est=rice_est_blast,
match:Maize_cDNA=maize_cdna_blast,
mRNA=genscan_prediction,exon=genscan_prediction</p>

<p style="margin-top: 1em">Grouping features by ID <br>
The GFF3 specification allows features like CDSes and
match_parts to be grouped together by sharing the same ID.
This loader does not support this method of grouping. <br>
Instead the parent feature must be explicitly created before
the parts and the parts must refer to the parent with the
Parent tag.</p>

<p style="margin-top: 1em">External Parent IDs <br>
The GFF3 specification states that IDs are only valid within
a single GFF file, so you can&rsquo;t have Parent tags that
refer to IDs in another file. By specificifying the <br>
&quot;allow_external_parent&quot; flag, you can relax this
restriction. A word of warning however: if the parent
feature&rsquo;s uniquename/ID was modified during loading
(to make it <br>
unique), this functionality won&rsquo;t work, as it
won&rsquo;t beable to find the original feature correctly.
Actually, it may be worse than not working, it may attach
child features <br>
to the wrong parent. This is why it is a bad idea to use
this functionality! Please use with caution.</p>

<p style="margin-top: 1em">AUTHORS <br>
Allen Day &lt;allenday@ucla.edu&gt;, Scott Cain
&lt;scain@cpan.org&gt;</p>

<p style="margin-top: 1em">Copyright (c) 2011</p>

<p style="margin-top: 1em">This library is free software;
you can redistribute it and/or modify it under the same
terms as Perl itself.</p>

<p style="margin-top: 1em">perl v5.24.1 2016-12-17
GMOD_BULK_LOAD_GFF3(1p)</p>
<hr>
</body>
</html>
