<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GMOD_GFF3_PREPROCESSOR(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GMOD_GFF3_PREPROCESSOR(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GMOD_GFF3_PREPROCESSOR(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
$0 - Prepares a GFF3 file for bulk loading into a chado database.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  % gmod_gff_preprocessor [options] --gfffile &lt;filename&gt;
</pre>
<h1 class="Sh" title="Sh" id="COMMAND-LINE_OPTIONS"><a class="selflink" href="#COMMAND-LINE_OPTIONS">COMMAND-LINE
  OPTIONS</a></h1>
<pre>
 --gfffile        The file containing GFF3 (optional, can read
                     from stdin)
 --outfile        The name kernel that will be used for naming result files
 --splitfile      Split the files into more manageable chunks, providing
                     an argument to control splitting
 --onlysplit      Split the files and then quit (ie, don't sort)
 --nosplit        Don't split the files (ie, only sort)
 --hasrefseq      Set this if the file contains a reference sequence line
                     (Only needed if not splitting files)
 --dbprofile      Specify a gmod.conf profile name (otherwise use default)
 --inheritance_tiers How many levels of inheritance do you expect tis file
                     to have (default: 3)
</pre>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
splitfile -- Just setting this flag to 1 will cause the file to be split by
  reference sequence. If you provide an optional argument, it will be further
  split according to these rules:
<div class="Pp"></div>
<pre>
 source=1     Splits files according to the value in the source column
 source=a,b,c Puts lines with sources that match (via regular expression)
                     'a', 'b', or 'c' in a separate file
 type=a,b,c   Puts lines with types that match 'a', 'b', or 'c' in a
                     separate file
</pre>
<div class="Pp"></div>
For example, if you wanted all of your analysis results to go in a separate
  file, you could indicate '--splitfile type=match', and all cDNA_match,
  EST_match and cross_genome_match features would go into separate files
  (separate by reference sequence).
<div class="Pp"></div>
inheritence_tiers -- The number of levels of inheritance this file has. For
  example, if the file has &quot;central dogma&quot; genes in it (gene/mRNA/
  exon,polypeptide), then it has 3. Up to 4 is supported but the higher the
  number, the more slowly it performs. If you don't know, 3 is a reasonable
  guess.
<h2 class="Ss" title="Ss" id="FASTA_sequence"><a class="selflink" href="#FASTA_sequence">FASTA
  sequence</a></h2>
If the GFF3 file contains FASTA sequence at the end, the sequence will be placed
  in a separate file with the extension '.fasta'. This fasta file can be loaded
  separately after the split and/or sorted GFF3 files are loaded, using the
  command:
<div class="Pp"></div>
<pre>
  gmod_bulk_load_gff3.pl -g &lt;fasta file name&gt;
</pre>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Scott Cain &lt;cain@cshl.org&gt;
<div class="Pp"></div>
Copyright (c) 2006-2007
<div class="Pp"></div>
This library is free software; you can redistribute it and/or modify it under
  the same terms as Perl itself.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-12-17</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
