<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GMT-MUSIC-BMR-CALC-BMR(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GMT-MUSIC-BMR-CALC-BMR(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GMT-MUSIC-BMR-CALC-BMR(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="gmt_music_bmr_calc-bmr"><a class="selflink" href="#gmt_music_bmr_calc-bmr">gmt
  music bmr calc-bmr</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gmt music bmr calc-bmr - Calculates mutation rates given per-gene coverage (from
  &quot;music bmr calc-covg&quot;), and a mutation list
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes gmt music bmr calc-bmr version 0.04 (2016-01-01 at
  23:10:19)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
gmt music bmr calc-bmr --bmr-output=? --roi-file=? --gene-mr-file=?
  --reference-sequence=? --bam-list=? --output-dir=? --maf-file=?
  [--skip-non-coding] [--skip-silent] [--bmr-groups=?] [--show-skipped]
  [--separate-truncations] [--merge-concurrent-muts] [--genes-to-ignore=?]
<div class="Pp"></div>
<pre>
 ... music bmr calc-bmr \
    --bam-list input_dir/bam_list \
    --maf-file input_dir/myMAF.tsv \
    --output-dir output_dir/ \
    --reference-sequence input_dir/all_sequences.fa \
    --roi-file input_dir/all_coding_exons.tsv
 ... music bmr calc-bmr \
    --bam-list input_dir/bam_list \
    --maf-file input_dir/myMAF.tsv \
    --output-dir output_dir/ \
    --reference-sequence input_dir/all_sequences.fa \
    --roi-file input_dir/all_coding_exons.tsv \
    --genes-to-ignore GENE1,GENE2
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">bmr-output <i>Number</i></dt>
  <dd class="It-tag">TODO</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">roi-file <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of ROIs [chr start stop gene_name] (See
      DESCRIPTION)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">gene-mr-file <i>Text</i></dt>
  <dd class="It-tag">TODO</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reference-sequence <i>Text</i></dt>
  <dd class="It-tag">Path to reference sequence in FASTA format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bam-list <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of BAM files [sample_name normal_bam
      tumor_bam] (See DESCRIPTION)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-dir <i>Text</i></dt>
  <dd class="It-tag">Directory where output files will be written (Use the same
      one used with calc-covg)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">maf-file <i>Text</i></dt>
  <dd class="It-tag">List of mutations using TCGA MAF specification v2.3</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">skip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Skip non-coding mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-non-coding 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Skip silent mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-silent 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bmr-groups <i>Integer</i></dt>
  <dd class="It-tag">Number of clusters of samples with comparable BMRs (See
      DESCRIPTION)
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '1' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">show-skipped <i>Boolean</i></dt>
  <dd class="It-tag">Report each skipped mutation, not just how many
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--noshow-skipped) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noshow-skipped <i>Boolean</i></dt>
  <dd class="It-tag">Make show-skipped 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">separate-truncations <i>Boolean</i></dt>
  <dd class="It-tag">Group truncational mutations as a separate category
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--noseparate-truncations) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noseparate-truncations <i>Boolean</i></dt>
  <dd class="It-tag">Make separate-truncations 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">merge-concurrent-muts <i>Boolean</i></dt>
  <dd class="It-tag">Multiple mutations of a gene in the same sample are treated
      as 1
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--nomerge-concurrent-muts) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">nomerge-concurrent-muts <i>Boolean</i></dt>
  <dd class="It-tag">Make merge-concurrent-muts 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">genes-to-ignore <i>Text</i></dt>
  <dd class="It-tag">Comma-delimited list of genes to ignore for background
      mutation rates</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Given a mutation list (MAF), and per-gene coverage data calculated using
  &quot;music bmr calc-covg&quot;), this script calculates overall Background
  Mutation Rate (BMR) and BMRs in the categories of AT/CG/CpG Transitions,
  AT/CG/CpG Transversions, and Indels. An optional category for truncational
  mutations can also be specified. The script generates a file with per-gene
  mutation rates that can be used with the tool that tests for significantly
  mutated genes (music smg).
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--roi-file</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The regions of interest (ROIs) of each gene are typically
    regions targeted for sequencing or are merged exon loci (from multiple
    transcripts) of genes with 2-bp flanks (splice junctions). ROIs from the
    same chromosome must be listed adjacent to each other in this file. This
    allows the underlying C-based code to run much more efficiently and avoid
    re-counting bases seen in overlapping ROIs (for overall covered base
    counts). For per-gene base counts, an overlapping base will be counted each
    time it appears in an ROI of the same gene. To avoid this, be sure to merge
    together overlapping ROIs of the same gene. BEDtools' mergeBed can help if
    used per gene.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--reference-sequence</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The reference sequence in FASTA format. If a reference
    sequence index is not found next to this file (a .fai file), it will be
    created.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--bam-list</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Provide a file containing sample names and normal/tumor BAM
    locations for each. Use the tab- delimited format [sample_name normal_bam
    tumor_bam] per line. Additional columns like clinical data are allowed, but
    ignored. The sample_name must be the same as the tumor sample names used in
    the MAF file (16th column, with the header Tumor_Sample_Barcode).</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--bmr-groups</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Ideally, we want to test the mutation rate (MR) of a gene
    in a sample against the background mutation rate (BMR) across that sample.
    But if the BMRs of some samples are comparable, we can instead test the MR
    of a gene across a group of samples with comparable BMR, against the overall
    BMR of that group. This argument specifies how many such groups you want to
    cluster samples into. By default, it is assumed that all samples have
    comparable BMRs (bmr-groups = 1).</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--output-dir</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">This should be the same output directory used when running
    &quot;music bmr calc-covg&quot;. The following outputs of this script will
    also be created/written: overall_bmrs: File containing categorized overall
    background mutation rates. gene_mrs: File containing categorized per-gene
    mutation rates.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--genes-to-ignore</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">A comma-delimited list of genes to ignore for overall BMR
    calculations. List genes that are known factors in this disease and whose
    mutations should not be classified as background.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2010-2011 Washington University in St. Louis.
<div class="Pp"></div>
It is released under the Lesser GNU Public License (LGPL) version 3. See the
  associated LICENSE file in this distribution.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Cyriac Kandoth, Ph.D.
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome-music-bmr</b>(1), <b>genome-music</b>(1), <b>genome</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
