<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:11 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMT-MUSIC-GALAXY(1p) User Contributed Perl Documentation
GMT-MUSIC-GALAXY(1p)</p>

<p style="margin-top: 1em">gmt music galaxy NAME <br>
gmt music galaxy - Run the full suite of MuSiC tools
sequentially.</p>

<p style="margin-top: 1em">VERSION <br>
This document describes gmt music galaxy version 0.04
(2016-01-01 at 23:10:19)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmt music galaxy [--output-dir=?] --bam-list=?
--output-bundle=? --roi-file=? --reference-sequence=?
--maf-file=? --pathway-file=?
[--numeric-clinical-data-file=?] <br>
[--categorical-clinical-data-file=?]
[--mutation-matrix-file=?] [--permutations=?]
[--normal-min-depth=?] [--tumor-min-depth=?] [--min-mapq=?]
[--show-skipped] <br>
[--genes-to-ignore=?] [--bmr=?] [--max-proximity=?]
[--bmr-modifier-file=?] [--numerical-data-test-method=?]
[--skip-low-mr-genes] [--max-fdr=?] [--genetic-data-type=?]
<br>
[--wu-annotation-headers] [--bmr-groups=?]
[--separate-truncations] [--merge-concurrent-muts]
[--skip-non-coding] [--skip-silent]
[--min-mut-genes-per-path=?] <br>
[--glm-model-file=?] [--processors=?] [--aa-range=?]
[--nuc-range=?] [--reference-build=?] [--show-known-hits]
[--glm-clinical-data-file=?] [--use-maf-in-glm]
[--omimaa-dir=?] <br>
[--cosmic-dir=?] [--verbose]
[--clinical-correlation-matrix-file=?]</p>

<p style="margin-top: 1em">This tool takes as parameters
all the information required to run the individual tools. An
example usage is:</p>

<p style="margin-top: 1em">... music play --bam-list
input/bams_to_analyze.txt --numeric-clinical-data-file
input/numeric_clinical_data.csv --maf-file input/myMAF.tsv
--output-dir play_output_dir --pathway-file input/pathway_db
--reference-sequence input/refseq/all_sequences.fa
--roi-file input/all_coding_regions.bed --genetic-data-type
gene</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
bam-list Text <br>
Tab delimited list of BAM files [sample_name normal_bam
tumor_bam]</p>

<p style="margin-top: 1em">output-bundle Text <br>
Location where Galaxy would like the bundle of Music outputs
to be saved</p>

<p style="margin-top: 1em">roi-file Text <br>
Tab delimited list of ROIs [chr start stop gene_name]</p>

<p style="margin-top: 1em">reference-sequence Text <br>
Path to reference sequence in FASTA format</p>

<p style="margin-top: 1em">maf-file Text <br>
List of mutations using TCGA MAF specifications v2.3</p>

<p style="margin-top: 1em">pathway-file Text <br>
Tab-delimited file of pathway information</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
output-dir <br>
Directory where output files and subdirectories will be
written</p>

<p style="margin-top: 1em">Default value &rsquo;&rsquo; if
not specified</p>

<p style="margin-top: 1em">numeric-clinical-data-file Text
<br>
Table of samples (y) vs. numeric clinical data category
(x)</p>

<p style="margin-top: 1em">categorical-clinical-data-file
Text <br>
Table of samples (y) vs. categorical clinical data category
(x)</p>

<p style="margin-top: 1em">mutation-matrix-file Text <br>
Optionally store the sample-vs-gene matrix used during
calculations.</p>

<p style="margin-top: 1em">permutations Number <br>
Number of permutations used to determine P-values</p>

<p style="margin-top: 1em">normal-min-depth Integer <br>
The minimum read depth to consider a Normal BAM base as
covered</p>

<p style="margin-top: 1em">tumor-min-depth Integer <br>
The minimum read depth to consider a Tumor BAM base as
covered</p>

<p style="margin-top: 1em">min-mapq Integer <br>
The minimum mapping quality of reads to consider towards
read depth counts</p>

<p style="margin-top: 1em">show-skipped Boolean <br>
Report each skipped mutation, not just how many</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--noshow-skipped) if not specified</p>

<p style="margin-top: 1em">noshow-skipped Boolean <br>
Make show-skipped &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">genes-to-ignore Text <br>
Comma-delimited list of genes to ignore for background
mutation rates</p>

<p style="margin-top: 1em">bmr Number <br>
Background mutation rate in the targeted regions</p>

<p style="margin-top: 1em">max-proximity Text <br>
Maximum AA distance between 2 mutations</p>

<p style="margin-top: 1em">bmr-modifier-file Text <br>
Tab delimited list of values per gene that modify BMR before
testing [gene_name bmr_modifier]</p>

<p style="margin-top: 1em">numerical-data-test-method Text
<br>
Either &rsquo;cor&rsquo; for Pearson Correlation or
&rsquo;wilcox&rsquo; for the Wilcoxon Rank-Sum Test for
numerical clinical data.</p>

<p style="margin-top: 1em">Default value &rsquo;cor&rsquo;
if not specified</p>

<p style="margin-top: 1em">skip-low-mr-genes Boolean <br>
Skip testing genes with MRs lower than the background MR</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-low-mr-genes Boolean <br>
Make skip-low-mr-genes &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">max-fdr Number <br>
The maximum allowed false discovery rate for a gene to be
considered an SMG</p>

<p style="margin-top: 1em">Default value &rsquo;0.2&rsquo;
if not specified</p>

<p style="margin-top: 1em">genetic-data-type Text <br>
Data in matrix file must be either &quot;gene&quot; or
&quot;variant&quot; type data</p>

<p style="margin-top: 1em">wu-annotation-headers Boolean
<br>
Use this to default to wustl annotation format headers</p>

<p style="margin-top: 1em">nowu-annotation-headers Boolean
<br>
Make wu-annotation-headers &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">bmr-groups Integer <br>
Number of clusters of samples with comparable BMRs</p>

<p style="margin-top: 1em">Default value &rsquo;1&rsquo; if
not specified</p>

<p style="margin-top: 1em">separate-truncations Boolean
<br>
Group truncational mutations as a separate category</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--noseparate-truncations) if not
specified</p>

<p style="margin-top: 1em">noseparate-truncations Boolean
<br>
Make separate-truncations &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">merge-concurrent-muts Boolean
<br>
Multiple mutations of a gene in the same sample are treated
as 1</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--nomerge-concurrent-muts) if not
specified</p>

<p style="margin-top: 1em">nomerge-concurrent-muts Boolean
<br>
Make merge-concurrent-muts &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">skip-non-coding Boolean <br>
Skip non-coding mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-non-coding Boolean <br>
Make skip-non-coding &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">skip-silent Boolean <br>
Skip silent mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-silent Boolean <br>
Make skip-silent &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">min-mut-genes-per-path Integer
<br>
Pathways with fewer mutated genes than this will be
ignored</p>

<p style="margin-top: 1em">Default value &rsquo;1&rsquo; if
not specified</p>

<p style="margin-top: 1em">glm-model-file Text <br>
File outlining the type of model, response variable,
covariants, etc. for the GLM analysis. (See
DESCRIPTION).</p>

<p style="margin-top: 1em">processors Integer <br>
Number of processors to use in SMG (requires
&rsquo;foreach&rsquo; and &rsquo;doMC&rsquo; R packages)</p>

<p style="margin-top: 1em">Default value &rsquo;1&rsquo; if
not specified</p>

<p style="margin-top: 1em">aa-range Integer <br>
Set how close a &rsquo;near&rsquo; match is when searching
for amino acid near hits</p>

<p style="margin-top: 1em">Default value &rsquo;2&rsquo; if
not specified</p>

<p style="margin-top: 1em">nuc-range Integer <br>
Set how close a &rsquo;near&rsquo; match is when searching
for nucleotide position near hits</p>

<p style="margin-top: 1em">Default value &rsquo;5&rsquo; if
not specified</p>

<p style="margin-top: 1em">reference-build Text <br>
Put either &quot;Build36&quot; or &quot;Build37&quot;</p>

<p style="margin-top: 1em">Default value
&rsquo;Build37&rsquo; if not specified</p>

<p style="margin-top: 1em">show-known-hits Boolean <br>
When a finding is novel, show known AA in that gene</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noshow-known-hits Boolean <br>
Make show-known-hits &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">glm-clinical-data-file Text <br>
Clinical traits, mutational profiles, other mixed clinical
data (See DESCRIPTION).</p>

<p style="margin-top: 1em">use-maf-in-glm Boolean <br>
Set this flag to use the variant matrix created from the MAF
file as variant input to GLM analysis.</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--nouse-maf-in-glm) if not
specified</p>

<p style="margin-top: 1em">nouse-maf-in-glm Boolean <br>
Make use-maf-in-glm &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">omimaa-dir Path <br>
omim amino acid mutation database folder</p>

<p style="margin-top: 1em">cosmic-dir Path <br>
cosmic amino acid mutation database folder</p>

<p style="margin-top: 1em">verbose Boolean <br>
turn on to display larger working output</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noverbose Boolean <br>
Make verbose &rsquo;false&rsquo;</p>


<p style="margin-top: 1em">clinical-correlation-matrix-file
Text <br>
Optionally store the sample-vs-gene matrix used internally
during calculations.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This command can be used to run all of the MuSiC analysis
tools on a set of data. Please see the individual tools for
further description of the parameters.</p>

<p style="margin-top: 1em">AUTHORS <br>
Thomas B. Mooney, M.S.</p>

<p style="margin-top: 1em">CREDITS <br>
Please see the credits for genome-music(1).</p>

<p style="margin-top: 1em">SEE ALSO <br>
genome-music(1), genome-music-path-scan(1),
genome-music-smg(1), genome-music-clinical-correlation(1),
genome-music-mutation-relation(1),
genome-music-cosmic-omim(1), genome- <br>
music-proximity(1), genome-music-pfam(1)</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GMT-MUSIC-GALAXY(1p)</p>
<hr>
</body>
</html>
