<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GMT-MUSIC-CLINICAL-CORRELATION(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GMT-MUSIC-CLINICAL-CORRELATION(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GMT-MUSIC-CLINICAL-CORRELATION(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="gmt_music_clinical-correlation"><a class="selflink" href="#gmt_music_clinical-correlation">gmt
  music clinical-correlation</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gmt music clinical-correlation - Correlate phenotypic traits against mutated
  genes, or against individual variants
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes gmt music clinical-correlation version 0.04 (2016-01-01
  at 23:10:19)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
gmt music clinical-correlation --bam-list=? --output-file=? [--maf-file=?]
  [--glm-clinical-data-file=?] [--use-maf-in-glm] [--skip-non-coding]
  [--skip-silent] [--clinical-correlation-matrix-file=?]
  [--input-clinical-correlation-matrix-file=?] [--genetic-data-type=?]
  [--numeric-clinical-data-file=?] [--numerical-data-test-method=?]
  [--categorical-clinical-data-file=?] [--glm-model-file=?]
<div class="Pp"></div>
<pre>
 ... music clinical-correlation \
        --bam-list /path/myBamList.tsv \
        --maf-file /path/myMAF.tsv \
        --numeric-clinical-data-file /path/myNumericData.tsv \
        --genetic-data-type 'gene' \
        --output-file /path/output_file
 ... music clinical-correlation \
        --maf-file /path/myMAF.tsv \
        --bam-list /path/myBamList.tsv \
        --numeric-clinical-data-file /path/myNumericData.tsv \
        --categorical-clinical-data-file /path/myClassData.tsv \
        --genetic-data-type 'gene' \
        --output-file /path/output_file
 ... music clinical-correlation \
        --maf-file /path/myMAF.tsv \
        --bam-list /path/myBamList.tsv \
        --output-file /path/output_file \
        --glm-model-file /path/model.tsv \
        --glm-clinical-data-file /path/glm_clinical_data.tsv \
        --use-maf-in-glm
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">bam-list <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of BAM files [sample_name, normal_bam,
      tumor_bam] (See Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-file <i>Text</i></dt>
  <dd class="It-tag">Results of clinical-correlation tool. Will have suffix
      added for data type</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">maf-file <i>Text</i></dt>
  <dd class="It-tag">List of mutations using TCGA MAF specification v2.3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">glm-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Clinical traits, mutational profiles, other mixed clinical
      data (See DESCRIPTION)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">use-maf-in-glm <i>Boolean</i></dt>
  <dd class="It-tag">Create a variant matrix from the MAF file as variant input
      to GLM analysis.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'false' (--nouse-maf-in-glm) if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">nouse-maf-in-glm <i>Boolean</i></dt>
  <dd class="It-tag">Make use-maf-in-glm 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Skip non-coding mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-non-coding 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Skip silent mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-silent 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">clinical-correlation-matrix-file <i>Text</i></dt>
  <dd class="It-tag">Specify a file to store the sample-vs-gene matrix created
      during calculations</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">input-clinical-correlation-matrix-file <i>Text</i></dt>
  <dd class="It-tag">Instead of creating this from the MAF, input the
      sample-vs-gene matrix for calculations</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">genetic-data-type <i>Text</i></dt>
  <dd class="It-tag">Correlate clinical data to &quot;gene&quot; or
      &quot;variant&quot; level data
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'gene' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">numeric-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Table of samples (y) vs. numeric clinical data category
    (x)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">numerical-data-test-method <i>Text</i></dt>
  <dd class="It-tag">Either 'cor' for Pearson Correlation or 'wilcox' for the
      Wilcoxon Rank-Sum Test for numerical clinical data
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'cor' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">categorical-clinical-data-file <i>Text</i></dt>
  <dd class="It-tag">Table of samples (y) vs. categorical clinical data category
      (x)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">glm-model-file <i>Text</i></dt>
  <dd class="It-tag">File outlining the type of model, response variable,
      covariants, etc. for the GLM analysis. (See DESCRIPTION)</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This command relates clinical traits and mutational data. Either one can perform
  correlation analysis between mutations recorded in a MAF and the particular
  phenotypic traits recorded in clinical data files for the same samples, or one
  can run a generalized linear model (GLM) analysis on the same types of data.
<div class="Pp"></div>
The clinical data files for correlation must be separated between numeric and
  categoric data and must follow these conventions:
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Headers are required</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Each file must include at least 1 sample_id column and 1
      attribute column, with the format being [sample_id
      clinical_data_attribute_1 clinical_data_attribute_2 ...]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The sample ID must match the sample ID listed in the MAF
      under &quot;Tumor_Sample_Barcode&quot; for relating the mutations of this
      sample.</dd>
</dl>
<div class="Pp"></div>
Note the importance of the headers: the header for each clinical_data_attribute
  will appear in the output file to denote relationships with the mutation data
  from the MAF.
<div class="Pp"></div>
Internally, the input data is fed into an R script which calculates a P-value
  representing the probability that the correlation seen between the mutations
  in each gene (or variant) and each phenotype trait are random. Lower P-values
  indicate lower randomness, or likely true correlations.
<div class="Pp"></div>
The results are saved to the output filename given with a suffix appended;
  &quot;.numeric.csv&quot; will be appended for results derived from numeric
  clinical data, and &quot;.categorical.csv&quot; will be appended for results
  derived from categorical clinical data. Also, &quot;.glm.csv&quot; will be
  appended to the output filename for GLM results.
<div class="Pp"></div>
The GLM analysis accepts a mixed numeric and categoric clinical data file, input
  using the parameter --glm-clinical-data-file. GLM clinical data must adhere to
  the formats described above for the correlation clinical data files. GLM also
  requires the user to input a --glm-model-file. This file requires specific
  headers and defines the analysis to be performed rather exactly. Here are the
  conventions required for this file:
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Columns must be ordered as such:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">[ analysis_type clinical_data_trait_name variant/gene_name
      covariates memo ]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The 'analysis_type' column must contain either
      &quot;Q&quot;, indicating a quantative trait, or &quot;B&quot;, indicating
      a binary trait will be examined.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The 'clinical_data_trait_name' is the name of a clinical
      data trait defined by being a header in the --glm-clinical-data-file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The 'variant/gene_name' can either be the name of one or
      more columns from the --glm-clinical-data-file, or the name of one or more
      mutated gene names from the MAF, separated by &quot;|&quot;. If this
      column is left blank, or instead contains &quot;NA&quot;, then each column
      from either the variant mutation matrix (--use-maf-in-glm) or
      alternatively the --glm-clinical-data-file is used consecutively as the
      variant column in independent analyses.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">'covariates' are the names of one or more columns from the
      --glm-clinical-data-file, separated by &quot;+&quot;.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">'memo' is any note deemed useful to the user. It will be
      printed in the output data file for reference.</dd>
</dl>
<div class="Pp"></div>
GLM analysis may be performed using solely the data input into
  --glm-clinical-data-file, as described above, or alternatively, mutational
  data from the MAF may be included as variants in the GLM analysis, as also
  described above. Use the --use-maf-in-glm flag to include the mutation matrix
  derived from the maf as variant data.
<div class="Pp"></div>
Note that all input files for both correlation and GLM analysis must be
  tab-separated.
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--bam-list</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Provide a file containing sample names and normal/tumor BAM
    locations for each. Use the tab- delimited format [sample_name normal_bam
    tumor_bam] per line. This tool only needs sample_name, so all other columns
    can be skipped. The sample_name must be the same as the tumor sample names
    used in the MAF file (16th column, with the header
    Tumor_Sample_Barcode).</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
Copyright (C) 2010-2011 Washington University in St. Louis.
<div class="Pp"></div>
It is released under the Lesser GNU Public License (LGPL) version 3. See the
  associated LICENSE file in this distribution.
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Nathan D. Dees, Ph.D.
 Qunyuan Zhang, Ph.D.
 William Schierding, M.S.
</pre>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>genome-music</b>(1), <b>genome</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
