<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:12 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMT-MUSIC-SMG(1p) User Contributed Perl Documentation
GMT-MUSIC-SMG(1p)</p>

<p style="margin-top: 1em">gmt music smg NAME <br>
gmt music smg - Identify significantly mutated genes.</p>

<p style="margin-top: 1em">VERSION <br>
This document describes gmt music smg version 0.04
(2016-01-01 at 23:10:18)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmt music smg --gene-mr-file=? --output-file=? [--max-fdr=?]
[--skip-low-mr-genes] [--bmr-modifier-file=?]
[--processors=?]</p>

<p style="margin-top: 1em">... music smg --gene-mr-file
output_dir/gene_mrs --output-file output_dir/smgs</p>

<p style="margin-top: 1em">(A &quot;gene-mr-file&quot; can
be generated using the tool &quot;music bmr
calc-bmr&quot;.)</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
gene-mr-file Text <br>
File with per-gene mutation rates (Created using &quot;music
bmr calc-bmr&quot;)</p>

<p style="margin-top: 1em">output-file Text <br>
Output file that will list significantly mutated genes and
their p-values</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
max-fdr Number <br>
The maximum allowed false discovery rate for a gene to be
considered an SMG</p>

<p style="margin-top: 1em">Default value &rsquo;0.2&rsquo;
if not specified</p>

<p style="margin-top: 1em">skip-low-mr-genes Boolean <br>
Skip testing genes with MRs lower than the background MR</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-low-mr-genes Boolean <br>
Make skip-low-mr-genes &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">bmr-modifier-file Text <br>
Tab delimited multipliers per gene that modify BMR before
testing [gene_name bmr_modifier]</p>

<p style="margin-top: 1em">processors Integer <br>
Number of processors to use (requires &rsquo;foreach&rsquo;
and &rsquo;doMC&rsquo; R packages)</p>

<p style="margin-top: 1em">Default value &rsquo;1&rsquo; if
not specified</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script runs R-based statistical tools to identify
Significantly Mutated Genes (SMGs), when given per-gene
mutation rates categorized by mutation type, and the overall
<br>
background mutation rates (BMRs) for each of those
categories (gene_mr_file, created using &quot;music bmr
calc-bmr&quot;).</p>

<p style="margin-top: 1em">P-values and false discovery
rates (FDRs) for each gene in gene_mr_file is calculated
using three tests: Fisher&rsquo;s Combined P-value test
(FCPT), Likelihood Ratio test (LRT), and <br>
the Convolution test (CT). For a gene, if its FDR for at
least 2 of these tests is &lt;= max_fdr, it will be output
as an SMG. Another output file with prefix
&quot;_detailed&quot; will have <br>
p-values and FDRs for all genes.</p>

<p style="margin-top: 1em">ARGUMENTS <br>
--bmr-modifier-file <br>
The user can provide a BMR modifier for each gene in the ROI
file, which is a multiplier for the categorized background
mutation rates, before testing them against the gene&rsquo;s
<br>
categorized mutation rates. Such a file can be used to
correct for regional or systematic bias in mutation rates
across the genome that may be correlated to CpG deamination
<br>
or DNA repair processes like transcription-coupled repair or
mismatch repair. Mutation rates have also been associated
with DNA replication timing, where higher mutation <br>
rates are seen in late replicating regions. Note that the
same per-gene multiplier is used on each mutation category
of BMR. Any genes from the ROI file that are not in the <br>
BMR modifier file will be tested against unmodified overall
BMRs per mutation category. BMR modifiers of &lt;=0 are not
permitted, because that&rsquo;s just silly. <br>
--skip-low-mr-genes <br>
Genes with consistently lower MRs than the BMRs across
mutation categories, may show up in the results as an SMG
(by CT or LRT). If such genes are not of interest, they may
<br>
be assigned a p-value of 1. This should also speed things
up. Genes with higher Indel or Truncation rates than the
background will not be skipped even if the gene&rsquo;s
overall <br>
MR is lower than the BMR. If bmr-modifiers are applied, this
step uses the modified BMRs instead.</p>

<p style="margin-top: 1em">AUTHORS <br>
Qunyuan Zhang, Ph.D. <br>
Cyriac Kandoth, Ph.D. <br>
Nathan D. Dees, Ph.D.</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GMT-MUSIC-SMG(1p)</p>
<hr>
</body>
</html>
