<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:11 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMT-MUSIC-COSMIC-OMIM(1p) User Contributed Perl
Documentation GMT-MUSIC-COSMIC-OMIM(1p)</p>

<p style="margin-top: 1em">gmt music cosmic-omim NAME <br>
gmt music cosmic-omim - Compare the amino acid changes of
supplied mutations to COSMIC and OMIM databases.</p>

<p style="margin-top: 1em">VERSION <br>
This document describes gmt music cosmic-omim (2016-01-01 at
23:10:19)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmt music cosmic-omim --maf-file=? --output-file=?
--reference-build=? [--omimaa-dir=?] [--cosmic-dir=?]
[--verbose] [--wu-annotation-headers] [--aa-range=?]
[--nuc-range=?] <br>
[--show-known-hits]</p>

<p style="margin-top: 1em">... music cosmic-omim --maf-file
input_dir/myMAF.tsv --output-file
output_dir/myMAF_output.tsv --no-verbose</p>

<p style="margin-top: 1em">... music cosmic-omim --maf-file
input_dir/myMAF.tsv --output-file
output_dir/myMAF_output.tsv --omimaa-dir omim_dir/
--cosmic-dir cosmic_dir/ --no-verbose</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
maf-file Path <br>
list of annotated mutations in MAF format (or any file with
MAF+annotation headers)</p>

<p style="margin-top: 1em">output-file Path <br>
Output file contains the input file with two columns
appended to the end, corresponding to cosmic and omim
mutation comparisons, respectively</p>

<p style="margin-top: 1em">reference-build Text <br>
Put either &quot;Build36&quot; or &quot;Build37&quot;</p>

<p style="margin-top: 1em">Default value
&rsquo;Build37&rsquo; if not specified</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
omimaa-dir Path <br>
omim amino acid mutation database folder</p>

<p style="margin-top: 1em">cosmic-dir Path <br>
cosmic amino acid mutation database folder</p>

<p style="margin-top: 1em">verbose Boolean <br>
Use this to display the larger working output</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--noverbose) if not specified</p>

<p style="margin-top: 1em">noverbose Boolean <br>
Make verbose &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">wu-annotation-headers Boolean
<br>
Use this if input MAF contains WUSTL annotation format
headers</p>

<p style="margin-top: 1em">Default value
&rsquo;false&rsquo; (--nowu-annotation-headers) if not
specified</p>

<p style="margin-top: 1em">nowu-annotation-headers Boolean
<br>
Make wu-annotation-headers &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">aa-range Integer <br>
Set how close a &rsquo;near&rsquo; match is when searching
for amino acid near hits</p>

<p style="margin-top: 1em">Default value &rsquo;2&rsquo; if
not specified</p>

<p style="margin-top: 1em">nuc-range Integer <br>
Set how close a &rsquo;near&rsquo; match is when searching
for nucleotide position near hits</p>

<p style="margin-top: 1em">Default value &rsquo;5&rsquo; if
not specified</p>

<p style="margin-top: 1em">show-known-hits Boolean <br>
When a finding is novel, show known AA in that gene</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noshow-known-hits Boolean <br>
Make show-known-hits &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This tool looks at the amino acid changes for the given set
of mutations and compares the genomic coordinates as well as
the affected amino acid to the coordinates and amino <br>
acids of all cancer-specific mutations listed in the Cosmic
and OMIM databases. The database files are specially
prepared for this task and provided with the MuSiC suite.
The <br>
tool reports various types of matches, including matches
within &quot;near proximity&quot;, where &quot;near
proximity&quot; is currently defined as a linear DNA
distance of 5 bases or 2 amino <br>
acids. (This type of matching helps to account for the
possibility of subtle differences in reported positions for
variants due to differences in transcript definitions or
other <br>
things of this nature.) Any site without a match in a
particular databases is reported as &quot;novel&quot; with
respect to that database.</p>

<p style="margin-top: 1em">The output of this script
returns each row the original input MAF file with two
columns appended to the end of each, one column for each of
the databases. Also included is a <br>
STDOUT printout of a summary of what was found in the input
MAF. Neither output can be suppressed in the current
version. The --verbose option is used to display working
notes <br>
that are useful for various purposes in debugging potential
MAF problems. The Omim and Cosmic directories must point to
the output of the downloader, named appropriately, as they
<br>
don&rsquo;t recognize the OMIM database in the raw download
format.</p>

<p style="margin-top: 1em">This tool only compares build 36
or build 37 coordinates that are specified in Cosmic to the
coordinates in your maf file. This is a weakness of the
Cosmic database (not all <br>
position entries are currently available for both builds)
that is out of our control.</p>

<p style="margin-top: 1em">In addition to the standard
version 2.3 MAF headers, there needs to be 3 columns
appended. These column headers in the MAF must have these
names in the header in order for the <br>
tool to find them: <br>
transcript_name - the transcript name, such as NM_000028
amino_acid_change - the amino acid change, such as p.R290H
<br>
c_position - the nucleotide position changed, such as
c.869</p>

<p style="margin-top: 1em">LICENSE <br>
Copyright (C) 2010-2011 Washington University in St.
Louis.</p>

<p style="margin-top: 1em">It is released under the Lesser
GNU Public License (LGPL) version 3. See the associated
LICENSE file in this distribution.</p>

<p style="margin-top: 1em">AUTHORS <br>
Brian Dunford-Shore <br>
David Larson, Ph.D. <br>
Michael C. Wendl, Ph.D. <br>
William Schierding, M.S.</p>

<p style="margin-top: 1em">CREDITS <br>
This tool depends on copies of data from the following
databases, packaged in a form useable for quick
analysis:</p>

<p style="margin-top: 1em">* COSMIC -
http://www.sanger.ac.uk/genetics/CGP/cosmic/ <br>
* OMIM - http://www.ncbi.nlm.nih.gov/omim</p>

<p style="margin-top: 1em">SEE ALSO <br>
genome-music(1), genome(1)</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GMT-MUSIC-COSMIC-OMIM(1p)</p>
<hr>
</body>
</html>
