<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GMT-MUSIC-PATH-SCAN(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GMT-MUSIC-PATH-SCAN(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GMT-MUSIC-PATH-SCAN(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="gmt_music_path-scan"><a class="selflink" href="#gmt_music_path-scan">gmt
  music path-scan</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gmt music path-scan - Find signifcantly mutated pathways in a cohort given a
  list of somatic mutations.
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes gmt music path-scan version 0.04 (2016-01-01 at
  23:10:19)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
gmt music path-scan --gene-covg-dir=? --bam-list=? --pathway-file=? --maf-file=?
  --output-file=? [--bmr=?] [--genes-to-ignore=?] [--min-mut-genes-per-path=?]
  [--skip-non-coding] [--skip-silent]
<div class="Pp"></div>
<pre>
 ... music path-scan \
        --bam-list input_dir/bam_file_list \
        --gene-covg-dir output_dir/gene_covgs/ \
        --maf-file input_dir/myMAF.tsv \
        --output-file output_dir/sm_pathways \
        --pathway-file input_dir/pathway_dbs/KEGG.txt \
        --bmr 8.7E-07
</pre>
<h1 class="Sh" title="Sh" id="REQUIRED_ARGUMENTS"><a class="selflink" href="#REQUIRED_ARGUMENTS">REQUIRED
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">gene-covg-dir <i>Text</i></dt>
  <dd class="It-tag">Directory containing per-gene coverage files (Created using
      music bmr calc-covg)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bam-list <i>Text</i></dt>
  <dd class="It-tag">Tab delimited list of BAM files [sample_name, normal_bam,
      tumor_bam] (See Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">pathway-file <i>Text</i></dt>
  <dd class="It-tag">Tab-delimited file of pathway information (See
    Description)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">maf-file <i>Text</i></dt>
  <dd class="It-tag">List of mutations using TCGA MAF specifications v2.3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">output-file <i>Text</i></dt>
  <dd class="It-tag">Output file that will list the significant pathways and
      their p-values</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONAL_ARGUMENTS"><a class="selflink" href="#OPTIONAL_ARGUMENTS">OPTIONAL
  ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">bmr <i>Number</i></dt>
  <dd class="It-tag">Background mutation rate in the targeted regions
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '1e-06' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">genes-to-ignore <i>Text</i></dt>
  <dd class="It-tag">Comma-delimited list of genes whose mutations should be
      ignored</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">min-mut-genes-per-path <i>Number</i></dt>
  <dd class="It-tag">Pathways with fewer mutated genes than this, will be
      ignored
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value '1' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Skip non-coding mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-non-coding <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-non-coding 'false'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">skip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Skip silent mutations from the provided MAF file
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value 'true' if not specified</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">noskip-silent <i>Boolean</i></dt>
  <dd class="It-tag">Make skip-silent 'false'</dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Only the following four columns in the MAF are used. All other columns may be
  left blank.
<div class="Pp"></div>
<pre>
 Col 1: Hugo_Symbol (Need not be HUGO, but must match gene names used in the pathway file)
 Col 2: Entrez_Gene_Id (Matching Entrez ID trump gene name matches between pathway file and MAF)
 Col 9: Variant_Classification
 Col 16: Tumor_Sample_Barcode (Must match the name in sample-list, or contain it as a substring)
</pre>
<div class="Pp"></div>
The Entrez_Gene_Id can also be left blank (or set to 0), but it is highly
  recommended, in case genes are named differently in the pathway file and the
  MAF file.
<h1 class="Sh" title="Sh" id="ARGUMENTS"><a class="selflink" href="#ARGUMENTS">ARGUMENTS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--pathway-file</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">This is a tab-delimited file prepared from a pathway
    database (such as KEGG), with the columns: [path_id, path_name, class,
    gene_line, diseases, drugs, description] The latter three columns are
    optional (but are available on KEGG). The gene_line contains the
    &quot;entrez_id:gene_name&quot; of all genes involved in this pathway, each
    separated by a &quot;|&quot; symbol.</dt>
  <dd class="It-tag">For example, a line in the pathway-file would look like:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
  hsa00061    Fatty acid biosynthesis    Lipid Metabolism    31:ACACA|32:ACACB|27349:MCAT|2194:FASN|54995:OXSM|55301:OLAH
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    Ensure that the gene names and entrez IDs used match those used in the MAF
      file. Entrez IDs are not mandatory (use a 0 if Entrez ID unknown). But if
      a gene name in the MAF does not match any gene name in this file, the
      entrez IDs are used to find a match (unless it's a 0).</dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--gene-covg-dir</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">This is usually the gene_covgs subdirectory created when
    you run &quot;music bmr calc-covg&quot;. It should contain files for each
    sample that report per-gene covered base counts.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--bam-list</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">Provide a file containing sample names and normal/tumor BAM
    locations for each. Use the tab- delimited format [sample_name normal_bam
    tumor_bam] per line. This tool only needs sample_name, so all other columns
    can be skipped. The sample_name must be the same as the tumor sample names
    used in the MAF file (16th column, with the header
    Tumor_Sample_Barcode).</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--bmr</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">The overall background mutation rate. This can be
    calculated using &quot;music bmr calc-bmr&quot;.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag">--genes-to-ignore</dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 4.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">A comma-delimited list of genes to ignore from the MAF
    file. This is useful when there are recurrently mutated genes like TP53
    which might mask the significance of other genes.</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<pre>
 Michael Wendl, Ph.D.
</pre>
<h1 class="Sh" title="Sh" id="CREDITS"><a class="selflink" href="#CREDITS">CREDITS</a></h1>
This module uses reformatted copies of data from the Kyoto Encyclopedia of Genes
  and Genomes (KEGG) database:
<div class="Pp"></div>
<pre>
 * KEGG - http://www.genome.jp/kegg/
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-01-01</td>
    <td class="foot-os">perl v5.22.1</td>
  </tr>
</table>
</body>
</html>
