<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:12 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMT-MUSIC-PROXIMITY(1p) User Contributed Perl
Documentation GMT-MUSIC-PROXIMITY(1p)</p>

<p style="margin-top: 1em">gmt music proximity NAME <br>
gmt music proximity - Perform a proximity analysis on a list
of mutations.</p>

<p style="margin-top: 1em">VERSION <br>
This document describes gmt music proximity version 0.04
(2016-01-01 at 23:10:19)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmt music proximity --maf-file=? --output-file=?
--output-dir=? [--max-proximity=?] [--skip-non-coding]
[--skip-silent]</p>

<p style="margin-top: 1em">... music proximity --maf-file
input_dir/myMAF.tsv --output-dir output_dir/ --max-proximity
15</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
maf-file Text <br>
List of mutations using TCGA MAF specifications v2.3</p>

<p style="margin-top: 1em">output-file Text <br>
TODO</p>

<p style="margin-top: 1em">output-dir Text <br>
Directory where output files will be written</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
max-proximity Text <br>
Maximum allowed AA distance between 2 mutations</p>

<p style="margin-top: 1em">Default value &rsquo;7&rsquo; if
not specified</p>

<p style="margin-top: 1em">skip-non-coding Boolean <br>
Skip non-coding mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-non-coding Boolean <br>
Make skip-non-coding &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">skip-silent Boolean <br>
Skip silent mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-silent Boolean <br>
Make skip-silent &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This module first calculates the amino acid position of each
mutation in the MAF file within its respective transcript.
Then, for each mutation, two values are calculated: 1) the
<br>
number of other mutations on the same transcript within the
proximity limit set by the max-proximity input parameter,
and 2) the distance to the closest other mutation in this
<br>
nearby set. Only mutations which have another mutation
within close proximity are reported in the output-file.</p>

<p style="margin-top: 1em">In addition to the standard
version 2.3 MAF headers, there needs to be 3 columns
appended. These column headers in the MAF must have these
names in the header in order for the <br>
tool to find them: <br>
transcript_name - the transcript name, such as NM_000028
amino_acid_change - the amino acid change, such as p.R290H
<br>
c_position - the nucleotide position changed, such as
c.869</p>

<p style="margin-top: 1em">The output is generated with the
folowing column headers: Mutations_Within_Proximity,
Nearest_Mutation, Gene, Transcript, Affected_Amino_Acid(s),
Chr, Start, Stop, Ref_Allele, <br>
Var_Allele, Sample</p>

<p style="margin-top: 1em">AUTHORS <br>
Nathan D. Dees, Ph.D. <br>
Dan Koboldt, M.S. <br>
Cyriac Kandoth, Ph.D.</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GMT-MUSIC-PROXIMITY(1p)</p>
<hr>
</body>
</html>
