<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:10 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMT-MUSIC(1p) User Contributed Perl Documentation
GMT-MUSIC(1p)</p>

<p style="margin-top: 1em">gmt music NAME <br>
gmt music - Mutational Significance in Cancer (Cancer
Mutation Analysis)</p>

<p style="margin-top: 1em">VERSION <br>
This document describes gmt music version 0.04 (2016-01-01
at 23:10:18)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmt music ...</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The MuSiC suite is a set of tools aimed at discovering the
significance of somatic mutations found within a given
cohort of cancer samples, and with respect to a variety of
<br>
external data sources. The standard inputs required are:</p>

<p style="margin-top: 1em">1. mapped reads in BAM format
<br>
2. predicted or validated SNVs or indels in mutation
annotation format (MAF) <br>
3. a list of regions of interest (typically the boundaries
of coding exons) <br>
4. any relevant numeric or categorical clinical data.</p>

<p style="margin-top: 1em">The formats for inputs 3. and 4.
are:</p>

<p style="margin-top: 1em">3. Regions of Interest File:
<br>
&Acirc;&middot; Do not use headers</p>

<p style="margin-top: 1em">&Acirc;&middot; 4 columns, which
are [chromosome start-position(1-based)
stop-position(1-based) gene_name]</p>

<p style="margin-top: 1em">4. Clinical Data Files: <br>
&Acirc;&middot; Headers are required</p>

<p style="margin-top: 1em">&Acirc;&middot; At least 1
sample_id column and 1 attribute column, with the format
being [sample_id clinical_data_attribute
clinical_data_attribute ...]</p>

<p style="margin-top: 1em">&Acirc;&middot; The sample_id
must match the sample_id listed in the MAF under
&quot;Tumor_Sample_Barcode&quot; for relating the mutations
of this sample.</p>

<p style="margin-top: 1em">&Acirc;&middot; The header for
each clinical_data_attribute will appear in the output file
to denote relationships with the mutation data from the
MAF.</p>

<p style="margin-top: 1em">Descriptions for the usage of
each tool (each sub-command) can be found separately.</p>

<p style="margin-top: 1em">The play command runs all of the
sub-commands serially on a selected input set.</p>

<p style="margin-top: 1em">SUB-COMMANDS <br>
GENERAL <br>
bmr <br>
Calculate gene coverages and background mutation rates.</p>

<p style="margin-top: 1em">clinical-correlation <br>
Correlate phenotypic traits against mutated genes, or
against individual variants</p>

<p style="margin-top: 1em">cosmic-omim <br>
Compare the amino acid changes of supplied mutations to
COSMIC and OMIM databases.</p>

<p style="margin-top: 1em">galaxy <br>
Run the full suite of MuSiC tools sequentially.</p>

<p style="margin-top: 1em">mutation-relation <br>
Identify relationships of mutation concurrency or mutual
exclusivity in genes across cases.</p>

<p style="margin-top: 1em">path-scan <br>
Find signifcantly mutated pathways in a cohort given a list
of somatic mutations.</p>

<p style="margin-top: 1em">pfam <br>
Add Pfam annotation to a MAF file</p>

<p style="margin-top: 1em">play <br>
Run the full suite of MuSiC tools sequentially.</p>

<p style="margin-top: 1em">plot <br>
Generate relevant plots and visualizations for MuSiC.</p>

<p style="margin-top: 1em">proximity <br>
Perform a proximity analysis on a list of mutations.</p>

<p style="margin-top: 1em">smg <br>
Identify significantly mutated genes.</p>

<p style="margin-top: 1em">survival <br>
Create survival plots and P-values for clinical and
mutational phenotypes.</p>

<p style="margin-top: 1em">LICENSE <br>
Copyright (C) 2007-2011 Washington University in St.
Louis.</p>

<p style="margin-top: 1em">It is released under the Lesser
GNU Public License (LGPL) version 3. See the associated
LICENSE file in this distribution.</p>

<p style="margin-top: 1em">AUTHORS <br>
This software is developed by the analysis and engineering
teams at The Genome Institute at Washington University
School of Medicine in St. Louis, with funding from the
National <br>
Human Genome Research Institute. Richard K. Wilson, P.I.</p>

<p style="margin-top: 1em">The primary authors of the MuSiC
suite are:</p>

<p style="margin-top: 1em">Nathan D. Dees, Ph.D. <br>
Cyriac Kandoth, Ph.D. <br>
Dan Koboldt, M.S. <br>
William Schierding, M.S. <br>
Michael Wendl, Ph.D. <br>
Qunyuan Zhang, Ph.D. <br>
Thomas B. Mooney, M.S.</p>

<p style="margin-top: 1em">CREDITS <br>
The MuSiC suite uses tabix, by Heng Li. See
http://samtools.sourceforge.net/tabix.shtml.</p>

<p style="margin-top: 1em">MuSiC depends on copies of data
from the following databases, packaged in a form useable for
quick analysis:</p>

<p style="margin-top: 1em">* KEGG -
http://www.genome.jp/kegg/ <br>
* COSMIC - http://www.sanger.ac.uk/genetics/CGP/cosmic/ <br>
* OMIM - http://www.ncbi.nlm.nih.gov/omim <br>
* Pfam - http://pfam.sanger.ac.uk/ <br>
* SMART - http://smart.embl-heidelberg.de/ <br>
* SUPERFAMILY - http://supfam.cs.bris.ac.uk/SUPERFAMILY/
<br>
* PatternScan - http://www.expasy.ch/prosite/</p>

<p style="margin-top: 1em">BUGS <br>
For defects with any software in the genome namespace,
contact <br>
genome-dev ~at~ genome.wustl.edu.</p>

<p style="margin-top: 1em">SEE ALSO <br>
genome(1)</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GMT-MUSIC(1p)</p>
<hr>
</body>
</html>
