<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:11 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMT-MUSIC-BMR-CALC-COVG-HELPER(1p) User Contributed Perl
Documentation GMT-MUSIC-BMR-CALC-COVG-HELPER(1p)</p>

<p style="margin-top: 1em">gmt music bmr calc-covg-helper
NAME <br>
gmt music bmr calc-covg-helper - Uses calcRoiCovg.c to count
covered bases per-gene for a tumor-normal pair of BAMs.</p>

<p style="margin-top: 1em">VERSION <br>
This document describes gmt music bmr calc-covg-helper
version 0.04 (2016-01-01 at 23:10:19)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmt music bmr calc-covg-helper --roi-file=?
--reference-sequence=? --normal-tumor-bam-pair=?
[--output-file=?] [--output-dir=?] [--normal-min-depth=?]
[--tumor-min-depth=?] <br>
[--min-mapq=?]</p>

<p style="margin-top: 1em">General usage:</p>

<p style="margin-top: 1em">... music bmr calc-covg-helper
--normal-tumor-bam-pair &quot;sample-name path/to/normal_bam
path/to/tumor_bam&quot; --reference-sequence
input_dir/all_sequences.fa --output-file output_file
--roi-file input_dir/all_coding_exons.tsv</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
roi-file Text <br>
Tab delimited list of ROIs [chr start stop gene_name] (See
Description)</p>

<p style="margin-top: 1em">reference-sequence Text <br>
Path to reference sequence in FASTA format</p>

<p style="margin-top: 1em">normal-tumor-bam-pair Text <br>
Tab delimited line with sample name, path to normal bam
file, and path to tumor bam file (See Description)</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
output-file Text <br>
Output file path. Specify either output-file or
output-directory.</p>

<p style="margin-top: 1em">output-dir Text <br>
Output directory path. Specify either output-file or
output-directory</p>

<p style="margin-top: 1em">normal-min-depth Integer <br>
The minimum read depth to consider a Normal BAM base as
covered</p>

<p style="margin-top: 1em">Default value &rsquo;6&rsquo; if
not specified</p>

<p style="margin-top: 1em">tumor-min-depth Integer <br>
The minimum read depth to consider a Tumor BAM base as
covered</p>

<p style="margin-top: 1em">Default value &rsquo;8&rsquo; if
not specified</p>

<p style="margin-top: 1em">min-mapq Integer <br>
The minimum mapping quality of reads to consider towards
read depth counts</p>

<p style="margin-top: 1em">Default value &rsquo;20&rsquo;
if not specified</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script counts bases with sufficient coverage in the
ROIs of each gene in the given pair of tumor-normal BAM
files and categorizes them into - AT, CG (non-CpG), and CpG
<br>
counts. It also adds up these base-counts across all ROIs of
each gene in the sample, but covered bases that lie within
overlapping ROIs are not counted more than once towards <br>
these total counts.</p>

<p style="margin-top: 1em">ARGUMENTS <br>
--roi-file <br>
The regions of interest (ROIs) of each gene are typically
regions targeted for sequencing or are merged exon loci
(from multiple transcripts) of genes with 2-bp flanks <br>
(splice junctions). ROIs from the same chromosome must be
listed adjacent to each other in this file. This allows the
underlying C-based code to run much more efficiently and
<br>
avoid re-counting bases seen in overlapping ROIs (for
overall covered base counts). For per-gene base counts, an
overlapping base will be counted each time it appears in an
<br>
ROI of the same gene. To avoid this, be sure to merge
together overlapping ROIs of the same gene. BEDtools&rsquo;
mergeBed can help if used per gene. <br>
--reference-sequence <br>
The reference sequence in FASTA format. If a reference
sequence index is not found next to this file (a .fai file),
it will be created. <br>
--normal-tumor-bam-pair <br>
&quot;sample-name path/to/normal_bam path/to/tumor_bam&quot;
<br>
--output-file <br>
Specify an output file where the per-ROI covered base counts
will be written</p>

<p style="margin-top: 1em">LICENSE <br>
Copyright (C) 2010-2011 Washington University in St.
Louis.</p>

<p style="margin-top: 1em">It is released under the Lesser
GNU Public License (LGPL) version 3. See the associated
LICENSE file in this distribution.</p>

<p style="margin-top: 1em">AUTHORS <br>
Cyriac Kandoth, Ph.D.</p>

<p style="margin-top: 1em">SEE ALSO <br>
genome-music-bmr(1), genome-music(1), genome(1)</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GMT-MUSIC-BMR-CALC-COVG-HELPER(1p)</p>
<hr>
</body>
</html>
