<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:12 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMT-MUSIC-SURVIVAL(1p) User Contributed Perl
Documentation GMT-MUSIC-SURVIVAL(1p)</p>

<p style="margin-top: 1em">gmt music survival NAME <br>
gmt music survival - Create survival plots and P-values for
clinical and mutational phenotypes.</p>

<p style="margin-top: 1em">VERSION <br>
This document describes gmt music survival version 0.04
(2016-01-01 at 23:10:19)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmt music survival --bam-list=? --output-dir=?
[--maf-file=?] [--skip-silent] [--genetic-data-type=?]
[--numeric-clinical-data-file=?]
[--categorical-clinical-data-file=?] <br>
[--glm-clinical-data-file=?] [--phenotypes-to-include=?]
[--legend-placement=?] [--skip-non-coding]</p>

<p style="margin-top: 1em">... music survival --bam-list
/path/myBamList.tsv --maf-file /path/myMAF.tsv
--numeric-clinical-data-file /path/myNumericData.tsv
--categorical-clinical-data-file /path/myClassData.tsv
--output-dir /path/output_directory</p>

<p style="margin-top: 1em">... music survival --bam-list
/path/myBamList.tsv --maf-file /path/myMAF.tsv
--glm-clinical-data-file /path/myGLMClinicalData.tsv
--output-dir /path/output_directory</p>

<p style="margin-top: 1em">... music survival --bam-list
/path/myBamList.tsv --maf-file /path/myMAF.tsv
--genetic-data-type &rsquo;gene&rsquo;
--glm-clinical-data-file /path/myGlmClinicalData.tsv
--phenotypes-to-include &rsquo;Race,Gender,TP53&rsquo;
--output-dir /path/output_directory</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
bam-list Text <br>
List of sample names to be included in the analysis. (See
Description)</p>

<p style="margin-top: 1em">output-dir Text <br>
Directory where output files will be written</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
maf-file Text <br>
List of mutations in MAF format</p>

<p style="margin-top: 1em">skip-silent Boolean <br>
Skip silent mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-silent Boolean <br>
Make skip-silent &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">genetic-data-type Text <br>
Correlate clinical data to &quot;gene&quot; or
&quot;variant&quot; level data</p>

<p style="margin-top: 1em">Default value &rsquo;gene&rsquo;
if not specified</p>

<p style="margin-top: 1em">numeric-clinical-data-file Text
<br>
Table of samples (y) vs. numeric clinical data category
(x)</p>

<p style="margin-top: 1em">categorical-clinical-data-file
Text <br>
Table of samples (y) vs. categorical clinical data category
(x)</p>

<p style="margin-top: 1em">glm-clinical-data-file Text <br>
Clinical traits, mutational profiles, other mixed clinical
data (See DESCRIPTION).</p>

<p style="margin-top: 1em">phenotypes-to-include Text <br>
Include only these genes and/or phenotypes in the anlaysis.
(COMMA-DELIMITED)</p>

<p style="margin-top: 1em">legend-placement Text <br>
Choose one of &rsquo;bottomleft&rsquo;,
&rsquo;topleft&rsquo;, &rsquo;topright&rsquo;, or
&rsquo;bottomright&rsquo;.</p>

<p style="margin-top: 1em">Default value
&rsquo;bottomleft&rsquo; if not specified</p>

<p style="margin-top: 1em">skip-non-coding Boolean <br>
Skip non-coding mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-non-coding Boolean <br>
Make skip-non-coding &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This command performs survival analysis and plots survival
curves for mutational data, as well as any clinical traits
of interest as specified via the --phenotypes-to-include
<br>
input parameter. The analyses performed include the
Kaplan-Meier estimator followed by the Cox Proportional
Hazards model. Outputs for each gene/clinical trait analyzed
include <br>
survival curves, a hazard ratio (with confidence intervals),
and P-values and FDRs describing the significance of the
difference between survivors and non-survivors.</p>

<p style="margin-top: 1em">All clinical data files are
searched for the required (case insensitive)
&quot;vital_status&quot; and
&quot;days_to_last_followup&quot; columns which are paired
to phenotypes via sample IDs for the <br>
survival analysis. The first column of all clinical data
files MUST contain the sample IDs, same as in other MuSiC
tools. By default, analysis is performed on every gene
present <br>
in the MAF. Optionally, the analysis may be limited to only
specific genes by listing them (comma delimited) after the
--phenotypes-to-include input parameter. Survival analysis
<br>
may also be performed on other columns in the clinical data
file by adding the column headers to the list of entries
specified after the --phenotypes-to-include input
parameter.</p>

<p style="margin-top: 1em">Here are some general
guildelines for creating clinical data input files:</p>

<p style="margin-top: 1em">&Acirc;&middot; Headers are
required.</p>

<p style="margin-top: 1em">&Acirc;&middot; The first column
of each clinical data file must contain sample IDs which
match those in both the --bam-list and the MAF variant list
(in the MAF, this is the <br>
Tumor_Sample_Barcode column, specifically).</p>

<p style="margin-top: 1em">&Acirc;&middot; In at least one
of the clinical data files input, columns with headers
&quot;vital_status&quot; and
&quot;days_to_last_followup&quot; (case insensitive) must
exist. &quot;vital_status&quot; must be <br>
delineated by 1&rsquo;s and 0&rsquo;s, where 0 denotes
&rsquo;living&rsquo;, and 1 denotes
&rsquo;deceased&rsquo;.</p>

<p style="margin-top: 1em">Note that all input files must
be tab-separated.</p>

<p style="margin-top: 1em">ARGUMENTS <br>
--bam-list <br>
Provide a file containing sample names and normal/tumor BAM
locations for each. Use the tab- delimited format
[sample_name normal_bam tumor_bam] per line. This tool only
<br>
needs sample_name, so all other columns can be skipped. The
sample_name must be the same as the tumor sample names used
in the MAF file (16th column, with the header <br>
Tumor_Sample_Barcode).</p>

<p style="margin-top: 1em">LICENSE <br>
Copyright (C) 2010-2011 Washington University in St.
Louis.</p>

<p style="margin-top: 1em">It is released under the Lesser
GNU Public License (LGPL) version 3. See the associated
LICENSE file in this distribution.</p>

<p style="margin-top: 1em">AUTHORS <br>
Nathan D. Dees, Ph.D. <br>
Qunyuan Zhang, Ph.D.</p>

<p style="margin-top: 1em">SEE ALSO <br>
genome-music(1), genome(1)</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GMT-MUSIC-SURVIVAL(1p)</p>
<hr>
</body>
</html>
