<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:14:11 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GMT-MUSIC-PATH-SCAN(1p) User Contributed Perl
Documentation GMT-MUSIC-PATH-SCAN(1p)</p>

<p style="margin-top: 1em">gmt music path-scan NAME <br>
gmt music path-scan - Find signifcantly mutated pathways in
a cohort given a list of somatic mutations.</p>

<p style="margin-top: 1em">VERSION <br>
This document describes gmt music path-scan version 0.04
(2016-01-01 at 23:10:19)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gmt music path-scan --gene-covg-dir=? --bam-list=?
--pathway-file=? --maf-file=? --output-file=? [--bmr=?]
[--genes-to-ignore=?] [--min-mut-genes-per-path=?]
[--skip-non-coding] <br>
[--skip-silent]</p>

<p style="margin-top: 1em">... music path-scan --bam-list
input_dir/bam_file_list --gene-covg-dir
output_dir/gene_covgs/ --maf-file input_dir/myMAF.tsv
--output-file output_dir/sm_pathways --pathway-file
input_dir/pathway_dbs/KEGG.txt --bmr 8.7E-07</p>

<p style="margin-top: 1em">REQUIRED ARGUMENTS <br>
gene-covg-dir Text <br>
Directory containing per-gene coverage files (Created using
music bmr calc-covg)</p>

<p style="margin-top: 1em">bam-list Text <br>
Tab delimited list of BAM files [sample_name, normal_bam,
tumor_bam] (See Description)</p>

<p style="margin-top: 1em">pathway-file Text <br>
Tab-delimited file of pathway information (See
Description)</p>

<p style="margin-top: 1em">maf-file Text <br>
List of mutations using TCGA MAF specifications v2.3</p>

<p style="margin-top: 1em">output-file Text <br>
Output file that will list the significant pathways and
their p-values</p>

<p style="margin-top: 1em">OPTIONAL ARGUMENTS <br>
bmr Number <br>
Background mutation rate in the targeted regions</p>

<p style="margin-top: 1em">Default value
&rsquo;1e-06&rsquo; if not specified</p>

<p style="margin-top: 1em">genes-to-ignore Text <br>
Comma-delimited list of genes whose mutations should be
ignored</p>

<p style="margin-top: 1em">min-mut-genes-per-path Number
<br>
Pathways with fewer mutated genes than this, will be
ignored</p>

<p style="margin-top: 1em">Default value &rsquo;1&rsquo; if
not specified</p>

<p style="margin-top: 1em">skip-non-coding Boolean <br>
Skip non-coding mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-non-coding Boolean <br>
Make skip-non-coding &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">skip-silent Boolean <br>
Skip silent mutations from the provided MAF file</p>

<p style="margin-top: 1em">Default value &rsquo;true&rsquo;
if not specified</p>

<p style="margin-top: 1em">noskip-silent Boolean <br>
Make skip-silent &rsquo;false&rsquo;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Only the following four columns in the MAF are used. All
other columns may be left blank.</p>

<p style="margin-top: 1em">Col 1: Hugo_Symbol (Need not be
HUGO, but must match gene names used in the pathway file)
<br>
Col 2: Entrez_Gene_Id (Matching Entrez ID trump gene name
matches between pathway file and MAF) <br>
Col 9: Variant_Classification <br>
Col 16: Tumor_Sample_Barcode (Must match the name in
sample-list, or contain it as a substring)</p>

<p style="margin-top: 1em">The Entrez_Gene_Id can also be
left blank (or set to 0), but it is highly recommended, in
case genes are named differently in the pathway file and the
MAF file.</p>

<p style="margin-top: 1em">ARGUMENTS <br>
--pathway-file <br>
This is a tab-delimited file prepared from a pathway
database (such as KEGG), with the columns: [path_id,
path_name, class, gene_line, diseases, drugs, description]
The <br>
latter three columns are optional (but are available on
KEGG). The gene_line contains the
&quot;entrez_id:gene_name&quot; of all genes involved in
this pathway, each separated by a &quot;|&quot; <br>
symbol. <br>
For example, a line in the pathway-file would look like:</p>

<p style="margin-top: 1em">hsa00061 Fatty acid biosynthesis
Lipid Metabolism
31:ACACA|32:ACACB|27349:MCAT|2194:FASN|54995:OXSM|55301:OLAH</p>

<p style="margin-top: 1em">Ensure that the gene names and
entrez IDs used match those used in the MAF file. Entrez IDs
are not mandatory (use a 0 if Entrez ID unknown). But if a
gene name in <br>
the MAF does not match any gene name in this file, the
entrez IDs are used to find a match (unless it&rsquo;s a
0).</p>

<p style="margin-top: 1em">--gene-covg-dir <br>
This is usually the gene_covgs subdirectory created when you
run &quot;music bmr calc-covg&quot;. It should contain files
for each sample that report per-gene covered base counts.
<br>
--bam-list <br>
Provide a file containing sample names and normal/tumor BAM
locations for each. Use the tab- delimited format
[sample_name normal_bam tumor_bam] per line. This tool only
<br>
needs sample_name, so all other columns can be skipped. The
sample_name must be the same as the tumor sample names used
in the MAF file (16th column, with the header <br>
Tumor_Sample_Barcode). <br>
--bmr <br>
The overall background mutation rate. This can be calculated
using &quot;music bmr calc-bmr&quot;. <br>
--genes-to-ignore <br>
A comma-delimited list of genes to ignore from the MAF file.
This is useful when there are recurrently mutated genes like
TP53 which might mask the significance of other <br>
genes.</p>

<p style="margin-top: 1em">AUTHORS <br>
Michael Wendl, Ph.D.</p>

<p style="margin-top: 1em">CREDITS <br>
This module uses reformatted copies of data from the Kyoto
Encyclopedia of Genes and Genomes (KEGG) database:</p>

<p style="margin-top: 1em">* KEGG -
http://www.genome.jp/kegg/</p>

<p style="margin-top: 1em">perl v5.22.1 2016-01-01
GMT-MUSIC-PATH-SCAN(1p)</p>
<hr>
</body>
</html>
