<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GSNAP(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GSNAP(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">GSNAP(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gsnap - Genomic Short-read Nucleotide Alignment Program
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gsnap</b> [ <i>OPTIONS</i>...] <i>&lt;FASTA file&gt;, or</i> cat &lt;FASTA
  file&gt; | gmap [OPTIONS...]
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_options_(must_include_-d)"><a class="selflink" href="#Input_options_(must_include_-d)">Input
  options (must include <b>-d</b>)</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b>, <b>--dir</b>=<i>directory</i></dt>
  <dd class="It-tag">Genome directory. Default (as specified by
      <b>--with-gmapdb</b> to the configure program) is
    <i>/var/cache/gmap</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b>, <b>--db</b>=<i>STRING</i></dt>
  <dd class="It-tag">Genome database</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--use-sarray</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether to use a suffix array, which will give increased
      speed. Allowed values: 0 (no), 1 (yes, plus GSNAP/GMAP algorithm,
      default), or 2 (yes, and use only suffix array algorithm). Note that
      suffix arrays will bias against SNP alleles in SNP-tolerant
    alignment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b>, <b>--kmer</b>=<i>INT</i></dt>
  <dd class="It-tag">kmer size to use in genome database (allowed values: 16 or
      less) If not specified, the program will find the highest available kmer
      size in the genome database</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sampling</b>=<i>INT</i></dt>
  <dd class="It-tag">Sampling to use in genome database. If not specified, the
      program will find the smallest available sampling value in the genome
      database within selected k-mer size</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b>, <b>--part</b>=<i>INT</i>/INT</dt>
  <dd class="It-tag">Process only the i-th out of every n sequences e.g., 0/100
      or 99/100 (useful for distributing jobs to a computer farm).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--input-buffer-size</b>=<i>INT</i></dt>
  <dd class="It-tag">Size of input buffer (program reads this many sequences at
      a time for efficiency) (default 1000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--barcode-length</b>=<i>INT</i></dt>
  <dd class="It-tag">Amount of barcode to remove from start of read (default
    0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--orientation</b>=<i>STRING</i></dt>
  <dd class="It-tag">Orientation of paired-end reads Allowed values: FR
      (fwd-rev, or typical Illumina; default), RF (rev-fwd, for circularized
      inserts), or FF (fwd-fwd, same strand)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fastq-id-start</b>=<i>INT</i></dt>
  <dd class="It-tag">Starting position of identifier in FASTQ header,
      space-delimited (&gt;= 1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fastq-id-end</b>=<i>INT</i></dt>
  <dd class="It-tag">Ending position of identifier in FASTQ header,
      space-delimited (&gt;= 1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Examples:</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">@HWUSI-EAS100R:6:73:941:1973#0/1</dt>
  <dd class="It-tag">start=1, end=1 (default) =&gt; identifier is
      HWUSI-EAS100R:6:73:941:1973#0</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">@SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345
    length=36</dt>
  <dd class="It-tag">start=1, end=1 =&gt; identifier is SRR001666.1 start=2,
      end=2 =&gt; identifier is 071112_SLXA-EAS1_s_7:5:1:817:345 start=1, end=2
      =&gt; identifier is SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--force-single-end</b></dt>
  <dd class="It-tag">When multiple FASTQ files are provided on the command line,
      GSNAP assumes they are matching paired-end files. This flag treats each
      file as single-end.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--filter-chastity</b>=<i>STRING</i></dt>
  <dd class="It-tag">Skips reads marked by the Illumina chastity program.
      Expecting a string after the accession having a 'Y' after the first colon,
      like this:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">@accession 1:Y:0:CTTGTA</dt>
  <dd class="It-tag">where the 'Y' signifies filtering by chastity. Values: off
      (default), either, both. For 'either', a 'Y' on either end of a paired-end
      read will be filtered. For 'both', a 'Y' is required on both ends of a
      paired-end read (or on the only end of a single-end read).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--allow-pe-name-mismatch</b></dt>
  <dd class="It-tag">Allows accession names of reads to mismatch in paired-end
      files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gunzip</b></dt>
  <dd class="It-tag">Uncompress gzipped input files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bunzip2</b></dt>
  <dd class="It-tag">Uncompress bzip2-compressed input files</dd>
</dl>
<div class="Pp"></div>
Computation options
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Note: GSNAP has an ultrafast algorithm for calculating
      mismatches up to and including</dd>
</dl>
<div class="Pp"></div>
((readlength+2)/kmer - 2) (&quot;ultrafast mismatches&quot;). The program will
  run fastest if max-mismatches (plus suboptimal-levels) is within that value.
  Also, indels, especially end indels, take longer to compute, although the
  algorithm is still designed to be fast.
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b>, <b>--batch</b>=<i>INT</i></dt>
  <dd class="It-tag">Batch mode (default = 2)
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     Mode Offsets Positions Genome Suffix array
    <br/>
     0 see note mmap mmap mmap
    <br/>
     1 see note mmap &amp; preload mmap mmap
    <br/>
     2 see note mmap &amp; preload mmap &amp; preload mmap &amp; preload
    <br/>
     3 see note allocate mmap &amp; preload mmap &amp; preload
    <br/>
     (default) 4 see note allocate allocate mmap &amp; preload
    <br/>
     5 see note allocate allocate allocate</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Note: For a single sequence, all data structures use
    mmap</dt>
  <dd class="It-tag">If mmap not available and allocate not chosen, then will
      use fileio (very slow)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Note about offsets: Expansion of offsets can be
    controlled</dt>
  <dd class="It-tag">independently by the <b>--expand-offsets</b> flag. However,
      offsets are accessed relatively fast in this version of GSNAP.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--use-shared-memory</b>=<i>INT</i></dt>
  <dd class="It-tag">If 1 (default), then allocated memory is shared among all
      processes on this node. If 0, then each process has private allocated
      memory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--preload-shared-memory</b></dt>
  <dd class="It-tag">Load files indicated by <b>--batch</b> mode into shared
      memory for use by other GMAP/GSNAP processes on this node, and then exit.
      Ignore any input files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--unload-shared-memory</b></dt>
  <dd class="It-tag">Unload files indicated by <b>--batch</b> mode into shared
      memory, or allow them to be unloaded when existing GMAP/GSNAP processes on
      this node are finished with them. Ignore any input files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--expand-offsets</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether to expand the genomic offsets index Values: 0 (no,
      default), or 1 (yes). Expansion gives faster alignment, but requires more
      memory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>, <b>--max-mismatches</b>=<i>FLOAT</i></dt>
  <dd class="It-tag">Maximum number of mismatches allowed (if not specified,
      then defaults to the ultrafast level of
      ((readlength+index_interval-1)/kmer - 2)) (By default, the genome index
      interval is 3, but this can be changed by providing a different value for
      <b>-q</b> to gmap_build when processing the genome.)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">If specified between 0.0 and 1.0, then treated as a
    fraction</dt>
  <dd class="It-tag">of each read length. Otherwise, treated as an integral
      number of mismatches (including indel and splicing penalties) For RNA-Seq,
      you may need to increase this value slightly to align reads extending past
      the ends of an exon.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-coverage</b>=<i>FLOAT</i></dt>
  <dd class="It-tag">Minimum coverage required for an alignment. If specified
      between 0.0 and 1.0, then treated as a fraction of each read length.
      Otherwise, treated as an integral number of base pairs. Default value is
      0.0.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--query-unk-mismatch</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether to count unknown (N) characters in the query as a
      mismatch (0=no (default), 1=yes)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--genome-unk-mismatch</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether to count unknown (N) characters in the genome as a
      mismatch (0=no, 1=yes (default))</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxsearch</b>=<i>INT</i></dt>
  <dd class="It-tag">Maximum number of alignments to find (default 1000). Must
      be larger than <b>--npaths</b>, which is the number to report. Keeping
      this number large will allow for random selection among multiple
      alignments. Reducing this number can speed up the program.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b>, <b>--indel-penalty</b>=<i>INT</i></dt>
  <dd class="It-tag">Penalty for an indel (default 2). Counts against mismatches
      allowed. To find indels, make indel-penalty less than or equal to
      max-mismatches. A value &lt; 2 can lead to false positives at read
    ends</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--indel-endlength</b>=<i>INT</i></dt>
  <dd class="It-tag">Minimum length at end required for indel alignments
      (default 4)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-y</b>, <b>--max-middle-insertions</b>=<i>INT</i></dt>
  <dd class="It-tag">Maximum number of middle insertions allowed (default is
      readlength - indel-endlength)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-z</b>,
  <b>--max-middle-deletions</b>=<i>INT</i> Maximum number of middle deletions
  allowed (default 30)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Y</b>, <b>--max-end-insertions</b>=<i>INT</i></dt>
  <dd class="It-tag">Maximum number of end insertions allowed (default 3)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b>, <b>--max-end-deletions</b>=<i>INT</i></dt>
  <dd class="It-tag">Maximum number of end deletions allowed (default 6)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b>, <b>--suboptimal-levels</b>=<i>INT</i></dt>
  <dd class="It-tag">Report suboptimal hits beyond best hit (default 0) All hits
      with best score plus suboptimal-levels are reported</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b>, <b>--adapter-strip</b>=<i>STRING</i></dt>
  <dd class="It-tag">Method for removing adapters from reads. Currently allowed
      values: off, paired. Default is &quot;off&quot;. To turn on, specify
      &quot;paired&quot;, which removes adapters from paired-end reads if they
      appear to be present.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trim-mismatch-score</b>=<i>INT</i></dt>
  <dd class="It-tag">Score to use for mismatches when trimming at ends (default
      is <b>-3</b>; to turn off trimming, specify 0). Warning: turning trimming
      off will give false positive mismatches at the ends of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trim-indel-score</b>=<i>INT</i></dt>
  <dd class="It-tag">Score to use for indels when trimming at ends (default is
      <b>-2</b>; to turn off trimming, specify 0). Warning: turning trimming off
      will give false positive indels at the ends of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--end-detail</b>=<i>STRING</i></dt>
  <dd class="It-tag">Amount of alignment detail at ends of read: high, medium,
      or low (default) Note: medium detail could increase speed by 20% or so,
      but will miss some splices at the ends of reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>, <b>--snpsdir</b>=<i>STRING</i></dt>
  <dd class="It-tag">Directory for SNPs index files (created using snpindex)
      (default is location of genome index files specified using <b>-D</b> and
      <b>-d</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--use-snps</b>=<i>STRING</i></dt>
  <dd class="It-tag">Use database containing known SNPs (in &lt;STRING&gt;.iit,
      built previously using snpindex) for tolerance to SNPs</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cmetdir</b>=<i>STRING</i></dt>
  <dd class="It-tag">Directory for methylcytosine index files (created using
      cmetindex) (default is location of genome index files specified using
      <b>-D</b>, <b>-V</b>, and <b>-d</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--atoidir</b>=<i>STRING</i></dt>
  <dd class="It-tag">Directory for A-to-I RNA editing index files (created using
      atoiindex) (default is location of genome index files specified using
      <b>-D</b>, <b>-V</b>, and <b>-d</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mode</b>=<i>STRING</i></dt>
  <dd class="It-tag">Alignment mode: standard (default), cmet-stranded,
      cmet-nonstranded, atoi-stranded, atoi-nonstranded, ttoc-stranded, or
      ttoc-nonstranded. Non-standard modes requires you to have previously run
      the cmetindex or atoiindex programs (which also cover the ttoc modes) on
      the genome</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>, <b>--nthreads</b>=<i>INT</i></dt>
  <dd class="It-tag">Number of worker threads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-anchors</b>=<i>INT</i></dt>
  <dd class="It-tag">Controls number of candidate segments returned by the
      complete set algorithm Default is 10. Can be increased to higher values to
      solve alignments with evenly spaced mismatches at close distances.
      However, higher values will cause GSNAP to run more slowly. A value of
      1000, for example, slows down the program by a factor of 10 or so.
      Therefore, change this value only if absolutely necessary.</dd>
</dl>
<div class="Pp"></div>
Options for GMAP alignment within GSNAP
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gmap-mode</b>=<i>STRING</i></dt>
  <dd class="It-tag">Cases to use GMAP for complex alignments containing
      multiple splices or indels Allowed values: none, all, pairsearch,
      terminal, improve</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">(or multiple values, separated by commas).</dt>
  <dd class="It-tag">Default: all, i.e., pairsearch,terminal,improve</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trigger-score-for-gmap</b>=<i>INT</i></dt>
  <dd class="It-tag">Try GMAP pairsearch on nearby genomic regions if best score
      (the total of both ends if paired-end) exceeds this value (default 5)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gmap-min-match-length</b>=<i>INT</i></dt>
  <dd class="It-tag">Keep GMAP hit only if it has this many consecutive matches
      (default 20)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gmap-allowance</b>=<i>INT</i></dt>
  <dd class="It-tag">Extra mismatch/indel score allowed for GMAP alignments
      (default 3)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-gmap-pairsearch</b>=<i>INT</i></dt>
  <dd class="It-tag">Perform GMAP pairsearch on nearby genomic regions up to
      this many many candidate ends (default 50). Requires pairsearch in
      <b>--gmap-mode</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-gmap-terminal</b>=<i>INT</i></dt>
  <dd class="It-tag">Perform GMAP terminal on nearby genomic regions up to this
      many candidate ends (default 50). Requires terminal in
    <b>--gmap-mode</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-gmap-improvement</b>=<i>INT</i></dt>
  <dd class="It-tag">Perform GMAP improvement on nearby genomic regions up to
      this many candidate ends (default 5). Requires improve in
      <b>--gmap-mode</b></dd>
</dl>
<div class="Pp"></div>
Splicing options for DNA-Seq
<dl class="Bl-tag">
  <dt class="It-tag"><b>--find-dna-chimeras</b>=<i>INT</i></dt>
  <dd class="It-tag">Look for distant splicing in DNA-Seq data (0=no (default),
      1=yes) Automatically inactivated for RNA-Seq data if <b>-N</b> or
      <b>-s</b> are specified)</dd>
</dl>
<div class="Pp"></div>
Splicing options for RNA-Seq
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b>, <b>--novelsplicing</b>=<i>INT</i></dt>
  <dd class="It-tag">Look for novel splicing (0=no (default), 1=yes)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--splicingdir</b>=<i>STRING</i></dt>
  <dd class="It-tag">Directory for splicing involving known sites or known
      introns, as specified by the <b>-s</b> or <b>--use-splicing</b> flag
      (default is directory computed from <b>-D</b> and <b>-d</b> flags). Note:
      can just give full pathname to the <b>-s</b> flag instead.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>, <b>--use-splicing</b>=<i>STRING</i></dt>
  <dd class="It-tag">Look for splicing involving known sites or known introns
      (in &lt;STRING&gt;.iit), at short or long distances See README
      instructions for the distinction between known sites and known
    introns</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ambig-splice-noclip</b></dt>
  <dd class="It-tag">For ambiguous known splicing at ends of the read, do not
      clip at the splice site, but extend instead into the intron. This flag
      makes sense only if you provide the <b>--use-splicing</b> flag, and you
      are trying to eliminate all soft clipping with
      <b>--trim-mismatch-score</b>= <i>0</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b>, <b>--localsplicedist</b>=<i>INT</i></dt>
  <dd class="It-tag">Definition of local novel splicing event (default
    200000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--novelend-splicedist</b>=<i>INT</i></dt>
  <dd class="It-tag">Distance to look for novel splices at the ends of reads
      (default 50000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>, <b>--local-splice-penalty</b>=<i>INT</i></dt>
  <dd class="It-tag">Penalty for a local splice (default 0). Counts against
      mismatches allowed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b>, <b>--distant-splice-penalty</b>=<i>INT</i></dt>
  <dd class="It-tag">Penalty for a distant splice (default 1). A distant splice
      is one where the intron length exceeds the value of <b>-w</b>, or
      <b>--localsplicedist</b>, or is an inversion, scramble, or translocation
      between two different chromosomes Counts against mismatches allowed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-K</b>,
    <b>--distant-splice-endlength</b>=<i>INT</i></dt>
  <dd class="It-tag">Minimum length at end required for distant spliced
      alignments (default 20, min allowed is the value of <b>-k</b>, or kmer
      size)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b>,
    <b>--shortend-splice-endlength</b>=<i>INT</i></dt>
  <dd class="It-tag">Minimum length at end required for short-end spliced
      alignments (default 2, but unless known splice sites are provided with the
      <b>-s</b> flag, GSNAP may still need the end length to be the value of
      <b>-k</b>, or kmer size to find a given splice</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--distant-splice-identity</b>=<i>FLOAT</i></dt>
  <dd class="It-tag">Minimum identity at end required for distant spliced
      alignments (default 0.95)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--antistranded-penalty</b>=<i>INT</i></dt>
  <dd class="It-tag">(Not currently implemented, since it leads to poor results)
      Penalty for antistranded splicing when using stranded RNA-Seq protocols. A
      positive value, such as 1, expects antisense on the first read and sense
      on the second read. Default is 0, which treats sense and antisense equally
      well</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--merge-distant-samechr</b></dt>
  <dd class="It-tag">Report distant splices on the same chromosome as a single
      splice, if possible. Will produce a single SAM line instead of two SAM
      lines, which is also done for translocations, inversions, and scramble
      events</dd>
</dl>
<div class="Pp"></div>
Options for paired-end reads
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pairmax-dna</b>=<i>INT</i></dt>
  <dd class="It-tag">Max total genomic length for DNA-Seq paired reads, or other
      reads without splicing (default 1000). Used if <b>-N</b> or <b>-s</b> is
      not specified. This value is also used for circular chromosomes when
      splicing in linear chromosomes is allowed</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pairmax-rna</b>=<i>INT</i></dt>
  <dd class="It-tag">Max total genomic length for RNA-Seq paired reads, or other
      reads that could have a splice (default 200000). Used if <b>-N</b> or
      <b>-s</b> is specified. Should probably match the value for <b>-w</b>,
      <b>--localsplicedist</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pairexpect</b>=<i>INT</i></dt>
  <dd class="It-tag">Expected paired-end length, used for calling splices in
      medial part of paired-end reads (default 200). Was turned off in previous
      versions, but reinstated.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pairdev</b>=<i>INT</i></dt>
  <dd class="It-tag">Allowable deviation from expected paired-end length, used
      for calling splices in medial part of paired-end reads (default 100). Was
      turned off in previous versions, but reinstated.</dd>
</dl>
<div class="Pp"></div>
Options for quality scores
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quality-protocol</b>=<i>STRING</i></dt>
  <dd class="It-tag">Protocol for input quality scores. Allowed values: illumina
      (ASCII 64-126) (equivalent to <b>-J</b> 64 <b>-j</b> <b>-31</b>) sanger
      (ASCII 33-126) (equivalent to <b>-J</b> 33 <b>-j</b> 0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Default is sanger (no quality print shift)</dt>
  <dd class="It-tag">SAM output files should have quality scores in sanger
      protocol</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Or you can customize this behavior with these flags:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-J</b>, <b>--quality-zero-score</b>=<i>INT</i></dt>
  <dd class="It-tag">FASTQ quality scores are zero at this ASCII value (default
      is 33 for sanger protocol; for Illumina, select 64)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-j</b>, <b>--quality-print-shift</b>=<i>INT</i></dt>
  <dd class="It-tag">Shift FASTQ quality scores by this amount in output
      (default is 0 for sanger protocol; to change Illumina input to Sanger
      output, select <b>-31</b>)</dd>
</dl>
<div class="Pp"></div>
Output options
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b>, <b>--npaths</b>=<i>INT</i></dt>
  <dd class="It-tag">Maximum number of paths to print (default 100).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b>, <b>--quiet-if-excessive</b></dt>
  <dd class="It-tag">If more than maximum number of paths are found, then
      nothing is printed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-O</b>, <b>--ordered</b></dt>
  <dd class="It-tag">Print output in same order as input (relevant only if there
      is more than one worker thread)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--show-refdiff</b></dt>
  <dd class="It-tag">For GSNAP output in SNP-tolerant alignment, shows all
      differences relative to the reference genome as lower case (otherwise, it
      shows all differences relative to both the reference and alternate
    genome)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--clip-overlap</b></dt>
  <dd class="It-tag">For paired-end reads whose alignments overlap, clip the
      overlapping region.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--merge-overlap</b></dt>
  <dd class="It-tag">For paired-end reads whose alignments overlap, merge the
      two ends into a single end (beta implementation)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--print-snps</b></dt>
  <dd class="It-tag">Print detailed information about SNPs in reads (works only
      if <b>-v</b> also selected) (not fully implemented yet)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--failsonly</b></dt>
  <dd class="It-tag">Print only failed alignments, those with no results</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nofails</b></dt>
  <dd class="It-tag">Exclude printing of failed alignments</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b>, <b>--format</b>=<i>STRING</i></dt>
  <dd class="It-tag">Another format type, other than default. Currently
      implemented: sam, m8 (BLAST tabular format)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--split-output</b>=<i>STRING</i></dt>
  <dd class="It-tag">Basename for multiple-file output, separately for
      nomapping, halfmapping_uniq, halfmapping_mult, unpaired_uniq,
      unpaired_mult, paired_uniq, paired_mult, concordant_uniq, and
      concordant_mult results</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>, <b>--output-file</b>=<i>STRING</i></dt>
  <dd class="It-tag">File name for a single stream of output results.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--failed-input</b>=<i>STRING</i></dt>
  <dd class="It-tag">Print completely failed alignments as input FASTA or FASTQ
      format, to the given file, appending .1 or .2, for paired-end data. If the
      <b>--split-output</b> flag is also given, this file is generated in
      addition to the output in the .nomapping file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--append-output</b></dt>
  <dd class="It-tag">When <b>--split-output</b> or <b>--failed-input</b> is
      given, this flag will append output to the existing files. Otherwise, the
      default is to create new files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--order-among-best</b>=<i>STRING</i></dt>
  <dd class="It-tag">Among alignments tied with the best score, order those
      alignments in this order. Allowed values: genomic, random (default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--output-buffer-size</b>=<i>INT</i></dt>
  <dd class="It-tag">Buffer size, in queries, for output thread (default 1000).
      When the number of results to be printed exceeds this size, the worker
      threads are halted until the backlog is cleared</dd>
</dl>
<div class="Pp"></div>
Options for SAM output
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-sam-headers</b></dt>
  <dd class="It-tag">Do not print headers beginning with '@'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--add-paired-nomappers</b></dt>
  <dd class="It-tag">Add nomapper lines as needed to make all paired-end results
      alternate between first end and second end</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--paired-flag-means-concordant</b>=<i>INT</i></dt>
  <dd class="It-tag">Whether the paired bit in the SAM flags means concordant
      only (1) or paired plus concordant (0, default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-headers-batch</b>=<i>INT</i></dt>
  <dd class="It-tag">Print headers only for this batch, as specified by
      <b>-q</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-use-0M</b></dt>
  <dd class="It-tag">Insert 0M in CIGAR between adjacent insertions and
      deletions Required by Picard, but can cause errors in other tools</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-multiple-primaries</b></dt>
  <dd class="It-tag">Allows multiple alignments to be marked as primary if they
      have equally good mapping scores</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--force-xs-dir</b></dt>
  <dd class="It-tag">For RNA-Seq alignments, disallows XS:A:? when the sense
      direction is unclear, and replaces this value arbitrarily with XS:A:+. May
      be useful for some programs, such as Cufflinks, that cannot handle XS:A:?.
      However, if you use this flag, the reported value of XS:A:+ in these cases
      will not be meaningful.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--md-lowercase-snp</b></dt>
  <dd class="It-tag">In MD string, when known SNPs are given by the <b>-v</b>
      flag, prints difference nucleotides as lower-case when they, differ from
      reference but match a known alternate allele</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--extend-soft-clips</b></dt>
  <dd class="It-tag">Extends alignments through soft clipped regions</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--action-if-cigar-error</b></dt>
  <dd class="It-tag">Action to take if there is a disagreement between CIGAR
      length and sequence length Allowed values: ignore, warning, noprint
      (default), abort</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read-group-id</b>=<i>STRING</i></dt>
  <dd class="It-tag">Value to put into read-group id (RG-ID) field</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read-group-name</b>=<i>STRING</i></dt>
  <dd class="It-tag">Value to put into read-group name (RG-SM) field</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read-group-library</b>=<i>STRING</i></dt>
  <dd class="It-tag">Value to put into read-group library (RG-LB) field</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read-group-platform</b>=<i>STRING</i></dt>
  <dd class="It-tag">Value to put into read-group library (RG-PL) field</dd>
</dl>
<div class="Pp"></div>
Help options
<dl class="Bl-tag">
  <dt class="It-tag"><b>--check</b></dt>
  <dd class="It-tag">Check compiler assumptions</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">Show version</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">Show this help message
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Other tools of GMAP suite are located in /usr/lib/gmap</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">gsnap 2017-01-14-1</td>
  </tr>
</table>
</body>
</html>
