<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:40 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GSNAP(1) User Commands GSNAP(1)</p>

<p style="margin-top: 1em">NAME <br>
gsnap - Genomic Short-read Nucleotide Alignment Program</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gsnap [OPTIONS...] &lt;FASTA file&gt;, or cat &lt;FASTA
file&gt; | gmap [OPTIONS...]</p>

<p style="margin-top: 1em">OPTIONS <br>
Input options (must include -d) <br>
-D, --dir=directory <br>
Genome directory. Default (as specified by --with-gmapdb to
the configure program) is /var/cache/gmap</p>

<p style="margin-top: 1em">-d, --db=STRING <br>
Genome database</p>

<p style="margin-top: 1em">--use-sarray=INT <br>
Whether to use a suffix array, which will give increased
speed. Allowed values: 0 (no), 1 (yes, plus GSNAP/GMAP
algorithm, default), or 2 (yes, and use only suffix array
<br>
algorithm). Note that suffix arrays will bias against SNP
alleles in SNP-tolerant alignment.</p>

<p style="margin-top: 1em">-k, --kmer=INT <br>
kmer size to use in genome database (allowed values: 16 or
less) If not specified, the program will find the highest
available kmer size in the genome database</p>

<p style="margin-top: 1em">--sampling=INT <br>
Sampling to use in genome database. If not specified, the
program will find the smallest available sampling value in
the genome database within selected k-mer size</p>

<p style="margin-top: 1em">-q, --part=INT/INT <br>
Process only the i-th out of every n sequences e.g., 0/100
or 99/100 (useful for distributing jobs to a computer
farm).</p>

<p style="margin-top: 1em">--input-buffer-size=INT <br>
Size of input buffer (program reads this many sequences at a
time for efficiency) (default 1000)</p>

<p style="margin-top: 1em">--barcode-length=INT <br>
Amount of barcode to remove from start of read (default
0)</p>

<p style="margin-top: 1em">--orientation=STRING <br>
Orientation of paired-end reads Allowed values: FR (fwd-rev,
or typical Illumina; default), RF (rev-fwd, for circularized
inserts), or FF (fwd-fwd, same strand)</p>

<p style="margin-top: 1em">--fastq-id-start=INT <br>
Starting position of identifier in FASTQ header,
space-delimited (&gt;= 1)</p>

<p style="margin-top: 1em">--fastq-id-end=INT <br>
Ending position of identifier in FASTQ header,
space-delimited (&gt;= 1)</p>

<p style="margin-top: 1em">Examples:</p>


<p style="margin-top: 1em">@HWUSI-EAS100R:6:73:941:1973#0/1
<br>
start=1, end=1 (default) =&gt; identifier is
HWUSI-EAS100R:6:73:941:1973#0</p>

<p style="margin-top: 1em">@SRR001666.1
071112_SLXA-EAS1_s_7:5:1:817:345 length=36 <br>
start=1, end=1 =&gt; identifier is SRR001666.1 start=2,
end=2 =&gt; identifier is 071112_SLXA-EAS1_s_7:5:1:817:345
start=1, end=2 =&gt; identifier is SRR001666.1 <br>
071112_SLXA-EAS1_s_7:5:1:817:345</p>

<p style="margin-top: 1em">--force-single-end <br>
When multiple FASTQ files are provided on the command line,
GSNAP assumes they are matching paired-end files. This flag
treats each file as single-end.</p>

<p style="margin-top: 1em">--filter-chastity=STRING <br>
Skips reads marked by the Illumina chastity program.
Expecting a string after the accession having a
&rsquo;Y&rsquo; after the first colon, like this:</p>

<p style="margin-top: 1em">@accession 1:Y:0:CTTGTA <br>
where the &rsquo;Y&rsquo; signifies filtering by chastity.
Values: off (default), either, both. For
&rsquo;either&rsquo;, a &rsquo;Y&rsquo; on either end of a
paired-end read will be filtered. For &rsquo;both&rsquo;,
<br>
a &rsquo;Y&rsquo; is required on both ends of a paired-end
read (or on the only end of a single-end read).</p>

<p style="margin-top: 1em">--allow-pe-name-mismatch <br>
Allows accession names of reads to mismatch in paired-end
files</p>

<p style="margin-top: 1em">--gunzip <br>
Uncompress gzipped input files</p>

<p style="margin-top: 1em">--bunzip2 <br>
Uncompress bzip2-compressed input files</p>

<p style="margin-top: 1em">Computation options</p>

<p style="margin-top: 1em">Note: GSNAP has an ultrafast
algorithm for calculating mismatches up to and including</p>

<p style="margin-top: 1em">((readlength+2)/kmer - 2)
(&quot;ultrafast mismatches&quot;). The program will run
fastest if max-mismatches (plus suboptimal-levels) is within
that value. Also, indels, especially end <br>
indels, take longer to compute, although the algorithm is
still designed to be fast.</p>

<p style="margin-top: 1em">-B, --batch=INT <br>
Batch mode (default = 2)</p>

<p style="margin-top: 1em">Mode Offsets Positions Genome
Suffix array <br>
0 see note mmap mmap mmap <br>
1 see note mmap &amp; preload mmap mmap <br>
2 see note mmap &amp; preload mmap &amp; preload mmap &amp;
preload <br>
3 see note allocate mmap &amp; preload mmap &amp; preload
<br>
(default) 4 see note allocate allocate mmap &amp; preload
<br>
5 see note allocate allocate allocate</p>

<p style="margin-top: 1em">Note: For a single sequence, all
data structures use mmap <br>
If mmap not available and allocate not chosen, then will use
fileio (very slow)</p>

<p style="margin-top: 1em">Note about offsets: Expansion of
offsets can be controlled <br>
independently by the --expand-offsets flag. However, offsets
are accessed relatively fast in this version of GSNAP.</p>

<p style="margin-top: 1em">--use-shared-memory=INT <br>
If 1 (default), then allocated memory is shared among all
processes on this node. If 0, then each process has private
allocated memory</p>

<p style="margin-top: 1em">--preload-shared-memory <br>
Load files indicated by --batch mode into shared memory for
use by other GMAP/GSNAP processes on this node, and then
exit. Ignore any input files.</p>

<p style="margin-top: 1em">--unload-shared-memory <br>
Unload files indicated by --batch mode into shared memory,
or allow them to be unloaded when existing GMAP/GSNAP
processes on this node are finished with them. Ignore any
<br>
input files.</p>

<p style="margin-top: 1em">--expand-offsets=INT <br>
Whether to expand the genomic offsets index Values: 0 (no,
default), or 1 (yes). Expansion gives faster alignment, but
requires more memory</p>

<p style="margin-top: 1em">-m, --max-mismatches=FLOAT <br>
Maximum number of mismatches allowed (if not specified, then
defaults to the ultrafast level of
((readlength+index_interval-1)/kmer - 2)) (By default, the
genome index <br>
interval is 3, but this can be changed by providing a
different value for -q to gmap_build when processing the
genome.)</p>

<p style="margin-top: 1em">If specified between 0.0 and
1.0, then treated as a fraction <br>
of each read length. Otherwise, treated as an integral
number of mismatches (including indel and splicing
penalties) For RNA-Seq, you may need to increase this value
<br>
slightly to align reads extending past the ends of an
exon.</p>

<p style="margin-top: 1em">--min-coverage=FLOAT <br>
Minimum coverage required for an alignment. If specified
between 0.0 and 1.0, then treated as a fraction of each read
length. Otherwise, treated as an integral number of <br>
base pairs. Default value is 0.0.</p>

<p style="margin-top: 1em">--query-unk-mismatch=INT <br>
Whether to count unknown (N) characters in the query as a
mismatch (0=no (default), 1=yes)</p>

<p style="margin-top: 1em">--genome-unk-mismatch=INT <br>
Whether to count unknown (N) characters in the genome as a
mismatch (0=no, 1=yes (default))</p>

<p style="margin-top: 1em">--maxsearch=INT <br>
Maximum number of alignments to find (default 1000). Must be
larger than --npaths, which is the number to report. Keeping
this number large will allow for random selec&acirc; <br>
tion among multiple alignments. Reducing this number can
speed up the program.</p>

<p style="margin-top: 1em">-i, --indel-penalty=INT <br>
Penalty for an indel (default 2). Counts against mismatches
allowed. To find indels, make indel-penalty less than or
equal to max-mismatches. A value &lt; 2 can lead to <br>
false positives at read ends</p>

<p style="margin-top: 1em">--indel-endlength=INT <br>
Minimum length at end required for indel alignments (default
4)</p>

<p style="margin-top: 1em">-y, --max-middle-insertions=INT
<br>
Maximum number of middle insertions allowed (default is
readlength - indel-endlength)</p>

<p style="margin-top: 1em">-z, --max-middle-deletions=INT
Maximum number of middle deletions allowed (default 30)</p>

<p style="margin-top: 1em">-Y, --max-end-insertions=INT
<br>
Maximum number of end insertions allowed (default 3)</p>

<p style="margin-top: 1em">-Z, --max-end-deletions=INT <br>
Maximum number of end deletions allowed (default 6)</p>

<p style="margin-top: 1em">-M, --suboptimal-levels=INT <br>
Report suboptimal hits beyond best hit (default 0) All hits
with best score plus suboptimal-levels are reported</p>

<p style="margin-top: 1em">-a, --adapter-strip=STRING <br>
Method for removing adapters from reads. Currently allowed
values: off, paired. Default is &quot;off&quot;. To turn on,
specify &quot;paired&quot;, which removes adapters from
paired-end <br>
reads if they appear to be present.</p>

<p style="margin-top: 1em">--trim-mismatch-score=INT <br>
Score to use for mismatches when trimming at ends (default
is -3; to turn off trimming, specify 0). Warning: turning
trimming off will give false positive mismatches at <br>
the ends of reads</p>

<p style="margin-top: 1em">--trim-indel-score=INT <br>
Score to use for indels when trimming at ends (default is
-2; to turn off trimming, specify 0). Warning: turning
trimming off will give false positive indels at the ends
<br>
of reads</p>

<p style="margin-top: 1em">--end-detail=STRING <br>
Amount of alignment detail at ends of read: high, medium, or
low (default) Note: medium detail could increase speed by
20% or so, but will miss some splices at the ends of <br>
reads</p>

<p style="margin-top: 1em">-V, --snpsdir=STRING <br>
Directory for SNPs index files (created using snpindex)
(default is location of genome index files specified using
-D and -d)</p>

<p style="margin-top: 1em">-v, --use-snps=STRING <br>
Use database containing known SNPs (in &lt;STRING&gt;.iit,
built previously using snpindex) for tolerance to SNPs</p>

<p style="margin-top: 1em">--cmetdir=STRING <br>
Directory for methylcytosine index files (created using
cmetindex) (default is location of genome index files
specified using -D, -V, and -d)</p>

<p style="margin-top: 1em">--atoidir=STRING <br>
Directory for A-to-I RNA editing index files (created using
atoiindex) (default is location of genome index files
specified using -D, -V, and -d)</p>

<p style="margin-top: 1em">--mode=STRING <br>
Alignment mode: standard (default), cmet-stranded,
cmet-nonstranded, atoi-stranded, atoi-nonstranded,
ttoc-stranded, or ttoc-nonstranded. Non-standard modes
requires you <br>
to have previously run the cmetindex or atoiindex programs
(which also cover the ttoc modes) on the genome</p>

<p style="margin-top: 1em">-t, --nthreads=INT <br>
Number of worker threads</p>

<p style="margin-top: 1em">--max-anchors=INT <br>
Controls number of candidate segments returned by the
complete set algorithm Default is 10. Can be increased to
higher values to solve alignments with evenly spaced
mis&acirc; <br>
matches at close distances. However, higher values will
cause GSNAP to run more slowly. A value of 1000, for
example, slows down the program by a factor of 10 or so.
<br>
Therefore, change this value only if absolutely
necessary.</p>

<p style="margin-top: 1em">Options for GMAP alignment
within GSNAP</p>

<p style="margin-top: 1em">--gmap-mode=STRING <br>
Cases to use GMAP for complex alignments containing multiple
splices or indels Allowed values: none, all, pairsearch,
terminal, improve</p>

<p style="margin-top: 1em">(or multiple values, separated
by commas). <br>
Default: all, i.e., pairsearch,terminal,improve</p>

<p style="margin-top: 1em">--trigger-score-for-gmap=INT
<br>
Try GMAP pairsearch on nearby genomic regions if best score
(the total of both ends if paired-end) exceeds this value
(default 5)</p>

<p style="margin-top: 1em">--gmap-min-match-length=INT <br>
Keep GMAP hit only if it has this many consecutive matches
(default 20)</p>

<p style="margin-top: 1em">--gmap-allowance=INT <br>
Extra mismatch/indel score allowed for GMAP alignments
(default 3)</p>

<p style="margin-top: 1em">--max-gmap-pairsearch=INT <br>
Perform GMAP pairsearch on nearby genomic regions up to this
many many candidate ends (default 50). Requires pairsearch
in --gmap-mode</p>

<p style="margin-top: 1em">--max-gmap-terminal=INT <br>
Perform GMAP terminal on nearby genomic regions up to this
many candidate ends (default 50). Requires terminal in
--gmap-mode</p>

<p style="margin-top: 1em">--max-gmap-improvement=INT <br>
Perform GMAP improvement on nearby genomic regions up to
this many candidate ends (default 5). Requires improve in
--gmap-mode</p>

<p style="margin-top: 1em">Splicing options for DNA-Seq</p>

<p style="margin-top: 1em">--find-dna-chimeras=INT <br>
Look for distant splicing in DNA-Seq data (0=no (default),
1=yes) Automatically inactivated for RNA-Seq data if -N or
-s are specified)</p>

<p style="margin-top: 1em">Splicing options for RNA-Seq</p>

<p style="margin-top: 1em">-N, --novelsplicing=INT <br>
Look for novel splicing (0=no (default), 1=yes)</p>

<p style="margin-top: 1em">--splicingdir=STRING <br>
Directory for splicing involving known sites or known
introns, as specified by the -s or --use-splicing flag
(default is directory computed from -D and -d flags). Note:
<br>
can just give full pathname to the -s flag instead.</p>

<p style="margin-top: 1em">-s, --use-splicing=STRING <br>
Look for splicing involving known sites or known introns (in
&lt;STRING&gt;.iit), at short or long distances See README
instructions for the distinction between known sites and
<br>
known introns</p>

<p style="margin-top: 1em">--ambig-splice-noclip <br>
For ambiguous known splicing at ends of the read, do not
clip at the splice site, but extend instead into the intron.
This flag makes sense only if you provide the <br>
--use-splicing flag, and you are trying to eliminate all
soft clipping with --trim-mismatch-score=0</p>

<p style="margin-top: 1em">-w, --localsplicedist=INT <br>
Definition of local novel splicing event (default
200000)</p>

<p style="margin-top: 1em">--novelend-splicedist=INT <br>
Distance to look for novel splices at the ends of reads
(default 50000)</p>

<p style="margin-top: 1em">-e, --local-splice-penalty=INT
<br>
Penalty for a local splice (default 0). Counts against
mismatches allowed</p>

<p style="margin-top: 1em">-E, --distant-splice-penalty=INT
<br>
Penalty for a distant splice (default 1). A distant splice
is one where the intron length exceeds the value of -w, or
--localsplicedist, or is an inversion, scramble, or <br>
translocation between two different chromosomes Counts
against mismatches allowed</p>

<p style="margin-top: 1em">-K,
--distant-splice-endlength=INT <br>
Minimum length at end required for distant spliced
alignments (default 20, min allowed is the value of -k, or
kmer size)</p>

<p style="margin-top: 1em">-l,
--shortend-splice-endlength=INT <br>
Minimum length at end required for short-end spliced
alignments (default 2, but unless known splice sites are
provided with the -s flag, GSNAP may still need the end <br>
length to be the value of -k, or kmer size to find a given
splice</p>

<p style="margin-top: 1em">--distant-splice-identity=FLOAT
<br>
Minimum identity at end required for distant spliced
alignments (default 0.95)</p>

<p style="margin-top: 1em">--antistranded-penalty=INT <br>
(Not currently implemented, since it leads to poor results)
Penalty for antistranded splicing when using stranded
RNA-Seq protocols. A positive value, such as 1, expects <br>
antisense on the first read and sense on the second read.
Default is 0, which treats sense and antisense equally
well</p>

<p style="margin-top: 1em">--merge-distant-samechr <br>
Report distant splices on the same chromosome as a single
splice, if possible. Will produce a single SAM line instead
of two SAM lines, which is also done for transloca&acirc;
<br>
tions, inversions, and scramble events</p>

<p style="margin-top: 1em">Options for paired-end reads</p>

<p style="margin-top: 1em">--pairmax-dna=INT <br>
Max total genomic length for DNA-Seq paired reads, or other
reads without splicing (default 1000). Used if -N or -s is
not specified. This value is also used for circu&acirc; <br>
lar chromosomes when splicing in linear chromosomes is
allowed</p>

<p style="margin-top: 1em">--pairmax-rna=INT <br>
Max total genomic length for RNA-Seq paired reads, or other
reads that could have a splice (default 200000). Used if -N
or -s is specified. Should probably match the <br>
value for -w, --localsplicedist.</p>

<p style="margin-top: 1em">--pairexpect=INT <br>
Expected paired-end length, used for calling splices in
medial part of paired-end reads (default 200). Was turned
off in previous versions, but reinstated.</p>

<p style="margin-top: 1em">--pairdev=INT <br>
Allowable deviation from expected paired-end length, used
for calling splices in medial part of paired-end reads
(default 100). Was turned off in previous versions, but <br>
reinstated.</p>

<p style="margin-top: 1em">Options for quality scores</p>

<p style="margin-top: 1em">--quality-protocol=STRING <br>
Protocol for input quality scores. Allowed values: illumina
(ASCII 64-126) (equivalent to -J 64 -j -31) sanger (ASCII
33-126) (equivalent to -J 33 -j 0)</p>

<p style="margin-top: 1em">Default is sanger (no quality
print shift) <br>
SAM output files should have quality scores in sanger
protocol</p>

<p style="margin-top: 1em">Or you can customize this
behavior with these flags:</p>

<p style="margin-top: 1em">-J, --quality-zero-score=INT
<br>
FASTQ quality scores are zero at this ASCII value (default
is 33 for sanger protocol; for Illumina, select 64)</p>

<p style="margin-top: 1em">-j, --quality-print-shift=INT
<br>
Shift FASTQ quality scores by this amount in output (default
is 0 for sanger protocol; to change Illumina input to Sanger
output, select -31)</p>

<p style="margin-top: 1em">Output options</p>

<p style="margin-top: 1em">-n, --npaths=INT <br>
Maximum number of paths to print (default 100).</p>

<p style="margin-top: 1em">-Q, --quiet-if-excessive <br>
If more than maximum number of paths are found, then nothing
is printed.</p>

<p style="margin-top: 1em">-O, --ordered <br>
Print output in same order as input (relevant only if there
is more than one worker thread)</p>

<p style="margin-top: 1em">--show-refdiff <br>
For GSNAP output in SNP-tolerant alignment, shows all
differences relative to the reference genome as lower case
(otherwise, it shows all differences relative to both the
<br>
reference and alternate genome)</p>

<p style="margin-top: 1em">--clip-overlap <br>
For paired-end reads whose alignments overlap, clip the
overlapping region.</p>

<p style="margin-top: 1em">--merge-overlap <br>
For paired-end reads whose alignments overlap, merge the two
ends into a single end (beta implementation)</p>

<p style="margin-top: 1em">--print-snps <br>
Print detailed information about SNPs in reads (works only
if -v also selected) (not fully implemented yet)</p>

<p style="margin-top: 1em">--failsonly <br>
Print only failed alignments, those with no results</p>

<p style="margin-top: 1em">--nofails <br>
Exclude printing of failed alignments</p>

<p style="margin-top: 1em">-A, --format=STRING <br>
Another format type, other than default. Currently
implemented: sam, m8 (BLAST tabular format)</p>

<p style="margin-top: 1em">--split-output=STRING <br>
Basename for multiple-file output, separately for nomapping,
halfmapping_uniq, halfmapping_mult, unpaired_uniq,
unpaired_mult, paired_uniq, paired_mult, concordant_uniq,
<br>
and concordant_mult results</p>

<p style="margin-top: 1em">-o, --output-file=STRING <br>
File name for a single stream of output results.</p>

<p style="margin-top: 1em">--failed-input=STRING <br>
Print completely failed alignments as input FASTA or FASTQ
format, to the given file, appending .1 or .2, for
paired-end data. If the --split-output flag is also given,
<br>
this file is generated in addition to the output in the
.nomapping file.</p>

<p style="margin-top: 1em">--append-output <br>
When --split-output or --failed-input is given, this flag
will append output to the existing files. Otherwise, the
default is to create new files.</p>

<p style="margin-top: 1em">--order-among-best=STRING <br>
Among alignments tied with the best score, order those
alignments in this order. Allowed values: genomic, random
(default)</p>

<p style="margin-top: 1em">--output-buffer-size=INT <br>
Buffer size, in queries, for output thread (default 1000).
When the number of results to be printed exceeds this size,
the worker threads are halted until the backlog is <br>
cleared</p>

<p style="margin-top: 1em">Options for SAM output</p>

<p style="margin-top: 1em">--no-sam-headers <br>
Do not print headers beginning with &rsquo;@&rsquo;</p>

<p style="margin-top: 1em">--add-paired-nomappers <br>
Add nomapper lines as needed to make all paired-end results
alternate between first end and second end</p>


<p style="margin-top: 1em">--paired-flag-means-concordant=INT
<br>
Whether the paired bit in the SAM flags means concordant
only (1) or paired plus concordant (0, default)</p>

<p style="margin-top: 1em">--sam-headers-batch=INT <br>
Print headers only for this batch, as specified by -q</p>

<p style="margin-top: 1em">--sam-use-0M <br>
Insert 0M in CIGAR between adjacent insertions and deletions
Required by Picard, but can cause errors in other tools</p>

<p style="margin-top: 1em">--sam-multiple-primaries <br>
Allows multiple alignments to be marked as primary if they
have equally good mapping scores</p>

<p style="margin-top: 1em">--force-xs-dir <br>
For RNA-Seq alignments, disallows XS:A:? when the sense
direction is unclear, and replaces this value arbitrarily
with XS:A:+. May be useful for some programs, such as <br>
Cufflinks, that cannot handle XS:A:?. However, if you use
this flag, the reported value of XS:A:+ in these cases will
not be meaningful.</p>

<p style="margin-top: 1em">--md-lowercase-snp <br>
In MD string, when known SNPs are given by the -v flag,
prints difference nucleotides as lower-case when they,
differ from reference but match a known alternate allele</p>

<p style="margin-top: 1em">--extend-soft-clips <br>
Extends alignments through soft clipped regions</p>

<p style="margin-top: 1em">--action-if-cigar-error <br>
Action to take if there is a disagreement between CIGAR
length and sequence length Allowed values: ignore, warning,
noprint (default), abort</p>

<p style="margin-top: 1em">--read-group-id=STRING <br>
Value to put into read-group id (RG-ID) field</p>

<p style="margin-top: 1em">--read-group-name=STRING <br>
Value to put into read-group name (RG-SM) field</p>

<p style="margin-top: 1em">--read-group-library=STRING <br>
Value to put into read-group library (RG-LB) field</p>

<p style="margin-top: 1em">--read-group-platform=STRING
<br>
Value to put into read-group library (RG-PL) field</p>

<p style="margin-top: 1em">Help options</p>

<p style="margin-top: 1em">--check <br>
Check compiler assumptions</p>

<p style="margin-top: 1em">--version <br>
Show version</p>

<p style="margin-top: 1em">--help Show this help
message</p>

<p style="margin-top: 1em">Other tools of GMAP suite are
located in /usr/lib/gmap</p>

<p style="margin-top: 1em">gsnap 2017-01-14-1 January 2017
GSNAP(1)</p>
<hr>
</body>
</html>
