<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-CDS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-CDS(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-CDS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-cds - Add CDS (coding sequence) features to exon features given in GFF3 file.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt cds</b> [option ...] [GFF3_file]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-minorflen</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">set the minimum length an open reading frame
  (ORF) must have to be added as a CDS feature (measured in amino acids)
  (default: 64)</div>
<div class="Pp"></div>
<b>-startcodon</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">require than an ORF must begin with a start
  codon (default: no)</div>
<div class="Pp"></div>
<b>-finalstopcodon</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">require that the final ORF must end with a
  stop codon (default: no)</div>
<div class="Pp"></div>
<b>-seqfile</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">set the sequence file from which to take the
  sequences (default: undefined)</div>
<div class="Pp"></div>
<b>-encseq</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">set the encoded sequence indexname from which
  to take the sequences (default: undefined)</div>
<div class="Pp"></div>
<b>-seqfiles</b>
<div style="margin-left: 4.00ex;">set the sequence files from which to extract
  the features use <i>--</i> to terminate the list of sequence files</div>
<div class="Pp"></div>
<b>-matchdesc</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">search the sequence descriptions from the
  input files for the desired sequence IDs (in GFF3), reporting the first match
  (default: no)</div>
<div class="Pp"></div>
<b>-matchdescstart</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">exactly match the sequence descriptions from
  the input files for the desired sequence IDs (in GFF3) from the beginning to
  the first whitespace (default: no)</div>
<div class="Pp"></div>
<b>-usedesc</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">use sequence descriptions to map the sequence
  IDs (in GFF3) to actual sequence entries. If a description contains a sequence
  range (e.g., III:1000001..2000000), the first part is used as sequence ID (
  <i>III</i>) and the first range position as offset (<i>1000001</i>) (default:
  no)</div>
<div class="Pp"></div>
<b>-regionmapping</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">set file containing sequence-region to
  sequence file mapping (default: undefined)</div>
<div class="Pp"></div>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">be verbose (default: no)</div>
<div class="Pp"></div>
<b>-o</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">redirect output to specified file (default:
  undefined)</div>
<div class="Pp"></div>
<b>-gzip</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write gzip compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-bzip2</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write bzip2 compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-force</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">force writing to output file (default:
  no)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<div style="height: 1.00em;">&#x00A0;</div>
File format for option <i>-regionmapping</i>:
<div style="height: 1.00em;">&#x00A0;</div>
The file supplied to option -regionmapping defines a &#x201C;mapping&#x201D;. A
  mapping maps the sequence-region entries given in the <i>GFF3_file</i> to a
  sequence file containing the corresponding sequence. Mappings can be defined
  in one of the following two forms:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
mapping = {
  chr1  = &quot;hs_ref_chr1.fa.gz&quot;,
  chr2  = &quot;hs_ref_chr2.fa.gz&quot;
}
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
or
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
function mapping(sequence_region)
  return &quot;hs_ref_&quot;..sequence_region..&quot;.fa.gz&quot;
end
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
The first form defines a Lua (http://www.lua.org) table named
  &#x201C;mapping&#x201D; which maps each sequence region to the corresponding
  sequence file. The second one defines a Lua function &#x201C;mapping&#x201D;,
  which has to return the sequence file name when it is called with the
  sequence_region as argument.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
