<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:45 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-COMPREADS-REFCO(1) GenomeTools Manual
GT-COMPREADS-REFCO(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-compreads-refcompress - Generates compact encoding for
fastq data using Reference Compressed Reads (RCR).</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt compreads refcompress [option ...] (-bam file -ref
file)</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-mquals [yes|no] <br>
store mapping quality for each read (default: no)</p>

<p style="margin-top: 1em">-quals [yes|no] <br>
store all quality values for each read, this implies
enabling of option &quot;vquals&quot; (default: no)</p>

<p style="margin-top: 1em">-vquals [yes|no] <br>
store quality values of read positions having variations
compared to reference (default: no)</p>

<p style="margin-top: 1em">-descs [yes|no] <br>
store read name for each read (default: no)</p>

<p style="margin-top: 1em">-ureads [yes|no] <br>
store unmapped reads in a separated fastq file (base name
will be the value given in name and suffix will be &quot;
_unmapped.fastq&quot; (default: no)</p>

<p style="margin-top: 1em">-ref [string] <br>
Index file (generated by the gt encseq tool) for reference
genome. (default: undefined)</p>

<p style="margin-top: 1em">-bam [string] <br>
File containing alignment of reads to genome (sorted
&quot;.bam&quot; file). (default: undefined)</p>

<p style="margin-top: 1em">-name [string] <br>
specify base name for RCR to be generated. If not set, base
name will be set to base name of value given for option
&quot;bam&quot; (default: undefined)</p>

<p style="margin-top: 1em">-help <br>
display help and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-COMPREADS-REFCO(1)</p>
<hr>
</body>
</html>
