<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:46 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-CSA(1) GenomeTools Manual GT-CSA(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-csa - Transform spliced alignments from GFF3 file into
consensus spliced alignments.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt csa [option ...] [GFF3_file]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-join-length [value] <br>
set join length for the spliced alignment clustering
(default: 300)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-o [filename] <br>
redirect output to specified file (default: undefined)</p>

<p style="margin-top: 1em">-gzip [yes|no] <br>
write gzip compressed output file (default: no)</p>

<p style="margin-top: 1em">-bzip2 [yes|no] <br>
write bzip2 compressed output file (default: no)</p>

<p style="margin-top: 1em">-force [yes|no] <br>
force writing to output file (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">EXAMPLE: <br>
Let&acirc;s assume we have a GFF3 file
csa_example_spliced_alignments.gff3 containing the following
four overlapping spliced alignments (represented as genes
with exons as children):</p>

<p style="margin-top: 1em">##gff-version 3 <br>
##sequence-region seq 1 290 <br>
seq . gene 1 209 . + . ID=gene1 <br>
seq . exon 1 90 . + . Parent=gene1 <br>
seq . exon 110 190 . + . Parent=gene1 <br>
seq . exon 201 209 . + . Parent=gene1 <br>
### <br>
seq . gene 1 290 . + . ID=gene2 <br>
seq . exon 1 90 . + . Parent=gene2 <br>
seq . exon 101 190 . + . Parent=gene2 <br>
seq . exon 201 290 . + . Parent=gene2 <br>
### <br>
seq . gene 10 290 . + . ID=gene3 <br>
seq . exon 10 90 . + . Parent=gene3 <br>
seq . exon 110 190 . + . Parent=gene3 <br>
seq . exon 201 290 . + . Parent=gene3 <br>
### <br>
seq . gene 181 290 . + . ID=gene4 <br>
seq . exon 181 190 . + . Parent=gene4 <br>
seq . exon 201 290 . + . Parent=gene4 <br>
###</p>

<p style="margin-top: 1em">To compute the consensus spliced
alignments we call:</p>

<p style="margin-top: 1em">$ gt csa
csa_example_spliced_alignments.gff3</p>

<p style="margin-top: 1em">Which returns:</p>

<p style="margin-top: 1em">##gff-version 3 <br>
##sequence-region seq 1 290 <br>
seq gt csa gene 1 290 . + . ID=gene1 <br>
seq gt csa mRNA 1 290 . + . ID=mRNA1;Parent=gene1 <br>
seq gt csa exon 1 90 . + . Parent=mRNA1 <br>
seq gt csa exon 110 190 . + . Parent=mRNA1 <br>
seq gt csa exon 201 290 . + . Parent=mRNA1 <br>
seq gt csa mRNA 1 290 . + . ID=mRNA2;Parent=gene1 <br>
seq gt csa exon 1 90 . + . Parent=mRNA2 <br>
seq gt csa exon 101 190 . + . Parent=mRNA2 <br>
seq gt csa exon 201 290 . + . Parent=mRNA2 <br>
###</p>

<p style="margin-top: 1em">As one can see, they have been
combined into a consensus spliced alignment (represented as
genes with mRNAs as children which in turn have exons as
children) with two alternative <br>
splice forms. The first and the third spliced alignment have
been combined into the first alternative splice form (mRNA1)
and the the second and the fourth spliced alignment into
<br>
the second alternative splice form (mRNA2).</p>

<p style="margin-top: 1em">As one can see, the second exon
from the first alternative splice form is shorter than the
corresponding exon from the second alternative splice
form.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-CSA(1)</p>
<hr>
</body>
</html>
