<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-COMPREADS-REFCO(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-COMPREADS-REFCO(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-COMPREADS-REFCO(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-compreads-refcompress - Generates compact encoding for fastq data using
  Reference Compressed Reads (RCR).
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt compreads refcompress</b> [option ...] (-bam file -ref file)
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">be verbose (default: no)</div>
<div class="Pp"></div>
<b>-mquals</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">store mapping quality for each read (default:
  no)</div>
<div class="Pp"></div>
<b>-quals</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">store all quality values for each read, this
  implies enabling of option &quot;vquals&quot; (default: no)</div>
<div class="Pp"></div>
<b>-vquals</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">store quality values of read positions having
  variations compared to reference (default: no)</div>
<div class="Pp"></div>
<b>-descs</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">store read name for each read (default:
  no)</div>
<div class="Pp"></div>
<b>-ureads</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">store unmapped reads in a separated fastq file
  (base name will be the value given in name and suffix will be &quot;
  _unmapped.fastq&quot; (default: no)</div>
<div class="Pp"></div>
<b>-ref</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">Index file (generated by the gt encseq tool)
  for reference genome. (default: undefined)</div>
<div class="Pp"></div>
<b>-bam</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">File containing alignment of reads to genome
  (sorted &quot;.bam&quot; file). (default: undefined)</div>
<div class="Pp"></div>
<b>-name</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">specify base name for RCR to be generated. If
  not set, base name will be set to base name of value given for option
  &quot;bam&quot; (default: undefined)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
