<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-CSA(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-CSA(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-CSA(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-csa - Transform spliced alignments from GFF3 file into consensus spliced
  alignments.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt csa</b> [option ...] [GFF3_file]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-join-length</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">set join length for the spliced alignment
  clustering (default: 300)</div>
<div class="Pp"></div>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">be verbose (default: no)</div>
<div class="Pp"></div>
<b>-o</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">redirect output to specified file (default:
  undefined)</div>
<div class="Pp"></div>
<b>-gzip</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write gzip compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-bzip2</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write bzip2 compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-force</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">force writing to output file (default:
  no)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<h1 class="Sh" title="Sh" id="EXAMPLE:"><a class="selflink" href="#EXAMPLE:">EXAMPLE:</a></h1>
Let&#x2019;s assume we have a GFF3 file
  <i>csa_example_spliced_alignments.gff3</i> containing the following four
  overlapping spliced alignments (represented as genes with exons as children):
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
##gff-version 3
##sequence-region   seq 1 290
seq .       gene    1       209     .       +       .       ID=gene1
seq .       exon    1       90      .       +       .       Parent=gene1
seq .       exon    110     190     .       +       .       Parent=gene1
seq .       exon    201     209     .       +       .       Parent=gene1
###
seq .       gene    1       290     .       +       .       ID=gene2
seq .       exon    1       90      .       +       .       Parent=gene2
seq .       exon    101     190     .       +       .       Parent=gene2
seq .       exon    201     290     .       +       .       Parent=gene2
###
seq .       gene    10      290     .       +       .       ID=gene3
seq .       exon    10      90      .       +       .       Parent=gene3
seq .       exon    110     190     .       +       .       Parent=gene3
seq .       exon    201     290     .       +       .       Parent=gene3
###
seq .       gene    181     290     .       +       .       ID=gene4
seq .       exon    181     190     .       +       .       Parent=gene4
seq .       exon    201     290     .       +       .       Parent=gene4
###
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
To compute the consensus spliced alignments we call:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
$ gt csa csa_example_spliced_alignments.gff3
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
Which returns:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
##gff-version 3
##sequence-region   seq 1 290
seq gt csa  gene    1       290     .       +       .       ID=gene1
seq gt csa  mRNA    1       290     .       +       .       ID=mRNA1;Parent=gene1
seq gt csa  exon    1       90      .       +       .       Parent=mRNA1
seq gt csa  exon    110     190     .       +       .       Parent=mRNA1
seq gt csa  exon    201     290     .       +       .       Parent=mRNA1
seq gt csa  mRNA    1       290     .       +       .       ID=mRNA2;Parent=gene1
seq gt csa  exon    1       90      .       +       .       Parent=mRNA2
seq gt csa  exon    101     190     .       +       .       Parent=mRNA2
seq gt csa  exon    201     290     .       +       .       Parent=mRNA2
###
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
As one can see, they have been combined into a consensus spliced alignment
  (represented as genes with mRNAs as children which in turn have exons as
  children) with two alternative splice forms. The first and the third spliced
  alignment have been combined into the first alternative splice form (mRNA1)
  and the the second and the fourth spliced alignment into the second
  alternative splice form (mRNA2).
<div style="height: 1.00em;">&#x00A0;</div>
As one can see, the second exon from the first alternative splice form is
  shorter than the corresponding exon from the second alternative splice form.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
