<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-EXTRACTSEQ(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-EXTRACTSEQ(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-EXTRACTSEQ(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-extractseq - Extract sequences from given sequence file(s) or fastaindex.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt extractseq</b> [option ...] [sequence_file(s)] | fastaindex
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-frompos</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">extract sequence from this position counting
  from 1 on (default: 0)</div>
<div class="Pp"></div>
<b>-topos</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">extract sequence up to this position counting
  from 1 on (default: 0)</div>
<div class="Pp"></div>
<b>-match</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">extract all sequences whose description
  matches the given pattern. The given pattern must be a valid extended regular
  expression. (default: undefined)</div>
<div class="Pp"></div>
<b>-keys</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">extract substrings for keys in specified file
  (default: undefined)</div>
<div class="Pp"></div>
<b>-width</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">set output width for FASTA sequence printing
  (0 disables formatting) (default: 0)</div>
<div class="Pp"></div>
<b>-o</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">redirect output to specified file (default:
  undefined)</div>
<div class="Pp"></div>
<b>-gzip</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write gzip compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-bzip2</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write bzip2 compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-force</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">force writing to output file (default:
  no)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<div style="height: 1.00em;">&#x00A0;</div>
The option -keys allows one to extract substrings or sequences from the given
  sequence file or from a fasta index. The substrings to be extracted are
  specified in a key file given as argument to this option. The key file must
  contain lines of the form
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
k
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
or
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
k i j
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
where k is a string (the key) and the optional i and j are positive integers
  such that i&#x21D0;j. k is the key and the optional numbers i and j specify
  the first position of the substring and the last position of the substring to
  be extracted. The positions are counted from 1. If k is identical to the
  string between the first first and second occurrence of the symbol | in a
  fasta header, then the fasta header and the corresponding sequence is output.
  For example in the fasta header
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
&gt;tr|A0AQI4|A0AQI4_9ARCH Putative ammonia monooxygenase (Fragment)
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
the fasta key is A0AQI4. If i and j are both specified, then the corresponding
  substring is shown in fasta format. In the latter case the header of the fasta
  formatted sequence in the output begins with
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
&gt;k i j
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
followed by the original original fasta header.
<div style="height: 1.00em;">&#x00A0;</div>
If the sequence input are fasta files, then the following holds:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;duplicated lines in the input file
  lead to only one sequence in the output</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;the sequences are output according to
  the order in the original sequence files</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;the formatting of the output can be
  controlled by the options <i>-width</i>, <i>-o</i>, <i>-gzip</i>, and
  <i>-bzip2</i></div>
<div style="height: 1.00em;">&#x00A0;</div>
If the sequence input comes from a fasta index (see below), the following holds:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;option <i>-width</i> is required</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;option <i>-o</i>, <i>-gzip</i> and
  <i>-bzip2</i> do not work</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;the sequences are output in the order
  the corresponding keys appear in the key file</div>
<div style="height: 1.00em;">&#x00A0;</div>
If the end of the argument list only contains one filename, say fastaindex, then
  it is checked if there is a file fastaindex.kys. This makes up part of the
  fasta index, which is constructed by calling the suffixerator tool as follows:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
gt suffixerator -protein -ssp -tis -des -sds -kys -indexname fastaindex \
  -db inputfile1 [inputfile2 ..]
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
This reads the protein sequence files given to the option <i>-db</i> and creates
  several files:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;a file fastaindex.esq representing the
  sequence.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;a file fastaindex.ssp specifying the
  sequence separator positions.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;a file fastaindex.des showing the
  fasta headers line by line.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;a file fastaindex.sds giving the
  sequence header delimiter positions.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;a file fastaindex.kys containing the
  keys in the fasta files.</div>
<div style="height: 1.00em;">&#x00A0;</div>
For the suffixerator command to work, the keys of the form |key| in the fasta
  header must satisfy the following constraints:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;they all have to be of the same
  length, not longer than 128, and not shorter than 1</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;they have to appear in lexicographic
  order</div>
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
