<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:47 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-EVAL(1) GenomeTools Manual GT-EVAL(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-eval - Compare annotation files and show accuracy
measures (prediction vs. reference).</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt eval reference_file prediction_file</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-nuc [yes|no] <br>
evaluate nucleotide level (memory consumption is
proportional to the input file sizes) (default: yes)</p>

<p style="margin-top: 1em">-ltr [yes|no] <br>
evaluate a LTR retrotransposon prediction instead of a gene
prediction (all LTR_retrotransposon elements are considered
to have an undetermined strand) (default: no)</p>

<p style="margin-top: 1em">-ltrdelta [value] <br>
set allowed delta for LTR borders to be considered equal
(default: 20)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-o [filename] <br>
redirect output to specified file (default: undefined)</p>

<p style="margin-top: 1em">-gzip [yes|no] <br>
write gzip compressed output file (default: no)</p>

<p style="margin-top: 1em">-bzip2 [yes|no] <br>
write bzip2 compressed output file (default: no)</p>

<p style="margin-top: 1em">-force [yes|no] <br>
force writing to output file (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">The program shows sensitivity
and specificity values for certain feature types (e.g.,
gene, mRNA, and exon). For some feature types the number of
missing and wrong features of <br>
that type is also shown. Thereby, &acirc;missing&acirc;
means the number of features of that type from the
&acirc;reference&acirc; without overlap to a feature of that
type from the &acirc;prediction&acirc;. Vice <br>
versa, &acirc;wrong&acirc; denotes the number of features of
that type from the &acirc;prediction&acirc; without overlap
to a feature of that type from the
&acirc;reference&acirc;.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-EVAL(1)</p>
<hr>
</body>
</html>
