<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:47 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-EXTRACTSEQ(1) GenomeTools Manual GT-EXTRACTSEQ(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-extractseq - Extract sequences from given sequence
file(s) or fastaindex.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt extractseq [option ...] [sequence_file(s)] |
fastaindex</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-frompos [value] <br>
extract sequence from this position counting from 1 on
(default: 0)</p>

<p style="margin-top: 1em">-topos [value] <br>
extract sequence up to this position counting from 1 on
(default: 0)</p>

<p style="margin-top: 1em">-match [string] <br>
extract all sequences whose description matches the given
pattern. The given pattern must be a valid extended regular
expression. (default: undefined)</p>

<p style="margin-top: 1em">-keys [filename] <br>
extract substrings for keys in specified file (default:
undefined)</p>

<p style="margin-top: 1em">-width [value] <br>
set output width for FASTA sequence printing (0 disables
formatting) (default: 0)</p>

<p style="margin-top: 1em">-o [filename] <br>
redirect output to specified file (default: undefined)</p>

<p style="margin-top: 1em">-gzip [yes|no] <br>
write gzip compressed output file (default: no)</p>

<p style="margin-top: 1em">-bzip2 [yes|no] <br>
write bzip2 compressed output file (default: no)</p>

<p style="margin-top: 1em">-force [yes|no] <br>
force writing to output file (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">The option -keys allows one to
extract substrings or sequences from the given sequence file
or from a fasta index. The substrings to be extracted are
specified in a key file <br>
given as argument to this option. The key file must contain
lines of the form</p>

<p style="margin-top: 1em">k</p>

<p style="margin-top: 1em">or</p>

<p style="margin-top: 1em">k i j</p>

<p style="margin-top: 1em">where k is a string (the key)
and the optional i and j are positive integers such that
i&acirc;j. k is the key and the optional numbers i and j
specify the first position of the <br>
substring and the last position of the substring to be
extracted. The positions are counted from 1. If k is
identical to the string between the first first and second
occurrence <br>
of the symbol | in a fasta header, then the fasta header and
the corresponding sequence is output. For example in the
fasta header</p>

<p style="margin-top: 1em">&gt;tr|A0AQI4|A0AQI4_9ARCH
Putative ammonia monooxygenase (Fragment)</p>

<p style="margin-top: 1em">the fasta key is A0AQI4. If i
and j are both specified, then the corresponding substring
is shown in fasta format. In the latter case the header of
the fasta formatted sequence <br>
in the output begins with</p>

<p style="margin-top: 1em">&gt;k i j</p>

<p style="margin-top: 1em">followed by the original
original fasta header.</p>

<p style="margin-top: 1em">If the sequence input are fasta
files, then the following holds:</p>

<p style="margin-top: 1em">&Acirc;&middot; duplicated lines
in the input file lead to only one sequence in the
output</p>

<p style="margin-top: 1em">&Acirc;&middot; the sequences
are output according to the order in the original sequence
files</p>

<p style="margin-top: 1em">&Acirc;&middot; the formatting
of the output can be controlled by the options -width, -o,
-gzip, and -bzip2</p>

<p style="margin-top: 1em">If the sequence input comes from
a fasta index (see below), the following holds:</p>

<p style="margin-top: 1em">&Acirc;&middot; option -width is
required</p>

<p style="margin-top: 1em">&Acirc;&middot; option -o, -gzip
and -bzip2 do not work</p>

<p style="margin-top: 1em">&Acirc;&middot; the sequences
are output in the order the corresponding keys appear in the
key file</p>

<p style="margin-top: 1em">If the end of the argument list
only contains one filename, say fastaindex, then it is
checked if there is a file fastaindex.kys. This makes up
part of the fasta index, which is <br>
constructed by calling the suffixerator tool as follows:</p>

<p style="margin-top: 1em">gt suffixerator -protein -ssp
-tis -des -sds -kys -indexname fastaindex -db inputfile1
[inputfile2 ..]</p>

<p style="margin-top: 1em">This reads the protein sequence
files given to the option -db and creates several files:</p>

<p style="margin-top: 1em">&Acirc;&middot; a file
fastaindex.esq representing the sequence.</p>

<p style="margin-top: 1em">&Acirc;&middot; a file
fastaindex.ssp specifying the sequence separator
positions.</p>

<p style="margin-top: 1em">&Acirc;&middot; a file
fastaindex.des showing the fasta headers line by line.</p>

<p style="margin-top: 1em">&Acirc;&middot; a file
fastaindex.sds giving the sequence header delimiter
positions.</p>

<p style="margin-top: 1em">&Acirc;&middot; a file
fastaindex.kys containing the keys in the fasta files.</p>

<p style="margin-top: 1em">For the suffixerator command to
work, the keys of the form |key| in the fasta header must
satisfy the following constraints:</p>

<p style="margin-top: 1em">&Acirc;&middot; they all have to
be of the same length, not longer than 128, and not shorter
than 1</p>

<p style="margin-top: 1em">&Acirc;&middot; they have to
appear in lexicographic order</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-EXTRACTSEQ(1)</p>
<hr>
</body>
</html>
