<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:47 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-EXTRACTFEAT(1) GenomeTools Manual
GT-EXTRACTFEAT(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-extractfeat - Extract features given in GFF3 file from
sequence file.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt extractfeat [option ...] [GFF3_file]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-type [string] <br>
set type of features to extract (default: undefined)</p>

<p style="margin-top: 1em">-join [yes|no] <br>
join feature sequences in the same subgraph into a single
one (default: no)</p>

<p style="margin-top: 1em">-translate [yes|no] <br>
translate the features (of a DNA sequence) into protein
(default: no)</p>

<p style="margin-top: 1em">-seqid [yes|no] <br>
add sequence ID of extracted features to FASTA descriptions
(default: no)</p>

<p style="margin-top: 1em">-target [yes|no] <br>
add target ID(s) of extracted features to FASTA descriptions
(default: no)</p>

<p style="margin-top: 1em">-coords [yes|no] <br>
add location of extracted features to FASTA descriptions
(default: no)</p>

<p style="margin-top: 1em">-retainids [yes|no] <br>
use ID attributes of extracted features as FASTA
descriptions (default: no)</p>

<p style="margin-top: 1em">-gcode [value] <br>
specify genetic code to use (default: 1)</p>

<p style="margin-top: 1em">-seqfile [filename] <br>
set the sequence file from which to take the sequences
(default: undefined)</p>

<p style="margin-top: 1em">-encseq [filename] <br>
set the encoded sequence indexname from which to take the
sequences (default: undefined)</p>

<p style="margin-top: 1em">-seqfiles <br>
set the sequence files from which to extract the features
use -- to terminate the list of sequence files</p>

<p style="margin-top: 1em">-matchdesc [yes|no] <br>
search the sequence descriptions from the input files for
the desired sequence IDs (in GFF3), reporting the first
match (default: no)</p>

<p style="margin-top: 1em">-matchdescstart [yes|no] <br>
exactly match the sequence descriptions from the input files
for the desired sequence IDs (in GFF3) from the beginning to
the first whitespace (default: no)</p>

<p style="margin-top: 1em">-usedesc [yes|no] <br>
use sequence descriptions to map the sequence IDs (in GFF3)
to actual sequence entries. If a description contains a
sequence range (e.g., III:1000001..2000000), the first <br>
part is used as sequence ID (III) and the first range
position as offset (1000001) (default: no)</p>

<p style="margin-top: 1em">-regionmapping [string] <br>
set file containing sequence-region to sequence file mapping
(default: undefined)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-width [value] <br>
set output width for FASTA sequence printing (0 disables
formatting) (default: 0)</p>

<p style="margin-top: 1em">-o [filename] <br>
redirect output to specified file (default: undefined)</p>

<p style="margin-top: 1em">-gzip [yes|no] <br>
write gzip compressed output file (default: no)</p>

<p style="margin-top: 1em">-bzip2 [yes|no] <br>
write bzip2 compressed output file (default: no)</p>

<p style="margin-top: 1em">-force [yes|no] <br>
force writing to output file (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">Genetic code numbers for option
-gcode:</p>

<p style="margin-top: 1em">1: Standard 2: Vertebrate
Mitochondrial 3: Yeast Mitochondrial 4: Mold Mitochondrial;
Protozoan Mitochondrial; Coelenterate Mitochondrial;
Mycoplasma; Spiroplasma 5: Invertebrate <br>
Mitochondrial 6: Ciliate Nuclear; Dasycladacean Nuclear;
Hexamita Nuclear 9: Echinoderm Mitochondrial; Flatworm
Mitochondrial 10: Euplotid Nuclear 11: Bacterial, Archaeal
and <br>
Plant Plastid 12: Alternative Yeast Nuclear 13: Ascidian
Mitochondrial 14: Alternative Flatworm Mitochondrial 15:
Blepharisma Macronuclear 16: Chlorophycean Mitochondrial 21:
<br>
Trematode Mitochondrial 22: Scenedesmus obliquus
Mitochondrial 23: Thraustochytrium Mitochondrial 24:
Pterobranchia Mitochondrial 25: Candidate Division SR1 and
Gracilibacteria</p>

<p style="margin-top: 1em">File format for option
-regionmapping:</p>

<p style="margin-top: 1em">The file supplied to option
-regionmapping defines a &acirc;mapping&acirc;. A mapping
maps the sequence-region entries given in the GFF3_file to a
sequence file containing the corresponding <br>
sequence. Mappings can be defined in one of the following
two forms:</p>

<p style="margin-top: 1em">mapping = { <br>
chr1 = &quot;hs_ref_chr1.fa.gz&quot;, <br>
chr2 = &quot;hs_ref_chr2.fa.gz&quot; <br>
}</p>

<p style="margin-top: 1em">or</p>

<p style="margin-top: 1em">function
mapping(sequence_region) <br>
return
&quot;hs_ref_&quot;..sequence_region..&quot;.fa.gz&quot;
<br>
end</p>

<p style="margin-top: 1em">The first form defines a Lua
(http://www.lua.org) table named &acirc;mapping&acirc; which
maps each sequence region to the corresponding sequence
file. The second one defines a Lua function <br>
&acirc;mapping&acirc;, which has to return the sequence file
name when it is called with the sequence_region as
argument.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-EXTRACTFEAT(1)</p>
<hr>
</body>
</html>
