<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-EVAL(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-EVAL(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-EVAL(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-eval - Compare annotation files and show accuracy measures (prediction vs.
  reference).
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt eval</b> reference_file prediction_file
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-nuc</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">evaluate nucleotide level (memory consumption
  is proportional to the input file sizes) (default: yes)</div>
<div class="Pp"></div>
<b>-ltr</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">evaluate a LTR retrotransposon prediction
  instead of a gene prediction (all LTR_retrotransposon elements are considered
  to have an undetermined strand) (default: no)</div>
<div class="Pp"></div>
<b>-ltrdelta</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">set allowed delta for LTR borders to be
  considered equal (default: 20)</div>
<div class="Pp"></div>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">be verbose (default: no)</div>
<div class="Pp"></div>
<b>-o</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">redirect output to specified file (default:
  undefined)</div>
<div class="Pp"></div>
<b>-gzip</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write gzip compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-bzip2</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write bzip2 compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-force</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">force writing to output file (default:
  no)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<div style="height: 1.00em;">&#x00A0;</div>
The program shows sensitivity and specificity values for certain feature types
  (e.g., gene, mRNA, and exon). For some feature types the number of missing and
  wrong features of that type is also shown. Thereby, &#x201C;missing&#x201D;
  means the number of features of that type from the &#x201C;reference&#x201D;
  without overlap to a feature of that type from the &#x201C;prediction&#x201D;.
  Vice versa, &#x201C;wrong&#x201D; denotes the number of features of that type
  from the &#x201C;prediction&#x201D; without overlap to a feature of that type
  from the &#x201C;reference&#x201D;.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
