<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:46 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-ENCSEQ-ENCODE(1) GenomeTools Manual
GT-ENCSEQ-ENCODE(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-encseq-encode - Encode sequence files (FASTA/FASTQ,
GenBank, EMBL) efficiently.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt encseq encode sequence_file [sequence_file [sequence_file
...]]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-showstats [yes|no] <br>
show compression results (default: no)</p>

<p style="margin-top: 1em">-ssp [yes|no] <br>
output sequence separator positions to file (default:
yes)</p>

<p style="margin-top: 1em">-des [yes|no] <br>
output sequence descriptions to file (default: yes)</p>

<p style="margin-top: 1em">-sds [yes|no] <br>
output sequence description separator positions to file
(default: yes)</p>

<p style="margin-top: 1em">-md5 [yes|no] <br>
output MD5 sums to file (default: yes)</p>

<p style="margin-top: 1em">-clipdesc [yes|no] <br>
clip descriptions after first whitespace (default: no)</p>

<p style="margin-top: 1em">-sat [string] <br>
specify kind of sequence representation by one of the
keywords direct, bytecompress, eqlen, bit, uchar, ushort,
uint32 (default: undefined)</p>

<p style="margin-top: 1em">-dna [yes|no] <br>
input is DNA sequence (default: no)</p>

<p style="margin-top: 1em">-protein [yes|no] <br>
input is protein sequence (default: no)</p>

<p style="margin-top: 1em">-plain [yes|no] <br>
process as plain text (default: no)</p>

<p style="margin-top: 1em">-indexname [string] <br>
specify name for index to be generated (default:
undefined)</p>

<p style="margin-top: 1em">-smap [string] <br>
specify file containing a symbol mapping (default:
undefined)</p>

<p style="margin-top: 1em">-lossless [yes|no] <br>
allow lossless original sequence retrieval (default: no)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help for basic options and exit</p>

<p style="margin-top: 1em">-help+ <br>
display help for all options and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-ENCSEQ-ENCODE(1)</p>
<hr>
</body>
</html>
