<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-GFF3(1) GenomeTools Manual GT-GFF3(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-gff3 - Parse, possibly transform, and output GFF3
files.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt gff3 [option ...] [GFF3_file ...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-sort [yes|no] <br>
sort the GFF3 features (memory consumption is proportional
to the input file size(s)) (default: no)</p>

<p style="margin-top: 1em">-sortlines [yes|no] <br>
sort the GFF3 features on a strict line basis (not sorted
asdefined by GenomeTools) (default: no)</p>

<p style="margin-top: 1em">-sortnum [yes|no] <br>
enable natural numeric sorting for sequence regions (not
sorted as defined by GenomeTools) (default: no)</p>

<p style="margin-top: 1em">-tidy [yes|no] <br>
try to tidy the GFF3 files up during parsing (default:
no)</p>

<p style="margin-top: 1em">-retainids [yes|no] <br>
when available, use the original IDs provided in the source
file (memory consumption is proportional to the input file
size(s)) (default: no)</p>

<p style="margin-top: 1em">-checkids [yes|no] <br>
make sure the ID attributes are unique within the scope of
each GFF3_file, as required by GFF3 specification (memory
consumption is proportional to the input file size(s)). <br>
If features with the same Parent attribute are not separated
by a # line the GFF3 parser tries to treat them as a
multi-line feature. This requires at least matching sequence
<br>
IDs and types. (default: no)</p>

<p style="margin-top: 1em">-addids [yes|no] <br>
add missing &quot;##sequence-region&quot; lines
automatically (default: yes)</p>

<p style="margin-top: 1em">-fixregionboundaries [yes|no]
<br>
automatically adjust &quot;##sequence-region&quot; lines to
contain all their features (memory consumption is
proportional to the input file size(s)) (default: no)</p>

<p style="margin-top: 1em">-addintrons [yes|no] <br>
add intron features between existing exon features (default:
no)</p>

<p style="margin-top: 1em">-offset [value] <br>
transform all features by the given offset</p>

<p style="margin-top: 1em">-offsetfile [filename] <br>
transform all features by the offsets given in file
(default: undefined)</p>

<p style="margin-top: 1em">-setsource [string] <br>
set the source value (2nd column) of each feature (default:
undefined)</p>

<p style="margin-top: 1em">-typecheck [string] <br>
use an ontology given in an OBO file to validate
parent-child relationships. If no argument is given, the
sofa.obo file from the gtdata/obo_files directory is used.
If an <br>
argument is given, it is used as an OBO filename. In the
case that such a file does not exist .obo is added to the
argument and loading the resulting filename from the <br>
gtdata/obo_files directory is attempted. (default:
undefined)</p>

<p style="margin-top: 1em">-xrfcheck [string] <br>
check Dbxref and Ontology_term attributes for correct syntax
according to a abbreviation definition file. If no argument
is given, the GO.xrf_abbs file from the <br>
gtdata/xrf_abbr directory is used. If an argument is given,
it is used as an specific filename for an abbreviation file.
In the case that such a file does not exist, <br>
.xrf_abbr is added to the argument and loading the resulting
filename from the gtdata/xrf_abbr directory is attempted.
(default: undefined)</p>

<p style="margin-top: 1em">-show [yes|no] <br>
show GFF3 output (default: yes)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-width [value] <br>
set output width for FASTA sequence printing (0 disables
formatting) (default: 0)</p>

<p style="margin-top: 1em">-o [filename] <br>
redirect output to specified file (default: undefined)</p>

<p style="margin-top: 1em">-gzip [yes|no] <br>
write gzip compressed output file (default: no)</p>

<p style="margin-top: 1em">-bzip2 [yes|no] <br>
write bzip2 compressed output file (default: no)</p>

<p style="margin-top: 1em">-force [yes|no] <br>
force writing to output file (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">File format for option
-offsetfile:</p>

<p style="margin-top: 1em">The file supplied to option
-offsetfile defines a mapping table named
&acirc;offsets&acirc;. It maps the sequence-region entries
given in the GFF3_file to offsets. It can be defined as <br>
follows:</p>

<p style="margin-top: 1em">offsets = { <br>
chr1 = 1000, <br>
chr2 = 500 <br>
}</p>

<p style="margin-top: 1em">When this example is used, all
features with seqid &acirc;chr1&acirc; will be offset by
1000 and all features with seqid &acirc;chr2&acirc; by
500.</p>

<p style="margin-top: 1em">If -offsetfile is used, offsets
for all sequence-regions contained in the given GFF3 files
must be defined.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-GFF3(1)</p>
<hr>
</body>
</html>
