<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-GFF3(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-GFF3(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-GFF3(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-gff3 - Parse, possibly transform, and output GFF3 files.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt gff3</b> [option ...] [GFF3_file ...]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-sort</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">sort the GFF3 features (memory consumption is
  proportional to the input file size(s)) (default: no)</div>
<div class="Pp"></div>
<b>-sortlines</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">sort the GFF3 features on a strict line basis
  (not sorted asdefined by GenomeTools) (default: no)</div>
<div class="Pp"></div>
<b>-sortnum</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">enable natural numeric sorting for sequence
  regions (not sorted as defined by GenomeTools) (default: no)</div>
<div class="Pp"></div>
<b>-tidy</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">try to tidy the GFF3 files up during parsing
  (default: no)</div>
<div class="Pp"></div>
<b>-retainids</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">when available, use the original IDs provided
  in the source file (memory consumption is proportional to the input file
  size(s)) (default: no)</div>
<div class="Pp"></div>
<b>-checkids</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">make sure the ID attributes are unique within
  the scope of each GFF3_file, as required by GFF3 specification (memory
  consumption is proportional to the input file size(s)). If features with the
  same Parent attribute are not separated by a <i>#</i> line the GFF3 parser
  tries to treat them as a multi-line feature. This requires at least matching
  sequence IDs and types. (default: no)</div>
<div class="Pp"></div>
<b>-addids</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">add missing &quot;##sequence-region&quot;
  lines automatically (default: yes)</div>
<div class="Pp"></div>
<b>-fixregionboundaries</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">automatically adjust
  &quot;##sequence-region&quot; lines to contain all their features (memory
  consumption is proportional to the input file size(s)) (default: no)</div>
<div class="Pp"></div>
<b>-addintrons</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">add intron features between existing exon
  features (default: no)</div>
<div class="Pp"></div>
<b>-offset</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">transform all features by the given
  offset</div>
<div class="Pp"></div>
<b>-offsetfile</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">transform all features by the offsets given in
  file (default: undefined)</div>
<div class="Pp"></div>
<b>-setsource</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">set the <i>source</i> value (2nd column) of
  each feature (default: undefined)</div>
<div class="Pp"></div>
<b>-typecheck</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">use an ontology given in an OBO file to
  validate parent-child relationships. If no argument is given, the sofa.obo
  file from the gtdata/obo_files directory is used. If an argument is given, it
  is used as an OBO filename. In the case that such a file does not exist
  <i>.obo</i> is added to the argument and loading the resulting filename from
  the gtdata/obo_files directory is attempted. (default: undefined)</div>
<div class="Pp"></div>
<b>-xrfcheck</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">check Dbxref and Ontology_term attributes for
  correct syntax according to a abbreviation definition file. If no argument is
  given, the GO.xrf_abbs file from the gtdata/xrf_abbr directory is used. If an
  argument is given, it is used as an specific filename for an abbreviation
  file. In the case that such a file does not exist, <i>.xrf_abbr</i> is added
  to the argument and loading the resulting filename from the gtdata/xrf_abbr
  directory is attempted. (default: undefined)</div>
<div class="Pp"></div>
<b>-show</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">show GFF3 output (default: yes)</div>
<div class="Pp"></div>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">be verbose (default: no)</div>
<div class="Pp"></div>
<b>-width</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">set output width for FASTA sequence printing
  (0 disables formatting) (default: 0)</div>
<div class="Pp"></div>
<b>-o</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">redirect output to specified file (default:
  undefined)</div>
<div class="Pp"></div>
<b>-gzip</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write gzip compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-bzip2</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write bzip2 compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-force</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">force writing to output file (default:
  no)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<div style="height: 1.00em;">&#x00A0;</div>
File format for option <i>-offsetfile</i>:
<div style="height: 1.00em;">&#x00A0;</div>
The file supplied to option <i>-offsetfile</i> defines a mapping table named
  &#x201C;offsets&#x201D;. It maps the sequence-region entries given in the
  GFF3_file to offsets. It can be defined as follows:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
offsets = {
  chr1  = 1000,
  chr2  = 500
}
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
When this example is used, all features with seqid &#x201C;chr1&#x201D; will be
  offset by 1000 and all features with seqid &#x201C;chr2&#x201D; by 500.
<div style="height: 1.00em;">&#x00A0;</div>
If <i>-offsetfile</i> is used, offsets for all sequence-regions contained in the
  given GFF3 files must be defined.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
