<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-HOP(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-HOP(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-HOP(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-hop - Cognate sequence-based homopolymer error correction.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt hop</b> -&lt;mode&gt; -c &lt;encseq&gt; -map &lt;sam/bam&gt; -reads
  &lt;fastq&gt; [options...]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-c</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">cognate sequence (encoded using gt encseq
  encode)</div>
<div class="Pp"></div>
<b>-map</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">mapping of reads to the cognate sequence it
  must be in SAM/BAM format, and sorted by coordinate (can be prepared e.g.
  using: samtools sort)</div>
<div class="Pp"></div>
<b>-sam</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">mapping file is SAM default: BAM</div>
<div class="Pp"></div>
<b>-aggressive</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">correct as much as possible</div>
<div class="Pp"></div>
<b>-moderate</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">mediate between sensitivity and
  precision</div>
<div class="Pp"></div>
<b>-conservative</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">correct only most likely errors</div>
<div class="Pp"></div>
<b>-expert</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">manually select correction criteria</div>
<div class="Pp"></div>
<b>-reads</b>
<div style="margin-left: 4.00ex;">uncorrected read file(s) in FastQ format; the
  corrected reads are output in the currect working directory in files which are
  named as the input files, each prepended by a prefix (see -outprefix option)
  -reads allows one to output the reads in the same order as in the input and is
  mandatory if the SAM contains more than a single primary alignment for each
  read (e.g. output of bwasw) see also -o option as an alternative</div>
<div class="Pp"></div>
<b>-outprefix</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">prefix for output filenames (corrected
  reads)when -reads is specified the prefix is prepended to each input filename
  (default: hop_)</div>
<div class="Pp"></div>
<b>-o</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">output file for corrected reads (see also
  -reads/-outprefix) if -o is used, reads are output in a single file in the
  order they are found in the SAM file (which usually differ from the original
  order) this will only work if the reads were aligned with a software which
  only includes 1 alignment for each read (e.g. bwa) (default: undefined)</div>
<div class="Pp"></div>
<b>-hmin</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">minimal homopolymer length in cognate sequence
  (default: 3)</div>
<div class="Pp"></div>
<b>-read-hmin</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">minimal homopolymer length in reads (default:
  2)</div>
<div class="Pp"></div>
<b>-qmax</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">maximal average quality of homopolymer in a
  read (default: 120)</div>
<div class="Pp"></div>
<b>-altmax</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">max support of alternate homopol. length; e.g.
  0.8 means: do not correct any read if homop. length in more than 80%% of the
  reads has the same value, different from the cognate if altmax is set to 1.0
  reads are always corrected (default: 0.800000)</div>
<div class="Pp"></div>
<b>-cogmin</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">min support of cognate sequence homopol.
  length; e.g. 0.1 means: do not correct any read if cognate homop. length is
  not present in at least 10%% of the reads if cogmin is set to 0.0 reads are
  always corrected</div>
<div class="Pp"></div>
<b>-mapqmin</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">minimal mapping quality (default: 21)</div>
<div class="Pp"></div>
<b>-covmin</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">minimal coverage; e.g. 5 means: do not correct
  any read if coverage (number of reads mapped over whole homopolymer) is less
  than 5 if covmin is set to 1 reads are always corrected (default: 1)</div>
<div class="Pp"></div>
<b>-allow-muliple</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">allow multiple corrections in a read (default:
  no)</div>
<div class="Pp"></div>
<b>-clenmax</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">maximal correction length default:
  unlimited</div>
<div class="Pp"></div>
<b>-ann</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">annotation of cognate sequence it must be
  sorted by coordinates on the cognate sequence (this can be e.g. done using: gt
  gff3 -sort) if -ann is used, corrections will be limited to homopolymers
  startingor ending inside the feature type indicated by -ft optionformat:
  sorted GFF3 (default: undefined)</div>
<div class="Pp"></div>
<b>-ft</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">feature type to use when -ann option is
  specified (default: CDS)</div>
<div class="Pp"></div>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">be verbose (default: no)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help for basic options and exit</div>
<div class="Pp"></div>
<b>-help+</b>
<div style="margin-left: 4.00ex;">display help for all options and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<div style="height: 1.00em;">&#x00A0;</div>
Correction mode:
<div style="height: 1.00em;">&#x00A0;</div>
One of the options <i>-aggressive</i>, <i>-moderate</i>, <i>-conservative</i> or
  <i>-expert</i> must be selected.
<div style="height: 1.00em;">&#x00A0;</div>
The <i>-aggressive</i>, <i>-moderate</i> and <i>-conservative</i> modes are
  presets of the criteria by which it is decided if an observed discrepancy in
  homopolymer length between cognate sequence and a read shall be corrected or
  not. A description of the single criteria is provided by using the
  <i>-help+</i>' option. The presets are equivalent to the following settings:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
                    -aggressive    -moderate      -conservative
-hmin               3              3              3
-read-hmin          1              1              2
-altmax             1.00           0.99           0.80
-refmin             0.00           0.00           0.10
-mapqmin            0              10             21
-covmin             1              1              1
-clenmax            unlimited      unlimited      unlimited
-allow-multiple     yes            yes            no
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
The aggressive mode tries to maximize the sensitivity, the conservative mode to
  minimize the false positives. An even more conservative set of corrections can
  be achieved using the <i>-ann</i> option (see <i>-help+</i>).
<div style="height: 1.00em;">&#x00A0;</div>
The <i>-expert</i> mode allows one to manually set each parameter; the default
  values are the same as in the <i>-conservative</i> mode.
<div style="height: 1.00em;">&#x00A0;</div>
(Finally, for evaluation purposes only, the <i>-state-of-truth</i> mode can be
  used: this mode assumes that the sequenced genome has been specified as
  cognate sequence and outputs an ideal list of corrections.)
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
