<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:49 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-HOP(1) GenomeTools Manual GT-HOP(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-hop - Cognate sequence-based homopolymer error
correction.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt hop -&lt;mode&gt; -c &lt;encseq&gt; -map &lt;sam/bam&gt;
-reads &lt;fastq&gt; [options...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-c [string] <br>
cognate sequence (encoded using gt encseq encode)</p>

<p style="margin-top: 1em">-map [string] <br>
mapping of reads to the cognate sequence it must be in
SAM/BAM format, and sorted by coordinate (can be prepared
e.g. using: samtools sort)</p>

<p style="margin-top: 1em">-sam [yes|no] <br>
mapping file is SAM default: BAM</p>

<p style="margin-top: 1em">-aggressive [yes|no] <br>
correct as much as possible</p>

<p style="margin-top: 1em">-moderate [yes|no] <br>
mediate between sensitivity and precision</p>

<p style="margin-top: 1em">-conservative [yes|no] <br>
correct only most likely errors</p>

<p style="margin-top: 1em">-expert [yes|no] <br>
manually select correction criteria</p>

<p style="margin-top: 1em">-reads <br>
uncorrected read file(s) in FastQ format; the corrected
reads are output in the currect working directory in files
which are named as the input files, each prepended by a <br>
prefix (see -outprefix option) -reads allows one to output
the reads in the same order as in the input and is mandatory
if the SAM contains more than a single primary <br>
alignment for each read (e.g. output of bwasw) see also -o
option as an alternative</p>

<p style="margin-top: 1em">-outprefix [string] <br>
prefix for output filenames (corrected reads)when -reads is
specified the prefix is prepended to each input filename
(default: hop_)</p>

<p style="margin-top: 1em">-o [string] <br>
output file for corrected reads (see also -reads/-outprefix)
if -o is used, reads are output in a single file in the
order they are found in the SAM file (which usually <br>
differ from the original order) this will only work if the
reads were aligned with a software which only includes 1
alignment for each read (e.g. bwa) (default: undefined)</p>

<p style="margin-top: 1em">-hmin [value] <br>
minimal homopolymer length in cognate sequence (default:
3)</p>

<p style="margin-top: 1em">-read-hmin [value] <br>
minimal homopolymer length in reads (default: 2)</p>

<p style="margin-top: 1em">-qmax [value] <br>
maximal average quality of homopolymer in a read (default:
120)</p>

<p style="margin-top: 1em">-altmax [value] <br>
max support of alternate homopol. length; e.g. 0.8 means: do
not correct any read if homop. length in more than 80%% of
the reads has the same value, different from the <br>
cognate if altmax is set to 1.0 reads are always corrected
(default: 0.800000)</p>

<p style="margin-top: 1em">-cogmin [value] <br>
min support of cognate sequence homopol. length; e.g. 0.1
means: do not correct any read if cognate homop. length is
not present in at least 10%% of the reads if cogmin is <br>
set to 0.0 reads are always corrected</p>

<p style="margin-top: 1em">-mapqmin [value] <br>
minimal mapping quality (default: 21)</p>

<p style="margin-top: 1em">-covmin [value] <br>
minimal coverage; e.g. 5 means: do not correct any read if
coverage (number of reads mapped over whole homopolymer) is
less than 5 if covmin is set to 1 reads are always <br>
corrected (default: 1)</p>

<p style="margin-top: 1em">-allow-muliple [yes|no] <br>
allow multiple corrections in a read (default: no)</p>

<p style="margin-top: 1em">-clenmax [value] <br>
maximal correction length default: unlimited</p>

<p style="margin-top: 1em">-ann [string] <br>
annotation of cognate sequence it must be sorted by
coordinates on the cognate sequence (this can be e.g. done
using: gt gff3 -sort) if -ann is used, corrections will be
<br>
limited to homopolymers startingor ending inside the feature
type indicated by -ft optionformat: sorted GFF3 (default:
undefined)</p>

<p style="margin-top: 1em">-ft [string] <br>
feature type to use when -ann option is specified (default:
CDS)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help for basic options and exit</p>

<p style="margin-top: 1em">-help+ <br>
display help for all options and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">Correction mode:</p>

<p style="margin-top: 1em">One of the options -aggressive,
-moderate, -conservative or -expert must be selected.</p>

<p style="margin-top: 1em">The -aggressive, -moderate and
-conservative modes are presets of the criteria by which it
is decided if an observed discrepancy in homopolymer length
between cognate sequence <br>
and a read shall be corrected or not. A description of the
single criteria is provided by using the -help+&rsquo;
option. The presets are equivalent to the following
settings:</p>

<p style="margin-top: 1em">-aggressive -moderate
-conservative <br>
-hmin 3 3 3 <br>
-read-hmin 1 1 2 <br>
-altmax 1.00 0.99 0.80 <br>
-refmin 0.00 0.00 0.10 <br>
-mapqmin 0 10 21 <br>
-covmin 1 1 1 <br>
-clenmax unlimited unlimited unlimited <br>
-allow-multiple yes yes no</p>

<p style="margin-top: 1em">The aggressive mode tries to
maximize the sensitivity, the conservative mode to minimize
the false positives. An even more conservative set of
corrections can be achieved using <br>
the -ann option (see -help+).</p>

<p style="margin-top: 1em">The -expert mode allows one to
manually set each parameter; the default values are the same
as in the -conservative mode.</p>

<p style="margin-top: 1em">(Finally, for evaluation
purposes only, the -state-of-truth mode can be used: this
mode assumes that the sequenced genome has been specified as
cognate sequence and outputs an <br>
ideal list of corrections.)</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-HOP(1)</p>
<hr>
</body>
</html>
