<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-LTRHARVEST(1) GenomeTools Manual GT-LTRHARVEST(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-ltrharvest - Predict LTR retrotransposons.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt ltrharvest [option ...] -index &lt;indexname&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-index [string] <br>
specify the name of the enhanced suffix array index
(mandatory) (default: undefined)</p>

<p style="margin-top: 1em">-range [start end] <br>
specify range in the input sequence(s) in which LTR pairs
are searched (default: [0..0])</p>

<p style="margin-top: 1em">-seed [value] <br>
specify minimum seed length for exact repeats (default:
30)</p>

<p style="margin-top: 1em">-minlenltr [value] <br>
specify minimum length for each LTR (default: 100)</p>

<p style="margin-top: 1em">-maxlenltr [value] <br>
specify maximum length for each LTR (default: 1000)</p>

<p style="margin-top: 1em">-mindistltr [value] <br>
specify minimum distance of LTR startpositions (default:
1000)</p>

<p style="margin-top: 1em">-maxdistltr [value] <br>
specify maximum distance of LTR startpositions (default:
15000)</p>

<p style="margin-top: 1em">-similar [value] <br>
specify similaritythreshold in range [1..100%] (default:
85.000000)</p>

<p style="margin-top: 1em">-mintsd [value] <br>
specify minimum length for each TSD (default: 4)</p>

<p style="margin-top: 1em">-maxtsd [value] <br>
specify maximum length for each TSD (default: 20)</p>

<p style="margin-top: 1em">-motif [string] <br>
specify 2 nucleotides startmotif + 2 nucleotides endmotif:
** (default: undefined)</p>

<p style="margin-top: 1em">-motifmis [value] <br>
specify maximum number of mismatches in motif [0,3]
(default: 4)</p>

<p style="margin-top: 1em">-vic [value] <br>
specify the number of nucleotides (to the left and to the
right) that will be searched for TSDs and/or motifs around
5&rsquo; and 3&rsquo; boundary of predicted LTR
retrotransposons <br>
(default: 60)</p>

<p style="margin-top: 1em">-overlaps [...] <br>
specify no|best|all (default: best)</p>

<p style="margin-top: 1em">-xdrop [value] <br>
specify xdropbelowscore for extension-alignment (default:
5)</p>

<p style="margin-top: 1em">-mat [value] <br>
specify matchscore for extension-alignment (default: 2)</p>

<p style="margin-top: 1em">-mis [value] <br>
specify mismatchscore for extension-alignment (default:
-2)</p>

<p style="margin-top: 1em">-ins [value] <br>
specify insertionscore for extension-alignment (default:
-3)</p>

<p style="margin-top: 1em">-del [value] <br>
specify deletionscore for extension-alignment (default:
-3)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
verbose mode (default: no)</p>

<p style="margin-top: 1em">-tabout [yes|no] <br>
show old tabular output instead of GFF3 on stdout (default:
yes)</p>

<p style="margin-top: 1em">-seqids [yes|no] <br>
use sequence descriptions instead of sequence numbers in
GFF3 output (default: no)</p>

<p style="margin-top: 1em">-md5 [yes|no] <br>
add MD5 hashes to seqids in GFF3 output (default: no)</p>

<p style="margin-top: 1em">-longoutput [yes|no] <br>
additional motif/TSD output (default: no)</p>

<p style="margin-top: 1em">-out [string] <br>
specify FASTA outputfilename (default: undefined)</p>

<p style="margin-top: 1em">-outinner [string] <br>
specify FASTA outputfilename for inner regions (default:
undefined)</p>

<p style="margin-top: 1em">-gff3 [string] <br>
specify GFF3 outputfilename (default: undefined)</p>

<p style="margin-top: 1em">-offset [value] <br>
offset added to GFF3 coordinates (default: 0)</p>

<p style="margin-top: 1em">-scan [yes|no] <br>
scan the index sequentially instead of mapping it into
memory entirely (default: yes)</p>

<p style="margin-top: 1em">-help <br>
display help for basic options and exit</p>

<p style="margin-top: 1em">-help+ <br>
display help for all options and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">ADDITIONAL INFORMATION <br>
For detailed information, please refer to the manual of
ltrharvest.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-LTRHARVEST(1)</p>
<hr>
</body>
</html>
