<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-LTRHARVEST(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-LTRHARVEST(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-LTRHARVEST(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-ltrharvest - Predict LTR retrotransposons.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt ltrharvest</b> [option ...] -index &lt;indexname&gt;
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-index</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">specify the name of the enhanced suffix array
  index (mandatory) (default: undefined)</div>
<div class="Pp"></div>
<b>-range</b> [<i>start</i> <i>end</i>]
<div style="margin-left: 4.00ex;">specify range in the input sequence(s) in
  which LTR pairs are searched (default: [0..0])</div>
<div class="Pp"></div>
<b>-seed</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify minimum seed length for exact repeats
  (default: 30)</div>
<div class="Pp"></div>
<b>-minlenltr</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify minimum length for each LTR (default:
  100)</div>
<div class="Pp"></div>
<b>-maxlenltr</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify maximum length for each LTR (default:
  1000)</div>
<div class="Pp"></div>
<b>-mindistltr</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify minimum distance of LTR startpositions
  (default: 1000)</div>
<div class="Pp"></div>
<b>-maxdistltr</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify maximum distance of LTR startpositions
  (default: 15000)</div>
<div class="Pp"></div>
<b>-similar</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify similaritythreshold in range [1..100%]
  (default: 85.000000)</div>
<div class="Pp"></div>
<b>-mintsd</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify minimum length for each TSD (default:
  4)</div>
<div class="Pp"></div>
<b>-maxtsd</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify maximum length for each TSD (default:
  20)</div>
<div class="Pp"></div>
<b>-motif</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">specify 2 nucleotides startmotif + 2
  nucleotides endmotif: <b>*</b>* (default: undefined)</div>
<div class="Pp"></div>
<b>-motifmis</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify maximum number of mismatches in motif
  [0,3] (default: 4)</div>
<div class="Pp"></div>
<b>-vic</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify the number of nucleotides (to the left
  and to the right) that will be searched for TSDs and/or motifs around 5' and
  3' boundary of predicted LTR retrotransposons (default: 60)</div>
<div class="Pp"></div>
<b>-overlaps</b> [<i>...</i>]
<div style="margin-left: 4.00ex;">specify no|best|all (default: best)</div>
<div class="Pp"></div>
<b>-xdrop</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify xdropbelowscore for
  extension-alignment (default: 5)</div>
<div class="Pp"></div>
<b>-mat</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify matchscore for extension-alignment
  (default: 2)</div>
<div class="Pp"></div>
<b>-mis</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify mismatchscore for extension-alignment
  (default: -2)</div>
<div class="Pp"></div>
<b>-ins</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify insertionscore for extension-alignment
  (default: -3)</div>
<div class="Pp"></div>
<b>-del</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify deletionscore for extension-alignment
  (default: -3)</div>
<div class="Pp"></div>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">verbose mode (default: no)</div>
<div class="Pp"></div>
<b>-tabout</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">show <i>old</i> tabular output instead of GFF3
  on stdout (default: yes)</div>
<div class="Pp"></div>
<b>-seqids</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">use sequence descriptions instead of sequence
  numbers in GFF3 output (default: no)</div>
<div class="Pp"></div>
<b>-md5</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">add MD5 hashes to seqids in GFF3 output
  (default: no)</div>
<div class="Pp"></div>
<b>-longoutput</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">additional motif/TSD output (default:
  no)</div>
<div class="Pp"></div>
<b>-out</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">specify FASTA outputfilename (default:
  undefined)</div>
<div class="Pp"></div>
<b>-outinner</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">specify FASTA outputfilename for inner regions
  (default: undefined)</div>
<div class="Pp"></div>
<b>-gff3</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">specify GFF3 outputfilename (default:
  undefined)</div>
<div class="Pp"></div>
<b>-offset</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">offset added to GFF3 coordinates (default:
  0)</div>
<div class="Pp"></div>
<b>-scan</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">scan the index sequentially instead of mapping
  it into memory entirely (default: yes)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help for basic options and exit</div>
<div class="Pp"></div>
<b>-help+</b>
<div style="margin-left: 4.00ex;">display help for all options and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<h1 class="Sh" title="Sh" id="ADDITIONAL_INFORMATION"><a class="selflink" href="#ADDITIONAL_INFORMATION">ADDITIONAL
  INFORMATION</a></h1>
For detailed information, please refer to the manual of ltrharvest.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
