<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-LTRDIGEST(1) GenomeTools Manual GT-LTRDIGEST(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-ltrdigest - Identifies and annotates sequence features in
LTR retrotransposon candidates.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt ltrdigest [option ...] gff3_file</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-outfileprefix [string] <br>
prefix for output files (e.g. foo will create files called
foo_*.csv and foo_*.fas) Omit this option for GFF3 output
only.</p>

<p style="margin-top: 1em">-metadata [yes|no] <br>
output metadata (run conditions) to separate file (default:
yes)</p>

<p style="margin-top: 1em">-seqnamelen [value] <br>
set maximal length of sequence names in FASTA headers (e.g.
for clustalw or similar tools) (default: 20)</p>

<p style="margin-top: 1em">-pptlen [start end] <br>
required PPT length range (default: [8..30])</p>

<p style="margin-top: 1em">-uboxlen [start end] <br>
required U-box length range (default: [3..30])</p>

<p style="margin-top: 1em">-uboxdist [value] <br>
allowed U-box distance range from PPT (default: 0)</p>

<p style="margin-top: 1em">-pptradius [value] <br>
radius around beginning of 3&rsquo; LTR to search for PPT
(default: 30)</p>

<p style="margin-top: 1em">-pptrprob [value] <br>
purine emission probability inside PPT (default:
0.970000)</p>

<p style="margin-top: 1em">-pptyprob [value] <br>
pyrimidine emission probability inside PPT (default:
0.030000)</p>

<p style="margin-top: 1em">-pptgprob [value] <br>
background G emission probability outside PPT (default:
0.250000)</p>

<p style="margin-top: 1em">-pptcprob [value] <br>
background C emission probability outside PPT (default:
0.250000)</p>

<p style="margin-top: 1em">-pptaprob [value] <br>
background A emission probability outside PPT (default:
0.250000)</p>

<p style="margin-top: 1em">-ppttprob [value] <br>
background T emission probability outside PPT (default:
0.250000)</p>

<p style="margin-top: 1em">-pptuprob [value] <br>
U/T emission probability inside U-box (default:
0.910000)</p>

<p style="margin-top: 1em">-trnas [filename] <br>
tRNA library in multiple FASTA format for PBS detection Omit
this option to disable PBS search.</p>

<p style="margin-top: 1em">-pbsalilen [start end] <br>
required PBS/tRNA alignment length range (default:
[11..30])</p>

<p style="margin-top: 1em">-pbsoffset [start end] <br>
allowed PBS offset from LTR boundary range (default:
[0..5])</p>

<p style="margin-top: 1em">-pbstrnaoffset [start end] <br>
allowed PBS/tRNA 3&rsquo; end alignment offset range
(default: [0..5])</p>

<p style="margin-top: 1em">-pbsmaxedist [value] <br>
maximal allowed PBS/tRNA alignment unit edit distance
(default: 1)</p>

<p style="margin-top: 1em">-pbsradius [value] <br>
radius around end of 5&rsquo; LTR to search for PBS
(default: 30)</p>

<p style="margin-top: 1em">-hmms <br>
profile HMM models for domain detection (separate by spaces,
finish with --) in HMMER3 format Omit this option to disable
pHMM search.</p>

<p style="margin-top: 1em">-pdomevalcutoff [value] <br>
global E-value cutoff for pHMM search default 1E-6</p>

<p style="margin-top: 1em">-pdomcutoff [...] <br>
model-specific score cutoff choose from TC (trusted cutoff)
| GA (gathering cutoff) | NONE (no cutoffs) (default:
NONE)</p>

<p style="margin-top: 1em">-aliout [yes|no] <br>
output pHMM to amino acid sequence alignments (default:
no)</p>

<p style="margin-top: 1em">-aaout [yes|no] <br>
output amino acid sequences for protein domain hits
(default: no)</p>

<p style="margin-top: 1em">-allchains [yes|no] <br>
output features from all chains and unchained features,
labeled with chain numbers (default: no)</p>

<p style="margin-top: 1em">-maxgaplen [value] <br>
maximal allowed gap size between fragments (in amino acids)
when chaining pHMM hits for a protein domain (default:
50)</p>

<p style="margin-top: 1em">-force_recreate [yes|no] <br>
force recreation of hmmpressed profiles (default: no)</p>

<p style="margin-top: 1em">-pbsmatchscore [value] <br>
match score for PBS/tRNA alignments (default: 5)</p>

<p style="margin-top: 1em">-pbsmismatchscore [value] <br>
mismatch score for PBS/tRNA alignments (default: -10)</p>

<p style="margin-top: 1em">-pbsinsertionscore [value] <br>
insertion score for PBS/tRNA alignments (default: -20)</p>

<p style="margin-top: 1em">-pbsdeletionscore [value] <br>
deletion score for PBS/tRNA alignments (default: -20)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-o [filename] <br>
redirect output to specified file (default: undefined)</p>

<p style="margin-top: 1em">-gzip [yes|no] <br>
write gzip compressed output file (default: no)</p>

<p style="margin-top: 1em">-bzip2 [yes|no] <br>
write bzip2 compressed output file (default: no)</p>

<p style="margin-top: 1em">-force [yes|no] <br>
force writing to output file (default: no)</p>

<p style="margin-top: 1em">-seqfile [filename] <br>
set the sequence file from which to take the sequences
(default: undefined)</p>

<p style="margin-top: 1em">-encseq [filename] <br>
set the encoded sequence indexname from which to take the
sequences (default: undefined)</p>

<p style="margin-top: 1em">-seqfiles <br>
set the sequence files from which to extract the features
use -- to terminate the list of sequence files</p>

<p style="margin-top: 1em">-matchdesc [yes|no] <br>
search the sequence descriptions from the input files for
the desired sequence IDs (in GFF3), reporting the first
match (default: no)</p>

<p style="margin-top: 1em">-matchdescstart [yes|no] <br>
exactly match the sequence descriptions from the input files
for the desired sequence IDs (in GFF3) from the beginning to
the first whitespace (default: no)</p>

<p style="margin-top: 1em">-usedesc [yes|no] <br>
use sequence descriptions to map the sequence IDs (in GFF3)
to actual sequence entries. If a description contains a
sequence range (e.g., III:1000001..2000000), the first <br>
part is used as sequence ID (III) and the first range
position as offset (1000001) (default: no)</p>

<p style="margin-top: 1em">-regionmapping [string] <br>
set file containing sequence-region to sequence file mapping
(default: undefined)</p>

<p style="margin-top: 1em">-help <br>
display help for basic options and exit</p>

<p style="margin-top: 1em">-help+ <br>
display help for all options and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-LTRDIGEST(1)</p>
<hr>
</body>
</html>
