<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-LTRDIGEST(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-LTRDIGEST(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-LTRDIGEST(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-ltrdigest - Identifies and annotates sequence features in LTR retrotransposon
  candidates.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt ltrdigest</b> [option ...] gff3_file
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-outfileprefix</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">prefix for output files (e.g. <i>foo</i> will
  create files called <i>foo_*.csv</i> and <i>foo_*.fas</i>) Omit this option
  for GFF3 output only.</div>
<div class="Pp"></div>
<b>-metadata</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">output metadata (run conditions) to separate
  file (default: yes)</div>
<div class="Pp"></div>
<b>-seqnamelen</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">set maximal length of sequence names in FASTA
  headers (e.g. for clustalw or similar tools) (default: 20)</div>
<div class="Pp"></div>
<b>-pptlen</b> [<i>start</i> <i>end</i>]
<div style="margin-left: 4.00ex;">required PPT length range (default:
  [8..30])</div>
<div class="Pp"></div>
<b>-uboxlen</b> [<i>start</i> <i>end</i>]
<div style="margin-left: 4.00ex;">required U-box length range (default:
  [3..30])</div>
<div class="Pp"></div>
<b>-uboxdist</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">allowed U-box distance range from PPT
  (default: 0)</div>
<div class="Pp"></div>
<b>-pptradius</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">radius around beginning of 3' LTR to search
  for PPT (default: 30)</div>
<div class="Pp"></div>
<b>-pptrprob</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">purine emission probability inside PPT
  (default: 0.970000)</div>
<div class="Pp"></div>
<b>-pptyprob</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">pyrimidine emission probability inside PPT
  (default: 0.030000)</div>
<div class="Pp"></div>
<b>-pptgprob</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">background G emission probability outside PPT
  (default: 0.250000)</div>
<div class="Pp"></div>
<b>-pptcprob</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">background C emission probability outside PPT
  (default: 0.250000)</div>
<div class="Pp"></div>
<b>-pptaprob</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">background A emission probability outside PPT
  (default: 0.250000)</div>
<div class="Pp"></div>
<b>-ppttprob</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">background T emission probability outside PPT
  (default: 0.250000)</div>
<div class="Pp"></div>
<b>-pptuprob</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">U/T emission probability inside U-box
  (default: 0.910000)</div>
<div class="Pp"></div>
<b>-trnas</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">tRNA library in multiple FASTA format for PBS
  detection Omit this option to disable PBS search.</div>
<div class="Pp"></div>
<b>-pbsalilen</b> [<i>start</i> <i>end</i>]
<div style="margin-left: 4.00ex;">required PBS/tRNA alignment length range
  (default: [11..30])</div>
<div class="Pp"></div>
<b>-pbsoffset</b> [<i>start</i> <i>end</i>]
<div style="margin-left: 4.00ex;">allowed PBS offset from LTR boundary range
  (default: [0..5])</div>
<div class="Pp"></div>
<b>-pbstrnaoffset</b> [<i>start</i> <i>end</i>]
<div style="margin-left: 4.00ex;">allowed PBS/tRNA 3' end alignment offset range
  (default: [0..5])</div>
<div class="Pp"></div>
<b>-pbsmaxedist</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">maximal allowed PBS/tRNA alignment unit edit
  distance (default: 1)</div>
<div class="Pp"></div>
<b>-pbsradius</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">radius around end of 5' LTR to search for PBS
  (default: 30)</div>
<div class="Pp"></div>
<b>-hmms</b>
<div style="margin-left: 4.00ex;">profile HMM models for domain detection
  (separate by spaces, finish with --) in HMMER3 format Omit this option to
  disable pHMM search.</div>
<div class="Pp"></div>
<b>-pdomevalcutoff</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">global E-value cutoff for pHMM search default
  1E-6</div>
<div class="Pp"></div>
<b>-pdomcutoff</b> [<i>...</i>]
<div style="margin-left: 4.00ex;">model-specific score cutoff choose from TC
  (trusted cutoff) | GA (gathering cutoff) | NONE (no cutoffs) (default:
  NONE)</div>
<div class="Pp"></div>
<b>-aliout</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">output pHMM to amino acid sequence alignments
  (default: no)</div>
<div class="Pp"></div>
<b>-aaout</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">output amino acid sequences for protein domain
  hits (default: no)</div>
<div class="Pp"></div>
<b>-allchains</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">output features from all chains and unchained
  features, labeled with chain numbers (default: no)</div>
<div class="Pp"></div>
<b>-maxgaplen</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">maximal allowed gap size between fragments (in
  amino acids) when chaining pHMM hits for a protein domain (default: 50)</div>
<div class="Pp"></div>
<b>-force_recreate</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">force recreation of hmmpressed profiles
  (default: no)</div>
<div class="Pp"></div>
<b>-pbsmatchscore</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">match score for PBS/tRNA alignments (default:
  5)</div>
<div class="Pp"></div>
<b>-pbsmismatchscore</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">mismatch score for PBS/tRNA alignments
  (default: -10)</div>
<div class="Pp"></div>
<b>-pbsinsertionscore</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">insertion score for PBS/tRNA alignments
  (default: -20)</div>
<div class="Pp"></div>
<b>-pbsdeletionscore</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">deletion score for PBS/tRNA alignments
  (default: -20)</div>
<div class="Pp"></div>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">be verbose (default: no)</div>
<div class="Pp"></div>
<b>-o</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">redirect output to specified file (default:
  undefined)</div>
<div class="Pp"></div>
<b>-gzip</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write gzip compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-bzip2</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write bzip2 compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-force</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">force writing to output file (default:
  no)</div>
<div class="Pp"></div>
<b>-seqfile</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">set the sequence file from which to take the
  sequences (default: undefined)</div>
<div class="Pp"></div>
<b>-encseq</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">set the encoded sequence indexname from which
  to take the sequences (default: undefined)</div>
<div class="Pp"></div>
<b>-seqfiles</b>
<div style="margin-left: 4.00ex;">set the sequence files from which to extract
  the features use <i>--</i> to terminate the list of sequence files</div>
<div class="Pp"></div>
<b>-matchdesc</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">search the sequence descriptions from the
  input files for the desired sequence IDs (in GFF3), reporting the first match
  (default: no)</div>
<div class="Pp"></div>
<b>-matchdescstart</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">exactly match the sequence descriptions from
  the input files for the desired sequence IDs (in GFF3) from the beginning to
  the first whitespace (default: no)</div>
<div class="Pp"></div>
<b>-usedesc</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">use sequence descriptions to map the sequence
  IDs (in GFF3) to actual sequence entries. If a description contains a sequence
  range (e.g., III:1000001..2000000), the first part is used as sequence ID (
  <i>III</i>) and the first range position as offset (<i>1000001</i>) (default:
  no)</div>
<div class="Pp"></div>
<b>-regionmapping</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">set file containing sequence-region to
  sequence file mapping (default: undefined)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help for basic options and exit</div>
<div class="Pp"></div>
<b>-help+</b>
<div style="margin-left: 4.00ex;">display help for all options and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
