<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:55 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-READJOINER-ASSE(1) GenomeTools Manual
GT-READJOINER-ASSE(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-readjoiner-assembly - Construct string graph and output
contigs.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt readjoiner assembly [option ...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-readset [string] <br>
specify the readset name (default: undefined)</p>

<p style="margin-top: 1em">-spmfiles [value] <br>
number of SPM files to read this must be equal to the value
of -j for the overlap phase (default: 1)</p>

<p style="margin-top: 1em">-l [value] <br>
specify the minimum SPM length (default: 0)</p>

<p style="margin-top: 1em">-depthcutoff [value] <br>
specify the minimal number of nodes in a contig (default:
3)</p>

<p style="margin-top: 1em">-lengthcutoff [value] <br>
specify the minimal length of a contig (default: 100)</p>

<p style="margin-top: 1em">-errors [yes|no] <br>
search graph features which may originate from sequencing
errors and remove them (default: no)</p>

<p style="margin-top: 1em">-bubble [value] <br>
number of rounds of p-bubble removal to perform (default:
3)</p>

<p style="margin-top: 1em">-deadend [value] <br>
number of rounds of dead end removal to perform a dead end
(default: 10)</p>

<p style="margin-top: 1em">-deadend-depth [value] <br>
specify the maximal depth of a path to an end-vertex by
which the path shall be considered a dead end (default:
10)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-q [yes|no] <br>
suppress standard output messages (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help for basic options and exit</p>

<p style="margin-top: 1em">-help+ <br>
display help for all options and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-READJOINER-ASSE(1)</p>
<hr>
</body>
</html>
