<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:56 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-READJOINER-PREF(1) GenomeTools Manual
GT-READJOINER-PREF(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-readjoiner-prefilter - Remove contained and low-quality
reads and encode read set in GtEncseq format.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt readjoiner prefilter [option ...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-readset [string] <br>
specify the readset name default: filename of first input
sequence_file</p>

<p style="margin-top: 1em">-db <br>
specify a list of input libraries (Fasta/FastQ); for
single-end libraries use the filename (which is not allowed
to contain : symbols); for paired-end libraries with reads
<br>
interleaved (f,r,f,r,...) in a single file use the notation
&lt;filename&gt;:&lt;insertlength&gt;[,&lt;stdev&gt;] (stdev
may be omitted); for paired-end with reads in two files (f,
r) use the <br>
notation
&lt;file_f&gt;:&lt;file_r&gt;:&lt;insertlength&gt;[,&lt;stdev&gt;]</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-q [yes|no] <br>
suppress standard output messages (default: no)</p>

<p style="margin-top: 1em">-des [yes|no] <br>
store Fasta IDs (or entire descriptionsif used together with
-clipdes no) warning: increases the memory requirement
(default: no)</p>

<p style="margin-top: 1em">-clipdes [yes|no] <br>
clip Fasta descriptions after first space set to false if
you need entire descriptions (default: yes)</p>

<p style="margin-top: 1em">-memdes [yes|no] <br>
use memory storage for descriptions (default: use temporary
disk storage)</p>

<p style="margin-top: 1em">-maxlow [value] <br>
maximal number of low-quality positions in a read default:
infinite</p>

<p style="margin-top: 1em">-lowqual [value] <br>
maximal quality for a position to be considered low-quality
(default: 3)</p>

<p style="margin-top: 1em">-phred64 [yes|no] <br>
use phred64 scores for FastQ format (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help for basic options and exit</p>

<p style="margin-top: 1em">-help+ <br>
display help for all options and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-READJOINER-PREF(1)</p>
<hr>
</body>
</html>
