<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:57 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-SELECT(1) GenomeTools Manual GT-SELECT(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-select - Select certain features (specified by the used
options) from given GFF3 file(s).</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt select [option ...] [GFF3_file ...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-retainids [yes|no] <br>
when available, use the original IDs provided in the source
file (memory consumption is proportional to the input file
size(s)) (default: no)</p>

<p style="margin-top: 1em">-seqid [string] <br>
select feature with the given sequence ID (all comments are
selected). (default: undefined)</p>

<p style="margin-top: 1em">-source [string] <br>
select feature with the given source (the source is column 2
in regular GFF3 lines) (default: undefined)</p>

<p style="margin-top: 1em">-contain [start end] <br>
select all features which are contained in the given range
(default: undefined)</p>

<p style="margin-top: 1em">-overlap [start end] <br>
select all features which do overlap with the given range
(default: undefined)</p>

<p style="margin-top: 1em">-strand [string] <br>
select all top-level features(i.e., features without
parents) whose strand equals the given one (must be one of
+-.?) (default: undefined)</p>

<p style="margin-top: 1em">-targetstrand [string] <br>
select all top-level features (i.e., features without
parents) which have exactly one target attribute whose
strand equals the given one (must be one of +-.?) (default:
<br>
undefined)</p>

<p style="margin-top: 1em">-targetbest [yes|no] <br>
if multiple top-level features (i.e., features without
parents) with exactly one target attribute have the same
target_id, keep only the feature with the best score. If
<br>
-targetstrand is used at the same time, this option is
applied after -targetstrand. Memory consumption is
proportional to the input file size(s). (default: no)</p>

<p style="margin-top: 1em">-hascds [yes|no] <br>
select all top-level features which do have a CDS child
(default: no)</p>

<p style="margin-top: 1em">-maxgenelength [value] <br>
select genes up to the given maximum length (default:
undefined)</p>

<p style="margin-top: 1em">-maxgenenum [value] <br>
select the first genes up to the given maximum number
(default: undefined)</p>

<p style="margin-top: 1em">-mingenescore [value] <br>
select genes with the given minimum score (default:
undefined)</p>

<p style="margin-top: 1em">-maxgenescore [value] <br>
select genes with the given maximum score (default:
undefined)</p>

<p style="margin-top: 1em">-minaveragessp [value] <br>
set the minimum average splice site probability (default:
undefined)</p>

<p style="margin-top: 1em">-rule_files <br>
specify Lua filter rule files to be used for selection
(terminate list with --)</p>

<p style="margin-top: 1em">-rule_logic [...] <br>
select how multiple Lua files should be combined choose from
AND|OR (default: AND)</p>

<p style="margin-top: 1em">-dropped_file [filename] <br>
save non-selected features to file (default: undefined)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-o [filename] <br>
redirect output to specified file (default: undefined)</p>

<p style="margin-top: 1em">-gzip [yes|no] <br>
write gzip compressed output file (default: no)</p>

<p style="margin-top: 1em">-bzip2 [yes|no] <br>
write bzip2 compressed output file (default: no)</p>

<p style="margin-top: 1em">-force [yes|no] <br>
force writing to output file (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">File format for option
-rule_files:</p>

<p style="margin-top: 1em">The files supplied to option
-rule_files define a function for filtering by user given
criteria (see example below):</p>

<p style="margin-top: 1em">function filter(gn) <br>
target = &quot;exon&quot; <br>
for curnode in gn:children() do <br>
if (curnode:get_type() == target) then <br>
return false <br>
end <br>
end <br>
return true <br>
end</p>

<p style="margin-top: 1em">The above function iterates over
all children of gn and checks whether there is a node of
type exon. If there is such a node the function returns
false, indicating that the <br>
parent node gn will not be sorted out.</p>

<p style="margin-top: 1em">NOTE: The function must be named
filter and must return false, indicating that the node
survived the filtering process.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-SELECT(1)</p>
<hr>
</body>
</html>
