<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:57 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-SEED_EXTEND(1) GenomeTools Manual
GT-SEED_EXTEND(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-seed_extend - Calculate local alignments using the seed
and extend algorithm.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt seed_extend [option ...] encseq_basename
[encseq_basename]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-ii [string] <br>
Input index for encseq encoded sequences</p>

<p style="margin-top: 1em">-qii [string] <br>
Query input index (encseq)</p>

<p style="margin-top: 1em">-seedlength [value] <br>
Minimum length of a seed default: logarithm of input length
to the basis alphabet size</p>

<p style="margin-top: 1em">-diagbandwidth [value] <br>
Logarithm of diagonal band width (for filter) (default:
6)</p>

<p style="margin-top: 1em">-mincoverage [value] <br>
Minimum coverage in two neighbouring diagonal bands (for
filter) default: 2.5 x seedlength</p>

<p style="margin-top: 1em">-maxfreq [value] <br>
Maximum frequency of a k-mer (for filter) (default:
undefined)</p>

<p style="margin-top: 1em">-memlimit [string] <br>
Maximum memory usage to determine the maximum frequency of a
k-mer (for filter) (default: undefined)</p>

<p style="margin-top: 1em">-extendxdrop [value] <br>
Extend seed to both sides using xdrop algorithm, /noptional
parameter specifies sensitivity (default: 97)</p>

<p style="margin-top: 1em">-xdropbelow [value] <br>
Specify xdrop cutoff score (0 means automatically defined
depending on minidentity)</p>

<p style="margin-top: 1em">-extendgreedy [value] <br>
Extend seed to both sides using greedy algorithm, optional
parameter specifies sensitivity (default: 97)</p>

<p style="margin-top: 1em">-l [value] <br>
Minimum alignment length (for seed extension) (default:
undefined)</p>

<p style="margin-top: 1em">-minidentity [value] <br>
Minimum identity of matches (for seed extension) (default:
80)</p>

<p style="margin-top: 1em">-a [value] <br>
show alignments/sequences (optional argument is number of
columns per line) (default: 70)</p>

<p style="margin-top: 1em">-no-reverse [yes|no] <br>
do not compute matches on reverse complemented strand
(default: no)</p>

<p style="margin-top: 1em">-no-forward [yes|no] <br>
do not compute matches on forward strand (default: no)</p>

<p style="margin-top: 1em">-parts [value] <br>
Divide data into specified number of parts (default: 1)</p>

<p style="margin-top: 1em">-kmerfile [yes|no] <br>
Use pre-calculated k-mers from file (if exist) (default:
yes)</p>

<p style="margin-top: 1em">-v [yes|no] <br>
be verbose (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-SEED_EXTEND(1)</p>
<hr>
</body>
</html>
