<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-SELECT(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-SELECT(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-SELECT(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-select - Select certain features (specified by the used options) from given
  GFF3 file(s).
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt select</b> [option ...] [GFF3_file ...]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-retainids</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">when available, use the original IDs provided
  in the source file (memory consumption is proportional to the input file
  size(s)) (default: no)</div>
<div class="Pp"></div>
<b>-seqid</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">select feature with the given sequence ID (all
  comments are selected). (default: undefined)</div>
<div class="Pp"></div>
<b>-source</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">select feature with the given source (the
  source is column 2 in regular GFF3 lines) (default: undefined)</div>
<div class="Pp"></div>
<b>-contain</b> [<i>start</i> <i>end</i>]
<div style="margin-left: 4.00ex;">select all features which are contained in the
  given range (default: undefined)</div>
<div class="Pp"></div>
<b>-overlap</b> [<i>start</i> <i>end</i>]
<div style="margin-left: 4.00ex;">select all features which do overlap with the
  given range (default: undefined)</div>
<div class="Pp"></div>
<b>-strand</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">select all top-level features(i.e., features
  without parents) whose strand equals the given one (must be one of
  <i>+-.?</i>) (default: undefined)</div>
<div class="Pp"></div>
<b>-targetstrand</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">select all top-level features (i.e., features
  without parents) which have exactly one target attribute whose strand equals
  the given one (must be one of <i>+-.?</i>) (default: undefined)</div>
<div class="Pp"></div>
<b>-targetbest</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">if multiple top-level features (i.e., features
  without parents) with exactly one target attribute have the same target_id,
  keep only the feature with the best score. If -targetstrand is used at the
  same time, this option is applied after -targetstrand. Memory consumption is
  proportional to the input file size(s). (default: no)</div>
<div class="Pp"></div>
<b>-hascds</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">select all top-level features which do have a
  CDS child (default: no)</div>
<div class="Pp"></div>
<b>-maxgenelength</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">select genes up to the given maximum length
  (default: undefined)</div>
<div class="Pp"></div>
<b>-maxgenenum</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">select the first genes up to the given maximum
  number (default: undefined)</div>
<div class="Pp"></div>
<b>-mingenescore</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">select genes with the given minimum score
  (default: undefined)</div>
<div class="Pp"></div>
<b>-maxgenescore</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">select genes with the given maximum score
  (default: undefined)</div>
<div class="Pp"></div>
<b>-minaveragessp</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">set the minimum average splice site
  probability (default: undefined)</div>
<div class="Pp"></div>
<b>-rule_files</b>
<div style="margin-left: 4.00ex;">specify Lua filter rule files to be used for
  selection (terminate list with <i>--</i>)</div>
<div class="Pp"></div>
<b>-rule_logic</b> [<i>...</i>]
<div style="margin-left: 4.00ex;">select how multiple Lua files should be
  combined choose from AND|OR (default: AND)</div>
<div class="Pp"></div>
<b>-dropped_file</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">save non-selected features to file (default:
  undefined)</div>
<div class="Pp"></div>
<b>-v</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">be verbose (default: no)</div>
<div class="Pp"></div>
<b>-o</b> [<i>filename</i>]
<div style="margin-left: 4.00ex;">redirect output to specified file (default:
  undefined)</div>
<div class="Pp"></div>
<b>-gzip</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write gzip compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-bzip2</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">write bzip2 compressed output file (default:
  no)</div>
<div class="Pp"></div>
<b>-force</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">force writing to output file (default:
  no)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<div style="height: 1.00em;">&#x00A0;</div>
File format for option <i>-rule_files</i>:
<div style="height: 1.00em;">&#x00A0;</div>
The files supplied to option <i>-rule_files</i> define a function for filtering
  by user given criteria (see example below):
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
function filter(gn)
  target = &quot;exon&quot;
  for curnode in gn:children() do
    if (curnode:get_type() == target) then
      return false
    end
  end
  return true
end
</pre>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
The above function iterates over all children of <i>gn</i> and checks whether
  there is a node of type <i>exon</i>. If there is such a node the function
  returns <i>false</i>, indicating that the parent node <i>gn</i> will not be
  sorted out.
<div style="height: 1.00em;">&#x00A0;</div>
NOTE: The function must be named <i>filter</i> and must return <i>false</i>,
  indicating that the node survived the filtering process.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
