<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GT-TIRVISH(1) GenomeTools Manual GT-TIRVISH(1)</p>

<p style="margin-top: 1em">NAME <br>
gt-tirvish - Identify Terminal Inverted Repeat (TIR)
elements,such as DNA transposons.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gt tirvish [option ...] -index INDEXNAME</p>

<p style="margin-top: 1em">DESCRIPTION <br>
-index [string] <br>
specify the name of the enhanced suffix array index
(mandatory) (default: undefined)</p>

<p style="margin-top: 1em">-seed [value] <br>
specify minimum seed length for exact repeats (default:
20)</p>

<p style="margin-top: 1em">-mintirlen [value] <br>
specify minimum length for each TIR (default: 100)</p>

<p style="margin-top: 1em">-maxtirlen [value] <br>
specify maximum length for each TIR (default: 1000)</p>

<p style="margin-top: 1em">-mintirdist [value] <br>
specify minimum distance of TIRs (default: 500)</p>

<p style="margin-top: 1em">-maxtirdist [value] <br>
specify maximum distance of TIRs (default: 10000)</p>

<p style="margin-top: 1em">-mat [value] <br>
specify matchscore for extension-alignment (default: 2)</p>

<p style="margin-top: 1em">-mis [value] <br>
specify mismatchscore for extension-alignment (default:
-2)</p>

<p style="margin-top: 1em">-ins [value] <br>
specify insertionscore for extension-alignment (default:
-3)</p>

<p style="margin-top: 1em">-del [value] <br>
specify deletionscore for extension-alignment (default:
-3)</p>

<p style="margin-top: 1em">-xdrop [value] <br>
specify xdropbelowscore for extension-alignment (default:
5)</p>

<p style="margin-top: 1em">-similar [value] <br>
specify TIR similarity threshold in therange [1..100%]
(default: 85.000000)</p>

<p style="margin-top: 1em">-overlaps [...] <br>
specify no|best|longest|all (default: best)</p>

<p style="margin-top: 1em">-mintsd [value] <br>
specify minimum length for each TSD (default: 2)</p>

<p style="margin-top: 1em">-maxtsd [value] <br>
specify maximum length for each TSD (default: 11)</p>

<p style="margin-top: 1em">-vic [value] <br>
specify the number of nucleotides (to the left and to the
right) that will be searched for TSDs around 5&rsquo; and
3&rsquo; boundary of predicted TIRs (default: 60)</p>

<p style="margin-top: 1em">-hmms <br>
profile HMM models for domain detection (separate by spaces,
finish with --) in HMMER3 format Omit this option to disable
pHMM search.</p>

<p style="margin-top: 1em">-pdomevalcutoff [value] <br>
global E-value cutoff for pHMM search default 1E-6</p>

<p style="margin-top: 1em">-pdomcutoff [...] <br>
model-specific score cutoff choose from TC (trusted cutoff)
| GA (gathering cutoff) | NONE (no cutoffs) (default:
GA)</p>

<p style="margin-top: 1em">-maxgaplen [value] <br>
maximal allowed gap size between fragments (in amino acids)
when chaining pHMM hits for a protein domain (default:
50)</p>

<p style="margin-top: 1em">-refseqs [string] <br>
specify the name of the gene sequences to scan for inside
candidates (default: undefined)</p>

<p style="margin-top: 1em">-seqids [yes|no] <br>
use sequence descriptions instead of sequence numbers in
GFF3 output (default: yes)</p>

<p style="margin-top: 1em">-md5 [yes|no] <br>
add MD5 hashes to seqids in GFF3 output (default: no)</p>

<p style="margin-top: 1em">-help <br>
display help for basic options and exit</p>

<p style="margin-top: 1em">-help+ <br>
display help for all options and exit</p>

<p style="margin-top: 1em">-version <br>
display version information and exit</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to &lt;gt-users@genometools.org&gt;.</p>

<p style="margin-top: 1em">GenomeTools 1.5.9 01/10/2017
GT-TIRVISH(1)</p>
<hr>
</body>
</html>
