<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GT-TIRVISH(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GT-TIRVISH(1)</td>
    <td class="head-vol">GenomeTools Manual</td>
    <td class="head-rtitle">GT-TIRVISH(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gt-tirvish - Identify Terminal Inverted Repeat (TIR) elements,such as DNA
  transposons.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>gt tirvish</b> [option ...] -index INDEXNAME
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>-index</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">specify the name of the enhanced suffix array
  index (mandatory) (default: undefined)</div>
<div class="Pp"></div>
<b>-seed</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify minimum seed length for exact repeats
  (default: 20)</div>
<div class="Pp"></div>
<b>-mintirlen</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify minimum length for each TIR (default:
  100)</div>
<div class="Pp"></div>
<b>-maxtirlen</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify maximum length for each TIR (default:
  1000)</div>
<div class="Pp"></div>
<b>-mintirdist</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify minimum distance of TIRs (default:
  500)</div>
<div class="Pp"></div>
<b>-maxtirdist</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify maximum distance of TIRs (default:
  10000)</div>
<div class="Pp"></div>
<b>-mat</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify matchscore for extension-alignment
  (default: 2)</div>
<div class="Pp"></div>
<b>-mis</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify mismatchscore for extension-alignment
  (default: -2)</div>
<div class="Pp"></div>
<b>-ins</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify insertionscore for extension-alignment
  (default: -3)</div>
<div class="Pp"></div>
<b>-del</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify deletionscore for extension-alignment
  (default: -3)</div>
<div class="Pp"></div>
<b>-xdrop</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify xdropbelowscore for
  extension-alignment (default: 5)</div>
<div class="Pp"></div>
<b>-similar</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify TIR similarity threshold in therange
  [1..100%] (default: 85.000000)</div>
<div class="Pp"></div>
<b>-overlaps</b> [<i>...</i>]
<div style="margin-left: 4.00ex;">specify no|best|longest|all (default:
  best)</div>
<div class="Pp"></div>
<b>-mintsd</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify minimum length for each TSD (default:
  2)</div>
<div class="Pp"></div>
<b>-maxtsd</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify maximum length for each TSD (default:
  11)</div>
<div class="Pp"></div>
<b>-vic</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">specify the number of nucleotides (to the left
  and to the right) that will be searched for TSDs around 5' and 3' boundary of
  predicted TIRs (default: 60)</div>
<div class="Pp"></div>
<b>-hmms</b>
<div style="margin-left: 4.00ex;">profile HMM models for domain detection
  (separate by spaces, finish with --) in HMMER3 format Omit this option to
  disable pHMM search.</div>
<div class="Pp"></div>
<b>-pdomevalcutoff</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">global E-value cutoff for pHMM search default
  1E-6</div>
<div class="Pp"></div>
<b>-pdomcutoff</b> [<i>...</i>]
<div style="margin-left: 4.00ex;">model-specific score cutoff choose from TC
  (trusted cutoff) | GA (gathering cutoff) | NONE (no cutoffs) (default:
  GA)</div>
<div class="Pp"></div>
<b>-maxgaplen</b> [<i>value</i>]
<div style="margin-left: 4.00ex;">maximal allowed gap size between fragments (in
  amino acids) when chaining pHMM hits for a protein domain (default: 50)</div>
<div class="Pp"></div>
<b>-refseqs</b> [<i>string</i>]
<div style="margin-left: 4.00ex;">specify the name of the gene sequences to scan
  for inside candidates (default: undefined)</div>
<div class="Pp"></div>
<b>-seqids</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">use sequence descriptions instead of sequence
  numbers in GFF3 output (default: yes)</div>
<div class="Pp"></div>
<b>-md5</b> [<i>yes|no</i>]
<div style="margin-left: 4.00ex;">add MD5 hashes to seqids in GFF3 output
  (default: no)</div>
<div class="Pp"></div>
<b>-help</b>
<div style="margin-left: 4.00ex;">display help for basic options and exit</div>
<div class="Pp"></div>
<b>-help+</b>
<div style="margin-left: 4.00ex;">display help for all options and exit</div>
<div class="Pp"></div>
<b>-version</b>
<div style="margin-left: 4.00ex;">display version information and exit</div>
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to &lt;gt-users@genometools.org&gt;.</div>
<table class="foot">
  <tr>
    <td class="foot-date">01/10/2017</td>
    <td class="foot-os">GenomeTools 1.5.9</td>
  </tr>
</table>
</body>
</html>
