<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:15:47 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GTF2GFF3(1p) User Contributed Perl Documentation
GTF2GFF3(1p)</p>

<p style="margin-top: 1em">if ($i == 0 &amp;&amp;
$START_IN_CDS == 0) { if ($strand == 1) { $exon-&gt;start =
$start-&gt;[0]{start}; } else { $exon-&gt;end =
$start-&gt;[0]{end}; } } if ($i == (scalar @{$CDSs} - 1)
&amp;&amp; <br>
$STOP_IN_CDS == 0) { if ($strand == 1) { $exon-&gt;{end} =
$stop-&gt;[0]{end}; } else { $exon-&gt;{start} =
$stop-&gt;[0]{start} } } NAME <br>
gtf2gff3 - Converts GTF formatted files to valid GFF3
files</p>

<p style="margin-top: 1em">VERSION <br>
This document describes version 0.1</p>

<p style="margin-top: 1em">SYNOPSIS <br>
gtf2gff3 --cfg gtf2gff3_MY_CONFIG.cfg gtf_file &gt;
gff3_file</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This script will convert GTF formatted files to valid GFF3
formatted files. It will map the value in column 3 ( <br>
may appear in that column in GTF files, you may edit the
config file to provide your own GTF feature to SO mapping.
The script will also build gene models from exons, CDSs and
<br>
other features given in the GTF file. It is currently tested
on Ensemble and Twinscan GTF, and it should work on any
other files that follow the same specification. It does not
<br>
work on GTF from the UCSC table browser because those files
use the same ID for gene and transcript, so it is impossible
to group multiple transcripts to a gene. See the README <br>
that came with the script for more info.</p>

<p style="margin-top: 1em">OPTIONS: <br>
--cfg <br>
Provide the filename for a config file. See the
configuration file provided with this script for format
details. Use this configuration file to modify the behavior
of the <br>
script. If no config file is given it looks for
./gtf2gff3.cfg, ~/gtf2gff3.cfg or /etc/gtf2gff3.cfg in that
order.</p>

<p style="margin-top: 1em">--help <br>
Provide a detailed man page style help message and then
exit.</p>

<p style="margin-top: 1em">DIAGNOSTICS <br>
&quot;ERROR: Missing or non-standard attributes:
parse_attributes&quot; <br>
A line in the GTF file did not have any attributes, or
it&rsquo;s attributes column was unparsable.</p>

<p style="margin-top: 1em">&quot;ERROR: Non-transcript gene
feature not supported. Please contact the author for
support: build_gene&quot; <br>
This warning indicates that a line was skipped because it
contained a non-transcript gene feature, and the code is not
currently equipped to handle this type of feature. <br>
This probably isn&rsquo;t too hard to add, so contact me if
you get this error and would like to have these features
supported.</p>

<p style="margin-top: 1em">&quot;ERROR: Must have at least
exons or CDSs to build a transcript: build_trnsc&quot; <br>
Some feature had a transcript_id and yet there were no exons
or CDSs associated with that transcript_id so the script
failed to build a transcript.</p>

<p style="margin-top: 1em">&quot;ERROR: seq_id conflict:
validate_and_finish_trnsc&quot; <br>
Found two features within the same transcript that
didn&rsquo;t share the same seq_id.</p>

<p style="margin-top: 1em">&quot;ERROR: source conflict:
validate_and_finish_trnsc&quot; <br>
Found two features within the same transcript that
didn&rsquo;t share the same source.</p>

<p style="margin-top: 1em">&quot;ERROR: type conflict:
validate_and_finish_trnsc&quot; <br>
Found two features within the same transcript that were
expected to share the same type and yet they
didn&rsquo;t.</p>

<p style="margin-top: 1em">&quot;ERROR: strand conflict:
validate_and_finish_trnsc&quot; <br>
Found two features within the same transcript that
didn&rsquo;t share the same strand.</p>

<p style="margin-top: 1em">&quot;ERROR: seq_id conflict:
validate_and_build_gene&quot; <br>
Found two features within the same gene that didn&rsquo;t
share the same seq_id.</p>

<p style="margin-top: 1em">&quot;ERROR: source conflict:
validate_and_build_gene&quot; <br>
Found two features within the same gene that didn&rsquo;t
share the same source.</p>

<p style="margin-top: 1em">&quot;ERROR: strand conflict:
validate_and_build_gene&quot; <br>
Found two features within the same gene that didn&rsquo;t
share the same strand.</p>

<p style="margin-top: 1em">&quot;ERROR: gene_id conflict:
validate_and_build_gene&quot; <br>
Found two features within the same gene that didn&rsquo;t
share the same gene_id.</p>

<p style="margin-top: 1em">&quot;FATAL: Can&rsquo;t open
GTF file: file_name for reading.&quot; <br>
Unable to open the GTF file for reading.</p>

<p style="margin-top: 1em">&quot;FATAL: Need exons or CDSs
to build transcripts: process_start&quot; <br>
A start_codon feature was annotated and yet there were no
exons or CDSs associated with that transcript_id so the
script failed.</p>

<p style="margin-top: 1em">&quot;FATAL: Untested code in
process_start. Contact the aurthor for support.&quot; <br>
The script is written to infer a start codon based on the
presence of a 5&rsquo; UTR, but we had no example GTF of
this type when we wrote the code, so we killed process
rather <br>
than run untested code. Contact the author for support.</p>

<p style="margin-top: 1em">&quot;FATAL: Invalid feature
set: process_start&quot; <br>
We tried to consider all possible ways of inferring a start
codon or inferring a a non-coding gene, and yet we&rsquo;ve
failed. Your combination of gene features doesn&rsquo;t make
<br>
sense to us. You should never get this error, and if you do,
we&rsquo;d really like to see the GTF file that generated
it. Please contact the author for support.</p>

<p style="margin-top: 1em">&quot;FATAL: Need exons or CDSs
to build transcripts: process_stop&quot; <br>
A stop_codon feature was annotated and yet there were no
exons or CDSs associated with that transcript_id so the
script failed.</p>

<p style="margin-top: 1em">&quot;FATAL: Untested code in
process_stop. Contact the aurthor for support.&quot; <br>
The script is written to infer a stop codon based on the
presence of a 3&rsquo; UTR, but we had no example GTF of
this type when we wrote the code, so we killed process
rather than <br>
run untested code. Contact the author for support.</p>

<p style="margin-top: 1em">&quot;FATAL: Invalid feature
set: process_stop&quot; <br>
We tried to consider all possible ways of inferring a stop
codon or inferring a a non-coding gene, and yet we&rsquo;ve
failed. Your combination of gene features doesn&rsquo;t make
sense <br>
to us. You should never get this error, and if you do,
we&rsquo;d really like to see the GTF file that generated
it. Please contact the author for support.</p>

<p style="margin-top: 1em">&quot;FATAL: Invalid feature
set: process_exon_CDS_UTR&quot; <br>
We tried to consider all possible ways of inferring exons,
CDSs and UTRs and yet we&rsquo;ve failed. Your combination
of gene features doesn&rsquo;t make sense to us. You really
should <br>
ever get this error, and if you do, we&rsquo;d really like
to see the GTF file that generated it. Please contact the
author for support.</p>

<p style="margin-top: 1em">&quot;FATAL: Array reference
required: sort_features.&quot; <br>
A user shouldn&rsquo;t be able to trigger this error. It
almost certainly indicates a software bug. Please contact
the author.</p>

<p style="margin-top: 1em">&quot;FATAL: Can&rsquo;t
determine strand in: sort_feature_types.&quot; <br>
This may indicate that your GTF file does not indicate the
strand for features that require it. It may also indicate a
software bug. Please contact the author.</p>

<p style="margin-top: 1em">&quot;FATAL: Hash reference
required: sort_feature_types.&quot; <br>
A user shouldn&rsquo;t be able to trigger this error. It
almost certainly indicates a software bug. Please contact
the author.</p>

<p style="margin-top: 1em">&quot;FATAL: Invalid value
passed to strand: strand.&quot; <br>
This may indicate that your GTF file does not indicate the
strand for features that require it. Consider using the
DEFAULT_STRAND parameter in the config file. It may also
<br>
indicate a software bug. Please contact the author.</p>

<p style="margin-top: 1em">CONFIGURATION AND ENVIRONMENT
<br>
A configuration file is provided with this script. The
script will look for that configuration file in
./gtf2gff3.cfg, ~/gtf2gff3.cfg or /etc/gtf2gff3.cfg in that
order. If the <br>
configuration file is not found in one of those locations
and one is not provided via the --cfg flag it will try to
choose some sane defaults, but you really should provide the
<br>
configuration file. See the supplied configuration file
itself as well as the README that came with this package for
format and details about the configuration file.</p>

<p style="margin-top: 1em">DEPENDENCIES <br>
This script requires the following perl packages that are
available from CPAN (www.cpan.org).</p>

<p style="margin-top: 1em">Getopt::Long; use
Config::Std;</p>

<p style="margin-top: 1em">INCOMPATIBILITIES <br>
None reported.</p>

<p style="margin-top: 1em">BUGS AND LIMITATIONS <br>
No bugs have been reported.</p>

<p style="margin-top: 1em">Please report any bugs or
feature requests to:
&lt;barry.moore@genetics.utah.edu&gt;</p>

<p style="margin-top: 1em">AUTHOR <br>
Barry Moore &lt;barry.moore@genetics.utah.edu&gt;</p>

<p style="margin-top: 1em">LICENCE AND COPYRIGHT <br>
Copyright (c) 2007, University of Utah</p>

<p style="margin-top: 1em">This module is free software;
you can redistribute it and/or <br>
modify it under the same terms as Perl itself.</p>

<p style="margin-top: 1em">DISCLAIMER OF WARRANTY <br>
BECAUSE THIS SOFTWARE IS LICENSED FREE OF CHARGE, THERE IS
NO WARRANTY FOR THE SOFTWARE, TO THE EXTENT PERMITTED BY
APPLICABLE LAW. EXCEPT WHEN OTHERWISE STATED IN WRITING THE
<br>
COPYRIGHT HOLDERS AND/OR OTHER PARTIES PROVIDE THE SOFTWARE
&quot;AS IS&quot; WITHOUT WARRANTY OF ANY KIND, EITHER
EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED <br>
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE. THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE
OF THE SOFTWARE IS WITH YOU. SHOULD THE SOFTWARE PROVE <br>
DEFECTIVE, YOU ASSUME THE COST OF ALL NECESSARY SERVICING,
REPAIR, OR CORRECTION.</p>

<p style="margin-top: 1em">IN NO EVENT UNLESS REQUIRED BY
APPLICABLE LAW OR AGREED TO IN WRITING WILL ANY COPYRIGHT
HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
REDISTRIBUTE THE SOFTWARE AS PERMITTED <br>
BY THE ABOVE LICENCE, BE LIABLE TO YOU FOR DAMAGES,
INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL
DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE THE
SOFTWARE <br>
(INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD
PARTIES OR A FAILURE OF THE SOFTWARE TO OPERATE WITH ANY
OTHER <br>
SOFTWARE), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN
ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-04-12
GTF2GFF3(1p)</p>
<hr>
</body>
</html>
