<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GTF2GFF3(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GTF2GFF3(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">GTF2GFF3(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="if_($i_==_0__________&amp;&amp;_$START_IN_CDS_==_0)_{_if_($strand_==_1)_{_$exon-&gt;start_=_$start-&gt;[0]{start};_}_else_{_$exon-&gt;end_=_$start-&gt;[0]{end};_}_}_if_($i_==_(scalar_@{$CDSs}_-_1)_&amp;&amp;_$STOP_IN_CDS_==_0)_{_if_($strand_==_1)_{_$exon-&gt;{end}_=_$stop-&gt;[0]{end};_}_else_{_$exon-&gt;{start}_=_$stop-&gt;[0]{start}_}_}"><a class="selflink" href="#if_($i_==_0__________&amp;&amp;_$START_IN_CDS_==_0)_{_if_($strand_==_1)_{_$exon-&gt;start_=_$start-&gt;[0]{start};_}_else_{_$exon-&gt;end_=_$start-&gt;[0]{end};_}_}_if_($i_==_(scalar_@{$CDSs}_-_1)_&amp;&amp;_$STOP_IN_CDS_==_0)_{_if_($strand_==_1)_{_$exon-&gt;{end}_=_$stop-&gt;[0]{end};_}_else_{_$exon-&gt;{start}_=_$stop-&gt;[0]{start}_}_}">
  if ($i == 0 &amp;&amp; $START_IN_CDS == 0) { if ($strand == 1) {
  $exon-&gt;start = $start-&gt;[0]{start}; } else { $exon-&gt;end =
  $start-&gt;[0]{end}; } } if ($i == (scalar @{$CDSs} - 1) &amp;&amp;
  $STOP_IN_CDS == 0) { if ($strand == 1) { $exon-&gt;{end} = $stop-&gt;[0]{end};
  } else { $exon-&gt;{start} = $stop-&gt;[0]{start} } }</a></h1>
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
gtf2gff3 - Converts GTF formatted files to valid GFF3 files
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This document describes version 0.1
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
gtf2gff3 --cfg gtf2gff3_MY_CONFIG.cfg gtf_file &gt; gff3_file
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This script will convert GTF formatted files to valid GFF3 formatted files. It
  will map the value in column 3 (\&quot;type\&quot; column) to valid SO, but
  because many non standard term may appear in that column in GTF files, you may
  edit the config file to provide your own GTF feature to SO mapping. The script
  will also build gene models from exons, CDSs and other features given in the
  GTF file. It is currently tested on Ensemble and Twinscan GTF, and it should
  work on any other files that follow the same specification. It does not work
  on GTF from the UCSC table browser because those files use the same ID for
  gene and transcript, so it is impossible to group multiple transcripts to a
  gene. See the README that came with the script for more info.
<h1 class="Sh" title="Sh" id="OPTIONS:"><a class="selflink" href="#OPTIONS:">OPTIONS:</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">--cfg</dt>
  <dd class="It-tag">Provide the filename for a config file. See the
      configuration file provided with this script for format details. Use this
      configuration file to modify the behavior of the script. If no config file
      is given it looks for ./gtf2gff3.cfg, ~/gtf2gff3.cfg or /etc/gtf2gff3.cfg
      in that order.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">--help</dt>
  <dd class="It-tag">Provide a detailed man page style help message and then
      exit.</dd>
</dl>
<h1 class="Sh" title="Sh" id="DIAGNOSTICS"><a class="selflink" href="#DIAGNOSTICS">DIAGNOSTICS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: Missing or non-standard attributes:
    parse_attributes&quot;</dt>
  <dd class="It-tag">A line in the GTF file did not have any attributes, or it's
      attributes column was unparsable.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: Non-transcript gene feature not supported.
    Please contact the author for support: build_gene&quot;</dt>
  <dd class="It-tag">This warning indicates that a line was skipped because it
      contained a non-transcript gene feature, and the code is not currently
      equipped to handle this type of feature. This probably isn't too hard to
      add, so contact me if you get this error and would like to have these
      features supported.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: Must have at least exons or CDSs to build a
    transcript: build_trnsc&quot;</dt>
  <dd class="It-tag">Some feature had a transcript_id and yet there were no
      exons or CDSs associated with that transcript_id so the script failed to
      build a transcript.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: seq_id conflict:
    validate_and_finish_trnsc&quot;</dt>
  <dd class="It-tag">Found two features within the same transcript that didn't
      share the same seq_id.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: source conflict:
    validate_and_finish_trnsc&quot;</dt>
  <dd class="It-tag">Found two features within the same transcript that didn't
      share the same source.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: type conflict:
    validate_and_finish_trnsc&quot;</dt>
  <dd class="It-tag">Found two features within the same transcript that were
      expected to share the same type and yet they didn't.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: strand conflict:
    validate_and_finish_trnsc&quot;</dt>
  <dd class="It-tag">Found two features within the same transcript that didn't
      share the same strand.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: seq_id conflict:
    validate_and_build_gene&quot;</dt>
  <dd class="It-tag">Found two features within the same gene that didn't share
      the same seq_id.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: source conflict:
    validate_and_build_gene&quot;</dt>
  <dd class="It-tag">Found two features within the same gene that didn't share
      the same source.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: strand conflict:
    validate_and_build_gene&quot;</dt>
  <dd class="It-tag">Found two features within the same gene that didn't share
      the same strand.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;ERROR: gene_id conflict:
    validate_and_build_gene&quot;</dt>
  <dd class="It-tag">Found two features within the same gene that didn't share
      the same gene_id.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Can't open GTF file: file_name for
    reading.&quot;</dt>
  <dd class="It-tag">Unable to open the GTF file for reading.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Need exons or CDSs to build transcripts:
    process_start&quot;</dt>
  <dd class="It-tag">A start_codon feature was annotated and yet there were no
      exons or CDSs associated with that transcript_id so the script
    failed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Untested code in process_start. Contact the
    aurthor for support.&quot;</dt>
  <dd class="It-tag">The script is written to infer a start codon based on the
      presence of a 5' UTR, but we had no example GTF of this type when we wrote
      the code, so we killed process rather than run untested code. Contact the
      author for support.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Invalid feature set: process_start&quot;</dt>
  <dd class="It-tag">We tried to consider all possible ways of inferring a start
      codon or inferring a a non-coding gene, and yet we've failed. Your
      combination of gene features doesn't make sense to us. You should never
      get this error, and if you do, we'd really like to see the GTF file that
      generated it. Please contact the author for support.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Need exons or CDSs to build transcripts:
    process_stop&quot;</dt>
  <dd class="It-tag">A stop_codon feature was annotated and yet there were no
      exons or CDSs associated with that transcript_id so the script
    failed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Untested code in process_stop. Contact the
    aurthor for support.&quot;</dt>
  <dd class="It-tag">The script is written to infer a stop codon based on the
      presence of a 3' UTR, but we had no example GTF of this type when we wrote
      the code, so we killed process rather than run untested code. Contact the
      author for support.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Invalid feature set: process_stop&quot;</dt>
  <dd class="It-tag">We tried to consider all possible ways of inferring a stop
      codon or inferring a a non-coding gene, and yet we've failed. Your
      combination of gene features doesn't make sense to us. You should never
      get this error, and if you do, we'd really like to see the GTF file that
      generated it. Please contact the author for support.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Invalid feature set:
    process_exon_CDS_UTR&quot;</dt>
  <dd class="It-tag">We tried to consider all possible ways of inferring exons,
      CDSs and UTRs and yet we've failed. Your combination of gene features
      doesn't make sense to us. You really should ever get this error, and if
      you do, we'd really like to see the GTF file that generated it. Please
      contact the author for support.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Array reference required:
    sort_features.&quot;</dt>
  <dd class="It-tag">A user shouldn't be able to trigger this error. It almost
      certainly indicates a software bug. Please contact the author.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Can't determine strand in:
    sort_feature_types.&quot;</dt>
  <dd class="It-tag">This may indicate that your GTF file does not indicate the
      strand for features that require it. It may also indicate a software bug.
      Please contact the author.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Hash reference required:
    sort_feature_types.&quot;</dt>
  <dd class="It-tag">A user shouldn't be able to trigger this error. It almost
      certainly indicates a software bug. Please contact the author.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&quot;FATAL: Invalid value passed to strand:
    strand.&quot;</dt>
  <dd class="It-tag">This may indicate that your GTF file does not indicate the
      strand for features that require it. Consider using the DEFAULT_STRAND
      parameter in the config file. It may also indicate a software bug. Please
      contact the author.</dd>
</dl>
<h1 class="Sh" title="Sh" id="CONFIGURATION_AND_ENVIRONMENT"><a class="selflink" href="#CONFIGURATION_AND_ENVIRONMENT">CONFIGURATION
  AND ENVIRONMENT</a></h1>
A configuration file is provided with this script. The script will look for that
  configuration file in ./gtf2gff3.cfg, ~/gtf2gff3.cfg or /etc/gtf2gff3.cfg in
  that order. If the configuration file is not found in one of those locations
  and one is not provided via the --cfg flag it will try to choose some sane
  defaults, but you really should provide the configuration file. See the
  supplied configuration file itself as well as the README that came with this
  package for format and details about the configuration file.
<h1 class="Sh" title="Sh" id="DEPENDENCIES"><a class="selflink" href="#DEPENDENCIES">DEPENDENCIES</a></h1>
This script requires the following perl packages that are available from CPAN
  (www.cpan.org).
<div class="Pp"></div>
Getopt::Long; use Config::Std;
<h1 class="Sh" title="Sh" id="INCOMPATIBILITIES"><a class="selflink" href="#INCOMPATIBILITIES">INCOMPATIBILITIES</a></h1>
None reported.
<h1 class="Sh" title="Sh" id="BUGS_AND_LIMITATIONS"><a class="selflink" href="#BUGS_AND_LIMITATIONS">BUGS
  AND LIMITATIONS</a></h1>
No bugs have been reported.
<div class="Pp"></div>
Please report any bugs or feature requests to:
  &lt;barry.moore@genetics.utah.edu&gt;
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Barry Moore &lt;barry.moore@genetics.utah.edu&gt;
<h1 class="Sh" title="Sh" id="LICENCE_AND_COPYRIGHT"><a class="selflink" href="#LICENCE_AND_COPYRIGHT">LICENCE
  AND COPYRIGHT</a></h1>
Copyright (c) 2007, University of Utah
<div class="Pp"></div>
<pre>
    This module is free software; you can redistribute it and/or
    modify it under the same terms as Perl itself.
</pre>
<h1 class="Sh" title="Sh" id="DISCLAIMER_OF_WARRANTY"><a class="selflink" href="#DISCLAIMER_OF_WARRANTY">DISCLAIMER
  OF WARRANTY</a></h1>
BECAUSE THIS SOFTWARE IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY FOR THE
  SOFTWARE, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT WHEN OTHERWISE
  STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES PROVIDE THE
  SOFTWARE &quot;AS IS&quot; WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR
  IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE ENTIRE RISK AS TO
  THE QUALITY AND PERFORMANCE OF THE SOFTWARE IS WITH YOU. SHOULD THE SOFTWARE
  PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL NECESSARY SERVICING, REPAIR, OR
  CORRECTION.
<div class="Pp"></div>
IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING WILL ANY
  COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR REDISTRIBUTE THE
  SOFTWARE AS PERMITTED BY THE ABOVE LICENCE, BE LIABLE TO YOU FOR DAMAGES,
  INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES ARISING
  OUT OF THE USE OR INABILITY TO USE THE SOFTWARE (INCLUDING BUT NOT LIMITED TO
  LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR
  THIRD PARTIES OR A FAILURE OF THE SOFTWARE TO OPERATE WITH ANY OTHER
  SOFTWARE), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE
  POSSIBILITY OF SUCH DAMAGES.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-04-12</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
