<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:12 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>GWAMA(1) General Commands Manual GWAMA(1)</p>

<p style="margin-top: 1em">NAME <br>
GWAMA - Genome-Wide Association Meta Analysis</p>

<p style="margin-top: 1em">SYNOPSIS <br>
GWAMA [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
GWAMA (Genome-Wide Association Meta Analysis) software
performs meta-analysis of the results of GWA studies of
binary or quantitative phenotypes. Fixed- and random-effect
meta- <br>
analyses are performed for both directly genotyped and
imputed SNPs using estimates of the allelic odds ratio and
95% confidence interval for binary traits, and estimates of
the <br>
allelic effect size and standard error for quantitative
phenotypes. GWAMA can be used for analysing the results of
all different genetic models (multiplicative, additive,
domi&acirc; <br>
nant, recessive). The software incorporates error trapping
facilities to identify strand alignment errors and allele
flipping, and performs tests of heterogeneity of effects
<br>
between studies.</p>

<p style="margin-top: 1em">OPTIONS <br>
-i, --filelist=filename <br>
Specify studies&rsquo; result files. Default = gwama.in</p>

<p style="margin-top: 1em">-o, --output=fileroot <br>
Specify file root for output of analysis. Default = gwama
(gwama.out, gwama.gc.out)</p>

<p style="margin-top: 1em">-r, --random <br>
Use random effect correction. Default = disabled</p>

<p style="margin-top: 1em">-gc, --genomic_control <br>
Use genomic control for adjusting studies&rsquo; result
files. Default = disabled</p>

<p style="margin-top: 1em">-gco, --genomic_control_output
<br>
Use genomic control on meta-analysis summary (i.e. results
of meta- analysis are corrected for gc). Default =
disabled</p>

<p style="margin-top: 1em">-qt, --quantitative <br>
Select quantitative trait version (BETA and SE columns).
Default = binary trait</p>

<p style="margin-top: 1em">-m, --map=filename <br>
Select file name for marker map.</p>

<p style="margin-top: 1em">-t, --threshold=0-1 <br>
The p-value threshold for showing direction in summary
effect directions. Default = 1</p>

<p style="margin-top: 1em">--no_alleles <br>
No allele information has been given. Expecting always the
same EA.</p>

<p style="margin-top: 1em">--indel_alleles <br>
Allele labes might contain more than single letter. No
strand checks.</p>

<p style="margin-top: 1em">--sex Run gender-differentiated
and gender- heterogeneity analysis (method described in
paper Magi, Lindgren &amp; Morris 2010). Gender info must be
provided in filelist file. (sec&acirc; <br>
ond column after file names is either M or F).</p>

<p style="margin-top: 1em">--filter <br>
Set a filtering based on column header. It needs 3
arguments: column name, equation
[&gt;,&lt;,&gt;=,&lt;=,==,!=], filter value. Multiple
filters can be set.</p>

<p style="margin-top: 1em">--name_marker <br>
Alternative header to marker name col.</p>

<p style="margin-top: 1em">--name_strand <br>
Alternative header to strand column.</p>

<p style="margin-top: 1em">--name_n <br>
Alternative header to sample size col.</p>

<p style="margin-top: 1em">--name_ea <br>
Alternative header to effect allele column.</p>

<p style="margin-top: 1em">--name_nea <br>
Alternative header to non-effect allele column.</p>

<p style="margin-top: 1em">--name_eaf <br>
Alternative header to effect allele frequency column.</p>

<p style="margin-top: 1em">--name_beta <br>
Alternative header to beta column.</p>

<p style="margin-top: 1em">--name_se <br>
Alternative header to std. err. col.</p>

<p style="margin-top: 1em">--name_or <br>
Alternative header to OR column.</p>

<p style="margin-top: 1em">--name_or_95l <br>
Alternative header to OR 95L column.</p>

<p style="margin-top: 1em">--name_or_95u <br>
Alternative header to OR 95U column.</p>

<p style="margin-top: 1em">-h, --help <br>
Print this help.</p>

<p style="margin-top: 1em">-v, --version <br>
Print GWAMA version number. This manual page must be
insufficient. Please use the help option for a quick
reminder of the options of gwama or turn to its home page
with <br>
the online documentation or the downloadable manual.</p>

<p style="margin-top: 1em">SEE ALSO <br>
http://www.geenivaramu.ee/en/tools/gwama</p>

<p style="margin-top: 1em">AUTHOR <br>
GWAMA was written by Reedik Magi and Andrew P Morris.</p>

<p style="margin-top: 1em">2.2.2 August 2016 GWAMA(1)</p>
<hr>
</body>
</html>
