<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>GWAMA(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">GWAMA(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">GWAMA(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
GWAMA - Genome-Wide Association Meta Analysis
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>GWAMA</b> [ <i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
GWAMA (Genome-Wide Association Meta Analysis) software performs meta-analysis of
  the results of GWA studies of binary or quantitative phenotypes. Fixed- and
  random-effect meta-analyses are performed for both directly genotyped and
  imputed SNPs using estimates of the allelic odds ratio and 95% confidence
  interval for binary traits, and estimates of the allelic effect size and
  standard error for quantitative phenotypes. GWAMA can be used for analysing
  the results of all different genetic models (multiplicative, additive,
  dominant, recessive). The software incorporates error trapping facilities to
  identify strand alignment errors and allele flipping, and performs tests of
  heterogeneity of effects between studies.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b>, <b>--filelist</b>=<i>filename</i></dt>
  <dd class="It-tag">Specify studies' result files. Default = gwama.in</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>, <b>--output</b>=<i>fileroot</i></dt>
  <dd class="It-tag">Specify file root for output of analysis. Default = gwama
      (gwama.out, gwama.gc.out)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b>, <b>--random</b></dt>
  <dd class="It-tag">Use random effect correction. Default = disabled</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gc</b>, <b>--genomic_control</b></dt>
  <dd class="It-tag">Use genomic control for adjusting studies' result files.
      Default = disabled</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gco</b>, <b>--genomic_control_output</b></dt>
  <dd class="It-tag">Use genomic control on meta-analysis summary (i.e. results
      of meta- analysis are corrected for gc). Default = disabled</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qt</b>, <b>--quantitative</b></dt>
  <dd class="It-tag">Select quantitative trait version (BETA and SE columns).
      Default = binary trait</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>, <b>--map</b>=<i>filename</i></dt>
  <dd class="It-tag">Select file name for marker map.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>, <b>--threshold</b>=<i>0-1</i></dt>
  <dd class="It-tag">The p-value threshold for showing direction in summary
      effect directions. Default = 1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no_alleles</b></dt>
  <dd class="It-tag">No allele information has been given. Expecting always the
      same EA.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--indel_alleles</b></dt>
  <dd class="It-tag">Allele labes might contain more than single letter. No
      strand checks.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sex</b></dt>
  <dd class="It-tag">Run gender-differentiated and gender- heterogeneity
      analysis (method described in paper Magi, Lindgren &amp; Morris 2010).
      Gender info must be provided in filelist file. (second column after file
      names is either M or F).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--filter</b></dt>
  <dd class="It-tag">Set a filtering based on column header. It needs 3
      arguments: column name, equation [&gt;,&lt;,&gt;=,&lt;=,==,!=], filter
      value. Multiple filters can be set.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_marker</b></dt>
  <dd class="It-tag">Alternative header to marker name col.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_strand</b></dt>
  <dd class="It-tag">Alternative header to strand column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_n</b></dt>
  <dd class="It-tag">Alternative header to sample size col.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_ea</b></dt>
  <dd class="It-tag">Alternative header to effect allele column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_nea</b></dt>
  <dd class="It-tag">Alternative header to non-effect allele column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_eaf</b></dt>
  <dd class="It-tag">Alternative header to effect allele frequency column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_beta</b></dt>
  <dd class="It-tag">Alternative header to beta column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_se</b></dt>
  <dd class="It-tag">Alternative header to std. err. col.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_or</b></dt>
  <dd class="It-tag">Alternative header to OR column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_or_95l</b></dt>
  <dd class="It-tag">Alternative header to OR 95L column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--name_or_95u</b></dt>
  <dd class="It-tag">Alternative header to OR 95U column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">Print this help.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--version</b></dt>
  <dd class="It-tag">Print GWAMA version number. This manual page must be
      insufficient. Please use the help option for a quick reminder of the
      options of gwama or turn to its home page with the online documentation or
      the downloadable manual.</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>http://www.geenivaramu.ee/en/tools/gwama</b>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
GWAMA was written by Reedik Magi and Andrew P Morris.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">August 2016</td>
    <td class="foot-os">2.2.2</td>
  </tr>
</table>
</body>
</html>
