<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>HARVESTTOOLS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">HARVESTTOOLS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">HARVESTTOOLS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
harvesttools - archiving and postprocessing for reference-compressed genomic
  multi-alignments
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>harvesttools</b> &lt;options&gt;
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Harvesttools is a utility for creating and interfacing with Gingr files, which
  are efficient archives that the Harvest Suite uses to store
  reference-compressed multi-alignments, phylogenetic trees, filtered variants
  and annotations. Though designed for use with Parsnp and Gingr, HarvestTools
  can also be used for generic conversion between standard bioinformatics file
  formats.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-i</b>
  &lt;Gingr input&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-b</b>
  &lt;bed filter intervals&gt;,&lt;filter
  name&gt;,&quot;&lt;description&gt;&quot;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-B</b>
  &lt;output backbone intervals&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-f</b>
  &lt;reference fasta&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-F</b>
  &lt;reference fasta out&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-g</b>
  &lt;reference genbank&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-a</b>
  &lt;MAF alignment input&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-m</b>
  &lt;multi-fasta alignment input&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-M</b>
  &lt;multi-fasta alignment output (concatenated LCBs)&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-n</b>
  &lt;Newick tree input&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-N</b>
  &lt;Newick tree output&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--midpoint-reroot</b>
  (reroot the tree at its midpoint after loading)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  &lt;Gingr output&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-S</b>
  &lt;output for multi-fasta SNPs&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-u</b> 0/1
  (update the branch values to reflect genome length)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-v</b>
  &lt;VCF input&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-V</b>
  &lt;VCF output&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-x</b>
  &lt;xmfa alignment file&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-X</b>
  &lt;output xmfa alignment file&gt;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-h</b>
  (show this help)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-q</b>
  (quiet mode)</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">March 2016</td>
    <td class="foot-os">harvesttools 1.2</td>
  </tr>
</table>
</body>
</html>
