<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:24 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>HARVESTTOOLS(1) User Commands HARVESTTOOLS(1)</p>

<p style="margin-top: 1em">NAME <br>
harvesttools - archiving and postprocessing for
reference-compressed genomic multi-alignments</p>

<p style="margin-top: 1em">SYNOPSIS <br>
harvesttools &lt;options&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Harvesttools is a utility for creating and interfacing with
Gingr files, which are efficient archives that the Harvest
Suite uses to store reference-compressed multi-alignments,
<br>
phylogenetic trees, filtered variants and annotations.
Though designed for use with Parsnp and Gingr, HarvestTools
can also be used for generic conversion between standard
bioin&acirc; <br>
formatics file formats.</p>

<p style="margin-top: 1em">OPTIONS <br>
-i &lt;Gingr input&gt;</p>

<p style="margin-top: 1em">-b &lt;bed filter
intervals&gt;,&lt;filter
name&gt;,&quot;&lt;description&gt;&quot;</p>

<p style="margin-top: 1em">-B &lt;output backbone
intervals&gt;</p>

<p style="margin-top: 1em">-f &lt;reference fasta&gt;</p>

<p style="margin-top: 1em">-F &lt;reference fasta
out&gt;</p>

<p style="margin-top: 1em">-g &lt;reference genbank&gt;</p>

<p style="margin-top: 1em">-a &lt;MAF alignment
input&gt;</p>

<p style="margin-top: 1em">-m &lt;multi-fasta alignment
input&gt;</p>

<p style="margin-top: 1em">-M &lt;multi-fasta alignment
output (concatenated LCBs)&gt;</p>

<p style="margin-top: 1em">-n &lt;Newick tree input&gt;</p>

<p style="margin-top: 1em">-N &lt;Newick tree
output&gt;</p>

<p style="margin-top: 1em">--midpoint-reroot (reroot the
tree at its midpoint after loading)</p>

<p style="margin-top: 1em">-o &lt;Gingr output&gt;</p>

<p style="margin-top: 1em">-S &lt;output for multi-fasta
SNPs&gt;</p>

<p style="margin-top: 1em">-u 0/1 (update the branch values
to reflect genome length)</p>

<p style="margin-top: 1em">-v &lt;VCF input&gt;</p>

<p style="margin-top: 1em">-V &lt;VCF output&gt;</p>

<p style="margin-top: 1em">-x &lt;xmfa alignment
file&gt;</p>

<p style="margin-top: 1em">-X &lt;output xmfa alignment
file&gt;</p>

<p style="margin-top: 1em">-h (show this help)</p>

<p style="margin-top: 1em">-q (quiet mode)</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">harvesttools 1.2 March 2016
HARVESTTOOLS(1)</p>
<hr>
</body>
</html>
