<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:46 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>HHCONSENSUS(1) User Commands HHCONSENSUS(1)</p>

<p style="margin-top: 1em">NAME <br>
hhconsensus - calculate the consensus sequence for an
A3M/FASTA input file</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hhconsensus -i &lt;file&gt; [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
HHconsensus 3.0.0 (15-03-2015) Calculate the consensus
sequence for an A3M/FASTA input file. (C) Johannes Soeding,
Michael Remmert, Andreas Biegert, Andreas Hauser Remmert M,
<br>
Biegert A, Hauser A, and Soding J. HHblits: Lightning-fast
iterative protein sequence searching by HMM-HMM alignment.
Nat. Methods 9:173-175 (2011).</p>

<p style="margin-top: 1em">-i &lt;file&gt; <br>
query alignment (A2M, A3M, or FASTA), or query HMM</p>

<p style="margin-top: 1em">Output options: <br>
-s &lt;file&gt; <br>
append consensus sequence in FASTA
(default=&lt;infile.seq&gt;)</p>

<p style="margin-top: 1em">-o &lt;file&gt; <br>
write alignment with consensus sequence in A3M</p>

<p style="margin-top: 1em">-oa3m &lt;file&gt; <br>
same</p>

<p style="margin-top: 1em">-oa2m &lt;file&gt; <br>
write alignment with consensus sequence in A2M</p>

<p style="margin-top: 1em">-ofas &lt;file&gt; <br>
write alignment with consensus sequence in FASTA</p>

<p style="margin-top: 1em">-v &lt;int&gt; <br>
verbose mode: 0:no screen output 1:only warings 2:
verbose</p>

<p style="margin-top: 1em">Filter input alignment (options
can be combined): <br>
-id [0,100] maximum pairwise sequence identity (%)
(def=100)</p>

<p style="margin-top: 1em">-diff [0,inf[ filter most
diverse set of sequences, keeping at least this</p>

<p style="margin-top: 1em">many sequences in each block of
&gt;50 columns (def=0)</p>

<p style="margin-top: 1em">-cov [0,100] minimum coverage
with query (%) (def=0)</p>

<p style="margin-top: 1em">-qid [0,100] minimum sequence
identity with query (%) (def=0)</p>

<p style="margin-top: 1em">-qsc [0,100] minimum score per
column with query (def=-20.0)</p>

<p style="margin-top: 1em">Input alignment format: <br>
-M a2m use A2M/A3M (default): upper case = Match; lower case
= Insert; &rsquo;-&rsquo; = Delete; &rsquo;.&rsquo; = gaps
aligned to inserts (may be omitted)</p>

<p style="margin-top: 1em">-M first <br>
use FASTA: columns with residue in 1st sequence are match
states</p>

<p style="margin-top: 1em">-M [0,100] <br>
use FASTA: columns with fewer than X% gaps are match
states</p>

<p style="margin-top: 1em">Other options: <br>
-maxres &lt;int&gt; max number of HMM columns
(def=20001)</p>

<p style="margin-top: 1em">Example: hhconsensus -i stdin -s
stdout</p>

<p style="margin-top: 1em">hhconsensus 3.0~beta2 January
2017 HHCONSENSUS(1)</p>
<hr>
</body>
</html>
