<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:47 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>HHFILTER(1) User Commands HHFILTER(1)</p>

<p style="margin-top: 1em">NAME <br>
hhfilter - filter an alignment by maximum sequence identity
of match states and minimum coverage</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hhfilter -i infile -o outfile [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
HHfilter 3.0.0 (15-03-2015) Filter an alignment by maximum
pairwise sequence identity, minimum coverage, minimum
sequence identity, or score per column to the first (seed)
<br>
sequence.n(C) Johannes Soeding, Michael Remmert, Andreas
Biegert, Andreas Hauser Remmert M, Biegert A, Hauser A, and
Soding J. HHblits: Lightning-fast iterative protein sequence
<br>
searching by HMM-HMM alignment. Nat. Methods 9:173-175
(2011).</p>

<p style="margin-top: 1em">-i &lt;file&gt; <br>
read input file in A3M/A2M or FASTA format</p>

<p style="margin-top: 1em">-o &lt;file&gt; <br>
write to output file in A3M format</p>

<p style="margin-top: 1em">-a &lt;file&gt; <br>
append to output file in A3M format</p>

<p style="margin-top: 1em">OPTIONS <br>
-v &lt;int&gt; <br>
verbose mode: 0:no screen output 1:only warings 2:
verbose</p>

<p style="margin-top: 1em">-id [0,100] maximum pairwise
sequence identity (%) (def=90)</p>

<p style="margin-top: 1em">-diff [0,inf[ <br>
filter MSA by selecting most diverse set of sequences,
keeping at least this many seqs in each MSA block of length
50 (def=0)</p>

<p style="margin-top: 1em">-cov [0,100] minimum coverage
with query (%) (def=0)</p>

<p style="margin-top: 1em">-qid [0,100] minimum sequence
identity with query (%) (def=0)</p>

<p style="margin-top: 1em">-qsc [0,100] minimum score per
column with query (def=-20.0)</p>

<p style="margin-top: 1em">-neff [1,inf] <br>
target diversity of alignment (default=off)</p>

<p style="margin-top: 1em">Input alignment format: <br>
-M a2m use A2M/A3M (default): upper case = Match; lower case
= Insert; &rsquo;-&rsquo; = Delete; &rsquo;.&rsquo; = gaps
aligned to inserts (may be omitted)</p>

<p style="margin-top: 1em">-M first <br>
use FASTA: columns with residue in 1st sequence are match
states</p>

<p style="margin-top: 1em">-M [0,100] <br>
use FASTA: columns with fewer than X% gaps are match
states</p>

<p style="margin-top: 1em">Example: hhfilter -id 50 -i
d1mvfd_.a2m -o d1mvfd_.fil.a2m</p>

<p style="margin-top: 1em">hhfilter 3.0~beta2 January 2017
HHFILTER(1)</p>
<hr>
</body>
</html>
