<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>HHFILTER(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">HHFILTER(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">HHFILTER(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
hhfilter - filter an alignment by maximum sequence identity of match states and
  minimum coverage
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hhfilter</b> <i>-i infile -o outfile </i>[<i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
HHfilter 3.0.0 (15-03-2015) Filter an alignment by maximum pairwise sequence
  identity, minimum coverage, minimum sequence identity, or score per column to
  the first (seed) sequence.n(C) Johannes Soeding, Michael Remmert, Andreas
  Biegert, Andreas Hauser Remmert M, Biegert A, Hauser A, and Soding J. HHblits:
  Lightning-fast iterative protein sequence searching by HMM-HMM alignment. Nat.
  Methods 9:173-175 (2011).
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> &lt;file&gt;</dt>
  <dd class="It-tag">read input file in A3M/A2M or FASTA format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> &lt;file&gt;</dt>
  <dd class="It-tag">write to output file in A3M format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> &lt;file&gt;</dt>
  <dd class="It-tag">append to output file in A3M format</dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b> &lt;int&gt;</dt>
  <dd class="It-tag">verbose mode: 0:no screen output 1:only warings 2:
    verbose</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-id</b></dt>
  <dd class="It-tag">[0,100] maximum pairwise sequence identity (%)
    (def=90)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-diff</b> [0,inf[</dt>
  <dd class="It-tag">filter MSA by selecting most diverse set of sequences,
      keeping at least this many seqs in each MSA block of length 50
    (def=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cov</b></dt>
  <dd class="It-tag">[0,100] minimum coverage with query (%) (def=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qid</b></dt>
  <dd class="It-tag">[0,100] minimum sequence identity with query (%)
    (def=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qsc</b></dt>
  <dd class="It-tag">[0,100] minimum score per column with query
    (def=-20.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-neff</b> [1,inf]</dt>
  <dd class="It-tag">target diversity of alignment (default=off)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_alignment_format:"><a class="selflink" href="#Input_alignment_format:">Input
  alignment format:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> a2m</dt>
  <dd class="It-tag">use A2M/A3M (default): upper case = Match; lower case =
      Insert; '-' = Delete; '.' = gaps aligned to inserts (may be omitted)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> first</dt>
  <dd class="It-tag">use FASTA: columns with residue in 1st sequence are match
      states</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> [0,100]</dt>
  <dd class="It-tag">use FASTA: columns with fewer than X% gaps are match
    states</dd>
</dl>
<div class="Pp"></div>
Example: hhfilter <b>-id</b> 50 <b>-i</b> d1mvfd_.a2m <b>-o</b>
  d1mvfd_.fil.a2m</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">hhfilter 3.0~beta2</td>
  </tr>
</table>
</body>
</html>
