<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>HHMAKE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">HHMAKE(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">HHMAKE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
hhmake - build an HMM from an input alignment or convert between HMMER format
  and HHsearch format
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hhmake</b> <i>-i file </i>[<i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
HHmake 3.0.0 (15-03-2015) Build an HMM from an input alignment in A2M, A3M, or
  FASTA format, or convert between HMMER format (.hmm) and HHsearch format
  (.hhm). Remmert M, Biegert A, Hauser A, and Soding J. HHblits: Lightning-fast
  iterative protein sequence searching by HMM-HMM alignment. Nat. Methods
  9:173-175 (2011). (C) Johannes Soeding, Michael Remmert, Andreas Biegert,
  Andreas Hauser
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> &lt;file&gt;</dt>
  <dd class="It-tag">query alignment (A2M, A3M, or FASTA), or query HMM</dd>
</dl>
<h2 class="Ss" title="Ss" id="Output_options:"><a class="selflink" href="#Output_options:">Output
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> &lt;file&gt;</dt>
  <dd class="It-tag">HMM file to be written to (default=&lt;infile.hhm&gt;)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> &lt;file&gt;</dt>
  <dd class="It-tag">HMM file to be appended to</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b> &lt;int&gt;</dt>
  <dd class="It-tag">verbose mode: 0:no screen output 1:only warings 2:
    verbose</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-seq</b> &lt;int&gt;</dt>
  <dd class="It-tag">max. number of query/template sequences displayed (def=10)
      Beware of overflows! All these sequences are stored in memory.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cons</b></dt>
  <dd class="It-tag">make consensus sequence master sequence of query MSA</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-name</b> &lt;name&gt;</dt>
  <dd class="It-tag">use this name for HMM (default: use name of first
    sequence)</dd>
</dl>
<div class="Pp"></div>
Filter query multiple sequence alignment
<dl class="Bl-tag">
  <dt class="It-tag"><b>-id</b></dt>
  <dd class="It-tag">[0,100] maximum pairwise sequence identity (%)
    (def=90)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-diff</b> [0,inf[</dt>
  <dd class="It-tag">filter MSA by selecting most diverse set of sequences,
      keeping at least this many seqs in each MSA block of length 50
    (def=100)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cov</b></dt>
  <dd class="It-tag">[0,100] minimum coverage with query (%) (def=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qid</b></dt>
  <dd class="It-tag">[0,100] minimum sequence identity with query (%)
    (def=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qsc</b></dt>
  <dd class="It-tag">[0,100] minimum score per column with query
    (def=-20.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-neff</b> [1,inf]</dt>
  <dd class="It-tag">target diversity of alignment (default=off)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_alignment_format:"><a class="selflink" href="#Input_alignment_format:">Input
  alignment format:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> a2m</dt>
  <dd class="It-tag">use A2M/A3M (default): upper case = Match; lower case =
      Insert; '-' = Delete; '.' = gaps aligned to inserts (may be omitted)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> first</dt>
  <dd class="It-tag">use FASTA: columns with residue in 1st sequence are match
      states</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> [0,100]</dt>
  <dd class="It-tag">use FASTA: columns with fewer than X% gaps are match
    states</dd>
</dl>
<h2 class="Ss" title="Ss" id="Pseudocount_(pc)_options:"><a class="selflink" href="#Pseudocount_(pc)_options:">Pseudocount
  (pc) options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Context specific hhm pseudocounts:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_contxt_mode</b> {0,..,3}</dt>
  <dd class="It-tag">position dependence of pc admixture 'tau' (pc mode,
      default=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">0: no pseudo counts:</dt>
  <dd class="It-tag">tau = 0</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">1: constant</dt>
  <dd class="It-tag">tau = a</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">2: diversity-dependent: tau = a/(1+((Neff[i]-1)/b)^c) 3:
      CSBlast admixture: tau = a(1+b)/(Neff[i]+b) (Neff[i]: number of effective
      seqs in local MSA around column i)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_contxt_a</b></dt>
  <dd class="It-tag">[0,1] overall pseudocount admixture (def=0.9)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_contxt_b</b></dt>
  <dd class="It-tag">[1,inf[ Neff threshold value for mode 2 (def=4.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_contxt_c</b></dt>
  <dd class="It-tag">[0,3] extinction exponent c for mode 2 (def=1.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Context independent hhm pseudocounts (used for templates;
      used for query if contxt file is not available):</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_nocontxt_mode</b> {0,..,3}</dt>
  <dd class="It-tag">position dependence of pc admixture 'tau' (pc mode,
      default=2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">0: no pseudo counts:</dt>
  <dd class="It-tag">tau = 0</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">1: constant</dt>
  <dd class="It-tag">tau = a</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">2: diversity-dependent: tau = a/(1+((Neff[i]-1)/b)^c)
      (Neff[i]: number of effective seqs in local MSA around column i)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_nocontxt_a</b></dt>
  <dd class="It-tag">[0,1] overall pseudocount admixture (def=1.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_nocontxt_b</b></dt>
  <dd class="It-tag">[1,inf[ Neff threshold value for mode 2 (def=1.5)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_nocontxt_c</b></dt>
  <dd class="It-tag">[0,3] extinction exponent c for mode 2 (def=1.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Context-specific pseudo-counts:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nocontxt</b></dt>
  <dd class="It-tag">use substitution-matrix instead of context-specific
      pseudocounts</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-contxt</b>
  &lt;file&gt; context file for computing context-specific pseudocounts
  (default=./data/context_data.crf)</div>
<h2 class="Ss" title="Ss" id="Other_options:"><a class="selflink" href="#Other_options:">Other
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-maxres</b> &lt;int&gt;</dt>
  <dd class="It-tag">max number of HMM columns (def=20001)</dd>
</dl>
<div class="Pp"></div>
Example: hhmake <b>-i</b> test.a3m</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">hhmake 3.0~beta2</td>
  </tr>
</table>
</body>
</html>
