<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:47 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>HHSEARCH(1) User Commands HHSEARCH(1)</p>

<p style="margin-top: 1em">NAME <br>
hhsearch - search a database of HMMs with a query alignment
or query HMM</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hhsearch -i query -d database [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
HHsearch 3.0.0 (15-03-2015) Search a database of HMMs with a
query alignment or query HMM (C) Johannes Soeding, Michael
Remmert, Andreas Biegert, Andreas Hauser Soding, J.
Pro&acirc; <br>
tein homology detection by HMM-HMM comparison.
Bioinformatics 21:951-960 (2005).</p>

<p style="margin-top: 1em">-i &lt;file&gt; <br>
input/query multiple sequence alignment (a2m, a3m, FASTA) or
HMM</p>

<p style="margin-top: 1em">&lt;file&gt; may be
&rsquo;stdin&rsquo; or &rsquo;stdout&rsquo; throughout.
Options:</p>

<p style="margin-top: 1em">-d &lt;name&gt; <br>
database name (e.g. uniprot20_29Feb2012) Multiple databases
may be specified with &rsquo;-d &lt;db1&gt; -d &lt;db2&gt;
...&rsquo;</p>

<p style="margin-top: 1em">-e [0,1] E-value cutoff for
inclusion in result alignment (def=0.001)</p>

<p style="margin-top: 1em">Input alignment format: <br>
-M a2m use A2M/A3M (default): upper case = Match; lower case
= Insert;</p>

<p style="margin-top: 1em">&rsquo;-&rsquo; = Delete;
&rsquo;.&rsquo; = gaps aligned to inserts (may be
omitted)</p>

<p style="margin-top: 1em">-M first <br>
use FASTA: columns with residue in 1st sequence are match
states</p>

<p style="margin-top: 1em">-M [0,100] <br>
use FASTA: columns with fewer than X% gaps are match
states</p>

<p style="margin-top: 1em">-tags/-notags <br>
do NOT / do neutralize His-, C-myc-, FLAG-tags, and trypsin
recognition sequence to background distribution
(def=-notags)</p>

<p style="margin-top: 1em">Output options: <br>
-o &lt;file&gt; <br>
write results in standard format to file
(default=&lt;infile.hhr&gt;)</p>

<p style="margin-top: 1em">-oa3m &lt;file&gt; <br>
write result MSA with significant matches in a3m format</p>

<p style="margin-top: 1em">-opsi &lt;file&gt; <br>
write result MSA of significant matches in PSI-BLAST
format</p>

<p style="margin-top: 1em">-ohhm &lt;file&gt; <br>
write HHM file for result MSA of significant matches</p>

<p style="margin-top: 1em">-add_cons <br>
generate consensus sequence as master sequence of query MSA
(default=don&rsquo;t)</p>

<p style="margin-top: 1em">-hide_cons <br>
don&rsquo;t show consensus sequence in alignments
(default=show)</p>

<p style="margin-top: 1em">-hide_pred <br>
don&rsquo;t show predicted 2ndary structure in alignments
(default=show)</p>

<p style="margin-top: 1em">-hide_dssp <br>
don&rsquo;t show DSSP 2ndary structure in alignments
(default=show)</p>

<p style="margin-top: 1em">-show_ssconf <br>
show confidences for predicted 2ndary structure in
alignments</p>

<p style="margin-top: 1em">-Ofas &lt;file&gt; <br>
write pairwise alignments in FASTA xor A2M (-Oa2m) xor A3M
(-Oa3m) format</p>

<p style="margin-top: 1em">-seq &lt;int&gt; <br>
max. number of query/template sequences displayed
(default=1)</p>

<p style="margin-top: 1em">-aliw &lt;int&gt; <br>
number of columns per line in alignment list
(default=80)</p>

<p style="margin-top: 1em">-p [0,100] <br>
minimum probability in summary and alignment list
(default=20)</p>

<p style="margin-top: 1em">-E [0,inf[ <br>
maximum E-value in summary and alignment list
(default=1E+06)</p>

<p style="margin-top: 1em">-Z &lt;int&gt; <br>
maximum number of lines in summary hit list
(default=500)</p>

<p style="margin-top: 1em">-z &lt;int&gt; <br>
minimum number of lines in summary hit list (default=10)</p>

<p style="margin-top: 1em">-B &lt;int&gt; <br>
maximum number of alignments in alignment list
(default=500)</p>

<p style="margin-top: 1em">-b &lt;int&gt; <br>
minimum number of alignments in alignment list
(default=10)</p>

<p style="margin-top: 1em">Filter options applied to query
MSA, database MSAs, and result MSA</p>

<p style="margin-top: 1em">-all show all sequences in
result MSA; do not filter result MSA</p>

<p style="margin-top: 1em">-id [0,100] maximum pairwise
sequence identity (def=90)</p>

<p style="margin-top: 1em">-diff [0,inf[ <br>
filter MSAs by selecting most diverse set of sequences,
keeping at least this many seqs in each MSA block of length
50 Zero and non-numerical values turn off the filter&acirc;
<br>
ing. (def=100)</p>

<p style="margin-top: 1em">-cov [0,100] minimum coverage
with master sequence (%) (def=0)</p>

<p style="margin-top: 1em">-qid [0,100] minimum sequence
identity with master sequence (%) (def=0)</p>

<p style="margin-top: 1em">-qsc [0,100] minimum score per
column with master sequence (default=-20.0)</p>

<p style="margin-top: 1em">-neff [1,inf] <br>
target diversity of multiple sequence alignment
(default=off)</p>

<p style="margin-top: 1em">-mark do not filter out
sequences marked by &quot;&gt;@&quot;in their name line</p>

<p style="margin-top: 1em">HMM-HMM alignment options: <br>
-norealign <br>
do NOT realign displayed hits with MAC algorithm
(def=realign)</p>

<p style="margin-top: 1em">-ovlp &lt;int&gt; <br>
banded alignment: forbid &lt;ovlp&gt; largest diagonals
|i-j| of DP matrix (def=0)</p>

<p style="margin-top: 1em">-mact [0,1[ <br>
posterior prob threshold for MAC realignment controlling
greediness at alignment ends: 0:global &gt;0.1:local
(default=0.35)</p>

<p style="margin-top: 1em">-glob/-loc <br>
use global/local alignment mode for searching/ranking
(def=local)</p>

<p style="margin-top: 1em">-realign <br>
realign displayed hits with max. accuracy (MAC)
algorithm</p>

<p style="margin-top: 1em">-excl &lt;range&gt; <br>
exclude query positions from the alignment, e.g.
&rsquo;1-33,97-168&rsquo;</p>

<p style="margin-top: 1em">-realign_max &lt;int&gt; <br>
realign max. &lt;int&gt; hits (default=500)</p>

<p style="margin-top: 1em">-alt &lt;int&gt; <br>
show up to this many alternative alignments with raw score
&gt; smin(def=4)</p>

<p style="margin-top: 1em">-smin &lt;float&gt; <br>
minimum raw score for alternative alignments (def=20.0)</p>

<p style="margin-top: 1em">-shift [-1,1] <br>
profile-profile score offset (def=-0.03)</p>

<p style="margin-top: 1em">-corr [0,1] <br>
weight of term for pair correlations (def=0.10)</p>

<p style="margin-top: 1em">-sc &lt;int&gt; amino acid score
(tja: template HMM at column j) (def=1)</p>

<p style="margin-top: 1em">0 = log2 Sum(tja*qia/pa) (pa: aa
background frequencies)</p>

<p style="margin-top: 1em">1 = log2 Sum(tja*qia/pqa) (pqa =
1/2*(pa+ta) )</p>

<p style="margin-top: 1em">2 = log2 Sum(tja*qia/ta) (ta:
av. aa freqs in template)</p>

<p style="margin-top: 1em">3 = log2 Sum(tja*qia/qa) (qa:
av. aa freqs in query)</p>

<p style="margin-top: 1em">5 local amino acid composition
correction</p>

<p style="margin-top: 1em">-ssm {0,..,4} <br>
0: no ss scoring</p>

<p style="margin-top: 1em">1,2: ss scoring after or during
alignment <br>
[default=2]</p>

<p style="margin-top: 1em">3,4: ss scoring after or during
alignment, predicted vs. predicted</p>

<p style="margin-top: 1em">-ssw [0,1] <br>
weight of ss score (def=0.11)</p>

<p style="margin-top: 1em">-ssa [0,1] <br>
SS substitution matrix = (1-ssa)*I +
ssa*full-SS-substition-matrix [def=1.00)</p>

<p style="margin-top: 1em">-wg use global sequence
weighting for realignment!</p>

<p style="margin-top: 1em">Gap cost options: <br>
-gapb [0,inf[ <br>
Transition pseudocount admixture (def=1.00)</p>

<p style="margin-top: 1em">-gapd [0,inf[ <br>
Transition pseudocount admixture for open gap
(default=0.15)</p>

<p style="margin-top: 1em">-gape [0,1.5] <br>
Transition pseudocount admixture for extend gap
(def=1.00)</p>

<p style="margin-top: 1em">-gapf ]0,inf] <br>
factor to increase/reduce gap open penalty for deletes
(def=0.60)</p>

<p style="margin-top: 1em">-gapg ]0,inf] <br>
factor to increase/reduce gap open penalty for inserts
(def=0.60)</p>

<p style="margin-top: 1em">-gaph ]0,inf] <br>
factor to increase/reduce gap extend penalty for
deletes(def=0.60)</p>

<p style="margin-top: 1em">-gapi ]0,inf] <br>
factor to increase/reduce gap extend penalty for
inserts(def=0.60)</p>

<p style="margin-top: 1em">-egq [0,inf[ penalty (bits) for
end gaps aligned to query residues (def=0.00)</p>

<p style="margin-top: 1em">-egt [0,inf[ penalty (bits) for
end gaps aligned to template residues (def=0.00)</p>

<p style="margin-top: 1em">Pseudocount (pc) options: <br>
Context specific hhm pseudocounts:</p>

<p style="margin-top: 1em">-pc_hhm_contxt_mode {0,..,3}
<br>
position dependence of pc admixture &rsquo;tau&rsquo; (pc
mode, default=2)</p>

<p style="margin-top: 1em">0: no pseudo counts: <br>
tau = 0</p>

<p style="margin-top: 1em">1: constant <br>
tau = a</p>

<p style="margin-top: 1em">2: diversity-dependent: tau =
a/(1+((Neff[i]-1)/b)^c) 3: CSBlast admixture: tau =
a(1+b)/(Neff[i]+b) (Neff[i]: number of effective seqs in
local MSA around column i)</p>

<p style="margin-top: 1em">-pc_hhm_contxt_a <br>
[0,1] overall pseudocount admixture (def=0.9)</p>

<p style="margin-top: 1em">-pc_hhm_contxt_b <br>
[1,inf[ Neff threshold value for mode 2 (def=4.0)</p>

<p style="margin-top: 1em">-pc_hhm_contxt_c <br>
[0,3] extinction exponent c for mode 2 (def=1.0)</p>

<p style="margin-top: 1em">Context independent hhm
pseudocounts (used for templates; used for query if contxt
file is not available):</p>

<p style="margin-top: 1em">-pc_hhm_nocontxt_mode {0,..,3}
<br>
position dependence of pc admixture &rsquo;tau&rsquo; (pc
mode, default=2)</p>

<p style="margin-top: 1em">0: no pseudo counts: <br>
tau = 0</p>

<p style="margin-top: 1em">1: constant <br>
tau = a</p>

<p style="margin-top: 1em">2: diversity-dependent: tau =
a/(1+((Neff[i]-1)/b)^c) (Neff[i]: number of effective seqs
in local MSA around column i)</p>

<p style="margin-top: 1em">-pc_hhm_nocontxt_a <br>
[0,1] overall pseudocount admixture (def=1.0)</p>

<p style="margin-top: 1em">-pc_hhm_nocontxt_b <br>
[1,inf[ Neff threshold value for mode 2 (def=1.5)</p>

<p style="margin-top: 1em">-pc_hhm_nocontxt_c <br>
[0,3] extinction exponent c for mode 2 (def=1.0)</p>

<p style="margin-top: 1em">Context-specific
pseudo-counts:</p>

<p style="margin-top: 1em">-nocontxt <br>
use substitution-matrix instead of context-specific
pseudocounts</p>

<p style="margin-top: 1em">-contxt &lt;file&gt; context
file for computing context-specific pseudocounts
(default=./data/context_data.crf)</p>

<p style="margin-top: 1em">-csw [0,inf] weight of central
position in cs pseudocount mode (def=1.6)</p>

<p style="margin-top: 1em">-csb [0,1] weight decay
parameter for positions in cs pc mode (def=0.9)</p>

<p style="margin-top: 1em">Other options: <br>
-v &lt;int&gt; <br>
verbose mode: 0:no screen output 1:only warings 2: verbose
(def=2)</p>

<p style="margin-top: 1em">-cpu &lt;int&gt; <br>
number of CPUs to use (for shared memory SMPs)
(default=2)</p>

<p style="margin-top: 1em">-scores &lt;file&gt; write
scores for all pairwise comparisons to file</p>

<p style="margin-top: 1em">-atab &lt;file&gt; write all
alignments in tabular layout to file</p>

<p style="margin-top: 1em">-maxres &lt;int&gt; <br>
max number of HMM columns (def=20001)</p>

<p style="margin-top: 1em">-maxmem [1,inf[ limit memory for
realignment (in GB) (def=3.0)</p>

<p style="margin-top: 1em">Example: hhsearch -i a.1.1.1.a3m
-d scop70_1.71</p>

<p style="margin-top: 1em">hhsearch 3.0~beta2 January 2017
HHSEARCH(1)</p>
<hr>
</body>
</html>
