<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>HHSEARCH(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">HHSEARCH(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">HHSEARCH(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
hhsearch - search a database of HMMs with a query alignment or query HMM
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hhsearch</b> <i>-i query -d database </i>[<i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
HHsearch 3.0.0 (15-03-2015) Search a database of HMMs with a query alignment or
  query HMM (C) Johannes Soeding, Michael Remmert, Andreas Biegert, Andreas
  Hauser Soding, J. Protein homology detection by HMM-HMM comparison.
  Bioinformatics 21:951-960 (2005).
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> &lt;file&gt;</dt>
  <dd class="It-tag">input/query multiple sequence alignment (a2m, a3m, FASTA)
      or HMM</dd>
</dl>
<div class="Pp"></div>
&lt;file&gt; may be 'stdin' or 'stdout' throughout. Options:
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> &lt;name&gt;</dt>
  <dd class="It-tag">database name (e.g. uniprot20_29Feb2012) Multiple databases
      may be specified with '-d &lt;db1&gt; <b>-d</b> &lt;db2&gt; ...'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b></dt>
  <dd class="It-tag">[0,1] E-value cutoff for inclusion in result alignment
      (def=0.001)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_alignment_format:"><a class="selflink" href="#Input_alignment_format:">Input
  alignment format:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> a2m</dt>
  <dd class="It-tag">use A2M/A3M (default): upper case = Match; lower case =
      Insert;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">'-' = Delete; '.' = gaps aligned to inserts (may be
      omitted)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> first</dt>
  <dd class="It-tag">use FASTA: columns with residue in 1st sequence are match
      states</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> [0,100]</dt>
  <dd class="It-tag">use FASTA: columns with fewer than X% gaps are match
    states</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-tags</b>/-notags</dt>
  <dd class="It-tag">do NOT / do neutralize His-, C-myc-, FLAG-tags, and trypsin
      recognition sequence to background distribution (def=-notags)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Output_options:"><a class="selflink" href="#Output_options:">Output
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> &lt;file&gt;</dt>
  <dd class="It-tag">write results in standard format to file
      (default=&lt;infile.hhr&gt;)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-oa3m</b> &lt;file&gt;</dt>
  <dd class="It-tag">write result MSA with significant matches in a3m
    format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-opsi</b> &lt;file&gt;</dt>
  <dd class="It-tag">write result MSA of significant matches in PSI-BLAST
    format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ohhm</b> &lt;file&gt;</dt>
  <dd class="It-tag">write HHM file for result MSA of significant matches</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-add_cons</b></dt>
  <dd class="It-tag">generate consensus sequence as master sequence of query MSA
      (default=don't)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-hide_cons</b></dt>
  <dd class="It-tag">don't show consensus sequence in alignments
    (default=show)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-hide_pred</b></dt>
  <dd class="It-tag">don't show predicted 2ndary structure in alignments
      (default=show)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-hide_dssp</b></dt>
  <dd class="It-tag">don't show DSSP 2ndary structure in alignments
      (default=show)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-show_ssconf</b></dt>
  <dd class="It-tag">show confidences for predicted 2ndary structure in
      alignments</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Ofas</b> &lt;file&gt;</dt>
  <dd class="It-tag">write pairwise alignments in FASTA xor A2M (<b>-Oa2m</b>)
      xor A3M ( <b>-Oa3m</b>) format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-seq</b> &lt;int&gt;</dt>
  <dd class="It-tag">max. number of query/template sequences displayed
      (default=1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-aliw</b> &lt;int&gt;</dt>
  <dd class="It-tag">number of columns per line in alignment list
    (default=80)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> [0,100]</dt>
  <dd class="It-tag">minimum probability in summary and alignment list
      (default=20)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b> [0,inf[</dt>
  <dd class="It-tag">maximum E-value in summary and alignment list
      (default=1E+06)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b> &lt;int&gt;</dt>
  <dd class="It-tag">maximum number of lines in summary hit list
    (default=500)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-z</b> &lt;int&gt;</dt>
  <dd class="It-tag">minimum number of lines in summary hit list
    (default=10)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b> &lt;int&gt;</dt>
  <dd class="It-tag">maximum number of alignments in alignment list
      (default=500)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b> &lt;int&gt;</dt>
  <dd class="It-tag">minimum number of alignments in alignment list
    (default=10)</dd>
</dl>
<div class="Pp"></div>
Filter options applied to query MSA, database MSAs, and result MSA
<dl class="Bl-tag">
  <dt class="It-tag"><b>-all</b></dt>
  <dd class="It-tag">show all sequences in result MSA; do not filter result
    MSA</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-id</b></dt>
  <dd class="It-tag">[0,100] maximum pairwise sequence identity (def=90)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-diff</b> [0,inf[</dt>
  <dd class="It-tag">filter MSAs by selecting most diverse set of sequences,
      keeping at least this many seqs in each MSA block of length 50 Zero and
      non-numerical values turn off the filtering. (def=100)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cov</b></dt>
  <dd class="It-tag">[0,100] minimum coverage with master sequence (%)
    (def=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qid</b></dt>
  <dd class="It-tag">[0,100] minimum sequence identity with master sequence (%)
      (def=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qsc</b></dt>
  <dd class="It-tag">[0,100] minimum score per column with master sequence
      (default=-20.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-neff</b> [1,inf]</dt>
  <dd class="It-tag">target diversity of multiple sequence alignment
      (default=off)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mark</b></dt>
  <dd class="It-tag">do not filter out sequences marked by &quot;&gt;@&quot;in
      their name line</dd>
</dl>
<h2 class="Ss" title="Ss" id="HMM-HMM_alignment_options:"><a class="selflink" href="#HMM-HMM_alignment_options:">HMM-HMM
  alignment options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-norealign</b></dt>
  <dd class="It-tag">do NOT realign displayed hits with MAC algorithm
      (def=realign)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ovlp</b> &lt;int&gt;</dt>
  <dd class="It-tag">banded alignment: forbid &lt;ovlp&gt; largest diagonals
      |i-j| of DP matrix (def=0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mact</b> [0,1[</dt>
  <dd class="It-tag">posterior prob threshold for MAC realignment controlling
      greediness at alignment ends: 0:global &gt;0.1:local (default=0.35)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-glob</b>/-loc</dt>
  <dd class="It-tag">use global/local alignment mode for searching/ranking
      (def=local)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-realign</b></dt>
  <dd class="It-tag">realign displayed hits with max. accuracy (MAC)
    algorithm</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-excl</b> &lt;range&gt;</dt>
  <dd class="It-tag">exclude query positions from the alignment, e.g.
      '1-33,97-168'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-realign_max</b> &lt;int&gt;</dt>
  <dd class="It-tag">realign max. &lt;int&gt; hits (default=500)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-alt</b> &lt;int&gt;</dt>
  <dd class="It-tag">show up to this many alternative alignments with raw score
      &gt; smin(def=4)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-smin</b> &lt;float&gt;</dt>
  <dd class="It-tag">minimum raw score for alternative alignments
    (def=20.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-shift</b> [-1,1]</dt>
  <dd class="It-tag">profile-profile score offset (def=-0.03)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-corr</b> [0,1]</dt>
  <dd class="It-tag">weight of term for pair correlations (def=0.10)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sc</b></dt>
  <dd class="It-tag">&lt;int&gt; amino acid score (tja: template HMM at column
      j) (def=1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">0</dt>
  <dd class="It-tag">= log2 Sum(tja*qia/pa) (pa: aa background frequencies)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">1</dt>
  <dd class="It-tag">= log2 Sum(tja*qia/pqa) (pqa = 1/2*(pa+ta) )</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">2</dt>
  <dd class="It-tag">= log2 Sum(tja*qia/ta) (ta: av. aa freqs in template)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">3</dt>
  <dd class="It-tag">= log2 Sum(tja*qia/qa) (qa: av. aa freqs in query)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">5</dt>
  <dd class="It-tag">local amino acid composition correction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ssm</b> {0,..,4}</dt>
  <dd class="It-tag">0: no ss scoring</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">1,2: ss scoring after or during alignment</dt>
  <dd class="It-tag">[default=2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">3,4: ss scoring after or during alignment, predicted vs.
      predicted</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ssw</b> [0,1]</dt>
  <dd class="It-tag">weight of ss score (def=0.11)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ssa</b> [0,1]</dt>
  <dd class="It-tag">SS substitution matrix = (1-ssa)*I +
      ssa*full-SS-substition-matrix [def=1.00)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wg</b></dt>
  <dd class="It-tag">use global sequence weighting for realignment!</dd>
</dl>
<h2 class="Ss" title="Ss" id="Gap_cost_options:"><a class="selflink" href="#Gap_cost_options:">Gap
  cost options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gapb</b> [0,inf[</dt>
  <dd class="It-tag">Transition pseudocount admixture (def=1.00)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gapd</b> [0,inf[</dt>
  <dd class="It-tag">Transition pseudocount admixture for open gap
      (default=0.15)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gape</b> [0,1.5]</dt>
  <dd class="It-tag">Transition pseudocount admixture for extend gap
    (def=1.00)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gapf</b> ]0,inf]</dt>
  <dd class="It-tag">factor to increase/reduce gap open penalty for deletes
      (def=0.60)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gapg</b> ]0,inf]</dt>
  <dd class="It-tag">factor to increase/reduce gap open penalty for inserts
      (def=0.60)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gaph</b> ]0,inf]</dt>
  <dd class="It-tag">factor to increase/reduce gap extend penalty for
      deletes(def=0.60)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gapi</b> ]0,inf]</dt>
  <dd class="It-tag">factor to increase/reduce gap extend penalty for
      inserts(def=0.60)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-egq</b></dt>
  <dd class="It-tag">[0,inf[ penalty (bits) for end gaps aligned to query
      residues (def=0.00)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-egt</b></dt>
  <dd class="It-tag">[0,inf[ penalty (bits) for end gaps aligned to template
      residues (def=0.00)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Pseudocount_(pc)_options:"><a class="selflink" href="#Pseudocount_(pc)_options:">Pseudocount
  (pc) options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Context specific hhm pseudocounts:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_contxt_mode</b> {0,..,3}</dt>
  <dd class="It-tag">position dependence of pc admixture 'tau' (pc mode,
      default=2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">0: no pseudo counts:</dt>
  <dd class="It-tag">tau = 0</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">1: constant</dt>
  <dd class="It-tag">tau = a</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">2: diversity-dependent: tau = a/(1+((Neff[i]-1)/b)^c) 3:
      CSBlast admixture: tau = a(1+b)/(Neff[i]+b) (Neff[i]: number of effective
      seqs in local MSA around column i)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_contxt_a</b></dt>
  <dd class="It-tag">[0,1] overall pseudocount admixture (def=0.9)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_contxt_b</b></dt>
  <dd class="It-tag">[1,inf[ Neff threshold value for mode 2 (def=4.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_contxt_c</b></dt>
  <dd class="It-tag">[0,3] extinction exponent c for mode 2 (def=1.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Context independent hhm pseudocounts (used for templates;
      used for query if contxt file is not available):</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_nocontxt_mode</b> {0,..,3}</dt>
  <dd class="It-tag">position dependence of pc admixture 'tau' (pc mode,
      default=2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">0: no pseudo counts:</dt>
  <dd class="It-tag">tau = 0</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">1: constant</dt>
  <dd class="It-tag">tau = a</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">2: diversity-dependent: tau = a/(1+((Neff[i]-1)/b)^c)
      (Neff[i]: number of effective seqs in local MSA around column i)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_nocontxt_a</b></dt>
  <dd class="It-tag">[0,1] overall pseudocount admixture (def=1.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_nocontxt_b</b></dt>
  <dd class="It-tag">[1,inf[ Neff threshold value for mode 2 (def=1.5)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pc_hhm_nocontxt_c</b></dt>
  <dd class="It-tag">[0,3] extinction exponent c for mode 2 (def=1.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Context-specific pseudo-counts:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nocontxt</b></dt>
  <dd class="It-tag">use substitution-matrix instead of context-specific
      pseudocounts</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-contxt</b>
  &lt;file&gt; context file for computing context-specific pseudocounts
  (default=./data/context_data.crf)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-csw</b></dt>
  <dd class="It-tag">[0,inf] weight of central position in cs pseudocount mode
      (def=1.6)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-csb</b></dt>
  <dd class="It-tag">[0,1] weight decay parameter for positions in cs pc mode
      (def=0.9)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Other_options:"><a class="selflink" href="#Other_options:">Other
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b> &lt;int&gt;</dt>
  <dd class="It-tag">verbose mode: 0:no screen output 1:only warings 2: verbose
      (def=2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cpu</b> &lt;int&gt;</dt>
  <dd class="It-tag">number of CPUs to use (for shared memory SMPs)
    (default=2)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-scores</b>
  &lt;file&gt; write scores for all pairwise comparisons to file</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-atab</b></dt>
  <dd class="It-tag">&lt;file&gt; write all alignments in tabular layout to
    file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-maxres</b> &lt;int&gt;</dt>
  <dd class="It-tag">max number of HMM columns (def=20001)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-maxmem</b>
  [1,inf[ limit memory for realignment (in GB) (def=3.0)</div>
<div class="Pp"></div>
Example: hhsearch <b>-i</b> a.1.1.1.a3m <b>-d</b> scop70_1.71</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">hhsearch 3.0~beta2</td>
  </tr>
</table>
</body>
</html>
