<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:50 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>HISAT2-ALIGN-L(1) User Commands HISAT2-ALIGN-L(1)</p>

<p style="margin-top: 1em">NAME <br>
hisat2-align-l - HISAT2 graph-based alignment of short
nucleotide reads to many genomes, large index binary</p>

<p style="margin-top: 1em">DESCRIPTION <br>
HISAT2 version 2.0.5 by Daehwan Kim (infphilo@gmail.com,
www.ccb.jhu.edu/people/infphilo) Usage:</p>

<p style="margin-top: 1em">hisat2-align [options]* -x
&lt;ht2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt;}
[-S &lt;sam&gt;]</p>

<p style="margin-top: 1em">&lt;ht2-idx&gt; <br>
Index filename prefix (minus trailing .X.ht2l).</p>

<p style="margin-top: 1em">&lt;m1&gt; Files with #1 mates,
paired with files in &lt;m2&gt;.</p>

<p style="margin-top: 1em">&lt;m2&gt; Files with #2 mates,
paired with files in &lt;m1&gt;.</p>

<p style="margin-top: 1em">&lt;r&gt; Files with unpaired
reads.</p>

<p style="margin-top: 1em">&lt;sam&gt; File for SAM output
(default: stdout)</p>

<p style="margin-top: 1em">&lt;m1&gt;, &lt;m2&gt;,
&lt;r&gt; can be comma-separated lists (no whitespace) and
can be specified many times. E.g. &rsquo;-U
file1.fq,file2.fq -U file3.fq&rsquo;.</p>

<p style="margin-top: 1em">Options (defaults in
parentheses):</p>

<p style="margin-top: 1em">Input:</p>

<p style="margin-top: 1em">-q query input files are FASTQ
.fq/.fastq (default)</p>

<p style="margin-top: 1em">--qseq query input files are in
Illumina&rsquo;s qseq format</p>

<p style="margin-top: 1em">-f query input files are
(multi-)FASTA .fa/.mfa</p>

<p style="margin-top: 1em">-r query input files are raw
one-sequence-per-line</p>

<p style="margin-top: 1em">-c &lt;m1&gt;, &lt;m2&gt;,
&lt;r&gt; are sequences themselves, not files</p>

<p style="margin-top: 1em">-s/--skip &lt;int&gt; <br>
skip the first &lt;int&gt; reads/pairs in the input
(none)</p>

<p style="margin-top: 1em">-u/--upto &lt;int&gt; <br>
stop after first &lt;int&gt; reads/pairs (no limit)</p>

<p style="margin-top: 1em">-5/--trim5 &lt;int&gt; <br>
trim &lt;int&gt; bases from 5&rsquo;/left end of reads
(0)</p>

<p style="margin-top: 1em">-3/--trim3 &lt;int&gt; <br>
trim &lt;int&gt; bases from 3&rsquo;/right end of reads
(0)</p>

<p style="margin-top: 1em">--phred33 <br>
qualities are Phred+33 (default)</p>

<p style="margin-top: 1em">--phred64 <br>
qualities are Phred+64</p>

<p style="margin-top: 1em">--int-quals <br>
qualities encoded as space-delimited integers</p>

<p style="margin-top: 1em">Alignment:</p>

<p style="margin-top: 1em">--n-ceil &lt;func&gt; <br>
func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)</p>

<p style="margin-top: 1em">--ignore-quals <br>
treat all quality values as 30 on Phred scale (off)</p>

<p style="margin-top: 1em">--nofw do not align forward
(original) version of read (off)</p>

<p style="margin-top: 1em">--norc do not align
reverse-complement version of read (off)</p>

<p style="margin-top: 1em">Spliced Alignment:</p>

<p style="margin-top: 1em">--pen-cansplice &lt;int&gt; <br>
penalty for a canonical splice site (0)</p>

<p style="margin-top: 1em">--pen-noncansplice &lt;int&gt;
<br>
penalty for a non-canonical splice site (12)</p>

<p style="margin-top: 1em">--pen-canintronlen &lt;func&gt;
<br>
penalty for long introns (G,-8,1) with canonical splice
sites</p>

<p style="margin-top: 1em">--pen-noncanintronlen
&lt;func&gt; <br>
penalty for long introns (G,-8,1) with noncanonical splice
sites</p>

<p style="margin-top: 1em">--min-intronlen &lt;int&gt; <br>
minimum intron length (20)</p>

<p style="margin-top: 1em">--max-intronlen &lt;int&gt; <br>
maximum intron length (500000)</p>

<p style="margin-top: 1em">--known-splicesite-infile
&lt;path&gt; <br>
provide a list of known splice sites</p>

<p style="margin-top: 1em">--novel-splicesite-outfile
&lt;path&gt; <br>
report a list of splice sites</p>

<p style="margin-top: 1em">--novel-splicesite-infile
&lt;path&gt; <br>
provide a list of novel splice sites</p>

<p style="margin-top: 1em">--no-temp-splicesite <br>
disable the use of splice sites found</p>

<p style="margin-top: 1em">--no-spliced-alignment <br>
disable spliced alignment</p>

<p style="margin-top: 1em">--rna-strandness &lt;string&gt;
<br>
Specify strand-specific information (unstranded)</p>

<p style="margin-top: 1em">--tmo Reports only those
alignments within known transcriptome</p>

<p style="margin-top: 1em">--dta Reports alignments
tailored for transcript assemblers</p>

<p style="margin-top: 1em">--dta-cufflinks <br>
Reports alignments tailored specifically for cufflinks</p>

<p style="margin-top: 1em">Scoring:</p>

<p style="margin-top: 1em">--ma &lt;int&gt; <br>
match bonus (0 for --end-to-end, 2 for --local)</p>

<p style="margin-top: 1em">--mp &lt;int&gt;,&lt;int&gt;
<br>
max and min penalties for mismatch; lower qual = lower
penalty &lt;2,6&gt;</p>

<p style="margin-top: 1em">--sp &lt;int&gt;,&lt;int&gt;
<br>
max and min penalties for soft-clipping; lower qual = lower
penalty &lt;1,2&gt;</p>

<p style="margin-top: 1em">--no-softclip <br>
no soft-clipping</p>

<p style="margin-top: 1em">--np &lt;int&gt; <br>
penalty for non-A/C/G/Ts in read/ref (1)</p>

<p style="margin-top: 1em">--rdg &lt;int&gt;,&lt;int&gt;
<br>
read gap open, extend penalties (5,3)</p>

<p style="margin-top: 1em">--rfg &lt;int&gt;,&lt;int&gt;
<br>
reference gap open, extend penalties (5,3)</p>

<p style="margin-top: 1em">--score-min &lt;func&gt; min
acceptable alignment score w/r/t read length <br>
(L,0.0,-0.2)</p>

<p style="margin-top: 1em">Reporting:</p>

<p style="margin-top: 1em">-k &lt;int&gt; (default: 5)
report up to &lt;int&gt; alns per read</p>

<p style="margin-top: 1em">Paired-end:</p>

<p style="margin-top: 1em">-I/--minins &lt;int&gt; <br>
minimum fragment length (0), only valid with
--no-spliced-alignment</p>

<p style="margin-top: 1em">-X/--maxins &lt;int&gt; <br>
maximum fragment length (500), only valid with
--no-spliced-alignment</p>

<p style="margin-top: 1em">--fr/--rf/--ff -1, -2 mates
align fw/rev, rev/fw, fw/fw (--fr)</p>

<p style="margin-top: 1em">--no-mixed <br>
suppress unpaired alignments for paired reads</p>

<p style="margin-top: 1em">--no-discordant <br>
suppress discordant alignments for paired reads</p>

<p style="margin-top: 1em">Output:</p>

<p style="margin-top: 1em">-t/--time <br>
print wall-clock time taken by search phases</p>

<p style="margin-top: 1em">--quiet <br>
print nothing to stderr except serious errors</p>

<p style="margin-top: 1em">--met-file &lt;path&gt; <br>
send metrics to file at &lt;path&gt; (off)</p>

<p style="margin-top: 1em">--met-stderr <br>
send metrics to stderr (off)</p>

<p style="margin-top: 1em">--met &lt;int&gt; <br>
report internal counters &amp; metrics every &lt;int&gt;
secs (1)</p>

<p style="margin-top: 1em">--no-head <br>
supppress header lines, i.e. lines starting with @</p>

<p style="margin-top: 1em">--no-sq <br>
supppress @SQ header lines</p>

<p style="margin-top: 1em">--rg-id &lt;text&gt; <br>
set read group id, reflected in @RG line and RG:Z: opt
field</p>

<p style="margin-top: 1em">--rg &lt;text&gt; <br>
add &lt;text&gt; (&quot;lab:value&quot;) to @RG line of SAM
header. Note: @RG line only printed when --rg-id is set.</p>

<p style="margin-top: 1em">--omit-sec-seq <br>
put &rsquo;*&rsquo; in SEQ and QUAL fields for secondary
alignments.</p>

<p style="margin-top: 1em">Performance:</p>

<p style="margin-top: 1em">-o/--offrate &lt;int&gt;
override offrate of index; must be &gt;= index&rsquo;s
offrate</p>

<p style="margin-top: 1em">-p/--threads &lt;int&gt; number
of alignment threads to launch (1)</p>

<p style="margin-top: 1em">--reorder <br>
force SAM output order to match order of input reads</p>

<p style="margin-top: 1em">--mm use memory-mapped I/O for
index; many &rsquo;hisat2&rsquo;s can share</p>

<p style="margin-top: 1em">Other:</p>

<p style="margin-top: 1em">--qc-filter <br>
filter out reads that are bad according to QSEQ filter</p>

<p style="margin-top: 1em">--seed &lt;int&gt; <br>
seed for random number generator (0)</p>

<p style="margin-top: 1em">--non-deterministic seed rand.
gen. arbitrarily instead of using read attributes</p>

<p style="margin-top: 1em">--remove-chrname <br>
remove &rsquo;chr&rsquo; from reference names in
alignment</p>

<p style="margin-top: 1em">--add-chrname <br>
add &rsquo;chr&rsquo; to reference names in alignment</p>

<p style="margin-top: 1em">--version <br>
print version information and quit</p>

<p style="margin-top: 1em">-h/--help <br>
print this usage message</p>

<p style="margin-top: 1em">64-bit Built on Debian unstable
Fri Dec 9 12:45:26 UTC 2016 Compiler: gcc version 6.2.1
20161202 (Ubuntu 6.2.1-5ubuntu2) Options: -O3 -funroll-loops
-g3 -Wdate-time -D_FOR&acirc; <br>
TIFY_SOURCE=2 -DPOPCNT_CAPABILITY Sizeof {int, long, long
long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}</p>

<p style="margin-top: 1em">hisat2-align-l version 2.0.5
December 2016 HISAT2-ALIGN-L(1)</p>
<hr>
</body>
</html>
