<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>HISAT2-ALIGN-S(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">HISAT2-ALIGN-S(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">HISAT2-ALIGN-S(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
hisat2-align-s - HISAT2 graph-based alignment of short nucleotide reads to many
  genomes, small index binary
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
HISAT2 version 2.0.5 by Daehwan Kim (infphilo@gmail.com,
  www.ccb.jhu.edu/people/infphilo) Usage:
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">hisat2-align [options]* <b>-x</b> &lt;ht2-idx&gt; {-1
      &lt;m1&gt; <b>-2</b> &lt;m2&gt; | <b>-U</b> &lt;r&gt;} [-S
    &lt;sam&gt;]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;ht2-idx&gt;</dt>
  <dd class="It-tag">Index filename prefix (minus trailing .X.ht2).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;m1&gt;</dt>
  <dd class="It-tag">Files with #1 mates, paired with files in &lt;m2&gt;.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;m2&gt;</dt>
  <dd class="It-tag">Files with #2 mates, paired with files in &lt;m1&gt;.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;r&gt;</dt>
  <dd class="It-tag">Files with unpaired reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;sam&gt;</dt>
  <dd class="It-tag">File for SAM output (default: stdout)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; can be comma-separated
      lists (no whitespace) and can be specified many times. E.g. '-U
      file1.fq,file2.fq <b>-U</b> file3.fq'.</dd>
</dl>
<div class="Pp"></div>
Options (defaults in parentheses):
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Input:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b></dt>
  <dd class="It-tag">query input files are FASTQ .fq/.fastq (default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--qseq</b></dt>
  <dd class="It-tag">query input files are in Illumina's qseq format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b></dt>
  <dd class="It-tag">query input files are (multi-)FASTA .fa/.mfa</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b></dt>
  <dd class="It-tag">query input files are raw one-sequence-per-line</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">&lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; are sequences themselves,
      not files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>/--skip &lt;int&gt;</dt>
  <dd class="It-tag">skip the first &lt;int&gt; reads/pairs in the input
    (none)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b>/--upto &lt;int&gt;</dt>
  <dd class="It-tag">stop after first &lt;int&gt; reads/pairs (no limit)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-5</b>/--trim5 &lt;int&gt;</dt>
  <dd class="It-tag">trim &lt;int&gt; bases from 5'/left end of reads (0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-3</b>/--trim3 &lt;int&gt;</dt>
  <dd class="It-tag">trim &lt;int&gt; bases from 3'/right end of reads (0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--phred33</b></dt>
  <dd class="It-tag">qualities are Phred+33 (default)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--phred64</b></dt>
  <dd class="It-tag">qualities are Phred+64</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--int-quals</b></dt>
  <dd class="It-tag">qualities encoded as space-delimited integers</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Alignment:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--n-ceil</b> &lt;func&gt;</dt>
  <dd class="It-tag">func for max # non-A/C/G/Ts permitted in aln
    (L,0,0.15)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ignore-quals</b></dt>
  <dd class="It-tag">treat all quality values as 30 on Phred scale (off)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nofw</b></dt>
  <dd class="It-tag">do not align forward (original) version of read (off)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--norc</b></dt>
  <dd class="It-tag">do not align reverse-complement version of read (off)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Spliced Alignment:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pen-cansplice</b> &lt;int&gt;</dt>
  <dd class="It-tag">penalty for a canonical splice site (0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pen-noncansplice</b> &lt;int&gt;</dt>
  <dd class="It-tag">penalty for a non-canonical splice site (12)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pen-canintronlen</b> &lt;func&gt;</dt>
  <dd class="It-tag">penalty for long introns (G,-8,1) with canonical splice
      sites</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pen-noncanintronlen</b> &lt;func&gt;</dt>
  <dd class="It-tag">penalty for long introns (G,-8,1) with noncanonical splice
      sites</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-intronlen</b> &lt;int&gt;</dt>
  <dd class="It-tag">minimum intron length (20)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-intronlen</b> &lt;int&gt;</dt>
  <dd class="It-tag">maximum intron length (500000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--known-splicesite-infile</b> &lt;path&gt;</dt>
  <dd class="It-tag">provide a list of known splice sites</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--novel-splicesite-outfile</b> &lt;path&gt;</dt>
  <dd class="It-tag">report a list of splice sites</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--novel-splicesite-infile</b> &lt;path&gt;</dt>
  <dd class="It-tag">provide a list of novel splice sites</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-temp-splicesite</b></dt>
  <dd class="It-tag">disable the use of splice sites found</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-spliced-alignment</b></dt>
  <dd class="It-tag">disable spliced alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rna-strandness</b> &lt;string&gt;</dt>
  <dd class="It-tag">Specify strand-specific information (unstranded)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tmo</b></dt>
  <dd class="It-tag">Reports only those alignments within known
    transcriptome</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dta</b></dt>
  <dd class="It-tag">Reports alignments tailored for transcript assemblers</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dta-cufflinks</b></dt>
  <dd class="It-tag">Reports alignments tailored specifically for cufflinks</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Scoring:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ma</b> &lt;int&gt;</dt>
  <dd class="It-tag">match bonus (0 for <b>--end-to-end</b>, 2 for
      <b>--local</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mp</b> &lt;int&gt;,&lt;int&gt;</dt>
  <dd class="It-tag">max and min penalties for mismatch; lower qual = lower
      penalty &lt;2,6&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sp</b> &lt;int&gt;,&lt;int&gt;</dt>
  <dd class="It-tag">max and min penalties for soft-clipping; lower qual = lower
      penalty &lt;1,2&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-softclip</b></dt>
  <dd class="It-tag">no soft-clipping</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--np</b> &lt;int&gt;</dt>
  <dd class="It-tag">penalty for non-A/C/G/Ts in read/ref (1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rdg</b> &lt;int&gt;,&lt;int&gt;</dt>
  <dd class="It-tag">read gap open, extend penalties (5,3)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rfg</b> &lt;int&gt;,&lt;int&gt;</dt>
  <dd class="It-tag">reference gap open, extend penalties (5,3)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--score-min</b> &lt;func&gt; min acceptable alignment
    score w/r/t read length</dt>
  <dd class="It-tag">(L,0.0,-0.2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Reporting:</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-k</b>
  &lt;int&gt; (default: 5) report up to &lt;int&gt; alns per read</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Paired-end:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-I</b>/--minins &lt;int&gt;</dt>
  <dd class="It-tag">minimum fragment length (0), only valid with
      <b>--no-spliced-alignment</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-X</b>/--maxins &lt;int&gt;</dt>
  <dd class="It-tag">maximum fragment length (500), only valid with
      <b>--no-spliced-alignment</b></dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--fr</b>/--rf/--ff
  <b>-1</b>, <b>-2</b> mates align fw/rev, rev/fw, fw/fw (<b>--fr</b>)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-mixed</b></dt>
  <dd class="It-tag">suppress unpaired alignments for paired reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-discordant</b></dt>
  <dd class="It-tag">suppress discordant alignments for paired reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Output:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>/--time</dt>
  <dd class="It-tag">print wall-clock time taken by search phases</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quiet</b></dt>
  <dd class="It-tag">print nothing to stderr except serious errors</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--met-file</b> &lt;path&gt;</dt>
  <dd class="It-tag">send metrics to file at &lt;path&gt; (off)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--met-stderr</b></dt>
  <dd class="It-tag">send metrics to stderr (off)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--met</b> &lt;int&gt;</dt>
  <dd class="It-tag">report internal counters &amp; metrics every &lt;int&gt;
      secs (1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-head</b></dt>
  <dd class="It-tag">supppress header lines, i.e. lines starting with @</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-sq</b></dt>
  <dd class="It-tag">supppress @SQ header lines</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rg-id</b> &lt;text&gt;</dt>
  <dd class="It-tag">set read group id, reflected in @RG line and RG:Z: opt
      field</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rg</b> &lt;text&gt;</dt>
  <dd class="It-tag">add &lt;text&gt; (&quot;lab:value&quot;) to @RG line of SAM
      header. Note: @RG line only printed when <b>--rg-id</b> is set.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--omit-sec-seq</b></dt>
  <dd class="It-tag">put '*' in SEQ and QUAL fields for secondary
    alignments.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Performance:</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>/--offrate
  &lt;int&gt; override offrate of index; must be &gt;= index's offrate</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-p</b>/--threads
  &lt;int&gt; number of alignment threads to launch (1)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--reorder</b></dt>
  <dd class="It-tag">force SAM output order to match order of input reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mm</b></dt>
  <dd class="It-tag">use memory-mapped I/O for index; many 'hisat2's can
    share</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Other:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--qc-filter</b></dt>
  <dd class="It-tag">filter out reads that are bad according to QSEQ filter</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b> &lt;int&gt;</dt>
  <dd class="It-tag">seed for random number generator (0)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--non-deterministic</b>
  seed rand. gen. arbitrarily instead of using read attributes</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--remove-chrname</b></dt>
  <dd class="It-tag">remove 'chr' from reference names in alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--add-chrname</b></dt>
  <dd class="It-tag">add 'chr' to reference names in alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">print version information and quit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>/--help</dt>
  <dd class="It-tag">print this usage message</dd>
</dl>
<div class="Pp"></div>
64-bit Built on Debian unstable Fri Dec 9 12:45:26 UTC 2016 Compiler: gcc
  version 6.2.1 20161202 (Ubuntu 6.2.1-5ubuntu2) Options: <b>-O3</b>
  <b>-funroll-loops</b> <b>-g3</b> <b>-Wdate-time</b>
  <b>-D_FORTIFY_SOURCE</b>=<i>2</i> <b>-DPOPCNT_CAPABILITY</b> Sizeof {int,
  long, long long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}</div>
<table class="foot">
  <tr>
    <td class="foot-date">December 2016</td>
    <td class="foot-os">hisat2-align-s version 2.0.5</td>
  </tr>
</table>
</body>
</html>
