<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:53 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmm2pfam(1) HMMER Manual hmm2pfam(1)</p>

<p style="margin-top: 1em">NAME <br>
hmm2pfam - search one or more sequences against an HMM
database</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmm2pfam [options] hmmfile seqfile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
hmm2pfam reads a sequence file seqfile and compares each
sequence in it, one at a time, against all the HMMs in
hmmfile looking for significantly similar sequence
matches.</p>

<p style="margin-top: 1em">hmmfile will be looked for first
in the current working directory, then in a directory named
by the environment variable HMMERDB. This lets
administrators install HMM library(s) <br>
such as Pfam in a common location.</p>

<p style="margin-top: 1em">There is a separate output
report for each sequence in seqfile. This report consists of
three sections: a ranked list of the best scoring HMMs, a
list of the best scoring <br>
domains in order of their occurrence in the sequence, and
alignments for all the best scoring domains. A sequence
score may be higher than a domain score for the same
sequence <br>
if there is more than one domain in the sequence; the
sequence score takes into account all the domains. All
sequences scoring above the -E and -T cutoffs are shown in
the first <br>
list, then every domain found in this list is shown in the
second list of domain hits. If desired, E-value and bit
score thresholds may also be applied to the domain list
using <br>
the --domE and --domT options.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Print brief help; includes version number and summary of
all options, including expert options.</p>

<p style="margin-top: 1em">-n Specify that models and
sequence are nucleic acid, not protein. Other HMMER programs
autodetect this; but because of the order in which hmm2pfam
accesses data, it can&rsquo;t <br>
reliably determine the correct &quot;alphabet&quot; by
itself.</p>

<p style="margin-top: 1em">-A &lt;n&gt; Limits the
alignment output to the &lt;n&gt; best scoring domains. -A0
shuts off the alignment output and can be used to reduce the
size of output files.</p>

<p style="margin-top: 1em">-E &lt;x&gt; Set the E-value
cutoff for the per-sequence ranked hit list to &lt;x&gt;,
where &lt;x&gt; is a positive real number. The default is
10.0. Hits with E-values better than (less than) <br>
this threshold will be shown.</p>

<p style="margin-top: 1em">-T &lt;x&gt; Set the bit score
cutoff for the per-sequence ranked hit list to &lt;x&gt;,
where &lt;x&gt; is a real number. The default is negative
infinity; by default, the threshold is con&acirc; <br>
trolled by E-value and not by bit score. Hits with bit
scores better than (greater than) this threshold will be
shown.</p>

<p style="margin-top: 1em">-Z &lt;n&gt; Calculate the
E-value scores as if we had seen a sequence database of
&lt;n&gt; sequences. The default is arbitrarily set to
59021, the size of Swissprot 34.</p>

<p style="margin-top: 1em">EXPERT OPTIONS <br>
--acc Report HMM accessions instead of names in the output
reports. Useful for high-throughput annotation, where the
data are being parsed for storage in a relational
database.</p>

<p style="margin-top: 1em">--compat <br>
Use the output format of HMMER 2.1.1, the 1998-2001 public
release; provided so 2.1.1 parsers don&rsquo;t have to be
rewritten.</p>

<p style="margin-top: 1em">--cpu &lt;n&gt; <br>
Sets the maximum number of CPUs that the program will run
on. The default is to use all CPUs in the machine. Overrides
the HMMER_NCPU environment variable. Only affects <br>
threaded versions of HMMER (the default on most
systems).</p>

<p style="margin-top: 1em">--cut_ga <br>
Use Pfam GA (gathering threshold) score cutoffs. Equivalent
to --globT &lt;GA1&gt; --domT &lt;GA2&gt;, but the GA1 and
GA2 cutoffs are read from each HMM in hmmfile individually.
<br>
hmm2build puts these cutoffs there if the alignment file was
annotated in a Pfam-friendly alignment format (extended
SELEX or Stockholm format) and the optional GA annota&acirc;
<br>
tion line was present. If these cutoffs are not set in the
HMM file, --cut_ga doesn&rsquo;t work.</p>

<p style="margin-top: 1em">--cut_tc <br>
Use Pfam TC (trusted cutoff) score cutoffs. Equivalent to
--globT &lt;TC1&gt; --domT &lt;TC2&gt;, but the TC1 and TC2
cutoffs are read from each HMM in hmmfile individually.
hmm2build <br>
puts these cutoffs there if the alignment file was annotated
in a Pfam-friendly alignment format (extended SELEX or
Stockholm format) and the optional TC annotation line <br>
was present. If these cutoffs are not set in the HMM file,
--cut_tc doesn&rsquo;t work.</p>

<p style="margin-top: 1em">--cut_nc <br>
Use Pfam NC (noise cutoff) score cutoffs. Equivalent to
--globT &lt;NC1&gt; --domT &lt;NC2&gt;, but the NC1 and NC2
cutoffs are read from each HMM in hmmfile individually.
hmm2build <br>
puts these cutoffs there if the alignment file was annotated
in a Pfam-friendly alignment format (extended SELEX or
Stockholm format) and the optional NC annotation line <br>
was present. If these cutoffs are not set in the HMM file,
--cut_nc doesn&rsquo;t work.</p>

<p style="margin-top: 1em">--domE &lt;x&gt; <br>
Set the E-value cutoff for the per-domain ranked hit list to
&lt;x&gt;, where &lt;x&gt; is a positive real number. The
default is infinity; by default, all domains in the
sequences <br>
that passed the first threshold will be reported in the
second list, so that the number of domains reported in the
per-sequence list is consistent with the number that <br>
appear in the per-domain list.</p>

<p style="margin-top: 1em">--domT &lt;x&gt; <br>
Set the bit score cutoff for the per-domain ranked hit list
to &lt;x&gt;, where &lt;x&gt; is a real number. The default
is negative infinity; by default, all domains in the
sequences <br>
that passed the first threshold will be reported in the
second list, so that the number of domains reported in the
per-sequence list is consistent with the number that <br>
appear in the per-domain list. Important note: only one
domain in a sequence is absolutely controlled by this
parameter, or by --domT. The second and subsequent domains
<br>
in a sequence have a de facto bit score threshold of 0
because of the details of how HMMER works. HMMER requires at
least one pass through the main model per sequence; to <br>
do more than one pass (more than one domain) the multidomain
alignment must have a better score than the single domain
alignment, and hence the extra domains must contrib&acirc;
<br>
ute positive score. See the Users&rsquo; Guide for more
detail.</p>

<p style="margin-top: 1em">--forward <br>
Use the Forward algorithm instead of the Viterbi algorithm
to determine the per-sequence scores. Per-domain scores are
still determined by the Viterbi algorithm. Some have <br>
argued that Forward is a more sensitive algorithm for
detecting remote sequence homologues; my experiments with
HMMER have not confirmed this, however.</p>

<p style="margin-top: 1em">--informat &lt;s&gt; <br>
Assert that the input seqfile is in format &lt;s&gt;; do not
run Babelfish format autodection. This increases the
reliability of the program somewhat, because the Babelfish
can <br>
make mistakes; particularly recommended for unattended,
high-throughput runs of HMMER. Valid format strings include
FASTA, GENBANK, EMBL, GCG, PIR, STOCKHOLM, SELEX, MSF, <br>
CLUSTAL, and PHYLIP. See the User&rsquo;s Guide for a
complete list.</p>

<p style="margin-top: 1em">--null2 <br>
Turn off the post hoc second null model. By default, each
alignment is rescored by a postprocessing step that takes
into account possible biased composition in either the <br>
HMM or the target sequence. This is almost essential in
database searches, especially with local alignment models.
There is a very small chance that this postprocessing <br>
might remove real matches, and in these cases --null2 may
improve sensitivity at the expense of reducing specificity
by letting biased composition hits through.</p>

<p style="margin-top: 1em">--pvm Run on a Parallel Virtual
Machine (PVM). The PVM must already be running. The client
program hmm2pfam-pvm must be installed on all the PVM nodes.
The HMM database hmmfile <br>
and an associated GSI index file hmmfile.gsi must also be
installed on all the PVM nodes. (The GSI index is produced
by the program hmm2index.) Because the PVM implemen&acirc;
<br>
tation is I/O bound, it is highly recommended that each node
have a local copy of hmmfile rather than NFS mounting a
shared copy. Optional PVM support must have been com&acirc;
<br>
piled into HMMER for --pvm to function.</p>

<p style="margin-top: 1em">--xnu Turn on XNU filtering of
target protein sequences. Has no effect on nucleic acid
sequences. In trial experiments, --xnu appears to perform
less well than the default post <br>
hoc null2 model.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Master man page, with full list of and guide to the
individual man pages: see hmmer2(1).</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide
(ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf);
or see the HMMER web page, http://hmmer.janelia.org/.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 1992-2003 HHMI/Washington University School of
Medicine. <br>
Freely distributed under the GNU General Public License
(GPL). <br>
See the file COPYING in your distribution for details on
redistribution conditions.</p>

<p style="margin-top: 1em">AUTHOR <br>
Sean Eddy <br>
HHMI/Dept. of Genetics <br>
Washington Univ. School of Medicine <br>
4566 Scott Ave. <br>
St Louis, MO 63110 USA <br>
http://www.genetics.wustl.edu/eddy/</p>

<p style="margin-top: 1em">HMMER 2.3.2 Oct 2003
hmm2pfam(1)</p>
<hr>
</body>
</html>
