<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>hmm2build(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">hmm2build(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">hmm2build(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">hmm2build - build a profile HMM from an alignment</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmm2build</b> <i>[options]</i> <i>hmmfile</i> <i>alignfile</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>hmm2build</b> reads a multiple sequence alignment file <i>alignfile</i> ,
  builds a new profile HMM, and saves the HMM in <i>hmmfile.</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<i>alignfile</i> may be in ClustalW, GCG MSF, SELEX, Stockholm, or aligned FASTA
  alignment format. The format is automatically detected.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
By default, the model is configured to find one or more nonoverlapping
  alignments to the complete model: multiple global alignments with respect to
  the model, and local with respect to the sequence. This is analogous to the
  behavior of the <b>hmmls</b> program of HMMER 1. To configure the model for
  multiple <i>local</i> alignments with respect to the model and local with
  respect to the sequence, a la the old program <b>hmmfs,</b> use the <b>-f</b>
  (fragment) option. More rarely, you may want to configure the model for a
  single global alignment (global with respect to both model and sequence),
  using the <b>-g</b> option; or to configure the model for a single local/local
  alignment (a la standard Smith/Waterman, or the old <b>hmmsw</b> program), use
  the <b>-s</b> option.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b></dt>
  <dd class="It-tag">Configure the model for finding multiple domains per
      sequence, where each domain can be a local (fragmentary) alignment. This
      is analogous to the old <b>hmmfs</b> program of HMMER 1.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">Configure the model for finding a single global alignment
      to a target sequence, analogous to the old <b>hmms</b> program of HMMER 1.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Print brief help; includes version number and summary of
      all options, including expert options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Name this HMM <i>&lt;s&gt;.</i> <i>&lt;s&gt;</i> can be any
      string of non-whitespace characters (e.g. one &quot;word&quot;). There is
      no length limit (at least not one imposed by HMMER; your shell will
      complain about command line lengths first).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Re-save the starting alignment to <i>&lt;f&gt;,</i> in
      Stockholm format. The columns which were assigned to match states will be
      marked with x's in an #=RF annotation line. If either the <b>--hand</b> or
      <b>--fast</b> construction options were chosen, the alignment may have
      been slightly altered to be compatible with Plan 7 transitions, so saving
      the final alignment and comparing to the starting alignment can let you
      view these alterations. See the User's Guide for more information on this
      arcane side effect.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">Configure the model for finding a single local alignment
      per target sequence. This is analogous to the standard Smith/Waterman
      algorithm or the <b>hmmsw</b> program of HMMER 1.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b></dt>
  <dd class="It-tag">Append this model to an existing <i>hmmfile</i> rather than
      creating <i>hmmfile.</i> Useful for building HMM libraries (like Pfam).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b></dt>
  <dd class="It-tag">Force overwriting of an existing <i>hmmfile.</i> Otherwise
      HMMER will refuse to clobber your existing HMM files, for safety's sake.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXPERT_OPTIONS"><a class="selflink" href="#EXPERT_OPTIONS">EXPERT
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--amino</b></dt>
  <dd class="It-tag">Force the sequence alignment to be interpreted as amino
      acid sequences. Normally HMMER autodetects whether the alignment is
      protein or DNA, but sometimes alignments are so small that autodetection
      is ambiguous. See <b>--nucleic.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--archpri</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the &quot;architecture prior&quot; used by MAP
      architecture construction to <i>&lt;x&gt;,</i> where <i>&lt;x&gt;</i> is a
      probability between 0 and 1. This parameter governs a geometric prior
      distribution over model lengths. As <i>&lt;x&gt;</i> increases, longer
      models are favored a priori. As <i>&lt;x&gt;</i> decreases, it takes more
      residue conservation in a column to make a column a &quot;consensus&quot;
      match column in the model architecture. The 0.85 default has been chosen
      empirically as a reasonable setting.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--binary</b></dt>
  <dd class="It-tag">Write the HMM to <i>hmmfile</i> in HMMER binary format
      instead of readable ASCII text.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cfile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save the observed emission and transition counts to
      <i>&lt;f&gt;</i> after the architecture has been determined (e.g. after
      residues/gaps have been assigned to match, delete, and insert states).
      This option is used in HMMER development for generating data files useful
      for training new Dirichlet priors. The format of count files is documented
      in the User's Guide.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fast</b></dt>
  <dd class="It-tag">Quickly and heuristically determine the architecture of the
      model by assigning all columns will more than a certain fraction of gap
      characters to insert states. By default this fraction is 0.5, and it can
      be changed using the <b>--gapmax</b> option. The default construction
      algorithm is a maximum a posteriori (MAP) algorithm, which is slower.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gapmax</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Controls the <i>--fast</i> model construction algorithm,
      but if <i>--fast</i> is not being used, has no effect. If a column has
      more than a fraction <i>&lt;x&gt;</i> of gap symbols in it, it gets
      assigned to an insert column. <i>&lt;x&gt;</i> is a frequency from 0 to 1,
      and by default is set to 0.5. Higher values of <i>&lt;x&gt;</i> mean more
      columns get assigned to consensus, and models get longer; smaller values
      of <i>&lt;x&gt;</i> mean fewer columns get assigned to consensus, and
      models get smaller. <i>&lt;x&gt;</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hand</b></dt>
  <dd class="It-tag">Specify the architecture of the model by hand: the
      alignment file must be in SELEX or Stockholm format, and the reference
      annotation line (#=RF in SELEX, #=GC RF in Stockholm) is used to specify
      the architecture. Any column marked with a non-gap symbol (such as an 'x',
      for instance) is assigned as a consensus (match) column in the model.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--idlevel</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Controls both the determination of effective sequence
      number and the behavior of the <i>--wblosum</i> weighting option. The
      sequence alignment is clustered by percent identity, and the number of
      clusters at a cutoff threshold of <i>&lt;x&gt;</i> is used to determine
      the effective sequence number. Higher values of <i>&lt;x&gt;</i> give more
      clusters and higher effective sequence numbers; lower values of
      <i>&lt;x&gt;</i> give fewer clusters and lower effective sequence numbers.
      <i>&lt;x&gt;</i> is a fraction from 0 to 1, and by default is set to 0.62
      (corresponding to the clustering level used in constructing the BLOSUM62
      substitution matrix).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Assert that the input <i>seqfile</i> is in format
      <i>&lt;s&gt;;</i> do not run Babelfish format autodection. This increases
      the reliability of the program somewhat, because the Babelfish can make
      mistakes; particularly recommended for unattended, high-throughput runs of
      HMMER. Valid format strings include FASTA, GENBANK, EMBL, GCG, PIR,
      STOCKHOLM, SELEX, MSF, CLUSTAL, and PHYLIP. See the User's Guide for a
      complete list.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noeff</b></dt>
  <dd class="It-tag">Turn off the effective sequence number calculation, and use
      the true number of sequences instead. This will usually reduce the
      sensitivity of the final model (so don't do it without good reason!)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nucleic</b></dt>
  <dd class="It-tag">Force the alignment to be interpreted as nucleic acid
      sequence, either RNA or DNA. Normally HMMER autodetects whether the
      alignment is protein or DNA, but sometimes alignments are so small that
      autodetection is ambiguous. See <b>--amino.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--null</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Read a null model from <i>&lt;f&gt;.</i> The default for
      protein is to use average amino acid frequencies from Swissprot 34 and p1
      = 350/351; for nucleic acid, the default is to use 0.25 for each base and
      p1 = 1000/1001. For documentation of the format of the null model file and
      further explanation of how the null model is used, see the User's Guide.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pam</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Apply a heuristic PAM- (substitution matrix-) based prior
      on match emission probabilities instead of the default mixture Dirichlet.
      The substitution matrix is read from <i>&lt;f&gt;.</i> See
      <b>--pamwgt.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The default Dirichlet state transition prior and insert emission prior are
      unaffected. Therefore in principle you could combine <b>--prior</b> with
      <b>--pam</b> but this isn't recommended, as it hasn't been tested. (
      <b>--pam</b> itself hasn't been tested much!)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pamwgt</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Controls the weight on a PAM-based prior. Only has effect
      if <b>--pam</b> option is also in use. <i>&lt;x&gt;</i> is a positive real
      number, 20.0 by default. <i>&lt;x&gt;</i> is the number of
      &quot;pseudocounts&quot; contriubuted by the heuristic prior. Very high
      values of <i>&lt;x&gt;</i> can force a scoring system that is entirely
      driven by the substitution matrix, making HMMER somewhat approximate
      Gribskov profiles.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pbswitch</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">For alignments with a very large number of sequences, the
      GSC, BLOSUM, and Voronoi weighting schemes are slow; they're O(N^2) for N
      sequences. Henikoff position-based weights (PB weights) are more
      efficient. At or above a certain threshold sequence number
      <i>&lt;n&gt;</i> <b>hmm2build</b> will switch from GSC, BLOSUM, or Voronoi
      weights to PB weights. To disable this switching behavior (at the cost of
      compute time, set <i>&lt;n&gt;</i> to be something larger than the number
      of sequences in your alignment. <i>&lt;n&gt;</i> is a positive integer;
      the default is 1000.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--prior</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Read a Dirichlet prior from <i>&lt;f&gt;,</i> replacing the
      default mixture Dirichlet. The format of prior files is documented in the
      User's Guide, and an example is given in the Demos directory of the HMMER
      distribution.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--swentry</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Controls the total probability that is distributed to local
      entries into the model, versus starting at the beginning of the model as
      in a global alignment. <i>&lt;x&gt;</i> is a probability from 0 to 1, and
      by default is set to 0.5. Higher values of <i>&lt;x&gt;</i> mean that hits
      that are fragments on their left (N or 5'-terminal) side will be penalized
      less, but complete global alignments will be penalized more. Lower values
      of <i>&lt;x&gt;</i> mean that fragments on the left will be penalized
      more, and global alignments on this side will be favored. This option only
      affects the configurations that allow local alignments, e.g. <b>-s</b> and
      <b>-f;</b> unless one of these options is also activated, this option has
      no effect. You have independent control over local/global alignment
      behavior for the N/C (5'/3') termini of your target sequences using
      <b>--swentry</b> and <b>--swexit.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--swexit</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Controls the total probability that is distributed to local
      exits from the model, versus ending an alignment at the end of the model
      as in a global alignment. <i>&lt;x&gt;</i> is a probability from 0 to 1,
      and by default is set to 0.5. Higher values of <i>&lt;x&gt;</i> mean that
      hits that are fragments on their right (C or 3'-terminal) side will be
      penalized less, but complete global alignments will be penalized more.
      Lower values of <i>&lt;x&gt;</i> mean that fragments on the right will be
      penalized more, and global alignments on this side will be favored. This
      option only affects the configurations that allow local alignments, e.g.
      <b>-s</b> and <b>-f;</b> unless one of these options is also activated,
      this option has no effect. You have independent control over local/global
      alignment behavior for the N/C (5'/3') termini of your target sequences
      using <b>--swentry</b> and <b>--swexit.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--verbose</b></dt>
  <dd class="It-tag">Print more possibly useful stuff, such as the individual
      scores for each sequence in the alignment.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wblosum</b></dt>
  <dd class="It-tag">Use the BLOSUM filtering algorithm to weight the sequences,
      instead of the default. Cluster the sequences at a given percentage
      identity (see <b>--idlevel);</b> assign each cluster a total weight of
      1.0, distributed equally amongst the members of that cluster.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wgsc</b></dt>
  <dd class="It-tag">Use the Gerstein/Sonnhammer/Chothia ad hoc sequence
      weighting algorithm. This is already the default, so this option has no
      effect (unless it follows another option in the -\-w family, in which case
      it overrides it).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wme</b></dt>
  <dd class="It-tag">Use the Krogh/Mitchison maximum entropy algorithm to
      &quot;weight&quot; the sequences. This supersedes the
      Eddy/Mitchison/Durbin maximum discrimination algorithm, which gives almost
      identical weights but is less robust. ME weighting seems to give a
      marginal increase in sensitivity over the default GSC weights, but takes a
      fair amount of time.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wnone</b></dt>
  <dd class="It-tag">Turn off all sequence weighting.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wpb</b></dt>
  <dd class="It-tag">Use the Henikoff position-based weighting scheme.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wvoronoi</b></dt>
  <dd class="It-tag">Use the Sibbald/Argos Voronoi sequence weighting algorithm
      in place of the default GSC weighting.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Master man page, with full list of and guide to the individual man pages: see
  <b>hmmer2(1).</b>
<div class="Pp"></div>
For complete documentation, see the user guide
  (ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf); or see the
  HMMER web page, http://hmmer.janelia.org/.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
Freely distributed under the GNU General Public License (GPL).
</pre>
See the file COPYING in your distribution for details on redistribution
  conditions.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Sean Eddy
HHMI/Dept. of Genetics
Washington Univ. School of Medicine
4566 Scott Ave.
St Louis, MO 63110 USA
http://www.genetics.wustl.edu/eddy/
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">Oct 2003</td>
    <td class="foot-os">HMMER 2.3.2</td>
  </tr>
</table>
</body>
</html>
