<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:53 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmm2emit(1) HMMER Manual hmm2emit(1)</p>

<p style="margin-top: 1em">NAME <br>
hmm2emit - generate sequences from a profile HMM</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmm2emit [options] hmmfile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
hmm2emit reads an HMM file from hmmfile containing one or
more HMMs, and generates a number of sequences from each
HMM; or, if the -c option is selected, generate a single
major&acirc; <br>
ity-rule consensus. This can be useful for various
applications in which one needs a simulation of sequences
consistent with a sequence family consensus.</p>

<p style="margin-top: 1em">By default, hmm2emit generates
10 sequences and outputs them in FASTA (unaligned)
format.</p>

<p style="margin-top: 1em">OPTIONS <br>
-a Write the generated sequences in an aligned format
(SELEX) rather than FASTA.</p>

<p style="margin-top: 1em">-c Predict a single
majority-rule consensus sequence instead of sampling
sequences from the HMM&rsquo;s probability distribution.
Highly conserved residues (p &gt;= 0.9 for DNA, p &gt;= <br>
0.5 for protein) are shown in upper case; others are shown
in lower case. Some insert states may become part of the
majority rule consensus, because they are used in &gt;= <br>
50% of generated sequences; when this happens,
insert-generated residues are simply shown as
&quot;x&quot;.</p>

<p style="margin-top: 1em">-h Print brief help; includes
version number and summary of all options, including expert
options.</p>

<p style="margin-top: 1em">-n &lt;n&gt; Generate &lt;n&gt;
sequences. Default is 10.</p>

<p style="margin-top: 1em">-o &lt;f&gt; Save the synthetic
sequences to file &lt;f&gt; rather than writing them to
stdout.</p>

<p style="margin-top: 1em">-q Quiet; suppress all output
except for the sequences themselves. Useful for piping or
directing the output.</p>

<p style="margin-top: 1em">EXPERT OPTIONS <br>
--seed &lt;n&gt; <br>
Set the random seed to &lt;n&gt;, where &lt;n&gt; is a
positive integer. The default is to use time() to generate a
different seed for each run, which means that two different
runs of <br>
hmm2emit on the same HMM will give slightly different
results. You can use this option to generate reproducible
results.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Master man page, with full list of and guide to the
individual man pages: see hmmer2(1).</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide
(ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf);
or see the HMMER web page, http://hmmer.janelia.org/.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 1992-2003 HHMI/Washington University School of
Medicine. <br>
Freely distributed under the GNU General Public License
(GPL). <br>
See the file COPYING in your distribution for details on
redistribution conditions.</p>

<p style="margin-top: 1em">AUTHOR <br>
Sean Eddy <br>
HHMI/Dept. of Genetics <br>
Washington Univ. School of Medicine <br>
4566 Scott Ave. <br>
St Louis, MO 63110 USA <br>
http://www.genetics.wustl.edu/eddy/</p>

<p style="margin-top: 1em">HMMER 2.3.2 Oct 2003
hmm2emit(1)</p>
<hr>
</body>
</html>
