<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>hmm2pfam(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">hmm2pfam(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">hmm2pfam(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">hmm2pfam - search one or more sequences against an HMM
    database</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmm2pfam</b> <i>[options]</i> <i>hmmfile</i> <i>seqfile</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>hmm2pfam</b> reads a sequence file <i>seqfile</i> and compares each sequence
  in it, one at a time, against all the HMMs in <i>hmmfile</i> looking for
  significantly similar sequence matches.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<i>hmmfile</i> will be looked for first in the current working directory, then
  in a directory named by the environment variable <i>HMMERDB.</i> This lets
  administrators install HMM library(s) such as Pfam in a common location.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
There is a separate output report for each sequence in <i>seqfile.</i> This
  report consists of three sections: a ranked list of the best scoring HMMs, a
  list of the best scoring domains in order of their occurrence in the sequence,
  and alignments for all the best scoring domains. A sequence score may be
  higher than a domain score for the same sequence if there is more than one
  domain in the sequence; the sequence score takes into account all the domains.
  All sequences scoring above the <i>-E</i> and <i>-T</i> cutoffs are shown in
  the first list, then <i>every</i> domain found in this list is shown in the
  second list of domain hits. If desired, E-value and bit score thresholds may
  also be applied to the domain list using the <i>--domE</i> and <i>--domT</i>
  options.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Print brief help; includes version number and summary of
      all options, including expert options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">Specify that models and sequence are nucleic acid, not
      protein. Other HMMER programs autodetect this; but because of the order in
      which <b>hmm2pfam</b> accesses data, it can't reliably determine the
      correct &quot;alphabet&quot; by itself.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Limits the alignment output to the <i>&lt;n&gt;</i> best
      scoring domains. <b>-A0</b> shuts off the alignment output and can be used
      to reduce the size of output files.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the E-value cutoff for the per-sequence ranked hit list
      to <i>&lt;x&gt;,</i> where <i>&lt;x&gt;</i> is a positive real number. The
      default is 10.0. Hits with E-values better than (less than) this threshold
      will be shown.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the bit score cutoff for the per-sequence ranked hit
      list to <i>&lt;x&gt;,</i> where <i>&lt;x&gt;</i> is a real number. The
      default is negative infinity; by default, the threshold is controlled by
      E-value and not by bit score. Hits with bit scores better than (greater
      than) this threshold will be shown.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Calculate the E-value scores as if we had seen a sequence
      database of <i>&lt;n&gt;</i> sequences. The default is arbitrarily set to
      59021, the size of Swissprot 34.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXPERT_OPTIONS"><a class="selflink" href="#EXPERT_OPTIONS">EXPERT
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--acc</b></dt>
  <dd class="It-tag">Report HMM accessions instead of names in the output
      reports. Useful for high-throughput annotation, where the data are being
      parsed for storage in a relational database.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--compat</b></dt>
  <dd class="It-tag">Use the output format of HMMER 2.1.1, the 1998-2001 public
      release; provided so 2.1.1 parsers don't have to be rewritten.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cpu</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the maximum number of CPUs that the program will run
      on. The default is to use all CPUs in the machine. Overrides the
      HMMER_NCPU environment variable. Only affects threaded versions of HMMER
      (the default on most systems).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_ga</b></dt>
  <dd class="It-tag">Use Pfam GA (gathering threshold) score cutoffs. Equivalent
      to --globT &lt;GA1&gt; --domT &lt;GA2&gt;, but the GA1 and GA2 cutoffs are
      read from each HMM in <i>hmmfile</i> individually. hmm2build puts these
      cutoffs there if the alignment file was annotated in a Pfam-friendly
      alignment format (extended SELEX or Stockholm format) and the optional GA
      annotation line was present. If these cutoffs are not set in the HMM file,
      <b>--cut_ga</b> doesn't work.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_tc</b></dt>
  <dd class="It-tag">Use Pfam TC (trusted cutoff) score cutoffs. Equivalent to
      --globT &lt;TC1&gt; --domT &lt;TC2&gt;, but the TC1 and TC2 cutoffs are
      read from each HMM in <i>hmmfile</i> individually. hmm2build puts these
      cutoffs there if the alignment file was annotated in a Pfam-friendly
      alignment format (extended SELEX or Stockholm format) and the optional TC
      annotation line was present. If these cutoffs are not set in the HMM file,
      <b>--cut_tc</b> doesn't work.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_nc</b></dt>
  <dd class="It-tag">Use Pfam NC (noise cutoff) score cutoffs. Equivalent to
      --globT &lt;NC1&gt; --domT &lt;NC2&gt;, but the NC1 and NC2 cutoffs are
      read from each HMM in <i>hmmfile</i> individually. hmm2build puts these
      cutoffs there if the alignment file was annotated in a Pfam-friendly
      alignment format (extended SELEX or Stockholm format) and the optional NC
      annotation line was present. If these cutoffs are not set in the HMM file,
      <b>--cut_nc</b> doesn't work.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the E-value cutoff for the per-domain ranked hit list
      to <i>&lt;x&gt;,</i> where <i>&lt;x&gt;</i> is a positive real number. The
      default is infinity; by default, all domains in the sequences that passed
      the first threshold will be reported in the second list, so that the
      number of domains reported in the per-sequence list is consistent with the
      number that appear in the per-domain list.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the bit score cutoff for the per-domain ranked hit list
      to <i>&lt;x&gt;,</i> where <i>&lt;x&gt;</i> is a real number. The default
      is negative infinity; by default, all domains in the sequences that passed
      the first threshold will be reported in the second list, so that the
      number of domains reported in the per-sequence list is consistent with the
      number that appear in the per-domain list. <i>Important note:</i> only one
      domain in a sequence is absolutely controlled by this parameter, or by
      <b>--domT.</b> The second and subsequent domains in a sequence have a de
      facto bit score threshold of 0 because of the details of how HMMER works.
      HMMER requires at least one pass through the main model per sequence; to
      do more than one pass (more than one domain) the multidomain alignment
      must have a better score than the single domain alignment, and hence the
      extra domains must contribute positive score. See the Users' Guide for
      more detail.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--forward</b></dt>
  <dd class="It-tag">Use the Forward algorithm instead of the Viterbi algorithm
      to determine the per-sequence scores. Per-domain scores are still
      determined by the Viterbi algorithm. Some have argued that Forward is a
      more sensitive algorithm for detecting remote sequence homologues; my
      experiments with HMMER have not confirmed this, however.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Assert that the input <i>seqfile</i> is in format
      <i>&lt;s&gt;;</i> do not run Babelfish format autodection. This increases
      the reliability of the program somewhat, because the Babelfish can make
      mistakes; particularly recommended for unattended, high-throughput runs of
      HMMER. Valid format strings include FASTA, GENBANK, EMBL, GCG, PIR,
      STOCKHOLM, SELEX, MSF, CLUSTAL, and PHYLIP. See the User's Guide for a
      complete list.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--null2</b></dt>
  <dd class="It-tag">Turn off the post hoc second null model. By default, each
      alignment is rescored by a postprocessing step that takes into account
      possible biased composition in either the HMM or the target sequence. This
      is almost essential in database searches, especially with local alignment
      models. There is a very small chance that this postprocessing might remove
      real matches, and in these cases <b>--null2</b> may improve sensitivity at
      the expense of reducing specificity by letting biased composition hits
      through.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pvm</b></dt>
  <dd class="It-tag">Run on a Parallel Virtual Machine (PVM). The PVM must
      already be running. The client program <b>hmm2pfam-pvm</b> must be
      installed on all the PVM nodes. The HMM database <i>hmmfile</i> and an
      associated GSI index file <i>hmmfile.</i>gsi must also be installed on all
      the PVM nodes. (The GSI index is produced by the program
      <b>hmm2index</b>.) Because the PVM implementation is I/O bound, it is
      highly recommended that each node have a local copy of <i>hmmfile</i>
      rather than NFS mounting a shared copy. Optional PVM support must have
      been compiled into HMMER for <b>--pvm</b> to function.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--xnu</b></dt>
  <dd class="It-tag">Turn on XNU filtering of target protein sequences. Has no
      effect on nucleic acid sequences. In trial experiments, <b>--xnu</b>
      appears to perform less well than the default post hoc null2 model.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Master man page, with full list of and guide to the individual man pages: see
  <b>hmmer2(1).</b>
<div class="Pp"></div>
For complete documentation, see the user guide
  (ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf); or see the
  HMMER web page, http://hmmer.janelia.org/.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
Freely distributed under the GNU General Public License (GPL).
</pre>
See the file COPYING in your distribution for details on redistribution
  conditions.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Sean Eddy
HHMI/Dept. of Genetics
Washington Univ. School of Medicine
4566 Scott Ave.
St Louis, MO 63110 USA
http://www.genetics.wustl.edu/eddy/
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">Oct 2003</td>
    <td class="foot-os">HMMER 2.3.2</td>
  </tr>
</table>
</body>
</html>
