<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>hmm2emit(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">hmm2emit(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">hmm2emit(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">hmm2emit - generate sequences from a profile HMM</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmm2emit</b> <i>[options]</i> <i>hmmfile</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>hmm2emit</b> reads an HMM file from <i>hmmfile</i> containing one or more
  HMMs, and generates a number of sequences from each HMM; or, if the <b>-c</b>
  option is selected, generate a single majority-rule consensus. This can be
  useful for various applications in which one needs a simulation of sequences
  consistent with a sequence family consensus.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
By default, <b>hmm2emit</b> generates 10 sequences and outputs them in FASTA
  (unaligned) format.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b></dt>
  <dd class="It-tag">Write the generated sequences in an aligned format (SELEX)
      rather than FASTA.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">Predict a single majority-rule consensus sequence instead
      of sampling sequences from the HMM's probability distribution. Highly
      conserved residues (p &gt;= 0.9 for DNA, p &gt;= 0.5 for protein) are
      shown in upper case; others are shown in lower case. Some insert states
      may become part of the majority rule consensus, because they are used in
      &gt;= 50% of generated sequences; when this happens, insert-generated
      residues are simply shown as &quot;x&quot;.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Print brief help; includes version number and summary of
      all options, including expert options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Generate <i>&lt;n&gt;</i> sequences. Default is 10.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save the synthetic sequences to file <i>&lt;f&gt;</i>
      rather than writing them to stdout.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b></dt>
  <dd class="It-tag">Quiet; suppress all output except for the sequences
      themselves. Useful for piping or directing the output.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXPERT_OPTIONS"><a class="selflink" href="#EXPERT_OPTIONS">EXPERT
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the random seed to <i>&lt;n&gt;,</i> where
      <i>&lt;n&gt;</i> is a positive integer. The default is to use
      <b>time()</b> to generate a different seed for each run, which means that
      two different runs of <b>hmm2emit</b> on the same HMM will give slightly
      different results. You can use this option to generate reproducible
      results.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Master man page, with full list of and guide to the individual man pages: see
  <b>hmmer2(1).</b>
<div class="Pp"></div>
For complete documentation, see the user guide
  (ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf); or see the
  HMMER web page, http://hmmer.janelia.org/.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
Freely distributed under the GNU General Public License (GPL).
</pre>
See the file COPYING in your distribution for details on redistribution
  conditions.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Sean Eddy
HHMI/Dept. of Genetics
Washington Univ. School of Medicine
4566 Scott Ave.
St Louis, MO 63110 USA
http://www.genetics.wustl.edu/eddy/
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">Oct 2003</td>
    <td class="foot-os">HMMER 2.3.2</td>
  </tr>
</table>
</body>
</html>
