<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:53 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmm2build(1) HMMER Manual hmm2build(1)</p>

<p style="margin-top: 1em">NAME <br>
hmm2build - build a profile HMM from an alignment</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmm2build [options] hmmfile alignfile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
hmm2build reads a multiple sequence alignment file alignfile
, builds a new profile HMM, and saves the HMM in
hmmfile.</p>

<p style="margin-top: 1em">alignfile may be in ClustalW,
GCG MSF, SELEX, Stockholm, or aligned FASTA alignment
format. The format is automatically detected.</p>

<p style="margin-top: 1em">By default, the model is
configured to find one or more nonoverlapping alignments to
the complete model: multiple global alignments with respect
to the model, and local with <br>
respect to the sequence. This is analogous to the behavior
of the hmmls program of HMMER 1. To configure the model for
multiple local alignments with respect to the model and <br>
local with respect to the sequence, a la the old program
hmmfs, use the -f (fragment) option. More rarely, you may
want to configure the model for a single global alignment
<br>
(global with respect to both model and sequence), using the
-g option; or to configure the model for a single
local/local alignment (a la standard Smith/Waterman, or the
old <br>
hmmsw program), use the -s option.</p>

<p style="margin-top: 1em">OPTIONS <br>
-f Configure the model for finding multiple domains per
sequence, where each domain can be a local (fragmentary)
alignment. This is analogous to the old hmmfs program of
<br>
HMMER 1.</p>

<p style="margin-top: 1em">-g Configure the model for
finding a single global alignment to a target sequence,
analogous to the old hmms program of HMMER 1.</p>

<p style="margin-top: 1em">-h Print brief help; includes
version number and summary of all options, including expert
options.</p>

<p style="margin-top: 1em">-n &lt;s&gt; Name this HMM
&lt;s&gt;. &lt;s&gt; can be any string of non-whitespace
characters (e.g. one &quot;word&quot;). There is no length
limit (at least not one imposed by HMMER; your shell will
<br>
complain about command line lengths first).</p>

<p style="margin-top: 1em">-o &lt;f&gt; Re-save the
starting alignment to &lt;f&gt;, in Stockholm format. The
columns which were assigned to match states will be marked
with x&rsquo;s in an #=RF annotation line. If either <br>
the --hand or --fast construction options were chosen, the
alignment may have been slightly altered to be compatible
with Plan 7 transitions, so saving the final alignment <br>
and comparing to the starting alignment can let you view
these alterations. See the User&rsquo;s Guide for more
information on this arcane side effect.</p>

<p style="margin-top: 1em">-s Configure the model for
finding a single local alignment per target sequence. This
is analogous to the standard Smith/Waterman algorithm or the
hmmsw program of HMMER 1.</p>

<p style="margin-top: 1em">-A Append this model to an
existing hmmfile rather than creating hmmfile. Useful for
building HMM libraries (like Pfam).</p>

<p style="margin-top: 1em">-F Force overwriting of an
existing hmmfile. Otherwise HMMER will refuse to clobber
your existing HMM files, for safety&rsquo;s sake.</p>

<p style="margin-top: 1em">EXPERT OPTIONS <br>
--amino <br>
Force the sequence alignment to be interpreted as amino acid
sequences. Normally HMMER autodetects whether the alignment
is protein or DNA, but sometimes alignments are so <br>
small that autodetection is ambiguous. See --nucleic.</p>

<p style="margin-top: 1em">--archpri &lt;x&gt; <br>
Set the &quot;architecture prior&quot; used by MAP
architecture construction to &lt;x&gt;, where &lt;x&gt; is a
probability between 0 and 1. This parameter governs a
geometric prior distribu&acirc; <br>
tion over model lengths. As &lt;x&gt; increases, longer
models are favored a priori. As &lt;x&gt; decreases, it
takes more residue conservation in a column to make a column
a &quot;con&acirc; <br>
sensus&quot; match column in the model architecture. The
0.85 default has been chosen empirically as a reasonable
setting.</p>

<p style="margin-top: 1em">--binary <br>
Write the HMM to hmmfile in HMMER binary format instead of
readable ASCII text.</p>

<p style="margin-top: 1em">--cfile &lt;f&gt; <br>
Save the observed emission and transition counts to
&lt;f&gt; after the architecture has been determined (e.g.
after residues/gaps have been assigned to match, delete, and
<br>
insert states). This option is used in HMMER development for
generating data files useful for training new Dirichlet
priors. The format of count files is documented in <br>
the User&rsquo;s Guide.</p>

<p style="margin-top: 1em">--fast Quickly and heuristically
determine the architecture of the model by assigning all
columns will more than a certain fraction of gap characters
to insert states. By default <br>
this fraction is 0.5, and it can be changed using the
--gapmax option. The default construction algorithm is a
maximum a posteriori (MAP) algorithm, which is slower.</p>

<p style="margin-top: 1em">--gapmax &lt;x&gt; <br>
Controls the --fast model construction algorithm, but if
--fast is not being used, has no effect. If a column has
more than a fraction &lt;x&gt; of gap symbols in it, it gets
<br>
assigned to an insert column. &lt;x&gt; is a frequency from
0 to 1, and by default is set to 0.5. Higher values of
&lt;x&gt; mean more columns get assigned to consensus, and
models <br>
get longer; smaller values of &lt;x&gt; mean fewer columns
get assigned to consensus, and models get smaller.
&lt;x&gt;</p>

<p style="margin-top: 1em">--hand Specify the architecture
of the model by hand: the alignment file must be in SELEX or
Stockholm format, and the reference annotation line (#=RF in
SELEX, #=GC RF in Stock&acirc; <br>
holm) is used to specify the architecture. Any column marked
with a non-gap symbol (such as an &rsquo;x&rsquo;, for
instance) is assigned as a consensus (match) column in the
model.</p>

<p style="margin-top: 1em">--idlevel &lt;x&gt; <br>
Controls both the determination of effective sequence number
and the behavior of the --wblosum weighting option. The
sequence alignment is clustered by percent identity, <br>
and the number of clusters at a cutoff threshold of
&lt;x&gt; is used to determine the effective sequence
number. Higher values of &lt;x&gt; give more clusters and
higher effective <br>
sequence numbers; lower values of &lt;x&gt; give fewer
clusters and lower effective sequence numbers. &lt;x&gt; is
a fraction from 0 to 1, and by default is set to 0.62
(correspond&acirc; <br>
ing to the clustering level used in constructing the
BLOSUM62 substitution matrix).</p>

<p style="margin-top: 1em">--informat &lt;s&gt; <br>
Assert that the input seqfile is in format &lt;s&gt;; do not
run Babelfish format autodection. This increases the
reliability of the program somewhat, because the Babelfish
can <br>
make mistakes; particularly recommended for unattended,
high-throughput runs of HMMER. Valid format strings include
FASTA, GENBANK, EMBL, GCG, PIR, STOCKHOLM, SELEX, MSF, <br>
CLUSTAL, and PHYLIP. See the User&rsquo;s Guide for a
complete list.</p>

<p style="margin-top: 1em">--noeff <br>
Turn off the effective sequence number calculation, and use
the true number of sequences instead. This will usually
reduce the sensitivity of the final model (so don&rsquo;t do
<br>
it without good reason!)</p>

<p style="margin-top: 1em">--nucleic <br>
Force the alignment to be interpreted as nucleic acid
sequence, either RNA or DNA. Normally HMMER autodetects
whether the alignment is protein or DNA, but sometimes
align&acirc; <br>
ments are so small that autodetection is ambiguous. See
--amino.</p>

<p style="margin-top: 1em">--null &lt;f&gt; <br>
Read a null model from &lt;f&gt;. The default for protein is
to use average amino acid frequencies from Swissprot 34 and
p1 = 350/351; for nucleic acid, the default is to use <br>
0.25 for each base and p1 = 1000/1001. For documentation of
the format of the null model file and further explanation of
how the null model is used, see the User&rsquo;s Guide.</p>

<p style="margin-top: 1em">--pam &lt;f&gt; <br>
Apply a heuristic PAM- (substitution matrix-) based prior on
match emission probabilities instead of the default mixture
Dirichlet. The substitution matrix is read from <br>
&lt;f&gt;. See --pamwgt.</p>

<p style="margin-top: 1em">The default Dirichlet state
transition prior and insert emission prior are unaffected.
Therefore in principle you could combine --prior with --pam
but this isn&rsquo;t recom&acirc; <br>
mended, as it hasn&rsquo;t been tested. ( --pam itself
hasn&rsquo;t been tested much!)</p>

<p style="margin-top: 1em">--pamwgt &lt;x&gt; <br>
Controls the weight on a PAM-based prior. Only has effect if
--pam option is also in use. &lt;x&gt; is a positive real
number, 20.0 by default. &lt;x&gt; is the number of
&quot;pseudo&acirc; <br>
counts&quot; contriubuted by the heuristic prior. Very high
values of &lt;x&gt; can force a scoring system that is
entirely driven by the substitution matrix, making HMMER
somewhat <br>
approximate Gribskov profiles.</p>

<p style="margin-top: 1em">--pbswitch &lt;n&gt; <br>
For alignments with a very large number of sequences, the
GSC, BLOSUM, and Voronoi weighting schemes are slow;
they&rsquo;re O(N^2) for N sequences. Henikoff
position-based <br>
weights (PB weights) are more efficient. At or above a
certain threshold sequence number &lt;n&gt; hmm2build will
switch from GSC, BLOSUM, or Voronoi weights to PB weights.
To <br>
disable this switching behavior (at the cost of compute
time, set &lt;n&gt; to be something larger than the number
of sequences in your alignment. &lt;n&gt; is a positive
integer; <br>
the default is 1000.</p>

<p style="margin-top: 1em">--prior &lt;f&gt; <br>
Read a Dirichlet prior from &lt;f&gt;, replacing the default
mixture Dirichlet. The format of prior files is documented
in the User&rsquo;s Guide, and an example is given in the
<br>
Demos directory of the HMMER distribution.</p>

<p style="margin-top: 1em">--swentry &lt;x&gt; <br>
Controls the total probability that is distributed to local
entries into the model, versus starting at the beginning of
the model as in a global alignment. &lt;x&gt; is a
prob&acirc; <br>
ability from 0 to 1, and by default is set to 0.5. Higher
values of &lt;x&gt; mean that hits that are fragments on
their left (N or 5&rsquo;-terminal) side will be penalized
less, <br>
but complete global alignments will be penalized more. Lower
values of &lt;x&gt; mean that fragments on the left will be
penalized more, and global alignments on this side will <br>
be favored. This option only affects the configurations that
allow local alignments, e.g. -s and -f; unless one of these
options is also activated, this option has no <br>
effect. You have independent control over local/global
alignment behavior for the N/C (5&rsquo;/3&rsquo;) termini
of your target sequences using --swentry and --swexit.</p>

<p style="margin-top: 1em">--swexit &lt;x&gt; <br>
Controls the total probability that is distributed to local
exits from the model, versus ending an alignment at the end
of the model as in a global alignment. &lt;x&gt; is a <br>
probability from 0 to 1, and by default is set to 0.5.
Higher values of &lt;x&gt; mean that hits that are fragments
on their right (C or 3&rsquo;-terminal) side will be
penalized <br>
less, but complete global alignments will be penalized more.
Lower values of &lt;x&gt; mean that fragments on the right
will be penalized more, and global alignments on this <br>
side will be favored. This option only affects the
configurations that allow local alignments, e.g. -s and -f;
unless one of these options is also activated, this option
<br>
has no effect. You have independent control over
local/global alignment behavior for the N/C
(5&rsquo;/3&rsquo;) termini of your target sequences using
--swentry and --swexit.</p>

<p style="margin-top: 1em">--verbose <br>
Print more possibly useful stuff, such as the individual
scores for each sequence in the alignment.</p>

<p style="margin-top: 1em">--wblosum <br>
Use the BLOSUM filtering algorithm to weight the sequences,
instead of the default. Cluster the sequences at a given
percentage identity (see --idlevel); assign each <br>
cluster a total weight of 1.0, distributed equally amongst
the members of that cluster.</p>

<p style="margin-top: 1em">--wgsc Use the
Gerstein/Sonnhammer/Chothia ad hoc sequence weighting
algorithm. This is already the default, so this option has
no effect (unless it follows another option in the <br>
-&minus;w family, in which case it overrides it).</p>

<p style="margin-top: 1em">--wme Use the Krogh/Mitchison
maximum entropy algorithm to &quot;weight&quot; the
sequences. This supersedes the Eddy/Mitchison/Durbin maximum
discrimination algorithm, which gives almost <br>
identical weights but is less robust. ME weighting seems to
give a marginal increase in sensitivity over the default GSC
weights, but takes a fair amount of time.</p>

<p style="margin-top: 1em">--wnone <br>
Turn off all sequence weighting.</p>

<p style="margin-top: 1em">--wpb Use the Henikoff
position-based weighting scheme.</p>

<p style="margin-top: 1em">--wvoronoi <br>
Use the Sibbald/Argos Voronoi sequence weighting algorithm
in place of the default GSC weighting.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Master man page, with full list of and guide to the
individual man pages: see hmmer2(1).</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide
(ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf);
or see the HMMER web page, http://hmmer.janelia.org/.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 1992-2003 HHMI/Washington University School of
Medicine. <br>
Freely distributed under the GNU General Public License
(GPL). <br>
See the file COPYING in your distribution for details on
redistribution conditions.</p>

<p style="margin-top: 1em">AUTHOR <br>
Sean Eddy <br>
HHMI/Dept. of Genetics <br>
Washington Univ. School of Medicine <br>
4566 Scott Ave. <br>
St Louis, MO 63110 USA <br>
http://www.genetics.wustl.edu/eddy/</p>

<p style="margin-top: 1em">HMMER 2.3.2 Oct 2003
hmm2build(1)</p>
<hr>
</body>
</html>
