<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:53 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmm2align(1) HMMER Manual hmm2align(1)</p>

<p style="margin-top: 1em">NAME <br>
hmm2align - align sequences to an HMM profile</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmm2align [options] hmm2file seqfile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
hmm2align reads an HMM file from hmmfile and a set of
sequences from seqfile, aligns the sequences to the profile
HMM, and outputs a multiple sequence alignment.</p>

<p style="margin-top: 1em">seqfile may be in any unaligned
or aligned file format accepted by HMMER. If it is in a
multiple alignment format (e.g. Stockholm, MSF, SELEX,
ClustalW), the existing alignment <br>
is ignored (i.e., the sequences are read as if they were
unaligned - hmm2align will align them the way it wants).</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Print brief help; includes version number and summary of
all options, including expert options.</p>

<p style="margin-top: 1em">-m Include in the alignment only
those symbols aligned to match states. Do not show symbols
assigned to insert states.</p>

<p style="margin-top: 1em">-o &lt;f&gt; Save alignment to
file &lt;f&gt; instead of to standard output.</p>

<p style="margin-top: 1em">-q quiet; suppress all output
except the alignment itself. Useful for piping or
redirecting the output.</p>

<p style="margin-top: 1em">EXPERT OPTIONS <br>
--informat &lt;s&gt; <br>
Assert that the input seqfile is in format &lt;s&gt;; do not
run Babelfish format autodection. This increases the
reliability of the program somewhat, because the Babelfish
can <br>
make mistakes; particularly recommended for unattended,
high-throughput runs of HMMER. Valid format strings include
FASTA, GENBANK, EMBL, GCG, PIR, STOCKHOLM, SELEX, MSF, <br>
CLUSTAL, and PHYLIP. See the User&rsquo;s Guide for a
complete list.</p>

<p style="margin-top: 1em">--mapali &lt;f&gt; <br>
Reads an alignment from file &lt;f&gt; and aligns it as a
single object to the HMM; e.g. the alignment in &lt;f&gt; is
held fixed. This allows you to align sequences to a model
with <br>
hmm2align and view them in the context of an existing
trusted multiple alignment. The alignment to the alignment
is defined by a &quot;map&quot; kept in the HMM, and so is
fast and <br>
guaranteed to be consistent with the way the HMM was
constructed from the alignment. The alignment in the file
&lt;f&gt; must be exactly the alignment that the HMM was
built <br>
from. Compare the --withali option.</p>

<p style="margin-top: 1em">--oneline <br>
Output the alignment with one line per sequence, rather than
interleaving the sequence alignment blocks. Only affects
Stockholm format output.</p>

<p style="margin-top: 1em">--outformat &lt;s&gt; <br>
Output the alignment in format &lt;s&gt;. The default is
Stockholm format. Valid formats include Stockholm, SELEX,
MSF, Clustal, Phylip, and A2M.</p>

<p style="margin-top: 1em">--withali &lt;f&gt; <br>
Reads an alignment from file &lt;f&gt; and aligns it as a
single object to the HMM; e.g. the alignment in &lt;f&gt; is
held fixed. This allows you to align sequences to a model
with <br>
hmm2align and view them in the context of an existing
trusted multiple alignment. The alignment to the alignment
is done with a heuristic (nonoptimal) dynamic programming
<br>
procedure, which may be somewhat slow and is not guaranteed
to be completely consistent with the way the HMM was
constructed (though it should be quite close). However, <br>
any alignment can be used, not just the alignment that the
HMM was built from. Compare the --mapali option.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Master man page, with full list of and guide to the
individual man pages: see hmmer2(1).</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide
(ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf);
or see the HMMER web page, http://hmmer.janelia.org/.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 1992-2003 HHMI/Washington University School of
Medicine. <br>
Freely distributed under the GNU General Public License
(GPL). <br>
See the file COPYING in your distribution for details on
redistribution conditions.</p>

<p style="margin-top: 1em">AUTHOR <br>
Sean Eddy <br>
HHMI/Dept. of Genetics <br>
Washington Univ. School of Medicine <br>
4566 Scott Ave. <br>
St Louis, MO 63110 USA <br>
http://www.genetics.wustl.edu/eddy/</p>

<p style="margin-top: 1em">HMMER 2.3.2 Oct 2003
hmm2align(1)</p>
<hr>
</body>
</html>
