<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>hmm2align(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">hmm2align(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">hmm2align(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">hmm2align - align sequences to an HMM profile</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmm2align</b> <i>[options]</i> <i>hmm2file</i> <i>seqfile</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>hmm2align</b> reads an HMM file from <i>hmmfile</i> and a set of sequences
  from <i>seqfile,</i> aligns the sequences to the profile HMM, and outputs a
  multiple sequence alignment.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<i>seqfile</i> may be in any unaligned or aligned file format accepted by HMMER.
  If it is in a multiple alignment format (e.g. Stockholm, MSF, SELEX,
  ClustalW), the existing alignment is ignored (i.e., the sequences are read as
  if they were unaligned - hmm2align will align them the way it wants).
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Print brief help; includes version number and summary of
      all options, including expert options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b></dt>
  <dd class="It-tag">Include in the alignment only those symbols aligned to
      match states. Do not show symbols assigned to insert states.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save alignment to file <i>&lt;f&gt;</i> instead of to
      standard output.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b></dt>
  <dd class="It-tag">quiet; suppress all output except the alignment itself.
      Useful for piping or redirecting the output.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXPERT_OPTIONS"><a class="selflink" href="#EXPERT_OPTIONS">EXPERT
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Assert that the input <i>seqfile</i> is in format
      <i>&lt;s&gt;;</i> do not run Babelfish format autodection. This increases
      the reliability of the program somewhat, because the Babelfish can make
      mistakes; particularly recommended for unattended, high-throughput runs of
      HMMER. Valid format strings include FASTA, GENBANK, EMBL, GCG, PIR,
      STOCKHOLM, SELEX, MSF, CLUSTAL, and PHYLIP. See the User's Guide for a
      complete list.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mapali</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Reads an alignment from file <i>&lt;f&gt;</i> and aligns it
      as a single object to the HMM; e.g. the alignment in <i>&lt;f&gt;</i> is
      held fixed. This allows you to align sequences to a model with
      <b>hmm2align</b> and view them in the context of an existing trusted
      multiple alignment. The alignment to the alignment is defined by a
      &quot;map&quot; kept in the HMM, and so is fast and guaranteed to be
      consistent with the way the HMM was constructed from the alignment. The
      alignment in the file <i>&lt;f&gt;</i> must be exactly the alignment that
      the HMM was built from. Compare the <b>--withali</b> option.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--oneline</b></dt>
  <dd class="It-tag">Output the alignment with one line per sequence, rather
      than interleaving the sequence alignment blocks. Only affects Stockholm
      format output.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Output the alignment in format <i>&lt;s&gt;.</i> The
      default is Stockholm format. Valid formats include Stockholm, SELEX, MSF,
      Clustal, Phylip, and A2M.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--withali</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Reads an alignment from file <i>&lt;f&gt;</i> and aligns it
      as a single object to the HMM; e.g. the alignment in <i>&lt;f&gt;</i> is
      held fixed. This allows you to align sequences to a model with
      <b>hmm2align</b> and view them in the context of an existing trusted
      multiple alignment. The alignment to the alignment is done with a
      heuristic (nonoptimal) dynamic programming procedure, which may be
      somewhat slow and is not guaranteed to be completely consistent with the
      way the HMM was constructed (though it should be quite close). However,
      any alignment can be used, not just the alignment that the HMM was built
      from. Compare the <b>--mapali</b> option.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Master man page, with full list of and guide to the individual man pages: see
  <b>hmmer2(1).</b>
<div class="Pp"></div>
For complete documentation, see the user guide
  (ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf); or see the
  HMMER web page, http://hmmer.janelia.org/.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
Freely distributed under the GNU General Public License (GPL).
</pre>
See the file COPYING in your distribution for details on redistribution
  conditions.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Sean Eddy
HHMI/Dept. of Genetics
Washington Univ. School of Medicine
4566 Scott Ave.
St Louis, MO 63110 USA
http://www.genetics.wustl.edu/eddy/
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">Oct 2003</td>
    <td class="foot-os">HMMER 2.3.2</td>
  </tr>
</table>
</body>
</html>
