<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmmalign(1) HMMER Manual hmmalign(1)</p>

<p style="margin-top: 1em">NAME <br>
hmmalign - align sequences to a profile HMM</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmmalign [options] &lt;hmmfile&gt; &lt;seqfile&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Perform a multiple sequence alignment of all the sequences
in &lt;seqfile&gt; by aligning them individually to the
profile HMM in &lt;hmmfile&gt;. The new alignment is output
to stdout in <br>
Stockholm format.</p>

<p style="margin-top: 1em">The &lt;hmmfile&gt; should
contain only a single profile. If it contains more, only the
first profile in the file will be used.</p>

<p style="margin-top: 1em">Either &lt;hmmfile&gt; or
&lt;seqfile&gt; (but not both) may be &rsquo;-&rsquo;
(dash), which means reading this input from stdin rather
than a file.</p>

<p style="margin-top: 1em">The sequences in &lt;seqfile&gt;
are aligned in unihit local alignment mode. Therefore they
should already be known to contain only a single domain (or
a fragment of one). The opti&acirc; <br>
mal alignment may assign some residues as nonhomologous (N
and C states), in which case these residues are still
included in the resulting alignment, but shoved to the outer
<br>
edges. To trim these unaligned nonhomologous residues from
the result, see the --trim option.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
all available options.</p>

<p style="margin-top: 1em">-o &lt;f&gt; Direct the output
alignment to file &lt;f&gt;, rather than to stdout.</p>

<p style="margin-top: 1em">--mapali &lt;f&gt; <br>
Merge the existing alignment in file &lt;f&gt; into the
result, where &lt;f&gt; is exactly the same alignment that
was used to build the model in &lt;hmmfile&gt;. This is done
using a map <br>
of alignment columns to consensus profile positions that is
stored in the &lt;hmmfile&gt;. The multiple alignment in
&lt;f&gt; will be exactly reproduced in its consensus
columns (as <br>
defined by the profile), but the displayed alignment in
insert columns may be altered, because insertions relative
to a profile are considered by convention to be <br>
unaligned data.</p>

<p style="margin-top: 1em">--trim Trim nonhomologous
residues (assigned to N and C states in the optimal
alignments) from the resulting multiple alignment
output.</p>

<p style="margin-top: 1em">--amino <br>
Specify that all sequences in &lt;seqfile&gt; are proteins.
By default, alphabet type is autodetected from looking at
the residue composition.</p>

<p style="margin-top: 1em">--dna Specify that all sequences
in &lt;seqfile&gt; are DNAs.</p>

<p style="margin-top: 1em">--rna Specify that all sequences
in &lt;seqfile&gt; are RNAs.</p>

<p style="margin-top: 1em">--informat &lt;s&gt; <br>
Declare that the input &lt;seqfile&gt; is in format
&lt;s&gt;. Accepted sequence file formats include FASTA,
EMBL, GenBank, DDBJ, UniProt, Stockholm, and SELEX. Default
is to autode&acirc; <br>
tect the format of the file.</p>

<p style="margin-top: 1em">--outformat &lt;s&gt; <br>
Specify that the output multiple alignment is in format
&lt;s&gt;. Currently the accepted multiple alignment
sequence file formats only include Stockholm and SELEX.</p>

<p style="margin-top: 1em">SEE ALSO <br>
See hmmer(1) for a master man page with a list of all the
individual man pages for programs in the HMMER package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
The Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
hmmalign(1)</p>
<hr>
</body>
</html>
