<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>hmmalign(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">hmmalign(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">hmmalign(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
hmmalign - align sequences to a profile HMM
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmmalign</b> <i>[options]</i> <i>&lt;hmmfile&gt;</i> <i>&lt;seqfile&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Perform a multiple sequence alignment of all the sequences in
  <i>&lt;seqfile&gt;</i> by aligning them individually to the profile HMM in
  <i>&lt;hmmfile&gt;.</i> The new alignment is output to <i>stdout</i> in
  Stockholm format.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The <i>&lt;hmmfile&gt;</i> should contain only a single profile. If it contains
  more, only the first profile in the file will be used.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Either <i>&lt;hmmfile&gt;</i> or <i>&lt;seqfile&gt;</i> (but not both) may be
  '-' (dash), which means reading this input from <i>stdin</i> rather than a
  file.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The sequences in <i>&lt;seqfile&gt;</i> are aligned in unihit local alignment
  mode. Therefore they should already be known to contain only a single domain
  (or a fragment of one). The optimal alignment may assign some residues as
  nonhomologous (N and C states), in which case these residues are still
  included in the resulting alignment, but shoved to the outer edges. To trim
  these unaligned nonhomologous residues from the result, see the <b>--trim</b>
  option.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and all
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Direct the output alignment to file <i>&lt;f&gt;,</i>
      rather than to <i>stdout.</i>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mapali</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Merge the existing alignment in file <i>&lt;f&gt;</i> into
      the result, where <i>&lt;f&gt;</i> is exactly the same alignment that was
      used to build the model in <i>&lt;hmmfile&gt;.</i> This is done using a
      map of alignment columns to consensus profile positions that is stored in
      the <i>&lt;hmmfile&gt;.</i> The multiple alignment in <i>&lt;f&gt;</i>
      will be exactly reproduced in its consensus columns (as defined by the
      profile), but the displayed alignment in insert columns may be altered,
      because insertions relative to a profile are considered by convention to
      be unaligned data.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trim</b></dt>
  <dd class="It-tag">Trim nonhomologous residues (assigned to N and C states in
      the optimal alignments) from the resulting multiple alignment output.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--amino</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>&lt;seqfile&gt;</i> are
      proteins. By default, alphabet type is autodetected from looking at the
      residue composition.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dna</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>&lt;seqfile&gt;</i> are
      DNAs.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rna</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>&lt;seqfile&gt;</i> are
      RNAs.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Declare that the input <i>&lt;seqfile&gt;</i> is in format
      <i>&lt;s&gt;.</i> Accepted sequence file formats include FASTA, EMBL,
      GenBank, DDBJ, UniProt, Stockholm, and SELEX. Default is to autodetect the
      format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Specify that the output multiple alignment is in format
      <i>&lt;s&gt;.</i> Currently the accepted multiple alignment sequence file
      formats only include Stockholm and SELEX.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>hmmer(1)</b> for a master man page with a list of all the individual man
  pages for programs in the HMMER package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your HMMER
  distribution (Userguide.pdf); or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2015 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2015</td>
    <td class="foot-os">HMMER 3.1b2</td>
  </tr>
</table>
</body>
</html>
