<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmmbuild(1) HMMER Manual hmmbuild(1)</p>

<p style="margin-top: 1em">NAME <br>
hmmbuild - construct profile HMM(s) from multiple sequence
alignment(s)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmmbuild [options] &lt;hmmfile_out&gt; &lt;msafile&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
For each multiple sequence alignment in &lt;msafile&gt;
build a profile HMM and save it to a new file
&lt;hmmfile_out&gt;.</p>

<p style="margin-top: 1em">&lt;msafile&gt; may be
&rsquo;-&rsquo; (dash), which means reading this input from
stdin rather than a file. To use &rsquo;-&rsquo;, you must
also specify the alignment file format with --informat
&lt;s&gt;, as in <br>
--informat stockholm (because of a current limitation in our
implementation, MSA file formats cannot be autodetected in a
nonrewindable input stream.)</p>

<p style="margin-top: 1em">&lt;hmmfile_out&gt; may not be
&rsquo;-&rsquo; (stdout), because sending the HMM file to
stdout would conflict with the other text output of the
program.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
all available options.</p>

<p style="margin-top: 1em">-n &lt;s&gt; Name the new
profile &lt;s&gt;. The default is to use the name of the
alignment (if one is present in the msafile, or, failing
that, the name of the hmmfile. If msafile con&acirc; <br>
tains more than one alignment, -n doesn&rsquo;t work, and
every alignment must have a name annotated in the msafile
(as in Stockholm #=GF ID annotation).</p>

<p style="margin-top: 1em">-o &lt;f&gt; Direct the summary
output to file &lt;f&gt;, rather than to stdout.</p>

<p style="margin-top: 1em">-O &lt;f&gt; After each model is
constructed, resave annotated, possibly modified source
alignments to a file &lt;f&gt; in Stockholm format. The
alignments are annotated with a reference <br>
annotation line indicating which columns were assigned as
consensus, and sequences are annotated with what relative
sequence weights were assigned. Some residues of the <br>
alignment may have been shifted to accommodate restrictions
of the Plan7 profile architecture, which disallows
transitions between insert and delete states.</p>

<p style="margin-top: 1em">OPTIONS FOR SPECIFYING THE
ALPHABET <br>
The alphabet type (amino, DNA, or RNA) is autodetected by
default, by looking at the composition of the msafile.
Autodetection is normally quite reliable, but occasionally
<br>
alphabet type may be ambiguous and autodetection can fail
(for instance, on tiny toy alignments of just a few
residues). To avoid this, or to increase robustness in
automated <br>
analysis pipelines, you may specify the alphabet type of
msafile with these options.</p>

<p style="margin-top: 1em">--amino <br>
Specify that all sequences in msafile are proteins.</p>

<p style="margin-top: 1em">--dna Specify that all sequences
in msafile are DNAs.</p>

<p style="margin-top: 1em">--rna Specify that all sequences
in msafile are RNAs.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING PROFILE
CONSTRUCTION <br>
These options control how consensus columns are defined in
an alignment.</p>

<p style="margin-top: 1em">--fast Define consensus columns
as those that have a fraction &gt;= symfrac of residues as
opposed to gaps. (See below for the --symfrac option.) This
is the default.</p>

<p style="margin-top: 1em">--hand Define consensus columns
in next profile using reference annotation to the multiple
alignment. This allows you to define any consensus columns
you like.</p>

<p style="margin-top: 1em">--symfrac &lt;x&gt; <br>
Define the residue fraction threshold necessary to define a
consensus column when using the --fast option. The default
is 0.5. The symbol fraction in each column is calcu&acirc;
<br>
lated after taking relative sequence weighting into account,
and ignoring gap characters corresponding to ends of
sequence fragments (as opposed to internal inser&acirc; <br>
tions/deletions). Setting this to 0.0 means that every
alignment column will be assigned as consensus, which may be
useful in some cases. Setting it to 1.0 means that <br>
only columns that include 0 gaps (internal
insertions/deletions) will be assigned as consensus.</p>

<p style="margin-top: 1em">--fragthresh &lt;x&gt; <br>
We only want to count terminal gaps as deletions if the
aligned sequence is known to be full-length, not if it is a
fragment (for instance, because only part of it was <br>
sequenced). HMMER uses a simple rule to infer fragments: if
the range of a sequence in the alignment (the number of
alignment columns between the first and last positions <br>
of the sequence) is less than or equal to a fraction
&lt;x&gt; times the alignment length in columns, then the
sequence is handled as a fragment. The default is 0.5.
Setting <br>
--fragthresh0 will define no (nonempty) sequence as a
fragment; you might want to do this if you know you&rsquo;ve
got a carefully curated alignment of full-length sequences.
<br>
Setting --fragthresh1 will define all sequences as
fragments; you might want to do this if you know your
alignment is entirely composed of fragments, such as
translated <br>
short reads in metagenomic shotgun data.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING RELATIVE
WEIGHTS <br>
HMMER uses an ad hoc sequence weighting algorithm to
downweight closely related sequences and upweight distantly
related ones. This has the effect of making models less
biased by <br>
uneven phylogenetic representation. For example, two
identical sequences would typically each receive half the
weight that one sequence would. These options control which
algo&acirc; <br>
rithm gets used.</p>

<p style="margin-top: 1em">--wpb Use the Henikoff
position-based sequence weighting scheme [Henikoff and
Henikoff, J. Mol. Biol. 243:574, 1994]. This is the
default.</p>

<p style="margin-top: 1em">--wgsc Use the
Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et
al, J. Mol. Biol. 235:1067, 1994].</p>

<p style="margin-top: 1em">--wblosum <br>
Use the same clustering scheme that was used to weight data
in calculating BLOSUM subsitution matrices [Henikoff and
Henikoff, Proc. Natl. Acad. Sci 89:10915, 1992]. <br>
Sequences are single-linkage clustered at an identity
threshold (default 0.62; see --wid) and within each cluster
of c sequences, each sequence gets relative weight 1/c.</p>

<p style="margin-top: 1em">--wnone <br>
No relative weights. All sequences are assigned uniform
weight.</p>

<p style="margin-top: 1em">--wid &lt;x&gt; <br>
Sets the identity threshold used by single-linkage
clustering when using --wblosum. Invalid with any other
weighting scheme. Default is 0.62.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING EFFECTIVE
SEQUENCE NUMBER <br>
After relative weights are determined, they are normalized
to sum to a total effective sequence number, eff_nseq. This
number may be the actual number of sequences in the
align&acirc; <br>
ment, but it is almost always smaller than that. The default
entropy weighting method (--eent) reduces the effective
sequence number to reduce the information content (relative
<br>
entropy, or average expected score on true homologs) per
consensus position. The target relative entropy is
controlled by a two-parameter function, where the two
parameters are <br>
settable with --ere and --esigma.</p>

<p style="margin-top: 1em">--eent Adjust effective sequence
number to achieve a specific relative entropy per position
(see --ere). This is the default.</p>

<p style="margin-top: 1em">--eclust <br>
Set effective sequence number to the number of
single-linkage clusters at a specific identity threshold
(see --eid). This option is not recommended; it&rsquo;s for
experiments <br>
evaluating how much better --eent is.</p>

<p style="margin-top: 1em">--enone <br>
Turn off effective sequence number determination and just
use the actual number of sequences. One reason you might
want to do this is to try to maximize the relative <br>
entropy/position of your model, which may be useful for
short models.</p>

<p style="margin-top: 1em">--eset &lt;x&gt; <br>
Explicitly set the effective sequence number for all models
to &lt;x&gt;.</p>

<p style="margin-top: 1em">--ere &lt;x&gt; <br>
Set the minimum relative entropy/position target to
&lt;x&gt;. Requires --eent. Default depends on the sequence
alphabet. For protein sequences, it is 0.59 bits/position;
for <br>
nucleotide sequences, it is 0.45 bits/position.</p>

<p style="margin-top: 1em">--esigma &lt;x&gt; <br>
Sets the minimum relative entropy contributed by an entire
model alignment, over its whole length. This has the effect
of making short models have higher relative entropy <br>
per position than --ere alone would give. The default is
45.0 bits.</p>

<p style="margin-top: 1em">--eid &lt;x&gt; <br>
Sets the fractional pairwise identity cutoff used by single
linkage clustering with the --eclust option. The default is
0.62.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING PRIORS <br>
By default, weighted counts are converted to mean posterior
probability parameter estimates using mixture Dirichlet
priors. Default mixture Dirichlet prior parameters for
pro&acirc; <br>
tein models and for nucleic acid (RNA and DNA) models are
built in. The following options allow you to override the
default priors.</p>

<p style="margin-top: 1em">--pnone <br>
Don&rsquo;t use any priors. Probability parameters will
simply be the observed frequencies, after relative sequence
weighting.</p>

<p style="margin-top: 1em">--plaplace <br>
Use a Laplace +1 prior in place of the default mixture
Dirichlet prior.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING E-VALUE
CALIBRATION <br>
The location parameters for the expected score distributions
for MSV filter scores, Viterbi filter scores, and Forward
scores require three short random sequence simulations.</p>

<p style="margin-top: 1em">--EmL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter mu for MSV filter E-values. Default is
200.</p>

<p style="margin-top: 1em">--EmN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter mu for MSV filter E-values. Default
is 200.</p>

<p style="margin-top: 1em">--EvL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter mu for Viterbi filter E-values. Default
is 200.</p>

<p style="margin-top: 1em">--EvN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter mu for Viterbi filter E-values.
Default is 200.</p>

<p style="margin-top: 1em">--EfL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter tau for Forward E-values. Default is
100.</p>

<p style="margin-top: 1em">--EfN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter tau for Forward E-values. Default is
200.</p>

<p style="margin-top: 1em">--Eft &lt;x&gt; <br>
Sets the tail mass fraction to fit in the simulation that
estimates the location parameter tau for Forward evalues.
Default is 0.04.</p>

<p style="margin-top: 1em">OTHER OPTIONS <br>
--cpu &lt;n&gt; <br>
Set the number of parallel worker threads to &lt;n&gt;. By
default, HMMER sets this to the number of CPU cores it
detects in your machine - that is, it tries to maximize the
<br>
use of your available processor cores. Setting &lt;n&gt;
higher than the number of available cores is of little if
any value, but you may want to set it to something less. You
<br>
can also control this number by setting an environment
variable, HMMER_NCPU.</p>

<p style="margin-top: 1em">This option is only available if
HMMER was compiled with POSIX threads support. This is the
default, but it may have been turned off for your site or
machine for some rea&acirc; <br>
son.</p>

<p style="margin-top: 1em">--informat &lt;s&gt; <br>
Declare that the input msafile is in format &lt;s&gt;.
Currently the accepted multiple alignment sequence file
formats include Stockholm, Aligned FASTA, Clustal, NCBI PSI-
<br>
BLAST, PHYLIP, Selex, and UCSC SAM A2M. Default is to
autodetect the format of the file.</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Seed the random number generator with &lt;n&gt;, an integer
&gt;= 0. If &lt;n&gt; is nonzero, any stochastic simulations
will be reproducible; the same command will give the same
<br>
results. If &lt;n&gt; is 0, the random number generator is
seeded arbitrarily, and stochastic simulations will vary
from run to run of the same command. The default seed is
<br>
42.</p>

<p style="margin-top: 1em">--w_beta &lt;x&gt; <br>
Window length tail mass. The upper bound, W, on the length
at which nhmmer expects to find an instance of the model is
set such that the fraction of all sequences gener&acirc;
<br>
ated by the model with length &gt;= W is less than
&lt;x&gt;. The default is 1e-7.</p>

<p style="margin-top: 1em">--w_length &lt;n&gt; <br>
Override the model instance length upper bound, W, which is
otherwise controlled by --w_beta. It should be larger than
the model length. The value of W is used deep in <br>
the acceleration pipeline, and modest changes are not
expected to impact results (though larger values of W do
lead to longer run time).</p>

<p style="margin-top: 1em">--mpi Run as a parallel MPI
program. Each alignment is assigned to a MPI worker node for
construction. (Therefore, the maximum parallelization cannot
exceed the number of align&acirc; <br>
ments in the input msafile.) This is useful when building
large profile libraries. This option is only available if
optional MPI capability was enabled at compile-time.</p>

<p style="margin-top: 1em">--stall <br>
For debugging MPI parallelization: arrest program execution
immediately after start, and wait for a debugger to attach
to the running process and release the arrest.</p>

<p style="margin-top: 1em">--maxinsertlen &lt;n&gt; <br>
Restrict insert length parameterization such that the
expected insert length at each position of the model is no
more than &lt;n&gt;.</p>

<p style="margin-top: 1em">SEE ALSO <br>
See hmmer(1) for a master man page with a list of all the
individual man pages for programs in the HMMER package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
hmmbuild(1)</p>
<hr>
</body>
</html>
