<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>hmmbuild(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">hmmbuild(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">hmmbuild(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
hmmbuild - construct profile HMM(s) from multiple sequence alignment(s)
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmmbuild</b> <i>[options]</i> <i>&lt;hmmfile_out&gt;</i>
  <i>&lt;msafile&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
For each multiple sequence alignment in <i>&lt;msafile&gt;</i> build a profile
  HMM and save it to a new file <i>&lt;hmmfile_out&gt;</i>.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<i>&lt;msafile&gt;</i> may be '-' (dash), which means reading this input from
  <i>stdin</i> rather than a file. To use '-', you must also specify the
  alignment file format with <b>--informat</b><i> &lt;s&gt;</i><b>,</b> as in
  <b>--informat stockholm</b> (because of a current limitation in our
  implementation, MSA file formats cannot be autodetected in a nonrewindable
  input stream.)
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<i>&lt;hmmfile_out&gt;</i> may not be '-' <i>(stdout)</i>, because sending the
  HMM file to <i>stdout</i> would conflict with the other text output of the
  program.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and all
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Name the new profile <i>&lt;s&gt;</i>. The default is to
      use the name of the alignment (if one is present in the <i>msafile</i>,
      or, failing that, the name of the <i>hmmfile</i>. If <i>msafile</i>
      contains more than one alignment, <i>-n</i> doesn't work, and every
      alignment must have a name annotated in the <i>msafile</i> (as in
      Stockholm #=GF ID annotation).
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Direct the summary output to file <i>&lt;f&gt;</i>, rather
      than to <i>stdout</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-O</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">After each model is constructed, resave annotated, possibly
      modified source alignments to a file <i>&lt;f&gt;</i> in Stockholm format.
      The alignments are annotated with a reference annotation line indicating
      which columns were assigned as consensus, and sequences are annotated with
      what relative sequence weights were assigned. Some residues of the
      alignment may have been shifted to accommodate restrictions of the Plan7
      profile architecture, which disallows transitions between insert and
      delete states.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_SPECIFYING_THE_ALPHABET"><a class="selflink" href="#OPTIONS_FOR_SPECIFYING_THE_ALPHABET">OPTIONS
  FOR SPECIFYING THE ALPHABET</a></h1>
The alphabet type (amino, DNA, or RNA) is autodetected by default, by looking at
  the composition of the <i>msafile</i>. Autodetection is normally quite
  reliable, but occasionally alphabet type may be ambiguous and autodetection
  can fail (for instance, on tiny toy alignments of just a few residues). To
  avoid this, or to increase robustness in automated analysis pipelines, you may
  specify the alphabet type of <i>msafile</i> with these options.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--amino</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>msafile</i> are proteins.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dna</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>msafile</i> are DNAs.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rna</b></dt>
  <dd class="It-tag">Specify that all sequences in <i>msafile</i> are RNAs.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_PROFILE_CONSTRUCTION"><a class="selflink" href="#OPTIONS_CONTROLLING_PROFILE_CONSTRUCTION">OPTIONS
  CONTROLLING PROFILE CONSTRUCTION</a></h1>
These options control how consensus columns are defined in an alignment.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fast</b></dt>
  <dd class="It-tag">Define consensus columns as those that have a fraction
      &gt;= <b>symfrac</b> of residues as opposed to gaps. (See below for the
      <b>--symfrac</b> option.) This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hand</b></dt>
  <dd class="It-tag">Define consensus columns in next profile using reference
      annotation to the multiple alignment. This allows you to define any
      consensus columns you like.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--symfrac</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Define the residue fraction threshold necessary to define a
      consensus column when using the <b>--fast</b> option. The default is 0.5.
      The symbol fraction in each column is calculated after taking relative
      sequence weighting into account, and ignoring gap characters corresponding
      to ends of sequence fragments (as opposed to internal
      insertions/deletions). Setting this to 0.0 means that every alignment
      column will be assigned as consensus, which may be useful in some cases.
      Setting it to 1.0 means that only columns that include 0 gaps (internal
      insertions/deletions) will be assigned as consensus.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fragthresh</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">We only want to count terminal gaps as deletions if the
      aligned sequence is known to be full-length, not if it is a fragment (for
      instance, because only part of it was sequenced). HMMER uses a simple rule
      to infer fragments: if the range of a sequence in the alignment (the
      number of alignment columns between the first and last positions of the
      sequence) is less than or equal to a fraction <i>&lt;x&gt;</i> times the
      alignment length in columns, then the sequence is handled as a fragment.
      The default is 0.5. Setting <b>--fragthresh</b><i>0</i> will define no
      (nonempty) sequence as a fragment; you might want to do this if you know
      you've got a carefully curated alignment of full-length sequences. Setting
      <b>--fragthresh</b><i>1</i> will define all sequences as fragments; you
      might want to do this if you know your alignment is entirely composed of
      fragments, such as translated short reads in metagenomic shotgun data.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_RELATIVE_WEIGHTS"><a class="selflink" href="#OPTIONS_CONTROLLING_RELATIVE_WEIGHTS">OPTIONS
  CONTROLLING RELATIVE WEIGHTS</a></h1>
HMMER uses an ad hoc sequence weighting algorithm to downweight closely related
  sequences and upweight distantly related ones. This has the effect of making
  models less biased by uneven phylogenetic representation. For example, two
  identical sequences would typically each receive half the weight that one
  sequence would. These options control which algorithm gets used.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wpb</b></dt>
  <dd class="It-tag">Use the Henikoff position-based sequence weighting scheme
      [Henikoff and Henikoff, J. Mol. Biol. 243:574, 1994]. This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wgsc</b></dt>
  <dd class="It-tag">Use the Gerstein/Sonnhammer/Chothia weighting algorithm
      [Gerstein et al, J. Mol. Biol. 235:1067, 1994].
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wblosum</b></dt>
  <dd class="It-tag">Use the same clustering scheme that was used to weight data
      in calculating BLOSUM subsitution matrices [Henikoff and Henikoff, Proc.
      Natl. Acad. Sci 89:10915, 1992]. Sequences are single-linkage clustered at
      an identity threshold (default 0.62; see <i>--wid</i>) and within each
      cluster of c sequences, each sequence gets relative weight 1/c.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wnone</b></dt>
  <dd class="It-tag">No relative weights. All sequences are assigned uniform
      weight.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wid</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the identity threshold used by single-linkage
      clustering when using <i>--wblosum</i>. Invalid with any other weighting
      scheme. Default is 0.62.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_EFFECTIVE_SEQUENCE_NUMBER"><a class="selflink" href="#OPTIONS_CONTROLLING_EFFECTIVE_SEQUENCE_NUMBER">OPTIONS
  CONTROLLING EFFECTIVE SEQUENCE NUMBER</a></h1>
After relative weights are determined, they are normalized to sum to a total
  effective sequence number, <i>eff_nseq</i>. This number may be the actual
  number of sequences in the alignment, but it is almost always smaller than
  that. The default entropy weighting method <i>(--eent)</i> reduces the
  effective sequence number to reduce the information content (relative entropy,
  or average expected score on true homologs) per consensus position. The target
  relative entropy is controlled by a two-parameter function, where the two
  parameters are settable with <i>--ere</i> and <i>--esigma</i>.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eent</b></dt>
  <dd class="It-tag">Adjust effective sequence number to achieve a specific
      relative entropy per position (see <i>--ere</i>). This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eclust</b></dt>
  <dd class="It-tag">Set effective sequence number to the number of
      single-linkage clusters at a specific identity threshold (see
      <i>--eid</i>). This option is not recommended; it's for experiments
      evaluating how much better <b>--eent</b> is.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--enone</b></dt>
  <dd class="It-tag">Turn off effective sequence number determination and just
      use the actual number of sequences. One reason you might want to do this
      is to try to maximize the relative entropy/position of your model, which
      may be useful for short models.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eset</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Explicitly set the effective sequence number for all models
      to <i>&lt;x&gt;</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ere</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the minimum relative entropy/position target to
      <i>&lt;x&gt;</i>. Requires <b>--eent</b>. Default depends on the sequence
      alphabet. For protein sequences, it is 0.59 bits/position; for nucleotide
      sequences, it is 0.45 bits/position.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--esigma</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the minimum relative entropy contributed by an entire
      model alignment, over its whole length. This has the effect of making
      short models have higher relative entropy per position than <i>--ere</i>
      alone would give. The default is 45.0 bits.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eid</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the fractional pairwise identity cutoff used by single
      linkage clustering with the <b>--eclust</b> option. The default is 0.62.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_PRIORS"><a class="selflink" href="#OPTIONS_CONTROLLING_PRIORS">OPTIONS
  CONTROLLING PRIORS</a></h1>
By default, weighted counts are converted to mean posterior probability
  parameter estimates using mixture Dirichlet priors. Default mixture Dirichlet
  prior parameters for protein models and for nucleic acid (RNA and DNA) models
  are built in. The following options allow you to override the default priors.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pnone</b></dt>
  <dd class="It-tag">Don't use any priors. Probability parameters will simply be
      the observed frequencies, after relative sequence weighting.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--plaplace</b></dt>
  <dd class="It-tag">Use a Laplace +1 prior in place of the default mixture
      Dirichlet prior.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_E-VALUE_CALIBRATION"><a class="selflink" href="#OPTIONS_CONTROLLING_E-VALUE_CALIBRATION">OPTIONS
  CONTROLLING E-VALUE CALIBRATION</a></h1>
The location parameters for the expected score distributions for MSV filter
  scores, Viterbi filter scores, and Forward scores require three short random
  sequence simulations.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EmL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter mu for MSV filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EmN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter mu for MSV filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EvL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter mu for Viterbi filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EvN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter mu for Viterbi filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EfL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter tau for Forward E-values. Default is 100.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EfN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter tau for Forward E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--Eft</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the tail mass fraction to fit in the simulation that
      estimates the location parameter tau for Forward evalues. Default is 0.04.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_OPTIONS"><a class="selflink" href="#OTHER_OPTIONS">OTHER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cpu</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the number of parallel worker threads to
      <i>&lt;n&gt;</i>. By default, HMMER sets this to the number of CPU cores
      it detects in your machine - that is, it tries to maximize the use of your
      available processor cores. Setting <i>&lt;n&gt;</i> higher than the number
      of available cores is of little if any value, but you may want to set it
      to something less. You can also control this number by setting an
      environment variable, <i>HMMER_NCPU</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    This option is only available if HMMER was compiled with POSIX threads
      support. This is the default, but it may have been turned off for your
      site or machine for some reason.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Declare that the input <i>msafile</i> is in format
      <i>&lt;s&gt;</i>. Currently the accepted multiple alignment sequence file
      formats include Stockholm, Aligned FASTA, Clustal, NCBI PSI-BLAST, PHYLIP,
      Selex, and UCSC SAM A2M. Default is to autodetect the format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Seed the random number generator with <i>&lt;n&gt;</i>, an
      integer &gt;= 0. If <i>&lt;n&gt;</i> is nonzero, any stochastic
      simulations will be reproducible; the same command will give the same
      results. If <i>&lt;n&gt;</i> is 0, the random number generator is seeded
      arbitrarily, and stochastic simulations will vary from run to run of the
      same command. The default seed is 42.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--w_beta</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Window length tail mass. The upper bound, <i>W</i>, on the
      length at which nhmmer expects to find an instance of the model is set
      such that the fraction of all sequences generated by the model with length
      <i>&gt;= W</i> is less than <i>&lt;x&gt;</i>. The default is 1e-7.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--w_length</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Override the model instance length upper bound, <i>W</i>,
      which is otherwise controlled by <b>--w_beta</b>. It should be larger than
      the model length. The value of <i>W</i> is used deep in the acceleration
      pipeline, and modest changes are not expected to impact results (though
      larger values of <i>W</i> do lead to longer run time).
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mpi</b></dt>
  <dd class="It-tag">Run as a parallel MPI program. Each alignment is assigned
      to a MPI worker node for construction. (Therefore, the maximum
      parallelization cannot exceed the number of alignments in the input
      <i>msafile</i>.) This is useful when building large profile libraries.
      This option is only available if optional MPI capability was enabled at
      compile-time.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stall</b></dt>
  <dd class="It-tag">For debugging MPI parallelization: arrest program execution
      immediately after start, and wait for a debugger to attach to the running
      process and release the arrest.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxinsertlen</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Restrict insert length parameterization such that the
      expected insert length at each position of the model is no more than
      <i>&lt;n&gt;</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>hmmer(1)</b> for a master man page with a list of all the individual man
  pages for programs in the HMMER package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your HMMER
  distribution (Userguide.pdf); or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2015 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2015</td>
    <td class="foot-os">HMMER 3.1b2</td>
  </tr>
</table>
</body>
</html>
