<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>HMMER(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">HMMER(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">HMMER(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
HMMER - profile HMMs for biological sequence analysis
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmmalign</b>
<br/>
 Align sequences to a profile
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmbuild</b>
<br/>
 Construct profile(s) from multiple sequence alignment(s)
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmconvert</b>
<br/>
 Convert profile file to various HMMER and non-HMMER formats
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmemit</b>
<br/>
 Sample sequences from a profile
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmfetch</b>
<br/>
 Retrieve profile HMM(s) from a file
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmpgmd</b>
<br/>
 Daemon used for hmmer.org web server, for phmmer-, hmmsearch-, and hmmscan-like
  search
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmpress</b>
<br/>
 Prepare an HMM database for hmmscan
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmscan</b>
<br/>
 Search protein sequence(s) against a protein profile database
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmsearch</b>
<br/>
 Search protein profile(s) against a protein sequence database
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmsim</b>
<br/>
 Collect profile score distributions on random sequences
<div style="height: 1.00em;">&#x00A0;</div>
<b>hmmstat</b>
<br/>
 Summary statistics for a profile file
<div style="height: 1.00em;">&#x00A0;</div>
<b>jackhmmer</b>
<br/>
 Iterative search of a protein sequence against a protein sequence database
<div style="height: 1.00em;">&#x00A0;</div>
<b>nhmmer</b>
<br/>
 Search nucleotide sequence(s), alignment(s), or profile(s) against a nucleotide
  sequence database
<div style="height: 1.00em;">&#x00A0;</div>
<b>nhmmscan</b>
<br/>
 Search nucleotide sequence(s) against a nucleotide profile database
<div style="height: 1.00em;">&#x00A0;</div>
<b>phmmer</b>
<br/>
 Search a protein sequence(s) against a protein sequence database
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
HMMER is a suite of several programs for biological sequence alignment and
  database homology search. It uses probabilistic models called &quot;profile
  hidden Markov models&quot; (profile HMMs) to represent the likely evolutionary
  homologs of a single sequence or a multiple alignment of a sequence family. A
  main avenue of research is to improve the evolutionary predictive models in
  HMMER to be able to recognize and accurately align increasingly remote
  homologs, distant in time.
<div style="height: 1.00em;">&#x00A0;</div>
HMMER is also used as an organizational tool, to group the exponentially growing
  number of biological sequences into a vastly smaller set of well-annotated
  sequence families. New sequences can be annotated by comparison against
  curated sequence family databases of prebuilt HMMER profiles, in addition or
  instead of comparison to the entire sequence database. Databases such as Pfam,
  SMART, and TIGRfams, among others, are based on this principle.
<div style="height: 1.00em;">&#x00A0;</div>
HMMER is used in three main modes: to search a sequence database for new
  homologs of a sequence or a sequence family; to search a profile database
  (like Pfam) to find what known family a query sequence belongs to, or what
  domains it has; and to automatically construct large multiple alignments (i.e.
  with an effectively unlimited number of sequences) using a profile
  representative of a sequence family.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
Suppose you have a multiple sequence alignment of a sequence family of interest,
  and you want to search a sequence database for additional homologs. The
  <b>hmmbuild</b> program builds profile(s) from multiple alignment(s). The
  <b>hmmsearch</b> program searches protein profile(s) against a protein
  sequence database, and <b>nhmmer</b> searches nucleotide profile(s) against a
  nucleotide sequence database.
<div style="height: 1.00em;">&#x00A0;</div>
Suppose you have a single sequence of interest, and you want to search a
  sequence database for additional homologs. The <b>phmmer</b> program searches
  a single protein sequence against a protein sequence database. The
  <b>jackhmmer</b> program does the same thing but iteratively -- homologs
  detected in a previous round are incorporated into a new profile, and the new
  profile is searched again. <b>phmmer</b> is used like BLASTP, and
  <b>jackhmmer</b> is used like a protein PSI-BLAST. The <b>nhmmer</b> program
  searches a single nucleotide sequence against a nucleotide sequence.
<div style="height: 1.00em;">&#x00A0;</div>
Suppose you have sequence(s) that you want to analyze using a HMMER-based
  profile HMM database like Pfam (http://pfam.sanger.ac.uk). The <b>hmmpress</b>
  program formats a profile HMM flatfile (such as the file you would download
  from Pfam) into a HMMER binary database. The <b>hmmscan</b> program searches
  protein sequence(s) against that database. The <b>nhmmscan</b> program can
  similarly search nucleotide sequence(s) against a pressed database of
  nucleotide profiles, such as from Dfam (http://dfam.janelia.org).
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
Suppose you want to align lots of sequences. You can construct a manageably
  small alignment of a representative set of sequences, build a profile with
  <b>hmmbuild</b>, and use the <b>hmmalign</b> program to align any number of
  sequences to that profile.
<div style="height: 1.00em;">&#x00A0;</div>
HMMER also includes some auxiliary tools for working with large profile
  databases. <b>hmmfetch</b> fetches one or more profiles from a database.
  <b>hmmstat</b> prints summary statistics about a profile file.
<div style="height: 1.00em;">&#x00A0;</div>
For compatibility with other profile software and previous versions of HMMER,
  the <b>hmmconvert</b> program converts profiles to a few other formats. We
  intend to add more support for other formats over time.
<div style="height: 1.00em;">&#x00A0;</div>
The <b>hmmemit</b> program generates (simulates) &quot;homologous&quot;
  sequences by sampling from a profile. It can also generate a
  &quot;consensus&quot; sequence.
<div style="height: 1.00em;">&#x00A0;</div>
The <b>hmmsim</b> program is a simulator used for collecting statistics about
  score distributions on random sequences.
<div style="height: 1.00em;">&#x00A0;</div>
Each program has its own man page.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
This is a summary man page for the entire HMMER3 package. See individual man
  pages <b>[hmmbuild(1)</b>, for example] for usage, options, and description of
  each program in the package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your HMMER
  distribution (Userguide.pdf); or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2015 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2015</td>
    <td class="foot-os">HMMER 3.1b2</td>
  </tr>
</table>
</body>
</html>
