<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>HMMER(1) HMMER Manual HMMER(1)</p>

<p style="margin-top: 1em">NAME <br>
HMMER - profile HMMs for biological sequence analysis</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmmalign <br>
Align sequences to a profile</p>

<p style="margin-top: 1em">hmmbuild <br>
Construct profile(s) from multiple sequence alignment(s)</p>

<p style="margin-top: 1em">hmmconvert <br>
Convert profile file to various HMMER and non-HMMER
formats</p>

<p style="margin-top: 1em">hmmemit <br>
Sample sequences from a profile</p>

<p style="margin-top: 1em">hmmfetch <br>
Retrieve profile HMM(s) from a file</p>

<p style="margin-top: 1em">hmmpgmd <br>
Daemon used for hmmer.org web server, for phmmer-,
hmmsearch-, and hmmscan-like search</p>

<p style="margin-top: 1em">hmmpress <br>
Prepare an HMM database for hmmscan</p>

<p style="margin-top: 1em">hmmscan <br>
Search protein sequence(s) against a protein profile
database</p>

<p style="margin-top: 1em">hmmsearch <br>
Search protein profile(s) against a protein sequence
database</p>

<p style="margin-top: 1em">hmmsim <br>
Collect profile score distributions on random sequences</p>

<p style="margin-top: 1em">hmmstat <br>
Summary statistics for a profile file</p>

<p style="margin-top: 1em">jackhmmer <br>
Iterative search of a protein sequence against a protein
sequence database</p>

<p style="margin-top: 1em">nhmmer <br>
Search nucleotide sequence(s), alignment(s), or profile(s)
against a nucleotide sequence database</p>

<p style="margin-top: 1em">nhmmscan <br>
Search nucleotide sequence(s) against a nucleotide profile
database</p>

<p style="margin-top: 1em">phmmer <br>
Search a protein sequence(s) against a protein sequence
database</p>

<p style="margin-top: 1em">DESCRIPTION <br>
HMMER is a suite of several programs for biological sequence
alignment and database homology search. It uses
probabilistic models called &quot;profile hidden Markov
models&quot; (profile <br>
HMMs) to represent the likely evolutionary homologs of a
single sequence or a multiple alignment of a sequence
family. A main avenue of research is to improve the
evolutionary <br>
predictive models in HMMER to be able to recognize and
accurately align increasingly remote homologs, distant in
time.</p>

<p style="margin-top: 1em">HMMER is also used as an
organizational tool, to group the exponentially growing
number of biological sequences into a vastly smaller set of
well-annotated sequence families. New <br>
sequences can be annotated by comparison against curated
sequence family databases of prebuilt HMMER profiles, in
addition or instead of comparison to the entire sequence
data&acirc; <br>
base. Databases such as Pfam, SMART, and TIGRfams, among
others, are based on this principle.</p>

<p style="margin-top: 1em">HMMER is used in three main
modes: to search a sequence database for new homologs of a
sequence or a sequence family; to search a profile database
(like Pfam) to find what known <br>
family a query sequence belongs to, or what domains it has;
and to automatically construct large multiple alignments
(i.e. with an effectively unlimited number of sequences)
<br>
using a profile representative of a sequence family.</p>

<p style="margin-top: 1em">Suppose you have a multiple
sequence alignment of a sequence family of interest, and you
want to search a sequence database for additional homologs.
The hmmbuild program builds <br>
profile(s) from multiple alignment(s). The hmmsearch program
searches protein profile(s) against a protein sequence
database, and nhmmer searches nucleotide profile(s) against
a <br>
nucleotide sequence database.</p>

<p style="margin-top: 1em">Suppose you have a single
sequence of interest, and you want to search a sequence
database for additional homologs. The phmmer program
searches a single protein sequence against <br>
a protein sequence database. The jackhmmer program does the
same thing but iteratively -- homologs detected in a
previous round are incorporated into a new profile, and the
new <br>
profile is searched again. phmmer is used like BLASTP, and
jackhmmer is used like a protein PSI-BLAST. The nhmmer
program searches a single nucleotide sequence against a
nucleo&acirc; <br>
tide sequence.</p>

<p style="margin-top: 1em">Suppose you have sequence(s)
that you want to analyze using a HMMER-based profile HMM
database like Pfam (http://pfam.sanger.ac.uk). The hmmpress
program formats a profile HMM <br>
flatfile (such as the file you would download from Pfam)
into a HMMER binary database. The hmmscan program searches
protein sequence(s) against that database. The nhmmscan
pro&acirc; <br>
gram can similarly search nucleotide sequence(s) against a
pressed database of nucleotide profiles, such as from Dfam
(http://dfam.janelia.org).</p>

<p style="margin-top: 1em">Suppose you want to align lots
of sequences. You can construct a manageably small alignment
of a representative set of sequences, build a profile with
hmmbuild, and use the hmma&acirc; <br>
lign program to align any number of sequences to that
profile.</p>

<p style="margin-top: 1em">HMMER also includes some
auxiliary tools for working with large profile databases.
hmmfetch fetches one or more profiles from a database.
hmmstat prints summary statistics <br>
about a profile file.</p>

<p style="margin-top: 1em">For compatibility with other
profile software and previous versions of HMMER, the
hmmconvert program converts profiles to a few other formats.
We intend to add more support for <br>
other formats over time.</p>

<p style="margin-top: 1em">The hmmemit program generates
(simulates) &quot;homologous&quot; sequences by sampling
from a profile. It can also generate a &quot;consensus&quot;
sequence.</p>

<p style="margin-top: 1em">The hmmsim program is a
simulator used for collecting statistics about score
distributions on random sequences.</p>

<p style="margin-top: 1em">Each program has its own man
page.</p>

<p style="margin-top: 1em">SEE ALSO <br>
This is a summary man page for the entire HMMER3 package.
See individual man pages [hmmbuild(1), for example] for
usage, options, and description of each program in the
package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
HMMER(1)</p>
<hr>
</body>
</html>
