<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmmer2(1) HMMER Manual hmmer2(1)</p>

<p style="margin-top: 1em">NAME <br>
HMMER - profile hidden Markov model software</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmm2align <br>
Align multiple sequences to a profile HMM.</p>

<p style="margin-top: 1em">hmm2build <br>
Build a profile HMM from a given multiple sequence
alignment.</p>

<p style="margin-top: 1em">hmm2calibrate <br>
Determine appropriate statistical significance parameters
for a profile HMM prior to doing database searches.</p>

<p style="margin-top: 1em">hmm2convert <br>
Convert HMMER profile HMMs to other formats, such as GCG
profiles.</p>

<p style="margin-top: 1em">hmm2emit <br>
Generate sequences probabilistically from a profile HMM.</p>

<p style="margin-top: 1em">hmm2fetch <br>
Retrieve an HMM from an HMM database</p>

<p style="margin-top: 1em">hmm2index <br>
Create a binary SSI index for an HMM database</p>

<p style="margin-top: 1em">hmm2pfam <br>
Search a profile HMM database with a sequence (i.e.,
annotate various kinds of domains in the query
sequence).</p>

<p style="margin-top: 1em">hmm2search <br>
Search a sequence database with a profile HMM (i.e., find
additional homologues of a modeled family).</p>

<p style="margin-top: 1em">DESCRIPTION <br>
These programs use profile hidden Markov models (profile
HMMs) to model the primary structure consensus of a family
of protein or nucleic acid sequences.</p>

<p style="margin-top: 1em">OPTIONS <br>
All HMMER programs give a brief summary of their
command-line syntax and options if invoked without any
arguments. When invoked with the single argument, -h (i.e.,
help), a pro&acirc; <br>
gram will report more verbose command-line usage
information, including rarely used, experimental, and expert
options. -h will report version numbers which are useful if
you <br>
need to report a bug or problem to me.</p>

<p style="margin-top: 1em">Each HMMER program has its own
man page briefly summarizing command line usage. There is
also a user&rsquo;s guide that came with the software
distribution, which includes a tutorial <br>
introduction and more detailed descriptions of the
programs.</p>

<p style="margin-top: 1em">See http://hmmer.janelia.org/
for on-line documentation and the current HMMER release.</p>

<p style="margin-top: 1em">In general, no command line
options should be needed by beginning users. The defaults
are set up for optimum performance in most situations.
Options that are single lowercase <br>
letters (e.g. -a ) are &quot;common&quot; options that are
expected to be frequently used and will be important in many
applications. Options that are single uppercase letters
(e.g. -B <br>
) are usually less common options, but also may be important
in some applications. Options that are full words (e.g.
--verbose ) are either rarely used, experimental, or expert
<br>
options. Some experimental options are only there for my own
ongoing experiments with HMMER, and may not be supported or
documented adequately.</p>

<p style="margin-top: 1em">SEQUENCE FILE FORMATS <br>
In general, HMMER attempts to read most common biological
sequence file formats. It autodetects the format of the
file. It also autodetects whether the sequences are protein
or <br>
nucleic acid. Standard IUPAC degeneracy codes are allowed in
addition to the usual 4-letter or 20-letter codes.</p>

<p style="margin-top: 1em">Unaligned sequences <br>
Unaligned sequence files may be in FASTA, Swissprot, EMBL,
GenBank, PIR, Intelligenetics, Strider, or GCG format. These
formats are documented in the User&rsquo;s Guide.</p>

<p style="margin-top: 1em">Sequence alignments <br>
Multiple sequence alignments may be in CLUSTALW, SELEX, or
GCG MSF format. These formats are documented in the
User&rsquo;s Guide.</p>

<p style="margin-top: 1em">ENVIRONMENT VARIABLES <br>
For ease of using large stable sequence and HMM databases,
HMMER looks for sequence files and HMM files in the current
working directory as well as in system directories
speci&acirc; <br>
fied by environment variables.</p>

<p style="margin-top: 1em">BLASTDB <br>
Specifies the directory location of sequence databases.
Example: /seqlibs/blast-db/. In installations that use BLAST
software, this environment variable is likely to <br>
already be set.</p>

<p style="margin-top: 1em">HMMERDB <br>
Specifies the directory location of HMM databases. Example:
/seqlibs/pfam/.</p>

<p style="margin-top: 1em">HMMER 2.3.2 Oct 2003
hmmer2(1)</p>
<hr>
</body>
</html>
