<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>hmmer2(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">hmmer2(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">hmmer2(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">HMMER - profile hidden Markov model software</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2align</b></dt>
  <dd class="It-tag">Align multiple sequences to a profile HMM.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2build</b></dt>
  <dd class="It-tag">Build a profile HMM from a given multiple sequence
      alignment.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2calibrate</b></dt>
  <dd class="It-tag">Determine appropriate statistical significance parameters
      for a profile HMM prior to doing database searches.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2convert</b></dt>
  <dd class="It-tag">Convert HMMER profile HMMs to other formats, such as GCG
      profiles.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2emit</b></dt>
  <dd class="It-tag">Generate sequences probabilistically from a profile HMM.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2fetch</b></dt>
  <dd class="It-tag">Retrieve an HMM from an HMM database
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2index</b></dt>
  <dd class="It-tag">Create a binary SSI index for an HMM database
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2pfam</b></dt>
  <dd class="It-tag">Search a profile HMM database with a sequence (i.e.,
      annotate various kinds of domains in the query sequence).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>hmm2search</b></dt>
  <dd class="It-tag">Search a sequence database with a profile HMM (i.e., find
      additional homologues of a modeled family).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
These programs use profile hidden Markov models (profile HMMs) to model the
  primary structure consensus of a family of protein or nucleic acid sequences.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
All <b>HMMER</b> programs give a brief summary of their command-line syntax and
  options if invoked without any arguments. When invoked with the single
  argument, <b>-h</b> (i.e., help), a program will report more verbose
  command-line usage information, including rarely used, experimental, and
  expert options. <b>-h</b> will report version numbers which are useful if you
  need to report a bug or problem to me.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Each <b>HMMER</b> program has its own man page briefly summarizing command line
  usage. There is also a user's guide that came with the software distribution,
  which includes a tutorial introduction and more detailed descriptions of the
  programs.
<div style="height: 1.00em;">&#x00A0;</div>
See http://hmmer.janelia.org/ for on-line documentation and the current HMMER
  release.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
In general, no command line options should be needed by beginning users. The
  defaults are set up for optimum performance in most situations. Options that
  are single lowercase letters (e.g. <b>-a</b> ) are &quot;common&quot; options
  that are expected to be frequently used and will be important in many
  applications. Options that are single uppercase letters (e.g. <b>-B</b> ) are
  usually less common options, but also may be important in some applications.
  Options that are full words (e.g. <b>--verbose</b> ) are either rarely used,
  experimental, or expert options. Some experimental options are only there for
  my own ongoing experiments with HMMER, and may not be supported or documented
  adequately.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEQUENCE_FILE_FORMATS"><a class="selflink" href="#SEQUENCE_FILE_FORMATS">SEQUENCE
  FILE FORMATS</a></h1>
In general, <b>HMMER</b> attempts to read most common biological sequence file
  formats. It autodetects the format of the file. It also autodetects whether
  the sequences are protein or nucleic acid. Standard IUPAC degeneracy codes are
  allowed in addition to the usual 4-letter or 20-letter codes.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Unaligned sequences</b></dt>
  <dd class="It-tag">Unaligned sequence files may be in FASTA, Swissprot, EMBL,
      GenBank, PIR, Intelligenetics, Strider, or GCG format. These formats are
      documented in the User's Guide.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Sequence alignments</b></dt>
  <dd class="It-tag">Multiple sequence alignments may be in CLUSTALW, SELEX, or
      GCG MSF format. These formats are documented in the User's Guide.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="ENVIRONMENT_VARIABLES"><a class="selflink" href="#ENVIRONMENT_VARIABLES">ENVIRONMENT
  VARIABLES</a></h1>
For ease of using large stable sequence and HMM databases, <b>HMMER</b> looks
  for sequence files and HMM files in the current working directory as well as
  in system directories specified by environment variables.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>BLASTDB</b></dt>
  <dd class="It-tag">Specifies the directory location of sequence databases.
      Example: <b>/seqlibs/blast-db/.</b> In installations that use BLAST
      software, this environment variable is likely to already be set.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>HMMERDB</b></dt>
  <dd class="It-tag">Specifies the directory location of HMM databases. Example:
      <b>/seqlibs/pfam/.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">Oct 2003</td>
    <td class="foot-os">HMMER 2.3.2</td>
  </tr>
</table>
</body>
</html>
