<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:55 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmmscan(1) HMMER Manual hmmscan(1)</p>

<p style="margin-top: 1em">NAME <br>
hmmscan - search protein sequence(s) against a protein
profile database</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmmscan [options] &lt;hmmdb&gt; &lt;seqfile&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
hmmscan is used to search protein sequences against
collections of protein profiles. For each sequence in
&lt;seqfile&gt;, use that query sequence to search the
target database of pro&acirc; <br>
files in &lt;hmmdb&gt;, and output ranked lists of the
profiles with the most significant matches to the
sequence.</p>

<p style="margin-top: 1em">The &lt;seqfile&gt; may contain
more than one query sequence. It can be in FASTA format, or
several other common sequence file formats (genbank, embl,
and uniprot, among others), or in <br>
alignment file formats (stockholm, aligned fasta, and
others). See the --qformat option for a complete list.</p>

<p style="margin-top: 1em">The &lt;hmmdb&gt; needs to be
press&rsquo;ed using hmmpress before it can be searched with
hmmscan. This creates four binary files, suffixed
.h3{fimp}.</p>

<p style="margin-top: 1em">The query &lt;seqfile&gt; may be
&rsquo;-&rsquo; (a dash character), in which case the query
sequences are read from a &lt;stdin&gt; pipe instead of from
a file. The &lt;hmmdb&gt; cannot be read from a <br>
&lt;stdin&gt; stream, because it needs to have those four
auxiliary binary files generated by hmmpress.</p>

<p style="margin-top: 1em">The output format is designed to
be human-readable, but is often so voluminous that reading
it is impractical, and parsing it is a pain. The --tblout
and --domtblout options save <br>
output in simple tabular formats that are concise and easier
to parse. The -o option allows redirecting the main output,
including throwing it away in /dev/null.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
all available options.</p>

<p style="margin-top: 1em">OPTIONS FOR CONTROLLING OUTPUT
<br>
-o &lt;f&gt; Direct the main human-readable output to a file
&lt;f&gt; instead of the default stdout.</p>

<p style="margin-top: 1em">--tblout &lt;f&gt; <br>
Save a simple tabular (space-delimited) file summarizing the
per-target output, with one data line per homologous target
model found.</p>

<p style="margin-top: 1em">--domtblout &lt;f&gt; <br>
Save a simple tabular (space-delimited) file summarizing the
per-domain output, with one data line per homologous domain
detected in a query sequence for each homologous <br>
model.</p>

<p style="margin-top: 1em">--pfamtblout &lt;f&gt; <br>
Save an especially succinct tabular (space-delimited) file
summarizing the per-target output, with one data line per
homologous target model found.</p>

<p style="margin-top: 1em">--acc Use accessions instead of
names in the main output, where available for profiles
and/or sequences.</p>

<p style="margin-top: 1em">--noali <br>
Omit the alignment section from the main output. This can
greatly reduce the output volume.</p>

<p style="margin-top: 1em">--notextw <br>
Unlimit the length of each line in the main output. The
default is a limit of 120 characters per line, which helps
in displaying the output cleanly on terminals and in <br>
editors, but can truncate target profile description
lines.</p>

<p style="margin-top: 1em">--textw &lt;n&gt; <br>
Set the main output&rsquo;s line length limit to &lt;n&gt;
characters per line. The default is 120.</p>

<p style="margin-top: 1em">OPTIONS FOR REPORTING THRESHOLDS
<br>
Reporting thresholds control which hits are reported in
output files (the main output, --tblout, and
--domtblout).</p>

<p style="margin-top: 1em">-E &lt;x&gt; In the per-target
output, report target profiles with an E-value of &lt;=
&lt;x&gt;. The default is 10.0, meaning that on average,
about 10 false positives will be reported per <br>
query, so you can see the top of the noise and decide for
yourself if it&rsquo;s really noise.</p>

<p style="margin-top: 1em">-T &lt;x&gt; Instead of
thresholding per-profile output on E-value, instead report
target profiles with a bit score of &gt;= &lt;x&gt;.</p>

<p style="margin-top: 1em">--domE &lt;x&gt; <br>
In the per-domain output, for target profiles that have
already satisfied the per-profile reporting threshold,
report individual domains with a conditional E-value of
&lt;= <br>
&lt;x&gt;. The default is 10.0. A conditional E-value means
the expected number of additional false positive domains in
the smaller search space of those comparisons that <br>
already satisfied the per-profile reporting threshold (and
thus must have at least one homologous domain already).</p>

<p style="margin-top: 1em">--domT &lt;x&gt; <br>
Instead of thresholding per-domain output on E-value,
instead report domains with a bit score of &gt;=
&lt;x&gt;.</p>

<p style="margin-top: 1em">OPTIONS FOR INCLUSION THRESHOLDS
<br>
Inclusion thresholds are stricter than reporting thresholds.
Inclusion thresholds control which hits are considered to be
reliable enough to be included in an output alignment <br>
or a subsequent search round. In hmmscan, which does not
have any alignment output (like hmmsearch or phmmer) nor any
iterative search steps (like jackhmmer), inclusion
thresh&acirc; <br>
olds have little effect. They only affect what domains get
marked as significant (!) or questionable (?) in domain
output.</p>

<p style="margin-top: 1em">--incE &lt;x&gt; <br>
Use an E-value of &lt;= &lt;x&gt; as the per-target
inclusion threshold. The default is 0.01, meaning that on
average, about 1 false positive would be expected in every
100 <br>
searches with different query sequences.</p>

<p style="margin-top: 1em">--incT &lt;x&gt; <br>
Instead of using E-values for setting the inclusion
threshold, instead use a bit score of &gt;= &lt;x&gt; as the
per-target inclusion threshold. It would be unusual to use
bit <br>
score thresholds with hmmscan, because you don&rsquo;t
expect a single score threshold to work for different
profiles; different profiles have slightly different
expected score <br>
distributions.</p>

<p style="margin-top: 1em">--incdomE &lt;x&gt; <br>
Use a conditional E-value of &lt;= &lt;x&gt; as the
per-domain inclusion threshold, in targets that have already
satisfied the overall per-target inclusion threshold. The
default <br>
is 0.01.</p>

<p style="margin-top: 1em">--incdomT &lt;x&gt; <br>
Instead of using E-values, instead use a bit score of &gt;=
&lt;x&gt; as the per-domain inclusion threshold. As with
--incT above, it would be unusual to use a single bit score
<br>
threshold in hmmscan.</p>

<p style="margin-top: 1em">OPTIONS FOR MODEL-SPECIFIC SCORE
THRESHOLDING <br>
Curated profile databases may define specific bit score
thresholds for each profile, superseding any thresholding
based on statistical significance alone.</p>

<p style="margin-top: 1em">To use these options, the
profile must contain the appropriate (GA, TC, and/or NC)
optional score threshold annotation; this is picked up by
hmmbuild from Stockholm format align&acirc; <br>
ment files. Each thresholding option has two scores: the
per-sequence threshold &lt;x1&gt; and the per-domain
threshold &lt;x2&gt; These act as if -T&lt;x1&gt;
--incT&lt;x1&gt; --domT&lt;x2&gt; --incdomT&lt;x2&gt; <br>
has been applied specifically using each model&rsquo;s
curated thresholds.</p>

<p style="margin-top: 1em">--cut_ga <br>
Use the GA (gathering) bit scores in the model to set
per-sequence (GA1) and per-domain (GA2) reporting and
inclusion thresholds. GA thresholds are generally considered
to <br>
be the reliable curated thresholds defining family
membership; for example, in Pfam, these thresholds define
what gets included in Pfam Full alignments based on searches
<br>
with Pfam Seed models.</p>

<p style="margin-top: 1em">--cut_nc <br>
Use the NC (noise cutoff) bit score thresholds in the model
to set per-sequence (NC1) and per-domain (NC2) reporting and
inclusion thresholds. NC thresholds are generally <br>
considered to be the score of the highest-scoring known
false positive.</p>

<p style="margin-top: 1em">--cut_tc <br>
Use the NC (trusted cutoff) bit score thresholds in the
model to set per-sequence (TC1) and per-domain (TC2)
reporting and inclusion thresholds. TC thresholds are
gener&acirc; <br>
ally considered to be the score of the lowest-scoring known
true positive that is above all known false positives.</p>

<p style="margin-top: 1em">CONTROL OF THE ACCELERATION
PIPELINE <br>
HMMER3 searches are accelerated in a three-step filter
pipeline: the MSV filter, the Viterbi filter, and the
Forward filter. The first filter is the fastest and most
approximate; <br>
the last is the full Forward scoring algorithm. There is
also a bias filter step between MSV and Viterbi. Targets
that pass all the steps in the acceleration pipeline are
then <br>
subjected to postprocessing -- domain identification and
scoring using the Forward/Backward algorithm.</p>

<p style="margin-top: 1em">Changing filter thresholds only
removes or includes targets from consideration; changing
filter thresholds does not alter bit scores, E-values, or
alignments, all of which are <br>
determined solely in postprocessing.</p>

<p style="margin-top: 1em">--max Turn off all filters,
including the bias filter, and run full Forward/Backward
postprocessing on every target. This increases sensitivity
somewhat, at a large cost in <br>
speed.</p>

<p style="margin-top: 1em">--F1 &lt;x&gt; <br>
Set the P-value threshold for the MSV filter step. The
default is 0.02, meaning that roughly 2% of the highest
scoring nonhomologous targets are expected to pass the
fil&acirc; <br>
ter.</p>

<p style="margin-top: 1em">--F2 &lt;x&gt; <br>
Set the P-value threshold for the Viterbi filter step. The
default is 0.001.</p>

<p style="margin-top: 1em">--F3 &lt;x&gt; <br>
Set the P-value threshold for the Forward filter step. The
default is 1e-5.</p>

<p style="margin-top: 1em">--nobias <br>
Turn off the bias filter. This increases sensitivity
somewhat, but can come at a high cost in speed, especially
if the query has biased residue composition (such as a <br>
repetitive sequence region, or if it is a membrane protein
with large regions of hydrophobicity). Without the bias
filter, too many sequences may pass the filter with <br>
biased queries, leading to slower than expected performance
as the computationally intensive Forward/Backward algorithms
shoulder an abnormally heavy load.</p>

<p style="margin-top: 1em">OTHER OPTIONS <br>
--nonull2 <br>
Turn off the null2 score corrections for biased
composition.</p>

<p style="margin-top: 1em">-Z &lt;x&gt; Assert that the
total number of targets in your searches is &lt;x&gt;, for
the purposes of per-sequence E-value calculations, rather
than the actual number of targets seen.</p>

<p style="margin-top: 1em">--domZ &lt;x&gt; <br>
Assert that the total number of targets in your searches is
&lt;x&gt;, for the purposes of per-domain conditional
E-value calculations, rather than the number of targets that
<br>
passed the reporting thresholds.</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Set the random number seed to &lt;n&gt;. Some steps in
postprocessing require Monte Carlo simulation. The default
is to use a fixed seed (42), so that results are exactly
<br>
reproducible. Any other positive integer will give different
(but also reproducible) results. A choice of 0 uses an
arbitrarily chosen seed.</p>

<p style="margin-top: 1em">--qformat &lt;s&gt; <br>
Assert that the query sequence file is in format &lt;s&gt;.
Accepted formats include fasta, embl, genbank, ddbj,
uniprot, stockholm, pfam, a2m, and afa.</p>

<p style="margin-top: 1em">--cpu &lt;n&gt; <br>
Set the number of parallel worker threads to &lt;n&gt;. By
default, HMMER sets this to the number of CPU cores it
detects in your machine - that is, it tries to maximize the
<br>
use of your available processor cores. Setting &lt;n&gt;
higher than the number of available cores is of little if
any value, but you may want to set it to something less. You
<br>
can also control this number by setting an environment
variable, HMMER_NCPU.</p>

<p style="margin-top: 1em">This option is only available if
HMMER was compiled with POSIX threads support. This is the
default, but it may have been turned off for your site or
machine for some rea&acirc; <br>
son.</p>

<p style="margin-top: 1em">--stall <br>
For debugging the MPI master/worker version: pause after
start, to enable the developer to attach debuggers to the
running master and worker(s) processes. Send SIGCONT <br>
signal to release the pause. (Under gdb: (gdb) signal
SIGCONT)</p>

<p style="margin-top: 1em">(Only available if optional MPI
support was enabled at compile-time.)</p>

<p style="margin-top: 1em">--mpi Run in MPI master/worker
mode, using mpirun.</p>

<p style="margin-top: 1em">(Only available if optional MPI
support was enabled at compile-time.)</p>

<p style="margin-top: 1em">SEE ALSO <br>
See hmmer(1) for a master man page with a list of all the
individual man pages for programs in the HMMER package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
hmmscan(1)</p>
<hr>
</body>
</html>
