<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:55 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>hmmsim(1) HMMER Manual hmmsim(1)</p>

<p style="margin-top: 1em">NAME <br>
hmmsim - collect score distributions on random sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
hmmsim [options] &lt;hmmfile&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The hmmsim program generates random sequences, scores them
with the model(s) in &lt;hmmfile&gt;, and outputs various
sorts of histograms, plots, and fitted distributions for the
<br>
resulting scores.</p>

<p style="margin-top: 1em">hmmsim is not a mainstream part
of the HMMER package. Most users would have no reason to use
it. It is used to develop and test the statistical methods
used to determine P-values <br>
and E-values in HMMER3. For example, it was used to generate
most of the results in a 2008 paper on H3&rsquo;s local
alignment statistics (PLoS Comp Bio 4:e1000069, 2008; <br>
http://www.ploscompbiol.org/doi/pcbi.1000069).</p>

<p style="margin-top: 1em">Because it is a research
testbed, you should not expect it to be as robust as other
programs in the package. For example, options may interact
in weird ways; we haven&rsquo;t tested <br>
nor tried to anticipate all different possible
combinations.</p>

<p style="margin-top: 1em">The main task is to fit a
maximum likelihood Gumbel distribution to Viterbi scores or
an maximum likelihood exponential tail to high-scoring
Forward scores, and to test that <br>
these fitted distributions obey the conjecture that lambda ~
log_2 for both the Viterbi Gumbel and the Forward
exponential tail.</p>

<p style="margin-top: 1em">The output is a table of
numbers, one row for each model. Four different parametric
fits to the score data are tested: (1) maximum likelihood
fits to both location (mu/tau) and <br>
slope (lambda) parameters; (2) assuming lambda=log_2,
maximum likelihood fit to the location parameter only; (3)
same but assuming an edge-corrected lambda, using current
proce&acirc; <br>
dures in H3 [Eddy, 2008]; and (4) using both parameters
determined by H3&rsquo;s current procedures. The standard
simple, quick and dirty statistic for goodness-of-fit is
&rsquo;E@10&rsquo;, the <br>
calculated E-value of the 10th ranked top hit, which we
expect to be about 10.</p>

<p style="margin-top: 1em">In detail, the columns of the
output are:</p>

<p style="margin-top: 1em">name Name of the model.</p>

<p style="margin-top: 1em">tailp Fraction of the highest
scores used to fit the distribution. For Viterbi, MSV, and
Hybrid scores, this defaults to 1.0 (a Gumbel distribution
is fitted to all the data). <br>
For Forward scores, this defaults to 0.02 (an exponential
tail is fitted to the highest 2% scores).</p>

<p style="margin-top: 1em">mu/tau Location parameter for
the maximum likelihood fit to the data.</p>

<p style="margin-top: 1em">lambda Slope parameter for the
maximum likelihood fit to the data.</p>

<p style="margin-top: 1em">E@10 The E-value calculated for
the 10th ranked high score (&rsquo;E@10&rsquo;) using the ML
mu/tau and lambda. By definition, this expected to be about
10, if E-value estimation were <br>
accurate.</p>

<p style="margin-top: 1em">mufix Location parameter, for a
maximum likelihood fit with a known (fixed) slope parameter
lambda of log_2 (0.693).</p>

<p style="margin-top: 1em">E@10fix <br>
The E-value calculated for the 10th ranked score using mufix
and the expected lambda = log_2 = 0.693.</p>

<p style="margin-top: 1em">mufix2 Location parameter, for a
maximum likelihood fit with an edge-effect-corrected
lambda.</p>

<p style="margin-top: 1em">E@10fix2 <br>
The E-value calculated for the 10th ranked score using
mufix2 and the edge-effect-corrected lambda.</p>

<p style="margin-top: 1em">pmu Location parameter as
determined by H3&rsquo;s estimation procedures.</p>

<p style="margin-top: 1em">plambda <br>
Slope parameter as determined by H3&rsquo;s estimation
procedures.</p>

<p style="margin-top: 1em">pE@10 The E-value calculated for
the 10th ranked score using pmu, plambda.</p>

<p style="margin-top: 1em">At the end of this table, one
more line is printed, starting with # and summarizing the
overall CPU time used by the simulations.</p>

<p style="margin-top: 1em">Some of the optional output
files are in xmgrace xy format. xmgrace is powerful and
freely available graph-plotting software.</p>

<p style="margin-top: 1em">MISCELLANEOUS OPTIONS <br>
-h Help; print a brief reminder of command line usage and
all available options.</p>

<p style="margin-top: 1em">-a Collect expected Viterbi
alignment length statistics from each simulated sequence.
This only works with Viterbi scores (the default; see
--vit). Two additional fields are <br>
printed in the output table for each model: the mean length
of Viterbi alignments, and the standard deviation.</p>

<p style="margin-top: 1em">-v (Verbose). Print the scores
too, one score per line.</p>

<p style="margin-top: 1em">-L &lt;n&gt; Set the length of
the randomly sampled (nonhomologous) sequences to &lt;n&gt;.
The default is 100.</p>

<p style="margin-top: 1em">-N &lt;n&gt; Set the number of
randomly sampled sequences to &lt;n&gt;. The default is
1000.</p>

<p style="margin-top: 1em">--mpi Run in MPI parallel mode,
under mpirun. It is parallelized at the level of sending one
profile at a time to an MPI worker process, so
parallelization only helps if you <br>
have more than one profile in the &lt;hmmfile&gt;, and you
want to have at least as many profiles as MPI worker
processes. (Only available if optional MPI support was
enabled <br>
at compile-time.)</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING OUTPUT <br>
-o &lt;f&gt; Save the main output table to a file &lt;f&gt;
rather than sending it to stdout.</p>

<p style="margin-top: 1em">--afile &lt;f&gt; <br>
When collecting Viterbi alignment statistics (the -a
option), for each sampled sequence, output two fields per
line to a file &lt;f&gt;: the length of the optimal
alignment, and <br>
the Viterbi bit score. Requires that the -a option is also
used.</p>

<p style="margin-top: 1em">--efile &lt;f&gt; <br>
Output a rank vs. E-value plot in XMGRACE xy format to file
&lt;f&gt;. The x-axis is the rank of this sequence, from
highest score to lowest; the y-axis is the E-value
calcu&acirc; <br>
lated for this sequence. E-values are calculated using
H3&rsquo;s default procedures (i.e. the pmu, plambda
parameters in the output table). You expect a rough match
between <br>
rank and E-value if E-values are accurately estimated.</p>

<p style="margin-top: 1em">--ffile &lt;f&gt; <br>
Output a &quot;filter power&quot; file to &lt;f&gt;: for
each model, a line with three fields: model name, number of
sequences passing the P-value threshold, and fraction of
sequences <br>
passing the P-value threshold. See --pthresh for setting the
P-value threshold, which defaults to 0.02 (the default MSV
filter threshold in H3). The P-values are as deter&acirc;
<br>
mined by H3&rsquo;s default procedures (the pmu,plambda
parameters in the output table). If all is well, you expect
to see filter power equal to the predicted P-value setting
<br>
of the threshold.</p>

<p style="margin-top: 1em">--pfile &lt;f&gt; <br>
Output cumulative survival plots (P(S&gt;x)) to file
&lt;f&gt; in XMGRACE xy format. There are three plots: (1)
the observed score distribution; (2) the maximum likelihood
fitted <br>
distribution; (3) a maximum likelihood fit to the location
parameter (mu/tau) while <br>
assuming lambda=log_2.</p>

<p style="margin-top: 1em">--xfile &lt;f&gt; <br>
Output the bit scores as a binary array of double-precision
floats (8 bytes per score) to file &lt;f&gt;. Programs like
Easel&rsquo;s esl-histplot can read such binary files. This
is <br>
useful when generating extremely large sample sizes.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING MODEL
CONFIGURATION (MODE) <br>
H3 only uses multihit local alignment ( --fs mode), and this
is where we believe the statistical fits. Unihit local
alignment scores (Smith/Waterman; --sw mode) also obey our
<br>
statistical conjectures. Glocal alignment statistics (either
multihit or unihit) are still not adequately understood nor
adequately fitted.</p>

<p style="margin-top: 1em">--fs Collect multihit local
alignment scores. This is the default. alignment as
&rsquo;fragment search mode&rsquo;.</p>

<p style="margin-top: 1em">--sw Collect unihit local
alignment scores. The H3 J state is disabled. alignment as
&rsquo;Smith/Waterman search mode&rsquo;.</p>

<p style="margin-top: 1em">--ls Collect multihit glocal
alignment scores. In glocal (global/local) alignment, the
entire model must align, to a subsequence of the target. The
H3 local entry/exit transi&acirc; <br>
tion probabilities are disabled. &rsquo;ls&rsquo; comes from
HMMER2&rsquo;s historical terminology for multihit local
alignment as &rsquo;local search mode&rsquo;.</p>

<p style="margin-top: 1em">--s Collect unihit glocal
alignment scores. Both the H3 J state and local entry/exit
transition probabilities are disabled. &rsquo;s&rsquo; comes
from HMMER2&rsquo;s historical terminology for <br>
unihit glocal alignment.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING SCORING
ALGORITHM <br>
--vit Collect Viterbi maximum likelihood alignment scores.
This is the default.</p>

<p style="margin-top: 1em">--fwd Collect Forward log-odds
likelihood scores, summed over alignment ensemble.</p>

<p style="margin-top: 1em">--hyb Collect
&rsquo;Hybrid&rsquo; scores, as described in papers by Yu
and Hwa (for instance, Bioinformatics 18:864, 2002). These
involve calculating a Forward matrix and taking the
maxi&acirc; <br>
mum cell value. The number itself is statistically somewhat
unmotivated, but the distribution is expected be a
well-behaved extreme value distribution (Gumbel).</p>

<p style="margin-top: 1em">--msv Collect MSV (multiple
ungapped segment Viterbi) scores, using H3&rsquo;s main
acceleration heuristic.</p>

<p style="margin-top: 1em">--fast For any of the above
options, use H3&rsquo;s optimized production implementation
(using SIMD vectorization). The default is to use the
implementations sacrifice a small amount <br>
of numerical precision. This can introduce confounding noise
into statistical simulations and fits, so when one gets
super-concerned about exact details, it&rsquo;s better to be
<br>
able to factor that source of noise out.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING FITTED TAIL
MASSES FOR FORWARD <br>
In some experiments, it was useful to fit Forward scores to
a range of different tail masses, rather than just one.
These options provide a mechanism for fitting an
evenly-spaced <br>
range of different tail masses. For each different tail
mass, a line is generated in the output.</p>

<p style="margin-top: 1em">--tmin &lt;x&gt; <br>
Set the lower bound on the tail mass distribution. (The
default is 0.02 for the default single tail mass.)</p>

<p style="margin-top: 1em">--tmax &lt;x&gt; <br>
Set the upper bound on the tail mass distribution. (The
default is 0.02 for the default single tail mass.)</p>

<p style="margin-top: 1em">--tpoints &lt;n&gt; <br>
Set the number of tail masses to sample, starting from
--tmin and ending at --tmax. (The default is 1, for the
default 0.02 single tail mass.)</p>

<p style="margin-top: 1em">--tlinear <br>
Sample a range of tail masses with uniform linear spacing.
The default is to use uniform logarithmic spacing.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING H3 PARAMETER
ESTIMATION METHODS <br>
H3 uses three short random sequence simulations to
estimating the location parameters for the expected score
distributions for MSV scores, Viterbi scores, and Forward
scores. <br>
These options allow these simulations to be modified.</p>

<p style="margin-top: 1em">--EmL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter mu for MSV E-values. Default is 200.</p>

<p style="margin-top: 1em">--EmN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter mu for MSV E-values. Default is
200.</p>

<p style="margin-top: 1em">--EvL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter mu for Viterbi E-values. Default is
200.</p>

<p style="margin-top: 1em">--EvN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter mu for Viterbi E-values. Default is
200.</p>

<p style="margin-top: 1em">--EfL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter tau for Forward E-values. Default is
100.</p>

<p style="margin-top: 1em">--EfN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter tau for Forward E-values. Default is
200.</p>

<p style="margin-top: 1em">--Eft &lt;x&gt; <br>
Sets the tail mass fraction to fit in the simulation that
estimates the location parameter tau for Forward evalues.
Default is 0.04.</p>

<p style="margin-top: 1em">DEBUGGING OPTIONS <br>
--stall <br>
For debugging the MPI master/worker version: pause after
start, to enable the developer to attach debuggers to the
running master and worker(s) processes. Send SIGCONT <br>
signal to release the pause. (Under gdb: (gdb) signal
SIGCONT) (Only available if optional MPI support was enabled
at compile-time.)</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Set the random number seed to &lt;n&gt;. The default is 0,
which makes the random number generator use an arbitrary
seed, so that different runs of hmmsim will almost certainly
<br>
generate a different statistical sample. For debugging, it
is useful to force reproducible results, by fixing a random
number seed.</p>

<p style="margin-top: 1em">EXPERIMENTAL OPTIONS <br>
These options were used in a small variety of different
exploratory experiments.</p>

<p style="margin-top: 1em">--bgflat <br>
Set the background residue distribution to a uniform
distribution, both for purposes of the null model used in
calculating scores, and for generating the random sequences.
<br>
The default is to use a standard amino acid background
frequency distribution.</p>

<p style="margin-top: 1em">--bgcomp <br>
Set the background residue distribution to the mean
composition of the profile. This was used in exploring some
of the effects of biased composition.</p>

<p style="margin-top: 1em">--x-no-lengthmodel <br>
Turn the H3 target sequence length model off. Set the
self-transitions for N,C,J and the null model to 350/351
instead; this emulates HMMER2. Not a good idea in general.
<br>
This was used to demonstrate one of the main H2 vs. H3
differences.</p>

<p style="margin-top: 1em">--nu &lt;x&gt; <br>
Set the nu parameter for the MSV algorithm -- the expected
number of ungapped local alignments per target sequence. The
default is 2.0, corresponding to a E-&gt;J transition <br>
probability of 0.5. This was used to test whether varying nu
has significant effect on result (it doesn&rsquo;t seem to,
within reason). This option only works if --msv is <br>
selected (it only affects MSV), and it will not work with
--fast (because the optimized implementations are hardwired
to assume nu=2.0).</p>

<p style="margin-top: 1em">--pthresh &lt;x&gt; <br>
Set the filter P-value threshold to use in generating filter
power files with --ffile. The default is 0.02 (which would
be appropriate for testing MSV scores, since this <br>
is the default MSV filter threshold in H3&rsquo;s
acceleration pipeline.) Other appropriate choices (matching
defaults in the acceleration pipeline) would be 0.001 for
Viterbi, <br>
and 1e-5 for Forward.</p>

<p style="margin-top: 1em">SEE ALSO <br>
See hmmer(1) for a master man page with a list of all the
individual man pages for programs in the HMMER package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
hmmsim(1)</p>
<hr>
</body>
</html>
