<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>hmmsim(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">hmmsim(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">hmmsim(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
hmmsim - collect score distributions on random sequences
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmmsim</b> <i>[options]</i> <i>&lt;hmmfile&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
The <b>hmmsim</b> program generates random sequences, scores them with the
  model(s) in <i>&lt;hmmfile&gt;</i>, and outputs various sorts of histograms,
  plots, and fitted distributions for the resulting scores.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
<b>hmmsim</b> is not a mainstream part of the HMMER package. Most users would
  have no reason to use it. It is used to develop and test the statistical
  methods used to determine P-values and E-values in HMMER3. For example, it was
  used to generate most of the results in a 2008 paper on H3's local alignment
  statistics (PLoS Comp Bio 4:e1000069, 2008;
  http://www.ploscompbiol.org/doi/pcbi.1000069).
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Because it is a research testbed, you should not expect it to be as robust as
  other programs in the package. For example, options may interact in weird
  ways; we haven't tested nor tried to anticipate all different possible
  combinations.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The main task is to fit a maximum likelihood Gumbel distribution to Viterbi
  scores or an maximum likelihood exponential tail to high-scoring Forward
  scores, and to test that these fitted distributions obey the conjecture that
  lambda ~ log_2 for both the Viterbi Gumbel and the Forward exponential tail.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The output is a table of numbers, one row for each model. Four different
  parametric fits to the score data are tested: (1) maximum likelihood fits to
  both location (mu/tau) and slope (lambda) parameters; (2) assuming
  lambda=log_2, maximum likelihood fit to the location parameter only; (3) same
  but assuming an edge-corrected lambda, using current procedures in H3 [Eddy,
  2008]; and (4) using both parameters determined by H3's current procedures.
  The standard simple, quick and dirty statistic for goodness-of-fit is 'E@10',
  the calculated E-value of the 10th ranked top hit, which we expect to be about
  10.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
In detail, the columns of the output are:
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>name</b></dt>
  <dd class="It-tag">Name of the model.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>tailp</b></dt>
  <dd class="It-tag">Fraction of the highest scores used to fit the
      distribution. For Viterbi, MSV, and Hybrid scores, this defaults to 1.0 (a
      Gumbel distribution is fitted to all the data). For Forward scores, this
      defaults to 0.02 (an exponential tail is fitted to the highest 2% scores).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mu/tau</b></dt>
  <dd class="It-tag">Location parameter for the maximum likelihood fit to the
      data.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>lambda</b></dt>
  <dd class="It-tag">Slope parameter for the maximum likelihood fit to the data.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>E@10</b></dt>
  <dd class="It-tag">The E-value calculated for the 10th ranked high score
      ('E@10') using the ML mu/tau and lambda. By definition, this expected to
      be about 10, if E-value estimation were accurate.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mufix</b></dt>
  <dd class="It-tag">Location parameter, for a maximum likelihood fit with a
      known (fixed) slope parameter lambda of log_2 (0.693).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>E@10fix</b></dt>
  <dd class="It-tag">The E-value calculated for the 10th ranked score using
      mufix and the expected lambda = log_2 = 0.693.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mufix2</b></dt>
  <dd class="It-tag">Location parameter, for a maximum likelihood fit with an
      edge-effect-corrected lambda.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>E@10fix2</b></dt>
  <dd class="It-tag">The E-value calculated for the 10th ranked score using
      mufix2 and the edge-effect-corrected lambda.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pmu</b></dt>
  <dd class="It-tag">Location parameter as determined by H3's estimation
      procedures.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>plambda</b></dt>
  <dd class="It-tag">Slope parameter as determined by H3's estimation
      procedures.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pE@10</b></dt>
  <dd class="It-tag">The E-value calculated for the 10th ranked score using pmu,
      plambda.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
At the end of this table, one more line is printed, starting with # and
  summarizing the overall CPU time used by the simulations.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Some of the optional output files are in xmgrace xy format. xmgrace is powerful
  and freely available graph-plotting software.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="MISCELLANEOUS_OPTIONS"><a class="selflink" href="#MISCELLANEOUS_OPTIONS">MISCELLANEOUS
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and all
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b></dt>
  <dd class="It-tag">Collect expected Viterbi alignment length statistics from
      each simulated sequence. This only works with Viterbi scores (the default;
      see <i>--vit</i>). Two additional fields are printed in the output table
      for each model: the mean length of Viterbi alignments, and the standard
      deviation.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b></dt>
  <dd class="It-tag">(Verbose). Print the scores too, one score per line.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the length of the randomly sampled (nonhomologous)
      sequences to <i>&lt;n&gt;</i>. The default is 100.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the number of randomly sampled sequences to
      <i>&lt;n&gt;</i>. The default is 1000.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mpi</b></dt>
  <dd class="It-tag">Run in MPI parallel mode, under <b>mpirun</b>. It is
      parallelized at the level of sending one profile at a time to an MPI
      worker process, so parallelization only helps if you have more than one
      profile in the <i>&lt;hmmfile&gt;</i>, and you want to have at least as
      many profiles as MPI worker processes. (Only available if optional MPI
      support was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_OUTPUT"><a class="selflink" href="#OPTIONS_CONTROLLING_OUTPUT">OPTIONS
  CONTROLLING OUTPUT</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save the main output table to a file <i>&lt;f&gt;</i>
      rather than sending it to stdout.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--afile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">When collecting Viterbi alignment statistics (the <i>-a</i>
      option), for each sampled sequence, output two fields per line to a file
      <i>&lt;f&gt;</i>: the length of the optimal alignment, and the Viterbi bit
      score. Requires that the <i>-a</i> option is also used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--efile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Output a rank vs. E-value plot in XMGRACE xy format to file
      <i>&lt;f&gt;</i>. The x-axis is the rank of this sequence, from highest
      score to lowest; the y-axis is the E-value calculated for this sequence.
      E-values are calculated using H3's default procedures (i.e. the pmu,
      plambda parameters in the output table). You expect a rough match between
      rank and E-value if E-values are accurately estimated.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ffile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Output a &quot;filter power&quot; file to <i>&lt;f&gt;</i>:
      for each model, a line with three fields: model name, number of sequences
      passing the P-value threshold, and fraction of sequences passing the
      P-value threshold. See <i>--pthresh</i> for setting the P-value threshold,
      which defaults to 0.02 (the default MSV filter threshold in H3). The
      P-values are as determined by H3's default procedures (the pmu,plambda
      parameters in the output table). If all is well, you expect to see filter
      power equal to the predicted P-value setting of the threshold.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pfile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Output cumulative survival plots (P(S&gt;x)) to file
      <i>&lt;f&gt;</i> in XMGRACE xy format. There are three plots: (1) the
      observed score distribution; (2) the maximum likelihood fitted
      distribution; (3) a maximum likelihood fit to the location parameter
      (mu/tau) while
    <br/>
     assuming lambda=log_2.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--xfile</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Output the bit scores as a binary array of double-precision
      floats (8 bytes per score) to file <i>&lt;f&gt;</i>. Programs like Easel's
      <b>esl-histplot</b> can read such binary files. This is useful when
      generating extremely large sample sizes.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_MODEL_CONFIGURATION_(MODE)"><a class="selflink" href="#OPTIONS_CONTROLLING_MODEL_CONFIGURATION_(MODE)">OPTIONS
  CONTROLLING MODEL CONFIGURATION (MODE)</a></h1>
H3 only uses multihit local alignment ( <i>--fs</i> mode), and this is where we
  believe the statistical fits. Unihit local alignment scores (Smith/Waterman;
  <i>--sw</i> mode) also obey our statistical conjectures. Glocal alignment
  statistics (either multihit or unihit) are still not adequately understood nor
  adequately fitted.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fs</b></dt>
  <dd class="It-tag">Collect multihit local alignment scores. This is the
      default. alignment as 'fragment search mode'.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sw</b></dt>
  <dd class="It-tag">Collect unihit local alignment scores. The H3 J state is
      disabled. alignment as 'Smith/Waterman search mode'.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ls</b></dt>
  <dd class="It-tag">Collect multihit glocal alignment scores. In glocal
      (global/local) alignment, the entire model must align, to a subsequence of
      the target. The H3 local entry/exit transition probabilities are disabled.
      'ls' comes from HMMER2's historical terminology for multihit local
      alignment as 'local search mode'.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--s</b></dt>
  <dd class="It-tag">Collect unihit glocal alignment scores. Both the H3 J state
      and local entry/exit transition probabilities are disabled. 's' comes from
      HMMER2's historical terminology for unihit glocal alignment.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_SCORING_ALGORITHM"><a class="selflink" href="#OPTIONS_CONTROLLING_SCORING_ALGORITHM">OPTIONS
  CONTROLLING SCORING ALGORITHM</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--vit</b></dt>
  <dd class="It-tag">Collect Viterbi maximum likelihood alignment scores. This
      is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fwd</b></dt>
  <dd class="It-tag">Collect Forward log-odds likelihood scores, summed over
      alignment ensemble.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hyb</b></dt>
  <dd class="It-tag">Collect 'Hybrid' scores, as described in papers by Yu and
      Hwa (for instance, Bioinformatics 18:864, 2002). These involve calculating
      a Forward matrix and taking the maximum cell value. The number itself is
      statistically somewhat unmotivated, but the distribution is expected be a
      well-behaved extreme value distribution (Gumbel).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--msv</b></dt>
  <dd class="It-tag">Collect MSV (multiple ungapped segment Viterbi) scores,
      using H3's main acceleration heuristic.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fast</b></dt>
  <dd class="It-tag">For any of the above options, use H3's optimized production
      implementation (using SIMD vectorization). The default is to use the
      implementations sacrifice a small amount of numerical precision. This can
      introduce confounding noise into statistical simulations and fits, so when
      one gets super-concerned about exact details, it's better to be able to
      factor that source of noise out.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_FITTED_TAIL_MASSES_FOR_FORWARD"><a class="selflink" href="#OPTIONS_CONTROLLING_FITTED_TAIL_MASSES_FOR_FORWARD">OPTIONS
  CONTROLLING FITTED TAIL MASSES FOR FORWARD</a></h1>
In some experiments, it was useful to fit Forward scores to a range of different
  tail masses, rather than just one. These options provide a mechanism for
  fitting an evenly-spaced range of different tail masses. For each different
  tail mass, a line is generated in the output.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tmin</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the lower bound on the tail mass distribution. (The
      default is 0.02 for the default single tail mass.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tmax</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the upper bound on the tail mass distribution. (The
      default is 0.02 for the default single tail mass.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tpoints</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the number of tail masses to sample, starting from
      <i>--tmin</i> and ending at <i>--tmax</i>. (The default is 1, for the
      default 0.02 single tail mass.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tlinear</b></dt>
  <dd class="It-tag">Sample a range of tail masses with uniform linear spacing.
      The default is to use uniform logarithmic spacing.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_H3_PARAMETER_ESTIMATION_METHODS"><a class="selflink" href="#OPTIONS_CONTROLLING_H3_PARAMETER_ESTIMATION_METHODS">OPTIONS
  CONTROLLING H3 PARAMETER ESTIMATION METHODS</a></h1>
H3 uses three short random sequence simulations to estimating the location
  parameters for the expected score distributions for MSV scores, Viterbi
  scores, and Forward scores. These options allow these simulations to be
  modified.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EmL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter mu for MSV E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EmN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter mu for MSV E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EvL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter mu for Viterbi E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EvN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter mu for Viterbi E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EfL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter tau for Forward E-values. Default is 100.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EfN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter tau for Forward E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--Eft</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the tail mass fraction to fit in the simulation that
      estimates the location parameter tau for Forward evalues. Default is 0.04.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="DEBUGGING_OPTIONS"><a class="selflink" href="#DEBUGGING_OPTIONS">DEBUGGING
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stall</b></dt>
  <dd class="It-tag">For debugging the MPI master/worker version: pause after
      start, to enable the developer to attach debuggers to the running master
      and worker(s) processes. Send SIGCONT signal to release the pause. (Under
      gdb: <i>(gdb) signal SIGCONT</i>) (Only available if optional MPI support
      was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the random number seed to <i>&lt;n&gt;</i>. The default
      is 0, which makes the random number generator use an arbitrary seed, so
      that different runs of <b>hmmsim</b> will almost certainly generate a
      different statistical sample. For debugging, it is useful to force
      reproducible results, by fixing a random number seed.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXPERIMENTAL_OPTIONS"><a class="selflink" href="#EXPERIMENTAL_OPTIONS">EXPERIMENTAL
  OPTIONS</a></h1>
These options were used in a small variety of different exploratory experiments.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bgflat</b></dt>
  <dd class="It-tag">Set the background residue distribution to a uniform
      distribution, both for purposes of the null model used in calculating
      scores, and for generating the random sequences. The default is to use a
      standard amino acid background frequency distribution.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bgcomp</b></dt>
  <dd class="It-tag">Set the background residue distribution to the mean
      composition of the profile. This was used in exploring some of the effects
      of biased composition.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--x-no-lengthmodel</b></dt>
  <dd class="It-tag">Turn the H3 target sequence length model off. Set the
      self-transitions for N,C,J and the null model to 350/351 instead; this
      emulates HMMER2. Not a good idea in general. This was used to demonstrate
      one of the main H2 vs. H3 differences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nu</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the nu parameter for the MSV algorithm -- the expected
      number of ungapped local alignments per target sequence. The default is
      2.0, corresponding to a E-&gt;J transition probability of 0.5. This was
      used to test whether varying nu has significant effect on result (it
      doesn't seem to, within reason). This option only works if <i>--msv</i> is
      selected (it only affects MSV), and it will not work with <i>--fast</i>
      (because the optimized implementations are hardwired to assume nu=2.0).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pthresh</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the filter P-value threshold to use in generating
      filter power files with <i>--ffile</i>. The default is 0.02 (which would
      be appropriate for testing MSV scores, since this is the default MSV
      filter threshold in H3's acceleration pipeline.) Other appropriate choices
      (matching defaults in the acceleration pipeline) would be 0.001 for
      Viterbi, and 1e-5 for Forward.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>hmmer(1)</b> for a master man page with a list of all the individual man
  pages for programs in the HMMER package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your HMMER
  distribution (Userguide.pdf); or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2015 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2015</td>
    <td class="foot-os">HMMER 3.1b2</td>
  </tr>
</table>
</body>
</html>
