<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>HOMGENEMAPPING(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">HOMGENEMAPPING(1)</td>
    <td class="head-vol">&#x00A0;</td>
    <td class="head-rtitle">HOMGENEMAPPING(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
homgenemapping - create summary of gene homology
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>homGeneMapping</b> [options] --gtfs=gffilenames.tbl --halfile=aln.hal
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>homGeneMapping</b> takes a set of gene predictions of different genomes and a
  hal alignment of the genomes and prints a summary for each gene, e.g.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;how many of its exons/introns are in
  agreement with genes of other genomes</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;how many of its exons/introns are
  supported by extrinsic evidence from any of the genomes</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;a list of geneids of homologous
  genes</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Mandatory_parameters"><a class="selflink" href="#Mandatory_parameters">Mandatory
  parameters</a></h2>
<b>--halfile=aln.hal</b>
<div style="margin-left: 4.00ex;">input hal file</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--gtfs=gtffilenames.tbl</b>
<div style="margin-left: 4.00ex;">a text file containing the locations of the
  input gene files and optionally the hints files (both in GTF format). The file
  is formatted as follows:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
name_of_genome_1  path/to/genefile/of/genome_1  path/to/hintsfile/of/genome_1
name_of_genome_2  path/to/genefile/of/genome_2  path/to/hintsfile/of/genome_2
...
name_of_genome_N  path/to/genefile/of/genome_N  path/to/hintsfile/of/genome_N
</pre>
</div>
</div>
<h2 class="Ss" title="Ss" id="Additional_options"><a class="selflink" href="#Additional_options">Additional
  options</a></h2>
<b>--cpus=N</b>
<div style="margin-left: 4.00ex;">N is the number of CPUs to use (default:
  1)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--noDupes</b>
<div style="margin-left: 4.00ex;">do not map between duplications in hal graph.
  (default: off)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--halLiftover_exec_dir=DIR</b>
<div style="margin-left: 4.00ex;">Directory that contains the executable
  halLiftover If not specified it must be in $PATH environment variable.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--tmpdir=DIR</b>
<div style="margin-left: 4.00ex;">a temporary file directory that stores lifted
  over files. (default 'tmp/' in current directory)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--outdir=DIR</b>
<div style="margin-left: 4.00ex;">file directory that stores output gene files.
  (default: current directory)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--printHomologs=FILE</b>
<div style="margin-left: 4.00ex;">prints disjunct sets of homologous transcripts
  to FILE, e.g.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">
<pre>
# 0     dana
# 1     dere
# 2     dgri
# 3     dmel
# 4     dmoj
# 5     dper
(0, jg4139.t1) (0, jg4140.t1) (1, jg7797.t1) (2, jg3247.t1) (4, jg6720.t1) (5, jg313.t1)
(1, jg14269.t1) (3, jg89.t1) (5, jg290.t1)
...
Two transcripts are in the same set, if all their exons/introns are homologs and their are
no additional exons/introns.
This option requires the Boost C++ Library
</pre>
</div>
</div>
<h1 class="Sh" title="Sh" id="EXAMPLE"><a class="selflink" href="#EXAMPLE">EXAMPLE</a></h1>
homGeneMapping --noDupes
  --halLiftover_exec_dir=~/tools/progressiveCactus/submodules/hal/bin
  --gtfs=gtffilenames.tbl --halfile=msca.hal
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
AUGUSTUS was written by M. Stanke, O. Keller, S. K&#x00F6;nig, L. Gerischer and
  L. Romoth.
<h1 class="Sh" title="Sh" id="ADDITIONAL_DOCUMENTATION"><a class="selflink" href="#ADDITIONAL_DOCUMENTATION">ADDITIONAL
  DOCUMENTATION</a></h1>
An exhaustive documentation can be found in the file
  /usr/share/augustus/README.TXT.</div>
<table class="foot">
  <tr>
    <td class="foot-date"></td>
    <td class="foot-os">&#x00A0;</td>
  </tr>
</table>
</body>
</html>
