<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:16:57 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>HOMGENEMAPPING(1) HOMGENEMAPPING(1)</p>

<p style="margin-top: 1em">NAME <br>
homgenemapping - create summary of gene homology</p>

<p style="margin-top: 1em">SYNOPSIS <br>
homGeneMapping [options] --gtfs=gffilenames.tbl
--halfile=aln.hal</p>

<p style="margin-top: 1em">DESCRIPTION <br>
homGeneMapping takes a set of gene predictions of different
genomes and a hal alignment of the genomes and prints a
summary for each gene, e.g.</p>

<p style="margin-top: 1em">&Acirc;&middot; how many of its
exons/introns are in agreement with genes of other
genomes</p>

<p style="margin-top: 1em">&Acirc;&middot; how many of its
exons/introns are supported by extrinsic evidence from any
of the genomes</p>

<p style="margin-top: 1em">&Acirc;&middot; a list of
geneids of homologous genes</p>

<p style="margin-top: 1em">OPTIONS <br>
Mandatory parameters <br>
--halfile=aln.hal <br>
input hal file</p>

<p style="margin-top: 1em">--gtfs=gtffilenames.tbl <br>
a text file containing the locations of the input gene files
and optionally the hints files (both in GTF format). The
file is formatted as follows:</p>

<p style="margin-top: 1em">name_of_genome_1
path/to/genefile/of/genome_1 path/to/hintsfile/of/genome_1
<br>
name_of_genome_2 path/to/genefile/of/genome_2
path/to/hintsfile/of/genome_2 <br>
... <br>
name_of_genome_N path/to/genefile/of/genome_N
path/to/hintsfile/of/genome_N</p>

<p style="margin-top: 1em">Additional options <br>
--cpus=N <br>
N is the number of CPUs to use (default: 1)</p>

<p style="margin-top: 1em">--noDupes <br>
do not map between duplications in hal graph. (default:
off)</p>

<p style="margin-top: 1em">--halLiftover_exec_dir=DIR <br>
Directory that contains the executable halLiftover If not
specified it must be in $PATH environment variable.</p>

<p style="margin-top: 1em">--tmpdir=DIR <br>
a temporary file directory that stores lifted over files.
(default &rsquo;tmp/&rsquo; in current directory)</p>

<p style="margin-top: 1em">--outdir=DIR <br>
file directory that stores output gene files. (default:
current directory)</p>

<p style="margin-top: 1em">--printHomologs=FILE <br>
prints disjunct sets of homologous transcripts to FILE,
e.g.</p>

<p style="margin-top: 1em"># 0 dana <br>
# 1 dere <br>
# 2 dgri <br>
# 3 dmel <br>
# 4 dmoj <br>
# 5 dper <br>
(0, jg4139.t1) (0, jg4140.t1) (1, jg7797.t1) (2, jg3247.t1)
(4, jg6720.t1) (5, jg313.t1) <br>
(1, jg14269.t1) (3, jg89.t1) (5, jg290.t1) <br>
... <br>
Two transcripts are in the same set, if all their
exons/introns are homologs and their are <br>
no additional exons/introns. <br>
This option requires the Boost C++ Library</p>

<p style="margin-top: 1em">EXAMPLE <br>
homGeneMapping --noDupes
--halLiftover_exec_dir=~/tools/progressiveCactus/submodules/hal/bin
--gtfs=gtffilenames.tbl --halfile=msca.hal</p>

<p style="margin-top: 1em">AUTHORS <br>
AUGUSTUS was written by M. Stanke, O. Keller, S.
K&Atilde;&para;nig, L. Gerischer and L. Romoth.</p>

<p style="margin-top: 1em">ADDITIONAL DOCUMENTATION <br>
An exhaustive documentation can be found in the file
/usr/share/augustus/README.TXT.</p>
 
<p style="margin-top: 1em">HOMGENEMAPPING(1)</p>
<hr>
</body>
</html>
