<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>IDBA_HYBRID(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">IDBA_HYBRID(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">IDBA_HYBRID(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
idba_hybrid - Iterative de Bruijn Graph Assembler for hybrid sequencing data
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>idba_hybrid</b> <i>-r read.fa -o output_dir </i>[<i>--reference ref.fa</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
IDBA-Tran is an iterative De Bruijn Graph De Novo short read assembler for
  transcriptome. It is purely de novo assembler based on only RNA sequencing
  reads. IDBA-Tran uses local assembly to reconstructing missing k-mers in
  low-expressed transcripts and then employs progressive cutoff on contigs to
  separate the graph into components. Each component corresponds to one gene in
  most cases and contains not many transcripts. A heuristic algorithm based on
  pair-end reads is then used to find the isoforms.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>, <b>--out</b> arg (=out)</dt>
  <dd class="It-tag">output directory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b>, <b>--read</b> arg</dt>
  <dd class="It-tag">fasta read file (&lt;=128)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read_level_2</b> arg</dt>
  <dd class="It-tag">paired-end reads fasta for second level scaffolds</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read_level_3</b> arg</dt>
  <dd class="It-tag">paired-end reads fasta for third level scaffolds</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read_level_4</b> arg</dt>
  <dd class="It-tag">paired-end reads fasta for fourth level scaffolds</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--read_level_5</b> arg</dt>
  <dd class="It-tag">paired-end reads fasta for fifth level scaffolds</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b>, <b>--long_read</b> arg</dt>
  <dd class="It-tag">fasta long read file (&gt;128)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--reference</b> arg</dt>
  <dd class="It-tag">reference genome</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mink</b> arg (=20)</dt>
  <dd class="It-tag">minimum k value (&lt;=124)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--maxk</b> arg (=100)</dt>
  <dd class="It-tag">maximum k value (&lt;=124)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--step</b> arg (=20)</dt>
  <dd class="It-tag">increment of k-mer of each iteration</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--inner_mink</b> arg (=10)</dt>
  <dd class="It-tag">inner minimum k value</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--inner_step</b> arg (=5)</dt>
  <dd class="It-tag">inner increment of k-mer</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--prefix</b> arg (=3)</dt>
  <dd class="It-tag">prefix length used to build sub k-mer table</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_count</b> arg (=2)</dt>
  <dd class="It-tag">minimum multiplicity for filtering k-mer when building the
      graph</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_support</b> arg (=1)</dt>
  <dd class="It-tag">minimum supoort in each iteration</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--num_threads</b> arg (=0)</dt>
  <dd class="It-tag">number of threads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_kmer</b> arg (=30)</dt>
  <dd class="It-tag">seed kmer size for alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_contig</b> arg (=200)</dt>
  <dd class="It-tag">minimum size of contig</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_region</b> arg (=500)</dt>
  <dd class="It-tag">minimum size of region in reference genome</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--similar</b> arg (=0.95)</dt>
  <dd class="It-tag">similarity for alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max_mismatch</b> arg (=3)</dt>
  <dd class="It-tag">max mismatch of error correction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_pairs</b> arg (=3)</dt>
  <dd class="It-tag">minimum number of pairs</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max_gap</b> arg (=50)</dt>
  <dd class="It-tag">maximum gap in reference</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no_local</b></dt>
  <dd class="It-tag">do not use local assembly</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no_coverage</b></dt>
  <dd class="It-tag">do not iterate on coverage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no_correct</b></dt>
  <dd class="It-tag">do not do correction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pre_correction</b></dt>
  <dd class="It-tag">perform pre-correction before assembly</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2014</td>
    <td class="foot-os">idba_hybrid 1.1.1</td>
  </tr>
</table>
</body>
</html>
