<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>INDELIBLE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">INDELIBLE(1)</td>
    <td class="head-vol"></td>
    <td class="head-rtitle">INDELIBLE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
indelible - powerful and flexible simulator of biological evolution
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>indelible</b>
<dl class="Bl-tag">
  <dt class="It-tag">No options can be passed via the commandline.</dt>
  <dd class="It-tag"></dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>indelible</b> can simulate evolution of
    multi-partitioned nucleotide, amino-acid, or codon data sets through the
    processes of insertion, deletion, and substitution in continous time. All
    common evolutionary models are supported.</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">The full documentation for <b>indelible</b> is available
    under <i>/usr/share/doc/indelible/help/index.html</i>.</dt>
  <dd class="It-tag"></dd>
</dl>
<h1 class="Sh" title="Sh" id="CONFIGURATION"><a class="selflink" href="#CONFIGURATION">CONFIGURATION</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>indelible</b> reads the settings for a simulation run
    from a file named <i>control.txt</i> in the current directory. The file is
    split into blocks, with each block controlling one particular aspect of the
    simulation. Blocks may contain further sub-blocks. Comments can be given in
    C or C++ style. If <b>indelible</b> is executed without a control file
    present, an example file is generated.</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>[TYPE]</b> type</dt>
  <dd class="It-tag">Every control file must begin with a <b>TYPE</b> block. The
      possible types are <i>NUCLEOTIDE</i>, <i>AMINOACID</i>, and
    <i>CODON</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>[SETTINGS]</b></dt>
  <dd class="It-tag">This block specifies non-essential user preferences such as
      output file types and formats, seeds for the random number generator, and
      whether to output detailed reports.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>[MODEL]</b> modelname</dt>
  <dd class="It-tag">This block starts a new evolutionary model to be used for
      simulation. The particular parameters including substitution models,
      indel-rates and codon-site models are controlled via submodels.
      <i>modelname</i> can be any name of your choosing.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>[TREE]</b> treename newick</dt>
  <dd class="It-tag">This block is used to specify a guide-tree with the given
      name <i>treename</i>. Evolutionary distances are represented as branch
      lengths for the tree in <i>NEWICK</i> format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>[BRANCHES]</b></dt>
  <dd class="It-tag">These blocks are used to simulate non-stationary and
      non-homogenous processes. Different models can be specified on different
      branches of the guide-tree allowing branches to have different models of
      substitution, indel length distribution, rate heterogeneity, base
      composition etc.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>[PARTITIONS]</b> partitionname</dt>
  <dd class="It-tag">During each run multiple sequence partitions can be
      simulated. For each partition the guide-tree, evolutionary model and
      sequence length has to be specified.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>[EVOLVE]</b></dt>
  <dd class="It-tag">Within this block multiple replicas of a partition may be
      generated.</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright &#x00A9; 2010 William Fletcher License GPLv3+: GNU GPL version 3 or
  later.
<div>&#x00A0;</div>
This is free software: you are free to change and redistribute it. There is NO
  WARRANTY, to the extent permitted by law. The full license text is available
  at &lt;http://gnu.org/licenses/gpl.html&gt;.
<h1 class="Sh" title="Sh" id="ACKNOWLEDGMENTS"><a class="selflink" href="#ACKNOWLEDGMENTS">ACKNOWLEDGMENTS</a></h1>
William Fletcher and Ziheng Yang (2009). INDELible: A Flexible Simulator of
  Biological Sequence Evolution, <i>Molecular Biology and Evolution,</i>
  26(8):1879-1888.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2010</td>
    <td class="foot-os">1.03</td>
  </tr>
</table>
</body>
</html>
