<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:18:36 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>INDELIBLE(1) INDELIBLE(1)</p>

<p style="margin-top: 1em">NAME <br>
indelible - powerful and flexible simulator of biological
evolution</p>

<p style="margin-top: 1em">SYNOPSIS <br>
indelible</p>

<p style="margin-top: 1em">No options can be passed via the
commandline.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
indelible can simulate evolution of multi-partitioned
nucleotide, amino-acid, or codon data sets through the
processes of insertion, deletion, and substitution in
continous time. <br>
All common evolutionary models are supported.</p>

<p style="margin-top: 1em">The full documentation for
indelible is available under
/usr/share/doc/indelible/help/index.html.</p>

<p style="margin-top: 1em">CONFIGURATION <br>
indelible reads the settings for a simulation run from a
file named control.txt in the current directory. The file is
split into blocks, with each block controlling one
particu&acirc; <br>
lar aspect of the simulation. Blocks may contain further
sub-blocks. Comments can be given in C or C++ style. If
indelible is executed without a control file present, an
example <br>
file is generated.</p>

<p style="margin-top: 1em">[TYPE] type <br>
Every control file must begin with a TYPE block. The
possible types are NUCLEOTIDE, AMINOACID, and CODON.</p>

<p style="margin-top: 1em">[SETTINGS] <br>
This block specifies non-essential user preferences such as
output file types and formats, seeds for the random number
generator, and whether to output detailed reports.</p>

<p style="margin-top: 1em">[MODEL] modelname <br>
This block starts a new evolutionary model to be used for
simulation. The particular parameters including substitution
models, indel-rates and codon-site models are con&acirc;
<br>
trolled via submodels. modelname can be any name of your
choosing.</p>

<p style="margin-top: 1em">[TREE] treename newick <br>
This block is used to specify a guide-tree with the given
name treename. Evolutionary distances are represented as
branch lengths for the tree in NEWICK format.</p>

<p style="margin-top: 1em">[BRANCHES] <br>
These blocks are used to simulate non-stationary and
non-homogenous processes. Different models can be specified
on different branches of the guide-tree allowing branches
<br>
to have different models of substitution, indel length
distribution, rate heterogeneity, base composition etc.</p>

<p style="margin-top: 1em">[PARTITIONS] partitionname <br>
During each run multiple sequence partitions can be
simulated. For each partition the guide-tree, evolutionary
model and sequence length has to be specified.</p>

<p style="margin-top: 1em">[EVOLVE] <br>
Within this block multiple replicas of a partition may be
generated.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 2010 William Fletcher License
GPLv3+: GNU GPL version 3 or later. <br>
This is free software: you are free to change and
redistribute it. There is NO WARRANTY, to the extent
permitted by law. The full license text is available at <br>
&lt;http://gnu.org/licenses/gpl.html&gt;.</p>

<p style="margin-top: 1em">ACKNOWLEDGMENTS <br>
William Fletcher and Ziheng Yang (2009). INDELible: A
Flexible Simulator of Biological Sequence Evolution,
Molecular Biology and Evolution, 26(8):1879-1888.</p>

<p style="margin-top: 1em">1.03 2010 INDELIBLE(1)</p>
<hr>
</body>
</html>
