<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:18:39 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>Infernal(1) Infernal Manual Infernal(1)</p>

<p style="margin-top: 1em">NAME <br>
Infernal - sequence analysis using profiles of RNA sequence
and secondary structure consensus</p>

<p style="margin-top: 1em">SYNOPSIS <br>
cmalign <br>
Align sequences to a covariance model</p>

<p style="margin-top: 1em">cmbuild <br>
Construct covariance model(s) from structurally annotated
RNA multiple sequence alignment(s)</p>

<p style="margin-top: 1em">cmcalibrate <br>
Fit exponential tails for covariance model E-value
determination</p>

<p style="margin-top: 1em">cmconvert <br>
Convert Infernal covariance model files</p>

<p style="margin-top: 1em">cmemit <br>
Sample sequences from a covariance model</p>

<p style="margin-top: 1em">cmfetch <br>
Retrieve covariance model(s) from a file</p>

<p style="margin-top: 1em">cmpress <br>
Prepare a covariance model database for cmscan</p>

<p style="margin-top: 1em">cmscan <br>
Search sequence(s) against a covariance model database</p>

<p style="margin-top: 1em">cmsearch <br>
Search covariance model(s) against a sequence database</p>

<p style="margin-top: 1em">cmstat <br>
summary statistics for a covariance model file</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Infernal is a suite of several programs for structural RNA
sequence alignment and database homology search. It uses
probabilistic models called &quot;covariance models&quot;
(CMs) to rep&acirc; <br>
resent the likely evolutionary homologs of a multiple
alignment (or single sequence) of a structural RNA sequence
family.</p>

<p style="margin-top: 1em">Along with the Rfam database of
RNA families and associated CMs (http://rfam.sanger.ac.uk),
Infernal can be used to annotate homologs of known
structural RNA families in genomes.</p>

<p style="margin-top: 1em">Infernal is closely related to
the HMMER software suite for sequence family analysis using
profile HMMs (http://hmmer.org), but is designed
specifically for structural RNA <br>
sequence families. In addition to modeling the conserved
sequence of a family as profile HMMs do, CMs model the
family&rsquo;s conserved, well-nested (non-pseudoknotted)
secondary <br>
structure as well. Consequently, CM search and alignment
methods are relatively computationally expensive. Infernal
uses profile HMMs as filters and for deriving constraints to
<br>
make the CM methods more practical.</p>

<p style="margin-top: 1em">Infernal is used in three main
modes: to search a sequence database for new homologs of an
RNA family (or annotate homologs in a genome); to search a
CM database (like Rfam) to <br>
find what known family a query sequence belongs to; and to
automatically construct large multiple alignments (i.e. with
an effectively unlimited number of sequences) using a CM
<br>
representative of a sequence family.</p>

<p style="margin-top: 1em">Suppose you have a structurally
annotated multiple sequence alignment of a RNA sequence
family of interest, and you want to search a sequence
database for additional homologs. <br>
The cmbuild program builds covariance model(s) from multiple
alignment(s) and cmcalibrate determines important parameters
for estimating the statistical significance of database <br>
hits to the model in subsequent searches.</p>

<p style="margin-top: 1em">The cmsearch program searches
CM(s) against a sequence database.</p>

<p style="margin-top: 1em">Suppose you have sequence(s)
that you want to analyze using a Infernal-based CM database
like Rfam (http://rfam.sanger.ac.uk). The cmpress program
formats a covariance model <br>
(such as the file you would download from Rfam) into a
Infernal binary database. The cmscan program searches
sequence(s) against that database.</p>

<p style="margin-top: 1em">Suppose you want to align lots
of sequences. You can construct a manageably small
structural alignment of a representative set of sequences,
build a CM with cmbuild, and use the <br>
cmalign program to align any number of sequences to that
CM.</p>

<p style="margin-top: 1em">Infernal also includes some
auxiliary tools for working with large CM databases. cmfetch
fetches one or more CMs from a database. cmstat prints
summary statistics about a CM <br>
file.</p>

<p style="margin-top: 1em">For compatibility with previous
versions of Infernal, as well as with HMMER, the cmconvert
program converts CM files to a few other formats.</p>

<p style="margin-top: 1em">The cmemit program generates
(simulates) &quot;homologous&quot; sequences by sampling
from a CM. It can also generate a &quot;consensus&quot;
sequence.</p>

<p style="margin-top: 1em">Each program has its own man
page.</p>

<p style="margin-top: 1em">SEE ALSO <br>
This is a summary man page for the entire Infernal package.
See individual man pages [cmbuild(1), for example] for
usage, options, and description of each program in the
pack&acirc; <br>
age.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your Infernal distribution
(Userguide.pdf); or see the Infernal web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2016 Howard Hughes Medical Institute. <br>
Freely distributed under a BSD open source license.</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your Infernal source distribution, or see the Infernal web
page ().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">Infernal 1.1.2 July 2016
Infernal(1)</p>
<hr>
</body>
</html>
