<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>Infernal(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">Infernal(1)</td>
    <td class="head-vol">Infernal Manual</td>
    <td class="head-rtitle">Infernal(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
Infernal - sequence analysis using profiles of RNA sequence and secondary
  structure consensus
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>cmalign</b>
<br/>
 Align sequences to a covariance model
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmbuild</b>
<br/>
 Construct covariance model(s) from structurally annotated RNA multiple sequence
  alignment(s)
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmcalibrate</b>
<br/>
 Fit exponential tails for covariance model E-value determination
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmconvert</b>
<br/>
 Convert Infernal covariance model files
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmemit</b>
<br/>
 Sample sequences from a covariance model
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmfetch</b>
<br/>
 Retrieve covariance model(s) from a file
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmpress</b>
<br/>
 Prepare a covariance model database for cmscan
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmscan</b>
<br/>
 Search sequence(s) against a covariance model database
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmsearch</b>
<br/>
 Search covariance model(s) against a sequence database
<div style="height: 1.00em;">&#x00A0;</div>
<b>cmstat</b>
<br/>
 summary statistics for a covariance model file
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Infernal is a suite of several programs for structural RNA sequence alignment
  and database homology search. It uses probabilistic models called
  &quot;covariance models&quot; (CMs) to represent the likely evolutionary
  homologs of a multiple alignment (or single sequence) of a structural RNA
  sequence family.
<div style="height: 1.00em;">&#x00A0;</div>
Along with the Rfam database of RNA families and associated CMs
  (http://rfam.sanger.ac.uk), Infernal can be used to annotate homologs of known
  structural RNA families in genomes.
<div style="height: 1.00em;">&#x00A0;</div>
Infernal is closely related to the HMMER software suite for sequence family
  analysis using profile HMMs (http://hmmer.org), but is designed specifically
  for structural RNA sequence families. In addition to modeling the conserved
  sequence of a family as profile HMMs do, CMs model the family's conserved,
  well-nested (non-pseudoknotted) secondary structure as well. Consequently, CM
  search and alignment methods are relatively computationally expensive.
  Infernal uses profile HMMs as filters and for deriving constraints to make the
  CM methods more practical.
<div style="height: 1.00em;">&#x00A0;</div>
Infernal is used in three main modes: to search a sequence database for new
  homologs of an RNA family (or annotate homologs in a genome); to search a CM
  database (like Rfam) to find what known family a query sequence belongs to;
  and to automatically construct large multiple alignments (i.e. with an
  effectively unlimited number of sequences) using a CM representative of a
  sequence family.
<div style="height: 1.00em;">&#x00A0;</div>
Suppose you have a structurally annotated multiple sequence alignment of a RNA
  sequence family of interest, and you want to search a sequence database for
  additional homologs. The <b>cmbuild</b> program builds covariance model(s)
  from multiple alignment(s) and <b>cmcalibrate</b> determines important
  parameters for estimating the statistical significance of database hits to the
  model in subsequent searches.
<div style="height: 1.00em;">&#x00A0;</div>
The <b>cmsearch</b> program searches CM(s) against a sequence database.
<div style="height: 1.00em;">&#x00A0;</div>
Suppose you have sequence(s) that you want to analyze using a Infernal-based CM
  database like Rfam (http://rfam.sanger.ac.uk). The <b>cmpress</b> program
  formats a covariance model (such as the file you would download from Rfam)
  into a Infernal binary database. The <b>cmscan</b> program searches
  sequence(s) against that database.
<div style="height: 1.00em;">&#x00A0;</div>
Suppose you want to align lots of sequences. You can construct a manageably
  small structural alignment of a representative set of sequences, build a CM
  with <b>cmbuild,</b> and use the <b>cmalign</b> program to align any number of
  sequences to that CM.
<div style="height: 1.00em;">&#x00A0;</div>
Infernal also includes some auxiliary tools for working with large CM databases.
  <b>cmfetch</b> fetches one or more CMs from a database. <b>cmstat</b> prints
  summary statistics about a CM file.
<div style="height: 1.00em;">&#x00A0;</div>
For compatibility with previous versions of Infernal, as well as with HMMER, the
  <b>cmconvert</b> program converts CM files to a few other formats.
<div style="height: 1.00em;">&#x00A0;</div>
The <b>cmemit</b> program generates (simulates) &quot;homologous&quot; sequences
  by sampling from a CM. It can also generate a &quot;consensus&quot; sequence.
<div style="height: 1.00em;">&#x00A0;</div>
Each program has its own man page.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
This is a summary man page for the entire Infernal package. See individual man
  pages <b>[cmbuild(1),</b> for example] for usage, options, and description of
  each program in the package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your Infernal
  distribution (Userguide.pdf); or see the Infernal web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2016 Howard Hughes Medical Institute.
Freely distributed under a BSD open source license.
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your Infernal source distribution, or see the Infernal web page
  ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2016</td>
    <td class="foot-os">Infernal 1.1.2</td>
  </tr>
</table>
</body>
</html>
