<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>ipcress(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">ipcress(1)</td>
    <td class="head-vol">PCR simulation tool</td>
    <td class="head-rtitle">ipcress(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
ipcress - In-silico PCR experiment simulation system
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>ipcress [ options ] &lt;primer file&gt; &lt;sequence paths&gt;</b>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>ipcress</b> is the <i>I</i>n-silico <i>PCR</i> <i>E</i>xperiment
  <i>S</i>imulation <i>S</i>ystem.
<div style="height: 1.00em;">&#x00A0;</div>
This is a tool for simulation of PCR experiments. You supply a file containing
  primers and a set of sequences, and it predicts PCR products.
<div class="Pp"></div>
Ipcress is similar to the e-PCR program from the NCBI, but is much faster, and
  does not suffer from problems identifying matches when there are ambiguity
  symbols near primer ends.
<div class="Pp"></div>
If you supply many primers pairs together, ipcress will simulate the PCR
  experiments in parallel, allowing genome wide simulation of large numbers of
  experiments. It uses many libraries from the <b>exonerate</b> sequence
  comparison tool.
<h1 class="Sh" title="Sh" id="INPUT_FORMAT"><a class="selflink" href="#INPUT_FORMAT">INPUT
  FORMAT</a></h1>
The input for ipcress is a simple white-space delimited file describing one
  experiment per line. Each line contains the following 5 fields:
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>id</b></dt>
  <dd class="It-tag">An identifier for this experiment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>primer_A</b></dt>
  <dd class="It-tag">Sequence for the first primer</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>primer_B</b></dt>
  <dd class="It-tag">Sequence for the second primer</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>min_product_len</b></dt>
  <dd class="It-tag">Minimum product length to report</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>max_product_len</b></dt>
  <dd class="It-tag">Maximum product length to report</dd>
</dl>
</div>
<div class="Pp"></div>
Here is an example line in this format:
<div style="height: 1.00em;">&#x00A0;</div>
<pre>
<div class="Pp"></div>
ID0001 CATGCATGCATGC CGATGCANGCATGCT 900 1100
<div style="height: 1.00em;">&#x00A0;</div>
</pre>
<h1 class="Sh" title="Sh" id="OUTPUT_FORMAT"><a class="selflink" href="#OUTPUT_FORMAT">OUTPUT
  FORMAT</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">The output format describes one PCR product per-line,</dt>
  <dd class="It-tag">and is prefixed by &quot;ipcress:&quot;, followed by the
      following 11 fields:</dd>
</dl>
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>sequence_id</b></dt>
  <dd class="It-tag">The sequence identifier</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>experiment_id</b></dt>
  <dd class="It-tag">The PCR experiment id</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>product_length</b></dt>
  <dd class="It-tag">The PCR product length</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>primer_5</b></dt>
  <dd class="It-tag">The 5' primer (either A or B)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pos_5</b></dt>
  <dd class="It-tag">Position of the 5' primer</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mismatch_5</b></dt>
  <dd class="It-tag">Number of mismatches on 5' primer</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>primer_3</b></dt>
  <dd class="It-tag">|</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pos_3</b></dt>
  <dd class="It-tag">| Same fields for the 3' primer</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mismatch_3</b></dt>
  <dd class="It-tag">|</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>description</b></dt>
  <dd class="It-tag">A description of the PCR product</dd>
</dl>
</div>
<div class="Pp"></div>
The description field is one of the following 4 strings:
<div class="Pp"></div>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>forward</b></dt>
  <dd class="It-tag">Normal product, primer A followed by B</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>revcomp</b></dt>
  <dd class="It-tag">Normal product, primer B followed by A</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>single_A</b></dt>
  <dd class="It-tag">Bad product generated by primer_A only</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>single_B</b></dt>
  <dd class="It-tag">Bad product generated by primer_B only</dd>
</dl>
</div>
<div class="Pp"></div>
There is also a human-readable output displayed, is not designed for parsing
  (see: --pretty below).
<h1 class="Sh" title="Sh" id="GENERAL_OPTIONS"><a class="selflink" href="#GENERAL_OPTIONS">GENERAL
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">Most arguments have short and long forms. The long
    forms</dt>
  <dd class="It-tag">are more likely to be stable over time, and hence should be
      used in scripts which call ipcress.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h | --shorthelp &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Show help. This will display a concise summary of the
      available options, defaults and values currently set.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help &lt;boolean&gt;</b></dt>
  <dd class="It-tag">This shows all the help options including the defaults, the
      value currently set, and the environment variable which may be used to set
      each parameter. There will be an indication of which options are
      mandatory. Mandatory options have no default, and must have a value
      supplied for ipcress to run. If mandatory options are used in order, their
      flags may be skipped from the command line (see examples below). Unlike
      this man page, the information from this option will always be up to date
      with the latest version of the program.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v | --version &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Display the version number. Also displays other information
      such as the build date and glib version used.</dd>
</dl>
<h1 class="Sh" title="Sh" id="FILE_INPUT_OPTIONS"><a class="selflink" href="#FILE_INPUT_OPTIONS">FILE
  INPUT OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i | --input &lt;path&gt;</b></dt>
  <dd class="It-tag">PCR experiment data in the ipcress file format described
      above.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s | --sequence &lt;paths&gt;</b></dt>
  <dd class="It-tag">Specify the sequences. Multiple files may be specified
      here, which reduces the FSM building overhead, and makes ipcress run
      faster than running the process separately.</dd>
</dl>
<h1 class="Sh" title="Sh" id="IPCRESS_PARAMETERS"><a class="selflink" href="#IPCRESS_PARAMETERS">IPCRESS
  PARAMETERS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m | --mismatch &lt;mismatches&gt;</b></dt>
  <dd class="It-tag">Specify the number of mismatches allowed per primer.
      Allowing mismatches reduces the speed of the program as a large primer
      neighbourhood must be constructed, and fewer experiments can be fitted in
      memory prior to each scan of the sequence databases.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M | --memory &lt;Mb&gt;</b></dt>
  <dd class="It-tag">Specify the amount of memory the program should use. The
      more memory made available ipcress, the faster it will run, as more PCR
      experiments can be conducted in each scan of the sequence databases. This
      does not include memory used during the scan (for storing partial results
      and sequences), so the actual amount of memory used will be slightly
      higher.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p | --pretty &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Display results in a human-readable format, not designed
      for parsing.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P | --products &lt;boolean&gt;</b></dt>
  <dd class="It-tag">Display PCR products as a FASTA format sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S | --seed &lt;length&gt;</b></dt>
  <dd class="It-tag">Specifiy the seed length for the wordneighbourhood for the
      FSM. If set to zero, the full primer is used. Shorter words reduce the
      size of the neighbourhood, but increase the time taken by ipcress to
      filter false positive matches.</dd>
</dl>
<h1 class="Sh" title="Sh" id="ENVIRONMENT"><a class="selflink" href="#ENVIRONMENT">ENVIRONMENT</a></h1>
Not documented yet.
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<b>ipcress test.ipcress sequence.fasta</b>
<div style="margin-left: 5.00ex;">This is the simplest way that ipcress can be
  used.</div>
<b>ipcress dbsts_human.ipcress --sequence ncbi30/*.fasta --mismatch 1</b>
<div style="margin-left: 5.00ex;">Compare a input file against a set of fasta
  files, allowing one mismatch in each primer.</div>
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
This documentation accompanies version 2.2.0 of the exonerate package.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Guy St.C. Slater. &lt;guy@ebi.ac.uk&gt;.
<div class="Pp"></div>
See the AUTHORS file accompanying the source code for a list of contributors.
<h1 class="Sh" title="Sh" id="AVAILABILITY"><a class="selflink" href="#AVAILABILITY">AVAILABILITY</a></h1>
This source code for the exonerate package is available under the terms of the
  GNU <i>general</i> public licence.
<div style="height: 1.00em;">&#x00A0;</div>
Please see the file COPYING which was distrubuted with this package, or
  http://www.gnu.org/licenses/gpl.txt for details.
<div style="height: 1.00em;">&#x00A0;</div>
This package has been developed as part of the ensembl project. Please see
  http://www.ensembl.org/ for more information.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>exonerate</b>(1),<b>e-PCR</b></div>
<table class="foot">
  <tr>
    <td class="foot-date">March 2003</td>
    <td class="foot-os">ipcress</td>
  </tr>
</table>
</body>
</html>
