<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:18:56 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ipcress(1) PCR simulation tool ipcress(1)</p>

<p style="margin-top: 1em">NAME <br>
ipcress - In-silico PCR experiment simulation system</p>

<p style="margin-top: 1em">SYNOPSIS <br>
ipcress [ options ] &lt;primer file&gt; &lt;sequence
paths&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
ipcress is the In-silico PCR Experiment Simulation
System.</p>

<p style="margin-top: 1em">This is a tool for simulation of
PCR experiments. You supply a file containing primers and a
set of sequences, and it predicts PCR products.</p>

<p style="margin-top: 1em">Ipcress is similar to the e-PCR
program from the NCBI, but is much faster, and does not
suffer from problems identifying matches when there are
ambiguity symbols near primer <br>
ends.</p>

<p style="margin-top: 1em">If you supply many primers pairs
together, ipcress will simulate the PCR experiments in
parallel, allowing genome wide simulation of large numbers
of experiments. It uses many <br>
libraries from the exonerate sequence comparison tool.</p>

<p style="margin-top: 1em">INPUT FORMAT <br>
The input for ipcress is a simple white-space delimited file
describing one experiment per line. Each line contains the
following 5 fields:</p>

<p style="margin-top: 1em">id An identifier for this
experiment <br>
primer_A Sequence for the first primer <br>
primer_B Sequence for the second primer <br>
min_product_len Minimum product length to report <br>
max_product_len Maximum product length to report</p>

<p style="margin-top: 1em">Here is an example line in this
format:</p>

<p style="margin-top: 1em">ID0001 CATGCATGCATGC
CGATGCANGCATGCT 900 1100</p>

<p style="margin-top: 1em">OUTPUT FORMAT <br>
The output format describes one PCR product per-line, <br>
and is prefixed by &quot;ipcress:&quot;, followed by the
following 11 fields:</p>

<p style="margin-top: 1em">sequence_id The sequence
identifier <br>
experiment_id The PCR experiment id <br>
product_length The PCR product length <br>
primer_5 The 5&rsquo; primer (either A or B) <br>
pos_5 Position of the 5&rsquo; primer <br>
mismatch_5 Number of mismatches on 5&rsquo; primer <br>
primer_3 | <br>
pos_3 | Same fields for the 3&rsquo; primer <br>
mismatch_3 | <br>
description A description of the PCR product</p>

<p style="margin-top: 1em">The description field is one of
the following 4 strings:</p>

<p style="margin-top: 1em">forward Normal product, primer A
followed by B <br>
revcomp Normal product, primer B followed by A <br>
single_A Bad product generated by primer_A only <br>
single_B Bad product generated by primer_B only</p>

<p style="margin-top: 1em">There is also a human-readable
output displayed, is not designed for parsing (see: --pretty
below).</p>

<p style="margin-top: 1em">GENERAL OPTIONS <br>
Most arguments have short and long forms. The long forms
<br>
are more likely to be stable over time, and hence should be
used in scripts which call ipcress.</p>

<p style="margin-top: 1em">-h | --shorthelp &lt;boolean&gt;
<br>
Show help. This will display a concise summary of the
available options, defaults and values currently set.</p>

<p style="margin-top: 1em">--help &lt;boolean&gt; <br>
This shows all the help options including the defaults, the
value currently set, and the environment variable which may
be used to set each parameter. There will be an <br>
indication of which options are mandatory. Mandatory options
have no default, and must have a value supplied for ipcress
to run. If mandatory options are used in order, <br>
their flags may be skipped from the command line (see
examples below). Unlike this man page, the information from
this option will always be up to date with the latest <br>
version of the program.</p>

<p style="margin-top: 1em">-v | --version &lt;boolean&gt;
<br>
Display the version number. Also displays other information
such as the build date and glib version used.</p>

<p style="margin-top: 1em">FILE INPUT OPTIONS <br>
-i | --input &lt;path&gt; <br>
PCR experiment data in the ipcress file format described
above.</p>

<p style="margin-top: 1em">-s | --sequence &lt;paths&gt;
<br>
Specify the sequences. Multiple files may be specified here,
which reduces the FSM building overhead, and makes ipcress
run faster than running the process separately.</p>

<p style="margin-top: 1em">IPCRESS PARAMETERS <br>
-m | --mismatch &lt;mismatches&gt; <br>
Specify the number of mismatches allowed per primer.
Allowing mismatches reduces the speed of the program as a
large primer neighbourhood must be constructed, and fewer
<br>
experiments can be fitted in memory prior to each scan of
the sequence databases.</p>

<p style="margin-top: 1em">-M | --memory &lt;Mb&gt; <br>
Specify the amount of memory the program should use. The
more memory made available ipcress, the faster it will run,
as more PCR experiments can be conducted in each scan <br>
of the sequence databases. This does not include memory used
during the scan (for storing partial results and sequences),
so the actual amount of memory used will be <br>
slightly higher.</p>

<p style="margin-top: 1em">-p | --pretty &lt;boolean&gt;
<br>
Display results in a human-readable format, not designed for
parsing.</p>

<p style="margin-top: 1em">-P | --products &lt;boolean&gt;
<br>
Display PCR products as a FASTA format sequence.</p>

<p style="margin-top: 1em">-S | --seed &lt;length&gt; <br>
Specifiy the seed length for the wordneighbourhood for the
FSM. If set to zero, the full primer is used. Shorter words
reduce the size of the neighbourhood, but increase <br>
the time taken by ipcress to filter false positive
matches.</p>

<p style="margin-top: 1em">ENVIRONMENT <br>
Not documented yet.</p>

<p style="margin-top: 1em">EXAMPLES <br>
ipcress test.ipcress sequence.fasta <br>
This is the simplest way that ipcress can be used. <br>
ipcress dbsts_human.ipcress --sequence ncbi30/*.fasta
--mismatch 1 <br>
Compare a input file against a set of fasta files, allowing
one mismatch in each primer.</p>

<p style="margin-top: 1em">VERSION <br>
This documentation accompanies version 2.2.0 of the
exonerate package.</p>

<p style="margin-top: 1em">AUTHOR <br>
Guy St.C. Slater. &lt;guy@ebi.ac.uk&gt;.</p>

<p style="margin-top: 1em">See the AUTHORS file
accompanying the source code for a list of contributors.</p>

<p style="margin-top: 1em">AVAILABILITY <br>
This source code for the exonerate package is available
under the terms of the GNU general public licence.</p>

<p style="margin-top: 1em">Please see the file COPYING
which was distrubuted with this package, or
http://www.gnu.org/licenses/gpl.txt for details.</p>

<p style="margin-top: 1em">This package has been developed
as part of the ensembl project. Please see
http://www.ensembl.org/ for more information.</p>

<p style="margin-top: 1em">SEE ALSO <br>
exonerate(1),e-PCR</p>

<p style="margin-top: 1em">ipcress March 2003
ipcress(1)</p>
<hr>
</body>
</html>
