<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>IPDSUMMARY(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">IPDSUMMARY(1)</td>
    <td class="head-vol"></td>
    <td class="head-rtitle">IPDSUMMARY(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
ipdSummary - Detect DNA base-modifications from kinetic signatures.
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
kineticsTool loads IPDs observed at each position in the genome, and compares
  those IPDs to value expected for unmodified DNA, and outputs the result of
  this statistical test. The expected IPD value for unmodified DNA can come from
  either an <i>in-silico control</i> or an <i>amplified control</i>. The in
  silico control is trained by PacBio and shipped with the package. It predicts
  predicts the IPD using the local sequence context around the current position.
  An amplified control dataset is generated by sequencing unmodified DNA with
  the same sequence as the test sample. An amplified control sample is usually
  generated by whole-genome amplification of the original sample.
<h2 class="Ss" title="Ss" id="Modification_Detection"><a class="selflink" href="#Modification_Detection">Modification
  Detection</a></h2>
The basic mode of kineticsTools does an independent comparison of IPDs at each
  position on the genome, for each strand, and emits various statistics to CSV
  and GFF (after applying a significance filter).
<h2 class="Ss" title="Ss" id="Modifications_Identification"><a class="selflink" href="#Modifications_Identification">Modifications
  Identification</a></h2>
<div style="margin-left: 0.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>kineticsTools also has a <i>Modification
    Identification</i> mode that can decode multi-site IPD 'fingerprints' into a
    reduced set of calls of specific modifications. This feature has the
    following benefits:</b></dt>
  <dd class="It-tag"></dd>
</dl>
<div style="margin-left: 7.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Different modifications occurring on the same base can be
      distinguished (for example m5C and m4C)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">The signal from one modification is combined into one
      statistic, improving sensitivity, removing extra peaks, and correctly
      centering the call</dd>
</dl>
</div>
<br/>
</div>
<br/>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
Please call this program with <b>--help</b> to see the available options.
<h1 class="Sh" title="Sh" id="ALGORITHM"><a class="selflink" href="#ALGORITHM">ALGORITHM</a></h1>
<h2 class="Ss" title="Ss" id="Synthetic_Control"><a class="selflink" href="#Synthetic_Control">Synthetic
  Control</a></h2>
Studies of the relationship between IPD and sequence context reveal that most of
  the variation in mean IPD across a genome can be predicted from a 12-base
  sequence context surrounding the active site of the DNA polymerase. The bounds
  of the relevant context window correspond to the window of DNA in contact with
  the polymerase, as seen in DNA/polymerase crystal structures. To simplify the
  process of finding DNA modifications with PacBio data, the tool includes a
  pre-trained lookup table mapping 12-mer DNA sequences to mean IPDs observed in
  C2 chemistry.
<h2 class="Ss" title="Ss" id="Filtering_and_Trimming"><a class="selflink" href="#Filtering_and_Trimming">Filtering
  and Trimming</a></h2>
kineticsTools uses the Mapping QV generated by BLASR and stored in the cmp.h5 or
  BAM file (or AlignmentSet) to ignore reads that aren't confidently mapped. The
  default minimum Mapping QV required is 10, implying that BLASR has <b>90\%</b>
  confidence that the read is correctly mapped. Because of the range of
  readlengths inherent in PacBio dataThis can be changed in using the
  --mapQvThreshold command line argument, or via the SMRTPortal configuration
  dialog for Modification Detection.
<div style="height: 1.00em;">&#x00A0;</div>
There are a few features of PacBio data that require special attention in order
  to achieve good modification detection performance. kineticsTools inspects the
  alignment between the observed bases and the reference sequence -- in order
  for an IPD measurement to be included in the analysis, the PacBio read
  sequence must match the reference sequence for <b>k</b> around the cognate
  base. In the current module <b>k=1</b> The IPD distribution at some locus be
  thought of as a mixture between the 'normal' incorporation process IPD, which
  is sensitive to the local sequence context and DNA modifications and a
  contaminating 'pause' process IPD which have a much longer duration (mean
  &gt;10x longer than normal), but happen rarely (~1% of IPDs). Note: Our
  current understanding is that pauses do not carry useful information about the
  methylation state of the DNA, however a more careful analysis may be
  warranted. Also note that modifications that drastically increase the Roughly
  1% of observed IPDs are generated by pause events. Capping observed IPDs at
  the global 99th percentile is motivated by theory from robust hypothesis
  testing. Some sequence contexts may have naturally longer IPDs, to avoid
  capping too much data at those contexts, the cap threshold is adjusted per
  context as follows: capThreshold = max(global99, 5*modelPrediction,
  percentile(ipdObservations, 75))
<h2 class="Ss" title="Ss" id="Statistical_Testing"><a class="selflink" href="#Statistical_Testing">Statistical
  Testing</a></h2>
We test the hypothesis that IPDs observed at a particular locus in the sample
  have a longer means than IPDs observed at the same locus in unmodified DNA. If
  we have generated a Whole Genome Amplified dataset, which removes DNA
  modifications, we use a case-control, two-sample t-test. This tool also
  provides a pre-calibrated 'synthetic control' model which predicts the
  unmodified IPD, given a 12 base sequence context. In the synthetic control
  case we use a one-sample t-test, with an adjustment to account for error in
  the synthetic control model.
<h1 class="Sh" title="Sh" id="EXAMPLE_USAGE"><a class="selflink" href="#EXAMPLE_USAGE">EXAMPLE
  USAGE</a></h1>
Basic use with BAM input, GFF+HDF5 output:
<div style="margin-left: 0.00ex;">
<div style="margin-left: 3.50ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<pre>
ipdSummary aligned.bam --reference ref.fasta --identify m6A,m4C --gff basemods.gff --csv_h5 kinetics.h5
</pre>
</div>
<br/>
</div>
<br/>
<div style="height: 1.00em;">&#x00A0;</div>
With cmp.h5 input, methyl fraction calculation and GFF+CSV output:
<div style="margin-left: 0.00ex;">
<div style="margin-left: 3.50ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<pre>
ipdSummary aligned.cmp.h5 --reference ref.fasta --identify m6A,m4C --methylFraction --gff basemods.gff --csv kinetics.csv
</pre>
</div>
<br/>
</div>
<br/>
<h1 class="Sh" title="Sh" id="INPUTS"><a class="selflink" href="#INPUTS">INPUTS</a></h1>
<h2 class="Ss" title="Ss" id="Aligned_Reads"><a class="selflink" href="#Aligned_Reads">Aligned
  Reads</a></h2>
A standard PacBio alignment file - either AlignmentSet XML, BAM, or cmp.h5 -
  containing alignments and IPD information supplies the kinetic data used to
  perform modification detection. The standard cmp.h5 file of a SMRTportal jobs
  is data/aligned_read.cmp.h5.
<h2 class="Ss" title="Ss" id="Reference_Sequence"><a class="selflink" href="#Reference_Sequence">Reference
  Sequence</a></h2>
The tool requires the reference sequence used to perform alignments. This can be
  either a FASTA file or a ReferenceSet XML.
<h1 class="Sh" title="Sh" id="OUTPUTS"><a class="selflink" href="#OUTPUTS">OUTPUTS</a></h1>
The modification detection tool provides results in a variety of formats
  suitable for in-depth statistical analysis, quick reference, and comsumption
  by visualization tools such as PacBio SMRTView. Results are generally indexed
  by reference position and reference strand. In all cases the strand value
  refers to the strand carrying the modification in DNA sample. Remember that
  the kinetic effect of the modification is observed in read sequences aligning
  to the opposite strand. So reads aligning to the positive strand carry
  information about modification on the negative strand and vice versa, but in
  this toolkit we alway report the strand containing the putative modification.
<div style="height: 1.00em;">&#x00A0;</div>
The following output options are available:
<div style="margin-left: 0.00ex;">
<div style="margin-left: 3.50ex;">
<div style="margin-left: 0.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag"><b>--gff FILENAME</b>: GFF format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag"><b>--csv FILENAME</b>: comma-separated value format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag"><b>--csv_h5 FILENAME</b>: compact binary equivalent of CSV
      in HDF5 format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag"><b>--bigwig FILENAME</b>: BigWig file (mostly only useful
      for SMRTView)</dd>
</dl>
</div>
<br/>
</div>
<br/>
</div>
<br/>
<h2 class="Ss" title="Ss" id="modifications.gff"><a class="selflink" href="#modifications.gff">modifications.gff</a></h2>
The modifications.gff is compliant with the GFF Version 3 specification (
  <i>http://www.sequenceontology.org/gff3.shtml</i>). Each template position /
  strand pair whose p-value exceeds the pvalue threshold appears as a row. The
  template position is 1-based, per the GFF spec. The strand column refers to
  the strand carrying the detected modification, which is the opposite strand
  from those used to detect the modification. The GFF confidence column is a
  Phred-transformed pvalue of detection.
<div style="height: 1.00em;">&#x00A0;</div>
<b>Note on genome browser compatibility</b>
<div style="height: 1.00em;">&#x00A0;</div>
The modifications.gff file will not work directly with most genome browsers. You
  will likely need to make a copy of the GFF file and convert the _seqid_
  columns from the generic 'ref0000x' names generated by PacBio, to the FASTA
  headers present in the original reference FASTA file. The mapping table is
  written in the header of the modifications.gff file in <b>#sequence-header</b>
  tags. This issue will be resolved in the 1.4 release of kineticsTools.
<div style="height: 1.00em;">&#x00A0;</div>
The auxiliary data column of the GFF file contains other statistics which may be
  useful downstream analysis or filtering. In particular the coverage level of
  the reads used to make the call, and +/- 20bp sequence context surrounding the
  site.
<dl class="Bl-tag">
  <dt class="It-tag">System Message: ERROR/3 (doc/manual.rst:, line 114)</dt>
  <dd class="It-tag">Malformed table. Text in column margin in table line
    2.</dd>
</dl>
<div style="margin-left: 0.00ex;">
<div style="margin-left: 3.50ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<pre>
================  ===========
Column      Description
================  ===========
seqid     Fasta contig name
source            Name of tool -- 'kinModCall'
type                    Modification type -- in identification mode this will be m6A, m4C, or m5C for identified bases, or the generic tag 'modified_base' if a kinetic event was detected that does not match a known modification signature
start                   Modification position on contig
end                     Modification position on contig
score                   Phred transformed p-value of detection - this is the single-site detection p-value
strand                  Sample strand containing modification
phase                   Not applicable
attributes              Extra fields relevant to base mods. IPDRatio is traditional IPDRatio, context is the reference sequence -20bp to +20bp around the modification, and coverage level is the number of IPD observations used after Mapping QV filtering and accuracy filtering. If the row results from an identified modification we also include an identificationQv tag with the from the modification identification procedure. identificationQv is the phred-transformed probability of an incorrect identification, for bases that were identified as having a particular modification. frac, fracLow, fracUp are the estimated fraction of molecules carrying the modification, and the 5% confidence intervals of the estimate. The methylated fraction estimation is a beta-level feature, and should only be used for exploratory purposes.
================  ===========
</pre>
</div>
<br/>
</div>
<br/>
<h2 class="Ss" title="Ss" id="modifications.csv"><a class="selflink" href="#modifications.csv">modifications.csv</a></h2>
The modifications.csv file contains one row for each (reference position,
  strand) pair that appeared in the dataset with coverage at least x. x defaults
  to 3, but is configurable with '--minCoverage' flag to ipdSummary.py. The
  reference position index is 1-based for compatibility with the gff file the R
  environment. Note that this output type scales poorly and is not recommended
  for large genomes; the HDF5 output should perform much better in these cases.
<h2 class="Ss" title="Ss" id="Output_columns"><a class="selflink" href="#Output_columns">Output
  columns</a></h2>
<b>in-silico control mode</b>
<table class="tbl">
  <colgroup>
    <col style="width: 15.00ex;"/>
    <col style="width: 138.00ex;"/>
  </colgroup>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>Column</td>
    <td>Description</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>refId</td>
    <td>reference sequence ID of this observation</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>tpl</td>
    <td>1-based template position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>strand</td>
    <td>native sample strand where kinetics were generated. '0' is the strand of
      the original FASTA, '1' is opposite strand from FASTA</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>base</td>
    <td>the cognate base at this position in the reference</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>score</td>
    <td>Phred-transformed pvalue that a kinetic deviation exists at this
      position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>tMean</td>
    <td>capped mean of normalized IPDs observed at this position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>tErr</td>
    <td>capped standard error of normalized IPDs observed at this position
      (standard deviation / sqrt(coverage)</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>modelPrediction</td>
    <td>normalized mean IPD predicted by the synthetic control model for this
      sequence context</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>ipdRatio</td>
    <td>tMean / modelPrediction</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>coverage</td>
    <td>count of valid IPDs at this position (see Filtering section for
      details)</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>frac</td>
    <td>estimate of the fraction of molecules that carry the modification</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>fracLow</td>
    <td>2.5% confidence bound of frac estimate</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>fracUpp</td>
    <td>97.5% confidence bound of frac estimate</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
</table>
<div style="height: 1.00em;">&#x00A0;</div>
<b>case-control mode</b>
<table class="tbl">
  <colgroup>
    <col style="width: 15.00ex;"/>
    <col style="width: 138.00ex;"/>
  </colgroup>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>Column</td>
    <td>Description</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>refId</td>
    <td>reference sequence ID of this observation</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>tpl</td>
    <td>1-based template position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>strand</td>
    <td>native sample strand where kinetics were generated. '0' is the strand of
      the original FASTA, '1' is opposite strand from FASTA</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>base</td>
    <td>the cognate base at this position in the reference</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>score</td>
    <td>Phred-transformed pvalue that a kinetic deviation exists at this
      position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>caseMean</td>
    <td>mean of normalized case IPDs observed at this position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>controlMean</td>
    <td>mean of normalized control IPDs observed at this position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>caseStd</td>
    <td>standard deviation of case IPDs observed at this position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>controlStd</td>
    <td>standard deviation of control IPDs observed at this position</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>ipdRatio</td>
    <td>tMean / modelPrediction</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>testStatistic</td>
    <td>t-test statistic</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>coverage</td>
    <td>mean of case and control coverage</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>controlCoverage</td>
    <td>count of valid control IPDs at this position (see Filtering section for
      details)</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>caseCoverage</td>
    <td>count of valid case IPDs at this position (see Filtering section for
      details)</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
</table>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>summarizeModifications</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">December 2015</td>
    <td class="foot-os"></td>
  </tr>
</table>
</body>
</html>
