<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>IPIG(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">IPIG(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">IPIG(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
ipig - Integrating PSMs Into Genome browser visualizations
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>ipig</b> <i>&lt;psm file&gt;</i> |-g|-c|-cg [&lt;config file&gt;]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
iPiG targets the integration of peptide spectrum matches (PSMs) from mass
  spectrometry (MS) peptide identifications into genomic visualisations provided
  by genome browser such as the UCSC genome browser (http://genome.ucsc.edu/).
<div class="Pp"></div>
iPiG takes PSMs from the MS standard format mzIdentML (*.mzid) or in text format
  and provides results in genome track formats (BED and GFF3 files), which can
  be easily imported into genome browsers.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><i>&lt;psm file&gt;</i></dt>
  <dd class="It-tag">indicates the file with the peptide spectrum matches
      (mzid/txt)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b>, <b>-gui</b></dt>
  <dd class="It-tag">starts the graphical user interface of iPiG</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>, <b>-control</b></dt>
  <dd class="It-tag">starts the gene control, necessary files have to be
      indicated in the configuration file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cg</b>, <b>-controlgui</b></dt>
  <dd class="It-tag">starts the graphical user interface of the gene
    control</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b>, <b>-downloader</b></dt>
  <dd class="It-tag">starts the download gui</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&lt;config file&gt;</dt>
  <dd class="It-tag">a different configuration file can be indicated (otherwise
      ipig.conf is loaded by default)</dd>
</dl>
<div class="Pp"></div>
additional requirements:
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">using a non-gui mode, a config file (ipig.conf by default)
      has to contain several additional parameters, e.g. indicating the
      reference genome etc.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">in a gui mode (<b>-g</b> and <b>-cg</b>), additional
      parameters can be indicated two ways, within the interface or with a
      config file as well.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">have a look into readme.txt and ipig.conf for examples and
      more details about the additional parameters</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2014</td>
    <td class="foot-os">ipig 2012-06-19</td>
  </tr>
</table>
</body>
</html>
