<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:19:01 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>IQTREE(1) User Commands IQTREE(1)</p>

<p style="margin-top: 1em">NAME <br>
iqtree - efficient phylogenetic software by maximum
likelihood</p>

<p style="margin-top: 1em">SYNOPSIS <br>
iqtree -s &lt;alignment&gt; [OPTIONS]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
IQ-TREE version 1.5.3 for Linux 64-bit built Feb 11 2017
Copyright &Acirc;&copy; 2011-2016 Nguyen Lam Tung, Olga
Chernomor, Arndt von Haeseler and Bui Quang Minh.</p>

<p style="margin-top: 1em">GENERAL OPTIONS: <br>
-? or -h <br>
Printing this help dialog</p>

<p style="margin-top: 1em">-s &lt;alignment&gt; <br>
Input alignment in PHYLIP/FASTA/NEXUS/CLUSTAL/MSF format</p>

<p style="margin-top: 1em">-st &lt;data_type&gt; <br>
BIN, DNA, AA, NT2AA, CODON, MORPH (default: auto-detect)</p>

<p style="margin-top: 1em">-q &lt;partition_file&gt; <br>
Edge-linked partition model (file in NEXUS/RAxML format)</p>

<p style="margin-top: 1em">-spp &lt;partition_file&gt; Like
-q option but allowing partition-specific rates</p>

<p style="margin-top: 1em">-sp &lt;partition_file&gt;
Edge-unlinked partition model (like -M option of RAxML)</p>

<p style="margin-top: 1em">-t &lt;start_tree_file&gt; or -t
BIONJ or -t RANDOM <br>
Starting tree (default: 99 parsimony tree and BIONJ)</p>

<p style="margin-top: 1em">-te &lt;user_tree_file&gt; Like
-t but fixing user tree (no tree search performed)</p>

<p style="margin-top: 1em">-o &lt;outgroup_taxon&gt; <br>
Outgroup taxon name for writing .treefile</p>

<p style="margin-top: 1em">-pre &lt;PREFIX&gt; <br>
Using &lt;PREFIX&gt; for output files (default:
aln/partition)</p>

<p style="margin-top: 1em">-seed &lt;number&gt; <br>
Random seed number, normally used for debugging purpose</p>

<p style="margin-top: 1em">-v, -vv, -vvv <br>
Verbose mode, printing more messages to screen</p>

<p style="margin-top: 1em">-quiet Silent mode, suppress
printing to screen (stdout)</p>

<p style="margin-top: 1em">-keep-ident <br>
Keep identical sequences (default: remove &amp; finally
add)</p>

<p style="margin-top: 1em">-safe Safe likelihood kernel to
avoid numerical underflow</p>

<p style="margin-top: 1em">-mem RAM <br>
Maximal RAM usage for memory saving mode</p>

<p style="margin-top: 1em">CHECKPOINTING TO RESUME STOPPED
RUN: <br>
-redo Redo analysis even for successful runs (default:
resume)</p>

<p style="margin-top: 1em">-cptime &lt;seconds&gt; <br>
Minimum checkpoint time interval (default: 20)</p>

<p style="margin-top: 1em">LIKELIHOOD MAPPING ANALYSIS:
<br>
-lmap &lt;#quartets&gt; <br>
Number of quartets for likelihood mapping analysis</p>

<p style="margin-top: 1em">-lmclust &lt;clustfile&gt; NEXUS
file containing clusters for likelihood mapping</p>

<p style="margin-top: 1em">-wql Print quartet
log-likelihoods to .quartetlh file</p>

<p style="margin-top: 1em">NEW STOCHASTIC TREE SEARCH
ALGORITHM: <br>
-ninit &lt;number&gt; <br>
Number of initial parsimony trees (default: 100)</p>

<p style="margin-top: 1em">-ntop &lt;number&gt; <br>
Number of top initial trees (default: 20)</p>

<p style="margin-top: 1em">-nbest &lt;number&gt; <br>
Number of best trees retained during search (defaut: 5)</p>

<p style="margin-top: 1em">-n &lt;#iterations&gt; <br>
Fix number of iterations to &lt;#iterations&gt; (default:
auto)</p>

<p style="margin-top: 1em">-nstop &lt;number&gt; <br>
Number of unsuccessful iterations to stop (default: 100)</p>

<p style="margin-top: 1em">-pers &lt;proportion&gt; <br>
Perturbation strength for randomized NNI (default: 0.5)</p>

<p style="margin-top: 1em">-sprrad &lt;number&gt; <br>
Radius for parsimony SPR search (default: 6)</p>

<p style="margin-top: 1em">-allnni <br>
Perform more thorough NNI search (default: off)</p>

<p style="margin-top: 1em">-g &lt;constraint_tree&gt;
(Multifurcating) topological constraint tree file</p>

<p style="margin-top: 1em">ULTRAFAST BOOTSTRAP: <br>
-bb &lt;#replicates&gt; <br>
Ultrafast bootstrap (&gt;=1000)</p>

<p style="margin-top: 1em">-wbt Write bootstrap trees to
.ufboot file (default: none)</p>

<p style="margin-top: 1em">-wbtl Like -wbt but also writing
branch lengths</p>

<p style="margin-top: 1em">-nm &lt;#iterations&gt; <br>
Maximum number of iterations (default: 1000)</p>

<p style="margin-top: 1em">-nstep &lt;#iterations&gt;
#Iterations for UFBoot stopping rule (default: 100)</p>

<p style="margin-top: 1em">-bcor &lt;min_corr&gt; <br>
Minimum correlation coefficient (default: 0.99)</p>

<p style="margin-top: 1em">-beps &lt;epsilon&gt; <br>
RELL epsilon to break tie (default: 0.5)</p>

<p style="margin-top: 1em">STANDARD NON-PARAMETRIC
BOOTSTRAP: <br>
-b &lt;#replicates&gt; <br>
Bootstrap + ML tree + consensus tree (&gt;=100)</p>

<p style="margin-top: 1em">-bc &lt;#replicates&gt; <br>
Bootstrap + consensus tree</p>

<p style="margin-top: 1em">-bo &lt;#replicates&gt; <br>
Bootstrap only</p>

<p style="margin-top: 1em">SINGLE BRANCH TEST: <br>
-alrt &lt;#replicates&gt; <br>
SH-like approximate likelihood ratio test (SH-aLRT)</p>

<p style="margin-top: 1em">-alrt 0 <br>
Parametric aLRT test (Anisimova and Gascuel 2006)</p>

<p style="margin-top: 1em">-abayes <br>
approximate Bayes test (Anisimova et al. 2011)</p>

<p style="margin-top: 1em">-lbp &lt;#replicates&gt; <br>
Fast local bootstrap probabilities</p>

<p style="margin-top: 1em">AUTOMATIC MODEL SELECTION: <br>
-m TESTONLY <br>
Standard model selection (like jModelTest, ProtTest)</p>

<p style="margin-top: 1em">-m TEST <br>
Like -m TESTONLY but followed by tree reconstruction</p>

<p style="margin-top: 1em">-m TESTNEWONLY <br>
Extended model selection incl. FreeRate (+R)
heterogeneity</p>

<p style="margin-top: 1em">-m TESTNEW <br>
Like -m TESTNEWONLY but followed by tree reconstruction</p>

<p style="margin-top: 1em">-m TESTMERGEONLY <br>
Select best-fit partition scheme (like PartitionFinder)</p>

<p style="margin-top: 1em">-m TESTMERGE <br>
Like -m TESTMERGEONLY but followed by tree
reconstruction</p>

<p style="margin-top: 1em">-m TESTNEWMERGEONLY <br>
Like -m TESTMERGEONLY but includes FreeRate
heterogeneity</p>

<p style="margin-top: 1em">-m TESTNEWMERGE <br>
Like -m TESTNEWMERGEONLY followed by tree reconstruction</p>

<p style="margin-top: 1em">-rcluster &lt;percent&gt; <br>
Percentage of partition pairs (relaxed clustering alg.)</p>

<p style="margin-top: 1em">-mset program <br>
Restrict search to models supported by other programs (i.e.,
raxml, phyml or mrbayes)</p>

<p style="margin-top: 1em">-mset m1,...,mk <br>
Restrict search to models in a comma-separated list (e.g.
-mset WAG,LG,JTT)</p>

<p style="margin-top: 1em">-msub source <br>
Restrict search to AA models designed for specific sources
(i.e., nuclear, mitochondrial, chloroplast or viral)</p>

<p style="margin-top: 1em">-mfreq f1,...,fk <br>
Restrict search to using a list of state frequencies
(default protein: -mfreq FU,F; codon: -mfreq
,F1x4,F3x4,F)</p>

<p style="margin-top: 1em">-mrate r1,...,rk <br>
Restrict search to using a list of rate-across-sites models
(e.g. -mrate E,I,G,I+G,R)</p>

<p style="margin-top: 1em">-cmin &lt;kmin&gt; <br>
Min #categories for FreeRate model [+R] (default: 2)</p>

<p style="margin-top: 1em">-cmax &lt;kmax&gt; <br>
Max #categories for FreeRate model [+R] (default: 10)</p>

<p style="margin-top: 1em">???merit AIC|AICc|BIC <br>
Optimality criterion to use (default: all)</p>

<p style="margin-top: 1em">-mtree Performing full tree
search for each model considered</p>

<p style="margin-top: 1em">-mredo Ignoring model results
computed earlier (default: no)</p>

<p style="margin-top: 1em">-madd mx1,...,mxk <br>
List of mixture models to also consider</p>

<p style="margin-top: 1em">-mdef &lt;nexus_file&gt; <br>
A model definition NEXUS file (see Manual)</p>

<p style="margin-top: 1em">SUBSTITUTION MODEL: <br>
-m &lt;model_name&gt;</p>

<p style="margin-top: 1em">DNA: HKY (default), JC, F81,
K2P, K3P, K81uf, TN/TrN, TNef, <br>
TIM, TIMef, TVM, TVMef, SYM, GTR, or 6-digit model
specification (e.g., 010010 = HKY)</p>

<p style="margin-top: 1em">Protein: LG (default), Poisson,
cpREV, mtREV, Dayhoff, mtMAM, <br>
JTT, WAG, mtART, mtZOA, VT, rtREV, DCMut, PMB, HIVb, HIVw,
JTTDCMut, FLU, Blosum62, GTR20</p>

<p style="margin-top: 1em">Protein mixture: C10,...,C60,
EX2, EX3, EHO, UL2, UL3, EX_EHO, LG4M, LG4X, <br>
JTTCF4G</p>

<p style="margin-top: 1em">Binary: JC2 (default), GTR2</p>

<p style="margin-top: 1em">Empirical codon: KOSI07,
SCHN05</p>

<p style="margin-top: 1em">Mechanistic codon: GY (default),
MG, MGK, GY0K, GY1KTS, GY1KTV, GY2K, <br>
MG1KTS, MG1KTV, MG2K</p>

<p style="margin-top: 1em">Semi-empirical codon: XX_YY
where XX is empirical and YY is mechanistic model</p>

<p style="margin-top: 1em">Morphology/SNP: MK (default),
ORDERED</p>

<p style="margin-top: 1em">Otherwise: Name of file
containing user-model parameters <br>
(rate parameters and state frequencies)</p>

<p style="margin-top: 1em">-m &lt;model_name&gt;+F or +FO
or +FU or +FQ (default: auto) <br>
counted, optimized, user-defined, equal state frequency</p>

<p style="margin-top: 1em">-m &lt;model_name&gt;+F1x4 or
+F3x4 <br>
Codon frequencies</p>

<p style="margin-top: 1em">-m &lt;model_name&gt;+ASC <br>
Ascertainment bias correction for morphological/SNP data</p>

<p style="margin-top: 1em">-m &quot;MIX{m1,...mK}&quot;
<br>
Mixture model with K components</p>

<p style="margin-top: 1em">-m &quot;FMIX{f1,...fK}&quot;
<br>
Frequency mixture model with K components</p>

<p style="margin-top: 1em">-mwopt Turn on optimizing
mixture weights (default: none)</p>

<p style="margin-top: 1em">RATE HETEROGENEITY AMONG SITES:
<br>
-m &lt;model_name&gt;+I or +G[n] or +I+G[n] or +R[n] <br>
Invar, Gamma, Invar+Gamma, or FreeRate model where
&rsquo;n&rsquo; is number of categories (default: n=4)</p>

<p style="margin-top: 1em">-a &lt;Gamma_shape&gt; <br>
Gamma shape parameter for site rates (default: estimate)</p>

<p style="margin-top: 1em">-amin &lt;min_shape&gt; <br>
Min Gamma shape parameter for site rates (default: 0.02)</p>

<p style="margin-top: 1em">-gmedian <br>
Median approximation for +G site rates (default: mean)</p>

<p style="margin-top: 1em">--opt-gamma-inv <br>
More thorough estimation for +I+G model parameters</p>

<p style="margin-top: 1em">-i &lt;p_invar&gt; <br>
Proportion of invariable sites (default: estimate)</p>

<p style="margin-top: 1em">-wsr Write site rates to .rate
file</p>

<p style="margin-top: 1em">-mh Computing site-specific
rates to .mhrate file using Meyer &amp; von Haeseler (2003)
method</p>

<p style="margin-top: 1em">SITE-SPECIFIC FREQUENCY MODEL:
<br>
-ft &lt;tree_file&gt; <br>
Input tree to infer site frequency model</p>

<p style="margin-top: 1em">-fs &lt;in_freq_file&gt; <br>
Input site frequency model file</p>

<p style="margin-top: 1em">-fmax Posterior maximum instead
of posterior mean approximation</p>

<p style="margin-top: 1em">CONSENSUS RECONSTRUCTION: <br>
-t &lt;tree_file&gt; <br>
Set of input trees for consensus reconstruction</p>

<p style="margin-top: 1em">-minsup &lt;threshold&gt; <br>
Min split support in range [0,1]; 0.5 for majority-rule
consensus (default: 0, i.e. extended consensus)</p>

<p style="margin-top: 1em">-bi &lt;burnin&gt; <br>
Discarding &lt;burnin&gt; trees at beginning of
&lt;treefile&gt;</p>

<p style="margin-top: 1em">-con Computing consensus tree to
.contree file</p>

<p style="margin-top: 1em">-net Computing consensus network
to .nex file</p>

<p style="margin-top: 1em">-sup &lt;target_tree&gt; <br>
Assigning support values for &lt;target_tree&gt; to
.suptree</p>

<p style="margin-top: 1em">-suptag &lt;name&gt; <br>
Node name (or ALL) to assign tree IDs where node occurs</p>

<p style="margin-top: 1em">ROBINSON-FOULDS DISTANCE: <br>
-rf_all <br>
Computing all-to-all RF distances of trees in
&lt;treefile&gt;</p>

<p style="margin-top: 1em">-rf &lt;treefile2&gt; <br>
Computing all RF distances between two sets of trees stored
in &lt;treefile&gt; and &lt;treefile2&gt;</p>

<p style="margin-top: 1em">-rf_adj <br>
Computing RF distances of adjacent trees in
&lt;treefile&gt;</p>

<p style="margin-top: 1em">TREE TOPOLOGY TEST: <br>
-z &lt;trees_file&gt; <br>
Evaluating a set of user trees</p>

<p style="margin-top: 1em">-zb &lt;#replicates&gt; <br>
Performing BP,KH,SH,ELW tests for trees passed via -z</p>

<p style="margin-top: 1em">-zw Also performing weighted-KH
and weighted-SH tests</p>

<p style="margin-top: 1em">-au Also performing
approximately unbiased (AU) test</p>

<p style="margin-top: 1em">GENERATING RANDOM TREES: <br>
-r &lt;num_taxa&gt; <br>
Create a random tree under Yule-Harding model</p>

<p style="margin-top: 1em">-ru &lt;num_taxa&gt; <br>
Create a random tree under Uniform model</p>

<p style="margin-top: 1em">-rcat &lt;num_taxa&gt; <br>
Create a random caterpillar tree</p>

<p style="margin-top: 1em">-rbal &lt;num_taxa&gt; <br>
Create a random balanced tree</p>

<p style="margin-top: 1em">-rcsg &lt;num_taxa&gt; <br>
Create a random circular split network</p>

<p style="margin-top: 1em">-rlen &lt;min_len&gt;
&lt;mean_len&gt; &lt;max_len&gt; <br>
min, mean, and max branch lengths of random trees</p>

<p style="margin-top: 1em">MISCELLANEOUS: <br>
-wt Write locally optimal trees into .treels file</p>

<p style="margin-top: 1em">-blfix Fix branch lengths of
user tree passed via -te</p>

<p style="margin-top: 1em">-blscale <br>
Scale branch lengths of user tree passed via -t</p>

<p style="margin-top: 1em">-blmin Min branch length for
optimization (default 0.000001)</p>

<p style="margin-top: 1em">-blmax Max branch length for
optimization (default 100)</p>

<p style="margin-top: 1em">-wsr Write site rates and
categories to .rate file</p>

<p style="margin-top: 1em">-wsl Write site log-likelihoods
to .sitelh file</p>

<p style="margin-top: 1em">-wslr Write site log-likelihoods
per rate category</p>

<p style="margin-top: 1em">-wslm Write site log-likelihoods
per mixture class</p>

<p style="margin-top: 1em">-wslmr Write site
log-likelihoods per mixture+rate class</p>

<p style="margin-top: 1em">-wspr Write site probabilities
per rate category</p>

<p style="margin-top: 1em">-wspm Write site probabilities
per mixture class</p>

<p style="margin-top: 1em">-wspmr Write site probabilities
per mixture+rate class</p>

<p style="margin-top: 1em">-wpl Write partition
log-likelihoods to .partlh file</p>

<p style="margin-top: 1em">-fconst f1,...,fN <br>
Add constant patterns into alignment (N=#nstates)</p>

<p style="margin-top: 1em">-me &lt;epsilon&gt; <br>
Logl epsilon for model parameter optimization (default
0.01)</p>

<p style="margin-top: 1em">--no-outfiles <br>
Suppress printing output files</p>

<p style="margin-top: 1em">iqtree 1.5.3+dfsg February 2017
IQTREE(1)</p>
<hr>
</body>
</html>
