<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>IQTREE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">IQTREE(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">IQTREE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
iqtree - efficient phylogenetic software by maximum likelihood
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>iqtree</b> <i>-s &lt;alignment&gt; </i>[<i>OPTIONS</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
IQ-TREE version 1.5.3 for Linux 64-bit built Feb 11 2017 Copyright &#x00A9;
  2011-2016 Nguyen Lam Tung, Olga Chernomor, Arndt von Haeseler and Bui Quang
  Minh.
<h2 class="Ss" title="Ss" id="GENERAL_OPTIONS:"><a class="selflink" href="#GENERAL_OPTIONS:">GENERAL
  OPTIONS:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">-? or <b>-h</b></dt>
  <dd class="It-tag">Printing this help dialog</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> &lt;alignment&gt;</dt>
  <dd class="It-tag">Input alignment in PHYLIP/FASTA/NEXUS/CLUSTAL/MSF
    format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-st</b> &lt;data_type&gt;</dt>
  <dd class="It-tag">BIN, DNA, AA, NT2AA, CODON, MORPH (default:
    auto-detect)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> &lt;partition_file&gt;</dt>
  <dd class="It-tag">Edge-linked partition model (file in NEXUS/RAxML
    format)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-spp</b>
  &lt;partition_file&gt; Like <b>-q</b> option but allowing partition-specific
  rates</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-sp</b>
  &lt;partition_file&gt; Edge-unlinked partition model (like <b>-M</b> option of
  RAxML)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> &lt;start_tree_file&gt; or <b>-t</b> BIONJ or
    <b>-t</b> RANDOM</dt>
  <dd class="It-tag">Starting tree (default: 99 parsimony tree and BIONJ)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-te</b>
  &lt;user_tree_file&gt; Like <b>-t</b> but fixing user tree (no tree search
  performed)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> &lt;outgroup_taxon&gt;</dt>
  <dd class="It-tag">Outgroup taxon name for writing .treefile</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pre</b> &lt;PREFIX&gt;</dt>
  <dd class="It-tag">Using &lt;PREFIX&gt; for output files (default:
      aln/partition)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-seed</b> &lt;number&gt;</dt>
  <dd class="It-tag">Random seed number, normally used for debugging
    purpose</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>-vv</b>, <b>-vvv</b></dt>
  <dd class="It-tag">Verbose mode, printing more messages to screen</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-quiet</b></dt>
  <dd class="It-tag">Silent mode, suppress printing to screen (stdout)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-keep-ident</b></dt>
  <dd class="It-tag">Keep identical sequences (default: remove &amp; finally
      add)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-safe</b></dt>
  <dd class="It-tag">Safe likelihood kernel to avoid numerical underflow</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mem</b> RAM</dt>
  <dd class="It-tag">Maximal RAM usage for memory saving mode</dd>
</dl>
<h2 class="Ss" title="Ss" id="CHECKPOINTING_TO_RESUME_STOPPED_RUN:"><a class="selflink" href="#CHECKPOINTING_TO_RESUME_STOPPED_RUN:">CHECKPOINTING
  TO RESUME STOPPED RUN:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-redo</b></dt>
  <dd class="It-tag">Redo analysis even for successful runs (default:
    resume)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cptime</b> &lt;seconds&gt;</dt>
  <dd class="It-tag">Minimum checkpoint time interval (default: 20)</dd>
</dl>
<h2 class="Ss" title="Ss" id="LIKELIHOOD_MAPPING_ANALYSIS:"><a class="selflink" href="#LIKELIHOOD_MAPPING_ANALYSIS:">LIKELIHOOD
  MAPPING ANALYSIS:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-lmap</b> &lt;#quartets&gt;</dt>
  <dd class="It-tag">Number of quartets for likelihood mapping analysis</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-lmclust</b>
  &lt;clustfile&gt; NEXUS file containing clusters for likelihood mapping</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wql</b></dt>
  <dd class="It-tag">Print quartet log-likelihoods to .quartetlh file</dd>
</dl>
<h2 class="Ss" title="Ss" id="NEW_STOCHASTIC_TREE_SEARCH_ALGORITHM:"><a class="selflink" href="#NEW_STOCHASTIC_TREE_SEARCH_ALGORITHM:">NEW
  STOCHASTIC TREE SEARCH ALGORITHM:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ninit</b> &lt;number&gt;</dt>
  <dd class="It-tag">Number of initial parsimony trees (default: 100)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ntop</b> &lt;number&gt;</dt>
  <dd class="It-tag">Number of top initial trees (default: 20)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nbest</b> &lt;number&gt;</dt>
  <dd class="It-tag">Number of best trees retained during search (defaut:
    5)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> &lt;#iterations&gt;</dt>
  <dd class="It-tag">Fix number of iterations to &lt;#iterations&gt; (default:
      auto)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nstop</b> &lt;number&gt;</dt>
  <dd class="It-tag">Number of unsuccessful iterations to stop (default:
    100)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pers</b> &lt;proportion&gt;</dt>
  <dd class="It-tag">Perturbation strength for randomized NNI (default:
    0.5)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sprrad</b> &lt;number&gt;</dt>
  <dd class="It-tag">Radius for parsimony SPR search (default: 6)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-allnni</b></dt>
  <dd class="It-tag">Perform more thorough NNI search (default: off)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-g</b>
  &lt;constraint_tree&gt; (Multifurcating) topological constraint tree
  file</div>
<h2 class="Ss" title="Ss" id="ULTRAFAST_BOOTSTRAP:"><a class="selflink" href="#ULTRAFAST_BOOTSTRAP:">ULTRAFAST
  BOOTSTRAP:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-bb</b> &lt;#replicates&gt;</dt>
  <dd class="It-tag">Ultrafast bootstrap (&gt;=1000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wbt</b></dt>
  <dd class="It-tag">Write bootstrap trees to .ufboot file (default: none)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wbtl</b></dt>
  <dd class="It-tag">Like <b>-wbt</b> but also writing branch lengths</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nm</b> &lt;#iterations&gt;</dt>
  <dd class="It-tag">Maximum number of iterations (default: 1000)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-nstep</b>
  &lt;#iterations&gt; #Iterations for UFBoot stopping rule (default: 100)</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-bcor</b> &lt;min_corr&gt;</dt>
  <dd class="It-tag">Minimum correlation coefficient (default: 0.99)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-beps</b> &lt;epsilon&gt;</dt>
  <dd class="It-tag">RELL epsilon to break tie (default: 0.5)</dd>
</dl>
<h2 class="Ss" title="Ss" id="STANDARD_NON-PARAMETRIC_BOOTSTRAP:"><a class="selflink" href="#STANDARD_NON-PARAMETRIC_BOOTSTRAP:">STANDARD
  NON-PARAMETRIC BOOTSTRAP:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b> &lt;#replicates&gt;</dt>
  <dd class="It-tag">Bootstrap + ML tree + consensus tree (&gt;=100)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-bc</b> &lt;#replicates&gt;</dt>
  <dd class="It-tag">Bootstrap + consensus tree</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-bo</b> &lt;#replicates&gt;</dt>
  <dd class="It-tag">Bootstrap only</dd>
</dl>
<h2 class="Ss" title="Ss" id="SINGLE_BRANCH_TEST:"><a class="selflink" href="#SINGLE_BRANCH_TEST:">SINGLE
  BRANCH TEST:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-alrt</b> &lt;#replicates&gt;</dt>
  <dd class="It-tag">SH-like approximate likelihood ratio test (SH-aLRT)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-alrt</b> 0</dt>
  <dd class="It-tag">Parametric aLRT test (Anisimova and Gascuel 2006)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-abayes</b></dt>
  <dd class="It-tag">approximate Bayes test (Anisimova et al. 2011)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-lbp</b> &lt;#replicates&gt;</dt>
  <dd class="It-tag">Fast local bootstrap probabilities</dd>
</dl>
<h2 class="Ss" title="Ss" id="AUTOMATIC_MODEL_SELECTION:"><a class="selflink" href="#AUTOMATIC_MODEL_SELECTION:">AUTOMATIC
  MODEL SELECTION:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> TESTONLY</dt>
  <dd class="It-tag">Standard model selection (like jModelTest, ProtTest)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> TEST</dt>
  <dd class="It-tag">Like <b>-m</b> TESTONLY but followed by tree
    reconstruction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> TESTNEWONLY</dt>
  <dd class="It-tag">Extended model selection incl. FreeRate (+R)
    heterogeneity</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> TESTNEW</dt>
  <dd class="It-tag">Like <b>-m</b> TESTNEWONLY but followed by tree
      reconstruction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> TESTMERGEONLY</dt>
  <dd class="It-tag">Select best-fit partition scheme (like
    PartitionFinder)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> TESTMERGE</dt>
  <dd class="It-tag">Like <b>-m</b> TESTMERGEONLY but followed by tree
      reconstruction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> TESTNEWMERGEONLY</dt>
  <dd class="It-tag">Like <b>-m</b> TESTMERGEONLY but includes FreeRate
      heterogeneity</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> TESTNEWMERGE</dt>
  <dd class="It-tag">Like <b>-m</b> TESTNEWMERGEONLY followed by tree
      reconstruction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rcluster</b> &lt;percent&gt;</dt>
  <dd class="It-tag">Percentage of partition pairs (relaxed clustering
    alg.)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mset</b> program</dt>
  <dd class="It-tag">Restrict search to models supported by other programs
      (i.e., raxml, phyml or mrbayes)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mset</b> m1,...,mk</dt>
  <dd class="It-tag">Restrict search to models in a comma-separated list (e.g.
      <b>-mset</b> WAG,LG,JTT)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-msub</b> source</dt>
  <dd class="It-tag">Restrict search to AA models designed for specific sources
      (i.e., nuclear, mitochondrial, chloroplast or viral)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mfreq</b> f1,...,fk</dt>
  <dd class="It-tag">Restrict search to using a list of state frequencies
      (default protein: <b>-mfreq</b> FU,F; codon: <b>-mfreq</b>
    ,F1x4,F3x4,F)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mrate</b> r1,...,rk</dt>
  <dd class="It-tag">Restrict search to using a list of rate-across-sites models
      (e.g. <b>-mrate</b> E,I,G,I+G,R)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cmin</b> &lt;kmin&gt;</dt>
  <dd class="It-tag">Min #categories for FreeRate model [+R] (default: 2)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cmax</b> &lt;kmax&gt;</dt>
  <dd class="It-tag">Max #categories for FreeRate model [+R] (default: 10)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">???merit AIC|AICc|BIC</dt>
  <dd class="It-tag">Optimality criterion to use (default: all)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mtree</b></dt>
  <dd class="It-tag">Performing full tree search for each model considered</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mredo</b></dt>
  <dd class="It-tag">Ignoring model results computed earlier (default: no)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-madd</b> mx1,...,mxk</dt>
  <dd class="It-tag">List of mixture models to also consider</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mdef</b> &lt;nexus_file&gt;</dt>
  <dd class="It-tag">A model definition NEXUS file (see Manual)</dd>
</dl>
<h2 class="Ss" title="Ss" id="SUBSTITUTION_MODEL:"><a class="selflink" href="#SUBSTITUTION_MODEL:">SUBSTITUTION
  MODEL:</a></h2>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-m</b>
  &lt;model_name&gt;</div>
<dl class="Bl-tag">
  <dt class="It-tag">DNA: HKY (default), JC, F81, K2P, K3P, K81uf, TN/TrN,
    TNef,</dt>
  <dd class="It-tag">TIM, TIMef, TVM, TVMef, SYM, GTR, or 6-digit model
      specification (e.g., 010010 = HKY)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Protein: LG (default), Poisson, cpREV, mtREV, Dayhoff,
    mtMAM,</dt>
  <dd class="It-tag">JTT, WAG, mtART, mtZOA, VT, rtREV, DCMut, PMB, HIVb, HIVw,
      JTTDCMut, FLU, Blosum62, GTR20</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Protein mixture: C10,...,C60, EX2, EX3, EHO, UL2, UL3,
    EX_EHO, LG4M, LG4X,</dt>
  <dd class="It-tag">JTTCF4G</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Binary: JC2 (default), GTR2</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Empirical codon: KOSI07, SCHN05</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Mechanistic codon: GY (default), MG, MGK, GY0K, GY1KTS,
    GY1KTV, GY2K,</dt>
  <dd class="It-tag">MG1KTS, MG1KTV, MG2K</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Semi-empirical codon: XX_YY where XX is empirical and YY is
      mechanistic model</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Morphology/SNP: MK (default), ORDERED</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Otherwise: Name of file containing user-model
    parameters</dt>
  <dd class="It-tag">(rate parameters and state frequencies)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &lt;model_name&gt;+F or +FO or +FU or +FQ
    (default: auto)</dt>
  <dd class="It-tag">counted, optimized, user-defined, equal state
    frequency</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &lt;model_name&gt;+F1x4 or +F3x4</dt>
  <dd class="It-tag">Codon frequencies</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &lt;model_name&gt;+ASC</dt>
  <dd class="It-tag">Ascertainment bias correction for morphological/SNP
    data</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &quot;MIX{m1,...mK}&quot;</dt>
  <dd class="It-tag">Mixture model with K components</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &quot;FMIX{f1,...fK}&quot;</dt>
  <dd class="It-tag">Frequency mixture model with K components</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mwopt</b></dt>
  <dd class="It-tag">Turn on optimizing mixture weights (default: none)</dd>
</dl>
<h2 class="Ss" title="Ss" id="RATE_HETEROGENEITY_AMONG_SITES:"><a class="selflink" href="#RATE_HETEROGENEITY_AMONG_SITES:">RATE
  HETEROGENEITY AMONG SITES:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> &lt;model_name&gt;+I or +G[n] or +I+G[n] or
    +R[n]</dt>
  <dd class="It-tag">Invar, Gamma, Invar+Gamma, or FreeRate model where 'n' is
      number of categories (default: n=4)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> &lt;Gamma_shape&gt;</dt>
  <dd class="It-tag">Gamma shape parameter for site rates (default:
    estimate)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-amin</b> &lt;min_shape&gt;</dt>
  <dd class="It-tag">Min Gamma shape parameter for site rates (default:
    0.02)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-gmedian</b></dt>
  <dd class="It-tag">Median approximation for +G site rates (default: mean)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--opt-gamma-inv</b></dt>
  <dd class="It-tag">More thorough estimation for +I+G model parameters</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> &lt;p_invar&gt;</dt>
  <dd class="It-tag">Proportion of invariable sites (default: estimate)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wsr</b></dt>
  <dd class="It-tag">Write site rates to .rate file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mh</b></dt>
  <dd class="It-tag">Computing site-specific rates to .mhrate file using Meyer
      &amp; von Haeseler (2003) method</dd>
</dl>
<h2 class="Ss" title="Ss" id="SITE-SPECIFIC_FREQUENCY_MODEL:"><a class="selflink" href="#SITE-SPECIFIC_FREQUENCY_MODEL:">SITE-SPECIFIC
  FREQUENCY MODEL:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ft</b> &lt;tree_file&gt;</dt>
  <dd class="It-tag">Input tree to infer site frequency model</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-fs</b> &lt;in_freq_file&gt;</dt>
  <dd class="It-tag">Input site frequency model file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-fmax</b></dt>
  <dd class="It-tag">Posterior maximum instead of posterior mean
    approximation</dd>
</dl>
<h2 class="Ss" title="Ss" id="CONSENSUS_RECONSTRUCTION:"><a class="selflink" href="#CONSENSUS_RECONSTRUCTION:">CONSENSUS
  RECONSTRUCTION:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> &lt;tree_file&gt;</dt>
  <dd class="It-tag">Set of input trees for consensus reconstruction</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minsup</b> &lt;threshold&gt;</dt>
  <dd class="It-tag">Min split support in range [0,1]; 0.5 for majority-rule
      consensus (default: 0, i.e. extended consensus)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-bi</b> &lt;burnin&gt;</dt>
  <dd class="It-tag">Discarding &lt;burnin&gt; trees at beginning of
      &lt;treefile&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-con</b></dt>
  <dd class="It-tag">Computing consensus tree to .contree file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-net</b></dt>
  <dd class="It-tag">Computing consensus network to .nex file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sup</b> &lt;target_tree&gt;</dt>
  <dd class="It-tag">Assigning support values for &lt;target_tree&gt; to
      .suptree</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-suptag</b> &lt;name&gt;</dt>
  <dd class="It-tag">Node name (or ALL) to assign tree IDs where node
    occurs</dd>
</dl>
<h2 class="Ss" title="Ss" id="ROBINSON-FOULDS_DISTANCE:"><a class="selflink" href="#ROBINSON-FOULDS_DISTANCE:">ROBINSON-FOULDS
  DISTANCE:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rf_all</b></dt>
  <dd class="It-tag">Computing all-to-all RF distances of trees in
      &lt;treefile&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rf</b> &lt;treefile2&gt;</dt>
  <dd class="It-tag">Computing all RF distances between two sets of trees stored
      in &lt;treefile&gt; and &lt;treefile2&gt;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rf_adj</b></dt>
  <dd class="It-tag">Computing RF distances of adjacent trees in
      &lt;treefile&gt;</dd>
</dl>
<h2 class="Ss" title="Ss" id="TREE_TOPOLOGY_TEST:"><a class="selflink" href="#TREE_TOPOLOGY_TEST:">TREE
  TOPOLOGY TEST:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-z</b> &lt;trees_file&gt;</dt>
  <dd class="It-tag">Evaluating a set of user trees</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-zb</b> &lt;#replicates&gt;</dt>
  <dd class="It-tag">Performing BP,KH,SH,ELW tests for trees passed via
      <b>-z</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-zw</b></dt>
  <dd class="It-tag">Also performing weighted-KH and weighted-SH tests</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-au</b></dt>
  <dd class="It-tag">Also performing approximately unbiased (AU) test</dd>
</dl>
<h2 class="Ss" title="Ss" id="GENERATING_RANDOM_TREES:"><a class="selflink" href="#GENERATING_RANDOM_TREES:">GENERATING
  RANDOM TREES:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> &lt;num_taxa&gt;</dt>
  <dd class="It-tag">Create a random tree under Yule-Harding model</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ru</b> &lt;num_taxa&gt;</dt>
  <dd class="It-tag">Create a random tree under Uniform model</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rcat</b> &lt;num_taxa&gt;</dt>
  <dd class="It-tag">Create a random caterpillar tree</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rbal</b> &lt;num_taxa&gt;</dt>
  <dd class="It-tag">Create a random balanced tree</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rcsg</b> &lt;num_taxa&gt;</dt>
  <dd class="It-tag">Create a random circular split network</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-rlen</b> &lt;min_len&gt; &lt;mean_len&gt;
    &lt;max_len&gt;</dt>
  <dd class="It-tag">min, mean, and max branch lengths of random trees</dd>
</dl>
<h2 class="Ss" title="Ss" id="MISCELLANEOUS:"><a class="selflink" href="#MISCELLANEOUS:">MISCELLANEOUS:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wt</b></dt>
  <dd class="It-tag">Write locally optimal trees into .treels file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-blfix</b></dt>
  <dd class="It-tag">Fix branch lengths of user tree passed via <b>-te</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-blscale</b></dt>
  <dd class="It-tag">Scale branch lengths of user tree passed via <b>-t</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-blmin</b></dt>
  <dd class="It-tag">Min branch length for optimization (default 0.000001)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-blmax</b></dt>
  <dd class="It-tag">Max branch length for optimization (default 100)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wsr</b></dt>
  <dd class="It-tag">Write site rates and categories to .rate file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wsl</b></dt>
  <dd class="It-tag">Write site log-likelihoods to .sitelh file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wslr</b></dt>
  <dd class="It-tag">Write site log-likelihoods per rate category</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wslm</b></dt>
  <dd class="It-tag">Write site log-likelihoods per mixture class</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wslmr</b></dt>
  <dd class="It-tag">Write site log-likelihoods per mixture+rate class</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wspr</b></dt>
  <dd class="It-tag">Write site probabilities per rate category</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wspm</b></dt>
  <dd class="It-tag">Write site probabilities per mixture class</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wspmr</b></dt>
  <dd class="It-tag">Write site probabilities per mixture+rate class</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-wpl</b></dt>
  <dd class="It-tag">Write partition log-likelihoods to .partlh file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-fconst</b> f1,...,fN</dt>
  <dd class="It-tag">Add constant patterns into alignment (N=#nstates)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-me</b> &lt;epsilon&gt;</dt>
  <dd class="It-tag">Logl epsilon for model parameter optimization (default
      0.01)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-outfiles</b></dt>
  <dd class="It-tag">Suppress printing output files</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2017</td>
    <td class="foot-os">iqtree 1.5.3+dfsg</td>
  </tr>
</table>
</body>
</html>
