<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:19:06 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>ISECT_MERCATOR_ALIGNMENT_GFF(1) User Commands
ISECT_MERCATOR_ALIGNMENT_GFF(1)</p>

<p style="margin-top: 1em">NAME <br>
isect_mercator_alignment_gff - Extracts subalignments from a
Mercator multiple alignment for the features in the GFF
file</p>

<p style="margin-top: 1em">SYNOPSIS <br>
isect_mercator_alignment_gff [options] &lt;genome&gt;
&lt;GFF file&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
isect_mercator_alignment_gff from FSA 1.15.9</p>

<p style="margin-top: 1em">Extracts subalignments from a
Mercator multiple alignment for the features in the GFF
file.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h, --help <br>
show this message</p>

<p style="margin-top: 1em">-v, --version <br>
show version information</p>

<p style="margin-top: 1em">-t, --type &lt;string&gt; <br>
only look at features of particular type</p>

<p style="margin-top: 1em">-D, --data &lt;directory&gt;
<br>
path to map, genome and alignment files</p>

<p style="margin-top: 1em">-M, --map &lt;directory&gt; <br>
path to map and genome files</p>

<p style="margin-top: 1em">-A, --align &lt;directory&gt;
<br>
path to alignment files</p>

<p style="margin-top: 1em">-L, --lazy <br>
warn, rather than die, if the subalignment can&rsquo;t be
obtained</p>

<p style="margin-top: 1em">-U, --truncate <br>
truncate unmappable sequence (rather than skipping) and show
truncated subalignment</p>

<p style="margin-top: 1em">-s, --stockholm <br>
use and display Stockholm-format alignments with
conservation statistics (default is multi-FASTA)</p>

<p style="margin-top: 1em">-e, --verbose <br>
report progress</p>

<p style="margin-top: 1em">PLEASE NOTE: While this program
is reasonably fast if the GFF is properly ordered by
chromosome and the start and end coordinates of features, it
will be *very slow* if the GFF <br>
is not sorted. Assumes that the &quot;seqid&quot; or
&quot;name&quot; field (the first field) of the GFF entries
holds the chromosome name.</p>

<p style="margin-top: 1em">Note that the GFF specification
defines coordinates to be 1-based and fully-closed,
therefore representing the interval [start, end].
Conformance to this specification is <br>
assumed internally.</p>

<p style="margin-top: 1em">If requested, unmappable
sequence will be truncated to the mappable portion; note
that the truncation will favor the beginning of the
requested sequence.</p>

<p style="margin-top: 1em">If a GFF feature is on the -
strand, then the corresponding subalignment will be
reverse-complemented.</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">isect_mercator_alignment_gff
1.15.9 December 2015 ISECT_MERCATOR_ALIGNMENT_GFF(1)</p>
<hr>
</body>
</html>
