<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>IVA(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">IVA(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">IVA(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
iva - iterative virus assembler
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
usage: iva [options] {-f reads_fwd <b>-r</b> reads_rev | <b>--fr</b> reads}
  &lt;output directory&gt;
<h2 class="Ss" title="Ss" id="positional_arguments:"><a class="selflink" href="#positional_arguments:">positional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">Output directory</dt>
  <dd class="It-tag">Name of output directory (must not already exist)</dd>
</dl>
<h2 class="Ss" title="Ss" id="optional_arguments:"><a class="selflink" href="#optional_arguments:">optional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_and_output:"><a class="selflink" href="#Input_and_output:">Input
  and output:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> filename[.gz], <b>--reads_fwd</b>
    filename[.gz]</dt>
  <dd class="It-tag">Name of forward reads fasta/q file. Must be used in
      conjunction with <b>--reads_rev</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> filename[.gz], <b>--reads_rev</b>
    filename[.gz]</dt>
  <dd class="It-tag">Name of reverse reads fasta/q file. Must be used in
      conjunction with <b>--reads_fwd</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fr</b> filename[.gz]</dt>
  <dd class="It-tag">Name of interleaved fasta/q file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--keep_files</b></dt>
  <dd class="It-tag">Keep intermediate files (could be many!). Default is to
      delete all unnecessary files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--contigs</b> filename[.gz]</dt>
  <dd class="It-tag">Fasta file of contigs to be extended. Incompatible with
      <b>--reference</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--reference</b> filename[.gz]</dt>
  <dd class="It-tag">EXPERIMENTAL! This option is EXPERIMENTAL, not recommended,
      and has not been tested! Fasta file of reference genome, or parts thereof.
      IVA will try to assemble one contig per sequence in this file.
      Incompatible with <b>--contigs</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--verbose</b></dt>
  <dd class="It-tag">Be verbose by printing messages to stdout. Use up to three
      times for increasing verbosity.</dd>
</dl>
<h2 class="Ss" title="Ss" id="SMALT_mapping_options:"><a class="selflink" href="#SMALT_mapping_options:">SMALT
  mapping options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b> INT, <b>--smalt_k</b> INT</dt>
  <dd class="It-tag">kmer hash length in SMALT (the <b>-k</b> option in smalt
      index) [19]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> INT, <b>--smalt_s</b> INT</dt>
  <dd class="It-tag">kmer hash step size in SMALT (the <b>-s</b> option in smalt
      index) [11]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-y</b> FLOAT, <b>--smalt_id</b> FLOAT</dt>
  <dd class="It-tag">Minimum identity threshold for mapping to be reported (the
      <b>-y</b> option in smalt map) [0.5]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Contig_options:"><a class="selflink" href="#Contig_options:">Contig
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ctg_first_trim</b> INT</dt>
  <dd class="It-tag">Number of bases to trim off the end of every contig before
      extending for the first time [25]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ctg_iter_trim</b> INT</dt>
  <dd class="It-tag">During iterative extension, number of bases to trim off the
      end of a contig when extension fails (then try extending again) [10]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ext_min_cov</b> INT</dt>
  <dd class="It-tag">Minimum kmer depth needed to use that kmer to extend a
      contig [10]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ext_min_ratio</b> FLOAT</dt>
  <dd class="It-tag">Sets N, where kmer for extension must be at least N times
      more abundant than next most common kmer [4]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ext_max_bases</b> INT</dt>
  <dd class="It-tag">Maximum number of bases to try to extend on each iteration
      [100]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ext_min_clip</b> INT</dt>
  <dd class="It-tag">Set minimum number of bases soft clipped off a read for
      those bases to be used for extension [3]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max_contigs</b> INT</dt>
  <dd class="It-tag">Maximum number of contigs allowed in the assembly. No more
      seeds generated if the cutoff is reached [50]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Seed_generation_options:"><a class="selflink" href="#Seed_generation_options:">Seed
  generation options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--make_new_seeds</b></dt>
  <dd class="It-tag">When no more contigs can be extended, generate a new seed.
      This is forced to be true when <b>--contigs</b> is not used</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_start_length</b> INT</dt>
  <dd class="It-tag">When making a seed sequence, use the most common kmer of
      this length. Default is to use the minimum of (median read length, 95).
      Warning: it is not recommended to set this higher than 95</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_stop_length</b> INT</dt>
  <dd class="It-tag">Stop extending seed using perfect matches from reads when
      this length is reached. Future extensions are then made by treating the
      seed as a contig [0.9*max_insert]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_min_kmer_cov</b> INT</dt>
  <dd class="It-tag">Minimum kmer coverage of initial seed [25]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_max_kmer_cov</b> INT</dt>
  <dd class="It-tag">Maximum kmer coverage of initial seed [1000000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_ext_max_bases</b> INT</dt>
  <dd class="It-tag">Maximum number of bases to try to extend on each iteration
      [50]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_overlap_length</b> INT</dt>
  <dd class="It-tag">Number of overlapping bases needed between read and seed to
      use that read to extend [seed_start_length]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_ext_min_cov</b> INT</dt>
  <dd class="It-tag">Minimum kmer depth needed to use that kmer to extend a
      contig [10]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed_ext_min_ratio</b> FLOAT</dt>
  <dd class="It-tag">Sets N, where kmer for extension must be at least N times
      more abundant than next most common kmer [4]</dd>
</dl>
<h2 class="Ss" title="Ss" id="Read_trimming_options:"><a class="selflink" href="#Read_trimming_options:">Read
  trimming options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trimmomatic</b> FILENAME</dt>
  <dd class="It-tag">Provide location of trimmomatic.jar file to enable read
      trimming. Required if <b>--adapters</b> used</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trimmo_qual</b> STRING</dt>
  <dd class="It-tag">Trimmomatic options used to quality trim reads [LEADING:10
      TRAILING:10 SLIDINGWINDOW:4:20]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--adapters</b> FILENAME</dt>
  <dd class="It-tag">Fasta file of adapter sequences to be trimmed off reads. If
      used, must also use <b>--trimmomatic</b>. Default is file of adapters
      supplied with IVA</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_trimmed_length</b> INT</dt>
  <dd class="It-tag">Minimum length of read after trimming [50]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pcr_primers</b> FILENAME</dt>
  <dd class="It-tag">FASTA file of primers. The first perfect match found to a
      sequence in the primers file will be trimmed off the start of each read.
      This is run after trimmomatic (if <b>--trimmomatic</b> used)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Other_options:"><a class="selflink" href="#Other_options:">Other
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> INT, <b>--max_insert</b> INT</dt>
  <dd class="It-tag">Maximum insert size (includes read length). Reads with
      inferred insert size more than the maximum will not be used to extend
      contigs [800]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> INT, <b>--threads</b> INT</dt>
  <dd class="It-tag">Number of threads to use [1]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--kmc_onethread</b></dt>
  <dd class="It-tag">Force kmc to use one thread. By default the value of
      <b>-t</b>/--threads is used when running kmc</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--strand_bias</b> FLOAT in [0,0.5]</dt>
  <dd class="It-tag">Set strand bias cutoff of mapped reads when trimming contig
      ends, in the interval [0,0.5]. A value of x means that a base needs
      min(fwd_depth, rev_depth) / total_depth &lt;= x. The only time this should
      be used is with libraries with overlapping reads (ie fragment length &lt;
      2*read length), and even then, it can make results worse. If used, try a
      low value like 0.1 first [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--test</b></dt>
  <dd class="It-tag">Run using built in test data. All other options will be
      ignored, except the mandatory output directory, and <b>--trimmomatic</b>
      and <b>--threads</b> can be also be used</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">show program's version number and exit</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">iva 1.0.8</td>
  </tr>
</table>
</body>
</html>
