<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:19:11 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>IVA(1) User Commands IVA(1)</p>

<p style="margin-top: 1em">NAME <br>
iva - iterative virus assembler</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: iva [options] {-f reads_fwd -r reads_rev | --fr
reads} &lt;output directory&gt;</p>

<p style="margin-top: 1em">positional arguments: <br>
Output directory <br>
Name of output directory (must not already exist)</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">Input and output: <br>
-f filename[.gz], --reads_fwd filename[.gz] <br>
Name of forward reads fasta/q file. Must be used in
conjunction with --reads_rev</p>

<p style="margin-top: 1em">-r filename[.gz], --reads_rev
filename[.gz] <br>
Name of reverse reads fasta/q file. Must be used in
conjunction with --reads_fwd</p>

<p style="margin-top: 1em">--fr filename[.gz] <br>
Name of interleaved fasta/q file</p>

<p style="margin-top: 1em">--keep_files <br>
Keep intermediate files (could be many!). Default is to
delete all unnecessary files</p>

<p style="margin-top: 1em">--contigs filename[.gz] <br>
Fasta file of contigs to be extended. Incompatible with
--reference</p>

<p style="margin-top: 1em">--reference filename[.gz] <br>
EXPERIMENTAL! This option is EXPERIMENTAL, not recommended,
and has not been tested! Fasta file of reference genome, or
parts thereof. IVA will try to assemble one contig <br>
per sequence in this file. Incompatible with --contigs</p>

<p style="margin-top: 1em">-v, --verbose <br>
Be verbose by printing messages to stdout. Use up to three
times for increasing verbosity.</p>

<p style="margin-top: 1em">SMALT mapping options: <br>
-k INT, --smalt_k INT <br>
kmer hash length in SMALT (the -k option in smalt index)
[19]</p>

<p style="margin-top: 1em">-s INT, --smalt_s INT <br>
kmer hash step size in SMALT (the -s option in smalt index)
[11]</p>

<p style="margin-top: 1em">-y FLOAT, --smalt_id FLOAT <br>
Minimum identity threshold for mapping to be reported (the
-y option in smalt map) [0.5]</p>

<p style="margin-top: 1em">Contig options: <br>
--ctg_first_trim INT <br>
Number of bases to trim off the end of every contig before
extending for the first time [25]</p>

<p style="margin-top: 1em">--ctg_iter_trim INT <br>
During iterative extension, number of bases to trim off the
end of a contig when extension fails (then try extending
again) [10]</p>

<p style="margin-top: 1em">--ext_min_cov INT <br>
Minimum kmer depth needed to use that kmer to extend a
contig [10]</p>

<p style="margin-top: 1em">--ext_min_ratio FLOAT <br>
Sets N, where kmer for extension must be at least N times
more abundant than next most common kmer [4]</p>

<p style="margin-top: 1em">--ext_max_bases INT <br>
Maximum number of bases to try to extend on each iteration
[100]</p>

<p style="margin-top: 1em">--ext_min_clip INT <br>
Set minimum number of bases soft clipped off a read for
those bases to be used for extension [3]</p>

<p style="margin-top: 1em">--max_contigs INT <br>
Maximum number of contigs allowed in the assembly. No more
seeds generated if the cutoff is reached [50]</p>

<p style="margin-top: 1em">Seed generation options: <br>
--make_new_seeds <br>
When no more contigs can be extended, generate a new seed.
This is forced to be true when --contigs is not used</p>

<p style="margin-top: 1em">--seed_start_length INT <br>
When making a seed sequence, use the most common kmer of
this length. Default is to use the minimum of (median read
length, 95). Warning: it is not recommended to set this <br>
higher than 95</p>

<p style="margin-top: 1em">--seed_stop_length INT <br>
Stop extending seed using perfect matches from reads when
this length is reached. Future extensions are then made by
treating the seed as a contig [0.9*max_insert]</p>

<p style="margin-top: 1em">--seed_min_kmer_cov INT <br>
Minimum kmer coverage of initial seed [25]</p>

<p style="margin-top: 1em">--seed_max_kmer_cov INT <br>
Maximum kmer coverage of initial seed [1000000]</p>

<p style="margin-top: 1em">--seed_ext_max_bases INT <br>
Maximum number of bases to try to extend on each iteration
[50]</p>

<p style="margin-top: 1em">--seed_overlap_length INT <br>
Number of overlapping bases needed between read and seed to
use that read to extend [seed_start_length]</p>

<p style="margin-top: 1em">--seed_ext_min_cov INT <br>
Minimum kmer depth needed to use that kmer to extend a
contig [10]</p>

<p style="margin-top: 1em">--seed_ext_min_ratio FLOAT <br>
Sets N, where kmer for extension must be at least N times
more abundant than next most common kmer [4]</p>

<p style="margin-top: 1em">Read trimming options: <br>
--trimmomatic FILENAME <br>
Provide location of trimmomatic.jar file to enable read
trimming. Required if --adapters used</p>

<p style="margin-top: 1em">--trimmo_qual STRING <br>
Trimmomatic options used to quality trim reads [LEADING:10
TRAILING:10 SLIDINGWINDOW:4:20]</p>

<p style="margin-top: 1em">--adapters FILENAME <br>
Fasta file of adapter sequences to be trimmed off reads. If
used, must also use --trimmomatic. Default is file of
adapters supplied with IVA</p>

<p style="margin-top: 1em">--min_trimmed_length INT <br>
Minimum length of read after trimming [50]</p>

<p style="margin-top: 1em">--pcr_primers FILENAME <br>
FASTA file of primers. The first perfect match found to a
sequence in the primers file will be trimmed off the start
of each read. This is run after trimmomatic (if <br>
--trimmomatic used)</p>

<p style="margin-top: 1em">Other options: <br>
-i INT, --max_insert INT <br>
Maximum insert size (includes read length). Reads with
inferred insert size more than the maximum will not be used
to extend contigs [800]</p>

<p style="margin-top: 1em">-t INT, --threads INT <br>
Number of threads to use [1]</p>

<p style="margin-top: 1em">--kmc_onethread <br>
Force kmc to use one thread. By default the value of
-t/--threads is used when running kmc</p>

<p style="margin-top: 1em">--strand_bias FLOAT in [0,0.5]
<br>
Set strand bias cutoff of mapped reads when trimming contig
ends, in the interval [0,0.5]. A value of x means that a
base needs min(fwd_depth, rev_depth) / total_depth &lt;=
<br>
x. The only time this should be used is with libraries with
overlapping reads (ie fragment length &lt; 2*read length),
and even then, it can make results worse. If used, try <br>
a low value like 0.1 first [0]</p>

<p style="margin-top: 1em">--test Run using built in test
data. All other options will be ignored, except the
mandatory output directory, and --trimmomatic and --threads
can be also be used</p>

<p style="margin-top: 1em">--version <br>
show program&rsquo;s version number and exit</p>

<p style="margin-top: 1em">iva 1.0.8 January 2017
IVA(1)</p>
<hr>
</body>
</html>
