<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>jackhmmer(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">jackhmmer(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">jackhmmer(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
jackhmmer - iteratively search sequence(s) against a protein database
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>jackhmmer</b> <i>[options]</i> <i>&lt;seqfile&gt;</i> <i>&lt;seqdb&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>jackhmmer</b> iteratively searches each query sequence in
  <i>&lt;seqfile&gt;</i> against the target sequence(s) in <i>&lt;seqdb&gt;</i>.
  The first iteration is identical to a <b>phmmer</b> search. For the next
  iteration, a multiple alignment of the query together with all target
  sequences satisfying <i>inclusion thresholds</i> is assembled, a profile is
  constructed from this alignment (identical to using <b>hmmbuild</b> on the
  alignment), and profile search of the <i>&lt;seqdb&gt;</i> is done (identical
  to an <b>hmmsearch</b> with the profile).
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The query <i>&lt;seqfile&gt;</i> may be '-' (a dash character), in which case
  the query sequences are read from a &lt;stdin&gt; pipe instead of from a file.
  The <i>&lt;seqdb&gt;</i> cannot be read from a &lt;stdin&gt; stream, because
  <b>jackhmmer</b> needs to do multiple passes over the database.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The output format is designed to be human-readable, but is often so voluminous
  that reading it is impractical, and parsing it is a pain. The <b>--tblout</b>
  and <b>--domtblout</b> options save output in simple tabular formats that are
  concise and easier to parse. The <b>-o</b> option allows redirecting the main
  output, including throwing it away in /dev/null.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and all
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the maximum number of iterations to <i>&lt;n&gt;</i>.
      The default is 5. If N=1, the result is equivalent to a <b>phmmer</b>
      search.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_OUTPUT"><a class="selflink" href="#OPTIONS_CONTROLLING_OUTPUT">OPTIONS
  CONTROLLING OUTPUT</a></h1>
By default, output for each iteration appears on stdout in a somewhat human
  readable, somewhat parseable format. These options allow redirecting that
  output or saving additional kinds of output to files, including checkpoint
  files for each iteration.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Direct the human-readable output to a file
      <i>&lt;f&gt;</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">After the final iteration, save an annotated multiple
      alignment of all hits satisfying inclusion thresholds (also including the
      original query) to <i>&lt;f&gt;</i> in Stockholm format.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tblout</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">After the final iteration, save a tabular summary of top
      sequence hits to <i>&lt;f&gt;</i> in a readily parseable, columnar,
      whitespace-delimited format.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domtblout</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">After the final iteration, save a tabular summary of top
      domain hits to <i>&lt;f&gt;</i> in a readily parseable, columnar,
      whitespace-delimited format.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--chkhmm</b><i> &lt;prefix&gt;</i></dt>
  <dd class="It-tag">At the start of each iteration, checkpoint the query HMM,
      saving it to a file named <i>&lt;prefix&gt;-&lt;n&gt;.hmm</i> where
      <i>&lt;n&gt;</i> is the iteration number (from 1..N).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--chkali</b><i> &lt;prefix&gt;</i></dt>
  <dd class="It-tag">At the end of each iteration, checkpoint an alignment of
      all domains satisfying inclusion thresholds (e.g. what will become the
      query HMM for the next iteration), saving it to a file named
      <i>&lt;checkpoint file prefix&gt;-&lt;n&gt;.sto</i> in Stockholm format,
      where <i>&lt;n&gt;</i> is the iteration number (from 1..N).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--acc</b></dt>
  <dd class="It-tag">Use accessions instead of names in the main output, where
      available for profiles and/or sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noali</b></dt>
  <dd class="It-tag">Omit the alignment section from the main output. This can
      greatly reduce the output volume.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--notextw</b></dt>
  <dd class="It-tag">Unlimit the length of each line in the main output. The
      default is a limit of 120 characters per line, which helps in displaying
      the output cleanly on terminals and in editors, but can truncate target
      profile description lines.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--textw</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the main output's line length limit to <i>&lt;n&gt;</i>
      characters per line. The default is 120.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_SINGLE_SEQUENCE_SCORING_(FIRST_ITERATION)"><a class="selflink" href="#OPTIONS_CONTROLLING_SINGLE_SEQUENCE_SCORING_(FIRST_ITERATION)">OPTIONS
  CONTROLLING SINGLE SEQUENCE SCORING (FIRST ITERATION)</a></h1>
By default, the first iteration uses a search model constructed from a single
  query sequence. This model is constructed using a standard 20x20 substitution
  matrix for residue probabilities, and two additional parameters for
  position-independent gap open and gap extend probabilities. These options
  allow the default single-sequence scoring parameters to be changed.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--popen</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the gap open probability for a single sequence query
      model to <i>&lt;x&gt;</i>. The default is 0.02. <i>&lt;x&gt;</i> must be
      &gt;= 0 and &lt; 0.5.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pextend</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the gap extend probability for a single sequence query
      model to <i>&lt;x&gt;</i>. The default is 0.4. <i>&lt;x&gt;</i> must be
      &gt;= 0 and &lt; 1.0.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mx</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Obtain residue alignment probabilities from the built-in
      substitution matrix named <i>&lt;s&gt;</i>. Several standard matrices are
      built-in, and do not need to be read from files. The matrix name
      <i>&lt;s&gt;</i> can be PAM30, PAM70, PAM120, PAM240, BLOSUM45, BLOSUM50,
      BLOSUM62, BLOSUM80, or BLOSUM90. Only one of the <b>--mx</b> and
      <b>--mxfile</b> options may be used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mxfile</b><i> &lt;mxfile&gt;</i></dt>
  <dd class="It-tag">Obtain residue alignment probabilities from the
      substitution matrix in file <i>&lt;mxfile&gt;</i>. The default score
      matrix is BLOSUM62 (this matrix is internal to HMMER and does not have to
      be available as a file). The format of a substitution matrix
      <i>&lt;mxfile&gt;</i> is the standard format accepted by BLAST, FASTA, and
      other sequence analysis software.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_REPORTING_THRESHOLDS"><a class="selflink" href="#OPTIONS_CONTROLLING_REPORTING_THRESHOLDS">OPTIONS
  CONTROLLING REPORTING THRESHOLDS</a></h1>
Reporting thresholds control which hits are reported in output files (the main
  output, <i>--tblout</i>, and <i>--domtblout</i>). In each iteration, sequence
  hits and domain hits are ranked by statistical significance (E-value) and
  output is generated in two sections called per-target and per-domain output.
  In per-target output, by default, all sequence hits with an E-value &lt;= 10
  are reported. In the per-domain output, for each target that has passed
  per-target reporting thresholds, all domains satisfying per-domain reporting
  thresholds are reported. By default, these are domains with conditional
  E-values of &lt;= 10. The following options allow you to change the default
  E-value reporting thresholds, or to use bit score thresholds instead.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Report sequences with E-values &lt;= <i>&lt;x&gt;</i> in
      per-sequence output. The default is 10.0.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Use a bit score threshold for per-sequence output instead
      of an E-value threshold (any setting of <b>-E</b> is ignored). Report
      sequences with a bit score of &gt;= <i>&lt;x&gt;</i>. By default this
      option is unset.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Declare the total size of the database to be
      <i>&lt;x&gt;</i> sequences, for purposes of E-value calculation. Normally
      E-values are calculated relative to the size of the database you actually
      searched (e.g. the number of sequences in <i>target_seqdb</i>). In some
      cases (for instance, if you've split your target sequence database into
      multiple files for parallelization of your search), you may know better
      what the actual size of your search space is.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Report domains with conditional E-values &lt;=
      <i>&lt;x&gt;</i> in per-domain output, in addition to the top-scoring
      domain per significant sequence hit. The default is 10.0.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Use a bit score threshold for per-domain output instead of
      an E-value threshold (any setting of <b>--domT</b> is ignored). Report
      domains with a bit score of &gt;= <i>&lt;x&gt;</i> in per-domain output,
      in addition to the top-scoring domain per significant sequence hit. By
      default this option is unset.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domZ</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Declare the number of significant sequences to be
      <i>&lt;x&gt;</i> sequences, for purposes of conditional E-value
      calculation for additional domain significance. Normally conditional
      E-values are calculated relative to the number of sequences passing
      per-sequence reporting threshold.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_INCLUSION_THRESHOLDS"><a class="selflink" href="#OPTIONS_CONTROLLING_INCLUSION_THRESHOLDS">OPTIONS
  CONTROLLING INCLUSION THRESHOLDS</a></h1>
Inclusion thresholds control which hits are included in the multiple alignment
  and profile constructed for the next search iteration. By default, a sequence
  must have a per-sequence E-value of &lt;= 0.001 (see <b>-E</b> option) to be
  included, and any additional domains in it besides the top-scoring one must
  have a conditional E-value of &lt;= 0.001 (see <b>--domE</b> option). The
  difference between reporting thresholds and inclusion thresholds is that
  inclusion thresholds control which hits actually get used in the next
  iteration (or the final output multiple alignment if the <i>-A</i> option is
  used), whereas reporting thresholds control what you see in output. Reporting
  thresholds are generally more loose so you can see borderline hits in the top
  of the noise that might be of interest.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Include sequences with E-values &lt;= <i>&lt;x&gt;</i> in
      subsequent iteration or final alignment output by <i>-A</i>. The default
      is 0.001.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Use a bit score threshold for per-sequence inclusion
      instead of an E-value threshold (any setting of <b>--incE</b> is ignored).
      Include sequences with a bit score of &gt;= <i>&lt;x&gt;</i>. By default
      this option is unset.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incdomE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Include domains with conditional E-values &lt;=
      <i>&lt;x&gt;</i> in subsequent iteration or final alignment output by
      <i>-A</i>, in addition to the top-scoring domain per significant sequence
      hit. The default is 0.001.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incdomT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Use a bit score threshold for per-domain inclusion instead
      of an E-value threshold (any setting of <b>--incT</b> is ignored). Include
      domains with a bit score of &gt;= <i>&lt;x&gt;</i>. By default this option
      is unset.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_ACCELERATION_HEURISTICS"><a class="selflink" href="#OPTIONS_CONTROLLING_ACCELERATION_HEURISTICS">OPTIONS
  CONTROLLING ACCELERATION HEURISTICS</a></h1>
HMMER3 searches are accelerated in a three-step filter pipeline: the MSV filter,
  the Viterbi filter, and the Forward filter. The first filter is the fastest
  and most approximate; the last is the full Forward scoring algorithm, slowest
  but most accurate. There is also a bias filter step between MSV and Viterbi.
  Targets that pass all the steps in the acceleration pipeline are then
  subjected to postprocessing -- domain identification and scoring using the
  Forward/Backward algorithm.
<div style="height: 1.00em;">&#x00A0;</div>
Essentially the only free parameters that control HMMER's heuristic filters are
  the P-value thresholds controlling the expected fraction of nonhomologous
  sequences that pass the filters. Setting the default thresholds higher will
  pass a higher proportion of nonhomologous sequence, increasing sensitivity at
  the expense of speed; conversely, setting lower P-value thresholds will pass a
  smaller proportion, decreasing sensitivity and increasing speed. Setting a
  filter's P-value threshold to 1.0 means it will passing all sequences, and
  effectively disables the filter.
<div style="height: 1.00em;">&#x00A0;</div>
Changing filter thresholds only removes or includes targets from consideration;
  changing filter thresholds does not alter bit scores, E-values, or alignments,
  all of which are determined solely in postprocessing.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max</b></dt>
  <dd class="It-tag">Maximum sensitivity. Turn off all filters, including the
      bias filter, and run full Forward/Backward postprocessing on every target.
      This increases sensitivity slightly, at a large cost in speed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F1</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">First filter threshold; set the P-value threshold for the
      MSV filter step. The default is 0.02, meaning that roughly 2% of the
      highest scoring nonhomologous targets are expected to pass the filter.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F2</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Second filter threshold; set the P-value threshold for the
      Viterbi filter step. The default is 0.001.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F3</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Third filter threshold; set the P-value threshold for the
      Forward filter step. The default is 1e-5.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nobias</b></dt>
  <dd class="It-tag">Turn off the bias filter. This increases sensitivity
      somewhat, but can come at a high cost in speed, especially if the query
      has biased residue composition (such as a repetitive sequence region, or
      if it is a membrane protein with large regions of hydrophobicity). Without
      the bias filter, too many sequences may pass the filter with biased
      queries, leading to slower than expected performance as the
      computationally intensive Forward/Backward algorithms shoulder an
      abnormally heavy load.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_PROFILE_CONSTRUCTION_(LATER_ITERATIONS)"><a class="selflink" href="#OPTIONS_CONTROLLING_PROFILE_CONSTRUCTION_(LATER_ITERATIONS)">OPTIONS
  CONTROLLING PROFILE CONSTRUCTION (LATER ITERATIONS)</a></h1>
These options control how consensus columns are defined in multiple alignments
  when building profiles. By default, <b>jackhmmer</b> always includes your
  original query sequence in the alignment result at every iteration, and
  consensus positions are defined by that query sequence: that is, a default
  <b>jackhmmer</b> profile is always the same length as your original query, at
  every iteration.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fast</b></dt>
  <dd class="It-tag">Define consensus columns as those that have a fraction
      &gt;= <b>symfrac</b> of residues as opposed to gaps. (See below for the
      <b>--symfrac</b> option.) Although this is the default profile
      construction option elsewhere (in <b>hmmbuild</b>, in particular), it may
      have undesirable effects in <b>jackhmmer</b>, because a profile could
      iteratively walk in sequence space away from your original query, leaving
      few or no consensus columns corresponding to its residues.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hand</b></dt>
  <dd class="It-tag">Define consensus columns in next profile using reference
      annotation to the multiple alignment. <b>jackhmmer</b> propagates
      reference annotation from the previous profile to the multiple alignment,
      and thence to the next profile. This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--symfrac</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Define the residue fraction threshold necessary to define a
      consensus column when using the <b>--fast</b> option. The default is 0.5.
      The symbol fraction in each column is calculated after taking relative
      sequence weighting into account, and ignoring gap characters corresponding
      to ends of sequence fragments (as opposed to internal
      insertions/deletions). Setting this to 0.0 means that every alignment
      column will be assigned as consensus, which may be useful in some cases.
      Setting it to 1.0 means that only columns that include 0 gaps (internal
      insertions/deletions) will be assigned as consensus.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fragthresh</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">We only want to count terminal gaps as deletions if the
      aligned sequence is known to be full-length, not if it is a fragment (for
      instance, because only part of it was sequenced). HMMER uses a simple rule
      to infer fragments: if the sequence length L is less than or equal to a
      fraction <i>&lt;x&gt;</i> times the alignment length in columns, then the
      sequence is handled as a fragment. The default is 0.5. Setting
      <b>--fragthresh</b><i>0</i> will define no (nonempty) sequence as a
      fragment; you might want to do this if you know you've got a carefully
      curated alignment of full-length sequences. Setting
      <b>--fragthresh</b><i>1</i> will define all sequences as fragments; you
      might want to do this if you know your alignment is entirely composed of
      fragments, such as translated short reads in metagenomic shotgun data.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_RELATIVE_WEIGHTS"><a class="selflink" href="#OPTIONS_CONTROLLING_RELATIVE_WEIGHTS">OPTIONS
  CONTROLLING RELATIVE WEIGHTS</a></h1>
Whenever a profile is built from a multiple alignment, HMMER uses an ad hoc
  sequence weighting algorithm to downweight closely related sequences and
  upweight distantly related ones. This has the effect of making models less
  biased by uneven phylogenetic representation. For example, two identical
  sequences would typically each receive half the weight that one sequence would
  (and this is why <b>jackhmmer</b> isn't concerned about always including your
  original query sequence in each iteration's alignment, even if it finds it
  again in the database you're searching). These options control which algorithm
  gets used.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wpb</b></dt>
  <dd class="It-tag">Use the Henikoff position-based sequence weighting scheme
      [Henikoff and Henikoff, J. Mol. Biol. 243:574, 1994]. This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wgsc</b></dt>
  <dd class="It-tag">Use the Gerstein/Sonnhammer/Chothia weighting algorithm
      [Gerstein et al, J. Mol. Biol. 235:1067, 1994].
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wblosum</b></dt>
  <dd class="It-tag">Use the same clustering scheme that was used to weight data
      in calculating BLOSUM subsitution matrices [Henikoff and Henikoff, Proc.
      Natl. Acad. Sci 89:10915, 1992]. Sequences are single-linkage clustered at
      an identity threshold (default 0.62; see <i>--wid</i>) and within each
      cluster of c sequences, each sequence gets relative weight 1/c.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wnone</b></dt>
  <dd class="It-tag">No relative weights. All sequences are assigned uniform
      weight.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wid</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the identity threshold used by single-linkage
      clustering when using <i>--wblosum</i>. Invalid with any other weighting
      scheme. Default is 0.62.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_EFFECTIVE_SEQUENCE_NUMBER"><a class="selflink" href="#OPTIONS_CONTROLLING_EFFECTIVE_SEQUENCE_NUMBER">OPTIONS
  CONTROLLING EFFECTIVE SEQUENCE NUMBER</a></h1>
After relative weights are determined, they are normalized to sum to a total
  effective sequence number, <i>eff_nseq</i>. This number may be the actual
  number of sequences in the alignment, but it is almost always smaller than
  that. The default entropy weighting method <i>(--eent)</i> reduces the
  effective sequence number to reduce the information content (relative entropy,
  or average expected score on true homologs) per consensus position. The target
  relative entropy is controlled by a two-parameter function, where the two
  parameters are settable with <i>--ere</i> and <i>--esigma</i>.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eent</b></dt>
  <dd class="It-tag">Adjust effective sequence number to achieve a specific
      relative entropy per position (see <i>--ere</i>). This is the default.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eclust</b></dt>
  <dd class="It-tag">Set effective sequence number to the number of
      single-linkage clusters at a specific identity threshold (see
      <i>--eid</i>). This option is not recommended; it's for experiments
      evaluating how much better <b>--eent</b> is.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--enone</b></dt>
  <dd class="It-tag">Turn off effective sequence number determination and just
      use the actual number of sequences. One reason you might want to do this
      is to try to maximize the relative entropy/position of your model, which
      may be useful for short models.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eset</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Explicitly set the effective sequence number for all models
      to <i>&lt;x&gt;</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ere</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the minimum relative entropy/position target to
      <i>&lt;x&gt;</i>. Requires <b>--eent</b>. Default depends on the sequence
      alphabet; for protein sequences, it is 0.59 bits/position.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--esigma</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the minimum relative entropy contributed by an entire
      model alignment, over its whole length. This has the effect of making
      short models have higher relative entropy per position than <i>--ere</i>
      alone would give. The default is 45.0 bits.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--eid</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the fractional pairwise identity cutoff used by single
      linkage clustering with the <b>--eclust</b> option. The default is 0.62.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_PRIORS"><a class="selflink" href="#OPTIONS_CONTROLLING_PRIORS">OPTIONS
  CONTROLLING PRIORS</a></h1>
In profile construction, by default, weighted counts are converted to mean
  posterior probability parameter estimates using mixture Dirichlet priors.
  Default mixture Dirichlet prior parameters for protein models and for nucleic
  acid (RNA and DNA) models are built in. The following options allow you to
  override the default priors.
<div style="height: 1.00em;">&#x00A0;</div>
<b>--pnone</b> Don't use any priors. Probability parameters will simply be the
  observed frequencies, after relative sequence weighting.
<div style="height: 1.00em;">&#x00A0;</div>
<b>--plaplace</b> Use a Laplace +1 prior in place of the default mixture
  Dirichlet prior.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_E-VALUE_CALIBRATION"><a class="selflink" href="#OPTIONS_CONTROLLING_E-VALUE_CALIBRATION">OPTIONS
  CONTROLLING E-VALUE CALIBRATION</a></h1>
Estimating the location parameters for the expected score distributions for MSV
  filter scores, Viterbi filter scores, and Forward scores requires three short
  random sequence simulations.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EmL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter mu for MSV filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EmN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter mu for MSV filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EvL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter mu for Viterbi filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EvN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter mu for Viterbi filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EfL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter tau for Forward E-values. Default is 100.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EfN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter tau for Forward E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--Eft</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the tail mass fraction to fit in the simulation that
      estimates the location parameter tau for Forward evalues. Default is 0.04.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_OPTIONS"><a class="selflink" href="#OTHER_OPTIONS">OTHER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nonull2</b></dt>
  <dd class="It-tag">Turn off the null2 score corrections for biased
      composition.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Assert that the total number of targets in your searches is
      <i>&lt;x&gt;</i>, for the purposes of per-sequence E-value calculations,
      rather than the actual number of targets seen.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domZ</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Assert that the total number of targets in your searches is
      <i>&lt;x&gt;</i>, for the purposes of per-domain conditional E-value
      calculations, rather than the number of targets that passed the reporting
      thresholds.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Seed the random number generator with <i>&lt;n&gt;</i>, an
      integer &gt;= 0. If <i>&lt;n&gt;</i> is &gt;0, any stochastic simulations
      will be reproducible; the same command will give the same results. If
      <i>&lt;n&gt;</i> is 0, the random number generator is seeded arbitrarily,
      and stochastic simulations will vary from run to run of the same command.
      The default seed is 42.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--qformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Declare that the input <i>query_seqfile</i> is in format
      <i>&lt;s&gt;</i>. Accepted sequence file formats include FASTA, EMBL,
      GenBank, DDBJ, UniProt, Stockholm, and SELEX. Default is to autodetect the
      format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Declare that the input <i>target_seqdb</i> is in format
      <i>&lt;s&gt;</i>. Accepted sequence file formats include FASTA, EMBL,
      GenBank, DDBJ, UniProt, Stockholm, and SELEX. Default is to autodetect the
      format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cpu</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the number of parallel worker threads to
      <i>&lt;n&gt;</i>. By default, HMMER sets this to the number of CPU cores
      it detects in your machine - that is, it tries to maximize the use of your
      available processor cores. Setting <i>&lt;n&gt;</i> higher than the number
      of available cores is of little if any value, but you may want to set it
      to something less. You can also control this number by setting an
      environment variable, <i>HMMER_NCPU</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    This option is only available if HMMER was compiled with POSIX threads
      support. This is the default, but it may have been turned off at
      compile-time for your site or machine for some reason.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stall</b></dt>
  <dd class="It-tag">For debugging the MPI master/worker version: pause after
      start, to enable the developer to attach debuggers to the running master
      and worker(s) processes. Send SIGCONT signal to release the pause. (Under
      gdb: <i>(gdb) signal SIGCONT</i>) (Only available if optional MPI support
      was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mpi</b></dt>
  <dd class="It-tag">Run in MPI master/worker mode, using <i>mpirun</i>. (Only
      available if optional MPI support was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>hmmer(1)</b> for a master man page with a list of all the individual man
  pages for programs in the HMMER package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your HMMER
  distribution (Userguide.pdf); or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2015 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2015</td>
    <td class="foot-os">HMMER 3.1b2</td>
  </tr>
</table>
</body>
</html>
