<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:19:15 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>jackhmmer(1) HMMER Manual jackhmmer(1)</p>

<p style="margin-top: 1em">NAME <br>
jackhmmer - iteratively search sequence(s) against a protein
database</p>

<p style="margin-top: 1em">SYNOPSIS <br>
jackhmmer [options] &lt;seqfile&gt; &lt;seqdb&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
jackhmmer iteratively searches each query sequence in
&lt;seqfile&gt; against the target sequence(s) in
&lt;seqdb&gt;. The first iteration is identical to a phmmer
search. For the next <br>
iteration, a multiple alignment of the query together with
all target sequences satisfying inclusion thresholds is
assembled, a profile is constructed from this alignment
(iden&acirc; <br>
tical to using hmmbuild on the alignment), and profile
search of the &lt;seqdb&gt; is done (identical to an
hmmsearch with the profile).</p>

<p style="margin-top: 1em">The query &lt;seqfile&gt; may be
&rsquo;-&rsquo; (a dash character), in which case the query
sequences are read from a &lt;stdin&gt; pipe instead of from
a file. The &lt;seqdb&gt; cannot be read from a <br>
&lt;stdin&gt; stream, because jackhmmer needs to do multiple
passes over the database.</p>

<p style="margin-top: 1em">The output format is designed to
be human-readable, but is often so voluminous that reading
it is impractical, and parsing it is a pain. The --tblout
and --domtblout options save <br>
output in simple tabular formats that are concise and easier
to parse. The -o option allows redirecting the main output,
including throwing it away in /dev/null.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
all available options.</p>

<p style="margin-top: 1em">-N &lt;n&gt; Set the maximum
number of iterations to &lt;n&gt;. The default is 5. If N=1,
the result is equivalent to a phmmer search.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING OUTPUT <br>
By default, output for each iteration appears on stdout in a
somewhat human readable, somewhat parseable format. These
options allow redirecting that output or saving additional
<br>
kinds of output to files, including checkpoint files for
each iteration.</p>

<p style="margin-top: 1em">-o &lt;f&gt; Direct the
human-readable output to a file &lt;f&gt;.</p>

<p style="margin-top: 1em">-A &lt;f&gt; After the final
iteration, save an annotated multiple alignment of all hits
satisfying inclusion thresholds (also including the original
query) to &lt;f&gt; in Stockholm format.</p>

<p style="margin-top: 1em">--tblout &lt;f&gt; <br>
After the final iteration, save a tabular summary of top
sequence hits to &lt;f&gt; in a readily parseable, columnar,
whitespace-delimited format.</p>

<p style="margin-top: 1em">--domtblout &lt;f&gt; <br>
After the final iteration, save a tabular summary of top
domain hits to &lt;f&gt; in a readily parseable, columnar,
whitespace-delimited format.</p>

<p style="margin-top: 1em">--chkhmm &lt;prefix&gt; <br>
At the start of each iteration, checkpoint the query HMM,
saving it to a file named &lt;prefix&gt;-&lt;n&gt;.hmm where
&lt;n&gt; is the iteration number (from 1..N).</p>

<p style="margin-top: 1em">--chkali &lt;prefix&gt; <br>
At the end of each iteration, checkpoint an alignment of all
domains satisfying inclusion thresholds (e.g. what will
become the query HMM for the next iteration), saving <br>
it to a file named &lt;checkpoint file
prefix&gt;-&lt;n&gt;.sto in Stockholm format, where
&lt;n&gt; is the iteration number (from 1..N).</p>

<p style="margin-top: 1em">--acc Use accessions instead of
names in the main output, where available for profiles
and/or sequences.</p>

<p style="margin-top: 1em">--noali <br>
Omit the alignment section from the main output. This can
greatly reduce the output volume.</p>

<p style="margin-top: 1em">--notextw <br>
Unlimit the length of each line in the main output. The
default is a limit of 120 characters per line, which helps
in displaying the output cleanly on terminals and in <br>
editors, but can truncate target profile description
lines.</p>

<p style="margin-top: 1em">--textw &lt;n&gt; <br>
Set the main output&rsquo;s line length limit to &lt;n&gt;
characters per line. The default is 120.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING SINGLE
SEQUENCE SCORING (FIRST ITERATION) <br>
By default, the first iteration uses a search model
constructed from a single query sequence. This model is
constructed using a standard 20x20 substitution matrix for
residue <br>
probabilities, and two additional parameters for
position-independent gap open and gap extend probabilities.
These options allow the default single-sequence scoring
parameters to <br>
be changed.</p>

<p style="margin-top: 1em">--popen &lt;x&gt; <br>
Set the gap open probability for a single sequence query
model to &lt;x&gt;. The default is 0.02. &lt;x&gt; must be
&gt;= 0 and &lt; 0.5.</p>

<p style="margin-top: 1em">--pextend &lt;x&gt; <br>
Set the gap extend probability for a single sequence query
model to &lt;x&gt;. The default is 0.4. &lt;x&gt; must be
&gt;= 0 and &lt; 1.0.</p>

<p style="margin-top: 1em">--mx &lt;s&gt; <br>
Obtain residue alignment probabilities from the built-in
substitution matrix named &lt;s&gt;. Several standard
matrices are built-in, and do not need to be read from
files. <br>
The matrix name &lt;s&gt; can be PAM30, PAM70, PAM120,
PAM240, BLOSUM45, BLOSUM50, BLOSUM62, BLOSUM80, or BLOSUM90.
Only one of the --mx and --mxfile options may be used.</p>

<p style="margin-top: 1em">--mxfile &lt;mxfile&gt; <br>
Obtain residue alignment probabilities from the substitution
matrix in file &lt;mxfile&gt;. The default score matrix is
BLOSUM62 (this matrix is internal to HMMER and does not <br>
have to be available as a file). The format of a
substitution matrix &lt;mxfile&gt; is the standard format
accepted by BLAST, FASTA, and other sequence analysis
software.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING REPORTING
THRESHOLDS <br>
Reporting thresholds control which hits are reported in
output files (the main output, --tblout, and --domtblout).
In each iteration, sequence hits and domain hits are ranked
by <br>
statistical significance (E-value) and output is generated
in two sections called per-target and per-domain output. In
per-target output, by default, all sequence hits with an E-
<br>
value &lt;= 10 are reported. In the per-domain output, for
each target that has passed per-target reporting thresholds,
all domains satisfying per-domain reporting thresholds are
<br>
reported. By default, these are domains with conditional
E-values of &lt;= 10. The following options allow you to
change the default E-value reporting thresholds, or to use
bit <br>
score thresholds instead.</p>

<p style="margin-top: 1em">-E &lt;x&gt; Report sequences
with E-values &lt;= &lt;x&gt; in per-sequence output. The
default is 10.0.</p>

<p style="margin-top: 1em">-T &lt;x&gt; Use a bit score
threshold for per-sequence output instead of an E-value
threshold (any setting of -E is ignored). Report sequences
with a bit score of &gt;= &lt;x&gt;. By default <br>
this option is unset.</p>

<p style="margin-top: 1em">-Z &lt;x&gt; Declare the total
size of the database to be &lt;x&gt; sequences, for purposes
of E-value calculation. Normally E-values are calculated
relative to the size of the database you <br>
actually searched (e.g. the number of sequences in
target_seqdb). In some cases (for instance, if you&rsquo;ve
split your target sequence database into multiple files for
par&acirc; <br>
allelization of your search), you may know better what the
actual size of your search space is.</p>

<p style="margin-top: 1em">--domE &lt;x&gt; <br>
Report domains with conditional E-values &lt;= &lt;x&gt; in
per-domain output, in addition to the top-scoring domain per
significant sequence hit. The default is 10.0.</p>

<p style="margin-top: 1em">--domT &lt;x&gt; <br>
Use a bit score threshold for per-domain output instead of
an E-value threshold (any setting of --domT is ignored).
Report domains with a bit score of &gt;= &lt;x&gt; in
per-domain <br>
output, in addition to the top-scoring domain per
significant sequence hit. By default this option is
unset.</p>

<p style="margin-top: 1em">--domZ &lt;x&gt; <br>
Declare the number of significant sequences to be &lt;x&gt;
sequences, for purposes of conditional E-value calculation
for additional domain significance. Normally conditional
<br>
E-values are calculated relative to the number of sequences
passing per-sequence reporting threshold.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING INCLUSION
THRESHOLDS <br>
Inclusion thresholds control which hits are included in the
multiple alignment and profile constructed for the next
search iteration. By default, a sequence must have a per-
<br>
sequence E-value of &lt;= 0.001 (see -E option) to be
included, and any additional domains in it besides the
top-scoring one must have a conditional E-value of &lt;=
0.001 (see --domE <br>
option). The difference between reporting thresholds and
inclusion thresholds is that inclusion thresholds control
which hits actually get used in the next iteration (or the
<br>
final output multiple alignment if the -A option is used),
whereas reporting thresholds control what you see in output.
Reporting thresholds are generally more loose so you can
<br>
see borderline hits in the top of the noise that might be of
interest.</p>

<p style="margin-top: 1em">--incE &lt;x&gt; <br>
Include sequences with E-values &lt;= &lt;x&gt; in
subsequent iteration or final alignment output by -A. The
default is 0.001.</p>

<p style="margin-top: 1em">--incT &lt;x&gt; <br>
Use a bit score threshold for per-sequence inclusion instead
of an E-value threshold (any setting of --incE is ignored).
Include sequences with a bit score of &gt;= &lt;x&gt;. By
<br>
default this option is unset.</p>

<p style="margin-top: 1em">--incdomE &lt;x&gt; <br>
Include domains with conditional E-values &lt;= &lt;x&gt; in
subsequent iteration or final alignment output by -A, in
addition to the top-scoring domain per significant sequence
<br>
hit. The default is 0.001.</p>

<p style="margin-top: 1em">--incdomT &lt;x&gt; <br>
Use a bit score threshold for per-domain inclusion instead
of an E-value threshold (any setting of --incT is ignored).
Include domains with a bit score of &gt;= &lt;x&gt;. By <br>
default this option is unset.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING ACCELERATION
HEURISTICS <br>
HMMER3 searches are accelerated in a three-step filter
pipeline: the MSV filter, the Viterbi filter, and the
Forward filter. The first filter is the fastest and most
approximate; <br>
the last is the full Forward scoring algorithm, slowest but
most accurate. There is also a bias filter step between MSV
and Viterbi. Targets that pass all the steps in the
accel&acirc; <br>
eration pipeline are then subjected to postprocessing --
domain identification and scoring using the Forward/Backward
algorithm.</p>

<p style="margin-top: 1em">Essentially the only free
parameters that control HMMER&rsquo;s heuristic filters are
the P-value thresholds controlling the expected fraction of
nonhomologous sequences that pass the <br>
filters. Setting the default thresholds higher will pass a
higher proportion of nonhomologous sequence, increasing
sensitivity at the expense of speed; conversely, setting
lower <br>
P-value thresholds will pass a smaller proportion,
decreasing sensitivity and increasing speed. Setting a
filter&rsquo;s P-value threshold to 1.0 means it will
passing all sequences, <br>
and effectively disables the filter.</p>

<p style="margin-top: 1em">Changing filter thresholds only
removes or includes targets from consideration; changing
filter thresholds does not alter bit scores, E-values, or
alignments, all of which are <br>
determined solely in postprocessing.</p>

<p style="margin-top: 1em">--max Maximum sensitivity. Turn
off all filters, including the bias filter, and run full
Forward/Backward postprocessing on every target. This
increases sensitivity slightly, <br>
at a large cost in speed.</p>

<p style="margin-top: 1em">--F1 &lt;x&gt; <br>
First filter threshold; set the P-value threshold for the
MSV filter step. The default is 0.02, meaning that roughly
2% of the highest scoring nonhomologous targets are <br>
expected to pass the filter.</p>

<p style="margin-top: 1em">--F2 &lt;x&gt; <br>
Second filter threshold; set the P-value threshold for the
Viterbi filter step. The default is 0.001.</p>

<p style="margin-top: 1em">--F3 &lt;x&gt; <br>
Third filter threshold; set the P-value threshold for the
Forward filter step. The default is 1e-5.</p>

<p style="margin-top: 1em">--nobias <br>
Turn off the bias filter. This increases sensitivity
somewhat, but can come at a high cost in speed, especially
if the query has biased residue composition (such as a <br>
repetitive sequence region, or if it is a membrane protein
with large regions of hydrophobicity). Without the bias
filter, too many sequences may pass the filter with <br>
biased queries, leading to slower than expected performance
as the computationally intensive Forward/Backward algorithms
shoulder an abnormally heavy load.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING PROFILE
CONSTRUCTION (LATER ITERATIONS) <br>
These options control how consensus columns are defined in
multiple alignments when building profiles. By default,
jackhmmer always includes your original query sequence in
the <br>
alignment result at every iteration, and consensus positions
are defined by that query sequence: that is, a default
jackhmmer profile is always the same length as your original
<br>
query, at every iteration.</p>

<p style="margin-top: 1em">--fast Define consensus columns
as those that have a fraction &gt;= symfrac of residues as
opposed to gaps. (See below for the --symfrac option.)
Although this is the default pro&acirc; <br>
file construction option elsewhere (in hmmbuild, in
particular), it may have undesirable effects in jackhmmer,
because a profile could iteratively walk in sequence space
<br>
away from your original query, leaving few or no consensus
columns corresponding to its residues.</p>

<p style="margin-top: 1em">--hand Define consensus columns
in next profile using reference annotation to the multiple
alignment. jackhmmer propagates reference annotation from
the previous profile to the <br>
multiple alignment, and thence to the next profile. This is
the default.</p>

<p style="margin-top: 1em">--symfrac &lt;x&gt; <br>
Define the residue fraction threshold necessary to define a
consensus column when using the --fast option. The default
is 0.5. The symbol fraction in each column is calcu&acirc;
<br>
lated after taking relative sequence weighting into account,
and ignoring gap characters corresponding to ends of
sequence fragments (as opposed to internal inser&acirc; <br>
tions/deletions). Setting this to 0.0 means that every
alignment column will be assigned as consensus, which may be
useful in some cases. Setting it to 1.0 means that <br>
only columns that include 0 gaps (internal
insertions/deletions) will be assigned as consensus.</p>

<p style="margin-top: 1em">--fragthresh &lt;x&gt; <br>
We only want to count terminal gaps as deletions if the
aligned sequence is known to be full-length, not if it is a
fragment (for instance, because only part of it was <br>
sequenced). HMMER uses a simple rule to infer fragments: if
the sequence length L is less than or equal to a fraction
&lt;x&gt; times the alignment length in columns, then the
<br>
sequence is handled as a fragment. The default is 0.5.
Setting --fragthresh0 will define no (nonempty) sequence as
a fragment; you might want to do this if you know <br>
you&rsquo;ve got a carefully curated alignment of
full-length sequences. Setting --fragthresh1 will define all
sequences as fragments; you might want to do this if you
know <br>
your alignment is entirely composed of fragments, such as
translated short reads in metagenomic shotgun data.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING RELATIVE
WEIGHTS <br>
Whenever a profile is built from a multiple alignment, HMMER
uses an ad hoc sequence weighting algorithm to downweight
closely related sequences and upweight distantly related
<br>
ones. This has the effect of making models less biased by
uneven phylogenetic representation. For example, two
identical sequences would typically each receive half the
weight <br>
that one sequence would (and this is why jackhmmer
isn&rsquo;t concerned about always including your original
query sequence in each iteration&rsquo;s alignment, even if
it finds it again in <br>
the database you&rsquo;re searching). These options control
which algorithm gets used.</p>

<p style="margin-top: 1em">--wpb Use the Henikoff
position-based sequence weighting scheme [Henikoff and
Henikoff, J. Mol. Biol. 243:574, 1994]. This is the
default.</p>

<p style="margin-top: 1em">--wgsc Use the
Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et
al, J. Mol. Biol. 235:1067, 1994].</p>

<p style="margin-top: 1em">--wblosum <br>
Use the same clustering scheme that was used to weight data
in calculating BLOSUM subsitution matrices [Henikoff and
Henikoff, Proc. Natl. Acad. Sci 89:10915, 1992]. <br>
Sequences are single-linkage clustered at an identity
threshold (default 0.62; see --wid) and within each cluster
of c sequences, each sequence gets relative weight 1/c.</p>

<p style="margin-top: 1em">--wnone <br>
No relative weights. All sequences are assigned uniform
weight.</p>

<p style="margin-top: 1em">--wid &lt;x&gt; <br>
Sets the identity threshold used by single-linkage
clustering when using --wblosum. Invalid with any other
weighting scheme. Default is 0.62.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING EFFECTIVE
SEQUENCE NUMBER <br>
After relative weights are determined, they are normalized
to sum to a total effective sequence number, eff_nseq. This
number may be the actual number of sequences in the
align&acirc; <br>
ment, but it is almost always smaller than that. The default
entropy weighting method (--eent) reduces the effective
sequence number to reduce the information content (relative
<br>
entropy, or average expected score on true homologs) per
consensus position. The target relative entropy is
controlled by a two-parameter function, where the two
parameters are <br>
settable with --ere and --esigma.</p>

<p style="margin-top: 1em">--eent Adjust effective sequence
number to achieve a specific relative entropy per position
(see --ere). This is the default.</p>

<p style="margin-top: 1em">--eclust <br>
Set effective sequence number to the number of
single-linkage clusters at a specific identity threshold
(see --eid). This option is not recommended; it&rsquo;s for
experiments <br>
evaluating how much better --eent is.</p>

<p style="margin-top: 1em">--enone <br>
Turn off effective sequence number determination and just
use the actual number of sequences. One reason you might
want to do this is to try to maximize the relative <br>
entropy/position of your model, which may be useful for
short models.</p>

<p style="margin-top: 1em">--eset &lt;x&gt; <br>
Explicitly set the effective sequence number for all models
to &lt;x&gt;.</p>

<p style="margin-top: 1em">--ere &lt;x&gt; <br>
Set the minimum relative entropy/position target to
&lt;x&gt;. Requires --eent. Default depends on the sequence
alphabet; for protein sequences, it is 0.59
bits/position.</p>

<p style="margin-top: 1em">--esigma &lt;x&gt; <br>
Sets the minimum relative entropy contributed by an entire
model alignment, over its whole length. This has the effect
of making short models have higher relative entropy <br>
per position than --ere alone would give. The default is
45.0 bits.</p>

<p style="margin-top: 1em">--eid &lt;x&gt; <br>
Sets the fractional pairwise identity cutoff used by single
linkage clustering with the --eclust option. The default is
0.62.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING PRIORS <br>
In profile construction, by default, weighted counts are
converted to mean posterior probability parameter estimates
using mixture Dirichlet priors. Default mixture Dirichlet
<br>
prior parameters for protein models and for nucleic acid
(RNA and DNA) models are built in. The following options
allow you to override the default priors.</p>

<p style="margin-top: 1em">--pnone Don&rsquo;t use any
priors. Probability parameters will simply be the observed
frequencies, after relative sequence weighting.</p>

<p style="margin-top: 1em">--plaplace Use a Laplace +1
prior in place of the default mixture Dirichlet prior.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING E-VALUE
CALIBRATION <br>
Estimating the location parameters for the expected score
distributions for MSV filter scores, Viterbi filter scores,
and Forward scores requires three short random sequence
sim&acirc; <br>
ulations.</p>

<p style="margin-top: 1em">--EmL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter mu for MSV filter E-values. Default is
200.</p>

<p style="margin-top: 1em">--EmN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter mu for MSV filter E-values. Default
is 200.</p>

<p style="margin-top: 1em">--EvL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter mu for Viterbi filter E-values. Default
is 200.</p>

<p style="margin-top: 1em">--EvN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter mu for Viterbi filter E-values.
Default is 200.</p>

<p style="margin-top: 1em">--EfL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter tau for Forward E-values. Default is
100.</p>

<p style="margin-top: 1em">--EfN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter tau for Forward E-values. Default is
200.</p>

<p style="margin-top: 1em">--Eft &lt;x&gt; <br>
Sets the tail mass fraction to fit in the simulation that
estimates the location parameter tau for Forward evalues.
Default is 0.04.</p>

<p style="margin-top: 1em">OTHER OPTIONS <br>
--nonull2 <br>
Turn off the null2 score corrections for biased
composition.</p>

<p style="margin-top: 1em">-Z &lt;x&gt; Assert that the
total number of targets in your searches is &lt;x&gt;, for
the purposes of per-sequence E-value calculations, rather
than the actual number of targets seen.</p>

<p style="margin-top: 1em">--domZ &lt;x&gt; <br>
Assert that the total number of targets in your searches is
&lt;x&gt;, for the purposes of per-domain conditional
E-value calculations, rather than the number of targets that
<br>
passed the reporting thresholds.</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Seed the random number generator with &lt;n&gt;, an integer
&gt;= 0. If &lt;n&gt; is &gt;0, any stochastic simulations
will be reproducible; the same command will give the same
results. <br>
If &lt;n&gt; is 0, the random number generator is seeded
arbitrarily, and stochastic simulations will vary from run
to run of the same command. The default seed is 42.</p>

<p style="margin-top: 1em">--qformat &lt;s&gt; <br>
Declare that the input query_seqfile is in format &lt;s&gt;.
Accepted sequence file formats include FASTA, EMBL, GenBank,
DDBJ, UniProt, Stockholm, and SELEX. Default is to <br>
autodetect the format of the file.</p>

<p style="margin-top: 1em">--tformat &lt;s&gt; <br>
Declare that the input target_seqdb is in format &lt;s&gt;.
Accepted sequence file formats include FASTA, EMBL, GenBank,
DDBJ, UniProt, Stockholm, and SELEX. Default is to <br>
autodetect the format of the file.</p>

<p style="margin-top: 1em">--cpu &lt;n&gt; <br>
Set the number of parallel worker threads to &lt;n&gt;. By
default, HMMER sets this to the number of CPU cores it
detects in your machine - that is, it tries to maximize the
<br>
use of your available processor cores. Setting &lt;n&gt;
higher than the number of available cores is of little if
any value, but you may want to set it to something less. You
<br>
can also control this number by setting an environment
variable, HMMER_NCPU.</p>

<p style="margin-top: 1em">This option is only available if
HMMER was compiled with POSIX threads support. This is the
default, but it may have been turned off at compile-time for
your site or <br>
machine for some reason.</p>

<p style="margin-top: 1em">--stall <br>
For debugging the MPI master/worker version: pause after
start, to enable the developer to attach debuggers to the
running master and worker(s) processes. Send SIGCONT <br>
signal to release the pause. (Under gdb: (gdb) signal
SIGCONT) (Only available if optional MPI support was enabled
at compile-time.)</p>

<p style="margin-top: 1em">--mpi Run in MPI master/worker
mode, using mpirun. (Only available if optional MPI support
was enabled at compile-time.)</p>

<p style="margin-top: 1em">SEE ALSO <br>
See hmmer(1) for a master man page with a list of all the
individual man pages for programs in the HMMER package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
jackhmmer(1)</p>
<hr>
</body>
</html>
