<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>JELLYFISH(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">JELLYFISH(1)</td>
    <td class="head-vol">k-mer counter </td>
    <td class="head-rtitle">JELLYFISH(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
Jellyfish is a software to count k-mers in DNA sequences.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
jellyfish count [ <b>-o</b><i>prefix</i>] [ <b>-m</b><i>merlength</i>] [
  <b>-t</b> <i>threads</i>] [ <b>-s</b><i>hashsize</i>] [ <b>--both-strands</b>]
  <i>fasta</i> [ <i>fasta ...</i> <i></i>]
<div>&#x00A0;</div>
jellyfish merge <i>hash1</i> <i>hash2</i> ...
<div>&#x00A0;</div>
jellyfish dump <i>hash</i>
<div>&#x00A0;</div>
jellyfish stats <i>hash</i>
<div>&#x00A0;</div>
jellyfish histo [ <b>-h</b><i>high</i>] [ <b>-l</b><i>low</i>] [
  <b>-i</b><i>increment</i>] <i>hash</i>
<div>&#x00A0;</div>
jellyfish query <i>hash</i>
<div>&#x00A0;</div>
jellyfish cite
<div>&#x00A0;</div>
<div class="Pp"></div>
Plus equivalent version for Quake mode: qhisto, qdump and qmerge.
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Jellyfish is a k-mer counter based on a multi-threaded hash table
  implementation.
<h2 class="Ss" title="Ss" id="COUNTING_AND_MERGING"><a class="selflink" href="#COUNTING_AND_MERGING">COUNTING
  AND MERGING</a></h2>
To count k-mers, use a command like:
<div class="Pp"></div>
<pre>
jellyfish count -m 22 -o output -c 3 -s 10000000 -t 32 input.fasta
<div class="Pp"></div>
</pre>
<div class="Pp"></div>
This will count the the 22-mers in input.fasta with 32 threads. The counter
  field in the hash uses only 3 bits and the hash has at least 10 million
  entries.
<div class="Pp"></div>
The output files will be named output_0, output_1, etc. (the prefix is specified
  with the <b>-o</b> switch). If the hash is large enough (has specified by the
  <b>-s</b> switch) to fit all the k-mers, there will be only one output file
  named output_0. If the hash filled up before all the mers were read, the hash
  is dumped to disk, zeroed out and reading in mers resumes. Multiple
  intermediary files will be present on the disks, named output_0, output_1,
  etc.
<div class="Pp"></div>
To obtain correct results from the other sub-commands (such as histo, stats,
  etc.), the multiple output files, if any, need to be merged into one with the
  merge command. For example with the following command:
<div class="Pp"></div>
<pre>
jellyfish merge -o output.jf output\_*
<div class="Pp"></div>
</pre>
<div class="Pp"></div>
Should you get many intermediary output files (say hundreds), the size of the
  hash table is too small. Rerunning Jellyfish with a larger size (option
  <b>-s</b>) is probably faster than merging all the intermediary files.
<h2 class="Ss" title="Ss" id="ORIENTATION"><a class="selflink" href="#ORIENTATION">ORIENTATION</a></h2>
When the orientation of the sequences in the input fasta file is not known, e.g.
  in sequencing reads, using <b>--both-strands</b> ( <b>-C</b>) makes the most
  sense.
<div class="Pp"></div>
For any k-mer m, its canonical representation is m itself or its
  reverse-complement, whichever comes first lexicographically. With the option
  <b>-C</b>, only the canonical representation of the mers are stored in the
  hash and the count value is the number of occurrences of both the mer and its
  reverse-complement.
<h2 class="Ss" title="Ss" id="CHOOSING_THE_HASH_SIZE"><a class="selflink" href="#CHOOSING_THE_HASH_SIZE">CHOOSING
  THE HASH SIZE</a></h2>
To achieve the best performance, a minimum number of intermediary files should
  be written to disk. So the parameter <b>-s</b> should be chosen to fit as many
  k-mers as possible (ideally all of them) while still fitting in memory.
<div class="Pp"></div>
We consider to examples: counting mers in sequencing reads and in a finished
  genome.
<div class="Pp"></div>
First, suppose we count k-mers in short sequencing reads: there are n reads and
  there is an average of 1 error per reads where each error generates k unique
  mers. If the genome size is G, the size of the hash (option <b>-s</b>) to fit
  all k-mers at once is estimated to: $(G + k*n)/0.8$. The division by 0.8
  compensates for the maximum usage of approximately $80%$ of the hash table.
<div class="Pp"></div>
On the other hand, when counting k-mers in an assembled sequence of length G,
  setting <b>-s</b> to G is appropriate.
<div class="Pp"></div>
As a matter of convenience, Jellyfish understands ISO suffixes for the size of
  the hash. Hence '-s 10M' stands 10 million entries while '-s 50G' stands for
  50 billion entries.
<div class="Pp"></div>
The actual memory usage of the hash table can be computed as follow. The actual
  size of the hash will be rounded up to the next power of 2: s=2^l. The
  parameter r is such that the maximum reprobe value ( <b>-p</b>) plus one is
  less than 2^r. Then the memory usage per entry in the hash is (in bits, not
  bytes) 2k-l+r+1. The total memory usage of the hash table in bytes is:
  2^l*(2k-l+r+1)/8.
<h2 class="Ss" title="Ss" id="CHOOSING_THE_COUNTING_FIELD_SIZE"><a class="selflink" href="#CHOOSING_THE_COUNTING_FIELD_SIZE">CHOOSING
  THE COUNTING FIELD SIZE</a></h2>
To save space, the hash table supports variable length counter, i.e. a k-mer
  occurring only a few times will use a small counter, a k-mer occurring many
  times will used multiple entries in the hash.
<div class="Pp"></div>
Important: the size of the couting field does NOT change the result, it only
  impacts the amount of memory used. In particular, there is no maximum value in
  the hash. Even if the counting field uses 5 bits, a k-mer occuring 2 million
  times will have a value reported of 2 million (i.e., it is not capped at 2^5).
<div class="Pp"></div>
The <b>-c</b> specify the length (in bits) of the counting field. The trade off
  is as follows: a low value will save space per entry in the hash but can
  potentially increase the number of entries used, hence maybe requiring a
  larger hash.
<div class="Pp"></div>
In practice, use a value for <b>-c</b> so that most of you k-mers require only 1
  entry. For example, to count k-mers in a genome, where most of the sequence is
  unique, use <b>-c</b><i>1</i> or <b>-c</b><i>2</i>. For sequencing reads, use
  a value for <b>-c</b> large enough to counts up to twice the coverage. For
  example, if the coverage is 10X, choose a counter length of 5 (
  <b>-c</b><i>5</i>) as $2^5 &gt; 20$.
<h1 class="Sh" title="Sh" id="SUBCOMMANDS_AND_OPTIONS"><a class="selflink" href="#SUBCOMMANDS_AND_OPTIONS">SUBCOMMANDS
  AND OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="COUNT"><a class="selflink" href="#COUNT">COUNT</a></h2>
Usage: jellyfish count [options] file:path+
<div class="Pp"></div>
Count k-mers or qmers in fasta or fastq files
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>,</dt>
  <dd class="It-tag"><b>--mer-len</b><i>=uint32</i>
    <br/>
     *Length of mer</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>,</dt>
  <dd class="It-tag"><b>--size</b><i>=uint64</i>
    <br/>
     *Hash size</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>,</dt>
  <dd class="It-tag"><b>--threads</b><i>=uint32</i>
    <br/>
     Number of threads (1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=string</i>
    <br/>
     Output prefix (mer_counts)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>,</dt>
  <dd class="It-tag"><b>--counter-len</b><i>=Length</i>
    <br/>
     in bits Length of counting field (7)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--out-counter-len</b><i>=Length</i></dt>
  <dd class="It-tag">
    <br/>
     in bytes Length of counter field in output (4)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b>,<b>--both-strands</b></dt>
  <dd class="It-tag">
    <br/>
     Count both strand, canonical representation (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b>,</dt>
  <dd class="It-tag"><b>--reprobes</b><i>=uint32</i>
    <br/>
     Maximum number of reprobes (62)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b>,<b>--raw</b></dt>
  <dd class="It-tag">
    <br/>
     Write raw database (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b>,<b>--quake</b></dt>
  <dd class="It-tag">
    <br/>
     Quake compatibility mode (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--quality-start</b><i>=uint32</i></dt>
  <dd class="It-tag">
    <br/>
     Starting ASCII for quality values (64)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-quality</b><i>=uint32</i></dt>
  <dd class="It-tag">
    <br/>
     Minimum quality. A base with lesser quality becomes an N (0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b>,</dt>
  <dd class="It-tag"><b>--lower-count</b><i>=uint64</i>
    <br/>
     Don't output k-mer with count &lt; lower-count</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b>,</dt>
  <dd class="It-tag"><b>--upper-count</b><i>=uint64</i>
    <br/>
     Don't output k-mer with count &gt; upper-count</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--matrix</b><i>=Matrix</i></dt>
  <dd class="It-tag">
    <br/>
     file Hash function binary matrix</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--timing</b><i>=Timing</i></dt>
  <dd class="It-tag">
    <br/>
     file Print timing information</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stats</b><i>=Stats</i></dt>
  <dd class="It-tag">
    <br/>
     file Print stats</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,<b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--full-help</b></dt>
  <dd class="It-tag">
    <br/>
     Detailed help</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="STATS"><a class="selflink" href="#STATS">STATS</a></h2>
Usage: jellyfish stats [options] db:path
<div class="Pp"></div>
Statistics
<div class="Pp"></div>
Display some statistics about the k-mers in the hash:
<div class="Pp"></div>
Unique: Number of k-mers which occur only once. Distinct: Number of k-mers, not
  counting multiplicity. Total: Number of k-mers, including multiplicity.
  Max_count: Maximum number of occurrence of a k-mer.
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b>,</dt>
  <dd class="It-tag"><b>--lower-count</b><i>=uint64</i>
    <br/>
     Don't consider k-mer with count &lt; lower-count</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b>,</dt>
  <dd class="It-tag"><b>--upper-count</b><i>=uint64</i>
    <br/>
     Don't consider k-mer with count &gt; upper-count</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>,<b>--verbose</b></dt>
  <dd class="It-tag">
    <br/>
     Verbose (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=string</i>
    <br/>
     Output file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,<b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--full-help</b></dt>
  <dd class="It-tag">
    <br/>
     Detailed help</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="HISTO"><a class="selflink" href="#HISTO">HISTO</a></h2>
Usage: jellyfish histo [options] db:path
<div class="Pp"></div>
Create an histogram of k-mer occurrences
<div class="Pp"></div>
Create an histogram with the number of k-mers having a given count. In bucket
  'i' are tallied the k-mers which have a count 'c' satisfying 'low+i*inc &lt;=
  c &lt; low+(i+1)*inc'. Buckets in the output are labeled by the low end point
  (low+i*inc).
<div class="Pp"></div>
The last bucket in the output behaves as a catchall: it tallies all k-mers with
  a count greater or equal to the low end point of this bucket.
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b>,</dt>
  <dd class="It-tag"><b>--low</b><i>=uint64</i>
    <br/>
     Low count value of histogram (1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,</dt>
  <dd class="It-tag"><b>--high</b><i>=uint64</i>
    <br/>
     High count value of histogram (10000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b>,</dt>
  <dd class="It-tag"><b>--increment</b><i>=uint64</i>
    <br/>
     Increment value for buckets (1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>,</dt>
  <dd class="It-tag"><b>--threads</b><i>=uint32</i>
    <br/>
     Number of threads (1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>,<b>--full</b></dt>
  <dd class="It-tag">
    <br/>
     Full histo. Don't skip count 0. (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=string</i>
    <br/>
     Output file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>,<b>--verbose</b></dt>
  <dd class="It-tag">
    <br/>
     Output information (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--full-help</b></dt>
  <dd class="It-tag">
    <br/>
     Detailed help</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="DUMP"><a class="selflink" href="#DUMP">DUMP</a></h2>
Usage: jellyfish dump [options] db:path
<div class="Pp"></div>
Dump k-mer counts
<div class="Pp"></div>
By default, dump in a fasta format where the header is the count and the
  sequence is the sequence of the k-mer. The column format is a 2 column output:
  k-mer count.
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>,<b>--column</b></dt>
  <dd class="It-tag">
    <br/>
     Column format (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>,<b>--tab</b></dt>
  <dd class="It-tag">
    <br/>
     Tab separator (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b>,</dt>
  <dd class="It-tag"><b>--lower-count</b><i>=uint64</i>
    <br/>
     Don't output k-mer with count &lt; lower-count</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b>,</dt>
  <dd class="It-tag"><b>--upper-count</b><i>=uint64</i>
    <br/>
     Don't output k-mer with count &gt; upper-count</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=string</i>
    <br/>
     Output file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,<b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="MERGE"><a class="selflink" href="#MERGE">MERGE</a></h2>
Usage: jellyfish merge [options] input:string+
<div class="Pp"></div>
Merge jellyfish databases
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>,</dt>
  <dd class="It-tag"><b>--buffer-size</b><i>=Buffer</i>
    <br/>
     length Length in bytes of input buffer (10000000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=string</i>
    <br/>
     Output file (mer_counts_merged.jf)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--out-counter-len</b><i>=uint32</i></dt>
  <dd class="It-tag">
    <br/>
     Length (in bytes) of counting field in output (4)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--out-buffer-size</b><i>=uint64</i></dt>
  <dd class="It-tag">
    <br/>
     Size of output buffer per thread (10000000)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>,<b>--verbose</b></dt>
  <dd class="It-tag">
    <br/>
     Be verbose (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,<b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="QUERY"><a class="selflink" href="#QUERY">QUERY</a></h2>
Usage: jellyfish query [options] db:path
<div class="Pp"></div>
Query from a compacted database
<div class="Pp"></div>
Query a hash. It reads k-mers from the standard input and write the counts on
  the standard output.
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b>,<b>--both-strands</b></dt>
  <dd class="It-tag">
    <br/>
     Both strands (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>,<b>--cary-bit</b></dt>
  <dd class="It-tag">
    <br/>
     Value field as the cary bit information (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b>,</dt>
  <dd class="It-tag"><b>--input</b><i>=file</i>
    <br/>
     Input file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=file</i>
    <br/>
     Output file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,<b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="QHISTO"><a class="selflink" href="#QHISTO">QHISTO</a></h2>
Usage: jellyfish qhisto [options] db:string
<div class="Pp"></div>
Create an histogram of k-mer occurences
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b>,</dt>
  <dd class="It-tag"><b>--low</b><i>=double</i>
    <br/>
     Low count value of histogram (0.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,</dt>
  <dd class="It-tag"><b>--high</b><i>=double</i>
    <br/>
     High count value of histogram (10000.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b>,</dt>
  <dd class="It-tag"><b>--increment</b><i>=double</i>
    <br/>
     Increment value for buckets (1.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>,<b>--full</b></dt>
  <dd class="It-tag">
    <br/>
     Full histo. Don't skip count 0. (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="QDUMP"><a class="selflink" href="#QDUMP">QDUMP</a></h2>
Usage: jellyfish qdump [options] db:path
<div class="Pp"></div>
Dump k-mer from a qmer database
<div class="Pp"></div>
By default, dump in a fasta format where the header is the count and the
  sequence is the sequence of the k-mer. The column format is a 2 column output:
  k-mer count.
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>,<b>--column</b></dt>
  <dd class="It-tag">
    <br/>
     Column format (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>,<b>--tab</b></dt>
  <dd class="It-tag">
    <br/>
     Tab separator (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b>,</dt>
  <dd class="It-tag"><b>--lower-count</b><i>=double</i>
    <br/>
     Don't output k-mer with count &lt; lower-count</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b>,</dt>
  <dd class="It-tag"><b>--upper-count</b><i>=double</i>
    <br/>
     Don't output k-mer with count &gt; upper-count</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>,<b>--verbose</b></dt>
  <dd class="It-tag">
    <br/>
     Be verbose (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=string</i>
    <br/>
     Output file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,<b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="QMERGE"><a class="selflink" href="#QMERGE">QMERGE</a></h2>
Usage: jellyfish merge [options] db:string+
<div class="Pp"></div>
Merge quake databases
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>,</dt>
  <dd class="It-tag"><b>--size</b><i>=uint64</i>
    <br/>
     *Merged hash table size</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>,</dt>
  <dd class="It-tag"><b>--mer-len</b><i>=uint32</i>
    <br/>
     *Mer length</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=string</i>
    <br/>
     Output file (merged.jf)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b>,</dt>
  <dd class="It-tag"><b>--reprobes</b><i>=uint32</i>
    <br/>
     Maximum number of reprobes (62)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,<b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--full-help</b></dt>
  <dd class="It-tag">
    <br/>
     Detailed help</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h2 class="Ss" title="Ss" id="CITE"><a class="selflink" href="#CITE">CITE</a></h2>
Usage: jellyfish cite [options]
<div class="Pp"></div>
How to cite Jellyfish's paper
<div class="Pp"></div>
Citation of paper
<div class="Pp"></div>
Options (default value in (), *required):
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b>,<b>--bibtex</b></dt>
  <dd class="It-tag">
    <br/>
     Bibtex format (false)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>,</dt>
  <dd class="It-tag"><b>--output</b><i>=string</i>
    <br/>
     Output file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--usage</b></dt>
  <dd class="It-tag">
    <br/>
     Usage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>,<b>--help</b></dt>
  <dd class="It-tag">
    <br/>
     This message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>,<b>--version</b></dt>
  <dd class="It-tag">
    <br/>
     Version</dd>
</dl>
<h1 class="Sh" title="Sh" id="VERSION"><a class="selflink" href="#VERSION">VERSION</a></h1>
Version: 1.1.4 of 2010/10/1
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>*</b></dt>
  <dd class="It-tag">jellyfish merge has not been parallelized and is relatively
      slow.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>*</b></dt>
  <dd class="It-tag">The hash table does not grow in memory automatically and
      jellyfish merge is not called automatically on the intermediary files (if
      any).</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT_&amp;_LICENSE"><a class="selflink" href="#COPYRIGHT_&amp;_LICENSE">COPYRIGHT
  &amp; LICENSE</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">Copyright</dt>
  <dd class="It-tag">(C)2010, Guillaume Marcais <b>guillaume@marcais.net</b> and
      Carl Kingsford <b>carlk@umiacs.umd.edu</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">License</dt>
  <dd class="It-tag">This program is free software: you can redistribute it
      and/or modify it under the terms of the GNU General Public License as
      published by the Free Software Foundation, either version 3 of the
      License, or (at your option) any later version.
    <div>&#x00A0;</div>
    This program is distributed in the hope that it will be useful, but WITHOUT
      ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
      FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
      more details.
    <div>&#x00A0;</div>
    You should have received a copy of the GNU General Public License along with
      this program. If not, see <b>http://www.gnu.org/licenses/</b>.</dd>
</dl>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
Guillaume Marcais
<div>&#x00A0;</div>
University of Maryland
<div>&#x00A0;</div>
<b>gmarcais@umd.edu</b>
<div class="Pp"></div>
Carl Kingsford
<div>&#x00A0;</div>
University of Maryland
<div>&#x00A0;</div>
<b>carlk@umiacs.umd.edu</b></div>
<table class="foot">
  <tr>
    <td class="foot-date">2010/10/1</td>
    <td class="foot-os">k-mer counter </td>
  </tr>
</table>
</body>
</html>
