<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>JOINGENES(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">JOINGENES(1)</td>
    <td class="head-vol">&#x00A0;</td>
    <td class="head-rtitle">JOINGENES(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
joingenes - merge several gene sets into one
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>joingenes</b> [parameters] --genesets=file1,file2,... --output=ofile
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
This program works in several steps:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;"> 1.divide the set of all transcripts into
  smaller sets, in which all transcripts are on the same sequence and are
  overlapping at least with one other transcript in this set (set is called
  &quot;overlap&quot;)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;"> 2.delete all duplications of transcripts and
  save the variant with the highest &quot;score&quot;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;"> 3.if sequence ranges are set for some
  transcripts, the program detects, whether the distance to that range is
  dangerously close</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;"> 4.join:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;if there is a transcript dangerously
  close to one/both end(s) of a sequence range, the program creates a copy
  without the corresponding terminal exon</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;if there is a transcript with start or
  stop codon in a set and a second one without this codon and they are
  &quot;joinable&quot;, than this step joins the corresponding terminal
  exons</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;"> 5.selection: selects the &quot;best&quot;
  gene structure out of all possible &quot;maximum&quot; gene structures
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;&quot;maximum&quot; gene structure is
  a set of transcripts from an overlap so that there is no other transcript in
  the overlap, which can be added to the set without producing a
  &quot;contradiction&quot;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;a gene structure is &quot;better&quot;
  than another one, if it has the transcript with the highest &quot;score&quot;,
  which is not present in the other gene structure.</div>
</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Mandatory_parameters:"><a class="selflink" href="#Mandatory_parameters:">Mandatory
  parameters:</a></h2>
<b>--genesets=file1,file2,...</b>/<b>-g file1,file2,...</b>
<div style="margin-left: 4.00ex;">where &quot;file1,file2,...,filen&quot; have
  to be data files with genesets in GTF format</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--output=ofile</b>/<b>-o ofile</b>
<div style="margin-left: 4.00ex;">where &quot;ofile&quot; is the name for an
  output file (GTF)</div>
<h2 class="Ss" title="Ss" id="Optional_parameters:"><a class="selflink" href="#Optional_parameters:">Optional
  parameters:</a></h2>
<b>--priorities=pr1,pr2,...</b>/<b>-p pr1,pr2,...</b>
<div style="margin-left: 4.00ex;">where &quot;pr1,pr2,...,prn&quot; have to be
  positiv integers (different from 0). Have to be as many as filenames are
  added. Bigger numbers means a higher priority. If no priorities are added, the
  program will set all priorties to 1. This option is only useful if there is
  more than one geneset. If there is a conflict between two transcripts, so that
  they can not be picked in the same genestructure, joingenes decides for the
  one with the highest priority.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--errordistance=x</b>/<b>-e x</b>
<div style="margin-left: 4.00ex;">where &quot;x&quot; is a non-negative integer.
  If a prediction is &#x21D0;x bases next to a prediction range border, the
  program supposes, that there could be a mistake. Default is 1000. To disable
  the function, set errordistance to a negative number (e.g. -1).</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--genemodel=x</b>/<b>-m x</b>
<div style="margin-left: 4.00ex;">where &quot;x&quot; is a genemodel from the
  set {eukaryote, bacterium}. Default is eukaryotic.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--alternatives</b>/<b>-a</b>
<div style="margin-left: 4.00ex;">If this flag is set, the program joins
  different genes if the transcripts of the genes are alternative
  variants.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--suppress=pr1,pr2,..</b>/<b>-s pr1,pr2,...</b>
<div style="margin-left: 4.00ex;">where &quot;pr1,pr2,...,prm&quot; have to be
  positive integers (different from 0). Default is none. If the core of a
  joined/non-joined transcript has one of these priorities it will not occur in
  the output file.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--stopincoding</b>/<b>-i</b>
<div style="margin-left: 4.00ex;">If this flag is set, the program joins the
  stop_codons to the CDS.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--nojoin</b>/<b>-j</b>
<div style="margin-left: 4.00ex;">If this flag is set, the program will not
  join/merge/shuffle; it will only decide between the unchanged input
  transcripts and output them.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--noselection</b>/<b>-l</b>
<div style="margin-left: 4.00ex;">If this flag is set, the program will NOT
  select at the end between &quot;contradictory&quot; transcripts.
  &quot;contradictory&quot; is self defined with respect to known biological
  terms. The selection works with a self defined scoring function.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<b>--onlycompare</b>/<b>-c</b>
<div style="margin-left: 4.00ex;">If this flag is set, it disables the normal
  function of the program and activates a compare and separate mode to separate
  equal transcripts from non equal ones.</div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
AUGUSTUS was written by M. Stanke, O. Keller, S. K&#x00F6;nig, L. Gerischer and
  L. Romoth.
<h1 class="Sh" title="Sh" id="ADDITIONAL_DOCUMENTATION"><a class="selflink" href="#ADDITIONAL_DOCUMENTATION">ADDITIONAL
  DOCUMENTATION</a></h1>
An exhaustive documentation can be found in the file
  /usr/share/augustus/README.TXT.</div>
<table class="foot">
  <tr>
    <td class="foot-date"></td>
    <td class="foot-os">&#x00A0;</td>
  </tr>
</table>
</body>
</html>
