<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:19:55 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>JOINGENES(1) JOINGENES(1)</p>

<p style="margin-top: 1em">NAME <br>
joingenes - merge several gene sets into one</p>

<p style="margin-top: 1em">SYNOPSIS <br>
joingenes [parameters] --genesets=file1,file2,...
--output=ofile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This program works in several steps:</p>

<p style="margin-top: 1em">1. divide the set of all
transcripts into smaller sets, in which all transcripts are
on the same sequence and are overlapping at least with one
other transcript in this set (set <br>
is called &quot;overlap&quot;)</p>

<p style="margin-top: 1em">2. delete all duplications of
transcripts and save the variant with the highest
&quot;score&quot;</p>

<p style="margin-top: 1em">3. if sequence ranges are set
for some transcripts, the program detects, whether the
distance to that range is dangerously close</p>

<p style="margin-top: 1em">4. join:</p>

<p style="margin-top: 1em">&Acirc;&middot; if there is a
transcript dangerously close to one/both end(s) of a
sequence range, the program creates a copy without the
corresponding terminal exon</p>

<p style="margin-top: 1em">&Acirc;&middot; if there is a
transcript with start or stop codon in a set and a second
one without this codon and they are &quot;joinable&quot;,
than this step joins the corresponding terminal <br>
exons</p>

<p style="margin-top: 1em">5. selection: selects the
&quot;best&quot; gene structure out of all possible
&quot;maximum&quot; gene structures</p>

<p style="margin-top: 1em">&Acirc;&middot;
&quot;maximum&quot; gene structure is a set of transcripts
from an overlap so that there is no other transcript in the
overlap, which can be added to the set without producing a
<br>
&quot;contradiction&quot;</p>

<p style="margin-top: 1em">&Acirc;&middot; a gene structure
is &quot;better&quot; than another one, if it has the
transcript with the highest &quot;score&quot;, which is not
present in the other gene structure.</p>

<p style="margin-top: 1em">OPTIONS <br>
Mandatory parameters: <br>
--genesets=file1,file2,.../-g file1,file2,... <br>
where &quot;file1,file2,...,filen&quot; have to be data
files with genesets in GTF format</p>

<p style="margin-top: 1em">--output=ofile/-o ofile <br>
where &quot;ofile&quot; is the name for an output file
(GTF)</p>

<p style="margin-top: 1em">Optional parameters: <br>
--priorities=pr1,pr2,.../-p pr1,pr2,... <br>
where &quot;pr1,pr2,...,prn&quot; have to be positiv
integers (different from 0). Have to be as many as filenames
are added. Bigger numbers means a higher priority. If no
priorities <br>
are added, the program will set all priorties to 1. This
option is only useful if there is more than one geneset. If
there is a conflict between two transcripts, so that they
<br>
can not be picked in the same genestructure, joingenes
decides for the one with the highest priority.</p>

<p style="margin-top: 1em">--errordistance=x/-e x <br>
where &quot;x&quot; is a non-negative integer. If a
prediction is &acirc;x bases next to a prediction range
border, the program supposes, that there could be a mistake.
Default is 1000. To <br>
disable the function, set errordistance to a negative number
(e.g. -1).</p>

<p style="margin-top: 1em">--genemodel=x/-m x <br>
where &quot;x&quot; is a genemodel from the set {eukaryote,
bacterium}. Default is eukaryotic.</p>

<p style="margin-top: 1em">--alternatives/-a <br>
If this flag is set, the program joins different genes if
the transcripts of the genes are alternative variants.</p>

<p style="margin-top: 1em">--suppress=pr1,pr2,../-s
pr1,pr2,... <br>
where &quot;pr1,pr2,...,prm&quot; have to be positive
integers (different from 0). Default is none. If the core of
a joined/non-joined transcript has one of these priorities
it will <br>
not occur in the output file.</p>

<p style="margin-top: 1em">--stopincoding/-i <br>
If this flag is set, the program joins the stop_codons to
the CDS.</p>

<p style="margin-top: 1em">--nojoin/-j <br>
If this flag is set, the program will not
join/merge/shuffle; it will only decide between the
unchanged input transcripts and output them.</p>

<p style="margin-top: 1em">--noselection/-l <br>
If this flag is set, the program will NOT select at the end
between &quot;contradictory&quot; transcripts.
&quot;contradictory&quot; is self defined with respect to
known biological terms. The <br>
selection works with a self defined scoring function.</p>

<p style="margin-top: 1em">--onlycompare/-c <br>
If this flag is set, it disables the normal function of the
program and activates a compare and separate mode to
separate equal transcripts from non equal ones.</p>

<p style="margin-top: 1em">AUTHORS <br>
AUGUSTUS was written by M. Stanke, O. Keller, S.
K&Atilde;&para;nig, L. Gerischer and L. Romoth.</p>

<p style="margin-top: 1em">ADDITIONAL DOCUMENTATION <br>
An exhaustive documentation can be found in the file
/usr/share/augustus/README.TXT.</p>
 
<p style="margin-top: 1em">JOINGENES(1)</p>
<hr>
</body>
</html>
