<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>KALIGN(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">KALIGN(1)</td>
    <td class="head-vol">Kalign User Manual</td>
    <td class="head-rtitle">KALIGN(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
kalign - performs multiple alignment of biological sequences.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 7.00ex; text-indent: -7.00ex;"><b>kalign</b>
  [ <i>infile.fasta</i>] [<i>outfile.fasta</i>] [Options]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 7.00ex; text-indent: -7.00ex;"><b>kalign</b>
  [-i&#x00A0; <i>infile.fasta</i>] [-o&#x00A0;<i>outfile.fasta</i>]
  [Options]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 7.00ex; text-indent: -7.00ex;"><b>kalign</b>
  [&lt;&#x00A0; <i>infile.fasta</i>] [&gt;&#x00A0;<i>outfile.fasta</i>]
  [Options]</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>Kalign</b> is a command line tool to perform multiple alignment of biological
  sequences. It employs the Muth?Manber string-matching algorithm, to improve
  both the accuracy and speed of the alignment. It uses global, progressive
  alignment approach, enriched by employing an approximate string-matching
  algorithm to calculate sequence distances and by incorporating local matches
  into the otherwise global alignment.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<b>-s</b> <b>-gpo</b> <b>-gapopen</b> <b>-gap_open</b> <i>x</i>
<div style="margin-left: 4.00ex;">Gap open penalty .</div>
<div class="Pp"></div>
<b>-e</b> <b>-gpe</b> <b>-gap_ext</b> <b>-gapextension</b> <i>x</i>
<div style="margin-left: 4.00ex;">Gap extension penalty.</div>
<div class="Pp"></div>
<b>-t</b> <b>-tgpe</b> <b>-terminal_gap_extension_penalty</b> <i>x</i>
<div style="margin-left: 4.00ex;">Terminal gap penalties.</div>
<div class="Pp"></div>
<b>-m</b> <b>-bonus</b> <b>-matrix_bonus</b> <i>x</i>
<div style="margin-left: 4.00ex;">A constant added to the substitution
  matrix.</div>
<div class="Pp"></div>
<b>-c</b> <b>-sort</b> <i>&lt;input, tree, gaps.&gt;</i>
<div style="margin-left: 4.00ex;">The order in which the sequences appear in the
  output alignment.</div>
<div class="Pp"></div>
<b>-g</b> <b>-feature</b>
<div style="margin-left: 4.00ex;">Selects feature mode and specifies which
  features are to be used: e.g. all, maxplp, STRUCT, PFAM-A?</div>
<div class="Pp"></div>
<b>-same_feature_score</b>
<div style="margin-left: 4.00ex;">Score for aligning same features.</div>
<div class="Pp"></div>
<b>-diff_feature_score</b>
<div style="margin-left: 4.00ex;">Penalty for aligning different features.</div>
<div class="Pp"></div>
<b>-d</b> <b>-distance</b> <i>&lt;wu, pair&gt;</i>
<div style="margin-left: 4.00ex;">Distance method</div>
<div class="Pp"></div>
<b>-b</b> <b>-tree</b> <b>-guide-tree</b> <i>&lt;nj, upgma&gt;</i>
<div style="margin-left: 4.00ex;">Guide tree method.</div>
<div class="Pp"></div>
<b>-z</b> <b>-zcutoff</b>
<div style="margin-left: 4.00ex;">Parameter used in the wu-manber based distance
  calculation.</div>
<div class="Pp"></div>
<b>-i</b> <b>-in</b> <b>-input</b>
<div style="margin-left: 4.00ex;">Name of the input file.</div>
<div class="Pp"></div>
<b>-o</b> <b>-out</b> <b>-output</b>
<div style="margin-left: 4.00ex;">Name of the output file.</div>
<div class="Pp"></div>
<b>-a</b> <b>-gap_inc</b>
<div style="margin-left: 4.00ex;">Increases gap penalties depending on the
  number of existing gaps.</div>
<div class="Pp"></div>
<b>-f</b> <b>-format</b> <i>&lt;fasta, msf, aln, clu, macsim&gt;</i>
<div style="margin-left: 4.00ex;">The output format.</div>
<div class="Pp"></div>
<b>-q</b> <b>-quiet</b>
<div style="margin-left: 4.00ex;">Print nothing to STDERR. Read nothing from
  STDIN.</div>
<h1 class="Sh" title="Sh" id="REFERENCES"><a class="selflink" href="#REFERENCES">REFERENCES</a></h1>
<div style="margin-left: 4.00ex;">&#x2022;Timo Lassmann and Erik L.L. Sonnhammer
  (2005) Kalign - an accurate and fast multiple sequence alignment algorithm.
  BMC Bioinformatics 6:298</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Timo Lassmann, Oliver Frings and Erik
  L. L. Sonnhammer (2009) Kalign2: high-performance multiple alignment of
  protein and nucleotide sequences allowing external features. Nucleic Acid
  Research 3:858?865.</div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<b>Timo Lassmann</b> &lt;timolassmann@gmail.com&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Upstream author of Kalign.</dd>
</dl>
<div class="Pp"></div>
<b>Charles Plessy</b> &lt;plessy@debian.org&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Wrote the manpage.</dd>
</dl>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
Copyright &#x00A9; 2004, 2005, 2006, 2007, 2008 Timo Lassmann
<div>&#x00A0;</div>
<div class="Pp"></div>
Kalign is free software. You can redistribute it and/or modify it under the
  terms of the GNU General Public License as published by the Free Software
  Foundation.
<div class="Pp"></div>
This manual page was written by Charles Plessy &lt;plessy@debian.org&gt; for the
  Debian(TM) system (but may be used by others). Permission is granted to copy,
  distribute and/or modify this document under the same terms as kalign itself.
<div class="Pp"></div>
On Debian systems, the complete text of the GNU General Public License version 2
  can be found in <i>/usr/share/common-licenses/GPL-2</i>.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 25, 2009</td>
    <td class="foot-os">kalign 2.04</td>
  </tr>
</table>
</body>
</html>
