<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:20:14 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>KALIGN(1) Kalign User Manual KALIGN(1)</p>

<p style="margin-top: 1em">NAME <br>
kalign - performs multiple alignment of biological
sequences.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
kalign [infile.fasta] [outfile.fasta] [Options]</p>

<p style="margin-top: 1em">kalign [-i infile.fasta] [-o
outfile.fasta] [Options]</p>

<p style="margin-top: 1em">kalign [&lt; infile.fasta] [&gt;
outfile.fasta] [Options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Kalign is a command line tool to perform multiple alignment
of biological sequences. It employs the Muth?Manber
string-matching algorithm, to improve both the accuracy and
speed <br>
of the alignment. It uses global, progressive alignment
approach, enriched by employing an approximate
string-matching algorithm to calculate sequence distances
and by <br>
incorporating local matches into the otherwise global
alignment.</p>

<p style="margin-top: 1em">OPTIONS <br>
-s -gpo -gapopen -gap_open x <br>
Gap open penalty .</p>

<p style="margin-top: 1em">-e -gpe -gap_ext -gapextension x
<br>
Gap extension penalty.</p>

<p style="margin-top: 1em">-t -tgpe
-terminal_gap_extension_penalty x <br>
Terminal gap penalties.</p>

<p style="margin-top: 1em">-m -bonus -matrix_bonus x <br>
A constant added to the substitution matrix.</p>

<p style="margin-top: 1em">-c -sort &lt;input, tree,
gaps.&gt; <br>
The order in which the sequences appear in the output
alignment.</p>

<p style="margin-top: 1em">-g -feature <br>
Selects feature mode and specifies which features are to be
used: e.g. all, maxplp, STRUCT, PFAM-A?</p>

<p style="margin-top: 1em">-same_feature_score <br>
Score for aligning same features.</p>

<p style="margin-top: 1em">-diff_feature_score <br>
Penalty for aligning different features.</p>

<p style="margin-top: 1em">-d -distance &lt;wu, pair&gt;
<br>
Distance method</p>

<p style="margin-top: 1em">-b -tree -guide-tree &lt;nj,
upgma&gt; <br>
Guide tree method.</p>

<p style="margin-top: 1em">-z -zcutoff <br>
Parameter used in the wu-manber based distance
calculation.</p>

<p style="margin-top: 1em">-i -in -input <br>
Name of the input file.</p>

<p style="margin-top: 1em">-o -out -output <br>
Name of the output file.</p>

<p style="margin-top: 1em">-a -gap_inc <br>
Increases gap penalties depending on the number of existing
gaps.</p>

<p style="margin-top: 1em">-f -format &lt;fasta, msf, aln,
clu, macsim&gt; <br>
The output format.</p>

<p style="margin-top: 1em">-q -quiet <br>
Print nothing to STDERR. Read nothing from STDIN.</p>

<p style="margin-top: 1em">REFERENCES <br>
&Acirc;&middot; Timo Lassmann and Erik L.L. Sonnhammer
(2005) Kalign - an accurate and fast multiple sequence
alignment algorithm. BMC Bioinformatics 6:298</p>

<p style="margin-top: 1em">&Acirc;&middot; Timo Lassmann,
Oliver Frings and Erik L. L. Sonnhammer (2009) Kalign2:
high-performance multiple alignment of protein and
nucleotide sequences allowing external features. <br>
Nucleic Acid Research 3:858?865.</p>

<p style="margin-top: 1em">AUTHORS <br>
Timo Lassmann &lt;timolassmann@gmail.com&gt; <br>
Upstream author of Kalign.</p>

<p style="margin-top: 1em">Charles Plessy
&lt;plessy@debian.org&gt; <br>
Wrote the manpage.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 2004, 2005, 2006, 2007, 2008 Timo
Lassmann</p>

<p style="margin-top: 1em">Kalign is free software. You can
redistribute it and/or modify it under the terms of the GNU
General Public License as published by the Free Software
Foundation.</p>

<p style="margin-top: 1em">This manual page was written by
Charles Plessy &lt;plessy@debian.org&gt; for the Debian(TM)
system (but may be used by others). Permission is granted to
copy, distribute and/or modify <br>
this document under the same terms as kalign itself.</p>

<p style="margin-top: 1em">On Debian systems, the complete
text of the GNU General Public License version 2 can be
found in /usr/share/common-licenses/GPL-2.</p>

<p style="margin-top: 1em">kalign 2.04 February 25, 2009
KALIGN(1)</p>
<hr>
</body>
</html>
