<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:20:38 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>KISSNP2(1) User Commands KISSNP2(1)</p>

<p style="margin-top: 1em">NAME <br>
kissnp2 - manual page for kissnp2 1.2.6</p>

<p style="margin-top: 1em">DESCRIPTION <br>
NAME kissnp, version DiscoSnp - kissnp submodule 1.2.6 -
Copyright INRIA - CeCILL License</p>

<p style="margin-top: 1em">SYNOPSIS kissnp2/kissnp2
&lt;readsC1.fasta&gt; [&lt;readsC2.fasta&gt;
[&lt;readsC3.fasta] ...] -o name [-t] [-e length] [-l] [-b]
[-k value] [-c value] [-g value] [-h]</p>

<p style="margin-top: 1em">or:</p>

<p style="margin-top: 1em">kissnp2/kissnp2
input_names_file.txt -o name [-t] [-e length] [-l] [-b] [-k
value] [-c value] [-g value] [-h]</p>

<p style="margin-top: 1em">with
&quot;input_names_file.txt&quot; being a file containing on
each line the name of read files</p>

<p style="margin-top: 1em">DESCRIPTION</p>

<p style="margin-top: 1em">&quot;kissnp2&quot;, detects
SNPs from read set(s). It should usually be followed by the
&quot;kissreads&quot; for recovering coverage and quality
information from reads.</p>

<p style="margin-top: 1em">MANDATORY</p>

<p style="margin-top: 1em">At least one read set. -o
file_name_prefix: where to write outputs and debruijn graph
structure files.</p>

<p style="margin-top: 1em">OPTIONS</p>

<p style="margin-top: 1em">-t extend found and stop at
first polymorphism (strict extension=unitigs) SNPs.
Uncompatible with -T</p>

<p style="margin-top: 1em">-T extend found and stop at
large polymorphism (extension=contigs) SNPs. Uncompatible
with -t</p>

<p style="margin-top: 1em">-e length: extend found SNPs
(option -t) and conserve only those whose min(left and right
extension) is bigger or equal to &quot;length&quot;</p>

<p style="margin-top: 1em">-l conserve low complexity SNPs.
Default: false (filter out low complexity results)</p>

<p style="margin-top: 1em">-b INT:</p>

<p style="margin-top: 1em">0: forbid SNPs for wich any of
the two paths is branching (high precision, low recall).
<br>
Default value</p>

<p style="margin-top: 1em">1: forbid SNPs for wich the two
paths are branching (e.g. the two paths can be created
either with a &rsquo;A&rsquo; or a &rsquo;C&rsquo; at the
same position 2: No limitation on branching (low <br>
precision, high recall)</p>

<p style="margin-top: 1em">-k size_seed: will use seeds of
length size_seed. Default: 27.</p>

<p style="margin-top: 1em">-c min_coverage: a sequence is
covered by at least min_coverage coherent reads. Default:
2</p>

<p style="margin-top: 1em">-C max_coverage: a sequence is
covered by at most max_coverage coherent reads. Default:
infiny (=2147483647 on your computer :) )</p>

<p style="margin-top: 1em">-g estimated_genome_size:
estimation of the size of the genome whose reads come
from.</p>

<p style="margin-top: 1em">It is in bp, does not need to be
accurate, only controls memory usage. Default: 3 billion</p>

<p style="margin-top: 1em">-h prints this message and
exit</p>

<p style="margin-top: 1em">DiscoSnp - kissnp submodule
1.2.6 - Copyright INRIA - CeCILL License The command line
was: kissnp2/kissnp2 -h</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation for kissnp2 is maintained as a
Texinfo manual. If the info and kissnp2 programs are
properly installed at your site, the command</p>

<p style="margin-top: 1em">info kissnp2</p>

<p style="margin-top: 1em">should give you access to the
complete manual.</p>

<p style="margin-top: 1em">kissnp2 1.2.6 November 2014
KISSNP2(1)</p>
<hr>
</body>
</html>
