<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>KISSNP2(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">KISSNP2(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">KISSNP2(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
kissnp2 - manual page for kissnp2 1.2.6
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
NAME kissnp, version DiscoSnp - kissnp submodule 1.2.6 - Copyright INRIA -
  CeCILL License
<div class="Pp"></div>
SYNOPSIS kissnp2/kissnp2 &lt;readsC1.fasta&gt; [&lt;readsC2.fasta&gt;
  [&lt;readsC3.fasta] ...] <b>-o</b> name [-t] [-e length] [-l] [-b] [-k value]
  [-c value] [-g value] [-h]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">or:</dd>
</dl>
<div class="Pp"></div>
kissnp2/kissnp2 input_names_file.txt <b>-o</b> name [-t] [-e length] [-l] [-b]
  [-k value] [-c value] [-g value] [-h]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">with &quot;input_names_file.txt&quot; being a file
      containing on each line the name of read files</dd>
</dl>
<div class="Pp"></div>
DESCRIPTION
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&quot;kissnp2&quot;, detects SNPs from read set(s). It
      should usually be followed by the &quot;kissreads&quot; for recovering
      coverage and quality information from reads.</dd>
</dl>
<div class="Pp"></div>
MANDATORY
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">At least one read set. <b>-o</b> file_name_prefix: where to
      write outputs and debruijn graph structure files.</dd>
</dl>
<div class="Pp"></div>
OPTIONS
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-t</b>
  extend found and stop at first polymorphism (strict extension=unitigs) SNPs.
  Uncompatible with <b>-T</b></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-T</b>
  extend found and stop at large polymorphism (extension=contigs) SNPs.
  Uncompatible with <b>-t</b></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-e</b>
  length: extend found SNPs (option <b>-t</b>) and conserve only those whose
  min(left and right extension) is bigger or equal to &quot;length&quot;</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-l</b>
  conserve low complexity SNPs. Default: false (filter out low complexity
  results)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-b</b>
  INT:</div>
<dl class="Bl-tag">
  <dt class="It-tag">0: forbid SNPs for wich any of the two paths is branching
    (high precision, low recall).</dt>
  <dd class="It-tag">Default value</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">1: forbid SNPs for wich the two paths are branching (e.g.
      the two paths can be created either with a 'A' or a 'C' at the same
      position 2: No limitation on branching (low precision, high recall)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-k</b>
  size_seed: will use seeds of length size_seed. Default: 27.</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-c</b>
  min_coverage: a sequence is covered by at least min_coverage coherent reads.
  Default: 2</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-C</b>
  max_coverage: a sequence is covered by at most max_coverage coherent reads.
  Default: infiny (=2147483647 on your computer :) )</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-g</b>
  estimated_genome_size: estimation of the size of the genome whose reads come
  from.</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">It is in bp, does not need to be accurate, only controls
      memory usage. Default: 3 billion</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-h</b>
  prints this message and exit</div>
<div class="Pp"></div>
DiscoSnp - kissnp submodule 1.2.6 - Copyright INRIA - CeCILL License The command
  line was: kissnp2/kissnp2 <b>-h</b>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The full documentation for <b>kissnp2</b> is maintained as a Texinfo manual. If
  the <b>info</b> and <b>kissnp2</b> programs are properly installed at your
  site, the command
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>info kissnp2</b></dd>
</dl>
<div class="Pp"></div>
should give you access to the complete manual.</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2014</td>
    <td class="foot-os">kissnp2 1.2.6</td>
  </tr>
</table>
</body>
</html>
