<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:20:38 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>KISSPLICE(1) User Commands KISSPLICE(1)</p>

<p style="margin-top: 1em">NAME <br>
kissplice - detection of polymorphisms in RNA-seq data.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
kissplice -h <br>
kissplice [OPTION] [-r READFILES]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Detects alternative splicing events and other kinds of
polymorphisms from READFILES (in FASTA or FASTQ format).</p>

<p style="margin-top: 1em">OPTIONS <br>
-h, --help <br>
Show this help message and exit.</p>

<p style="margin-top: 1em">-r READFILES <br>
Input fasta/q read files (multiple, such as &quot;-r file1
-r file2...&quot;).</p>

<p style="margin-top: 1em">-k KVAL <br>
k-mer size (default=25).</p>

<p style="margin-top: 1em">-l LLMAX <br>
Maximal length of the shorter path (default: 2k-1).</p>

<p style="margin-top: 1em">-m LL_MIN <br>
Minimum length of the shorter path (default 2k-8).</p>

<p style="margin-top: 1em">-M UL_MAX <br>
Maximum length of the longest path (default 1000), skipped
exons longer than UL_MAX are not reported.</p>

<p style="margin-top: 1em">-g GRAPH_PREFIX <br>
Path and prefix to pre-built de Bruijn graph (suffixed by
.edges/.nodes) if jointly used with -r, graph used to find
bubbles and reads used for quantification.</p>

<p style="margin-top: 1em">-o OUT_DIR <br>
Path to store the results (default = ./results).</p>

<p style="margin-top: 1em">-d PATH_TO_TMP <br>
Specific directory (absolute path) where to build temporary
files (default temporary directory otherwise).</p>

<p style="margin-top: 1em">-t NBPROCS <br>
Number of cores (must be &lt;= number of physical
cores).</p>

<p style="margin-top: 1em">-s Don&rsquo;t output SNPs
(saves time).</p>

<p style="margin-top: 1em">-v Verbose mode.</p>

<p style="margin-top: 1em">-u Keep the nodes/edges file for
unfinished bccs.</p>

<p style="margin-top: 1em">-c MIN_COV <br>
Discard k-mers tha are present strictly less than this
number of times in the dataset. (default 2).</p>

<p style="margin-top: 1em">-C MIN_RELATIVE_COV <br>
Discard edges with relative coverage below MIN_RELATIVE_COV
expressed as a percentage in [0,1). (default 0.02).</p>

<p style="margin-top: 1em">-z GENOME_SIZE <br>
Estimated number of nodes in the De-Bruin Graph. (default =
1000000000).</p>

<p style="margin-top: 1em">-e MIN_EDIT_DIST <br>
Classify as inexact repeats those bubbles whose paths&rsquo;
edit distance is smaller than MIN_EDIT_DIST (default 3).</p>

<p style="margin-top: 1em">-y MAX_CYCLES <br>
Maximal number of bubble enumerations in each bcc. If
exceeded, no bubble is output for the bcc (default
10000).</p>

<p style="margin-top: 1em">--timeout TIMEOUT <br>
Max amount of time (in seconds) spent for enumerating
bubbles in each bcc. If exceeded, no bubble is output for
the bcc (default 900).</p>

<p style="margin-top: 1em">--version <br>
Display program&rsquo;s version number and exit.</p>

<p style="margin-top: 1em">kissplice 1.8.3-1 June 2013
KISSPLICE(1)</p>
<hr>
</body>
</html>
