<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>KISSREADS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">KISSREADS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">KISSREADS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
kissreads - manual page for kissreads 1.2.6
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
NAME kissReads, version kissreads module 1.2.6 - Copyright INRIA - CeCILL
  License
<div class="Pp"></div>
SYNOPSIS kissreads/kissreads &lt;toCheck.fasta&gt; &lt;readsC1.fasta&gt;
  [&lt;readsC2.fasta&gt; [&lt;readsC3.fasta] ...] [-k value] [-c value] [-d
  value] [-O value] [-o name] [-u name] [-n] [-I] [-i index_stride] [-m
  align_file] [-s] [-f] [-h]
<div class="Pp"></div>
DESCRIPTION Checks for each sequence contained into the toCheck.fasta if it is
  read coherent (each position is covered by at least &quot;min_coverage&quot;
  read(s)) with reads from readsA.fasta or readsB.fasta A sequence s from
  toCheck is treated as follow:
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">if (s coherent with at least one read set): output the
      sequence as follows</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">&gt;original fasta
      comment|C1:min&lt;avg-corr_avg&lt;max|C2:min&lt;avg-cor_avg&lt;max|C3...:
      &gt;s</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">With A:min&lt;avg-cor_avg&lt;max standing for : value of
      the position having minimal coverage in readsA.fasta &lt; average coverage
      in readsA.fasta - R-squarred corrected average in readsA.fa &lt; value of
      the position having maximal coverage in readsA.fasta The coverage is the
      number of reads that perfectly mapped a position Any other situation (s
      not coherent with any): couple non read coherent, not outputed</dd>
</dl>
<div class="Pp"></div>
OPTIONS
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-k</b>
  size_seed: will use seeds of length size_seed. Default: 25.</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-O</b>
  minimal_read_overlap: a read can be mapped if its overlap is a least
  &quot;minimal_read_overlap&quot;. Default: k</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-c</b>
  min_coverage: a sequence is covered by at least min_coverage coherent reads.
  Default: 2</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-d</b>
  max_substitutions: Maximal number of substitutions authorized between a read
  and a fragment. Note that no substitution is allowed on the central position
  while anaylizing the kissnp output. Default: 1.</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  file_name: write read-coherent outputs. Default: standard output</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-u</b>
  file_name: write unread-coherent outputs. Default: standard output</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-n</b> the
  input file (toCheck.fasta) is a kissnp output (incompatible with <b>-I</b>
  option)</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">in this case: 1/ only the upper characters are considered
      (no mapping done on the extensions) and 2/ the central position (where the
      SNP occurs) is strictly mapped, no subsitution is authorized on this
      position.</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-I</b> the
  input file (toCheck.fasta) is an Intl output (incompatible with <b>-n</b>
  option)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-i</b>
  index_stride (int value). This is a heuristic for limiting the memory
  footprint. Instead of indexing each kmer of the sequences contained into the
  toCheck.fasta, kissreads indexes kmers occurring each &quot;index_stride&quot;
  position. Default = 1 (no heuristic)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-t</b> max
  number of threads (also limited by number of input files)</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-m</b>
  align_file, write a file of reads mapped to sequences in file align_file</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-s</b>
  silent mode</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-f</b>
  outputs coherent events in a standard fasta file format</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-h</b>
  prints this message and exit</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The full documentation for <b>kissreads</b> is maintained as a Texinfo manual.
  If the <b>info</b> and <b>kissreads</b> programs are properly installed at
  your site, the command
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>info kissreads</b></dd>
</dl>
<div class="Pp"></div>
should give you access to the complete manual.</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2014</td>
    <td class="foot-os">kissreads 1.2.6</td>
  </tr>
</table>
</body>
</html>
