<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>LAMBDA(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">LAMBDA(1)</td>
    <td class="head-vol"></td>
    <td class="head-rtitle">LAMBDA(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
lambda - the Local Aligner for Massive Biological DatA
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>lambda</b> [<i>OPTIONS</i>] <i>-q QUERY.fasta</i> <i>-d DATABASE.fasta</i>
  [<i>-o output.m8</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Lambda is a local aligner optimized for many query sequences and searches in
  protein space. It is compatible to BLAST, but much faster than BLAST and many
  other comparable tools.
<div style="height: 1.00em;">&#x00A0;</div>
Detailed information is available in the wiki:
  &lt;https://github.com/seqan/lambda/wiki&gt;
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">Display the help message.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-hh</b>, <b>--full-help</b></dt>
  <dd class="It-tag">Display the help message with advanced options.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">Display version information.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--copyright</b></dt>
  <dd class="It-tag">Display long copyright information.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--verbosity</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Display more/less diagnostic output during operation: 0
      [only errors]; 1 [default]; 2 [+run-time, options and statistics]. In
      range [0..2]. Default: <i>1</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_Options:"><a class="selflink" href="#Input_Options:">Input
  Options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b>, <b>--query</b> <i>INPUT_FILE</i></dt>
  <dd class="It-tag">Query sequences. Valid filetypes are: <i>.sam[.*]</i>,
      <i>.raw[.*]</i>, <i>.gbk[.*]</i>, <i>.frn[.*]</i>, <i>.fq[.*]</i>,
      <i>.fna[.*]</i>, <i>.ffn[.*]</i>, <i>.fastq[.*]</i>, <i>.fasta[.*]</i>,
      <i>.faa[.*]</i>, <i>.fa[.*]</i>, <i>.embl[.*]</i>, and <i>.bam</i>, where
      * is any of the following extensions: <i>gz</i>, <i>bz2</i>, and
      <i>bgzf</i> for transparent (de)compression.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b>, <b>--database</b> <i>INPUT_FILE</i></dt>
  <dd class="It-tag">Path to original database sequences (a precomputed index
      with .sa or .fm needs to exist!). Valid filetypes are: <i>.sam[.*]</i>,
      <i>.raw[.*]</i>, <i>.gbk[.*]</i>, <i>.frn[.*]</i>, <i>.fq[.*]</i>,
      <i>.fna[.*]</i>, <i>.ffn[.*]</i>, <i>.fastq[.*]</i>, <i>.fasta[.*]</i>,
      <i>.faa[.*]</i>, <i>.fa[.*]</i>, <i>.embl[.*]</i>, and <i>.bam</i>, where
      * is any of the following extensions: <i>gz</i>, <i>bz2</i>, and
      <i>bgzf</i> for transparent (de)compression.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-di</b>, <b>--db-index-type</b> <i>STRING</i></dt>
  <dd class="It-tag">database index is in this format. One of <i>sa</i> and
      <i>fm</i>. Default: <i>fm</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Output_Options:"><a class="selflink" href="#Output_Options:">Output
  Options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>, <b>--output</b> <i>OUTPUT_FILE</i></dt>
  <dd class="It-tag">File to hold reports on hits (.m* are blastall -m* formats;
      .m8 is tab-seperated, .m9 is tab-seperated with with comments, .m0 is
      pairwise format). Valid filetypes are: <i>.sam[.*]</i>, <i>.m9[.*]</i>,
      <i>.m8[.*]</i>, <i>.m0[.*]</i>, and <i>.bam</i>, where * is any of the
      following extensions: <i>gz</i>, <i>bz2</i>, and <i>bgzf</i> for
      transparent (de)compression. Default: <i>output.m8</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-oc</b>, <b>--output-columns</b> <i>STRING</i></dt>
  <dd class="It-tag">Print specified column combination and/or order (.m8 and
      .m9 outputs only); call -oc help for more details. Default:
    <i>std</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-id</b>, <b>--percent-identity</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Output only matches above this threshold (checked before
      e-value check). In range [0..100]. Default: <i>0</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>, <b>--e-value</b> <i>DOUBLE</i></dt>
  <dd class="It-tag">Output only matches that score below this threshold. In
      range [0..inf]. Default: <i>0.1</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-nm</b>, <b>--num-matches</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Print at most this number of matches per query. In range
      [1..inf]. Default: <i>500</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-with-refheader</b> <i>STRING</i></dt>
  <dd class="It-tag">BAM files require all subject names to be written to the
      header. For SAM this is not required, so Lambda does not automatically do
      it to save space (especially for protein database this is a lot!). If you
      still want them with SAM, e.g. for better BAM compatibility, use this
      option. One of <i>on</i> and <i>off</i>. Default: <i>off</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-bam-seq</b> <i>STRING</i></dt>
  <dd class="It-tag">Write matching DNA subsequence into SAM/BAM file (BLASTN).
      For BLASTX and TBLASTX the matching protein sequence is
      &quot;untranslated&quot; and positions retransformed to the original
      sequence. For BLASTP and TBLASTN there is no DNA sequence so a
      &quot;*&quot; is written to the SEQ column. The matching protein sequence
      can be written as an optional tag, see --sam-bam-tags. If set to uniq than
      the sequence is omitted iff it is identical to the previous match's
      subsequence. One of <i>always</i>, <i>uniq</i>, and <i>never</i>. Default:
      <i>uniq</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-bam-tags</b> <i>STRING</i></dt>
  <dd class="It-tag">Write the specified optional columns to the SAM/BAM file.
      Call --sam-bam-tags help for more details. Default: <i>AS NM ZE ZI
    ZF</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam-bam-clip</b> <i>STRING</i></dt>
  <dd class="It-tag">Whether to hard-clip or soft-clip the regions beyond the
      local match. Soft-clipping retains the full sequence in the output file,
      but obviously uses more space. One of <i>hard</i> and <i>soft</i>.
      Default: <i>hard</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="General_Options:"><a class="selflink" href="#General_Options:">General
  Options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>, <b>--threads</b> <i>INTEGER</i></dt>
  <dd class="It-tag">number of threads to run concurrently.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-qi</b>, <b>--query-index-type</b> <i>STRING</i></dt>
  <dd class="It-tag">controls double-indexing. One of <i>radix</i> and
      <i>none</i>. Default: <i>none</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Alphabets_and_Translation:"><a class="selflink" href="#Alphabets_and_Translation:">Alphabets
  and Translation:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b>, <b>--program</b> <i>STRING</i></dt>
  <dd class="It-tag">Blast Operation Mode. One of <i>blastn</i>, <i>blastp</i>,
      <i>blastx</i>, <i>tblastn</i>, and <i>tblastx</i>. Default:
    <i>blastx</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b>, <b>--genetic-code</b> <i>INTEGER</i></dt>
  <dd class="It-tag">The translation table to use for nucl -&gt; amino acid
      translation(not for BlastN, BlastP). See
      https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c for ids
      (default is generic). Six frames are generated. Default: <i>1</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ar</b>, <b>--alphabet-reduction</b> <i>STRING</i></dt>
  <dd class="It-tag">Alphabet Reduction for seeding phase (ignored for BLASTN).
      One of <i>none</i> and <i>murphy10</i>. Default: <i>murphy10</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Seeding_/_Filtration:"><a class="selflink" href="#Seeding_/_Filtration:">Seeding
  / Filtration:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sl</b>, <b>--seed-length</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Length of the seeds (default = 14 for BLASTN). Default:
      <i>10</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-so</b>, <b>--seed-offset</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Offset for seeding (if unset = seed-length,
      non-overlapping; default = 5 for BLASTN). Default: <i>10</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sd</b>, <b>--seed-delta</b> <i>INTEGER</i></dt>
  <dd class="It-tag">maximum seed distance. Default: <i>1</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Miscellaneous_Heuristics:"><a class="selflink" href="#Miscellaneous_Heuristics:">Miscellaneous
  Heuristics:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ps</b>, <b>--pre-scoring</b> <i>INTEGER</i></dt>
  <dd class="It-tag">evaluate score of a region NUM times the size of the seed
      before extension (0 -&gt; no pre-scoring, 1 -&gt; evaluate seed, n-&gt;
      area around seed, as well; default = 1 if no reduction is used). In range
      [1..inf]. Default: <i>2</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pt</b>, <b>--pre-scoring-threshold</b>
    <i>DOUBLE</i></dt>
  <dd class="It-tag">minimum average score per position in pre-scoring region.
      Default: <i>2</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pd</b>, <b>--filter-putative-duplicates</b>
    <i>STRING</i></dt>
  <dd class="It-tag">filter hits that will likely duplicate a match already
      found. One of <i>on</i> and <i>off</i>. Default: <i>on</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pa</b>, <b>--filter-putative-abundant</b>
    <i>STRING</i></dt>
  <dd class="It-tag">If the maximum number of matches per query are found
      already, stop searching if the remaining realm looks unfeasable. One of
      <i>on</i> and <i>off</i>. Default: <i>on</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Scoring:"><a class="selflink" href="#Scoring:">Scoring:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-sc</b>, <b>--scoring-scheme</b> <i>INTEGER</i></dt>
  <dd class="It-tag">use '45' for Blosum45; '62' for Blosum62 (default); '80'
      for Blosum80; [ignored for BlastN] Default: <i>62</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ge</b>, <b>--score-gap</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Score per gap character (default = -2 for BLASTN). Default:
      <i>-1</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-go</b>, <b>--score-gap-open</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Additional cost for opening gap (default = -5 for BLASTN).
      Default: <i>-11</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ma</b>, <b>--score-match</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Match score [only BLASTN]) Default: <i>2</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mi</b>, <b>--score-mismatch</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Mismatch score [only BLASTN] Default: <i>-3</i>.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Extension:"><a class="selflink" href="#Extension:">Extension:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-x</b>, <b>--x-drop</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Stop Banded extension if score x below the maximum seen (-1
      means no xdrop). In range [-1..inf]. Default: <i>30</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b>, <b>--band</b> <i>INTEGER</i></dt>
  <dd class="It-tag">Size of the DP-band used in extension (-3 means log2 of
      query length; -2 means sqrt of query length; -1 means full dp; n means
      band of size 2n+1) In range [-3..inf]. Default: <i>-3</i>.</dd>
</dl>
<h1 class="Sh" title="Sh" id="TUNING"><a class="selflink" href="#TUNING">TUNING</a></h1>
Tuning the seeding parameters and (de)activating alphabet reduction has a strong
  influence on both speed and sensitivity. We recommend the following
  alternative profiles for protein searches:
<div style="height: 1.00em;">&#x00A0;</div>
fast (high similarity): -ar none -sl 7 -sd 0
<div style="height: 1.00em;">&#x00A0;</div>
sensitive (lower similarity): -so 5
<div style="height: 1.00em;">&#x00A0;</div>
For further information see the wiki:
  &lt;https://github.com/seqan/lambda/wiki&gt;
<h1 class="Sh" title="Sh" id="LEGAL"><a class="selflink" href="#LEGAL">LEGAL</a></h1>
<b>lambda Copyright:</b> 2013-2017 Hannes Hauswedell, released under the GNU GPL
  v3 (or later); 2016-2017 Knut Reinert and Freie Universit&#x00E4;t Berlin,
  released under the 3-clause-BSDL
<div>&#x00A0;</div>
<b>SeqAn Copyright:</b> 2006-2015 Knut Reinert, FU-Berlin; released under the
  3-clause BSDL.
<div>&#x00A0;</div>
<b>In your academic works please cite:</b> Hauswedell et al (2014); doi:
  10.1093/bioinformatics/btu439
<div>&#x00A0;</div>
For full copyright and/or warranty information see <b>--copyright</b>.</div>
<table class="foot">
  <tr>
    <td class="foot-date">Jan 21 2017</td>
    <td class="foot-os">lambda 1.0.1 </td>
  </tr>
</table>
</body>
</html>
