<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:21:14 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>LAMBDA_INDEXER(1) LAMBDA_INDEXER(1)</p>

<p style="margin-top: 1em">NAME <br>
lambda_indexer - indexer for creating lambda-compatible
databases</p>

<p style="margin-top: 1em">SYNOPSIS <br>
lambda_indexer [OPTIONS] -d DATABASE.fasta</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Lambda is a local aligner optimized for many query sequences
and searches in protein space. It is compatible to BLAST,
but much faster than BLAST and many other comparable
tools.</p>

<p style="margin-top: 1em">Detailed information is
available in the wiki:
&lt;https://github.com/seqan/lambda/wiki&gt;</p>

<p style="margin-top: 1em">This is the indexer_binary for
creating lambda-compatible databases.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h, --help <br>
Display the help message.</p>

<p style="margin-top: 1em">-hh, --full-help <br>
Display the help message with advanced options.</p>

<p style="margin-top: 1em">--version <br>
Display version information.</p>

<p style="margin-top: 1em">--copyright <br>
Display long copyright information.</p>

<p style="margin-top: 1em">-v, --verbosity INTEGER <br>
Display more/less diagnostic output during operation: 0
[only errors]; 1 [default]; 2 [+run-time, options and
statistics]. In range [0..2]. Default: 1.</p>

<p style="margin-top: 1em">Input Options: <br>
-d, --database INPUT_FILE <br>
Database sequences. Valid filetypes are: .sam[.*], .raw[.*],
.gbk[.*], .frn[.*], .fq[.*], .fna[.*], .ffn[.*], .fastq[.*],
.fasta[.*], .faa[.*], .fa[.*], .embl[.*], and <br>
.bam, where * is any of the following extensions: gz, bz2,
and bgzf for transparent (de)compression.</p>

<p style="margin-top: 1em">-s, --segfile INPUT_FILE <br>
SEG intervals for database(optional). Valid filetype is:
.seg.</p>

<p style="margin-top: 1em">Output Options: <br>
-di, --db-index-type STRING <br>
Suffix array or full-text minute space. One of sa and fm.
Default: fm.</p>

<p style="margin-top: 1em">--truncate-ids STRING <br>
Truncate IDs at first whitespace. This saves a lot of space
and is irrelevant for all LAMBDA output formats other than
BLAST Pairwise (.m0). One of on and off. Default: <br>
on.</p>

<p style="margin-top: 1em">Alphabets and Translation: <br>
-p, --program STRING <br>
Blast Operation Mode. One of blastn, blastp, blastx,
tblastn, and tblastx. Default: blastx.</p>

<p style="margin-top: 1em">-g, --genetic-code INTEGER <br>
The translation table to use (not for BlastN, BlastP). See
https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c
for ids (default is generic). Default: 1.</p>

<p style="margin-top: 1em">-ar, --alphabet-reduction STRING
<br>
Alphabet Reduction for seeding phase (ignored for BLASTN).
One of none and murphy10. Default: murphy10.</p>

<p style="margin-top: 1em">Algorithm: <br>
-a, --algorithm STRING <br>
Algorithm for SA construction (also used for FM; see Memory
Requirements below!). One of mergesort, quicksortbuckets,
quicksort, radixsort, and skew7ext. Default: radix&acirc;
<br>
sort.</p>

<p style="margin-top: 1em">-t, --threads INTEGER <br>
number of threads to run concurrently (ignored if a ==
skew7ext).</p>

<p style="margin-top: 1em">-td, --tmp-dir STRING <br>
temporary directory used by skew, defaults to working
directory.</p>

<p style="margin-top: 1em">REMARKS <br>
Please see the wiki
(&lt;https://github.com/seqan/lambda/wiki&gt;) for more
information on which indexes to chose and which algorithms
to pick.</p>

<p style="margin-top: 1em">Note that the indexes created
are binary and not compatible between different CPU
endiannesses. Also the on-disk format is still subject to
change between Lambda versions.</p>

<p style="margin-top: 1em">LEGAL <br>
lambda_indexer Copyright: 2013-2017 Hannes Hauswedell,
released under the GNU GPL v3 (or later); 2016-2017 Knut
Reinert and Freie Universit&Atilde;&curren;t Berlin,
released under the 3-clause- <br>
BSDL <br>
SeqAn Copyright: 2006-2015 Knut Reinert, FU-Berlin; released
under the 3-clause BSDL. <br>
In your academic works please cite: Hauswedell et al (2014);
doi: 10.1093/bioinformatics/btu439 <br>
For full copyright and/or warranty information see
--copyright.</p>

<p style="margin-top: 1em">lambda_indexer 1.0.1 Jan 21 2017
LAMBDA_INDEXER(1)</p>
<hr>
</body>
</html>
