<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:21:14 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>LAMBDA(1) LAMBDA(1)</p>

<p style="margin-top: 1em">NAME <br>
lambda - the Local Aligner for Massive Biological DatA</p>

<p style="margin-top: 1em">SYNOPSIS <br>
lambda [OPTIONS] -q QUERY.fasta -d DATABASE.fasta [-o
output.m8]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Lambda is a local aligner optimized for many query sequences
and searches in protein space. It is compatible to BLAST,
but much faster than BLAST and many other comparable
tools.</p>

<p style="margin-top: 1em">Detailed information is
available in the wiki:
&lt;https://github.com/seqan/lambda/wiki&gt;</p>

<p style="margin-top: 1em">OPTIONS <br>
-h, --help <br>
Display the help message.</p>

<p style="margin-top: 1em">-hh, --full-help <br>
Display the help message with advanced options.</p>

<p style="margin-top: 1em">--version <br>
Display version information.</p>

<p style="margin-top: 1em">--copyright <br>
Display long copyright information.</p>

<p style="margin-top: 1em">-v, --verbosity INTEGER <br>
Display more/less diagnostic output during operation: 0
[only errors]; 1 [default]; 2 [+run-time, options and
statistics]. In range [0..2]. Default: 1.</p>

<p style="margin-top: 1em">Input Options: <br>
-q, --query INPUT_FILE <br>
Query sequences. Valid filetypes are: .sam[.*], .raw[.*],
.gbk[.*], .frn[.*], .fq[.*], .fna[.*], .ffn[.*], .fastq[.*],
.fasta[.*], .faa[.*], .fa[.*], .embl[.*], and .bam, <br>
where * is any of the following extensions: gz, bz2, and
bgzf for transparent (de)compression.</p>

<p style="margin-top: 1em">-d, --database INPUT_FILE <br>
Path to original database sequences (a precomputed index
with .sa or .fm needs to exist!). Valid filetypes are:
.sam[.*], .raw[.*], .gbk[.*], .frn[.*], .fq[.*], .fna[.*],
<br>
.ffn[.*], .fastq[.*], .fasta[.*], .faa[.*], .fa[.*],
.embl[.*], and .bam, where * is any of the following
extensions: gz, bz2, and bgzf for transparent
(de)compression.</p>

<p style="margin-top: 1em">-di, --db-index-type STRING <br>
database index is in this format. One of sa and fm. Default:
fm.</p>

<p style="margin-top: 1em">Output Options: <br>
-o, --output OUTPUT_FILE <br>
File to hold reports on hits (.m* are blastall -m* formats;
.m8 is tab-seperated, .m9 is tab-seperated with with
comments, .m0 is pairwise format). Valid filetypes are: <br>
.sam[.*], .m9[.*], .m8[.*], .m0[.*], and .bam, where * is
any of the following extensions: gz, bz2, and bgzf for
transparent (de)compression. Default: output.m8.</p>

<p style="margin-top: 1em">-oc, --output-columns STRING
<br>
Print specified column combination and/or order (.m8 and .m9
outputs only); call -oc help for more details. Default:
std.</p>

<p style="margin-top: 1em">-id, --percent-identity INTEGER
<br>
Output only matches above this threshold (checked before
e-value check). In range [0..100]. Default: 0.</p>

<p style="margin-top: 1em">-e, --e-value DOUBLE <br>
Output only matches that score below this threshold. In
range [0..inf]. Default: 0.1.</p>

<p style="margin-top: 1em">-nm, --num-matches INTEGER <br>
Print at most this number of matches per query. In range
[1..inf]. Default: 500.</p>

<p style="margin-top: 1em">--sam-with-refheader STRING <br>
BAM files require all subject names to be written to the
header. For SAM this is not required, so Lambda does not
automatically do it to save space (especially for protein
<br>
database this is a lot!). If you still want them with SAM,
e.g. for better BAM compatibility, use this option. One of
on and off. Default: off.</p>

<p style="margin-top: 1em">--sam-bam-seq STRING <br>
Write matching DNA subsequence into SAM/BAM file (BLASTN).
For BLASTX and TBLASTX the matching protein sequence is
&quot;untranslated&quot; and positions retransformed to the
origi&acirc; <br>
nal sequence. For BLASTP and TBLASTN there is no DNA
sequence so a &quot;*&quot; is written to the SEQ column.
The matching protein sequence can be written as an optional
tag, see <br>
--sam-bam-tags. If set to uniq than the sequence is omitted
iff it is identical to the previous match&rsquo;s
subsequence. One of always, uniq, and never. Default:
uniq.</p>

<p style="margin-top: 1em">--sam-bam-tags STRING <br>
Write the specified optional columns to the SAM/BAM file.
Call --sam-bam-tags help for more details. Default: AS NM ZE
ZI ZF.</p>

<p style="margin-top: 1em">--sam-bam-clip STRING <br>
Whether to hard-clip or soft-clip the regions beyond the
local match. Soft-clipping retains the full sequence in the
output file, but obviously uses more space. One of <br>
hard and soft. Default: hard.</p>

<p style="margin-top: 1em">General Options: <br>
-t, --threads INTEGER <br>
number of threads to run concurrently.</p>

<p style="margin-top: 1em">-qi, --query-index-type STRING
<br>
controls double-indexing. One of radix and none. Default:
none.</p>

<p style="margin-top: 1em">Alphabets and Translation: <br>
-p, --program STRING <br>
Blast Operation Mode. One of blastn, blastp, blastx,
tblastn, and tblastx. Default: blastx.</p>

<p style="margin-top: 1em">-g, --genetic-code INTEGER <br>
The translation table to use for nucl -&gt; amino acid
translation(not for BlastN, BlastP). See
https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c
for ids <br>
(default is generic). Six frames are generated. Default:
1.</p>

<p style="margin-top: 1em">-ar, --alphabet-reduction STRING
<br>
Alphabet Reduction for seeding phase (ignored for BLASTN).
One of none and murphy10. Default: murphy10.</p>

<p style="margin-top: 1em">Seeding / Filtration: <br>
-sl, --seed-length INTEGER <br>
Length of the seeds (default = 14 for BLASTN). Default:
10.</p>

<p style="margin-top: 1em">-so, --seed-offset INTEGER <br>
Offset for seeding (if unset = seed-length, non-overlapping;
default = 5 for BLASTN). Default: 10.</p>

<p style="margin-top: 1em">-sd, --seed-delta INTEGER <br>
maximum seed distance. Default: 1.</p>

<p style="margin-top: 1em">Miscellaneous Heuristics: <br>
-ps, --pre-scoring INTEGER <br>
evaluate score of a region NUM times the size of the seed
before extension (0 -&gt; no pre-scoring, 1 -&gt; evaluate
seed, n-&gt; area around seed, as well; default = 1 if no
<br>
reduction is used). In range [1..inf]. Default: 2.</p>

<p style="margin-top: 1em">-pt, --pre-scoring-threshold
DOUBLE <br>
minimum average score per position in pre-scoring region.
Default: 2.</p>

<p style="margin-top: 1em">-pd,
--filter-putative-duplicates STRING <br>
filter hits that will likely duplicate a match already
found. One of on and off. Default: on.</p>

<p style="margin-top: 1em">-pa, --filter-putative-abundant
STRING <br>
If the maximum number of matches per query are found
already, stop searching if the remaining realm looks
unfeasable. One of on and off. Default: on.</p>

<p style="margin-top: 1em">Scoring: <br>
-sc, --scoring-scheme INTEGER <br>
use &rsquo;45&rsquo; for Blosum45; &rsquo;62&rsquo; for
Blosum62 (default); &rsquo;80&rsquo; for Blosum80; [ignored
for BlastN] Default: 62.</p>

<p style="margin-top: 1em">-ge, --score-gap INTEGER <br>
Score per gap character (default = -2 for BLASTN). Default:
-1.</p>

<p style="margin-top: 1em">-go, --score-gap-open INTEGER
<br>
Additional cost for opening gap (default = -5 for BLASTN).
Default: -11.</p>

<p style="margin-top: 1em">-ma, --score-match INTEGER <br>
Match score [only BLASTN]) Default: 2.</p>

<p style="margin-top: 1em">-mi, --score-mismatch INTEGER
<br>
Mismatch score [only BLASTN] Default: -3.</p>

<p style="margin-top: 1em">Extension: <br>
-x, --x-drop INTEGER <br>
Stop Banded extension if score x below the maximum seen (-1
means no xdrop). In range [-1..inf]. Default: 30.</p>

<p style="margin-top: 1em">-b, --band INTEGER <br>
Size of the DP-band used in extension (-3 means log2 of
query length; -2 means sqrt of query length; -1 means full
dp; n means band of size 2n+1) In range [-3..inf]. <br>
Default: -3.</p>

<p style="margin-top: 1em">TUNING <br>
Tuning the seeding parameters and (de)activating alphabet
reduction has a strong influence on both speed and
sensitivity. We recommend the following alternative profiles
for pro&acirc; <br>
tein searches:</p>

<p style="margin-top: 1em">fast (high similarity): -ar none
-sl 7 -sd 0</p>

<p style="margin-top: 1em">sensitive (lower similarity):
-so 5</p>

<p style="margin-top: 1em">For further information see the
wiki: &lt;https://github.com/seqan/lambda/wiki&gt;</p>

<p style="margin-top: 1em">LEGAL <br>
lambda Copyright: 2013-2017 Hannes Hauswedell, released
under the GNU GPL v3 (or later); 2016-2017 Knut Reinert and
Freie Universit&Atilde;&curren;t Berlin, released under the
3-clause-BSDL <br>
SeqAn Copyright: 2006-2015 Knut Reinert, FU-Berlin; released
under the 3-clause BSDL. <br>
In your academic works please cite: Hauswedell et al (2014);
doi: 10.1093/bioinformatics/btu439 <br>
For full copyright and/or warranty information see
--copyright.</p>

<p style="margin-top: 1em">lambda 1.0.1 Jan 21 2017
LAMBDA(1)</p>
<hr>
</body>
</html>
