<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:21:17 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>LASTAL8(1) User Commands LASTAL8(1)</p>

<p style="margin-top: 1em">NAME <br>
lastal8 - genome-scale comparison of biological
sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
lastal8 [options] lastdb-name fasta-sequence-file(s)</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Find and align similar sequences.</p>

<p style="margin-top: 1em">Cosmetic options: -h, --help:
show all options and their default settings, and exit -V,
--version: show version information, and exit -v: be
verbose: write messages about what <br>
lastal is doing -f: output format: TAB, MAF, BlastTab,
BlastTab+ (default=MAF)</p>

<p style="margin-top: 1em">E-value options (default
settings): -D: query letters per random alignment (1e+06)
-E: maximum expected alignments per square giga
(1e+18/D/refSize/numOfStrands)</p>

<p style="margin-top: 1em">Score options (default
settings): -r: match score (2 if -M, else 6 if 0&lt;Q&lt;5,
else 1 if DNA) -q: mismatch cost (3 if -M, else 18 if
0&lt;Q&lt;5, else 1 if DNA) -p: match/mismatch <br>
score matrix (protein-protein: BL62, DNA-protein: BL80) -a:
gap existence cost (DNA: 7, protein: 11, 0&lt;Q&lt;5: 21)
-b: gap extension cost (DNA: 1, protein: 2, 0&lt;Q&lt;5: 9)
-A: inser&acirc; <br>
tion existence cost (a) -B: insertion extension cost (b) -c:
unaligned residue pair cost (off) -F: frameshift cost (off)
-x: maximum score drop for gapped alignments (max[y, <br>
e-1]) -y: maximum score drop for gapless alignments (t*10)
-z: maximum score drop for final gapped alignments (x) -d:
minimum score for gapless alignments (min[e,
t*ln(1000*ref&acirc; <br>
Size/n)]) -e: minimum score for gapped alignments</p>

<p style="margin-top: 1em">Initial-match options (default
settings): -m: maximum initial matches per query position
(10) -l: minimum length for initial matches (1) -L: maximum
length for initial matches <br>
(infinity) -k: use initial matches starting at every k-th
position in each query (1) -W: use &quot;minimum&quot;
positions in sliding windows of W consecutive positions</p>

<p style="margin-top: 1em">Miscellaneous options (default
settings): -s: strand: 0=reverse, 1=forward, 2=both (2 for
DNA, 1 for protein) -S: score matrix applies to forward
strand of: 0=reference, 1=query <br>
(0) -K: omit alignments whose query range lies in &gt;= K
others with &gt; score (off) -C: omit gapless alignments in
&gt;= C others with &gt; score-per-length (off) -P: number
of parallel <br>
threads (1) -i: query batch size (8 KiB, unless there is
&gt; 1 thread or lastdb volume) -M: find minimum-difference
alignments (faster but cruder) -T: type of alignment:
0=local, <br>
1=overlap (0) -n: maximum gapless alignments per query
position (infinity if m=0, else m) -R: repeat-marking
options (the same as was used for lastdb) -u: mask lowercase
during <br>
extensions: 0=never, 1=gapless,</p>

<p style="margin-top: 1em">2=gapless+postmask, 3=always (2
if lastdb -c and Q&lt;5, else 0)</p>

<p style="margin-top: 1em">-w: suppress repeats inside
exact matches, offset by &lt;= this distance (1000) -G:
genetic code file -t: &rsquo;temperature&rsquo; for
calculating probabilities (1/lambda) -g: &rsquo;gamma&rsquo;
parame&acirc; <br>
ter for gamma-centroid and LAMA (1) -j: output type: 0=match
counts, 1=gapless, 2=redundant gapped, 3=gapped,</p>

<p style="margin-top: 1em">4=column ambiguity estimates,
5=gamma-centroid, 6=LAMA, 7=expected counts (3)</p>

<p style="margin-top: 1em">-Q: input format: 0=fasta,
1=fastq-sanger, 2=fastq-solexa, 3=fastq-illumina,</p>

<p style="margin-top: 1em">4=prb, 5=PSSM (0)</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to: last-align (ATmark) googlegroups (dot) com
<br>
LAST home page: http://last.cbrc.jp/</p>

<p style="margin-top: 1em">lastal8 830 January 2017
LASTAL8(1)</p>
<hr>
</body>
</html>
