<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>LASTAL(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">LASTAL(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">LASTAL(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
lastal - genome-scale comparison of biological sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>lastal</b> [ <i>options</i>] <i>lastdb-name fasta-sequence-file(s)</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Find and align similar sequences.
<div class="Pp"></div>
Cosmetic options: <b>-h</b>, <b>--help</b>: show all options and their default
  settings, and exit <b>-V</b>, <b>--version</b>: show version information, and
  exit <b>-v</b>: be verbose: write messages about what lastal is doing
  <b>-f</b>: output format: TAB, MAF, BlastTab, BlastTab+ (default=MAF)
<div class="Pp"></div>
E-value options (default settings): <b>-D</b>: query letters per random
  alignment (1e+06) <b>-E</b>: maximum expected alignments per square giga
  (1e+18/D/refSize/numOfStrands)
<div class="Pp"></div>
Score options (default settings): <b>-r</b>: match score (2 if <b>-M</b>, else 6
  if 0&lt;Q&lt;5, else 1 if DNA) <b>-q</b>: mismatch cost (3 if <b>-M</b>, else
  18 if 0&lt;Q&lt;5, else 1 if DNA) <b>-p</b>: match/mismatch score matrix
  (protein-protein: BL62, DNA-protein: BL80) <b>-a</b>: gap existence cost (DNA:
  7, protein: 11, 0&lt;Q&lt;5: 21) <b>-b</b>: gap extension cost (DNA: 1,
  protein: 2, 0&lt;Q&lt;5: 9) <b>-A</b>: insertion existence cost (a) <b>-B</b>:
  insertion extension cost (b) <b>-c</b>: unaligned residue pair cost (off)
  <b>-F</b>: frameshift cost (off) <b>-x</b>: maximum score drop for gapped
  alignments (max[y, e-1]) <b>-y</b>: maximum score drop for gapless alignments
  (t*10) <b>-z</b>: maximum score drop for final gapped alignments (x)
  <b>-d</b>: minimum score for gapless alignments (min[e, t*ln(1000*refSize/n)])
  <b>-e</b>: minimum score for gapped alignments
<div class="Pp"></div>
Initial-match options (default settings): <b>-m</b>: maximum initial matches per
  query position (10) <b>-l</b>: minimum length for initial matches (1)
  <b>-L</b>: maximum length for initial matches (infinity) <b>-k</b>: use
  initial matches starting at every k-th position in each query (1) <b>-W</b>:
  use &quot;minimum&quot; positions in sliding windows of W consecutive
  positions
<div class="Pp"></div>
Miscellaneous options (default settings): <b>-s</b>: strand: 0=reverse,
  1=forward, 2=both (2 for DNA, 1 for protein) <b>-S</b>: score matrix applies
  to forward strand of: 0=reference, 1=query (0) <b>-K</b>: omit alignments
  whose query range lies in &gt;= K others with &gt; score (off) <b>-C</b>: omit
  gapless alignments in &gt;= C others with &gt; score-per-length (off)
  <b>-P</b>: number of parallel threads (1) <b>-i</b>: query batch size (8 KiB,
  unless there is &gt; 1 thread or lastdb volume) <b>-M</b>: find
  minimum-difference alignments (faster but cruder) <b>-T</b>: type of
  alignment: 0=local, 1=overlap (0) <b>-n</b>: maximum gapless alignments per
  query position (infinity if m=0, else m) <b>-R</b>: repeat-marking options
  (the same as was used for lastdb) <b>-u</b>: mask lowercase during extensions:
  0=never, 1=gapless,
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">2=gapless+postmask, 3=always (2 if lastdb <b>-c</b> and
      Q&lt;5, else 0)</dd>
</dl>
<div class="Pp"></div>
<b>-w</b>: suppress repeats inside exact matches, offset by &lt;= this distance
  (1000) <b>-G</b>: genetic code file <b>-t</b>: 'temperature' for calculating
  probabilities (1/lambda) <b>-g</b>: 'gamma' parameter for gamma-centroid and
  LAMA (1) <b>-j</b>: output type: 0=match counts, 1=gapless, 2=redundant
  gapped, 3=gapped,
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">4=column ambiguity estimates, 5=gamma-centroid, 6=LAMA,
      7=expected counts (3)</dd>
</dl>
<div class="Pp"></div>
<b>-Q</b>: input format: 0=fasta, 1=fastq-sanger, 2=fastq-solexa,
  3=fastq-illumina,
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">4=prb, 5=PSSM (0)</dd>
</dl>
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
Report bugs to: last-align (ATmark) googlegroups (dot) com
<div>&#x00A0;</div>
LAST home page: http://last.cbrc.jp/</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2017</td>
    <td class="foot-os">lastal 830</td>
  </tr>
</table>
</body>
</html>
