<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:21:18 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>LAST-TRAIN(1) User Commands LAST-TRAIN(1)</p>

<p style="margin-top: 1em">NAME <br>
last-train - Try to find suitable score parameters for
aligning the given sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
last-train [options] lastdb-name sequence-file(s)</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Try to find suitable score parameters for aligning the given
sequences.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">Training options:</p>

<p style="margin-top: 1em">--revsym <br>
force reverse-complement symmetry</p>

<p style="margin-top: 1em">--matsym <br>
force symmetric substitution matrix</p>

<p style="margin-top: 1em">--gapsym <br>
force insertion/deletion symmetry</p>

<p style="margin-top: 1em">--pid=PID <br>
skip alignments with &gt; PID% identity (default: 100)</p>

<p style="margin-top: 1em">--sample=N <br>
use a random sample of N letters (default: 1000000)</p>

<p style="margin-top: 1em">Initial parameter options:</p>

<p style="margin-top: 1em">-r SCORE <br>
match score (default: 6 if Q&gt;0, else 5)</p>

<p style="margin-top: 1em">-q COST <br>
mismatch cost (default: 18 if Q&gt;0, else 5)</p>

<p style="margin-top: 1em">-p NAME <br>
match/mismatch score matrix</p>

<p style="margin-top: 1em">-a COST <br>
gap existence cost (default: 21 if Q&gt;0, else 15)</p>

<p style="margin-top: 1em">-b COST <br>
gap extension cost (default: 9 if Q&gt;0, else 3)</p>

<p style="margin-top: 1em">-A COST <br>
insertion existence cost</p>

<p style="margin-top: 1em">-B COST <br>
insertion extension cost</p>

<p style="margin-top: 1em">Alignment options:</p>

<p style="margin-top: 1em">-D LENGTH <br>
query letters per random alignment (default: 1e6)</p>

<p style="margin-top: 1em">-E EG2 maximum expected
alignments per square giga</p>

<p style="margin-top: 1em">-s STRAND <br>
0=reverse, 1=forward, 2=both (default: 2 if DNA, else 1)</p>

<p style="margin-top: 1em">-S NUMBER <br>
score matrix applies to forward strand of: 0=reference,
1=query (default: 1)</p>

<p style="margin-top: 1em">-T NUMBER <br>
type of alignment: 0=local, 1=overlap (default: 0)</p>

<p style="margin-top: 1em">-m COUNT <br>
maximum initial matches per query position (default: 10)</p>

<p style="margin-top: 1em">-P THREADS <br>
number of parallel threads</p>

<p style="margin-top: 1em">-Q NUMBER <br>
input format: 0=fasta, 1=fastq-sanger</p>

<p style="margin-top: 1em">last-train 830 January 2017
LAST-TRAIN(1)</p>
<hr>
</body>
</html>
