<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>LEAFF(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">LEAFF(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">LEAFF(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
leaff - sequence library utilities and applications
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>leaff</b> [-f fasta-file] [options]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
LEAFF (Let's Extract Anything From Fasta) is a utility program for working with
  multi-fasta files. In addition to providing random access to the base level,
  it includes several analysis functions.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
SOURCE FILES
<br/>
 -f file: use sequence in 'file' (-F is also allowed for historical reasons)
<br/>
 -A file: read actions from 'file'
<div style="height: 1.00em;">&#x00A0;</div>
SOURCE FILE EXAMINATION
<br/>
 -d: print the number of sequences in the fasta
<br/>
 -i name: print an index, labelling the source 'name'
<div style="height: 1.00em;">&#x00A0;</div>
OUTPUT OPTIONS
<br/>
 -6 &lt;#&gt;: insert a newline every 60 letters
<br/>
 (if the next arg is a number, newlines are inserted every
<br/>
 n letters, e.g., -6 80. Disable line breaks with -6 0,
<br/>
 or just don't use -6!)
<br/>
 -e beg end: Print only the bases from position 'beg' to position 'end'
<br/>
 (space based, relative to the FORWARD sequence!) If
<br/>
 beg == end, then the entire sequence is printed. It is an
<br/>
 error to specify beg &gt; end, or beg &gt; len, or end &gt; len.
<br/>
 -ends n Print n bases from each end of the sequence. One input
<br/>
 sequence generates two output sequences, with '_5' or '_3'
<br/>
 appended to the ID. If 2n &gt;= length of the sequence, the
<br/>
 sequence itself is printed, no ends are extracted (they
<br/>
 overlap).
<br/>
 -C: complement the sequences
<br/>
 -H: DON'T print the defline
<br/>
 -h: Use the next word as the defline (&quot;-H -H&quot; will reset to the
<br/>
 original defline
<br/>
 -R: reverse the sequences
<br/>
 -u: uppercase all bases
<div style="height: 1.00em;">&#x00A0;</div>
SEQUENCE SELECTION
<br/>
 -G n s l: print n randomly generated sequences, 0 &lt; s &lt;= length &lt;= l
<br/>
 -L s l: print all sequences such that s &lt;= length &lt; l
<br/>
 -N l h: print all sequences such that l &lt;= % N composition &lt; h
<br/>
 (NOTE 0.0 &lt;= l &lt; h &lt; 100.0)
<br/>
 (NOTE that you cannot print sequences with 100% N
<br/>
 This is a useful bug).
<br/>
 -q file: print sequences from the seqid list in 'file'
<br/>
 -r num: print 'num' randomly picked sequences
<br/>
 -s seqid: print the single sequence 'seqid'
<br/>
 -S f l: print all the sequences from ID 'f' to 'l' (inclusive)
<br/>
 -W: print all sequences (do the whole file)
<div style="height: 1.00em;">&#x00A0;</div>
LONGER HELP
<br/>
 -help analysis
<br/>
 -help examples
<div style="height: 1.00em;">&#x00A0;</div>
ANALYSIS FUNCTIONS
<br/>
 --findduplicates a.fasta
<br/>
 Reports sequences that are present more than once. Output
<br/>
 is a list of pairs of deflines, separated by a newline.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --mapduplicates a.fasta b.fasta
<br/>
 Builds a map of IIDs from a.fasta and b.fasta that have
<br/>
 identical sequences. Format is &quot;IIDa &lt;-&gt; IIDb&quot;
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --md5 a.fasta:
<br/>
 Don't print the sequence, but print the md5 checksum
<br/>
 (of the entire sequence) followed by the entire defline.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --partition prefix [ n[gmk]bp | n ] a.fasta
<br/>
 --partitionmap [ n[gmk]bp | n ] a.fasta
<br/>
 Partition the sequences into roughly equal size pieces of
<br/>
 size nbp, nkbp, nmbp or ngbp; or into n roughly equal sized
<br/>
 parititions. Sequences larger that the partition size are
<br/>
 in a partition by themself. --partitionmap writes a
<br/>
 description of the partition to stdout; --partiton creates
<br/>
 a fasta file 'prefix-###.fasta' for each partition.
<br/>
 Example: -F some.fasta --partition parts 130mbp
<br/>
 -F some.fasta --partition parts 16
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --segment prefix n a.fasta
<br/>
 Splits the sequences into n files, prefix-###.fasta.
<br/>
 Sequences are not reordered; the first n sequences are in
<br/>
 the first file, the next n in the second file, etc.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --gccontent a.fasta
<br/>
 Reports the GC content over a sliding window of
<br/>
 3, 5, 11, 51, 101, 201, 501, 1001, 2001 bp.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --testindex a.fasta
<br/>
 Test the index of 'file'. If index is up-to-date, leaff
<br/>
 exits successfully, else, leaff exits with code 1. If an
<br/>
 index file is supplied, that one is tested, otherwise, the
<br/>
 default index file name is used.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --dumpblocks a.fasta
<br/>
 Generates a list of the blocks of N and non-N. Output
<br/>
 format is 'base seq# beg end len'. 'N 84 483 485 2' means
<br/>
 that a block of 2 N's starts at space-based position 483
<br/>
 in sequence ordinal 84. A '.' is the end of sequence
<br/>
 marker.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --errors L N C P a.fasta
<br/>
 For every sequence in the input file, generate new
<br/>
 sequences including simulated sequencing errors.
<br/>
 L -- length of the new sequence. If zero, the length
<br/>
 of the original sequence will be used.
<br/>
 N -- number of subsequences to generate. If L=0, all
<br/>
 subsequences will be the same, and you should use
<br/>
 C instead.
<br/>
 C -- number of copies to generate. Each of the N
<br/>
 subsequences will have C copies, each with different
<br/>
 errors.
<br/>
 P -- probability of an error.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 HINT: to simulate ESTs from genes, use L=500, N=10, C=10
<br/>
 -- make C=10 sequencer runs of N=10 EST sequences
<br/>
 of length 500bp each.
<br/>
 to simulate mRNA from genes, use L=0, N=10, C=10
<br/>
 to simulate reads from genomes, use L=800, N=10, C=1
<br/>
 -- of course, N= should be increased to give the
<br/>
 appropriate depth of coverage
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --stats a.fasta
<br/>
 Reports size statistics; number, N50, sum, largest.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 --seqstore out.seqStore
<br/>
 Converts the input file (-f) to a seqStore file (for instance,
<br/>
 for use with the Celera assembler or sim4db).
<h1 class="Sh" title="Sh" id="NOTES"><a class="selflink" href="#NOTES">NOTES</a></h1>
Please note that options are ORDER DEPENDENT. Sequences are printed whenever a
  SEQUENCE SELECTION option occurs on the command line. OUTPUT OPTIONS are not
  reset when a sequence is printed.
<div class="Pp"></div>
SEQUENCES are numbered starting at ZERO, not one!
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
1. Print the first 10 bases of the fourth sequence in file 'genes':
<div>&#x00A0;</div>
	leaff -f genes -e 0 10 -s 3
<div style="height: 1.00em;">&#x00A0;</div>
2. Print the first 10 bases of the fourth and fifth sequences:
<div>&#x00A0;</div>
	leaff -f genes -e 0 10 -s 3 -s 4
<div style="height: 1.00em;">&#x00A0;</div>
3. Print the fourth and fifth sequences reverse complemented, and the sixth
<br/>
 sequence forward. The second set of -R -C toggle off reverse-complement:
<div>&#x00A0;</div>
	leaff -f genes -R -C -s 3 -s 4 -R -C -s 5
<div style="height: 1.00em;">&#x00A0;</div>
4. Convert file 'genes' to a seqStore 'genes.seqStore'.
<div>&#x00A0;</div>
	leaff -f genes --seqstore genes.seqStore
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
README.leaff
<div>&#x00A0;</div>
http://kmer.sourceforge.net/wiki/index.php/LEAFF_User%27s_Guide
<div>&#x00A0;</div>
http://kmer.sourceforge.net/wiki/index.php/LEAFF_Programming_Example</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2016</td>
    <td class="foot-os"></td>
  </tr>
</table>
</body>
</html>
