<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:21:37 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>LEAFF(1) General Commands Manual LEAFF(1)</p>

<p style="margin-top: 1em">NAME <br>
leaff - sequence library utilities and applications</p>

<p style="margin-top: 1em">SYNOPSIS <br>
leaff [-f fasta-file] [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
LEAFF (Let&rsquo;s Extract Anything From Fasta) is a utility
program for working with multi-fasta files. In addition to
providing random access to the base level, it includes
several <br>
analysis functions.</p>

<p style="margin-top: 1em">OPTIONS <br>
SOURCE FILES <br>
-f file: use sequence in &rsquo;file&rsquo; (-F is also
allowed for historical reasons) <br>
-A file: read actions from &rsquo;file&rsquo;</p>

<p style="margin-top: 1em">SOURCE FILE EXAMINATION <br>
-d: print the number of sequences in the fasta <br>
-i name: print an index, labelling the source
&rsquo;name&rsquo;</p>

<p style="margin-top: 1em">OUTPUT OPTIONS <br>
-6 &lt;#&gt;: insert a newline every 60 letters <br>
(if the next arg is a number, newlines are inserted every
<br>
n letters, e.g., -6 80. Disable line breaks with -6 0, <br>
or just don&rsquo;t use -6!) <br>
-e beg end: Print only the bases from position
&rsquo;beg&rsquo; to position &rsquo;end&rsquo; <br>
(space based, relative to the FORWARD sequence!) If <br>
beg == end, then the entire sequence is printed. It is an
<br>
error to specify beg &gt; end, or beg &gt; len, or end &gt;
len. <br>
-ends n Print n bases from each end of the sequence. One
input <br>
sequence generates two output sequences, with
&rsquo;_5&rsquo; or &rsquo;_3&rsquo; <br>
appended to the ID. If 2n &gt;= length of the sequence, the
<br>
sequence itself is printed, no ends are extracted (they <br>
overlap). <br>
-C: complement the sequences <br>
-H: DON&rsquo;T print the defline <br>
-h: Use the next word as the defline (&quot;-H -H&quot; will
reset to the <br>
original defline <br>
-R: reverse the sequences <br>
-u: uppercase all bases</p>

<p style="margin-top: 1em">SEQUENCE SELECTION <br>
-G n s l: print n randomly generated sequences, 0 &lt; s
&lt;= length &lt;= l <br>
-L s l: print all sequences such that s &lt;= length &lt; l
<br>
-N l h: print all sequences such that l &lt;= % N
composition &lt; h <br>
(NOTE 0.0 &lt;= l &lt; h &lt; 100.0) <br>
(NOTE that you cannot print sequences with 100% N <br>
This is a useful bug). <br>
-q file: print sequences from the seqid list in
&rsquo;file&rsquo; <br>
-r num: print &rsquo;num&rsquo; randomly picked sequences
<br>
-s seqid: print the single sequence &rsquo;seqid&rsquo; <br>
-S f l: print all the sequences from ID &rsquo;f&rsquo; to
&rsquo;l&rsquo; (inclusive) <br>
-W: print all sequences (do the whole file)</p>

<p style="margin-top: 1em">LONGER HELP <br>
-help analysis <br>
-help examples</p>

<p style="margin-top: 1em">ANALYSIS FUNCTIONS <br>
--findduplicates a.fasta <br>
Reports sequences that are present more than once. Output
<br>
is a list of pairs of deflines, separated by a newline.</p>

<p style="margin-top: 1em">--mapduplicates a.fasta b.fasta
<br>
Builds a map of IIDs from a.fasta and b.fasta that have <br>
identical sequences. Format is &quot;IIDa &lt;-&gt;
IIDb&quot;</p>

<p style="margin-top: 1em">--md5 a.fasta: <br>
Don&rsquo;t print the sequence, but print the md5 checksum
<br>
(of the entire sequence) followed by the entire defline.</p>

<p style="margin-top: 1em">--partition prefix [ n[gmk]bp |
n ] a.fasta <br>
--partitionmap [ n[gmk]bp | n ] a.fasta <br>
Partition the sequences into roughly equal size pieces of
<br>
size nbp, nkbp, nmbp or ngbp; or into n roughly equal sized
<br>
parititions. Sequences larger that the partition size are
<br>
in a partition by themself. --partitionmap writes a <br>
description of the partition to stdout; --partiton creates
<br>
a fasta file &rsquo;prefix-###.fasta&rsquo; for each
partition. <br>
Example: -F some.fasta --partition parts 130mbp <br>
-F some.fasta --partition parts 16</p>

<p style="margin-top: 1em">--segment prefix n a.fasta <br>
Splits the sequences into n files, prefix-###.fasta. <br>
Sequences are not reordered; the first n sequences are in
<br>
the first file, the next n in the second file, etc.</p>

<p style="margin-top: 1em">--gccontent a.fasta <br>
Reports the GC content over a sliding window of <br>
3, 5, 11, 51, 101, 201, 501, 1001, 2001 bp.</p>

<p style="margin-top: 1em">--testindex a.fasta <br>
Test the index of &rsquo;file&rsquo;. If index is
up-to-date, leaff <br>
exits successfully, else, leaff exits with code 1. If an
<br>
index file is supplied, that one is tested, otherwise, the
<br>
default index file name is used.</p>

<p style="margin-top: 1em">--dumpblocks a.fasta <br>
Generates a list of the blocks of N and non-N. Output <br>
format is &rsquo;base seq# beg end len&rsquo;. &rsquo;N 84
483 485 2&rsquo; means <br>
that a block of 2 N&rsquo;s starts at space-based position
483 <br>
in sequence ordinal 84. A &rsquo;.&rsquo; is the end of
sequence <br>
marker.</p>

<p style="margin-top: 1em">--errors L N C P a.fasta <br>
For every sequence in the input file, generate new <br>
sequences including simulated sequencing errors. <br>
L -- length of the new sequence. If zero, the length <br>
of the original sequence will be used. <br>
N -- number of subsequences to generate. If L=0, all <br>
subsequences will be the same, and you should use <br>
C instead. <br>
C -- number of copies to generate. Each of the N <br>
subsequences will have C copies, each with different <br>
errors. <br>
P -- probability of an error.</p>

<p style="margin-top: 1em">HINT: to simulate ESTs from
genes, use L=500, N=10, C=10 <br>
-- make C=10 sequencer runs of N=10 EST sequences <br>
of length 500bp each. <br>
to simulate mRNA from genes, use L=0, N=10, C=10 <br>
to simulate reads from genomes, use L=800, N=10, C=1 <br>
-- of course, N= should be increased to give the <br>
appropriate depth of coverage</p>

<p style="margin-top: 1em">--stats a.fasta <br>
Reports size statistics; number, N50, sum, largest.</p>

<p style="margin-top: 1em">--seqstore out.seqStore <br>
Converts the input file (-f) to a seqStore file (for
instance, <br>
for use with the Celera assembler or sim4db).</p>

<p style="margin-top: 1em">NOTES <br>
Please note that options are ORDER DEPENDENT. Sequences are
printed whenever a SEQUENCE SELECTION option occurs on the
command line. OUTPUT OPTIONS are not reset when a sequence
<br>
is printed.</p>

<p style="margin-top: 1em">SEQUENCES are numbered starting
at ZERO, not one!</p>

<p style="margin-top: 1em">EXAMPLES <br>
1. Print the first 10 bases of the fourth sequence in file
&rsquo;genes&rsquo;: <br>
leaff -f genes -e 0 10 -s 3</p>

<p style="margin-top: 1em">2. Print the first 10 bases of
the fourth and fifth sequences: <br>
leaff -f genes -e 0 10 -s 3 -s 4</p>

<p style="margin-top: 1em">3. Print the fourth and fifth
sequences reverse complemented, and the sixth <br>
sequence forward. The second set of -R -C toggle off
reverse-complement: <br>
leaff -f genes -R -C -s 3 -s 4 -R -C -s 5</p>

<p style="margin-top: 1em">4. Convert file
&rsquo;genes&rsquo; to a seqStore
&rsquo;genes.seqStore&rsquo;. <br>
leaff -f genes --seqstore genes.seqStore</p>

<p style="margin-top: 1em">SEE ALSO <br>
README.leaff <br>

http://kmer.sourceforge.net/wiki/index.php/LEAFF_User%27s_Guide
<br>

http://kmer.sourceforge.net/wiki/index.php/LEAFF_Programming_Example</p>

<p style="margin-top: 1em">January 2016 LEAFF(1)</p>
<hr>
</body>
</html>
