<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>LOCUSPOCUS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">LOCUSPOCUS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">LOCUSPOCUS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
locuspocus - calculate locus coordinates for the given gene annotation
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>locuspocus</b> [ <i>options</i>] <i>gff3file1 </i>[<i>gff3file2 gff3file3
  </i>...]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Basic options:
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b>|--debug</dt>
  <dd class="It-tag">print detailed debugging messages to terminal (standard
      error)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>|--help</dt>
  <dd class="It-tag">print this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>|--version</dt>
  <dd class="It-tag">print version number and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">iLocus parsing:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b>|--delta: INT</dt>
  <dd class="It-tag">when parsing interval loci, use the following delta to
      extend gene loci and include potential regulatory regions; default is
    500</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b>|--skipends</dt>
  <dd class="It-tag">when enumerating interval loci, exclude unannotated (and
      presumably incomplete) iLoci at either end of the sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>|--endsonly</dt>
  <dd class="It-tag">report only incomplete iLocus fragments at the unannotated
      ends of sequences (complement of <b>--skipends</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-y</b>|--skipiiloci</dt>
  <dd class="It-tag">do not report intergenic iLoci</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Refinement options:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b>|--refine</dt>
  <dd class="It-tag">by default genes are grouped in the same iLocus if they
      have any overlap; 'refine' mode allows for a more nuanced handling of
      overlapping genes</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b>|--cds</dt>
  <dd class="It-tag">use CDS rather than UTRs for determining gene overlap;
      implies 'refine' mode</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>|--minoverlap: INT</dt>
  <dd class="It-tag">the minimum number of nucleotides two genes must overlap to
      be grouped in the same iLocus; default is 1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Output options:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b>|--namefmt: STR</dt>
  <dd class="It-tag">provide a printf-style format string to override the
      default ID format for newly created loci; default is 'locus%lu' (locus1,
      locus2, etc) for loci and 'iLocus%lu' (iLocus1, iLocus2, etc) for interval
      loci; note the format string should include a single %lu specifier to be
      filled in with a long unsigned integer value</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b>|--ilens: FILE</dt>
  <dd class="It-tag">create a file with the lengths of each intergenic
    iLocus</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b>|--genemap: FILE</dt>
  <dd class="It-tag">print a mapping from each gene annotation to its
      corresponding locus to the given file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b>|--outfile: FILE</dt>
  <dd class="It-tag">name of file to which results will be written; default is
      terminal (standard output)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b>|--retainids</dt>
  <dd class="It-tag">retain original feature IDs from input files; conflicts
      will arise if input contains duplicated ID values</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>|--transmap: FILE</dt>
  <dd class="It-tag">print a mapping from each transcript annotation to its
      corresponding locus to the given file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-V</b>|--verbose</dt>
  <dd class="It-tag">include all locus subfeatures (genes, RNAs, etc) in the
      GFF3 output; default includes only locus features</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Input options:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>|--filter: TYPE</dt>
  <dd class="It-tag">comma-separated list of feature types to use in
      constructing loci/iLoci; default is 'gene'</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b>|--parent: CT:PT</dt>
  <dd class="It-tag">if a feature of type $CT exists without a parent, create a
      parent for this feature with type $PT; for example, mRNA:gene will create
      a gene feature as a parent for any top-level mRNA feature; this option can
      be specified multiple times</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b>|--pseudo</dt>
  <dd class="It-tag">correct erroneously labeled pseudogenes</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2016</td>
    <td class="foot-os">locuspocus 0.15.1</td>
  </tr>
</table>
</body>
</html>
