<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:22:25 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>LOCUSPOCUS(1) User Commands LOCUSPOCUS(1)</p>

<p style="margin-top: 1em">NAME <br>
locuspocus - calculate locus coordinates for the given gene
annotation</p>

<p style="margin-top: 1em">SYNOPSIS <br>
locuspocus [options] gff3file1 [gff3file2 gff3file3 ...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Basic options:</p>

<p style="margin-top: 1em">-d|--debug <br>
print detailed debugging messages to terminal (standard
error)</p>

<p style="margin-top: 1em">-h|--help <br>
print this help message and exit</p>

<p style="margin-top: 1em">-v|--version <br>
print version number and exit</p>

<p style="margin-top: 1em">iLocus parsing:</p>

<p style="margin-top: 1em">-l|--delta: INT <br>
when parsing interval loci, use the following delta to
extend gene loci and include potential regulatory regions;
default is 500</p>

<p style="margin-top: 1em">-s|--skipends <br>
when enumerating interval loci, exclude unannotated (and
presumably incomplete) iLoci at either end of the
sequence</p>

<p style="margin-top: 1em">-e|--endsonly <br>
report only incomplete iLocus fragments at the unannotated
ends of sequences (complement of --skipends)</p>

<p style="margin-top: 1em">-y|--skipiiloci <br>
do not report intergenic iLoci</p>

<p style="margin-top: 1em">Refinement options:</p>

<p style="margin-top: 1em">-r|--refine <br>
by default genes are grouped in the same iLocus if they have
any overlap; &rsquo;refine&rsquo; mode allows for a more
nuanced handling of overlapping genes</p>

<p style="margin-top: 1em">-c|--cds <br>
use CDS rather than UTRs for determining gene overlap;
implies &rsquo;refine&rsquo; mode</p>

<p style="margin-top: 1em">-m|--minoverlap: INT <br>
the minimum number of nucleotides two genes must overlap to
be grouped in the same iLocus; default is 1</p>

<p style="margin-top: 1em">Output options:</p>

<p style="margin-top: 1em">-n|--namefmt: STR <br>
provide a printf-style format string to override the default
ID format for newly created loci; default is
&rsquo;locus%lu&rsquo; (locus1, locus2, etc) for loci and
&rsquo;iLocus%lu&rsquo; (iLo&acirc; <br>
cus1, iLocus2, etc) for interval loci; note the format
string should include a single %lu specifier to be filled in
with a long unsigned integer value</p>

<p style="margin-top: 1em">-i|--ilens: FILE <br>
create a file with the lengths of each intergenic iLocus</p>

<p style="margin-top: 1em">-g|--genemap: FILE <br>
print a mapping from each gene annotation to its
corresponding locus to the given file</p>

<p style="margin-top: 1em">-o|--outfile: FILE <br>
name of file to which results will be written; default is
terminal (standard output)</p>

<p style="margin-top: 1em">-T|--retainids <br>
retain original feature IDs from input files; conflicts will
arise if input contains duplicated ID values</p>

<p style="margin-top: 1em">-t|--transmap: FILE <br>
print a mapping from each transcript annotation to its
corresponding locus to the given file</p>

<p style="margin-top: 1em">-V|--verbose <br>
include all locus subfeatures (genes, RNAs, etc) in the GFF3
output; default includes only locus features</p>

<p style="margin-top: 1em">Input options:</p>

<p style="margin-top: 1em">-f|--filter: TYPE <br>
comma-separated list of feature types to use in constructing
loci/iLoci; default is &rsquo;gene&rsquo;</p>

<p style="margin-top: 1em">-p|--parent: CT:PT <br>
if a feature of type $CT exists without a parent, create a
parent for this feature with type $PT; for example,
mRNA:gene will create a gene feature as a parent for any
<br>
top-level mRNA feature; this option can be specified
multiple times</p>

<p style="margin-top: 1em">-u|--pseudo <br>
correct erroneously labeled pseudogenes</p>

<p style="margin-top: 1em">locuspocus 0.15.1 January 2016
LOCUSPOCUS(1)</p>
<hr>
</body>
</html>
