<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MACS2_FILTERDUP(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MACS2_FILTERDUP(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">MACS2_FILTERDUP(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
macs2_filterdup - Model-based Analysis for ChIP-Sequencing
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
usage: macs2 filterdup [-h] <b>-i</b> IFILE [IFILE ...]
<dl class="Bl-tag">
  <dt class="It-tag">[-f
    {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}]</dt>
  <dd class="It-tag">[-g GSIZE] [-s TSIZE] [-p PVALUE] [--keep-dup
      KEEPDUPLICATES] [--verbose VERBOSE] [--outdir OUTDIR] [-o OUTPUTFILE]
    [-d]</dd>
</dl>
<h2 class="Ss" title="Ss" id="optional_arguments:"><a class="selflink" href="#optional_arguments:">optional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> IFILE [IFILE ...], <b>--ifile</b> IFILE [IFILE
    ...]</dt>
  <dd class="It-tag">ChIP-seq alignment file. If multiple files are given as '-t
      A B C', then they will all be read and combined. Note that pair-end data
      is not supposed to work with this command. REQUIRED.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>
    {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}, <b>--format</b>
    {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}</dt>
  <dd class="It-tag">Format of tag file, &quot;AUTO&quot;, &quot;BED&quot; or
      &quot;ELAND&quot; or &quot;ELANDMULTI&quot; or &quot;ELANDEXPORT&quot; or
      &quot;SAM&quot; or &quot;BAM&quot; or &quot;BOWTIE&quot;. The default AUTO
      option will let 'macs2 filterdup' decide which format the file is. Please
      check the definition in README file if you choose
      ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT:
    &quot;AUTO&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b> GSIZE, <b>--gsize</b> GSIZE</dt>
  <dd class="It-tag">Effective genome size. It can be 1.0e+9 or 1000000000, or
      shortcuts:'hs' for human (2.7e9), 'mm' for mouse (1.87e9), 'ce' for C.
      elegans (9e7) and 'dm' for fruitfly (1.2e8), DEFAULT:hs</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> TSIZE, <b>--tsize</b> TSIZE</dt>
  <dd class="It-tag">Tag size. This will override the auto detected tag size.
      DEFAULT: Not set</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> PVALUE, <b>--pvalue</b> PVALUE</dt>
  <dd class="It-tag">Pvalue cutoff for binomial distribution test.
    DEFAULT:1e-5</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--keep-dup</b> KEEPDUPLICATES</dt>
  <dd class="It-tag">It controls the 'macs2 filterdup' behavior towards
      duplicate tags/pairs at the exact same location <b>--</b> the same
      coordination and the same strand. The 'auto' option makes 'macs2
      filterdup' calculate the maximum tags at the exact same location based on
      binomal distribution using given <b>-p</b> as pvalue cutoff; and the 'all'
      option keeps every tags (useful if you only want to convert formats). If
      an integer is given, at most this number of tags will be kept at the same
      location. Note, MACS2 callpeak function uses KEEPDUPLICATES=1 as default.
      Note, if you've used samtools or picard to flag reads as 'PCR/Optical
      duplicate' in bit 1024, MACS2 will still read them although the reads may
      be decided by MACS2 as duplicate later. Default: auto</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--verbose</b> VERBOSE</dt>
  <dd class="It-tag">Set verbose level. 0: only show critical message, 1: show
      additional warning message, 2: show process information, 3: show debug
      messages. If you want to know where are the duplicate reads, use 3.
      DEFAULT:2</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outdir</b> OUTDIR</dt>
  <dd class="It-tag">If specified all output files will be written to that
      directory. Default: the current working directory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> OUTPUTFILE, <b>--ofile</b> OUTPUTFILE</dt>
  <dd class="It-tag">Output BED file name. If not specified, will write to
      standard output. DEFAULT: stdout</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b>, <b>--dry-run</b></dt>
  <dd class="It-tag">When set, filterdup will only output numbers instead of
      writing output files, including maximum allowable duplicates, total number
      of reads before filtering, total number of reads after filtering, and
      redundant rate. Default: not set</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">May 2016</td>
    <td class="foot-os">macs2_filterdup 2.1.1.20160309</td>
  </tr>
</table>
</body>
</html>
