<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:28 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MACSYFINDER(1) User Commands MACSYFINDER(1)</p>

<p style="margin-top: 1em">NAME <br>
macsyfinder - detection of macromolecular systems in protein
datasets</p>

<p style="margin-top: 1em">SYNOPSIS <br>
macsyfinder [-h] [--sequence-db SEQUENCE_DB] [--db-type
{unordered_replicon,ordered_replicon,gembase,unordered}]
[--replicon-topology {linear,circular}] [--topology-file
TOPOL&acirc; <br>
OGY_FILE] [--idx] [--inter-gene-max-space
INTER_GENE_MAX_SPACE INTER_GENE_MAX_SPACE]
[--min-mandatory-genes-required MIN_MANDATORY_GENES_REQUIRED
MIN_MANDATORY_GENES_REQUIRED] <br>
[--min-genes-required MIN_GENES_REQUIRED MIN_GENES_REQUIRED]
[--max-nb-genes MAX_NB_GENES MAX_NB_GENES] [--multi-loci
MULTI_LOCI] [--hmmer HMMER_EXE] [--index-db INDEX_DB_EXE]
<br>
[--e-value-search E_VALUE_RES] [--i-evalue-select
I_EVALUE_SEL] [--coverage-profile COVERAGE_PROFILE] [-d
DEF_DIR] [-o OUT_DIR] [-r RES_SEARCH_DIR]
[--res-search-suffix <br>
RES_SEARCH_SUFFIX] [--res-extract-suffix RES_EXTRACT_SUFFIX]
[-p PROFILE_DIR] [--profile-suffix PROFILE_SUFFIX] [-w
WORKER_NB] [-v] [--log LOG_FILE] [--config CFG_FILE]
[--previ&acirc; <br>
ous-run PREVIOUS_RUN] systems [systems ...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
MacSyFinder is a program to model and detect macromolecular
systems, genetic pathways... in protein datasets. In
prokaryotes, these systems have often evolutionarily
conserved <br>
properties: they are made of conserved components, and are
encoded in compact loci (conserved genetic architecture).
The user models these systems with MacSyFinder to reflect
<br>
these conserved features, and to allow their efficient
detection</p>

<p style="margin-top: 1em">OPTIONS <br>
positional arguments: <br>
systems <br>
The systems to detect. This is an obligatory option with no
keyword associated to it. To detect all the protein
secretion systems and related appendages: set to
&quot;all&quot; <br>
(case insensitive). Otherwise, a single or multiple systems
can be specified. For example: &quot;T2SS T4P&quot;.</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">Input dataset options: <br>
--sequence-db SEQUENCE_DB <br>
Path to the sequence dataset in fasta format.</p>

<p style="margin-top: 1em">--db-type
{unordered_replicon,ordered_replicon,gembase,unordered} <br>
The type of dataset to deal with.
&quot;unordered_replicon&quot; corresponds to a
non-assembled genome, &quot;unordered&quot; to a metagenomic
dataset, &quot;ordered_replicon&quot; to an assembled <br>
genome, and &quot;gembase&quot; to a set of replicons where
sequence identifiers follow this convention:
&quot;&gt;RepliconName SequenceID&quot;.</p>

<p style="margin-top: 1em">--replicon-topology
{linear,circular} <br>
The topology of the replicons (this option is meaningful
only if the db_type is &rsquo;ordered_replicon&rsquo; or
&rsquo;gembase&rsquo;.</p>

<p style="margin-top: 1em">--topology-file TOPOLOGY_FILE
<br>
Topology file path. The topology file allows one to specify
a topology (linear or circular) for each replicon (this
option is meaningful only if the db_type is <br>
&rsquo;ordered_replicon&rsquo; or &rsquo;gembase&rsquo;. A
topology file is a tabular file with two columns: the 1st is
the replicon name, and the 2nd the corresponding topology:
&quot;RepliconA lin&acirc; <br>
ear&quot;</p>

<p style="margin-top: 1em">--idx Forces to build the
indexes for the sequence dataset even if they were
presviously computed and present at the dataset location
(default = False)</p>

<p style="margin-top: 1em">Systems detection options: <br>
--inter-gene-max-space INTER_GENE_MAX_SPACE
INTER_GENE_MAX_SPACE <br>
Co-localization criterion: maximum number of components
non-matched by a profile allowed between two matched
components for them to be considered contiguous. Option only
<br>
meaningful for &rsquo;ordered&rsquo; datasets. The first
value must match to a system, the second to a number of
components. This option can be repeated several times: <br>
&quot;--inter-gene-max-space T2SS 12 --inter-gene-max-space
Flagellum 20&quot;</p>

<p style="margin-top: 1em">--min-mandatory-genes-required
MIN_MANDATORY_GENES_REQUIRED MIN_MANDATORY_GENES_REQUIRED
<br>
The minimal number of mandatory genes required for system
assessment. The first value must correspond to a system
name, the second value to an integer. This option can be
<br>
repeated several times: &quot;--minmandatory-genes-required
T2SS 15 --min-mandatorygenes-required Flagellum 10&quot;</p>

<p style="margin-top: 1em">--min-genes-required
MIN_GENES_REQUIRED MIN_GENES_REQUIRED <br>
The minimal number of genes required for system assessment
(includes both &rsquo;mandatory&rsquo; and
&rsquo;accessory&rsquo; components). The first value must
correspond to a system name, the <br>
second value to an integer. This option can be repeated
several times: &quot;--min-genesrequired T2SS 15
--min-genes-required Flagellum 10&quot;</p>

<p style="margin-top: 1em">--max-nb-genes MAX_NB_GENES
MAX_NB_GENES <br>
The maximal number of genes required for system assessment.
The first value must correspond to a system name, the second
value to an integer. This option can be repeated <br>
several times: &quot;--max-nb-genes T2SS 5 --max-nb-genes
Flagellum 10</p>

<p style="margin-top: 1em">--multi-loci MULTI_LOCI <br>
Allow the storage of multi-loci systems for the specified
systems. The systems are specified as a comma separated list
(--multi-loci sys1,sys2) default is False</p>

<p style="margin-top: 1em">Options for Hmmer execution and
hits filtering: <br>
--hmmer HMMER_EXE <br>
Path to the Hmmer program.</p>

<p style="margin-top: 1em">--index-db INDEX_DB_EXE <br>
The indexer to be used for Hmmer. The value can be either
&rsquo;makeblastdb&rsquo; or &rsquo;formatdb&rsquo; or the
path to one of these binary (default = makeblastb)</p>

<p style="margin-top: 1em">--e-value-search E_VALUE_RES
<br>
Maximal e-value for hits to be reported during Hmmer search.
(default = 1)</p>

<p style="margin-top: 1em">--i-evalue-select I_EVALUE_SEL
<br>
Maximal independent e-value for Hmmer hits to be selected
for system detection. (default = 0.001)</p>

<p style="margin-top: 1em">--coverage-profile
COVERAGE_PROFILE <br>
Minimal profile coverage required in the hit alignment to
allow the hit selection for system detection. (default =
0.5)</p>

<p style="margin-top: 1em">Path options: <br>
-d DEF_DIR, --def DEF_DIR <br>
Path to the systems definition files.</p>

<p style="margin-top: 1em">-o OUT_DIR, --out-dir OUT_DIR
<br>
Path to the directory where to store results. if outdir is
specified res-search-dir will be ignored.</p>

<p style="margin-top: 1em">-r RES_SEARCH_DIR,
--res-search-dir RES_SEARCH_DIR <br>
Path to the directory where to store MacSyFinder search
results directories (default current working directory).</p>

<p style="margin-top: 1em">--res-search-suffix
RES_SEARCH_SUFFIX <br>
The suffix to give to Hmmer raw output files.</p>

<p style="margin-top: 1em">--res-extract-suffix
RES_EXTRACT_SUFFIX <br>
The suffix to give to filtered hits output files.</p>

<p style="margin-top: 1em">-p PROFILE_DIR, --profile-dir
PROFILE_DIR <br>
Path to the profiles directory.</p>

<p style="margin-top: 1em">--profile-suffix PROFILE_SUFFIX
<br>
The suffix of profile files. For each &rsquo;Gene&rsquo;
element, the corresponding profile is searched in the
&rsquo;profile_dir&rsquo;, in a file which name is based on
the Gene name + the pro&acirc; <br>
file suffix. For instance, if the Gene is named
&rsquo;gspG&rsquo; and the suffix is &rsquo;.hmm3&rsquo;,
then the profile should be placed at the specified location
and be named &rsquo;gspG.hmm3&rsquo;</p>

<p style="margin-top: 1em">General options: <br>
-w WORKER_NB, --worker WORKER_NB <br>
Number of workers to be used by MacSyFinder. In the case the
user wants to run MacSyFinder in a multithread mode. (0 mean
all cores will be used, default 1)</p>

<p style="margin-top: 1em">-v, --verbosity <br>
Increases the verbosity level. There are 4 levels: Error
messages (default), Warning (-v), Info (-vv) and
Debug.(-vvv)</p>

<p style="margin-top: 1em">--log LOG_FILE <br>
Path to the directory where to store the
&rsquo;macsyfinder.log&rsquo; log file.</p>

<p style="margin-top: 1em">--config CFG_FILE <br>
Path to a putative MacSyFinder configuration file to be
used.</p>

<p style="margin-top: 1em">--previous-run PREVIOUS_RUN <br>
Path to a previous MacSyFinder run directory. It allows one
to skip the Hmmer search step on same dataset, as it uses
previous run results and thus parameters regarding <br>
Hmmer detection. The configuration file from this previous
run will be used. (conflict with options --config,
--sequence-db, --profile-suffix, --resextract-suffix, <br>
--e-value-res, --db-type, --hmmer)</p>

<p style="margin-top: 1em">For more details, visit the
MacSyFinder website and see the MacSyFinder
documentation.</p>

<p style="margin-top: 1em">macsyfinder 1.0.2 January 2015
MACSYFINDER(1)</p>
<hr>
</body>
</html>
