<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:27 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MACS2_PREDICTD(1) User Commands MACS2_PREDICTD(1)</p>

<p style="margin-top: 1em">NAME <br>
macs2_predictd - Model-based Analysis for
ChIP-Sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: macs2 predictd [-h] -i IFILE [IFILE ...]</p>

<p style="margin-top: 1em">[-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}]
<br>
[-g GSIZE] [-s TSIZE] [--bw BW] [-m MFOLD MFOLD] [--outdir
OUTDIR] [--rfile RFILE] [--verbose VERBOSE]</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">-i IFILE [IFILE ...], --ifile
IFILE [IFILE ...] <br>
ChIP-seq alignment file. If multiple files are given as
&rsquo;-t A B C&rsquo;, then they will all be read and
combined. Note that pair-end data is not supposed to work
with this com&acirc; <br>
mand. REQUIRED.</p>

<p style="margin-top: 1em">-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE},
--format
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}
<br>
Format of tag file, &quot;AUTO&quot;, &quot;BED&quot; or
&quot;ELAND&quot; or &quot;ELANDMULTI&quot; or
&quot;ELANDEXPORT&quot; or &quot;SAM&quot; or
&quot;BAM&quot; or &quot;BOWTIE&quot; or &quot;BAMPE&quot;
or &quot;BEDPE&quot;. The default AUTO option will let MACS
<br>
decide which format the file is. Please check the definition
in README file if you choose
ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT:
&quot;AUTO&quot;</p>

<p style="margin-top: 1em">-g GSIZE, --gsize GSIZE <br>
Effective genome size. It can be 1.0e+9 or 1000000000, or
shortcuts:&rsquo;hs&rsquo; for human (2.7e9),
&rsquo;mm&rsquo; for mouse (1.87e9), &rsquo;ce&rsquo; for C.
elegans (9e7) and &rsquo;dm&rsquo; for fruitfly <br>
(1.2e8), Default:hs</p>

<p style="margin-top: 1em">-s TSIZE, --tsize TSIZE <br>
Tag size. This will override the auto detected tag size.
DEFAULT: Not set</p>

<p style="margin-top: 1em">--bw BW <br>
Band width for picking regions to compute fragment size.
This value is only used while building the shifting model.
DEFAULT: 300</p>

<p style="margin-top: 1em">-m MFOLD MFOLD, --mfold MFOLD
MFOLD <br>
Select the regions within MFOLD range of highconfidence
enrichment ratio against background to build model.
Fold-enrichment in regions must be lower than upper limit,
and <br>
higher than the lower limit. Use as &quot;-m 10 30&quot;.
DEFAULT:5 50</p>

<p style="margin-top: 1em">--outdir OUTDIR <br>
If specified all output files will be written to that
directory. Default: the current working directory</p>

<p style="margin-top: 1em">--rfile RFILE <br>
PREFIX of filename of R script for drawing X-correlation
figure. DEFAULT:&rsquo;predictd&rsquo; and R file will be
predicted_model.R</p>

<p style="margin-top: 1em">--verbose VERBOSE <br>
Set verbose level of runtime message. 0: only show critical
message, 1: show additional warning message, 2: show process
information, 3: show debug messages. DEFAULT:2</p>

<p style="margin-top: 1em">macs2_predictd 2.1.1.20160309
May 2016 MACS2_PREDICTD(1)</p>
<hr>
</body>
</html>
