<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:27 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MACS2_FILTERDUP(1) User Commands MACS2_FILTERDUP(1)</p>

<p style="margin-top: 1em">NAME <br>
macs2_filterdup - Model-based Analysis for
ChIP-Sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: macs2 filterdup [-h] -i IFILE [IFILE ...]</p>

<p style="margin-top: 1em">[-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}] <br>
[-g GSIZE] [-s TSIZE] [-p PVALUE] [--keep-dup
KEEPDUPLICATES] [--verbose VERBOSE] [--outdir OUTDIR] [-o
OUTPUTFILE] [-d]</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">-i IFILE [IFILE ...], --ifile
IFILE [IFILE ...] <br>
ChIP-seq alignment file. If multiple files are given as
&rsquo;-t A B C&rsquo;, then they will all be read and
combined. Note that pair-end data is not supposed to work
with this com&acirc; <br>
mand. REQUIRED.</p>

<p style="margin-top: 1em">-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE},
--format
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE} <br>
Format of tag file, &quot;AUTO&quot;, &quot;BED&quot; or
&quot;ELAND&quot; or &quot;ELANDMULTI&quot; or
&quot;ELANDEXPORT&quot; or &quot;SAM&quot; or
&quot;BAM&quot; or &quot;BOWTIE&quot;. The default AUTO
option will let &rsquo;macs2 filterdup&rsquo; decide <br>
which format the file is. Please check the definition in
README file if you choose
ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT:
&quot;AUTO&quot;</p>

<p style="margin-top: 1em">-g GSIZE, --gsize GSIZE <br>
Effective genome size. It can be 1.0e+9 or 1000000000, or
shortcuts:&rsquo;hs&rsquo; for human (2.7e9),
&rsquo;mm&rsquo; for mouse (1.87e9), &rsquo;ce&rsquo; for C.
elegans (9e7) and &rsquo;dm&rsquo; for fruitfly <br>
(1.2e8), DEFAULT:hs</p>

<p style="margin-top: 1em">-s TSIZE, --tsize TSIZE <br>
Tag size. This will override the auto detected tag size.
DEFAULT: Not set</p>

<p style="margin-top: 1em">-p PVALUE, --pvalue PVALUE <br>
Pvalue cutoff for binomial distribution test.
DEFAULT:1e-5</p>

<p style="margin-top: 1em">--keep-dup KEEPDUPLICATES <br>
It controls the &rsquo;macs2 filterdup&rsquo; behavior
towards duplicate tags/pairs at the exact same location --
the same coordination and the same strand. The
&rsquo;auto&rsquo; option makes <br>
&rsquo;macs2 filterdup&rsquo; calculate the maximum tags at
the exact same location based on binomal distribution using
given -p as pvalue cutoff; and the &rsquo;all&rsquo; option
keeps every <br>
tags (useful if you only want to convert formats). If an
integer is given, at most this number of tags will be kept
at the same location. Note, MACS2 callpeak function <br>
uses KEEPDUPLICATES=1 as default. Note, if you&rsquo;ve used
samtools or picard to flag reads as &rsquo;PCR/Optical
duplicate&rsquo; in bit 1024, MACS2 will still read them
although the <br>
reads may be decided by MACS2 as duplicate later. Default:
auto</p>

<p style="margin-top: 1em">--verbose VERBOSE <br>
Set verbose level. 0: only show critical message, 1: show
additional warning message, 2: show process information, 3:
show debug messages. If you want to know where are <br>
the duplicate reads, use 3. DEFAULT:2</p>

<p style="margin-top: 1em">--outdir OUTDIR <br>
If specified all output files will be written to that
directory. Default: the current working directory</p>

<p style="margin-top: 1em">-o OUTPUTFILE, --ofile
OUTPUTFILE <br>
Output BED file name. If not specified, will write to
standard output. DEFAULT: stdout</p>

<p style="margin-top: 1em">-d, --dry-run <br>
When set, filterdup will only output numbers instead of
writing output files, including maximum allowable
duplicates, total number of reads before filtering, total
number <br>
of reads after filtering, and redundant rate. Default: not
set</p>

<p style="margin-top: 1em">macs2_filterdup 2.1.1.20160309
May 2016 MACS2_FILTERDUP(1)</p>
<hr>
</body>
</html>
