<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MACS2(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MACS2(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">MACS2(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
macs2 - macs2 - Model-based Analysis for ChIP-Sequencing
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
usage: macs2 [-h] [--version]
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">{callpeak,bdgpeakcall,bdgbroadcall,bdgcmp,bdgopt,cmbreps,bdgdiff,filterdup,predictd,pileup,randsample,refinepeak}
      ...</dd>
</dl>
<div class="Pp"></div>
macs2 <b>--</b> Model-based Analysis for ChIP-Sequencing
<h2 class="Ss" title="Ss" id="positional_arguments:"><a class="selflink" href="#positional_arguments:">positional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">{callpeak,bdgpeakcall,bdgbroadcall,bdgcmp,bdgopt,cmbreps,bdgdiff,filterdup,predictd,pileup,randsample,refinepeak}</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">callpeak</dt>
  <dd class="It-tag">Main MACS2 Function: Call peaks from alignment
    results.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bdgpeakcall</dt>
  <dd class="It-tag">Call peaks from bedGraph output. Note: All regions on the
      same chromosome in the bedGraph file should be continuous so only bedGraph
      files from MACS2 are accpetable.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bdgbroadcall</dt>
  <dd class="It-tag">Call broad peaks from bedGraph output. Note: All regions on
      the same chromosome in the bedGraph file should be continuous so only
      bedGraph files from MACS2 are accpetable.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bdgcmp</dt>
  <dd class="It-tag">Deduct noise by comparing two signal tracks in bedGraph.
      Note: All regions on the same chromosome in the bedGraph file should be
      continuous so only bedGraph files from MACS2 are accpetable.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bdgopt</dt>
  <dd class="It-tag">Operations on score column of bedGraph file. Note: All
      regions on the same chromosome in the bedGraph file should be continuous
      so only bedGraph files from MACS2 are accpetable.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">cmbreps</dt>
  <dd class="It-tag">Combine BEDGraphs of scores from replicates. Note: All
      regions on the same chromosome in the bedGraph file should be continuous
      so only bedGraph files from MACS2 are accpetable.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bdgdiff</dt>
  <dd class="It-tag">Differential peak detection based on paired four bedgraph
      files. Note: All regions on the same chromosome in the bedGraph file
      should be continuous so only bedGraph files from MACS2 are
    accpetable.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">filterdup</dt>
  <dd class="It-tag">Remove duplicate reads at the same position, then convert
      acceptable format to BED format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">predictd</dt>
  <dd class="It-tag">Predict d or fragment size from alignment results. *Will
      NOT filter duplicates*</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">pileup</dt>
  <dd class="It-tag">Pileup aligned reads with a given extension size (fragment
      size or d in MACS language). Note there will be no step for duplicate
      reads filtering or sequencing depth scaling, so you may need to do certain
      pre/postprocessing.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">randsample</dt>
  <dd class="It-tag">Randomly sample number/percentage of total reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">refinepeak</dt>
  <dd class="It-tag">(Experimental) Take raw reads alignment, refine peak
      summits and give scores measuring balance of waston/crick tags. Inspired
      by SPP.</dd>
</dl>
<h2 class="Ss" title="Ss" id="optional_arguments:"><a class="selflink" href="#optional_arguments:">optional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">show program's version number and exit</dd>
</dl>
<div class="Pp"></div>
For command line options of each command, type: macs2 COMMAND <b>-h</b></div>
<table class="foot">
  <tr>
    <td class="foot-date">May 2016</td>
    <td class="foot-os">macs2 2.1.1.20160309</td>
  </tr>
</table>
</body>
</html>
