<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:27 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MACS2_PILEUP(1) User Commands MACS2_PILEUP(1)</p>

<p style="margin-top: 1em">NAME <br>
macs2_pileup - Model-based Analysis for ChIP-Sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: macs2 pileup [-h] -i IFILE [IFILE ...] -o OUTPUTFILE
[--outdir OUTDIR]</p>

<p style="margin-top: 1em">[-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}] <br>
[-B] [--extsize EXTSIZE] [--verbose VERBOSE]</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">-i IFILE [IFILE ...], --ifile
IFILE [IFILE ...] <br>
ChIP-seq alignment file. If multiple files are given as
&rsquo;-t A B C&rsquo;, then they will all be read and
combined. Note that pair-end data is not supposed to work
with this com&acirc; <br>
mand. REQUIRED.</p>

<p style="margin-top: 1em">-o OUTPUTFILE, --ofile
OUTPUTFILE <br>
Output bedGraph file name. If not specified, will write to
standard output. REQUIRED.</p>

<p style="margin-top: 1em">--outdir OUTDIR <br>
If specified all output files will be written to that
directory. Default: the current working directory</p>

<p style="margin-top: 1em">-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE},
--format
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE} <br>
Format of tag file, &quot;AUTO&quot;, &quot;BED&quot; or
&quot;ELAND&quot; or &quot;ELANDMULTI&quot; or
&quot;ELANDEXPORT&quot; or &quot;SAM&quot; or
&quot;BAM&quot; or &quot;BOWTIE&quot;. The default AUTO
option will let &rsquo;macs2 pileup&rsquo; decide which <br>
format the file is. Please check the definition in README
file if you choose
ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT:
&quot;AUTO&quot;</p>

<p style="margin-top: 1em">-B, --both-direction <br>
By default, any read will be extended towards downstream
direction by extension size. So it&rsquo;s [0,size-1]
(1-based index system) for plus strand read and [-size+1,0]
for <br>
minus strand read where position 0 is 5&rsquo; end of the
aligned read. Default behavior can simulate MACS2 way of
piling up ChIP sample reads where extension size is set as
<br>
fragment size/d. If this option is set as on, aligned reads
will be extended in both upstream and downstream directions
by extension size. It means [-size,size] where 0 is <br>
the 5&rsquo; end of a aligned read. It can partially
simulate MACS2 way of piling up control reads. However MACS2
local bias is calculated by maximizing the expected pileup
over <br>
a ChIP fragment size/d estimated from 10kb, 1kb, d and whole
genome background. DEFAULT: False</p>

<p style="margin-top: 1em">--extsize EXTSIZE <br>
The extension size in bps. Each alignment read will become a
EXTSIZE of fragment, then be piled up. Check description for
-B for detail. It&rsquo;s twice the &lsquo;shiftsize&lsquo;
in old <br>
MACSv1 language. DEFAULT: 200</p>

<p style="margin-top: 1em">--verbose VERBOSE <br>
Set verbose level. 0: only show critical message, 1: show
additional warning message, 2: show process information, 3:
show debug messages. If you want to know where are <br>
the duplicate reads, use 3. DEFAULT:2</p>

<p style="margin-top: 1em">macs2_pileup 2.1.1.20160309 May
2016 MACS2_PILEUP(1)</p>
<hr>
</body>
</html>
