<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MACSYFINDER(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MACSYFINDER(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">MACSYFINDER(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
macsyfinder - detection of macromolecular systems in protein datasets
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>macsyfinder</b> [-h] [--sequence-db SEQUENCE_DB] [--db-type
  {unordered_replicon,ordered_replicon,gembase,unordered}] [--replicon-topology
  {linear,circular}] [--topology-file TOPOLOGY_FILE] [--idx]
  [--inter-gene-max-space INTER_GENE_MAX_SPACE INTER_GENE_MAX_SPACE]
  [--min-mandatory-genes-required MIN_MANDATORY_GENES_REQUIRED
  MIN_MANDATORY_GENES_REQUIRED] [--min-genes-required MIN_GENES_REQUIRED
  MIN_GENES_REQUIRED] [--max-nb-genes MAX_NB_GENES MAX_NB_GENES] [--multi-loci
  MULTI_LOCI] [--hmmer HMMER_EXE] [--index-db INDEX_DB_EXE] [--e-value-search
  E_VALUE_RES] [--i-evalue-select I_EVALUE_SEL] [--coverage-profile
  COVERAGE_PROFILE] [-d DEF_DIR] [-o OUT_DIR] [-r RES_SEARCH_DIR]
  [--res-search-suffix RES_SEARCH_SUFFIX] [--res-extract-suffix
  RES_EXTRACT_SUFFIX] [-p PROFILE_DIR] [--profile-suffix PROFILE_SUFFIX] [-w
  WORKER_NB] [-v] [--log LOG_FILE] [--config CFG_FILE] [--previous-run
  PREVIOUS_RUN] systems [systems ...]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
MacSyFinder is a program to model and detect macromolecular systems, genetic
  pathways... in protein datasets. In prokaryotes, these systems have often
  evolutionarily conserved properties: they are made of conserved components,
  and are encoded in compact loci (conserved genetic architecture). The user
  models these systems with MacSyFinder to reflect these conserved features, and
  to allow their efficient detection
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="positional_arguments:"><a class="selflink" href="#positional_arguments:">positional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">systems</dt>
  <dd class="It-tag">The systems to detect. This is an obligatory option with no
      keyword associated to it. To detect all the protein secretion systems and
      related appendages: set to &quot;all&quot; (case insensitive). Otherwise,
      a single or multiple systems can be specified. For example: &quot;T2SS
      T4P&quot;.</dd>
</dl>
<h2 class="Ss" title="Ss" id="optional_arguments:"><a class="selflink" href="#optional_arguments:">optional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_dataset_options:"><a class="selflink" href="#Input_dataset_options:">Input
  dataset options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sequence-db</b> SEQUENCE_DB</dt>
  <dd class="It-tag">Path to the sequence dataset in fasta format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--db-type</b>
    {unordered_replicon,ordered_replicon,gembase,unordered}</dt>
  <dd class="It-tag">The type of dataset to deal with.
      &quot;unordered_replicon&quot; corresponds to a non-assembled genome,
      &quot;unordered&quot; to a metagenomic dataset,
      &quot;ordered_replicon&quot; to an assembled genome, and
      &quot;gembase&quot; to a set of replicons where sequence identifiers
      follow this convention: &quot;&gt;RepliconName SequenceID&quot;.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--replicon-topology</b> {linear,circular}</dt>
  <dd class="It-tag">The topology of the replicons (this option is meaningful
      only if the db_type is 'ordered_replicon' or 'gembase'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--topology-file</b> TOPOLOGY_FILE</dt>
  <dd class="It-tag">Topology file path. The topology file allows one to specify
      a topology (linear or circular) for each replicon (this option is
      meaningful only if the db_type is 'ordered_replicon' or 'gembase'. A
      topology file is a tabular file with two columns: the 1st is the replicon
      name, and the 2nd the corresponding topology: &quot;RepliconA
    linear&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--idx</b></dt>
  <dd class="It-tag">Forces to build the indexes for the sequence dataset even
      if they were presviously computed and present at the dataset location
      (default = False)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Systems_detection_options:"><a class="selflink" href="#Systems_detection_options:">Systems
  detection options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--inter-gene-max-space</b> INTER_GENE_MAX_SPACE
    INTER_GENE_MAX_SPACE</dt>
  <dd class="It-tag">Co-localization criterion: maximum number of components
      non-matched by a profile allowed between two matched components for them
      to be considered contiguous. Option only meaningful for 'ordered'
      datasets. The first value must match to a system, the second to a number
      of components. This option can be repeated several times:
      &quot;--inter-gene-max-space T2SS 12 <b>--inter-gene-max-space</b>
      Flagellum 20&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-mandatory-genes-required</b>
    MIN_MANDATORY_GENES_REQUIRED MIN_MANDATORY_GENES_REQUIRED</dt>
  <dd class="It-tag">The minimal number of mandatory genes required for system
      assessment. The first value must correspond to a system name, the second
      value to an integer. This option can be repeated several times:
      &quot;--minmandatory-genes-required T2SS 15
      <b>--min-mandatorygenes-required</b> Flagellum 10&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min-genes-required</b> MIN_GENES_REQUIRED
    MIN_GENES_REQUIRED</dt>
  <dd class="It-tag">The minimal number of genes required for system assessment
      (includes both 'mandatory' and 'accessory' components). The first value
      must correspond to a system name, the second value to an integer. This
      option can be repeated several times: &quot;--min-genesrequired T2SS 15
      <b>--min-genes-required</b> Flagellum 10&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max-nb-genes</b> MAX_NB_GENES MAX_NB_GENES</dt>
  <dd class="It-tag">The maximal number of genes required for system assessment.
      The first value must correspond to a system name, the second value to an
      integer. This option can be repeated several times: &quot;--max-nb-genes
      T2SS 5 <b>--max-nb-genes</b> Flagellum 10</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--multi-loci</b> MULTI_LOCI</dt>
  <dd class="It-tag">Allow the storage of multi-loci systems for the specified
      systems. The systems are specified as a comma separated list (
      <b>--multi-loci</b> sys1,sys2) default is False</dd>
</dl>
<h2 class="Ss" title="Ss" id="Options_for_Hmmer_execution_and_hits_filtering:"><a class="selflink" href="#Options_for_Hmmer_execution_and_hits_filtering:">Options
  for Hmmer execution and hits filtering:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--hmmer</b> HMMER_EXE</dt>
  <dd class="It-tag">Path to the Hmmer program.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--index-db</b> INDEX_DB_EXE</dt>
  <dd class="It-tag">The indexer to be used for Hmmer. The value can be either
      'makeblastdb' or 'formatdb' or the path to one of these binary (default =
      makeblastb)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--e-value-search</b> E_VALUE_RES</dt>
  <dd class="It-tag">Maximal e-value for hits to be reported during Hmmer
      search. (default = 1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--i-evalue-select</b> I_EVALUE_SEL</dt>
  <dd class="It-tag">Maximal independent e-value for Hmmer hits to be selected
      for system detection. (default = 0.001)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--coverage-profile</b> COVERAGE_PROFILE</dt>
  <dd class="It-tag">Minimal profile coverage required in the hit alignment to
      allow the hit selection for system detection. (default = 0.5)</dd>
</dl>
<h2 class="Ss" title="Ss" id="Path_options:"><a class="selflink" href="#Path_options:">Path
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> DEF_DIR, <b>--def</b> DEF_DIR</dt>
  <dd class="It-tag">Path to the systems definition files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> OUT_DIR, <b>--out-dir</b> OUT_DIR</dt>
  <dd class="It-tag">Path to the directory where to store results. if outdir is
      specified res-search-dir will be ignored.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> RES_SEARCH_DIR, <b>--res-search-dir</b>
    RES_SEARCH_DIR</dt>
  <dd class="It-tag">Path to the directory where to store MacSyFinder search
      results directories (default current working directory).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--res-search-suffix</b> RES_SEARCH_SUFFIX</dt>
  <dd class="It-tag">The suffix to give to Hmmer raw output files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--res-extract-suffix</b> RES_EXTRACT_SUFFIX</dt>
  <dd class="It-tag">The suffix to give to filtered hits output files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> PROFILE_DIR, <b>--profile-dir</b>
    PROFILE_DIR</dt>
  <dd class="It-tag">Path to the profiles directory.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--profile-suffix</b> PROFILE_SUFFIX</dt>
  <dd class="It-tag">The suffix of profile files. For each 'Gene' element, the
      corresponding profile is searched in the 'profile_dir', in a file which
      name is based on the Gene name + the profile suffix. For instance, if the
      Gene is named 'gspG' and the suffix is '.hmm3', then the profile should be
      placed at the specified location and be named 'gspG.hmm3'</dd>
</dl>
<h2 class="Ss" title="Ss" id="General_options:"><a class="selflink" href="#General_options:">General
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b> WORKER_NB, <b>--worker</b> WORKER_NB</dt>
  <dd class="It-tag">Number of workers to be used by MacSyFinder. In the case
      the user wants to run MacSyFinder in a multithread mode. (0 mean all cores
      will be used, default 1)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--verbosity</b></dt>
  <dd class="It-tag">Increases the verbosity level. There are 4 levels: Error
      messages (default), Warning ( <b>-v</b>), Info (<b>-vv</b>) and Debug.(
      <b>-vvv</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--log</b> LOG_FILE</dt>
  <dd class="It-tag">Path to the directory where to store the 'macsyfinder.log'
      log file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--config</b> CFG_FILE</dt>
  <dd class="It-tag">Path to a putative MacSyFinder configuration file to be
      used.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--previous-run</b> PREVIOUS_RUN</dt>
  <dd class="It-tag">Path to a previous MacSyFinder run directory. It allows one
      to skip the Hmmer search step on same dataset, as it uses previous run
      results and thus parameters regarding Hmmer detection. The configuration
      file from this previous run will be used. (conflict with options
      <b>--config</b>, <b>--sequence-db</b>, <b>--profile-suffix</b>,
      <b>--resextract-suffix</b>, <b>--e-value-res</b>, <b>--db-type</b>,
      <b>--hmmer</b>)</dd>
</dl>
<div class="Pp"></div>
For more details, visit the MacSyFinder website and see the MacSyFinder
  documentation.</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2015</td>
    <td class="foot-os">macsyfinder 1.0.2</td>
  </tr>
</table>
</body>
</html>
