<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:26 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MACS2(1) User Commands MACS2(1)</p>

<p style="margin-top: 1em">NAME <br>
macs2 - macs2 - Model-based Analysis for ChIP-Sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: macs2 [-h] [--version]</p>


<p style="margin-top: 1em">{callpeak,bdgpeakcall,bdgbroadcall,bdgcmp,bdgopt,cmbreps,bdgdiff,filterdup,predictd,pileup,randsample,refinepeak}
...</p>

<p style="margin-top: 1em">macs2 -- Model-based Analysis
for ChIP-Sequencing</p>

<p style="margin-top: 1em">positional arguments: <br>

{callpeak,bdgpeakcall,bdgbroadcall,bdgcmp,bdgopt,cmbreps,bdgdiff,filterdup,predictd,pileup,randsample,refinepeak}</p>

<p style="margin-top: 1em">callpeak <br>
Main MACS2 Function: Call peaks from alignment results.</p>

<p style="margin-top: 1em">bdgpeakcall <br>
Call peaks from bedGraph output. Note: All regions on the
same chromosome in the bedGraph file should be continuous so
only bedGraph files from MACS2 are accpetable.</p>

<p style="margin-top: 1em">bdgbroadcall <br>
Call broad peaks from bedGraph output. Note: All regions on
the same chromosome in the bedGraph file should be
continuous so only bedGraph files from MACS2 are
accpetable.</p>

<p style="margin-top: 1em">bdgcmp Deduct noise by comparing
two signal tracks in bedGraph. Note: All regions on the same
chromosome in the bedGraph file should be continuous so only
bedGraph files from <br>
MACS2 are accpetable.</p>

<p style="margin-top: 1em">bdgopt Operations on score
column of bedGraph file. Note: All regions on the same
chromosome in the bedGraph file should be continuous so only
bedGraph files from MACS2 are <br>
accpetable.</p>

<p style="margin-top: 1em">cmbreps <br>
Combine BEDGraphs of scores from replicates. Note: All
regions on the same chromosome in the bedGraph file should
be continuous so only bedGraph files from MACS2 are <br>
accpetable.</p>

<p style="margin-top: 1em">bdgdiff <br>
Differential peak detection based on paired four bedgraph
files. Note: All regions on the same chromosome in the
bedGraph file should be continuous so only bedGraph files
<br>
from MACS2 are accpetable.</p>

<p style="margin-top: 1em">filterdup <br>
Remove duplicate reads at the same position, then convert
acceptable format to BED format.</p>

<p style="margin-top: 1em">predictd <br>
Predict d or fragment size from alignment results. *Will NOT
filter duplicates*</p>

<p style="margin-top: 1em">pileup Pileup aligned reads with
a given extension size (fragment size or d in MACS
language). Note there will be no step for duplicate reads
filtering or sequencing depth scal&acirc; <br>
ing, so you may need to do certain pre/postprocessing.</p>

<p style="margin-top: 1em">randsample <br>
Randomly sample number/percentage of total reads.</p>

<p style="margin-top: 1em">refinepeak <br>
(Experimental) Take raw reads alignment, refine peak summits
and give scores measuring balance of waston/crick tags.
Inspired by SPP.</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">--version <br>
show program&rsquo;s version number and exit</p>

<p style="margin-top: 1em">For command line options of each
command, type: macs2 COMMAND -h</p>

<p style="margin-top: 1em">macs2 2.1.1.20160309 May 2016
MACS2(1)</p>
<hr>
</body>
</html>
