<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:27 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MACS2_CALLPEAK(1) User Commands MACS2_CALLPEAK(1)</p>

<p style="margin-top: 1em">NAME <br>
macs2_callpeak - Model-based Analysis for
ChIP-Sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: macs2 callpeak [-h] -t TFILE [TFILE ...] [-c [CFILE
[CFILE ...]]]</p>

<p style="margin-top: 1em">[-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}]
<br>
[-g GSIZE] [--keep-dup KEEPDUPLICATES] [--buffer-size
BUFFER_SIZE] [--outdir OUTDIR] [-n NAME] [-B] [--verbose
VERBOSE] [--trackline] [--SPMR] [-s TSIZE] [--bw BW] [-m
<br>
MFOLD MFOLD] [--fix-bimodal] [--nomodel] [--shift SHIFT]
[--extsize EXTSIZE] [-q QVALUE | -p PVALUE] [--to-large]
[--ratio RATIO] [--down-sample] [--seed SEED] [--tempdir
<br>
TEMPDIR] [--nolambda] [--slocal SMALLLOCAL] [--llocal
LARGELOCAL] [--broad] [--broad-cutoff BROADCUTOFF]
[--cutoff-analysis] [--call-summits] [--fe-cutoff
FECUTOFF]</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">Input files arguments: <br>
-t TFILE [TFILE ...], --treatment TFILE [TFILE ...] <br>
ChIP-seq treatment file. If multiple files are given as
&rsquo;-t A B C&rsquo;, then they will all be read and
pooled together. REQUIRED.</p>

<p style="margin-top: 1em">-c [CFILE [CFILE ...]],
--control [CFILE [CFILE ...]] <br>
Control file. If multiple files are given as &rsquo;-c A B
C&rsquo;, they will be pooled to estimate ChIP-seq
background noise.</p>

<p style="margin-top: 1em">-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE},
--format
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}
<br>
Format of tag file, &quot;AUTO&quot;, &quot;BED&quot; or
&quot;ELAND&quot; or &quot;ELANDMULTI&quot; or
&quot;ELANDEXPORT&quot; or &quot;SAM&quot; or
&quot;BAM&quot; or &quot;BOWTIE&quot; or &quot;BAMPE&quot;
or &quot;BEDPE&quot;. The default AUTO option will let MACS
<br>
decide which format (except for BAMPE and BEDPE which should
be implicitly set) the file is. Please check the definition
in README. Please note that if the format is set <br>
as BAMPE or BEDPE, MACS2 will call its special Paired-end
mode to call peaks by piling up the actual ChIPed fragments
defined by both aligned ends, instead of predicting <br>
the fragment size first and extending reads. Also please
note that the BEDPE only contains three columns, and is NOT
the same BEDPE format used by BEDTOOLS. DEFAULT: <br>
&quot;AUTO&quot;</p>

<p style="margin-top: 1em">-g GSIZE, --gsize GSIZE <br>
Effective genome size. It can be 1.0e+9 or 1000000000, or
shortcuts:&rsquo;hs&rsquo; for human (2.7e9),
&rsquo;mm&rsquo; for mouse (1.87e9), &rsquo;ce&rsquo; for C.
elegans (9e7) and &rsquo;dm&rsquo; for fruitfly <br>
(1.2e8), Default:hs</p>

<p style="margin-top: 1em">--keep-dup KEEPDUPLICATES <br>
It controls the MACS behavior towards duplicate tags at the
exact same location -- the same coordination and the same
strand. The &rsquo;auto&rsquo; option makes MACS calculate
the <br>
maximum tags at the exact same location based on binomal
distribution using 1e-5 as pvalue cutoff; and the
&rsquo;all&rsquo; option keeps every tags. If an integer is
given, at most <br>
this number of tags will be kept at the same location. Note,
if you&rsquo;ve used samtools or picard to flag reads as
&rsquo;PCR/Optical duplicate&rsquo; in bit 1024, MACS2 will
still read <br>
them although the reads may be decided by MACS2 as duplicate
later. The default is to keep one tag at the same location.
Default: 1</p>

<p style="margin-top: 1em">--buffer-size BUFFER_SIZE <br>
Buffer size for incrementally increasing internal array size
to store reads alignment information. In most cases, you
don&rsquo;t have to change this parameter. However, if <br>
there are large number of chromosomes/contigs/scaffolds in
your alignment, it&rsquo;s recommended to specify a smaller
buffer size in order to decrease memory usage (but it will
<br>
take longer time to read alignment files). Minimum memory
requested for reading an alignment file is about # of
CHROMOSOME * BUFFER_SIZE * 2 Bytes. DEFAULT: 100000</p>

<p style="margin-top: 1em">Output arguments: <br>
--outdir OUTDIR <br>
If specified all output files will be written to that
directory. Default: the current working directory</p>

<p style="margin-top: 1em">-n NAME, --name NAME <br>
Experiment name, which will be used to generate output file
names. DEFAULT: &quot;NA&quot;</p>

<p style="margin-top: 1em">-B, --bdg <br>
Whether or not to save extended fragment pileup, and local
lambda tracks (two files) at every bp into a bedGraph file.
DEFAULT: False</p>

<p style="margin-top: 1em">--verbose VERBOSE <br>
Set verbose level of runtime message. 0: only show critical
message, 1: show additional warning message, 2: show process
information, 3: show debug messages. DEFAULT:2</p>

<p style="margin-top: 1em">--trackline <br>
Tells MACS to include trackline with bedGraph files. To
include this trackline while displaying bedGraph at UCSC
genome browser, can show name and description of the file
<br>
as well. However my suggestion is to convert bedGraph to
bigWig, then show the smaller and faster binary bigWig file
at UCSC genome browser, as well as downstream analy&acirc;
<br>
sis. Require -B to be set. Default: Not include
trackline.</p>

<p style="margin-top: 1em">--SPMR If True, MACS will save
signal per million reads for fragment pileup profiles.
Require -B to be set. Default: False</p>

<p style="margin-top: 1em">Shifting model arguments: <br>
-s TSIZE, --tsize TSIZE <br>
Tag size. This will override the auto detected tag size.
DEFAULT: Not set</p>

<p style="margin-top: 1em">--bw BW <br>
Band width for picking regions to compute fragment size.
This value is only used while building the shifting model.
DEFAULT: 300</p>

<p style="margin-top: 1em">-m MFOLD MFOLD, --mfold MFOLD
MFOLD <br>
Select the regions within MFOLD range of highconfidence
enrichment ratio against background to build model.
Fold-enrichment in regions must be lower than upper limit,
and <br>
higher than the lower limit. Use as &quot;-m 10 30&quot;.
DEFAULT:5 50</p>

<p style="margin-top: 1em">--fix-bimodal <br>
Whether turn on the auto pair model process. If set, when
MACS failed to build paired model, it will use the nomodel
settings, the --exsize parameter to extend each tags <br>
towards 3&rsquo; direction. Not to use this automate
fixation is a default behavior now. DEFAULT: False</p>

<p style="margin-top: 1em">--nomodel <br>
Whether or not to build the shifting model. If True, MACS
will not build model. by default it means shifting size =
100, try to set extsize to change it. DEFAULT: False</p>

<p style="margin-top: 1em">--shift SHIFT <br>
(NOT the legacy --shiftsize option!) The arbitrary shift in
bp. Use discretion while setting it other than default
value. When NOMODEL is set, MACS will use this value to <br>
move cutting ends (5&rsquo;) towards 5&rsquo;-&gt;3&rsquo;
direction then apply EXTSIZE to extend them to fragments.
When this value is negative, ends will be moved toward
3&rsquo;-&gt;5&rsquo; direction. <br>
Recommended to keep it as default 0 for ChIP-Seq datasets,
or -1 * half of EXTSIZE together with EXTSIZE option for
detecting enriched cutting loci such as certain DNA&acirc;
<br>
seI-Seq datasets. Note, you can&rsquo;t set values other
than 0 if format is BAMPE or BEDPE for paired-end data.
DEFAULT: 0.</p>

<p style="margin-top: 1em">--extsize EXTSIZE <br>
The arbitrary extension size in bp. When nomodel is true,
MACS will use this value as fragment size to extend each
read towards 3&rsquo; end, then pile them up. It&rsquo;s
exactly <br>
twice the number of obsolete SHIFTSIZE. In previous
language, each read is moved 5&rsquo;-&gt;3&rsquo; direction
to middle of fragment by 1/2 d, then extended to both
direction with 1/2 <br>
d. This is equivalent to say each read is extended towards
5&rsquo;-&gt;3&rsquo; into a d size fragment. DEFAULT: 200.
EXTSIZE and SHIFT can be combined when necessary. Check
SHIFT <br>
option.</p>

<p style="margin-top: 1em">Peak calling arguments: <br>
-q QVALUE, --qvalue QVALUE <br>
Minimum FDR (q-value) cutoff for peak detection. DEFAULT:
0.05. -q, and -p are mutually exclusive.</p>

<p style="margin-top: 1em">-p PVALUE, --pvalue PVALUE <br>
Pvalue cutoff for peak detection. DEFAULT: not set. -q, and
-p are mutually exclusive. If pvalue cutoff is set, qvalue
will not be calculated and reported as -1 in the <br>
final .xls file.</p>

<p style="margin-top: 1em">--to-large <br>
When set, scale the small sample up to the bigger sample. By
default, the bigger dataset will be scaled down towards the
smaller dataset, which will lead to smaller <br>
p/qvalues and more specific results. Keep in mind that
scaling down will bring down background noise more. DEFAULT:
False</p>

<p style="margin-top: 1em">--ratio RATIO <br>
When set, use a custom scaling ratio of ChIP/control (e.g.
calculated using NCIS) for linear scaling. DEFAULT:
ingore</p>

<p style="margin-top: 1em">--down-sample <br>
When set, random sampling method will scale down the bigger
sample. By default, MACS uses linear scaling. Warning: This
option will make your result unstable and irrepro&acirc;
<br>
ducible since each time, random reads would be selected.
Consider to use &rsquo;randsample&rsquo; script instead.
&lt;not implmented&gt;If used together with --SPMR, 1
million unique reads <br>
will be randomly picked.&lt;/not implemented&gt; Caution:
due to the implementation, the final number of selected
reads may not be as you expected! DEFAULT: False</p>

<p style="margin-top: 1em">--seed SEED <br>
Set the random seed while down sampling data. Must be a
non-negative integer in order to be effective. DEFAULT: not
set</p>

<p style="margin-top: 1em">--tempdir TEMPDIR <br>
Optional directory to store temp files. DEFAULT: /tmp</p>

<p style="margin-top: 1em">--nolambda <br>
If True, MACS will use fixed background lambda as local
lambda for every peak region. Normally, MACS calculates a
dynamic local lambda to reflect the local bias due to <br>
potential chromatin structure.</p>

<p style="margin-top: 1em">--slocal SMALLLOCAL <br>
The small nearby region in basepairs to calculate dynamic
lambda. This is used to capture the bias near the peak
summit region. Invalid if there is no control data. If you
<br>
set this to 0, MACS will skip slocal lambda calculation.
*Note* that MACS will always perform a d-size local lambda
calculation. The final local bias should be the maximum <br>
of the lambda value from d, slocal, and llocal size windows.
DEFAULT: 1000</p>

<p style="margin-top: 1em">--llocal LARGELOCAL <br>
The large nearby region in basepairs to calculate dynamic
lambda. This is used to capture the surround bias. If you
set this to 0, MACS will skip llocal lambda calcula&acirc;
<br>
tion. *Note* that MACS will always perform a d-size local
lambda calculation. The final local bias should be the
maximum of the lambda value from d, slocal, and llocal <br>
size windows. DEFAULT: 10000.</p>

<p style="margin-top: 1em">--broad <br>
If set, MACS will try to call broad peaks by linking nearby
highly enriched regions. The linking region is controlled by
another cutoff through --linking-cutoff. The max&acirc; <br>
imum linking region length is 4 times of d from MACS.
DEFAULT: False</p>

<p style="margin-top: 1em">--broad-cutoff BROADCUTOFF <br>
Cutoff for broad region. This option is not available unless
--broad is set. If -p is set, this is a pvalue cutoff,
otherwise, it&rsquo;s a qvalue cutoff. DEFAULT: 0.1</p>

<p style="margin-top: 1em">--cutoff-analysis <br>
While set, MACS2 will analyze number or total length of
peaks that can be called by different p-value cutoff then
output a summary table to help user decide a better
cut&acirc; <br>
off. The table will be saved in NAME_cutoff_analysis.txt
file. Note, minlen and maxgap may affect the results.
WARNING: May take ~30 folds longer time to finish. DEFAULT:
<br>
False</p>

<p style="margin-top: 1em">Post-processing options: <br>
--call-summits <br>
If set, MACS will use a more sophisticated signal processing
approach to find subpeak summits in each enriched peak
region. DEFAULT: False</p>

<p style="margin-top: 1em">--fe-cutoff FECUTOFF <br>
When set, the value will be used to filter out peaks with
low fold-enrichment. Note, MACS2 use 1.0 as pseudocount
while calculating fold-enrichment. DEFAULT: 1.0</p>

<p style="margin-top: 1em">macs2_callpeak 2.1.1.20160309
May 2016 MACS2_CALLPEAK(1)</p>
<hr>
</body>
</html>
