<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:26 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MACS2_BDGCMP(1) User Commands MACS2_BDGCMP(1)</p>

<p style="margin-top: 1em">NAME <br>
macs2_bdgcmp - Model-based Analysis for ChIP-Sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: macs2 bdgcmp [-h] -t TFILE -c CFILE [-S SFACTOR] [-p
PSEUDOCOUNT]</p>

<p style="margin-top: 1em">[-m
{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
[{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} ...]] <br>
[--outdir OUTDIR] (--o-prefix OPREFIX | -o OFILE [OFILE
...])</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">-t TFILE, --tfile TFILE <br>
Treatment bedGraph file, e.g. *_treat_pileup.bdg from
MACSv2. REQUIRED</p>

<p style="margin-top: 1em">-c CFILE, --cfile CFILE <br>
Control bedGraph file, e.g. *_control_lambda.bdg from
MACSv2. REQUIRED</p>

<p style="margin-top: 1em">-S SFACTOR, --scaling-factor
SFACTOR <br>
Scaling factor for treatment and control track. Keep it as
1.0 or default in most cases. Set it ONLY while you have
SPMR output from MACS2 callpeak, and plan to calculate <br>
scores as MACS2 callpeak module. If you want to simulate
&rsquo;callpeak&rsquo; w/o &rsquo;--to-large&rsquo;,
calculate effective smaller sample size after filtering
redudant reads in million <br>
(e.g., put 31.415926 if effective reads are 31,415,926) and
input it for &rsquo;-S&rsquo;; for &rsquo;callpeak
--to-large&rsquo;, calculate effective reads in larger
sample. DEFAULT: 1.0</p>

<p style="margin-top: 1em">-p PSEUDOCOUNT, --pseudocount
PSEUDOCOUNT <br>
The pseudocount used for calculating logLR, logFE or FE. The
count will be applied after normalization of sequencing
depth. DEFAULT: 0.0, no pseudocount is applied.</p>

<p style="margin-top: 1em">-m
{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
[{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} ...],
--method {ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
<br>
[{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} ...] <br>
Method to use while calculating a score in any bin by
comparing treatment value and control value. Available
choices are: ppois, qpois, subtract, logFE, logLR, and
slogLR. <br>
They represent Poisson Pvalue (-log10(pvalue) form) using
control as lambda and treatment as observation, q-value
through a BH process for poisson pvalues, subtraction <br>
from treatment, linear scale fold enrichment, log10 fold
enrichment(need to set pseudocount), log10 likelihood
between ChIP-enriched model and open chromatin model(need to
<br>
set pseudocount), symmetric log10 likelihood between two
ChIP-enrichment models, or maximum value between the two
tracks. Default option is ppois.</p>

<p style="margin-top: 1em">--outdir OUTDIR <br>
If specified all output files will be written to that
directory. Default: the current working directory</p>

<p style="margin-top: 1em">--o-prefix OPREFIX <br>
The PREFIX of output bedGraph file to write scores. If it is
given as A, and method is &rsquo;ppois&rsquo;, output file
will be A_ppois.bdg. Mutually exclusive with -o/--ofile.</p>

<p style="margin-top: 1em">-o OFILE [OFILE ...], --ofile
OFILE [OFILE ...] <br>
Output filename. Mutually exclusive with --o-prefix. The
number and the order of arguments for --ofile must be the
same as for -m.</p>

<p style="margin-top: 1em">macs2_bdgcmp 2.1.1.20160309 May
2016 MACS2_BDGCMP(1)</p>
<hr>
</body>
</html>
