<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MACS2_CALLPEAK(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MACS2_CALLPEAK(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">MACS2_CALLPEAK(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
macs2_callpeak - Model-based Analysis for ChIP-Sequencing
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
usage: macs2 callpeak [-h] <b>-t</b> TFILE [TFILE ...] [-c [CFILE [CFILE ...]]]
<dl class="Bl-tag">
  <dt class="It-tag">[-f
    {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}]</dt>
  <dd class="It-tag">[-g GSIZE] [--keep-dup KEEPDUPLICATES] [--buffer-size
      BUFFER_SIZE] [--outdir OUTDIR] [-n NAME] [-B] [--verbose VERBOSE]
      [--trackline] [--SPMR] [-s TSIZE] [--bw BW] [-m MFOLD MFOLD]
      [--fix-bimodal] [--nomodel] [--shift SHIFT] [--extsize EXTSIZE] [-q QVALUE
      | <b>-p</b> PVALUE] [--to-large] [--ratio RATIO] [--down-sample] [--seed
      SEED] [--tempdir TEMPDIR] [--nolambda] [--slocal SMALLLOCAL] [--llocal
      LARGELOCAL] [--broad] [--broad-cutoff BROADCUTOFF] [--cutoff-analysis]
      [--call-summits] [--fe-cutoff FECUTOFF]</dd>
</dl>
<h2 class="Ss" title="Ss" id="optional_arguments:"><a class="selflink" href="#optional_arguments:">optional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<h2 class="Ss" title="Ss" id="Input_files_arguments:"><a class="selflink" href="#Input_files_arguments:">Input
  files arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> TFILE [TFILE ...], <b>--treatment</b> TFILE
    [TFILE ...]</dt>
  <dd class="It-tag">ChIP-seq treatment file. If multiple files are given as '-t
      A B C', then they will all be read and pooled together. REQUIRED.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b> [CFILE [CFILE ...]], <b>--control</b> [CFILE
    [CFILE ...]]</dt>
  <dd class="It-tag">Control file. If multiple files are given as '-c A B C',
      they will be pooled to estimate ChIP-seq background noise.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>
    {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE},
    <b>--format</b>
    {AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE,BAMPE,BEDPE}</dt>
  <dd class="It-tag">Format of tag file, &quot;AUTO&quot;, &quot;BED&quot; or
      &quot;ELAND&quot; or &quot;ELANDMULTI&quot; or &quot;ELANDEXPORT&quot; or
      &quot;SAM&quot; or &quot;BAM&quot; or &quot;BOWTIE&quot; or
      &quot;BAMPE&quot; or &quot;BEDPE&quot;. The default AUTO option will let
      MACS decide which format (except for BAMPE and BEDPE which should be
      implicitly set) the file is. Please check the definition in README. Please
      note that if the format is set as BAMPE or BEDPE, MACS2 will call its
      special Paired-end mode to call peaks by piling up the actual ChIPed
      fragments defined by both aligned ends, instead of predicting the fragment
      size first and extending reads. Also please note that the BEDPE only
      contains three columns, and is NOT the same BEDPE format used by BEDTOOLS.
      DEFAULT: &quot;AUTO&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b> GSIZE, <b>--gsize</b> GSIZE</dt>
  <dd class="It-tag">Effective genome size. It can be 1.0e+9 or 1000000000, or
      shortcuts:'hs' for human (2.7e9), 'mm' for mouse (1.87e9), 'ce' for C.
      elegans (9e7) and 'dm' for fruitfly (1.2e8), Default:hs</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--keep-dup</b> KEEPDUPLICATES</dt>
  <dd class="It-tag">It controls the MACS behavior towards duplicate tags at the
      exact same location <b>--</b> the same coordination and the same strand.
      The 'auto' option makes MACS calculate the maximum tags at the exact same
      location based on binomal distribution using 1e-5 as pvalue cutoff; and
      the 'all' option keeps every tags. If an integer is given, at most this
      number of tags will be kept at the same location. Note, if you've used
      samtools or picard to flag reads as 'PCR/Optical duplicate' in bit 1024,
      MACS2 will still read them although the reads may be decided by MACS2 as
      duplicate later. The default is to keep one tag at the same location.
      Default: 1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--buffer-size</b> BUFFER_SIZE</dt>
  <dd class="It-tag">Buffer size for incrementally increasing internal array
      size to store reads alignment information. In most cases, you don't have
      to change this parameter. However, if there are large number of
      chromosomes/contigs/scaffolds in your alignment, it's recommended to
      specify a smaller buffer size in order to decrease memory usage (but it
      will take longer time to read alignment files). Minimum memory requested
      for reading an alignment file is about # of CHROMOSOME * BUFFER_SIZE * 2
      Bytes. DEFAULT: 100000</dd>
</dl>
<h2 class="Ss" title="Ss" id="Output_arguments:"><a class="selflink" href="#Output_arguments:">Output
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outdir</b> OUTDIR</dt>
  <dd class="It-tag">If specified all output files will be written to that
      directory. Default: the current working directory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> NAME, <b>--name</b> NAME</dt>
  <dd class="It-tag">Experiment name, which will be used to generate output file
      names. DEFAULT: &quot;NA&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b>, <b>--bdg</b></dt>
  <dd class="It-tag">Whether or not to save extended fragment pileup, and local
      lambda tracks (two files) at every bp into a bedGraph file. DEFAULT:
    False</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--verbose</b> VERBOSE</dt>
  <dd class="It-tag">Set verbose level of runtime message. 0: only show critical
      message, 1: show additional warning message, 2: show process information,
      3: show debug messages. DEFAULT:2</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--trackline</b></dt>
  <dd class="It-tag">Tells MACS to include trackline with bedGraph files. To
      include this trackline while displaying bedGraph at UCSC genome browser,
      can show name and description of the file as well. However my suggestion
      is to convert bedGraph to bigWig, then show the smaller and faster binary
      bigWig file at UCSC genome browser, as well as downstream analysis.
      Require <b>-B</b> to be set. Default: Not include trackline.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--SPMR</b></dt>
  <dd class="It-tag">If True, MACS will save signal per million reads for
      fragment pileup profiles. Require <b>-B</b> to be set. Default: False</dd>
</dl>
<h2 class="Ss" title="Ss" id="Shifting_model_arguments:"><a class="selflink" href="#Shifting_model_arguments:">Shifting
  model arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> TSIZE, <b>--tsize</b> TSIZE</dt>
  <dd class="It-tag">Tag size. This will override the auto detected tag size.
      DEFAULT: Not set</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bw</b> BW</dt>
  <dd class="It-tag">Band width for picking regions to compute fragment size.
      This value is only used while building the shifting model. DEFAULT:
    300</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> MFOLD MFOLD, <b>--mfold</b> MFOLD MFOLD</dt>
  <dd class="It-tag">Select the regions within MFOLD range of highconfidence
      enrichment ratio against background to build model. Fold-enrichment in
      regions must be lower than upper limit, and higher than the lower limit.
      Use as &quot;-m 10 30&quot;. DEFAULT:5 50</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fix-bimodal</b></dt>
  <dd class="It-tag">Whether turn on the auto pair model process. If set, when
      MACS failed to build paired model, it will use the nomodel settings, the
      <b>--exsize</b> parameter to extend each tags towards 3' direction. Not to
      use this automate fixation is a default behavior now. DEFAULT: False</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nomodel</b></dt>
  <dd class="It-tag">Whether or not to build the shifting model. If True, MACS
      will not build model. by default it means shifting size = 100, try to set
      extsize to change it. DEFAULT: False</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--shift</b> SHIFT</dt>
  <dd class="It-tag">(NOT the legacy <b>--shiftsize</b> option!) The arbitrary
      shift in bp. Use discretion while setting it other than default value.
      When NOMODEL is set, MACS will use this value to move cutting ends (5')
      towards 5'-&gt;3' direction then apply EXTSIZE to extend them to
      fragments. When this value is negative, ends will be moved toward
      3'-&gt;5' direction. Recommended to keep it as default 0 for ChIP-Seq
      datasets, or <b>-1</b> * half of EXTSIZE together with EXTSIZE option for
      detecting enriched cutting loci such as certain DNAseI-Seq datasets. Note,
      you can't set values other than 0 if format is BAMPE or BEDPE for
      paired-end data. DEFAULT: 0.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--extsize</b> EXTSIZE</dt>
  <dd class="It-tag">The arbitrary extension size in bp. When nomodel is true,
      MACS will use this value as fragment size to extend each read towards 3'
      end, then pile them up. It's exactly twice the number of obsolete
      SHIFTSIZE. In previous language, each read is moved 5'-&gt;3' direction to
      middle of fragment by 1/2 d, then extended to both direction with 1/2 d.
      This is equivalent to say each read is extended towards 5'-&gt;3' into a d
      size fragment. DEFAULT: 200. EXTSIZE and SHIFT can be combined when
      necessary. Check SHIFT option.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Peak_calling_arguments:"><a class="selflink" href="#Peak_calling_arguments:">Peak
  calling arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> QVALUE, <b>--qvalue</b> QVALUE</dt>
  <dd class="It-tag">Minimum FDR (q-value) cutoff for peak detection. DEFAULT:
      0.05. <b>-q</b>, and <b>-p</b> are mutually exclusive.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> PVALUE, <b>--pvalue</b> PVALUE</dt>
  <dd class="It-tag">Pvalue cutoff for peak detection. DEFAULT: not set.
      <b>-q</b>, and <b>-p</b> are mutually exclusive. If pvalue cutoff is set,
      qvalue will not be calculated and reported as <b>-1</b> in the final .xls
      file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--to-large</b></dt>
  <dd class="It-tag">When set, scale the small sample up to the bigger sample.
      By default, the bigger dataset will be scaled down towards the smaller
      dataset, which will lead to smaller p/qvalues and more specific results.
      Keep in mind that scaling down will bring down background noise more.
      DEFAULT: False</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--ratio</b> RATIO</dt>
  <dd class="It-tag">When set, use a custom scaling ratio of ChIP/control (e.g.
      calculated using NCIS) for linear scaling. DEFAULT: ingore</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--down-sample</b></dt>
  <dd class="It-tag">When set, random sampling method will scale down the bigger
      sample. By default, MACS uses linear scaling. Warning: This option will
      make your result unstable and irreproducible since each time, random reads
      would be selected. Consider to use 'randsample' script instead. &lt;not
      implmented&gt;If used together with <b>--SPMR</b>, 1 million unique reads
      will be randomly picked.&lt;/not implemented&gt; Caution: due to the
      implementation, the final number of selected reads may not be as you
      expected! DEFAULT: False</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b> SEED</dt>
  <dd class="It-tag">Set the random seed while down sampling data. Must be a
      non-negative integer in order to be effective. DEFAULT: not set</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tempdir</b> TEMPDIR</dt>
  <dd class="It-tag">Optional directory to store temp files. DEFAULT:
      <i>/tmp</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nolambda</b></dt>
  <dd class="It-tag">If True, MACS will use fixed background lambda as local
      lambda for every peak region. Normally, MACS calculates a dynamic local
      lambda to reflect the local bias due to potential chromatin
    structure.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--slocal</b> SMALLLOCAL</dt>
  <dd class="It-tag">The small nearby region in basepairs to calculate dynamic
      lambda. This is used to capture the bias near the peak summit region.
      Invalid if there is no control data. If you set this to 0, MACS will skip
      slocal lambda calculation. *Note* that MACS will always perform a d-size
      local lambda calculation. The final local bias should be the maximum of
      the lambda value from d, slocal, and llocal size windows. DEFAULT:
    1000</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--llocal</b> LARGELOCAL</dt>
  <dd class="It-tag">The large nearby region in basepairs to calculate dynamic
      lambda. This is used to capture the surround bias. If you set this to 0,
      MACS will skip llocal lambda calculation. *Note* that MACS will always
      perform a d-size local lambda calculation. The final local bias should be
      the maximum of the lambda value from d, slocal, and llocal size windows.
      DEFAULT: 10000.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--broad</b></dt>
  <dd class="It-tag">If set, MACS will try to call broad peaks by linking nearby
      highly enriched regions. The linking region is controlled by another
      cutoff through <b>--linking-cutoff</b>. The maximum linking region length
      is 4 times of d from MACS. DEFAULT: False</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--broad-cutoff</b> BROADCUTOFF</dt>
  <dd class="It-tag">Cutoff for broad region. This option is not available
      unless <b>--broad</b> is set. If <b>-p</b> is set, this is a pvalue
      cutoff, otherwise, it's a qvalue cutoff. DEFAULT: 0.1</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cutoff-analysis</b></dt>
  <dd class="It-tag">While set, MACS2 will analyze number or total length of
      peaks that can be called by different p-value cutoff then output a summary
      table to help user decide a better cutoff. The table will be saved in
      NAME_cutoff_analysis.txt file. Note, minlen and maxgap may affect the
      results. WARNING: May take ~30 folds longer time to finish. DEFAULT:
    False</dd>
</dl>
<h2 class="Ss" title="Ss" id="Post-processing_options:"><a class="selflink" href="#Post-processing_options:">Post-processing
  options:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--call-summits</b></dt>
  <dd class="It-tag">If set, MACS will use a more sophisticated signal
      processing approach to find subpeak summits in each enriched peak region.
      DEFAULT: False</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fe-cutoff</b> FECUTOFF</dt>
  <dd class="It-tag">When set, the value will be used to filter out peaks with
      low fold-enrichment. Note, MACS2 use 1.0 as pseudocount while calculating
      fold-enrichment. DEFAULT: 1.0</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">May 2016</td>
    <td class="foot-os">macs2_callpeak 2.1.1.20160309</td>
  </tr>
</table>
</body>
</html>
