<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MACS2_BDGCMP(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MACS2_BDGCMP(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">MACS2_BDGCMP(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
macs2_bdgcmp - Model-based Analysis for ChIP-Sequencing
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
usage: macs2 bdgcmp [-h] <b>-t</b> TFILE <b>-c</b> CFILE [-S SFACTOR] [-p
  PSEUDOCOUNT]
<dl class="Bl-tag">
  <dt class="It-tag">[-m {ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
    [{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} ...]]</dt>
  <dd class="It-tag">[--outdir OUTDIR] ( <b>--o-prefix</b> OPREFIX | <b>-o</b>
      OFILE [OFILE ...])</dd>
</dl>
<h2 class="Ss" title="Ss" id="optional_arguments:"><a class="selflink" href="#optional_arguments:">optional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> TFILE, <b>--tfile</b> TFILE</dt>
  <dd class="It-tag">Treatment bedGraph file, e.g. *_treat_pileup.bdg from
      MACSv2. REQUIRED</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b> CFILE, <b>--cfile</b> CFILE</dt>
  <dd class="It-tag">Control bedGraph file, e.g. *_control_lambda.bdg from
      MACSv2. REQUIRED</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b> SFACTOR, <b>--scaling-factor</b> SFACTOR</dt>
  <dd class="It-tag">Scaling factor for treatment and control track. Keep it as
      1.0 or default in most cases. Set it ONLY while you have SPMR output from
      MACS2 callpeak, and plan to calculate scores as MACS2 callpeak module. If
      you want to simulate 'callpeak' w/o '--to-large', calculate effective
      smaller sample size after filtering redudant reads in million (e.g., put
      31.415926 if effective reads are 31,415,926) and input it for '-S'; for
      'callpeak <b>--to-large</b>', calculate effective reads in larger sample.
      DEFAULT: 1.0</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b> PSEUDOCOUNT, <b>--pseudocount</b>
    PSEUDOCOUNT</dt>
  <dd class="It-tag">The pseudocount used for calculating logLR, logFE or FE.
      The count will be applied after normalization of sequencing depth.
      DEFAULT: 0.0, no pseudocount is applied.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> {ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
    [{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} ...], <b>--method</b>
    {ppois,qpois,subtract,logFE,FE,logLR,slogLR,max}
    [{ppois,qpois,subtract,logFE,FE,logLR,slogLR,max} ...]</dt>
  <dd class="It-tag">Method to use while calculating a score in any bin by
      comparing treatment value and control value. Available choices are: ppois,
      qpois, subtract, logFE, logLR, and slogLR. They represent Poisson Pvalue (
      <b>-log10</b>(pvalue) form) using control as lambda and treatment as
      observation, q-value through a BH process for poisson pvalues, subtraction
      from treatment, linear scale fold enrichment, log10 fold enrichment(need
      to set pseudocount), log10 likelihood between ChIP-enriched model and open
      chromatin model(need to set pseudocount), symmetric log10 likelihood
      between two ChIP-enrichment models, or maximum value between the two
      tracks. Default option is ppois.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outdir</b> OUTDIR</dt>
  <dd class="It-tag">If specified all output files will be written to that
      directory. Default: the current working directory</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--o-prefix</b> OPREFIX</dt>
  <dd class="It-tag">The PREFIX of output bedGraph file to write scores. If it
      is given as A, and method is 'ppois', output file will be A_ppois.bdg.
      Mutually exclusive with <b>-o</b>/--ofile.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b> OFILE [OFILE ...], <b>--ofile</b> OFILE [OFILE
    ...]</dt>
  <dd class="It-tag">Output filename. Mutually exclusive with <b>--o-prefix</b>.
      The number and the order of arguments for <b>--ofile</b> must be the same
      as for <b>-m</b>.</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">May 2016</td>
    <td class="foot-os">macs2_bdgcmp 2.1.1.20160309</td>
  </tr>
</table>
</body>
</html>
