<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:27 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MACS2_RANDSAMPLE(1) User Commands
MACS2_RANDSAMPLE(1)</p>

<p style="margin-top: 1em">NAME <br>
macs2_randsample - Model-based Analysis for
ChIP-Sequencing</p>

<p style="margin-top: 1em">DESCRIPTION <br>
usage: macs2 randsample [-h] -t TFILE [TFILE ...] (-p
PERCENTAGE | -n NUMBER)</p>

<p style="margin-top: 1em">[--seed SEED] [-o OUTPUTFILE]
[--outdir OUTDIR] <br>
[-s TSIZE] [-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE}]
[--verbose VERBOSE]</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">-t TFILE [TFILE ...], --tfile
TFILE [TFILE ...] <br>
ChIP-seq alignment file. If multiple files are given as
&rsquo;-t A B C&rsquo;, then they will all be read and
combined. Note that pair-end data is not supposed to work
with this com&acirc; <br>
mand. REQUIRED.</p>

<p style="margin-top: 1em">-p PERCENTAGE, --percentage
PERCENTAGE <br>
Percentage of tags you want to keep. Input 80.0 for 80%.
This option can&rsquo;t be used at the same time with
-n/--num. REQUIRED</p>

<p style="margin-top: 1em">-n NUMBER, --number NUMBER <br>
Number of tags you want to keep. Input 8000000 or 8e+6 for 8
million. This option can&rsquo;t be used at the same time
with -p/--percent. Note that the number of tags in output
<br>
is approximate as the number specified here. REQUIRED</p>

<p style="margin-top: 1em">--seed SEED <br>
Set the random seed while down sampling data. Must be a
non-negative integer in order to be effective. DEFAULT: not
set</p>

<p style="margin-top: 1em">-o OUTPUTFILE, --ofile
OUTPUTFILE <br>
Output BED file name. If not specified, will write to
standard output. DEFAULT: stdout</p>

<p style="margin-top: 1em">--outdir OUTDIR <br>
If specified all output files will be written to that
directory. Default: the current working directory</p>

<p style="margin-top: 1em">-s TSIZE, --tsize TSIZE <br>
Tag size. This will override the auto detected tag size.
DEFAULT: Not set</p>

<p style="margin-top: 1em">-f
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE},
--format
{AUTO,BAM,SAM,BED,ELAND,ELANDMULTI,ELANDEXPORT,BOWTIE} <br>
Format of tag file, &quot;AUTO&quot;, &quot;BED&quot; or
&quot;ELAND&quot; or &quot;ELANDMULTI&quot; or
&quot;ELANDEXPORT&quot; or &quot;SAM&quot; or
&quot;BAM&quot; or &quot;BOWTIE&quot;. The default AUTO
option will macs2 randsample decide which <br>
format the file is. Please check the definition in README
file if you choose
ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT:
&quot;AUTO&quot;</p>

<p style="margin-top: 1em">--verbose VERBOSE <br>
Set verbose level. 0: only show critical message, 1: show
additional warning message, 2: show process information, 3:
show debug messages. If you want to know where are <br>
the duplicate reads, use 3. DEFAULT:2</p>

<p style="margin-top: 1em">macs2_randsample 2.1.1.20160309
May 2016 MACS2_RANDSAMPLE(1)</p>
<hr>
</body>
</html>
