<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MAFFT(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MAFFT(1)</td>
    <td class="head-vol">Mafft Manual</td>
    <td class="head-rtitle">MAFFT(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="THIS_MANUAL_IS_FOR_V6.2XX_(2007)"><a class="selflink" href="#THIS_MANUAL_IS_FOR_V6.2XX_(2007)">THIS
  MANUAL IS FOR V6.2XX (2007)</a></h1>
Recent versions (v7.1xx; 2013 Jan.) have more features than those described
  below. See also the tips page at
  http://mafft.cbrc.jp/alignment/software/tips0.html
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<div style="margin-left: 0.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
mafft - Multiple alignment program for amino acid or nucleotide sequences</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  [ <b>options</b>] <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>linsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>ginsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>einsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 7.00ex; text-indent: -7.00ex;"><b>fftnsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>fftns</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>nwns</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>nwnsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 14.00ex; text-indent: -14.00ex;"><b>mafft-profile</b>
  <i>group1</i> <i>group2</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<i>input</i>, <i>group1</i> and <i>group2</i> must be in FASTA format.</div>
</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<div style="margin-left: 0.00ex;"><b>MAFFT</b> is a multiple sequence alignment
  program for unix-like operating systems. It offers a range of multiple
  alignment methods.</div>
<h2 class="Ss" title="Ss" id="Accuracy-oriented_methods:"><a class="selflink" href="#Accuracy-oriented_methods:">Accuracy-oriented
  methods:</a></h2>
<div style="margin-left: 4.00ex;">&#x2022;L-INS-i (probably most accurate;
  recommended for &lt;200 sequences; iterative refinement method incorporating
  local pairwise alignment information):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--localpair</b> <b>--maxiterate</b>&#x00A0;<i>1000</i> <i>input</i>
  [&gt;&#x00A0; <i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>linsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;G-INS-i (suitable for sequences of
  similar lengths; recommended for &lt;200 sequences; iterative refinement
  method incorporating global pairwise alignment information):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--globalpair</b> <b>--maxiterate</b>&#x00A0;<i>1000</i> <i>input</i>
  [&gt;&#x00A0; <i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>ginsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;E-INS-i (suitable for sequences
  containing large unalignable regions; recommended for &lt;200 sequences):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--ep</b>&#x00A0;<i>0</i> <b>--genafpair</b>
  <b>--maxiterate</b>&#x00A0;<i>1000</i> <i>input</i>
  [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>einsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]
<div>&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
For E-INS-i, the <b>--ep</b> <i>0</i> option is recommended to allow large
  gaps.</div>
</div>
<h2 class="Ss" title="Ss" id="Speed-oriented_methods:"><a class="selflink" href="#Speed-oriented_methods:">Speed-oriented
  methods:</a></h2>
<div style="margin-left: 4.00ex;">&#x2022;FFT-NS-i (iterative refinement method;
  two cycles only):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--retree</b>&#x00A0;<i>2</i> <b>--maxiterate</b>&#x00A0;<i>2</i>
  <i>input</i> [&gt;&#x00A0; <i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 7.00ex; text-indent: -7.00ex;"><b>fftnsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;FFT-NS-i (iterative refinement method;
  max. 1000 iterations):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--retree</b>&#x00A0;<i>2</i> <b>--maxiterate</b>&#x00A0;<i>1000</i>
  <i>input</i> [&gt;&#x00A0; <i>output</i>]</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;FFT-NS-2 (fast; progressive method):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--retree</b>&#x00A0;<i>2</i> <b>--maxiterate</b>&#x00A0;<i>0</i>
  <i>input</i> [&gt;&#x00A0; <i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>fftns</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;FFT-NS-1 (very fast; recommended for
  &gt;2000 sequences; progressive method with a rough guide tree):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--retree</b>&#x00A0;<i>1</i> <b>--maxiterate</b>&#x00A0;<i>0</i>
  <i>input</i> [&gt;&#x00A0; <i>output</i>]</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;NW-NS-i (iterative refinement method
  without FFT approximation; two cycles only):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--retree</b>&#x00A0;<i>2</i> <b>--maxiterate</b>&#x00A0;<i>2</i>
  <b>--nofft</b>&#x00A0; <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 7.00ex; text-indent: -7.00ex;"><b>nwnsi</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;NW-NS-2 (fast; progressive method
  without the FFT approximation):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--retree</b>&#x00A0;<i>2</i> <b>--maxiterate</b>&#x00A0;<i>0</i>
  <b>--nofft</b>&#x00A0; <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>nwns</b>
  <i>input</i> [&gt;&#x00A0;<i>output</i>]</div>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;NW-NS-PartTree-1 (recommended for
  ~10,000 to ~50,000 sequences; progressive method with the PartTree algorithm):
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>mafft</b>
  <b>--retree</b>&#x00A0;<i>1</i> <b>--maxiterate</b>&#x00A0;<i>0</i>
  <b>--nofft</b>&#x00A0; <b>--parttree</b> <i>input</i>
  [&gt;&#x00A0;<i>output</i>]</div>
</div>
<h2 class="Ss" title="Ss" id="Group-to-group_alignments"><a class="selflink" href="#Group-to-group_alignments">Group-to-group
  alignments</a></h2>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"></div>
<div style="margin-left: 4.00ex;"><b>mafft-profile</b> <i>group1</i>
  <i>group2</i> [&gt;&#x00A0; <i>output</i>]
<div style="height: 1.00em;">&#x00A0;</div>
or:
<div style="height: 1.00em;">&#x00A0;</div>
<b>mafft</b> <b>--maxiterate</b>&#x00A0;<i>1000</i>
  <b>--seed</b>&#x00A0;<i>group1</i> <b>--seed</b>&#x00A0;<i>group2</i>
  /dev/null [&gt;&#x00A0; <i>output</i>]</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Algorithm"><a class="selflink" href="#Algorithm">Algorithm</a></h2>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
<b>--auto</b>
<div style="margin-left: 4.00ex;">Automatically selects an appropriate strategy
  from L-INS-i, FFT-NS-i and FFT-NS-2, according to data size. Default: off
  (always FFT-NS-2)</div>
<div class="Pp"></div>
<b>--6merpair</b>
<div style="margin-left: 4.00ex;">Distance is calculated based on the number of
  shared 6mers. Default: on</div>
<div class="Pp"></div>
<b>--globalpair</b>
<div style="margin-left: 4.00ex;">All pairwise alignments are computed with the
  Needleman-Wunsch algorithm. More accurate but slower than --6merpair. Suitable
  for a set of globally alignable sequences. Applicable to up to ~200 sequences.
  A combination with --maxiterate 1000 is recommended (G-INS-i). Default: off
  (6mer distance is used)</div>
<div class="Pp"></div>
<b>--localpair</b>
<div style="margin-left: 4.00ex;">All pairwise alignments are computed with the
  Smith-Waterman algorithm. More accurate but slower than --6merpair. Suitable
  for a set of locally alignable sequences. Applicable to up to ~200 sequences.
  A combination with --maxiterate 1000 is recommended (L-INS-i). Default: off
  (6mer distance is used)</div>
<div class="Pp"></div>
<b>--genafpair</b>
<div style="margin-left: 4.00ex;">All pairwise alignments are computed with a
  local algorithm with the generalized affine gap cost (Altschul 1998). More
  accurate but slower than --6merpair. Suitable when large internal gaps are
  expected. Applicable to up to ~200 sequences. A combination with --maxiterate
  1000 is recommended (E-INS-i). Default: off (6mer distance is used)</div>
<div class="Pp"></div>
<b>--fastapair</b>
<div style="margin-left: 4.00ex;">All pairwise alignments are computed with
  FASTA (Pearson and Lipman 1988). FASTA is required. Default: off (6mer
  distance is used)</div>
<div class="Pp"></div>
<b>--weighti</b> <i>number</i>
<div style="margin-left: 4.00ex;">Weighting factor for the consistency term
  calculated from pairwise alignments. Valid when either of --globalpair,
  --localpair, --genafpair, --fastapair or --blastpair is selected. Default:
  2.7</div>
<div class="Pp"></div>
<b>--retree</b> <i>number</i>
<div style="margin-left: 4.00ex;">Guide tree is built <i>number</i> times in the
  progressive stage. Valid with 6mer distance. Default: 2</div>
<div class="Pp"></div>
<b>--maxiterate</b> <i>number</i>
<div style="margin-left: 4.00ex;"><i>number</i> cycles of iterative refinement
  are performed. Default: 0</div>
<div class="Pp"></div>
<b>--fft</b>
<div style="margin-left: 4.00ex;">Use FFT approximation in group-to-group
  alignment. Default: on</div>
<div class="Pp"></div>
<b>--nofft</b>
<div style="margin-left: 4.00ex;">Do not use FFT approximation in group-to-group
  alignment. Default: off</div>
<div class="Pp"></div>
<b>--noscore</b>
<div style="margin-left: 4.00ex;">Alignment score is not checked in the
  iterative refinement stage. Default: off (score is checked)</div>
<div class="Pp"></div>
<b>--memsave</b>
<div style="margin-left: 4.00ex;">Use the Myers-Miller (1988) algorithm.
  Default: automatically turned on when the alignment length exceeds 10,000
  (aa/nt).</div>
<div class="Pp"></div>
<b>--parttree</b>
<div style="margin-left: 4.00ex;">Use a fast tree-building method (PartTree,
  Katoh and Toh 2007) with the 6mer distance. Recommended for a large number
  (&gt; ~10,000) of sequences are input. Default: off</div>
<div class="Pp"></div>
<b>--dpparttree</b>
<div style="margin-left: 4.00ex;">The PartTree algorithm is used with distances
  based on DP. Slightly more accurate and slower than --parttree. Recommended
  for a large number (&gt; ~10,000) of sequences are input. Default: off</div>
<div class="Pp"></div>
<b>--fastaparttree</b>
<div style="margin-left: 4.00ex;">The PartTree algorithm is used with distances
  based on FASTA. Slightly more accurate and slower than --parttree. Recommended
  for a large number (&gt; ~10,000) of sequences are input. FASTA is required.
  Default: off</div>
<div class="Pp"></div>
<b>--partsize</b> <i>number</i>
<div style="margin-left: 4.00ex;">The number of partitions in the PartTree
  algorithm. Default: 50</div>
<div class="Pp"></div>
<b>--groupsize</b> <i>number</i>
<div style="margin-left: 4.00ex;">Do not make alignment larger than
  <i>number</i> sequences. Valid only with the --*parttree options. Default: the
  number of input sequences</div>
</div>
<h2 class="Ss" title="Ss" id="Parameter"><a class="selflink" href="#Parameter">Parameter</a></h2>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
<b>--op</b> <i>number</i>
<div style="margin-left: 4.00ex;">Gap opening penalty at group-to-group
  alignment. Default: 1.53</div>
<div class="Pp"></div>
<b>--ep</b> <i>number</i>
<div style="margin-left: 4.00ex;">Offset value, which works like gap extension
  penalty, for group-to-group alignment. Default: 0.123</div>
<div class="Pp"></div>
<b>--lop</b> <i>number</i>
<div style="margin-left: 4.00ex;">Gap opening penalty at local pairwise
  alignment. Valid when the --localpair or --genafpair option is selected.
  Default: -2.00</div>
<div class="Pp"></div>
<b>--lep</b> <i>number</i>
<div style="margin-left: 4.00ex;">Offset value at local pairwise alignment.
  Valid when the --localpair or --genafpair option is selected. Default:
  0.1</div>
<div class="Pp"></div>
<b>--lexp</b> <i>number</i>
<div style="margin-left: 4.00ex;">Gap extension penalty at local pairwise
  alignment. Valid when the --localpair or --genafpair option is selected.
  Default: -0.1</div>
<div class="Pp"></div>
<b>--LOP</b> <i>number</i>
<div style="margin-left: 4.00ex;">Gap opening penalty to skip the alignment.
  Valid when the --genafpair option is selected. Default: -6.00</div>
<div class="Pp"></div>
<b>--LEXP</b> <i>number</i>
<div style="margin-left: 4.00ex;">Gap extension penalty to skip the alignment.
  Valid when the --genafpair option is selected. Default: 0.00</div>
<div class="Pp"></div>
<b>--bl</b> <i>number</i>
<div style="margin-left: 4.00ex;">BLOSUM <i>number</i> matrix (Henikoff and
  Henikoff 1992) is used. <i>number</i>=30, 45, 62 or 80. Default: 62</div>
<div class="Pp"></div>
<b>--jtt</b> <i>number</i>
<div style="margin-left: 4.00ex;">JTT PAM <i>number</i> (Jones et al. 1992)
  matrix is used. <i>number</i>&gt;0. Default: BLOSUM62</div>
<div class="Pp"></div>
<b>--tm</b> <i>number</i>
<div style="margin-left: 4.00ex;">Transmembrane PAM <i>number</i> (Jones et al.
  1994) matrix is used. <i>number</i>&gt;0. Default: BLOSUM62</div>
<div class="Pp"></div>
<b>--aamatrix</b> <i>matrixfile</i>
<div style="margin-left: 4.00ex;">Use a user-defined AA scoring matrix. The
  format of <i>matrixfile</i> is the same to that of BLAST. Ignored when
  nucleotide sequences are input. Default: BLOSUM62</div>
<div class="Pp"></div>
<b>--fmodel</b>
<div style="margin-left: 4.00ex;">Incorporate the AA/nuc composition information
  into the scoring matrix. Default: off</div>
</div>
<h2 class="Ss" title="Ss" id="Output"><a class="selflink" href="#Output">Output</a></h2>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
<b>--clustalout</b>
<div style="margin-left: 4.00ex;">Output format: clustal format. Default: off
  (fasta format)</div>
<div class="Pp"></div>
<b>--inputorder</b>
<div style="margin-left: 4.00ex;">Output order: same as input. Default: on</div>
<div class="Pp"></div>
<b>--reorder</b>
<div style="margin-left: 4.00ex;">Output order: aligned. Default: off
  (inputorder)</div>
<div class="Pp"></div>
<b>--treeout</b>
<div style="margin-left: 4.00ex;">Guide tree is output to the <i>input</i>.tree
  file. Default: off</div>
<div class="Pp"></div>
<b>--quiet</b>
<div style="margin-left: 4.00ex;">Do not report progress. Default: off</div>
</div>
<h2 class="Ss" title="Ss" id="Input"><a class="selflink" href="#Input">Input</a></h2>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
<b>--nuc</b>
<div style="margin-left: 4.00ex;">Assume the sequences are nucleotide. Default:
  auto</div>
<div class="Pp"></div>
<b>--amino</b>
<div style="margin-left: 4.00ex;">Assume the sequences are amino acid. Default:
  auto</div>
<div class="Pp"></div>
<b>--seed</b> <i>alignment1</i> [<b>--seed</b> <i>alignment2</i> <b>--seed</b>
  <i>alignment3</i> ...]
<div style="margin-left: 4.00ex;">Seed alignments given in <i>alignment_n</i>
  (fasta format) are aligned with sequences in <i>input</i>. The alignment
  within every seed is preserved.</div>
</div>
<h1 class="Sh" title="Sh" id="FILES"><a class="selflink" href="#FILES">FILES</a></h1>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
Mafft stores the input sequences and other files in a temporary directory, which
  by default is located in <i>/tmp</i>.</div>
<h1 class="Sh" title="Sh" id="ENVIONMENT"><a class="selflink" href="#ENVIONMENT">ENVIONMENT</a></h1>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
<b>MAFFT_BINARIES</b>
<div style="margin-left: 4.00ex;">Indicates the location of the binary files
  used by mafft. By default, they are searched in <i>/usr/local/lib/mafft</i>,
  but on Debian systems, they are searched in <i>/usr/lib/mafft</i>.</div>
<div class="Pp"></div>
<b>FASTA_4_MAFFT</b>
<div style="margin-left: 4.00ex;">This variable can be set to indicate to mafft
  the location to the fasta34 program if it is not in the PATH.</div>
</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
<b>mafft-homologs</b>(1)</div>
<h1 class="Sh" title="Sh" id="REFERENCES"><a class="selflink" href="#REFERENCES">REFERENCES</a></h1>
<div style="margin-left: 0.00ex;"></div>
<h2 class="Ss" title="Ss" id="In_English"><a class="selflink" href="#In_English">In
  English</a></h2>
<div style="margin-left: 4.00ex;">&#x2022;Katoh and Toh (Bioinformatics
  23:372-374, 2007) PartTree: an algorithm to build an approximate tree from a
  large number of unaligned sequences (describes the PartTree algorithm).</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Katoh, Kuma, Toh and Miyata (Nucleic
  Acids Res. 33:511-518, 2005) MAFFT version 5: improvement in accuracy of
  multiple sequence alignment (describes [ancestral versions of] the G-INS-i,
  L-INS-i and E-INS-i strategies)</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Katoh, Misawa, Kuma and Miyata
  (Nucleic Acids Res. 30:3059-3066, 2002) MAFFT: a novel method for rapid
  multiple sequence alignment based on fast Fourier transform (describes the
  FFT-NS-1, FFT-NS-2 and FFT-NS-i strategies)</div>
<h2 class="Ss" title="Ss" id="In_Japanese"><a class="selflink" href="#In_Japanese">In
  Japanese</a></h2>
<div style="margin-left: 4.00ex;">&#x2022;Katoh and Misawa (Seibutsubutsuri
  46:312-317, 2006) Multiple Sequence Alignments: the Next Generation</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;">&#x2022;Katoh and Kuma (Kagaku to Seibutsu
  44:102-108, 2006) Jissen-teki Multiple Alignment</div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
<div style="margin-left: 0.00ex;">
<div class="Pp"></div>
<b>Kazutaka Katoh</b> &lt;kazutaka.katoh_at_aist.go.jp&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Wrote Mafft.</dd>
</dl>
<div class="Pp"></div>
<b>Charles Plessy</b> &lt;charles-debian-nospam_at_plessy.org&gt;
<div style="height: 0.00ex;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag">
  <div class="Pp"></div>
  </dt>
  <dd class="It-tag">Wrote this manpage in DocBook XML for the Debian
      distribution, using Mafft's homepage as a template.</dd>
</dl>
</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<div style="margin-left: 0.00ex;">Copyright &#x00A9; 2002-2007 Kazutaka Katoh
  (mafft)
<div>&#x00A0;</div>
Copyright &#x00A9; 2007 Charles Plessy (this manpage)
<div>&#x00A0;</div>
<div class="Pp"></div>
Mafft and its manpage are offered under the following conditions:
<div class="Pp"></div>
Redistribution and use in source and binary forms, with or without modification,
  are permitted provided that the following conditions are met:
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;"> 1.Redistributions of source code must retain
  the above copyright notice, this list of conditions and the following
  disclaimer.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;"> 2.Redistributions in binary form must
  reproduce the above copyright notice, this list of conditions and the
  following disclaimer in the documentation and/or other materials provided with
  the distribution.</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="margin-left: 4.00ex;"> 3.The name of the author may not be used to
  endorse or promote products derived from this software without specific prior
  written permission.</div>
<div class="Pp"></div>
THIS SOFTWARE IS PROVIDED BY THE AUTHOR &quot;AS IS&quot; AND ANY EXPRESS OR
  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
  EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
  BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
  IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  POSSIBILITY OF SUCH DAMAGE.
<div>&#x00A0;</div>
</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2007-06-09</td>
    <td class="foot-os">mafft 6.240</td>
  </tr>
</table>
</body>
</html>
