<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:29 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MAFFT-HOMOLOGS(1) Mafft Manual MAFFT-HOMOLOGS(1)</p>

<p style="margin-top: 1em">NAME <br>
mafft-homologs - aligns sequences together with homologues
automatically collected from SwissProt via NCBI BLAST</p>

<p style="margin-top: 1em">SYNOPSIS <br>
mafft-homologs [options] input [&gt; output]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The accuracy of an alignment of a few distantly related
sequences is considerably improved when being aligned
together with their close homologs. The reason for the
improvement <br>
is probably the same as that for PSI-BLAST. That is, the
positions of highly conserved residues, those with many gaps
and other additional information is brought by close <br>
homologs. According to Katoh et al. (2005), the improvement
by adding close homologs is 10% or so, which is comparable
to the improvement by incorporating structural information
<br>
of a pair of sequences. Mafft-homologs in a mafft server
works like this:</p>

<p style="margin-top: 1em">1. Collect a number (50 by
default) of close homologs (E=1e-10 by default) of the input
sequences.</p>

<p style="margin-top: 1em">2. Align the input sequences and
homologs all together using the L-INS-i strategy.</p>

<p style="margin-top: 1em">3. Remove the homologs.</p>

<p style="margin-top: 1em">OPTIONS <br>
-a n <br>
The number of collected sequences (default: 50).</p>

<p style="margin-top: 1em">-e n <br>
Threshold value (default: 1e-10).</p>

<p style="margin-top: 1em">-o xxx <br>
Options for mafft (default: &quot; --op 1.53 --ep 0.123
--maxiterate 1000 --localpair --reorder&quot;).</p>

<p style="margin-top: 1em">-l <br>
Locally carries out BLAST searches instead of NCBI BLAST
(requires locally installed BLAST and a database).</p>

<p style="margin-top: 1em">-f <br>
Outputs collected homologues also (default: off).</p>

<p style="margin-top: 1em">-w <br>
entire sequences are subjected to BLAST search (default:
well-aligned region only)</p>

<p style="margin-top: 1em">REQUIREMENTS <br>
MAFFT version &gt; 5.58.</p>

<p style="margin-top: 1em">Either of</p>

<p style="margin-top: 1em">lynx (when remote BLAST server
is used)</p>

<p style="margin-top: 1em">BLAST and a protein sequence
database (when local BLAST is used)</p>

<p style="margin-top: 1em">REFERENCES <br>
Katoh, Kuma, Toh and Miyata (Nucleic Acids Res. 33:511-518,
2005) MAFFT version 5: improvement in accuracy of multiple
sequence alignment.</p>

<p style="margin-top: 1em">SEE ALSO <br>
mafft(1)</p>

<p style="margin-top: 1em">AUTHORS <br>
Kazutaka Katoh &lt;katoh_at_bioreg.kyushu-u.ac.jp.&gt; <br>
Wrote Mafft.</p>

<p style="margin-top: 1em">Charles Plessy
&lt;charles-debian-nospam@plessy.org&gt; <br>
Wrote this manpage in DocBook XML for the Debian
distribution, using Mafft&rsquo;s homepage as a
template.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 2002-2007 Kazutaka Katoh (mafft)
<br>
Copyright &Acirc;&copy; 2007 Charles Plessy (this
manpage)</p>

<p style="margin-top: 1em">Mafft and its manpage are
offered under the following conditions:</p>

<p style="margin-top: 1em">Redistribution and use in source
and binary forms, with or without modification, are
permitted provided that the following conditions are
met:</p>

<p style="margin-top: 1em">1. Redistributions of source
code must retain the above copyright notice, this list of
conditions and the following disclaimer.</p>

<p style="margin-top: 1em">2. Redistributions in binary
form must reproduce the above copyright notice, this list of
conditions and the following disclaimer in the documentation
and/or other materials <br>
provided with the distribution.</p>

<p style="margin-top: 1em">3. The name of the author may
not be used to endorse or promote products derived from this
software without specific prior written permission.</p>

<p style="margin-top: 1em">THIS SOFTWARE IS PROVIDED BY THE
AUTHOR &quot;AS IS&quot; AND ANY EXPRESS OR IMPLIED
WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A <br>
PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
NOT <br>
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
<br>
CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
DAMAGE.</p>

<p style="margin-top: 1em">mafft-homologs 2.1 2007-06-09
MAFFT-HOMOLOGS(1)</p>
<hr>
</body>
</html>
