<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:29 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MAFFT(1) Mafft Manual MAFFT(1)</p>

<p style="margin-top: 1em">THIS MANUAL IS FOR V6.2XX (2007)
<br>
Recent versions (v7.1xx; 2013 Jan.) have more features than
those described below. See also the tips page at
http://mafft.cbrc.jp/alignment/software/tips0.html</p>

<p style="margin-top: 1em">NAME <br>
mafft - Multiple alignment program for amino acid or
nucleotide sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
mafft [options] input [&gt; output]</p>

<p style="margin-top: 1em">linsi input [&gt; output]</p>

<p style="margin-top: 1em">ginsi input [&gt; output]</p>

<p style="margin-top: 1em">einsi input [&gt; output]</p>

<p style="margin-top: 1em">fftnsi input [&gt; output]</p>

<p style="margin-top: 1em">fftns input [&gt; output]</p>

<p style="margin-top: 1em">nwns input [&gt; output]</p>

<p style="margin-top: 1em">nwnsi input [&gt; output]</p>

<p style="margin-top: 1em">mafft-profile group1 group2
[&gt; output]</p>

<p style="margin-top: 1em">input, group1 and group2 must be
in FASTA format.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
MAFFT is a multiple sequence alignment program for unix-like
operating systems. It offers a range of multiple alignment
methods.</p>

<p style="margin-top: 1em">Accuracy-oriented methods: <br>
&Acirc;&middot; L-INS-i (probably most accurate; recommended
for &lt;200 sequences; iterative refinement method
incorporating local pairwise alignment information):</p>

<p style="margin-top: 1em">mafft --localpair --maxiterate
1000 input [&gt; output]</p>

<p style="margin-top: 1em">linsi input [&gt; output]</p>

<p style="margin-top: 1em">&Acirc;&middot; G-INS-i
(suitable for sequences of similar lengths; recommended for
&lt;200 sequences; iterative refinement method incorporating
global pairwise alignment information):</p>

<p style="margin-top: 1em">mafft --globalpair --maxiterate
1000 input [&gt; output]</p>

<p style="margin-top: 1em">ginsi input [&gt; output]</p>

<p style="margin-top: 1em">&Acirc;&middot; E-INS-i
(suitable for sequences containing large unalignable
regions; recommended for &lt;200 sequences):</p>

<p style="margin-top: 1em">mafft --ep 0 --genafpair
--maxiterate 1000 input [&gt; output]</p>

<p style="margin-top: 1em">einsi input [&gt; output]</p>

<p style="margin-top: 1em">For E-INS-i, the --ep 0 option
is recommended to allow large gaps.</p>

<p style="margin-top: 1em">Speed-oriented methods: <br>
&Acirc;&middot; FFT-NS-i (iterative refinement method; two
cycles only):</p>

<p style="margin-top: 1em">mafft --retree 2 --maxiterate 2
input [&gt; output]</p>

<p style="margin-top: 1em">fftnsi input [&gt; output]</p>

<p style="margin-top: 1em">&Acirc;&middot; FFT-NS-i
(iterative refinement method; max. 1000 iterations):</p>

<p style="margin-top: 1em">mafft --retree 2 --maxiterate
1000 input [&gt; output]</p>

<p style="margin-top: 1em">&Acirc;&middot; FFT-NS-2 (fast;
progressive method):</p>

<p style="margin-top: 1em">mafft --retree 2 --maxiterate 0
input [&gt; output]</p>

<p style="margin-top: 1em">fftns input [&gt; output]</p>

<p style="margin-top: 1em">&Acirc;&middot; FFT-NS-1 (very
fast; recommended for &gt;2000 sequences; progressive method
with a rough guide tree):</p>

<p style="margin-top: 1em">mafft --retree 1 --maxiterate 0
input [&gt; output]</p>

<p style="margin-top: 1em">&Acirc;&middot; NW-NS-i
(iterative refinement method without FFT approximation; two
cycles only):</p>

<p style="margin-top: 1em">mafft --retree 2 --maxiterate 2
--nofft input [&gt; output]</p>

<p style="margin-top: 1em">nwnsi input [&gt; output]</p>

<p style="margin-top: 1em">&Acirc;&middot; NW-NS-2 (fast;
progressive method without the FFT approximation):</p>

<p style="margin-top: 1em">mafft --retree 2 --maxiterate 0
--nofft input [&gt; output]</p>

<p style="margin-top: 1em">nwns input [&gt; output]</p>

<p style="margin-top: 1em">&Acirc;&middot; NW-NS-PartTree-1
(recommended for ~10,000 to ~50,000 sequences; progressive
method with the PartTree algorithm):</p>

<p style="margin-top: 1em">mafft --retree 1 --maxiterate 0
--nofft --parttree input [&gt; output]</p>

<p style="margin-top: 1em">Group-to-group alignments</p>

<p style="margin-top: 1em">mafft-profile group1 group2
[&gt; output]</p>

<p style="margin-top: 1em">or:</p>

<p style="margin-top: 1em">mafft --maxiterate 1000 --seed
group1 --seed group2 /dev/null [&gt; output]</p>

<p style="margin-top: 1em">OPTIONS <br>
Algorithm <br>
--auto <br>
Automatically selects an appropriate strategy from L-INS-i,
FFT-NS-i and FFT-NS-2, according to data size. Default: off
(always FFT-NS-2)</p>

<p style="margin-top: 1em">--6merpair <br>
Distance is calculated based on the number of shared 6mers.
Default: on</p>

<p style="margin-top: 1em">--globalpair <br>
All pairwise alignments are computed with the
Needleman-Wunsch algorithm. More accurate but slower than
--6merpair. Suitable for a set of globally alignable
sequences. <br>
Applicable to up to ~200 sequences. A combination with
--maxiterate 1000 is recommended (G-INS-i). Default: off
(6mer distance is used)</p>

<p style="margin-top: 1em">--localpair <br>
All pairwise alignments are computed with the Smith-Waterman
algorithm. More accurate but slower than --6merpair.
Suitable for a set of locally alignable sequences. <br>
Applicable to up to ~200 sequences. A combination with
--maxiterate 1000 is recommended (L-INS-i). Default: off
(6mer distance is used)</p>

<p style="margin-top: 1em">--genafpair <br>
All pairwise alignments are computed with a local algorithm
with the generalized affine gap cost (Altschul 1998). More
accurate but slower than --6merpair. Suitable when <br>
large internal gaps are expected. Applicable to up to ~200
sequences. A combination with --maxiterate 1000 is
recommended (E-INS-i). Default: off (6mer distance is
used)</p>

<p style="margin-top: 1em">--fastapair <br>
All pairwise alignments are computed with FASTA (Pearson and
Lipman 1988). FASTA is required. Default: off (6mer distance
is used)</p>

<p style="margin-top: 1em">--weighti number <br>
Weighting factor for the consistency term calculated from
pairwise alignments. Valid when either of --globalpair,
--localpair, --genafpair, --fastapair or --blastpair is <br>
selected. Default: 2.7</p>

<p style="margin-top: 1em">--retree number <br>
Guide tree is built number times in the progressive stage.
Valid with 6mer distance. Default: 2</p>

<p style="margin-top: 1em">--maxiterate number <br>
number cycles of iterative refinement are performed.
Default: 0</p>

<p style="margin-top: 1em">--fft <br>
Use FFT approximation in group-to-group alignment. Default:
on</p>

<p style="margin-top: 1em">--nofft <br>
Do not use FFT approximation in group-to-group alignment.
Default: off</p>

<p style="margin-top: 1em">--noscore <br>
Alignment score is not checked in the iterative refinement
stage. Default: off (score is checked)</p>

<p style="margin-top: 1em">--memsave <br>
Use the Myers-Miller (1988) algorithm. Default:
automatically turned on when the alignment length exceeds
10,000 (aa/nt).</p>

<p style="margin-top: 1em">--parttree <br>
Use a fast tree-building method (PartTree, Katoh and Toh
2007) with the 6mer distance. Recommended for a large number
(&gt; ~10,000) of sequences are input. Default: off</p>

<p style="margin-top: 1em">--dpparttree <br>
The PartTree algorithm is used with distances based on DP.
Slightly more accurate and slower than --parttree.
Recommended for a large number (&gt; ~10,000) of sequences
are <br>
input. Default: off</p>

<p style="margin-top: 1em">--fastaparttree <br>
The PartTree algorithm is used with distances based on
FASTA. Slightly more accurate and slower than --parttree.
Recommended for a large number (&gt; ~10,000) of sequences
are <br>
input. FASTA is required. Default: off</p>

<p style="margin-top: 1em">--partsize number <br>
The number of partitions in the PartTree algorithm. Default:
50</p>

<p style="margin-top: 1em">--groupsize number <br>
Do not make alignment larger than number sequences. Valid
only with the --*parttree options. Default: the number of
input sequences</p>

<p style="margin-top: 1em">Parameter <br>
--op number <br>
Gap opening penalty at group-to-group alignment. Default:
1.53</p>

<p style="margin-top: 1em">--ep number <br>
Offset value, which works like gap extension penalty, for
group-to-group alignment. Default: 0.123</p>

<p style="margin-top: 1em">--lop number <br>
Gap opening penalty at local pairwise alignment. Valid when
the --localpair or --genafpair option is selected. Default:
-2.00</p>

<p style="margin-top: 1em">--lep number <br>
Offset value at local pairwise alignment. Valid when the
--localpair or --genafpair option is selected. Default:
0.1</p>

<p style="margin-top: 1em">--lexp number <br>
Gap extension penalty at local pairwise alignment. Valid
when the --localpair or --genafpair option is selected.
Default: -0.1</p>

<p style="margin-top: 1em">--LOP number <br>
Gap opening penalty to skip the alignment. Valid when the
--genafpair option is selected. Default: -6.00</p>

<p style="margin-top: 1em">--LEXP number <br>
Gap extension penalty to skip the alignment. Valid when the
--genafpair option is selected. Default: 0.00</p>

<p style="margin-top: 1em">--bl number <br>
BLOSUM number matrix (Henikoff and Henikoff 1992) is used.
number=30, 45, 62 or 80. Default: 62</p>

<p style="margin-top: 1em">--jtt number <br>
JTT PAM number (Jones et al. 1992) matrix is used.
number&gt;0. Default: BLOSUM62</p>

<p style="margin-top: 1em">--tm number <br>
Transmembrane PAM number (Jones et al. 1994) matrix is used.
number&gt;0. Default: BLOSUM62</p>

<p style="margin-top: 1em">--aamatrix matrixfile <br>
Use a user-defined AA scoring matrix. The format of
matrixfile is the same to that of BLAST. Ignored when
nucleotide sequences are input. Default: BLOSUM62</p>

<p style="margin-top: 1em">--fmodel <br>
Incorporate the AA/nuc composition information into the
scoring matrix. Default: off</p>

<p style="margin-top: 1em">Output <br>
--clustalout <br>
Output format: clustal format. Default: off (fasta
format)</p>

<p style="margin-top: 1em">--inputorder <br>
Output order: same as input. Default: on</p>

<p style="margin-top: 1em">--reorder <br>
Output order: aligned. Default: off (inputorder)</p>

<p style="margin-top: 1em">--treeout <br>
Guide tree is output to the input.tree file. Default:
off</p>

<p style="margin-top: 1em">--quiet <br>
Do not report progress. Default: off</p>

<p style="margin-top: 1em">Input <br>
--nuc <br>
Assume the sequences are nucleotide. Default: auto</p>

<p style="margin-top: 1em">--amino <br>
Assume the sequences are amino acid. Default: auto</p>

<p style="margin-top: 1em">--seed alignment1 [--seed
alignment2 --seed alignment3 ...] <br>
Seed alignments given in alignment_n (fasta format) are
aligned with sequences in input. The alignment within every
seed is preserved.</p>

<p style="margin-top: 1em">FILES <br>
Mafft stores the input sequences and other files in a
temporary directory, which by default is located in
/tmp.</p>

<p style="margin-top: 1em">ENVIONMENT <br>
MAFFT_BINARIES <br>
Indicates the location of the binary files used by mafft. By
default, they are searched in /usr/local/lib/mafft, but on
Debian systems, they are searched in /usr/lib/mafft.</p>

<p style="margin-top: 1em">FASTA_4_MAFFT <br>
This variable can be set to indicate to mafft the location
to the fasta34 program if it is not in the PATH.</p>

<p style="margin-top: 1em">SEE ALSO <br>
mafft-homologs(1)</p>

<p style="margin-top: 1em">REFERENCES <br>
In English <br>
&Acirc;&middot; Katoh and Toh (Bioinformatics 23:372-374,
2007) PartTree: an algorithm to build an approximate tree
from a large number of unaligned sequences (describes the
PartTree <br>
algorithm).</p>

<p style="margin-top: 1em">&Acirc;&middot; Katoh, Kuma, Toh
and Miyata (Nucleic Acids Res. 33:511-518, 2005) MAFFT
version 5: improvement in accuracy of multiple sequence
alignment (describes [ancestral versions of] <br>
the G-INS-i, L-INS-i and E-INS-i strategies)</p>

<p style="margin-top: 1em">&Acirc;&middot; Katoh, Misawa,
Kuma and Miyata (Nucleic Acids Res. 30:3059-3066, 2002)
MAFFT: a novel method for rapid multiple sequence alignment
based on fast Fourier transform (describes <br>
the FFT-NS-1, FFT-NS-2 and FFT-NS-i strategies)</p>

<p style="margin-top: 1em">In Japanese <br>
&Acirc;&middot; Katoh and Misawa (Seibutsubutsuri
46:312-317, 2006) Multiple Sequence Alignments: the Next
Generation</p>

<p style="margin-top: 1em">&Acirc;&middot; Katoh and Kuma
(Kagaku to Seibutsu 44:102-108, 2006) Jissen-teki Multiple
Alignment</p>

<p style="margin-top: 1em">AUTHORS <br>
Kazutaka Katoh &lt;kazutaka.katoh_at_aist.go.jp&gt; <br>
Wrote Mafft.</p>

<p style="margin-top: 1em">Charles Plessy
&lt;charles-debian-nospam_at_plessy.org&gt; <br>
Wrote this manpage in DocBook XML for the Debian
distribution, using Mafft&rsquo;s homepage as a
template.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright &Acirc;&copy; 2002-2007 Kazutaka Katoh (mafft)
<br>
Copyright &Acirc;&copy; 2007 Charles Plessy (this
manpage)</p>

<p style="margin-top: 1em">Mafft and its manpage are
offered under the following conditions:</p>

<p style="margin-top: 1em">Redistribution and use in source
and binary forms, with or without modification, are
permitted provided that the following conditions are
met:</p>

<p style="margin-top: 1em">1. Redistributions of source
code must retain the above copyright notice, this list of
conditions and the following disclaimer.</p>

<p style="margin-top: 1em">2. Redistributions in binary
form must reproduce the above copyright notice, this list of
conditions and the following disclaimer in the documentation
and/or other materials <br>
provided with the distribution.</p>

<p style="margin-top: 1em">3. The name of the author may
not be used to endorse or promote products derived from this
software without specific prior written permission.</p>

<p style="margin-top: 1em">THIS SOFTWARE IS PROVIDED BY THE
AUTHOR &quot;AS IS&quot; AND ANY EXPRESS OR IMPLIED
WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A <br>
PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
NOT <br>
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
<br>
CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
DAMAGE.</p>

<p style="margin-top: 1em">mafft 6.240 2007-06-09
MAFFT(1)</p>
<hr>
</body>
</html>
