<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:53 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MAP_GFF_COORDS(1) User Commands MAP_GFF_COORDS(1)</p>

<p style="margin-top: 1em">NAME <br>
map_gff_coords - Map coordinates of GFF features from one
genome to another using a Mercator multiple alignment</p>

<p style="margin-top: 1em">SYNOPSIS <br>
map_gff_coords [options] &lt;source genome&gt; &lt;source
GFF file&gt; &lt;target genome&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
map_gff_coords from FSA 1.15.9</p>

<p style="margin-top: 1em">Map coordinates of GFF features
from one genome to another using a Mercator multiple
alignment.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h, --help <br>
show this message</p>

<p style="margin-top: 1em">-t, --type &lt;string&gt; <br>
only look at features of particular type</p>

<p style="margin-top: 1em">-D, --data &lt;directory&gt;
<br>
path to map, genome and alignment files</p>

<p style="margin-top: 1em">-M, --map &lt;directory&gt; <br>
path to map and genome files</p>

<p style="margin-top: 1em">-A, --align &lt;directory&gt;
<br>
path to alignment files</p>

<p style="margin-top: 1em">-L, --lazy <br>
warn, rather than die, if the subalignment can&rsquo;t be
obtained</p>

<p style="margin-top: 1em">-U, --truncate <br>
truncate unmappable sequence (rather than skipping) and show
truncated subalignment</p>

<p style="margin-top: 1em">-f, --force-entry <br>
if a feature can&rsquo;t be mapped, then add an empty entry
to the GFF file (rather than skipping it entirely); implies
--lazy</p>

<p style="margin-top: 1em">-e, --verbose <br>
report progress</p>

<p style="margin-top: 1em">PLEASE NOTE: While this program
is reasonably fast if the GFF is properly ordered by
chromosome and the start and end coordinates of features, it
will be *very slow* if the GFF <br>
is not sorted. Assumes that the &quot;seqid&quot; or
&quot;name&quot; field (the first field) of the GFF entries
holds the chromosome name.</p>

<p style="margin-top: 1em">Note that the GFF specification
defines coordinates to be 1-based and fully-closed,
therefore representing the interval [start, end].
Conformance to this specification is <br>
assumed internally.</p>

<p style="margin-top: 1em">If requested, unmappable
sequence will be truncated to the mappable portion; note
that the truncation will favor the beginning of the
requested sequence.</p>

<p style="margin-top: 1em">If a GFF feature is on the +
strand for the source genome but the corresponding
homologous region in the target genome is on the - strand,
then the mapped GFF feature will be <br>
reported as on the - strand.</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">map_gff_coords 1.15.9 December
2015 MAP_GFF_COORDS(1)</p>
<hr>
</body>
</html>
