<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MAP_GFF_COORDS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MAP_GFF_COORDS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">MAP_GFF_COORDS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
map_gff_coords - Map coordinates of GFF features from one genome to another
  using a Mercator multiple alignment
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>map_gff_coords</b> [ <i>options</i>] <i>&lt;source genome&gt; &lt;source GFF
  file&gt; &lt;target genome&gt;</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
map_gff_coords from FSA 1.15.9
<div class="Pp"></div>
Map coordinates of GFF features from one genome to another using a Mercator
  multiple alignment.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this message</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b>, <b>--type</b> &lt;string&gt;</dt>
  <dd class="It-tag">only look at features of particular type</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b>, <b>--data</b> &lt;directory&gt;</dt>
  <dd class="It-tag">path to map, genome and alignment files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b>, <b>--map</b> &lt;directory&gt;</dt>
  <dd class="It-tag">path to map and genome files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b>, <b>--align</b> &lt;directory&gt;</dt>
  <dd class="It-tag">path to alignment files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b>, <b>--lazy</b></dt>
  <dd class="It-tag">warn, rather than die, if the subalignment can't be
      obtained</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b>, <b>--truncate</b></dt>
  <dd class="It-tag">truncate unmappable sequence (rather than skipping) and
      show truncated subalignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>, <b>--force-entry</b></dt>
  <dd class="It-tag">if a feature can't be mapped, then add an empty entry to
      the GFF file (rather than skipping it entirely); implies
    <b>--lazy</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>, <b>--verbose</b></dt>
  <dd class="It-tag">report progress</dd>
</dl>
<div class="Pp"></div>
PLEASE NOTE: While this program is reasonably fast if the GFF is properly
  ordered by chromosome and the start and end coordinates of features, it will
  be *very slow* if the GFF is not sorted. Assumes that the &quot;seqid&quot; or
  &quot;name&quot; field (the first field) of the GFF entries holds the
  chromosome name.
<div class="Pp"></div>
Note that the GFF specification defines coordinates to be 1-based and
  fully-closed, therefore representing the interval [start, end]. Conformance to
  this specification is assumed internally.
<div class="Pp"></div>
If requested, unmappable sequence will be truncated to the mappable portion;
  note that the truncation will favor the beginning of the requested sequence.
<div class="Pp"></div>
If a GFF feature is on the + strand for the source genome but the corresponding
  homologous region in the target genome is on the - strand, then the mapped GFF
  feature will be reported as on the - strand.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">December 2015</td>
    <td class="foot-os">map_gff_coords 1.15.9</td>
  </tr>
</table>
</body>
</html>
