<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MAPDAMAGE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MAPDAMAGE(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">MAPDAMAGE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
mapDamage - tracking and quantifying damage patterns in ancient DNA sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>mapDamage</b> [ <i>options</i>] <i>-i BAMfile -r reference.fasta</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
MapDamage is a computational framework written in Python and R, which tracks and
  quantifies DNA damage patterns among ancient DNA sequencing reads generated by
  Next-Generation Sequencing platforms.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">show program's version number and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Input files:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b> FILENAME, <b>--input</b>=<i>FILENAME</i></dt>
  <dd class="It-tag">SAM/BAM file, must contain a valid header, use '-' for
      reading a BAM from stdin</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> REF, <b>--reference</b>=<i>REF</i></dt>
  <dd class="It-tag">Reference file in FASTA format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">General options:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> DOWNSAMPLE,
    <b>--downsample</b>=<i>DOWNSAMPLE</i></dt>
  <dd class="It-tag">Downsample to a randomly selected fraction of the reads (if
      0 &lt; DOWNSAMPLE &lt; 1), or a fixed number of randomly selected reads
      (if DOWNSAMPLE &gt;= 1). By default, no downsampling is performed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--downsample-seed</b>=<i>DOWNSAMPLE_SEED</i></dt>
  <dd class="It-tag">Seed value to use for downsampling. See documentation for
      py module 'random' for default behavior.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--merge-reference-sequences</b></dt>
  <dd class="It-tag">Ignore referece sequence names when tabulating reads (using
      '*' instead). Useful for alignments with a large number of reference
      sequnces, which may otherwise result in excessive memory or disk usage due
      to the number of tables generated.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> LENGTH, <b>--length</b>=<i>LENGTH</i></dt>
  <dd class="It-tag">read length, in nucleotides to consider [70]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> AROUND, <b>--around</b>=<i>AROUND</i></dt>
  <dd class="It-tag">nucleotides to retrieve before/after reads [10]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b> MINQUAL,
    <b>--min-basequal</b>=<i>MINQUAL</i></dt>
  <dd class="It-tag">minimun base quality Phred score considered, Phred-33
      assumed [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> FOLDER, <b>--folder</b>=<i>FOLDER</i></dt>
  <dd class="It-tag">folder name to store results [results_FILENAME]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b>, <b>--fasta</b></dt>
  <dd class="It-tag">Write alignments in a FASTA file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--plot-only</b></dt>
  <dd class="It-tag">Run only plotting from a valid result folder</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b>, <b>--quiet</b></dt>
  <dd class="It-tag">Disable any output to stdout</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b>, <b>--verbose</b></dt>
  <dd class="It-tag">Display progression information during parsing</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mapdamage-modules</b>=<i>MAPDAMAGE_MODULES</i></dt>
  <dd class="It-tag">Override the system wide installed mapDamage module</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Options for graphics:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-y</b> YMAX, <b>--ymax</b>=<i>YMAX</i></dt>
  <dd class="It-tag">graphical y-axis limit for nucleotide misincorporation
      frequencies [0.3]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> READPLOT, <b>--readplot</b>=<i>READPLOT</i></dt>
  <dd class="It-tag">read length, in nucleotides, considered for plotting
      nucleotide misincorporations [25]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b> REFPLOT, <b>--refplot</b>=<i>REFPLOT</i></dt>
  <dd class="It-tag">the number of reference nucleotides to consider for ploting
      base composition in the region located upstream and downstream of every
      read [10]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> TITLE, <b>--title</b>=<i>TITLE</i></dt>
  <dd class="It-tag">title used for plots []</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Options for the statistical estimation:</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rand</b>=<i>RAND</i></dt>
  <dd class="It-tag">Number of random starting points for the likelihood
      optimization [30]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--burn</b>=<i>BURN</i></dt>
  <dd class="It-tag">Number of burnin iterations [10000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--adjust</b>=<i>ADJUST</i></dt>
  <dd class="It-tag">Number of adjust proposal variance parameters iterations
      [10]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--iter</b>=<i>ITER</i></dt>
  <dd class="It-tag">Number of final MCMC iterations [50000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--forward</b></dt>
  <dd class="It-tag">Using only the 5' end of the seqs [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--reverse</b></dt>
  <dd class="It-tag">Using only the 3' end of the seqs [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--var-disp</b></dt>
  <dd class="It-tag">Variable dispersion in the overhangs [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--jukes-cantor</b></dt>
  <dd class="It-tag">Use Jukes Cantor instead of HKY85 [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--diff-hangs</b></dt>
  <dd class="It-tag">The overhangs are different for 5' and 3' [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--fix-nicks</b></dt>
  <dd class="It-tag">Fix the nick frequency vector (Only C.T from the 5' end and
      G.A from the 3' end) [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--use-raw-nick-freq</b></dt>
  <dd class="It-tag">Use the raw nick frequency vector without smoothing
    [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--single-stranded</b></dt>
  <dd class="It-tag">Single stranded protocol [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--theme-bw</b></dt>
  <dd class="It-tag">Use black and white theme in post. pred. plot [False]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seq-length</b>=<i>SEQ_LENGTH</i></dt>
  <dd class="It-tag">How long sequence to use from each side [12]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stats-only</b></dt>
  <dd class="It-tag">Run only statistical estimation from a valid result
    folder</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rescale</b></dt>
  <dd class="It-tag">Rescale the quality scores in the BAM file using the output
      from the statistical estimation</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rescale-only</b></dt>
  <dd class="It-tag">Run only rescaling from a valid result folder</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--rescale-out</b>=<i>RESCALE_OUT</i></dt>
  <dd class="It-tag">Write the rescaled BAM to this file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no-stats</b></dt>
  <dd class="It-tag">Disabled statistical estimation, active by default</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--check-R-packages</b></dt>
  <dd class="It-tag">Check if the R modules are working</dd>
</dl>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Report bugs to aginolhac@snm.ku.dk, MSchubert@snm.ku.dk or
  jonsson.hakon@gmail.com
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">August 2016</td>
    <td class="foot-os">mapDamage 2.0.6</td>
  </tr>
</table>
</body>
</html>
