<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:52 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MAPDAMAGE(1) User Commands MAPDAMAGE(1)</p>

<p style="margin-top: 1em">NAME <br>
mapDamage - tracking and quantifying damage patterns in
ancient DNA sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
mapDamage [options] -i BAMfile -r reference.fasta</p>

<p style="margin-top: 1em">DESCRIPTION <br>
MapDamage is a computational framework written in Python and
R, which tracks and quantifies DNA damage patterns among
ancient DNA sequencing reads generated by Next-Generation
<br>
Sequencing platforms.</p>

<p style="margin-top: 1em">OPTIONS <br>
--version <br>
show program&rsquo;s version number and exit</p>

<p style="margin-top: 1em">-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">Input files:</p>

<p style="margin-top: 1em">-i FILENAME, --input=FILENAME
<br>
SAM/BAM file, must contain a valid header, use
&rsquo;-&rsquo; for reading a BAM from stdin</p>

<p style="margin-top: 1em">-r REF, --reference=REF <br>
Reference file in FASTA format</p>

<p style="margin-top: 1em">General options:</p>

<p style="margin-top: 1em">-n DOWNSAMPLE,
--downsample=DOWNSAMPLE <br>
Downsample to a randomly selected fraction of the reads (if
0 &lt; DOWNSAMPLE &lt; 1), or a fixed number of randomly
selected reads (if DOWNSAMPLE &gt;= 1). By default, no
downsam&acirc; <br>
pling is performed.</p>


<p style="margin-top: 1em">--downsample-seed=DOWNSAMPLE_SEED
<br>
Seed value to use for downsampling. See documentation for py
module &rsquo;random&rsquo; for default behavior.</p>

<p style="margin-top: 1em">--merge-reference-sequences <br>
Ignore referece sequence names when tabulating reads (using
&rsquo;*&rsquo; instead). Useful for alignments with a large
number of reference sequnces, which may otherwise result in
<br>
excessive memory or disk usage due to the number of tables
generated.</p>

<p style="margin-top: 1em">-l LENGTH, --length=LENGTH <br>
read length, in nucleotides to consider [70]</p>

<p style="margin-top: 1em">-a AROUND, --around=AROUND <br>
nucleotides to retrieve before/after reads [10]</p>

<p style="margin-top: 1em">-Q MINQUAL,
--min-basequal=MINQUAL <br>
minimun base quality Phred score considered, Phred-33
assumed [0]</p>

<p style="margin-top: 1em">-d FOLDER, --folder=FOLDER <br>
folder name to store results [results_FILENAME]</p>

<p style="margin-top: 1em">-f, --fasta <br>
Write alignments in a FASTA file</p>

<p style="margin-top: 1em">--plot-only <br>
Run only plotting from a valid result folder</p>

<p style="margin-top: 1em">-q, --quiet <br>
Disable any output to stdout</p>

<p style="margin-top: 1em">-v, --verbose <br>
Display progression information during parsing</p>


<p style="margin-top: 1em">--mapdamage-modules=MAPDAMAGE_MODULES
<br>
Override the system wide installed mapDamage module</p>

<p style="margin-top: 1em">Options for graphics:</p>

<p style="margin-top: 1em">-y YMAX, --ymax=YMAX <br>
graphical y-axis limit for nucleotide misincorporation
frequencies [0.3]</p>

<p style="margin-top: 1em">-m READPLOT, --readplot=READPLOT
<br>
read length, in nucleotides, considered for plotting
nucleotide misincorporations [25]</p>

<p style="margin-top: 1em">-b REFPLOT, --refplot=REFPLOT
<br>
the number of reference nucleotides to consider for ploting
base composition in the region located upstream and
downstream of every read [10]</p>

<p style="margin-top: 1em">-t TITLE, --title=TITLE <br>
title used for plots []</p>

<p style="margin-top: 1em">Options for the statistical
estimation:</p>

<p style="margin-top: 1em">--rand=RAND <br>
Number of random starting points for the likelihood
optimization [30]</p>

<p style="margin-top: 1em">--burn=BURN <br>
Number of burnin iterations [10000]</p>

<p style="margin-top: 1em">--adjust=ADJUST <br>
Number of adjust proposal variance parameters iterations
[10]</p>

<p style="margin-top: 1em">--iter=ITER <br>
Number of final MCMC iterations [50000]</p>

<p style="margin-top: 1em">--forward <br>
Using only the 5&rsquo; end of the seqs [False]</p>

<p style="margin-top: 1em">--reverse <br>
Using only the 3&rsquo; end of the seqs [False]</p>

<p style="margin-top: 1em">--var-disp <br>
Variable dispersion in the overhangs [False]</p>

<p style="margin-top: 1em">--jukes-cantor <br>
Use Jukes Cantor instead of HKY85 [False]</p>

<p style="margin-top: 1em">--diff-hangs <br>
The overhangs are different for 5&rsquo; and 3&rsquo;
[False]</p>

<p style="margin-top: 1em">--fix-nicks <br>
Fix the nick frequency vector (Only C.T from the 5&rsquo;
end and G.A from the 3&rsquo; end) [False]</p>

<p style="margin-top: 1em">--use-raw-nick-freq <br>
Use the raw nick frequency vector without smoothing
[False]</p>

<p style="margin-top: 1em">--single-stranded <br>
Single stranded protocol [False]</p>

<p style="margin-top: 1em">--theme-bw <br>
Use black and white theme in post. pred. plot [False]</p>

<p style="margin-top: 1em">--seq-length=SEQ_LENGTH <br>
How long sequence to use from each side [12]</p>

<p style="margin-top: 1em">--stats-only <br>
Run only statistical estimation from a valid result
folder</p>

<p style="margin-top: 1em">--rescale <br>
Rescale the quality scores in the BAM file using the output
from the statistical estimation</p>

<p style="margin-top: 1em">--rescale-only <br>
Run only rescaling from a valid result folder</p>

<p style="margin-top: 1em">--rescale-out=RESCALE_OUT <br>
Write the rescaled BAM to this file</p>

<p style="margin-top: 1em">--no-stats <br>
Disabled statistical estimation, active by default</p>

<p style="margin-top: 1em">--check-R-packages <br>
Check if the R modules are working</p>

<p style="margin-top: 1em">BUGS <br>
Report bugs to aginolhac@snm.ku.dk, MSchubert@snm.ku.dk or
jonsson.hakon@gmail.com</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">mapDamage 2.0.6 August 2016
MAPDAMAGE(1)</p>
<hr>
</body>
</html>
