<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:55 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MAPSEMBLER_EXTEND(1) User Commands
MAPSEMBLER_EXTEND(1)</p>

<p style="margin-top: 1em">NAME <br>
mapsembler_extend - manual page for mapsembler_extend
2.2.3+dfsg</p>

<p style="margin-top: 1em">DESCRIPTION <br>
NAME mapsembler_extend, version 1.0.0 - Copyright INRIA -
CeCILL License</p>

<p style="margin-top: 1em">SYNOPSIS
./mapsembler2_extend/mapsembler_extend
&lt;extrem_kmers.fasta&gt; &lt;readsC1.fasta&gt;
[&lt;readsC2.fasta&gt; [&lt;readsC3.fasta] ...] [-t
extension_type] [-k value] [-c value] [-g value] <br>
[-i index_name] [-o name] [-h]</p>

<p style="margin-top: 1em">DESCRIPTION</p>

<p style="margin-top: 1em">TODO</p>

<p style="margin-top: 1em">OPTIONS</p>

<p style="margin-top: 1em">-t extension_type. Default:
1</p>

<p style="margin-top: 1em">1: a strict sequence: any
branching stops the extension 2: a consensus sequence:
contiging approach 3: a strict graph: any branching is
conserved in the graph 4: a consen&acirc; <br>
sus graph: &quot;small&quot; polymorphism is merged, but
&quot;large&quot; structures are represented</p>

<p style="margin-top: 1em">-k size_kmers: Size of the
k-mers used duriung the extension phase Default: 31.
Accepted range, depends on the compilation (make k=42 for
instance)</p>

<p style="margin-top: 1em">-c min_coverage: a sequence is
covered by at least min_coverage coherent reads. Default:
2</p>

<p style="margin-top: 1em">-g estimated_genome_size:
estimation of the size of the genome whose reads come
from.</p>

<p style="margin-top: 1em">It is in bp, does not need to be
accurate, only controls memory usage. Default: 3 billion</p>

<p style="margin-top: 1em">-x node_len: limit max of nodes
length. Default: 40</p>

<p style="margin-top: 1em">-y graph_max_depth: limit max of
graph depth.Default: 10000</p>

<p style="margin-top: 1em">-i index_name: stores the index
files in files starting with this prefix name. Can be
re-used latter. Default: &quot;index&quot;</p>

<p style="margin-top: 1em">IF THE FILE
&quot;index_name.bloom&quot; EXISTS: the index is not
re-created</p>

<p style="margin-top: 1em">-o file_name_prefix: where to
write outputs. Default: &quot;res_mapsembler&quot;</p>

<p style="margin-top: 1em">-p search_mod: kind of prosses
Breadth or Depth. Default: Breadth</p>

<p style="margin-top: 1em">-h prints this message and
exit</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation for mapsembler_extend is maintained
as a Texinfo manual. If the info and mapsembler_extend
programs are properly installed at your site, the
command</p>

<p style="margin-top: 1em">info mapsembler_extend</p>

<p style="margin-top: 1em">should give you access to the
complete manual.</p>

<p style="margin-top: 1em">mapsembler_extend 2.2.3+dfsg
September 2016 MAPSEMBLER_EXTEND(1)</p>
<hr>
</body>
</html>
