<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MAPSEMBLER2_KISSREADS(1) User Commands
MAPSEMBLER2_KISSREADS(1)</p>

<p style="margin-top: 1em">NAME <br>
mapsembler2_kissreads - manual page for
mapsembler2_kissreads 2.2.3+dfsg</p>

<p style="margin-top: 1em">DESCRIPTION <br>
This is kissreads, version kissreads module 1.2.2 -
Copyright INRIA - CeCILL License The command line was:
./kissreads/mapsembler2_kissreads -h NAME kissReads, version
kissreads <br>
module 1.2.2 - Copyright INRIA - CeCILL License</p>

<p style="margin-top: 1em">SYNOPSIS
./kissreads/mapsembler2_kissreads &lt;toCheck.fasta&gt;
&lt;readsC1.fasta&gt; [&lt;readsC2.fasta&gt;
[&lt;readsC3.fasta] ...] [-k value] [-c value] [-d value]
[-o name] [-u name] [-n] [-I] <br>
[-i index_stride] [-m align_file] [-s] [-f] [-h]</p>

<p style="margin-top: 1em">DESCRIPTION Checks for each
sequence contained into the toCheck.fasta if it is read
coherent (each position is covered by at least
&quot;min_coverage&quot; read(s)) with reads from <br>
readsA.fasta or readsB.fasta A sequence s from toCheck is
treated as follow:</p>

<p style="margin-top: 1em">if (s coherent with at least one
read set): output the sequence as follows</p>

<p style="margin-top: 1em">&gt;original fasta
comment|C1:min&lt;avg-corr_avg&lt;max|C2:min&lt;avg-cor_avg&lt;max|C3...:
&gt;s</p>

<p style="margin-top: 1em">With A:min&lt;avg-cor_avg&lt;max
standing for : value of the position having minimal coverage
in readsA.fasta &lt; average coverage in readsA.fasta -
R-squarred corrected average <br>
in readsA.fa &lt; value of the position having maximal
coverage in readsA.fasta The coverage is the number of reads
that perfectly mapped a position Any other situation (s <br>
not coherent with any): couple non read coherent, not
outputed</p>

<p style="margin-top: 1em">OPTIONS</p>

<p style="margin-top: 1em">-k size_seed: will use seeds of
length size_seed. Default: 25.</p>

<p style="margin-top: 1em">-c min_coverage: a sequence is
covered by at least min_coverage coherent reads. Default:
2</p>

<p style="margin-top: 1em">-d max_substitutions: Maximal
number of substitutions authorized between a read and a
fragment. Note that no substitution is allowed on the
central position while anaylizing the <br>
kissnp output. Default: 1.</p>

<p style="margin-top: 1em">-o file_name: write
read-coherent outputs. Default: standard output</p>

<p style="margin-top: 1em">-u file_name: write
unread-coherent outputs. Default: standard output</p>

<p style="margin-top: 1em">-n the input file
(toCheck.fasta) is a kissnp output (incompatible with -I
option)</p>

<p style="margin-top: 1em">in this case: 1/ only the upper
characters are considered (no mapping done on the
extensions) and 2/ the central position (where the SNP
occurs) is strictly mapped, no <br>
subsitution is authorized on this position.</p>

<p style="margin-top: 1em">-I the input file
(toCheck.fasta) is an Intl output (incompatible with -n
option)</p>

<p style="margin-top: 1em">-i index_stride (int value).
This is a heuristic for limiting the memory footprint.
Instead of indexing each kmer of the sequences contained
into the toCheck.fasta, kissreads <br>
indexes kmers occurring each &quot;index_stride&quot;
position. Default = 1 (no heuristic)</p>

<p style="margin-top: 1em">-t max number of threads (also
limited by number of input files)</p>

<p style="margin-top: 1em">-m align_file, write a file of
reads mapped to sequences in file align_file</p>

<p style="margin-top: 1em">-s silent mode</p>

<p style="margin-top: 1em">--version get the kissReads
version and exit</p>

<p style="margin-top: 1em">-f outputs coherent events in a
standard fasta file format</p>

<p style="margin-top: 1em">-h prints this message and
exit</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation for mapsembler2_kissreads is
maintained as a Texinfo manual. If the info and
mapsembler2_kissreads programs are properly installed at
your site, the com&acirc; <br>
mand</p>

<p style="margin-top: 1em">info mapsembler2_kissreads</p>

<p style="margin-top: 1em">should give you access to the
complete manual.</p>

<p style="margin-top: 1em">mapsembler2_kissreads 2.2.3+dfsg
September 2016 MAPSEMBLER2_KISSREADS(1)</p>
<hr>
</body>
</html>
