<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MAQ(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MAQ(1)</td>
    <td class="head-vol">Bioinformatics Tools</td>
    <td class="head-rtitle">MAQ(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
Maq - Mapping and Assembly with Qualities
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>maq</b> <i>command</i> [<i>options</i>] <i>arguments</i>
<div class="Pp"></div>
<b>maq.pl</b> <i>command</i> [<i>options</i>] <i>arguments</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Maq is a software that builds mapping assemblies from short reads generated by
  the next-generation sequencing machines. It is particularly designed for
  Illumina-Solexa 1G Genetic Analyzer, and has a preliminary functionality to
  handle AB SOLiD data.
<div class="Pp"></div>
With Maq you can:
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Fast align Illumina/SOLiD reads to the reference genome.
      With the default options, one million pairs of reads can be mapped to the
      human genome in about 10 CPU hours with less than 1G memory.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Accurately measure the error probability of the alignment
      of each individual read.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Call the consensus genotypes, including homozygous and
      heterozygous polymorphisms, with a Phred probabilistic quality assigned to
      each base.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Find short indels with paired end reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Accurately find large scale genomic deletions and
      translocations with paired end reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Discover potential CNVs by checking read depth.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Evaluate the accuracy of raw base qualities from sequencers
      and help to check the systematic errors.</dd>
</dl>
<div class="Pp"></div>
However, Maq can <b>NOT</b>:
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Do <i>de novo</i> assembly. (Maq can only call the
      consensus by mapping reads to a known reference.)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Map shorts reads against themselves. (Maq can only find
      complete overlap between reads.)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Align capillary reads or 454 reads to the reference. (Maq
      cannot align reads longer than 63bp.)</dd>
</dl>
<h1 class="Sh" title="Sh" id="MAQ_COMMANDS"><a class="selflink" href="#MAQ_COMMANDS">MAQ
  COMMANDS</a></h1>
<b>Key Commands</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>fasta2bfa</b></dt>
  <dd class="It-tag"><b>maq</b> <b>fasta2bfa</b> <i>in.ref.fasta</i>
      <i>out.ref.bfa</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert sequences in FASTA format to Maq's BFA (binary FASTA) format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>fastq2bfq</b></dt>
  <dd class="It-tag"><b>maq</b> <b>fastq2bfq</b> [<b>-n</b> <i>nreads</i>]
      <i>in.read.fastq</i> <i>out.read.bfq</i>&#x23AA;<i>out.prefix</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert reads in FASTQ format to Maq's BFQ (binary FASTQ) format.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> <i>INT</i></dt>
  <dd class="It-tag">number of reads per file [not specified]</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>map</b></dt>
  <dd class="It-tag"><b>maq</b> <b>map</b> [<b>-n</b> <i>nmis</i>] [<b>-a</b>
      <i>maxins</i>] [<b>-c</b>] [<b>-1</b> <i>len1</i>] [ <b>-2</b>
      <i>len2</i>] [ <b>-d</b> <i>adap3</i>] [<b>-m</b> <i>mutrate</i>]
      [<b>-u</b> <i>unmapped</i>] [-e <i>maxerr</i>] [<b>-M</b> c&#x23AA;g]
      [<b>-N</b>] [ <b>-H</b> <i>allhits</i>] [<b>-C</b> <i>maxhits</i>]
      <i>out.aln.map</i> <i>in.ref.bfa</i> <i>in.read1.bfq</i>
      [<i>in.read2.bfq</i>] 2&gt; <i>out.map.log</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Map reads to the reference sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> <i>INT</i></dt>
  <dd class="It-tag">Number of maximum mismatches that can always be found
    [2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum outer distance for a correct read pair [250]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum outer distance of two RF paied read (0 for disable)
      [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">Map reads in the colour space (for SOLiD only)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b> <i>INT</i></dt>
  <dd class="It-tag">Read length for the first read, 0 for auto [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-2</b> <i>INT</i></dt>
  <dd class="It-tag">Read length for the second read, 0 for auto [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> <i>FLOAT</i></dt>
  <dd class="It-tag">Mutation rate between the reference sequences and the reads
      [0.001]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> <i>FILE</i></dt>
  <dd class="It-tag">Specify a file containing a single line of the 3'-adapter
      sequence [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b> <i>FILE</i></dt>
  <dd class="It-tag">Dump unmapped reads and reads containing more than
      <i>nmis</i> mismatches to a separate file [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b> <i>INT</i></dt>
  <dd class="It-tag">Threshold on the sum of mismatching base qualities
    [70]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-H</b> <i>FILE</i></dt>
  <dd class="It-tag">Dump multiple/all 01-mismatch hits to <i>FILE</i>
    [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum number of hits to output. Unlimited if larger than
      512. [250]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-M</b> c&#x23AA;g</dt>
  <dd class="It-tag">methylation alignment mode. All C (or G) on the forward
      strand will be changed to T (or A). This option is for testing only.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b></dt>
  <dd class="It-tag">store the mismatch position in the output file
      <i>out.aln.map</i>. When this option is in use, the maximum allowed read
      length is 55bp.</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<b>NOTE:</b>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Paired end reads should be prepared in two files, one for
      each end, with reads are sorted in the same order. This means the k-th
      read in the first file is mated with the k-th read in the second file. The
      corresponding read names must be identical up to the tailing `/1' or `/2'.
      For example, such a pair of read names are allowed: `EAS1_1_5_100_200/1'
      and `EAS1_1_5_100_200/2'. The tailing `/[12]' is usually generated by the
      GAPipeline to distinguish the two ends in a pair.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">The output is a compressed binary file. It is affected by
      the endianness.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">The best way to run this command is to provide about 1 to 3
      million reads as input. More reads consume more memory.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Option <b>-n</b> controls the sensitivity of the alignment.
      By default, a hit with up to 2 mismatches can be always found. Higher
      <b>-n</b> finds more hits and also improves the accuracy of mapping
      qualities. However, this is done at the cost of speed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Alignments with many high-quality mismatches should be
      discarded as false alignments or possible contaminations. This behaviour
      is controlled by option <b>-e</b>. The <b>-e</b> threshold is only
      calculated approximately because base qualities are divided by 10 at a
      certain stage of the alignment. The <b>-Q</b> option in the
      <b>assemble</b> command precisely set the threshold.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">A pair of reads are said to be correctly paired if and only
      if the orientation is <i>FR</i> and the outer distance of the pair is no
      larger than <i>maxins</i>. There is no limit on the minimum insert size.
      This setting is determined by the paired end alignment algorithm used in
      Maq. Requiring a minimum insert size will lead to some wrong alignments
      with highly overestimated mapping qualities.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Currently, read pairs from Illumina/Solexa long-insert
      library have RF read orientation. The maximum insert size is set by option
      <b>-A</b>. However, long-insert library is also mixed with a small
      fraction of short-insert read pairs. <b>-a</b> should also be set
      correctly.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Sometimes 5'-end or even the entire 3'-adapter sequence may
      be sequenced. Providing <b>-d</b> renders Maq to eliminate the adapter
      contaminations.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Given 2 million reads as input, <b>maq</b> usually takes
      800MB memory.</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mapmerge</b></dt>
  <dd class="It-tag"><b>maq</b> <b>mapmerge</b> <i>out.aln.map</i>
      <i>in.aln1.map</i> <i>in.aln2.map</i> [...]
    <div style="height: 1.00em;">&#x00A0;</div>
    Merge a batch of read alignments together.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>NOTE:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">In theory, this command can merge unlimited number of
      alignments. However, as mapmerge will be reading all the inputs at the
      same time, it may hit the limit of the maximum number of opening files set
      by the OS. At present, this has to be manually solved by endusers.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Command <b>mapmerge</b> can be used to merge alignment
      files with different read lengths. All the subsequent analyses do not
      assume fixed length any more.</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>rmdup</b></dt>
  <dd class="It-tag"><b>maq</b> <b>rmdup</b> <i>out.rmdup.map</i>
      <i>in.ori.map</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Remove pairs with identical outer coordinates. In principle, pairs with
      identical outer coordinates should happen rarely. However, due to the
      amplification in sample preparation, this occurs much more frequently than
      by chance. Practical analyses show that removing duplicates helps to
      improve the overall accuracy of SNP calling.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>assemble</b></dt>
  <dd class="It-tag"><b>maq</b> <b>assemble</b> [<b>-sp</b>] [<b>-m</b>
      <i>maxmis</i>] [ <b>-Q</b> <i>maxerr</i>] [<b>-r</b> <i>hetrate</i>]
      [<b>-t</b> <i>coef</i>] [<b>-q</b> <i>minQ</i>] [<b>-N</b> <i>nHap</i>]
      <i>out.cns</i> <i>in.ref.bfa</i> <i>in.aln.map</i> 2&gt;
      <i>out.cns.log</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Call the consensus sequences from read mapping.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b> <i>FLOAT</i></dt>
  <dd class="It-tag">Error dependency coefficient [0.93]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> <i>FLOAT</i></dt>
  <dd class="It-tag">Fraction of heterozygotes among all sites [0.001]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">Take single end mapping quality as the final mapping
      quality; otherwise paired end mapping quality will be used</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b></dt>
  <dd class="It-tag">Discard paired end reads that are not mapped in correct
      pairs</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum number of mismatches allowed for a read to be used
      in consensus calling [7]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum allowed sum of quality values of mismatched bases
      [60]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum mapping quality allowed for a read to be used in
      consensus calling [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b> <i>INT</i></dt>
  <dd class="It-tag">Number of haplotypes in the pool (&gt;=2) [2]</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<b>NOTE:</b>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Option <b>-Q</b> sets a limit on the maximum sum of
      mismatching base qualities. Reads containing many high-quality mismatches
      should be discarded.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Option <b>-N</b> sets the number of haplotypes in a pool.
      It is designed for resequencing of samples by pooling multiple
      strains/individuals together. For diploid genome resequencing, this option
      equals 2.</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>glfgen</b></dt>
  <dd class="It-tag"><b>maq</b> <b>glfgen</b> [<b>-sp</b>] [<b>-m</b>
      <i>maxmis</i>] [ <b>-Q</b> <i>maxerr</i>] [<b>-r</b> <i>hetrate</i>]
      [<b>-t</b> <i>coef</i>] [<b>-q</b> <i>minQ</i>] [<b>-N</b> <i>nHap</i>]
      <i>out.cns</i> <i>in.ref.bfa</i> <i>in.aln.map</i> 2&gt;
      <i>out.cns.log</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Calculate log-likelihood for all genotypes and store the results in GLF
      format (Genotyping Likelihood Format). Please check MAQ website for
      detailed descriptions of the file format and the related utilities.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>indelpe</b></dt>
  <dd class="It-tag"><b>maq</b> <b>indelpe</b> <i>in.ref.bfa</i>
      <i>in.aln.map</i> &gt; <i>out.indelpe</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Call consistent indels from paired end reads. The output is TAB delimited
      with each line consisting of chromosome, start position, type of the
      indel, number of reads across the indel, size of the indel and
      inserted/deleted nucleotides (separated by colon), number of indels on the
      reverse strand, number of indels on the forward strand, 5' sequence ahead
      of the indel, 3' sequence following the indel, number of reads aligned
      without indels and three additional columns for filters.
    <div style="height: 1.00em;">&#x00A0;</div>
    At the 3rd column, type of the indel, a star indicates the indel is
      confirmed by reads from both strands, a plus means the indel is hit by at
      least two reads but from the same strand, a minus shows the indel is only
      found on one read, and a dot means the indel is too close to another indel
      and is filtered out.
    <div style="height: 1.00em;">&#x00A0;</div>
    Users are recommended to run through `maq.pl indelpe' to correct the number
      of reads mapped without indels. For more details, see the `maq.pl indelpe'
      section.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>indelsoa</b></dt>
  <dd class="It-tag"><b>maq</b> <b>indelsoa</b> <i>in.ref.bfa</i>
      <i>in.aln.map</i> &gt; <i>out.indelsoa</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Call potential homozygous indels and break points by detecting the abnormal
      alignment pattern around indels and break points. The output is also TAB
      delimited with each line consisting of chromosome, approximate coordinate,
      length of the abnormal region, number of reads mapped across the position,
      number of reads on the left-hand side of the position and number of reads
      on the right-hand side. The last column can be ignored.
    <div style="height: 1.00em;">&#x00A0;</div>
    The output contains many false positives. A recommended filter could be:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
  awk '$5+$6-$4 &gt;= 3 &amp;&amp; $4 &lt;= 1' in.indelsoa
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    Note that this command does not aim to be an accurate indel detector, but
      mainly helps to avoid some false positives in substitution calling. In
      addition, it only works well given deep depth (~40X for example);
      otherwise the false negative rate would be very high.
    <div>&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
<b>Format Converting</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>sol2sanger</b></dt>
  <dd class="It-tag"><b>maq</b> <b>sol2sanger</b> <i>in.sol.fastq</i>
      <i>out.sanger.fastq</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert Solexa FASTQ to standard/Sanger FASTQ format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>bfq2fastq</b></dt>
  <dd class="It-tag"><b>maq</b> <b>bfq2fastq</b> <i>in.read.bfq</i>
      <i>out.read.fastq</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert Maq's BFQ format to standard FASTQ format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mapass2maq</b></dt>
  <dd class="It-tag"><b>maq</b> <b>mapass2maq</b> <i>in.mapass2.map</i>
      <i>out.maq.map</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert obsolete mapass2's map format to Maq's map format. The old format
      does not contain read names.
    <div>&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
<b>Information Extracting</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mapview</b></dt>
  <dd class="It-tag"><b>maq</b> <b>mapview</b> [<b>-bN</b>] <i>in.aln.map</i>
      &gt; <i>out.aln.txt</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Display the read alignment in plain text. For reads aligned before the
      Smith-Waterman alignment, each line consists of read name, chromosome,
      position, strand, insert size from the outer coorniates of a pair, paired
      flag, mapping quality, single-end mapping quality, alternative mapping
      quality, number of mismatches of the best hit, sum of qualities of
      mismatched bases of the best hit, number of 0-mismatch hits of the first
      24bp, number of 1-mismatch hits of the first 24bp on the reference, length
      of the read, read sequence and its quality. Alternative mapping quality
      always equals to mapping quality if the reads are not paired. If reads are
      paired, it equals to the smaller mapping quality of the two ends. This
      alternative mapping quality is actually the mapping quality of an abnormal
      pair.
    <div style="height: 1.00em;">&#x00A0;</div>
    The fifth column, paired flag, is a bitwise flag. Its lower 4 bits give the
      orientation: 1 stands for FF, 2 for FR, 4 for RF, and 8 for RR, where FR
      means that the read with smaller coordinate is on the forward strand, and
      its mate is on the reverse strand. Only FR is allowed for a correct pair.
      The higher bits of this flag give further information. If the pair meets
      the paired end requirement, 16 will be set. If the two reads are mapped to
      different chromosomes, 32 will be set. If one of the two reads cannot be
      mapped at all, 64 will be set. The flag for a correct pair always equals
      to 18.
    <div style="height: 1.00em;">&#x00A0;</div>
    For reads aligned by the Smith-Waterman alignment afterwards, the flag is
      always 130. A line consists of read name, chromosome, position, strand,
      insert size, flag (always 130), position of the indel on the read (0 if no
      indel), length of the indels (positive for insertions and negative for
      deletions), mapping quality of its mate, number of mismatches of the best
      hit, sum of qualities of mismatched bases of the best hit, two zeros,
      length of the read, read sequence and its quality. The mate of a
      130-flagged read always gets a flag 18.
    <div style="height: 1.00em;">&#x00A0;</div>
    Flag 192 indicates that the read is not mapped but its mate is mapped. For
      such a read pair, one read has flag 64 and the other has 192.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b></dt>
  <dd class="It-tag">do not display the read sequence and the quality</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b></dt>
  <dd class="It-tag">display the positions where mismatches occur. This flag
      only works with a .map file generated by `maq map -N'.</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mapcheck</b></dt>
  <dd class="It-tag"><b>maq</b> <b>mapcheck</b> [<b>-s</b>] [<b>-m</b>
      <i>maxmis</i>] [ <b>-q</b> <i>minQ</i>] <i>in.ref.bfa</i>
      <i>in.aln.map</i> &gt; <i>out.mapcheck</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Read quality check. The mapcheck first reports the composition and the depth
      of the reference. After that there is a form. The first column indicates
      the position on a read. Following four columns which show the nucleotide
      composition, substitution rates between the reference and reads will be
      given. These rates and the numbers in the following columns are scaled to
      999 and rounded to nearest integer. The next group of columns show the
      distribution of base qualities along the reads at a quality interval of
      10. A decay in quality can usually be observed, which means bases at the
      end of read are less accurate. The last group of columns present the
      fraction of substitutions for read bases at a quality interval. This
      measures the accuracy of base quality estimation. Idealy, we expect to see
      1 in the 3? column, 10 in the 2? column and 100 in the 1? column.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">Take single end mapping quality as the final mapping
      quality</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum number of mismatahces allowed for a read to be
      counted [4]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum mapping quality allowed for a read to be counted
      [30]</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>pileup</b></dt>
  <dd class="It-tag"><b>maq</b> <b>pileup</b> [<b>-spvP</b>] [<b>-m</b>
      <i>maxmis</i>] [ <b>-Q</b> <i>maxerr</i>] [<b>-q</b> <i>minQ</i>]
      [<b>-l</b> <i>sitefile</i>] <i>in.ref.bfa</i> <i>in.aln.map</i> &gt;
      <i>out.pileup</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Display the alignment in a `pileup' text format. Each line consists of
      chromosome, position, reference base, depth and the bases on reads that
      cover this position. If <b>-v</b> is added on the command line, base
      qualities and mapping qualities will be presented in the sixth and seventh
      columns in order.
    <div style="height: 1.00em;">&#x00A0;</div>
    The fifth column always starts with `@'. In this column, read bases
      identical to the reference are showed in comma `,' or dot `.', and read
      bases different from the reference in letters. A comma or a upper case
      indicates that the base comes from a read aligned on the forward strand,
      while a dot or a lower case on the reverse strand.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command is for users who want to develop their own SNP callers.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">Take single end mapping quality as the final mapping
      quality</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b></dt>
  <dd class="It-tag">Discard paired end reads that are not mapped as correct
      pairs</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b></dt>
  <dd class="It-tag">Output verbose information including base qualities and
      mapping qualities</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum number of mismatches allowed for a read to be used
      [7]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum allowed number of quality values of mismatches
    [60]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum mapping quality allowed for a read to be used
    [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b> <i>FILE</i></dt>
  <dd class="It-tag">File containing the sites at which pileup will be printed
      out. In this file the first column gives the names of the reference and
      the second the coordinates. Additional columns will be ignored.
    [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P</b></dt>
  <dd class="It-tag">also output the base position on the read</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cns2fq</b></dt>
  <dd class="It-tag"><b>maq</b> <b>cns2fq</b> [<b>-Q</b> <i>minMapQ</i>]
      [<b>-n</b> <i>minNeiQ</i>] [<b>-d</b> <i>minDepth</i>] [<b>-D</b>
      <i>maxDepth</i>] <i>in.cns</i> &gt; <i>out.cns.fastq</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Extract the consensus sequences in FASTQ format. In the sequence lines,
      bases in lower case are essentially repeats or do not have sufficient
      coverage; bases in upper case indicate regions where SNPs can be reliably
      called. In the quality lines, ASCII of a character minus 33 gives the
      PHRED quality.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum mapping quality [40]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum read depth [3]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum neighbouring quality [20]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum read dpeth. &gt;=255 for unlimited. [255]</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cns2snp</b></dt>
  <dd class="It-tag"><b>maq</b> <b>cns2snp</b> <i>in.cns</i> &gt; <i>out.snp</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Extract SNP sites. Each line consists of chromosome, position, reference
      base, consensus base, Phred-like consensus quality, read depth, the
      average number of hits of reads covering this position, the highest
      mapping quality of the reads covering the position, the minimum consensus
      quality in the 3bp flanking regions at each side of the site (6bp in
      total), the second best call, log likelihood ratio of the second best and
      the third best call, and the third best call.
    <div style="height: 1.00em;">&#x00A0;</div>
    The 5th column is the key criterion when you judge the reliability of a SNP.
      However, as this quality is only calculated assuming site independency,
      you should also consider other columns to get more accurate SNP calls.
      Script command ` <b>maq.pl</b> <b>SNPfilter</b>' is designed for this (see
      below).
    <div style="height: 1.00em;">&#x00A0;</div>
    The 7th column implies whether the site falls in a repetitive region. If no
      read covering the site can be mapped with high mapping quality, the
      flanking region is possibly repetitive or in the lack of good reads. A SNP
      at such site is usually not reliable.
    <div style="height: 1.00em;">&#x00A0;</div>
    The 8th column roughly gives the copy number of the flanking region in the
      reference genome. In most cases, this number approaches 1.00, which means
      the region is about unique. Sometimes you may see non-zero read depth but
      0.00 at the 7th column. This indicates that all the reads covering the
      position have at least two mismatches. Maq only counts the number of 0-
      and 1-mismatch hits to the reference. This is due to a complex technical
      issue.
    <div style="height: 1.00em;">&#x00A0;</div>
    The 9th column gives the neighbouring quality. Filtering on this column is
      also required to get reliable SNPs. This idea is inspired by NQS, although
      NQS is initially designed for a single read instead of a consensus.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cns2view</b></dt>
  <dd class="It-tag"><b>maq</b> <b>cns2view</b> <i>in.cns</i> &gt;
      <i>out.view</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Show detailed information at all sites. The output format is identical to
      <b>cns2snp</b> report.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cns2ref</b></dt>
  <dd class="It-tag"><b>maq</b> <b>cns2ref</b> <i>in.cns</i> &gt;
      <i>out.ref.fasta</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Extract the reference sequence.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cns2win</b></dt>
  <dd class="It-tag"><b>maq</b> <b>cns2win</b> [<b>-w</b> <i>winsize</i>]
      [<b>-c</b> <i>chr</i>] [<b>-b</b> <i>begin</i>] [ <b>-e</b> <i>end</i>]
      [<b>-q</b> <i>minQ</i>] <i>in.cns</i> &gt; <i>out.win</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Extract information averaged in a tilling window. The output is TAB
      delimited, which consists of reference name, coordinate divided by
      1,000,000, SNP rate, het rate, raw read depth, read depth in approximately
      unique regions, the average number of hits of reads in the window and
      percent GC.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b> <i>INT</i></dt>
  <dd class="It-tag">Size of a window [1000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b> <i>STR</i></dt>
  <dd class="It-tag">Destinated reference sequence; otherwise all references
      will be used [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b> <i>INT</i></dt>
  <dd class="It-tag">Start position, 0 for no constraint [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b> <i>INT</i></dt>
  <dd class="It-tag">End position, 0 for no constraint [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum consensus quality of the sites to be used [0]</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<div>&#x00A0;</div>
<div class="Pp"></div>
<b>Simulation Related</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>fakemut</b></dt>
  <dd class="It-tag"><b>maq</b> <b>fakemut</b> [<b>-r</b> <i>mutrate</i>]
      [<b>-R</b> <i>indelfrac</i>] <i>in.ref.fasta</i> &gt;
      <i>out.fakeref.fasta</i> 2&gt; <i>out.fake.snp</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Randomly introduce substitutions and indels to the reference. Substitutions
      and sinlge base-pair indels can be added.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> <i>FLOAT</i></dt>
  <dd class="It-tag">Mutation rate [0.001]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b> <i>FLOAT</i></dt>
  <dd class="It-tag">Fraction of mutations to be indels [0.1]</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>simutrain</b></dt>
  <dd class="It-tag"><b>maq</b> <b>simutrain</b> <i>out.simupars.dat</i>
      <i>in.read.fastq</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Estimate/train parameters for read simulation.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>simulate</b></dt>
  <dd class="It-tag"><b>maq</b> <b>simulate</b> [<b>-d</b> <i>insize</i>]
      [<b>-s</b> <i>stdev</i>] [<b>-N</b> <i>nReads</i>] [ <b>-1</b>
      <i>readLen1</i>] [ <b>-2</b> <i>readLen2</i>] [<b>-r</b> <i>mutRate</i>]
      [<b>-R</b> <i>indelFrac</i>] [<b>-h</b>] <i>out.read1.fastq</i>
      <i>out.read2.fastq</i> <i>in.ref.fasta</i> <i>in.simupars.dat</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Simulate paired end reads. File <i>in.simupars.dat</i> determines the read
      lengths and quality distribution. It is generated from <b>simutrain</b>,
      or can be downloaded from Maq website. In the output read files, a read
      name consists of the reference sequence name and the outer coordinates of
      the pair of simulated reads. By default, <b>simulate</b> assumes reads
      come from a diploid sequence which is generated by adding two different
      sets of mutations, including one base-pair indels, to <i>in.ref.fasta</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> <i>INT</i></dt>
  <dd class="It-tag">mean of the outer distance of insert sizes [170]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> <i>INT</i></dt>
  <dd class="It-tag">standard deviation of insert sizes [20]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b> <i>INT</i></dt>
  <dd class="It-tag">number of pairs of reads to be generated [1000000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b> <i>INT</i></dt>
  <dd class="It-tag">length of the first read [set by
    <i>in.simupars.dat</i>]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-2</b> <i>INT</i></dt>
  <dd class="It-tag">length of the second read [set by
    <i>in.simupars.dat</i>]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b> <i>FLOAT</i></dt>
  <dd class="It-tag">mutation rate [0.001]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b> <i>FLOAT</i></dt>
  <dd class="It-tag">fraction of 1bp indels [0.1]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">add all mutations to <i>in.ref.fasta</i> and generate reads
      from the single mutated sequence (haploid mode)</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<b>NOTE:</b>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">Reads generated from this command are independent, which
      deviates from the truth. Whereas alignment evaluation is less affected by
      this, evaluation on SNP calling should be performed with caution. Error
      dependency may be one of the major causes of wrong SNP calls.</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>simustat</b></dt>
  <dd class="It-tag"><b>maq</b> <b>simustat</b> <i>in.simu-aln.map</i> &gt;
      <i>out.simustat</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Evaluate mapping qualities from simulated reads.
    <div>&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
<b>SOLiD Related</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>fasta2csfa</b></dt>
  <dd class="It-tag"><b>maq</b> <b>fasta2csfa</b> <i>in.nucl-ref.fasta</i> &gt;
      <i>out.colour-ref.fasta</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert nucleotide FASTA to colour-coded FASTA. Flag <b>-c</b> should be
      then applied to <b>map</b> command. In the output, letter `A' stands for
      color 0, `C' for 1, `G' for 2 and `T' for 3. Each sequence in the output
      is 1bp shorter than the input.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>csmap2nt</b></dt>
  <dd class="It-tag"><b>maq</b> <b>csmap2nt</b> <i>out.nt.map</i>
      <i>in.ref.nt.bfa</i> <i>in.cs.map</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert color alignment to nucleotide alignment. The input
      <i>in.ref.nt.bfa</i> is the nucleotide binary FASTA reference file. It
      must correspond to the original file from which the color reference is
      converted. Nucleotide consensus can be called from the resultant
      alignment.
    <div>&#x00A0;</div>
  </dd>
</dl>
<div class="Pp"></div>
<b>Miscellaneous/Advanced Commands</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>submap</b></dt>
  <dd class="It-tag"><b>maq</b> <b>submap</b> [<b>-q</b> <i>minMapQ</i>]
      [<b>-Q</b> <i>maxSumErr</i>] [<b>-m</b> <i>maxMM</i>] [ <b>-p</b>]
      <i>out.map</i> <i>in.map</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Filter bad alignments in <i>in.map</i>. Command-line options are described
      in the ` <b>assemble</b>' command.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>eland2maq</b></dt>
  <dd class="It-tag"><b>maq</b> <b>eland2maq</b> [<b>-q</b> <i>defqual</i>]
      <i>out.map</i> <i>in.list</i> <i>in.eland</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert eland alignment to maq's .map format. File <i>in.list</i> consists
      of the sequence names that appear at the seventh column of the eland
      alignment file <i>in.eland</i> and the name you expect to see in maq
      alignment. The following is an example:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
  cX.fa chrX
  c1.fa chr1
  c2.fa chr2
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    If you are aligning reads in several batches using eland, it is important to
      use the same <i>in.list</i> for the conversion. In addition, maq will load
      all the alignments and sort them in the memory. If you have concatenate
      several eland outputs into one huge file, you should separate it into
      smaller files to prevent maq from eating all your machine memory.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command actually aims to show Eland alignment in Maqview. As no quality
      information is available, the resultant maq alignment file should not be
      used to call consensus genotypes.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>export2maq</b></dt>
  <dd class="It-tag"><b>maq</b> <b>export2maq</b> [<b>-1</b> <i>read1len</i>]
      [<b>-2</b> <i>read2len</i>] [<b>-a</b> <i>maxdist</i>] [<b>-n</b>]
      <i>out.map</i> <i>in.list</i> <i>in.export</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert Illumina's Export format to Maq's <i>.map</i> format. Export format
      is a new alignment format since SolexaPipeline-0.3.0 which also calculates
      mapping qualities like maq. The resultant file can be used to call
      consensus genotypes as most of necessary information is available for maq
      to do this accurately.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b> <i>INT</i></dt>
  <dd class="It-tag">Length of the first read [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-2</b> <i>INT</i></dt>
  <dd class="It-tag">Length of the second read [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum outer distance for a correct read pair [250]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">Retain filtered reads</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<h1 class="Sh" title="Sh" id="MAQ-PERL_COMMANDS"><a class="selflink" href="#MAQ-PERL_COMMANDS">MAQ-PERL
  COMMANDS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>demo</b></dt>
  <dd class="It-tag"><b>maq.pl</b> <b>demo</b> [<b>-h</b>] [<b>-s</b>]
      [<b>-N</b> <i>nPairs</i>] [<b>-d</b> <i>outDir</i>] <i>in.fasta</i>
      <i>in.simudat</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Demonstrate the use of <b>maq</b> and its companion scripts. This command
      will simulate reads from a FASTA file <i>in.fasta</i>. The sequence length
      and qualities are determined by <i>in.simudat</i> which is generated from
      <b>maq simutrain</b> or can be downloaded from Maq website. The simulated
      reads will then be mapped with <b>maq.pl easyrun</b>. The alignment
      accuracy is evaluated by <b>maq simustat</b>, the consensus accuracy by
      <b>maq</b> <b>simucns</b>, and the SNP accuracy by <b>maq_eval.pl</b>.
    <div style="height: 1.00em;">&#x00A0;</div>
    By default, paired end reads will be simulated and a diploid sequence will
      be generated from the input by adding mutations to either haploid type.
      The insert size and mutation rate are controlled by <b>maq</b>
      <b>simulate</b>.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">simulate a haploid sequence instead of a diploid
    sequence</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">use single-end mode to align reads instead of paired-end
      mode</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b> <i>INT</i></dt>
  <dd class="It-tag">number of pairs of reads to be simulated [1000000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> <i>DIR</i></dt>
  <dd class="It-tag">output directory [maqdemo]</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<b>NOTE:</b>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">The output files from <b>maq_eval.pl</b> have not been
      documented, but you may make a good guess at some of these files.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">This command just demonstrates the use of the maq suite.
      The accuracy on real data is almost always lower than what you see from
      pure simulation.</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>easyrun</b></dt>
  <dd class="It-tag"><b>maq.pl</b> <b>easyrun</b> [<b>-1</b> <i>read1Len</i>]
      [<b>-d</b> <i>out.dir</i>] [<b>-n</b> <i>nReads</i>] [<b>-A</b>
      <i>3adapter</i>] [ <b>-e</b> <i>minDep</i>] [<b>-q</b> <i>minCnsQ</i>] [
      <b>-p</b>] [<b>-2</b> <i>read2Len</i>] [<b>-a</b> <i>maxIns</i>]
      [<b>-S</b>] [ <b>-N</b>] <i>in.ref.fasta</i> <i>in1.fastq</i>
      [<i>in2.fastq</i>]
    <div style="height: 1.00em;">&#x00A0;</div>
    Analyses pipeline for small genomes. Easyrun command will run most of
      analyses implemented in <b>maq</b>. By default, <b>easyrun</b> assumes all
      the input read sequences files are single-end and independent; when
      <b>-p</b> is specified, two read sequence files are required, one for each
      end.
    <div style="height: 1.00em;">&#x00A0;</div>
    Several files will be generated in <i>out.dir</i>, among which the following
      files are the key output:</dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><i>cns.final.snp</i></dt>
  <dd class="It-tag">final SNP calls with low quality ones filtered out</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>cns.fq</i></dt>
  <dd class="It-tag">consensus sequences and qualities in the FASTQ format</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<b>OPTIONS:</b>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> <i>DIR</i></dt>
  <dd class="It-tag">output directory [easyrun]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> <i>INT</i></dt>
  <dd class="It-tag">number of reads/pairs in one batch of alignment
    [2000000]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">apply split-read analysis of short indels (maybe very
    slow)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b> <i>INT</i></dt>
  <dd class="It-tag">number of haplotypes/strains in the pool (&gt;=2) [2]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b> <i>FILE</i></dt>
  <dd class="It-tag">file for 3'-adapter. The file should contain a single line
      of sequence [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b> <i>INT</i></dt>
  <dd class="It-tag">length of the first read, 0 for auto [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b> <i>INT</i></dt>
  <dd class="It-tag">minimum read depth required to call a SNP (for SNPfilter)
      [3]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> <i>INT</i></dt>
  <dd class="It-tag">minimum consensus quality for SNPs in <i>cns.final.snp</i>
      [30]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b></dt>
  <dd class="It-tag">switch to paired end alignment mode</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-2</b> <i>INT</i></dt>
  <dd class="It-tag">length of the second read when <b>-p</b> is applied
    [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b> <i>INT</i></dt>
  <dd class="It-tag">maximum insert size when <b>-p</b> is applied [250]</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;">
<div style="height: 1.00em;">&#x00A0;</div>
<b>NOTES:</b>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">For SNP calling on pooled samples, users should set correct
      ` <b>-N</b>' as well as ` <b>-E</b> 0'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">*</dt>
  <dd class="It-tag">The input file can be maq's binary format. <b>maq.pl</b>
      will automatically detect the file format.</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>SNPfilter</b></dt>
  <dd class="It-tag"><b>maq.pl</b> <b>SNPfilter</b> [<b>-d</b> <i>minDep</i>]
      [<b>-D</b> <i>maxDep</i>] [<b>-Q</b> <i>maxMapQ</i>] [<b>-q</b>
      <i>minCnsQ</i>] [ <b>-w</b> <i>indelWinSize</i>] [<b>-n</b>
      <i>minNeiQ</i>] [ <b>-F</b> <i>in.indelpe</i>] [<b>-f</b>
      <i>in.indelsoa</i>] [ <b>-s</b> <i>minScore</i>] [<b>-m</b>
      <i>maxAcross</i>] [ <b>-a</b>] [<b>-N</b> <i>maxWinSNP</i>] [<b>-W</b>
      <i>densWinSize</i>] <i>in.cns2snp.snp</i> &gt; <i>out.filtered.snp</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Rule out SNPs that are covered by few reads (specified by <b>-d</b>), by too
      many reads (specified by <b>-D</b>), near (specified by <b>-w</b>) to a
      potential indel, falling in a possible repetitve region (characterized by
      <b>-Q</b>), or having low-quality neighbouring bases (specified by
      <b>-n</b>). If <i>maxWinSNP</i> or more SNPs appear in any
      <i>densWinSize</i> window, they will also be filtered out together.
    <div style="height: 1.00em;">&#x00A0;</div>
     <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 11.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum read depth required to call a SNP [3]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum read depth required to call a SNP (&lt;255,
      otherwise ignored) [256]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b> <i>INT</i></dt>
  <dd class="It-tag">Required maximum mapping quality of reads covering the SNP
      [40]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum consensus quality [20]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum adjacent consensus quality [20]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b> <i>INT</i></dt>
  <dd class="It-tag">Size of the window around the potential indels. SNPs that
      are close to indels will be suppressed [3]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b> <i>FILE</i></dt>
  <dd class="It-tag">The <b>indelpe</b> output [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b> <i>FILE</i></dt>
  <dd class="It-tag">The <b>indelsoa</b> output [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b> <i>INT</i></dt>
  <dd class="It-tag">Minimum score for a soa-indel to be considered [3]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b> <i>INT</i></dt>
  <dd class="It-tag">Maximum number of reads that can be mapped across a
      soa-indel [1]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-a</b></dt>
  <dd class="It-tag">Alternative filter for single end alignment</dd>
</dl>
</div>
<div style="margin-left: 11.00ex;"></div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>indelpe</b></dt>
  <dd class="It-tag"><b>maq.pl</b> <b>indelpe</b> <i>in.indelpe</i> &gt;
      <i>out.indelpe</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Correct the number of reads mapped without indels for homopolymer tracts.
      This command modify the 4th, 10th and the last three columns of
      <i>in.indelpe</i> and output the result in <i>out.indelpe</i>. After the
      correction, the following <b>awk</b> command gives putative homozygous
      indels:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
  awk '($3==&quot;*&quot;&#x23AA;&#x23AA;$3==&quot;+&quot;) &amp;&amp; $6+$7&gt;=3 &amp;&amp; ($6+$7)/$4&gt;=0.75'
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    and the following gives heterozygotes:
    <div style="height: 1.00em;">&#x00A0;</div>
    <pre>
  awk '($3==&quot;*&quot;&#x23AA;&#x23AA;$3==&quot;+&quot;) &amp;&amp; $6+$7&gt;=3 &amp;&amp; ($6+$7)/$4&lt;0.75'
    </pre>
    <div style="height: 1.00em;">&#x00A0;</div>
    Please note that this <b>indelpe</b> command just implements several
      heuristic rules. It does not correct for impure homopolymer runs or
      di-nucleotide/triplet repeats. Consequently, the two awk commands only
      give approximate hom/het indels.</dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Easyrun script:
    <br/>
     maq.pl easyrun -d easyrun ref.fasta part1.fastq part2.fastq</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">&#x2022;</dt>
  <dd class="It-tag">Key commands behind easyrun:
    <br/>
     maq fasta2bfa ref.fasta ref.bfa;
    <br/>
     maq fastq2bfq part1.fastq part1.bfq;
    <br/>
     maq fastq2bfq part2.fastq part2.bfq;
    <br/>
     maq map part1.map ref.bfa part1.bfq;
    <br/>
     maq map part2.map ref.bfa part2.bfq;
    <br/>
     maq mapmerge aln.map part1.map part2.map;
    <br/>
     maq assemble cns.cns ref.bfa aln.map;</dd>
</dl>
<h1 class="Sh" title="Sh" id="LICENSE"><a class="selflink" href="#LICENSE">LICENSE</a></h1>
GNU General Public License, version 3 (GPLv3)
<h1 class="Sh" title="Sh" id="AVAILABILITY"><a class="selflink" href="#AVAILABILITY">AVAILABILITY</a></h1>
&lt;http://maq.sourceforge.net&gt;
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Heng Li &lt;lh3@sanger.ac.uk&gt;</div>
<table class="foot">
  <tr>
    <td class="foot-date">2008-09-22</td>
    <td class="foot-os">maq-0.7.1</td>
  </tr>
</table>
</body>
</html>
