<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:55 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MAQ(1) Bioinformatics Tools MAQ(1)</p>

<p style="margin-top: 1em">NAME <br>
Maq - Mapping and Assembly with Qualities</p>

<p style="margin-top: 1em">SYNOPSIS <br>
maq command [options] arguments</p>

<p style="margin-top: 1em">maq.pl command [options]
arguments</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Maq is a software that builds mapping assemblies from short
reads generated by the next-generation sequencing machines.
It is particularly designed for Illumina-Solexa 1G Genetic
<br>
Analyzer, and has a preliminary functionality to handle AB
SOLiD data.</p>

<p style="margin-top: 1em">With Maq you can:</p>

<p style="margin-top: 1em">&Acirc;&middot; Fast align
Illumina/SOLiD reads to the reference genome. With the
default options, one million pairs of reads can be mapped to
the human genome in about 10 CPU hours with less <br>
than 1G memory.</p>

<p style="margin-top: 1em">&Acirc;&middot; Accurately
measure the error probability of the alignment of each
individual read.</p>

<p style="margin-top: 1em">&Acirc;&middot; Call the
consensus genotypes, including homozygous and heterozygous
polymorphisms, with a Phred probabilistic quality assigned
to each base.</p>

<p style="margin-top: 1em">&Acirc;&middot; Find short
indels with paired end reads.</p>

<p style="margin-top: 1em">&Acirc;&middot; Accurately find
large scale genomic deletions and translocations with paired
end reads.</p>

<p style="margin-top: 1em">&Acirc;&middot; Discover
potential CNVs by checking read depth.</p>

<p style="margin-top: 1em">&Acirc;&middot; Evaluate the
accuracy of raw base qualities from sequencers and help to
check the systematic errors.</p>

<p style="margin-top: 1em">However, Maq can NOT:</p>

<p style="margin-top: 1em">&Acirc;&middot; Do de novo
assembly. (Maq can only call the consensus by mapping reads
to a known reference.)</p>

<p style="margin-top: 1em">&Acirc;&middot; Map shorts reads
against themselves. (Maq can only find complete overlap
between reads.)</p>

<p style="margin-top: 1em">&Acirc;&middot; Align capillary
reads or 454 reads to the reference. (Maq cannot align reads
longer than 63bp.)</p>

<p style="margin-top: 1em">MAQ COMMANDS <br>
Key Commands</p>

<p style="margin-top: 1em">fasta2bfa maq fasta2bfa
in.ref.fasta out.ref.bfa</p>

<p style="margin-top: 1em">Convert sequences in FASTA
format to Maq&rsquo;s BFA (binary FASTA) format.</p>

<p style="margin-top: 1em">fastq2bfq maq fastq2bfq [-n
nreads] in.read.fastq
out.read.bfq&acirc;&ordf;out.prefix</p>

<p style="margin-top: 1em">Convert reads in FASTQ format to
Maq&rsquo;s BFQ (binary FASTQ) format.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-n INT number of reads per file
[not specified]</p>

<p style="margin-top: 1em">map maq map [-n nmis] [-a
maxins] [-c] [-1 len1] [-2 len2] [-d adap3] [-m mutrate] [-u
unmapped] [-e maxerr] [-M c&acirc;&ordf;g] [-N] [-H allhits]
[-C maxhits] out.aln.map in.ref.bfa <br>
in.read1.bfq [in.read2.bfq] 2&gt; out.map.log</p>

<p style="margin-top: 1em">Map reads to the reference
sequences.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-n INT Number of maximum
mismatches that can always be found [2]</p>

<p style="margin-top: 1em">-a INT Maximum outer distance
for a correct read pair [250]</p>

<p style="margin-top: 1em">-A INT Maximum outer distance of
two RF paied read (0 for disable) [0]</p>

<p style="margin-top: 1em">-c Map reads in the colour space
(for SOLiD only)</p>

<p style="margin-top: 1em">-1 INT Read length for the first
read, 0 for auto [0]</p>

<p style="margin-top: 1em">-2 INT Read length for the
second read, 0 for auto [0]</p>

<p style="margin-top: 1em">-m FLOAT Mutation rate between
the reference sequences and the reads [0.001]</p>

<p style="margin-top: 1em">-d FILE Specify a file
containing a single line of the 3&rsquo;-adapter sequence
[null]</p>

<p style="margin-top: 1em">-u FILE Dump unmapped reads and
reads containing more than nmis mismatches to a separate
file [null]</p>

<p style="margin-top: 1em">-e INT Threshold on the sum of
mismatching base qualities [70]</p>

<p style="margin-top: 1em">-H FILE Dump multiple/all
01-mismatch hits to FILE [null]</p>

<p style="margin-top: 1em">-C INT Maximum number of hits to
output. Unlimited if larger than 512. [250]</p>

<p style="margin-top: 1em">-M c&acirc;&ordf;g methylation
alignment mode. All C (or G) on the forward strand will be
changed to T (or A). This option is for testing only.</p>

<p style="margin-top: 1em">-N store the mismatch position
in the output file out.aln.map. When this option is in use,
the maximum allowed read length is 55bp.</p>

<p style="margin-top: 1em">NOTE:</p>

<p style="margin-top: 1em">* Paired end reads should be
prepared in two files, one for each end, with reads are
sorted in the same order. This means the k-th read in the
first file is mated with <br>
the k-th read in the second file. The corresponding read
names must be identical up to the tailing &lsquo;/1&rsquo;
or &lsquo;/2&rsquo;. For example, such a pair of read names
are allowed: <br>
&lsquo;EAS1_1_5_100_200/1&rsquo; and
&lsquo;EAS1_1_5_100_200/2&rsquo;. The tailing
&lsquo;/[12]&rsquo; is usually generated by the GAPipeline
to distinguish the two ends in a pair.</p>

<p style="margin-top: 1em">* The output is a compressed
binary file. It is affected by the endianness.</p>

<p style="margin-top: 1em">* The best way to run this
command is to provide about 1 to 3 million reads as input.
More reads consume more memory.</p>

<p style="margin-top: 1em">* Option -n controls the
sensitivity of the alignment. By default, a hit with up to 2
mismatches can be always found. Higher -n finds more hits
and also improves the <br>
accuracy of mapping qualities. However, this is done at the
cost of speed.</p>

<p style="margin-top: 1em">* Alignments with many
high-quality mismatches should be discarded as false
alignments or possible contaminations. This behaviour is
controlled by option -e. The -e <br>
threshold is only calculated approximately because base
qualities are divided by 10 at a certain stage of the
alignment. The -Q option in the assemble command pre&acirc;
<br>
cisely set the threshold.</p>

<p style="margin-top: 1em">* A pair of reads are said to be
correctly paired if and only if the orientation is FR and
the outer distance of the pair is no larger than maxins.
There is no limit <br>
on the minimum insert size. This setting is determined by
the paired end alignment algorithm used in Maq. Requiring a
minimum insert size will lead to some wrong <br>
alignments with highly overestimated mapping qualities.</p>

<p style="margin-top: 1em">* Currently, read pairs from
Illumina/Solexa long-insert library have RF read
orientation. The maximum insert size is set by option -A.
However, long-insert library is <br>
also mixed with a small fraction of short-insert read pairs.
-a should also be set correctly.</p>

<p style="margin-top: 1em">* Sometimes 5&rsquo;-end or even
the entire 3&rsquo;-adapter sequence may be sequenced.
Providing -d renders Maq to eliminate the adapter
contaminations.</p>

<p style="margin-top: 1em">* Given 2 million reads as
input, maq usually takes 800MB memory.</p>

<p style="margin-top: 1em">mapmerge maq mapmerge
out.aln.map in.aln1.map in.aln2.map [...]</p>

<p style="margin-top: 1em">Merge a batch of read alignments
together.</p>

<p style="margin-top: 1em">NOTE:</p>

<p style="margin-top: 1em">* In theory, this command can
merge unlimited number of alignments. However, as mapmerge
will be reading all the inputs at the same time, it may hit
the limit of the <br>
maximum number of opening files set by the OS. At present,
this has to be manually solved by endusers.</p>

<p style="margin-top: 1em">* Command mapmerge can be used
to merge alignment files with different read lengths. All
the subsequent analyses do not assume fixed length any
more.</p>

<p style="margin-top: 1em">rmdup maq rmdup out.rmdup.map
in.ori.map</p>

<p style="margin-top: 1em">Remove pairs with identical
outer coordinates. In principle, pairs with identical outer
coordinates should happen rarely. However, due to the
amplification in sample <br>
preparation, this occurs much more frequently than by
chance. Practical analyses show that removing duplicates
helps to improve the overall accuracy of SNP calling.</p>

<p style="margin-top: 1em">assemble maq assemble [-sp] [-m
maxmis] [-Q maxerr] [-r hetrate] [-t coef] [-q minQ] [-N
nHap] out.cns in.ref.bfa in.aln.map 2&gt; out.cns.log</p>

<p style="margin-top: 1em">Call the consensus sequences
from read mapping.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-t FLOAT Error dependency
coefficient [0.93]</p>

<p style="margin-top: 1em">-r FLOAT Fraction of
heterozygotes among all sites [0.001]</p>

<p style="margin-top: 1em">-s Take single end mapping
quality as the final mapping quality; otherwise paired end
mapping quality will be used</p>

<p style="margin-top: 1em">-p Discard paired end reads that
are not mapped in correct pairs</p>

<p style="margin-top: 1em">-m INT Maximum number of
mismatches allowed for a read to be used in consensus
calling [7]</p>

<p style="margin-top: 1em">-Q INT Maximum allowed sum of
quality values of mismatched bases [60]</p>

<p style="margin-top: 1em">-q INT Minimum mapping quality
allowed for a read to be used in consensus calling [0]</p>

<p style="margin-top: 1em">-N INT Number of haplotypes in
the pool (&gt;=2) [2]</p>

<p style="margin-top: 1em">NOTE:</p>

<p style="margin-top: 1em">* Option -Q sets a limit on the
maximum sum of mismatching base qualities. Reads containing
many high-quality mismatches should be discarded.</p>

<p style="margin-top: 1em">* Option -N sets the number of
haplotypes in a pool. It is designed for resequencing of
samples by pooling multiple strains/individuals together.
For diploid genome <br>
resequencing, this option equals 2.</p>

<p style="margin-top: 1em">glfgen maq glfgen [-sp] [-m
maxmis] [-Q maxerr] [-r hetrate] [-t coef] [-q minQ] [-N
nHap] out.cns in.ref.bfa in.aln.map 2&gt; out.cns.log</p>

<p style="margin-top: 1em">Calculate log-likelihood for all
genotypes and store the results in GLF format (Genotyping
Likelihood Format). Please check MAQ website for detailed
descriptions of <br>
the file format and the related utilities.</p>

<p style="margin-top: 1em">indelpe maq indelpe in.ref.bfa
in.aln.map &gt; out.indelpe</p>

<p style="margin-top: 1em">Call consistent indels from
paired end reads. The output is TAB delimited with each line
consisting of chromosome, start position, type of the indel,
number of reads <br>
across the indel, size of the indel and inserted/deleted
nucleotides (separated by colon), number of indels on the
reverse strand, number of indels on the forward <br>
strand, 5&rsquo; sequence ahead of the indel, 3&rsquo;
sequence following the indel, number of reads aligned
without indels and three additional columns for filters.</p>

<p style="margin-top: 1em">At the 3rd column, type of the
indel, a star indicates the indel is confirmed by reads from
both strands, a plus means the indel is hit by at least two
reads but from <br>
the same strand, a minus shows the indel is only found on
one read, and a dot means the indel is too close to another
indel and is filtered out.</p>

<p style="margin-top: 1em">Users are recommended to run
through &lsquo;maq.pl indelpe&rsquo; to correct the number
of reads mapped without indels. For more details, see the
&lsquo;maq.pl indelpe&rsquo; section.</p>

<p style="margin-top: 1em">indelsoa maq indelsoa in.ref.bfa
in.aln.map &gt; out.indelsoa</p>

<p style="margin-top: 1em">Call potential homozygous indels
and break points by detecting the abnormal alignment pattern
around indels and break points. The output is also TAB
delimited with <br>
each line consisting of chromosome, approximate coordinate,
length of the abnormal region, number of reads mapped across
the position, number of reads on the left-hand <br>
side of the position and number of reads on the right-hand
side. The last column can be ignored.</p>

<p style="margin-top: 1em">The output contains many false
positives. A recommended filter could be:</p>

<p style="margin-top: 1em">awk &rsquo;$5+$6-$4 &gt;= 3
&amp;&amp; $4 &lt;= 1&rsquo; in.indelsoa</p>

<p style="margin-top: 1em">Note that this command does not
aim to be an accurate indel detector, but mainly helps to
avoid some false positives in substitution calling. In
addition, it only <br>
works well given deep depth (~40X for example); otherwise
the false negative rate would be very high.</p>

<p style="margin-top: 1em">Format Converting</p>

<p style="margin-top: 1em">sol2sanger maq sol2sanger
in.sol.fastq out.sanger.fastq</p>

<p style="margin-top: 1em">Convert Solexa FASTQ to
standard/Sanger FASTQ format.</p>

<p style="margin-top: 1em">bfq2fastq maq bfq2fastq
in.read.bfq out.read.fastq</p>

<p style="margin-top: 1em">Convert Maq&rsquo;s BFQ format
to standard FASTQ format.</p>

<p style="margin-top: 1em">mapass2maq maq mapass2maq
in.mapass2.map out.maq.map</p>

<p style="margin-top: 1em">Convert obsolete mapass2&rsquo;s
map format to Maq&rsquo;s map format. The old format does
not contain read names.</p>

<p style="margin-top: 1em">Information Extracting</p>

<p style="margin-top: 1em">mapview maq mapview [-bN]
in.aln.map &gt; out.aln.txt</p>

<p style="margin-top: 1em">Display the read alignment in
plain text. For reads aligned before the Smith-Waterman
alignment, each line consists of read name, chromosome,
position, strand, insert <br>
size from the outer coorniates of a pair, paired flag,
mapping quality, single-end mapping quality, alternative
mapping quality, number of mismatches of the best hit, <br>
sum of qualities of mismatched bases of the best hit, number
of 0-mismatch hits of the first 24bp, number of 1-mismatch
hits of the first 24bp on the reference, length <br>
of the read, read sequence and its quality. Alternative
mapping quality always equals to mapping quality if the
reads are not paired. If reads are paired, it equals <br>
to the smaller mapping quality of the two ends. This
alternative mapping quality is actually the mapping quality
of an abnormal pair.</p>

<p style="margin-top: 1em">The fifth column, paired flag,
is a bitwise flag. Its lower 4 bits give the orientation: 1
stands for FF, 2 for FR, 4 for RF, and 8 for RR, where FR
means that the <br>
read with smaller coordinate is on the forward strand, and
its mate is on the reverse strand. Only FR is allowed for a
correct pair. The higher bits of this flag give <br>
further information. If the pair meets the paired end
requirement, 16 will be set. If the two reads are mapped to
different chromosomes, 32 will be set. If one of the <br>
two reads cannot be mapped at all, 64 will be set. The flag
for a correct pair always equals to 18.</p>

<p style="margin-top: 1em">For reads aligned by the
Smith-Waterman alignment afterwards, the flag is always 130.
A line consists of read name, chromosome, position, strand,
insert size, flag <br>
(always 130), position of the indel on the read (0 if no
indel), length of the indels (positive for insertions and
negative for deletions), mapping quality of its <br>
mate, number of mismatches of the best hit, sum of qualities
of mismatched bases of the best hit, two zeros, length of
the read, read sequence and its quality. The <br>
mate of a 130-flagged read always gets a flag 18.</p>

<p style="margin-top: 1em">Flag 192 indicates that the read
is not mapped but its mate is mapped. For such a read pair,
one read has flag 64 and the other has 192.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-b do not display the read
sequence and the quality</p>

<p style="margin-top: 1em">-N display the positions where
mismatches occur. This flag only works with a .map file
generated by &lsquo;maq map -N&rsquo;.</p>

<p style="margin-top: 1em">mapcheck maq mapcheck [-s] [-m
maxmis] [-q minQ] in.ref.bfa in.aln.map &gt;
out.mapcheck</p>

<p style="margin-top: 1em">Read quality check. The mapcheck
first reports the composition and the depth of the
reference. After that there is a form. The first column
indicates the position on a <br>
read. Following four columns which show the nucleotide
composition, substitution rates between the reference and
reads will be given. These rates and the numbers in <br>
the following columns are scaled to 999 and rounded to
nearest integer. The next group of columns show the
distribution of base qualities along the reads at a quality
<br>
interval of 10. A decay in quality can usually be observed,
which means bases at the end of read are less accurate. The
last group of columns present the fraction of <br>
substitutions for read bases at a quality interval. This
measures the accuracy of base quality estimation. Idealy, we
expect to see 1 in the 3? column, 10 in the 2? <br>
column and 100 in the 1? column.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-s Take single end mapping
quality as the final mapping quality</p>

<p style="margin-top: 1em">-m INT Maximum number of
mismatahces allowed for a read to be counted [4]</p>

<p style="margin-top: 1em">-q INT Minimum mapping quality
allowed for a read to be counted [30]</p>

<p style="margin-top: 1em">pileup maq pileup [-spvP] [-m
maxmis] [-Q maxerr] [-q minQ] [-l sitefile] in.ref.bfa
in.aln.map &gt; out.pileup</p>

<p style="margin-top: 1em">Display the alignment in a
&lsquo;pileup&rsquo; text format. Each line consists of
chromosome, position, reference base, depth and the bases on
reads that cover this position. If <br>
-v is added on the command line, base qualities and mapping
qualities will be presented in the sixth and seventh columns
in order.</p>

<p style="margin-top: 1em">The fifth column always starts
with &lsquo;@&rsquo;. In this column, read bases identical
to the reference are showed in comma &lsquo;,&rsquo; or dot
&lsquo;.&rsquo;, and read bases different from the <br>
reference in letters. A comma or a upper case indicates that
the base comes from a read aligned on the forward strand,
while a dot or a lower case on the reverse <br>
strand.</p>

<p style="margin-top: 1em">This command is for users who
want to develop their own SNP callers.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-s Take single end mapping
quality as the final mapping quality</p>

<p style="margin-top: 1em">-p Discard paired end reads that
are not mapped as correct pairs</p>

<p style="margin-top: 1em">-v Output verbose information
including base qualities and mapping qualities</p>

<p style="margin-top: 1em">-m INT Maximum number of
mismatches allowed for a read to be used [7]</p>

<p style="margin-top: 1em">-Q INT Maximum allowed number of
quality values of mismatches [60]</p>

<p style="margin-top: 1em">-q INT Minimum mapping quality
allowed for a read to be used [0]</p>

<p style="margin-top: 1em">-l FILE File containing the
sites at which pileup will be printed out. In this file the
first column gives the names of the reference and the second
the coordinates. <br>
Additional columns will be ignored. [null]</p>

<p style="margin-top: 1em">-P also output the base position
on the read</p>

<p style="margin-top: 1em">cns2fq maq cns2fq [-Q minMapQ]
[-n minNeiQ] [-d minDepth] [-D maxDepth] in.cns &gt;
out.cns.fastq</p>

<p style="margin-top: 1em">Extract the consensus sequences
in FASTQ format. In the sequence lines, bases in lower case
are essentially repeats or do not have sufficient coverage;
bases in upper <br>
case indicate regions where SNPs can be reliably called. In
the quality lines, ASCII of a character minus 33 gives the
PHRED quality.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-Q INT Minimum mapping quality
[40]</p>

<p style="margin-top: 1em">-d INT Minimum read depth
[3]</p>

<p style="margin-top: 1em">-n INT Minimum neighbouring
quality [20]</p>

<p style="margin-top: 1em">-D INT Maximum read dpeth.
&gt;=255 for unlimited. [255]</p>

<p style="margin-top: 1em">cns2snp maq cns2snp in.cns &gt;
out.snp</p>

<p style="margin-top: 1em">Extract SNP sites. Each line
consists of chromosome, position, reference base, consensus
base, Phred-like consensus quality, read depth, the average
number of hits of <br>
reads covering this position, the highest mapping quality of
the reads covering the position, the minimum consensus
quality in the 3bp flanking regions at each side of <br>
the site (6bp in total), the second best call, log
likelihood ratio of the second best and the third best call,
and the third best call.</p>

<p style="margin-top: 1em">The 5th column is the key
criterion when you judge the reliability of a SNP. However,
as this quality is only calculated assuming site
independency, you should also <br>
consider other columns to get more accurate SNP calls.
Script command &lsquo;maq.pl SNPfilter&rsquo; is designed
for this (see below).</p>

<p style="margin-top: 1em">The 7th column implies whether
the site falls in a repetitive region. If no read covering
the site can be mapped with high mapping quality, the
flanking region is pos&acirc; <br>
sibly repetitive or in the lack of good reads. A SNP at such
site is usually not reliable.</p>

<p style="margin-top: 1em">The 8th column roughly gives the
copy number of the flanking region in the reference genome.
In most cases, this number approaches 1.00, which means the
region is <br>
about unique. Sometimes you may see non-zero read depth but
0.00 at the 7th column. This indicates that all the reads
covering the position have at least two mis&acirc; <br>
matches. Maq only counts the number of 0- and 1-mismatch
hits to the reference. This is due to a complex technical
issue.</p>

<p style="margin-top: 1em">The 9th column gives the
neighbouring quality. Filtering on this column is also
required to get reliable SNPs. This idea is inspired by NQS,
although NQS is initially <br>
designed for a single read instead of a consensus.</p>

<p style="margin-top: 1em">cns2view maq cns2view in.cns
&gt; out.view</p>

<p style="margin-top: 1em">Show detailed information at all
sites. The output format is identical to cns2snp report.</p>

<p style="margin-top: 1em">cns2ref maq cns2ref in.cns &gt;
out.ref.fasta</p>

<p style="margin-top: 1em">Extract the reference
sequence.</p>

<p style="margin-top: 1em">cns2win maq cns2win [-w winsize]
[-c chr] [-b begin] [-e end] [-q minQ] in.cns &gt;
out.win</p>

<p style="margin-top: 1em">Extract information averaged in
a tilling window. The output is TAB delimited, which
consists of reference name, coordinate divided by 1,000,000,
SNP rate, het rate, <br>
raw read depth, read depth in approximately unique regions,
the average number of hits of reads in the window and
percent GC.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-w INT Size of a window
[1000]</p>

<p style="margin-top: 1em">-c STR Destinated reference
sequence; otherwise all references will be used [null]</p>

<p style="margin-top: 1em">-b INT Start position, 0 for no
constraint [0]</p>

<p style="margin-top: 1em">-e INT End position, 0 for no
constraint [0]</p>

<p style="margin-top: 1em">-q INT Minimum consensus quality
of the sites to be used [0]</p>

<p style="margin-top: 1em">Simulation Related</p>

<p style="margin-top: 1em">fakemut maq fakemut [-r mutrate]
[-R indelfrac] in.ref.fasta &gt; out.fakeref.fasta 2&gt;
out.fake.snp</p>

<p style="margin-top: 1em">Randomly introduce substitutions
and indels to the reference. Substitutions and sinlge
base-pair indels can be added.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-r FLOAT Mutation rate
[0.001]</p>

<p style="margin-top: 1em">-R FLOAT Fraction of mutations
to be indels [0.1]</p>

<p style="margin-top: 1em">simutrain maq simutrain
out.simupars.dat in.read.fastq</p>

<p style="margin-top: 1em">Estimate/train parameters for
read simulation.</p>

<p style="margin-top: 1em">simulate maq simulate [-d
insize] [-s stdev] [-N nReads] [-1 readLen1] [-2 readLen2]
[-r mutRate] [-R indelFrac] [-h] out.read1.fastq
out.read2.fastq in.ref.fasta in.simu&acirc; <br>
pars.dat</p>

<p style="margin-top: 1em">Simulate paired end reads. File
in.simupars.dat determines the read lengths and quality
distribution. It is generated from simutrain, or can be
downloaded from Maq <br>
website. In the output read files, a read name consists of
the reference sequence name and the outer coordinates of the
pair of simulated reads. By default, simulate <br>
assumes reads come from a diploid sequence which is
generated by adding two different sets of mutations,
including one base-pair indels, to in.ref.fasta.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-d INT mean of the outer
distance of insert sizes [170]</p>

<p style="margin-top: 1em">-s INT standard deviation of
insert sizes [20]</p>

<p style="margin-top: 1em">-N INT number of pairs of reads
to be generated [1000000]</p>

<p style="margin-top: 1em">-1 INT length of the first read
[set by in.simupars.dat]</p>

<p style="margin-top: 1em">-2 INT length of the second read
[set by in.simupars.dat]</p>

<p style="margin-top: 1em">-r FLOAT mutation rate
[0.001]</p>

<p style="margin-top: 1em">-R FLOAT fraction of 1bp indels
[0.1]</p>

<p style="margin-top: 1em">-h add all mutations to
in.ref.fasta and generate reads from the single mutated
sequence (haploid mode)</p>

<p style="margin-top: 1em">NOTE:</p>

<p style="margin-top: 1em">* Reads generated from this
command are independent, which deviates from the truth.
Whereas alignment evaluation is less affected by this,
evaluation on SNP calling <br>
should be performed with caution. Error dependency may be
one of the major causes of wrong SNP calls.</p>

<p style="margin-top: 1em">simustat maq simustat
in.simu-aln.map &gt; out.simustat</p>

<p style="margin-top: 1em">Evaluate mapping qualities from
simulated reads.</p>

<p style="margin-top: 1em">SOLiD Related</p>

<p style="margin-top: 1em">fasta2csfa maq fasta2csfa
in.nucl-ref.fasta &gt; out.colour-ref.fasta</p>

<p style="margin-top: 1em">Convert nucleotide FASTA to
colour-coded FASTA. Flag -c should be then applied to map
command. In the output, letter &lsquo;A&rsquo; stands for
color 0, &lsquo;C&rsquo; for 1, &lsquo;G&rsquo; for 2 and
<br>
&lsquo;T&rsquo; for 3. Each sequence in the output is 1bp
shorter than the input.</p>

<p style="margin-top: 1em">csmap2nt maq csmap2nt out.nt.map
in.ref.nt.bfa in.cs.map</p>

<p style="margin-top: 1em">Convert color alignment to
nucleotide alignment. The input in.ref.nt.bfa is the
nucleotide binary FASTA reference file. It must correspond
to the original file from <br>
which the color reference is converted. Nucleotide consensus
can be called from the resultant alignment.</p>

<p style="margin-top: 1em">Miscellaneous/Advanced
Commands</p>

<p style="margin-top: 1em">submap maq submap [-q minMapQ]
[-Q maxSumErr] [-m maxMM] [-p] out.map in.map</p>

<p style="margin-top: 1em">Filter bad alignments in in.map.
Command-line options are described in the
&lsquo;assemble&rsquo; command.</p>

<p style="margin-top: 1em">eland2maq maq eland2maq [-q
defqual] out.map in.list in.eland</p>

<p style="margin-top: 1em">Convert eland alignment to
maq&rsquo;s .map format. File in.list consists of the
sequence names that appear at the seventh column of the
eland alignment file in.eland and <br>
the name you expect to see in maq alignment. The following
is an example:</p>

<p style="margin-top: 1em">cX.fa chrX <br>
c1.fa chr1 <br>
c2.fa chr2</p>

<p style="margin-top: 1em">If you are aligning reads in
several batches using eland, it is important to use the same
in.list for the conversion. In addition, maq will load all
the alignments and <br>
sort them in the memory. If you have concatenate several
eland outputs into one huge file, you should separate it
into smaller files to prevent maq from eating all <br>
your machine memory.</p>

<p style="margin-top: 1em">This command actually aims to
show Eland alignment in Maqview. As no quality information
is available, the resultant maq alignment file should not be
used to call con&acirc; <br>
sensus genotypes.</p>

<p style="margin-top: 1em">export2maq maq export2maq [-1
read1len] [-2 read2len] [-a maxdist] [-n] out.map in.list
in.export</p>

<p style="margin-top: 1em">Convert Illumina&rsquo;s Export
format to Maq&rsquo;s .map format. Export format is a new
alignment format since SolexaPipeline-0.3.0 which also
calculates mapping qualities like <br>
maq. The resultant file can be used to call consensus
genotypes as most of necessary information is available for
maq to do this accurately.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-1 INT Length of the first read
[0]</p>

<p style="margin-top: 1em">-2 INT Length of the second read
[0]</p>

<p style="margin-top: 1em">-a INT Maximum outer distance
for a correct read pair [250]</p>

<p style="margin-top: 1em">-n Retain filtered reads</p>

<p style="margin-top: 1em">MAQ-PERL COMMANDS <br>
demo maq.pl demo [-h] [-s] [-N nPairs] [-d outDir] in.fasta
in.simudat</p>

<p style="margin-top: 1em">Demonstrate the use of maq and
its companion scripts. This command will simulate reads from
a FASTA file in.fasta. The sequence length and qualities are
determined by <br>
in.simudat which is generated from maq simutrain or can be
downloaded from Maq website. The simulated reads will then
be mapped with maq.pl easyrun. The alignment <br>
accuracy is evaluated by maq simustat, the consensus
accuracy by maq simucns, and the SNP accuracy by
maq_eval.pl.</p>

<p style="margin-top: 1em">By default, paired end reads
will be simulated and a diploid sequence will be generated
from the input by adding mutations to either haploid type.
The insert size and <br>
mutation rate are controlled by maq simulate.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-h simulate a haploid sequence
instead of a diploid sequence</p>

<p style="margin-top: 1em">-s use single-end mode to align
reads instead of paired-end mode</p>

<p style="margin-top: 1em">-N INT number of pairs of reads
to be simulated [1000000]</p>

<p style="margin-top: 1em">-d DIR output directory
[maqdemo]</p>

<p style="margin-top: 1em">NOTE:</p>

<p style="margin-top: 1em">* The output files from
maq_eval.pl have not been documented, but you may make a
good guess at some of these files.</p>

<p style="margin-top: 1em">* This command just demonstrates
the use of the maq suite. The accuracy on real data is
almost always lower than what you see from pure
simulation.</p>

<p style="margin-top: 1em">easyrun maq.pl easyrun [-1
read1Len] [-d out.dir] [-n nReads] [-A 3adapter] [-e minDep]
[-q minCnsQ] [-p] [-2 read2Len] [-a maxIns] [-S] [-N]
in.ref.fasta in1.fastq <br>
[in2.fastq]</p>

<p style="margin-top: 1em">Analyses pipeline for small
genomes. Easyrun command will run most of analyses
implemented in maq. By default, easyrun assumes all the
input read sequences files are <br>
single-end and independent; when -p is specified, two read
sequence files are required, one for each end.</p>

<p style="margin-top: 1em">Several files will be generated
in out.dir, among which the following files are the key
output:</p>

<p style="margin-top: 1em">cns.final.snp final SNP calls
with low quality ones filtered out</p>

<p style="margin-top: 1em">cns.fq consensus sequences and
qualities in the FASTQ format</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-d DIR output directory
[easyrun]</p>

<p style="margin-top: 1em">-n INT number of reads/pairs in
one batch of alignment [2000000]</p>

<p style="margin-top: 1em">-S apply split-read analysis of
short indels (maybe very slow)</p>

<p style="margin-top: 1em">-N INT number of
haplotypes/strains in the pool (&gt;=2) [2]</p>

<p style="margin-top: 1em">-A FILE file for
3&rsquo;-adapter. The file should contain a single line of
sequence [null]</p>

<p style="margin-top: 1em">-1 INT length of the first read,
0 for auto [0]</p>

<p style="margin-top: 1em">-e INT minimum read depth
required to call a SNP (for SNPfilter) [3]</p>

<p style="margin-top: 1em">-q INT minimum consensus quality
for SNPs in cns.final.snp [30]</p>

<p style="margin-top: 1em">-p switch to paired end
alignment mode</p>

<p style="margin-top: 1em">-2 INT length of the second read
when -p is applied [0]</p>

<p style="margin-top: 1em">-a INT maximum insert size when
-p is applied [250]</p>

<p style="margin-top: 1em">NOTES:</p>

<p style="margin-top: 1em">* For SNP calling on pooled
samples, users should set correct &lsquo;-N&rsquo; as well
as &lsquo;-E 0&rsquo;.</p>

<p style="margin-top: 1em">* The input file can be
maq&rsquo;s binary format. maq.pl will automatically detect
the file format.</p>

<p style="margin-top: 1em">SNPfilter maq.pl SNPfilter [-d
minDep] [-D maxDep] [-Q maxMapQ] [-q minCnsQ] [-w
indelWinSize] [-n minNeiQ] [-F in.indelpe] [-f in.indelsoa]
[-s minScore] [-m maxAcross] [-a] <br>
[-N maxWinSNP] [-W densWinSize] in.cns2snp.snp &gt;
out.filtered.snp</p>

<p style="margin-top: 1em">Rule out SNPs that are covered
by few reads (specified by -d), by too many reads (specified
by -D), near (specified by -w) to a potential indel, falling
in a possible <br>
repetitve region (characterized by -Q), or having
low-quality neighbouring bases (specified by -n). If
maxWinSNP or more SNPs appear in any densWinSize window,
they <br>
will also be filtered out together.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-d INT Minimum read depth
required to call a SNP [3]</p>

<p style="margin-top: 1em">-D INT Maximum read depth
required to call a SNP (&lt;255, otherwise ignored)
[256]</p>

<p style="margin-top: 1em">-Q INT Required maximum mapping
quality of reads covering the SNP [40]</p>

<p style="margin-top: 1em">-q INT Minimum consensus quality
[20]</p>

<p style="margin-top: 1em">-n INT Minimum adjacent
consensus quality [20]</p>

<p style="margin-top: 1em">-w INT Size of the window around
the potential indels. SNPs that are close to indels will be
suppressed [3]</p>

<p style="margin-top: 1em">-F FILE The indelpe output
[null]</p>

<p style="margin-top: 1em">-f FILE The indelsoa output
[null]</p>

<p style="margin-top: 1em">-s INT Minimum score for a
soa-indel to be considered [3]</p>

<p style="margin-top: 1em">-m INT Maximum number of reads
that can be mapped across a soa-indel [1]</p>

<p style="margin-top: 1em">-a Alternative filter for single
end alignment</p>

<p style="margin-top: 1em">indelpe maq.pl indelpe
in.indelpe &gt; out.indelpe</p>

<p style="margin-top: 1em">Correct the number of reads
mapped without indels for homopolymer tracts. This command
modify the 4th, 10th and the last three columns of
in.indelpe and output the <br>
result in out.indelpe. After the correction, the following
awk command gives putative homozygous indels:</p>

<p style="margin-top: 1em">awk
&rsquo;($3==&quot;*&quot;&acirc;&ordf;&acirc;&ordf;$3==&quot;+&quot;)
&amp;&amp; $6+$7&gt;=3 &amp;&amp;
($6+$7)/$4&gt;=0.75&rsquo;</p>

<p style="margin-top: 1em">and the following gives
heterozygotes:</p>

<p style="margin-top: 1em">awk
&rsquo;($3==&quot;*&quot;&acirc;&ordf;&acirc;&ordf;$3==&quot;+&quot;)
&amp;&amp; $6+$7&gt;=3 &amp;&amp;
($6+$7)/$4&lt;0.75&rsquo;</p>

<p style="margin-top: 1em">Please note that this indelpe
command just implements several heuristic rules. It does not
correct for impure homopolymer runs or di-nucleotide/triplet
repeats. Conse&acirc; <br>
quently, the two awk commands only give approximate hom/het
indels.</p>

<p style="margin-top: 1em">EXAMPLES <br>
&Acirc;&middot; Easyrun script: <br>
maq.pl easyrun -d easyrun ref.fasta part1.fastq
part2.fastq</p>

<p style="margin-top: 1em">&Acirc;&middot; Key commands
behind easyrun: <br>
maq fasta2bfa ref.fasta ref.bfa; <br>
maq fastq2bfq part1.fastq part1.bfq; <br>
maq fastq2bfq part2.fastq part2.bfq; <br>
maq map part1.map ref.bfa part1.bfq; <br>
maq map part2.map ref.bfa part2.bfq; <br>
maq mapmerge aln.map part1.map part2.map; <br>
maq assemble cns.cns ref.bfa aln.map;</p>

<p style="margin-top: 1em">LICENSE <br>
GNU General Public License, version 3 (GPLv3)</p>

<p style="margin-top: 1em">AVAILABILITY <br>
&lt;http://maq.sourceforge.net&gt;</p>

<p style="margin-top: 1em">AUTHOR <br>
Heng Li &lt;lh3@sanger.ac.uk&gt;</p>

<p style="margin-top: 1em">maq-0.7.1 2008-09-22 MAQ(1)</p>
<hr>
</body>
</html>
