<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:56 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MAQVIEW(1) Bioinformatics Tools MAQVIEW(1)</p>

<p style="margin-top: 1em">NAME <br>
Maqview - Maq alignment indexer and viewer</p>

<p style="margin-top: 1em">SYNOPSIS <br>
maqindex -i|-v|-b [-c in.cns] in.map [chr[:start[-end]]
[...]]</p>

<p style="margin-top: 1em">maqview [-c in.cns] in.map</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Maqview is a set of programs that achieves fast random
access to the alignment file generated by &lsquo;maq
map&rsquo;, and displays the alignment with a nice GUI
(Graphical User Interface). <br>
It is highly efficient in speed, memory and disk usage.
Maqview is based on OpenGL and is known to work on both Mac
OS X and Linux. Porting to Windows is in principle easy.</p>

<p style="margin-top: 1em">maqindex <br>
maqindex -i|-v|-b [-c in.cns] in.map [chr[:start[-end]]
[...]]</p>

<p style="margin-top: 1em">-i Index the alignment file</p>

<p style="margin-top: 1em">-v Print the alignment in the
specified region in the &lsquo;maq mapview&rsquo; format</p>

<p style="margin-top: 1em">-b Dump the alignment in the
specified region in the binary .map format</p>

<p style="margin-top: 1em">-c FILE Index the Maq consensus
file for viewing in maqview at the same time</p>

<p style="margin-top: 1em">Program maqindex indexes a maq
alignment file in.map or quickly retrieves all the reads in
one or multiple regions. One of -i, -v and -b must be used.
For -v or -b, at least one <br>
region like &lsquo;chrX&rsquo;, &lsquo;chrX:1000&rsquo; or
&lsquo;chrX:1,000-2,000&rsquo; must be specified. Multiple
regions are allowed.</p>

<p style="margin-top: 1em">maqview <br>
maqview [-c in.cns] in.map</p>

<p style="margin-top: 1em">-c FILE The Maq consensus
file</p>

<p style="margin-top: 1em">Program maqview displays the
read alignment in a graphical window. When in.cns is
specified, the top sequence is the reference, followed by
the Maq consensus sequence; otherwise, <br>
the majority-rule consensus will be calculated.</p>

<p style="margin-top: 1em">Maqview has two views: sequence
view and box view. In the sequence view, read sequences will
be printed on the screen. Darker bases indicate lower base
qualities and red ones <br>
show the differences in comparison to the reference if
in.cns is specified or to the majority-rule consensus if
not. In the box view, different types of nucleotides are
<br>
represented as colour boxes with green for A, cyan for C,
orange for G, red for T and dark gray for N. The saturation
of colours indicates the base qualities and the thickness
<br>
lines of reads shows the mapping qualities of read
alignments. Zooming in/out is supported only in the box
view. In both views, the status bar at the bottom of the
window will <br>
show some information about the key touches, and read names
and base qualities pointed by the mouse.</p>

<p style="margin-top: 1em">Navigating the alignment is
accomplished by vaious key bindings.</p>

<p style="margin-top: 1em">&Acirc;&middot; Key
bindings:</p>

<p style="margin-top: 1em">? Display help <br>
&lt;Esc&gt; Exit <br>
&lt;F1&gt; Sequence view <br>
&lt;F2&gt; Box view <br>
p Switch to the previous reference sequence <br>
n Switch to the next reference sequence <br>
r Refresh</p>

<p style="margin-top: 1em">&lt;digits&gt; &lt;Enter&gt;
Move to the position pointed by &lt;digits&gt; <br>
h / &lt;Left&gt; Move left by one base <br>
l / &lt;Right&gt; Move right by one base <br>
k / &lt;Up&gt; Move up by one line <br>
j / &lt;Down&gt; Move down by one line <br>
&gt; Move to the next read <br>
&lt; Move to the previous read <br>
g / &lt;Home&gt; Move to the beginning of the current
reference <br>
G / &lt;End&gt; Move to the end of the current reference
<br>
u / &lt;PageUp&gt; Move left by one page <br>
&lt;Space&gt;/&lt;PageDn&gt; Move right by one page <br>
&lt;Shift&gt; &lt;Arrow&gt; Move by 100 bases <br>
&lt;Ctrl&gt; &lt;Arrow&gt; Move by 1000 bases</p>

<p style="margin-top: 1em">&lt;Ctrl&gt; + Add a new view
<br>
&lt;Ctrl&gt; - Remove the current view</p>

<p style="margin-top: 1em">&lt;Enter&gt; Toggle on/off
right auto-scrolling <br>
^b Toggle on/off left auto-scrolling <br>
+ Zoom in <br>
- Zoom out <br>
0 Zoom to the default scale <br>
q Toggle show/hide mapping qualities <br>
e Toggle display SE/PE mapping qualities <br>
o Open a new alignment file</p>

<p style="margin-top: 1em">&Acirc;&middot; Enter an integer
and then press &lt;Enter&gt; to jump to the required
position.</p>

<p style="margin-top: 1em">&Acirc;&middot; Enter an integer
and then press a key other than &lt;Enter&gt; to speed up
scrolling by the specified factor.</p>

<p style="margin-top: 1em">&Acirc;&middot; Left click and
drag the yellow cross in the square to move along the
sequence. Right click the reference to highlight a
column.</p>

<p style="margin-top: 1em">LICENSE <br>
GNU General Public License (GPL)</p>

<p style="margin-top: 1em">AVAILABILITY <br>
&lt;http://maq.sourceforge.net&gt;</p>

<p style="margin-top: 1em">AUTHORS <br>
Jue Ruan &lt;ruanjue@genomics.org.cn&gt; for writing and
maintaining the whole software.</p>

<p style="margin-top: 1em">Heng Li &lt;lh3@sanger.ac.uk&gt;
for testing the program and drafting the documentations.</p>

<p style="margin-top: 1em">Mengyao Zhao
&lt;zhaomengyao@gmail.com&gt; for designing the color
schemes.</p>

<p style="margin-top: 1em">maqview-0.2.4 2015-12-17
MAQVIEW(1)</p>
<hr>
</body>
</html>
