<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:23:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>MATCHER(1e) EMBOSS Manual for Debian MATCHER(1e)</p>

<p style="margin-top: 1em">NAME <br>
matcher - Waterman-Eggert local alignment of two
sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
matcher -asequence sequence -bsequence sequence [-datafile
matrix] [-alternatives integer] [-gapopen integer]
[-gapextend integer] -outfile align</p>

<p style="margin-top: 1em">matcher -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
matcher is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Alignment:Local&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-asequence sequence</p>

<p style="margin-top: 1em">-bsequence sequence</p>

<p style="margin-top: 1em">-datafile matrix <br>
This is the scoring matrix file used when comparing
sequences. By default it is the file &rsquo;EBLOSUM62&rsquo;
(for proteins) or the file &rsquo;EDNAFULL&rsquo; (for
nucleic sequences). These <br>
files are found in the &rsquo;data&rsquo; directory of the
EMBOSS installation.</p>

<p style="margin-top: 1em">Additional section <br>
-alternatives integer <br>
This sets the number of alternative matches output. By
default only the highest scoring alignment is shown. A value
of 2 gives you other reasonable alignments. In some cases,
<br>
for example multidomain proteins of cDNA and genomic DNA
comparisons, there may be other interesting and significant
alignments. Default value: 1</p>

<p style="margin-top: 1em">-gapopen integer <br>
The gap penalty is the score taken away when a gap is
created. The best value depends on the choice of comparison
matrix. The default value of 14 assumes you are using the
<br>
EBLOSUM62 matrix for protein sequences, or a value of 16 and
the EDNAFULL matrix for nucleotide sequences. Default value:
@($(acdprotein)? 14 : 16)</p>

<p style="margin-top: 1em">-gapextend integer <br>
The gap length, or gap extension, penalty is added to the
standard gap penalty for each base or residue in the gap.
This is how long gaps are penalized. Usually you will <br>
expect a few long gaps rather than many short gaps, so the
gap extension penalty should be lower than the gap penalty.
An exception is where one or both sequences are single <br>
reads with possible sequencing errors in which case you
would expect many single base gaps. You can get this result
by setting the gap penalty to zero (or very low) and using
<br>
the gap extension penalty to control gap scoring. Default
value: @($(acdprotein)? 4 : 4)</p>

<p style="margin-top: 1em">Output section <br>
-outfile align</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
matcher is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
MATCHER(1e)</p>
<hr>
</body>
</html>
