<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>MATCHER(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">MATCHER(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">MATCHER(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
matcher - Waterman-Eggert local alignment of two sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 8.00ex; text-indent: -8.00ex;"><b>matcher</b>
  <b>-asequence&#x00A0;</b><b></b><i>sequence</i>
  <b>-bsequence&#x00A0;</b><b></b> <i>sequence</i>
  [<b>-datafile&#x00A0;</b><b></b> <i>matrix</i>]
  [<b>-alternatives&#x00A0;</b><b></b><i>integer</i>]
  [<b>-gapopen&#x00A0;</b><b></b><i>integer</i>]
  [<b>-gapextend&#x00A0;</b><b></b> <i>integer</i>]
  <b>-outfile&#x00A0;</b><b></b> <i>align</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 8.00ex; text-indent: -8.00ex;"><b>matcher</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>matcher</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Alignment:Local&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-asequence</b> <i>sequence</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-bsequence</b> <i>sequence</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-datafile</b> <i>matrix</i>
<div style="margin-left: 4.00ex;">This is the scoring matrix file used when
  comparing sequences. By default it is the file 'EBLOSUM62' (for proteins) or
  the file 'EDNAFULL' (for nucleic sequences). These files are found in the
  'data' directory of the EMBOSS installation.</div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-alternatives</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This sets the number of alternative matches
  output. By default only the highest scoring alignment is shown. A value of 2
  gives you other reasonable alignments. In some cases, for example multidomain
  proteins of cDNA and genomic DNA comparisons, there may be other interesting
  and significant alignments. Default value: 1</div>
<div class="Pp"></div>
<b>-gapopen</b> <i>integer</i>
<div style="margin-left: 4.00ex;">The gap penalty is the score taken away when a
  gap is created. The best value depends on the choice of comparison matrix. The
  default value of 14 assumes you are using the EBLOSUM62 matrix for protein
  sequences, or a value of 16 and the EDNAFULL matrix for nucleotide sequences.
  Default value: @($(acdprotein)? 14 : 16)</div>
<div class="Pp"></div>
<b>-gapextend</b> <i>integer</i>
<div style="margin-left: 4.00ex;">The gap length, or gap extension, penalty is
  added to the standard gap penalty for each base or residue in the gap. This is
  how long gaps are penalized. Usually you will expect a few long gaps rather
  than many short gaps, so the gap extension penalty should be lower than the
  gap penalty. An exception is where one or both sequences are single reads with
  possible sequencing errors in which case you would expect many single base
  gaps. You can get this result by setting the gap penalty to zero (or very low)
  and using the gap extension penalty to control gap scoring. Default value:
  @($(acdprotein)? 4 : 4)</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outfile</b> <i>align</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
matcher is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
