<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:27:04 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>NEEDLE(1e) EMBOSS Manual for Debian NEEDLE(1e)</p>

<p style="margin-top: 1em">NAME <br>
needle - Needleman-Wunsch global alignment of two
sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
needle -asequence sequence -bsequence seqall [-datafile
matrixf] -gapopen float -gapextend float [-endweight
boolean] [-endopen float] [-endextend float] -brief boolean
<br>
-outfile align</p>

<p style="margin-top: 1em">needle -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
needle is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Alignment:Global&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-asequence sequence</p>

<p style="margin-top: 1em">-bsequence seqall</p>

<p style="margin-top: 1em">-datafile matrixf <br>
This is the scoring matrix file used when comparing
sequences. By default it is the file &rsquo;EBLOSUM62&rsquo;
(for proteins) or the file &rsquo;EDNAFULL&rsquo; (for
nucleic sequences). These <br>
files are found in the &rsquo;data&rsquo; directory of the
EMBOSS installation.</p>

<p style="margin-top: 1em">Required section <br>
-gapopen float <br>
The gap open penalty is the score taken away when a gap is
created. The best value depends on the choice of comparison
matrix. The default value assumes you are using the <br>
EBLOSUM62 matrix for protein sequences, and the EDNAFULL
matrix for nucleotide sequences. Default value:
@($(acdprotein)? 10.0 : 10.0 )</p>

<p style="margin-top: 1em">-gapextend float <br>
The gap extension, penalty is added to the standard gap
penalty for each base or residue in the gap. This is how
long gaps are penalized. Usually you will expect a few long
<br>
gaps rather than many short gaps, so the gap extension
penalty should be lower than the gap penalty. An exception
is where one or both sequences are single reads with <br>
possible sequencing errors in which case you would expect
many single base gaps. You can get this result by setting
the gap open penalty to zero (or very low) and using the
<br>
gap extension penalty to control gap scoring. Default value:
@($(acdprotein)? 0.5 : 0.5 )</p>

<p style="margin-top: 1em">Additional section <br>
-endweight boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-endopen float <br>
The end gap open penalty is the score taken away when an end
gap is created. The best value depends on the choice of
comparison matrix. The default value assumes you are <br>
using the EBLOSUM62 matrix for protein sequences, and the
EDNAFULL matrix for nucleotide sequences. Default value:
@($(acdprotein)? 10.0 : 10.0 )</p>

<p style="margin-top: 1em">-endextend float <br>
The end gap extension, penalty is added to the end gap
penalty for each base or residue in the end gap. This is how
long end gaps are penalized. Default value: <br>
@($(acdprotein)? 0.5 : 0.5 )</p>

<p style="margin-top: 1em">Output section <br>
-brief boolean <br>
Brief identity and similarity Default value: Y</p>

<p style="margin-top: 1em">-outfile align</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
needle is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
NEEDLE(1e)</p>
<hr>
</body>
</html>
