<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:27:18 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>nhmmer(1) HMMER Manual nhmmer(1)</p>

<p style="margin-top: 1em">NAME <br>
nhmmer - search DNA/RNA queries against a DNA/RNA sequence
database</p>

<p style="margin-top: 1em">SYNOPSIS <br>
nhmmer [options] &lt;queryfile&gt; &lt;seqdb&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
nhmmer is used to search one or more nucleotide queries
against a nucleotide sequence database. For each query in
&lt;queryfile&gt;, use that query to search the target
database of <br>
sequences in &lt;seqdb&gt;, and output a ranked list of the
hits with the most significant matches to the query. A query
may be either a profile model built using hmmbuild, a
sequence <br>
alignment, or a single sequence. Sequence based queries can
be in a number of formats (see --qformat), and can typically
be autodetected. Note that only Stockholm format supports
<br>
queries made up of more than one sequence alignment.</p>

<p style="margin-top: 1em">Either the query
&lt;queryfile&gt; or the target &lt;seqdb&gt; may be
&rsquo;-&rsquo; (a dash character), in which case the query
file or target database input will be read from a
&lt;stdin&gt; pipe instead <br>
of from a file. Only one input source can come through
&lt;stdin&gt;, not both. If the query is sequence-based and
passed via &lt;stdin&gt;, the --qformat flag must be used.
If the &lt;query&acirc; <br>
file&gt; contains more than one query, then &lt;seqdb&gt;
cannot come from &lt;stdin&gt;, because we can&rsquo;t
rewind the streaming target database to search it with
another profile.</p>

<p style="margin-top: 1em">If the query is sequence-based,
and not from &lt;stdin&gt;, a new file containing the HMM(s)
built from the input(s) in &lt;queryfile&gt; may optionally
be produced, with the filename set <br>
using the --hmmout flag.</p>

<p style="margin-top: 1em">The output format is designed to
be human-readable, but is often so voluminous that reading
it is impractical, and parsing it is a pain. The --tblout
option saves output in a <br>
simple tabular format that is concise and easier to parse.
The -o option allows redirecting the main output, including
throwing it away in /dev/null.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
all available options.</p>

<p style="margin-top: 1em">OPTIONS FOR CONTROLLING OUTPUT
<br>
-o &lt;f&gt; Direct the main human-readable output to a file
&lt;f&gt; instead of the default stdout.</p>

<p style="margin-top: 1em">-A &lt;f&gt; Save a multiple
alignment of all significant hits (those satisfying
inclusion thresholds) to the file &lt;f&gt;.</p>

<p style="margin-top: 1em">--tblout &lt;f&gt; <br>
Save a simple tabular (space-delimited) file summarizing the
per-target output, with one data line per homologous target
sequence found.</p>

<p style="margin-top: 1em">--dfamtblout &lt;f&gt; <br>
Save a tabular (space-delimited) file summarizing the
per-hit output, similar to --tblout but more succinct.</p>

<p style="margin-top: 1em">--aliscoresout &lt;f&gt; <br>
Save to file a list of per-position scores for each hit.
This is useful, for example, in identifying regions of high
score density for use in resolving overlapping hits <br>
from different models.</p>

<p style="margin-top: 1em">--hmmout &lt;f&gt; <br>
If &lt;queryfile&gt; is sequence-based, write the
internally-computed HMM(s) to &lt;f&gt;.</p>

<p style="margin-top: 1em">--acc Use accessions instead of
names in the main output, where available for profiles
and/or sequences.</p>

<p style="margin-top: 1em">--noali <br>
Omit the alignment section from the main output. This can
greatly reduce the output volume.</p>

<p style="margin-top: 1em">--notextw <br>
Unlimit the length of each line in the main output. The
default is a limit of 120 characters per line, which helps
in displaying the output cleanly on terminals and in <br>
editors, but can truncate target profile description
lines.</p>

<p style="margin-top: 1em">--textw &lt;n&gt; <br>
Set the main output&rsquo;s line length limit to &lt;n&gt;
characters per line. The default is 120.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING REPORTING
THRESHOLDS <br>
Reporting thresholds control which hits are reported in
output files (the main output, --tblout, and --dfamtblout).
Hits are ranked by statistical significance (E-value).</p>

<p style="margin-top: 1em">-E &lt;x&gt; Report target
sequences with an E-value of &lt;= &lt;x&gt;. The default is
10.0, meaning that on average, about 10 false positives will
be reported per query, so you can see the <br>
top of the noise and decide for yourself if it&rsquo;s
really noise.</p>

<p style="margin-top: 1em">-T &lt;x&gt; Instead of
thresholding output on E-value, instead report target
sequences with a bit score of &gt;= &lt;x&gt;.</p>

<p style="margin-top: 1em">OPTIONS FOR INCLUSION THRESHOLDS
<br>
Inclusion thresholds are stricter than reporting thresholds.
Inclusion thresholds control which hits are considered to be
reliable enough to be included in an output alignment <br>
or a subsequent search round, or marked as significant
(&quot;!&quot;) as opposed to questionable (&quot;?&quot;)
in hit output.</p>

<p style="margin-top: 1em">--incE &lt;x&gt; <br>
Use an E-value of &lt;= &lt;x&gt; as the inclusion
threshold. The default is 0.01, meaning that on average,
about 1 false positive would be expected in every 100
searches with <br>
different query sequences.</p>

<p style="margin-top: 1em">--incT &lt;x&gt; <br>
Instead of using E-values for setting the inclusion
threshold, use a bit score of &gt;= &lt;x&gt; as the
inclusion threshold. By default this option is unset.</p>

<p style="margin-top: 1em">OPTIONS FOR MODEL-SPECIFIC SCORE
THRESHOLDING <br>
Curated profile databases may define specific bit score
thresholds for each profile, superseding any thresholding
based on statistical significance alone.</p>

<p style="margin-top: 1em">To use these options, the
profile must contain the appropriate (GA, TC, and/or NC)
optional score threshold annotation; this is picked up by
hmmbuild from Stockholm format align&acirc; <br>
ment files. For a nucleotide model, each thresholding option
has a single per-hit threshold &lt;x&gt; This acts as if
-T&lt;x&gt; --incT&lt;x&gt; has been applied specifically
using each model&rsquo;s <br>
curated thresholds.</p>

<p style="margin-top: 1em">--cut_ga <br>
Use the GA (gathering) bit score threshold in the model to
set per-hit reporting and inclusion thresholds. GA
thresholds are generally considered to be the reliable <br>
curated thresholds defining family membership; for example,
in Dfam, these thresholds are applied when annotating a
genome with a model of a family known to be found in <br>
that organism. They may allow for minimal expected false
discovery rate.</p>

<p style="margin-top: 1em">--cut_nc <br>
Use the NC (noise cutoff) bit score threshold in the model
to set per-hit reporting and inclusion thresholds. NC
thresholds are less stringent than GA; in the context of
<br>
Pfam, they are generally used to store the score of the
highest-scoring known false positive.</p>

<p style="margin-top: 1em">--cut_tc <br>
Use the NC (trusted cutoff) bit score threshold in the model
to set per-hit reporting and inclusion thresholds. TC
thresholds are more stringent than GA, and are generally
<br>
considered to be the score of the lowest-scoring known true
positive that is above all known false positives; for
example, in Dfam, these thresholds are applied when
anno&acirc; <br>
tating a genome with a model of a family not known to be
found in that organism.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING THE
ACCELERATION PIPELINE <br>
HMMER3 searches are accelerated in a three-step filter
pipeline: the scanning-SSV filter, the Viterbi filter, and
the Forward filter. The first filter is the fastest and most
<br>
approximate; the last is the full Forward scoring algorithm.
There is also a bias filter step between SSV and Viterbi.
Targets that pass all the steps in the acceleration
pipe&acirc; <br>
line are then subjected to postprocessing -- domain
identification and scoring using the Forward/Backward
algorithm.</p>

<p style="margin-top: 1em">Changing filter thresholds only
removes or includes targets from consideration; changing
filter thresholds does not alter bit scores, E-values, or
alignments, all of which are <br>
determined solely in postprocessing.</p>

<p style="margin-top: 1em">--max Turn off (nearly) all
filters, including the bias filter, and run full
Forward/Backward postprocessing on most of the target
sequence. In contrast to phmmer and hmm&acirc; <br>
search, where this flag really does turn off the filters
entirely, the --max flag in nhmmer sets the scanning-SSV
filter threshold to 0.4, not 1.0. Use of this flag <br>
increases sensitivity somewhat, at a large cost in
speed.</p>

<p style="margin-top: 1em">--F1 &lt;x&gt; <br>
Set the P-value threshold for the SSV filter step. The
default is 0.02, meaning that roughly 2% of the highest
scoring nonhomologous targets are expected to pass the
fil&acirc; <br>
ter.</p>

<p style="margin-top: 1em">--F2 &lt;x&gt; <br>
Set the P-value threshold for the Viterbi filter step. The
default is 0.001.</p>

<p style="margin-top: 1em">--F3 &lt;x&gt; <br>
Set the P-value threshold for the Forward filter step. The
default is 1e-5.</p>

<p style="margin-top: 1em">--nobias <br>
Turn off the bias filter. This increases sensitivity
somewhat, but can come at a high cost in speed, especially
if the query has biased residue composition (such as a <br>
repetitive sequence region, or if it is a membrane protein
with large regions of hydrophobicity). Without the bias
filter, too many sequences may pass the filter with <br>
biased queries, leading to slower than expected performance
as the computationally intensive Forward/Backward algorithms
shoulder an abnormally heavy load.</p>

<p style="margin-top: 1em">OPTIONS FOR SPECIFYING THE
ALPHABET <br>
The alphabet type of the target database (DNA or RNA) is
autodetected by default, by looking at the composition of
the &lt;seqdb&gt;. Autodetection is normally quite reliable,
but <br>
occasionally alphabet type may be ambiguous and
autodetection can fail (for instance, when the first
sequence starts with a run of ambiguous characters). To
avoid this, or to <br>
increase robustness in automated analysis pipelines, you may
specify the alphabet type of &lt;seqdb&gt; with these
options.</p>

<p style="margin-top: 1em">--dna Specify that all sequences
in &lt;seqdb&gt; are DNAs.</p>

<p style="margin-top: 1em">--rna Specify that all sequences
in &lt;seqdb&gt; are RNAs.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING SEED SEARCH
HEURISTIC <br>
When searching with nhmmer, one may optionally precompute a
binary version of the target database, using makehmmerdb,
then search against that database. Using default settings,
<br>
this yields a roughly 10-fold acceleration with small loss
of sensitivity on benchmarks. This is achieved using a
heuristic method that searches for seeds (ungapped
alignments) <br>
around which full processing is done. This is essentially a
replacement to the SSV stage. (This method has been
extensively tested, but should still be treated as somewhat
exper&acirc; <br>
imental.) The following options only impact nhmmer if the
value of --tformat is hmmerdb.</p>

<p style="margin-top: 1em">Changing parameters for this
seed-finding step will impact both speed and sensitivity -
typically faster search leads to lower sensitivity.</p>

<p style="margin-top: 1em">--seed_max_depth &lt;n&gt; <br>
The seed step requires that a seed reach a specified bit
score in length no longer than &lt;n&gt;. By default, this
value is 15. Longer seeds allow a greater chance of meeting
<br>
the bit score threshold, leading to diminished filtering
(greater sensitivity, slower run time).</p>

<p style="margin-top: 1em">--seed_sc_thresh &lt;x&gt; <br>
The seed must reach score &lt;x&gt; (in bits). The default
is 15.0 bits. A higher threshold increases filtering
stringency, leading to faster run times and lower
sensitivity.</p>

<p style="margin-top: 1em">--seed_sc_density &lt;x&gt; <br>
Either all prefixes or all suffixes of a seed must have bit
density (bits per aligned position) of at least &lt;x&gt;.
The default is 0.8 bits/position. An increase in the
den&acirc; <br>
sity requirement leads to increased filtering stringency,
thus faster run times and lower sensitivity.</p>

<p style="margin-top: 1em">--seed_drop_max_len &lt;n&gt;
<br>
A seed may not have a run of length &lt;n&gt; in which the
score drops by --seed_drop_lim or more. Basically, this
prunes seeds that go through long slightly-negative seed
<br>
extensions. The default is 4. Increasing the limit causes
(slightly) diminished filtering efficiency, thus slower run
times and higher sensitivity. (minor tuning option)</p>

<p style="margin-top: 1em">--seed_drop_lim &lt;x&gt; <br>
In a seed, there may be no run of length --seed_drop_max_len
in which the score drops by --seed_drop_lim. The default is
0.3 bits. Larger numbers mean less filtering. <br>
(minor tuning option)</p>

<p style="margin-top: 1em">--seed_req_pos &lt;n&gt; <br>
A seed must contain a run of at least &lt;n&gt;
positive-scoring matches. The default is 5. Larger values
mean increased filtering. (minor tuning option)</p>

<p style="margin-top: 1em">--seed_ssv_length &lt;n&gt; <br>
After finding a short seed, an ungapped alignment is
extended in both directions in an attempt to meet the --F1
score threshold. The window through which this ungapped <br>
alignment extends is length &lt;n&gt;. The default is 70.
Decreasing this value slightly reduces run time, at a small
risk of reduced sensitivity. (minor tuning option)</p>

<p style="margin-top: 1em">OTHER OPTIONS <br>
--tformat &lt;s&gt; <br>
Assert that the target sequence database file is in format
&lt;s&gt;. Accepted formats include fasta, embl, genbank,
ddbj, uniprot, stockholm, pfam, a2m, afa, and hmmerfm. The
<br>
default is to autodetect the format of the file. The format
hmmerfm indicates that the database file is a binary file
produced using makehmmerdb (this format is not cur&acirc;
<br>
rently autodetected).</p>

<p style="margin-top: 1em">--qformat &lt;s&gt; <br>
Declare that the input &lt;queryfile&gt; is in format
&lt;s&gt;. This is used when the query is sequence-based,
rather than made up of profile model(s). Currently the
accepted mul&acirc; <br>
tiple alignment sequence file formats include Stockholm,
Aligned FASTA, Clustal, NCBI PSI-BLAST, PHYLIP, Selex, and
UCSC SAM A2M. Default is to autodetect the format of <br>
the file.</p>

<p style="margin-top: 1em">--nonull2 <br>
Turn off the null2 score corrections for biased
composition.</p>

<p style="margin-top: 1em">-Z &lt;x&gt; For the purposes of
per-hit E-value calculations, Assert that the total size of
the target database is &lt;x&gt; million nucleotides, rather
than the actual number of targets <br>
seen.</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Set the random number seed to &lt;n&gt;. Some steps in
postprocessing require Monte Carlo simulation. The default
is to use a fixed seed (42), so that results are exactly
<br>
reproducible. Any other positive integer will give different
(but also reproducible) results. A choice of 0 uses a
randomly chosen seed.</p>

<p style="margin-top: 1em">--w_beta &lt;x&gt; <br>
Window length tail mass. The upper bound, W, on the length
at which nhmmer expects to find an instance of the model is
set such that the fraction of all sequences gener&acirc;
<br>
ated by the model with length &gt;= W is less than
&lt;x&gt;. The default is 1e-7. This flag may be used to
override the value of W established for the model by
hmmbuild, or when <br>
the query is sequence-based.</p>

<p style="margin-top: 1em">--w_length &lt;n&gt; <br>
Override the model instance length upper bound, W, which is
otherwise controlled by --w_beta. It should be larger than
the model length. The value of W is used deep in <br>
the acceleration pipeline, and modest changes are not
expected to impact results (though larger values of W do
lead to longer run time). This flag may be used to override
<br>
the value of W established for the model by hmmbuild, or
when the query is sequence-based.</p>

<p style="margin-top: 1em">--toponly <br>
Only search the top strand. By default both the query
sequence and its reverse-complement are searched.</p>

<p style="margin-top: 1em">--bottomonly <br>
Only search the bottom (reverse-complement) strand. By
default both the query sequence and its reverse-complement
are searched.</p>

<p style="margin-top: 1em">--cpu &lt;n&gt; <br>
Set the number of parallel worker threads to &lt;n&gt;. By
default, HMMER sets this to the number of CPU cores it
detects in your machine - that is, it tries to maximize the
<br>
use of your available processor cores. Setting &lt;n&gt;
higher than the number of available cores is of little if
any value, but you may want to set it to something less. You
<br>
can also control this number by setting an environment
variable, HMMER_NCPU.</p>

<p style="margin-top: 1em">This option is only available if
HMMER was compiled with POSIX threads support. This is the
default, but it may have been turned off at compile-time for
your site or <br>
machine for some reason.</p>

<p style="margin-top: 1em">--stall <br>
For debugging the MPI master/worker version: pause after
start, to enable the developer to attach debuggers to the
running master and worker(s) processes. Send SIGCONT <br>
signal to release the pause. (Under gdb: (gdb) signal
SIGCONT) (Only available if optional MPI support was enabled
at compile-time.)</p>

<p style="margin-top: 1em">--mpi Run in MPI master/worker
mode, using mpirun. (Only available if optional MPI support
was enabled at compile-time.)</p>

<p style="margin-top: 1em">SEE ALSO <br>
See hmmer(1) for a master man page with a list of all the
individual man pages for programs in the HMMER package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
nhmmer(1)</p>
<hr>
</body>
</html>
