<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>nhmmscan(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">nhmmscan(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">nhmmscan(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
nhmmscan - search nucleotide sequence(s) against a nucleotide profile database
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>hmmscan</b> <i>[options]</i> <i>&lt;hmmdb&gt;</i> <i>&lt;seqfile&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>nhmmscan</b> is used to search nucleotide sequences against collections of
  nucleotide profiles. For each sequence in <i>&lt;seqfile&gt;</i>, use that
  query sequence to search the target database of profiles in
  <i>&lt;hmmdb&gt;</i>, and output ranked lists of the profiles with the most
  significant matches to the sequence.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The <i>&lt;seqfile&gt;</i> may contain more than one query sequence. It can be
  in FASTA format, or several other common sequence file formats (genbank, embl,
  and uniprot, among others), or in alignment file formats (stockholm, aligned
  fasta, and others). See the <i>--qformat</i> option for a complete list.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The <i>&lt;hmmdb&gt;</i> needs to be press'ed using <b>hmmpress</b> before it
  can be searched with <b>hmmscan</b>. This creates four binary files, suffixed
  .h3{fimp}.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The query <i>&lt;seqfile&gt;</i> may be '-' (a dash character), in which case
  the query sequences are read from a &lt;stdin&gt; pipe instead of from a file.
  The <i>&lt;hmmdb&gt;</i> cannot be read from a &lt;stdin&gt; stream, because
  it needs to have those four auxiliary binary files generated by
  <b>hmmpress</b>.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The output format is designed to be human-readable, but is often so voluminous
  that reading it is impractical, and parsing it is a pain. The <b>--tblout</b>
  option saves output in a simple tabular format that is concise and easier to
  parse. The <b>-o</b> option allows redirecting the main output, including
  throwing it away in /dev/null.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and all
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_CONTROLLING_OUTPUT"><a class="selflink" href="#OPTIONS_FOR_CONTROLLING_OUTPUT">OPTIONS
  FOR CONTROLLING OUTPUT</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Direct the main human-readable output to a file
      <i>&lt;f&gt;</i> instead of the default stdout.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tblout</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save a simple tabular (space-delimited) file summarizing
      the per-hit output, with one data line per homologous target model hit
      found.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dfamtblout</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save a tabular (space-delimited) file summarizing the
      per-hit output, similar to <b>--tblout</b> but more succinct.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--aliscoresout</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save to file a list of per-position scores for each hit.
      This is useful, for example, in identifying regions of high score density
      for use in resolving overlapping hits from different models.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--acc</b></dt>
  <dd class="It-tag">Use accessions instead of names in the main output, where
      available for profiles and/or sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noali</b></dt>
  <dd class="It-tag">Omit the alignment section from the main output. This can
      greatly reduce the output volume.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--notextw</b></dt>
  <dd class="It-tag">Unlimit the length of each line in the main output. The
      default is a limit of 120 characters per line, which helps in displaying
      the output cleanly on terminals and in editors, but can truncate target
      profile description lines.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--textw</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the main output's line length limit to <i>&lt;n&gt;</i>
      characters per line. The default is 120.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_REPORTING_THRESHOLDS"><a class="selflink" href="#OPTIONS_FOR_REPORTING_THRESHOLDS">OPTIONS
  FOR REPORTING THRESHOLDS</a></h1>
Reporting thresholds control which hits are reported in output files (the main
  output, <i>--tblout</i>, and <i>--dfamtblout</i>). Hits are ranked by
  statistical significance (E-value).
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Report target profiles with an E-value of &lt;=
      <i>&lt;x&gt;</i>. The default is 10.0, meaning that on average, about 10
      false positives will be reported per query, so you can see the top of the
      noise and decide for yourself if it's really noise.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Instead of thresholding output on E-value, instead report
      target profiles with a bit score of &gt;= <i>&lt;x&gt;</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_INCLUSION_THRESHOLDS"><a class="selflink" href="#OPTIONS_FOR_INCLUSION_THRESHOLDS">OPTIONS
  FOR INCLUSION THRESHOLDS</a></h1>
Inclusion thresholds are stricter than reporting thresholds. Inclusion
  thresholds control which hits are considered to be reliable enough to be
  included in an output alignment or a subsequent search round. In
  <b>nhmmscan</b>, which does not have any alignment output (like
  <b>nhmmer</b>), inclusion thresholds have little effect. They only affect what
  hits get marked as significant (!) or questionable (?) in hit output.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Use an E-value of &lt;= <i>&lt;x&gt;</i> as the inclusion
      threshold. The default is 0.01, meaning that on average, about 1 false
      positive would be expected in every 100 searches with different query
      sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Instead of using E-values for setting the inclusion
      threshold, use a bit score of &gt;= <i>&lt;x&gt;</i> as the inclusion
      threshold. It would be unusual to use bit score thresholds with
      <i>hmmscan</i>, because you don't expect a single score threshold to work
      for different profiles; different profiles have slightly different
      expected score distributions.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_MODEL-SPECIFIC_SCORE_THRESHOLDING"><a class="selflink" href="#OPTIONS_FOR_MODEL-SPECIFIC_SCORE_THRESHOLDING">OPTIONS
  FOR MODEL-SPECIFIC SCORE THRESHOLDING</a></h1>
Curated profile databases may define specific bit score thresholds for each
  profile, superseding any thresholding based on statistical significance alone.
<div style="height: 1.00em;">&#x00A0;</div>
To use these options, the profile must contain the appropriate (GA, TC, and/or
  NC) optional score threshold annotation; this is picked up by <b>hmmbuild</b>
  from Stockholm format alignment files. For a nucleotide model, each
  thresholding option has a single per-hit threshold &lt;x&gt; This acts as if
  <b>-T</b><i>&lt;x&gt;</i> <b>--incT</b><i>&lt;x&gt;</i> has been applied
  specifically using each model's curated thresholds.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_ga</b></dt>
  <dd class="It-tag">Use the GA (gathering) bit score threshold in the model to
      set per-hit reporting and inclusion thresholds. GA thresholds are
      generally considered to be the reliable curated thresholds defining family
      membership; for example, in Dfam, these thresholds are applied when
      annotating a genome with a model of a family known to be found in that
      organism. They may allow for minimal expected false discovery rate.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_nc</b></dt>
  <dd class="It-tag">Use the NC (noise cutoff) bit score threshold in the model
      to set per-hit reporting and inclusion thresholds. NC thresholds are less
      stringent than GA; in the context of Pfam, they are generally used to
      store the score of the highest-scoring known false positive.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cut_tc</b></dt>
  <dd class="It-tag">Use the NC (trusted cutoff) bit score threshold in the
      model to set per-hit reporting and inclusion thresholds. TC thresholds are
      more stringent than GA, and are generally considered to be the score of
      the lowest-scoring known true positive that is above all known false
      positives; for example, in Dfam, these thresholds are applied when
      annotating a genome with a model of a family not known to be found in that
      organism.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="CONTROL_OF_THE_ACCELERATION_PIPELINE"><a class="selflink" href="#CONTROL_OF_THE_ACCELERATION_PIPELINE">CONTROL
  OF THE ACCELERATION PIPELINE</a></h1>
HMMER3 searches are accelerated in a three-step filter pipeline: the
  scanning-SSV filter, the Viterbi filter, and the Forward filter. The first
  filter is the fastest and most approximate; the last is the full Forward
  scoring algorithm. There is also a bias filter step between SSV and Viterbi.
  Targets that pass all the steps in the acceleration pipeline are then
  subjected to postprocessing -- domain identification and scoring using the
  Forward/Backward algorithm.
<div style="height: 1.00em;">&#x00A0;</div>
Changing filter thresholds only removes or includes targets from consideration;
  changing filter thresholds does not alter bit scores, E-values, or alignments,
  all of which are determined solely in postprocessing.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max</b></dt>
  <dd class="It-tag">Turn off (nearly) all filters, including the bias filter,
      and run full Forward/Backward postprocessing on most of the target
      sequence. In contrast to <b>hmmscan,</b> where this flag really does turn
      off the filters entirely, the <i>--max</i> flag in <b>nhmmscan</b> sets
      the scanning-SSV filter threshold to 0.4, not 1.0. Use of this flag
      increases sensitivity somewhat, at a large cost in speed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F1</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the P-value threshold for the MSV filter step. The
      default is 0.02, meaning that roughly 2% of the highest scoring
      nonhomologous targets are expected to pass the filter.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F2</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the P-value threshold for the Viterbi filter step. The
      default is 0.001.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F3</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the P-value threshold for the Forward filter step. The
      default is 1e-5.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nobias</b></dt>
  <dd class="It-tag">Turn off the bias filter. This increases sensitivity
      somewhat, but can come at a high cost in speed, especially if the query
      has biased residue composition (such as a repetitive sequence region, or
      if it is a membrane protein with large regions of hydrophobicity). Without
      the bias filter, too many sequences may pass the filter with biased
      queries, leading to slower than expected performance as the
      computationally intensive Forward/Backward algorithms shoulder an
      abnormally heavy load.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_OPTIONS"><a class="selflink" href="#OTHER_OPTIONS">OTHER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nonull2</b></dt>
  <dd class="It-tag">Turn off the null2 score corrections for biased
      composition.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Assert that the total number of targets in your searches is
      <i>&lt;x&gt;</i>, for the purposes of per-sequence E-value calculations,
      rather than the actual number of targets seen.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the random number seed to <i>&lt;n&gt;</i>. Some steps
      in postprocessing require Monte Carlo simulation. The default is to use a
      fixed seed (42), so that results are exactly reproducible. Any other
      positive integer will give different (but also reproducible) results. A
      choice of 0 uses an arbitrarily chosen seed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--qformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Assert that the query sequence file is in format
      <i>&lt;s&gt;</i>. Accepted formats include <i>fasta</i>, <i>embl</i>,
      <i>genbank</i>, <i>ddbj</i>, <i>uniprot</i>, <i>stockholm</i>,
      <i>pfam</i>, <i>a2m</i>, and <i>afa</i>. The default is to autodetect the
      format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--w_beta</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Window length tail mass. The upper bound, <i>W</i>, on the
      length at which nhmmer expects to find an instance of the model is set
      such that the fraction of all sequences generated by the model with length
      <i>&gt;= W</i> is less than <i>&lt;x&gt;</i>. The default is 1e-7. This
      flag may be used to override the value of <i>W</i> established for the
      model by <b>hmmbuild</b>.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--w_length</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Override the model instance length upper bound, <i>W</i>,
      which is otherwise controlled by <b>--w_beta</b>. It should be larger than
      the model length. The value of <i>W</i> is used deep in the acceleration
      pipeline, and modest changes are not expected to impact results (though
      larger values of <i>W</i> do lead to longer run time). This flag may be
      used to override the value of <i>W</i> established for the model by
      <b>hmmbuild</b>.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--toponly</b></dt>
  <dd class="It-tag">Only search the top strand. By default both the query
      sequence and its reverse-complement are searched.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--bottomonly</b></dt>
  <dd class="It-tag">Only search the bottom (reverse-complement) strand. By
      default both the query sequence and its reverse-complement are searched.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cpu</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the number of parallel worker threads to
      <i>&lt;n&gt;</i>. By default, HMMER sets this to the number of CPU cores
      it detects in your machine - that is, it tries to maximize the use of your
      available processor cores. Setting <i>&lt;n&gt;</i> higher than the number
      of available cores is of little if any value, but you may want to set it
      to something less. You can also control this number by setting an
      environment variable, <i>HMMER_NCPU</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    This option is only available if HMMER was compiled with POSIX threads
      support. This is the default, but it may have been turned off for your
      site or machine for some reason.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stall</b></dt>
  <dd class="It-tag">For debugging the MPI master/worker version: pause after
      start, to enable the developer to attach debuggers to the running master
      and worker(s) processes. Send SIGCONT signal to release the pause. (Under
      gdb: <i>(gdb) signal SIGCONT</i>)
    <div style="height: 1.00em;">&#x00A0;</div>
    (Only available if optional MPI support was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mpi</b></dt>
  <dd class="It-tag">Run in MPI master/worker mode, using <i>mpirun</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    (Only available if optional MPI support was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>hmmer(1)</b> for a master man page with a list of all the individual man
  pages for programs in the HMMER package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your HMMER
  distribution (Userguide.pdf); or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2015 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2015</td>
    <td class="foot-os">HMMER 3.1b2</td>
  </tr>
</table>
</body>
</html>
