<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:29:57 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>PARSEVAL(1) User Commands PARSEVAL(1)</p>

<p style="margin-top: 1em">NAME <br>
parseval - comparative analysis of two alternative sources
of annotation</p>

<p style="margin-top: 1em">SYNOPSIS <br>
parseval [options] reference.gff3 prediction.gff3</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Basic options:</p>

<p style="margin-top: 1em">-d|--debug: <br>
Print debugging messages</p>

<p style="margin-top: 1em">-h|--help: <br>
Print help message and exit</p>

<p style="margin-top: 1em">-l|--delta: INT <br>
Extend gene loci by this many nucleotides; default is 0</p>

<p style="margin-top: 1em">-V|--verbose: <br>
Print verbose warning messages</p>

<p style="margin-top: 1em">-v|--version: <br>
Print version number and exit</p>

<p style="margin-top: 1em">Output options:</p>

<p style="margin-top: 1em">-a|--datashare: STRING <br>
Location from which to copy shared data for HTML output (if
&lsquo;make install&rsquo; has not yet been run)</p>

<p style="margin-top: 1em">-f|--outformat: STRING <br>
Indicate desired output format; possible options:
&rsquo;csv&rsquo;, &rsquo;text&rsquo;, or &rsquo;html&rsquo;
(default=&rsquo;text&rsquo;); in &rsquo;text&rsquo; or
&rsquo;csv&rsquo; mode, will create a single file; in
&rsquo;html&rsquo; mode, will cre&acirc; <br>
ate a directory</p>

<p style="margin-top: 1em">-g|--nogff3: <br>
Do no print GFF3 output corresponding to each comparison</p>

<p style="margin-top: 1em">-o|--outfile: FILENAME <br>
File/directory to which output will be written; default is
the terminal (STDOUT)</p>

<p style="margin-top: 1em">-p|--nopng: <br>
In HTML output mode, skip generation of PNG graphics for
each gene locus</p>

<p style="margin-top: 1em">-s|--summary: <br>
Only print summary statistics, do not print individual
comparisons</p>

<p style="margin-top: 1em">-w|--overwrite: <br>
Force overwrite of any existing output files</p>

<p style="margin-top: 1em">-x|--refrlabel: STRING <br>
Optional label for reference annotations</p>

<p style="margin-top: 1em">-y|--predlabel: STRING <br>
Optional label for prediction annotations</p>

<p style="margin-top: 1em">Filtering options:</p>

<p style="margin-top: 1em">-k|--makefilter <br>
Create a default configuration file for filtering reported
results and quit, performing no comparisons</p>

<p style="margin-top: 1em">-r|--filterfile: STRING <br>
Use the indicated configuration file to filter reported
results;</p>

<p style="margin-top: 1em">-t|--maxtrans: INT <br>
Maximum transcripts allowed per locus; use 0 to disable
limit; default is 32</p>

<p style="margin-top: 1em">parseval 0.15.1 January 2016
PARSEVAL(1)</p>
<hr>
</body>
</html>
