<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>phmmer(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">phmmer(1)</td>
    <td class="head-vol">HMMER Manual</td>
    <td class="head-rtitle">phmmer(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
phmmer - search protein sequence(s) against a protein sequence database
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>phmmer</b> <i>[options]</i> <i>&lt;seqfile&gt;</i> <i>&lt;seqdb&gt;</i>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>phmmer</b> is used to search one or more query protein sequences against a
  protein sequence database. For each query sequence in <i>&lt;seqfile&gt;</i>,
  use that sequence to search the target database of sequences in
  <i>&lt;seqdb&gt;</i>, and output ranked lists of the sequences with the most
  significant matches to the query.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Either the query <i>&lt;seqfile&gt;</i> or the target <i>&lt;seqdb&gt;</i> may
  be '-' (a dash character), in which case the query sequences or target
  database input will be read from a &lt;stdin&gt; pipe instead of from a file.
  Only one input source can come through &lt;stdin&gt;, not both. An exception
  is that if the <i>&lt;seqfile&gt;</i> contains more than one query sequence,
  then <i>&lt;seqdb&gt;</i> cannot come from &lt;stdin&gt;, because we can't
  rewind the streaming target database to search it with another query.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The output format is designed to be human-readable, but is often so voluminous
  that reading it is impractical, and parsing it is a pain. The <b>--tblout</b>
  and <b>--domtblout</b> options save output in simple tabular formats that are
  concise and easier to parse. The <b>-o</b> option allows redirecting the main
  output, including throwing it away in /dev/null.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Help; print a brief reminder of command line usage and all
      available options.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_FOR_CONTROLLING_OUTPUT"><a class="selflink" href="#OPTIONS_FOR_CONTROLLING_OUTPUT">OPTIONS
  FOR CONTROLLING OUTPUT</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Direct the main human-readable output to a file
      <i>&lt;f&gt;</i> instead of the default stdout.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save a multiple alignment of all significant hits (those
      satisfying inclusion thresholds) to the file <i>&lt;f&gt;</i> in Stockholm
      format.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tblout</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save a simple tabular (space-delimited) file summarizing
      the per-target output, with one data line per homologous target sequence
      found.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domtblout</b><i> &lt;f&gt;</i></dt>
  <dd class="It-tag">Save a simple tabular (space-delimited) file summarizing
      the per-domain output, with one data line per homologous domain detected
      in a query sequence for each homologous model.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--acc</b></dt>
  <dd class="It-tag">Use accessions instead of names in the main output, where
      available for profiles and/or sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--noali</b></dt>
  <dd class="It-tag">Omit the alignment section from the main output. This can
      greatly reduce the output volume.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--notextw</b></dt>
  <dd class="It-tag">Unlimit the length of each line in the main output. The
      default is a limit of 120 characters per line, which helps in displaying
      the output cleanly on terminals and in editors, but can truncate target
      profile description lines.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--textw</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the main output's line length limit to <i>&lt;n&gt;</i>
      characters per line. The default is 120.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_SCORING_SYSTEM"><a class="selflink" href="#OPTIONS_CONTROLLING_SCORING_SYSTEM">OPTIONS
  CONTROLLING SCORING SYSTEM</a></h1>
The probability model in <b>phmmer</b> is constructed by inferring residue
  probabilities from a standard 20x20 substitution score matrix, plus two
  additional parameters for position-independent gap open and gap extend
  probabilities.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--popen</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the gap open probability for a single sequence query
      model to <i>&lt;x&gt;</i>. The default is 0.02. <i>&lt;x&gt;</i> must be
      &gt;= 0 and &lt; 0.5.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pextend</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Set the gap extend probability for a single sequence query
      model to <i>&lt;x&gt;</i>. The default is 0.4. <i>&lt;x&gt;</i> must be
      &gt;= 0 and &lt; 1.0.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mx</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Obtain residue alignment probabilities from the built-in
      substitution matrix named <i>&lt;s&gt;</i>. Several standard matrices are
      built-in, and do not need to be read from files. The matrix name
      <i>&lt;s&gt;</i> can be PAM30, PAM70, PAM120, PAM240, BLOSUM45, BLOSUM50,
      BLOSUM62, BLOSUM80, or BLOSUM90. Only one of the <b>--mx</b> and
      <b>--mxfile</b> options may be used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mxfile</b><i> &lt;mxfile&gt;</i></dt>
  <dd class="It-tag">Obtain residue alignment probabilities from the
      substitution matrix in file <i>&lt;mxfile&gt;</i>. The default score
      matrix is BLOSUM62 (this matrix is internal to HMMER and does not have to
      be available as a file). The format of a substitution matrix
      <i>&lt;mxfile&gt;</i> is the standard format accepted by BLAST, FASTA, and
      other sequence analysis software. Only one of the <b>--mx</b> and
      <b>--mxfile</b> options may be used.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_REPORTING_THRESHOLDS"><a class="selflink" href="#OPTIONS_CONTROLLING_REPORTING_THRESHOLDS">OPTIONS
  CONTROLLING REPORTING THRESHOLDS</a></h1>
Reporting thresholds control which hits are reported in output files (the main
  output, <i>--tblout</i>, and <i>--domtblout</i>). Sequence hits and domain
  hits are ranked by statistical significance (E-value) and output is generated
  in two sections called per-target and per-domain output. In per-target output,
  by default, all sequence hits with an E-value &lt;= 10 are reported. In the
  per-domain output, for each target that has passed per-target reporting
  thresholds, all domains satisfying per-domain reporting thresholds are
  reported. By default, these are domains with conditional E-values of &lt;= 10.
  The following options allow you to change the default E-value reporting
  thresholds, or to use bit score thresholds instead.
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">In the per-target output, report target sequences with an
      E-value of &lt;= <i>&lt;x&gt;</i>. The default is 10.0, meaning that on
      average, about 10 false positives will be reported per query, so you can
      see the top of the noise and decide for yourself if it's really noise.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Instead of thresholding per-profile output on E-value,
      instead report target sequences with a bit score of &gt;=
      <i>&lt;x&gt;</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">In the per-domain output, for target sequences that have
      already satisfied the per-profile reporting threshold, report individual
      domains with a conditional E-value of &lt;= <i>&lt;x&gt;</i>. The default
      is 10.0. A conditional E-value means the expected number of additional
      false positive domains in the smaller search space of those comparisons
      that already satisfied the per-target reporting threshold (and thus must
      have at least one homologous domain already).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Instead of thresholding per-domain output on E-value,
      instead report domains with a bit score of &gt;= <i>&lt;x&gt;</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_INCLUSION_THRESHOLDS"><a class="selflink" href="#OPTIONS_CONTROLLING_INCLUSION_THRESHOLDS">OPTIONS
  CONTROLLING INCLUSION THRESHOLDS</a></h1>
Inclusion thresholds are stricter than reporting thresholds. They control which
  hits are included in any output multiple alignment (the <i>-A</i> option) and
  which domains are marked as significant (&quot;!&quot;) as opposed to
  questionable (&quot;?&quot;) in domain output.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Use an E-value of &lt;= <i>&lt;x&gt;</i> as the per-target
      inclusion threshold. The default is 0.01, meaning that on average, about 1
      false positive would be expected in every 100 searches with different
      query sequences.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Instead of using E-values for setting the inclusion
      threshold, instead use a bit score of &gt;= <i>&lt;x&gt;</i> as the
      per-target inclusion threshold. By default this option is unset.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incdomE</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Use a conditional E-value of &lt;= <i>&lt;x&gt;</i> as the
      per-domain inclusion threshold, in targets that have already satisfied the
      overall per-target inclusion threshold. The default is 0.01.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--incdomT</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Instead of using E-values, use a bit score of &gt;=
      <i>&lt;x&gt;</i> as the per-domain inclusion threshold. By default this
      option is unset.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_THE_ACCELERATION_PIPELINE"><a class="selflink" href="#OPTIONS_CONTROLLING_THE_ACCELERATION_PIPELINE">OPTIONS
  CONTROLLING THE ACCELERATION PIPELINE</a></h1>
HMMER3 searches are accelerated in a three-step filter pipeline: the MSV filter,
  the Viterbi filter, and the Forward filter. The first filter is the fastest
  and most approximate; the last is the full Forward scoring algorithm, slowest
  but most accurate. There is also a bias filter step between MSV and Viterbi.
  Targets that pass all the steps in the acceleration pipeline are then
  subjected to postprocessing -- domain identification and scoring using the
  Forward/Backward algorithm.
<div style="height: 1.00em;">&#x00A0;</div>
Essentially the only free parameters that control HMMER's heuristic filters are
  the P-value thresholds controlling the expected fraction of nonhomologous
  sequences that pass the filters. Setting the default thresholds higher will
  pass a higher proportion of nonhomologous sequence, increasing sensitivity at
  the expense of speed; conversely, setting lower P-value thresholds will pass a
  smaller proportion, decreasing sensitivity and increasing speed. Setting a
  filter's P-value threshold to 1.0 means it will passing all sequences, and
  effectively disables the filter.
<div style="height: 1.00em;">&#x00A0;</div>
Changing filter thresholds only removes or includes targets from consideration;
  changing filter thresholds does not alter bit scores, E-values, or alignments,
  all of which are determined solely in postprocessing.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max</b></dt>
  <dd class="It-tag">Maximum sensitivity. Turn off all filters, including the
      bias filter, and run full Forward/Backward postprocessing on every target.
      This increases sensitivity slightly, at a large cost in speed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F1</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">First filter threshold; set the P-value threshold for the
      MSV filter step. The default is 0.02, meaning that roughly 2% of the
      highest scoring nonhomologous targets are expected to pass the filter.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F2</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Second filter threshold; set the P-value threshold for the
      Viterbi filter step. The default is 0.001.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--F3</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Third filter threshold; set the P-value threshold for the
      Forward filter step. The default is 1e-5.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nobias</b></dt>
  <dd class="It-tag">Turn off the bias filter. This increases sensitivity
      somewhat, but can come at a high cost in speed, especially if the query
      has biased residue composition (such as a repetitive sequence region, or
      if it is a membrane protein with large regions of hydrophobicity). Without
      the bias filter, too many sequences may pass the filter with biased
      queries, leading to slower than expected performance as the
      computationally intensive Forward/Backward algorithms shoulder an
      abnormally heavy load.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS_CONTROLLING_E-VALUE_CALIBRATION"><a class="selflink" href="#OPTIONS_CONTROLLING_E-VALUE_CALIBRATION">OPTIONS
  CONTROLLING E-VALUE CALIBRATION</a></h1>
Estimating the location parameters for the expected score distributions for MSV
  filter scores, Viterbi filter scores, and Forward scores requires three short
  random sequence simulations.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EmL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter mu for MSV filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EmN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter mu for MSV filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EvL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter mu for Viterbi filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EvN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter mu for Viterbi filter E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EfL</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the sequence length in simulation that estimates the
      location parameter tau for Forward E-values. Default is 100.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--EfN</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Sets the number of sequences in simulation that estimates
      the location parameter tau for Forward E-values. Default is 200.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--Eft</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Sets the tail mass fraction to fit in the simulation that
      estimates the location parameter tau for Forward evalues. Default is 0.04.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="OTHER_OPTIONS"><a class="selflink" href="#OTHER_OPTIONS">OTHER
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nonull2</b></dt>
  <dd class="It-tag">Turn off the null2 score corrections for biased
      composition.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Z</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Assert that the total number of targets in your searches is
      <i>&lt;x&gt;</i>, for the purposes of per-sequence E-value calculations,
      rather than the actual number of targets seen.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--domZ</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Assert that the total number of targets in your searches is
      <i>&lt;x&gt;</i>, for the purposes of per-domain conditional E-value
      calculations, rather than the number of targets that passed the reporting
      thresholds.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Seed the random number generator with <i>&lt;n&gt;</i>, an
      integer &gt;= 0. If <i>&lt;n&gt;</i> is &gt;0, any stochastic simulations
      will be reproducible; the same command will give the same results. If
      <i>&lt;n&gt;</i> is 0, the random number generator is seeded arbitrarily,
      and stochastic simulations will vary from run to run of the same command.
      The default seed is 42.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--qformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Declare that the input <i>&lt;seqfile&gt;</i> is in format
      <i>&lt;s&gt;</i>. Accepted formats include <i>fasta</i>, <i>embl</i>,
      <i>genbank</i>, <i>ddbj</i>, <i>uniprot</i>, <i>stockholm</i>,
      <i>pfam</i>, <i>a2m</i>, and <i>afa</i>. The default is to autodetect the
      format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tformat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Declare that the input <i>&lt;seqdb&gt;</i> is in format
      <i>&lt;s&gt;</i>. Accepted formats include <i>fasta</i>, <i>embl</i>,
      <i>genbank</i>, <i>ddbj</i>, <i>uniprot</i>, <i>stockholm</i>,
      <i>pfam</i>, <i>a2m</i>, and <i>afa</i>. The default is to autodetect the
      format of the file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--cpu</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Set the number of parallel worker threads to
      <i>&lt;n&gt;</i>. By default, HMMER sets this to the number of CPU cores
      it detects in your machine - that is, it tries to maximize the use of your
      available processor cores. Setting <i>&lt;n&gt;</i> higher than the number
      of available cores is of little if any value, but you may want to set it
      to something less. You can also control this number by setting an
      environment variable, <i>HMMER_NCPU</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    This option is only available if HMMER was compiled with POSIX threads
      support. This is the default, but it may have been turned off at
      compile-time for your site or machine for some reason.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--stall</b></dt>
  <dd class="It-tag">For debugging the MPI master/worker version: pause after
      start, to enable the developer to attach debuggers to the running master
      and worker(s) processes. Send SIGCONT signal to release the pause. (Under
      gdb: <i>(gdb) signal SIGCONT</i>) (Only available if optional MPI support
      was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mpi</b></dt>
  <dd class="It-tag">Run in MPI master/worker mode, using <i>mpirun</i>. (Only
      available if optional MPI support was enabled at compile-time.)
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
See <b>hmmer(1)</b> for a master man page with a list of all the individual man
  pages for programs in the HMMER package.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
For complete documentation, see the user guide that came with your HMMER
  distribution (Userguide.pdf); or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<pre>
Copyright (C) 2015 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
For additional information on copyright and licensing, see the file called
  COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">February 2015</td>
    <td class="foot-os">HMMER 3.1b2</td>
  </tr>
</table>
</body>
</html>
