<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:31:31 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>phmmer(1) HMMER Manual phmmer(1)</p>

<p style="margin-top: 1em">NAME <br>
phmmer - search protein sequence(s) against a protein
sequence database</p>

<p style="margin-top: 1em">SYNOPSIS <br>
phmmer [options] &lt;seqfile&gt; &lt;seqdb&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
phmmer is used to search one or more query protein sequences
against a protein sequence database. For each query sequence
in &lt;seqfile&gt;, use that sequence to search the target
<br>
database of sequences in &lt;seqdb&gt;, and output ranked
lists of the sequences with the most significant matches to
the query.</p>

<p style="margin-top: 1em">Either the query &lt;seqfile&gt;
or the target &lt;seqdb&gt; may be &rsquo;-&rsquo; (a dash
character), in which case the query sequences or target
database input will be read from a &lt;stdin&gt; pipe <br>
instead of from a file. Only one input source can come
through &lt;stdin&gt;, not both. An exception is that if the
&lt;seqfile&gt; contains more than one query sequence, then
&lt;seqdb&gt; can&acirc; <br>
not come from &lt;stdin&gt;, because we can&rsquo;t rewind
the streaming target database to search it with another
query.</p>

<p style="margin-top: 1em">The output format is designed to
be human-readable, but is often so voluminous that reading
it is impractical, and parsing it is a pain. The --tblout
and --domtblout options save <br>
output in simple tabular formats that are concise and easier
to parse. The -o option allows redirecting the main output,
including throwing it away in /dev/null.</p>

<p style="margin-top: 1em">OPTIONS <br>
-h Help; print a brief reminder of command line usage and
all available options.</p>

<p style="margin-top: 1em">OPTIONS FOR CONTROLLING OUTPUT
<br>
-o &lt;f&gt; Direct the main human-readable output to a file
&lt;f&gt; instead of the default stdout.</p>

<p style="margin-top: 1em">-A &lt;f&gt; Save a multiple
alignment of all significant hits (those satisfying
inclusion thresholds) to the file &lt;f&gt; in Stockholm
format.</p>

<p style="margin-top: 1em">--tblout &lt;f&gt; <br>
Save a simple tabular (space-delimited) file summarizing the
per-target output, with one data line per homologous target
sequence found.</p>

<p style="margin-top: 1em">--domtblout &lt;f&gt; <br>
Save a simple tabular (space-delimited) file summarizing the
per-domain output, with one data line per homologous domain
detected in a query sequence for each homologous <br>
model.</p>

<p style="margin-top: 1em">--acc Use accessions instead of
names in the main output, where available for profiles
and/or sequences.</p>

<p style="margin-top: 1em">--noali <br>
Omit the alignment section from the main output. This can
greatly reduce the output volume.</p>

<p style="margin-top: 1em">--notextw <br>
Unlimit the length of each line in the main output. The
default is a limit of 120 characters per line, which helps
in displaying the output cleanly on terminals and in <br>
editors, but can truncate target profile description
lines.</p>

<p style="margin-top: 1em">--textw &lt;n&gt; <br>
Set the main output&rsquo;s line length limit to &lt;n&gt;
characters per line. The default is 120.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING SCORING
SYSTEM <br>
The probability model in phmmer is constructed by inferring
residue probabilities from a standard 20x20 substitution
score matrix, plus two additional parameters for position-
<br>
independent gap open and gap extend probabilities.</p>

<p style="margin-top: 1em">--popen &lt;x&gt; <br>
Set the gap open probability for a single sequence query
model to &lt;x&gt;. The default is 0.02. &lt;x&gt; must be
&gt;= 0 and &lt; 0.5.</p>

<p style="margin-top: 1em">--pextend &lt;x&gt; <br>
Set the gap extend probability for a single sequence query
model to &lt;x&gt;. The default is 0.4. &lt;x&gt; must be
&gt;= 0 and &lt; 1.0.</p>

<p style="margin-top: 1em">--mx &lt;s&gt; <br>
Obtain residue alignment probabilities from the built-in
substitution matrix named &lt;s&gt;. Several standard
matrices are built-in, and do not need to be read from
files. <br>
The matrix name &lt;s&gt; can be PAM30, PAM70, PAM120,
PAM240, BLOSUM45, BLOSUM50, BLOSUM62, BLOSUM80, or BLOSUM90.
Only one of the --mx and --mxfile options may be used.</p>

<p style="margin-top: 1em">--mxfile &lt;mxfile&gt; <br>
Obtain residue alignment probabilities from the substitution
matrix in file &lt;mxfile&gt;. The default score matrix is
BLOSUM62 (this matrix is internal to HMMER and does not <br>
have to be available as a file). The format of a
substitution matrix &lt;mxfile&gt; is the standard format
accepted by BLAST, FASTA, and other sequence analysis
software. Only <br>
one of the --mx and --mxfile options may be used.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING REPORTING
THRESHOLDS <br>
Reporting thresholds control which hits are reported in
output files (the main output, --tblout, and --domtblout).
Sequence hits and domain hits are ranked by statistical
sig&acirc; <br>
nificance (E-value) and output is generated in two sections
called per-target and per-domain output. In per-target
output, by default, all sequence hits with an E-value &lt;=
10 are <br>
reported. In the per-domain output, for each target that has
passed per-target reporting thresholds, all domains
satisfying per-domain reporting thresholds are reported. By
<br>
default, these are domains with conditional E-values of
&lt;= 10. The following options allow you to change the
default E-value reporting thresholds, or to use bit score
thresholds <br>
instead.</p>

<p style="margin-top: 1em">-E &lt;x&gt; In the per-target
output, report target sequences with an E-value of &lt;=
&lt;x&gt;. The default is 10.0, meaning that on average,
about 10 false positives will be reported per <br>
query, so you can see the top of the noise and decide for
yourself if it&rsquo;s really noise.</p>

<p style="margin-top: 1em">-T &lt;x&gt; Instead of
thresholding per-profile output on E-value, instead report
target sequences with a bit score of &gt;= &lt;x&gt;.</p>

<p style="margin-top: 1em">--domE &lt;x&gt; <br>
In the per-domain output, for target sequences that have
already satisfied the per-profile reporting threshold,
report individual domains with a conditional E-value of
&lt;= <br>
&lt;x&gt;. The default is 10.0. A conditional E-value means
the expected number of additional false positive domains in
the smaller search space of those comparisons that <br>
already satisfied the per-target reporting threshold (and
thus must have at least one homologous domain already).</p>

<p style="margin-top: 1em">--domT &lt;x&gt; <br>
Instead of thresholding per-domain output on E-value,
instead report domains with a bit score of &gt;=
&lt;x&gt;.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING INCLUSION
THRESHOLDS <br>
Inclusion thresholds are stricter than reporting thresholds.
They control which hits are included in any output multiple
alignment (the -A option) and which domains are marked as
<br>
significant (&quot;!&quot;) as opposed to questionable
(&quot;?&quot;) in domain output.</p>

<p style="margin-top: 1em">--incE &lt;x&gt; <br>
Use an E-value of &lt;= &lt;x&gt; as the per-target
inclusion threshold. The default is 0.01, meaning that on
average, about 1 false positive would be expected in every
100 <br>
searches with different query sequences.</p>

<p style="margin-top: 1em">--incT &lt;x&gt; <br>
Instead of using E-values for setting the inclusion
threshold, instead use a bit score of &gt;= &lt;x&gt; as the
per-target inclusion threshold. By default this option is
unset.</p>

<p style="margin-top: 1em">--incdomE &lt;x&gt; <br>
Use a conditional E-value of &lt;= &lt;x&gt; as the
per-domain inclusion threshold, in targets that have already
satisfied the overall per-target inclusion threshold. The
default <br>
is 0.01.</p>

<p style="margin-top: 1em">--incdomT &lt;x&gt; <br>
Instead of using E-values, use a bit score of &gt;=
&lt;x&gt; as the per-domain inclusion threshold. By default
this option is unset.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING THE
ACCELERATION PIPELINE <br>
HMMER3 searches are accelerated in a three-step filter
pipeline: the MSV filter, the Viterbi filter, and the
Forward filter. The first filter is the fastest and most
approximate; <br>
the last is the full Forward scoring algorithm, slowest but
most accurate. There is also a bias filter step between MSV
and Viterbi. Targets that pass all the steps in the
accel&acirc; <br>
eration pipeline are then subjected to postprocessing --
domain identification and scoring using the Forward/Backward
algorithm.</p>

<p style="margin-top: 1em">Essentially the only free
parameters that control HMMER&rsquo;s heuristic filters are
the P-value thresholds controlling the expected fraction of
nonhomologous sequences that pass the <br>
filters. Setting the default thresholds higher will pass a
higher proportion of nonhomologous sequence, increasing
sensitivity at the expense of speed; conversely, setting
lower <br>
P-value thresholds will pass a smaller proportion,
decreasing sensitivity and increasing speed. Setting a
filter&rsquo;s P-value threshold to 1.0 means it will
passing all sequences, <br>
and effectively disables the filter.</p>

<p style="margin-top: 1em">Changing filter thresholds only
removes or includes targets from consideration; changing
filter thresholds does not alter bit scores, E-values, or
alignments, all of which are <br>
determined solely in postprocessing.</p>

<p style="margin-top: 1em">--max Maximum sensitivity. Turn
off all filters, including the bias filter, and run full
Forward/Backward postprocessing on every target. This
increases sensitivity slightly, <br>
at a large cost in speed.</p>

<p style="margin-top: 1em">--F1 &lt;x&gt; <br>
First filter threshold; set the P-value threshold for the
MSV filter step. The default is 0.02, meaning that roughly
2% of the highest scoring nonhomologous targets are <br>
expected to pass the filter.</p>

<p style="margin-top: 1em">--F2 &lt;x&gt; <br>
Second filter threshold; set the P-value threshold for the
Viterbi filter step. The default is 0.001.</p>

<p style="margin-top: 1em">--F3 &lt;x&gt; <br>
Third filter threshold; set the P-value threshold for the
Forward filter step. The default is 1e-5.</p>

<p style="margin-top: 1em">--nobias <br>
Turn off the bias filter. This increases sensitivity
somewhat, but can come at a high cost in speed, especially
if the query has biased residue composition (such as a <br>
repetitive sequence region, or if it is a membrane protein
with large regions of hydrophobicity). Without the bias
filter, too many sequences may pass the filter with <br>
biased queries, leading to slower than expected performance
as the computationally intensive Forward/Backward algorithms
shoulder an abnormally heavy load.</p>

<p style="margin-top: 1em">OPTIONS CONTROLLING E-VALUE
CALIBRATION <br>
Estimating the location parameters for the expected score
distributions for MSV filter scores, Viterbi filter scores,
and Forward scores requires three short random sequence
sim&acirc; <br>
ulations.</p>

<p style="margin-top: 1em">--EmL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter mu for MSV filter E-values. Default is
200.</p>

<p style="margin-top: 1em">--EmN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter mu for MSV filter E-values. Default
is 200.</p>

<p style="margin-top: 1em">--EvL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter mu for Viterbi filter E-values. Default
is 200.</p>

<p style="margin-top: 1em">--EvN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter mu for Viterbi filter E-values.
Default is 200.</p>

<p style="margin-top: 1em">--EfL &lt;n&gt; <br>
Sets the sequence length in simulation that estimates the
location parameter tau for Forward E-values. Default is
100.</p>

<p style="margin-top: 1em">--EfN &lt;n&gt; <br>
Sets the number of sequences in simulation that estimates
the location parameter tau for Forward E-values. Default is
200.</p>

<p style="margin-top: 1em">--Eft &lt;x&gt; <br>
Sets the tail mass fraction to fit in the simulation that
estimates the location parameter tau for Forward evalues.
Default is 0.04.</p>

<p style="margin-top: 1em">OTHER OPTIONS <br>
--nonull2 <br>
Turn off the null2 score corrections for biased
composition.</p>

<p style="margin-top: 1em">-Z &lt;x&gt; Assert that the
total number of targets in your searches is &lt;x&gt;, for
the purposes of per-sequence E-value calculations, rather
than the actual number of targets seen.</p>

<p style="margin-top: 1em">--domZ &lt;x&gt; <br>
Assert that the total number of targets in your searches is
&lt;x&gt;, for the purposes of per-domain conditional
E-value calculations, rather than the number of targets that
<br>
passed the reporting thresholds.</p>

<p style="margin-top: 1em">--seed &lt;n&gt; <br>
Seed the random number generator with &lt;n&gt;, an integer
&gt;= 0. If &lt;n&gt; is &gt;0, any stochastic simulations
will be reproducible; the same command will give the same
results. <br>
If &lt;n&gt; is 0, the random number generator is seeded
arbitrarily, and stochastic simulations will vary from run
to run of the same command. The default seed is 42.</p>

<p style="margin-top: 1em">--qformat &lt;s&gt; <br>
Declare that the input &lt;seqfile&gt; is in format
&lt;s&gt;. Accepted formats include fasta, embl, genbank,
ddbj, uniprot, stockholm, pfam, a2m, and afa. The default is
to autode&acirc; <br>
tect the format of the file.</p>

<p style="margin-top: 1em">--tformat &lt;s&gt; <br>
Declare that the input &lt;seqdb&gt; is in format &lt;s&gt;.
Accepted formats include fasta, embl, genbank, ddbj,
uniprot, stockholm, pfam, a2m, and afa. The default is to
autode&acirc; <br>
tect the format of the file.</p>

<p style="margin-top: 1em">--cpu &lt;n&gt; <br>
Set the number of parallel worker threads to &lt;n&gt;. By
default, HMMER sets this to the number of CPU cores it
detects in your machine - that is, it tries to maximize the
<br>
use of your available processor cores. Setting &lt;n&gt;
higher than the number of available cores is of little if
any value, but you may want to set it to something less. You
<br>
can also control this number by setting an environment
variable, HMMER_NCPU.</p>

<p style="margin-top: 1em">This option is only available if
HMMER was compiled with POSIX threads support. This is the
default, but it may have been turned off at compile-time for
your site or <br>
machine for some reason.</p>

<p style="margin-top: 1em">--stall <br>
For debugging the MPI master/worker version: pause after
start, to enable the developer to attach debuggers to the
running master and worker(s) processes. Send SIGCONT <br>
signal to release the pause. (Under gdb: (gdb) signal
SIGCONT) (Only available if optional MPI support was enabled
at compile-time.)</p>

<p style="margin-top: 1em">--mpi Run in MPI master/worker
mode, using mpirun. (Only available if optional MPI support
was enabled at compile-time.)</p>

<p style="margin-top: 1em">SEE ALSO <br>
See hmmer(1) for a master man page with a list of all the
individual man pages for programs in the HMMER package.</p>

<p style="margin-top: 1em">For complete documentation, see
the user guide that came with your HMMER distribution
(Userguide.pdf); or see the HMMER web page ().</p>

<p style="margin-top: 1em">COPYRIGHT <br>
Copyright (C) 2015 Howard Hughes Medical Institute. <br>
Freely distributed under the GNU General Public License
(GPLv3).</p>

<p style="margin-top: 1em">For additional information on
copyright and licensing, see the file called COPYRIGHT in
your HMMER source distribution, or see the HMMER web page
().</p>

<p style="margin-top: 1em">AUTHOR <br>
Eddy/Rivas Laboratory <br>
Janelia Farm Research Campus <br>
19700 Helix Drive <br>
Ashburn VA 20147 USA <br>
http://eddylab.org</p>

<p style="margin-top: 1em">HMMER 3.1b2 February 2015
phmmer(1)</p>
<hr>
</body>
</html>
