<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>PLS2FASTA(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">PLS2FASTA(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">PLS2FASTA(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
pls2fasta - convert plx.h5/bax.h5/fofn files to fasta or fastq files
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>pls2fasta</b> <i>in.bax.h5</i> <i>out.fasta</i> [<i>options</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Although fasta files are provided with every run, they are not trimmed nor split
  into subreads. This program takes additional annotation information, such as
  the subread coordinates and high quality regions, and uses them to create
  fasta sequences that are substrings of all bases called. Most of the time, you
  will want to trim low quality reads, so you should specify
  <b>-trimByRegion</b>.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><i>in.bax.h5</i></dt>
  <dd class="It-tag">Input plx.h5/bax.h5/fofn file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>out.fasta</i></dt>
  <dd class="It-tag">Output fasta/fastq file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-trimByRegion</b></dt>
  <dd class="It-tag">Trim away low quality regions.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-maskByRegion</b></dt>
  <dd class="It-tag">Mask low quality regions with 'N'.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-regionTable</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Optional HDF file with a <i>/PulseData/Regions</i>
    dataset.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minSubreadLength</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Do not write subreads less than the specified length.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-noSplitSubreads</b></dt>
  <dd class="It-tag">Do not split reads on adapter sequences.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-holeNumber</b></dt>
  <dd class="It-tag">Only print this hole number (or list of numbers).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-fastq</b></dt>
  <dd class="It-tag">Print in FASTQ format with quality.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-ccs</b></dt>
  <dd class="It-tag">Print de novo circular consensus (ccs) sequences</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-lineLength &#x2002;value</b></dt>
  <dd class="It-tag">Specify fasta/fastq line length</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minReadScore</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Minimum read score to print a read. The score is a number
      between 0 and 1000 and represents the expected accuracy percentage * 10. A
      typical value would be between 750 and 800. This does not apply to ccs
      reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-best</b></dt>
  <dd class="It-tag">If a ccs sequence exists, print this. Otherwise, print the
      longest subread. This does not support fastq.</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>blasr</b>(1) <b>loadPulses</b>(1) <b>samFilter</b>(1) <b>samtoh5</b>(1)
  <b>samtom4</b>(1) <b>sawriter</b>(1) <b>sdpMatcher</b>(1)
  <b>toAfg</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2015</td>
    <td class="foot-os">pls2fasta 3ca7fe8</td>
  </tr>
</table>
</body>
</html>
