<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:31:52 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>PLS2FASTA(1) User Commands PLS2FASTA(1)</p>

<p style="margin-top: 1em">NAME <br>
pls2fasta - convert plx.h5/bax.h5/fofn files to fasta or
fastq files</p>

<p style="margin-top: 1em">SYNOPSIS <br>
pls2fasta in.bax.h5 out.fasta [options]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Although fasta files are provided with every run, they are
not trimmed nor split into subreads. This program takes
additional annotation information, such as the subread
coordi&acirc; <br>
nates and high quality regions, and uses them to create
fasta sequences that are substrings of all bases called.
Most of the time, you will want to trim low quality reads,
so you <br>
should specify -trimByRegion.</p>

<p style="margin-top: 1em">OPTIONS <br>
in.bax.h5 <br>
Input plx.h5/bax.h5/fofn file.</p>

<p style="margin-top: 1em">out.fasta <br>
Output fasta/fastq file.</p>

<p style="margin-top: 1em">-trimByRegion <br>
Trim away low quality regions.</p>

<p style="margin-top: 1em">-maskByRegion <br>
Mask low quality regions with &rsquo;N&rsquo;.</p>

<p style="margin-top: 1em">-regionTable value <br>
Optional HDF file with a /PulseData/Regions dataset.</p>

<p style="margin-top: 1em">-minSubreadLength value <br>
Do not write subreads less than the specified length.</p>

<p style="margin-top: 1em">-noSplitSubreads <br>
Do not split reads on adapter sequences.</p>

<p style="margin-top: 1em">-holeNumber <br>
Only print this hole number (or list of numbers).</p>

<p style="margin-top: 1em">-fastq Print in FASTQ format
with quality.</p>

<p style="margin-top: 1em">-ccs Print de novo circular
consensus (ccs) sequences</p>

<p style="margin-top: 1em">-lineLength value <br>
Specify fasta/fastq line length</p>

<p style="margin-top: 1em">-minReadScore value <br>
Minimum read score to print a read. The score is a number
between 0 and 1000 and represents the expected accuracy
percentage * 10. A typical value would be between 750 <br>
and 800. This does not apply to ccs reads.</p>

<p style="margin-top: 1em">-best If a ccs sequence exists,
print this. Otherwise, print the longest subread. This does
not support fastq.</p>

<p style="margin-top: 1em">SEE ALSO <br>
blasr(1) loadPulses(1) samFilter(1) samtoh5(1) samtom4(1)
sawriter(1) sdpMatcher(1) toAfg(1)</p>

<p style="margin-top: 1em">pls2fasta 3ca7fe8 July 2015
PLS2FASTA(1)</p>
<hr>
</body>
</html>
