<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>PMRNA(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">PMRNA(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">PMRNA(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
pmrna - filter out all but the primary isoform from each gene of the input
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>pmrna</b> [ <i>options</i>] <i>&lt; annot.gff3 &gt; new.gff3</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Options:
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>|--help</dt>
  <dd class="It-tag">print this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b>|--introns</dt>
  <dd class="It-tag">flag indicating that introns are declared explicitly and do
      not need to be inferred from exon features; default is to infer
    introns</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b>|--locus</dt>
  <dd class="It-tag">report a single representative mRNA for each locus instead
      of each gene</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>|--map: FILE</dt>
  <dd class="It-tag">write each gene/mRNA mapping to the specified file</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b>|--pseudogenes</dt>
  <dd class="It-tag">disable pseudogene detection and correction</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2016</td>
    <td class="foot-os">pmrna 0.15.1</td>
  </tr>
</table>
</body>
</html>
