<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:34:59 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>RASMOL(1) General Commands Manual RASMOL(1)</p>

<p style="margin-top: 1em">NAME <br>
rasmol - Molecular Graphics Visualisation Tool v2.7.5</p>

<p style="margin-top: 1em">SYNOPSIS <br>
rasmol [-nodiplay] [[-format ] filename] [-script
scriptfile]</p>

<p style="margin-top: 1em">FORMATS <br>
-pdb Protein Data Bank <br>
-mdl MDL&rsquo;s MOL File Format <br>
-mol2 Tripos&rsquo; Sybyl MOL2 Format <br>
-xyz MSC&rsquo;s XYZ (XMol) Format <br>
-mopac MOPAC Input or Output File Format <br>
-alchemy Alchemy File Format <br>
-charmm CHARMm File Format <br>
-cif IUCr CIF or CIF File Format</p>

<p style="margin-top: 1em">NOTICES <br>
This software has been created from several sources. Much of
the code is from RasMol 2.6, as created by Roger Sayle. The
torsion angle code, new POVRAY3 code and other features <br>
are derived from the RasMol2.6x1 revisions by Arne Mueller.
The Ramachandran printer plot code was derived from fisipl
created by Frances C. Bernstein. See the Protein Data <br>
Bank program tape.</p>

<p style="margin-top: 1em">The code to display multiple
molecules and to allow bond rotation is derived in large
part from the UCB mods by Gary Grossman and Marco Molinaro,
included with permission of <br>
Eileen Lewis of the ModularCHEM Consortium.</p>

<p style="margin-top: 1em">The CIF modifications make use
of a library based in part on CBFlib by Paul J. Ellis and
Herbert J. Bernstein. Parts of CBFlib is loosely based on
the CIFPARSE software package <br>
from the NDB at Rutgers university. Please type the RasMol
commands help copying, help general, help IUCR, help CBFlib,
<br>
and help CIFPARSE for applicable notices. Please type help
copyright for copyright notices. If you use RasMol V2.6 or
an earlier version, type the RasMol command help
oldno&acirc; <br>
tice.</p>

<p style="margin-top: 1em">COPYING <br>
This version is based directly on RasMol version 2.7.4.2, on
RasMol verion 2.7.4.2, on RasMol version 2.7.4, on RasMol
version 2.7.3.1, on RasMol version 2.7.3, on RasMol version
<br>
2.7.2.1.1, Rasmol version 2.7.2, RasMol version 2.7.1.1 and
RasTop version 1.3 and indirectly on the RasMol 2.5-ucb and
2.6-ucb versions and version 2.6_CIF.2, RasMol 2.6x1 and
<br>
RasMol_2.6.4.</p>

<p style="margin-top: 1em">RasMol 2.7.5 may be distributed
under the terms of the GNU General Public License (the GPL),
see</p>


<p style="margin-top: 1em">http://www.gnu.org/licenses/gpl.txt</p>

<p style="margin-top: 1em">or the file GPL or type the
command help GPL</p>

<p style="margin-top: 1em">or RasMol 2.7.5 may be
distributed under the RASMOL license. See the file NOTICE or
type the command help RASLIC</p>

<p style="margin-top: 1em">GPL <br>
GNU GENERAL PUBLIC LICENSE <br>
Version 2, June 1991</p>

<p style="margin-top: 1em">Copyright (C) 1989, 1991 Free
Software Foundation, Inc. <br>
59 Temple Place, Suite 330, Boston, MA 02111-1307 USA <br>
Everyone is permitted to copy and distribute verbatim copies
<br>
of this license document, but changing it is not
allowed.</p>

<p style="margin-top: 1em">Preamble</p>

<p style="margin-top: 1em">The licenses for most software
are designed to take away your freedom to share and change
it. By contrast, the GNU General Public License is intended
to guarantee your <br>
freedom to share and change free software--to make sure the
software is free for all its users. This General Public
License applies to most of the Free Software Founda&acirc;
<br>
tion&rsquo;s software and to any other program whose authors
commit to using it. (Some other Free Software Foundation
software is covered by the GNU Library General Public <br>
License instead.) You can apply it to your programs,
too.</p>

<p style="margin-top: 1em">When we speak of free software,
we are referring to freedom, not price. Our General Public
Licenses are designed to make sure that you have the freedom
to distribute <br>
copies of free software (and charge for this service if you
wish), that you receive source code or can get it if you
want it, that you can change the software or use <br>
pieces of it in new free programs; and that you know you can
do these things.</p>

<p style="margin-top: 1em">To protect your rights, we need
to make restrictions that forbid anyone to deny you these
rights or to ask you to surrender the rights. These
restrictions translate to <br>
certain responsibilities for you if you distribute copies of
the software, or if you modify it.</p>

<p style="margin-top: 1em">For example, if you distribute
copies of such a program, whether gratis or for a fee, you
must give the recipients all the rights that you have. You
must make sure that <br>
they, too, receive or can get the source code. And you must
show them these terms so they know their rights.</p>

<p style="margin-top: 1em">We protect your rights with two
steps: (1) copyright the software, and (2) offer you this
license which gives you legal permission to copy, distribute
and/or modify the <br>
software.</p>

<p style="margin-top: 1em">Also, for each author&rsquo;s
protection and ours, we want to make certain that everyone
understands that there is no warranty for this free
software. If the software is mod&acirc; <br>
ified by someone else and passed on, we want its recipients
to know that what they have is not the original, so that any
problems introduced by others will not reflect on <br>
the original authors&rsquo; reputations.</p>

<p style="margin-top: 1em">Finally, any free program is
threatened constantly by software patents. We wish to avoid
the danger that redistributors of a free program will
individually obtain <br>
patent licenses, in effect making the program proprietary.
To prevent this, we have made it clear that any patent must
be licensed for everyone&rsquo;s free use or not licensed
<br>
at all.</p>

<p style="margin-top: 1em">The precise terms and conditions
for copying, distribution and modification follow.</p>

<p style="margin-top: 1em">GNU GENERAL PUBLIC LICENSE <br>
TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND
MODIFICATION</p>

<p style="margin-top: 1em">0. This License applies to any
program or other work which contains a notice placed by the
copyright holder saying it may be distributed under the
terms of this General <br>
Public License. The &quot;Program&quot;, below, refers to
any such program or work, and a &quot;work based on the
Program&quot; means either the Program or any derivative
work under copy&acirc; <br>
right law: that is to say, a work containing the Program or
a portion of it, either verbatim or with modifications
and/or translated into another language. (Hereinafter, <br>
translation is included without limitation in the term
&quot;modification&quot;.) Each licensee is addressed as
&quot;you&quot;.</p>

<p style="margin-top: 1em">Activities other than copying,
distribution and modification are not covered by this
License; they are outside its scope. The act of running the
Program is not <br>
restricted, and the output from the Program is covered only
if its contents constitute a work based on the Program
(independent of having been made by running the Pro&acirc;
<br>
gram). Whether that is true depends on what the Program
does.</p>

<p style="margin-top: 1em">1. You may copy and distribute
verbatim copies of the Program&rsquo;s source code as you
receive it, in any medium, provided that you conspicuously
and appropriately publish <br>
on each copy an appropriate copyright notice and disclaimer
of warranty; keep intact all the notices that refer to this
License and to the absence of any warranty; and <br>
give any other recipients of the Program a copy of this
License along with the Program.</p>

<p style="margin-top: 1em">You may charge a fee for the
physical act of transferring a copy, and you may at your
option offer warranty protection in exchange for a fee.</p>

<p style="margin-top: 1em">2. You may modify your copy or
copies of the Program or any portion of it, thus forming a
work based on the Program, and copy and distribute such
modifications or work <br>
under the terms of Section 1 above, provided that you also
meet all of these conditions:</p>

<p style="margin-top: 1em">a) You must cause the modified
files to carry prominent notices <br>
stating that you changed the files and the date of any
change.</p>

<p style="margin-top: 1em">b) You must cause any work that
you distribute or publish, that in <br>
whole or in part contains or is derived from the Program or
any <br>
part thereof, to be licensed as a whole at no charge to all
third <br>
parties under the terms of this License.</p>

<p style="margin-top: 1em">c) If the modified program
normally reads commands interactively <br>
when run, you must cause it, when started running for such
<br>
interactive use in the most ordinary way, to print or
display an <br>
announcement including an appropriate copyright notice and a
<br>
notice that there is no warranty (or else, saying that you
provide <br>
a warranty) and that users may redistribute the program
under <br>
these conditions, and telling the user how to view a copy of
this <br>
License. (Exception: if the Program itself is interactive
but <br>
does not normally print such an announcement, your work
based on <br>
the Program is not required to print an announcement.)</p>

<p style="margin-top: 1em">These requirements apply to the
modified work as a whole. If identifiable sections of that
work are not derived from the Program, and can be reasonably
considered inde&acirc; <br>
pendent and separate works in themselves, then this License,
and its terms, do not apply to those sections when you
distribute them as separate works. But when you dis&acirc;
<br>
tribute the same sections as part of a whole which is a work
based on the Program, the distribution of the whole must be
on the terms of this License, whose permissions <br>
for other licensees extend to the entire whole, and thus to
each and every part regardless of who wrote it.</p>

<p style="margin-top: 1em">Thus, it is not the intent of
this section to claim rights or contest your rights to work
written entirely by you; rather, the intent is to exercise
the right to control <br>
the distribution of derivative or collective works based on
the Program.</p>

<p style="margin-top: 1em">In addition, mere aggregation of
another work not based on the Program with the Program (or
with a work based on the Program) on a volume of a storage
or distribution <br>
medium does not bring the other work under the scope of this
License.</p>

<p style="margin-top: 1em">3. You may copy and distribute
the Program (or a work based on it, under Section 2) in
object code or executable form under the terms of Sections 1
and 2 above provided <br>
that you also do one of the following:</p>

<p style="margin-top: 1em">a) Accompany it with the
complete corresponding machine-readable <br>
source code, which must be distributed under the terms of
Sections <br>
1 and 2 above on a medium customarily used for software
interchange; or,</p>

<p style="margin-top: 1em">b) Accompany it with a written
offer, valid for at least three <br>
years, to give any third party, for a charge no more than
your <br>
cost of physically performing source distribution, a
complete <br>
machine-readable copy of the corresponding source code, to
be <br>
distributed under the terms of Sections 1 and 2 above on a
medium <br>
customarily used for software interchange; or,</p>

<p style="margin-top: 1em">c) Accompany it with the
information you received as to the offer <br>
to distribute corresponding source code. (This alternative
is <br>
allowed only for noncommercial distribution and only if you
<br>
received the program in object code or executable form with
such <br>
an offer, in accord with Subsection b above.)</p>

<p style="margin-top: 1em">The source code for a work means
the preferred form of the work for making modifications to
it. For an executable work, complete source code means all
the source code for <br>
all modules it contains, plus any associated interface
definition files, plus the scripts used to control
compilation and installation of the executable. However, as
a <br>
special exception, the source code distributed need not
include anything that is normally distributed (in either
source or binary form) with the major components (com&acirc;
<br>
piler, kernel, and so on) of the operating system on which
the executable runs, unless that component itself
accompanies the executable.</p>

<p style="margin-top: 1em">If distribution of executable or
object code is made by offering access to copy from a
designated place, then offering equivalent access to copy
the source code from the <br>
same place counts as distribution of the source code, even
though third parties are not compelled to copy the source
along with the object code.</p>

<p style="margin-top: 1em">4. You may not copy, modify,
sublicense, or distribute the Program except as expressly
provided under this License. Any attempt otherwise to copy,
modify, sublicense or <br>
distribute the Program is void, and will automatically
terminate your rights under this License. However, parties
who have received copies, or rights, from you under this
<br>
License will not have their licenses terminated so long as
such parties remain in full compliance.</p>

<p style="margin-top: 1em">5. You are not required to
accept this License, since you have not signed it. However,
nothing else grants you permission to modify or distribute
the Program or its de&acirc; <br>
rivative works. These actions are prohibited by law if you
do not accept this License. Therefore, by modifying or
distributing the Program (or any work based on the
Pro&acirc; <br>
gram), you indicate your acceptance of this License to do
so, and all its terms and conditions for copying,
distributing or modifying the Program or works based on
it.</p>

<p style="margin-top: 1em">6. Each time you redistribute
the Program (or any work based on the Program), the
recipient automatically receives a license from the original
licensor to copy, distrib&acirc; <br>
ute or modify the Program subject to these terms and
conditions. You may not impose any further restrictions on
the recipients&rsquo; exercise of the rights granted herein.
<br>
You are not responsible for enforcing compliance by third
parties to this License.</p>

<p style="margin-top: 1em">7. If, as a consequence of a
court judgment or allegation of patent infringement or for
any other reason (not limited to patent issues), conditions
are imposed on you <br>
(whether by court order, agreement or otherwise) that
contradict the conditions of this License, they do not
excuse you from the conditions of this License. If you
cannot <br>
distribute so as to satisfy simultaneously your obligations
under this License and any other pertinent obligations, then
as a consequence you may not distribute the Pro&acirc; <br>
gram at all. For example, if a patent license would not
permit royalty-free redistribution of the Program by all
those who receive copies directly or indirectly through <br>
you, then the only way you could satisfy both it and this
License would be to refrain entirely from distribution of
the Program.</p>

<p style="margin-top: 1em">If any portion of this section
is held invalid or unenforceable under any particular
circumstance, the balance of the section is intended to
apply and the section as a <br>
whole is intended to apply in other circumstances.</p>

<p style="margin-top: 1em">It is not the purpose of this
section to induce you to infringe any patents or other
property right claims or to contest validity of any such
claims; this section has the <br>
sole purpose of protecting the integrity of the free
software distribution system, which is implemented by public
license practices. Many people have made generous con&acirc;
<br>
tributions to the wide range of software distributed through
that system in reliance on consistent application of that
system; it is up to the author/donor to decide if he <br>
or she is willing to distribute software through any other
system and a licensee cannot impose that choice.</p>

<p style="margin-top: 1em">This section is intended to make
thoroughly clear what is believed to be a consequence of the
rest of this License.</p>

<p style="margin-top: 1em">8. If the distribution and/or
use of the Program is restricted in certain countries either
by patents or by copyrighted interfaces, the original
copyright holder who <br>
places the Program under this License may add an explicit
geographical distribution limitation excluding those
countries, so that distribution is permitted only in or <br>
among countries not thus excluded. In such case, this
License incorporates the limitation as if written in the
body of this License.</p>

<p style="margin-top: 1em">9. The Free Software Foundation
may publish revised and/or new versions of the General
Public License from time to time. Such new versions will be
similar in spirit to <br>
the present version, but may differ in detail to address new
problems or concerns.</p>

<p style="margin-top: 1em">Each version is given a
distinguishing version number. If the Program specifies a
version number of this License which applies to it and
&quot;any later version&quot;, you have the <br>
option of following the terms and conditions either of that
version or of any later version published by the Free
Software Foundation. If the Program does not specify a <br>
version number of this License, you may choose any version
ever published by the Free Software Foundation.</p>

<p style="margin-top: 1em">10. If you wish to incorporate
parts of the Program into other free programs whose
distribution conditions are different, write to the author
to ask for permission. For <br>
software which is copyrighted by the Free Software
Foundation, write to the Free Software Foundation; we
sometimes make exceptions for this. Our decision will be
guided <br>
by the two goals of preserving the free status of all
derivatives of our free software and of promoting the
sharing and reuse of software generally.</p>

<p style="margin-top: 1em">NO WARRANTY</p>

<p style="margin-top: 1em">11. BECAUSE THE PROGRAM IS
LICENSED FREE OF CHARGE, THERE IS NO WARRANTY FOR THE
PROGRAM, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT
WHEN OTHERWISE STATED IN <br>
WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES PROVIDE
THE PROGRAM &quot;AS IS&quot; WITHOUT WARRANTY OF ANY KIND,
EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO,
<br>
THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
PARTICULAR PURPOSE. THE ENTIRE RISK AS TO THE QUALITY AND
PERFORMANCE OF THE PROGRAM IS WITH YOU. SHOULD THE <br>
PROGRAM PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL
NECESSARY SERVICING, REPAIR OR CORRECTION.</p>

<p style="margin-top: 1em">12. IN NO EVENT UNLESS REQUIRED
BY APPLICABLE LAW OR AGREED TO IN WRITING WILL ANY COPYRIGHT
HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
REDISTRIBUTE THE PROGRAM AS <br>
PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES, INCLUDING ANY
GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES
ARISING OUT OF THE USE OR INABILITY TO USE THE PROGRAM <br>
(INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD
PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY <br>
OTHER PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN
ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.</p>

<p style="margin-top: 1em">END OF TERMS AND CONDITIONS</p>

<p style="margin-top: 1em">How to Apply These Terms to Your
New Programs</p>

<p style="margin-top: 1em">If you develop a new program,
and you want it to be of the greatest possible use to the
public, the best way to achieve this is to make it free
software which everyone <br>
can redistribute and change under these terms.</p>

<p style="margin-top: 1em">To do so, attach the following
notices to the program. It is safest to attach them to the
start of each source file to most effectively convey the
exclusion of war&acirc; <br>
ranty; and each file should have at least the
&quot;copyright&quot; line and a pointer to where the full
notice is found.</p>

<p style="margin-top: 1em">&lt;one line to give the
program&rsquo;s name and a brief idea of what it does.&gt;
<br>
Copyright (C) &lt;year&gt; &lt;name of author&gt;</p>

<p style="margin-top: 1em">This program is free software;
you can redistribute it and/or modify <br>
it under the terms of the GNU General Public License as
published by <br>
the Free Software Foundation; either version 2 of the
License, or <br>
(at your option) any later version.</p>

<p style="margin-top: 1em">This program is distributed in
the hope that it will be useful, <br>
but WITHOUT ANY WARRANTY; without even the implied warranty
of <br>
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
<br>
GNU General Public License for more details.</p>

<p style="margin-top: 1em">You should have received a copy
of the GNU General Public License <br>
along with this program; if not, write to the Free Software
<br>
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
02111-1307 USA</p>

<p style="margin-top: 1em">Also add information on how to
contact you by electronic and paper mail.</p>

<p style="margin-top: 1em">If the program is interactive,
make it output a short notice like this when it starts in an
interactive mode:</p>

<p style="margin-top: 1em">Gnomovision version 69,
Copyright (C) year name of author <br>
Gnomovision comes with ABSOLUTELY NO WARRANTY; for details
type &lsquo;show w&rsquo;. <br>
This is free software, and you are welcome to redistribute
it <br>
under certain conditions; type &lsquo;show c&rsquo; for
details.</p>

<p style="margin-top: 1em">The hypothetical commands
&lsquo;show w&rsquo; and &lsquo;show c&rsquo; should show
the appropriate parts of the General Public License. Of
course, the commands you use may be called something <br>
other than &lsquo;show w&rsquo; and &lsquo;show c&rsquo;;
they could even be mouse-clicks or menu items--whatever
suits your program.</p>

<p style="margin-top: 1em">You should also get your
employer (if you work as a programmer) or your school, if
any, to sign a &quot;copyright disclaimer&quot; for the
program, if necessary. Here is a sample; <br>
alter the names:</p>

<p style="margin-top: 1em">Yoyodyne, Inc., hereby disclaims
all copyright interest in the program <br>
&lsquo;Gnomovision&rsquo; (which makes passes at compilers)
written by James Hacker.</p>

<p style="margin-top: 1em">&lt;signature of Ty Coon&gt;, 1
April 1989 <br>
Ty Coon, President of Vice</p>

<p style="margin-top: 1em">This General Public License does
not permit incorporating your program into proprietary
programs. If your program is a subroutine library, you may
consider it more useful <br>
to permit linking proprietary applications with the library.
If this is what you want to do, use the GNU Library General
Public License instead of this License.</p>

<p style="margin-top: 1em">RASLIC If you do not use the
GPL, the following license terms apply:</p>

<p style="margin-top: 1em">RasMol License</p>

<p style="margin-top: 1em">Even though the authors of the
various documents and software found here have made a good
faith effort to ensure that the documents are correct and
that the software per&acirc; <br>
forms according to its documentation, and we would greatly
appreciate hearing of any problems you may encounter, the
programs and documents any files created by the pro&acirc;
<br>
grams are provided **AS IS** without any warranty as to
correctness, merchantability or fitness for any particular
or general use.</p>

<p style="margin-top: 1em">THE RESPONSIBILITY FOR ANY
ADVERSE CONSEQUENCES FROM THE USE OF PROGRAMS OR DOCUMENTS
OR ANY FILE OR FILES CREATED BY USE OF THE PROGRAMS OR
DOCUMENTS LIES SOLELY WITH THE <br>
USERS OF THE PROGRAMS OR DOCUMENTS OR FILE OR FILES AND NOT
WITH AUTHORS OF THE PROGRAMS OR DOCUMENTS.</p>

<p style="margin-top: 1em">Subject to your acceptance of
the conditions stated above, and your respect for the terms
and conditions stated in the notices below, if you are not
going to make any mod&acirc; <br>
ifications or create derived works, you are given permission
to freely copy and distribute this package, provided you do
the following:</p>

<p style="margin-top: 1em">1. Either include the complete
documentation, especially the file NOTICE, with what you
distribute or provide a clear indication where people can
get a copy of the doc&acirc; <br>
umentation; and</p>

<p style="margin-top: 1em">2. Please give credit where
credit is due citing the version and original authors
properly; and</p>

<p style="margin-top: 1em">3. Please do not give anyone the
impression that the original authors are providing a
warranty of any kind.</p>

<p style="margin-top: 1em">If you would like to use major
pieces of RasMol in some other program, make modifications
to RasMol, or in some other way make what a lawyer would
call a &quot;derived work&quot;, <br>
you are not only permitted to do so, you are encouraged to
do so. In addition to the things we discussed above, please
do the following:</p>

<p style="margin-top: 1em">4. Please explain in your
documentation how what you did differs from this version of
RasMol; and</p>

<p style="margin-top: 1em">5. Please make your modified
source code available.</p>

<p style="margin-top: 1em">This version of RasMol is _not_
in the public domain, but it is given freely to the
community in the hopes of advancing science. If you make
changes, please make them in <br>
a responsible manner, and please offer us the opportunity to
include those changes in future versions of RasMol.</p>

<p style="margin-top: 1em">General Notice <br>
The following notice applies to this work as a whole and to
the works included within it:</p>

<p style="margin-top: 1em">* Creative endeavors depend on
the lively exchange of ideas. There are laws and customs
which establish rights and responsibilities for authors and
the users of what <br>
authors create. This notice is not intended to prevent you
from using the software and documents in this package, but
to ensure that there are no misunderstandings about <br>
terms and conditions of such use.</p>

<p style="margin-top: 1em">* Please read the following
notice carefully. If you do not understand any portion of
this notice, please seek appropriate professional legal
advice before making use of <br>
the software and documents included in this software
package. In addition to whatever other steps you may be
obliged to take to respect the intellectual property rights
of <br>
the various parties involved, if you do make use of the
software and documents in this package, please give credit
where credit is due by citing this package, its authors <br>
and the URL or other source from which you obtained it, or
equivalent primary references in the literature with the
same authors.</p>

<p style="margin-top: 1em">* Some of the software and
documents included within this software package are the
intellectual property of various parties, and placement in
this package does not in any <br>
way imply that any such rights have in any way been waived
or diminished.</p>

<p style="margin-top: 1em">* With respect to any software
or documents for which a copyright exists, ALL RIGHTS ARE
RESERVED TO THE OWNERS OF SUCH COPYRIGHT.</p>

<p style="margin-top: 1em">* Even though the authors of the
various documents and software found here have made a good
faith effort to ensure that the documents are correct and
that the software <br>
performs according to its documentation, and we would
greatly appreciate hearing of any problems you may
encounter, the programs and documents and any files created
by the <br>
programs are provided **AS IS** without any warranty as to
correctness, merchantability or fitness for any particular
or general use.</p>

<p style="margin-top: 1em">* THE RESPONSIBILITY FOR ANY
ADVERSE CONSEQUENCES FROM THE USE OF PROGRAMS OR DOCUMENTS
OR ANY FILE OR FILES CREATED BY USE OF THE PROGRAMS OR
DOCUMENTS LIES SOLELY WITH <br>
THE USERS OF THE PROGRAMS OR DOCUMENTS OR FILE OR FILES AND
NOT WITH AUTHORS OF THE PROGRAMS OR DOCUMENTS.</p>

<p style="margin-top: 1em">See the files GPL and RASLIC for
two alternate ways to license this package.</p>

<p style="margin-top: 1em">RasMol V2.6 Notice <br>
The following notice applies to RasMol V 2.6 and older
RasMol versions.</p>

<p style="margin-top: 1em">Information in this document is
subject to change without notice and does not represent a
commitment on the part of the supplier. This package is
sold/distributed subject <br>
to the condition that it shall not, by way of trade or
otherwise, be lent, re-sold, hired out or otherwise
circulated without the supplier&rsquo;s prior consent, in
any form of <br>
packaging or cover other than that in which it was produced.
No part of this manual or accompanying software may be
reproduced, stored in a retrieval system on optical or <br>
magnetic disk, tape or any other medium, or transmitted in
any form or by any means, electronic, mechanical,
photocopying, recording or otherwise for any purpose other
<br>
than the purchaser&rsquo;s personal use.</p>

<p style="margin-top: 1em">This product is not to be used
in the planning, construction, maintenance, operation or use
of any nuclear facility nor the flight, navigation or
communication of aircraft <br>
or ground support equipment. The author shall not be liable,
in whole or in part, for any claims or damages arising from
such use, including death, bankruptcy or outbreak <br>
of war.</p>

<p style="margin-top: 1em">IUCR Policy <br>
The IUCr Policy for the Protection and the Promotion of the
STAR File and CIF Standards for Exchanging and Archiving
Electronic Data.</p>

<p style="margin-top: 1em">Overview</p>

<p style="margin-top: 1em">The Crystallographic Information
File (CIF)[1] is a standard for information interchange
promulgated by the International Union of Crystallography
(IUCr). CIF (Hall, Allen <br>
&amp; Brown, 1991) is the recommended method for submitting
publications to Acta Crystallographica Section C and reports
of crystal structure determinations to other sections <br>
of Acta Crystallographica and many other journals. The
syntax of a CIF is a subset of the more general STAR File[2]
format. The CIF and STAR File approaches are used <br>
increasingly in the structural sciences for data exchange
and archiving, and are having a significant influence on
these activities in other fields.</p>

<p style="margin-top: 1em">Statement of intent</p>

<p style="margin-top: 1em">The IUCr&rsquo;s interest in the
STAR File is as a general data interchange standard for
science, and its interest in the CIF, a conformant
derivative of the STAR File, is as a <br>
concise data exchange and archival standard for
crystallography and structural science.</p>

<p style="margin-top: 1em">Protection of the standards</p>

<p style="margin-top: 1em">To protect the STAR File and the
CIF as standards for interchanging and archiving electronic
data, the IUCr, on behalf of the scientific community,</p>

<p style="margin-top: 1em">* holds the copyrights on the
standards themselves,</p>

<p style="margin-top: 1em">* owns the associated trademarks
and service marks, and</p>

<p style="margin-top: 1em">* holds a patent on the STAR
File.</p>

<p style="margin-top: 1em">These intellectual property
rights relate solely to the interchange formats, not to the
data contained therein, nor to the software used in the
generation, access or <br>
manipulation of the data.</p>

<p style="margin-top: 1em">Promotion of the standards</p>

<p style="margin-top: 1em">The sole requirement that the
IUCr, in its protective role, imposes on software purporting
to process STAR File or CIF data is that the following
conditions be met prior <br>
to sale or distribution.</p>

<p style="margin-top: 1em">* Software claiming to read
files written to either the STAR File or the CIF standard
must be able to extract the pertinent data from a file
conformant to the STAR File <br>
syntax, or the CIF syntax, respectively.</p>

<p style="margin-top: 1em">* Software claiming to write
files in either the STAR File, or the CIF, standard must
produce files that are conformant to the STAR File syntax,
or the CIF syntax, <br>
respectively.</p>

<p style="margin-top: 1em">* Software claiming to read
definitions from a specific data dictionary approved by the
IUCr must be able to extract any pertinent definition which
is conformant to the <br>
dictionary definition language (DDL)[3] associated with that
dictionary.</p>

<p style="margin-top: 1em">The IUCr, through its Committee
on CIF Standards, will assist any developer to verify that
software meets these conformance conditions.</p>

<p style="margin-top: 1em">Glossary of terms</p>

<p style="margin-top: 1em">[1] CIF:</p>

<p style="margin-top: 1em">is a data file conformant to the
file syntax defined at
http://www.iucr.org/iucr-top/cif/spec/index.html</p>

<p style="margin-top: 1em">[2] STAR File:</p>

<p style="margin-top: 1em">is a data file conformant to the
file syntax defined at
http://www.iucr.org/iucr-top/cif/spec/star/index.html</p>

<p style="margin-top: 1em">[3] DDL:</p>

<p style="margin-top: 1em">is a language used in a data
dictionary to define data items in terms of
&quot;attributes&quot;. Dictionaries currently approved by
the IUCr, and the DDL versions used to construct <br>
these dictionaries, are listed at
http://www.iucr.org/iucr-top/cif/spec/ddl/index.html</p>

<p style="margin-top: 1em">Last modified: 30 September
2000</p>

<p style="margin-top: 1em">IUCr Policy Copyright (C) 2000
International Union of Crystallography</p>

<p style="margin-top: 1em">CBFLIB The following Disclaimer
Notice applies to CBFlib V0.1, from which this code in part
is derived.</p>

<p style="margin-top: 1em">* The items furnished herewith
were developed under the sponsorship of the U.S. Government.
Neither the U.S., nor the U.S. D.O.E., nor the Leland
Stanford Junior Univer&acirc; <br>
sity, nor their employees, makes any warranty, express or
implied, or assumes any liability or responsibility for
accuracy, completeness or usefulness of any information,
<br>
apparatus, product or process disclosed, or represents that
its use will not infringe privately-owned rights. Mention of
any product, its manufacturer, or suppliers shall <br>
not, nor is it intended to, imply approval, disapproval, or
fitness for any particular use. The U.S. and the University
at all times retain the right to use and dissemi&acirc; <br>
nate the furnished items for any purpose whatsoever.</p>

<p style="margin-top: 1em">Notice 91 02 01</p>

<p style="margin-top: 1em">CIFPARSE <br>
Portions of this software are loosely based on the CIFPARSE
software package from the NDB at Rutgers University. See</p>


<p style="margin-top: 1em">http://ndbserver.rutgers.edu/NDB/mmcif/software</p>

<p style="margin-top: 1em">CIFPARSE is part of the NDBQUERY
application, a program component of the Nucleic Acid
Database Project [ H. M. Berman, W. K. Olson, D. L.
Beveridge, J. K. Westbrook, A. <br>
Gelbin, T. Demeny, S. H. Shieh, A. R. Srinivasan, and B.
Schneider. (1992). The Nucleic Acid Database: A
Comprehensive Relational Database of Three-Dimensional
Structures <br>
of Nucleic Acids. Biophys J., 63, 751-759.], whose
cooperation is gratefully acknowledged, especially in the
form of design concepts created by J. Westbrook.</p>

<p style="margin-top: 1em">Please be aware of the following
notice in the CIFPARSE API:</p>

<p style="margin-top: 1em">This software is provided
WITHOUT WARRANTY OF MERCHANTABILITY OR FITNESS FOR A
PARTICULAR PURPOSE OR ANY OTHER WARRANTY, EXPRESS OR
IMPLIED. RUTGERS MAKE NO REPRESENTATION <br>
OR WARRANTY THAT THE SOFTWARE WILL NOT INFRINGE ANY PATENT,
COPYRIGHT OR OTHER PROPRIETARY RIGHT.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
RasMol is a molecular graphics program intended for the
visualisation of proteins, nucleic acids and small
molecules. The program is aimed at display, teaching and
generation of <br>
publication quality images. RasMol runs on wide range of
architectures and operating systems including Microsoft
Windows, Apple Macintosh, UNIX and VMS systems. UNIX and VMS
<br>
versions require an 8, 24 or 32 bit colour X Windows display
(X11R4 or later). The X Windows version of RasMol provides
optional support for a hardware dials box and accelerated
<br>
shared memory communication (via the XInput and MIT-SHM
extensions) if available on the current X Server.</p>

<p style="margin-top: 1em">The program reads in a molecule
coordinate file and interactively displays the molecule on
the screen in a variety of colour schemes and molecule
representations. Currently <br>
available representations include depth-cued wireframes,
&rsquo;Dreiding&rsquo; sticks, spacefilling (CPK) spheres,
ball and stick, solid and strand biomolecular ribbons, atom
labels and dot <br>
surfaces.</p>

<p style="margin-top: 1em">Up to 5 molecules may be loaded
and displayed at once. Any one or all of <br>
the molecules may be rotated and translated.</p>

<p style="margin-top: 1em">The RasMol help facility can be
accessed by typing &quot;help &lt;topic&gt;&quot; or
&quot;help &lt;topic&gt; &lt;subtopic&gt;&quot; from the
command line. A complete list of RasMol commands may be
displayed by typ&acirc; <br>
ing &quot;help commands&quot;. A single question mark may
also be used to abbreviate the keyword &quot;help&quot;.
Please type &quot;help notices&quot; for important
notices.</p>

<p style="margin-top: 1em">COMMANDS <br>
RasMol allows the execution of interactive commands typed at
the RasMol&gt; prompt in the terminal window. Each command
must be given on a separate line. Keywords are case
insensi&acirc; <br>
tive and may be entered in either upper or lower case
letters. All whitespace characters are ignored except to
separate keywords and their arguments.</p>

<p style="margin-top: 1em">All commands may be prefixed by
a parenthesized atom expression to temporarily select
certain atoms just for the execution of that one command.
After execution of the command, <br>
the previous selection is restored except for the commands
select , restrict and script.</p>

<p style="margin-top: 1em">The commands/keywords currently
recognised by RasMol are given below.</p>

<p style="margin-top: 1em">Backbone <br>
The RasMol backbone command permits the representation of a
polypeptide backbone as a series of bonds connecting the
adjacent alpha carbons of each amino acid in a chain. <br>
The display of these backbone &rsquo;bonds&rsquo; is turned
on and off by the command parameter in the same way as with
the wireframe command. The command backbone off turns off
the <br>
selected &rsquo;bonds&rsquo;, and backbone on or with a
number turns them on. The number can be used to specify the
cylinder radius of the representation in either Angstrom or
RasMol <br>
units. A parameter value of 500 (2.0 Angstroms) or above
results in a &quot;Parameter value too large&quot; error.
Backbone objects may be coloured using the RasMol colour
backbone <br>
command.</p>

<p style="margin-top: 1em">The reserved word backbone is
also used as a predefined set (&quot;help sets&quot;) and as
a parameter to the set hbond and set ssbond commands. The
RasMol command trace renders a <br>
smoothed backbone, in contrast to backbone which connects
alpha carbons with straight lines.</p>

<p style="margin-top: 1em">The backbone may be displayed
with dashed lines by use of the backbone dash command.</p>

<p style="margin-top: 1em">Background <br>
The RasMol background command is used to set the colour of
the &quot;canvas&quot; background. The colour may be given
as either a colour name or a comma separated triple of Red,
<br>
Green and Blue (RGB) components enclosed in square brackets.
Typing the command help colours will give a list of the
predefined colour names recognised by RasMol. When <br>
running under X Windows, RasMol also recognises colours in
the X server&rsquo;s colour name database.</p>

<p style="margin-top: 1em">The background command is
synonymous with the RasMol set background command.</p>

<p style="margin-top: 1em">Bond The RasMol command bond
&lt;number&gt; &lt;number&gt; + adds the designated bond to
the drawing, increasing the bond order if the bond already
exists. The command bond &lt;number&gt; &lt;num&acirc; <br>
ber&gt; pick selects the two atoms specified by the atom
serial numbers as the two ends of a bond around which the
rotate bond &lt;angle&gt; command will be applied. If no
bond <br>
exists, it is created.</p>

<p style="margin-top: 1em">Rotation around a previously
picked bond may be specified by the rotate bond
&lt;angle&gt; command, or may also be controlled with the
mouse, using the bond rotate on/off or the <br>
equivalent rotate bond on/off commands.</p>

<p style="margin-top: 1em">Bulgarian <br>
The RasMol Bulgarian command sets the menus and messages to
the Bulgarian versions.</p>

<p style="margin-top: 1em">This command may not work
correctly unless appropriate fonts have been installed. The
commands Bulgarian, Chinese, English, French, Italian,
Russian and Spanish may be <br>
used to select Bulgarian, Chinese, English, French, Italian,
Japanese, Russian and Spanish menus and messages if the
appropriate fonts have been installed.</p>

<p style="margin-top: 1em">Cartoon <br>
The RasMol cartoon command does a display of a molecule
ribbons as Richardson (MolScript) style protein cartoons,
implemented as thick (deep) ribbons. The easiest way to <br>
obtain a cartoon representation of a protein is to use the
Cartoons option on the Display menu. The cartoon command
represents the currently selected residues as a deep <br>
ribbon with width specified by the command&rsquo;s argument.
Using the command without a parameter results in the
ribbon&rsquo;s width being taken from the protein&rsquo;s
secondary struc&acirc; <br>
ture, as described in the ribbons command. By default, the
C-termini of beta-sheets are displayed as arrow heads. This
may be enabled and disabled using the set cartoons <br>
command. The depth of the cartoon may be adjusted using the
set cartoons &lt;number&gt; command. The set cartoons
command without any parameters returns these two options to
<br>
their default values.</p>

<p style="margin-top: 1em">Centre The RasMol centre command
defines the point about which the rotate command and the
scroll bars rotate the current molecule. Without a parameter
the centre command resets <br>
the centre of rotation to be the centre of gravity of the
molecule. If an atom expression is specified, RasMol rotates
the molecule about the centre of gravity of the set <br>
of atoms specified by the expression. Hence, if a single
atom is specified by the expression, that atom will remain
&rsquo;stationary&rsquo; during rotations.</p>

<p style="margin-top: 1em">Type help expression for more
information on RasMol atom expressions.</p>

<p style="margin-top: 1em">Alternatively the centring may
be given as a comma separated triple of [CenX, CenY, CenZ]
offsets in RasMol units (1/250 of an Angstrom) from the
centre of gravity. The <br>
triple must be enclosed in square brackets.</p>

<p style="margin-top: 1em">The optional forms centre ...
translate and centre ... center may be used to specify use
of a translated centre of rotation (not necessarily in the
centre of the canvas) <br>
or a centre of rotation which is placed at the centre of the
canvas. Starting with RasMol 2.7.2, the default is to center
the new axis on the canvas.</p>

<p style="margin-top: 1em">Chinese <br>
The RasMol Chinese command sets the menus and messages to
the Chinese versions.</p>

<p style="margin-top: 1em">This command may not work
correctly unless appropriate fonts have been installed. The
commands Bulgarian, Chinese, English, French, Italian,
Russian and Spanish may be <br>
used to select Bulgarian, Chinese, English, French, Italian,
Japanese, Russian and Spanish menus and messages if the
appropriate fonts have been installed.</p>

<p style="margin-top: 1em">Clipboard <br>
The RasMol clipboard command places a copy of the currently
displayed image on the local graphics
&rsquo;clipboard&rsquo;. Note: this command is not yet
supported on UNIX or VMS <br>
machines. It is intended to make transfering images between
applications easier under Microsoft Windows or on an Apple
Macintosh.</p>

<p style="margin-top: 1em">When using RasMol on a UNIX or
VMS system this functionality may be achieved by generating
a raster image in a format that can be read by the receiving
program using the <br>
RasMol write command.</p>

<p style="margin-top: 1em">Colour Colour the atoms (or
other objects) of the selected region. The colour may be
given as either a colour name or a comma separated triple of
Red, Green and Blue (RGB) compo&acirc; <br>
nents enclosed in square brackets. Typing the command help
colours will give a list of all the predefined colour names
recognised by RasMol.</p>

<p style="margin-top: 1em">Allowed objects are atoms,
bonds, backbone, ribbons, labels, dots, hbonds, map, and
ssbonds. If no object is specified, the default keyword atom
is assumed. Some colour <br>
schemes are defined for certain object types. The colour
scheme none can be applied to all objects except atoms and
dots, stating that the selected objects have no colour <br>
of their own, but use the colour of their associated atoms
(i.e. the atoms they connect). Atom objects can also be
coloured by alt, amino, chain, charge, cpk, group, <br>
model, shapely, structure, temperature or user. Hydrogen
bonds can also be coloured by type and dot surfaces can also
be coloured by electrostatic potential. For more <br>
information type help colour &lt;colour&gt;. Map objects may
be coloured by specific color of by nearest atom.</p>

<p style="margin-top: 1em">ColourMode <br>
ColourMode allows the user to switch between using the new
colour method. At present, the new coloring technique is the
same as the old one, but to preserve compatibility <br>
for older scripts it may be wise to add a &quot;colormode
on&quot; near the top of your script somewhere, if the
script was designed for version 2.7.3 of RasMol or earlier.
The new <br>
color method, when completed, aims to fix a few bugs in the
coloring routines.</p>

<p style="margin-top: 1em">Connect <br>
The RasMol connect command is used to force RasMol to
(re)calculate the connectivity of the current molecule. If
the original input file contained connectivity
informa&acirc; <br>
tion, this is discarded. The command connect false uses a
fast heuristic algorithm that is suitable for determining
bonding in large bio-molecules such as proteins and <br>
nucleic acids. The command connect true uses a slower more
accurate algorithm based upon covalent radii that is more
suitable to small molecules containing inorganic ele&acirc;
<br>
ments or strained rings. If no parameters are given, RasMol
determines which algorithm to use based on the number of
atoms in the input file. Greater than 255 atoms causes <br>
RasMol to use the faster implementation. This is the method
used to determine bonding, if necessary, when a molecule is
first read in using the load command.</p>

<p style="margin-top: 1em">Defer The RasMol defer command
adds the command given to the macro with given name, if no
name is given, the command is added to the macro with a
blank name. The command zap is <br>
a special case. In that case the macro is erased. If no name
is given the command must begin with a selection, e.g. defer
(selection).spacefill</p>

<p style="margin-top: 1em">The deferred commands
accumulated under the given name can be executed using the
execute command</p>

<p style="margin-top: 1em">Define The RasMol define command
allows the user to associate an arbitrary set of atoms with
a unique identifier. This allows the definition of
user-defined sets. These sets are <br>
declared statically, i.e. once defined the contents of the
set do not change, even if the expression defining them
depends on the current transformation and representation
<br>
of the molecule.</p>

<p style="margin-top: 1em">Depth The RasMol depth command
enables, disables or positions the back-clipping plane of
the molecule. The program only draws those portions of the
molecule that are closer to <br>
the viewer than the clipping plane. Integer values range
from zero at the very back of the molecule to 100 which is
completely in front of the molecule. Intermediate val&acirc;
<br>
ues determine the percentage of the molecule to be
drawn.</p>

<p style="margin-top: 1em">This command interacts with the
slab &lt;value&gt; command, which clips to the front of a
given z-clipping plane.</p>

<p style="margin-top: 1em">Dots The RasMol dots command is
used to generate a van der Waals&rsquo; dot surface around
the currently selected atoms. Dot surfaces display regularly
spaced points on a sphere of <br>
van der Waals&rsquo; radius about each selected atom. Dots
that would are &rsquo;buried&rsquo; within the van der
Waals&rsquo; radius of any other atom (selected or not) are
not displayed. The <br>
command dots on deletes any existing dot surface and
generates a dots surface around the currently selected atom
set with a default dot density of 100. The command dots <br>
off deletes any existing dot surface. The dot density may be
specified by providing a numeric parameter between 1 and
1000. This value approximately corresponds to the <br>
number of dots on the surface of a medium sized atom.</p>

<p style="margin-top: 1em">By default, the colour of each
point on a dot surface is the colour of its closest atom at
the time the surface is generated. The colour of the whole
dot surface may be <br>
changed using the colour dots command.</p>

<p style="margin-top: 1em">Echo The RasMol echo command is
used to display a message in the RasMol command/terminal
window. The string parameter may optionally be delimited in
double quote characters. If <br>
no parameter is specified, the echo command displays a blank
line. This command is particularly useful for displaying
text from within a RasMol script file.</p>

<p style="margin-top: 1em">English <br>
The RasMol English command sets the menus and messages to
the English versions.</p>

<p style="margin-top: 1em">This command may not work
correctly unless appropriate fonts have been installed. The
commands Bulgarian, Chinese, English, French, Italian,
Russian and Spanish may be <br>
used to select Bulgarian, Chinese, English, French, Italian,
Japanese, Russian and Spanish menus and messages if the
appropriate fonts have been installed.</p>

<p style="margin-top: 1em">Execute <br>
The RasMol execute command:</p>

<p style="margin-top: 1em">1. saves the old poise of the
molecule (translation, rotation and zoom)</p>

<p style="margin-top: 1em">2. executes the specified macro
suppressing both screen updates and recording</p>

<p style="margin-top: 1em">3. animates motion of the newly
rendered molecule linearly from the old poise to the new
poise</p>

<p style="margin-top: 1em">The macro must have been
previously defined by calls to the defer command.</p>

<p style="margin-top: 1em">The animation of the motion
depends on the prior settings of the record command.</p>

<p style="margin-top: 1em">French The RasMol French command
sets the menus and messages to the French versions.</p>

<p style="margin-top: 1em">This command may not work
correctly unless appropriate fonts have been installed. The
commands Bulgarian, Chinese, English, French, Italian,
Russian and Spanish may be <br>
used to select Bulgarian, Chinese, English, French, Italian,
Japanese, Russian and Spanish menus and messages if the
appropriate fonts have been installed.</p>

<p style="margin-top: 1em">HBonds The RasMol hbond command
is used to represent the hydrogen bonding of the protein
molecule&rsquo;s backbone. This information is useful in
assessing the protein&rsquo;s secondary <br>
structure. Hydrogen bonds are represented as either dotted
lines or cylinders between the donor and acceptor residues.
The first time the hbond command is used, the pro&acirc;
<br>
gram searches the structure of the molecule to find hydrogen
bonded residues and reports the number of bonds to the user.
The command hbonds on displays the selected <br>
&rsquo;bonds&rsquo; as dotted lines, and the hbonds off
turns off their display. The colour of hbond objects may be
changed by the colour hbond command. Initially, each
hydrogen bond <br>
has the colours of its connected atoms.</p>

<p style="margin-top: 1em">By default the dotted lines are
drawn between the accepting oxygen and the donating
nitrogen. By using the set hbonds command the alpha carbon
positions of the appropriate <br>
residues may be used instead. This is especially useful when
examining proteins in backbone representation.</p>

<p style="margin-top: 1em">Help The RasMol help command
provides on-line help on the given topic.</p>

<p style="margin-top: 1em">Italian <br>
The RasMol Italian command sets the menus and messages to
the Italian versions.</p>

<p style="margin-top: 1em">This command may not work
correctly unless appropriate fonts have been installed. The
commands Bulgarian, Chinese, English, French, Italian,
Russian and Spanish may be <br>
used to select Bulgarian, Chinese, English, French, Italian,
Japanese, Russian and Spanish menus and messages if the
appropriate fonts have been installed.</p>

<p style="margin-top: 1em">Japanese <br>
The RasMol Japanese command sets the menus and messages to
the Japanese versions.</p>

<p style="margin-top: 1em">This command may not work
correctly unless appropriate fonts have been installed. The
commands Bulgarian, Chinese, English, French, Italian,
Russian and Spanish may be <br>
used to select Bulgarian, Chinese, English, French, Italian,
Japanese, Russian and Spanish menus and messages if the
appropriate fonts have been installed.</p>

<p style="margin-top: 1em">Label The RasMol label command
allows an arbitrary formatted text string to be associated
with each currently selected atom. This string may contain
embedded &rsquo;expansion speci&acirc; <br>
fiers&rsquo; which display properties of the atom being
labelled. An expansion specifier consists of a
&rsquo;%&rsquo; character followed by a single alphabetic
character specifying the <br>
property to be displayed. An actual &rsquo;%&rsquo;
character may be displayed by using the expansion specifier
&rsquo;%%&rsquo;.</p>

<p style="margin-top: 1em">Atom labelling for the currently
selected atoms may be turned off with the command label off.
By default, if no string is given as a parameter, RasMol
uses labels appro&acirc; <br>
priate for the current molecule.</p>

<p style="margin-top: 1em">The colour of each label may be
changed using the colour label command. By default, each
label is drawn in the same colour as the atom to which it is
attached. The size <br>
and spacing of the displayed text may be changed using the
set fontsize command. The width of the strokes in the
displayed text may be changed <br>
using the set fontstroke <br>
command.</p>

<p style="margin-top: 1em">Load Load a molecule coordinate
file into RasMol. Valid molecule file formats are pdb
(Protein Data Bank format), mdl (Molecular Design
Limited&rsquo;s MOL file format), alchemy <br>
(Tripos&rsquo; Alchemy file format), mol2 (Tripos&rsquo;
Sybyl Mol2 file format), charmm (CHARMm file format), xyz
(MSC&rsquo;s XMol XYZ file format), mopac (J. P.
Stewart&rsquo;s MOPAC file for&acirc; <br>
mat) or cif (IUCr CIF or mmCIF file format). If no file
format is specified, PDB, CIF, or mmCIF is assumed by
default. Up to 20 molecules may be loaded at a time. If <br>
CHEM_COMP ligand models are included in an mmCIF file, they
will be loaded as NMR models, first giving the all the NMR
models for model coordinates if specified and then <br>
giving all the NMR models for ideal model coordinates.</p>

<p style="margin-top: 1em">To delete a molecule prior to
loading another use the RasMol zap command. To select a
molecule for manipulation use the RasMol molecule &lt;n&gt;
command.</p>

<p style="margin-top: 1em">The load command selects all the
atoms in the molecule, centres it on the screen and renders
it as a CPK coloured wireframe model. If the molecule
contains no bonds (i.e. <br>
contains only alpha carbons), it is drawn as an alpha carbon
backbone. If the file specifies fewer bonds than atoms,
RasMol determines connectivity using the connect com&acirc;
<br>
mand.</p>

<p style="margin-top: 1em">The load inline command also
allows the storing of atom coordinates in scripts to allow
better integration with WWW browsers. A load command
executed inside a script file <br>
may specify the keyword inline instead of a conventional
filename. This option specifies that the coordinates of the
molecule to load are stored in the same file as the <br>
currently executing commands.</p>

<p style="margin-top: 1em">Map The RasMol map commands
manipulate electron density maps in coordination with the
display of molecules. These commands are very memory
intensive and may not work on <br>
machines with limited memory. Each molecule may have as many
maps as available memory permits. Maps may be read from
files or generated from Gaussian density distribu&acirc;
<br>
tions around atoms.</p>

<p style="margin-top: 1em">map colour, to colour a map
according to a given colour scheme, map generate, to
generate a map from selected atoms based on
pseudo-Gaussians, map level, to set the con&acirc; <br>
touring level for selected maps, map load, to load a map
from a file, map mask to designate a mask for the selected
maps, map resolution, to set the resolution for con&acirc;
<br>
touring selected maps, map restrict, to select one or more
maps and to disable all others, map save, to save map
information to a file, map scale, control the scaling of
<br>
pseudo-Gaussians when generating maps, map select, to select
one or more maps, map show, to display information about one
or more maps or about the parameters to be used <br>
in generating or loading the next map, map spacing, to set
the spacing betwen contour lines of selected maps, map
spread, to set the variance of the Gaussians for map
gen&acirc; <br>
eration as a fraction of the atomic radius, and map zap to
delete previously generated or loaded maps.</p>

<p style="margin-top: 1em">The effect of map generate and
map load commands is modified by the map mask command which
limits the portion of the display space that can be
considered for display of <br>
maps.</p>

<p style="margin-top: 1em">Map colour <br>
The RasMol map colour command colours the selected maps
according to the specified colour scheme. The colour scheme
may be a colour name or and RBG triple in brackets, or <br>
the keyword atom to cause the map points to be coloured by
the color of the nearest atom.</p>

<p style="margin-top: 1em">Map generate <br>
The RasMol map generate command generates a map from
whatever atoms are currently selected, by summing electron
densities approximated by Gaussian distributions. The <br>
height of each Gaussian is determined by the setting of the
map scale command. In the default of map scale true, each
Gaussian has a height proportional element type of <br>
the atom. If the optional &rsquo;LRSurf&rsquo; parameter is
given or if map scale false has been executed, each Gaussian
is scaled so that the Gaussian contour level 1 is at the van
<br>
der Waals radius. In either case a standard deviation
determined by the most recently specified spread or
resolution is used. If a non-zero spread has been given the
<br>
radius of the atom is multiplied by the spread to find the
standard deviation. The default is 2/3rds. If a resolution
has been given, the spread is inferred as 2/3rds of <br>
the resolution.</p>

<p style="margin-top: 1em">For example, if the resolution
is given as 1., and the atom in question is a Carbon with a
van der Waals radius of 468 RasMol units (1.87 Angstroms),
the inferred spead is <br>
.6667, and the standard deviation of the Gaussian is taken
as 1.25 Angstroms.</p>

<p style="margin-top: 1em">If the spread has been set to
zero, the spread for each atom is determined from the van
der Waals radius and the probe atom radius to simulate the
effect of a Lee-Richards <br>
surface.</p>

<p style="margin-top: 1em">If no specific map was given by
the map selector, the new map is given the next available
map number.</p>

<p style="margin-top: 1em">If a specific map was given by
the map selector, the new map replaces that map. If more
than one map was given by the map selector, the new map
replaces the lowest num&acirc; <br>
bered of the selected maps. In any case the new map becomes
the currently selected map.</p>

<p style="margin-top: 1em">The map is displayed as dots,
mesh or a surface, depending on the last map rendering mode
selected or the mode selected on the command itself.</p>

<p style="margin-top: 1em">Map level <br>
The RasMol map level command sets the contour level to be
used in creating subsequent representations of generated or
loaded maps. If the keyword MEAN in used the level <br>
is relative to the mean of the map data. Otherwise the level
is absolute.</p>

<p style="margin-top: 1em">In general, a lower level
results in a map containing more of the displayed volume,
while a higher level results in a map containing less of the
displayed volume.</p>

<p style="margin-top: 1em">Map load <br>
The RasMol map load command loads a map file into RasMol.
The valid formats are CCP4 map format and imgCIF format.</p>

<p style="margin-top: 1em">If no specific map was given by
the map selector, the new map is given the next available
map number.</p>

<p style="margin-top: 1em">If a specific map was given by
the map selector, the new map replaces that map. If more
than one map was given by the map selector, the new map
replaces the lowest num&acirc; <br>
bered of the selected maps. In any case the new map becomes
the currently selected map.</p>

<p style="margin-top: 1em">The map is displayed as dots,
mesh or a surface depending on the last map rendering mode
selected.</p>

<p style="margin-top: 1em">Map mask <br>
The RasMol map mask command specifies a mask to be used to
limit the display space to be used for making
representations of other maps or removes an earlier mask
specifi&acirc; <br>
cation.</p>

<p style="margin-top: 1em">The &rsquo;selected&rsquo;
option indicates that the mask is to be created from the
currently selected atoms. The &rsquo;&lt;number&gt;&rsquo;
option indicates that the mask is to be copied from the <br>
map of the number specified. The &rsquo;none&rsquo; option
removes the previously specified mask, if any.</p>

<p style="margin-top: 1em">The map selector specifies the
map or maps to which the specified mask will the applied.
For example, &rsquo;map next mask selected&rsquo; specifies
that the currently selected atoms <br>
are to be used to generate a mask to be applied to any maps
created by subsequent &rsquo;map load&rsquo; or &rsquo;map
generate&rsquo; commands.</p>

<p style="margin-top: 1em">Any map may be used as a mask.
The portions of the mask map greater than than or equal to
the average value of the mask map allow the values of the
map being masked to be <br>
used as given. The portions of the mask map lower than the
average value of the mask map cause the values of the map
being masked to be treated as if they were equal to <br>
the lowest data value of the map being masked.</p>

<p style="margin-top: 1em">Map resolution <br>
The RasMol map resolution command specifies the resolution
in RasMol units or, if a number containing a decimal point
is given, the resolution in Angstroms to be used in <br>
generating and in representing maps.</p>

<p style="margin-top: 1em">The resolution is used at the
map spacing for representations of maps, indicating the
separation between contour levels (see the map spacing
command) and to infer the map <br>
spread to be used in generated maps from selected atoms (see
the map spread command). The map spread is set to two thirds
of the specified resolution.</p>

<p style="margin-top: 1em">Map restrict <br>
The RasMol map restrict command selects particular maps to
make them active for subsequent map commands. This is
similar to the map select command, but does disables the
<br>
display of the maps that were not selected.</p>

<p style="margin-top: 1em">Map save <br>
The RasMol map save command saves an imgCIF map file.</p>

<p style="margin-top: 1em">If no specific map was given by
the map selector, the currently selected maps and their
masks are written to the file, one map and mask pair per
data block.</p>

<p style="margin-top: 1em">Map scale <br>
The RasMol map scale command selects the scaling of
pseudo-Gaussians in the map generate commands. In the
default of map scale true, each Gaussian has a height
propor&acirc; <br>
tional element type of the atom. If map scale false has been
executed, each Gaussian is scaled so that the Gaussian
contour level 1 is at the van der Waals radius. In <br>
either case a standard deviation determined by the most
recently specified spread or resolution is used.</p>

<p style="margin-top: 1em">Map select <br>
The RasMol map select command selects particular maps to
make them active for subsequent map commands. This is
similar to the map restrict command, but does not disable
<br>
the display of the maps that were not selected.</p>

<p style="margin-top: 1em">If the optional atom parameter
is given, the command selects the atoms with centres closest
to the map points. The radius of the search may be specified
by the parameter <br>
search_radius. The default is to look for atoms within 4
Angstroms plus the probe radius. If the optional within
parameter is given, the new selection is taken from <br>
within the currently selected atoms. If the options add
parameter is given, the new selection is added to the
currently selected atoms. The default is to search within
<br>
all atoms.</p>

<p style="margin-top: 1em">Map show <br>
The RasMol map show command causes information about the
maps specified by the map selector to be written to the
command window.</p>

<p style="margin-top: 1em">Map spacing <br>
The RasMol map spacing command specifies the spacing to be
used between contour lines in creating representations of
maps. The spacing is typically <br>
given in Angstroms with a decimal point, but may also be
specified in RasMol units (250ths of an Angstom) as an
integer. For maps loaded in grid coordinates that spacing
<br>
is parallel to the cell edges. The default spacing is one
half Angstrom.</p>

<p style="margin-top: 1em">Map spread <br>
The RasMol map spread command specifies the reciprocal of
the number of standard deviations per radius to be used in
generating maps as sums of Gaussians centered on <br>
atomic positions. The default spread is one two thirds (i.e.
each radius covers 1.5 standard deviations).</p>

<p style="margin-top: 1em">If the spread has been set to
zero, the spread for each atom is determined from the van
der Waals radius and the probe atom radius to simulate the
effect of a Lee-Richards <br>
surface.</p>

<p style="margin-top: 1em">Map zap <br>
The RasMol map zap command removes the data and
representations of the maps specified by the map selector.
The map numbers of maps that have not been removed are not
<br>
changed.</p>

<p style="margin-top: 1em">Molecule <br>
The RasMol molecule command selects one of up to 5
previously loaded molecules for active manipulation. While
all the molcules are displayed and may be rotated
collec&acirc; <br>
tively (see the rotate all command), only one molecule at a
time time is active for manipulation by the commands which
control the details of rendering.</p>

<p style="margin-top: 1em">Monitor <br>
The RasMol monitor command allows the display of distance
monitors. A distance monitor is a dashed (dotted) line
between an arbitrary pair of atoms, optionally labelled by
<br>
the distance between them. The RasMol command monitor
&lt;number&gt; &lt;number&gt; adds such a distance monitor
between the two atoms specified by the atom serial numbers
given as <br>
parameters</p>

<p style="margin-top: 1em">Distance monitors are turned off
with the command monitors off. By default, monitors display
the distance between its two end points as a label at the
centre of the moni&acirc; <br>
tor. These distance labels may be turned off with the
command set monitors off, and re-enabled with the command
set monitors on. Like most other representations, the <br>
colour of a monitor is taken from the colour of its end
points unless specified by the colour monitors command.</p>

<p style="margin-top: 1em">Distance monitors may also be
added to a molecule interactively with the mouse, using the
set picking monitor command. Clicking on an atom results in
its being identified <br>
on the rasmol command line. In addition every atom picked
increments a modulo counter such that, in monitor mode,
every second atom displays the distance between this atom
<br>
and the previous one. The shift key may be used to form
distance monitors between a fixed atom and several
consecutive positions. A distance monitor may also be
removed <br>
(toggled) by selecting the appropriate pair of atom end
points a second time.</p>

<p style="margin-top: 1em">Notoggle <br>
The RasMol NoToggle command enables or disables the use of
the toggle ability that is used by some of the other RasMol
commands. When no boolean value is specified, <br>
NoToggle mode is ENABLED. When NoToggle mode is ENABLED, all
toggle functionality is DISABLED. To turn it off, one must
explicitly set notoggle off.</p>

<p style="margin-top: 1em">Some commands which use the
toggle feature are: ColourMode. More functions that utilize
this capability may be added at a later date.</p>

<p style="margin-top: 1em">Pause The RasMol pause command
is used in script files to stop the script file for local
manipulation by a mouse, until any key is pushed to restart
the script file. Wait is <br>
synonymous with pause. This command may be executed in
RasMol script files to suspend the sequential execution of
commands and allow the user to examine the current <br>
image. When RasMol executes a pause command in a script
file, it suspends execution of the rest of the file,
refreshes the image on the screen and allows the
manipula&acirc; <br>
tion of the image using the mouse and scroll bars, or
resizing of the graphics window. Once a key is pressed,
control returns to the script file at the line following
<br>
the pause command. While a script is suspended the molecule
may be rotated, translated, scaled, slabbed and picked as
usual, but all menu commands are disabled.</p>

<p style="margin-top: 1em">Play The RasMol play command
specifies the recording medium from which to play back a
movie. The playback frame start time is given in seconds to
millisecond precision. Since <br>
we are working on computers, the medium is specified as a
set of files, each marked with the playback frame start time
in milliseconds as part of the name. The place in <br>
the name at which to look for the playback frame start time
in milliseconds is marked by the characters
&quot;ssssss&quot; with an appropriate number of digits.
RasMol accepts <br>
either upper or lower case s&rsquo;s or decimal digits to
mark the place for the time. The play off and play eject
commands effectively remove the specified medium from use.
If <br>
no medium is specified, play off suspends playing and play
on resumes playing. Normally play starts immediately and
runs to the end of the medium. However, if play off <br>
and/or or some combination of play from and play until is
entered before play type medium, those settings will be
used.</p>

<p style="margin-top: 1em">As of release 2.7.5, RasMol
support play from scripts and data files.</p>

<p style="margin-top: 1em">Print The RasMol print command
sends the currently displayed image to the local default
printer using the operating system&rsquo;s native printer
driver. Note: this command is not yet <br>
supported under UNIX or VMS. It is intended to take
advantage of Microsoft Windows and Apple Macintosh printer
drivers. For example, this allows images to be printed <br>
directly on a dot matrix printer.</p>

<p style="margin-top: 1em">When using RasMol on a UNIX or
VMS system this functionality may be achieved by either
generating a PostScript file using the RasMol write ps or
write vectps commands and <br>
printing that or generating a raster image file and using a
utility to dump that to the local printer.</p>

<p style="margin-top: 1em">Quit Exit from the RasMol
program. The RasMol commands exit and quit are synonymous,
except within nested scripts. In that case, exit terminates
only the current level, while <br>
quit terminates all nested levels of scripts.</p>

<p style="margin-top: 1em">Record The RasMol record command
specifies the recording medium to hold the movie. Since we
are working on computers, the medium is specified as a
template for a set of files, <br>
each marked with the playback frame start time in
milliseconds (rather than as seconds to avoid embedding a
decimal point) as part of the name. The place in the name to
be <br>
replaced with the playback frame start time in milliseconds
is marked by the characters &quot;ssssss&quot; with an
appropriate number of digits. RasMol accepts either upper or
lower <br>
case s&rsquo;s or decimal digits to mark the place for the
time. The record off commands remove the specified medium
from use. If no medium is specified, record off suspends
<br>
recording and record on resumes recording with the next
available time on the same medium. The screen is the default
medium and is, by default, on. Writing to disk must be <br>
explicitly specified so that the disk does not get filled up
unintentionally. The type of a recording medium may be an
image type such as gif, pict or png to record the <br>
actual screen images or script to record the RasMol commands
used to generate the frames.</p>

<p style="margin-top: 1em">Normally recording starts at
playback frame start time 0 seconds. A non-zero starting
time in seconds can be specified with the record from
command as in record from 25 <br>
or record from 37.25 to help in organizing scenes of movies
to be assembled later in an appropriate order. The record
until command allows an upper limit to be set on <br>
recording time in seconds. The default is to have no limit.
Issuing the commands</p>

<p style="margin-top: 1em">record from 600</p>

<p style="margin-top: 1em">record until 1800</p>

<p style="margin-top: 1em">would result in a 20 minute
movie segment intended to start 10 minutes into a longer
movie. These commands allow control over rewriting selected
time segments.</p>

<p style="margin-top: 1em">Refresh <br>
The RasMol refresh command redraws the current image. This
is useful in scripts to ensure application of a complex list
of parameter changes.</p>

<p style="margin-top: 1em">Renumber <br>
The RasMol renumber command sequentially numbers the
residues in a macromolecular chain. The optional parameter
specifies the value of the first residue in the sequence.
<br>
By default, this value is one. For proteins, each amino acid
is numbered consecutively from the N terminus to the C
terminus. For nucleic acids, each base is numbered from <br>
the 5&rsquo; terminus to the 3&rsquo; terminus. All chains
in the current database are renumbered and gaps in the
original sequence are ignored. The starting value for
numbering may <br>
be negative.</p>

<p style="margin-top: 1em">Reset The RasMol reset command
restores the original viewing transformation and centre of
rotation. The scale is set to its default value, zoom 100,
the centre of rotation is <br>
set to the geometric centre of the currently loaded
molecule, centre all, this centre is translated to the
middle of the screen and the viewpoint set to the default
orien&acirc; <br>
tation.</p>

<p style="margin-top: 1em">This command should not be
mistaken for the RasMol zap command which deletes the
currently stored molecule, returning the program to its
initial state.</p>

<p style="margin-top: 1em">Restrict <br>
The RasMol restrict command both defines the currently
selected region of the molecule and disables the
representation of (most of) those parts of the molecule no
longer <br>
selected. All subsequent RasMol commands that modify a
molecule&rsquo;s colour or representation affect only the
currently selected region. The parameter of a restrict
command <br>
is a RasMol atom expression that is evaluated for every atom
of the current molecule. This command is very similar to the
RasMol select command, except restrict disables <br>
the wireframe, spacefill and backbone representations in the
non-selected region.</p>

<p style="margin-top: 1em">Type &quot;help expression&quot;
for more information on RasMol atom expressions or see
section Atom Expressions.</p>

<p style="margin-top: 1em">Ribbons <br>
The RasMol ribbons command displays the currently loaded
protein or nucleic acid as a smooth solid &quot;ribbon&quot;
surface passing along the backbone of the protein. The
ribbon <br>
is drawn between each amino acid whose alpha carbon is
currently selected. The colour of the ribbon is changed by
the RasMol colour ribbon command. If the current ribbon <br>
colour is none (the default), the colour is taken from the
alpha carbon at each position along its length.</p>

<p style="margin-top: 1em">The width of the ribbon at each
position is determined by the optional parameter in the
usual RasMol units. By default the width of the ribbon is
taken from the secondary <br>
structure of the protein or a constant value of 720 (2.88
Angstroms) for nucleic acids. The default width of protein
alpha helices and beta sheets is 380 (1.52 Angstroms) <br>
and 100 (0.4 Angstroms) for turns and random coil. The
secondary structure assignment is either from the PDB file
or calculated using the DSSP algorithm as used by the <br>
structure command. This command is similar to the RasMol
command strands which renders the biomolecular ribbon as
parallel depth-cued curves.</p>

<p style="margin-top: 1em">Rotate Rotate the molecule about
the specified axis. Permitted values for the axis parameter
are &quot;x&quot;, &quot;y&quot;, &quot;z&quot; and
&quot;bond&quot;. The integer parameter states the angle in
degrees for <br>
the structure to be rotated. For the X and Y axes, positive
values move the closest point up and right, and negative
values move it down and left, respectively. For the Z <br>
axis, a positive rotation acts clockwise and a negative
angle anti-clockwise.</p>

<p style="margin-top: 1em">Alternatively, this command may
be used to specify which rotations the mouse or dials will
control. If rotate bond true is selected, the horizontal
scroll bar will con&acirc; <br>
trol rotation around the axis selected by the bond src dst
pick command. If rotate all true is selected, and multiple
molecules have been loaded, then all molecules will <br>
rotate together. In all other cases, the mouseand dials
control the the rotation of the molecule selected by the
molecule n command.</p>

<p style="margin-top: 1em">Russian <br>
The RasMol Russian command sets the menus and messages to
the Russian versions.</p>

<p style="margin-top: 1em">This command may not work
correctly unless appropriate fonts have been installed. The
commands Bulgarian, Chinese, English, French, Italian,
Russian and Spanish may be <br>
used to select Bulgarian, Chinese, English, French, Italian,
Japanese, Russian and Spanish menus and messages if the
appropriate fonts have been installed.</p>

<p style="margin-top: 1em">Save Save the currently selected
set of atoms in a Protein Data Bank (PDB), MDL, Alchemy(tm)
or XYZ format file. The distinction between this command and
the RasMol write com&acirc; <br>
mand has been dropped. The only difference is that without a
format specifier the save command generates a PDB file and
the write command generates a GIF image.</p>

<p style="margin-top: 1em">Script The RasMol script command
reads a set of RasMol commands sequentially from a text file
and executes them. This allows sequences of commonly used
commands to be stored and <br>
performed by single command. A RasMol script file may
contain a further script command up to a maximum
&quot;depth&quot; of 10, allowing complicated sequences of
actions to be exe&acirc; <br>
cuted. RasMol ignores all characters after the first
&rsquo;#&rsquo; character on each line allowing the scripts
to be annotated. Script files are often also annotated using
the Ras&acirc; <br>
Mol echo command.</p>

<p style="margin-top: 1em">The most common way to generate
a RasMol script file is to use the write script or write
rasmol commands to output the sequence of commands that are
needed to regenerate <br>
the current view, representation and colouring of the
currently displayed molecule.</p>

<p style="margin-top: 1em">The RasMol command source is
synonymous with the script command.</p>

<p style="margin-top: 1em">Select Define the currently
selected region of the molecule. All subsequent RasMol
commands that manipulate a molecule or modify its colour or
representation only affect the cur&acirc; <br>
rently selected region. The parameter of a select command is
a RasMol expression that is evaluated for every atom of the
current molecule. The currently selected (active) <br>
region of the molecule are those atoms that cause the
expression to evaluate true. To select the whole molecule
use the RasMol command select all. The behaviour of the <br>
select command without any parameters is determined by the
RasMol hetero and hydrogen parameters.</p>

<p style="margin-top: 1em">Type &quot;help expression&quot;
for more information on RasMol atom expressions or see
section Atom Expressions.</p>

<p style="margin-top: 1em">Set The RasMol set command
allows the user to alter various internal program parameters
such as those controlling rendering options. Each parameter
has its own set or permis&acirc; <br>
sible parameter options. Typically, omitting the paramter
option resets that parameter to its default value. A list of
valid parameter names is given below.</p>

<p style="margin-top: 1em">Show The RasMol show command
display details of the status of the currently loaded
molecule. The command show information lists the
molecule&rsquo;s name, classification, PDB code <br>
and the number of atoms, chains, groups it contains. If
hydrogen bonding, disulphide bridges or secondary structure
have been determined, the number of hbonds, ssbonds, <br>
helices, ladders and turns are also displayed, respectively.
The command show centre shows any non-zero centering values
selected by the centre [CenX, CenY, CenZ] command. <br>
The command show phipsi shows the phi and psi angles of the
currently selected residues and the omega angles of cis
peptide bonds. The command show RamPrint (or &rsquo;show
RPP&rsquo; <br>
or &rsquo;show RamachandranPrinterPlot&rsquo;) shows a
simple Ramachandran printer plot in the style of Frances
Bernstein&rsquo;s fisipl program. The command show rotation
(or &rsquo;show rot&rsquo; <br>
or &rsquo;show &rsquo;rotate&rsquo;) shows the currently
selected values of z, y, x and bond rotations, if any. The
command show selected (or &rsquo;show selected group&rsquo;
or &rsquo;show selected chain&rsquo; <br>
or &rsquo;show selected atom&rsquo; ) shows the groups
(default), chains or atoms of the current selection. The
command show sequence lists the residues that comprise each
chain of <br>
the molecule. The command show symmetry shows the space
group and unit cell of the molecule. The command show
translation shows any non-zero translation values selected
<br>
by the translate &lt;axis&gt; &lt;value&gt; command. The
command show zoom shows any non-zero zoom value selected by
the zoom &lt;value&gt; command.</p>

<p style="margin-top: 1em">Slab The RasMol slab command
enables, disables or positions the z-clipping plane of the
molecule. The program only draws those portions of the
molecule that are further from <br>
the viewer than the slabbing plane. Integer values range
from zero at the very back of the molecule to 100 which is
completely in front of the molecule. Intermediate val&acirc;
<br>
ues determine the percentage of the molecule to be
drawn.</p>

<p style="margin-top: 1em">This command interacts with the
depth &lt;value&gt; command, which clips to the rear of a
given z-clipping plane.</p>

<p style="margin-top: 1em">Spacefill <br>
The RasMol spacefill command is used to represent all of the
currently selected atoms as solid spheres. This command is
used to produce both union-of-spheres and ball-and- <br>
stick models of a molecule. The command, spacefill true, the
default, represents each atom as a sphere of van der Waals
radius. The command spacefill off turns off the <br>
representation of the selected atom as spheres. A sphere
radius may be specified as an integer in RasMol units
(1/250th Angstrom) or a value containing a decimal point. A
<br>
value of 500 (2.0 Angstroms) or greater results in a
&quot;Parameter value too large&quot; error.</p>

<p style="margin-top: 1em">The temperature option sets the
radius of each sphere to the value stored in its temperature
field. Zero or negative values have no effect and values
greater than 2.0 are <br>
truncated to 2.0. The user option allows the radius of each
sphere to be specified by additional lines in the
molecule&rsquo;s PDB file using Raster 3D&rsquo;s COLOUR
record exten&acirc; <br>
sion.</p>

<p style="margin-top: 1em">The RasMol command cpk is
synonymous with the spacefill command.</p>

<p style="margin-top: 1em">The RasMol command cpknew is
synonymous with the spacefill command, except that a
slightly different set of colours is used.</p>

<p style="margin-top: 1em">Spanish <br>
The RasMol Spanish command sets the menus and messages to
the Spanish versions.</p>

<p style="margin-top: 1em">This command may not work
correctly unless appropriate fonts have been installed. The
commands Bulgarian, Chinese, English, French, Italian,
Russian and Spanish may be <br>
used to select Bulgarian, Chinese, English, French, Italian,
Japanese, Russian and Spanish menus and messages if the
appropriate fonts have been installed.</p>

<p style="margin-top: 1em">SSBonds <br>
The RasMol ssbonds command is used to represent the
disulphide bridges of the protein molecule as either dotted
lines or cylinders between the connected cysteines. The <br>
first time that the ssbonds command is used, the program
searches the structure of the protein to find half-cysteine
pairs (cysteines whose sulphurs are within 3 Angstroms <br>
of each other) and reports the number of bridges to the
user. The command ssbonds on displays the selected
&quot;bonds&quot; as dotted lines, and the command ssbonds
off disables <br>
the display of ssbonds in the currently selected area.
Selection of disulphide bridges is identical to normal
bonds, and may be adjusted using the RasMol set bondmode
com&acirc; <br>
mand. The colour of disulphide bonds may be changed using
the colour ssbonds command. By default, each disulphide bond
has the colours of its connected atoms.</p>

<p style="margin-top: 1em">By default disulphide bonds are
drawn between the sulphur atoms within the cysteine groups.
By using the set ssbonds command the position of the
cysteine&rsquo;s alpha carbons <br>
may be used instead.</p>

<p style="margin-top: 1em">Star The RasMol star command is
used to represent all of the currently selected atoms as
stars (six strokes, one each in the x, -x, y, -y, z and -z
directions). The commands <br>
select not bonded followed by star 75 are useful to mark
unbonded atoms in a wireframe display with less overhead
than provided by spacefill 75. This can be done
automat&acirc; <br>
ically for all subsequent wireframe displays with the
command set bondmode not bonded.</p>

<p style="margin-top: 1em">The command star true, the
default, represents each atom as a star with strokes length
equal to van der Waals radius. The command star off turns
off the representation of <br>
the selected atom as stars. A star stroke length may be
specified as an integer in RasMol units (1/250th Angstrom)
or a value containing a decimal point. A value of 500 <br>
(2.0 Angstroms) or greater results in a &quot;Parameter
value too large&quot; error.</p>

<p style="margin-top: 1em">The temperature option sets the
stroke length of each star to the value stored in its
temperature field. Zero or negative values have no effect
and values greater than 2.0 <br>
are truncated to 2.0. The user option allows the stroke
length of each star to be specified by additional lines in
the molecule&rsquo;s PDB file using Raster 3D&rsquo;s COLOUR
record <br>
extension.</p>

<p style="margin-top: 1em">The RasMol spacefill command can
be used for more artistic rendering of atoms as spheres.</p>

<p style="margin-top: 1em">Stereo The RasMol stereo command
provides side-by-side stereo display of images. Stereo
viewing of a molecule may be turned on (and off) either by
selecting Stereo from the <br>
Options menu, or by typing the commands stereo on or stereo
off.</p>

<p style="margin-top: 1em">Starting with RasMol version
2.7.2.1, the Stereo menu selection and the command stereo
without arguments cycle from the initial state of stereo off
to stereo on in cross- <br>
eyed mode to stereo on in wall-eyed mode and then back to
stereo off.</p>

<p style="margin-top: 1em">The separation angle between the
two views may be adjusted with the set stereo [-]
&lt;number&gt; command, where positive values result in
crossed eye viewing and negative val&acirc; <br>
ues in relaxed (wall-eyed) viewing. The inclusion of [-]
&lt;number&gt; in the stereo command, as for example in
stereo 3 or stereo -5, also controls angle and
direction.</p>

<p style="margin-top: 1em">The stereo command is only
partially implemented. When stereo is turned on, the image
is not properly recentred. (This can be done with a
translate x -&lt;number&gt; <br>
command.) It is not supported in vector PostScript output
files, is not saved by the write script command, and in
general is not yet properly interfaced with several <br>
other features of the program.</p>

<p style="margin-top: 1em">Strands <br>
The RasMol strands command displays the currently loaded
protein or nucleic acid as a smooth &quot;ribbon&quot; of
depth-cued curves passing along the backbone of the protein.
The <br>
ribbon is composed of a number of strands that run parallel
to one another along the peptide plane of each residue. The
ribbon is drawn between each amino acid whose alpha <br>
carbon is currently selected. The colour of the ribbon is
changed by the RasMol colour ribbon command. If the current
ribbon colour is none (the default), the colour is <br>
taken from the alpha carbon at each position along its
length. The central and outermost strands may be coloured
independently using the colour ribbon1 and colour ribbon2
<br>
commands, respectively. The number of strands in the ribbon
may be altered using the RasMol set strands command.</p>

<p style="margin-top: 1em">The width of the ribbon at each
position is determined by the optional parameter in the
usual RasMol units. By default the width of the ribbon is
taken from the secondary <br>
structure of the protein or a constant value of 720 for
nucleic acids (which produces a ribbon 2.88 Angstroms wide).
The default width of protein alpha helices and beta <br>
sheets is 380 (1.52 Angstroms) and 100 (0.4 Angstroms) for
turns and random coil. The secondary structure assignment is
either from the PDB file or calculated using the <br>
DSSP algorithm as used by the structure command. This
command is similar to the RasMol command ribbons which
renders the biomolecular ribbon as a smooth shaded
surface.</p>

<p style="margin-top: 1em">Structure <br>
The RasMol structure command calculates secondary structure
assignments for the currently loaded protein. If the
original PDB file contained structural assignment records
<br>
(HELIX, SHEET and TURN) these are discarded. Initially, the
hydrogen bonds of the current molecule are found, if this
hasn&rsquo;t been done already. The secondary structure is
<br>
then determined using Kabsch and Sander&rsquo;s DSSP
algorithm. Once finished the program reports the number of
helices, strands and turns found.</p>

<p style="margin-top: 1em">Surface <br>
The RasMol surface command renders a Lee-Richards molecular
surface resulting from rolling a probe atom on the selected
atoms. The value given specifies the radius of the <br>
probe. If given in the first form, the evolute of the
surface of the probe is shown (the solvent excluded
surface). If given in the second form, the envelope of the
<br>
positions of the center of the probe is shown (the solvent
accessible surface).</p>

<p style="margin-top: 1em">Trace The RasMol trace command
displays a smooth spline between consecutive alpha carbon
positions. This spline does not pass exactly through the
alpha carbon position of each <br>
residue, but follows the same path as ribbons, strands and
cartoons. Note that residues may be displayed as ribbons,
strands, cartoons or as a trace. Enabling one of <br>
these representations disables the others. However, a
residue may be displayed simultaneously as backbone and as
one of the above representations. This may change in <br>
future versions of RasMol. Prior to version 2.6, trace was
synonymous with backbone.</p>

<p style="margin-top: 1em">Trace temperature displays the
backbone as a wider cylinder at high temperature factors and
thinner at lower. This representation is useful to X-ray
crystallographers and <br>
NMR spectroscopists.</p>

<p style="margin-top: 1em">Translate <br>
The RasMol translate command moves the position of the
centre of the molecule on the screen. The axis parameter
specifies along which axis the molecule is to be moved and
<br>
the integer parameter specifies the absolute position of the
molecule centre from the middle of the screen. Permitted
values for the axis parameter are &quot;x&quot;,
&quot;y&quot; and &quot;z&quot;. <br>
Displacement values must be between -100 and 100 which
correspond to moving the current molecule just off the
screen. A positive &quot;x&quot; displacement moves the
molecule to the <br>
right, and a positive &quot;y&quot; displacement moves the
molecule down the screen. The pair of commands translate x 0
and translate y 0 centres the molecule on the screen.</p>

<p style="margin-top: 1em">UnBond The RasMol command unbond
&lt;number&gt; &lt;number&gt; removes the designated bond
from the drawing.</p>

<p style="margin-top: 1em">The command unbond without
arguments removes a bond previously picked by the bond
&lt;number&gt; &lt;number&gt; pick command.</p>

<p style="margin-top: 1em">Wireframe <br>
The RasMol wireframe command represents each bond within the
selected region of the molecule as a cylinder, a line or a
depth-cued vector. The display of bonds as depth- <br>
cued vectors (drawn darker the further away from the viewer)
is turned on by the command wireframe or wireframe on. The
selected bonds are displayed as cylinders by spec&acirc;
<br>
ifying a radius either as an integer in RasMol units or
containing a decimal point as a value in Angstroms. A
parameter value of 500 (2.0 Angstroms) or above results in
<br>
an &quot;Parameter value too large&quot; error. Bonds may be
coloured using the colour bonds command.</p>

<p style="margin-top: 1em">If the selected bonds involved
atoms of alternate conformers then the bonds are narrowed in
the middle to a radius of .8 of the specified radius (or to
the radius specifed <br>
as the optional second parameter).</p>

<p style="margin-top: 1em">Non-bonded atoms, which could
become invisible in an ordinary wireframe display can be
marked by a preceding set bondmode not bonded command. If
nearly co-linear bonds to <br>
atoms cause them to be difficult to see in a wireframe
display, the set bondmode all command will add markers for
all atoms in subsequent wireframe command executions.</p>

<p style="margin-top: 1em">Write Write the current image to
a file in a standard format. Currently supported image file
formats include bmp (Microsoft bitmap) and gif (Compuserve
GIF), iris (IRIS RGB), <br>
ppm (Portable Pixmap), ras (Sun rasterfile), ps and epsf
(Encapsulated PostScript), monops (Monochrome Encapsulated
PostScript), pict (Apple PICT), vectps (Vector Post&acirc;
<br>
script). The write command may also be used to generate
command scripts for other graphics programs. The format
script writes out a file containing the RasMol script
com&acirc; <br>
mands to reproduce the current image. The format molscript
writes out the commands required to render the current view
of the molecule as ribbons in Per Kraulis&rsquo; Molscript
<br>
program and the format kinemage the commands for David
Richardson&rsquo;s program Mage. The following formats are
useful for further processing: povray (POVRay 2), povray3
<br>
(POVRay 3 -- under development), vrml (VRML file). Finally,
several formats are provided to provide phi-psi data for
listing or for phipsi (phi-psi data as an annotated <br>
list with cis omegas), ramachan and RDF and
RamachandranDataFile (phi-psi data as columns of numbers for
gnuplot), RPP and RamachandranPrinterPlot (phi-psi data as a
<br>
printer plot).</p>

<p style="margin-top: 1em">The distinction between this
command and the RasMol save command has been dropped. The
only difference is that without a format specifier the save
command generates a PDB <br>
file and the write command generates a GIF image.</p>

<p style="margin-top: 1em">Zap Deletes the contents of the
current database and resets parameter variables to their
initial default state.</p>

<p style="margin-top: 1em">Zoom Change the magnification of
the currently displayed image. Boolean parameters either
magnify or reset the scale of current molecule. An integer
parameter specifies the <br>
desired magnification as a percentage of the default scale.
The minimum parameter value is 10; the maximum parameter
value is dependent upon the size of the molecule being <br>
displayed. For medium sized proteins this is about 500.</p>

<p style="margin-top: 1em">SET PARAMETERS <br>
RasMol has a number of internal parameters that may be
modified using the set command. These parameters control a
number of program options such as rendering options and
mouse <br>
button mappings.</p>

<p style="margin-top: 1em">picking play.fps radius
record.aps</p>

<p style="margin-top: 1em">Set Ambient <br>
The RasMol ambient parameter is used to control the amount
of ambient (or surrounding) light in the scene. The ambient
value must be between 0 and 100. It controls the <br>
percentage intensity of the darkest shade of an object. For
a solid object, this is the intensity of surfaces facing
away from the light source or in shadow. For depth- <br>
cued objects this is the intensity of objects furthest from
the viewer.</p>

<p style="margin-top: 1em">This parameter is commonly used
to correct for monitors with different &quot;gamma
values&quot; (brightness), to change how light or dark a
hardcopy image appears when printed or to <br>
alter the feeling of depth for wireframe or ribbon
representations.</p>

<p style="margin-top: 1em">Set Axes <br>
The RasMol axes parameter controls the display of orthogonal
coordinate axes on the current display. The coordinate axes
are those used in the molecule data file, and the <br>
origin is the centre of the molecule&rsquo;s bounding box.
The set axes command is similar to the commands set boundbox
and set unitcell that display the bounding box and the <br>
crystallographic unit cell, respectively.</p>

<p style="margin-top: 1em">Set Backfade <br>
The RasMol backfade parameter is used to control backfade to
the specified background colour, rather than black. This is
controlled by the commands set backfade on and <br>
set backfade off. For example, this may be used to generate
depth-cued images that fade to white, rather than black.</p>

<p style="margin-top: 1em">Set Background <br>
The RasMol background parameter is used to set the colour of
the &quot;canvas&quot; background. The colour may be given
as either a colour name or a comma separated triple of Red,
<br>
Green, Blue (RGB) components enclosed in square brackets.
Typing the command help colours will give a list of the
predefined colour names recognised by RasMol. When
run&acirc; <br>
ning under X Windows, RasMol also recognises colours in the
X server&rsquo;s colour name database.</p>

<p style="margin-top: 1em">The command set background is
synonymous with the RasMol command background.</p>

<p style="margin-top: 1em">Set BondMode <br>
The RasMol set bondmode command controls the mechanism used
to select individual bonds and modifies the display of
bonded and non-bonded atoms by subsequent wireframe
com&acirc; <br>
mands.</p>

<p style="margin-top: 1em">When using the select and
restrict commands, a given bond will be selected if i) the
bondmode is or and either of the connected atoms is
selected, or ii) the bondmode is <br>
and and both atoms connected by the bond are selected. Hence
an individual bond may be uniquely identified by using the
command set bondmode and and then uniquely select&acirc;
<br>
ing the atoms at both ends.</p>

<p style="margin-top: 1em">The bondmode [all | none | not
bonded] commands add star 75 or spacefill 75 markers for the
designated atoms to wireframe displays. Stars are used when
the specified <br>
wireframe radius is zero.</p>

<p style="margin-top: 1em">Set Bonds <br>
The RasMol bonds parameter is used to control display of
double and triple bonds as multiple lines or cylinders.
Currently bond orders are only read from MDL Mol files, <br>
Sybyl Mol2 format files, Tripos Alchemy format files, CIF
and mmCIF, and suitable PDB files. Double (and triple) bonds
are specified in some PDB files by specifying a <br>
given bond twice (and three times) in CONECT records. The
command set bonds on enables the display of bond orders, and
the command set bonds off disables them.</p>

<p style="margin-top: 1em">Set BoundBox <br>
The RasMol boundbox parameter controls the display of the
current molecule&rsquo;s bounding box on the display. The
bounding box is orthogonal to the data file&rsquo;s original
coor&acirc; <br>
dinate axes. The set boundbox command is similar to the
commands set axes and set unitcell that display orthogonal
coordinate axes and the bounding box, respectively.</p>

<p style="margin-top: 1em">Set Cartoon <br>
The RasMol cartoon parameter is used to control display of
the cartoon version of the ribbons display. By default, the
C-termini of beta-sheets are displayed as arrow <br>
heads. This may be enabled and disabled using the set
cartoons &lt;boolean&gt; command. The depth of the cartoon
may be adjusted using the cartoons &lt;number&gt; command.
The set <br>
cartoons command without any parameters returns these two
options to <br>
their default values.</p>

<p style="margin-top: 1em">Set CisAngle <br>
The RasMol cisangle parameter controls the cutoff angle for
identifying cis peptide <br>
bonds. If no value is given, the cutoff is set to 90
degrees.</p>

<p style="margin-top: 1em">Set Display <br>
This command controls the display mode within RasMol. By
default, set display normal, RasMol displays the molecule in
the representation specified by the user. The command <br>
set display selected changes the display mode such that the
molecule is temporarily drawn so as to indicate currently
selected portion of the molecule. The user specified <br>
colour scheme and representation remains unchanged. In this
representation all selected atoms are shown in yellow and
all non selected atoms are shown in blue. The colour <br>
of the background is also changed to a dark grey to indicate
the change of display mode. This command is typically only
used by external Graphical User Interfaces (GUIs).</p>

<p style="margin-top: 1em">Set FontSize <br>
The RasMol set fontsize command is used to control the size
of the characters that form atom labels. This value
corresponds to the height of the displayed character in <br>
pixels. The maximum value of fontsize is 48 pixels, and the
default value is 8 pixels high. Fixed or proportional
spacing may be selected by appending the &quot;FS&quot; or
&quot;PS&quot; <br>
modifiers, respectively. The default is &quot;FS&quot;. To
display atom labels on the screen use the RasMol label
command and to change the colour of displayed labels, use
the <br>
colour labels command.</p>

<p style="margin-top: 1em">Set FontStroke <br>
The RasMol set fontstroke command is used to control the
size of the stroke width of the characters that form atom
labels. This value is the radius in pixels of cylinders <br>
used to form the strokes. The special value of &quot;0&quot;
is the default used for the normal single pixel stroke
width, which allows for rapid drawing and rotation of the
image. <br>
Non-zero values are provided to allow for more artistic atom
labels for publication at the expense of extra time in
rendering the image.</p>

<p style="margin-top: 1em">When wider strokes are used, a
larger font size is recommend, e.g. by using the RasMol set
fontsize 24 PS command, followed by set fontstroke 2</p>

<p style="margin-top: 1em">To display atom labels on the
screen use the RasMol label command, and to change the
colour of displayed labels use the colour labels
command.</p>

<p style="margin-top: 1em">Set HBonds <br>
The RasMol hbonds parameter determines whether hydrogen
bonds are drawn between the donor and acceptor atoms of the
hydrogen bond, set hbonds sidechain or between the <br>
alpha carbon atoms of the protein backbone and between the
phosphorous atoms of the nucleic acid backbone, set hbonds
backbone. The actual display of hydrogen bonds is <br>
controlled by the hbonds command. Drawing hydrogen bonds
between protein alpha carbons or nucleic acid phosphorous
atoms is useful when the rest of the molecule is shown <br>
in only a schematic representation such as backbone, ribbons
or strands. This parameter is similar to the RasMol ssbonds
parameter.</p>

<p style="margin-top: 1em">Set Hetero <br>
The RasMol hetero parameter is used to modify the
&rsquo;default&rsquo; behaviour of the RasMol select
command, i.e. the behaviour of select without any
parameters. When this value is <br>
false, the default select region does not include any
heterogeneous atoms (refer to the predefined set hetero ).
When this value is true, the default select region may <br>
contain hetero atoms. This parameter is similar to the
RasMol hydrogen parameter which determines whether hydrogen
atoms should be included in the default set. If both <br>
hetero and hydrogen are true, select without any parameters
is equivalent to select all.</p>

<p style="margin-top: 1em">Set HourGlass <br>
The RasMol hourglass parameter allows the user to enable and
disable the use of the &rsquo;hour glass&rsquo; cursor used
by RasMol to indicate that the program is currently busy
draw&acirc; <br>
ing the next frame. The command set hourglass on enables the
indicator, whilst set hourglass off prevents RasMol from
changing the cursor. This is useful when spinning the <br>
molecule, running a sequence of commands from a script file
or using interprocess communication to execute complex
sequences of commands. In these cases a
&rsquo;flashing&rsquo; cur&acirc; <br>
sor may be distracting.</p>

<p style="margin-top: 1em">Set Hydrogen <br>
The RasMol hydrogen parameter is used to modify the
&quot;default&quot; behaviour of the RasMol select command,
i.e. the behaviour of select without any parameters. When
this value <br>
is false, the default select region does not include any
hydrogen, deuterium or tritium atoms (refer to the
predefined set hydrogen ). When this value is true, the
default <br>
select region may contain hydrogen atoms. This parameter is
similar to the RasMol hetero parameter which determines
whether heterogeneous atoms should be included in the <br>
default set. If both hydrogen and hetero are true, select
without any parameters is equivalent to select all.</p>

<p style="margin-top: 1em">Set Kinemage <br>
The RasMol set kinemage command controls the amount of
detail stored in a Kinemage output file generated by the
RasMol write kinemage command. The output kinemage files
<br>
are intended to be displayed by David Richardson&rsquo;s
Mage program. set kinemage false, the default, only stores
the currently displayed representation in the generated
out&acirc; <br>
put file. The command set kinemage true, generates a more
complex Kinemage that contains both the wireframe and
backbone representations as well as the coordinate axes,
<br>
bounding box and crystal unit cell.</p>

<p style="margin-top: 1em">Set Menus <br>
The RasMol set menus command enables the canvas
window&rsquo;s menu buttons or menu bar. This command is
typically only used by graphical user interfaces or to
create as large <br>
an image as possible when using Microsoft Windows.</p>

<p style="margin-top: 1em">Set Monitor <br>
The RasMol set monitor command enables monitors. The
distance monitor labels may be turned off with the command
set monitor off, and re-enabled with the command set
moni&acirc; <br>
tor on.</p>

<p style="margin-top: 1em">Set Mouse <br>
The RasMol set mouse command sets the rotation, translation,
scaling and zooming mouse bindings. The default value is
rasmol which is suitable for two button mice (for <br>
three button mice the second and third buttons are
synonymous); X-Y rotation is controlled by the first button,
and X-Y translation by the second. Additional functions are
<br>
controlled by holding a modifier key on the keyboard.
[Shift] and the first button performs scaling, [shift] and
the second button performs Z-rotation, and [control] and
<br>
the first mouse button controls the clipping plane. The
insight and quanta options provide the same mouse bindings
as other packages for experienced users.</p>

<p style="margin-top: 1em">Set Picking <br>
The RasMol set picking series of commands affects how a user
may interact with a molecule displayed on the screen in
RasMol.</p>

<p style="margin-top: 1em">Enabling/Disabling Atom
Identification Picking: Clicking on an atom with the mouse
results in identification and the display of its residue
name, residue number, atom <br>
name, atom serial number and chain in the command window.
This behavior may be disabled with the command set picking
none and restored with the command set picking ident. <br>
The command set picking coord adds the atomic coordinates of
the atom to the display.</p>

<p style="margin-top: 1em">Disabling picking, by using set
picking off is useful when executing the pause command in
RasMol scripts as it prevents the display of spurious
message on the command line <br>
while the script is suspended.</p>

<p style="margin-top: 1em">Measuring Distances, Angles and
Torsions: Interactive measurement of distances, angles and
torsions is achieved using the commands: set picking
distance, set picking moni&acirc; <br>
tor, set picking angle and set picking torsion,
respectively. In these modes, clicking on an atom results in
it being identified on the rasmol command line. In addition
<br>
every atom picked increments a modulo counter such that in
distance mode, every second atom displays the distance (or
distance monitor) between this atom and the previous <br>
one. In angle mode, every third atom displays the angle
between the previous three atoms and in torsion mode every
fourth atom displays the torsion between the last four <br>
atoms. By holding down the shift key while picking an atom,
this modulo counter is not incremented and allows, for
example, the distances of consecutive atoms from a fixed
<br>
atom to be displayed. See the monitor command for how to
control the display of distance monitor lines and
labels.</p>

<p style="margin-top: 1em">Labelling Atoms with the Mouse:
The mouse may also be used to toggle the display of an atom
label on a given atom. The RasMol command set picking label
removes a label <br>
from a picked atom if it already has one or displays a
concise label at that atom position otherwise.</p>

<p style="margin-top: 1em">Centring Rotation with the
Mouse: A molecule may be centred on a specified atom
position using the RasMol commands set picking centre or set
picking center. In this mode, <br>
picking an atom causes all futher rotations to be about that
point.</p>

<p style="margin-top: 1em">Picking a Bond as a Rotation
Axis: Any bond may be picked as an axis of rotation for the
portion of the molecule beyond the second atom selected.
This feature should be <br>
used with caution, since, naturally, it changes the
conformation of the molecule. After executing set picking
bond or using the equivalent &quot;Pick Bond&quot; in the
&quot;Settings&quot; <br>
menu, a bond to be rotated is picked with the same sort of
mouse clicks as are used for picking atoms for a distance
measurement. Normally this should be done where a <br>
bond exists, but if no bond exists, it will be added. The
bond cannot be used for rotation if it is part of a ring of
any size. All bonds selected for rotation are <br>
remembered so that they can be properly reported when
writing a script, but only the most recently selected bond
may be actively rotated.</p>

<p style="margin-top: 1em">Enabling Atom/Group/Chain
Selection Picking: Atoms, groups and chains may be selected
(as if with the select command), with the set picking atom,
set picking group, set <br>
picking chain commands. For each of these commands, the
shift key may be used to have a new selection added to the
old, and the control key may be used to have a new <br>
selection deleted from the old. When the set picking atom
command is given, the mouse can be used to pick or to drag a
box around the atoms for which selection is desired. <br>
When the set picking group command is given, picking any an
atom will cause selection of all atoms which agree in
residue number with the picked atom, even if in different
<br>
chains. When the set picking chain command is given, picking
any atom will cause selection of all atoms which agree in
chain identifier with the picked atom.</p>

<p style="margin-top: 1em">Set Play <br>
The RasMol set play.fps command gives the number of frames
per second for playback by the play command (default 24
frames per second).</p>

<p style="margin-top: 1em">In the current release of
RasMol, the play timing is not controlled by this
parameter.</p>

<p style="margin-top: 1em">Set Radius <br>
The RasMol set radius command is used to alter the behaviour
of the RasMol dots command depending upon the value of the
solvent parameter. When solvent is true, the <br>
radius parameter controls whether a true van der
Waals&rsquo; surface is generated by the dots command. If
the value of radius is anything other than zero, that value
is used as <br>
the radius of each atom instead of its true vdW value. When
the value of solvent is true, this parameter determines the
&rsquo;probe sphere&rsquo; (solvent) radius. The parameter
may <br>
be given as an integer in rasmol units or containing a
decimal point in Angstroms. The default value of this
parameter is determined by the value of solvent and changing
<br>
solvent resets radius to its new default value.</p>

<p style="margin-top: 1em">Set Record <br>
The RasMol set record.aps gives the maximum on-screen
velocity in Angstroms per second in animating translations,
rotations and zooms (default 10 A/second).</p>

<p style="margin-top: 1em">The RasMol set record.aps
command gives number of frames per second for recording by
the record command (default 24 frames per second).</p>

<p style="margin-top: 1em">The RasMol set record.dwell
command sets the time in seconds to dwell on a change in
appearance (default .5 sec).</p>

<p style="margin-top: 1em">Set ShadePower <br>
The shadepower parameter (adopted from RasTop) determines
the shade repartition (the contrast) used in rendering solid
objects. This value between 0 and 100 adjusts shad&acirc;
<br>
ing on an object surface oriented along the direction to the
light source. Changing the shadepower parameter does not
change the maximum or the minimum values of this <br>
shading, as does changing the ambient parameter. A value of
100 concentrates the light on the top of spheres, giving a
highly specular, glassy rendering (see the <br>
specpower parameter). A value of 0 distributes the light on
the entire object.</p>

<p style="margin-top: 1em">This implementation of
shadepower differs from the one in RasTop only in the choice
of range (0 to 100 versus -20 to 20 in RasTop).</p>

<p style="margin-top: 1em">Set Shadow <br>
The RasMol set shadow command enables and disables
ray-tracing of the currently rendered image. Currently only
the spacefilling representation is shadowed or can cast <br>
shadows. Enabling shadowing will automatically disable the
Z-clipping (slabbing) plane using the command slab off.
Ray-tracing typically takes about several seconds for a <br>
moderately sized protein. It is recommended that shadowing
be normally disabled whilst the molecule is being
transformed or manipulated, and only enabled once an
appropi&acirc; <br>
ate viewpoint is selected, to provide a greater impression
of depth.</p>

<p style="margin-top: 1em">Set SlabMode <br>
The RasMol slabmode parameter controls the rendering method
of objects cut by the slabbing (z-clipping) plane. Valid
slabmode parameters are &quot;reject&quot;,
&quot;half&quot;, &quot;hollow&quot;, <br>
&quot;solid&quot; and &quot;section&quot;.</p>

<p style="margin-top: 1em">Set Solvent <br>
The RasMol set solvent command is used to control the
behaviour of the RasMol dots command. Depending upon the
value of the solvent parameter, the dots command either
gen&acirc; <br>
erates a van der Waals&rsquo; or a solvent accessible
surface around the currently selected set of atoms. Changing
this parameter automatically resets the value of the RasMol
<br>
radius parameter. The command set solvent false, the default
value, indicates that a van der Waals&rsquo; surface should
be generated and resets the value of radius to zero. <br>
The command set solvent true indicates that a
&rsquo;Connolly&rsquo; or &rsquo;Richards&rsquo; solvent
accessible surface should be drawn and sets the radius
parameter, the solvent radius, to 1.2 <br>
Angstroms (or 300 RasMol units).</p>

<p style="margin-top: 1em">Set Specular <br>
The RasMol set specular command enables and disables the
display of specular highlights on solid objects drawn by
RasMol. Specular highlights appear as white reflections <br>
of the light source on the surface of the object. The
current RasMol implementation uses an approximation function
to generate this highlight.</p>

<p style="margin-top: 1em">The specular highlights on the
surfaces of solid objects may be altered by using the
specular reflection coefficient, which is altered using the
RasMol set specpower com&acirc; <br>
mand.</p>

<p style="margin-top: 1em">Set SpecPower <br>
The specpower parameter determines the shininess of solid
objects rendered by RasMol. This value between 0 and 100
adjusts the reflection coefficient used in specular <br>
highlight calculations. The specular highlights are enabled
and disabled by the RasMol set specular command. Values
around 20 or 30 produce plastic looking surfaces. High <br>
values represent more shiny surfaces such as metals, while
lower values produce more diffuse/dull surfaces.</p>

<p style="margin-top: 1em">Set SSBonds <br>
The RasMol ssbonds parameter determines whether disulphide
bridges are drawn between the sulphur atoms in the sidechain
(the default) or between the alpha carbon atoms in <br>
the backbone of the cysteines residues. The actual display
of disulphide bridges is controlled by the ssbonds command.
Drawing disulphide bridges between alpha carbons is <br>
useful when the rest of the protein is shown in only a
schematic representation such as backbone, ribbons or
strands. This parameter is similar to the RasMol hbonds <br>
parameter.</p>

<p style="margin-top: 1em">Set Stereo <br>
The RasMol set stereo parameter controls the separation
between the left and right images. Turning stereo on and off
doesn&rsquo;t reposition the centre of the molecule.</p>

<p style="margin-top: 1em">Stereo viewing of a molecule may
be turned on (and off) either by selecting Stereo from the
Options menu, or by typing the commands stereo on or stereo
off.</p>

<p style="margin-top: 1em">The separation angle between the
two views may be adjusted with the set stereo [-]
&lt;number&gt; command, where positive values result in
crossed eye viewing and negative val&acirc; <br>
ues in relaxed (wall-eyed) viewing. Currently, stereo
viewing is not supported in vector PostScript output
files.</p>

<p style="margin-top: 1em">Set Strands <br>
The RasMol strands parameter controls the number of parallel
strands that are displayed in the ribbon representations of
proteins. The permissible values for this parame&acirc; <br>
ter are 1, 2, 3, 4, 5 and 9. The default value is 5. The
number of strands is constant for all ribbons being
displayed. However, the ribbon width (the separation between
<br>
strands) may be controlled on a residue by residue basis
using the RasMol ribbons command.</p>

<p style="margin-top: 1em">Set Transparent <br>
The RasMol transparent parameter controls the writing of
transparent GIFs by the write gif &lt;filename&gt; command.
This may be controlled by the set transparent on and set
<br>
transparent off commands.</p>

<p style="margin-top: 1em">Set UnitCell <br>
The RasMol unitcell parameter controls the display of the
crystallographic unit cell on the current display. The
crystal cell is only enabled if the appropriate crystal <br>
symmetry information is contained in the PDB, CIF or mmCIF
data file. The RasMol command show symmetry display details
of the crystal&rsquo;s space group and unit cell axes. The
<br>
set unitcell command is similar to the commands set axes and
set boundbox that display orthogonal coordinate axes and the
bounding box, respectively.</p>

<p style="margin-top: 1em">Set VectPS <br>
The RasMol vectps parameter is use to control the way in
which the RasMol write command generates vector PostScript
output files. The command set vectps on enables the use <br>
of black outlines around spheres and cylinder bonds
producing &quot;cartoon-like&quot; high resolution output.
However, the current implementation of RasMol incorrectly
cartoons <br>
spheres that are intersected by more than one other sphere.
Hence &quot;ball and stick&quot; models are rendered
correctly but not large spacefilling spheres models. Cartoon
out&acirc; <br>
lines can be disabled, the default, by the command set
vectps off.</p>

<p style="margin-top: 1em">Set Write <br>
The RasMol write parameter controls the use of the save and
write commands within scripts, but it may only be executed
from the command line. By default, this value is <br>
false, prohibiting the generation of files in any scripts
executed at start-up (such as those launched from a WWW
browser). However, animators may start up RasMol
interac&acirc; <br>
tively: type set write on and then execute a script to
generate each frame using the source command.</p>

<p style="margin-top: 1em">ATOM EXPRESSIONS <br>
RasMol atom expressions uniquely identify an arbitrary group
of atoms within a molecule. Atom expressions are composed of
either primitive expressions, predefined sets,
compari&acirc; <br>
son operators, within expressions, or logical (boolean)
combinations of the above expression types.</p>

<p style="margin-top: 1em">The logical operators allow
complex queries to be constructed out of simpler ones using
the standard boolean connectives and, or and not. These may
be abbreviated by the symbols <br>
&quot;&amp;&quot;, &quot;|&quot; and &quot;!&quot;,
respectively. Parentheses (brackets) may be used to alter
the precedence of the operators. For convenience, a comma
may also be used for boolean disjunction.</p>

<p style="margin-top: 1em">The atom expression is evaluated
for each atom, hence protein and backbone selects protein
backbone atoms, not the protein and [nucleic] acid backbone
atoms!</p>

<p style="margin-top: 1em">Primitive Expressions <br>
RasMol primitive expressions are the fundamental building
blocks of atom expressions. There are two types of primitive
expression. The first type is used to identify a <br>
given residue number or range of residue numbers. A single
residue is identified by its number (position in the
sequence), and a range is specified by lower and upper <br>
bounds separated by a hyphen character. For example select
5,6,7,8 is also select 5-8. Note that this selects the given
residue numbers in all macromolecule chains.</p>

<p style="margin-top: 1em">The second type of primitive
expression specifies a sequence of fields that must match
for a given atom. The first part specifies a residue (or
group of residues) and an <br>
optional second part specifies the atoms within those
residues. The first part consists of a residue name,
optionally followed by a residue number and/or chain
identifier.</p>

<p style="margin-top: 1em">The second part consists of a
period character followed by an atom name. An atom name may
be up to four alphabetic or numeric characters. An optional
semicolon followed <br>
by an alternate conformation identifier may be appended. An
optional slash followed by a model number may also be
appended.</p>

<p style="margin-top: 1em">An asterisk may be used as a
wild card for a whole field and a question mark as a single
character wildcard.</p>

<p style="margin-top: 1em">Comparison Operators <br>
Parts of a molecule may also be distinguished using
equality, inequality and ordering operators on their
properties. The format of such comparison expression is a
property <br>
name, followed by a comparison operator and then an integer
value.</p>

<p style="margin-top: 1em">The atom properties that may be
used in RasMol are atomno for the atom serial number, elemno
for the atom&rsquo;s atomic number (element), resno for the
residue number, radius <br>
for the spacefill radius in RasMol units (or zero if not
represented as a sphere) and temperature for the PDB
isotropic temperature value.</p>

<p style="margin-top: 1em">The equality operator is denoted
either &quot;=&quot; or &quot;==&quot;. The inequality
operator as either &quot;&lt;&gt;&quot;, &quot;!=&quot; or
&quot;/=&quot;. The ordering operators are &quot;&lt;&quot;
for less than, &quot;&lt;=&quot; for less <br>
than or equal to, &quot;&gt;&quot; for greater than, and
&quot;&gt;&quot; for greater than or equal to.</p>

<p style="margin-top: 1em">Within Expressions <br>
A RasMol within expression allows atoms to be selected on
their proximity to another set of atoms. A within expression
takes two parameters separated by a comma and sur&acirc;
<br>
rounded by parentheses. The first argument is an integer
value called the &quot;cut-off&quot; distance of the within
expression and the second argument is any valid atom
expres&acirc; <br>
sion. The cut-off distance is expressed in either integer
RasMol units or Angstroms containing a decimal point. An
atom is selected if it is within the cut-off distance <br>
of any of the atoms defined by the second argument. This
allows complex expressions to be constructed containing
nested within expressions.</p>

<p style="margin-top: 1em">For example, the command select
within(3.2,backbone) selects any atom within a 3.2 Angstrom
radius of any atom in a protein or nucleic acid backbone.
Within expressions <br>
are particularly useful for selecting the atoms around an
active site.</p>

<p style="margin-top: 1em">Predefined Sets <br>
RasMol atom expressions may contain predefined sets. These
sets are single keywords that represent portions of a
molecule of interest. Predefined sets are often
abbrevia&acirc; <br>
tions of primitive atom expressions. In some cases the use
of predefined sets allows selection of areas of a molecule
that could not otherwise be distinguished. A list <br>
of the currently predefined sets is given below. In addition
to the sets listed here, RasMol also treats element names
(and their plurals) as predefined sets containing <br>
all atoms of that element type, i.e. the command select
oxygen is equivalent to the command select elemno=8.</p>

<p style="margin-top: 1em">Predefined Sets <br>
AT Set This set contains the atoms in the complementary
nucleotides adenosine and thymidine (A and T, respectively).
All nucleotides are classified as either the set at or the
<br>
set cg This set is equivalent to the RasMol atom expressions
a,t, and nucleic and not cg.</p>

<p style="margin-top: 1em">Acidic Set <br>
The set of acidic amino acids. These are the residue types
Asp and Glu. All amino acids are classified as either
acidic, basic or neutral. This set is equivalent to the <br>
RasMol atom expressions asp, glu and amino and not (basic or
neutral).</p>

<p style="margin-top: 1em">Acyclic Set <br>
The set of atoms in amino acids not containing a cycle or
ring. All amino acids are classified as either cyclic or
acyclic. This set is equivalent to the RasMol atom <br>
expression amino and not cyclic.</p>

<p style="margin-top: 1em">Aliphatic Set <br>
This set contains the aliphatic amino acids. These are the
amino acids Ala, Gly, Ile, Leu and Val. This set is
equivalent to the RasMol atom expression ala, gly, ile, <br>
leu, val.</p>

<p style="margin-top: 1em">Alpha Set <br>
The set of alpha carbons in the protein molecule. This set
is approximately equivalent to the RasMol atom expression
*.CA. This command should not be confused with the <br>
predefined set helix which contains the atoms in the amino
acids of the protein&rsquo;s alpha helices.</p>

<p style="margin-top: 1em">Amino Set <br>
This set contains all the atoms contained in amino acid
residues. This is useful for distinguishing the protein from
the nucleic acid and heterogeneous atoms in the cur&acirc;
<br>
rent molecule database.</p>

<p style="margin-top: 1em">Aromatic Set <br>
The set of atoms in amino acids containing aromatic rings.
These are the amino acids His, Phe, Trp and Tyr. Because
they contain aromatic rings all members of this set <br>
are member of the predefined set cyclic. This set is
equivalent to the RasMol atom expressions his, phe, trp, tyr
and cyclic and not pro.</p>

<p style="margin-top: 1em">Backbone Set <br>
This set contains the four atoms of each amino acid that
form the polypeptide N-C-C-O backbone of proteins, and the
atoms of the sugar phosphate backbone of nucleic acids. <br>
Use the RasMol predefined sets protein and nucleic to
distinguish between the two forms of backbone. Atoms in
nucleic acids and proteins are either backbone or sidechain.
<br>
This set is equivalent to the RasMol expression (protein or
nucleic) and not sidechain.</p>

<p style="margin-top: 1em">The predefined set mainchain is
synonymous with the set backbone.</p>

<p style="margin-top: 1em">Basic Set <br>
The set of basic amino acids. These are the residue types
Arg, His and Lys. All amino acids are classified as either
acidic, basic or neutral. This set is equivalent to <br>
the RasMol atom expressions arg, his, lys and amino and not
(acidic or neutral).</p>

<p style="margin-top: 1em">Bonded Set <br>
This set contain all the atoms in the current molecule
database that are bonded to at least one other atom.</p>

<p style="margin-top: 1em">Buried Set <br>
This set contains the atoms in those amino acids that tend
(prefer) to be buried inside protein, away from contact with
solvent molecules. This set refers to the amino <br>
acids preference and not the actual solvent accessibility
for the current protein. All amino acids are classified as
either surface or buried. This set is equivalent to <br>
the RasMol atom expression amino and not surface.</p>

<p style="margin-top: 1em">CG Set This set contains the
atoms in the complementary nucleotides cytidine and
guanosine (C and G, respectively). All nucleotides are
classified as either the set at or the set <br>
cg This set is equivalent to the RasMol atom expressions c,g
and nucleic and not at.</p>

<p style="margin-top: 1em">Charged Set <br>
This set contains the charged amino acids. These are the
amino acids that are either acidic or basic. Amino acids are
classified as being either charged or neutral. This <br>
set is equivalent to the RasMol atom expressions acidic or
basic and amino and not neutral.</p>

<p style="margin-top: 1em">Cyclic Set <br>
The set of atoms in amino acids containing a cycle or rings.
All amino acids are classified as either cyclic or acyclic.
This set consists of the amino acids His, Phe, <br>
Pro, Trp and Tyr. The members of the predefined set aromatic
are members of this set. The only cyclic but non-aromatic
amino acid is proline. This set is equivalent to <br>
the RasMol atom expressions his, phe, pro, trp, tyr and
aromatic or pro and amino and not acyclic.</p>

<p style="margin-top: 1em">Cystine Set <br>
This set contains the atoms of cysteine residues that form
part of a disulphide bridge, i.e. half cystines. RasMol
automatically determines disulphide bridges, if neither <br>
the predefined set cystine nor the RasMol ssbonds command
have been used since the molecule was loaded. The set of
free cysteines may be determined using the RasMol atom <br>
expression cys and not cystine.</p>

<p style="margin-top: 1em">Helix Set <br>
This set contains all atoms that form part of a protein
alpha helix as determined by either the PDB file author or
Kabsch and Sander&rsquo;s DSSP algorithm. By default, RasMol
<br>
uses the secondary structure determination given in the PDB
file if it exists. Otherwise, it uses the DSSP algorithm as
used by the RasMol structure command.</p>

<p style="margin-top: 1em">This predefined set should not
be confused with the predefined set alpha which contains the
alpha carbon atoms of a protein.</p>

<p style="margin-top: 1em">Hetero Set <br>
This set contains all the heterogeneous atoms in the
molecule. These are the atoms described by HETATM entries in
the PDB file. These typically contain water, cofactors <br>
and other solvents and ligands. All hetero atoms are
classified as either ligand or solvent atoms. These
heterogeneous solvent atoms are further classified as either
water <br>
or ions.</p>

<p style="margin-top: 1em">Hydrogen Set <br>
This predefined set contains all the hydrogen, deuterium and
tritium atoms of the current molecule. This predefined set
is equivalent to the RasMol atom expression <br>
elemno=1.</p>

<p style="margin-top: 1em">Hydrophobic Set <br>
This set contains all the hydrophobic amino acids. These are
the amino acids Ala, Leu, Val, Ile, Pro, Phe, Met and Trp.
All amino acids are classified as either <br>
hydrophobic or polar. This set is equivalent to the RasMol
atom expressions ala, leu, val, ile, pro, phe, met, trp and
amino and not polar.</p>

<p style="margin-top: 1em">Ions Set <br>
This set contains all the heterogeneous phosphate and
sulphate ions in the current molecule data file. A large
number of these ions are sometimes associated with protein
<br>
and nucleic acid structures determined by X-ray
crystallography. These atoms tend to clutter an image. All
hetero atoms are classified as either ligand or solvent
atoms. <br>
All solvent atoms are classified as either water or
ions.</p>

<p style="margin-top: 1em">Large Set <br>
All amino acids are classified as either small, medium or
large. This set is equivalent to the RasMol atom expression
amino and not (small or medium).</p>

<p style="margin-top: 1em">Ligand Set <br>
This set contains all the heterogeneous cofactor and ligand
moieties that are contained in the current molecule data
file. This set is defined to be all hetero atoms that <br>
are not solvent atoms. Hence this set is equivalent to the
RasMol atom expression hetero and not solvent.</p>

<p style="margin-top: 1em">Medium Set <br>
All amino acids are classified as either small, medium or
large. This set is equivalent to the RasMol atom expression
amino and not (large or small).</p>

<p style="margin-top: 1em">Neutral Set <br>
The set of neutral amino acids. All amino acids are
classified as either acidic, basic or neutral. This set is
equivalent to the RasMol atom expression amino and not <br>
(acidic or basic).</p>

<p style="margin-top: 1em">Nucleic Set <br>
The set of all atoms in nucleic acids, which consists of the
four nucleotide bases adenosine, cytidine, guanosine and
thymidine (A, C, G and T, respectively). All neu&acirc; <br>
cleotides are classified as either purine or pyrimidine.
This set is equivalent to the RasMol atom expressions
a,c,g,t and purine or pyrimidine. The symbols for RNA
nu&acirc; <br>
cleotides (U, +U, I, 1MA, 5MC, OMC, 1MG, 2MG, M2G, 7MG, OMG,
YG, H2U, 5MU, and PSU) are also recognized as members of
this set.</p>

<p style="margin-top: 1em">Polar Set <br>
This set contains the polar amino acids. All amino acids are
classified as either hydrophobic or polar. This set is
equivalent to the RasMol atom expression amino and <br>
not hydrophobic.</p>

<p style="margin-top: 1em">Protein Set <br>
The set of all atoms in proteins. This consists of the
RasMol predefined set amino and common post-translation
modifications.</p>

<p style="margin-top: 1em">Purine Set <br>
The set of purine nucleotides. These are the bases adenosine
and guanosine (A and G, respectively). All nucleotides are
either purines or pyrimidines. This set is <br>
equivalent to the RasMol atom expressions a,g and nucleic
and not pyrimidine.</p>

<p style="margin-top: 1em">Pyrimidine Set <br>
The set of pyrimidine nucleotides. These are the bases
cytidine and thymidine (C and T, respectively). All
nucleotides are either purines or pyrimidines. This set is
<br>
equivalent to the RasMol atom expressions c,t and nucleic
and not purine.</p>

<p style="margin-top: 1em">Selected Set <br>
This set contains the set of atoms in the currently selected
region. The currently selected region is defined by the
preceding select or restrict command and not the atom <br>
expression containing the selected keyword.</p>

<p style="margin-top: 1em">Sheet Set <br>
This set contains all atoms that form part of a protein beta
sheet as determined by either the PDB file author or Kabsch
and Sander&rsquo;s DSSP algorithm. By default, RasMol <br>
uses the secondary structure determination given in the PDB
file if it exists. Otherwise, it uses the DSSP algorithm as
used by the RasMol structure command.</p>

<p style="margin-top: 1em">Sidechain Set <br>
This set contains the functional sidechains of any amino
acids and the base of each nucleotide. These are the atoms
not part of the polypeptide N-C-C-O backbone of pro&acirc;
<br>
teins or the sugar phosphate backbone of nucleic acids. Use
the RasMol predefined sets protein and nucleic to
distinguish between the two forms of sidechain. Atoms in
<br>
nucleic acids and proteins are either backbone or sidechain.
This set is equivalent to the RasMol expression (protein or
nucleic) and not backbone.</p>

<p style="margin-top: 1em">Small Set <br>
All amino acids are classified as either small, medium or
large. This set is equivalent to the RasMol atom expression
amino and not (medium or large).</p>

<p style="margin-top: 1em">Solvent Set <br>
This set contains the solvent atoms in the molecule
coordinate file. These are the heterogeneous water
molecules, phosphate and sulphate ions. All hetero atoms are
clas&acirc; <br>
sified as either ligand or solvent atoms. All solvent atoms
are classified as either water or ions. This set is
equivalent to the RasMol atom expressions hetero and not
<br>
ligand and water or ions.</p>

<p style="margin-top: 1em">Surface Set <br>
This set contains the atoms in those amino acids that tend
(prefer) to be on the surface of proteins, in contact with
solvent molecules. This set refers to the amino acids <br>
preference and not the actual solvent accessibility for the
current protein. All amino acids are classified as either
surface or buried. This set is equivalent to the <br>
RasMol atom expression amino and not buried.</p>

<p style="margin-top: 1em">Turn Set <br>
This set contains all atoms that form part of a protein
turns as determined by either the PDB file author or Kabsch
and Sander&rsquo;s DSSP algorithm. By default, RasMol uses
<br>
the secondary structure determination given in the PDB file
if it exists. Otherwise, it uses the DSSP algorithm as used
by the RasMol structure command.</p>

<p style="margin-top: 1em">Water Set <br>
This set contains all the heterogeneous water molecules in
the current database. A large number of water molecules are
sometimes associated with protein and nucleic acid <br>
structures determined by X-ray crystallography. These atoms
tend to clutter an image. All hetero atoms are classified as
either ligand or solvent atoms. The solvent atoms <br>
are further classified as either water or ions.</p>

<p style="margin-top: 1em">Set Summary <br>
The table below summarises RasMol&rsquo;s classification of
the common amino acids.</p>

<p style="margin-top: 1em">COLOUR SCHEMES <br>
The RasMol colour command allows different objects (such as
atoms, bonds and ribbon segments) to be given a specified
colour. Typically this colour is either a RasMol predefined
<br>
colour name or an RGB triple. Additionally RasMol also
supports alt, amino, chain, charge, cpk, group, model,
shapely, structure, temperature or user colour schemes for
atoms, <br>
and hbond type colour scheme for hydrogen bonds and
electrostatic potential colour scheme for dot surfaces. The
24 currently predefined colour names are Black, Blue,
BlueTint, <br>
Brown, Cyan, Gold, Grey, Green, GreenBlue, GreenTint,
HotPink, Magenta, Orange, Pink, PinkTint, Purple, Red,
RedOrange, SeaGreen, SkyBlue, Violet, White, Yellow and
YellowTint</p>

<p style="margin-top: 1em">If you frequently wish to use a
colour not predefined, you can write a one-line script. For
example, if you make the file grey.col containing the line,
colour [180,180,180] <br>
#grey, then the command script grey.col colours the
currently selected atom set grey.</p>

<p style="margin-top: 1em">Alt Colours <br>
The RasMol alt (Alternate Conformer) colour scheme codes the
base structure with one colour and applies a limited number
of colours to each alternate conformer. In a Ras&acirc; <br>
Mol built for 8-bit colour systems, 4 colours are allowed
for alternate conformers. Otherwise, 8 colours are
available.</p>

<p style="margin-top: 1em">Amino Colours <br>
The RasMol amino colour scheme colours amino acids according
to traditional amino acid properties. The purpose of
colouring is to identify amino acids in an unusual or <br>
surprising environment. The outer parts of a protein that
are polar are visible (bright) colours and non-polar
residues darker. Most colours are hallowed by tradition.
<br>
This colour scheme is similar to the shapely scheme.</p>

<p style="margin-top: 1em">Chain Colours <br>
The RasMol chain colour scheme assigns each macromolecular
chain a unique colour. This colour scheme is particularly
useful for distinguishing the parts of multimeric <br>
structure or the individual &rsquo;strands&rsquo; of a DNA
chain. Chain can be selected from the RasMol Colours
menu.</p>

<p style="margin-top: 1em">Charge Colours <br>
The RasMol charge colour scheme colour codes each atom
according to the charge value stored in the input file (or
beta factor field of PDB files). High values are coloured
<br>
in blue (positive) and lower values coloured in red
(negative). Rather than use a fixed scale this scheme
determines the maximum and minimum values of the
charge/tempera&acirc; <br>
ture field and interpolates from red to blue appropriately.
Hence, green cannot be assumed to be &rsquo;no net
charge&rsquo; charge.</p>

<p style="margin-top: 1em">The difference between the
charge and temperature colour schemes is that increasing
temperature values proceed from blue to red, whereas
increasing charge values go from <br>
red to blue.</p>

<p style="margin-top: 1em">If the charge/temperature field
stores reasonable values it is possible to use the RasMol
colour dots potential command to colour code a dot surface
(generated by the dots <br>
command) by electrostatic potential.</p>

<p style="margin-top: 1em">CPK Colours <br>
The RasMol cpk colour scheme is based upon the colours of
the popular plastic spacefilling models which were developed
by Corey, Pauling and later improved by Kultun. This <br>
colour scheme colours &rsquo;atom&rsquo; objects by the atom
(element) type. This is the scheme conventionally used by
chemists. The assignment of the most commonly used element
<br>
types to colours is given below.</p>

<p style="margin-top: 1em">Group Colours <br>
The RasMol group colour scheme colour codes residues by
their position in a macromolecular chain. Each chain is
drawn as a smooth spectrum from blue through green, yellow
<br>
and orange to red. Hence the N terminus of proteins and
5&rsquo; terminus of nucleic acids are coloured red and the
C terminus of proteins and 3&rsquo; terminus of nucleic
acids are <br>
drawn in blue. If a chain has a large number of
heterogeneous molecules associated with it, the
macromolecule may not be drawn in the full
&rsquo;range&rsquo; of the spectrum. Group <br>
can be selected from the RasMol Colours menu.</p>

<p style="margin-top: 1em">If a chain has a large number of
heterogeneous molecules associated with it, the
macromolecule may not be drawn in the full range of the
spectrum. When RasMol performs <br>
group colouring it decides the range of colours it uses from
the residue numbering given in the PDB file. Hence the
lowest residue number is displayed in blue and the <br>
highest residue number is displayed as red. Unfortunately,
if a PDB file contains a large number of heteroatoms, such
as water molecules, that occupy the high residue num&acirc;
<br>
bers, the protein is displayed in the blue-green end of the
spectrum and the waters in the yellow-red end of the
spectrum. This is aggravated by there typically being many
<br>
more water molecules than amino acid residues. The solution
to this problem is to use the command set hetero off before
applying the group colour scheme. This can also be <br>
achieved by toggling Hetero Atoms on the Options menu before
selecting Group on the Colour menu. This command instructs
RasMol to only use non-hetero residues in the group <br>
colour scaling.</p>

<p style="margin-top: 1em">NMR Model Colours <br>
The RasMol model colour scheme codes each NMR model with a
distinct colour. The NMR model number is taken as a numeric
value. High values are coloured in blue and lower <br>
values coloured in red. Rather than use a fixed scale this
scheme determines the maximum value of the NMR model number
and interpolates from red to blue appropriately.</p>

<p style="margin-top: 1em">Shapely Colours <br>
The RasMol shapely colour scheme colour codes residues by
amino acid property. This scheme is based upon Bob
Fletterick&rsquo;s &quot;Shapely Models&quot;. Each amino
acid and nucleic <br>
acid residue is given a unique colour. The shapely colour
scheme is used by David Bacon&rsquo;s Raster3D program. This
colour scheme is similar to the amino colour scheme.</p>

<p style="margin-top: 1em">Structure Colours <br>
The RasMol structure colour scheme colours the molecule by
protein secondary structure. Alpha helices are coloured
magenta, [240,0,128], beta sheets are coloured yellow, <br>
[255,255,0], turns are coloured pale blue, [96,128,255] and
all other residues are coloured white. The secondary
structure is either read from the PDB file (HELIX, SHEET
<br>
and TURN records), if available, or determined using Kabsch
and Sander&rsquo;s DSSP algorithm. The RasMol structure
command may be used to force DSSP&rsquo;s structure
assignment to <br>
be used.</p>

<p style="margin-top: 1em">Temperature Colours <br>
The RasMol temperature colour scheme colour codes each atom
according to the anisotropic temperature (beta) value stored
in the PDB file. Typically this gives a measure of <br>
the mobility/uncertainty of a given atom&rsquo;s position.
High values are coloured in warmer (red) colours and lower
values in colder (blue) colours. This feature is often used
<br>
to associate a &quot;scale&quot; value [such as amino acid
variability in viral mutants] with each atom in a PDB file,
and colour the molecule appropriately.</p>

<p style="margin-top: 1em">The difference between the
temperature and charge colour schemes is that increasing
temperature values proceed from blue to red, whereas
increasing charge values go from <br>
red to blue.</p>

<p style="margin-top: 1em">User Colours <br>
The RasMol user colour scheme allows RasMol to use the
colour scheme stored in the PDB file. The colours for each
atom are stored in COLO records placed in the PDB data <br>
file. This convention was introduced by David Bacon&rsquo;s
Raster3D program.</p>

<p style="margin-top: 1em">HBond Type Colours <br>
The RasMol type colour scheme applies only to hydrogen
bonds, hence is used in the command colour hbonds type. This
scheme colour codes each hydrogen bond according to <br>
the distance along a protein chain between hydrogen bond
donor and acceptor. This schematic representation was
introduced by Belhadj-Mostefa and Milner-White. This
repre&acirc; <br>
sentation gives a good insight into protein secondary
structure (hbonds forming alpha helices appear red, those
forming sheets appear yellow and those forming turns appear
<br>
magenta).</p>

<p style="margin-top: 1em">Potential Colours <br>
The RasMol potential colour scheme applies only to dot
surfaces, hence is used in the command colour dots
potential. This scheme colours each currently displayed dot
by <br>
the electrostatic potential at that point in space. This
potential is calculated using Coulomb&rsquo;s law taking the
temperature/charge field of the input file to be the charge
<br>
assocated with that atom. This is the same interpretation
used by the colour charge command. Like the charge colour
scheme low values are blue/white and high values are <br>
red.</p>

<p style="margin-top: 1em">Amino Acid Codes <br>
The following table lists the names, single letter and three
letter codes of each of the amino acids.</p>

<p style="margin-top: 1em">Booleans <br>
A boolean parameter is a truth value. Valid boolean values
are &rsquo;true&rsquo; and used by RasMol to either enable
or disable a representation or option.</p>

<p style="margin-top: 1em">FILE FORMATS <br>
Protein Data Bank Files</p>

<p style="margin-top: 1em">If you do not have the PDB
documentation, you may find the following summary of the PDB
file format useful. The Protein Data Bank is a
computer-based archival database for macro&acirc; <br>
molecular structures. The database was established in 1971
by Brookhaven National Laboratory, Upton, New York, as a
public domain repository for resolved crystallographic
struc&acirc; <br>
tures. The Bank uses a uniform format to store atomic
coordinates and partial bond connectivities as derived from
crystallographic studies. In 1999 the Protein Data Bank
moved <br>
to the Research Collaboratory for Structural Biology.</p>

<p style="margin-top: 1em">PDB file entries consist of
records of 80 characters each. Using the punched card
analogy, columns 1 to 6 contain a record-type identifier,
the columns 7 to 70 contain data. In <br>
older entries, columns 71 to 80 are normally blank, but may
contain sequence information added by library management
programs. In new entries conforming to the 1996 PDB format,
<br>
there is other information in those columns. The first four
characters of the record identifier are sufficient to
identify the type of record uniquely, and the syntax of each
<br>
record is independent of the order of records within any
entry for a particular macromolecule.</p>

<p style="margin-top: 1em">The only record types that are
of major interest to the RasMol program are the ATOM and
HETATM records which describe the position of each atom.
ATOM/HETATM records contain stan&acirc; <br>
dard atom names and residue abbreviations, along with
sequence identifiers, coordinates in Angstrom units,
occupancies and thermal motion factors. The exact details
are given <br>
below as a FORTRAN format statement. The &quot;fmt&quot;
column indicates use of the field in all PDB formats, in the
1992 and earlier formats or in the 1996 and later
formats.</p>

<p style="margin-top: 1em">Residues occur in order starting
from the N-terminal residue for proteins and
5&rsquo;-terminus for nucleic acids. If the residue sequence
is known, certain atom serial numbers may be <br>
omitted to allow for future insertion of any missing atoms.
Within each residue, atoms are ordered in a standard manner,
starting with the backbone (N-C-C-O for proteins) and <br>
proceeding in increasing remoteness from the alpha carbon,
along the side chain.</p>

<p style="margin-top: 1em">HETATM records are used to
define post-translational modifications and cofactors
associated with the main molecule. TER records are
interpreted as breaks in the main molecule&rsquo;s <br>
backbone.</p>

<p style="margin-top: 1em">If present, RasMol also inspects
HEADER, COMPND, HELIX, SHEET, TURN, CONECT, CRYST1, SCALE,
MODEL, ENDMDL, EXPDTA and END records. Information such as
the name, database code, <br>
revision date and classification of the molecule are
extracted from HEADER and COMPND records, initial secondary
structure assignments are taken from HELIX, SHEET and TURN
<br>
records, and the end of the file may be indicated by an END
record.</p>

<p style="margin-top: 1em">RasMol Interpretation of PDB
fields <br>
Atoms located at 9999.000, 9999.000, 9999.000 are assumed to
be Insight pseudo atoms and are ignored by RasMol. Atom
names beginning &rsquo; Q&rsquo; are also assumed to be
pseudo <br>
atoms or position markers.</p>

<p style="margin-top: 1em">When a data file contains an NMR
structure, multiple conformations may be placed in a single
PDB file delimited by pairs of MODEL and ENDMDL records.
RasMol displays all <br>
the NMR models contained in the file.</p>

<p style="margin-top: 1em">Residue names &quot;CSH&quot;,
&quot;CYH&quot; and &quot;CSM&quot; are considered
pseudonyms for cysteine &quot;CYS&quot;. Residue names
&quot;WAT&quot;, &quot;H20&quot;, &quot;SOL&quot; and
&quot;TIP&quot; are considered pseudonyms for water
&quot;HOH&quot;. <br>
The residue name &quot;D20&quot; is consider heavy water
&quot;DOD&quot;. The residue name &quot;SUL&quot; is
considered a sulphate ion &quot;SO4&quot;. The residue name
&quot;CPR&quot; is considered to be cis-proline and <br>
is translated as &quot;PRO&quot;. The residue name
&quot;TRY&quot; is considered a pseudonym for tryptophan
&quot;TRP&quot;.</p>

<p style="margin-top: 1em">RasMol uses the HETATM fields to
define the sets hetero, water, solvent and ligand. Any group
with the name &quot;HOH&quot;, &quot;DOD&quot;,
&quot;SO4&quot; or &quot;PO4&quot; (or aliased to one of
these names <br>
by the preceding rules) is considered a solvent and is
considered to be defined by a HETATM field.</p>

<p style="margin-top: 1em">RasMol only respects CONECT
connectivity records in PDB files containing fewer than 256
atoms. This is explained in more detail in the section on
determining molecule con&acirc; <br>
nectivity. CONECT records that define a bond more than once
are interpreted as specifying the bond order of that bond,
i.e. a bond specified twice is a double bond and a <br>
bond specified three (or more) times is a triple bond. This
is not a standard PDB feature.</p>

<p style="margin-top: 1em">PDB Colour Scheme Specification
<br>
RasMol also accepts the supplementary COLO record type in
the PDB files. This record format was introduced by David
Bacon&rsquo;s Raster3D program for specifying the colour
<br>
scheme to be used when rendering the molecule. This
extension is not currently supported by the PDB. The COLO
record has the same basic record type as the ATOM and HETATM
<br>
records described above.</p>

<p style="margin-top: 1em">Colours are assigned to atoms
using a matching process. The Mask field is used in the
matching process as follows. First RasMol reads in and
remembers all the ATOM, HETATM <br>
and COLO records in input order. When the user-defined
(&rsquo;User&rsquo;) colour scheme is selected, RasMol goes
through each remembered ATOM/HETATM record in turn, and
searches for <br>
a COLO record that matches in all of columns 7 through 30.
The first such COLO record to be found determines the colour
and radius of the atom.</p>

<p style="margin-top: 1em">Note that the Red, Green and
Blue components are in the same positions as the X, Y, and Z
components of an ATOM or HETA record, and the van der Waals
radius goes in the <br>
place of the Occupancy. The Red, Green and Blue components
must all be in the range 0 to 1.</p>

<p style="margin-top: 1em">In order that one COLO record
can provide colour and radius specifications for more than
one atom (e.g. based on residue, atom type, or any other
criterion for which <br>
labels can be given somewhere in columns 7 through 30), a
&rsquo;don&rsquo;t-care&rsquo; character, the hash mark
&quot;#&quot; (number or sharp sign) is used. This
character, when found in a COLO <br>
record, matches any character in the corresponding column in
a ATOM/HETATM record. All other characters must match
identically to count as a match. As an extension to the <br>
specification, any atom that fails to match a COLO record is
displayed in white.</p>

<p style="margin-top: 1em">Multiple NMR Models <br>
RasMol loads all of the NMR models from a PDB file no matter
which command is used: load pdb &lt;filename&gt; or load
nmrpdb &lt;filename&gt;</p>

<p style="margin-top: 1em">Once multiple NMR conformations
have been loaded they may be manipulated with the atom
expression extensions described in Primitive Expressions. In
particular, the com&acirc; <br>
mand restrict */1 will restrict the display to the first
model only.</p>

<p style="margin-top: 1em">CIF and mmCIF Format Files <br>
CIF is the IUCr standard for presentation of small molecules
and mmCIF is intended as the replacement for the fixed-field
PDB format for presentation of macromolecular <br>
structures. RasMol can accept data sets in either
format.</p>

<p style="margin-top: 1em">There are many useful sites on
the World Wide Web where information tools and software
related to CIF, mmCIF and the PDB can be found. The
following are good starting <br>
points for exploration:</p>

<p style="margin-top: 1em">The International Union of
Crystallography (IUCr) provides access to software,
dictionaries, policy statements and documentation relating
to CIF and mmCIF at: IUCr, <br>
Chester, England (www.iucr.org/iucr-top/cif/) with many
mirror sites.</p>

<p style="margin-top: 1em">The Nucleic Acid Database
Project provides access to its entries, software and
documentation, with an mmCIF page giving access to the
dictionary and mmCIF software tools <br>
at Rutgers University, New Jersey, USA
(http://ndbserver.rutgers.edu/NDB/mmcif) with many mirror
sites.</p>

<p style="margin-top: 1em">This version of RasMol restricts
CIF or mmCIF tag values to essentially the same conventions
as are used for the fixed-field PDB format. Thus chain
identifiers and alter&acirc; <br>
nate conformation identifiers are limited to a single
character, atom names are limited to 4 characters, etc.
RasMol interprets the following CIF and mmCIF tags: A search
<br>
is made through multiple data blocks for the desired tags,
so a single dataset may be composed from multiple data
blocks, but multiple data sets may not be stacked in the
<br>
same file.</p>

<p style="margin-top: 1em">MACHINE-SPECIFIC SUPPORT <br>
In the following sections, support for Monochrome X-Windows,
Tcl/Tk IPC, UNIX sockets based IPC, Compiling RasWin with
Borland and MetroWerks are described.</p>

<p style="margin-top: 1em">Monochrome X-Windows Support
<br>
RasMol supports the many monochrome UNIX workstations
typically found in academia, such as low-end SUN
workstations and NCD X-terminals. The X11 version of RasMol
(when <br>
compiled in 8 bit mode) now detects black and white
X-Windows displays and enables dithering automatically. The
use of run-time error diffusion dithering means that all
<br>
display modes of RasMol are available when in monochrome
mode. For best results, users should experiment with the set
ambient command to ensure the maximum contrast in <br>
resulting images.</p>

<p style="margin-top: 1em">Tcl/Tk IPC support <br>
Version 4 of Tk graphics library changed the protocol used
to communicate between Tk applications. RasMol version 2.6
was modified such that it could communicate with both <br>
this new protocol and the previous version 3 protocol
supported by RasMol v2.5. Although Tcl/Tk 3.x applications
may only communicate with other 3.x applications and <br>
Tcl/Tk 4.x applications with other 4.x applications, these
changes allow RasMol to communicate between processes with
both protocols (potentially concurrently).</p>

<p style="margin-top: 1em">UNIX sockets based IPC <br>
The UNIX implementation of RasMol supports BSD-style socket
communication. An identical socket mechanism is also being
developed for VMS, Apple Macintosh and Microsoft <br>
Windows systems. This should allow RasMol to interactively
display results of a computation on a remote host. The
current protocol acts as a TCP/IP server on port 21069 <br>
that executes command lines until either the command exit or
the command quit is typed. The command exit from the RasMol
server, the command quit both disconnects the cur&acirc;
<br>
rent session and terminates RasMol. This functionality may
be tested using the UNIX command telnet &lt;hostname&gt;
21069.</p>

<p style="margin-top: 1em">Compiling RasWin with Borland
and MetroWerks <br>
A number of changes were made to the source code in the
transition from version 2.5 to 2.6 to allow the Microsoft
Windows version of RasMol to compile using the Borland <br>
C/C++ compiler. These fixes include name changes for the
standard library and special code to avoid a bug in
_fmemset. Additional changes were made in the transition
from <br>
2.6 to 2.7 to allow compilation with the MetroWerks
compilers.</p>

<p style="margin-top: 1em">BIBLIOGRAPHY <br>
Molecular Graphics</p>

<p style="margin-top: 1em">[1] Nelson Max, &quot;Computer
Representation of Molecular Surfaces&quot;, IEEE Computer
Graphics and Applications, pp.21-29, August 1983.</p>

<p style="margin-top: 1em">[2] Arthur M. Lesk,
&quot;Protein Architecture: A Practical Approach&quot;, IRL
Press Publishers, 1991.</p>

<p style="margin-top: 1em">Molecular Graphics Programs</p>

<p style="margin-top: 1em">[3] Per J. Kraulis,
&quot;MOLSCRIPT: A Program to Produce both Detailed and
Schematic Plots of Protein Structures&quot;, Journal of
Applied Crystallography, Vol.24, pp.946-950, 1991.</p>

<p style="margin-top: 1em">[4] David Bacon and Wayne F.
Anderson, &quot;A Fast Algorithm for Rendering Space-Filling
Molecule Pictures&quot;, Journal of Molecular Graphics,
Vol.6, No.4, pp.219-220, December 1988.</p>

<p style="margin-top: 1em">[5] David C. Richardson and Jane
S. Richardson, &quot;The Kinemage: A tool for Scientific
Communication&quot;, Protein Science, Vol.1, No.1,pp.3-9,
January 1992.</p>

<p style="margin-top: 1em">[6] Mike Carson, &quot;RIBBONS
2.0&quot;, Journal of Applied Crystallography, Vol.24,
pp.958-961, 1991.</p>

<p style="margin-top: 1em">[7] Conrad C. Huang, Eric F.
Pettersen, Teri E. Klein, Thomas E. Ferrin and Robert
Langridge, &quot;Conic: A Fast Renderer for Space-Filling
Molecules with Shadows&quot;, Journal of <br>
Molecular Graphics, Vol.9, No.4, pp.230-236, December
1991.</p>

<p style="margin-top: 1em">Molecular Biology Algorithms</p>

<p style="margin-top: 1em">[8] Wolfgang Kabsch and
Christian Sander, &quot;Dictionary of Protein Secondary
Structure: Pattern Recognition of Hydrogen-Bonded and
Geometrical Features&quot;, Biopolymers, Vol.22, <br>
pp.2577-2637, 1983.</p>

<p style="margin-top: 1em">[9] Michael L. Connolly,
&quot;Solvent-Accessible Surfaces of Proteins and Nucleic
Acids&quot;, Science, Vol.221, No.4612, pp.709-713, August
1983.</p>

<p style="margin-top: 1em">[10] Khaled Belhadj-Mostefa, Ron
Poet and E. James Milner-White, &quot;Displaying Inter-Main
Chain Hydrogen Bond Patterns in Proteins&quot;, Journal of
Molecular Graphics, Vol.9, No.3, <br>
pp.194-197, September 1991.</p>

<p style="margin-top: 1em">[11] Mike Carson, &quot;Ribbon
Models of Macromolecules&quot;, Journal of Molecular
Graphics, Vol.5, No.2, pp.103-106, June 1987.</p>

<p style="margin-top: 1em">[12] Mike Carson and Charles E.
Bugg, &quot;Algorithm for Ribbon Models of Proteins&quot;,
Journal of Molecular Graphics, Vol.4, No.2, pp.121-122, June
1986.</p>

<p style="margin-top: 1em">[13] H. Iijima, J. B. Dunbar Jr.
and G. Marshall, &quot;Calibration of Effective van der
Waals Atomic Contact Radii for Proteins and Peptides&quot;,
Proteins: Structure, Functions and <br>
Genetics, Vol.2, pp.330-339,1987.</p>

<p style="margin-top: 1em">Graphics Algorithms</p>

<p style="margin-top: 1em">[14] J. Foley, A. van Dam, S.
Feiner and J. Hughes, &quot;Computer Graphics: Principles
and Practice&quot;, 2nd Edition, Addison Wesley Publishers,
1990.</p>

<p style="margin-top: 1em">[15] J. Cleary and G. Wyvill,
&quot;Analysis of an Algorithm for Fast Ray Tracing using
Uniform Space Subdivision&quot;, The Visual Computer, Vol.4,
pp.65-83, 1988.</p>

<p style="margin-top: 1em">[16] Thomas
Porter,&quot;Spherical Shading&quot;, Computer Graphics
Vol.12, ACM SIGGRAPH, pp.282-285, 1978.</p>

<p style="margin-top: 1em">[17] Jean-Michel Cense,
&quot;Exact Visibility Calculation for Space-Filling
Molecular Models&quot;, Journal of Molecular Graphics,
Vol.9, No.3, pp.191-193, September 1991.</p>

<p style="margin-top: 1em">[18] Chris Schafmeister,
&quot;Fast Algorithm for Generating CPK Images on Graphics
Workstations&quot;, Journal of Molecular Graphics, Vol.8,
No.4, pp.201-206, December 1990.</p>

<p style="margin-top: 1em">[19] Bruce A. Johnson,
&quot;MSURF: A Rapid and General Program for the
Representation of Molecular Surfaces&quot;, Journal of
Molecular Graphics, Vol.5, No.3, pp.167-169, September
1987.</p>

<p style="margin-top: 1em">File Formats</p>

<p style="margin-top: 1em">[20] Frances C. Bernstein et
al., &quot;The Protein Data Bank: A Computer-Based Archival
File for Macromolecular Structures&quot;, Journal of
Molecular Biology, Vol.112, pp.535-542, 1977.</p>

<p style="margin-top: 1em">[21] Arthur Dalby, James G.
Nourse, W. Douglas Hounshell, Ann K. I. Gushurst, David L.
Grier, Burton A. Leland and John Laufer, &quot;Description
of Several Chemical File Formats <br>
Used by Computer Programs Developed at Molecular Design
Limited&quot;, Journal of Chemical Information and Computer
Sciences, Vol.32, No.3, pp.244-255, 1992.</p>

<p style="margin-top: 1em">[22] Adobe Systems Inc.,
&quot;PostScript Language Reference Manual&quot;,
Addison-Wesley Publishers, Reading, Mass., 1985.</p>

<p style="margin-top: 1em">[23] Philip E. Bourne et al.,
&quot;The Macromolecular Crystallographic Information File
(mmCIF)&quot;, Meth. Enzymol. (1997) 277, 571-590.</p>

<p style="margin-top: 1em">[24] Sydney R. Hall, &quot;The
STAR File: a New Format for Electronic Data Transfer and
Archiving&quot;, Journal of Chemical Information and
Computer Sciences, Vol. 31, 326-333, 1991.</p>

<p style="margin-top: 1em">SEE ALSO <br>
The RasMol User Manual!</p>

<p style="margin-top: 1em">AUTHOR <br>
1992-1998 by Roger Sayle (rasmol@ggr.co.uk)</p>

<p style="margin-top: 1em">July 2009 RASMOL(1)</p>
<hr>
</body>
</html>
