<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>RASMOL(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">RASMOL(1)</td>
    <td class="head-vol">General Commands Manual</td>
    <td class="head-rtitle">RASMOL(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
rasmol - Molecular Graphics Visualisation Tool v2.7.5
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>rasmol</b> [<b>-nodiplay</b>] [[<b>-</b><i>format </i>] <i>filename</i>]
  [<b>-script </b><i>scriptfile</i>]
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="FORMATS"><a class="selflink" href="#FORMATS">FORMATS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-pdb</b></dt>
  <dd class="It-tag">Protein Data Bank</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mdl</b></dt>
  <dd class="It-tag">MDL's MOL File Format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mol2</b></dt>
  <dd class="It-tag">Tripos' Sybyl MOL2 Format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-xyz</b></dt>
  <dd class="It-tag">MSC's XYZ (XMol) Format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-mopac</b></dt>
  <dd class="It-tag">MOPAC Input or Output File Format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-alchemy</b></dt>
  <dd class="It-tag">Alchemy File Format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-charmm</b></dt>
  <dd class="It-tag">CHARMm File Format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-cif</b></dt>
  <dd class="It-tag">IUCr CIF or CIF File Format</dd>
</dl>
<h1 class="Sh" title="Sh" id="NOTICES"><a class="selflink" href="#NOTICES">NOTICES</a></h1>
This software has been created from several sources. Much of the code is from
  RasMol 2.6, as created by Roger Sayle. The torsion angle code, new POVRAY3
  code and other features are derived from the RasMol2.6x1 revisions by Arne
  Mueller. The Ramachandran printer plot code was derived from fisipl created by
  Frances C. Bernstein. See the Protein Data Bank program tape.
<div style="height: 1.00em;">&#x00A0;</div>
The code to display multiple molecules and to allow bond rotation is derived in
  large part from the UCB mods by Gary Grossman and Marco Molinaro, included
  with permission of Eileen Lewis of the ModularCHEM Consortium.
<div style="height: 1.00em;">&#x00A0;</div>
The CIF modifications make use of a library based in part on CBFlib by Paul J.
  Ellis and Herbert J. Bernstein. Parts of CBFlib is loosely based on the
  CIFPARSE software package from the NDB at Rutgers university. Please type the
  RasMol commands <b>help copying,</b> <b>help general,</b> <b>help IUCR,</b>
  <b>help CBFlib,</b>
<br/>
 and <b>help CIFPARSE</b> for applicable notices. Please type <b>help
  copyright</b> for copyright notices. If you use RasMol V2.6 or an earlier
  version, type the RasMol command <b>help oldnotice.</b>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COPYING"><a class="selflink" href="#COPYING">COPYING</a></h1>
This version is based directly on RasMol version 2.7.4.2, on RasMol verion
  2.7.4.2, on RasMol version 2.7.4, on RasMol version 2.7.3.1, on RasMol version
  2.7.3, on RasMol version 2.7.2.1.1, Rasmol version 2.7.2, RasMol version
  2.7.1.1 and RasTop version 1.3 and indirectly on the RasMol 2.5-ucb and
  2.6-ucb versions and version 2.6_CIF.2, RasMol 2.6x1 and RasMol_2.6.4.
<div style="height: 1.00em;">&#x00A0;</div>
RasMol 2.7.5 may be distributed under the terms of the GNU General Public
  License (the GPL), see
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 http://www.gnu.org/licenses/gpl.txt
<div style="height: 1.00em;">&#x00A0;</div>
or the file GPL or type the command <b>help GPL</b>
<div style="height: 1.00em;">&#x00A0;</div>
or RasMol 2.7.5 may be distributed under the RASMOL license. See the file NOTICE
  or type the command <b>help RASLIC</b>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>GPL</b></dt>
  <dd class="It-tag">
    <br/>
     GNU GENERAL PUBLIC LICENSE
    <br/>
     Version 2, June 1991
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     Copyright (C) 1989, 1991 Free Software Foundation, Inc.
    <br/>
     59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
    <br/>
     Everyone is permitted to copy and distribute verbatim copies
    <br/>
     of this license document, but changing it is not allowed.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     Preamble
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     The licenses for most software are designed to take away your freedom to
      share and change it. By contrast, the GNU General Public License is
      intended to guarantee your freedom to share and change free software--to
      make sure the software is free for all its users. This General Public
      License applies to most of the Free Software Foundation's software and to
      any other program whose authors commit to using it. (Some other Free
      Software Foundation software is covered by the GNU Library General Public
      License instead.) You can apply it to your programs, too.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     When we speak of free software, we are referring to freedom, not price. Our
      General Public Licenses are designed to make sure that you have the
      freedom to distribute copies of free software (and charge for this service
      if you wish), that you receive source code or can get it if you want it,
      that you can change the software or use pieces of it in new free programs;
      and that you know you can do these things.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     To protect your rights, we need to make restrictions that forbid anyone to
      deny you these rights or to ask you to surrender the rights. These
      restrictions translate to certain responsibilities for you if you
      distribute copies of the software, or if you modify it.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     For example, if you distribute copies of such a program, whether gratis or
      for a fee, you must give the recipients all the rights that you have. You
      must make sure that they, too, receive or can get the source code. And you
      must show them these terms so they know their rights.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     We protect your rights with two steps: (1) copyright the software, and (2)
      offer you this license which gives you legal permission to copy,
      distribute and/or modify the software.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     Also, for each author's protection and ours, we want to make certain that
      everyone understands that there is no warranty for this free software. If
      the software is modified by someone else and passed on, we want its
      recipients to know that what they have is not the original, so that any
      problems introduced by others will not reflect on the original authors'
      reputations.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     Finally, any free program is threatened constantly by software patents. We
      wish to avoid the danger that redistributors of a free program will
      individually obtain patent licenses, in effect making the program
      proprietary. To prevent this, we have made it clear that any patent must
      be licensed for everyone's free use or not licensed at all.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     The precise terms and conditions for copying, distribution and modification
      follow.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     GNU GENERAL PUBLIC LICENSE
    <br/>
     TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     0. This License applies to any program or other work which contains a
      notice placed by the copyright holder saying it may be distributed under
      the terms of this General Public License. The &quot;Program&quot;, below,
      refers to any such program or work, and a &quot;work based on the
      Program&quot; means either the Program or any derivative work under
      copyright law: that is to say, a work containing the Program or a portion
      of it, either verbatim or with modifications and/or translated into
      another language. (Hereinafter, translation is included without limitation
      in the term &quot;modification&quot;.) Each licensee is addressed as
      &quot;you&quot;.
    <div style="height: 1.00em;">&#x00A0;</div>
    Activities other than copying, distribution and modification are not covered
      by this License; they are outside its scope. The act of running the
      Program is not restricted, and the output from the Program is covered only
      if its contents constitute a work based on the Program (independent of
      having been made by running the Program). Whether that is true depends on
      what the Program does.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     1. You may copy and distribute verbatim copies of the Program's source code
      as you receive it, in any medium, provided that you conspicuously and
      appropriately publish on each copy an appropriate copyright notice and
      disclaimer of warranty; keep intact all the notices that refer to this
      License and to the absence of any warranty; and give any other recipients
      of the Program a copy of this License along with the Program.
    <div style="height: 1.00em;">&#x00A0;</div>
    You may charge a fee for the physical act of transferring a copy, and you
      may at your option offer warranty protection in exchange for a fee.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     2. You may modify your copy or copies of the Program or any portion of it,
      thus forming a work based on the Program, and copy and distribute such
      modifications or work under the terms of Section 1 above, provided that
      you also meet all of these conditions:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     a) You must cause the modified files to carry prominent notices
    <br/>
     stating that you changed the files and the date of any change.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     b) You must cause any work that you distribute or publish, that in
    <br/>
     whole or in part contains or is derived from the Program or any
    <br/>
     part thereof, to be licensed as a whole at no charge to all third
    <br/>
     parties under the terms of this License.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     c) If the modified program normally reads commands interactively
    <br/>
     when run, you must cause it, when started running for such
    <br/>
     interactive use in the most ordinary way, to print or display an
    <br/>
     announcement including an appropriate copyright notice and a
    <br/>
     notice that there is no warranty (or else, saying that you provide
    <br/>
     a warranty) and that users may redistribute the program under
    <br/>
     these conditions, and telling the user how to view a copy of this
    <br/>
     License. (Exception: if the Program itself is interactive but
    <br/>
     does not normally print such an announcement, your work based on
    <br/>
     the Program is not required to print an announcement.)
    <div style="height: 1.00em;">&#x00A0;</div>
    These requirements apply to the modified work as a whole. If identifiable
      sections of that work are not derived from the Program, and can be
      reasonably considered independent and separate works in themselves, then
      this License, and its terms, do not apply to those sections when you
      distribute them as separate works. But when you distribute the same
      sections as part of a whole which is a work based on the Program, the
      distribution of the whole must be on the terms of this License, whose
      permissions for other licensees extend to the entire whole, and thus to
      each and every part regardless of who wrote it.
    <div style="height: 1.00em;">&#x00A0;</div>
    Thus, it is not the intent of this section to claim rights or contest your
      rights to work written entirely by you; rather, the intent is to exercise
      the right to control the distribution of derivative or collective works
      based on the Program.
    <div style="height: 1.00em;">&#x00A0;</div>
    In addition, mere aggregation of another work not based on the Program with
      the Program (or with a work based on the Program) on a volume of a storage
      or distribution medium does not bring the other work under the scope of
      this License.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     3. You may copy and distribute the Program (or a work based on it, under
      Section 2) in object code or executable form under the terms of Sections 1
      and 2 above provided that you also do one of the following:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     a) Accompany it with the complete corresponding machine-readable
    <br/>
     source code, which must be distributed under the terms of Sections
    <br/>
     1 and 2 above on a medium customarily used for software interchange; or,
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     b) Accompany it with a written offer, valid for at least three
    <br/>
     years, to give any third party, for a charge no more than your
    <br/>
     cost of physically performing source distribution, a complete
    <br/>
     machine-readable copy of the corresponding source code, to be
    <br/>
     distributed under the terms of Sections 1 and 2 above on a medium
    <br/>
     customarily used for software interchange; or,
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     c) Accompany it with the information you received as to the offer
    <br/>
     to distribute corresponding source code. (This alternative is
    <br/>
     allowed only for noncommercial distribution and only if you
    <br/>
     received the program in object code or executable form with such
    <br/>
     an offer, in accord with Subsection b above.)
    <div style="height: 1.00em;">&#x00A0;</div>
    The source code for a work means the preferred form of the work for making
      modifications to it. For an executable work, complete source code means
      all the source code for all modules it contains, plus any associated
      interface definition files, plus the scripts used to control compilation
      and installation of the executable. However, as a special exception, the
      source code distributed need not include anything that is normally
      distributed (in either source or binary form) with the major components
      (compiler, kernel, and so on) of the operating system on which the
      executable runs, unless that component itself accompanies the executable.
    <div style="height: 1.00em;">&#x00A0;</div>
    If distribution of executable or object code is made by offering access to
      copy from a designated place, then offering equivalent access to copy the
      source code from the same place counts as distribution of the source code,
      even though third parties are not compelled to copy the source along with
      the object code.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     4. You may not copy, modify, sublicense, or distribute the Program except
      as expressly provided under this License. Any attempt otherwise to copy,
      modify, sublicense or distribute the Program is void, and will
      automatically terminate your rights under this License. However, parties
      who have received copies, or rights, from you under this License will not
      have their licenses terminated so long as such parties remain in full
      compliance.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     5. You are not required to accept this License, since you have not signed
      it. However, nothing else grants you permission to modify or distribute
      the Program or its derivative works. These actions are prohibited by law
      if you do not accept this License. Therefore, by modifying or distributing
      the Program (or any work based on the Program), you indicate your
      acceptance of this License to do so, and all its terms and conditions for
      copying, distributing or modifying the Program or works based on it.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     6. Each time you redistribute the Program (or any work based on the
      Program), the recipient automatically receives a license from the original
      licensor to copy, distribute or modify the Program subject to these terms
      and conditions. You may not impose any further restrictions on the
      recipients' exercise of the rights granted herein. You are not responsible
      for enforcing compliance by third parties to this License.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     7. If, as a consequence of a court judgment or allegation of patent
      infringement or for any other reason (not limited to patent issues),
      conditions are imposed on you (whether by court order, agreement or
      otherwise) that contradict the conditions of this License, they do not
      excuse you from the conditions of this License. If you cannot distribute
      so as to satisfy simultaneously your obligations under this License and
      any other pertinent obligations, then as a consequence you may not
      distribute the Program at all. For example, if a patent license would not
      permit royalty-free redistribution of the Program by all those who receive
      copies directly or indirectly through you, then the only way you could
      satisfy both it and this License would be to refrain entirely from
      distribution of the Program.
    <div style="height: 1.00em;">&#x00A0;</div>
    If any portion of this section is held invalid or unenforceable under any
      particular circumstance, the balance of the section is intended to apply
      and the section as a whole is intended to apply in other circumstances.
    <div style="height: 1.00em;">&#x00A0;</div>
    It is not the purpose of this section to induce you to infringe any patents
      or other property right claims or to contest validity of any such claims;
      this section has the sole purpose of protecting the integrity of the free
      software distribution system, which is implemented by public license
      practices. Many people have made generous contributions to the wide range
      of software distributed through that system in reliance on consistent
      application of that system; it is up to the author/donor to decide if he
      or she is willing to distribute software through any other system and a
      licensee cannot impose that choice.
    <div style="height: 1.00em;">&#x00A0;</div>
    This section is intended to make thoroughly clear what is believed to be a
      consequence of the rest of this License.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     8. If the distribution and/or use of the Program is restricted in certain
      countries either by patents or by copyrighted interfaces, the original
      copyright holder who places the Program under this License may add an
      explicit geographical distribution limitation excluding those countries,
      so that distribution is permitted only in or among countries not thus
      excluded. In such case, this License incorporates the limitation as if
      written in the body of this License.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     9. The Free Software Foundation may publish revised and/or new versions of
      the General Public License from time to time. Such new versions will be
      similar in spirit to the present version, but may differ in detail to
      address new problems or concerns.
    <div style="height: 1.00em;">&#x00A0;</div>
    Each version is given a distinguishing version number. If the Program
      specifies a version number of this License which applies to it and
      &quot;any later version&quot;, you have the option of following the terms
      and conditions either of that version or of any later version published by
      the Free Software Foundation. If the Program does not specify a version
      number of this License, you may choose any version ever published by the
      Free Software Foundation.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     10. If you wish to incorporate parts of the Program into other free
      programs whose distribution conditions are different, write to the author
      to ask for permission. For software which is copyrighted by the Free
      Software Foundation, write to the Free Software Foundation; we sometimes
      make exceptions for this. Our decision will be guided by the two goals of
      preserving the free status of all derivatives of our free software and of
      promoting the sharing and reuse of software generally.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     NO WARRANTY
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     11. BECAUSE THE PROGRAM IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
      FOR THE PROGRAM, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT WHEN
      OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES
      PROVIDE THE PROGRAM &quot;AS IS&quot; WITHOUT WARRANTY OF ANY KIND, EITHER
      EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
      WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
      ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE PROGRAM IS WITH YOU.
      SHOULD THE PROGRAM PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL NECESSARY
      SERVICING, REPAIR OR CORRECTION.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     12. IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
      WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
      REDISTRIBUTE THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES,
      INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES
      ARISING OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT NOT
      LIMITED TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES
      SUSTAINED BY YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE
      WITH ANY OTHER PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN
      ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     END OF TERMS AND CONDITIONS
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     How to Apply These Terms to Your New Programs
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     If you develop a new program, and you want it to be of the greatest
      possible use to the public, the best way to achieve this is to make it
      free software which everyone can redistribute and change under these
      terms.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     To do so, attach the following notices to the program. It is safest to
      attach them to the start of each source file to most effectively convey
      the exclusion of warranty; and each file should have at least the
      &quot;copyright&quot; line and a pointer to where the full notice is
      found.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     &lt;one line to give the program's name and a brief idea of what it
      does.&gt;
    <br/>
     Copyright (C) &lt;year&gt; &lt;name of author&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     This program is free software; you can redistribute it and/or modify
    <br/>
     it under the terms of the GNU General Public License as published by
    <br/>
     the Free Software Foundation; either version 2 of the License, or
    <br/>
     (at your option) any later version.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     This program is distributed in the hope that it will be useful,
    <br/>
     but WITHOUT ANY WARRANTY; without even the implied warranty of
    <br/>
     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    <br/>
     GNU General Public License for more details.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     You should have received a copy of the GNU General Public License
    <br/>
     along with this program; if not, write to the Free Software
    <br/>
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
    <div style="height: 1.00em;">&#x00A0;</div>
    Also add information on how to contact you by electronic and paper mail.
    <div style="height: 1.00em;">&#x00A0;</div>
    If the program is interactive, make it output a short notice like this when
      it starts in an interactive mode:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     Gnomovision version 69, Copyright (C) year name of author
    <br/>
     Gnomovision comes with ABSOLUTELY NO WARRANTY; for details type `show w'.
    <br/>
     This is free software, and you are welcome to redistribute it
    <br/>
     under certain conditions; type `show c' for details.
    <div style="height: 1.00em;">&#x00A0;</div>
    The hypothetical commands `show w' and `show c' should show the appropriate
      parts of the General Public License. Of course, the commands you use may
      be called something other than `show w' and `show c'; they could even be
      mouse-clicks or menu items--whatever suits your program.
    <div style="height: 1.00em;">&#x00A0;</div>
    You should also get your employer (if you work as a programmer) or your
      school, if any, to sign a &quot;copyright disclaimer&quot; for the
      program, if necessary. Here is a sample; alter the names:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     Yoyodyne, Inc., hereby disclaims all copyright interest in the program
    <br/>
     `Gnomovision' (which makes passes at compilers) written by James Hacker.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     &lt;signature of Ty Coon&gt;, 1 April 1989
    <br/>
     Ty Coon, President of Vice
    <div style="height: 1.00em;">&#x00A0;</div>
    This General Public License does not permit incorporating your program into
      proprietary programs. If your program is a subroutine library, you may
      consider it more useful to permit linking proprietary applications with
      the library. If this is what you want to do, use the GNU Library General
      Public License instead of this License.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>RASLIC</b></dt>
  <dd class="It-tag">If you do not use the GPL, the following license terms
      apply:
    <div style="height: 1.00em;">&#x00A0;</div>
    RasMol License
    <div style="height: 1.00em;">&#x00A0;</div>
    Even though the authors of the various documents and software found here
      have made a good faith effort to ensure that the documents are correct and
      that the software performs according to its documentation, and we would
      greatly appreciate hearing of any problems you may encounter, the programs
      and documents any files created by the programs are provided **AS IS**
      without any warranty as to correctness, merchantability or fitness for any
      particular or general use.
    <div style="height: 1.00em;">&#x00A0;</div>
    THE RESPONSIBILITY FOR ANY ADVERSE CONSEQUENCES FROM THE USE OF PROGRAMS OR
      DOCUMENTS OR ANY FILE OR FILES CREATED BY USE OF THE PROGRAMS OR DOCUMENTS
      LIES SOLELY WITH THE USERS OF THE PROGRAMS OR DOCUMENTS OR FILE OR FILES
      AND NOT WITH AUTHORS OF THE PROGRAMS OR DOCUMENTS.
    <div style="height: 1.00em;">&#x00A0;</div>
    Subject to your acceptance of the conditions stated above, and your respect
      for the terms and conditions stated in the notices below, if you are not
      going to make any modifications or create derived works, you are given
      permission to freely copy and distribute this package, provided you do the
      following:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     1. Either include the complete documentation, especially the file NOTICE,
      with what you distribute or provide a clear indication where people can
      get a copy of the documentation; and
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     2. Please give credit where credit is due citing the version and original
      authors properly; and
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     3. Please do not give anyone the impression that the original authors are
      providing a warranty of any kind.
    <div style="height: 1.00em;">&#x00A0;</div>
    If you would like to use major pieces of RasMol in some other program, make
      modifications to RasMol, or in some other way make what a lawyer would
      call a &quot;derived work&quot;, you are not only permitted to do so, you
      are encouraged to do so. In addition to the things we discussed above,
      please do the following:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     4. Please explain in your documentation how what you did differs from this
      version of RasMol; and
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     5. Please make your modified source code available.
    <div style="height: 1.00em;">&#x00A0;</div>
    This version of RasMol is _not_ in the public domain, but it is given freely
      to the community in the hopes of advancing science. If you make changes,
      please make them in a responsible manner, and please offer us the
      opportunity to include those changes in future versions of RasMol.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>General Notice</b></dt>
  <dd class="It-tag">The following notice applies to this work as a whole and to
      the works included within it:
    <div style="height: 1.00em;">&#x00A0;</div>
    * Creative endeavors depend on the lively exchange of ideas. There are laws
      and customs which establish rights and responsibilities for authors and
      the users of what authors create. This notice is not intended to prevent
      you from using the software and documents in this package, but to ensure
      that there are no misunderstandings about terms and conditions of such
      use.
    <div style="height: 1.00em;">&#x00A0;</div>
    * Please read the following notice carefully. If you do not understand any
      portion of this notice, please seek appropriate professional legal advice
      before making use of the software and documents included in this software
      package. In addition to whatever other steps you may be obliged to take to
      respect the intellectual property rights of the various parties involved,
      if you do make use of the software and documents in this package, please
      give credit where credit is due by citing this package, its authors and
      the URL or other source from which you obtained it, or equivalent primary
      references in the literature with the same authors.
    <div style="height: 1.00em;">&#x00A0;</div>
    * Some of the software and documents included within this software package
      are the intellectual property of various parties, and placement in this
      package does not in any way imply that any such rights have in any way
      been waived or diminished.
    <div style="height: 1.00em;">&#x00A0;</div>
    * With respect to any software or documents for which a copyright exists,
      ALL RIGHTS ARE RESERVED TO THE OWNERS OF SUCH COPYRIGHT.
    <div style="height: 1.00em;">&#x00A0;</div>
    * Even though the authors of the various documents and software found here
      have made a good faith effort to ensure that the documents are correct and
      that the software performs according to its documentation, and we would
      greatly appreciate hearing of any problems you may encounter, the programs
      and documents and any files created by the programs are provided **AS IS**
      without any warranty as to correctness, merchantability or fitness for any
      particular or general use.
    <div style="height: 1.00em;">&#x00A0;</div>
    * THE RESPONSIBILITY FOR ANY ADVERSE CONSEQUENCES FROM THE USE OF PROGRAMS
      OR DOCUMENTS OR ANY FILE OR FILES CREATED BY USE OF THE PROGRAMS OR
      DOCUMENTS LIES SOLELY WITH THE USERS OF THE PROGRAMS OR DOCUMENTS OR FILE
      OR FILES AND NOT WITH AUTHORS OF THE PROGRAMS OR DOCUMENTS.
    <div style="height: 1.00em;">&#x00A0;</div>
    See the files GPL and RASLIC for two alternate ways to license this package.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>RasMol V2.6 Notice</b></dt>
  <dd class="It-tag">The following notice applies to RasMol V 2.6 and older
      RasMol versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    Information in this document is subject to change without notice and does
      not represent a commitment on the part of the supplier. This package is
      sold/distributed subject to the condition that it shall not, by way of
      trade or otherwise, be lent, re-sold, hired out or otherwise circulated
      without the supplier's prior consent, in any form of packaging or cover
      other than that in which it was produced. No part of this manual or
      accompanying software may be reproduced, stored in a retrieval system on
      optical or magnetic disk, tape or any other medium, or transmitted in any
      form or by any means, electronic, mechanical, photocopying, recording or
      otherwise for any purpose other than the purchaser's personal use.
    <div style="height: 1.00em;">&#x00A0;</div>
    This product is not to be used in the planning, construction, maintenance,
      operation or use of any nuclear facility nor the flight, navigation or
      communication of aircraft or ground support equipment. The author shall
      not be liable, in whole or in part, for any claims or damages arising from
      such use, including death, bankruptcy or outbreak of war.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>IUCR Policy</b></dt>
  <dd class="It-tag"><b>The IUCr Policy for the Protection and the Promotion of
      the STAR File and</b> <b>CIF Standards for Exchanging and Archiving
      Electronic Data.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Overview</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The Crystallographic Information File (CIF)[1] is a standard for information
      interchange promulgated by the International Union of Crystallography
      (IUCr). CIF (Hall, Allen &amp; Brown, 1991) is the recommended method for
      submitting publications to Acta Crystallographica Section C and reports of
      crystal structure determinations to other sections of Acta
      Crystallographica and many other journals. The syntax of a CIF is a subset
      of the more general STAR File[2] format. The CIF and STAR File approaches
      are used increasingly in the structural sciences for data exchange and
      archiving, and are having a significant influence on these activities in
      other fields.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Statement of intent</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The IUCr's interest in the STAR File is as a general data interchange
      standard for science, and its interest in the CIF, a conformant derivative
      of the STAR File, is as a concise data exchange and archival standard for
      crystallography and structural science.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Protection of the standards</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    To protect the STAR File and the CIF as standards for interchanging and
      archiving electronic data, the IUCr, on behalf of the scientific
      community,
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     * holds the copyrights on the standards themselves,
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     * owns the associated trademarks and service marks, and
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     * holds a patent on the STAR File.
    <div style="height: 1.00em;">&#x00A0;</div>
    These intellectual property rights relate solely to the interchange formats,
      not to the data contained therein, nor to the software used in the
      generation, access or manipulation of the data.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Promotion of the standards</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The sole requirement that the IUCr, in its protective role, imposes on
      software purporting to process STAR File or CIF data is that the following
      conditions be met prior to sale or distribution.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     * Software claiming to read files written to either the STAR File or the
      CIF standard must be able to extract the pertinent data from a file
      conformant to the STAR File syntax, or the CIF syntax, respectively.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     * Software claiming to write files in either the STAR File, or the CIF,
      standard must produce files that are conformant to the STAR File syntax,
      or the CIF syntax, respectively.
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     * Software claiming to read definitions from a specific data dictionary
      approved by the IUCr must be able to extract any pertinent definition
      which is conformant to the dictionary definition language (DDL)[3]
      associated with that dictionary.
    <div style="height: 1.00em;">&#x00A0;</div>
    The IUCr, through its Committee on CIF Standards, will assist any developer
      to verify that software meets these conformance conditions.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Glossary of terms</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    [1] CIF:
    <div style="height: 1.00em;">&#x00A0;</div>
    is a data file conformant to the file syntax defined at
      http://www.iucr.org/iucr-top/cif/spec/index.html
    <div style="height: 1.00em;">&#x00A0;</div>
    [2] STAR File:
    <div style="height: 1.00em;">&#x00A0;</div>
    is a data file conformant to the file syntax defined at
      http://www.iucr.org/iucr-top/cif/spec/star/index.html
    <div style="height: 1.00em;">&#x00A0;</div>
    [3] DDL:
    <div style="height: 1.00em;">&#x00A0;</div>
    is a language used in a data dictionary to define data items in terms of
      &quot;attributes&quot;. Dictionaries currently approved by the IUCr, and
      the DDL versions used to construct these dictionaries, are listed at
      http://www.iucr.org/iucr-top/cif/spec/ddl/index.html
    <div style="height: 1.00em;">&#x00A0;</div>
    Last modified: 30 September 2000
    <div style="height: 1.00em;">&#x00A0;</div>
    IUCr Policy Copyright (C) 2000 International Union of Crystallography
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>CBFLIB</b></dt>
  <dd class="It-tag">The following Disclaimer Notice applies to CBFlib V0.1,
      from which this code in part is derived.
    <div style="height: 1.00em;">&#x00A0;</div>
    * The items furnished herewith were developed under the sponsorship of the
      U.S. Government. Neither the U.S., nor the U.S. D.O.E., nor the Leland
      Stanford Junior University, nor their employees, makes any warranty,
      express or implied, or assumes any liability or responsibility for
      accuracy, completeness or usefulness of any information, apparatus,
      product or process disclosed, or represents that its use will not infringe
      privately-owned rights. Mention of any product, its manufacturer, or
      suppliers shall not, nor is it intended to, imply approval, disapproval,
      or fitness for any particular use. The U.S. and the University at all
      times retain the right to use and disseminate the furnished items for any
      purpose whatsoever.
    <div style="height: 1.00em;">&#x00A0;</div>
    Notice 91 02 01
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>CIFPARSE</b></dt>
  <dd class="It-tag">Portions of this software are loosely based on the CIFPARSE
      software package from the NDB at Rutgers University. See
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     http://ndbserver.rutgers.edu/NDB/mmcif/software
    <div style="height: 1.00em;">&#x00A0;</div>
    CIFPARSE is part of the NDBQUERY application, a program component of the
      Nucleic Acid Database Project [ H. M. Berman, W. K. Olson, D. L.
      Beveridge, J. K. Westbrook, A. Gelbin, T. Demeny, S. H. Shieh, A. R.
      Srinivasan, and B. Schneider. (1992). The Nucleic Acid Database: A
      Comprehensive Relational Database of Three-Dimensional Structures of
      Nucleic Acids. Biophys J., 63, 751-759.], whose cooperation is gratefully
      acknowledged, especially in the form of design concepts created by J.
      Westbrook.
    <div style="height: 1.00em;">&#x00A0;</div>
    Please be aware of the following notice in the CIFPARSE API:
    <div style="height: 1.00em;">&#x00A0;</div>
    This software is provided WITHOUT WARRANTY OF MERCHANTABILITY OR FITNESS FOR
      A PARTICULAR PURPOSE OR ANY OTHER WARRANTY, EXPRESS OR IMPLIED. RUTGERS
      MAKE NO REPRESENTATION OR WARRANTY THAT THE SOFTWARE WILL NOT INFRINGE ANY
      PATENT, COPYRIGHT OR OTHER PROPRIETARY RIGHT.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
RasMol is a molecular graphics program intended for the visualisation of
  proteins, nucleic acids and small molecules. The program is aimed at display,
  teaching and generation of publication quality images. RasMol runs on wide
  range of architectures and operating systems including Microsoft Windows,
  Apple Macintosh, UNIX and VMS systems. UNIX and VMS versions require an 8, 24
  or 32 bit colour X Windows display (X11R4 or later). The X Windows version of
  RasMol provides optional support for a hardware dials box and accelerated
  shared memory communication (via the XInput and MIT-SHM extensions) if
  available on the current X Server.
<div style="height: 1.00em;">&#x00A0;</div>
The program reads in a molecule coordinate file and interactively displays the
  molecule on the screen in a variety of colour schemes and molecule
  representations. Currently available representations include depth-cued
  wireframes, 'Dreiding' sticks, spacefilling (CPK) spheres, ball and stick,
  solid and strand biomolecular ribbons, atom labels and dot surfaces.
<div style="height: 1.00em;">&#x00A0;</div>
Up to 5 molecules may be loaded and displayed at once. Any one or all of
<br/>
 the molecules may be rotated and translated.
<div style="height: 1.00em;">&#x00A0;</div>
The RasMol help facility can be accessed by typing &quot;help
  &lt;topic&gt;&quot; or &quot;help &lt;topic&gt; &lt;subtopic&gt;&quot; from
  the command line. A complete list of RasMol commands may be displayed by
  typing &quot;help commands&quot;. A single question mark may also be used to
  abbreviate the keyword &quot;help&quot;. Please type &quot;help notices&quot;
  for important notices.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="COMMANDS"><a class="selflink" href="#COMMANDS">COMMANDS</a></h1>
RasMol allows the execution of interactive commands typed at the
  <b>RasMol&gt;</b> prompt in the terminal window. Each command must be given on
  a separate line. Keywords are case insensitive and may be entered in either
  upper or lower case letters. All whitespace characters are ignored except to
  separate keywords and their arguments.
<div style="height: 1.00em;">&#x00A0;</div>
All commands may be prefixed by a parenthesized <b>atom expression</b> to
  temporarily select certain atoms just for the execution of that one command.
  After execution of the command, the previous selection is restored except for
  the commands <b>select</b> , <b>restrict</b> and <b>script.</b>
<div style="height: 1.00em;">&#x00A0;</div>
The commands/keywords currently recognised by RasMol are given below.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Backbone</b></dt>
  <dd class="It-tag">The RasMol <b>backbone</b> command permits the
      representation of a polypeptide backbone as a series of bonds connecting
      the adjacent alpha carbons of each amino acid in a chain. The display of
      these backbone 'bonds' is turned on and off by the command parameter in
      the same way as with the <b>wireframe</b> command. The command <b>backbone
      off</b> turns off the selected 'bonds', and <b>backbone on</b> or with a
      number turns them on. The number can be used to specify the cylinder
      radius of the representation in either Angstrom or RasMol units. A
      parameter value of 500 (2.0 Angstroms) or above results in a
      &quot;Parameter value too large&quot; error. Backbone objects may be
      coloured using the RasMol <b>colour backbone</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
    The reserved word backbone is also used as a predefined set (&quot;help
      sets&quot;) and as a parameter to the <b>set hbond</b> and <b>set
      ssbond</b> commands. The RasMol command <b>trace</b> renders a smoothed
      backbone, in contrast to <b>backbone</b> which connects alpha carbons with
      straight lines.
    <div style="height: 1.00em;">&#x00A0;</div>
    The backbone may be displayed with dashed lines by use of the <b>backbone
      dash</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Background</b></dt>
  <dd class="It-tag">The RasMol <b>background</b> command is used to set the
      colour of the &quot;canvas&quot; background. The colour may be given as
      either a colour name or a comma separated triple of Red, Green and Blue
      (RGB) components enclosed in square brackets. Typing the command <b>help
      colours</b> will give a list of the predefined colour names recognised by
      RasMol. When running under X Windows, RasMol also recognises colours in
      the X server's colour name database.
    <div style="height: 1.00em;">&#x00A0;</div>
    The <b>background</b> command is synonymous with the RasMol <b>set
      background</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Bond</b></dt>
  <dd class="It-tag">The RasMol command <b>bond &lt;number&gt; &lt;number&gt;
      +</b> adds the designated bond to the drawing, increasing the bond order
      if the bond already exists. The command <b>bond &lt;number&gt;
      &lt;number&gt; pick</b> selects the two atoms specified by the atom serial
      numbers as the two ends of a bond around which the <b>rotate bond
      &lt;angle&gt;</b> command will be applied. If no bond exists, it is
      created.
    <div style="height: 1.00em;">&#x00A0;</div>
    Rotation around a previously picked bond may be specified by the <b>rotate
      bond &lt;angle&gt;</b> command, or may also be controlled with the mouse,
      using the <b>bond rotate on/off</b> or the equivalent <b>rotate bond
      on/off</b> commands.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Bulgarian</b></dt>
  <dd class="It-tag">The RasMol <b>Bulgarian</b> command sets the menus and
      messages to the Bulgarian versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command may not work correctly unless appropriate fonts have been
      installed. The commands <b>Bulgarian,</b> <b>Chinese,</b> <b>English,</b>
      <b>French,</b> <b>Italian,</b> <b>Russian</b> and <b>Spanish</b> may be
      used to select Bulgarian, Chinese, English, French, Italian, Japanese,
      Russian and Spanish menus and messages if the appropriate fonts have been
      installed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Cartoon</b></dt>
  <dd class="It-tag">The RasMol <b>cartoon</b> command does a display of a
      molecule <b>ribbons</b> as Richardson (MolScript) style protein
      <b>cartoons,</b> implemented as thick (deep) ribbons. The easiest way to
      obtain a cartoon representation of a protein is to use the <b>Cartoons</b>
      option on the <b>Display</b> menu. The <b>cartoon</b> command represents
      the currently selected residues as a deep ribbon with width specified by
      the command's argument. Using the command without a parameter results in
      the ribbon's width being taken from the protein's secondary structure, as
      described in the <b>ribbons</b> command. By default, the C-termini of
      beta-sheets are displayed as arrow heads. This may be enabled and disabled
      using the <b>set cartoons</b> command. The depth of the cartoon may be
      adjusted using the <b>set cartoons &lt;number&gt;</b> command. The <b>set
      cartoons</b> command without any parameters returns these two options to
      their default values.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Centre</b></dt>
  <dd class="It-tag">The RasMol <b>centre</b> command defines the point about
      which the <b>rotate</b> command and the scroll bars rotate the current
      molecule. Without a parameter the centre command resets the centre of
      rotation to be the centre of gravity of the molecule. If an atom
      expression is specified, RasMol rotates the molecule about the centre of
      gravity of the set of atoms specified by the expression. Hence, if a
      single atom is specified by the expression, that atom will remain
      'stationary' during rotations.
    <div style="height: 1.00em;">&#x00A0;</div>
    Type <b>help expression</b> for more information on RasMol atom expressions.
    <div style="height: 1.00em;">&#x00A0;</div>
    Alternatively the centring may be given as a comma separated triple of
      [CenX, CenY, CenZ] offsets in RasMol units (1/250 of an Angstrom) from the
      centre of gravity. The triple must be enclosed in square brackets.
    <div style="height: 1.00em;">&#x00A0;</div>
    The optional forms <b>centre ... translate</b> and <b>centre ... center</b>
      may be used to specify use of a translated centre of rotation (not
      necessarily in the centre of the canvas) or a centre of rotation which is
      placed at the centre of the canvas. Starting with RasMol 2.7.2, the
      default is to center the new axis on the canvas.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Chinese</b></dt>
  <dd class="It-tag">The RasMol <b>Chinese</b> command sets the menus and
      messages to the Chinese versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command may not work correctly unless appropriate fonts have been
      installed. The commands <b>Bulgarian,</b> <b>Chinese,</b> <b>English,</b>
      <b>French,</b> <b>Italian,</b> <b>Russian</b> and <b>Spanish</b> may be
      used to select Bulgarian, Chinese, English, French, Italian, Japanese,
      Russian and Spanish menus and messages if the appropriate fonts have been
      installed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Clipboard</b></dt>
  <dd class="It-tag">The RasMol <b>clipboard</b> command places a copy of the
      currently displayed image on the local graphics 'clipboard'. Note: this
      command is not yet supported on UNIX or VMS machines. It is intended to
      make transfering images between applications easier under Microsoft
      Windows or on an Apple Macintosh.
    <div style="height: 1.00em;">&#x00A0;</div>
    When using RasMol on a UNIX or VMS system this functionality may be achieved
      by generating a raster image in a format that can be read by the receiving
      program using the RasMol <b>write</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Colour</b></dt>
  <dd class="It-tag">Colour the atoms (or other objects) of the selected region.
      The colour may be given as either a colour name or a comma separated
      triple of Red, Green and Blue (RGB) components enclosed in square
      brackets. Typing the command <b>help colours</b> will give a list of all
      the predefined colour names recognised by RasMol.
    <div style="height: 1.00em;">&#x00A0;</div>
    Allowed objects are <b>atoms,</b> <b>bonds,</b> <b>backbone,</b>
      <b>ribbons,</b> <b>labels,</b> <b>dots,</b> <b>hbonds,</b> <b>map,</b> and
      <b>ssbonds.</b> If no object is specified, the default keyword <b>atom</b>
      is assumed. Some colour schemes are defined for certain object types. The
      colour scheme <b>none</b> can be applied to all objects except atoms and
      dots, stating that the selected objects have no colour of their own, but
      use the colour of their associated atoms (i.e. the atoms they connect).
      <b>Atom</b> objects can also be coloured by <b>alt,</b> <b>amino,</b>
      <b>chain,</b> <b>charge,</b> <b>cpk,</b> <b>group,</b> <b>model,</b>
      <b>shapely,</b> <b>structure,</b> <b>temperature</b> or <b>user.</b>
      Hydrogen bonds can also be coloured by <b>type</b> and dot surfaces can
      also be coloured by <b>electrostatic potential.</b> For more information
      type <b>help colour &lt;colour&gt;.</b> Map objects may be coloured by
      specific color of by nearest atom.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>ColourMode</b></dt>
  <dd class="It-tag">ColourMode allows the user to switch between using the new
      <b>colour</b> method. At present, the new coloring technique is the same
      as the old one, but to preserve compatibility for older scripts it may be
      wise to add a &quot;colormode on&quot; near the top of your script
      somewhere, if the script was designed for version 2.7.3 of RasMol or
      earlier. The new color method, when completed, aims to fix a few bugs in
      the coloring routines.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Connect</b></dt>
  <dd class="It-tag">The RasMol <b>connect</b> command is used to force RasMol
      to (re)calculate the connectivity of the current molecule. If the original
      input file contained connectivity information, this is discarded. The
      command <b>connect false</b> uses a fast heuristic algorithm that is
      suitable for determining bonding in large bio-molecules such as proteins
      and nucleic acids. The command <b>connect true</b> uses a slower more
      accurate algorithm based upon covalent radii that is more suitable to
      small molecules containing inorganic elements or strained rings. If no
      parameters are given, RasMol determines which algorithm to use based on
      the number of atoms in the input file. Greater than 255 atoms causes
      RasMol to use the faster implementation. This is the method used to
      determine bonding, if necessary, when a molecule is first read in using
      the <b>load</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Defer</b></dt>
  <dd class="It-tag">The RasMol <b>defer</b> command adds the command given to
      the macro with given name, if no name is given, the command is added to
      the macro with a blank name. The command <b>zap</b> is a special case. In
      that case the macro is erased. If no name is given the command must begin
      with a selection, e.g. <b>defer (selection).spacefill</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The deferred commands accumulated under the given name can be executed using
      the <b>execute</b> command
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Define</b></dt>
  <dd class="It-tag">The RasMol <b>define</b> command allows the user to
      associate an arbitrary set of atoms with a unique identifier. This allows
      the definition of user-defined sets. These sets are declared statically,
      i.e. once defined the contents of the set do not change, even if the
      expression defining them depends on the current transformation and
      representation of the molecule.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Depth</b></dt>
  <dd class="It-tag">The RasMol <b>depth</b> command enables, disables or
      positions the back-clipping plane of the molecule. The program only draws
      those portions of the molecule that are closer to the viewer than the
      clipping plane. Integer values range from zero at the very back of the
      molecule to 100 which is completely in front of the molecule. Intermediate
      values determine the percentage of the molecule to be drawn.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command interacts with the <b>slab &lt;value&gt;</b> command, which
      clips to the front of a given z-clipping plane.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Dots</b></dt>
  <dd class="It-tag">The RasMol <b>dots</b> command is used to generate a van
      der Waals' dot surface around the currently selected atoms. Dot surfaces
      display regularly spaced points on a sphere of van der Waals' radius about
      each selected atom. Dots that would are 'buried' within the van der Waals'
      radius of any other atom (selected or not) are not displayed. The command
      <b>dots on</b> deletes any existing dot surface and generates a dots
      surface around the currently selected atom set with a default dot density
      of 100. The command <b>dots off</b> deletes any existing dot surface. The
      dot density may be specified by providing a numeric parameter between 1
      and 1000. This value approximately corresponds to the number of dots on
      the surface of a medium sized atom.
    <div style="height: 1.00em;">&#x00A0;</div>
    By default, the colour of each point on a dot surface is the colour of its
      closest atom at the time the surface is generated. The colour of the whole
      dot surface may be changed using the <b>colour dots</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Echo</b></dt>
  <dd class="It-tag">The RasMol <b>echo</b> command is used to display a message
      in the RasMol command/terminal window. The string parameter may optionally
      be delimited in double quote characters. If no parameter is specified, the
      <b>echo</b> command displays a blank line. This command is particularly
      useful for displaying text from within a RasMol <b>script</b> file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>English</b></dt>
  <dd class="It-tag">The RasMol <b>English</b> command sets the menus and
      messages to the English versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command may not work correctly unless appropriate fonts have been
      installed. The commands <b>Bulgarian,</b> <b>Chinese,</b> <b>English,</b>
      <b>French,</b> <b>Italian,</b> <b>Russian</b> and <b>Spanish</b> may be
      used to select Bulgarian, Chinese, English, French, Italian, Japanese,
      Russian and Spanish menus and messages if the appropriate fonts have been
      installed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Execute</b></dt>
  <dd class="It-tag">The RasMol <b>execute</b> command:
    <div style="height: 1.00em;">&#x00A0;</div>
    1. saves the old poise of the molecule (translation, rotation and zoom)
    <div style="height: 1.00em;">&#x00A0;</div>
    2. executes the specified macro suppressing both screen updates and
      recording
    <div style="height: 1.00em;">&#x00A0;</div>
    3. animates motion of the newly rendered molecule linearly from the old
      poise to the new poise
    <div style="height: 1.00em;">&#x00A0;</div>
    The macro must have been previously defined by calls to the <b>defer</b>
      command.
    <div style="height: 1.00em;">&#x00A0;</div>
    The animation of the motion depends on the prior settings of the
      <b>record</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>French</b></dt>
  <dd class="It-tag">The RasMol <b>French</b> command sets the menus and
      messages to the French versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command may not work correctly unless appropriate fonts have been
      installed. The commands <b>Bulgarian,</b> <b>Chinese,</b> <b>English,</b>
      <b>French,</b> <b>Italian,</b> <b>Russian</b> and <b>Spanish</b> may be
      used to select Bulgarian, Chinese, English, French, Italian, Japanese,
      Russian and Spanish menus and messages if the appropriate fonts have been
      installed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>HBonds</b></dt>
  <dd class="It-tag">The RasMol <b>hbond</b> command is used to represent the
      hydrogen bonding of the protein molecule's backbone. This information is
      useful in assessing the protein's secondary structure. Hydrogen bonds are
      represented as either dotted lines or cylinders between the donor and
      acceptor residues. The first time the <b>hbond</b> command is used, the
      program searches the structure of the molecule to find hydrogen bonded
      residues and reports the number of bonds to the user. The command
      <b>hbonds on</b> displays the selected 'bonds' as dotted lines, and the
      <b>hbonds off</b> turns off their display. The colour of hbond objects may
      be changed by the <b>colour hbond</b> command. Initially, each hydrogen
      bond has the colours of its connected atoms.
    <div style="height: 1.00em;">&#x00A0;</div>
    By default the dotted lines are drawn between the accepting oxygen and the
      donating nitrogen. By using the <b>set hbonds</b> command the alpha carbon
      positions of the appropriate residues may be used instead. This is
      especially useful when examining proteins in backbone representation.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Help</b></dt>
  <dd class="It-tag">The RasMol <b>help</b> command provides on-line help on the
      given topic.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Italian</b></dt>
  <dd class="It-tag">The RasMol <b>Italian</b> command sets the menus and
      messages to the Italian versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command may not work correctly unless appropriate fonts have been
      installed. The commands <b>Bulgarian,</b> <b>Chinese,</b> <b>English,</b>
      <b>French,</b> <b>Italian,</b> <b>Russian</b> and <b>Spanish</b> may be
      used to select Bulgarian, Chinese, English, French, Italian, Japanese,
      Russian and Spanish menus and messages if the appropriate fonts have been
      installed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Japanese</b></dt>
  <dd class="It-tag">The RasMol <b>Japanese</b> command sets the menus and
      messages to the Japanese versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command may not work correctly unless appropriate fonts have been
      installed. The commands <b>Bulgarian,</b> <b>Chinese,</b> <b>English,</b>
      <b>French,</b> <b>Italian,</b> <b>Russian</b> and <b>Spanish</b> may be
      used to select Bulgarian, Chinese, English, French, Italian, Japanese,
      Russian and Spanish menus and messages if the appropriate fonts have been
      installed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Label</b></dt>
  <dd class="It-tag">The RasMol <b>label</b> command allows an arbitrary
      formatted text string to be associated with each currently selected atom.
      This string may contain embedded 'expansion specifiers' which display
      properties of the atom being labelled. An expansion specifier consists of
      a '%' character followed by a single alphabetic character specifying the
      property to be displayed. An actual '%' character may be displayed by
      using the expansion specifier '%%'.
    <div style="height: 1.00em;">&#x00A0;</div>
    Atom labelling for the currently selected atoms may be turned off with the
      command <b>label off.</b> By default, if no string is given as a
      parameter, RasMol uses labels appropriate for the current molecule.
    <div style="height: 1.00em;">&#x00A0;</div>
    The colour of each label may be changed using the <b>colour label</b>
      command. By default, each label is drawn in the same colour as the atom to
      which it is attached. The size and spacing of the displayed text may be
      changed using the <b>set fontsize</b> command. The width of the strokes in
      the displayed text may be changed
    <br/>
     using the <b>set fontstroke</b>
    <br/>
     command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Load</b></dt>
  <dd class="It-tag">Load a molecule coordinate file into RasMol. Valid molecule
      file formats are <b>pdb</b> (Protein Data Bank format), <b>mdl</b>
      (Molecular Design Limited's MOL file format), <b>alchemy</b> (Tripos'
      Alchemy file format), <b>mol2</b> (Tripos' Sybyl Mol2 file format),
      <b>charmm</b> (CHARMm file format), <b>xyz</b> (MSC's XMol XYZ file
      format), <b>mopac</b> (J. P. Stewart's MOPAC file format) or <b>cif</b>
      (IUCr CIF or mmCIF file format). If no file format is specified,
      <b>PDB,</b> <b>CIF,</b> or <b>mmCIF</b> is assumed by default. Up to 20
      molecules may be loaded at a time. If CHEM_COMP ligand models are included
      in an mmCIF file, they will be loaded as NMR models, first giving the all
      the NMR models for model coordinates if specified and then giving all the
      NMR models for ideal model coordinates.
    <div style="height: 1.00em;">&#x00A0;</div>
    To delete a molecule prior to loading another use the RasMol <b>zap</b>
      command. To select a molecule for manipulation use the RasMol <b>molecule
      &lt;n&gt;</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
    The <b>load</b> command selects all the atoms in the molecule, centres it on
      the screen and renders it as a CPK coloured wireframe model. If the
      molecule contains no bonds (i.e. contains only alpha carbons), it is drawn
      as an alpha carbon backbone. If the file specifies fewer bonds than atoms,
      RasMol determines connectivity using the <b>connect</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
    The <b>load inline</b> command also allows the storing of atom coordinates
      in scripts to allow better integration with WWW browsers. A load command
      executed inside a script file may specify the keyword <b>inline</b>
      instead of a conventional filename. This option specifies that the
      coordinates of the molecule to load are stored in the same file as the
      currently executing commands.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map</b></dt>
  <dd class="It-tag">The RasMol <b>map</b> commands manipulate electron density
      maps in coordination with the display of molecules. These commands are
      very memory intensive and may not work on machines with limited memory.
      Each molecule may have as many maps as available memory permits. Maps may
      be read from files or generated from Gaussian density distributions around
      atoms.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>map colour,</b> to colour a map according to a given colour scheme,
      <b>map generate,</b> to generate a map from selected atoms based on
      pseudo-Gaussians, <b>map level,</b> to set the contouring level for
      selected maps, <b>map load,</b> to load a map from a file, <b>map mask</b>
      to designate a mask for the selected maps, <b>map resolution,</b> to set
      the resolution for contouring selected maps, <b>map restrict,</b> to
      select one or more maps and to disable all others, <b>map save,</b> to
      save map information to a file, <b>map scale,</b> <b>control the scaling
      of pseudo-Gaussians when generating maps,</b> <b>map select,</b> to select
      one or more maps, <b>map show,</b> to display information about one or
      more maps or about the parameters to be used in generating or loading the
      next map, <b>map spacing,</b> to set the spacing betwen contour lines of
      selected maps, <b>map spread,</b> to set the variance of the Gaussians for
      map generation as a fraction of the atomic radius, and <b>map zap</b> to
      delete previously generated or loaded maps.
    <div style="height: 1.00em;">&#x00A0;</div>
    The effect of <b>map generate</b> and <b>map load</b> commands is modified
      by the <b>map mask</b> command which limits the portion of the display
      space that can be considered for display of maps.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map colour</b></dt>
  <dd class="It-tag">The RasMol <b>map colour</b> command colours the selected
      maps according to the specified colour scheme. The colour scheme may be a
      colour name or and RBG triple in brackets, or the keyword <b>atom</b> to
      cause the map points to be coloured by the color of the nearest atom.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map generate</b></dt>
  <dd class="It-tag">The RasMol <b>map generate</b> command generates a map from
      whatever atoms are currently selected, by summing electron densities
      approximated by Gaussian distributions. The height of each Gaussian is
      determined by the setting of the <b>map scale</b> command. In the default
      of map scale true, each Gaussian has a height proportional element type of
      the atom. If the optional 'LRSurf' parameter is given or if map scale
      false has been executed, each Gaussian is scaled so that the Gaussian
      contour level 1 is at the van der Waals radius. In either case a standard
      deviation determined by the most recently specified spread or resolution
      is used. If a non-zero spread has been given the radius of the atom is
      multiplied by the spread to find the standard deviation. The default is
      2/3rds. If a resolution has been given, the spread is inferred as 2/3rds
      of the resolution.
    <div style="height: 1.00em;">&#x00A0;</div>
    For example, if the resolution is given as 1., and the atom in question is a
      Carbon with a van der Waals radius of 468 RasMol units (1.87 Angstroms),
      the inferred spead is .6667, and the standard deviation of the Gaussian is
      taken as 1.25 Angstroms.
    <div style="height: 1.00em;">&#x00A0;</div>
    If the spread has been set to zero, the spread for each atom is determined
      from the van der Waals radius and the probe atom radius to simulate the
      effect of a Lee-Richards surface.
    <div style="height: 1.00em;">&#x00A0;</div>
    If no specific map was given by the map selector, the new map is given the
      next available map number.
    <div style="height: 1.00em;">&#x00A0;</div>
    If a specific map was given by the map selector, the new map replaces that
      map. If more than one map was given by the map selector, the new map
      replaces the lowest numbered of the selected maps. In any case the new map
      becomes the currently selected map.
    <div style="height: 1.00em;">&#x00A0;</div>
    The map is displayed as dots, mesh or a surface, depending on the last map
      rendering mode selected or the mode selected on the command itself.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map level</b></dt>
  <dd class="It-tag">The RasMol <b>map level</b> command sets the contour level
      to be used in creating subsequent representations of generated or loaded
      maps. If the keyword MEAN in used the level is relative to the mean of the
      map data. Otherwise the level is absolute.
    <div style="height: 1.00em;">&#x00A0;</div>
    In general, a lower level results in a map containing more of the displayed
      volume, while a higher level results in a map containing less of the
      displayed volume.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map load</b></dt>
  <dd class="It-tag">The RasMol <b>map load</b> command loads a map file into
      RasMol. The valid formats are CCP4 map format and imgCIF format.
    <div style="height: 1.00em;">&#x00A0;</div>
    If no specific map was given by the map selector, the new map is given the
      next available map number.
    <div style="height: 1.00em;">&#x00A0;</div>
    If a specific map was given by the map selector, the new map replaces that
      map. If more than one map was given by the map selector, the new map
      replaces the lowest numbered of the selected maps. In any case the new map
      becomes the currently selected map.
    <div style="height: 1.00em;">&#x00A0;</div>
    The map is displayed as dots, mesh or a surface depending on the last map
      rendering mode selected.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map mask</b></dt>
  <dd class="It-tag">The RasMol <b>map mask</b> command specifies a mask to be
      used to limit the display space to be used for making representations of
      other maps or removes an earlier mask specification.
    <div style="height: 1.00em;">&#x00A0;</div>
    The 'selected' option indicates that the mask is to be created from the
      currently selected atoms. The '&lt;number&gt;' option indicates that the
      mask is to be copied from the map of the number specified. The 'none'
      option removes the previously specified mask, if any.
    <div style="height: 1.00em;">&#x00A0;</div>
    The map selector specifies the map or maps to which the specified mask will
      the applied. For example, 'map next mask selected' specifies that the
      currently selected atoms are to be used to generate a mask to be applied
      to any maps created by subsequent 'map load' or 'map generate' commands.
    <div style="height: 1.00em;">&#x00A0;</div>
    Any map may be used as a mask. The portions of the mask map greater than
      than or equal to the average value of the mask map allow the values of the
      map being masked to be used as given. The portions of the mask map lower
      than the average value of the mask map cause the values of the map being
      masked to be treated as if they were equal to the lowest data value of the
      map being masked.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map resolution</b></dt>
  <dd class="It-tag">The RasMol <b>map resolution</b> command specifies the
      resolution in RasMol units or, if a number containing a decimal point is
      given, the resolution in Angstroms to be used in generating and in
      representing maps.
    <div style="height: 1.00em;">&#x00A0;</div>
    The resolution is used at the map spacing for representations of maps,
      indicating the separation between contour levels (see the <b>map
      spacing</b> command) and to infer the map spread to be used in generated
      maps from selected atoms (see the <b>map spread</b> command). The map
      spread is set to two thirds of the specified resolution.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map restrict</b></dt>
  <dd class="It-tag">The RasMol <b>map restrict</b> command selects particular
      maps to make them active for subsequent map commands. This is similar to
      the <b>map select</b> command, but does disables the display of the maps
      that were not selected.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map save</b></dt>
  <dd class="It-tag">The RasMol <b>map save</b> command saves an imgCIF map
      file.
    <div style="height: 1.00em;">&#x00A0;</div>
    If no specific map was given by the map selector, the currently selected
      maps and their masks are written to the file, one map and mask pair per
      data block.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map scale</b></dt>
  <dd class="It-tag">The RasMol <b>map scale</b> command selects the scaling of
      pseudo-Gaussians in the <b>map generate</b> commands. In the default of
      map scale true, each Gaussian has a height proportional element type of
      the atom. If map scale false has been executed, each Gaussian is scaled so
      that the Gaussian contour level 1 is at the van der Waals radius. In
      either case a standard deviation determined by the most recently specified
      spread or resolution is used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map select</b></dt>
  <dd class="It-tag">The RasMol <b>map select</b> command selects particular
      maps to make them active for subsequent map commands. This is similar to
      the <b>map restrict</b> command, but does not disable the display of the
      maps that were not selected.
    <div style="height: 1.00em;">&#x00A0;</div>
    If the optional <b>atom</b> parameter is given, the command selects the
      atoms with centres closest to the map points. The radius of the search may
      be specified by the parameter <b>search_radius.</b> The default is to look
      for atoms within 4 Angstroms plus the probe radius. If the optional
      <b>within</b> parameter is given, the new selection is taken from within
      the currently selected atoms. If the options <b>add</b> parameter is
      given, the new selection is added to the currently selected atoms. The
      default is to search within all atoms.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map show</b></dt>
  <dd class="It-tag">The RasMol <b>map show</b> command causes information about
      the maps specified by the map selector to be written to the command
      window.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map spacing</b></dt>
  <dd class="It-tag">The RasMol <b>map spacing</b> command specifies the spacing
      to be used between contour lines in creating representations of maps. The
      spacing is typically
    <br/>
     given in Angstroms with a decimal point, but may also be specified in
      RasMol units (250ths of an Angstom) as an integer. For maps loaded in grid
      coordinates that spacing is parallel to the cell edges. The default
      spacing is one half Angstrom.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map spread</b></dt>
  <dd class="It-tag">The RasMol <b>map spread</b> command specifies the
      reciprocal of the number of standard deviations per radius to be used in
      generating maps as sums of Gaussians centered on atomic positions. The
      default spread is one two thirds (i.e. each radius covers 1.5 standard
      deviations).
    <div style="height: 1.00em;">&#x00A0;</div>
    If the spread has been set to zero, the spread for each atom is determined
      from the van der Waals radius and the probe atom radius to simulate the
      effect of a Lee-Richards surface.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Map zap</b></dt>
  <dd class="It-tag">The RasMol <b>map zap</b> command removes the data and
      representations of the maps specified by the map selector. The map numbers
      of maps that have not been removed are not changed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Molecule</b></dt>
  <dd class="It-tag">The RasMol <b>molecule</b> command selects one of up to 5
      previously loaded molecules for active manipulation. While all the
      molcules are displayed and may be rotated collectively (see the <b>rotate
      all</b> command), only one molecule at a time time is active for
      manipulation by the commands which control the details of rendering.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Monitor</b></dt>
  <dd class="It-tag">The RasMol <b>monitor</b> command allows the display of
      distance monitors. A distance monitor is a dashed (dotted) line between an
      arbitrary pair of atoms, optionally labelled by the distance between them.
      The RasMol command <b>monitor &lt;number&gt; &lt;number&gt;</b> adds such
      a distance monitor between the two atoms specified by the atom serial
      numbers given as parameters
    <div style="height: 1.00em;">&#x00A0;</div>
    Distance monitors are turned off with the command <b>monitors off.</b> By
      default, monitors display the distance between its two end points as a
      label at the centre of the monitor. These distance labels may be turned
      off with the command <b>set monitors off,</b> and re-enabled with the
      command <b>set monitors on.</b> Like most other representations, the
      colour of a monitor is taken from the colour of its end points unless
      specified by the <b>colour monitors</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
    Distance monitors may also be added to a molecule interactively with the
      mouse, using the <b>set picking monitor</b> command. Clicking on an atom
      results in its being identified on the rasmol command line. In addition
      every atom picked increments a modulo counter such that, in monitor mode,
      every second atom displays the distance between this atom and the previous
      one. The shift key may be used to form distance monitors between a fixed
      atom and several consecutive positions. A distance monitor may also be
      removed (toggled) by selecting the appropriate pair of atom end points a
      second time.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Notoggle</b></dt>
  <dd class="It-tag">The RasMol <b>NoToggle</b> command enables or disables the
      use of the toggle ability that is used by some of the other RasMol
      commands. When no boolean value is specified, NoToggle mode is ENABLED.
      When NoToggle mode is ENABLED, all toggle functionality is DISABLED. To
      turn it off, one must explicitly set <b>notoggle off.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    Some commands which use the toggle feature are: <b>ColourMode.</b> More
      functions that utilize this capability may be added at a later date.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Pause</b></dt>
  <dd class="It-tag">The RasMol <b>pause</b> command is used in script files to
      stop the script file for local manipulation by a mouse, until any key is
      pushed to restart the script file. <b>Wait</b> is synonymous with
      <b>pause.</b> This command may be executed in RasMol script files to
      suspend the sequential execution of commands and allow the user to examine
      the current image. When RasMol executes a <b>pause</b> command in a script
      file, it suspends execution of the rest of the file, refreshes the image
      on the screen and allows the manipulation of the image using the mouse and
      scroll bars, or resizing of the graphics window. Once a key is pressed,
      control returns to the script file at the line following the <b>pause</b>
      command. While a script is suspended the molecule may be rotated,
      translated, scaled, slabbed and picked as usual, but all menu commands are
      disabled.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Play</b></dt>
  <dd class="It-tag">The RasMol <b>play</b> command specifies the recording
      medium from which to play back a movie. The playback frame start time is
      given in seconds to millisecond precision. Since we are working on
      computers, the medium is specified as a set of files, each marked with the
      playback frame start time in milliseconds as part of the name. The place
      in the name at which to look for the playback frame start time in
      milliseconds is marked by the characters &quot;ssssss&quot; with an
      appropriate number of digits. RasMol accepts either upper or lower case
      s's or decimal digits to mark the place for the time. The play off and
      play eject commands effectively remove the specified medium from use. If
      no medium is specified, play off suspends playing and play on resumes
      playing. Normally play starts immediately and runs to the end of the
      medium. However, if play off and/or or some combination of play from and
      play until is entered before <b>play type medium,</b> those settings will
      be used.
    <div style="height: 1.00em;">&#x00A0;</div>
    As of release 2.7.5, RasMol support play from scripts and data files.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Print</b></dt>
  <dd class="It-tag">The RasMol <b>print</b> command sends the currently
      displayed image to the local default printer using the operating system's
      native printer driver. Note: this command is not yet supported under UNIX
      or VMS. It is intended to take advantage of Microsoft Windows and Apple
      Macintosh printer drivers. For example, this allows images to be printed
      directly on a dot matrix printer.
    <div style="height: 1.00em;">&#x00A0;</div>
    When using RasMol on a UNIX or VMS system this functionality may be achieved
      by either generating a PostScript file using the RasMol <b>write ps</b> or
      <b>write vectps</b> commands and printing that or generating a raster
      image file and using a utility to dump that to the local printer.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Quit</b></dt>
  <dd class="It-tag">Exit from the RasMol program. The RasMol commands
      <b>exit</b> and <b>quit</b> are synonymous, except within nested scripts.
      In that case, <b>exit</b> terminates only the current level, while
      <b>quit</b> terminates all nested levels of scripts.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Record</b></dt>
  <dd class="It-tag">The RasMol <b>record</b> command specifies the recording
      medium to hold the movie. Since we are working on computers, the medium is
      specified as a template for a set of files, each marked with the playback
      frame start time in milliseconds (rather than as seconds to avoid
      embedding a decimal point) as part of the name. The place in the name to
      be replaced with the playback frame start time in milliseconds is marked
      by the characters &quot;ssssss&quot; with an appropriate number of digits.
      RasMol accepts either upper or lower case s's or decimal digits to mark
      the place for the time. The record off commands remove the specified
      medium from use. If no medium is specified, record off suspends recording
      and record on resumes recording with the next available time on the same
      medium. The screen is the default medium and is, by default, on. Writing
      to disk must be explicitly specified so that the disk does not get filled
      up unintentionally. The type of a recording medium may be an image type
      such as gif, pict or png to record the actual screen images or script to
      record the RasMol commands used to generate the frames.
    <div style="height: 1.00em;">&#x00A0;</div>
    Normally recording starts at playback frame start time 0 seconds. A non-zero
      starting time in seconds can be specified with the <b>record from</b>
      command as in <b>record from 25</b> or <b>record from 37.25</b> to help in
      organizing scenes of movies to be assembled later in an appropriate order.
      The <b>record until</b> command allows an upper limit to be set on
      recording time in seconds. The default is to have no limit. Issuing the
      commands
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>record from 600</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>record until 1800</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    would result in a 20 minute movie segment intended to start 10 minutes into
      a longer movie. These commands allow control over rewriting selected time
      segments.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Refresh</b></dt>
  <dd class="It-tag">The RasMol <b>refresh</b> command redraws the current
      image. This is useful in scripts to ensure application of a complex list
      of parameter changes.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Renumber</b></dt>
  <dd class="It-tag">The RasMol <b>renumber</b> command sequentially numbers the
      residues in a macromolecular chain. The optional parameter specifies the
      value of the first residue in the sequence. By default, this value is one.
      For proteins, each amino acid is numbered consecutively from the N
      terminus to the C terminus. For nucleic acids, each base is numbered from
      the 5' terminus to the 3' terminus. All chains in the current database are
      renumbered and gaps in the original sequence are ignored. The starting
      value for numbering may be negative.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Reset</b></dt>
  <dd class="It-tag">The RasMol <b>reset</b> command restores the original
      viewing transformation and centre of rotation. The scale is set to its
      default value, <b>zoom 100,</b> the centre of rotation is set to the
      geometric centre of the currently loaded molecule, <b>centre all,</b> this
      centre is translated to the middle of the screen and the viewpoint set to
      the default orientation.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command should not be mistaken for the RasMol <b>zap</b> command which
      deletes the currently stored molecule, returning the program to its
      initial state.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Restrict</b></dt>
  <dd class="It-tag">The RasMol <b>restrict</b> command both defines the
      currently selected region of the molecule and disables the representation
      of (most of) those parts of the molecule no longer selected. All
      subsequent RasMol commands that modify a molecule's colour or
      representation affect only the currently selected region. The parameter of
      a <b>restrict</b> command is a RasMol atom expression that is evaluated
      for every atom of the current molecule. This command is very similar to
      the RasMol <b>select</b> command, except <b>restrict</b> disables the
      <b>wireframe,</b> <b>spacefill</b> and <b>backbone</b> representations in
      the non-selected region.
    <div style="height: 1.00em;">&#x00A0;</div>
    Type &quot;help expression&quot; for more information on RasMol atom
      expressions or see section <b>Atom Expressions.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Ribbons</b></dt>
  <dd class="It-tag">The RasMol <b>ribbons</b> command displays the currently
      loaded protein or nucleic acid as a smooth solid &quot;ribbon&quot;
      surface passing along the backbone of the protein. The ribbon is drawn
      between each amino acid whose alpha carbon is currently selected. The
      colour of the ribbon is changed by the RasMol <b>colour ribbon</b>
      command. If the current ribbon colour is <b>none</b> (the default), the
      colour is taken from the alpha carbon at each position along its length.
    <div style="height: 1.00em;">&#x00A0;</div>
    The width of the ribbon at each position is determined by the optional
      parameter in the usual RasMol units. By default the width of the ribbon is
      taken from the secondary structure of the protein or a constant value of
      720 (2.88 Angstroms) for nucleic acids. The default width of protein alpha
      helices and beta sheets is 380 (1.52 Angstroms) and 100 (0.4 Angstroms)
      for turns and random coil. The secondary structure assignment is either
      from the PDB file or calculated using the DSSP algorithm as used by the
      <b>structure</b> command. This command is similar to the RasMol command
      <b>strands</b> which renders the biomolecular ribbon as parallel
      depth-cued curves.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Rotate</b></dt>
  <dd class="It-tag">Rotate the molecule about the specified axis. Permitted
      values for the axis parameter are &quot;x&quot;, &quot;y&quot;,
      &quot;z&quot; and &quot;bond&quot;. The integer parameter states the angle
      in degrees for the structure to be rotated. For the X and Y axes, positive
      values move the closest point up and right, and negative values move it
      down and left, respectively. For the Z axis, a positive rotation acts
      clockwise and a negative angle anti-clockwise.
    <div style="height: 1.00em;">&#x00A0;</div>
    Alternatively, this command may be used to specify which rotations the mouse
      or dials will control. If <b>rotate bond true</b> is selected, the
      horizontal scroll bar will control rotation around the axis selected by
      the <b>bond src dst pick</b> command. If <b>rotate all true</b> is
      selected, and multiple molecules have been loaded, then all molecules will
      rotate together. In all other cases, the mouseand dials control the the
      rotation of the molecule selected by the <b>molecule n</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Russian</b></dt>
  <dd class="It-tag">The RasMol <b>Russian</b> command sets the menus and
      messages to the Russian versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command may not work correctly unless appropriate fonts have been
      installed. The commands <b>Bulgarian,</b> <b>Chinese,</b> <b>English,</b>
      <b>French,</b> <b>Italian,</b> <b>Russian</b> and <b>Spanish</b> may be
      used to select Bulgarian, Chinese, English, French, Italian, Japanese,
      Russian and Spanish menus and messages if the appropriate fonts have been
      installed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Save</b></dt>
  <dd class="It-tag">Save the currently selected set of atoms in a Protein Data
      Bank (PDB), MDL, Alchemy(tm) or XYZ format file. The distinction between
      this command and the RasMol <b>write</b> command has been dropped. The
      only difference is that without a format specifier the <b>save</b> command
      generates a <b>PDB</b> file and the <b>write</b> command generates a
      <b>GIF</b> image.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Script</b></dt>
  <dd class="It-tag">The RasMol <b>script</b> command reads a set of RasMol
      commands sequentially from a text file and executes them. This allows
      sequences of commonly used commands to be stored and performed by single
      command. A RasMol script file may contain a further script command up to a
      maximum &quot;depth&quot; of 10, allowing complicated sequences of actions
      to be executed. RasMol ignores all characters after the first '#'
      character on each line allowing the scripts to be annotated. Script files
      are often also annotated using the RasMol <b>echo</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
    The most common way to generate a RasMol script file is to use the <b>write
      script</b> or <b>write rasmol</b> commands to output the sequence of
      commands that are needed to regenerate the current view, representation
      and colouring of the currently displayed molecule.
    <div style="height: 1.00em;">&#x00A0;</div>
    The RasMol command <b>source</b> is synonymous with the <b>script</b>
      command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Select</b></dt>
  <dd class="It-tag">Define the currently selected region of the molecule. All
      subsequent RasMol commands that manipulate a molecule or modify its colour
      or representation only affect the currently selected region. The parameter
      of a <b>select</b> command is a RasMol expression that is evaluated for
      every atom of the current molecule. The currently selected (active) region
      of the molecule are those atoms that cause the expression to evaluate
      true. To select the whole molecule use the RasMol command <b>select
      all.</b> The behaviour of the <b>select</b> command without any parameters
      is determined by the RasMol <b>hetero</b> and <b>hydrogen</b> parameters.
    <div style="height: 1.00em;">&#x00A0;</div>
    Type &quot;help expression&quot; for more information on RasMol atom
      expressions or see section <b>Atom Expressions.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set</b></dt>
  <dd class="It-tag">The RasMol <b>set</b> command allows the user to alter
      various internal program parameters such as those controlling rendering
      options. Each parameter has its own set or permissible parameter options.
      Typically, omitting the paramter option resets that parameter to its
      default value. A list of valid parameter names is given below.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Show</b></dt>
  <dd class="It-tag">The RasMol <b>show</b> command display details of the
      status of the currently loaded molecule. The command <b>show
      information</b> lists the molecule's name, classification, PDB code and
      the number of atoms, chains, groups it contains. If hydrogen bonding,
      disulphide bridges or secondary structure have been determined, the number
      of hbonds, ssbonds, helices, ladders and turns are also displayed,
      respectively. The command <b>show centre</b> shows any non-zero centering
      values selected by the <b>centre [CenX, CenY, CenZ]</b> command. The
      command <b>show phipsi</b> shows the phi and psi angles of the currently
      selected residues and the omega angles of cis peptide bonds. The command
      <b>show RamPrint</b> (or 'show RPP' or 'show RamachandranPrinterPlot')
      shows a simple Ramachandran printer plot in the style of Frances
      Bernstein's fisipl program. The command <b>show rotation</b> (or 'show
      rot' or 'show 'rotate') shows the currently selected values of z, y, x and
      bond rotations, if any. The command <b>show selected</b> (or 'show
      selected group' or 'show selected chain' or 'show selected atom' ) shows
      the groups (default), chains or atoms of the current selection. The
      command <b>show sequence</b> lists the residues that comprise each chain
      of the molecule. The command <b>show symmetry</b> shows the space group
      and unit cell of the molecule. The command <b>show translation</b> shows
      any non-zero translation values selected by the <b>translate &lt;axis&gt;
      &lt;value&gt;</b> command. The command <b>show zoom</b> shows any non-zero
      zoom value selected by the <b>zoom &lt;value&gt;</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Slab</b></dt>
  <dd class="It-tag">The RasMol <b>slab</b> command enables, disables or
      positions the z-clipping plane of the molecule. The program only draws
      those portions of the molecule that are further from the viewer than the
      slabbing plane. Integer values range from zero at the very back of the
      molecule to 100 which is completely in front of the molecule. Intermediate
      values determine the percentage of the molecule to be drawn.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command interacts with the <b>depth &lt;value&gt;</b> command, which
      clips to the rear of a given z-clipping plane.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Spacefill</b></dt>
  <dd class="It-tag">The RasMol <b>spacefill</b> command is used to represent
      all of the currently selected atoms as solid spheres. This command is used
      to produce both union-of-spheres and ball-and-stick models of a molecule.
      The command, <b>spacefill true,</b> the default, represents each atom as a
      sphere of van der Waals radius. The command <b>spacefill off</b> turns off
      the representation of the selected atom as spheres. A sphere radius may be
      specified as an integer in RasMol units (1/250th Angstrom) or a value
      containing a decimal point. A value of 500 (2.0 Angstroms) or greater
      results in a &quot;Parameter value too large&quot; error.
    <div style="height: 1.00em;">&#x00A0;</div>
    The <b>temperature</b> option sets the radius of each sphere to the value
      stored in its temperature field. Zero or negative values have no effect
      and values greater than 2.0 are truncated to 2.0. The <b>user</b> option
      allows the radius of each sphere to be specified by additional lines in
      the molecule's PDB file using Raster 3D's COLOUR record extension.
    <div style="height: 1.00em;">&#x00A0;</div>
    The RasMol command <b>cpk</b> is synonymous with the <b>spacefill</b>
      command.
    <div style="height: 1.00em;">&#x00A0;</div>
    The RasMol command <b>cpknew</b> is synonymous with the <b>spacefill</b>
      command, except that a slightly different set of colours is used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Spanish</b></dt>
  <dd class="It-tag">The RasMol <b>Spanish</b> command sets the menus and
      messages to the Spanish versions.
    <div style="height: 1.00em;">&#x00A0;</div>
    This command may not work correctly unless appropriate fonts have been
      installed. The commands <b>Bulgarian,</b> <b>Chinese,</b> <b>English,</b>
      <b>French,</b> <b>Italian,</b> <b>Russian</b> and <b>Spanish</b> may be
      used to select Bulgarian, Chinese, English, French, Italian, Japanese,
      Russian and Spanish menus and messages if the appropriate fonts have been
      installed.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>SSBonds</b></dt>
  <dd class="It-tag">The RasMol <b>ssbonds</b> command is used to represent the
      disulphide bridges of the protein molecule as either dotted lines or
      cylinders between the connected cysteines. The first time that the
      <b>ssbonds</b> command is used, the program searches the structure of the
      protein to find half-cysteine pairs (cysteines whose sulphurs are within 3
      Angstroms of each other) and reports the number of bridges to the user.
      The command <b>ssbonds on</b> displays the selected &quot;bonds&quot; as
      dotted lines, and the command <b>ssbonds off</b> disables the display of
      ssbonds in the currently selected area. Selection of disulphide bridges is
      identical to normal bonds, and may be adjusted using the RasMol <b>set
      bondmode</b> command. The colour of disulphide bonds may be changed using
      the <b>colour ssbonds</b> command. By default, each disulphide bond has
      the colours of its connected atoms.
    <div style="height: 1.00em;">&#x00A0;</div>
    By default disulphide bonds are drawn between the sulphur atoms within the
      cysteine groups. By using the <b>set ssbonds</b> command the position of
      the cysteine's alpha carbons may be used instead.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Star</b></dt>
  <dd class="It-tag">The RasMol <b>star</b> command is used to represent all of
      the currently selected atoms as stars (six strokes, one each in the x, -x,
      y, -y, z and -z directions). The commands <b>select not bonded</b>
      followed by <b>star 75</b> are useful to mark unbonded atoms in a
      <b>wireframe</b> display with less overhead than provided by <b>spacefill
      75.</b> This can be done automatically for all subsequent wireframe
      displays with the command <b>set bondmode not bonded.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The command <b>star true,</b> the default, represents each atom as a star
      with strokes length equal to van der Waals radius. The command <b>star
      off</b> turns off the representation of the selected atom as stars. A star
      stroke length may be specified as an integer in RasMol units (1/250th
      Angstrom) or a value containing a decimal point. A value of 500 (2.0
      Angstroms) or greater results in a &quot;Parameter value too large&quot;
      error.
    <div style="height: 1.00em;">&#x00A0;</div>
    The <b>temperature</b> option sets the stroke length of each star to the
      value stored in its temperature field. Zero or negative values have no
      effect and values greater than 2.0 are truncated to 2.0. The <b>user</b>
      option allows the stroke length of each star to be specified by additional
      lines in the molecule's PDB file using Raster 3D's COLOUR record
      extension.
    <div style="height: 1.00em;">&#x00A0;</div>
    The RasMol <b>spacefill</b> command can be used for more artistic rendering
      of atoms as spheres.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Stereo</b></dt>
  <dd class="It-tag">The RasMol <b>stereo</b> command provides side-by-side
      stereo display of images. Stereo viewing of a molecule may be turned on
      (and off) either by selecting <b>Stereo</b> from the <b>Options</b> menu,
      or by typing the commands <b>stereo on</b> or <b>stereo off.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    Starting with RasMol version 2.7.2.1, the <b>Stereo</b> menu selection and
      the command <b>stereo</b> without arguments cycle from the initial state
      of <b>stereo off</b> to <b>stereo on</b> in cross-eyed mode to <b>stereo
      on</b> in wall-eyed mode and then back to <b>stereo off.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The separation angle between the two views may be adjusted with the <b>set
      stereo [-] &lt;number&gt;</b> command, where positive values result in
      crossed eye viewing and negative values in relaxed (wall-eyed) viewing.
      The inclusion of <b>[-] &lt;number&gt;</b> in the <b>stereo</b> command,
      as for example in <b>stereo 3</b> or <b>stereo -5,</b> also controls angle
      and direction.
    <div style="height: 1.00em;">&#x00A0;</div>
    The stereo command is only partially implemented. When stereo is turned on,
      the image is not properly recentred. (This can be done with a <b>translate
      x -&lt;number&gt;</b>
    <br/>
     command.) It is not supported in vector PostScript output files, is not
      saved by the <b>write script</b> command, and in general is not yet
      properly interfaced with several other features of the program.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Strands</b></dt>
  <dd class="It-tag">The RasMol <b>strands</b> command displays the currently
      loaded protein or nucleic acid as a smooth &quot;ribbon&quot; of
      depth-cued curves passing along the backbone of the protein. The ribbon is
      composed of a number of strands that run parallel to one another along the
      peptide plane of each residue. The ribbon is drawn between each amino acid
      whose alpha carbon is currently selected. The colour of the ribbon is
      changed by the RasMol <b>colour ribbon</b> command. If the current ribbon
      colour is <b>none</b> (the default), the colour is taken from the alpha
      carbon at each position along its length. The central and outermost
      strands may be coloured independently using the <b>colour ribbon1</b> and
      <b>colour ribbon2</b> commands, respectively. The number of strands in the
      ribbon may be altered using the RasMol <b>set strands</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
    The width of the ribbon at each position is determined by the optional
      parameter in the usual RasMol units. By default the width of the ribbon is
      taken from the secondary structure of the protein or a constant value of
      720 for nucleic acids (which produces a ribbon 2.88 Angstroms wide). The
      default width of protein alpha helices and beta sheets is 380 (1.52
      Angstroms) and 100 (0.4 Angstroms) for turns and random coil. The
      secondary structure assignment is either from the PDB file or calculated
      using the DSSP algorithm as used by the <b>structure</b> command. This
      command is similar to the RasMol command <b>ribbons</b> which renders the
      biomolecular ribbon as a smooth shaded surface.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Structure</b></dt>
  <dd class="It-tag">The RasMol <b>structure</b> command calculates secondary
      structure assignments for the currently loaded protein. If the original
      PDB file contained structural assignment records (HELIX, SHEET and TURN)
      these are discarded. Initially, the hydrogen bonds of the current molecule
      are found, if this hasn't been done already. The secondary structure is
      then determined using Kabsch and Sander's DSSP algorithm. Once finished
      the program reports the number of helices, strands and turns found.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Surface</b></dt>
  <dd class="It-tag">The RasMol <b>surface</b> command renders a Lee-Richards
      molecular surface resulting from rolling a probe atom on the selected
      atoms. The value given specifies the radius of the probe. If given in the
      first form, the evolute of the surface of the probe is shown (the solvent
      excluded surface). If given in the second form, the envelope of the
      positions of the center of the probe is shown (the solvent accessible
      surface).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Trace</b></dt>
  <dd class="It-tag">The RasMol <b>trace</b> command displays a smooth spline
      between consecutive alpha carbon positions. This spline does not pass
      exactly through the alpha carbon position of each residue, but follows the
      same path as <b>ribbons,</b> <b>strands</b> and <b>cartoons.</b> Note that
      residues may be displayed as <b>ribbons,</b> <b>strands,</b>
      <b>cartoons</b> or as a <b>trace.</b> Enabling one of these
      representations disables the others. However, a residue may be displayed
      simultaneously as backbone and as one of the above representations. This
      may change in future versions of RasMol. Prior to version 2.6,
      <b>trace</b> was synonymous with <b>backbone.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Trace temperature</b> displays the backbone as a wider cylinder at high
      temperature factors and thinner at lower. This representation is useful to
      X-ray crystallographers and NMR spectroscopists.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Translate</b></dt>
  <dd class="It-tag">The RasMol <b>translate</b> command moves the position of
      the centre of the molecule on the screen. The axis parameter specifies
      along which axis the molecule is to be moved and the integer parameter
      specifies the absolute position of the molecule centre from the middle of
      the screen. Permitted values for the axis parameter are &quot;x&quot;,
      &quot;y&quot; and &quot;z&quot;. Displacement values must be between -100
      and 100 which correspond to moving the current molecule just off the
      screen. A positive &quot;x&quot; displacement moves the molecule to the
      right, and a positive &quot;y&quot; displacement moves the molecule down
      the screen. The pair of commands <b>translate x 0</b> and <b>translate y
      0</b> centres the molecule on the screen.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>UnBond</b></dt>
  <dd class="It-tag">The RasMol command <b>unbond &lt;number&gt;
      &lt;number&gt;</b> removes the designated bond from the drawing.
    <div style="height: 1.00em;">&#x00A0;</div>
    The command <b>unbond</b> without arguments removes a bond previously picked
      by the <b>bond &lt;number&gt; &lt;number&gt; pick</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Wireframe</b></dt>
  <dd class="It-tag">The RasMol <b>wireframe</b> command represents each bond
      within the selected region of the molecule as a cylinder, a line or a
      depth-cued vector. The display of bonds as depth-cued vectors (drawn
      darker the further away from the viewer) is turned on by the command
      <b>wireframe</b> or <b>wireframe on.</b> The selected bonds are displayed
      as cylinders by specifying a radius either as an integer in RasMol units
      or containing a decimal point as a value in Angstroms. A parameter value
      of 500 (2.0 Angstroms) or above results in an &quot;Parameter value too
      large&quot; error. Bonds may be coloured using the <b>colour bonds</b>
      command.
    <div style="height: 1.00em;">&#x00A0;</div>
    If the selected bonds involved atoms of alternate conformers then the bonds
      are narrowed in the middle to a radius of .8 of the specified radius (or
      to the radius specifed as the optional second parameter).
    <div style="height: 1.00em;">&#x00A0;</div>
    Non-bonded atoms, which could become invisible in an ordinary
      <b>wireframe</b> display can be marked by a preceding <b>set bondmode not
      bonded</b> command. If nearly co-linear bonds to atoms cause them to be
      difficult to see in a wireframe display, the <b>set bondmode all</b>
      command will add markers for <b>all</b> atoms in subsequent
      <b>wireframe</b> command executions.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Write</b></dt>
  <dd class="It-tag">Write the current image to a file in a standard format.
      Currently supported image file formats include <b>bmp</b> (Microsoft
      bitmap) and <b>gif</b> (Compuserve GIF), <b>iris</b> (IRIS RGB),
      <b>ppm</b> (Portable Pixmap), <b>ras</b> (Sun rasterfile), <b>ps</b> and
      <b>epsf</b> (Encapsulated PostScript), <b>monops</b> (Monochrome
      Encapsulated PostScript), <b>pict</b> (Apple PICT), <b>vectps</b> (Vector
      Postscript). The <b>write</b> command may also be used to generate command
      scripts for other graphics programs. The format <b>script</b> writes out a
      file containing the RasMol <b>script</b> commands to reproduce the current
      image. The format <b>molscript</b> writes out the commands required to
      render the current view of the molecule as ribbons in Per Kraulis'
      Molscript program and the format <b>kinemage</b> the commands for David
      Richardson's program Mage. The following formats are useful for further
      processing: <b>povray</b> (POVRay 2), <b>povray3</b> (POVRay 3 -- under
      development), <b>vrml</b> (VRML file). Finally, several formats are
      provided to provide phi-psi data for listing or for <b>phipsi</b> (phi-psi
      data as an annotated list with cis omegas), <b>ramachan</b> and <b>RDF</b>
      and <b>RamachandranDataFile</b> (phi-psi data as columns of numbers for
      gnuplot), <b>RPP</b> and <b>RamachandranPrinterPlot</b> (phi-psi data as a
      printer plot).
    <div style="height: 1.00em;">&#x00A0;</div>
    The distinction between this command and the RasMol <b>save</b> command has
      been dropped. The only difference is that without a format specifier the
      <b>save</b> command generates a <b>PDB</b> file and the <b>write</b>
      command generates a <b>GIF</b> image.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Zap</b></dt>
  <dd class="It-tag">Deletes the contents of the current database and resets
      parameter variables to their initial default state.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Zoom</b></dt>
  <dd class="It-tag">Change the magnification of the currently displayed image.
      Boolean parameters either magnify or reset the scale of current molecule.
      An integer parameter specifies the desired magnification as a percentage
      of the default scale. The minimum parameter value is 10; the maximum
      parameter value is dependent upon the size of the molecule being
      displayed. For medium sized proteins this is about 500.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SET_PARAMETERS"><a class="selflink" href="#SET_PARAMETERS">SET
  PARAMETERS</a></h1>
RasMol has a number of internal parameters that may be modified using the
  <b>set</b> command. These parameters control a number of program options such
  as rendering options and mouse button mappings.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 picking play.fps radius record.aps
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Ambient</b></dt>
  <dd class="It-tag">The RasMol <b>ambient</b> parameter is used to control the
      amount of ambient (or surrounding) light in the scene. The <b>ambient</b>
      value must be between 0 and 100. It controls the percentage intensity of
      the darkest shade of an object. For a solid object, this is the intensity
      of surfaces facing away from the light source or in shadow. For depth-cued
      objects this is the intensity of objects furthest from the viewer.
    <div style="height: 1.00em;">&#x00A0;</div>
    This parameter is commonly used to correct for monitors with different
      &quot;gamma values&quot; (brightness), to change how light or dark a
      hardcopy image appears when printed or to alter the feeling of depth for
      wireframe or ribbon representations.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Axes</b></dt>
  <dd class="It-tag">The RasMol <b>axes</b> parameter controls the display of
      orthogonal coordinate axes on the current display. The coordinate axes are
      those used in the molecule data file, and the origin is the centre of the
      molecule's bounding box. The <b>set axes</b> command is similar to the
      commands <b>set boundbox</b> and <b>set unitcell</b> that display the
      bounding box and the crystallographic unit cell, respectively.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Backfade</b></dt>
  <dd class="It-tag">The RasMol <b>backfade</b> parameter is used to control
      backfade to the specified background colour, rather than black. This is
      controlled by the commands <b>set backfade on</b> and <b>set backfade
      off.</b> For example, this may be used to generate depth-cued images that
      fade to white, rather than black.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Background</b></dt>
  <dd class="It-tag">The RasMol <b>background</b> parameter is used to set the
      colour of the &quot;canvas&quot; background. The colour may be given as
      either a colour name or a comma separated triple of Red, Green, Blue (RGB)
      components enclosed in square brackets. Typing the command <b>help
      colours</b> will give a list of the predefined colour names recognised by
      RasMol. When running under X Windows, RasMol also recognises colours in
      the X server's colour name database.
    <div style="height: 1.00em;">&#x00A0;</div>
    The command <b>set background</b> is synonymous with the RasMol command
      <b>background.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set BondMode</b></dt>
  <dd class="It-tag">The RasMol <b>set bondmode</b> command controls the
      mechanism used to select individual bonds and modifies the display of
      bonded and non-bonded atoms by subsequent <b>wireframe</b> commands.
    <div style="height: 1.00em;">&#x00A0;</div>
    When using the <b>select</b> and <b>restrict</b> commands, a given bond will
      be selected if i) the bondmode is <b>or</b> and either of the connected
      atoms is selected, or ii) the bondmode is <b>and</b> and both atoms
      connected by the bond are selected. Hence an individual bond may be
      uniquely identified by using the command <b>set bondmode and</b> and then
      uniquely selecting the atoms at both ends.
    <div style="height: 1.00em;">&#x00A0;</div>
    The <b>bondmode [all | none | not bonded]</b> commands add <b>star 75</b> or
      <b>spacefill 75</b> markers for the designated atoms to <b>wireframe</b>
      displays. Stars are used when the specified wireframe radius is zero.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Bonds</b></dt>
  <dd class="It-tag">The RasMol <b>bonds</b> parameter is used to control
      display of double and triple bonds as multiple lines or cylinders.
      Currently bond orders are only read from MDL Mol files, Sybyl Mol2 format
      files, Tripos Alchemy format files, CIF and mmCIF, and suitable PDB files.
      Double (and triple) bonds are specified in some PDB files by specifying a
      given bond twice (and three times) in CONECT records. The command <b>set
      bonds on</b> enables the display of bond orders, and the command <b>set
      bonds off</b> disables them.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set BoundBox</b></dt>
  <dd class="It-tag">The RasMol <b>boundbox</b> parameter controls the display
      of the current molecule's bounding box on the display. The bounding box is
      orthogonal to the data file's original coordinate axes. The <b>set
      boundbox</b> command is similar to the commands <b>set axes</b> and <b>set
      unitcell</b> that display orthogonal coordinate axes and the bounding box,
      respectively.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Cartoon</b></dt>
  <dd class="It-tag">The RasMol <b>cartoon</b> parameter is used to control
      display of the cartoon version of the <b>ribbons</b> display. By default,
      the C-termini of beta-sheets are displayed as arrow heads. This may be
      enabled and disabled using the <b>set cartoons &lt;boolean&gt;</b>
      command. The depth of the cartoon may be adjusted using the <b>cartoons
      &lt;number&gt;</b> command. The <b>set cartoons</b> command without any
      parameters returns these two options to
    <br/>
     their default values.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set CisAngle</b></dt>
  <dd class="It-tag">The RasMol <b>cisangle</b> parameter controls the cutoff
      angle for identifying cis peptide
    <br/>
     bonds. If no value is given, the cutoff is set to 90 degrees.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Display</b></dt>
  <dd class="It-tag">This command controls the display mode within RasMol. By
      default, <b>set display normal,</b> RasMol displays the molecule in the
      representation specified by the user. The command <b>set display
      selected</b> changes the display mode such that the molecule is
      temporarily drawn so as to indicate currently selected portion of the
      molecule. The user specified colour scheme and representation remains
      unchanged. In this representation all selected atoms are shown in yellow
      and all non selected atoms are shown in blue. The colour of the background
      is also changed to a dark grey to indicate the change of display mode.
      This command is typically only used by external Graphical User Interfaces
      (GUIs).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set FontSize</b></dt>
  <dd class="It-tag">The RasMol <b>set fontsize</b> command is used to control
      the size of the characters that form atom labels. This value corresponds
      to the height of the displayed character in pixels. The maximum value of
      <b>fontsize</b> is 48 pixels, and the default value is 8 pixels high.
      Fixed or proportional spacing may be selected by appending the
      &quot;FS&quot; or &quot;PS&quot; modifiers, respectively. The default is
      &quot;FS&quot;. To display atom labels on the screen use the RasMol
      <b>label</b> command and to change the colour of displayed labels, use the
      <b>colour labels</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set FontStroke</b></dt>
  <dd class="It-tag">The RasMol <b>set fontstroke</b> command is used to control
      the size of the stroke width of the characters that form atom labels. This
      value is the radius in pixels of cylinders used to form the strokes. The
      special value of &quot;0&quot; is the default used for the normal single
      pixel stroke width, which allows for rapid drawing and rotation of the
      image. Non-zero values are provided to allow for more artistic atom labels
      for publication at the expense of extra time in rendering the image.
    <div style="height: 1.00em;">&#x00A0;</div>
    When wider strokes are used, a larger font size is recommend, e.g. by using
      the RasMol <b>set fontsize 24 PS</b> command, followed by <b>set
      fontstroke 2</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    To display atom labels on the screen use the RasMol <b>label</b> command,
      and to change the colour of displayed labels use the <b>colour labels</b>
      command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set HBonds</b></dt>
  <dd class="It-tag">The RasMol <b>hbonds</b> parameter determines whether
      hydrogen bonds are drawn between the donor and acceptor atoms of the
      hydrogen bond, <b>set hbonds sidechain</b> or between the alpha carbon
      atoms of the protein backbone and between the phosphorous atoms of the
      nucleic acid backbone, <b>set hbonds backbone.</b> The actual display of
      hydrogen bonds is controlled by the <b>hbonds</b> command. Drawing
      hydrogen bonds between protein alpha carbons or nucleic acid phosphorous
      atoms is useful when the rest of the molecule is shown in only a schematic
      representation such as <b>backbone,</b> <b>ribbons</b> or <b>strands.</b>
      This parameter is similar to the RasMol <b>ssbonds</b> parameter.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Hetero</b></dt>
  <dd class="It-tag">The RasMol <b>hetero</b> parameter is used to modify the
      'default' behaviour of the RasMol <b>select</b> command, i.e. the
      behaviour of <b>select</b> without any parameters. When this value is
      <b>false,</b> the default <b>select</b> region does not include any
      heterogeneous atoms (refer to the predefined set <b>hetero</b> ). When
      this value is <b>true,</b> the default <b>select</b> region may contain
      hetero atoms. This parameter is similar to the RasMol <b>hydrogen</b>
      parameter which determines whether hydrogen atoms should be included in
      the default set. If both <b>hetero</b> and <b>hydrogen</b> are
      <b>true,</b> <b>select</b> without any parameters is equivalent to
      <b>select all.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set HourGlass</b></dt>
  <dd class="It-tag">The RasMol <b>hourglass</b> parameter allows the user to
      enable and disable the use of the 'hour glass' cursor used by RasMol to
      indicate that the program is currently busy drawing the next frame. The
      command <b>set hourglass on</b> enables the indicator, whilst <b>set
      hourglass off</b> prevents RasMol from changing the cursor. This is useful
      when spinning the molecule, running a sequence of commands from a script
      file or using interprocess communication to execute complex sequences of
      commands. In these cases a 'flashing' cursor may be distracting.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Hydrogen</b></dt>
  <dd class="It-tag">The RasMol <b>hydrogen</b> parameter is used to modify the
      &quot;default&quot; behaviour of the RasMol <b>select</b> command, i.e.
      the behaviour of <b>select</b> without any parameters. When this value is
      <b>false,</b> the default <b>select</b> region does not include any
      hydrogen, deuterium or tritium atoms (refer to the predefined set
      <b>hydrogen</b> ). When this value is <b>true,</b> the default
      <b>select</b> region may contain hydrogen atoms. This parameter is similar
      to the RasMol <b>hetero</b> parameter which determines whether
      heterogeneous atoms should be included in the default set. If both
      <b>hydrogen</b> and <b>hetero</b> are <b>true,</b> <b>select</b> without
      any parameters is equivalent to <b>select all.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Kinemage</b></dt>
  <dd class="It-tag">The RasMol <b>set kinemage</b> command controls the amount
      of detail stored in a Kinemage output file generated by the RasMol
      <b>write kinemage</b> command. The output kinemage files are intended to
      be displayed by David Richardson's Mage program. <b>set kinemage
      false,</b> the default, only stores the currently displayed representation
      in the generated output file. The command <b>set kinemage true,</b>
      generates a more complex Kinemage that contains both the wireframe and
      backbone representations as well as the coordinate axes, bounding box and
      crystal unit cell.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Menus</b></dt>
  <dd class="It-tag">The RasMol <b>set menus</b> command enables the canvas
      window's menu buttons or menu bar. This command is typically only used by
      graphical user interfaces or to create as large an image as possible when
      using Microsoft Windows.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Monitor</b></dt>
  <dd class="It-tag">The RasMol <b>set monitor</b> command enables
      <b>monitors.</b> The distance monitor labels may be turned off with the
      command <b>set monitor off,</b> and re-enabled with the command <b>set
      monitor on.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Mouse</b></dt>
  <dd class="It-tag">The RasMol <b>set mouse</b> command sets the rotation,
      translation, scaling and zooming mouse bindings. The default value is
      <b>rasmol</b> which is suitable for two button mice (for three button mice
      the second and third buttons are synonymous); X-Y rotation is controlled
      by the first button, and X-Y translation by the second. Additional
      functions are controlled by holding a modifier key on the keyboard.
      [Shift] and the first button performs scaling, [shift] and the second
      button performs Z-rotation, and [control] and the first mouse button
      controls the clipping plane. The <b>insight</b> and <b>quanta</b> options
      provide the same mouse bindings as other packages for experienced users.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Picking</b></dt>
  <dd class="It-tag">The RasMol <b>set picking</b> series of commands affects
      how a user may interact with a molecule displayed on the screen in RasMol.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Enabling/Disabling Atom Identification Picking:</b> Clicking on an atom
      with the mouse results in identification and the display of its residue
      name, residue number, atom name, atom serial number and chain in the
      command window. This behavior may be disabled with the command <b>set
      picking none</b> and restored with the command <b>set picking ident.</b>
      The command <b>set picking coord</b> adds the atomic coordinates of the
      atom to the display.
    <div style="height: 1.00em;">&#x00A0;</div>
    Disabling picking, by using <b>set picking off</b> is useful when executing
      the <b>pause</b> command in RasMol scripts as it prevents the display of
      spurious message on the command line while the script is suspended.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Measuring Distances, Angles and Torsions:</b> Interactive measurement of
      distances, angles and torsions is achieved using the commands: <b>set
      picking distance,</b> <b>set picking monitor,</b> <b>set picking angle</b>
      and <b>set picking torsion,</b> respectively. In these modes, clicking on
      an atom results in it being identified on the rasmol command line. In
      addition every atom picked increments a modulo counter such that in
      distance mode, every second atom displays the distance (or distance
      monitor) between this atom and the previous one. In angle mode, every
      third atom displays the angle between the previous three atoms and in
      torsion mode every fourth atom displays the torsion between the last four
      atoms. By holding down the shift key while picking an atom, this modulo
      counter is not incremented and allows, for example, the distances of
      consecutive atoms from a fixed atom to be displayed. See the
      <b>monitor</b> command for how to control the display of distance monitor
      lines and labels.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Labelling Atoms with the Mouse:</b> The mouse may also be used to toggle
      the display of an atom label on a given atom. The RasMol command <b>set
      picking label</b> removes a label from a picked atom if it already has one
      or displays a concise label at that atom position otherwise.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Centring Rotation with the Mouse:</b> A molecule may be centred on a
      specified atom position using the RasMol commands <b>set picking
      centre</b> or <b>set picking center.</b> In this mode, picking an atom
      causes all futher rotations to be about that point.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Picking a Bond as a Rotation Axis:</b> Any bond may be picked as an axis
      of rotation for the portion of the molecule beyond the second atom
      selected. This feature should be used with caution, since, naturally, it
      changes the conformation of the molecule. After executing <b>set picking
      bond</b> or using the equivalent &quot;Pick Bond&quot; in the
      &quot;Settings&quot; menu, a bond to be rotated is picked with the same
      sort of mouse clicks as are used for picking atoms for a distance
      measurement. Normally this should be done where a bond exists, but if no
      bond exists, it will be added. The bond cannot be used for rotation if it
      is part of a ring of any size. All bonds selected for rotation are
      remembered so that they can be properly reported when writing a script,
      but only the most recently selected bond may be actively rotated.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Enabling Atom/Group/Chain Selection Picking:</b> Atoms, groups and chains
      may be selected (as if with the <b>select</b> command), with the <b>set
      picking atom,</b> <b>set picking group,</b> <b>set picking chain</b>
      commands. For each of these commands, the shift key may be used to have a
      new selection added to the old, and the control key may be used to have a
      new selection deleted from the old. When the <b>set picking atom</b>
      command is given, the mouse can be used to pick or to drag a box around
      the atoms for which selection is desired. When the <b>set picking
      group</b> command is given, picking any an atom will cause selection of
      all atoms which agree in residue number with the picked atom, even if in
      different chains. When the <b>set picking chain</b> command is given,
      picking any atom will cause selection of all atoms which agree in chain
      identifier with the picked atom.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Play</b></dt>
  <dd class="It-tag">The RasMol <b>set play.fps</b> command gives the number of
      frames per second for playback by the <b>play</b> command (default 24
      frames per second).
    <div style="height: 1.00em;">&#x00A0;</div>
    In the current release of RasMol, the play timing is not controlled by this
      parameter.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Radius</b></dt>
  <dd class="It-tag">The RasMol <b>set radius</b> command is used to alter the
      behaviour of the RasMol <b>dots</b> command depending upon the value of
      the <b>solvent</b> parameter. When <b>solvent</b> is <b>true,</b> the
      <b>radius</b> parameter controls whether a true van der Waals' surface is
      generated by the <b>dots</b> command. If the value of <b>radius</b> is
      anything other than zero, that value is used as the radius of each atom
      instead of its true vdW value. When the value of <b>solvent</b> is
      <b>true,</b> this parameter determines the 'probe sphere' (solvent)
      radius. The parameter may be given as an integer in rasmol units or
      containing a decimal point in Angstroms. The default value of this
      parameter is determined by the value of <b>solvent</b> and changing
      <b>solvent</b> resets <b>radius</b> to its new default value.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Record</b></dt>
  <dd class="It-tag">The RasMol <b>set record.aps</b> gives the maximum
      on-screen velocity in Angstroms per second in animating translations,
      rotations and zooms (default 10 A/second).
    <div style="height: 1.00em;">&#x00A0;</div>
    The RasMol <b>set record.aps</b> command gives number of frames per second
      for recording by the <b>record</b> command (default 24 frames per second).
    <div style="height: 1.00em;">&#x00A0;</div>
    The RasMol <b>set record.dwell</b> command sets the time in seconds to dwell
      on a change in appearance (default .5 sec).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set ShadePower</b></dt>
  <dd class="It-tag">The <b>shadepower</b> parameter (adopted from RasTop)
      determines the shade repartition (the contrast) used in rendering solid
      objects. This value between 0 and 100 adjusts shading on an object surface
      oriented along the direction to the light source. Changing the shadepower
      parameter does not change the maximum or the minimum values of this
      shading, as does changing the <b>ambient</b> parameter. A value of 100
      concentrates the light on the top of spheres, giving a highly specular,
      glassy rendering (see the <b>specpower</b> parameter). A value of 0
      distributes the light on the entire object.
    <div style="height: 1.00em;">&#x00A0;</div>
    This implementation of shadepower differs from the one in RasTop only in the
      choice of range (0 to 100 versus -20 to 20 in RasTop).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Shadow</b></dt>
  <dd class="It-tag">The RasMol <b>set shadow</b> command enables and disables
      ray-tracing of the currently rendered image. Currently only the
      spacefilling representation is shadowed or can cast shadows. Enabling
      shadowing will automatically disable the Z-clipping (slabbing) plane using
      the command <b>slab off.</b> Ray-tracing typically takes about several
      seconds for a moderately sized protein. It is recommended that shadowing
      be normally disabled whilst the molecule is being transformed or
      manipulated, and only enabled once an appropiate viewpoint is selected, to
      provide a greater impression of depth.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set SlabMode</b></dt>
  <dd class="It-tag">The RasMol <b>slabmode</b> parameter controls the rendering
      method of objects cut by the slabbing (z-clipping) plane. Valid slabmode
      parameters are &quot;reject&quot;, &quot;half&quot;, &quot;hollow&quot;,
      &quot;solid&quot; and &quot;section&quot;.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Solvent</b></dt>
  <dd class="It-tag">The RasMol <b>set solvent</b> command is used to control
      the behaviour of the RasMol <b>dots</b> command. Depending upon the value
      of the <b>solvent</b> parameter, the <b>dots</b> command either generates
      a van der Waals' or a solvent accessible surface around the currently
      selected set of atoms. Changing this parameter automatically resets the
      value of the RasMol <b>radius</b> parameter. The command <b>set solvent
      false,</b> the default value, indicates that a van der Waals' surface
      should be generated and resets the value of <b>radius</b> to zero. The
      command <b>set solvent true</b> indicates that a 'Connolly' or 'Richards'
      solvent accessible surface should be drawn and sets the <b>radius</b>
      parameter, the solvent radius, to 1.2 Angstroms (or 300 RasMol units).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Specular</b></dt>
  <dd class="It-tag">The RasMol <b>set specular</b> command enables and disables
      the display of specular highlights on solid objects drawn by RasMol.
      Specular highlights appear as white reflections of the light source on the
      surface of the object. The current RasMol implementation uses an
      approximation function to generate this highlight.
    <div style="height: 1.00em;">&#x00A0;</div>
    The specular highlights on the surfaces of solid objects may be altered by
      using the specular reflection coefficient, which is altered using the
      RasMol <b>set specpower</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set SpecPower</b></dt>
  <dd class="It-tag">The <b>specpower</b> parameter determines the shininess of
      solid objects rendered by RasMol. This value between 0 and 100 adjusts the
      reflection coefficient used in specular highlight calculations. The
      specular highlights are enabled and disabled by the RasMol <b>set
      specular</b> command. Values around 20 or 30 produce plastic looking
      surfaces. High values represent more shiny surfaces such as metals, while
      lower values produce more diffuse/dull surfaces.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set SSBonds</b></dt>
  <dd class="It-tag">The RasMol <b>ssbonds</b> parameter determines whether
      disulphide bridges are drawn between the sulphur atoms in the sidechain
      (the default) or between the alpha carbon atoms in the backbone of the
      cysteines residues. The actual display of disulphide bridges is controlled
      by the <b>ssbonds</b> command. Drawing disulphide bridges between alpha
      carbons is useful when the rest of the protein is shown in only a
      schematic representation such as <b>backbone,</b> <b>ribbons</b> or
      <b>strands.</b> This parameter is similar to the RasMol <b>hbonds</b>
      parameter.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Stereo</b></dt>
  <dd class="It-tag">The RasMol <b>set stereo</b> parameter controls the
      separation between the left and right images. Turning stereo on and off
      doesn't reposition the centre of the molecule.
    <div style="height: 1.00em;">&#x00A0;</div>
    Stereo viewing of a molecule may be turned on (and off) either by selecting
      <b>Stereo</b> from the <b>Options</b> menu, or by typing the commands
      <b>stereo on</b> or <b>stereo off.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The separation angle between the two views may be adjusted with the <b>set
      stereo [-] &lt;number&gt;</b> command, where positive values result in
      crossed eye viewing and negative values in relaxed (wall-eyed) viewing.
      Currently, stereo viewing is not supported in <b>vector PostScript</b>
      output files.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Strands</b></dt>
  <dd class="It-tag">The RasMol <b>strands</b> parameter controls the number of
      parallel strands that are displayed in the ribbon representations of
      proteins. The permissible values for this parameter are 1, 2, 3, 4, 5 and
      9. The default value is 5. The number of strands is constant for all
      ribbons being displayed. However, the ribbon width (the separation between
      strands) may be controlled on a residue by residue basis using the RasMol
      <b>ribbons</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Transparent</b></dt>
  <dd class="It-tag">The RasMol <b>transparent</b> parameter controls the
      writing of transparent GIFs by the <b>write gif &lt;filename&gt;</b>
      command. This may be controlled by the <b>set transparent on</b> and
      <b>set transparent off</b> commands.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set UnitCell</b></dt>
  <dd class="It-tag">The RasMol <b>unitcell</b> parameter controls the display
      of the crystallographic unit cell on the current display. The crystal cell
      is only enabled if the appropriate crystal symmetry information is
      contained in the PDB, CIF or mmCIF data file. The RasMol command <b>show
      symmetry</b> display details of the crystal's space group and unit cell
      axes. The <b>set unitcell</b> command is similar to the commands <b>set
      axes</b> and <b>set boundbox</b> that display orthogonal coordinate axes
      and the bounding box, respectively.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set VectPS</b></dt>
  <dd class="It-tag">The RasMol <b>vectps</b> parameter is use to control the
      way in which the RasMol <b>write</b> command generates vector PostScript
      output files. The command <b>set vectps on</b> enables the use of black
      outlines around spheres and cylinder bonds producing
      &quot;cartoon-like&quot; high resolution output. However, the current
      implementation of RasMol incorrectly cartoons spheres that are intersected
      by more than one other sphere. Hence &quot;ball and stick&quot; models are
      rendered correctly but not large spacefilling spheres models. Cartoon
      outlines can be disabled, the default, by the command <b>set vectps
      off.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Write</b></dt>
  <dd class="It-tag">The RasMol <b>write</b> parameter controls the use of the
      <b>save</b> and <b>write</b> commands within scripts, but it may only be
      executed from the command line. By default, this value is <b>false,</b>
      prohibiting the generation of files in any scripts executed at start-up
      (such as those launched from a WWW browser). However, animators may start
      up RasMol interactively: type <b>set write on</b> and then execute a
      script to generate each frame using the source command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="ATOM_EXPRESSIONS"><a class="selflink" href="#ATOM_EXPRESSIONS">ATOM
  EXPRESSIONS</a></h1>
RasMol atom expressions uniquely identify an arbitrary group of atoms within a
  molecule. Atom expressions are composed of either primitive expressions,
  predefined sets, comparison operators, <b>within</b> expressions, or logical
  (boolean) combinations of the above expression types.
<div style="height: 1.00em;">&#x00A0;</div>
The logical operators allow complex queries to be constructed out of simpler
  ones using the standard boolean connectives <b>and,</b> <b>or</b> and
  <b>not.</b> These may be abbreviated by the symbols &quot;&amp;&quot;,
  &quot;|&quot; and &quot;!&quot;, respectively. Parentheses (brackets) may be
  used to alter the precedence of the operators. For convenience, a comma may
  also be used for boolean disjunction.
<div style="height: 1.00em;">&#x00A0;</div>
The atom expression is evaluated for each atom, hence <b>protein and
  backbone</b> selects protein backbone atoms, not the protein and [nucleic]
  acid backbone atoms!
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Primitive Expressions</b></dt>
  <dd class="It-tag">RasMol primitive expressions are the fundamental building
      blocks of atom expressions. There are two types of primitive expression.
      The first type is used to identify a given residue number or range of
      residue numbers. A single residue is identified by its number (position in
      the sequence), and a range is specified by lower and upper bounds
      separated by a hyphen character. For example <b>select 5,6,7,8</b> is also
      <b>select 5-8.</b> Note that this selects the given residue numbers in all
      macromolecule chains.
    <div style="height: 1.00em;">&#x00A0;</div>
    The second type of primitive expression specifies a sequence of fields that
      must match for a given atom. The first part specifies a residue (or group
      of residues) and an optional second part specifies the atoms within those
      residues. The first part consists of a residue name, optionally followed
      by a residue number and/or chain identifier.
    <div style="height: 1.00em;">&#x00A0;</div>
    The second part consists of a period character followed by an atom name. An
      atom name may be up to four alphabetic or numeric characters. An optional
      semicolon followed by an alternate conformation identifier may be
      appended. An optional slash followed by a model number may also be
      appended.
    <div style="height: 1.00em;">&#x00A0;</div>
    An asterisk may be used as a wild card for a whole field and a question mark
      as a single character wildcard.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Comparison Operators</b></dt>
  <dd class="It-tag">Parts of a molecule may also be distinguished using
      equality, inequality and ordering operators on their properties. The
      format of such comparison expression is a property name, followed by a
      comparison operator and then an integer value.
    <div style="height: 1.00em;">&#x00A0;</div>
    The atom properties that may be used in RasMol are <b>atomno</b> for the
      atom serial number, <b>elemno</b> for the atom's atomic number (element),
      <b>resno</b> for the residue number, <b>radius</b> for the spacefill
      radius in RasMol units (or zero if not represented as a sphere) and
      <b>temperature</b> for the PDB isotropic temperature value.
    <div style="height: 1.00em;">&#x00A0;</div>
    The equality operator is denoted either &quot;=&quot; or &quot;==&quot;. The
      inequality operator as either &quot;&lt;&gt;&quot;, &quot;!=&quot; or
      &quot;/=&quot;. The ordering operators are &quot;&lt;&quot; for less than,
      &quot;&lt;=&quot; for less than or equal to, &quot;&gt;&quot; for greater
      than, and &quot;&gt;&quot; for greater than or equal to.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Within Expressions</b></dt>
  <dd class="It-tag">A RasMol <b>within</b> expression allows atoms to be
      selected on their proximity to another set of atoms. A <b>within</b>
      expression takes two parameters separated by a comma and surrounded by
      parentheses. The first argument is an integer value called the
      &quot;cut-off&quot; distance of the within expression and the second
      argument is any valid atom expression. The cut-off distance is expressed
      in either integer RasMol units or Angstroms containing a decimal point. An
      atom is selected if it is within the cut-off distance of any of the atoms
      defined by the second argument. This allows complex expressions to be
      constructed containing nested <b>within</b> expressions.
    <div style="height: 1.00em;">&#x00A0;</div>
    For example, the command <b>select within(3.2,backbone)</b> selects any atom
      within a 3.2 Angstrom radius of any atom in a protein or nucleic acid
      backbone. <b>Within</b> expressions are particularly useful for selecting
      the atoms around an active site.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Predefined Sets</b></dt>
  <dd class="It-tag">RasMol atom expressions may contain predefined sets. These
      sets are single keywords that represent portions of a molecule of
      interest. Predefined sets are often abbreviations of primitive atom
      expressions. In some cases the use of predefined sets allows selection of
      areas of a molecule that could not otherwise be distinguished. A list of
      the currently predefined sets is given below. In addition to the sets
      listed here, RasMol also treats element names (and their plurals) as
      predefined sets containing all atoms of that element type, i.e. the
      command <b>select oxygen</b> is equivalent to the command <b>select
      elemno=8.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="Predefined_Sets"><a class="selflink" href="#Predefined_Sets">Predefined
  Sets</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>AT Set</b></dt>
  <dd class="It-tag">This set contains the atoms in the complementary
      nucleotides adenosine and thymidine (A and T, respectively). All
      nucleotides are classified as either the set <b>at</b> or the set
      <b>cg</b> This set is equivalent to the RasMol atom expressions
      <b>a,t,</b> and <b>nucleic and not cg.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Acidic Set</b></dt>
  <dd class="It-tag">The set of acidic amino acids. These are the residue types
      Asp and Glu. All amino acids are classified as either <b>acidic,</b>
      <b>basic</b> <b>or</b> <b>neutral.</b> This set is equivalent to the
      RasMol atom expressions <b>asp, glu</b> and <b>amino and not (basic or
      neutral).</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Acyclic Set</b></dt>
  <dd class="It-tag">The set of atoms in amino acids not containing a cycle or
      ring. All amino acids are classified as either <b>cyclic</b> or
      <b>acyclic.</b> This set is equivalent to the RasMol atom expression
      <b>amino and not cyclic.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Aliphatic Set</b></dt>
  <dd class="It-tag">This set contains the aliphatic amino acids. These are the
      amino acids Ala, Gly, Ile, Leu and Val. This set is equivalent to the
      RasMol atom expression <b>ala, gly, ile, leu, val.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Alpha Set</b></dt>
  <dd class="It-tag">The set of alpha carbons in the protein molecule. This set
      is approximately equivalent to the RasMol atom expression <b>*.CA.</b>
      This command should not be confused with the predefined set <b>helix</b>
      which contains the atoms in the amino acids of the protein's alpha
      helices.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Amino Set</b></dt>
  <dd class="It-tag">This set contains all the atoms contained in amino acid
      residues. This is useful for distinguishing the protein from the nucleic
      acid and heterogeneous atoms in the current molecule database.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Aromatic Set</b></dt>
  <dd class="It-tag">The set of atoms in amino acids containing aromatic rings.
      These are the amino acids His, Phe, Trp and Tyr. Because they contain
      aromatic rings all members of this set are member of the predefined set
      <b>cyclic.</b> This set is equivalent to the RasMol atom expressions
      <b>his, phe, trp, tyr</b> and <b>cyclic and not pro.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Backbone Set</b></dt>
  <dd class="It-tag">This set contains the four atoms of each amino acid that
      form the polypeptide N-C-C-O backbone of proteins, and the atoms of the
      sugar phosphate backbone of nucleic acids. Use the RasMol predefined sets
      <b>protein</b> and <b>nucleic</b> to distinguish between the two forms of
      backbone. Atoms in nucleic acids and proteins are either <b>backbone</b>
      or <b>sidechain.</b> This set is equivalent to the RasMol expression
      <b>(protein or nucleic) and not sidechain.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    The predefined set <b>mainchain</b> is synonymous with the set
      <b>backbone.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Basic Set</b></dt>
  <dd class="It-tag">The set of basic amino acids. These are the residue types
      Arg, His and Lys. All amino acids are classified as either <b>acidic,</b>
      <b>basic</b> or <b>neutral.</b> This set is equivalent to the RasMol atom
      expressions <b>arg, his, lys</b> and <b>amino and not (acidic or
      neutral).</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Bonded Set</b></dt>
  <dd class="It-tag">This set contain all the atoms in the current molecule
      database that are bonded to at least one other atom.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Buried Set</b></dt>
  <dd class="It-tag">This set contains the atoms in those amino acids that tend
      (prefer) to be buried inside protein, away from contact with solvent
      molecules. This set refers to the amino acids preference and not the
      actual solvent accessibility for the current protein. All amino acids are
      classified as either <b>surface</b> or <b>buried.</b> This set is
      equivalent to the RasMol atom expression <b>amino and not surface.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>CG Set</b></dt>
  <dd class="It-tag">This set contains the atoms in the complementary
      nucleotides cytidine and guanosine (C and G, respectively). All
      nucleotides are classified as either the set <b>at</b> or the set
      <b>cg</b> This set is equivalent to the RasMol atom expressions <b>c,g</b>
      and <b>nucleic and not at.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Charged Set</b></dt>
  <dd class="It-tag">This set contains the charged amino acids. These are the
      amino acids that are either <b>acidic</b> or <b>basic.</b> Amino acids are
      classified as being either <b>charged</b> or <b>neutral.</b> This set is
      equivalent to the RasMol atom expressions <b>acidic or basic</b> and
      <b>amino and not neutral.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Cyclic Set</b></dt>
  <dd class="It-tag">The set of atoms in amino acids containing a cycle or
      rings. All amino acids are classified as either <b>cyclic</b> or
      <b>acyclic.</b> This set consists of the amino acids His, Phe, Pro, Trp
      and Tyr. The members of the predefined set <b>aromatic</b> are members of
      this set. The only cyclic but non-aromatic amino acid is proline. This set
      is equivalent to the RasMol atom expressions <b>his, phe, pro, trp,
      tyr</b> and <b>aromatic or pro</b> and <b>amino and not acyclic.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Cystine Set</b></dt>
  <dd class="It-tag">This set contains the atoms of cysteine residues that form
      part of a disulphide bridge, i.e. half cystines. RasMol automatically
      determines disulphide bridges, if neither the predefined set
      <b>cystine</b> nor the RasMol <b>ssbonds</b> command have been used since
      the molecule was loaded. The set of free cysteines may be determined using
      the RasMol atom expression <b>cys and not cystine.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Helix Set</b></dt>
  <dd class="It-tag">This set contains all atoms that form part of a protein
      alpha helix as determined by either the PDB file author or Kabsch and
      Sander's DSSP algorithm. By default, RasMol uses the secondary structure
      determination given in the PDB file if it exists. Otherwise, it uses the
      DSSP algorithm as used by the RasMol <b>structure</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
    This predefined set should not be confused with the predefined set
      <b>alpha</b> which contains the alpha carbon atoms of a protein.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Hetero Set</b></dt>
  <dd class="It-tag">This set contains all the heterogeneous atoms in the
      molecule. These are the atoms described by HETATM entries in the PDB file.
      These typically contain water, cofactors and other solvents and ligands.
      All <b>hetero</b> atoms are classified as either <b>ligand</b> or
      <b>solvent</b> atoms. These heterogeneous <b>solvent</b> atoms are further
      classified as either <b>water</b> or <b>ions.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Hydrogen Set</b></dt>
  <dd class="It-tag">This predefined set contains all the hydrogen, deuterium
      and tritium atoms of the current molecule. This predefined set is
      equivalent to the RasMol atom expression <b>elemno=1.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Hydrophobic Set</b></dt>
  <dd class="It-tag">This set contains all the hydrophobic amino acids. These
      are the amino acids Ala, Leu, Val, Ile, Pro, Phe, Met and Trp. All amino
      acids are classified as either <b>hydrophobic</b> or <b>polar.</b> This
      set is equivalent to the RasMol atom expressions <b>ala, leu, val, ile,
      pro, phe, met, trp</b> and <b>amino and not polar.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Ions Set</b></dt>
  <dd class="It-tag">This set contains all the heterogeneous phosphate and
      sulphate ions in the current molecule data file. A large number of these
      ions are sometimes associated with protein and nucleic acid structures
      determined by X-ray crystallography. These atoms tend to clutter an image.
      All <b>hetero</b> atoms are classified as either <b>ligand</b> or
      <b>solvent</b> atoms. All <b>solvent</b> atoms are classified as either
      <b>water</b> or <b>ions.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Large Set</b></dt>
  <dd class="It-tag">All amino acids are classified as either <b>small,</b>
      <b>medium</b> or <b>large.</b> This set is equivalent to the RasMol atom
      expression <b>amino and not (small or medium).</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Ligand Set</b></dt>
  <dd class="It-tag">This set contains all the heterogeneous cofactor and ligand
      moieties that are contained in the current molecule data file. This set is
      defined to be all <b>hetero</b> atoms that are not <b>solvent</b> atoms.
      Hence this set is equivalent to the RasMol atom expression <b>hetero and
      not solvent.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Medium Set</b></dt>
  <dd class="It-tag">All amino acids are classified as either <b>small,</b>
      <b>medium</b> or <b>large.</b> This set is equivalent to the RasMol atom
      expression <b>amino and not (large or small).</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Neutral Set</b></dt>
  <dd class="It-tag">The set of neutral amino acids. All amino acids are
      classified as either <b>acidic,</b> <b>basic</b> or <b>neutral.</b> This
      set is equivalent to the RasMol atom expression <b>amino and not (acidic
      or basic).</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Nucleic Set</b></dt>
  <dd class="It-tag">The set of all atoms in nucleic acids, which consists of
      the four nucleotide bases adenosine, cytidine, guanosine and thymidine (A,
      C, G and T, respectively). All neucleotides are classified as either
      <b>purine</b> or <b>pyrimidine.</b> This set is equivalent to the RasMol
      atom expressions <b>a,c,g,t</b> and <b>purine or pyrimidine.</b> The
      symbols for RNA nucleotides (U, +U, I, 1MA, 5MC, OMC, 1MG, 2MG, M2G, 7MG,
      OMG, YG, H2U, 5MU, and PSU) are also recognized as members of this set.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Polar Set</b></dt>
  <dd class="It-tag">This set contains the polar amino acids. All amino acids
      are classified as either <b>hydrophobic</b> or <b>polar.</b> This set is
      equivalent to the RasMol atom expression <b>amino and not hydrophobic.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Protein Set</b></dt>
  <dd class="It-tag">The set of all atoms in proteins. This consists of the
      RasMol predefined set <b>amino</b> and common post-translation
      modifications.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Purine Set</b></dt>
  <dd class="It-tag">The set of purine nucleotides. These are the bases
      adenosine and guanosine (A and G, respectively). All nucleotides are
      either <b>purines</b> or <b>pyrimidines.</b> This set is equivalent to the
      RasMol atom expressions <b>a,g</b> and <b>nucleic and not pyrimidine.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Pyrimidine Set</b></dt>
  <dd class="It-tag">The set of pyrimidine nucleotides. These are the bases
      cytidine and thymidine (C and T, respectively). All nucleotides are either
      <b>purines</b> or <b>pyrimidines.</b> This set is equivalent to the RasMol
      atom expressions <b>c,t</b> and <b>nucleic and not purine.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Selected Set</b></dt>
  <dd class="It-tag">This set contains the set of atoms in the currently
      selected region. The currently selected region is defined by the preceding
      <b>select</b> or <b>restrict</b> command and not the atom expression
      containing the <b>selected</b> keyword.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Sheet Set</b></dt>
  <dd class="It-tag">This set contains all atoms that form part of a protein
      beta sheet as determined by either the PDB file author or Kabsch and
      Sander's DSSP algorithm. By default, RasMol uses the secondary structure
      determination given in the PDB file if it exists. Otherwise, it uses the
      DSSP algorithm as used by the RasMol <b>structure</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Sidechain Set</b></dt>
  <dd class="It-tag">This set contains the functional sidechains of any amino
      acids and the base of each nucleotide. These are the atoms not part of the
      polypeptide N-C-C-O backbone of proteins or the sugar phosphate backbone
      of nucleic acids. Use the RasMol predefined sets <b>protein</b> and
      <b>nucleic</b> to distinguish between the two forms of sidechain. Atoms in
      nucleic acids and proteins are either <b>backbone</b> or <b>sidechain.</b>
      This set is equivalent to the RasMol expression <b>(protein or nucleic)
      and not backbone.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Small Set</b></dt>
  <dd class="It-tag">All amino acids are classified as either <b>small,</b>
      <b>medium</b> or <b>large.</b> This set is equivalent to the RasMol atom
      expression <b>amino and not (medium or large).</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Solvent Set</b></dt>
  <dd class="It-tag">This set contains the solvent atoms in the molecule
      coordinate file. These are the heterogeneous water molecules, phosphate
      and sulphate ions. All <b>hetero</b> atoms are classified as either
      <b>ligand</b> or <b>solvent</b> atoms. All <b>solvent</b> atoms are
      classified as either <b>water</b> or <b>ions.</b> This set is equivalent
      to the RasMol atom expressions <b>hetero and not ligand</b> and <b>water
      or ions.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Surface Set</b></dt>
  <dd class="It-tag">This set contains the atoms in those amino acids that tend
      (prefer) to be on the surface of proteins, in contact with solvent
      molecules. This set refers to the amino acids preference and not the
      actual solvent accessibility for the current protein. All amino acids are
      classified as either <b>surface</b> or <b>buried.</b> This set is
      equivalent to the RasMol atom expression <b>amino and not buried.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Turn Set</b></dt>
  <dd class="It-tag">This set contains all atoms that form part of a protein
      turns as determined by either the PDB file author or Kabsch and Sander's
      DSSP algorithm. By default, RasMol uses the secondary structure
      determination given in the PDB file if it exists. Otherwise, it uses the
      DSSP algorithm as used by the RasMol <b>structure</b> command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Water Set</b></dt>
  <dd class="It-tag">This set contains all the heterogeneous water molecules in
      the current database. A large number of water molecules are sometimes
      associated with protein and nucleic acid structures determined by X-ray
      crystallography. These atoms tend to clutter an image. All <b>hetero</b>
      atoms are classified as either <b>ligand</b> or <b>solvent</b> atoms. The
      <b>solvent</b> atoms are further classified as either <b>water</b> or
      <b>ions.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Set Summary</b></dt>
  <dd class="It-tag">The table below summarises RasMol's classification of the
      common amino acids.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="COLOUR_SCHEMES"><a class="selflink" href="#COLOUR_SCHEMES">COLOUR
  SCHEMES</a></h1>
The RasMol <b>colour</b> command allows different objects (such as atoms, bonds
  and ribbon segments) to be given a specified colour. Typically this colour is
  either a RasMol predefined colour name or an RGB triple. Additionally RasMol
  also supports <b>alt,</b> <b>amino,</b> <b>chain,</b> <b>charge,</b>
  <b>cpk,</b> <b>group,</b> <b>model,</b> <b>shapely,</b> <b>structure,</b>
  <b>temperature</b> or <b>user</b> colour schemes for atoms, and <b>hbond
  type</b> colour scheme for hydrogen bonds and <b>electrostatic potential</b>
  colour scheme for dot surfaces. The 24 currently predefined colour names are
  Black, Blue, BlueTint, Brown, Cyan, Gold, Grey, Green, GreenBlue, GreenTint,
  HotPink, Magenta, Orange, Pink, PinkTint, Purple, Red, RedOrange, SeaGreen,
  SkyBlue, Violet, White, Yellow and YellowTint
<div style="height: 1.00em;">&#x00A0;</div>
If you frequently wish to use a colour not predefined, you can write a one-line
  script. For example, if you make the file <b>grey.col</b> containing the line,
  <b>colour [180,180,180] #grey,</b> then the command <b>script grey.col</b>
  colours the currently selected atom set grey.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Alt Colours</b></dt>
  <dd class="It-tag">The RasMol <b>alt</b> (Alternate Conformer) colour scheme
      codes the base structure with one colour and applies a limited number of
      colours to each alternate conformer. In a RasMol built for 8-bit colour
      systems, 4 colours are allowed for alternate conformers. Otherwise, 8
      colours are available.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Amino Colours</b></dt>
  <dd class="It-tag">The RasMol <b>amino</b> colour scheme colours amino acids
      according to traditional amino acid properties. The purpose of colouring
      is to identify amino acids in an unusual or surprising environment. The
      outer parts of a protein that are polar are visible (bright) colours and
      non-polar residues darker. Most colours are hallowed by tradition. This
      colour scheme is similar to the <b>shapely</b> scheme.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Chain Colours</b></dt>
  <dd class="It-tag">The RasMol <b>chain</b> colour scheme assigns each
      macromolecular chain a unique colour. This colour scheme is particularly
      useful for distinguishing the parts of multimeric structure or the
      individual 'strands' of a DNA chain. <b>Chain</b> can be selected from the
      RasMol <b>Colours</b> menu.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Charge Colours</b></dt>
  <dd class="It-tag">The RasMol <b>charge</b> colour scheme colour codes each
      atom according to the charge value stored in the input file (or beta
      factor field of PDB files). High values are coloured in blue (positive)
      and lower values coloured in red (negative). Rather than use a fixed scale
      this scheme determines the maximum and minimum values of the
      charge/temperature field and interpolates from red to blue appropriately.
      Hence, green cannot be assumed to be 'no net charge' charge.
    <div style="height: 1.00em;">&#x00A0;</div>
    The difference between the <b>charge</b> and <b>temperature</b> colour
      schemes is that increasing temperature values proceed from blue to red,
      whereas increasing charge values go from red to blue.
    <div style="height: 1.00em;">&#x00A0;</div>
    If the charge/temperature field stores reasonable values it is possible to
      use the RasMol <b>colour dots potential</b> command to colour code a dot
      surface (generated by the <b>dots</b> command) by electrostatic potential.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>CPK Colours</b></dt>
  <dd class="It-tag">The RasMol <b>cpk</b> colour scheme is based upon the
      colours of the popular plastic spacefilling models which were developed by
      Corey, Pauling and later improved by Kultun. This colour scheme colours
      'atom' objects by the atom (element) type. This is the scheme
      conventionally used by chemists. The assignment of the most commonly used
      element types to colours is given below.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Group Colours</b></dt>
  <dd class="It-tag">The RasMol <b>group</b> colour scheme colour codes residues
      by their position in a macromolecular chain. Each chain is drawn as a
      smooth spectrum from blue through green, yellow and orange to red. Hence
      the N terminus of proteins and 5' terminus of nucleic acids are coloured
      red and the C terminus of proteins and 3' terminus of nucleic acids are
      drawn in blue. If a chain has a large number of heterogeneous molecules
      associated with it, the macromolecule may not be drawn in the full 'range'
      of the spectrum. <b>Group</b> can be selected from the RasMol
      <b>Colours</b> menu.
    <div style="height: 1.00em;">&#x00A0;</div>
    If a chain has a large number of heterogeneous molecules associated with it,
      the macromolecule may not be drawn in the full range of the spectrum. When
      RasMol performs group colouring it decides the range of colours it uses
      from the residue numbering given in the PDB file. Hence the lowest residue
      number is displayed in blue and the highest residue number is displayed as
      red. Unfortunately, if a PDB file contains a large number of heteroatoms,
      such as water molecules, that occupy the high residue numbers, the protein
      is displayed in the blue-green end of the spectrum and the waters in the
      yellow-red end of the spectrum. This is aggravated by there typically
      being many more water molecules than amino acid residues. The solution to
      this problem is to use the command <b>set hetero off</b> before applying
      the group colour scheme. This can also be achieved by toggling <b>Hetero
      Atoms</b> on the <b>Options</b> menu before selecting <b>Group</b> on the
      <b>Colour</b> menu. This command instructs RasMol to only use non-hetero
      residues in the group colour scaling.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>NMR Model Colours</b></dt>
  <dd class="It-tag">The RasMol <b>model</b> colour scheme codes each NMR model
      with a distinct colour. The NMR model number is taken as a numeric value.
      High values are coloured in blue and lower values coloured in red. Rather
      than use a fixed scale this scheme determines the maximum value of the NMR
      model number and interpolates from red to blue appropriately.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Shapely Colours</b></dt>
  <dd class="It-tag">The RasMol <b>shapely</b> colour scheme colour codes
      residues by amino acid property. This scheme is based upon Bob
      Fletterick's &quot;Shapely Models&quot;. Each amino acid and nucleic acid
      residue is given a unique colour. The <b>shapely</b> colour scheme is used
      by David Bacon's Raster3D program. This colour scheme is similar to the
      <b>amino</b> colour scheme.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Structure Colours</b></dt>
  <dd class="It-tag">The RasMol <b>structure</b> colour scheme colours the
      molecule by protein secondary structure. Alpha helices are coloured
      magenta, [240,0,128], beta sheets are coloured yellow, [255,255,0], turns
      are coloured pale blue, [96,128,255] and all other residues are coloured
      white. The secondary structure is either read from the PDB file (HELIX,
      SHEET and TURN records), if available, or determined using Kabsch and
      Sander's DSSP algorithm. The RasMol <b>structure</b> command may be used
      to force DSSP's structure assignment to be used.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Temperature Colours</b></dt>
  <dd class="It-tag">The RasMol <b>temperature</b> colour scheme colour codes
      each atom according to the anisotropic temperature (beta) value stored in
      the PDB file. Typically this gives a measure of the mobility/uncertainty
      of a given atom's position. High values are coloured in warmer (red)
      colours and lower values in colder (blue) colours. This feature is often
      used to associate a &quot;scale&quot; value [such as amino acid
      variability in viral mutants] with each atom in a PDB file, and colour the
      molecule appropriately.
    <div style="height: 1.00em;">&#x00A0;</div>
    The difference between the <b>temperature</b> and <b>charge</b> colour
      schemes is that increasing temperature values proceed from blue to red,
      whereas increasing charge values go from red to blue.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>User Colours</b></dt>
  <dd class="It-tag">The RasMol <b>user</b> colour scheme allows RasMol to use
      the colour scheme stored in the PDB file. The colours for each atom are
      stored in COLO records placed in the PDB data file. This convention was
      introduced by David Bacon's Raster3D program.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>HBond Type Colours</b></dt>
  <dd class="It-tag">The RasMol <b>type</b> colour scheme applies only to
      hydrogen bonds, hence is used in the command <b>colour hbonds type.</b>
      This scheme colour codes each hydrogen bond according to the distance
      along a protein chain between hydrogen bond donor and acceptor. This
      schematic representation was introduced by Belhadj-Mostefa and
      Milner-White. This representation gives a good insight into protein
      secondary structure (hbonds forming alpha helices appear red, those
      forming sheets appear yellow and those forming turns appear magenta).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Potential Colours</b></dt>
  <dd class="It-tag">The RasMol <b>potential</b> colour scheme applies only to
      dot surfaces, hence is used in the command <b>colour dots potential.</b>
      This scheme colours each currently displayed dot by the electrostatic
      potential at that point in space. This potential is calculated using
      Coulomb's law taking the temperature/charge field of the input file to be
      the charge assocated with that atom. This is the same interpretation used
      by the <b>colour charge</b> command. Like the <b>charge</b> colour scheme
      low values are blue/white and high values are red.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Amino Acid Codes</b></dt>
  <dd class="It-tag">The following table lists the names, single letter and
      three letter codes of each of the amino acids.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Booleans</b></dt>
  <dd class="It-tag">A boolean parameter is a truth value. Valid boolean values
      are 'true' and used by RasMol to either enable or disable a representation
      or option.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="FILE_FORMATS"><a class="selflink" href="#FILE_FORMATS">FILE
  FORMATS</a></h1>
<b>Protein Data Bank Files</b>
<div style="height: 1.00em;">&#x00A0;</div>
If you do not have the PDB documentation, you may find the following summary of
  the PDB file format useful. The Protein Data Bank is a computer-based archival
  database for macromolecular structures. The database was established in 1971
  by Brookhaven National Laboratory, Upton, New York, as a public domain
  repository for resolved crystallographic structures. The Bank uses a uniform
  format to store atomic coordinates and partial bond connectivities as derived
  from crystallographic studies. In 1999 the Protein Data Bank moved to the
  Research Collaboratory for Structural Biology.
<div style="height: 1.00em;">&#x00A0;</div>
PDB file entries consist of records of 80 characters each. Using the punched
  card analogy, columns 1 to 6 contain a record-type identifier, the columns 7
  to 70 contain data. In older entries, columns 71 to 80 are normally blank, but
  may contain sequence information added by library management programs. In new
  entries conforming to the 1996 PDB format, there is other information in those
  columns. The first four characters of the record identifier are sufficient to
  identify the type of record uniquely, and the syntax of each record is
  independent of the order of records within any entry for a particular
  macromolecule.
<div style="height: 1.00em;">&#x00A0;</div>
The only record types that are of major interest to the RasMol program are the
  ATOM and HETATM records which describe the position of each atom. ATOM/HETATM
  records contain standard atom names and residue abbreviations, along with
  sequence identifiers, coordinates in Angstrom units, occupancies and thermal
  motion factors. The exact details are given below as a FORTRAN format
  statement. The &quot;fmt&quot; column indicates use of the field in all PDB
  formats, in the 1992 and earlier formats or in the 1996 and later formats.
<div style="height: 1.00em;">&#x00A0;</div>
Residues occur in order starting from the N-terminal residue for proteins and
  5'-terminus for nucleic acids. If the residue sequence is known, certain atom
  serial numbers may be omitted to allow for future insertion of any missing
  atoms. Within each residue, atoms are ordered in a standard manner, starting
  with the backbone (N-C-C-O for proteins) and proceeding in increasing
  remoteness from the alpha carbon, along the side chain.
<div style="height: 1.00em;">&#x00A0;</div>
HETATM records are used to define post-translational modifications and cofactors
  associated with the main molecule. TER records are interpreted as breaks in
  the main molecule's backbone.
<div style="height: 1.00em;">&#x00A0;</div>
If present, RasMol also inspects HEADER, COMPND, HELIX, SHEET, TURN, CONECT,
  CRYST1, SCALE, MODEL, ENDMDL, EXPDTA and END records. Information such as the
  name, database code, revision date and classification of the molecule are
  extracted from HEADER and COMPND records, initial secondary structure
  assignments are taken from HELIX, SHEET and TURN records, and the end of the
  file may be indicated by an END record.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>RasMol Interpretation of PDB fields</b></dt>
  <dd class="It-tag">Atoms located at 9999.000, 9999.000, 9999.000 are assumed
      to be Insight pseudo atoms and are ignored by RasMol. Atom names beginning
      ' Q' are also assumed to be pseudo atoms or position markers.
    <div style="height: 1.00em;">&#x00A0;</div>
    When a data file contains an NMR structure, multiple conformations may be
      placed in a single PDB file delimited by pairs of MODEL and ENDMDL
      records. RasMol displays all the NMR models contained in the file.
    <div style="height: 1.00em;">&#x00A0;</div>
    Residue names &quot;CSH&quot;, &quot;CYH&quot; and &quot;CSM&quot; are
      considered pseudonyms for cysteine &quot;CYS&quot;. Residue names
      &quot;WAT&quot;, &quot;H20&quot;, &quot;SOL&quot; and &quot;TIP&quot; are
      considered pseudonyms for water &quot;HOH&quot;. The residue name
      &quot;D20&quot; is consider heavy water &quot;DOD&quot;. The residue name
      &quot;SUL&quot; is considered a sulphate ion &quot;SO4&quot;. The residue
      name &quot;CPR&quot; is considered to be cis-proline and is translated as
      &quot;PRO&quot;. The residue name &quot;TRY&quot; is considered a
      pseudonym for tryptophan &quot;TRP&quot;.
    <div style="height: 1.00em;">&#x00A0;</div>
    RasMol uses the HETATM fields to define the sets hetero, water, solvent and
      ligand. Any group with the name &quot;HOH&quot;, &quot;DOD&quot;,
      &quot;SO4&quot; or &quot;PO4&quot; (or aliased to one of these names by
      the preceding rules) is considered a solvent and is considered to be
      defined by a HETATM field.
    <div style="height: 1.00em;">&#x00A0;</div>
    RasMol only respects CONECT connectivity records in PDB files containing
      fewer than 256 atoms. This is explained in more detail in the section on
      determining molecule connectivity. CONECT records that define a bond more
      than once are interpreted as specifying the bond order of that bond, i.e.
      a bond specified twice is a double bond and a bond specified three (or
      more) times is a triple bond. This is not a standard PDB feature.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>PDB Colour Scheme Specification</b></dt>
  <dd class="It-tag">RasMol also accepts the supplementary COLO record type in
      the PDB files. This record format was introduced by David Bacon's Raster3D
      program for specifying the colour scheme to be used when rendering the
      molecule. This extension is not currently supported by the PDB. The COLO
      record has the same basic record type as the ATOM and HETATM records
      described above.
    <div style="height: 1.00em;">&#x00A0;</div>
    Colours are assigned to atoms using a matching process. The Mask field is
      used in the matching process as follows. First RasMol reads in and
      remembers all the ATOM, HETATM and COLO records in input order. When the
      user-defined ('User') colour scheme is selected, RasMol goes through each
      remembered ATOM/HETATM record in turn, and searches for a COLO record that
      matches in all of columns 7 through 30. The first such COLO record to be
      found determines the colour and radius of the atom.
    <div style="height: 1.00em;">&#x00A0;</div>
    Note that the Red, Green and Blue components are in the same positions as
      the X, Y, and Z components of an ATOM or HETA record, and the van der
      Waals radius goes in the place of the Occupancy. The Red, Green and Blue
      components must all be in the range 0 to 1.
    <div style="height: 1.00em;">&#x00A0;</div>
    In order that one COLO record can provide colour and radius specifications
      for more than one atom (e.g. based on residue, atom type, or any other
      criterion for which labels can be given somewhere in columns 7 through
      30), a 'don't-care' character, the hash mark &quot;#&quot; (number or
      sharp sign) is used. This character, when found in a COLO record, matches
      any character in the corresponding column in a ATOM/HETATM record. All
      other characters must match identically to count as a match. As an
      extension to the specification, any atom that fails to match a COLO record
      is displayed in white.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Multiple NMR Models</b></dt>
  <dd class="It-tag">RasMol loads all of the NMR models from a PDB file no
      matter which command is used: <b>load pdb &lt;filename&gt;</b> or <b>load
      nmrpdb &lt;filename&gt;</b>
    <div style="height: 1.00em;">&#x00A0;</div>
    Once multiple NMR conformations have been loaded they may be manipulated
      with the atom expression extensions described in <b>Primitive
      Expressions.</b> In particular, the command <b>restrict */1</b> will
      restrict the display to the first model only.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>CIF and mmCIF Format Files</b></dt>
  <dd class="It-tag">CIF is the IUCr standard for presentation of small
      molecules and mmCIF is intended as the replacement for the fixed-field PDB
      format for presentation of macromolecular structures. RasMol can accept
      data sets in either format.
    <div style="height: 1.00em;">&#x00A0;</div>
    There are many useful sites on the World Wide Web where information tools
      and software related to CIF, mmCIF and the PDB can be found. The following
      are good starting points for exploration:
    <div style="height: 1.00em;">&#x00A0;</div>
    The International Union of Crystallography (IUCr) provides access to
      software, dictionaries, policy statements and documentation relating to
      CIF and mmCIF at: IUCr, Chester, England (www.iucr.org/iucr-top/cif/) with
      many mirror sites.
    <div style="height: 1.00em;">&#x00A0;</div>
    The Nucleic Acid Database Project provides access to its entries, software
      and documentation, with an mmCIF page giving access to the dictionary and
      mmCIF software tools at Rutgers University, New Jersey, USA
      (http://ndbserver.rutgers.edu/NDB/mmcif) with many mirror sites.
    <div style="height: 1.00em;">&#x00A0;</div>
    This version of RasMol restricts CIF or mmCIF tag values to essentially the
      same conventions as are used for the fixed-field PDB format. Thus chain
      identifiers and alternate conformation identifiers are limited to a single
      character, atom names are limited to 4 characters, etc. RasMol interprets
      the following CIF and mmCIF tags: A search is made through multiple data
      blocks for the desired tags, so a single dataset may be composed from
      multiple data blocks, but multiple data sets may not be stacked in the
      same file.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="MACHINE-SPECIFIC_SUPPORT"><a class="selflink" href="#MACHINE-SPECIFIC_SUPPORT">MACHINE-SPECIFIC
  SUPPORT</a></h1>
In the following sections, support for <b>Monochrome X-Windows,</b> <b>Tcl/Tk
  IPC,</b> <b>UNIX sockets based IPC,</b> <b>Compiling RasWin with Borland and
  MetroWerks</b> are described.
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Monochrome X-Windows Support</b></dt>
  <dd class="It-tag">RasMol supports the many monochrome UNIX workstations
      typically found in academia, such as low-end SUN workstations and NCD
      X-terminals. The X11 version of RasMol (when compiled in 8 bit mode) now
      detects black and white X-Windows displays and enables dithering
      automatically. The use of run-time error diffusion dithering means that
      all display modes of RasMol are available when in monochrome mode. For
      best results, users should experiment with the set ambient command to
      ensure the maximum contrast in resulting images.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Tcl/Tk IPC support</b></dt>
  <dd class="It-tag">Version 4 of Tk graphics library changed the protocol used
      to communicate between Tk applications. RasMol version 2.6 was modified
      such that it could communicate with both this new protocol and the
      previous version 3 protocol supported by RasMol v2.5. Although Tcl/Tk 3.x
      applications may only communicate with other 3.x applications and Tcl/Tk
      4.x applications with other 4.x applications, these changes allow RasMol
      to communicate between processes with both protocols (potentially
      concurrently).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>UNIX sockets based IPC</b></dt>
  <dd class="It-tag">The UNIX implementation of RasMol supports BSD-style socket
      communication. An identical socket mechanism is also being developed for
      VMS, Apple Macintosh and Microsoft Windows systems. This should allow
      RasMol to interactively display results of a computation on a remote host.
      The current protocol acts as a TCP/IP server on port 21069 that executes
      command lines until either the command <b>exit</b> or the command
      <b>quit</b> is typed. The command exit from the RasMol server, the command
      <b>quit</b> both disconnects the current session and terminates RasMol.
      This functionality may be tested using the UNIX command <b>telnet
      &lt;hostname&gt; 21069.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Compiling RasWin with Borland and MetroWerks</b></dt>
  <dd class="It-tag">A number of changes were made to the source code in the
      transition from version 2.5 to 2.6 to allow the Microsoft Windows version
      of RasMol to compile using the Borland C/C++ compiler. These fixes include
      name changes for the standard library and special code to avoid a bug in
      _fmemset. Additional changes were made in the transition from 2.6 to 2.7
      to allow compilation with the MetroWerks compilers.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="BIBLIOGRAPHY"><a class="selflink" href="#BIBLIOGRAPHY">BIBLIOGRAPHY</a></h1>
<b>Molecular Graphics</b>
<div style="height: 1.00em;">&#x00A0;</div>
[1] Nelson Max, &quot;Computer Representation of Molecular Surfaces&quot;, IEEE
  Computer Graphics and Applications, pp.21-29, August 1983.
<div style="height: 1.00em;">&#x00A0;</div>
[2] Arthur M. Lesk, &quot;Protein Architecture: A Practical Approach&quot;, IRL
  Press Publishers, 1991.
<div style="height: 1.00em;">&#x00A0;</div>
<b>Molecular Graphics Programs</b>
<div style="height: 1.00em;">&#x00A0;</div>
[3] Per J. Kraulis, &quot;MOLSCRIPT: A Program to Produce both Detailed and
  Schematic Plots of Protein Structures&quot;, Journal of Applied
  Crystallography, Vol.24, pp.946-950, 1991.
<div style="height: 1.00em;">&#x00A0;</div>
[4] David Bacon and Wayne F. Anderson, &quot;A Fast Algorithm for Rendering
  Space-Filling Molecule Pictures&quot;, Journal of Molecular Graphics, Vol.6,
  No.4, pp.219-220, December 1988.
<div style="height: 1.00em;">&#x00A0;</div>
[5] David C. Richardson and Jane S. Richardson, &quot;The Kinemage: A tool for
  Scientific Communication&quot;, Protein Science, Vol.1, No.1,pp.3-9, January
  1992.
<div style="height: 1.00em;">&#x00A0;</div>
[6] Mike Carson, &quot;RIBBONS 2.0&quot;, Journal of Applied Crystallography,
  Vol.24, pp.958-961, 1991.
<div style="height: 1.00em;">&#x00A0;</div>
[7] Conrad C. Huang, Eric F. Pettersen, Teri E. Klein, Thomas E. Ferrin and
  Robert Langridge, &quot;Conic: A Fast Renderer for Space-Filling Molecules
  with Shadows&quot;, Journal of Molecular Graphics, Vol.9, No.4, pp.230-236,
  December 1991.
<div style="height: 1.00em;">&#x00A0;</div>
<b>Molecular Biology Algorithms</b>
<div style="height: 1.00em;">&#x00A0;</div>
[8] Wolfgang Kabsch and Christian Sander, &quot;Dictionary of Protein Secondary
  Structure: Pattern Recognition of Hydrogen-Bonded and Geometrical
  Features&quot;, Biopolymers, Vol.22, pp.2577-2637, 1983.
<div style="height: 1.00em;">&#x00A0;</div>
[9] Michael L. Connolly, &quot;Solvent-Accessible Surfaces of Proteins and
  Nucleic Acids&quot;, Science, Vol.221, No.4612, pp.709-713, August 1983.
<div style="height: 1.00em;">&#x00A0;</div>
[10] Khaled Belhadj-Mostefa, Ron Poet and E. James Milner-White,
  &quot;Displaying Inter-Main Chain Hydrogen Bond Patterns in Proteins&quot;,
  Journal of Molecular Graphics, Vol.9, No.3, pp.194-197, September 1991.
<div style="height: 1.00em;">&#x00A0;</div>
[11] Mike Carson, &quot;Ribbon Models of Macromolecules&quot;, Journal of
  Molecular Graphics, Vol.5, No.2, pp.103-106, June 1987.
<div style="height: 1.00em;">&#x00A0;</div>
[12] Mike Carson and Charles E. Bugg, &quot;Algorithm for Ribbon Models of
  Proteins&quot;, Journal of Molecular Graphics, Vol.4, No.2, pp.121-122, June
  1986.
<div style="height: 1.00em;">&#x00A0;</div>
[13] H. Iijima, J. B. Dunbar Jr. and G. Marshall, &quot;Calibration of Effective
  van der Waals Atomic Contact Radii for Proteins and Peptides&quot;, Proteins:
  Structure, Functions and Genetics, Vol.2, pp.330-339,1987.
<div style="height: 1.00em;">&#x00A0;</div>
<b>Graphics Algorithms</b>
<div style="height: 1.00em;">&#x00A0;</div>
[14] J. Foley, A. van Dam, S. Feiner and J. Hughes, &quot;Computer Graphics:
  Principles and Practice&quot;, 2nd Edition, Addison Wesley Publishers, 1990.
<div style="height: 1.00em;">&#x00A0;</div>
[15] J. Cleary and G. Wyvill, &quot;Analysis of an Algorithm for Fast Ray
  Tracing using Uniform Space Subdivision&quot;, The Visual Computer, Vol.4,
  pp.65-83, 1988.
<div style="height: 1.00em;">&#x00A0;</div>
[16] Thomas Porter,&quot;Spherical Shading&quot;, Computer Graphics Vol.12, ACM
  SIGGRAPH, pp.282-285, 1978.
<div style="height: 1.00em;">&#x00A0;</div>
[17] Jean-Michel Cense, &quot;Exact Visibility Calculation for Space-Filling
  Molecular Models&quot;, Journal of Molecular Graphics, Vol.9, No.3,
  pp.191-193, September 1991.
<div style="height: 1.00em;">&#x00A0;</div>
[18] Chris Schafmeister, &quot;Fast Algorithm for Generating CPK Images on
  Graphics Workstations&quot;, Journal of Molecular Graphics, Vol.8, No.4,
  pp.201-206, December 1990.
<div style="height: 1.00em;">&#x00A0;</div>
[19] Bruce A. Johnson, &quot;MSURF: A Rapid and General Program for the
  Representation of Molecular Surfaces&quot;, Journal of Molecular Graphics,
  Vol.5, No.3, pp.167-169, September 1987.
<div style="height: 1.00em;">&#x00A0;</div>
<b>File Formats</b>
<div style="height: 1.00em;">&#x00A0;</div>
[20] Frances C. Bernstein et al., &quot;The Protein Data Bank: A Computer-Based
  Archival File for Macromolecular Structures&quot;, Journal of Molecular
  Biology, Vol.112, pp.535-542, 1977.
<div style="height: 1.00em;">&#x00A0;</div>
[21] Arthur Dalby, James G. Nourse, W. Douglas Hounshell, Ann K. I. Gushurst,
  David L. Grier, Burton A. Leland and John Laufer, &quot;Description of Several
  Chemical File Formats Used by Computer Programs Developed at Molecular Design
  Limited&quot;, Journal of Chemical Information and Computer Sciences, Vol.32,
  No.3, pp.244-255, 1992.
<div style="height: 1.00em;">&#x00A0;</div>
[22] Adobe Systems Inc., &quot;PostScript Language Reference Manual&quot;,
  Addison-Wesley Publishers, Reading, Mass., 1985.
<div style="height: 1.00em;">&#x00A0;</div>
[23] Philip E. Bourne et al., &quot;The Macromolecular Crystallographic
  Information File (mmCIF)&quot;, Meth. Enzymol. (1997) 277, 571-590.
<div style="height: 1.00em;">&#x00A0;</div>
[24] Sydney R. Hall, &quot;The STAR File: a New Format for Electronic Data
  Transfer and Archiving&quot;, Journal of Chemical Information and Computer
  Sciences, Vol. 31, 326-333, 1991.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The RasMol User Manual!
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
1992-1998 by Roger Sayle (rasmol@ggr.co.uk)</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2009</td>
    <td class="foot-os"></td>
  </tr>
</table>
</body>
</html>
