<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>RAY(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">RAY(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">RAY(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
Ray - assemble genomes in parallel using the message-passing interface
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<br/>
 mpiexec -n NUMBER_OF_RANKS Ray -k KMERLENGTH -p l1_1.fastq l1_2.fastq -p
  l2_1.fastq l2_2.fastq -o test
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 mpiexec -n NUMBER_OF_RANKS Ray Ray.conf # with commands in a file
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<br/>
 The Ray genome assembler is built on top of the RayPlatform, a generic
  plugin-based
<br/>
 distributed and parallel compute engine that uses the message-passing interface
<br/>
 for passing messages.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Ray targets several applications:
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 - de novo genome assembly (with Ray vanilla)
<br/>
 - de novo meta-genome assembly (with Ray M&#x00C3;&#x00A9;ta)
<br/>
 - de novo transcriptome assembly (works, but not tested a lot)
<br/>
 - quantification of contig abundances
<br/>
 - quantification of microbiome consortia members (with Ray Communities)
<br/>
 - quantification of transcript expression
<br/>
 - taxonomy profiling of samples (with Ray Communities)
<br/>
 - gene ontology profiling of samples (with Ray Ontologies)
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<br/>
 -help
<br/>
 Displays this help page.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -version
<br/>
 Displays Ray version and compilation options.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Using a configuration file
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Ray can be launched with
<br/>
 mpiexec -n 16 Ray Ray.conf
<br/>
 The configuration file can include comments (starting with #).
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 K-mer length
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -k kmerLength
<br/>
 Selects the length of k-mers. The default value is 21.
<br/>
 It must be odd because reverse-complement vertices are stored together.
<br/>
 The maximum length is defined at compilation by MAXKMERLENGTH
<br/>
 Larger k-mers utilise more memory.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Inputs
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -p leftSequenceFile rightSequenceFile [averageOuterDistance standardDeviation]
<br/>
 Provides two files containing paired-end reads.
<br/>
 averageOuterDistance and standardDeviation are automatically computed if not
  provided.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -i interleavedSequenceFile [averageOuterDistance standardDeviation]
<br/>
 Provides one file containing interleaved paired-end reads.
<br/>
 averageOuterDistance and standardDeviation are automatically computed if not
  provided.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -s sequenceFile
<br/>
 Provides a file containing single-end reads.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Outputs
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -o outputDirectory
<br/>
 Specifies the directory for outputted files. Default is RayOutput
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Assembly options (defaults work well)
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -disable-recycling
<br/>
 Disables read recycling during the assembly
<br/>
 reads will be set free in 3 cases:
<br/>
 1. the distance did not match for a pair
<br/>
 2. the read has not met its mate
<br/>
 3. the library population indicates a wrong placement
<br/>
 see Constrained traversal of repeats with paired sequences.
<br/>
 S&#x00E9;bastien Boisvert, &#x00C9;l&#x00E9;nie Godzaridis, Fran&#x00E7;ois
  Laviolette &amp; Jacques Corbeil.
<br/>
 First Annual RECOMB Satellite Workshop on Massively Parallel Sequencing, March
  26-27 2011, Vancouver, BC, Canada.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -disable-scaffolder
<br/>
 Disables the scaffolder.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -minimum-contig-length minimumContigLength
<br/>
 Changes the minimum contig length, default is 100 nucleotides
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -color-space
<br/>
 Runs in color-space
<br/>
 Needs csfasta files. Activated automatically if csfasta files are provided.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -use-maximum-seed-coverage maximumSeedCoverageDepth
<br/>
 Ignores any seed with a coverage depth above this threshold.
<br/>
 The default is 4294967295.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -use-minimum-seed-coverage minimumSeedCoverageDepth
<br/>
 Sets the minimum seed coverage depth.
<br/>
 Any path with a coverage depth lower than this will be discarded. The default
  is 0.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Distributed storage engine (all these values are for each MPI rank)
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -bloom-filter-bits bits
<br/>
 Sets the number of bits for the Bloom filter
<br/>
 Default is 268435456 bits, 0 bits disables the Bloom filter.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -hash-table-buckets buckets
<br/>
 Sets the initial number of buckets. Must be a power of 2 !
<br/>
 Default value: 268435456
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -hash-table-buckets-per-group buckets
<br/>
 Sets the number of buckets per group for sparse storage
<br/>
 Default value: 64, Must be between &gt;=1 and &lt;= 64
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -hash-table-load-factor-threshold threshold
<br/>
 Sets the load factor threshold for real-time resizing
<br/>
 Default value: 0.75, must be &gt;= 0.5 and &lt; 1
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -hash-table-verbosity
<br/>
 Activates verbosity for the distributed storage engine
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Biological abundances
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -search searchDirectory
<br/>
 Provides a directory containing fasta files to be searched in the de Bruijn
  graph.
<br/>
 Biological abundances will be written to RayOutput/BiologicalAbundances
<br/>
 See Documentation/BiologicalAbundances.txt
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -one-color-per-file
<br/>
 Sets one color per file instead of one per sequence.
<br/>
 By default, each sequence in each file has a different color.
<br/>
 For files with large numbers of sequences, using one single color per file may
  be more efficient.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Taxonomic profiling with colored de Bruijn graphs
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -with-taxonomy Genome-to-Taxon.tsv TreeOfLife-Edges.tsv Taxon-Names.tsv
<br/>
 Provides a taxonomy.
<br/>
 Computes and writes detailed taxonomic profiles.
<br/>
 See Documentation/Taxonomy.txt for details.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -gene-ontology OntologyTerms.txt Annotations.txt
<br/>
 Provides an ontology and annotations.
<br/>
 OntologyTerms.txt is fetched from http://geneontology.org
<br/>
 Annotations.txt is a 2-column file (EMBL_CDS handle	&amp;	gene ontology
  identifier)
<br/>
 See Documentation/GeneOntology.txt
<br/>
 Other outputs
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -enable-neighbourhoods
<br/>
 Computes contig neighborhoods in the de Bruijn graph
<br/>
 Output file: RayOutput/NeighbourhoodRelations.txt
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -amos
<br/>
 Writes the AMOS file called RayOutput/AMOS.afg
<br/>
 An AMOS file contains read positions on contigs.
<br/>
 Can be opened with software with graphical user interface.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -write-kmers
<br/>
 Writes k-mer graph to RayOutput/kmers.txt
<br/>
 The resulting file is not utilised by Ray.
<br/>
 The resulting file is very large.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -write-read-markers
<br/>
 Writes read markers to disk.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -write-seeds
<br/>
 Writes seed DNA sequences to RayOutput/Rank&lt;rank&gt;.RaySeeds.fasta
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -write-extensions
<br/>
 Writes extension DNA sequences to
  RayOutput/Rank&lt;rank&gt;.RayExtensions.fasta
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -write-contig-paths
<br/>
 Writes contig paths with coverage values
<br/>
 to RayOutput/Rank&lt;rank&gt;.RayContigPaths.txt
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -write-marker-summary
<br/>
 Writes marker statistics.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Memory usage
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -show-memory-usage
<br/>
 Shows memory usage. Data is fetched from /proc on GNU/Linux
<br/>
 Needs __linux__
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -show-memory-allocations
<br/>
 Shows memory allocation events
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Algorithm verbosity
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -show-extension-choice
<br/>
 Shows the choice made (with other choices) during the extension.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -show-ending-context
<br/>
 Shows the ending context of each extension.
<br/>
 Shows the children of the vertex where extension was too difficult.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -show-distance-summary
<br/>
 Shows summary of outer distances used for an extension path.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -show-consensus
<br/>
 Shows the consensus when a choice is done.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Checkpointing
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -write-checkpoints checkpointDirectory
<br/>
 Write checkpoint files
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -read-checkpoints checkpointDirectory
<br/>
 Read checkpoint files
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -read-write-checkpoints checkpointDirectory
<br/>
 Read and write checkpoint files
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Message routing for large number of cores
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -route-messages
<br/>
 Enables the Ray message router. Disabled by default.
<br/>
 Messages will be routed accordingly so that any rank can communicate directly
  with only a few others.
<br/>
 Without -route-messages, any rank can communicate directly with any other rank.
<br/>
 Files generated: Routing/Connections.txt, Routing/Routes.txt and
  Routing/RelayEvents.txt
<br/>
 and Routing/Summary.txt
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -connection-type type
<br/>
 Sets the connection type for routes.
<br/>
 Accepted values are debruijn, hypercube, polytope, group, random, kautz and
  complete. Default is debruijn.
<br/>
 debruijn: a full de Bruijn graph a given alphabet and diameter
<br/>
 hypercube: a hypercube, alphabet is {0,1} and the vertices is a power of 2
<br/>
 polytope: a convex regular polytope, alphabet is {0,1,...,B-1} and the vertices
  is a power of B
<br/>
 group: silly model where one representative per group can communicate with
  outsiders
<br/>
 random: Erd&#x0151;s-R&#x00E9;nyi model
<br/>
 kautz: a full de Kautz graph, which is a subgraph of a de Bruijn graph
<br/>
 complete: a full graph with all the possible connections
<br/>
 With the type debruijn, the number of ranks must be a power of something.
<br/>
 Examples: 256 = 16*16, 512=8*8*8, 49=7*7, and so on.
<br/>
 Otherwise, don't use debruijn routing but use another one
<br/>
 With the type kautz, the number of ranks n must be n=(k+1)*k^(d-1) for some k
  and d
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -routing-graph-degree degree
<br/>
 Specifies the outgoing degree for the routing graph.
<br/>
 See Documentation/Routing.txt
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Hardware testing
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -test-network-only
<br/>
 Tests the network and returns.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -write-network-test-raw-data
<br/>
 Writes one additional file per rank detailing the network test.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -exchanges NumberOfExchanges
<br/>
 Sets the number of exchanges
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -disable-network-test
<br/>
 Skips the network test.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Debugging
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -verify-message-integrity
<br/>
 Checks message data reliability for any non-empty message.
<br/>
 add '-D CONFIG_SSE_4_2' in the Makefile to use hardware instruction (SSE 4.2)
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -run-profiler
<br/>
 Runs the profiler as the code runs. By default, only show granularity warnings.
<br/>
 Running the profiler increases running times.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -with-profiler-details
<br/>
 Shows number of messages sent and received in each methods during in each time
  slices (epochs). Needs -run-profiler.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -show-communication-events
<br/>
 Shows all messages sent and received.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -show-read-placement
<br/>
 Shows read placement in the graph during the extension.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -debug-bubbles
<br/>
 Debugs bubble code.
<br/>
 Bubbles can be due to heterozygous sites or sequencing errors or other
  (unknown) events
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -debug-seeds
<br/>
 Debugs seed code.
<br/>
 Seeds are paths in the graph that are likely unique.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -debug-fusions
<br/>
 Debugs fusion code.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 -debug-scaffolder
<br/>
 Debug the scaffolder.
<h1 class="Sh" title="Sh" id="FILES"><a class="selflink" href="#FILES">FILES</a></h1>
<br/>
 Input files
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Note: file format is determined with file extension.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 .fasta
<br/>
 .fasta.gz (needs HAVE_LIBZ=y at compilation)
<br/>
 .fasta.bz2 (needs HAVE_LIBBZ2=y at compilation)
<br/>
 .fastq
<br/>
 .fastq.gz (needs HAVE_LIBZ=y at compilation)
<br/>
 .fastq.bz2 (needs HAVE_LIBBZ2=y at compilation)
<br/>
 .sff (paired reads must be extracted manually)
<br/>
 .csfasta (color-space reads)
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Outputted files
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Scaffolds
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/Scaffolds.fasta
<br/>
 	The scaffold sequences in FASTA format
<br/>
 RayOutput/ScaffoldComponents.txt
<br/>
 	The components of each scaffold
<br/>
 RayOutput/ScaffoldLengths.txt
<br/>
 	The length of each scaffold
<br/>
 RayOutput/ScaffoldLinks.txt
<br/>
 	Scaffold links
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Contigs
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/Contigs.fasta
<br/>
 	Contiguous sequences in FASTA format
<br/>
 RayOutput/ContigLengths.txt
<br/>
 	The lengths of contiguous sequences
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Summary
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/OutputNumbers.txt
<br/>
 	Overall numbers for the assembly
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 de Bruijn graph
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/CoverageDistribution.txt
<br/>
 	The distribution of coverage values
<br/>
 RayOutput/CoverageDistributionAnalysis.txt
<br/>
 	Analysis of the coverage distribution
<br/>
 RayOutput/degreeDistribution.txt
<br/>
 	Distribution of ingoing and outgoing degrees
<br/>
 RayOutput/kmers.txt
<br/>
 	k-mer graph, required option: -write-kmers
<br/>
 The resulting file is not utilised by Ray.
<br/>
 The resulting file is very large.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Assembly steps
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/SeedLengthDistribution.txt
<br/>
 Distribution of seed length
<br/>
 RayOutput/Rank&lt;rank&gt;.OptimalReadMarkers.txt
<br/>
 Read markers.
<br/>
 RayOutput/Rank&lt;rank&gt;.RaySeeds.fasta
<br/>
 Seed DNA sequences, required option: -write-seeds
<br/>
 RayOutput/Rank&lt;rank&gt;.RayExtensions.fasta
<br/>
 Extension DNA sequences, required option: -write-extensions
<br/>
 RayOutput/Rank&lt;rank&gt;.RayContigPaths.txt
<br/>
 Contig paths with coverage values, required option: -write-contig-paths
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Paired reads
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/LibraryStatistics.txt
<br/>
 	Estimation of outer distances for paired reads
<br/>
 RayOutput/Library&lt;LibraryNumber&gt;.txt
<br/>
 Frequencies for observed outer distances (insert size + read lengths)
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Partition
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/NumberOfSequences.txt
<br/>
 Number of reads in each file
<br/>
 RayOutput/SequencePartition.txt
<br/>
 	Sequence partition
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Ray software
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/RayVersion.txt
<br/>
 	The version of Ray
<br/>
 RayOutput/RayCommand.txt
<br/>
 	The exact same command provided
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 AMOS
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/AMOS.afg
<br/>
 	Assembly representation in AMOS format, required option: -amos
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 Communication
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 RayOutput/MessagePassingInterface.txt 	 	Number of messages sent
<br/>
 RayOutput/NetworkTest.txt 	 	Latencies in microseconds
<br/>
 RayOutput/Rank&lt;rank&gt;NetworkTestData.txt 	 	Network test raw data
<h1 class="Sh" title="Sh" id="DOCUMENTATION"><a class="selflink" href="#DOCUMENTATION">DOCUMENTATION</a></h1>
<br/>
 - mpiexec -n 1 Ray -help|less (always up-to-date)
<br/>
 - This help page (always up-to-date)
<br/>
 - The directory Documentation/
<br/>
 - Manual (Portable Document Format): InstructionManual.tex (in Documentation)
<br/>
 - Mailing list archives:
  http://sourceforge.net/mailarchive/forum.php?forum_name=denovoassembler-users
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<br/>
 Written by S&#x00E9;bastien Boisvert.
<h1 class="Sh" title="Sh" id="REPORTING_BUGS"><a class="selflink" href="#REPORTING_BUGS">REPORTING
  BUGS</a></h1>
<br/>
 Report bugs to denovoassembler-users@lists.sourceforge.net
<br/>
 Home page: &lt;http://denovoassembler.sourceforge.net/&gt;
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
<br/>
 This program is free software: you can redistribute it and/or modify
<br/>
 it under the terms of the GNU General Public License as published by
<br/>
 the Free Software Foundation, version 3 of the License.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 This program is distributed in the hope that it will be useful,
<br/>
 but WITHOUT ANY WARRANTY; without even the implied warranty of
<br/>
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
<br/>
 GNU General Public License for more details.
<div style="height: 1.00em;">&#x00A0;</div>
<br/>
 You have received a copy of the GNU General Public License
<br/>
 along with this program (see LICENSE).
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">November 2012</td>
    <td class="foot-os">Ray 2.1.0</td>
  </tr>
</table>
</body>
</html>
