<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:35:03 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>RAY(1) User Commands RAY(1)</p>

<p style="margin-top: 1em">NAME <br>
Ray - assemble genomes in parallel using the message-passing
interface</p>

<p style="margin-top: 1em">SYNOPSIS <br>
mpiexec -n NUMBER_OF_RANKS Ray -k KMERLENGTH -p l1_1.fastq
l1_2.fastq -p l2_1.fastq l2_2.fastq -o test</p>

<p style="margin-top: 1em">mpiexec -n NUMBER_OF_RANKS Ray
Ray.conf # with commands in a file</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The Ray genome assembler is built on top of the RayPlatform,
a generic plugin-based <br>
distributed and parallel compute engine that uses the
message-passing interface <br>
for passing messages.</p>

<p style="margin-top: 1em">Ray targets several
applications:</p>

<p style="margin-top: 1em">- de novo genome assembly (with
Ray vanilla) <br>
- de novo meta-genome assembly (with Ray
M&Atilde;&Acirc;&copy;ta) <br>
- de novo transcriptome assembly (works, but not tested a
lot) <br>
- quantification of contig abundances <br>
- quantification of microbiome consortia members (with Ray
Communities) <br>
- quantification of transcript expression <br>
- taxonomy profiling of samples (with Ray Communities) <br>
- gene ontology profiling of samples (with Ray
Ontologies)</p>

<p style="margin-top: 1em">OPTIONS <br>
-help <br>
Displays this help page.</p>

<p style="margin-top: 1em">-version <br>
Displays Ray version and compilation options.</p>

<p style="margin-top: 1em">Using a configuration file</p>

<p style="margin-top: 1em">Ray can be launched with <br>
mpiexec -n 16 Ray Ray.conf <br>
The configuration file can include comments (starting with
#).</p>

<p style="margin-top: 1em">K-mer length</p>

<p style="margin-top: 1em">-k kmerLength <br>
Selects the length of k-mers. The default value is 21. <br>
It must be odd because reverse-complement vertices are
stored together. <br>
The maximum length is defined at compilation by
MAXKMERLENGTH <br>
Larger k-mers utilise more memory.</p>

<p style="margin-top: 1em">Inputs</p>

<p style="margin-top: 1em">-p leftSequenceFile
rightSequenceFile [averageOuterDistance standardDeviation]
<br>
Provides two files containing paired-end reads. <br>
averageOuterDistance and standardDeviation are automatically
computed if not provided.</p>

<p style="margin-top: 1em">-i interleavedSequenceFile
[averageOuterDistance standardDeviation] <br>
Provides one file containing interleaved paired-end reads.
<br>
averageOuterDistance and standardDeviation are automatically
computed if not provided.</p>

<p style="margin-top: 1em">-s sequenceFile <br>
Provides a file containing single-end reads.</p>

<p style="margin-top: 1em">Outputs</p>

<p style="margin-top: 1em">-o outputDirectory <br>
Specifies the directory for outputted files. Default is
RayOutput</p>

<p style="margin-top: 1em">Assembly options (defaults work
well)</p>

<p style="margin-top: 1em">-disable-recycling <br>
Disables read recycling during the assembly <br>
reads will be set free in 3 cases: <br>
1. the distance did not match for a pair <br>
2. the read has not met its mate <br>
3. the library population indicates a wrong placement <br>
see Constrained traversal of repeats with paired sequences.
<br>
S&Atilde;&copy;bastien Boisvert, &Atilde;l&Atilde;&copy;nie
Godzaridis, Fran&Atilde;&sect;ois Laviolette &amp; Jacques
Corbeil. <br>
First Annual RECOMB Satellite Workshop on Massively Parallel
Sequencing, March 26-27 2011, Vancouver, BC, Canada.</p>

<p style="margin-top: 1em">-disable-scaffolder <br>
Disables the scaffolder.</p>

<p style="margin-top: 1em">-minimum-contig-length
minimumContigLength <br>
Changes the minimum contig length, default is 100
nucleotides</p>

<p style="margin-top: 1em">-color-space <br>
Runs in color-space <br>
Needs csfasta files. Activated automatically if csfasta
files are provided.</p>

<p style="margin-top: 1em">-use-maximum-seed-coverage
maximumSeedCoverageDepth <br>
Ignores any seed with a coverage depth above this threshold.
<br>
The default is 4294967295.</p>

<p style="margin-top: 1em">-use-minimum-seed-coverage
minimumSeedCoverageDepth <br>
Sets the minimum seed coverage depth. <br>
Any path with a coverage depth lower than this will be
discarded. The default is 0.</p>

<p style="margin-top: 1em">Distributed storage engine (all
these values are for each MPI rank)</p>

<p style="margin-top: 1em">-bloom-filter-bits bits <br>
Sets the number of bits for the Bloom filter <br>
Default is 268435456 bits, 0 bits disables the Bloom
filter.</p>

<p style="margin-top: 1em">-hash-table-buckets buckets <br>
Sets the initial number of buckets. Must be a power of 2 !
<br>
Default value: 268435456</p>

<p style="margin-top: 1em">-hash-table-buckets-per-group
buckets <br>
Sets the number of buckets per group for sparse storage <br>
Default value: 64, Must be between &gt;=1 and &lt;= 64</p>


<p style="margin-top: 1em">-hash-table-load-factor-threshold
threshold <br>
Sets the load factor threshold for real-time resizing <br>
Default value: 0.75, must be &gt;= 0.5 and &lt; 1</p>

<p style="margin-top: 1em">-hash-table-verbosity <br>
Activates verbosity for the distributed storage engine</p>

<p style="margin-top: 1em">Biological abundances</p>

<p style="margin-top: 1em">-search searchDirectory <br>
Provides a directory containing fasta files to be searched
in the de Bruijn graph. <br>
Biological abundances will be written to
RayOutput/BiologicalAbundances <br>
See Documentation/BiologicalAbundances.txt</p>

<p style="margin-top: 1em">-one-color-per-file <br>
Sets one color per file instead of one per sequence. <br>
By default, each sequence in each file has a different
color. <br>
For files with large numbers of sequences, using one single
color per file may be more efficient.</p>

<p style="margin-top: 1em">Taxonomic profiling with colored
de Bruijn graphs</p>

<p style="margin-top: 1em">-with-taxonomy
Genome-to-Taxon.tsv TreeOfLife-Edges.tsv Taxon-Names.tsv
<br>
Provides a taxonomy. <br>
Computes and writes detailed taxonomic profiles. <br>
See Documentation/Taxonomy.txt for details.</p>

<p style="margin-top: 1em">-gene-ontology OntologyTerms.txt
Annotations.txt <br>
Provides an ontology and annotations. <br>
OntologyTerms.txt is fetched from http://geneontology.org
<br>
Annotations.txt is a 2-column file (EMBL_CDS handle &amp;
gene ontology identifier) <br>
See Documentation/GeneOntology.txt <br>
Other outputs</p>

<p style="margin-top: 1em">-enable-neighbourhoods <br>
Computes contig neighborhoods in the de Bruijn graph <br>
Output file: RayOutput/NeighbourhoodRelations.txt</p>

<p style="margin-top: 1em">-amos <br>
Writes the AMOS file called RayOutput/AMOS.afg <br>
An AMOS file contains read positions on contigs. <br>
Can be opened with software with graphical user
interface.</p>

<p style="margin-top: 1em">-write-kmers <br>
Writes k-mer graph to RayOutput/kmers.txt <br>
The resulting file is not utilised by Ray. <br>
The resulting file is very large.</p>

<p style="margin-top: 1em">-write-read-markers <br>
Writes read markers to disk.</p>

<p style="margin-top: 1em">-write-seeds <br>
Writes seed DNA sequences to
RayOutput/Rank&lt;rank&gt;.RaySeeds.fasta</p>

<p style="margin-top: 1em">-write-extensions <br>
Writes extension DNA sequences to
RayOutput/Rank&lt;rank&gt;.RayExtensions.fasta</p>

<p style="margin-top: 1em">-write-contig-paths <br>
Writes contig paths with coverage values <br>
to RayOutput/Rank&lt;rank&gt;.RayContigPaths.txt</p>

<p style="margin-top: 1em">-write-marker-summary <br>
Writes marker statistics.</p>

<p style="margin-top: 1em">Memory usage</p>

<p style="margin-top: 1em">-show-memory-usage <br>
Shows memory usage. Data is fetched from /proc on GNU/Linux
<br>
Needs __linux__</p>

<p style="margin-top: 1em">-show-memory-allocations <br>
Shows memory allocation events</p>

<p style="margin-top: 1em">Algorithm verbosity</p>

<p style="margin-top: 1em">-show-extension-choice <br>
Shows the choice made (with other choices) during the
extension.</p>

<p style="margin-top: 1em">-show-ending-context <br>
Shows the ending context of each extension. <br>
Shows the children of the vertex where extension was too
difficult.</p>

<p style="margin-top: 1em">-show-distance-summary <br>
Shows summary of outer distances used for an extension
path.</p>

<p style="margin-top: 1em">-show-consensus <br>
Shows the consensus when a choice is done.</p>

<p style="margin-top: 1em">Checkpointing</p>

<p style="margin-top: 1em">-write-checkpoints
checkpointDirectory <br>
Write checkpoint files</p>

<p style="margin-top: 1em">-read-checkpoints
checkpointDirectory <br>
Read checkpoint files</p>

<p style="margin-top: 1em">-read-write-checkpoints
checkpointDirectory <br>
Read and write checkpoint files</p>

<p style="margin-top: 1em">Message routing for large number
of cores</p>

<p style="margin-top: 1em">-route-messages <br>
Enables the Ray message router. Disabled by default. <br>
Messages will be routed accordingly so that any rank can
communicate directly with only a few others. <br>
Without -route-messages, any rank can communicate directly
with any other rank. <br>
Files generated: Routing/Connections.txt, Routing/Routes.txt
and Routing/RelayEvents.txt <br>
and Routing/Summary.txt</p>

<p style="margin-top: 1em">-connection-type type <br>
Sets the connection type for routes. <br>
Accepted values are debruijn, hypercube, polytope, group,
random, kautz and complete. Default is debruijn. <br>
debruijn: a full de Bruijn graph a given alphabet and
diameter <br>
hypercube: a hypercube, alphabet is {0,1} and the vertices
is a power of 2 <br>
polytope: a convex regular polytope, alphabet is
{0,1,...,B-1} and the vertices is a power of B <br>
group: silly model where one representative per group can
communicate with outsiders <br>
random: Erd&Aring;s-R&Atilde;&copy;nyi model <br>
kautz: a full de Kautz graph, which is a subgraph of a de
Bruijn graph <br>
complete: a full graph with all the possible connections
<br>
With the type debruijn, the number of ranks must be a power
of something. <br>
Examples: 256 = 16*16, 512=8*8*8, 49=7*7, and so on. <br>
Otherwise, don&rsquo;t use debruijn routing but use another
one <br>
With the type kautz, the number of ranks n must be
n=(k+1)*k^(d-1) for some k and d</p>

<p style="margin-top: 1em">-routing-graph-degree degree
<br>
Specifies the outgoing degree for the routing graph. <br>
See Documentation/Routing.txt</p>

<p style="margin-top: 1em">Hardware testing</p>

<p style="margin-top: 1em">-test-network-only <br>
Tests the network and returns.</p>

<p style="margin-top: 1em">-write-network-test-raw-data
<br>
Writes one additional file per rank detailing the network
test.</p>

<p style="margin-top: 1em">-exchanges NumberOfExchanges
<br>
Sets the number of exchanges</p>

<p style="margin-top: 1em">-disable-network-test <br>
Skips the network test.</p>

<p style="margin-top: 1em">Debugging</p>

<p style="margin-top: 1em">-verify-message-integrity <br>
Checks message data reliability for any non-empty message.
<br>
add &rsquo;-D CONFIG_SSE_4_2&rsquo; in the Makefile to use
hardware instruction (SSE 4.2)</p>

<p style="margin-top: 1em">-run-profiler <br>
Runs the profiler as the code runs. By default, only show
granularity warnings. <br>
Running the profiler increases running times.</p>

<p style="margin-top: 1em">-with-profiler-details <br>
Shows number of messages sent and received in each methods
during in each time slices (epochs). Needs
-run-profiler.</p>

<p style="margin-top: 1em">-show-communication-events <br>
Shows all messages sent and received.</p>

<p style="margin-top: 1em">-show-read-placement <br>
Shows read placement in the graph during the extension.</p>

<p style="margin-top: 1em">-debug-bubbles <br>
Debugs bubble code. <br>
Bubbles can be due to heterozygous sites or sequencing
errors or other (unknown) events</p>

<p style="margin-top: 1em">-debug-seeds <br>
Debugs seed code. <br>
Seeds are paths in the graph that are likely unique.</p>

<p style="margin-top: 1em">-debug-fusions <br>
Debugs fusion code.</p>

<p style="margin-top: 1em">-debug-scaffolder <br>
Debug the scaffolder.</p>

<p style="margin-top: 1em">FILES <br>
Input files</p>

<p style="margin-top: 1em">Note: file format is determined
with file extension.</p>

<p style="margin-top: 1em">.fasta <br>
.fasta.gz (needs HAVE_LIBZ=y at compilation) <br>
.fasta.bz2 (needs HAVE_LIBBZ2=y at compilation) <br>
.fastq <br>
.fastq.gz (needs HAVE_LIBZ=y at compilation) <br>
.fastq.bz2 (needs HAVE_LIBBZ2=y at compilation) <br>
.sff (paired reads must be extracted manually) <br>
.csfasta (color-space reads)</p>

<p style="margin-top: 1em">Outputted files</p>

<p style="margin-top: 1em">Scaffolds</p>

<p style="margin-top: 1em">RayOutput/Scaffolds.fasta <br>
The scaffold sequences in FASTA format <br>
RayOutput/ScaffoldComponents.txt <br>
The components of each scaffold <br>
RayOutput/ScaffoldLengths.txt <br>
The length of each scaffold <br>
RayOutput/ScaffoldLinks.txt <br>
Scaffold links</p>

<p style="margin-top: 1em">Contigs</p>

<p style="margin-top: 1em">RayOutput/Contigs.fasta <br>
Contiguous sequences in FASTA format <br>
RayOutput/ContigLengths.txt <br>
The lengths of contiguous sequences</p>

<p style="margin-top: 1em">Summary</p>

<p style="margin-top: 1em">RayOutput/OutputNumbers.txt <br>
Overall numbers for the assembly</p>

<p style="margin-top: 1em">de Bruijn graph</p>


<p style="margin-top: 1em">RayOutput/CoverageDistribution.txt
<br>
The distribution of coverage values <br>
RayOutput/CoverageDistributionAnalysis.txt <br>
Analysis of the coverage distribution <br>
RayOutput/degreeDistribution.txt <br>
Distribution of ingoing and outgoing degrees <br>
RayOutput/kmers.txt <br>
k-mer graph, required option: -write-kmers <br>
The resulting file is not utilised by Ray. <br>
The resulting file is very large.</p>

<p style="margin-top: 1em">Assembly steps</p>


<p style="margin-top: 1em">RayOutput/SeedLengthDistribution.txt
<br>
Distribution of seed length <br>
RayOutput/Rank&lt;rank&gt;.OptimalReadMarkers.txt <br>
Read markers. <br>
RayOutput/Rank&lt;rank&gt;.RaySeeds.fasta <br>
Seed DNA sequences, required option: -write-seeds <br>
RayOutput/Rank&lt;rank&gt;.RayExtensions.fasta <br>
Extension DNA sequences, required option: -write-extensions
<br>
RayOutput/Rank&lt;rank&gt;.RayContigPaths.txt <br>
Contig paths with coverage values, required option:
-write-contig-paths</p>

<p style="margin-top: 1em">Paired reads</p>

<p style="margin-top: 1em">RayOutput/LibraryStatistics.txt
<br>
Estimation of outer distances for paired reads <br>
RayOutput/Library&lt;LibraryNumber&gt;.txt <br>
Frequencies for observed outer distances (insert size + read
lengths)</p>

<p style="margin-top: 1em">Partition</p>

<p style="margin-top: 1em">RayOutput/NumberOfSequences.txt
<br>
Number of reads in each file <br>
RayOutput/SequencePartition.txt <br>
Sequence partition</p>

<p style="margin-top: 1em">Ray software</p>

<p style="margin-top: 1em">RayOutput/RayVersion.txt <br>
The version of Ray <br>
RayOutput/RayCommand.txt <br>
The exact same command provided</p>

<p style="margin-top: 1em">AMOS</p>

<p style="margin-top: 1em">RayOutput/AMOS.afg <br>
Assembly representation in AMOS format, required option:
-amos</p>

<p style="margin-top: 1em">Communication</p>


<p style="margin-top: 1em">RayOutput/MessagePassingInterface.txt
Number of messages sent <br>
RayOutput/NetworkTest.txt Latencies in microseconds <br>
RayOutput/Rank&lt;rank&gt;NetworkTestData.txt Network test
raw data</p>

<p style="margin-top: 1em">DOCUMENTATION <br>
- mpiexec -n 1 Ray -help|less (always up-to-date) <br>
- This help page (always up-to-date) <br>
- The directory Documentation/ <br>
- Manual (Portable Document Format): InstructionManual.tex
(in Documentation) <br>
- Mailing list archives:
http://sourceforge.net/mailarchive/forum.php?forum_name=denovoassembler-users</p>

<p style="margin-top: 1em">AUTHOR <br>
Written by S&Atilde;&copy;bastien Boisvert.</p>

<p style="margin-top: 1em">REPORTING BUGS <br>
Report bugs to denovoassembler-users@lists.sourceforge.net
<br>
Home page:
&lt;http://denovoassembler.sourceforge.net/&gt;</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This program is free software: you can redistribute it
and/or modify <br>
it under the terms of the GNU General Public License as
published by <br>
the Free Software Foundation, version 3 of the License.</p>

<p style="margin-top: 1em">This program is distributed in
the hope that it will be useful, <br>
but WITHOUT ANY WARRANTY; without even the implied warranty
of <br>
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
<br>
GNU General Public License for more details.</p>

<p style="margin-top: 1em">You have received a copy of the
GNU General Public License <br>
along with this program (see LICENSE).</p>

<p style="margin-top: 1em">Ray 2.1.0 November 2012
RAY(1)</p>
<hr>
</body>
</html>
