<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:35:31 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>RESTRICT(1e) EMBOSS Manual for Debian RESTRICT(1e)</p>

<p style="margin-top: 1em">NAME <br>
restrict - Report restriction enzyme cleavage sites in a
nucleotide sequence</p>

<p style="margin-top: 1em">SYNOPSIS <br>
restrict -sequence seqall -datafile datafile -mfile datafile
-sitelen integer -enzymes string -min integer -max integer
-solofragment boolean -single boolean -blunt boolean <br>
-sticky boolean -ambiguity boolean -plasmid boolean
-methylation boolean -commercial boolean -limit boolean
-alphabetic boolean -fragments boolean -name boolean <br>
-outfile report</p>

<p style="margin-top: 1em">restrict -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
restrict is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Nucleic:Restriction&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall</p>

<p style="margin-top: 1em">-datafile datafile</p>

<p style="margin-top: 1em">-mfile datafile <br>
Default value: Emethylsites.dat</p>

<p style="margin-top: 1em">Required section <br>
-sitelen integer <br>
This sets the minimum length of the restriction enzyme
recognition site. Any enzymes with sites shorter than this
will be ignored. Default value: 4</p>

<p style="margin-top: 1em">-enzymes string <br>
The name &rsquo;all&rsquo; reads in all enzyme names from
the REBASE database. You can specify enzymes by giving their
names with commas between then, such as: <br>
&rsquo;HincII,hinfI,ppiI,hindiii&rsquo;. The case of the
names is not important. You can specify a file of enzyme
names to read in by giving the name of the file holding the
enzyme names <br>
with a &rsquo;@&rsquo; character in front of it, for
example, &rsquo;@enz.list&rsquo;. Blank lines and lines
starting with a hash character or &rsquo;!&rsquo; are
ignored and all other lines are concatenated <br>
together with a comma character &rsquo;,&rsquo; and then
treated as the list of enzymes to search for. An example of
a file of enzyme names is: ! my enzymes HincII, ppiII !
other enzymes <br>
hindiii HinfI PpiI Default value: all</p>

<p style="margin-top: 1em">Advanced section <br>
-min integer <br>
This sets the minimum number of cuts for any restriction
enzyme that will be considered. Any enzymes that cut fewer
times than this will be ignored. Default value: 1</p>

<p style="margin-top: 1em">-max integer <br>
This sets the maximum number of cuts for any restriction
enzyme that will be considered. Any enzymes that cut more
times than this will be ignored. Default value:
2000000000</p>

<p style="margin-top: 1em">-solofragment boolean <br>
This gives the fragment lengths of the forward sense strand
produced by complete restriction by each restriction enzyme
on its own. Results are added to the tail section of <br>
the report. Default value: N</p>

<p style="margin-top: 1em">-single boolean <br>
If this is set then this forces the values of the mincuts
and maxcuts qualifiers to both be 1. Any other value you may
have set them to will be ignored. Default value: N</p>

<p style="margin-top: 1em">-blunt boolean <br>
This allows those enzymes which cut at the same position on
the forward and reverse strands to be considered. Default
value: Y</p>

<p style="margin-top: 1em">-sticky boolean <br>
This allows those enzymes which cut at different positions
on the forward and reverse strands, leaving an overhang, to
be considered. Default value: Y</p>

<p style="margin-top: 1em">-ambiguity boolean <br>
This allows those enzymes which have one or more
&rsquo;N&rsquo; ambiguity codes in their pattern to be
considered Default value: Y</p>

<p style="margin-top: 1em">-plasmid boolean <br>
If this is set then this allows searches for restriction
enzyme recognition site and cut positions that span the end
of the sequence to be considered. Default value: N</p>

<p style="margin-top: 1em">-methylation boolean <br>
If this is set then RE recognition sites will not match
methylated bases. Default value: N</p>

<p style="margin-top: 1em">-commercial boolean <br>
If this is set, then only those enzymes with a commercial
supplier will be searched for. This qualifier is ignored if
you have specified an explicit list of enzymes to search
<br>
for, rather than searching through &rsquo;all&rsquo; the
enzymes in the REBASE database. It is assumed that, if you
are asking for an explicit enzyme, then you probably know
where to get <br>
it from and so all enzymes names that you have asked to be
searched for, and which cut, will be reported whether or not
they have a commercial supplier. Default value: Y</p>

<p style="margin-top: 1em">Output section <br>
-limit boolean <br>
This limits the reporting of enzymes to just one enzyme from
each group of isoschizomers. The enzyme chosen to represent
an isoschizomer group is the prototype indicated in <br>
the data file &rsquo;embossre.equ&rsquo;, which is created
by the program &rsquo;rebaseextract&rsquo;. If you prefer
different prototypes to be used, make a copy of embossre.equ
in your home <br>
directory and edit it. If this value is set to be false then
all of the input enzymes will be reported. You might like to
set this to false if you are supplying an explicit <br>
set of enzymes rather than searching &rsquo;all&rsquo; of
them. Default value: Y</p>

<p style="margin-top: 1em">-alphabetic boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-fragments boolean <br>
This gives the fragment lengths of the forward sense strand
produced by complete restriction using all of the input
enzymes together. Results are added to the tail section of
<br>
the report. Default value: N</p>

<p style="margin-top: 1em">-name boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-outfile report</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
restrict is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
RESTRICT(1e)</p>
<hr>
</body>
</html>
