<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>RESTRICT(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">RESTRICT(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">RESTRICT(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
restrict - Report restriction enzyme cleavage sites in a nucleotide sequence
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>restrict</b>
  <b>-sequence&#x00A0;</b><b></b><i>seqall</i>
  <b>-datafile&#x00A0;</b><b></b><i>datafile</i>
  <b>-mfile&#x00A0;</b><b></b><i>datafile</i>
  <b>-sitelen&#x00A0;</b><b></b><i>integer</i>
  <b>-enzymes&#x00A0;</b><b></b><i>string</i>
  <b>-min&#x00A0;</b><b></b><i>integer</i>
  <b>-max&#x00A0;</b><b></b><i>integer</i>
  <b>-solofragment&#x00A0;</b><b></b><i>boolean</i>
  <b>-single&#x00A0;</b><b></b> <i>boolean</i>
  <b>-blunt&#x00A0;</b><b></b><i>boolean</i>
  <b>-sticky&#x00A0;</b><b></b><i>boolean</i>
  <b>-ambiguity&#x00A0;</b><b></b><i>boolean</i>
  <b>-plasmid&#x00A0;</b><b></b><i>boolean</i>
  <b>-methylation&#x00A0;</b><b></b> <i>boolean</i>
  <b>-commercial&#x00A0;</b><b></b><i>boolean</i>
  <b>-limit&#x00A0;</b><b></b><i>boolean</i>
  <b>-alphabetic&#x00A0;</b><b></b><i>boolean</i>
  <b>-fragments&#x00A0;</b><b></b> <i>boolean</i>
  <b>-name&#x00A0;</b><b></b><i>boolean</i>
  <b>-outfile&#x00A0;</b><b></b><i>report</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>restrict</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>restrict</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Nucleic:Restriction&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-sequence</b> <i>seqall</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-datafile</b> <i>datafile</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-mfile</b> <i>datafile</i>
<div style="margin-left: 4.00ex;">Default value: Emethylsites.dat</div>
<h2 class="Ss" title="Ss" id="Required_section"><a class="selflink" href="#Required_section">Required
  section</a></h2>
<b>-sitelen</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This sets the minimum length of the
  restriction enzyme recognition site. Any enzymes with sites shorter than this
  will be ignored. Default value: 4</div>
<div class="Pp"></div>
<b>-enzymes</b> <i>string</i>
<div style="margin-left: 4.00ex;">The name 'all' reads in all enzyme names from
  the REBASE database. You can specify enzymes by giving their names with commas
  between then, such as: 'HincII,hinfI,ppiI,hindiii'. The case of the names is
  not important. You can specify a file of enzyme names to read in by giving the
  name of the file holding the enzyme names with a '@' character in front of it,
  for example, '@enz.list'. Blank lines and lines starting with a hash character
  or '!' are ignored and all other lines are concatenated together with a comma
  character ',' and then treated as the list of enzymes to search for. An
  example of a file of enzyme names is: ! my enzymes HincII, ppiII ! other
  enzymes hindiii HinfI PpiI Default value: all</div>
<h2 class="Ss" title="Ss" id="Advanced_section"><a class="selflink" href="#Advanced_section">Advanced
  section</a></h2>
<b>-min</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This sets the minimum number of cuts for any
  restriction enzyme that will be considered. Any enzymes that cut fewer times
  than this will be ignored. Default value: 1</div>
<div class="Pp"></div>
<b>-max</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This sets the maximum number of cuts for any
  restriction enzyme that will be considered. Any enzymes that cut more times
  than this will be ignored. Default value: 2000000000</div>
<div class="Pp"></div>
<b>-solofragment</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This gives the fragment lengths of the forward
  sense strand produced by complete restriction by each restriction enzyme on
  its own. Results are added to the tail section of the report. Default value:
  N</div>
<div class="Pp"></div>
<b>-single</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this is set then this forces the values of
  the mincuts and maxcuts qualifiers to both be 1. Any other value you may have
  set them to will be ignored. Default value: N</div>
<div class="Pp"></div>
<b>-blunt</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This allows those enzymes which cut at the
  same position on the forward and reverse strands to be considered. Default
  value: Y</div>
<div class="Pp"></div>
<b>-sticky</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This allows those enzymes which cut at
  different positions on the forward and reverse strands, leaving an overhang,
  to be considered. Default value: Y</div>
<div class="Pp"></div>
<b>-ambiguity</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This allows those enzymes which have one or
  more 'N' ambiguity codes in their pattern to be considered Default value:
  Y</div>
<div class="Pp"></div>
<b>-plasmid</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this is set then this allows searches for
  restriction enzyme recognition site and cut positions that span the end of the
  sequence to be considered. Default value: N</div>
<div class="Pp"></div>
<b>-methylation</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this is set then RE recognition sites will
  not match methylated bases. Default value: N</div>
<div class="Pp"></div>
<b>-commercial</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this is set, then only those enzymes with a
  commercial supplier will be searched for. This qualifier is ignored if you
  have specified an explicit list of enzymes to search for, rather than
  searching through 'all' the enzymes in the REBASE database. It is assumed
  that, if you are asking for an explicit enzyme, then you probably know where
  to get it from and so all enzymes names that you have asked to be searched
  for, and which cut, will be reported whether or not they have a commercial
  supplier. Default value: Y</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-limit</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This limits the reporting of enzymes to just
  one enzyme from each group of isoschizomers. The enzyme chosen to represent an
  isoschizomer group is the prototype indicated in the data file 'embossre.equ',
  which is created by the program 'rebaseextract'. If you prefer different
  prototypes to be used, make a copy of embossre.equ in your home directory and
  edit it. If this value is set to be false then all of the input enzymes will
  be reported. You might like to set this to false if you are supplying an
  explicit set of enzymes rather than searching 'all' of them. Default value:
  Y</div>
<div class="Pp"></div>
<b>-alphabetic</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<div class="Pp"></div>
<b>-fragments</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This gives the fragment lengths of the forward
  sense strand produced by complete restriction using all of the input enzymes
  together. Results are added to the tail section of the report. Default value:
  N</div>
<div class="Pp"></div>
<b>-name</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<div class="Pp"></div>
<b>-outfile</b> <i>report</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
restrict is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
