<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>RUN_GUBBINS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">RUN_GUBBINS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">RUN_GUBBINS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
run_gubbins - phylogenetic analysis of genome sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>run_gubbins</b> [-h] [--outgroup OUTGROUP] [--starting_tree STARTING_TREE]
  [--use_time_stamp] [--verbose] [--no_cleanup] [--tree_builder TREE_BUILDER]
  [--iterations ITERATIONS] [--min_snps MIN_SNPS] [--filter_percentage
  FILTER_PERCENTAGE] [--prefix PREFIX] [--threads THREADS] [--converge_method
  CONVERGE_METHOD] [--version] [--min_window_size MIN_WINDOW_SIZE]
  [--max_window_size MAX_WINDOW_SIZE] alignment_filename
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Gubbins supports rapid phylogenetic analysis of large samples of recombinant
  bacterial whole genome sequences.
<div class="Pp"></div>
Gubbins (Genealogies Unbiased By recomBinations In Nucleotide Sequences) is an
  algorithm that iteratively identifies loci containing elevated densities of
  base substitutions while concurrently constructing a phylogeny based on the
  putative point mutations outside of these regions. Simulations demonstrate the
  algorithm generates highly accurate reconstructions under realistic models of
  short-term bacterial evolution, and can be run in only a few hours on
  alignments of hundreds of bacterial genome sequences.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="positional_arguments:"><a class="selflink" href="#positional_arguments:">positional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">alignment_filename</dt>
  <dd class="It-tag">Multifasta alignment file</dd>
</dl>
<h2 class="Ss" title="Ss" id="optional_arguments:"><a class="selflink" href="#optional_arguments:">optional
  arguments:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b>, <b>--help</b></dt>
  <dd class="It-tag">show this help message and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--outgroup</b> OUTGROUP, <b>-o</b> OUTGROUP</dt>
  <dd class="It-tag">Outgroup name for rerooting. A list of comma separated
      names can be used if they form a clade</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--starting_tree</b> STARTING_TREE, <b>-s</b>
    STARTING_TREE</dt>
  <dd class="It-tag">Starting tree</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--use_time_stamp</b>, <b>-u</b></dt>
  <dd class="It-tag">Use a time stamp in file names</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--verbose</b>, <b>-v</b></dt>
  <dd class="It-tag">Turn on debugging</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--no_cleanup</b>, <b>-n</b></dt>
  <dd class="It-tag">Don't cleanup intermediate files</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--tree_builder</b> TREE_BUILDER, <b>-t</b>
    TREE_BUILDER</dt>
  <dd class="It-tag">Application to use for tree building
      [raxml|fasttree|hybrid], default RAxML</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--iterations</b> ITERATIONS, <b>-i</b> ITERATIONS</dt>
  <dd class="It-tag">Maximum No. of iterations, default is 5</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_snps</b> MIN_SNPS, <b>-m</b> MIN_SNPS</dt>
  <dd class="It-tag">Min SNPs to identify a recombination block, default is
    3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--filter_percentage</b> FILTER_PERCENTAGE, <b>-f</b>
    FILTER_PERCENTAGE</dt>
  <dd class="It-tag">Filter out taxa with more than this percentage of gaps,
      default is 25</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--prefix</b> PREFIX, <b>-p</b> PREFIX</dt>
  <dd class="It-tag">Add a prefix to the final output filenames</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--threads</b> THREADS, <b>-c</b> THREADS</dt>
  <dd class="It-tag">Number of threads to run with RAXML, but only if a PTHREADS
      version is available</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--converge_method</b> CONVERGE_METHOD, <b>-z</b>
    CONVERGE_METHOD</dt>
  <dd class="It-tag">Criteria to use to know when to halt iterations [weigh
      ted_robinson_foulds|robinson_foulds|recombination]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--version</b></dt>
  <dd class="It-tag">show program's version number and exit</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--min_window_size</b> MIN_WINDOW_SIZE, <b>-a</b>
    MIN_WINDOW_SIZE</dt>
  <dd class="It-tag">Minimum window size, default 100</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--max_window_size</b> MAX_WINDOW_SIZE, <b>-b</b>
    MAX_WINDOW_SIZE</dt>
  <dd class="It-tag">Maximum window size, default 10000</dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Croucher N. J., Page A. J., Connor T. R., Delaney A. J., Keane J. A., Bentley S.
  D., Parkhill J., Harris S.R. &quot;Rapid phylogenetic analysis of large
  samples of recombinant bacterial whole genome sequences using Gubbins&quot;.
  Nucleic Acids Res. 2015 Feb 18;43(3):e15. doi: 10.1093/nar/gku1196.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">Julu 2016</td>
    <td class="foot-os">run_gubbins 2.0.0</td>
  </tr>
</table>
</body>
</html>
