<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:36:21 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>RUN_MAPSEMBLER2_PIPELINE(1) User Commands
RUN_MAPSEMBLER2_PIPELINE(1)</p>

<p style="margin-top: 1em">NAME <br>
run_mapsembler2_pipeline - manual page for
run_mapsembler2_pipeline 2.2.3+dfsg</p>

<p style="margin-top: 1em">SYNOPSIS <br>
run_mapsembler_and_phaser.sh -s &lt;starter.fasta&gt; -r
&lt;reads.faste&gt; -t [1/2/3/4]&lt;options&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
run_mapsembler_pipeline.sh, a pipelining
mapsembler2_extremities, mapsembler2_extend and
kissread_g</p>

<p style="margin-top: 1em">-s: file containing starters
(fasta) -r list of reads separated by space, surrounded by
the &rsquo;&quot;&rsquo; character. Note that reads may be
in fasta or fastq format, gzipped or not. <br>
Example: -r &quot;data_sample/reads_sequence1.fasta
data_sample/reads_sequence2.fasta.gz&quot;. -t: kind of
assembly: 1=unitig (fasta), 2=contig (fasta), 3=unitig
(graph), 4=con&acirc; <br>
tig(graph)</p>

<p style="margin-top: 1em">&lt;options&gt;: <br>
-p prefix. All out files will start with this prefix.
Example: -p my_prefix -k value. Set the length of used
kmers. Must fit the compiled value. Default=31. Example -k
31 <br>
-c value. Set the minimal coverage. Default=5. Example -c 5
-d value. Set the number of authorized substitutions used
while mapping reads on found SNPs. Default=1. Exam&acirc;
<br>
ple: -d 1 -g value. Estimated genome size. Used only to
control memory usage. e.g. 3 billion (3000000000) uses 4Gb
of RAM. Default=10 million. Example: -d 10000000 -f <br>
value. Set the process of search in the graph (1=Breadth and
2=Depth). Default=1. Example: -f 1 -x value. Set the maximal
nodes length . Default=40. Example: -x 40 -y <br>
value. Set the maximal graph depth . Default=10000. Example:
-y 10000 -h Prints this message and exist</p>

<p style="margin-top: 1em">Any further question: read the
readme file or contact us: pierre.peterlongo@inria.fr</p>

<p style="margin-top: 1em">SEE ALSO <br>
The full documentation for run_mapsembler2_pipeline is
maintained as a Texinfo manual. If the info and
run_mapsembler2_pipeline programs are properly installed at
your site, the <br>
command</p>

<p style="margin-top: 1em">info
run_mapsembler2_pipeline</p>

<p style="margin-top: 1em">should give you access to the
complete manual.</p>

<p style="margin-top: 1em">run_mapsembler2_pipeline
2.2.3+dfsg September 2016 RUN_MAPSEMBLER2_PIPELINE(1)</p>
<hr>
</body>
</html>
