<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:36:20 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>RUN_GUBBINS(1) User Commands RUN_GUBBINS(1)</p>

<p style="margin-top: 1em">NAME <br>
run_gubbins - phylogenetic analysis of genome sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
run_gubbins [-h] [--outgroup OUTGROUP] [--starting_tree
STARTING_TREE] [--use_time_stamp] [--verbose] [--no_cleanup]
[--tree_builder TREE_BUILDER] [--iterations ITERATIONS] <br>
[--min_snps MIN_SNPS] [--filter_percentage
FILTER_PERCENTAGE] [--prefix PREFIX] [--threads THREADS]
[--converge_method CONVERGE_METHOD] [--version]
[--min_window_size MIN_WIN&acirc; <br>
DOW_SIZE] [--max_window_size MAX_WINDOW_SIZE]
alignment_filename</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Gubbins supports rapid phylogenetic analysis of large
samples of recombinant bacterial whole genome sequences.</p>

<p style="margin-top: 1em">Gubbins (Genealogies Unbiased By
recomBinations In Nucleotide Sequences) is an algorithm that
iteratively identifies loci containing elevated densities of
base substitutions <br>
while concurrently constructing a phylogeny based on the
putative point mutations outside of these regions.
Simulations demonstrate the algorithm generates highly
accurate recon&acirc; <br>
structions under realistic models of short-term bacterial
evolution, and can be run in only a few hours on alignments
of hundreds of bacterial genome sequences.</p>

<p style="margin-top: 1em">OPTIONS <br>
positional arguments: <br>
alignment_filename <br>
Multifasta alignment file</p>

<p style="margin-top: 1em">optional arguments: <br>
-h, --help <br>
show this help message and exit</p>

<p style="margin-top: 1em">--outgroup OUTGROUP, -o OUTGROUP
<br>
Outgroup name for rerooting. A list of comma separated names
can be used if they form a clade</p>

<p style="margin-top: 1em">--starting_tree STARTING_TREE,
-s STARTING_TREE <br>
Starting tree</p>

<p style="margin-top: 1em">--use_time_stamp, -u <br>
Use a time stamp in file names</p>

<p style="margin-top: 1em">--verbose, -v <br>
Turn on debugging</p>

<p style="margin-top: 1em">--no_cleanup, -n <br>
Don&rsquo;t cleanup intermediate files</p>

<p style="margin-top: 1em">--tree_builder TREE_BUILDER, -t
TREE_BUILDER <br>
Application to use for tree building
[raxml|fasttree|hybrid], default RAxML</p>

<p style="margin-top: 1em">--iterations ITERATIONS, -i
ITERATIONS <br>
Maximum No. of iterations, default is 5</p>

<p style="margin-top: 1em">--min_snps MIN_SNPS, -m MIN_SNPS
<br>
Min SNPs to identify a recombination block, default is 3</p>

<p style="margin-top: 1em">--filter_percentage
FILTER_PERCENTAGE, -f FILTER_PERCENTAGE <br>
Filter out taxa with more than this percentage of gaps,
default is 25</p>

<p style="margin-top: 1em">--prefix PREFIX, -p PREFIX <br>
Add a prefix to the final output filenames</p>

<p style="margin-top: 1em">--threads THREADS, -c THREADS
<br>
Number of threads to run with RAXML, but only if a PTHREADS
version is available</p>

<p style="margin-top: 1em">--converge_method
CONVERGE_METHOD, -z CONVERGE_METHOD <br>
Criteria to use to know when to halt iterations [weigh
ted_robinson_foulds|robinson_foulds|recombination]</p>

<p style="margin-top: 1em">--version <br>
show program&rsquo;s version number and exit</p>

<p style="margin-top: 1em">--min_window_size
MIN_WINDOW_SIZE, -a MIN_WINDOW_SIZE <br>
Minimum window size, default 100</p>

<p style="margin-top: 1em">--max_window_size
MAX_WINDOW_SIZE, -b MAX_WINDOW_SIZE <br>
Maximum window size, default 10000</p>

<p style="margin-top: 1em">SEE ALSO <br>
Croucher N. J., Page A. J., Connor T. R., Delaney A. J.,
Keane J. A., Bentley S. D., Parkhill J., Harris S.R.
&quot;Rapid phylogenetic analysis of large samples of
recombinant bacte&acirc; <br>
rial whole genome sequences using Gubbins&quot;. Nucleic
Acids Res. 2015 Feb 18;43(3):e15. doi:
10.1093/nar/gku1196.</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">run_gubbins 2.0.0 Julu 2016
RUN_GUBBINS(1)</p>
<hr>
</body>
</html>
