<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>RUN_MAPSEMBLER2_PIPELINE(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">RUN_MAPSEMBLER2_PIPELINE(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">RUN_MAPSEMBLER2_PIPELINE(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
run_mapsembler2_pipeline - manual page for run_mapsembler2_pipeline 2.2.3+dfsg
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>run_mapsembler_and_phaser.sh</b> <i>-s &lt;starter.fasta&gt; -r
  &lt;reads.faste&gt; -t </i>[ <i>1/2/3/4</i>]<i>&lt;options&gt;</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
run_mapsembler_pipeline.sh, a pipelining mapsembler2_extremities,
  mapsembler2_extend and kissread_g
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>-s</b>: file containing starters (fasta) <b>-r</b> list
      of reads separated by space, surrounded by the '&quot;' character. Note
      that reads may be in fasta or fastq format, gzipped or not. Example:
      <b>-r</b> &quot;data_sample/reads_sequence1.fasta
      data_sample/reads_sequence2.fasta.gz&quot;. <b>-t</b>: kind of assembly:
      1=unitig (fasta), 2=contig (fasta), 3=unitig (graph), 4=contig(graph)</dd>
</dl>
<h2 class="Ss" title="Ss" id="&lt;options&gt;:"><a class="selflink" href="#&lt;options&gt;:">&lt;options&gt;:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>-p</b> prefix. All out files will start with this
      prefix. Example: <b>-p</b> my_prefix <b>-k</b> value. Set the length of
      used kmers. Must fit the compiled value. Default=31. Example <b>-k</b> 31
      <b>-c</b> value. Set the minimal coverage. Default=5. Example <b>-c</b> 5
      <b>-d</b> value. Set the number of authorized substitutions used while
      mapping reads on found SNPs. Default=1. Example: <b>-d</b> 1 <b>-g</b>
      value. Estimated genome size. Used only to control memory usage. e.g. 3
      billion (3000000000) uses 4Gb of RAM. Default=10 million. Example:
      <b>-d</b> 10000000 <b>-f</b> value. Set the process of search in the graph
      (1=Breadth and 2=Depth). Default=1. Example: <b>-f</b> 1 <b>-x</b> value.
      Set the maximal nodes length . Default=40. Example: <b>-x</b> 40 <b>-y</b>
      value. Set the maximal graph depth . Default=10000. Example: <b>-y</b>
      10000 <b>-h</b> Prints this message and exist</dd>
</dl>
<div class="Pp"></div>
Any further question: read the readme file or contact us:
  pierre.peterlongo@inria.fr
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
The full documentation for <b>run_mapsembler2_pipeline</b> is maintained as a
  Texinfo manual. If the <b>info</b> and <b>run_mapsembler2_pipeline</b>
  programs are properly installed at your site, the command
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag"><b>info run_mapsembler2_pipeline</b></dd>
</dl>
<div class="Pp"></div>
should give you access to the complete manual.</div>
<table class="foot">
  <tr>
    <td class="foot-date">September 2016</td>
    <td class="foot-os">run_mapsembler2_pipeline 2.2.3+dfsg</td>
  </tr>
</table>
</body>
</html>
