<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:36:21 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>JMODELTEST(1) User Commands JMODELTEST(1)</p>

<p style="margin-top: 1em">NAME <br>
jmodeltest - HPC selection of models of nucleotide
substitution</p>

<p style="margin-top: 1em">SYNOPSIS <br>
jmodeltest -d sequenceFileName [-getPhylip] [-ckp
checkpointFileName.ckp] [-n executionName] [-t
fixed|BIONJ|ML] [-u userTreeFileName] [-o outputFileName]
[-S NNI|SPR|BEST] <br>
[-AIC] [-AICc] [-BIC] [-DT] [-c confidenceInterval] [-s
3|5|7|11|203] [-f] [-i] [-g numberOfCategories] [-uLNL]
[-dLRT] [-h confidenceInterval] [-hLRT] [-O {ftvwxgp}] [-a]
[-z] <br>
[-p] [-v] [-w] [-tr numberOfThreads] [-machinesfile
machinesFileName]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
jModelTest is a tool to carry out statistical selection of
best-fit models of nucleotide substitution. It implements
five different model selection strategies: hierarchical and
<br>
dynamical likelihood ratio tests (hLRT and dLRT), Akaike and
Bayesian information criteria (AIC and BIC), and a decision
theory method (DT). It also provides estimates of model <br>
selection uncertainty, parameter importances and
model-averaged parameter estimates, including model-averaged
tree topologies. jModelTest 2 includes High Performance
Computing <br>
(HPC) capabilities and additional features like new
strategies for tree optimization, model- averaged
phylogenetic trees (both topology and branch length),
heuristic filtering <br>
and automatic logging of user activity.</p>

<p style="margin-top: 1em">OPTIONS <br>
-a</p>

<p style="margin-top: 1em">estimate model-averaged
phylogeny for each active criterion (e.g., -a) (default is
false)</p>

<p style="margin-top: 1em">-AIC</p>

<p style="margin-top: 1em">calculate the Akaike Information
Criterion (e.g., -AIC) (default is false)</p>

<p style="margin-top: 1em">-AICc</p>

<p style="margin-top: 1em">calculate the corrected Akaike
Information Criterion (e.g., -AICc) (default is false)</p>

<p style="margin-top: 1em">-BIC</p>

<p style="margin-top: 1em">calculate the Bayesian
Information Criterion (e.g., -BIC) (default is false)</p>

<p style="margin-top: 1em">-DT</p>

<p style="margin-top: 1em">calculate the decision theory
criterion (e.g., -DT) (default is false)</p>

<p style="margin-top: 1em">-c confidenceInterval</p>

<p style="margin-top: 1em">confidence interval (e.g., -c
90) (default is 100)</p>

<p style="margin-top: 1em">-ckp checkpointFileName</p>

<p style="margin-top: 1em">Loads a checkpointing file</p>

<p style="margin-top: 1em">-d sequenceFileName</p>

<p style="margin-top: 1em">input data file (e.g., -d
data.phy)</p>

<p style="margin-top: 1em">-dLRT</p>

<p style="margin-top: 1em">do dynamical likelihood ratio
tests (e.g., -dLRT)(default is false)</p>

<p style="margin-top: 1em">-f</p>

<p style="margin-top: 1em">include models with unequals
base frecuencies (e.g., -f) (default is false)</p>

<p style="margin-top: 1em">-g numberOfCategories</p>

<p style="margin-top: 1em">include models with rate
variation among sites and number of categories (e.g., -g 8)
(default is false &amp; 4 categories)</p>

<p style="margin-top: 1em">-G threshold</p>

<p style="margin-top: 1em">heuristic search. Requires a
threshold &gt; 0 (e.g., -G 0.1)</p>

<p style="margin-top: 1em">-getPhylip</p>

<p style="margin-top: 1em">converts the input file into
phylip format</p>

<p style="margin-top: 1em">-h confidenceInterval</p>

<p style="margin-top: 1em">confidence level for the hLRTs
(e.g., -a0.002) (default is 0.01)</p>

<p style="margin-top: 1em">-H informationCriterion</p>

<p style="margin-top: 1em">information criterion for
clustering search (AIC, AICc, BIC). (default is BIC) this
argument applies only for 203 substitution schemes (e.g., -s
203 -H AIC)</p>

<p style="margin-top: 1em">-help</p>

<p style="margin-top: 1em">displays this help message</p>

<p style="margin-top: 1em">-hLRT</p>

<p style="margin-top: 1em">do hierarchical likelihood ratio
tests (default is false) hypothesis testing order can be
specified with -O argument</p>

<p style="margin-top: 1em">-i</p>

<p style="margin-top: 1em">include models with a proportion
invariable sites (e.g., -i) (default is false)</p>

<p style="margin-top: 1em">-machinesfile
manchinesFileName</p>

<p style="margin-top: 1em">gets the processors per host
from a machines file</p>

<p style="margin-top: 1em">-n executionName</p>

<p style="margin-top: 1em">execution name for appending to
the log filenames (default: current time yyyyMMddhhmmss)</p>

<p style="margin-top: 1em">-o outputFileName</p>

<p style="margin-top: 1em">set output file (e.g., -o
jmodeltest.out)</p>

<p style="margin-top: 1em">-O hypothesisOrder</p>

<p style="margin-top: 1em">hypothesis order for the hLRTs
(e.g., -hLRT -O gpftv) (default is ftvwxgp)</p>

<p style="margin-top: 1em">f=freq, t=titvi,
v=2ti4tv(subst=3)/2ti(subst&gt;3), w=2tv, x=4tv, g=gamma,
p=pinv this argument is used only if -hLRT argument is set
&rsquo;f&rsquo;,&rsquo;t&rsquo;,&rsquo;v&rsquo;,&rsquo;g&rsquo;,&rsquo;p&rsquo;
are mandatory in any <br>
order. &rsquo;w&rsquo; is optional, and &rsquo;x&rsquo;
requires &rsquo;w&rsquo; to be present thus, length should
be 5, 6 *including &rsquo;w&rsquo;) or 7 (including both
&rsquo;w&rsquo; and &rsquo;x&rsquo;) e.g., -hLRT -O
gpfvwxt</p>

<p style="margin-top: 1em">-p</p>

<p style="margin-top: 1em">calculate parameter importances
(e.g., -p) (default is false)</p>

<p style="margin-top: 1em">-r</p>

<p style="margin-top: 1em">backward selection for the hLRT
(e.g., -r) (default is forward)</p>

<p style="margin-top: 1em">-s
numberOfSubstitutionSchemes</p>

<p style="margin-top: 1em">number of substitution schemes
(e.g., -s 11) (it has to be 3,5,7,11,203; default is 3)</p>

<p style="margin-top: 1em">--set-local-config
localConfigurationFile</p>

<p style="margin-top: 1em">set a local configuration file
in replacement of conf/jmodeltest.conf</p>

<p style="margin-top: 1em">--set-property
propertyName=propertyValue</p>

<p style="margin-top: 1em">set a new value for a property
contained in the configuration file
(conf/jmodeltest.conf)</p>

<p style="margin-top: 1em">-S NNI|SPR|BEST</p>

<p style="margin-top: 1em">tree topology search operation
option (NNI (fast), SPR (a bit slower), BEST (best of NNI
and SPR)) (default is BEST)</p>

<p style="margin-top: 1em">-t fixed|BIONJ|ML</p>

<p style="margin-top: 1em">base tree for likelihood
calculations (e.g., -t BIONJ)</p>

<p style="margin-top: 1em">fixed (common BIONJ-JC
topology)</p>

<p style="margin-top: 1em">BIONJ (Neighbor-Joining
topology)</p>

<p style="margin-top: 1em">ML (Maximum Likelihood topology)
(default)</p>

<p style="margin-top: 1em">-tr numberOfThreads</p>

<p style="margin-top: 1em">number of threads to execute
(default is 4)</p>

<p style="margin-top: 1em">-u treeFileName</p>

<p style="margin-top: 1em">user tree for likelihood
calculations <br>
(e.g., -u data.tre)</p>

<p style="margin-top: 1em">-uLnL</p>

<p style="margin-top: 1em">calculate delta AIC,AICc,BIC
against unconstrained likelihood (e.g., -uLnL)</p>

<p style="margin-top: 1em">(default is false if the input
alignment has gaps or ambiguous characters)</p>

<p style="margin-top: 1em">-v</p>

<p style="margin-top: 1em">do model averaging and parameter
importances (e.g., -v) (default is false)</p>

<p style="margin-top: 1em">-w</p>

<p style="margin-top: 1em">write PAUP block (e.g., -w)
(default is false)</p>

<p style="margin-top: 1em">-z</p>

<p style="margin-top: 1em">strict consensus type for
model-averaged phylogeny (e.g., -z) (default is majority
rule)</p>

<p style="margin-top: 1em">EXAMPLE <br>
jmodeltest -d sequenceFileName -i -f -g 4 -BIC -AIC -AICc
-DT -v -a -w</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">jmodeltest 2.1.10 August 2016
JMODELTEST(1)</p>
<hr>
</body>
</html>
