<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>JMODELTEST(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">JMODELTEST(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">JMODELTEST(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
jmodeltest - HPC selection of models of nucleotide substitution
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>jmodeltest</b> <b>-d</b> sequenceFileName [-getPhylip] [-ckp
  checkpointFileName.ckp] [-n executionName] [-t fixed|BIONJ|ML] [-u
  userTreeFileName] [-o outputFileName] [-S NNI|SPR|BEST] [-AIC] [-AICc] [-BIC]
  [-DT] [-c confidenceInterval] [-s 3|5|7|11|203] [-f] [-i] [-g
  numberOfCategories] [-uLNL] [-dLRT] [-h confidenceInterval] [-hLRT] [-O
  {ftvwxgp}] [-a] [-z] [-p] [-v] [-w] [-tr numberOfThreads] [-machinesfile
  machinesFileName]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
jModelTest is a tool to carry out statistical selection of best-fit models of
  nucleotide substitution. It implements five different model selection
  strategies: hierarchical and dynamical likelihood ratio tests (hLRT and dLRT),
  Akaike and Bayesian information criteria (AIC and BIC), and a decision theory
  method (DT). It also provides estimates of model selection uncertainty,
  parameter importances and model-averaged parameter estimates, including
  model-averaged tree topologies. jModelTest 2 includes High Performance
  Computing (HPC) capabilities and additional features like new strategies for
  tree optimization, model- averaged phylogenetic trees (both topology and
  branch length), heuristic filtering and automatic logging of user activity.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-a</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">estimate model-averaged phylogeny for each active criterion
      (e.g., <b>-a</b>) (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-AIC</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">calculate the Akaike Information Criterion (e.g.,
      <b>-AIC</b>) (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-AICc</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">calculate the corrected Akaike Information Criterion (e.g.,
      <b>-AICc</b>) (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-BIC</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">calculate the Bayesian Information Criterion (e.g.,
      <b>-BIC</b>) (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-DT</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">calculate the decision theory criterion (e.g., <b>-DT</b>)
      (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-c</b>
  confidenceInterval</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">confidence interval (e.g., <b>-c</b> 90) (default is
    100)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-ckp</b>
  checkpointFileName</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Loads a checkpointing file</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-d</b>
  sequenceFileName</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">input data file (e.g., <b>-d</b> data.phy)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-dLRT</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">do dynamical likelihood ratio tests (e.g.,
      <b>-dLRT</b>)(default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-f</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">include models with unequals base frecuencies (e.g.,
      <b>-f</b>) (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-g</b>
  numberOfCategories</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">include models with rate variation among sites and number
      of categories (e.g., <b>-g</b> 8) (default is false &amp; 4
    categories)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-G</b>
  threshold</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">heuristic search. Requires a threshold &gt; 0 (e.g.,
      <b>-G</b> 0.1)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-getPhylip</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">converts the input file into phylip format</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-h</b>
  confidenceInterval</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">confidence level for the hLRTs (e.g., <b>-a0</b>.002)
      (default is 0.01)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-H</b>
  informationCriterion</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">information criterion for clustering search (AIC, AICc,
      BIC). (default is BIC) this argument applies only for 203 substitution
      schemes (e.g., <b>-s</b> 203 <b>-H</b> AIC)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-help</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">displays this help message</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-hLRT</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">do hierarchical likelihood ratio tests (default is false)
      hypothesis testing order can be specified with <b>-O</b> argument</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-i</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">include models with a proportion invariable sites (e.g.,
      <b>-i</b>) (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-machinesfile</b>
  manchinesFileName</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">gets the processors per host from a machines file</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-n</b>
  executionName</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">execution name for appending to the log filenames (default:
      current time yyyyMMddhhmmss)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-o</b>
  outputFileName</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">set output file (e.g., <b>-o</b> jmodeltest.out)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-O</b>
  hypothesisOrder</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">hypothesis order for the hLRTs (e.g., <b>-hLRT</b>
      <b>-O</b> gpftv) (default is ftvwxgp)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">f=freq, t=titvi, v=2ti4tv(subst=3)/2ti(subst&gt;3), w=2tv,
      x=4tv, g=gamma, p=pinv this argument is used only if <b>-hLRT</b> argument
      is set 'f','t','v','g','p' are mandatory in any order. 'w' is optional,
      and 'x' requires 'w' to be present thus, length should be 5, 6 *including
      'w') or 7 (including both 'w' and 'x') e.g., <b>-hLRT</b> <b>-O</b>
      gpfvwxt</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-p</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">calculate parameter importances (e.g., <b>-p</b>) (default
      is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-r</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">backward selection for the hLRT (e.g., <b>-r</b>) (default
      is forward)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-s</b>
  numberOfSubstitutionSchemes</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">number of substitution schemes (e.g., <b>-s</b> 11) (it has
      to be 3,5,7,11,203; default is 3)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--set-local-config</b>
  localConfigurationFile</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">set a local configuration file in replacement of
      conf/jmodeltest.conf</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>--set-property</b>
  propertyName=propertyValue</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">set a new value for a property contained in the
      configuration file (conf/jmodeltest.conf)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-S</b>
  NNI|SPR|BEST</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">tree topology search operation option (NNI (fast), SPR (a
      bit slower), BEST (best of NNI and SPR)) (default is BEST)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-t</b>
  fixed|BIONJ|ML</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">base tree for likelihood calculations (e.g., <b>-t</b>
      BIONJ)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">fixed (common BIONJ-JC topology)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">BIONJ (Neighbor-Joining topology)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">ML (Maximum Likelihood topology) (default)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-tr</b>
  numberOfThreads</div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">number of threads to execute (default is 4)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-u</b>
  treeFileName</div>
<dl class="Bl-tag">
  <dt class="It-tag">user tree for likelihood calculations</dt>
  <dd class="It-tag">(e.g., <b>-u</b> data.tre)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-uLnL</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">calculate delta AIC,AICc,BIC against unconstrained
      likelihood (e.g., <b>-uLnL</b>)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(default is false if the input alignment has gaps or
      ambiguous characters)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-v</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">do model averaging and parameter importances (e.g.,
      <b>-v</b>) (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-w</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">write PAUP block (e.g., <b>-w</b>) (default is false)</dd>
</dl>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 5.00ex; text-indent: -5.00ex;"><b>-z</b></div>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">strict consensus type for model-averaged phylogeny (e.g.,
      <b>-z</b>) (default is majority rule)</dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLE"><a class="selflink" href="#EXAMPLE">EXAMPLE</a></h1>
jmodeltest <b>-d</b> sequenceFileName <b>-i</b> <b>-f</b> <b>-g</b> 4
  <b>-BIC</b> <b>-AIC</b> <b>-AICc</b> <b>-DT</b> <b>-v</b> <b>-a</b> <b>-w</b>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">August 2016</td>
    <td class="foot-os">jmodeltest 2.1.10</td>
  </tr>
</table>
</body>
</html>
