<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:36:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SAM-STATS(1) User Commands SAM-STATS(1)</p>

<p style="margin-top: 1em">NAME <br>
sam-stats - ea-utils: produce digested statistics</p>

<p style="margin-top: 1em">SYNOPSIS <br>
sam-stats [options] [file1] [file2...filen]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Version: 1.38.681</p>

<p style="margin-top: 1em">Produces lots of easily digested
statistics for the files listed</p>

<p style="margin-top: 1em">Options (default in parens):</p>

<p style="margin-top: 1em">-D Keep track of multiple
alignments -O PREFIX Output prefix enabling extended output
(see below) -R FIL Coverage/RNA output (coverage, 3&rsquo;
bias, etc, <br>
implies -A) -A Report all chr sigs, even if there are more
than 1000 -b INT Number of reads to sample for per-base
stats (1M) -S INT Size of <br>
ascii-signature (30) -x FIL File extension for handling
multiple files (stats) -M Only overwrite if newer (requires
-x, or multiple files) -B <br>
Input is bam, don&rsquo;t bother looking at magic -z
Don&rsquo;t fail when zero entries in sam</p>

<p style="margin-top: 1em">OUTPUT:</p>

<p style="margin-top: 1em">If one file is specified, then
the output is to standard out. If multiple files are
specified, or if the -x option is supplied, the output file
is &lt;filename&gt;.&lt;ext&gt;. Default <br>
extension is &rsquo;stats&rsquo;.</p>

<p style="margin-top: 1em">Complete Stats:</p>

<p style="margin-top: 1em">&lt;STATS&gt; <br>
: mean, max, stdev, median, Q1 (25 percentile), Q3</p>

<p style="margin-top: 1em">reads : # of entries in the sam
file, might not be # reads</p>

<p style="margin-top: 1em">phred : phred scale used</p>

<p style="margin-top: 1em">bsize : # reads used for qual
stats</p>

<p style="margin-top: 1em">mapped reads <br>
: number of aligned reads (unique probe id sequences)</p>

<p style="margin-top: 1em">mapped bases <br>
: total of the lengths of the aligned reads</p>

<p style="margin-top: 1em">forward <br>
: number of forward-aligned reads</p>

<p style="margin-top: 1em">reverse <br>
: number of reverse-aligned reads</p>

<p style="margin-top: 1em">snp rate <br>
: mismatched bases / total bases (snv rate)</p>

<p style="margin-top: 1em">ins rate <br>
: insert bases / total bases</p>

<p style="margin-top: 1em">del rate <br>
: deleted bases / total bases</p>

<p style="margin-top: 1em">pct mismatch <br>
: percent of reads that have mismatches</p>

<p style="margin-top: 1em">pct align <br>
: percent of reads that aligned</p>

<p style="margin-top: 1em">len &lt;STATS&gt; <br>
: read length stats, ignored if fixed-length</p>

<p style="margin-top: 1em">mapq &lt;STATS&gt; <br>
: stats for mapping qualities</p>

<p style="margin-top: 1em">insert &lt;STATS&gt; <br>
: stats for insert sizes</p>

<p style="margin-top: 1em">%&lt;CHR&gt; : percentage of
mapped bases per chr, followed by a signature</p>

<p style="margin-top: 1em">Subsampled stats (1M reads max):
<br>
base qual &lt;STATS&gt; : stats for base qualities
%A,%T,%C,%G : base percentages</p>

<p style="margin-top: 1em">Meaning of the per-chromosome
signature: <br>
A ascii-histogram of mapped reads by chromosome position. It
is only output if the original SAM/BAM has a header. The
values are the log2 of the # of mapped reads at each <br>
position + ascii &rsquo;0&rsquo;.</p>

<p style="margin-top: 1em">Extended output mode produces a
set of files: <br>
.stats : primary output</p>

<p style="margin-top: 1em">.fastx : fastx-toolkit
compatible output</p>

<p style="margin-top: 1em">.rcov : per-reference counts
&amp; coverage</p>

<p style="margin-top: 1em">.xdist : mismatch
distribution</p>

<p style="margin-top: 1em">.ldist : length distribution (if
applicable)</p>

<p style="margin-top: 1em">.mqdist <br>
: mapping quality distribution</p>

<p style="margin-top: 1em">sam-stats 1.1.2 July 2015
SAM-STATS(1)</p>
<hr>
</body>
</html>
