<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>SAM-STATS(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">SAM-STATS(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">SAM-STATS(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
sam-stats - ea-utils: produce digested statistics
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>sam-stats</b> [ <i>options</i>] [<i>file1</i>] [<i>file2</i>...<i>filen</i>]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Version: 1.38.681
<div class="Pp"></div>
Produces lots of easily digested statistics for the files listed
<div class="Pp"></div>
Options (default in parens):
<div class="Pp"></div>
<b>-D</b> Keep track of multiple alignments <b>-O</b> PREFIX Output prefix
  enabling extended output (see below) <b>-R</b> FIL Coverage/RNA output
  (coverage, 3' bias, etc, implies <b>-A</b>) <b>-A</b> Report all chr sigs,
  even if there are more than 1000 <b>-b</b> INT Number of reads to sample for
  per-base stats (1M) <b>-S</b> INT Size of ascii-signature (30) <b>-x</b> FIL
  File extension for handling multiple files (stats) <b>-M</b> Only overwrite if
  newer (requires <b>-x</b>, or multiple files) <b>-B</b> Input is bam, don't
  bother looking at magic <b>-z</b> Don't fail when zero entries in sam
<div class="Pp"></div>
OUTPUT:
<div class="Pp"></div>
If one file is specified, then the output is to standard out. If multiple files
  are specified, or if the <b>-x</b> option is supplied, the output file is
  &lt;filename&gt;.&lt;ext&gt;. Default extension is 'stats'.
<div class="Pp"></div>
Complete Stats:
<dl class="Bl-tag">
  <dt class="It-tag">&lt;STATS&gt;</dt>
  <dd class="It-tag">: mean, max, stdev, median, Q1 (25 percentile), Q3</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reads</dt>
  <dd class="It-tag">: # of entries in the sam file, might not be # reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">phred</dt>
  <dd class="It-tag">: phred scale used</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">bsize</dt>
  <dd class="It-tag">: # reads used for qual stats</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">mapped reads</dt>
  <dd class="It-tag">: number of aligned reads (unique probe id sequences)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">mapped bases</dt>
  <dd class="It-tag">: total of the lengths of the aligned reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">forward</dt>
  <dd class="It-tag">: number of forward-aligned reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">reverse</dt>
  <dd class="It-tag">: number of reverse-aligned reads</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">snp rate</dt>
  <dd class="It-tag">: mismatched bases / total bases (snv rate)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">ins rate</dt>
  <dd class="It-tag">: insert bases / total bases</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">del rate</dt>
  <dd class="It-tag">: deleted bases / total bases</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">pct mismatch</dt>
  <dd class="It-tag">: percent of reads that have mismatches</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">pct align</dt>
  <dd class="It-tag">: percent of reads that aligned</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">len &lt;STATS&gt;</dt>
  <dd class="It-tag">: read length stats, ignored if fixed-length</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">mapq &lt;STATS&gt;</dt>
  <dd class="It-tag">: stats for mapping qualities</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">insert &lt;STATS&gt;</dt>
  <dd class="It-tag">: stats for insert sizes</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">%&lt;CHR&gt;</dt>
  <dd class="It-tag">: percentage of mapped bases per chr, followed by a
      signature</dd>
</dl>
<h2 class="Ss" title="Ss" id="Subsampled_stats_(1M_reads_max):"><a class="selflink" href="#Subsampled_stats_(1M_reads_max):">Subsampled
  stats (1M reads max):</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">base qual &lt;STATS&gt; : stats for base qualities
      %A,%T,%C,%G : base percentages</dd>
</dl>
<h2 class="Ss" title="Ss" id="Meaning_of_the_per-chromosome_signature:"><a class="selflink" href="#Meaning_of_the_per-chromosome_signature:">Meaning
  of the per-chromosome signature:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">A ascii-histogram of mapped reads by chromosome position.
      It is only output if the original SAM/BAM has a header. The values are the
      log2 of the # of mapped reads at each position + ascii '0'.</dd>
</dl>
<h2 class="Ss" title="Ss" id="Extended_output_mode_produces_a_set_of_files:"><a class="selflink" href="#Extended_output_mode_produces_a_set_of_files:">Extended
  output mode produces a set of files:</a></h2>
<dl class="Bl-tag">
  <dt class="It-tag">.stats</dt>
  <dd class="It-tag">: primary output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">.fastx</dt>
  <dd class="It-tag">: fastx-toolkit compatible output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">.rcov</dt>
  <dd class="It-tag">: per-reference counts &amp; coverage</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">.xdist</dt>
  <dd class="It-tag">: mismatch distribution</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">.ldist</dt>
  <dd class="It-tag">: length distribution (if applicable)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">.mqdist</dt>
  <dd class="It-tag">: mapping quality distribution</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2015</td>
    <td class="foot-os">sam-stats 1.1.2</td>
  </tr>
</table>
</body>
</html>
