<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>SAMFILTER(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">SAMFILTER(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">SAMFILTER(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
samFilter - filter nucleotide sequence alignments in SAM files
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>samFilter</b> <i>file.sam</i> <i>reference.fasta</i> <i>out.sam</i>
  [<i>options</i>]
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><i>file.sam</i></dt>
  <dd class="It-tag">Input SAM file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>reference.fasta</i></dt>
  <dd class="It-tag">Reference used to generate reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>out.sam</i></dt>
  <dd class="It-tag">Output SAM file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minAlnLength</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">(50) Report alignments only if their lengths are greater
      than <i>value</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minAlignLength</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Alias of <b>-minAlnLength</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minLength</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Alias of <b>-minAlnLength</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minPctSimilarity</b><i>&#x2002;value</i></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">(70) Report alignments only if their percentage similairty
      is greater than <i>value&#x2002;.</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minPctIdentity</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Alias of <b>-minPctSimilarity</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minPctAccuracy</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">(70) Report alignments only if their percentage accuray is
      greater than <i>value</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-minAccuracy</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Alias of <b>-minPctAccuracy</b></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-hitPolicy</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">(randombest) Specify a policy to treat multiple hits from
      [all, allbest, random, randombest, leftmost]</dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><i>all</i></dt>
  <dd class="It-tag">report all alignments.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>allbest</i></dt>
  <dd class="It-tag">report all equally top scoring alignments.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>random</i></dt>
  <dd class="It-tag">report a random alignment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>randombest</i></dt>
  <dd class="It-tag">report a random alignment from multiple equally top scoring
      alignments.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>leftmost</i></dt>
  <dd class="It-tag">report an alignment which has the best alignmentscore and
      has the smallest mapping coordinate in any reference.</dd>
</dl>
</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-scoreSign</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">(-1) Whether higher or lower scores are better.</dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag">-1</dt>
  <dd class="It-tag">lower is better</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">1</dt>
  <dd class="It-tag">higher is better.</dd>
</dl>
</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-scoreCutoff</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">(INF) Report alignments only if their scores are no worse
      than <i>value</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-seed</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">(1) Seed for random number generator. If seed is 0, then
      use current time as seed.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-holeNumbers</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">A string of comma-delimited hole number ranges to output
      hits, such as '1,2,10-12'. This requires hit titles to be in SMRT read
      title format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-smrtTitle</b></dt>
  <dd class="It-tag">Use this option when filtering alignments generated by
      programs other than <b>blasr</b>(1), e.g. bwa-sw or gmap. Parse read
      coordinates from the SMRT read title. The title is in the format
      <i>/name/hole/coordinates</i>, where coordinates are in the format
      \d+_\d+, and represent the interval of the read that was aligned.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-titleTable</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Use this experimental option when filtering alignments
      generated by <b>blasr</b>(1) with <b>-titleTable</b> titleTableName, in
      which case reference titles in SAM are represented by their indices (e.g.,
      0, 1, 2, ...) in the title table.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-filterAdapterOnly</b><i>&#x2002;value</i></dt>
  <dd class="It-tag">Use this option to remove reads which can only map to
      adapters specified in the GFF file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b></dt>
  <dd class="It-tag">Be verbose.</dd>
</dl>
<h1 class="Sh" title="Sh" id="NOTES"><a class="selflink" href="#NOTES">NOTES</a></h1>
Because SAM has optional tags that have different meanings in different
  programs, careful usage is required in order to have proper output. The
  &quot;xs&quot; tag in bwa-sw is used to show the suboptimal score, but in
  PacBio SAM (<b>blasr</b>(1)) it is defined as the start in the query sequence
  of the alignment. When <b>-smrtTitle</b> is specified, the xs tag is ignored,
  but when it is not specified, the coordinates given by the xs and xe tags are
  used to define the interval of a read that is aligned. The CIGAR string is
  relative to this interval.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>blasr</b>(1) <b>loadPulses</b>(1) <b>pls2fasta</b>(1) <b>samtoh5</b>(1)
  <b>samtom4</b>(1) <b>sawriter</b>(1) <b>sdpMatcher</b>(1)
  <b>toAfg</b>(1)</div>
<table class="foot">
  <tr>
    <td class="foot-date">July 2015</td>
    <td class="foot-os">samFilter 3ca7fe8</td>
  </tr>
</table>
</body>
</html>
