<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:36:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SAMFILTER(1) User Commands SAMFILTER(1)</p>

<p style="margin-top: 1em">NAME <br>
samFilter - filter nucleotide sequence alignments in SAM
files</p>

<p style="margin-top: 1em">SYNOPSIS <br>
samFilter file.sam reference.fasta out.sam [options]</p>

<p style="margin-top: 1em">OPTIONS <br>
file.sam <br>
Input SAM file.</p>

<p style="margin-top: 1em">reference.fasta <br>
Reference used to generate reads.</p>

<p style="margin-top: 1em">out.sam <br>
Output SAM file.</p>

<p style="margin-top: 1em">-minAlnLength value <br>
(50) Report alignments only if their lengths are greater
than value.</p>

<p style="margin-top: 1em">-minAlignLength value <br>
Alias of -minAlnLength</p>

<p style="margin-top: 1em">-minLength value <br>
Alias of -minAlnLength</p>

<p style="margin-top: 1em">-minPctSimilarity value</p>

<p style="margin-top: 1em">(70) Report alignments only if
their percentage similairty is greater than value .</p>

<p style="margin-top: 1em">-minPctIdentity value <br>
Alias of -minPctSimilarity</p>

<p style="margin-top: 1em">-minPctAccuracy value <br>
(70) Report alignments only if their percentage accuray is
greater than value.</p>

<p style="margin-top: 1em">-minAccuracy value <br>
Alias of -minPctAccuracy</p>

<p style="margin-top: 1em">-hitPolicy value <br>
(randombest) Specify a policy to treat multiple hits from
[all, allbest, random, randombest, leftmost]</p>

<p style="margin-top: 1em">all report all alignments.</p>

<p style="margin-top: 1em">allbest <br>
report all equally top scoring alignments.</p>

<p style="margin-top: 1em">random report a random
alignment.</p>

<p style="margin-top: 1em">randombest <br>
report a random alignment from multiple equally top scoring
alignments.</p>

<p style="margin-top: 1em">leftmost <br>
report an alignment which has the best alignmentscore and
has the smallest mapping coordinate in any reference.</p>

<p style="margin-top: 1em">-scoreSign value <br>
(-1) Whether higher or lower scores are better.</p>

<p style="margin-top: 1em">-1 lower is better</p>

<p style="margin-top: 1em">1 higher is better.</p>

<p style="margin-top: 1em">-scoreCutoff value <br>
(INF) Report alignments only if their scores are no worse
than value.</p>

<p style="margin-top: 1em">-seed value <br>
(1) Seed for random number generator. If seed is 0, then use
current time as seed.</p>

<p style="margin-top: 1em">-holeNumbers value <br>
A string of comma-delimited hole number ranges to output
hits, such as &rsquo;1,2,10-12&rsquo;. This requires hit
titles to be in SMRT read title format.</p>

<p style="margin-top: 1em">-smrtTitle <br>
Use this option when filtering alignments generated by
programs other than blasr(1), e.g. bwa-sw or gmap. Parse
read coordinates from the SMRT read title. The title is in
<br>
the format /name/hole/coordinates, where coordinates are in
the format +_+, and represent the interval of the read that
was aligned.</p>

<p style="margin-top: 1em">-titleTable value <br>
Use this experimental option when filtering alignments
generated by blasr(1) with -titleTable titleTableName, in
which case reference titles in SAM are represented by <br>
their indices (e.g., 0, 1, 2, ...) in the title table.</p>

<p style="margin-top: 1em">-filterAdapterOnly value <br>
Use this option to remove reads which can only map to
adapters specified in the GFF file.</p>

<p style="margin-top: 1em">-v Be verbose.</p>

<p style="margin-top: 1em">NOTES <br>
Because SAM has optional tags that have different meanings
in different programs, careful usage is required in order to
have proper output. The &quot;xs&quot; tag in bwa-sw is used
to <br>
show the suboptimal score, but in PacBio SAM (blasr(1)) it
is defined as the start in the query sequence of the
alignment. When -smrtTitle is specified, the xs tag is
ignored, <br>
but when it is not specified, the coordinates given by the
xs and xe tags are used to define the interval of a read
that is aligned. The CIGAR string is relative to this
inter&acirc; <br>
val.</p>

<p style="margin-top: 1em">SEE ALSO <br>
blasr(1) loadPulses(1) pls2fasta(1) samtoh5(1) samtom4(1)
sawriter(1) sdpMatcher(1) toAfg(1)</p>

<p style="margin-top: 1em">samFilter 3ca7fe8 July 2015
SAMFILTER(1)</p>
<hr>
</body>
</html>
