<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:36:49 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>samtools(1) Bioinformatics tools samtools(1)</p>

<p style="margin-top: 1em">NAME <br>
samtools - Utilities for the Sequence Alignment/Map (SAM)
format</p>

<p style="margin-top: 1em">bcftools - Utilities for the
Binary Call Format (BCF) and VCF</p>

<p style="margin-top: 1em">SYNOPSIS <br>
samtools view -bt ref_list.txt -o aln.bam aln.sam.gz</p>

<p style="margin-top: 1em">samtools sort aln.bam
aln.sorted</p>

<p style="margin-top: 1em">samtools index
aln.sorted.bam</p>

<p style="margin-top: 1em">samtools idxstats
aln.sorted.bam</p>

<p style="margin-top: 1em">samtools view aln.sorted.bam
chr2:20,100,000-20,200,000</p>

<p style="margin-top: 1em">samtools merge out.bam in1.bam
in2.bam in3.bam</p>

<p style="margin-top: 1em">samtools faidx ref.fasta</p>

<p style="margin-top: 1em">samtools pileup -vcf ref.fasta
aln.sorted.bam</p>

<p style="margin-top: 1em">samtools mpileup -C50 -gf
ref.fasta -r chr3:1,000-2,000 in1.bam in2.bam</p>

<p style="margin-top: 1em">samtools tview aln.sorted.bam
ref.fasta</p>

<p style="margin-top: 1em">bcftools index in.bcf</p>

<p style="margin-top: 1em">bcftools view in.bcf
chr2:100-200 &gt; out.vcf</p>

<p style="margin-top: 1em">bcftools view -Nvm0.99 in.bcf
&gt; out.vcf 2&gt; out.afs</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Samtools is a set of utilities that manipulate alignments in
the BAM format. It imports from and exports to the SAM
(Sequence Alignment/Map) format, does sorting, merging and
<br>
indexing, and allows to retrieve reads in any regions
swiftly.</p>

<p style="margin-top: 1em">Samtools is designed to work on
a stream. It regards an input file &lsquo;-&rsquo; as the
standard input (stdin) and an output file &lsquo;-&rsquo; as
the standard output (stdout). Several commands can <br>
thus be combined with Unix pipes. Samtools always output
warning and error messages to the standard error output
(stderr).</p>

<p style="margin-top: 1em">Samtools is also able to open a
BAM (not SAM) file on a remote FTP or HTTP server if the BAM
file name starts with &lsquo;ftp://&rsquo; or
&lsquo;http://&rsquo;. Samtools checks the current working
<br>
directory for the index file and will download the index
upon absence. Samtools does not retrieve the entire
alignment file unless it is asked to do so.</p>

<p style="margin-top: 1em">SAMTOOLS COMMANDS AND OPTIONS
<br>
view samtools view [-bchuHS] [-t in.refList] [-o output] [-f
reqFlag] [-F skipFlag] [-q minMapQ] [-l library] [-r
readGroup] [-R rgFile] &lt;in.bam&gt;|&lt;in.sam&gt;
[region1 [...]]</p>

<p style="margin-top: 1em">Extract/print all or sub
alignments in SAM or BAM format. If no region is specified,
all the alignments will be printed; otherwise only
alignments overlapping the spec&acirc; <br>
ified regions will be output. An alignment may be given
multiple times if it is overlapping several regions. A
region can be presented, for example, in the following <br>
format: &lsquo;chr2&rsquo; (the whole chr2),
&lsquo;chr2:1000000&rsquo; (region starting from
1,000,000bp) or &lsquo;chr2:1,000,000-2,000,000&rsquo;
(region between 1,000,000 and 2,000,000bp including the <br>
end points). The coordinate is 1-based.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-b Output in the BAM format.</p>

<p style="margin-top: 1em">-f INT Only output alignments
with all bits in INT present in the FLAG field. INT can be
in hex in the format of /^0x[0-9A-F]+/ [0]</p>

<p style="margin-top: 1em">-F INT Skip alignments with bits
present in INT [0]</p>

<p style="margin-top: 1em">-h Include the header in the
output.</p>

<p style="margin-top: 1em">-H Output the header only.</p>

<p style="margin-top: 1em">-l STR Only output reads in
library STR [null]</p>

<p style="margin-top: 1em">-o FILE Output file [stdout]</p>

<p style="margin-top: 1em">-q INT Skip alignments with MAPQ
smaller than INT [0]</p>

<p style="margin-top: 1em">-r STR Only output reads in read
group STR [null]</p>

<p style="margin-top: 1em">-R FILE Output reads in read
groups listed in FILE [null]</p>

<p style="margin-top: 1em">-s FLOAT Fraction of
templates/pairs to subsample; the integer part is treated as
the seed for the random number generator [-1]</p>

<p style="margin-top: 1em">-S Input is in SAM. If @SQ
header lines are absent, the &lsquo;-t&rsquo; option is
required.</p>

<p style="margin-top: 1em">-c Instead of printing the
alignments, only count them and print the total number. All
filter options, such as &lsquo;-f&rsquo;, &lsquo;-F&rsquo;
and &lsquo;-q&rsquo; , are taken into account.</p>

<p style="margin-top: 1em">-t FILE This file is
TAB-delimited. Each line must contain the reference name and
the length of the reference, one line for each distinct
reference; additional fields <br>
are ignored. This file also defines the order of the
reference sequences in sorting. If you run &lsquo;samtools
faidx &lt;ref.fa&gt;&rsquo;, the resultant index file <br>
&lt;ref.fa&gt;.fai can be used as this &lt;in.ref_list&gt;
file.</p>

<p style="margin-top: 1em">-u Output uncompressed BAM. This
option saves time spent on compression/decomprssion and is
thus preferred when the output is piped to another samtools
command.</p>

<p style="margin-top: 1em">tview samtools tview [-p
chr:pos] [-s STR] [-d display] &lt;in.sorted.bam&gt;
[ref.fasta]</p>

<p style="margin-top: 1em">Text alignment viewer (based on
the ncurses library). In the viewer, press &lsquo;?&rsquo;
for help and press &lsquo;g&rsquo; to check the alignment
start from a region in the format like <br>
&lsquo;chr10:10,000,000&rsquo; or &lsquo;=10,000,000&rsquo;
when viewing the same reference sequence.</p>

<p style="margin-top: 1em">Options:</p>

<p style="margin-top: 1em">-d display Output as (H)tml or
(C)urses or (T)ext</p>

<p style="margin-top: 1em">-p chr:pos Go directly to this
position</p>

<p style="margin-top: 1em">-s STR Display only reads from
this sample or read group</p>

<p style="margin-top: 1em">mpileup samtools mpileup
[-EBugp] [-C capQcoef] [-r reg] [-f in.fa] [-l list] [-M
capMapQ] [-Q minBaseQ] [-q minMapQ] in.bam [in2.bam
[...]]</p>

<p style="margin-top: 1em">Generate BCF or pileup for one
or multiple BAM files. Alignment records are grouped by
sample identifiers in @RG header lines. If sample
identifiers are absent, each <br>
input file is regarded as one sample.</p>

<p style="margin-top: 1em">In the pileup format (without
-uor-g), each line represents a genomic position, consisting
of chromosome name, coordinate, reference base, read bases,
read qualities <br>
and alignment mapping qualities. Information on match,
mismatch, indel, strand, mapping quality and start and end
of a read are all encoded at the read base column. At <br>
this column, a dot stands for a match to the reference base
on the forward strand, a comma for a match on the reverse
strand, a &rsquo;&gt;&rsquo; or &rsquo;&lt;&rsquo; for a
reference skip, &lsquo;ACGTN&rsquo; <br>
for a mismatch on the forward strand and &lsquo;acgtn&rsquo;
for a mismatch on the reverse strand. A pattern
&lsquo;+[0-9]+[ACGTNacgtn]+&rsquo; indicates there is an
insertion between this <br>
reference position and the next reference position. The
length of the insertion is given by the integer in the
pattern, followed by the inserted sequence. Similarly, a
<br>
pattern &lsquo;-[0-9]+[ACGTNacgtn]+&rsquo; represents a
deletion from the reference. The deleted bases will be
presented as &lsquo;*&rsquo; in the following lines. Also at
the read base col&acirc; <br>
umn, a symbol &lsquo;^&rsquo; marks the start of a read. The
ASCII of the character following &lsquo;^&rsquo; minus 33
gives the mapping quality. A symbol &lsquo;$&rsquo; marks
the end of a read segment.</p>

<p style="margin-top: 1em">Input Options:</p>

<p style="margin-top: 1em">-6 Assume the quality is in the
Illumina 1.3+ encoding. -A Do not skip anomalous read pairs
in variant calling.</p>

<p style="margin-top: 1em">-B Disable probabilistic
realignment for the computation of base alignment quality
(BAQ). BAQ is the Phred-scaled probability of a read base
being misaligned. <br>
Applying this option greatly helps to reduce false SNPs
caused by misalignments.</p>

<p style="margin-top: 1em">-b FILE List of input BAM files,
one file per line [null]</p>

<p style="margin-top: 1em">-C INT Coefficient for
downgrading mapping quality for reads containing excessive
mismatches. Given a read with a phred-scaled probability q
of being generated from <br>
the mapped position, the new mapping quality is about
sqrt((INT-q)/INT)*INT. A zero value disables this
functionality; if enabled, the recommended value for <br>
BWA is 50. [0]</p>

<p style="margin-top: 1em">-d INT At a position, read
maximally INT reads per input BAM. [250]</p>

<p style="margin-top: 1em">-E Extended BAQ computation.
This option helps sensitivity especially for MNPs, but may
hurt specificity a little bit.</p>

<p style="margin-top: 1em">-f FILE The faidx-indexed
reference file in the FASTA format. The file can be
optionally compressed by razip. [null]</p>

<p style="margin-top: 1em">-l FILE BED or position list
file containing a list of regions or sites where pileup or
BCF should be generated [null]</p>

<p style="margin-top: 1em">-q INT Minimum mapping quality
for an alignment to be used [0]</p>

<p style="margin-top: 1em">-Q INT Minimum base quality for
a base to be considered [13]</p>

<p style="margin-top: 1em">-r STR Only generate pileup in
region STR [all sites]</p>

<p style="margin-top: 1em">Output Options:</p>

<p style="margin-top: 1em">-D Output per-sample read
depth</p>

<p style="margin-top: 1em">-g Compute genotype likelihoods
and output them in the binary call format (BCF).</p>

<p style="margin-top: 1em">-S Output per-sample
Phred-scaled strand bias P-value</p>

<p style="margin-top: 1em">-u Similar to -g except that the
output is uncompressed BCF, which is preferred for
piping.</p>

<p style="margin-top: 1em">Options for Genotype Likelihood
Computation (for -g or -u):</p>

<p style="margin-top: 1em">-e INT Phred-scaled gap
extension sequencing error probability. Reducing INT leads
to longer indels. [20]</p>

<p style="margin-top: 1em">-h INT Coefficient for modeling
homopolymer errors. Given an l-long homopolymer run, the
sequencing error of an indel of size s is modeled as
INT*s/l. [100]</p>

<p style="margin-top: 1em">-I Do not perform INDEL
calling</p>

<p style="margin-top: 1em">-L INT Skip INDEL calling if the
average per-sample depth is above INT. [250]</p>

<p style="margin-top: 1em">-o INT Phred-scaled gap open
sequencing error probability. Reducing INT leads to more
indel calls. [40]</p>

<p style="margin-top: 1em">-p Apply -m and -F thresholds
per sample to increase sensitivity of calling. By default
both options are applied to reads pooled from all
samples.</p>

<p style="margin-top: 1em">-P STR Comma dilimited list of
platforms (determined by @RG-PL) from which indel candidates
are obtained. It is recommended to collect indel candidates
from sequenc&acirc; <br>
ing technologies that have low indel error rate such as
ILLUMINA. [all]</p>

<p style="margin-top: 1em">reheader samtools reheader
&lt;in.header.sam&gt; &lt;in.bam&gt;</p>

<p style="margin-top: 1em">Replace the header in in.bam
with the header in in.header.sam. This command is much
faster than replacing the header with a BAM-&gt;SAM-&gt;BAM
conversion.</p>

<p style="margin-top: 1em">cat samtools cat [-h header.sam]
[-o out.bam] &lt;in1.bam&gt; &lt;in2.bam&gt; [ ... ]</p>

<p style="margin-top: 1em">Concatenate BAMs. The sequence
dictionary of each input BAM must be identical, although
this command does not check this. This command uses a
similar trick to reheader <br>
which enables fast BAM concatenation.</p>

<p style="margin-top: 1em">sort samtools sort [-nof] [-m
maxMem] &lt;in.bam&gt; &lt;out.prefix&gt;</p>

<p style="margin-top: 1em">Sort alignments by leftmost
coordinates. File &lt;out.prefix&gt;.bam will be created.
This command may also create temporary files
&lt;out.prefix&gt;.%d.bam when the whole align&acirc; <br>
ment cannot be fitted into memory (controlled by option
-m).</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-o Output the final alignment to
the standard output.</p>

<p style="margin-top: 1em">-n Sort by read names rather
than by chromosomal coordinates</p>

<p style="margin-top: 1em">-f Use &lt;out.prefix&gt; as the
full output path and do not append .bam suffix.</p>

<p style="margin-top: 1em">-m INT Approximately the maximum
required memory. [500000000]</p>

<p style="margin-top: 1em">merge samtools merge [-nur1f]
[-h inh.sam] [-R reg] &lt;out.bam&gt; &lt;in1.bam&gt;
&lt;in2.bam&gt; [...]</p>

<p style="margin-top: 1em">Merge multiple sorted
alignments. The header reference lists of all the input BAM
files, and the @SQ headers of inh.sam, if any, must all
refer to the same set of ref&acirc; <br>
erence sequences. The header reference list and (unless
overridden by -h) &lsquo;@&rsquo; headers of in1.bam will be
copied to out.bam, and the headers of other files will be
<br>
ignored.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-1 Use zlib compression level 1
to comrpess the output</p>

<p style="margin-top: 1em">-f Force to overwrite the output
file if present.</p>

<p style="margin-top: 1em">-h FILE Use the lines of FILE as
&lsquo;@&rsquo; headers to be copied to out.bam, replacing
any header lines that would otherwise be copied from
in1.bam. (FILE is actually in SAM <br>
format, though any alignment records it may contain are
ignored.)</p>

<p style="margin-top: 1em">-n The input alignments are
sorted by read names rather than by chromosomal
coordinates</p>

<p style="margin-top: 1em">-R STR Merge files in the
specified region indicated by STR [null]</p>

<p style="margin-top: 1em">-r Attach an RG tag to each
alignment. The tag value is inferred from file names.</p>

<p style="margin-top: 1em">-u Uncompressed BAM output</p>

<p style="margin-top: 1em">index samtools index
&lt;aln.bam&gt;</p>

<p style="margin-top: 1em">Index sorted alignment for fast
random access. Index file &lt;aln.bam&gt;.bai will be
created.</p>

<p style="margin-top: 1em">idxstats samtools idxstats
&lt;aln.bam&gt;</p>

<p style="margin-top: 1em">Retrieve and print stats in the
index file. The output is TAB delimited with each line
consisting of reference sequence name, sequence length, #
mapped reads and # <br>
unmapped reads.</p>

<p style="margin-top: 1em">faidx samtools faidx
&lt;ref.fasta&gt; [region1 [...]]</p>

<p style="margin-top: 1em">Index reference sequence in the
FASTA format or extract subsequence from indexed reference
sequence. If no region is specified, faidx will index the
file and create <br>
&lt;ref.fasta&gt;.fai on the disk. If regions are
speficified, the subsequences will be retrieved and printed
to stdout in the FASTA format. The input file can be
compressed <br>
in the RAZF format.</p>

<p style="margin-top: 1em">fixmate samtools fixmate
&lt;in.nameSrt.bam&gt; &lt;out.bam&gt;</p>

<p style="margin-top: 1em">Fill in mate coordinates, ISIZE
and mate related flags from a name-sorted alignment.</p>

<p style="margin-top: 1em">rmdup samtools rmdup [-sS]
&lt;input.srt.bam&gt; &lt;out.bam&gt;</p>

<p style="margin-top: 1em">Remove potential PCR duplicates:
if multiple read pairs have identical external coordinates,
only retain the pair with highest mapping quality. In the
paired-end mode, <br>
this command ONLY works with FR orientation and requires
ISIZE is correctly set. It does not work for unpaired reads
(e.g. two ends mapped to different chromosomes or <br>
orphan reads).</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-s Remove duplicate for
single-end reads. By default, the command works for
paired-end reads only.</p>

<p style="margin-top: 1em">-S Treat paired-end reads and
single-end reads.</p>

<p style="margin-top: 1em">calmd samtools calmd [-EeubSr]
[-C capQcoef] &lt;aln.bam&gt; &lt;ref.fasta&gt;</p>

<p style="margin-top: 1em">Generate the MD tag. If the MD
tag is already present, this command will give a warning if
the MD tag generated is different from the existing tag.
Output SAM by <br>
default.</p>

<p style="margin-top: 1em">OPTIONS:</p>

<p style="margin-top: 1em">-A When used jointly with -r
this option overwrites the original base quality.</p>

<p style="margin-top: 1em">-e Convert a the read base to =
if it is identical to the aligned reference base. Indel
caller does not support the = bases at the moment.</p>

<p style="margin-top: 1em">-u Output uncompressed BAM</p>

<p style="margin-top: 1em">-b Output compressed BAM</p>

<p style="margin-top: 1em">-S The input is SAM with header
lines</p>

<p style="margin-top: 1em">-C INT Coefficient to cap
mapping quality of poorly mapped reads. See the pileup
command for details. [0]</p>

<p style="margin-top: 1em">-r Compute the BQ tag (without
-A) or cap base quality by BAQ (with -A).</p>

<p style="margin-top: 1em">-E Extended BAQ calculation.
This option trades specificity for sensitivity, though the
effect is minor.</p>

<p style="margin-top: 1em">targetcut samtools targetcut [-Q
minBaseQ] [-i inPenalty] [-0 em0] [-1 em1] [-2 em2] [-f ref]
&lt;in.bam&gt;</p>

<p style="margin-top: 1em">This command identifies target
regions by examining the continuity of read depth, computes
haploid consensus sequences of targets and outputs a SAM
with each sequence <br>
corresponding to a target. When option -f is in use, BAQ
will be applied. This command is only designed for cutting
fosmid clones from fosmid pool sequencing [Ref. <br>
Kitzman et al. (2010)].</p>

<p style="margin-top: 1em">phase samtools phase [-AF] [-k
len] [-b prefix] [-q minLOD] [-Q minBaseQ]
&lt;in.bam&gt;</p>

<p style="margin-top: 1em">Call and phase heterozygous
SNPs. OPTIONS:</p>

<p style="margin-top: 1em">-A Drop reads with ambiguous
phase.</p>

<p style="margin-top: 1em">-b STR Prefix of BAM output.
When this option is in use, phase-0 reads will be saved in
file STR.0.bam and phase-1 reads in STR.1.bam. Phase unknown
reads will be <br>
randomly allocated to one of the two files. Chimeric reads
with switch errors will be saved in STR.chimeric.bam.
[null]</p>

<p style="margin-top: 1em">-F Do not attempt to fix
chimeric reads.</p>

<p style="margin-top: 1em">-k INT Maximum length for local
phasing. [13]</p>

<p style="margin-top: 1em">-q INT Minimum Phred-scaled LOD
to call a heterozygote. [40]</p>

<p style="margin-top: 1em">-Q INT Minimum base quality to
be used in het calling. [13]</p>

<p style="margin-top: 1em">BCFTOOLS COMMANDS AND OPTIONS
<br>
view bcftools view [-AbFGNQSucgv] [-D seqDict] [-l listLoci]
[-s listSample] [-i gapSNPratio] [-t mutRate] [-p varThres]
[-m varThres] [-P prior] [-1 nGroup1] [-d minFrac] <br>
[-U nPerm] [-X permThres] [-T trioType] in.bcf [region]</p>

<p style="margin-top: 1em">Convert between BCF and VCF,
call variant candidates and estimate allele frequencies.</p>

<p style="margin-top: 1em">Input/Output Options:</p>

<p style="margin-top: 1em">-A Retain all possible alternate
alleles at variant sites. By default, the view command
discards unlikely alleles.</p>

<p style="margin-top: 1em">-b Output in the BCF format. The
default is VCF.</p>

<p style="margin-top: 1em">-D FILE Sequence dictionary
(list of chromosome names) for VCF-&gt;BCF conversion
[null]</p>

<p style="margin-top: 1em">-F Indicate PL is generated by
r921 or before (ordering is different).</p>

<p style="margin-top: 1em">-G Suppress all individual
genotype information.</p>

<p style="margin-top: 1em">-l FILE List of sites at which
information are outputted [all sites]</p>

<p style="margin-top: 1em">-N Skip sites where the REF
field is not A/C/G/T</p>

<p style="margin-top: 1em">-Q Output the QCALL likelihood
format</p>

<p style="margin-top: 1em">-s FILE List of samples to use.
The first column in the input gives the sample names and the
second gives the ploidy, which can only be 1 or 2. When the
2nd column is <br>
absent, the sample ploidy is assumed to be 2. In the output,
the ordering of samples will be identical to the one in
FILE. [null]</p>

<p style="margin-top: 1em">-S The input is VCF instead of
BCF.</p>

<p style="margin-top: 1em">-u Uncompressed BCF output
(force -b).</p>

<p style="margin-top: 1em">Consensus/Variant Calling
Options:</p>

<p style="margin-top: 1em">-c Call variants using Bayesian
inference. This option automatically invokes option -e.</p>

<p style="margin-top: 1em">-d FLOAT When -v is in use, skip
loci where the fraction of samples covered by reads is below
FLOAT. [0]</p>

<p style="margin-top: 1em">-e Perform max-likelihood
inference only, including estimating the site allele
frequency, testing Hardy-Weinberg equlibrium and testing
associations with LRT.</p>

<p style="margin-top: 1em">-g Call per-sample genotypes at
variant sites (force -c)</p>

<p style="margin-top: 1em">-i FLOAT Ratio of INDEL-to-SNP
mutation rate [0.15]</p>

<p style="margin-top: 1em">-m FLOAT New model for improved
multiallelic and rare-variant calling. Another ALT allele is
accepted if P(chi^2) of LRT exceeds the FLOAT threshold. The
parameter <br>
seems robust and the actual value usually does not affect
the results much; a good value to use is 0.99. This is the
recommended calling method. [0]</p>

<p style="margin-top: 1em">-p FLOAT A site is considered to
be a variant if P(ref|D)&lt;FLOAT [0.5]</p>

<p style="margin-top: 1em">-P STR Prior or initial allele
frequency spectrum. If STR can be full, cond2, flat or the
file consisting of error output from a previous variant
calling run.</p>

<p style="margin-top: 1em">-t FLOAT Scaled muttion rate for
variant calling [0.001]</p>

<p style="margin-top: 1em">-T STR Enable pair/trio calling.
For trio calling, option -s is usually needed to be applied
to configure the trio members and their ordering. In the
file supplied <br>
to the option -s, the first sample must be the child, the
second the father and the third the mother. The valid values
of STR are &lsquo;pair&rsquo;, &lsquo;trioauto&rsquo;,
&lsquo;tri&acirc; <br>
oxd&rsquo; and &lsquo;trioxs&rsquo;, where
&lsquo;pair&rsquo; calls differences between two input
samples, and &lsquo;trioxd&rsquo; (&lsquo;trioxs&rsquo;)
specifies that the input is from the X chromosome non-PAR
<br>
regions and the child is a female (male). [null]</p>

<p style="margin-top: 1em">-v Output variant sites only
(force -c)</p>

<p style="margin-top: 1em">Contrast Calling and Association
Test Options:</p>

<p style="margin-top: 1em">-1 INT Number of group-1
samples. This option is used for dividing the samples into
two groups for contrast SNP calling or association test.
When this option is in <br>
use, the following VCF INFO will be outputted: PC2, PCHI2
and QCHI2. [0]</p>

<p style="margin-top: 1em">-U INT Number of permutations
for association test (effective only with -1) [0]</p>

<p style="margin-top: 1em">-X FLOAT Only perform
permutations for P(chi^2)&lt;FLOAT (effective only with -U)
[0.01]</p>

<p style="margin-top: 1em">index bcftools index in.bcf</p>

<p style="margin-top: 1em">Index sorted BCF for random
access.</p>

<p style="margin-top: 1em">cat bcftools cat in1.bcf
[in2.bcf [...]]]</p>

<p style="margin-top: 1em">Concatenate BCF files. The input
files are required to be sorted and have identical samples
appearing in the same order.</p>

<p style="margin-top: 1em">SAM FORMAT <br>
Sequence Alignment/Map (SAM) format is TAB-delimited. Apart
from the header lines, which are started with the
&lsquo;@&rsquo; symbol, each alignment line consists of:</p>


<p style="margin-top: 1em">&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;
<br>
&acirc;Col &acirc; Field &acirc; Description &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc; 1 &acirc; QNAME &acirc; Query template/pair NAME
&acirc; <br>
&acirc; 2 &acirc; FLAG &acirc; bitwise FLAG &acirc; <br>
&acirc; 3 &acirc; RNAME &acirc; Reference sequence NAME
&acirc; <br>
&acirc; 4 &acirc; POS &acirc; 1-based leftmost
POSition/coordinate of clipped sequence &acirc; <br>
&acirc; 5 &acirc; MAPQ &acirc; MAPping Quality
(Phred-scaled) &acirc; <br>
&acirc; 6 &acirc; CIAGR &acirc; extended CIGAR string
&acirc; <br>
&acirc; 7 &acirc; MRNM &acirc; Mate Reference sequence NaMe
(&lsquo;=&rsquo; if same as RNAME) &acirc; <br>
&acirc; 8 &acirc; MPOS &acirc; 1-based Mate POSistion
&acirc; <br>
&acirc; 9 &acirc; TLEN &acirc; inferred Template LENgth
(insert size) &acirc; <br>
&acirc;10 &acirc; SEQ &acirc; query SEQuence on the same
strand as the reference &acirc; <br>
&acirc;11 &acirc; QUAL &acirc; query QUALity (ASCII-33 gives
the Phred base quality) &acirc; <br>
&acirc;12+ &acirc; OPT &acirc; variable OPTional fields in
the format TAG:VTYPE:VALUE &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;</p>

<p style="margin-top: 1em">Each bit in the FLAG field is
defined as:</p>


<p style="margin-top: 1em">&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;
<br>
&acirc; Flag &acirc; Chr &acirc; Description &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc;0x0001 &acirc; p &acirc; the read is paired in
sequencing &acirc; <br>
&acirc;0x0002 &acirc; P &acirc; the read is mapped in a
proper pair &acirc; <br>
&acirc;0x0004 &acirc; u &acirc; the query sequence itself is
unmapped &acirc; <br>
&acirc;0x0008 &acirc; U &acirc; the mate is unmapped &acirc;
<br>
&acirc;0x0010 &acirc; r &acirc; strand of the query (1 for
reverse) &acirc; <br>
&acirc;0x0020 &acirc; R &acirc; strand of the mate &acirc;
<br>
&acirc;0x0040 &acirc; 1 &acirc; the read is the first read
in a pair &acirc; <br>
&acirc;0x0080 &acirc; 2 &acirc; the read is the second read
in a pair &acirc; <br>
&acirc;0x0100 &acirc; s &acirc; the alignment is not primary
&acirc; <br>
&acirc;0x0200 &acirc; f &acirc; the read fails
platform/vendor quality checks &acirc; <br>
&acirc;0x0400 &acirc; d &acirc; the read is either a PCR or
an optical duplicate &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;
<br>
where the second column gives the string representation of
the FLAG field.</p>

<p style="margin-top: 1em">VCF FORMAT <br>
The Variant Call Format (VCF) is a TAB-delimited format with
each data line consists of the following fields:</p>


<p style="margin-top: 1em">&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;
<br>
&acirc;Col &acirc; Field &acirc; Description &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc; 1 &acirc; CHROM &acirc; CHROMosome name &acirc; <br>
&acirc; 2 &acirc; POS &acirc; the left-most POSition of the
variant &acirc; <br>
&acirc; 3 &acirc; ID &acirc; unique variant IDentifier
&acirc; <br>
&acirc; 4 &acirc; REF &acirc; the REFerence allele &acirc;
<br>
&acirc; 5 &acirc; ALT &acirc; the ALTernate allele(s),
separated by comma &acirc; <br>
&acirc; 6 &acirc; QUAL &acirc; variant/reference QUALity
&acirc; <br>
&acirc; 7 &acirc; FILTER &acirc; FILTers applied &acirc;
<br>
&acirc; 8 &acirc; INFO &acirc; INFOrmation related to the
variant, separated by semi-colon &acirc; <br>
&acirc; 9 &acirc; FORMAT &acirc; FORMAT of the genotype
fields, separated by colon (optional) &acirc; <br>
&acirc;10+ &acirc; SAMPLE &acirc; SAMPLE genotypes and
per-sample information (optional) &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;</p>

<p style="margin-top: 1em">The following table gives the
INFO tags used by samtools and bcftools.</p>


<p style="margin-top: 1em">&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&not;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;
<br>
&acirc; Tag &acirc; Format &acirc; Description &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&frac14;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&curren;
<br>
&acirc;AF1 &acirc; double &acirc; Max-likelihood estimate of
the site allele frequency (AF) of the first ALT allele
&acirc; <br>
&acirc;DP &acirc; int &acirc; Raw read depth (without
quality filtering) &acirc; <br>
&acirc;DP4 &acirc; int[4] &acirc; # high-quality reference
forward bases, ref reverse, alternate for and alt rev bases
&acirc; <br>
&acirc;FQ &acirc; int &acirc; Consensus quality. Positive:
sample genotypes different; negative: otherwise &acirc; <br>
&acirc;MQ &acirc; int &acirc; Root-Mean-Square mapping
quality of covering reads &acirc; <br>
&acirc;PC2 &acirc; int[2] &acirc; Phred probability of AF in
group1 samples being larger (,smaller) than in group2
&acirc; <br>
&acirc;PCHI2 &acirc; double &acirc; Posterior weighted chi^2
P-value between group1 and group2 samples &acirc; <br>
&acirc;PV4 &acirc; double[4] &acirc; P-value for strand
bias, baseQ bias, mapQ bias and tail distance bias &acirc;
<br>
&acirc;QCHI2 &acirc; int &acirc; Phred-scaled PCHI2 &acirc;
<br>
&acirc;RP &acirc; int &acirc; # permutations yielding a
smaller PCHI2 &acirc; <br>
&acirc;CLR &acirc; int &acirc; Phred log ratio of genotype
likelihoods with and without the trio/pair constraint
&acirc; <br>
&acirc;UGT &acirc; string &acirc; Most probable genotype
configuration without the trio constraint &acirc; <br>
&acirc;CGT &acirc; string &acirc; Most probable
configuration with the trio constraint &acirc; <br>
&acirc;VDB &acirc; float &acirc; Tests variant positions
within reads. Intended for filtering RNA-seq artifacts
around splice sites &acirc; <br>
&acirc;RPB &acirc; float &acirc; Mann-Whitney rank-sum test
for tail distance bias &acirc; <br>
&acirc;HWE &acirc; float &acirc; Hardy-Weinberg equilibrium
test, Wigginton et al., PMID: 15789306 &acirc; <br>

&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acute;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;&acirc;</p>

<p style="margin-top: 1em">EXAMPLES <br>
o Import SAM to BAM when @SQ lines are present in the
header:</p>

<p style="margin-top: 1em">samtools view -bS aln.sam &gt;
aln.bam</p>

<p style="margin-top: 1em">If @SQ lines are absent:</p>

<p style="margin-top: 1em">samtools faidx ref.fa <br>
samtools view -bt ref.fa.fai aln.sam &gt; aln.bam</p>

<p style="margin-top: 1em">where ref.fa.fai is generated
automatically by the faidx command.</p>

<p style="margin-top: 1em">o Attach the RG tag while
merging sorted alignments:</p>

<p style="margin-top: 1em">perl -e &rsquo;print
&quot;@RGID:gaSM:hsLB:gaPL:Illumina0RGID:454SM:hsLB:454PL:4540&rsquo;
&gt; rg.txt <br>
samtools merge -rh rg.txt merged.bam ga.bam 454.bam</p>

<p style="margin-top: 1em">The value in a RG tag is
determined by the file name the read is coming from. In this
example, in the merged.bam, reads from ga.bam will be
attached RG:Z:ga, while reads from <br>
454.bam will be attached RG:Z:454.</p>

<p style="margin-top: 1em">o Call SNPs and short INDELs for
one diploid individual:</p>

<p style="margin-top: 1em">samtools mpileup -ugf ref.fa
aln.bam | bcftools view -bvcg - &gt; var.raw.bcf <br>
bcftools view var.raw.bcf | vcfutils.pl varFilter -D 100
&gt; var.flt.vcf</p>

<p style="margin-top: 1em">The -D option of varFilter
controls the maximum read depth, which should be adjusted to
about twice the average read depth. One may consider to add
-C50 to mpileup if mapping <br>
quality is overestimated for reads containing excessive
mismatches. Applying this option usually helps BWA-short but
may not other mappers.</p>

<p style="margin-top: 1em">o Generate the consensus
sequence for one diploid individual:</p>

<p style="margin-top: 1em">samtools mpileup -uf ref.fa
aln.bam | bcftools view -cg - | vcfutils.pl vcf2fq &gt;
cns.fq</p>

<p style="margin-top: 1em">o Call somatic mutations from a
pair of samples:</p>

<p style="margin-top: 1em">samtools mpileup -DSuf ref.fa
aln.bam | bcftools view -bvcgT pair - &gt; var.bcf</p>

<p style="margin-top: 1em">In the output INFO field, CLR
gives the Phred-log ratio between the likelihood by treating
the two samples independently, and the likelihood by
requiring the genotype to be <br>
identical. This CLR is effectively a score measuring the
confidence of somatic calls. The higher the better.</p>

<p style="margin-top: 1em">o Call de novo and somatic
mutations from a family trio:</p>

<p style="margin-top: 1em">samtools mpileup -DSuf ref.fa
aln.bam | bcftools view -bvcgT pair -s samples.txt - &gt;
var.bcf</p>

<p style="margin-top: 1em">File samples.txt should consist
of three lines specifying the member and order of samples
(in the order of child-father-mother). Similarly, CLR gives
the Phred-log likelihood <br>
ratio with and without the trio constraint. UGT shows the
most likely genotype configuration without the trio
constraint, and CGT gives the most likely genotype
configuration <br>
satisfying the trio constraint.</p>

<p style="margin-top: 1em">o Phase one individual:</p>

<p style="margin-top: 1em">samtools calmd -AEur aln.bam
ref.fa | samtools phase -b prefix - &gt; phase.out</p>

<p style="margin-top: 1em">The calmd command is used to
reduce false heterozygotes around INDELs.</p>

<p style="margin-top: 1em">o Call SNPs and short indels for
multiple diploid individuals:</p>

<p style="margin-top: 1em">samtools mpileup -P ILLUMINA
-ugf ref.fa *.bam | bcftools view -bcvg - &gt; var.raw.bcf
<br>
bcftools view var.raw.bcf | vcfutils.pl varFilter -D 2000
&gt; var.flt.vcf</p>

<p style="margin-top: 1em">Individuals are identified from
the SM tags in the @RG header lines. Individuals can be
pooled in one alignment file; one individual can also be
separated into multiple files. <br>
The -P option specifies that indel candidates should be
collected only from read groups with the @RG-PL tag set to
ILLUMINA. Collecting indel candidates from reads sequenced
<br>
by an indel-prone technology may affect the performance of
indel calling.</p>

<p style="margin-top: 1em">Note that there is a new calling
model which can be invoked by</p>

<p style="margin-top: 1em">bcftools view -m0.99 ...</p>

<p style="margin-top: 1em">which fixes some severe
limitations of the default method.</p>

<p style="margin-top: 1em">For filtering, best results seem
to be achieved by first applying the SnpGap filter and then
applying some machine learning approach</p>

<p style="margin-top: 1em">vcf-annotate -f SnpGap=n <br>
vcf filter ...</p>

<p style="margin-top: 1em">Both can be found in the
vcftools and htslib package (links below).</p>

<p style="margin-top: 1em">o Derive the allele frequency
spectrum (AFS) on a list of sites from multiple
individuals:</p>

<p style="margin-top: 1em">samtools mpileup -Igf ref.fa
*.bam &gt; all.bcf <br>
bcftools view -bl sites.list all.bcf &gt; sites.bcf <br>
bcftools view -cGP cond2 sites.bcf &gt; /dev/null 2&gt;
sites.1.afs <br>
bcftools view -cGP sites.1.afs sites.bcf &gt; /dev/null
2&gt; sites.2.afs <br>
bcftools view -cGP sites.2.afs sites.bcf &gt; /dev/null
2&gt; sites.3.afs <br>
......</p>

<p style="margin-top: 1em">where sites.list contains the
list of sites with each line consisting of the reference
sequence name and position. The following bcftools commands
estimate AFS by EM.</p>

<p style="margin-top: 1em">o Dump BAQ applied alignment for
other SNP callers:</p>

<p style="margin-top: 1em">samtools calmd -bAr aln.bam &gt;
aln.baq.bam</p>

<p style="margin-top: 1em">It adds and corrects the NM and
MD tags at the same time. The calmd command also comes with
the -C option, the same as the one in pileup and mpileup.
Apply if it helps.</p>

<p style="margin-top: 1em">LIMITATIONS <br>
o Unaligned words used in bam_import.c, bam_endian.h, bam.c
and bam_aux.c.</p>

<p style="margin-top: 1em">o Samtools paired-end rmdup does
not work for unpaired reads (e.g. orphan reads or ends
mapped to different chromosomes). If this is a concern,
please use Picard&rsquo;s MarkDuplicate <br>
which correctly handles these cases, although a little
slower.</p>

<p style="margin-top: 1em">AUTHOR <br>
Heng Li from the Sanger Institute wrote the C version of
samtools. Bob Handsaker from the Broad Institute implemented
the BGZF library and Jue Ruan from Beijing Genomics
Insti&acirc; <br>
tute wrote the RAZF library. John Marshall and Petr Danecek
contribute to the source code and various people from the
1000 Genomes Project have contributed to the SAM format <br>
specification.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Samtools website: &lt;http://samtools.sourceforge.net&gt;
<br>
Samtools latest source:
&lt;https://github.com/samtools/samtools&gt; <br>
VCFtools website with stable link to VCF specification:
&lt;http://vcftools.sourceforge.net&gt; <br>
HTSlib website:
&lt;https://github.com/samtools/htslib&gt;</p>

<p style="margin-top: 1em">samtools-0.1.19 15 March 2013
samtools(1)</p>
<hr>
</body>
</html>
