<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:36:48 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SAMTOH5(1) User Commands SAMTOH5(1)</p>

<p style="margin-top: 1em">NAME <br>
samtoh5 - convert a SAM file to cmp.h5 format</p>

<p style="margin-top: 1em">SYNOPSIS <br>
samtoh5 in.sam reference.fasta out.cmp.h5 [options]</p>

<p style="margin-top: 1em">OPTIONS <br>
in.sam Input SAM file.</p>

<p style="margin-top: 1em">reference.fasta <br>
Reference used to generate reads.</p>

<p style="margin-top: 1em">out.cmp.h5 <br>
Output cmp.h5 file.</p>

<p style="margin-top: 1em">-smrtTitle <br>
Use this option when converting alignments generated from
reads produced by the pls2fasta(1) from bas.h5 files by
parsing read coordinates from the SMRT read title. The <br>
title is in the format /name/hole/coordinates, where
coordinates are in the format +_+, and represent the
interval of the read that was aligned.</p>

<p style="margin-top: 1em">-readType value <br>
Set the read type: &rsquo;standard&rsquo;,
&rsquo;strobe&rsquo;, &rsquo;CCS&rsquo;, or
&rsquo;cDNA&rsquo;</p>

<p style="margin-top: 1em">-verbosity value <br>
Set desired verbosity.</p>

<p style="margin-top: 1em">-useShortRefName <br>
Use abbreviated reference names obtained from file.sam
instead of using full names from reference.fasta.</p>

<p style="margin-top: 1em">-copyQVs <br>
Copy all QVs available in the SAM file into the cmp.h5 file.
This includes things like InsertionQV and DeletionTag.</p>

<p style="margin-top: 1em">NOTES <br>
Because SAM has optional tags that have different meanings
in different programs, careful usage is required in order to
have proper output. The &quot;xs&quot; tag in bwa-sw is used
to show <br>
the suboptimal score, but in PacBio SAM (blasr(1)) it is
defined as the start in the query sequence of the alignment.
When -smrtTitle is specified, the xs tag is ignored, but
<br>
when it is not specified, the coordinates given by the xs
and xe tags are used to define the interval of a read that
is aligned. The CIGAR string is relative to this
interval.</p>

<p style="margin-top: 1em">SEE ALSO <br>
blasr(1) loadPulses(1) pls2fasta(1) samFilter(1) samtom4(1)
sawriter(1) sdpMatcher(1) toAfg(1)</p>

<p style="margin-top: 1em">samtoh5 3ca7fe8 July 2015
SAMTOH5(1)</p>
<hr>
</body>
</html>
