<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>samtools(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">samtools(1)</td>
    <td class="head-vol">Bioinformatics tools</td>
    <td class="head-rtitle">samtools(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
samtools - Utilities for the Sequence Alignment/Map (SAM) format
<div style="height: 1.00em;">&#x00A0;</div>
bcftools - Utilities for the Binary Call Format (BCF) and VCF
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
samtools view -bt ref_list.txt -o aln.bam aln.sam.gz
<div class="Pp"></div>
samtools sort aln.bam aln.sorted
<div class="Pp"></div>
samtools index aln.sorted.bam
<div class="Pp"></div>
samtools idxstats aln.sorted.bam
<div class="Pp"></div>
samtools view aln.sorted.bam chr2:20,100,000-20,200,000
<div class="Pp"></div>
samtools merge out.bam in1.bam in2.bam in3.bam
<div class="Pp"></div>
samtools faidx ref.fasta
<div class="Pp"></div>
samtools pileup -vcf ref.fasta aln.sorted.bam
<div class="Pp"></div>
samtools mpileup -C50 -gf ref.fasta -r chr3:1,000-2,000 in1.bam in2.bam
<div class="Pp"></div>
samtools tview aln.sorted.bam ref.fasta
<div class="Pp"></div>
bcftools index in.bcf
<div class="Pp"></div>
bcftools view in.bcf chr2:100-200 &gt; out.vcf
<div class="Pp"></div>
bcftools view -Nvm0.99 in.bcf &gt; out.vcf 2&gt; out.afs
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Samtools is a set of utilities that manipulate alignments in the BAM format. It
  imports from and exports to the SAM (Sequence Alignment/Map) format, does
  sorting, merging and indexing, and allows to retrieve reads in any regions
  swiftly.
<div style="height: 1.00em;">&#x00A0;</div>
Samtools is designed to work on a stream. It regards an input file `-' as the
  standard input (stdin) and an output file `-' as the standard output (stdout).
  Several commands can thus be combined with Unix pipes. Samtools always output
  warning and error messages to the standard error output (stderr).
<div style="height: 1.00em;">&#x00A0;</div>
Samtools is also able to open a BAM (not SAM) file on a remote FTP or HTTP
  server if the BAM file name starts with `ftp://' or `http://'. Samtools checks
  the current working directory for the index file and will download the index
  upon absence. Samtools does not retrieve the entire alignment file unless it
  is asked to do so.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SAMTOOLS_COMMANDS_AND_OPTIONS"><a class="selflink" href="#SAMTOOLS_COMMANDS_AND_OPTIONS">SAMTOOLS
  COMMANDS AND OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>view</b></dt>
  <dd class="It-tag">samtools view [-bchuHS] [-t in.refList] [-o output] [-f
      reqFlag] [-F skipFlag] [-q minMapQ] [-l library] [-r readGroup] [-R
      rgFile] &lt;in.bam&gt;|&lt;in.sam&gt; [region1 [...]]
    <div style="height: 1.00em;">&#x00A0;</div>
    Extract/print all or sub alignments in SAM or BAM format. If no region is
      specified, all the alignments will be printed; otherwise only alignments
      overlapping the specified regions will be output. An alignment may be
      given multiple times if it is overlapping several regions. A region can be
      presented, for example, in the following format: `chr2' (the whole chr2),
      `chr2:1000000' (region starting from 1,000,000bp) or
      `chr2:1,000,000-2,000,000' (region between 1,000,000 and 2,000,000bp
      including the end points). The coordinate is 1-based.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b></dt>
  <dd class="It-tag">Output in the BAM format.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Only output alignments with all bits in INT present in the
      FLAG field. INT can be in hex in the format of /^0x[0-9A-F]+/ [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Skip alignments with bits present in INT [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Include the header in the output.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-H</b></dt>
  <dd class="It-tag">Output the header only.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Only output reads in library STR [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">Output file [stdout]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Skip alignments with MAPQ smaller than INT [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Only output reads in read group STR [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">Output reads in read groups listed in <i>FILE</i>
    [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b><i>&#x00A0;FLOAT</i></dt>
  <dd class="It-tag">Fraction of templates/pairs to subsample; the integer part
      is treated as the seed for the random number generator [-1]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">Input is in SAM. If @SQ header lines are absent, the
      <b>`-t'</b> option is required.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">Instead of printing the alignments, only count them and
      print the total number. All filter options, such as <b>`-f',</b>
      <b>`-F'</b> and <b>`-q'</b> , are taken into account.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">This file is TAB-delimited. Each line must contain the
      reference name and the length of the reference, one line for each distinct
      reference; additional fields are ignored. This file also defines the order
      of the reference sequences in sorting. If you run `samtools faidx
      &lt;ref.fa&gt;', the resultant index file <i>&lt;ref.fa&gt;.fai</i> can be
      used as this <i>&lt;in.ref_list&gt;</i> file.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b></dt>
  <dd class="It-tag">Output uncompressed BAM. This option saves time spent on
      compression/decomprssion and is thus preferred when the output is piped to
      another samtools command.</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>tview</b></dt>
  <dd class="It-tag">samtools tview [<b>-p</b> <i>chr:pos</i>] [<b>-s</b>
      <i>STR</i>] [<b>-d</b> <i>display</i>] &lt;in.sorted.bam&gt; [ref.fasta]
    <div style="height: 1.00em;">&#x00A0;</div>
    Text alignment viewer (based on the ncurses library). In the viewer, press
      `?' for help and press `g' to check the alignment start from a region in
      the format like `chr10:10,000,000' or `=10,000,000' when viewing the same
      reference sequence.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Options:</b></dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b><i>&#x00A0;display</i></dt>
  <dd class="It-tag">Output as (H)tml or (C)urses or (T)ext</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b><i>&#x00A0;chr:pos</i></dt>
  <dd class="It-tag">Go directly to this position</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Display only reads from this sample or read group</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>mpileup</b></dt>
  <dd class="It-tag">samtools mpileup [<b>-EBugp</b>] [<b>-C</b>
      <i>capQcoef</i>] [<b>-r</b> <i>reg</i>] [<b>-f</b> <i>in.fa</i>]
      [<b>-l</b> <i>list</i>] [<b>-M</b> <i>capMapQ</i>] [<b>-Q</b>
      <i>minBaseQ</i>] [<b>-q</b> <i>minMapQ</i>] <i>in.bam</i> [<i>in2.bam</i>
      [<i>...</i>]]
    <div style="height: 1.00em;">&#x00A0;</div>
    Generate BCF or pileup for one or multiple BAM files. Alignment records are
      grouped by sample identifiers in @RG header lines. If sample identifiers
      are absent, each input file is regarded as one sample.
    <div style="height: 1.00em;">&#x00A0;</div>
    In the pileup format (without <b>-u</b>or<b>-g</b>), each line represents a
      genomic position, consisting of chromosome name, coordinate, reference
      base, read bases, read qualities and alignment mapping qualities.
      Information on match, mismatch, indel, strand, mapping quality and start
      and end of a read are all encoded at the read base column. At this column,
      a dot stands for a match to the reference base on the forward strand, a
      comma for a match on the reverse strand, a '&gt;' or '&lt;' for a
      reference skip, `ACGTN' for a mismatch on the forward strand and `acgtn'
      for a mismatch on the reverse strand. A pattern `\+[0-9]+[ACGTNacgtn]+'
      indicates there is an insertion between this reference position and the
      next reference position. The length of the insertion is given by the
      integer in the pattern, followed by the inserted sequence. Similarly, a
      pattern `-[0-9]+[ACGTNacgtn]+' represents a deletion from the reference.
      The deleted bases will be presented as `*' in the following lines. Also at
      the read base column, a symbol `^' marks the start of a read. The ASCII of
      the character following `^' minus 33 gives the mapping quality. A symbol
      `$' marks the end of a read segment.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>Input Options:</b></dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-6</b></dt>
  <dd class="It-tag">Assume the quality is in the Illumina 1.3+ encoding.
      <b>-A</b> Do not skip anomalous read pairs in variant calling.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b></dt>
  <dd class="It-tag">Disable probabilistic realignment for the computation of
      base alignment quality (BAQ). BAQ is the Phred-scaled probability of a
      read base being misaligned. Applying this option greatly helps to reduce
      false SNPs caused by misalignments.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">List of input BAM files, one file per line [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Coefficient for downgrading mapping quality for reads
      containing excessive mismatches. Given a read with a phred-scaled
      probability q of being generated from the mapped position, the new mapping
      quality is about sqrt((INT-q)/INT)*INT. A zero value disables this
      functionality; if enabled, the recommended value for BWA is 50. [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">At a position, read maximally <i>INT</i> reads per input
      BAM. [250]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b></dt>
  <dd class="It-tag">Extended BAQ computation. This option helps sensitivity
      especially for MNPs, but may hurt specificity a little bit.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">The <b>faidx</b>-indexed reference file in the FASTA
      format. The file can be optionally compressed by <b>razip</b>. [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">BED or position list file containing a list of regions or
      sites where pileup or BCF should be generated [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Minimum mapping quality for an alignment to be used
    [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Minimum base quality for a base to be considered [13]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Only generate pileup in region <i>STR</i> [all sites]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Output Options:</b></dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b></dt>
  <dd class="It-tag">Output per-sample read depth</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">Compute genotype likelihoods and output them in the binary
      call format (BCF).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">Output per-sample Phred-scaled strand bias P-value</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b></dt>
  <dd class="It-tag">Similar to <b>-g</b> except that the output is uncompressed
      BCF, which is preferred for piping.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Options for Genotype Likelihood Computation (for -g or
    -u):</b></dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Phred-scaled gap extension sequencing error probability.
      Reducing <i>INT</i> leads to longer indels. [20]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Coefficient for modeling homopolymer errors. Given an
      <i>l</i>-long homopolymer run, the sequencing error of an indel of size
      <i>s</i> is modeled as <i>INT</i>*<i>s</i>/<i>l</i>. [100]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-I</b></dt>
  <dd class="It-tag">Do not perform INDEL calling</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-L</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Skip INDEL calling if the average per-sample depth is above
      <i>INT</i>. [250]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Phred-scaled gap open sequencing error probability.
      Reducing <i>INT</i> leads to more indel calls. [40]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b></dt>
  <dd class="It-tag">Apply -m and -F thresholds per sample to increase
      sensitivity of calling. By default both options are applied to reads
      pooled from all samples.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Comma dilimited list of platforms (determined by
      <b>@RG-PL</b>) from which indel candidates are obtained. It is recommended
      to collect indel candidates from sequencing technologies that have low
      indel error rate such as ILLUMINA. [all]</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>reheader</b></dt>
  <dd class="It-tag">samtools reheader &lt;in.header.sam&gt; &lt;in.bam&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    Replace the header in <i>in.bam</i> with the header in <i>in.header.sam.</i>
      This command is much faster than replacing the header with a
      BAM-&gt;SAM-&gt;BAM conversion.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cat</b></dt>
  <dd class="It-tag">samtools cat [-h header.sam] [-o out.bam] &lt;in1.bam&gt;
      &lt;in2.bam&gt; [ ... ]
    <div style="height: 1.00em;">&#x00A0;</div>
    Concatenate BAMs. The sequence dictionary of each input BAM must be
      identical, although this command does not check this. This command uses a
      similar trick to <b>reheader</b> which enables fast BAM concatenation.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>sort</b></dt>
  <dd class="It-tag">samtools sort [-nof] [-m maxMem] &lt;in.bam&gt;
      &lt;out.prefix&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    Sort alignments by leftmost coordinates. File <i>&lt;out.prefix&gt;.bam</i>
      will be created. This command may also create temporary files
      <i>&lt;out.prefix&gt;.%d.bam</i> when the whole alignment cannot be fitted
      into memory (controlled by option -m).
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-o</b></dt>
  <dd class="It-tag">Output the final alignment to the standard output.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">Sort by read names rather than by chromosomal
    coordinates</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b></dt>
  <dd class="It-tag">Use <i>&lt;out.prefix&gt;</i> as the full output path and
      do not append <i>.bam</i> suffix.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Approximately the maximum required memory. [500000000]</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>merge</b></dt>
  <dd class="It-tag">samtools merge [-nur1f] [-h inh.sam] [-R reg]
      &lt;out.bam&gt; &lt;in1.bam&gt; &lt;in2.bam&gt; [...]
    <div style="height: 1.00em;">&#x00A0;</div>
    Merge multiple sorted alignments. The header reference lists of all the
      input BAM files, and the @SQ headers of <i>inh.sam</i>, if any, must all
      refer to the same set of reference sequences. The header reference list
      and (unless overridden by <b>-h</b>) `@' headers of <i>in1.bam</i> will be
      copied to <i>out.bam</i>, and the headers of other files will be ignored.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b></dt>
  <dd class="It-tag">Use zlib compression level 1 to comrpess the output</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-f</b></dt>
  <dd class="It-tag">Force to overwrite the output file if present.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">Use the lines of <i>FILE</i> as `@' headers to be copied to
      <i>out.bam</i>, replacing any header lines that would otherwise be copied
      from <i>in1.bam</i>. (<i>FILE</i> is actually in SAM format, though any
      alignment records it may contain are ignored.)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">The input alignments are sorted by read names rather than
      by chromosomal coordinates</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Merge files in the specified region indicated by <i>STR</i>
      [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b></dt>
  <dd class="It-tag">Attach an RG tag to each alignment. The tag value is
      inferred from file names.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b></dt>
  <dd class="It-tag">Uncompressed BAM output</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>index</b></dt>
  <dd class="It-tag">samtools index &lt;aln.bam&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    Index sorted alignment for fast random access. Index file
      <i>&lt;aln.bam&gt;.bai</i> will be created.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>idxstats</b></dt>
  <dd class="It-tag">samtools idxstats &lt;aln.bam&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    Retrieve and print stats in the index file. The output is TAB delimited with
      each line consisting of reference sequence name, sequence length, # mapped
      reads and # unmapped reads.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>faidx</b></dt>
  <dd class="It-tag">samtools faidx &lt;ref.fasta&gt; [region1 [...]]
    <div style="height: 1.00em;">&#x00A0;</div>
    Index reference sequence in the FASTA format or extract subsequence from
      indexed reference sequence. If no region is specified, <b>faidx</b> will
      index the file and create <i>&lt;ref.fasta&gt;.fai</i> on the disk. If
      regions are speficified, the subsequences will be retrieved and printed to
      stdout in the FASTA format. The input file can be compressed in the
      <b>RAZF</b> format.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>fixmate</b></dt>
  <dd class="It-tag">samtools fixmate &lt;in.nameSrt.bam&gt; &lt;out.bam&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    Fill in mate coordinates, ISIZE and mate related flags from a name-sorted
      alignment.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>rmdup</b></dt>
  <dd class="It-tag">samtools rmdup [-sS] &lt;input.srt.bam&gt; &lt;out.bam&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    Remove potential PCR duplicates: if multiple read pairs have identical
      external coordinates, only retain the pair with highest mapping quality.
      In the paired-end mode, this command <b>ONLY</b> works with FR orientation
      and requires ISIZE is correctly set. It does not work for unpaired reads
      (e.g. two ends mapped to different chromosomes or orphan reads).
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b></dt>
  <dd class="It-tag">Remove duplicate for single-end reads. By default, the
      command works for paired-end reads only.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">Treat paired-end reads and single-end reads.</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>calmd</b></dt>
  <dd class="It-tag">samtools calmd [-EeubSr] [-C capQcoef] &lt;aln.bam&gt;
      &lt;ref.fasta&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    Generate the MD tag. If the MD tag is already present, this command will
      give a warning if the MD tag generated is different from the existing tag.
      Output SAM by default.
    <div style="height: 1.00em;">&#x00A0;</div>
    <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b></dt>
  <dd class="It-tag">When used jointly with <b>-r</b> this option overwrites the
      original base quality.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b></dt>
  <dd class="It-tag">Convert a the read base to = if it is identical to the
      aligned reference base. Indel caller does not support the = bases at the
      moment.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b></dt>
  <dd class="It-tag">Output uncompressed BAM</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b></dt>
  <dd class="It-tag">Output compressed BAM</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">The input is SAM with header lines</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-C</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Coefficient to cap mapping quality of poorly mapped reads.
      See the <b>pileup</b> command for details. [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b></dt>
  <dd class="It-tag">Compute the BQ tag (without -A) or cap base quality by BAQ
      (with -A).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-E</b></dt>
  <dd class="It-tag">Extended BAQ calculation. This option trades specificity
      for sensitivity, though the effect is minor.</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>targetcut</b></dt>
  <dd class="It-tag">samtools targetcut [-Q minBaseQ] [-i inPenalty] [-0 em0]
      [-1 em1] [-2 em2] [-f ref] &lt;in.bam&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    This command identifies target regions by examining the continuity of read
      depth, computes haploid consensus sequences of targets and outputs a SAM
      with each sequence corresponding to a target. When option <b>-f</b> is in
      use, BAQ will be applied. This command is <b>only</b> designed for cutting
      fosmid clones from fosmid pool sequencing [Ref. Kitzman et al.
    (2010)].</dd>
</dl>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>phase</b></dt>
  <dd class="It-tag">samtools phase [-AF] [-k len] [-b prefix] [-q minLOD] [-Q
      minBaseQ] &lt;in.bam&gt;
    <div style="height: 1.00em;">&#x00A0;</div>
    Call and phase heterozygous SNPs. <b>OPTIONS:</b></dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b></dt>
  <dd class="It-tag">Drop reads with ambiguous phase.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Prefix of BAM output. When this option is in use, phase-0
      reads will be saved in file <b>STR</b>.0.bam and phase-1 reads in
      <b>STR</b>.1.bam. Phase unknown reads will be randomly allocated to one of
      the two files. Chimeric reads with switch errors will be saved in
      <b>STR</b>.chimeric.bam. [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b></dt>
  <dd class="It-tag">Do not attempt to fix chimeric reads.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-k</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Maximum length for local phasing. [13]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-q</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Minimum Phred-scaled LOD to call a heterozygote. [40]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Minimum base quality to be used in het calling. [13]</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="BCFTOOLS_COMMANDS_AND_OPTIONS"><a class="selflink" href="#BCFTOOLS_COMMANDS_AND_OPTIONS">BCFTOOLS
  COMMANDS AND OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>view</b></dt>
  <dd class="It-tag"><b>bcftools view</b> [<b>-AbFGNQSucgv</b>] [<b>-D</b>
      <i>seqDict</i>] [<b>-l</b> <i>listLoci</i>] [<b>-s</b> <i>listSample</i>]
      [<b>-i</b> <i>gapSNPratio</i>] [<b>-t</b> <i>mutRate</i>] [<b>-p</b>
      <i>varThres</i>] [<b>-m</b> <i>varThres</i>] [<b>-P</b> <i>prior</i>]
      [<b>-1</b> <i>nGroup1</i>] [<b>-d</b> <i>minFrac</i>] [<b>-U</b>
      <i>nPerm</i>] [<b>-X</b> <i>permThres</i>] [<b>-T</b> <i>trioType</i>]
      <i>in.bcf</i> [<i>region</i>]
    <div style="height: 1.00em;">&#x00A0;</div>
    Convert between BCF and VCF, call variant candidates and estimate allele
      frequencies.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<div style="margin-left: 5.00ex;">
<dl class="Bl-tag">
  <dt class="It-tag"><b>Input/Output Options:</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-A</b></dt>
  <dd class="It-tag">Retain all possible alternate alleles at variant sites. By
      default, the view command discards unlikely alleles.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b></dt>
  <dd class="It-tag">Output in the BCF format. The default is VCF.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-D</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">Sequence dictionary (list of chromosome names) for
      VCF-&gt;BCF conversion [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-F</b></dt>
  <dd class="It-tag">Indicate PL is generated by r921 or before (ordering is
      different).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-G</b></dt>
  <dd class="It-tag">Suppress all individual genotype information.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">List of sites at which information are outputted [all
      sites]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-N</b></dt>
  <dd class="It-tag">Skip sites where the REF field is not A/C/G/T</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-Q</b></dt>
  <dd class="It-tag">Output the QCALL likelihood format</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-s</b><i>&#x00A0;FILE</i></dt>
  <dd class="It-tag">List of samples to use. The first column in the input gives
      the sample names and the second gives the ploidy, which can only be 1 or
      2. When the 2nd column is absent, the sample ploidy is assumed to be 2. In
      the output, the ordering of samples will be identical to the one in
      <i>FILE</i>. [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-S</b></dt>
  <dd class="It-tag">The input is VCF instead of BCF.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b></dt>
  <dd class="It-tag">Uncompressed BCF output (force -b).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Consensus/Variant Calling Options:</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-c</b></dt>
  <dd class="It-tag">Call variants using Bayesian inference. This option
      automatically invokes option <b>-e</b>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b><i>&#x00A0;FLOAT</i></dt>
  <dd class="It-tag">When <b>-v</b> is in use, skip loci where the fraction of
      samples covered by reads is below FLOAT. [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b></dt>
  <dd class="It-tag">Perform max-likelihood inference only, including estimating
      the site allele frequency, testing Hardy-Weinberg equlibrium and testing
      associations with LRT.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b></dt>
  <dd class="It-tag">Call per-sample genotypes at variant sites (force -c)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-i</b><i>&#x00A0;FLOAT</i></dt>
  <dd class="It-tag">Ratio of INDEL-to-SNP mutation rate [0.15]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b><i>&#x00A0;FLOAT</i></dt>
  <dd class="It-tag">New model for improved multiallelic and rare-variant
      calling. Another ALT allele is accepted if P(chi^2) of LRT exceeds the
      FLOAT threshold. The parameter seems robust and the actual value usually
      does not affect the results much; a good value to use is 0.99. This is the
      recommended calling method. [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b><i>&#x00A0;FLOAT</i></dt>
  <dd class="It-tag">A site is considered to be a variant if P(ref|D)&lt;FLOAT
      [0.5]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-P</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Prior or initial allele frequency spectrum. If STR can be
      <i>full</i>, <i>cond2</i>, <i>flat</i> or the file consisting of error
      output from a previous variant calling run.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b><i>&#x00A0;FLOAT</i></dt>
  <dd class="It-tag">Scaled muttion rate for variant calling [0.001]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-T</b><i>&#x00A0;STR</i></dt>
  <dd class="It-tag">Enable pair/trio calling. For trio calling, option
      <b>-s</b> is usually needed to be applied to configure the trio members
      and their ordering. In the file supplied to the option <b>-s</b>, the
      first sample must be the child, the second the father and the third the
      mother. The valid values of <i>STR</i> are `pair', `trioauto', `trioxd'
      and `trioxs', where `pair' calls differences between two input samples,
      and `trioxd' (`trioxs') specifies that the input is from the X chromosome
      non-PAR regions and the child is a female (male). [null]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-v</b></dt>
  <dd class="It-tag">Output variant sites only (force -c)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>Contrast Calling and Association Test Options:</b></dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Number of group-1 samples. This option is used for dividing
      the samples into two groups for contrast SNP calling or association test.
      When this option is in use, the following VCF INFO will be outputted: PC2,
      PCHI2 and QCHI2. [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-U</b><i>&#x00A0;INT</i></dt>
  <dd class="It-tag">Number of permutations for association test (effective only
      with <b>-1</b>) [0]</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-X</b><i>&#x00A0;FLOAT</i></dt>
  <dd class="It-tag">Only perform permutations for P(chi^2)&lt;FLOAT (effective
      only with <b>-U</b>) [0.01]</dd>
</dl>
</div>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>index</b></dt>
  <dd class="It-tag"><b>bcftools index</b> <i>in.bcf</i>
    <div style="height: 1.00em;">&#x00A0;</div>
    Index sorted BCF for random access.</dd>
</dl>
<div style="height: 1.00em;">&#x00A0;</div>
<dl class="Bl-tag">
  <dt class="It-tag"><b>cat</b></dt>
  <dd class="It-tag"><b>bcftools cat</b> <i>in1.bcf</i> [<i>in2.bcf
      </i>[<i>...</i>]]]
    <div style="height: 1.00em;">&#x00A0;</div>
    Concatenate BCF files. The input files are required to be sorted and have
      identical samples appearing in the same order.</dd>
</dl>
<h1 class="Sh" title="Sh" id="SAM_FORMAT"><a class="selflink" href="#SAM_FORMAT">SAM
  FORMAT</a></h1>
Sequence Alignment/Map (SAM) format is TAB-delimited. Apart from the header
  lines, which are started with the `@' symbol, each alignment line consists of:
<div style="height: 1.00em;">&#x00A0;</div>
<table class="tbl">
  <colgroup>
    <col style="width: 7.00ex;"/>
    <col style="width: 5.00ex;"/>
    <col style="width: 56.00ex;"/>
  </colgroup>
  <tr>
    <td>Col</td>
    <td>Field</td>
    <td>Description</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>1</td>
    <td>QNAME</td>
    <td>Query template/pair NAME</td>
  </tr>
  <tr>
    <td>2</td>
    <td>FLAG</td>
    <td>bitwise FLAG</td>
  </tr>
  <tr>
    <td>3</td>
    <td>RNAME</td>
    <td>Reference sequence NAME</td>
  </tr>
  <tr>
    <td>4</td>
    <td>POS</td>
    <td>1-based leftmost POSition/coordinate of clipped sequence</td>
  </tr>
  <tr>
    <td>5</td>
    <td>MAPQ</td>
    <td>MAPping Quality (Phred-scaled)</td>
  </tr>
  <tr>
    <td>6</td>
    <td>CIAGR</td>
    <td>extended CIGAR string</td>
  </tr>
  <tr>
    <td>7</td>
    <td>MRNM</td>
    <td>Mate Reference sequence NaMe (`=' if same as RNAME)</td>
  </tr>
  <tr>
    <td>8</td>
    <td>MPOS</td>
    <td>1-based Mate POSistion</td>
  </tr>
  <tr>
    <td>9</td>
    <td>TLEN</td>
    <td>inferred Template LENgth (insert size)</td>
  </tr>
  <tr>
    <td>10</td>
    <td>SEQ</td>
    <td>query SEQuence on the same strand as the reference</td>
  </tr>
  <tr>
    <td>11</td>
    <td>QUAL</td>
    <td>query QUALity (ASCII-33 gives the Phred base quality)</td>
  </tr>
  <tr>
    <td>12+</td>
    <td>OPT</td>
    <td>variable OPTional fields in the format TAG:VTYPE:VALUE</td>
  </tr>
</table>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Each bit in the FLAG field is defined as:
<div style="height: 1.00em;">&#x00A0;</div>
<table class="tbl">
  <colgroup>
    <col style="width: 6.00ex;"/>
    <col style="width: 3.00ex;"/>
    <col style="width: 48.00ex;"/>
  </colgroup>
  <tr>
    <td>Flag</td>
    <td>Chr</td>
    <td>Description</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>0x0001</td>
    <td>p</td>
    <td>the read is paired in sequencing</td>
  </tr>
  <tr>
    <td>0x0002</td>
    <td>P</td>
    <td>the read is mapped in a proper pair</td>
  </tr>
  <tr>
    <td>0x0004</td>
    <td>u</td>
    <td>the query sequence itself is unmapped</td>
  </tr>
  <tr>
    <td>0x0008</td>
    <td>U</td>
    <td>the mate is unmapped</td>
  </tr>
  <tr>
    <td>0x0010</td>
    <td>r</td>
    <td>strand of the query (1 for reverse)</td>
  </tr>
  <tr>
    <td>0x0020</td>
    <td>R</td>
    <td>strand of the mate</td>
  </tr>
  <tr>
    <td>0x0040</td>
    <td>1</td>
    <td>the read is the first read in a pair</td>
  </tr>
  <tr>
    <td>0x0080</td>
    <td>2</td>
    <td>the read is the second read in a pair</td>
  </tr>
  <tr>
    <td>0x0100</td>
    <td>s</td>
    <td>the alignment is not primary</td>
  </tr>
  <tr>
    <td>0x0200</td>
    <td>f</td>
    <td>the read fails platform/vendor quality checks</td>
  </tr>
  <tr>
    <td>0x0400</td>
    <td>d</td>
    <td>the read is either a PCR or an optical duplicate</td>
  </tr>
</table>
<div style="height: 1.00em;">&#x00A0;</div>
where the second column gives the string representation of the FLAG field.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="VCF_FORMAT"><a class="selflink" href="#VCF_FORMAT">VCF
  FORMAT</a></h1>
The Variant Call Format (VCF) is a TAB-delimited format with each data line
  consists of the following fields:
<table class="tbl">
  <colgroup>
    <col style="width: 7.00ex;"/>
    <col style="width: 6.00ex;"/>
    <col style="width: 60.00ex;"/>
  </colgroup>
  <tr>
    <td>Col</td>
    <td>Field</td>
    <td>Description</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>1</td>
    <td>CHROM</td>
    <td>CHROMosome name</td>
  </tr>
  <tr>
    <td>2</td>
    <td>POS</td>
    <td>the left-most POSition of the variant</td>
  </tr>
  <tr>
    <td>3</td>
    <td>ID</td>
    <td>unique variant IDentifier</td>
  </tr>
  <tr>
    <td>4</td>
    <td>REF</td>
    <td>the REFerence allele</td>
  </tr>
  <tr>
    <td>5</td>
    <td>ALT</td>
    <td>the ALTernate allele(s), separated by comma</td>
  </tr>
  <tr>
    <td>6</td>
    <td>QUAL</td>
    <td>variant/reference QUALity</td>
  </tr>
  <tr>
    <td>7</td>
    <td>FILTER</td>
    <td>FILTers applied</td>
  </tr>
  <tr>
    <td>8</td>
    <td>INFO</td>
    <td>INFOrmation related to the variant, separated by semi-colon</td>
  </tr>
  <tr>
    <td>9</td>
    <td>FORMAT</td>
    <td>FORMAT of the genotype fields, separated by colon (optional)</td>
  </tr>
  <tr>
    <td>10+</td>
    <td>SAMPLE</td>
    <td>SAMPLE genotypes and per-sample information (optional)</td>
  </tr>
</table>
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The following table gives the <b>INFO</b> tags used by samtools and bcftools.
<div style="height: 1.00em;">&#x00A0;</div>
<table class="tbl">
  <colgroup>
    <col style="width: 5.00ex;"/>
    <col style="width: 9.00ex;"/>
    <col style="width: 98.00ex;"/>
  </colgroup>
  <tr>
    <td>Tag</td>
    <td>Format</td>
    <td>Description</td>
  </tr>
  <tr>
    <td colspan="0"></td>
  </tr>
  <tr>
    <td>AF1</td>
    <td>double</td>
    <td>Max-likelihood estimate of the site allele frequency (AF) of the first
      ALT allele</td>
  </tr>
  <tr>
    <td>DP</td>
    <td>int</td>
    <td>Raw read depth (without quality filtering)</td>
  </tr>
  <tr>
    <td>DP4</td>
    <td>int[4]</td>
    <td># high-quality reference forward bases, ref reverse, alternate for and
      alt rev bases</td>
  </tr>
  <tr>
    <td>FQ</td>
    <td>int</td>
    <td>Consensus quality. Positive: sample genotypes different; negative:
      otherwise</td>
  </tr>
  <tr>
    <td>MQ</td>
    <td>int</td>
    <td>Root-Mean-Square mapping quality of covering reads</td>
  </tr>
  <tr>
    <td>PC2</td>
    <td>int[2]</td>
    <td>Phred probability of AF in group1 samples being larger (,smaller) than
      in group2</td>
  </tr>
  <tr>
    <td>PCHI2</td>
    <td>double</td>
    <td>Posterior weighted chi^2 P-value between group1 and group2 samples</td>
  </tr>
  <tr>
    <td>PV4</td>
    <td>double[4]</td>
    <td>P-value for strand bias, baseQ bias, mapQ bias and tail distance
      bias</td>
  </tr>
  <tr>
    <td>QCHI2</td>
    <td>int</td>
    <td>Phred-scaled PCHI2</td>
  </tr>
  <tr>
    <td>RP</td>
    <td>int</td>
    <td># permutations yielding a smaller PCHI2</td>
  </tr>
  <tr>
    <td>CLR</td>
    <td>int</td>
    <td>Phred log ratio of genotype likelihoods with and without the trio/pair
      constraint</td>
  </tr>
  <tr>
    <td>UGT</td>
    <td>string</td>
    <td>Most probable genotype configuration without the trio constraint</td>
  </tr>
  <tr>
    <td>CGT</td>
    <td>string</td>
    <td>Most probable configuration with the trio constraint</td>
  </tr>
  <tr>
    <td>VDB</td>
    <td>float</td>
    <td>Tests variant positions within reads. Intended for filtering RNA-seq
      artifacts around splice sites</td>
  </tr>
  <tr>
    <td>RPB</td>
    <td>float</td>
    <td>Mann-Whitney rank-sum test for tail distance bias</td>
  </tr>
  <tr>
    <td>HWE</td>
    <td>float</td>
    <td>Hardy-Weinberg equilibrium test, Wigginton et al., PMID: 15789306</td>
  </tr>
</table>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Import SAM to BAM when <b>@SQ</b> lines are present in the
      header:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools view -bS aln.sam &gt; aln.bam
    <div style="height: 1.00em;">&#x00A0;</div>
    If <b>@SQ</b> lines are absent:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools faidx ref.fa
    <br/>
     samtools view -bt ref.fa.fai aln.sam &gt; aln.bam
    <div style="height: 1.00em;">&#x00A0;</div>
    where <i>ref.fa.fai</i> is generated automatically by the <b>faidx</b>
      command.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Attach the <b>RG</b> tag while merging sorted alignments:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     perl -e 'print
      &quot;@RG\tID:ga\tSM:hs\tLB:ga\tPL:Illumina\n@RG\tID:454\tSM:hs\tLB:454\tPL:454\n&quot;'
      &gt; rg.txt
    <br/>
     samtools merge -rh rg.txt merged.bam ga.bam 454.bam
    <div style="height: 1.00em;">&#x00A0;</div>
    The value in a <b>RG</b> tag is determined by the file name the read is
      coming from. In this example, in the <i>merged.bam</i>, reads from
      <i>ga.bam</i> will be attached <i>RG:Z:ga</i>, while reads from
      <i>454.bam</i> will be attached <i>RG:Z:454</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Call SNPs and short INDELs for one diploid individual:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools mpileup -ugf ref.fa aln.bam | bcftools view -bvcg - &gt;
      var.raw.bcf
    <br/>
     bcftools view var.raw.bcf | vcfutils.pl varFilter -D 100 &gt; var.flt.vcf
    <div style="height: 1.00em;">&#x00A0;</div>
    The <b>-D</b> option of varFilter controls the maximum read depth, which
      should be adjusted to about twice the average read depth. One may consider
      to add <b>-C50</b> to <b>mpileup</b> if mapping quality is overestimated
      for reads containing excessive mismatches. Applying this option usually
      helps <b>BWA-short</b> but may not other mappers.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Generate the consensus sequence for one diploid individual:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools mpileup -uf ref.fa aln.bam | bcftools view -cg - | vcfutils.pl
      vcf2fq &gt; cns.fq
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Call somatic mutations from a pair of samples:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools mpileup -DSuf ref.fa aln.bam | bcftools view -bvcgT pair - &gt;
      var.bcf
    <div style="height: 1.00em;">&#x00A0;</div>
    In the output INFO field, <i>CLR</i> gives the Phred-log ratio between the
      likelihood by treating the two samples independently, and the likelihood
      by requiring the genotype to be identical. This <i>CLR</i> is effectively
      a score measuring the confidence of somatic calls. The higher the better.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Call de novo and somatic mutations from a family trio:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools mpileup -DSuf ref.fa aln.bam | bcftools view -bvcgT pair -s
      samples.txt - &gt; var.bcf
    <div style="height: 1.00em;">&#x00A0;</div>
    File <i>samples.txt</i> should consist of three lines specifying the member
      and order of samples (in the order of child-father-mother). Similarly,
      <i>CLR</i> gives the Phred-log likelihood ratio with and without the trio
      constraint. <i>UGT</i> shows the most likely genotype configuration
      without the trio constraint, and <i>CGT</i> gives the most likely genotype
      configuration satisfying the trio constraint.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Phase one individual:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools calmd -AEur aln.bam ref.fa | samtools phase -b prefix - &gt;
      phase.out
    <div style="height: 1.00em;">&#x00A0;</div>
    The <b>calmd</b> command is used to reduce false heterozygotes around
      INDELs.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Call SNPs and short indels for multiple diploid
      individuals:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools mpileup -P ILLUMINA -ugf ref.fa *.bam | bcftools view -bcvg - &gt;
      var.raw.bcf
    <br/>
     bcftools view var.raw.bcf | vcfutils.pl varFilter -D 2000 &gt; var.flt.vcf
    <div style="height: 1.00em;">&#x00A0;</div>
    Individuals are identified from the <b>SM</b> tags in the <b>@RG</b> header
      lines. Individuals can be pooled in one alignment file; one individual can
      also be separated into multiple files. The <b>-P</b> option specifies that
      indel candidates should be collected only from read groups with the
      <b>@RG-PL</b> tag set to <i>ILLUMINA</i>. Collecting indel candidates from
      reads sequenced by an indel-prone technology may affect the performance of
      indel calling.
    <div style="height: 1.00em;">&#x00A0;</div>
    Note that there is a new calling model which can be invoked by
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     bcftools view -m0.99 ...
    <div style="height: 1.00em;">&#x00A0;</div>
    which fixes some severe limitations of the default method.
    <div style="height: 1.00em;">&#x00A0;</div>
    For filtering, best results seem to be achieved by first applying the
      <i>SnpGap</i> filter and then applying some machine learning approach
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     vcf-annotate -f SnpGap=n
    <br/>
     vcf filter ...
    <div style="height: 1.00em;">&#x00A0;</div>
    Both can be found in the <b>vcftools</b> and <b>htslib</b> package (links
      below).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Derive the allele frequency spectrum (AFS) on a list of
      sites from multiple individuals:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools mpileup -Igf ref.fa *.bam &gt; all.bcf
    <br/>
     bcftools view -bl sites.list all.bcf &gt; sites.bcf
    <br/>
     bcftools view -cGP cond2 sites.bcf &gt; /dev/null 2&gt; sites.1.afs
    <br/>
     bcftools view -cGP sites.1.afs sites.bcf &gt; /dev/null 2&gt; sites.2.afs
    <br/>
     bcftools view -cGP sites.2.afs sites.bcf &gt; /dev/null 2&gt; sites.3.afs
    <br/>
     ......
    <div style="height: 1.00em;">&#x00A0;</div>
    where <i>sites.list</i> contains the list of sites with each line consisting
      of the reference sequence name and position. The following <b>bcftools</b>
      commands estimate AFS by EM.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Dump BAQ applied alignment for other SNP callers:
    <div style="height: 1.00em;">&#x00A0;</div>
    <br/>
     samtools calmd -bAr aln.bam &gt; aln.baq.bam
    <div style="height: 1.00em;">&#x00A0;</div>
    It adds and corrects the <b>NM</b> and <b>MD</b> tags at the same time. The
      <b>calmd</b> command also comes with the <b>-C</b> option, the same as the
      one in <b>pileup</b> and <b>mpileup</b>. Apply if it helps.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="LIMITATIONS"><a class="selflink" href="#LIMITATIONS">LIMITATIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Unaligned words used in bam_import.c, bam_endian.h, bam.c
      and bam_aux.c.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">o</dt>
  <dd class="It-tag">Samtools paired-end rmdup does not work for unpaired reads
      (e.g. orphan reads or ends mapped to different chromosomes). If this is a
      concern, please use Picard's MarkDuplicate which correctly handles these
      cases, although a little slower.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Heng Li from the Sanger Institute wrote the C version of samtools. Bob Handsaker
  from the Broad Institute implemented the BGZF library and Jue Ruan from
  Beijing Genomics Institute wrote the RAZF library. John Marshall and Petr
  Danecek contribute to the source code and various people from the 1000 Genomes
  Project have contributed to the SAM format specification.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Samtools website: &lt;http://samtools.sourceforge.net&gt;
<div>&#x00A0;</div>
Samtools latest source: &lt;https://github.com/samtools/samtools&gt;
<div>&#x00A0;</div>
VCFtools website with stable link to VCF specification:
  &lt;http://vcftools.sourceforge.net&gt;
<div>&#x00A0;</div>
HTSlib website: &lt;https://github.com/samtools/htslib&gt;</div>
<table class="foot">
  <tr>
    <td class="foot-date">15 March 2013</td>
    <td class="foot-os">samtools-0.1.19</td>
  </tr>
</table>
</body>
</html>
