<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:37:04 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>DEBIAN/SCORE_CONSERVATION(1) User Commands
DEBIAN/SCORE_CONSERVATION(1)</p>

<p style="margin-top: 1em">NAME <br>
score_conservation - score protein sequence conservation</p>

<p style="margin-top: 1em">SYNOPSIS <br>
score_conservation [options] ALIGNFILE</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Score protein sequence conservation in ALIGNFILE. ALIGNFILE
must be in FASTA, CLUSTAL or Stockholm format.</p>

<p style="margin-top: 1em">The following conservation
scoring methods are implemented: <br>
* sum of pairs <br>
* weighted sum of pairs <br>
* Shannon entropy <br>
* Shannon entropy with property groupings (Mirny and
Shakhnovich 1995, <br>
Valdar and Thornton 2001) <br>
* relative entropy with property groupings (Williamson 1995)
<br>
* von Neumann entropy (Caffrey et al 2004) <br>
* relative entropy (Samudrala and Wang 2006) <br>
* Jensen-Shannon divergence (Capra and Singh 2007)</p>

<p style="margin-top: 1em">A window-based extension that
incorporates the estimated conservation of sequentially
adjacent residues into the score for each column is also
given. This window approach can be <br>
applied to any of the conservation scoring methods.</p>

<p style="margin-top: 1em">With default parameters
score_conservation(1) computes the conservation scores for
the alignment using the Jensen-Shannon divergence and a
window -w of 3.</p>

<p style="margin-top: 1em">The sequence-specific output can
be used as the conservation input for concavity(1).</p>

<p style="margin-top: 1em">Conservation is highly
predictive in identifying catalytic sites and residues near
bound ligands.</p>

<p style="margin-top: 1em">REFERENCES <br>
Capra JA and Singh M. Predicting functionally important
residues from sequence conservation. Bioinformatics,
23(15):1875-82, 2007.</p>

<p style="margin-top: 1em">OPTIONS <br>
-a [NAME] <br>
Reference sequence. Print scores in reference to the named
sequence (ignoring gaps). Default prints the entire
column.</p>

<p style="margin-top: 1em">-b [0-1] <br>
Lambda for window heuristic linear combination.
Default=.5.</p>

<p style="margin-top: 1em">Equation:</p>

<p style="margin-top: 1em">&quot;score = (1 - lambda) *
average_score_over_window_around_middle + lambda *
score_of_middle&quot;</p>

<p style="margin-top: 1em">-d [FILE] <br>
Background distribution file, e.g.
distributions/swissprot.distribution. Default=built-in
BLOSUM62.</p>

<p style="margin-top: 1em">-g [0-1)] <br>
Gap cutoff. Do not score columns that contain more than gap
cutoff fraction gaps. Default=.3.</p>

<p style="margin-top: 1em">-h Print help.</p>

<p style="margin-top: 1em">-l [true|false] <br>
Use sequence weighting. Default=true.</p>

<p style="margin-top: 1em">-m [FILE] <br>
Similarity matrix file, e.g. matrix/blosum62.bla or .qij.
Default=matrix/blosum62.bla.</p>

<p style="margin-top: 1em">Some methods, e.g.
js_divergence, do not use this.</p>

<p style="margin-top: 1em">-n [true|false] <br>
Normalize scores. Print the z-score (over the alignment) of
each column raw score. Default=false.</p>

<p style="margin-top: 1em">-o FILE <br>
Output file. Default: standard output stream.</p>

<p style="margin-top: 1em">-p [true|false] <br>
Use gap penalty. Lower the score of columns that contain
gaps, proportionally to the sum weight of the gapped
sequences. Default=true.</p>

<p style="margin-top: 1em">-s [METHOD] <br>
Conservation estimation method, one of shannon_entropy
property_entropy property_relative_entropy vn_entropy
relative_entropy js_divergence sum_of_pairs. <br>
Default=js_divergence.</p>

<p style="margin-top: 1em">-w [0-INT] <br>
Window size. Number of residues on either side included in
the window. Default=3.</p>

<p style="margin-top: 1em">EXAMPLES <br>
Note: you may have to copy and uncompress the example data
files before running the following examples.</p>

<p style="margin-top: 1em">Compute conservation scores for
the alignment using the Jensen-Shannon divergence with
default settings and print out the scores: <br>
score_conservation
/usr/share/doc/conservation-code/examples/2plc__hssp-filtered.aln</p>

<p style="margin-top: 1em">Score an alignment using
Jensen-Shannon divergence, a window of size 3 (on either
side of the residue), and the swissprot background
distribution: <br>
score_conservation -s js_divergence -w 3 -d
/usr/share/conservation-code/distributions/swissprot.distribution
/usr/share/doc/conservation-code/examples/2plc__hssp-filtered.aln</p>

<p style="margin-top: 1em">FILES <br>
Distributions <br>
/usr/share/conservation-code/distributions</p>

<p style="margin-top: 1em">Matrices <br>
/usr/share/conservation-code/matrix</p>

<p style="margin-top: 1em">SEE ALSO <br>
Homepage
&lt;http://compbio.cs.princeton.edu/conservation/&gt; <br>
Publication
&lt;http://bioinformatics.oxfordjournals.org/cgi/content/full/23/15/1875&gt;
<br>
concavity(1)</p>

<p style="margin-top: 1em">20110309.0 2016-06-02
DEBIAN/SCORE_CONSERVATION(1)</p>
<hr>
</body>
</html>
