<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>DEBIAN/SCORE_CONSERVATION(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">DEBIAN/SCORE_CONSERVATION(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">DEBIAN/SCORE_CONSERVATION(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
score_conservation - score protein sequence conservation
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
score_conservation [options] ALIGNFILE
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Score protein sequence conservation in <b>ALIGNFILE</b>. <b>ALIGNFILE</b> must
  be in FASTA, CLUSTAL or Stockholm format.
<div class="Pp"></div>
The following conservation scoring methods are implemented:
<br/>
 * sum of pairs
<br/>
 * weighted sum of pairs
<br/>
 * Shannon entropy
<br/>
 * Shannon entropy with property groupings (Mirny and Shakhnovich 1995,
<br/>
 Valdar and Thornton 2001)
<br/>
 * relative entropy with property groupings (Williamson 1995)
<br/>
 * von Neumann entropy (Caffrey et al 2004)
<br/>
 * relative entropy (Samudrala and Wang 2006)
<br/>
 * Jensen-Shannon divergence (Capra and Singh 2007)
<div class="Pp"></div>
A window-based extension that incorporates the estimated conservation of
  sequentially adjacent residues into the score for each column is also given.
  This window approach can be applied to any of the conservation scoring
  methods.
<div class="Pp"></div>
With default parameters <i>score_conservation</i>(1) computes the conservation
  scores for the alignment using the Jensen-Shannon divergence and a window
  <b>-w</b> of <i>3</i>.
<div class="Pp"></div>
The sequence-specific output can be used as the conservation input for
  <i>concavity</i>(1).
<div class="Pp"></div>
Conservation is highly predictive in identifying catalytic sites and residues
  near bound ligands.
<h1 class="Sh" title="Sh" id="REFERENCES"><a class="selflink" href="#REFERENCES">REFERENCES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">Capra JA and Singh M. Predicting functionally important
    residues from sequence conservation. Bioinformatics, 23(15):1875-82,
    2007.</dt>
  <dd class="It-tag"></dd>
</dl>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">-a [NAME]</dt>
  <dd class="It-tag">Reference sequence. Print scores in reference to the named
      sequence (ignoring gaps). Default prints the entire column.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-b [0-1]</dt>
  <dd class="It-tag">Lambda for window heuristic linear combination.
      Default=<i>.5</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    Equation:
    <div style="height: 1.00em;">&#x00A0;</div>
    &quot;score = (1 - lambda) * average_score_over_window_around_middle +
      lambda * score_of_middle&quot;</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-d [FILE]</dt>
  <dd class="It-tag">Background distribution file, e.g.
      <i>distributions/swissprot.distribution</i>. Default=built-in
    BLOSUM62.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-g [0-1)]</dt>
  <dd class="It-tag">Gap cutoff. Do not score columns that contain more than gap
      cutoff fraction gaps. Default= <i>.3</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-h</dt>
  <dd class="It-tag">Print help.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-l [true|false]</dt>
  <dd class="It-tag">Use sequence weighting. Default=<i>true</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-m [FILE]</dt>
  <dd class="It-tag">Similarity matrix file, e.g. <i>matrix/blosum62.bla</i> or
      .qij. Default= <i>matrix/blosum62.bla</i>.
    <div style="height: 1.00em;">&#x00A0;</div>
    Some methods, e.g. <i>js_divergence</i>, do not use this.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-n [true|false]</dt>
  <dd class="It-tag">Normalize scores. Print the z-score (over the alignment) of
      each column raw score. Default= <i>false</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-o FILE</dt>
  <dd class="It-tag">Output file. Default: standard output stream.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-p [true|false]</dt>
  <dd class="It-tag">Use gap penalty. Lower the score of columns that contain
      gaps, proportionally to the sum weight of the gapped sequences. Default=
      <i>true</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-s [METHOD]</dt>
  <dd class="It-tag">Conservation estimation method, one of <i>shannon_entropy
      property_entropy property_relative_entropy vn_entropy relative_entropy
      js_divergence sum_of_pairs</i>. Default= <i>js_divergence</i>.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-w [0-INT]</dt>
  <dd class="It-tag">Window size. Number of residues on either side included in
      the window. Default= <i>3</i>.</dd>
</dl>
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
Note: you may have to copy and uncompress the example data files before running
  the following examples.
<dl class="Bl-tag">
  <dt class="It-tag">Compute conservation scores for the alignment using the
    Jensen-Shannon divergence with default settings and print out the
    scores:</dt>
  <dd class="It-tag">
    <pre>
 score_conservation /usr/share/doc/conservation-code/examples/2plc__hssp-filtered.aln
    </pre>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Score an alignment using Jensen-Shannon divergence, a
    window of size 3 (on either side of the residue), and the swissprot
    background distribution:</dt>
  <dd class="It-tag">
    <pre>
 score_conservation -s js_divergence -w 3 -d \
  /usr/share/conservation-code/distributions/swissprot.distribution \
  /usr/share/doc/conservation-code/examples/2plc__hssp-filtered.aln
    </pre>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="FILES"><a class="selflink" href="#FILES">FILES</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">Distributions</dt>
  <dd class="It-tag"><i>/usr/share/conservation-code/distributions</i></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Matrices</dt>
  <dd class="It-tag"><i>/usr/share/conservation-code/matrix</i></dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">Homepage
    &lt;http://compbio.cs.princeton.edu/conservation/&gt;</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">Publication
    &lt;http://bioinformatics.oxfordjournals.org/cgi/content/full/23/15/1875&gt;</dt>
  <dd class="It-tag"></dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><i>concavity</i>(1)</dt>
  <dd class="It-tag"></dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">2016-06-02</td>
    <td class="foot-os">20110309.0</td>
  </tr>
</table>
</body>
</html>
