<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>SEQNR(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">SEQNR(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">SEQNR(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
seqnr - Removes redundancy from DHF files.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>seqnr</b>
  <b>-dhfinpath&#x00A0;</b><b></b><i>dirlist</i>
  <b>-dosing&#x00A0;</b><b></b><i>toggle</i>
  <b>-singletsdir&#x00A0;</b><b></b><i>directory</i>
  <b>-dosets&#x00A0;</b><b></b> <i>toggle</i>
  <b>-insetsdir&#x00A0;</b><b></b><i>directory</i>
  [<b>-matrix&#x00A0;</b><b></b> <i>matrixf</i>]
  <b>-mode&#x00A0;</b><b></b><i>list</i>
  <b>-thresh&#x00A0;</b><b></b><i>float</i>
  <b>-threshlow&#x00A0;</b><b></b><i>float</i>
  <b>-threshup&#x00A0;</b><b></b><i>float</i>
  [<b>-gapopen&#x00A0;</b><b></b><i>float</i>] [
  <b>-gapextend&#x00A0;</b><b></b> <i>float</i>]
  <b>-dhfoutdir&#x00A0;</b><b></b> <i>outdir</i>
  <b>-dored&#x00A0;</b><b></b><i>toggle</i>
  <b>-redoutdir&#x00A0;</b><b></b><i>outdir</i>
  <b>-logfile&#x00A0;</b><b></b><i>outfile</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 6.00ex; text-indent: -6.00ex;"><b>seqnr</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>seqnr</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Utils:Database creation&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-dhfinpath</b> <i>dirlist</i>
<div style="margin-left: 4.00ex;">This option specifies the location of DHF
  files (domain hits files) (input). A 'domain hits file' contains database hits
  (sequences) with domain classification information, in the DHF format (FASTA
  or EMBL-like). The hits are relatives to a SCOP or CATH family and are found
  from a search of a sequence database. Files containing hits retrieved by
  PSIBLAST are generated by using SEQSEARCH. Default value: ./</div>
<div class="Pp"></div>
<b>-dosing</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">This option specifies whether to use singlet
  sequences (e.g. DHF files) to filter input. Optionally, up to two further
  directories of sequences may be read: these are considered in the redundancy
  calculation but never appear in the output files. Default value: Y</div>
<div class="Pp"></div>
<b>-singletsdir</b> <i>directory</i>
<div style="margin-left: 4.00ex;">This option specifies the location of singlet
  filter sequences (e.g. DHF files) (input). A 'domain hits file' contains
  database hits (sequences) with domain classification information, in the DHF
  format (FASTA or EMBL-like). The hits are relatives to a SCOP or CATH family
  and are found from a search of a sequence database. Files containing hits
  retrieved by PSIBLAST are generated by using SEQSEARCH. Default value:
  ./</div>
<div class="Pp"></div>
<b>-dosets</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">This option specifies whether to use sets of
  sequences (e.g. DHF files) to filter input. Optionally, up to two further
  directories of sequences may be read: these are considered in the redundancy
  calculation but never appear in the output files. Default value: Y</div>
<div class="Pp"></div>
<b>-insetsdir</b> <i>directory</i>
<div style="margin-left: 4.00ex;">This option specifies location of sets of
  filter sequences (e.g. DAF files) (input). A 'domain alignment file' contains
  a sequence alignment of domains belonging to the same SCOP or CATH family. The
  file is in clustal format annotated with domain family classification
  information. The files generated by using SCOPALIGN will contain a
  structure-based sequence alignment of domains of known structure only. Such
  alignments can be extended with sequence relatives (of unknown structure) by
  using SEQALIGN. Default value: ./</div>
<div class="Pp"></div>
<b>-matrix</b> <i>matrixf</i>
<div style="margin-left: 4.00ex;">This option specifies the residue substitution
  matrix that is used for sequence comparison. Default value: EBLOSUM62</div>
<h2 class="Ss" title="Ss" id="Required_section"><a class="selflink" href="#Required_section">Required
  section</a></h2>
<b>-mode</b> <i>list</i>
<div style="margin-left: 4.00ex;">This option specifies whether to remove
  redundancy at a single threshold % sequence similarity or remove redundancy
  outside a range of acceptable threshold % similarity. All permutations of
  pair-wise sequence alignments are calculated for each set of input sequences
  in turn using the EMBOSS implementation of the Needleman and Wunsch global
  alignment algorithm. Redundant sequences are removed in one of two modes as
  follows: (i) If a pair of proteins achieve greater than a threshold percentage
  sequence similarity (specified by the user) the shortest sequence is
  discarded. (ii) If a pair of proteins have a percentage sequence similarity
  that lies outside an acceptable range (specified by the user) the shortest
  sequence is discarded. Default value: 1</div>
<div class="Pp"></div>
<b>-thresh</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the % sequence identity
  redundancy threshold. The % sequence identity redundancy threshold determines
  the redundancy calculation. If a pair of proteins achieve greater than this
  threshold the shortest sequence is discarded. Default value: 95.0</div>
<div class="Pp"></div>
<b>-threshlow</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the % sequence identity
  redundancy threshold (lower limit). The % sequence identity redundancy
  threshold determines the redundancy calculation. If a pair of proteins have a
  percentage sequence similarity that lies outside an acceptable range the
  shortest sequence is discarded. Default value: 30.0</div>
<div class="Pp"></div>
<b>-threshup</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the % sequence identity
  redundancy threshold (upper limit). The % sequence identity redundancy
  threshold determines the redundancy calculation. If a pair of proteins have a
  percentage sequence similarity that lies outside an acceptable range the
  shortest sequence is discarded. Default value: 90.0</div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-gapopen</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the gap insertion
  penalty. The gap insertion penalty is the score taken away when a gap is
  created. The best value depends on the choice of comparison matrix. The
  default value assumes you are using the EBLOSUM62 matrix for protein
  sequences, and the EDNAFULL matrix for nucleotide sequences. Default value:
  10</div>
<div class="Pp"></div>
<b>-gapextend</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the gap extension
  penalty. The gap extension, penalty is added to the standard gap penalty for
  each base or residue in the gap. This is how long gaps are penalized. Usually
  you will expect a few long gaps rather than many short gaps, so the gap
  extension penalty should be lower than the gap penalty. Default value:
  0.5</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-dhfoutdir</b> <i>outdir</i>
<div style="margin-left: 4.00ex;">This option specifies the location of DHF
  files (domain hits files) of non-redundant sequences (output). A 'domain hits
  file' contains database hits (sequences) with domain classification
  information, in the DHF format (FASTA or EMBL-like). The hits are relatives to
  a SCOP or CATH family and are found from a search of a sequence database.
  Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH.
  Default value: ./</div>
<div class="Pp"></div>
<b>-dored</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">This option specifies whether to retain
  redundant sequences. If this option is set a DHF file (domain hits file) of
  redundant sequences is written. Default value: N</div>
<div class="Pp"></div>
<b>-redoutdir</b> <i>outdir</i>
<div style="margin-left: 4.00ex;">This option specifies the location of DHF
  files (domain hits files) of redundant sequences (output). A 'domain hits
  file' contains database hits (sequences) with domain classification
  information, in the DHF format (FASTA or EMBL-like). The hits are relatives to
  a SCOP or CATH family and are found from a search of a sequence database.
  Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH.
  Default value: ./</div>
<div class="Pp"></div>
<b>-logfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of SEQNR log
  file (output). The log file contains messages about any errors arising while
  SEQNR ran. Default value: seqnr.log</div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
seqnr is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">08/11/2010</td>
    <td class="foot-os">DOMSEARCH 0.1.0++20100721</td>
  </tr>
</table>
</body>
</html>
