<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:37:19 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SEQNR(1e) EMBOSS Manual for Debian SEQNR(1e)</p>

<p style="margin-top: 1em">NAME <br>
seqnr - Removes redundancy from DHF files.</p>

<p style="margin-top: 1em">SYNOPSIS <br>
seqnr -dhfinpath dirlist -dosing toggle -singletsdir
directory -dosets toggle -insetsdir directory [-matrix
matrixf] -mode list -thresh float -threshlow float -threshup
float <br>
[-gapopen float] [-gapextend float] -dhfoutdir outdir -dored
toggle -redoutdir outdir -logfile outfile</p>

<p style="margin-top: 1em">seqnr -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
seqnr is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Utils:Database creation&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-dhfinpath dirlist <br>
This option specifies the location of DHF files (domain hits
files) (input). A &rsquo;domain hits file&rsquo; contains
database hits (sequences) with domain classification
information, <br>
in the DHF format (FASTA or EMBL-like). The hits are
relatives to a SCOP or CATH family and are found from a
search of a sequence database. Files containing hits
retrieved by <br>
PSIBLAST are generated by using SEQSEARCH. Default value:
./</p>

<p style="margin-top: 1em">-dosing toggle <br>
This option specifies whether to use singlet sequences (e.g.
DHF files) to filter input. Optionally, up to two further
directories of sequences may be read: these are <br>
considered in the redundancy calculation but never appear in
the output files. Default value: Y</p>

<p style="margin-top: 1em">-singletsdir directory <br>
This option specifies the location of singlet filter
sequences (e.g. DHF files) (input). A &rsquo;domain hits
file&rsquo; contains database hits (sequences) with domain
classification <br>
information, in the DHF format (FASTA or EMBL-like). The
hits are relatives to a SCOP or CATH family and are found
from a search of a sequence database. Files containing hits
<br>
retrieved by PSIBLAST are generated by using SEQSEARCH.
Default value: ./</p>

<p style="margin-top: 1em">-dosets toggle <br>
This option specifies whether to use sets of sequences (e.g.
DHF files) to filter input. Optionally, up to two further
directories of sequences may be read: these are <br>
considered in the redundancy calculation but never appear in
the output files. Default value: Y</p>

<p style="margin-top: 1em">-insetsdir directory <br>
This option specifies location of sets of filter sequences
(e.g. DAF files) (input). A &rsquo;domain alignment
file&rsquo; contains a sequence alignment of domains
belonging to the same <br>
SCOP or CATH family. The file is in clustal format annotated
with domain family classification information. The files
generated by using SCOPALIGN will contain a <br>
structure-based sequence alignment of domains of known
structure only. Such alignments can be extended with
sequence relatives (of unknown structure) by using SEQALIGN.
<br>
Default value: ./</p>

<p style="margin-top: 1em">-matrix matrixf <br>
This option specifies the residue substitution matrix that
is used for sequence comparison. Default value:
EBLOSUM62</p>

<p style="margin-top: 1em">Required section <br>
-mode list <br>
This option specifies whether to remove redundancy at a
single threshold % sequence similarity or remove redundancy
outside a range of acceptable threshold % similarity. All
<br>
permutations of pair-wise sequence alignments are calculated
for each set of input sequences in turn using the EMBOSS
implementation of the Needleman and Wunsch global <br>
alignment algorithm. Redundant sequences are removed in one
of two modes as follows: (i) If a pair of proteins achieve
greater than a threshold percentage sequence similarity <br>
(specified by the user) the shortest sequence is discarded.
(ii) If a pair of proteins have a percentage sequence
similarity that lies outside an acceptable range (specified
<br>
by the user) the shortest sequence is discarded. Default
value: 1</p>

<p style="margin-top: 1em">-thresh float <br>
This option specifies the % sequence identity redundancy
threshold. The % sequence identity redundancy threshold
determines the redundancy calculation. If a pair of proteins
<br>
achieve greater than this threshold the shortest sequence is
discarded. Default value: 95.0</p>

<p style="margin-top: 1em">-threshlow float <br>
This option specifies the % sequence identity redundancy
threshold (lower limit). The % sequence identity redundancy
threshold determines the redundancy calculation. If a <br>
pair of proteins have a percentage sequence similarity that
lies outside an acceptable range the shortest sequence is
discarded. Default value: 30.0</p>

<p style="margin-top: 1em">-threshup float <br>
This option specifies the % sequence identity redundancy
threshold (upper limit). The % sequence identity redundancy
threshold determines the redundancy calculation. If a <br>
pair of proteins have a percentage sequence similarity that
lies outside an acceptable range the shortest sequence is
discarded. Default value: 90.0</p>

<p style="margin-top: 1em">Additional section <br>
-gapopen float <br>
This option specifies the gap insertion penalty. The gap
insertion penalty is the score taken away when a gap is
created. The best value depends on the choice of comparison
<br>
matrix. The default value assumes you are using the
EBLOSUM62 matrix for protein sequences, and the EDNAFULL
matrix for nucleotide sequences. Default value: 10</p>

<p style="margin-top: 1em">-gapextend float <br>
This option specifies the gap extension penalty. The gap
extension, penalty is added to the standard gap penalty for
each base or residue in the gap. This is how long gaps <br>
are penalized. Usually you will expect a few long gaps
rather than many short gaps, so the gap extension penalty
should be lower than the gap penalty. Default value: 0.5</p>

<p style="margin-top: 1em">Output section <br>
-dhfoutdir outdir <br>
This option specifies the location of DHF files (domain hits
files) of non-redundant sequences (output). A &rsquo;domain
hits file&rsquo; contains database hits (sequences) with
domain <br>
classification information, in the DHF format (FASTA or
EMBL-like). The hits are relatives to a SCOP or CATH family
and are found from a search of a sequence database. Files
<br>
containing hits retrieved by PSIBLAST are generated by using
SEQSEARCH. Default value: ./</p>

<p style="margin-top: 1em">-dored toggle <br>
This option specifies whether to retain redundant sequences.
If this option is set a DHF file (domain hits file) of
redundant sequences is written. Default value: N</p>

<p style="margin-top: 1em">-redoutdir outdir <br>
This option specifies the location of DHF files (domain hits
files) of redundant sequences (output). A &rsquo;domain hits
file&rsquo; contains database hits (sequences) with domain
<br>
classification information, in the DHF format (FASTA or
EMBL-like). The hits are relatives to a SCOP or CATH family
and are found from a search of a sequence database. Files
<br>
containing hits retrieved by PSIBLAST are generated by using
SEQSEARCH. Default value: ./</p>

<p style="margin-top: 1em">-logfile outfile <br>
This option specifies the name of SEQNR log file (output).
The log file contains messages about any errors arising
while SEQNR ran. Default value: seqnr.log</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
seqnr is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">DOMSEARCH 0.1.0++20100721
08/11/2010 SEQNR(1e)</p>
<hr>
</body>
</html>
