<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:37:20 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>seqstat(1) Biosquid Manual seqstat(1)</p>

<p style="margin-top: 1em">NAME <br>
seqstat - show statistics and format for a sequence file</p>

<p style="margin-top: 1em">SYNOPSIS <br>
seqstat [options] seqfile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
seqstat reads a sequence file seqfile and shows a number of
simple statistics about it.</p>

<p style="margin-top: 1em">The printed statistics include
the name of the format, the residue type of the first
sequence (protein, RNA, or DNA), the number of sequences,
the total number of residues, and <br>
the average and range of the sequence lengths.</p>

<p style="margin-top: 1em">OPTIONS <br>
-a Show additional verbose information: a table with one
line per sequence showing name, length, and description
line. These lines are prefixed with a * character to enable
<br>
easily grep&rsquo;ing them out and sorting them.</p>

<p style="margin-top: 1em">-h Print brief help; includes
version number and summary of all options, including expert
options.</p>

<p style="margin-top: 1em">-B (Babelfish). Autodetect and
read a sequence file format other than the default (FASTA).
Almost any common sequence file format is recognized
(including Genbank, EMBL, <br>
SWISS-PROT, PIR, and GCG unaligned sequence formats, and
Stockholm, GCG MSF, and Clustal alignment formats). See the
printed documentation for a complete list of supported <br>
formats.</p>

<p style="margin-top: 1em">EXPERT OPTIONS <br>
--informat &lt;s&gt; <br>
Specify that the sequence file is in format &lt;s&gt;,
rather than the default FASTA format. Common examples
include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF, or
<br>
PHYLIP; see the printed documentation for a complete list of
accepted format names. This option overrides the default
expected format (FASTA) and the -B Babelfish autode&acirc;
<br>
tection option.</p>

<p style="margin-top: 1em">--quiet <br>
Suppress the verbose header (program name, release number
and date, the parameters and options in effect).</p>

<p style="margin-top: 1em">SEE ALSO <br>
afetch(1), alistat(1), compalign(1), compstruct(1),
revcomp(1), seqsplit(1), sfetch(1), shuffle(1), sindex(1),
sreformat(1), stranslate(1), weight(1).</p>

<p style="margin-top: 1em">AUTHOR <br>
Biosquid and its documentation are Copyright (C) 1992-2003
HHMI/Washington University School of Medicine Freely
distributed under the GNU General Public License (GPL) See
COPYING <br>
in the source code distribution for more details, or contact
me.</p>

<p style="margin-top: 1em">Sean Eddy <br>
HHMI/Department of Genetics <br>
Washington University School of Medicine <br>
4444 Forest Park Blvd., Box 8510 <br>
St Louis, MO 63108 USA <br>
Phone: 1-314-362-7666 <br>
FAX : 1-314-362-2157 <br>
Email: eddy@genetics.wustl.edu</p>

<p style="margin-top: 1em">Biosquid 1.9g January 2003
seqstat(1)</p>
<hr>
</body>
</html>
