<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:37:52 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SHOWFEAT(1e) EMBOSS Manual for Debian SHOWFEAT(1e)</p>

<p style="margin-top: 1em">NAME <br>
showfeat - Display features of a sequence in pretty
format</p>

<p style="margin-top: 1em">SYNOPSIS <br>
showfeat -sequence seqall [-sourcematch string] [-typematch
string] [-tagmatch string] [-valuematch string] [-sort list]
[-joinfeatures boolean] [-annotation range] -html boolean
<br>
-id boolean -description boolean -scale boolean -width
integer -collapse boolean -forward boolean -reverse boolean
-unknown boolean -strand boolean -origin boolean <br>
-position boolean -type boolean -tags boolean -values
boolean -stricttags boolean -outfile outfile</p>

<p style="margin-top: 1em">showfeat -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
showfeat is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Display,Feature tables&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall</p>

<p style="margin-top: 1em">Additional section <br>
-sourcematch string <br>
By default any feature source in the feature table is shown.
You can set this to match any feature source you wish to
show. The source name is usually either the name of the <br>
program that detected the feature or it is the feature table
(eg: EMBL) that the feature came from. The source may be
wildcarded by using &rsquo;*&rsquo;. If you wish to show
more than <br>
one source, separate their names with the character
&rsquo;|&rsquo;, eg: gene* | embl Default value: *</p>

<p style="margin-top: 1em">-typematch string <br>
By default any feature type in the feature table is shown.
You can set this to match any feature type you wish to show.
See http://www.ebi.ac.uk/embl/WebFeat/ for a list of <br>
the EMBL feature types and see Appendix A of the Swissprot
user manual in http://www.expasy.org/sprot/userman.html for
a list of the Swissprot feature types. The type may be <br>
wildcarded by using &rsquo;*&rsquo;. If you wish to show
more than one type, separate their names with the character
&rsquo;|&rsquo;, eg: *UTR | intron Default value: *</p>

<p style="margin-top: 1em">-tagmatch string <br>
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, <br>
&rsquo;/db_xref&rsquo;, &rsquo;/EC_number&rsquo;,
&rsquo;/evidence&rsquo;, &rsquo;/exception&rsquo;,
&rsquo;/function&rsquo;, &rsquo;/gene&rsquo;,
&rsquo;/label&rsquo;, &rsquo;/map&rsquo;,
&rsquo;/note&rsquo;, &rsquo;/number&rsquo;,
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;, <br>
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Some of these tags also have
values, for example &rsquo;/gene&rsquo; can have the value
of the gene <br>
name. By default any feature tag in the feature table is
shown. You can set this to match any feature tag you wish to
show. The tag may be wildcarded by using &rsquo;*&rsquo;. If
you <br>
wish to show more than one tag, separate their names with
the character &rsquo;|&rsquo;, eg: gene | label Default
value: *</p>

<p style="margin-top: 1em">-valuematch string <br>
Tag values are the values associated with a feature tag.
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of <br>
feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, &rsquo;/db_xref&rsquo;,
&rsquo;/EC_number&rsquo;, &rsquo;/evidence&rsquo;,
&rsquo;/exception&rsquo;, &rsquo;/function&rsquo;,
&rsquo;/gene&rsquo;, &rsquo;/label&rsquo;,
&rsquo;/map&rsquo;, &rsquo;/note&rsquo;,
&rsquo;/number&rsquo;, <br>
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;,
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Only some of these tags can have
values, <br>
for example &rsquo;/gene&rsquo; can have the value of the
gene name. By default any feature tag value in the feature
table is shown. You can set this to match any feature tag
value you <br>
wish to show. The tag value may be wildcarded by using
&rsquo;*&rsquo;. If you wish to show more than one tag
value, separate their names with the character
&rsquo;|&rsquo;, eg: pax* | 10 Default <br>
value: *</p>

<p style="margin-top: 1em">-sort list <br>
Default value: start</p>

<p style="margin-top: 1em">-joinfeatures boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-annotation range <br>
Regions to annotate by marking. If this is left blank, then
no annotation is added. A set of regions is specified by a
set of pairs of positions followed by optional text. <br>
The positions are integers. They are followed by any text
(but not digits when on the command-line). Examples of
region specifications are: 24-45 new domain 56-78 match to
<br>
Mouse 1-100 First part 120-156 oligo A file of ranges to
annotate (one range per line) can be specified as
&rsquo;@filename&rsquo;.</p>

<p style="margin-top: 1em">Advanced section <br>
-html boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-id boolean <br>
Set this to be false if you do not wish to display the ID
name of the sequence. Default value: Y</p>

<p style="margin-top: 1em">-description boolean <br>
Set this to be false if you do not wish to display the
description of the sequence. Default value: Y</p>

<p style="margin-top: 1em">-scale boolean <br>
Set this to be false if you do not wish to display the scale
line. Default value: Y</p>

<p style="margin-top: 1em">-width integer <br>
You can expand (or contract) the width of the
ASCII-character graphics display of the positions of the
features using this value. For example, a width of 80
characters would <br>
cover a standard page width and a width a 10 characters
would be nearly unreadable. If the width is set to less than
4, the graphics lines and the scale line will not be <br>
displayed. Default value: 60</p>

<p style="margin-top: 1em">-collapse boolean <br>
If this is set, then features from the same source and of
the same type and sense are all printed on the same line.
For instance if there are several features from the EMBL
<br>
feature table (ie. the same source) which are all of type
&rsquo;exon&rsquo; in the same sense, then they will all be
displayed on the same line. This makes it hard to
distinguish <br>
overlapping features. If this is set to false then each
feature is displayed on a separate line making it easier to
distinguish where features start and end. Default value:
N</p>

<p style="margin-top: 1em">-forward boolean <br>
Set this to be false if you do not wish to display forward
sense features. Default value: Y</p>

<p style="margin-top: 1em">-reverse boolean <br>
Set this to be false if you do not wish to display reverse
sense features. Default value: Y</p>

<p style="margin-top: 1em">-unknown boolean <br>
Set this to be false if you do not wish to display unknown
sense features. (ie. features with no directionality - all
protein features are of this type and some nucleic <br>
features (for example, CG-rich regions)). Default value:
Y</p>

<p style="margin-top: 1em">-strand boolean <br>
Set this if you wish to display the strand of the features.
Protein features are always directionless (indicated by
&rsquo;0&rsquo;), forward is indicated by &rsquo;+&rsquo;
and reverse is &rsquo;-&rsquo;. <br>
Default value: N</p>

<p style="margin-top: 1em">-origin boolean <br>
Set this if you wish to display the origin of the features.
The source name is usually either the name of the program
that detected the feature or it is the name of the <br>
feature table (eg: EMBL) that the feature came from. Default
value: N</p>

<p style="margin-top: 1em">-position boolean <br>
Set this if you wish to display the start and end position
of the features. If several features are being displayed on
the same line, then the start and end positions will be <br>
joined by a comma, for example:
&rsquo;189-189,225-225&rsquo;. Default value: N</p>

<p style="margin-top: 1em">-type boolean <br>
Set this to be false if you do not wish to display the type
of the features. Default value: Y</p>

<p style="margin-top: 1em">-tags boolean <br>
Set this to be false if you do not wish to display the tags
and values of the features. Default value: N</p>

<p style="margin-top: 1em">-values boolean <br>
Set this to be false if you do not wish to display the tag
values of the features. If this is set to be false, only the
tag names will be displayed. If the tags are not <br>
displayed, then the values will not be displayed. The value
of the &rsquo;translation&rsquo; tag is never displayed as
it is often extremely long. Default value: Y</p>

<p style="margin-top: 1em">-stricttags boolean <br>
By default if any tag/value pair in a feature matches the
specified tag and value, then all the tags/value pairs of
that feature will be displayed. If this is set to be true,
<br>
then only those tag/value pairs in a feature that match the
specified tag and value will be displayed. Default value:
N</p>

<p style="margin-top: 1em">Output section <br>
-outfile outfile</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
showfeat is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
SHOWFEAT(1e)</p>
<hr>
</body>
</html>
