<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>SHOWSEQ(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">SHOWSEQ(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">SHOWSEQ(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
showseq - Displays sequences with features in pretty format
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 8.00ex; text-indent: -8.00ex;"><b>showseq</b>
  <b>-sequence&#x00A0;</b><b></b><i>seqall</i>
  <b>-mfile&#x00A0;</b><b></b><i>datafile</i>
  <b>-format&#x00A0;</b><b></b><i>list</i>
  <b>-things&#x00A0;</b><b></b><i>list</i> [
  <b>-translate&#x00A0;</b><b></b><i>range</i>] [
  <b>-revtranslate&#x00A0;</b><b></b><i>range</i>]
  [<b>-uppercase&#x00A0;</b><b></b> <i>range</i>]
  [<b>-highlight&#x00A0;</b><b></b> <i>range</i>]
  [<b>-annotation&#x00A0;</b><b></b> <i>range</i>]
  [<b>-enzymes&#x00A0;</b><b></b> <i>string</i>]
  [<b>-table&#x00A0;</b><b></b><i>list</i>]
  [<b>-sourcematch&#x00A0;</b><b></b><i>string</i>]
  [<b>-typematch&#x00A0;</b><b></b><i>string</i>]
  [<b>-sensematch&#x00A0;</b><b></b> <i>integer</i>]
  [<b>-minscore&#x00A0;</b><b></b><i>float</i>]
  [<b>-maxscore&#x00A0;</b><b></b><i>float</i>]
  [<b>-tagmatch&#x00A0;</b><b></b><i>string</i>]
  [<b>-valuematch&#x00A0;</b><b></b> <i>string</i>]
  [<b>-stricttags&#x00A0;</b><b></b><i>boolean</i>]
  <b>-flatreformat&#x00A0;</b><b></b><i>boolean</i>
  <b>-mincuts&#x00A0;</b><b></b> <i>integer</i>
  <b>-maxcuts&#x00A0;</b><b></b><i>integer</i>
  <b>-sitelen&#x00A0;</b><b></b><i>integer</i>
  <b>-single&#x00A0;</b><b></b><i>boolean</i>
  <b>-blunt&#x00A0;</b><b></b><i>boolean</i>
  <b>-sticky&#x00A0;</b><b></b><i>boolean</i>
  <b>-ambiguity&#x00A0;</b><b></b><i>boolean</i>
  <b>-plasmid&#x00A0;</b><b></b><i>boolean</i>
  <b>-methylation&#x00A0;</b><b></b> <i>boolean</i>
  <b>-commercial&#x00A0;</b><b></b><i>boolean</i>
  <b>-limit&#x00A0;</b><b></b><i>boolean</i>
  <b>-orfminsize&#x00A0;</b><b></b><i>integer</i>
  <b>-threeletter&#x00A0;</b><b></b> <i>boolean</i>
  <b>-number&#x00A0;</b><b></b> <i>boolean</i>
  <b>-width&#x00A0;</b><b></b><i>integer</i>
  <b>-length&#x00A0;</b><b></b><i>integer</i>
  <b>-margin&#x00A0;</b><b></b><i>integer</i>
  <b>-name&#x00A0;</b><b></b><i>boolean</i>
  <b>-description&#x00A0;</b><b></b><i>boolean</i>
  <b>-offset&#x00A0;</b><b></b><i>integer</i>
  <b>-html&#x00A0;</b><b></b><i>boolean</i>
  <b>-outfile&#x00A0;</b><b></b><i>outfile</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 8.00ex; text-indent: -8.00ex;"><b>showseq</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>showseq</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Nucleic:Translation,Nucleic:Restriction&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-sequence</b> <i>seqall</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-mfile</b> <i>datafile</i>
<div style="margin-left: 4.00ex;">Default value: Emethylsites.dat</div>
<h2 class="Ss" title="Ss" id="Required_section"><a class="selflink" href="#Required_section">Required
  section</a></h2>
<b>-format</b> <i>list</i>
<div style="margin-left: 4.00ex;">Default value: 2</div>
<div class="Pp"></div>
<b>-things</b> <i>list</i>
<div style="margin-left: 4.00ex;">Specify a list of one or more code characters
  in the order in which you wish things to be displayed one above the other down
  the page. For example if you wish to see things displayed in the order:
  sequence, complement sequence, ticks line, frame 1 translation, blank line;
  then you should enter 'S,C,T,1,B'. Default value: B,N,T,S,A,F</div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-translate</b> <i>range</i>
<div style="margin-left: 4.00ex;">Regions to translate (if translating). If this
  is left blank the complete sequence is translated. A set of regions is
  specified by a set of pairs of positions. The positions are integers. They are
  separated by any non-digit, non-alpha character. Examples of region
  specifications are: 24-45, 56-78 1:45, 67=99;765..888</div>
<div class="Pp"></div>
<b>-revtranslate</b> <i>range</i>
<div style="margin-left: 4.00ex;">Regions to translate (if translating). If this
  is left blank the complete sequence is translated. A set of regions is
  specified by a set of pairs of positions. The positions are integers. They are
  separated by any non-digit, non-alpha character. Examples of region
  specifications are: 78-56, 45-24, 888..765, 99=67; 45:1</div>
<div class="Pp"></div>
<b>-uppercase</b> <i>range</i>
<div style="margin-left: 4.00ex;">Regions to put in uppercase. If this is left
  blank, then the sequence case is left alone. A set of regions is specified by
  a set of pairs of positions. The positions are integers. They are separated by
  any non-digit, non-alpha character. Examples of region specifications are:
  24-45, 56-78 1:45, 67=99;765..888 1,5,8,10,23,45,57,99</div>
<div class="Pp"></div>
<b>-highlight</b> <i>range</i>
<div style="margin-left: 4.00ex;">Regions to colour if formatting for HTML. If
  this is left blank, then the sequence is left alone. A set of regions is
  specified by a set of pairs of positions. The positions are integers. They are
  followed by any valid HTML font colour. Examples of region specifications are:
  24-45 blue 56-78 orange 1-100 green 120-156 red A file of ranges to colour
  (one range per line) can be specified as '@filename'.</div>
<div class="Pp"></div>
<b>-annotation</b> <i>range</i>
<div style="margin-left: 4.00ex;">Regions to annotate by marking. If this is
  left blank, then no annotation is added. A set of regions is specified by a
  set of pairs of positions followed by optional text. The positions are
  integers. They are followed by any text (but not digits when on the
  command-line). Examples of region specifications are: 24-45 new domain 56-78
  match to Mouse 1-100 First part 120-156 oligo A file of ranges to annotate
  (one range per line) can be specified as '@filename'.</div>
<div class="Pp"></div>
<b>-enzymes</b> <i>string</i>
<div style="margin-left: 4.00ex;">The name 'all' reads in all enzyme names from
  the REBASE database. You can specify enzymes by giving their names with commas
  between then, such as: 'HincII,hinfI,ppiI,hindiii'. The case of the names is
  not important. You can specify a file of enzyme names to read in by giving the
  name of the file holding the enzyme names with a '@' character in front of it,
  for example, '@enz.list'. Blank lines and lines starting with a hash character
  or '!' are ignored and all other lines are concatenated together with a comma
  character ',' and then treated as the list of enzymes to search for. An
  example of a file of enzyme names is: ! my enzymes HincII, ppiII ! other
  enzymes hindiii HinfI PpiI Default value: all</div>
<div class="Pp"></div>
<b>-table</b> <i>list</i>
<div style="margin-left: 4.00ex;"></div>
<h2 class="Ss" title="Ss" id="Feature_display_options"><a class="selflink" href="#Feature_display_options">Feature
  display options</a></h2>
<b>-sourcematch</b> <i>string</i>
<div style="margin-left: 4.00ex;">By default any feature source in the feature
  table is shown. You can set this to match any feature source you wish to show.
  The source name is usually either the name of the program that detected the
  feature or it is the feature table (eg: EMBL) that the feature came from. The
  source may be wildcarded by using '*'. If you wish to show more than one
  source, separate their names with the character '|', eg: gene* | embl Default
  value: *</div>
<div class="Pp"></div>
<b>-typematch</b> <i>string</i>
<div style="margin-left: 4.00ex;">By default any feature type in the feature
  table is shown. You can set this to match any feature type you wish to show.
  See http://www.ebi.ac.uk/embl/WebFeat/ for a list of the EMBL feature types
  and see Appendix A of the Swissprot user manual in
  http://www.expasy.org/sprot/userman.html for a list of the Swissprot feature
  types. The type may be wildcarded by using '*'. If you wish to show more than
  one type, separate their names with the character '|', eg: *UTR | intron
  Default value: *</div>
<div class="Pp"></div>
<b>-sensematch</b> <i>integer</i>
<div style="margin-left: 4.00ex;">By default any feature type in the feature
  table is shown. You can set this to match any feature sense you wish to show.
  0 - any sense, 1 - forward sense, -1 - reverse sense</div>
<div class="Pp"></div>
<b>-minscore</b> <i>float</i>
<div style="margin-left: 4.00ex;">Minimum score of feature to display (see also
  maxscore) Default value: 0.0</div>
<div class="Pp"></div>
<b>-maxscore</b> <i>float</i>
<div style="margin-left: 4.00ex;">Maximum score of feature to display. If both
  minscore and maxscore are zero (the default), then any score is ignored
  Default value: 0.0</div>
<div class="Pp"></div>
<b>-tagmatch</b> <i>string</i>
<div style="margin-left: 4.00ex;">Tags are the types of extra values that a
  feature may have. For example in the EMBL feature table, a 'CDS' type of
  feature may have the tags '/codon', '/codon_start', '/db_xref', '/EC_number',
  '/evidence', '/exception', '/function', '/gene', '/label', '/map', '/note',
  '/number', '/partial', '/product', '/protein_id', '/pseudo', '/standard_name',
  '/translation', '/transl_except', '/transl_table', or '/usedin'. Some of these
  tags also have values, for example '/gene' can have the value of the gene
  name. By default any feature tag in the feature table is shown. You can set
  this to match any feature tag you wish to show. The tag may be wildcarded by
  using '*'. If you wish to show more than one tag, separate their names with
  the character '|', eg: gene | label Default value: *</div>
<div class="Pp"></div>
<b>-valuematch</b> <i>string</i>
<div style="margin-left: 4.00ex;">Tag values are the values associated with a
  feature tag. Tags are the types of extra values that a feature may have. For
  example in the EMBL feature table, a 'CDS' type of feature may have the tags
  '/codon', '/codon_start', '/db_xref', '/EC_number', '/evidence', '/exception',
  '/function', '/gene', '/label', '/map', '/note', '/number', '/partial',
  '/product', '/protein_id', '/pseudo', '/standard_name', '/translation',
  '/transl_except', '/transl_table', or '/usedin'. Only some of these tags can
  have values, for example '/gene' can have the value of the gene name. By
  default any feature tag value in the feature table is shown. You can set this
  to match any feature tag value you wish to show. The tag value may be
  wildcarded by using '*'. If you wish to show more than one tag value, separate
  their names with the character '|', eg: pax* | 10 Default value: *</div>
<div class="Pp"></div>
<b>-stricttags</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">By default if any tag/value pair in a feature
  matches the specified tag and value, then all the tags/value pairs of that
  feature will be displayed. If this is set to be true, then only those
  tag/value pairs in a feature that match the specified tag and value will be
  displayed. Default value: N</div>
<h2 class="Ss" title="Ss" id="Advanced_section"><a class="selflink" href="#Advanced_section">Advanced
  section</a></h2>
<h2 class="Ss" title="Ss" id="Restriction_map_options"><a class="selflink" href="#Restriction_map_options">Restriction
  map options</a></h2>
<b>-flatreformat</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This changes the output format to one where
  the recognition site is indicated by a row of '===' characters and the cut
  site is pointed to by a '&gt;' character in the forward sense, or a '&lt;' in
  the reverse sense strand. Default value: N</div>
<div class="Pp"></div>
<b>-mincuts</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This sets the minimum number of cuts for any
  restriction enzyme that will be considered. Any enzymes that cut fewer times
  than this will be ignored. Default value: 1</div>
<div class="Pp"></div>
<b>-maxcuts</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This sets the maximum number of cuts for any
  restriction enzyme that will be considered. Any enzymes that cut more times
  than this will be ignored. Default value: 2000000000</div>
<div class="Pp"></div>
<b>-sitelen</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This sets the minimum length of the
  restriction enzyme recognition site. Any enzymes with sites shorter than this
  will be ignored. Default value: 4</div>
<div class="Pp"></div>
<b>-single</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this is set then this forces the values of
  the mincuts and maxcuts qualifiers to both be 1. Any other value you may have
  set them to will be ignored. Default value: N</div>
<div class="Pp"></div>
<b>-blunt</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This allows those enzymes which cut at the
  same position on the forward and reverse strands to be considered. Default
  value: Y</div>
<div class="Pp"></div>
<b>-sticky</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This allows those enzymes which cut at
  different positions on the forward and reverse strands, leaving an overhang,
  to be considered. Default value: Y</div>
<div class="Pp"></div>
<b>-ambiguity</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This allows those enzymes which have one or
  more 'N' ambiguity codes in their pattern to be considered Default value:
  Y</div>
<div class="Pp"></div>
<b>-plasmid</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this is set then this allows searches for
  restriction enzyme recognition site and cut positions that span the end of the
  sequence to be considered. Default value: N</div>
<div class="Pp"></div>
<b>-methylation</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this is set then RE recognition sites will
  not match methylated bases. Default value: N</div>
<div class="Pp"></div>
<b>-commercial</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">If this is set, then only those enzymes with a
  commercial supplier will be searched for. This qualifier is ignored if you
  have specified an explicit list of enzymes to search for, rather than
  searching through 'all' the enzymes in the REBASE database. It is assumed
  that, if you are asking for an explicit enzyme, then you probably know where
  to get it from and so all enzymes names that you have asked to be searched
  for, and which cut, will be reported whether or not they have a commercial
  supplier. Default value: Y</div>
<div class="Pp"></div>
<b>-limit</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">This limits the reporting of enzymes to just
  one enzyme from each group of isoschizomers. The enzyme chosen to represent an
  isoschizomer group is the prototype indicated in the data file 'embossre.equ',
  which is created by the program 'rebaseextract'. If you prefer different
  prototypes to be used, make a copy of embossre.equ in your home directory and
  edit it. If this value is set to be false then all of the input enzymes will
  be reported. You might like to set this to false if you are supplying an
  explicit set of enzymes rather than searching 'all' of them. Default value:
  Y</div>
<div class="Pp"></div>
<b>-orfminsize</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This sets the minimum size of Open Reading
  Frames (ORFs) to display in the translations. All other translation regions
  are masked by changing the amino acids to '-' characters.</div>
<div class="Pp"></div>
<b>-threeletter</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<div class="Pp"></div>
<b>-number</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<div class="Pp"></div>
<b>-width</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Default value: 60</div>
<div class="Pp"></div>
<b>-length</b> <i>integer</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-margin</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Default value: 10</div>
<div class="Pp"></div>
<b>-name</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Set this to be false if you do not wish to
  display the ID name of the sequence Default value: Y</div>
<div class="Pp"></div>
<b>-description</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Set this to be false if you do not wish to
  display the description of the sequence Default value: Y</div>
<div class="Pp"></div>
<b>-offset</b> <i>integer</i>
<div style="margin-left: 4.00ex;">Default value: 1</div>
<div class="Pp"></div>
<b>-html</b> <i>boolean</i>
<div style="margin-left: 4.00ex;">Default value: N</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
showseq is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
