<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:37:52 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SHOWPEP(1e) EMBOSS Manual for Debian SHOWPEP(1e)</p>

<p style="margin-top: 1em">NAME <br>
showpep - Displays protein sequences with features in pretty
format</p>

<p style="margin-top: 1em">SYNOPSIS <br>
showpep -sequence seqall -format list -things list
[-uppercase range] [-highlight range] [-annotation range]
[-sourcematch string] [-typematch string] [-minscore float]
<br>
[-maxscore float] [-tagmatch string] [-valuematch string]
[-stricttags boolean] -threeletter boolean -number boolean
-width integer -length integer -margin integer <br>
-name boolean -description boolean -offset integer -html
boolean -outfile outfile</p>

<p style="margin-top: 1em">showpep -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
showpep is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Display&quot; command group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall</p>

<p style="margin-top: 1em">Required section <br>
-format list <br>
Default value: 2</p>

<p style="margin-top: 1em">-things list <br>
Specify a list of one or more code characters in the order
in which you wish things to be displayed one above the other
down the page. For example if you wish to see things <br>
displayed in the order: sequence, ticks line, blank line;
then you should enter &rsquo;S,T,B&rsquo;. Default value:
B,N,T,S,A,F</p>

<p style="margin-top: 1em">Additional section <br>
-uppercase range <br>
Regions to put in uppercase. If this is left blank, then the
sequence case is left alone. A set of regions is specified
by a set of pairs of positions. The positions are <br>
integers. They are separated by any non-digit, non-alpha
character. Examples of region specifications are: 24-45,
56-78 1:45, 67=99;765..888 1,5,8,10,23,45,57,99</p>

<p style="margin-top: 1em">-highlight range <br>
Regions to colour if formatting for HTML. If this is left
blank, then the sequence is left alone. A set of regions is
specified by a set of pairs of positions. The positions <br>
are integers. They are followed by any valid HTML font
colour. Examples of region specifications are: 24-45 blue
56-78 orange 1-100 green 120-156 red A file of ranges to
<br>
colour (one range per line) can be specified as
&rsquo;@filename&rsquo;.</p>

<p style="margin-top: 1em">-annotation range <br>
Regions to annotate by marking. If this is left blank, then
no annotation is added. A set of regions is specified by a
set of pairs of positions followed by optional text. <br>
The positions are integers. They are followed by any text
(but not digits when on the command-line). Examples of
region specifications are: 24-45 new domain 56-78 match to
<br>
Mouse 1-100 First part 120-156 oligo A file of ranges to
annotate (one range per line) can be specified as
&rsquo;@filename&rsquo;.</p>

<p style="margin-top: 1em">Feature display options <br>
-sourcematch string <br>
By default any feature source in the feature table is shown.
You can set this to match any feature source you wish to
show. The source name is usually either the name of the <br>
program that detected the feature or it is the feature table
(eg: EMBL) that the feature came from. The source may be
wildcarded by using &rsquo;*&rsquo;. If you wish to show
more than <br>
one source, separate their names with the character
&rsquo;|&rsquo;, eg: gene* | embl Default value: *</p>

<p style="margin-top: 1em">-typematch string <br>
By default any feature type in the feature table is shown.
You can set this to match any feature type you wish to show.
See http://www.ebi.ac.uk/embl/WebFeat/ for a list of <br>
the EMBL feature types and see Appendix A of the Swissprot
user manual in http://www.expasy.org/sprot/userman.html for
a list of the Swissprot feature types. The type may be <br>
wildcarded by using &rsquo;*&rsquo;. If you wish to show
more than one type, separate their names with the character
&rsquo;|&rsquo;, eg: *UTR | intron Default value: *</p>

<p style="margin-top: 1em">-minscore float <br>
Minimum score of feature to display (see also maxscore)
Default value: 0.0</p>

<p style="margin-top: 1em">-maxscore float <br>
Maximum score of feature to display. If both minscore and
maxscore are zero (the default), then any score is ignored
Default value: 0.0</p>

<p style="margin-top: 1em">-tagmatch string <br>
Tags are the types of extra values that a feature may have.
By default any feature tag in the feature table is shown.
You can set this to match any feature tag you wish to <br>
show. The tag may be wildcarded by using &rsquo;*&rsquo;. If
you wish to show more than one tag, separate their names
with the character &rsquo;|&rsquo;, eg: gene | label Default
value: *</p>

<p style="margin-top: 1em">-valuematch string <br>
Tag values are the values associated with a feature tag.
Tags are the types of extra values that a feature may have.
By default any feature tag value in the feature table is
<br>
shown. You can set this to match any feature tag value you
wish to show. The tag value may be wildcarded by using
&rsquo;*&rsquo;. If you wish to show more than one tag
value, separate <br>
their names with the character &rsquo;|&rsquo;, eg: pax* |
10 Default value: *</p>

<p style="margin-top: 1em">-stricttags boolean <br>
By default if any tag/value pair in a feature matches the
specified tag and value, then all the tags/value pairs of
that feature will be displayed. If this is set to be true,
<br>
then only those tag/value pairs in a feature that match the
specified tag and value will be displayed. Default value:
N</p>

<p style="margin-top: 1em">Advanced section <br>
-threeletter boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-number boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-width integer <br>
Default value: 60</p>

<p style="margin-top: 1em">-length integer</p>

<p style="margin-top: 1em">-margin integer <br>
Default value: 10</p>

<p style="margin-top: 1em">-name boolean <br>
Set this to be false if you do not wish to display the ID
name of the sequence Default value: Y</p>

<p style="margin-top: 1em">-description boolean <br>
Set this to be false if you do not wish to display the
description of the sequence Default value: Y</p>

<p style="margin-top: 1em">-offset integer <br>
Default value: 1</p>

<p style="margin-top: 1em">-html boolean <br>
Default value: N</p>

<p style="margin-top: 1em">Output section <br>
-outfile outfile</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
showpep is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
SHOWPEP(1e)</p>
<hr>
</body>
</html>
