<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:37:53 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SHOWSEQ(1e) EMBOSS Manual for Debian SHOWSEQ(1e)</p>

<p style="margin-top: 1em">NAME <br>
showseq - Displays sequences with features in pretty
format</p>

<p style="margin-top: 1em">SYNOPSIS <br>
showseq -sequence seqall -mfile datafile -format list
-things list [-translate range] [-revtranslate range]
[-uppercase range] [-highlight range] [-annotation range]
<br>
[-enzymes string] [-table list] [-sourcematch string]
[-typematch string] [-sensematch integer] [-minscore float]
[-maxscore float] [-tagmatch string] <br>
[-valuematch string] [-stricttags boolean] -flatreformat
boolean -mincuts integer -maxcuts integer -sitelen integer
-single boolean -blunt boolean -sticky boolean <br>
-ambiguity boolean -plasmid boolean -methylation boolean
-commercial boolean -limit boolean -orfminsize integer
-threeletter boolean -number boolean -width integer <br>
-length integer -margin integer -name boolean -description
boolean -offset integer -html boolean -outfile outfile</p>

<p style="margin-top: 1em">showseq -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
showseq is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the
&quot;Nucleic:Translation,Nucleic:Restriction&quot; command
<br>
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall</p>

<p style="margin-top: 1em">-mfile datafile <br>
Default value: Emethylsites.dat</p>

<p style="margin-top: 1em">Required section <br>
-format list <br>
Default value: 2</p>

<p style="margin-top: 1em">-things list <br>
Specify a list of one or more code characters in the order
in which you wish things to be displayed one above the other
down the page. For example if you wish to see things <br>
displayed in the order: sequence, complement sequence, ticks
line, frame 1 translation, blank line; then you should enter
&rsquo;S,C,T,1,B&rsquo;. Default value: B,N,T,S,A,F</p>

<p style="margin-top: 1em">Additional section <br>
-translate range <br>
Regions to translate (if translating). If this is left blank
the complete sequence is translated. A set of regions is
specified by a set of pairs of positions. The positions <br>
are integers. They are separated by any non-digit, non-alpha
character. Examples of region specifications are: 24-45,
56-78 1:45, 67=99;765..888</p>

<p style="margin-top: 1em">-revtranslate range <br>
Regions to translate (if translating). If this is left blank
the complete sequence is translated. A set of regions is
specified by a set of pairs of positions. The positions <br>
are integers. They are separated by any non-digit, non-alpha
character. Examples of region specifications are: 78-56,
45-24, 888..765, 99=67; 45:1</p>

<p style="margin-top: 1em">-uppercase range <br>
Regions to put in uppercase. If this is left blank, then the
sequence case is left alone. A set of regions is specified
by a set of pairs of positions. The positions are <br>
integers. They are separated by any non-digit, non-alpha
character. Examples of region specifications are: 24-45,
56-78 1:45, 67=99;765..888 1,5,8,10,23,45,57,99</p>

<p style="margin-top: 1em">-highlight range <br>
Regions to colour if formatting for HTML. If this is left
blank, then the sequence is left alone. A set of regions is
specified by a set of pairs of positions. The positions <br>
are integers. They are followed by any valid HTML font
colour. Examples of region specifications are: 24-45 blue
56-78 orange 1-100 green 120-156 red A file of ranges to
<br>
colour (one range per line) can be specified as
&rsquo;@filename&rsquo;.</p>

<p style="margin-top: 1em">-annotation range <br>
Regions to annotate by marking. If this is left blank, then
no annotation is added. A set of regions is specified by a
set of pairs of positions followed by optional text. <br>
The positions are integers. They are followed by any text
(but not digits when on the command-line). Examples of
region specifications are: 24-45 new domain 56-78 match to
<br>
Mouse 1-100 First part 120-156 oligo A file of ranges to
annotate (one range per line) can be specified as
&rsquo;@filename&rsquo;.</p>

<p style="margin-top: 1em">-enzymes string <br>
The name &rsquo;all&rsquo; reads in all enzyme names from
the REBASE database. You can specify enzymes by giving their
names with commas between then, such as: <br>
&rsquo;HincII,hinfI,ppiI,hindiii&rsquo;. The case of the
names is not important. You can specify a file of enzyme
names to read in by giving the name of the file holding the
enzyme names <br>
with a &rsquo;@&rsquo; character in front of it, for
example, &rsquo;@enz.list&rsquo;. Blank lines and lines
starting with a hash character or &rsquo;!&rsquo; are
ignored and all other lines are concatenated <br>
together with a comma character &rsquo;,&rsquo; and then
treated as the list of enzymes to search for. An example of
a file of enzyme names is: ! my enzymes HincII, ppiII !
other enzymes <br>
hindiii HinfI PpiI Default value: all</p>

<p style="margin-top: 1em">-table list</p>

<p style="margin-top: 1em">Feature display options <br>
-sourcematch string <br>
By default any feature source in the feature table is shown.
You can set this to match any feature source you wish to
show. The source name is usually either the name of the <br>
program that detected the feature or it is the feature table
(eg: EMBL) that the feature came from. The source may be
wildcarded by using &rsquo;*&rsquo;. If you wish to show
more than <br>
one source, separate their names with the character
&rsquo;|&rsquo;, eg: gene* | embl Default value: *</p>

<p style="margin-top: 1em">-typematch string <br>
By default any feature type in the feature table is shown.
You can set this to match any feature type you wish to show.
See http://www.ebi.ac.uk/embl/WebFeat/ for a list of <br>
the EMBL feature types and see Appendix A of the Swissprot
user manual in http://www.expasy.org/sprot/userman.html for
a list of the Swissprot feature types. The type may be <br>
wildcarded by using &rsquo;*&rsquo;. If you wish to show
more than one type, separate their names with the character
&rsquo;|&rsquo;, eg: *UTR | intron Default value: *</p>

<p style="margin-top: 1em">-sensematch integer <br>
By default any feature type in the feature table is shown.
You can set this to match any feature sense you wish to
show. 0 - any sense, 1 - forward sense, -1 - reverse
sense</p>

<p style="margin-top: 1em">-minscore float <br>
Minimum score of feature to display (see also maxscore)
Default value: 0.0</p>

<p style="margin-top: 1em">-maxscore float <br>
Maximum score of feature to display. If both minscore and
maxscore are zero (the default), then any score is ignored
Default value: 0.0</p>

<p style="margin-top: 1em">-tagmatch string <br>
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, <br>
&rsquo;/db_xref&rsquo;, &rsquo;/EC_number&rsquo;,
&rsquo;/evidence&rsquo;, &rsquo;/exception&rsquo;,
&rsquo;/function&rsquo;, &rsquo;/gene&rsquo;,
&rsquo;/label&rsquo;, &rsquo;/map&rsquo;,
&rsquo;/note&rsquo;, &rsquo;/number&rsquo;,
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;, <br>
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Some of these tags also have
values, for example &rsquo;/gene&rsquo; can have the value
of the gene <br>
name. By default any feature tag in the feature table is
shown. You can set this to match any feature tag you wish to
show. The tag may be wildcarded by using &rsquo;*&rsquo;. If
you <br>
wish to show more than one tag, separate their names with
the character &rsquo;|&rsquo;, eg: gene | label Default
value: *</p>

<p style="margin-top: 1em">-valuematch string <br>
Tag values are the values associated with a feature tag.
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of <br>
feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, &rsquo;/db_xref&rsquo;,
&rsquo;/EC_number&rsquo;, &rsquo;/evidence&rsquo;,
&rsquo;/exception&rsquo;, &rsquo;/function&rsquo;,
&rsquo;/gene&rsquo;, &rsquo;/label&rsquo;,
&rsquo;/map&rsquo;, &rsquo;/note&rsquo;,
&rsquo;/number&rsquo;, <br>
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;,
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Only some of these tags can have
values, <br>
for example &rsquo;/gene&rsquo; can have the value of the
gene name. By default any feature tag value in the feature
table is shown. You can set this to match any feature tag
value you <br>
wish to show. The tag value may be wildcarded by using
&rsquo;*&rsquo;. If you wish to show more than one tag
value, separate their names with the character
&rsquo;|&rsquo;, eg: pax* | 10 Default <br>
value: *</p>

<p style="margin-top: 1em">-stricttags boolean <br>
By default if any tag/value pair in a feature matches the
specified tag and value, then all the tags/value pairs of
that feature will be displayed. If this is set to be true,
<br>
then only those tag/value pairs in a feature that match the
specified tag and value will be displayed. Default value:
N</p>

<p style="margin-top: 1em">Advanced section <br>
Restriction map options <br>
-flatreformat boolean <br>
This changes the output format to one where the recognition
site is indicated by a row of &rsquo;===&rsquo; characters
and the cut site is pointed to by a &rsquo;&gt;&rsquo;
character in the forward <br>
sense, or a &rsquo;&lt;&rsquo; in the reverse sense strand.
Default value: N</p>

<p style="margin-top: 1em">-mincuts integer <br>
This sets the minimum number of cuts for any restriction
enzyme that will be considered. Any enzymes that cut fewer
times than this will be ignored. Default value: 1</p>

<p style="margin-top: 1em">-maxcuts integer <br>
This sets the maximum number of cuts for any restriction
enzyme that will be considered. Any enzymes that cut more
times than this will be ignored. Default value:
2000000000</p>

<p style="margin-top: 1em">-sitelen integer <br>
This sets the minimum length of the restriction enzyme
recognition site. Any enzymes with sites shorter than this
will be ignored. Default value: 4</p>

<p style="margin-top: 1em">-single boolean <br>
If this is set then this forces the values of the mincuts
and maxcuts qualifiers to both be 1. Any other value you may
have set them to will be ignored. Default value: N</p>

<p style="margin-top: 1em">-blunt boolean <br>
This allows those enzymes which cut at the same position on
the forward and reverse strands to be considered. Default
value: Y</p>

<p style="margin-top: 1em">-sticky boolean <br>
This allows those enzymes which cut at different positions
on the forward and reverse strands, leaving an overhang, to
be considered. Default value: Y</p>

<p style="margin-top: 1em">-ambiguity boolean <br>
This allows those enzymes which have one or more
&rsquo;N&rsquo; ambiguity codes in their pattern to be
considered Default value: Y</p>

<p style="margin-top: 1em">-plasmid boolean <br>
If this is set then this allows searches for restriction
enzyme recognition site and cut positions that span the end
of the sequence to be considered. Default value: N</p>

<p style="margin-top: 1em">-methylation boolean <br>
If this is set then RE recognition sites will not match
methylated bases. Default value: N</p>

<p style="margin-top: 1em">-commercial boolean <br>
If this is set, then only those enzymes with a commercial
supplier will be searched for. This qualifier is ignored if
you have specified an explicit list of enzymes to search
<br>
for, rather than searching through &rsquo;all&rsquo; the
enzymes in the REBASE database. It is assumed that, if you
are asking for an explicit enzyme, then you probably know
where to get <br>
it from and so all enzymes names that you have asked to be
searched for, and which cut, will be reported whether or not
they have a commercial supplier. Default value: Y</p>

<p style="margin-top: 1em">-limit boolean <br>
This limits the reporting of enzymes to just one enzyme from
each group of isoschizomers. The enzyme chosen to represent
an isoschizomer group is the prototype indicated in <br>
the data file &rsquo;embossre.equ&rsquo;, which is created
by the program &rsquo;rebaseextract&rsquo;. If you prefer
different prototypes to be used, make a copy of embossre.equ
in your home <br>
directory and edit it. If this value is set to be false then
all of the input enzymes will be reported. You might like to
set this to false if you are supplying an explicit <br>
set of enzymes rather than searching &rsquo;all&rsquo; of
them. Default value: Y</p>

<p style="margin-top: 1em">-orfminsize integer <br>
This sets the minimum size of Open Reading Frames (ORFs) to
display in the translations. All other translation regions
are masked by changing the amino acids to &rsquo;-&rsquo;
<br>
characters.</p>

<p style="margin-top: 1em">-threeletter boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-number boolean <br>
Default value: N</p>

<p style="margin-top: 1em">-width integer <br>
Default value: 60</p>

<p style="margin-top: 1em">-length integer</p>

<p style="margin-top: 1em">-margin integer <br>
Default value: 10</p>

<p style="margin-top: 1em">-name boolean <br>
Set this to be false if you do not wish to display the ID
name of the sequence Default value: Y</p>

<p style="margin-top: 1em">-description boolean <br>
Set this to be false if you do not wish to display the
description of the sequence Default value: Y</p>

<p style="margin-top: 1em">-offset integer <br>
Default value: 1</p>

<p style="margin-top: 1em">-html boolean <br>
Default value: N</p>

<p style="margin-top: 1em">Output section <br>
-outfile outfile</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
showseq is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
SHOWSEQ(1e)</p>
<hr>
</body>
</html>
