<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:37:56 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>shuffle(1) Biosquid Manual shuffle(1)</p>

<p style="margin-top: 1em">NAME <br>
shuffle - randomize the sequences in a sequence file</p>

<p style="margin-top: 1em">SYNOPSIS <br>
shuffle [options] seqfile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
shuffle reads a sequence file seqfile, randomizes each
sequence, and prints the randomized sequences in FASTA
format on standard output. The sequence names are unchanged;
this <br>
allows you to track down the source of each randomized
sequence if necessary.</p>

<p style="margin-top: 1em">The default is to simply shuffle
each input sequence, preserving monosymbol composition
exactly. To shuffle each sequence while preserving both its
monosymbol and disymbol compo&acirc; <br>
sition exactly, use the -d option.</p>

<p style="margin-top: 1em">The -0 and -1 options allow you
to generate sequences with the same Markov properties as
each input sequence. With -0, for each input sequence, 0th
order Markov statistics are <br>
collected (e.g. symbol composition), and a new sequence is
generated with the same composition. With -1, the generated
sequence has the same 1st order Markov properties as the
<br>
input sequence (e.g. the same disymbol frequencies).</p>

<p style="margin-top: 1em">Note that the default and -0, or
-d and -1, are similar; the shuffling algorithms preserve
composition exactly, while the Markov algorithms only expect
to generate a sequence of <br>
similar composition on average.</p>

<p style="margin-top: 1em">Other shuffling algorithms are
also available, as documented below in the options.</p>

<p style="margin-top: 1em">OPTIONS <br>
-0 Calculate 0th order Markov frequencies of each input
sequence (e.g. residue composition); generate output
sequence using the same 0th order Markov frequencies.</p>

<p style="margin-top: 1em">-1 Calculate 1st order Markov
frequencies for each input sequence (e.g. diresidue
composition); generate output sequence using the same 1st
order Markov frequencies. The <br>
first residue of the output sequence is always the same as
the first residue of the input sequence.</p>

<p style="margin-top: 1em">-d Shuffle the input sequence
while preserving both monosymbol and disymbol composition
exactly. Uses an algorithm published by S.F. Altschul and
B.W. Erickson, Mol. Biol. <br>
Evol. 2:526-538, 1985.</p>

<p style="margin-top: 1em">-h Print brief help; includes
version number and summary of all options, including expert
options.</p>

<p style="margin-top: 1em">-l Look only at the length of
each input sequence; generate an i.i.d. output protein
sequence of that length, using monoresidue frequencies
typical of proteins (taken from <br>
Swissprot 35).</p>

<p style="margin-top: 1em">-n &lt;n&gt; Make &lt;n&gt;
different randomizations of each input sequence in seqfile,
rather than the default of one.</p>

<p style="margin-top: 1em">-r Generate the output sequence
by reversing the input sequence. (Therefore only one
&quot;randomization&quot; per input sequence is possible, so
it&rsquo;s not worth using -n if you use <br>
reversal.)</p>

<p style="margin-top: 1em">-t &lt;n&gt; Truncate each input
sequence to a fixed length of exactly &lt;n&gt; residues. If
the input sequence is shorter than &lt;n&gt; it is discarded
(therefore the output file may contain <br>
fewer sequences than the input file). If the input sequence
is longer than &lt;n&gt; a contiguous subsequence is
randomly chosen.</p>

<p style="margin-top: 1em">-w &lt;n&gt; Regionally shuffle
each input sequence in window sizes of &lt;n&gt;, preserving
local residue composition in each window. Probably a better
shuffling algorithm for biosequences <br>
with nonstationary residue composition (e.g. composition
that is varying along the sequence, such as between
different isochores in human genome sequence).</p>

<p style="margin-top: 1em">-B (Babelfish). Autodetect and
read a sequence file format other than the default (FASTA).
Almost any common sequence file format is recognized
(including Genbank, EMBL, <br>
SWISS-PROT, PIR, and GCG unaligned sequence formats, and
Stockholm, GCG MSF, and Clustal alignment formats). See the
printed documentation for a complete list of supported <br>
formats.</p>

<p style="margin-top: 1em">EXPERT OPTIONS <br>
--informat &lt;s&gt; <br>
Specify that the sequence file is in format &lt;s&gt;,
rather than the default FASTA format. Common examples
include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF, or
<br>
PHYLIP; see the printed documentation for a complete list of
accepted format names. This option overrides the default
expected format (FASTA) and the -B Babelfish autode&acirc;
<br>
tection option.</p>

<p style="margin-top: 1em">--nodesc <br>
Do not output any sequence description in the output file,
only the sequence names.</p>

<p style="margin-top: 1em">--seed &lt;s&gt; <br>
Set the random number seed to &lt;s&gt;. If you want
reproducible results, use the same seed each time. By
default, shuffle uses a different seed each time, so does
not gener&acirc; <br>
ate the same output in subsequent runs with the same
input.</p>

<p style="margin-top: 1em">SEE ALSO <br>
afetch(1), alistat(1), compalign(1), compstruct(1),
revcomp(1), seqsplit(1), seqstat(1), sfetch(1), sindex(1),
sreformat(1), stranslate(1), weight(1).</p>

<p style="margin-top: 1em">AUTHOR <br>
Biosquid and its documentation are Copyright (C) 1992-2003
HHMI/Washington University School of Medicine Freely
distributed under the GNU General Public License (GPL) See
COPYING <br>
in the source code distribution for more details, or contact
me.</p>

<p style="margin-top: 1em">Sean Eddy <br>
HHMI/Department of Genetics <br>
Washington University School of Medicine <br>
4444 Forest Park Blvd., Box 8510 <br>
St Louis, MO 63108 USA <br>
Phone: 1-314-362-7666 <br>
FAX : 1-314-362-2157 <br>
Email: eddy@genetics.wustl.edu</p>

<p style="margin-top: 1em">Biosquid 1.9g January 2003
shuffle(1)</p>
<hr>
</body>
</html>
