<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>shuffle(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">shuffle(1)</td>
    <td class="head-vol">Biosquid Manual</td>
    <td class="head-rtitle">shuffle(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">shuffle - randomize the sequences in a sequence file</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>shuffle</b> <i>[options]</i> <i>seqfile</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>shuffle</b> reads a sequence file <i>seqfile,</i> randomizes each sequence,
  and prints the randomized sequences in FASTA format on standard output. The
  sequence names are unchanged; this allows you to track down the source of each
  randomized sequence if necessary.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The default is to simply shuffle each input sequence, preserving monosymbol
  composition exactly. To shuffle each sequence while preserving both its
  monosymbol and disymbol composition exactly, use the <i>-d</i> option.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The <i>-0</i> and <i>-1</i> options allow you to generate sequences with the
  same Markov properties as each input sequence. With <i>-0,</i> for each input
  sequence, 0th order Markov statistics are collected (e.g. symbol composition),
  and a new sequence is generated with the same composition. With <i>-1,</i> the
  generated sequence has the same 1st order Markov properties as the input
  sequence (e.g. the same disymbol frequencies).
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Note that the default and <i>-0,</i> or <i>-d</i> and <i>-1,</i> are similar;
  the shuffling algorithms preserve composition exactly, while the Markov
  algorithms only expect to generate a sequence of similar composition on
  average.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Other shuffling algorithms are also available, as documented below in the
  options.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-0</b></dt>
  <dd class="It-tag">Calculate 0th order Markov frequencies of each input
      sequence (e.g. residue composition); generate output sequence using the
      same 0th order Markov frequencies.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-1</b></dt>
  <dd class="It-tag">Calculate 1st order Markov frequencies for each input
      sequence (e.g. diresidue composition); generate output sequence using the
      same 1st order Markov frequencies. The first residue of the output
      sequence is always the same as the first residue of the input sequence.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b></dt>
  <dd class="It-tag">Shuffle the input sequence while preserving both monosymbol
      and disymbol composition exactly. Uses an algorithm published by S.F.
      Altschul and B.W. Erickson, Mol. Biol. Evol. 2:526-538, 1985.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Print brief help; includes version number and summary of
      all options, including expert options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b></dt>
  <dd class="It-tag">Look only at the length of each input sequence; generate an
      i.i.d. output protein sequence of that length, using monoresidue
      frequencies typical of proteins (taken from Swissprot 35).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Make <i>&lt;n&gt;</i> different randomizations of each
      input sequence in <i>seqfile,</i> rather than the default of one.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b></dt>
  <dd class="It-tag">Generate the output sequence by reversing the input
      sequence. (Therefore only one &quot;randomization&quot; per input sequence
      is possible, so it's not worth using <i>-n</i> if you use reversal.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-t</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Truncate each input sequence to a fixed length of exactly
      <i>&lt;n&gt;</i> residues. If the input sequence is shorter than
      <i>&lt;n&gt;</i> it is discarded (therefore the output file may contain
      fewer sequences than the input file). If the input sequence is longer than
      <i>&lt;n&gt;</i> a contiguous subsequence is randomly chosen.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-w</b><i> &lt;n&gt;</i></dt>
  <dd class="It-tag">Regionally shuffle each input sequence in window sizes of
      <i>&lt;n&gt;,</i> preserving local residue composition in each window.
      Probably a better shuffling algorithm for biosequences with nonstationary
      residue composition (e.g. composition that is varying along the sequence,
      such as between different isochores in human genome sequence).
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-B</b></dt>
  <dd class="It-tag">(Babelfish). Autodetect and read a sequence file format
      other than the default (FASTA). Almost any common sequence file format is
      recognized (including Genbank, EMBL, SWISS-PROT, PIR, and GCG unaligned
      sequence formats, and Stockholm, GCG MSF, and Clustal alignment formats).
      See the printed documentation for a complete list of supported formats.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXPERT_OPTIONS"><a class="selflink" href="#EXPERT_OPTIONS">EXPERT
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Specify that the sequence file is in format
      <i>&lt;s&gt;,</i> rather than the default FASTA format. Common examples
      include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF, or PHYLIP; see
      the printed documentation for a complete list of accepted format names.
      This option overrides the default expected format (FASTA) and the
      <i>-B</i> Babelfish autodetection option.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nodesc</b></dt>
  <dd class="It-tag">Do not output any sequence description in the output file,
      only the sequence names.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--seed</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Set the random number seed to <i>&lt;s&gt;.</i> If you want
      reproducible results, use the same seed each time. By default,
      <b>shuffle</b> uses a different seed each time, so does not generate the
      same output in subsequent runs with the same input.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>afetch</b>(1), <b>alistat</b>(1), <b>compalign</b>(1), <b>compstruct</b>(1),
  <b>revcomp</b>(1), <b>seqsplit</b>(1), <b>seqstat</b>(1), <b>sfetch</b>(1),
  <b>sindex</b>(1), <b>sreformat</b>(1), <b>stranslate</b>(1), <b>weight</b>(1).
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Biosquid and its documentation are Copyright (C) 1992-2003 HHMI/Washington
  University School of Medicine Freely distributed under the GNU General Public
  License (GPL) See COPYING in the source code distribution for more details, or
  contact me.
<div style="height: 1.00em;">&#x00A0;</div>
<pre>
Sean Eddy
HHMI/Department of Genetics
Washington University School of Medicine
4444 Forest Park Blvd., Box 8510
St Louis, MO 63108 USA
Phone: 1-314-362-7666
FAX  : 1-314-362-2157
Email: eddy@genetics.wustl.edu
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2003</td>
    <td class="foot-os">Biosquid 1.9g</td>
  </tr>
</table>
</body>
</html>
