<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>SIBsim4(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">SIBsim4(1)</td>
    <td class="head-vol">User Manuals</td>
    <td class="head-rtitle">SIBsim4(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
SIBsim4 - align RNA sequences with a DNA sequence, allowing for introns
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>SIBsim4 [</b> <i>options</i> <b>]</b> <i>dna rna_db</i>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>SIBsim4</b> is a similarity-based tool for aligning a collection of expressed
  sequences (EST, mRNA) with a genomic DNA sequence.
<div style="height: 1.00em;">&#x00A0;</div>
Launching <b>SIBsim4</b> without any arguments will print the options list,
  along with their default values.
<div style="height: 1.00em;">&#x00A0;</div>
<b>SIBsim4</b> employs a blast-based technique to first determine the basic
  matching blocks representing the &quot;exon cores&quot;. In this first stage,
  it detects all possible exact matches of W-mers (i.e., DNA words of size W)
  between the two sequences and extends them to maximal scoring gap-free
  segments. In the second stage, the exon cores are extended into the adjacent
  as-yet-unmatched fragments using greedy alignment algorithms, and heuristics
  are used to favor configurations that conform to the splice-site recognition
  signals (e.g., GT-AG). If necessary, the process is repeated with less
  stringent parameters on the unmatched fragments.
<div style="height: 1.00em;">&#x00A0;</div>
By default, <b>SIBsim4</b> searches both strands and reports the best matches,
  measured by the number of matching nucleotides found in the alignment. The
  <b>R</b> command line option can be used to restrict the search to one
  orientation (strand) only.
<div style="height: 1.00em;">&#x00A0;</div>
Currently, four major alignment display options are supported, controlled by the
  <b>A</b> option. By default, only the endpoints, overall similarity, and
  orientation of the introns are reported. An arrow sign ('-&gt;' or '&lt;-')
  indicates the orientation of the intron. The sign `==' marks the absence from
  the alignment of a cDNA fragment starting at that position.
<div style="height: 1.00em;">&#x00A0;</div>
In the description below, the term <b>MSP</b> denotes a maximal scoring pair,
  that is, a pair of highly similar fragments in the two sequences, obtained
  during the blast-like procedure by extending a W-mer hit by matches and
  perhaps a few mismatches.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">-A &lt;int&gt;</dt>
  <dd class="It-tag">output format
    <br/>
     0: exon endpoints only
    <br/>
     1: alignment text
    <br/>
     3: both exon endpoints and alignment text
    <br/>
     4: both exon endpoints and alignment text with polyA info
    <div style="height: 1.00em;">&#x00A0;</div>
    Note that 2 is unimplemented.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 0.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-C &lt;int&gt;</dt>
  <dd class="It-tag">MSP score threshold for the second pass.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 12.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-c &lt;int&gt;</dt>
  <dd class="It-tag">minimum score cutoff value. Alignments which have scores
      below this value are not reported.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 50.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-E &lt;int&gt;</dt>
  <dd class="It-tag">cutoff value.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 3.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-f &lt;int&gt;</dt>
  <dd class="It-tag">score filter in percent. When multiple hits are detected
      for the same RNA element, only those having a score within this percentage
      of the maximal score for that RNA element are reported. Setting this value
      to 0 disables filtering and all hits will be reported, provided their
      score is above the cutoff value specified through the <b>c</b> option.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 75.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-g &lt;int&gt;</dt>
  <dd class="It-tag">join exons when gap on genomic and RNA have lengths which
      differ at most by this percentage.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 10.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-H &lt;int&gt;</dt>
  <dd class="It-tag">report chimeric transcripts when the best score is lower
      than this percentage of the overall RNA coverage and the chimera score is
      greater than this percentage of the RNA length (0 disables this report)
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 75.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-I &lt;int&gt;</dt>
  <dd class="It-tag">window width in which to search for intron splicing.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 6.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-K &lt;int&gt;</dt>
  <dd class="It-tag">MSP score threshold for the first pass.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 16.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-L &lt;str&gt;</dt>
  <dd class="It-tag">a comma separated list of forward splice-types.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is &quot;GTAG,GCAG,GTAC,ATAC&quot;.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-M &lt;int&gt;</dt>
  <dd class="It-tag">scoring splice sites, evaluate match within M nucleotides.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 10.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-o &lt;int&gt;</dt>
  <dd class="It-tag">when printing results, offset nt positions in dna sequence
      by this amount.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 0.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-q &lt;int&gt;</dt>
  <dd class="It-tag">penalty for a nucleotide mismatch.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is -5.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-R &lt;int&gt;</dt>
  <dd class="It-tag">direction of search
    <br/>
     0: search the '+' (direct) strand only
    <br/>
     1: search the '-' strand only
    <br/>
     2: search both strands
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 2.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-r &lt;int&gt;</dt>
  <dd class="It-tag">reward for a nucleotide match.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 1.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-S &lt;int&gt;</dt>
  <dd class="It-tag">splice site indels search breadth. While determining the
      best position of a splice site, <b>SIBsim4</b> will evaluate adding at
      most this number of insertions and deletions on the DNA strand on each
      side of the splice junction.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 2.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-s &lt;int&gt;</dt>
  <dd class="It-tag">split score in percent. While linking MSP, if two
      consecutive group of exons appear like they could be part of two different
      copies of the same gene, they will be tested to see if the score of each
      individual group relative to the best overall score is greater than this
      value. If both groups have a relative score above this threshold they will
      be split.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 75.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-W &lt;int&gt;</dt>
  <dd class="It-tag">word size.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 12.</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag">-X &lt;int&gt;</dt>
  <dd class="It-tag">value for terminating word extensions.
    <div style="height: 1.00em;">&#x00A0;</div>
    Default value is 12.</dd>
</dl>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">April 2007</td>
    <td class="foot-os">Bioinformatics</td>
  </tr>
</table>
</body>
</html>
