<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:37:56 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SIBsim4(1) User Manuals SIBsim4(1)</p>

<p style="margin-top: 1em">NAME <br>
SIBsim4 - align RNA sequences with a DNA sequence, allowing
for introns</p>

<p style="margin-top: 1em">SYNOPSIS <br>
SIBsim4 [ options ] dna rna_db</p>

<p style="margin-top: 1em">DESCRIPTION <br>
SIBsim4 is a similarity-based tool for aligning a collection
of expressed sequences (EST, mRNA) with a genomic DNA
sequence.</p>

<p style="margin-top: 1em">Launching SIBsim4 without any
arguments will print the options list, along with their
default values.</p>

<p style="margin-top: 1em">SIBsim4 employs a blast-based
technique to first determine the basic matching blocks
representing the &quot;exon cores&quot;. In this first
stage, it detects all possible exact matches of <br>
W-mers (i.e., DNA words of size W) between the two sequences
and extends them to maximal scoring gap-free segments. In
the second stage, the exon cores are extended into the <br>
adjacent as-yet-unmatched fragments using greedy alignment
algorithms, and heuristics are used to favor configurations
that conform to the splice-site recognition signals (e.g.,
<br>
GT-AG). If necessary, the process is repeated with less
stringent parameters on the unmatched fragments.</p>

<p style="margin-top: 1em">By default, SIBsim4 searches
both strands and reports the best matches, measured by the
number of matching nucleotides found in the alignment. The R
command line option can be <br>
used to restrict the search to one orientation (strand)
only.</p>

<p style="margin-top: 1em">Currently, four major alignment
display options are supported, controlled by the A option.
By default, only the endpoints, overall similarity, and
orientation of the introns are <br>
reported. An arrow sign (&rsquo;-&gt;&rsquo; or
&rsquo;&lt;-&rsquo;) indicates the orientation of the
intron. The sign &lsquo;==&rsquo; marks the absence from the
alignment of a cDNA fragment starting at that position.</p>

<p style="margin-top: 1em">In the description below, the
term MSP denotes a maximal scoring pair, that is, a pair of
highly similar fragments in the two sequences, obtained
during the blast-like procedure <br>
by extending a W-mer hit by matches and perhaps a few
mismatches.</p>

<p style="margin-top: 1em">OPTIONS <br>
-A &lt;int&gt; <br>
output format <br>
0: exon endpoints only <br>
1: alignment text <br>
3: both exon endpoints and alignment text <br>
4: both exon endpoints and alignment text with polyA
info</p>

<p style="margin-top: 1em">Note that 2 is
unimplemented.</p>

<p style="margin-top: 1em">Default value is 0.</p>

<p style="margin-top: 1em">-C &lt;int&gt; <br>
MSP score threshold for the second pass.</p>

<p style="margin-top: 1em">Default value is 12.</p>

<p style="margin-top: 1em">-c &lt;int&gt; <br>
minimum score cutoff value. Alignments which have scores
below this value are not reported.</p>

<p style="margin-top: 1em">Default value is 50.</p>

<p style="margin-top: 1em">-E &lt;int&gt; <br>
cutoff value.</p>

<p style="margin-top: 1em">Default value is 3.</p>

<p style="margin-top: 1em">-f &lt;int&gt; <br>
score filter in percent. When multiple hits are detected for
the same RNA element, only those having a score within this
percentage of the maximal score for that RNA ele&acirc; <br>
ment are reported. Setting this value to 0 disables
filtering and all hits will be reported, provided their
score is above the cutoff value specified through the c <br>
option.</p>

<p style="margin-top: 1em">Default value is 75.</p>

<p style="margin-top: 1em">-g &lt;int&gt; <br>
join exons when gap on genomic and RNA have lengths which
differ at most by this percentage.</p>

<p style="margin-top: 1em">Default value is 10.</p>

<p style="margin-top: 1em">-H &lt;int&gt; <br>
report chimeric transcripts when the best score is lower
than this percentage of the overall RNA coverage and the
chimera score is greater than this percentage of the RNA
<br>
length (0 disables this report)</p>

<p style="margin-top: 1em">Default value is 75.</p>

<p style="margin-top: 1em">-I &lt;int&gt; <br>
window width in which to search for intron splicing.</p>

<p style="margin-top: 1em">Default value is 6.</p>

<p style="margin-top: 1em">-K &lt;int&gt; <br>
MSP score threshold for the first pass.</p>

<p style="margin-top: 1em">Default value is 16.</p>

<p style="margin-top: 1em">-L &lt;str&gt; <br>
a comma separated list of forward splice-types.</p>

<p style="margin-top: 1em">Default value is
&quot;GTAG,GCAG,GTAC,ATAC&quot;.</p>

<p style="margin-top: 1em">-M &lt;int&gt; <br>
scoring splice sites, evaluate match within M
nucleotides.</p>

<p style="margin-top: 1em">Default value is 10.</p>

<p style="margin-top: 1em">-o &lt;int&gt; <br>
when printing results, offset nt positions in dna sequence
by this amount.</p>

<p style="margin-top: 1em">Default value is 0.</p>

<p style="margin-top: 1em">-q &lt;int&gt; <br>
penalty for a nucleotide mismatch.</p>

<p style="margin-top: 1em">Default value is -5.</p>

<p style="margin-top: 1em">-R &lt;int&gt; <br>
direction of search <br>
0: search the &rsquo;+&rsquo; (direct) strand only <br>
1: search the &rsquo;-&rsquo; strand only <br>
2: search both strands</p>

<p style="margin-top: 1em">Default value is 2.</p>

<p style="margin-top: 1em">-r &lt;int&gt; <br>
reward for a nucleotide match.</p>

<p style="margin-top: 1em">Default value is 1.</p>

<p style="margin-top: 1em">-S &lt;int&gt; <br>
splice site indels search breadth. While determining the
best position of a splice site, SIBsim4 will evaluate adding
at most this number of insertions and deletions on <br>
the DNA strand on each side of the splice junction.</p>

<p style="margin-top: 1em">Default value is 2.</p>

<p style="margin-top: 1em">-s &lt;int&gt; <br>
split score in percent. While linking MSP, if two
consecutive group of exons appear like they could be part of
two different copies of the same gene, they will be tested
<br>
to see if the score of each individual group relative to the
best overall score is greater than this value. If both
groups have a relative score above this threshold they <br>
will be split.</p>

<p style="margin-top: 1em">Default value is 75.</p>

<p style="margin-top: 1em">-W &lt;int&gt; <br>
word size.</p>

<p style="margin-top: 1em">Default value is 12.</p>

<p style="margin-top: 1em">-X &lt;int&gt; <br>
value for terminating word extensions.</p>

<p style="margin-top: 1em">Default value is 12.</p>

<p style="margin-top: 1em">Bioinformatics April 2007
SIBsim4(1)</p>
<hr>
</body>
</html>
