<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>SKIPREDUNDANT(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">SKIPREDUNDANT(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">SKIPREDUNDANT(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
skipredundant - Remove redundant sequences from an input set
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 14.00ex; text-indent: -14.00ex;"><b>skipredundant</b>
  <b>-feature&#x00A0;</b><b></b><i>toggle</i>
  <b>-sequences&#x00A0;</b><b></b><i>seqset</i>
  [<b>-datafile&#x00A0;</b><b></b><i>matrixf</i>]
  <b>-mode&#x00A0;</b><b></b><i>list</i>
  <b>-threshold&#x00A0;</b><b></b><i>float</i>
  <b>-minthreshold&#x00A0;</b><b></b> <i>float</i>
  <b>-maxthreshold&#x00A0;</b><b></b><i>float</i>
  <b>-gapopen&#x00A0;</b><b></b><i>float</i>
  <b>-gapextend&#x00A0;</b><b></b><i>float</i>
  <b>-outseq&#x00A0;</b><b></b><i>seqoutall</i>
  <b>-redundantoutseq&#x00A0;</b><b></b><i>seqoutall</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 14.00ex; text-indent: -14.00ex;"><b>skipredundant</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>skipredundant</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Edit&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-feature</b> <i>toggle</i>
<div style="margin-left: 4.00ex;">Sequence feature information will be retained
  if this option is set.</div>
<div class="Pp"></div>
<b>-sequences</b> <i>seqset</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-datafile</b> <i>matrixf</i>
<div style="margin-left: 4.00ex;">This is the scoring matrix file used when
  comparing sequences. By default it is the file 'EBLOSUM62' (for proteins) or
  the file 'EDNAFULL' (for nucleic sequences). These files are found in the
  'data' directory of the EMBOSS installation.</div>
<h2 class="Ss" title="Ss" id="Required_section"><a class="selflink" href="#Required_section">Required
  section</a></h2>
<b>-mode</b> <i>list</i>
<div style="margin-left: 4.00ex;">This option specifies whether to remove
  redundancy at a single threshold percentage sequence similarity or remove
  redundancy outside a range of acceptable threshold percentage similarity. All
  permutations of pair-wise sequence alignments are calculated for each set of
  input sequences in turn using the EMBOSS implementation of the Needleman and
  Wunsch global alignment algorithm. Redundant sequences are removed in one of
  two modes as follows: (i) If a pair of proteins achieve greater than a
  threshold percentage sequence similarity (specified by the user) the shortest
  sequence is discarded. (ii) If a pair of proteins have a percentage sequence
  similarity that lies outside an acceptable range (specified by the user) the
  shortest sequence is discarded. Default value: 1</div>
<div class="Pp"></div>
<b>-threshold</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the percentage sequence
  identity redundancy threshold. The percentage sequence identity redundancy
  threshold determines the redundancy calculation. If a pair of proteins achieve
  greater than this threshold the shortest sequence is discarded. Default value:
  95.0</div>
<div class="Pp"></div>
<b>-minthreshold</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the percentage sequence
  identity redundancy threshold (lower limit). The percentage sequence identity
  redundancy threshold determines the redundancy calculation. If a pair of
  proteins have a percentage sequence similarity that lies outside an acceptable
  range the shortest sequence is discarded. Default value: 30.0</div>
<div class="Pp"></div>
<b>-maxthreshold</b> <i>float</i>
<div style="margin-left: 4.00ex;">This option specifies the percentage sequence
  identity redundancy threshold (upper limit). The percentage sequence identity
  redundancy threshold determines the redundancy calculation. If a pair of
  proteins have a percentage sequence similarity that lies outside an acceptable
  range the shortest sequence is discarded. Default value: 90.0</div>
<div class="Pp"></div>
<b>-gapopen</b> <i>float</i>
<div style="margin-left: 4.00ex;">The gap open penalty is the score taken away
  when a gap is created. The best value depends on the choice of comparison
  matrix. The default value assumes you are using the EBLOSUM62 matrix for
  protein sequences, and the EDNAFULL matrix for nucleotide sequences. Default
  value: @($(acdprotein)? 10.0 : 10.0 )</div>
<div class="Pp"></div>
<b>-gapextend</b> <i>float</i>
<div style="margin-left: 4.00ex;">The gap extension, penalty is added to the
  standard gap penalty for each base or residue in the gap. This is how long
  gaps are penalized. Usually you will expect a few long gaps rather than many
  short gaps, so the gap extension penalty should be lower than the gap penalty.
  An exception is where one or both sequences are single reads with possible
  sequencing errors in which case you would expect many single base gaps. You
  can get this result by setting the gap open penalty to zero (or very low) and
  using the gap extension penalty to control gap scoring. Default value:
  @($(acdprotein)? 0.5 : 0.5 )</div>
<h2 class="Ss" title="Ss" id="Advanced_section"><a class="selflink" href="#Advanced_section">Advanced
  section</a></h2>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outseq</b> <i>seqoutall</i>
<div style="margin-left: 4.00ex;"></div>
<div class="Pp"></div>
<b>-redundantoutseq</b> <i>seqoutall</i>
<div style="margin-left: 4.00ex;"></div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
skipredundant is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">05/11/2012</td>
    <td class="foot-os">EMBOSS 6.4.0</td>
  </tr>
</table>
</body>
</html>
