<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:38:05 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SKIPREDUNDANT(1e) EMBOSS Manual for Debian
SKIPREDUNDANT(1e)</p>

<p style="margin-top: 1em">NAME <br>
skipredundant - Remove redundant sequences from an input
set</p>

<p style="margin-top: 1em">SYNOPSIS <br>
skipredundant -feature toggle -sequences seqset [-datafile
matrixf] -mode list -threshold float -minthreshold float
-maxthreshold float -gapopen float -gapextend float <br>
-outseq seqoutall -redundantoutseq seqoutall</p>

<p style="margin-top: 1em">skipredundant -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
skipredundant is a command line program from EMBOSS
(&acirc;the European Molecular Biology Open Software
Suite&acirc;). It is part of the &quot;Edit&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-feature toggle <br>
Sequence feature information will be retained if this option
is set.</p>

<p style="margin-top: 1em">-sequences seqset</p>

<p style="margin-top: 1em">-datafile matrixf <br>
This is the scoring matrix file used when comparing
sequences. By default it is the file &rsquo;EBLOSUM62&rsquo;
(for proteins) or the file &rsquo;EDNAFULL&rsquo; (for
nucleic sequences). These <br>
files are found in the &rsquo;data&rsquo; directory of the
EMBOSS installation.</p>

<p style="margin-top: 1em">Required section <br>
-mode list <br>
This option specifies whether to remove redundancy at a
single threshold percentage sequence similarity or remove
redundancy outside a range of acceptable threshold <br>
percentage similarity. All permutations of pair-wise
sequence alignments are calculated for each set of input
sequences in turn using the EMBOSS implementation of the
<br>
Needleman and Wunsch global alignment algorithm. Redundant
sequences are removed in one of two modes as follows: (i) If
a pair of proteins achieve greater than a threshold <br>
percentage sequence similarity (specified by the user) the
shortest sequence is discarded. (ii) If a pair of proteins
have a percentage sequence similarity that lies outside <br>
an acceptable range (specified by the user) the shortest
sequence is discarded. Default value: 1</p>

<p style="margin-top: 1em">-threshold float <br>
This option specifies the percentage sequence identity
redundancy threshold. The percentage sequence identity
redundancy threshold determines the redundancy calculation.
If a <br>
pair of proteins achieve greater than this threshold the
shortest sequence is discarded. Default value: 95.0</p>

<p style="margin-top: 1em">-minthreshold float <br>
This option specifies the percentage sequence identity
redundancy threshold (lower limit). The percentage sequence
identity redundancy threshold determines the redundancy <br>
calculation. If a pair of proteins have a percentage
sequence similarity that lies outside an acceptable range
the shortest sequence is discarded. Default value: 30.0</p>

<p style="margin-top: 1em">-maxthreshold float <br>
This option specifies the percentage sequence identity
redundancy threshold (upper limit). The percentage sequence
identity redundancy threshold determines the redundancy <br>
calculation. If a pair of proteins have a percentage
sequence similarity that lies outside an acceptable range
the shortest sequence is discarded. Default value: 90.0</p>

<p style="margin-top: 1em">-gapopen float <br>
The gap open penalty is the score taken away when a gap is
created. The best value depends on the choice of comparison
matrix. The default value assumes you are using the <br>
EBLOSUM62 matrix for protein sequences, and the EDNAFULL
matrix for nucleotide sequences. Default value:
@($(acdprotein)? 10.0 : 10.0 )</p>

<p style="margin-top: 1em">-gapextend float <br>
The gap extension, penalty is added to the standard gap
penalty for each base or residue in the gap. This is how
long gaps are penalized. Usually you will expect a few long
<br>
gaps rather than many short gaps, so the gap extension
penalty should be lower than the gap penalty. An exception
is where one or both sequences are single reads with <br>
possible sequencing errors in which case you would expect
many single base gaps. You can get this result by setting
the gap open penalty to zero (or very low) and using the
<br>
gap extension penalty to control gap scoring. Default value:
@($(acdprotein)? 0.5 : 0.5 )</p>

<p style="margin-top: 1em">Advanced section <br>
Output section <br>
-outseq seqoutall</p>

<p style="margin-top: 1em">-redundantoutseq seqoutall</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
skipredundant is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
SKIPREDUNDANT(1e)</p>
<hr>
</body>
</html>
