<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>SMITHWATERMAN(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">SMITHWATERMAN(1)</td>
    <td class="head-vol">User Commands</td>
    <td class="head-rtitle">SMITHWATERMAN(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
smithwaterman - determine similar regions between two strings or nucleotide or
  protein sequences
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>smithwaterman</b> [options] &lt;reference sequence&gt; &lt;query sequence&gt;
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
The Smith&#x2013;Waterman algorithm performs local sequence alignment; that is,
  for determining similar regions between two strings or nucleotide or protein
  sequences. Instead of looking at the total sequence, the Smith&#x2013;Waterman
  algorithm compares segments of all possible lengths and optimizes the
  similarity measure.
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-m</b>, <b>--match-score</b></dt>
  <dd class="It-tag">the match score (default 10.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b>, <b>--mismatch-score</b></dt>
  <dd class="It-tag">the mismatch score (default <b>-9</b>.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-g</b>, <b>--gap-open-penalty</b></dt>
  <dd class="It-tag">the gap open penalty (default 15.0)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-z</b>, <b>--entropy-gap-open-penalty</b></dt>
  <dd class="It-tag">enable entropy scaling of the gap open penalty</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-e</b>, <b>--gap-extend-penalty</b></dt>
  <dd class="It-tag">the gap extend penalty (default 6.66)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b>, <b>--repeat-gap-extend-penalty</b></dt>
  <dd class="It-tag">use repeat information when generating gap extension
      penalties</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-b</b>, <b>--bandwidth</b></dt>
  <dd class="It-tag">bandwidth to use (default 0, or non-banded algorithm)</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-p</b>, <b>--print-alignment</b></dt>
  <dd class="It-tag">print out the alignment</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-R</b>, <b>--reverse-complement</b></dt>
  <dd class="It-tag">report the reverse-complement alignment if it scores
    better</dd>
</dl>
<div class="Pp"></div>
When called with literal reference and query sequences, smithwaterman prints the
  cigar match positional string and the match position for the query sequence
  against the reference sequence.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
This manpage was written by Andreas Tille for the Debian distribution and can be
  used for any other usage of the program.</div>
<table class="foot">
  <tr>
    <td class="foot-date">June 2016</td>
    <td class="foot-os">smithwaterman 0.0+20151117</td>
  </tr>
</table>
</body>
</html>
