<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:38:11 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>SMITHWATERMAN(1) User Commands SMITHWATERMAN(1)</p>

<p style="margin-top: 1em">NAME <br>
smithwaterman - determine similar regions between two
strings or nucleotide or protein sequences</p>

<p style="margin-top: 1em">SYNOPSIS <br>
smithwaterman [options] &lt;reference sequence&gt; &lt;query
sequence&gt;</p>

<p style="margin-top: 1em">DESCRIPTION <br>
The Smith&acirc;Waterman algorithm performs local sequence
alignment; that is, for determining similar regions between
two strings or nucleotide or protein sequences. Instead of
look&acirc; <br>
ing at the total sequence, the Smith&acirc;Waterman
algorithm compares segments of all possible lengths and
optimizes the similarity measure.</p>

<p style="margin-top: 1em">OPTIONS <br>
-m, --match-score <br>
the match score (default 10.0)</p>

<p style="margin-top: 1em">-n, --mismatch-score <br>
the mismatch score (default -9.0)</p>

<p style="margin-top: 1em">-g, --gap-open-penalty <br>
the gap open penalty (default 15.0)</p>

<p style="margin-top: 1em">-z, --entropy-gap-open-penalty
<br>
enable entropy scaling of the gap open penalty</p>

<p style="margin-top: 1em">-e, --gap-extend-penalty <br>
the gap extend penalty (default 6.66)</p>

<p style="margin-top: 1em">-r, --repeat-gap-extend-penalty
<br>
use repeat information when generating gap extension
penalties</p>

<p style="margin-top: 1em">-b, --bandwidth <br>
bandwidth to use (default 0, or non-banded algorithm)</p>

<p style="margin-top: 1em">-p, --print-alignment <br>
print out the alignment</p>

<p style="margin-top: 1em">-R, --reverse-complement <br>
report the reverse-complement alignment if it scores
better</p>

<p style="margin-top: 1em">When called with literal
reference and query sequences, smithwaterman prints the
cigar match positional string and the match position for the
query sequence against the reference <br>
sequence.</p>

<p style="margin-top: 1em">AUTHOR <br>
This manpage was written by Andreas Tille for the Debian
distribution and can be used for any other usage of the
program.</p>

<p style="margin-top: 1em">smithwaterman 0.0+20151117 June
2016 SMITHWATERMAN(1)</p>
<hr>
</body>
</html>
