<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:38:58 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>sreformat(1) Biosquid Manual sreformat(1)</p>

<p style="margin-top: 1em">NAME <br>
sreformat - convert sequence file to different format</p>

<p style="margin-top: 1em">SYNOPSIS <br>
sreformat [options] format seqfile</p>

<p style="margin-top: 1em">DESCRIPTION <br>
sreformat reads the sequence file seqfile in any supported
format, reformats it into a new format specified by format,
then prints the reformatted text.</p>

<p style="margin-top: 1em">Supported input formats include
(but are not limited to) the unaligned formats FASTA,
Genbank, EMBL, SWISS-PROT, PIR, and GCG, and the aligned
formats Stockholm, Clustal, GCG <br>
MSF, and Phylip.</p>

<p style="margin-top: 1em">Available unaligned output file
format codes include fasta (FASTA format); embl
(EMBL/SWISSPROT format); genbank (Genbank format); gcg (GCG
single sequence format); gcgdata (GCG <br>
flatfile database format); pir (PIR/CODATA flatfile format);
raw (raw sequence, no other information).</p>

<p style="margin-top: 1em">The available aligned output
file format codes include stockholm (PFAM/Stockholm format);
msf (GCG MSF format); a2m (an aligned FASTA format); PHYLIP
(Felsenstein&rsquo;s PHYLIP for&acirc; <br>
mat); and clustal (Clustal V/W/X format); and selex (the old
SELEX/HMMER/Pfam annotated alignment format);</p>

<p style="margin-top: 1em">All thee codes are interpreted
case-insensitively (e.g. MSF, Msf, or msf all work).</p>

<p style="margin-top: 1em">Unaligned format files cannot be
reformatted to aligned formats. However, aligned formats can
be reformatted to unaligned formats -- gap characters are
simply stripped out.</p>

<p style="margin-top: 1em">This program was originally
named reformat, but that name clashes with a GCG program of
the same name.</p>

<p style="margin-top: 1em">OPTIONS <br>
-d DNA; convert U&rsquo;s to T&rsquo;s, to make sure a
nucleic acid sequence is shown as DNA not RNA. See -r.</p>

<p style="margin-top: 1em">-h Print brief help; includes
version number and summary of all options, including expert
options.</p>

<p style="margin-top: 1em">-l Lowercase; convert all
sequence residues to lower case. See -u.</p>

<p style="margin-top: 1em">-n For DNA/RNA sequences,
converts any character that&rsquo;s not unambiguous RNA/DNA
(e.g. ACGTU/acgtu) to an N. Used to convert IUPAC ambiguity
codes to N&rsquo;s, for software that <br>
can&rsquo;t handle all IUPAC codes (some public RNA folding
codes, for example). If the file is an alignment, gap
characters are also left unchanged. If sequences are not
<br>
nucleic acid sequences, this option will corrupt the data in
a predictable fashion.</p>

<p style="margin-top: 1em">-r RNA; convert T&rsquo;s to
U&rsquo;s, to make sure a nucleic acid sequence is shown as
RNA not DNA. See -d.</p>

<p style="margin-top: 1em">-u Uppercase; convert all
sequence residues to upper case. See -l.</p>

<p style="margin-top: 1em">-x For DNA sequences, convert
non-IUPAC characters (such as X&rsquo;s) to N&rsquo;s. This
is for compatibility with benighted people who insist on
using X instead of the IUPAC ambiguity <br>
character N. (X is for ambiguity in an amino acid
residue).</p>

<p style="margin-top: 1em">Warning: like the -n option, the
code doesn&rsquo;t check that you are actually giving it
DNA. It simply literally just converts non-IUPAC DNA symbols
to N. So if you acciden&acirc; <br>
tally give it protein sequence, it will happily convert most
every amino acid residue to an N.</p>

<p style="margin-top: 1em">EXPERT OPTIONS <br>
--gapsym &lt;c&gt; <br>
Convert all gap characters to &lt;c&gt;. Used to prepare
alignment files for programs with strict requirements for
gap symbols. Only makes sense if the input seqfile is an
<br>
alignment.</p>

<p style="margin-top: 1em">--informat &lt;s&gt; <br>
Specify that the sequence file is in format &lt;s&gt;,
rather than allowing the program to autodetect the file
format. Common examples include Genbank, EMBL, GCG, PIR,
Stock&acirc; <br>
holm, Clustal, MSF, or PHYLIP; see the printed documentation
for a complete list of accepted format names.</p>

<p style="margin-top: 1em">--mingap <br>
If seqfile is an alignment, remove any columns that contain
100% gap characters, minimizing the overall length of the
alignment. (Often useful if you&rsquo;ve extracted a
sub&acirc; <br>
set of aligned sequences from a larger alignment.)</p>

<p style="margin-top: 1em">--nogap <br>
Remove any aligned columns that contain any gap symbols at
all. Useful as a prelude to phylogenetic analyses, where you
only want to analyze columns containing 100% <br>
residues, so you want to strip out any columns with gaps in
them. Only makes sense if the file is an alignment file.</p>

<p style="margin-top: 1em">--pfam For SELEX alignment
output format only, put the entire alignment in one block
(don&rsquo;t wrap into multiple blocks). This is close to
the format used internally by Pfam in <br>
Stockholm and Cambridge.</p>

<p style="margin-top: 1em">--sam Try to convert gap
characters to UC Santa Cruz SAM style, where a . means a gap
in an insert column, and a - means a deletion in a
consensus/match column. This only works <br>
for converting aligned file formats, and only if the
alignment already adheres to the SAM convention of upper
case for residues in consensus/match columns, and lower case
<br>
for residues in insert columns. This is true, for instance,
of all alignments produced by old versions of HMMER. (HMMER2
produces alignments that adhere to SAM&rsquo;s conven&acirc;
<br>
tions even in gap character choice.) This option was added
to allow Pfam alignments to be reformatted into something
more suitable for profile HMM construction using the <br>
UCSC SAM software.</p>

<p style="margin-top: 1em">--samfrac &lt;x&gt; <br>
Try to convert the alignment gap characters and residue
cases to UC Santa Cruz SAM style, where a . means a gap in
an insert column and a - means a deletion in a consen&acirc;
<br>
sus/match column, and upper case means match/consensus
residues and lower case means inserted resiudes. This will
only work for converting aligned file formats, but unlike
<br>
the --sam option, it will work regardless of whether the
file adheres to the upper/lower case residue convention.
Instead, any column containing more than a fraction
&lt;x&gt; <br>
of gap characters is interpreted as an insert column, and
all other columns are interpreted as match columns. This
option was added to allow Pfam alignments to be refor&acirc;
<br>
matted into something more suitable for profile HMM
construction using the UCSC SAM software.</p>

<p style="margin-top: 1em">--wussify <br>
Convert RNA secondary structure annotation strings (both
consensus and individual) from old &quot;KHS&quot; format,
&gt;&lt;, to the new WUSS notation, &lt;&gt;. If the
notation is already in <br>
WUSS format, this option will screw it up, without warning.
Only SELEX and Stockholm format files have secondary
structure markup at present.</p>

<p style="margin-top: 1em">--dewuss <br>
Convert RNA secondary structure annotation strings from the
new WUSS notation, &lt;&gt;, back to the old KHS format,
&gt;&lt;. If the annotation is already in KHS, this option
will <br>
corrupt it, without warning. Only SELEX and Stockholm format
files have secondary structure markup.</p>

<p style="margin-top: 1em">SEE ALSO <br>
afetch(1), alistat(1), compalign(1), compstruct(1),
revcomp(1), seqsplit(1), seqstat(1), sfetch(1), shuffle(1),
sindex(1), stranslate(1), weight(1).</p>

<p style="margin-top: 1em">AUTHOR <br>
Biosquid and its documentation are Copyright (C) 1992-2003
HHMI/Washington University School of Medicine Freely
distributed under the GNU General Public License (GPL) See
COPYING <br>
in the source code distribution for more details, or contact
me.</p>

<p style="margin-top: 1em">Sean Eddy <br>
HHMI/Department of Genetics <br>
Washington University School of Medicine <br>
4444 Forest Park Blvd., Box 8510 <br>
St Louis, MO 63108 USA <br>
Phone: 1-314-362-7666 <br>
FAX : 1-314-362-2157 <br>
Email: eddy@genetics.wustl.edu</p>

<p style="margin-top: 1em">Biosquid 1.9g January 2003
sreformat(1)</p>
<hr>
</body>
</html>
