<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>sreformat(1)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">sreformat(1)</td>
    <td class="head-vol">Biosquid Manual</td>
    <td class="head-rtitle">sreformat(1)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag">sreformat - convert sequence file to different format</dt>
  <dd class="It-tag">
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>sreformat</b> <i>[options]</i> <i>format</i> <i>seqfile</i>
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>sreformat</b> reads the sequence file <i>seqfile</i> in any supported format,
  reformats it into a new format specified by <i>format,</i> then prints the
  reformatted text.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Supported input formats include (but are not limited to) the unaligned formats
  FASTA, Genbank, EMBL, SWISS-PROT, PIR, and GCG, and the aligned formats
  Stockholm, Clustal, GCG MSF, and Phylip.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Available unaligned output file format codes include <i>fasta</i> (FASTA
  format); <i>embl</i> (EMBL/SWISSPROT format); <i>genbank</i> (Genbank format);
  <i>gcg</i> (GCG single sequence format); <i>gcgdata</i> (GCG flatfile database
  format); <i>pir</i> (PIR/CODATA flatfile format); <i>raw</i> (raw sequence, no
  other information).
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
The available aligned output file format codes include <i>stockholm</i>
  (PFAM/Stockholm format); <i>msf</i> (GCG MSF format); <i>a2m</i> (an aligned
  FASTA format); <i>PHYLIP</i> (Felsenstein's PHYLIP format); and <i>clustal</i>
  (Clustal V/W/X format); and <i>selex</i> (the old SELEX/HMMER/Pfam annotated
  alignment format);
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
All thee codes are interpreted case-insensitively (e.g. MSF, Msf, or msf all
  work).
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
Unaligned format files cannot be reformatted to aligned formats. However,
  aligned formats can be reformatted to unaligned formats -- gap characters are
  simply stripped out.
<div style="height: 1.00em;">&#x00A0;</div>
<div class="Pp"></div>
This program was originally named <b>reformat,</b> but that name clashes with a
  GCG program of the same name.
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-d</b></dt>
  <dd class="It-tag">DNA; convert U's to T's, to make sure a nucleic acid
      sequence is shown as DNA not RNA. See <b>-r.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-h</b></dt>
  <dd class="It-tag">Print brief help; includes version number and summary of
      all options, including expert options.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-l</b></dt>
  <dd class="It-tag">Lowercase; convert all sequence residues to lower case. See
      <b>-u.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-n</b></dt>
  <dd class="It-tag">For DNA/RNA sequences, converts any character that's not
      unambiguous RNA/DNA (e.g. ACGTU/acgtu) to an N. Used to convert IUPAC
      ambiguity codes to N's, for software that can't handle all IUPAC codes
      (some public RNA folding codes, for example). If the file is an alignment,
      gap characters are also left unchanged. If sequences are not nucleic acid
      sequences, this option will corrupt the data in a predictable fashion.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-r</b></dt>
  <dd class="It-tag">RNA; convert T's to U's, to make sure a nucleic acid
      sequence is shown as RNA not DNA. See <b>-d.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-u</b></dt>
  <dd class="It-tag">Uppercase; convert all sequence residues to upper case. See
      <b>-l.</b>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>-x</b></dt>
  <dd class="It-tag">For DNA sequences, convert non-IUPAC characters (such as
      X's) to N's. This is for compatibility with benighted people who insist on
      using X instead of the IUPAC ambiguity character N. (X is for ambiguity in
      an amino acid residue).</dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"></dt>
  <dd class="It-tag">Warning: like the <b>-n</b> option, the code doesn't check
      that you are actually giving it DNA. It simply literally just converts
      non-IUPAC DNA symbols to N. So if you accidentally give it protein
      sequence, it will happily convert most every amino acid residue to an N.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="EXPERT_OPTIONS"><a class="selflink" href="#EXPERT_OPTIONS">EXPERT
  OPTIONS</a></h1>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--gapsym</b><i> &lt;c&gt;</i></dt>
  <dd class="It-tag">Convert all gap characters to <i>&lt;c&gt;.</i> Used to
      prepare alignment files for programs with strict requirements for gap
      symbols. Only makes sense if the input <i>seqfile</i> is an alignment.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--informat</b><i> &lt;s&gt;</i></dt>
  <dd class="It-tag">Specify that the sequence file is in format
      <i>&lt;s&gt;,</i> rather than allowing the program to autodetect the file
      format. Common examples include Genbank, EMBL, GCG, PIR, Stockholm,
      Clustal, MSF, or PHYLIP; see the printed documentation for a complete list
      of accepted format names.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--mingap</b></dt>
  <dd class="It-tag">If <i>seqfile</i> is an alignment, remove any columns that
      contain 100% gap characters, minimizing the overall length of the
      alignment. (Often useful if you've extracted a subset of aligned sequences
      from a larger alignment.)
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--nogap</b></dt>
  <dd class="It-tag">Remove any aligned columns that contain any gap symbols at
      all. Useful as a prelude to phylogenetic analyses, where you only want to
      analyze columns containing 100% residues, so you want to strip out any
      columns with gaps in them. Only makes sense if the file is an alignment
      file.
    <div style="height: 1.00em;">&#x00A0;</div>
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--pfam</b></dt>
  <dd class="It-tag">For SELEX alignment output format only, put the entire
      alignment in one block (don't wrap into multiple blocks). This is close to
      the format used internally by Pfam in Stockholm and Cambridge.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--sam</b></dt>
  <dd class="It-tag">Try to convert gap characters to UC Santa Cruz SAM style,
      where a . means a gap in an insert column, and a - means a deletion in a
      consensus/match column. This only works for converting aligned file
      formats, and only if the alignment already adheres to the SAM convention
      of upper case for residues in consensus/match columns, and lower case for
      residues in insert columns. This is true, for instance, of all alignments
      produced by old versions of HMMER. (HMMER2 produces alignments that adhere
      to SAM's conventions even in gap character choice.) This option was added
      to allow Pfam alignments to be reformatted into something more suitable
      for profile HMM construction using the UCSC SAM software.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--samfrac</b><i> &lt;x&gt;</i></dt>
  <dd class="It-tag">Try to convert the alignment gap characters and residue
      cases to UC Santa Cruz SAM style, where a . means a gap in an insert
      column and a - means a deletion in a consensus/match column, and upper
      case means match/consensus residues and lower case means inserted
      resiudes. This will only work for converting aligned file formats, but
      unlike the <b>--sam</b> option, it will work regardless of whether the
      file adheres to the upper/lower case residue convention. Instead, any
      column containing more than a fraction <i>&lt;x&gt;</i> of gap characters
      is interpreted as an insert column, and all other columns are interpreted
      as match columns. This option was added to allow Pfam alignments to be
      reformatted into something more suitable for profile HMM construction
      using the UCSC SAM software.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--wussify</b></dt>
  <dd class="It-tag">Convert RNA secondary structure annotation strings (both
      consensus and individual) from old &quot;KHS&quot; format, &gt;&lt;, to
      the new WUSS notation, &lt;&gt;. If the notation is already in WUSS
      format, this option will screw it up, without warning. Only SELEX and
      Stockholm format files have secondary structure markup at present.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<dl class="Bl-tag">
  <dt class="It-tag"><b>--dewuss</b></dt>
  <dd class="It-tag">Convert RNA secondary structure annotation strings from the
      new WUSS notation, &lt;&gt;, back to the old KHS format, &gt;&lt;. If the
      annotation is already in KHS, this option will corrupt it, without
      warning. Only SELEX and Stockholm format files have secondary structure
      markup.
    <div style="height: 1.00em;">&#x00A0;</div>
  </dd>
</dl>
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
<b>afetch</b>(1), <b>alistat</b>(1), <b>compalign</b>(1), <b>compstruct</b>(1),
  <b>revcomp</b>(1), <b>seqsplit</b>(1), <b>seqstat</b>(1), <b>sfetch</b>(1),
  <b>shuffle</b>(1), <b>sindex</b>(1), <b>stranslate</b>(1), <b>weight</b>(1).
<div style="height: 1.00em;">&#x00A0;</div>
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Biosquid and its documentation are Copyright (C) 1992-2003 HHMI/Washington
  University School of Medicine Freely distributed under the GNU General Public
  License (GPL) See COPYING in the source code distribution for more details, or
  contact me.
<div style="height: 1.00em;">&#x00A0;</div>
<pre>
Sean Eddy
HHMI/Department of Genetics
Washington University School of Medicine
4444 Forest Park Blvd., Box 8510
St Louis, MO 63108 USA
Phone: 1-314-362-7666
FAX  : 1-314-362-2157
Email: eddy@genetics.wustl.edu
</pre>
<div style="height: 1.00em;">&#x00A0;</div>
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">January 2003</td>
    <td class="foot-os">Biosquid 1.9g</td>
  </tr>
</table>
</body>
</html>
