<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>SSEMATCH(1e)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">SSEMATCH(1e)</td>
    <td class="head-vol">EMBOSS Manual for Debian</td>
    <td class="head-rtitle">SSEMATCH(1e)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
ssematch - Search a DCF file for secondary structure matches.
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>ssematch</b>
  <b>-ssinfile&#x00A0;</b><b></b><i>infile</i>
  <b>-dcfinfile&#x00A0;</b><b></b><i>infile</i>
  [<b>-datafile&#x00A0;</b><b></b><i>matrixf</i>]
  <b>-maxhits&#x00A0;</b><b></b><i>integer</i>
  [<b>-rgapopen&#x00A0;</b><b></b><i>float</i>]
  [<b>-rgapextend&#x00A0;</b><b></b> <i>float</i>]
  [<b>-egapopen&#x00A0;</b><b></b> <i>float</i>]
  [<b>-egapextend&#x00A0;</b><b></b> <i>float</i>]
  <b>-outssfile&#x00A0;</b><b></b> <i>outfile</i>
  <b>-outsefile&#x00A0;</b><b></b> <i>outfile</i>
  <b>-logfile&#x00A0;</b><b></b><i>outfile</i></div>
<div class="Pp"></div>
<div class="Pp" style="margin-left: 9.00ex; text-indent: -9.00ex;"><b>ssematch</b>
  <b>-help</b></div>
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
<b>ssematch</b> is a command line program from EMBOSS (&#x201C;the European
  Molecular Biology Open Software Suite&#x201D;). It is part of the
  &quot;Utils:Database creation&quot; command group(s).
<h1 class="Sh" title="Sh" id="OPTIONS"><a class="selflink" href="#OPTIONS">OPTIONS</a></h1>
<h2 class="Ss" title="Ss" id="Input_section"><a class="selflink" href="#Input_section">Input
  section</a></h2>
<b>-ssinfile</b> <i>infile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of the file of
  secondary structure (input).</div>
<div class="Pp"></div>
<b>-dcfinfile</b> <i>infile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of the DCF file
  (domain classification file) (input). A 'domain classification file' contains
  classification and other data for domains from SCOP or CATH, in DCF format
  (EMBL-like). The files are generated by using SCOPPARSE and CATHPARSE. Domain
  sequence information can be added to the file by using DOMAINSEQS.</div>
<div class="Pp"></div>
<b>-datafile</b> <i>matrixf</i>
<div style="margin-left: 4.00ex;">This option specifies the secondary structure
  substitution matrix. Default value: SSSUB</div>
<h2 class="Ss" title="Ss" id="Required_section"><a class="selflink" href="#Required_section">Required
  section</a></h2>
<b>-maxhits</b> <i>integer</i>
<div style="margin-left: 4.00ex;">This option specifies the number of
  top-scoring matches to report. Default value: 5</div>
<h2 class="Ss" title="Ss" id="Additional_section"><a class="selflink" href="#Additional_section">Additional
  section</a></h2>
<b>-rgapopen</b> <i>float</i>
<div style="margin-left: 4.00ex;">This options specifies the gap insertion
  penalty for reside-based alignment. The gap insertion penalty is the score
  taken away when a gap is created. The best value depends on the choice of
  comparison matrix. The default value assumes you are using the EBLOSUM62
  matrix for protein sequences, and the EDNAFULL matrix for nucleotide
  sequences. Default value: 10</div>
<div class="Pp"></div>
<b>-rgapextend</b> <i>float</i>
<div style="margin-left: 4.00ex;">This options specifies the gap extension
  penalty for residue-based alignment. The gap extension, penalty is added to
  the standard gap penalty for each base or residue in the gap. This is how long
  gaps are penalized. Usually you will expect a few long gaps rather than many
  short gaps, so the gap extension penalty should be lower than the gap penalty.
  Default value: 0.5</div>
<div class="Pp"></div>
<b>-egapopen</b> <i>float</i>
<div style="margin-left: 4.00ex;">This options specifies the gap insertion
  penalty for element-based alignment. The gap insertion penalty is the score
  taken away when a gap is created. The best value depends on the choice of
  comparison matrix. The default value assumes you are using the EBLOSUM62
  matrix for protein sequences, and the EDNAFULL matrix for nucleotide
  sequences. Default value: 10</div>
<div class="Pp"></div>
<b>-egapextend</b> <i>float</i>
<div style="margin-left: 4.00ex;">This options specifies the gap extension
  penalty for secondary structure element-based alignment. The gap extension,
  penalty is added to the standard gap penalty for each base or residue in the
  gap. This is how long gaps are penalized. Usually you will expect a few long
  gaps rather than many short gaps, so the gap extension penalty should be lower
  than the gap penalty. Default value: 0.5</div>
<h2 class="Ss" title="Ss" id="Output_section"><a class="selflink" href="#Output_section">Output
  section</a></h2>
<b>-outssfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of the file
  containing top-scoring domains for residue-based alignment (output).A 'domain
  classification file' contains classification and other data for domains from
  SCOP or CATH, in DCF format (EMBL-like).</div>
<div class="Pp"></div>
<b>-outsefile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of the file
  containing top-scoring domains for secondary structure element-based alignment
  (output). A 'domain classification file' contains classification and other
  data for domains from SCOP or CATH, in DCF format (EMBL-like).</div>
<div class="Pp"></div>
<b>-logfile</b> <i>outfile</i>
<div style="margin-left: 4.00ex;">This option specifies the name of the ssematch
  log file (output). Default value: ssematch.log</div>
<h1 class="Sh" title="Sh" id="BUGS"><a class="selflink" href="#BUGS">BUGS</a></h1>
Bugs can be reported to the Debian Bug Tracking system
  (http://bugs.debian.org/emboss), or directly to the EMBOSS developers
  (http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
ssematch is fully documented via the <b>tfm</b>(1) system.
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
<b>Debian Med Packaging Team</b>
  &lt;debian-med-packaging@lists.alioth.debian.org&gt;
<div style="margin-left: 4.00ex;">Wrote the script used to autogenerate this
  manual page.</div>
<h1 class="Sh" title="Sh" id="COPYRIGHT"><a class="selflink" href="#COPYRIGHT">COPYRIGHT</a></h1>
This manual page was autogenerated from an Ajax Control Definition of the EMBOSS
  package. It can be redistributed under the same terms as EMBOSS itself.
<div style="height: 1.00em;">&#x00A0;</div>
</div>
<table class="foot">
  <tr>
    <td class="foot-date">08/11/2010</td>
    <td class="foot-os">DOMAINATRIX 0.1.0+20100721</td>
  </tr>
</table>
</body>
</html>
