<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:41:13 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>TWOFEAT(1e) EMBOSS Manual for Debian TWOFEAT(1e)</p>

<p style="margin-top: 1em">NAME <br>
twofeat - Finds neighbouring pairs of features in
sequence(s)</p>

<p style="margin-top: 1em">SYNOPSIS <br>
twofeat -sequence seqall [-asource string] -atype string
[-asense list] [-aminscore float] [-amaxscore float] [-atag
string] [-avalue string] [-bsource string] -btype string
<br>
[-bsense list] [-bminscore float] [-bmaxscore float] [-btag
string] [-bvalue string] [-overlap list] -minrange integer
-maxrange integer [-rangetype list] [-sense list] <br>
[-order list] [-twoout toggle] -typeout string -outfile
report</p>

<p style="margin-top: 1em">twofeat -help</p>

<p style="margin-top: 1em">DESCRIPTION <br>
twofeat is a command line program from EMBOSS (&acirc;the
European Molecular Biology Open Software Suite&acirc;). It
is part of the &quot;Feature tables&quot; command
group(s).</p>

<p style="margin-top: 1em">OPTIONS <br>
Input section <br>
-sequence seqall</p>

<p style="margin-top: 1em">First feature options <br>
-asource string <br>
By default any feature source in the feature table is
allowed. You can set this to match any feature source you
wish to allow. The source name is usually either the name of
<br>
the program that detected the feature or it is the feature
table (eg: EMBL) that the feature came from. The source may
be wildcarded by using &rsquo;*&rsquo;. If you wish to allow
more <br>
than one source, separate their names with the character
&rsquo;|&rsquo;, eg: gene* | embl Default value: *</p>

<p style="margin-top: 1em">-atype string <br>
By default every feature in the feature table is allowed.
You can set this to be any feature type you wish to allow.
See http://www.ebi.ac.uk/embl/WebFeat/ for a list of the
<br>
EMBL feature types and see Appendix A of the Swissprot user
manual in http://www.expasy.org/sprot/userman.html for a
list of the Swissprot feature types. The type may be <br>
wildcarded by using &rsquo;*&rsquo;. If you wish to allow
more than one type, separate their names with the character
&rsquo;|&rsquo;, eg: *UTR | intron Default value: *</p>

<p style="margin-top: 1em">-asense list <br>
By default any feature sense is allowed. You can set this to
match the required sense.</p>

<p style="margin-top: 1em">-aminscore float <br>
If this is greater than or equal to the maximum score, then
any score is allowed. Default value: 0.0</p>

<p style="margin-top: 1em">-amaxscore float <br>
If this is less than or equal to the maximum score, then any
score is permitted. Default value: 0.0</p>

<p style="margin-top: 1em">-atag string <br>
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, <br>
&rsquo;/db_xref&rsquo;, &rsquo;/EC_number&rsquo;,
&rsquo;/evidence&rsquo;, &rsquo;/exception&rsquo;,
&rsquo;/function&rsquo;, &rsquo;/gene&rsquo;,
&rsquo;/label&rsquo;, &rsquo;/map&rsquo;,
&rsquo;/note&rsquo;, &rsquo;/number&rsquo;,
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;, <br>
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Some of these tags also have
values, for example &rsquo;/gene&rsquo; can have the value
of the gene <br>
name. By default any feature tag in the feature table is
allowed. You can set this to match any feature tag you wish
to allow. The tag may be wildcarded by using
&rsquo;*&rsquo;. If you <br>
wish to allow more than one tag, separate their names with
the character &rsquo;|&rsquo;, eg: gene | label Default
value: *</p>

<p style="margin-top: 1em">-avalue string <br>
Tag values are the values associated with a feature tag.
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of <br>
feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, &rsquo;/db_xref&rsquo;,
&rsquo;/EC_number&rsquo;, &rsquo;/evidence&rsquo;,
&rsquo;/exception&rsquo;, &rsquo;/function&rsquo;,
&rsquo;/gene&rsquo;, &rsquo;/label&rsquo;,
&rsquo;/map&rsquo;, &rsquo;/note&rsquo;,
&rsquo;/number&rsquo;, <br>
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;,
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Only some of these tags can have
values, <br>
for example &rsquo;/gene&rsquo; can have the value of the
gene name. By default any feature tag value in the feature
table is allowed. You can set this to match any feature tag
value you <br>
wish to allow. The tag value may be wildcarded by using
&rsquo;*&rsquo;. If you wish to allow more than one tag
value, separate their names with the character
&rsquo;|&rsquo;, eg: pax* | 10 Default <br>
value: *</p>

<p style="margin-top: 1em">Second feature options <br>
-bsource string <br>
By default any feature source in the feature table is
allowed. You can set this to match any feature source you
wish to allow. The source name is usually either the name of
<br>
the program that detected the feature or it is the feature
table (eg: EMBL) that the feature came from. The source may
be wildcarded by using &rsquo;*&rsquo;. If you wish to allow
more <br>
than one source, separate their names with the character
&rsquo;|&rsquo;, eg: gene* | embl Default value: *</p>

<p style="margin-top: 1em">-btype string <br>
By default every feature in the feature table is allowed.
You can set this to be any feature type you wish to allow.
See http://www.ebi.ac.uk/embl/WebFeat/ for a list of the
<br>
EMBL feature types and see Appendix A of the Swissprot user
manual in http://www.expasy.org/sprot/userman.html for a
list of the Swissprot feature types. The type may be <br>
wildcarded by using &rsquo;*&rsquo;. If you wish to allow
more than one type, separate their names with the character
&rsquo;|&rsquo;, eg: *UTR | intron Default value: *</p>

<p style="margin-top: 1em">-bsense list <br>
By default any feature sense is allowed. You can set this to
match the required sense.</p>

<p style="margin-top: 1em">-bminscore float <br>
If this is greater than or equal to the maximum score, then
any score is allowed. Default value: 0.0</p>

<p style="margin-top: 1em">-bmaxscore float <br>
If this is less than or equal to the maximum score, then any
score is permitted. Default value: 0.0</p>

<p style="margin-top: 1em">-btag string <br>
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, <br>
&rsquo;/db_xref&rsquo;, &rsquo;/EC_number&rsquo;,
&rsquo;/evidence&rsquo;, &rsquo;/exception&rsquo;,
&rsquo;/function&rsquo;, &rsquo;/gene&rsquo;,
&rsquo;/label&rsquo;, &rsquo;/map&rsquo;,
&rsquo;/note&rsquo;, &rsquo;/number&rsquo;,
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;, <br>
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Some of these tags also have
values, for example &rsquo;/gene&rsquo; can have the value
of the gene <br>
name. By default any feature tag in the feature table is
allowed. You can set this to match any feature tag you wish
to allow. The tag may be wildcarded by using
&rsquo;*&rsquo;. If you <br>
wish to allow more than one tag, separate their names with
the character &rsquo;|&rsquo;, eg: gene | label Default
value: *</p>

<p style="margin-top: 1em">-bvalue string <br>
Tag values are the values associated with a feature tag.
Tags are the types of extra values that a feature may have.
For example in the EMBL feature table, a &rsquo;CDS&rsquo;
type of <br>
feature may have the tags &rsquo;/codon&rsquo;,
&rsquo;/codon_start&rsquo;, &rsquo;/db_xref&rsquo;,
&rsquo;/EC_number&rsquo;, &rsquo;/evidence&rsquo;,
&rsquo;/exception&rsquo;, &rsquo;/function&rsquo;,
&rsquo;/gene&rsquo;, &rsquo;/label&rsquo;,
&rsquo;/map&rsquo;, &rsquo;/note&rsquo;,
&rsquo;/number&rsquo;, <br>
&rsquo;/partial&rsquo;, &rsquo;/product&rsquo;,
&rsquo;/protein_id&rsquo;, &rsquo;/pseudo&rsquo;,
&rsquo;/standard_name&rsquo;, &rsquo;/translation&rsquo;,
&rsquo;/transl_except&rsquo;, &rsquo;/transl_table&rsquo;,
or &rsquo;/usedin&rsquo;. Only some of these tags can have
values, <br>
for example &rsquo;/gene&rsquo; can have the value of the
gene name. By default any feature tag value in the feature
table is allowed. You can set this to match any feature tag
value you <br>
wish to allow. The tag value may be wildcarded by using
&rsquo;*&rsquo;. If you wish to allow more than one tag
value, separate their names with the character
&rsquo;|&rsquo;, eg: pax* | 10 Default <br>
value: *</p>

<p style="margin-top: 1em">Feature relation options <br>
-overlap list <br>
This allows you to specify the allowed overlaps of the
features A and B. You can allow any or no overlaps, specify
that they must or must not overlap, that one must or must
<br>
not be wholly enclosed within another feature. Default
value: A</p>

<p style="margin-top: 1em">-minrange integer <br>
If this is greater or equal to &rsquo;maxrange&rsquo;, then
no min or max range is specified</p>

<p style="margin-top: 1em">-maxrange integer <br>
If this is less than or equal to &rsquo;minrange&rsquo;,
then no min or max range is specified</p>

<p style="margin-top: 1em">-rangetype list <br>
This allows you to specify the positions from which the
allowed minimum or maximum distance between the features is
measured Default value: N</p>

<p style="margin-top: 1em">-sense list <br>
This allows you to specify the required sense that the two
features must be on. This is ignored (always
&rsquo;Any&rsquo;) when looking at protein sequence
features. Default value: A</p>

<p style="margin-top: 1em">-order list <br>
This allows you to specify the required order of the two
features. The order is measured from the start positions of
the features. This criterion is always applied despite <br>
the specified overlap type required. Default value: A</p>

<p style="margin-top: 1em">Output section <br>
-twoout toggle <br>
If you set this to be true, then the two features themselves
will be written out. If it is left as false, then a single
feature will be written out covering the two features <br>
you found. Default value: N</p>

<p style="margin-top: 1em">-typeout string <br>
If you have specified that the pairs of features that are
found should be reported as one feature in the output, then
you can specify the &rsquo;type&rsquo; name of the new
feature here. <br>
By default every feature in the feature table is allowed.
See http://www.ebi.ac.uk/embl/WebFeat/ for a list of the
EMBL feature types and see Appendix A of the Swissprot user
<br>
manual in http://www.expasy.org/sprot/userman.html for a
list of the Swissprot feature types. If you specify an
invalid feature type name, then the default name <br>
&rsquo;misc_feature&rsquo; is used. Default value:
misc_feature</p>

<p style="margin-top: 1em">-outfile report</p>

<p style="margin-top: 1em">BUGS <br>
Bugs can be reported to the Debian Bug Tracking system
(http://bugs.debian.org/emboss), or directly to the EMBOSS
developers <br>

(http://sourceforge.net/tracker/?group_id=93650&amp;atid=605031).</p>

<p style="margin-top: 1em">SEE ALSO <br>
twofeat is fully documented via the tfm(1) system.</p>

<p style="margin-top: 1em">AUTHOR <br>
Debian Med Packaging Team
&lt;debian-med-packaging@lists.alioth.debian.org&gt; <br>
Wrote the script used to autogenerate this manual page.</p>

<p style="margin-top: 1em">COPYRIGHT <br>
This manual page was autogenerated from an Ajax Control
Definition of the EMBOSS package. It can be redistributed
under the same terms as EMBOSS itself.</p>

<p style="margin-top: 1em">EMBOSS 6.4.0 05/11/2012
TWOFEAT(1e)</p>
<hr>
</body>
</html>
