<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:41:58 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>VARCALL(1) User Commands VARCALL(1)</p>

<p style="margin-top: 1em">NAME <br>
varcall - ea-utils: output summry stats or perform variant
calling</p>

<p style="margin-top: 1em">SYNOPSIS <br>
varcall &lt;-s|-v&gt; &lt;-f REF&gt; [options] bam1
[bam2...]</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Version: 0.9.632 (BETA)</p>

<p style="margin-top: 1em">Either outputs summry stats for
the list of files, or performs variant calling</p>

<p style="margin-top: 1em">Options (later options override
earlier):</p>

<p style="margin-top: 1em">-s Calculate statistics -v
Calculate variants bases on supplied parameters (see -S) -f
Reference fasta (required if using bams, ignored otherwise)
-m <br>
Min locii depth (0) -a Min allele depth (0) -p Min allele
pct by quality (0) -q Min qual (3) -Q Min mapping quality
(0) -b Min pct <br>
balance (strand/total) (0) -D FLOAT Max duplicate read
fraction (depth/length per position) (1) -B Turn off BAQ
correction (false) -R Homopolymer repeat <br>
indel filtering (8) -e FLOAT Alpha filter to use, requires
-l or -S (.05) -g FLOAT Global minimum error rate (default:
assume phred is ok) -l INT Number of locii in <br>
total pileup used for bonferroni (1 mil) -x CHR:POS Output
this pos only, then quit -N FIL Output noise stats to FIL -S
FIL Read in statistics and params from a previ&acirc; <br>
ous run with -s (do this!) -A ANNOT Calculate in-target
stats using the annotation file (requires -o) -o PREFIX
Output prefix (note: overlaps with -N)</p>

<p style="margin-top: 1em">Input files</p>

<p style="margin-top: 1em">Files must be sorted bam files
with bai index files available. Alternatively, a single
pileup file can be supplied.</p>

<p style="margin-top: 1em">Output files</p>

<p style="margin-top: 1em">Varcalls go to stdout. Stats go
to stdout, or stderr if varcalling too</p>

<p style="margin-top: 1em">If an output prefix is used,
files are created as follows: <br>
PREFIX.var <br>
Variant calls in tab delimited &rsquo;varcall&rsquo;
format</p>

<p style="margin-top: 1em">PREFIX.eav <br>
Variant calls in tab delimited &rsquo;ea-var&rsquo;
format</p>

<p style="margin-top: 1em">PREFIX.vcf <br>
Variant calls, in vcf format</p>

<p style="margin-top: 1em">PREFIX.varsum <br>
Summary of variant calls</p>

<p style="margin-top: 1em">PREFIX.tgt <br>
On-target stats detail</p>

<p style="margin-top: 1em">PREFIX.tgtsum <br>
Summary of on-target stats</p>

<p style="margin-top: 1em">PREFIX.noise <br>
Noise stats detail</p>

<p style="margin-top: 1em">Stats Output:</p>

<p style="margin-top: 1em">Contains mean, median, quartile
information for depth, base quality, read len, mapping
quality, indel levels. Also estimates parameters suitable
for variant calls, and can be <br>
passed directly to this program for variant calls</p>

<p style="margin-top: 1em">Filtering Details:</p>

<p style="margin-top: 1em">varcall 1.1.2 July 2015
VARCALL(1)</p>
<hr>
</body>
</html>
