<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>vcftools(man)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">vcftools(man)</td>
    <td class="head-vol">05 January 2016</td>
    <td class="head-rtitle">vcftools(man)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
vcftools v0.1.14 - Utilities for the variant call format (VCF) and binary
  variant call format (BCF)
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<b>vcftools</b> [ <b>--vcf</b> FILE | <b>--gzvcf</b> FILE | <b>--bcf</b> FILE] [
  <b>--out</b> OUTPUT PREFIX ] [ FILTERING OPTIONS ] [ OUTPUT OPTIONS ]
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
vcftools is a suite of functions for use on genetic variation data in the form
  of VCF and BCF files. The tools provided will be used mainly to summarize
  data, run calculations on data, filter out data, and convert data into other
  useful file formats.
<h1 class="Sh" title="Sh" id="EXAMPLES"><a class="selflink" href="#EXAMPLES">EXAMPLES</a></h1>
Output allele frequency for all sites in the input vcf file from chromosome 1
<div style="margin-left: 2.00ex;"><b>vcftools</b> --gzvcf input_file.vcf.gz
  --freq --chr 1 --out chr1_analysis</div>
<div class="Pp"></div>
Output a new vcf file from the input vcf file that removes any indel sites
<div style="margin-left: 2.00ex;"><b>vcftools</b> --vcf input_file.vcf
  --remove-indels --recode --recode-INFO-all --out SNPs_only</div>
<div class="Pp"></div>
Output file comparing the sites in two vcf files
<div style="margin-left: 2.00ex;"><b>vcftools</b> --gzvcf input_file1.vcf.gz
  --gzdiff input_file2.vcf.gz --diff-site --out in1_v_in2</div>
<div class="Pp"></div>
Output a new vcf file to standard out without any sites that have a filter tag,
  then compress it with gzip
<div style="margin-left: 2.00ex;"><b>vcftools</b> --gzvcf input_file.vcf.gz
  --remove-filtered-all --recode --stdout | gzip -c &gt;
  output_PASS_only.vcf.gz</div>
<div class="Pp"></div>
Output a Hardy-Weinberg p-value for every site in the bcf file that does not
  have any missing genotypes
<div style="margin-left: 2.00ex;"><b>vcftools</b> --bcf input_file.bcf --hardy
  --max-missing 1.0 --out output_noMissing</div>
<div class="Pp"></div>
Output nucleotide diversity at a list of positions
<div style="margin-left: 2.00ex;">zcat input_file.vcf.gz | <b>vcftools</b> --vcf
  - --site-pi --positions SNP_list.txt --out nucleotide_diversity</div>
<h1 class="Sh" title="Sh" id="BASIC_OPTIONS"><a class="selflink" href="#BASIC_OPTIONS">BASIC
  OPTIONS</a></h1>
These options are used to specify the input and output files.
<h2 class="Ss" title="Ss" id="INPUT_FILE_OPTIONS"><a class="selflink" href="#INPUT_FILE_OPTIONS">INPUT
  FILE OPTIONS</a></h2>
<div style="margin-left: 2.00ex;"><b>--vcf</b> <i>&lt;input_filename&gt;</i>
<div style="margin-left: 2.00ex;">This option defines the VCF file to be
  processed. VCFtools expects files in VCF format v4.0, v4.1 or v4.2. The latter
  two are supported with some small limitations. If the user provides a dash
  character '-' as a file name, the program expects a VCF file to be piped in
  through standard in.</div>
<div class="Pp"></div>
<b>--gzvcf</b> <i>&lt;input_filename&gt;</i>
<div style="margin-left: 2.00ex;">This option can be used in place of the --vcf
  option to read compressed (gzipped) VCF files directly.</div>
<div class="Pp"></div>
<b>--bcf</b> <i>&lt;input_filename&gt;</i>
<div style="margin-left: 2.00ex;">This option can be used in place of the --vcf
  option to read BCF2 files directly. You do not need to specify if this file is
  compressed with BGZF encoding. If the user provides a dash character '-' as a
  file name, the program expects a BCF2 file to be piped in through standard
  in.</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_FILE_OPTIONS"><a class="selflink" href="#OUTPUT_FILE_OPTIONS">OUTPUT
  FILE OPTIONS</a></h2>
<div style="margin-left: 2.00ex;"><b>--out</b> <i>&lt;output_prefix&gt;</i>
<div style="margin-left: 2.00ex;">This option defines the output filename prefix
  for all files generated by vcftools. For example, if &lt;prefix&gt; is set to
  output_filename, then all output files will be of the form output_filename.***
  . If this option is omitted, all output files will have the prefix
  &quot;out.&quot; in the current working directory.</div>
<div class="Pp"></div>
<b>--stdout</b>
<div>&#x00A0;</div>
<b>-c</b>
<div style="margin-left: 2.00ex;">These options direct the vcftools output to
  standard out so it can be piped into another program or written directly to a
  filename of choice. However, a select few output functions cannot be written
  to standard out.</div>
<div class="Pp"></div>
<b>--temp</b> <i>&lt;temporary_directory&gt;</i>
<div style="margin-left: 2.00ex;">This option can be used to redirect any
  temporary files that vcftools creates into a specified directory.</div>
</div>
<h1 class="Sh" title="Sh" id="SITE_FILTERING_OPTIONS"><a class="selflink" href="#SITE_FILTERING_OPTIONS">SITE
  FILTERING OPTIONS</a></h1>
These options are used to include or exclude certain sites from any analysis
  being performed by the program.
<h2 class="Ss" title="Ss" id="POSITION_FILTERING"><a class="selflink" href="#POSITION_FILTERING">POSITION
  FILTERING</a></h2>
<div style="margin-left: 2.00ex;"><b>--chr</b> <i>&lt;chromosome&gt;</i>
<div>&#x00A0;</div>
<b>--not-chr</b> <i>&lt;chromosome&gt;</i>
<div style="margin-left: 2.00ex;">Includes or excludes sites with indentifiers
  matching &lt;chromosome&gt;. These options may be used multiple times to
  include or exclude more than one chromosome.</div>
<div class="Pp"></div>
<b>--from-bp</b> <i>&lt;integer&gt;</i>
<div>&#x00A0;</div>
<b>--to-bp</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">These options specify a lower bound and upper
  bound for a range of sites to be processed. Sites with positions less than or
  greater than these values will be excluded. These options can only be used in
  conjunction with a single usage of --chr. Using one of these does not require
  use of the other.</div>
<div class="Pp"></div>
<b>--positions</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--exclude-positions</b> <i>&lt;filename&gt;</i>
<div style="margin-left: 2.00ex;">Include or exclude a set of sites on the basis
  of a list of positions in a file. Each line of the input file should contain a
  (tab-separated) chromosome and position. The file can have comment lines that
  start with a &quot;#&quot;, they will be ignored.</div>
<div class="Pp"></div>
<b>--positions-overlap</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--exclude-positions-overlap</b> <i>&lt;filename&gt;</i>
<div style="margin-left: 2.00ex;">Include or exclude a set of sites on the basis
  of the reference allele overlapping with a list of positions in a file. Each
  line of the input file should contain a (tab-separated) chromosome and
  position. The file can have comment lines that start with a &quot;#&quot;,
  they will be ignored.</div>
<div class="Pp"></div>
<b>--bed</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--exclude-bed</b> <i>&lt;filename&gt;</i>
<div style="margin-left: 2.00ex;">Include or exclude a set of sites on the basis
  of a BED file. Only the first three columns (chrom, chromStart and chromEnd)
  are required. The BED file is expected to have a header line. A site will be
  kept or excluded if any part of any allele (REF or ALT) at a site is within
  the range of one of the BED entries.</div>
<div class="Pp"></div>
<b>--thin</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Thin sites so that no two sites are within the
  specified distance from one another.</div>
<div class="Pp"></div>
<b>--mask</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--invert-mask</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--mask-min</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">These options are used to specify a FASTA-like
  mask file to filter with. The mask file contains a sequence of integer digits
  (between 0 and 9) for each position on a chromosome that specify if a site at
  that position should be filtered or not.
<div>&#x00A0;</div>
An example mask file would look like:
<div style="margin-left: 2.00ex;"><i>&gt;1</i>
<div>&#x00A0;</div>
<i>0000011111222...</i>
<div>&#x00A0;</div>
<i>&gt;2</i>
<div>&#x00A0;</div>
<i>2222211111000...</i></div>
In this example, sites in the VCF file located within the first 5 bases of the
  start of chromosome 1 would be kept, whereas sites at position 6 onwards would
  be filtered out. And sites after the 11th position on chromosome 2 would be
  filtered out as well.
<div>&#x00A0;</div>
The &quot;--invert-mask&quot; option takes the same format mask file as the
  &quot;--mask&quot; option, however it inverts the mask file before filtering
  with it.
<div>&#x00A0;</div>
And the &quot;--mask-min&quot; option specifies a threshold mask value between 0
  and 9 to filter positions by. The default threshold is 0, meaning only sites
  with that value or lower will be kept.</div>
</div>
<h2 class="Ss" title="Ss" id="SITE_ID_FILTERING"><a class="selflink" href="#SITE_ID_FILTERING">SITE
  ID FILTERING</a></h2>
<div style="margin-left: 2.00ex;"><b>--snp</b> <i>&lt;string&gt;</i>
<div style="margin-left: 2.00ex;">Include SNP(s) with matching ID (e.g. a dbSNP
  rsID). This command can be used multiple times in order to include more than
  one SNP.</div>
<div class="Pp"></div>
<b>--snps</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--exclude</b> <i>&lt;filename&gt;</i>
<div style="margin-left: 2.00ex;">Include or exclude a list of SNPs given in a
  file. The file should contain a list of SNP IDs (e.g. dbSNP rsIDs), with one
  ID per line. No header line is expected.</div>
</div>
<h2 class="Ss" title="Ss" id="VARIANT_TYPE_FILTERING"><a class="selflink" href="#VARIANT_TYPE_FILTERING">VARIANT
  TYPE FILTERING</a></h2>
<div style="margin-left: 2.00ex;"><b>--keep-only-indels</b>
<div>&#x00A0;</div>
<b>--remove-indels</b>
<div style="margin-left: 2.00ex;">Include or exclude sites that contain an
  indel. For these options &quot;indel&quot; means any variant that alters the
  length of the REF allele.</div>
</div>
<h2 class="Ss" title="Ss" id="FILTER_FLAG_FILTERING"><a class="selflink" href="#FILTER_FLAG_FILTERING">FILTER
  FLAG FILTERING</a></h2>
<div style="margin-left: 2.00ex;"><b>--remove-filtered-all</b>
<div style="margin-left: 2.00ex;">Removes all sites with a FILTER flag other
  than PASS.</div>
<div class="Pp"></div>
<b>--keep-filtered</b> <i>&lt;string&gt;</i>
<div>&#x00A0;</div>
<b>--remove-filtered</b> <i>&lt;string&gt;</i>
<div style="margin-left: 2.00ex;">Includes or excludes all sites marked with a
  specific FILTER flag. These options may be used more than once to specify
  multiple FILTER flags.</div>
</div>
<h2 class="Ss" title="Ss" id="INFO_FIELD_FILTERING"><a class="selflink" href="#INFO_FIELD_FILTERING">INFO
  FIELD FILTERING</a></h2>
<div style="margin-left: 2.00ex;"><b>--keep-INFO</b> <i>&lt;string&gt;</i>
<div>&#x00A0;</div>
<b>--remove-INFO</b> <i>&lt;string&gt;</i>
<div style="margin-left: 2.00ex;">Includes or excludes all sites with a specific
  INFO flag. These options only filter on the presence of the flag and not its
  value. These options can be used multiple times to specify multiple INFO
  flags.</div>
</div>
<h2 class="Ss" title="Ss" id="ALLELE_FILTERING"><a class="selflink" href="#ALLELE_FILTERING">ALLELE
  FILTERING</a></h2>
<div style="margin-left: 2.00ex;"><b>--maf</b> <i>&lt;float&gt;</i>
<div>&#x00A0;</div>
<b>--max-maf</b> <i>&lt;float&gt;</i>
<div style="margin-left: 2.00ex;">Include only sites with a Minor Allele
  Frequency greater than or equal to the &quot;--maf&quot; value and less than
  or equal to the &quot;--max-maf&quot; value. One of these options may be used
  without the other. Allele frequency is defined as the number of times an
  allele appears over all individuals at that site, divided by the total number
  of non-missing alleles at that site.</div>
<div class="Pp"></div>
<b>--non-ref-af</b> <i>&lt;float&gt;</i>
<div>&#x00A0;</div>
<b>--max-non-ref-af</b> <i>&lt;float&gt;</i>
<div>&#x00A0;</div>
<b>--non-ref-ac</b> <i>&lt;integer&gt;</i>
<div>&#x00A0;</div>
<b>--max-non-ref-ac</b> <i>&lt;integer&gt;</i>
<div class="Pp"></div>
<b>--non-ref-af-any</b> <i>&lt;float&gt;</i>
<div>&#x00A0;</div>
<b>--max-non-ref-af-any</b> <i>&lt;float&gt;</i>
<div>&#x00A0;</div>
<b>--non-ref-ac-any</b> <i>&lt;integer&gt;</i>
<div>&#x00A0;</div>
<b>--max-non-ref-ac-any</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Include only sites with all Non-Reference
  (ALT) Allele Frequencies (af) or Counts (ac) within the range specified, and
  including the specified value. The default options require <b>all</b> alleles
  to meet the specified criteria, whereas the options appended with
  &quot;any&quot; require only one allele to meet the criteria. The Allele
  frequency is defined as the number of times an allele appears over all
  individuals at that site, divided by the total number of non-missing alleles
  at that site.</div>
<div class="Pp"></div>
<b>--mac</b> <i>&lt;integer&gt;</i>
<div>&#x00A0;</div>
<b>--max-mac</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Include only sites with Minor Allele Count
  greater than or equal to the &quot;--mac&quot; value and less than or equal to
  the &quot;--max-mac&quot; value. One of these options may be used without the
  other. Allele count is simply the number of times that allele appears over all
  individuals at that site.</div>
<div class="Pp"></div>
<b>--min-alleles</b> <i>&lt;integer&gt;</i>
<div>&#x00A0;</div>
<b>--max-alleles</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Include only sites with a number of alleles
  greater than or equal to the &quot;--min-alleles&quot; value and less than or
  equal to the &quot;--max-alleles&quot; value. One of these options may be used
  without the other.
<div>&#x00A0;</div>
For example, to include only bi-allelic sites, one could use:
<div>&#x00A0;</div>
<div style="margin-left: 2.00ex;"><b>vcftools</b> --vcf file1.vcf --min-alleles
  2 --max-alleles 2</div>
</div>
</div>
<h2 class="Ss" title="Ss" id="GENOTYPE_VALUE_FILTERING"><a class="selflink" href="#GENOTYPE_VALUE_FILTERING">GENOTYPE
  VALUE FILTERING</a></h2>
<div style="margin-left: 2.00ex;"><b>--min-meanDP</b> <i>&lt;float&gt;</i>
<div>&#x00A0;</div>
<b>--max-meanDP</b> <i>&lt;float&gt;</i>
<div style="margin-left: 2.00ex;">Includes only sites with mean depth values
  (over all included individuals) greater than or equal to the
  &quot;--min-meanDP&quot; value and less than or equal to the
  &quot;--max-meanDP&quot; value. One of these options may be used without the
  other. These options require that the &quot;DP&quot; FORMAT tag is included
  for each site.</div>
<div class="Pp"></div>
<b>--hwe</b> <i>&lt;float&gt;</i>
<div style="margin-left: 2.00ex;">Assesses sites for Hardy-Weinberg Equilibrium
  using an exact test, as defined by Wigginton, Cutler and Abecasis (2005).
  Sites with a p-value below the threshold defined by this option are taken to
  be out of HWE, and therefore excluded.</div>
<div class="Pp"></div>
<b>--max-missing</b> <i>&lt;float&gt;</i>
<div style="margin-left: 2.00ex;">Exclude sites on the basis of the proportion
  of missing data (defined to be between 0 and 1, where 0 allows sites that are
  completely missing and 1 indicates no missing data allowed).</div>
<div class="Pp"></div>
<b>--max-missing-count</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Exclude sites with more than this number of
  missing genotypes over all individuals.</div>
<div class="Pp"></div>
<b>--phased</b>
<div style="margin-left: 2.00ex;">Excludes all sites that contain unphased
  genotypes.</div>
</div>
<h2 class="Ss" title="Ss" id="MISCELLANEOUS_FILTERING"><a class="selflink" href="#MISCELLANEOUS_FILTERING">MISCELLANEOUS
  FILTERING</a></h2>
<div style="margin-left: 2.00ex;"><b>--minQ</b> <i>&lt;float&gt;</i>
<div style="margin-left: 2.00ex;">Includes only sites with Quality value above
  this threshold.</div>
</div>
<h1 class="Sh" title="Sh" id="INDIVIDUAL_FILTERING_OPTIONS"><a class="selflink" href="#INDIVIDUAL_FILTERING_OPTIONS">INDIVIDUAL
  FILTERING OPTIONS</a></h1>
These options are used to include or exclude certain individuals from any
  analysis being performed by the program.
<div>&#x00A0;</div>
<div style="margin-left: 2.00ex;"><b>--indv</b> <i>&lt;string&gt;</i>
<div>&#x00A0;</div>
<b>--remove-indv</b> <i>&lt;string&gt;</i>
<div style="margin-left: 2.00ex;">Specify an individual to be kept or removed
  from the analysis. This option can be used multiple times to specify multiple
  individuals. If both options are specified, then the &quot;--indv&quot; option
  is executed before the &quot;--remove-indv option&quot;.</div>
<div class="Pp"></div>
<b>--keep</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--remove</b> <i>&lt;filename&gt;</i>
<div style="margin-left: 2.00ex;">Provide files containing a list of individuals
  to either include or exclude in subsequent analysis. Each individual ID (as
  defined in the VCF headerline) should be included on a separate line. If both
  options are used, then the &quot;--keep&quot; option is executed before the
  &quot;--remove&quot; option. When multiple files are provided, the union of
  individuals from all keep files subtracted by the union of individuals from
  all remove files are kept. No header line is expected.</div>
<div class="Pp"></div>
<b>--max-indv</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Randomly thins individuals so that only the
  specified number are retained.</div>
</div>
<h1 class="Sh" title="Sh" id="GENOTYPE_FILTERING_OPTIONS"><a class="selflink" href="#GENOTYPE_FILTERING_OPTIONS">GENOTYPE
  FILTERING OPTIONS</a></h1>
These options are used to exclude genotypes from any analysis being performed by
  the program. If excluded, these values will be treated as missing.
<div>&#x00A0;</div>
<div style="margin-left: 2.00ex;"><b>--remove-filtered-geno-all</b>
<div style="margin-left: 2.00ex;">Excludes all genotypes with a FILTER flag not
  equal to &quot;.&quot; (a missing value) or PASS.</div>
<div class="Pp"></div>
<b>--remove-filtered-geno</b> <i>&lt;string&gt;</i>
<div style="margin-left: 2.00ex;">Excludes genotypes with a specific FILTER
  flag.</div>
<div class="Pp"></div>
<b>--minGQ</b> <i>&lt;float&gt;</i>
<div style="margin-left: 2.00ex;">Exclude all genotypes with a quality below the
  threshold specified. This option requires that the &quot;GQ&quot; FORMAT tag
  is specified for all sites.</div>
<div class="Pp"></div>
<b>--minDP</b> <i>&lt;float&gt;</i>
<div>&#x00A0;</div>
<b>--maxDP</b> <i>&lt;float&gt;</i>
<div style="margin-left: 2.00ex;">Includes only genotypes greater than or equal
  to the &quot;--minDP&quot; value and less than or equal to the
  &quot;--maxDP&quot; value. This option requires that the &quot;DP&quot; FORMAT
  tag is specified for all sites.</div>
</div>
<h1 class="Sh" title="Sh" id="OUTPUT_OPTIONS"><a class="selflink" href="#OUTPUT_OPTIONS">OUTPUT
  OPTIONS</a></h1>
These options specify which analyses or conversions to perform on the data that
  passed through all specified filters.
<h2 class="Ss" title="Ss" id="OUTPUT_ALLELE_STATISTICS"><a class="selflink" href="#OUTPUT_ALLELE_STATISTICS">OUTPUT
  ALLELE STATISTICS</a></h2>
<div style="margin-left: 2.00ex;"><b>--freq</b>
<div>&#x00A0;</div>
<b>--freq2</b>
<div style="margin-left: 2.00ex;">Outputs the allele frequency for each site in
  a file with the suffix &quot;.frq&quot;. The second option is used to suppress
  output of any information about the alleles.</div>
<div class="Pp"></div>
<b>--counts</b>
<div>&#x00A0;</div>
<b>--counts2</b>
<div style="margin-left: 2.00ex;">Outputs the raw allele counts for each site in
  a file with the suffix &quot;.frq.count&quot;. The second option is used to
  suppress output of any information about the alleles.</div>
<div class="Pp"></div>
<b>--derived</b>
<div style="margin-left: 2.00ex;">For use with the previous four frequency and
  count options only. Re-orders the output file columns so that the ancestral
  allele appears first. This option relies on the ancestral allele being
  specified in the VCF file using the AA tag in the INFO field.</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_DEPTH_STATISTICS"><a class="selflink" href="#OUTPUT_DEPTH_STATISTICS">OUTPUT
  DEPTH STATISTICS</a></h2>
<div style="margin-left: 2.00ex;"><b>--depth</b>
<div style="margin-left: 2.00ex;">Generates a file containing the mean depth per
  individual. This file has the suffix &quot;.idepth&quot;.</div>
<div class="Pp"></div>
<b>--site-depth</b>
<div style="margin-left: 2.00ex;">Generates a file containing the depth per site
  summed across all individuals. This output file has the suffix
  &quot;.ldepth&quot;.</div>
<div class="Pp"></div>
<b>--site-mean-depth</b>
<div style="margin-left: 2.00ex;">Generates a file containing the mean depth per
  site averaged across all individuals. This output file has the suffix
  &quot;.ldepth.mean&quot;.</div>
<div class="Pp"></div>
<b>--geno-depth</b>
<div style="margin-left: 2.00ex;">Generates a (possibly very large) file
  containing the depth for each genotype in the VCF file. Missing entries are
  given the value -1. The file has the suffix &quot;.gdepth&quot;.</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_LD_STATISTICS"><a class="selflink" href="#OUTPUT_LD_STATISTICS">OUTPUT
  LD STATISTICS</a></h2>
<div style="margin-left: 2.00ex;"><b>--hap-r2</b>
<div style="margin-left: 2.00ex;">Outputs a file reporting the r2, D, and D'
  statistics using phased haplotypes. These are the traditional measures of LD
  often reported in the population genetics literature. The output file has the
  suffix &quot;.hap.ld&quot;. This option assumes that the VCF input file has
  phased haplotypes.</div>
<div class="Pp"></div>
<b>--geno-r2</b>
<div style="margin-left: 2.00ex;">Calculates the squared correlation coefficient
  between genotypes encoded as 0, 1 and 2 to represent the number of
  non-reference alleles in each individual. This is the same as the LD measure
  reported by PLINK. The D and D' statistics are only available for phased
  genotypes. The output file has the suffix &quot;.geno.ld&quot;.</div>
<div class="Pp"></div>
<b>--geno-chisq</b>
<div style="margin-left: 2.00ex;">If your data contains sites with more than two
  alleles, then this option can be used to test for genotype independence via
  the chi-squared statistic. The output file has the suffix
  &quot;.geno.chisq&quot;.</div>
<div class="Pp"></div>
<b>--hap-r2-positions</b> <i>&lt;positions list file&gt;</i>
<div>&#x00A0;</div>
<b>--geno-r2-positions</b> <i>&lt;positions list file&gt;</i>
<div style="margin-left: 2.00ex;">Outputs a file reporting the r2 statistics of
  the sites contained in the provided file verses all other sites. The output
  files have the suffix &quot;.list.hap.ld&quot; or &quot;.list.geno.ld&quot;,
  depending on which option is used.</div>
<div class="Pp"></div>
<b>--ld-window</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">This optional parameter defines the maximum
  number of SNPs between the SNPs being tested for LD in the
  &quot;--hap-r2&quot;, &quot;--geno-r2&quot;, and &quot;--geno-chisq&quot;
  functions.</div>
<div class="Pp"></div>
<b>--ld-window-bp</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">This optional parameter defines the maximum
  number of physical bases between the SNPs being tested for LD in the
  &quot;--hap-r2&quot;, &quot;--geno-r2&quot;, and &quot;--geno-chisq&quot;
  functions.</div>
<div class="Pp"></div>
<b>--ld-window-min</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">This optional parameter defines the minimum
  number of SNPs between the SNPs being tested for LD in the
  &quot;--hap-r2&quot;, &quot;--geno-r2&quot;, and &quot;--geno-chisq&quot;
  functions.</div>
<div class="Pp"></div>
<b>--ld-window-bp-min</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">This optional parameter defines the minimum
  number of physical bases between the SNPs being tested for LD in the
  &quot;--hap-r2&quot;, &quot;--geno-r2&quot;, and &quot;--geno-chisq&quot;
  functions.</div>
<div class="Pp"></div>
<b>--min-r2</b> <i>&lt;float&gt;</i>
<div style="margin-left: 2.00ex;">This optional parameter sets a minimum value
  for r2, below which the LD statistic is not reported by the
  &quot;--hap-r2&quot;, &quot;--geno-r2&quot;, and &quot;--geno-chisq&quot;
  functions.</div>
<div class="Pp"></div>
<b>--interchrom-hap-r2</b>
<div>&#x00A0;</div>
<b>--interchrom-geno-r2</b>
<div style="margin-left: 2.00ex;">Outputs a file reporting the r2 statistics for
  sites on different chromosomes. The output files have the suffix
  &quot;.interchrom.hap.ld&quot; or &quot;.interchrom.geno.ld&quot;, depending
  on the option used.</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_TRANSITION/TRANSVERSION_STATISTICS"><a class="selflink" href="#OUTPUT_TRANSITION/TRANSVERSION_STATISTICS">OUTPUT
  TRANSITION/TRANSVERSION STATISTICS</a></h2>
<div style="margin-left: 2.00ex;"><b>--TsTv</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Calculates the Transition / Transversion ratio
  in bins of size defined by this option. Only uses bi-allelic SNPs. The
  resulting output file has the suffix &quot;.TsTv&quot;.</div>
<div class="Pp"></div>
<b>--TsTv-summary</b>
<div style="margin-left: 2.00ex;">Calculates a simple summary of all Transitions
  and Transversions. The output file has the suffix
  &quot;.TsTv.summary&quot;.</div>
<div class="Pp"></div>
<b>--TsTv-by-count</b>
<div style="margin-left: 2.00ex;">Calculates the Transition / Transversion ratio
  as a function of alternative allele count. Only uses bi-allelic SNPs. The
  resulting output file has the suffix &quot;.TsTv.count&quot;.</div>
<div class="Pp"></div>
<b>--TsTv-by-qual</b>
<div style="margin-left: 2.00ex;">Calculates the Transition / Transversion ratio
  as a function of SNP quality threshold. Only uses bi-allelic SNPs. The
  resulting output file has the suffix &quot;.TsTv.qual&quot;.</div>
<div class="Pp"></div>
<b>--FILTER-summary</b>
<div style="margin-left: 2.00ex;">Generates a summary of the number of SNPs and
  Ts/Tv ratio for each FILTER category. The output file has the suffix
  &quot;.FILTER.summary&quot;.</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_NUCLEOTIDE_DIVERGENCE_STATISTICS"><a class="selflink" href="#OUTPUT_NUCLEOTIDE_DIVERGENCE_STATISTICS">OUTPUT
  NUCLEOTIDE DIVERGENCE STATISTICS</a></h2>
<div style="margin-left: 2.00ex;"><b>--site-pi</b>
<div style="margin-left: 2.00ex;">Measures nucleotide divergency on a per-site
  basis. The output file has the suffix &quot;.sites.pi&quot;.</div>
<div class="Pp"></div>
<b>--window-pi</b> <i>&lt;integer&gt;</i>
<div>&#x00A0;</div>
<b>--window-pi-step</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Measures the nucleotide diversity in windows,
  with the number provided as the window size. The output file has the suffix
  &quot;.windowed.pi&quot;. The latter is an optional argument used to specify
  the step size in between windows.</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_FST_STATISTICS"><a class="selflink" href="#OUTPUT_FST_STATISTICS">OUTPUT
  FST STATISTICS</a></h2>
<div style="margin-left: 2.00ex;"><b>--weir-fst-pop</b> <i>&lt;filename&gt;</i>
<div style="margin-left: 2.00ex;">This option is used to calculate an Fst
  estimate from Weir and Cockerham's 1984 paper. This is the preferred
  calculation of Fst. The provided file must contain a list of individuals (one
  individual per line) from the VCF file that correspond to one population. This
  option can be used multiple times to calculate Fst for more than two
  populations. These files will also be included as &quot;--keep&quot; options.
  By default, calculations are done on a per-site basis. The output file has the
  suffix &quot;.weir.fst&quot;.</div>
<div class="Pp"></div>
<b>--fst-window-size</b> <i>&lt;integer&gt;</i>
<div>&#x00A0;</div>
<b>--fst-window-step</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">These options can be used with
  &quot;--weir-fst-pop&quot; to do the Fst calculations on a windowed basis
  instead of a per-site basis. These arguments specify the desired window size
  and the desired step size between windows.</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_OTHER_STATISTICS"><a class="selflink" href="#OUTPUT_OTHER_STATISTICS">OUTPUT
  OTHER STATISTICS</a></h2>
<div style="margin-left: 2.00ex;"><b>--het</b>
<div style="margin-left: 2.00ex;">Calculates a measure of heterozygosity on a
  per-individual basis. Specfically, the inbreeding coefficient, F, is estimated
  for each individual using a method of moments. The resulting file has the
  suffix &quot;.het&quot;.</div>
<div class="Pp"></div>
<b>--hardy</b>
<div style="margin-left: 2.00ex;">Reports a p-value for each site from a
  Hardy-Weinberg Equilibrium test (as defined by Wigginton, Cutler and Abecasis
  (2005)). The resulting file (with suffix &quot;.hwe&quot;) also contains the
  Observed numbers of Homozygotes and Heterozygotes and the corresponding
  Expected numbers under HWE.</div>
<div class="Pp"></div>
<b>--TajimaD</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Outputs Tajima's D statistic in bins with size
  of the specified number. The output file has the suffix
  &quot;.Tajima.D&quot;.</div>
<div class="Pp"></div>
<b>--indv-freq-burden</b>
<div style="margin-left: 2.00ex;">This option calculates the number of variants
  within each individual of a specific frequency. The resulting file has the
  suffix &quot;.ifreqburden&quot;.</div>
<div class="Pp"></div>
<b>--LROH</b>
<div style="margin-left: 2.00ex;">This option will identify and output Long Runs
  of Homozygosity. The output file has the suffix &quot;.LROH&quot;. This
  function is experimental, and will use a lot of memory if applied to large
  datasets.</div>
<div class="Pp"></div>
<b>--relatedness</b>
<div style="margin-left: 2.00ex;">This option is used to calculate and output a
  relatedness statistic based on the method of Yang et al, Nature Genetics 2010
  (doi:10.1038/ng.608). Specifically, calculate the unadjusted Ajk statistic.
  Expectation of Ajk is zero for individuals within a populations, and one for
  an individual with themselves. The output file has the suffix
  &quot;.relatedness&quot;.</div>
<div class="Pp"></div>
<b>--relatedness2</b>
<div style="margin-left: 2.00ex;">This option is used to calculate and output a
  relatedness statistic based on the method of Manichaikul et al.,
  BIOINFORMATICS 2010 (doi:10.1093/bioinformatics/btq559). The output file has
  the suffix &quot;.relatedness2&quot;.</div>
<div class="Pp"></div>
<b>--site-quality</b>
<div style="margin-left: 2.00ex;">Generates a file containing the per-site SNP
  quality, as found in the QUAL column of the VCF file. This file has the suffix
  &quot;.lqual&quot;.</div>
<div class="Pp"></div>
<b>--missing-indv</b>
<div style="margin-left: 2.00ex;">Generates a file reporting the missingness on
  a per-individual basis. The file has the suffix &quot;.imiss&quot;.</div>
<div class="Pp"></div>
<b>--missing-site</b>
<div style="margin-left: 2.00ex;">Generates a file reporting the missingness on
  a per-site basis. The file has the suffix &quot;.lmiss&quot;.</div>
<div class="Pp"></div>
<b>--SNPdensity</b> <i>&lt;integer&gt;</i>
<div style="margin-left: 2.00ex;">Calculates the number and density of SNPs in
  bins of size defined by this option. The resulting output file has the suffix
  &quot;.snpden&quot;.</div>
<div class="Pp"></div>
<b>--kept-sites</b>
<div style="margin-left: 2.00ex;">Creates a file listing all sites that have
  been kept after filtering. The file has the suffix
  &quot;.kept.sites&quot;.</div>
<div class="Pp"></div>
<b>--removed-sites</b>
<div style="margin-left: 2.00ex;">Creates a file listing all sites that have
  been removed after filtering. The file has the suffix
  &quot;.removed.sites&quot;.</div>
<div class="Pp"></div>
<b>--singletons</b>
<div style="margin-left: 2.00ex;">This option will generate a file detailing the
  location of singletons, and the individual they occur in. The file reports
  both true singletons, and private doubletons (i.e. SNPs where the minor allele
  only occurs in a single individual and that individual is homozygotic for that
  allele). The output file has the suffix &quot;.singletons&quot;.</div>
<div class="Pp"></div>
<b>--hist-indel-len</b>
<div style="margin-left: 2.00ex;">This option will generate a histogram file of
  the length of all indels (including SNPs). It shows both the count and the
  percentage of all indels for indel lengths that occur at least once in the
  input file. SNPs are considered indels with length zero. The output file has
  the suffix &quot;.indel.hist&quot;.</div>
<div class="Pp"></div>
<b>--hapcount</b> <i>&lt;BED file&gt;</i>
<div style="margin-left: 2.00ex;">This option will output the number of unique
  haplotypes within user specified bins, as defined by the BED file. The output
  file has the suffix &quot;.hapcount&quot;.</div>
<div class="Pp"></div>
<b>--mendel</b> <i>&lt;PED file&gt;</i>
<div style="margin-left: 2.00ex;">This option is use to report mendel errors
  identified in trios. The command requires a PLINK-style PED file, with the
  first four columns specifying a family ID, the child ID, the father ID, and
  the mother ID. The output of this command has the suffix
  &quot;.mendel&quot;.</div>
<div class="Pp"></div>
<b>--extract-FORMAT-info</b> <i>&lt;string&gt;</i>
<div style="margin-left: 2.00ex;">Extract information from the genotype fields
  in the VCF file relating to a specfied FORMAT identifier. The resulting output
  file has the suffix &quot;.&lt;FORMAT_ID&gt;.FORMAT&quot;. For example, the
  following command would extract the all of the GT (i.e. Genotype) entries:
<div>&#x00A0;</div>
<div style="margin-left: 2.00ex;"><b>vcftools</b> --vcf file1.vcf
  --extract-FORMAT-info GT</div>
</div>
<div class="Pp"></div>
<b>--get-INFO</b> <i>&lt;string&gt;</i>
<div style="margin-left: 2.00ex;">This option is used to extract information
  from the INFO field in the VCF file. The &lt;string&gt; argument specifies the
  INFO tag to be extracted, and the option can be used multiple times in order
  to extract multiple INFO entries. The resulting file, with suffix
  &quot;.INFO&quot;, contains the required INFO information in a tab-separated
  table. For example, to extract the NS and DB flags, one would use the command:
<div>&#x00A0;</div>
<div style="margin-left: 2.00ex;"><b>vcftools</b> --vcf file1.vcf --get-INFO NS
  --get-INFO DB</div>
</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_VCF_FORMAT"><a class="selflink" href="#OUTPUT_VCF_FORMAT">OUTPUT
  VCF FORMAT</a></h2>
<div style="margin-left: 2.00ex;"><b>--recode</b>
<div>&#x00A0;</div>
<b>--recode-bcf</b>
<div style="margin-left: 2.00ex;">These options are used to generate a new file
  in either VCF or BCF from the input VCF or BCF file after applying the
  filtering options specified by the user. The output file has the suffix
  &quot;.recode.vcf&quot; or &quot;.recode.bcf&quot;. By default, the INFO
  fields are removed from the output file, as the INFO values may be invalidated
  by the recoding (e.g. the total depth may need to be recalculated if
  individuals are removed). This behavior may be overriden by the following
  options. By default, BCF files are written out as BGZF compressed files.</div>
<div class="Pp"></div>
<b>--recode-INFO</b> <i>&lt;string&gt;</i>
<div>&#x00A0;</div>
<b>--recode-INFO-all</b>
<div style="margin-left: 2.00ex;">These options can be used with the above
  recode options to define an INFO key name to keep in the output file. This
  option can be used multiple times to keep more of the INFO fields. The second
  option is used to keep all INFO values in the original file.</div>
<div class="Pp"></div>
<b>--contigs</b> <i>&lt;string&gt;</i>
<div style="margin-left: 2.00ex;">This option can be used in conjuction with the
  --recode-bcf when the input file does not have any contig declarations. This
  option expects a file name with one contig header per line. These lines are
  included in the output file.</div>
</div>
<h2 class="Ss" title="Ss" id="OUTPUT_OTHER_FORMATS"><a class="selflink" href="#OUTPUT_OTHER_FORMATS">OUTPUT
  OTHER FORMATS</a></h2>
<div style="margin-left: 2.00ex;"><b>--012</b>
<div style="margin-left: 2.00ex;">This option outputs the genotypes as a large
  matrix. Three files are produced. The first, with suffix &quot;.012&quot;,
  contains the genotypes of each individual on a separate line. Genotypes are
  represented as 0, 1 and 2, where the number represent that number of
  non-reference alleles. Missing genotypes are represented by -1. The second
  file, with suffix &quot;.012.indv&quot; details the individuals included in
  the main file. The third file, with suffix &quot;.012.pos&quot; details the
  site locations included in the main file.</div>
<div class="Pp"></div>
<b>--IMPUTE</b>
<div style="margin-left: 2.00ex;">This option outputs phased haplotypes in
  IMPUTE reference-panel format. As IMPUTE requires phased data, using this
  option also implies --phased. Unphased individuals and genotypes are therefore
  excluded. Only bi-allelic sites are included in the output. Using this option
  generates three files. The IMPUTE haplotype file has the suffix
  &quot;.impute.hap&quot;, and the IMPUTE legend file has the suffix
  &quot;.impute.hap.legend&quot;. The third file, with suffix
  &quot;.impute.hap.indv&quot;, details the individuals included in the
  haplotype file, although this file is not needed by IMPUTE.</div>
<div class="Pp"></div>
<b>--ldhat</b>
<div>&#x00A0;</div>
<b>--ldhelmet</b>
<div>&#x00A0;</div>
<b>--ldhat-geno</b>
<div style="margin-left: 2.00ex;">These options output data in LDhat/LDhelmet
  format. This option requires the &quot;--chr&quot; filter option to also be
  used. The two first options output phased data only, and therefore also
  implies &quot;--phased&quot; be used, leading to unphased individuals and
  genotypes being excluded. For LDhelmet, only snps will be considered, and
  therefore it implies &quot;--remove-indels&quot;. The second option treats all
  of the data as unphased, and therefore outputs LDhat files in
  genotype/unphased format. Two output files are generated with the suffixes
  &quot;.ldhat.sites&quot; and &quot;.ldhat.locs&quot;, which correspond to the
  LDhat &quot;sites&quot; and &quot;locs&quot; input files respectively; for
  LDhelmet, the two files generated have the suffixes &quot;.ldhelmet.snps&quot;
  and &quot;.ldhelmet.pos&quot;, which corresponds to the &quot;SNPs&quot; and
  &quot;positions&quot; files.</div>
<div class="Pp"></div>
<b>--BEAGLE-GL</b>
<div>&#x00A0;</div>
<b>--BEAGLE-PL</b>
<div style="margin-left: 2.00ex;">These options output genotype likelihood
  information for input into the BEAGLE program. The VCF file is required to
  contain FORMAT fields with &quot;GL&quot; or &quot;PL&quot; tags, which can
  generally be output by SNP callers such as the GATK. Use of this option
  requires a chromosome to be specified via the &quot;--chr&quot; option. The
  resulting output file has the suffix &quot;.BEAGLE.GL&quot; or
  &quot;.BEAGLE.PL&quot; and contains genotype likelihoods for biallelic sites.
  This file is suitable for input into BEAGLE via the &quot;like=&quot;
  argument.</div>
<div class="Pp"></div>
<b>--plink</b>
<div>&#x00A0;</div>
<b>--plink-tped</b>
<div>&#x00A0;</div>
<b>--chrom-map</b>
<div style="margin-left: 2.00ex;">These options output the genotype data in
  PLINK PED format. With the first option, two files are generated, with
  suffixes &quot;.ped&quot; and &quot;.map&quot;. Note that only bi-allelic loci
  will be output. Further details of these files can be found in the PLINK
  documentation.
<div>&#x00A0;</div>
Note: The first option can be very slow on large datasets. Using the --chr
  option to divide up the dataset is advised, or alternatively use the
  --plink-tped option which outputs the files in the PLINK transposed format
  with suffixes &quot;.tped&quot; and &quot;.tfam&quot;.
<div>&#x00A0;</div>
For usage with variant sites in species other than humans, the --chrom-map
  option may be used to specify a file name that has a tab-delimited mapping of
  chromosome name to a desired integer value with one line per chromosome. This
  file must contain a mapping for every chromosome value found in the
  file.</div>
</div>
<h1 class="Sh" title="Sh" id="COMPARISON_OPTIONS"><a class="selflink" href="#COMPARISON_OPTIONS">COMPARISON
  OPTIONS</a></h1>
These options are used to compare the original variant file to another variant
  file and output the results. All of the diff functions require both files to
  contain the same chromosomes and that the files be sorted in the same order.
  If one of the files contains chromosomes that the other file does not, use the
  --not-chr filter to remove them from the analysis.
<h2 class="Ss" title="Ss" id="DIFF_VCF_FILE"><a class="selflink" href="#DIFF_VCF_FILE">DIFF
  VCF FILE</a></h2>
<div style="margin-left: 2.00ex;"><b>--diff</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--gzdiff</b> <i>&lt;filename&gt;</i>
<div>&#x00A0;</div>
<b>--diff-bcf</b> <i>&lt;filename&gt;</i>
<div style="margin-left: 2.00ex;">These options compare the original input file
  to this specified VCF, gzipped VCF, or BCF file. These options must be
  specified with one additional option described below in order to specify what
  type of comparison is to be performed. See the examples section for typical
  usage.</div>
</div>
<h2 class="Ss" title="Ss" id="DIFF_OPTIONS"><a class="selflink" href="#DIFF_OPTIONS">DIFF
  OPTIONS</a></h2>
<div style="margin-left: 2.00ex;"><b>--diff-site</b>
<div style="margin-left: 2.00ex;">Outputs the sites that are common / unique to
  each file. The output file has the suffix
  &quot;.diff.sites_in_files&quot;.</div>
<div class="Pp"></div>
<b>--diff-indv</b>
<div style="margin-left: 2.00ex;">Outputs the individuals that are common /
  unique to each file. The output file has the suffix
  &quot;.diff.indv_in_files&quot;.</div>
<div class="Pp"></div>
<b>--diff-site-discordance</b>
<div style="margin-left: 2.00ex;">This option calculates discordance on a site
  by site basis. The resulting output file has the suffix
  &quot;.diff.sites&quot;.</div>
<div class="Pp"></div>
<b>--diff-indv-discordance</b>
<div style="margin-left: 2.00ex;">This option calculates discordance on a
  per-individual basis. The resulting output file has the suffix
  &quot;.diff.indv&quot;.</div>
<div class="Pp"></div>
<b>--diff-indv-map</b> <i>&lt;filename&gt;</i>
<div style="margin-left: 2.00ex;">This option allows the user to specify a
  mapping of individual IDs in the second file to those in the first file. The
  program expects the file to contain a tab-delimited line containing an
  individual's name in file one followed by that same individual's name in file
  two with one mapping per line.</div>
<div class="Pp"></div>
<b>--diff-discordance-matrix</b>
<div style="margin-left: 2.00ex;">This option calculates a discordance matrix.
  This option only works with bi-allelic loci with matching alleles that are
  present in both files. The resulting output file has the suffix
  &quot;.diff.discordance.matrix&quot;.</div>
<div class="Pp"></div>
<b>--diff-switch-error</b>
<div style="margin-left: 2.00ex;">This option calculates phasing errors
  (specifically &quot;switch errors&quot;). This option creates an output file
  describing switch errors found between sites, with suffix
  &quot;.diff.switch&quot;.</div>
</div>
<h1 class="Sh" title="Sh" id="AUTHORS"><a class="selflink" href="#AUTHORS">AUTHORS</a></h1>
Adam Auton (adam.auton@einstein.yu.edu)
<div>&#x00A0;</div>
Anthony Marcketta (anthony.marcketta@einstein.yu.edu)</div>
<table class="foot">
  <tr>
    <td class="foot-date">page</td>
    <td class="foot-os">1</td>
  </tr>
</table>
</body>
</html>
