<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8"/>
  <style>
    table.head, table.foot { width: 100%; }
    td.head-rtitle, td.foot-os { text-align: right; }
    td.head-vol { text-align: center; }
    div.Pp { margin: 1ex 0ex; }
  </style>
  <title>WIGGLE2GFF3(1p)</title>
</head>
<body>
<table class="head">
  <tr>
    <td class="head-ltitle">WIGGLE2GFF3(1p)</td>
    <td class="head-vol">User Contributed Perl Documentation</td>
    <td class="head-rtitle">WIGGLE2GFF3(1p)</td>
  </tr>
</table>
<div class="manual-text">
<h1 class="Sh" title="Sh" id="NAME"><a class="selflink" href="#NAME">NAME</a></h1>
wiggle2gff3.pl - Converts UCSC WIG format files into gff3 files
<h1 class="Sh" title="Sh" id="SYNOPSIS"><a class="selflink" href="#SYNOPSIS">SYNOPSIS</a></h1>
<pre>
  wiggle2gff3.pl [options] WIG_FILE &gt; load_data.gff3
</pre>
<div class="Pp"></div>
Converts UCSC WIG format files into gff3 files suitable for loading into GBrowse
  databases. This is used for high-density quantitative data such as CNV, SNP
  and expression arrays.
<h1 class="Sh" title="Sh" id="DESCRIPTION"><a class="selflink" href="#DESCRIPTION">DESCRIPTION</a></h1>
Use this converter when you have dense quantitative data to display using the
  xyplot, density, or heatmap glyphs, and too many data items (thousands) to
  load into GBrowse. It creates one or more space- efficient binary files
  containing the quantitative data, as well as a small GFF3 file that can be
  loaded into Chado or other GBrowse databases.
<div class="Pp"></div>
Typical usage is as follows:
<div class="Pp"></div>
<pre>
  % wiggle2gff3.pl --method=microarray_oligo my_data.wig &gt; my_data.gff3
</pre>
<h2 class="Ss" title="Ss" id="Options"><a class="selflink" href="#Options">Options</a></h2>
The following options are accepted:
<div class="Pp"></div>
<pre>
 --method=&lt;method name&gt;   Set the method for the GFF3 lines representing 
                           each quantitative data point in the track.
                           The default is &quot;microarray_oligo.&quot;
 --source=&lt;source&gt;        Set the source field for the GFF3 file. The default is
                           none.
 --gff3                   Create a GFF3-format file (the default)
 --featurefile            Create a &quot;featurefile&quot; format file -- this is the
                           simplified format used for GBrowse uploads. This
                           option is incompatible with the --gff3 option.
 --sample                 If true, then very large files (&gt;5 MB) will be sampled
                           to obtain minimum, maximum and standard deviation; otherwise
                           the entire file will be scanned to obtain these statistics. 
                           This will process the files faster but may miss outlier
                           values.
 --path=&lt;path&gt;            Specify the directory in which to place the binary wiggle
                           files. The default is the current temporary directory
                           (/tmp or whatever is appropriate for your operating system).
 --base=&lt;path&gt;            Same as &quot;--path&quot;.
 --trackname              specify the trackname base for the wigfile creation
 --help                   This documentation.
</pre>
<div class="Pp"></div>
This script will accept a variety of option styles, including abbreviated
  options (&quot;--meth=foo&quot;), single character options (&quot;-m
  foo&quot;), and other common variants.
<h2 class="Ss" title="Ss" id="Binary_wiggle_files"><a class="selflink" href="#Binary_wiggle_files">Binary
  wiggle files</a></h2>
The binary &quot;wiggle&quot; files created by this utility are readable using
  the Bio::Graphics::Wiggle module. The quantitative data is scaled to the range
  of 1-255 (losing lots of precision, but still more than enough for data
  visualization), and stored in a packed format in which each file corresponds
  to the length of a single chromosome or contig.
<div class="Pp"></div>
Once created, the binary files should not be moved or renamed, unless you are
  careful to make corresponding changes to the pathnames given by the
  &quot;wigfile&quot; attribute in the GFF3 file feature lines. You should also
  be careful about using the cp command to copy the binary files; they are
  formatted with &quot;holes&quot; in such a way that missing data does not take
  up any space on disk. If you cp them, the holes will fill up with zeroes and
  the space savings will be lost. Better to use the &quot;tar&quot; command with
  its --sparse option to move the files from one place to another.
<h2 class="Ss" title="Ss" id="Example_WIG_File"><a class="selflink" href="#Example_WIG_File">Example
  WIG File</a></h2>
This example is from &lt;http://genome.ucsc.edu/goldenPath/help/wiggle.html&gt;:
<div class="Pp"></div>
<pre>
 # filename: example.wig
 #
 #       300 base wide bar graph, autoScale is on by default == graphing
 #       limits will dynamically change to always show full range of data
 #       in viewing window, priority = 20 positions this as the second graph
 #       Note, zero-relative, half-open coordinate system in use for bed format
 track type=wiggle_0 name=&quot;Bed Format&quot; description=&quot;BED format&quot; \
     visibility=full color=200,100,0 altColor=0,100,200 priority=20
 chr19 59302000 59302300 -1.0
 chr19 59302300 59302600 -0.75
 chr19 59302600 59302900 -0.50
 chr19 59302900 59303200 -0.25
 chr19 59303200 59303500 0.0
 chr19 59303500 59303800 0.25
 chr19 59303800 59304100 0.50
 chr19 59304100 59304400 0.75
 chr19 59304400 59304700 1.00
 #       150 base wide bar graph at arbitrarily spaced positions,
 #       threshold line drawn at y=11.76
 #       autoScale off viewing range set to [0:25]
 #       priority = 10 positions this as the first graph
 #       Note, one-relative coordinate system in use for this format
 track type=wiggle_0 name=&quot;variableStep&quot; description=&quot;variableStep format&quot; \
     visibility=full autoScale=off viewLimits=0.0:25.0 color=255,200,0 \
     yLineMark=11.76 yLineOnOff=on priority=10
 variableStep chrom=chr19 span=150
 59304701 10.0
 59304901 12.5
 59305401 15.0
 59305601 17.5
 59305901 20.0
 59306081 17.5
 59306301 15.0
 59306691 12.5
 59307871 10.0
 #       200 base wide points graph at every 300 bases, 50 pixel high graph
 #       autoScale off and viewing range set to [0:1000]
 #       priority = 30 positions this as the third graph
 #       Note, one-relative coordinate system in use for this format
 track type=wiggle_0 name=&quot;fixedStep&quot; description=&quot;fixed step&quot; visibility=full \
     autoScale=off viewLimits=0:1000 color=0,200,100 maxHeightPixels=100:50:20 \
     graphType=points priority=30
 fixedStep chrom=chr19 start=59307401 step=300 span=200
 1000
  900
  800
  700
  600
  500
  400
  300
  200
  100
</pre>
<div class="Pp"></div>
You can convert this into a loadable GFF3 file with the following command:
<div class="Pp"></div>
<pre>
 wiggle2gff3.pl --meth=example --so=example --path=/var/gbrowse/db example.wig \
              &gt; example.gff3
</pre>
<div class="Pp"></div>
The output will look like this:
<div class="Pp"></div>
<pre>
 ##gff-version 3
 chr19  example example 59302001        59304700        .       .       .       Name=Bed Format;wigfile=/var/gbrowse/db/track001.chr19.1199828298.wig
 chr19  example example 59304701        59308020        .       .       .       Name=variableStep;wigfile=/var/gbrowse/db/track002.chr19.1199828298.wig
 chr19  example example 59307401        59310400        .       .       .       Name=fixedStep;wigfile=/var/gbrowse/db/track003.chr19.1199828298.wig
</pre>
<h1 class="Sh" title="Sh" id="PROBLEMS"><a class="selflink" href="#PROBLEMS">PROBLEMS</a></h1>
This script has trouble with wig files from very fragmented genomes (&gt;100K
  scaffolds). In this case, you may wish to run split_wig.pl, which splits the
  original wig file into a series of smaller files with a maximum of 900
  scaffolds each. It then runs wiggle2gff3.pl for each subfile and stores the
  results in separate folders.
<h1 class="Sh" title="Sh" id="SEE_ALSO"><a class="selflink" href="#SEE_ALSO">SEE
  ALSO</a></h1>
Bio::DB::GFF, bp_bulk_load_gff.pl, bp_fast_load_gff.pl, bp_load_gff.pl,
  bp_seqfeature_load.pl
<h1 class="Sh" title="Sh" id="AUTHOR"><a class="selflink" href="#AUTHOR">AUTHOR</a></h1>
Lincoln Stein &lt;lstein@cshl.org&gt;.
<div class="Pp"></div>
Copyright (c) 2008 Cold Spring Harbor Laboratory
<div class="Pp"></div>
This package is free software; you can redistribute it and/or modify it under
  the terms of the GPL (either version 1, or at your option, any later version)
  or the Artistic License 2.0. Refer to LICENSE for the full license text. See
  DISCLAIMER.txt for disclaimers of warranty.</div>
<table class="foot">
  <tr>
    <td class="foot-date">2017-04-12</td>
    <td class="foot-os">perl v5.24.1</td>
  </tr>
</table>
</body>
</html>
