<!-- Creator     : groff version 1.22.3 -->
<!-- CreationDate: Sun Aug 27 16:42:54 2017 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title></title>
</head>
<body>

<hr>


<p>WIGGLE2GFF3(1p) User Contributed Perl Documentation
WIGGLE2GFF3(1p)</p>

<p style="margin-top: 1em">NAME <br>
wiggle2gff3.pl - Converts UCSC WIG format files into gff3
files</p>

<p style="margin-top: 1em">SYNOPSIS <br>
wiggle2gff3.pl [options] WIG_FILE &gt; load_data.gff3</p>

<p style="margin-top: 1em">Converts UCSC WIG format files
into gff3 files suitable for loading into GBrowse databases.
This is used for high-density quantitative data such as CNV,
SNP and expression <br>
arrays.</p>

<p style="margin-top: 1em">DESCRIPTION <br>
Use this converter when you have dense quantitative data to
display using the xyplot, density, or heatmap glyphs, and
too many data items (thousands) to load into GBrowse. It
<br>
creates one or more space- efficient binary files containing
the quantitative data, as well as a small GFF3 file that can
be loaded into Chado or other GBrowse databases.</p>

<p style="margin-top: 1em">Typical usage is as follows:</p>

<p style="margin-top: 1em">% wiggle2gff3.pl
--method=microarray_oligo my_data.wig &gt; my_data.gff3</p>

<p style="margin-top: 1em">Options <br>
The following options are accepted:</p>

<p style="margin-top: 1em">--method=&lt;method name&gt; Set
the method for the GFF3 lines representing <br>
each quantitative data point in the track. <br>
The default is &quot;microarray_oligo.&quot;</p>

<p style="margin-top: 1em">--source=&lt;source&gt; Set the
source field for the GFF3 file. The default is <br>
none.</p>

<p style="margin-top: 1em">--gff3 Create a GFF3-format file
(the default)</p>

<p style="margin-top: 1em">--featurefile Create a
&quot;featurefile&quot; format file -- this is the <br>
simplified format used for GBrowse uploads. This <br>
option is incompatible with the --gff3 option.</p>

<p style="margin-top: 1em">--sample If true, then very
large files (&gt;5 MB) will be sampled <br>
to obtain minimum, maximum and standard deviation; otherwise
<br>
the entire file will be scanned to obtain these statistics.
<br>
This will process the files faster but may miss outlier <br>
values.</p>

<p style="margin-top: 1em">--path=&lt;path&gt; Specify the
directory in which to place the binary wiggle <br>
files. The default is the current temporary directory <br>
(/tmp or whatever is appropriate for your operating
system).</p>

<p style="margin-top: 1em">--base=&lt;path&gt; Same as
&quot;--path&quot;.</p>

<p style="margin-top: 1em">--trackname specify the
trackname base for the wigfile creation</p>

<p style="margin-top: 1em">--help This documentation.</p>

<p style="margin-top: 1em">This script will accept a
variety of option styles, including abbreviated options
(&quot;--meth=foo&quot;), single character options (&quot;-m
foo&quot;), and other common variants.</p>

<p style="margin-top: 1em">Binary wiggle files <br>
The binary &quot;wiggle&quot; files created by this utility
are readable using the Bio::Graphics::Wiggle module. The
quantitative data is scaled to the range of 1-255 (losing
lots of <br>
precision, but still more than enough for data
visualization), and stored in a packed format in which each
file corresponds to the length of a single chromosome or
contig.</p>

<p style="margin-top: 1em">Once created, the binary files
should not be moved or renamed, unless you are careful to
make corresponding changes to the pathnames given by the
&quot;wigfile&quot; attribute in the GFF3 <br>
file feature lines. You should also be careful about using
the cp command to copy the binary files; they are formatted
with &quot;holes&quot; in such a way that missing data does
not take <br>
up any space on disk. If you cp them, the holes will fill up
with zeroes and the space savings will be lost. Better to
use the &quot;tar&quot; command with its --sparse option to
move the <br>
files from one place to another.</p>

<p style="margin-top: 1em">Example WIG File <br>
This example is from
&lt;http://genome.ucsc.edu/goldenPath/help/wiggle.html&gt;:</p>

<p style="margin-top: 1em"># filename: example.wig <br>
# <br>
# 300 base wide bar graph, autoScale is on by default ==
graphing <br>
# limits will dynamically change to always show full range
of data <br>
# in viewing window, priority = 20 positions this as the
second graph <br>
# Note, zero-relative, half-open coordinate system in use
for bed format <br>
track type=wiggle_0 name=&quot;Bed Format&quot;
description=&quot;BED format&quot; visibility=full
color=200,100,0 altColor=0,100,200 priority=20 <br>
chr19 59302000 59302300 -1.0 <br>
chr19 59302300 59302600 -0.75 <br>
chr19 59302600 59302900 -0.50 <br>
chr19 59302900 59303200 -0.25 <br>
chr19 59303200 59303500 0.0 <br>
chr19 59303500 59303800 0.25 <br>
chr19 59303800 59304100 0.50 <br>
chr19 59304100 59304400 0.75 <br>
chr19 59304400 59304700 1.00 <br>
# 150 base wide bar graph at arbitrarily spaced positions,
<br>
# threshold line drawn at y=11.76 <br>
# autoScale off viewing range set to [0:25] <br>
# priority = 10 positions this as the first graph <br>
# Note, one-relative coordinate system in use for this
format <br>
track type=wiggle_0 name=&quot;variableStep&quot;
description=&quot;variableStep format&quot; visibility=full
autoScale=off viewLimits=0.0:25.0 color=255,200,0
yLineMark=11.76 yLineOnOff=on priority=10 <br>
variableStep chrom=chr19 span=150 <br>
59304701 10.0 <br>
59304901 12.5 <br>
59305401 15.0 <br>
59305601 17.5 <br>
59305901 20.0 <br>
59306081 17.5 <br>
59306301 15.0 <br>
59306691 12.5 <br>
59307871 10.0 <br>
# 200 base wide points graph at every 300 bases, 50 pixel
high graph <br>
# autoScale off and viewing range set to [0:1000] <br>
# priority = 30 positions this as the third graph <br>
# Note, one-relative coordinate system in use for this
format <br>
track type=wiggle_0 name=&quot;fixedStep&quot;
description=&quot;fixed step&quot; visibility=full
autoScale=off viewLimits=0:1000 color=0,200,100
maxHeightPixels=100:50:20 graphType=points priority=30 <br>
fixedStep chrom=chr19 start=59307401 step=300 span=200 <br>
1000 <br>
900 <br>
800 <br>
700 <br>
600 <br>
500 <br>
400 <br>
300 <br>
200 <br>
100</p>

<p style="margin-top: 1em">You can convert this into a
loadable GFF3 file with the following command:</p>

<p style="margin-top: 1em">wiggle2gff3.pl --meth=example
--so=example --path=/var/gbrowse/db example.wig &gt;
example.gff3</p>

<p style="margin-top: 1em">The output will look like
this:</p>

<p style="margin-top: 1em">##gff-version 3</p>

<p style="margin-top: 1em">chr19 example example 59302001
59304700 . . . Name=Bed
Format;wigfile=/var/gbrowse/db/track001.chr19.1199828298.wig
<br>
chr19 example example 59304701 59308020 . . .
Name=variableStep;wigfile=/var/gbrowse/db/track002.chr19.1199828298.wig
<br>
chr19 example example 59307401 59310400 . . .
Name=fixedStep;wigfile=/var/gbrowse/db/track003.chr19.1199828298.wig</p>

<p style="margin-top: 1em">PROBLEMS <br>
This script has trouble with wig files from very fragmented
genomes (&gt;100K scaffolds). In this case, you may wish to
run split_wig.pl, which splits the original wig file into a
<br>
series of smaller files with a maximum of 900 scaffolds
each. It then runs wiggle2gff3.pl for each subfile and
stores the results in separate folders.</p>

<p style="margin-top: 1em">SEE ALSO <br>
Bio::DB::GFF, bp_bulk_load_gff.pl, bp_fast_load_gff.pl,
bp_load_gff.pl, bp_seqfeature_load.pl</p>

<p style="margin-top: 1em">AUTHOR <br>
Lincoln Stein &lt;lstein@cshl.org&gt;.</p>

<p style="margin-top: 1em">Copyright (c) 2008 Cold Spring
Harbor Laboratory</p>

<p style="margin-top: 1em">This package is free software;
you can redistribute it and/or modify it under the terms of
the GPL (either version 1, or at your option, any later
version) or the Artistic <br>
License 2.0. Refer to LICENSE for the full license text. See
DISCLAIMER.txt for disclaimers of warranty.</p>

<p style="margin-top: 1em">perl v5.24.1 2017-04-12
WIGGLE2GFF3(1p)</p>
<hr>
</body>
</html>
